{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c3b57a69",
   "metadata": {},
   "outputs": [],
   "source": [
    "import langchain_community\n",
    "import langchain_text_splitters\n",
    "from langchain_community.document_loaders import PyPDFLoader, pdf\n",
    "from langchain_community.embeddings.sentence_transformer import (\n",
    "    SentenceTransformerEmbeddings,\n",
    ")\n",
    "from langchain_community.vectorstores import Chroma\n",
    "from langchain_text_splitters import CharacterTextSplitter, RecursiveCharacterTextSplitter\n",
    "import os\n",
    "import pprint\n",
    "import re\n",
    "from langchain_core.documents import Document\n",
    "from langchain_core.prompts.prompt import PromptTemplate\n",
    "from langchain_core.output_parsers import JsonOutputParser\n",
    "from langchain_ollama import ChatOllama\n",
    "from pydantic import BaseModel, Field\n",
    "import json"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "328c2061",
   "metadata": {},
   "source": [
    "## Google"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e03c65a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "from langchain_google_genai import GoogleGenerativeAI\n",
    "from langchain_core.messages import HumanMessage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "357e9e5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('/data/local/aparraga/Bioinformatician/RAG/api_google.txt') as f:\n",
    "    \n",
    "    api_key = json.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "23cd2e8b",
   "metadata": {},
   "outputs": [],
   "source": [
    "llm = GoogleGenerativeAI(model=\"gemini-2.0-flash\",api_key=api_key['key'],temperature=0.2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "4bc69aff",
   "metadata": {},
   "outputs": [],
   "source": [
    "# load the document and split it into pages\n",
    "loader = PyPDFLoader(\"Publications/Parraga-Leo2023.pdf\", mode=\"single\")\n",
    "doc = loader.load()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "18353b37",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "67457\n"
     ]
    }
   ],
   "source": [
    "print(len(doc[0].page_content))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "cd521499",
   "metadata": {},
   "outputs": [],
   "source": [
    "class paper(BaseModel):\n",
    "\n",
    "    PaperTitle: str = Field(description=\"The full title of the research paper\")\n",
    "    Publication: str = Field(description=\"Year: The year the paper was published\")\n",
    "    Authors: str = Field(description=\"The full names of all authors of the paper\")\n",
    "    Email: str = Field(description=\"The email address of the author (if provided)\")\n",
    "    Abstract: str = Field(description=\"The full text of the paper's abstract.\")\n",
    "    Introduction: str = Field(description=\"The full text fo the paper's introduction. Bear in mind that it can have other names such as background.\")\n",
    "    Methods: str = Field(description=\"The full text fo the paper's methods. Don't take the information from abstract.\")\n",
    "    Results: str = Field(description=\"The full text fo the paper's results. Don't take the information from abstract. Please take all possible text of results, this section could be divided into different sections.\")\n",
    "    Discussion: str = Field(description=\"The full text fo the paper's discussion if provided. Otherwise leave the filed blank. It should be entitled 'Discussion'\")\n",
    "    Conclusion: str = Field(description=\"The full text fo the paper's conclusion if provided. Otherwise leave the filed blank. Please don't take this information form abstract.\")\n",
    "    URL: str = Field(description=\"the link where you can find the article. This link is also known as DOI.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "6f763428",
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = JsonOutputParser(pydantic_object=paper)\n",
    "\n",
    "prompt = PromptTemplate(\n",
    "    template=\"\"\"\n",
    "    You are an expert in analyzing scientific research papers. Please carefully read the provided research paper above and extract the following key information:\n",
    "Extract these nine (9) properties from the research paper:\n",
    "\n",
    "Paper Title: The full title of the research paper\n",
    "Publication Year: The year the paper was published\n",
    "\n",
    "Authors: The full names of all authors of the paper\n",
    "\n",
    "Email: The email address of the author (if provided)\n",
    "\n",
    "Abstract: The full text of the paper's abstract\n",
    "\n",
    "Introduction: The full text fo the paper's introduction. Bear in mind that it can have other names such as background.\n",
    "\n",
    "Methods: The full text fo the paper's methods. Don't take the information from abstract.\n",
    "\n",
    "Results: The full text fo the paper's results. Don't take the information from abstract. Please take all possible text of results, this section could be divided into different sections.\n",
    "\n",
    "Discussion:The full text fo the paper's discussion if provided. Otherwise leave the filed blank. It shoudl be entitled \"Discussion\"\n",
    "\n",
    "Conclusion: The full text fo the paper's conclusion if provided. Otherwise leave the filed blank. Please don't take this information form abstract.\n",
    "\n",
    "URL: the link where you can find the article. This link is also known as DOI.\n",
    "\n",
    "\n",
    "Guidelines:\n",
    "\n",
    "\n",
    "The extracted information should be factual and accurate to the document. Be extremely concise, except for the Abstract, Introduction, Methods, Results. Discussion and Conclusion which should be copied in full.\n",
    "The extracted entities should be self-contained and easily understood without the rest of the paper. If a property is missing from the paper, please leave the field empty rather than guessing.\n",
    "Answer in JSON format. The JSON should contain 9 keys: \"PaperTitle\", \"PublicationYear\", \"Authors\", \"Email\", \"Abstract\", \"Introduction\", \"Methods\",\"Results\", \"URL\".\n",
    "\n",
    "\n",
    "Format instructions: \\n{format_instructions}\\n\n",
    "\n",
    "    \n",
    "The article is this:\\n{query}\\n\"\"\",\n",
    "\n",
    "    input_variables=[\"query\"],\n",
    "\n",
    "    partial_variables={\"format_instructions\": parser.get_format_instructions()}\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "aabf31d5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PromptTemplate(input_variables=['query'], input_types={}, partial_variables={'format_instructions': 'The output should be formatted as a JSON instance that conforms to the JSON schema below.\\n\\nAs an example, for the schema {\"properties\": {\"foo\": {\"title\": \"Foo\", \"description\": \"a list of strings\", \"type\": \"array\", \"items\": {\"type\": \"string\"}}}, \"required\": [\"foo\"]}\\nthe object {\"foo\": [\"bar\", \"baz\"]} is a well-formatted instance of the schema. The object {\"properties\": {\"foo\": [\"bar\", \"baz\"]}} is not well-formatted.\\n\\nHere is the output schema:\\n```\\n{\"properties\": {\"PaperTitle\": {\"description\": \"The full title of the research paper\", \"title\": \"Papertitle\", \"type\": \"string\"}, \"Publication\": {\"description\": \"Year: The year the paper was published\", \"title\": \"Publication\", \"type\": \"string\"}, \"Authors\": {\"description\": \"The full names of all authors of the paper\", \"title\": \"Authors\", \"type\": \"string\"}, \"Email\": {\"description\": \"The email address of the author (if provided)\", \"title\": \"Email\", \"type\": \"string\"}, \"Abstract\": {\"description\": \"The full text of the paper\\'s abstract.\", \"title\": \"Abstract\", \"type\": \"string\"}, \"Introduction\": {\"description\": \"The full text fo the paper\\'s introduction. Bear in mind that it can have other names such as background.\", \"title\": \"Introduction\", \"type\": \"string\"}, \"Methods\": {\"description\": \"The full text fo the paper\\'s methods. Don\\'t take the information from abstract.\", \"title\": \"Methods\", \"type\": \"string\"}, \"Results\": {\"description\": \"The full text fo the paper\\'s results. Don\\'t take the information from abstract. Please take all possible text of results, this section could be divided into different sections.\", \"title\": \"Results\", \"type\": \"string\"}, \"Discussion\": {\"description\": \"The full text fo the paper\\'s discussion if provided. Otherwise leave the filed blank. It shoudl be entitled \\'Discussion\\'\", \"title\": \"Discussion\", \"type\": \"string\"}, \"Conclusion\": {\"description\": \"The full text fo the paper\\'s conclusion if provided. Otherwise leave the filed blank. Please don\\'t take this information form abstract.\", \"title\": \"Conclusion\", \"type\": \"string\"}, \"URL\": {\"description\": \"the link where you can find the article. This link is also known as DOI. \", \"title\": \"Url\", \"type\": \"string\"}}, \"required\": [\"PaperTitle\", \"Publication\", \"Authors\", \"Email\", \"Abstract\", \"Introduction\", \"Methods\", \"Results\", \"Discussion\", \"Conclusion\", \"URL\"]}\\n```'}, template='\\n    You are an expert in analyzing scientific research papers. Please carefully read the provided research paper above and extract the following key information:\\nExtract these nine (9) properties from the research paper:\\n\\nPaper Title: The full title of the research paper\\nPublication Year: The year the paper was published\\n\\nAuthors: The full names of all authors of the paper\\n\\nEmail: The email address of the author (if provided)\\n\\nAbstract: The full text of the paper\\'s abstract\\n\\nIntroduction: The full text fo the paper\\'s introduction. Bear in mind that it can have other names such as background.\\n\\nMethods: The full text fo the paper\\'s methods. Don\\'t take the information from abstract.\\n\\nResults: The full text fo the paper\\'s results. Don\\'t take the information from abstract. Please take all possible text of results, this section could be divided into different sections.\\n\\nDiscussion:The full text fo the paper\\'s discussion if provided. Otherwise leave the filed blank. It shoudl be entitled \"Discussion\"\\n\\nConclusion: The full text fo the paper\\'s conclusion if provided. Otherwise leave the filed blank. Please don\\'t take this information form abstract.\\n\\nURL: the link where you can find the article. This link is also known as DOI.\\n\\n\\nGuidelines:\\n\\n\\nThe extracted information should be factual and accurate to the document. Be extremely concise, except for the Abstract, Introduction, Methods, Results. Discussion and Conclusion which should be copied in full.\\nThe extracted entities should be self-contained and easily understood without the rest of the paper. If a property is missing from the paper, please leave the field empty rather than guessing.\\nAnswer in JSON format. The JSON should contain 9 keys: \"PaperTitle\", \"PublicationYear\", \"Authors\", \"Email\", \"Abstract\", \"Introduction\", \"Methods\",\"Results\", \"URL\".\\n\\n\\nFormat instructions: \\n{format_instructions}\\n\\n\\n    \\nThe article is this:\\n{query}\\n')\n",
       "| GoogleGenerativeAI(model='gemini-2.0-flash', google_api_key=SecretStr('**********'), temperature=0.2, client=ChatGoogleGenerativeAI(model='models/gemini-2.0-flash', google_api_key=SecretStr('**********'), temperature=0.2, client=<google.ai.generativelanguage_v1beta.services.generative_service.client.GenerativeServiceClient object at 0x7feaa46506d0>, default_metadata=(), model_kwargs={}))\n",
       "| JsonOutputParser(pydantic_object=<class '__main__.paper'>)"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chain = prompt | llm | parser\n",
    "chain"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "e1f44809",
   "metadata": {},
   "outputs": [],
   "source": [
    "my_json = []\n",
    "my_json.append(chain.invoke({\"query\": doc[0].page_content}))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "94faaacc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'PaperTitle': 'Deciphering a\\xa0shared transcriptomic regulation and\\xa0the\\xa0relative contribution of\\xa0each regulator type through\\xa0endometrial gene expression signatures',\n",
       " 'Publication': '2023',\n",
       " 'Authors': 'Antonio Parraga‑Leo, Patricia Sebastian‑Leon, Almudena Devesa‑Peiro, Diana Marti‑Garcia, Nuria Pellicer, Jose Remohi, Francisco Dominguez and Patricia Diaz‑Gimeno',\n",
       " 'Email': 'patricia.diaz@ivirma.com; patricia_diaz@iislafe.es',\n",
       " 'Abstract': 'Backgorund While various endometrial biomarkers have been characterized at the transcriptomic and functional level, there is generally a poor overlap among studies, making it unclear to what extent their upstream regulators (e.g., ovarian hormones, transcription factors (TFs) and microRNAs (miRNAs)) realistically contribute to menstrual cycle progression and function. Unmasking the intricacies of the molecular interactions in the endometrium from a novel systemic point of view will help gain a more accurate perspective of endometrial regulation and a better explanation the molecular etiology of endometrial‑factor infertility.\\nMethods An in-silico analysis was carried out to identify which regulators consistently target the gene biomarkers proposed in studies related to endometrial progression and implantation failure (19 gene lists/signatures were included). The roles of these regulators, and of genes related to progesterone and estrogens, were then analysed in transcriptomic datasets compiled from samples collected throughout the menstrual cycle (n = 129), and the expression of selected TFs were prospectively validated in an independent cohort of healthy participants (n = 19).\\nResults A total of 3,608 distinct genes from the 19 gene lists were associated with endometrial progression and implantation failure. The lists’ regulation was significantly favoured by TFs (89% (17/19) of gene lists) and progesterone (47% (8 /19) of gene lists), rather than miRNAs (5% (1/19) of gene lists) or estrogen (0% (0/19) of gene lists), respectively (FDR < 0.05). Exceptionally, two gene lists that were previously associated with implantation failure and unexplained infertility were less hormone‑dependent, but primarily regulated by estrogen. Although endometrial progression genes were mainly targeted by hormones rather than non‑hormonal contributors (odds ratio = 91.94, FDR < 0.05), we identified 311 TFs and 595 miRNAs not previously associated with ovarian hormones. We highlight CTCF, GATA6, hsa‑miR‑15a‑5p, hsa‑miR‑218‑5p, hsa‑miR‑107, hsa‑miR‑103a‑3p, and hsa‑miR‑128‑3p, as overlapping novel master regulators of endometrial function. The gene expression changes of selected regulators through‑out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their potential regulatory roles in the endometrium.',\n",
       " 'Introduction': 'Background\\nThe endometrium is the innermost layer of the uterus, \\nwhich undergoes dynamic histological, physiological \\nand molecular changes that allow it to synchronize with \\nembryo development, facilitate embryo implantation, \\nand ultimately, establish a successful pregnancy [1].\\nImplantation is a crucial and complex limiting step \\nfor conception [2] which occurs between days 19 to 24 \\nof a normal menstrual cycle. Consequently, this period \\nof time is called the window of implantation (WOI) [3], \\nand is characterized by abrupt transcriptomic changes \\nin the endometrial tissue [4]. Alterations in either WOI \\nestablishment and endometrial progression may lead to \\nimplantation failures, which, along with biochemical mis-\\ncarriage, account for more than 50% of pregnancy losses \\nat pre-clinical stages [5, 6]. These alterations have broadly \\nbeen classified as being related to a displaced WOI, \\ncaused by variable timing of endometrial progression, or \\na disrupted WOI, where impaired endometrial function \\nprevents the establishment of an effective WOI [7].\\nInitially, the WOI was postulated to mainly be regu -\\nlated by the ovarian hormones (namely, estrogen and \\nprogesterone) acting via their respective nuclear recep -\\ntors (i.e., ESR1/2 and PGR). However, evidence has shown \\nthat these receptors cooperate with other transcription \\nfactors (TFs) and co-regulators to mediate uterine physi -\\nology [8]. In fact, TFs (such as homeobox TFs, FOXA2, \\nand KLF9) have recently been reported as key regulators \\nfor the establishment of endometrial receptivity [9, 10], \\nand microRNAs (miRNAs) have been found to similarly \\nintervene in this complex and multifactorial process as \\ntranscriptional regulators [11]. While the involvement of \\nboth types of regulators was widely reported in embryo \\nimplantation [12] and other infertility-related diseases, \\nincluding recurrent implantation failure (RIF) [13], the \\ncomplex interactions between the different types of \\nregulators and their respective contributions to endo -\\nmetrial progression and function have not been investi -\\ngated from a systemic point of view. This paradigm shift \\napproaches genes and regulators in an integrative way, by \\nconsidering how they interact with and coordinate each \\nother to carry out cellular processes, rather than rein -\\nforcing the premise they work independently [14]. By \\nmore accurately reflecting the biological milieu, systemic \\napproaches take gene-based discoveries to the next level, \\nhelping to generate hypotheses with relevant clinical and \\nmolecular implications [15, 16].\\nPrevious studies of endometrial transcriptomics have \\nbeen used to define the WOI, compare the endometrium \\nof healthy women as their menstrual cycles progressed \\n[4, 17–19]; address endometrial differences related to age \\n[20] or the dysfunction of the WOI between patients with \\nimplantation failure and healthy controls [21]; and finally, \\ncorrect for the menstrual cycle bias that can mask impor-\\ntant biomarkers [22]. Notwithstanding, the candidate \\nbiomarkers of endometrial receptivity reported in each \\nof the aforementioned studies overlap poorly [7]. Indeed, \\nrecent comparisons conducted by Sebastian-Leon and \\ncolleagues [7] found no congruities between 16 different \\nreported gene lists of endometrial receptivity, and only \\nfair to moderate functional agreements between some of \\nthe included signatures. Despite previous transcriptional \\nregulation studies, there have been no holistic studies of \\nthis regulatory process, which robustly analyse the driv -\\ners of transcription across a higher number of studies \\n(that each searched for endometrial receptivity biomark -\\ners) and map the relative contributions of each type of \\nregulator to endometrial progression regulation (Ovarian \\nhormones [i.e., estrogen, progesterone], TFs, miRNAs).\\nThus, the aim of this study was to utilize available \\ngenomic data regarding transcriptional regulators to \\nidentify overlapping mediators of endometrial transcrip -\\ntional regulation (i.e., ovarian hormones, TFs, and miR -\\nNAs) and determine the relative contribution of each \\ntype of regulator among the previously reported gene \\nsignatures. These findings could help unveil the master \\nregulators and principal type of endometrial regulation, \\npaving the way for further research aimed at improving \\nwoman´s reproductive health.',\n",
       " 'Methods': 'Methods\\nA detailed study design is depicted in Supplementary \\nFig. S1.\\nAnnotating gene lists associated with\\xa0endometrial \\nprogression and\\xa0function\\nEndometrial progression and implantation failure gene \\nlists [7] were retrieved and updated using a public data \\nrepository (i.e., Gene Expression Omnibus (GEO)).\\nKeywords used for the GEO search included: endometrial \\nreceptivity, mid-secretory endometrium, RIF, recurrent \\nimplantation failure, endometrium, unexplained infer -\\ntility, and implantation failure. The resulting datasets \\nwere filtered by their publication date (i.e., from January \\n2018 to October 2020, to expand the systematic search of \\n[7], number of samples (> 3 samples for each condition), \\nand species (i.e., Homo sapiens), with no restrictions on \\npublication language. The original signatures were then \\nretrieved from their corresponding publications. The \\ngenes prioritized in the original publication were exclu -\\nsively selected and their names were annotated with \\nHUGO Gene Nomenclature Committee (HGNC) gene \\nname using biomaRt R-package v.3.10 [23]. Each signa -\\nture was labelled with the name of first author of the cor -\\nresponding study. Our annotated gene lists were selected \\nas representative signatures of endometrial progression \\nand function for subsequent analyses (Supplementary \\nFig. S1A).\\nIdentifying hormonal and\\xa0non‑hormonal gene regulators \\nof\\xa0endometrial progression and\\xa0function\\nTo study the hormonal regulation of endometrial \\nprogression, the Kyoto Encyclopaedia of Genes and \\nGenomes (KEGG) [24], and Gene Ontology (GO) [25] \\ndatabases were consulted for ovarian hormone-related \\ngenes, using relevant keywords (i.e., progesterone, estro-\\ngen, oestrogen and estradiol) and selecting those pathways \\nand functions that contained them (Supplementary Fig. \\nS1B). Genes associated with the obtained human KEGG \\npathways and GO functions were grouped according to \\nwhether they were unique to progesterone (P4 gene set) \\nor estrogen (E2 gene set) or related to both hormones \\n(P4 and E2 gene set). Gene targets of the nuclear proges -\\nterone (PGR) and estrogen (ESR1, ESR2) receptors were \\nadded to their corresponding gene sets using DoRothEA \\n(Discriminant Regulon Expression Analysis) database \\n[26], considering only manually-curated or ChiP-Seq \\nexperimentally-validated gene-TF relationships. Finally, \\nthe P4 and E2 gene sets were independently mapped to \\neach gene list.\\nMeanwhile, to evaluate the non-hormonal regula -\\ntors of endometrial progression and function, DoRo -\\nthEA [26] and TarBase [27] databases were consulted to \\nobtain TFs and miRNAs. DoRothEA was filtered as pre -\\nviously described herein, while TarBase, which only con -\\ntains miRNA-gene target relationships manually curated \\nfrom publications or experimentally-validated in high-\\nthroughput datasets [27], was filtered by species (i.e., \\nhuman). Finally, a functional over-representation analysis \\nwas carried out to identify which of the total annotated \\nTFs or miRNAs were significantly associated with a par -\\nticular gene list (Supplementary Fig. S1B).\\nTo evaluate the relative contribution of each type of \\nregulator (i.e., P4, E2, TFs, and miRNAs) on endometrial \\nprogression and function, three different comparisons \\nwere carried out applying Fisher’s exact tests. First, hor -\\nmonal regulators were evaluated using the relative pro -\\nportion of P4- and E2-related genes included in each list, \\nwith respect to the total of number of genes founded in \\nthe P4 or E2 gene sets. Then, non-hormonal regulators \\nwere evaluated using the relative proportion of over-\\nrepresented miRNAs and TFs with respect to the total of \\nmiRNAs and TFs with a at least one target in the gene \\nlist. Finally, the relative proportion of genes under hor -\\nmonal (i.e., P4 and/or E2-annotated genes) versus non-\\nhormonal (i.e., miRNA and/or TF-annotated genes) \\nregulation was evaluated considering the miRNAs/TFs \\nfound within the individual P4 or E2 gene sets as hormo -\\nnal regulators.\\nPrioritizing key regulators in\\xa0endometrial progression \\nand\\xa0function\\nWe built regulatory networks to systemically analyse \\nendometrial regulation. Nodes represented gene lists \\nand regulators (miRNAs or TFs), while edges indicated \\nsignificant enrichment among the lists and their corre -\\nsponding regulators (false discovery rate (FDR) ≤ 0.05). \\nTo select the most influential regulators (i.e., those which \\ntargeted genes in most of our signatures), we studied the \\ndegree distribution of the networks (i.e., the number of \\ngene lists regulated by each molecule) and prioritized the \\nmiRNAs and TFs which surpassed the relative maximum \\n(i.e., 1.50 times the interquartile range (IQR)) number of \\nrelationships (gene list–regulators). All networks were \\nbuilt and analysed with Cytoscape version 3.7 [28] (Sup -\\nplementary Fig. S1B).\\nDataset construction and\\xa0processing for\\xa0in‑silico validation\\nTo evaluate how the miRNAs’ expression change in the \\nendometrium throughout the menstrual cycle, we used \\nthe raw data from GSE44558 GEO dataset [29], deriving \\nfrom 20 endometrial samples collected throughout the \\nmenstrual cycle: four in early-proliferative (EPF), four \\nin late-proliferative (LPF), four in early-secretory (ESE), \\nfour in mid-secretory (MSE) and four in late-secretory \\n(LSE).\\nTo evaluate how the TFs’ expression change in the \\nendometrium throughout the menstrual cycle, we \\nanalysed the integrated endometrial dataset previ -\\nously created by our group, and reported in [30], which \\ncompiled data from five prior publications (GSE98386, \\nGSE29981, GSE4888, GSE119209 and GSE86491). \\nThis dataset included endometrial gene expression\\ndata from 109 participants with normal endometrium, \\nwhere biopsies were collected in proliferative (PF) \\n(n = 29), ESE (n  = 29), MSE (n  = 43) and LSE (n  = 8) \\nphases of the menstrual cycle (Table S1 ).\\nBoth TF and miRNA datasets were processed using \\nthe limma R-package [31]. Expression data were log \\ntransformed and quantile normalized, prior to explora -\\ntory analyses which sought out possible outliers and \\nbatch effects. Relative gene expression ranges of low \\n(1–10%), medium (11–50%), and high (51–100%) were \\nestablished based on the expression of all the genes \\nincluded in our dataset.\\nWet lab validation cohorts\\nTo corroborate the menstrual cycle-related expression \\nchanges observed with CTCF and GATA6 in our in-\\nsilico analysis, we conducted a prospective study as an \\nexternal validation, using an independent sample set. \\nWe included a cohort of 19 healthy Spanish women \\n(obtaining a total of 20 biopsies), between the ages of \\n22–35 and with a body mass index of 22.80 ± 2.76\\xa0 kg/\\nm2. Endometrial samples were collected and staged in \\nthe menstrual cycle according to the follicle growth to \\ncontrol ovulation and LH levels in urine. Samples were \\ngrouped into PF (n  = 5), ESE (n  = 5), MSE (n  = 5) and \\nLSE (n = 5) phases and categorized into two groups, PF/\\nESE and MSE/LSE.\\nRNA extraction and\\xa0RT‑qPCR\\nTotal RNA was extracted from the endometrial biopsy \\nsamples using the miRNeasy Mini Kit (Qiagen, Ger -\\nmany), and reverse transcribed into cDNA using the \\nPrimeScript reagent kit (TAKARA, Japan). RT-qPCR \\nreactions were carried out in duplicate, using fluores -\\ncent Power-up SYBR Green (Thermo Fisher Scientific \\nMA, USA) in a final volume of 10 µL. Primer sequences \\nare shown in Table S2. Samples underwent 40 cycles of \\namplification, under standard conditions, using a Ste -\\npOnePlus™ System (Applied Biosystems, MA, USA). \\nRelative mRNA expression was calculated using the  2−\\nΔΔCt method [32], and normalized to the expression of \\nGAPDH housekeeping gene.\\nStatistical analysis\\nFisher’s exact test was used to evaluate independent \\nproportions. Mean expression changes across the cycle \\nwere studied using an analysis of variance (ANOVA) fol -\\nlowed by a pairwise t-test or the Wilcoxon test when only \\ntwo groups were available. P-values from multiple test \\ncomparisons were corrected by FDR and considered as \\nsignificant when FDR ≤ 0.05. All statistical analyses were \\nperformed through R software (version 3.5) [33].',\n",
       " 'Results': 'Gene signatures associated to\\xa0endometrial progression \\nand\\xa0function\\nOf the 19 gene lists used for our analysis, eleven were \\nobtained from published studies evaluating control \\npatients throughout the menstrual cycle [4, 17–19, \\n34–40], and the other eight were derived from studies \\ncomparing patients with RIF or unexplained infertility \\nto controls [21,\\xa0 41–47] (Table\\xa01). Unifying all the afore -\\nmentioned signatures, we compiled 3,608 genes related \\nto endometrial progression and function.\\nHormonal regulation of\\xa0endometrial progression is\\xa0largely \\ndriven by\\xa0progesterone\\nWe identified 7,540 and 698 genes related to estrogen \\nand progesterone hormones, respectively. However, as \\ndetermined by the relative contribution of each type of \\nhormone within each gene list (Fig.\\xa0 1A), 17/19 (89%) \\nsignatures favoured regulation by progesterone rather \\nthan estrogen. These differences were significant in \\n47% of the signatures (FDR < 0.05). Exceptionally, the \\ngenes reported by the Altmae2010 and Koot signa -\\ntures, that were related to unexplained infertility and \\nRIF, had a preference for estrogen regulation (Fig.\\xa0 1A). \\nWith respect to the proportion of progesterone-related \\ngenes, Altmae2010 showed 2.1 times more estrogen-\\nrelated genes while Koot showed 1.7 times more. More -\\nover, comparing the proportion of genes regulated by \\nboth hormones, between signatures revealed that Alt -\\nmae2010 and Koot exhibited a lower hormone-depend -\\nent behaviour overall, with respect to the remaining \\nsignatures (Fig.\\xa0 1B), highlighting the role of non-hor -\\nmonal regulators.\\nNon‑hormonal regulation of\\xa0endometrial progression \\nand\\xa0function\\nWe identified a total of 770 miRNAs and 555 TFs regu -\\nlating at least one of the 3,608 genes reported in the \\n19 gene lists. Further, we observed that 29% of the \\nmiRNAs and 44% of the TFs were associated with the \\novarian hormones (i.e., progesterone and/or estrogen). \\nSignificantly more genes were targeted by hormo -\\nnally-influenced regulators rather than non-hormonal \\nregulators (P -value < 0.01, odds ratio =  91.94). Never -\\ntheless, we also identified 595 miRNAs (77.27%) and \\n311 TFs (56.04%) that were unrelated to progesterone \\nand/or estrogen, highlighting a substantial proportion \\nof hormonally-independent regulators of the men -\\nstrual cycle.\\nHsa‑miR‑15a‑5p, hsa‑miR‑128‑3p, hsa‑miR‑218‑5p, \\nhsa‑miR‑27b‑3p, hsa‑miR‑107, hsa‑miR‑424‑5p, \\nhsa‑miR‑195‑5p, hsa‑miR‑103a‑3p, has‑let‑7b‑5p \\nand\\xa0hsa‑miR‑22‑3p are key players in\\xa0the\\xa0regulation \\nof\\xa0endometrial progression and\\xa0function\\nTo bridge the gap of endometrial-based evidence in \\nDoRothEA and TarBase databases, we aimed to iden -\\ntify the common miRNAs that mediate gene expres -\\nsion related to endometrial progression and function. \\nWe evaluated the number of endometrial gene lists \\nthat each miRNA controlled (Fig.\\xa0 2A). This strategy \\nhighlighted the most influential miRNAs that were \\ncommonly found across the endometrial gene lists. \\nNotably, 85.80% of miRNAs regulated genes from one \\nto three signatures, and only 59 miRNAs (14.20%) were \\nassociated with more than three signatures.  Ultimately, \\n24 miRNAs were prioritized for regulating genes of at \\nleast six signatures.\\nNext, we validated the gene expression trends for \\nthese 24 prioritized miRNAs in an independent endo -\\nmetrial gene expression dataset (GSE44730) (Fig.\\xa0 2B). \\nThis dataset included the miRNA expression in endo -\\nmetrial biopsies from 20 healthy women in differ -\\nent phases of the menstrual cycle. We found that \\ntwo of our prioritized miRNAs were not expressed \\nin the endometrial tissue, and only 10 miRNAs (i.e., \\nhsa-miR-15a-5p, hsa-miR-128-3p, hsa-miR-218-5p, \\nhsa-miR-27b-3p, hsa-miR-107, hsa-miR-424-5p, hsa-\\nmiR-195-5p, hsa-miR-103a-3p, has-let-7b-5p and hsa-\\nmiR-22-3p) showed significant changes across the \\nmenstrual cycle (Fig.\\xa0 2B). Of these, only the expression \\nof hsa-miR-22-3p significantly increased from ESE \\nto MSE stages (FDR = 0.02, Fig.\\xa0 2B), highlighting the \\npotential inhibitory role of this miRNA in endometrial \\nfunction. The expression of the remaining miRNAs \\ndecreased during the MSE phase.\\nCTCF and\\xa0GATA6 as\\xa0distinguished transcriptional \\nregulators of\\xa0the\\xa0human menstrual cycle\\nWe repeated the regulation network analysis with the \\nTFs, to identify the prominent TFs commonly regu -\\nlating the genes related to endometrial progression \\nand function in the reported signatures. We observed \\n78.80% of the TFs regulated genes in up to 10 signa -\\ntures, and only 99 TFs (21.20%) regulated genes in over \\n10 (Fig.\\xa0 3A). We highlighted CTCF (a CCCTC-binding \\nfactor, which functions as a transcriptomic repressor) \\nas the TF with the most influential regulation of endo -\\nmetrial progression and function across 95% of studies \\n(18 gene lists, excluding that of Punyadeera, Fig.\\xa0 3A). \\nOther distinguished TFs that regulated up to 15 gene \\nlists included AR, CEBPA, CEBPB, CEBPD, CREB1, \\nEGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, \\nGATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, \\nNR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, \\nTCF7L2, TFAP2C and TP53. Notably, GATA6 was \\nselected due to novel association with endometrial pro -\\ngression and function.\\nTo infer the expression of CTCF and GATA6  in \\nendometrial tissue throughout the menstrual cycle, \\nwe analysed their expression using a relevant endo -\\nmetrial dataset recently created by our group [30] \\n(Table S1 ). We observed that, relative to the global \\nexpression of endometrial genes, CTCF was broadly \\nexpressed while GATA6  was moderately expressed. \\nIndeed, CTCF was significantly down-regulated \\nbetween the PF and ESE phases (FDR = 2.10E-03)  \\nand ESE to the MSE phase (FDR = 2.10E-04), while \\nGATA6 expression increased from the PF to the \\nMSE phase (FDR = 1.70E-11) and from the ESE  \\nto MSE phase (FDR = 9.80E-15). These findings \\nsupported the role of these TFs at the beginning \\nof the menstrual cycle and during the MSE phase \\n(Fig.\\xa03 B).\\nThese in-silico  validations were then prospec -\\ntively corroborated by RT-qPCR analysis in an inde -\\npendent sample set of 20 endometrial biopsies. We \\nobserved a similar expression profile of CTCF and \\nGATA6 throughout the menstrual cycle, with signifi -\\ncant changes between the endometrial phases. Nota -\\nbly, CTCF expression was decreased between the PF/\\nESE and MSE/LSE phases (FDR = 0.01) while GATA6 \\nexpression increased from the PF/ESE to MSE/LSE \\nphases (FDR = 1.55E-04) (Fig.\\xa03 C).',\n",
       " 'Discussion': 'Discussion\\nThis data-driven approach exposed the common tran -\\nscriptional regulators among 19 studies who proposed \\nvariable biomarkers of endometrial progression and \\nfunction. In this study, we focused on understanding \\nthe relative contribution of both the hormonal and non-\\nhormonal regulation, from an alternative holistic per -\\nspective. We applied data-driven hypothesis research \\nthat, unlike the traditional scientific method, allowed us \\nto generate new hypotheses based on all available bio -\\nlogically-relevant knowledge [16], observing the molec -\\nular relationships from a wider scale view. Besides, \\nsome traditional molecular procedures such as PCRs \\nare also performed to corroborate the insight uncovered \\nwith this approach. We highlighted a larger influence \\nof progesterone-related genes and TFs in endome -\\ntrial progression and observed a larger contribution \\nof miRNAs and estrogen-related genes in endometrial \\npathology. Furthermore, we unveiled CTCF, GATA6, \\nhsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-\\nmiR-103a-3p and hsa-miR-128-3p as master regulators \\nin the endometrium, and these findings were reinforced \\nby the dynamic changes in their expression throughout \\nthe menstrual cycle (dually validated with in-silico  and \\nprospective analyses).\\nThe menstrual cycle is perceived to be under tight \\nregulation by the ovarian hormones, progesterone \\nand estrogen [48]. Supporting the role of progester -\\none in the secretory endometrium [49], and reinforc -\\ning its indispensable role in endometrial regulation \\n[50, 51], we found progesterone and its related genes \\nhad a larger influence in menstrual cycle progression \\nthan estrogen. As the RIF-related gene lists reported \\nby Altmae2010 and Koot were generally controlled by \\nestrogen, this further suggested that while progester -\\none could be crucial for a suitable endometrial progres-\\nsion and mid-secretory phase acquisition, estrogenic \\nirregularities may be more conducive to endometrial \\nfunction and molecular pathology. Indeed, this hypoth -\\nesis corroborated previous reports of most uterine dis -\\norders being estrogen-dependent [44, 52]. Supporting \\nthe findings from another recent study from our group, \\nwhere the Altmae2010 and Koot signatures were classi -\\nfied as highly-predictive signatures to identify endome -\\ntrial pathology in comparison with a prediction model \\nthat detected endometrial progression [7 ]. This study \\ndistinguished these signatures (related to implantation \\nfailure and unexplained infertility) based on the mag -\\nnitude of their miRNA and estrogen-regulation, which \\ncontrasted with the prominent influence of the TFs over \\nthe remaining signatures more related to endometrial \\nprogression. All these facts reinforce the hypothesis \\nthat implantation failure is mainly caused by impaired \\nmiRNA expression [53].\\nAlthough we observed a clear hormonal influence in \\nnumerous annotated regulators, this study transcended \\nthe simplistic paradigm of the bihormonal regulation of \\nendometrial function by identifying a plethora of medi -\\nators, including 595 miRNAs and 311 TFs, that were \\npreviously unrelated to these ovarian hormones. These \\nfindings set the foundation for new discoveries, reveal -\\ning alternative pathways and new actors in the regula -\\ntion of the human endometrium, and deepening our \\nunderstanding of the complex regulatory mechanisms \\nbehind endometrial progression and unexplained \\ninfertility.\\nMicroRNAs are well known to be involved in embryo \\nimplantation [12] and receptivity control [54, 55]. Using \\nnetwork analysis, we unveiled the 10 common miR -\\nNAs regulating endometrial progression and function \\nacross published biomarker signatures (hsa-miR-22-3p, \\nhsa-miR-107, hsa-miR-103a-3p, hsa-miR-128-3p, hsa-\\nmiR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, hsa-\\nmiR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p). \\nTheir significant gene expression changes in the secre -\\ntory endometrium support their potential role in the \\nendometrial regulation through an inhibition during \\nthe WOI. Notably, hsa-miR-424-5p, hsa-miR-27b-3p \\nand hsa-miR-195-5p have been previously associated \\nwith RIF [11, 54, 56], while has-let-7b-5p was associ -\\nated with human endometrial receptivity [57], and \\nhsa-miR-22-3p was overexpressed in a decidualization \\nmodel [58]. Moreover, hsa-miR-15a-5p was previously \\nassociated with endometriosis [59]. Notwithstanding, \\nthis study is the first to associate hsa-miR-15a-5p, hsa-\\nmiR-218-5p, hsa-miR-103a-3p, hsa-miR-107, and hsa-\\nmiR-128-3p to endometrial progression and functional \\nregulation in humans.\\nDue to their direct relationship with progesterone and \\nestrogen [60], we expected to find TFs, such as ESR1  \\nand PGR, involved in endometrial progression, how -\\never, we additionally identified other TFs with an even \\nlarger influence in this process. For instance, STAT3  \\nwas associated with 17/19 signatures (FDR < 0.05), and \\nis known to regulate uterine epithelial junctional reor -\\nganization and stromal proliferation, which are critical \\nfor implantation [61]. SP1, which similarly regulated \\n17/19 signatures (FDR < 0.05), acts as a downstream \\nparacrine target of progesterone to regulate estrogen \\ninactivation, and could have a predominant role dur -\\ning the WOI [62, 63]. By focusing on the most univer -\\nsal, overlapping TFs that were not previously related \\nto implantation, we ultimately prioritized CTCF and \\nGATA6 as novel key regulators of endometrial pro -\\ngression, however, their specific molecular actions in\\nendometrial progression and function were beyond the \\nscope of this project and merit further investigation.\\nCTCF is a conserved zinc finger protein whose regu -\\nlatory functions are well characterized throughout the \\nhuman body [64]. CTCF acts as a transcriptional repres -\\nsor in RNA polymerase II (Pol II) pausing and imprint -\\ning and\\xa0 X-chromosome inactivation [65], as well as an \\ninsulator, blocking the interaction between enhancers \\nand the promoters of neighbouring genes [66]. Consider -\\ning these attributed functions and its variable expression \\nthroughout the menstrual cycle, we propose that CTCF \\nexerts an inhibitory role in endometrial tissue during \\nthe PF phase of the menstrual cycle. With the significant \\ndownregulation of CTCF in the secretory endometrium, \\ndually validated in-silico  and experimentally herein, its \\ninhibited genes would be derepressed and become tran -\\nscriptionally active during the WOI. This interpretation \\nsupports previous findings from our group demonstrat -\\ning that, during the WOI, a global transcriptional dere -\\npression may be required for implantation and early \\nembryo development [63]. Transcriptional derepression \\nhas been associated with multiple human disease states \\nand should be investigated further within the context \\nof endometrial-factor infertility. Despite previous asso -\\nciations of CTCF with endometriosis [67], its implica -\\ntion in endometrial progression and function has not \\nbeen directly proposed until now. Nevertheless, the role \\nof CTCF in endometrial receptivity has been revealed \\nthrough its interaction with HOXA10, a gene mainly \\nexpressed in endometrium involved in functions such as \\nendometrial proliferation and differentiation, the forma-\\ntion of pinopodes or embryo implantation [68]. Indeed, \\nthis study proposed that an overexpression of CTCF can \\nlead to a drop in HOXA10 expression, affecting endome-\\ntrial proliferation and endometrial function and corrobo-\\nrating the fact that CTCF must decrease its expression as \\nwe can observe in our findings.\\nGATA6, the second prioritized TF that overlapped \\nacross the gene lists, is a transcription factor belong -\\ning to the GATA family, a highly conserved family of \\nsix zinc finger proteins [69]. Besides its essential role in \\nembryonic development [70], GATA6 plays a key role \\nin endometriosis and regulates steroidogenic genes [71]. \\nTogether with evidence supporting both an activating \\nand inhibitory role for GATA6 [72, 73], the significant \\nchanges we observed in its expression across the men-\\nstrual cycle reflect its presumed involvement in the endo-\\nmetrium and suggest an activating role in endometrial \\nprogression and endometrial receptivity acquisition. Like \\nCTCF, this is the first time that GATA6 is proposed as a \\nkey TF in endometrial progression and function.\\nAlthough the exact molecular mechanisms underlying \\nendometrial regulation remain elusive, emerging exper -\\nimental technologies such as High-throughput Chro -\\nmosome Conformation Capture (Hi-C) or chromatin \\ninteraction analysis with paired-end tags (ChIA-PET) \\nthat respectively study general chromatin interactions \\nor chromatin interactions involving a specific protein \\n[74, 75] may improve our understanding of genetic \\nregulation in the future. Nevertheless, as collaborative \\nefforts unveil new regulators and gene targets in humans \\n(such as the ENCODE project [76]), the complexities of \\ntranscriptional regulation can be exposed by in-silico  \\nanalyses like those described herein, and we can gain a \\nbetter understanding of the processes involved in men -\\nstrual cycle progression and endometrial competence.\\nFurthermore, it should be noted that results of gene \\nexpression studies are influenced by different variables, \\nsuch as the experimental platforms employed, study \\ndesigns, or patients’ characteristics of each independent \\nstudy, adding variability [30, 77, 78]. Despite this fact, \\nwe have found that our holistic approach overcomes this \\nundesired variability effect by finding common upstream \\nmaster regulators across studies and predicting new \\nrelationships between hormones, TFs and miRNAs. \\nThis approach also provides the basis for future single-\\nmolecule studies that aim to elucidate new regulatory \\npathways in endometrial progression and function. Our \\nresults lay the foundation'}"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "my_json[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7733fdb9",
   "metadata": {},
   "source": [
    "## Others"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a10b4d6d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Devesa-Peiro2020.pdf\n",
      "Henarejos-Castillo2020.pdf\n",
      "Sebastian-Leon2021.pdf\n",
      "Marti-Garcia2024(1).pdf\n",
      "Henarejos-Castillo2021.pdf\n",
      "Henarejos-Castillo2022.pdf\n",
      "Diaz-Gimeno2024.pdf\n",
      "Devesa-Peiro2021.pdf\n",
      "Henarejos-Castillo2024.pdf\n",
      "parraga-leo_2023.pdf\n",
      "Marti-Garcia2024.pdf\n",
      "Diaz-Gimeno2022.pdf\n",
      "Devesa-Peiro2022.pdf\n",
      "Sebastian-Leon2018.pdf\n",
      "Diaz-Gimeno2017.pdf\n"
     ]
    }
   ],
   "source": [
    "# Load several documents\n",
    "\n",
    "storage = []\n",
    "\n",
    "for root, dirs, files in os.walk(\"Data\",): # for server all the path\n",
    "\n",
    "    for file in files:\n",
    "        print(file)\n",
    "\n",
    "        loader = PyPDFLoader(os.path.join(root,file))\n",
    "        pages = loader.load_and_split()\n",
    "\n",
    "        # split it into chunks\n",
    "        text_splitter = CharacterTextSplitter(chunk_size=1000, chunk_overlap=50)\n",
    "        docs = text_splitter.split_documents(pages)\n",
    "        storage.extend(docs) # This  is a list of lists\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "4e00cd7b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load several documents\n",
    "\n",
    "storage = []\n",
    "\n",
    "for root, dirs, files in os.walk(\"Data\",): # for server all the path\n",
    "\n",
    "    for file in files:\n",
    "\n",
    "        loader = PyPDFLoader(os.path.join(root,file),mode='page')\n",
    "        pages = loader.load() # List of pages\n",
    "        \n",
    "        text_splitter = RecursiveCharacterTextSplitter(chunk_size=500, chunk_overlap=100)\n",
    "        docs = text_splitter.split_documents(pages)\n",
    "\n",
    "        storage.extend(docs) # Add several items at the same time\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "04f51224",
   "metadata": {},
   "outputs": [],
   "source": [
    "# load the document and split it into pages\n",
    "loader = PyPDFLoader(\"Publications/Parraga-Leo2023.pdf\")\n",
    "pages = loader.load_and_split()\n",
    "\n",
    "# split it into chunks\n",
    "text_splitter = RecursiveCharacterTextSplitter(chunk_size=3000, chunk_overlap=100)\n",
    "storage = text_splitter.split_documents(pages)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "fcdd0572",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('out the menstrual cycle (FDR < 0.05), dually validated in-silico and through '\n",
      " 'endometrial biopsies, corroborated their \\n'\n",
      " 'potential regulatory roles in the endometrium.\\n'\n",
      " '*Correspondence:\\n'\n",
      " 'Patricia Diaz‑Gimeno\\n'\n",
      " 'patricia.diaz@ivirma.com; patricia_diaz@iislafe.es\\n'\n",
      " 'Full list of author information is available at the end of the article')\n"
     ]
    }
   ],
   "source": [
    "pprint.pprint(storage[2].page_content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "7725a5a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "model = ChatOllama(\n",
    "    model=\"gemma3:4b\",\n",
    "    temperature=0.1,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "61f23c4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "class paper(BaseModel):\n",
    "    text: str = Field(description=\"Main text that explains the reuslts, introductions, methods or conclusions\")\n",
    "    doi: str = Field(description=\"link with the url of scientifc article known as doi.\")\n",
    "    citation: str = Field(description=\"add the citation of the manuscript is presented in the text.\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "88db094b",
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = JsonOutputParser(pydantic_object=paper)\n",
    "\n",
    "prompt = PromptTemplate(\n",
    "    template=\"\"\"\n",
    "    You are an expert analyzing scientifc research papers. Read the text carefully to provide the request information.\n",
    "    Remove the potential information that is not related with the scientific article.\n",
    "\n",
    "    If it appears a link with the doi of the artcile, take it. Otherwise write down UNKOWN.\n",
    "\n",
    "    Help me to extract the information and create an json file with it. If you don't know the\n",
    "    answer, say UNKNOWN.\n",
    "    \n",
    "    Format instructions: \\n{format_instructions}\\n\n",
    "    \n",
    "    Case:\\n{query}\\n\"\"\",\n",
    "\n",
    "    input_variables=[\"query\"],\n",
    "    \n",
    "    partial_variables={\"format_instructions\": parser.get_format_instructions()},\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "8ccc261f",
   "metadata": {},
   "outputs": [],
   "source": [
    "chain = prompt | model | parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "94b83fdd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Parraga‑Leo et al. \n",
      "Reproductive Biology and Endocrinology           (2023) 21:84  \n",
      "https://doi.org/10.1186/s12958‑023‑01131‑4\n",
      "RESEARCH Open Access\n",
      "© The Author(s) 2023. Open Access This article is licensed under a Creative Commons Attribution 4.0 International License, which \n",
      "permits use, sharing, adaptation, distribution and reproduction in any medium or format, as long as you give appropriate credit to the \n",
      "original author(s) and the source, provide a link to the Creative Commons licence, and indicate if changes were made. The images or \n",
      "other third party material in this article are included in the article’s Creative Commons licence, unless indicated otherwise in a credit line \n",
      "to the material. If material is not included in the article’s Creative Commons licence and your intended use is not permitted by statutory \n",
      "regulation or exceeds the permitted use, you will need to obtain permission directly from the copyright holder. To view a copy of this \n",
      "licence, visit http://creativecommons.org/licenses/by/4.0/. The Creative Commons Public Domain Dedication waiver (http://creativecom‑\n",
      "mons.org/publicdomain/zero/1.0/) applies to the data made available in this article, unless otherwise stated in a credit line to the data.\n",
      "Reproductive Biology\n",
      "and Endocrinology\n",
      "Deciphering a shared transcriptomic \n",
      "regulation and the relative contribution of each \n",
      "regulator type through endometrial gene \n",
      "expression signatures\n",
      "Antonio Parraga‑Leo1,2, Patricia Sebastian‑Leon1, Almudena Devesa‑Peiro1,2, Diana Marti‑Garcia1,2, \n",
      "Nuria Pellicer1,3, Jose Remohi2,3, Francisco Dominguez1 and Patricia Diaz‑Gimeno1*   \n",
      "Abstract \n",
      "Backgorund While various endometrial biomarkers have been characterized at the transcriptomic and functional \n",
      "level, there is generally a poor overlap among studies, making it unclear to what extent their upstream regulators \n",
      "(e.g., ovarian hormones, transcription factors (TFs) and microRNAs (miRNAs)) realistically contribute to menstrual cycle \n",
      "progression and function. Unmasking the intricacies of the molecular interactions in the endometrium from a novel \n",
      "systemic point of view will help gain a more accurate perspective of endometrial regulation and a better explanation \n",
      "the molecular etiology of endometrial‑factor infertility.\n",
      "Methods An in-silico analysis was carried out to identify which regulators consistently target the gene biomark‑\n",
      "ers proposed in studies related to endometrial progression and implantation failure (19 gene lists/signatures were \n",
      "included). The roles of these regulators, and of genes related to progesterone and estrogens, were then analysed \n",
      "in transcriptomic datasets compiled from samples collected throughout the menstrual cycle (n = 129), and the expres‑\n",
      "sion of selected TFs were prospectively validated in an independent cohort of healthy participants (n = 19).\n",
      "Results A total of 3,608 distinct genes from the 19 gene lists were associated with endometrial progression\n",
      "and implantation failure. The lists’ regulation was significantly favoured by TFs (89% (17/19) of gene lists) and pro‑\n",
      "gesterone (47% (8 /19) of gene lists), rather than miRNAs (5% (1/19) of gene lists) or estrogen (0% (0/19) of gene \n",
      "lists), respectively (FDR < 0.05). Exceptionally, two gene lists that were previously associated with implantation failure \n",
      "and unexplained infertility were less hormone‑dependent, but primarily regulated by estrogen. Although endometrial \n",
      "progression genes were mainly targeted by hormones rather than non‑hormonal contributors (odds ratio = 91.94, \n",
      "FDR < 0.05), we identified 311 TFs and 595 miRNAs not previously associated with ovarian hormones. We highlight \n",
      "CTCF, GATA6, hsa‑miR‑15a‑5p, hsa‑miR‑218‑5p, hsa‑miR‑107, hsa‑miR‑103a‑3p, and hsa‑miR‑128‑3p, as overlapping \n",
      "novel master regulators of endometrial function. The gene expression changes of selected regulators through‑\n",
      "out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their\n",
      "out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their \n",
      "potential regulatory roles in the endometrium.\n",
      "*Correspondence:\n",
      "Patricia Diaz‑Gimeno\n",
      "patricia.diaz@ivirma.com; patricia_diaz@iislafe.es\n",
      "Full list of author information is available at the end of the article\n",
      "Page 2 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "Conclusions This study revealed novel hormonal and non‑hormonal regulators and their relative contributions \n",
      "to endometrial progression and pathology, providing new leads for the potential causes of endometrial‑factor \n",
      "infertility.\n",
      "Keywords CTCF, GATA6, Progesterone, Estrogen, miRNAs, TFs, Endometrial receptivity, Recurrent implantation failure, \n",
      "Infertility, Menstrual cycle regulation\n",
      "Background\n",
      "The endometrium is the innermost layer of the uterus, \n",
      "which undergoes dynamic histological, physiological \n",
      "and molecular changes that allow it to synchronize with \n",
      "embryo development, facilitate embryo implantation, \n",
      "and ultimately, establish a successful pregnancy [1].\n",
      "Implantation is a crucial and complex limiting step \n",
      "for conception [2] which occurs between days 19 to 24 \n",
      "of a normal menstrual cycle. Consequently, this period \n",
      "of time is called the window of implantation (WOI) [3], \n",
      "and is characterized by abrupt transcriptomic changes \n",
      "in the endometrial tissue [4]. Alterations in either WOI \n",
      "establishment and endometrial progression may lead to \n",
      "implantation failures, which, along with biochemical mis-\n",
      "carriage, account for more than 50% of pregnancy losses \n",
      "at pre-clinical stages [5, 6]. These alterations have broadly \n",
      "been classified as being related to a displaced WOI, \n",
      "caused by variable timing of endometrial progression, or \n",
      "a disrupted WOI, where impaired endometrial function \n",
      "prevents the establishment of an effective WOI [7].\n",
      "Initially, the WOI was postulated to mainly be regu -\n",
      "lated by the ovarian hormones (namely, estrogen and \n",
      "progesterone) acting via their respective nuclear recep -\n",
      "tors (i.e., ESR1/2 and PGR). However, evidence has shown \n",
      "that these receptors cooperate with other transcription \n",
      "factors (TFs) and co-regulators to mediate uterine physi -\n",
      "ology [8]. In fact, TFs (such as homeobox TFs, FOXA2, \n",
      "and KLF9) have recently been reported as key regulators \n",
      "for the establishment of endometrial receptivity [9, 10], \n",
      "and microRNAs (miRNAs) have been found to similarly \n",
      "intervene in this complex and multifactorial process as \n",
      "transcriptional regulators [11]. While the involvement of \n",
      "both types of regulators was widely reported in embryo \n",
      "implantation [12] and other infertility-related diseases, \n",
      "including recurrent implantation failure (RIF) [13], the \n",
      "complex interactions between the different types of \n",
      "regulators and their respective contributions to endo -\n",
      "metrial progression and function have not been investi -\n",
      "gated from a systemic point of view. This paradigm shift \n",
      "approaches genes and regulators in an integrative way, by \n",
      "considering how they interact with and coordinate each \n",
      "other to carry out cellular processes, rather than rein -\n",
      "forcing the premise they work independently [14]. By \n",
      "more accurately reflecting the biological milieu, systemic \n",
      "approaches take gene-based discoveries to the next level,\n",
      "helping to generate hypotheses with relevant clinical and \n",
      "molecular implications [15, 16].\n",
      "Previous studies of endometrial transcriptomics have \n",
      "been used to define the WOI, compare the endometrium \n",
      "of healthy women as their menstrual cycles progressed \n",
      "[4, 17–19]; address endometrial differences related to age \n",
      "[20] or the dysfunction of the WOI between patients with \n",
      "implantation failure and healthy controls [21]; and finally, \n",
      "correct for the menstrual cycle bias that can mask impor-\n",
      "tant biomarkers [22]. Notwithstanding, the candidate \n",
      "biomarkers of endometrial receptivity reported in each \n",
      "of the aforementioned studies overlap poorly [7]. Indeed, \n",
      "recent comparisons conducted by Sebastian-Leon and \n",
      "colleagues [7] found no congruities between 16 different \n",
      "reported gene lists of endometrial receptivity, and only \n",
      "fair to moderate functional agreements between some of \n",
      "the included signatures. Despite previous transcriptional \n",
      "regulation studies, there have been no holistic studies of\n",
      "fair to moderate functional agreements between some of \n",
      "the included signatures. Despite previous transcriptional \n",
      "regulation studies, there have been no holistic studies of \n",
      "this regulatory process, which robustly analyse the driv -\n",
      "ers of transcription across a higher number of studies \n",
      "(that each searched for endometrial receptivity biomark -\n",
      "ers) and map the relative contributions of each type of \n",
      "regulator to endometrial progression regulation (Ovarian \n",
      "hormones [i.e., estrogen, progesterone], TFs, miRNAs).\n",
      "Thus, the aim of this study was to utilize available \n",
      "genomic data regarding transcriptional regulators to \n",
      "identify overlapping mediators of endometrial transcrip -\n",
      "tional regulation (i.e., ovarian hormones, TFs, and miR -\n",
      "NAs) and determine the relative contribution of each \n",
      "type of regulator among the previously reported gene \n",
      "signatures. These findings could help unveil the master \n",
      "regulators and principal type of endometrial regulation, \n",
      "paving the way for further research aimed at improving \n",
      "woman´s reproductive health.\n",
      "Methods\n",
      "A detailed study design is depicted in Supplementary \n",
      "Fig. S1.\n",
      "Annotating gene lists associated with endometrial \n",
      "progression and function\n",
      "Endometrial progression and implantation failure gene \n",
      "lists [7] were retrieved and updated using a public data \n",
      "repository (i.e., Gene Expression Omnibus (GEO)).\n",
      "Page 3 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Keywords used for the GEO search included: endometrial \n",
      "receptivity, mid-secretory endometrium, RIF, recurrent \n",
      "implantation failure, endometrium, unexplained infer -\n",
      "tility, and implantation failure. The resulting datasets \n",
      "were filtered by their publication date (i.e., from January \n",
      "2018 to October 2020, to expand the systematic search of \n",
      "[7], number of samples (> 3 samples for each condition), \n",
      "and species (i.e., Homo sapiens), with no restrictions on \n",
      "publication language. The original signatures were then \n",
      "retrieved from their corresponding publications. The \n",
      "genes prioritized in the original publication were exclu -\n",
      "sively selected and their names were annotated with \n",
      "HUGO Gene Nomenclature Committee (HGNC) gene \n",
      "name using biomaRt R-package v.3.10 [23]. Each signa -\n",
      "ture was labelled with the name of first author of the cor -\n",
      "responding study. Our annotated gene lists were selected \n",
      "as representative signatures of endometrial progression \n",
      "and function for subsequent analyses (Supplementary \n",
      "Fig. S1A).\n",
      "Identifying hormonal and non‑hormonal gene regulators \n",
      "of endometrial progression and function\n",
      "To study the hormonal regulation of endometrial \n",
      "progression, the Kyoto Encyclopaedia of Genes and \n",
      "Genomes (KEGG) [24], and Gene Ontology (GO) [25] \n",
      "databases were consulted for ovarian hormone-related \n",
      "genes, using relevant keywords (i.e., progesterone, estro-\n",
      "gen, oestrogen and estradiol) and selecting those pathways \n",
      "and functions that contained them (Supplementary Fig. \n",
      "S1B). Genes associated with the obtained human KEGG \n",
      "pathways and GO functions were grouped according to \n",
      "whether they were unique to progesterone (P4 gene set) \n",
      "or estrogen (E2 gene set) or related to both hormones \n",
      "(P4 and E2 gene set). Gene targets of the nuclear proges -\n",
      "terone (PGR) and estrogen (ESR1, ESR2) receptors were \n",
      "added to their corresponding gene sets using DoRothEA \n",
      "(Discriminant Regulon Expression Analysis) database \n",
      "[26], considering only manually-curated or ChiP-Seq \n",
      "experimentally-validated gene-TF relationships. Finally, \n",
      "the P4 and E2 gene sets were independently mapped to \n",
      "each gene list.\n",
      "Meanwhile, to evaluate the non-hormonal regula -\n",
      "tors of endometrial progression and function, DoRo -\n",
      "thEA [26] and TarBase [27] databases were consulted to \n",
      "obtain TFs and miRNAs. DoRothEA was filtered as pre -\n",
      "viously described herein, while TarBase, which only con -\n",
      "tains miRNA-gene target relationships manually curated \n",
      "from publications or experimentally-validated in high-\n",
      "throughput datasets [27], was filtered by species (i.e., \n",
      "human). Finally, a functional over-representation analysis \n",
      "was carried out to identify which of the total annotated \n",
      "TFs or miRNAs were significantly associated with a par -\n",
      "ticular gene list (Supplementary Fig. S1B).\n",
      "To evaluate the relative contribution of each type of \n",
      "regulator (i.e., P4, E2, TFs, and miRNAs) on endometrial\n",
      "progression and function, three different comparisons \n",
      "were carried out applying Fisher’s exact tests. First, hor -\n",
      "monal regulators were evaluated using the relative pro -\n",
      "portion of P4- and E2-related genes included in each list, \n",
      "with respect to the total of number of genes founded in \n",
      "the P4 or E2 gene sets. Then, non-hormonal regulators \n",
      "were evaluated using the relative proportion of over-\n",
      "represented miRNAs and TFs with respect to the total of \n",
      "miRNAs and TFs with a at least one target in the gene \n",
      "list. Finally, the relative proportion of genes under hor -\n",
      "monal (i.e., P4 and/or E2-annotated genes) versus non-\n",
      "hormonal (i.e., miRNA and/or TF-annotated genes) \n",
      "regulation was evaluated considering the miRNAs/TFs \n",
      "found within the individual P4 or E2 gene sets as hormo -\n",
      "nal regulators.\n",
      "Prioritizing key regulators in endometrial progression \n",
      "and function\n",
      "We built regulatory networks to systemically analyse \n",
      "endometrial regulation. Nodes represented gene lists\n",
      "nal regulators.\n",
      "Prioritizing key regulators in endometrial progression \n",
      "and function\n",
      "We built regulatory networks to systemically analyse \n",
      "endometrial regulation. Nodes represented gene lists \n",
      "and regulators (miRNAs or TFs), while edges indicated \n",
      "significant enrichment among the lists and their corre -\n",
      "sponding regulators (false discovery rate (FDR) ≤ 0.05). \n",
      "To select the most influential regulators (i.e., those which \n",
      "targeted genes in most of our signatures), we studied the \n",
      "degree distribution of the networks (i.e., the number of \n",
      "gene lists regulated by each molecule) and prioritized the \n",
      "miRNAs and TFs which surpassed the relative maximum \n",
      "(i.e., 1.50 times the interquartile range (IQR)) number of \n",
      "relationships (gene list–regulators). All networks were \n",
      "built and analysed with Cytoscape version 3.7 [28] (Sup -\n",
      "plementary Fig. S1B).\n",
      "Dataset construction and processing for in‑silico validation\n",
      "To evaluate how the miRNAs’ expression change in the \n",
      "endometrium throughout the menstrual cycle, we used \n",
      "the raw data from GSE44558 GEO dataset [29], deriving \n",
      "from 20 endometrial samples collected throughout the \n",
      "menstrual cycle: four in early-proliferative (EPF), four \n",
      "in late-proliferative (LPF), four in early-secretory (ESE), \n",
      "four in mid-secretory (MSE) and four in late-secretory \n",
      "(LSE).\n",
      "To evaluate how the TFs’ expression change in the \n",
      "endometrium throughout the menstrual cycle, we \n",
      "analysed the integrated endometrial dataset previ -\n",
      "ously created by our group, and reported in [30], which \n",
      "compiled data from five prior publications (GSE98386, \n",
      "GSE29981, GSE4888, GSE119209 and GSE86491). \n",
      "This dataset included endometrial gene expression\n",
      "Page 4 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "data from 109 participants with normal endometrium, \n",
      "where biopsies were collected in proliferative (PF) \n",
      "(n = 29), ESE (n  = 29), MSE (n  = 43) and LSE (n  = 8) \n",
      "phases of the menstrual cycle (Table S1 ).\n",
      "Both TF and miRNA datasets were processed using \n",
      "the limma R-package [31]. Expression data were log \n",
      "transformed and quantile normalized, prior to explora -\n",
      "tory analyses which sought out possible outliers and \n",
      "batch effects. Relative gene expression ranges of low \n",
      "(1–10%), medium (11–50%), and high (51–100%) were \n",
      "established based on the expression of all the genes \n",
      "included in our dataset.\n",
      "Wet lab validation cohorts\n",
      "To corroborate the menstrual cycle-related expression \n",
      "changes observed with CTCF and GATA6 in our in-\n",
      "silico analysis, we conducted a prospective study as an \n",
      "external validation, using an independent sample set. \n",
      "We included a cohort of 19 healthy Spanish women \n",
      "(obtaining a total of 20 biopsies), between the ages of \n",
      "22–35 and with a body mass index of 22.80 ± 2.76  kg/\n",
      "m2. Endometrial samples were collected and staged in \n",
      "the menstrual cycle according to the follicle growth to \n",
      "control ovulation and LH levels in urine. Samples were \n",
      "grouped into PF (n  = 5), ESE (n  = 5), MSE (n  = 5) and \n",
      "LSE (n = 5) phases and categorized into two groups, PF/\n",
      "ESE and MSE/LSE.\n",
      "RNA extraction and RT‑qPCR\n",
      "Total RNA was extracted from the endometrial biopsy \n",
      "samples using the miRNeasy Mini Kit (Qiagen, Ger -\n",
      "many), and reverse transcribed into cDNA using the \n",
      "PrimeScript reagent kit (TAKARA, Japan). RT-qPCR \n",
      "reactions were carried out in duplicate, using fluores -\n",
      "cent Power-up SYBR Green (Thermo Fisher Scientific \n",
      "MA, USA) in a final volume of 10 µL. Primer sequences \n",
      "are shown in Table S2. Samples underwent 40 cycles of \n",
      "amplification, under standard conditions, using a Ste -\n",
      "pOnePlus™ System (Applied Biosystems, MA, USA). \n",
      "Relative mRNA expression was calculated using the  2−\n",
      "ΔΔCt method [32], and normalized to the expression of \n",
      "GAPDH housekeeping gene.\n",
      "Statistical analysis\n",
      "Fisher’s exact test was used to evaluate independent \n",
      "proportions. Mean expression changes across the cycle \n",
      "were studied using an analysis of variance (ANOVA) fol -\n",
      "lowed by a pairwise t-test or the Wilcoxon test when only \n",
      "two groups were available. P-values from multiple test \n",
      "comparisons were corrected by FDR and considered as \n",
      "significant when FDR ≤ 0.05. All statistical analyses were \n",
      "performed through R software (version 3.5) [33].\n",
      "Results\n",
      "Gene signatures associated to endometrial progression \n",
      "and function\n",
      "Of the 19 gene lists used for our analysis, eleven were \n",
      "obtained from published studies evaluating control \n",
      "patients throughout the menstrual cycle [4, 17–19, \n",
      "34–40], and the other eight were derived from studies \n",
      "comparing patients with RIF or unexplained infertility \n",
      "to controls [21,  41–47] (Table 1). Unifying all the afore -\n",
      "mentioned signatures, we compiled 3,608 genes related \n",
      "to endometrial progression and function.\n",
      "Hormonal regulation of endometrial progression is largely \n",
      "driven by progesterone\n",
      "We identified 7,540 and 698 genes related to estrogen \n",
      "and progesterone hormones, respectively. However, as \n",
      "determined by the relative contribution of each type of \n",
      "hormone within each gene list (Fig.  1A), 17/19 (89%) \n",
      "signatures favoured regulation by progesterone rather \n",
      "than estrogen. These differences were significant in \n",
      "47% of the signatures (FDR < 0.05). Exceptionally, the \n",
      "genes reported by the Altmae2010 and Koot signa -\n",
      "tures, that were related to unexplained infertility and \n",
      "RIF, had a preference for estrogen regulation (Fig.  1A). \n",
      "With respect to the proportion of progesterone-related \n",
      "genes, Altmae2010 showed 2.1 times more estrogen-\n",
      "related genes while Koot showed 1.7 times more. More -\n",
      "over, comparing the proportion of genes regulated by \n",
      "both hormones, between signatures revealed that Alt -\n",
      "related genes while Koot showed 1.7 times more. More -\n",
      "over, comparing the proportion of genes regulated by \n",
      "both hormones, between signatures revealed that Alt -\n",
      "mae2010 and Koot exhibited a lower hormone-depend -\n",
      "ent behaviour overall, with respect to the remaining \n",
      "signatures (Fig.  1B), highlighting the role of non-hor -\n",
      "monal regulators.\n",
      "Non‑hormonal regulation of endometrial progression \n",
      "and function\n",
      "We identified a total of 770 miRNAs and 555 TFs regu -\n",
      "lating at least one of the 3,608 genes reported in the \n",
      "19 gene lists. Further, we observed that 29% of the \n",
      "miRNAs and 44% of the TFs were associated with the \n",
      "ovarian hormones (i.e., progesterone and/or estrogen). \n",
      "Significantly more genes were targeted by hormo -\n",
      "nally-influenced regulators rather than non-hormonal \n",
      "regulators (P -value < 0.01, odds ratio =  91.94). Never -\n",
      "theless, we also identified 595 miRNAs (77.27%) and \n",
      "311 TFs (56.04%) that were unrelated to progesterone \n",
      "and/or estrogen, highlighting a substantial proportion \n",
      "of hormonally-independent regulators of the men -\n",
      "strual cycle.\n",
      "Page 5 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Table 1 Characteristics of the 19 gene lists included in this study. Gene lists were divided according to their evaluated condition. Each \n",
      "signature was labelled using the name of first author, followed by the publication year in the case of duplicates. Endometrial staging \n",
      "was based on different variables including urinary LH, histological analysis, ultrasound evaluation, or the number of days since the hCG \n",
      "trigger. Participants included healthy controls and women with infertility‑related conditions. Various platforms were used to evaluate \n",
      "gene expression. The thresholds applied for gene refinement and the final number of genes per signature are also presented\n",
      "Abbreviations: Δ delta  Ct, adj-P-value Adjusted P‑value, C Control group, d Day of menstrual cycle, EPF Early proliferative, ESE Early secretory, |FC| Absolute fold‑change, \n",
      "FC Fold change, HRT Hormone replacement therapy, IF Implantation failure, LH Luteinizing hormone, LH + X X days post LH surge, LPF Late proliferative, LSE Late‑\n",
      "secretory, M Miscarriage, MPF Mid‑proliferative, MSE Mid‑secretory, n number of samples, N/A Not applicable, OP Ongoing pregnancy, PF Proliferative, PFP Proportion \n",
      "of false positives, PR Pre‑receptive, R Receptive, RIF Recurrent implantation failure, RRA  Robust Rank Aggregation algorithm, SF Spontaneously fertile women, UI \n",
      "Women with unexplained infertility, vs versus\n",
      "Condition Gene signature Endometrial staging Participants Platform Threshold No. genes References\n",
      "Endometrial progres‑\n",
      "sion\n",
      "Altmäe2017 N/A N/A N/A RRA 57 [34]\n",
      "Borthwick Urinary LH; Histologi‑\n",
      "cally confirmed\n",
      "PF d9‑11 (n = 5) vs R \n",
      "LH + 6–8 (n = 5)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "N/A 116 [18]\n",
      "Carrascosa N/A N/A Quantitative RT‑PCR Based on literature 187 [35]\n",
      "Carson Urinary LH; Histologi‑\n",
      "cally confirmed\n",
      "ESE LH + 2–4 (n = \n",
      "3) vs MSE LH + 7–9 \n",
      "(n = 3)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "|FC|> 2 695 [17]\n",
      "Diaz‑Gimeno Urinary LH PR LH + 1–5 (n = \n",
      "15) vs R\n",
      "Agilent custom gene \n",
      "expression microarray\n",
      "|FC| > 2 234 [36]\n",
      "Kao Urinary LH; Histologi‑\n",
      "cally confirmed\n",
      "LPF d8‑10 (n = 4) vs \n",
      "MSE LH + 8‑10 (n = 7)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "|FC|> 2 and P‑value \n",
      "<0.05\n",
      "340 [37]\n",
      "Mirkin Urinary LH; Histologi‑\n",
      "cal dating\n",
      "ESE LH + 3 (n = 3) vs \n",
      "MSE LH + 8 (n = 5)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "|FC|>2 and adj‑P‑\n",
      "value < 0.05\n",
      "105 [19]\n",
      "Ponnampalam Histological dating EPF (n = 5) MPF (n = 7) \n",
      "LPF (n = 3) ESE (n = 7) \n",
      "MSE (n = 8) LS (n = 7) \n",
      "Mense (n = 6)\n",
      "Custom Adj‑P‑value <0.05 306 [38]\n",
      "Punyadeera Histological dating Mense (n = 2) vs LPF \n",
      "(n=2)\n",
      "Affymetrix HG‑U133A FC and P‑value 50 [39]\n",
      "Riesewijk Urinary LH; Histologi‑\n",
      "cal dating\n",
      "PR LH + 2 (n = 5) vs R \n",
      "LH + 7 (n = 5)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "|FC|> 3 in at least four \n",
      "out of five women.\n",
      "196 [40]\n",
      "Talbi Histological dating P (n = 6) vs ESE (n = 3) \n",
      "vs MSE (n = 8) vs LSE \n",
      "(n = 6)\n",
      "Affymetrix HG‑U133 \n",
      "Plus 2.0\n",
      "|FC|> 1.5 and adj‑P‑\n",
      "value < 0.05\n",
      "317 [4]\n",
      "|FC|> 1.5 and adj‑P‑\n",
      "value < 0.05\n",
      "317 [4]\n",
      "Implantation failure Altmäe2010 Urinary LH UI (n = 4) vs C (n = 5) Whole Human \n",
      "Genome Oligo \n",
      "Microarray (Agilent \n",
      "Technologies) \n",
      "|FC|> 2 and PFP < 0.05 184 [21]\n",
      "Bastu 6‑7 days after LH \n",
      "surge\n",
      "IF (n = 24) vs Fertile \n",
      "women (n = 24)\n",
      "Agilent‑039494\n",
      "SurePrint G3 Human\n",
      "GE v2 8x60K\n",
      "Log2(FC) ≥ 1 and adj‑\n",
      "P‑value ≤0.05 |FC|> 2 \n",
      "and PFP < 0.05\n",
      "524 [41]\n",
      "Bersinger N/A IF (n = 3) vs M (n = 3) \n",
      "vs OP (n = 3)\n",
      "Affymetrix Human \n",
      "Genome U133A 2.0 \n",
      "Chips\n",
      "Not specified 93 [42]\n",
      "Bhagwat N/A PR vs R Multiple platforms Based on literature 177 [43]\n",
      "Koot Urinary LH RIF (n = 43) vs C (n \n",
      "= 72)\n",
      "Human whole \n",
      "genome gene expres‑\n",
      "sion microarrays V2 \n",
      "(Agilent, Belgium)\n",
      "Machine learning \n",
      "predictor\n",
      "310 [44]\n",
      "Lédée Ultrasound evaluation IF (n = 30) vs Fertile \n",
      "women (n = 15)\n",
      "Affymetrix Human\n",
      "Genome U133 Plus \n",
      "2.0 Array\n",
      "Adj‑P‑value <0.01 \n",
      "and Δ > 1.35\n",
      "322 [45]\n",
      "Pathare 6‑7 days after hCG \n",
      "administration\n",
      "IF (n = 10) vs Healthy \n",
      "oocyte donors (n = 8)\n",
      "Illumina HumanHT‑12 \n",
      "V4.0 expression \n",
      "beadchip\n",
      "|FC|> 2 818 [46]\n",
      "Shi Urinary LH; Ultrasound \n",
      "evaluation\n",
      "Pathare 6‑7 days after hCG \n",
      "administration\n",
      "IF (n = 10) vs Healthy \n",
      "oocyte donors (n = 8)\n",
      "Illumina HumanHT‑12 \n",
      "V4.0 expression \n",
      "beadchip\n",
      "|FC|> 2 818 [46]\n",
      "Shi Urinary LH; Ultrasound \n",
      "evaluation\n",
      "RIF (n = 12) vs C (n \n",
      "= 10)\n",
      "Agilent‑052909\n",
      "CBC lncRNAmRNA V3\n",
      "|FC|≥ 2 and adj‑P‑\n",
      "value ≤ 0.05\n",
      "281 [47]\n",
      "Page 6 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "Of the respective miRNAs and TFs targeting genes \n",
      "related to endometrial progression and function, \n",
      "417 miRNAs and 467 TFs were significantly over-  \n",
      "represented in at least one gene list (FDR < 0.05, Tables \n",
      "S3-S4). Further, most signatures were significantly \n",
      "governed by TFs rather than miRNAs (4 > odds \n",
      "ratio > 164, FDR < 0.05, 17 lists [89%]), with the excep -\n",
      "tion of Koot’s signature that was governed by miRNAs \n",
      "and showed the opposite (odds ratio = 0.35, FDR < 0.05, \n",
      "Table S5).\n",
      "Fig. 1 Estrogen and progesterone‑mediated regulation of endometrial progression and function. A Eight out of nineteen gene lists (47%) \n",
      "had significant differences between the proportion of genes regulated by progesterone versus estrogen. (*FDR ≤ 0.05). B Descriptive analysis \n",
      "of hormonal regulation, normalized by the size of each gene list. Gene lists clustered according to higher (blue) or lower (orange) contribution \n",
      "of hormonal regulation\n",
      "Page 7 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Hsa‑miR‑15a‑5p, hsa‑miR‑128‑3p, hsa‑miR‑218‑5p, \n",
      "hsa‑miR‑27b‑3p, hsa‑miR‑107, hsa‑miR‑424‑5p, \n",
      "hsa‑miR‑195‑5p, hsa‑miR‑103a‑3p, has‑let‑7b‑5p \n",
      "and hsa‑miR‑22‑3p are key players in the regulation \n",
      "of endometrial progression and function\n",
      "To bridge the gap of endometrial-based evidence in \n",
      "DoRothEA and TarBase databases, we aimed to iden -\n",
      "tify the common miRNAs that mediate gene expres -\n",
      "sion related to endometrial progression and function. \n",
      "We evaluated the number of endometrial gene lists \n",
      "that each miRNA controlled (Fig.  2A). This strategy \n",
      "highlighted the most influential miRNAs that were \n",
      "commonly found across the endometrial gene lists. \n",
      "Notably, 85.80% of miRNAs regulated genes from one \n",
      "to three signatures, and only 59 miRNAs (14.20%) were \n",
      "associated with more than three signatures.  Ultimately, \n",
      "24 miRNAs were prioritized for regulating genes of at \n",
      "least six signatures.\n",
      "Next, we validated the gene expression trends for \n",
      "these 24 prioritized miRNAs in an independent endo -\n",
      "metrial gene expression dataset (GSE44730) (Fig.  2B). \n",
      "This dataset included the miRNA expression in endo -\n",
      "metrial biopsies from 20 healthy women in differ -\n",
      "ent phases of the menstrual cycle. We found that \n",
      "two of our prioritized miRNAs were not expressed \n",
      "in the endometrial tissue, and only 10 miRNAs (i.e., \n",
      "hsa-miR-15a-5p, hsa-miR-128-3p, hsa-miR-218-5p, \n",
      "hsa-miR-27b-3p, hsa-miR-107, hsa-miR-424-5p, hsa-\n",
      "miR-195-5p, hsa-miR-103a-3p, has-let-7b-5p and hsa-\n",
      "miR-22-3p) showed significant changes across the \n",
      "menstrual cycle (Fig.  2B). Of these, only the expression \n",
      "of hsa-miR-22-3p significantly increased from ESE \n",
      "to MSE stages (FDR = 0.02, Fig.  2B), highlighting the \n",
      "potential inhibitory role of this miRNA in endometrial \n",
      "function. The expression of the remaining miRNAs \n",
      "decreased during the MSE phase.\n",
      "CTCF and GATA6 as distinguished transcriptional \n",
      "regulators of the human menstrual cycle\n",
      "We repeated the regulation network analysis with the \n",
      "TFs, to identify the prominent TFs commonly regu -\n",
      "lating the genes related to endometrial progression \n",
      "and function in the reported signatures. We observed \n",
      "78.80% of the TFs regulated genes in up to 10 signa -\n",
      "tures, and only 99 TFs (21.20%) regulated genes in over \n",
      "10 (Fig.  3A). We highlighted CTCF (a CCCTC-binding \n",
      "factor, which functions as a transcriptomic repressor) \n",
      "as the TF with the most influential regulation of endo -\n",
      "metrial progression and function across 95% of studies \n",
      "(18 gene lists, excluding that of Punyadeera, Fig.  3A). \n",
      "Other distinguished TFs that regulated up to 15 gene \n",
      "lists included AR, CEBPA, CEBPB, CEBPD, CREB1, \n",
      "EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, \n",
      "GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, \n",
      "NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, \n",
      "TCF7L2, TFAP2C and TP53. Notably, GATA6 was \n",
      "selected due to novel association with endometrial pro -\n",
      "gression and function.\n",
      "gression and function.\n",
      "To infer the expression of CTCF and GATA6  in \n",
      "endometrial tissue throughout the menstrual cycle, \n",
      "we analysed their expression using a relevant endo -\n",
      "metrial dataset recently created by our group [30] \n",
      "(Table S1 ). We observed that, relative to the global \n",
      "expression of endometrial genes, CTCF was broadly \n",
      "expressed while GATA6  was moderately expressed. \n",
      "Indeed, CTCF was significantly down-regulated \n",
      "between the PF and ESE phases (FDR = 2.10E-03)  \n",
      "and ESE to the MSE phase (FDR = 2.10E-04), while \n",
      "GATA6 expression increased from the PF to the \n",
      "MSE phase (FDR = 1.70E-11) and from the ESE  \n",
      "to MSE phase (FDR = 9.80E-15). These findings \n",
      "supported the role of these TFs at the beginning \n",
      "of the menstrual cycle and during the MSE phase \n",
      "(Fig. 3 B).\n",
      "These in-silico  validations were then prospec -\n",
      "tively corroborated by RT-qPCR analysis in an inde -\n",
      "pendent sample set of 20 endometrial biopsies. We \n",
      "observed a similar expression profile of CTCF and \n",
      "GATA6 throughout the menstrual cycle, with signifi -\n",
      "pendent sample set of 20 endometrial biopsies. We \n",
      "observed a similar expression profile of CTCF and \n",
      "GATA6 throughout the menstrual cycle, with signifi -\n",
      "cant changes between the endometrial phases. Nota -\n",
      "bly, CTCF expression was decreased between the PF/\n",
      "ESE and MSE/LSE phases (FDR = 0.01) while GATA6 \n",
      "expression increased from the PF/ESE to MSE/LSE \n",
      "phases (FDR = 1.55E-04) (Fig. 3 C).\n",
      "Fig. 2 MicroRNAs universally regulating endometrial progression and function and their expression throughout the menstrual cycle. A Regulatory \n",
      "network of miRNAs forming relationships with numerous gene lists (2–12); miRNAs targeting specific gene lists were placed in the centre \n",
      "of the network. The prioritized miRNAs (outlined in red) included has‑miR‑16‑5p, has‑miR‑138‑5p, has‑miR‑21‑3p, has‑miR‑205‑5p, has‑miR‑27a‑5p, \n",
      "has‑miR‑15a‑5p, has‑miR‑155‑5p, has‑miR‑129–2‑3p, has‑miR‑147a, has‑miR‑146a‑5p, has‑miR‑107, has‑miR‑424‑5p, has‑miR‑203a‑3p, \n",
      "has‑miR‑195‑5p, has‑miR‑124‑3p, has‑miR‑128‑3p, has‑miR‑22‑3p, has‑miR‑27b‑3p, has‑let‑7b‑5p, has‑miR‑1343‑3p, has‑miR‑1‑3p, has‑miR‑103a‑3p, \n",
      "has‑miR‑23b‑3p, has‑miR‑218‑5p. B Expression profiles of the 10 prioritized miRNAs at different phases of the menstrual cycle. Significant changes \n",
      "between phases are denoted with asterisks (* FDR ≤ 0.05, ** FDR ≤ 0.01. Abbreviations: EPF, early proliferative phase; LPF, late proliferative phase; ESE, \n",
      "early‑secretory phase; MSE, mid‑secretory phase; LSE, late‑secretory phase\n",
      "(See figure on next page.)\n",
      "Page 8 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "Fig. 2 (See legend on previous page.)\n",
      "Page 9 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Fig. 3 Transcription factors universally regulating endometrial progression and function and their expression throughout the menstrual \n",
      "cycle. A Regulatory network of TFs forming relationships with numerous gene lists (2–18). TFs targeting specific gene lists were placed \n",
      "in the center of the network. CTCF stood out as the only TF that targeted genes belonging to 18/19 (95%) different signatures. B In-silico analysis \n",
      "of CTCF and GATA6 expression throughout the menstrual cycle. Significant changes between phases are marked with asterisks (**FDR ≤ 0.01, \n",
      "***FDR ≤ 0.001). C RT‑qPCR validation. Bar plot shows the relative mRNA expression of CTCF and GATA6 in endometrial biopsies collected \n",
      "throughout the menstrual cycle. Data is presented as a mean ± SD. Significant differences of CTCF (*FDR < 0.05) and GATA6 (***FDR < 0.001) \n",
      "expression were observed between proliferative/early‑secretory (PF/ESE) and mid‑secretory/late‑secretory phases (MSE/LSE). Abbreviations in (B \n",
      "and C): PF, Proliferative; ESE, early‑secretory; MSE, mid‑secretory; LSE, late‑secretory; SD, standard deviation\n",
      "Page 10 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "Discussion\n",
      "This data-driven approach exposed the common tran -\n",
      "scriptional regulators among 19 studies who proposed \n",
      "variable biomarkers of endometrial progression and \n",
      "function. In this study, we focused on understanding \n",
      "the relative contribution of both the hormonal and non-\n",
      "hormonal regulation, from an alternative holistic per -\n",
      "spective. We applied data-driven hypothesis research \n",
      "that, unlike the traditional scientific method, allowed us \n",
      "to generate new hypotheses based on all available bio -\n",
      "logically-relevant knowledge [16], observing the molec -\n",
      "ular relationships from a wider scale view. Besides, \n",
      "some traditional molecular procedures such as PCRs \n",
      "are also performed to corroborate the insight uncovered \n",
      "with this approach. We highlighted a larger influence \n",
      "of progesterone-related genes and TFs in endome -\n",
      "trial progression and observed a larger contribution \n",
      "of miRNAs and estrogen-related genes in endometrial \n",
      "pathology. Furthermore, we unveiled CTCF, GATA6, \n",
      "hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-\n",
      "miR-103a-3p and hsa-miR-128-3p as master regulators \n",
      "in the endometrium, and these findings were reinforced \n",
      "by the dynamic changes in their expression throughout \n",
      "the menstrual cycle (dually validated with in-silico  and \n",
      "prospective analyses).\n",
      "The menstrual cycle is perceived to be under tight \n",
      "regulation by the ovarian hormones, progesterone \n",
      "and estrogen [48]. Supporting the role of progester -\n",
      "one in the secretory endometrium [49], and reinforc -\n",
      "ing its indispensable role in endometrial regulation \n",
      "[50, 51], we found progesterone and its related genes \n",
      "had a larger influence in menstrual cycle progression \n",
      "than estrogen. As the RIF-related gene lists reported \n",
      "by Altmae2010 and Koot were generally controlled by \n",
      "estrogen, this further suggested that while progester -\n",
      "one could be crucial for a suitable endometrial progres -\n",
      "sion and mid-secretory phase acquisition, estrogenic \n",
      "irregularities may be more conducive to endometrial \n",
      "function and molecular pathology. Indeed, this hypoth -\n",
      "esis corroborated previous reports of most uterine dis -\n",
      "orders being estrogen-dependent [44, 52]. Supporting \n",
      "the findings from another recent study from our group, \n",
      "where the Altmae2010 and Koot signatures were classi -\n",
      "fied as highly-predictive signatures to identify endome -\n",
      "trial pathology in comparison with a prediction model \n",
      "that detected endometrial progression [7 ]. This study \n",
      "distinguished these signatures (related to implantation \n",
      "failure and unexplained infertility) based on the mag -\n",
      "nitude of their miRNA and estrogen-regulation, which \n",
      "contrasted with the prominent influence of the TFs over \n",
      "the remaining signatures more related to endometrial \n",
      "progression. All these facts reinforce the hypothesis \n",
      "that implantation failure is mainly caused by impaired \n",
      "miRNA expression [53].\n",
      "Although we observed a clear hormonal influence in\n",
      "numerous annotated regulators, this study transcended \n",
      "the simplistic paradigm of the bihormonal regulation of \n",
      "endometrial function by identifying a plethora of medi -\n",
      "ators, including 595 miRNAs and 311 TFs, that were \n",
      "previously unrelated to these ovarian hormones. These \n",
      "findings set the foundation for new discoveries, reveal -\n",
      "ing alternative pathways and new actors in the regula -\n",
      "tion of the human endometrium, and deepening our \n",
      "understanding of the complex regulatory mechanisms \n",
      "behind endometrial progression and unexplained \n",
      "infertility.\n",
      "MicroRNAs are well known to be involved in embryo \n",
      "implantation [12] and receptivity control [54, 55]. Using \n",
      "network analysis, we unveiled the 10 common miR -\n",
      "NAs regulating endometrial progression and function \n",
      "across published biomarker signatures (hsa-miR-22-3p, \n",
      "hsa-miR-107, hsa-miR-103a-3p, hsa-miR-128-3p, hsa-\n",
      "miR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, hsa-\n",
      "miR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p).\n",
      "hsa-miR-107, hsa-miR-103a-3p, hsa-miR-128-3p, hsa-\n",
      "miR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, hsa-\n",
      "miR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p). \n",
      "Their significant gene expression changes in the secre -\n",
      "tory endometrium support their potential role in the \n",
      "endometrial regulation through an inhibition during \n",
      "the WOI. Notably, hsa-miR-424-5p, hsa-miR-27b-3p \n",
      "and hsa-miR-195-5p have been previously associated \n",
      "with RIF [11, 54, 56], while has-let-7b-5p was associ -\n",
      "ated with human endometrial receptivity [57], and \n",
      "hsa-miR-22-3p was overexpressed in a decidualization \n",
      "model [58]. Moreover, hsa-miR-15a-5p was previously \n",
      "associated with endometriosis [59]. Notwithstanding, \n",
      "this study is the first to associate hsa-miR-15a-5p, hsa-\n",
      "miR-218-5p, hsa-miR-103a-3p, hsa-miR-107, and hsa-\n",
      "miR-128-3p to endometrial progression and functional \n",
      "regulation in humans.\n",
      "Due to their direct relationship with progesterone and \n",
      "estrogen [60], we expected to find TFs, such as ESR1  \n",
      "and PGR, involved in endometrial progression, how -\n",
      "ever, we additionally identified other TFs with an even \n",
      "larger influence in this process. For instance, STAT3  \n",
      "was associated with 17/19 signatures (FDR < 0.05), and \n",
      "is known to regulate uterine epithelial junctional reor -\n",
      "ganization and stromal proliferation, which are critical \n",
      "for implantation [61]. SP1, which similarly regulated \n",
      "17/19 signatures (FDR < 0.05), acts as a downstream \n",
      "paracrine target of progesterone to regulate estrogen \n",
      "inactivation, and could have a predominant role dur -\n",
      "ing the WOI [62, 63]. By focusing on the most univer -\n",
      "sal, overlapping TFs that were not previously related \n",
      "to implantation, we ultimately prioritized CTCF and \n",
      "GATA6 as novel key regulators of endometrial pro -\n",
      "gression, however, their specific molecular actions in\n",
      "Page 11 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "endometrial progression and function were beyond the \n",
      "scope of this project and merit further investigation.\n",
      "CTCF is a conserved zinc finger protein whose regu -\n",
      "latory functions are well characterized throughout the \n",
      "human body [64]. CTCF acts as a transcriptional repres -\n",
      "sor in RNA polymerase II (Pol II) pausing and imprint -\n",
      "ing and  X-chromosome inactivation [65], as well as an \n",
      "insulator, blocking the interaction between enhancers \n",
      "and the promoters of neighbouring genes [66]. Consider -\n",
      "ing these attributed functions and its variable expression \n",
      "throughout the menstrual cycle, we propose that CTCF \n",
      "exerts an inhibitory role in endometrial tissue during \n",
      "the PF phase of the menstrual cycle. With the significant \n",
      "downregulation of CTCF in the secretory endometrium, \n",
      "dually validated in-silico  and experimentally herein, its \n",
      "inhibited genes would be derepressed and become tran -\n",
      "scriptionally active during the WOI. This interpretation \n",
      "supports previous findings from our group demonstrat -\n",
      "ing that, during the WOI, a global transcriptional dere -\n",
      "pression may be required for implantation and early \n",
      "embryo development [63]. Transcriptional derepression \n",
      "has been associated with multiple human disease states \n",
      "and should be investigated further within the context \n",
      "of endometrial-factor infertility. Despite previous asso -\n",
      "ciations of CTCF with endometriosis [67], its implica -\n",
      "tion in endometrial progression and function has not \n",
      "been directly proposed until now. Nevertheless, the role \n",
      "of CTCF in endometrial receptivity has been revealed \n",
      "through its interaction with HOXA10, a gene mainly \n",
      "expressed in endometrium involved in functions such as \n",
      "endometrial proliferation and differentiation, the forma -\n",
      "tion of pinopodes or embryo implantation [68]. Indeed, \n",
      "this study proposed that an overexpression of CTCF can \n",
      "lead to a drop in HOXA10 expression, affecting endome -\n",
      "trial proliferation and endometrial function and corrobo-\n",
      "rating the fact that CTCF must decrease its expression as \n",
      "we can observe in our findings.\n",
      "GATA6, the second prioritized TF that overlapped \n",
      "across the gene lists, is a transcription factor belong -\n",
      "ing to the GATA family, a highly conserved family of \n",
      "six zinc finger proteins [69]. Besides its essential role in \n",
      "embryonic development [70], GATA6 plays a key role \n",
      "in endometriosis and regulates steroidogenic genes [71]. \n",
      "Together with evidence supporting both an activating \n",
      "and inhibitory role for GATA6 [72, 73], the significant \n",
      "changes we observed in its expression across the men -\n",
      "strual cycle reflect its presumed involvement in the endo-\n",
      "metrium and suggest an activating role in endometrial \n",
      "progression and endometrial receptivity acquisition. Like \n",
      "CTCF, this is the first time that GATA6 is proposed as a \n",
      "key TF in endometrial progression and function.\n",
      "Although the exact molecular mechanisms underlying\n",
      "endometrial regulation remain elusive, emerging exper -\n",
      "imental technologies such as High-throughput Chro -\n",
      "mosome Conformation Capture (Hi-C) or chromatin \n",
      "interaction analysis with paired-end tags (ChIA-PET) \n",
      "that respectively study general chromatin interactions \n",
      "or chromatin interactions involving a specific protein \n",
      "[74, 75] may improve our understanding of genetic \n",
      "regulation in the future. Nevertheless, as collaborative \n",
      "efforts unveil new regulators and gene targets in humans \n",
      "(such as the ENCODE project [76]), the complexities of \n",
      "transcriptional regulation can be exposed by in-silico  \n",
      "analyses like those described herein, and we can gain a \n",
      "better understanding of the processes involved in men -\n",
      "strual cycle progression and endometrial competence.\n",
      "Furthermore, it should be noted that results of gene \n",
      "expression studies are influenced by different variables, \n",
      "such as the experimental platforms employed, study \n",
      "designs, or patients’ characteristics of each independent\n",
      "expression studies are influenced by different variables, \n",
      "such as the experimental platforms employed, study \n",
      "designs, or patients’ characteristics of each independent \n",
      "study, adding variability [30, 77, 78]. Despite this fact, \n",
      "we have found that our holistic approach overcomes this \n",
      "undesired variability effect by finding common upstream \n",
      "master regulators across studies and predicting new \n",
      "relationships between hormones, TFs and miRNAs. \n",
      "This approach also provides the basis for future single-\n",
      "molecule studies that aim to elucidate new regulatory \n",
      "pathways in endometrial progression and function. Our \n",
      "results lay the foundation for further molecular studies \n",
      "that can validate the function(s) of the prevailing endo -\n",
      "metrial regulators we prioritized.\n",
      "Conclusion\n",
      "The study shows for the first time, the relative contri -\n",
      "bution of estrogens, progesterone, TFs and miRNAs in \n",
      "endometrial function and progression. Endometrial pro -\n",
      "gression is mainly influenced by progesterone-related \n",
      "genes and TFs, whereas miRNAs and estrogen-related \n",
      "genes play a larger role in endometrial pathology. \n",
      "Moreover, we highlight novel common transcriptional \n",
      "regulators such as CTCF, GATA6, hsa-miR-15a-5p, hsa-\n",
      "miR-218-5p, hsa-miR-107, hsa-miR-103a-3p, and hsa-\n",
      "miR-128-3p across 19 studies that propose biomarkers in \n",
      "endometrial regulation. These results reveal the molecu -\n",
      "lar mechanism underlying endometrial regulation and lay \n",
      "the foundation for the development of targeted therapies \n",
      "for patients with endometrial-factor infertility.\n",
      "Abbreviations\n",
      "ANOVA  Analysis of variance\n",
      "ChIA‑PET  Chromatin interaction analysis with paired‑end tags\n",
      "EPF  Early proliferative\n",
      "ESE  Early secretory\n",
      "FDR  False Discovery Rate\n",
      "GEO  Gene Expression Omnibus\n",
      "Page 12 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "GO  Gene Ontology\n",
      "HGNC  HUGO Gene Nomenclature Committee\n",
      "Hi‑C  High‑throughput Chromosome Conformation Capture\n",
      "IQR  Interquartile range\n",
      "KEGG  Kyoto Encyclopaedia of Genes and Genomes\n",
      "miRNAs  MicroRNAs\n",
      "MSE  Mid Secretory\n",
      "LPF  Late proliferative\n",
      "LSE  Late secretory\n",
      "RIF  Recurrent Implantation Failure\n",
      "Pol II  RNA polymerase II\n",
      "TFs  Transcription Factors\n",
      "WOI  Window of implantation\n",
      "Supplementary Information\n",
      "The online version contains supplementary material available at https:// doi. \n",
      "org/ 10. 1186/ s12958‑ 023‑ 01131‑4.\n",
      "Additional file 1: Supplementary Figure S1. Study design and \n",
      "workflow. Steps performed throughout the study. (A) We reviewed 19 \n",
      "publications related to endometrial progression and function, extracted \n",
      "the gene signatures from the original publication, and created corre‑\n",
      "sponding gene lists that we could annotate. (B) We performed different \n",
      "functional analyses based on the type of regulator (i.e., ovarian hormones, \n",
      "transcription factors and miRNAs). We used online databases to obtain the \n",
      "information of the regulatory process, at the hormonal (Gene Ontology, \n",
      "Kyoto Encyclopedia of Genes and Genomes, Dorothea), transcription \n",
      "factor (Dorothea) and miRNA (Tarbase) level and analysed the results \n",
      "accordingly. Finally, we employed an integrative analysis and prioritized \n",
      "the most relevant regulators for subsequent validation. TF, Transcription \n",
      "factors; miRNA, microRNA.\n",
      "Additional file 2: Supplementary Table S1. Characteristics of endo‑\n",
      "metrial transcriptomic datasets used to build the integrated dataset. ID, \n",
      "identifier of Gene Expression Omnibus database; Source, first author and \n",
      "year of the publication if it is available; No. samples; number of samples \n",
      "included; Age, range of patient age; Participants, reason for endome‑\n",
      "trial sample collection; Cycle Type, type of menstrual cycle of subjects \n",
      "included; Platform, platform used to measure gene expression; No. genes, \n",
      "number of genes measured in each dataset, Endometrial phase, number \n",
      "of samples in each phase of menstrual cycle. PF, proliferative; ESE, early \n",
      "secretory; MSE, mid secretory, LSE, late secretory. NA, Not Available.\n",
      "Additional file 3: Supplementary Table S2. Primer sequences for CTCF \n",
      "and GATA6. Forward and reverse primers used for RT‑qPCR validation of \n",
      "CTCF and GATA6 transcription factors.\n",
      "Additional file 4: Supplementary Table S3. Results of miRNA enrich‑\n",
      "ment. Excel file with the over‑representation analysis results of the micro‑\n",
      "RNAs (miRNAs). Each sheet contains the results of a specific list of genes. \n",
      "In each sheet, the following information is included: ID, gene symbol ID of \n",
      "the miRNA; GeneRatio, the proportion of miRNA targets in the gene list; \n",
      "BgRatio, the proportion of miRNA targets in the whole genome; OddsRa‑\n",
      "tio; the odd ratios for each miRNA, pvalue; the P‑value obtained for each \n",
      "miRNA; Padjust, P‑value adjusted by FDR for each miRNA; geneID, gene\n",
      "symbol ID of the miRNA targets.\n",
      "Additional file 5: Supplementary Table S4. Results of TFs enrichment. \n",
      "Excel file with the over‑representation analysis results of the transcription \n",
      "factors (TFs). Each sheet contains the results of a specific list of genes. In \n",
      "each sheet, the following information is included: ID, gene symbol of the \n",
      "TF; GeneRatio, the proportion of TF targets in the gene list; BgRatio, the \n",
      "proportion of TF targets in the whole genome; OddsRatio; the odd ratios \n",
      "for each TF, pvalue; the P‑value obtained for each TF; padjust, P‑value \n",
      "adjusted by FDR for each TF; geneID, the gene symbol ID of the TF targets.\n",
      "Additional file 6: Supplementary Table S5. Relative contribution of \n",
      "each regulator type (TFs or miRNAs) in each gene list. A Fisher’s test was \n",
      "performed to calculate the proportional differences between transcrip‑\n",
      "tion factors (TFs) and microRNAs (miRNAs). Resulting P‑values were \n",
      "adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene\n",
      "tion factors (TFs) and microRNAs (miRNAs). Resulting P‑values were \n",
      "adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene \n",
      "lists were mainly regulated by TFs, whereas odds ratios < 1 indicate a \n",
      "higher regulation by miRNAs. Odds Ratio with a value of “Inf” indicate \n",
      "that miRNAs were not found for this gene list. Percentages indicate the \n",
      "proportion between the significant regulators in each gene signature and \n",
      "regulators with at least one target gene in the corresponding signature. \n",
      "(***FDR ≤0.001).\n",
      "Acknowledgements\n",
      "The authors would like to thank Yassmin Medina‑Laver and Alicia Quiñonero \n",
      "for their advice and technical support during the experimental validation. \n",
      "Moreover, the authors thank Rosalba Lopez for her constructive criticism and \n",
      "editorial services in preparing the manuscript for publication.\n",
      "Author’s contribution\n",
      "P . D.‑G. conceived the original idea of this work and designed the study with the \n",
      "help of P .S.‑L. P .D.‑G. with the help of J.R. promoted the study and P .D.‑G. supervised \n",
      "it. All the in‑silico and experimental procedures were done by A.P .‑L. and supervised \n",
      "by P .D.‑G. In‑silico procedures were also supervised by P .S.‑L. Gene lists were col‑\n",
      "lected by A.P .‑L. with help from P .S.‑L. and A.D.‑P . miRNA dataset was collected and \n",
      "pre‑processed by A.P .‑L. with the help from P .S.‑L. and A.D.‑P . The collection of genes \n",
      "related to ovarian hormones was carried out by A.P .‑L. with the supervision of P .D.‑G. \n",
      "Statistical analyses were carried out by A.P .‑L. with the help of P .S.‑L. and supervised \n",
      "by P .D.‑G. miRNAs and TFs analysis was done by A.P .‑L. supervised by P .S.‑L. Networks \n",
      "and figures were performed by A.P .‑L., with the help of A.D.‑P . being supervised by \n",
      "P .D.‑G. Samples were collected by J.R. and N.P . Experimental validation was carried \n",
      "out by A.P .‑L. and D.M.‑G. and co‑supervised by F.D. and P .D‑G. Interpretation regard‑\n",
      "ing the results and their effect in the receptivity was carried out by A.P .‑L. and P .D.‑G. \n",
      "with the help of A.D.‑P . The manuscript was written by A.P .‑L., P .D.‑G., and A.D‑P , and \n",
      "was revised by all co‑authors (P .S.‑L., J.R., D.M.‑G., N.P . and F.D.).\n",
      "Funding\n",
      "This study was supported by Fundación IVI (1706‑FIVI‑048‑PD); the Instituto \n",
      "de Salud Carlos III (ISCIII) and co‑funded by the European Regional Develop‑\n",
      "ment Fund “A way to make Europe” (PI19/00537 [P .D.‑G]). Patricia Diaz‑Gimeno is \n",
      "supported by Instituto de Salud Carlos III (ISCIII) through Miguel Servet program \n",
      "(CP20/00118 [P .D.‑G]) and co‑funded by the European Union. Spanish Ministry of \n",
      "Science, Innovation and Universities (FPU18/01777 [A.P .‑L.], FPU15/01398 [A.D.‑P .], \n",
      "FPU19/03247 [D.M.‑G). Patricia Sebastian‑Leon is funded by Instituto de Salud \n",
      "Carlos III (ISCIII) through Sara Borrell program (CD21/00132[P .S.‑L]) and co‑funded \n",
      "by the European Union. Francisco Dominguez was supported by Instituto de \n",
      "Salud Carlos III(ISCIII) through Miguel Servet program CPII18/00002 and co‑funded\n",
      "by the European Union (Fonda Social Europeo), “El FSE invierte en tu futuro” .\n",
      "Availability of data and materials\n",
      "The data used to analyze the TF and miRNA expression throughout the men‑\n",
      "strual cycle and in infertility are available in Gene Expression Omnibus, with \n",
      "IDs GSE98386, GSE29981, GSE4888, GSE119209, GSE86491, and GSE44558.\n",
      "Declarations\n",
      "Ethics approval and consent to participate\n",
      "This research evaluated endometrial gene expression data from women throughout \n",
      "the menstrual cycle. Raw endometrial gene expression data (from previous studies) \n",
      "was downloaded from the publicly‑available functional genomic dataset repository \n",
      "Gene Expression Omnibus (GEO) [79] for curation/processing. In accordance with \n",
      "GEO policies, all patient data was anonymized and encrypted, and therefore an addi‑\n",
      "tional institutional review board approval was not required for our in-silico validation \n",
      "of regulator gene expression changes in endometrial tissue. On the other hand, for\n",
      "tional institutional review board approval was not required for our in-silico validation \n",
      "of regulator gene expression changes in endometrial tissue. On the other hand, for \n",
      "the experimental validation of gene expression changes in endometrial biopsies, \n",
      "all individuals gave written informed consent for endometrial gene expression \n",
      "evaluation prior to participating (Ethics Committee of the Instituto Valenciano de \n",
      "Infertilidad [IVI Valencia], Valencia, Spain [1810‑FIVI‑066‑PD/1706‑FIVI‑048‑PD]).\n",
      "Consent for publication\n",
      "Not applicable.\n",
      "Competing interests\n",
      "The authors declare no competing interests.\n",
      "Page 13 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Author details\n",
      "1 IVIRMA Global Research Alliance, IVI Foundation, Instituto de Investigación \n",
      "Sanitaria La Fe (IIS La Fe), Av. Fernando Abril Martorell 106, Torre A, Planta \n",
      "1ª, 46026 Valencia, Valencia, Spain. 2 Department of Pediatrics, Obstetrics \n",
      "and Gynaecology, Universidad de Valencia, Av. Blasco Ibáñez 15, 46010 Valen‑\n",
      "cia, Valencia, Spain. 3 IVIRMA Global Research Alliance, IVIRMA Valencia, Plaza de \n",
      "La Policia Local 3, 46015 Valencia, Spain. \n",
      "Received: 21 June 2023   Accepted: 22 August 2023\n",
      "References\n",
      " 1. Strowitzki T, Germeyer A, Popovici R, von Wolff M. The human endo‑\n",
      "metrium as a fertility‑determining factor. Hum Reprod Update. \n",
      "2006;12:617–30.\n",
      " 2. Cakmak H, Taylor HS. Implantation failure: Molecular mechanisms and \n",
      "clinical treatment. Hum Reprod Update. 2011;17:242–53.\n",
      " 3. Lessey BA, Young SL. What exactly is endometrial receptivity? Fertil Steril. \n",
      "2019;111:611–7.\n",
      " 4. Talbi S, Hamilton AE, Vo KC, Tulac S, Overgaard MT, Dosiou C, et al. \n",
      "Molecular phenotyping of human endometrium distinguishes menstrual \n",
      "cycle phases and underlying biological processes in normo‑ovulatory \n",
      "women. Endocrinology. 2006;147:1097–121.\n",
      " 5. Somigliana E, Vigano P , Busnelli A, Paffoni A, Vegetti W, Vercellini P . \n",
      "Repeated implantation failure at the crossroad between statistics, clinics \n",
      "and over‑diagnosis. Reprod Biomed Online. 2018;36:32–8.\n",
      " 6. Craciunas L, Gallos I, Chu J, Bourne T, Quenby S, Brosens JJ, et al. Con‑\n",
      "ventional and modern markers of endometrial receptivity: a systematic \n",
      "review and meta‑analysis. Hum Reprod Update. 2019;25:202–23.\n",
      " 7. Sebastian‑Leon P , Garrido N, Remohí J, Pellicer A, Diaz‑Gimeno P . Asyn‑\n",
      "chronous and pathological windows of implantation: Two causes of \n",
      "recurrent implantation failure. Hum Reprod. 2018;33:626–35.\n",
      " 8. Vasquez YM, DeMayo FJ. Role of nuclear receptors in blastocyst implanta‑\n",
      "tion. Semin Cell Dev Biol. 2013;24:724–35.\n",
      " 9. Wang H, Zhang S, Lin H, Kong S, Wang S, Wang H, et al. Physiological \n",
      "and molecular determinants of embryo implantation. Mol Aspects Med. \n",
      "2013;34:939–80.\n",
      " 10. Kelleher AM, Behura SK, Burns GW, Young SL, Demayo FJ, Spencer TE. Inte‑\n",
      "grative analysis of the forkhead box A2 (FOXA2) cistrome for the human \n",
      "endometrium. FASEB J. 2019;33:8543–54.\n",
      " 11. Revel A, Achache H, Stevens J, Smith Y, Reich R. MicroRNAs are \n",
      "associated with human embryo implantation defects. Hum Reprod. \n",
      "2011;26:2830–40.\n",
      " 12. Galliano D, Pellicer A. MicroRNA and implantation. Fertil Steril. \n",
      "2014;101:1531–44.\n",
      " 13. Xu H, Zhou M, Cao Y, Zhang D, Han M, Gao X, et al. Genome‑wide analysis \n",
      "of long noncoding RNAs, microRNAs, and mRNAs forming a compet‑\n",
      "ing endogenous RNA network in repeated implantation failure. Gene. \n",
      "2019;720: 144056.\n",
      " 14. Copley RR. The animal in the genome: comparative genomics and evolu‑\n",
      "tion. Philos Trans R Soc Lond B: Biol Sci. 2008;363:1453–61.\n",
      "15. Jacob F. The statue within: an autobiography. CSHL Press; 1995.\n",
      " 16. Yanai I, Lercher M. Night science. Genome Biol. 2019;20:179.\n",
      " 17. Carson DD, Lagow E, Thathiah A, Al‑Shami R, Farach‑Carson MC, Vernon \n",
      "M, et al. Changes in gene expression during the early to mid‑luteal \n",
      "(receptive phase) transition in human endometrium detected by high‑\n",
      "density microarray screening. Mol Hum Reprod. 2002;8:871–9.\n",
      " 18. Borthwick JM, Charnock‑Jones DS, Tom BD, Hull ML, Teirney R, Phillips SC, \n",
      "et al. Determination of the transcript profile of human endometrium. Mol \n",
      "Hum Reprod. 2003;9:19–33.\n",
      " 19. Mirkin S, Arslan M, Churikov D, Corica A, Diaz JI, Williams S, et al. In search \n",
      "of candidate genes critically expressed in the human endometrium dur‑\n",
      "ing the window of implantation. Hum Reprod. 2005;20:2104–17.\n",
      " 20. Devesa‑Peiro A, Sebastian‑Leon P , Parraga‑Leo A, Pellicer A, Diaz‑Gimeno \n",
      "P . Breaking the ageing paradigm in endometrium: endometrial gene \n",
      "expression related to cilia and ageing hallmarks in women over 35 years. \n",
      "Hum Reprod. 2022;37:762–76.\n",
      "P . Breaking the ageing paradigm in endometrium: endometrial gene \n",
      "expression related to cilia and ageing hallmarks in women over 35 years. \n",
      "Hum Reprod. 2022;37:762–76.\n",
      " 21. Altmae S, Martinez‑Conejero JA, Salumets A, Simon C, Horcajadas JA, \n",
      "Stavreus‑Evers A. Endometrial gene expression analysis at the time of \n",
      "embryo implantation in women with unexplained infertility. Mol Hum \n",
      "Reprod. 2010;16:178–87.\n",
      " 22. Devesa‑Peiro A, Sebastian‑Leon P , Pellicer A, Diaz‑Gimeno P . Guidelines for \n",
      "biomarker discovery in endometrium: correcting for menstrual cycle bias \n",
      "reveals new genes associated with uterine disorders. Mol Hum Reprod. \n",
      "2021;27:11.\n",
      " 23. Durinck S, Spellman PT, Birney E, Huber W. Mapping identifiers for the \n",
      "integration of genomic datasets with the R/ Bioconductor package \n",
      "biomaRt. Nat Protoc. 2009;4:1184–91.\n",
      " 24. Kanehisa M, Goto S. KEGG: Kyoto encyclopedia of genes and genomes. \n",
      "Nucleic Acids Res. 2000;27:27–30.\n",
      " 25. Ashburner M, Ball CA, Blake JA, Botstein D, Butler H, Cherry JM, et al. Gene \n",
      "ontology: tool for the unification of biology. Nat Genet. 2000;25:25–9.\n",
      " 26. Garcia‑Alonso L, Holland CH, Ibrahim MM, Turei D, Saez‑Rodriguez J. \n",
      "Benchmark and integration of resources for the estimation of human \n",
      "transcription factor activities. Genome Res. 2019;29:1363. https:// doi. org/ \n",
      "10. 1101/ gr. 240663. 118.\n",
      " 27. Karagkouni D, Paraskevopoulou MD, Chatzopoulos S, Vlachos IS, Tast‑\n",
      "soglou S, Kanellos I, et al. DIANA‑TarBase v8: a decade‑long collection of \n",
      "experimentally supported miRNA‑gene interactions. Nucleic Acids Res. \n",
      "2018;46:239–45.\n",
      " 28. Franz M, Lopes CT, Huck G, Dong Y, Sumer O, Bader GD. Cytoscape. \n",
      "js: a graph theory library for visualisation and analysis. Bioinformatics. \n",
      "2016;32:309–11.\n",
      " 29. Vilella F, Moreno‑Moya JM, Balaguer N, Grasso A, Herrero M, Martínez S, \n",
      "et al. Hsa‑miR‑30d, secreted by the human endometrium, is taken up \n",
      "by the pre‑implantation embryo and might modify its transcriptome. \n",
      "Development. 2015;142:3210–21.\n",
      " 30. Henarejos‑Castillo I, Sebastian‑Leon P , Devesa‑Peiro A, Pellicer A, Diaz‑\n",
      "Gimeno P . SARS‑CoV‑2 infection risk assessment in the endometrium: \n",
      "viral infection‑related gene expression across the menstrual cycle. Fertil \n",
      "Steril. 2020;114:223–32.\n",
      " 31. Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, et al. Limma powers \n",
      "differential expression analyses for RNA sequencing and microarray stud‑\n",
      "ies. Nucleic Acids Res. 2015;43:e47.\n",
      " 32. Livak KJ, Schmittgen TD. Analysis of relative gene expression data \n",
      "using real‑time quantitative PCR and the 2‑ΔΔCT method. Methods. \n",
      "2001;25:402–8.\n",
      " 33. R Development Core Team, R Core Team, R Development Core Team. A \n",
      "Language and Environment for Statistical Computing. R Foundation for \n",
      "Statistical Computing. 2018;2. https:// www. R‑‑ proje ct. org.\n",
      " 34. Altmäe S, Koel M, Võsa U, Adler P , Suhorutšenko M, Laisk‑Podar T, et al. \n",
      "Meta‑signature of human endometrial receptivity: a meta‑analysis and \n",
      "validation study of transcriptomic biomarkers. Scient Rep. 2017;7:10077.\n",
      "35. Carrascosa J, Horcajadas J, Moreno‑Moya JM. Chapter 15 ‑ the molecular \n",
      "signature of the endometrial receptivity: research and clinical application. \n",
      "In: Horcajadas JA, Gosálvez J, editors. Reproductomics: the omics revolu‑\n",
      "tion and its impact on human reproductive medicine. Academic Press; \n",
      "2018. p. 301–279.\n",
      " 36. Díaz‑Gimeno P , Horcajadas JA, Martínez‑Conejero JA, Esteban FJ, Alamá \n",
      "P , Pellicer A, et al. A genomic diagnostic tool for human endometrial \n",
      "receptivity based on the transcriptomic signature. Fertil Steril. 2011;95:50.\n",
      " 37. Kao LC, Tulac S, Lobo S, Imani B, Yang JP , Germeyer A, et al. Global gene \n",
      "profiling in human endometrium during the window of implantation. \n",
      "Endocrinology. 2002;143:2119–38.\n",
      " 38. Ponnampalam AP , Weston GC, Trajstman AC, Susil B, Rogers PAW. Molecu‑\n",
      "lar classification of human endometrial cycle stages by transcriptional \n",
      "profiling. Mol Hum Reprod. 2004;10:879–93.\n",
      " 39. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al.\n",
      "lar classification of human endometrial cycle stages by transcriptional \n",
      "profiling. Mol Hum Reprod. 2004;10:879–93.\n",
      " 39. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al. \n",
      "Oestrogen‑modulated gene expression in the human endometrium. Cell \n",
      "Mol Life Sci. 2005;62:239–50.\n",
      " 40. Riesewijk A, Martín J, van Os R, Horcajadas JA, Polman J, Pellicer A, et al. \n",
      "Gene expression profiling of human endometrial receptivity on days LH+2 \n",
      "versus LH+7 by microarray technology. Mol Hum Reprod. 2003;9:253–64.\n",
      " 41. Bastu E, Demiral I, Gunel T, Ulgen E, Gumusoglu E, Hosseini MK, et al. \n",
      "Potential marker pathways in the endometrium that may cause recurrent \n",
      "implantation failure. Reprod Sci. 2019;26:879–90.\n",
      " 42. Bersinger NA, Wunder DM, Birkhäuser MH, Mueller MD. Gene expression \n",
      "in cultured endometrium from women with different outcomes follow‑\n",
      "ing IVF. Mol Hum Reprod. 2008;14:475–84.\n",
      "Page 14 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "•\n",
      " \n",
      "fast, convenient online submission\n",
      " •\n",
      "  \n",
      "thorough peer review by experienced researchers in your ﬁeld\n",
      "• \n",
      " \n",
      "rapid publication on acceptance\n",
      "• \n",
      " \n",
      "support for research data, including large and complex data types\n",
      "•\n",
      "  \n",
      "gold Open Access which fosters wider collaboration and increased citations \n",
      " \n",
      "maximum visibility for your research: over 100M website views per year •\n",
      "  At BMC, research is always in progress.\n",
      "Learn more biomedcentral.com/submissions\n",
      "Ready to submit y our researc hReady to submit y our researc h  ?  Choose BMC and benefit fr om: ?  Choose BMC and benefit fr om: \n",
      " 43. Bhagwat SR, Chandrashekar DS, Kakar R, Davuluri S, Bajpai AK, Nayak S, \n",
      "et al. Endometrial receptivity: a revisit to functional genomics studies \n",
      "on human endometrium and creation of HGEx ‑ERdb. PLoS ONE. 2013;8: \n",
      "e58419.\n",
      " 44. Koot YEM, Van Hooff SR, Boomsma CM, Van Leenen D, Koerkamp \n",
      "MJAG, Goddijn M, et al. An endometrial gene expression signature \n",
      "accurately predicts recurrent implantation failure after IVF. Sci Rep. \n",
      "2015;2016(6):1–12.\n",
      " 45. Lédée N, Munaut C, Aubert J, Sérazin V, Rahmati M, Chaouat GG, et al. \n",
      "Specific and extensive endometrial deregulation is present before \n",
      "conception in IVF/ICSI repeated implantation failures (IF) or recurrent \n",
      "miscarriages. J Pathol. 2011;225:554–64.\n",
      " 46. Pathare ADS, Zaveri K, Hinduja I. Downregulation of genes related to \n",
      "immune and inflammatory response in IVF implantation failure cases under \n",
      "controlled ovarian stimulation. Am J Reprod Immunol. 2017;78: e12679.\n",
      " 47. Shi C, Han HJ, Fan LJ, Guan J, Zheng XB, Chen X, et al. Diverse endometrial \n",
      "mRNA signatures during the window of implantation in patients with \n",
      "repeated implantation failure. Hum Fertil. 2018;21:183–94.\n",
      " 48. Messinisi IE. Ovarian feedback, mechanism of action and possible clinical \n",
      "implications. Hum Reprod Update. 2006;12:557–71.\n",
      " 49. Edwards RG. Implantation, interception and contraception. Hum Reprod \n",
      "(Oxford, England). 1994;9:985–95.\n",
      " 50. Young SL. Oestrogen and progesterone action on endometrium: a \n",
      "translational approach to understanding endometrial receptivity. Reprod \n",
      "Biomed Online. 2013;27:497–505.\n",
      " 51. Bhurke AS, Bagchi IC, Bagchi MK. Progesterone‑regulated endometrial \n",
      "factors controlling implantation. Am J Reprod Immunol. 2016;75:237–45.\n",
      " 52. Devesa‑Peiro A, Sebastian‑Leon P , Garcia‑ Garcia F, Arnau V, Aleman \n",
      "A, Pellicer A, et al. Uterine disorders affecting female fertility: what \n",
      "are the molecular functions altered in endometrium? Fertil Steril. \n",
      "2020;113:1261–74.\n",
      " 53. von Grothusen C, Frisendahl C, Modhukur V, Lalitkumar PG, Peters M, \n",
      "Faridani OR, et al. Uterine fluid microRNAs are dysregulated in women \n",
      "with recurrent implantation failure. Hum Reprod. 2022;37:734–46.\n",
      " 54. Altmäe S, Martinez‑Conejero JA, Esteban FJ, Ruiz‑Alonso M, Stavreus‑\n",
      "Evers A, Horcajadas JA, et al. MicroRNAs miR‑30b, miR‑30d, and miR‑494\n",
      "regulate human endometrial receptivity. Reprod Sci. 2013;20:308–17.\n",
      " 55. Di Pietro C, Caruso S, Battaglia R, IraciSareri M, La Ferlita A, Strino F, et al. \n",
      "MiR‑27a‑3p and miR‑124–3p, upregulated in endometrium and serum \n",
      "from women affected by Chronic Endometritis, are new potential \n",
      "molecular markers of endometrial receptivity. Am J Reprod Immunol. \n",
      "2018;80:12858.\n",
      " 56. Shi C, Shen H, Fan L ‑ JJ, Guan J, Zheng X ‑BB, Chen X, et al. Endometrial \n",
      "MicroRNA signature during the window of implantation changed in \n",
      "patients with repeated implantation failure. Chin Med J. 2017;130:566–73.\n",
      " 57. La Ferlita A, Battaglia R, Andronico F, Caruso S, Cianci A, Purrello M, \n",
      "et al. Non‑coding RNAs in endometrial physiopathology. Int J Mol Sci. \n",
      "2018;19:2120.\n",
      " 58. Estella C, Herrer I, Moreno‑Moya JM, Quiñonero A, Martínez S, Pellicer A, \n",
      "et al. miRNA signature and dicer requirement during human endometrial \n",
      "stromal decidualization in vitro. PLoS ONE. 2012;7:41080.\n",
      " 59. Liu D, Liang Y, Chen M, Yang F, Yao S. Knockdown of circ_0075503 sup ‑\n",
      "stromal decidualization in vitro. PLoS ONE. 2012;7:41080.\n",
      " 59. Liu D, Liang Y, Chen M, Yang F, Yao S. Knockdown of circ_0075503 sup ‑\n",
      "presses cell migration and invasion by regulating miR‑15a‑5p and KLF12 \n",
      "in endometriosis. Mol Cell Biochem. 2021;476:3845–56.\n",
      " 60. Chi RPA, Wang T, Adams N, Wu SP , Young SL, Spencer TE, et al. Human \n",
      "endometrial transcriptome and progesterone receptor cistrome reveal \n",
      "important pathways and epithelial regulators. J Clin Endocrinol Metab. \n",
      "2020;105:1–21.\n",
      " 61. Pawar S, Starosvetsky E, Orvis GD, Behringer RR, Bagchi IC, Bagchi MK. \n",
      "STAT3 regulates uterine epithelial remodeling and epithelial‑stromal \n",
      "crosstalk during implantation. Mol Endocrinol. 2013;27:1996–2012.\n",
      " 62. Cheng YH, Imir A, Suzuki T, Fenkci V, Yilmaz B, Sasano H, et al. SP1 and SP3 \n",
      "mediate progesterone‑ dependent induction of the 17beta hydroxyster ‑\n",
      "oid dehydrogenase type 2 gene in human endometrium. Biol Reprod. \n",
      "2006;75:605–14.\n",
      " 63. Sebastian‑Leon P , Devesa‑Peiro A, Aleman A, Parraga‑Leo A, Arnau V, \n",
      "Pellicer A, et al. Transcriptional changes through menstrual cycle reveal a \n",
      "global transcriptional derepression underlying the molecular mechanism \n",
      "involved in the window of implantation. Mol Hum Reprod. 2021. https:// \n",
      "doi. org/ 10. 1093/ molehr/ gaab0 27.\n",
      " 64. Kim S, Yu NK, Kaang BK. CTCF as a multifunctional protein in genome \n",
      "regulation and gene expression. Exp Mol Med. 2015;47: e166.\n",
      " 65. Shukla S, Kavak E, Gregory M, Imashimizu M, Shutinoski B, Kashlev M, \n",
      "et al. CTCF‑promoted RNA polymerase II pausing links DNA methylation \n",
      "to splicing. Nature. 2011;479:74–9.\n",
      " 66. Phillips JE, Corces VGCTCF. Master weaver of the genome. Cell. \n",
      "2009;137:1194–211.\n",
      " 67. Guo J, Cao B, Xu X, Wu F, Zhu B. Novel CTCF mutations in Chinese \n",
      "patients with ovarian endometriosis. Mol Med Rep. 2018;18:1031–6.\n",
      " 68. Geng J, Cui C, Yin Y, Zhao Y, Zhang C. LncRNA NEAT1 affects endome ‑\n",
      "trial receptivity by regulating HOXA10 promoter activity. Cell Cycle. \n",
      "2022;21:1932–44.\n",
      " 69. Viger RS, Guittot SM, Anttonen M, Wilson DB, Heikinheimo M. Role of the \n",
      "GATA family of transcription factors in endocrine development, function, \n",
      "and disease. Mol Endocrinol. 2008;22:781.\n",
      " 70. Xuan S, Sussel L. GATA4 and GATA6 regulate pancreatic endoderm \n",
      "identity through inhibition of hedgehog signaling. Development (Cam‑\n",
      "bridge). 2016;143:780–6.\n",
      " 71. Bernardi LA, Dyson MT, Tokunaga H, Sison C, Oral M, Robins JC, et al. The \n",
      "essential role of GATA6 in the activation of estrogen synthesis in endome‑\n",
      "triosis. Reprod Sci. 2019;26:60.\n",
      " 72. Yin F, Herring BP . GATA‑6 can act as a positive or negative regula‑\n",
      "tor of smooth muscle‑specific gene expression *. J Biol Chem. \n",
      "2005;280:4745–52.\n",
      " 73. Zhai J, Luo G. GATA6‑induced FN1 activation promotes the proliferation, \n",
      "invasion and migration of oral squamous cell carcinoma cells. Mol Med \n",
      "Rep. 2022;25:102.\n",
      " 74. Fullwood MJ, Liu MH, Pan YF, Liu J, Xu H, Mohamed YB, et al. An \n",
      "oestrogen‑receptor‑α‑bound human chromatin interactome. Nature. \n",
      "2009;462:58–64.\n",
      "2009;462:58–64.\n",
      " 75. Lieberman‑Aiden E, Van Berkum NL, Williams L, Imakaev M, Ragoczy \n",
      "T, Telling A, et al. Comprehensive mapping of long range interactions \n",
      "reveals folding principles of the human genome. Science. 2009;326:289.\n",
      " 76. Birney E, Stamatoyannopoulos JA, Dutta A, Guigó R, Gingeras TR, \n",
      "Margulies EH, et al. Identification and analysis of functional elements \n",
      "in 1% of the human genome by the ENCODE pilot project. Nature. \n",
      "2007;447:799–816.\n",
      " 77. Tajti F, Kuppe C, Antoranz A, Ibrahim MM, Kim H, Ceccarelli F, et al. A func‑\n",
      "tional landscape of CKD entities from public transcriptomic data. Kidney \n",
      "Int Rep. 2020. https:// doi. org/ 10. 1016/j. ekir. 2019. 11. 005.\n",
      " 78. Henarejos‑ Castillo I, Devesa‑Peiro A, de Miguel‑ Gomez L, Sebastian‑Leon \n",
      "P , Romeu M, Aleman A, et al. Predicted COVID ‑19 molecular effects on \n",
      "endometrium reveal key dysregulated genes and functions. Mol Hum \n",
      "Reprod. 2022;28:gaac035.\n",
      " 79. Barrett T, Wilhite SE, Ledoux P , Evangelista C, Kim IF, Tomashevsky M, et al.\n",
      "endometrium reveal key dysregulated genes and functions. Mol Hum \n",
      "Reprod. 2022;28:gaac035.\n",
      " 79. Barrett T, Wilhite SE, Ledoux P , Evangelista C, Kim IF, Tomashevsky M, et al. \n",
      "NCBI GEO: archive for functional genomics data sets–update. Nucleic \n",
      "Acids Res. 2013;41(Database issue):D991‑5.\n",
      "Publisher’s Note\n",
      "Springer Nature remains neutral with regard to jurisdictional claims in pub‑\n",
      "lished maps and institutional affiliations.\n"
     ]
    }
   ],
   "source": [
    "my_json = []\n",
    "\n",
    "for docs in storage:\n",
    "    print(docs.page_content)\n",
    "\n",
    "    my_json.append(chain.invoke({\"query\": docs.page_content}))\n",
    "\n",
    "\n",
    "#chain.invoke({\"query\": storage[0].page_content})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "19cb09a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'text': 'A total of 3,608 distinct genes from the 19 gene lists were associated with endometrial progression',\n",
       "  'doi': '10.1186/s12958‑023‑01131‑4',\n",
       "  'citation': 'Parraga‑Leo et al. (2023)'},\n",
       " {'text': 'The lists’ regulation was significantly favoured by TFs (89% (17/19) of gene lists) and progesterone (47% (8 /19) of gene lists), rather than miRNAs (5% (1/19) of gene lists) or estrogen (0% (0/19) of gene lists), respectively (FDR < 0.05). Exceptionally, two gene lists that were previously associated with implantation failure and unexplained infertility were less hormone‑dependent, but primarily regulated by estrogen. Although endometrial progression genes were mainly targeted by hormones rather than non‑hormonal contributors (odds ratio = 91.94, FDR < 0.05), we identified 311 TFs and 595 miRNAs not previously associated with ovarian hormones. We highlight CTCF, GATA6, hsa‑miR‑15a‑5p, hsa‑miR‑218‑5p, hsa‑miR‑107, hsa‑miR‑103a‑3p, and hsa‑miR‑128‑3p, as overlapping novel master regulators of endometrial function. The gene expression changes of selected regulators through‑out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their potential regulatory roles in the endometrium.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'This study revealed novel hormonal and non‑hormonal regulators and their relative contributions to endometrial progression and pathology, providing new leads for the potential causes of endometrial‑factor infertility. The endometrium is the innermost layer of the uterus, which undergoes dynamic histological, physiological and molecular changes that allow it to synchronize with embryo development, facilitate embryo implantation, and ultimately, establish a successful pregnancy. Implantation is a crucial and complex limiting step for conception which occurs between days 19 to 24 of a normal menstrual cycle. Consequently, this period of time is called the window of implantation (WOI) and is characterized by abrupt transcriptomic changes in the endometrial tissue. Alterations in either WOI establishment and endometrial progression may lead to implantation failures, which, along with biochemical miscarriage, account for more than 50% of pregnancy losses at pre-clinical stages. These alterations have broadly been classified as being related to a displaced WOI, caused by variable timing of endometrial progression, or a disrupted WOI, where impaired endometrial function prevents the establishment of an effective WOI. Initially, the WOI was postulated to mainly be regu -lated by the ovarian hormones (namely, estrogen and progesterone) acting via their respective nuclear receptors (i.e., ESR1/2 and PGR). However, evidence has shown that these receptors cooperate with other transcription factors (TFs) and co-regulators to mediate uterine physiology [8]. In fact, TFs (such as homeobox TFs, FOXA2, and KLF9) have recently been reported as key regulators for the establishment of endometrial receptivity [9, 10], and microRNAs (miRNAs) have been found to similarly intervene in this complex and multifactorial process as transcriptional regulators [11]. While the involvement of both types of regulators was widely reported in embryo implantation [12] and other infertility-related diseases, including recurrent implantation failure (RIF) [13], the complex interactions between the different types of regulators and their respective contributions to endometrial progression and function have not been investigated from a systemic point of view. This paradigm shift approaches genes and regulators in an integrative way, by considering how they interact with and coordinate each other to carry out cellular processes, rather than reinforcing the premise they work independently [14]. By more accurately reflecting the biological milieu, systemic approaches take gene-based discoveries to the next level.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Previous studies of endometrial transcriptomics have been used to define the WOI, compare the endometrium of healthy women as their menstrual cycles progressed [4, 17–19]; address endometrial differences related to age [20] or the dysfunction of the WOI between patients with implantation failure and healthy controls [21]; and finally, correct for the menstrual cycle bias that can mask important biomarkers [22]. Notwithstanding, the candidate biomarkers of endometrial receptivity reported in each of the aforementioned studies overlap poorly [7]. Indeed, recent comparisons conducted by Sebastian-Leon and colleagues [7] found no congruities between 16 different reported gene lists of endometrial receptivity, and only fair to moderate functional agreements between some of the included signatures. Despite previous transcriptional regulation studies, there have been no holistic studies of endometrial receptivity.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'fair to moderate functional agreements between some of the included signatures. Despite previous transcriptional regulation studies, there have been no holistic studies of this regulatory process, which robustly analyse the drivers of transcription across a higher number of studies (that each searched for endometrial receptivity biomarkers) and map the relative contributions of each type of regulator to endometrial progression regulation (Ovarian hormones [i.e., estrogen, progesterone], TFs, miRNAs). Thus, the aim of this study was to utilize available genomic data regarding transcriptional regulators to identify overlapping mediators of endometrial transcriptional regulation (i.e., ovarian hormones, TFs, and miRNAs) and determine the relative contribution of each type of regulator among the previously reported gene signatures. These findings could help unveil the master regulators and principal type of endometrial regulation, paving the way for further research aimed at improving woman´s reproductive health. Annotating gene lists associated with endometrial progression and function. Endometrial progression and implantation failure gene lists [7] were retrieved and updated using a public data repository (i.e., Gene Expression Omnibus (GEO)).',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'To study the hormonal regulation of endometrial progression, the Kyoto Encyclopaedia of Genes and Genomes (KEGG) [24], and Gene Ontology (GO) [25] databases were consulted for ovarian hormone-related genes, using relevant keywords (i.e., progesterone, estrogen, oestrogen and estradiol) and selecting those pathways and functions that contained them (Supplementary Fig. S1B). Genes associated with the obtained human KEGG pathways and GO functions were grouped according to whether they were unique to progesterone (P4 gene set) or estrogen (E2 gene set) or related to both hormones (P4 and E2 gene set). Gene targets of the nuclear progesterone (PGR) and estrogen (ESR1, ESR2) receptors were added to their corresponding gene sets using DoRothEA (Discriminant Regulon Expression Analysis) database [26], considering only manually-curated or ChiP-Seq experimentally-validated gene-TF relationships. Finally, the P4 and E2 gene sets were independently mapped to each gene list.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Prioritizing key regulators in endometrial progression and function\\nWe built regulatory networks to systemically analyse endometrial regulation.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'We built regulatory networks to systemically analyse endometrial regulation. Nodes represented gene lists and regulators (miRNAs or TFs), while edges indicated significant enrichment among the lists and their corresponding regulators (false discovery rate (FDR) ≤ 0.05). To select the most influential regulators (i.e., those which targeted genes in most of our signatures), we studied the degree distribution of the networks (i.e., the number of gene lists regulated by each molecule) and prioritized the miRNAs and TFs which surpassed the relative maximum (i.e., 1.50 times the interquartile range (IQR)) number of relationships (gene list–regulators). All networks were built and analysed with Cytoscape version 3.7 [28] (Supplementary Fig. S1B). To evaluate how the miRNAs’ expression change in the endometrium throughout the menstrual cycle, we used the raw data from GSE44558 GEO dataset [29], deriving from 20 endometrial samples collected throughout the menstrual cycle: four in early-proliferative (EPF), four in late-proliferative (LPF), four in early-secretory (ESE), four in mid-secretory (MSE) and four in late-secretory (LSE). To evaluate how the TFs’ expression change in the endometrium throughout the menstrual cycle, we analysed the integrated endometrial dataset previously created by our group, and reported in [30], which compiled data from five prior publications (GSE98386, GSE29981, GSE4888, GSE119209 and GSE86491). This dataset included endometrial gene expression',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Of the 19 gene lists used for our analysis, eleven were obtained from published studies evaluating control patients throughout the menstrual cycle [4, 17–19, 34–40], and the other eight were derived from studies comparing patients with RIF or unexplained infertility to controls [21, 41–47] (Table 1). Unifying all the afore-',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'Table 1'},\n",
       " {'text': 'mentioned signatures, we compiled 3,608 genes related to endometrial progression and function. Hormonal regulation of endometrial progression is largely driven by progesterone. We identified 7,540 and 698 genes related to estrogen and progesterone hormones, respectively. However, as determined by the relative contribution of each type of hormone within each gene list (Fig. 1A), 17/19 (89%) signatures favoured regulation by progesterone rather than estrogen. These differences were significant in 47% of the signatures (FDR < 0.05). Exceptionally, the genes reported by the Altmae2010 and Koot signatures, that were related to unexplained infertility and RIF, had a preference for estrogen regulation (Fig. 1A). With respect to the proportion of progesterone-related genes, Altmae2010 showed 2.1 times more estrogen-related genes while Koot showed 1.7 times more. More over, comparing the proportion of genes regulated by both hormones, between signatures revealed that Alt -',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'We identified a total of 770 miRNAs and 555 TFs regulating at least one of the 3,608 genes reported in the 19 gene lists. Further, we observed that 29% of the miRNAs and 44% of the TFs were associated with the ovarian hormones (i.e., progesterone and/or estrogen). Significantly more genes were targeted by hormonally-influenced regulators rather than non-hormonal regulators (P -value < 0.01, odds ratio =  91.94). Nevertheless, we also identified 595 miRNAs (77.27%) and 311 TFs (56.04%) that were unrelated to progesterone and/or estrogen, highlighting a substantial proportion of hormonally-independent regulators of the menstrual cycle.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Table 1 Characteristics of the 19 gene lists included in this study. Gene lists were divided according to their evaluated condition. Each signature was labelled using the name of first author, followed by the publication year in the case of duplicates. Endometrial staging was based on different variables including urinary LH, histological analysis, ultrasound evaluation, or the number of days since the hCG trigger. Participants included healthy controls and women with infertility‑related conditions. Various platforms were used to evaluate gene expression. The thresholds applied for gene refinement and the final number of genes per signature are also presented\\nAbbreviations: Δ delta  Ct, adj-P-value Adjusted P‑value, C Control group, d Day of menstrual cycle, EPF Early proliferative, ESE Early secretory, |FC| Absolute fold‑change, FC Fold change, HRT Hormone replacement therapy, IF Implantation failure, LH Luteinizing hormone, LH + X X days post LH surge, LPF Late proliferative, LSE Late‑secretory, M Miscarriage, MPF Mid‑proliferative, MSE Mid‑secretory, n number of samples, N/A Not applicable, OP Ongoing pregnancy, PF Proliferative, PFP Proportion of false positives, PR Pre‑receptive, R Receptive, RIF Recurrent implantation failure, RRA Robust Rank Aggregation algorithm, SF Spontaneously fertile women, UI Women with unexplained infertility, vs versus\\nCondition Gene signature Endometrial staging Participants Platform Threshold No. genes References\\nEndometrial progres‑ sion\\nAltmäe2017 N/A N/A N/A RRA 57 [34]\\nBorthwick Urinary LH; Histologi‑ cally confirmed PF d9‑11 (n = 5) vs R LH + 6–8 (n = 5)\\nAffymetrix Genechip Hu95A N/A 116 [18]\\nCarrascosa N/A N/A Quantitative RT‑PCR Based on literature 187 [35]\\nCarson Urinary LH; Histologically confirmed ESE LH + 2–4 (n = 3) vs MSE LH + 7–9 (n = 3)\\nAffymetrix Genechip Hu95A |FC|> 2 695 [17]\\nDiaz‑Gimeno Urinary LH PR LH + 1–5 (n = 15) vs R\\nAgilent custom gene expression microarray |FC| > 2 234 [36]\\nKao Urinary LH; Histologically confirmed LPF d8‑10 (n = 4) vs MSE LH + 8‑10 (n = 7)\\nAffymetrix Genechip Hu95A |FC|> 2 and P‑value <0.05 340 [37]\\nMirkin Urinary LH; Histological dating ESE LH + 3 (n = 3) vs MSE LH + 8 (n = 5)\\nAffymetrix Genechip Hu95A |FC|>2 and adj‑P‑value < 0.05 105 [19]\\nPonnampalam Histological dating EPF (n = 5) MPF (n = 7) LPF (n = 3) ESE (n = 7) MSE (n = 8) LS (n = 7) Mense (n = 6)\\nCustom Adj‑P‑value <0.05 306 [38]\\nPunyadeera Histological dating Mense (n = 2) vs LPF (n=2)\\nAffymetrix HG‑U133A FC and P‑value 50 [39]\\nRiesewijk Urinary LH; Histological dating PR LH + 2 (n = 5) vs R LH + 7 (n = 5)\\nAffymetrix Genechip Hu95A |FC|> 3 in at least four out of five women. 196 [40]\\nTalbi Histological dating P (n = 6) vs ESE (n = 3) vs MSE (n = 8) vs LSE (n = 6)\\nAffymetrix HG‑U133 Plus 2.0 |FC|> 1.5 and adj‑P‑value < 0.05 317 [4]',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Implantation failure Altmäe2010 Urinary LH UI (n = 4) vs C (n = 5) Whole Human Genome Oligo Microarray (Agilent Technologies) |FC|> 2 and PFP < 0.05 184 [21] Bastu 6‑7 days after LH surge IF (n = 24) vs Fertile women (n = 24) Agilent‑039494 SurePrint G3 Human GE v2 8x60K Log2(FC) ≥ 1 and adj‑P‑value ≤0.05 |FC|> 2 and PFP < 0.05 524 [41] Bersinger N/A IF (n = 3) vs M (n = 3) vs OP (n = 3) Affymetrix Human Genome U133A 2.0 Chips Not specified 93 [42] Bhagwat N/A PR vs R Multiple platforms Based on literature 177 [43] Koot Urinary LH RIF (n = 43) vs C (n = 72) Human whole genome gene expres‑sion microarrays V2 (Agilent, Belgium) Machine learning predictor 310 [44] Lédée Ultrasound evaluation IF (n = 30) vs Fertile women (n = 15) Affymetrix Human Genome U133 Plus 2.0 Array Adj‑P‑value <0.01 and Δ > 1.35 322 [45] Pathare 6‑7 days after hCG administration IF (n = 10) vs Healthy oocyte donors (n = 8) Illumina HumanHT‑12 V4.0 expression beadchip |FC|> 2 818 [46] Shi Urinary LH; Ultrasound evaluation',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'IF (n = 10) vs Healthy oocyte donors (n = 8) Illumina HumanHT‑12 V4.0 expression beadchip |FC|> 2 818 [46] Shi Urinary LH; Ultrasound evaluation RIF (n = 12) vs C (n \\n= 10) Agilent‑052909 CBC lncRNAmRNA V3 |FC|≥ 2 and adj‑P‑value ≤ 0.05 281 [47]',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Of the respective miRNAs and TFs targeting genes related to endometrial progression and function, 417 miRNAs and 467 TFs were significantly over- represented in at least one gene list (FDR < 0.05, Tables S3-S4). Further, most signatures were significantly governed by TFs rather than miRNAs (4 > odds ratio > 164, FDR < 0.05, 17 lists [89%]), with the exception of Koot’s signature that was governed by miRNAs and showed the opposite (odds ratio = 0.35, FDR < 0.05, Table S5). Eight out of nineteen gene lists (47%) had significant differences between the proportion of genes regulated by progesterone versus estrogen. (*FDR ≤ 0.05). Descriptive analysis of hormonal regulation, normalized by the size of each gene list. Gene lists clustered according to higher (blue) or lower (orange) contribution of hormonal regulation',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Parraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology (2023) 21:84 \\n\\nHsa‑miR‑15a‑5p, hsa‑miR‑128‑3p, hsa‑miR‑218‑5p, \\nhsa‑miR‑27b‑3p, hsa‑miR‑107, hsa‑miR‑424‑5p, \\nhsa‑miR‑195‑5p, hsa‑miR‑103a‑3p, has‑let‑7b‑5p and hsa‑miR‑22‑3p are key players in the regulation of endometrial progression and function\\n\\nTo bridge the gap of endometrial-based evidence in DoRothEA and TarBase databases, we aimed to identify the common miRNAs that mediate gene expression related to endometrial progression and function. We evaluated the number of endometrial gene lists that each miRNA controlled (Fig. 2A). This strategy highlighted the most influential miRNAs that were commonly found across the endometrial gene lists. Notably, 85.80% of miRNAs regulated genes from one to three signatures, and only 59 miRNAs (14.20%) were associated with more than three signatures. Ultimately, 24 miRNAs were prioritized for regulating genes of at least six signatures.\\n\\nNext, we validated the gene expression trends for these 24 prioritized miRNAs in an independent endometrial gene expression dataset (GSE44730) (Fig. 2B). This dataset included the miRNA expression in endometrial biopsies from 20 healthy women in different phases of the menstrual cycle. We found that two of our prioritized miRNAs were not expressed in the endometrial tissue, and only 10 miRNAs (i.e., hsa-miR-15a-5p, hsa-miR-128-3p, hsa-miR-218-5p, hsa-miR-27b-3p, hsa-miR-107, hsa-miR-424-5p, hsa-miR-195-5p, hsa-miR-103a-3p, has-let-7b-5p and hsa-miR-22-3p) showed significant changes across the menstrual cycle (Fig. 2B). Of these, only the expression of hsa-miR-22-3p significantly increased from ESE to MSE stages (FDR = 0.02, Fig. 2B), highlighting the potential inhibitory role of this miRNA in endometrial function. The expression of the remaining miRNAs decreased during the MSE phase.\\n\\nCTCF and GATA6 as distinguished transcriptional regulators of the human menstrual cycle\\n\\nWe repeated the regulation network analysis with the TFs, to identify the prominent TFs commonly regulating the genes related to endometrial progression and function in the reported signatures. We observed 78.80% of the TFs regulated genes in up to 10 signatures, and only 99 TFs (21.20%) regulated genes in over 10 (Fig. 3A). We highlighted CTCF (a CCCTC-binding factor, which functions as a transcriptomic repressor) as the TF with the most influential regulation of endometrial progression and function across 95% of studies (18 gene lists, excluding that of Punyadeera, Fig. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53. Notably, GATA6 was selected due to novel association with endometrial progression and function.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'Parraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology (2023) 21:84'},\n",
       " {'text': 'To infer the expression of CTCF and GATA6  in \\nendometrial tissue throughout the menstrual cycle, \\nwe analysed their expression using a relevant endo - \\nmentrial dataset recently created by our group [30] \\n(Table S1 ). We observed that, relative to the global \\nexpression of endometrial genes, CTCF was broadly \\nexpressed while GATA6  was moderately expressed. \\nIndeed, CTCF was significantly down-regulated \\nbetween the PF and ESE phases (FDR = 2.10E-03)  \\nand ESE to the MSE phase (FDR = 2.10E-04), while \\nGATA6 expression increased from the PF to the \\nMSE phase (FDR = 1.70E-11) and from the ESE  \\nto MSE phase (FDR = 9.80E-15). These findings \\nsupported the role of these TFs at the beginning \\nof the menstrual cycle and during the MSE phase \\n(Fig.\\xa03 B). \\nThese in-silico  validations were then prospec - \\ntively corroborated by RT-qPCR analysis in an inde - \\npendent sample set of 20 endometrial biopsies. We \\nobserved a similar expression profile of CTCF and \\nGATA6 throughout the menstrual cycle, with signifi-\\n',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with significant changes between the endometrial phases. Notably, CTCF expression was decreased between the PF/ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C). The prioritized miRNAs included has‑miR‑16‑5p, has‑miR‑138‑5p, has‑miR‑21‑3p, has‑miR‑205‑5p, has‑miR‑27a‑5p, has‑miR‑15a‑5p, has‑miR‑155‑5p, has‑miR‑129–2‑3p, has‑miR‑147a, has‑miR‑146a‑5p, has‑miR‑107, has‑miR‑424‑5p, has‑miR‑203a‑3p, has‑miR‑195‑5p, has‑miR‑124‑3p, has‑miR‑128‑3p, has‑miR‑22‑3p, has‑miR‑27b‑3p, has‑let‑7b‑5p, has‑miR‑1343‑3p, has‑miR‑1‑3p, has‑miR‑103a‑3p, has‑miR‑23b‑3p, has‑miR‑218‑5p. Significant changes between phases are denoted with asterisks (* FDR ≤ 0.05, ** FDR ≤ 0.01.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Fig. 2 (See legend on previous page.)',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Page 9 of 14\\nParraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology (2023) 21:84 \\n\\nFig. 3 Transcription factors universally regulating endometrial progression and function and their expression throughout the menstrual cycle. A Regulatory network of TFs forming relationships with numerous gene lists (2–18). TFs targeting specific gene lists were placed in the center of the network. CTCF stood out as the only TF that targeted genes belonging to 18/19 (95%) different signatures. B In-silico analysis of CTCF and GATA6 expression throughout the menstrual cycle. Significant changes between phases are marked with asterisks (**FDR ≤ 0.01, ***FDR ≤ 0.001). C RT‑qPCR validation. Bar plot shows the relative mRNA expression of CTCF and GATA6 in endometrial biopsies collected throughout the menstrual cycle. Data is presented as a mean ± SD. Significant differences of CTCF (*FDR < 0.05) and GATA6 (***FDR < 0.001) expression were observed between proliferative/early‑secretory (PF/ESE) and mid‑secretory/late‑secretory phases (MSE/LSE). Abbreviations in (B and C): PF, Proliferative; ESE, early‑secretory; MSE, mid‑secretory; LSE, late‑secretory; SD, standard deviation',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'This data-driven approach exposed the common transcriptional regulators among 19 studies who proposed variable biomarkers of endometrial progression and function. In this study, we focused on understanding the relative contribution of both the hormonal and non-hormonal regulation, from an alternative holistic perspective. We applied data-driven hypothesis research that, unlike the traditional scientific method, allowed us to generate new hypotheses based on all available biologically-relevant knowledge [16], observing the molecular relationships from a wider scale view. Besides, some traditional molecular procedures such as PCRs are also performed to corroborate the insight uncovered with this approach. We highlighted a larger influence of progesterone-related genes and TFs in endometrial progression and observed a larger contribution of miRNAs and estrogen-related genes in endometrial pathology. Furthermore, we unveiled CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-miR-103a-3p and hsa-miR-128-3p as master regulators in the endometrium, and these findings were reinforced by the dynamic changes in their expression throughout the menstrual cycle (dually validated with in-silico and prospective analyses). The menstrual cycle is perceived to be under tight regulation by the ovarian hormones, progesterone and estrogen [48]. Supporting the role of progesterone in the secretory endometrium [49], and reinforcing its indispensable role in endometrial regulation [50, 51], we found progesterone and its related genes had a larger influence in menstrual cycle progression than estrogen. As the RIF-related gene lists reported by Altmae2010 and Koot were generally controlled by estrogen, this further suggested that while progesterone could be crucial for a suitable endometrial progression and mid-secretory phase acquisition, estrogenic irregularities may be more conducive to endometrial function and molecular pathology. Indeed, this hypothesis corroborated previous reports of most uterine disorders being estrogen-dependent [44, 52]. Supporting the findings from another recent study from our group, where the Altmae2010 and Koot signatures were classified as highly-predictive signatures to identify endometrial pathology in comparison with a prediction model that detected endometrial progression [7 ]. This study distinguished these signatures (related to implantation failure and unexplained infertility) based on the magnitude of their miRNA and estrogen-regulation, which contrasted with the prominent influence of the TFs over the remaining signatures more related to endometrial progression. All these facts reinforce the hypothesis that implantation failure is mainly caused by impaired miRNA expression [53].',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'This study transcended the simplistic paradigm of the bihormonal regulation of endometrial function by identifying a plethora of mediators, including 595 miRNAs and 311 TFs, that were previously unrelated to these ovarian hormones. These findings set the foundation for new discoveries, revealing alternative pathways and new actors in the regulation of the human endometrium, and deepening our understanding of the complex regulatory mechanisms behind endometrial progression and unexplained infertility. MicroRNAs are well known to be involved in embryo implantation and receptivity control. Using network analysis, we unveiled the 10 common miRNAs regulating endometrial progression and function across published biomarker signatures (hsa-miR-22-3p, hsa-miR-107, hsa-miR-103a-3p, hsa-miR-128-3p, hsa-miR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, hsa-miR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p).',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Their significant gene expression changes in the secre - \\ntory endometrium support their potential role in the \\nendometrial regulation through an inhibition during \\nthe WOI. Notably, hsa-miR-424-5p, hsa-miR-27b-3p \\nand hsa-miR-195-5p have been previously associated \\nwith RIF [11, 54, 56], while has-let-7b-5p was associ - \\nated with human endometrial receptivity [57], and \\nhsa-miR-22-3p was overexpressed in a decidualization \\nmodel [58]. Moreover, hsa-miR-15a-5p was previously \\nassociated with endometriosis [59]. Notwithstanding, \\nthis study is the first to associate hsa-miR-15a-5p, hsa- \\nmiR-218-5p, hsa-miR-103a-3p, hsa-miR-107, and hsa- \\nmiR-128-3p to endometrial progression and functional \\nregulation in humans.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Page 11 of 14\\nParraaga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \\n\\nendometrial progression and function were beyond the scope of this project and merit further investigation. CTCF is a conserved zinc finger protein whose regulatory functions are well characterized throughout the human body [64]. CTCF acts as a transcriptional repressor in RNA polymerase II (Pol II) pausing and imprinting and X-chromosome inactivation [65], as well as an insulator, blocking the interaction between enhancers and the promoters of neighbouring genes [66]. Considering these attributed functions and its variable expression throughout the menstrual cycle, we propose that CTCF exerts an inhibitory role in endometrial tissue during the PF phase of the menstrual cycle. With the significant downregulation of CTCF in the secretory endometrium, dually validated in-silico and experimentally herein, its inhibited genes would be derepressed and become transcriptionally active during the WOI. This interpretation supports previous findings from our group demonstrating that, during the WOI, a global transcriptional derepression may be required for implantation and early embryo development [63]. Transcriptional derepression has been associated with multiple human disease states and should be investigated further within the context of endometrial-factor infertility. Despite previous associations of CTCF with endometriosis [67], its implication in endometrial progression and function has not been directly proposed until now. Nevertheless, the role of CTCF in endometrial receptivity has been revealed through its interaction with HOXA10, a gene mainly expressed in endometrium involved in functions such as endometrial proliferation and differentiation, the formation of pinopodes or embryo implantation [68]. Indeed, this study proposed that an overexpression of CTCF can lead to a drop in HOXA10 expression, affecting endometrial proliferation and endometrial function and corroborating the fact that CTCF must decrease its expression as we can observe in our findings. GATA6, the second prioritized TF that overlapped across the gene lists, is a transcription factor belonging to the GATA family, a highly conserved family of six zinc finger proteins [69]. Besides its essential role in embryonic development [70], GATA6 plays a key role in endometriosis and regulates steroidogenic genes [71]. Together with evidence supporting both an activating and inhibitory role for GATA6 [72, 73], the significant changes we observed in its expression across the menstrual cycle reflect its presumed involvement in the endometrium and suggest an activating role in endometrial progression and endometrial receptivity acquisition. Like CTCF, this is the first time that GATA6 is proposed as a key TF in endometrial progression and function.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'endometrial regulation remain elusive, emerging experimental technologies such as High-throughput Chromosome Conformation Capture (Hi-C) or chromatin interaction analysis with paired-end tags (ChIA-PET) that respectively study general chromatin interactions or chromatin interactions involving a specific protein [74, 75] may improve our understanding of genetic regulation in the future. Nevertheless, as collaborative efforts unveil new regulators and gene targets in humans (such as the ENCODE project [76]), the complexities of transcriptional regulation can be exposed by in-silico analyses like those described herein, and we can gain a better understanding of the processes involved in menstrual cycle progression and endometrial competence. Furthermore, it should be noted that results of gene expression studies are influenced by different variables, such as the experimental platforms employed, study designs, or patients’ characteristics of each independent',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'expression studies are influenced by different variables, such as the experimental platforms employed, study designs, or patients’ characteristics of each independent study, adding variability [30, 77, 78]. Despite this fact, we have found that our holistic approach overcomes this undesired variability effect by finding common upstream master regulators across studies and predicting new relationships between hormones, TFs and miRNAs. This approach also provides the basis for future single-molecule studies that aim to elucidate new regulatory pathways in endometrial progression and function. Our results lay the foundation for further molecular studies that can validate the function(s) of the prevailing endometrial regulators we prioritized. The study shows for the first time, the relative contribution of estrogens, progesterone, TFs and miRNAs in endometrial function and progression. Endometrial progression is mainly influenced by progesterone-related genes and TFs, whereas miRNAs and estrogen-related genes play a larger role in endometrial pathology. Moreover, we highlight novel common transcriptional regulators such as CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-miR-103a-3p, and hsa-miR-128-3p across 19 studies that propose biomarkers in endometrial regulation. These results reveal the molecular mechanism underlying endometrial regulation and lay the foundation for the development of targeted therapies for patients with endometrial-factor infertility.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'We reviewed 19 publications related to endometrial progression and function, extracted the gene signatures from the original publication, and created corresponding gene lists that we could annotate. We performed different functional analyses based on the type of regulator (i.e., ovarian hormones, transcription factors and miRNAs). We used online databases to obtain the information of the regulatory process, at the hormonal (Gene Ontology, Kyoto Encyclopedia of Genes and Genomes, Dorothea), transcription factor (Dorothea) and miRNA (Tarbase) level and analysed the results accordingly. Finally, we employed an integrative analysis and prioritized the most relevant regulators for subsequent validation. TF, Transcription factors; miRNA, microRNA.',\n",
       "  'doi': 'https://doi.org/10.1186/s12958-023-01131-4',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Additional file 5: Supplementary Table S4. Results of TFs enrichment. Excel file with the over‑representation analysis results of the transcription factors (TFs). Each sheet contains the results of a specific list of genes. In each sheet, the following information is included: ID, gene symbol of the TF; GeneRatio, the proportion of TF targets in the gene list; BgRatio, the proportion of TF targets in the whole genome; OddsRatio; the odd ratios for each TF, pvalue; the P‑value obtained for each TF; padjust, P‑value adjusted by FDR for each TF; geneID, the gene symbol ID of the TF targets. Additional file 6: Supplementary Table S5. Relative contribution of each regulator type (TFs or miRNAs) in each gene list. A Fisher’s test was performed to calculate the proportional differences between transcription factors (TFs) and microRNAs (miRNAs). Resulting P‑values were adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Resulting P‑values were adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene lists were mainly regulated by TFs, whereas odds ratios < 1 indicate a higher regulation by miRNAs. Odds Ratio with a value of “Inf” indicate that miRNAs were not found for this gene list. Percentages indicate the proportion between the significant regulators in each gene signature and regulators with at least one target gene in the corresponding signature. (***FDR ≤0.001).',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'The data used to analyze the TF and miRNA expression throughout the men‑\\ntstrual cycle and in infertility are available in Gene Expression Omnibus, with \\nIDs GSE98386, GSE29981, GSE4888, GSE119209, GSE86491, and GSE44558. The raw endometrial gene expression data (from previous studies) was downloaded from the publicly‑available functional genomic dataset repository Gene Expression Omnibus (GEO) [79] for curation/processing. In accordance with GEO policies, all patient data was anonymized and encrypted, and therefore an additional institutional review board approval was not required for our in-silico validation of regulator gene expression changes in endometrial tissue.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'tional institutional review board approval was not required for our in-silico validation of regulator gene expression changes in endometrial tissue. On the other hand, for the experimental validation of gene expression changes in endometrial biopsies, all individuals gave written informed consent for endometrial gene expression evaluation prior to participating (Ethics Committee of the Instituto Valenciano de Infertilidad [IVI Valencia], Valencia, Spain [1810‑FIVI‑066‑PD/1706‑FIVI‑048‑PD]).',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'Ethics Committee of the Instituto Valenciano de Infertilidad [IVI Valencia], Valencia, Spain [1810‑FIVI‑066‑PD/1706‑FIVI‑048‑PD]'},\n",
       " {'text': 'Parraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology (2023) 21:84 \\n\\nAuthor details\\n1 IVIRMA Global Research Alliance, IVI Foundation, Instituto de Investigación Sanitaria La Fe (IIS La Fe), Av. Fernando Abril Martorell 106, Torre A, Planta 1ª, 46026 Valencia, Valencia, Spain. 2 Department of Pediatrics, Obstetrics and Gynaecology, Universidad de Valencia, Av. Blasco Ibáñez 15, 46010 Valen‑cia, Valencia, Spain. 3 IVIRMA Global Research Alliance, IVIRMA Valencia, Plaza de La Policia Local 3, 46015 Valencia, Spain.\\nReceived: 21 June 2023 Accepted: 22 August 2023\\n\\nReferences\\n1. Strowitzki T, Germeyer A, Popovici R, von Wolff M. The human endometrium as a fertility-determining factor. Hum Reprod Update. 2006;12:617–30.\\n2. Cakmak H, Taylor HS. Implantation failure: Molecular mechanisms and clinical treatment. Hum Reprod Update. 2011;17:242–53.\\n3. Lessey BA, Young SL. What exactly is endometrial receptivity? Fertil Steril. 2019;111:611–7.\\n4. Talbi S, Hamilton AE, Vo KC, Tulac S, Overgaard MT, Dosiou C, et al. Molecular phenotyping of human endometrium distinguishes menstrual cycle phases and underlying biological processes in normo‑ovulatory women. Endocrinology. 2006;147:1097–121.\\n5. Somigliana E, Vigano P , Busnelli A, Paffoni A, Vegetti W, Vercellini P . Repeated implantation failure at the crossroad between statistics, clinics and over‑diagnosis. Reprod Biomed Online. 2018;36:32–8.\\n6. Craciunas L, Gallos I, Chu J, Bourne T, Quenby S, Brosens JJ, et al. Conventional and modern markers of endometrial receptivity: a systematic review and meta‑analysis. Hum Reprod Update. 2019;25:202–23.\\n7. Sebastian‑Leon P , Garrido N, Remohí J, Pellicer A, Diaz‑Gimeno P . Asynchronous and pathological windows of implantation: Two causes of recurrent implantation failure. Hum Reprod. 2018;33:626–35.\\n8. Vasquez YM, DeMayo FJ. Role of nuclear receptors in blastocyst implantation. Semin Cell Dev Biol. 2013;24:724–35.\\n9. Wang H, Zhang S, Lin H, Kong S, Wang S, Wang H, et al. Physiological and molecular determinants of embryo implantation. Mol Aspects Med. 2013;34:939–80.\\n10. Kelleher AM, Behura SK, Burns GW, Young SL, Demayo FJ, Spencer TE. Integrative analysis of the forkhead box A2 (FOXA2) cistrome for the human endometrium. FASEB J. 2019;33:8543–54.\\n11. Revel A, Achache H, Stevens J, Smith Y, Reich R. MicroRNAs are associated with human embryo implantation defects. Hum Reprod. 2011;26:2830–40.\\n12. Galliano D, Pellicer A. MicroRNA and implantation. Fertil Steril. 2014;101:1531–44.\\n13. Xu H, Zhou M, Cao Y, Zhang D, Han M, Gao X, et al. Genome‑wide analysis of long noncoding RNAs, microRNAs, and mRNAs forming a competing endogenous RNA network in repeated implantation failure. Gene. 2019;720: 144056.\\n14. Copley RR. The animal in the genome: comparative genomics and evolution. Philos Trans R Soc Lond B: Biol Sci. 2008;363:1453–61.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Breaking the ageing paradigm in endometrium: endometrial gene expression related to cilia and ageing hallmarks in women over 35 years.',\n",
       "  'doi': '37:762–76',\n",
       "  'citation': 'Devesa‑Peiro A, Sebastian‑Leon P , Parraga‑Leo A, Pellicer A, Diaz‑Gimeno P .'},\n",
       " {'text': 'Meta‑signature of human endometrial receptivity: a meta‑analysis and validation study of transcriptomic biomarkers.',\n",
       "  'doi': 'https:// doi. org/ 10. 1101/ gr. 240663.',\n",
       "  'citation': 'Altmäe S, Koel M, Võsa U, Adler P , Suhorutšenko M, Laisk‑Podar T, et al. Scient Rep. 2017;7:10077.'},\n",
       " {'text': 'Chapter 15 ‑ the molecular signature of the endometrial receptivity: research and clinical application. In: Horcajadas JA, Gosálvez J, editors. Reproductomics: the omics revolu‑tion and its impact on human reproductive medicine. Academic Press; 2018. p. 301–279.\\n\\n36. Díaz‑Gimeno P , Horcajadas JA, Martínez‑Conejero JA, Esteban FJ, Alamá P , Pellicer A, et al. A genomic diagnostic tool for human endometrial receptivity based on the transcriptomic signature. Fertil Steril. 2011;95:50.\\n\\n37. Kao LC, Tulac S, Lobo S, Imani B, Yang JP , Germeyer A, et al. Global gene profiling in human endometrium during the window of implantation. Endocrinology. 2002;143:2119–38.\\n\\n38. Ponnampalam AP , Weston GC, Trajstman AC, Susil B, Rogers PAW. Molecular classification of human endometrial cycle stages by transcriptional profiling. Mol Hum Reprod. 2004;10:879–93.\\n\\n39. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'lar classification of human endometrial cycle stages by transcriptional profiling. Mol Hum Reprod. 2004;10:879–93. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al. Oestrogen‑modulated gene expression in the human endometrium. Cell Mol Life Sci. 2005;62:239–50. Riesewijk A, Martín J, van Os R, Horcajadas JA, Polman J, Pellicer A, et al. Gene expression profiling of human endometrial receptivity on days LH+2 versus LH+7 by microarray technology. Mol Hum Reprod. 2003;9:253–64. Bastu E, Demiral I, Gunel T, Ulgen E, Gumusoglu E, Hosseini MK, et al. Potential marker pathways in the endometrium that may cause recurrent implantation failure. Reprod Sci. 2019;26:879–90. Bersinger NA, Wunder DM, Birkhäuser MH, Mueller MD. Gene expression in cultured endometrium from women with different outcomes follow‑following IVF. Mol Hum Reprod. 2008;14:475–84.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Page 14 of 14Parraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology           (2023) 21:84 \\n•\\n\\nfast, convenient online submission\\n •\\n\\nthorough peer review by experienced researchers in your ﬁeld\\n• \\n\\nrapid publication on acceptance\\n• \\n\\nsupport for research data, including large and complex data types\\n\\ngold Open Access which fosters wider collaboration and increased citations\\n\\nmaximum visibility for your research: over 100M website views per year •\\n  At BMC, research is always in progress. Learn more biomedcentral.com/submissions\\nReady to submit y our researc hReady to submit y our researc h  ?  Choose BMC and benefit fr om: ?  Choose BMC and benefit fr om: 43. Bhagwat SR, Chandrashekar DS, Kakar R, Davuluri S, Bajpai AK, Nayak S, et al. Endometrial receptivity: a revisit to functional genomics studies on human endometrium and creation of HGEx ‑ERdb. PLoS ONE. 2013;8: e58419.\\n44. Koot YEM, Van Hooff SR, Boomsma CM, Van Leenen D, Koerkamp MJAG, Goddijn M, et al. An endometrial gene expression signature accurately predicts recurrent implantation failure after IVF. Sci Rep. 2015;2016(6):1–12.\\n45. Lédée N, Munaut C, Aubert J, Sérazin V, Rahmati M, Chaouat GG, et al. Specific and extensive endometrial deregulation is present before conception in IVF/ICSI repeated implantation failures (IF) or recurrent miscarriages. J Pathol. 2011;225:554–64.\\n46. Pathare ADS, Zaveri K, Hinduja I. Downregulation of genes related to immune and inflammatory response in IVF implantation failure cases under controlled ovarian stimulation. Am J Reprod Immunol. 2017;78: e12679.\\n47. Shi C, Han HJ, Fan LJ, Guan J, Zheng XB, Chen X, et al. Diverse endometrial mRNA signatures during the window of implantation in patients with repeated implantation failure. Hum Fertil. 2018;21:183–94.\\n48. Messinisi IE. Ovarian feedback, mechanism of action and possible clinical implications. Hum Reprod Update. 2006;12:557–71.\\n49. Edwards RG. Implantation, interception and contraception. Hum Reprod (Oxford, England). 1994;9:985–95.\\n50. Young SL. Oestrogen and progesterone action on endometrium: a translational approach to understanding endometrial receptivity. Reprod Biomed Online. 2013;27:497–505.\\n51. Bhurke AS, Bagchi IC, Bagchi MK. Progesterone‑regulated endometrial factors controlling implantation. Am J Reprod Immunol. 2016;75:237–45.\\n52. Devesa‑Peiro A, Sebastian‑Leon P , Garcia‑ Garcia F, Arnau V, Aleman A, Pellicer A, et al. Uterine disorders affecting female fertility: what are the molecular functions altered in endometrium? Fertil Steril. 2020;113:1261–74.\\n53. von Grothusen C, Frisendahl C, Modhukur V, Lalitkumar PG, Peters M, Faridani OR, et al. Uterine fluid microRNAs are dysregulated in women with recurrent implantation failure. Hum Reprod. 2022;37:734–46.\\n54. Altmäe S, Martinez‑Conejero JA, Esteban FJ, Ruiz‑Alonso M, Stavreus‑Evers A, Horcajadas JA, et al. MicroRNAs miR‑30b, miR‑30d, and miR‑494',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'regulate human endometrial receptivity.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'Reprod Sci. 2013;20:308–17.'},\n",
       " {'text': 'stromal decidualization in vitro. PLoS ONE. 2012;7:41080. Liu D, Liang Y, Chen M, Yang F, Yao S. Knockdown of circ_0075503 sup ‑ presses cell migration and invasion by regulating miR‑15a‑5p and KLF12 in endometriosis. Mol Cell Biochem. 2021;476:3845–56. Chi RPA, Wang T, Adams N, Wu SP , Young SL, Spencer TE, et al. Human endometrial transcriptome and progesterone receptor cistrome reveal important pathways and epithelial regulators. J Clin Endocrinol Metab. 2020;105:1–21. Pawar S, Starosvetsky E, Orvis GD, Behringer RR, Bagchi IC, Bagchi MK. STAT3 regulates uterine epithelial remodeling and epithelial‑stromal crosstalk during implantation. Mol Endocrinol. 2013;27:1996–2012. Cheng YH, Imir A, Suzuki T, Fenkci V, Yilmaz B, Sasano H, et al. SP1 and SP3 mediate progesterone‑ dependent induction of the 17beta hydroxyster ‑ oid dehydrogenase type 2 gene in human endometrium. Biol Reprod. 2006;75:605–14. Sebastian‑Leon P , Devesa‑Peiro A, Aleman A, Parraga‑Leo A, Arnau V, Pellicer A, et al. Transcriptional changes through menstrual cycle reveal a global transcriptional derepression underlying the molecular mechanism involved in the window of implantation. Mol Hum Reprod. 2021. https:// doi. org/ 10. 1093/ molehr/ gaab0 27. Kim S, Yu NK, Kaang BK. CTCF as a multifunctional protein in genome regulation and gene expression. Exp Mol Med. 2015;47: e166. Shukla S, Kavak E, Gregory M, Imashimizu M, Shutinoski B, Kashlev M, et al. CTCF‑promoted RNA polymerase II pausing links DNA methylation to splicing. Nature. 2011;479:74–9. Phillips JE, Corces VGCTCF. Master weaver of the genome. Cell. 2009;137:1194–211. Guo J, Cao B, Xu X, Wu F, Zhu B. Novel CTCF mutations in Chinese patients with ovarian endometriosis. Mol Med Rep. 2018;18:1031–6. Geng J, Cui C, Yin Y, Zhao Y, Zhang C. LncRNA NEAT1 affects endome ‑ trial receptivity by regulating HOXA10 promoter activity. Cell Cycle. 2022;21:1932–44. Viger RS, Guittot SM, Anttonen M, Wilson DB, Heikinheimo M. Role of the GATA family of transcription factors in endocrine development, function, and disease. Mol Endocrinol. 2008;22:781. Xuan S, Sussel L. GATA4 and GATA6 regulate pancreatic endoderm identity through inhibition of hedgehog signaling. Development (Cam‑ bridge). 2016;143:780–6. Bernardi LA, Dyson MT, Tokunaga H, Sison C, Oral M, Robins JC, et al. The essential role of GATA6 in the activation of estrogen synthesis in endome ‑ triosis. Reprod Sci. 2019;26:60. Yin F, Herring BP . GATA‑6 can act as a positive or negative regula ‑ tor of smooth muscle‑specific gene expression *. J Biol Chem. 2005;280:4745–52. Zhai J, Luo G. GATA6‑induced FN1 activation promotes the proliferation, invasion and migration of oral squamous cell carcinoma cells. Mol Med ‑ Rep. 2022;25:102. Fullwood MJ, Liu MH, Pan YF, Liu J, Xu H, Mohamed YB, et al. An oestrogen‑receptor‑α‑bound human chromatin interactome. Nature. 2009;462:58–64.',\n",
       "  'doi': 'https://doi.org/10.1093/molehr/gaab027',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': '75. Lieberman‑Aiden E, Van Berkum NL, Williams L, Imakaev M, Ragoczy T, Telling A, et al. Comprehensive mapping of long range interactions reveals folding principles of the human genome. Science. 2009;326:289.\\n76. Birney E, Stamatoyannopoulos JA, Dutta A, Guigó R, Gingeras TR, Margulies EH, et al. Identification and analysis of functional elements in 1% of the human genome by the ENCODE pilot project. Nature. 2007;447:799–816.\\n77. Tajti F, Kuppe C, Antoranz A, Ibrahim MM, Kim H, Ceccarelli F, et al. A func‑\\n tional landscape of CKD entities from public transcriptomic data. Kidney Int Rep. 2020. https:// doi. org/ 10. 1016/j. ekir. 2019. 11. 005.\\n78. Henarejos‑ Castillo I, Devesa‑Peiro A, de Miguel‑ Gomez L, Sebastian‑Leon P , Romeu M, Aleman A, et al. Predicted COVID ‑19 molecular effects on \\n endometrium reveal key dysregulated genes and functions. Mol Hum Reprod. 2022.gaac035.\\n79. Barrett T, Wilhite SE, Ledoux P , Evangelista C, Kim IF, Tomashevsky M, et al.',\n",
       "  'doi': 'https:// doi. org/ 10. 1016/j. ekir. 2019. 11. 005',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'endometrium reveal key dysregulated genes and functions.',\n",
       "  'doi': 'gaac035',\n",
       "  'citation': 'UNKOWN'}]"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "my_json\n",
    "\n",
    "# my_json['author'] = \"parraga-leo\"\n",
    "# my_json['title'] = storage[0].metadata['title']\n",
    "# my_json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a7cfb13b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2455317/1728583740.py:10: LangChainDeprecationWarning: The class `HuggingFaceEmbeddings` was deprecated in LangChain 0.2.2 and will be removed in 1.0. An updated version of the class exists in the :class:`~langchain-huggingface package and should be used instead. To use it run `pip install -U :class:`~langchain-huggingface` and import as `from :class:`~langchain_huggingface import HuggingFaceEmbeddings``.\n",
      "  embedding_function = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")\n",
      "/home/aparraga/anaconda3/envs/RAG/lib/python3.9/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "# load the document and split it into pages\n",
    "loader = PyPDFLoader(\"/data/local/aparraga/Bioinformatician/RAG/Publications/Parraga-Leo2023.pdf\")\n",
    "pages = loader.load_and_split()\n",
    "\n",
    "# split it into chunks\n",
    "text_splitter = CharacterTextSplitter(chunk_size=1000, chunk_overlap=50)\n",
    "storage = text_splitter.split_documents(pages)\n",
    "\n",
    "# create the open-source embedding function\n",
    "embedding_function = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "4a0b29cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A total of 3,608 distinct genes from the 19 gene lists were associated with endometrial progression\n",
      "The lists’ regulation was significantly favoured by TFs (89% (17/19) of gene lists) and progesterone (47% (8 /19) of gene lists), rather than miRNAs (5% (1/19) of gene lists) or estrogen (0% (0/19) of gene lists), respectively (FDR < 0.05). Exceptionally, two gene lists that were previously associated with implantation failure and unexplained infertility were less hormone‑dependent, but primarily regulated by estrogen. Although endometrial progression genes were mainly targeted by hormones rather than non‑hormonal contributors (odds ratio = 91.94, FDR < 0.05), we identified 311 TFs and 595 miRNAs not previously associated with ovarian hormones. We highlight CTCF, GATA6, hsa‑miR‑15a‑5p, hsa‑miR‑218‑5p, hsa‑miR‑107, hsa‑miR‑103a‑3p, and hsa‑miR‑128‑3p, as overlapping novel master regulators of endometrial function. The gene expression changes of selected regulators through‑out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their\n",
      "out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their potential regulatory roles in the endometrium.\n",
      "This study revealed novel hormonal and non‑hormonal regulators and their relative contributions to endometrial progression and pathology, providing new leads for the potential causes of endometrial‑factor infertility. The endometrium is the innermost layer of the uterus, which undergoes dynamic histological, physiological and molecular changes that allow it to synchronize with embryo development, facilitate embryo implantation, and ultimately, establish a successful pregnancy. Implantation is a crucial and complex limiting step for conception which occurs between days 19 to 24 of a normal menstrual cycle. Consequently, this period of time is called the window of implantation (WOI) and is characterized by abrupt transcriptomic changes in the endometrial tissue. Alterations in either WOI establishment and endometrial progression may lead to implantation failures, which, along with biochemical miscarriage, account for more than 50% of pregnancy losses at pre-clinical stages. These alterations have broadly been classified as being related to a displaced WOI, caused by variable timing of endometrial progression, or a disrupted WOI, where impaired endometrial function prevents the establishment of an effective WOI. Initially, the WOI was postulated to mainly be regu -lated by the ovarian hormones (namely, estrogen and progesterone) acting via their respective nuclear receptors (i.e., ESR1/2 and PGR). However, evidence has shown that these receptors cooperate with other transcription factors (TFs) and co-regulators to mediate uterine physiology [8]. In fact, TFs (such as homeobox TFs, FOXA2, and KLF9) have recently been reported as key regulators for the establishment of endometrial receptivity [9, 10], and microRNAs (miRNAs) have been found to similarly intervene in this complex and multifactorial process as transcriptional regulators [11]. While the involvement of both types of regulators was widely reported in embryo implantation [12] and other infertility-related diseases, including recurrent implantation failure (RIF) [13], the complex interactions between the different types of regulators and their respective contributions to endometrial progression and function have not been investigated from a systemic point of view. This paradigm shift approaches genes and regulators in an integrative way, by considering how they interact with and coordinate each other to carry out cellular processes, rather than reinforcing the premise they work independently [14]. By more accurately reflecting the biological milieu, systemic approaches take gene-based discoveries to the next level.\n",
      "Previous studies of endometrial transcriptomics have been used to define the WOI, compare the endometrium of healthy women as their menstrual cycles progressed [4, 17–19]; address endometrial differences related to age [20] or the dysfunction of the WOI between patients with implantation failure and healthy controls [21]; and finally, correct for the menstrual cycle bias that can mask important biomarkers [22]. Notwithstanding, the candidate biomarkers of endometrial receptivity reported in each of the aforementioned studies overlap poorly [7]. Indeed, recent comparisons conducted by Sebastian-Leon and colleagues [7] found no congruities between 16 different reported gene lists of endometrial receptivity, and only fair to moderate functional agreements between some of the included signatures. Despite previous transcriptional regulation studies, there have been no holistic studies of endometrial receptivity.\n",
      "fair to moderate functional agreements between some of the included signatures. Despite previous transcriptional regulation studies, there have been no holistic studies of this regulatory process, which robustly analyse the drivers of transcription across a higher number of studies (that each searched for endometrial receptivity biomarkers) and map the relative contributions of each type of regulator to endometrial progression regulation (Ovarian hormones [i.e., estrogen, progesterone], TFs, miRNAs). Thus, the aim of this study was to utilize available genomic data regarding transcriptional regulators to identify overlapping mediators of endometrial transcriptional regulation (i.e., ovarian hormones, TFs, and miRNAs) and determine the relative contribution of each type of regulator among the previously reported gene signatures. These findings could help unveil the master regulators and principal type of endometrial regulation, paving the way for further research aimed at improving woman´s reproductive health. Annotating gene lists associated with endometrial progression and function. Endometrial progression and implantation failure gene lists [7] were retrieved and updated using a public data repository (i.e., Gene Expression Omnibus (GEO)).\n",
      "To study the hormonal regulation of endometrial progression, the Kyoto Encyclopaedia of Genes and Genomes (KEGG) [24], and Gene Ontology (GO) [25] databases were consulted for ovarian hormone-related genes, using relevant keywords (i.e., progesterone, estrogen, oestrogen and estradiol) and selecting those pathways and functions that contained them (Supplementary Fig. S1B). Genes associated with the obtained human KEGG pathways and GO functions were grouped according to whether they were unique to progesterone (P4 gene set) or estrogen (E2 gene set) or related to both hormones (P4 and E2 gene set). Gene targets of the nuclear progesterone (PGR) and estrogen (ESR1, ESR2) receptors were added to their corresponding gene sets using DoRothEA (Discriminant Regulon Expression Analysis) database [26], considering only manually-curated or ChiP-Seq experimentally-validated gene-TF relationships. Finally, the P4 and E2 gene sets were independently mapped to each gene list.\n",
      "Prioritizing key regulators in endometrial progression and function\n",
      "We built regulatory networks to systemically analyse endometrial regulation.\n",
      "We built regulatory networks to systemically analyse endometrial regulation. Nodes represented gene lists and regulators (miRNAs or TFs), while edges indicated significant enrichment among the lists and their corresponding regulators (false discovery rate (FDR) ≤ 0.05). To select the most influential regulators (i.e., those which targeted genes in most of our signatures), we studied the degree distribution of the networks (i.e., the number of gene lists regulated by each molecule) and prioritized the miRNAs and TFs which surpassed the relative maximum (i.e., 1.50 times the interquartile range (IQR)) number of relationships (gene list–regulators). All networks were built and analysed with Cytoscape version 3.7 [28] (Supplementary Fig. S1B). To evaluate how the miRNAs’ expression change in the endometrium throughout the menstrual cycle, we used the raw data from GSE44558 GEO dataset [29], deriving from 20 endometrial samples collected throughout the menstrual cycle: four in early-proliferative (EPF), four in late-proliferative (LPF), four in early-secretory (ESE), four in mid-secretory (MSE) and four in late-secretory (LSE). To evaluate how the TFs’ expression change in the endometrium throughout the menstrual cycle, we analysed the integrated endometrial dataset previously created by our group, and reported in [30], which compiled data from five prior publications (GSE98386, GSE29981, GSE4888, GSE119209 and GSE86491). This dataset included endometrial gene expression\n",
      "Of the 19 gene lists used for our analysis, eleven were obtained from published studies evaluating control patients throughout the menstrual cycle [4, 17–19, 34–40], and the other eight were derived from studies comparing patients with RIF or unexplained infertility to controls [21, 41–47] (Table 1). Unifying all the afore-\n",
      "mentioned signatures, we compiled 3,608 genes related to endometrial progression and function. Hormonal regulation of endometrial progression is largely driven by progesterone. We identified 7,540 and 698 genes related to estrogen and progesterone hormones, respectively. However, as determined by the relative contribution of each type of hormone within each gene list (Fig. 1A), 17/19 (89%) signatures favoured regulation by progesterone rather than estrogen. These differences were significant in 47% of the signatures (FDR < 0.05). Exceptionally, the genes reported by the Altmae2010 and Koot signatures, that were related to unexplained infertility and RIF, had a preference for estrogen regulation (Fig. 1A). With respect to the proportion of progesterone-related genes, Altmae2010 showed 2.1 times more estrogen-related genes while Koot showed 1.7 times more. More over, comparing the proportion of genes regulated by both hormones, between signatures revealed that Alt -\n",
      "We identified a total of 770 miRNAs and 555 TFs regulating at least one of the 3,608 genes reported in the 19 gene lists. Further, we observed that 29% of the miRNAs and 44% of the TFs were associated with the ovarian hormones (i.e., progesterone and/or estrogen). Significantly more genes were targeted by hormonally-influenced regulators rather than non-hormonal regulators (P -value < 0.01, odds ratio =  91.94). Nevertheless, we also identified 595 miRNAs (77.27%) and 311 TFs (56.04%) that were unrelated to progesterone and/or estrogen, highlighting a substantial proportion of hormonally-independent regulators of the menstrual cycle.\n",
      "Table 1 Characteristics of the 19 gene lists included in this study. Gene lists were divided according to their evaluated condition. Each signature was labelled using the name of first author, followed by the publication year in the case of duplicates. Endometrial staging was based on different variables including urinary LH, histological analysis, ultrasound evaluation, or the number of days since the hCG trigger. Participants included healthy controls and women with infertility‑related conditions. Various platforms were used to evaluate gene expression. The thresholds applied for gene refinement and the final number of genes per signature are also presented\n",
      "Abbreviations: Δ delta  Ct, adj-P-value Adjusted P‑value, C Control group, d Day of menstrual cycle, EPF Early proliferative, ESE Early secretory, |FC| Absolute fold‑change, FC Fold change, HRT Hormone replacement therapy, IF Implantation failure, LH Luteinizing hormone, LH + X X days post LH surge, LPF Late proliferative, LSE Late‑secretory, M Miscarriage, MPF Mid‑proliferative, MSE Mid‑secretory, n number of samples, N/A Not applicable, OP Ongoing pregnancy, PF Proliferative, PFP Proportion of false positives, PR Pre‑receptive, R Receptive, RIF Recurrent implantation failure, RRA Robust Rank Aggregation algorithm, SF Spontaneously fertile women, UI Women with unexplained infertility, vs versus\n",
      "Condition Gene signature Endometrial staging Participants Platform Threshold No. genes References\n",
      "Endometrial progres‑ sion\n",
      "Altmäe2017 N/A N/A N/A RRA 57 [34]\n",
      "Borthwick Urinary LH; Histologi‑ cally confirmed PF d9‑11 (n = 5) vs R LH + 6–8 (n = 5)\n",
      "Affymetrix Genechip Hu95A N/A 116 [18]\n",
      "Carrascosa N/A N/A Quantitative RT‑PCR Based on literature 187 [35]\n",
      "Carson Urinary LH; Histologically confirmed ESE LH + 2–4 (n = 3) vs MSE LH + 7–9 (n = 3)\n",
      "Affymetrix Genechip Hu95A |FC|> 2 695 [17]\n",
      "Diaz‑Gimeno Urinary LH PR LH + 1–5 (n = 15) vs R\n",
      "Agilent custom gene expression microarray |FC| > 2 234 [36]\n",
      "Kao Urinary LH; Histologically confirmed LPF d8‑10 (n = 4) vs MSE LH + 8‑10 (n = 7)\n",
      "Affymetrix Genechip Hu95A |FC|> 2 and P‑value <0.05 340 [37]\n",
      "Mirkin Urinary LH; Histological dating ESE LH + 3 (n = 3) vs MSE LH + 8 (n = 5)\n",
      "Affymetrix Genechip Hu95A |FC|>2 and adj‑P‑value < 0.05 105 [19]\n",
      "Ponnampalam Histological dating EPF (n = 5) MPF (n = 7) LPF (n = 3) ESE (n = 7) MSE (n = 8) LS (n = 7) Mense (n = 6)\n",
      "Custom Adj‑P‑value <0.05 306 [38]\n",
      "Punyadeera Histological dating Mense (n = 2) vs LPF (n=2)\n",
      "Affymetrix HG‑U133A FC and P‑value 50 [39]\n",
      "Riesewijk Urinary LH; Histological dating PR LH + 2 (n = 5) vs R LH + 7 (n = 5)\n",
      "Affymetrix Genechip Hu95A |FC|> 3 in at least four out of five women. 196 [40]\n",
      "Talbi Histological dating P (n = 6) vs ESE (n = 3) vs MSE (n = 8) vs LSE (n = 6)\n",
      "Affymetrix HG‑U133 Plus 2.0 |FC|> 1.5 and adj‑P‑value < 0.05 317 [4]\n",
      "Implantation failure Altmäe2010 Urinary LH UI (n = 4) vs C (n = 5) Whole Human Genome Oligo Microarray (Agilent Technologies) |FC|> 2 and PFP < 0.05 184 [21] Bastu 6‑7 days after LH surge IF (n = 24) vs Fertile women (n = 24) Agilent‑039494 SurePrint G3 Human GE v2 8x60K Log2(FC) ≥ 1 and adj‑P‑value ≤0.05 |FC|> 2 and PFP < 0.05 524 [41] Bersinger N/A IF (n = 3) vs M (n = 3) vs OP (n = 3) Affymetrix Human Genome U133A 2.0 Chips Not specified 93 [42] Bhagwat N/A PR vs R Multiple platforms Based on literature 177 [43] Koot Urinary LH RIF (n = 43) vs C (n = 72) Human whole genome gene expres‑sion microarrays V2 (Agilent, Belgium) Machine learning predictor 310 [44] Lédée Ultrasound evaluation IF (n = 30) vs Fertile women (n = 15) Affymetrix Human Genome U133 Plus 2.0 Array Adj‑P‑value <0.01 and Δ > 1.35 322 [45] Pathare 6‑7 days after hCG administration IF (n = 10) vs Healthy oocyte donors (n = 8) Illumina HumanHT‑12 V4.0 expression beadchip |FC|> 2 818 [46] Shi Urinary LH; Ultrasound evaluation\n",
      "IF (n = 10) vs Healthy oocyte donors (n = 8) Illumina HumanHT‑12 V4.0 expression beadchip |FC|> 2 818 [46] Shi Urinary LH; Ultrasound evaluation RIF (n = 12) vs C (n \n",
      "= 10) Agilent‑052909 CBC lncRNAmRNA V3 |FC|≥ 2 and adj‑P‑value ≤ 0.05 281 [47]\n",
      "Of the respective miRNAs and TFs targeting genes related to endometrial progression and function, 417 miRNAs and 467 TFs were significantly over- represented in at least one gene list (FDR < 0.05, Tables S3-S4). Further, most signatures were significantly governed by TFs rather than miRNAs (4 > odds ratio > 164, FDR < 0.05, 17 lists [89%]), with the exception of Koot’s signature that was governed by miRNAs and showed the opposite (odds ratio = 0.35, FDR < 0.05, Table S5). Eight out of nineteen gene lists (47%) had significant differences between the proportion of genes regulated by progesterone versus estrogen. (*FDR ≤ 0.05). Descriptive analysis of hormonal regulation, normalized by the size of each gene list. Gene lists clustered according to higher (blue) or lower (orange) contribution of hormonal regulation\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \n",
      "\n",
      "Hsa‑miR‑15a‑5p, hsa‑miR‑128‑3p, hsa‑miR‑218‑5p, \n",
      "hsa‑miR‑27b‑3p, hsa‑miR‑107, hsa‑miR‑424‑5p, \n",
      "hsa‑miR‑195‑5p, hsa‑miR‑103a‑3p, has‑let‑7b‑5p and hsa‑miR‑22‑3p are key players in the regulation of endometrial progression and function\n",
      "\n",
      "To bridge the gap of endometrial-based evidence in DoRothEA and TarBase databases, we aimed to identify the common miRNAs that mediate gene expression related to endometrial progression and function. We evaluated the number of endometrial gene lists that each miRNA controlled (Fig. 2A). This strategy highlighted the most influential miRNAs that were commonly found across the endometrial gene lists. Notably, 85.80% of miRNAs regulated genes from one to three signatures, and only 59 miRNAs (14.20%) were associated with more than three signatures. Ultimately, 24 miRNAs were prioritized for regulating genes of at least six signatures.\n",
      "\n",
      "Next, we validated the gene expression trends for these 24 prioritized miRNAs in an independent endometrial gene expression dataset (GSE44730) (Fig. 2B). This dataset included the miRNA expression in endometrial biopsies from 20 healthy women in different phases of the menstrual cycle. We found that two of our prioritized miRNAs were not expressed in the endometrial tissue, and only 10 miRNAs (i.e., hsa-miR-15a-5p, hsa-miR-128-3p, hsa-miR-218-5p, hsa-miR-27b-3p, hsa-miR-107, hsa-miR-424-5p, hsa-miR-195-5p, hsa-miR-103a-3p, has-let-7b-5p and hsa-miR-22-3p) showed significant changes across the menstrual cycle (Fig. 2B). Of these, only the expression of hsa-miR-22-3p significantly increased from ESE to MSE stages (FDR = 0.02, Fig. 2B), highlighting the potential inhibitory role of this miRNA in endometrial function. The expression of the remaining miRNAs decreased during the MSE phase.\n",
      "\n",
      "CTCF and GATA6 as distinguished transcriptional regulators of the human menstrual cycle\n",
      "\n",
      "We repeated the regulation network analysis with the TFs, to identify the prominent TFs commonly regulating the genes related to endometrial progression and function in the reported signatures. We observed 78.80% of the TFs regulated genes in up to 10 signatures, and only 99 TFs (21.20%) regulated genes in over 10 (Fig. 3A). We highlighted CTCF (a CCCTC-binding factor, which functions as a transcriptomic repressor) as the TF with the most influential regulation of endometrial progression and function across 95% of studies (18 gene lists, excluding that of Punyadeera, Fig. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53. Notably, GATA6 was selected due to novel association with endometrial progression and function.\n",
      "To infer the expression of CTCF and GATA6  in \n",
      "endometrial tissue throughout the menstrual cycle, \n",
      "we analysed their expression using a relevant endo - \n",
      "mentrial dataset recently created by our group [30] \n",
      "(Table S1 ). We observed that, relative to the global \n",
      "expression of endometrial genes, CTCF was broadly \n",
      "expressed while GATA6  was moderately expressed. \n",
      "Indeed, CTCF was significantly down-regulated \n",
      "between the PF and ESE phases (FDR = 2.10E-03)  \n",
      "and ESE to the MSE phase (FDR = 2.10E-04), while \n",
      "GATA6 expression increased from the PF to the \n",
      "MSE phase (FDR = 1.70E-11) and from the ESE  \n",
      "to MSE phase (FDR = 9.80E-15). These findings \n",
      "supported the role of these TFs at the beginning \n",
      "of the menstrual cycle and during the MSE phase \n",
      "(Fig. 3 B). \n",
      "These in-silico  validations were then prospec - \n",
      "tively corroborated by RT-qPCR analysis in an inde - \n",
      "pendent sample set of 20 endometrial biopsies. We \n",
      "observed a similar expression profile of CTCF and \n",
      "GATA6 throughout the menstrual cycle, with signifi-\n",
      "\n",
      "We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with significant changes between the endometrial phases. Notably, CTCF expression was decreased between the PF/ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C). The prioritized miRNAs included has‑miR‑16‑5p, has‑miR‑138‑5p, has‑miR‑21‑3p, has‑miR‑205‑5p, has‑miR‑27a‑5p, has‑miR‑15a‑5p, has‑miR‑155‑5p, has‑miR‑129–2‑3p, has‑miR‑147a, has‑miR‑146a‑5p, has‑miR‑107, has‑miR‑424‑5p, has‑miR‑203a‑3p, has‑miR‑195‑5p, has‑miR‑124‑3p, has‑miR‑128‑3p, has‑miR‑22‑3p, has‑miR‑27b‑3p, has‑let‑7b‑5p, has‑miR‑1343‑3p, has‑miR‑1‑3p, has‑miR‑103a‑3p, has‑miR‑23b‑3p, has‑miR‑218‑5p. Significant changes between phases are denoted with asterisks (* FDR ≤ 0.05, ** FDR ≤ 0.01.\n",
      "Fig. 2 (See legend on previous page.)\n",
      "Page 9 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \n",
      "\n",
      "Fig. 3 Transcription factors universally regulating endometrial progression and function and their expression throughout the menstrual cycle. A Regulatory network of TFs forming relationships with numerous gene lists (2–18). TFs targeting specific gene lists were placed in the center of the network. CTCF stood out as the only TF that targeted genes belonging to 18/19 (95%) different signatures. B In-silico analysis of CTCF and GATA6 expression throughout the menstrual cycle. Significant changes between phases are marked with asterisks (**FDR ≤ 0.01, ***FDR ≤ 0.001). C RT‑qPCR validation. Bar plot shows the relative mRNA expression of CTCF and GATA6 in endometrial biopsies collected throughout the menstrual cycle. Data is presented as a mean ± SD. Significant differences of CTCF (*FDR < 0.05) and GATA6 (***FDR < 0.001) expression were observed between proliferative/early‑secretory (PF/ESE) and mid‑secretory/late‑secretory phases (MSE/LSE). Abbreviations in (B and C): PF, Proliferative; ESE, early‑secretory; MSE, mid‑secretory; LSE, late‑secretory; SD, standard deviation\n",
      "This data-driven approach exposed the common transcriptional regulators among 19 studies who proposed variable biomarkers of endometrial progression and function. In this study, we focused on understanding the relative contribution of both the hormonal and non-hormonal regulation, from an alternative holistic perspective. We applied data-driven hypothesis research that, unlike the traditional scientific method, allowed us to generate new hypotheses based on all available biologically-relevant knowledge [16], observing the molecular relationships from a wider scale view. Besides, some traditional molecular procedures such as PCRs are also performed to corroborate the insight uncovered with this approach. We highlighted a larger influence of progesterone-related genes and TFs in endometrial progression and observed a larger contribution of miRNAs and estrogen-related genes in endometrial pathology. Furthermore, we unveiled CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-miR-103a-3p and hsa-miR-128-3p as master regulators in the endometrium, and these findings were reinforced by the dynamic changes in their expression throughout the menstrual cycle (dually validated with in-silico and prospective analyses). The menstrual cycle is perceived to be under tight regulation by the ovarian hormones, progesterone and estrogen [48]. Supporting the role of progesterone in the secretory endometrium [49], and reinforcing its indispensable role in endometrial regulation [50, 51], we found progesterone and its related genes had a larger influence in menstrual cycle progression than estrogen. As the RIF-related gene lists reported by Altmae2010 and Koot were generally controlled by estrogen, this further suggested that while progesterone could be crucial for a suitable endometrial progression and mid-secretory phase acquisition, estrogenic irregularities may be more conducive to endometrial function and molecular pathology. Indeed, this hypothesis corroborated previous reports of most uterine disorders being estrogen-dependent [44, 52]. Supporting the findings from another recent study from our group, where the Altmae2010 and Koot signatures were classified as highly-predictive signatures to identify endometrial pathology in comparison with a prediction model that detected endometrial progression [7 ]. This study distinguished these signatures (related to implantation failure and unexplained infertility) based on the magnitude of their miRNA and estrogen-regulation, which contrasted with the prominent influence of the TFs over the remaining signatures more related to endometrial progression. All these facts reinforce the hypothesis that implantation failure is mainly caused by impaired miRNA expression [53].\n",
      "This study transcended the simplistic paradigm of the bihormonal regulation of endometrial function by identifying a plethora of mediators, including 595 miRNAs and 311 TFs, that were previously unrelated to these ovarian hormones. These findings set the foundation for new discoveries, revealing alternative pathways and new actors in the regulation of the human endometrium, and deepening our understanding of the complex regulatory mechanisms behind endometrial progression and unexplained infertility. MicroRNAs are well known to be involved in embryo implantation and receptivity control. Using network analysis, we unveiled the 10 common miRNAs regulating endometrial progression and function across published biomarker signatures (hsa-miR-22-3p, hsa-miR-107, hsa-miR-103a-3p, hsa-miR-128-3p, hsa-miR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, hsa-miR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p).\n",
      "Their significant gene expression changes in the secre - \n",
      "tory endometrium support their potential role in the \n",
      "endometrial regulation through an inhibition during \n",
      "the WOI. Notably, hsa-miR-424-5p, hsa-miR-27b-3p \n",
      "and hsa-miR-195-5p have been previously associated \n",
      "with RIF [11, 54, 56], while has-let-7b-5p was associ - \n",
      "ated with human endometrial receptivity [57], and \n",
      "hsa-miR-22-3p was overexpressed in a decidualization \n",
      "model [58]. Moreover, hsa-miR-15a-5p was previously \n",
      "associated with endometriosis [59]. Notwithstanding, \n",
      "this study is the first to associate hsa-miR-15a-5p, hsa- \n",
      "miR-218-5p, hsa-miR-103a-3p, hsa-miR-107, and hsa- \n",
      "miR-128-3p to endometrial progression and functional \n",
      "regulation in humans.\n",
      "Page 11 of 14\n",
      "Parraaga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \n",
      "\n",
      "endometrial progression and function were beyond the scope of this project and merit further investigation. CTCF is a conserved zinc finger protein whose regulatory functions are well characterized throughout the human body [64]. CTCF acts as a transcriptional repressor in RNA polymerase II (Pol II) pausing and imprinting and X-chromosome inactivation [65], as well as an insulator, blocking the interaction between enhancers and the promoters of neighbouring genes [66]. Considering these attributed functions and its variable expression throughout the menstrual cycle, we propose that CTCF exerts an inhibitory role in endometrial tissue during the PF phase of the menstrual cycle. With the significant downregulation of CTCF in the secretory endometrium, dually validated in-silico and experimentally herein, its inhibited genes would be derepressed and become transcriptionally active during the WOI. This interpretation supports previous findings from our group demonstrating that, during the WOI, a global transcriptional derepression may be required for implantation and early embryo development [63]. Transcriptional derepression has been associated with multiple human disease states and should be investigated further within the context of endometrial-factor infertility. Despite previous associations of CTCF with endometriosis [67], its implication in endometrial progression and function has not been directly proposed until now. Nevertheless, the role of CTCF in endometrial receptivity has been revealed through its interaction with HOXA10, a gene mainly expressed in endometrium involved in functions such as endometrial proliferation and differentiation, the formation of pinopodes or embryo implantation [68]. Indeed, this study proposed that an overexpression of CTCF can lead to a drop in HOXA10 expression, affecting endometrial proliferation and endometrial function and corroborating the fact that CTCF must decrease its expression as we can observe in our findings. GATA6, the second prioritized TF that overlapped across the gene lists, is a transcription factor belonging to the GATA family, a highly conserved family of six zinc finger proteins [69]. Besides its essential role in embryonic development [70], GATA6 plays a key role in endometriosis and regulates steroidogenic genes [71]. Together with evidence supporting both an activating and inhibitory role for GATA6 [72, 73], the significant changes we observed in its expression across the menstrual cycle reflect its presumed involvement in the endometrium and suggest an activating role in endometrial progression and endometrial receptivity acquisition. Like CTCF, this is the first time that GATA6 is proposed as a key TF in endometrial progression and function.\n",
      "endometrial regulation remain elusive, emerging experimental technologies such as High-throughput Chromosome Conformation Capture (Hi-C) or chromatin interaction analysis with paired-end tags (ChIA-PET) that respectively study general chromatin interactions or chromatin interactions involving a specific protein [74, 75] may improve our understanding of genetic regulation in the future. Nevertheless, as collaborative efforts unveil new regulators and gene targets in humans (such as the ENCODE project [76]), the complexities of transcriptional regulation can be exposed by in-silico analyses like those described herein, and we can gain a better understanding of the processes involved in menstrual cycle progression and endometrial competence. Furthermore, it should be noted that results of gene expression studies are influenced by different variables, such as the experimental platforms employed, study designs, or patients’ characteristics of each independent\n",
      "expression studies are influenced by different variables, such as the experimental platforms employed, study designs, or patients’ characteristics of each independent study, adding variability [30, 77, 78]. Despite this fact, we have found that our holistic approach overcomes this undesired variability effect by finding common upstream master regulators across studies and predicting new relationships between hormones, TFs and miRNAs. This approach also provides the basis for future single-molecule studies that aim to elucidate new regulatory pathways in endometrial progression and function. Our results lay the foundation for further molecular studies that can validate the function(s) of the prevailing endometrial regulators we prioritized. The study shows for the first time, the relative contribution of estrogens, progesterone, TFs and miRNAs in endometrial function and progression. Endometrial progression is mainly influenced by progesterone-related genes and TFs, whereas miRNAs and estrogen-related genes play a larger role in endometrial pathology. Moreover, we highlight novel common transcriptional regulators such as CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-miR-103a-3p, and hsa-miR-128-3p across 19 studies that propose biomarkers in endometrial regulation. These results reveal the molecular mechanism underlying endometrial regulation and lay the foundation for the development of targeted therapies for patients with endometrial-factor infertility.\n",
      "We reviewed 19 publications related to endometrial progression and function, extracted the gene signatures from the original publication, and created corresponding gene lists that we could annotate. We performed different functional analyses based on the type of regulator (i.e., ovarian hormones, transcription factors and miRNAs). We used online databases to obtain the information of the regulatory process, at the hormonal (Gene Ontology, Kyoto Encyclopedia of Genes and Genomes, Dorothea), transcription factor (Dorothea) and miRNA (Tarbase) level and analysed the results accordingly. Finally, we employed an integrative analysis and prioritized the most relevant regulators for subsequent validation. TF, Transcription factors; miRNA, microRNA.\n",
      "Additional file 5: Supplementary Table S4. Results of TFs enrichment. Excel file with the over‑representation analysis results of the transcription factors (TFs). Each sheet contains the results of a specific list of genes. In each sheet, the following information is included: ID, gene symbol of the TF; GeneRatio, the proportion of TF targets in the gene list; BgRatio, the proportion of TF targets in the whole genome; OddsRatio; the odd ratios for each TF, pvalue; the P‑value obtained for each TF; padjust, P‑value adjusted by FDR for each TF; geneID, the gene symbol ID of the TF targets. Additional file 6: Supplementary Table S5. Relative contribution of each regulator type (TFs or miRNAs) in each gene list. A Fisher’s test was performed to calculate the proportional differences between transcription factors (TFs) and microRNAs (miRNAs). Resulting P‑values were adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene\n",
      "Resulting P‑values were adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene lists were mainly regulated by TFs, whereas odds ratios < 1 indicate a higher regulation by miRNAs. Odds Ratio with a value of “Inf” indicate that miRNAs were not found for this gene list. Percentages indicate the proportion between the significant regulators in each gene signature and regulators with at least one target gene in the corresponding signature. (***FDR ≤0.001).\n",
      "The data used to analyze the TF and miRNA expression throughout the men‑\n",
      "tstrual cycle and in infertility are available in Gene Expression Omnibus, with \n",
      "IDs GSE98386, GSE29981, GSE4888, GSE119209, GSE86491, and GSE44558. The raw endometrial gene expression data (from previous studies) was downloaded from the publicly‑available functional genomic dataset repository Gene Expression Omnibus (GEO) [79] for curation/processing. In accordance with GEO policies, all patient data was anonymized and encrypted, and therefore an additional institutional review board approval was not required for our in-silico validation of regulator gene expression changes in endometrial tissue.\n",
      "tional institutional review board approval was not required for our in-silico validation of regulator gene expression changes in endometrial tissue. On the other hand, for the experimental validation of gene expression changes in endometrial biopsies, all individuals gave written informed consent for endometrial gene expression evaluation prior to participating (Ethics Committee of the Instituto Valenciano de Infertilidad [IVI Valencia], Valencia, Spain [1810‑FIVI‑066‑PD/1706‑FIVI‑048‑PD]).\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \n",
      "\n",
      "Author details\n",
      "1 IVIRMA Global Research Alliance, IVI Foundation, Instituto de Investigación Sanitaria La Fe (IIS La Fe), Av. Fernando Abril Martorell 106, Torre A, Planta 1ª, 46026 Valencia, Valencia, Spain. 2 Department of Pediatrics, Obstetrics and Gynaecology, Universidad de Valencia, Av. Blasco Ibáñez 15, 46010 Valen‑cia, Valencia, Spain. 3 IVIRMA Global Research Alliance, IVIRMA Valencia, Plaza de La Policia Local 3, 46015 Valencia, Spain.\n",
      "Received: 21 June 2023 Accepted: 22 August 2023\n",
      "\n",
      "References\n",
      "1. Strowitzki T, Germeyer A, Popovici R, von Wolff M. The human endometrium as a fertility-determining factor. Hum Reprod Update. 2006;12:617–30.\n",
      "2. Cakmak H, Taylor HS. Implantation failure: Molecular mechanisms and clinical treatment. Hum Reprod Update. 2011;17:242–53.\n",
      "3. Lessey BA, Young SL. What exactly is endometrial receptivity? Fertil Steril. 2019;111:611–7.\n",
      "4. Talbi S, Hamilton AE, Vo KC, Tulac S, Overgaard MT, Dosiou C, et al. Molecular phenotyping of human endometrium distinguishes menstrual cycle phases and underlying biological processes in normo‑ovulatory women. Endocrinology. 2006;147:1097–121.\n",
      "5. Somigliana E, Vigano P , Busnelli A, Paffoni A, Vegetti W, Vercellini P . Repeated implantation failure at the crossroad between statistics, clinics and over‑diagnosis. Reprod Biomed Online. 2018;36:32–8.\n",
      "6. Craciunas L, Gallos I, Chu J, Bourne T, Quenby S, Brosens JJ, et al. Conventional and modern markers of endometrial receptivity: a systematic review and meta‑analysis. Hum Reprod Update. 2019;25:202–23.\n",
      "7. Sebastian‑Leon P , Garrido N, Remohí J, Pellicer A, Diaz‑Gimeno P . Asynchronous and pathological windows of implantation: Two causes of recurrent implantation failure. Hum Reprod. 2018;33:626–35.\n",
      "8. Vasquez YM, DeMayo FJ. Role of nuclear receptors in blastocyst implantation. Semin Cell Dev Biol. 2013;24:724–35.\n",
      "9. Wang H, Zhang S, Lin H, Kong S, Wang S, Wang H, et al. Physiological and molecular determinants of embryo implantation. Mol Aspects Med. 2013;34:939–80.\n",
      "10. Kelleher AM, Behura SK, Burns GW, Young SL, Demayo FJ, Spencer TE. Integrative analysis of the forkhead box A2 (FOXA2) cistrome for the human endometrium. FASEB J. 2019;33:8543–54.\n",
      "11. Revel A, Achache H, Stevens J, Smith Y, Reich R. MicroRNAs are associated with human embryo implantation defects. Hum Reprod. 2011;26:2830–40.\n",
      "12. Galliano D, Pellicer A. MicroRNA and implantation. Fertil Steril. 2014;101:1531–44.\n",
      "13. Xu H, Zhou M, Cao Y, Zhang D, Han M, Gao X, et al. Genome‑wide analysis of long noncoding RNAs, microRNAs, and mRNAs forming a competing endogenous RNA network in repeated implantation failure. Gene. 2019;720: 144056.\n",
      "14. Copley RR. The animal in the genome: comparative genomics and evolution. Philos Trans R Soc Lond B: Biol Sci. 2008;363:1453–61.\n",
      "Breaking the ageing paradigm in endometrium: endometrial gene expression related to cilia and ageing hallmarks in women over 35 years.\n",
      "Meta‑signature of human endometrial receptivity: a meta‑analysis and validation study of transcriptomic biomarkers.\n",
      "Chapter 15 ‑ the molecular signature of the endometrial receptivity: research and clinical application. In: Horcajadas JA, Gosálvez J, editors. Reproductomics: the omics revolu‑tion and its impact on human reproductive medicine. Academic Press; 2018. p. 301–279.\n",
      "\n",
      "36. Díaz‑Gimeno P , Horcajadas JA, Martínez‑Conejero JA, Esteban FJ, Alamá P , Pellicer A, et al. A genomic diagnostic tool for human endometrial receptivity based on the transcriptomic signature. Fertil Steril. 2011;95:50.\n",
      "\n",
      "37. Kao LC, Tulac S, Lobo S, Imani B, Yang JP , Germeyer A, et al. Global gene profiling in human endometrium during the window of implantation. Endocrinology. 2002;143:2119–38.\n",
      "\n",
      "38. Ponnampalam AP , Weston GC, Trajstman AC, Susil B, Rogers PAW. Molecular classification of human endometrial cycle stages by transcriptional profiling. Mol Hum Reprod. 2004;10:879–93.\n",
      "\n",
      "39. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al.\n",
      "lar classification of human endometrial cycle stages by transcriptional profiling. Mol Hum Reprod. 2004;10:879–93. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al. Oestrogen‑modulated gene expression in the human endometrium. Cell Mol Life Sci. 2005;62:239–50. Riesewijk A, Martín J, van Os R, Horcajadas JA, Polman J, Pellicer A, et al. Gene expression profiling of human endometrial receptivity on days LH+2 versus LH+7 by microarray technology. Mol Hum Reprod. 2003;9:253–64. Bastu E, Demiral I, Gunel T, Ulgen E, Gumusoglu E, Hosseini MK, et al. Potential marker pathways in the endometrium that may cause recurrent implantation failure. Reprod Sci. 2019;26:879–90. Bersinger NA, Wunder DM, Birkhäuser MH, Mueller MD. Gene expression in cultured endometrium from women with different outcomes follow‑following IVF. Mol Hum Reprod. 2008;14:475–84.\n",
      "Page 14 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "•\n",
      "\n",
      "fast, convenient online submission\n",
      " •\n",
      "\n",
      "thorough peer review by experienced researchers in your ﬁeld\n",
      "• \n",
      "\n",
      "rapid publication on acceptance\n",
      "• \n",
      "\n",
      "support for research data, including large and complex data types\n",
      "\n",
      "gold Open Access which fosters wider collaboration and increased citations\n",
      "\n",
      "maximum visibility for your research: over 100M website views per year •\n",
      "  At BMC, research is always in progress. Learn more biomedcentral.com/submissions\n",
      "Ready to submit y our researc hReady to submit y our researc h  ?  Choose BMC and benefit fr om: ?  Choose BMC and benefit fr om: 43. Bhagwat SR, Chandrashekar DS, Kakar R, Davuluri S, Bajpai AK, Nayak S, et al. Endometrial receptivity: a revisit to functional genomics studies on human endometrium and creation of HGEx ‑ERdb. PLoS ONE. 2013;8: e58419.\n",
      "44. Koot YEM, Van Hooff SR, Boomsma CM, Van Leenen D, Koerkamp MJAG, Goddijn M, et al. An endometrial gene expression signature accurately predicts recurrent implantation failure after IVF. Sci Rep. 2015;2016(6):1–12.\n",
      "45. Lédée N, Munaut C, Aubert J, Sérazin V, Rahmati M, Chaouat GG, et al. Specific and extensive endometrial deregulation is present before conception in IVF/ICSI repeated implantation failures (IF) or recurrent miscarriages. J Pathol. 2011;225:554–64.\n",
      "46. Pathare ADS, Zaveri K, Hinduja I. Downregulation of genes related to immune and inflammatory response in IVF implantation failure cases under controlled ovarian stimulation. Am J Reprod Immunol. 2017;78: e12679.\n",
      "47. Shi C, Han HJ, Fan LJ, Guan J, Zheng XB, Chen X, et al. Diverse endometrial mRNA signatures during the window of implantation in patients with repeated implantation failure. Hum Fertil. 2018;21:183–94.\n",
      "48. Messinisi IE. Ovarian feedback, mechanism of action and possible clinical implications. Hum Reprod Update. 2006;12:557–71.\n",
      "49. Edwards RG. Implantation, interception and contraception. Hum Reprod (Oxford, England). 1994;9:985–95.\n",
      "50. Young SL. Oestrogen and progesterone action on endometrium: a translational approach to understanding endometrial receptivity. Reprod Biomed Online. 2013;27:497–505.\n",
      "51. Bhurke AS, Bagchi IC, Bagchi MK. Progesterone‑regulated endometrial factors controlling implantation. Am J Reprod Immunol. 2016;75:237–45.\n",
      "52. Devesa‑Peiro A, Sebastian‑Leon P , Garcia‑ Garcia F, Arnau V, Aleman A, Pellicer A, et al. Uterine disorders affecting female fertility: what are the molecular functions altered in endometrium? Fertil Steril. 2020;113:1261–74.\n",
      "53. von Grothusen C, Frisendahl C, Modhukur V, Lalitkumar PG, Peters M, Faridani OR, et al. Uterine fluid microRNAs are dysregulated in women with recurrent implantation failure. Hum Reprod. 2022;37:734–46.\n",
      "54. Altmäe S, Martinez‑Conejero JA, Esteban FJ, Ruiz‑Alonso M, Stavreus‑Evers A, Horcajadas JA, et al. MicroRNAs miR‑30b, miR‑30d, and miR‑494\n",
      "regulate human endometrial receptivity.\n",
      "stromal decidualization in vitro. PLoS ONE. 2012;7:41080. Liu D, Liang Y, Chen M, Yang F, Yao S. Knockdown of circ_0075503 sup ‑ presses cell migration and invasion by regulating miR‑15a‑5p and KLF12 in endometriosis. Mol Cell Biochem. 2021;476:3845–56. Chi RPA, Wang T, Adams N, Wu SP , Young SL, Spencer TE, et al. Human endometrial transcriptome and progesterone receptor cistrome reveal important pathways and epithelial regulators. J Clin Endocrinol Metab. 2020;105:1–21. Pawar S, Starosvetsky E, Orvis GD, Behringer RR, Bagchi IC, Bagchi MK. STAT3 regulates uterine epithelial remodeling and epithelial‑stromal crosstalk during implantation. Mol Endocrinol. 2013;27:1996–2012. Cheng YH, Imir A, Suzuki T, Fenkci V, Yilmaz B, Sasano H, et al. SP1 and SP3 mediate progesterone‑ dependent induction of the 17beta hydroxyster ‑ oid dehydrogenase type 2 gene in human endometrium. Biol Reprod. 2006;75:605–14. Sebastian‑Leon P , Devesa‑Peiro A, Aleman A, Parraga‑Leo A, Arnau V, Pellicer A, et al. Transcriptional changes through menstrual cycle reveal a global transcriptional derepression underlying the molecular mechanism involved in the window of implantation. Mol Hum Reprod. 2021. https:// doi. org/ 10. 1093/ molehr/ gaab0 27. Kim S, Yu NK, Kaang BK. CTCF as a multifunctional protein in genome regulation and gene expression. Exp Mol Med. 2015;47: e166. Shukla S, Kavak E, Gregory M, Imashimizu M, Shutinoski B, Kashlev M, et al. CTCF‑promoted RNA polymerase II pausing links DNA methylation to splicing. Nature. 2011;479:74–9. Phillips JE, Corces VGCTCF. Master weaver of the genome. Cell. 2009;137:1194–211. Guo J, Cao B, Xu X, Wu F, Zhu B. Novel CTCF mutations in Chinese patients with ovarian endometriosis. Mol Med Rep. 2018;18:1031–6. Geng J, Cui C, Yin Y, Zhao Y, Zhang C. LncRNA NEAT1 affects endome ‑ trial receptivity by regulating HOXA10 promoter activity. Cell Cycle. 2022;21:1932–44. Viger RS, Guittot SM, Anttonen M, Wilson DB, Heikinheimo M. Role of the GATA family of transcription factors in endocrine development, function, and disease. Mol Endocrinol. 2008;22:781. Xuan S, Sussel L. GATA4 and GATA6 regulate pancreatic endoderm identity through inhibition of hedgehog signaling. Development (Cam‑ bridge). 2016;143:780–6. Bernardi LA, Dyson MT, Tokunaga H, Sison C, Oral M, Robins JC, et al. The essential role of GATA6 in the activation of estrogen synthesis in endome ‑ triosis. Reprod Sci. 2019;26:60. Yin F, Herring BP . GATA‑6 can act as a positive or negative regula ‑ tor of smooth muscle‑specific gene expression *. J Biol Chem. 2005;280:4745–52. Zhai J, Luo G. GATA6‑induced FN1 activation promotes the proliferation, invasion and migration of oral squamous cell carcinoma cells. Mol Med ‑ Rep. 2022;25:102. Fullwood MJ, Liu MH, Pan YF, Liu J, Xu H, Mohamed YB, et al. An oestrogen‑receptor‑α‑bound human chromatin interactome. Nature. 2009;462:58–64.\n",
      "75. Lieberman‑Aiden E, Van Berkum NL, Williams L, Imakaev M, Ragoczy T, Telling A, et al. Comprehensive mapping of long range interactions reveals folding principles of the human genome. Science. 2009;326:289.\n",
      "76. Birney E, Stamatoyannopoulos JA, Dutta A, Guigó R, Gingeras TR, Margulies EH, et al. Identification and analysis of functional elements in 1% of the human genome by the ENCODE pilot project. Nature. 2007;447:799–816.\n",
      "77. Tajti F, Kuppe C, Antoranz A, Ibrahim MM, Kim H, Ceccarelli F, et al. A func‑\n",
      " tional landscape of CKD entities from public transcriptomic data. Kidney Int Rep. 2020. https:// doi. org/ 10. 1016/j. ekir. 2019. 11. 005.\n",
      "78. Henarejos‑ Castillo I, Devesa‑Peiro A, de Miguel‑ Gomez L, Sebastian‑Leon P , Romeu M, Aleman A, et al. Predicted COVID ‑19 molecular effects on \n",
      " endometrium reveal key dysregulated genes and functions. Mol Hum Reprod. 2022.gaac035.\n",
      "79. Barrett T, Wilhite SE, Ledoux P , Evangelista C, Kim IF, Tomashevsky M, et al.\n",
      "endometrium reveal key dysregulated genes and functions.\n"
     ]
    }
   ],
   "source": [
    "# create the chroma client\n",
    "import uuid\n",
    "import chromadb\n",
    "from chromadb.config import Settings\n",
    "\n",
    "# create the open-source embedding function\n",
    "embedding_function = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")\n",
    "\n",
    "client = chromadb.HttpClient(host='localhost', port=8000, settings=Settings(allow_reset=True))\n",
    "# client.list_collections()\n",
    "# client.reset()  # resets the database\n",
    "collection = client.get_or_create_collection(\"tfm_APL\")\n",
    "# collection = client.get_collection('tfm')\n",
    "\n",
    "for i, doc in enumerate(my_json): #storage:\n",
    "    print(doc['text'])\n",
    "    collection.add(\n",
    "        ids=[str(uuid.uuid1())], metadatas=storage[i].metadata, documents=str(doc['text'])\n",
    "    )\n",
    "\n",
    "# tell LangChain to use our client and collection name\n",
    "db = Chroma(\n",
    "    client=client,\n",
    "    collection_name=\"tfm\",\n",
    "    embedding_function=embedding_function,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "ab621f32",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "query = \"What is the role of CTCF?\"\n",
    "docs = db.similarity_search(query,k=4)\n",
    "\n",
    "fullcontent =''\n",
    "\n",
    "for doc in docs:\n",
    "    print(docs)\n",
    "    fullcontent ='. '.join([fullcontent,doc.page_content])\n",
    "#print(docs[0].page_content)\n",
    "#print(len(docs))\n",
    "\n",
    "print(fullcontent)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "a16b9f1e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'author': 'Antonio Parraga-Leo',\n",
      " 'creationdate': '2023-09-12T04:25:09+00:00',\n",
      " 'creator': 'Adobe InDesign 15.1 (Windows)',\n",
      " 'crossmarkdomainexclusive': 'true',\n",
      " 'crossmarkdomains[1]': 'springer.com',\n",
      " 'crossmarkdomains[2]': 'springerlink.com',\n",
      " 'crossmarkmajorversiondate': '2010-04-23',\n",
      " 'doi': '10.1186/s12958-023-01131-4',\n",
      " 'keywords': ';GATA6;Progesterone;Estrogen;miRNAs;TFs;Endometrial '\n",
      "             'receptivity;Recurrent implantation failure;Infertility;Menstrual '\n",
      "             'cycle regulation',\n",
      " 'moddate': '2023-11-19T01:55:46+01:00',\n",
      " 'page': 0,\n",
      " 'page_label': '1',\n",
      " 'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 '\n",
      "             '1T3XT BVBA (SPRINGER SBM; licensed version)',\n",
      " 'robots': 'noindex',\n",
      " 'source': 'Data/parraga-leo_2023.pdf',\n",
      " 'subject': 'Reproductive Biology and Endocrinology, '\n",
      "            'https://doi.org/10.1186/s12958-023-01131-4',\n",
      " 'title': 'Deciphering a shared transcriptomic regulation and the relative '\n",
      "          'contribution of each regulator type through endometrial gene '\n",
      "          'expression signatures',\n",
      " 'total_pages': 14,\n",
      " 'trapped': '/False'}\n"
     ]
    }
   ],
   "source": [
    "uuid.uuid1()\n",
    "pprint.pprint(storage[2].metadata)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "60f90a42",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_system_message_rag(content):\n",
    "        return f\"\"\"You are an expert consultant helping executive advisors to get relevant information from scientific articles and code related to reproduction and bioinformatics.\n",
    "\n",
    "        Generate your response by following the steps below:\n",
    "        1. Recursively break down the question into smaller questions to better understand it.\n",
    "        2. For each question/directive:\n",
    "            2a. Select the most relevant information from the context in light of the conversation history.\n",
    "        3. Generate a draft response using selected information.\n",
    "        4. Remove duplicate content from draft response.\n",
    "        5. Generate your final response after adjusting it to increase accuracy and relevance.\n",
    "        6. Do not try to summarize the answers, explain it properly.\n",
    "        6. Only show your final response! \n",
    "        \n",
    "        Constraints:\n",
    "        1. DO NOT PROVIDE ANY EXPLANATION OR DETAILS OR MENTION THAT YOU WERE GIVEN CONTEXT. Only do that when questions are related to coding.\n",
    "        2. Don't mention that you are not able to find the answer in the provided context.\n",
    "        3. Ignore the part of the content that only contains references.\n",
    "        3. Don't make up the answers by yourself.\n",
    "        4. Try your best to provide answer from the given context.\n",
    "\n",
    "        CONTENT:\n",
    "        {content}\n",
    "        \"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "0281c263",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_ques_response_prompt(question):\n",
    "    return f\"\"\"\n",
    "    ==============================================================\n",
    "    Based on the above context, please provide the answer to the following question:\n",
    "    {question}\n",
    "    \"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "id": "bec23937",
   "metadata": {},
   "outputs": [],
   "source": [
    "from ollama import Client\n",
    "client = Client(host='http://localhost:11434')\n",
    "stream = client.chat(model='gemma3:12b', messages=[\n",
    "{\"role\": \"system\", \"content\": get_system_message_rag(fullcontent)},            \n",
    "{\"role\": \"user\", \"content\": get_ques_response_prompt(query)}\n",
    "],stream=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d86936a",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# stream = client.chat(model='mistral', messages=[\n",
    "#     {\"role\": \"system\", \"content\": get_system_message_rag(content)},            \n",
    "#     {\"role\": \"user\", \"content\": get_ques_response_prompt(query)}\n",
    "#     ],stream=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "ca1140ec",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are an expert consultant helping executive advisors to get relevant information from scientific articles and code related to reproduction and bioinformatics.\n",
      "\n",
      "        Generate your response by following the steps below:\n",
      "        1. Recursively break down the question into smaller questions to better understand it.\n",
      "        2. For each question/directive:\n",
      "            2a. Select the most relevant information from the context in light of the conversation history.\n",
      "        3. Generate a draft response using selected information.\n",
      "        4. Remove duplicate content from draft response.\n",
      "        5. Generate your final response after adjusting it to increase accuracy and relevance.\n",
      "        6. Do not try to summarize the answers, explain it properly.\n",
      "        6. Only show your final response! \n",
      "        \n",
      "        Constraints:\n",
      "        1. DO NOT PROVIDE ANY EXPLANATION OR DETAILS OR MENTION THAT YOU WERE GIVEN CONTEXT. Only do that when questions are related to coding.\n",
      "        2. Don't mention that you are not able to find the answer in the provided context.\n",
      "        3. Ignore the part of the content that only contains references.\n",
      "        3. Don't make up the answers by yourself.\n",
      "        4. Try your best to provide answer from the given context.\n",
      "\n",
      "        CONTENT:\n",
      "        . 1. Introduction\n",
      "Recurrent implantation failure (RIF) is a highly demanding and complex clinical\n",
      "phenomenon that, despite advancements in assisted reproductive technology, remains a\n",
      "major issue affecting approximately 15% of women undergoing in vitro fertilization [1].\n",
      "Despite the extensive relevant literature, there is still no widely accepted definition. The\n",
      "definition that until recently had the broadest acceptance for RIF is that of Coughlan et al.,\n",
      "who define RIF as the failure to achieve clinical pregnancy after the transfer of at least four\n",
      "J. Clin. Med.2024, 13, 2145. https://doi.org/10.3390/jcm13072145 https://www.mdpi.com/journal/jcm. 1. Introduction\n",
      "Recurrent implantation failure (RIF) is a highly demanding and complex clinical\n",
      "phenomenon that, despite advancements in assisted reproductive technology, remains a\n",
      "major issue affecting approximately 15% of women undergoing in vitro fertilization [1].\n",
      "Despite the extensive relevant literature, there is still no widely accepted definition. The\n",
      "definition that until recently had the broadest acceptance for RIF is that of Coughlan et al.,\n",
      "who define RIF as the failure to achieve clinical pregnancy after the transfer of at least four\n",
      "J. Clin. Med.2024, 13, 2145. https://doi.org/10.3390/jcm13072145 https://www.mdpi.com/journal/jcm. 2\n",
      "Vol:.(1234567890)Scientific Reports |        (2024) 14:19822  | https://doi.org/10.1038/s41598-024-70778-9\n",
      "www.nature.com/scientificreports/\n",
      "Recurrent implantation failure (RIF) is a clinical complication that physicians encounter during in-vitro fertiliza-\n",
      "tion (IVF)1. RIF is defined by the inability to achieve a clinical pregnancy after the transfer of at least five high-\n",
      "quality embryos in at least two or more IVF cycles, according to the European Society of Human Reproduction \n",
      "and Embryology (ESHRE)  guidelines2. This condition suggests that the embryo does not implant successfully in \n",
      "the uterine lining. In contrast, recurrent miscarriage is characterized by the loss of two or more clinical pregnan-\n",
      "cies, which means the embryo initially implants and starts to develop but is subsequently lost. These conditions \n",
      "require different diagnostic and therapeutic approaches due to their distinct natures and underlying  causes3. RIF \n",
      "could be related to the type and quality of clinical action and also the population of cases studied. Implantation \n",
      "failure may also be related to many factors that are classified into three categories: reduced endometrial recep-\n",
      "tivity, embryonic defects and multifactorial  causes4. RIF cases may also be idiopathic, which means there is no \n",
      "identifiable problem in  parents5. Many genes and proteins that are expressed during the window of implantation \n",
      "have been identified in earlier  studies6,7. Up-regulated genes that related to endometrial receptivity are recognized \n",
      "to adjust cytokine-cytokine interaction pathways, complement and coagulation cascades, extracellular membrane \n",
      "receptor interaction, and inhibition of matrix-metalloproteinase  pathway8–11. Expression of these genes prior \n",
      "to the window of implantation may prepare the endometrium in terms of structural and functional for embryo \n",
      "attachment. However, the ways to control such a complex set of genes involved in the regulation of endometrial \n",
      "receptivity remains largely  unknown7. Generally, 62.25% of the receptivity-associated genes that up-regulated \n",
      "during window of implantation, encode the extracellular and plasma membrane  proteins12. These proteins may \n",
      "be important in embryo adhesion and attendant signal transmission pathways. The important roles of ion chan-\n",
      "nels in indirect activation of signal pathways that regulate endometrial receptivity and embryo implantation have \n",
      "been emphasized in recent  studies7,13–15.\n",
      "Endometrial ion channels play a major role in endometrial receptivity regulation and embryo  implantation16. \n",
      "Many ion channels are involved in embryo implantation due to their role in the regulation of the volume of intra \n",
      "uterine fluid, decidualization and indirectly expression of the implantation associated  genes17. Abnormal expres-\n",
      "sion or action of ion channels in the endometrium can lead to disruption of endometrial receptivity and embryo \n",
      "implantation failure. To date, more than 14 types of ion channels have been identified in human endometrium, \n",
      "including Epithelial Na+ Channel (ENaC), Cystic Fibrosis Transmembrane Conductance Regulator (CFTR) and \n",
      "various types of calcium and potassium  channels18–20.\n",
      "Uterine secretion absorption during pre-implantation phase is an innate phenomenon in many species that \n",
      "enables the embryo to stay near the uterine lumen before the penetration and implantation processes and to \n",
      "maintain its connection with the uterine  endometrium21. Earlier studies have shown that ENaC is necessary \n",
      "to absorb uterine fluid and is thought to be responsible for decreasing uterine fluid during the pre-implan-\n",
      "tation phase. In recent years, the essential role of ENaC in the onset of embryo implantation has come to be \n",
      " recognized17. It has been shown that ENaC is activated by a protease that is released from the embryo, resulting \n",
      "a series of events occurring in endometrial epithelial cells including membrane depolarization events, activa-. 2\n",
      "Vol:.(1234567890)Scientific Reports |        (2024) 14:19822  | https://doi.org/10.1038/s41598-024-70778-9\n",
      "www.nature.com/scientificreports/\n",
      "Recurrent implantation failure (RIF) is a clinical complication that physicians encounter during in-vitro fertiliza-\n",
      "tion (IVF)1. RIF is defined by the inability to achieve a clinical pregnancy after the transfer of at least five high-\n",
      "quality embryos in at least two or more IVF cycles, according to the European Society of Human Reproduction \n",
      "and Embryology (ESHRE)  guidelines2. This condition suggests that the embryo does not implant successfully in \n",
      "the uterine lining. In contrast, recurrent miscarriage is characterized by the loss of two or more clinical pregnan-\n",
      "cies, which means the embryo initially implants and starts to develop but is subsequently lost. These conditions \n",
      "require different diagnostic and therapeutic approaches due to their distinct natures and underlying  causes3. RIF \n",
      "could be related to the type and quality of clinical action and also the population of cases studied. Implantation \n",
      "failure may also be related to many factors that are classified into three categories: reduced endometrial recep-\n",
      "tivity, embryonic defects and multifactorial  causes4. RIF cases may also be idiopathic, which means there is no \n",
      "identifiable problem in  parents5. Many genes and proteins that are expressed during the window of implantation \n",
      "have been identified in earlier  studies6,7. Up-regulated genes that related to endometrial receptivity are recognized \n",
      "to adjust cytokine-cytokine interaction pathways, complement and coagulation cascades, extracellular membrane \n",
      "receptor interaction, and inhibition of matrix-metalloproteinase  pathway8–11. Expression of these genes prior \n",
      "to the window of implantation may prepare the endometrium in terms of structural and functional for embryo \n",
      "attachment. However, the ways to control such a complex set of genes involved in the regulation of endometrial \n",
      "receptivity remains largely  unknown7. Generally, 62.25% of the receptivity-associated genes that up-regulated \n",
      "during window of implantation, encode the extracellular and plasma membrane  proteins12. These proteins may \n",
      "be important in embryo adhesion and attendant signal transmission pathways. The important roles of ion chan-\n",
      "nels in indirect activation of signal pathways that regulate endometrial receptivity and embryo implantation have \n",
      "been emphasized in recent  studies7,13–15.\n",
      "Endometrial ion channels play a major role in endometrial receptivity regulation and embryo  implantation16. \n",
      "Many ion channels are involved in embryo implantation due to their role in the regulation of the volume of intra \n",
      "uterine fluid, decidualization and indirectly expression of the implantation associated  genes17. Abnormal expres-\n",
      "sion or action of ion channels in the endometrium can lead to disruption of endometrial receptivity and embryo \n",
      "implantation failure. To date, more than 14 types of ion channels have been identified in human endometrium, \n",
      "including Epithelial Na+ Channel (ENaC), Cystic Fibrosis Transmembrane Conductance Regulator (CFTR) and \n",
      "various types of calcium and potassium  channels18–20.\n",
      "Uterine secretion absorption during pre-implantation phase is an innate phenomenon in many species that \n",
      "enables the embryo to stay near the uterine lumen before the penetration and implantation processes and to \n",
      "maintain its connection with the uterine  endometrium21. Earlier studies have shown that ENaC is necessary \n",
      "to absorb uterine fluid and is thought to be responsible for decreasing uterine fluid during the pre-implan-\n",
      "tation phase. In recent years, the essential role of ENaC in the onset of embryo implantation has come to be \n",
      " recognized17. It has been shown that ENaC is activated by a protease that is released from the embryo, resulting \n",
      "a series of events occurring in endometrial epithelial cells including membrane depolarization events, activa-\n",
      "        \n",
      "\n",
      "    ==============================================================\n",
      "    Based on the above context, please provide the answer to the following question:\n",
      "    What is the implantation failure and what is its cause? Give me a detail explanation of implantation failure, not only about RIF\n",
      "    \n",
      "####### THINKING OF ANSWER............ \n"
     ]
    }
   ],
   "source": [
    "\n",
    "print(get_system_message_rag(fullcontent))\n",
    "print(get_ques_response_prompt(query))\n",
    "print(\"####### THINKING OF ANSWER............ \")\n",
    "full_answer = ''\n",
    "\n",
    "for chunk in stream:\n",
    "    # print(chunk['message']['content'], end='', flush=True)\n",
    "    full_answer =''.join([full_answer,chunk['message']['content']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "id": "fdd86269",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Implantation failure refers to the inability of an embryo to successfully '\n",
      " 'attach to and be supported by the uterine lining (endometrium), preventing '\n",
      " 'pregnancy from progressing. This can occur in the context of Recurrent '\n",
      " \"Implantation Failure (RIF), but it's a broader phenomenon that can also \"\n",
      " 'contribute to recurrent miscarriage.\\n'\n",
      " '\\n'\n",
      " '**RIF Definition:** According to the European Society of Human Reproduction '\n",
      " 'and Embryology (ESHRE) guidelines, RIF is defined as the inability to '\n",
      " 'achieve a clinical pregnancy after the transfer of at least five '\n",
      " 'high-quality embryos in at least two or more IVF cycles. This suggests that '\n",
      " \"the embryo doesn't implant properly.\\n\"\n",
      " '\\n'\n",
      " \"**Distinction from Recurrent Miscarriage:** It's important to differentiate \"\n",
      " 'implantation failure from recurrent miscarriage. In recurrent miscarriage, '\n",
      " 'the embryo initially implants and begins to develop, but is subsequently '\n",
      " 'lost. These conditions require different diagnostic and therapeutic '\n",
      " 'approaches.\\n'\n",
      " '\\n'\n",
      " '**Causes of Implantation Failure:** Implantation failure can stem from '\n",
      " 'various factors, broadly categorized as:\\n'\n",
      " '\\n'\n",
      " '*   **Reduced Endometrial Receptivity:** This refers to issues with the '\n",
      " 'endometrium that prevent it from adequately supporting embryo implantation.  '\n",
      " 'Numerous genes and proteins expressed during the window of implantation play '\n",
      " 'a role, affecting pathways like cytokine-cytokine interaction, complement '\n",
      " 'and coagulation cascades, extracellular membrane receptor interaction, and '\n",
      " 'inhibition of the matrix-metalloproteinase pathway.  Approximately 62.25% of '\n",
      " 'receptivity-associated genes encode extracellular and plasma membrane '\n",
      " 'proteins that are crucial for embryo adhesion and signal transmission. '\n",
      " 'Abnormal expression or function of ion channels, such as Epithelial Na+ '\n",
      " 'Channel (ENaC), Cystic Fibrosis Transmembrane Conductance Regulator (CFTR), '\n",
      " 'and various calcium and potassium channels, can disrupt endometrial '\n",
      " 'receptivity.\\n'\n",
      " '*   **Embryonic Defects:** Problems with the embryo itself can hinder '\n",
      " \"implantation. While not explicitly detailed in the provided text, it's \"\n",
      " 'acknowledged as a potential cause.\\n'\n",
      " '*   **Multifactorial Causes:** These are situations where multiple factors '\n",
      " 'contribute to implantation failure.\\n'\n",
      " '*   **Idiopathic Causes:** In some cases, the cause of implantation failure '\n",
      " 'remains unknown (idiopathic).\\n'\n",
      " '\\n'\n",
      " '**Role of Ion Channels:** Endometrial ion channels play a significant role '\n",
      " 'in endometrial receptivity and embryo implantation. They impact uterine '\n",
      " 'fluid volume regulation, decidualization, and the expression of '\n",
      " 'implantation-associated genes.  Uterine secretion absorption during the '\n",
      " 'pre-implantation phase is also crucial, with ENaC considered responsible for '\n",
      " 'decreasing uterine fluid and activated by a protease released from the '\n",
      " 'embryo.')\n"
     ]
    }
   ],
   "source": [
    "pprint.pprint(full_answer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "9bb19adb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on method chat in module ollama._client:\n",
      "\n",
      "chat(model: str = '', messages: Optional[Sequence[Union[Mapping[str, Any], ollama._types.Message]]] = None, *, tools: Optional[Sequence[Union[Mapping[str, Any], ollama._types.Tool, Callable]]] = None, stream: bool = False, think: Optional[bool] = None, format: Union[Literal['', 'json'], dict[str, Any], NoneType] = None, options: Union[Mapping[str, Any], ollama._types.Options, NoneType] = None, keep_alive: Union[float, str, NoneType] = None) -> Union[ollama._types.ChatResponse, collections.abc.Iterator[ollama._types.ChatResponse]] method of ollama._client.Client instance\n",
      "    Create a chat response using the requested model.\n",
      "    \n",
      "    Args:\n",
      "      tools:\n",
      "        A JSON schema as a dict, an Ollama Tool or a Python Function.\n",
      "        Python functions need to follow Google style docstrings to be converted to an Ollama Tool.\n",
      "        For more information, see: https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings\n",
      "      stream: Whether to stream the response.\n",
      "      format: The format of the response.\n",
      "    \n",
      "    Example:\n",
      "      def add_two_numbers(a: int, b: int) -> int:\n",
      "        '''\n",
      "        Add two numbers together.\n",
      "    \n",
      "        Args:\n",
      "          a: First number to add\n",
      "          b: Second number to add\n",
      "    \n",
      "        Returns:\n",
      "          int: The sum of a and b\n",
      "        '''\n",
      "        return a + b\n",
      "    \n",
      "      client.chat(model='llama3.2', tools=[add_two_numbers], messages=[...])\n",
      "    \n",
      "    Raises `RequestError` if a model is not provided.\n",
      "    \n",
      "    Raises `ResponseError` if the request could not be fulfilled.\n",
      "    \n",
      "    Returns `ChatResponse` if `stream` is `False`, otherwise returns a `ChatResponse` generator.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(client.chat)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.21"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
