{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "18772ea0",
   "metadata": {},
   "source": [
    "# Prepare documents for RAG"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "579d11f6",
   "metadata": {},
   "source": [
    "## Libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "c3b57a69",
   "metadata": {},
   "outputs": [],
   "source": [
    "import langchain_community\n",
    "import langchain_text_splitters\n",
    "from langchain_community.document_loaders import PyPDFLoader, pdf\n",
    "from langchain_community.embeddings.sentence_transformer import (\n",
    "    SentenceTransformerEmbeddings,\n",
    ")\n",
    "from langchain_community.vectorstores import Chroma\n",
    "from langchain_text_splitters import CharacterTextSplitter, RecursiveCharacterTextSplitter\n",
    "import os\n",
    "import pprint\n",
    "import re\n",
    "from langchain_core.documents import Document\n",
    "from langchain_core.prompts.prompt import PromptTemplate\n",
    "from langchain_core.output_parsers import JsonOutputParser\n",
    "from langchain_ollama import ChatOllama\n",
    "from pydantic import BaseModel, Field\n",
    "import json\n",
    "import uuid\n",
    "import chromadb\n",
    "from chromadb.config import Settings\n",
    "import unicodedata\n",
    "from langchain_google_genai import GoogleGenerativeAI\n",
    "import uuid\n",
    "# from chromadb.utils.embedding_functions import SentenceTransformerEmbeddingFunction\n",
    "import pickle as pkl\n",
    "import requests\n",
    "import subprocess\n",
    "import pandas as pd\n",
    "from bs4 import BeautifulSoup"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "328c2061",
   "metadata": {},
   "source": [
    "## Google option"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9bc6ea96",
   "metadata": {},
   "source": [
    "#### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7be2125f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/antonioparragaleo/Documents/RAG'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "357e9e5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('api_google.txt') as f:\n",
    "    \n",
    "    api_key = json.load(f)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9b413d0a",
   "metadata": {},
   "source": [
    "#### Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7784f8c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def clean_text(text):\n",
    "    # Normalize weird Unicode characters to their closest ASCII equivalent\n",
    "    text = unicodedata.normalize(\"NFKC\", text)\n",
    "    # Replace non-breaking hyphens and dashes with ASCII hyphen\n",
    "    text = re.sub(r'[\\u2010-\\u2015]', '-', text)\n",
    "     # Remove form feed characters (\\x0c), common page breaks\n",
    "    text = text.replace('\\x0c', ' ')\n",
    "    # Replace non-breaking spaces (\\xa0) with regular space\n",
    "    text = text.replace('\\xa0', ' ')\n",
    "    # 3. Fix hyphenated line breaks (e.g., treat-\\nment -> treatment)\n",
    "    text = re.sub(r\"(\\w+)-\\n(\\w+)\", r\"\\1\\2\", text)\n",
    "    # Remove numbered citations like (1), (9, 10), (5–7), etc.\n",
    "    text = re.sub(r\"\\(\\s?\\d+(?:\\s?(?:,|-)\\s?\\d+)*\\s?\\)\", \"\", text)\n",
    "    # Remove mid-sentence line breaks: \"word\\nword\" -> \"word word\"\n",
    "    text = re.sub(r'(?<!\\n)\\n(?!\\n)', ' ', text)\n",
    "    # Replace multiple newlines with just two (preserve paragraphs)\n",
    "    text = re.sub(r'\\n{2,}', '\\n\\n', text)\n",
    "    return text.strip()\n",
    "\n",
    "def get_system_message_rag(content):\n",
    "        return f\"\"\"You are an expert consultant helping executive advisors to get relevant information from scientific articles and code related to reproduction and bioinformatics.\n",
    "\n",
    "        Generate your response by following the steps below:\n",
    "        1. Recursively break down the question into smaller questions to better understand it.\n",
    "        2. For each question/directive:\n",
    "            2a. Select the most relevant information from the context in light of the conversation history.\n",
    "        3. Generate a draft response using selected information.\n",
    "        4. Remove duplicate content from draft response.\n",
    "        5. Generate your final response after adjusting it to increase accuracy and relevance.\n",
    "        6. Do not try to summarize the answers, explain it properly.\n",
    "        7. When you provide information, you must also provide the reference of the article.\n",
    "        8. Do not look up on internet.\n",
    "        9. Only show your final response! \n",
    "        \n",
    "        Constraints:\n",
    "        1. DO NOT PROVIDE ANY EXPLANATION OR DETAILS OR MENTION THAT YOU WERE GIVEN CONTEXT. Only do that when questions are related to coding.\n",
    "        2. Don't mention that you are not able to find the answer in the provided context.\n",
    "        3. Ignore the part of the content that only contains references.\n",
    "        3. Don't make up the answers by yourself.\n",
    "        4. Try your best to provide answer from the given context.\n",
    "\n",
    "        CONTENT:\n",
    "        {content}\n",
    "        \"\"\"\n",
    "\n",
    "def get_ques_response_prompt(question, context):\n",
    "    return f\"\"\"\n",
    "    Context\\n:\n",
    "    {context}\n",
    "    ==============================================================\n",
    "    Based on the above context, please provide the answer to the following question\\n:\n",
    "    {question}\n",
    "    \"\"\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d6746c38",
   "metadata": {},
   "source": [
    "#### Model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "23cd2e8b",
   "metadata": {},
   "outputs": [],
   "source": [
    "llm = GoogleGenerativeAI(model=\"gemini-2.5-flash\",api_key=api_key['key'],temperature=0.2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a914329c",
   "metadata": {},
   "source": [
    "#### Load PDFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "8d164fb0",
   "metadata": {},
   "outputs": [],
   "source": [
    "paths = []\n",
    "for root, dirs, files in os.walk(\"Data2\"):\n",
    "    for f in files:\n",
    "        if f.endswith('.pdf'):\n",
    "            paths.append(os.path.join(root,f))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "96d7da10",
   "metadata": {},
   "source": [
    "We create a folder with each article to pre-process the references"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "0e4874f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[None, None]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[os.mkdir(re.sub(r'.pdf','',p)) for p in paths]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "a0f6a65c",
   "metadata": {},
   "outputs": [],
   "source": [
    "articles = []\n",
    "for file in paths:\n",
    "    loader = loader = PyPDFLoader(file, mode=\"single\")\n",
    "    doc = loader.load()[0] # As this function provides a list, we select the first element.\n",
    "    doc.page_content = clean_text(doc.page_content)\n",
    "    \n",
    "    articles.append(doc)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "62bb9420",
   "metadata": {},
   "source": [
    "#### Json generation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "cd521499",
   "metadata": {},
   "outputs": [],
   "source": [
    "class paper(BaseModel):\n",
    "\n",
    "    PaperTitle: str = Field(description=\"The full title of the research paper\")\n",
    "    Publication: str = Field(description=\"Year: The year the paper was published\")\n",
    "    Authors: str = Field(description=\"The full names of all authors of the paper\")\n",
    "    Email: str = Field(description=\"The email address of the author (if provided)\")\n",
    "    Abstract: str = Field(description=\"The full text of the paper's abstract.\")\n",
    "    Introduction: str = Field(description=\"The full text fo the paper's introduction. Bear in mind that it can have other names such as background.\")\n",
    "    Methods: str = Field(description=\"The full text fo the paper's methods. Don't take the information from abstract.\")\n",
    "    Results: str = Field(description=\"The full text fo the paper's results. Don't take the information from abstract. Please take all possible text of results, this section could be divided into different sections.\")\n",
    "    Discussion: str = Field(description=\"The full text fo the paper's discussion if provided. Otherwise leave the filed blank. It should be entitled 'Discussion'\")\n",
    "    Conclusion: str = Field(description=\"The full text fo the paper's conclusion if provided. Otherwise leave the filed blank. Please don't take this information form abstract.\")\n",
    "    URL: str = Field(description=\"the link where you can find the article. This link is also known as DOI. This url is quite relevant and you must find it. It usually starts with https://doi.org/\")\n",
    "    Journal: str = Field(description=\"provide the name of the journal, e.g Nature.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "6f763428",
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = JsonOutputParser(pydantic_object=paper)\n",
    "\n",
    "prompt = PromptTemplate(\n",
    "    template=\"\"\"\n",
    "    You are an expert in analyzing scientific research papers. Please carefully read the provided research paper above and extract the following key information:\n",
    "Extract these nine (10) properties from the research paper:\n",
    "\n",
    "Paper Title: The full title of the research paper\n",
    "\n",
    "Publication Year: The year the paper was published\n",
    "\n",
    "Authors: The full names of all authors of the paper\n",
    "\n",
    "Email: The email address of the author (if provided)\n",
    "\n",
    "Abstract: The full text of the paper's abstract\n",
    "\n",
    "Introduction: The full text fo the paper's introduction. Bear in mind that it can have other names such as background.\n",
    "\n",
    "Methods: The full text fo the paper's methods. Don't take the information from abstract.\n",
    "\n",
    "Results: The full text fo the paper's results. Don't take the information from abstract. Please take all possible text of results, this section could be divided into different sections.\n",
    "\n",
    "Discussion:The full text fo the paper's discussion if provided. Otherwise leave the filed blank. It shoudl be entitled \"Discussion\"\n",
    "\n",
    "Conclusion: The full text fo the paper's conclusion if provided. Otherwise leave the filed blank. Please don't take this information form abstract.\n",
    "\n",
    "URL: the link where you can find the article. This link is also known as DOI.\n",
    "\n",
    "Journal: provide the name of the journal, e.g Nature.\n",
    "\n",
    "\n",
    "Guidelines:\n",
    "\n",
    "\n",
    "The extracted information should be factual and accurate to the document. Be extremely concise, except for the Abstract, Introduction, Methods, Results. Discussion and Conclusion which should be copied in full.\n",
    "The extracted entities should be self-contained and easily understood without the rest of the paper. If a property is missing from the paper, please leave the field empty rather than guessing.\n",
    "Answer in JSON format. The JSON should contain 9 keys: \"PaperTitle\", \"PublicationYear\", \"Authors\", \"Email\", \"Abstract\", \"Introduction\", \"Methods\",\"Results\", \"URL\", \"Reference\".\n",
    "\n",
    "\n",
    "Format instructions: \\n{format_instructions}\\n\n",
    "\n",
    "    \n",
    "The article is this:\\n{query}\\n\"\"\",\n",
    "\n",
    "    input_variables=[\"query\"],\n",
    "\n",
    "    partial_variables={\"format_instructions\": parser.get_format_instructions()}\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "aabf31d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "chain = prompt | llm | parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "e1f44809",
   "metadata": {},
   "outputs": [],
   "source": [
    "my_json = []\n",
    "for doc in articles:\n",
    "    my_json.append(chain.invoke({\"query\": doc.page_content}))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dc1ecc8e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# with open('info_articles.pkl','wb') as f:\n",
    "#     pkl.dump(my_json,f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "4d4b49c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# already DOI (To avoid repeat)\n",
    "\n",
    "includedDOI = [ re.sub(r'https://doi.org/', '',a['URL']) for a in my_json]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7147d0f3",
   "metadata": {},
   "source": [
    "## References"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "973d0869",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-10 22:03:28,424 - DEBUG - pubmed2pdf.cli - Full log mode activated\n",
      "2025-08-10 22:03:28,424 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18192189\n",
      "2025-08-10 22:03:29,182 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18192189 failed from error list index out of range\n",
      "2025-08-10 22:03:29,182 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23994285\n",
      "2025-08-10 22:03:30,565 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23994285 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 22:03:30,565 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10592173\n",
      "2025-08-10 22:03:31,277 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10592173 failed from error list index out of range\n",
      "2025-08-10 22:03:31,277 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18174356\n",
      "2025-08-10 22:03:31,983 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18174356 failed from error list index out of range\n",
      "2025-08-10 22:03:31,983 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16672246\n",
      "2025-08-10 22:03:32,721 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16672246 failed from error list index out of range\n",
      "2025-08-10 22:03:32,721 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21964334\n",
      "2025-08-10 22:03:34,983 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15550397\n",
      "2025-08-10 22:03:37,672 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15550397 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 22:03:37,672 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10802651\n",
      "2025-08-10 22:03:40,007 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18539642\n",
      "2025-08-10 22:03:40,918 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18539642 failed from error list index out of range\n",
      "2025-08-10 22:03:40,918 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22025212\n",
      "2025-08-10 22:03:41,808 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 22025212 failed from error list index out of range\n",
      "2025-08-10 22:03:41,808 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22902743\n",
      "2025-08-10 22:03:44,103 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29102484\n",
      "2025-08-10 22:03:45,361 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29102484 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 22:03:45,361 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35147192\n",
      "2025-08-10 22:03:46,069 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35147192 failed from error list index out of range\n",
      "2025-08-10 22:03:46,070 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16807381\n",
      "2025-08-10 22:03:46,732 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16807381 failed from error list index out of range\n",
      "2025-08-10 22:03:46,732 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33830236\n",
      "2025-08-10 22:03:47,419 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33830236 failed from error list index out of range\n",
      "2025-08-10 22:03:47,419 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24882617\n",
      "2025-08-10 22:03:48,760 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24882617 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 22:03:48,760 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21849299\n",
      "2025-08-10 22:03:49,431 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 21849299 failed from error list index out of range\n",
      "2025-08-10 22:03:49,431 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12529417\n",
      "2025-08-10 22:03:50,075 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12529417 failed from error list index out of range\n",
      "2025-08-10 22:03:50,076 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30624659\n",
      "2025-08-10 22:03:50,760 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30624659 failed from error list index out of range\n",
      "2025-08-10 22:03:50,760 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32641214\n",
      "2025-08-10 22:03:52,064 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32641214 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 22:03:52,064 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19815776\n",
      "2025-08-10 22:03:53,019 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19815776 failed from error argument of type 'NoneType' is not iterable\n",
      "2025-08-10 22:03:53,020 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30037059\n",
      "2025-08-10 22:03:53,723 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30037059 failed from error list index out of range\n",
      "2025-08-10 22:03:53,723 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33576824\n",
      "2025-08-10 22:03:54,425 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33576824 failed from error list index out of range\n",
      "2025-08-10 22:03:54,425 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29402198\n",
      "2025-08-10 22:03:57,600 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29663566\n",
      "2025-08-10 22:03:58,426 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29663566 failed from error list index out of range\n",
      "2025-08-10 22:03:58,426 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34117589\n",
      "2025-08-10 22:04:00,741 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26415722\n",
      "2025-08-10 22:04:01,420 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26415722 failed from error list index out of range\n",
      "2025-08-10 22:04:01,421 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19933690\n",
      "2025-08-10 22:04:02,110 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19933690 failed from error list index out of range\n",
      "2025-08-10 22:04:02,110 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16832043\n",
      "2025-08-10 22:04:02,752 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16832043 failed from error list index out of range\n",
      "2025-08-10 22:04:02,753 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11846609\n",
      "2025-08-10 22:04:04,154 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 11846609 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 22:04:04,154 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32482256\n",
      "2025-08-10 22:04:05,462 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32482256 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 22:04:05,462 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31511876\n",
      "2025-08-10 22:04:07,210 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 31511876 failed from error list index out of range\n",
      "2025-08-10 22:04:07,210 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23933037\n",
      "2025-08-10 22:04:08,429 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23933037 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 22:04:08,430 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26932670\n",
      "2025-08-10 22:04:09,166 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26932670 failed from error list index out of range\n",
      "2025-08-10 22:04:09,166 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17571346\n",
      "2025-08-10 22:04:09,166 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30081718\n",
      "2025-08-10 22:04:11,902 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29452422\n",
      "2025-08-10 22:04:12,649 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29452422 failed from error list index out of range\n",
      "2025-08-10 22:04:12,650 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25605792\n",
      "2025-08-10 22:04:13,365 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 25605792 failed from error list index out of range\n",
      "2025-08-10 22:04:13,365 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26395145\n",
      "2025-08-10 22:04:14,103 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26395145 failed from error list index out of range\n",
      "2025-08-10 22:04:14,104 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19563753\n",
      "2025-08-10 22:04:15,369 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19563753 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 22:04:15,369 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35088888\n",
      "2025-08-10 22:04:15,369 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31340985\n",
      "2025-08-10 22:04:15,369 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30951376\n",
      "2025-08-10 22:04:16,228 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30951376 failed from error list index out of range\n",
      "2025-08-10 22:04:16,228 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35574918\n",
      "2025-08-10 22:04:16,907 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35574918 failed from error list index out of range\n",
      "2025-08-10 22:04:16,907 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15501903\n",
      "2025-08-10 22:04:17,598 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15501903 failed from error list index out of range\n",
      "2025-08-10 22:04:17,599 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29156006\n",
      "2025-08-10 22:04:18,342 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29156006 failed from error list index out of range\n",
      "2025-08-10 22:04:18,343 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28855728\n",
      "2025-08-10 22:04:18,343 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23555582\n",
      "2025-08-10 22:04:18,343 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28523980\n",
      "2025-08-10 22:04:19,049 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28523980 failed from error list index out of range\n",
      "2025-08-10 22:04:19,049 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30929718\n",
      "2025-08-10 22:04:20,366 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30929718 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 22:04:20,366 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22911744\n",
      "2025-08-10 22:04:20,366 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19890323\n",
      "2025-08-10 22:04:22,857 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36205711\n",
      "2025-08-10 22:04:23,576 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36205711 failed from error list index out of range\n",
      "2025-08-10 22:04:23,576 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35085395\n",
      "2025-08-10 22:04:24,264 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35085395 failed from error list index out of range\n",
      "2025-08-10 22:04:24,264 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31462323\n",
      "2025-08-10 22:04:24,264 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24100212\n",
      "2025-08-10 22:04:25,137 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24100212 failed from error list index out of range\n",
      "2025-08-10 22:04:25,137 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19617889\n",
      "2025-08-10 22:04:27,384 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12728018\n",
      "2025-08-10 22:04:28,058 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12728018 failed from error list index out of range\n",
      "2025-08-10 22:04:28,059 - INFO - pubmed2pdf.cli - Trying to fetch pmid 7962413\n",
      "2025-08-10 22:04:28,756 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 7962413 failed from error list index out of range\n",
      "2025-08-10 22:04:28,756 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15666095\n",
      "2025-08-10 22:04:30,917 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16306079\n",
      "2025-08-10 22:04:31,620 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16306079 failed from error list index out of range\n",
      "2025-08-10 22:04:31,620 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12021176\n",
      "2025-08-10 22:04:32,255 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12021176 failed from error list index out of range\n",
      "2025-08-10 22:04:32,255 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20729534\n",
      "2025-08-10 22:04:32,969 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 20729534 failed from error list index out of range\n",
      "2025-08-10 22:04:32,969 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32043035\n",
      "2025-08-10 22:04:34,188 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32043035 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 22:04:34,188 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26804062\n",
      "2025-08-10 22:04:37,069 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26804062 failed from error list index out of range\n",
      "2025-08-10 22:04:37,069 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23290997\n",
      "2025-08-10 22:04:38,477 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23290997 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 22:04:38,477 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12200466\n",
      "2025-08-10 22:04:39,185 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12200466 failed from error list index out of range\n",
      "2025-08-10 22:04:39,185 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15878921\n",
      "2025-08-10 22:04:39,896 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15878921 failed from error list index out of range\n",
      "2025-08-10 22:04:39,896 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28229988\n",
      "2025-08-10 22:04:40,768 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28229988 failed from error list index out of range\n",
      "2025-08-10 22:04:40,768 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20619403\n",
      "2025-08-10 22:04:42,150 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 20619403 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in Data2/Parraga-Leo2023\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mKeyboardInterrupt\u001b[39m                         Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[86]\u001b[39m\u001b[32m, line 69\u001b[39m\n\u001b[32m     66\u001b[39m     \u001b[38;5;28mprint\u001b[39m(\u001b[33mf\u001b[39m\u001b[33m'\u001b[39m\u001b[33m The article with \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mpmid\u001b[38;5;132;01m}\u001b[39;00m\u001b[33m is not avaiable\u001b[39m\u001b[33m'\u001b[39m)\n\u001b[32m     68\u001b[39m soup = BeautifulSoup(response.text, \u001b[33m\"\u001b[39m\u001b[33mhtml.parser\u001b[39m\u001b[33m\"\u001b[39m)\n\u001b[32m---> \u001b[39m\u001b[32m69\u001b[39m my_json_ref.append(\u001b[43mchain\u001b[49m\u001b[43m.\u001b[49m\u001b[43minvoke\u001b[49m\u001b[43m(\u001b[49m\u001b[43m{\u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mquery\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43msoup\u001b[49m\u001b[43m.\u001b[49m\u001b[43mfind\u001b[49m\u001b[43m(\u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43marticle\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m.\u001b[49m\u001b[43mget_text\u001b[49m\u001b[43m(\u001b[49m\u001b[43mseparator\u001b[49m\u001b[43m=\u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstrip\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m}\u001b[49m\u001b[43m)\u001b[49m)\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/langchain_core/runnables/base.py:3047\u001b[39m, in \u001b[36mRunnableSequence.invoke\u001b[39m\u001b[34m(self, input, config, **kwargs)\u001b[39m\n\u001b[32m   3045\u001b[39m                 input_ = context.run(step.invoke, input_, config, **kwargs)\n\u001b[32m   3046\u001b[39m             \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[32m-> \u001b[39m\u001b[32m3047\u001b[39m                 input_ = context.run(step.invoke, input_, config)\n\u001b[32m   3048\u001b[39m \u001b[38;5;66;03m# finish the root run\u001b[39;00m\n\u001b[32m   3049\u001b[39m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mBaseException\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m e:\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/langchain_core/language_models/llms.py:389\u001b[39m, in \u001b[36mBaseLLM.invoke\u001b[39m\u001b[34m(self, input, config, stop, **kwargs)\u001b[39m\n\u001b[32m    378\u001b[39m \u001b[38;5;129m@override\u001b[39m\n\u001b[32m    379\u001b[39m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34minvoke\u001b[39m(\n\u001b[32m    380\u001b[39m     \u001b[38;5;28mself\u001b[39m,\n\u001b[32m   (...)\u001b[39m\u001b[32m    385\u001b[39m     **kwargs: Any,\n\u001b[32m    386\u001b[39m ) -> \u001b[38;5;28mstr\u001b[39m:\n\u001b[32m    387\u001b[39m     config = ensure_config(config)\n\u001b[32m    388\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m (\n\u001b[32m--> \u001b[39m\u001b[32m389\u001b[39m         \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mgenerate_prompt\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    390\u001b[39m \u001b[43m            \u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_convert_input\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43minput\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    391\u001b[39m \u001b[43m            \u001b[49m\u001b[43mstop\u001b[49m\u001b[43m=\u001b[49m\u001b[43mstop\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    392\u001b[39m \u001b[43m            \u001b[49m\u001b[43mcallbacks\u001b[49m\u001b[43m=\u001b[49m\u001b[43mconfig\u001b[49m\u001b[43m.\u001b[49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mcallbacks\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    393\u001b[39m \u001b[43m            \u001b[49m\u001b[43mtags\u001b[49m\u001b[43m=\u001b[49m\u001b[43mconfig\u001b[49m\u001b[43m.\u001b[49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mtags\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    394\u001b[39m \u001b[43m            \u001b[49m\u001b[43mmetadata\u001b[49m\u001b[43m=\u001b[49m\u001b[43mconfig\u001b[49m\u001b[43m.\u001b[49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mmetadata\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    395\u001b[39m \u001b[43m            \u001b[49m\u001b[43mrun_name\u001b[49m\u001b[43m=\u001b[49m\u001b[43mconfig\u001b[49m\u001b[43m.\u001b[49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mrun_name\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    396\u001b[39m \u001b[43m            \u001b[49m\u001b[43mrun_id\u001b[49m\u001b[43m=\u001b[49m\u001b[43mconfig\u001b[49m\u001b[43m.\u001b[49m\u001b[43mpop\u001b[49m\u001b[43m(\u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mrun_id\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    397\u001b[39m \u001b[43m            \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    398\u001b[39m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    399\u001b[39m         .generations[\u001b[32m0\u001b[39m][\u001b[32m0\u001b[39m]\n\u001b[32m    400\u001b[39m         .text\n\u001b[32m    401\u001b[39m     )\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/langchain_core/language_models/llms.py:766\u001b[39m, in \u001b[36mBaseLLM.generate_prompt\u001b[39m\u001b[34m(self, prompts, stop, callbacks, **kwargs)\u001b[39m\n\u001b[32m    757\u001b[39m \u001b[38;5;129m@override\u001b[39m\n\u001b[32m    758\u001b[39m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34mgenerate_prompt\u001b[39m(\n\u001b[32m    759\u001b[39m     \u001b[38;5;28mself\u001b[39m,\n\u001b[32m   (...)\u001b[39m\u001b[32m    763\u001b[39m     **kwargs: Any,\n\u001b[32m    764\u001b[39m ) -> LLMResult:\n\u001b[32m    765\u001b[39m     prompt_strings = [p.to_string() \u001b[38;5;28;01mfor\u001b[39;00m p \u001b[38;5;129;01min\u001b[39;00m prompts]\n\u001b[32m--> \u001b[39m\u001b[32m766\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mgenerate\u001b[49m\u001b[43m(\u001b[49m\u001b[43mprompt_strings\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstop\u001b[49m\u001b[43m=\u001b[49m\u001b[43mstop\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcallbacks\u001b[49m\u001b[43m=\u001b[49m\u001b[43mcallbacks\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/langchain_core/language_models/llms.py:973\u001b[39m, in \u001b[36mBaseLLM.generate\u001b[39m\u001b[34m(self, prompts, stop, callbacks, tags, metadata, run_name, run_id, **kwargs)\u001b[39m\n\u001b[32m    958\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m (\u001b[38;5;28mself\u001b[39m.cache \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;129;01mand\u001b[39;00m get_llm_cache() \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m) \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m.cache \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mFalse\u001b[39;00m:\n\u001b[32m    959\u001b[39m     run_managers = [\n\u001b[32m    960\u001b[39m         callback_manager.on_llm_start(\n\u001b[32m    961\u001b[39m             \u001b[38;5;28mself\u001b[39m._serialized,\n\u001b[32m   (...)\u001b[39m\u001b[32m    971\u001b[39m         )\n\u001b[32m    972\u001b[39m     ]\n\u001b[32m--> \u001b[39m\u001b[32m973\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_generate_helper\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    974\u001b[39m \u001b[43m        \u001b[49m\u001b[43mprompts\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    975\u001b[39m \u001b[43m        \u001b[49m\u001b[43mstop\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    976\u001b[39m \u001b[43m        \u001b[49m\u001b[43mrun_managers\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    977\u001b[39m \u001b[43m        \u001b[49m\u001b[43mnew_arg_supported\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43mbool\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mnew_arg_supported\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    978\u001b[39m \u001b[43m        \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    979\u001b[39m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    980\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(missing_prompts) > \u001b[32m0\u001b[39m:\n\u001b[32m    981\u001b[39m     run_managers = [\n\u001b[32m    982\u001b[39m         callback_managers[idx].on_llm_start(\n\u001b[32m    983\u001b[39m             \u001b[38;5;28mself\u001b[39m._serialized,\n\u001b[32m   (...)\u001b[39m\u001b[32m    990\u001b[39m         \u001b[38;5;28;01mfor\u001b[39;00m idx \u001b[38;5;129;01min\u001b[39;00m missing_prompt_idxs\n\u001b[32m    991\u001b[39m     ]\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/langchain_core/language_models/llms.py:792\u001b[39m, in \u001b[36mBaseLLM._generate_helper\u001b[39m\u001b[34m(self, prompts, stop, run_managers, new_arg_supported, **kwargs)\u001b[39m\n\u001b[32m    781\u001b[39m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34m_generate_helper\u001b[39m(\n\u001b[32m    782\u001b[39m     \u001b[38;5;28mself\u001b[39m,\n\u001b[32m    783\u001b[39m     prompts: \u001b[38;5;28mlist\u001b[39m[\u001b[38;5;28mstr\u001b[39m],\n\u001b[32m   (...)\u001b[39m\u001b[32m    788\u001b[39m     **kwargs: Any,\n\u001b[32m    789\u001b[39m ) -> LLMResult:\n\u001b[32m    790\u001b[39m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[32m    791\u001b[39m         output = (\n\u001b[32m--> \u001b[39m\u001b[32m792\u001b[39m             \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_generate\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    793\u001b[39m \u001b[43m                \u001b[49m\u001b[43mprompts\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    794\u001b[39m \u001b[43m                \u001b[49m\u001b[43mstop\u001b[49m\u001b[43m=\u001b[49m\u001b[43mstop\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    795\u001b[39m \u001b[43m                \u001b[49m\u001b[38;5;66;43;03m# TODO: support multiple run managers\u001b[39;49;00m\n\u001b[32m    796\u001b[39m \u001b[43m                \u001b[49m\u001b[43mrun_manager\u001b[49m\u001b[43m=\u001b[49m\u001b[43mrun_managers\u001b[49m\u001b[43m[\u001b[49m\u001b[32;43m0\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mif\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mrun_managers\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01melse\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[32m    797\u001b[39m \u001b[43m                \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    798\u001b[39m \u001b[43m            \u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    799\u001b[39m             \u001b[38;5;28;01mif\u001b[39;00m new_arg_supported\n\u001b[32m    800\u001b[39m             \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28mself\u001b[39m._generate(prompts, stop=stop)\n\u001b[32m    801\u001b[39m         )\n\u001b[32m    802\u001b[39m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mBaseException\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[32m    803\u001b[39m         \u001b[38;5;28;01mfor\u001b[39;00m run_manager \u001b[38;5;129;01min\u001b[39;00m run_managers:\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/langchain_google_genai/llms.py:114\u001b[39m, in \u001b[36mGoogleGenerativeAI._generate\u001b[39m\u001b[34m(self, prompts, stop, run_manager, **kwargs)\u001b[39m\n\u001b[32m    112\u001b[39m generations = []\n\u001b[32m    113\u001b[39m \u001b[38;5;28;01mfor\u001b[39;00m prompt \u001b[38;5;129;01min\u001b[39;00m prompts:\n\u001b[32m--> \u001b[39m\u001b[32m114\u001b[39m     chat_result = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mclient\u001b[49m\u001b[43m.\u001b[49m\u001b[43m_generate\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    115\u001b[39m \u001b[43m        \u001b[49m\u001b[43m[\u001b[49m\u001b[43mHumanMessage\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcontent\u001b[49m\u001b[43m=\u001b[49m\u001b[43mprompt\u001b[49m\u001b[43m)\u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    116\u001b[39m \u001b[43m        \u001b[49m\u001b[43mstop\u001b[49m\u001b[43m=\u001b[49m\u001b[43mstop\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    117\u001b[39m \u001b[43m        \u001b[49m\u001b[43mrun_manager\u001b[49m\u001b[43m=\u001b[49m\u001b[43mrun_manager\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    118\u001b[39m \u001b[43m        \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    119\u001b[39m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    120\u001b[39m     generations.append(\n\u001b[32m    121\u001b[39m         [\n\u001b[32m    122\u001b[39m             Generation(\n\u001b[32m   (...)\u001b[39m\u001b[32m    130\u001b[39m         ]\n\u001b[32m    131\u001b[39m     )\n\u001b[32m    132\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m LLMResult(generations=generations)\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/langchain_google_genai/chat_models.py:1441\u001b[39m, in \u001b[36mChatGoogleGenerativeAI._generate\u001b[39m\u001b[34m(self, messages, stop, run_manager, tools, functions, safety_settings, tool_config, generation_config, cached_content, tool_choice, **kwargs)\u001b[39m\n\u001b[32m   1414\u001b[39m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34m_generate\u001b[39m(\n\u001b[32m   1415\u001b[39m     \u001b[38;5;28mself\u001b[39m,\n\u001b[32m   1416\u001b[39m     messages: List[BaseMessage],\n\u001b[32m   (...)\u001b[39m\u001b[32m   1427\u001b[39m     **kwargs: Any,\n\u001b[32m   1428\u001b[39m ) -> ChatResult:\n\u001b[32m   1429\u001b[39m     request = \u001b[38;5;28mself\u001b[39m._prepare_request(\n\u001b[32m   1430\u001b[39m         messages,\n\u001b[32m   1431\u001b[39m         stop=stop,\n\u001b[32m   (...)\u001b[39m\u001b[32m   1439\u001b[39m         **kwargs,\n\u001b[32m   1440\u001b[39m     )\n\u001b[32m-> \u001b[39m\u001b[32m1441\u001b[39m     response: GenerateContentResponse = \u001b[43m_chat_with_retry\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m   1442\u001b[39m \u001b[43m        \u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m=\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   1443\u001b[39m \u001b[43m        \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   1444\u001b[39m \u001b[43m        \u001b[49m\u001b[43mgeneration_method\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mclient\u001b[49m\u001b[43m.\u001b[49m\u001b[43mgenerate_content\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   1445\u001b[39m \u001b[43m        \u001b[49m\u001b[43mmetadata\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mdefault_metadata\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   1446\u001b[39m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m   1447\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m _response_to_result(response)\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/langchain_google_genai/chat_models.py:231\u001b[39m, in \u001b[36m_chat_with_retry\u001b[39m\u001b[34m(generation_method, **kwargs)\u001b[39m\n\u001b[32m    222\u001b[39m         \u001b[38;5;28;01mraise\u001b[39;00m e\n\u001b[32m    224\u001b[39m params = (\n\u001b[32m    225\u001b[39m     {k: v \u001b[38;5;28;01mfor\u001b[39;00m k, v \u001b[38;5;129;01min\u001b[39;00m kwargs.items() \u001b[38;5;28;01mif\u001b[39;00m k \u001b[38;5;129;01min\u001b[39;00m _allowed_params_prediction_service}\n\u001b[32m    226\u001b[39m     \u001b[38;5;28;01mif\u001b[39;00m (request := kwargs.get(\u001b[33m\"\u001b[39m\u001b[33mrequest\u001b[39m\u001b[33m\"\u001b[39m))\n\u001b[32m   (...)\u001b[39m\u001b[32m    229\u001b[39m     \u001b[38;5;28;01melse\u001b[39;00m kwargs\n\u001b[32m    230\u001b[39m )\n\u001b[32m--> \u001b[39m\u001b[32m231\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_chat_with_retry\u001b[49m\u001b[43m(\u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mparams\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/tenacity/__init__.py:338\u001b[39m, in \u001b[36mBaseRetrying.wraps.<locals>.wrapped_f\u001b[39m\u001b[34m(*args, **kw)\u001b[39m\n\u001b[32m    336\u001b[39m copy = \u001b[38;5;28mself\u001b[39m.copy()\n\u001b[32m    337\u001b[39m wrapped_f.statistics = copy.statistics  \u001b[38;5;66;03m# type: ignore[attr-defined]\u001b[39;00m\n\u001b[32m--> \u001b[39m\u001b[32m338\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mcopy\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkw\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/tenacity/__init__.py:477\u001b[39m, in \u001b[36mRetrying.__call__\u001b[39m\u001b[34m(self, fn, *args, **kwargs)\u001b[39m\n\u001b[32m    475\u001b[39m retry_state = RetryCallState(retry_object=\u001b[38;5;28mself\u001b[39m, fn=fn, args=args, kwargs=kwargs)\n\u001b[32m    476\u001b[39m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[32m--> \u001b[39m\u001b[32m477\u001b[39m     do = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43miter\u001b[49m\u001b[43m(\u001b[49m\u001b[43mretry_state\u001b[49m\u001b[43m=\u001b[49m\u001b[43mretry_state\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    478\u001b[39m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(do, DoAttempt):\n\u001b[32m    479\u001b[39m         \u001b[38;5;28;01mtry\u001b[39;00m:\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/tenacity/__init__.py:378\u001b[39m, in \u001b[36mBaseRetrying.iter\u001b[39m\u001b[34m(self, retry_state)\u001b[39m\n\u001b[32m    376\u001b[39m result = \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[32m    377\u001b[39m \u001b[38;5;28;01mfor\u001b[39;00m action \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m.iter_state.actions:\n\u001b[32m--> \u001b[39m\u001b[32m378\u001b[39m     result = \u001b[43maction\u001b[49m\u001b[43m(\u001b[49m\u001b[43mretry_state\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    379\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m result\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/tenacity/__init__.py:400\u001b[39m, in \u001b[36mBaseRetrying._post_retry_check_actions.<locals>.<lambda>\u001b[39m\u001b[34m(rs)\u001b[39m\n\u001b[32m    398\u001b[39m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34m_post_retry_check_actions\u001b[39m(\u001b[38;5;28mself\u001b[39m, retry_state: \u001b[33m\"\u001b[39m\u001b[33mRetryCallState\u001b[39m\u001b[33m\"\u001b[39m) -> \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[32m    399\u001b[39m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m (\u001b[38;5;28mself\u001b[39m.iter_state.is_explicit_retry \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m.iter_state.retry_run_result):\n\u001b[32m--> \u001b[39m\u001b[32m400\u001b[39m         \u001b[38;5;28mself\u001b[39m._add_action_func(\u001b[38;5;28;01mlambda\u001b[39;00m rs: \u001b[43mrs\u001b[49m\u001b[43m.\u001b[49m\u001b[43moutcome\u001b[49m\u001b[43m.\u001b[49m\u001b[43mresult\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m)\n\u001b[32m    401\u001b[39m         \u001b[38;5;28;01mreturn\u001b[39;00m\n\u001b[32m    403\u001b[39m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m.after \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/concurrent/futures/_base.py:449\u001b[39m, in \u001b[36mFuture.result\u001b[39m\u001b[34m(self, timeout)\u001b[39m\n\u001b[32m    447\u001b[39m     \u001b[38;5;28;01mraise\u001b[39;00m CancelledError()\n\u001b[32m    448\u001b[39m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28mself\u001b[39m._state == FINISHED:\n\u001b[32m--> \u001b[39m\u001b[32m449\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m__get_result\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    451\u001b[39m \u001b[38;5;28mself\u001b[39m._condition.wait(timeout)\n\u001b[32m    453\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m._state \u001b[38;5;129;01min\u001b[39;00m [CANCELLED, CANCELLED_AND_NOTIFIED]:\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/concurrent/futures/_base.py:401\u001b[39m, in \u001b[36mFuture.__get_result\u001b[39m\u001b[34m(self)\u001b[39m\n\u001b[32m    399\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m._exception:\n\u001b[32m    400\u001b[39m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[32m--> \u001b[39m\u001b[32m401\u001b[39m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;28mself\u001b[39m._exception\n\u001b[32m    402\u001b[39m     \u001b[38;5;28;01mfinally\u001b[39;00m:\n\u001b[32m    403\u001b[39m         \u001b[38;5;66;03m# Break a reference cycle with the exception in self._exception\u001b[39;00m\n\u001b[32m    404\u001b[39m         \u001b[38;5;28mself\u001b[39m = \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/tenacity/__init__.py:480\u001b[39m, in \u001b[36mRetrying.__call__\u001b[39m\u001b[34m(self, fn, *args, **kwargs)\u001b[39m\n\u001b[32m    478\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(do, DoAttempt):\n\u001b[32m    479\u001b[39m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[32m--> \u001b[39m\u001b[32m480\u001b[39m         result = \u001b[43mfn\u001b[49m\u001b[43m(\u001b[49m\u001b[43m*\u001b[49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    481\u001b[39m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mBaseException\u001b[39;00m:  \u001b[38;5;66;03m# noqa: B902\u001b[39;00m\n\u001b[32m    482\u001b[39m         retry_state.set_exception(sys.exc_info())  \u001b[38;5;66;03m# type: ignore[arg-type]\u001b[39;00m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/langchain_google_genai/chat_models.py:206\u001b[39m, in \u001b[36m_chat_with_retry.<locals>._chat_with_retry\u001b[39m\u001b[34m(**kwargs)\u001b[39m\n\u001b[32m    203\u001b[39m \u001b[38;5;129m@retry_decorator\u001b[39m\n\u001b[32m    204\u001b[39m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34m_chat_with_retry\u001b[39m(**kwargs: Any) -> Any:\n\u001b[32m    205\u001b[39m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[32m--> \u001b[39m\u001b[32m206\u001b[39m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mgeneration_method\u001b[49m\u001b[43m(\u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    207\u001b[39m     \u001b[38;5;66;03m# Do not retry for these errors.\u001b[39;00m\n\u001b[32m    208\u001b[39m     \u001b[38;5;28;01mexcept\u001b[39;00m google.api_core.exceptions.FailedPrecondition \u001b[38;5;28;01mas\u001b[39;00m exc:\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/google/ai/generativelanguage_v1beta/services/generative_service/client.py:835\u001b[39m, in \u001b[36mGenerativeServiceClient.generate_content\u001b[39m\u001b[34m(self, request, model, contents, retry, timeout, metadata)\u001b[39m\n\u001b[32m    832\u001b[39m \u001b[38;5;28mself\u001b[39m._validate_universe_domain()\n\u001b[32m    834\u001b[39m \u001b[38;5;66;03m# Send the request.\u001b[39;00m\n\u001b[32m--> \u001b[39m\u001b[32m835\u001b[39m response = \u001b[43mrpc\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    836\u001b[39m \u001b[43m    \u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    837\u001b[39m \u001b[43m    \u001b[49m\u001b[43mretry\u001b[49m\u001b[43m=\u001b[49m\u001b[43mretry\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    838\u001b[39m \u001b[43m    \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m=\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    839\u001b[39m \u001b[43m    \u001b[49m\u001b[43mmetadata\u001b[49m\u001b[43m=\u001b[49m\u001b[43mmetadata\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    840\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    842\u001b[39m \u001b[38;5;66;03m# Done; return the response.\u001b[39;00m\n\u001b[32m    843\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m response\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/google/api_core/gapic_v1/method.py:131\u001b[39m, in \u001b[36m_GapicCallable.__call__\u001b[39m\u001b[34m(self, timeout, retry, compression, *args, **kwargs)\u001b[39m\n\u001b[32m    128\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m._compression \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[32m    129\u001b[39m     kwargs[\u001b[33m\"\u001b[39m\u001b[33mcompression\u001b[39m\u001b[33m\"\u001b[39m] = compression\n\u001b[32m--> \u001b[39m\u001b[32m131\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mwrapped_func\u001b[49m\u001b[43m(\u001b[49m\u001b[43m*\u001b[49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/google/api_core/retry/retry_unary.py:294\u001b[39m, in \u001b[36mRetry.__call__.<locals>.retry_wrapped_func\u001b[39m\u001b[34m(*args, **kwargs)\u001b[39m\n\u001b[32m    290\u001b[39m target = functools.partial(func, *args, **kwargs)\n\u001b[32m    291\u001b[39m sleep_generator = exponential_sleep_generator(\n\u001b[32m    292\u001b[39m     \u001b[38;5;28mself\u001b[39m._initial, \u001b[38;5;28mself\u001b[39m._maximum, multiplier=\u001b[38;5;28mself\u001b[39m._multiplier\n\u001b[32m    293\u001b[39m )\n\u001b[32m--> \u001b[39m\u001b[32m294\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mretry_target\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    295\u001b[39m \u001b[43m    \u001b[49m\u001b[43mtarget\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    296\u001b[39m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_predicate\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    297\u001b[39m \u001b[43m    \u001b[49m\u001b[43msleep_generator\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    298\u001b[39m \u001b[43m    \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_timeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    299\u001b[39m \u001b[43m    \u001b[49m\u001b[43mon_error\u001b[49m\u001b[43m=\u001b[49m\u001b[43mon_error\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    300\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/google/api_core/retry/retry_unary.py:147\u001b[39m, in \u001b[36mretry_target\u001b[39m\u001b[34m(target, predicate, sleep_generator, timeout, on_error, exception_factory, **kwargs)\u001b[39m\n\u001b[32m    145\u001b[39m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[32m    146\u001b[39m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[32m--> \u001b[39m\u001b[32m147\u001b[39m         result = \u001b[43mtarget\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    148\u001b[39m         \u001b[38;5;28;01mif\u001b[39;00m inspect.isawaitable(result):\n\u001b[32m    149\u001b[39m             warnings.warn(_ASYNC_RETRY_WARNING)\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/google/api_core/timeout.py:130\u001b[39m, in \u001b[36mTimeToDeadlineTimeout.__call__.<locals>.func_with_timeout\u001b[39m\u001b[34m(*args, **kwargs)\u001b[39m\n\u001b[32m    126\u001b[39m         remaining_timeout = \u001b[38;5;28mself\u001b[39m._timeout\n\u001b[32m    128\u001b[39m     kwargs[\u001b[33m\"\u001b[39m\u001b[33mtimeout\u001b[39m\u001b[33m\"\u001b[39m] = remaining_timeout\n\u001b[32m--> \u001b[39m\u001b[32m130\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[43m*\u001b[49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/google/api_core/grpc_helpers.py:76\u001b[39m, in \u001b[36m_wrap_unary_errors.<locals>.error_remapped_callable\u001b[39m\u001b[34m(*args, **kwargs)\u001b[39m\n\u001b[32m     73\u001b[39m \u001b[38;5;129m@functools\u001b[39m.wraps(callable_)\n\u001b[32m     74\u001b[39m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34merror_remapped_callable\u001b[39m(*args, **kwargs):\n\u001b[32m     75\u001b[39m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[32m---> \u001b[39m\u001b[32m76\u001b[39m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mcallable_\u001b[49m\u001b[43m(\u001b[49m\u001b[43m*\u001b[49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m     77\u001b[39m     \u001b[38;5;28;01mexcept\u001b[39;00m grpc.RpcError \u001b[38;5;28;01mas\u001b[39;00m exc:\n\u001b[32m     78\u001b[39m         \u001b[38;5;28;01mraise\u001b[39;00m exceptions.from_grpc_error(exc) \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34;01mexc\u001b[39;00m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/grpc/_interceptor.py:277\u001b[39m, in \u001b[36m_UnaryUnaryMultiCallable.__call__\u001b[39m\u001b[34m(self, request, timeout, metadata, credentials, wait_for_ready, compression)\u001b[39m\n\u001b[32m    268\u001b[39m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34m__call__\u001b[39m(\n\u001b[32m    269\u001b[39m     \u001b[38;5;28mself\u001b[39m,\n\u001b[32m    270\u001b[39m     request: Any,\n\u001b[32m   (...)\u001b[39m\u001b[32m    275\u001b[39m     compression: Optional[grpc.Compression] = \u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[32m    276\u001b[39m ) -> Any:\n\u001b[32m--> \u001b[39m\u001b[32m277\u001b[39m     response, ignored_call = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_with_call\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    278\u001b[39m \u001b[43m        \u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    279\u001b[39m \u001b[43m        \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m=\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    280\u001b[39m \u001b[43m        \u001b[49m\u001b[43mmetadata\u001b[49m\u001b[43m=\u001b[49m\u001b[43mmetadata\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    281\u001b[39m \u001b[43m        \u001b[49m\u001b[43mcredentials\u001b[49m\u001b[43m=\u001b[49m\u001b[43mcredentials\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    282\u001b[39m \u001b[43m        \u001b[49m\u001b[43mwait_for_ready\u001b[49m\u001b[43m=\u001b[49m\u001b[43mwait_for_ready\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    283\u001b[39m \u001b[43m        \u001b[49m\u001b[43mcompression\u001b[49m\u001b[43m=\u001b[49m\u001b[43mcompression\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    284\u001b[39m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    285\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m response\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/grpc/_interceptor.py:329\u001b[39m, in \u001b[36m_UnaryUnaryMultiCallable._with_call\u001b[39m\u001b[34m(self, request, timeout, metadata, credentials, wait_for_ready, compression)\u001b[39m\n\u001b[32m    326\u001b[39m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m exception:  \u001b[38;5;66;03m# pylint:disable=broad-except\u001b[39;00m\n\u001b[32m    327\u001b[39m         \u001b[38;5;28;01mreturn\u001b[39;00m _FailureOutcome(exception, sys.exc_info()[\u001b[32m2\u001b[39m])\n\u001b[32m--> \u001b[39m\u001b[32m329\u001b[39m call = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_interceptor\u001b[49m\u001b[43m.\u001b[49m\u001b[43mintercept_unary_unary\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    330\u001b[39m \u001b[43m    \u001b[49m\u001b[43mcontinuation\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mclient_call_details\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrequest\u001b[49m\n\u001b[32m    331\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    332\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m call.result(), call\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/google/ai/generativelanguage_v1beta/services/generative_service/transports/grpc.py:79\u001b[39m, in \u001b[36m_LoggingClientInterceptor.intercept_unary_unary\u001b[39m\u001b[34m(self, continuation, client_call_details, request)\u001b[39m\n\u001b[32m     64\u001b[39m     grpc_request = {\n\u001b[32m     65\u001b[39m         \u001b[33m\"\u001b[39m\u001b[33mpayload\u001b[39m\u001b[33m\"\u001b[39m: request_payload,\n\u001b[32m     66\u001b[39m         \u001b[33m\"\u001b[39m\u001b[33mrequestMethod\u001b[39m\u001b[33m\"\u001b[39m: \u001b[33m\"\u001b[39m\u001b[33mgrpc\u001b[39m\u001b[33m\"\u001b[39m,\n\u001b[32m     67\u001b[39m         \u001b[33m\"\u001b[39m\u001b[33mmetadata\u001b[39m\u001b[33m\"\u001b[39m: \u001b[38;5;28mdict\u001b[39m(request_metadata),\n\u001b[32m     68\u001b[39m     }\n\u001b[32m     69\u001b[39m     _LOGGER.debug(\n\u001b[32m     70\u001b[39m         \u001b[33mf\u001b[39m\u001b[33m\"\u001b[39m\u001b[33mSending request for \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mclient_call_details.method\u001b[38;5;132;01m}\u001b[39;00m\u001b[33m\"\u001b[39m,\n\u001b[32m     71\u001b[39m         extra={\n\u001b[32m   (...)\u001b[39m\u001b[32m     76\u001b[39m         },\n\u001b[32m     77\u001b[39m     )\n\u001b[32m---> \u001b[39m\u001b[32m79\u001b[39m response = \u001b[43mcontinuation\u001b[49m\u001b[43m(\u001b[49m\u001b[43mclient_call_details\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m     80\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m logging_enabled:  \u001b[38;5;66;03m# pragma: NO COVER\u001b[39;00m\n\u001b[32m     81\u001b[39m     response_metadata = response.trailing_metadata()\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/grpc/_interceptor.py:315\u001b[39m, in \u001b[36m_UnaryUnaryMultiCallable._with_call.<locals>.continuation\u001b[39m\u001b[34m(new_details, request)\u001b[39m\n\u001b[32m    306\u001b[39m (\n\u001b[32m    307\u001b[39m     new_method,\n\u001b[32m    308\u001b[39m     new_timeout,\n\u001b[32m   (...)\u001b[39m\u001b[32m    312\u001b[39m     new_compression,\n\u001b[32m    313\u001b[39m ) = _unwrap_client_call_details(new_details, client_call_details)\n\u001b[32m    314\u001b[39m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[32m--> \u001b[39m\u001b[32m315\u001b[39m     response, call = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_thunk\u001b[49m\u001b[43m(\u001b[49m\u001b[43mnew_method\u001b[49m\u001b[43m)\u001b[49m\u001b[43m.\u001b[49m\u001b[43mwith_call\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    316\u001b[39m \u001b[43m        \u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    317\u001b[39m \u001b[43m        \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m=\u001b[49m\u001b[43mnew_timeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    318\u001b[39m \u001b[43m        \u001b[49m\u001b[43mmetadata\u001b[49m\u001b[43m=\u001b[49m\u001b[43mnew_metadata\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    319\u001b[39m \u001b[43m        \u001b[49m\u001b[43mcredentials\u001b[49m\u001b[43m=\u001b[49m\u001b[43mnew_credentials\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    320\u001b[39m \u001b[43m        \u001b[49m\u001b[43mwait_for_ready\u001b[49m\u001b[43m=\u001b[49m\u001b[43mnew_wait_for_ready\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    321\u001b[39m \u001b[43m        \u001b[49m\u001b[43mcompression\u001b[49m\u001b[43m=\u001b[49m\u001b[43mnew_compression\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    322\u001b[39m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    323\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m _UnaryOutcome(response, call)\n\u001b[32m    324\u001b[39m \u001b[38;5;28;01mexcept\u001b[39;00m grpc.RpcError \u001b[38;5;28;01mas\u001b[39;00m rpc_error:\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/grpc/_channel.py:1195\u001b[39m, in \u001b[36m_UnaryUnaryMultiCallable.with_call\u001b[39m\u001b[34m(self, request, timeout, metadata, credentials, wait_for_ready, compression)\u001b[39m\n\u001b[32m   1183\u001b[39m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34mwith_call\u001b[39m(\n\u001b[32m   1184\u001b[39m     \u001b[38;5;28mself\u001b[39m,\n\u001b[32m   1185\u001b[39m     request: Any,\n\u001b[32m   (...)\u001b[39m\u001b[32m   1190\u001b[39m     compression: Optional[grpc.Compression] = \u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[32m   1191\u001b[39m ) -> Tuple[Any, grpc.Call]:\n\u001b[32m   1192\u001b[39m     (\n\u001b[32m   1193\u001b[39m         state,\n\u001b[32m   1194\u001b[39m         call,\n\u001b[32m-> \u001b[39m\u001b[32m1195\u001b[39m     ) = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_blocking\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m   1196\u001b[39m \u001b[43m        \u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmetadata\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcredentials\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mwait_for_ready\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcompression\u001b[49m\n\u001b[32m   1197\u001b[39m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m   1198\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m _end_unary_response_blocking(state, call, \u001b[38;5;28;01mTrue\u001b[39;00m, \u001b[38;5;28;01mNone\u001b[39;00m)\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/grpc/_channel.py:1162\u001b[39m, in \u001b[36m_UnaryUnaryMultiCallable._blocking\u001b[39m\u001b[34m(self, request, timeout, metadata, credentials, wait_for_ready, compression)\u001b[39m\n\u001b[32m   1145\u001b[39m state.target = _common.decode(\u001b[38;5;28mself\u001b[39m._target)\n\u001b[32m   1146\u001b[39m call = \u001b[38;5;28mself\u001b[39m._channel.segregated_call(\n\u001b[32m   1147\u001b[39m     cygrpc.PropagationConstants.GRPC_PROPAGATE_DEFAULTS,\n\u001b[32m   1148\u001b[39m     \u001b[38;5;28mself\u001b[39m._method,\n\u001b[32m   (...)\u001b[39m\u001b[32m   1160\u001b[39m     \u001b[38;5;28mself\u001b[39m._registered_call_handle,\n\u001b[32m   1161\u001b[39m )\n\u001b[32m-> \u001b[39m\u001b[32m1162\u001b[39m event = \u001b[43mcall\u001b[49m\u001b[43m.\u001b[49m\u001b[43mnext_event\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m   1163\u001b[39m _handle_event(event, state, \u001b[38;5;28mself\u001b[39m._response_deserializer)\n\u001b[32m   1164\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m state, call\n",
      "\u001b[36mFile \u001b[39m\u001b[32msrc/python/grpcio/grpc/_cython/_cygrpc/channel.pyx.pxi:388\u001b[39m, in \u001b[36mgrpc._cython.cygrpc.SegregatedCall.next_event\u001b[39m\u001b[34m()\u001b[39m\n",
      "\u001b[36mFile \u001b[39m\u001b[32msrc/python/grpcio/grpc/_cython/_cygrpc/channel.pyx.pxi:211\u001b[39m, in \u001b[36mgrpc._cython.cygrpc._next_call_event\u001b[39m\u001b[34m()\u001b[39m\n",
      "\u001b[36mFile \u001b[39m\u001b[32msrc/python/grpcio/grpc/_cython/_cygrpc/channel.pyx.pxi:205\u001b[39m, in \u001b[36mgrpc._cython.cygrpc._next_call_event\u001b[39m\u001b[34m()\u001b[39m\n",
      "\u001b[36mFile \u001b[39m\u001b[32msrc/python/grpcio/grpc/_cython/_cygrpc/completion_queue.pyx.pxi:97\u001b[39m, in \u001b[36mgrpc._cython.cygrpc._latent_event\u001b[39m\u001b[34m()\u001b[39m\n",
      "\u001b[36mFile \u001b[39m\u001b[32msrc/python/grpcio/grpc/_cython/_cygrpc/completion_queue.pyx.pxi:80\u001b[39m, in \u001b[36mgrpc._cython.cygrpc._internal_latent_event\u001b[39m\u001b[34m()\u001b[39m\n",
      "\u001b[36mFile \u001b[39m\u001b[32msrc/python/grpcio/grpc/_cython/_cygrpc/completion_queue.pyx.pxi:61\u001b[39m, in \u001b[36mgrpc._cython.cygrpc._next\u001b[39m\u001b[34m()\u001b[39m\n",
      "\u001b[31mKeyboardInterrupt\u001b[39m: "
     ]
    }
   ],
   "source": [
    "my_json_ref = []\n",
    "pmid = {}\n",
    "HEADERS = {\n",
    "        \"User-Agent\": (\n",
    "            \"Mozilla/5.0 (Windows NT 10.0; Win64; x64) \"\n",
    "            \"AppleWebKit/537.36 (KHTML, like Gecko) \"\n",
    "            \"Chrome/115.0.0.0 Safari/537.36\"\n",
    "        )\n",
    "    }\n",
    "for i, article in enumerate(my_json):\n",
    "    path_name = re.sub(r'.pdf','',paths[i]).split('/')[1]\n",
    "    pmid[path_name] = []\n",
    "\n",
    "    if 'https://' in article['URL']:\n",
    "        doi = re.sub(r'https://doi.org/', '', article['URL'])\n",
    "        response = requests.get('https://api.crossref.org/works/'+doi)\n",
    "        list_reference = response.json()['message'].get('reference',None)\n",
    "        article['List_references'] = list(set([x.get('DOI') for x in list_reference if x.get('DOI') and x.get('DOI') not in includedDOI]))\n",
    "\n",
    "        includedDOI.extend(article['List_references'])\n",
    "\n",
    "        # We retrieve the pmid from each DOI\n",
    "\n",
    "        for a in article['List_references']:\n",
    "            \n",
    "            pmid[path_name].append(get_pmid_from_doi(a))\n",
    "\n",
    "        # We try to get the pdf\n",
    "\n",
    "        command = [\n",
    "        \"python3\", \"-m\", \"pubmed2pdf\", \"pdf\",\n",
    "        \"--out\", os.path.join(\"Data2\",path_name),\n",
    "        \"--pmids\", f\"{', '.join(pmid[path_name])}\",\n",
    "        \"--verbose\",\n",
    "        \"--errors\", os.path.join(\"Data2\",path_name,\"pubmed2pdf_log.txt\")\n",
    "        ]\n",
    "        subprocess.run(command)\n",
    "\n",
    "        # We process pdfs\n",
    "\n",
    "        for pdf in [os.path.join(\"Data2\",path_name,f) for f in os.listdir(os.path.join(\"Data2\",path_name)) if f.endswith('.pdf')]:\n",
    "\n",
    "            loader = loader = PyPDFLoader(file, mode=\"single\")\n",
    "            doc = loader.load()[0] # As this function provides a list, we select the first element.\n",
    "            doc.page_content = clean_text(doc.page_content)\n",
    "\n",
    "            my_json_ref.append(chain.invoke({\"query\": doc.page_content}))\n",
    "        \n",
    "\n",
    "        # We try to get the html otherwise\n",
    "\n",
    "        to_retrieve = pd.read_csv(os.path.join(\"Data2\",path_name,\"pubmed2pdf_log.txt\"), header=None,dtype='str') # We get those that resulted in a mistake\n",
    "        to_retrieve.columns = [\"PMID\"]\n",
    "        to_retrieve = to_retrieve['PMID'].tolist()\n",
    "        to_retrieve.extend([re.sub(r'.html','',f)  for f in os.listdir(os.path.join(\"Data2\",path_name)) if f.endswith('.html')]) # We add those in html format since they do not have information\n",
    "        \n",
    "        [os.remove(os.path.join(\"Data2\",path_name,f))  for f in os.listdir(os.path.join(\"Data2\",path_name)) if f.endswith('.html')] # Then, we remove from directory\n",
    "        \n",
    "        \n",
    "        for pmid in to_retrieve:\n",
    "            pmcid = get_pmcid_from_pmid(pmid)\n",
    "\n",
    "            try:\n",
    "                response = requests.get(url = f\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC{pmcid}/\", headers=HEADERS)\n",
    "            except:\n",
    "                print(f' The article with {pmid} is not avaiable')\n",
    "\n",
    "            soup = BeautifulSoup(response.text, \"html.parser\")\n",
    "            my_json_ref.append(chain.invoke({\"query\": soup.find(\"article\").get_text(separator=\"\\n\", strip=True)}))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "550033d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'PaperTitle': 'Predicting risk of endometrial failure: a biomarker signature that identifies a novel disruption independent of endometrial timing in patients undergoing hormonal replacement cycles',\n",
       "  'Publication': '2024',\n",
       "  'Authors': 'Patricia Diaz-Gimeno, Ph.D., Patricia Sebastian-Leon, Ph.D., Katharina Spath, Ph.D., Diana Marti-Garcia, M.Sc., Josefa Maria Sanchez-Reyes, Ph.D., Maria del Carmen Vidal, Ph.D., Almudena Devesa-Peiro, Ph.D., Immaculada Sanchez-Ribas, Ph.D., Asunta Martinez-Martinez, M.Sc., Nuria Pellicer, M.D., Ph.D., Dagan Wells, Ph.D., and Antonio Pellicer, M.D., Ph.D.',\n",
       "  'Email': 'patricia.diaz@ivirma.com or patricia_diaz@iislafe.es',\n",
       "  'Abstract': 'Objective: To propose a new gene expression signature that identifies endometrial disruptions independent of endometrial luteal phase timing and predicts if patients are at risk of endometrial failure. Design: Multicentric, prospective study. Setting: Reproductive medicine research department in a public hospital affiliated with private fertility clinics and a reproductive genetics laboratory. Patients: Caucasian women (n1⁄4 281; 39.4/C6 4.8 years old with a body mass index of 22.9/C6 3.5 kg/m 2) undergoing hormone replacement therapy between July 2018 and July 2021. Endometrial samples from 217 patients met RNA quality criteria for signature discovery and analysis. Intervention(s): Endometrial biopsies collected in the mid-secretory phase. Main Outcome Measure(s):Endometrial luteal phase timing-corrected expression of 404 genes and reproductive outcomes of thefirst single embryo transfer (SET) after biopsy collection to identify prognostic biomarkers of endometrial failure. Results: Removal of endometrial timing variation from gene expression data allowed patients to be stratified into poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognosis groups on the basis of their clinical and transcriptomic profiles. Significant differences were found between endometrial prognosis groups in terms of reproductive rates: pregnancy (44.6% vs. 79.6%), live birth (25.6% vs. 77.6%), clinical miscarriage (22.2% vs. 2.6%), and biochemical miscarriage (20.4% vs. 0%). The relative risk of endometrial failure for patients predicted as a poor endometrial prognosis was 3.3 times higher than those with a good prognosis. The differences in gene expression between both profiles were proposed as a biomarker, coined the endometrial failure risk (EFR) signature. Poor prognosis profiles were characterized by 59 upregulated and 63 downregulated genes mainly involved in regulation (17.0%), metabolism (8.4%), immune response, and inflammation (7.8%). This EFR signature had a median accuracy of 0.92 (min1⁄40.88, max1⁄40.94), median sensitivity of 0.96 (min1⁄40.91, max 1⁄4 0.98), and median specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88), positioning itself as a promising biomarker for endometrial evaluation.',\n",
       "  'Introduction': 'A key factor for reproductive success in assisted reproduction treatments is the status of the maternal endometrium during embryo implantation and fetal development. The endometrial cycle is reflected by the cyclic structural and functional changes of the endometrium across the menstrual cycle, particularly during the mid-secretory phase, to prepare for the‘‘window of implantation’’ (WOI) . After successful embryo implantation, the decidua (specialized layer of the endometrium) actively encapsulates the trophectoderm to support placentation and provide adequate vascularization for optimal fetal growth . Thus, approaches that predict and prevent endometrial-factor infertility could substantially improve assisted reproduction treatment outcomes by supporting the establishment and maintenance of pregnancy . There is a lack of consensus on the minimum number of implantation failures with good-quality embryos derived from ovum donation or with confirmed euploid karyotypes required to diagnose recurrent implantation failure (RIF) . The disparities in clinical classification and heterogeneous etiology of RIF confound its diagnosis and leave the exact prevalence unclear . As clinical symptoms cannot be prevented and do not provide sufficient criteria to stratify patients, patients might be misclassi fied and receive inefficient treatment. Indeed, most studies using clinical criteria for RIF identified no or few differentially expressed genes between samples of affected patients , reflecting the molecular heterogeneity of RIF. To overcome these healthcare challenges, high-throughput molecular technologies are combined with artificial intelligence (AI) algorithms to provide a deeper understanding of infertilityrelated pathogenesis . These molecular insights are then applied in genomics-driven precision medicine approaches to characterize complex diseases, and ultimately, improve patient diagnosis and prognosis . Employing genomics in reproductive medicine has the potential to go beyond treating the clinical manifestations of disease progression , by leveraging the patients’ heterogeneous molecular profiles for precise stratification and revealing new targets for preventive measures . In this regard, current diagnostic tools for endometrialfactor infertility use transcriptomics to date the endometrial biopsy in the mid-secretory phase, exceeding the accuracy and objectivity of traditional endometrial progression staging by Noyes’ histological criteria . However, the clinical utility of synchronizing the endometrium by adjusting the timing of progesterone (P4) administration on the basis of the patient’s transcriptomic endometrial dating profile remains controversial . Independent randomized clinical trials found no improvement in reproductive outcomes , indicating the inefficacy of this clinical intervention and highlighting the need for alternative treatments and/or a refined target patient population that bene fits from this strategy. Combining these precision medicine approaches with transcriptomics and AI algorithms; our group recently proposed a novel molecular taxonomy of RIF and a new clinical algorithm for endometrial-factor infertility that considered 2 molecular causes of implantation failure: a WOI displacement that indicates a dysregulated endometrial timing; and a WOI disruption that indicates an endometrial alteration independent of endometrial timing . Transcriptomic analysis revealed these 2 WOI phenotypes were presented exclusively or simultaneously and patients with RIF had significantly more WOI disruptions but a similar frequency of WOI displacements, compared with controls . Thesefindings suggested that WOI disruptions had stronger associations with implantation failure than WOI displacements and shifted the paradigm for the study and treatment of endometrialfactor infertility . In addition, this work by SebastianLeon et al. highlighted the importance of removing the transcriptomic variation due to cyclical endometrial tissue changes (ie, the effect of endometrial timing) that, if not corrected, could mask the significant disruptions in endometrial gene expression. However, given this hypothesis-driven research was conducted retrospectively and in silico with publicly available data, the reproductive outcomes of the cycle after the biopsy collection were not available . This clinical follow-up is required to discover biomarkers for WOI disruption in patients with a higher risk of endometrial failure. On the basis of these genomics-driven precision medicine strategies, this proof-of-concept study aimed to discover a biomarker panel that could identify endometrial disruption, independent of endometrial timing, to predict, and ultimately prevent, poor reproductive outcomes in in vitro fertilization (IVF) patients.',\n",
       "  'Methods': 'MATERIAL AND METHODS Ethics statement This study was approved by the Ethics Committee of the Instituto Valenciano de Infertilidad, Valencia, Spain (1810-FIVI-066-PD/1810-FIVI-048-PD). Written informed consent was obtained from all recruited patients (n1⁄4 281). Participants, collection of endometrial biopsies, and reproductive outcomes This prospective, multicentric study was conducted between July 2018 and July 2021 at 5 private fertility clinics in Spain. Participants (n1⁄4 281) were scheduled for endometrial evaluation due to medical indications orR2 implantation failures of idiopathic origin. All patients met the following inclusion criteria: 18- 50 years old, with a body mass index (BMI) of 19- 30 kg/m 2, undergoing hormone replacement therapy for single embryo transfer (SET), presenting an endometrial thickness >6.5 mm with trilaminar structure in proliferative phase before P4 administration, and serum levels of estradiol >100 pg/mL and P4<1 ng/mL on the 10th day of estrogen treatment . Endometrial biopsies were collected after an average of 114.5/C6 7.2 h of P4 (duration ranged from 82 to 172 h), and processed as we previously described . Baseline patient characteristics including age, BMI, the number of IVF attempts, the quality of the embryo transferred (defined in theSupplemental Methods, available online), and reproductive outcomes (ie, pregnancy, live birth, and biochemical and clinical miscarriage) for each attempt, were obtained from internal medical records. Information about IVF attempts in other fertility clinics was included if recorded (only in 47.7% of patients). Study design First, a custom RNA-seq gene panel (the endometrial failure risk [EFR] panel) comprising 404 biomarkers of endometrial luteal phase timing and/or potential disruption was designed and employed to determine the risk of endometrial failure among 217 IVF patients (Supplemental Fig. 1, step 1 available online). The endometrial timing-corrected expression of the EFR panel was used to study the differential expression associated with endometrial disruption together with clinical characteristics of IVF patients to stratify the population as having poor or good endometrial prognoses, according to a semi-supervised learning process (Supplemental Fig. 1, step 2). The distinct transcriptomic patterns of each prognosis group were associated with the differences in reproductive outcomes of SET after biopsy collection, and the differentially expressed gene signature was proposed as a biomarker of endometrial disruption (Supplemental Fig. 1, step 3). Finally, the gene expression signature was used to supervise machine learning algorithms to evaluate the biomarker capacity of this signature to distinguish endometrial competence. The predictive performance of the models was assessed through the accuracy, sensibility, and specificity over 100 iterations of fivefold strati fied cross-validation ( Supplemental Fig. 1 , step 4). Constructing the EFR gene panel Supplemental Figure 2illustrates the workflow for the EFR panel gene selection. A whole-transcriptome endometrial data set (GSE58144, n1⁄4 115) retrieved from Gene Expression Omnibus was processed to correct the endometrial progression effect, using linear models in limma (version 3.46.0) , and unmask genes related to endometrial disruption . These genes were ranked according to an informativity score (CorrelationAttributeEval) in Weka (version 3.8.2, December 22, 2017) to evaluate their worth by measuring their correlation with endometrial pathology. Gene sets with increasing gene signature size were developed by adding the most informative genes in a stepwise manner, and their predictive performance was analyzed using Support Vector Machine and Random Forest algorithms. After calculating predictive performance with a stratified fivefold cross-validation (default parameters; 80:20; 100 iterations) using the Weka R package , the optimal signature size was determined for each algorithm, as the one with the fewest genes and highest accuracy. Finally, genes we previously prioritized for transcriptomic endometrial dating w e r ea l s oi n c l u d e di nt h e EFR panel to be able to remove the endometrial luteal phase timing variation. Endometrial biopsy processing, sequencing, and transcriptomic preprocessing Total RNA was extracted, and only samples that met our quality criteria were sequenced using the TruSeq Targeted RNA Illumina protocol and our custom EFR panel, in the Illumina NextSeq 550 platform with a paired-end design of 150 cycles and 4 M reads/sample, as previously described . Raw transcriptomic data was processed and normalized using limma, to filter low-quality raw counts (Q< 30) and low-read samples (<2M ). Possible batch effects were detected using principal components analysis and corrected with linear models in limma. Finally, endometrial luteal phase timing effects were detected using our transcriptomic endometrial dating (TED) model (see Supplemental Methods) and removed using limma. EFR biomarker signature identification A preliminary classification of patients was based on the patients’ history of reproductive outcomes and transcriptomic patterns corrected for endometrial luteal phase timing. This approach eliminated the gene expression variation related to the cyclic tissue changes that mask endometrial disruption and was used before applying semi-supervised machine learning on the basis of self-labeled techniques (see Supplemental Methods). Once, the genomic-driven disease taxonomy was established, the differentially expressed genes obtained between the final good and poor prognosis groups were proposed as the EFR biomarker signature. The limma R package was used to identify the statistically signi ficant differentially expressed genes (false discovery rate [FDR]-adjustedP value <.05). Functional characterization of the EFR signature The biomarker signature genes were annotated for functional interpretation databases by consulting Kyoto Encyclopaedia of Genes and Genomes human pathways (release 99.0, August 1, 2021) and Gene Ontology (version July 2, 2021; employing only experimental annotations) . Functional terms were manually summarized into categories according to their description in the database and presented as a bar plot using ggplot2 . EFR signature capability as a biomarker of endometrial disruption To estimate the prediction capability of the EFR signature to distinguish poor and good endometrial prognosis profiles, we implemented predictive models on the basis of the Support Vector Machine algorithm. Using a stratified fivefold crossvalidation process (repeated 100 times), the accuracy, sensitivity, and specificity were estimated. The range of values obtained across the 100 iterations was presented as a boxplot using ggplot2. Statistical analysis Rates for reproductive outcomes (ie, pregnancy rate [PR], cumulative pregnancy rate [CPR], live birth rate [LBR], biochemical miscarriage rate [BMR], and clinical miscarriage rate [CMR]) were calculated as de fined in the Supplemental Methods. Descriptive statistics were used for transcriptomic and clinical characteristics of patients to compare prognosis or WOI displacement groups. Continuous variables were presented as an overall mean /C6 standard deviation whereas discrete variables were presented as counts and percentages. Groups were compared using the Wilcoxon rank-sum test for continuous variables and Fisher’s exact test for discrete variables. All statistical analyses were conducted in R. All graphical results were generated with ggplot2. Gene expression validation by quantitative realtime polymerase chain reaction Quantitative real-time polymerase chain reaction (RT-qPCR) was performed to validate the expression of the top 4 differentially expressed genes between patients with poor (n1⁄4 10) and good (n1⁄4 10) endometrial prognoses. RNA (200- 220 ng) was reverse transcribed into cDNA using the PrimeScript Reagent Kit (Perfect Real Time, Takara, Japan) on a Thermocycler T3000 (Biometra, Ireland). Quantitative real-time polymerase chain reaction was performed on a StepOnePlus Real-Time PCR System (Applied Biosystems, USA) using Power-Up SYBR Green (Thermo Fisher Scientific, USA) and standard conditions.Supplemental Table 1(available online) lists the specific primers (Invitrogen, Thermo Fisher Scientific) used. Universal human RNA (Agilent, Spain) and water were respectively included as positive and negative controls. Each sample was analyzed in duplicate and relative gene expression (normalized to beta actin) was calculated using the DDCt method .',\n",
       "  'Results': 'RESULTS Preliminary sequencing with the 404-gene EFR panel The EFR panel evaluated 404 genes: 139 that predict endometrial disruption and 310 that serve as biomarkers of endometrial dating or timing. Notably, 45 genes overlapped in these 2 endometrial factors. Of the 281 endometrial samples collected, we excluded 56 samples for having low-quality or insufficient RNA; 6 samples containing a low number of sequencing reads (Supplemental Fig. 3A); and 2 samples that had low gene expression after normalization ( Supplemental Fig. 3B ). Thus, 217 patients were considered in the downstream analysis. After correcting batch effects from sequencing runs (Supplemental Fig. 3C) and removing the transcriptomic variation due to endometrial cyclic changes (endometrial luteal phase timing;Supplemental Fig. 3D), there were no other confounding technical or clinical factors (Supplemental Fig. 4). A genomic-driven endometrial taxonomy for patient stratification on the basis of a novel endometrial disruption independent of endometrial timing Forty-four patients were initially categorized as acute RIF on the basis of the inability to carry a pregnancy to term after 3- 9 SETs. Alternatively, 40 patients were categorized as acute Fertile because they reached full-term pregnancy at thefirst SET after biopsy. Notably, after this clinical acute criteria, 133 (61.3%) patients remained unclassified for having insufficient attempts (Supplemental Fig. 5A), and no transcriptomic differences were found between the acute RIF and acute Fertile groups (the lowest FDR was 0.3027). Incorporation of transcriptomic data with this initial acute clinical classification allowed us to stratify patients with poor (n1⁄4 137, 63.1%) or good (n1⁄4 49, 22.6%) endometrial prognoses; leaving only 31 samples (14.3%) unclassified because of the lack of recognition by the algorithms (Supplemental Fig. 5B). EFR biomarker signature identification and functional characterization A total of 122 genes were identified as significantly different (30.2% of all panel genes, FDR< 0.05) between patients with good and poor endometrial prognoses, including 63 downregulated and 59 upregulated in patients with poor endometrial prognosis (Table 1 and Fig. 1A), and proposed as the EFR biomarker signature. Transcriptomically, endometrial biopsies clustered by prognosis (Fig. 1B), and the signature genes had functions that were mainly related to regulation (17%); metabolism (8.4%); the immune system and inflammation (7.8%); protein processes (7.3%), signaling (6.9%), transport (6.1%), and the cell cycle, proliferation, and differentiation (5.3%) (Fig. 1C andSupplemental Table 2). The poor prognosis profile was distinguished by a downregulation of 27/39 (69.2%) functional groups. We highlight the functions with higher proportions of downregulated genes as steroidogenesis and steroid signaling (77.8%), hormone-related (80.0%), the nervous system (81.3%), gametogenesis and reproduction (85.7%), and oxidative stress and response to O 2 (88.9%), as well as cell action potential and polarization (100%) (Fig. 1D). Few functional groups, including cell migration, extracellular matrix, and glucose homeostasis contained a larger proportion of upregulated genes (60.9%, 62.5%, and 62.5% upregulated genes, respectively;Fig. 1D). Differences in clinical reproductive outcomes authenticated the EFR signature’s clinical relevance There were no significant differences in the age or BMI of patients predicted to have good/poor endometrial prognosis (all P>.05). Women predicted to have poor endometrial prognosis previously underwent significantly more SETs than those with good endometrial prognosis (2.8 vs. 1.6;P<.0001), however, there were no significant differences in embryo quality (P1⁄4.4679) between groups (Supplemental Table 3). The lack of differences in the proportion of out-of-phase endometria (57% in poor vs. 47% in good,P1⁄4.25) between good and poor prognosis groups (Supplemental Table 3) distinguished this signature from established endometrial luteal phase timing phenotypes (displaced or on-time). Therefore, the significant differences (P<.0001) between the reproductive outcomes of women predicted to have poor and good endometrial prognoses validated the clinical relevance of our transcriptomic signature. Specifically, the poor endometrial prognosis was associated with a 34.5% decline in PR (P1⁄43.8E-05), a 52% reduction in LBR (P1⁄45E-10), a 19.6% rise in CMR (P1⁄4.0066), and 20.4% increase in BMR (P1⁄4.0023) (Fig. 2A). Furthermore, patients with a poor endometrial prognosis only achieved a CPR of 42.3% after the ninth SET, whereas patients with a good prognosis had an 83.7% CPR at thefifth attempt (Fig. 2B). The relative risk of implantation failure or miscarriage in women predicted to have poor endometrial prognosis was respectively 2.7 and 11.5 compared with women with good endometrial prognosis. However, when failures in implantation or ongoing pregnancy were considered together, the relative risk of endometrial failure (not achieving a live birth in the next SET) for patients with poor endometrial prognosis was 3.3. EFR signature biomarker’s predictive capability The 186 samples labeled with poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognoses were used to evaluate the signature’s predictive performance. The 122 differentially expressed genes between poor/good endometrial prognoses proposed as the EFR biomarker signature were employed to train the model. Cross-validation yielded a median accuracy of 0.92 (min1⁄4 0.88, max 1⁄4 0.94), sensitivity of 0.96 (min1⁄4 0.91, max 1⁄4 0.98), and specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88) (Fig. 2C). Notably, 163 of 186 (87.6%) samples were classified correctly in almost 90% of all cross-validation models, whereas 151 of 186 (81.2%) samples were well classified correctly in 100% of the iterations. Twelve and eleven samples were misclassified consistently, in <50% or 50%- 90% of models, respectively (Fig. 2D). Clinical algorithm results after evaluating both WOI factors Next, we grouped samples according to both potential causes of implantation failure, endometrial luteal phase timing (displaced or on-time) (Fig. 3A), and endometrial disruption (good or poor prognosis) (Fig. 2A) and compared the patients’ reproductive outcomes. There were no significant differences between the patients with displaced and on-time WOI in terms of BMI (P1⁄4.8740), the average number of SETs (P1⁄4.0879), and embryo quality (P1⁄41). However, patients with a displaced WOI were signi ficantly older (40.4 vs. 38.3 years old; P1⁄4.0044) (Supplemental Table 4). Once patients were stratified by prognosis (endometrial disruption), the proportion of patients with poor prognosis was similar (P1⁄4.2458) between displaced and on-time groups (Fig. 3B). Experimental validation of genes related to the EFR signature The expression of the top 4 differentially expressed genes in the poor prognosis group was validated by RT-qPCR. Progestagen-associated endometrial protein and G-coupled protein receptor 110, which were found upregulated by RNA-seq, were significantly overexpressed in the poor prognosis group, compared with the good prognosis group (fold change [FC]1⁄4 6.157 [P1⁄4.0288] and FC1⁄4 11.803 [P1⁄4.0350], respectively; Supplemental Figs. 6Aand B). Meanwhile, the expression of fibrinogen beta chain and solute carrier 16 member 6, which were found downregulated by RNA-seq, did not reach a statistically significant difference (FC1⁄4 1.582 [P1⁄4.2475] and FC1⁄4 2.311 [P1⁄4.4376], respectively;Supplemental Figs. 6Cand D).',\n",
       "  'Discussion': 'DISCUSSION Advances in the diagnosis and treatment of RIF have been limited by its elusive etiology, the discrepancies in clinical classification (mainly due to the lack of consensus over the minimum number of pregnancy attempts) , and the heterogeneous molecular profiles of affected patients . On the basis of current clinical practice, patients with RIF may be misclassified and/or receive inef ficient treatment . Thus, the aim of this study was to propose a novel gene expression signature that could identify endometrial disruptions independent of endometrial luteal phase timing. Using a semi-supervised learning AI strategy together with the endometrial gene expression obtained from our custom EFR panel, we stratified patients according to poor or good endometrial prognoses and demonstrated how this signature reliably predicted gene expression patterns that corresponded with significant differences in PR, LBR, CMR, and BMR. When considering both WOI factors (disruption and timing) in our study cohort, poor endometrial prognoses (disruption) affected 73.7% of patients, whereas WOI displacements only affected 54.3%; 41.9% of patients presented a displaced WOI and were predicted to have a poor endometrial prognosis, whereas only 12.4% of patients presented with a displaced WOI and were predicted to have a good endometrial prognosis. These results offer a possible explanation as to why other studies reported there was no clinical benefito f personalizing embryo transfers on the basis of transcriptomic dating of the endometrium . According to our clinical algorithm, only patients with a displaced WOI and a good endometrial prognosis (12.4% of our cohort) would be better off with this approach. Patients with poor endometrial prognosis (73.7% of our cohort) would have suboptimal reproductive outcomes even though their WOI was on-time. This highlights an unmet healthcare need for IVF patients with endometrial disruptions and underscores the importance of additional prospective studies to further characterize patients on the basis of these 2 molecular causes for trying tailored treatments. Contemplating endometrial luteal phase timing together with a prevalent disruption or poor endometrial prognosis shifts the paradigm in studies about the endometrium. Our innovative methodology based on transcriptomics and AI models led us to develop new endometrial biomarker taxonomies related to endometrial luteal phase timing and/or endometrial disruption . There were 30 genes overlapping between the previously published TED signature and the EFR signature presented herein, emphasizing their correlation despite the lower FC of biomarkers for endometrial disruption versus endometrial timing . AI algorithms offer powerful approaches for patient stratification, by combining high-throughput molecular data and clinical variables . Clinical translation of these approaches will require careful supervision of how the machine learning is initially trained . However, a semi-supervised learning methodology implementing accurate clinical diagnostic criteria and patient-specific genomic data to establish relevant molecular taxonomies will be indispensable for advancing precision medicine , especially within the context of endometrial-factor infertility. Predictive biomarkers can stratify patients with endometrial failure even if the number of their attempts was insufficient to meet clinical RIF classification criteria . To our knowledge, this is thefirst study to apply semi-supervised learning to the endometrium, conferring an advantage over previous models that were only trained with clinical classifications and did not leverage the underlying molecular heterogeneity of the condition . The lack of gene expression differences between RIF and control patients with our initial clinical classification reinforced findings from Macklon’s group and the molecular heterogeneity among patients with RIF . The EFR signature comprised 122 genes, including 44 genes newly associated with an endometrial gene expression signature and a single common gene with Koot’s pioneer endometrial timing-independent signature for endometrial pathology . Here, we only considered biopsies collected in the endometrial mid-secretory phase, when patients had received P4 for 82- 172 h. Although this signature is independent of endometrial luteal phase timing, further studies are required to confirm if the signature is detectable in other phases of the endometrial cycle. Progestagen-associated endometrial protein is well characterized in endometrial tissue and was the most upregulated gene (FC1⁄4 3.18) in the poor endometrial prognosis pro file. However, the levels we observed were 10-fold lower compared with previous studies where changes in endometrial timing were considered . Alternatively,fibrinogen beta chain, related to the plasma membrane and vesicle transport by secretory granules, was the most downregulated gene. This gene accumulated polymorphisms that hinder the production offibrinogen and its associated anti-inflammatory activity and was recently related to the dysregulated pathways in RIF . FOXP1, the most predictive gene, is a transcription factor mediated by estrogen that has established functions in regulation, leukocyte differentiation, and response to lipids and chemokines. This gene is associated with breast endometrial cancer , and endometriosis-related fibrosis , but was never included in an endometrial gene signature until now. With FOXP1 and the FOX transcription factor family being associated with the regulation of the menstrual cycle and implantation , this evidence reinforces our prior study and affirms that alterations in endometrial regulation are an underlying cause of endometrial failure. Our clinical algorithm considers the 2 WOI factors, endometrial timing and disruption, to stratify IVF patients on the basis of 4 possible phenotypes: displaced/on-time and poor/ good prognosis. This model is promising for preventive and precision medicine in endometrial-factor infertility, as it combines the robust prediction parameters of the TED model and the EFR biomarker signature’s predictive capability reported herein. In contrast to other transcriptomic-based endometrial dating tools , our bioinformatic methodology removes the molecular variation associated with the cyclic menstrual changes, providing deeper insights for identifying a new kind of disruption independent of endometrial luteal phase timing by the EFR signature. Furthermore, the EFR biomarker signature was developed on the basis of reproductive outcomes and effectively classifies patients predicted to have positive or negative reproductive outcomes in the first SET after biopsy collection. All these facts, point out to a more potential clinical utility than existing tools that only predict the WOI without showing significant differences in reproductive outcomes as have been reported and we have shown in ourfindings. Although our strategy appears promising, the classification presented herein needs further re finement before clinical implementation. Larger sample sizes and validation in independent samples are required to ensure model generalization . Healthcare professionals will need to be adequately trained to ensure technical reproducibility. Longitudinal assessments including several menstrual cycles in the same patient are recommended to confirm inter-cycle reproducibility and to gain a comprehensive understanding of the reliability of the EFR panel. Additional studies should assess the relationship between a poor endometrial prognosis and the presence of benign disorders that potentially compromise endometrial function and/or embryo implantation . Inspired by the MammaPrint risk signature for breast cancer , the EFR signature could become a nextgeneration tool used to classify patients on the basis of endometrial prognosis, to predict reproductive outcomes before undergoing IVF, potentially experiencing RIF and/or losing valuable embryos. Although treatments for patients with poor endometrial prognosis remain to be determined, clinically identifying these patients is the first step toward improving their care.',\n",
       "  'Conclusion': 'Conclusion(s): The EFR signature revealed a novel endometrial disruption, independent of endometrial luteal phase timing, present in 73.7% of patients. This EFR signature stratified patients into 2 significantly distinct and clinically relevant prognosis profiles providing opportunities for personalized therapy. Nevertheless, further validations are needed before implementing this gene signature as an artificial intelligence (AI)-based tool to reduce the risk of patients experiencing endometrial failure. (Fertil Steril /C2102024;-:-- -. /C2112024 by American Society for Reproductive Medicine.)',\n",
       "  'URL': 'https://doi.org/10.1016/j.fertnstert.2024.03.015',\n",
       "  'Journal': 'Fertil Steril'},\n",
       " {'PaperTitle': 'Predicting risk of endometrial failure: a biomarker signature that identifies a novel disruption independent of endometrial timing in patients undergoing hormonal replacement cycles',\n",
       "  'Publication': '2024',\n",
       "  'Authors': 'Patricia Diaz-Gimeno, Ph.D., Patricia Sebastian-Leon, Ph.D., Katharina Spath, Ph.D., Diana Marti-Garcia, M.Sc., Josefa Maria Sanchez-Reyes, Ph.D., Maria del Carmen Vidal, Ph.D., Almudena Devesa-Peiro, Ph.D., Immaculada Sanchez-Ribas, Ph.D., Asunta Martinez-Martinez, M.Sc., Nuria Pellicer, M.D., Ph.D., Dagan Wells, Ph.D., and Antonio Pellicer, M.D., Ph.D.',\n",
       "  'Email': 'patricia.diaz@ivirma.com or patricia_diaz@iislafe.es',\n",
       "  'Abstract': 'Objective: To propose a new gene expression signature that identifies endometrial disruptions independent of endometrial luteal phase timing and predicts if patients are at risk of endometrial failure. Design: Multicentric, prospective study. Setting: Reproductive medicine research department in a public hospital affiliated with private fertility clinics and a reproductive genetics laboratory. Patients: Caucasian women (n1⁄4 281; 39.4/C6 4.8 years old with a body mass index of 22.9/C6 3.5 kg/m 2) undergoing hormone replacement therapy between July 2018 and July 2021. Endometrial samples from 217 patients met RNA quality criteria for signature discovery and analysis. Intervention(s): Endometrial biopsies collected in the mid-secretory phase. Main Outcome Measure(s):Endometrial luteal phase timing-corrected expression of 404 genes and reproductive outcomes of thefirst single embryo transfer (SET) after biopsy collection to identify prognostic biomarkers of endometrial failure. Results: Removal of endometrial timing variation from gene expression data allowed patients to be stratified into poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognosis groups on the basis of their clinical and transcriptomic profiles. Significant differences were found between endometrial prognosis groups in terms of reproductive rates: pregnancy (44.6% vs. 79.6%), live birth (25.6% vs. 77.6%), clinical miscarriage (22.2% vs. 2.6%), and biochemical miscarriage (20.4% vs. 0%). The relative risk of endometrial failure for patients predicted as a poor endometrial prognosis was 3.3 times higher than those with a good prognosis. The differences in gene expression between both profiles were proposed as a biomarker, coined the endometrial failure risk (EFR) signature. Poor prognosis profiles were characterized by 59 upregulated and 63 downregulated genes mainly involved in regulation (17.0%), metabolism (8.4%), immune response, and inflammation (7.8%). This EFR signature had a median accuracy of 0.92 (min1⁄40.88, max1⁄40.94), median sensitivity of 0.96 (min1⁄40.91, max 1⁄4 0.98), and median specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88), positioning itself as a promising biomarker for endometrial evaluation. Conclusion(s): The EFR signature revealed a novel endometrial disruption, independent of endometrial luteal phase timing, present in 73.7% of patients. This EFR signature stratified patients into 2 significantly distinct and clinically relevant prognosis profiles providing opportunities for personalized therapy. Nevertheless, further validations are needed before implementing this gene signature as an artificial intelligence (AI)-based tool to reduce the risk of patients experiencing endometrial failure. (Fertil Steril /C2102024;-:-- -. /C2112024 by American Society for Reproductive Medicine.) Key Words: artificial intelligence, precision medicine, patient stratification, endometrial-factor infertility, gene expression signature',\n",
       "  'Introduction': 'A key factor for reproductive success in assisted reproduction treatments is the status of the maternal endometrium during embryo implantation and fetal development. The endometrial cycle is reflected by the cyclic structural and functional changes of the endometrium across the menstrual cycle, particularly during the mid-secretory phase, to prepare for the‘‘window of implantation’’ (WOI) . After successful embryo implantation, the decidua (specialized layer of the endometrium) actively encapsulates the trophectoderm to support placentation and provide adequate vascularization for optimal fetal growth . Thus, approaches that predict and prevent endometrial-factor infertility could substantially improve assisted reproduction treatment outcomes by supporting the establishment and maintenance of pregnancy . There is a lack of consensus on the minimum number of implantation failures with good-quality embryos derived from ovum donation or with confirmed euploid karyotypes required to diagnose recurrent implantation failure (RIF) . The disparities in clinical classification and heterogeneous etiology of RIF confound its diagnosis and leave the exact prevalence unclear . As clinical symptoms cannot be prevented and do not provide sufficient criteria to stratify patients, patients might be misclassi fied and receive inefficient treatment. Indeed, most studies using clinical criteria for RIF identified no or few differentially expressed genes between samples of affected patients , reflecting the molecular heterogeneity of RIF. To overcome these healthcare challenges, high-throughput molecular technologies are combined with artificial intelligence (AI) algorithms to provide a deeper understanding of infertilityrelated pathogenesis . These molecular insights are then applied in genomics-driven precision medicine approaches to characterize complex diseases, and ultimately, improve patient diagnosis and prognosis . Employing genomics in reproductive medicine has the potential to go beyond treating the clinical manifestations of disease progression , by leveraging the patients’ heterogeneous molecular profiles for precise stratification and revealing new targets for preventive measures . In this regard, current diagnostic tools for endometrialfactor infertility use transcriptomics to date the endometrial biopsy in the mid-secretory phase, exceeding the accuracy and objectivity of traditional endometrial progression staging by Noyes’ histological criteria . However, the clinical utility of synchronizing the endometrium by adjusting the timing of progesterone (P4) administration on the basis of the patient’s transcriptomic endometrial dating profile remains controversial . Independent randomized clinical trials found no improvement in reproductive outcomes , indicating the inefficacy of this clinical intervention and highlighting the need for alternative treatments and/or a refined target patient population that bene fits from this strategy. Combining these precision medicine approaches with transcriptomics and AI algorithms; our group recently proposed a novel molecular taxonomy of RIF and a new clinical algorithm for endometrial-factor infertility that considered 2 molecular causes of implantation failure: a WOI displacement that indicates a dysregulated endometrial timing; and a WOI disruption that indicates an endometrial alteration independent of endometrial timing . Transcriptomic analysis revealed these 2 WOI phenotypes were presented exclusively or simultaneously and patients with RIF had significantly more WOI disruptions but a similar frequency of WOI displacements, compared with controls . Thesefindings suggested that WOI disruptions had stronger associations with implantation failure than WOI displacements and shifted the paradigm for the study and treatment of endometrialfactor infertility . In addition, this work by SebastianLeon et al. highlighted the importance of removing the transcriptomic variation due to cyclical endometrial tissue changes (ie, the effect of endometrial timing) that, if not corrected, could mask the significant disruptions in endometrial gene expression. However, given this hypothesis-driven research was conducted retrospectively and in silico with publicly available data, the reproductive outcomes of the cycle after the biopsy collection were not available . This clinical follow-up is required to discover biomarkers for WOI disruption in patients with a higher risk of endometrial failure. On the basis of these genomics-driven precision medicine strategies, this proof-of-concept study aimed to discover a biomarker panel that could identify endometrial disruption, independent of endometrial timing, to predict, and ultimately prevent, poor reproductive outcomes in in vitro fertilization (IVF) patients.',\n",
       "  'Methods': 'MATERIAL AND METHODS\\nEthics statement\\nThis study was approved by the Ethics Committee of the Instituto Valenciano de Infertilidad, Valencia, Spain (1810-FIVI-066-PD/1810-FIVI-048-PD). Written informed consent was obtained from all recruited patients (n1⁄4 281).\\nParticipants, collection of endometrial biopsies, and reproductive outcomes\\nThis prospective, multicentric study was conducted between July 2018 and July 2021 at 5 private fertility clinics in Spain. Participants (n1⁄4 281) were scheduled for endometrial evaluation due to medical indications orR2 implantation failures of idiopathic origin. All patients met the following inclusion criteria: 18- 50 years old, with a body mass index (BMI) of 19- 30 kg/m 2, undergoing hormone replacement therapy for single embryo transfer (SET), presenting an endometrial thickness >6.5 mm with trilaminar structure in proliferative phase before P4 administration, and serum levels of estradiol >100 pg/mL and P4<1 ng/mL on the 10th day of estrogen treatment . Endometrial biopsies were collected after an average of 114.5/C6 7.2 h of P4 (duration ranged from 82 to 172 h), and processed as we previously described . Baseline patient characteristics including age, BMI, the number of IVF attempts, the quality of the embryo transferred (defined in theSupplemental Methods, available online), and reproductive outcomes (ie, pregnancy, live birth, and biochemical and clinical miscarriage) for each attempt, were obtained from internal medical records. Information about IVF attempts in other fertility clinics was included if recorded (only in 47.7% of patients).\\nStudy design\\nFirst, a custom RNA-seq gene panel (the endometrial failure risk [EFR] panel) comprising 404 biomarkers of endometrial luteal phase timing and/or potential disruption was designed and employed to determine the risk of endometrial failure among 217 IVF patients (Supplemental Fig. 1, step 1 available online). The endometrial timing-corrected expression of the EFR panel was used to study the differential expression associated with endometrial disruption together with clinical characteristics of IVF patients to stratify the population as having poor or good endometrial prognoses, according to a semi-supervised learning process (Supplemental Fig. 1, step 2). The distinct transcriptomic patterns of each prognosis group were associated with the differences in reproductive outcomes of SET after biopsy collection, and the differentially expressed gene signature was proposed as a biomarker of endometrial disruption (Supplemental Fig. 1, step 3). Finally, the gene expression signature was used to supervise machine learning algorithms to evaluate the biomarker capacity of this signature to distinguish endometrial competence. The predictive performance of the models was assessed through the accuracy, sensibility, and specificity over 100 iterations of fivefold strati fied cross-validation ( Supplemental Fig. 1 , step 4).\\nConstructing the EFR gene panel\\nSupplemental Figure 2illustrates the workflow for the EFR panel gene selection. A whole-transcriptome endometrial data set (GSE58144, n1⁄4 115) retrieved from Gene Expression Omnibus was processed to correct the endometrial progression effect, using linear models in limma (version 3.46.0) , and unmask genes related to endometrial disruption . These genes were ranked according to an informativity score (CorrelationAttributeEval) in Weka (version 3.8.2, December 22, 2017) to evaluate their worth by measuring their correlation with endometrial pathology. Gene sets with increasing gene signature size were developed by adding the most informative genes in a stepwise manner, and their predictive performance was analyzed using Support Vector Machine and Random Forest algorithms. After calculating predictive performance with a stratified fivefold cross-validation (default parameters; 80:20; 100 iterations) using the Weka R package , the optimal signature size was determined for each algorithm, as the one with the fewest genes and highest accuracy. Finally, genes we previously prioritized for transcriptomic endometrial dating w e r ea l s oi n c l u d e di nt h e EFR panel to be able to remove the endometrial luteal phase timing variation.\\nEndometrial biopsy processing, sequencing, and transcriptomic preprocessing\\nTotal RNA was extracted, and only samples that met our quality criteria were sequenced using the TruSeq Targeted RNA Illumina protocol and our custom EFR panel, in the Illumina NextSeq 550 platform with a paired-end design of 150 cycles and 4 M reads/sample, as previously described . Raw transcriptomic data was processed and normalized using limma, to filter low-quality raw counts (Q< 30) and low-read samples (<2M ). Possible batch effects were detected using principal components analysis and corrected with linear models in limma. Finally, endometrial luteal phase timing effects were detected using our transcriptomic endometrial dating (TED) model (see Supplemental Methods) and removed using limma.\\nEFR biomarker signature identification\\nA preliminary classification of patients was based on the patients’ history of reproductive outcomes and transcriptomic patterns corrected for endometrial luteal phase timing. This approach eliminated the gene expression variation related to the cyclic tissue changes that mask endometrial disruption and was used before applying semi-supervised machine learning on the basis of self-labeled techniques (see Supplemental Methods). Once, the genomic-driven disease taxonomy was established, the differentially expressed genes obtained between the final good and poor prognosis groups were proposed as the EFR biomarker signature. The limma R package was used to identify the statistically signi ficant differentially expressed genes (false discovery rate [FDR]-adjustedP value <.05).\\nFunctional characterization of the EFR signature\\nThe biomarker signature genes were annotated for functional interpretation databases by consulting Kyoto Encyclopaedia of Genes and Genomes human pathways (release 99.0, August 1, 2021) and Gene Ontology (version July 2, 2021; employing only experimental annotations) . Functional terms were manually summarized into categories according VOL. - NO. - / - 2024 3 Fertil Steril® to their description in the database and presented as a bar plot using ggplot2 .\\nEFR signature capability as a biomarker of endometrial disruption\\nTo estimate the prediction capability of the EFR signature to distinguish poor and good endometrial prognosis profiles, we implemented predictive models on the basis of the Support Vector Machine algorithm. Using a stratified fivefold crossvalidation process (repeated 100 times), the accuracy, sensitivity, and specificity were estimated. The range of values obtained across the 100 iterations was presented as a boxplot using ggplot2.\\nStatistical analysis\\nRates for reproductive outcomes (ie, pregnancy rate [PR], cumulative pregnancy rate [CPR], live birth rate [LBR], biochemical miscarriage rate [BMR], and clinical miscarriage rate [CMR]) were calculated as de fined in the Supplemental Methods. Descriptive statistics were used for transcriptomic and clinical characteristics of patients to compare prognosis or WOI displacement groups. Continuous variables were presented as an overall mean /C6 standard deviation whereas discrete variables were presented as counts and percentages. Groups were compared using the Wilcoxon rank-sum test for continuous variables and Fisher’s exact test for discrete variables. All statistical analyses were conducted in R. All graphical results were generated with ggplot2.\\nGene expression validation by quantitative realtime polymerase chain reaction\\nQuantitative real-time polymerase chain reaction (RT-qPCR) was performed to validate the expression of the top 4 differentially expressed genes between patients with poor (n1⁄4 10) and good (n1⁄4 10) endometrial prognoses. RNA (200- 220 ng) was reverse transcribed into cDNA using the PrimeScript Reagent Kit (Perfect Real Time, Takara, Japan) on a Thermocycler T3000 (Biometra, Ireland). Quantitative real-time polymerase chain reaction was performed on a StepOnePlus Real-Time PCR System (Applied Biosystems, USA) using Power-Up SYBR Green (Thermo Fisher Scientific, USA) and standard conditions.Supplemental Table 1(available online) lists the specific primers (Invitrogen, Thermo Fisher Scientific) used. Universal human RNA (Agilent, Spain) and water were respectively included as positive and negative controls. Each sample was analyzed in duplicate and relative gene expression (normalized to beta actin) was calculated using the DDCt method .',\n",
       "  'Results': 'Preliminary sequencing with the 404-gene EFR panel\\nThe EFR panel evaluated 404 genes: 139 that predict endometrial disruption and 310 that serve as biomarkers of endometrial dating or timing. Notably, 45 genes overlapped in these 2 endometrial factors. Of the 281 endometrial samples collected, we excluded 56 samples for having low-quality or insufficient RNA; 6 samples containing a low number of sequencing reads (Supplemental Fig. 3A); and 2 samples that had low gene expression after normalization ( Supplemental Fig. 3B ). Thus, 217 patients were considered in the downstream analysis. After correcting batch effects from sequencing runs (Supplemental Fig. 3C) and removing the transcriptomic variation due to endometrial cyclic changes (endometrial luteal phase timing;Supplemental Fig. 3D), there were no other confounding technical or clinical factors (Supplemental Fig. 4).\\nA genomic-driven endometrial taxonomy for patient stratification on the basis of a novel endometrial disruption independent of endometrial timing\\nForty-four patients were initially categorized as acute RIF on the basis of the inability to carry a pregnancy to term after 3- 9 SETs. Alternatively, 40 patients were categorized as acute Fertile because they reached full-term pregnancy at thefirst SET after biopsy. Notably, after this clinical acute criteria, 133 (61.3%) patients remained unclassified for having insufficient attempts (Supplemental Fig. 5A), and no transcriptomic differences were found between the acute RIF and acute Fertile groups (the lowest FDR was 0.3027). Incorporation of transcriptomic data with this initial acute clinical classification allowed us to stratify patients with poor (n1⁄4 137, 63.1%) or good (n1⁄4 49, 22.6%) endometrial prognoses; leaving only 31 samples (14.3%) unclassified because of the lack of recognition by the algorithms (Supplemental Fig. 5B).\\nEFR biomarker signature identification and functional characterization\\nA total of 122 genes were identified as significantly different (30.2% of all panel genes, FDR< 0.05) between patients with good and poor endometrial prognoses, including 63 downregulated and 59 upregulated in patients with poor endometrial prognosis (Table 1 and Fig. 1A), and proposed as the EFR biomarker signature. Transcriptomically, endometrial biopsies clustered by prognosis (Fig. 1B), and the signature genes had functions that were mainly related to regulation (17%); metabolism (8.4%); the immune system and inflammation (7.8%); protein processes (7.3%), signaling (6.9%), transport (6.1%), and the cell cycle, proliferation, and differentiation (5.3%) (Fig. 1C andSupplemental Table 2). The poor prognosis profile was distinguished by a downregulation of 27/39 (69.2%) functional groups. We highlight the functions with higher proportions of downregulated genes as steroidogenesis and steroid signaling (77.8%), hormone-related (80.0%), the nervous system (81.3%), gametogenesis and reproduction (85.7%), and oxidative stress and response to O 2 (88.9%), as well as cell action potential and polarization (100%) (Fig. 1D). Few functional groups, including cell migration, extracellular matrix, and glucose homeostasis contained a larger proportion of upregulated genes (60.9%, 62.5%, and 62.5% upregulated genes, respectively;Fig. 1D).\\nDifferences in clinical reproductive outcomes authenticated the EFR signature’s clinical relevance\\nThere were no significant differences in the age or BMI of patients predicted to have good/poor endometrial prognosis (all P>.05). Women predicted to have poor endometrial prognosis previously underwent significantly more SETs than those with good endometrial prognosis (2.8 vs. 1.6;P<.0001), however, there were no significant differences in embryo quality (P1⁄4.4679) between groups (Supplemental Table 3). The lack TABLE 1 Continued. Gene ID Gene name FC FDR BDH1 3-Hydroxybutyrate Dehydrogenase 1 -1.1744 0.0030 ECI2 Enoyl-CoA Delta Isomerase 2 -1.1862 0.0178 ANK3 Ankyrin 3 -1.2084 0.0433 IL15 Interleukin 15 -1.2233 0.0198 GSN Gelsolin -1.2241 0.0076 GLIPR1 GLI Pathogenesis Related 1 -1.2251 0.0166 CBR3 Carbonyl Reductase 3 -1.2252 0.0124 ATP6V0E2 ATPase H þ Transporting V0 Subunit E2 -1.2304 0.0082 DYNLT3 Dynein Light Chain Tctex-Type 3 -1.2323 0.0199 S100A4 S100 Calcium Binding Protein A4 -1.2478 0.0041 BUB1B BUB1 Mitotic Checkpoint Serine/Threonine Kinase B -1.2482 0.0451 IDH1 Isocitrate Dehydrogenase (NADP( þ)) 1 -1.2488 0.0124 PLA2G16 Phospholipase A And Acyltransferase 3 -1.2498 0.0124 EPHB3 EPH Receptor B3 -1.2530 0.0396 OAZ3 Ornithine Decarboxylase Antizyme 3 -1.2581 0.0013 PYCR1 Pyrroline-5-Carboxylate Reductase 1 -1.2597 0.0007 FAM155B NALCN Channel Auxiliary Factor 2 -1.2618 0.0116 ECHS1 Enoyl-CoA Hydratase, Short Chain 1 -1.2621 3.68E-05 CSRP2 Cysteine And Glycine Rich Protein 2 -1.2647 0.0023 TLR4 Toll Like Receptor 4 -1.2716 0.0065 CTSW Cathepsin W -1.2764 0.0322 CEP55 Centrosomal Protein 55 -1.3021 0.0264 BARD1 BRCA1 Associated RING Domain 1 -1.3064 0.0014 PLA2G4A Phospholipase A2 Group IVA -1.3101 0.0008 ATP1B1 ATPase Na þ/Kþ Transporting Subunit Beta 1 -1.3202 0.0113 FOXO1 Forkhead Box O1 -1.3330 0.0018 BIRC3 Baculoviral IAP Repeat Containing 3 -1.3496 0.0007 CRISP3 Cysteine Rich Secretory Protein 3 -1.3498 0.0397 LMCD1 LIM And Cysteine Rich Domains 1 -1.3627 0.0012 PSMB8 Proteasome 20S Subunit Beta 8 -1.3783 3.68E-05 CD81 CD81 Molecule -1.3888 0.0104 ALDH1A3 Aldehyde Dehydrogenase 1 Family Member A3 -1.3891 0.0007 SQLE Squalene Epoxidase -1.3905 0.0001 KMO Kynurenine 3-Monooxygenase -1.4107 0.0434 RPRM Reprimo, TP53 Dependent G2 Arrest Mediator Homolog -1.4121 0.0061 SPP1 Secreted Phosphoprotein 1 -1.4130 0.0242 ANGPTL1 Angiopoietin Like 1 -1.4381 0.0017 HPRT1 Hypoxanthine Phosphoribosyltransferase 1 -1.4394 1.43E-05 ATP6V1A ATPase H þ Transporting V1 Subunit A -1.4484 0.0040 CDK1 Cyclin Dependent Kinase 1 -1.4651 0.0084 ENPEP Glutamyl Aminopeptidase -1.4704 0.0441 TH Tyrosine Hydroxylase -1.5149 0.0113 G0S2 G0/G1 Switch 2 -1.5183 0.0226 PROS1 Protein S -1.5297 0.0025 STAR Steroidogenic Acute Regulatory Protein -1.5342 0.0005 DKK1 Dickkopf WNT Signaling Pathway Inhibitor 1 -1.5425 0.0005 BANK1 B Cell Scaffold Protein With Ankyrin Repeats 1 -1.5553 0.0005 KCNJ2 Potassium Inwardly Rectifying Channel Subfamily J Member 2 -1.6602 0.0119 SLC7A2 Solute Carrier Family 7 Member 2 -1.6919 0.0038 PROK1 Prokineticin 1 -1.7218 0.0106 OLFM4 Olfactomedin 4 -1.7759 0.0458 IFNG Interferon Gamma -1.9666 0.0033 CXCL13 C-X-C Motif Chemokine Ligand 13 -2.0690 0.0001 SLC16A6 Solute Carrier Family 16 Member 6 -2.2113 9.12E-07 FGB Fibrinogen Beta Chain -2.3652 0.0108 Complete list of the 122 genes included in the endometrial failure risk (EFR) signature. Gene name and ID, fold change (FC) related to poor prognosis, and adjustedP value by false discovery rate (FDR) are indicated. Diaz-Gimeno. Endometrial Failure Risk signature. Fertil Steril 2024. 6 VOL. - NO. - / - 2024 REPRODUCTIVE SCIENCE of differences in the proportion of out-of-phase endometria (57% in poor vs. 47% in good,P1⁄4.25) between good and poor prognosis groups (Supplemental Table 3) distinguished this signature from established endometrial luteal phase timing phenotypes (displaced or on-time). Therefore, the significant differences (P<.0001) between the reproductive outcomes of women predicted to have poor and good endometrial prognoses validated the clinical relevance of our transcriptomic signature. Specifically, the poor endometrial prognosis was associated with a 34.5% decline in PR FIGURE 1 Transcriptomic and functional characterization of the endometrial failure risk (EFR) signature. (A) Volcano plot showing the 122 EFR signature genes. Red and blue dots were used to differentiate EFR genes that were significantly upregulated or downregulated in poor prognosis samples, respectively, whereas gray dots represented genes (n1⁄4 282) whose expression was not significantly different (FDR> 0.05) between poor and good prognoses groups and were not included in EFR signature. (B) Principal component analysis highlighting the transcriptomic behavior of endometrial samples from patients predicted to have poor (orange) or good (green) endometrial prognosis on the basis of genes included in EFR signature. Notably, the transcriptomic variance explained by the first and second principal components (PC1 and PC2, respectively) reached 23.1%. (C) Pie chart depicting the overall proportion of EFR signature genes included in each functional category. (D) The EFR signature genes were classified into 39 functional groups. Red and blue bars were used to differentiate genes that were significantly upregulated or downregulated in poor prognosis samples, respectively. FDR1⁄4 false discovery rate. Figure created withBioRender.com. Diaz-Gimeno. Endometrial Failure Risk signature. Fertil Steril 2024. VOL. - NO. - / - 2024 7 Fertil Steril® (P1⁄43.8E-05), a 52% reduction in LBR (P1⁄45E-10), a 19.6% rise in CMR (P1⁄4.0066), and 20.4% increase in BMR (P1⁄4.0023) (Fig. 2A). Furthermore, patients with a poor endometrial prognosis only achieved a CPR of 42.3% after the ninth SET, whereas patients with a good prognosis had an 83.7% CPR at thefifth attempt (Fig. 2B). The relative risk of implantation failure or miscarriage in women predicted to have poor endometrial prognosis was respectively 2.7 and 11.5 compared with women with good endometrial prognosis. However, when failures in implantation or ongoing pregnancy were considered together, the relative risk of endometrial failure (not achieving a live birth in the next SET) for patients with poor endometrial prognosis was 3.3.\\nEFR signature biomarker’s predictive capability\\nThe 186 samples labeled with poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognoses were used to evaluate the signature’s predictive performance. The 122 differentially expressed genes between poor/good endometrial prognoses proposed as the EFR biomarker signature were employed to train the model. Cross-validation yielded a median accuracy of 0.92 (min1⁄4 0.88, max 1⁄4 0.94), sensitivity of 0.96 (min1⁄4 0.91, max 1⁄4 0.98), and specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88) (Fig. 2C). Notably, 163 of 186 (87.6%) samples were classified correctly in almost 90% of all cross-validation models, whereas 151 of 186 (81.2%) samples were well classified correctly in 100% of the iterations. Twelve and eleven samples were misclassified consistently, in <50% or 50%- 90% of models, respectively (Fig. 2D).\\nClinical algorithm results after evaluating both WOI factors\\nNext, we grouped samples according to both potential causes of implantation failure, endometrial luteal phase timing (displaced or on-time) (Fig. 3A), and endometrial disruption (good or poor prognosis) (Fig. 2A) and compared the patients’ reproductive outcomes. There were no significant differences between the patients with displaced and on-time WOI in terms of BMI (P1⁄4.8740), the average number of SETs (P1⁄4.0879), and embryo quality (P1⁄41). However, patients with a displaced WOI were signi ficantly older (40.4 vs. 38.3 years old; P1⁄4.0044) (Supplemental Table 4). Once patients were stratified by prognosis (endometrial disruption), the proportion of FIGURE 2 Clinical meaning and prediction capability of the EFR signature. (A) Clinical outcomes of the single embryo transfer after biopsy collection. Rates for pregnancy, live birth, and clinical and biochemical miscarriage were calculated as described in theSupplemental Methods. Statistical differences were evaluated using Fisher’s exact test. (B) Comparison of cumulative pregnancy rates. Notably, many patients predicted to have a good endometrial prognosis achieved pregnancy with a single transfer (63.3%) and reached the maximum number of pregnancies (83.7%) at the fifth transfer, whereas patients predicted to have poor endometrial prognosis were only able to achieve a cumulative pregnancy rate of 42.3% at the ninth transfer. (C) Boxplots for the EFR signature prediction parameters (sensitivity, specificity, and accuracy) were calculated over 100 iterations of fivefold stratified cross-validation. (D) Histogram showing the proportion of patients that were successfully classified during the aforementioned cross-validation procedure. Figure created withBioRender.com. Diaz-Gimeno. Endometrial Failure Risk signature. Fertil Steril 2024. 8 VOL. - NO. - / - 2024 REPRODUCTIVE SCIENCE patients with poor prognosis was similar (P1⁄4.2458) between displaced and on-time groups (Fig. 3B).\\nExperimental validation of genes related to the EFR signature\\nThe expression of the top 4 differentially expressed genes in the poor prognosis group was validated by RT-qPCR. Progestagen-associated endometrial protein and G-coupled protein receptor 110, which were found upregulated by RNA-seq, were significantly overexpressed in the poor prognosis group, compared with the good prognosis group (fold change [FC]1⁄4 6.157 [P1⁄4.0288] and FC1⁄4 11.803 [P1⁄4.0350], respectively; Supplemental Figs. 6Aand B). Meanwhile, the expression of fibrinogen beta chain and solute carrier 16 member 6, which were found downregulated by RNA-seq, did not reach a statistically significant difference (FC1⁄4 1.582 [P1⁄4.2475] and FC1⁄4 2.311 [P1⁄4.4376], respectively;Supplemental Figs. 6Cand D).',\n",
       "  'Discussion': 'Advances in the diagnosis and treatment of RIF have been limited by its elusive etiology, the discrepancies in clinical classification (mainly due to the lack of consensus over the minimum number of pregnancy attempts) , and the heterogeneous molecular profiles of affected patients . On the basis of current clinical practice, patients with RIF may be misclassified and/or receive inef ficient treatment . Thus, the aim of this study was to propose a novel gene expression signature that could identify endometrial disruptions independent of endometrial luteal phase timing. Using a semi-supervised learning AI strategy together with the endometrial gene expression obtained from our custom EFR panel, we stratified patients according to poor or good endometrial prognoses and demonstrated how this signature reliably predicted gene expression patterns that corresponded with significant differences in PR, LBR, CMR, and BMR. When considering both WOI factors (disruption and timing) in our study cohort, poor endometrial prognoses (disruption) affected 73.7% of patients, whereas WOI displacements only affected 54.3%; 41.9% of patients presented a displaced WOI and were predicted to have a poor endometrial prognosis, whereas only 12.4% of patients presented with a displaced WOI and were predicted to have a good endometrial prognosis. These results offer a possible explanation as to why other studies reported there was no clinical benefito f personalizing embryo transfers on the basis of transcriptomic dating of the endometrium . According to our clinical algorithm, only patients with a displaced WOI and a good endometrial prognosis (12.4% of our cohort) would be better off with this approach. Patients with poor endometrial prognosis (73.7% of our cohort) would have suboptimal reproductive outcomes even though their WOI was on-time. This highlights an unmet healthcare need for IVF patients with endometrial disruptions and underscores the importance of additional prospective studies to further characterize patients on the basis of these 2 molecular causes for trying tailored treatments. FIGURE 3 Evaluation of reproductive outcomes after single embryo transfer (SET) on the basis of the 2 WOI factors. Patients were initially classified according to transcriptomic endometrial dating by TED ( A), as per current clinical practice. This approach distinguishes between patients whose window of implantation (WOI) was considered displaced or on time. These groups were then stratified according to their endometrial disruption (B), as determined by our semi-supervised stratification. In all cases, the reproductive outcomes of thefirst SET after biopsy collection were compared using Fisher/C19 s exact test. NS 1⁄4 no significant differences; TED 1⁄4 transcriptomic endometrial dating. * P<.05; ** P<.01; ***P<.001. Figure created withBioRender.com. Diaz-Gimeno. Endometrial Failure Risk signature. Fertil Steril 2024. VOL. - NO. - / - 2024 9 Fertil Steril® Contemplating endometrial luteal phase timing together with a prevalent disruption or poor endometrial prognosis shifts the paradigm in studies about the endometrium. Our innovative methodology based on transcriptomics and AI models led us to develop new endometrial biomarker taxonomies related to endometrial luteal phase timing and/or endometrial disruption . There were 30 genes overlapping between the previously published TED signature and the EFR signature presented herein, emphasizing their correlation despite the lower FC of biomarkers for endometrial disruption versus endometrial timing . AI algorithms offer powerful approaches for patient stratification, by combining high-throughput molecular data and clinical variables . Clinical translation of these approaches will require careful supervision of how the machine learning is initially trained . However, a semi-supervised learning methodology implementing accurate clinical diagnostic criteria and patient-specific genomic data to establish relevant molecular taxonomies will be indispensable for advancing precision medicine , especially within the context of endometrial-factor infertility. Predictive biomarkers can stratify patients with endometrial failure even if the number of their attempts was insufficient to meet clinical RIF classification criteria . To our knowledge, this is thefirst study to apply semi-supervised learning to the endometrium, conferring an advantage over previous models that were only trained with clinical classifications and did not leverage the underlying molecular heterogeneity of the condition . The lack of gene expression differences between RIF and control patients with our initial clinical classification reinforced findings from Macklon’s group and the molecular heterogeneity among patients with RIF . The EFR signature comprised 122 genes, including 44 genes newly associated with an endometrial gene expression signature and a single common gene with Koot’s pioneer endometrial timing-independent signature for endometrial pathology . Here, we only considered biopsies collected in the endometrial mid-secretory phase, when patients had received P4 for 82- 172 h. Although this signature is independent of endometrial luteal phase timing, further studies are required to confirm if the signature is detectable in other phases of the endometrial cycle. Progestagen-associated endometrial protein is well characterized in endometrial tissue and was the most upregulated gene (FC1⁄4 3.18) in the poor endometrial prognosis pro file. However, the levels we observed were 10-fold lower compared with previous studies where changes in endometrial timing were considered . Alternatively,fibrinogen beta chain, related to the plasma membrane and vesicle transport by secretory granules, was the most downregulated gene. This gene accumulated polymorphisms that hinder the production offibrinogen and its associated anti-inflammatory activity and was recently related to the dysregulated pathways in RIF . FOXP1, the most predictive gene, is a transcription factor mediated by estrogen that has established functions in regulation, leukocyte differentiation, and response to lipids and chemokines. This gene is associated with breast endometrial cancer , and endometriosis-related fibrosis , but was never included in an endometrial gene signature until now. With FOXP1 and the FOX transcription factor family being associated with the regulation of the menstrual cycle and implantation , this evidence reinforces our prior study and affirms that alterations in endometrial regulation are an underlying cause of endometrial failure. Our clinical algorithm considers the 2 WOI factors, endometrial timing and disruption, to stratify IVF patients on the basis of 4 possible phenotypes: displaced/on-time and poor/ good prognosis. This model is promising for preventive and precision medicine in endometrial-factor infertility, as it combines the robust prediction parameters of the TED model and the EFR biomarker signature’s predictive capability reported herein. In contrast to other transcriptomic-based endometrial dating tools , our bioinformatic methodology removes the molecular variation associated with the cyclic menstrual changes, providing deeper insights for identifying a new kind of disruption independent of endometrial luteal phase timing by the EFR signature. Furthermore, the EFR biomarker signature was developed on the basis of reproductive outcomes and effectively classifies patients predicted to have positive or negative reproductive outcomes in the first SET after biopsy collection. All these facts, point out to a more potential clinical utility than existing tools that only predict the WOI without showing significant differences in reproductive outcomes as have been reported and we have shown in ourfindings. Although our strategy appears promising, the classification presented herein needs further re finement before clinical implementation. Larger sample sizes and validation in independent samples are required to ensure model generalization . Healthcare professionals will need to be adequately trained to ensure technical reproducibility. Longitudinal assessments including several menstrual cycles in the same patient are recommended to confirm inter-cycle reproducibility and to gain a comprehensive understanding of the reliability of the EFR panel. Additional studies should assess the relationship between a poor endometrial prognosis and the presence of benign disorders that potentially compromise endometrial function and/or embryo implantation . Inspired by the MammaPrint risk signature for breast cancer , the EFR signature could become a nextgeneration tool used to classify patients on the basis of endometrial prognosis, to predict reproductive outcomes before undergoing IVF, potentially experiencing RIF and/or losing valuable embryos. Although treatments for patients with poor endometrial prognosis remain to be determined, clinically identifying these patients is the first step toward improving their care.',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1016/j.fertnstert.2024.03.015',\n",
       "  'Journal': 'Fertil Steril'},\n",
       " {'PaperTitle': 'Predicting risk of endometrial failure: a biomarker signature that identifies a novel disruption independent of endometrial timing in patients undergoing hormonal replacement cycles',\n",
       "  'Publication': '2024',\n",
       "  'Authors': 'Patricia Diaz-Gimeno, Ph.D., Patricia Sebastian-Leon, Ph.D., Katharina Spath, Ph.D., Diana Marti-Garcia, M.Sc., Josefa Maria Sanchez-Reyes, Ph.D., Maria del Carmen Vidal, Ph.D., Almudena Devesa-Peiro, Ph.D., Immaculada Sanchez-Ribas, Ph.D., Asunta Martinez-Martinez, M.Sc., Nuria Pellicer, M.D., Ph.D., Dagan Wells, Ph.D., and Antonio Pellicer, M.D., Ph.D.',\n",
       "  'Email': 'patricia.diaz@ivirma.com or patricia_diaz@iislafe.es',\n",
       "  'Abstract': 'Objective: To propose a new gene expression signature that identifies endometrial disruptions independent of endometrial luteal phase timing and predicts if patients are at risk of endometrial failure. Design: Multicentric, prospective study. Setting: Reproductive medicine research department in a public hospital affiliated with private fertility clinics and a reproductive genetics laboratory. Patients: Caucasian women (n1⁄4 281; 39.4/C6 4.8 years old with a body mass index of 22.9/C6 3.5 kg/m 2) undergoing hormone replacement therapy between July 2018 and July 2021. Endometrial samples from 217 patients met RNA quality criteria for signature discovery and analysis. Intervention(s): Endometrial biopsies collected in the mid-secretory phase. Main Outcome Measure(s):Endometrial luteal phase timing-corrected expression of 404 genes and reproductive outcomes of thefirst single embryo transfer (SET) after biopsy collection to identify prognostic biomarkers of endometrial failure. Results: Removal of endometrial timing variation from gene expression data allowed patients to be stratified into poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognosis groups on the basis of their clinical and transcriptomic profiles. Significant differences were found between endometrial prognosis groups in terms of reproductive rates: pregnancy (44.6% vs. 79.6%), live birth (25.6% vs. 77.6%), clinical miscarriage (22.2% vs. 2.6%), and biochemical miscarriage (20.4% vs. 0%). The relative risk of endometrial failure for patients predicted as a poor endometrial prognosis was 3.3 times higher than those with a good prognosis. The differences in gene expression between both profiles were proposed as a biomarker, coined the endometrial failure risk (EFR) signature. Poor prognosis profiles were characterized by 59 upregulated and 63 downregulated genes mainly involved in regulation (17.0%), metabolism (8.4%), immune response, and inflammation (7.8%). This EFR signature had a median accuracy of 0.92 (min1⁄40.88, max1⁄40.94), median sensitivity of 0.96 (min1⁄40.91, max 1⁄4 0.98), and median specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88), positioning itself as a promising biomarker for endometrial evaluation. Conclusion(s): The EFR signature revealed a novel endometrial disruption, independent of endometrial luteal phase timing, present in 73.7% of patients. This EFR signature stratified patients into 2 significantly distinct and clinically relevant prognosis profiles providing opportunities for personalized therapy. Nevertheless, further validations are needed before implementing this gene signature as an artificial intelligence (AI)-based tool to reduce the risk of patients experiencing endometrial failure. (Fertil Steril /C2102024;-:-- -. /C2112024 by American Society for Reproductive Medicine.) Key Words: artificial intelligence, precision medicine, patient stratification, endometrial-factor infertility, gene expression signature',\n",
       "  'Introduction': 'A key factor for reproductive success in assisted reproduction treatments is the status of the maternal endometrium during embryo implantation and fetal development. The endometrial cycle is reflected by the cyclic structural and functional changes of the endometrium across the menstrual cycle, particularly during the mid-secretory phase, to prepare for the‘‘window of implantation’’ (WOI) . After successful embryo implantation, the decidua (specialized layer of the endometrium) actively encapsulates the trophectoderm to support placentation and provide adequate vascularization for optimal fetal growth . Thus, approaches that predict and prevent endometrial-factor infertility could substantially improve assisted reproduction treatment outcomes by supporting the establishment and maintenance of pregnancy . There is a lack of consensus on the minimum number of implantation failures with good-quality embryos derived from ovum donation or with confirmed euploid karyotypes required to diagnose recurrent implantation failure (RIF) . The disparities in clinical classification and heterogeneous etiology of RIF confound its diagnosis and leave the exact prevalence unclear . As clinical symptoms cannot be prevented and do not provide sufficient criteria to stratify patients, patients might be misclassi fied and receive inefficient treatment. Indeed, most studies using clinical criteria for RIF identified no or few differentially expressed genes between samples of affected patients , reflecting the molecular heterogeneity of RIF. To overcome these healthcare challenges, high-throughput molecular technologies are combined with artificial intelligence (AI) algorithms to provide a deeper understanding of infertilityrelated pathogenesis . These molecular insights are then applied in genomics-driven precision medicine approaches to characterize complex diseases, and ultimately, improve patient diagnosis and prognosis . Employing genomics in reproductive medicine has the potential to go beyond treating the clinical manifestations of disease progression , by leveraging the patients’ heterogeneous molecular profiles for precise stratification and revealing new targets for preventive measures . In this regard, current diagnostic tools for endometrialfactor infertility use transcriptomics to date the endometrial biopsy in the mid-secretory phase, exceeding the accuracy and objectivity of traditional endometrial progression staging by Noyes’ histological criteria . However, the clinical utility of synchronizing the endometrium by adjusting the timing of progesterone (P4) administration on the basis of the patient’s transcriptomic endometrial dating profile remains controversial . Independent randomized clinical trials found no improvement in reproductive outcomes , indicating the inefficacy of this clinical intervention and highlighting the need for alternative treatments and/or a refined target patient population that bene fits from this strategy. Combining these precision medicine approaches with transcriptomics and AI algorithms; our group recently proposed a novel molecular taxonomy of RIF and a new clinical algorithm for endometrial-factor infertility that considered 2 molecular causes of implantation failure: a WOI displacement that indicates a dysregulated endometrial timing; and a WOI disruption that indicates an endometrial alteration independent of endometrial timing . Transcriptomic analysis revealed these 2 WOI phenotypes were presented exclusively or simultaneously and patients with RIF had significantly more WOI disruptions but a similar frequency of WOI displacements, compared with controls . Thesefindings suggested that WOI disruptions had stronger associations with implantation failure than WOI displacements and shifted the paradigm for the study and treatment of endometrialfactor infertility . In addition, this work by SebastianLeon et al. highlighted the importance of removing the transcriptomic variation due to cyclical endometrial tissue changes (ie, the effect of endometrial timing) that, if not corrected, could mask the significant disruptions in endometrial gene expression. However, given this hypothesis-driven research was conducted retrospectively and in silico with publicly available data, the reproductive outcomes of the cycle after the biopsy collection were not available . This clinical follow-up is required to discover biomarkers for WOI disruption in patients with a higher risk of endometrial failure. On the basis of these genomics-driven precision medicine strategies, this proof-of-concept study aimed to discover a biomarker panel that could identify endometrial disruption, independent of endometrial timing, to predict, and ultimately prevent, poor reproductive outcomes in in vitro fertilization (IVF) patients.',\n",
       "  'Methods': 'Ethics statement This study was approved by the Ethics Committee of the Instituto Valenciano de Infertilidad, Valencia, Spain (1810-FIVI-066-PD/1810-FIVI-048-PD). Written informed consent was obtained from all recruited patients (n1⁄4 281).\\nParticipants, collection of endometrial biopsies, and reproductive outcomes This prospective, multicentric study was conducted between July 2018 and July 2021 at 5 private fertility clinics in Spain. Participants (n1⁄4 281) were scheduled for endometrial evaluation due to medical indications orR2 implantation failures of idiopathic origin. All patients met the following inclusion criteria: 18- 50 years old, with a body mass index (BMI) of 19- 30 kg/m 2, undergoing hormone replacement therapy for single embryo transfer (SET), presenting an endometrial thickness >6.5 mm with trilaminar structure in proliferative phase before P4 administration, and serum levels of estradiol >100 pg/mL and P4<1 ng/mL on the 10th day of estrogen treatment . Endometrial biopsies were collected after an average of 114.5/C6 7.2 h of P4 (duration ranged from 82 to 172 h), and processed as we previously described . Baseline patient characteristics including age, BMI, the number of IVF attempts, the quality of the embryo transferred (defined in theSupplemental Methods, available online), and reproductive outcomes (ie, pregnancy, live birth, and biochemical and clinical miscarriage) for each attempt, were obtained from internal medical records. Information about IVF attempts in other fertility clinics was included if recorded (only in 47.7% of patients).\\nStudy design First, a custom RNA-seq gene panel (the endometrial failure risk [EFR] panel) comprising 404 biomarkers of endometrial luteal phase timing and/or potential disruption was designed and employed to determine the risk of endometrial failure among 217 IVF patients (Supplemental Fig. 1, step 1 available online). The endometrial timing-corrected expression of the EFR panel was used to study the differential expression associated with endometrial disruption together with clinical characteristics of IVF patients to stratify the population as having poor or good endometrial prognoses, according to a semi-supervised learning process (Supplemental Fig. 1, step 2). The distinct transcriptomic patterns of each prognosis group were associated with the differences in reproductive outcomes of SET after biopsy collection, and the differentially expressed gene signature was proposed as a biomarker of endometrial disruption (Supplemental Fig. 1, step 3). Finally, the gene expression signature was used to supervise machine learning algorithms to evaluate the biomarker capacity of this signature to distinguish endometrial competence. The predictive performance of the models was assessed through the accuracy, sensibility, and specificity over 100 iterations of fivefold strati fied cross-validation ( Supplemental Fig. 1 , step 4).\\nConstructing the EFR gene panel Supplemental Figure 2illustrates the workflow for the EFR panel gene selection. A whole-transcriptome endometrial data set (GSE58144, n1⁄4 115) retrieved from Gene Expression Omnibus was processed to correct the endometrial progression effect, using linear models in limma (version 3.46.0) , and unmask genes related to endometrial disruption . These genes were ranked according to an informativity score (CorrelationAttributeEval) in Weka (version 3.8.2, December 22, 2017) to evaluate their worth by measuring their correlation with endometrial pathology. Gene sets with increasing gene signature size were developed by adding the most informative genes in a stepwise manner, and their predictive performance was analyzed using Support Vector Machine and Random Forest algorithms. After calculating predictive performance with a stratified fivefold cross-validation (default parameters; 80:20; 100 iterations) using the Weka R package , the optimal signature size was determined for each algorithm, as the one with the fewest genes and highest accuracy. Finally, genes we previously prioritized for transcriptomic endometrial dating w e r ea l s oi n c l u d e di nt h e EFR panel to be able to remove the endometrial luteal phase timing variation.\\nEndometrial biopsy processing, sequencing, and transcriptomic preprocessing Total RNA was extracted, and only samples that met our quality criteria were sequenced using the TruSeq Targeted RNA Illumina protocol and our custom EFR panel, in the Illumina NextSeq 550 platform with a paired-end design of 150 cycles and 4 M reads/sample, as previously described . Raw transcriptomic data was processed and normalized using limma, to filter low-quality raw counts (Q< 30) and low-read samples (<2M ). Possible batch effects were detected using principal components analysis and corrected with linear models in limma. Finally, endometrial luteal phase timing effects were detected using our transcriptomic endometrial dating (TED) model (see Supplemental Methods) and removed using limma.\\nEFR biomarker signature identification A preliminary classification of patients was based on the patients’ history of reproductive outcomes and transcriptomic patterns corrected for endometrial luteal phase timing. This approach eliminated the gene expression variation related to the cyclic tissue changes that mask endometrial disruption and was used before applying semi-supervised machine learning on the basis of self-labeled techniques (see Supplemental Methods). Once, the genomic-driven disease taxonomy was established, the differentially expressed genes obtained between the final good and poor prognosis groups were proposed as the EFR biomarker signature. The limma R package was used to identify the statistically signi ficant differentially expressed genes (false discovery rate [FDR]-adjustedP value <.05).\\nFunctional characterization of the EFR signature The biomarker signature genes were annotated for functional interpretation databases by consulting Kyoto Encyclopaedia of Genes and Genomes human pathways (release 99.0, August 1, 2021) and Gene Ontology (version July 2, 2021; employing only experimental annotations) . Functional terms were manually summarized into categories according to their description in the database and presented as a bar plot using ggplot2 .\\nEFR signature capability as a biomarker of endometrial disruption To estimate the prediction capability of the EFR signature to distinguish poor and good endometrial prognosis profiles, we implemented predictive models on the basis of the Support Vector Machine algorithm. Using a stratified fivefold crossvalidation process (repeated 100 times), the accuracy, sensitivity, and specificity were estimated. The range of values obtained across the 100 iterations was presented as a boxplot using ggplot2.\\nStatistical analysis Rates for reproductive outcomes (ie, pregnancy rate [PR], cumulative pregnancy rate [CPR], live birth rate [LBR], biochemical miscarriage rate [BMR], and clinical miscarriage rate [CMR]) were calculated as de fined in the Supplemental Methods. Descriptive statistics were used for transcriptomic and clinical characteristics of patients to compare prognosis or WOI displacement groups. Continuous variables were presented as an overall mean /C6 standard deviation whereas discrete variables were presented as counts and percentages. Groups were compared using the Wilcoxon rank-sum test for continuous variables and Fisher’s exact test for discrete variables. All statistical analyses were conducted in R. All graphical results were generated with ggplot2.\\nGene expression validation by quantitative realtime polymerase chain reaction Quantitative real-time polymerase chain reaction (RT-qPCR) was performed to validate the expression of the top 4 differentially expressed genes between patients with poor (n1⁄4 10) and good (n1⁄4 10) endometrial prognoses. RNA (200- 220 ng) was reverse transcribed into cDNA using the PrimeScript Reagent Kit (Perfect Real Time, Takara, Japan) on a Thermocycler T3000 (Biometra, Ireland). Quantitative real-time polymerase chain reaction was performed on a StepOnePlus Real-Time PCR System (Applied Biosystems, USA) using Power-Up SYBR Green (Thermo Fisher Scientific, USA) and standard conditions.Supplemental Table 1(available online) lists the specific primers (Invitrogen, Thermo Fisher Scientific) used. Universal human RNA (Agilent, Spain) and water were respectively included as positive and negative controls. Each sample was analyzed in duplicate and relative gene expression (normalized to beta actin) was calculated using the DDCt method .',\n",
       "  'Results': 'Preliminary sequencing with the 404-gene EFR panel The EFR panel evaluated 404 genes: 139 that predict endometrial disruption and 310 that serve as biomarkers of endometrial dating or timing. Notably, 45 genes overlapped in these 2 endometrial factors. Of the 281 endometrial samples collected, we excluded 56 samples for having low-quality or insufficient RNA; 6 samples containing a low number of sequencing reads (Supplemental Fig. 3A); and 2 samples that had low gene expression after normalization ( Supplemental Fig. 3B ). Thus, 217 patients were considered in the downstream analysis. After correcting batch effects from sequencing runs (Supplemental Fig. 3C) and removing the transcriptomic variation due to endometrial cyclic changes (endometrial luteal phase timing;Supplemental Fig. 3D), there were no other confounding technical or clinical factors (Supplemental Fig. 4).\\nA genomic-driven endometrial taxonomy for patient stratification on the basis of a novel endometrial disruption independent of endometrial timing Forty-four patients were initially categorized as acute RIF on the basis of the inability to carry a pregnancy to term after 3- 9 SETs. Alternatively, 40 patients were categorized as acute Fertile because they reached full-term pregnancy at thefirst SET after biopsy. Notably, after this clinical acute criteria, 133 (61.3%) patients remained unclassified for having insufficient attempts (Supplemental Fig. 5A), and no transcriptomic differences were found between the acute RIF and acute Fertile groups (the lowest FDR was 0.3027). Incorporation of transcriptomic data with this initial acute clinical classification allowed us to stratify patients with poor (n1⁄4 137, 63.1%) or good (n1⁄4 49, 22.6%) endometrial prognoses; leaving only 31 samples (14.3%) unclassified because of the lack of recognition by the algorithms (Supplemental Fig. 5B).\\nEFR biomarker signature identification and functional characterization A total of 122 genes were identified as significantly different (30.2% of all panel genes, FDR< 0.05) between patients with good and poor endometrial prognoses, including 63 downregulated and 59 upregulated in patients with poor endometrial prognosis (Table 1 and Fig. 1A), and proposed as the EFR biomarker signature. Transcriptomically, endometrial biopsies clustered by prognosis (Fig. 1B), and the signature genes had functions that were mainly related to regulation (17%); metabolism (8.4%); the immune system and inflammation (7.8%); protein processes (7.3%), signaling (6.9%), transport (6.1%), and the cell cycle, proliferation, and differentiation (5.3%) (Fig. 1C andSupplemental Table 2). The poor prognosis profile was distinguished by a downregulation of 27/39 (69.2%) functional groups. We highlight the functions with higher proportions of downregulated genes as steroidogenesis and steroid signaling (77.8%), hormone-related (80.0%), the nervous system (81.3%), gametogenesis and reproduction (85.7%), and oxidative stress and response to O 2 (88.9%), as well as cell action potential and polarization (100%) (Fig. 1D). Few functional groups, including cell migration, extracellular matrix, and glucose homeostasis contained a larger proportion of upregulated genes (60.9%, 62.5%, and 62.5% upregulated genes, respectively;Fig. 1D).\\nDifferences in clinical reproductive outcomes authenticated the EFR signature’s clinical relevance There were no significant differences in the age or BMI of patients predicted to have good/poor endometrial prognosis (all P>.05). Women predicted to have poor endometrial prognosis previously underwent significantly more SETs than those with good endometrial prognosis (2.8 vs. 1.6;P<.0001), however, there were no significant differences in embryo quality (P1⁄4.4679) between groups (Supplemental Table 3). The lack of differences in the proportion of out-of-phase endometria (57% in poor vs. 47% in good,P1⁄4.25) between good and poor prognosis groups (Supplemental Table 3) distinguished this signature from established endometrial luteal phase timing phenotypes (displaced or on-time). Therefore, the significant differences (P<.0001) between the reproductive outcomes of women predicted to have poor and good endometrial prognoses validated the clinical relevance of our transcriptomic signature. Specifically, the poor endometrial prognosis was associated with a 34.5% decline in PR (P1⁄43.8E-05), a 52% reduction in LBR (P1⁄45E-10), a 19.6% rise in CMR (P1⁄4.0066), and 20.4% increase in BMR (P1⁄4.0023) (Fig. 2A). Furthermore, patients with a poor endometrial prognosis only achieved a CPR of 42.3% after the ninth SET, whereas patients with a good prognosis had an 83.7% CPR at thefifth attempt (Fig. 2B). The relative risk of implantation failure or miscarriage in women predicted to have poor endometrial prognosis was respectively 2.7 and 11.5 compared with women with good endometrial prognosis. However, when failures in implantation or ongoing pregnancy were considered together, the relative risk of endometrial failure (not achieving a live birth in the next SET) for patients with poor endometrial prognosis was 3.3.\\nEFR signature biomarker’s predictive capability The 186 samples labeled with poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognoses were used to evaluate the signature’s predictive performance. The 122 differentially expressed genes between poor/good endometrial prognoses proposed as the EFR biomarker signature were employed to train the model. Cross-validation yielded a median accuracy of 0.92 (min1⁄4 0.88, max 1⁄4 0.94), sensitivity of 0.96 (min1⁄4 0.91, max 1⁄4 0.98), and specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88) (Fig. 2C). Notably, 163 of 186 (87.6%) samples were classified correctly in almost 90% of all cross-validation models, whereas 151 of 186 (81.2%) samples were well classified correctly in 100% of the iterations. Twelve and eleven samples were misclassified consistently, in <50% or 50%- 90% of models, respectively (Fig. 2D).\\nClinical algorithm results after evaluating both WOI factors Next, we grouped samples according to both potential causes of implantation failure, endometrial luteal phase timing (displaced or on-time) (Fig. 3A), and endometrial disruption (good or poor prognosis) (Fig. 2A) and compared the patients’ reproductive outcomes. There were no significant differences between the patients with displaced and on-time WOI in terms of BMI (P1⁄4.8740), the average number of SETs (P1⁄4.0879), and embryo quality (P1⁄41). However, patients with a displaced WOI were signi ficantly older (40.4 vs. 38.3 years old; P1⁄4.0044) (Supplemental Table 4). Once patients were stratified by prognosis (endometrial disruption), the proportion of patients with poor prognosis was similar (P1⁄4.2458) between displaced and on-time groups (Fig. 3B).\\nExperimental validation of genes related to the EFR signature The expression of the top 4 differentially expressed genes in the poor prognosis group was validated by RT-qPCR. Progestagen-associated endometrial protein and G-coupled protein receptor 110, which were found upregulated by RNA-seq, were significantly overexpressed in the poor prognosis group, compared with the good prognosis group (fold change [FC]1⁄4 6.157 [P1⁄4.0288] and FC1⁄4 11.803 [P1⁄4.0350], respectively; Supplemental Figs. 6Aand B). Meanwhile, the expression of fibrinogen beta chain and solute carrier 16 member 6, which were found downregulated by RNA-seq, did not reach a statistically significant difference (FC1⁄4 1.582 [P1⁄4.2475] and FC1⁄4 2.311 [P1⁄4.4376], respectively;Supplemental Figs. 6Cand D).',\n",
       "  'Discussion': 'Advances in the diagnosis and treatment of RIF have been limited by its elusive etiology, the discrepancies in clinical classification (mainly due to the lack of consensus over the minimum number of pregnancy attempts) , and the heterogeneous molecular profiles of affected patients . On the basis of current clinical practice, patients with RIF may be misclassified and/or receive inef ficient treatment . Thus, the aim of this study was to propose a novel gene expression signature that could identify endometrial disruptions independent of endometrial luteal phase timing. Using a semi-supervised learning AI strategy together with the endometrial gene expression obtained from our custom EFR panel, we stratified patients according to poor or good endometrial prognoses and demonstrated how this signature reliably predicted gene expression patterns that corresponded with significant differences in PR, LBR, CMR, and BMR. When considering both WOI factors (disruption and timing) in our study cohort, poor endometrial prognoses (disruption) affected 73.7% of patients, whereas WOI displacements only affected 54.3%; 41.9% of patients presented a displaced WOI and were predicted to have a poor endometrial prognosis, whereas only 12.4% of patients presented with a displaced WOI and were predicted to have a good endometrial prognosis. These results offer a possible explanation as to why other studies reported there was no clinical benefito f personalizing embryo transfers on the basis of transcriptomic dating of the endometrium . According to our clinical algorithm, only patients with a displaced WOI and a good endometrial prognosis (12.4% of our cohort) would be better off with this approach. Patients with poor endometrial prognosis (73.7% of our cohort) would have suboptimal reproductive outcomes even though their WOI was on-time. This highlights an unmet healthcare need for IVF patients with endometrial disruptions and underscores the importance of additional prospective studies to further characterize patients on the basis of these 2 molecular causes for trying tailored treatments. Contemplating endometrial luteal phase timing together with a prevalent disruption or poor endometrial prognosis shifts the paradigm in studies about the endometrium. Our innovative methodology based on transcriptomics and AI models led us to develop new endometrial biomarker taxonomies related to endometrial luteal phase timing and/or endometrial disruption . There were 30 genes overlapping between the previously published TED signature and the EFR signature presented herein, emphasizing their correlation despite the lower FC of biomarkers for endometrial disruption versus endometrial timing . AI algorithms offer powerful approaches for patient stratification, by combining high-throughput molecular data and clinical variables . Clinical translation of these approaches will require careful supervision of how the machine learning is initially trained . However, a semi-supervised learning methodology implementing accurate clinical diagnostic criteria and patient-specific genomic data to establish relevant molecular taxonomies will be indispensable for advancing precision medicine , especially within the context of endometrial-factor infertility. Predictive biomarkers can stratify patients with endometrial failure even if the number of their attempts was insufficient to meet clinical RIF classification criteria . To our knowledge, this is thefirst study to apply semi-supervised learning to the endometrium, conferring an advantage over previous models that were only trained with clinical classifications and did not leverage the underlying molecular heterogeneity of the condition . The lack of gene expression differences between RIF and control patients with our initial clinical classification reinforced findings from Macklon’s group and the molecular heterogeneity among patients with RIF . The EFR signature comprised 122 genes, including 44 genes newly associated with an endometrial gene expression signature and a single common gene with Koot’s pioneer endometrial timing-independent signature for endometrial pathology . Here, we only considered biopsies collected in the endometrial mid-secretory phase, when patients had received P4 for 82- 172 h. Although this signature is independent of endometrial luteal phase timing, further studies are required to confirm if the signature is detectable in other phases of the endometrial cycle. Progestagen-associated endometrial protein is well characterized in endometrial tissue and was the most upregulated gene (FC1⁄4 3.18) in the poor endometrial prognosis pro file. However, the levels we observed were 10-fold lower compared with previous studies where changes in endometrial timing were considered . Alternatively,fibrinogen beta chain, related to the plasma membrane and vesicle transport by secretory granules, was the most downregulated gene. This gene accumulated polymorphisms that hinder the production offibrinogen and its associated anti-inflammatory activity and was recently related to the dysregulated pathways in RIF . FOXP1, the most predictive gene, is a transcription factor mediated by estrogen that has established functions in regulation, leukocyte differentiation, and response to lipids and chemokines. This gene is associated with breast endometrial cancer , and endometriosis-related fibrosis , but was never included in an endometrial gene signature until now. With FOXP1 and the FOX transcription factor family being associated with the regulation of the menstrual cycle and implantation , this evidence reinforces our prior study and affirms that alterations in endometrial regulation are an underlying cause of endometrial failure. Our clinical algorithm considers the 2 WOI factors, endometrial timing and disruption, to stratify IVF patients on the basis of 4 possible phenotypes: displaced/on-time and poor/ good prognosis. This model is promising for preventive and precision medicine in endometrial-factor infertility, as it combines the robust prediction parameters of the TED model and the EFR biomarker signature’s predictive capability reported herein. In contrast to other transcriptomic-based endometrial dating tools , our bioinformatic methodology removes the molecular variation associated with the cyclic menstrual changes, providing deeper insights for identifying a new kind of disruption independent of endometrial luteal phase timing by the EFR signature. Furthermore, the EFR biomarker signature was developed on the basis of reproductive outcomes and effectively classifies patients predicted to have positive or negative reproductive outcomes in the first SET after biopsy collection. All these facts, point out to a more potential clinical utility than existing tools that only predict the WOI without showing significant differences in reproductive outcomes as have been reported and we have shown in ourfindings. Although our strategy appears promising, the classification presented herein needs further re finement before clinical implementation. Larger sample sizes and validation in independent samples are required to ensure model generalization . Healthcare professionals will need to be adequately trained to ensure technical reproducibility. Longitudinal assessments including several menstrual cycles in the same patient are recommended to confirm inter-cycle reproducibility and to gain a comprehensive understanding of the reliability of the EFR panel. Additional studies should assess the relationship between a poor endometrial prognosis and the presence of benign disorders that potentially compromise endometrial function and/or embryo implantation . Inspired by the MammaPrint risk signature for breast cancer , the EFR signature could become a nextgeneration tool used to classify patients on the basis of endometrial prognosis, to predict reproductive outcomes before undergoing IVF, potentially experiencing RIF and/or losing valuable embryos. Although treatments for patients with poor endometrial prognosis remain to be determined, clinically identifying these patients is the first step toward improving their care.',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1016/j.fertnstert.2024.03.015',\n",
       "  'Journal': 'Fertil Steril'},\n",
       " {'PaperTitle': 'Predicting risk of endometrial failure: a biomarker signature that identifies a novel disruption independent of endometrial timing in patients undergoing hormonal replacement cycles',\n",
       "  'Publication': '2024',\n",
       "  'Authors': 'Patricia Diaz-Gimeno, Ph.D., Patricia Sebastian-Leon, Ph.D., Katharina Spath, Ph.D., Diana Marti-Garcia, M.Sc., Josefa Maria Sanchez-Reyes, Ph.D., Maria del Carmen Vidal, Ph.D., Almudena Devesa-Peiro, Ph.D., Immaculada Sanchez-Ribas, Ph.D., Asunta Martinez-Martinez, M.Sc., Nuria Pellicer, M.D., Ph.D., Dagan Wells, Ph.D., and Antonio Pellicer, M.D., Ph.D.',\n",
       "  'Email': 'patricia.diaz@ivirma.com or patricia_diaz@iislafe.es',\n",
       "  'Abstract': 'Objective: To propose a new gene expression signature that identifies endometrial disruptions independent of endometrial luteal phase timing and predicts if patients are at risk of endometrial failure. Design: Multicentric, prospective study. Setting: Reproductive medicine research department in a public hospital affiliated with private fertility clinics and a reproductive genetics laboratory. Patients: Caucasian women (n1⁄4 281; 39.4/C6 4.8 years old with a body mass index of 22.9/C6 3.5 kg/m 2) undergoing hormone replacement therapy between July 2018 and July 2021. Endometrial samples from 217 patients met RNA quality criteria for signature discovery and analysis. Intervention(s): Endometrial biopsies collected in the mid-secretory phase. Main Outcome Measure(s):Endometrial luteal phase timing-corrected expression of 404 genes and reproductive outcomes of thefirst single embryo transfer (SET) after biopsy collection to identify prognostic biomarkers of endometrial failure. Results: Removal of endometrial timing variation from gene expression data allowed patients to be stratified into poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognosis groups on the basis of their clinical and transcriptomic profiles. Significant differences were found between endometrial prognosis groups in terms of reproductive rates: pregnancy (44.6% vs. 79.6%), live birth (25.6% vs. 77.6%), clinical miscarriage (22.2% vs. 2.6%), and biochemical miscarriage (20.4% vs. 0%). The relative risk of endometrial failure for patients predicted as a poor endometrial prognosis was 3.3 times higher than those with a good prognosis. The differences in gene expression between both profiles were proposed as a biomarker, coined the endometrial failure risk (EFR) signature. Poor prognosis profiles were characterized by 59 upregulated and 63 downregulated genes mainly involved in regulation (17.0%), metabolism (8.4%), immune response, and inflammation (7.8%). This EFR signature had a median accuracy of 0.92 (min1⁄40.88, max1⁄40.94), median sensitivity of 0.96 (min1⁄40.91, max 1⁄4 0.98), and median specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88), positioning itself as a promising biomarker for endometrial evaluation.',\n",
       "  'Introduction': 'A key factor for reproductive success in assisted reproduction treatments is the status of the maternal endometrium during embryo implantation and fetal development. The endometrial cycle is reflected by the cyclic structural and functional changes of the endometrium across the menstrual cycle, particularly during the mid-secretory phase, to prepare for the‘‘window of implantation’’ (WOI) . After successful embryo implantation, the decidua (specialized layer of the endometrium) actively encapsulates the trophectoderm to support placentation and provide adequate vascularization for optimal fetal growth . Thus, approaches that predict and prevent endometrial-factor infertility could substantially improve assisted reproduction treatment outcomes by supporting the establishment and maintenance of pregnancy . There is a lack of consensus on the minimum number of implantation failures with good-quality embryos derived from ovum donation or with confirmed euploid karyotypes required to diagnose recurrent implantation failure (RIF) . The disparities in clinical classification and heterogeneous etiology of RIF confound its diagnosis and leave the exact prevalence unclear . As clinical symptoms cannot be prevented and do not provide sufficient criteria to stratify patients, patients might be misclassi fied and receive inefficient treatment. Indeed, most studies using clinical criteria for RIF identified no or few differentially expressed genes between samples of affected patients , reflecting the molecular heterogeneity of RIF. To overcome these healthcare challenges, high-throughput molecular technologies are combined with artificial intelligence (AI) algorithms to provide a deeper understanding of infertilityrelated pathogenesis . These molecular insights are then applied in genomics-driven precision medicine approaches to characterize complex diseases, and ultimately, improve patient diagnosis and prognosis . Employing genomics in reproductive medicine has the potential to go beyond treating the clinical manifestations of disease progression , by leveraging the patients’ heterogeneous molecular profiles for precise stratification and revealing new targets for preventive measures . In this regard, current diagnostic tools for endometrialfactor infertility use transcriptomics to date the endometrial biopsy in the mid-secretory phase, exceeding the accuracy and objectivity of traditional endometrial progression staging by Noyes’ histological criteria . However, the clinical utility of synchronizing the endometrium by adjusting the timing of progesterone (P4) administration on the basis of the patient’s transcriptomic endometrial dating profile remains controversial . Independent randomized clinical trials found no improvement in reproductive outcomes , indicating the inefficacy of this clinical intervention and highlighting the need for alternative treatments and/or a refined target patient population that bene fits from this strategy. Combining these precision medicine approaches with transcriptomics and AI algorithms; our group recently proposed a novel molecular taxonomy of RIF and a new clinical algorithm for endometrial-factor infertility that considered 2 molecular causes of implantation failure: a WOI displacement that indicates a dysregulated endometrial timing; and a WOI disruption that indicates an endometrial alteration independent of endometrial timing . Transcriptomic analysis revealed these 2 WOI phenotypes were presented exclusively or simultaneously and patients with RIF had significantly more WOI disruptions but a similar frequency of WOI displacements, compared with controls . Thesefindings suggested that WOI disruptions had stronger associations with implantation failure than WOI displacements and shifted the paradigm for the study and treatment of endometrialfactor infertility . In addition, this work by SebastianLeon et al.  highlighted the importance of removing the transcriptomic variation due to cyclical endometrial tissue changes (ie, the effect of endometrial timing) that, if not corrected, could mask the significant disruptions in endometrial gene expression. However, given this hypothesis-driven research was conducted retrospectively and in silico with publicly available data, the reproductive outcomes of the cycle after the biopsy collection were not available . This clinical follow-up is required to discover biomarkers for WOI disruption in patients with a higher risk of endometrial failure. On the basis of these genomics-driven precision medicine strategies, this proof-of-concept study aimed to discover a biomarker panel that could identify endometrial disruption, independent of endometrial timing, to predict, and ultimately prevent, poor reproductive outcomes in in vitro fertilization (IVF) patients.',\n",
       "  'Methods': 'MATERIAL AND METHODS Ethics statement This study was approved by the Ethics Committee of the Instituto Valenciano de Infertilidad, Valencia, Spain (1810-FIVI-066-PD/1810-FIVI-048-PD). Written informed consent was obtained from all recruited patients (n1⁄4 281). Participants, collection of endometrial biopsies, and reproductive outcomes This prospective, multicentric study was conducted between July 2018 and July 2021 at 5 private fertility clinics in Spain. Participants (n1⁄4 281) were scheduled for endometrial evaluation due to medical indications orR2 implantation failures of idiopathic origin. All patients met the following inclusion criteria: 18- 50 years old, with a body mass index (BMI) of 19- 30 kg/m 2, undergoing hormone replacement therapy for single embryo transfer (SET), presenting an endometrial thickness >6.5 mm with trilaminar structure in proliferative phase before P4 administration, and serum levels of estradiol >100 pg/mL and P4<1 ng/mL on the 10th day of estrogen treatment . Endometrial biopsies were collected after an average of 114.5/C6 7.2 h of P4 (duration ranged from 82 to 172 h), and processed as we previously described . Baseline patient characteristics including age, BMI, the number of IVF attempts, the quality of the embryo transferred (defined in theSupplemental Methods, available online), and reproductive outcomes (ie, pregnancy, live birth, and biochemical and clinical miscarriage) for each attempt, were obtained from internal medical records. Information about IVF attempts in other fertility clinics was included if recorded (only in 47.7% of patients). Study design First, a custom RNA-seq gene panel (the endometrial failure risk [EFR] panel) comprising 404 biomarkers of endometrial luteal phase timing and/or potential disruption was designed and employed to determine the risk of endometrial failure among 217 IVF patients (Supplemental Fig. 1, step 1 available online). The endometrial timing-corrected expression of the EFR panel was used to study the differential expression associated with endometrial disruption together with clinical characteristics of IVF patients to stratify the population as having poor or good endometrial prognoses, according to a semi-supervised learning process (Supplemental Fig. 1, step 2). The distinct transcriptomic patterns of each prognosis group were associated with the differences in reproductive outcomes of SET after biopsy collection, and the differentially expressed gene signature was proposed as a biomarker of endometrial disruption (Supplemental Fig. 1, step 3). Finally, the gene expression signature was used to supervise machine learning algorithms to evaluate the biomarker capacity of this signature to distinguish endometrial competence. The predictive performance of the models was assessed through the accuracy, sensibility, and specificity over 100 iterations of fivefold strati fied cross-validation ( Supplemental Fig. 1 , step 4). Constructing the EFR gene panel Supplemental Figure 2illustrates the workflow for the EFR panel gene selection. A whole-transcriptome endometrial data set (GSE58144, n1⁄4 115) retrieved from Gene Expression Omnibus  was processed to correct the endometrial progression effect, using linear models in limma (version 3.46.0) , and unmask genes related to endometrial disruption . These genes were ranked according to an informativity score (CorrelationAttributeEval) in Weka (version 3.8.2, December 22, 2017)  to evaluate their worth by measuring their correlation with endometrial pathology. Gene sets with increasing gene signature size were developed by adding the most informative genes in a stepwise manner, and their predictive performance was analyzed using Support Vector Machine  and Random Forest  algorithms. After calculating predictive performance with a stratified fivefold cross-validation (default parameters; 80:20; 100 iterations) using the Weka R package , the optimal signature size was determined for each algorithm, as the one with the fewest genes and highest accuracy. Finally, genes we previously prioritized for transcriptomic endometrial dating w e r ea l s oi n c l u d e di nt h e EFR panel to be able to remove the endometrial luteal phase timing variation. Endometrial biopsy processing, sequencing, and transcriptomic preprocessing Total RNA was extracted, and only samples that met our quality criteria were sequenced using the TruSeq Targeted RNA Illumina protocol and our custom EFR panel, in the Illumina NextSeq 550 platform with a paired-end design of 150 cycles and 4 M reads/sample, as previously described . Raw transcriptomic data was processed and normalized using limma, to filter low-quality raw counts (Q< 30) and low-read samples (<2M ). Possible batch effects were detected using principal components analysis and corrected with linear models in limma. Finally, endometrial luteal phase timing effects were detected using our transcriptomic endometrial dating (TED) model  (see Supplemental Methods) and removed using limma. EFR biomarker signature identification A preliminary classification of patients was based on the patients’ history of reproductive outcomes and transcriptomic patterns corrected for endometrial luteal phase timing. This approach eliminated the gene expression variation related to the cyclic tissue changes that mask endometrial disruption  and was used before applying semi-supervised machine learning on the basis of self-labeled techniques (see Supplemental Methods). Once, the genomic-driven disease taxonomy was established, the differentially expressed genes obtained between the final good and poor prognosis groups were proposed as the EFR biomarker signature. The limma R package was used to identify the statistically signi ficant differentially expressed genes (false discovery rate [FDR]-adjustedP value <.05). Functional characterization of the EFR signature The biomarker signature genes were annotated for functional interpretation databases by consulting Kyoto Encyclopaedia of Genes and Genomes human pathways  (release 99.0, August 1, 2021) and Gene Ontology (version July 2, 2021; employing only experimental annotations) . Functional terms were manually summarized into categories according to their description in the database and presented as a bar plot using ggplot2 . EFR signature capability as a biomarker of endometrial disruption To estimate the prediction capability of the EFR signature to distinguish poor and good endometrial prognosis profiles, we implemented predictive models on the basis of the Support Vector Machine algorithm. Using a stratified fivefold crossvalidation process (repeated 100 times), the accuracy, sensitivity, and specificity were estimated. The range of values obtained across the 100 iterations was presented as a boxplot using ggplot2. Statistical analysis Rates for reproductive outcomes (ie, pregnancy rate [PR], cumulative pregnancy rate [CPR], live birth rate [LBR], biochemical miscarriage rate [BMR], and clinical miscarriage rate [CMR]) were calculated as de fined in the Supplemental Methods. Descriptive statistics were used for transcriptomic and clinical characteristics of patients to compare prognosis or WOI displacement groups. Continuous variables were presented as an overall mean /C6 standard deviation whereas discrete variables were presented as counts and percentages. Groups were compared using the Wilcoxon rank-sum test for continuous variables and Fisher’s exact test for discrete variables. All statistical analyses were conducted in R. All graphical results were generated with ggplot2. Gene expression validation by quantitative realtime polymerase chain reaction Quantitative real-time polymerase chain reaction (RT-qPCR) was performed to validate the expression of the top 4 differentially expressed genes between patients with poor (n1⁄4 10) and good (n1⁄4 10) endometrial prognoses. RNA (200- 220 ng) was reverse transcribed into cDNA using the PrimeScript Reagent Kit (Perfect Real Time, Takara, Japan) on a Thermocycler T3000 (Biometra, Ireland). Quantitative real-time polymerase chain reaction was performed on a StepOnePlus Real-Time PCR System (Applied Biosystems, USA) using Power-Up SYBR Green (Thermo Fisher Scientific, USA) and standard conditions.Supplemental Table 1(available online) lists the specific primers (Invitrogen, Thermo Fisher Scientific) used. Universal human RNA (Agilent, Spain) and water were respectively included as positive and negative controls. Each sample was analyzed in duplicate and relative gene expression (normalized to beta actin) was calculated using the DDCt method .',\n",
       "  'Results': 'RESULTS Preliminary sequencing with the 404-gene EFR panel The EFR panel evaluated 404 genes: 139 that predict endometrial disruption and 310 that serve as biomarkers of endometrial dating or timing. Notably, 45 genes overlapped in these 2 endometrial factors. Of the 281 endometrial samples collected, we excluded 56 samples for having low-quality or insufficient RNA; 6 samples containing a low number of sequencing reads (Supplemental Fig. 3A); and 2 samples that had low gene expression after normalization ( Supplemental Fig. 3B ). Thus, 217 patients were considered in the downstream analysis. After correcting batch effects from sequencing runs (Supplemental Fig. 3C) and removing the transcriptomic variation due to endometrial cyclic changes (endometrial luteal phase timing;Supplemental Fig. 3D), there were no other confounding technical or clinical factors (Supplemental Fig. 4). A genomic-driven endometrial taxonomy for patient stratification on the basis of a novel endometrial disruption independent of endometrial timing Forty-four patients were initially categorized as acute RIF on the basis of the inability to carry a pregnancy to term after 3- 9 SETs. Alternatively, 40 patients were categorized as acute Fertile because they reached full-term pregnancy at thefirst SET after biopsy. Notably, after this clinical acute criteria, 133 (61.3%) patients remained unclassified for having insufficient attempts (Supplemental Fig. 5A), and no transcriptomic differences were found between the acute RIF and acute Fertile groups (the lowest FDR was 0.3027). Incorporation of transcriptomic data with this initial acute clinical classification allowed us to stratify patients with poor (n1⁄4 137, 63.1%) or good (n1⁄4 49, 22.6%) endometrial prognoses; leaving only 31 samples (14.3%) unclassified because of the lack of recognition by the algorithms (Supplemental Fig. 5B). EFR biomarker signature identification and functional characterization A total of 122 genes were identified as significantly different (30.2% of all panel genes, FDR< 0.05) between patients with good and poor endometrial prognoses, including 63 downregulated and 59 upregulated in patients with poor endometrial prognosis (Table 1 and Fig. 1A), and proposed as the EFR biomarker signature. Transcriptomically, endometrial biopsies clustered by prognosis (Fig. 1B), and the signature genes had functions that were mainly related to regulation (17%); metabolism (8.4%); the immune system and inflammation (7.8%); protein processes (7.3%), signaling (6.9%), transport (6.1%), and the cell cycle, proliferation, and differentiation (5.3%) (Fig. 1C andSupplemental Table 2). The poor prognosis profile was distinguished by a downregulation of 27/39 (69.2%) functional groups. We highlight the functions with higher proportions of downregulated genes as steroidogenesis and steroid signaling (77.8%), hormone-related (80.0%), the nervous system (81.3%), gametogenesis and reproduction (85.7%), and oxidative stress and response to O 2 (88.9%), as well as cell action potential and polarization (100%) (Fig. 1D). Few functional groups, including cell migration, extracellular matrix, and glucose homeostasis contained a larger proportion of upregulated genes (60.9%, 62.5%, and 62.5% upregulated genes, respectively;Fig. 1D). Differences in clinical reproductive outcomes authenticated the EFR signature’s clinical relevance There were no significant differences in the age or BMI of patients predicted to have good/poor endometrial prognosis (all P>.05). Women predicted to have poor endometrial prognosis previously underwent significantly more SETs than those with good endometrial prognosis (2.8 vs. 1.6;P<.0001), however, there were no significant differences in embryo quality (P1⁄4.4679) between groups (Supplemental Table 3). The lack of differences in the proportion of out-of-phase endometria (57% in poor vs. 47% in good,P1⁄4.25) between good and poor prognosis groups (Supplemental Table 3) distinguished this signature from established endometrial luteal phase timing phenotypes (displaced or on-time). Therefore, the significant differences (P<.0001) between the reproductive outcomes of women predicted to have poor and good endometrial prognoses validated the clinical relevance of our transcriptomic signature. Specifically, the poor endometrial prognosis was associated with a 34.5% decline in PR (P1⁄43.8E-05), a 52% reduction in LBR (P1⁄45E-10), a 19.6% rise in CMR (P1⁄4.0066), and 20.4% increase in BMR (P1⁄4.0023) (Fig. 2A). Furthermore, patients with a poor endometrial prognosis only achieved a CPR of 42.3% after the ninth SET, whereas patients with a good prognosis had an 83.7% CPR at thefifth attempt (Fig. 2B). The relative risk of implantation failure or miscarriage in women predicted to have poor endometrial prognosis was respectively 2.7 and 11.5 compared with women with good endometrial prognosis. However, when failures in implantation or ongoing pregnancy were considered together, the relative risk of endometrial failure (not achieving a live birth in the next SET) for patients with poor endometrial prognosis was 3.3. EFR signature biomarker’s predictive capability The 186 samples labeled with poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognoses were used to evaluate the signature’s predictive performance. The 122 differentially expressed genes between poor/good endometrial prognoses proposed as the EFR biomarker signature were employed to train the model. Cross-validation yielded a median accuracy of 0.92 (min1⁄4 0.88, max 1⁄4 0.94), sensitivity of 0.96 (min1⁄4 0.91, max 1⁄4 0.98), and specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88) (Fig. 2C). Notably, 163 of 186 (87.6%) samples were classified correctly in almost 90% of all cross-validation models, whereas 151 of 186 (81.2%) samples were well classified correctly in 100% of the iterations. Twelve and eleven samples were misclassified consistently, in <50% or 50%- 90% of models, respectively (Fig. 2D). Clinical algorithm results after evaluating both WOI factors Next, we grouped samples according to both potential causes of implantation failure, endometrial luteal phase timing (displaced or on-time) (Fig. 3A), and endometrial disruption (good or poor prognosis) (Fig. 2A) and compared the patients’ reproductive outcomes. There were no significant differences between the patients with displaced and on-time WOI in terms of BMI (P1⁄4.8740), the average number of SETs (P1⁄4.0879), and embryo quality (P1⁄41). However, patients with a displaced WOI were signi ficantly older (40.4 vs. 38.3 years old; P1⁄4.0044) (Supplemental Table 4). Once patients were stratified by prognosis (endometrial disruption), the proportion of patients with poor prognosis was similar (P1⁄4.2458) between displaced and on-time groups (Fig. 3B). Experimental validation of genes related to the EFR signature The expression of the top 4 differentially expressed genes in the poor prognosis group was validated by RT-qPCR. Progestagen-associated endometrial protein and G-coupled protein receptor 110, which were found upregulated by RNA-seq, were significantly overexpressed in the poor prognosis group, compared with the good prognosis group (fold change [FC]1⁄4 6.157 [P1⁄4.0288] and FC1⁄4 11.803 [P1⁄4.0350], respectively; Supplemental Figs. 6Aand B). Meanwhile, the expression of fibrinogen beta chain and solute carrier 16 member 6, which were found downregulated by RNA-seq, did not reach a statistically significant difference (FC1⁄4 1.582 [P1⁄4.2475] and FC1⁄4 2.311 [P1⁄4.4376], respectively;Supplemental Figs. 6Cand D).',\n",
       "  'Discussion': 'Advances in the diagnosis and treatment of RIF have been limited by its elusive etiology, the discrepancies in clinical classification (mainly due to the lack of consensus over the minimum number of pregnancy attempts) , and the heterogeneous molecular profiles of affected patients . On the basis of current clinical practice, patients with RIF may be misclassified and/or receive inef ficient treatment . Thus, the aim of this study was to propose a novel gene expression signature that could identify endometrial disruptions independent of endometrial luteal phase timing. Using a semi-supervised learning AI strategy together with the endometrial gene expression obtained from our custom EFR panel, we stratified patients according to poor or good endometrial prognoses and demonstrated how this signature reliably predicted gene expression patterns that corresponded with significant differences in PR, LBR, CMR, and BMR. When considering both WOI factors (disruption and timing) in our study cohort, poor endometrial prognoses (disruption) affected 73.7% of patients, whereas WOI displacements only affected 54.3%; 41.9% of patients presented a displaced WOI and were predicted to have a poor endometrial prognosis, whereas only 12.4% of patients presented with a displaced WOI and were predicted to have a good endometrial prognosis. These results offer a possible explanation as to why other studies reported there was no clinical benefito f personalizing embryo transfers on the basis of transcriptomic dating of the endometrium . According to our clinical algorithm, only patients with a displaced WOI and a good endometrial prognosis (12.4% of our cohort) would be better off with this approach. Patients with poor endometrial prognosis (73.7% of our cohort) would have suboptimal reproductive outcomes even though their WOI was on-time. This highlights an unmet healthcare need for IVF patients with endometrial disruptions  and underscores the importance of additional prospective studies to further characterize patients on the basis of these 2 molecular causes for trying tailored treatments. Contemplating endometrial luteal phase timing together with a prevalent disruption or poor endometrial prognosis shifts the paradigm in studies about the endometrium. Our innovative methodology based on transcriptomics and AI models led us to develop new endometrial biomarker taxonomies related to endometrial luteal phase timing and/or endometrial disruption . There were 30 genes overlapping between the previously published TED signature  and the EFR signature presented herein, emphasizing their correlation despite the lower FC of biomarkers for endometrial disruption versus endometrial timing . AI algorithms offer powerful approaches for patient stratification, by combining high-throughput molecular data and clinical variables . Clinical translation of these approaches will require careful supervision of how the machine learning is initially trained . However, a semi-supervised learning methodology implementing accurate clinical diagnostic criteria and patient-specific genomic data to establish relevant molecular taxonomies will be indispensable for advancing precision medicine , especially within the context of endometrial-factor infertility. Predictive biomarkers can stratify patients with endometrial failure even if the number of their attempts was insufficient to meet clinical RIF classification criteria . To our knowledge, this is thefirst study to apply semi-supervised learning to the endometrium, conferring an advantage over previous models that were only trained with clinical classifications and did not leverage the underlying molecular heterogeneity of the condition . The lack of gene expression differences between RIF and control patients with our initial clinical classification reinforced findings from Macklon’s group and the molecular heterogeneity among patients with RIF . The EFR signature comprised 122 genes, including 44 genes newly associated with an endometrial gene expression signature and a single common gene with Koot’s pioneer endometrial timing-independent signature for endometrial pathology . Here, we only considered biopsies collected in the endometrial mid-secretory phase, when patients had received P4 for 82- 172 h. Although this signature is independent of endometrial luteal phase timing, further studies are required to confirm if the signature is detectable in other phases of the endometrial cycle. Progestagen-associated endometrial protein is well characterized in endometrial tissue and was the most upregulated gene (FC1⁄4 3.18) in the poor endometrial prognosis pro file. However, the levels we observed were 10-fold lower compared with previous studies where changes in endometrial timing were considered . Alternatively,fibrinogen beta chain, related to the plasma membrane and vesicle transport by secretory granules, was the most downregulated gene. This gene accumulated polymorphisms that hinder the production offibrinogen and its associated anti-inflammatory activity  and was recently related to the dysregulated pathways in RIF . FOXP1, the most predictive gene, is a transcription factor mediated by estrogen that has established functions in regulation, leukocyte differentiation, and response to lipids and chemokines. This gene is associated with breast  endometrial cancer , and endometriosis-related fibrosis , but was never included in an endometrial gene signature until now. With FOXP1 and the FOX transcription factor family being associated with the regulation of the menstrual cycle and implantation , this evidence reinforces our prior study  and affirms that alterations in endometrial regulation are an underlying cause of endometrial failure. Our clinical algorithm considers the 2 WOI factors, endometrial timing and disruption, to stratify IVF patients on the basis of 4 possible phenotypes: displaced/on-time and poor/ good prognosis. This model is promising for preventive and precision medicine in endometrial-factor infertility, as it combines the robust prediction parameters of the TED model  and the EFR biomarker signature’s predictive capability reported herein. In contrast to other transcriptomic-based endometrial dating tools , our bioinformatic methodology removes the molecular variation associated with the cyclic menstrual changes, providing deeper insights for identifying a new kind of disruption independent of endometrial luteal phase timing by the EFR signature. Furthermore, the EFR biomarker signature was developed on the basis of reproductive outcomes and effectively classifies patients predicted to have positive or negative reproductive outcomes in the first SET after biopsy collection. All these facts, point out to a more potential clinical utility than existing tools that only predict the WOI without showing significant differences in reproductive outcomes as have been reported  and we have shown in ourfindings. Although our strategy appears promising, the classification presented herein needs further re finement before clinical implementation. Larger sample sizes and validation in independent samples are required to ensure model generalization . Healthcare professionals will need to be adequately trained to ensure technical reproducibility. Longitudinal assessments including several menstrual cycles in the same patient are recommended to confirm inter-cycle reproducibility and to gain a comprehensive understanding of the reliability of the EFR panel. Additional studies should assess the relationship between a poor endometrial prognosis and the presence of benign disorders that potentially compromise endometrial function and/or embryo implantation . Inspired by the MammaPrint risk signature for breast cancer , the EFR signature could become a nextgeneration tool used to classify patients on the basis of endometrial prognosis, to predict reproductive outcomes before undergoing IVF, potentially experiencing RIF and/or losing valuable embryos. Although treatments for patients with poor endometrial prognosis remain to be determined, clinically identifying these patients is the first step toward improving their care.',\n",
       "  'Conclusion': 'Conclusion(s): The EFR signature revealed a novel endometrial disruption, independent of endometrial luteal phase timing, present in 73.7% of patients. This EFR signature stratified patients into 2 significantly distinct and clinically relevant prognosis profiles providing opportunities for personalized therapy. Nevertheless, further validations are needed before implementing this gene signature as an artificial intelligence (AI)-based tool to reduce the risk of patients experiencing endometrial failure. (Fertil Steril /C2102024;-:-- -. /C2112024 by American Society for Reproductive Medicine.)',\n",
       "  'URL': 'https://doi.org/10.1016/j.fertnstert.2024.03.015',\n",
       "  'Journal': 'Fertil Steril®'},\n",
       " {'PaperTitle': 'Predicting risk of endometrial failure: a biomarker signature that identifies a novel disruption independent of endometrial timing in patients undergoing hormonal replacement cycles',\n",
       "  'Publication': '2024',\n",
       "  'Authors': 'Patricia Diaz-Gimeno, Ph.D., Patricia Sebastian-Leon, Ph.D., Katharina Spath, Ph.D., Diana Marti-Garcia, M.Sc., Josefa Maria Sanchez-Reyes, Ph.D., Maria del Carmen Vidal, Ph.D., Almudena Devesa-Peiro, Ph.D., Immaculada Sanchez-Ribas, Ph.D., Asunta Martinez-Martinez, M.Sc., Nuria Pellicer, M.D., Ph.D., Dagan Wells, Ph.D., and Antonio Pellicer, M.D., Ph.D.',\n",
       "  'Email': 'patricia.diaz@ivirma.com or patricia_diaz@iislafe.es',\n",
       "  'Abstract': 'Objective: To propose a new gene expression signature that identifies endometrial disruptions independent of endometrial luteal phase timing and predicts if patients are at risk of endometrial failure. Design: Multicentric, prospective study. Setting: Reproductive medicine research department in a public hospital affiliated with private fertility clinics and a reproductive genetics laboratory. Patients: Caucasian women (n1⁄4 281; 39.4/C6 4.8 years old with a body mass index of 22.9/C6 3.5 kg/m 2) undergoing hormone replacement therapy between July 2018 and July 2021. Endometrial samples from 217 patients met RNA quality criteria for signature discovery and analysis. Intervention(s): Endometrial biopsies collected in the mid-secretory phase. Main Outcome Measure(s):Endometrial luteal phase timing-corrected expression of 404 genes and reproductive outcomes of thefirst single embryo transfer (SET) after biopsy collection to identify prognostic biomarkers of endometrial failure. Results: Removal of endometrial timing variation from gene expression data allowed patients to be stratified into poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognosis groups on the basis of their clinical and transcriptomic profiles. Significant differences were found between endometrial prognosis groups in terms of reproductive rates: pregnancy (44.6% vs. 79.6%), live birth (25.6% vs. 77.6%), clinical miscarriage (22.2% vs. 2.6%), and biochemical miscarriage (20.4% vs. 0%). The relative risk of endometrial failure for patients predicted as a poor endometrial prognosis was 3.3 times higher than those with a good prognosis. The differences in gene expression between both profiles were proposed as a biomarker, coined the endometrial failure risk (EFR) signature. Poor prognosis profiles were characterized by 59 upregulated and 63 downregulated genes mainly involved in regulation (17.0%), metabolism (8.4%), immune response, and inflammation (7.8%). This EFR signature had a median accuracy of 0.92 (min1⁄40.88, max1⁄40.94), median sensitivity of 0.96 (min1⁄40.91, max 1⁄4 0.98), and median specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88), positioning itself as a promising biomarker for endometrial evaluation. Conclusion(s): The EFR signature revealed a novel endometrial disruption, independent of endometrial luteal phase timing, present in 73.7% of patients. This EFR signature stratified patients into 2 significantly distinct and clinically relevant prognosis profiles providing opportunities for personalized therapy. Nevertheless, further validations are needed before implementing this gene signature as an artificial intelligence (AI)-based tool to reduce the risk of patients experiencing endometrial failure. (Fertil Steril /C2102024;-:-- -. /C2112024 by American Society for Reproductive Medicine.)',\n",
       "  'Introduction': 'A key factor for reproductive success in assisted reproduction treatments is the status of the maternal endometrium during embryo implantation and fetal development. The endometrial cycle is reflected by the cyclic structural and functional changes of the endometrium across the menstrual cycle, particularly during the mid-secretory phase, to prepare for the‘‘window of implantation’’ (WOI) . After successful embryo implantation, the decidua (specialized layer of the endometrium) actively encapsulates the trophectoderm to support placentation and provide adequate vascularization for optimal fetal growth . Thus, approaches that predict and prevent endometrial-factor infertility could substantially improve assisted reproduction treatment outcomes by supporting the establishment and maintenance of pregnancy . There is a lack of consensus on the minimum number of implantation failures with good-quality embryos derived from ovum donation or with confirmed euploid karyotypes required to diagnose recurrent implantation failure (RIF) . The disparities in clinical classification and heterogeneous etiology of RIF confound its diagnosis and leave the exact prevalence unclear . As clinical symptoms cannot be prevented and do not provide sufficient criteria to stratify patients, patients might be misclassi fied and receive inefficient treatment. Indeed, most studies using clinical criteria for RIF identified no or few differentially expressed genes between samples of affected patients , reflecting the molecular heterogeneity of RIF. To overcome these healthcare challenges, high-throughput molecular technologies are combined with artificial intelligence (AI) algorithms to provide a deeper understanding of infertilityrelated pathogenesis . These molecular insights are then applied in genomics-driven precision medicine approaches to characterize complex diseases, and ultimately, improve patient diagnosis and prognosis . Employing genomics in reproductive medicine has the potential to go beyond treating the clinical manifestations of disease progression , by leveraging the patients’ heterogeneous molecular profiles for precise stratification and revealing new targets for preventive measures . In this regard, current diagnostic tools for endometrialfactor infertility use transcriptomics to date the endometrial biopsy in the mid-secretory phase, exceeding the accuracy and objectivity of traditional endometrial progression staging by Noyes’ histological criteria . However, the clinical utility of synchronizing the endometrium by adjusting the timing of progesterone (P4) administration on the basis of the patient’s transcriptomic endometrial dating profile remains controversial . Independent randomized clinical trials found no improvement in reproductive outcomes , indicating the inefficacy of this clinical intervention and highlighting the need for alternative treatments and/or a refined target patient population that bene fits from this strategy. Combining these precision medicine approaches with transcriptomics and AI algorithms; our group recently proposed a novel molecular taxonomy of RIF and a new clinical algorithm for endometrial-factor infertility that considered 2 molecular causes of implantation failure: a WOI displacement that indicates a dysregulated endometrial timing; and a WOI disruption that indicates an endometrial alteration independent of endometrial timing . Transcriptomic analysis revealed these 2 WOI phenotypes were presented exclusively or simultaneously and patients with RIF had significantly more WOI disruptions but a similar frequency of WOI displacements, compared with controls . Thesefindings suggested that WOI disruptions had stronger associations with implantation failure than WOI displacements and shifted the paradigm for the study and treatment of endometrialfactor infertility . In addition, this work by SebastianLeon et al. highlighted the importance of removing the transcriptomic variation due to cyclical endometrial tissue changes (ie, the effect of endometrial timing) that, if not corrected, could mask the significant disruptions in endometrial gene expression. However, given this hypothesis-driven research was conducted retrospectively and in silico with publicly available data, the reproductive outcomes of the cycle after the biopsy collection were not available . This clinical follow-up is required to discover biomarkers for WOI disruption in patients with a higher risk of endometrial failure. On the basis of these genomics-driven precision medicine strategies, this proof-of-concept study aimed to discover a biomarker panel that could identify endometrial disruption, independent of endometrial timing, to predict, and ultimately prevent, poor reproductive outcomes in in vitro fertilization (IVF) patients.',\n",
       "  'Methods': 'MATERIAL AND METHODS Ethics statement This study was approved by the Ethics Committee of the Instituto Valenciano de Infertilidad, Valencia, Spain (1810-FIVI-066-PD/1810-FIVI-048-PD). Written informed consent was obtained from all recruited patients (n1⁄4 281). Participants, collection of endometrial biopsies, and reproductive outcomes This prospective, multicentric study was conducted between July 2018 and July 2021 at 5 private fertility clinics in Spain. Participants (n1⁄4 281) were scheduled for endometrial evaluation due to medical indications orR2 implantation failures of idiopathic origin. All patients met the following inclusion criteria: 18- 50 years old, with a body mass index (BMI) of 19- 30 kg/m 2, undergoing hormone replacement therapy for single embryo transfer (SET), presenting an endometrial thickness >6.5 mm with trilaminar structure in proliferative phase before P4 administration, and serum levels of estradiol >100 pg/mL and P4<1 ng/mL on the 10th day of estrogen treatment . Endometrial biopsies were collected after an average of 114.5/C6 7.2 h of P4 (duration ranged from 82 to 172 h), and processed as we previously described . Baseline patient characteristics including age, BMI, the number of IVF attempts, the quality of the embryo transferred (defined in theSupplemental Methods, available online), and reproductive outcomes (ie, pregnancy, live birth, and biochemical and clinical miscarriage) for each attempt, were obtained from internal medical records. Information about IVF attempts in other fertility clinics was included if recorded (only in 47.7% of patients). Study design First, a custom RNA-seq gene panel (the endometrial failure risk [EFR] panel) comprising 404 biomarkers of endometrial luteal phase timing and/or potential disruption was designed and employed to determine the risk of endometrial failure among 217 IVF patients (Supplemental Fig. 1, step 1 available online). The endometrial timing-corrected expression of the EFR panel was used to study the differential expression associated with endometrial disruption together with clinical characteristics of IVF patients to stratify the population as having poor or good endometrial prognoses, according to a semi-supervised learning process (Supplemental Fig. 1, step 2). The distinct transcriptomic patterns of each prognosis group were associated with the differences in reproductive outcomes of SET after biopsy collection, and the differentially expressed gene signature was proposed as a biomarker of endometrial disruption (Supplemental Fig. 1, step 3). Finally, the gene expression signature was used to supervise machine learning algorithms to evaluate the biomarker capacity of this signature to distinguish endometrial competence. The predictive performance of the models was assessed through the accuracy, sensibility, and specificity over 100 iterations of fivefold strati fied cross-validation ( Supplemental Fig. 1 , step 4). Constructing the EFR gene panel Supplemental Figure 2illustrates the workflow for the EFR panel gene selection. A whole-transcriptome endometrial data set (GSE58144, n1⁄4 115) retrieved from Gene Expression Omnibus was processed to correct the endometrial progression effect, using linear models in limma (version 3.46.0) , and unmask genes related to endometrial disruption . These genes were ranked according to an informativity score (CorrelationAttributeEval) in Weka (version 3.8.2, December 22, 2017) to evaluate their worth by measuring their correlation with endometrial pathology. Gene sets with increasing gene signature size were developed by adding the most informative genes in a stepwise manner, and their predictive performance was analyzed using Support Vector Machine and Random Forest algorithms. After calculating predictive performance with a stratified fivefold cross-validation (default parameters; 80:20; 100 iterations) using the Weka R package , the optimal signature size was determined for each algorithm, as the one with the fewest genes and highest accuracy. Finally, genes we previously prioritized for transcriptomic endometrial dating w e r ea l s oi n c l u d e di nt h e EFR panel to be able to remove the endometrial luteal phase timing variation. Endometrial biopsy processing, sequencing, and transcriptomic preprocessing Total RNA was extracted, and only samples that met our quality criteria were sequenced using the TruSeq Targeted RNA Illumina protocol and our custom EFR panel, in the Illumina NextSeq 550 platform with a paired-end design of 150 cycles and 4 M reads/sample, as previously described . Raw transcriptomic data was processed and normalized using limma, to filter low-quality raw counts (Q< 30) and low-read samples (<2M ). Possible batch effects were detected using principal components analysis and corrected with linear models in limma. Finally, endometrial luteal phase timing effects were detected using our transcriptomic endometrial dating (TED) model (see Supplemental Methods) and removed using limma. EFR biomarker signature identification A preliminary classification of patients was based on the patients’ history of reproductive outcomes and transcriptomic patterns corrected for endometrial luteal phase timing. This approach eliminated the gene expression variation related to the cyclic tissue changes that mask endometrial disruption and was used before applying semi-supervised machine learning on the basis of self-labeled techniques (see Supplemental Methods). Once, the genomic-driven disease taxonomy was established, the differentially expressed genes obtained between the final good and poor prognosis groups were proposed as the EFR biomarker signature. The limma R package was used to identify the statistically signi ficant differentially expressed genes (false discovery rate [FDR]-adjustedP value <.05). Functional characterization of the EFR signature The biomarker signature genes were annotated for functional interpretation databases by consulting Kyoto Encyclopaedia of Genes and Genomes human pathways (release 99.0, August 1, 2021) and Gene Ontology (version July 2, 2021; employing only experimental annotations) . Functional terms were manually summarized into categories according to their description in the database and presented as a bar plot using ggplot2 . EFR signature capability as a biomarker of endometrial disruption To estimate the prediction capability of the EFR signature to distinguish poor and good endometrial prognosis profiles, we implemented predictive models on the basis of the Support Vector Machine algorithm. Using a stratified fivefold crossvalidation process (repeated 100 times), the accuracy, sensitivity, and specificity were estimated. The range of values obtained across the 100 iterations was presented as a boxplot using ggplot2. Statistical analysis Rates for reproductive outcomes (ie, pregnancy rate [PR], cumulative pregnancy rate [CPR], live birth rate [LBR], biochemical miscarriage rate [BMR], and clinical miscarriage rate [CMR]) were calculated as de fined in the Supplemental Methods. Descriptive statistics were used for transcriptomic and clinical characteristics of patients to compare prognosis or WOI displacement groups. Continuous variables were presented as an overall mean /C6 standard deviation whereas discrete variables were presented as counts and percentages. Groups were compared using the Wilcoxon rank-sum test for continuous variables and Fisher’s exact test for discrete variables. All statistical analyses were conducted in R. All graphical results were generated with ggplot2. Gene expression validation by quantitative realtime polymerase chain reaction Quantitative real-time polymerase chain reaction (RT-qPCR) was performed to validate the expression of the top 4 differentially expressed genes between patients with poor (n1⁄4 10) and good (n1⁄4 10) endometrial prognoses. RNA (200- 220 ng) was reverse transcribed into cDNA using the PrimeScript Reagent Kit (Perfect Real Time, Takara, Japan) on a Thermocycler T3000 (Biometra, Ireland). Quantitative real-time polymerase chain reaction was performed on a StepOnePlus Real-Time PCR System (Applied Biosystems, USA) using Power-Up SYBR Green (Thermo Fisher Scientific, USA) and standard conditions.Supplemental Table 1(available online) lists the specific primers (Invitrogen, Thermo Fisher Scientific) used. Universal human RNA (Agilent, Spain) and water were respectively included as positive and negative controls. Each sample was analyzed in duplicate and relative gene expression (normalized to beta actin) was calculated using the DDCt method .',\n",
       "  'Results': 'RESULTS Preliminary sequencing with the 404-gene EFR panel The EFR panel evaluated 404 genes: 139 that predict endometrial disruption and 310 that serve as biomarkers of endometrial dating or timing. Notably, 45 genes overlapped in these 2 endometrial factors. Of the 281 endometrial samples collected, we excluded 56 samples for having low-quality or insufficient RNA; 6 samples containing a low number of sequencing reads (Supplemental Fig. 3A); and 2 samples that had low gene expression after normalization ( Supplemental Fig. 3B ). Thus, 217 patients were considered in the downstream analysis. After correcting batch effects from sequencing runs (Supplemental Fig. 3C) and removing the transcriptomic variation due to endometrial cyclic changes (endometrial luteal phase timing;Supplemental Fig. 3D), there were no other confounding technical or clinical factors (Supplemental Fig. 4). A genomic-driven endometrial taxonomy for patient stratification on the basis of a novel endometrial disruption independent of endometrial timing Forty-four patients were initially categorized as acute RIF on the basis of the inability to carry a pregnancy to term after 3- 9 SETs. Alternatively, 40 patients were categorized as acute Fertile because they reached full-term pregnancy at thefirst SET after biopsy. Notably, after this clinical acute criteria, 133 (61.3%) patients remained unclassified for having insufficient attempts (Supplemental Fig. 5A), and no transcriptomic differences were found between the acute RIF and acute Fertile groups (the lowest FDR was 0.3027). Incorporation of transcriptomic data with this initial acute clinical classification allowed us to stratify patients with poor (n1⁄4 137, 63.1%) or good (n1⁄4 49, 22.6%) endometrial prognoses; leaving only 31 samples (14.3%) unclassified because of the lack of recognition by the algorithms (Supplemental Fig. 5B). EFR biomarker signature identification and functional characterization A total of 122 genes were identified as significantly different (30.2% of all panel genes, FDR< 0.05) between patients with good and poor endometrial prognoses, including 63 downregulated and 59 upregulated in patients with poor endometrial prognosis (Table 1 and Fig. 1A), and proposed as the EFR biomarker signature. Transcriptomically, endometrial biopsies clustered by prognosis (Fig. 1B), and the signature genes had functions that were mainly related to regulation (17%); metabolism (8.4%); the immune system and inflammation (7.8%); protein processes (7.3%), signaling (6.9%), transport (6.1%), and the cell cycle, proliferation, and differentiation (5.3%) (Fig. 1C andSupplemental Table 2). The poor prognosis profile was distinguished by a downregulation of 27/39 (69.2%) functional groups. We highlight the functions with higher proportions of downregulated genes as steroidogenesis and steroid signaling (77.8%), hormone-related (80.0%), the nervous system (81.3%), gametogenesis and reproduction (85.7%), and oxidative stress and response to O 2 (88.9%), as well as cell action potential and polarization (100%) (Fig. 1D). Few functional groups, including cell migration, extracellular matrix, and glucose homeostasis contained a larger proportion of upregulated genes (60.9%, 62.5%, and 62.5% upregulated genes, respectively;Fig. 1D). Differences in clinical reproductive outcomes authenticated the EFR signature’s clinical relevance There were no significant differences in the age or BMI of patients predicted to have good/poor endometrial prognosis (all P>.05). Women predicted to have poor endometrial prognosis previously underwent significantly more SETs than those with good endometrial prognosis (2.8 vs. 1.6;P<.0001), however, there were no significant differences in embryo quality (P1⁄4.4679) between groups (Supplemental Table 3). The lack of differences in the proportion of out-of-phase endometria (57% in poor vs. 47% in good,P1⁄4.25) between good and poor prognosis groups (Supplemental Table 3) distinguished this signature from established endometrial luteal phase timing phenotypes (displaced or on-time). Therefore, the significant differences (P<.0001) between the reproductive outcomes of women predicted to have poor and good endometrial prognoses validated the clinical relevance of our transcriptomic signature. Specifically, the poor endometrial prognosis was associated with a 34.5% decline in PR (P1⁄43.8E-05), a 52% reduction in LBR (P1⁄45E-10), a 19.6% rise in CMR (P1⁄4.0066), and 20.4% increase in BMR (P1⁄4.0023) (Fig. 2A). Furthermore, patients with a poor endometrial prognosis only achieved a CPR of 42.3% after the ninth SET, whereas patients with a good prognosis had an 83.7% CPR at thefifth attempt (Fig. 2B). The relative risk of implantation failure or miscarriage in women predicted to have poor endometrial prognosis was respectively 2.7 and 11.5 compared with women with good endometrial prognosis. However, when failures in implantation or ongoing pregnancy were considered together, the relative risk of endometrial failure (not achieving a live birth in the next SET) for patients with poor endometrial prognosis was 3.3. EFR signature biomarker’s predictive capability The 186 samples labeled with poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognoses were used to evaluate the signature’s predictive performance. The 122 differentially expressed genes between poor/good endometrial prognoses proposed as the EFR biomarker signature were employed to train the model. Cross-validation yielded a median accuracy of 0.92 (min1⁄4 0.88, max 1⁄4 0.94), sensitivity of 0.96 (min1⁄4 0.91, max 1⁄4 0.98), and specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88) (Fig. 2C). Notably, 163 of 186 (87.6%) samples were classified correctly in almost 90% of all cross-validation models, whereas 151 of 186 (81.2%) samples were well classified correctly in 100% of the iterations. Twelve and eleven samples were misclassified consistently, in <50% or 50%- 90% of models, respectively (Fig. 2D). Clinical algorithm results after evaluating both WOI factors Next, we grouped samples according to both potential causes of implantation failure, endometrial luteal phase timing (displaced or on-time) (Fig. 3A), and endometrial disruption (good or poor prognosis) (Fig. 2A) and compared the patients’ reproductive outcomes. There were no significant differences between the patients with displaced and on-time WOI in terms of BMI (P1⁄4.8740), the average number of SETs (P1⁄4.0879), and embryo quality (P1⁄41). However, patients with a displaced WOI were signi ficantly older (40.4 vs. 38.3 years old; P1⁄4.0044) (Supplemental Table 4). Once patients were stratified by prognosis (endometrial disruption), the proportion of patients with poor prognosis was similar (P1⁄4.2458) between displaced and on-time groups (Fig. 3B). Experimental validation of genes related to the EFR signature The expression of the top 4 differentially expressed genes in the poor prognosis group was validated by RT-qPCR. Progestagen-associated endometrial protein and G-coupled protein receptor 110, which were found upregulated by RNA-seq, were significantly overexpressed in the poor prognosis group, compared with the good prognosis group (fold change [FC]1⁄4 6.157 [P1⁄4.0288] and FC1⁄4 11.803 [P1⁄4.0350], respectively; Supplemental Figs. 6Aand B). Meanwhile, the expression of fibrinogen beta chain and solute carrier 16 member 6, which were found downregulated by RNA-seq, did not reach a statistically significant difference (FC1⁄4 1.582 [P1⁄4.2475] and FC1⁄4 2.311 [P1⁄4.4376], respectively;Supplemental Figs. 6Cand D).',\n",
       "  'Discussion': 'DISCUSSION Advances in the diagnosis and treatment of RIF have been limited by its elusive etiology, the discrepancies in clinical classification (mainly due to the lack of consensus over the minimum number of pregnancy attempts) , and the heterogeneous molecular profiles of affected patients . On the basis of current clinical practice, patients with RIF may be misclassified and/or receive inef ficient treatment . Thus, the aim of this study was to propose a novel gene expression signature that could identify endometrial disruptions independent of endometrial luteal phase timing. Using a semi-supervised learning AI strategy together with the endometrial gene expression obtained from our custom EFR panel, we stratified patients according to poor or good endometrial prognoses and demonstrated how this signature reliably predicted gene expression patterns that corresponded with significant differences in PR, LBR, CMR, and BMR. When considering both WOI factors (disruption and timing) in our study cohort, poor endometrial prognoses (disruption) affected 73.7% of patients, whereas WOI displacements only affected 54.3%; 41.9% of patients presented a displaced WOI and were predicted to have a poor endometrial prognosis, whereas only 12.4% of patients presented with a displaced WOI and were predicted to have a good endometrial prognosis. These results offer a possible explanation as to why other studies reported there was no clinical benefito f personalizing embryo transfers on the basis of transcriptomic dating of the endometrium . According to our clinical algorithm, only patients with a displaced WOI and a good endometrial prognosis (12.4% of our cohort) would be better off with this approach. Patients with poor endometrial prognosis (73.7% of our cohort) would have suboptimal reproductive outcomes even though their WOI was on-time. This highlights an unmet healthcare need for IVF patients with endometrial disruptions and underscores the importance of additional prospective studies to further characterize patients on the basis of these 2 molecular causes for trying tailored treatments. Contemplating endometrial luteal phase timing together with a prevalent disruption or poor endometrial prognosis shifts the paradigm in studies about the endometrium. Our innovative methodology based on transcriptomics and AI models led us to develop new endometrial biomarker taxonomies related to endometrial luteal phase timing and/or endometrial disruption . There were 30 genes overlapping between the previously published TED signature and the EFR signature presented herein, emphasizing their correlation despite the lower FC of biomarkers for endometrial disruption versus endometrial timing . AI algorithms offer powerful approaches for patient stratification, by combining high-throughput molecular data and clinical variables . Clinical translation of these approaches will require careful supervision of how the machine learning is initially trained . However, a semi-supervised learning methodology implementing accurate clinical diagnostic criteria and patient-specific genomic data to establish relevant molecular taxonomies will be indispensable for advancing precision medicine , especially within the context of endometrial-factor infertility. Predictive biomarkers can stratify patients with endometrial failure even if the number of their attempts was insufficient to meet clinical RIF classification criteria . To our knowledge, this is thefirst study to apply semi-supervised learning to the endometrium, conferring an advantage over previous models that were only trained with clinical classifications and did not leverage the underlying molecular heterogeneity of the condition . The lack of gene expression differences between RIF and control patients with our initial clinical classification reinforced findings from Macklon’s group and the molecular heterogeneity among patients with RIF . The EFR signature comprised 122 genes, including 44 genes newly associated with an endometrial gene expression signature and a single common gene with Koot’s pioneer endometrial timing-independent signature for endometrial pathology . Here, we only considered biopsies collected in the endometrial mid-secretory phase, when patients had received P4 for 82- 172 h. Although this signature is independent of endometrial luteal phase timing, further studies are required to confirm if the signature is detectable in other phases of the endometrial cycle. Progestagen-associated endometrial protein is well characterized in endometrial tissue and was the most upregulated gene (FC1⁄4 3.18) in the poor endometrial prognosis pro file. However, the levels we observed were 10-fold lower compared with previous studies where changes in endometrial timing were considered . Alternatively,fibrinogen beta chain, related to the plasma membrane and vesicle transport by secretory granules, was the most downregulated gene. This gene accumulated polymorphisms that hinder the production offibrinogen and its associated anti-inflammatory activity and was recently related to the dysregulated pathways in RIF . FOXP1, the most predictive gene, is a transcription factor mediated by estrogen that has established functions in regulation, leukocyte differentiation, and response to lipids and chemokines. This gene is associated with breast endometrial cancer , and endometriosis-related fibrosis , but was never included in an endometrial gene signature until now. With FOXP1 and the FOX transcription factor family being associated with the regulation of the menstrual cycle and implantation , this evidence reinforces our prior study and affirms that alterations in endometrial regulation are an underlying cause of endometrial failure. Our clinical algorithm considers the 2 WOI factors, endometrial timing and disruption, to stratify IVF patients on the basis of 4 possible phenotypes: displaced/on-time and poor/ good prognosis. This model is promising for preventive and precision medicine in endometrial-factor infertility, as it combines the robust prediction parameters of the TED model and the EFR biomarker signature’s predictive capability reported herein. In contrast to other transcriptomic-based endometrial dating tools , our bioinformatic methodology removes the molecular variation associated with the cyclic menstrual changes, providing deeper insights for identifying a new kind of disruption independent of endometrial luteal phase timing by the EFR signature. Furthermore, the EFR biomarker signature was developed on the basis of reproductive outcomes and effectively classifies patients predicted to have positive or negative reproductive outcomes in the first SET after biopsy collection. All these facts, point out to a more potential clinical utility than existing tools that only predict the WOI without showing significant differences in reproductive outcomes as have been reported and we have shown in ourfindings. Although our strategy appears promising, the classification presented herein needs further re finement before clinical implementation. Larger sample sizes and validation in independent samples are required to ensure model generalization . Healthcare professionals will need to be adequately trained to ensure technical reproducibility. Longitudinal assessments including several menstrual cycles in the same patient are recommended to confirm inter-cycle reproducibility and to gain a comprehensive understanding of the reliability of the EFR panel. Additional studies should assess the relationship between a poor endometrial prognosis and the presence of benign disorders that potentially compromise endometrial function and/or embryo implantation . Inspired by the MammaPrint risk signature for breast cancer , the EFR signature could become a nextgeneration tool used to classify patients on the basis of endometrial prognosis, to predict reproductive outcomes before undergoing IVF, potentially experiencing RIF and/or losing valuable embryos. Although treatments for patients with poor endometrial prognosis remain to be determined, clinically identifying these patients is the first step toward improving their care.',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1016/j.fertnstert.2024.03.015',\n",
       "  'Journal': 'Fertil Steril®'},\n",
       " {'PaperTitle': 'Predicting risk of endometrial failure: a biomarker signature that identifies a novel disruption independent of endometrial timing in patients undergoing hormonal replacement cycles',\n",
       "  'Publication': '2024',\n",
       "  'Authors': 'Patricia Diaz-Gimeno, Ph.D., Patricia Sebastian-Leon, Ph.D., Katharina Spath, Ph.D., Diana Marti-Garcia, M.Sc., Josefa Maria Sanchez-Reyes, Ph.D., Maria del Carmen Vidal, Ph.D., Almudena Devesa-Peiro, Ph.D., Immaculada Sanchez-Ribas, Ph.D., Asunta Martinez-Martinez, M.Sc., Nuria Pellicer, M.D., Ph.D., Dagan Wells, Ph.D., and Antonio Pellicer, M.D., Ph.D.',\n",
       "  'Email': 'patricia.diaz@ivirma.com or patricia_diaz@iislafe.es',\n",
       "  'Abstract': 'Objective: To propose a new gene expression signature that identifies endometrial disruptions independent of endometrial luteal phase timing and predicts if patients are at risk of endometrial failure. Design: Multicentric, prospective study. Setting: Reproductive medicine research department in a public hospital affiliated with private fertility clinics and a reproductive genetics laboratory. Patients: Caucasian women (n1⁄4 281; 39.4/C6 4.8 years old with a body mass index of 22.9/C6 3.5 kg/m 2) undergoing hormone replacement therapy between July 2018 and July 2021. Endometrial samples from 217 patients met RNA quality criteria for signature discovery and analysis. Intervention(s): Endometrial biopsies collected in the mid-secretory phase. Main Outcome Measure(s):Endometrial luteal phase timing-corrected expression of 404 genes and reproductive outcomes of thefirst single embryo transfer (SET) after biopsy collection to identify prognostic biomarkers of endometrial failure. Results: Removal of endometrial timing variation from gene expression data allowed patients to be stratified into poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognosis groups on the basis of their clinical and transcriptomic profiles. Significant differences were found between endometrial prognosis groups in terms of reproductive rates: pregnancy (44.6% vs. 79.6%), live birth (25.6% vs. 77.6%), clinical miscarriage (22.2% vs. 2.6%), and biochemical miscarriage (20.4% vs. 0%). The relative risk of endometrial failure for patients predicted as a poor endometrial prognosis was 3.3 times higher than those with a good prognosis. The differences in gene expression between both profiles were proposed as a biomarker, coined the endometrial failure risk (EFR) signature. Poor prognosis profiles were characterized by 59 upregulated and 63 downregulated genes mainly involved in regulation (17.0%), metabolism (8.4%), immune response, and inflammation (7.8%). This EFR signature had a median accuracy of 0.92 (min1⁄40.88, max1⁄40.94), median sensitivity of 0.96 (min1⁄40.91, max 1⁄4 0.98), and median specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88), positioning itself as a promising biomarker for endometrial evaluation. Conclusion(s): The EFR signature revealed a novel endometrial disruption, independent of endometrial luteal phase timing, present in 73.7% of patients. This EFR signature stratified patients into 2 significantly distinct and clinically relevant prognosis profiles providing opportunities for personalized therapy. Nevertheless, further validations are needed before implementing this gene signature as an artificial intelligence (AI)-based tool to reduce the risk of patients experiencing endometrial failure. (Fertil Steril /C2102024;-:-- -. /C2112024 by American Society for Reproductive Medicine.)',\n",
       "  'Introduction': 'A key factor for reproductive success in assisted reproduction treatments is the status of the maternal endometrium during embryo implantation and fetal development. The endometrial cycle is reflected by the cyclic structural and functional changes of the endometrium across the menstrual cycle, particularly during the mid-secretory phase, to prepare for the‘‘window of implantation’’ (WOI) . After successful embryo implantation, the decidua (specialized layer of the endometrium) actively encapsulates the trophectoderm to support placentation and provide adequate vascularization for optimal fetal growth . Thus, approaches that predict and prevent endometrial-factor infertility could substantially improve assisted reproduction treatment outcomes by supporting the establishment and maintenance of pregnancy . There is a lack of consensus on the minimum number of implantation failures with good-quality embryos derived from ovum donation or with confirmed euploid karyotypes required to diagnose recurrent implantation failure (RIF) . The disparities in clinical classification and heterogeneous etiology of RIF confound its diagnosis and leave the exact prevalence unclear . As clinical symptoms cannot be prevented and do not provide sufficient criteria to stratify patients, patients might be misclassi fied and receive inefficient treatment. Indeed, most studies using clinical criteria for RIF identified no or few differentially expressed genes between samples of affected patients , reflecting the molecular heterogeneity of RIF. To overcome these healthcare challenges, high-throughput molecular technologies are combined with artificial intelligence (AI) algorithms to provide a deeper understanding of infertilityrelated pathogenesis . These molecular insights are then applied in genomics-driven precision medicine approaches to characterize complex diseases, and ultimately, improve patient diagnosis and prognosis . Employing genomics in reproductive medicine has the potential to go beyond treating the clinical manifestations of disease progression , by leveraging the patients’ heterogeneous molecular profiles for precise stratification and revealing new targets for preventive measures . In this regard, current diagnostic tools for endometrialfactor infertility use transcriptomics to date the endometrial biopsy in the mid-secretory phase, exceeding the accuracy and objectivity of traditional endometrial progression staging by Noyes’ histological criteria . However, the clinical utility of synchronizing the endometrium by adjusting the timing of progesterone (P4) administration on the basis of the patient’s transcriptomic endometrial dating profile remains controversial . Independent randomized clinical trials found no improvement in reproductive outcomes , indicating the inefficacy of this clinical intervention and highlighting the need for alternative treatments and/or a refined target patient population that bene fits from this strategy. Combining these precision medicine approaches with transcriptomics and AI algorithms; our group recently proposed a novel molecular taxonomy of RIF and a new clinical algorithm for endometrial-factor infertility that considered 2 molecular causes of implantation failure: a WOI displacement that indicates a dysregulated endometrial timing; and a WOI disruption that indicates an endometrial alteration independent of endometrial timing . Transcriptomic analysis revealed these 2 WOI phenotypes were presented exclusively or simultaneously and patients with RIF had significantly more WOI disruptions but a similar frequency of WOI displacements, compared with controls . Thesefindings suggested that WOI disruptions had stronger associations with implantation failure than WOI displacements and shifted the paradigm for the study and treatment of endometrialfactor infertility . In addition, this work by SebastianLeon et al. highlighted the importance of removing the transcriptomic variation due to cyclical endometrial tissue changes (ie, the effect of endometrial timing) that, if not corrected, could mask the significant disruptions in endometrial gene expression. However, given this hypothesis-driven research was conducted retrospectively and in silico with publicly available data, the reproductive outcomes of the cycle after the biopsy collection were not available . This clinical follow-up is required to discover biomarkers for WOI disruption in patients with a higher risk of endometrial failure. On the basis of these genomics-driven precision medicine strategies, this proof-of-concept study aimed to discover a biomarker panel that could identify endometrial disruption, independent of endometrial timing, to predict, and ultimately prevent, poor reproductive outcomes in in vitro fertilization (IVF) patients.',\n",
       "  'Methods': 'MATERIAL AND METHODS Ethics statement This study was approved by the Ethics Committee of the Instituto Valenciano de Infertilidad, Valencia, Spain (1810-FIVI-066-PD/1810-FIVI-048-PD). Written informed consent was obtained from all recruited patients (n1⁄4 281). 2 VOL. - NO. - / - 2024 REPRODUCTIVE SCIENCE Participants, collection of endometrial biopsies, and reproductive outcomes This prospective, multicentric study was conducted between July 2018 and July 2021 at 5 private fertility clinics in Spain. Participants (n1⁄4 281) were scheduled for endometrial evaluation due to medical indications orR2 implantation failures of idiopathic origin. All patients met the following inclusion criteria: 18- 50 years old, with a body mass index (BMI) of 19- 30 kg/m 2, undergoing hormone replacement therapy for single embryo transfer (SET), presenting an endometrial thickness >6.5 mm with trilaminar structure in proliferative phase before P4 administration, and serum levels of estradiol >100 pg/mL and P4<1 ng/mL on the 10th day of estrogen treatment . Endometrial biopsies were collected after an average of 114.5/C6 7.2 h of P4 (duration ranged from 82 to 172 h), and processed as we previously described . Baseline patient characteristics including age, BMI, the number of IVF attempts, the quality of the embryo transferred (defined in theSupplemental Methods, available online), and reproductive outcomes (ie, pregnancy, live birth, and biochemical and clinical miscarriage) for each attempt, were obtained from internal medical records. Information about IVF attempts in other fertility clinics was included if recorded (only in 47.7% of patients). Study design First, a custom RNA-seq gene panel (the endometrial failure risk [EFR] panel) comprising 404 biomarkers of endometrial luteal phase timing and/or potential disruption was designed and employed to determine the risk of endometrial failure among 217 IVF patients (Supplemental Fig. 1, step 1 available online). The endometrial timing-corrected expression of the EFR panel was used to study the differential expression associated with endometrial disruption together with clinical characteristics of IVF patients to stratify the population as having poor or good endometrial prognoses, according to a semi-supervised learning process (Supplemental Fig. 1, step 2). The distinct transcriptomic patterns of each prognosis group were associated with the differences in reproductive outcomes of SET after biopsy collection, and the differentially expressed gene signature was proposed as a biomarker of endometrial disruption (Supplemental Fig. 1, step 3). Finally, the gene expression signature was used to supervise machine learning algorithms to evaluate the biomarker capacity of this signature to distinguish endometrial competence. The predictive performance of the models was assessed through the accuracy, sensibility, and specificity over 100 iterations of fivefold strati fied cross-validation ( Supplemental Fig. 1 , step 4). Constructing the EFR gene panel Supplemental Figure 2illustrates the workflow for the EFR panel gene selection. A whole-transcriptome endometrial data set (GSE58144, n1⁄4 115) retrieved from Gene Expression Omnibus was processed to correct the endometrial progression effect, using linear models in limma (version 3.46.0) , and unmask genes related to endometrial disruption . These genes were ranked according to an informativity score (CorrelationAttributeEval) in Weka (version 3.8.2, December 22, 2017) to evaluate their worth by measuring their correlation with endometrial pathology. Gene sets with increasing gene signature size were developed by adding the most informative genes in a stepwise manner, and their predictive performance was analyzed using Support Vector Machine and Random Forest algorithms. After calculating predictive performance with a stratified fivefold cross-validation (default parameters; 80:20; 100 iterations) using the Weka R package , the optimal signature size was determined for each algorithm, as the one with the fewest genes and highest accuracy. Finally, genes we previously prioritized for transcriptomic endometrial dating w e r ea l s oi n c l u d e di nt h e EFR panel to be able to remove the endometrial luteal phase timing variation. Endometrial biopsy processing, sequencing, and transcriptomic preprocessing Total RNA was extracted, and only samples that met our quality criteria were sequenced using the TruSeq Targeted RNA Illumina protocol and our custom EFR panel, in the Illumina NextSeq 550 platform with a paired-end design of 150 cycles and 4 M reads/sample, as previously described . Raw transcriptomic data was processed and normalized using limma, to filter low-quality raw counts (Q< 30) and low-read samples (<2M ). Possible batch effects were detected using principal components analysis and corrected with linear models in limma. Finally, endometrial luteal phase timing effects were detected using our transcriptomic endometrial dating (TED) model (see Supplemental Methods) and removed using limma. EFR biomarker signature identification A preliminary classification of patients was based on the patients’ history of reproductive outcomes and transcriptomic patterns corrected for endometrial luteal phase timing. This approach eliminated the gene expression variation related to the cyclic tissue changes that mask endometrial disruption and was used before applying semi-supervised machine learning on the basis of self-labeled techniques (see Supplemental Methods). Once, the genomic-driven disease taxonomy was established, the differentially expressed genes obtained between the final good and poor prognosis groups were proposed as the EFR biomarker signature. The limma R package was used to identify the statistically signi ficant differentially expressed genes (false discovery rate [FDR]-adjustedP value <.05). Functional characterization of the EFR signature The biomarker signature genes were annotated for functional interpretation databases by consulting Kyoto Encyclopaedia of Genes and Genomes human pathways (release 99.0, August 1, 2021) and Gene Ontology (version July 2, 2021; employing only experimental annotations) . Functional terms were manually summarized into categories according VOL. - NO. - / - 2024 3 Fertil Steril® to their description in the database and presented as a bar plot using ggplot2 . EFR signature capability as a biomarker of endometrial disruption To estimate the prediction capability of the EFR signature to distinguish poor and good endometrial prognosis profiles, we implemented predictive models on the basis of the Support Vector Machine algorithm. Using a stratified fivefold crossvalidation process (repeated 100 times), the accuracy, sensitivity, and specificity were estimated. The range of values obtained across the 100 iterations was presented as a boxplot using ggplot2. Statistical analysis Rates for reproductive outcomes (ie, pregnancy rate [PR], cumulative pregnancy rate [CPR], live birth rate [LBR], biochemical miscarriage rate [BMR], and clinical miscarriage rate [CMR]) were calculated as de fined in the Supplemental Methods. Descriptive statistics were used for transcriptomic and clinical characteristics of patients to compare prognosis or WOI displacement groups. Continuous variables were presented as an overall mean /C6 standard deviation whereas discrete variables were presented as counts and percentages. Groups were compared using the Wilcoxon rank-sum test for continuous variables and Fisher’s exact test for discrete variables. All statistical analyses were conducted in R. All graphical results were generated with ggplot2. Gene expression validation by quantitative realtime polymerase chain reaction Quantitative real-time polymerase chain reaction (RT-qPCR) was performed to validate the expression of the top 4 differentially expressed genes between patients with poor (n1⁄4 10) and good (n1⁄4 10) endometrial prognoses. RNA (200- 220 ng) was reverse transcribed into cDNA using the PrimeScript Reagent Kit (Perfect Real Time, Takara, Japan) on a Thermocycler T3000 (Biometra, Ireland). Quantitative real-time polymerase chain reaction was performed on a StepOnePlus Real-Time PCR System (Applied Biosystems, USA) using Power-Up SYBR Green (Thermo Fisher Scientific, USA) and standard conditions.Supplemental Table 1(available online) lists the specific primers (Invitrogen, Thermo Fisher Scientific) used. Universal human RNA (Agilent, Spain) and water were respectively included as positive and negative controls. Each sample was analyzed in duplicate and relative gene expression (normalized to beta actin) was calculated using the DDCt method .',\n",
       "  'Results': 'RESULTS Preliminary sequencing with the 404-gene EFR panel The EFR panel evaluated 404 genes: 139 that predict endometrial disruption and 310 that serve as biomarkers of endometrial dating or timing. Notably, 45 genes overlapped in these 2 endometrial factors. Of the 281 endometrial samples collected, we excluded 56 samples for having low-quality or insufficient RNA; 6 samples containing a low number of sequencing reads (Supplemental Fig. 3A); and 2 samples that had low gene expression after normalization ( Supplemental Fig. 3B ). Thus, 217 patients were considered in the downstream analysis. After correcting batch effects from sequencing runs (Supplemental Fig. 3C) and removing the transcriptomic variation due to endometrial cyclic changes (endometrial luteal phase timing;Supplemental Fig. 3D), there were no other confounding technical or clinical factors (Supplemental Fig. 4). A genomic-driven endometrial taxonomy for patient stratification on the basis of a novel endometrial disruption independent of endometrial timing Forty-four patients were initially categorized as acute RIF on the basis of the inability to carry a pregnancy to term after 3- 9 SETs. Alternatively, 40 patients were categorized as acute Fertile because they reached full-term pregnancy at thefirst SET after biopsy. Notably, after this clinical acute criteria, 133 (61.3%) patients remained unclassified for having insufficient attempts (Supplemental Fig. 5A), and no transcriptomic differences were found between the acute RIF and acute Fertile groups (the lowest FDR was 0.3027). Incorporation of transcriptomic data with this initial acute clinical classification allowed us to stratify patients with poor (n1⁄4 137, 63.1%) or good (n1⁄4 49, 22.6%) endometrial prognoses; leaving only 31 samples (14.3%) unclassified because of the lack of recognition by the algorithms (Supplemental Fig. 5B). EFR biomarker signature identification and functional characterization A total of 122 genes were identified as significantly different (30.2% of all panel genes, FDR< 0.05) between patients with good and poor endometrial prognoses, including 63 downregulated and 59 upregulated in patients with poor endometrial prognosis (Table 1 and Fig. 1A), and proposed as the EFR biomarker signature. Transcriptomically, endometrial biopsies clustered by prognosis (Fig. 1B), and the signature genes had functions that were mainly related to regulation (17%); metabolism (8.4%); the immune system and inflammation (7.8%); protein processes (7.3%), signaling (6.9%), transport (6.1%), and the cell cycle, proliferation, and differentiation (5.3%) (Fig. 1C andSupplemental Table 2). The poor prognosis profile was distinguished by a downregulation of 27/39 (69.2%) functional groups. We highlight the functions with higher proportions of downregulated genes as steroidogenesis and steroid signaling (77.8%), hormone-related (80.0%), the nervous system (81.3%), gametogenesis and reproduction (85.7%), and oxidative stress and response to O 2 (88.9%), as well as cell action potential and polarization (100%) (Fig. 1D). Few functional groups, including cell migration, extracellular matrix, and glucose homeostasis contained a larger proportion of upregulated genes (60.9%, 62.5%, and 62.5% upregulated genes, respectively;Fig. 1D). TABLE 1 Endometrial Failure Risk (EFR) signature’s genes. Gene ID Gene name FC FDR PAEP Progestagen Associated Endometrial Protein 3.1873 9.12E-07 GPR110 Adhesion G Protein-Coupled Receptor F1 2.3304 3.68E-05 CHST1 Carbohydrate Sulfotransferase 1 2.0292 0.0004 POSTN Periostin 1.8936 0.0009 BCL2L10 BCL2 Like 10 1.8549 0.0001 TSPAN8 Tetraspanin 8 1.8098 0.0179 CYP3A5 Cytochrome P450 Family 3 Subfamily A Member 5 1.7966 0.0092 SFRP4 Secreted Frizzled Related Protein 4 1.7778 1.23E-06 BAI2 Adhesion G Protein-Coupled Receptor B2 1.7138 0.0018 ITGA8 Integrin Subunit Alpha 8 1.7033 0.0005 PMEPA1 Prostate Transmembrane Protein, Androgen Induced 1 1.7026 3.35E-07 CLU Clusterin 1.6771 3.68E-05 MUC16 Mucin 16, Cell Surface Associated 1.6288 0.0001 MYH7B Myosin Heavy Chain 7B 1.6034 0.0116 SCGB2A2 Secretoglobin Family 2A Member 2 1.5859 0.0326 STC1 Stannioclacin 1 1.5800 0.0033 FOXP1 Forkhead Box P1 1.5332 3.94E-13 C10orf10 DEPP1 Autophagy Regulator 1.5093 0.0004 LRRC17 Leucine Rich Repeat Containing 17 1.5020 0.0001 SERTAD4 SERTA Domain Containing 4 1.4805 7.63E-07 COL16A1 Collagen Type XVI Alpha 1 Chain 1.4786 0.0024 C1orf133 SERTAD4 Antisense RNA 1 1.4381 0.0065 MAPK8IP3 Mitogen-Activated Protein Kinase 8 Interacting Protein 3 1.4275 0.0003 ABCC3 ATP Binding Cassette Subfamily C Member 3 1.4181 0.0124 MFAP5 Micro fibril Associated Protein 5 1.4030 0.0033 MFAP2 Micro fibril Associated Protein 2 1.3990 0.0005 GPX3 Glutathione Peroxidase 3 1.3960 0.0198 EDN3 Endothelin 3 1.3627 0.0053 SORCS1 Sortilin Related VPS10 Domain Containing Receptor 1 1.3563 0.0322 DUSP2 Dual Speci ficity Phosphatase 2 1.3545 0.0264 SLPI Secretory Leukocyte Peptidase Inhibitor 1.3463 0.0197 BICD1 BICD Cargo Adaptor 1 1.3427 1.99E-05 CRABP2 Cellular Retinoic Acid Binding Protein 2 1.3395 0.0040 ARID5B AT-Rich Interaction Domain 5B 1.3247 1.31E-05 EVC EvC Ciliary Complex Subunit 1 1.3157 0.0011 C2CD4B C2 Calcium Dependent Domain Containing 4B 1.3091 0.0109 ANO1 Anoctamin 1 1.3014 0.0053 CLDN4 Claudin 4 1.2867 0.0216 IGF2 Insulin Like Growth Factor 2 1.2854 0.0134 SYNE2 Spectrin Repeat Containing Nuclear Envelope Protein 2 1.2756 5.71E-06 VAV3 Vav Guanine Nucleotide Exchange Factor 3 1.2644 0.0286 ITGA9 Integrin Subunit Alpha 9 1.2562 0.0084 OBFC2A Nucleic Acid Binding Protein 1 1.2266 0.0032 WEE1 WEE1 G2 Checkpoint Kinase 1.2190 6.54E-06 RAB40B RAB40B, Member RAS Oncogene Family 1.2100 0.0042 BCL6 BCL6 Transcription Repressor 1.1891 0.0084 ID4 Inhibitor Of DNA Binding 4, HLH Protein 1.1878 0.0156 CD55 CD55 Molecule (Cromer Blood Group) 1.1806 0.0493 LETM2 Leucine Zipper And EF-Hand Containing Transmembrane Protein 2 1.1754 0.0179 CYBRD1 Cytochrome B Reducer 1 1.1647 0.0116 LIMK2 LIM Domain Kinase 2 1.1614 0.0022 C14orf45 Basal Body Orientation Factor 1 1.1526 0.0053 C16orf70 Phagosome Assembly Factor 1 1.1514 0.0022 FOSL2 FOS Like 2, AP-1 Transcription Factor Subunit 1.1412 0.0193 SCAMP1 Secretory Carrier Membrane Protein 1 1.1296 0.0264 CTDSP2 CTD Small Phosphatase 2 1.1099 0.0166 RNF19A Ring Finger Protein 19A, RBR E3 Ubiquitin Protein Ligase 1.1058 0.0250 HMGN3 High Mobility Group Nucleosomal Binding Domain 3 1.0988 0.0286 ACTN1 Actinin Alpha 1 1.0983 0.0322 ENY2 ENY2 Transcription And Export Complex 2 Subunit -1.0932 0.0466 PSMB10 Proteasome 20S Subunit Beta 10 -1.1231 0.0053 ARHGDIA Rho GDP Dissociation Inhibitor Alpha -1.1245 0.0433 PRPF4 Pre-MRNA Processing Factor 4 -1.1397 0.0348 COPG1 COPI Coat Complex Subunit Gamma 1 -1.1433 0.0006 RACGAP1 Rac GTPase Activating Protein 1 -1.1496 0.0042 PNP Purine Nucleoside Phosphorylase -1.1519 0.0143 STEAP4 STEAP4 Metalloreductase -1.1550 0.0270 Diaz-Gimeno. Endometrial Failure Risk signature. Fertil Steril 2024. VOL. - NO. - / - 2024 5 Fertil Steril® Differences in clinical reproductive outcomes authenticated the EFR signature’s clinical relevance There were no significant differences in the age or BMI of patients predicted to have good/poor endometrial prognosis (all P>.05). Women predicted to have poor endometrial prognosis previously underwent significantly more SETs than those with good endometrial prognosis (2.8 vs. 1.6;P<.0001), however, there were no significant differences in embryo quality (P1⁄4.4679) between groups (Supplemental Table 3). The lack TABLE 1 Continued. Gene ID Gene name FC FDR BDH1 3-Hydroxybutyrate Dehydrogenase 1 -1.1744 0.0030 ECI2 Enoyl-CoA Delta Isomerase 2 -1.1862 0.0178 ANK3 Ankyrin 3 -1.2084 0.0433 IL15 Interleukin 15 -1.2233 0.0198 GSN Gelsolin -1.2241 0.0076 GLIPR1 GLI Pathogenesis Related 1 -1.2251 0.0166 CBR3 Carbonyl Reductase 3 -1.2252 0.0124 ATP6V0E2 ATPase H þ Transporting V0 Subunit E2 -1.2304 0.0082 DYNLT3 Dynein Light Chain Tctex-Type 3 -1.2323 0.0199 S100A4 S100 Calcium Binding Protein A4 -1.2478 0.0041 BUB1B BUB1 Mitotic Checkpoint Serine/Threonine Kinase B -1.2482 0.0451 IDH1 Isocitrate Dehydrogenase (NADP( þ)) 1 -1.2488 0.0124 PLA2G16 Phospholipase A And Acyltransferase 3 -1.2498 0.0124 EPHB3 EPH Receptor B3 -1.2530 0.0396 OAZ3 Ornithine Decarboxylase Antizyme 3 -1.2581 0.0013 PYCR1 Pyrroline-5-Carboxylate Reductase 1 -1.2597 0.0007 FAM155B NALCN Channel Auxiliary Factor 2 -1.2618 0.0116 ECHS1 Enoyl-CoA Hydratase, Short Chain 1 -1.2621 3.68E-05 CSRP2 Cysteine And Glycine Rich Protein 2 -1.2647 0.0023 TLR4 Toll Like Receptor 4 -1.2716 0.0065 CTSW Cathepsin W -1.2764 0.0322 CEP55 Centrosomal Protein 55 -1.3021 0.0264 BARD1 BRCA1 Associated RING Domain 1 -1.3064 0.0014 PLA2G4A Phospholipase A2 Group IVA -1.3101 0.0008 ATP1B1 ATPase Na þ/Kþ Transporting Subunit Beta 1 -1.3202 0.0113 FOXO1 Forkhead Box O1 -1.3330 0.0018 BIRC3 Baculoviral IAP Repeat Containing 3 -1.3496 0.0007 CRISP3 Cysteine Rich Secretory Protein 3 -1.3498 0.0397 LMCD1 LIM And Cysteine Rich Domains 1 -1.3627 0.0012 PSMB8 Proteasome 20S Subunit Beta 8 -1.3783 3.68E-05 CD81 CD81 Molecule -1.3888 0.0104 ALDH1A3 Aldehyde Dehydrogenase 1 Family Member A3 -1.3891 0.0007 SQLE Squalene Epoxidase -1.3905 0.0001 KMO Kynurenine 3-Monooxygenase -1.4107 0.0434 RPRM Reprimo, TP53 Dependent G2 Arrest Mediator Homolog -1.4121 0.0061 SPP1 Secreted Phosphoprotein 1 -1.4130 0.0242 ANGPTL1 Angiopoietin Like 1 -1.4381 0.0017 HPRT1 Hypoxanthine Phosphoribosyltransferase 1 -1.4394 1.43E-05 ATP6V1A ATPase H þ Transporting V1 Subunit A -1.4484 0.0040 CDK1 Cyclin Dependent Kinase 1 -1.4651 0.0084 ENPEP Glutamyl Aminopeptidase -1.4704 0.0441 TH Tyrosine Hydroxylase -1.5149 0.0113 G0S2 G0/G1 Switch 2 -1.5183 0.0226 PROS1 Protein S -1.5297 0.0025 STAR Steroidogenic Acute Regulatory Protein -1.5342 0.0005 DKK1 Dickkopf WNT Signaling Pathway Inhibitor 1 -1.5425 0.0005 BANK1 B Cell Scaffold Protein With Ankyrin Repeats 1 -1.5553 0.0005 KCNJ2 Potassium Inwardly Rectifying Channel Subfamily J Member 2 -1.6602 0.0119 SLC7A2 Solute Carrier Family 7 Member 2 -1.6919 0.0038 PROK1 Prokineticin 1 -1.7218 0.0106 OLFM4 Olfactomedin 4 -1.7759 0.0458 IFNG Interferon Gamma -1.9666 0.0033 CXCL13 C-X-C Motif Chemokine Ligand 13 -2.0690 0.0001 SLC16A6 Solute Carrier Family 16 Member 6 -2.2113 9.12E-07 FGB Fibrinogen Beta Chain -2.3652 0.0108 Complete list of the 122 genes included in the endometrial failure risk (EFR) signature. Gene name and ID, fold change (FC) related to poor prognosis, and adjustedP value by false discovery rate (FDR) are indicated. Diaz-Gimeno. Endometrial Failure Risk signature. Fertil Steril 2024. 6 VOL. - NO. - / - 2024 REPRODUCTIVE SCIENCE of differences in the proportion of out-of-phase endometria (57% in poor vs. 47% in good,P1⁄4.25) between good and poor prognosis groups (Supplemental Table 3) distinguished this signature from established endometrial luteal phase timing phenotypes (displaced or on-time). Therefore, the significant differences (P<.0001) between the reproductive outcomes of women predicted to have poor and good endometrial prognoses validated the clinical relevance of our transcriptomic signature. Specifically, the poor endometrial prognosis was associated with a 34.5% decline in PR FIGURE 1 Transcriptomic and functional characterization of the endometrial failure risk (EFR) signature. (A) Volcano plot showing the 122 EFR signature genes. Red and blue dots were used to differentiate EFR genes that were significantly upregulated or downregulated in poor prognosis samples, respectively, whereas gray dots represented genes (n1⁄4 282) whose expression was not significantly different (FDR> 0.05) between poor and good prognoses groups and were not included in EFR signature. (B) Principal component analysis highlighting the transcriptomic behavior of endometrial samples from patients predicted to have poor (orange) or good (green) endometrial prognosis on the basis of genes included in EFR signature. Notably, the transcriptomic variance explained by the first and second principal components (PC1 and PC2, respectively) reached 23.1%. (C) Pie chart depicting the overall proportion of EFR signature genes included in each functional category. (D) The EFR signature genes were classified into 39 functional groups. Red and blue bars were used to differentiate genes that were significantly upregulated or downregulated in poor prognosis samples, respectively. FDR1⁄4 false discovery rate. Figure created withBioRender.com. Diaz-Gimeno. Endometrial Failure Risk signature. Fertil Steril 2024. VOL. - NO. - / - 2024 7 Fertil Steril® (P1⁄43.8E-05), a 52% reduction in LBR (P1⁄45E-10), a 19.6% rise in CMR (P1⁄4.0066), and 20.4% increase in BMR (P1⁄4.0023) (Fig. 2A). Furthermore, patients with a poor endometrial prognosis only achieved a CPR of 42.3% after the ninth SET, whereas patients with a good prognosis had an 83.7% CPR at thefifth attempt (Fig. 2B). The relative risk of implantation failure or miscarriage in women predicted to have poor endometrial prognosis was respectively 2.7 and 11.5 compared with women with good endometrial prognosis. However, when failures in implantation or ongoing pregnancy were considered together, the relative risk of endometrial failure (not achieving a live birth in the next SET) for patients with poor endometrial prognosis was 3.3. EFR signature biomarker’s predictive capability The 186 samples labeled with poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognoses were used to evaluate the signature’s predictive performance. The 122 differentially expressed genes between poor/good endometrial prognoses proposed as the EFR biomarker signature were employed to train the model. Cross-validation yielded a median accuracy of 0.92 (min1⁄4 0.88, max 1⁄4 0.94), sensitivity of 0.96 (min1⁄4 0.91, max 1⁄4 0.98), and specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88) (Fig. 2C). Notably, 163 of 186 (87.6%) samples were classified correctly in almost 90% of all cross-validation models, whereas 151 of 186 (81.2%) samples were well classified correctly in 100% of the iterations. Twelve and eleven samples were misclassified consistently, in <50% or 50%- 90% of models, respectively (Fig. 2D). Clinical algorithm results after evaluating both WOI factors Next, we grouped samples according to both potential causes of implantation failure, endometrial luteal phase timing (displaced or on-time) (Fig. 3A), and endometrial disruption (good or poor prognosis) (Fig. 2A) and compared the patients’ reproductive outcomes. There were no significant differences between the patients with displaced and on-time WOI in terms of BMI (P1⁄4.8740), the average number of SETs (P1⁄4.0879), and embryo quality (P1⁄41). However, patients with a displaced WOI were signi ficantly older (40.4 vs. 38.3 years old; P1⁄4.0044) (Supplemental Table 4). Once patients were stratified by prognosis (endometrial disruption), the proportion of FIGURE 2 Clinical meaning and prediction capability of the EFR signature. (A) Clinical outcomes of the single embryo transfer after biopsy collection. Rates for pregnancy, live birth, and clinical and biochemical miscarriage were calculated as described in theSupplemental Methods. Statistical differences were evaluated using Fisher’s exact test. (B) Comparison of cumulative pregnancy rates. Notably, many patients predicted to have a good endometrial prognosis achieved pregnancy with a single transfer (63.3%) and reached the maximum number of pregnancies (83.7%) at the fifth transfer, whereas patients predicted to have poor endometrial prognosis were only able to achieve a cumulative pregnancy rate of 42.3% at the ninth transfer. (C) Boxplots for the EFR signature prediction parameters (sensitivity, specificity, and accuracy) were calculated over 100 iterations of fivefold stratified cross-validation. (D) Histogram showing the proportion of patients that were successfully classified during the aforementioned cross-validation procedure. Figure created withBioRender.com. Diaz-Gimeno. Endometrial Failure Risk signature. Fertil Steril 2024. 8 VOL. - NO. - / - 2024 REPRODUCTIVE SCIENCE patients with poor prognosis was similar (P1⁄4.2458) between displaced and on-time groups (Fig. 3B). Experimental validation of genes related to the EFR signature The expression of the top 4 differentially expressed genes in the poor prognosis group was validated by RT-qPCR. Progestagen-associated endometrial protein and G-coupled protein receptor 110, which were found upregulated by RNA-seq, were significantly overexpressed in the poor prognosis group, compared with the good prognosis group (fold change [FC]1⁄4 6.157 [P1⁄4.0288] and FC1⁄4 11.803 [P1⁄4.0350], respectively; Supplemental Figs. 6Aand B). Meanwhile, the expression of fibrinogen beta chain and solute carrier 16 member 6, which were found downregulated by RNA-seq, did not reach a statistically significant difference (FC1⁄4 1.582 [P1⁄4.2475] and FC1⁄4 2.311 [P1⁄4.4376], respectively;Supplemental Figs. 6Cand D).',\n",
       "  'Discussion': 'DISCUSSION Advances in the diagnosis and treatment of RIF have been limited by its elusive etiology, the discrepancies in clinical classification (mainly due to the lack of consensus over the minimum number of pregnancy attempts) , and the heterogeneous molecular profiles of affected patients . On the basis of current clinical practice, patients with RIF may be misclassified and/or receive inef ficient treatment . Thus, the aim of this study was to propose a novel gene expression signature that could identify endometrial disruptions independent of endometrial luteal phase timing. Using a semi-supervised learning AI strategy together with the endometrial gene expression obtained from our custom EFR panel, we stratified patients according to poor or good endometrial prognoses and demonstrated how this signature reliably predicted gene expression patterns that corresponded with significant differences in PR, LBR, CMR, and BMR. When considering both WOI factors (disruption and timing) in our study cohort, poor endometrial prognoses (disruption) affected 73.7% of patients, whereas WOI displacements only affected 54.3%; 41.9% of patients presented a displaced WOI and were predicted to have a poor endometrial prognosis, whereas only 12.4% of patients presented with a displaced WOI and were predicted to have a good endometrial prognosis. These results offer a possible explanation as to why other studies reported there was no clinical benefito f personalizing embryo transfers on the basis of transcriptomic dating of the endometrium . According to our clinical algorithm, only patients with a displaced WOI and a good endometrial prognosis (12.4% of our cohort) would be better off with this approach. Patients with poor endometrial prognosis (73.7% of our cohort) would have suboptimal reproductive outcomes even though their WOI was on-time. This highlights an unmet healthcare need for IVF patients with endometrial disruptions and underscores the importance of additional prospective studies to further characterize patients on the basis of these 2 molecular causes for trying tailored treatments. Contemplating endometrial luteal phase timing together with a prevalent disruption or poor endometrial prognosis shifts the paradigm in studies about the endometrium. Our innovative methodology based on transcriptomics and AI models led us to develop new endometrial biomarker taxonomies related to endometrial luteal phase timing and/or endometrial disruption . There were 30 genes overlapping between the previously published TED signature and the EFR signature presented herein, emphasizing their correlation despite the lower FC of biomarkers for endometrial disruption versus endometrial timing . AI algorithms offer powerful approaches for patient stratification, by combining high-throughput molecular data and clinical variables . Clinical translation of these approaches will require careful supervision of how the machine learning is initially trained . However, a semi-supervised learning methodology implementing accurate clinical diagnostic criteria and patient-specific genomic data to establish relevant molecular taxonomies will be indispensable for advancing precision medicine , especially within the context of endometrial-factor infertility. Predictive biomarkers can stratify patients with endometrial failure even if the number of their attempts was insufficient to meet clinical RIF classification criteria . To our knowledge, this is thefirst study to apply semi-supervised learning to the endometrium, conferring an advantage over previous models that were only trained with clinical classifications and did not leverage the underlying molecular heterogeneity of the condition . The lack of gene expression differences between RIF and control patients with our initial clinical classification reinforced findings from Macklon’s group and the molecular heterogeneity among patients with RIF . The EFR signature comprised 122 genes, including 44 genes newly associated with an endometrial gene expression signature and a single common gene with Koot’s pioneer endometrial timing-independent signature for endometrial pathology . Here, we only considered biopsies collected in the endometrial mid-secretory phase, when patients had received P4 for 82- 172 h. Although this signature is independent of endometrial luteal phase timing, further studies are required to confirm if the signature is detectable in other phases of the endometrial cycle. Progestagen-associated endometrial protein is well characterized in endometrial tissue and was the most upregulated gene (FC1⁄4 3.18) in the poor endometrial prognosis pro file. However, the levels we observed were 10-fold lower compared with previous studies where changes in endometrial timing were considered . Alternatively,fibrinogen beta chain, related to the plasma membrane and vesicle transport by secretory granules, was the most downregulated gene. This gene accumulated polymorphisms that hinder the production offibrinogen and its associated anti-inflammatory activity and was recently related to the dysregulated pathways in RIF . FOXP1, the most predictive gene, is a transcription factor mediated by estrogen that has established functions in regulation, leukocyte differentiation, and response to lipids and chemokines. This gene is associated with breast endometrial cancer , and endometriosis-related fibrosis , but was never included in an endometrial gene signature until now. With FOXP1 and the FOX transcription factor family being associated with the regulation of the menstrual cycle and implantation , this evidence reinforces our prior study and affirms that alterations in endometrial regulation are an underlying cause of endometrial failure. Our clinical algorithm considers the 2 WOI factors, endometrial timing and disruption, to stratify IVF patients on the basis of 4 possible phenotypes: displaced/on-time and poor/ good prognosis. This model is promising for preventive and precision medicine in endometrial-factor infertility, as it combines the robust prediction parameters of the TED model and the EFR biomarker signature’s predictive capability reported herein. In contrast to other transcriptomic-based endometrial dating tools , our bioinformatic methodology removes the molecular variation associated with the cyclic menstrual changes, providing deeper insights for identifying a new kind of disruption independent of endometrial luteal phase timing by the EFR signature. Furthermore, the EFR biomarker signature was developed on the basis of reproductive outcomes and effectively classifies patients predicted to have positive or negative reproductive outcomes in the first SET after biopsy collection. All these facts, point out to a more potential clinical utility than existing tools that only predict the WOI without showing significant differences in reproductive outcomes as have been reported and we have shown in ourfindings. Although our strategy appears promising, the classification presented herein needs further re finement before clinical implementation. Larger sample sizes and validation in independent samples are required to ensure model generalization . Healthcare professionals will need to be adequately trained to ensure technical reproducibility. Longitudinal assessments including several menstrual cycles in the same patient are recommended to confirm inter-cycle reproducibility and to gain a comprehensive understanding of the reliability of the EFR panel. Additional studies should assess the relationship between a poor endometrial prognosis and the presence of benign disorders that potentially compromise endometrial function and/or embryo implantation . Inspired by the MammaPrint risk signature for breast cancer , the EFR signature could become a nextgeneration tool used to classify patients on the basis of endometrial prognosis, to predict reproductive outcomes before undergoing IVF, potentially experiencing RIF and/or losing valuable embryos. Although treatments for patients with poor endometrial prognosis remain to be determined, clinically identifying these patients is the first step toward improving their care.',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1016/j.fertnstert.2024.03.015',\n",
       "  'Journal': 'Fertil Steril'},\n",
       " {'PaperTitle': 'Predicting risk of endometrial failure: a biomarker signature that identifies a novel disruption independent of endometrial timing in patients undergoing hormonal replacement cycles',\n",
       "  'Publication': '2024',\n",
       "  'Authors': 'Patricia Diaz-Gimeno, Ph.D., Patricia Sebastian-Leon, Ph.D., Katharina Spath, Ph.D., Diana Marti-Garcia, M.Sc., Josefa Maria Sanchez-Reyes, Ph.D., Maria del Carmen Vidal, Ph.D., Almudena Devesa-Peiro, Ph.D., Immaculada Sanchez-Ribas, Ph.D., Asunta Martinez-Martinez, M.Sc., Nuria Pellicer, M.D., Ph.D., Dagan Wells, Ph.D., and Antonio Pellicer, M.D., Ph.D.',\n",
       "  'Email': 'patricia.diaz@ivirma.com or patricia_diaz@iislafe.es',\n",
       "  'Abstract': 'Objective: To propose a new gene expression signature that identifies endometrial disruptions independent of endometrial luteal phase timing and predicts if patients are at risk of endometrial failure. Design: Multicentric, prospective study. Setting: Reproductive medicine research department in a public hospital affiliated with private fertility clinics and a reproductive genetics laboratory. Patients: Caucasian women (n1⁄4 281; 39.4/C6 4.8 years old with a body mass index of 22.9/C6 3.5 kg/m 2) undergoing hormone replacement therapy between July 2018 and July 2021. Endometrial samples from 217 patients met RNA quality criteria for signature discovery and analysis. Intervention(s): Endometrial biopsies collected in the mid-secretory phase. Main Outcome Measure(s):Endometrial luteal phase timing-corrected expression of 404 genes and reproductive outcomes of thefirst single embryo transfer (SET) after biopsy collection to identify prognostic biomarkers of endometrial failure. Results: Removal of endometrial timing variation from gene expression data allowed patients to be stratified into poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognosis groups on the basis of their clinical and transcriptomic profiles. Significant differences were found between endometrial prognosis groups in terms of reproductive rates: pregnancy (44.6% vs. 79.6%), live birth (25.6% vs. 77.6%), clinical miscarriage (22.2% vs. 2.6%), and biochemical miscarriage (20.4% vs. 0%). The relative risk of endometrial failure for patients predicted as a poor endometrial prognosis was 3.3 times higher than those with a good prognosis. The differences in gene expression between both profiles were proposed as a biomarker, coined the endometrial failure risk (EFR) signature. Poor prognosis profiles were characterized by 59 upregulated and 63 downregulated genes mainly involved in regulation (17.0%), metabolism (8.4%), immune response, and inflammation (7.8%). This EFR signature had a median accuracy of 0.92 (min1⁄40.88, max1⁄40.94), median sensitivity of 0.96 (min1⁄40.91, max 1⁄4 0.98), and median specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88), positioning itself as a promising biomarker for endometrial evaluation. Conclusion(s): The EFR signature revealed a novel endometrial disruption, independent of endometrial luteal phase timing, present in 73.7% of patients. This EFR signature stratified patients into 2 significantly distinct and clinically relevant prognosis profiles providing opportunities for personalized therapy. Nevertheless, further validations are needed before implementing this gene signature as an artificial intelligence (AI)-based tool to reduce the risk of patients experiencing endometrial failure. (Fertil Steril /C2102024;-:-- -. /C2112024 by American Society for Reproductive Medicine.)',\n",
       "  'Introduction': 'A key factor for reproductive success in assisted reproduction treatments is the status of the maternal endometrium during embryo implantation and fetal development. The endometrial cycle is reflected by the cyclic structural and functional changes of the endometrium across the menstrual cycle, particularly during the mid-secretory phase, to prepare for the‘‘window of implantation’’ (WOI) . After successful embryo implantation, the decidua (specialized layer of the endometrium) actively encapsulates the trophectoderm to support placentation and provide adequate vascularization for optimal fetal growth . Thus, approaches that predict and prevent endometrial-factor infertility could substantially improve assisted reproduction treatment outcomes by supporting the establishment and maintenance of pregnancy . There is a lack of consensus on the minimum number of implantation failures with good-quality embryos derived from ovum donation or with confirmed euploid karyotypes required to diagnose recurrent implantation failure (RIF) . The disparities in clinical classification and heterogeneous etiology of RIF confound its diagnosis and leave the exact prevalence unclear . As clinical symptoms cannot be prevented and do not provide sufficient criteria to stratify patients, patients might be misclassi fied and receive inefficient treatment. Indeed, most studies using clinical criteria for RIF identified no or few differentially expressed genes between samples of affected patients , reflecting the molecular heterogeneity of RIF. To overcome these healthcare challenges, high-throughput molecular technologies are combined with artificial intelligence (AI) algorithms to provide a deeper understanding of infertilityrelated pathogenesis . These molecular insights are then applied in genomics-driven precision medicine approaches to characterize complex diseases, and ultimately, improve patient diagnosis and prognosis . Employing genomics in reproductive medicine has the potential to go beyond treating the clinical manifestations of disease progression , by leveraging the patients’ heterogeneous molecular profiles for precise stratification and revealing new targets for preventive measures . In this regard, current diagnostic tools for endometrialfactor infertility use transcriptomics to date the endometrial biopsy in the mid-secretory phase, exceeding the accuracy and objectivity of traditional endometrial progression staging by Noyes’ histological criteria . However, the clinical utility of synchronizing the endometrium by adjusting the timing of progesterone (P4) administration on the basis of the patient’s transcriptomic endometrial dating profile remains controversial . Independent randomized clinical trials found no improvement in reproductive outcomes , indicating the inefficacy of this clinical intervention and highlighting the need for alternative treatments and/or a refined target patient population that bene fits from this strategy. Combining these precision medicine approaches with transcriptomics and AI algorithms; our group recently proposed a novel molecular taxonomy of RIF and a new clinical algorithm for endometrial-factor infertility that considered 2 molecular causes of implantation failure: a WOI displacement that indicates a dysregulated endometrial timing; and a WOI disruption that indicates an endometrial alteration independent of endometrial timing . Transcriptomic analysis revealed these 2 WOI phenotypes were presented exclusively or simultaneously and patients with RIF had significantly more WOI disruptions but a similar frequency of WOI displacements, compared with controls . Thesefindings suggested that WOI disruptions had stronger associations with implantation failure than WOI displacements and shifted the paradigm for the study and treatment of endometrialfactor infertility . In addition, this work by SebastianLeon et al. highlighted the importance of removing the transcriptomic variation due to cyclical endometrial tissue changes (ie, the effect of endometrial timing) that, if not corrected, could mask the significant disruptions in endometrial gene expression. However, given this hypothesis-driven research was conducted retrospectively and in silico with publicly available data, the reproductive outcomes of the cycle after the biopsy collection were not available . This clinical follow-up is required to discover biomarkers for WOI disruption in patients with a higher risk of endometrial failure. On the basis of these genomics-driven precision medicine strategies, this proof-of-concept study aimed to discover a biomarker panel that could identify endometrial disruption, independent of endometrial timing, to predict, and ultimately prevent, poor reproductive outcomes in in vitro fertilization (IVF) patients.',\n",
       "  'Methods': 'Ethics statement This study was approved by the Ethics Committee of the Instituto Valenciano de Infertilidad, Valencia, Spain (1810-FIVI-066-PD/1810-FIVI-048-PD). Written informed consent was obtained from all recruited patients (n1⁄4 281). Participants, collection of endometrial biopsies, and reproductive outcomes This prospective, multicentric study was conducted between July 2018 and July 2021 at 5 private fertility clinics in Spain. Participants (n1⁄4 281) were scheduled for endometrial evaluation due to medical indications orR2 implantation failures of idiopathic origin. All patients met the following inclusion criteria: 18- 50 years old, with a body mass index (BMI) of 19- 30 kg/m 2, undergoing hormone replacement therapy for single embryo transfer (SET), presenting an endometrial thickness >6.5 mm with trilaminar structure in proliferative phase before P4 administration, and serum levels of estradiol >100 pg/mL and P4<1 ng/mL on the 10th day of estrogen treatment . Endometrial biopsies were collected after an average of 114.5/C6 7.2 h of P4 (duration ranged from 82 to 172 h), and processed as we previously described . Baseline patient characteristics including age, BMI, the number of IVF attempts, the quality of the embryo transferred (defined in theSupplemental Methods, available online), and reproductive outcomes (ie, pregnancy, live birth, and biochemical and clinical miscarriage) for each attempt, were obtained from internal medical records. Information about IVF attempts in other fertility clinics was included if recorded (only in 47.7% of patients). Study design First, a custom RNA-seq gene panel (the endometrial failure risk [EFR] panel) comprising 404 biomarkers of endometrial luteal phase timing and/or potential disruption was designed and employed to determine the risk of endometrial failure among 217 IVF patients (Supplemental Fig. 1, step 1 available online). The endometrial timing-corrected expression of the EFR panel was used to study the differential expression associated with endometrial disruption together with clinical characteristics of IVF patients to stratify the population as having poor or good endometrial prognoses, according to a semi-supervised learning process (Supplemental Fig. 1, step 2). The distinct transcriptomic patterns of each prognosis group were associated with the differences in reproductive outcomes of SET after biopsy collection, and the differentially expressed gene signature was proposed as a biomarker of endometrial disruption (Supplemental Fig. 1, step 3). Finally, the gene expression signature was used to supervise machine learning algorithms to evaluate the biomarker capacity of this signature to distinguish endometrial competence. The predictive performance of the models was assessed through the accuracy, sensibility, and specificity over 100 iterations of fivefold strati fied cross-validation ( Supplemental Fig. 1 , step 4). Constructing the EFR gene panel Supplemental Figure 2illustrates the workflow for the EFR panel gene selection. A whole-transcriptome endometrial data set (GSE58144, n1⁄4 115) retrieved from Gene Expression Omnibus was processed to correct the endometrial progression effect, using linear models in limma (version 3.46.0) , and unmask genes related to endometrial disruption . These genes were ranked according to an informativity score (CorrelationAttributeEval) in Weka (version 3.8.2, December 22, 2017) to evaluate their worth by measuring their correlation with endometrial pathology. Gene sets with increasing gene signature size were developed by adding the most informative genes in a stepwise manner, and their predictive performance was analyzed using Support Vector Machine and Random Forest algorithms. After calculating predictive performance with a stratified fivefold cross-validation (default parameters; 80:20; 100 iterations) using the Weka R package , the optimal signature size was determined for each algorithm, as the one with the fewest genes and highest accuracy. Finally, genes we previously prioritized for transcriptomic endometrial dating w e r ea l s oi n c l u d e di nt h e EFR panel to be able to remove the endometrial luteal phase timing variation. Endometrial biopsy processing, sequencing, and transcriptomic preprocessing Total RNA was extracted, and only samples that met our quality criteria were sequenced using the TruSeq Targeted RNA Illumina protocol and our custom EFR panel, in the Illumina NextSeq 550 platform with a paired-end design of 150 cycles and 4 M reads/sample, as previously described . Raw transcriptomic data was processed and normalized using limma, to filter low-quality raw counts (Q< 30) and low-read samples (<2M ). Possible batch effects were detected using principal components analysis and corrected with linear models in limma. Finally, endometrial luteal phase timing effects were detected using our transcriptomic endometrial dating (TED) model (see Supplemental Methods) and removed using limma. EFR biomarker signature identification A preliminary classification of patients was based on the patients’ history of reproductive outcomes and transcriptomic patterns corrected for endometrial luteal phase timing. This approach eliminated the gene expression variation related to the cyclic tissue changes that mask endometrial disruption and was used before applying semi-supervised machine learning on the basis of self-labeled techniques (see Supplemental Methods). Once, the genomic-driven disease taxonomy was established, the differentially expressed genes obtained between the final good and poor prognosis groups were proposed as the EFR biomarker signature. The limma R package was used to identify the statistically signi ficant differentially expressed genes (false discovery rate [FDR]-adjustedP value <.05). Functional characterization of the EFR signature The biomarker signature genes were annotated for functional interpretation databases by consulting Kyoto Encyclopaedia of Genes and Genomes human pathways (release 99.0, August 1, 2021) and Gene Ontology (version July 2, 2021; employing only experimental annotations) . Functional terms were manually summarized into categories according to their description in the database and presented as a bar plot using ggplot2 . EFR signature capability as a biomarker of endometrial disruption To estimate the prediction capability of the EFR signature to distinguish poor and good endometrial prognosis profiles, we implemented predictive models on the basis of the Support Vector Machine algorithm. Using a stratified fivefold crossvalidation process (repeated 100 times), the accuracy, sensitivity, and specificity were estimated. The range of values obtained across the 100 iterations was presented as a boxplot using ggplot2. Statistical analysis Rates for reproductive outcomes (ie, pregnancy rate [PR], cumulative pregnancy rate [CPR], live birth rate [LBR], biochemical miscarriage rate [BMR], and clinical miscarriage rate [CMR]) were calculated as de fined in the Supplemental Methods. Descriptive statistics were used for transcriptomic and clinical characteristics of patients to compare prognosis or WOI displacement groups. Continuous variables were presented as an overall mean /C6 standard deviation whereas discrete variables were presented as counts and percentages. Groups were compared using the Wilcoxon rank-sum test for continuous variables and Fisher’s exact test for discrete variables. All statistical analyses were conducted in R. All graphical results were generated with ggplot2. Gene expression validation by quantitative realtime polymerase chain reaction Quantitative real-time polymerase chain reaction (RT-qPCR) was performed to validate the expression of the top 4 differentially expressed genes between patients with poor (n1⁄4 10) and good (n1⁄4 10) endometrial prognoses. RNA (200- 220 ng) was reverse transcribed into cDNA using the PrimeScript Reagent Kit (Perfect Real Time, Takara, Japan) on a Thermocycler T3000 (Biometra, Ireland). Quantitative real-time polymerase chain reaction was performed on a StepOnePlus Real-Time PCR System (Applied Biosystems, USA) using Power-Up SYBR Green (Thermo Fisher Scientific, USA) and standard conditions.Supplemental Table 1(available online) lists the specific primers (Invitrogen, Thermo Fisher Scientific) used. Universal human RNA (Agilent, Spain) and water were respectively included as positive and negative controls. Each sample was analyzed in duplicate and relative gene expression (normalized to beta actin) was calculated using the DDCt method .',\n",
       "  'Results': 'Preliminary sequencing with the 404-gene EFR panel The EFR panel evaluated 404 genes: 139 that predict endometrial disruption and 310 that serve as biomarkers of endometrial dating or timing. Notably, 45 genes overlapped in these 2 endometrial factors. Of the 281 endometrial samples collected, we excluded 56 samples for having low-quality or insufficient RNA; 6 samples containing a low number of sequencing reads (Supplemental Fig. 3A); and 2 samples that had low gene expression after normalization ( Supplemental Fig. 3B ). Thus, 217 patients were considered in the downstream analysis. After correcting batch effects from sequencing runs (Supplemental Fig. 3C) and removing the transcriptomic variation due to endometrial cyclic changes (endometrial luteal phase timing;Supplemental Fig. 3D), there were no other confounding technical or clinical factors (Supplemental Fig. 4). A genomic-driven endometrial taxonomy for patient stratification on the basis of a novel endometrial disruption independent of endometrial timing Forty-four patients were initially categorized as acute RIF on the basis of the inability to carry a pregnancy to term after 3- 9 SETs. Alternatively, 40 patients were categorized as acute Fertile because they reached full-term pregnancy at thefirst SET after biopsy. Notably, after this clinical acute criteria, 133 (61.3%) patients remained unclassified for having insufficient attempts (Supplemental Fig. 5A), and no transcriptomic differences were found between the acute RIF and acute Fertile groups (the lowest FDR was 0.3027). Incorporation of transcriptomic data with this initial acute clinical classification allowed us to stratify patients with poor (n1⁄4 137, 63.1%) or good (n1⁄4 49, 22.6%) endometrial prognoses; leaving only 31 samples (14.3%) unclassified because of the lack of recognition by the algorithms (Supplemental Fig. 5B). EFR biomarker signature identification and functional characterization A total of 122 genes were identified as significantly different (30.2% of all panel genes, FDR< 0.05) between patients with good and poor endometrial prognoses, including 63 downregulated and 59 upregulated in patients with poor endometrial prognosis (Table 1 and Fig. 1A), and proposed as the EFR biomarker signature. Transcriptomically, endometrial biopsies clustered by prognosis (Fig. 1B), and the signature genes had functions that were mainly related to regulation (17%); metabolism (8.4%); the immune system and inflammation (7.8%); protein processes (7.3%), signaling (6.9%), transport (6.1%), and the cell cycle, proliferation, and differentiation (5.3%) (Fig. 1C andSupplemental Table 2). The poor prognosis profile was distinguished by a downregulation of 27/39 (69.2%) functional groups. We highlight the functions with higher proportions of downregulated genes as steroidogenesis and steroid signaling (77.8%), hormone-related (80.0%), the nervous system (81.3%), gametogenesis and reproduction (85.7%), and oxidative stress and response to O 2 (88.9%), as well as cell action potential and polarization (100%) (Fig. 1D). Few functional groups, including cell migration, extracellular matrix, and glucose homeostasis contained a larger proportion of upregulated genes (60.9%, 62.5%, and 62.5% upregulated genes, respectively;Fig. 1D). Differences in clinical reproductive outcomes authenticated the EFR signature’s clinical relevance There were no significant differences in the age or BMI of patients predicted to have good/poor endometrial prognosis (all P>.05). Women predicted to have poor endometrial prognosis previously underwent significantly more SETs than those with good endometrial prognosis (2.8 vs. 1.6;P<.0001), however, there were no significant differences in embryo quality (P1⁄4.4679) between groups (Supplemental Table 3). The lack of differences in the proportion of out-of-phase endometria (57% in poor vs. 47% in good,P1⁄4.25) between good and poor prognosis groups (Supplemental Table 3) distinguished this signature from established endometrial luteal phase timing phenotypes (displaced or on-time). Therefore, the significant differences (P<.0001) between the reproductive outcomes of women predicted to have poor and good endometrial prognoses validated the clinical relevance of our transcriptomic signature. Specifically, the poor endometrial prognosis was associated with a 34.5% decline in PR (P1⁄43.8E-05), a 52% reduction in LBR (P1⁄45E-10), a 19.6% rise in CMR (P1⁄4.0066), and 20.4% increase in BMR (P1⁄4.0023) (Fig. 2A). Furthermore, patients with a poor endometrial prognosis only achieved a CPR of 42.3% after the ninth SET, whereas patients with a good prognosis had an 83.7% CPR at thefifth attempt (Fig. 2B). The relative risk of implantation failure or miscarriage in women predicted to have poor endometrial prognosis was respectively 2.7 and 11.5 compared with women with good endometrial prognosis. However, when failures in implantation or ongoing pregnancy were considered together, the relative risk of endometrial failure (not achieving a live birth in the next SET) for patients with poor endometrial prognosis was 3.3. EFR signature biomarker’s predictive capability The 186 samples labeled with poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognoses were used to evaluate the signature’s predictive performance. The 122 differentially expressed genes between poor/good endometrial prognoses proposed as the EFR biomarker signature were employed to train the model. Cross-validation yielded a median accuracy of 0.92 (min1⁄4 0.88, max 1⁄4 0.94), sensitivity of 0.96 (min1⁄4 0.91, max 1⁄4 0.98), and specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88) (Fig. 2C). Notably, 163 of 186 (87.6%) samples were classified correctly in almost 90% of all cross-validation models, whereas 151 of 186 (81.2%) samples were well classified correctly in 100% of the iterations. Twelve and eleven samples were misclassified consistently, in <50% or 50%- 90% of models, respectively (Fig. 2D). Clinical algorithm results after evaluating both WOI factors Next, we grouped samples according to both potential causes of implantation failure, endometrial luteal phase timing (displaced or on-time) (Fig. 3A), and endometrial disruption (good or poor prognosis) (Fig. 2A) and compared the patients’ reproductive outcomes. There were no significant differences between the patients with displaced and on-time WOI in terms of BMI (P1⁄4.8740), the average number of SETs (P1⁄4.0879), and embryo quality (P1⁄41). However, patients with a displaced WOI were signi ficantly older (40.4 vs. 38.3 years old; P1⁄4.0044) (Supplemental Table 4). Once patients were stratified by prognosis (endometrial disruption), the proportion of patients with poor prognosis was similar (P1⁄4.2458) between displaced and on-time groups (Fig. 3B). Experimental validation of genes related to the EFR signature The expression of the top 4 differentially expressed genes in the poor prognosis group was validated by RT-qPCR. Progestagen-associated endometrial protein and G-coupled protein receptor 110, which were found upregulated by RNA-seq, were significantly overexpressed in the poor prognosis group, compared with the good prognosis group (fold change [FC]1⁄4 6.157 [P1⁄4.0288] and FC1⁄4 11.803 [P1⁄4.0350], respectively; Supplemental Figs. 6Aand B). Meanwhile, the expression of fibrinogen beta chain and solute carrier 16 member 6, which were found downregulated by RNA-seq, did not reach a statistically significant difference (FC1⁄4 1.582 [P1⁄4.2475] and FC1⁄4 2.311 [P1⁄4.4376], respectively;Supplemental Figs. 6Cand D).',\n",
       "  'Discussion': 'Advances in the diagnosis and treatment of RIF have been limited by its elusive etiology, the discrepancies in clinical classification (mainly due to the lack of consensus over the minimum number of pregnancy attempts) , and the heterogeneous molecular profiles of affected patients . On the basis of current clinical practice, patients with RIF may be misclassified and/or receive inef ficient treatment . Thus, the aim of this study was to propose a novel gene expression signature that could identify endometrial disruptions independent of endometrial luteal phase timing. Using a semi-supervised learning AI strategy together with the endometrial gene expression obtained from our custom EFR panel, we stratified patients according to poor or good endometrial prognoses and demonstrated how this signature reliably predicted gene expression patterns that corresponded with significant differences in PR, LBR, CMR, and BMR. When considering both WOI factors (disruption and timing) in our study cohort, poor endometrial prognoses (disruption) affected 73.7% of patients, whereas WOI displacements only affected 54.3%; 41.9% of patients presented a displaced WOI and were predicted to have a poor endometrial prognosis, whereas only 12.4% of patients presented with a displaced WOI and were predicted to have a good endometrial prognosis. These results offer a possible explanation as to why other studies reported there was no clinical benefito f personalizing embryo transfers on the basis of transcriptomic dating of the endometrium . According to our clinical algorithm, only patients with a displaced WOI and a good endometrial prognosis (12.4% of our cohort) would be better off with this approach. Patients with poor endometrial prognosis (73.7% of our cohort) would have suboptimal reproductive outcomes even though their WOI was on-time. This highlights an unmet healthcare need for IVF patients with endometrial disruptions and underscores the importance of additional prospective studies to further characterize patients on the basis of these 2 molecular causes for trying tailored treatments. Contemplating endometrial luteal phase timing together with a prevalent disruption or poor endometrial prognosis shifts the paradigm in studies about the endometrium. Our innovative methodology based on transcriptomics and AI models led us to develop new endometrial biomarker taxonomies related to endometrial luteal phase timing and/or endometrial disruption . There were 30 genes overlapping between the previously published TED signature and the EFR signature presented herein, emphasizing their correlation despite the lower FC of biomarkers for endometrial disruption versus endometrial timing . AI algorithms offer powerful approaches for patient stratification, by combining high-throughput molecular data and clinical variables . Clinical translation of these approaches will require careful supervision of how the machine learning is initially trained . However, a semi-supervised learning methodology implementing accurate clinical diagnostic criteria and patient-specific genomic data to establish relevant molecular taxonomies will be indispensable for advancing precision medicine , especially within the context of endometrial-factor infertility. Predictive biomarkers can stratify patients with endometrial failure even if the number of their attempts was insufficient to meet clinical RIF classification criteria . To our knowledge, this is thefirst study to apply semi-supervised learning to the endometrium, conferring an advantage over previous models that were only trained with clinical classifications and did not leverage the underlying molecular heterogeneity of the condition . The lack of gene expression differences between RIF and control patients with our initial clinical classification reinforced findings from Macklon’s group and the molecular heterogeneity among patients with RIF . The EFR signature comprised 122 genes, including 44 genes newly associated with an endometrial gene expression signature and a single common gene with Koot’s pioneer endometrial timing-independent signature for endometrial pathology . Here, we only considered biopsies collected in the endometrial mid-secretory phase, when patients had received P4 for 82- 172 h. Although this signature is independent of endometrial luteal phase timing, further studies are required to confirm if the signature is detectable in other phases of the endometrial cycle. Progestagen-associated endometrial protein is well characterized in endometrial tissue and was the most upregulated gene (FC1⁄4 3.18) in the poor endometrial prognosis pro file. However, the levels we observed were 10-fold lower compared with previous studies where changes in endometrial timing were considered . Alternatively,fibrinogen beta chain, related to the plasma membrane and vesicle transport by secretory granules, was the most downregulated gene. This gene accumulated polymorphisms that hinder the production offibrinogen and its associated anti-inflammatory activity and was recently related to the dysregulated pathways in RIF . FOXP1, the most predictive gene, is a transcription factor mediated by estrogen that has established functions in regulation, leukocyte differentiation, and response to lipids and chemokines. This gene is associated with breast endometrial cancer , and endometriosis-related fibrosis , but was never included in an endometrial gene signature until now. With FOXP1 and the FOX transcription factor family being associated with the regulation of the menstrual cycle and implantation , this evidence reinforces our prior study and affirms that alterations in endometrial regulation are an underlying cause of endometrial failure. Our clinical algorithm considers the 2 WOI factors, endometrial timing and disruption, to stratify IVF patients on the basis of 4 possible phenotypes: displaced/on-time and poor/ good prognosis. This model is promising for preventive and precision medicine in endometrial-factor infertility, as it combines the robust prediction parameters of the TED model and the EFR biomarker signature’s predictive capability reported herein. In contrast to other transcriptomic-based endometrial dating tools , our bioinformatic methodology removes the molecular variation associated with the cyclic menstrual changes, providing deeper insights for identifying a new kind of disruption independent of endometrial luteal phase timing by the EFR signature. Furthermore, the EFR biomarker signature was developed on the basis of reproductive outcomes and effectively classifies patients predicted to have positive or negative reproductive outcomes in the first SET after biopsy collection. All these facts, point out to a more potential clinical utility than existing tools that only predict the WOI without showing significant differences in reproductive outcomes as have been reported and we have shown in ourfindings. Although our strategy appears promising, the classification presented herein needs further re finement before clinical implementation. Larger sample sizes and validation in independent samples are required to ensure model generalization . Healthcare professionals will need to be adequately trained to ensure technical reproducibility. Longitudinal assessments including several menstrual cycles in the same patient are recommended to confirm inter-cycle reproducibility and to gain a comprehensive understanding of the reliability of the EFR panel. Additional studies should assess the relationship between a poor endometrial prognosis and the presence of benign disorders that potentially compromise endometrial function and/or embryo implantation . Inspired by the MammaPrint risk signature for breast cancer , the EFR signature could become a nextgeneration tool used to classify patients on the basis of endometrial prognosis, to predict reproductive outcomes before undergoing IVF, potentially experiencing RIF and/or losing valuable embryos. Although treatments for patients with poor endometrial prognosis remain to be determined, clinically identifying these patients is the first step toward improving their care.',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1016/j.fertnstert.2024.03.015',\n",
       "  'Journal': 'Fertil Steril®'},\n",
       " {'PaperTitle': 'The animal in the genome: comparative genomics and evolution',\n",
       "  'Publication': '2008',\n",
       "  'Authors': 'Richard R Copley',\n",
       "  'Email': 'copley@well.ox.ac.uk',\n",
       "  'Abstract': 'Comparisons between completely sequenced metazoan genomes have generally emphasized how similar their encoded protein content is, even when the comparison is between phyla. Given the manifest differences between phyla and, in particular, intuitive notions that some animals are more complex than others, this creates something of a paradox. Simplistic explanations have included arguments such as increased numbers of genes; greater numbers of protein products produced through alternative splicing; increased numbers of regulatory non-coding RNAs and increased complexity of the cis-regulatory code. An obvious value of complete genome sequences lies in their ability to provide us with inventories of such components. I examine progress being made in linking genome content to the pattern of animal evolution, and argue that the gap between genomic and phenotypic complexity can only be understood through the totality of interacting components.',\n",
       "  'Introduction': 'Complete genome sequences provide limits to our imaginations. Even just a few years before the human genome was available in rough draft form, it was widely believed to encode at least 50\\u200a000 genes (Fields et al. 1994; Nature Genetics Editorial 2000). In contrast, the initial publications estimated 25–40\\u200a000 protein-coding genes (Lander et al. 2001; Venter et al. 2001), and since then estimates have generally carried a downward momentum, most recently approaching 20\\u200a000 (Goodstadt & Ponting 2006; Pennisi 2007). Although this number is higher than 16\\u200a000 or so found in invertebrate chordates (Dehal et al. 2002), it is roughly the same total as the nematode worm Caenorhabditis elegans (Hillier et al. 2005). Whether or not these low numbers of protein-coding genes for vertebrates stand the test of time, the sense of unease surrounding the lack of correlation between organismal complexity (often measured in numbers of distinct cell types) and protein-coding gene count is evident from the framing of the ‘G-value paradox’ by Hahn & Wray (2002), and the various explanations that have been put forward to ease it, including, for example, miRNAs (Sempere et al. 2006), non-protein-coding DNA (Taft et al. 2007) and alternative splicing (Kim et al. 2007).\\nSimilar gene counts are, of course, a crude measure of biological complexity. There is no reason why two genomes should not encode very different sets of protein-coding genes, but still have similar overall totals. Within the field of animal evolution and the evolution of development (evo–devo), however, the G-value paradox has a particular resonance. Studies in different animal phyla have repeatedly shown the reuse of a core set of developmental genes, the so-called ‘toolkit’ (Carroll et al. 2005), with the HOX genes in particular taking on an iconic significance. Broadly, toolkit genes come from a handful of transcription factor families, defined by the presence of particular structural domains such as the helix-turn-helix (HTH), including the homeobox genes; zinc fingers (ZnFs); leucine zippers and the helix-loop-helix (HLH). As well as transcription factors, there are seven well-conserved pathways responsible for intercellular signalling (Pires-daSilva & Sommer 2003), many of which appear to be present in sponges, the earliest branching clade of animals (Nichols et al. 2006). An extreme interpretation of these data is provided by Davidson (2006): ‘if we focus explicitly on the genes encoding transcription factors, and […] signalling systems required for developmental spatial regulation, there is almost no qualitative variation among the genomes of bilaterians’.\\nGiven all this, where in the genome do the phenotypic differences between animal taxa arise? The undoubted conservation of the protein-coding developmental genes has, particularly in the evo–devo field with its morphological concerns, focused attention on cis-enhancer elements affecting transcription (Carroll et al. 2005; Davidson 2006; Simpson 2007; Wray 2007), although there are alternative views emphasizing the importance of different kinds of regulatory elements (Alonso & Wilkins 2005) and different protein classes, such as structural genes (Hoekstra & Coyne 2007). As well as the presence of particular genes, the role of gene loss, especially with regard to secondarily simplified organisms such as tunicates and nematodes, is also likely to be of major significance. Below I outline some major themes being developed by large-scale genome comparisons, principally of nematodes, insects and vertebrates. My aim is not to present an exhaustive account, but to highlight areas where functionally relevant species-specific differences may arise, within apparently conserved systems. Although I concentrate on the evolution of the systems regulating animal development, this is not to lose sight of the things being regulated: the proteins involved in making nematode cuticles, or asynchronous flight muscles in insects, or the human brain and adaptive immune system, to name but a few, are what make it necessary to evolve those systems.',\n",
       "  'Methods': '',\n",
       "  'Results': '2. Gene duplication\\nUsefully summarizing the differences and similarities between more than 10\\u200a000 protein-coding genes from several species at once is not necessarily straightforward. Although pairwise similarities between sequences are easy to compute, they suffer from the imposition of arbitrary cut-offs and are less easy to interpret than measures that explicitly reflect phylogeny. Genes in different species are most obviously compared by grouping into sets of orthologues (i.e. genes related by speciation events) and paralogues (genes related by intra-genome duplication events). Closely related species share large numbers of orthologues: 93% of dog (Canis familiaris) and 82% of the marsupial Monodelphis domesticus gene predictions have orthologues in human (Goodstadt et al. 2007). The Linnean hierarchy, however, is not necessarily a good guide of genomic relatedness by this definition of similarity. Within the nematodes, 65% of C. elegans genes share an orthologue with Caenorhabditis briggsae, despite their being from the same genus (Stein et al. 2003). For more distantly related genomes, orthologue counts can drop rapidly. This may be as much a sign of difficulties in reliably assigning gene orthology on a large scale, as a real indication of the extents of the conserved cores.\\nParalogues often arise via tandem duplication of genes, giving rise to localized clusters of functionally related genes. As these are the regions where gene content is evolving most rapidly between closely related species, the functions of these genes are of special interest for understanding animal-specific differences. For the most part, for any two closely related vertebrate genomes, the functional classes of genes duplicated in this way are similar—olfaction and chemosensation, reproduction and effectors of the immune response—although the duplications have occurred independently in each lineage (Emes et al. 2003). These large groups of paralogues often show evidence of adaptive evolution in their amino acid sequences, suggesting that new functions have been selected for (Emes et al. 2004a, b).\\nThe recurrent nature of duplications within particular functional classes, coupled with the observed diversifying selection suggests that they are a standard adaptive genomic response to environmental challenges. Does similar rapid duplication occur in the kinds of genes, such as transcription factors, that might be implicated in development? A growing number of examples are known. Perhaps most dramatically, in mice a set of 32 tandemly duplicated homeoboxes have arisen from apparently one or two genes in the common ancestor of humans and rodents; they are believed to play a role in germ cell development and embryonic stem cell differentiation (Maclean et al. 2005; Jackson et al. 2006).\\nZnF containing transcription factors have undergone independent rounds of gene duplication in insects and tetrapods. In insects a set of ZnFs is found to co-occur with a ZnF-associated domain (ZAD; Chung et al. 2007); this ZAD class is found in approximately 100 and 150 copies in Drosophila melanogaster and the mosquito Anopheles gambiae, respectively; there is only a single copy in vertebrates (Chung et al. 2007). In D. melanogaster, many are expressed in the female germ line, suggesting a role in oocyte development or embryogenesis (Chung et al. 2007). An analogous story is found with Krüppel-associated box (KRAB) containing Zn fingers in tetrapods. Successive independent tandem duplication events have occurred in different mammalian lineages, leading to over 400 copies in the human genome (Huntley et al. 2006). The KRAB domain itself appears to have been co-opted from a progenitor sequence conserved throughout eukaryotes (Birtle & Ponting 2006), however, it has evolved so much as to make this similarity difficult to detect; clearly identifiable KRAB domains are specific to tetrapods. Their functions are largely unknown, and have not been tied to any general aspects of tetrapod-specific biology. As such, why the family as a whole has expanded is a puzzle.\\nNematodes too exhibit lineage-specific expansions of particular transcription factor families, most notably, the nuclear hormone receptors (NHRs). The C. elegans genome encodes 284, far more than the 48 in human and 21 in D. melanogaster. The bulk of these (greater than 200) have arisen from an apparently nematode-specific expansion of a unique gene (Lander et al. 2001; Robinson-Rechavi et al. 2005). Once more, the reasons for such a dramatic lineage-specific expansion of a particular transcription factor family, and any links to taxon-specific biology, are obscure, although it has been speculated that C. elegans relies less on combinatorial reuse of different transcription factors (Antebi 2006). A less dramatic lineage-specific expansion occurs in the case of the T-box-containing transcription factors: there are 21 in C. elegans, with 17 arising from a lineage-specific expansion when compared with D. melanogaster and humans. Ascertaining when and in which taxa these duplications found in C. elegans took place is currently frustrated by a lack of nematode genome sequences—currently, only those of C. elegans and C. briggsae have been published. These T-box genes as a set map to several genomic locations, suggesting that they have arisen over a more protracted time scale than the examples discussed above; some, at least, have known roles in the development of C. elegans (Poole & Hobert 2006).\\n3. The invention of new genes\\nA number of gene families appear to be metazoan novelties, with no clear sequence similarity to other genes outside the Metazoa, but present in the more basal animal phyla, such as cnidarians and sponges. These include key families involved in animal development, such as T-box and SMAD transcription factors, and signalling molecules such as WNTs and fibroblast growth factors (FGFs; Putnam et al. 2007). Was the invention of such families a prerequisite for the evolution of the Metazoa, and were analogous protein inventions required for the evolution of particular taxa, such as insects and vertebrates? Analysis of three-dimensional structures (i.e. the protein fold itself) suggests a more subtle transition than large-scale evolution of new protein folds. In many cases, examination of protein three-dimensional structural similarities shows that these genes have distant homologues in non-metazoan genomes. The MH1 (DNA binding) domain of SMADs, for instance, is probably homologous to a family of homing endonucleases found in all kingdoms of life (Grishin 2001); the T-box shares structural similarities indicative of homology with a variety of other transcription factors, such as STAT DNA-binding domains, which are found in other eukaryotes (Murzin et al. 1995; Soler-Lopez et al. 2004); and the signalling domain of metazoan hedgehog proteins shares detailed similarities with members of a family of bacterial peptidases, suggesting that they too are likely to be homologous (Murzin et al. 1995). In these cases, the novel families are likely to be cases of rapid sequence evolution, accompanying functional shifts, within stem lineages leading to the Metazoa. Sparse sequence sampling of non-fungal and metazoan eukaryotic genomes may contribute to the apparent co-origin of these protein domains with the animals.\\nAs this type of domain evolution is occurring from pre-existing domain types, the process fits within a standard framework of accelerated point mutation and selection for new functions. The invention of the domain type is not a key innovation in itself; rather, it can be seen as the extension of functional diversification of subfamilies of the kind that is apparent when comparing more closely related species. The fact that so many new domain types are found to be coincident within the origin of metazoans suggests that the selective pressures giving rise to this kind of accelerated sequence evolution were greater in the metazoan stem lineage.\\nAn example of a more recent domain innovation is found in the Drosophila gene brinker, which plays a key role in the establishment of dorsoventral patterning. Although the protein-coding sequence of its DNA-binding domain is well conserved in insects, using current sequence databases it shows no significant sequence similarity to proteins from any other taxa (figure 1), although there is weak (non-significant) similarity to pogo-like transposases, and the structure, which is only folded when complexed with DNA, suggests similarity to various transcription factors (Cordier et al. 2006).\\n4. Evolution of transcription factors: the animal in the orthologue\\nLineage-specific duplication followed by sequence divergence provides one route to species-specific biology, but what scope is there for lineage-specific functional shifts within orthologous genes? In the absence of gene duplication, it is hard to imagine how the DNA specificity of a particular factor might be significantly changed in such a way that it targets new genes, without deleterious consequences. The modular structure of proteins, however, suggests that other routes of functional evolution are available. A protein may have pleiotropic effects, but that is not the same as saying that every amino acid in the protein will be directly involved in all those effects. A recent illustrative example from the hox gene Ultrabithorax, is of an insect-specific ‘QA’ protein motif, found outside the homeodomain. The region is involved in limb repression; the effects of deleting the motif are strong in some tissues but close to undetectable in others (Hittinger et al. 2005). Clearly, changes in the protein-coding sequences of transcription factors, apart from their more obvious DNA-binding residues, must be integrated into our understanding of the evolution of developmental regulation.\\nThe majority of residues in metazoan transcription factors do not fall within regions of well-defined globular structure, with many belonging to so-called ‘intrinsically disordered’ regions—regions that may form a structure when complexed with other macromolecules (Liu et al. 2006; Minezaki et al. 2006). The specific sequences of these regions are typically not obviously conserved between paralogues; because they are unique to particular families they are not covered in domain databases such as SMART and Pfam (Finn et al. 2006; Letunic et al. 2006). The lack of extreme conservation between distant species has sometimes masked the fact that within closely related species, these regions are conserved. Comparisons of orthologous sequences from closely related genomes (e.g. vertebrates or drosophilids) often show that substantial proportions of these non-domain sequences are undergoing strong purifying selection—they accumulate many more synonymous nucleotide changes than non-synonymous changes—and are thus functional. For the large part, precisely what these biological functions are is unknown; two possibilities, however, suggest themselves. Firstly, they may have relatively uninteresting non-specific effects, such as facilitating folding of the major domain (e.g. by reducing aggregation) or acting as spacers between globular domains. Secondly, and more interestingly from the point of view of animal evolution, they may include short linear peptide motifs that mediate protein–protein interactions (Dyson & Wright 2005; Neduva & Russell 2005; Neduva et al. 2005).\\nThere are numerous examples of regulatory motifs found outside of transcription factor domains. Many hox proteins include a YPWM-like hexapeptide motif that interacts with other homeodomain-containing proteins (In der Rieden et al. 2004); Drosophila ftz orthologues have lost this motif but acquired an LXXLL motif coupled to a new role in segmentation (Lohr & Pick 2005); and an N-terminal SSYF-like motif believed to be involved in transcriptional activation is conserved across Hox orthologues and paralogues from different phyla (Tour et al. 2005). Interaction motifs can be coupled with signalling pathways to create cell-type specificity. They can, for instance, be regulated by phosphorylation, such that the phosphorylation status governs what interactions can be made (e.g. Sapkota et al. 2007), or alternative splicing can result in protein–protein interaction motifs being included or excluded from particular cell types, providing additional layers of regulatory complexity that are likely to be species specific (Neduva & Russell 2005).\\nThe challenge of identifying small regulatory motifs means that their species distributions, and how their presence might produce taxon-specific differences in protein functions, have not been well studied. Examples that tie cleanly to one taxonomic group are less common, but an interesting case has been proposed in bilaterian orthologues of the Brachyury gene. These possess an N-terminal motif that is not found in non-bilaterian Metazoa (Marcellini 2006), which instead have a well-defined EH1-like motif (Copley 2005). The bilaterian motif is believed to be responsible for an interaction with Smad1, and hence to link gastrulation to bilateral pattern formation (Marcellini 2006).\\n5. Enhancers: transcription factor-binding sites and ultraconserved regions\\nTheoretical considerations have led to an intense focus on transcription factor-binding sites (TFBSs) as a major molecular source of morphological novelty (Wray et al. 2003; Carroll et al. 2005; Davidson 2006; Wray 2007), although see (Hoekstra & Coyne 2007) for a critique. Individual TFBSs show rapid turnover in comparisons of closely related genomes, with many being lineage specific (Dermitzakis & Clark 2002; Moses et al. 2006). This dynamic nature may not be revealed in the phenotype—patterns of gene expression may be conserved even though regulatory sequences change at the molecular level (Ludwig et al. 2000; Romano & Wray 2003; Fisher et al. 2006). On the other hand, the gain and the loss of individual TFBSs have been implicated in several recent cases of morphological evolution, in both vertebrates and invertebrates (reviewed in Simpson (2007) and Wray (2007)). The relationship between individual TFBSs and enhancer function is clearly not straightforward, beyond the fact that clustering of individual binding sites can identify some enhancer regions (Markstein et al. 2002). Cases of functional linkages between particular transcription factors have been proposed, for example, between dorsal, twist, Su(H) and an unidentified motif in neurogenic ectoderm formation in Diptera (Markstein et al. 2004), and even a coupling originating prior to the origin of Bilateria, of hairy and E(spl) promoting neural cell fate (Rebeiz et al. 2005).\\nComparisons of vertebrate genomes have revealed large regions (more than 100 nucleotides) of extreme conservation of non-coding sequences (conserved non-coding elements (CNEs); Bejerano et al. 2004). These regions are often found near transcription factors and other developmental genes (Sandelin et al. 2004). Outside of the vertebrates, there is evidence for similar regions occurring near developmental genes in flies (Glazov et al. 2005) and nematodes (Vavouri et al. 2007). Although in many cases the conserved regions are even found near orthologous genes, there is no evidence that they are homologous; they appear to have evolved independently in each of the phyla (Vavouri et al. 2007). Experimental evidence from vertebrates shows that many instances have roles as tissue-specific enhancer elements (Woolfe et al. 2005; Pennacchio et al. 2006).\\nThe length and lack of inter-phylum conservation of CNEs is in contrast to individual TFBSs. The DNA specificity of orthologous transcription factors is usually well conserved over large phylogenetic distances, but typical TFBSs are short, of the order of 6–10 nucleotides. An obvious possibility is that longer CNEs are composed of overlapping or adjacent TFBSs. This would suggest a tight packing of transcription factor proteins on the genomic DNA of these CNEs. There is direct evidence for this: some fragments of highly conserved non-coding sequences are present in crystal structures of transcription factor complexes. An atomic model based on known crystal structures of the interferon-β enhancer, for example, shows 50 consecutive nucleotides in contact with eight different proteins; these nucleotides are well conserved in mammalian species (Panne et al. 2007; see figure 2 for another example). Given that such structures exist, it is not such a leap to imagine 16 proteins binding to 100 nucleotides, or even bigger complexes. This suggests a model where CNE enhancer regions controlling orthologous genes in different phyla are controlled by multiple TFBSs, although not necessarily the same transcription factors or in the same orientation. Moreover, the tight packing of transcription factors on the genomic DNA suggests that the proteins themselves may be co-adapted to interact with each other and aid the cooperative formation of enhancer complexes. Previously, Ruvinsky & Ruvkun (2003) have presented experimental evidence that enhancers and transcription factors co-evolve in this way, with neuronal and muscle-specific enhancer elements from D. melanogaster failing to drive expression in homologous tissue types in C. elegans, and Dover and co-workers have argued for coevolution of bicoid protein and hunchback regulatory regions (McGregor et al. 2001; Shaw et al. 2002).\\nIf protein–protein interactions between transcription factors are often required for the formation of enhancer complexes, close analysis of transcription factor sequence and structure may reveal evidence for co-adaptation of proteins, such as the Hox hexapeptide motif, through which homeotic proteins form complexes with TALE class homeodomains (LaRonde-LeBlanc & Wolberger 2003). We might expect instances of co-adapted transcription factor combinations to be taxon specific, to match the taxon specificity of enhancer sequences.\\n6. Alternative splicing\\nNot all CNEs are associated with enhancer regions. There is good evidence that many are involved in regulating alternative splicing events, including the alternative splicing of mRNAs of proteins which themselves regulate alternative splicing (Lareau et al. 2007; Ni et al. 2007). The presence of highly conserved control elements to regulate alternative splicing indicates that the functional consequences are of importance. Although large very conserved elements may be the exception rather than the rule, detailed comparative analyses have identified smaller conserved motifs regulating alternative splicing, for instance in nematodes (Kabat et al. 2006) and vertebrates (Sorek & Ast 2003; Yeo et al. 2005).\\nAlternative splicing is often touted as a mechanism by which proteomic complexity is increased. Although early reports suggested that levels of alternative splicing were comparable in vertebrates and invertebrates (Brett et al. 2002), more recent studies suggest that there is indeed more alternative splicing of transcripts in vertebrates (Kim et al. 2007), suggesting a link with increased phenotypic complexity. How relevant is alternative splicing for species-specific biology and morphological differences? Quantitatively, the gene products that appear to be most affected by alternative splicing are typically involved in nervous and immune system function (Modrek et al. 2001). There are, however, ample examples of alternatively spliced transcription factors—as many as 63% of mouse transcription factors have variant exons (Taneri et al. 2004). Although the differences in molecular roles of the alternatively spliced products are often unknown, the genes themselves include developmental classics such as members of Hox, SMAD and T-box families (Fan et al. 2004; Dunn et al. 2005; Noro et al. 2006), although they do not necessarily present obvious morphological correlates (Yoder & Carroll 2006). Alternative splicing of modular proteins is an obvious route through which functions can be changed, by including or excluding particular combinations of domains. In this regard, it is interesting that alternative splicing often affects intrinsically disordered regions outside known protein domains (Romero et al. 2006)—this again points to a critical role for finely tuned protein–protein interactions among transcriptional regulators.\\nThere are few known cases of distant conservation of alternative splice variants of transcription factors; typically, examples are conserved within phyla at best. Widening the search to other classes of gene again suggests that splice variants are not conserved over long periods, although it should be remembered that transcript coverage of most species from which evidence of alternative splicing is obtained is very restricted. Perhaps the best counter-example is currently that of fibroblast growth factor receptor 2 (FGFR2), where an exon configuration diagnostic of mutually exclusive alternative splicing is found in both vertebrates and the sea urchin Strongylocentrotus purpuratus (Mistry et al. 2003). Examples of orthologous ion channel encoding genes showing similar alternative splicing patterns in D. melanogaster, C. elegans and humans are likely to be cases of parallel evolution (Copley 2004). The shared ability of vertebrates and at least insects and C. elegans to produce alternative transcripts in a regulated manner, but the absence of large numbers of conserved alternative splicing between protostomes and deuterostomes suggests that gene products have become alternatively spliced in parallel between different lineages, while at the same time hinting that the functions performed by alternative splice variants may, over time, be replaced by different genomic solutions.',\n",
       "  'Discussion': 'Although most major classes of protein involved in animal development may be conserved throughout the Metazoa, detailed comparative analysis of these gene types reveals a more dynamic picture, with frequent gene duplication, gene loss, couplings with new motifs and other processes such as alternative splicing and regulation by micro-RNAs, all of which are likely to be important for a full understanding of function.\\nCis-regulatory variation may well be revealed to be quantitatively the most common form of variation between species, but it seems probable that the cumulative effects of multiple cis-regulatory changes will have required that protein networks evolve to accommodate and correctly regulate changed enhancer structures.\\nOur knowledge of animal evolution and the picture presented here is currently based on a very small sampling of almost exclusively nematode, insect and vertebrate genomes. Although this situation is beginning to change, the fact that many important functional regions, especially those that do not encode proteins, are only revealed by having sets of closely related genome sequences, and that there are 35 or so animal phyla gives some idea of the enormity of the challenges ahead. The rapidly falling costs of genome sequencing do, however, give grounds for optimism.',\n",
       "  'Conclusion': 'Although most major classes of protein involved in animal development may be conserved throughout the Metazoa, detailed comparative analysis of these gene types reveals a more dynamic picture, with frequent gene duplication, gene loss, couplings with new motifs and other processes such as alternative splicing and regulation by micro-RNAs, all of which are likely to be important for a full understanding of function.\\nCis-regulatory variation may well be revealed to be quantitatively the most common form of variation between species, but it seems probable that the cumulative effects of multiple cis-regulatory changes will have required that protein networks evolve to accommodate and correctly regulate changed enhancer structures.\\nOur knowledge of animal evolution and the picture presented here is currently based on a very small sampling of almost exclusively nematode, insect and vertebrate genomes. Although this situation is beginning to change, the fact that many important functional regions, especially those that do not encode proteins, are only revealed by having sets of closely related genome sequences, and that there are 35 or so animal phyla gives some idea of the enormity of the challenges ahead. The rapidly falling costs of genome sequencing do, however, give grounds for optimism.',\n",
       "  'URL': 'https://doi.org/10.1098/rstb.2007.2235',\n",
       "  'Journal': 'Philos Trans R Soc Lond B Biol Sci'},\n",
       " {'PaperTitle': 'Role of Nuclear Receptors in Blastocyst Implantation',\n",
       "  'Publication': '2013',\n",
       "  'Authors': 'YM Vasquez, FJ DeMayo',\n",
       "  'Email': '',\n",
       "  'Abstract': 'The regulation of blastocyst implantation in the uterus is orchestrated by the ovarian hormones estrogen and progesterone. These hormones act via their nuclear receptors to direct the transcriptional activity of the endometrial compartments and create a defined period in which the uterus is permissive to embryo implantation termed the “window of receptivity”. Additional members of the nuclear receptor family have also been described to have a potential role in endometrial function. Much of what we know about the function of these nuclear receptors during implantation we have learned from the use of mouse models. Transgenic murine models with targeted gene ablation have allowed us to identify a complex network of paracrine signaling between the endometrial epithelium and stroma. While some of the critical molecules have been identified, the mechanism underlying the intricate communication between endometrial compartments during the implantation window has not been fully elucidated. Defining this mechanism will help identify markers of a receptive uterine environment, ultimately providing a useful tool to help improve the fertility outlook for reproductively challenged couples. The aim of this review is to outline our current understanding of how nuclear receptors and their effector molecules regulate blastocyst implantation in the endometrium.',\n",
       "  'Introduction': '1. Introduction: The Nuclear Hormone Receptor Gene Superfamily\\nThe nuclear receptor family is composed of a large number of intracellular proteins with unique roles in transcriptional regulation and very specific patterns of expression. Sequencing of the human genome has identified 48 nuclear receptors (NRs) and allowed for the phylogenetic classification into six evolutionary groups. NRs have modular domains for transactivation, ligand-binding and DNA-binding [1]. Sequence variation at the DNA binding domain allows NRs to bind discrete DNA sequences in the regulatory regions of genes known as hormone response elements (HRE). Along with DNA sequence binding similarity, members of each group also share dimerization abilities. Endogenous molecules such as steroid hormones, thyroid hormones, oxysterols and fat-soluble vitamins have been identified to regulate the activity of more than half of all NRs. Members with no known ligand are called orphan receptors. Naturally occurring and synthetic compounds have been progressively identified to bind some of these orphan receptors, resulting in their newer classification as “adopted receptors”. Various ligands have been identified to stabilize different combinations of NR dimer pairs, promote specific post-translational modifications, regulate DNA binding and cellular localization or promote NR degradation. Ligand bound or unbound NRs can engage in transcriptional crosstalk with pioneer transcription factors, co-regulators, chromatin remodeling enzymes and transcriptional machinery. These interactions result in the activation or repression of gene transcription and are highly context specific. Ultimately, NRs mediate cellular responses to hormones and ligands by regulating various cellular processes including cell survival, proliferation and differentiation [1]. Several of these members have been described to have pivotal roles in the regulation of endometrial function and dysfunction. For this reason, NRs have been extensively studied and targeted for the pharmacological manipulation of various cellular processes in the treatment gynecological malignancies and infertility.\\nThis review will discuss the role of NRs in the regulation of the “window of receptivity”, a short and tightly regulated period in the female reproductive cycle in which the endometrium becomes receptive to the implanting blastocyst. It is only during the window of receptivity that conditions are permissive of blastocyst implantation for the initiation of pregnancy [2]. A number of studies in humans and non-human primates have looked at some of the morphological and transcriptional changes occurring during the window of receptivity. However, most of what we know about uterine biology and implantation comes from murine studies and transgenic models [3]. Several protocols have been developed to successfully mimic events of early pregnancy in the mouse, providing investigators with useful strategies to interrogate implantation mechanisms [4]. The NRs show very dynamic and compartment specific expressions through the window of receptivity in the murine endometrium (Figure 1.)\\nThe review will begin by providing a brief comparative description of the female reproductive cycle in women and mice, highlighting the role of the ovarian steroid hormone estrogen and progesterone. These hormones play a central role in the regulation of implantation via their cognate nuclear receptors to govern the proliferation and differentiation of the endometrial compartments through an intricate mechanism of paracrine crosstalk [5]. Finally, there will be a brief description of other members of the nuclear receptor family known to be expressed in the endometrium. However, characterization of these NRs during implantation may not be fully developed in many cases due to lack of appropriate model systems. Final remarks will highlight important advances and future directions in the molecular understanding of NR biology during implantation.\\nFigure 1. Peri-implantation Expression of the Nuclear Receptors.\\nOpen in a new tab\\nThe implantation period in the mouse can be divided into pre-receptive, receptive and post-receptive. The nuclear receptors ERα, PR and COUP-TFII have distinct spatiotemporal expression patterns during implantation.',\n",
       "  'Methods': '',\n",
       "  'Results': '2. The Window of Receptivity\\nUterine receptivity to the implanting blastocyst is regulated by the ovarian steroid hormones estrogen and progesterone [6]. In humans, a rise in estrogen levels occurs during the ovulatory/follicular phase of the menstrual cycle. Estrogen stimulates proliferation and growth of the epithelial and stromal compartments of the endometrium. Differentiation of the ovarian granulosa cells into the corpus luteum results in the production of progesterone and the transition into the secretory phase. At this time, proliferation in the epithelial compartment is ceased and the stromal compartment undergoes a progesterone dependent process of differentiation termed decidualization. In species where the blastocyst invades through the epithelial layer into the underlying stroma, decidualization is absolutely required for implantation of the blastocyst and pregnancy progression. Decidualization is a highly complex differentiation process involving an extensive reprogramming of families of genes that result in changes in stromal cell morphology, steroid responsiveness, secretory profile, resistance to oxidative stress and extra cellular matrix remodeling [7, 8]. In humans, the decidual reaction occurs recurrently as part of the menstrual cycle and is a blastocyst-independent process. Decidualization initiates near the spiral arteries and spreads throughout the endometrium. The uterus becomes receptive 7-10 days after ovulation during the mid-secretory phase. During the second half of the secretory phase, the endometrium becomes refractory and non-receptive, ultimately concluding in menses in the absence of pregnancy. This menstrual cycle is repeated every 28-30 days [9].\\nIn contrast, mice have shorter and often irregular cycles. The estrous cycle in mice last 4-5 days and can be divided into 4 phases: proestrous, estrous, metestrous and diestrous. In proestrous, the ovary accumulates a mature batch of follicles ready for ovulation during the subsequent estrous phase. Female mice are most receptive to mating during the estrous phase. The morning after mating, termed day 0.5 of pregnancy, a preovulatory surge of estrogen acts upon the uterine epithelium to stimulate proliferation. Estrogen subsequently decreases by day 1.5. At day 2.5, the formation of corpora lutea results in the production of progesterone and the proliferation of the endometrial stroma. Copulation is able to extend this metestrous stage for up to 10 days in a state called pseudopregnancy, but in the absence of pregnancy the uterus enters the diestrous phase. On day 3.5 of pregnancy there is a nidatory estrogen spike that is absolutely critical for inducing uterine receptivity [10, 11]. The window of receptivity opens on day 4.0, allowing for implantation to occur. Unlike humans, decidualization of the murine stroma requires mechanical stimulation by the blastocyst at the site of attachment. The decidual reaction subsequently spreads throughout the implantation chamber [12]. In both species, sustained progesterone signaling is critical for the maintenance of pregnancy.\\n3. Estrogen Receptor\\nThe steroid hormone estrogen acts via the nuclear receptors estrogen receptor α (NR3A1) and estrogen receptor β (NR3A2). The estrogen receptor- α (ERα) and β (ERβ) isoforms arise from two distinct genes. ERα is predominantly expressed in the mammary glands, pituitary, hypothalamus, ovarian theca cells and uterus. In contrast, ERβ is primarily expressed in the ovarian granulosa cells, lung and prostate [13]. In both human and macaque endometrium, ERα and ERβ are expressed in the nuclei of the glandular epithelium and the stroma. Expression of ERα is high in the glands and stroma during the estrogen dominated proliferative phase but declines during the secretory phase [14–16]. Interestingly, the functionalis and basalis layers of the endometrium display subtle differences in the expression of ERα and ERβ during artificial cycles in the macaque [15]. ERα expression is spatially and temporally regulated during implantation in the mouse. Its expression can be observed strongly in the glandular epithelium and more moderately in the luminal epithelium and subepithelial stroma at day 3.5 of pregnancy. At day 4.5, strong expression persists in the glands while most luminal and subepithelial expression is detected in the mesometrial pole of the implantation chamber. Decidual cells surrounding the blastocyst lack ERα expression while the undifferentiated stromal cells maintain strong expression [17].\\nUterine responsiveness to estradiol has been attributed primarily to the action of ERα, due to the lack of uterine stimulation and mitotic growth response to estrogen observed in the ERα null mice (ERαKO) [18, 19]. The role of ERβ remains controversial. While one study reported that ablation of ERβ in mice (ERβKO) did not result in a significant difference in induction of estrogen responsive genes compared to wild-type mice [19], a separate study reported a significant difference in estrogen and progesterone responses in the ERβKO when an exaggerated estrogen responsiveness and lack of differentiation of the epithelium following acute estrogen treatment was observed [20]. The ERαKO mouse has a hypoplastic uterus and is infertile due to an implantation failure [21]. In contrast, the ERβKO mouse maintains normal implantation, further suggesting estrogen signals primarily via the α isoform [22]. Interestingly, the ratio of ERα to ERβ is altered in ectopic lesions of women with endometriosis, a disease characterized by growth endometrial tissue outside the uterine cavity causing inflammation and chronic pain. Lesions have been shown to be dependent on estrogen and resistant to progesterone signaling [23]. Together with the observation of exaggerated estrogen responsiveness in the ERβKO, this evidence indicates a distinct role of the ER isoforms in the regulation of uterine function and homeostasis.\\nEstrogen regulation of gene transcription is mediated by ERα interaction with chromatin near the promoter regions of estrogen responsive genes. The ERα cistrome was determined in the murine uterus using chromatin immunoprecipitation (ChIP) followed by whole genome sequencing (ChIP-seq). ChIP-seq identified 5,184 binding sites after vehicle treatment and 17,240 binding sites after treatment with estradiol for one hour. Motif analysis of these binding sites revealed the highest enrichment was for estrogen response element (ERE) motifs, but also reported were significant enrichments in the motifs of other members of the NR family including: peroxisome proliferator associated receptor γ, the thyroid, vitamin D, androgen and glucocorticoid receptors, COUP-TFII, DAX1, retinoid-related orphan receptor-α and estrogen-related receptor. Additionally, there were significant increases in many other non-NR transcription factors, most notably the homeobox (Hox) motifs. ERα binding was overlapped with transcriptional datasets for genes increased or decreased by estrogen relative to vehicle after 2, 6 or 24 hours. This overlap revealed that genes that were up regulated after 2 or 6 hours of estrogen treatment exhibited enriched binding proximal to the transcriptional start site (<10kb) while, in contrast, ER was found to be typically bound at more distal regions surrounding genes that were down regulated. Among the estrogen up-regulated genes were Igf1 and Cdk1, important regulators of early and late estrogen mitogenic events, respectively [24].\\nIn the mouse, the nidatory estrogen surge in the morning of the fourth day of natural pregnancy coincides with the up-regulated expression of the IL-6 family member leukemia inhibitory factor (Lif) [25]. Estrogen directly mediates Lif signaling via binding of ERα to both the Lif and Lifr genes and regulated their expression in the estrogen treated murine uterus [24]. Lif is a secreted glycoprotein that binds to LIF receptor (LIFR) and gp130 and activates the downstream signal transducer STAT [26]. Both of these receptors are expressed by the blastocyst as well as the uterine luminal epithelium, identifying a potential mechanism of autocrine/paracrine signaling during implantation. LIF expression is seen in the glandular epithelium at day 3.5 and subsequently in the stroma at the site of blastocyst attachment [27]. It was shown that in mice lacking Lif (Lif−/−), the uterus did not respond to the presence of an embryo and was unable to produce a decidual response by artificial stimulation. Interestingly, intraperitoneal injection of LIF was able to rescue implantation and was sufficient to substitute for the nidatory estrogen surge [28]. Altogether this evidence suggests that the nidatory estrogen surge, mediated by ERα, is only required to induce Lif and that the glandular secretion of LIF is only required to initiate the window of receptivity but not subsequently required for the maintenance of pregnancy [28].\\nLIF expression is also high in humans around the time of implantation [6]. Clinical data has shown that endometria of women with unexplained infertility and multiple implantation failures often display significantly lower levels of Lif during the midsecretory phase of their menstrual cycle when compared to healthy fertile controls [29].\\n3.1 Estrogen Regulation of Proliferation\\nThe molecular pathways regulated by estrogen have been extensively evaluated in the human and murine endometrium and have identified the conserved role of estrogen in the regulation of proliferation [9]. In humans, the regeneration of the endometrial layers after every menstrual cycle initiates with the proliferation of the basal compartments before an increase in circulating estrogen [30]. The proliferative activity dramatically increases with increasing levels of estrogen and peaks between days 8 and 10 of the menstrual cycle [31]. The epithelial and stromal compartments of the endometrium are highly sensitive to estrogen stimulation and their response has been shown to occur in a biphasic mode during which distinct sets of genes are regulated during early and late events. These responses have been extensively characterized at the physiological, morphological and molecular level. In the mouse early uterine responses to estrogen include transcription of early cell cycle genes, hyperemia, infiltration of immune cells and water imbibition into the uterine tissue [32, 33]. Later responses include the further infiltration of immune cells, increased uterine weight and the induction of late cell cycle genes resulting in robust DNA synthesis and mitosis of epithelial cells [33].\\nEpithelial proliferation in the ovariectomized murine endometrium can be observed by a wave of DNA synthesis starting 6 hours after exposure to estrogen. The wave peaks approximately 12-15 hours with the progression through the G2 and M phases of the cell cycle [34, 35]. Estrogen stimulates the activation of the insulin-like growth factor 1 receptor (IGF1R) which leads to the activation of the phosphoinositide 3-kinase (PI3K)/AKT pathway to execute an inhibitory phosphorylation of the glycogen synthase kinase (GSK-3β) [36]. Inactivation of GSK-3β limits the inhibitory phosphorylation of cyclin D1 and allows for its nuclear translocation. Estrogen also induces cyclin E- and cyclin A-cdk2 kinases which promote the hyperphosphorylation of Rb and p107 proteins. Hyperphosphorylated Rb and p107 proteins are unable to inhibit E2F-mediated transcriptional transactivation. The result is the activation of the DNA replication machinery and entry into S-phase of the cell cycle ensuring robust proliferation of the uterine epithelium [37]. The inhibition of proliferation is subsequently necessary for successful implantation [38]. The proposed mechanisms for this inhibition are regulated by progesterone and will be discussed ahead in this review.\\n3.2 Compartment Specific Roles of ER\\nERα expression is under a very strict hormonal and temporal control, indicative of spatiotemporal roles during implantation [17]. The most substantial evidence for the compartment specific roles for ERα was the observation that the uterine epithelium in neonatal mice displayed a mitogenic response to estrogen stimulation despite lack of epithelial ERα expression. Because the mesenchymal/stromal cells of neonatal uteri express ERα, it was hypothesized that the epithelial mitogenic response to estrogen was mediated by stromal ERα paracrine action [39–41]. In order to better elucidate the compartment specific roles of ERα, studies used tissue recombinants of epithelial and stromal cells isolated from wild-type and ERα knock-out uteri were conducted. Tissue recombinants were grafted under the kidney capsule of intact nude mice and estrogen dependent induction of proliferation was measured by [3H] thymidine incorporation into DNA. Wild-type reconstitutions and those with wild-type stroma both exhibited robust levels of epithelial [3H] thymidine labeling. In contrast, tissue grafts containing wild-type epithelium and ERα knock-out stroma were unable to stimulate epithelial proliferation in response to estrogen [42]. This evidence indicated that stromal ERα was both necessary and sufficient to induce estrogen-dependent epithelial proliferation.\\nThe epithelial-specific role for ERα was evaluated in vivo using an epithelial specific ERα knock-out mouse model. This mouse model selectively ablates the Esr1 gene by utilizing the Esr1-loxP and the epithelial specific Wnt7aCre expressing system (UTEpiαERKO). Selective ablation of ERα in the uterine epithelium results in female infertility due to an impaired estrogen response. Interestingly, estrogen stimulated DNA synthesis and expression of mitogenic mediators of epithelial proliferation in wild-type and UTEpiαERKO mice. However, expression of the luminal epithelial estrogen target lactoferrin (Ltf) and a subset of estrogen targets were compromised in the UTEpiαERKO mice. The expression of the apoptotic inhibitor Birc1a was significantly lower after uterine growth was induced by estrogen treatment for 3 days. Consequently, UTEpiαERKO displayed a significant increase in apoptosis. This evidence suggests the role of epithelial ERα is to prevent epithelial apoptosis and ensuring a full epithelial response, while stromal ERα is responsible for estrogen-driven epithelial proliferation [43]. These observations are consistent with the tissue reconstitution studies and collectively illustrate the differential roles for ERα during implantation.\\n4. Progesterone Receptor\\nThe hormone progesterone acts via the progesterone receptor (NR3C3). Progesterone receptor (PR) exists as two isoforms, PR-A and PR-B, arising from alternative promoter usage in the same gene. PR isoforms are each composed of multiple domains including the activation domain (AF-1) in the N-terminus, the DNA binding domain and the ligand binding domain which contains a second activation domain (AF-2). The PR-B isoform has an additional 164 amino acids in the N-terminus which contains an additional activation domain (AF-3) [44]. There has been evidence of a third truncated isoform, PR-C. This isoform lacks the N-terminus region containing the DNA binding domain, thereby hindering binding of PR-C to DNA. Because PR-C retains the ability to bind hormone, it has been implicated in the functional withdrawal of progesterone in the regulation of parturition [45]. The total levels of PR and the ratio of PR-A/PR-B vary across reproductive tissues. Since PR-A lacks the amino terminal transactivation, it may serve to dampen PR-B driven transcriptional activation when these isoforms are co-expressed [46]. PR-A and PR-B can form heterodimers that result in a dominant repression of PR-B activity [47]. The differential transactivation properties of each isoforms and stoichiometry collectively result in a complex mechanism of progesterone responsiveness in specific tissues and cellular contexts.\\nMolecular profiling of the human endometrium throughout the menstrual cycle indicates expression of PR is highest in the estrogen-driven proliferative phase and decreases in the progesterone-dominated secretory phase [16]. These observations are consistent with the evidence that PR expression is induced by estrogen action on an ERE in the promoter region of PR while progesterone inhibits its own expression by action of PR/ER regulation of the same ERE [48]. Expression of both PR isoforms is observed in the murine uterus, and ablation of both isoforms (PRKO) results in multiple reproductive abnormalities, including a hyperplastic response to estrogen and progesterone and an implantation defect [49]. However, specific ablation of PR-A (PRAKO) and PR-B (PRBKO) individually has shed light on the role of these isoforms in the mouse uterus. Ablation of PR-B resulted in no discernible uterine phenotype and display normal fertility. PRBKO mice display reduced pregnancy-associated mammary gland morphogenesis, indicating PR-B is a major regulator of mammary gland maturation during pregnancy [50]. Ablation of PR-A phenocopies the PRKO mouse phenotype and indicates that the PR-A isoform is the predominant functional isoform in the mouse uterus [51]. Contrary to what is observed in wild type mice where progesterone antagonizes estrogen-induced epithelial proliferation, progesterone treatment induced proliferation of the uterine epithelium in the PRAKO mice and this proliferation was shown to be dependent on PR-B [51]. This PR-B-dependent gain of function in proliferative activity indicates an important role for PR-A in the regulation of the potentially adverse effects of PR-B-driven proliferation as well as inhibition of estrogen-driven hyperplasia. Collectively, these observations identify functional differences between these two isoforms in response to progesterone in the regulation of epithelial proliferation during early pregnancy.\\n4.1 Progesterone Receptor Targets\\nOne of the earlier murine studies identified progesterone targets by comparing the uterine gene expression of wild-type and PRKO mice after acute and chronic treatment with progesterone [52]. However, in this study the most robust PR-dependent gene expression change occurred after acute progesterone treatment with the up-regulation of 55 genes. Among these was Indian Hedgehog (Ihh), a morphogen expressed in the luminal epithelium shown to be absolutely required for implantation and decidual response [53]. Chronic progesterone treatment resulted in the robust down-regulation of 102 genes including many of the transcription factors, transport proteins, signal transduction, cell growth, and enzyme genes that had been induced by acute progesterone treatment. The progesterone-PR axis was shown to regulate retinoic acid metabolism, known to play a role in female fertility and the Klks gene family associated with regulation of the inflammatory process, vasodilatation and vascular permeability [54–57]. The P4-PR axis was also shown to regulate other important transcription factors and co-regulators including Gata2 and Cited1 [52].\\nPR regulation of gene expression occurs by direct binding of the receptor in the regulatory promoter regions of targets genes [58]. In order to identify direct genomic targets of PR in the murine endometrium the genome-wide PR cistrome was determined using ChIP-seq. In the absence of ligand, ChIP-seq identified 6,367 PR-binding sites. The number of binding sites increased nearly 3-fold (18,432) after acute progesterone treatment. Over 73% of these intervals contain progesterone response elements (PRE) or half-PREs. Intervals were enriched in the promoter regions proximal to the transcriptional start sites of genes. PR binding datasets were overlapped with microarray gene expression comparing genes significantly induced by acute progesterone treatment. This analysis confirmed PR binding on both up-regulated (Gata2, Egfr, Ihh, Fkbp5, Cyp26a1) and down-regulated (Pgr, Wnt7a, Lifr) progesterone target genes [59]. Most notably, the ChIP-seq and microarray datasets overlap identified a novel role for progesterone in the regulation of uterine circadian rhythm discussed in greater detail further ahead in this review.\\n4.2 Progesterone Inhibition of Estrogen Signaling\\nThe use of progesterone supplementation has been an invaluable resource to Assisted Reproductive Technology (ART) efforts [60, 61]. However, little is known about the action of progesterone during the implantation window in humans because the numerous molecular profiling efforts have been limited by the technologies available at the time or imprecise methods for endometrial cycle dating [62–66]. Nevertheless, these studies have consistently identified a conserved role of progesterone in the inhibition of estrogen-driven epithelial proliferation. The antiproliferative action of progesterone in the endometrium has also been the focus of extensive research for its potential therapeutic role regulating progression of estrogen-dependent pathologies such as endometrial cancer and endometriosis [67]. Contrasting mechanisms have been proposed for the progesterone-mediated inhibition of estrogen induced epithelial cell proliferation. One line of evidence supports a paracrine mechanism while the alternative proposes a direct role of PR in the uterine epithelial cells.\\nThe paracrine mechanism for the inhibition of epithelial proliferation was strongly supported when the basic helix-loop-helix transcription factor Hand2 was shown to play an essential role in the regulation of growth factor signaling in a progesterone-dependent manner. Hand2 was identified as a significantly regulated gene in the stroma of mice treated with the PR antagonist RU486 [68]. Conditional ablation of Hand2 using the PRcre/+ Hand2f/f (Hand2d/d) bigenic mouse model results in infertility due to an implantation defect. Progesterone inhibition of epithelial proliferation and down regulation of estrogen driven genes including Mucin 1 (Muc1) and Lactoferrin (Ltf) was altered in the Hand2d/d mice. Progesterone responsive genes, including Ihh, Alox15 and Irg1 were not affected. Interestingly, several members of the fibroblast growth factor family (FGFs) and HB-EGF showed a significantly high level of expression in Hand2d/d mice. Fgfs act on Fgf receptors (Fgfr) to activate downstream extracellular signal regulated kinase (ERK) 1 and 2 or the PI3K/AKT pathways. Interestingly, the Fgfr’s are expressed on day 1 and 4 of pregnancy in the epithelium of mice. In the uterus of Hand2d/d, aberrant activation of Fgfr was observed on day 4 of pregnancy, suggesting a mechanism for stromal Hand2 mediated regulation of ERK1/2 and PI3K/AKT signaling in the epithelium [69]. Collectively these results identified a complex mechanism of stromal-epithelial crosstalk regulates proliferation during implantation.\\nThe direct action of progesterone on epithelial cell proliferation is based on the observation that progesterone has been shown to completely inhibit estrogen-induced DNA synthesis and cell proliferation [70, 71]. It was found that progesterone inhibits the estrogen regulation of cyclin D1 nuclear translocation resulting in the hypophosphorylation of Rb and p107 proteins and a block in G1-S phase progression [72]. PR also regulates the members of the MCM family of proteins to inhibit epithelial proliferation [37, 73]. Substantial evidence implicates the Kruppel-like factor family in this mechanism. The Kruppel-like factor family is composed of zinc finger transcription factors involved in many cellular processes including proliferation, differentiation, apoptosis and development [74]. KLF4 and KLF15 play a significant role in the regulation of epithelial proliferation. KLF4 is up regulated only in the epithelium of ovariectomized mice after treatment in estradiol and inhibited when mice were pretreated with progesterone. In contrast, expression of KLF15 is dramatically increased after estrogen and progesterone treatment in the epithelium and moderately in the stroma. The temporal expression of these KLFs was associated with steroid control of epithelial proliferation via transcriptional regulation of Mcm2, a positive regulator of proliferation required for DNA replication licensing. Estradiol treatment stimulated ERα, KLF4 and RNAPol II binding at transcriptional regulatory sites of the Mcm2 and induced Mcm2 expression. Treatment with estradiol and progesterone ablated this binding but favored binding of KLF15. KLF15-mediated estrogen and progesterone inhibition of Mcm2 promoter activity, Mcm2 expression and proliferation was confirmed in the Ishikawa epithelial cancer cell line, providing a mechanism for KLF regulation of epithelial proliferation. [75]\\nThe above observations suggest that there are direct actions of progesterone and PR, as well as paracrine mechanisms of action. The above conflicting conclusions likely arise from the use of different hormone treatment in these murine models. These strategies could be mimicking events that occur synchronously and synergistically across compartments during different stages of early pregnancy. A more refined hypothesis would be that both of these mechanisms could be occurring at specific times during early pregnancy and disruption of either of these results in the disruption of implantation. This holistic hypothesis highlights the complexity of progesterone signaling and underscores the need to develop new strategies to address remaining questions. Therefore, understanding the compartmental specific expression and action of PR in the uterus is critical.\\n4.3 Compartment Specific Roles of PR\\nExpression of PR is observed in the luminal epithelium and subepithelial stroma at day 3.5 of pregnancy. By day 4.5, PR expression in the luminal epithelium ceases and is restricted to the subepithelial stroma at the site of implantation and is undetected in the glandular and luminal epithelium. Expression becomes strong and persistent in the stromal compartment throughout the process of decidualization [17]. In order to begin elucidating the compartment specific roles of PR, studies used tissue recombinants of epithelial and stromal cells isolated from wild-type and PRKO uteri. Tissue recombinants were grafted under the kidney capsule of intact nude mice. Subsequently, mice were ovariectomized and treated with estradiol alone or in combination with progesterone. All tissue grafts combinations (wild-type epithelium with wild-type stroma, PRKO epithelium with wild-type stroma, wild-type epithelium and PRKO stroma) displayed comparable levels of epithelial [3H] thymidine labeling in response to estradiol. Tissue recombinants containing wild-type stroma, independent of the expression of PR in the epithelial tissue, displayed a significant inhibition of estradiol-induced epithelial proliferation with progesterone co-treatment. Tissue grafts with PRKO stroma with wild-type or PRKO epithelium, were not able to suppress epithelial proliferation in response to progesterone treatment. This indicated that expression of PR in the stroma was both necessary and sufficient to inhibit the estrogen-induced epithelial proliferation by progesterone. These results provided evidence for the existence of paracrine signals from the stroma to the epithelium to mediate progesterone signaling.\\nIn order to understand progesterone signaling in compartment specific manner in vivo under the physiological context of implantation the epithelial PR was ablated using a bigenic mouse model expressing Cre recombinase under the epithelial specific promoter of Wnt7a and containing PR allele flanked by loxP sites. This mouse model selectively ablates the Pgr gene by utilizing the Pgr-loxP and the epithelial specific Wnt7a Cre/+ expressing system (Wnt7aCre PRf/−). As a consequence of disrupted progesterone signaling, the epithelial PR knock-out mice were infertile due to a defect in blastocyst attachment and an absent decidual response. Expression of the epithelial progesterone targets, Ihh and Areg where significantly down regulated when PR was ablated in the luminal epithelium [76]. In the absence of epithelial PR, the progesterone-dependent down-regulation of estrogen targets cyclin D1 and MCM3 was not observed. In accordance with the defect in down regulation of estrogen proliferation targets, it was observed that ablation of epithelial PR results in retained proliferation of the luminal epithelium at days 2.5 and 3.5 of natural pregnancy. These results are in disagreement with the findings determined by tissue reconstitution studies and demonstrated an essential role of the epithelial PR in the progesterone inhibition of estrogen-driven epithelial proliferation [76]. Notably, it was shown that epithelial PR regulated epithelial expression of Ihh after acute progesterone stimulation. However, the natural pregnancy data indicated only a trend supportive of the hypothesis that epithelial PR was necessary and sufficient for this induction. The inconsistent observations between tissue reconstitution studies and the Wnt7aCre PRf/− mouse model underscore the urgent need to investigate role of the stromal PR in the regulation of pre-implantation events in vivo. As mentioned above, the differences in these observations may be due to hormonal treatment and the time in which the proliferation is evaluated. In the pre-implantation uterus PR is expressed in the epithelial compartment and its actions to regulate gene expression and cell proliferation are direct. During the window of receptivity, PR expression ceases in the epithelium and increases in the stromal compartment. At this time the paracrine mechanism of regulation may dominate.\\n5. Androgen Receptor\\nThe androgen receptor (NR3C4) has been extensively studied for its role in male reproductive tissues. However, the androgen receptor (AR) is also expressed in female reproductive tissues including the ovary and the uterus [77]. AR is expressed in both stromal and epithelial compartments throughout the menstrual cycle, with expression being highest in the stromal compartment during the proliferative phase and lowest in the secretory phase [78]. Expression of AR is elevated in the endometrium of women with polycystic ovarian syndrome (PCOS) and premature ovarian failure (POF). [77, 79, 80] Although expression of AR decreases in the decidualizing stroma during the secretory phase, androgen signaling seems to play a critical role in the regulation of a distinct subset of genes during in vitro decidualization of primary human endometrial stromal cells (HESC). Addition of dihydrotestosterone (DHT) enhances the progestin and cAMP-dependent activation of the promoter for the decidual marker prolactin (PRL) and enhances the expression of PRL in a dose dependent manner. Gene expression analysis in decidualizing HESC identified genes involved in cytoskeletal organization, cell motility and cell cycle regulation to be deregulated after AR depletion. Additionally, stromal cells failed to adequately form differentiation-dependent stress fibers [81]. It has been suggested that AR has antiproliferative effects that overlap or may even complement those of progesterone in estrogen-induced replication [82]. Knockdown of AR significantly promotes cell proliferation and partially reverses the hyperphosphorylation of RB associated with HESC differentiation [83]. Additionally, androgens have been shown to inhibit cell growth and DNA synthesis in HESC and cancer cell lines [84, 85]. However, it is unknown if this mechanism plays a role in the regulation of the window of receptivity. The AR knockout (ARKO) mice displayed reduced fertility over time. The late infertility defect was attributed to the development of a POF phenotype. However, from an early time point ARKO mice displayed a reduction in litter size, smaller uteri, reduced uterine growth at estrous, abnormal placental development and placentamegaly [77]. While expression of AR and a physiological role in uterine function is evident in the human and mouse, its function in the regulation of uterine receptivity is still not fully understood.\\n6. Glucocorticoid Receptor\\nGlucocorticoids are important regulators of many physiological functions including metabolism, response to infection, stress response, blood pressure maintenance and electrolyte homeostasis [86]. Glucocorticoid secretion from the adrenal cortex is regulated by the hypothalamic-pituitary-adrenal axis and its activity is regulated by the local action of metabolizing enzymes [87–89]. Glucocorticoids bind the glucocorticoid receptor (NR3C1) to regulate glucocorticoid responsive gene expression [90]. GR was the first nuclear receptor to be cloned and sequenced [91]. GR is expressed in many tissues throughout development, including the placenta [92–94]. In humans, GR is highly expressed throughout the menstrual cycle in the fibroblasts, endothelial cells and lymphocytes in the stromal compartment, but absent in the glands [95]. In human endometrial stromal cells, glucocorticoids have been shown to inhibit the expression of PRL and stimulate the production of the PRL suppressing lipocortin-1 [96]. Additionally, glucocorticoids have been shown to regulate the production of prostaglandins, critical regulators of stromal vascular permeability during implantation [97]. However, GR null mice die shortly after birth due to respiratory insufficiency thus hindering the study of the role of GR in the adult uterus during implantation [98]. Generating a conditional GR knock-out model will be pivotal to elucidate the uterine specific role of GR in vivo.\\n7. Retinoic Acid Receptors (RAR and RXR)\\nRetinoic acid (RA) is one of the most potent and naturally-occurring metabolite of Vitamin A [99]. RA signaling is involved in embryonic development, growth and reproduction [100, 101]. RAs regulate target gene expression by binding the retinoic acid receptor (RAR) and the retinoic X receptor (RXR) [101]. There are three known subtypes of the RA receptor subfamily, RAR-α (NR1B1), RAR-β (NR1B2) and RAR-γ (NR1B3). Similarly, there are three subtypes of the RXR, RXR-α (NR2B1), RXR-β (NR2B2) and RXR-γ (NR2B3) [102]. Immunohistochemical identification of the RAR and RXR though out the menstrual cycle in women revealed distinct patterns in endometrial compartment and subcellular localization. All RARs can be found in the nuclei of the luminal epithelium during the proliferative phase of the cycle. The staining for RAR in the nuclei was significantly decreased during the secretory phase, but not for RXR. Interestingly, RAR staining in the cytoplasm increased in the secretory phase. Nuclear export of RAR would suggest its activity is predominant during the proliferative phase, while RXR remains active throughout the menstrual cycle. A similar pattern expression for RAR and RXR can be observed in the stromal compartment during the proliferative and secretory phase [103].\\nThe changes in RAR localization are positively correlated with serum estradiol and negatively correlated with progesterone serum levels [104]. Treatment of post-menopausal women with Premarin or a mixture of estrone and equilin sulfates for 3 months results in the up-regulation of retinaldehyde dehydrogenase (RALDH), critical enzyme in RA biosynthesis, RAR-α and RA-regulated genes. Interestingly, the premenopausal endometrium expresses high levels of these RA signaling components during the proliferative phase, while the RA catabolic enzymes retinoic acid 4-hydroxylase (CYP26A1) and tissue transglutaminase are increased in the secretory phase [105]. Similarly, expression of RALDH1, RALDH2 and CYP26A1 has been shown to fluctuate in the murine uterus throughout the estrous cycle and during implantation [54].\\nRA has been shown to inhibit the morphological transformation and suppress the expression of PRL and IGFBP1 in decidualizing HESC. The effect of RA appears to be partially mediated by the inhibition of cAMP signaling under this estrogen and medroxyprogesterone treatment and was only partially compensated by addition of cAMP. Treatment of cultured rat endometrial cells with estrogen increases the production of RA [106]. Additionally, treatment of rats with pregnant mare serum gonadotropin (PMSG), a pharmacological strategy to induce ovulation, results in the induction of RA by the estrogenized uterus. [54] Altogether, this evidence suggests a potential role for RA signaling in the regulation of estrogen-induced proliferation during the window of receptivity. Unfortunately, the double-null mice lacking RARα/RARγ or RARβ/RARγ die before or immediately after birth due to severe skeletal defects, hindering the study of the RAR during peri-implantation in the adult uterus [107, 108].\\n8. Thyroid Hormone Receptor\\nThyroid hormones (TH) are important in processes including growth, development, differentiation, metabolism and reproduction [109]. Hypothyroidism and hyperthyroidism have been known to affect the metabolism of sex hormones, ovarian function and have been associated with menstrual irregularities and infertility [110]. There are two major types of THs, thyroxine (T4), which is the main secretory product and the pro-hormone triiodothyronine (T3) [109]. T3 binds the two major isoforms of the thyroid hormone receptor (TR), TRα (NR1A1) and TRβ (NR1A2). There are two TRα isoforms, TRα1 and TRα2. The TRα2 isoform, however, cannot bind T3 and transactivate TH-responsive genes. Additionally, there are two TRβ isoforms, TRβ1 and TRβ2 [111]. Ablation of the TRβ1 in the mouse (Thrb−/−) results in a hyperactive goiter and elevated thyroid hormone levels.[112] In contrast, ablation of TRα1 in the mouse (TRa1−/−) results in mild hypothyroidism.[113] Unfortunately, disruption of the endocrine system in both of these mouse models hinders the ability to study the specific role of these TRs in endometrial biology.\\nIt has been reported that TRα1, TRα2 and TRβ1 transcripts are regulated by progesterone in the endometrium. Treatment with the anti-progestin RU486 on day LH+8 of the menstrual cycle results in the down-regulation of TRα1 and TRα2 and the up-regulation of TRβ1 [114]. Both protein and message levels of these TRs are detected throughout the menstrual cycle in healthy women. TRα1 is higher in the midsecretory phase while TRα2 was highest in the proliferative phase. TRβ1 is highest in the luminal epithelium in the midsecretory phase and highest in the glandular epithelium in the mid to late secretory phase. In the stromal compartment, TRβ1 was expressed consistently throughout the menstrual cycle. Interestingly, TRβ2 was undetected at the message or protein level by immunostaining, but was present by Western blotting [115]. Collectively, these results underscore the potentially significant, yet largely unknown function of TRs in endometrial receptivity.\\n9. Peroxisome proliferator-activated receptor\\nThe peroxisome proliferator-activated receptor (PPAR) family is composed of three forms, the PPARα (NR1C1) PPARδ, also referred to as PPARβ/δ, (NR1C2) and PPARγ (NR1C3) [116]. These receptors bind several natural and synthetic ligands to regulate lipid and glucose metabolism, energy homeostasis, inflammation and cell proliferation and differentiation [117, 118]. PPAR expression is abundant in tissues like the liver or adipose, critical sites of metabolic regulation. Interestingly, PPARs can also be found in reproductive tissues inducing the hypothalamus, ovary, uterus and placenta in rodents [119]. PPARγ is expressed in the stroma at day 2.5, but subsequently decreases to a low level during implantation. In contrast, PPARα and PPARδ are constantly expressed in the glandular epithelium at days 0.5-6.5 of pregnancy. Both PPARs are present in the endometrial stroma of during days 4.5-6.5 of the rat pregnancy [120]. PPARδ has been observed in the stroma at the site of implantation in mouse and rat [121, 122]. PPARδ has been suggested to play a role in placental function, intrauterine growth restriction (IUGR), inflammatory pathways and mediating COX2-derived prostacyclin (PGI2) signaling during pregnancy [123–126]. PPARα activation has been linked to the negative regulation of lipid peroxidation and nitric oxide (NO) production during placentation [127].\\nGenetic ablation of PPARα in the mice results in an increase in abortion rate. PPARα−/− mice display greater neonatal mortality, indicating an important role for PPARα in placental function [128]. Similarly, PPARδ null mice (PPARβ/δ−/−) display placental abnormalities and embryonic death during early pregnancy due to alteration in cyclo-oxygenase-2-derived prostacyclin mediated implantation [126, 129]. PPARγ null embryos (PPARγ−/−) die very early in development due to defects in labyrinth formation during placentation [130]. Due to the early embryonic lethality of the PPARγ−/− mouse, the role of PPARγ in the adult endometrium remains unknown. It has been shown that the regulation of metabolism for energy homeostasis plays an important role in female reproduction, particularly in the function of the placenta. However, early events in pregnancy also require fine-tuning of metabolism to prepare the uterus for implantation and decidual transformation suggesting a role for the PPAR in the preparation of uterine receptivity. The mechanism by which PPARs regulate early implantation events remain to be fully elucidated.\\n10. The Orphan Nuclear Receptors\\n10.1 COUP-TFII\\nCOUP-TFII (NR2F2), chicken ovalbumin upstream promoter-transcription factor II, is an orphan member of the nuclear receptor family best known for its role in the regulation of angiogenesis during mouse development, glucose metabolism and cancer progression [131]. COUP-TFII is expressed consistently throughout implantation in the endometrial stromal compartment. Conditional ablation of COUP-TFII in the endometrium using the PRCre/+ COUP-TFII f/f (COUP-TFII d/d) bigenic mouse results in infertility due to a defect in blastocyst attachment and absent decidual response. The endometria of COUP-TFII d/d mice exhibited unopposed estrogen signaling in the luminal epithelium, a phenotype similar to that observed after endometrial ablation of Ihh [53]. Interestingly, Ihh was shown to be an upstream regulator of COUP-TFII expression [53]. Consistent with an increase in total ERα and its coactivator SRC-1, a proportional increase in activated phospho-ERα as well as the persistent expression of the estrogen target gene Ltf was observed in COUP-TFII d/d [132]. Furthermore, COUP-TFII d/d displayed a defect in epithelial maturation with a persistent expression of Muc1, glycocalyx and an inappropriate remodeling of the microvilli. Collectively these defects imposed a barrier to implantation and subsequent decidualization [132].\\nCOUP-TFII plays a central role in the mechanism of stromal-epithelial crosstalk during implantation by regulating the WNT and bone-morphogenic-protein (BMP)-pathways. Administration of a pure ERα antagonist ICI 182,780 (ICI) rescued the expression of Bmp2 and Wnt4 in COUP-TFII d/d mice, molecules that are absolutely critical for stromal cell decidualization [133–135]. This resulted in the partial rescue the implantation defect and decidual response. Interestingly, the decidual defect in COUP-TFII d/d could also be rescued by the introduction of recombinant BMP2 into the uterine horn [132]. Similar to COUP-TFII, Bmp2 and Wnt4 are expressed in the stromal compartment during implantation. Conditional ablation of Bmp2 in the uterus using the PRCre/+ Bmp2f/f (Bmp2d/d) bigenic mouse model resulted in infertility. While attachment, proliferation and vascularization were normal in the Bmp2d/d model, endometrial stromal cells were unable to differentiate due to compromised Ptgs1 expression and a decrease in activating PR phosphorylations. The Bmp2 target, Wnt4, was also conditionally ablated in the murine endometrium using the PRcre/+ Wnt4f/f bigenic mouse model. Ablation of Wnt4 results in subfertility due to defects in stromal cell differentiation and progesterone responsiveness. Collectively, these results indicate a central role for COUP-TFII during implantation involving BMP and WNT signaling in the activation of progesterone signaling, regulation of epithelial maturation and stromal preparation for decidualization.\\n10.2 SF-1 and LRH-1\\nThe steroidogenic factor-1 (NR5A1) and the liver receptor homolog-1 (NR5A2) are highly homologous orphan NRs involved in the regulation of steroidogenic genes. Steroidogenic factor-1 (SF-1) expression is predominantly observed in Leydig and Sertoli cells of the testis, granulosa and theca cells of the ovary, all the layers of the adrenal cortex, pituitary and hypothalamus [136, 137]. SF-1 regulates the expression of the enzymes required for steroidogenesis including STAR, aromatase, CYP11A1 and CYP17A1. In the endometrium expression of SF-1 is silenced due to heavy promoter methylation. Therefore, SF-1 it is hypothesized not to play a role in blastocyst implantation. In endometriosis, aberrant demethylation of the SF-1 promoter in stromal cells results in the activation of SF-1 expression and consequently the up-regulation of steroidogenic enzymes. This condition is hypothesized to promote de novo estradiol synthesis and growth of endometriotic lesions [138]. It is known that COUP-TFII provides additional repression of the aromatase promoter in the endometrium [139]. Given that SF-1 is the stronger competitor in this process, is clear why SF-1 must remain silent in order to maintain endometrial homeostasis. The physiological and molecular consequences of activating SF-1 expression in the uterus remain unknown, but would provide insight into the role of SF-1 in endometrial pathologies.\\nLiver receptor homolog-1 (LRH-1) is known to be involved in liver bile acid metabolism, ovarian steroidogenesis and cell proliferation. It is expressed in the granulosa and luteal cells in the ovary where it was found to be critical for the regulation of cumulus expansion, luteinization and ovulation [140]. A recent study evaluated the role of LRH-1 in luteal function using the PRCre/+ NR5A2f/f (NR5A2d/d) bigenic mouse. These mice exhibited luteal insufficiently that resulted in implantation failure. Although implantation failure was rescued by progesterone supplementation, defects in implantation site spacing resulted in compromised placental formation, fetal growth retardation and fetal death. Furthermore, expression of preimplantation genes Hoxa10, Wnt4, Wnt5a, Ihh and Bmp2 was compromised in NR5A2d/d mice and was not rescued by progesterone supplementation. LRH-1 was also shown to be indispensable for mouse and human stromal cell decidualization [141]. Finally, it has been shown that both SF1 and LRH-1 cooperate to activate steroidogenesis in endometrial cancer cell lines, providing a possible mechanism for the observation that de novo estradiol synthesis seem to provide tumor cells with a growth advantage [142]. Collectively these results underscore the need to further evaluate the roles of LRH-1 and SF-1 in uterine function and disease.\\n10.3 Circadian Rhythm: ReverbA and RORC\\nRegulators of the circadian rhythm form part of the endogenous timing system of biological systems. These transcriptional regulators were originally identified in the suprachiasmatic nucleus in the brain but recent evidence demonstrated this system works in individual tissues, including reproductive tissues [143]. Recently, it was reported that progesterone might have a role in the regulation of this circadian biology. Overlapping PR binding with expression data after acute progesterone exposure of ovariectomized mice identified circadian rhythm biology and identified new gene target of PR including Clock, Npas2, Cry1 and Per1, Nr1d1, Rorc and Ppargc1a [59]. Rev-erbα (NR1D1) and RAR-related orphan receptor C (NR1F3) are orphan members of the NR superfamily. Together with the peroxisome proliferator-activated receptor gamma, coactivator 1 alpha (Ppargc1a), these NRs negatively regulate Clock, Npas2, Cry1 and Per1. Interestingly, ReverbA and RORC are negatively regulated by acute progesterone stimulation, while Clock, Npas2, Cry1 and Per1 are up-regulated by progesterone [59]. This evidence underlines the previously unknown role of PR in the regulation of uterine circadian biology. Given that circadian rhythms are known to be important in hypothalamic-pituitary-gonadal axis, it is no surprise to find relationships between disrupted rhythm and menstrual irregularities as well as decrease pregnancy rates [144]. These observations open up a new field examining the role of circadian rhythm in blastocyst receptivity and fertility.',\n",
       "  'Discussion': '',\n",
       "  'Conclusion': '11. Conclusions and Future Directions\\nImplantation in mammals is a very complex process that requires intricate crosstalk between molecules within compartments of the endometrium for the tight regulation of receptivity (Figure 2.)\\nThe ovarian steroids estrogen and progesterone orchestrate the transcriptional programs required for changes in cell cycle progression and differentiation observed in the endometrium during implantation. It is clear that ER and PR play a central role in this process. Future studies should focus on delineating isoform specific targets and functions during implantation and disease. The roles of many of the NRs with endometrial expression have not been fully elucidated due to lack of appropriate in vivo systems. Tissue specific mouse models are currently the most efficient strategy for interrogating molecules that regulated the window of receptivity and have already defined the roles of several important regulators. (Summarized in Table 1.) Conditional mouse models utilizing PRcre, circumvent many of the issue associated with whole body knock-outs and will likely continue to shed light into the roles of NRs and downstream molecules. Significant advances in the understanding of paracrine crosstalk could also be achieved by utilizing compartment specific and inducible Cre models to answer questions about the spatiotemporal roles of these NRs and their effector molecules.\\nFigure 2. Model for Stromal-Epithelial Paracrine Signaling during Implantation.\\nOpen in a new tab\\nExtensive crosstalk between the epithelial and stromal compartments in the endometrium is required for establishing the window of receptivity. Solid arrows (→) indicate activating interactions, (—/) indicate inhibition or repression and dashed arrows represent potential induction that remain to be elucidated.\\nTable 1. Summary of Implantation Phenotypes in Mouse Models.\\nTarget\\nGene\\nMouse Model\\nMouse Uterine Phenotype\\nNR Ablation\\nNr3a1 (Esr1)\\nERαKO\\nInfertile, hypoplastic, no implantation,\\ndecidual response persists with progesterone\\npriming.\\nNr3a2 (Esr2)\\nERβKO\\nNormal fertility uterine phenotype.\\nExaggerated estrogen responsiveness.\\nepithelial Esr1\\nUTEpiαERKO\\nInfertile, loss of lactoferrin expression and\\nenhanced uterine apoptosis.\\nNr3c3 (Pgr)\\nPRKO, PRAKO,\\nPRBKO\\nInfertile, no implantation, absent decidual\\nresponse, unopposed estrogen signaling.\\nPRKO phenotype in PRAKO only.\\nepithelial Pgr\\nWnt7a\\nCre/+\\nPR\\nf/f\\nInfertile, no implantation, absent decidual\\nresponse, unopposed estrogen signaling.\\nNr3c4 (AR)\\nARKO\\nReduced litter size, smaller uterus and\\nreduced growth in response to hormones,\\nabnormal placental development and\\nplacentamegaly.\\nNr1c1\\n(PPARα)\\nPPARα\\n−/−\\nIncrease in abortion rate and greater\\nneonatal mortality.\\nNr1c2\\n(PPARβ/δ)\\nPPARβ/δ\\n−/−\\nPlacental abnormalities and embryonic death\\nduring early pregnancy due to alteration in\\ncyclo-oxygenase-2-derived prostacyclin.\\nNr1c3\\n(PPARγ)\\nPPARγ\\n−/−\\nEarly death of embryo due to defects in\\nlabyrinth formation during placentation.\\nNr2f2\\n(COUP-TFII)\\nPR\\nCre/+\\nCOUP-\\nTFII\\nf/f\\nInfertile, no implantation, absent decidual\\nresponse, unopposed estrogen signaling in\\nluminal epithelium.\\nNr5a2\\n(LRH-1)\\nPR\\nCre/+\\nNR5A2\\nf/f\\nInfertile, luteal insufficiency, implantation\\nfailure, defect decidualization and\\nplacentation.\\nNR Target Ablation\\nIhh\\nPR\\nCre/+\\nIhh\\nf/f\\nInfertile, absent decidual response,\\nunopposed estrogen signaling in luminal\\nepithelium\\nBmp2\\nPR\\nCre/+\\nBmp2\\nf/f\\nInfertile, decrease Cox2, absent decidual\\nresponse.\\nWnt4\\nPR\\nCre/+\\nWnt4\\nf/f\\nSubfertile, reduced number of glands,\\nsquamous cell metaplasia in response to\\nprolonged estrogen treatment, defective\\ndecidualization and progesterone response.\\nLif\\nLif\\n−/−\\nInfertile, no implantation, embryos remain\\ndormant.\\nHand2\\nPR\\nCre/+\\nHand2\\nf/f\\nInfertile, no implantation, altered\\nprogesterone responsiveness, persistent\\nluminal epithelial proliferation.\\nOpen in a new tab\\nThe use of ChIP-Seq in combination with expression profiling techniques such as microarray and RNA-Seq have been very powerful tools in the identification of chromatin interactions and direct transcriptional targets of NRs. It is no doubt the field of uterine biology will continue to reap the benefits of the advances in high content technology for the better understanding of NR action in blastocyst implantation. The ultimate goal of these efforts is to identify molecular markers of uterine receptivity. The knowledge of these will facilitate efforts in assisted reproductive clinics and improve the outlook of reproductively challenged couples. Additionally, many of the signaling pathways involved in regulating implantation are disrupted in gynecological malignancies such as endometriosis and endometrial cancer. A greater understanding of the role of hormone and nuclear receptor signaling in the regulation of proliferation and differentiation in the endometrium could identify more effective strategies for the treatment of these pathological conditions.',\n",
       "  'URL': 'https://doi.org/10.1016/j.semcdb.2013.08.004',\n",
       "  'Journal': 'Semin Cell Dev Biol'},\n",
       " {'PaperTitle': 'KEGG: Kyoto Encyclopedia of Genes and Genomes',\n",
       "  'Publication': '2000',\n",
       "  'Authors': 'Minoru Kanehisa, Susumu Goto',\n",
       "  'Email': 'kanehisa@kuicr.kyoto-u.ac.jp',\n",
       "  'Abstract': 'KEGG (Kyoto Encyclopedia of Genes and Genomes) is a knowledge base for systematic analysis of gene functions, linking genomic information with higher order functional information. The genomic information is stored in the GENES database, which is a collection of gene catalogs for all the completely sequenced genomes and some partial genomes with up-to-date annotation of gene functions. The higher order functional information is stored in the PATHWAY database, which contains graphical representations of cellular processes, such as metabolism, membrane transport, signal transduction and cell cycle. The PATHWAY database is supplemented by a set of ortholog group tables for the information about conserved subpathways (pathway motifs), which are often encoded by positionally coupled genes on the chromosome and which are especially useful in predicting gene functions. A third database in KEGG is LIGAND for the information about chemical compounds, enzyme molecules and enzymatic reactions. KEGG provides Java graphics tools for browsing genome maps, comparing two genome maps and manipulating expression maps, as well as computational tools for sequence comparison, graph comparison and path computation. The KEGG databases are daily updated and made freely available (http://www.genome.ad.jp/kegg/ ).',\n",
       "  'Introduction': 'While the genome sequencing projects rapidly determine gene catalogs for an increasing number of organisms, functional annotation of individual genes is still largely incomplete. KEGG (Kyoto Encyclopedia of Genes and Genomes) is an effort to link genomic information with higher order functional information by computerizing current knowledge on cellular processes and by standardizing gene annotations. Generally speaking, the biological function of the living cell is a result of many interacting molecules; it cannot be attributed to just a single gene or a single molecule ( 1 ). The functional assignment in KEGG is a process of linking a set of genes in the genome with a network of interacting molecules in the cell, such as a pathway or a complex, representing a higher order biological function.\\nThe KEGG project was initiated in May 1995 under the Human Genome Program of the Ministry of Education, Science, Sports and Culture in Japan ( 2 ). All the data in KEGG and associated software tools are made available as part of the Japanese GenomeNet service ( 3 ). KEGG consists of three databases: PATHWAY for representation of higher order functions in terms of the network of interacting molecules, GENES for the collection of gene catalogs for all the completely sequenced genomes and some partial genomes, and LIGAND ( 4 ) for the collection of chemical compounds in the cell, enzyme molecules and enzymatic reactions. The overall architecture of the KEGG system is basically the same as previously reported ( 5 ). The user may enter the KEGG system top-down starting from the pathway (functional) information or bottom-up starting from the genomic information at the KEGG table of contents page (http://www.genome.ad.jp/kegg/kegg2.html ).',\n",
       "  'Methods': 'GENOMIC INFORMATION\\nGENES database\\nThe current status of the KEGG databases is summarized in Table 1 . During the past year, we have made every effort to keep up with the data increase of complete genome sequences, and also with the imminent data explosion of gene expression profiles. The number of GENES entries for just 29 species—human, mouse, Drosophila, Arabidopsis, Schizosaccharomyce pombe, and 24\\xa0completely sequenced genomes—totals ~110 000 entries, which is already larger than the number of entries in the well-annotated SWISS-PROT database ( 6 ). The GENES database contains the bare minimum information for each gene as shown in Table 2 , but it is intended to be a resource containing up-to-date, standardized descriptions of gene functions. GENES also serves as a gateway to a number of other resources containing more detailed information.\\nTable 1. The summary of KEGG release 12.0 (October 1999).\\nDatabase\\nContent\\nPATHWAY\\n2706 entries for pathway diagrams constructed from 143 manually drawn diagrams\\nGENES\\n110 018 entries in 24 complete genomes and 12 partial genomes\\nLIGAND\\n5645 entries in the COMPOUND section\\n3705 entries in the ENZYME section\\n5207 reactions in the REACTION section\\nAuxiliary data\\nContent\\nOrtholog group table\\n61 tables\\nGenome map\\n23 complete genomes and one partial genome\\nComparative genome map\\n23 × 23 complete genome comparisons\\nExpression map\\nFour sets of expression maps\\nGene catalog\\n53 catalogs\\nMolecular catalog\\nEight catalogs\\nDisease catalog\\nThree catalogs\\nOpen in a new tab\\nTable 2. The data content of the GENES database entry.\\nField\\nContent\\nLinks\\nData source\\nENTRY\\nEntry identifier (gene accession number)\\nLinkDB database\\nGenBank or original database\\nNAME\\nGene names and alternative names\\nGenBank or original database\\nDEFINITION\\nAnnotation of gene function\\nLIGAND/ENZYME database, SWISS-PROT database and PubMed database\\nGenBank, original database, SWISS-PROT and KEGG\\nCLASS\\nClassification of genes according to the KEGG pathways\\nKEGG/PATHWAY database\\nKEGG\\nPOSITION\\nChromosomal position\\nKEGG/GENOME map\\nGenBank\\nDBLINKS\\nOutside links\\nOriginal databases and NCBI Entrez database\\nCODON_USAGE\\nCodon usage\\nComputed\\nAASEQ\\nAmino acid sequence\\nsee footnote\\na\\nGenBank or original database\\nNTSEQ\\nNucleotide sequence\\nGenBank or original database\\nOpen in a new tab\\na\\nComputational links are available including sequence similarity searches (FASTA and BLAST), motif search (MOTIF), membrane protein predictions (SOSUI and TSEG), and cellular localization site prediction (PSORT).\\nWe developed various computational tools for the maintenance of the GENES database, especially for extraction of information from GenBank, which is not a trivial task, and for assisting systematic annotation of gene functions. The overall flow of both computerized and manual processes is illustrated in Figure 1 . A web-based annotation tool is used, together with other computational tools, to assign EC numbers, to assign ortholog identifiers, to incorporate new experimental evidence from literature, and to annotate predictions based on pathway construction. As described below, the ortholog identifiers will be used as primary keys for automatic matching of genes in the genome and gene products in the pathway.\\nFigure 1.\\nOpen in a new tab\\nProcedures used to organize and annotate the GENES database.\\nGene expression profiles\\nThe backbone retrieval system for the GENES database is the DBGET/LinkDB system ( 7 ), but there are additional ways of accessing this database. One is the Java-based genome map browser for graphical manipulation of gene positions on the chromosome. The other is what we call the hierarchical text browser for handling functional hierarchy of gene catalogs. Here we report another Java graphics browser, the expression map browser, for analysis of gene expression profiles obtained by cDNA microarray or oligonucleotide array experiments. The vast amount of data generated by such functional genomics experiments are likely to contain valuable information, which will supplement genomic sequence information toward understanding higher biological functions of the cell. A preliminary version of the expression map browser is linked to both the KEGG pathway data and the genome map data, so that the user may examine if, for example, a group of co-regulated genes are also correlated in the pathway or are encoded in a cluster of genes on the chromosome.\\nPATHWAY INFORMATION\\nPATHWAY database\\nCurrently the best organized part of the KEGG/PATHWAY database is metabolism, which is represented by ~90 graphical diagrams for the reference metabolic pathways. Each reference pathway can be viewed as a network of enzymes or a network of EC numbers. Once enzyme genes are identified in the genome based on sequence similarity and positional correlation of genes, and the EC numbers are properly assigned, organism-specific pathways can be constructed computationally by correlating genes in the genome with gene products (enzymes) in the reference pathways according to the matching EC numbers. We are trying to extend this mechanism to include various regulatory pathways, such as signal transduction, cell cycle and apoptosis.\\nBecause the metabolic pathway, especially for intermediary metabolism, is well conserved among most organisms from mammals to bacteria, it is possible to manually draw one reference pathway and then to computationally generate many organism-specific pathways. In contrast, the regulatory pathways are far more divergent and are difficult to combine into common reference pathway diagrams. Thus, we basically draw a pathway diagram separately for each organism. At the same time, we are trying to identify groups of organisms that share common pathways or assemblies and whose diagrams may be combined. Examples include one common apoptosis pathway diagram for human and mouse, three ribosome assembly diagrams separately for bacteria, archaea and eukaryotes.\\nThe other related problem is the absence of proper identifiers for functions in the regulatory pathways. The EC numbers in the metabolic pathways play roles as identifiers of the nodes (enzymes) and also as keys for linking with the genomic information. We are preparing for the introduction of the ortholog identifiers to extend such capabilities of the EC numbers. The ortholog identifiers will be used to identify nodes (proteins) in the regulatory pathways and also to link with the genomic information. In addition, the ortholog identifiers will replace the EC numbers in the metabolic pathways in order to distinguish multiple genes that match one EC number, for example, different subunits of an enzyme complex or different genes expressed under different conditions.\\nOrtholog group tables\\nOrthologs are identified in KEGG not only by sequence similarity of individual genes but also by examining if all constituent members are found for a functional group, such as a conserved subpathway or a molecular complex. The KEGG ortholog group table is a representation of three features: whether an organism contains a complete set of genes that constitutes a functional group, whether those genes are physically coupled on the chromosome, and what are orthologous genes among different organisms. Currently there are 61 ortholog group tables, which contain, for example, a gene cluster in the genome coding for a functionally related enzyme cluster in the metabolic pathway. In KEGG such correlated clusters are first detected by a heuristic graph comparison algorithm, and then manually edited and compiled into the ortholog group tables. There are two types of graph comparisons that we use: genome–pathway and genome–genome comparisons ( 1 ). An ortholog group table is a composite of such pairwise comparisons, representing a conserved portion of the pathway, or what we call a pathway motif.\\nGeneralized protein–protein interaction\\nThe KEGG pathway representation focuses on the network of gene products, mostly proteins but including functional RNAs. As illustrated in Figure 2 , the metabolic pathway is a network of indirect protein–protein interactions, which is actually a network of enzyme–enzyme relations. In contrast, the regulatory pathway often consists of direct protein–protein interactions, such as binding and phosphorylation, and another class of indirect protein–protein interactions, which are relations of transcription factors and transcribed gene products via gene expressions. The generalized protein–protein interaction network that includes these three types of interactions is an abstract network, but it is especially useful to link with genomic information because the nodes (gene products) of this network can be directly correlated with the nodes (genes) in the genome. With this concept of generalized protein–protein interaction network, we are expanding the collection of manually drawn reference pathway diagrams.\\nFigure 2.\\nOpen in a new tab\\nThe generalized protein–protein interaction includes an indirect protein–protein interaction by two successive enzymes, a direct protein–protein interaction, and another indirect protein–protein interaction by gene expression. The nodes of the generalized protein–protein interaction network are gene products, which can be directly correlated with genes in the genome.',\n",
       "  'Results': 'GENOMIC INFORMATION\\nGENES database\\nThe current status of the KEGG databases is summarized in Table 1 . During the past year, we have made every effort to keep up with the data increase of complete genome sequences, and also with the imminent data explosion of gene expression profiles. The number of GENES entries for just 29 species—human, mouse, Drosophila, Arabidopsis, Schizosaccharomyce pombe, and 24\\xa0completely sequenced genomes—totals ~110 000 entries, which is already larger than the number of entries in the well-annotated SWISS-PROT database ( 6 ). The GENES database contains the bare minimum information for each gene as shown in Table 2 , but it is intended to be a resource containing up-to-date, standardized descriptions of gene functions. GENES also serves as a gateway to a number of other resources containing more detailed information.\\nTable 1. The summary of KEGG release 12.0 (October 1999).\\nDatabase\\nContent\\nPATHWAY\\n2706 entries for pathway diagrams constructed from 143 manually drawn diagrams\\nGENES\\n110 018 entries in 24 complete genomes and 12 partial genomes\\nLIGAND\\n5645 entries in the COMPOUND section\\n3705 entries in the ENZYME section\\n5207 reactions in the REACTION section\\nAuxiliary data\\nContent\\nOrtholog group table\\n61 tables\\nGenome map\\n23 complete genomes and one partial genome\\nComparative genome map\\n23 × 23 complete genome comparisons\\nExpression map\\nFour sets of expression maps\\nGene catalog\\n53 catalogs\\nMolecular catalog\\nEight catalogs\\nDisease catalog\\nThree catalogs\\nOpen in a new tab\\nTable 2. The data content of the GENES database entry.\\nField\\nContent\\nLinks\\nData source\\nENTRY\\nEntry identifier (gene accession number)\\nLinkDB database\\nGenBank or original database\\nNAME\\nGene names and alternative names\\nGenBank or original database\\nDEFINITION\\nAnnotation of gene function\\nLIGAND/ENZYME database, SWISS-PROT database and PubMed database\\nGenBank, original database, SWISS-PROT and KEGG\\nCLASS\\nClassification of genes according to the KEGG pathways\\nKEGG/PATHWAY database\\nKEGG\\nPOSITION\\nChromosomal position\\nKEGG/GENOME map\\nGenBank\\nDBLINKS\\nOutside links\\nOriginal databases and NCBI Entrez database\\nCODON_USAGE\\nCodon usage\\nComputed\\nAASEQ\\nAmino acid sequence\\nsee footnote\\na\\nGenBank or original database\\nNTSEQ\\nNucleotide sequence\\nGenBank or original database\\nOpen in a new tab\\na\\nComputational links are available including sequence similarity searches (FASTA and BLAST), motif search (MOTIF), membrane protein predictions (SOSUI and TSEG), and cellular localization site prediction (PSORT).\\nWe developed various computational tools for the maintenance of the GENES database, especially for extraction of information from GenBank, which is not a trivial task, and for assisting systematic annotation of gene functions. The overall flow of both computerized and manual processes is illustrated in Figure 1 . A web-based annotation tool is used, together with other computational tools, to assign EC numbers, to assign ortholog identifiers, to incorporate new experimental evidence from literature, and to annotate predictions based on pathway construction. As described below, the ortholog identifiers will be used as primary keys for automatic matching of genes in the genome and gene products in the pathway.\\nFigure 1.\\nOpen in a new tab\\nProcedures used to organize and annotate the GENES database.\\nGene expression profiles\\nThe backbone retrieval system for the GENES database is the DBGET/LinkDB system ( 7 ), but there are additional ways of accessing this database. One is the Java-based genome map browser for graphical manipulation of gene positions on the chromosome. The other is what we call the hierarchical text browser for handling functional hierarchy of gene catalogs. Here we report another Java graphics browser, the expression map browser, for analysis of gene expression profiles obtained by cDNA microarray or oligonucleotide array experiments. The vast amount of data generated by such functional genomics experiments are likely to contain valuable information, which will supplement genomic sequence information toward understanding higher biological functions of the cell. A preliminary version of the expression map browser is linked to both the KEGG pathway data and the genome map data, so that the user may examine if, for example, a group of co-regulated genes are also correlated in the pathway or are encoded in a cluster of genes on the chromosome.\\nPATHWAY INFORMATION\\nPATHWAY database\\nCurrently the best organized part of the KEGG/PATHWAY database is metabolism, which is represented by ~90 graphical diagrams for the reference metabolic pathways. Each reference pathway can be viewed as a network of enzymes or a network of EC numbers. Once enzyme genes are identified in the genome based on sequence similarity and positional correlation of genes, and the EC numbers are properly assigned, organism-specific pathways can be constructed computationally by correlating genes in the genome with gene products (enzymes) in the reference pathways according to the matching EC numbers. We are trying to extend this mechanism to include various regulatory pathways, such as signal transduction, cell cycle and apoptosis.\\nBecause the metabolic pathway, especially for intermediary metabolism, is well conserved among most organisms from mammals to bacteria, it is possible to manually draw one reference pathway and then to computationally generate many organism-specific pathways. In contrast, the regulatory pathways are far more divergent and are difficult to combine into common reference pathway diagrams. Thus, we basically draw a pathway diagram separately for each organism. At the same time, we are trying to identify groups of organisms that share common pathways or assemblies and whose diagrams may be combined. Examples include one common apoptosis pathway diagram for human and mouse, three ribosome assembly diagrams separately for bacteria, archaea and eukaryotes.\\nThe other related problem is the absence of proper identifiers for functions in the regulatory pathways. The EC numbers in the metabolic pathways play roles as identifiers of the nodes (enzymes) and also as keys for linking with the genomic information. We are preparing for the introduction of the ortholog identifiers to extend such capabilities of the EC numbers. The ortholog identifiers will be used to identify nodes (proteins) in the regulatory pathways and also to link with the genomic information. In addition, the ortholog identifiers will replace the EC numbers in the metabolic pathways in order to distinguish multiple genes that match one EC number, for example, different subunits of an enzyme complex or different genes expressed under different conditions.\\nOrtholog group tables\\nOrthologs are identified in KEGG not only by sequence similarity of individual genes but also by examining if all constituent members are found for a functional group, such as a conserved subpathway or a molecular complex. The KEGG ortholog group table is a representation of three features: whether an organism contains a complete set of genes that constitutes a functional group, whether those genes are physically coupled on the chromosome, and what are orthologous genes among different organisms. Currently there are 61 ortholog group tables, which contain, for example, a gene cluster in the genome coding for a functionally related enzyme cluster in the metabolic pathway. In KEGG such correlated clusters are first detected by a heuristic graph comparison algorithm, and then manually edited and compiled into the ortholog group tables. There are two types of graph comparisons that we use: genome–pathway and genome–genome comparisons ( 1 ). An ortholog group table is a composite of such pairwise comparisons, representing a conserved portion of the pathway, or what we call a pathway motif.\\nGeneralized protein–protein interaction\\nThe KEGG pathway representation focuses on the network of gene products, mostly proteins but including functional RNAs. As illustrated in Figure 2 , the metabolic pathway is a network of indirect protein–protein interactions, which is actually a network of enzyme–enzyme relations. In contrast, the regulatory pathway often consists of direct protein–protein interactions, such as binding and phosphorylation, and another class of indirect protein–protein interactions, which are relations of transcription factors and transcribed gene products via gene expressions. The generalized protein–protein interaction network that includes these three types of interactions is an abstract network, but it is especially useful to link with genomic information because the nodes (gene products) of this network can be directly correlated with the nodes (genes) in the genome. With this concept of generalized protein–protein interaction network, we are expanding the collection of manually drawn reference pathway diagrams.\\nFigure 2.\\nOpen in a new tab\\nThe generalized protein–protein interaction includes an indirect protein–protein interaction by two successive enzymes, a direct protein–protein interaction, and another indirect protein–protein interaction by gene expression. The nodes of the generalized protein–protein interaction network are gene products, which can be directly correlated with genes in the genome.',\n",
       "  'Discussion': '',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1093/nar/28.1.27',\n",
       "  'Journal': 'Nucleic Acids Res'},\n",
       " {'PaperTitle': 'Role of the GATA Family of Transcription Factors in Endocrine Development, Function, and Disease',\n",
       "  'Publication': '2008',\n",
       "  'Authors': 'Robert S Viger, Séverine Mazaud Guittot, Mikko Anttonen, David B Wilson, Markku Heikinheimo',\n",
       "  'Email': 'robert.viger@crchul.ulaval.ca',\n",
       "  'Abstract': 'The WGATAR motif is a common nucleotide sequence found in the transcriptional regulatory regions of numerous genes. In vertebrates, these motifs are bound by one of six factors (GATA1 to GATA6) that constitute the GATA family of transcriptional regulatory proteins. Although originally considered for their roles in hematopoietic cells and the heart, GATA factors are now known to be expressed in a wide variety of tissues where they act as critical regulators of cell-specific gene expression. This includes multiple endocrine organs such as the pituitary, pancreas, adrenals, and especially the gonads. Insights into the functional roles played by GATA factors in adult organ systems have been hampered by the early embryonic lethality associated with the different Gata-null mice. This is now being overcome with the generation of tissue-specific knockout models and other knockdown strategies. These approaches, together with the increasing number of human GATA-related pathologies have greatly broadened the scope of GATA-dependent genes and, importantly, have shown that GATA action is not necessarily limited to early development. This has been particularly evident in endocrine organs where GATA factors appear to contribute to the transcription of multiple hormone-encoding genes. This review provides an overview of the GATA family of transcription factors as they relate to endocrine function and disease.',\n",
       "  'Introduction': 'GATA FACTORS ARE a group of evolutionarily conserved transcriptional regulators that play crucial roles in the development and differentiation of all eukaryotic organisms. In vertebrates, the GATA family comprises six members (named GATA1 to -6) that can be separated into two subgroups based on spatial and temporal expression patterns. GATA1/2/3 are expressed in hematopoietic cell lineages and are essential for erythroid and megakaryocyte differentiation, the proliferation of hematopoietic stem cells, and the development of T lymphocytes (1). Their expression, however, is not limited to hematopoietic cells but is also present in the brain, spinal cord, and inner ear where they play important roles in the development of these structures (2, 3, 4). In contrast, the GATA4/5/6 proteins are mainly found in tissues of mesodermal and endodermal origin such as the heart, gut, and gonads (5). Disruption of the Gata4 and Gata6 genes in mice results in early embryo lethality due to defects in heart tube formation and extraembryonic endoderm development, respectively (6, 7, 8, 9). Although loss of GATA5 function is not lethal, female Gata5-null mice exhibit genitourinary abnormalities (10). The physiological requirement for GATA function in humans is further supported by the more recent identification of GATA gene mutations associated with human diseases, such as dyserythropoietic anemia and thrombocytopenia (GATA1 mutation); hypoparathyroidism, deafness, and renal dysplasia (HDR syndrome; GATA3 mutation); and congenital heart defects (GATA4 mutation) (11, 12, 13). All vertebrate GATA proteins contain a conserved DNA-binding domain composed of two multifunctional zinc fingers involved in DNA-binding and protein-protein interaction with other transcriptional partners and/or cofactors (Fig. 1). Because members of the GATA family share a highly conserved DNA-binding domain, there is an apparent redundancy in DNA binding properties to target GATA sequences (14, 15). These in vitro properties contrast, however, with their often nonredundant roles in vivo (6, 7, 9, 16, 17, 18). The specificity of GATA action is governed, in part, through protein-protein interactions with other transcriptional partners. Indeed, there is now an extensive list of ubiquitously expressed or cell-restricted factors that are known to cooperate with GATA factors to control tissue-specific transcription in the hematopoietic system, the heart, and many other tissues (19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45). The activity of GATA factors has also been shown to be modulated by posttranslational modifications such as sumoylation, acetylation, and phosphorylation. The effect of these modifications most often involves enhanced transcriptional activity due to changes in GATA nuclear localization, DNA-binding, protein stability, and/or cofactor recruitment (see Table 1 for references). With respect to endocrine cells, GATA4 phosphorylation has been the best studied posttranslational modification to date (described further below). Although originally classified as markers of hematopoietic and cardiac cells, the expression pattern and functions attributed to GATA factors extend well beyond these cell lineages. Interestingly, GATA factors have been described in most, if not all, of the endocrine organs where they are emerging as important regulators of numerous genes coding for hormones or proteins/enzymes involved in hormone biosynthesis. A comprehensive list of putative GATA-regulated genes in endocrine tissues is presented in Table 2. In the sections that follow, our current understanding of the expression and role of GATA factors in different endocrine organs is presented.',\n",
       "  'Methods': '',\n",
       "  'Results': 'GATA FACTORS IN ENDOCRINE TISSUES Hypothalamus and Pituitary The expression of Gata4 in migrating GnRH neurons at embryonic d 13.5 (e13.5) but not in mature GnRH-secreting neurons in mouse has led to the hypothesis that GATA factors could play a role in the differentiation of the small population of neurosecretory cells that mediate central nervous system control of reproduction (46). In hypothalamic neuronal GT1-7 cells, GATA4 activates GnRH (Gnrh1) transcription through binding to two GATA motifs arranged in tandem repeat within the GnRH neuron-specific enhancer (46). Because the GATA-binding site is also involved in Gnrh1 gene regulation in the mature hypothalamus, the role for GATA in Gnrh1 expression would not appear to be limited to cell differentiation. The absence of GATA4 binding activity in adult hypothalamus nuclear extracts does not exclude the possibility of the binding of another, yet to be identified, GATA family member or some other transcriptional regulator (46). The mature pituitary gland orchestrates the homeostasis of key endocrine organs by regulating their function in response to signals coming from brain and the periphery. Among the six adult differentiated hormone-secreting cell types, both thyrotrope and gonadotrope cells express GATA2 from e10.5 onward, whereas GATA3 is transiently expressed during development (47). The secretory products of thyrotrope and gonadotrope cells are heterodimers that share a common α-glycoprotein subunit (αGSU) and a specific β-subunit (FSHβ, LHβ, and thyrotropin-β). Interestingly, genes encoding αGSU (Cga) and thyrotropin-β (Tshb) are GATA2 target genes (21, 48, 49). Moreover, although a functional GATA binding site has been described in the GnRH receptor promoter in pituitary cells, the GATA factor that binds has not yet been identified (50). GATA2 is involved in both gonadotrope and thyrotrope terminal cell specification. The detailed analysis of several transgenic mouse models such as expression of a dominant-negative form of GATA2 under the control of Cga (αGSU) regulatory sequences and overexpression of GATA2 directed by the Pou1f1 (Pit1) promoter, have highlighted the complex mechanisms whereby interactions between Pit1 and GATA2 contribute to pituitary cell fate determination (47, 51). In gonadotropes where GATA2 is expressed in the absence of Pit1, GATA2 promotes the expression of gonadotrope-specific genes. In thyrotropes, where GATA2 and Pit1 are coexpressed, thyrotrope-specific genes are up-regulated by the binding of both factors to adjacent DNA cis-elements, whereas GATA motif-containing gonadotrope-specific genes are down-regulated. This discrepancy comes from the ability of Pit1 to interact with a critical DNA-binding zinc finger of GATA2, via its homeodomain, thus preventing GATA2 DNA-binding and target gene transactivation (47). Because of the early embryonic lethality of Gata2-null mice resulting from severe anemia consistent with the critical role of GATA2 in both primitive hematopoiesis of the yolk sac and definitive hematopoiesis of the liver (18), the role of GATA2 in later development was obscured. However, a recent study reporting pituitary-specific Gata2 knockout mice has shed further light into its role in pituitary function (52). Loss of GATA2 in the anterior pituitary results in a decrease in gonadotrope and thyrotrope cell numbers at birth and a marked decrease in secretory capacity of these cells in the adult (52). Although Gata2 pituitary knockout mice have defects in the same pituitary cell populations as the dominant-negative GATA2 transgenic mice (47), the extent of the defect is much less severe (52). This difference has been proposed to be due to compensation by GATA3 that is up-regulated in the Gata2 pituitary knockout mice and that would normally be targeted by the GATA2 dominant-negative. Thus, GATA2 has been proposed to have a dual role in pituitary function: the initial specification and/or proliferation of thyrotropes and the later maintenance of hormone expression in mature thyrotropes and gonadotropes (52). Consistent with this hypothesis, GATA2 is found in most αGSU-positive and thyrotropin-secreting human pituitary adenomas (53). Thyroid and Parathyroid The parathyroid glands regulate calcium balance in the body through the secretion and action of PTH. Developmentally, parathyroid glands, together with the thyroid, thymus, and ultimobranchial bodies, originate from the pharyngeal region by the complex interactions between endoderm and neural crest derivatives (54). GATA3 expression has been reported in human embryos in the second and third branchial arches, from which the parathyroids derive (55). Although a consensus GATA binding site has been identified in the proximal promoter of the murine PTH (Pth) gene (56), its functional relevance remains unclear, and to date, no other potential GATA3 target genes have been identified in the developing parathyroid. Homozygous Gata3 knockout mice display mid-gestation lethality (16), in part due to a noradrenaline deficiency of the sympathetic nervous system, which can be rescued by pharmacological treatment with catechol intermediates (57). Because of this early embryonic lethality, the generation of parathyroid-conditional Gata3 transgenic mice would therefore be of particular interest to gain insights into the role of GATA3 in adult parathyroid gland physiology. However, the importance of GATA3 in parathyroid function has not come from transgenic mice but rather from study of patients suffering hypoparathyroidism, sensorineural deafness, and renal anomaly (HDR) syndrome. HDR syndrome has been linked to deletions or mutations of GATA3 that affect the integrity of its two zinc fingers (13, 58, 59, 60, 61, 62, 63). Whereas mutations involving the first zinc finger result in either a loss of interaction with Friend of GATA (FOG) transcriptional partners or altered DNA-binding affinity, those involving the GATA3 second zinc finger or adjacent basic amino acids result in a loss of DNA-binding (58). Although no GATA factor has yet been reported to regulate thyroid development, at least two thyroid-expressed genes, Nkx2–5 and Nkx2–1, have been shown to be under GATA control in the heart and lung, respectively (64, 65). Therefore, a similar regulation in the thyroid would not be unlikely. Pancreas The mammalian pancreas is a mixed exocrine and endocrine gland that is formed from the fusion of dorsal and ventral buds of the foregut endoderm. In recent years, our understanding of pancreatic development has benefited from the identification of essential transcription factors implicated in this process such as pancreatic and duodenal homeobox 1 (PDX1) and neurogenin 3 (NEUROG3) (66, 67). The cardiac subfamily of GATA factors are well-known markers of several endodermal derivatives including yolk sac, lung, stomach, gut, and pancreas (5). GATA4 and GATA6 have overlapping expression profiles in the early pancreas (68, 69). GATA4 then becomes limited to exocrine tissue and GATA6 to a subpopulation of endocrine cells (68). GATA6, but not GATA4, has also been shown to interact with Nkx2.2 (68), a critical islet transcription factor. In the same study, transgenic mice expressing a GATA4- or GATA6-Engrailed dominant repressor fusion protein in the pancreatic epithelium and in the islet revealed an important role for GATA6, but not GATA4, in early pancreas specification (68). Expression of the GATA6-Engrailed transgene had two distinct phenotypes: agenesis of the pancreas or a reduction in pancreatic tissue. In the latter, all pancreatic cell types were significantly reduced, having few differentiated endocrine cells within enlarged ductal structures (68). In contrast to these findings, others have reported that GATA4 colocalizes with early glucagon-positive cells as early as e12.5 in the mouse and that GATA4 is able to transactivate the glucagon (Gcg) gene promoter in vitro (70). The same study also demonstrated that mutation of the GATA binding element in the Gcg promoter reduces its basal promoter activity in glucagon producing cells by 55%. In another study, the development of the ventral but not dorsal pancreas is completely absent in Gata4−/− embryos, and Gata6−/− embryos display a similar but less dramatic phenotype (71). GATA4 has been found to be frequently overexpressed and infrequently methylated in human pancreatic cancers, whereas GATA5 is likely hypermethylated during pancreatic cancer development (72). Testis Reproductive tissues are prominent sites of GATA expression. In the gonads, GATA-like proteins are evolutionarily conserved because they are found in a variety species ranging from worms to humans (73, 74). In mammals, four GATA factors exhibit gonadal expression (GATA1/2/4/6), and with the exception of GATA2 (75), this expression is exclusive to the somatic cell population (73, 74) (Fig. 2). The testis contains two major somatic cell types that facilitate spermatogenesis: Sertoli cells, which are the supporting cell lineage, and Leydig cells, the steroidogenic cell lineage. GATA factors have been described in both cell types in several species starting from the early stages of testis differentiation to adult cells (74). Sertoli cells express GATA4 from the onset of their differentiation through to adulthood where its level of expression is considered to be constant regardless of the stage of germ cell maturation (76, 77, 78, 79, 80, 81, 82, 83). GATA6 is also present early and is maintained in adult Sertoli cells (78, 80, 84). Finally, a third GATA factor, GATA1, is expressed by Sertoli cells, but the temporal expression of GATA1 in this lineage differs from that of other GATA members. GATA1 is first detected in the immature animal in Sertoli cells of all the tubules, concomitantly with the first wave of spermatogenesis (83, 85). In the adult animal, GATA1 expression in Sertoli cells in stages VI–IX of the seminiferous epithelium cycle coincides with the androgen-sensitive VII–VIII stages and appears to be dependent on the presence of maturing germ cells (79, 85). Although both GATA4 and GATA6 have been described in Leydig cells (78, 83, 84), GATA4 is by far the predominant GATA factor of this cell type, especially in rodents. GATA4 has also been reported in Sertoli and Leydig cell tumors (86). Because of the critical role of GATA factors in hematopoietic and cardiac development and the corresponding early embryonic lethality associated with Gata-null mice, alternative strategies have had to be developed to unravel their function in testis differentiation and physiology. Conditional knockout of GATA1 in Sertoli cells failed to yield a significant phenotype, most likely due to the redundancy of multiple GATA factors in this cell type (87). Other approaches, however, have begun to give clues into the roles of GATA4 in testis cell differentiation. For example, mice homozygous for a GATA4 variant that cannot interact with its cofactor FOG2 (Gata4ki/ki) or homozygous for a FOG2-null mutant (Fog2−/−) exhibit abnormal testis morphogenesis resulting from a failure of Sertoli cells to differentiate (88). On a molecular level, this appears to be due to a significant reduction in transcription of the testis-determining gene Sry (88), as well as a block in Sox9 expression (89, 90). Whether Sry and Sox9 are direct targets of a GATA4/FOG2 complex, however, remains to be demonstrated. The Gata4ki/ki and Fog2−/− mice also fail to express anti-Müllerian hormone (Amh) as well as several steroidogenic markers [cytochrome P450 side chain cleavage (Cyp11a1), cytochrome P450 17α-hydroxylase/17,20 lyase (Cyp17a1), and 3β-hydroxysteroid dehydrogenase type 1 (Hsd3b1)], suggesting that steroidogenic cell differentiation and/or gene expression is equally under GATA4/FOG2 control (88). Indeed, a recent study using chimeric mice has shown that that GATA4 is directly required for Leydig cell differentiation in the fetal mouse testis (91). Additional indirect support for a role for GATA factors in Leydig cells has come from the characterization of GATA-like protein 1 [GLP1 (Glp1)] knockout mice (92). Glp1 is expressed in Leydig cells where it appears to act as a repressor of GATA action (92). In males, loss of GLP1 function correlates with an early impairment of germ cell differentiation, suggesting that a complex regulation of GATA proteins in Leydig cells by GLP1, FOG2, and perhaps other modulatory factors, is crucial for their function. Although these mouse studies have illustrated the involvement of GATA factors for early testis cell differentiation, other strategies are still needed for the direct assessment of their function in the mature testis. Important insights into GATA function in the testis have also come from promoter studies of putative target genes in cell lines or primary cultures of Sertoli and Leydig cells. Most of these GATA gene targets are either correlated with early differentiation (Dmrt1, Sry, and Sox9), or hormone synthesis such as the inhibin subunits (Inha and Inhb), Amh, and multiple proteins and enzymes of the steroidogenic pathway in Leydig cells (see Table 2). Interestingly, recent studies have identified genes encoding junctional proteins such as claudin 11 and Coxsackie- and adenovirus receptor-like membrane protein (CLMP) as novel GATA targets in the testis (93, 94). Although a scaffolding protein of the blood-testis barrier, Cldn11 expression has been shown to be under germ cell and androgen regulation (95, 96), with the androgen regulation apparently mediated by GATA and NFYA proteins (94). Despite the long list of target genes for GATA factors in testicular cells, we have only begun to scratch the surface as to how these factors are themselves regulated. At the transcriptional level, it is now evident that Gata1 expression in mouse Sertoli cells is dependent on a cell-specific promoter (97, 98, 99) and is negatively regulated by FSH and cAMP (100). In contrast to GATA1, gonadotropins and/or cAMP agonists have been shown to induce Gata4 and/or Gata6 expression in several gonadal cell lines, including MSC-1 Sertoli cells and mLTC-1 Leydig cells (78, 101). The dissection of the Gata4 promoter has begun to reveal some of its transcriptional control mechanisms (102, 103). In transgenic mice, the first 5 kb of the 5′-flanking region of the Gata4 gene is sufficient to recapitulate endogenous Gata4 expression in Sertoli cells throughout development (102). Subsequent in vitro studies identified two GC-box motifs and especially one E-box element within this −118-bp region that are crucial for Gata4 transcriptional activity in both testicular and cardiac cells (102, 103). In Sertoli cells, members of the upstream transcription factor (USF) family of factors, especially USF2, bind to and activate the Gata4 promoter via this critical E-box motif (102). Gata4 regulatory sequences driving expression in steroidogenic (Leydig) cells of the testis and ovarian cells have yet to be identified. In both Sertoli and Leydig cells, GATA4 activity is modulated via cooperative interactions with other transcriptional regulators and/or cofactors (73, 74, 104). These include the nuclear receptors steroidogenic factor 1 (SF-1/NR5A1) and liver receptor homolog 1 (LRH-1/NR5A2), which cooperate with GATA4 to enhance transcription from the Inha, Amh, P450 aromatase (Cyp19a1), and 3β-hydroxysteroid dehydrogenase type 2 (HSD3B2) promoters (27, 35, 36, 105). In testicular cells, GATA1 and GATA4 also interact with FOG1 and FOG2 (25, 26, 106). Although the FOG proteins do not bind directly to DNA, they function as either enhancers or repressors of GATA transcriptional activity depending on the cell and promoter context being studied (25, 26, 28, 106, 107, 108, 109). On GATA-dependent gonadal promoters, however, in vitro data suggest that FOG proteins play a strictly repressive role (110). In the mouse testis, FOG1 and FOG2 are coexpressed with GATA1/4/6 in Sertoli and Leydig cells (79, 110). FOG1 is first detected in Sertoli cells at e15.5; expression is maintained throughout development but eventually becomes restricted to stage VII–XII tubules in the adult (79). FOG2 is expressed earlier than FOG1 and can be detected as early as the undifferentiated gonad stage in both sexes (76). After sex determination, testicular FOG2 expression is rapidly down-regulated by e15.5, leaving only the testis capsule and Leydig cells as significant sites of expression (79). FOG2 then reappears after birth and, like FOG1, eventually becomes restricted to stage VII–XI tubules in the adult testis (79). Different mouse models have clearly identified a role for a GATA4/FOG2 complex in early testis differentiation, acting upstream of Sry and Sox9 (88, 89, 90). However, it remains to be shown whether these genes and other GATA targets in the testis are directly modulated by either FOG-mediated coactivation or corepression. GATA4 is also a target for posttranslational modifications such as phosphorylation (34, 111, 112, 113, 114, 115, 116, 117). In gonadal cells, activation of the cAMP/protein kinase A (PKA) signaling pathway is a major mechanism for conveying the hormone responsiveness of many genes. In this classic pathway, PKA phosphorylates cAMP response element (CRE)-binding protein (CREB), which is then recruited along with the CREB-binding protein (CBP) coactivator to activate genes possessing CRE in their promoters. In MA-10 Leydig cells, GATA4 is directly phosphorylated by PKA on a specific serine residue (Ser261) (114). Phosphorylation of GATA4 allows for synergistic interactions with different transcription factors including CCAAT/enhancer-binding protein (C/EBPβ) and NR5A2 (34, 35) and permits recruitment of the CBP transcriptional coactivator much like CREB (114). This leads to robust stimulation of transcription from gonadal promoters such as Star, Inha, and HSD3B2 (34, 35, 36, 114). PKA is not the sole kinase known to target the GATA4 protein. Indeed, in the heart, phosphorylation of GATA4 at Ser105 via the MAPK signaling pathway plays a crucial role in modulating its GATA4 activity in cardiomyocytes (111, 112, 118). Although GATA4 Ser105 has been shown to be constitutively phosphorylated in porcine granulosa cells (119), and appears to contribute to the FSH-induced up-regulation of Cyp19a1 expression in these same cells in the rat (120), its significance in testicular cells remains to be demonstrated. Ovary The fetal ovary has been shown to express three GATA factors: GATA2, -4, and -6 (Fig. 2). In the mouse, GATA2 is specific for germ cells but only for a brief period during fetal development (e11.5–e15.5) (75). GATA4 is first detected in the mouse coelomic epithelium and porcine urogenital ridge of both XX and XY embryos (76, 79, 81, 83). At this time, GATA4 and its cofactor FOG2 label the somatic cell population of the developing gonad (76, 79). Subsequently, GATA4 and FOG2 continue to be expressed at high levels in ovarian somatic cells over the entire fetal period (76). GATA4 expression patterns are similar in the fetal rat and porcine ovary; the somatic, pregranulosa or follicular cells are GATA4 positive, whereas germ cells do not express this factor (80, 81). In the fetal rat ovary, GATA6 is present in both postmitotic germ cells and the ovarian epithelium (80). In the human fetal ovary, GATA4 is strongly expressed in pregranulosa/stromal cells as early as gestational wk 13 and 14, when primordial follicles are not yet evident (121). By wk 16–33, GATA4 localizes to granulosa cells (121). Like GATA4, GATA6 and FOG2 are expressed in human fetal somatic cells but can also be detected in occasional germ cells (unpublished data). Thus, much like the testis, the abundant expression of GATA proteins, particularly GATA4, in the fetal ovary suggest that they play key roles in the early steps of ovarian differentiation. In the postnatal ovary, GATA4 and GATA6 are the predominant GATA factors. GATA4 expression is negligible in primordial follicles of both the mouse and human ovary (76, 80, 101, 122, 123). GATA4 expression initiates in granulosa cells at the onset of follicle growth (i.e. their initial recruitment) and their transformation into primary follicles (76, 101, 122, 123). As follicular growth proceeds, granulosa cells of healthy preantral and antral follicles express GATA4, where it localizes to cumulus and mural granulosa cells, theca cells, and stromal cells (76, 80, 101, 122, 123, 124). In luteal glands, GATA4 expression is low or negligible in mice and humans (76, 101, 123). However, in the rat and pig, GATA4 is expressed in peripheral cells of new luteal glands; expression is down-regulated as these structures regress (80, 124). Thus, GATA4 expression is clearly linked to both structural and functional changes in the postnatal ovary of different mammalian species. GATA6 is also expressed in multiple cell types of the mouse, pig, and human postnatal ovary. These include oocytes, luteal glands, and granulosa and theca cells of larger follicles (80, 101, 122, 123, 124). Whereas GATA4 expression in granulosa cells is usually down-regulated in follicles undergoing atresia, GATA6 expression in rodents is retained at this stage (80, 101). In humans, GATA6 and FOG2 expression patterns are similar; primary follicles have negligible or weak immunoreactivity, but in later follicular stages, GATA6 and FOG2 proteins are detected in most granulosa cells, occasional oocytes, and luteal glands (122, 123). In the mouse, only a few granulosa cells of primordial follicles readily express FOG2 protein. In antral follicles, FOG2 expression is often down-regulated, whereas GATA4 expression is retained (76). Of the endocrine and paracrine regulators of folliculogenesis, gonadotropins as well as TGFβ signaling have been linked to the regulation of ovarian GATA function. Gonadotropins have a fundamental impact on ovarian folliculogenesis, especially FSH, which is required for the development from preantral to antral follicles (125). In FSH-unresponsive ovaries of women with an inactivating mutation in the FSH receptor gene (FSHR), GATA4 is absent in the large pool of primordial follicles as compared with healthy ovaries (126). However, granulosa cells in the few primary follicles seen in these patients do express GATA4 protein, suggesting that either FSH or FSH receptor is not required for the initial activation of GATA4 expression. In Fshr-null mice bearing follicular arrest at the preantral stage, Gata4 expression is, however, impaired in preantral follicles compared with age-matched wild-type animals (127). In vitro studies have further established the link between gonadotropins and the regulation of GATA4 and/or GATA6 action in the ovary. Like testicular cells (114), the transcriptional activity of GATA proteins is likely induced by FSH/cAMP/PKA-mediated phosphorylation in granulosa cells (124). In murine granulosa tumor cells, which resemble normal granulosa cells, FSH and forskolin up-regulate Gata4 mRNA levels (101). Moreover, treatment of immature 3-wk-old mice with pregnant mare serum gonadotropin (PMSG) enhances follicular expression of Gata4 and Gata6 transcripts (101). Similarly, in primary rat granulosa cells, FSH up-regulates GATA4 mRNA and protein levels (120), whereas in immature human granulosa cells, obtained from in vitro maturation of oocytes during fertility treatments, FSH was shown to up-regulate GATA6 mRNA levels (128). In cultured preovulatory human granulosa cells, which luteinize during in vitro culture, treatments with recombinant human chorionic gonadotropin or 8-Br-cAMP had no effect on GATA4 and FOG2 steady-state mRNA levels, whereas GATA6 mRNA levels were modestly but significantly up-regulated (123). The effects on GATA6 rather than GATA4 likely reflect the luteinized phenotype of human granulosa cells. TGFβ regulation of GATA target genes has been previously demonstrated in both cardiac and T helper cells (129, 130). Recent results also support a role for GATA4 in TGFβ signaling in the ovary. In mouse granulosa tumor cells TGFβ is able to up-regulate GATA4 protein levels (131). Furthermore, TGFβ activation of the Inha promoter, a GATA target in both the testis and ovary (Table 2), is blocked when the critical GATA binding sites are mutated (131). GATA4 interacts with Smad3 in granulosa tumor cells, suggesting a role for GATA4 in the transcriptional complex activated by the TGFβ-signaling pathway in the ovary. The proposed role for TGFβ and other signaling mechanisms in the regulation of GATA4 activity in gonadal cells is presented in Fig. 3. Much like the testis, important insights into the role of GATA factors in ovarian function have come from an analysis of potential target gene promoters in ovarian cells. Because somatic cells of both the ovary and testis express similar GATA proteins, it is not surprising that GATA factors have similar target genes in these two tissues (Table 2). To date, proposed GATA-dependent genes in the ovary include Star (132, 133, 134), GNRHR (135), Inha (131), Cyp19a1 (120, 136, 137), 17β-hydroxysteroid dehydrogenase type 1 (HSD17B1) (138), Cyp11a1 (139), and Bcl2 (unpublished data). One of the first GATA target genes to be described in gonadal cells was AMH (83). In vitro and in vivo studies have provided strong evidence that the mouse and human AMH promoter, at least in the testis, is an important target for GATA4 (140, 141). Although exclusive to the fetal testis, the anti-Müllerian hormone (AMH) is nonetheless produced by granulosa cells of the postnatal ovary where GATA4 is also found (142). Therefore, a similar role for GATA4 in AMH gene regulation in the ovary is likely. Placenta GATA factors in placental cells were initially described in studies of αGSU (CGA) transcription in the human trophoblast-derived JEG-3 choriocarcinoma cell line (48). Nuclear extracts of JEG-3 cells were shown to contain GATA2 and GATA3 protein capable of binding to a consensus GATA motif in the proximal CGA promoter (48). Moreover, mutation of the αGSU GATA element dramatically reduced CGA promoter activity in JEG-3 cells, suggesting that GATA2 and/or GATA3 is an important contributor to CGA transcription in the placenta (48). Subsequent Northern blot and in situ hybridization studies revealed that the Gata2 and Gata3 transcripts are abundantly expressed in mouse placenta, localizing predominantly to trophoblast giant cells (143). In these cells, the murine or ovine genes encoding for the placental hormones placental lactogen 1 (PRL3D1) and proliferin (PRL2C2) have been proposed to be regulated by GATA2 and/or GATA3 (143, 144, 145). Indeed, although mice lacking either the Gata2 or Gata3 gene develop trophoblast giant cells, placental Prl3d1 expression is nonetheless reduced by 50% in these animals (145). For the Prl2c2 gene, a similar 50% reduction in expression was observed in Gata3-null placentas, and this reduction was even more pronounced (5- to 6-fold) in Gata2-null placentas (145). Proliferin is an angiogenic hormone known to stimulate endothelial cell migration and neovascularization of the placenta (146). Consistent with a role for GATA2 in Prl2c2 expression, decidual tissue adjacent to Gata2-null placentas has been reported to exhibit less neovascularization as compared with wild-type tissue (145). In vitro studies have also identified other placenta-expressed gene promoters as potential targets for regulation by GATA or GATA-like factors. These include the mouse promoters for adenosine deaminase (Ada) (147) and Cyp11a1 (139) and the human promoters for leukemia inhibitory factor receptor (LIFR) (148), GnRH receptor (GNRHR) (149), (HSD17B1) (150), (HSD3B1) (151), and (CYP17A1) (38). Whether these genes are actual targets in vivo, however, awaits further genetic studies. Adrenal The adrenal cortex arises from mesodermal precursors that also give rise to gonadal steroidogenic cells (152). The human fetal adrenal cortex consists of a large inner zone, termed the fetal zone, and a thin outer rim of more immature cells, known as the definitive zone (152). After birth, the fetal zone regresses. In late gestation, the definitive zone begins to partition into anatomically and functionally distinct compartments: the zona glomerulosa, zona fasiculata, and zona reticularis (152). In the mouse adrenal, the zona glomerulosa and zona fasciculata are well defined, but there is no discernable zona reticularis. The adrenal cortex of the young mouse contains an additional layer, termed the x-zone, which is adjacent to the medulla and is analogous to the fetal zone of the human adrenal cortex (153). GATA6 is abundantly expressed in the adrenal cortex throughout development (38, 154, 155). During human fetal development, GATA6 is evident in both the fetal and definitive zones of the adrenal cortex (154). In the adult, GATA6 is expressed in the zona reticularis and to a lesser extent the zona fasciculata (154, 156, 157), but its pattern of expression during early childhood and adrenarche has not been established. Based on this distribution and promoter analyses described below, GATA6 is postulated to play a key role in the production of adrenal androgens and possibly also glucocorticoids (156). In the mouse adrenal, GATA6 is expressed in both the fetal and adult adrenal cortex (154), suggesting a role beyond merely adrenal androgen biosynthesis. Although evidence for the involvement of GATA6 in adrenal steroidogenesis is accumulating, genetic proof that GATA6 is required for adrenocortical function is lacking. By comparison, GATA4 has a more restricted pattern of expression during adrenocortical development and therefore is presumed to have a limited role in the function of this tissue (154). During human development, GATA4 mRNA is evident in the fetal zone of the adrenal, but there is only weak expression of this transcript in the adrenal cortex postnatally (154). Similarly, Gata4 is transiently expressed in the mouse adrenal cortex during fetal but not postnatal development (154), a pattern reminiscent of Cyp17a1 in the mouse (158). Given their overlapping patterns of expression, it is possible that GATA4 enhances expression of Cyp17a1 in the fetal adrenal, although there is no direct proof of this. Studies of mice harboring Gata4 loss-of-function mutations suggest that this transcription factor is not required for the differentiation of fetal adrenocortical cells (88, 154). Consistent with its proposed role in the biosynthesis of adrenal androgens and other corticoids, GATA6 has been shown to act in synergy with SF-1 (NR5A1) and other factors to enhance the transcription of CYP11A1 (156), CYP17A1 (38), cytochrome b5 (CYB5) (159), hydroxysteroid sulfotransferase (SULT2A1) (156, 160, 161), and HSD3B2 (35) in cell lines (see Table 2). In the case of the CYP17A1 promoter, GATA6 does not act through a GATA response element but instead through interactions with another transcription factor, specificity protein 1 (SP1) (38). In the case of CYB5, GATA6 enhances promoter activity through interactions with specificity protein 3 (SP3) (159). GATA4 can substitute for GATA6 in transactivation studies of the CYP17A1 promoter (38), suggesting that GATA4 may serve to augment CYP17A1 expression during fetal development. In addition to normal adrenal cortex, GATA4 may regulate gene expression in disease states (described below). The differentiation, growth, and survival of steroidogenic cells in the adrenal gland are controlled by a diverse array of hormones. Endocrine hormones and paracrine factors traditionally associated with the function of gonadal steroidogenic cells, such as LH, inhibins, and activins, also influence the differentiation, proliferation and function of adrenocortical cells, both in physiological and pathophysiological states. There is no direct evidence that GATA6 is hormonally regulated in the normal adrenal. Steady-state levels of Gata6 mRNA in the adrenal gland do not change markedly in mice treated with ACTH or dexamethasone (157), nor does there appear to be a diurnal variation in GATA6 expression in the adrenal. Administration of dibutyryl cAMP to NCI-H295A human adrenocortical cells has been shown to increase GATA6 mRNA levels, raising the possibility of hormonal regulation of GATA6 in the human adrenal cortex, but the physiological relevance of this finding in vivo remains to be established. GATA6 protein is down-regulated in response to mitogenic signals in some cell types (162, 163), but there is no evidence supporting a role for GATA6 during proliferation in the adrenal cortex. Human chorionic gonadotropin up-regulates Gata4 and LH receptor (Lhcgr) expression in a dose-dependent manner (164). GATA4 has a role in the induced, but not basal, regulation of the LH receptor (Lhcgr) promoter in adrenocortical tumor cells. Ectopic expression of GATA4 in the adrenal heralds the appearance of neoplastic cells (164), and this factor may serve to integrate intracellular signals evoked by different hormones. GATA FACTORS AND ENDOCRINE DISEASE Gonadal Disorders and Tumorigenesis To date, reports of gonadal disorders involving GATA4 or its associated cofactors have been scarce. Human mutations in the NR5A1 gene have been identified in individuals with 46,XY sex reversal with gonadal dysgenesis and adrenal insufficiency (165, 166, 167). Some of the cases involved retained Müllerian duct derivatives, indicating insufficient AMH production during fetal life. One of the heterozygous NR5A1 mutations (G35E) was found to act as a dominant-negative competitor of the synergism between GATA4 and the wild-type NR5A1 protein at the level of the AMH promoter (168). This suggested that proper GATA4/SF-1 is essential for normal AMH transcription in humans (168). Although not targeting the GATA4 gene itself, a recently described de novo t(8;10) chromosomal translocation associated with heart defects and hypergonadotropic hypogonadism has been linked to the gene encoding the GATA4 cofactor FOG2 (169). The resulting truncated FOG2 protein, however, did not appear to impair AMH expression because the patient had no signs of Müllerian duct retention. Thus, unlike the mouse (88), FOG2 might be dispensable for the initiation of testicular development in humans. In vitro studies have suggested roles for GATA factors in the regulation of steroidogenesis in both gonadal and adrenal cells (104, 156). Patients with polycystic ovary syndrome (PCOS) have hyperandrogenism of both ovarian and adrenal origin. The gene expression pattern as well as the response to hormonal stimulation of PCOS theca cells differs significantly from their normal counterparts (170, 171). Interestingly, steady-state GATA6 mRNA levels are augmented in PCOS theca cells due to increased gene transcription and stability of the GATA6 transcripts (172). No sequence variations at the GATA6 locus, however, were associated with PCOS patients (172), suggesting that regulatory factors acting at the level of the GATA6 promoter might be differentially expressed in normal vs. PCOS theca cells. Ovarian carcinomas account for up to 90% of ovarian cancer and typically have a poor prognosis related to a delay in detection. They originate from the surface epithelium and occur in various forms, the most common being serous carcinoma. The serous carcinomas commonly exhibit loss of heterozygosity (LOH) in the locus 8p21–23, which bears the GATA4 gene. In mucinous carcinomas, this LOH is less common (173). Of interest, GATA4 protein expression is lost in practically all serous carcinomas (516 of 528 samples studied) but in only some mucinous carcinomas (26 of 75 samples studied) (173). Moreover, in the mucinous carcinomas, the loss of GATA4 expression correlates inversely with the grade and clinical stage of the tumors (173). In general, LOH correlates with loss of GATA4 expression in the tumors, suggesting that loss of the other allele is enough to completely silence GATA4 expression. More recent findings indicate that the GATA4 gene is often silenced in ovarian carcinomas by histone modifications. In various ovarian cancer cell lines, GATA4 expression is usually negative due to methylation of its gene promoter (174, 175). However, none of the analyzed serous carcinoma samples (n =7) exhibited a GATA4 methylated state (175), indicating that LOH is the only reason for the absence of GATA4 expression in the serous subtype. In one study, GATA4 was found to be expressed in most of the studied ovarian carcinomas of different subtypes (43 of 50 samples); in contrast to previous findings (173), 30 of these tumors were of serous type, and only six were GATA4 negative (176). Instead, the authors found a frequent loss of GATA6 protein correlated with a loss of several markers of normal epithelium such as disabled-2, collagen IV, and laminin (176). Taken together, loss of GATA4 and/or GATA6 expression, due to LOH or histone modifications, is likely involved in the dedifferentiation of the ovarian epithelium and ovarian carcinogenesis. Granulosa cell tumors (GCTs) are rare, accounting for 3–5% of all ovarian cancers (176, 177). They are steroidogenically active, typically causing precocious puberty, disturbances in menstrual cycle, and endometrial hyperplasia or cancer (176, 177). Although poorly understood, the pathogenesis of GCTs involves defects in the regulation of granulosa cell proliferation (178). Recent findings suggest that GATA4 and GATA6 play a role in granulosa cell tumorigenesis (122). The majority of the analyzed 80 GCTs expressed GATA4 and GATA6 as well as NR5A1 and FOG2 protein at levels comparable with those in normal granulosa cells (122). However, negative or weak GATA6 protein expression is associated with a large tumor size, suggesting that GATA6 has a role in suppressing proliferation (122). This hypothesis is supported by findings in both glomerular and vascular smooth muscle cells, in which GATA6 is able to induce cell cycle arrest by activating and/or stabilizing p21cip1 (Cdkn1a) expression (163, 179). Thus, GATA4 likely has an opposing role to GATA6 in granulosa cell tumorigenesis. GATA4 expression associates with aggressive behavior in a subgroup of GCTs, suggesting that it may facilitate proliferation and suppress apoptosis in granulosa cells (122). The vast majority of the analyzed GCTs expressed GATA4 at an intermediate or high level corresponding to its expression in normal granulosa cells (122). GATA4 expression associates with clinical stage (122); GATA4 expression is high in 60% of tumors spread outside the ovary (stage Ic and beyond). Furthermore, high GATA4 expression correlates with recurrence risk of the GCT patients, suggesting that GATA4 immunostaining may have value as a prognostic tool in human GCTs. The reasons for high GATA4 expression in aggressive GCTs, however, remain unknown and therefore need further study. GCTs and adrenal tumors (described below) of both mice and humans exhibit similarities in their GATA expression patterns. Given the common embryonic origin of adrenocortical cells and ovarian granulosa cells, the GATA4/GATA6 ratio may serve as an essential element in the balanced regulation of proliferation vs. apoptosis. Potential target genes involved in these processes in the ovary include cyclin D2 (Ccnd2) and the antiapoptotic genes Bcl2 and Bcl2l1, which have been shown to be under GATA control in other organs (180, 181, 182). Cyclin D2 is essential for granulosa cell proliferation and plays a role in GCT development (183, 184). Our unpublished results show that GATA4 expression is associated with CCND2 and BCL2 expression in GCTs and that the proximal promoters of both genes are activated by GATA4 in granulosa tumor cells. Adrenal Disorders and Tumorigenesis Although not normally expressed in the postnatal adrenal cortex, ectopic GATA4 production has been linked to disease states in this tissue (e.g. adrenocortical tumors; see below). Given their expression patterns and target genes, it is reasonable to postulate that GATA factors may be relevant to adrenal insufficiency or other diseases characterized by aberrant steroidogenic gene expression alone or together with other factors. For example, a de novo heterozygous mutation (G35E) in the NR5A1 gene has been shown to be responsible for adrenal insufficiency and pseudohermaphroditism (185). This mutant specifically fails to synergize with GATA factors on the HSD3B2 promoter (35). GATA6 immunoreactivity is significantly diminished in most human adrenocortical adenomas and carcinomas (157, 186). This decrease in GATA6 expression may be an important event for the escape of tumor cells from normal control mechanisms, although there is no direct proof of this premise. Consistent with its proposed role in adrenal androgen biosynthesis in the zona reticularis, GATA6 expression is retained in a significant fraction of virilizing adrenocortical tumors, especially carcinomas (187, 188). GATA4 mRNA is expressed in only a small proportion of human adenomas and carcinomas (157, 187, 188). Larger amounts of GATA4 mRNA have been found among those presenting aggressive clinical behavior, although its value as a prognostic marker remains to be established (189). Gonadectomy causes cells in the mouse adrenal cortex to transform into sex steroid-producing cells that are histologically and functionally similar to gonadal tissue (190). This phenomenon is strain dependent; susceptible strains include CE, DBA/2J, and NU/J. When these mice are gonadectomized, small, basophilic cells, known as A cells, accumulate in the subcapsular region of the adrenal cortex (191). Type A cells express GATA4 and AMH type 2 receptor (AMHR2) but lack expression of steroidogenic markers (164, 192). Functionally, A cells resemble stromal cells of the postmenopausal ovary, which can metabolize cholesterol into oxysterols but have limited capacity for the synthesis of sex steroids (193). Later, foci of large, sex steroid-producing cells, termed B cells, appear in the adrenal cortex (194). B cells express Gata4, Nr5a1, Lhcgr, Inha, Amh, estrogen receptor α (Esr1), Cyp17a1, and an ovary-specific alternative splice variant of Cyp19a1 (164, 192). Type B cells resemble follicular theca cells from women with PCOS, a condition marked by chronically elevated LH levels (195). Traditional adrenocortical cell markers, such as the melanocortin 2 receptor (MC2R) or enzymes for the synthesis of corticosterone or aldosterone [e.g. steroid 21-hydroxylase (Cyp21a1), P450 11β-hydroxylase (Cyp11b1), and aldosterone synthase (Cyp11b2)] are not expressed in A or B cells (164). Given that many of the genes expressed in A and B cells, such as Lhcgr, Inha, Amh, and Cyp17a1, harbor GATA-binding sites in their promoters, increased GATA4 levels may be postulated to lead to their enhanced expression in these cells. When Inha promoter-SV40 T-antigen transgenic mice are subject to prepubertal gonadectomy, they develop sex steroid-producing adrenocortical carcinomas by 5–7 months of age (196, 197). These adrenocortical carcinomas generally arise in the X-zone, but subcapsular pates of type A and B cells can accompany the X-zone lesions. The molecular markers of adrenocortical neoplasia in these mice, Lhcgr, Inha, Gata4, and Cyp17a1, are similar to those described in gonadectomized DBA/2J mice. GATA6 is also down-regulated during adrenocortical tumorigenesis in these transgenic mice. The phenomenon of gonadectomy-induced adrenocortical neoplasia has also been observed in other small animals, including rats, guinea pigs, hamsters, and ferrets (198). In the ferret adrenal, GATA4 immunoreactivity can be seen in both small and large neoplastic cells and seems to correlate with the degree of nuclear and cytoplasmic atypia (199). GATA4 is particularly abundant in anaplastic myxoid adrenocortical carcinomas but is absent from the spindle cell component of adrenocortical tumors (199). Thus, taken together, it seems likely that GATA4 serves an essential function in gonadectomy-induced adrenocortical tumorigenesis, given its established role in differentiation of gonadal stroma (88).',\n",
       "  'Discussion': '',\n",
       "  'Conclusion': 'Over the past decade, it has become plainly evident that the critical nature of GATA regulatory proteins is not simply a matter of the cardiac and hematopoietic systems. Indeed, the abundant expression of GATA proteins in multiple cell types of various endocrine organs along with their ever-expanding list of target genes is a strong indication that these factors are essential regulators of cell-specific gene expression required for the development, differentiation, and function of endocrine cells. One must be wary, however, that most studies to date have relied heavily on in vitro transactivation assays, DNA-binding analyses, and to a lesser extent chromatin immunoprecipitation. Although steadily increasing, the number of reports relying on genetics and other in vivo models remain limited as are cases of human endocrine-related disorders associated with GATA mutations and/or deregulated GATA function. Although several studies have implicated GATA factors in endocrine tumors, the underlying mechanisms remain to be elucidated. The future challenge is then to continue applying genetic models to validate the bulk of in vitro data that is currently available. For some endocrine organs, this appears to be a somewhat daunting task. This is particularly true of the gonads and adrenal where the overlapping expression of different GATA proteins in various cell types (see Fig. 2) and the potential for compensatory mechanisms render individual Gata gene knockout approaches in mice difficult to interpret. A case in point is the Sertoli cell-specific knockout of the Gata1 gene in the adult testis, which produced no overt testicular phenotype (87). In these particular cases, more sophisticated approaches will undoubtedly be required.',\n",
       "  'URL': 'https://doi.org/10.1210/me.2007-0513',\n",
       "  'Journal': 'Mol Endocrinol'},\n",
       " {'PaperTitle': 'Reconstructing the female reproductive system using 3D bioprinting in tissue engineering',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Heesuh Yi, Gaeun Lee, Sanghyeok Park, Juhyeong Ha, Dayeong Choi, Jihoon Ko, Jungho Ahn',\n",
       "  'Email': 'koch@gachon.ac.kr; jhahn513@skku.edu',\n",
       "  'Abstract': 'Three-dimensional bioprinting enables the precise fabrication of complex biological tissues through the layer-by-layer deposition of living cells and biomaterials, offering a promising strategy for reconstructing the female reproductive system. This technology has facilitated the development of in vitro models for tissues such as the endometrium, ovary, cervix, and vagina, providing improved structural fidelity and functional relevance. By leveraging bioinks, including decellularized extracellular matrix and advanced bioprinting techniques, researchers can recreate the intricate microarchitectures and vascular networks required for tissue functionality. These bioprinted systems serve as high-fidelity microphysiological systems for studying reproductive health, modeling disease progression, and evaluating therapeutic responses. Moreover, the integration of artificial intelligence into bioprinting workflows enhances reproducibility, scalability, and patient-specific customization. This review summarizes recent advances in reproductive tissue bioprinting and highlights its potential to transform regenerative gynecology and personalized reproductive healthcare.',\n",
       "  'Introduction': \"The female reproductive system is a complex network of organs and tissues—including the ovaries, fallopian tubes, uterus, cervix, and vagina—that work together to ensure fertility, hormonal regulation, and reproductive health [1]. The ovaries produce both oocytes and key sex hormones, such as estrogen and progesterone (P4), which regulate the menstrual cycle and induce cyclical remodeling of the endometrium. After ovulation, the oocyte is typically fertilized in the fallopian tube and transported to the uterus, where the hormonally primed endometrium facilitates embryo implantation. If implantation fails, hormonal withdrawal triggers the shedding of the endometrial lining during menstruation [2]. The uterus, particularly its inner lining called the endometrium, undergoes proliferative and secretory phases under the influence of ovarian hormones, creating an optimal environment for embryo implantation and fetal development; if implantation does not occur, hormone withdrawal leads to the shedding of the functional endometrial layer during menstruation [3]. The cervix connects the uterus to the vagina, regulates the passage of sperm and menstrual blood, and produces mucus whose properties change with hormonal fluctuations, while also forming a barrier against pathogens during pregnancy [4]. The vagina is a fibromuscular canal that serves as the site for sperm deposition, the birth canal during delivery, and a protective barrier against external microorganisms, with its epithelium maintained by estrogen [5]. These organs are highly dynamic, with their structure and function tightly regulated by cyclical endocrine signals, cellular interactions, and extracellular matrix (ECM) remodeling, all of which are essential for reproductive success. Understanding these complex biological processes is critical for the rational design of engineered tissue models that aim to recapitulate native female reproductive function [6, 7].\\nReproduction is a vital biological process essential for species survival, involving the coordinated interaction of specialized tissues, organs, and hormones. In the female reproductive system, hormones such as estradiol (E2) and P4 regulate the proliferative and secretory phases of the endometrium, creating an optimal environment for implantation and fetal development. The luteinizing hormone (LH) surge triggers ovulation, enabling fertilization in the fallopian tubes and subsequent zygote transport to the uterus. During pregnancy, the cervix serves as a protective barrier, safeguarding against external microorganisms. Recent advancements in reproductive biology have deepened our understanding of these processes, offering critical insights into tissue development and function [6, 7].\\nThe study of human reproduction requires a multidisciplinary approach. While animal models and two-dimensional (2D) cell cultures have advanced reproductive biology research, they present significant limitations. Animal models often fail to replicate human reproductive physiology accurately [8, 9], and 2D models cannot capture the dynamic, three-dimensional (3D) complexity of living tissues. These challenges not only highlight the urgent need for advanced in vitro platforms but also underscore the critical importance of evaluating how far current strategies have come and where they continue to fall short.\\n3D bioprinting, an emerging technology that precisely combines living cells with biomaterials, offers a solution for addressing these limitations. By recreating the intricate architecture and functionality of native tissues, bioprinting surpasses conventional tissue-engineering approaches [10]. This technology has rapidly evolved, incorporating methods such as extrusion-based, inkjet-based, laser-assisted and Vat-polymerization-based bioprinting to improve precision and cell viability [11]. Bioprinting's ability to spatially organize multiple cell types and biomaterials enables the creation of tissue models with high structural and functional fidelity [12], providing valuable tools for regenerative medicine, drug testing, and disease modeling [[13], [14], [15]].\\nSuccessful bioprinting of female reproductive tissues requires careful consideration of several design criteria: (1) Hormonal responsiveness, to mimic physiological responses to estrogen and P4; (2) Matrix composition, reflecting the tissue-specific ECM for cell support and signaling; (3) Vascularization, ensuring nutrient and oxygen supply in thick constructs; (4) Multi-cellularity, to recapitulate the complex cellular architecture; and (5) Mechanical properties, matching the elasticity and strength of native tissues. These criteria are critical for developing physiologically relevant and functional bioprinted constructs.\\nIn reproductive medicine, 3D bioprinting has demonstrated significant potential in addressing critical challenges, including restoring ovarian function, regenerating the endometrium, and reconstructing vaginal tissue. Regulatory advancements, such as the FDA Modernization Act 2.0/3.0, have further accelerated the development of nonclinical bioprinted models by supporting in vitro and in silico methods alongside traditional animal testing [16]. These bioengineered systems now enable more accurate evaluation of drug effects on reproductive tissues, addressing longstanding gaps in maternal and fetal health research.\\nThis review offers a detailed overview of recent progress in 3D bioprinting for female reproductive tissue engineering, with a focus on key organs such as the ovary, endometrium, cervix, andvagina. It discusses organ-specific bioink development, bioprinting techniques, and the functional restoration of reproductive tissues. In addition, the review addresses persistent challenges including vascularization, innervation, and barriers to clinical translation, while also exploring future directions involving artificial intelligence (AI) and multi-organ platforms. By integrating these perspectives, the review emphasizes the potential of 3D bioprinting to advance reproductive health research and personalized therapeutic strategies.\",\n",
       "  'Methods': '',\n",
       "  'Results': '',\n",
       "  'Discussion': '',\n",
       "  'Conclusion': '3D bioprinting has emerged as a transformative technology in female reproductive tissue engineering, offering unprecedented precision in replicating the structural, cellular, and functional complexities of reproductive organs. This review provides a comprehensive, organ-specific analysis of 3D bioprinting applications across the entire female reproductive system, including the ovary, endometrium, cervix, vagina, and placenta. Key current challenges are examined, including material-related issues, technical and structural limitations of bioprinting platforms, and barriers to clinical translation. In addition, this review explores advanced future directions in reproductive tissue bioprinting, with emphasis on vascularization and innervation strategies, the integration of AI for personalized precision bioprinting, and the commercialization landscape and translational outlook of this rapidly evolving field.\\nBuilding upon these insights, recent innovations in 3D bioprinting are enabling the development of physiologically relevant in vitro models that hold significant promises for restoring fertility, treating gynecological disorders, and advancing personalized reproductive medicine. Despite this rapid progress, several critical challenges persist, including the need for standardized organ-specific bioinks, and the translational gap between bench-scale constructs and clinical applications. Ultimately, the integration of bioprinting, regenerative medicine, and precision engineering offers a promising frontier for reproductive health, with bioprinted tissues poised to advance both our understanding of reproduction and the development of future therapies.',\n",
       "  'URL': 'https://doi.org/10.1016/j.mtbio.2025.102127',\n",
       "  'Journal': 'Mater Today Bio'},\n",
       " {'PaperTitle': 'Sox9 in the epicardium: Implications for cell invasion, differentiation, and coronary vascular development',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Andrew B Harvey, Allison M Trouten, Renélyn A Wolters, Jenna R Drummond, Raymond N Deepe, Hannah G Tarolli, Inara Devji, Silvia G Vaena, Martin J Romeo, Robin Muise-Helmericks, Paula S Ramos, Russell A Norris, Ge Tao, Andy Wessels',\n",
       "  'Email': 'harveyan@musc.edu',\n",
       "  'Abstract': 'The epicardium is the mesothelial lining of the heart and is a source of progenitor cells during heart development, giving rise to an invasive population of mesenchymal cells which differentiate into cardiac fibroblasts, mural cells, and other cell types essential for heart structure and function. Previously, we showed that epicardial-specific deletion of the gene encoding SRY-box transcription factor 9 (SOX9) impairs epicardial-derived cell invasion and reduces their contribution to the atrioventricular valve mesenchyme. In this study, we use single-cell RNA-sequencing to investigate broader roles of Sox9 in the epicardium as it relates to epicardial invasion, differentiation, and vascular development. We identified transcriptional changes indicative of decreased epicardial-to-mesenchymal transformation consistent with histological observations. Immunofluorescence analyses revealed defective epicardial attachment and decreased epicardial-derived cell invasion into the ventricular myocardium associated with delayed coronary plexus formation. Sox9-deficient epicardial cells exhibited elevated expression of vascular smooth muscle cell genes, suggesting that Sox9 may influence epicardial cell fate decisions. This study expands our understanding of the role of Sox9 in epicardial biology, demonstrating an important function in regulating epicardial cell invasion, differentiation, and coronary vasculature development. These insights provide a foundation for further investigations into epicardial-mediated mechanisms underlying congenital heart abnormalities.',\n",
       "  'Introduction': 'The epicardium acts as an important source of progenitor cells which give rise to multiple cell types essential for heart development. The proepicardial organ (PEO), the extracardiac precursor to the epicardium, is a coelomic mesothelium-derived cluster of cells situated dorsal to the developing heart [1–5]. Around embryonic day (E)9.5 in mouse, cells from the PEO migrate towards the surface of the heart where they spread across the myocardium to form a thin, continuous layer of epicardial cells [1]. A portion of these cells will subsequently undergo an epicardial-to-mesenchymal transformation (EpiMT) generating an invasive population of epicardial-derived cells (EPDCs). Around E12.5, EPDCs invade the myocardium and proceed to differentiate into multiple different cell types: predominantly cardiac fibroblasts and mural cells (vascular smooth muscle cells [VSMCs] and pericytes), but also, to a lesser extent, endothelial cells of the coronary vasculature [6–9]. We have previously reported on a subpopulation of EPDCs at the atrioventricular (AV) junction that contribute to the mesenchyme of the AV valves, where they play an important role in maintaining extracellular matrix homeostasis [10–13]. The epicardium and EPDCs act as a source of signals which modulate the growth and maturation of the myocardium and patterning of the coronary vasculature [14–16]. Despite significant advances in understanding the cell types derived from the epicardium, the timing and molecular mechanisms governing their migration and differentiation remain poorly understood.\\nSRY-Box Transcription Factor 9 (SOX9), is a high mobility group domain transcription factor with critical roles in embryonic development. While homozygous null mutations in Sox9 are embryonic lethal [17], cell-type specific deletion of Sox9 using the Cre-LoxP system has provided insight into its diverse roles in cardiovascular development. Deletion of Sox9 from the endocardial or cardiac neural crest cell lineages results in defective cardiac cushion formation, the precursors to the valves [18, 19]. Additionally, deletion of Sox9 from the second heart field results in a high penetrance of ventricular septal defects, often in combination with primary atrial septa defects [20, 21]. Recently, we identified a role for Sox9 in the epicardium, showing that epicardial-specific deletion of Sox9 leads to a reduced contribution of EPDCs to the AV valves and reduced EPDC invasion into the ventricular myocardium [13]. In that study, we focused mainly on the tissues of the AV junction, particularly the mitral valve. We reported that reduced contribution of EPDCs to the posterior leaflet of the mitral valve during development due to loss of Sox9 leads to myxomatous valve degeneration postnatally [13]. Our previous study demonstrated that the presence of EPDCs in the valves is essential for maintaining valve extracellular matrix homeostasis [13].\\nHere, we used single-cell RNA-sequencing to further characterize the role of Sox9 in epicardial development. Transcriptomic analyses of E14.5 epicardial-Sox9 knockout and control hearts indicated dysregulation of genes implicated in epicardial invasion, differentiation, and vascular development. Immunofluorescence analyses revealed defective epicardial attachment and delay in coronary plexus formation associated with decreased EPDC invasion in epicardial-Sox9 knockout hearts. Sox9-deficient epicardial cells expressed higher levels of VSMC genes suggestive of a role for Sox9 in epicardial cell fate decisions. Our data provide insight into the role of Sox9 in coordinating epicardial cell behaviors important for proper heart development.',\n",
       "  'Methods': 'Animals\\nAll animal experiments were conducted according to NIH guidelines (Guide for the Care and Use of Laboratory Animals) and protocols approved by the MUSC Institutional Animal Care and Use Committee (IACUC) with protocol ID number IACUC-2020-01140. The mWt1/IRES/GFP-Cre (Wt1Cre) mouse was described previously [11]. The B6.129S7-Sox9tm2Crm/J (Sox9fl/fl) mouse and the B6.129(Cg)-Gt(Rosa)26Sortm4(ACTB-tdTomato,EGFPLuo/J (Rosa26mTmG dual fluorescence lineage trace reporter expressing membrane bound tdTomato or eGFP, abbreviated R26mG) mouse was obtained from the Jackson Laboratory. Epicardial-specific Sox9 knockout animals (Wt1cre; Sox9fl/fl; R26mG) and littermate control (Wt1cre; Sox9fl/+; R26mG or Wt1cre; Sox9+/+; R26mG) animals were generated by crossing Wt1cre; Sox9fl/+ males with Sox9fl/+; R26mG or Sox9fl/fl; R26mG females. Mice heterozygous at the Sox9 locus (Wt1cre; Sox9fl/+; R26mG) did not present with any of the phenotypes described in Wt1cre; Sox9fl/fl; R26mG animals and were thus deemed appropriate controls for histological, immunofluorescence, and transcriptomic analyses. Pregnant dams were euthanized by isoflurane induction followed by cervical dislocation in accordance with NIH guidelines. Embryos were considered day 0.5 at midday on the day of vaginal plug detection, and staging was confirmed upon isolation using distinctive features described in The House Mouse: Atlas of Embryonic Development [84]. Both male and female specimens were generated, and combined data for both sexes are shown.\\nHistology, immunofluorescence, and microscopy\\nEmbryos were dissected in 1x PBS and fixed in 4% paraformaldehyde (PFA) either for 4 hours at room temperature or overnight at 4°C. Embryos were then processed through a series of graded alcohols, cleared in toluene, and embedded in paraffin. Hematoxylin and Eosin (H&E) or immunofluorescence staining was performed on 5µm tissue sections. Slides were deparaffinized in xylenes and rehydrated through graded alcohols. Antigen retrieval was performed by pressure cooking slides in citric acid-based antigen unmasking solution (Vector, H3300) for 1\\u2009min. Sections were incubated for 30 mins with 1% bovine serum albumin (BSA) prior to immunostaining to minimize non-specific binding of primary antibodies. Sections were incubated in primary antibody overnight at 4°C, washed in 1x PBS, and incubated with secondary antibody for 1 hour at room temperature. Serial dilutions were performed to determine optimal antibody concentrations, and secondary only controls were used to confirm signal specificity for the primary antibody. Primary and secondary antibodies and dilutions can be found in S4 Table. Slides were coverslipped with SlowFade Gold Antifade Reagent with DAPI (Invitrogen). Brightfield images were acquired using Olympus BX40. Fluorescence images were acquired with Zeiss AxioImager II and Leica TCS SP8 microscopes.\\nEpicardial explant culture and scratch assay\\nEpicardial explant culture was carried out essentially as previously described [85]. Briefly, Falcon® 4-well culture slides (Corning 354114) were precoated with rat tail collagen (Corning CB-40236), coated with media (M199\\u2009+\\u200910% Fetal Bovine Serum), and warmed in incubator for 30 mins, before removing media. Hearts were microdissected from E12.5 embryos, placed dorsal side down in individual chambers, and incubated at 37°C, 5% CO2 for 45 mins to allow adherence to slide. After 45 mins, 200uL of media was added to each chamber, and slides were incubated overnight at 37°C, 5% CO2. The next day, hearts were carefully removed and media was changed. After removal of the heart, explant cultures were incubated for 24 hours to allow for closure of acellular area where heart was attached to the plate. The next day, after confirming confluency of explant culture, cultures were scratched with a p10 pipette tip to create acellular area and imaged at t\\u2009=\\u20090,8,12, and 16 hours post scratch on a Zeiss AxioImager II microscope. Remaining acellular area was quantified in Adobe Photoshop and plotted by genotype in GraphPad Prism 9.\\nSingle-cell RNA-sequencing\\nEmbryos were dissected at E14.5, tail clipped, and placed in 2% BSA in ice cold PBS during genotyping with the KAPA Fast Genotyping Kit (~2 hours). Hearts were then pooled by genotype, dissected and dissociated in 200ug/mL Liberase TM (Sigma-Aldrich) dissolved in Hibernate E minus calcium media (Transnetyx). Single-cell suspensions loaded onto a 10X Genomics Next GEM Chip and emulsified with 3’ Single Cell Next GEM beads using a ChromiumTM Controller (10X Genomics). All embryos were dissected, digested, and barcoded on the same day to minimize variation. From barcoded cDNAs, gene expression libraries were constructed using ChromiumTM Single Cell 3′ Library Kits (10X Genomics) at the Translational Science Laboratory (Medical University of South Carolina). Next-generation sequencing was performed on each sample using an Illumina NovaSeq S4 flow cell at the VANTAGE facility (Vanderbilt University Medical Center). Fastq files were aligned to the mm10 reference transcriptome using the count function in the Cell Ranger software (v7.1, 10x Genomics, Pleasanton, CA, USA). Quality control was performed using a pipeline including exclusion of cells containing high mitochondrial read fraction, low detection of unique genes/ number of unique molecular identifiers per cell, and doublet filtering. Dimensional reduction and unsupervised clustering were performed RunUMAP and FindClusters functions in Seurat. Differential expression analyses were carried out in Seurat using DESeq2. Visualization and graphics were generated in Ggplot2. Functional analyses were conducted using ToppFun [32].\\nWestern blot\\nProtein was harvested from dissected embryonic hearts using 1X RIPA buffer (Thermo, 89900) with 1X Protease Inhibitor Cocktail (Abcam, ab271306), boiled at 95 °C for 5 mins with 4X Laemmli Buffer (BioRad, 1610747) with 10% 2-mercaptoethanol, and separated by gel electrophoresis using a BioRad mini blot system with 4−20% Mini PROTEAN TGX Stain-Free Protein Gels (BioRad, 456-8093). Protein was transferred to a nitrocellulose membrane using a BioRad Trans-Blot Turbo Semi-Dry Transfer System with Trans-Blot Turbo Mini Nitrocellulose Transfer Packs (BioRad, 170-4158). Membranes were washed in Ponceau S total protein stain (Thermo, A40000478) and imaged on a BioRad ChemiDoc MP imaging system for downstream normalization. Membranes were washed in Tris Buffered Saline with 1% Tween-20 (TBST) to remove Ponceau S, then blocked with 5% skim milk diluted in TBST. Membranes were then incubated overnight at 4 °C in primary antibody diluted 1:1000 in blocking buffer. Following three 10-min washes in TBST, membranes were then incubated at room temperature for 1 hour in HRP-conjugated secondary antibody diluted 1:7500 in blocking buffer. Following three 10-min washes in TBST, membranes were exposed to Pierce ECL Western Blotting Substrate (ThermoFisher, 32209) and imaged on a BioRad ChemiDoc MP imaging system.\\nProximity-based image analysis\\nTo assess the spatial relationship between epicardial-derived cells (marked by GFP) and coronary vasculature (marked by CD109), a custom image analysis pipeline was developed in R (source code available at https://github.com/andrewbharvey/proximity-based-image-analysis). Images were acquired from n\\u2009=\\u20094 embryos per genotype with 6 images analyzed per specimen. For each image, regions of interest were cropped to include only the compact myocardium, excluding the epicardium proper and trabecular myocardium. Channels were segmented and GFP+ and CD109\\u2009+\\u2009objects identified by Otsu thresholding. For each GFP+ cell, the shortest distance to the nearest CD109\\u2009+\\u2009neighbor was calculated. Distances were then used to categorize GFP+ cells into bins based on proximity. A threshold of 22 pixels derived from the mean distance between GFP+ and CD109\\u2009+\\u2009cells in control specimens was applied to stratify cells as either “perivascular” (<22 pixels) or “interstitial” (≥22 pixels). This binning enabled quantification of how loss of Sox9 influences spatial proximity to vasculature as a proxy for determining whether loss of Sox9 preferentially affects mural cells (perivascular) or interstitial fibroblasts (interstitial). Measurements were analyzed and visualized using GraphPad Prism 9.\\nStatistic and quantifications\\nStatistical analyses were performed in GraphPad Prism 9. Sample sizes and statistical tests are labeled in figure captions. For differential expression analyses of scRNA-seq data, the Wald test was used with Benjamini Hochberg False Discovery Rate adjustment. Genes were considered significant which had a Padj <\\u20090.05 and a |Fold Change|\\u2009>\\u20091.25. Chi-squared analysis was used to compare relative proportions of cell numbers within epicardial clusters between experimental groups (α\\u2009=\\u20090.0025). Fisher’s Exact Test was used for analysis of epicardial attachment in n\\u2009=\\u20097 specimens per genotype (α\\u2009=\\u20090.05). Simple linear regression analysis was used to analyze correlation between EPDC invasion (GFP+ fluorescent area) and presence of vasculature (CD109\\u2009+\\u2009fluorescent area) in cropped images of ventricular wall, with R2 values of 0.75, 0.50, and 0.25 described as substantial, moderate, and weak correlations, respectively, as per Henseler, 2009 [86]. Fluorescence area of GFP, CD109, and DAPI were calculated in CellProfiler 3.1.8. For these measurements, n\\u2009=\\u20093 specimens per genotype were analyzed, 4 sections (spaced at least 50µm apart) were analyzed per specimen, and 6 fields of view spaced equally around the left and right ventricles were analyzed per section. All data was analyzed and visualized in GraphPad Prism 9. For Western blot analysis of SOX18 levels, n\\u2009=\\u20096 specimens per genotype were analyzed from 3 independent E15.5 litters. For relative quantification, band intensities were determined in Adobe Photoshop, normalized to total protein intensity from Ponceau S, and compared in GraphPad Prism 9 (Student’s unpaired t-test, α\\u2009=\\u20090.05).',\n",
       "  'Results': 'SOX9 regulates epicardial-derived cell invasion during heart development\\nCells from the proepicardial organ (PEO), the extracardiac precursor to the epicardium, migrate towards and envelop the developing mouse heart, beginning around E9.5 [1] (Fig 1A). In our previous study, we described the generation of epicardial-specific Sox9 knockout mice using the previously described Wt1-Cre mouse (Wt1Cre) [11, 13]. By crossing in a Rosa26mTmG reporter allele (abbreviated R26mG), we were able to trace the epicardium and its progeny by GFP expression detected by immunofluorescence. For all histological and immunofluorescence analyses, mice heterozygous at the Sox9 locus (Wt1Cre; Sox9fl/+; R26mG) were phenotypically indistinguishable from Wt1Cre; Sox9+/+; R26mG specimens across all measured parameters, including epicardial attachment, EPDC invasion, myocardial thickness, and valve morphology [13]. They were thus deemed appropriate controls for comparison to Wt1Cre; Sox9fl/fl; R26mG littermates. This was consistent with prior studies using Sox9 conditional alleles in related contexts [13, 20, 21].\\nFig 1. SOX9 regulates epicardial-derived cell invasion.\\nOpen in a new tab\\n(A) Schematic showing epicardial formation from the proepicardial organ. Boxed region indicates area shown in (B–C’). (B–C’) Immunofluorescence shows in vivo SOX9 expression (red) in control (B, B’) and Wt1Cre; Sox9fl/fl; R26mG (C, C’) embryonic hearts at E10.5. MF20 (blue) is used as a myocardial marker. The Wt1Cre lineage is marked by GFP (green). Arrowheads in (B’) highlight SOX9\\u2009+\\u2009epicardial cells as they migrate from the proepicardial organ (PEO) to envelop the myocardium. Arrows in (C’) indicate epicardial cells which have undergone Cre-mediated recombination and no longer express SOX9.\\n(D) Schematic of epicardial explant and scratch assay. (E, F) Fluorescence images of epicardial explants from control and Wt1Cre; Sox9fl/fl; R26mG hearts at t\\u2009=\\u20090 and t\\u2009=\\u200916 hours post scratch (E), quantified in (F) (n\\u2009=\\u20092). Datapoints represent mean ±s.d.\\n(G–J) SOX9 expression in control (G, I) and Wt1Cre; Sox9fl/fl; R26mG (H, J) embryonic hearts at E12.5 (G, H) and E14.5 (I, J). By E12.5, SOX9 is no longer prominently expressed in the epicardium proper, but it is upregulated in a subset of EPDCs as they undergo EpiMT and invade the underlying ventricular myocardium (arrowheads, (G)).\\n(K) Wt1Cre; Sox9fl/fl; R26mG specimens display reduced EPDC invasion in comparison to both homozygous wildtype and heterozygous specimens (Wt1Cre; Sox9+/+; R26mG and Wt1Cre; Sox9fl/+; R26mG). Datapoints are mean\\u2009±\\u2009s.d. (n\\u2009=\\u20093). (Control hearts are from littermate matched Wt1Cre; Sox9fl/+; R26mG specimens. Atr, atrium; AVC, atrioventricular cushion; Epi, epicardium; LV, left ventricle; PEO, proepicardial organ; Ven, ventricle).\\nWe examined SOX9 expression in control and Wt1Cre; Sox9fl/fl; R26mG hearts by immunofluorescence in the early stages of the establishment of the epicardium. At E10.5, SOX9 was robustly expressed in the PEO, in epicardial cells as they migrate over the surface of the myocardium, and in the AV cushions of control specimens (Fig 1B, B’, arrowheads). In Wt1Cre; Sox9fl/fl; R26mG specimens, SOX9 was absent in GFP-labeled cells that had undergone Cre-mediated recombination in both the PEO and epicardium (Fig 1C, C’, arrows). Importantly, however, SOX9 was still present in GFP-negative populations of the PEO and epicardium at this stage, suggesting incomplete recombination by the Wt1Cre at this timepoint. Notably, the early establishment of the epicardium appeared grossly normal in Wt1Cre; Sox9fl/fl; R26mG specimens, though incomplete recombination at this stage precludes the conclusion that Sox9 does not play a role in this process.\\nAfter epicardial cells cover the surface of the heart, a subset will undergo an epicardial-to-mesenchymal transformation (EpiMT), giving rise to epicardial-derived cells (EPDCs), which will migrate into the underlying myocardium [6, 8, 22, 23]. To determine if SOX9 affects the migratory capacity of epicardial cells, we performed a scratch closure migration assay on epicardial explants from control (Wt1Cre; Sox9fl/+; R26mG) and Wt1Cre; Sox9fl/fl; R26mG hearts (Fig 1D). By E12.5, all epicardial cells expressed GFP, indicating effective Cre-mediated recombination. Fluorescence imaging of explant cultures confirmed low non-epicardial cell contamination (Fig 1E). Epicardial cells explanted from Wt1Cre; Sox9fl/fl; R26mG hearts showed diminished ability to migrate into the acellular area after scratch when compared to controls (Fig 1E, 1F).\\nTo examine EPDC invasion in vivo, we examined hearts by immunofluorescence at E12.5 and E14.5. At E12.5, SOX9 expression was negligible in the epicardium proper, but became upregulated in EPDCs undergoing epicardial-to-mesenchymal transformation (EpiMT) as they invaded the ventricular myocardium (Fig 1G, arrowheads). In Wt1Cre; Sox9fl/fl; R26mG hearts, the loss of Sox9 results in diminished EPDC invasion into the myocardium (Fig 1H). At E14.5, control hearts displayed extensive EPDC invasion into the myocardium, with heterogeneous expression of SOX9 (Fig 1I). In contrast, Wt1Cre; Sox9fl/fl; R26mG hearts exhibited a marked reduction of EPDC infiltration (Fig 1J). The reduction in EPDC invasion at E14.5 was quantified to be approximately 40% in Wt1Cre; Sox9fl/fl; R26mG hearts compared to both Wt1Cre; Sox9+/+; R26mG and Wt1Cre; Sox9fl/+; R26mG specimens (Fig 1K). We previously reported that reduced EPDC invasion was also accompanied by the presence of a thinner compact myocardial wall and decreased cardiomyocyte proliferation [13]. Overall, these findings suggest that Sox9 plays an important role in the migratory capacity of EPDCs as they transform to an invasive mesenchymal state.\\nSingle-cell RNA-sequencing of E14.5 hearts\\nTo better understand how loss of Sox9 in the epicardium affects heart development, we employed single-cell RNA-sequencing (scRNA-seq) to profile the transcriptome from Wt1Cre; Sox9+/+; R26mG (n\\u2009=\\u20093 pooled), Wt1Cre; Sox9fl/+; R26mG (n\\u2009=\\u20091), and Wt1Cre; Sox9fl/fl; R26mG (n\\u2009=\\u20093 pooled) hearts at E14.5. To enrich for cell populations of interest, we removed most of the left and right atrial appendage during microdissection (Fig 2A). Unbiased clustering visualized through a uniform manifold approximation and projection plot (UMAP) identified 17 transcriptionally distinct clusters (Fig 2B). After quality control, we obtained a dataset of 21,647 single cell transcriptomes across the three genotypes. Clusters were further merged based upon expression of known cell type markers into 13 clusters (Fig 2C). Cells from each genotype were present in each of the 13 clusters, which ranged in size from 46 cells to 4,976 cells (Fig 2D, S1 Fig). Given the absence of an observable phenotype in Wt1Cre; Sox9fl/+; R26mG specimens, Wt1Cre; Sox9+/+; R26mG and Wt1Cre; Sox9fl/+; R26mG data were grouped together as controls in downstream differential expression analyses for comparison to Wt1Cre; Sox9fl/fl; R26mG to prioritize discovery genes associated with the knockout phenotypes.\\nFig 2. Single-cell RNA-sequencing of E14.5 hearts.\\nOpen in a new tab\\n(A) Schematic showing the scRNA-seq sample preparation and experimental workflow.\\n(B) Uniform Manifold Approximation Projection (UMAP) representation of unbiased clustering analysis of 21,918 single-cell transcriptomes identifying 17 transcriptionally distinct clusters.\\n(C) Final annotated clusters after exclusion of low-quality clusters (14 and 15) and merging of similar cardiomyocyte clusters (1 and 7, 0 and 11).\\n(D) UMAP colored by genotype.\\n(E) Dot plot of gene expression for representative marker genes (x-axis) for each cell type cluster (y-axis) represented in (C).\\n(F) Frequency plot depicting number of cells in each cluster, split by group, with the size of the bubble indicating the proportion of the total number of cells in the group contained in that cluster, to normalized to the size of the control group.\\n(G) Biological Processes from functional enrichment analysis of significantly upregulated and downregulated genes (padj <\\u20090.05, |FC|\\u2009>\\u20091.25) in Wt1Cre; Sox9fl/fl; R26mG epicardial cells compared to controls. (H,I) Violin plots indicating relative expression levels of a subset of differentially expressed genes in control and Wt1Cre; Sox9fl/fl; R26mG epicardial cells. These genes have been previously implicated in important epicardial processes such as EpiMT, EPDC differentiation, and epicardial attachment. (Atr CM, atrial cardiomyocytes; AVC, atrioventricular cushion; AVC EC, atrioventricular cushion endothelial cells; Endo EC, endocardial endothelial cells; Epi, epicardium; FB, fibroblasts; Leuk, leukocytes; Mel, melanocytes; Pro AVC, proliferating atrioventricular cushion; Pro CM, proliferating cardiomyocytes; SMC, smooth muscle cells; Vasc EC, vascular endothelial cells; Vent CM, ventricular cardiomyocytes).\\nWe performed differential expression analysis for each cluster compared to all other clusters to manually annotate cell types (S1 Table). A subset of the markers used to identify clusters in Fig 2C are represented in a dot plot in Fig 2E. Epicardial (Epi) cells were identified by expression of the mesothelial and epicardial markers Upk3b, Wt1, and Tbx18 [11, 24, 25]. Fibroblasts (FB) were identified by high expression of Tcf21, Spon2, and Col1a1 [26]. Additionally, detection of the Rosa26mTmG reporter transcript, enhanced green fluorescent protein (Egfp), allowed for identification of the epicardial lineage, notably highly expressed in the Epi and FB clusters. The smooth muscle cell (SMC) cluster, marked by expression of Rgs5, Cspg4, and Cxcl12 [27] was largely Egfp-negative, indicating that this is likely a cluster of outflow tract SMCs of cardiac neural crest and second heart field origin. A subset of cells within the FB cluster, which are Egfp+, Rgs5+, Kcnj8+, and Cspg4+, likely represent mural cells of epicardial origin [28]. A full list of cell type markers can be found in S1 Table. Feature plots of a subset of these markers overlayed on the UMAP plot are displayed in S1 Fig.\\nComparison of cell type proportions between control and Wt1Cre; Sox9fl/fl; R26mG hearts revealed differences in cellular composition between groups (Fig 2F, S1 Fig). Notably, the Epi, FB, and Vasc EC clusters were substantially reduced in Wt1Cre; Sox9fl/fl; R26mG compared to controls, while the proportions of cardiomyocyte (CM) clusters were reciprocally increased (Fig 2F, S1 Fig). The reduced proportion of epicardial cells suggests that loss of Sox9 may impair epicardial survival or proliferation, leading to a smaller epicardial population, although we did not observe gross abnormalities in the epicardium proper at E14.5 (Fig 1J). Given that fibroblasts arise from the epicardium following EpiMT, a nearly 40% proportional decrease in the FB cluster in Wt1Cre; Sox9fl/fl; R26mG compared to controls further supports a reduction in EPDC presence in Wt1Cre; Sox9fl/fl; R26mG hearts, reduced in a similar proportion in quantifications from immunofluorescent stains (Fig 1K). While vascular endothelial cells are not predominantly derived from the epicardium [29], their reduction is likely an indirect consequence of diminished EPDCs in the myocardial wall. The epicardium and EPDCs contribute signals that regulate vascularization in a paracrine manner [9, 30, 31]. Thus, the loss of these supportive cues in Wt1Cre; Sox9fl/fl; R26mG hearts may underlie the observed decrease in Vasc EC. These findings suggest that Sox9 is essential for maintaining epicardial integrity, with broad consequences for downstream mesenchymal and endothelial lineages.\\nDysregulation of genes implicated in epicardial invasion and differentiation\\nWe next performed differential expression analysis between the epicardial cluster in control and Wt1Cre; Sox9fl/fl; R26mG specimens (S2 Table). We input differentially expressed upregulated and downregulated genes (padj <\\u20090.05, Log2 |FC|\\u2009>\\u20091.25) to ToppGene suite’s functional enrichment analysis tool, ToppFun [32], which revealed upregulation of genes in cytoskeletal and epithelial-barrier related pathways in Wt1Cre; Sox9fl/fl; R26mG epicardial cells compared to controls (Fig 2G). Genes that were downregulated were enriched for biological processes related to vasculature development and angiogenesis (Fig 2G).\\nDownregulated genes included various transcriptional regulators, secreted factors, and ECM components that have been previously implicated in a variety of processes in epicardial biology. Wt1Cre; Sox9fl/fl; R26mG epicardial cells displayed dysregulation of factors known to influence EpiMT including Cebpb, Cebpd, and Tcf21 [33, 34] (Fig 2H). While TCF21 and C/EBP-family transcription factors have been shown to be positive regulators of EpiMT that were downregulated in Wt1Cre; Sox9fl/fl; R26mG epicardial cells, we also saw upregulation of Cldn3 and Crip1, negative regulators of EMT [35, 36] (S2 Fig). Additionally, we observed downregulation of two transcription factors reported to influence epicardial heterogeneity and differentiation potential, Bnc1 and Tcf21 [34, 37–39] (Fig 2H). Epicardial cells from Wt1Cre; Sox9fl/fl; R26mG hearts expressed lower levels of secreted factors known to influence myocardial growth when compared to controls, including cardiomyocyte mitogens Fgf2 and Fgf9 [40, 41], and canonical Wnt signaling ligand Wnt5a [42], offering a potential explanation for decreased cardiomyocyte proliferation described in this model [13] (S2 Fig). Overall, these data support a model in which the loss of Sox9 results in a transcriptional program rendering the epicardium less conducive to undergoing EpiMT, supporting our immunofluorescence observations of reduced numbers of EPDCs within the myocardial wall at this timepoint (Fig 1). Additionally, we identified downregulation of two type IV collagen chains, Col4a2 and Col4a5, in Wt1Cre; Sox9fl/fl; R26mG epicardial cells when compared to controls (Fig 2I).\\nSOX9 mediates epicardial attachment to underlying myocardium\\nType IV collagen is the most abundant protein component of basement membranes—a dense ECM layer on the basal surface of epithelia that forms a scaffolding anchoring the epithelium to the underlying tissue or matrix [43]. Morphological assessment of hearts at E12.5 revealed defective attachment of the epicardium to the underlying myocardium in 5/7 of Wt1Cre; Sox9fl/fl; R26mG specimens, while all control littermates displayed a close apposition of the epicardium to the myocardium (Fig 3A–3C). When examined by immunofluorescence, Wt1Cre; Sox9fl/fl; R26mG specimens displayed reduced expression of type IV collagen (COL4) in regions of blistering when compared to controls at E12.5 (Fig 3D–E”’). To further evaluate the temporal dynamics of COL4 expression and epicardial integrity, we performed additional immunostaining at E14.5 and E16.5. At E14.5, while subepicardial expression of COL4 remained reduced in Wt1Cre; Sox9fl/fl; R26mG specimens compared to controls, the epicardium appeared properly attached in all specimens examined (Fig 3F–G”’). This suggests that Sox9 is not required for maintaining epicardial attachment at this stage. By E16.5, COL4 levels were similarly low in both control and Wt1Cre; Sox9fl/fl; R26mG hearts (S3 Fig), indicating a broader downregulation of this basement membrane component through development. These findings suggest that Sox9 may play a temporally specific role during the initial attachment of the epicardium at E12.5, but is dispensable for maintenance of this attachment at later stages. There is supporting evidence that SOX9 activates the Col4a2 promoter in mouse kidney mesangial cells [44], and that siRNA-mediated knockdown of SOX9 leads to a reduction in COL4 protein in human mesangial cells [45], although this regulation may be indirect. Taken together, our results support a model in which Sox9 regulates expression of basement membrane components such as COL4 during early epicardial development, and that disruption of this regulation may contribute to the epicardial blistering observed at E12.5 in Wt1Cre; Sox9fl/fl; R26mG specimens.\\nFig 3. Loss of Sox9 results in defective epicardial attachment.\\nOpen in a new tab\\n(A–C) Hematoxylin and Eosin stains show control (A) and Wt1Cre; Sox9fl/fl; R26mG (B) hearts at E12.5. While the epicardium was adhered to the myocardium in all control specimens examined at this time point, arrows in (B) highlight epicardial blistering seen in 5/7 Wt1Cre; Sox9fl/fl; R26mG specimens as quantified in (C) (Fisher’s Exact Test). (D–E”’) Immunofluorescent stains show the epicardium (GFP, green) and the basement membrane type 4 collagen (COL4, red) in control (D–D”’) and Wt1Cre; Sox9fl/fl; R26mG (E–E”’) embryonic hearts at E12.5. Arrowheads in (D’–D”’) highlight the epicardium anchored to the underlying myocardium (Myo) by a continuous COL4\\u2009+\\u2009basement membrane, while arrows in (E’–E”’) indicate lower COL4 expression in regions of epicardial blistering. (F–G”’) Immunofluorescent stains at E14.5 showing the epicardium and COL4 in control (F–F”’) and Wt1Cre; Sox9fl/fl; R26mG (G–G”’) hearts. At this stage, the epicardium is properly attached in both groups, though COL4 expression remains lower in Wt1Cre; Sox9fl/fl; R26mG specimens. Arrowheads in (F’–F”’) highlight the subepicardial expression of COL4, while arrows in (G’–G”’) indicate reduced subepicardial COL4 expression. (Control hearts are from littermate matched Wt1Cre; Sox9fl/+; R26mG specimens. Endo, endocardium; LV, left ventricle; Myo, myocardium; OFT, outflow tract; RV, right ventricle).\\nEPDC invasion correlates with vascular plexus formation\\nEPDCs play important roles in the formation of the coronary vascular network, with many mouse models with perturbed epicardial formation and EPDC invasion also displaying significant vascular defects [42, 46–51]. Given the proportional decrease in Vasc ECs in Wt1Cre; Sox9fl/fl; R26mG hearts (S1 Fig) and the downregulation of genes related to vascular development (Fig 2G), we wanted to investigate the relationship between reduced EPDC invasion observed in Wt1Cre; Sox9fl/fl; R26mG specimens (Fig 1) and the development of the coronary vascular plexus. To do this, we quantified the fluorescent area of EPDCs (GFP) and vasculature as marked by CD109—a cell surface protein we identified previously as marking the coronary vascular plexus [13]—within the compact myocardium of the ventricular walls and performed a linear regression analysis (Fig 4A–4C). Regression analysis revealed a positive correlation of moderate strength between EPDC invasion and CD109\\u2009+\\u2009vasculature (R2=\\u20090.57, Fig 4C). Datapoints from Wt1Cre; Sox9fl/fl; R26mG tissue sections tended to fall lower along the line of best fit (Fig 4C, red 95% confidence ellipse), indicative of less EPDC invasion and less CD109\\u2009+\\u2009vasculature in Wt1Cre; Sox9fl/fl; R26mG specimens relative to controls, which fell higher along the line (Fig 4C, blue 95% confidence ellipse). This correlation reinforces a large body of existing research highlighting the importance of the presence of EPDCs in the myocardial wall for proper vascular development.\\nFig 4. Reduced EPDC invasion correlates with deficient coronary vascular plexus development.\\nOpen in a new tab\\n(A, B) Immunofluorescent stains marking EPDCs (GFP, green), vasculature (CD109, red), and myocardium (MF20, blue) in control (A) and Wt1Cre; Sox9fl/fl; R26mG (B) hearts at E14.5. Area between dotted lines represents compact ventricular myocardium.\\n(C) Linear regression analysis reveals a positive correlation (R2=\\u20090.57) between invading EPDCs and vasculature within the myocardial wall. Datapoints represent quantification from individual image fields, 18 fields per specimen, n\\u2009=\\u20093 specimens per genotype. 95% confidence ellipses show that sections from control hearts (red ellipse) tend to have more EPDC invasion and more CD109\\u2009+\\u2009vasculature than Wt1Cre; Sox9fl/fl; R26mG specimens (blue ellipse).\\n(D) Magnification of the endothelial portion of the UMAP from Fig 2C with a feature plot of Sox18 shows that Sox18 is primarily expressed in the Vasc EC cluster. Violin plot shows Sox18 expression levels in in control and Wt1Cre; Sox9fl/fl; R26mG Vasc EC cells. Dots represent relative expression level of individual cells.\\n(E, F) Immunofluorescent stains marking vasculature (PECAM1, green), myocardium (MF20, blue), and SOX18 (red) suggest a decrease in SOX18 expression in the vasculature of Wt1Cre; Sox9fl/fl; R26mG specimens compared to controls at E15.5.\\n(G) Western blot confirms downregulation of SOX18 in protein isolated from E15.5 whole hearts (n\\u2009=\\u20096, 3 independent litters). Data are mean ±s.d. (Student’s unpaired t-test). (Control hearts are from littermate matched Wt1Cre; Sox9fl/+; R26mG specimens. RCA, right coronary artery; RV, right ventricle).\\nWe next went back to our scRNA-seq data to probe for genes differentially expressed by genotype in the Vasc EC cluster (S2 Table). This revealed a nearly 3.5-fold downregulation of the Sox18 transcript in Wt1Cre; Sox9fl/fl; R26mG Vasc ECs compared to controls (Fig 4D). Sox18 encodes a pro-angiogenic transcription factor that plays cooperative roles with Sox7 and Sox17 in regulating angiogenesis [52, 53]. Mutations in Sox18 lead to vascular defects and rupture in the Ragged spontaneous mutant mouse [54–56]. Immunofluorescent labeling for SOX18 (Fig 4E, 4F) and Western blot analysis of protein isolated from control and Wt1Cre; Sox9fl/fl; R26mG hearts at E15.5 (Fig 4G) confirmed downregulation of SOX18 protein in Wt1Cre; Sox9fl/fl; R26mG specimens, suggesting a potential role for Sox18 in the delayed coronary vascular plexus phenotype. Further investigation is required to determine mechanisms by which loss of Sox9 in the epicardial lineage leads to downregulation of Sox18 in Vasc ECs, and whether this downregulation directly contributes to delayed coronary plexus formation in Wt1Cre; Sox9fl/fl; R26mG specimens.\\nEpicardial subpopulation enriched in Wt1Cre; Sox9fl/fl; R26mG hearts\\nTo investigate transcriptional distinctions between genotypes in more depth, we performed subsetting and reclustering on the Epi cluster (Fig 5A). This revealed four epicardial subpopulations, each expressing epicardial marker Wt1 and reporter gene Egfp (Fig 5B). Chi-square analysis of cell frequencies in each subpopulation revealed a significant enrichment of control cells in the Epi 1 subpopulation, and an enrichment of Wt1Cre; Sox9fl/fl; R26mG cells in Epi 2.(Fig 5C). When we looked at marker genes expressed in these subpopulations, we saw that Epi 1 is marked by genes with reported roles in coronary vascular development: vascular endothelial growth factor C (Vegfc), which promotes the formation of the coronary plexus when secreted by the epicardium [29]; dipeptidyl-peptidase 4 (Dpp4), a cell surface protease that cleaves extracellular matrix and matrix-embedded growth factors to enable cell invasion and migration promoting vascularization [57]; Sulfatase 1 (Sulf1), a heparan sulfate-editing enzyme promotes cardiac angiogenesis by freeing vascular endothelial growth factors; and semaphoring 3D (Sema3d), a marker of cells from the proepicardial organ which contribute to the coronary vascular endothelium [58] (Fig 5D, S3 Table). Interestingly, Epi 1 is also enriched for Erbb4 (Fig 5D), a neuregulin receptor that regulates cell growth and survival in the myocardium [59] but has not yet been interrogated in the epicardium. Epi 2, which was enriched with Wt1Cre; Sox9fl/fl; R26mG cells, was characterized by expression of genes often used as markers of VSMC and pericyte differentiation, including alpha-smooth muscle actin (Acta2) [60], transgelin 1 and 2 (Tagln and Tagln2) [61–63], elastin (Eln) [64, 65], and GATA-binding 6 (Gata6) [66, 67] (Fig 5E, S3 Table). These genes not only marked the Epi 2 cluster but were expressed in higher levels in Wt1Cre; Sox9fl/fl; R26mG epicardial cells compared to controls (Fig 5F). Epi 3 was marked by high levels of ribosomal genes indicative of cellular stress, while the Pro Epi cluster expressed high levels of proliferation genes including Mki67, Cenpa, and Top2a (S3 Table).\\nFig 5. Single-cell transcriptomic profiling of control and Wt1Cre; Sox9fl/fl; R26mG epicardial cells.\\nOpen in a new tab\\n(A) UMAP depicting a subsetting and reclustering of the epicardial cell cluster.\\n(B) Feature plots of epicardial marker Wt1 and reporter transcript Egfp indicated expression in all subpopulations of epicardial cells.\\n(C) Relative proportion of cells in each cluster from (A) split by experimental group. Frequencies are listed in parentheses, and asterisks indicate significant differences in frequency between groups (chi-squared analysis, * p\\u2009<\\u20090.0025).\\n(D) Feature plots projected on UMAP from (A) indicating expression of a subset of genes enriched in Epi 1 with red indicating high expression.\\n(E) Feature plots projected on UMAP from (A) indicating expression of a subset of genes enriched in Epi 2 with red indicating high expression.\\n(F) Violin plots indicating relative expression levels of genes from (E) in control and Wt1Cre; Sox9fl/fl; R26mG epicardial cells. In all violins, dots represent relative expression level of individual cells.\\nTo further characterize these subpopulations, we looked at the epicardial-derived fibroblast marker Tcf21 and determined this was not enriched in any specific subpopulation of the epicardium (S2 Fig), although Tcf21 was significantly downregulated in Wt1Cre; Sox9fl/fl; R26mG epicardium and fibroblast clusters compared to controls (Fig 3D, S4 Fig, S2 Table). Nor were the EMT markers Snai1 or Snai2 enriched in any subpopulation of epicardial cells (S2 Fig), suggesting that Epi 2 is not a cluster undergoing EpiMT, but may represent a cluster of epicardial cells prematurely differentiating into VSMCs, or primed for the mural cell fate. Interestingly, Tbx18 expression was notably weak in this subpopulation (S2 Fig), which has been reported to have an inhibitory effect on VSMC differentiation [68, 69]. These observations are supported by prior literature independently describing roles for Sox9, Tcf21, and Tbx18 in the repression of SMC differentiation [38, 68–73], such that loss of Sox9 and subsequent downregulation of Tcf21 in Wt1Cre; Sox9fl/fl; R26mG hearts leads to enrichment of a Tbx18-low epicardial subpopulation expressing high levels of VSMC markers. An important limitation of these data to note is that the E14.5 timepoint is before coronary VSMCs have matured, making further investigation of later timepoints critical for confirming a role for Sox9 in epicardial-derived VSMC fate determination. Loss of Sox9 alters the spatial distribution of EPDCs.\\nLoss of Sox9 results in depletion of interstitial EPDCs\\nIt is notoriously difficult to distinguish vascular smooth muscle cells (VSMCs) from fibroblasts using transcriptional markers, as many commonly used genes are expressed in both lineages [74]. To investigate whether loss of Sox9 preferentially affects one EPDC lineage over the other, we used a proximity-based approach to assess the spatial relationship between GFP+ EPDCs and the coronary vasculature. We reasoned that EPDCs in close proximity to coronary vasculature are more likely to be mural cells (VSMCs and pericytes), while those located further from vasculature are more likely to be fibroblasts. At E14.5, VSMC coverage of the developing coronary plexus is patchy and incomplete, as demonstrated by Volz et al. [75], so we analyzed ventricular sections at E16.5 (Fig 6A–6G). This reflects a timepoint at which coronary arteries have begun to mature and mural cells have been recruited to the vessel wall. We performed immunofluorescent staining for EPDCs (GFP), coronary endothelial cells (CD109), and myocardium (MF20), and we generated image analysis code to detect GFP+ and CD109\\u2009+\\u2009objects in cropped regions of compact myocardium (Fig 6B–6G).\\nFig 6. Altered spatial distribution of EPDCs in Wt1Cre; Sox9fl/fl; R26mG hearts.\\nOpen in a new tab\\n(A) Schematic depicting the ventricular regions analyzed (boxed).\\n(B–G) Proximity analysis workflow. Immunofluorescent staining of E16.5 heart sections from control (B) and Wt1Cre; Sox9fl/fl; R26mG (E) hearts showing EPDCs (GFP, green), coronary vasculature (CD109, red), and myocardium (MF20, blue). (C) and (F) are representative images used for analysis, showing GFP and CD109 channels cropped to remove the epicardium proper, trabeculated myocardium, and ventricular lumen. (D) and (G) show the GFP+ (green) and CD109+ (red) objects detected by Otsu’s auto-thresholding method.\\n(H) Quantification of GFP+ fluorescence area, first normalized to DAPI area to account for myocardial area differences, then normalized to control (set to 1.0) for comparison between genotypes (n\\u2009=\\u20094, Student’s unpaired t-test).\\n(I) Nearest-neighbor analysis reveals a decreased average distance between GFP+ object centroids and their nearest CD109\\u2009+\\u2009neighbor in Wt1Cre; Sox9fl/fl; R26mG specimens. Each datapoint is the average distance to the nearest CD109\\u2009+\\u2009neighbor for each GFP+ object in a single image. Datapoint shapes correspond to technical replicates (images) for each specimen (n\\u2009=\\u20094 specimens, 6 images per specimen, Student’s unpaired t-test).\\n(J) Proportion of GFP+ fluorescence area classified as perivascular (black) or interstitial (gray), based on proximity to nearest CD109\\u2009+\\u2009neighbor, as described in Materials and Methods.\\n(K) Table summarizing object detection counts for perivascular and interstitial GFP+ objects by genotype. Chi-square analysis indicates a significant shift in spatial distribution (p\\u2009<\\u20090.0001). (L–M’) Immunofluorescence shows SOX9 expression (blue), primarily in EPDCs (GFP, green) positioned interstitially (arrowheads), while those positioned near vasculature (CD109, red), are largely SOX9-negative (arrows). Data are mean\\u2009±\\u2009s.d. (Control hearts are from littermate matched Wt1Cre; Sox9fl/+; R26mG specimens. LA, left atrium; LV, left ventricle; RA, right atrium; RV, right ventricle).\\nFluorescence quantifications revealed that the overall GFP+ area was significantly reduced in Wt1Cre; Sox9fl/fl; R26mG hearts compared to controls, confirming a sustained reduction in EPDC invasion at later stages (Fig 6H). To explore whether this reduction reflected a shift in EPDC localization, we measured the average distance from each GFP+ cell to its nearest CD109\\u2009+\\u2009neighbor. In Wt1Cre; Sox9fl/fl; R26mG hearts, GFP+ cells were significantly closer to the vasculature than in controls (Fig 6I), suggesting that loss of Sox9 alters the spatial distribution of EPDCs within the myocardium. To better understand this shift, we binned GFP+ cells into “perivascular” or “interstitial” groups based on their proximity to the nearest CD109\\u2009+\\u2009cell. This analysis revealed that the reduction in GFP+ area in mutants was primarily due to loss of interstitial EPDCs, with perivascular EPDCs remaining relatively unaffected (Fig 6J). Object counts confirmed a significant redistribution of EPDCs in the mutant hearts (Fig 6K).\\nLastly, we examined SOX9 protein expression at E14.5 and E16.5, which confirmed that SOX9 was primarily expressed in interstitial EPDCs (Fig 6L–M’, arrowheads) and largely absent from perivascular cells (Fig 6L–M’, arrows). These findings suggest that SOX9 is important for the maintenance of the epicardial-derived fibroblast lineage, and that its loss results in a selective depletion of this population.',\n",
       "  'Discussion': 'This report provides insight into the roles of Sox9 in the epicardium as it relates to the regulation of migratory and differentiation events important for proper heart development. Epicardial-specific loss of Sox9 led to a delay in the attachment of the epicardium to the myocardium, as evidenced by a high prevalence of epicardial blistering or blebbing away from the myocardium in E12.5 Wt1Cre; Sox9fl/fl; R26mG specimens. While this blistering is no longer observed at E14.5, reduced invasion in Wt1Cre; Sox9fl/fl; R26mG points to an important role for Sox9 in regulating the proper attachment and spatiotemporally regulated invasion of EPDCs into the myocardium.\\nOur single-cell transcriptomic analysis identified key regulators of epicardial development that were differentially expressed in Wt1Cre; Sox9fl/fl; R26mG specimens compared to controls at E14.5. Notably, loss of Sox9 led to downregulation of two transcription factors reported to influence epicardial heterogeneity and differentiation potential, Bnc1 [39] and Tcf21 [34, 37, 38], as well as dysregulation of factors influencing EpiMT including Tcf21 [34], Cebpb and Cebpd [33], and Crip1 [36]. In human pluripotent stem cell-derived epicardium (hPSC-epi), BNC1 and TCF21 marked distinct subpopulations, with TCF21\\u2009+\\u2009hPSC-epi cells retaining the ability to differentiate into both fibroblasts and VSMCs, while BNC1\\u2009+\\u2009hPSC-epi cells predominantly differentiated into VSMCs [39]. While these subpopulations were not specifically observed in our dataset, downregulation of both Bnc1 and Tcf21 in Wt1Cre; Sox9fl/fl; R26mG epicardial cells is suggestive of an epicardium with diminished differentiation capacity. In mice, Tcf21 represses transcription of VSMC genes, and regulates lineage-specific EpiMT of cardiac fibroblasts [34, 73]. Additional regulators of EpiMT include C/EBP-family transcription factors, which promote Wt1 and Raldh2 expression in the epicardium, and are reactivated after ischemic injury [33]. Additionally, lower expression of basement membrane collagens and growth factors secreted by the epicardium provide potential explanations for the epicardial blistering and myocardial hypoproliferation phenotypes observed in Wt1Cre; Sox9fl/fl; R26mG specimens.\\nWe observed a correlation between EPDC invasion and the presence of the coronary vascular plexus in this model, where reduced EPDCs in the myocardium of Wt1Cre; Sox9fl/fl; R26mG specimens correlates with reduction in CD109-expressing vasculature within the myocardial wall. This observation bolsters a large body of literature describing the importance of EPDCs in the proper development and patterning of the coronary vascular plexus [31, 42, 69, 76, 77]. We identified the pro-angiogenic transcription factor SOX18 [52–54] as a potential mediator of this phenotype, though the mechanisms of how signaling from EPDCs may influence SOX18 expression in Vasc ECs remains unclear.\\nWe discovered in Wt1Cre; Sox9fl/fl; R26mG hearts an expansion of a subpopulation of epicardial cells with characteristics of both epicardial cells and differentiating VSMCs (Epi 2), and a corresponding reduction of epicardial cells marked by expression of proangiogenic factors (Epi 1). Previous literature has implicated Sox9 in playing an inhibitory role in VSMC differentiation by suppressing the transcription of VSMC- specific gene programs (Acta2, Cnn1, Tagln) through its interaction with myocardin (MYOCD) [70, 71]. Repression of Sox9 by the canonical Notch ligand Jagged1 (JAG1) has also been reported to be important for VSMC differentiation, as the lack of this repression, or the misexpression of Sox9, resulted in a differentiation of VSMC precursors into chondrocytes in the developing aorta [72]. Additional roles for Tcf21 and Tbx18 in inhibiting or repressing VSMC differentiation have also been described previously: epicardial loss of Tcf21 or epicardial expression of an transcriptional activator form of Tbx18 (instead of its wildtype transcriptional repressor activity [78]) each result in premature differentiation of epicardial cells into VSMCs [38, 69, 73]. The studies alongside our transcriptomic observations support a model in which the loss of Sox9 and subsequent downregulation of Tcf21 in Wt1Cre; Sox9fl/fl; R26mG epicardium leads to expansion of a Tbx18-low subpopulation of epicardial cells expressing higher levels of VSMC genes.\\nAn unbiased proximity-based image analysis approach to interrogate the spatial relationship between EPDCs and coronary vasculature supported our hypothesis that loss of Sox9 is preferentially detrimental to the interstitial mesenchymal EPDC populations, while leaving the perivascular EPDC populations intact. Future studies can incorporate additional timepoints to further investigate maturation and differentiation trajectories at a transcriptional level and dissect how loss of Sox9 influences these trajectories.\\nTogether, these data identify an important role for Sox9 in epicardial attachment, invasion, and differentiation that is important for proper myocardial and coronary vascular development.\\nFig 7 is a model illustrating the phenotypic consequences of loss of Sox9 in the epicardial lineage. Sox9 deletion leads to aberrant epicardial attachment, reduced EPDC invasion, and a preference towards the VSMC fate. Reduced expression of Tcf21 in Wt1Cre; Sox9fl/fl; R26mG specimens may mediate lineage specific effects of Sox9 deletion, and reduced expression of Sox18 may contribute to a deficiency in coronary plexus development, but further analyses are required to confirm mechanisms underlying these phenotypes.\\nUnderstanding how epicardial cells function to regulate the growth and development of the myocardium and coronary vasculature has intriguing therapeutic implications. Existing evidence supports that the “reactivation” or “reprogramming” of the adult epicardium to a developmental state may promote revascularization and improve ejection fraction after ischemic injury to the myocardium, though there are significant limitations preventing translation of these findings into clinical practice currently [33, 79–83]. These include limitations in how we understand the molecular initiation and termination of these activation signals, as well as the potential for functional divergence between the human epicardial response when compared to model organisms such as mouse. Further research into mechanisms of epicardial invasion, differentiation, and function could illuminate novel approaches to improve cardiac outcomes in patients with congenital heart disease, diseases of the coronary system, or ischemic injury.',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1371/journal.pone.0325852',\n",
       "  'Journal': 'PLoS One'},\n",
       " {'PaperTitle': 'Endometriosis and ovulatory menstruation: beyond the Sampson principle',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Serdar E Bulun',\n",
       "  'Email': 's-bulun@northwestern.edu',\n",
       "  'Abstract': 'Endometriosis is an estrogen-dependent chronic inflammatory syndrome characterized by viable endometrial tissue outside the uterine cavity and associated with pain and infertility. Endometriosis, as tissue or a pathological process, is dynamic in that its establishment and progression require repeated episodes of retrograde travel of shed endometrial tissue, which implants in the lower abdominal cavity following ovulatory cycles and survives. Estrogen-rich follicular fluid released onto peritoneal surfaces during ovulation may also support endometriotic implants. DNA evidence indicates that endometriosis originates from eutopic endometrial tissue, which may reach the abdominal cavity in a retrograde manner primarily via the uterine tubes. Unlike uterine bleeding associated with non-ovulatory circumstances, retrograde menstruation following an ovulation maximizes shedding of epithelial cells localized to deep invaginations of the basalis portion of the endometrium, which likely carry somatic cancer-driver mutations such as KRAS. The attached endometrial stromal cells are mostly mutation free but display epigenetic defects including overexpression of aromatase and estrogen receptor-β and downregulation of progesterone receptor, causing estrogen excess and progesterone resistance. These tissue clones may form implants in involuting ovarian corpus luteum cysts and peritoneal surfaces and induce tissue remodeling and fibrosis, manifested as deep-infiltrating endometriosis. The first-line treatment for chronic pelvic pain associated with endometriosis is suppression of ovulation, with the goal of relieving pain. Infertility is often managed using in vitro fertilization, which improves the embryo quality and alters endometrial development.',\n",
       "  'Introduction': 'This Review focuses on the role of the menstrual event that follows ovulation in the pathophysiology of endometriosis. Endometriosis is a unique inflammatory syndrome resulting from repeated episodes of ovulatory menstrual bleeding in a subset of people who menstruate, a population comprising diverse gender and biological sex identifications. This Review will mainly refer to women because they are highly represented in the clinical literature (1, 2). Endometriosis, as tissue or a pathological process, is not static but dynamic in that repeated episodes of retrograde travel of shed endometrial tissue, which implants and survives in the lower abdominal cavity following ovulatory cycles, are indispensable for its establishment and progression (1, 2). Its primary symptoms are painful menstrual periods, chronic pelvic pain, painful intercourse, and infertility (3–6). The dependence of endometriosis on estrogen to drive the inflammatory process and pain separates it from other chronic inflammatory syndromes such as Crohn’s disease or acid reflux esophagitis (7, 8). Until recently, a prevailing notion that prioritized the surgical description of endometriosis over the disease process unfortunately limited this syndrome to a narrow anatomic definition, i.e., the ectopic presence of endometrium-like tissue outside of the uterine cavity; this definition profoundly stalled progress for developing long-term and patient-centered management strategies (9, 10). This arbitrary weighted and purely anatomic classification system had limited value to predict relief of pain or achievement of pregnancy in response to various regimens of symptom management (11, 12). A recent shift toward developing new paradigms to understand and manage the symptoms of endometriosis through the various stages of the disease process puts patients first (4, 6, 13, 14). This new overall approach recognizes its early manifestations during adolescence, the long history of disease, its recurrent symptoms, and the overall limitations of the currently available medical and surgical approaches (Figure 1).\\nFigure 1. The clinical course of endometriosis over a lifetime.\\nOpen in a new tab\\nThe top row summarizes typical patient experiences beginning in the teenage years and continuing through their 40s and 50s. The lower row lists pain management strategies recommended during each decade of life. Although hysterectomy and bilateral removal of tubes and ovaries are frequently used, they represent severely radical approaches and may not provide pain relief in a subset of patients. COCs, combined oral contraceptives; NSAID, nonsteroidal antiinflammatory drug; GnRH, gonadotropin-releasing hormone.\\nA number of hypotheses have been proposed to explain the presence of endometrial tissue outside the uterine cavity (15). This Review focuses on the mechanism, which is supported by an extraordinarily large body of consistent scientific evidence that indicates endometriosis and its symptoms are caused by retrograde travel of menstrual blood containing molecularly defective endometrial tissue fragments onto the lower abdominal peritoneal surfaces and the ovaries following an ovulation (Figure 2) (16–19). In 1927, John Sampson was the first to provide evidence that this retrograde spread into the pelvic cavity could take place via the uterine tubes or venous circulation (18–20). Also known as Sampson’s theory, this straightforward mechanism was recently demonstrated via mapping of the identical DNA variants or mutations in the epithelial cells of the endometrium to the ectopic endometriotic implants in hundreds of endometriosis patients (21–25). Adenomyosis, a disorder defined as the presence of multiple islands of endometrial tissue in the myometrium, is closely related to endometriosis (26, 27). Adenomyosis results from survival and growth of entrapped fragments of basalis endometrium in the myometrial layer (25, 27). This concept of intracavitary endometrium as the source of pelvic endometriosis and its symptoms has also been emphasized in a number of recent review articles (4, 6, 14).\\nFigure 2. Ovulatory cycle.\\nOpen in a new tab\\nRepetitious ovulatory cycles initiated by hypothalamic gonadotropin-releasing hormone (GnRH) neurons are essential for the establishment and progression of endometriosis (see Figure 3). GnRH, which is secreted in a pulsatile manner, acts on its seven-transmembrane G protein–coupled receptor on the gonadotroph and induces sequential secretion of follicle-stimulating hormone (FSH) and luteinizing hormone (LH) from the pituitary. FSH and LH, via their specific G protein–coupled receptors on ovarian granulosa and theca cells, stimulate sequential secretion of estradiol and progesterone. Estradiol during the proliferative phase builds the functionalis layer of the endometrium, whereas progesterone during the secretory phase stimulates endometrial differentiation. In the absence of pregnancy, endometrial tissue is shed through the vagina and also in a retrograde manner through the uterine tubes into the lower abdominal cavity.\\nThere is also consensus that endometriosis likely affects a woman chronically from adolescence until menopause (Figure 1) (4, 14). The paradigm that dysmenorrhea should be defined as primary or secondary (secondary being associated with pelvic pathology) is unhelpful and implies that imaging or laparoscopy is necessary to make a presumptive diagnosis of endometriosis (28). Labeling severe menstrual cramps as primary dysmenorrhea during adolescence just because the patient has not gone through a workup including laparoscopy assumes, by definition, that dysmenorrhea can be managed without understanding of its etiology if ovulation is suppressed together with regimens of NSAIDs for an undefined period of time (28, 29). This diagnostic approach does not allow the clinician to consider long-term preventative measures for the progress of endometriosis. Moderate to severe dysmenorrhea experienced during adolescence likely represents the initial stages of the long disease process of endometriosis (Figure 1). The modern, patient-centric definition of endometriosis is more clinically helpful, since it is used to describe the symptom complex that a patient experiences throughout their reproductive life (Figure 1) (4, 6, 14). An endometriosis patient may or may not have the surgically demonstrable presence of endometrium-like tissue in their lower abdominal organs, including the ovaries, because microscopic disease may be missed. In approximately 10% of patients endometriosis and its symptoms persist after menopause or even after the surgical removal of their uterus and ovaries (8, 30, 31).\\nEndometriosis is the most common cause of painful menses and chronic pelvic pain in women; and it is often associated with infertility (5, 32, 33). Worldwide, approximately 200 million reproductive-age women suffer from endometriosis and experience painful and heavy menstrual bleeding, painful intercourse, and pelvic pain outside of menses (34, 35). More than 100,000 hysterectomies and many more laparoscopies for the diagnosis and conservative removal or ablation of endometriosis are performed annually in the United States (36, 37). Multiple studies showed that the suffering and economic burden associated with endometriosis are very high and resemble those of other chronic inflammatory conditions such as Crohn’s disease and rheumatoid arthritis, except that endometriosis may be more prevalent (38, 39). The analyses used in these studies focus on health care costs and usually severely underestimate productivity loss and decreased quality of life (38, 39). There are additional obstetrical risks associated with endometriosis: a history of endometriosis increases the risk of early pregnancy loss, preterm delivery, neonatal admission to the intensive care unit, and gestational diabetes (40, 41).\\nDiffuse adenomyosis is associated with prolonged menstrual bleeding, painful menses, and recurrent pregnancy loss (26, 27, 42). The presence of adenomyosis together with endometriosis elevates the risk of placenta previa or a cesarean delivery (40).\\nThe hallmark symptoms of endometriosis, i.e., dysmenorrhea, dyspareunia, and chronic pelvic pain, are associated with retrograde travel of menstrual blood and tissue and an inflammatory response (Figure 2) (1, 34, 43–46). Stimulation of the nociceptors on peritoneal surfaces by the menstrual tissue or established ectopic implants is thought to be the major source of pain (47, 48). Nerve-like tissue has been described in the endometrium of patients with endometriosis, raising the question of whether intracavitary endometrium may also contribute to the generation of pain (49–51). Additionally, nociceptive input from the spinal cord to higher brain areas may amplify receptor activation at peripheral tissues (47).\\nIn the majority of patients, endometriosis first manifests as moderate to severe adolescent dysmenorrhea, which either goes untreated or is managed with NSAIDs or combination oral contraceptives (Figure 1) (52). Many of these patients eventually develop persistent pain between their periods and experience painful intercourse (53). Only a fraction of these patients undergo laparoscopy that may reveal pelvic endometriotic implants and facilitate their surgical removal to provide temporary pain relief (Figure 1) (53, 54). Pain recurs in up to 90% of these patients within 2 years despite adjuvant ovarian suppression; this eventually leads to repeated laparoscopic procedures in an attempt at pain control (Figure 1) (54). This population of patients with relapsed symptoms constitutes the largest and the most challenging group among all patients with endometriosis, since after each treatment attempt the recurrent symptoms progressively become harder to manage (54). Many of these patients eventually resort to surgical removal of the uterus and the ovaries (Figure 1) (31). Even these radical approaches do not provide pain relief in at least 10% of patients (Figure 1) (8, 30, 31).\\nAs adjuvant approaches to surgical resection, suppression of ovulation or menses for pain control is attempted by the use of progestin-containing intrauterine devices, combination oral contraceptives, or gonadotropin-releasing hormone (GnRH) analogs before or after surgical procedures (55). Given that the majority of cases of endometriosis occur as a result of retrograde travel of endometrial tissue to the abdominal cavity, it is my opinion that long-term suppression of ovulatory menses should be the mainstay of prevention or therapy. Surprisingly, this approach as the primary treatment of endometriosis is seldom considered, recommended, or used (2, 56).\\nOvulatory menstruation per se is essential for the development of symptomatic endometriosis (Figure 2) (2, 14, 56). Only menstruating species develop endometriosis (2, 56–59). Women with conditions precluding spontaneous menstruation, such as Müllerian aplasia or Turner syndrome (caused by loss or partial loss of an X chromosome), do not develop endometriosis, although the presence of a vestigial Müllerian remnant has been published in a rare patient with Turner syndrome (60). Endometriosis and anovulatory disorders such as polycystic ovary syndrome rarely occur together (61, 62).\\nRepetitious menstrual episodes that follow ovulation represent the single most impactful and indispensable factor that initiates the process of endometriosis and worsens its existing symptoms, for the following reasons (Figure 2): First, during an ovulatory cycle, sequential exposure to estradiol for 2 weeks followed by added secretion of progesterone for another 2 weeks maximizes endometrial thickness right before this large mass of tissue is shed (Figure 3A) (63–65). Physiologically developed spiral arterioles and endometrial vessels during this period may lead to rapid vasoconstriction and separation at a plane close to the myometrium, producing large quantities of viable endometrial tissue available for transtubal migration onto the pelvic peritoneal surfaces and ovaries (66). Thus, large quantities of viable endometrial tissue are likely shed after a nonpregnant ovulatory cycle, and some of this tissue together with blood reaches into the lower abdominal cavity via the tubes (Figure 3A). Second, massive quantities of estradiol (as high as 130,000 pmol/L in the follicular fluid of approximately 5 mL) and pain-inducing substances such as prostaglandin E2 are released directly onto the pelvic peritoneal surfaces during an ovulation (67, 68). These strikingly high quantities of estradiol probably contribute to the implantation and growth of viable endometrial fragments on peritoneal or ovarian surfaces. Third, an ovarian endometrioma may be formed in a hemorrhagic corpus luteum cyst that recently released an egg (Figure 2) (69). The steroid hormone–rich cyst cavity serves as a growth medium for endometrial tissue to survive and thrive. Fourth, compared with women having regular predictable periods, the prevalence of endometriosis is lower in anovulatory states such as polycystic ovary syndrome, hypothalamic amenorrhea, or breastfeeding (61, 70). Fifth, early menarche followed by cyclic periods is associated with a higher risk of endometriosis (2, 71). Sixth, during infertility treatment of a patient with endometriosis, back-to-back cycles of ovulation induction followed by intrauterine insemination worsen endometriosis symptoms (72), possibly because these simple but low-fecundity treatments must be repeated multiple times, exposing the patient to multiple ovulatory menstrual cycles. On the other hand, more intense forms of ovulation induction used for in vitro fertilization (IVF) usually do not worsen endometriosis as much as intrauterine induction cycles, possibly because although IVF is associated with very high levels of estradiol, it does not trigger multiple and frequent episodes of ovulatory menstruation (72).\\nFigure 3. Not all forms of uterine bleeding increase the risk of endometriosis.\\nOpen in a new tab\\n(A) Menstrual tissue after a nonpregnant ovulatory cycle. In addition to inducing stromal and epithelial differentiation, progesterone during the secretory phase of the menstrual cycle stimulates the coiling of the spiral arteries (64, 65). When progesterone production is interrupted in a nonpregnant cycle, the spiral arteries of the basalis and functionalis layers go through rapid and intense vasoconstriction and coagulation, leading to the separation of the thick functionalis layer in its entirety. This leads to the presence of large numbers of viable stromal, epithelial, and endothelial cells along with blood in the menstrual tissue. Some of these tissue fragments travel through the tubes into the pelvic cavity. During this process, mutated epithelial cells and epigenetically abnormal populations of stromal cells are deposited on the pelvic peritoneum or the ovaries. (B) Menstrual tissue associated with withdrawal bleeding in a patient on an oral contraceptive (OC). Because the endometrial tissue that is chronically exposed to an estrogen-progestin combination or progestin-only OC does not build a thick functionalis layer, temporary cessation of OC gives rise to menstrual tissue composed primarily of blood and containing very few stromal or epithelial cells (65, 76). This is also the case for the breakthrough bleeding the patient experiences while they are on hormone-containing OC pills used in a continuous manner.\\nAdditionally, time-honored clinical data support the postulate that repetitious menstrual episodes that follow ovulation contribute to the development or worsening of endometriosis symptoms. For example, the conventional use of cyclic combination oral contraceptive pills involving withdrawal bleeding is associated with cyclic uterine bleeding that is much lighter than spontaneous ovulatory menses (Figure 3B) (65, 73–76). This may be because a more superficial and thinner portion of endometrial tissue is present in the menstrual tissue (Figure 3B). Oral contraceptive use, which disrupts or suppresses ovulatory menses, reduces endometriosis symptoms such as dysmenorrhea or chronic pelvic pain (74). Breakthrough bleeding associated with other anovulatory conditions such as polycystic ovary syndrome or postmenopausal hormone replacement does not increase the risk of endometriosis (61, 70). It follows that suppression of ovulation should be pursued as a first-line approach for preventing the development of endometriosis or alleviating its symptoms (77).\\nThe fundamental mechanism of endometriosis is relatively straightforward. Following each nonpregnant ovulatory cycle, menstruation ensues. This routinely involves retrograde travel of blood and floating fragments of endometrial tissue through the uterine tubes onto the ovaries and pelvic peritoneal surfaces in the majority of premenopausal women (78). Although most ovulatory women are presumed to have retrograde menstruation, only a fraction of this population seems to develop endometriosis and its symptoms (1). Quantitatively, obstructive Müllerian abnormalities such as imperforate hymen or transverse vaginal septum strikingly increase retrograde menstrual tissue and very often give rise to endometriosis (79). Premenopausal women with heavy menstrual bleeding also have higher risk of developing endometriosis (80, 81). Qualitatively, endometrial tissue with cancer-driver mutations and epigenetic defects leading to abnormal gene expression is more likely to survive in ectopic locations and give rise to endometriosis (22). All these observations with mechanistic implications, however, likely require the presence of numerous back-to-back ovulatory cycles for endometrial fragments to repeatedly implant and accumulate in an ovarian corpus luteum cyst (ovarian endometrioma), peritoneal surfaces proximal to tubal ostia (peritoneal endometriosis), or the peritoneal pouch between the rectum and the vagina (deep-infiltrating endometriosis) (Figures 2 and 3) (22, 69).\\nIn addition to the retrograde menstruation theory based on ample scientific evidence from human samples, there has been considerable controversy regarding whether other biological processes contribute to endometriosis (15). Based on a report that human donor-derived endometrial cells were detected in eutopic endometrial biopsy samples from human bone marrow recipients, a mouse model was developed to demonstrate that circulating stem cells may contribute to endometriosis (82, 83). Additionally, the peritoneal metaplasia or Müllerian remnant hypothesis was primarily based on some 11 published case reports of male patients, whereby small uterine tissue near the bladder or inguinal canal was discovered because of pain symptoms, frequently stimulated by exogenous estrogen administration to treat prostate cancer (84, 85). The deficiency of or resistance to anti-Müllerian hormone (AMH) during embryonic development most likely caused the persistence of intact uterine tissue, including the myometrium and endometrium, in the pelvis of each of these 46,XY individuals (84, 85). In my opinion, these rare cases of persistent Müllerian tissue in men are not relevant to the hypothesis that peritoneal tissue in women may differentiate to endometriosis via metaplasia or that endometriosis may arise from Müllerian tissue remnants (84, 85).',\n",
       "  'Methods': '',\n",
       "  'Results': 'A pathologist typically makes a tissue diagnosis of endometriosis by identifying two of the following three types of cells: (a) endometrial stromal cells, (b) endometrial epithelial cells, and (c) hemosiderin-containing macrophages indicative of chronic bleeding (34). Stromal cells constitute the bulk of endometriotic lesions. Epithelial cells may or may not be present on the surface of an endometriotic lesion and less frequently line the invaginating or glandular portions of these lesions. The origin of the macrophages containing blood pigment in endometriotic tissue is not well understood.\\nEpithelial cells of endometriosis.\\nEpithelial cells of peritoneal, deep-infiltrating, or ovarian endometriotic lesions share identical cancer-driver mutations with the eutopic endometrial epithelial cells of the same patients (22). At least four different laboratories mapped identical DNA variants or mutations in eutopic endometrium to matched tissues of pelvic endometriosis within the same individuals (21–25, 46, 86). This provided irrefutable evidence that all three forms of pelvic endometriotic lesions originated from the eutopically located endometrium (45).\\nInterestingly, starting from the teenage years, the endometrium of an average woman progressively accumulates cancer-driver mutations in oligoclones of epithelial cells located in the deepest portions of invaginating mucosal crypts, also called glands (87). Hundreds of cancer-driver genes may be mutated in various endometrial epithelial cell populations of an adult woman (21–25, 46, 86, 87). Given the frequency of this mutational process in the majority of women, it is considered normal (22). What is not well understood is how these mutations contribute to the disease process in a subset of women (44).\\nIn eutopic endometrial tissue, although the most commonly mutated gene is PIK3CA, other gain-of-function mutations such as those that affect KRAS are also frequently detected (21–25, 46, 86, 87). Although mutations sporadically affect small numbers of cells distributed across the eutopic endometrium, the majority of endometriotic epithelial cells carry these mutations in the same individual (21–25, 46, 86, 87). In particular, epithelial cells with KRAS mutations are enriched in endometriotic lesions, suggesting that autonomously activated KRAS may confer survival advantages for endometriotic tissue (21, 22, 24, 25, 46, 86). Intriguingly, a number of these mutations in endometriosis also serve as driver mutations for epithelial ovarian cancer (22, 24, 25, 46, 86). Although clear cell and endometrioid ovarian cancers are more strongly associated with an ovarian endometrioma, history of high rates of ovulatory activity is a well-established risk factor for all histological types of epithelial ovarian cancer (44, 88, 89). These data collectively suggest that ovulation followed by retrograde menstruation leads to the seeding of endometrial tissue fragments, including mutated epithelial cells, in an involuting corpus luteum cyst or other inclusion cysts, a small portion of which may eventually evolve into a benign or malignant ovarian neoplasm over decades (44, 69).\\nAdditionally, considerable differences were found between eutopic endometrial and endometriotic epithelial cells with respect to mRNA or protein expression. One of these proteins, 17β-hydroxysteroid dehydrogenase type 2 (HSD17B2), which physiologically catalyzes the enzymatic conversion of estradiol to biologically less potent estrone in eutopic endometrial epithelium during the implantation window, was severely deficient in endometriotic epithelial cells (90–92). The underlying mechanism involves a paracrine relationship between endometrial stromal and epithelial cells (93). It turned out that progesterone via its receptor (PGR) physiologically induces retinol uptake and metabolism in decidualized endometrial stromal cells (94). Consequently, increased levels of retinoic acid, in turn, stimulate HSD17B2 expression and activity in endometrial epithelial cells (93, 95, 96). This paracrine induction of epithelial HSD17B2 expression does not occur in endometriotic tissue because of decreased stromal PGR levels (93–97). Another implantation window marker, glycodelin (PAEP), is expressed at strikingly lower levels in endometriotic epithelial cells compared with the eutopic endometrial tissue (98). PAEP mRNA and protein are significantly lower in eutopic endometrium of women with versus without endometriosis (99). PAEP expression in endometrial tissue was reported to be significantly lower in implantation failure during IVF (100).\\nStromal cells of endometriosis.\\nFollowing the seminal 1994 publication by Ryan et al., which described how to maintain primary endometriotic stromal cells isolated from ovarian endometriomas in culture, a number of groups conducted mechanistic studies on these cells and compared them with eutopic endometrial stromal cells in culture (101, 102). These studies started by demonstrating differential mRNA or protein expression between eutopic and endometriotic stromal cells. They collectively revealed that endometriotic stromal cells bear widespread epigenetic aberrations involving pathological overexpression of transcription factors and nuclear receptors, which transform these cells from a normal endometrial stromal cell to an endometriotic stromal cell (34). Most notably, ectopic expression of NR5A1 (SF-1) and GATA6 together was sufficient to transform a normal endometrial stromal cell to an endometriotic stromal cell that can convert cholesterol to estradiol via activation of the steroidogenic cascade, including aromatase (103). These differences in mRNA or protein expression were also demonstrated in the eutopic endometrium of endometriosis patients, although the difference between eutopic endometrial tissues of women with versus without endometriosis was more subtle compared with expression profiling between endometrium and endometriosis (35, 99, 104). For example, expression of the decidualization markers prolactin and insulin-like growth factor–binding protein-1 (IGFBP1) is reduced in endometriotic stromal cells and in the stromal cells of the endometrium of women with endometriosis compared with disease-free endometrial stromal cells (105).\\nEndometriosis originates from eutopic endometrium of endometriosis patients.\\nAnalyses of epithelial and stromal cells of eutopic endometrium of patients with or without endometriosis and endometriotic lesions with respect to analyses for mutations, DNA methylation, histone modification, and gene expression indicate the following (34): Eutopic endometrium of endometriosis patients harbors oligoclones or localized populations of stromal cells with aberrant mRNA and protein expression and epithelial cells with mutations (22, 25, 104, 106). The great majority of stromal cells in endometriotic peritoneal implants, deep-infiltrating endometriosis, or ovarian endometriomas contain similar aberrant levels of mRNA/protein, most notably NR5A1, GATA6, and aromatase, in part due to DNA methylation differences, which control gene expression (106–108). Likewise, the great majority of epithelial cells in all forms of endometriotic lesions bear an identical driver mutation found in oligoclones of eutopic endometrial cells of the same individual (21–25, 46, 86). The most logical explanation is that if these tissue fragments with molecular aberrations travel to the pelvic cavity via retrograde menstruation, they have a higher potential to implant, survive, and grow in response to estrogen.\\nSomatic mutations\\nEutopic endometrial tissue of all women accumulates somatic mutations primarily in epithelial cells, starting from the teenage years (87). The mutation burden increases with advancing age (87). In particular, one laboratory reported significant differences between the eutopic endometrial tissues of women with and without adenomyosis or endometriosis with respect to the types of mutations (25). They compared gene mutations in histologically normal-appearing endometrial tissues of patients with adenomyosis, patients with neither adenomyosis nor endometriosis, and patients with endometriosis but without co-occurring adenomyosis (25). In general, the most commonly mutated gene in all three groups was PIK3CA (25). Mutation or variant allele frequency in eutopic endometrium was similar with respect to the PIK3CA or PPP2R1A genes in all three groups. KRAS (G12/G13) mutation or variant allele frequency was significantly higher in endometrial tissues of adenomyosis patients compared with disease-free patients (25). In endometrial tissues of patients with endometriosis but without co-occurring adenomyosis, the KRAS mutation or variant allele frequency was higher, but the difference did not reach significance (25). Although other studies did not make this direct comparison, their data were also suggestive that the eutopic endometrium of endometriosis patients contained a higher frequency of KRAS mutations compared with those of endometriosis-free patients (22, 86). This set of observations raises the question of whether the individuals who develop endometriosis are more likely to have endometrial tissue fragments with KRAS mutations in their retrograde menstrual tissue material.\\nOn the other hand, the mutation landscapes of endometriotic or adenomyotic tissues are clearly different from those of eutopic endometrium of the same patient (22, 25, 86). For example, adenomyotic tissue contains almost exclusively KRAS mutations (25). Similarly, the majority of epithelial mutations in endometriotic tissue affect the KRAS gene (22, 25). Moreover, identical epithelial mutations that affect both eutopic endometrium and endometriosis have been mapped in the matched tissues of each individual patient (22, 25). These observations collectively suggest that endometriosis or adenomyosis originates from eutopic endometrial tissue clones with epithelial mutations that enable these tissues to survive outside of the uterine cavity (21–25, 46, 86).\\nEpigenetic defects\\nDNA methylation.\\nThe initial epigenetic studies on endometriosis focused on DNA methylation (108–110). They demonstrated differences between the stromal cells of eutopic endometrium and ovarian endometriomas with respect to the CpG methylation of the promoters or coding regions of distinct transcription factors or nuclear receptors (111). In general, genes such as NR5A1, ESR2 (encoding estrogen receptor-β), and GATA6 are heavily methylated and suppressed in endometrial stromal cells, whereas they are less methylated and are expressed in endometriotic stromal cells (108–110). In contrast, GATA2 is methylated and suppressed in endometriosis, whereas it is highly expressed and serves as a critical mediator in progesterone-induced differentiation or decidualization of eutopic endometrial stromal cells (108, 112). In parallel, the failure to downregulate DNA methyltransferase-3B expression in response to physiological differentiating stimuli in endometriotic cells compared with healthy endometrial stroma may contribute to a defective epigenetic fingerprint and gene expression pattern in endometriosis (113). Mechanistically, it is not well understood how individual genes such as NR5A1, ESR2, GATA6, or GATA2 are differentially methylated in each cell type (108, 112).\\nOther studies also demonstrated that aberrations in global DNA methylation play a key role in the pathophysiology of endometriosis (114–116). One report suggested that estrogen and progesterone regulate DNA methylome in endometrial cells, and this process is perturbed in endometriosis (115). A recent report suggested that menstrual cycle phase was a major source of DNA methylation of the eutopic endometrium (116). This differential methylation study of endometrium from disease-free or endometriosis patients highlighted candidate genes contributing to disease risk (116).\\nHistone modification.\\nA limited number of studies have been published regarding the histone modification patterns in endometriosis (117). One study assessed acetylation/methylation levels of histones in whole tissues of endometriosis, eutopic endometrium from endometriosis patients, and endometrium from disease-free individuals (117). Globally, the endometriotic lesions were hypoacetylated at H3 but not H4 compared with disease-free eutopic endometrium (117). Lysine (K)-specific acetylation/methylation levels of histones showed that endometriosis tissue contained significantly lower levels of acetylated H3K9 and H4K16 in comparison with eutopic endometrium from patients or disease-free subjects (117). Both endometriotic and eutopic endometrial tissues from patients were hypermethylated at H3K4, H3K9, and H3K27 compared with normal endometrium of disease-free women. ChIP analysis of endometriotic lesions showed hypoacetylation of H3/H4 around promoter regions of genes previously demonstrated to be downregulated in endometriosis, such as HOXA10, ESR1, CDH1, and CDKN1A, compared with disease-free endometrium (117). On the other hand, the NR5A1 promoter region was enriched with acetylated H3 and H4 in endometriosis versus disease-free tissues, correlating with its reported overexpression in endometriosis (117).\\nEpigenetic defects affecting nuclear receptors in endometriotic stromal cells lead to aberrant expression of critical genes. For example, NR5A1 expression in endometriotic stromal cells activates most of the genes in the steroidogenic cascade, including aromatase, leading to the conversion of cholesterol to estradiol, whereas this is not observed in the eutopic endometrium of disease-free individuals (Figure 4) (102, 103, 107, 118, 119). Excessive ESR2 expression leads to suppression of ESR1 (ERα) (120). Ectopic overexpression of NR5A1 and GATA6 in a normal endometrial stromal cell is sufficient to confer an endometriotic phenotype with respect to steroidogenesis and failure to differentiate properly (Figure 4) (103, 108). Thus, epigenetic stromal defects in endometriosis profoundly influence the biology of endometriosis.\\nFigure 4. The epigenetic defects that cause gene expression abnormalities in endometriotic cells.\\nOpen in a new tab\\nThe transcription factors NR5A1, GATA6, and ESR2 are pathologically hypomethylated and thus upregulated in the stromal cells of endometriosis, whereas PGR is downregulated in these cells, compared with normal endometrial stromal cells. Consequently, PGE2 stimulates the steroidogenic enzymes, including aromatase, to increase the production of estradiol, which stimulates COX-2 to elevate PGE2 production. PGE2 itself induces COX-2 expression through a short feedback loop. This vicious cycle causes continuous local production of estradiol and PGE2 in endometriotic tissue. Low PGR levels in endometriotic stromal cells lead to progesterone resistance and decreased uptake of retinol, leading to decreased retinoic acid production. The enzyme HSD17B2, which is physiologically regulated by retinoic acid in normal endometrium, is downregulated in endometriotic epithelial cells. Deficient HSD17B2 leads to decreased metabolism of estradiol, further contributing to increased estradiol levels in endometriotic tissue.\\nProgesterone resistance\\nProgesterone resistance in endometriotic tissue or eutopic endometrium has been described by numerous laboratories and clinicians in different contexts (35, 91–93, 97, 104, 121–124). It was originally suggested upon the observation that the enzyme HSD17B2, which is involved in inactivation of estradiol, is present in endometrial epithelium but absent in matched endometriotic tissue during the luteal phase, although both tissues are simultaneously exposed to circulating progesterone (Figure 4) (91). Follow-up studies demonstrated that progesterone physiologically acted directly on endometrial stromal cells to alter retinoid uptake and biosynthesis of retinoic acid, which in turn induced epithelial HSD17B2 expression in a paracrine fashion (Figure 4) (93, 96, 125). Pathologically, deficient PGR expression in endometriotic stromal cells disrupted this physiological mechanism and thus the capacity of endometrial epithelial cells to inactivate estradiol during the secretory phase (Figure 4) (93, 94, 96, 125).\\nA number of other laboratories also reported deficient expression of genes regulated by progesterone in endometriotic tissue or eutopic endometrium of endometriosis patients (99, 104, 126, 127). These included IGFBP1, PAEP, HOXA10, HOXA11, CYP26A1, FOXO1A, FKBP4, and BCL2 (99, 104, 126, 127). HOXA10 hypermethylation was demonstrated as a mechanism for decreased HOXA10 expression in the endometrium of women with endometriosis (127). Thus, inappropriate methylation of progesterone-regulated genes may be a potential mechanism for progesterone resistance. The underlying mechanisms for the other suppressed genes such as epithelial PAEP in tissues of women with endometriosis, however, have remained unclear.\\nProstaglandin production and action\\nProstaglandin E2 (PGE2), a product of the COX/PGES cascade, is the most intensively studied eicosanoid that contributes to inflammatory pain (47). PGE2 also plays a central role in the pathophysiology of endometriosis (Figure 4) (48). In endometrium and inflamed tissues such as the peritoneum adjacent to endometriosis, upregulated COX-2 and perhaps also COX-1 produce prostaglandin H2 (PGH2) from arachidonic acid released by phospholipase A2, and then upregulated PGE synthase subtypes produce PGE2 from PGH2 (Figure 4) (128, 129). PGE2 interacts with the G protein–coupled EP2 and EP4 receptors in the nerve endings or nociceptors, leading to activation of PKA and also EPAC (exchange protein activated by cAMP) to produce PGE2-mediated pain (47, 48). In addition to the PGE2 receptors at the nerve endings, spinal EP2 receptors via PKA activation also contribute to the amplification of PGE2-evoked nociceptive input from the spinal cord to higher brain areas where pain is actually perceived (47).\\nPeritoneal fluid concentrations of PGE2 are higher in women with endometriosis, and, in addition to evoking pain, elevated PGE2 plays key roles in survival and growth of endometriotic lesions (130, 131). Inhibition of PGE2 synthesis suppresses endometriosis and chronic pelvic pain in women and decreases growth of experimental endometriotic lesions in animal models (132, 133). PGE2 in endometriotic stromal cells induces the expression of aromatase and other steroidogenic enzymes via the EP2/PKA/cAMP pathway and generates estradiol (Figure 4) (102, 103, 119, 134). Estradiol, in turn, stimulates COX-2 expression via ESR2 to enhance PGE2 production in uterine cells (Figure 4) (135). Moreover, PGE2, via a short feedback loop, induces COX-2 expression via the activation of NF-κB in endometrial stromal cells, leading to a vicious cycle that continuously produces estradiol and PGE2 in endometriotic tissue (Figure 4) (102, 129).\\nSelective inhibition of EP2 and EP4 decreases survival and growth of human endometriotic cells via altering the expression of a number of gene products, including matrix metalloproteinases (MMPs) and tissue inhibitors of metalloproteinases (TIMPs) (136). Selective inhibition of EP2/EP4 in an experimental animal model also decreased angiogenesis and innervation of endometriotic lesions, suppressed proinflammatory state of dorsal root ganglia neurons, and significantly reduced proinflammatory, estrogen-dominant, and progesterone-resistant molecular environment in endometriotic tissue (137).\\nNSAIDs, inhibitors of COX-1 and/or COX-2, suppress endometriosis-associated inflammatory pain by reducing generation of eicosanoids, mainly PGE2, while they exhibit gastrointestinal, renal, and cardiovascular toxicities (132, 138). Selective inhibitors of microsomal PGE synthase-1 and subtype-selective antagonists of PGE2 receptors, particularly EP2 and EP4, may be useful as future treatments of endometriosis pain with minimized side effects (137). Thus, PGE2 has a substantial impact on endometriosis-induced pain, and pharmacological intervention in PGE2 signaling at pelvic peritoneal tissues or central nervous system may serve as a future therapeutic strategy for the treatment of endometriosis-induced pain (47).',\n",
       "  'Discussion': '',\n",
       "  'Conclusion': 'Pelvic endometriotic tissue evidently originates from shed endometrial fragments that travel in a retrograde manner primarily through the tubes (19, 20, 45). In patients with endometriosis, relatively larger populations of stromal and epithelial cells in the eutopic endometrium display widespread epigenetic abnormalities and mutations that favor increased estradiol and PGE2 production and action, progesterone resistance, autonomous activation of KRAS and its downstream survival pathways, and heightened cytokine production and immune cell recruitment (Figure 4). Although these molecular abnormalities have been consistently observed by multiple laboratories, the upstream reproductive process that is indispensable for the establishment of endometriosis and its symptoms remains as repetitious episodes of ovulation followed by retrograde menstruation. That is why the risk of developing endometriosis is extremely low in the absence of ovulatory menses. Ovulation is associated with relatively larger quantities of shed endometrial functionalis tissue that separates from the basalis and travels to the pelvic cavity, increases blood estradiol levels, and directly releases ovulatory fluid with massive amounts of estradiol onto the peritoneal surfaces. The resulting corpus luteum cyst potentially serves as the foundation for a new endometrioma formation. Therefore, the process of ovulation per se strikingly increases the risk for endometriosis in multiple ways.\\nEliminating or diminishing ovulation or ovulatory menses in the long term should be the first-line strategy for treating pain caused by endometriosis or potentially preventing endometriosis. Surprisingly, this has not been explicitly addressed by clinicians or researchers. Traditionally, progestins, combination oral contraceptives, or injectable gonadotropin-releasing hormone (GnRH) agonists have been used to suppress ovulatory activity regulated by follicle-stimulating hormone (FSH) and luteinizing hormone (LH) (139–142). More recently, oral GnRH antagonists with estrogen or progestin addback have been introduced (143, 144). The key side effects of GnRH suppression have been hot flashes and decreased bone mineralization (145). New pharmaceutical approaches to suppress ovulation long-term with fewer side effects are urgently needed for prevention and treatment of endometriosis. I speculate that exploiting the inhibin, activin, and follistatin-related pathways or blocking the gonadotropin receptors may produce beneficial future pharmaceutical targets.',\n",
       "  'URL': 'https://doi.org/10.1172/JCI188787',\n",
       "  'Journal': 'The Journal of Clinical Investigation'},\n",
       " {'PaperTitle': 'Outcomes of Ultimpro Testing Implementation for Patients With Unexplained Recurrent Implantation Failure: A Single‐Center Retrospective Cohort Study',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Genevieve Genest, Alyssa Hochberg, Martine Boivin, Alexie Ferreira, Zhiyang Liu, Coralie Beauchamp, Marc Beltempo, Michael H Dahan, Elias M Dahdouh, Ezgi Demirtas, Robert Hemmings, Wael Jamal, Isaac‐Jacques Kadoch, Louise Lapensée, Neal Mahutte, Simon Phillips, Shauna Reinblatt, Camille Sylvestre, William Buckett, Bruce D Mazer, Ciriaco A Piccirillo',\n",
       "  'Email': '',\n",
       "  'Abstract': 'Problem\\nA proportion of patients with unexplained recurrent implantation failure (uRIF) may have endometrial immune dysregulation amenable to treatment. Ultimpro (Matrice lab Innove, Paris, France) is an endometrial immunophenotype test which identifies the absence or presence and type of immune dysregulation, proposing personalized protocol adjustments for a future embryo transfer (ET).\\nMethod of Study\\n2.5‐year single‐center retrospective cohort analysis of 121 patients who underwent Ultimpro testing as a pilot study (January 2021—May 2023). Group 1 included 93 patients with uRIF, and Group 2 included 28 patients without RIF. Outcomes from Group 1 were compared to Group 3, a control cohort of 94 uRIF patients, outcomes for Group 2 were compared to Canadian average ongoing pregnancy rates (OPR) per ET. Primary outcome was pregnancy >32\\xa0weeks or live birth. Statistical analyses included the Student\\nt\\n‐test or Mann–Whitney\\nU\\ntest for continuous variables and the Fisher exact test for categorical data.\\nResults\\nIn Group 1, 46/93 (49.5%) patients had a successful pregnancy compared to control (31/94 [33%],\\np\\n=\\xa00.026); when stratifying for RIF severity, patients with ≥\\xa05 failed blastocyst transfers benefitted most from Ultimpro (21/38 [55.3%] vs. control 5/25 (20.0%),\\np\\n=\\xa00.0084). In Group 2, rates of OPR per ET with Ultimpro recommendations did not differ from outcomes for non‐RIF patients (9/28 [32.1%]; 95% CI: 0.15–0.52\\xa0vs. 46.2%).\\nConclusions\\nIn patients with uRIF, our preliminary findings suggest that application of Ultimpro recommendations was associated with a higher rate of successful outcomes; this was not the case for patients without RIF. These findings provide the rationale for an independent randomized controlled trial evaluating the large‐scale use of Ultimpro in patients with uRIF.',\n",
       "  'Introduction': \"Infertility is a significant health problem, affecting 1/6 couples globally (2023 WHO statistics); up to 10% of reproductive age women in North America seek medical assistance to conceive [\\n1\\n]. In vitro fertilization has revolutionized the treatment of infertility; however, success rates remain low with 20%–30% live birth rate (LBR) per embryo transfer (ET) [\\n2\\n,\\n3\\n]. Up to 10% of couples resorting to IVF will have recurrent implantation failure (RIF), generally defined as the absence of clinical pregnancy after 3 high quality blastocyst transfers [\\n4\\n]. Over 50% of such patients will be diagnosed with unexplained RIF (uRIF) after exclusion of anatomical, hormonal and infectious causes, further compounding the psychological and financial burdens these couples face [\\n5\\n]. Although embryo aneuploidy likely explains a high proportion of uRIF cases [\\n6\\n], additional factors must be considered especially in young couples with high quality embryos and normal parental karyotypes.\\nIndeed, immune mediated RIF (IM‐RIF) is increasingly recognized as a host‐derived factor that can contribute to reproductive failure. An appropriate endometrial immune response is necessary for organized decidualization, endometrial receptivity, and efficient embryo implantation [\\n7\\n]. Multiple immune cell types are involved in coordinating this response, with the natural killer cell (NK cell) being the predominant leukocyte in the endometrium during the window of implantation [\\n8\\n] (Figure\\nS1\\n). Uterine NK cells (uNK cells) exhibit significant differences compared to their peripheral blood counterparts (pNK cells). They are CD56\\nbright\\nCD16\\ndim\\n, express a low cytotoxic profile and secrete different cytokines and growth factors [\\n9\\n]. During the luteal phase, uNK cells secrete pro‐angiogenic factors and matrix metalloproteases, participating in spiral artery remodeling and loosening the endometrial extracellular matrix to facilitate embryo implantation [\\n8\\n,\\n10\\n]. The uNK cells are also thought to be involved in embryo recognition, sensing embryo quality [\\n11\\n] and guiding trophoblast invasion into the endometrium [\\n9\\n,\\n12\\n]. Therefore, it is possible that uNK cell dysfunction (both overactive and insufficient NK cell responses) during implantation contributes to reproductive failure [\\n13\\n]. Although peripheral blood NK cells do not reflect uNK cell populations [\\n14\\n,\\n15\\n], nor correlate with reproductive outcomes [\\n16\\n,\\n17\\n]; endometrial uNK cell evaluation may offer more insight into implantation failure [\\n18\\n].\\nThe Ultimpro test (Matrice Lab Innove, Paris, France) offers targeted uNK cell testing to identify immune dysregulation contributing to reproductive failure. Briefly, by studying differential transcriptional [\\n19\\n] and protein expression profiles [\\n20\\n,\\n21\\n,\\n22\\n,\\n23\\n] between patients with unexplained RIF and fertile controls, the company has developed biomarkers to assess uNK function and behavior. Endometrial biopsies are performed in the mid‐luteal phase (8–11\\xa0days after LH surge in monitored natural cycles and on Days 6–8 of progesterone in HRT cycles). Immunohistochemistry is performed on formalin preserved tissue sections using an anti‐CD56 antibody. The rest of the sample is used for RNA extraction and an RT‐PCR panel is performed for the markers IL‐15, IL‐18, Fn‐14, and TWEAK. CD56 is a uNK cell surface marker and is used to quantitate the amount of uNK cells present in each biopsy sample. The cytokine IL‐15 is involved in NK cell mobilization, proliferation, and maturation, whereas IL‐18 is a pro‐angiogenic cytokine involved in regulating uNK cell phenotype, endometrial vascular remodeling and trophoblast invasion [\\n24\\n]. When in excess, IL‐15 and IL‐18 can stimulate NK cell cytotoxicity and contribute to disorganized spiral artery remodeling. TNF Weak inducer of Apoptosis (TWEAK) and its receptor, fibroblast growth factor‐inducible molecule 14 (Fn‐14), have been proposed to suppress NK cell cytotoxicity and regulate IL‐15 and ‐18 when in excess [\\n25\\n]. The TWEAK/Fn14 axis is also proposed to be involved in tissue regeneration and repair; excessive TWEAK/Fn14 can contribute to tissue damage and aberrant endometrial remodeling [\\n26\\n].\\nHigh IL‐18/TWEAK and/or IL‐15/Fn14 mRNA levels and/or elevated CD56 staining characterize the “over‐active” endometrial immune profile where the endometrial microenvironment is pro‐inflammatory, uNK cell recruitment is excessive and vascular remodeling is disorganized. On the contrary, low IL‐18/TWEAK and/or IL‐15/Fn14 mRNA and/or CD56 staining characterize the “underactive” endometrial immune profile, where uNK recruitment is deficient and the endometrial micro‐environment exhibits insufficient inflammatory stimulus for normal vascular remodeling and implantation to occur. A “mixed” endometrial immune profile denotes a combination of over‐immune or underactive phenotypes whereby NK cells may be immature or insufficiently recruited, but the endometrium has inflammatory characteristics which may induce a deleterious NK cell phenotype [\\n27\\n]. Depending on the patient's profile, treatment recommendations to normalize immune profile for ET are made by the company and applied for subsequent ETs (Figure\\n1\\n). Of note, CD56 levels, IL‐18/TWEAK and IL‐15/Fn14 ratios are compared to proprietary norms established by the MatriceLAB Innove company. The readout addressed to physicians provides a numerical value for these markers and whether they are higher, lower or within norm. The interpretation of individual patient results is performed by the company.\\nA retrospective cohort study [\\n28\\n], a prospective cohort study [\\n29\\n] and a randomized controlled trial [\\n30\\n] led by the company suggest a possible benefit of Ultimpro for patients with RIF. However, there has been no independent validation of their CD56 or cytokine ratios, limited evidence of inter‐laboratory reproducibility [\\n31\\n] and thus far no independent studies assessing clinical performance of the test. Although the ability to diagnose and correct immune dysregulation contributing to implantation failure would likely improve the management of uRIF patients, Ultimpro testing is expensive, invasive, and data amassed so far is insufficient to recommend widespread clinical implementation. An independent randomized controlled trial (RCT) is needed to assess the clinical utility of the test in uRIF patients but first, patient populations most likely to benefit from Ultimpro testing must be identified.\\nIn this study, we present a retrospective review of our clinic's 2.5‐year outcomes, comparing success rates of Ultimpro to “expected outcomes” from a historical cohort of patients with uRIF not having undergone Ultimpro testing and not having received immunomodulatory treatments.\",\n",
       "  'Methods': \"2. Methods\\n2.1. Study Design and Clinical Setting\\nThis is a single center retrospective cohort study of patients followed at the McGill University Health Center Reproductive Immunology clinic (MRIC). The MRIC is the reference center in the province of Quebec of the evaluation of patients with suspected immune‐mediated reproductive failure. We aimed to determine if Ultimpro improved ongoing pregnancy rates (OPR) in patients with unexplained RIF. This study was approved by the MUHC research ethics board (MUHC REB #2025‐10617). All methods were performed in accordance with relevant guidelines and regulations. This manuscript was written according to Strengthening the Reporting of Observational studies in Epidemiology (STROBE) guidelines.\\n2.2. Study Population\\nAll patients undergoing Ultimpro testing and ET between January 2021 and May 2023 were included. Patients were divided into two groups. Group 1 included patients with RIF fitting our inclusion criteria (Table\\n1\\n, Figure\\n2\\n). Criteria for Ultimpro testing are: ≥\\xa03 consecutive high‐quality (≥\\xa03BB Gardner's criteria) previous blastocyst transfer failures, age\\xa0≤\\xa042 (<45 if using oocyte donation), body mass index (BMI)\\xa0<\\xa035, non‐smokers, endometrium\\xa0≥\\xa07\\xa0mm at time of transfer, absence of hydrosalpinges, normal uterine cavity, absence or correction of endocrinopathy (medically controlled hypo/hyperthyroidism, hyper‐prolactinemia, diabetes) and absence of HIV, hepatitis B/C and syphilis. All patients in Group 1 were required to have at least one high quality blastocyst (Gardner ≥\\xa03 BB) remaining for transfer. A significant number of patients without RIF or not fitting our Ultimpro criteria requested testing against medical advice. These patients were included in the study, but as a separate group (Group 2) (Figure\\n2\\n). The first ET treated with Ultimpro recommendations was used as the index ET. Patients for whom the index embryo lost significant quality upon thaw (Gardner grade C or lower for the trophectoderm or the inner cell mass) were excluded. Only patients for whom the index ET is known were included in the analysis. In patients who conceived, only those pregnant ≥\\xa032\\xa0weeks gestation at the time of analysis were included.\\n2.3. Ultimpro Protocol\\nPatients referred to the MRIC for uRIF were presented with the option of performing Ultimpro testing. Patients accepting Ultimpro testing were provided with a kit containing the collection tube. The referring fertility specialist then performed endometrial biopsy on a hormone replacement therapy cycle (on Days 6–8 of progesterone) or on a natural cycle (LH + 8–11 days) without glucocorticoids or acetylsalicylic acid. Samples were shipped to the Matrice Lab Innove laboratory according to manufacturer instructions. Ultimpro treatment recommendations were followed for each patient for their index ET, only patients who were compliant to treatment were included in our study (Figure\\n1\\n). Although Ultimpro suggests repeat biopsy with treatment application, repeat biopsies were not performed because of time and financial constraints to the patient.\\n2.3.1. Control Group\\nWe used an unmatched RIF control group (Group 3) from a previous study [\\n32\\n] as a comparator for Group 1 outcomes. Briefly, 1061 patient files were screened from the MRIC (170 patients) and the MUHC Reproductive Center (891 patients) and included patients undergoing ET between January 2020 and December 2021. We included 94 controls fitting or inclusion criteria (≥\\xa03 previous high quality ET failures and respecting our inclusion criteria (Table\\n1\\n, Figure\\n2\\n)). The last blastocyst transfer on record during the study period was included as the index ET. BMI was not available for all control patients, but weight was. We excluded patients with recoded weight over 95\\xa0kg (BMI 34.5 for an average 165\\xa0cm woman). Patients were also excluded if they had previously been treated with immunomodulatory treatment for any of their ETs (including glucocorticoids, intravenous immunoglobulin, intralipids, or calcineurin inhibitors). Importantly, there was no overlap between Group 3 and the Ultimpro cohort (Groups 1 and 2). All data were collected via retrospective chart review using institutional databases. Only patients for whom the index ET is known were included in the analysis. In patients who conceived, only those whose outcomes (live birth or miscarriage) are known were included in the analysis.\\n2.4. Variable Definition\\nRecurrent implantation failure was defined as ≥\\xa03 previous consecutive high quality blastocyst transfer failures (Gardner grade ≥\\xa03BB). Patients with primary RIF, secondary RIF, and RIF with donor oocytes were included in Groups 1 and 2 (Table\\n1\\n). Primary outcomes included successful pregnancy, ET failure or miscarriage. A biochemical pregnancy was defined as a positive quantitative hCG blood test without presence of an intra‐uterine sac. Clinical pregnancy was defined as the presence of an intra‐uterine gestational sac. Successful pregnancy was defined as either a ≥\\xa032\\xa0weeks ongoing pregnancy or live birth occurring ≥\\xa024 gestational weeks. Unsuccessful outcomes included miscarriage (<24 weeks gestation after serologic confirmation of pregnancy) or ET failure (hCG\\xa0<\\xa05, 10\\xa0days after ET). Secondary outcomes included maternal side effects from medication, as well as maternal pregnancy and neonatal outcomes.\\n2.5. Statistical Analysis\\nPatients were stratified into Group 1 (uRIF patients fitting our RIF inclusion criteria for Ultimpro testing, (Table\\n1\\n)) and Group 2 (patients not fitting our criteria) to preserve group homogeny (Figure\\n2\\n). Index ET outcomes for patients in Group 1 were compared those of Group 3. ET outcomes for patients in Group 2 were compared to expected outcomes of non‐RIF women (stratified by maternal age at oocyte retrieval) from the national Canadian ART database (CARTR 2023) and from a representative Quebec Fertility clinic (Ovo Clinic).\\nCategorical data were presented as percentage, continuous variables were presented as mean and minimum or maximum values. Shapiro–Wilk test was used to assess normal distribution of the quantitative parameters. The Student\\nt\\n‐test or Mann–Whitney\\nU\\ntest was used for continuous variables and the Fisher exact test for categorical data. The Wilson score interval with continuity correction was used to calculate 95% confidence intervals. Analyses were performed using R version 3.6.0 (R Foundation for Statistical Computing, Vienna, Austria). The authors have used appropriate statistical methods for analysis. Raw data are available on demand for editorial review.\\nMultivariable analyses were conducted to compare the primary outcomes between groups using three models to account for potential confounders. In model 1, we adjusted for potential confounders (maternal age at time of oocyte retrieval (categorical as ≤35 vs. >35), AMH (normal vs. low; normal was defined as AMH level of ≥\\xa01\\xa0ng/mL or use of oocyte donation), number of previous ET (categorical as <5 vs. ≥\\xa05) and PGTA (yes/no). For models 2 and 3, we used propensity score analysis. A propensity score was estimated using a multivariable logistic regression model with maternal age at time of oocyte retrieval (used as continuous in order to optimize matching and potentially reduce bias by avoiding matching a 35\\xa0year old with a 22\\xa0year old since both would be grouped as ≤35), AMH (normal vs. low), number of previous ET (categorical <5 vs. ≥\\xa05), and PGTA (yes/no).\\nFor the model 2, we adjusted for the propensity score (used as a covariate). In model 3, we conducted a propensity score matching analysis. Matching was performed using the SAS macro match.sas and was based on a caliper width of 0.1‐fold the SD of the logit‐transformed propensity scores. The association of the outcome with intervention versus control groups in the matched sample was examined using logistic regression analyses using Generalized Estimated Equations (GEE) with an unstructured correlation.\\nEthics Approval and Consent to Participate\\nThis study was approved by the McGill University Health Center (MUHC) ethics board, study number MUHC REB #2025‐10617.\",\n",
       "  'Results': \"3. Results\\n3.1. Patient Cohorts (Groups 1 and 2)\\nBetween January 1, 2021 and May 31, 2023, a total of 128 patients underwent Ultimpro testing. Seven patients were excluded from analysis (five patients were lost to follow‐up, one patient resorted to a gestational carrier and one patient's last embryo degenerated at thaw). All included patients adhered strictly to Ultimpro recommendations and were followed in clinic until cessation of treatment. Group 1 was comprised of 93 patients with unexplained RIF fitting our criteria (Table\\n1\\n). Group 2 was comprised of 28 patients who did not fit our RIF criteria as follows (each patient may have several): age >42 with autologous oocytes or >45 with oocyte donation (\\nn\\n=\\xa05), thin endometrium (\\nn\\n=\\xa05), poor quality embryos (\\nn\\n=\\xa014), <3 previous blastocyst transfers prior to Ultimpro testing (\\nn\\n=\\xa05), BMI >35 (\\nn\\n=\\xa01), partner with balanced translocation, and no pre‐implantation genetic screening of embryos (\\nn\\n=\\xa01) (Figure\\n2\\n).\\n3.1.1. Control Cohort (Group 3)\\nBetween January 2020 and December 2021, 1061 patient files from the MUHC Reproductive Center and MRIC were screened. Ninety‐four patients met our uRIF inclusion criteria and had not received any previous immunomodulatory treatment; they were included in our control group (Group 3) as previously described [\\n32\\n] (Figure\\n2\\n). Patients undergoing oocyte donation were excluded to preserve group homogeneity.\\n3.1.2. Patient Characteristics\\nBaseline patient characteristics are presented in Table\\n2\\n. Overall, patients in Group 1 had a higher median maternal age at oocyte retrieval (35\\xa0vs. 34,\\np\\n<\\xa00.001) and a higher mean number of previous high quality blastocyst transfers (4.74\\xa0vs. 4.13,\\np\\n=\\xa00.012) than Group 3. AMH levels were similar between Group 1 and Group 3. Patients in Group 1 had higher rates of pre‐implantation genetic screening (PGS) compared to Group 3 (34/93 [36.5%] vs. 13/94 [13.8%],\\np\\n=\\xa00.0004).\\nPatients in Group 2 had a higher mean maternal age at ET than both Group 1 and Group 3 (39.7\\xa0vs. 36.2,\\np\\n=\\xa00.002; 39.7\\xa0vs. 34,\\np\\n=\\xa00.0001, respectively), lower AMH levels for those undergoing autologous ET (1.65\\xa0vs. 3.22,\\np\\n=\\xa00.01; 1.65\\xa0vs. 3.5,\\np\\n=\\xa00.002, respectively), a lower number of previously failed high quality ETs (1.92\\xa0vs. 4.74, p<0.0001; 1.92\\xa0vs. 4.13, p<0.0001, respectively) and lower total number of previously failed ETs (3.6\\xa0vs. 4.74,\\np\\n=\\xa00.01; 3.6\\xa0vs. 4.85,\\np\\n=\\xa00.0012). Mean maternal BMI and proportion of patients with unexplained infertility was similar between all groups.\\n3.2. Control Group Outcomes\\nIn Group 3, there were 31 live births (31/94, 33%, 95% CI [0.24–0.43]). This is lower than expected compared to 2023 Canadian statistics (OPR per blastocyst transfer 46.2%) and compared to outcomes from a large representative Quebec Fertility clinic (Ovo Clinic, OPR 53.8%) (Table\\nS3\\n).\\n3.3. Ultimpro Outcomes\\nIn Group 1, there were 46 successful pregnancies (46/93, 49.5% [95% CI: 0.39–0.59]), which was higher than in Group 3 (31/94, 33% [95% CI: 0.24–0.43];\\np\\n=\\xa00.026) (Table\\n3\\n); unadjusted OR 1.99 (95% CI: 1.10–3.59) (Table\\n5\\n) and comparable to expected outcomes according to Canadian 2023 CARTR statistics and to outcomes from a large representative Quebec Fertility clinic (Table\\nS3\\n). Because the rate of successful pregnancy per ET likely decreases with the number of previously failed ETs [\\n2\\n], we stratified patients by RIF severity. Only patients with more severe RIF phenotypes (≥\\xa05 previously failed blastocyst transfers) had significantly higher success rates compared to control (21/38 [55.2%, 95% CI: 0.38–0.71]) vs. 5/25 [20%],\\np\\n=\\xa00.0084). Patients with less severe phenotypes trended toward more successful pregnancies than control, but this was not significant. Without stratifying for RIF severity, patients in the underactive endometrial immune profile had significantly more successful pregnancies than Group 3 (17/26 [64.5%] vs. 30/94 [32%],\\np\\n=\\xa00.003). This was not the case for the other three profiles (Table\\n3\\n). Notably, PGS nor maternal age did not affect successful Ultimpro outcomes (Tables\\n2\\nand\\n5\\n).\\nMultivariable analyses were conducted to compare rates of successful pregnancy between Groups 1 and 3 (Table\\n5\\n). In Model 1, we adjusted for potential cofounders (maternal age at oocyte retrieval, AMH level, number of previous ETs and pre‐implantation genetic testing for aneuploidy (PGT‐A). In Model 2, we adjusted for the propensity score (used as a covariate) which included maternal age as a continuous variable as well as AMH level, number of previous ET failures and PGT‐A. For both models, we were able to demonstrate a beneficial effect for Ultimpro testing with adjusted OR 2.00 (95% CI: 1.10–3.99) and 1.99 (95% CI: 1.10–3.99), respectively (Table\\n5\\n). Propensity score analysis was also conducted. Unmatched patients were excluded from the analysis and 48 patients were matched. As expected, this loss of sample size affected the statistical significance of our results, but nonetheless, we retain the size effect of our previous analyses (OR 2.28 [95% CI 0.87–5.97], Table\\n5\\n).\\nMiscarriage rates were higher in the over‐active endometrial immune profile group (9/41, 22%) compared to the other three groups (underactive 0/36 (0%), mixed 3/18 (16.7%), balanced 1/8 (12.5%) endometrial immune profiles), but this was not significant (\\np\\n=\\xa00.73 and\\np\\n=\\xa01, respectively, for mixed and balanced endometrial immune profiles). Of note, in the over‐active endometrial immune profile group, there were three anembryonic pregnancies and one aneuploidy on product of conception testing (47XX, +21); this may reflect miscarriage due to embryo aneuploidy rather than Ultimpro protocol failure. Compared to miscarriage rates in the general population, which range from 15% to 25% depending on maternal age [\\n33\\n] miscarriage rates in patients undergoing Ultimpro did not differ significantly (Table\\n4\\n). In Group 1, maternal characteristics were similar for patients who had a successful pregnancy, miscarried or failed their ET; both patients with successful pregnancies and miscarriages were more likely to have unexplained infertility compared to patients who failed to conceive post index ET (Table\\nS1\\n).\\nIn Group 2, there were nine successful pregnancies (9/28, 32.1%), 14 patients failed ET (50%), and five patients miscarried (17.9%). Since patients in Group 2 did not have RIF, we compared their outcomes to national Canadian averages of ongoing clinical pregnancy per ET (2023 CARTR statistics). In patients not fitting our criteria for RIF, Ultimpro did not improve rates of successful pregnancies (Table\\n3\\n, Table\\nS3\\n). Like Group 1, the miscarriage rate in Group 2 is comparable to age‐expected rates in the general population.\\n3.4. Pregnancy and Obstetric Outcomes\\nFor Group 1, 10 patients reported pregnancy complications (10/46, 21.7%). Median maternal age was 36 (range 29–42), median number of previously failed ET was 5 (range 3–8) and median BMI was 21.7 (range 18.6–30.1); one patient had a normal profile (1/8, 12.5%), four patients had an overactive profile (4/41, 9.7%), four patients had an underactive profile (4/26, 15.4%) and one patient had a mixed profile (1/18, 5.5%). Birthweight was available for 30 neonates (Table\\nS4\\n). For Group 2, six uncomplicated pregnancies and live births were recorded. Birthweights were available for five neonates; no neonatal complications were reported. Of the three ongoing pregnancies, one patient (overactive profile) has developed steroid‐induced GDB in her first trimester requiring insulin, she is currently 37\\xa0weeks pregnant. For the Group 3, birthweights were available for the six patients that delivered within the MUHC. Pregnancy complications were reported in four patients (4/31 (13%)) (Table\\nS4\\n).\",\n",
       "  'Discussion': \"Unexplained recurrent implantation failure will affect up to 10% of couples undergoing IVF [\\n34\\n,\\n35\\n]. Although the expected OPR per blastocyst transfer approach 46% in Canada (2023 CARTR report), this percentage is likely lower for patients after a uRIF diagnosis, with historical cohorts (including our control group) estimating 12%–30% per subsequent ET [\\n35\\n,\\n36\\n].\\nImmune‐mediated RIF (IM‐RIF) is mainly a diagnosis of exclusion based on lack of alternative explanation, failure of previous therapy, and sometimes on variances in peripheral blood NK cell testing. Because of limited ability to select patients with a diagnosis of IM‐RIF, studies evaluating the performance of immunomodulatory therapies to improve RIF outcomes have not shown convincing evidence of therapeutic benefit [\\n7\\n]. By assessing endometrial NK cells and factors known to influence their behavior during the WOI, Ultimpro may thus provide a more representative view of in‐situ endometrial immune events contributing to IM‐RIF and a better personalized approach to immune therapy. However, Ultimpro is an invasive test carrying low albeit present risks (infection, uterine perforation, pain), is expensive (1600–2000$) and can cause delays for ET (1–3\\xa0months). Although studies performed to date seem encouraging, they are observational and sometimes lack appropriate controls [\\n27\\n,\\n28\\n,\\n29\\n,\\n37\\n]. Furthermore, studies are all performed by the company marketing Ultimpro, providing potential biases. Thus, there is a need to (1) identify if and which patients would most benefit from Ultimpro testing and (2) determine if the risk‐cost/benefit ratio is clinically reasonable.\\nThe McGill University Health Center Reproductive Immunology Clinic (MRIC) is the Quebec reference center for patients with suspected IM‐RIF. In collaboration with referring colleagues from Quebec‐wide fertility clinics, we decided to pilot Ultimpro testing as of January 2021. Our criteria for Ultimpro testing was stringent (Table\\n1\\n), as we aimed to identify patients most likely to have an underlying diagnosis of IM‐RIF by excluding cofounding variables which can contribute to implantation failure (Group 1). We accounted for embryo quality (only patients whose index ET was a high blastocyst (Garner grade ≥\\xa03BB) were included, endometrial preparation, body mass index and maternal age at oocyte retrieval and ET as well as presence of pre‐implantation genetic screening (PGS). We did not assess endometrial receptivity as ERA (Igenomix) testing do not seem to improve pregnancy rates [\\n38\\n,\\n39\\n,\\n40\\n] and have fallen out of favor in Quebec fertility clinics. Our control population (Group 3) was selected using the same criteria as we wanted to determine ET outcomes in patients with uRIF without additional immune treatment. Overall LBR per index ET was lower for Group 3 (31/94, 33%) compared to Canadian (2023 CARTR Statistics) OPR per ET (Table\\nS3\\n) and in line with previous reports [\\n35\\n,\\n36\\n]. We also noted an inverse correlation between probability of live birth and number of previously failed ETs (Table\\n3\\n).\\nThere were some differences between Group 1 and Group 3. Maternal age at oocyte retrieval (35\\xa0vs. 34,\\np\\n=\\xa00.023) and the mean number of previous blastocyst transfer failures before index ET (4.74\\xa0vs. 4.14,\\np\\n=\\xa00.012) was higher in Group 1 compared to Group 3; Group 1 likely represents a group with a worse prognosis than Group 3. Nevertheless, patients treated according to Ultimpro recommendations had higher rates of successful pregnancies overall (46/93 [49.5%] vs. 31/94 [33%],\\np\\n=\\xa00.026). Importantly, patients in Group 1 had higher rates of PGS compared to Group 3 (34/93 [36.5%] vs. 13/94 [13.8%],\\np\\n=\\xa00.0004). However, PGS did significantly affect pregnancy outcomes (Table\\nS5\\n).\\nWhen stratifying by severity of RIF phenotypes, only patients with ≥\\xa05 previously failed blastocyst transfers benefitted significantly from Ultimpro (Table\\n3\\n). Although there was a trend for benefit in groups <5 previous blastocyst transfer failures, this was not significant. It is possible that only patients with a more severe RIF phenotype have true immune‐mediated RIF and correcting the underlying deficit alone will improve outcomes. Therefore, the significantly higher pregnancy rates seen in Group 1 may be solely driven by patients with ≥\\xa05 previous ET failures. Another explanation is that our study is underpowered to detect differences in outcomes between Group 1 and Group 3 for patients with milder RIF phenotypes; the baseline LBR in the control group with <5 previous ET failures is relatively high (37.6%), and more patients would be needed to detect differences.\\nAnother interesting finding in this study is that, without stratifying for RIF severity, patients with underactive endometrial immune profiles had significantly more successful pregnancies than controls (17/26 [65.4%] vs. 30/94 [32%],\\np\\n=\\xa00.003). Ultimpro recommendations for such a profile include mid‐luteal phase scratching on cycle prior to ET and luteal phase hCG support. Although previous studies have shown that endometrial scratch therapy may have marginal effects for improving IVF outcomes [\\n7\\n,\\n41\\n], perhaps this intervention would be most beneficial in well‐selected patients. Additionally, only 60% of patients in Group 1 received glucocorticoids (all patients with overactive and mixed endometrial immune profiles) (Figure\\n1\\n, Table\\n4\\n). Glucocorticoids would likely have suppressed the normal endometrial immune events required for implantation in patients with balanced and underactive endometrial immune profiles, thus decreasing implantation rates. This is an important concept as previous studies evaluating the utility of glucocorticoids for RIF did not select patients based on immune profiles and may have inappropriately concluded that steroids are not a useful adjunct for uRIF [\\n42\\n,\\n43\\n].\\nLastly, we included all patients who had Ultimpro testing but did not meet our testing criteria as a separate group (Group 2). These patients were excluded because they either did not have RIF (<3 previous ET failures) or because they had a probable explanation for RIF (advanced maternal age, thin endometrium, poor quality embryos, BMI\\xa0>\\xa035, balanced parental translocation). As expected, treatment recommendations according to Ultimpro results did not improve the rate of ongoing pregnancy per ET compared to overall Canadian statistics.\\nInterestingly, the complication rate was higher in Group 1 compared to Group 3 (Table\\nS4\\n). Two patients receiving glucocorticoids (GC) developed gestational diabetes (GDB) at 28\\xa0weeks requiring insulin (two other patients with an underactive profile developed GDB treated with diet, likely unrelated to Ultimpro treatment as there was no GC exposure). In patients receiving GC, steroid taper is performed at 6\\xa0weeks gestation (lasting 3–4\\xa0weeks). No patients receiving GC developed GDB in their first trimester. We cannot exclude that GC treatment in the first trimester predisposes to GDB in the third trimester. None of the patients in our cohort delivered macrosomic infants, no deliveries were complicated by shoulder dystocia.\\nPlacenta‐related complications were higher in Group 1. Two pregnancies were complicated by intra‐uterine growth restriction (IUGR). However, one patient had a balanced immune profile and did not receive any adjunctive treatment, and the other (underactive profile) delivered a neonate later diagnosed with a monogenic disorder explaining the IUGR. Two patients developed term preeclampsia without severity features (one with an underactive profile with conception post endometrial scratch and the other with an overactive profile receiving GC early in pregnancy). Lastly, one patient (overactive profile) had a stillbirth. She presented to hospital with acute fatty liver of pregnancy, thrombocytopenia, and acute kidney failure at term. She was on salicylic acid and low molecular weight heparin (known positive lupus anticoagulant, no previous history of miscarriages or clotting) and had received GC in early pregnancy. Her case was reviewed in a multidisciplinary morbidity and mortality meeting, and it was concluded that her medications did not contribute to her adverse pregnancy outcome.\\nStudies performed in patients with recurrent pregnancy loss do not show that glucocorticoid use in early pregnancy adversely affects pregnancy outcomes [\\n44\\n,\\n45\\n]; but if used at higher doses and continued throughout pregnancy, there are increased risks of gestational diabetes and pre‐term birth [\\n46\\n]. Although pre‐conception/early pregnancy glucocorticoids administration is likely safe, patients should be counseled and stringently monitored for gestational diabetes. Although we did not observe an increased risk of pre‐term birth in Group 1, patients should be counseled that this is a possibility with GC use early in pregnancy.\\nSimilarly, endometrial scratching prior to ET does not seem to significantly affect pregnancy outcomes [\\n47\\n]. It is therefore unlikely that the adverse pregnancy or obstetrical outcomes in Group 1 are attributable to treatments administered in early pregnancy, but not impossible. However, whether the higher complication rate in Group 1 is due to the underlying etiology of the infertility or to the treatment administered prior to conception/early pregnancy remains to be determined. Indeed, patients with uRIF may represent an at‐risk obstetric population irrespective of treatment administered and should be treated as such once pregnant. Unfortunately, longitudinal studies evaluating pregnancy outcomes in patients with uRIF are lacking. Perhaps an unfavorable endometrial microenvironment at conception may translate into placental pathology later in pregnancy; treatment aiming to correct the endometrial defect at implantation may not be sufficient to prevent adverse pregnancy outcomes. Long term follow‐up (from conception until parturition), particularly for patients with more severe RIF phenotypes, should be incorporated in all studies assessing treatment efficacy in patients with uRIF.\\nOur study has both strengths and limitations. In terms of strengths, patients were stringently selected, ensuring the most homogenous patient population possible. Patients are followed until delivery, providing the opportunity to describe not only OPR, but also LBR and pregnancy/neonatal complications. In terms of limitations, it is a retrospective, observational study which may hold undetected biases. Furthermore, the interpretation of test results is performed by Ultimpro using proprietary norms. Although the Ultimpro test was developed respecting a rigorous scientific method [\\n19\\n,\\n23\\n,\\n48\\n,\\n49\\n,\\n50\\n,\\n51\\n] and clinical implementation [\\n27\\n,\\n28\\n,\\n30\\n], thresholds for identifying immune dysregulation are not detailed, and there has been no independent validation of these thresholds or proof of inter‐laboratory reproducibility. Treatment recommendations are provided by MatriceLab Innove according to established algorithms and not by the treating physician.\\nGroup 3 is small and unmatched (reflecting the low incidence of true RIF (≤5%) [\\n6\\n]), but patients were selected with the same criteria, thus providing an adequate comparator for Group 1. Although most maternal characteristics were similar between Group 1 and 3, Group 1 had a higher number of mean ET failures prior to index pregnancy. Group 1 may thus represent a patient population with worse prognosis (which would emphasize the positive effect of Ultimpro). Group 3 was selected from an earlier timepoint (2020–2021) than Group 1 (2021–2023), but success rates per transferrable embryo did not significantly change during this time (2020–2023 Ovo/MUHC Reproductive Center statistics, CARTR 2020–2023 statistics).\\nA matched control group within the same timeframe as the Ultimpro cohort would have been ideal. However, since 2021–2022, most Quebec fertility clinics refer patients with ≥\\xa03 ET failures to the MRIC. It would now be difficult to find a comparable RIF control cohort within Quebec without crossover with the intervention group. Additionally, patients are referred to the MRIC from over 10 different fertility clinics in Quebec and Ontario, we were unable to include information of IVF cycle parameters, stimulation/ET protocols or conditions of embryo culture as this data are often not available in the patient's chart. In this study, we cannot account for variability in laboratory practices and their effect on ET success rates. Of note, the rate of pre‐implantation genetic screening (PGS) was higher in Group 1 than Group 3 but did not impact outcomes (Table\\nS5\\n).\\nNext, while overall patient numbers were robust (93 patients in Group 1 and 94 patients in Group 3), the number of patients with moderate‐severe RIF (≥\\xa04 previously failed high quality blastocyst transfers) was lower. Nevertheless, we were able to show a beneficial effect of Ultimpro for such patients.\\nAnother limitation is that the rate of pre‐implantation genetic testing for embryo aneuploidy was low (PGT‐A) (36% in Group 1 and 13.8% in the Control group). Although “true RIF” should incorporate ET failure with euploid embryos, PGT‐A testing is controversial for patients <35 as OPR is similar per ET with and without PGT‐A (2023 CARTR statistics). Additionally, because PGT‐A is not covered in Quebec, it is rarely performed for first IVF cycles or in patients <38\\xa0years of age. Further, we did not perform products of conception (POC) testing in all patients who miscarried and cannot determine if miscarriages occurring with Ultimpro recommendations represent a treatment failure or embryo aneuploidy. Most miscarriages were biochemical or <6\\xa0weeks; one patient had an 8‐week loss after positive viability scan but lost her pregnancy spontaneously at home. Only one patient who miscarried was able to perform POC testing and had a proven aneuploid loss (47XX, +21).\",\n",
       "  'Conclusion': 'Although our results are encouraging for Ultimpro use in patients with uRIF, it is too early to recommend routine incorporation of Ultimpro in IVF clinics as standard practice. A prospective validation cohort in our Center is currently underway, but a randomized controlled trial (RCT) is needed to define the role of Ultimpro in the management of uRIF patients. However, RCTs are expensive and time consuming; they must be well designed and adequately powered to maximize the information gained. Therefore, observational studies such as this one are important in identifying key variables and populations to target for an eventual RCT. Ultimpro is likely going to prove a useful clinical adjunct for select RIF patients, but it may be detrimental to others (specifically patients without RIF or patients with explained RIF). Potential risks of biopsy, adverse pregnancy outcomes related to treatment and cost of Ultimpro testing must be carefully weighed against perceived benefits; all of which can be addressed by a RCT.\\nAlthough Ultimpro currently assesses NK cell markers, the uterine immune microenvironment is also enriched in other key immune cell types such as T, B, and dendritic cells. Broadening the scope of the endometrial immune phenotyping may further define relevant immune profiles and their use in improved diagnostic accuracy. Indeed, studies using genomic approaches such as single cell transcriptomics to profile the underlying immune pathways in mid‐luteal phase endometrial samples are currently underway. These may eventually validate Ultimpro biomarkers and broad diagnostic categories but would also shed more light into normal immune events permitting implantation as well as pathogenic events underlying implantation failure. Although our independent pilot study shows promising preliminary results, it may be too soon to recommend incorporation of Ultimpro testing into mainstream clinical practice. Awaiting the results of an independent multi‐centered randomized controlled trial, Ultimpro should be used within a research setting (registry, longitudinal cohort study). This permits periodic practice audits for efficacy but more importantly, for drug side effects as well as maternal, pregnancy and fetal outcomes. Lastly, as physicians who deal with vulnerable and often desperate couples, new technology offering clinical guidance and improved pregnancy rates is attractive. However, careful consideration of whom this testing is offered to is of the utmost importance.',\n",
       "  'URL': 'https://doi.org/10.1111/aji.70131',\n",
       "  'Journal': 'American Journal of Reproductive Immunology'},\n",
       " {'PaperTitle': 'Variability and implications of recurrent implantation failure definitions used in the scientific literature: a systematic review',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Jessica K Lu, Yin Jun Law, Ning Zhang, Evangelia T Katsika, Efstratios M Kolibianakis, Christos A Venetis',\n",
       "  'Email': 'venetis@gmail.com',\n",
       "  'Abstract': 'STUDY QUESTION How is recurrent implantation failure (RIF) defined in published literature and what is the prognostic agreement of these definitions with recently introduced RIF criteria by ESHRE? SUMMARY ANSWER RIF definitions used in current clinical studies are highly variable and only a low proportion of published studies on RIF meet the ESHRE RIF diagnostic threshold. WHAT IS KNOWN ALREADY RIF is a key cause of ART failure and growing focus of ART research. However, RIF remains poorly and inconsistently defined in published literature, thereby making the interpretation and clinical applicability of RIF research difficult and highly problematic. STUDY DESIGN, SIZE, DURATION The electronic databases EMBASE (Ovid), PubMed, Cochrane Central Register Of Controlled Trials (CENTRAL), Scopus, and Web of Science were systematically searched up to 30 June 2024 using the search terms ‘recurrent implantation failure’ and ‘repeated implantation failure’ for original peer-reviewed journal articles that included RIF patients. PARTICIPANTS/MATERIALS, SETTING, METHODS The following data were manually extracted from eligible full-text articles: study methodology and characteristics, ART characteristics, and the RIF definition used. Extracted RIF definitions were analysed according to predetermined specifiers. The prognostic profile of these RIF definitions was compared with the 2023 ESHRE-recommended threshold for RIF diagnosis. MAIN RESULTS AND THE ROLE OF CHANCE The literature search identified 9853 studies, of which 748 were eligible for inclusion. Of these 748 studies, 589 studies (78.7%) provided one RIF definition, 83 studies (11.1%) used two definitions, three studies (0.4%) provided three or more definitions while 73 studies (9.8%) did not provide a definition for RIF. Of the 838 RIF definitions retrieved, there were a total of 503 unique RIF definitions. The three most common specifiers used to define RIF were embryo morphological quality (n\\u2009=\\u2009491, 58.6% of RIF definitions), number of transfer events (n\\u2009=\\u2009439, 52.4%), and cumulative number of embryos transferred (n\\u2009=\\u2009326, 38.9%). RIF was most frequently diagnosed as ‘failure of ≥3 embryo transfer events’ (n\\u2009=\\u200926) and ‘failure of ≥3 stimulated cycles’ (n\\u2009=\\u200922). The threshold for defining RIF based on the cumulative number of embryos transferred in total was significantly higher for cleavage-stage embryos compared to blastocysts (incidence rate ratio 2.15, P\\u2009<\\u20090.001). In most cases, the RIF definitions used did not meet the ESHRE-recommended RIF diagnostic threshold of >60% cumulative predicted chance of implantation. LIMITATIONS, REASONS FOR CAUTION This systematic review excluded abstracts and case-series. Several studies provided RIF definitions with limited detail or ambiguous terminology with potential for misclassification or misinterpretation. WIDER IMPLICATIONS OF THE FINDINGS There remains a high degree of variability and discrepancy between RIF definitions used in current clinical studies on RIF. Furthermore, the low proportion of studies meeting the ESHRE RIF diagnostic threshold casts doubts on whether the populations in these studies were truly RIF patients. As such, published research findings should be interpreted with caution. To enable wider clinical applicability of future research on the aetiology of and therapeutic interventions for RIF, it is imperative that a standardized RIF definition is meticulously implemented. STUDY FUNDING/COMPETING INTEREST(S) No specific external funding was sought or obtained for this study. All authors report no conflicts of interest with regard to this study. TRIAL REGISTRATION NUMBER This trial was registered in PROSPERO (CRD42022295349). Keywords: recurrent implantation failure, repeated implantation failure, definition, assisted reproductive technologies, ART, IVF WHAT DOES THIS MEAN FOR PATIENTS? In assisted reproduction, the term ‘recurrent implantation failure (RIF)’ is used to describe the lack of pregnancy despite multiple IVF cycles and embryo transfer attempts. However, previous research has shown that the definition of RIF is highly variable in published studies on this topic. This study has evaluated the extent of this variation and analysed how available scientific research defines RIF. Results demonstrate that the majority of published studies use their own unique definition for RIF and no widely accepted RIF definition currently exists. Furthermore, only a small proportion of published RIF definitions meet the recently introduced criteria recommended by the internationally recognized European Society for Human Reproduction and Embryology (ESHRE). This is problematic as it means that available RIF research cannot be widely applied to all RIF patients, who may be at risk of undergoing tests or treatments for RIF that are unnecessary or unhelpful. To enable more relevant and robust clinical research on RIF, a standardized RIF definition needs to be implemented.',\n",
       "  'Introduction': 'Embryo implantation is a complex process that requires synchrony between a competent blastocyst and a receptive endometrium via the successful interplay of embryonic and maternal factors. Recurrent or repeated implantation failure (RIF) is a poorly defined clinical entity encompassing the lack of successful embryo implantation following multiple embryo transfers in ART cycles. Despite advancements in ART, RIF still poses a major clinical challenge as a key cause of recurrent IVF and ICSI treatment failure. RIF has been reported to affect up to 15% of women undergoing IVF treatment (Margalioth et al., 2006; Busnelli et al., 2020), including women who respond appropriately to controlled ovarian stimulation and produce morphologically good-quality embryos. Not surprisingly, RIF has been the focus of an increasing body of research over the last two decades (Wang et al., 2022), including studies examining its aetiology and relevant therapeutic interventions. However, the high inconsistency regarding its definition renders the interpretation of published evidence highly problematic. This inconsistency was firstly described a decade ago by Polanski et al. (2014) in a systematic review assessing the use of RIF definitions in scientific literature, which found that the most commonly described definitions were ‘three or more failed treatment cycles’ or ‘two or more failed cycles’ (Polanski et al., 2014). Since 2014, however, the number of studies in the field of RIF has increased exponentially and newer definitions of RIF have been proposed, taking into account modern therapeutic strategies, such as preimplantation genetic testing (PGT) for aneuploidy (Pagidas et al., 2008; Pirtea et al., 2021) and elective preservation of all embryos (Magdi et al., 2016, 2017; Rigos et al., 2021). Additionally, researchers have proposed individual algorithms for diagnosing a patient with RIF, on the basis of important prognostic factors such as age, number, and quality of embryos (Somigliana et al., 2018; Ata et al., 2021; Rozen et al., 2021). Recently, ESHRE has introduced criteria which aim to standardize the diagnosis of RIF. The present systematic review aims to provide an updated overview and analysis of the definitions of RIF used in published literature and to assess their prognostic agreement with the recently introduced RIF criteria by ESHRE.',\n",
       "  'Methods': 'This systematic review was registered in PROSPERO (CRD42022295349) and conducted in accordance with the PRISMA guidelines (Page et al., 2021) (Fig. 1). Figure 1. Open in a new tab Preferred Reporting Items for Systematic Reviews and Meta-analyses (PRISMA) flow diagram. Eligibility criteria All original research articles published in peer-reviewed journals of any study design or language reporting to have included patients with RIF (either as the primary study group or subgroup) were included. Case reports, conference abstracts, reviews, and textbook chapters were excluded. Sources Studies were identified by systematically searching the databases EMBASE (Ovid), PubMed, Cochrane Central Register of Controlled Trials (CENTRAL), Scopus, and Web of Science from their inception until 30 June 2024. Search strategy The literature search was performed by three independent reviewers (J.K.L., Y.J.L., and N.Z.) using a predetermined search strategy with the following search terms: ‘recurrent implantation failure’, ‘repeated implantation failure’. Study selection Search results were collated in reference management software (Endnote version 20; Clarivate Analytics; Philadelphia, PA, USA) and duplicates were removed prior to screening. Compiled studies were sequentially screened by three independent reviewers (J.K.L., Y.J.L., and N.Z.), first by scanning article titles for relevance to reproductive medicine and assisted reproductive technology, followed by scrutinizing abstracts and then the full-text article to select studies that met eligibility criteria for inclusion in this systematic review. Reference lists of relevant publications were manually searched for additional eligible references. Any disagreement between reviewers about study selection was resolved by a fourth reviewer (C.A.V.). Data extraction Four independent reviewers (J.K.L., Y.J.L, N.Z., and E.T.K.) extracted the following predefined data from eligible full-text articles: methodology (journal, study type and design, study period, setting, country and region of origin, main research question), study characteristics (inclusion and exclusion criteria, maternal age limit, sample size, whether the study subjects comprised of RIF patients entirely or as a subgroup only, female age, cause of infertility if known, RIF incidence if reported), ART characteristics (fertilization method, embryo type, i.e. fresh and/or frozen, number of embryos transferred per patient, use of PGT, embryo developmental stage, and quality), and the RIF definition used. Studies were grouped by type (interventional, predictive, prognostic, diagnostic) according to the clinical question addressed: interventional studies analysed the effectiveness of treatments for RIF, predictive studies analysed aetiological and/or risk factors associated with RIF, prognostic studies analysed outcomes in women with RIF and diagnostic studies analysed modalities for diagnosing RIF. Only RIF definitions that were provided in the ‘Materials and methods’ section of eligible studies and/or were explicitly stated as being used for patient recruitment were accepted. Statistical analysis To facilitate interpretation of and comparison between the RIF definitions extracted, worded definitions were deconstructed into their core components according to the following specifiers: cumulative number of embryos transferred, number of embryo transfer events, number of embryos transferred per cycle, embryo developmental stage at transfer and morphological quality, embryo type (fresh and/or frozen), whether genetic testing for aneuploidy was performed, number of stimulated cycles, whether failed cycles were consecutive, maternal age, maternal ovarian reserve, how failed implantation was defined, and any additional specifiers not mentioned above. A set of predetermined rules was devised to ensure this process of interpreting and transcribing worded RIF definitions using these specifiers was consistent and reproducible (Supplementary File S1). Statistical analysis was performed to determine the use and variability in RIF definitions described in the literature, including the proportion of studies using each of the core components listed above. Recorded combinations of RIF definition components were analysed to determine the most common RIF definitions applied in published literature. To examine whether differing embryo developmental stage and maternal age were reflected in published RIF definitions, subgroup analyses were undertaken to compare RIF definitions using cleavage-stage versus blastocyst-stage embryos and for maternal age <40 versus ≥40\\u2009years. Binomial logistic regression was used to assess for association between year of publication and embryo developmental stage for defining RIF. Multinomial logistic regression was conducted to quantify the effect of publication year on the embryo type (fresh and/or frozen) and developmental stage (cleavage or blastocyst) used and the region of study origin. Poisson regression was used to examine the effect of publication year on the number of specifiers used to define RIF, as well as the effect of embryo developmental stage on the minimum number of: (a) cumulative embryos transferred, (b) embryo transfer events, and (c) stimulated cycles required to meet the definition of RIF. The results were evaluated using odds ratios (ORs), relative risk ratios (RRRs), and incidence rate ratios (IRRs), respectively, and statistical significance was defined as P\\u2009<\\u20090.05. To assess the prognostic profile and comparability of RIF definitions in the literature, we applied the thresholds for maternal age, total number of embryo transfer events, and embryo ploidy status used by published RIF definitions to aggregated ART data generated from European registries recently published by ESHRE (ESHRE Working Group on Recurrent Implantation Failure et al., 2023) to estimate the cumulative likelihood of pregnancy for each reported definition using the following four models: ‘Worst case scenario’ model—Maternal age was taken as the maximum reported age of included participants. For studies that did not specify cut-off values for maternal age or transfer events, maternal age was assumed to be ≥40\\u2009years and embryo transfer events ≥2 as the minimum thresholds for this analysis. ‘Optimistic theoretical cohort’ model—Maternal age was assumed to be <35 years. The number of embryo transfer events was assumed to be ≥2 if not otherwise specified. ‘Available data’ model—Only studies reporting both maximum maternal age and minimum number of embryo transfers were included. ‘Optimistic theoretical minimum embryo transfer’ model—Maternal age was assumed to be <35\\u2009years for all studies reporting a minimum number of embryo transfers. It should be noted that this additional analysis was a post hoc decision as the ESHRE criteria on RIF had been published in 2023 (ESHRE Working Group on Recurrent Implantation Failure et al., 2023) (after protocol registration) and it was considered highly relevant to assess the concordance of the scientific literature with the newly introduced criteria.',\n",
       "  'Results': 'Characteristics of included studies The systematic search resulted in 9853 studies and the full text of 2433 articles were assessed for eligibility. A total of 748 studies met the inclusion criteria and were included in the review (Supplementary File S2). The characteristics of these studies are summarized in Table 1. The number of included studies by publication year increased exponentially over time (averaging 17.0% per year) (Fig. 2), particularly in Asia, which has produced 71.9% of included studies published after 2014, and over half (50.2%) of the included studies published after 2022 have originated from China (Supplementary Fig. S1). Table 1. Characteristics of included studies. Study characteristics Number of studies (n) % Total included studies 748 100 Country China 236 31.5 Iran 88 11.8 Turkey 52 6.9 Spain 29 3.9 Other (51 other countries) 343 46.0 Region of origin Africa 13 1.7 Asia 497 66.5 Europe 182 24.3 North America 32 4.3 Oceania 10 1.3 South America 12 1.7 Not specified 3 0.4 Study design Case-control 224 30.0 Cohort 287 38.4 Cross-sectional 14 1.9 Interventional (non-RCT) 32 4.3 Observational 82 11.0 Pilot 16 2.1 RCT 73 9.8 Other 20 2.7 Study approach Prospective 298 39.8 Retrospective 435 58.2 N/A (cross-sectional) 15 2.0 Article type Diagnostic 98 13.1 Interventional 287 38.4 Predictive 311 41.6 Prognostic 52 6.9 Setting Hospital 370 49.5 Fertility centre/clinic 267 35.7 Research institute 28 3.7 University 16 2.1 Database 12 1.6 NR 55 7.4 Total sample size (n\\u2009=\\u2009739, excluding nine studies that did not report) Participants (n) Minimum 8 Maximum 45\\u2009921 Mean 356.5 Standard deviation 1896.7 RIF sample size (n\\u2009=\\u2009716, excluding 32 studies that did not report) Participants (n) Minimum 3 Maximum 45\\u2009921 Mean 242.3 Standard deviation 1822.4 Open in a new tab NR, not reported; RCT, randomized controlled trial; RIF, recurrent implantation failure. Figure 2. Open in a new tab Number of articles published per year included in this systematic review. Note: In 2024, articles with publication date up to 30 June were included in this systematic review. The studies were performed in 57 countries, most frequently originating from China (n\\u2009=\\u2009236, 31.5%), Iran (n\\u2009=\\u200988, 11.8%), and Turkey (n\\u2009=\\u200952, 7.0%), as illustrated in Fig. 3. Of note, the most common study designs were cohort (n\\u2009=\\u2009287, 38.4%) and case-control (n\\u2009=\\u2009224, 30.0%), while randomized control trials (RCT) only comprised of 9.8% (n\\u2009=\\u200973) of the included studies. Figure 3. Open in a new tab Distribution of country of origin for articles included in the review. RIF definitions Of the 748 included studies, 589 studies (78.7%) used a single definition, 83 studies (11.1%) used two definitions, three studies (0.4%) provided three or more definitions, while 73 studies (9.8%) did not provide a definition for RIF. Justification of RIF definition Of the 675 studies that reported at least one definition (765 definitions provided in total), 520 studies (68.0%) did not justify their definition/s of RIF using published data while 245 studies (32.0%) based their definition/s on published literature. RIF definitions: analysis of criteria used Details of the 838 extracted RIF definitions are summarized in Table 2. The three most common specifiers used to define RIF were embryo morphological quality (n\\u2009=\\u2009491, 58.6%), followed by number of transfer events (n\\u2009=\\u2009439, 52.4%) and cumulative number of embryos transferred (n\\u2009=\\u2009326, 38.9%) (Fig. 4). The number of specifiers used to define RIF ranged from 0 to 9, with a median of 3 and standard deviation 1.8 (Supplementary Fig. S2). On analysis of all combinations of specifiers applied, RIF was most commonly defined by a minimum number of embryo transfer events or a minimum number of stimulated cycles alone (Fig. 5). Table 2. Analysis of criteria used in RIF definitions provided by included studies. Criteria used to define RIF Outcome n (%) Range Mean Median SD Min cumulative # embryos transferred 2–15 5.0 4 2.7 2 embryos 38 (4.5%) 3 embryos 37 (4.4%) 4 embryos 155 (18.5%) ≥5 embryos 96 (11.5%) NR 512 (61.1%) Min # embryo transfer events 2–6 2.7 3 0.6 2 transfers 157 (18.7%) 3 transfers 263 (31.4%) ≥4 transfers 19 (2.3%) NR 399 (47.6%) Min # embryos transferred per cycle 1–4 1.4 1 0.7 1 embryo 55 (6.6%) 2 embryos 17 (2.0%) ≥3 embryos 6 (0.7%) NR 760 (90.7%) Embryo developmental stage Cleavage only 59 (7.0%) Blastocyst only 76 (9.1%) Cleavage or blastocyst 36 (4.3%) NR 667 (79.6%) Embryo morphological quality Top quality 166 (19.8%) At least good quality 319 (38.1%) At least fair quality 5 (0.6%) At least poor quality 1 (0.1%) NR 347 (41.4%) Embryo type (fresh, frozen) Fresh only 43 (5.1%) Frozen only 9 (1.1%) Fresh or frozen 153 (18.3%) NR 633 (75.5%) Embryo ploidy Euploid 10 (1.2%) Not tested 1 (0.1%) NR 827 (98.7%) Min # stimulated cycles 1\\u2009cycle 8 (1.0%) 1–6 2.6 3 0.7 2 cycles 109 (13.0%) 3 cycles 150 (17.9%) 4 cycles 11 (1.3%) ≥5 cycles 3 (0.4%) NR 557 (66.4%) Consecutive failed ART cycles Required 92 (11.0%) NR 746 (89.0%) Maternal upper age limit (inclusive) 30–49 39.0 39 2.4 30–34 7 (0.8%) 35–39 105 (12.5%) 40–44 31 (3.7%) 45–49 4 (0.5%) NR 691 (82.5%) Good maternal ovarian/hormonal reserve Required 21 (2.5%) NR 817 (97.5%) Defined measure of failed implantation Lack of positive urine and/or serum hCG 64 (7.6%) Lack of visible gestational sac on US 12 (1.4%) Lack of clinical pregnancy 138 (16.5%) Lack of viable pregnancy/delivery 3 (0.4%) NR 621 (74.1%) Additional specifiers (e.g. idiopathic RIF, endometrial thickness at implantation, etc.) Reported 130 (15.5%) NR 708 (84.5%) Open in a new tab NR, not reported; RIF, recurrent implantation failure; #, number of. Figure 4. Open in a new tab Specifiers used to define recurrent implantation failure. Figure 5. Open in a new tab Common combinations of specifiers used to define RIF (n, %). Min., minimum; max., maximum; #, number of; RIF, recurrent implantation failure. Common RIF definitions Of the 838 RIF definitions retrieved from eligible studies, there were a total of 503 unique RIF definitions. RIF was most frequently diagnosed as ‘failure of ≥3 transfer events’, occurring 26 times, followed by ‘failure of ≥3 stimulated cycles’ (22 times) and ‘failure to achieve clinical pregnancy after transfer of at least four good quality embryos in a minimum of three fresh or frozen transfer events in a woman of age <40\\u2009years’ (19 times). Other repeated RIF definitions included ‘failure of ≥2 transfer events’ (18 times), ‘failure of ≥3 transfer events with at least good quality embryos’ (14 times), and ‘failure of ≥3 stimulated cycles top quality embryos’ (14 times). Subgroup analyses Subgroup analysis of RIF definitions was conducted hierarchically based on female age (less than 40\\u2009years) and embryo developmental stage (cleavage- versus blastocyst-only embryos), as detailed in Table 3. RIF definition criteria analysed included minimum cumulative number of embryos transferred, minimum number of embryo transfer events, minimum number of stimulated cycles, embryo developmental stage, and embryo morphological quality. Table 3. Subgroup analysis of criteria used in RIF definitions provided by included studies. Criteria used to define RIF Outcome n (%) Range Mean Median SD Female age <40\\u2009years (n\\u2009=\\u2009112) Min cumulative # embryos transferred 2–11 4.2 4 1.8 2 embryos 7 (6.3%) 3 embryos 5 (4.5%) 4 embryos 55 (49.1%) ≥5 embryos 6 (5.4%) NR 39 (34.8%) Min # embryo transfer events 2–6 2.9 3 0.7 2 transfers 18 (16.1%) 3 transfers 52 (46.4%) ≥4 transfers 5 (4.5%) NR 37 (33.0%) Min # stimulated cycles 2–4 2.7 3 0.6 2 cycles 10 (8.9%) 3 cycles 15 (13.4%) 4 cycles 2 (1.8%) NR 85 (75.9%) Embryo developmental stage Cleavage only 6 (5.4%) Blastocyst only 10 (8.9%) Cleavage or blastocyst 5 (4.5%) NR 91 (81.2%) Embryo morphological quality Top quality 25 (22.3%) At least good quality 53 (47.3%) NR 34 (30.4%) Defined measure of implantation failure Lack of positive urine or serum hCG 1 (0.9%) Lack of visible gestational sac on US 2 (1.8%) Lack of clinical pregnancy 46 (41.1%) NR 63 (56.2%) Cleavage only embryos (n\\u2009=\\u200959) Min cumulative # embryos transferred 4–11 5.9 4 2.6 4–5 embryos 28 (47.4%) 6–7 embryos 4 (6.7%) 8–9 embryos 7 (11.9%) ≥10 embryos 8 (13.6%) NR 12 (20.4%) Min # embryo transfer events 2–4 2.5 2 0.7 2 transfers 19 (32.2%) 3 transfers 9 (15.3%) 4 transfers 3 (5.1%) NR 28 (47.4%) Min # stimulated cycles 1–4 2.3 2 0.6 1\\u2009cycle 1 (1.7%) 2 cycles 15 (25.4%) 3 cycles 5 (8.5%) 4 cycles 1 (1.7%) NR 37 (62.7%) Embryo morphological quality Top quality 11 (18.6%) At least good quality 40 (67.8%) NR 8 (13.6%) Defined measure of failed implantation Lack of positive urine and/or serum hCG 19 (32.2) Lack of clinical pregnancy 13 (22.0%) NR 27 (45.8%) Blastocyst only embryos (n\\u2009=\\u200976) Min cumulative # embryos transferred 2–4 2.7 3 0.8 2 embryos 25 (32.9%) 3 embryos 14 (18.4%) 4 embryos 12 (15.8%) NR 25 (32.9%) Min # embryo transfer events 2–6 2.6 2 0.8 2 transfers 30 (39.5%) 3 transfers 22 (29.0%) ≥4 transfers 4 (5.2%) NR 20 (26.3%) Min # stimulated cycles 1–3 2.5 3 0.8 1\\u2009cycle 1 (1.3%) 2 cycles 2 (2.6%) 3 cycles 5 (6.6%) NR 68 (89.5%) Embryo morphological quality Top quality 21 (27.6%) At least good quality 35 (46.1%) NR 20 (26.3%) Defined measure of implantation failure Lack of positive urine and/or serum hCG 8 (10.5%) Lack of clinical pregnancy 18 (23.7%) NR 50 (65.8%) Open in a new tab NR, not reported; RIF, recurrent implantation failure; US, ultrasound; #, number of. Regression analyses Over time, studies were significantly more likely to use both fresh and frozen embryo transfers for RIF definition (RRR\\u2009=\\u20091.23, 95% CI 1.13–1.33, P\\u2009<\\u20090.001) compared to the use of only fresh embryo transfers (Table 4). There was no significant association between the year of publication and the number of studies using cleavage or blastocyst-stage embryo transfers in their RIF definition. Over time, significantly fewer studies originated from Europe, North America, Oceania, and South America compared to studies originating from Asia (Table 4). Table 4. Multinomial regression analysis of factors associated with year of publication. Variable Relative risk ratio (95% CI) P-value Embryo developmental stage Year of publication Blastocyst only (reference) Cleavage-stage only 0.945 (0.868–1.028) 0.189 Embryo type Year of publication Fresh embryo transfer only (reference) Frozen embryo transfer only 1.425 (1.069–1.900) 0.016 * Fresh or frozen embryo transfer 1.225 (1.130–1.328) <0.001 * Region of origin Year of publication Asia (reference) Africa 0.984 (0.874–1.106) 0.783 Europe 0.914 (0.885–0.944) <0.001 * North America 0.859 (0.813–0.907) <0.001 * Oceania 0.843 (0.772–0.920) <0.001 * South America 0.877 (0.805–0.955) 0.003 * Open in a new tab * P\\u2009<\\u20090.05. The threshold for defining RIF according to the cumulative number of embryos transferred was significantly higher for cleavage-stage embryos compared to blastocysts (IRR\\u2009=\\u20092.15, 95% CI 1.76–2.64, P\\u2009<\\u20090.001) (Table 5). No significant associations were detected between: (i) the year of publication and number of specifiers included in definitions used, (ii) the developmental stage of embryos transferred and the minimum number of embryo transfer events to meet the definition of RIF, and (iii) the developmental stage of embryos transferred and the minimum number of stimulated cycles required to meet the definition of RIF (Table 5). Table 5. Poisson regression analyses between year of publication and number of specifiers of the RIF definitions, as well as between embryo developmental stage and (a) the minimum number of embryos transferred cumulatively, (b) the minimum number of embryo transfer events, and (c) the minimum number of stimulated cycles used to define RIF. Variable Incidence rate ratio (95% CI) P-value Number of specifiers used to define RIF Year of publication 1.006 (0.998–1.013) 0.166 Embryo developmental stage Minimum number of embryos transferred cumulatively to define RIF Blastocyst (reference) Cleavage-stage 2.155 (1.759–2.640) <0.001* Minimum number of embryo transfer events to define RIF Blastocyst (reference) Cleavage-stage 0.959 (0.728–1.265) 0.768 Minimum number of stimulated cycles to define RIF Blastocyst (reference) Cleavage-stage 0.909 (0.541–1.527) 0.719 Open in a new tab RIF, recurrent implantation failure. * P\\u2009<\\u20090.05. Prognostic models Based on the four models we proposed to assess the prognostic profile of patients included in the published studies, the proportion of RIF definitions that met the ESHRE-recommended RIF diagnostic threshold of >60% cumulative likelihood of pregnancy (ESHRE Working Group on Recurrent Implantation Failure et al., 2023) was 2.1% for Model 1, 34.5% for Model 2, 6.4% for Model 3, and 65.1% for Model 4 (Table 6). Table 6. Prognostic models estimating the cumulative likelihood of pregnancy. Prognostic model 1 a (n\\u2009=\\u2009838) Cumulative likelihood of pregnancy (%) Frequency (% of total) 27.8 494 (59.0) 38.6 209 (24.9) 45.1 50 (6.0) 47.8 9 (1.1) 53.1 5 (0.6) 55.6 2 (0.2) 59.3 51 (6.1) 62.3 3 (0.4) 69.9 2 (0.2) 77.7 1 (0.1) 78 1 (0.1) 82.4 7 (0.9) 89.7 1 (0.1) 92.6 2 (0.2) 95.4 1 (0.1) Prognostic model 2 b (n\\u2009=\\u2009838) Cumulative likelihood of pregnancy (%) Frequency (% of total) 53.1 549 (65.5) 67.9 260 (31.0) 78 12 (1.4) 84.9 3 (0.4) 89.7 4 (0.5) 90 7 (0.8) 96.8 3 (0.4) Prognosis model 3 c (n\\u2009=\\u200994) Cumulative likelihood of pregnancy (%) Frequency (% of total) 27.8 9 (9.6) 38.6 8 (8.5) 45.1 16 (17.0) 47.8 2 (2.1) 53.1 2 (2.1) 59.3 51 (54.2) 69.9 2 (2.1) 77.7 1 (1.1) 78 1 (1.1) 89.7 1 (1.1) 95.4 1 (1.1) Prognostic model 4 d (n\\u2009=\\u2009439) Cumulative likelihood of pregnancy (%) Frequency (% of total) 53.1 153 (34.9) 67.9 260 (59.2) 78 12 (2.7) 84.9 3 (0.7) 89.7 4 (0.9) 90 4 (0.9) 96.8 3 (0.7) Open in a new tab a Prognostic model 1: ‘Worst case scenario’—if not otherwise specified, maternal age was assumed to be ≥40\\u2009years and number of embryo transfers was assumed to be 2. b Prognostic model 2: ‘Optimistic theoretical cohort’—maternal age was assumed <35\\u2009years and if not otherwise specified, the number of embryo transfers was assumed to be 2. c Prognostic model 3: ‘Available data model’—only studies reporting maximum maternal age and minimum number of embryos transferred were included. d Prognostic model 4: ‘Theoretical minimum embryo transfer’ model—maternal age was assumed to be <35\\u2009years for all studies reporting a minimum number of embryo transfers. Cumulative likelihood of pregnancy based on aggregate data published by ESHRE (ESHRE Working Group on Recurrent Implantation Failure et al., 2023).',\n",
       "  'Discussion': 'RIF has been a major challenge for patients and clinicians since the early days of ART and substantial research efforts have been made to identify relevant therapeutic interventions. This systematic review analysed the research output regarding RIF, clearly demonstrating the increasing interest in this topic over the past decade. Importantly this systematic review showcased the marked variability in the definition of RIF and the evolution of the use of different components and descriptors over the last four decades. The global interest in RIF is demonstrated by the widespread distribution of available studies originating from 55 different countries. Interestingly, Asia has had a major contribution in this field over the past decade, contributing annually more than 70% of published research since 2014 (Supplementary Fig. S1). Despite the widely acknowledged importance of RIF, no consistent or internationally accepted clinical definition of RIF currently exists and RIF is not yet recognized in the International Committee Monitoring Assisted Reproductive Technologies (ICMART) glossary (Zegers-Hochschild et al., 2017). While most studies (91.5%) provided some form of RIF definition, the majority of these definitions (62.3%) were described using terminology that was ambiguous or lacked clarity, requiring assumptions to be made (Supplementary File S1). For example, by using terms such as ‘IVF-embryo transfer/ICSI-embryo transfer attempts’ of ‘ART cycles’ without specifying whether this refers to the number of stimulation cycles or embryo transfer events, there exists potential disparity between the intended versus interpreted RIF definition. It should be noted also that the term ‘RIF’ is often erroneously used synonymously with ‘recurrent/repeated IVF failure’, the latter of which incorporates other causes of lack of clinical pregnancy/live birth following ART such as recurrent miscarriage, poor response to ovarian stimulation, poor embryo quality or development etc.—this further contributes to the confusion surrounding the definition of RIF. Even more concerning though is the striking variability in the definitions of RIF used in the literature. Out of the 838 definitions used in the 748 reports analysed, there were 503 unique definitions reported, which renders the comparison of different interventions for RIF and the synthesis of available evidence an impossible task. This creates significant issues around the interpretation of the available literature, particularly as it pertains to the population sampled by each study, limiting its external validity (i.e. generalizability). This greatly diminishes the value of published research and represents an unfortunate waste of resources with minimal, if any, translational value. It is also not surprising that most studies (67.6%) did not explain how they formulated their definition of RIF, which raises of possibility of convenience sampling (while also permitting selection bias) and further devalues the available evidence. The need for a homogeneous, a priori definition of conditions has been previously highlighted in the case of poor ovarian response in ART which eventually led to the introduction of the Bologna criteria (Ferraretti et al., 2011) which, although not perfect, have represented a major step in the right direction. Similarly, recently the ESHRE working group on RIF recognized the need for a more standardized definition of RIF based on the cumulative probability of pregnancy depending on several prognostic factors such as female age, the developmental stage of the embryos (cleavage versus blastocyst), embryo ploidy, and the number of embryos transferred (Coughlan et al., 2014; Ata et al., 2021; Rozen et al., 2021; ESHRE Working Group on Recurrent Implantation Failure et al., 2023; Pirtea et al., 2023). In that respect, this systematic review identified that the most common specifier used in the definition of RIF was embryo morphological quality. This underlines the importance researchers place on the prognostic value of embryo morphology, which has been shown to have a correlation with embryo ploidy in blastocysts (Capalbo et al., 2014) and even with the probability of ongoing pregnancy in case of euploid blastocysts (Irani et al., 2017). Another important determinant of success is the developmental stage of the embryo, however, this seems to have been underutilized in the definitions of RIF in the literature. The most plausible explanation is the fact that extended blastocyst culture has been introduced only in the last two decades and has had variable uptake in certain areas of the world, particularly in Asia (Kushnir et al., 2017; Glujovsky et al., 2022). This is also the likely reason behind the infrequent use of the number of embryos transferred per embryo transfer event in the definition of RIF. Single embryo transfer has also been popularized in the last decade (particularly along with the establishment of blastocyst culture) (van Montfoort et al., 2006; Practice Committee of the American Society for Reproductive Medicine, 2021), and, for this reason, the transfer of multiple cleavage stage embryos per transfer event has been the norm. Therefore, most researchers have opted for the use of cumulative number of embryos transferred, and this indeed has been found to be higher for cleavage-stage embryos compared to blastocysts. Another important determinant of pregnancy achievement is embryo ploidy which has been repeatedly shown to be superior to embryo morphology (Yang et al., 2012; Capalbo et al., 2014; Munné et al., 2019). The implantation potential of a euploid blastocyst has been reported to range between 50% and 70% (Forman et al., 2013) and therefore the probability of not having achieved a pregnancy after the sequential transfer of three euploid blastocysts has been shown to be <7% (Pirtea et al., 2021). Recently, an updated analysis suggested that the transfer of up to five euploid blastocysts (Gill et al., 2023) leads to >98% cumulative probability of pregnancy. Alongside aneuploidy, the Pirtea et al. (2021) and Gill et al. (2023) studies both excluded other known factors affecting implantation potential, including intrauterine pathology and advanced maternal age (>45\\u2009years), to investigate the existence and estimate the prevalence of otherwise unexplained RIF. For this reason, it has been recently argued that in order to limit the burden of a false diagnosis, RIF should only be diagnosed after known causes of implantation failure have been investigated and excluded or, if euploidy status is unknown, after the cumulative probability of implantation has been calculated from an age-adjusted equivalent number of untested blastocysts (Ata et al., 2021; Pirtea et al., 2023). Considering the lack of universal availability of PGT, its associated costs and the concerns that have been raised regarding embryo biopsy and the ability of PGT to enhance the cumulative chance of pregnancy, researchers have suggested introducing female age as a surrogate predictor of the probability of embryo euploidy (Ata et al., 2021). Unfortunately, this has been used as a component of the RIF definition in less than one in seven of the studies. Recently proposed models to define RIF have incorporated female age (Coughlan et al., 2014; Ata et al., 2021; ESHRE Working Group on Recurrent Implantation Failure et al., 2023) which is likely to reduce overdiagnosis and increase homogeneity in the population examined. The diagnosis of RIF on the basis of prognosis has been a subject of debate recently, where several authors have proposed different cut-offs of cumulative chance of pregnancy over which diagnostic and therapeutic interventions are justified. Venetis and Kolibianakis (2019) proposed a cut off of 85%, Rozen et al. (2021) suggested a cut-off of 80%, and Ata et al. (2021) proposed a cut-off of 95%. ESHRE, following polling among the members of the working group, suggested a more relaxed definition by proposing a cut-off of 60%. All these cut-offs are arbitrary in nature and proper studies, including health economic assessments, need to be performed to ascertain what is the cut-off that has the best sensitivity and specificity to identify populations that benefit from further diagnostic and therapeutic interventions or their inclusion in clinical research. One of the most revealing aspects of this systematic review is the assessment of the prognostic profile of the population included in each publication. Based on this analysis, for three out of four scenarios, very few of the studies satisfy the criteria of the RIF population according to the >60% cut-off proposed by ESHRE (ESHRE Working Group on Recurrent Implantation Failure et al., 2023). This essentially could be translated to significant overdiagnosis of RIF both in previously published research and therefore, in everyday clinical practice. Further to the potential inappropriate use of diagnostic and therapeutic interventions in populations that do not need them, testing an intervention that might be beneficial for RIF in a sample of patients where very few are truly RIF will likely lead to effect dilution and therefore failure to demonstrate the benefit of the intervention (Ata et al., 2021). This systematic review is also characterized by limitations that need to be discussed. Abstracts and case-series were excluded as they were considered to provide limited information regarding RIF definition. Most importantly, even by including only published manuscripts, several studies had limited data and required certain assumptions to be made. Even though these were made on the basis on predetermined criteria to ensure consistency and reproducibility, the potential for misclassification cannot be excluded. For this reason, in the case of ascertaining the prognostic profile of the populations examined based on the reported definitions, several scenarios were utilized in order to evaluate the robustness of our findings when the assumptions made varied.',\n",
       "  'Conclusion': 'In conclusion, the present systematic review clearly shows the clinical conundrum that RIF has been over the past few decades and the complete lack of a homogeneous definition in the relevant research. This, along the fact that in most studies the included population does not meet the RIF criteria of ESHRE, should caution the readers in how they should interpret the published evidence for RIF diagnostic and therapeutic interventions and emphasize the need for future research on the basis of the recent, more standardized definitions proposed.',\n",
       "  'URL': 'https://doi.org/10.1093/hropen/hoaf033',\n",
       "  'Journal': 'Hum Reprod Open'},\n",
       " {'PaperTitle': 'Uterine fluid microRNAs are dysregulated in women with recurrent implantation failure',\n",
       "  'Publication': '2022',\n",
       "  'Authors': 'Carolina von Grothusen, Caroline Frisendahl, Vijayachitra Modhukur, Parameswaran Grace Lalitkumar, Maire Peters, Omid R Faridani, Andres Salumets, Nageswara Rao Boggavarapu, Kristina Gemzell-Danielsson',\n",
       "  'Email': 'caroline.frisendahl@ki.se',\n",
       "  'Abstract': 'STUDY QUESTION Is the composition of microRNAs (miRNAs) in uterine fluid (UF) of women with recurrent implantation failure (RIF) different from that of healthy fertile women?\\nSUMMARY ANSWER The composition of miRNAs in UF of women with RIF is different from that of healthy fertile women and the dysregulated miRNAs are associated with impaired endometrial receptivity and embryo implantation.\\nWHAT IS KNOWN ALREADY It has previously been demonstrated that the miRNAs secreted from endometrial cells into the UF contribute to the achievement of endometrial receptivity. Endometrial miRNAs are dysregulated in women with RIF.\\nSTUDY DESIGN, SIZE, DURATION In this descriptive laboratory case–control study, miRNA abundancy was compared between UF collected during implantation phase from healthy fertile women (n = 17) and women with RIF (n = 34), which was defined as three failed IVF cycles with high-quality embryos.\\nPARTICIPANTS/MATERIALS, SETTING, METHODS Recruitment of study subjects and sampling of UF were performed at two university clinics in Stockholm, Sweden and Tartu, Estonia. The study participants monitored their menstrual cycles using an LH test kit. The UF samples were collected on Day LH + 7–9 by flushing with saline. Samples were processed for small RNA sequencing and mapped for miRNAs. The differential abundance of miRNAs in UF was compared between the two groups using differential expression analysis (DESeq2). Further downstream analyses, including miRNA target gene prediction (miRTarBase), Kyoto Encyclopedia of Genes and Genomes (KEGG) pathway analysis (g:Profiler) and external validation using relevant published data, were performed on the dysregulated miRNAs. Two miRNAs were technically validated with quantitative real-time PCR (RT-PCR).\\nMAIN RESULTS AND THE ROLE OF CHANCE After processing of the sequencing data, there were 15 samples in the healthy fertile group and 33 samples in the RIF group. We found 61 differentially abundant UF miRNAs (34 upregulated and 27 downregulated) in RIF compared to healthy women with a false discovery rate of <0.05 and a fold change (FC) of ≤−2 or ≥2. When analyzed with published literature, we found that several of the differentially abundant miRNAs are expressed in endometrial epithelial cells and have been reported in endometrial extracellular vesicles and in association with endometrial receptivity and RIF. Their predicted target genes were further expressed both in the trophectodermal cells of blastocyst-stage embryos and endometrial mid-secretory epithelial cells, as assessed by publicly available single-cell transcriptome-sequencing studies. Pathway analysis further revealed that 25 pathways, having key roles in endometrial receptivity and implantation, were significantly enriched. Hsa-miR-486-5p (FC −20.32; P-value = 0.004) and hsa-miR-92b-3p (FC −9.72; P-value = 0.004) were successfully technically validated with RT-PCR.\\nLARGE SCALE DATA The data are available in Gene Expression Omnibus (GEO) at https://www.ncbi.nlm.nih.gov/geo/ with GEO accession number: GSE173289.\\nLIMITATIONS, REASONS FOR CAUTION This is a descriptive study with a limited number of study participants. Moreover, the identified differentially abundant miRNAs should be validated in a larger study cohort, and the predicted miRNA target genes and enriched pathways in RIF need to be confirmed and further explored in vitro.\\nWIDER IMPLICATIONS OF THE FINDINGS RIF is a major challenge in the current IVF setting with no diagnostic markers nor effective treatment options at hand. For the first time, total miRNAs have been extensively mapped in receptive phase UF of both healthy women with proven fertility and women diagnosed with RIF. Our observations shed further light on the molecular mechanisms behind RIF, with possible implications in future biomarker and clinical treatment studies.\\nSTUDY FUNDING/COMPETING INTEREST(S) This work was financially supported by the Swedish Research Council (2017-00932), a joint grant from Region Stockholm and Karolinska Institutet (ALF Medicine 2020, FoUI-954072), Estonian Research Council (PRG1076), Horizon 2020 innovation (ERIN, EU952516) and European Commission and Enterprise Estonia (EU48695). The authors have no competing interests to declare for the current study.',\n",
       "  'Introduction': 'Embryo implantation is a delicate and tightly regulated process. It can only occur during a very short time frame in the luteal phase of the menstrual cycle known as the window of implantation (WOI). The WOI opens around 6 days after the LH surge (Day LH + 6) and closes around LH + 10 (Harper, 1992).\\nMuch research has been put into understanding the molecular specifics of the WOI. Most studies have focused on the tissue and cellular level of the endometrium while a perhaps less explored area, but one that is gaining in attention, is the uterine cavity microenvironment and the composition of the uterine fluid (UF). Tubal and peritoneal fluid as well as plasma exudate and endometrial secretions all contribute to the UF mix and molecular components include lipids, carbohydrates, amino acids, prostaglandins and proteins (Berlanga et al., 2011), with marked changes in composition during the WOI (Gemzell-Danielsson and Hamberg, 1994). Nucleic acids have also been identified in UF, and more specifically, in microRNA (miRNA)-containing extracellular vesicles (EVs), suggesting a role for miRNAs in embryo-endometrial communication (Ng et al., 2013; Li et al., 2020). miRNAs are important short regulatory RNA molecules that negatively affect gene expression by either translational repression or degradation of mRNA (Krol et al., 2010). All in all, UF constitutes the optimal microenvironment to facilitate implantation and is an important factor to understand in the WOI context.\\nIn the ART setting, women with recurrent implantation failure (RIF) present a special challenge to accurately diagnose and treat. Defined by three failed implantations despite good-quality embryos, RIF forms an umbrella term that might include several different underlying conditions and pathologies (Simon and Laufer, 2012). Efforts have been made to understand the molecular mechanisms in RIF. The endometrial receptivity array (ERA) was developed based on bulk endometrial tissue gene expression of mRNAs and uses endometrial biopsies to pinpoint the individual WOI in women for personalized embryo transfer (pET) in IVF treatment (Díaz-Gimeno et al., 2011). The ERA test can thus detect a ‘non-receptive’ or ‘out-of-phase’ endometrium in RIF patients and provide a treatment option with pET (Ruiz-Alonso et al., 2013). Using biopsy material and modern transcriptomics, it has also been shown that miRNAs are dysregulated in the endometrium in women with RIF (Rekker et al., 2018). The use of biopsies has however its disadvantages and there is a need for a less invasive method to sample the endometrium. UF has been suggested as a possibility to replace the need for biopsies. Recent protein analysis of UF has indeed shown that the UF composition differs in RIF patients and can be used to assess endometrial receptivity in the IVF setting (Kasvandik et al., 2020).\\nIn the present study, we set out to investigate the composition of total miRNAs in receptive phase UF from healthy fertile women as well as from RIF patients. We utilized a high throughput small RNA sequencing technique developed for detection of miRNAs in minute concentrations (Hagemann-Jensen et al., 2018). Our aim was to draw a detailed map of UF miRNAs that are involved in the implantation process and may be affected in women with disturbed implantation.',\n",
       "  'Methods': 'Ethics\\nThis study was approved by the regional ethics committees at Karolinska Institutet (approval no. 2016/795-31/4) and University of Tartu (276/M-15). Study subjects were included only after written informed consent was obtained.\\nStudy population\\nTwo study groups were included: healthy fertile women and women diagnosed with RIF. UF samples were collected from 17 healthy fertile women during the time period of March 2017 to September 2019 at the Karolinska University Hospital, Stockholm, Sweden. All the women were recruited as healthy study volunteers through advertisements on university campus, on social media and the website studentkaninen.se. The study participants met the following inclusion criteria: age 18–40 years, proven fertile (at least one spontaneous pregnancy that had not ended in a miscarriage), regular menstrual cycles (21–35 days) and no use of hormonal or intrauterine contraception for at least three months prior to sample collection. Exclusion criteria included known systemic, endocrinological or gynecological diseases, infertility, regular use or use prior to sample collection, of non-steroidal anti-inflammatory drugs, ongoing pregnancy or breast-feeding, ectopic pregnancy and more than two miscarriages.\\nUF samples were collected from 34 women diagnosed with RIF during the time period of April 2013 to June 2015 at Nova Vita Clinic, Tallinn, Estonia. RIF was defined as at least three failed IVF/ICSI cycles despite using high-quality embryos according to standard clinical routine. The following inclusion criteria had to be full-filled: minimum three failed IVF/ICSI cycles after either fresh or frozen embryo transfers, age 18–42 and no use of hormonal preparations for 3 months prior to sample collection and an endometrial thickness of minimum 7 mm at the time of sample collection. Women with congenital uterine abnormalities, known systemic or endocrinological comorbidities and/or ongoing pregnancy were excluded from the study.\\nAll women at both study sites were checked for absence of Chlamydia trachomatis, Mycoplasma genitalium and Neisseria gonorrhoeae vaginally and in urine, and for hepatitis B and HIV in the blood.\\nCollection of uterine fluid\\nStudy participants were instructed to monitor LH levels in urine using a self-test kit (Clearblue, SPD Swiss Precision Diagnostics GmbH, Switzerland or BabyTime® hLH urine cassette, Pharmanova, Beit Shemesh, Israel) and to contact the study coordinator at the local research clinic when the LH surge was detected (Day LH + 0). Samples were collected on Day LH + 7 to 9 from both healthy fertile women and from RIF patients. UF samples were collected vaginally using a neonatal feeding tube, 8 French (Vygon, France) or an intrauterine insemination catheter (Cooper Surgical, Trumbull, CT, USA) depending on availability, placed close to the inner opening of the cervical canal. Then 1–2 ml of sterile clinical grade saline was flushed into the uterine cavity using a syringe, incubated for 30 s and then aspirated and collected in a 1.5 ml Eppendorf tube. Samples were centrifuged at 2000 rpm/400× g for 5 min at 4°C to remove cells and debris and then the supernatant was transferred to a new tube and stored at −80°C until further processing.\\nRNA extraction\\nTotal RNA was extracted from 50 µl of UF mixed with 300 µl of Guanidium-based RLT plus buffer (Qiagen, Hilden, Germany), washed twice with ethanol, re-suspended in RNase free water and stored at −80°C until further processing. RNA quantity was checked using the Qubit RNA High Sensitivity Assay Kit (Invitrogen, Oregon, USA) on a Qubit 4 Fluorometer (Invitrogen, Singapore).\\nSmall RNA library preparation and next-generation sequencing\\nSmall RNA library preparation for sequencing was adapted from Hagemann-Jensen et al. (2018). cDNA libraries were prepared on a 96-well-plate using 3 µl of input RNA from each sample. A thermal cycler from BIOER Life Touch (Techtum, China) was used in all steps. After the second PCR, the cDNA libraries were purified individually with AMPure XP beads (Beckman Coulter Inc., USA) in a 1:1 bead to sample ratio. The quality of the purified cDNA product was checked on a High Sensitivity DNA chip (Agilent Technologies, USA) using an Agilent 2100 Bioanalyzer System (Agilent Technologies, USA) and the quantity was measured with a Qubit 4 Fluorometer (Invitrogen, Singapore) using the Qubit 1X dsDNA High Sensitivity Kit (Invitrogen, Oregon, USA). Samples were indexed using customized barcodes (IDT Technologies, USA). From each library, 5 ng of cDNA was pooled and small RNA sequencing was performed on an Illumina NextSeq 550 platform at the SciLifeLab, National Genomics Infrastructure, Karolinska Institutet.\\nTechnical validation of microRNAs with real-time PCR\\nTechnical validation using quantitative real-time PCR (RT-PCR) was done in the same cohort of samples used for sequencing. A couple of miRNAs were selected based on published literature, novelty and availability of TaqMan primers. First, 5 ng of RNA was reverse transcribed to cDNA using the TaqMan Advanced microRNA cDNA synthesis Kit (Applied Biosystems, USA), and this was followed by RT-PCR using TaqMan advanced microRNA assays (Applied Biosystems, USA) on a One Step Plus Real-time PCR system (Applied Biosystems, Singapore) according to the manufactures protocol. Each sample was run in triplicates to allow for assessment of technical variability. Fold change (FC) was calculated using the comparative Ct-method. An endogenous control miRNA was selected by performing RT-PCR for 30 commonly expressed miRNAs using TaqMan™ Advanced miRNA Human Endogenous Controls 96-well Plate (Thermo Fischer, Catalogue No: A34643). Based on a high and stable abundance in UF, hsa-miR-423-5p was chosen as endogenous control. The abundancy pattern of the reference miRNA in the sequencing data is shown in Supplementary Fig. S1A. One sample in the healthy group was excluded since it was considered as an extreme outlier with CT values >39 for the housekeeping gene in all PCR runs. RT-PCR primer assay IDs are available in Supplementary Table SI.\\nStatistical analysis\\nDifferences in miRNA abundance between the two groups after validation with RT-PCR were analyzed with Mann–Whitney U-test. Box-plot figures of the miRNA abundancy are presented as relative quantification of miRNA abundancy. A CT value <35 was used as a cutoff for sample inclusion in the statistical analysis. A P-value <0.05 was considered statistically significant. Data were analyzed using SPSS version 27 software (IBM).\\nBioinformatic analysis\\nPre-processing\\nThe small RNA sequencing data have been deposited into the Gene Expression Omnibus (GEO) with accession number: GSE173289. The initial quality of the FASTQ files was assessed using the software FastQC (version 0.11.9). All samples passed the initial quality control. The subsequent processing of the data was performed according to a previously published pipeline (Hagemann-Jensen et al., 2018). First, the unique molecular identifier (UMI) sequences were removed from the FASTQ files and appended to their respective headers. The sequencing data were then quality checked for overrepresented sequences and guanine-cytosine and adapter content. The Illumina small RNA 3′-adapter and the two cytosine-adenine bases linked to the UMI were finally removed with Cutadapt (version 1.7.1).\\nRead alignment, annotation and quantification\\nThe trimmed reads were aligned to hg38 using STAR (version 2.7.5b). Samples with an overall mapping percentage <60% were excluded from the analysis. Aligned reads were filtered for a read length of 18–40 nucleotides. Next, PCR amplicons were collapsed based on UMIs. The filtered, aligned molecules were annotated using miRBase (miRNAs), GtRNAdb (transfer RNAs (tRNAs)) and GENCODE transcripts (small nucleolar RNAS (snoRNAs), small nuclear RNAs (snRNAs), Y RNAs and small Cajal body-specific RNA (scaRNA)). PCR duplicates and precursor molecules were removed from the annotated reads. Quantification of the aligned and annotated molecules were performed according to the custom-made published pipeline (Hagemann-Jensen et al., 2018).\\nDifferential expression analysis\\nTo identify differentially abundant miRNAs in our data, we used the Bioconductor software DESeq2 (1.30.0). Differentially abundant miRNAs with a false discovery rate (FDR) <0.05 and a FC of ≤−2 or ≥2 were considered statistically significant.\\nmicroRNA target gene prediction and functional enrichment analyses\\nTarget genes were retrieved for the miRNAs with an FDR <0.05 and an FC ≤−2 or ≥2 from miRTarBase (Release 8.0) (Chou et al., 2018). MiRTarBase is a database with experimentally validated miRNA target genes. Only target genes with strong experimental evidence were included in the downstream analysis. Biological Kyoto Encyclopedia of Genes and Genomes (KEGG) pathways and Gene ontology molecular functions and cellular components enriched for the predicted miRNA target genes were identified using the web tool g:Profiler (version e101_eg48_p14_baf17f0) (Raudvere et al., 2019) by utilizing the default g:SCS algorithm. Protein–protein interaction (PPI) clusters were also predicted based on predicted miRNA target genes using the Metascape MCODE enrichment analysis (version v3.5.20211101) (Zhou et al., 2019). An adjusted P-value of <0.05 was considered statistically significant.\\nValidation in external datasets\\nDysregulated miRNAs were compared to relevant previously published endometrial miRNA data (Ng et al., 2013; Logan et al., 2018; Rekker et al., 2018). Target genes of dysregulated miRNAs were matched with genes expressed in trophectoderm of Days 5–7 human blastocyst-stage embryos reported in the single-cell dataset published by Petropoulos et al. (2016). 10× Genomics single-cell sequencing data from endometrial tissue published by Wang et al. (2020) was downloaded from https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE111976. R (version 3.6.1) and the Seurat package (version 3.1.1) were used in R-Studio (version 1.2.1335) to subset mid-secretory cells and find marker genes expressed in epithelial cells.',\n",
       "  'Results': 'Small RNA molecules in uterine fluid\\nTo explore the content of small RNAs in UF from healthy fertile women and women with RIF, we utilized next-generation small RNA sequencing and subsequent bioinformatic analyses (Fig. 1). Two samples, one in each group, did not meet the alignment criterion and one sample in the healthy group did not pass the read-length filter and were subsequently excluded. In total, 15 healthy and 33 RIF samples remained after preprocessing of the data (Fig. 1). After initial filtering and annotation, we identified 1155 miRNAs, 706 tRNA fragments, 538 Y RNA-derived small RNAs, 465 snRNA-derived small RNAs, 393 snoRNA-derived small RNAs and 24 scaRNA-derived small RNAs (Fig. 2A). Hierarchical clustering of samples was performed based on normalized miRNA counts and showed clustering of samples according to group (Supplementary Fig. S1B). The clinical characteristics for the groups are shown in Table I.\\nFigure 1.\\nOpen in a new tab\\nOverview of the study design.\\nBP, biological processes; CC, cellular component; DE, differential expression; MF, molecular functions; miRNA, microRNA; nt, nucleotides; PPI, protein-protein interactions; RIF, recurrent implantation failure; UF, uterine fluid.\\nFigure 2.\\nOpen in a new tab\\nOverview of the small RNA sequencing data.\\n(A) Pie chart showing the distribution, in percentage, of small RNA types in uterine fluid from healthy fertile women and women with RIF. (B) Volcano plot of upregulated (red) and downregulated (blue) miRNAs in uterine fluid from women with RIF. (C and D) Heatmaps based on z-scores showing the abundancy pattern of the upregulated (C) and downregulated (D) miRNAs. Each column in the heatmap represents a sample. The following cutoffs for significance were used for the differentially abundant miRNAs presented in B–D: FDR <0.05 and FC ≤−2 or ≥2. FC, fold change; FDR, false discovery rate; miRNA, microRNA; NS, not significant; RIF, recurrent implantation failure; scaRNA, small Cajal body-specific RNA; snoRNA, small nucleolar RNA; snRNA, small nuclear RNA; tRNA, transfer RNA; UF, uterine fluid.\\nTable I.\\nClinical characteristics of the healthy fertile and RIF group.\\nParameter\\nHealthy fertile (n = 15)\\nRIF (n = 33)\\nAge (years)\\nmedian (range)\\n32 (22–39)\\n35 (26–42)\\nBMI (kg/m2)\\nmedian (range)\\n24 (20–33)\\n23 (18–39)\\nMenstrual cycle length (days)\\nmedian (range)\\n28 (26–30)\\n28 (24–32)\\nNumber of pregnancies\\nmedian (range)\\n1 (1–4)\\n1 (0–4)\\nLive birth (≥1)\\n%\\n58a\\n18\\nMiscarriage (≥1)\\n%\\n1b\\n36\\nEctopic pregnancy (≥1)\\n%\\n0\\n21\\nAbortion (≥1)\\n%\\n55b\\n12\\nNumber of fresh embryo transfers\\nmedian (range)\\n–\\n3 (0–10)\\nNumber of frozen embryo transfers\\nmedian (range)\\n–\\n1 (0–4)\\nNumber of IVF failures\\nmedian (range)\\n–\\n4 (3–10)\\nDuration of infertility (years)\\nmedian (range)\\n–\\n6 (2–14)\\nOpen in a new tab\\nValues are presented as median and range or percentage.\\nn, number; RIF, recurrent implantation failure.\\naThree missing values.\\nbFour missing values.\\nDifferentially abundant UF miRNAs in women with RIF\\nTo further investigate the role of miRNAs in RIF, the differential abundance of miRNAs in UF between the two groups was analyzed using DESeq2. We identified 61 miRNAs that were dysregulated in women with RIF compared to healthy fertile women (FDR < 0.05 and FC ≤−2 or ≥2) (Table II and Fig. 2B–D). Out of these miRNAs, 34 were upregulated and 27 downregulated in RIF. The downregulated miRNAs showed a more stable and uniform abundancy pattern (Fig. 2C and D).\\nTable II.\\nDifferentially abundant miRNAs in receptive phase uterine fluid from women with recurrent implantation failure compared to healthy fertile women (FC ≤−2 or ≥2 and FDR < 0.05).\\nUpregulated miRNA\\nDownregulated miRNA\\nmiRNA ID\\nFC\\nFDR\\nmiRNA ID\\nFC\\nFDR\\nhsa-miR-6821-5p\\n17.7\\n<0.001\\nhsa-miR-96-5p\\n−4.2\\n0.008\\nhsa-miR-4521\\n11.0\\n<0.001\\nhsa-miR-186-5p\\n−4.1\\n<0.001\\nhsa-miR-483-5p\\n10.9\\n<0.001\\nhsa-miR-628-3p\\n−3.7\\n0.004\\nhsa-miR-4421\\n9.7\\n0.001\\nhsa-miR-183-5p\\n−3.6\\n<0.001\\nhsa-miR-1268b\\n6.9\\n0.002\\nhsa-miR-126-5p\\n−3.4\\n0.003\\nhsa-miR-3130-3p\\n6.6\\n0.005\\nhsa-miR-451a\\n−3.4\\n0.003\\nhsa-miR-127-5p\\n6.2\\n<0.001\\nhsa-miR-374a-5p\\n−3.4\\n<0.001\\nhsa-miR-671-5p\\n6.1\\n<0.001\\nhsa-miR-224-5p\\n−3.4\\n<0.001\\nhsa-miR-504-5p\\n5.7\\n0.046\\nhsa-miR-203a-3p\\n−3.2\\n0.002\\nhsa-miR-6724-5p\\n5.4\\n0.030\\nhsa-miR-361-5p\\n−3.2\\n<0.001\\nhsa-miR-4731-5p\\n4.7\\n0.024\\nhsa-miR-877-5p\\n−3.2\\n0.002\\nhsa-miR-378c\\n4.0\\n0.019\\nhsa-miR-363-3p\\n−3.0\\n0.002\\nhsa-miR-193b-3p\\n3.7\\n0.003\\nhsa-miR-15b-5p\\n−3.0\\n0.006\\nhsa-miR-23a-5p\\n3.7\\n<0.001\\nhsa-miR-486-5p\\n−2.9\\n<0.001\\nhsa-miR-3614-5p\\n3.6\\n0.007\\nhsa-miR-92b-3p\\n−2.9\\n<0.001\\nhsa-miR-1247-3p\\n3.6\\n0.016\\nhsa-miR-10a-3p\\n−2.8\\n0.025\\nhsa-miR-769-5p\\n3.6\\n<0.001\\nhsa-miR-200b-5p\\n−2.6\\n0.002\\nhsa-miR-143-5p\\n3.4\\n0.016\\nhsa-miR-192-5p\\n−2.5\\n0.002\\nhsa-miR-370-3p\\n3.2\\n<0.001\\nhsa-miR-151b\\n−2.4\\n<0.001\\nhsa-miR-1247-5p\\n3.2\\n0.017\\nhsa-miR-101-3p\\n−2.4\\n0.003\\nhsa-miR-1307-3p\\n2.9\\n0.004\\nhsa-miR-142-5p\\n−2.4\\n0.003\\nhsa-miR-127-3p\\n2.9\\n<0.001\\nhsa-miR-450b-5p\\n−2.4\\n0.013\\nhsa-miR-550a-5p\\n2.9\\n0.002\\nhsa-miR-503-5p\\n−2.3\\n0.013\\nhsa-miR-25-5p\\n2.9\\n0.009\\nhsa-miR-342-3p\\n−2.2\\n0.008\\nhsa-miR-27b-5p\\n2.7\\n0.016\\nhsa-miR-30c-5p\\n−2.1\\n0.002\\nhsa-miR-491-5p\\n2.7\\n0.043\\nhsa-miR-10b-3p\\n−2.1\\n0.008\\nhsa-miR-574-5p\\n2.6\\n<0.001\\nhsa-miR-182-5p\\n−2.0\\n0.001\\nhsa-miR-769-3p\\n2.6\\n0.012\\nhsa-miR-193a-5p\\n2.4\\n0.001\\nhsa-miR-21-3p\\n2.3\\n0.030\\nhsa-miR-221-3p\\n2.2\\n0.046\\nhsa-miR-125a-3p\\n2.1\\n0.005\\nhsa-miR-744-5p\\n2.1\\n0.029\\nhsa-miR-320a\\n2.1\\n0.002\\nOpen in a new tab\\nSmall RNA sequencing was used to detect miRNAs in uterine fluid. Differential expression analysis was performed with DESeq2.\\nFC, fold change; FDR, false discovery rate; miRNAs, microRNAs.\\nTarget genes of differentially abundant miRNAs in RIF\\nAs miRNAs function by targeting mRNA transcripts and negatively regulating their expression (Carthew and Sontheimer, 2009), we explored the target genes for the dysregulated miRNAs in UF from RIF patients. We utilized the web-based tool miRTarBase (Chou et al., 2018) and filtered for experimentally validated targets with strong evidence. This yielded a list with 108 potential target genes for our upregulated miRNAs and 357 potential target genes for our downregulated miRNAs (Supplementary Table SII).\\nBiological processes and pathways critical for endometrial receptivity and implantation are enriched for the identified miRNA target genes\\nIn order to put the target genes of the dysregulated miRNAs in a biological context, we utilized the web-based tool g:Profiler (Raudvere et al., 2019) to perform functional enrichment analysis on our target gene lists. When looking at enriched KEGG (Kanehisa and Goto, 2000) pathways, we found 69 significantly enriched pathways for the target genes of upregulated miRNAs and 110 significantly enriched pathways for the target genes of downregulated miRNAs (Supplementary Table SIII). Among the significantly enriched pathways, we could identify 25 unique pathways that have been previously reported to be involved in endometrial receptivity and implantation (Fig. 3A and B, references listed in Supplementary Table SIV). The top 10 statistically most significant pathways out of those associated with endometrial receptivity and implantation and their associated target genes overlapping between the pathways for the up- and downregulated miRNAs, are shown in Fig. 3C and 3D, respectively.\\nFigure 3.\\nOpen in a new tab\\nBiological pathways related to endometrial receptivity and implantation.\\n(A and B) Bar charts showing the number of target genes per pathway for all significant KEGG pathways (adjusted P-value < 0.05) related to endometrial receptivity and/or implantation for up- (A) and downregulated (B) UF miRNAs in RIF. (C and D) The top 10 most significant KEGG pathways (adjusted P-value < 0.05) related to endometrial receptivity and/or implantation and their associated target genes for up- (C) and downregulated (D) UF miRNAs in RIF. Each target gene presented in the figure was present in a minimum of three pathways for the up and downregulated miRNAs respectively. miRNA, microRNA; RIF, recurrent implantation failure; UF, uterine fluid.\\nTarget genes enriched in the top 10 pathways associated with RIF and implantation for up- and downregulated miRNAs (presented in Fig. 3C and D) were further analyzed for gene ontology molecular functions, cellular components and biological processes as well as Metascape MCODE PPI clusters. Protein, enzyme and transcription factor binding were among the most significant molecular functions predicted to be related to the target gene products of both up- and downregulated miRNAs (Supplementary Fig. S2A and B). Examples of the most significant cellular components in which the target gene product may exert its function were cytoplasm, membrane microdomain and membrane raft for the upregulated miRNA target genes (Supplementary Fig. S3A) and transferase complex, protein kinase complex and cytosol for the downregulated miRNA target genes (Supplementary Fig. S3B). Some of the enriched biological processes were negative regulation of developmental processes and regulation of cell population proliferation for the upregulated miRNAs (Supplementary Fig. S4A) and phosphorylation and regulation of cell death for the downregulated miRNAs (Supplementary Fig. S4B). PPI network analysis identified three clusters of densely connected protein products of our upregulated miRNA target genes (Supplementary Fig. S5) and four clusters for the downregulated miRNA target genes (Supplementary Fig. S6). Individual PPI cluster scores are presented in Supplementary Table SV.\\nDifferentially abundant UF miRNAs are found in endometrial tissue and extracellular vesicles during the WOI\\nNext, we wanted to explore our results in the endometrial context. Logan et al. (2018) have profiled miRNAs in endometrial epithelial cells of healthy women using microarray technology. Out of 61 of our dysregulated miRNAs, 55 overlapped with miRNAs reported to be expressed in endometrial epithelial cells of that study (Fig. 4A, Supplementary Table SVI) (Logan et al., 2018).\\nNg et al. (2013) studied EVs (exosomes and microvesicles) in UF and the miRNA content of EVs derived from the endometrial epithelial cell-line (ECC-1) in vitro. We found that 19 of our dysregulated miRNAs were among the 229 miRNAs present in ECC-1 EVs reported by Ng et al. (2013) (Fig. 4B), suggesting that our results could reflect a dysregulation of the endometrial epithelial secretome.\\nFigure 4.\\nOpen in a new tab\\nExternal validation of dysregulated miRNAs.\\n(A–D) Venn diagrams showing overlap with our dysregulated UF miRNAs in RIF and microRNAs that are: (A) expressed in endometrial epithelial cells (Logan et al., 2018), (B) present in extracellular vesicles from endometrial epithelial cells (Ng et al., 2013), (C and D) differentially expressed in the endometrium of (C) women with RIF and (D) healthy fertile women during the WOI (Rekker et al., 2018). EV, extracellular vesicle; miRNA, microRNA; RIF, recurrent implantation failure; UF, uterine fluid; WOI, window of implantation.\\nRekker et al. (2018) studied miRNA expression in endometrial tissue at the time of the WOI in RIF patients and reported 21 dysregulated miRNAs compared to healthy women. Three out of these were found similarly dysregulated in our UF-derived miRNAs (Fig. 4C). In the same study, they also reported differentially expressed miRNAs in endometrial tissue in healthy women between the pre-receptive phase and the WOI, thereby describing normal miRNA regulation during implantation. When we compared our up- and downregulated miRNAs to the ones reported by Rekker et al. (2018), we found nine miRNAs that were present in both datasets (Fig. 4D). We also noted that some had opposite expressional direction between the two studies, which further supports dysregulation of these miRNAs during the WOI in RIF.\\nDifferentially abundant miRNAs’ targets exist in both endometrial and embryonic cells\\nUF miRNAs have the potential to act on both the endometrium and on the embryo. To investigate the possible location of our targets of the dysregulated miRNAs, we searched for the presence of our target genes in relevant cell types utilizing publicly available single-cell transcriptome sequencing data.\\nThe extensive mapping of human endometrial cell types by Wang et al. (2020) was used to look at the specific gene expression of endometrial epithelial cells during the WOI. We found that several of the target genes for the dysregulated miRNAs in RIF were active in human endometrial epithelial cells during the WOI, indicating that our dysregulated miRNAs may target genes in the epithelial cells of the endometrium during implantation (Fig. 5A).\\nFigure 5.\\nOpen in a new tab\\nExternal validation of predicted target genes of dysregulated miRNAs.\\n(A and B) Venn diagrams showing overlap with the predicted target genes of our dysregulated miRNAs and (A) genes expressed in the endometrial epithelial cells (Wang et al., 2020) and (B) lineage-specific genes in Days 5–7 trophectoderm (Petropoulos et al., 2016) during the WOI. miRNA, microRNA; RIF, recurrent implantation failure; UF, uterine fluid.\\nUsing the comprehensive single-cell RNA sequencing study on human embryos by Petropoulos et al. (2016), we looked at the specific gene expression (P-value < 0.05) in Days 5–7 trophoblasts of blastocyst-stage embryos and compared to our miRNAs target genes (Fig. 5B). We found that some of the target genes were indeed active in peri-implantation trophoblasts, suggesting that they may be affected by the dysregulation of miRNAs in RIF.\\nSelected UF miRNAs validated by qPCR\\nTo technically validate our results of the small RNA sequencing data, the relative abundance of two dysregulated miRNAs (hsa-miR-486-5p and hsa-miR-92b-3p) were examined with RT-PCR. We chose to validate one miRNA, hsa-miR-486-5p, that has been previously linked to implantation failure (Yang et al., 2018), and one miRNA, hsa-miR-92b-3p, which has not been studied in the context of RIF. In addition, both miRNAs were among the most abundant miRNAs and showed a stable abundancy pattern in both groups (Fig. 2C and D). In line with the sequencing results, both miRNAs were downregulated in the RIF group with a FC of −20.32 (P-value = 0.004) for hsa-miR-486-5p and a FC of −9.72 (P-value = 0.004) for hsa-miR-92b-3p (Supplementary Fig. S7, Fig. 2D).',\n",
       "  'Discussion': 'RIF is a major challenge in the current IVF setting with no diagnostic markers nor effective treatment options at hand. For the first time, total miRNAs have been extensively mapped in receptive phase UF of both healthy women with proven fertility and women diagnosed with RIF. We show the presence of several dysregulated miRNAs in the UF of women with RIF and validate their significance and potential role in implantation using external datasets and pathway enrichment analysis.\\nA receptive endometrium is a pre-requisite for a successful implantation and poor endometrial receptivity has been suggested as a putative cause of RIF (Ruiz-Alonso et al., 2013). In a recent study, Li et al. (2020) compared the abundancy of miRNAs specifically in EVs from UF collected from healthy women in natural cycles and from women undergoing IVF after ovarian stimulation. Several of our miRNAs overlap with the reported dysregulated miRNAs in receptive phase of natural and ovarian stimulation cycles of that study (Li et al., 2020). Furthermore, hsa-miR-320a was reported as downregulated in women with a successful IVF pregnancy, which is in line with our results where this miRNA was reported as upregulated in RIF (Li et al., 2020). In another study on UF and endometrial receptivity, Vilella et al. (2015) reported 27 WOI-specific miRNAs in UF of healthy women utilizing microarray technology. Two of our dysregulated miRNAs, miR-183-5p and miR-320a, overlapped with the results of that study, implicating a role for them in endometrial receptivity (Vilella et al., 2015). Indeed, downregulation of miR-183-5p, along with miR-182-5p and miR-96-5p, as we show in RIF in our study, have been linked to poor endometrial receptivity in a functional in vitro study by Akbar et al. (2020). Notably, they demonstrated that reduced expression of miR-183-5p both in vitro in a human endometrial cell line and in vivo in the endometrium of mice reduced embryo attachment whereas upregulation promoted it, suggesting that downregulation of miR-183-5p could potentially contribute to poor endometrial receptivity and thus implantation failure in women with RIF (Akbar et al., 2020).\\nFurther supporting a role of our dysregulated miRNAs in RIF, external validation of our findings with a study by Rekker et al. (2018) demonstrated an overlap with differentially expressed miRNAs in the endometrial tissue of women with RIF. Interestingly, two of the overlapping miRNAs, miR-224-5p and miR-127-5p, have previously been shown to target genes of the implantation associated pathway PI3K/AKT, which was one of the most significantly enriched pathways in our study (Grewal et al., 2008; Fabi et al., 2017; Kuang et al., 2019; Li et al., 2019; Qiao et al., 2020). In addition, aberrant expression of miR-127-3p (sharing the same precursor miRNA as miR-127-5p) that was upregulated in our study as well as one of our downregulated miRNAs that we also technically validated, miR-486-5p, have been associated with poor embryo-transfer outcome and recurrent miscarriage in a study by Yang et al. (2018). BCL6, a well-known endometrial receptivity gene, is also one of the target genes of miR-127-3p, which further support a role of this miRNA in endometrial receptivity and subsequently in RIF (Chen et al., 2013).\\nIn line with previous published literature outlined above, we show that the predicted target genes of our dysregulated miRNAs are significantly enriched in numerous pathways having key roles in implantation, including pathways associated with endometrial receptivity such as JAK/STAT, vascular endothelial growth factor (VEGF) and estrogen-mediated signaling (Young, 2013; Rosario and Stewart, 2016; Guo et al., 2021). These results provide clues about the biological contexts of the dysregulated miRNAs in RIF and indeed, similar to our study, VEGF, P53, JAK/STAT, HIF-1, IL-17 and TGF-beta signaling have previously been associated with RIF (Rekker et al., 2018; Shi et al., 2018; Wang et al., 2019; Yu et al., 2019). Further studies are however needed to investigate how the pathways are affected. We also show that target genes of our miRNAs overlap with genes expressed in the endometrial mid-secretory epithelium (Wang et al., 2020). Hence, we speculate that miRNAs in UF could potentially regulate genes crucial for endometrial receptivity through paracrine signaling. In RIF, the implications of this could be impaired endometrial receptivity and implantation, for example, by deregulation of VEGFA (Guo et al., 2021), which is a predicted target gene for several of the dysregulated UF miRNAs.\\nEmbryo endometrial cross-talk is crucial for a successful implantation. It has previously been reported that miRNAs secreted from the endometrium in EVs can be taken up by the embryo and subsequently affect embryo adhesion during implantation (Vilella et al., 2015). Furthermore, Liu et al. (2020) recently demonstrated that EVs specifically secreted from endometrial cells of RIF patients negatively affects growth and invasion of the embryo during implantation. The overlap of our data with miRNAs identified in EVs secreted from ECC-1 cells reported by Ng et al. (2013), suggest that some of the miRNAs identified in our study could be transported in EVs from the endometrium to the embryo. In line with this observation, we showed that some of the predicted target genes of our UF miRNAs are indeed expressed in the blastocyst, including CDH1, encoding the cell-to-cell adhesion protein E-cadherin forming adherens junctions between cells (Petropoulos et al., 2016). E-cadherin is known to be involved in embryo adhesion during implantation and it also indirectly affects Wnt-signaling, one of the important pathways regulating embryo-endometrium crosstalk (Shih et al., 2002; Nelson and Nusse, 2004; Chen et al., 2009; Kokkinos et al., 2010). Indeed, Wnt-signaling and adherens junctions were two of the significantly enriched pathways in our study. Accordingly, our data together with previously published literature, collectively implies that an abnormal content of miRNAs in UF of women with RIF may not only affect endometrial receptivity but also crosstalk between the embryo and the endometrium during implantation. Aside from EVs, it is likely that some of the identified miRNAs may exist in UF as part of apoptotic bodies, free or bound to proteins that originate from endometrial secretions, plasma exudates or peritoneal fluid, which all contribute to the mix of components in UF (Berlanga et al., 2011; Bhusane et al., 2016).\\nOur results present an explorative overview of differentially abundant miRNAs in RIF and suggest possible target genes and cells of these miRNAs and thus lays out direction for future investigations into the functional role of miRNAs in RIF. Further in vitro studies are needed to validate the predicted target genes and to understand how identified miRNAs might act on cells to affect the implantation process. Another key aspect to further address is what possible mechanisms underly the dysregulation of miRNAs in RIF UF. Possible explanations that require further investigation range from genetic to epigenetic mechanisms to defects in exocytosis or exosomal pathways.\\nOur findings also constitute a pilot investigation for the use of UF miRNAs as non-invasive biomarkers for the RIF condition. Thus, they could in the future potentially assist in clinical decisions and timing of embryo transfer if the risk of implantation failure is predicted as high. We show here that robust and biologically relevant differences in miRNA abundancy, some of which have not been reported in this context before, can be detected between healthy and pathological samples of UF that was collected with a minimally invasive procedure. Of course, the possible implications of this need to be further studied in larger clinical settings but it is a promising starting point. As we could see a very stable abundancy pattern in several of the downregulated miRNAs, including the ones chosen for technical validation (hsa-miR-486-5p and hsa-miR-92b-3p), we suggest future investigations looking into these as potential biomarkers. In the clinical setting, a diagnostic marker analyzed by PCR or a panel of markers analyzed by sequencing would be convenient, cheap and readily available tools.\\nSome limitations should be considered when interpreting the data of this study. Recent studies suggest that the endometrium is a limited issue in implantation failure when applying preimplantation testing for aneuploidy (PGT-A) on the embryos (Pirtea et al., 2021, 2021). However, the usefulness of PGT-A is still debatable as some studies have shown no benefit of PGT-A to live-birth rates (Sarkar et al., 2021) or have demonstrated improved live-birth rates only for women over the age of 35 (Simopoulou et al., 2021), suggesting that currently available evidence is insufficient to support PGT-A in routine clinical practice (Madero et al., 2021). Therefore, we used the most common definition of RIF which only includes morphological scoring for embryo quality assessment. Although collection of UF is a minimally invasive procedure, minor amounts of blood contamination may still occur due to accidental scratching of the catheter to the wall of the uterine cavity or cervical canal. However, to avoid this we used the smallest diameter possible of the catheter, placed it close to the inner opening of the cervical canal during UF collection, and only included the supernatant of the UF after centrifugation. In order to avoid contamination of cervical mucus, extra care was taken to not aspirate during the passage through the cervical canal. Furthermore, a batch effect due to different UF collection sites cannot be excluded. We did, however, use the same protocol at both collection sites and carefully normalized the cDNA input concentrations when submitting the cDNA libraries for sequencing, which was performed in the same laboratory for all samples. Finally, although we have applied strict clinical inclusion and exclusion criteria for all the study participants and collected the samples in the same phase of the menstrual cycle in order to make the study population of the two groups as homogenous as possible, the sample size is limited. Furthermore, we used previously published data from endometrial samples, ECC-1 cells in vitro and embryos to validate our results (Ng et al., 2013; Petropoulos et al., 2016; Logan et al., 2018; Rekker et al., 2018; Wang et al., 2020). Although we found several interesting overlaps with our data, the heterogeneity of the bio-specimens and in vivo/in vitro settings should be considered when interpreting the results.',\n",
       "  'Conclusion': 'In conclusion, we show that miRNAs can be extensively mapped in receptive phase UF and that several miRNAs are differentially abundant in UF from women with RIF compared to healthy fertile women. Based on external validation and pathway analysis, we speculate that an aberrant composition of miRNAs in UF may contribute to RIF by affecting endometrial receptivity and/or embryo-endometrial crosstalk during implantation. However, functional in vitro and in vivo studies are needed to establish their role in the pathogenesis of RIF. In addition, the identified UF miRNAs constitute a promising panel of non-invasive biomarkers for RIF that could be further studied in larger clinical trials to evaluate their clinical utility.',\n",
       "  'URL': 'https://doi.org/10.1093/humrep/deac019',\n",
       "  'Journal': 'Hum Reprod'},\n",
       " {'PaperTitle': 'Endometriosis and ovulatory menstruation: beyond the Sampson principle',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Serdar E Bulun',\n",
       "  'Email': 's-bulun@northwestern.edu',\n",
       "  'Abstract': 'Endometriosis is an estrogen-dependent chronic inflammatory syndrome characterized by viable endometrial tissue outside the uterine cavity and associated with pain and infertility. Endometriosis, as tissue or a pathological process, is dynamic in that its establishment and progression require repeated episodes of retrograde travel of shed endometrial tissue, which implants in the lower abdominal cavity following ovulatory cycles and survives. Estrogen-rich follicular fluid released onto peritoneal surfaces during ovulation may also support endometriotic implants. DNA evidence indicates that endometriosis originates from eutopic endometrial tissue, which may reach the abdominal cavity in a retrograde manner primarily via the uterine tubes. Unlike uterine bleeding associated with non-ovulatory circumstances, retrograde menstruation following an ovulation maximizes shedding of epithelial cells localized to deep invaginations of the basalis portion of the endometrium, which likely carry somatic cancer-driver mutations such as KRAS. The attached endometrial stromal cells are mostly mutation free but display epigenetic defects including overexpression of aromatase and estrogen receptor-β and downregulation of progesterone receptor, causing estrogen excess and progesterone resistance. These tissue clones may form implants in involuting ovarian corpus luteum cysts and peritoneal surfaces and induce tissue remodeling and fibrosis, manifested as deep-infiltrating endometriosis. The first-line treatment for chronic pelvic pain associated with endometriosis is suppression of ovulation, with the goal of relieving pain. Infertility is often managed using in vitro fertilization, which improves the embryo quality and alters endometrial development.',\n",
       "  'Introduction': 'This Review focuses on the role of the menstrual event that follows ovulation in the pathophysiology of endometriosis. Endometriosis is a unique inflammatory syndrome resulting from repeated episodes of ovulatory menstrual bleeding in a subset of people who menstruate, a population comprising diverse gender and biological sex identifications. This Review will mainly refer to women because they are highly represented in the clinical literature (1, 2). Endometriosis, as tissue or a pathological process, is not static but dynamic in that repeated episodes of retrograde travel of shed endometrial tissue, which implants and survives in the lower abdominal cavity following ovulatory cycles, are indispensable for its establishment and progression (1, 2). Its primary symptoms are painful menstrual periods, chronic pelvic pain, painful intercourse, and infertility (3–6). The dependence of endometriosis on estrogen to drive the inflammatory process and pain separates it from other chronic inflammatory syndromes such as Crohn’s disease or acid reflux esophagitis (7, 8). Until recently, a prevailing notion that prioritized the surgical description of endometriosis over the disease process unfortunately limited this syndrome to a narrow anatomic definition, i.e., the ectopic presence of endometrium-like tissue outside of the uterine cavity; this definition profoundly stalled progress for developing long-term and patient-centered management strategies (9, 10). This arbitrary weighted and purely anatomic classification system had limited value to predict relief of pain or achievement of pregnancy in response to various regimens of symptom management (11, 12). A recent shift toward developing new paradigms to understand and manage the symptoms of endometriosis through the various stages of the disease process puts patients first (4, 6, 13, 14). This new overall approach recognizes its early manifestations during adolescence, the long history of disease, its recurrent symptoms, and the overall limitations of the currently available medical and surgical approaches (Figure 1).\\nFigure 1. The clinical course of endometriosis over a lifetime.\\nOpen in a new tab\\nThe top row summarizes typical patient experiences beginning in the teenage years and continuing through their 40s and 50s. The lower row lists pain management strategies recommended during each decade of life. Although hysterectomy and bilateral removal of tubes and ovaries are frequently used, they represent severely radical approaches and may not provide pain relief in a subset of patients. COCs, combined oral contraceptives; NSAID, nonsteroidal antiinflammatory drug; GnRH, gonadotropin-releasing hormone.\\nA number of hypotheses have been proposed to explain the presence of endometrial tissue outside the uterine cavity (15). This Review focuses on the mechanism, which is supported by an extraordinarily large body of consistent scientific evidence that indicates endometriosis and its symptoms are caused by retrograde travel of menstrual blood containing molecularly defective endometrial tissue fragments onto the lower abdominal peritoneal surfaces and the ovaries following an ovulation (Figure 2) (16–19). In 1927, John Sampson was the first to provide evidence that this retrograde spread into the pelvic cavity could take place via the uterine tubes or venous circulation (18–20). Also known as Sampson’s theory, this straightforward mechanism was recently demonstrated via mapping of the identical DNA variants or mutations in the epithelial cells of the endometrium to the ectopic endometriotic implants in hundreds of endometriosis patients (21–25). Adenomyosis, a disorder defined as the presence of multiple islands of endometrial tissue in the myometrium, is closely related to endometriosis (26, 27). Adenomyosis results from survival and growth of entrapped fragments of basalis endometrium in the myometrial layer (25, 27). This concept of intracavitary endometrium as the source of pelvic endometriosis and its symptoms has also been emphasized in a number of recent review articles (4, 6, 14).\\nFigure 2. Ovulatory cycle.\\nOpen in a new tab\\nRepetitious ovulatory cycles initiated by hypothalamic gonadotropin-releasing hormone (GnRH) neurons are essential for the establishment and progression of endometriosis (see Figure 3). GnRH, which is secreted in a pulsatile manner, acts on its seven-transmembrane G protein–coupled receptor on the gonadotroph and induces sequential secretion of follicle-stimulating hormone (FSH) and luteinizing hormone (LH) from the pituitary. FSH and LH, via their specific G protein–coupled receptors on ovarian granulosa and theca cells, stimulate sequential secretion of estradiol and progesterone. Estradiol during the proliferative phase builds the functionalis layer of the endometrium, whereas progesterone during the secretory phase stimulates endometrial differentiation. In the absence of pregnancy, endometrial tissue is shed through the vagina and also in a retrograde manner through the uterine tubes into the lower abdominal cavity.\\nThere is also consensus that endometriosis likely affects a woman chronically from adolescence until menopause (Figure 1) (4, 14). The paradigm that dysmenorrhea should be defined as primary or secondary (secondary being associated with pelvic pathology) is unhelpful and implies that imaging or laparoscopy is necessary to make a presumptive diagnosis of endometriosis (28). Labeling severe menstrual cramps as primary dysmenorrhea during adolescence just because the patient has not gone through a workup including laparoscopy assumes, by definition, that dysmenorrhea can be managed without understanding of its etiology if ovulation is suppressed together with regimens of NSAIDs for an undefined period of time (28, 29). This diagnostic approach does not allow the clinician to consider long-term preventative measures for the progress of endometriosis. Moderate to severe dysmenorrhea experienced during adolescence likely represents the initial stages of the long disease process of endometriosis (Figure 1). The modern, patient-centric definition of endometriosis is more clinically helpful, since it is used to describe the symptom complex that a patient experiences throughout their reproductive life (Figure 1) (4, 6, 14). An endometriosis patient may or may not have the surgically demonstrable presence of endometrium-like tissue in their lower abdominal organs, including the ovaries, because microscopic disease may be missed. In approximately 10% of patients endometriosis and its symptoms persist after menopause or even after the surgical removal of their uterus and ovaries (8, 30, 31).\\nEpidemiology, symptoms, and clinical course\\nEndometriosis is the most common cause of painful menses and chronic pelvic pain in women; and it is often associated with infertility (5, 32, 33). Worldwide, approximately 200 million reproductive-age women suffer from endometriosis and experience painful and heavy menstrual bleeding, painful intercourse, and pelvic pain outside of menses (34, 35). More than 100,000 hysterectomies and many more laparoscopies for the diagnosis and conservative removal or ablation of endometriosis are performed annually in the United States (36, 37). Multiple studies showed that the suffering and economic burden associated with endometriosis are very high and resemble those of other chronic inflammatory conditions such as Crohn’s disease and rheumatoid arthritis, except that endometriosis may be more prevalent (38, 39). The analyses used in these studies focus on health care costs and usually severely underestimate productivity loss and decreased quality of life (38, 39). There are additional obstetrical risks associated with endometriosis: a history of endometriosis increases the risk of early pregnancy loss, preterm delivery, neonatal admission to the intensive care unit, and gestational diabetes (40, 41).\\nDiffuse adenomyosis is associated with prolonged menstrual bleeding, painful menses, and recurrent pregnancy loss (26, 27, 42). The presence of adenomyosis together with endometriosis elevates the risk of placenta previa or a cesarean delivery (40).\\nThe hallmark symptoms of endometriosis, i.e., dysmenorrhea, dyspareunia, and chronic pelvic pain, are associated with retrograde travel of menstrual blood and tissue and an inflammatory response (Figure 2) (1, 34, 43–46). Stimulation of the nociceptors on peritoneal surfaces by the menstrual tissue or established ectopic implants is thought to be the major source of pain (47, 48). Nerve-like tissue has been described in the endometrium of patients with endometriosis, raising the question of whether intracavitary endometrium may also contribute to the generation of pain (49–51). Additionally, nociceptive input from the spinal cord to higher brain areas may amplify receptor activation at peripheral tissues (47).\\nIn the majority of patients, endometriosis first manifests as moderate to severe adolescent dysmenorrhea, which either goes untreated or is managed with NSAIDs or combination oral contraceptives (Figure 1) (52). Many of these patients eventually develop persistent pain between their periods and experience painful intercourse (53). Only a fraction of these patients undergo laparoscopy that may reveal pelvic endometriotic implants and facilitate their surgical removal to provide temporary pain relief (Figure 1) (53, 54). Pain recurs in up to 90% of these patients within 2 years despite adjuvant ovarian suppression; this eventually leads to repeated laparoscopic procedures in an attempt at pain control (Figure 1) (54). This population of patients with relapsed symptoms constitutes the largest and the most challenging group among all patients with endometriosis, since after each treatment attempt the recurrent symptoms progressively become harder to manage (54). Many of these patients eventually resort to surgical removal of the uterus and the ovaries (Figure 1) (31). Even these radical approaches do not provide pain relief in at least 10% of patients (Figure 1) (8, 30, 31).\\nAs adjuvant approaches to surgical resection, suppression of ovulation or menses for pain control is attempted by the use of progestin-containing intrauterine devices, combination oral contraceptives, or gonadotropin-releasing hormone (GnRH) analogs before or after surgical procedures (55). Given that the majority of cases of endometriosis occur as a result of retrograde travel of endometrial tissue to the abdominal cavity, it is my opinion that long-term suppression of ovulatory menses should be the mainstay of prevention or therapy. Surprisingly, this approach as the primary treatment of endometriosis is seldom considered, recommended, or used (2, 56).',\n",
       "  'Methods': '',\n",
       "  'Results': 'Role of ovulatory menses in endometriosis\\nOvulatory menstruation per se is essential for the development of symptomatic endometriosis (Figure 2) (2, 14, 56). Only menstruating species develop endometriosis (2, 56–59). Women with conditions precluding spontaneous menstruation, such as Müllerian aplasia or Turner syndrome (caused by loss or partial loss of an X chromosome), do not develop endometriosis, although the presence of a vestigial Müllerian remnant has been published in a rare patient with Turner syndrome (60). Endometriosis and anovulatory disorders such as polycystic ovary syndrome rarely occur together (61, 62).\\nRepetitious menstrual episodes that follow ovulation represent the single most impactful and indispensable factor that initiates the process of endometriosis and worsens its existing symptoms, for the following reasons (Figure 2): First, during an ovulatory cycle, sequential exposure to estradiol for 2 weeks followed by added secretion of progesterone for another 2 weeks maximizes endometrial thickness right before this large mass of tissue is shed (Figure 3A) (63–65). Physiologically developed spiral arterioles and endometrial vessels during this period may lead to rapid vasoconstriction and separation at a plane close to the myometrium, producing large quantities of viable endometrial tissue available for transtubal migration onto the pelvic peritoneal surfaces and ovaries (66). Thus, large quantities of viable endometrial tissue are likely shed after a nonpregnant ovulatory cycle, and some of this tissue together with blood reaches into the lower abdominal cavity via the tubes (Figure 3A). Second, massive quantities of estradiol (as high as 130,000 pmol/L in the follicular fluid of approximately 5 mL) and pain-inducing substances such as prostaglandin E2 are released directly onto the pelvic peritoneal surfaces during an ovulation (67, 68). These strikingly high quantities of estradiol probably contribute to the implantation and growth of viable endometrial fragments on peritoneal or ovarian surfaces. Third, an ovarian endometrioma may be formed in a hemorrhagic corpus luteum cyst that recently released an egg (Figure 2) (69). The steroid hormone–rich cyst cavity serves as a growth medium for endometrial tissue to survive and thrive. Fourth, compared with women having regular predictable periods, the prevalence of endometriosis is lower in anovulatory states such as polycystic ovary syndrome, hypothalamic amenorrhea, or breastfeeding (61, 70). Fifth, early menarche followed by cyclic periods is associated with a higher risk of endometriosis (2, 71). Sixth, during infertility treatment of a patient with endometriosis, back-to-back cycles of ovulation induction followed by intrauterine insemination worsen endometriosis symptoms (72), possibly because these simple but low-fecundity treatments must be repeated multiple times, exposing the patient to multiple ovulatory menstrual cycles. On the other hand, more intense forms of ovulation induction used for in vitro fertilization (IVF) usually do not worsen endometriosis as much as intrauterine induction cycles, possibly because although IVF is associated with very high levels of estradiol, it does not trigger multiple and frequent episodes of ovulatory menstruation (72).\\nFigure 3. Not all forms of uterine bleeding increase the risk of endometriosis.\\nOpen in a new tab\\n(A) Menstrual tissue after a nonpregnant ovulatory cycle. In addition to inducing stromal and epithelial differentiation, progesterone during the secretory phase of the menstrual cycle stimulates the coiling of the spiral arteries (64, 65). When progesterone production is interrupted in a nonpregnant cycle, the spiral arteries of the basalis and functionalis layers go through rapid and intense vasoconstriction and coagulation, leading to the separation of the thick functionalis layer in its entirety. This leads to the presence of large numbers of viable stromal, epithelial, and endothelial cells along with blood in the menstrual tissue. Some of these tissue fragments travel through the tubes into the pelvic cavity. During this process, mutated epithelial cells and epigenetically abnormal populations of stromal cells are deposited on the pelvic peritoneum or the ovaries. (B) Menstrual tissue associated with withdrawal bleeding in a patient on an oral contraceptive (OC). Because the endometrial tissue that is chronically exposed to an estrogen-progestin combination or progestin-only OC does not build a thick functionalis layer, temporary cessation of OC gives rise to menstrual tissue composed primarily of blood and containing very few stromal or epithelial cells (65, 76). This is also the case for the breakthrough bleeding the patient experiences while they are on hormone-containing OC pills used in a continuous manner.\\nAdditionally, time-honored clinical data support the postulate that repetitious menstrual episodes that follow ovulation contribute to the development or worsening of endometriosis symptoms. For example, the conventional use of cyclic combination oral contraceptive pills involving withdrawal bleeding is associated with cyclic uterine bleeding that is much lighter than spontaneous ovulatory menses (Figure 3B) (65, 73–76). This may be because a more superficial and thinner portion of endometrial tissue is present in the menstrual tissue (Figure 3B). Oral contraceptive use, which disrupts or suppresses ovulatory menses, reduces endometriosis symptoms such as dysmenorrhea or chronic pelvic pain (74). Breakthrough bleeding associated with other anovulatory conditions such as polycystic ovary syndrome or postmenopausal hormone replacement does not increase the risk of endometriosis (61, 70). It follows that suppression of ovulation should be pursued as a first-line approach for preventing the development of endometriosis or alleviating its symptoms (77).\\nKey mechanisms of endometriosis\\nThe fundamental mechanism of endometriosis is relatively straightforward. Following each nonpregnant ovulatory cycle, menstruation ensues. This routinely involves retrograde travel of blood and floating fragments of endometrial tissue through the uterine tubes onto the ovaries and pelvic peritoneal surfaces in the majority of premenopausal women (78). Although most ovulatory women are presumed to have retrograde menstruation, only a fraction of this population seems to develop endometriosis and its symptoms (1). Quantitatively, obstructive Müllerian abnormalities such as imperforate hymen or transverse vaginal septum strikingly increase retrograde menstrual tissue and very often give rise to endometriosis (79). Premenopausal women with heavy menstrual bleeding also have higher risk of developing endometriosis (80, 81). Qualitatively, endometrial tissue with cancer-driver mutations and epigenetic defects leading to abnormal gene expression is more likely to survive in ectopic locations and give rise to endometriosis (22). All these observations with mechanistic implications, however, likely require the presence of numerous back-to-back ovulatory cycles for endometrial fragments to repeatedly implant and accumulate in an ovarian corpus luteum cyst (ovarian endometrioma), peritoneal surfaces proximal to tubal ostia (peritoneal endometriosis), or the peritoneal pouch between the rectum and the vagina (deep-infiltrating endometriosis) (Figures 2 and 3) (22, 69).\\nIn addition to the retrograde menstruation theory based on ample scientific evidence from human samples, there has been considerable controversy regarding whether other biological processes contribute to endometriosis (15). Based on a report that human donor-derived endometrial cells were detected in eutopic endometrial biopsy samples from human bone marrow recipients, a mouse model was developed to demonstrate that circulating stem cells may contribute to endometriosis (82, 83). Additionally, the peritoneal metaplasia or Müllerian remnant hypothesis was primarily based on some 11 published case reports of male patients, whereby small uterine tissue near the bladder or inguinal canal was discovered because of pain symptoms, frequently stimulated by exogenous estrogen administration to treat prostate cancer (84, 85). The deficiency of or resistance to anti-Müllerian hormone (AMH) during embryonic development most likely caused the persistence of intact uterine tissue, including the myometrium and endometrium, in the pelvis of each of these 46,XY individuals (84, 85). In my opinion, these rare cases of persistent Müllerian tissue in men are not relevant to the hypothesis that peritoneal tissue in women may differentiate to endometriosis via metaplasia or that endometriosis may arise from Müllerian tissue remnants (84, 85).\\nCellular origins\\nA pathologist typically makes a tissue diagnosis of endometriosis by identifying two of the following three types of cells: (a) endometrial stromal cells, (b) endometrial epithelial cells, and (c) hemosiderin-containing macrophages indicative of chronic bleeding (34). Stromal cells constitute the bulk of endometriotic lesions. Epithelial cells may or may not be present on the surface of an endometriotic lesion and less frequently line the invaginating or glandular portions of these lesions. The origin of the macrophages containing blood pigment in endometriotic tissue is not well understood.\\nEpithelial cells of endometriosis.\\nEpithelial cells of peritoneal, deep-infiltrating, or ovarian endometriotic lesions share identical cancer-driver mutations with the eutopic endometrial epithelial cells of the same patients (22). At least four different laboratories mapped identical DNA variants or mutations in eutopic endometrium to matched tissues of pelvic endometriosis within the same individuals (21–25, 46, 86). This provided irrefutable evidence that all three forms of pelvic endometriotic lesions originated from the eutopically located endometrium (45).\\nInterestingly, starting from the teenage years, the endometrium of an average woman progressively accumulates cancer-driver mutations in oligoclones of epithelial cells located in the deepest portions of invaginating mucosal crypts, also called glands (87). Hundreds of cancer-driver genes may be mutated in various endometrial epithelial cell populations of an adult woman (21–25, 46, 86, 87). Given the frequency of this mutational process in the majority of women, it is considered normal (22). What is not well understood is how these mutations contribute to the disease process in a subset of women (44).\\nIn eutopic endometrial tissue, although the most commonly mutated gene is PIK3CA, other gain-of-function mutations such as those that affect KRAS are also frequently detected (21–25, 46, 86, 87). Although mutations sporadically affect small numbers of cells distributed across the eutopic endometrium, the majority of endometriotic epithelial cells carry these mutations in the same individual (21–25, 46, 86, 87). In particular, epithelial cells with KRAS mutations are enriched in endometriotic lesions, suggesting that autonomously activated KRAS may confer survival advantages for endometriotic tissue (21, 22, 24, 25, 46, 86). Intriguingly, a number of these mutations in endometriosis also serve as driver mutations for epithelial ovarian cancer (22, 24, 25, 46, 86). Although clear cell and endometrioid ovarian cancers are more strongly associated with an ovarian endometrioma, history of high rates of ovulatory activity is a well-established risk factor for all histological types of epithelial ovarian cancer (44, 88, 89). These data collectively suggest that ovulation followed by retrograde menstruation leads to the seeding of endometrial tissue fragments, including mutated epithelial cells, in an involuting corpus luteum cyst or other inclusion cysts, a small portion of which may eventually evolve into a benign or malignant ovarian neoplasm over decades (44, 69).\\nAdditionally, considerable differences were found between eutopic endometrial and endometriotic epithelial cells with respect to mRNA or protein expression. One of these proteins, 17β-hydroxysteroid dehydrogenase type 2 (HSD17B2), which physiologically catalyzes the enzymatic conversion of estradiol to biologically less potent estrone in eutopic endometrial epithelium during the implantation window, was severely deficient in endometriotic epithelial cells (90–92). The underlying mechanism involves a paracrine relationship between endometrial stromal and epithelial cells (93). It turned out that progesterone via its receptor (PGR) physiologically induces retinol uptake and metabolism in decidualized endometrial stromal cells (94). Consequently, increased levels of retinoic acid, in turn, stimulate HSD17B2 expression and activity in endometrial epithelial cells (93, 95, 96). This paracrine induction of epithelial HSD17B2 expression does not occur in endometriotic tissue because of decreased stromal PGR levels (93–97). Another implantation window marker, glycodelin (PAEP), is expressed at strikingly lower levels in endometriotic epithelial cells compared with the eutopic endometrial tissue (98). PAEP mRNA and protein are significantly lower in eutopic endometrium of women with versus without endometriosis (99). PAEP expression in endometrial tissue was reported to be significantly lower in implantation failure during IVF (100).\\nStromal cells of endometriosis.\\nFollowing the seminal 1994 publication by Ryan et al., which described how to maintain primary endometriotic stromal cells isolated from ovarian endometriomas in culture, a number of groups conducted mechanistic studies on these cells and compared them with eutopic endometrial stromal cells in culture (101, 102). These studies started by demonstrating differential mRNA or protein expression between eutopic and endometriotic stromal cells. They collectively revealed that endometriotic stromal cells bear widespread epigenetic aberrations involving pathological overexpression of transcription factors and nuclear receptors, which transform these cells from a normal endometrial stromal cell to an endometriotic stromal cell (34). Most notably, ectopic expression of NR5A1 (SF-1) and GATA6 together was sufficient to transform a normal endometrial stromal cell to an endometriotic stromal cell that can convert cholesterol to estradiol via activation of the steroidogenic cascade, including aromatase (103). These differences in mRNA or protein expression were also demonstrated in the eutopic endometrium of endometriosis patients, although the difference between eutopic endometrial tissues of women with versus without endometriosis was more subtle compared with expression profiling between endometrium and endometriosis (35, 99, 104). For example, expression of the decidualization markers prolactin and insulin-like growth factor–binding protein-1 (IGFBP1) is reduced in endometriotic stromal cells and in the stromal cells of the endometrium of women with endometriosis compared with disease-free endometrial stromal cells (105).\\nEndometriosis originates from eutopic endometrium of endometriosis patients.\\nAnalyses of epithelial and stromal cells of eutopic endometrium of patients with or without endometriosis and endometriotic lesions with respect to analyses for mutations, DNA methylation, histone modification, and gene expression indicate the following (34): Eutopic endometrium of endometriosis patients harbors oligoclones or localized populations of stromal cells with aberrant mRNA and protein expression and epithelial cells with mutations (22, 25, 104, 106). The great majority of stromal cells in endometriotic peritoneal implants, deep-infiltrating endometriosis, or ovarian endometriomas contain similar aberrant levels of mRNA/protein, most notably NR5A1, GATA6, and aromatase, in part due to DNA methylation differences, which control gene expression (106–108). Likewise, the great majority of epithelial cells in all forms of endometriotic lesions bear an identical driver mutation found in oligoclones of eutopic endometrial cells of the same individual (21–25, 46, 86). The most logical explanation is that if these tissue fragments with molecular aberrations travel to the pelvic cavity via retrograde menstruation, they have a higher potential to implant, survive, and grow in response to estrogen.\\nSomatic mutations\\nEutopic endometrial tissue of all women accumulates somatic mutations primarily in epithelial cells, starting from the teenage years (87). The mutation burden increases with advancing age (87). In particular, one laboratory reported significant differences between the eutopic endometrial tissues of women with and without adenomyosis or endometriosis with respect to the types of mutations (25). They compared gene mutations in histologically normal-appearing endometrial tissues of patients with adenomyosis, patients with neither adenomyosis nor endometriosis, and patients with endometriosis but without co-occurring adenomyosis (25). In general, the most commonly mutated gene in all three groups was PIK3CA (25). Mutation or variant allele frequency in eutopic endometrium was similar with respect to the PIK3CA or PPP2R1A genes in all three groups. KRAS (G12/G13) mutation or variant allele frequency was significantly higher in endometrial tissues of adenomyosis patients compared with disease-free patients (25). In endometrial tissues of patients with endometriosis but without co-occurring adenomyosis, the KRAS mutation or variant allele frequency was higher, but the difference did not reach significance (25). Although other studies did not make this direct comparison, their data were also suggestive that the eutopic endometrium of endometriosis patients contained a higher frequency of KRAS mutations compared with those of endometriosis-free patients (22, 86). This set of observations raises the question of whether the individuals who develop endometriosis are more likely to have endometrial tissue fragments with KRAS mutations in their retrograde menstrual tissue material.\\nOn the other hand, the mutation landscapes of endometriotic or adenomyotic tissues are clearly different from those of eutopic endometrium of the same patient (22, 25, 86). For example, adenomyotic tissue contains almost exclusively KRAS mutations (25). Similarly, the majority of epithelial mutations in endometriotic tissue affect the KRAS gene (22, 25). Moreover, identical epithelial mutations that affect both eutopic endometrium and endometriosis have been mapped in the matched tissues of each individual patient (22, 25). These observations collectively suggest that endometriosis or adenomyosis originates from eutopic endometrial tissue clones with epithelial mutations that enable these tissues to survive outside of the uterine cavity (21–25, 46, 86).\\nEpigenetic defects\\nDNA methylation.\\nThe initial epigenetic studies on endometriosis focused on DNA methylation (108–110). They demonstrated differences between the stromal cells of eutopic endometrium and ovarian endometriomas with respect to the CpG methylation of the promoters or coding regions of distinct transcription factors or nuclear receptors (111). In general, genes such as NR5A1, ESR2 (encoding estrogen receptor-β), and GATA6 are heavily methylated and suppressed in endometrial stromal cells, whereas they are less methylated and are expressed in endometriotic stromal cells (108–110). In contrast, GATA2 is methylated and suppressed in endometriosis, whereas it is highly expressed and serves as a critical mediator in progesterone-induced differentiation or decidualization of eutopic endometrial stromal cells (108, 112). In parallel, the failure to downregulate DNA methyltransferase-3B expression in response to physiological differentiating stimuli in endometriotic cells compared with healthy endometrial stroma may contribute to a defective epigenetic fingerprint and gene expression pattern in endometriosis (113). Mechanistically, it is not well understood how individual genes such as NR5A1, ESR2, GATA6, or GATA2 are differentially methylated in each cell type (108, 112).\\nOther studies also demonstrated that aberrations in global DNA methylation play a key role in the pathophysiology of endometriosis (114–116). One report suggested that estrogen and progesterone regulate DNA methylome in endometrial cells, and this process is perturbed in endometriosis (115). A recent report suggested that menstrual cycle phase was a major source of DNA methylation of the eutopic endometrium (116). This differential methylation study of endometrium from disease-free or endometriosis patients highlighted candidate genes contributing to disease risk (116).\\nHistone modification.\\nA limited number of studies have been published regarding the histone modification patterns in endometriosis (117). One study assessed acetylation/methylation levels of histones in whole tissues of endometriosis, eutopic endometrium from endometriosis patients, and endometrium from disease-free individuals (117). Globally, the endometriotic lesions were hypoacetylated at H3 but not H4 compared with disease-free eutopic endometrium (117). Lysine (K)-specific acetylation/methylation levels of histones showed that endometriosis tissue contained significantly lower levels of acetylated H3K9 and H4K16 in comparison with eutopic endometrium from patients or disease-free subjects (117). Both endometriotic and eutopic endometrial tissues from patients were hypermethylated at H3K4, H3K9, and H3K27 compared with normal endometrium of disease-free women. ChIP analysis of endometriotic lesions showed hypoacetylation of H3/H4 around promoter regions of genes previously demonstrated to be downregulated in endometriosis, such as HOXA10, ESR1, CDH1, and CDKN1A, compared with disease-free endometrium (117). On the other hand, the NR5A1 promoter region was enriched with acetylated H3 and H4 in endometriosis versus disease-free tissues, correlating with its reported overexpression in endometriosis (117).\\nEpigenetic defects affecting nuclear receptors in endometriotic stromal cells lead to aberrant expression of critical genes. For example, NR5A1 expression in endometriotic stromal cells activates most of the genes in the steroidogenic cascade, including aromatase, leading to the conversion of cholesterol to estradiol, whereas this is not observed in the eutopic endometrium of disease-free individuals (Figure 4) (102, 103, 107, 118, 119). Excessive ESR2 expression leads to suppression of ESR1 (ERα) (120). Ectopic overexpression of NR5A1 and GATA6 in a normal endometrial stromal cell is sufficient to confer an endometriotic phenotype with respect to steroidogenesis and failure to differentiate properly (Figure 4) (103, 108). Thus, epigenetic stromal defects in endometriosis profoundly influence the biology of endometriosis.\\nFigure 4. The epigenetic defects that cause gene expression abnormalities in endometriotic cells.\\nOpen in a new tab\\nThe transcription factors NR5A1, GATA6, and ESR2 are pathologically hypomethylated and thus upregulated in the stromal cells of endometriosis, whereas PGR is downregulated in these cells, compared with normal endometrial stromal cells. Consequently, PGE2 stimulates the steroidogenic enzymes, including aromatase, to increase the production of estradiol, which stimulates COX-2 to elevate PGE2 production. PGE2 itself induces COX-2 expression through a short feedback loop. This vicious cycle causes continuous local production of estradiol and PGE2 in endometriotic tissue. Low PGR levels in endometriotic stromal cells lead to progesterone resistance and decreased uptake of retinol, leading to decreased retinoic acid production. The enzyme HSD17B2, which is physiologically regulated by retinoic acid in normal endometrium, is downregulated in endometriotic epithelial cells. Deficient HSD17B2 leads to decreased metabolism of estradiol, further contributing to increased estradiol levels in endometriotic tissue.\\nProgesterone resistance\\nProgesterone resistance in endometriotic tissue or eutopic endometrium has been described by numerous laboratories and clinicians in different contexts (35, 91–93, 97, 104, 121–124). It was originally suggested upon the observation that the enzyme HSD17B2, which is involved in inactivation of estradiol, is present in endometrial epithelium but absent in matched endometriotic tissue during the luteal phase, although both tissues are simultaneously exposed to circulating progesterone (Figure 4) (91). Follow-up studies demonstrated that progesterone physiologically acted directly on endometrial stromal cells to alter retinoid uptake and biosynthesis of retinoic acid, which in turn induced epithelial HSD17B2 expression in a paracrine fashion (Figure 4) (93, 96, 125). Pathologically, deficient PGR expression in endometriotic stromal cells disrupted this physiological mechanism and thus the capacity of endometrial epithelial cells to inactivate estradiol during the secretory phase (Figure 4) (93, 94, 96, 125).\\nA number of other laboratories also reported deficient expression of genes regulated by progesterone in endometriotic tissue or eutopic endometrium of endometriosis patients (99, 104, 126, 127). These included IGFBP1, PAEP, HOXA10, HOXA11, CYP26A1, FOXO1A, FKBP4, and BCL2 (99, 104, 126, 127). HOXA10 hypermethylation was demonstrated as a mechanism for decreased HOXA10 expression in the endometrium of women with endometriosis (127). Thus, inappropriate methylation of progesterone-regulated genes may be a potential mechanism for progesterone resistance. The underlying mechanisms for the other suppressed genes such as epithelial PAEP in tissues of women with endometriosis, however, have remained unclear.\\nProstaglandin production and action\\nProstaglandin E2 (PGE2), a product of the COX/PGES cascade, is the most intensively studied eicosanoid that contributes to inflammatory pain (47). PGE2 also plays a central role in the pathophysiology of endometriosis (Figure 4) (48). In endometrium and inflamed tissues such as the peritoneum adjacent to endometriosis, upregulated COX-2 and perhaps also COX-1 produce prostaglandin H2 (PGH2) from arachidonic acid released by phospholipase A2, and then upregulated PGE synthase subtypes produce PGE2 from PGH2 (Figure 4) (128, 129). PGE2 interacts with the G protein–coupled EP2 and EP4 receptors in the nerve endings or nociceptors, leading to activation of PKA and also EPAC (exchange protein activated by cAMP) to produce PGE2-mediated pain (47, 48). In addition to the PGE2 receptors at the nerve endings, spinal EP2 receptors via PKA activation also contribute to the amplification of PGE2-evoked nociceptive input from the spinal cord to higher brain areas where pain is actually perceived (47).\\nPeritoneal fluid concentrations of PGE2 are higher in women with endometriosis, and, in addition to evoking pain, elevated PGE2 plays key roles in survival and growth of endometriotic lesions (130, 131). Inhibition of PGE2 synthesis suppresses endometriosis and chronic pelvic pain in women and decreases growth of experimental endometriotic lesions in animal models (132, 133). PGE2 in endometriotic stromal cells induces the expression of aromatase and other steroidogenic enzymes via the EP2/PKA/cAMP pathway and generates estradiol (Figure 4) (102, 103, 119, 134). Estradiol, in turn, stimulates COX-2 expression via ESR2 to enhance PGE2 production in uterine cells (Figure 4) (135). Moreover, PGE2, via a short feedback loop, induces COX-2 expression via the activation of NF-κB in endometrial stromal cells, leading to a vicious cycle that continuously produces estradiol and PGE2 in endometriotic tissue (Figure 4) (102, 129).\\nSelective inhibition of EP2 and EP4 decreases survival and growth of human endometriotic cells via altering the expression of a number of gene products, including matrix metalloproteinases (MMPs) and tissue inhibitors of metalloproteinases (TIMPs) (136). Selective inhibition of EP2/EP4 in an experimental animal model also decreased angiogenesis and innervation of endometriotic lesions, suppressed proinflammatory state of dorsal root ganglia neurons, and significantly reduced proinflammatory, estrogen-dominant, and progesterone-resistant molecular environment in endometriotic tissue (137).\\nNSAIDs, inhibitors of COX-1 and/or COX-2, suppress endometriosis-associated inflammatory pain by reducing generation of eicosanoids, mainly PGE2, while they exhibit gastrointestinal, renal, and cardiovascular toxicities (132, 138). Selective inhibitors of microsomal PGE synthase-1 and subtype-selective antagonists of PGE2 receptors, particularly EP2 and EP4, may be useful as future treatments of endometriosis pain with minimized side effects (137). Thus, PGE2 has a substantial impact on endometriosis-induced pain, and pharmacological intervention in PGE2 signaling at pelvic peritoneal tissues or central nervous system may serve as a future therapeutic strategy for the treatment of endometriosis-induced pain (47).',\n",
       "  'Discussion': '',\n",
       "  'Conclusion': 'Pelvic endometriotic tissue evidently originates from shed endometrial fragments that travel in a retrograde manner primarily through the tubes (19, 20, 45). In patients with endometriosis, relatively larger populations of stromal and epithelial cells in the eutopic endometrium display widespread epigenetic abnormalities and mutations that favor increased estradiol and PGE2 production and action, progesterone resistance, autonomous activation of KRAS and its downstream survival pathways, and heightened cytokine production and immune cell recruitment (Figure 4). Although these molecular abnormalities have been consistently observed by multiple laboratories, the upstream reproductive process that is indispensable for the establishment of endometriosis and its symptoms remains as repetitious episodes of ovulation followed by retrograde menstruation. That is why the risk of developing endometriosis is extremely low in the absence of ovulatory menses. Ovulation is associated with relatively larger quantities of shed endometrial functionalis tissue that separates from the basalis and travels to the pelvic cavity, increases blood estradiol levels, and directly releases ovulatory fluid with massive amounts of estradiol onto the peritoneal surfaces. The resulting corpus luteum cyst potentially serves as the foundation for a new endometrioma formation. Therefore, the process of ovulation per se strikingly increases the risk for endometriosis in multiple ways.\\nEliminating or diminishing ovulation or ovulatory menses in the long term should be the first-line strategy for treating pain caused by endometriosis or potentially preventing endometriosis. Surprisingly, this has not been explicitly addressed by clinicians or researchers. Traditionally, progestins, combination oral contraceptives, or injectable gonadotropin-releasing hormone (GnRH) agonists have been used to suppress ovulatory activity regulated by follicle-stimulating hormone (FSH) and luteinizing hormone (LH) (139–142). More recently, oral GnRH antagonists with estrogen or progestin addback have been introduced (143, 144). The key side effects of GnRH suppression have been hot flashes and decreased bone mineralization (145). New pharmaceutical approaches to suppress ovulation long-term with fewer side effects are urgently needed for prevention and treatment of endometriosis. I speculate that exploiting the inhibin, activin, and follistatin-related pathways or blocking the gonadotropin receptors may produce beneficial future pharmaceutical targets.',\n",
       "  'URL': 'https://doi.org/10.1172/JCI188787',\n",
       "  'Journal': 'J Clin Invest'},\n",
       " {'PaperTitle': 'miRNA in Endometriosis—A New Hope or an Illusion?',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Anna Dryja-Brodowska, Bogdan Obrzut, Maciej Obrzut, Dorota Darmochwał-Kolarz',\n",
       "  'Email': 'bobrzut@ur.edu.pl',\n",
       "  'Abstract': 'Background: Endometriosis is a complex, estrogen-dependent condition that can significantly impact women’s quality of life and fertility. Current diagnostic strategies remain invasive and often prolonged, demonstrating the need for reliable, non-invasive biomarkers. In this context, microRNAs (miRNAs), due to their stability in blood and regulatory roles in inflammation and cell proliferation, have emerged as promising candidates.\\nMethods: This review systematically analyzes 17 studies published between 2010 and 2025 that investigated the diagnostic utility of circulating and tissue-based miRNAs in endometriosis.\\nResults: A wide range of dysregulated miRNAs was identified, with miR-125b-5p, miR-451a, and miR-3613-5p showing the most consistent alterations across studies. However, diagnostic performance varied considerably—largely due to methodological heterogeneity. Key differences were observed in sample type (serum, plasma, endometrium), patient selection, and control group definition. The menstrual cycle phase and hormonal status were often not matched or reported, limiting reproducibility.\\nConclusions: Despite encouraging findings, the current evidence base is weakened by inconsistent protocols and limited validation. Standardized, multicenter research with well-characterized patient cohorts is essential to the establishment of clinically applicable miRNA-based diagnostics. If validated, miRNAs may offer a transformative, non-invasive approach for earlier detection and improved management of endometriosis.',\n",
       "  'Introduction': 'Endometriosis is a chronic, estrogen-dependent inflammatory disease in which endometrium-like tissue develops outside its natural location—within the ovaries, fallopian tubes and other pelvic structures, and—in rare cases—in other parts of the body [1].\\nAccording to the World Health Organization (WHO), endometriosis affects 6–13% of the reproductive age population [2]. Some sources indicate that the incidence in the general population may be as high as 15–18% [3,4] and, in women struggling with infertility, this percentage can be as high as 30–50% [5]. There are also reports of isolated cases of endometriosis in men [6].\\nPain is the dominant symptom of this disease [7,8,9]. It is not only a physical symptom but also a factor influencing the emotional and social life of patients [10]. Its chronicity and variability can lead to serious consequences, in terms of both physical and mental health—including depression, weakening of relationships with loved ones and problems with self-esteem [11,12]. Women suffering from endometriosis may avoid professional, educational, social, or family activities, that have further consequences in the form of reduced economic activity and a decrease in the number of people participating in the workforce [13,14,15].\\nEndometriosis is a disease that is often difficult to diagnose due to its varied symptoms, which may resemble other diseases such as primary dysmenorrhea, pelvic inflammatory disease, intestinal problems, urinary tract inflammation, or degenerative changes in the spine, which presents a diagnostic challenge for doctors of all specialties [16]. It is a difficult to diagnose disease due to the lack of non-invasive tests that would allow its detection [17].\\nUntil recently, laparoscopy was considered the gold standard in diagnosing endometriosis [18]; however, this is an invasive procedure and carries a risk of complications. Furthermore, the absence of endometriosis foci during surgery does not necessarily entail a negative diagnosis. The latest ESHRE guidelines recommend diagnosing endometriosis based on an interview and/or imaging tests, i.e., ultrasound or MRI [19]. According to the ESHRE guidelines, surgery should be proposed to patients with symptoms that do not respond to pharmacological treatment or with symptoms but with negative imaging test results.\\nTherefore, the search for a non-invasive marker enabling the diagnosis of endometriosis has become a priority for researchers in recent years [20].\\nAmong the numerous non-invasive markers of endometriosis being studied are microRNA.\\nmicroRNAs (miRNAs) are short, endogenous RNA molecules of about 22 nucleotides in length that play a key role in the regulation of gene expression at the post-transcriptional level [21]. They act by binding to mRNA and inhibiting translation, leading to the suppression of target gene expression [22].\\nIn addition, miRNAs can influence chromatin modifications and DNA methylation, which further regulates gene expression [23].\\nmiRNAs are involved in many biological processes such as cell division, proliferation, cell differentiation, apoptosis and stress response [24]. Because of their high stability in biological fluids and tissue specificity, miRNAs could be promising candidates for diagnosing endometriosis and other diseases [25,26].\\nIn recent years, researchers have shown that specific miRNAs are important in endometriosis [27].\\nIn this article, we aim to assess whether we are indeed approaching the discovery of a new, non-invasive biomarker for endometriosis, as suggested by the growing body of research, or whether this is merely an illusion resulting from the large number of papers on the subject.\\nWe aim to explore the current state of knowledge on this subject by analyzing the available articles that may indicate a real breakthrough in the diagnosis of endometriosis and to identify possible barriers that may prevent full achievement of this goal.',\n",
       "  'Methods': '2. Materials and Methods\\n2.1. Protocol\\nThe protocol for this systematic review was designed following the guidelines of the Preferred Reporting Items for Systematic Review and Meta-analysis Protocols (PRISMA) [28]. The present review protocol is officially registered in the INPLASY database on 6 June 2025 (INPLASY202560027).\\n2.2. Search Strategy\\nWe conducted a thorough literature review using electronic databases such as Pubmed and Google Scholar, considering articles that examined miRNA as potential biomarkers in endometriosis. Articles were published in English from 2010 to March 2025. We used keywords such as: micro RNA, miRNA, biomarker, endometriosis, mcRNA and a combination of them. We found 727 articles.\\nInclusion criteria contained: randomized clinical trials, meta-analyses, retrospective and prospective studies; human studies only; publication year between 2010 and 2025; articles in English; accessible as full-text articles; study population ≥ 24; samples taken from blood, serum, plasma, or endometrium; journal Impact Factor at the time of publication ≥ 2.\\nExclusion criteria included: not accessible as full-text articles; articles in a language other than English; non-human study population; Impact Factor < 2; publication year earlier than 2010; study population < 24; review articles. The exclusion of studies published in journals with an impact factor (IF) below 2 was implemented to reduce the risk of including poorly controlled or non-peer-reviewed studies and to prevent inclusion of repetitive or lower-quality results from less rigorous publications. While this criterion may have omitted some relevant data, it was implemented to strengthen the overall credibility and scientific rigor of the review.\\nFull-text articles were obtained for abstracts that appeared to meet the inclusion criteria. We additionally conducted a manual review of the bibliographies of the included articles to identify additional relevant studies.\\nTaking into account the inclusion and exclusion criteria listed below, we selected 17 articles for this review.\\n2.3. Data Collection Process\\nThe articles included in our review were thoroughly analyzed by two independent researchers. Authors performed data extraction using a Microsoft Excel form (Microsoft Corporation, Redmond, WA, USA). The following data were obtained from each study: the name of the first author, year of publication, the journal’s Impact Factor at the time of publication, the nationality of the patients included in the study, the number of participants in the study group and control groups, the phase of the menstrual cycle—whether it influenced miRNA expression, key dysregulated miRNA, control group characteristics, and the main outcomes.\\nA formal risk of bias assessment was not performed due to the heterogeneity of study designs included. However two independent reviewers evaluated each study for eligibility based on the predefined criteria outlined in the protocol. Any differences between the reviewing authors were clarified in joint discussions. Based on the 17 studies included in this review, several recurring sources of potential bias were identified. Selection bias was common, as control groups often included women with other gynecologic conditions rather than healthy, asymptomatic individuals. Many studies did not report or control for the menstrual cycle phase at the time of sample collection. Sample types (serum, plasma, tissue) and analytic platforms varied widely, limiting comparability. Small sample sizes and lack of independent validation cohorts further reduced the generalizability of findings. Such factors may have influenced the comparability and reproducibility of results across studies.\\nThe results of the identified studies were organized into a table, with the most significant elements discussed in detail. The study selection process is illustrated in the PRISMA flow diagram (Figure 1).',\n",
       "  'Results': '3. Results\\nThis review included 17 studies examining miRNA in women with endometriosis compared to controls [29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45]. Despite the diversity of methods and study populations, common patterns and important differences in study designs can be identified.\\nTable 1 provides an overview of all included studies and their key characteristics such as the name of the first author, the year of publication, the country where study was performed, the journal’s Impact Factor at the time of publication, the number of patients included in study group and control group, whether menstrual phase was taken into account and its result on miRNA expression, dysregulated miRNA-upregulated or downregulated, sample type and characteristic of control group.\\nTable 1. Characteristics of studies included in the systematic review.\\nAuthor, Year\\nCountry of Study\\nIF\\nStudy Group (n)\\nControl Group (n)\\nMenstrual phase Considered (Yes/No)-Effect on miRNA Expression\\nDysregulated miRNA\\nSample Type\\nCharacteristic of Control Group\\nPetracco et al., 2011 [29]\\nUSA\\n6.3\\n32\\n50\\nyes-no effect\\nmiR-135a ↑, miR-135b ↑\\nendometrial tissue\\nwomen undergoing operation-no data-women with polyps, uterine fibroids were excluded\\nWang et al., 2013 [30]\\nChina\\n6.3\\n60\\n25\\nyes-no effect\\nmiR-199a ↑, miR-122 ↑, miR-145 ↓, miR-542-3p ↓\\nserum\\nwomen undergoing lpsc-mainly due to tubal infertility\\nBraza-Boïls et al., 2014 [31]\\nSpain\\n4.8\\n51\\n32\\nyes-no effect\\nmiR-202-3p ↑, miR-424-5p ↑, miR-449b-3p ↑, miR-556-3p ↑\\nendometrial tissue, endometriotic lesions\\nwomen undergoing lpsc for sterilization\\nCho et al., 2015 [32]\\nSouth Korea\\n4.4\\n24\\n24\\nyes-effect\\nlet-7b ↓, let-7d ↓, let-7f ↓, miR-135a ↑\\nserum\\nwomen undergoing lpsc due to adnexal or peritubal cysts\\nRekker et al., 2015 [33]\\nEstonia\\n4.4\\n61\\n65\\nyes-effect\\nmiR-200a ↓, miR-200b ↓, miR-141 ↓\\nplasma\\nwomen undergoing lpsc with abdominal pain, infertility, PCOS or healthy-without lpsc\\nCosar et al., 2016 [34]\\nUSA\\n6.3\\n24\\n24\\nyes-no effect\\nmiR-125b-5p ↑, miR-150-5p ↑, miR-342-3p ↑, miR-143-3p ↓, miR-451a ↑, miR-3613-5p ↓\\nserum\\nwomen undergoing lpsc due to adnexal or peritubal cysts\\nZhou et al., 2016 [35]\\nChina\\n4.6\\n22\\n20\\nyes, but only midsecretory-no data\\nmiR-196a ↑\\nEndometrial tissue\\nwomen undergoing lpsc for infertility\\nPateisky et al., 2018 [36]\\nAustria\\n2.9\\n51\\n41\\nyes-effect\\nmiR-154-5p ↓, miR-196b-5p ↓, miR-378a-3p ↑, miR-33a-5p ↓\\nplasma\\nwomen undergoing lpsc with fibroids, infertility, adnexal cysts\\nNisenblat et al., 2019 [37]\\nAustralia\\n5.8\\n51\\n27\\nyes- effect\\nmiR-155 ↓, miR-574-3p ↓, miR-139-3p ↓\\nplasma\\nwomen undergoing lpsc with abdominal pain and infertility\\nVanhie et al., 2019 [38]\\nBelgium\\n5.7\\n60\\n30\\nyes-no effect\\nmiR-125b-5p ↑, miR-28-5p ↑, miR-29a-3p ↑\\nplasma\\nwomen undergoing lpsc due to infertility, abdominal pain\\nMoustafa et al., 2020 [39]\\nUSA\\n8.8\\n41\\n59\\nyes-no effect\\nmiR-125b-5p ↑, miR-150-5p ↑, miR-342-3p ↑, miR-451a ↑, miR-3613-5p ↓, let-7b ↓\\nserum\\nwomen undergoing lpsc due to suspected benign pelvic lesions\\nPapari et al., 2020 [40]\\nIran\\n7.3\\n53\\n53\\nno-no data\\nmiR-199a-3p ↓, miR-143-3p ↓, miR-340-5p ↓, let-7b-5p, miR-21-5p ↑, miR-17-5p ↓, miR-20a-5p ↓, miR-103a-3p ↓\\nplasma\\nwomen undergoing lpsc with fibroids, infertility, pelvic tumors\\nFarismadan et al., 2021 [41]\\nIran, Iraq\\n2.4\\n260\\n260\\nno-no data\\nmiR-146a ↑, miR-149 ↑, miR-196a2 –, miR-499 (their SNPs)\\nblood\\nwomen undergoing lpsc due to urinary incontinence, pelvic organ prolapse, adnexal cysts\\nWalasik et al., 2023 [42]\\nPoland\\n2.5\\n24\\n25\\nno-no data\\nmiR-3613-5p ↓, miR-451a ↑\\nplasma\\nwomen undergoing lpsc due to adnexal cysts or diagnostic lpsc\\nChico-Sordo et al., 2024 [43]\\nSpain\\n5.6\\n77\\n82\\nyes, but only in control group-no effect\\nmiR-30c-5p ↓\\nserum\\nwomen undergoing lpsc for sterilization or as oocyte donors\\nLiu et al., 2025 [44]\\nChina\\n3.7\\n60\\n40\\nno-no data\\nmiR-1229-5p ↑\\nendometriotic lesions\\nwomen undergoing operation for CIN III\\nWang et al., 2025 [45]\\nChina\\n2.3\\n100\\n80\\nno-no data\\nmiR-375-3p ↓\\nserum\\nno operation-healthy women, with no symptoms and with normal levels of sex hormones\\nNote: ↑ indicates upregulated expression of miRNA; ↓ indicates downregulated expression of miRNA.\\n3.1. Sample Type\\nThe vast majority of the studies included in this review (12 out of 17) used serum or plasma samples to evaluate miRNA levels as potential non-invasive biomarkers [30,32,33,34,36,37,38,39,40,41,42,43,45]. Serum was used in six studies [30,32,34,39,43,45]. The same number of studies used plasma [33,36,37,38,40,42]. Only one study [21] used white blood cells to investigate single nucleotide polymorphisms (SNPs) in miRNA genes, rather than miRNA expression itself.\\nThree out of 17 studies [29,31,35] focused on the analysis of eutopic and/or ectopic endometrial tissue and only one study [44] used only endometrial lesions for analysis.\\n3.2. Validation Method\\nRT-qPCR was used as a validation method in 14 out of the 17 studies [29,30,31,32,33,34,35,36,37,38,39,40,42,44]. The remaining three studies were designed differently [41,43,45]. Farismadan et al. analyzed genetic polymorphisms (SNPs) rather than expression levels [41]. Wang et al. (2025) used miRNA microarrays, but did not clearly describe the qPCR-based validation of individual miRNAs [45]. Chico-Sordo et al. used multiplex analysis and machine learning, but did not report qRT-PCR validation [43].\\n3.3. Number of Studies by Continent\\nA review of the included studies revealed a predominance of research conducted in Asia (n = 7) [30,32,35,40,41,44,45], particularly in countries such as China, Korea, Iran, and Iraq (Table 2). European countries accounted for six studies (Spain, Austria, Belgium, Poland, and Estonia) [31,33,36,38,42,43], while a smaller number of studies originated from North America (n = 3, all from the USA) [29,34,39] and Oceania (n = 1, Australia) [37].\\nTable 2. Number of studies by continent.\\nContinent\\nNumber of Studies\\nAsia\\n7 (China, Korea, Iran, Iraq)\\nEurope\\n6 (Spain, Austria, Belgium, Poland, Estonia)\\nNorth America\\n3 (USA)\\nOceania\\n1 (Australia)\\n3.4. Study and Control Group Characteristic\\nAll women in the study groups from selected publications had endometriosis confirmed surgically and histopathologically. The reviewed studies showed inconsistent reporting of coexisting disorders.\\nAcross the 17 studies, the characteristics of the control groups varied considerably.\\nIn 12 out of 17 studies, the control participants underwent laparoscopic surgery because of pelvic pathology [29,32,33,34,36,37,38,39,40,41,42,44]. The most common indications in this subgroup included tubal abnormalities, benign adnexal masses, e.g., ovarian or paraovarian cysts, uterine fibroids or pelvic pain.\\nBesides the above-listed pelvic pathologies, infertility as a co-indication was also mentioned in six papers [30,35,36,37,38,40].\\nIn two studies, controls underwent laparoscopy specifically for sterilization [31,43].\\nThe study of Farismadan et al. included controls undergoing surgery for CIN III (cervical intraepithelial neoplasia); however, the type of surgical procedure was not specified in the study [41]. In the research of Petracco et al., the authors did not specify whether the women in the control group underwent surgery [29].\\nNotably, only one study [45] involved a control group consisting of healthy, non-operated women who were asymptomatic and had normal levels of reproductive hormones.\\n3.5. Phase of Menstrual Cycle\\nThe menstrual cycle phase was considered in 10 out of the 17 included studies [29,30,31,32,33,34,36,37,38,39]. However, only three of these studies demonstrated a significant influence of menstrual phase on miRNA expression [32,36,37]. The phase-dependent miRNAs identified in these three studies have been visualized in two separate diagrams (Figure 2 and Figure 3), presented as heatmaps, to illustrate differential expression patterns across menstrual phases in women with endometriosis and healthy controls.\\nFigure 2. Diagram (heatmap) illustrating miRNA expression in endometriosis group depend on menstrual phase.\\nFigure 3. Diagram (heatmap) illustrating miRNA expression in the control group depends on the menstrual phase.\\nOne study considered the cycle phase only in the control group [43].\\nIn one study, all participants in the endometriosis group were in the mid-secretory phase [35].\\nOf the 17 studies, 5 do not take into account the menstrual cycle phase [40,41,42,44,45].\\n3.6. Dysregulated miRNA Across Studies\\nAnalysis of dysregulated miRNAs revealed a set of molecules commonly identified in multiple studies. Among these, miR-125b-5p was reported upregulated in three studies [34,38,39], as were miR-451a [34,39,42] and miR-3613-5p was downregulated [34,39,42].\\nOther recurrent miRNAs included miR-135a upregulated [29,32], let-7b downregulated [32,39]; miR-150-5p upregulated [34,39]; miR-342-3p upregulated [34,39] and miR-143-3p downregulated [34,40].\\nAnalysis of dysregulated miRNAs by sample type revealed that miRNA was both upregulated and downregulated in serum as well as in plasma [30,32,33,34,36,37,38,39,40,41,42,43,45].\\nIn contrast, in all studies analyzing endometrial or endometriotic tissue, all miRNAs investigated were upregulated [29,31,35,44].\\n3.7. Impact Factor of the Studies\\nThe impact factor (IF) of the journals in which the analyzed studies were published ranged from 2.3 to 8.8. The median IF was 4.8. This distribution indicates that this area is supported by a reliable source of peer-reviewed scientific publications.',\n",
       "  'Discussion': '4. Discussion\\nmiRNAs play a key role in the pathogenesis of endometriosis, influencing many biological and molecular processes involved in the development and persistence of endometrial lesions outside the uterine cavity [46,47]. They contribute to the development of endometriosis by modulating processes such as inflammation, cellular proliferation and apoptosis, angiogenesis, extracellular matrix remodeling, tissue repair and TGFβ-regulated pathways, hypoxic injury and resistance to progesterone [48,49,50].\\nNumerous genes involved in endometriosis are regulated by miRNAs. Among the most prominent are ERα/β, KRAS4A/4B, CYP19, IL-6, HOXA9, HOXA10, and EDN1—all relevant to disease pathophysiology. miRNAs further modulate mTOR signaling and the VEGF pathway, affecting cell growth, migration, and angiogenesis—key mechanisms in lesion development [51,52,53,54].\\nIn recent years, miRNAs have gained enormous interest as potential non-invasive diagnostic biomarkers in endometriosis.\\nEfforts to develop non-invasive diagnostic tools for endometriosis continue to intensify, reflecting the condition’s widespread nature and its profound clinical and societal consequences [20]. In the United States alone, the annual economic burden of endometriosis has been estimated at $78 billion, factoring in healthcare costs and productivity losses [15]. A large multicenter study involving over 1400 women demonstrated that patients with endometriosis worked, on average, 10.8 h less per week compared to controls, primarily due to pain-related impairment [13].\\nMoreover, chronic inflammation and persistent immune activation associated with the disease may contribute to increased risk of comorbidities, including cardiovascular disease, autoimmune disorders, and endocrine dysfunctions [55,56].\\nThese associations further emphasize the urgency to identify effective, non-invasive diagnostic strategies that could facilitate earlier detection and targeted interventions.\\nSeveral systematic reviews have examined circulating miRNAs in the context of endometriosis diagnosis, yet they consistently report inconsistent findings and a lack of validation in independent cohorts. Agrawal et al. [57] identified 42 dysregulated miRNAs across nine studies, but only one miRNA (miR-20a) was consistently reported, underscoring the limited reproducibility of candidate markers. Similarly, Leonova et al. [58] found that out of 63 miRNAs reported in 18 studies, only 14 were replicated, attributing these discrepancies to methodological variability. Earlier reviews, such as the Cochrane analysis by Nisenblat et al. (2016) [59], evaluated a broad range of blood biomarkers but gave minimal attention to miRNAs and did not assess their diagnostic value in depth.\\nOur systematic review offers a focused and up-to-date synthesis of recent miRNA studies, critically evaluates their methodological quality, and proposes structured recommendations to enhance their diagnostic use.\\nThis review of 17 studies reveals both the great potential of miRNA as new non-invasive biomarker for endometriosis and significant methodological and interpretive limitations related to their clinical application. These challenges must be critically examined before any meaningful clinical proceeding can be proposed.\\nOne of the main limitations of this review, as with any comprehensive analysis, is the substantial methodological variability across studies—including differences in sample materials, experimental models, analytical techniques, and outcome measures—which often leads to conflicting results and complicates the interpretation of the overall evidence.\\nA recurring issue across the literature is the limited size of study cohorts. Many investigations included small, often non-representative groups, both in the study and control group. Such sample sizes undermine statistical reliability and make it difficult to generalize findings. Additionally, subgroup analyses based on disease stage, lesion localization, or symptom severity are typically underpowered or absent altogether, leading to oversimplified conclusions.\\nCompounding this issue is the heterogeneity of control groups used across studies. Rather than recruiting healthy, asymptomatic women, many researchers have included patients undergoing surgery for other gynecological conditions such as fibroids, cysts, or unexplained infertility. These comorbidities may independently influence miRNA expression profiles, confounding comparisons and obscuring disease-specific signatures. Numerous studies have demonstrated that miRNAs are dysregulated in uterine fibroids [60,61,62]. In asymptomatic women undergoing sterilization, studies have shown that the incidence of endometriosis varies between 2 and 43% [63,64]. In women with unexplained infertility, the incidence of endometriosis may range from 30% to 63.2% [65]. Among women hospitalized for chronic pelvic pain, endometriosis is diagnosed in 45% to 82% of cases [66]. Furthermore, inflammatory or hormonal abnormalities present in these conditions may mimic or mask endometriosis-related miRNA changes. On the other hand, it is important to recognize that laparoscopy, currently considered the gold standard for confirming endometriosis, remains an invasive procedure associated with operative risks and ethical concerns. As a result, designing a truly “healthy” control group, that can be definitively confirmed as endometriosis-free without surgical exploration, presents a significant methodological challenge. Identifying appropriate control participants is further complicated by the fact that endometriosis may be asymptomatic [17]. Enrolling women without any gynecological complaints and excluding laparoscopy for ethical reasons may lead to undetected cases within control cohorts. This limitation highlights the delicate balance between diagnostic accuracy and ethical feasibility in the selection of appropriate control populations for endometriosis research.\\nIn many of the reviewed studies, the risk of incorrect classification of participants into endometriosis or control groups could not be excluded. Publications did not provide sufficient details concerning the diagnostic criteria, surgical techniques employed, and the qualifications or experience level of the surgical team.\\nA further confounder is the neglect of menstrual cycle phase. It is well documented that miRNA expression, particularly in endometrial and blood-derived samples, is modulated by hormonal fluctuations [67,68,69,70]. In the majority of studies, the menstrual cycle phase is either overlooked or insufficiently controlled, leading to biological variability. Controlling for or stratifying by cycle phase is essential, particularly when assessing tissues or fluids sensitive to estrogen and progesterone levels.\\nAn additional limitation identified across the reviewed studies is the lack of geographic and ethnic diversity, with the majority of data derived from populations in Asia and Europe. Given that miRNA expression can be influenced by genetic, environmental and lifestyle factors [71], the inclusion of multi-ethnic cohorts is essential to ensure broader applicability and generalizability of findings.\\nA major obstacle in the field remains the inconsistency and poor reproducibility of results. Although several miRNAs have been identified as dysregulated in more than one study, especially miR-125b-5p, miR-451a, miR196 and let-7 family [32,34,35,36,38,39,41,42], the vast majority of reported miRNAs were unique to single studies [29,30,31,33,37,40,43,44,45]. This lack of replication limits the credibility of individual biomarkers and underscores the need for independent validation in diverse cohorts. This also suggests that miRNA profiles are highly context-dependent and susceptible to variations in methodology, sample handling, or population characteristics.\\nAnother technical limitation relates to the lack of standardization in validation methods. Studies differ in their use of platforms such as RT-qPCR, microarray or NGS, normalization strategies, and analytical pipelines. Even when similar technologies are used, differences in RNA extraction, quality control, or thresholding can lead to divergent results. Without harmonized protocols, cross-study comparisons remain difficult, and findings lack the reliability needed for biomarker development.\\nIn the article published by Rahmioglu et al. in 2014 [72], as part of the WERF EPHect (World Endometriosis Research Foundation Endometriosis Phenome and Biobanking Harmonization Project), the authors proposed recommended standard operating procedures (SOPs) for the collection, processing, and storage of biological fluid samples in endometriosis research. The aim of this international initiative, involving 34 clinical and academic centers across 16 countries, was to harmonize methodologies and enhance the comparability of findings across studies [72,73].\\nTaken together, these limitations highlight the urgent need for more rigorously designed, multicenter studies with larger sample sizes, homogeneous and well-defined control groups, standardized laboratory workflows, and thorough reporting of biological variables.\\nOnly through such collaborative efforts can we advance toward identifying reliable, non-invasive biomarkers capable of aiding early and accurate diagnosis of endometriosis.\\nIn light of the challenges described above, future research must shift from fragmented, exploratory studies toward collaborative, multicenter projects that follow standardized protocols and include larger, well-defined cohorts. Such studies should not only aim to replicate existing findings but also evaluate the diagnostic value of miRNA signatures in prospective, blinded settings.\\nImportantly, cycle phase stratification and documentation must become routine in all miRNA-related studies, especially when sampling endometrial tissue or blood. Likewise, consensus must be reached on best-practice laboratory techniques—from RNA isolation to quantification and data normalization—to enhance reproducibility and facilitate meta-analyses.\\nThe goal should be to establish validated, reproducible biomarkers that complement current imaging and clinical assessment, thereby reducing diagnostic delay and guiding personalized treatment.\\nUntil such rigorous efforts are undertaken, miRNAs should be considered as promising, but not yet ready-for-practice diagnostic tools. Their clinical translation will depend on overcoming the limitations discussed above and demonstrating consistent, replicable diagnostic performance across diverse populations.',\n",
       "  'Conclusion': '5. Conclusions\\nmiRNAs offer an exciting perspective in the context of developing non-invasive diagnostic biomarkers for endometriosis, but the current state of knowledge requires further research. Expression variability, methodological differences, lack of standardization, and lack of reproducible results in independent studies significantly limit their clinical value.\\nBefore miRNAs find their place in everyday diagnostics, large, multicenter studies with a uniform protocol, standardization of data isolation, analysis, and reporting methods, consideration of biological and clinical variables, integration of miRNA data with other biomarkers (metabolomics, immune), and validation of candidates in prospective studies and ethnically diverse cohorts are necessary.\\nmiRNAs remain a direction worthy of further research, but at present they are more of a scientific illusion than a clinical hope. Initial progress has been made, but considerable research is still required.',\n",
       "  'URL': 'https://doi.org/10.3390/jcm14144849',\n",
       "  'Journal': 'Journal of Clinical Medicine'},\n",
       " {'PaperTitle': 'Maternal plasma microRNAs as potential biomarkers for triaging pregnancies of unknown location and ectopic pregnancy diagnosis',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Christopher Kyriacou, Sung Hye Kim, Maria Arianoglou, Shabnam Bobdiwala, Margaret Pikovsky, Nina Parker, Jennifer Barcroft, Maya Al-Memar, Phillip R Bennett, David A MacIntyre, Tom Bourne, Vasso Terzidou',\n",
       "  'Email': 'v.terzidou@imperial.ac.uk',\n",
       "  'Abstract': 'Background\\nPregnancy of unknown location (PUL) is classified if an early pregnancy is not visualised on transvaginal ultrasonography (TVUS). Biomarkers currently used to triage PUL outcomes have varying accuracy. Delayed or missed diagnosis of ectopic pregnancies (EP) continue to cause significant morbidity and mortality. We investigated whether maternal plasma microRNAs (miRNAs) can predict and differentiate high-risk EP from viable (VIUP) or non-viable (NVIUP) intrauterine pregnancies.\\nMethods\\nPlasma was collected from women with PUL/EP (n\\xa0=\\xa0120), mostly between four to eight weeks’ gestation, where outcomes of EP (n\\xa0=\\xa039), VIUP (n\\xa0=\\xa058) and NVIUP (miscarriage, n\\xa0=\\xa023) were determined using TVUS. Nanostring nCounter miRNA assay was used to examine the expression of ∼800 miRNAs in 22 women. Differentially expressed miRNAs were validated using RT-qPCR in 98 women.\\nResults\\nNanostring nCounter miRNA assay identified 19 miRNAs which were expressed significantly higher in EP/NVIUP compared with VIUP. Two miRNAs were validated in a second, separate validation cohort using RT-qPCR: hsa-miR-21-5p in EP was 2.8-fold higher than in VIUP (\\np\\n=\\xa00.03, ROC AUC\\xa0=\\xa00.64), and hsa-miR-411-5p had 0.2-fold decreased expression (\\np\\n=\\xa00.02, ROC AUC\\xa0=\\xa00.66). Combining the divergent miRNAs as a ratio improved discrimination of EP from VIUP (\\np\\n<\\xa00.001, ROC AUC\\xa0=\\xa00.74).\\nConclusion\\nPlasma miRNAs are differentially expressed in EP and VIUP and are detectable as early as four gestational weeks. Exploring miRNA targets may further understanding of EP pathophysiology, offering the potential to use miRNA as predictive and diagnostic markers in early pregnancy.',\n",
       "  'Introduction': 'Pregnancy of unknown location (PUL) is classified in early pregnancy when a pregnancy is not visualised on transvaginal ultrasonography (TVUS) following a positive urine pregnancy test (UPT) [\\n1\\n]. PUL rates have been reported to vary between five to 42\\xa0% of all pregnancies [\\n[2]\\n,\\n[3]\\n,\\n[4]\\n,\\n[5]\\n]. Management of PUL is based on the triage of these pregnancies to high versus low risk of clinical complications to decide community-based follow-up arrangements pending final diagnosis [\\n1\\n,\\n6\\n]. High-risk outcomes include ectopic pregnancies (EP – pregnancies partially or completely external to the endometrial cavity) or persistent PUL (PPUL – PUL with three or more serial beta-human chorionic gonadotrophin (hCG) values that vary by <15\\xa0%), and low-risk outcomes comprise intrauterine pregnancies (viable – VIUP, or non-viable – NVIUP/miscarriage) or failed PUL (FPUL – PUL with a negative UPT two weeks from initial assessment when presenting with a positive UPT) [\\n1\\n,\\n6\\n]. Ectopic pregnancies continue to carry significant risk of maternal morbidity and mortality if diagnosis is delayed or missed [\\n7\\n].\\nWhilst ectopic pregnancies, viable intrauterine pregnancies and miscarriages are diagnosed using TVUS, persistent PUL and failed PUL are more heterogeneous and require TVUS with the aid of serial hCG values (PPUL) or a negative urine pregnancy test (FPUL) [\\n1\\n]. It is likely that both PPUL and FPUL are very early intrauterine or ectopic pregnancies that are too small to confidently identify on TVUS [\\n6\\n].\\nBiomarkers are used for PUL triage to facilitate screening of cases as efficiently and as accurately as possible pending a final diagnosis. Biomarkers currently available in clinical practice include hCG (a marker of trophoblast function), and progesterone (a marker of ovarian corpus luteal function) [\\n1\\n,\\n8\\n,\\n9\\n]. These are used in isolation, as a ratio (utilising serial measurements over time), or within multinomial logistic regression models [\\n1\\n,\\n[8]\\n,\\n[9]\\n,\\n[10]\\n,\\n[11]\\n,\\n[12]\\n,\\n[13]\\n,\\n[14]\\n,\\n[15]\\n]. The best available model (the M6) discriminates those at high-risk and low-risk of clinical complications well (area under the curve, AUC between 0.86–0.89), however, it has limited population generalisability [\\n9\\n]. Although novel marker candidates have been investigated, many are not appropriate or suitable for clinical use. Biomarkers used in isolation have poor sensitivity or specificity, with much of the evidence based on studies with small numbers, that have not been externally validated [\\n9\\n,\\n[16]\\n,\\n[17]\\n,\\n[18]\\n]. There is thus a clinical need to improve our ability to triage PUL pending diagnosis.\\nMicroRNAs (miRNAs) are single stranded, non-coding, 19–25 nucleotide molecules which regulate messenger RNA stability and transcription [\\n19\\n,\\n20\\n]. A proportion of miRNAs are shed into the circulation in a stable and quantifiable form [\\n19\\n,\\n21\\n,\\n22\\n]. They are important gene regulators, and their expression profiles change with pathophysiology [\\n19\\n,\\n21\\n]. They have been implicated in cardiac, pulmonary, neurological, endocrinological, haematological, rheumatological and oncological processes or pathologies [\\n[23]\\n,\\n[24]\\n,\\n[25]\\n,\\n[26]\\n,\\n[27]\\n,\\n[28]\\n,\\n[29]\\n,\\n[30]\\n,\\n[31]\\n,\\n[32]\\n]. MiRNAs have been investigated in reproductive health and pregnancy outcomes. Outside of pregnancy, they are implicated in reproductive tract and fibroid development, as well as endometriosis [\\n[32]\\n,\\n[33]\\n,\\n[34]\\n,\\n[35]\\n,\\n[36]\\n,\\n[37]\\n]. MiRNAs appear to also play a role in abnormal trophoblast invasion and choriocarcinoma [\\n[38]\\n,\\n[39]\\n,\\n[40]\\n]. Antenatally and intrapartum, miRNAs have been linked to mesenchymal hamartomas, neural tube defects, gestational diabetes, pre-eclampsia, fetal growth restriction, pre-term birth and uterine contractility [\\n19\\n,\\n21\\n,\\n[41]\\n,\\n[42]\\n,\\n[43]\\n,\\n[44]\\n,\\n[45]\\n,\\n[46]\\n,\\n[47]\\n,\\n[48]\\n,\\n[49]\\n,\\n[50]\\n].\\nIn early pregnancy, variable miRNA expression has been associated with blastocyst development, site and success of implantation, and thus miscarriage as well as EP [\\n20\\n,\\n22\\n,\\n31\\n,\\n[51]\\n,\\n[52]\\n,\\n[53]\\n,\\n[54]\\n,\\n[55]\\n,\\n[56]\\n,\\n[57]\\n,\\n[58]\\n,\\n[59]\\n,\\n[60]\\n,\\n[61]\\n]. MiRNAs have been suggested as novel biomarkers that could improve PUL outcome prediction beyond current clinical capabilities [\\n53\\n]. The objective of this study was to evaluate the potential of circulating miRNA markers in pregnancy of unknown location triage and ectopic pregnancy diagnosis.',\n",
       "  'Methods': \"2. Methods\\n2.1. Sex as a biological variable\\nWomen alone were included in this work given the nature and relevance of the conditions under investigation.\\n2.2. Recruitment and sample collection\\nWomen with a positive urine pregnancy test attending the early pregnancy assessment units (EPAU) at Imperial College Healthcare NHS Trust in London, UK, between July 2018 and December 2020 were recruited following classification of PUL or diagnosis of EP, mostly between four to eight weeks' gestation. TVUS\\xa0±\\xa0PUL triage was performed as per local clinical guidelines until an outcome was confirmed, and ectopic pregnancies, miscarriages (NVIUP) and viable intrauterine pregnancies were included in the analysis. VIUP were confirmed at the pregnancy dating scan, usually at 12–14 weeks'gestation. FPUL and PPUL were excluded from analysis due to their heterogeneity (FPUL and PPUL cohorts are likely to contain both failing or persistent intrauterine or ectopic pregnancies that cannot be discerned using TVUS).\\nWhole blood samples were collected in EDTA tubes (K\\n2\\nEDTA BD Vacutainer®), and plasma was isolated by centrifugation at 1962×\\ng\\nfor 10\\xa0min at 4\\xa0°C (Megafuge 40R, Thermo Scientific, Waltham, USA). Isolated plasma was stored in 500\\xa0μL aliquots in RNase-free microtubes at −80\\xa0°C.\\n2.3. Study approval\\nThe samples used in this project were collected following the approval of two studies by the Health Research Authority, the Health and Care Research Wales Research Ethics Committee, and either the North of Scotland Research Ethics Committee, or the London-Dulwich Research Ethics Committee. Their references are 14/NS/1078 (Assessment of Biomarkers in Pregnancy of Unknown Location and Ectopic pregnancy – ABPEP:\\nClinicalTrials.gov\\nNCT04738370\\n) and 19/LO/1154 (Assessment of Biomarkers in Ectopic Pregnancy – AMBER:\\nClinicalTrials.gov\\nNCT04176549\\n). Written informed consent was received prior to participation.\\n2.4. Data availability\\nFindings for this observational cohort study have been reported in accordance with the STROBE guidelines. Raw count data from the Nanostring nCounter miRNA profiling assay are available in\\nSupplementary Table S1\\n.\\n2.5. RNA extraction\\nRNA extraction was performed using Quick-cfRNA Serum & Plasma Kit (Zymo Research, Orange, USA – RRID:\\nSCR_008968\\n). Plasma samples were thawed on ice prior to centrifugation at 12,000×\\ng\\nfor 15\\xa0min to remove any cell debris or precipitates as per manufacturer's instructions. The upper 200\\xa0μL of plasma was used for the RNA extraction to avoid cellular and/or platelet contamination. Samples were evaluated for haemolysis macroscopically, and via the use of spectrophotometric measurement of haemoglobin at 414\\xa0nm using Nanodrop 1000 (Thermo Scientific, Waltham, USA – RRID:\\nSCR_016517\\n).\\nIn accordance with the manufacturer's protocol, the spike-in control cel-miR-254 (Qiagen, Hilden, Germany – RRID:\\nSCR_008539\\n) was added to each sample following lysis and denaturation of plasma with Quick-cfRNA Digestion Buffer to allow normalisation of any technical variation that may have occurred during RNA extraction.\\n2.6. nCounter microRNA assay and data analysis\\nThe RNA samples for the discovery cohort were evaluated using nCounter cell-free miRNA profiling (Nanostring, Seattle, USA – RRID:\\nSCR_021712\\n), allowing direct fluorescence-based molecular barcoding of 798 individual miRNAs per sample without amplification. The nCounter Human v3 miRNA panel was used for each sample and it accounts for more than 95\\xa0% of all observed sequencing reads in miRbase version 21 (\\nhttps://www.mirbase.org/\\n), including all miRNAs denoted as ‘high confidence’ [\\n62\\n].\\nCounts of miRNAs were analysed using nSolver version 4.0 (Nanostring, Seattle, USA – RRID:\\nSCR_003420\\n) software, with normalisation to the 100 most expressed targets. Normalised miRNA counts were then compared between the outcome groups, EP, NVIUP, and VIUP. The miRNAs expressed below background level in more than 50\\xa0% of at least one outcome group were excluded from analysis (background level defined as two standard deviations above mean negative control counts). The raw count data is available in\\nSupplementary Table S1\\n.\\n2.7. Reverse transcription, real-time quantitative polymerase chain reaction (RT-qPCR) and data analysis\\nReverse transcription and RT-qPCR was used for technical validation of the discovery work, as well as for the biological validation study which utilised a second, separate validation cohort. Following a literature review, three additional miRNA markers were also added to the biological validation work. Reverse transcription of the extracted RNA samples was achieved using the miRCURY LNA Universal\\nRT\\nmiRNA cDNA synthesis kits following the manufacturer's instructions (Qiagen). As part of the protocol, the spike-in control, UniSp6, was added (Qiagen) to allow normalisation of any variation during reverse transcription.\\nEndogenous miRNA controls were identified using NormFinder (Microsoft, Redmond, USA – RRID:\\nSCR_003387\\n) software. The most stable combination of miRNAs was chosen for the normalisation of RT-qPCR data [\\n63\\n]. The stability values created by NormFinder using miRNA counts profiled using nCounter and analysed using nSolver (Nanosting) reflect the intra- and inter-group variation in the expression of each individual and combination of miRNAs. The combination of two miRNAs with greatest stability were used. The NormFinder analysis is also available in\\nSupplementary Table S1\\n.\\nRT-qPCR was performed as per manufacturer's instructions using custom PCR panels with embedded LNA™ primers (Qiagen, Hilden, Germany – RRID:\\nSCR_008539\\n), and miRCURY LNA SYBR® Green master mix (Qiagen, Hilden, Germany – RRID:\\nSCR_008539\\n. RT-qPCR was performed in duplicate for the technical validation samples.\\nRT-qPCR reactions were carried out on the StepOnePlus™ Real-Time PCR system (Applied Biosystems, Waltham, USA – RRID:\\nSCR_015805\\n). The cycle conditions included initial PCR heat activation at 95\\xa0°C for 2\\xa0min, followed by 45 cycles of denaturation at 95\\xa0°C for 10\\xa0s and annealing/extension at 56\\xa0°C for 60\\xa0s. Melt curve analyses from 60 to 95\\xa0°C confirmed a single PCR product.\\nRaw fluorescence data were collected using the StepOne Software version 2.3 (Applied Biosystems, Waltham, USA – RRID:\\nSCR_014281\\n) for evaluation of Cq values and primer efficiencies using the LinRegPCR program version 2017.1 [\\n64\\n]. Following this, Cq data was processed by normalising to the inter-plate calibrator (UniSp3), UniSp6 reverse transcription control, cel-miR-254 RNA extraction control and the endogenous controls. Univariate analyses were then performed to compare miRNA fold changes between the different outcome groups. Any participant samples or miRNAs that did not consistently amplify were excluded. For the remainder, any normalised Cq values above 45 or below one were given a value of 45 (the maximum number of cycles in PCR reaction).\\n2.8. Statistics\\nStatistical software used included nSolver version 4.0 (Nanostring) and PRISM version 8.0 (GraphPad, San Diego, USA – RRID:\\nSCR_002798\\n). Demographics (age, ethnicity, gestational age to initial TVUS, and Body Mass Index – BMI) for all patient groups were tested for normality using the Shapiro-Wilk test with mean (standard deviation) and median (interquartile range) appropriately presented. One-way ANOVA/Kruskal-Wallis analyses were selected depending on data distribution, with categorical data analyses using Fisher's. nSolver (Nanostring) statistical analysis allowed comparison of log-transformed geometric means of expression counts assuming unequal variance between different outcome groups in the discovery work using an unpaired two-tailed\\nt\\n-test. MiRNAs with\\np\\n<\\xa00.05 and a false discovery rate (FDR)\\xa0<\\xa00.05 were considered to be differentially expressed between outcomes. Count data were plotted on the logarithmic scale using geometric mean and geometric standard deviation (SD), with statistically significant\\np\\nvalues presented. For the technical and biological validation work, univariate analyses of the relative expression fold changes for each miRNA were compared between outcome groups. Data from all miRNA candidates in each outcome group were tested for normality using the Shapiro-Wilk test, and unpaired two-tailed\\nt\\n-test/Mann-Whitney U analysis was selected depending on data distribution. Fold change data were plotted on the logarithmic scale using geometric mean and geometric SD, with statistically significant\\np\\nvalues presented. For ROC curves, area under the ROC curve (AUC) with 95\\xa0% confidence intervals (95\\xa0% CI) were also included. Significance was again determined by\\np\\nvalue, with a level <0.05 deemed significant. Whilst no formal power calculation was performed due to the discovery nature of the study, the study population numbers were comparable to similar published work exploring miRNAs in fetal growth restriction and preterm birth [\\n19\\n,\\n21\\n].\",\n",
       "  'Results': \"3. Results\\n3.1. Clinical outcome and patient characteristics\\nA total of 290 women were recruited from EPAU and 120 women (41.4\\xa0%) with outcomes of either ectopic pregnancy, miscarriage (NVIUP) or viable intrauterine pregnancy were used for further analysis (\\nFig. 1\\n). 170 women (58.6\\xa0%) were excluded from this study due to their outcome: 141 failed PUL, 9 persistent PUL, 8 IUP of uncertain viability and 12 lost to follow up (LTFU). For the discovery cohort and technical validation using RT-qPCR, 22 women (18.3\\xa0%) were selected for analysis, where 7 had EP, 7 NVIUP, and 8 VIUP. One ectopic pregnancy sample did not consistently amplify and was excluded from the technical validation. For the validation cohort, 98 women (81.7\\xa0%) were selected where 32 were EP, 16 NVIUP and 50 VIUP. One miscarriage sample subsequently did not consistently amplify and was excluded.\\nFig. 1.\\nOpen in a new tab\\nFlowchart of patient recruitment and breakdown of data available for discovery, technical and biological validation studies\\nAbbreviations: FPUL: Failed PUL; LTFU: Lost to follow up; PPUL: Persistent PUL; IPUV: Intrauterine pregnancy of unknown viability; PUL: Pregnancy of unknown location; EP: Ectopic pregnancy; NVIUP: Non-viable intrauterine pregnancy; VIUP: Viable intrauterine pregnancy.\\nThe demographics for all three outcome groups for the separate discovery and biological validation cohorts were similar, except for the participants age in the biological validation cohort when comparing NVIUP with VIUP (\\np\\n=\\xa00.02, Kruskal-Wallis analysis), and when comparing ‘Black’ and ‘Other’ ethnicities within viable intrauterine and ectopic pregnancy cohorts (\\np\\n=\\xa00.04, Fisher's exact test) (\\nTable 1\\n,\\nTable 2\\n). The presence of ectopic pregnancy risk factors within the outcome groups of both the discovery and biological validation cohorts are described in\\nSupplementary Tables S2 and S3\\n, with the demographics and characteristics of each outcome group further divided by the ethnicity groups of\\nTable 1\\n,\\nTable 2\\nin\\nSupplementary Tables S4 and S5\\n.\\nTable 1.\\nDemographics of discovery and technical validation cohort (N\\xa0=\\xa022) for microRNA analysis, divided by outcome.\\nDemographics\\nVIUP (N\\xa0=\\xa08)\\nEP (N\\xa0=\\xa07)\\nNVIUP (N\\xa0=\\xa07)\\np\\n-value\\nMean age - years (SD)\\n32 (5.9)\\n35.3 (3.9)\\n34.4 (6.4)\\n0.60; 0.77\\nIndividual\\np\\n-values\\n0.60, 0.65\\n0.60, 0.77\\n0.65, 0.77\\nEthnicity N (%)\\nCaucasian\\n5 (62.5)\\n4 (57.1)\\n3 (42.9)\\n0.44; 0.99\\nAsian\\n3 (37.5)\\n2 (28.6)\\n3 (42.9)\\nBlack\\n0 (0.0)\\n1 (14.3)\\n0 (0.0)\\nOther\\n0 (0.0)\\n0 (0.0)\\n1 (14.3)\\nIndividual\\np\\n-value ranges\\n0.44–0.99\\n0.99\\n0.44–0.99\\nMean gestational age - days (SD)\\n32.3 (4.8)\\n37.3 (12.9)\\n44.6 (11.8)\\n0.092; 0.36\\nIndividual\\np\\n-values\\n0.092, 0.36\\n0.36\\n0.092, 0.36\\nMedian BMI - kg/m\\n2\\n(IQR)\\n24.8 (21.5–31.3)\\n24.4 (21.9–31.2)\\n25.8 (22.2–32.0)\\na\\n0.99\\nIndividual\\np\\n-values\\n0.99\\n0.99\\n0.99\\nOpen in a new tab\\np-value corresponds to one-way ANOVA (parametric, continuous), Kruskal-Wallis (non-parametric, continuous), or Fisher's (categorical) for the difference in study participants' demographics between outcome groups. For Fisher's, each outcome and ethnicity group were individually tested against one another. Individual\\np\\n-values are specific to each outcome group under comparison, with overall\\np\\n-value ranges listed in the end column.\\nAbbreviations: SD = Standard deviation; IQR = Interquartile range; VIUP = Viable intrauterine pregnancy; EP\\xa0=\\xa0Ectopic pregnancy; NVIUP = Non-viable intrauterine pregnancy; NS = Not significant; BMI = Body mass index.\\na\\nBMI not available for 1 NVIUP.\\nTable 2.\\nDemographics of biological validation cohort (N\\xa0=\\xa097) for microRNA analysis, divided by outcome.\\nDemographics\\nVIUP (N\\xa0=\\xa050)\\nEP (N\\xa0=\\xa032)\\nNVIUP (N\\xa0=\\xa015)\\np\\n-value\\nMedian age - years (IQR)\\n31 (25.8–35)\\n32 (27–37.5)\\n34 (31–42)\\na\\n0.018\\na\\n; 0.68\\nIndividual\\np\\n-values\\n0.018\\na\\n, 0.68\\n0.26, 0.68\\n0.018\\na\\n, 0.26\\nEthnicity N (%)\\nCaucasian\\n28 (56.0)\\n16 (50.0)\\n7 (46.7)\\n0.041\\nb\\n; 0.99\\nAsian\\n7 (14.0)\\n6 (18.8)\\n3 (20.0)\\nBlack\\n8 (16.0)\\nb\\n1 (3.1)\\nb\\n3 (20.0)\\nOther\\n7 (14.0)\\nb\\n9 (28.1)\\nb\\n2 (13.3)\\nIndividual\\np\\n-value ranges\\n0.041\\nb\\n–0.99\\n0.041\\nb\\n–0.99\\n0.077–0.99\\nMedian gestational age - days (IQR)\\n33 (31.0–38.5)\\nc\\n42.5 (27.5–48.8)\\nd\\n40 (34.0–44.0)\\n0.18; 0.99\\nIndividual\\np\\n-values\\n0.18, 0.47\\n0.18, 0.99\\n0.47, 0.99\\nMedian BMI - kg/m\\n2\\n(IQR)\\n26.4 (22.8–28.8)\\ne\\n23.7 (20.6–26.7)\\nf\\n24 (20.3–28.3)\\n0.46; 0.99\\nIndividual\\np\\n-values\\n0.46, 0.60\\n0.46, 0.99\\n0.60, 0.99\\nOpen in a new tab\\np-value corresponds to one-way ANOVA (parametric, continuous), Kruskal-Wallis (non-parametric, continuous), or Fisher's (categorical) for the difference in study participants' demographics between outcome groups. For Fisher's, each outcome and ethnicity group were individually tested against one another. Individual\\np\\n-values are specific to each outcome group under comparison, with overall\\np\\n-value ranges listed in the end column.\\nAbbreviations: IQR = Interquartile range; VIUP = Viable intrauterine pregnancy; EP\\xa0=\\xa0Ectopic pregnancy; NVIUP = Non-viable intrauterine pregnancy; NS = Not significant; BMI = Body mass index.\\na\\nSignificant difference comparing NVIUP with VIUP.\\nb\\nSignificant difference comparing ‘Black’ and ‘Other’ ethnicities within VIUP and EP cohorts.\\nc\\nGestational age not available for 4 VIUP.\\nd\\nGestational age not available for 4\\xa0EP.\\ne\\nBMI not available for 1 VIUP.\\nf\\nBMI not available for 1\\xa0EP.\\n3.2. Plasma miRNA expression profiling using nCounter assay\\nA total of 62 miRNAs were found to be expressed above the background in more than 50\\xa0% of samples in any one of the outcome groups (\\nTable S6\\n). The expression of these miRNAs were compared between outcome groups using nSolver (Nanostring) with 19 miRNAs identified to be significantly different between control (VIUP) and ectopic pregnancy\\xa0±\\xa0miscarriage (NVIUP) outcome groups (\\nTable 3\\n,\\nTable 4\\n,\\nTable 5\\n). There were eight miRNAs expressed significantly higher only in EP compared to VIUP (\\nTable 3\\n,\\nFig. 2\\n), seven only in NVIUP when compared to VIUP (\\nTable 4\\n,\\nFig. 3\\n), and four miRNAs which were increased in both EP and NVIUP when compared to VIUP (\\nTable 5\\n,\\nFig. 4\\n).\\nTable 3.\\nDiscovery microRNA markers meeting significance when comparing viable intrauterine pregnancy with ectopic pregnancy (N\\xa0=\\xa015) using nCounter.\\nMicroRNA\\nVIUP (N\\xa0=\\xa08) vs EP (N\\xa0=\\xa07) Ratio (95\\xa0% CI)\\nRelative change\\np\\n-value\\nhsa-miR-222-3p\\n4.5∗ (1.7–12.1)\\nEP\\xa0>\\xa0VIUP\\n0.0060\\nhsa-miR-20a-5p\\xa0+\\xa0hsa-miR-20b-5p\\na\\n6.0∗ (1.4–25.0)\\nEP\\xa0>\\xa0VIUP\\n0.019\\nhsa-miR-21-5p\\n5.4∗ (1.3–23.0)\\nEP\\xa0>\\xa0VIUP\\n0.027\\nhsa-miR-185-5p\\n4.8∗ (1.2–18.8)\\nEP\\xa0>\\xa0VIUP\\n0.028\\nhsa-miR-148a-3p\\n5.1∗ (1.1–23.2)\\nEP\\xa0>\\xa0VIUP\\n0.036\\nhsa-miR-19b-3p\\n4.5∗ (1.1–18.8)\\nEP\\xa0>\\xa0VIUP\\n0.038\\nhsa-miR-1285-5p\\n4.4∗ (1.1–18.1)\\nEP\\xa0>\\xa0VIUP\\n0.038\\nhsa-miR-22-3p\\n4.6∗ (1.0–20.5)\\nEP\\xa0>\\xa0VIUP\\n0.044\\nOpen in a new tab\\np-value corresponds to unpaired two-tailed\\nt\\n-test, for the difference between outcome groups.\\nOrdered by\\np\\n-value; ∗Designates ratio significance.\\nAbbreviations: 95\\xa0% CI\\xa0=\\xa095\\xa0% confidence interval; VIUP = Viable intrauterine pregnancy; EP\\xa0=\\xa0Ectopic pregnancy.\\na\\nNanostring assay ran both these targets together.\\nTable 4.\\nDiscovery microRNA markers meeting significance when comparing viable intrauterine pregnancy with non-viable intrauterine pregnancy (N\\xa0=\\xa015) using nCounter.\\nMicroRNA\\nVIUP (N\\xa0=\\xa08) vs NVIUP (N\\xa0=\\xa07) Ratio (95\\xa0% CI)\\nRelative change\\np\\n-value\\nhas-miR-107\\n5.3∗ (1.8–15.2)\\nNVIUP\\xa0>\\xa0VIUP\\n0.0047\\nhsa-miR-223-3p\\n2.0∗ (1.1–3.7)\\nNVIUP\\xa0>\\xa0VIUP\\n0.028\\nhsa-let-7d-5p\\n3.9∗ (1.1–13.9)\\nNVIUP\\xa0>\\xa0VIUP\\n0.035\\nhsa-miR-191-5p\\n2.5∗ (1.1–5.8)\\nNVIUP\\xa0>\\xa0VIUP\\n0.037\\nhsa-miR-29b-3p\\n4.5∗ (1.0–19.9)\\nNVIUP\\xa0>\\xa0VIUP\\n0.045\\nhsa-miR-374a-5p\\n1.6∗ (1.0–2.5)\\nNVIUP\\xa0>\\xa0VIUP\\n0.046\\nhsa-miR-199a-3p\\xa0+\\xa0hsa-miR-199b-3p\\na\\n3.7∗ (1.0–13.2)\\nNVIUP\\xa0>\\xa0VIUP\\n0.047\\nOpen in a new tab\\np-value corresponds to unpaired two-tailed\\nt\\n-test, for the difference between outcome groups.\\nOrdered by\\np\\n-value; ∗Designates ratio significance.\\nAbbreviations: 95\\xa0% CI\\xa0=\\xa095\\xa0% confidence interval; VIUP = Viable intrauterine pregnancy; NVIUP = Non-viable intrauterine pregnancy.\\na\\nNanostring assay ran both these targets together.\\nTable 5.\\nDiscovery microRNA markers meeting significance when comparing viable intrauterine pregnancy with both ectopic pregnancy and non-viable intrauterine pregnancy (N\\xa0=\\xa022) using nCounter.\\nMicroRNA\\nVIUP (N\\xa0=\\xa08) vs EP (N\\xa0=\\xa07) Ratio (95\\xa0% CI)\\nVIUP (N\\xa0=\\xa08) vs NVIUP (N\\xa0=\\xa07) Ratio (95\\xa0% CI)\\nRelative change\\np\\n-value\\nhsa-miR-1246\\n13.9∗ (2.8–70.3)\\n6.0∗ (1.2–29.0)\\nEP\\xa0>\\xa0VIUP\\n0.0062\\nNVIUP\\xa0>\\xa0VIUP\\n0.030\\nhsa-miR-130a-3p\\n5.7∗ (1.3–26.0)\\n8.2∗ (1.9–35.2)\\nEP\\xa0>\\xa0VIUP\\n0.026\\nNVIUP\\xa0>\\xa0VIUP\\n0.0090\\nhsa-miR-320e\\n5.3∗ (1.4–20.6)\\n3.9∗ (1.3–11.2)\\nEP\\xa0>\\xa0VIUP\\n0.020\\nNVIUP\\xa0>\\xa0VIUP\\n0.017\\nhsa-miR-30d-5p\\n3.0∗ (1.2–7.3)\\n3.3∗ (1.2–9.1)\\nEP\\xa0>\\xa0VIUP\\n0.022\\nNVIUP\\xa0>\\xa0VIUP\\n0.023\\nOpen in a new tab\\np\\n-value corresponds to unpaired two-tailed\\nt\\n-test, for the difference between outcome groups.\\nOrdered by\\np\\n-value of EP/NVIUP vs VIUP for each microRNA; ∗Designates ratio significance.\\nAbbreviations: 95\\xa0% CI\\xa0=\\xa095\\xa0% confidence interval; VIUP = Viable intrauterine pregnancy; EP\\xa0=\\xa0Ectopic pregnancy; NVIUP = Non-viable intrauterine pregnancy.\\nFig. 2.\\nOpen in a new tab\\nEight discriminatory miRNAs in early pregnancy plasma when comparing VIUP with EP (N\\xa0=\\xa015) in discovery study. nCounter probe counts for A: hsa-miR-222-3p, B: hsa-miR-20a-5p\\xa0+\\xa0hsa-miR-20b-5p, C: hsa-miR-21-5p, D: hsa-miR-185-5p, E: hsa-miR-148a-3p, F: hsa-miR-19b-3p, G: hsa-miR-1285-5p, and H: hsa-miR-22-3p detected in human plasma in women in early pregnancy who subsequently had a confirmed VIUP (N\\xa0=\\xa08, green), EP (N\\xa0=\\xa07, red), or NVIUP (N\\xa0=\\xa07, purple)\\n∗Statistical significance of differences between groups was tested using Nanostring nSolver platform. Graphs show geometric mean with geometric standard deviation and\\np\\n-values, with outcome on x axis, and log(10) counts on y axis. Only statistically significant differences are presented.\\nAbbreviations: VIUP = Viable intrauterine pregnancy; EP\\xa0=\\xa0Ectopic pregnancy; NVIUP = Non-viable intrauterine pregnancy.\\nFig. 3.\\nOpen in a new tab\\nSeven discriminatory miRNAs in early pregnancy plasma when comparing VIUP with NVIUP (N\\xa0=\\xa015) in discovery study. nCounter probe counts for A: hsa-miR-107, B: hsa-miR-223-3p, C: hsa-let-7d-5p, D: hsa-miR-191-5p, E: hsa-miR-29b-3p, F: hsa-miR-374a-5p, and G: hsa-miR-199a-3p\\xa0+\\xa0hsa-miR-199b-3p detected in human plasma in women in early pregnancy who subsequently had a confirmed VIUP (N\\xa0=\\xa08, green), EP (N\\xa0=\\xa07, red), or NVIUP (N\\xa0=\\xa07, purple)\\n∗Statistical significance of differences between groups was tested using Nanostring nSolver platform. Graphs show geometric mean with geometric standard deviation and\\np\\n-values, with outcome on x axis, and log(10) counts on y axis. Only statistically significant differences are presented.\\nAbbreviations: VIUP = Viable intrauterine pregnancy; EP\\xa0=\\xa0Ectopic pregnancy; NVIUP = Non-viable intrauterine pregnancy.\\nFig. 4.\\nOpen in a new tab\\nFour discriminatory miRNAs in early pregnancy plasma when comparing VIUP with EP and NVIUP (N\\xa0=\\xa022) in discovery study. nCounter probe counts for A: hsa-miR-1246, B: hsa-miR-130a-3p, C: hsa-miR-320e, and D: hsa-miR-30d-5p detected in human plasma in women in early pregnancy who subsequently had a confirmed VIUP (N\\xa0=\\xa08, green), EP (N\\xa0=\\xa07, red), or NVIUP (N\\xa0=\\xa07, purple)\\n∗Statistical significance of differences between groups was tested using Nanostring nSolver platform. Graphs show geometric mean with geometric standard deviation and\\np\\n-values, with outcome on x axis, and log(10) counts on y axis. Statistically significant differences are presented.\\nAbbreviations: VIUP = Viable intrauterine pregnancy; EP\\xa0=\\xa0Ectopic pregnancy; NVIUP = Non-viable intrauterine pregnancy.\\n3.3. Technical validation of nCounter microRNA profiling using RT-qPCR\\nWe performed technical validation of the 19 miRNAs identified using nCounter (Nanostring). Two miRNAs, hsa-miR-126-3p and hsa-let-7g-5p were identified to have the highest inter- and intra-group stability using NormFinder and were used as endogenous controls [\\n63\\n]. Univariate analysis of the DeltaCq values for each miRNA candidate are shown in\\nTable 3\\n. From the 19 miRNAs examined, one miRNA (hsa-miR-21-5p) showed significant increase in EP compared to VIUP (\\nTable 6\\n), and two miRNAs (hsa-miR-130a-3p and hsa-miR-374a-5p) were found to be significantly increased in NVIUP compared to VIUP (\\nTable 7\\n,\\nTable 8\\n).\\nTable 6.\\nTechnical validation of microRNA markers comparing viable intrauterine pregnancy with ectopic pregnancy (N\\xa0=\\xa014) using reverse transcription quantitative polymerase chain reaction (RT-qPCR).\\nMicroRNA\\nVIUP (N\\xa0=\\xa08)\\nEP (N\\xa0=\\xa06)\\nVIUP vs EP Fold change\\nRelative change\\np\\n-value\\nhas-miR-21-5p\\n1.0 (0.28)\\n3.2 (2.75)\\n3.2∗\\nEP\\xa0>\\xa0VIUP\\n0.0013\\nhsa-miR-222-3p\\n1.0 (0.79)\\n1.8 (1.01)\\n1.8\\nEP\\xa0>\\xa0VIUP\\n0.12\\nhsa-miR-148a-3p\\n1.0 (0.47)\\n2.2 (2.05)\\n2.2\\nEP\\xa0>\\xa0VIUP\\n0.13\\nhsa-miR-185-5p\\n1.0 (0.28)\\n1.2 (0.57)\\n1.2\\nEP\\xa0>\\xa0VIUP\\n0.45\\nhsa-miR-20a-5p\\n1.0 (0.43)\\n1.2 (0.52)\\n1.2\\nEP\\xa0>\\xa0VIUP\\n0.52\\nhsa-miR-19b-3p\\n1.0 (0.48)\\n1.2 (0.81)\\n1.2\\nEP\\xa0>\\xa0VIUP\\n0.54\\nhsa-miR-1285-5p\\n1.0 (0.37)\\n0.9 (0.45)\\n0.9\\nEP\\xa0<\\xa0VIUP\\n0.57\\nhsa-miR-20b-5p\\n1.0 (0.32)\\n1.1 (0.58)\\n1.1\\nEP\\xa0>\\xa0VIUP\\n0.79\\nhsa-miR-22-3p\\n1.0 (0.45)\\n1.2 (0.89)\\n1.2\\nEP\\xa0>\\xa0VIUP\\n0.85\\nOpen in a new tab\\nMean DeltaCq (SD) presented in outcome columns.\\np\\n-value corresponds to unpaired two-tailed\\nt\\n-test (parametric, continuous), or Mann-Whitney U (non-parametric, continuous), for the difference in fold change between outcome groups.\\nOrdered by\\np\\n-value; ∗Designates relative expression fold change significance.\\nAbbreviations: VIUP = Viable intrauterine pregnancy; EP\\xa0=\\xa0Ectopic pregnancy; SD = Standard deviation.\\nTable 7.\\nTechnical validation of microRNA markers comparing viable intrauterine pregnancy with non-viable intrauterine pregnancy (N\\xa0=\\xa015) using reverse transcription quantitative polymerase chain reaction (RT-qPCR).\\nMicroRNA\\nVIUP (N\\xa0=\\xa08)\\nNVIUP (N\\xa0=\\xa07)\\nVIUP vs NVIUP Fold change\\nRelative change\\np\\n-value\\nhsa-miR-374a-5p\\n1.0 (0.14)\\n2.0 (1.20)\\n2.0∗\\nNVIUP\\xa0>\\xa0VIUP\\n0.0022\\nhsa-miR-199a-3p\\n1.0 (0.87)\\n2.0 (1.34)\\n2.0\\nNVIUP\\xa0>\\xa0VIUP\\n0.054\\nhsa-miR-107\\n1.0 (0.32)\\n1.5 (0.52)\\n1.5\\nNVIUP\\xa0>\\xa0VIUP\\n0.058\\nhsa-miR-223-3p\\n1.0 (0.77)\\n1.6 (1.09)\\n1.6\\nNVIUP\\xa0>\\xa0VIUP\\n0.072\\nhsa-miR-29b-3p\\n1.0 (0.35)\\n1.2 (0.22)\\n1.2\\nNVIUP\\xa0>\\xa0VIUP\\n0.19\\nhsa-miR-191-5p\\n1.0 (0.25)\\n1.3 (0.62)\\n1.3\\nNVIUP\\xa0>\\xa0VIUP\\n0.28\\nhsa-let-7d-5p\\n1.0 (0.60)\\n0.9 (0.36)\\n0.9\\nNVIUP\\xa0<\\xa0VIUP\\n0.70\\nOpen in a new tab\\nMean DeltaCq (SD) presented in outcome columns.\\np\\n-value corresponds to unpaired two-tailed\\nt\\n-test (parametric, continuous), or Mann-Whitney U (non-parametric, continuous), for the difference in fold change between outcome groups.\\nOrdered by\\np\\n-value; ∗Designates relative expression fold change significance.\\nAbbreviations: VIUP = Viable intrauterine pregnancy; NVIUP = Non-viable intrauterine pregnancy; SD = Standard deviation.\\nTable 8.\\nTechnical validation of microRNA markers comparing viable intrauterine pregnancy with both ectopic pregnancy and non-viable intrauterine pregnancy (N\\xa0=\\xa021) using reverse transcription quantitative polymerase chain reaction (RT-qPCR).\\nMicroRNA\\nVIUP (N\\xa0=\\xa08)\\nEP (N\\xa0=\\xa06)\\nNVIUP (N\\xa0=\\xa07)\\nVIUP vs EP Fold change\\nVIUP vs NVIUP Fold change\\nRelative change\\np\\n-value\\nhsa-miR-130a-3p\\n1.0 (0.61)\\n1.3 (0.57)\\n2.2 (1.61)\\n1.3\\n2.2∗\\nEP\\xa0>\\xa0VIUP\\n0.14\\nNVIUP\\xa0>\\xa0VIUP\\n0.014\\nhsa-miR-30d-5p\\n1.0 (0.38)\\n1.7 (1.01)\\n1.5 (0.68)\\n1.7\\n1.5\\nEP\\xa0>\\xa0VIUP\\n0.11\\nNVIUP\\xa0>\\xa0VIUP\\n0.094\\nhsa-miR-1246\\n1.0 (0.63)\\n1.3 (1.36)\\n0.5 (0.29)\\n1.3\\n0.5\\nEP\\xa0>\\xa0VIUP\\n0.95\\nNVIUP\\xa0<\\xa0VIUP\\n0.10\\nhsa-miR-320e\\n1.0 (0.62)\\n1.8 (1.43)\\n1.4 (1.20)\\n1.8\\n1.4\\nEP\\xa0>\\xa0VIUP\\n0.41\\nNVIUP\\xa0>\\xa0VIUP\\n0.96\\nOpen in a new tab\\nMean DeltaCq (SD) presented in outcome columns.\\np\\n-value corresponds to unpaired two-tailed\\nt\\n-test (parametric, continuous), or Mann-Whitney U (non-parametric, continuous), for the difference in fold change between outcome groups.\\nOrdered by\\np\\n-value of EP/NVIUP vs VIUP for each microRNA; ∗Designates relative expression fold change significance.\\nAbbreviations: VIUP = Viable intrauterine pregnancy; EP\\xa0=\\xa0Ectopic pregnancy; NVIUP = Non-viable intrauterine pregnancy; SD = Standard deviation.\\n3.4. Biological validation of microRNA candidates using RT-qPCR\\nBiological validation was performed for the three technically validated miRNAs (hsa-miR-21-5p, hsa-miR-130a-3p and hsa-miR-374a-5p) in a separate patient population utilised as a second validation cohort (\\nFig. 1\\n, n\\xa0=\\xa098). Three additional miRNA markers were also added following literature review: hsa-miR-411-5p, hsa-miR-324-3p and hsa-miR-873-3p [\\n52\\n,\\n54\\n,\\n57\\n,\\n59\\n]. These three additional miRNAs were reported to be expressed at lower levels in ectopic pregnancy when compared to viable intrauterine pregnancy (or termination of pregnancy patients). Whilst this trend was confirmed by the findings of the nCounter assay (\\nTable S1\\n), where the differences in Nanostring counts between EP and VIUP were either significant (hsa-miR-411-5p,\\np\\n=\\xa00.040) or borderline (hsa-miR-324-3p,\\np\\n=\\xa00.090; hsa-miR-873-3,\\np\\n=\\xa00.069), they were excluded from further discovery cohort analysis due to their low levels of expression (below background level in more than 50\\xa0% of at least one outcome group). However, in this separate biological validation, we aimed to exploit the inverse expression relationship between these miRNAs and the increasing miRNAs identified in the discovery cohort, to improve the discrimination of ectopic pregnancy.\\nTwo of the six miRNAs examined (hsa-miR-21-5p and hsa-miR-411-5p) were biologically validated (\\nTable 9\\n). Hsa-miR-21-5p was significantly increased in EP compared to VIUP (\\np\\n=\\xa00.035) with a mean fold change of 2.8 (SD 5.46), and hsa-miR-411-5p was decreased in EP (\\np\\n=\\xa00.016) with a mean fold change of 0.2 (SD 0.20) (\\nTable 9\\n,\\nFig. 5\\n). Hsa-miR-130a-3p, hsa-miR-374a-5p and hsa-miR-324-3p did not show any significant differences between outcome groups (\\nTable 9\\n,\\nTable 10\\n,\\nFig. S1\\n), and hsa-miR-873-3p failed to amplify during RT-qPCR. Six of the 485 normalised Cq values obtained from the five miRNAs that did amplify were given a value of 45 during the analysis (1.2\\xa0%).\\nTable 9.\\nBiological validation of microRNA markers comparing viable intrauterine pregnancy with ectopic pregnancy (N\\xa0=\\xa082) using reverse transcription quantitative polymerase chain reaction (RT-qPCR).\\nMicroRNA\\nVIUP (N\\xa0=\\xa050)\\nEP (N\\xa0=\\xa032)\\nVIUP vs EP Fold change\\nRelative change\\np\\n-value\\nhsa-miR-411-5p\\n1.0 (3.65)\\n0.2 (0.20)\\n0.2∗\\nEP\\xa0<\\xa0VIUP\\n0.016\\nhsa-miR-21-5p\\n1.0 (0.44)\\n2.8 (5.46)\\n2.8∗\\nEP\\xa0>\\xa0VIUP\\n0.035\\nhsa-miR-324-3p\\n1.0 (0.83)\\n1.0 (0.74)\\n1.0\\nEP = VIUP\\n0.62\\nOpen in a new tab\\nMean DeltaCq (SD) presented in outcome columns.\\np\\n-value corresponds to Mann-Whitney U (non-parametric, continuous). For the difference in fold change between outcome groups.\\nOrdered by\\np\\n-value; ∗Designates relative expression fold change significance.\\nAbbreviations: VIUP = Viable intrauterine pregnancy; EP\\xa0=\\xa0Ectopic pregnancy; SD = Standard deviation.\\nFig. 5.\\nOpen in a new tab\\nTwo discriminatory miRNAs in early pregnancy plasma when comparing VIUP with EP (N\\xa0=\\xa082) in biological validation study. Fold changes for A: hsa-miR-21-5p, and B: hsa-miR-411-5p in women in early pregnancy who subsequently had a confirmed VIUP (N\\xa0=\\xa050, green), or EP (N\\xa0=\\xa032, red)\\n∗Statistical significance of differences between groups was tested using Mann-Whitney (non-parametric, continuous). Graphs show geometric mean with geometric standard deviation and\\np\\n-values, with outcome on x axis, and log(10) fold change on y axis. Statistically significant differences are presented.\\nAbbreviations: VIUP = Viable intrauterine pregnancy; EP\\xa0=\\xa0Ectopic pregnancy.\\nTable 10.\\nBiological validation of microRNA markers comparing viable intrauterine pregnancy with non-viable intrauterine pregnancy (N\\xa0=\\xa065) using reverse transcription quantitative polymerase chain reaction (RT-qPCR).\\nMicroRNA\\nVIUP (N\\xa0=\\xa050)\\nNVIUP (N\\xa0=\\xa015)\\nVIUP vs NVIUP Fold change\\nRelative change\\np\\n-value\\nhsa-miR-374a-5p\\n1.0 (0.42)\\n0.9 (0.51)\\n0.9\\nNVIUP\\xa0<\\xa0VIUP\\n0.26\\nhsa-miR-130a-3p\\n1.0 (0.60)\\n0.9 (0.39)\\n0.9\\nNVIUP\\xa0<\\xa0VIUP\\n0.68\\nOpen in a new tab\\nMean DeltaCq (SD) presented in outcome columns.\\nP\\n-value corresponds to unpaired two-tailed\\nt\\n-test (parametric, continuous), or Mann-Whitney U (non-parametric, continuous), for the difference in fold change between outcome groups.\\nOrdered by\\np\\n-value; Relative expression fold change presented.\\nAbbreviations: VIUP = Viable intrauterine pregnancy; NVIUP = Non-viable intrauterine pregnancy; SD = Standard deviation.\\n3.5. Assessment of the predictive ability and diagnostic test performance of identified miRNA markers\\nThe predictive ability of the two biologically validated miRNA markers for ectopic pregnancy were assessed using ROC curve analyses (\\nFig. 6\\n). As individual markers, hsa-miR-21-5p and hsa-miR-411-5p showed ROC AUC of 0.64 (\\np\\n=\\xa00.035) and 0.66 (\\np\\n=\\xa00.017), respectively. To explore the two markers in combination for the prediction of EP, the inverse correlation of hsa-miR-21-5p and hsa-miR-411-5p expression was used to create a ratio (\\nFig. 7\\n). The ratio of hsa-miR-21-5p/hsa-miR-411-5p showed significant difference between EP and viable intrauterine pregnancies (\\np\\n=\\xa00.0003), and the ratio revealed improved predictive ability with ROC AUC of 0.74 (\\np\\n=\\xa00.0004).\\nFig. 6.\\nOpen in a new tab\\nTwo discriminatory miRNAs in early pregnancy plasma when comparing VIUP with EP (N\\xa0=\\xa082) in biological validation study. Area under the Receiver operating characteristic (ROC) curve (AUC) for A: hsa-miR-21-5p, and B: hsa-miR-411-5p, in women in early pregnancy who subsequently had a confirmed VIUP (N\\xa0=\\xa050), or EP (N\\xa0=\\xa032)\\nROC curve with area, 95\\xa0% confidence interval and\\np\\n-value presented. Statistically significant differences are presented.\\nAbbreviations: ROC = Receiver operating characteristic.\\nFig. 7.\\nOpen in a new tab\\nTwo discriminatory miRNAs in early pregnancy plasma when comparing VIUP with EP (N\\xa0=\\xa082) in biological validation study. A: Fold changes, and B: area under the Receiver operating characteristic (ROC) curve (AUC) for hsa-miR-21-5p/hsa-miR-411-5p ratio, in women in early pregnancy who subsequently had a confirmed VIUP (N\\xa0=\\xa050, green), or EP (N\\xa0=\\xa032, red)\\n∗Statistical significance of differences between groups was tested using Mann-Whitney (non-parametric, continuous). Graph shows geometric mean with geometric standard deviation and\\np\\n-values, with outcome on x axis, and log(10) fold change on y axis.\\nROC curve with area, 95\\xa0% confidence interval and\\np\\n-value presented.\\nStatistically significant differences are presented.\\nAbbreviations: VIUP = Viable intrauterine pregnancy; EP\\xa0=\\xa0Ectopic pregnancy; ROC = Receiver operating characteristic.\\n3.6. Target gene and pathway enrichment analysis\\nAn in-silico analysis of the genes and pathways regulated by hsa-miR-21-5p and hsa-miR-411-5p was undertaken to further understand their potential role in ectopic pregnancy pathophysiology. The miRDIP miRNA data integration portal (University Health Network, Toronto, Canada – RRID:\\nSCR_016770\\n) identified 413 genes with binding sites for these miRNA markers [\\n20\\n,\\n[65]\\n,\\n[66]\\n,\\n[67]\\n]. Pathway analysis was then performed using the Reactome (Reactome project, Ontario/New York/Hinxton/Oregon, Canada/USA/UK – RRID:\\nSCR_003485\\n) platform [\\n[68]\\n,\\n[69]\\n,\\n[70]\\n,\\n[71]\\n,\\n[72]\\n,\\n[73]\\n].\\nOf the 1129 pathways identified using the Reactome platform, 162 met significance (defined as\\np\\n≤\\xa00.05) and are presented in\\nTable S7\\n. These include signal transduction pathways via growth factor receptors, transcription regulation pathways, oestrogen dependent gene expression, and cytokine signalling, particularly interleukins (IL-4, IL-6, IL-12, and IL-13). IL-6, a proinflammatory interleukin, has been associated with ovarian hormonal stimulation, follicle genesis, and implantation, with higher levels identified in the fallopian tubes of women with EP [\\n53\\n]. Pilot work previously presented suggested a difference in both IL-6 and oestradiol expression in EP [\\n74\\n,\\n75\\n].\",\n",
       "  'Discussion': '4. Discussion\\nThere is an ongoing clinical need to improve pregnancy of unknown location triage to reduce potential life-threatening consequences of missed or delayed ectopic pregnancy diagnoses. In this study, we have explored miRNAs as potential novel markers that may play a role in improving PUL triage pending diagnosis using transvaginal ultrasonography. We report the identification of 19 miRNAs differentially expressed in the three outcome groups (VIUP, NVIUP and EP) through maternal plasma miRNA profiling in our discovery work. Three of these were technically validated using RT-qPCR, and two miRNA markers, hsa-miR-21-5p and hsa-miR-411-5p, were further validated in a second, separate validation cohort. We found both validated miRNAs were able to differentiate viable intrauterine pregnancies from ectopic pregnancies, with hsa-miR-21-5p increasing and the hsa-miR-411-5p decreasing significantly in EP. The combination of their relative expression as a ratio demonstrated good predictive ability for EP with ROC AUC of 0.74, indicating their potential use in EP prediction as early as four weeks’ gestation.\\nSeveral putative miRNAs biomarkers of ectopic pregnancy have been previously reported (\\nTable S8\\n) [\\n52\\n,\\n54\\n,\\n[56]\\n,\\n[57]\\n,\\n[58]\\n,\\n[59]\\n,\\n61\\n,\\n76\\n,\\n77\\n]. Dysregulation of these miRNAs have been hypothesized to affect endometrial receptivity, production of reproductive regulators such as kisspeptin, influence implantation site (with associated tissue trauma and inflammation), vascularity and trophoblast formation [\\n53\\n,\\n54\\n,\\n59\\n,\\n78\\n,\\n79\\n]. Whilst there were no peer-reviewed articles that linked hsa-miR-21-5p to EP, there was one paper that discussed hsa-miR-411-5p in this manner [\\n54\\n].\\nThere are key differences between the literature available and our work: a) none of the studies had focussed on a PUL population, b) many used tissue, or serum samples, c) most had low study numbers, and d) the majority of EP included had later gestational ages. We have examined a unique population in this context with early miRNA detection suggesting the ability of prompt EP prediction and detection (with consequent early intervention), which could reduce morbidity. Plasma collection from women with PUL is a non-invasive (unlike tissue), easily tolerated and well accepted blood test, which unlike serum is not affected by white blood cells or platelets where miRNAs are released during clotting. Yi Feng et al. has previously examined tissue samples from women undergoing termination of pregnancy and compared with fallopian tube samples (invasive) from women undergoing salpingectomy for tubal EP (not PUL) [\\n54\\n]. Amongst the miRNAs identified, lower expression of hsa-miR-411-5p was noted in 20\\xa0EP with an average gestational age of nine weeks.\\nFollowing our in-silico analyses, a separate literature review of hsa-miR-21-5p confirmed its involvement in the pathophysiology of multiple pregnancy complications. These include dysregulation of endometrial receptivity, proliferation, apoptosis and migration via the PDCD4/AKT, PI3K/AKY (FOXO3) and JAK/STAT3 pathways, hypoxia-related changes, increased postpartum cardiovascular risk in women previously affected by gestational diabetes, success of in-vitro fertilisation and ovarian reserve, placentation, gestational hypertension, pre-eclampsia and intrahepatic cholestasis of pregnancy, preterm premature rupture of membranes and pre term birth, placental and fetal growth [\\n[80]\\n,\\n[81]\\n,\\n[82]\\n,\\n[83]\\n,\\n[84]\\n,\\n[85]\\n,\\n[86]\\n,\\n[87]\\n,\\n[88]\\n,\\n[89]\\n,\\n[90]\\n,\\n[91]\\n,\\n[92]\\n,\\n[93]\\n,\\n[94]\\n].\\nPTEN (phosphates and tensin homologue deleted on chromosome ten) is quoted as one of the most significant pathways within the in-silico analysis. However, PTEN is also linked to hsa-miR-411-5p expression [\\n95\\n]. Whilst PTEN overexpression negatively affects cell survival, hsa-miR-411-5p causes downregulation, allowing trophoblast proliferation, migration, and invasion [\\n95\\n]. Nuclear paraspeckle assembly transcript 1 (NEAT1) inhibits hsa-miR-411-5p expression, upregulating PTEN which in turn promotes trophoblast apoptosis and pre-eclampsia [\\n95\\n].\\nWhilst these findings support both hsa-miR-21-5p and hsa-miR-411-5p potentially contributing to ectopic pregnancy aetiology in a biologically plausible manner, gaining a functional understanding may prove beneficial when considering their use in combination with other epigenetic markers, proteins or hormones not yet identified to further improve EP prediction, diagnosis, and subsequent management. Given the complexity of early pregnancy processes and pathophysiology, opportunities to use various statistical or algorithmic approaches beyond the scope of this paper may also subsequently arise.\\nLimitations of the study pertain to the overall numbers, the specific demographics, and the subjective assessment of ultrasound, which all could potentially confound results as well as limit findings. However, whilst the proportions of each outcome group were not reflective of population prevalence, and patients were recruited following either PUL classification or EP diagnosis, the enriched EP population, including ectopic pregnancies at diagnosis, allowed this relatively uncommon early pregnancy outcome to be evaluated with greater statistical power, with total population numbers comparable to similar published work, and matched demographically as closely as the population allowed [\\n19\\n,\\n21\\n]. The results, from a Health Policy perspective, lay important groundwork for ongoing research into how miRNA can contribute to improving day to day clinical workflow and patient safety in early pregnancy, both of which would likely also carry economic benefits. Unfortunately, from a total of 290 women, 170 were excluded due to their outcome diagnosis of failed or persistent PUL, both heterogenic groups where pregnancy location could not be confidently differentiated, as well as those lost to follow up or with an intrauterine pregnancy of uncertain viability. It is also likely that miRNAs differentiating VIUP with miscarriages (NVIUP) did not validate due to the heterogeneity of the NVIUP population following a PUL classification. For example, intrauterine pregnancies of unknown viability (IPUV) may have miscarried in the first trimester with or without the development of fetal cardiac activity.\\nValidation in a larger population, the inclusion of all pregnancy of unknown location outcomes, and a sensitivity analysis are the next steps in rigorously evaluating the potential clinical usefulness of hsa-miR-21-5p and hsa-miR-411-5p, both alone and as part of ratio. To facilitate such a large deployment of these miRNAs will require the digital signal of these candidate biomarkers to be merged with the digital signals of two or more independent modalities. This fusion of technology would increase generalisability and access to the markers without compromising workflow efficiency and precision, allowing outpatient early pregnancy use in a manner that is rapid but accurate.\\nWe have demonstrated the potential of circulating miRNA markers for PUL triage by predicting EP in early pregnancy, which can be measured via the gold-standard RT-qPCR method that is an efficient, cost-effective, and readily available technology for translational use in clinical practice. With better diagnostics, the potential for further advancements in ectopic pregnancy therapeutics is also high, using proposed vehicles such as nanoparticles for targeted interventions in the management of pregnancy complications [\\n96\\n].',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1016/j.ncrna.2025.05.005',\n",
       "  'Journal': 'Noncoding RNA Research'},\n",
       " {'PaperTitle': 'A Systematic Evidence‐Based Review Regarding miRNA Polymorphisms in Recurrent Implantation Failure',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Bogdan Doroftei, Ovidiu‐Dumitru Ilie, Ana‐Maria Dabuleanu, Mara Doroftei, Ciprian Ilea, Sergiu Timofeiov, Anca Bivoleanu, Elena Tataranu',\n",
       "  'Email': 'ilie_ovidiu-dumitru@d.umfiasi.ro',\n",
       "  'Abstract': 'Background This systematic review aimed to evaluate whether specific single nucleotide polymorphisms (SNPs) in miRNAs are associated with recurrent implantation failure (RIF).\\nMethods A comprehensive literature search was conducted across PubMed‐MEDLINE, Web of Science, Scopus, and the Excerpta Medica DataBASE.\\nResults The Newcastle‐Ottawa Scale (NOS) yielded an intermediate to high quality, with one study rated with 6 stars, and the remaining four with 7 stars. RIF risk‐related genotypes included miR‐196a, miR‐449b, miR‐34a, miR‐146aCG+GG‐miR‐196a2CC, miR‐149TT‐miR‐196a2CC, miR‐196a2CC‐miR‐499AA, miR‐608GC‐miR‐938CC, miR‐27aAG‐miR‐423CC/miR‐604AG/GG and miR‐34aC>A AA‐miR‐130aG>A GG. Protective combinations included miR‐1302‐3, miR‐631II‐miR‐1302‐3CT, and miR‐938CC‐miR‐1302‐3CT. Protective allele combinations G‐T‐T‐A, C‐T, T‐T‐G, T‐T and G‐C‐A‐G, G‐A‐G, A‐G‐G were less frequent in RIF cases, whereas A‐T‐C, T‐C‐C‐T, T‐C‐T, A‐C‐G‐A, A‐A‐G‐G, G‐A‐A‐A, A‐A‐C‐A and G‐G‐A haplotypes were more commonly associated with increased risk. Notably, miR‐608 GC+CC, miR‐1302‐3 CC, miR‐27a AG+GG, miR‐423 CA+AA, miR‐604 AG+GG, miR‐222 GT+TT, and miR‐34a GA+AA were associated with altered coagulation parameters. Additionally, miR‐222 correlated with decreased creatinine levels, the G>T mutation with elevated follicle‐stimulating hormone (FSH), miR‐34aC>A AA genotype with reduced thyroid‐stimulating hormone (TSH) levels, and CA+AA with increased blood urea nitrogen (BUN) levels.\\nConclusions This systematic review highlights that specific miRNA SNPs and haplotype combinations are significantly associated with either increased susceptibility to or protection against RIF.',\n",
       "  'Introduction': '1. Introduction\\nThe field of human‐assisted reproductive technology (ART) has advanced significantly in the management of infertility, demonstrating improved outcomes over time [1]. These advances are reflected in rising global success rates of pregnancies achieved through ART intervention [2]. Nonetheless, concerns persist regarding the risk of implantation failure (IF), which remains a subject of ongoing debate in clinical and research settings [3]. Among ART procedures, in vitro fertilization‐embryo transfer (IVF‐ET) can result in recurrent implantation failure (RIF) [4], a commonly encountered clinical challenge [5, 6], with an estimated prevalence of up to 10% of IVF attempts [7].\\nDue to the inconsistent application of the term RIF and the lack of a universally accepted definition, this condition is often variably described. A commonly cited definition refers to RIF as the failure to conceive a pregnancy after at least three transfers of high‐quality embryos [8]. Other guidelines adopt broader criteria, defining RIF as two or more consecutive failed transfers, which may include a cumulative total of 4–10 embryos across multiple IVF‐ET cycles or intracytoplasmic sperm injection (ICSI), whether using fresh or frozen embryos or, in some definitions, at least two transferred blastocysts [5, 6, 9].\\nGiven the limited understanding of the pathogenesis of RIF [10, 11] and its complex, multifactorial etiology [12], comprehensive molecular investigations are necessary to uncover its underlying biological mechanisms [13]. In this context, microRNAs (miRNAs) have emerged as key epigenetic regulators of gene expression. They primarily function by mediating post‐transcriptional gene silencing [14, 15] through binding to the 3′‐untranslated regions (3′‐UTRs) of target mRNAs, resulting in either mRNA degradation or translational repression [14, 16].\\nA growing body of evidence highlights the essential role of miRNAs in maintaining normal cellular functions [17]. Although miRNAs constitute only about 1%–3% of the human genome, they are estimated to regulate the expression of approximately 30% of human genes [18, 19]. Accumulating data have revealed that miRNAs are intricately involved in fundamental biological processes such as cell proliferation, differentiation, and apoptosis [20]. Notably, dysregulated miRNA expression has been linked to impaired fertility in mammals, including disruptions in human implantation [15].\\nAt the molecular level, altered microRNA activity may contribute to an increased susceptibility to RIF, endometriosis (EMS), and impaired endometrial receptivity (ER) among other reproductive disorders [21]. In light of this, the present systematic review aims to evaluate the quality of existing research and to identify microRNA polymorphisms that are most frequently associated with RIF.',\n",
       "  'Methods': \"2. Materials and Methods\\n2.1. Methodology\\nThis systematic review protocol was developed in accordance with the 2020 Preferred Reporting Items for Systematic Reviews and Meta‐Analyses (PRISMA) guidelines [22].\\n2.2. Ethics Statement\\nThis manuscript did not require approval from an Institutional Review Board (IRB) or consent from any third party, as the research data analyzed were extracted from previously published studies in peer‐reviewed journals.\\n2.3. Sources Inquiry\\nA comprehensive literature search was conducted across four major academic databases: PubMed‐MEDLINE—United States National Library of Medicine (NLM, 1996), Web of Science (WOS) (Clarivate Analytics, 1997), Scopus (Elsevier, 2004) [23, 24, 25], and Excerpta Medica dataBASE (EMBASE) (Elsevier, 1947). These databases were selected due to their complementary coverage and variation in both quantitative and qualitative indexing [26]. The search spanned from January 1st, 2014, to December 1st, 2024, to capture the most current and impactful evidence available in the field. Search strategies were built using both MeSH (Medical Subject Headings) and Emtree terms. The term ‘MicroRNA’ [MicroRNAs] [MeSH] (2003) was used as a major topic [Majr] and can be identified with the MeSH Unique ID: D035683, Tree Number(s): D13.150.650.319, D13.444.735.150.319, D13.444.735.790.552.500 in combination with ‘Polymorphism’ [Polymorphism, Genetic] [MeSH] (2005)—MeSH Unique ID: D011110, Tree Number(s): G05.365.795 and ‘Recurrent Implantation Failure’. As RIF is not recognized as an official MeSH term, the search string incorporated its abbreviation as a free‐text keyword (accessed on 1.12.2024).\\n2.4. String Strategy\\nBoolean operators such as ‘AND’ or ‘OR’ or ‘NOT’ were employed to construct the initial search strategy (Search #1) as a single query. Additional combinations were used to refine a second series (Search #2). Full details of the search strings and syntax are provided in Supporting Information S1.\\n2.5. Selection of Studies\\nPotentially eligible references were imported into Mendeley—Reference Management Software (v. 1.19.8) (Elsevier, 2013), which has demonstrated relatively high accuracy in duplicate detection, with a reported sensitivity and specificity of 93%, comparable to that of Ovid/Rayyan (97%) and Covidence (96%) [27]. The ‘Check for Duplicates’ function was applied and supplemented by a manual screening to ensure consistency across records. The titles ± abstracts were independently reviewed by each contributor. Full‐text articles meeting the inclusion criteria were evaluated by O.‐D.I., A.‐M.D. (C)., S.T., and B.D. Any discrepancies or conflicting assessments were resolved unanimously through discussion and consensus. A summarizing table of the retrieved records is provided in Supporting Information S2.\\n2.6. Objective\\nTo address the primary objective in identifying miR polymorphisms associated with either an increased or decreased risk of RIF, we developed a Patient (P), Intervention (I), Comparator (C), and Outcome (O) (PICO) model. This structured model guided the formulation of the research question and eligibility criteria and is presented in Supporting Information S3.\\n2.7. Extraction of Data\\nStandard methodological data were extracted by O.‐D.I., E.T., and B.D. from the included studies using a structured tabular form in Microsoft Excel 2010 (Microsoft Corporation, Redmond, WA, USA). The spreadsheet was also utilized for sorting and coding raw data. Extracted variables included: first author's name and year of publication, country, study settings, study design, population and sample size, microRNA, method of detection, and identified microRNA polymorphism(s).\\n2.8. Quality Assessment\\nO.‐D.I., and B.D. independently evaluated the methodological quality of the included studies using the Newcastle‐Ottawa Scale (NOS) [28], appropriate for the appraisal of non‐randomized study designs. The NOS assesses studies across three domains: Selection (score 0–4), Comparability (score 0–2), Exposure (score 0–3). Quality classification was determined based on the total number of asterisks awarded per item: high (7–9), intermediate (4–6), and low quality (0–3).\\n2.9. Criteria of Inclusion and Exclusion\\nTo be deemed eligible for inclusion, studies were required to present original data, be written exclusively in English, and be published in peer‐reviewed journals, regardless of adherence to the IMRaD structure. Additionally, only studies involving human subjects were considered. All other types of publications and analyses were automatically excluded.\",\n",
       "  'Results': \"3. Results\\n3.1. Final Number of Entries and Inclusion\\nOut of a total of 27 retrieved records (7 from PubMed‐MEDLINE, 11 from WOS, 4 from Scopus, and 5 from EMBASE), 11 initially met the inclusion criteria for the review after duplicates were removed. During the full‐text screening phase, six studies were excluded due to not meeting our predefined eligibility criteria. The main reasons for exclusion included: use of non‐human (animal) models, studies lacking original data on microRNA polymorphisms associated with RIF, focus on experimental models not directly related to miRNA genetics, and review articles without primary data. Additionally, some studies applied network or bioinformatics analyses without investigating specific single nucleotide polymorphisms (SNPs) or functional impacts on miRNA pathways relevant to RIF. A detailed summary is provided in Figure 1. A complete list of the included manuscripts organized chronologically is provided in Supporting Information S4.\\nFIGURE 1.\\nOpen in a new tab\\nPRISMA flow diagram of the eligible case–control studies.\\n3.2. Study Characteristics\\nThe five included studies were published over distinct time periods: one in 2016 [29], two in 2019 [30, 31], one in 2020 [32], and one in 2023 [33]. Most were conducted by researchers based in East Asia, particularly South Korea [30, 31, 32, 33], with collaborations involving Japanese institutions [29] or dual affiliations with universities in the USA [33]. These case–control studies collectively enrolled 1799 women, 1161 controls, and 638 women diagnosed with RIF. Associations between miRNA polymorphisms and RIF were identified using various methods, including a combined approach [29], Polymerase‐Chain Reaction (PCR)‐Restriction Fragment Length Polymorphism (RFLP) [30, 31, 32], or Real‐Time (RT)‐qPCR [33] (Table 1).\\nTABLE 1.\\nMain characteristics of the eligible studies reporting miRNAs polymorphisms in RIF.\\nAuthor's first name and year of publication\\nCountry\\nStudy settings\\nStudy design\\nPopulation and sample type\\nmicroRNA\\nMethod of detection\\nmicroRNA polymorphism(s)\\nReferences\\nCho et al. 2016\\nSouth Korea, Japan\\nn = 234\\n*\\nn = 120\\n**\\nCase–control\\nBlood samples from\\nn = 354 women\\nmiR‐146a (rs2910164)\\nmiR‐149 (rs2292832)\\nmiR‐196a (rs11614913)\\nmiR‐499 (rs3746444)\\nPCR‐RFLP\\nqRT‐PCR\\nmiR‐146aCG+GG/miR‐196a2CC genotype was associated with an increased risk of RIF\\nG‐T‐T‐A (miR‐146a/miR‐149/miR‐196a2/miR‐499) and G‐T‐T (miR‐146a/miR‐149/miR‐196a2) genotype frequencies were less frequent in RIF\\n[29]\\nLee et al. 2019\\nSouth Korea\\nn = 212\\n*\\nn = 119\\n**\\nCase–control\\nBlood samples from\\nn = 331 women\\nmiR‐605 (rs2043556)\\nmiR‐608 (rs4919510)\\nmiR‐631 (rs5745925)\\nmiR‐938 (rs12416605)\\nmiR‐1302‐3 (rs7589328)\\nPCR‐RFLP\\nmiR‐938C/1302–3T allele combinations were associated with a decreased risk of RIF\\nT allele of miR‐1302–3C>T was associated with a decreased risk of RIF\\nmiR‐605A/938T/1302–3C allele combination was associated with an increased risk of RIF\\n[30]\\nLee et al. 2019\\nSouth Korea\\nn = 228\\n*\\nn = 118\\n**\\nCase–control\\nBlood samples from\\nn = 346 women\\nmiR‐25 (rs1527423)\\nmiR‐32 (rs7041716)\\nmiR‐125a (rs12976445)\\nmiR‐222 (rs34678647)\\nPCR‐RFLP\\nmiR‐25T/miR‐125aT/miR‐222G was associated with a reduced risk of RIF\\nmiR‐25T/miR‐125aT allele combinations were associated with a reduced risk of RIF\\nmiR‐25T/miR‐32C/miR‐125aC/miR‐222T allele combinations were associated with an increased risk of RIF\\n[31]\\nKim et al. 2020\\nSouth Korea\\nn = 219\\n*\\nn = 120\\n**\\nCase–control\\nBlood samples from\\nn = 339 women\\nmiR‐27a (rs895819)\\nmiR‐423 (rs6505162)\\nmiR‐449b (rs10061133)\\nmiR‐604 (rs2368393)\\nPCR‐RFLP\\nmiR‐27aA>G, miR‐449bAG+GG, and miR‐604A>G was associated with an increased risk of RIF\\n[32]\\nLee et al. 2023\\nSouth Korea\\nn = 268\\n*\\nn = 161\\n**\\nCase–control\\nBlood samples from\\nn = 429 women\\nmiR‐218‐2 (rs11134527)\\nmiR‐34a (rs2666433)\\n(rs6577555) miR‐130a\\n(rs731384)\\nqRT‐PCR\\nmiR‐34aC>A AA genotype was associated with an increased risk of RIF\\n[33]\\nOpen in a new tab\\n* Control group.\\n** RIF group.\\n3.3. Methodological Heterogeneity and Statistical Considerations\\nThe five studies included in this analysis did not uniformly define RIF, leading to significant heterogeneity. While all studies required the failure to achieve pregnancy after multiple fresh IVF‐ET cycles with embryos that had cleaved into more than 10 cells, notable differences arose regarding the number of failed cycles and the criteria related to both embryo quantity and quality.\\nSpecifically, three studies [29, 30, 31] defined RIF as the inability to conceive after two completed IVF‐ET cycles involving a total of more than 10 cleaved embryos, focusing on a quantity‐based threshold. In contrast, one study [32] applied stricter criteria by requiring failure after two cycles with transfers of only one or two good‐quality embryos, each cleaved into more than 10 cells, emphasizing embryo quality rather than quantity. Another study [33] broadened the definition by requiring failure after more than two IVF‐ET cycles, reflecting a higher severity threshold.\\nThese discrepancies suggest that some studies may have included women with repeated transfers of lower‐quality embryos, while others focused on patients experiencing failure despite transfer of fewer but higher‐quality embryos. This heterogeneity likely influenced patient selection and clinical characteristics across studies, which in turn impacts the strength, comparability, and generalizability of the associations reported with miRNA polymorphisms.\\nConsequently, these definitional inconsistencies may obscure true genetic effects or limit detection of important subgroup‐specific findings, highlighting the need for cautious interpretation and for standardized RIF definitions in future research.\\nAdditionally, the authors employed a range of statistical methods and software tools to evaluate the association between miRNA polymorphisms and RIF, demonstrating methodological heterogeneity. Most studies calculated sample size and power using QUANTO [29] or G*POWER [30], with a desired power threshold of 80% and a significance level of p < 0.05. Genotype and allele frequencies were assessed using Fisher's exact tests or logistic regression, and odds ratios (ORs), adjusted odds ratios (AORs), and 95% confidence intervals (CIs) were calculated.\\nThree studies used multivariate logistic regression models that adjusted for confounders such as maternal age [30, 31, 32]. However, other key clinical variables such as body mass index (BMI), endometrial thickness, hormonal levels, or number of embryos transferred were either inconsistently reported or not included in the models, limiting the capacity to control for all relevant confounding factors. Hardy–Weinberg equilibrium (HWE) testing was uniformly performed to validate genotype distributions.\\nTo address multiple comparisons, some studies applied Bonferroni correction [29] or the false discovery rate (FDR) approach [30], strengthening the robustness of the reported associations. Combined genotype and haplotype risk scores were typically derived from genotype interaction models, with three studies employing multifactor dimensionality reduction (MDR) [29, 30, 32] to identify synergistic gene–gene effects. However, most studies did not specify whether the cumulative risk was estimated using additive, multiplicative, or interaction‐based models. Only one study explicitly calculated inferred genotype frequencies from MDR results to estimate the combined effect of miRNA variants [29], while the remaining studies reported ORs for genotype combinations without detailing whether a composite or weighted risk score was used. Consequently, the interpretation of cumulative genetic risk remains uncertain due to the lack of standardized or transparent scoring methodology.\\nGiven the limited number of eligible studies (n = 5), the potential for publication bias cannot be excluded. Although formal assessments such as Egger's regression or funnel plot asymmetry analysis were not feasible, the possibility of selective reporting should be considered when interpreting these findings. To enhance methodological transparency and improve future meta‐analytic work, future research should adopt standardized statistical modeling approaches, adjust for key clinical confounders, and clearly report the derivation and interpretation of cumulative genetic risk scores.\\n3.4. Quality of Case–Control Studies Assessment\\nAs shown in Table 2 and Figure 2, the overall quality of the eligible case–control studies ranged from intermediate to high across all three assessment items, with the exception of Lee et al. [31], which did not adequately characterize the control group under the Selection criterion. In contrast, the remaining four studies provided detailed descriptions of participant enrollment. Non‐RIF control participants were required to have a regular menstrual cycle, a history of at least one naturally conceived pregnancy, no history of pregnancy loss or preeclampsia, and a normal karyotype [46XX] [29, 30, 32, 33]. Regarding the Outcome criterion, a score for the ‘Non‐Response Rate’ could not be assigned; however, the total quality scores were 6 [31] and 7 for the other four studies [29, 30, 32, 33].\\nTABLE 2.\\nQuality assessment of eligible case–control studies based on NOS.\\nStudy\\nCase definition\\nCase representativeness\\nControls selection\\nControls definition\\nComparability\\nExposure ascertainment\\nMethod ascertainment\\nNon‐response rate\\nTotal\\nCho et al. [29]\\n*\\n*\\n*\\n*\\n*\\n*\\n*\\n—\\n7\\nLee et al. [30]\\n*\\n*\\n*\\n*\\n*\\n*\\n*\\n—\\n7\\nLee et al. [31]\\n*\\n*\\n*\\n—\\n*\\n*\\n*\\n—\\n6\\nKim et al. [32]\\n*\\n*\\n*\\n*\\n*\\n*\\n*\\n—\\n7\\nLee et al. [33]\\n*\\n*\\n*\\n*\\n*\\n*\\n*\\n—\\n7\\nOpen in a new tab\\nNote: “*” represents 1 point within the NOS scale for each item, whereas “—” represents 0 points.\\nFIGURE 2.\\nOpen in a new tab\\nGraphical representation of the overall score following NOS.\\n3.5. Differential Profile Between Participants\\n3.5.1. Genotype Frequencies in RIF\\nIF is widely understood to result from multifactorial biological disruptions, and miRNA polymorphisms have emerged as plausible contributors to this complexity. Variations in miRNA sequences can alter gene regulatory networks, particularly those involved in ER, immune modulation, and hormone signaling. Thus, assessing the frequency of specific miRNA genotypes in women with RIF may help identify genetic predispositions. This section reviews the associations between distinct miRNA genotypes and RIF occurrence, exploring their potential as risk or protective factors depending on IF frequency and individual genetic background. Among the reviewed studies, Lee et al. [31] reported no significant differences in genotype frequencies between RIF and healthy individuals. In contrast, other investigations have indicated either a protective effect [30] or increased susceptibility [29, 32, 33]. For instance, Cho et al. [29] were among the first to observe an association between the miR‐196a2C>T polymorphism and RIF risk, although this was not statistically significant (p > 0.05) relative to wild‐type (WT) homozygotes. The AOR was 1.488, with a 95% CI ranging from 0.749 to 2.955, with no major differences in the number of IFs. When considering the combined CT+CC genotype, the AOR was 1.230, with a 95% CI of 0.720–2.099. Kim et al. [32] later demonstrated a significant association between the miR‐449bA>G polymorphism and increased RIF, particularly for the AA genotype compared to AG+GG (AOR = 1.584, 95% CI = 1.008–2.490, p = 0.046). The risk increased in women who had experienced at least four IFs. The AG+GG genotype was associated with a higher RIF risk compared to AA (AOR = 1.932, 95% CI = 1.122–3.327, p = 0.018). Subsequently, Lee et al. [33] identified the miR‐34aC>A AA genotype as significantly associated with more than two IFs (AOR = 2.264, 95% CI = 1.007–5.092, p = 0.048). Consistent findings indicated that patients with this genotype, particularly those with multiple IFs, are at increased risk. Specifically, for women with more than three IFs, the AA genotype had an AOR of 2.322, 95% CI of 1.052–5.125, and p = 0.034, while the recessive comparison (CC+CA vs. AA) yielded an AOR = 2.406, 95% CI = 1.068–5.419, p = 0.037. With four or more IFs, the AOR increased to 2.783 and 95% CI of 1.185–6.536, p = 0.019, and for CC+CA versus AA, the AOR was 2.457, 95% CI = 1.077–5.606, p = 0.033, further reinforcing a genetic influence in RIF [33].\\nConversely, Lee et al. [30] found that the miR‐1302‐3 CT genotype (AOR = 0.234; 95% CI = 0.089–0.618, p = 0.003) was associated with a decreased RIF risk, and this effect persisted with the CT+TT combination; AOR = 0.227; 95% CI = 0.086–0.598, p = 0.003. In women with more than three IFs, the CT genotype had an AOR of 0.261 with a 95% CI of 0.099–0.690 and p = 0.007, and the CT+TT genotype showed an AOR of 0.253 with a 95% CI of 0.096–0.668, and p = 0.006. For four IFs, the CT genotype had an AOR of 0.301 and 95% CI between 0.103 and 0.882, and p = 0.029, and the CT+TT combination showed an AOR of 0.292, 95% CI of 0.100–0.853, and p = 0.024.\\n3.5.2. Allele Combinations in RIF\\nBeyond the effects of individual genotypes, combinations of alleles from different miRNA loci may interact epistatically, impacting implantation‐related processes such as cytokine balance, cellular adhesion, and angiogenesis. Given that gene regulation by miRNAs is cumulative, evaluating haplotypes and combinations of alleles provides a more comprehensive understanding of genetic susceptibility. This section explores the allele combinations that have been reported to influence the risk of RIF, either increasing susceptibility or providing protection. It emphasizes statistically significant haplotypes found in patient cohorts. A comprehensive analysis of the haplotype data across multiple studies has identified several microRNA polymorphisms combinations that appear to modulate susceptibility to RIF. Specifically, certain haplotypes were consistently associated with a reduced risk of RIF. These included G‐T‐T‐A [29], C‐T [30], T‐T‐G, T‐T [31], and G‐C‐A‐G, G‐A‐G, A‐G‐G [32]. Statistically, these associations were supported by Fisher's exact test, followed by Bonferroni or FDR corrections for multiple comparisons. For example, the G‐T‐T‐A haplotype showed a protective effect with an AOR = 0.443, 95% CI = 0.261–0.753, p = 0.002; Bonferroni‐adjusted p = 0.035 [29], while the C‐T combination demonstrated a similarly reduced risk; OR = 0.259, 95% CI = 0.100–0.674, FDR‐p‐value = 0.003 [30]. Protective effects were also observed for T‐T‐G: AOR = 0.528, 95% CI = 0.282–0.990, p = 0.044, and T‐T: AOR = 0.510, 95% CI = 0.285–0.913, p = 0.022 [31], as well as for G‐C‐A‐G: OR = 0.248, 95% CI = 0.115–0.537, p = 0.0001, G‐A‐G: OR = 0.337, 95% CI = 0.174–0.652, p = 0.001, FDR = 0.007, and A‐G‐G: OR = 0.094, 95% CI = 0.005–1.636, p = 0.030, FDR = 0.210 [32].\\nIn contrast, several haplotypes were found to be significantly associated with an increased risk of RIF. Among the most notable was A‐T‐C (OR = 31.670, 95% CI = 1.802–556.500, p = 0.0003) [30]. Other risk‐enhancing combinations included T‐C‐C‐T (AOR = 1.496, 95% CI = 1.000–2.237, p = 0.049) and T‐C‐T (AOR = 1.585, 95% CI = 1.071–2.345, p = 0.021) [31], as well as A‐C‐G‐A (OR = 2.352, 95% CI = 1.260–4.390, p = 0.007), A‐A‐G‐G (OR = 8.818, 95% CI = 1.004–77.460, p = 0.030), and G‐A‐A‐A (OR = 5.291, 95% CI = 1.627–17.210, p = 0.006) [32]. Additional risk‐associated haplotypes identified included A‐A‐C‐A (OR = 3.687, 95% CI = 1.078–12.61, p = 0.034) and G‐G‐A (OR = 2.111, 95% CI = 1.044–4.269, p = 0.034) [33].\\n3.5.3. Combined Genotypes in RIF\\nIndividual miRNAs frequently target overlapping sets of mRNAs, implying combined effects of multiple miRNA polymorphisms that may result in more significant physiological outcomes. These synergistic or antagonistic genotype patterns could influence critical reproductive processes such as trophoblast invasion, endometrial remodeling, and immune tolerance. This section explores how combined miRNA genotypes affect IF, with the aim of identifying profiles that either increase or reduce the risk based on the co‐occurrence of polymorphisms. Cho et al. [29] reported that individuals carrying the combined genotypes miR‐146aCG+GG and miR‐196a2CC have a significantly higher susceptibility to RIF (AOR = 2.022, 95% CI = 1.044–3.916, p = 0.037). Other combinations involving miR‐196a2CC, specifically with miR‐149TT (AOR = 1.580, 95% CI = 0.811–3.076) and with miR‐499AA (AOR = 1.449, 95% CI = 0.801–2.621) were also associated with elevated risk, although these did not reach statistical significance. Complementary findings revealed that the miR‐608GC with miR‐938CC genotypes, when combined, were significantly associated with increased RIF risk (OR = 1.811, 95% CI = 1.024–3.202, p = 0.041). In contrast, other genotype combinations appeared to confer a protective effect. Notably, individuals carrying both miR‐631II and miR‐1302‐3CT (OR = 0.244, 95% CI = 0.092–0.650, p = 0.005) and miR‐938CC with miR‐1302‐3CT (OR = 0.270, 95% CI = 0.101–0.718, p = 0.009) had a markedly lower prevalence of RIF [30]. Further evidence supports the role of miRNA combinations in modulating implantation outcomes. The presence of miR‐27aAG with miR‐423CC or miR‐604AG/GG genotype was significantly associated with RIF; (AG/CC: AOR = 0.417; 95% CI = 0.223–0.780; p = 0.006, respectively, AG/AG: AOR = 0.329; 95% CI = 0.153–0.706; p = 0.004 and AG/GG: AOR = 0.193, 95% CI = 0.054–0.693, p = 0.012) [32]. Lastly, the combination of miR‐34aC>A AA and miR‐130aG>A GG genotype as a potential factor was significantly associated with RIF (AOR = 2.881, 95% CI = 1.132–7.332, p = 0.026) [33].\\n3.5.4. Coagulation Factors\\nIn the context of implantation biology, proper regulation of coagulation is essential for establishing and maintaining ER and vascular integrity. Aberrations in coagulation parameters may disrupt this delicate balance and contribute to IF. Several studies have explored how miRNA polymorphisms may influence coagulation profiles in women with RIF, shedding light on a potential molecular link between genetic variability and thrombotic risk in early pregnancy. The outcomes of hematological evaluations aimed at determining the potential risk of blood clots indicated that prothrombin time (PT), activated partial thromboplastin time (aPTT), and platelet count (PLT) exhibited synergistic influences on particular microRNA polymorphisms associated with RIF [30, 31, 32, 33]. The number of PLT in both groups was similar, indicated by the non‐significant difference (p > 0.05) [30, 31, 32, 33]. Nonetheless, additional analyses confirm significant differences in PT (p < 0.05) per Lee et al. [31, 33], contrasting with earlier studies (p > 0.05) [30, 32]. According to the data, similar findings were noted for aPTT in three [30, 31, 32] out of five eligible articles [33], except for the one that concentrated solely on the RIF group [29]. Considering harmful effects, the aPTT showed a significant change (p < 0.05) [30, 31, 32] compared to the findings of Lee et al. [33] as indicated by p = 0.224. In addition to the aforementioned, specific genotypes showed a strong correlation with RIF across all three coagulation factors. Among the key genotype variants were miR‐608 GC+CC, miR‐1302‐3 CC [30], miR‐27a AG+GG, miR‐423 CA+AA, miR‐604 AG+GG [32], miR‐222 GT+TT [31], and miR‐34a GA+AA [33].\\n3.5.5. Hormonal Factors\\nGiven that hormone signaling is fundamental to endometrial preparation and embryo implantation, any miRNA‐mediated disruption in hormone levels could plausibly interfere with reproductive outcomes. Recent evidence suggests that miRNA polymorphisms may exert regulatory control over key reproductive hormones, further linking genetic variability to implantation physiology. The analysis of hormones associated with reproduction in RIF individuals revealed significant changes in estradiol (E2) and luteinizing hormone (LH) levels (p < 0.05). In comparison to E2 and LH, follicle‐stimulating hormone (FSH) levels were identified to be within the normal limits (p > 0.05) [30, 31, 32, 33]. It has been noted that FSH levels could increase depending on the presence of the miR‐222 G>T mutation [31], while thyroid‐stimulating hormone (TSH) is generally found to be reduced in individuals possessing the miR‐34aC>A AA genotype (p = 0.003).\\n3.5.6. Renal Factors\\nAlthough renal function may seem tangential to implantation, emerging evidence suggests that systemic metabolic stress, reflected in altered renal biomarkers, can influence uterine receptivity. In this regard, miRNA polymorphisms may play a role in shaping the renal microenvironment and, by extension, the broader physiological milieu necessary for successful implantation. Regarding other predisposing factors that may influence pregnancy outcomes, blood urea nitrogen (BUN) and creatinine levels were higher in RIF patients compared to controls (p < 0.0001) [31, 33]. In addition to this distinguishing characteristic, the authors demonstrated that miR‐222 polymorphisms result in a reduction of creatinine levels (p < 0.05) [31], whereas the miR‐34aC>A CA+AA genotype was associated with an increased level of BUN (p = 0.015) [33].\",\n",
       "  'Discussion': \"4. Discussion\\nIn this systematic review, we aimed to summarize available evidence from case–control studies regarding the frequencies and combinations of miRNA genotypes and alleles significantly contributing to RIF. Among the five eligible studies, four received a high‐quality score of 7 [29, 30, 32, 33], while one study was rated as intermediate with a score of 6 [31] due to insufficient characterization of control participants' medical histories.\\nTo improve interpretability, Table 3 presents a consolidated summary of the most significant miRNA polymorphisms and haplotype combinations associated with RIF, highlighting the direction of risk, effect sizes (ORs/AORs), and corresponding biological mechanisms reported in the literature. To further elucidate the potential functional impact of these miRNA variants, we compiled a list of both experimentally validated and bioinformatically predicted target genes for the key miRNAs identified in this study. These target genes are implicated in critical biological pathways relevant to RIF, including ER, immune regulation, and hormonal signaling, as summarized in Table 4.\\nTABLE 3.\\nSummary of miRNA polymorphisms and associated risk of RIF with proposed mechanisms.\\nmiRNA genotype/combination\\nRisk direction\\nAOR/OR (95% CI)\\nProposed mechanism\\nReferences\\nmiR‐146a CG+GG+miR‐196a2 CC\\n↑\\n2.022 (1.044–3.916)\\nInflammatory signaling, ER\\n[29]\\nmiR‐196a2 CC+miR‐149 TT\\n↑\\n1.580 (0.811–3.076)\\nER\\n[29]\\nmiR‐196a2 CC+miR‐499 AA\\n↑\\n1.449 (0.801–2.621)\\nUnknown\\n[29]\\nG‐T‐T‐A haplotype\\n↓\\n0.443 (0.261–0.753)\\nImmunomodulatory via NF‐κB\\n[29]\\nmiR‐1302‐3 CT/CT+TT\\n↓\\n0.234 (0.089–0.618)/0.227 (0.086–0.598)\\nHormonal regulation, coagulation modulation\\n[30]\\nmiR‐608 GC+miR‐938 CC\\n↑\\n1.811 (1.024–3.202)\\nCoagulation regulation\\n[30]\\nmiR‐631 II+miR‐1302‐3 CT\\n↓\\n0.244 (0.092–0.650)\\nImmunological protection\\n[30]\\nmiR‐938 CC+miR‐1302‐3 CT\\n↓\\n0.270 (0.101–0.718)\\nProtective interaction\\n[30]\\nA‐T‐C haplotype\\n↑\\n31.67 (1.802–556.5)\\nStrong genetic susceptibility\\n[30]\\nC‐T haplotype\\n↓\\n0.259 (0.100–0.674)\\nImmunological modulation\\n[30]\\nT‐T haplotype\\n↓\\n0.510 (0.285–0.913)\\nGenetic protective factor\\n[31]\\nT‐C‐T combination\\n↑\\n1.585 (1.071–2.345)\\nEndometrial dysfunction\\n[31]\\nT‐C‐C‐T haplotype\\n↑\\n1.496 (1.000–2.237)\\nUnknown\\n[31]\\nmiR‐222 GT+TT\\n↑\\nNot provided\\nFSH dysregulation\\n[31]\\nmiR‐449b AA\\n↑\\n1.584 (1.008–2.490)\\n≥ 4 IFs\\n[32]\\nmiR‐27a AG+miR‐423 CC\\n↓\\n0.417 (0.223–0.780)\\nHormonal modulation\\n[32]\\nmiR‐27a AG+miR‐604 AG\\n↓\\n0.329 (0.153–0.706)\\nHormonal modulation\\n[32]\\nmiR‐27a AG+miR‐604 GG\\n↓\\n0.193 (0.054–0.693)\\nHormonal modulation\\n[32]\\nG‐C‐A‐G haplotype\\n↓\\n0.248 (0.115–0.537)\\nGenetic protective factor\\n[32]\\nG‐A‐G haplotype\\n↓\\n0.337 (0.174–0.652)\\nUnknown\\n[32]\\nA‐G‐G haplotype\\n↓\\n0.094 (0.005–1.636)\\nUnknown\\n[32]\\nA‐C‐G‐A haplotype\\n↑\\n2.352 (1.260–4.390)\\nUnknown\\n[32]\\nA‐A‐G‐G haplotype\\n↑\\n8.818 (1.004–77.460)\\nUnknown\\n[32]\\nG‐A‐A‐A haplotype\\n↑\\n5.291 (1.627–17.210)\\nUnknown\\n[32]\\nmiR‐34a C>A AA\\n↑\\n2.322 (1.052–5.125)\\nHormonal and renal dysfunction\\n[33]\\nmiR‐34a C>A AA+miR‐130a G>A GG\\n↑\\n2.881 (1.132–7.332)\\nHormonal/genetic synergy\\n[33]\\nA‐A‐C‐A haplotype\\n↑\\n3.687 (1.078–12.61)\\nUnknown\\n[33]\\nG‐G‐A haplotype\\n↑\\n2.111 (1.044–4.269)\\nUnknown\\n[33]\\nOpen in a new tab\\nAbbreviations: ↑, increased risk; ↓, decreased risk; AOR, adjusted odds ratio; CI, confidence interval; IFs, implantation failures; OR, odds ratio.\\nTABLE 4.\\nKey microRNAs and their validated or predicted target genes associated with RIF.\\nmiRNA\\nTarget gene(s)\\nValidation status\\nBioinformatic source(s)\\nBiological role\\nRelevance to implantation\\nReference(s)\\nmiR‐146a\\nIRAK1, TRAF6\\nValidated\\nmiRTarBase, Literature\\nInflammatory signaling via NF‐κB\\nRegulation of endometrial immune balance\\n[29]\\nmiR‐149\\nGIT1, FOXM1\\nPredicted\\nTargetScan, miRDB\\nCell proliferation, migration\\nTrophoblast invasion, ER\\n[29]\\nmiR‐196a2\\nHOXB8, ANXA1, ERG\\nPredicted\\nTargetScan, DIANA‐TarBase\\nCell differentiation, immune signaling\\nEndometrial patterning, vascular remodeling\\n[29]\\nmiR‐608\\nTP53, IL1A\\nPredicted\\nNot consistently predicted in TargetScan/miRTarBase\\nInflammation, apoptosis\\nImmune surveillance, endometrial immune tolerance\\n[30]\\nmiR‐938\\nIL6, CXCL10\\nPredicted\\nTargetScan, miRDB\\nImmune response\\nModulation of endometrial immune tolerance\\n[30]\\nmiR‐1302‐3p\\nZC3H12C\\nPredicted\\nTargetScan\\nImmune regulation\\nCytokine homeostasis and inflammation\\n[30]\\nmiR‐222\\nCDKN1B, ERα (ESR1)\\nValidated\\nmiRTarBase, Literature\\nCell cycle, E2 signaling\\nAffects E2 signaling and endometrial cell receptivity\\n[31]\\nmiR‐27a\\nPPARG, VEGFA\\nPredicted\\nTargetScan, DIANA‐TarBase\\nAngiogenesis, hormonal regulation\\nVascularization, E2‐mediated endometrial development\\n[32]\\nmiR‐449b\\nCDK6, NOTCH1\\nPredicted\\nTargetScan, miRDB\\nCell cycle regulation, decidualization\\nStromal cell differentiation, epithelial receptivity\\n[32]\\nmiR‐423\\nPA2G4\\nPredicted\\nTargetScan, miRDB\\nCell proliferation, differentiation\\nEndometrial cell growth and receptivity\\n[32]\\nmiR‐604\\nFGF2, MMP9\\nPredicted\\nTargetScan, DIANA‐TarBase\\nExtracellular matrix remodeling, cell proliferation\\nTrophoblast invasion, ER\\n[32]\\nmiR‐34a\\nBCL2, SIRT1\\nValidated\\nmiRTarBase, Literature\\nApoptosis, hormonal sensitivity\\nControl of stromal cell survival, progesterone response\\n[33]\\nOpen in a new tab\\nmiR‐146a has been experimentally validated to target key inflammatory signaling molecules IRAK1 and TRAF6, modulating the NF‐κB pathway to maintain a balanced immune environment within the endometrium. This immunoregulatory function is crucial, as an aberrant inflammatory response may lead to rejection of the embryo and IF. Taganov et al. [34] first identified miR‐146a as a negative feedback regulator of NF‐κB signaling via suppression of IRAK1 and TRAF6 in innate immune responses. Further studies demonstrated that miR‐146a attenuates NF‐κB‐driven cytokine production in immune cells [35] and may participate in implantation‐related immune tolerance, as evidenced by altered expression in ART and pregnancy complications [36].\\nSimilarly, miR‐222 targets cell cycle regulator CDKN1B and E2 receptor ESR1, thereby controlling cell proliferation and hormonal signaling that orchestrate endometrial preparation and receptivity. Dysregulation of these pathways could impair E2‐mediated proliferation of endometrial epithelial cells during the implantation window. The pivotal role of miR‐34a in modulating apoptosis via BCL2 and hormonal sensitivity through SIRT1 further underscores the importance of tightly regulated stromal cell survival and progesterone responsiveness in decidualization and embryo acceptance. In support, Mahfouz et al. [37] showed that miR‐34a regulates SIRT1 and FoxO1 expression in endometrial tissue, contributing to reproductive pathology such as EMS.\\nBeyond these experimentally confirmed interactions, bioinformatic predictions highlight a suite of additional miRNAs potentially involved in implantation biology. For example, miR‐149 is predicted to regulate GIT1 and FOXM1, genes associated with cell proliferation and migration, both of which are critical for trophoblast invasion and endometrial remodeling. Chen et al. [38] demonstrated miR‐149 targeting of GIT1 in breast cancer models, offering mechanistic insights into its role in regulating cell motility and integrin signaling. Likewise, miR‐196a2 potentially targets HOXB8, ANXA1, and ERG, genes implicated in cellular differentiation and immune modulation, suggesting involvement in endometrial patterning and vascular remodeling essential for placental development.\\nFurther predicted interactions highlight the roles of angiogenesis and structural remodeling in implantation. miR‐27a is thought to target PPARG and VEGFA, implicating it in endometrial vascularization necessary for embryo support. Other miRNAs such as miR‐449b, miR‐423, and miR‐604 may influence stromal cell decidualization, epithelial receptivity, and extracellular matrix remodeling via predicted targets like CDK6, NOTCH1, PA2G4, FGF2, and MMP9. These processes are integral to successful implantation. Although miR‐608 and miR‐938 are predicted to target immune‐regulatory genes such as TP53, IL1A, IL6, and CXCL10, their associations remain inconsistent across prediction platforms and lack experimental validation. Nonetheless, their involvement in immune homeostasis warrants further investigation in the context of maternal‐fetal immune tolerance.\\nmiR‐1302‐3p represents a less‐characterized candidate in implantation biology. Although it has been detected in various tissues, including the placenta, its specific function in ER remains largely unknown. Current bioinformatic analyses, such as those from miRDB, have predicted potential target genes for miR‐1302‐3p, but these have not been validated in reproductive tissues [39]. Its limited characterization suggests that future studies are needed to clarify any potential roles in implantation.\\nCurrent research identified several miRNAs associated with RIF risk, including miR‐196a [29], miR‐449b [32], and miR‐34a [33], contrasting with miR‐1302‐3 [30], which exhibited differing genotype frequencies. Genotype combinations linked to increased RIF risk include miR‐146aCG+GG‐miR‐196a2CC, miR‐149TT‐miR‐196a2CC, miR‐196a2CC‐miR‐499AA [29], miR‐608GC‐miR‐938CC [30], miR‐27aAG‐miR‐423CC/miR‐604AG/GG [32] and miR‐34aC>A AA‐miR‐130aG>A GG [33]. Conversely, combinations such as miR‐631II‐miR‐1302‐3CT and miR‐938CC‐miR‐1302‐3CT were associated with a reduced risk [30]. Similarly, certain allele combinations were less frequently reported in RIF patients, including G‐T‐T‐A [29], C‐T [30], T‐T‐G, T‐T [31], and G‐C‐A‐G, G‐A‐G, A‐G‐G [32], while other haplotypes such as A‐T‐C [30], T‐C‐C‐T, T‐C‐T [31], A‐C‐G‐A, A‐A‐G‐G, G‐A‐A‐A [32], A‐A‐C‐A, and G‐G‐A [33] showed increased risk.\\nER is a complex, pivotal process for the successful embryo implantation [40, 41] regulated by differentially expressed miRNAs [42], especially in RIF [40]. These miRNAs play essential roles in the early stages of pregnancy, including decidualization and placental development [43]. Abnormal miRNA expression has been linked to blastocyst IF [44] and other reproductive disorders [45, 46] by affecting cellular processes such as cell adhesion and junctions [47, 48]. Genes related to RIF are enriched in Wnt signaling [49] and cyclin‐D2‐mediated cell cycle pathways [48], with dysregulation potentially impairing blastocyst implantation [41] and folliculogenesis [50]. MiRNA dysfunction is also implicated in premature ovarian failure (POF) [51], recurrent spontaneous abortion (RSA) [52], and recurrent pregnancy loss (RPL) [53].\\nEmerging studies reinforce the clinical importance of miRNA polymorphisms and expression profiles in the pathophysiology of RIF. For instance, Park et al. [54] demonstrated that allele combinations such as miR‐25T/miR‐125aT/miR‐222G reduced RIF risk, whereas miR‐25T/miR‐32 C/miR‐125a C/miR‐222T were linked to an increased risk. Notably, the miR‐222 GT+TT genotypes in conjunction with prolonged PT (≥ 12 s) significantly elevated the risk of RIF, highlighting a gene–environment interaction where miRNA variants may influence coagulation pathways critical for implantation.\\nFunctional analyses further elucidate molecular mechanisms. von Grothusen et al. [55] reported downregulation of hsa‐miR‐486‐5p and hsa‐miR‐92b‐3p in uterine fluid of RIF patients, implicating disrupted signaling pathways vital for endometrial preparation and embryo implantation. Similarly, Zhang et al. [56] found decreased miR‐30d‐5p in RIF endometrial samples, with upregulation of its target suppressor of cytokine signaling 1 (SOCS1) and reduced levels of implantation markers such as leukemia inhibitory factor (LIF) and phosphorylated STAT3. These molecular changes may impair pinopode formation and decidualization, ultimately hindering successful embryo attachment.\\nThese findings extend the clinical relevance of miRNA profiling by linking specific polymorphisms and expression changes to coagulation, hormonal, and immune pathways involved in implantation, underscoring the potential for miRNA‐based diagnostics and personalized therapies.\\nFrom a clinical perspective, these polymorphisms offer implications beyond diagnostics and may guide personalized therapeutic strategies. For example, variations in miR‐222 [31] and miR‐34a [33] influence FSH and TSH levels, suggesting a potential role for tailored hormonal supplementation. Patients with miR‐222 G>T variants exhibited elevated FSH [31], and those with the miR‐34a AA genotype showed decreased TSH [33], both of which could contribute to suboptimal ER and support the rationale for endocrine profiling and individualized hormonal therapy during IVF cycles.\\nAdditionally, several polymorphisms correlate with alterations in coagulation markers (PT, aPTT), which may warrant personalized thromboprophylaxis. Variants such as miR‐608GC+CC, miR‐1302‐3CC [30], miR‐27a AG+GG [32], and miR‐34a GA+AA [33] demonstrated significant associations with coagulation factor abnormalities. Given the established role of thrombophilia in RIF and miscarriage [57, 58, 59, 60, 61, 62], these findings could support the use of low‐dose aspirin or anticoagulants in genetically susceptible patients, although further clinical trials are needed.\\nMoreover, the immune milieu is another potential therapeutic target. The influence of E2 on immune tolerance via IL‐35 [63] and the interaction of miRNA polymorphisms with cytokine production (e.g., transforming growth factor beta—TGF‐β, IL‐10, IL‐35) highlight the importance of immune modulation. Variants impacting miR‐449 [32] and miR‐222 [31] may alter inflammatory signaling, supporting future exploration of immunomodulatory therapies such as corticosteroids or intravenous immunoglobulin (IVIg) in patients with RIF and relevant polymorphisms.\\nThus, while current studies focus primarily on diagnostic and prognostic associations, expanding the translational scope of miRNA polymorphism research to include personalized therapeutic interventions represents a promising future direction. These genetic insights may one day enable the development of individualized treatment protocols based on genotype‐specific profiles, ultimately improving implantation outcomes and pregnancy rates in patients with RIF.\\nTo accurately interpret the role of miRNAs in RIF, it is crucial to distinguish between two key aspects of miRNA biology: (1) miRNA polymorphisms, typically SNPs, which are inherited genetic variations that remain stable throughout an individual's lifetime and (2) miRNA expression levels, which are dynamic and responsive to environmental, hormonal, and epigenetic factors. Polymorphisms can affect miRNA biogenesis, maturation, or target binding, serving as static genetic risk markers for RIF [64]. Conversely, miRNA expression is modulated by various environmental exposures such as inflammation, smoking, nutritional status, and endocrine‐disrupting chemicals (EDCs) [65]. These expression changes are potentially reversible and thus represent promising therapeutic targets [66].\\nWhile miRNA polymorphisms represent stable genetic predispositions, growing evidence highlights that miRNA expression is also highly dynamic and responsive to environmental and epigenetic factors. Inflammatory states, oxidative stress (OS), and exposure to pollutants or EDCs can significantly alter miRNA profiles. For instance, exposure to diesel exhaust particles has been shown to disrupt miRNA expression in human airway cells, affecting gene networks involved in inflammation and disease [67]. Similarly, EDCs such as bisphenol A (BPA) and phthalates have been associated with dysregulation of miRNAs in reproductive tissues. BPA exposure was linked to increased miR‐146a levels in placental cells, correlating with adverse pregnancy outcomes [68], while phthalates have been shown to modulate expression of the miR‐34 family, which governs critical implantation‐related processes such as apoptosis and cell cycle regulation [69].\\nLifestyle factors also play a pivotal role in modulating miRNA expression and may interact with genetic predispositions to influence reproductive outcomes. Smoking, for example, has been linked to the upregulation of pro‐inflammatory miRNAs such as miR‐21 and miR‐146a, which may impair ER and increase the risk of IF [70]. In contrast, diets rich in fruits, vegetables, and omega‐3 fatty acids are associated with increased expression of protective miRNAs such as miR‐let‐7a and miR‐328 and a reduction in pro‐inflammatory miRNAs [70].\\nThese findings underscore the relevance of modifiable environmental and nutritional exposures in shaping the miRNA landscape. Integrating such factors with genetic screening may enhance clinical understanding of RIF pathophysiology and support the development of personalized interventions aimed at improving implantation success.\\n4.1. Strengths and Limitations of the Study\\nTo the authors' best knowledge, this is the first systematic review conducted on the role of miRNA polymorphisms in RIF, with the included case–control studies demonstrating intermediate‐to‐high methodological quality. A noteworthy strength is the uniformity of study settings, as all participants were enrolled from a single fertility center, specifically CHA Bundang Medical Center in Seongnam, South Korea, ensuring consistent clinical protocols and laboratory techniques.\\nHowever, several limitations must be considered when interpreting the findings. First, the exact molecular mechanisms by which the identified miRNA polymorphisms contribute to RIF remain to be fully elucidated. Second, the entire study population was comprised exclusively of Korean women, which limits the generalizability of these findings to broader populations. Ethnic differences in both miRNA polymorphism frequencies and expression levels are well‐documented in the literature. Third, none of the reviewed studies adjusted for potential confounding variables such as patient comorbidities, IVF protocols, or environmental exposures, which may interact with genetic predispositions to affect implantation success. Fourth, the sample sizes in the included studies were relatively modest, potentially limiting statistical power and the ability to detect less prevalent polymorphisms. Future investigations should consider larger, multi‐ethnic cohorts and incorporate environmental and clinical data to allow for more comprehensive and generalizable conclusions.\\nRawlings‐Goss et al. [71] identified 31 miRNA variants with significant allele frequency differences between African and non‐African populations, including a novel deletion in hsa‐mir‐4640 with functional implications for mRNA targeting. This emphasizes the influence of population‐specific architectures on disease susceptibility. Additionally, differential expression patterns of 351 miRNAs across individuals of African and European ancestry have been observed, potentially contributing to population‐specific disease phenotypes and treatment responses [72]. Gong et al. [73] investigated miRNA expression in breast cancer tissues from women of African and European ancestry. The researchers identified several miRNAs that were differentially expressed between the two groups, with some miRNAs showing distinct expression patterns in E2 receptor‐negative (ER−) tumors. For instance, miR‐105‐5p and miR‐767‐5p were downregulated in African American women compared to European American women, while miR‐187‐3p and miR‐937‐3p were upregulated in African American women.\\nXu et al. [74] conducted a meta‐analysis focusing on Asian populations and found that certain miRNA polymorphisms, such as rs2910164 in miR‐146a, rs11614913 in miR‐196a2, and rs3746444 in miR‐499, exhibited varying associations with cancer risk across different Asian subpopulations, highlighting the influence of genetic background on disease susceptibility. Similarly, Alimena et al. [75] investigated serum miRNA profiles across diverse racial and ethnic groups and discovered significant variations in miRNA expression levels, emphasizing the necessity for population‐specific validation of miRNA‐based biomarkers. Sonehara et al. [76] conducted a comprehensive analysis of miRNA expression quantitative trait loci (miRNA‐eQTLs) in a Japanese cohort, identifying 1275 cis‐miRNA‐eQTL variants for 40 miRNAs. Notably, 25 of these miRNAs with eQTLs were unreported in European studies, and five had lead variants monomorphic in European populations, highlighting the presence of population‐specific genetic architectures influencing miRNA expression. Similarly, a meta‐analysis by Yan et al. [77] revealed that the association between miRNA‐196a2 rs11614913 polymorphism and cervical cancer risk among different ethnic groups, highlighting the necessity for population‐specific studies.\\nTherefore, while the current findings provide valuable insights into miRNA polymorphisms associated with RIF in Korean women, caution should be exercised when extrapolating these results to other populations. Future research should aim to validate these associations in diverse ethnic cohorts to enhance the translational relevance and applicability of miRNA‐based diagnostics and therapeutic interventions in reproductive medicine.\",\n",
       "  'Conclusion': '5. Conclusions\\nIn conclusion, this systematic review highlighted that certain frequencies and combinations of genotypes and alleles of miRNAs are linked with either a protective role or an increased risk of RIF. Interactions with specific parameters can negatively impact pregnancy progression and may prompt further studies to explore miRNAs as potential prognostic or diagnostic biomarkers for RIF. This manuscript clarifies the essential evidence and supports the development of future randomized controlled trials (RCTs) to effectively translate findings into clinical practice by designing management strategies for women diagnosed with RIF. Prospectively, the integration of advanced transcriptomic technologies promises to refine our understanding of ER and its disruption in RIF. Techniques such as single‐cell RNA sequencing (scRNA‐seq) and spatial transcriptomics offer unprecedented resolution in mapping cell‐type‐specific gene expression and spatial organization within the endometrial microenvironment. Additionally, the profiling of circulating miRNAs from blood or uterine fluid may serve as a minimally invasive biomarker approach for patient stratification and monitoring treatment response. As these tools become increasingly accessible, they are expected to complement genomic association studies by unveiling dynamic regulatory networks, improving diagnostic precision, and informing personalized therapeutic strategies in RIF.',\n",
       "  'URL': 'https://doi.org/10.1002/rmb2.12670',\n",
       "  'Journal': 'Reproductive Medicine and Biology'},\n",
       " {'PaperTitle': 'Clinical outcomes following endometrial receptivity assessment-guided personalized euploid embryo transfer in patients with previous implantation failures',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'María Ruiz-Alonso, Carlos Gómez, Tiffany Stankewicz, José A Castellón, Antonio Díez-Juan, Eva Gómez, Carmen Rubio, Carlos Simón, Diana Valbuena',\n",
       "  'Email': '',\n",
       "  'Abstract': 'This study aimed to investigate potential improvements when implementing endometrial receptivity analysis (ERA)-guided personalized embryo transfer (pET) using euploid blastocyst in patients with one or more previous failed embryo transfers. A total of 270 patients with one or more previous failed embryo transfers were enrolled between 2017 and 2021 in this multicenter retrospective study. These patients were divided into two groups: study cases (ERA-guided pET) (n\\u2009=\\u2009200) or controls (standard embryo transfer) (n\\u2009=\\u200970). Clinical outcomes compared were pregnancy rate (PR), implantation rate (IR), ongoing pregnancy rate (OPR), and live birth rate (LBR). Clinical results in terms of PR, OPR, and LBR were significantly higher when performing ERA-guided pET (PR: 65.0%; OPR: 49.0%; LBR: 48.2%) compared to standard embryo transfer (PR: 37.1%; OPR: 27.1%; LBR: 26.1%) (P\\u2009<\\u20090.01). Logistic regression was applied to examine the correlation between ERA and the primary outcome measured, OPR, including demographic variables as covariates. The effect of ERA was significantly associated with OPR (P\\u2009=\\u20090.002; aOR 2.8, 95% CI 1.5–5.5); furthermore, OPR decreased significantly when body mass index (BMI) values increased (P\\u2009=\\u20090.04; aOR 0.9, 95% CI 0.8–0.98). These findings support the potential of ERA-guided pET to improve clinical outcomes and address the challenges encountered by patients with previous implantation failures.',\n",
       "  'Introduction': 'While Homo sapiens have evolved to boast high levels of intelligence, our fertility rates have not kept pace. A systematic review published in 2022 estimated that the infertility rate in reproductive-aged couples could reach 17.5%1. The window of implantation (WOI), spanning 30–36 h, marks the critical period when the maternal endometrium becomes receptive to blastocyst implantation. This window typically occurs six to eight days after the luteinizing hormone (LH) surge in natural cycles or four to seven days following endogenous or exogenous progesterone exposure2–4.\\nDuring the 2000´s, research focusing on bulk endometrial transcriptomics identified gene expression profiles related to the different phases of the menstrual cycle. Four independent groups simultaneously reported the transcriptomic profiling of the secretory phase of the human endometrium during natural cycles, searching for the WOI5–8. Two additional groups extended this transcriptomic characterization across the menstrual cycle9,10. Subsequent studies expanded their scope to include controlled ovarian stimulation (COS) cycles11–13 and even refractory cycles in patients with inert intrauterine devices (IUD)14.\\nThe pioneering transcriptomics-based “Endometrial Receptivity Analysis” (ERA) test was designed to diagnose endometrial receptivity status in infertile patients experiencing implantation failure originating from the endometrium15. The ERA test enables personalized embryo transfer (pET) by aligning the individualized WOI of each patient with the developmental stage of their blastocyst. In addition, single-cell RNA sequencing-based research has validated this comprehensive transcriptomics dataset16.\\nUnfortunately, the results of ERA-guided pET remain contentious after over a decade of clinical application. Thus far, 43 publications containing varying levels of evidence have meticulously scrutinized the clinical outcomes of ERA-guided pET across specific clinical indications and geographical locations, leading to divergent conclusions (Supplementary Table 1 provides comprehensive coverage of published studies).\\nNotably, four recent meta-analyses - considered the pinnacle of the evidence-based pyramid – have been published. Three of the four suggest the potential utility of assessing endometrial receptivity in patients experiencing repeated implantation failure (RIF) of endometrial origin17–19, while a study by Arian20 suggested no benefit. Recently, findings from the first randomized controlled trial evaluating the ERA test in patients with RIF supported the clinical benefit of euploid pET21.\\nSince first appearing in the early 1990´s, PGT-A has become a critical tool in any in vitro fertilization (IVF) center, helping professionals to increase pregnancy rates and decrease miscarriage rates22. Nevertheless, the transfer of a euploid embryo does not always lead to the achievement of a pregnancy owing to the implication of additional factors, such as the endometrial factor. Thus, combining PGT-A and the ERA test may provide an important clinical benefit to IVF patients. Multiple studies sought to explore the potential benefits of this approach, with reported results both for21,23–26 and against27–31 the implementation of this combination. Of note, most of these studies involved RIF or first appointment population, with low representation of patients with just 1–2 previous implantation failures, which is where the scientific evidence remains limited.\\nThis study aims to explore an intermediate perspective by assessing the clinical outcomes of ERA-guided pET in patients who have experienced at least one failed embryo transfer and underwent preimplantation genetic testing for aneuploidy (PGT-A).',\n",
       "  'Methods': 'Study design\\nThis is a multicenter retrospective study comparing ERA-guided pET versus standard embryo transfer, both with euploid blastocysts, in patients with one or more previous failed embryo transfers.\\nPatient population\\nThis study involves 270 patients with at least one previous failed embryo transfer performed between 2017 and 2021. Data were exported from our electronic health database and anonymized in compliance with specific legislation on biomedical research, personal data protection, and bioethics. The study was evaluated and approved by the independent Scientific Committee and the Ethical Committee of the Hospital Clínico Universitario de Valencia (approval No. 2021/018) and it was performed in accordance with relevant guidelines and regulations. The consent form waiver for this retrospective study was also approved by the independent Scientific Committee and the Ethical Committee of the Hospital Clínico Universitario de Valencia. Patients were divided into two groups: the study group underwent ERA-guided, euploid pET (n = 200), and the control group underwent standard euploid embryo transfer (n = 70). The transfer of euploid embryos assures the lack of chromosomal abnormalities in embryos as a potential cause of implantation failure or miscarriage. This study reports one embryo transfer per patient, considering the first embryo transfer after performing the ERA test in the pET group. All embryo transfers for both groups were performed in hormone replacement therapy (HRT) cycles. Single and double embryo transfers (SET and DET, respectively) were considered in this study. Variables such as age, body mass index (BMI), previous failed attempts, number of metaphase II (MII) and fertilized oocytes, number of vitrified embryos and transferred embryo quality were compared between the two groups. Embryo quality was evaluated at the inner cell mass and trophectoderm levels according to Gardner’s criteria32, classifying the embryos as optimal (categories A + B) or suboptimal (categories C + D).\\nERA endometrial biopsy, analysis, and interpretation of results\\nAn endometrial biopsy for ERA was obtained during an HRT cycle. The type and dosage of progesterone varied slightly depending on the clinic’s standard of care. For example, a typical HRT cycle involves estradiol priming (delivered orally [6 mg daily] or via patches [two patches every two days]) on the first or second day of the menstrual cycle. Ultrasound assessment is performed seven to ten days after estradiol priming starts. Progesterone intake starts when a trilaminar endometrium >\\u20096 mm is reached with serum progesterone levels of <\\u20091 ng/ml (within 24 h before exogenous progesterone starts). Progesterone is administered for five full days (approximately 120 h), with 400 mg of micronized vaginal progesterone given every 12 h (800 mg per day). Overall, the first intake of progesterone occurs on day P + 0, with endometrial biopsies obtained on day P + 5.\\nThe endometrial biopsy isolates a small uterine lining sample at the fundus. A pipette is inserted through the vagina and cervix into the uterus, where a small piece of tissue is extracted.\\nThe ERA molecular diagnostic tool utilizes next-generation sequencing (NGS) to analyze the expression levels of 248 genes related to endometrial receptivity status. The ERA computational predictor then identifies the specific transcriptomic signatures for each endometrial stage: proliferative, pre-receptive, receptive, late receptive, and post-receptive. A pET recommendation is then offered based on this result, indicating the optimal moment of receptivity during the menstrual cycle. For patients with a receptive result, pET is recommended following the same conditions and timing in which the ERA biopsy was collected. For patients who received a “pre-receptive” result, pET is recommended a specified number of hours/days later in the cycle in relation to the time the ERA biopsy was collected. For patients who received a “post-receptive” or a “late receptive” result, pET is recommended at a specified number of hours/days earlier in relation to the time the ERA biopsy was collected.\\nEmbryo transfer and clinical outcomes\\nAfter identifying the patient’s WOI, ERA-guided pET was performed for patients in the ERA group. Participants in the control group underwent standard embryo transfer. Transfers of euploid embryos were performed in HRT cycles in all cases for both groups.\\nClinical outcomes compared between both groups were pregnancy rate (PR), implantation rate (IR), ongoing pregnancy rate (OPR), and live birth rate (LBR). PR was calculated as the percentage of all patients positive for free beta human chorionic gonadotrophin (βhCG); IR was defined as the number of gestational sacs observed by vaginal ultrasound at the fifth gestational week divided by the total number of embryos transferred; OPR was calculated as the percentage of pregnancies that continued beyond the 12 th gestational week for all embryo transfers; and LBR was calculated as the number of deliveries that resulted in at least one live birth per embryo transfer from the total of patients receiving an embryo transfer with completed follow-up until delivery.\\nStatistical analyses\\nThe Wilcoxon rank-sum test was applied to quantitative comparisons. Categorical variables were expressed in percentages and compared using Chi-square or Fisher exact tests. Multivariate logistic regression was used to evaluate the association between ERA and OPR. Maternal age, BMI, previous failed attempts, number of embryos transferred, and embryo quality were covariates to calculate the adjusted odds ratio (aOR). The corresponding confidence intervals were also computed for each coefficient. Statistical analysis was performed using statistical software R (version 4.2). p values ≤\\u20090.05 were considered statistically different.',\n",
       "  'Results': 'In total, 270 patients were enrolled in this study: 200 in the ERA-guided pET group and 70 in the standard euploid embryo transfer (control) group. Among the pET group, 117 patients displayed a receptive result (58.5%), while 83 exhibited a displaced WOI (41.5%), with 74 (89.2%) being pre-receptive, 6 (7.2%) late receptive and 3 (3.6%) post-receptive. We compared demographic variables, including age, BMI, number of previous failed embryo transfers, number of MII oocytes, number of fertilized oocytes, number of vitrified embryos, and the number of embryos transferred, and morphological analyses of embryo quality of transferred blastocysts between groups (Table 1). These comparisons revealed no significant differences between the pET and control groups.\\nWe observed significant differences in clinical outcomes when comparing patients undergoing ERA-guided pET versus standard embryo transfer of euploid blastocysts. The PR reached 65.0% in the pET group, significantly surpassing the 37.1% observed in the control (P < 0.001). Additionally, we observed a significantly higher OPR in the pET group (49.0%) compared to control (27.1%) (P = 0.002). We obtained follow-up information until delivery in 197/200 patients in the pET group and 69/70 patients in the control group, revealing a significantly higher LBR in the pET group (48.2%) compared to the control group (26.1%) (P = 0.002). The remaining early pregnancy loss rates (clinical miscarriages, biochemical and ectopic pregnancy rates) remained similar in both groups (Table 2).\\nA sub-analysis including just cycles with SET provided a similar trend, with significant differences for the pET group compared to the control group (Supplementary Table 2).\\nWe next conducted a subclassification based on the number of previous failed embryo transfers (one or two or more previous failed embryo transfers) (Table 3). For those cases with one previous failed embryo transfer, the comparison revealed a higher IR (56.6% versus 28.6%; P = 0.01) and PR (65.7% versus 35.3%; P = 0.007) when applying ERA. Furthermore, we found a significantly higher OPR (47.1% versus 23.5%; P = 0.04) and LBR (46.4% versus 23.5%; P = 0.04) in the pET group.\\nIn cases with two or more previous failed embryo transfers, we found a significantly higher IR in the pET group versus the control group (55.8% versus 29.6%; P = 0.004) (Table 3) and a significantly higher PR in the pET group versus the control group (64.6% versus 38.9%; P = 0.01). For the OPR, we observed a similar trend in favor of the pET group as for cases with one previous failed embryo transfer; however, this difference failed to reach significance due to the smaller sample size (50.0% versus 30.6%; P = 0.06) (Table 3). Finally, we found a significantly higher LBR in the pET group compared to the control group (49.2% versus 28.6%; P = 0.047) (Table 3).\\nWe also compared clinical outcomes in the pET group between patients with standard and displaced WOIs. We did not encounter significant differences in the pET group, as transfers were performed according to their personalized WOI (Table 4).\\nFinally, we undertook a multivariant binomial logistic regression to study the relationship between ERA-guided pET and the principal studied variable (OPR), considering interfering control variables. This analysis revealed that ERA significantly influenced the OPR (P = 0.002; aOR 2.8, 95% CI 1.5–5.5) (Supplementary Table 3). Additionally, we discovered that the BMI significantly impacted the OPR (P = 0.04; aOR 0.9, 95% CI 0.8–0.98) (Supplementary Table 3).',\n",
       "  'Discussion': 'The WOI represents a specific time frame during the menstrual cycle when the endometrium allows embryo implantation; therefore, understanding and accurately pinpointing the WOI remains important for a successful embryo transfer. Endogenous factors play a significant role in shaping the WOI, with hormonal imbalances, irregular menstrual cycles, high BMI, or conditions like polycystic ovary syndrome known as disrupting factors33–36.\\nAn extensive body of research has revealed the greater prevalence of implantation difficulties in certain groups, such as RIF patients or older individuals37,38. RIF - a challenging condition associated with multiple unsuccessful embryo transfers - could arise due to issues related to the WOI; therefore, determining the timing of endometrial receptivity and employing ERA-guided pET could represent an important means of improving implantation2,39.\\nOur study reveals that ERA-guided pET supports a significantly higher PR, OPR, and LBR than standard embryo transfer (65.0% versus 37.1%, 49.0% versus 27.1%, and 48.2% versus 26.1%, respectively). These results provide evidence that ERA-guided pET notably enhances clinical outcomes, specifically in cases involving patients with previous implantation failures when the use of PGT-A rules out the embryo factor. Furthermore, multivariate testing revealed a relationship between BMI and OPR in PGT-A patients, underlining the relevance of female weight to outcomes when minimizing the impact of the embryo factor35,36. Obese women suffer from worse results and a high number of miscarriages that negatively affect LBR after euploid embryo transfer40–42. These data reinforce the importance of endometrial factors on reproductive outcomes, given the relationship between BMI and the WOI36,43.\\nSubclassification based on the number of previous failed embryo transfers (one or two or more) further demonstrated improved clinical outcomes with ERA-guided pET compared to standard ET, revealing that the ERA effectively addressed cases of implantation failure (Table 3). A deeper analysis of the pET group revealed that the correction in patients with a displaced WOI supported clinical results similar to those without WOI displacement.\\nOf note, the gradual nature of the changes in the endometrial transcriptome during the menstrual cycle means that not all cases of WOI displacement have equal relevance at the clinical level. The possibly frequent occurrence of an endometrium existing in a transition between pre-receptivity and receptivity would make the adjustment of progesterone exposure timing irrelevant in a healthy population with a normal hormonal balance, regular menstrual cycles, and no prior implantation failures. This has been addressed in some studies20 27, raising fundamental questions regarding the appropriate application of ERA-guided pET in assisted reproductive technologies. However, most of these studies were focused on good prognosis patients, where the benefit of ERA is less evident. In contrast, recent studies, including one randomized controlled trial and recent meta-analyses, have shown the potential benefits of ERA-guided pET in patients with RIF2,21,39. Thus, selecting patients with previous failed embryo transfers may serve as a filter to select clinically relevant WOI displacements; employing ERA-guided pET in these cases can make a significant difference, as observed in the present study and other studies reporting similar findings2,39.\\nConsidering the evolving technological and clinical landscape, these findings warrant further prospective controlled trials to confirm and generalize our findings. These prospective studies would support the comprehensive investigation of other factors influencing implantation success to ensure a holistic approach to improving fertility treatment outcomes and to refine the criteria for selecting the most appropriate transfer method for each patient.\\nWhile this analysis depends on pre-existing data and suffers from a risk of bias due to the study’s retrospective nature, our findings offer a solid foundation for prospective future research.',\n",
       "  'Conclusion': 'In this study we have found that 41.5% of patients from the ERA-guided pET group presented WOI displacement. These patients showed similar clinical outcomes to those with a non-displaced WOI, after WOI correction. Furthermore, results have shown that ERA-guided pET on euploid embryos improved clinical outcomes in patients with at least one previous failure attempt. These results support that ERA-guided pET has the potential to enhance clinical outcomes, demonstrating chances in tackling the difficulties faced by patients with previous implantation failure These findings underscore the potential of personalized medicine to increase reproductive success and warrant further investigation in large well-designed clinical trials.',\n",
       "  'URL': 'https://doi.org/10.1038/s41598-025-01056-5',\n",
       "  'Journal': 'Sci Rep'},\n",
       " {'PaperTitle': 'Effects of standard and low doses of estradiol on markers of endometrial receptivity in frozen-thawed embryo transfer cycles: Double-blind, randomized clinical trial',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Nichamon Parkpinyo, Isarin Thanaboonyawat, Japarath Prechapanich, Pitak Laokirkkiat, Roungsin Choavaratana, Suchanan Hanamornroongruang, Somsin Petyim',\n",
       "  'Email': 'somsin101@yahoo.com',\n",
       "  'Abstract': 'Background\\nEndometrial receptivity biomarkers, specifically the regulatory proteins HOXA-10 and HOXA-11 as well as the integrin αvβ3, play vital roles in implantation during the window of implantation. This Double-blinded, randomized clinical trial compares the effects of two initial doses of oral estrogen regimens on endometrial receptivity markers during the window of implantation in endometrial preparation for frozen-thawed embryo transfer.\\nMethods and findings\\nThe study includes infertile patients who underwent in vitro fertilization and planned frozen-thawed embryo transfer at the infertility clinic between June 2018 and March 2019. Fifty patients were randomized to a low-dose (4 mg/day) or standard-dose (6 mg/day) estradiol group for artificial endometrial preparation. On the first visit (day 12 of the cycle), measurements of mean endometrial thickness and estradiol and progesterone levels were taken. Following this visit, patients received 600 mg daily of micronized progesterone for 7 days. On the second visit (day 19 of the cycle), hormonal levels were reassessed, and an endometrial biopsy was performed for immunohistochemical analysis. The primary outcome was the expression level of HOXA-10. Secondary outcomes included the expression levels of HOXA-11 and integrin αvβ3, mean endometrial thickness, as well as serum estradiol and progesterone levels measured at various phases of the endometrial cycle. The mean age of the participants was 36 years. The standard-dose group exhibited significantly greater intensity scores for the expression of the regulatory proteins HOXA-10 and HOXA-11 and the integrin αVβ3 than did the low-dose group (75.84\\u2009±\\u200911.25 vs 61.53\\u2009±\\u200911.05, 107.08\\u2009±\\u200919.42 vs 87.62\\u2009±\\u20099.40, and 90.25\\u2009±\\u200910.42 vs 76.32\\u2009±\\u200912.98, respectively; P <\\u20090.001). The groups had no significant differences in mean serum estradiol level, progesterone level, or endometrial thickness during the artificial cycle at the first or second visit.\\nConclusions\\nOptimal artificial endometrial preparation for embryo transfer in a frozen-thawed embryo transfer cycle is crucial for maximizing implantation outcomes. This study suggests that the administration of a fixed standard dose of 6 mg of estradiol for artificial endometrial preparation should be considered.',\n",
       "  'Introduction': 'Globally, in vitro fertilization is being increasingly utilized. Advances in assisted reproductive technologies and ovarian hyperstimulation protocols have enabled the cryopreservation of standard-quality embryos for subsequent frozen-thawed embryo transfer cycles. Understanding the nuances of endometrial preparation in the frozen-thawed embryo transfer cycle is pivotal for successful implantation [1]. Exogenous administration of estrogens and progestogens in a precise sequence can artificially induce optimal endometrial preparation [2]. Various protocols for artificial endometrial preparation have been explored, with constant and increasing doses of estrogen being the most noteworthy. Madero et al reported a significantly greater biochemical pregnancy rate with a constant-dose regimen than with an increasing-dose protocol [3].\\nEmbryo implantation, a critical determinant of pregnancy success, progresses through three stages: apposition, adhesion, and invasion [4]. Success hinges on synchronizing embryo development and endometrial receptivity [5]. While impaired embryogenesis accounts for approximately one-third of implantation failures, aberrant endometrial receptivity is responsible for the remaining cases [6]. Endometrial receptivity, particularly during the window of implantation, is instrumental in accepting an embryo. The window of implantation is a finite period characterized by critical cellular and molecular events that facilitate implantation, typically occurring between days 20 and 24 of the menstrual cycle or 7–9 days postovulation [7, 8].\\nDiagnostic methods for determining endometrial receptivity include endometrial thickness measurement, endometrial histological dating [9], and molecular marker assessment via transcriptomics [10, 11], proteomics [12], and secretomics [13]. Biomarkers, specifically the regulatory proteins HOXA-10 and HOXA-11 as well as the integrin αvβ3, play vital roles in implantation during the window of implantation [14–18]. The HOXA-10 and HOXA-11 transcriptional regulators are significantly expressed in the endometrial glands and stroma during the mid-luteal phase, the time when implantation occurs [17]. These regulators are vital for uterine organogenesis and functional differentiation of the endometrium [15]. Integrins are essential heterodimeric glycoproteins that fulfill numerous roles within the endometrium [6]. Notably, the integrin αvβ3, which is crucial for implantation, appears on luminal and glandular cell surfaces as the window of implantation commences.\\nPrevious studies examining estrogen use in artificial endometrial preparation for frozen-thawed embryo transfer cycles have focused primarily on endometrial thickness or histological dating, with neither providing a full representation of endometrial quality or quantity [19]. The influence of endometrial factors on pregnancy outcomes presents challenges in accurately assessing endometrial receptivity [3, 20]. Optimal estradiol dosing is crucial for developing a standard-quality endometrium, creating the conditions for a progesterone-induced secretory phase and ensuring a receptive endometrium for embryo implantation.\\nThis double-blinded, randomized clinical trial aimed to evaluate the impact of two different estrogen doses on endometrial receptivity markers during the window of implantation in endometrial preparation for frozen-thawed embryo transfer. The primary outcome was the expression of the endometrial receptivity gene HOXA-10. Secondary outcomes included the expression of other endometrial receptivity markers—HOXA-11 and integrin αvβ3—along with mean endometrial thickness, and serum levels of estradiol and progesterone measured at various stages of the endometrial cycle.',\n",
       "  'Methods': 'Study design\\nThis double-blinded, randomized clinical trial was conducted over 10 months from June 2018 to March 2019 at the infertility clinic of the Faculty of Medicine Siriraj Hospital, Bangkok, Thailand. The Siriraj Institutional Review Board authorized the study protocol (approval number 653/2560[EC2]), and the research was registered with the Thai Clinical Trials Registry (TCTR20180718003). Before study enrollment, all participants received comprehensive counseling and provided written informed consent.\\nStudy population\\nFifty infertility patients scheduled for frozen-thawed embryo transfer in the in vitro fertilization program were recruited for the study. The inclusion criteria were age between 18 and 45 years; normal follicle-stimulating hormone, luteinizing hormone, and estradiol levels during the early follicular phase; and regular menstrual cycles during the 3 months before enrollment. The exclusion criteria were contraindications to estrogen use (for example, endometrial cancer, unexplained bleeding, thromboembolism, or liver disease); uterine cavity abnormalities (such as synechiae or polyps); or estrogen use during the 3 months before enrollment.\\nStudy procedure\\nThe demographic data collected comprised age, body mass index, live birth and miscarriage history, curettage history, and infertility duration. Patients were randomized into low-dose or standard-dose estradiol hemihydrate (Estrofem) groups using computer-generated randomization, with 25 individuals per group. Allocation concealment was maintained with sealed envelopes, and both patients and the investigator were blinded to the assignments. The gynecologists who assessed the endometrial thickness were also blinded to the treatment allocation. The low-dose group received 4 mg/day of EstroFem, which was taken as two capsules containing 1-mg of Estrofem in the morning and evening (total dose is 4 mg/day). The standard-dose group was prescribed 6 mg/day, which involved taking one capsule containing 1-mg of Estrofem and one capsule containing 2-mg of Estrofem twice daily (total dose is 6 mg/day). The capsules containing 1-mg and 2-mg of Estrofem look alike. Treatment began on day 2 of the menstrual cycle and continued for 10 days.\\nAt the initial follow-up, which was conducted on day 12 of the menstrual cycle, endometrial thickness was assessed via transvaginal ultrasonography (Xario 100; Toshiba Medical Systems, Japan) via a sagittal view of the uterus. A single experienced gynecologist, who was blinded to the treatment allocation, performed this assessment. Blood samples for estradiol and progesterone level determination were drawn. The compliance with the medication was verified during the study period by counting the returned capsules in the medication pack. Following the initial visit, participants commenced a regimen of micronized progesterone (Utrogestan), prescribed at a daily dosage of 600 mg. This dosage was administered in three doses daily for 7 consecutive days. At the second follow-up (day 19 of the cycle), blood samples were collected to measure the estradiol and progesterone levels, and an endometrial biopsy was obtained. In the subsequent menstrual cycle, standard procedures for endometrial preparation were followed in anticipation of frozen-thawed embryo transfer.\\nEndometrial biopsy procedure\\nDuring the second follow-up (day 19 of the menstrual cycle), an endometrial biopsy was performed on all patients utilizing the endometrial aspiration technique with a Wallach Endocell endometrial sampling device (CooperSurgical, USA). The procedure entailed the insertion of a vaginal speculum, followed by the identification and cleaning of the cervix and cervical canal. The cervical mucus was removed, and the cervix was disinfected with a normal saline solution. The Endocell device was then carefully inserted through the cervical os to the uterine fundus, where negative pressure was applied to aspirate the endometrial tissue. Sampling occurred continuously throughout the uterine cavity, and each procedure was carried out by the same operator. The collected endometrial tissue was promptly placed in a tissue cassette for paraffin embedding and submerged in 10% neutral-buffered formaldehyde for fixation overnight in preparation for subsequent immunohistochemical analysis. Morphological examination of all endometrial tissues was performed according to the histological dating criteria established by Noyes [21]. To ensure objectivity, a senior pathologist, who was blinded to the study details, was tasked with interpreting the histological dating and immunobiological findings.\\nImmunohistochemical staining protocol\\nAfter endometrial biopsy, the specimens were immediately fixed in 10% neutral-buffered formaldehyde and subsequently embedded in paraffin wax. The embedded tissues were sectioned into 3-µm slices, affixed onto SuperFrosted Plus adhesion slides (Thermo Scientific, UK) and incubated overnight at 60°C. The specimens were then autostained using the Ventana BenchMark Ultra Staining System (N750-BMKU-FS 05342716001; Roche) following an optimized immunohistochemical staining protocol for detecting the expression of the regulatory proteins HOXA-10 and HOXA-11 and the integrin αvβ3. For HOXA-10 detection, antigen retrieval was performed at 95°C for 64 minutes in OptiView CC1 (pH 6.0), and sections were then incubated with a rabbit polyclonal anti-human HOXA-10 primary antibody (GTX37412, GeneTex) at a 1:50 dilution at 37°C. The milder protocol for HOXA-11 and integrin αvβ3 involved antigen retrieval in Mild Ultra CC1 (pH 6.0) for 52 minutes. This was followed by incubation with rabbit polyclonal anti-human HOXA-11 (GTX48983, GeneTex, 1:200 dilution) and mouse monoclonal anti-human αVβ3 integrin (MAB1976, Merck, 1:1500 dilution) primary antibodies at 37°C. Detection was carried out using an Ultraview Universal DAB Detection Kit. Finally, the sections were counterstained with hematoxylin, mounted with Permount solution, and secured with a coverslip. The process of staining and analysis of the specimens for the HOXA-10, HOXA-11 and for integrin αvβ3 expression were standardized before assessment.\\nControl measures included the use of slides of normal secretory-phase endometrial epithelium with referral staining procedures as positive controls. Slides processed without primary antibodies served as negative controls.\\nImmunohistochemical staining analysis\\nThe intensity of immunohistochemical staining for the regulatory proteins HOXA-10 and HOXA-11 and for integrin αvβ3 was quantified via intensity scores using ImageJ 1.46r/Java 1.6.0_20 (64-bit, USA). This scoring focused on the glandular epithelium of each section and was independently conducted by two observers. The final scores for each protein were derived by averaging the scores assigned to each compartment.\\nStatistical analysis\\nDemographic and descriptive statistics are presented as the mean\\u2009±\\u2009SD or median (min–max), as appropriate. Comparative analyses between groups were conducted using an independent t test for normally distributed data and the Mann–Whitney U test for nonnormally distributed data. Pearson’s correlation was used to assess the relationship between the mean endometrial thickness and immunohistochemical staining intensity in the different groups. Data analysis was performed using PASW Statistics, version 18 (SPSS Inc, Chicago, IL, USA), with P values\\u2009<\\u20090.05 indicating statistical significance.\\nThe sample size calculation was based on the previous study [17]. The sample size was calculated based on HOXA-10. HOXA-10 functions as an upstream master regulator, controlling the expression of other key genes involved in uterine development and endometrial receptivity [17]. The formula for testing two independent means was used. With a Type I error (α) of 0.05 and a power of 90%, a sample size of 46 women was necessary for the study. When accounting for a dropout rate of 10%, the total sample size needed was 50 women. The appropriate statistical test is the Independent samples t-test. The effect size measured using Cohen’s d is 0.97.',\n",
       "  'Results': 'Among the 78 patients assessed for eligibility, 50 were included in this study and randomized into two groups. In the low-dose group, one patient’s data were excluded due to a diagnosis of endometrioid adenocarcinoma of the uterus identified in the final pathological report. In the standard-dose group, data from one patient were discarded due to incorrect drug consumption. Therefore, the analyses were performed on a per-protocol basis (Fig 1). All 48 patients in the final analysis compliant to study protocol. Demographic and clinical characteristics (age, body mass index, number of live births and miscarriages, history of curettage, and duration of infertility) were recorded (Table 1). The mean ages were 36.25\\u2009±\\u20094.28 years in the low-dose group and 36.92\\u2009±\\u20092.60 years in the standard-dose group.\\nFig 1. Flow diagram of the double-blinded, randomized clinical study design.\\nOpen in a new tab\\nTable 1. Demographic and clinical characteristics and outcomes of the study participants.\\nVariables\\nLow-dose estradiol (n\\u2009=\\u200924)\\nStandard-dose estradiol (n\\u2009=\\u200924)\\nP\\nAge (yrs)\\na\\nBody mass index (kg/m\\n2\\n)\\na\\nDuration of infertility (yrs)\\na\\nLive births\\nb\\nMiscarriages\\nb\\nCurettages\\nb\\n36.25\\u2009±\\u20094.28\\n22.40\\u2009±\\u20094.33\\n6.79\\u2009±\\u20093.68\\n0 (0-3)\\n0 (0-3)\\n0 (0-2)\\n36.92\\u2009±\\u20092.60\\n22.79\\u2009±\\u20094.76\\n6.71\\u2009±\\u20093.21\\n0 (0-1)\\n0.5 (0-3)\\n0 (0-2)\\n0.517\\n0.775\\n0.934\\n0.292\\n0.313\\n0.196\\nOpen in a new tab\\nNote:\\na\\nThe data are presented as means\\u2009±\\u2009standard deviations.\\nb\\nThe data are presented as medians (min–max).\\nExpression of the regulatory proteins HOXA-10 and HOXA-11 and integrin αvβ3\\nImmunohistochemical analysis revealed differential expression of the regulatory proteins HOXA-10 and HOXA-11 and of integrin αvβ3 between the groups. The intensity of HOXA-10 expression was significantly greater in the standard-dose group than in the low-dose group (75.84\\u2009±\\u200911.25 vs 61.53\\u2009±\\u200911.05; P <\\u20090.001; Figs 2, 3a, and 3b). Similarly, HOXA-11 expression was significantly greater in the standard-dose group (107.08\\u2009±\\u200919.42 vs 87.62\\u2009±\\u20099.40; P <\\u20090.001; Figs 2, 3e, and 3f). The intensity of integrin αvβ3 expression was also significantly greater in the standard-dose group (90.25\\u2009±\\u200910.42 vs 76.32\\u2009±\\u200912.98; P <\\u20090.001; Fig 2, 3i, and 3j).\\nFig 2. Comparative immunohistochemical staining intensities of the regulatory proteins HOXA-10 and HOXA-11 and integrin αvβ3 in glandular tissue: low-dose (4 mg/day) vs standard-dose (6 mg/day) estradiol groups.\\nOpen in a new tab\\nThe data plotted as mean\\u2009±\\u2009SD. * P <\\u20090.001.\\nFig 3. Differential expression of the regulatory proteins HOXA-10 and HOXA-11 and integrin αvβ3 in endometrial tissue: immunohistochemistry analysis across low estradiol doses (a, e, i) and standard estradiol doses (b, f, J) with positive controls (c, g, k) and negative controls (d, h, l).\\nOpen in a new tab\\nRepresentative images of (a, b, c) HOX A10, (e, f, g) HOX A11, and (i, j, k) integrin αvβ3 expression in low estradiol doses, standard estradiol doses, and positive control. Magnification\\u2009=\\u2009x400.\\nLocalization of the regulatory proteins HOXA-10 and HOXA-11 and integrin αvβ3\\nThe specificity of the immunohistochemical staining was confirmed using normal secretory-phase endometrial epithelium as a positive control. HOXA-10 was predominantly localized in the cytoplasm of glandular epithelial cells and was less abundant in the nuclei of stromal cells (Fig 3c). HOXA-11 exhibited similar cytoplasmic localization (Fig 3g). The integrin αVβ3 was expressed in both glandular epithelial and stromal cells (Fig 3k). Histological evaluation of endometrial biopsies confirmed an adequate secretory-phase endometrium for days 18 and 19, according to the Noyes criteria, characterized by decreased vacuolization, basal nuclei, and improved cellular alignment.\\nHormone levels and endometrial thickness\\nThe serum hormone levels and mean endometrial thickness data collected during the artificial cycle are detailed in Table 2. The mean serum estradiol levels at the first visit were 220.55\\u2009±\\u2009117.25 pg/ml in the low-dose group and 229.98\\u2009±\\u200980.06 pg/ml in the standard-dose group, with no significant difference (P =\\u20090.746; Table 2). At the second visit, estradiol levels were observed to be lower than those recorded at the first visit in both groups, although these changes did not reach statistical significance (P =\\u20090.317; Table 2). The mean serum progesterone levels at the first and second visits were not significantly different between the groups (0.16 [0.10, 0.22] ng/ml vs 0.14 [0.06, 0.20] ng/ml, median [IQR]; P =\\u20090.215; and 11.69 [7.51, 14.45] ng/ml vs 10.82 [7.59, 13.65] ng/ml; P =\\u20090.592, respectively). The mean endometrial thickness on day 12 of the cycle was 7.84\\u2009±\\u20091.69 mm in the low-dose group and 8.75\\u2009±\\u20092.22 mm in the standard-dose group, without a statistically significant difference (P =\\u20090.115; Table 2). Pearson’s correlation analysis revealed no significant relationship between the mean endometrial thickness and the intensity of immunohistochemical staining for the regulatory proteins HOXA-10 and HOXA-11 or for integrin αvβ3 (r\\u2009=\\u2009−0.107, −0.074, and −0.052, respectively; P >\\u20090.05).\\nTable 2. Comparative analysis of serum estradiol and progesterone levels and endometrial thickness during artificial cycles among different treatment groups.\\nVariables\\nLow dose (n\\u2009=\\u200924)\\nStandard dose (n\\u2009=\\u200924)\\nP\\nEstradiol on day 12 (pg/ml)\\na\\nEstradiol on day 19 (pg/ml)\\na\\nProgesterone on day 12 (ng/ml)\\nb\\nProgesterone on day 19 (ng/ml)\\nb\\nEndometrial thickness on Day 12 (mm)\\na\\n220.55\\u2009±\\u2009117.25\\n181.20\\u2009±\\u200990.71\\n0.16 (0.10, 0.22)\\n11.69 (7.51, 14.45)\\n7.84\\u2009±\\u20091.69\\n229.98\\u2009±\\u200980.06\\n205.72\\u2009±\\u200976.46\\n0.14 (0.06, 0.20)\\n10.82 (7.59, 13.65)\\n8.75\\u2009±\\u20092.22\\n0.746\\n0.317\\n0.215\\n0.592\\n0.115\\nOpen in a new tab\\nNote:\\na\\nThe data are presented as means\\u2009±\\u2009standard deviations. Independent t test was used to compare between groups.\\nb\\nThe data are presented as medians (interquartile ranges). Mann–Whitney U test was used to compare between groups).',\n",
       "  'Discussion': 'Achieving a high-quality endometrium is pivotal for successful embryo implantation, especially in in vitro fertilization treatment. Creating a receptive endometrium is crucial for FET cycles, with programmed preparation offering limited monitoring and flexible scheduling [22]. Prior studies indicate that the impact of endometrial thickness on endometrial receptivity is controversial. However, in our practice, we believe that pregnancy rates are highest when endometrial thickness ranges from 8–14 mm in FET cycles [23, 24].\\nThis double-blinded, randomized clinical trial assessed the effects of estradiol dosage on artificial endometrial preparation. The results indicated that a standard dose of estradiol (6 mg/day), initiated on day 2 of the menstrual cycle, led to an average endometrial thickness of 8 mm. In contrast, a lower dose (4 mg/day) resulted in slightly less thickening. However, no significant difference in endometrial thickness was observed between the two dosage groups. Histological assessment using the Noyes criteria revealed a similar transformation of the endometrium into the secretory phase across both groups, with adequate secretory-phase characteristics on days 18 and 19 of the cycle. These findings imply that both estradiol regimens could be effective for endometrial secretory transformation and initiating artificial endometrial preparation in FET cycles, aligning with the findings of Lewin et al [19].\\nWhile assessing mean endometrial thickness serves as a practical morphological marker for endometrial receptivity, it is essential to recognize the influence of various molecular and protein changes on implantation. Lessey et al [25] previously identified a range of specific proteins and biochemical markers that are indicative of endometrial receptivity during the implantation window. The purpose of the present investigation was to assess the impact of estrogen dosage on the expression of the regulatory proteins HOXA-10 and HOXA-11 and the integrin αvβ3 in endometrial preparations for frozen-thawed embryo transfer. Our findings revealed that, despite the absence of a statistically significant difference in mean endometrial thickness between the standard- and low-dose groups, the expression levels of the regulatory proteins HOXA-10 and HOXA-11 and integrin αvβ3 were notably greater in the standard-dose group. Given the established importance of these biochemical markers in endometrial receptivity, our study suggests that employing a standard dose of estradiol (6 mg/day) in the artificial preparation of the endometrium may enhance its receptivity, potentially improving the outcomes of frozen-thawed embryo transfers.\\nIn this investigation, the cellular localization patterns of HOXA10 and HOXA11 were shown to parallel those observed in prior research [17, 18, 26]. Indeed, IHC study showed HOXA10 and HOXA11 located abundantly in endometrial grand in low doses but remarkably presented in both the endometrial gland and stomal tissue in standard doses. These proteins are essential for embryo implantation during the receptive phase of the endometrium (the implantation window). These findings underscore the critical role of these genes in the implantation process. Interestingly, the location of HOXA10 and HOXA11 proteins showed in the cytoplasm of glandular epithelial but in the nucleus in stromal cells. The function of these transcription factors in the cytoplasm remained unclear but might be related to the post-transcription process, such as the regulation of protein synthesis. Indeed, the function of HOXA10 and HOXA11 in the endometrial is crucially related to the implantation of embryo. Studies have shown that targeted disruption of HOXA-10 is linked to implantation failure in murine models [27]. In addition, the expression pattern of the integrin αVβ3, recognized as one of the pivotal endometrial receptivity biomarkers associated with infertility [16], was consistent with that documented in earlier research [28]. The localization of the integrin αVβ3, particularly at the apex of luminal and glandular cell surfaces during the window of implantation, has been directly correlated with successful implantation. These findings further emphasize the integral role of the integrin αVβ3 in establishing a receptive endometrial environment conducive to embryo implantation. However, variations in gene expression have been observed during several reproductive anomalies, including endometriosis, polycystic ovarian syndrome (PCOS), leiomyoma, polyps, adenomyosis, and hydrosalpinx [29]\\nHormonally, both low and standard estadiol dosages achieved adequate estradiol levels, with no significant difference between groups. The reason why the endometrial thickness and markers of endometrial receptivity are higher in the higher dose group and yet the serum levels are similar is that serum estradiol levels may fluctuate after oral administration. Intriguingly, the serum estradiol concentration decreased following progesterone administration in both groups. This phenomenon may be attributed to progesterone’s role in downregulating estrogen receptor concentrations and promoting the metabolic conversion of estradiol to weaker metabolites, thereby influencing the hormonal milieu favorable for implantation [30].\\nArtificial endometrial preparation, commonly utilized by in vitro fertilization centers, is favored because of its simplicity and manipulability. The mode of progesterone administration for secretory transformation of the endometrium includes oral, intramuscular, and vaginal routes. Although conclusive evidence on the optimal form of progesterone supplementation is lacking [31], the selection between these forms does not significantly influence clinical outcomes such as pregnancy and live birth rates [32]. Patient preferences and convenience are key factors in choosing progesterone preparations. In this study, vaginal micronized progesterone was used, which, although associated with lower systemic serum levels than intramuscular injections, is known for its direct, local absorption, resulting in standarder tissue concentrations in the gynecological tract and endometrium [33]. Baseline serum progesterone levels were established during the first visit, acknowledging that elevated levels in the late follicular phase can markedly alter gene expression in the endometrium [34]. Subsequent serum measurements confirmed progesterone absorption, but no significant intergroup differences were observed.\\nThis study is the first double-blinded, randomized clinical trial to evaluate the impact of different estrogen doses on endometrial receptivity markers during FET cycle preparation. However, the comparison of only two fixed doses of estrogen points to the need for more comprehensive research. Future studies should explore various aspects of endometrial preparation protocols to potentially improve implantation and pregnancy outcomes.',\n",
       "  'Conclusion': 'Identifying the optimal approach for artificial endometrial preparation and the receptive window is crucial for enhancing implantation success in FET cycles. The findings of this study suggest that a fixed daily oral dose of 6 mg of estradiol was associated with improved markers of endometrial receptivity as compared with a fixed daily oral dose of 4 mg of estradiol.',\n",
       "  'URL': 'https://doi.org/10.1371/journal.pone.0328871',\n",
       "  'Journal': 'PLoS One'},\n",
       " {'PaperTitle': 'SARS-CoV-2 infection risk assessment in the endometrium: viral infection-related gene expression across the menstrual cycle',\n",
       "  'Publication': '2020',\n",
       "  'Authors': 'Ismael Henarejos-Castillo, Patricia Sebastian-Leon, Almudena Devesa-Peiro, Antonio Pellicer, Patricia Diaz-Gimeno',\n",
       "  'Email': '',\n",
       "  'Abstract': 'Objective\\nTo determine the susceptibility of the endometrium to infection by—and thereby potential damage from—SARS-CoV-2.\\nDesign\\nAnalysis of SARS-Cov-2 infection-related gene expression from endometrial transcriptomic data sets.\\nSetting\\nInfertility research department affiliated with a public hospital.\\nPatient(s)\\nGene expression data from five studies in 112 patients with normal endometrium collected throughout the menstrual cycle.\\nIntervention(s)\\nNone.\\nMain Outcome Measure(s)\\nGene expression and correlation between viral infectivity genes and age throughout the menstrual cycle.\\nResult(s)\\nGene expression was high for TMPRSS4, CTSL, CTSB, FURIN, MX1, and BSG; medium for TMPRSS2; and low for ACE2. ACE2, TMPRSS4, CTSB, CTSL, and MX1 expression increased toward the window of implantation. TMPRSS4 expression was positively correlated with ACE2, CTSB, CTSL, MX1, and FURIN during several cycle phases; TMPRSS2 was not statistically significantly altered across the cycle. ACE2, TMPRSS4, CTSB, CTSL, BSG, and MX1 expression increased with age, especially in early phases of the cycle.\\nConclusion(s)\\nEndometrial tissue is likely safe from SARS-CoV-2 cell entry based on ACE2 and TMPRSS2 expression, but susceptibility increases with age. Further, TMPRSS4, along with BSG-mediated viral entry into cells, could imply a susceptible environment for SARS-CoV-2 entry via different mechanisms. Additional studies are warranted to determine the true risk of endometrial infection by SARS-CoV-2 and implications for fertility treatments.',\n",
       "  'Introduction': 'The rapid international spread of novel coronavirus disease 2019 (COVID-19) (1) has resulted in 5,103,006 confirmed cases and 333,401deaths worldwide as of May 23, 2020 (2). COVID-19 is caused by SARS-coronavirus 2 (SARS-CoV-2) infection in the lower respiratory tract (3), but the mechanisms underlying infection remain poorly understood (4). The rapid spread of SARS-CoV-2, genetically closely related to severe acute respiratory syndrome coronavirus (SARS-CoV-1) (5), has resulted in a global health emergency with long-term consequences on economics markets, nutritional habits, and mental and physical well-being (5, 6, 7, 8, 9).\\nAs information is still emerging about the health consequences of COVID-19, assisted reproductive treatments (ARTs) have been delayed due to fear of the unknown impact of SARS-Cov-2 on fertility (10, 11). Fertility is compromised by age, and the longer treatments are delayed, the less likely that successful outcomes will be achieved (12). Furthermore, effects of SARS-CoV-2 infection increase in severity with host age (13, 14, 15), meaning that women of more advanced reproductive age undergoing ART could be at higher risk of infection. It is therefore critical to determine exactly how the virus affects reproductive physiology. Although the studies to date have been discordant, vertical transmission during pregnancy seems to occur infrequently, ranging from 0 to 11% (11, 16, 17, 18). To date there is no information about how infection affects early implantation and development.\\nThe ability of SARS-CoV-2 to damage tissue is determined by its capacity to enter and infect cells in that tissue (4). The SARS-CoV-2 entry point on the cell is angiotensin-converting enzyme 2 (ACE2) (19, 20), which plays a key role in the renin-angiotensin system, cleaving angiotensin II to angiotensin 1–7. The system is disrupted after SARS-CoV-2 gains cell entry and down-regulates the expression of ACE2, leading to up-regulation of the proinflammatory response by angiotensin II (21, 22, 23). ACE2 exhibits moderately increased expression with age (24), which could explain disease severity in older people. Another path of entry, using Basigin (BSG) as receptor instead of ACE2, has been proposed (25).\\nTo enter the cell, SARS-CoV-2 uses its spike S protein to bind to ACE2, which leads to fusion with the cell membrane and endocytosis (4, 20, 26). TMPRSS2, a transmembrane protease, cleaves the S protein (27). Cleavage is necessary for the virus to bind to ACE2 and spread through the infected host (19). Other proteases are under investigation as possible implications in SARS-CoV-2 infectivity related to S protein cleaving. TMPRSS4 increased virus infectivity on its own, at least in gut epithelial cells (28), while cathepsins B and L (CTSB and CTSL, respectively) had residual cleaving activity of viral S protein in TMPRSS2- cells (19). FURIN, another protease predicted to cleave S protein, presents alongside ACE2 in epithelial layers of several oral mucosal tissues (29, 30). MX dynamin-like GTPase 1 (MX1) regulates neutrophil infiltration, favoring infection through protein S modification by neutrophil elastase (31).\\nThe clinical presentation of COVID-19 ranges from mild respiratory symptoms to severe progressive pneumonia, gastrointestinal symptoms, fecal shedding, multiorgan failure, and even death (32, 33), but few studies have focused on the virus’s effect on fertility and damage to reproductive tissues, or on concerns regarding the use of reproductive treatments. Leydig and Sertoli cells in the testis (34, 35, 36, 37), oocytes (38), and ovarian tissue (37) are likely to experience damage due to their medium-high expression of the ACE2 receptor. The endometrium is crucial for human reproduction and embryo implantation, but studies of the effect of SARS-CoV-2 infection on menstrual cycle progression have not been performed. Delineating the virus’s impact on the tissue is important for determining risk to ART, given that a healthy endometrium is needed for embryo implantation and growth.\\nThe endometrium is a complex tissue subjected to a cycle of cell death and renewal approximately every 28 days (39). Numerous transcriptomic studies have sought to understand gene expression changes throughout the menstrual cycle (40), and most of these data sets are available in public repositories, such as the Gene Expression Omnibus (GEO) database (41). According to the Human Protein Atlas (HPA) (42), ACE transcript is in low abundance in endometrium and not present as protein. According to HPA expression levels, TMPRSS4 and FURIN RNA expression is low, and protein levels are medium, whereas CTSB, MX1, and BSG have medium RNA expression and high protein expression (43). However, there is little information on how the virus could affect endometrial receptivity and embryo implantation.\\nWe analyzed the impact of SARS-CoV-2 infection on the endometrium by measuring endometrial ACE2, TMPRSS2, TMPRSS4, CTSB, CTSL, FURIN, MX1, and BSG gene expression. Transcriptomic data sets available across the phases of endometrial progression were used to evaluate molecularly the risk of SARS-CoV-2 infection during the COVID-19 pandemic.',\n",
       "  'Methods': 'Search and selection of SARS-CoV-2 infectivity-related proteins\\nA thorough literature search identified proteins related to the SARS-CoV-2 disease-causing mechanism, including genes related to cell entry and genes with implications in health and fertility, among others. The search for relevant genes associated with viral infectivity was made chronologically up to May 10, 2020. The keywords searched in PubMed (44) included all possible combinations between “SARS-CoV-2,” “COVID-19,” “coronavirus,” “cell entry mechanisms,” “long-term implications,” and “fertility.”\\nSearch and selection of endometrial transcriptomic data sets\\nPublic transcriptomic data sets were used to analyze expression of SARS-CoV-2 infectivity-related genes throughout the menstrual cycle. Endometrial transcriptomic experiments for control patients (without any known endometrial pathology) were systematically searched in the GEO database (41) with no restrictions on publication date or language and according to preferred reporting items for systematic reviews and meta-analyses (PRISMA) guidelines (45). Keywords were: uter∗ OR endometr∗, filtered by “homo sapiens.” Experiments were selected if\\n•\\nRNA was extracted directly from human endometrial biopsies.\\n•\\nEndometrial biopsies were collected at different times during the menstrual cycle.\\n•\\nCycle phase at the time of biopsy was available for all samples.\\n•\\nEndometrial gene expression was evaluated by microarray or RNA sequencing using Affymetrix, Illumina, or Agilent platforms.\\n•\\nRaw gene expression data were freely available to download from GEO.\\nPreprocessing and integrative analysis\\nTranscriptomic raw data were downloaded from the GEO database and processed according to the standards of the technology used (microarrays or RNA-seq) using limma R-package (version 3.34.9) (46). Expression data from each experiment were annotated with biomaRt (version 2.30.0) (47), log transformed, and quantile normalized using limma (46). Principal components analysis (PCA) was done using the prcomp() function, and the scores were displayed using the ggplot2 R-package (48) to look for possible batch effects and were corrected if using linear models. Outliers were deleted from posterior analysis if there was a distinct transcriptional behavior according to PCA.\\nTo integrate the included endometrial transcriptomic experiments, several steps were followed as recommended by Tajti et al. (49): after being independently normalized, selected studies were joined in a unique data set, and batch effects were corrected using linear models (limma R package) (43). A relative expression value of low, medium, and high expression were established. The thresholds respectively correspond to 1% to 10%, 11% to 50%, and 51% to 100% categories of gene expression values of the entire integrated data set.\\nPairwise differential expression analysis between experiments was performed using limma to identify genes with expression differences. P values were corrected using false discovery rates (FDR) (50), and genes differentially expressed between experiments were removed from the analysis (FDR <.05).\\nDifferential gene expression throughout the menstrual cycle\\nAn analysis of variance (ANOVA) was performed for each selected gene related to SARS-CoV-2 infectivity to assess which genes showed statistically significant differences between endometrial phases. Analysis of variance was followed by a pairwise t-test to determine statistically significant differences between phases. Fold changes in gene expression from phase to phase were also calculated. The mean expression and confidence intervals for each gene in each phase were represented using ggplot2 R-package (version 3.0.0) (45).\\nCoexpression of infectivity genes and the effect of age\\nPearson’s correlation (51) was used to assess coexpression between each pair of selected proteins related to SARS-CoV-2 infectivity and to analyze the effect of age (from 23 to 50 years) on the expression of each protein. All programming and statistical tests (t-test, Pearson’s correlation, and analysis of variance) were implemented in the R environment (version 3.4.4, 2018-03-15) (52).',\n",
       "  'Results': 'Menstrual cycle clock in integrated endometrial data sets\\nFrom a total of 694 studies retrieved from GEO, our search identified five unique studies that evaluated endometrial gene expression in women with normal endometrium, comprising 112 samples (Table 1; see Supplemental Fig. 1, available online, for detailed filtering steps). All studies were analyzed separately for correction of possible batch effects, and three samples were excluded from analysis (details in Supplemental Fig. 2, available online). A unique combined data set with a population of 109 patients was obtained, comprising 29 samples in the proliferative phase, 29 in the early secretory phase, 43 in the medium secretory phase, and eight in the late secretory phase (Table 1).\\nThe 109 samples were grouped depending on the experiment rather than by the menstrual cycle phase (Fig. 1A, left). To make data sets comparable after the integration of all samples, this batch effect was removed (Fig. 1A, right). The resulting gene expression samples showed a behavior based on menstrual cycle phases rather than study nature (Fig. 1B).\\nViral infectivity genes show a different expression landscape across the menstrual cycle\\nThe gene expression landscapes for each viral gene across the cycle are shown in Figure 2B (56). TMPRSS4, CTSL, CTSB, FURIN, MX1, and BSG were highly expressed throughout the cycle; TMPRSS2 was moderately expressed, and ACE2 was low (Fig. 2A). Gene expression of viral proteins depended on menstrual cycle phase. Calculated P values and adjusted P values are presented in Supplemental Fig. 3A (available online). All genes except TMPRSS2 (P=.053) had statistically significant changes in expression across the menstrual cycle (Fig. 2A). Specifically, the genes most affected by menstrual cycle progression (P<.0001) were ACE2, which increased expression from early secretory to midsecretory; TMPRSS4, whose expression increased from proliferative to midsecretory and from early secretory to midsecretory (P<.0001); CTSL and CTSB, which increased from early secretory to midsecretory (P<.0001); TMPRSS2 with decreased expression from proliferative to early secretory (P<.05); and BSG and FURIN, which increased from proliferative to midsecretory (P<.01) (Fig. 2A). All genes, including TMPRSS2, showed increased expression from early secretory to midsecretory, as indicated in Supplemental Figure 3B. Detailed expression changes across the menstrual cycle are provided for each gene in Supplemental Figure 4 (available online).\\nCorrelations between genes showed activations and repressions between them throughout the menstrual cycle (Supplemental Table 1, available online). ACE2 and TMPRSS4 were positively correlated in the early secretory phase (0.58). A weak correlation (activation) was found for ACE2 with TMPRSS4 and CTSL in the window of implantation (0.17 and 0.25, respectively). Coactivations were also detected in the proliferative phase between CTSB and TMPRSS4; the early secretory phase between FURIN and BSG; the midsecretory phase between CTSB with CTSL, and MX1 with CTSB, CTSL, and TMPRSS4; and the late secretory phase between TMPRSS4 and CTSB, FURIN, CTSB, and CTSL. Based on our results, high coactivation values were prioritized to build a molecular scheme of SARS-CoV-2 plausible infection of the endometrium based on two mechanisms of entry: one via ACE2 receptor and the other via BSG receptor. This scheme is shown in Figure 2B.\\nIn this model, TMPRSS4, which is highly expressed in all phases of the menstrual cycle, especially during the window of implantation, is the protease responsible for protein S cleaving, giving SARS-CoV-2 the capacity to bind to ACE2 and infect cells even if ACE2 is lowly expressed. The model adds CTSL, CTSB, and FURIN as novel actors in the SARS-CoV-2 mechanism of cell entry. These proteases are highly expressed and coactivate with TMPRSS4. They could also help with the cleaving of protein S in different sites, thereby increasing infectivity. We also included MX1, which regulates neutrophil activity and is highly expressed, bringing neutrophils into the tissue with elastases to help cleave S protein. Finally, BSG, the alternative receptor for SARS-CoV-2, also showed high expression, which would favor infectivity in this model, and activation with FURIN, which would favor infectivity in this model.\\nAge influences the expression of viral genes throughout the menstrual cycle\\nFor this analysis, only samples from the study Talbi et al. (53) were used because it was the only study that provided the age for each sample (in a range of 23–50 years old; n = 27). ACE2 showed increased expression with age in the proliferative, early secretory, and midsecretory phases (0.4, 0.73, 0.44, respectively; Fig. 3A). Most correlations by age were stronger in early phases of the menstrual cycle (proliferative and early secretory) than in subsequent endometrial stages (Fig. 3B). TMPRSS4 and MX1 were highly correlated with age in the early secretory phase; TMPRSS4, CTSL, and CTSB had positive correlation in the proliferative phase; and TMPRSS4 and BSG had positive correlation in the late secretory phase. In addition, high negative correlations were detected for BSG in the early secretory phase and for FURIN in the midsecretory phase.',\n",
       "  'Discussion': 'It is important to understand the consequences of SARS-CoV-2 infection as the virus continues to spread worldwide. Fertility, specifically embryo implantation, is impacted by changes in endometrial gene expression throughout the menstrual cycle. Between menstrual cycle phases, gene expression varies to accompany physiologic changes (57). The differences in gene expression from one phase to another create a changing landscape for viral infection, and viral proteins themselves alter their expression throughout the menstrual cycle, setting the stage for a cyclical landscape of viral infectivity risk.\\nACE2, the receptor far for SARS-CoV-2 cell entry, showed low expression in the endometrium in our study, as reported previously in the Human Protein Atlas (HPA) (42). ACE2 is also reduced in critical tissues such as lungs (24); however, we cannot suggest that low levels of ACE2 expression imply no effect of the virus on the tissue. Further studies are needed to elucidate the effect of decreased ACE2 in the endometrium. It is interesting that ACE2 increased (fold change = 2.47) from the early secretory to midsecretory phases, implying an increase in ACE2 in the window of implantation and a high risk of viral infectivity at this stage of the menstrual cycle.\\nSeveral possible mechanisms of SARS-CoV-2 cell entry have been proposed (19, 28). TMPRSS2, the most-reported protease involved with SARS-CoV-2 infectivity alongside ACE2, had medium endometrial expression in our study. However, there was no correlation between TMPRSS2, ACE2, and the rest of the genes studied. These results imply that the endometrium should be safe against SARS-CoV-2 infectivity mediated by TMPRSS2, though the expression of other proteases associated with S protein cleavage show a different landscape than TMPRSS2.\\nTMPRSS4 statistically significantly changes its expression throughout the menstrual cycle, showing a statistically significant increase in the midsecretory phase. This protein also showed an interesting correlation with other genes, including an increase alongside ACE2 in the early secretory phase. Although TMPRSS2 is implicated in cell entry, the fact that S proteins could be targeted by TMPRSS4 and that TMPRSS4 increased infectivity in gut epithelial cells (26, 28) could mean a vulnerability of the endometrium to infection by SARS-CoV-2 mediated by TMPRSS4. Furthermore, TMPRSS4 up-regulation was correlated with CTSL and CTSB in all phases except the early secretory phase. Cell infection by SARS-CoV-2 was observed in TMRPSS2- cell lines expressing both CTSL and CTSB, so these proteins may have a residual function of cleaving viral S protein (19).\\nWe propose that a synergy of CTSL and CTSB with TMPRSS4 through most of the cycle favors SARS-CoV-2 infectivity in the endometrium. FURIN, another protease predicted to cleave protein S (37, 58, 59), also showed a positive correlation with TMPRSS4 in the late secretory phase; with both targeting S protein, higher infectivity may occur. Finally, TMPRSS4 was positively correlated with MX1 in the midsecretory phase. MX1 actively participates in SARS-CoV-2 infection by attracting neutrophils to infected tissue and employing their elastases to cleave protein S along with TMPRSS2 (31); if TMPRSS4 participates alongside MX1, a more severe infection could occur during the window of implantation.\\nTMPRSS4, CTSB, CTSL, FURIN, and MX1 showed statistically significant expression changes throughout the cycle. Protein expression of TMPRSS4, CTSB, FURIN, and MX1 (42) supports a susceptible environment for infection in the early secretory and midsecretory phases. Future studies are needed to fully understand the mechanism of SARS-CoV-2 cell entry and the roles of TMPRSS4, CTSB, CTSL, FURIN, and MX1 in the endometrium.\\nBSG, the alternative receptor for SARS-CoV-2 entry (25), showed high expression with statistically significant changes between phases and strong activation with FURIN through most of the cycle; however, its repression was correlated with ACE2 and TMPRSS2. Our results may indicate that only FURIN could be involved in S viral protein cleavage, if necessary, for SARS-CoV-2 binding to BSG, though other unknown proteases could also be involved. Further studies are needed to provide insight on BSG function in SARS-CoV-2 infection in the endometrium.\\nWe also investigated the effect of age on viral gene expression throughout the menstrual cycle. In a recent study in lungs, ACE2 expression was detected as highly variable among individuals ranging from 0.17 transcripts per million to 14.5 transcripts per million in the same demographic group and highlighted a skewness of expression that statistically significantly increased with age (24). Our results showed a positive correlation between age and ACE2 from the proliferative to midsecretory phases, especially in the early secretory phase, meaning that the endometrium in older women (to age 50) could be more susceptible to SARS-CoV-2 infection. TMPRSS2 decreased with age, particularly in the midsecretory and late secretory phases, which may mean that cell-mediated entry of the virus is lowered with increased age. However, TMPRSS4 increased in expression from the proliferative to early secretory phases similar to CTSB, CTSL, and MX1. These results, in conjunction with ACE2 expression changes, highlighted the same behavior related to age as in the lungs (24), implying a higher risk of endometrial infection in older women, especially in the early phases of the menstrual cycle, as well as potential implantation problems. However, our results are limited to only one study with 27 patients, and there are no prospective studies showing the real clinical implications.\\nACE2 plays an important role in endometrial changes because it cleaves angiotensin II and impacts spiral artery vasoconstriction, posterior proliferation, and renewal of tissue (60, 61, 62). ACE2 down-regulation by SARS-CoV-2 could alter the endometrial balance (63); however, given that the landscape of the tissue is modified with each menstrual cycle renewal, the consequences of SARS-CoV-2 infection could be reset from cycle to cycle.\\nBecause our study integrates data sets from GEO, this approach is limited by the study design. This also means that results depend on the experimental transcriptomic procedures of prior reported results and the sample cohort included in each study. However, the inclusion criteria for the selected studies were very careful, and uniform and raw data were preprocessed to control undesired effects. Likewise, the virus differentially affects individuals due to each person’s genetic profile (31, 64). Further prospective research is needed to determine the mechanisms of SARS-CoV-2 infection in the endometrium and how it could affect the fertility of each patient.',\n",
       "  'Conclusion': 'Although TMPRSS2 expression implies a safe environment against infection, the roles of proteases and neutrophil regulating proteins TMPRSS4, CTSB, CTSL, FURIN, and MX1 could imply susceptibility of infection, especially during the early secretory and midsecretory phases. Additionally, the high expression of BSG could imply an alternative mechanism independent of ACE2 for endometrial viral infection. Our findings show that viral gene expression increases with age, suggesting that the endometrium of older women undergoing ART is at higher risk of viral infection. A careful approach is advised, with precautions implemented when resuming ART.',\n",
       "  'URL': 'https://doi.org/10.1016/j.fertnstert.2020.06.026',\n",
       "  'Journal': 'Fertil Steril'},\n",
       " {'PaperTitle': 'Comprehensive mapping of long range interactions reveals folding principles of the human genome',\n",
       "  'Publication': '2009',\n",
       "  'Authors': 'Erez Lieberman-Aiden, Nynke L van Berkum, Louise Williams, Maxim Imakaev, Tobias Ragoczy, Agnes Telling, Ido Amit, Bryan R Lajoie, Peter J Sabo, Michael O Dorschner, Richard Sandstrom, Bradley Bernstein, M A Bender, Mark Groudine, Andreas Gnirke, John Stamatoyannopoulos, Leonid A Mirny, Eric S Lander, Job Dekker',\n",
       "  'Email': 'lander@broadinstitute.org; job.dekker@umassmed.edu',\n",
       "  'Abstract': 'We describe Hi-C, a method that probes the three-dimensional architecture of whole genomes by coupling proximity-based ligation with massively parallel sequencing. We constructed spatial proximity maps of the human genome with Hi-C at a resolution of 1Mb. These maps confirm the presence of chromosome territories and the spatial proximity of small, gene rich chromosomes. We identified an additional level of genome organization that is characterized by the spatial segregation of open and closed chromatin to form two genome-wide compartments. At the megabase scale, the chromatin conformation is consistent with a fractal globule, a knot-free conformation that enables maximally dense packing while preserving the ability to easily fold and unfold any genomic locus. The fractal globule is distinct from the more commonly used globular equilibrium model. Our results demonstrate the power of Hi-C to map the dynamic conformations of whole genomes.',\n",
       "  'Introduction': 'The three-dimensional conformation of chromosomes is involved in compartmentalizing the nucleus and bringing widely separated functional elements into close spatial proximity (1-5). Understanding how chromosomes fold can provide insight into the complex relationships between chromatin structure, gene activity, and the functional state of the cell. Yet beyond the scale of nucleosomes, little is known about chromatin organization. Long-range interactions between specific pairs of loci can be evaluated with Chromosome Conformation Capture (3C), using spatially constrained ligation followed by locus-specific PCR (6). Adaptations of 3C have extended the process with the use of inverse PCR (4C) (7, 8) or multiplexed ligation-mediated amplification (5C) (9). Still, these techniques require choosing a set of target loci and do not allow unbiased genome-wide analysis.',\n",
       "  'Methods': 'Here we report a method named Hi-C that adapts the above approach to enable purification of ligation products followed by massively parallel sequencing. Hi-C allows unbiased identification of chromatin interactions across an entire genome. Briefly: cells are crosslinked with formaldehyde; DNA is digested with a restriction enzyme that leaves a 5′-overhang; the 5′-overhang is filled, including a biotinylated residue; and the resulting blunt-end fragments are ligated under dilute conditions that favor ligation events between the cross-linked DNA fragments. The resulting DNA sample contains ligation products consisting of fragments that were originally in close spatial proximity in the nucleus, marked with biotin at the junction. A Hi-C library is created by shearing the DNA and selecting the biotin-containing fragments with streptavidin beads. The library is then analyzed using massively parallel DNA sequencing, producing a catalog of interacting fragments (Fig. 1A, SOM).',\n",
       "  'Results': 'We created a Hi-C library from a karyotypically normal human lymphoblastoid cell line (GM06990) and sequenced it on two lanes of an Illumina Genome Analyzer, generating 8.4 million read pairs that could be uniquely aligned to the human genome reference sequence; of these, 6.7 million corresponded to long-range contacts between segments greater than >20 Kb apart.\\nWe constructed a genome-wide contact matrix M by dividing the genome into 1 Mb regions (‘loci’) and defining the matrix entry mij to be the number of ligation products between locus i and locus j (SOM). This matrix reflects an ensemble average of the interactions present in the original sample of cells; it can be visually represented as a heatmap, with intensity indicating contact frequency (Fig. 1B).\\nWe tested whether Hi-C results were reproducible by repeating the experiment using the same restriction enzyme (HindIII) and using a different one (NcoI). We observed that contact matrices for these new libraries (Fig 1C, D) were extremely similar to the original contact matrix (Pearson’s r=0.990 [HindIII] and r=0.814 [NcoI]; p was negligible [<10−300] in both cases). We therefore combined the three datasets in subsequent analyses.\\nWe first tested whether our data are consistent with known features of genome organization (1) – specifically, chromosome territories (the tendency of distant loci on the same chromosome to be near one another in space) and patterns in sub-nuclear positioning (the tendency of certain chromosome pairs to be near one another).\\nWe calculated the average intrachromosomal contact probability, In(s), for pairs of loci separated by a genomic distance s (distance in base pairs along the nucleotide sequence) on chromosome n. In(s) decreases monotonically on every chromosome, suggesting polymer-like behavior in which the three-dimensional distance between loci increases with increasing genomic distance; these findings are in agreement with 3C and fluorescence in situ hybridization (FISH) (6, 10). Even at distances greater than 200 Mb, In(s) is always much greater than the average contact probability between different chromosomes (Fig. 2A). This implies the existence of chromosome territories.\\nInterchromosomal contact probabilities between pairs of chromosomes (Fig. 2B) show that small, gene-rich chromosomes (chromosomes 16, 17, 19, 20, 21, 22) preferentially interact with each other. This is consistent with FISH studies showing that these chromosomes frequently co-localize in the center of the nucleus (11, 12). Interestingly, chromosome 18, which is small but gene-poor, does not interact frequently with the other small chromosomes; this agrees with FISH studies showing that chromosome 18 tends to be located near the nuclear periphery (13).\\nWe then zoomed in on individual chromosomes to explore whether there are chromosomal regions that preferentially associate with each other. Because sequence proximity strongly influences contact probability, we defined a normalized contact matrix M* by dividing each entry in the contact matrix by the genome-wide average contact probability for loci at that genomic distance (SOM). The normalized matrix shows many large blocks of enriched and depleted interactions generating a ‘plaid’ pattern (Fig. 3B). If two loci (here 1 Mb regions) are nearby in space, we reasoned that they will share neighbors and have correlated interaction profiles. We therefore defined a correlation matrix C in which cij is the Pearson correlation between the ith row and jth column of M*. This process dramatically sharpened the plaid pattern (Fig. 3C); 71% of the resulting matrix entries represent statistically significant correlations (p ≤ 0.05).\\nThe plaid pattern suggests that each chromosome can be decomposed into two sets of loci (arbitrarily labeled A and B) such that contacts within each set are enriched and contacts between sets are depleted. We partitioned each chromosome in this way using principal component analysis. For all but two chromosomes, the first principal component (PC) clearly corresponded to the plaid pattern (positive values defining one set, negative values the other) (Fig. S1). For chromosomes 4 and 5, the first PC corresponded to the two chromosome arms, but the second PC corresponded to the plaid pattern. The entries of the PC vector reflected the sharp transitions from compartment to compartment observed within the plaid heatmaps. Moreover, the plaid patterns within each chromosome were consistent across chromosomes: the labels (A and B) could be assigned on each chromosome so that sets on different chromosomes carrying the same label had correlated contact profiles, and those carrying different labels had anticorrelated contact profiles (Fig. 3D). These results imply that the entire genome can be partitioned into two spatial compartments such that greater interaction occurs within each compartment rather than across compartments.\\nThe Hi-C data imply that regions tend be closer in space if they belong to the same compartment (A vs. B) than if they do not. We tested this using 3D-FISH, probing four loci (L1, L2, L3, and L4) on chromosome 14 that alternate between the two compartments (L1 and L3 in compartment A; L2 and L4 in compartment B) (Fig. 3E, F). 3D-FISH showed that L3 tends to be closer to L1 than to L2, despite the fact that L2 lies between L1 and L3 in the linear genome sequence (Fig. 3E). Similarly, we found that L2 is closer to L4 than to L3 (Fig. 3F). Comparable results were obtained for four consecutive loci on chromosome 22 (Fig. S2A, B). Taken together, these observations confirm the spatial compartmentalization of the genome inferred from Hi-C. More generally, a strong correlation was observed between the number of Hi-C reads mij and the three-dimensional distance between locus i and locus j as measured by FISH (Spearman’s rho=0.874, p=0.0002 [Fig. S3]), suggesting that Hi-C read count may serve as a proxy for distance.\\nUpon close examination of the Hi-C data, we noted that pairs of loci in compartment B showed a consistently higher interaction frequency at a given genomic distance than pairs of loci in compartment A (Fig. S4). This suggests that compartment B is more densely packed (14). The FISH data are consistent with this observation; loci in compartment B exhibited a stronger tendency for close spatial localization.\\nTo explore whether the two spatial compartments correspond to known features of the genome, we compared the compartments identified in our 1 Mb correlation maps to known genetic and epigenetic features. Compartment A correlates strongly with the presence of genes (Spearman’s rho=0.431, p<10−137), higher expression (via genome-wide mRNA expression, Spearman’s rho=0.476, p<10−145 [Fig. S5]), and accessible chromatin (as measured by DNAseI sensitivity, Spearman’s rho=0.651, p negligible) (15, 16). Compartment A also shows enrichment for both activating (H3K36 trimethylation, Spearman’s rho=0.601, p<10−296) and repressive (H3K27 trimethylation, Spearman’s rho=0.282, p<10−56) chromatin marks (17). We repeated the above analysis at a resolution of 100 kb (Fig. 3G) and saw that while the correlation of compartment A with all other genomic and epigenetic features remained strong (Spearman’s rho>0.4, p negligible), the correlation with the sole repressive mark, H3K27 trimethylation, was dramatically attenuated (Spearman’s rho=0.046, p<10−15). On the basis of these results we concluded that compartment A is more closely associated with open, accessible, actively transcribed chromatin.\\nWe repeated our experiment with K562 cells, an erythroleukemia cell line with an aberrant karyotype (18). We again observed two compartments; these were similar in composition to those observed in GM06990 cells (Pearson’s r=0.732, p negligible [Fig. S6]) and showed strong correlation with open and closed chromatin states as indicated by DNAseI sensitivity (Spearman’s rho=0.455, p<10−154).\\nThe compartment patterns in K562 and GM are similar, but there are many loci in the open compartment in one cell type and the closed compartment in the other (Fig. 3H). Examining these discordant loci on karyotypically normal chromosomes in K562 (18), we observed a strong correlation between the compartment pattern in a cell type and chromatin accessibility in that same cell type (GM06990, Spearman’s rho=0.384, p=0.012; K562, Spearman’s rho=0.366, p=0.017). Thus, even in a highly rearranged genome, spatial compartmentalization correlates strongly with chromatin state.\\nOur results demonstrate that open and closed chromatin domains throughout the genome occupy different spatial compartments in the nucleus. These findings expand upon studies of individual loci that have observed particular instances of such interactions; both between distantly located active genes, and between distantly located inactive genes (8, 19-23).\\nFinally, we sought to explore the internal structure of the open and closed chromatin domains that correspond to the compartments seen in the plaid correlation maps. We closely examined the average behavior of intrachromosomal contact probability as a function of genomic distance, calculating the genome-wide distribution I(s). When plotted on log log axes, I(s) exhibits a prominent power law scaling between ~500 kb and ~7 Mb, where contact probability scales as s−1 (Fig. 4A). This range corresponds to the known size of open and closed chromatin domains.\\nPower-law dependencies can arise from polymer-like behavior (24). Various authors have proposed that chromosomal regions can be modeled as an ‘equilibrium globule’ – a compact, densely knotted configuration originally used to describe a polymer in a poor solvent at equilibrium (25, 26). (Historically, this specific model has often been referred to simply as a ‘globule’; some authors have used the term ‘equilibrium globule’ to distinguish it from other globular states [See below].) Grosberg et al. proposed an alternative model, theorizing that polymers, including interphase DNA, can self-organize into a long-lived, non-equilibrium conformation that they described as a ‘fractal globule’ (27, 28). This highly compact state is formed by an unentangled polymer when it crumples into a series of small globules in a ‘beads-on-a-string’ configuration. These beads serve as monomers in subsequent rounds of spontaneous crumpling until only a single globule-of-globules-of-globules remains. The resulting structure resembles a Peano curve, a continuous fractal trajectory that densely fills three-dimensional space without crossing itself (29). Fractal globules are an attractive structure for chromatin segments because they lack knots (30) and would facilitate unfolding and refolding, e.g. during gene activation, gene repression, or the cell cycle. In a fractal globule, contiguous regions of the genome tend to form spatial sectors whose size corresponds to the length of the original region (Fig. 4C). In contrast, an equilibrium globule is highly knotted and lacks such sectors; instead, linear and spatial positions are largely decorrelated after at most a few megabases (Fig. 4C). The fractal globule has not previously been observed (28).\\nThe ‘equilibrium globule’ and ‘fractal globule’ models make very different predictions concerning the scaling of contact probability with genomic distance s. The equilibrium globule model predicts that contact probability will scale as s−3/2, which we do not observe in our data. We analytically derived the contact probability for a fractal globule and found that it decays as s−1 (SOM); this corresponds closely with the prominent scaling we observed (−1.08).\\nThe equilibrium and fractal globule models also make differing predictions about the three-dimensional distance between pairs of loci (s1/2 for an equilibrium globule, s1/3 for a fractal globule). While three-dimensional distance is not directly measured by Hi-C, we note that a recent paper using 3D-FISH reported an s1/3 scaling for genomic distances between 500kb and 2Mb (26).\\nWe used Monte Carlo simulations to construct ensembles of fractal globules and equilibrium globules (500 each). The properties of the ensembles matched the theoretically-derived scalings for contact probability (fractal: s−1, equilibrium: s−3/2) and three dimensional distance (fractal: s1/3, equilibrium: s1/2). These simulations also illustrated the lack of entanglements [measured using the knot-theoretic Alexander polynomial (31)] and the formation of spatial sectors within a fractal globule (Fig. 4B).',\n",
       "  'Discussion': '',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1126/science.1181369',\n",
       "  'Journal': 'Science'},\n",
       " {'PaperTitle': 'Non-Coding RNAs in Endometrial Physiopathology',\n",
       "  'Publication': '2018',\n",
       "  'Authors': 'Alessandro La Ferlita, Rosalia Battaglia, Francesca Andronico, Salvatore Caruso, Antonio Cianci, Michele Purrello, Cinzia Di Pietro',\n",
       "  'Email': 'dipietro@unict.it',\n",
       "  'Abstract': 'The Human Genome Project led to the discovery that about 80% of our DNA is transcribed in RNA molecules. Only 2% of the human genome is translated into proteins, the rest mostly produces molecules called non-coding RNAs, which are a heterogeneous class of RNAs involved in different steps of gene regulation. They have been classified, according to their length, into small non-coding RNAs and long non-coding RNAs, or to their function, into housekeeping non-coding RNAs and regulatory non-coding RNAs. Their involvement has been widely demonstrated in all cellular processes, as well as their dysregulation in human pathologies. In this review, we discuss the function of non-coding RNAs in endometrial physiology, analysing their involvement in embryo implantation. Moreover, we explore their role in endometrial pathologies such as endometrial cancer, endometriosis and chronic endometritis.',\n",
       "  'Introduction': '1. Introduction At the beginning of the Human Genome Project (HGP), in the late 1990s, researchers hypothesized that our genome comprised about 100,000 protein-coding genes [1]. Over the years, this number has continuously decreased. In 2001, the International Human Genome Sequencing Consortium (IHGSC) published the initial sequence of the human genome and proposed that the number of protein-coding genes was about 30,000 [2]. The end of the sequencing of the human genome in 2004 revealed that the number of genes encoding for proteins were only 20–25,000 [3]. The latest version of human GENCODE (available online: https://www.gencodegenes.org/) established that the number of genes encoding for protein was 19,901 that is the 34.09% of human genes (Figure 1a). In spite of the low number of gene encoding proteins the ENCODE project, based on High Throughput Sequencing technologies and advances in bioinformatics, has provided a detailed landscape of transcription in human cells. (Figure 1b). Figure 1. Open in a new tab Overview of functional elements of the human genome and of the associated transcriptome. Pie chart showing the percentage of human genes (a) and transcripts (b) based on the current Release (Release 28, GRCh38.p12) of GENCODE. Classification is referred to Gene/Transcript Biotypes in GENCODE & Ensembl and to VEGA descriptions (available online: http://vega.archive.ensembl.org/info/about/gene_and_transcript_types.html). The non-coding transcriptome includes both small and long non-coding RNAs; pseudogenes incorporate processed, unprocessed, transcribed, translated, polymorphic and unitary sequences; other includes IG/TCR and their pseudogenes, together with non-stop decay and non-sense mediated decay. The non-coding RNAs (ncRNAs) could be classified, according on their size, into small ncRNAs (less than 200 nucleotides in length) and in long ncRNAs [(lncRNAs) longer than 200 nucleotides]. Alternatively, a further classification based on their function, split the ncRNAs into housekeeping and regulatory: the housekeeping ncRNAs, including ribosomal RNA (rRNA), transfer RNA (tRNA), small nuclear RNA (snRNA) and small nucleolar RNA (snoRNA), are expressed in all cell types and carry out essential functions in the cells, while the regulatory ncRNAs, including several classes of small and long molecules, assist in the regulation of gene expression, controlling different points of the central dogma (Figure 2). Figure 2. Open in a new tab Schematic diagram illustrating the classification of ncRNAs according to their biological role and their length. The interest of scientific literature in regulatory ncRNAs is generated by the important roles that these molecules perform regulating cell proliferation, differentiation, migration, cell death and angiogenesis. Consequently, their altered expression is involved in different human pathologies. Another interesting characteristic for the potential implications for human health is that ncRNAs are present in all biological fluids associated with protein complexes or enclosed within extracellular vesicles (EVs) such as microvesicles or exosomes. Extracellular ncRNAs also show altered expression in different human pathologies, thus their role has been proposed as non-invasive biomarkers, prognostic factors and also therapeutic targets in cancer or in other complex diseases. Among the regulatory ncRNAs we found: microRNAs (miRNAs), small interfering RNAs (siRNAs), Piwi-associated RNAs (piRNAs), long non-coding RNAs (lncRNAs), circular RNAs (circRNAs) and the tRNA derived small RNAs (tsRNAs) (Figure 2). To date, the most studied molecules are miRNAs, lncRNAs and circRNAs. miRNAs are long 18–25 nucleotide (nt) single-stranded RNAs, evolutionarily conserved, which negatively modulate the expression of their target mRNAs. They bind to the 3′ Untranslated Region (3′ UTR) of specific mRNA targets, leading to translational repression, or mRNA cleavage [4, 5, 6] miRNAs are very important molecules in the regulation of gene expression at the post-transcriptional level, a single miRNA can control the expression of several mRNAs and a single mRNA may be targeted by more than one miRNA, thus creating a complex network of cooperative regulation [7]. lncRNAs are the most heterogeneous class of non-protein-coding RNAs with lengths ranging from 200 to 100,000 nt. They include transcripts that may be classified as: (a) intergenic lncRNAs; (b) intronic lncRNAs; (c) sense or antisense transcripts; (d) pseudogenes; and (e) retrotransposons [8]. lncRNAs regulate gene expression at different levels, including chromatin modification, alternative splicing, protein localization and activity and can protect 3′ UTR of mRNAs from miRNA binding, increasing their stability [8]. Several recent studies have also shown that lncRNAs are critically involved in a wide range of biological processes, such as cell cycle regulation, pluripotency, differentiation and cell death [9, 10, 11, 12]. CircRNAs are a recently discovered class of circular single strand RNA molecules, covalently closed, which are resistant by the exonuclease [13]. Their function and the biological process in which they are involved remain mainly unknown but recent evidence suggests that circRNAs may play an important role in RNA–RNA interactions. In some cases, circRNAs exhibit multiple binding sites for the same miRNA and for this reason it has been suggested that they could represent potential molecular sponges for sequestering the most abundant miRNAs [14]. It means that they can act as a negative regulation of miRNAs by competing, using their binding sites, with the miRNA–mRNA target interaction. Their important role in gene regulation is demonstrated by some papers, which show that deregulation of circRNAs is associated in different types of cancer [15]. Scientific community recognized the important role played by regulatory ncRNAs in cell physiology and also the effects caused by their altered expression in all human diseases but many other studies will be necessary. In fact, sometimes, we are able to identify the differential expression of one or more ncRNAs in a particular disease and also to evaluate their prognostic value but we are not able to correlate accurately specific expression profiles to specific phenotype alteration, because of the complexity and the redundancy of the circuits regulating gene expression. In this review, we will point up the ncRNA, particularly miRNA and lncRNAs, found expressed in endometrium in physiological conditions, focusing on their involvement in embryo implantation. Moreover, we will analyse the alteration of their expression in the pathogenesis of some endometrial diseases, as endometrial cancer, endometriosis and chronic endometritis.',\n",
       "  'Methods': '',\n",
       "  'Results': '2. ncRNAs and Embryo Implantation The success of embryo implantation is related to blastocyst quality as well as to the endometrium receptivity. Both the embryo and the maternal tissues are able to reciprocally exchange signals; the proper secretion of signal molecules and their uptake allow the successful implantation [16, 17]. Several mediators such as growth factors, cytokines, chemokines, lipids, matrix-degrading enzymes and integrins, whose expression is regulated by oestrogen (E) and progesterone (P), influence endometrial receptivity [18]. Many genes have been discovered that have important roles in embryo implantation and in 2011 a genomic tool was developed to assess the endometrial receptivity of woman during In Vitro Fertilization cycles (IVF). The Endometrial Receptivity Array (ERA) test, based on microarray technology, analyses the expression profile of 238 genes encoding proteins which are related with the implantation process, during the Window of Implantation (WOI). The ERA test has been shown to be more accurate than histological evaluation to assess endometrial receptivity. This has subsequently led to the new concept of personalized embryo transfer, by using endometrial biomarkers as a therapeutic strategy for patients with recurrent implantation failure [18, 19, 20, 21, 22] According to the rising interest in the role of ncRNAs in the regulation of gene expression, different studies have explored the involvement of miRNAs in implantation, focusing on the dialog between embryo and maternal tissues. Different miRNAs, associated with endometrial receptivity, have been identified in endometrial biopsies as well as in endometrial fluids (Table 1). Table 1. miRNAs involved in embryo receptivity. miRNAs Species Sample Reference mmu-let-7a mouse endometrium [23] mmu-let-7b mouse endometrium [23] hsa-let-7e* human exosomes from endometrial cell lines [34] hsa-let-7f-2* human exosomes from endometrial cell lines [34] hsa-miR-30b human endometrium [24] hsa-miR-30d human endometrium/endometrial fluid [24, 32] hsa-miR-122 human exosomes from endometrial cell lines [34] hsa-miR-124 human exosomes from endometrial cell lines [34] hsa-miR-125b human endometrial cell lines [30] mmu-miR-126-3p mouse endometrium [29] hsa-miR-129* human exosomes from endometrial cell lines [34] hsa-miR-142-3p human exosomes from endometrial cell lines [34] hsa-miR-145 human endometrium [26] mmu-miR-199a mouse endometrium [25] hsa-miR-222* human exosomes from endometrial cell lines [34] hsa-miR-376c human exosomes from endometrial cell lines [34] hsa-miR-409-3p human exosomes from endometrial cell lines [34] mmu-miR-429 mouse endometrium [28] hsa-miR-432 human exosomes from endometrial cell lines [34] hsa-miR-451 human exosomes from endometrial cell lines [34] hsa-miR-494 human endometrium [24] hsa-miR-520h human exosomes from endometrial cell lines [34] hsa-miR-1248 human exosomes from endometrial cell lines [34] Open in a new tab 2.2. miRNAs in Endometrial Receptivity Different studies, on animal models and humans, confirmed the important role played by miRNAs in endometrial physiology by regulating the changes in gene expression levels during the different phases of the endometrial cycle. In fact, many of them were found differentially expressed at each stage of the endometrial cycle. MiR-30b and miR-30d were found to be significantly up-regulated and miR-494 down-regulated in the receptive endometrium (LH+7) compared with the pre-receptive endometrium (LH+2) from healthy fertile women. The bioinformatic prediction of the target genes of these miRNAs showed that they are involved in the cyclic remodelling of the endometrium, including endometrial maturation to the receptive state [23]. Mucin 1 (Muc1) is an integral transmembrane mucin glycoprotein expressed on the apical surface of the endometrium, acting as an inhibitor of embryo attachment. In mice, the expression of Muc1 decreased significantly during the WOI and could be due to a negative regulation mediated by miR-199a, let-7a and let-7b [24, 25]. Insulin-like growth factor 1 receptor (IGF1R) is an important receptor, up-regulated in the endometrium during the receptive stage, which is closely related to embryo implantation. In fact, its increase on the surfaces of the endometrium might contribute to adhesive interaction with the embryo. It has been demonstrated that a high expression of miR-145 having as target the mRNA coding for IGF1R, inhibits embryo attachment. Mir-145 up-regulation has been shown in the endometrium of RIF patients [26]. Another very important characteristic for endometrial receptivity is the epithelial-mesenchymal transition (EMT). In EMT, the cells lose their polarity and display a remodelling of cell junctions in order to facilitate the interaction with embryo trophectoderm [27]. As an important suppressor of EMT, miR-429 exhibited a down-regulation during implantation in mice. Enhancement of miR-429 resulted in suppression of the migratory and invasive capacities of cells, probably through targeting protocadherin 8, leading to reduced implantation sites [28]. On the contrary, miR-126-3p was specifically up-regulated in implantation sites, promoting cell migratory and invasive capacity by regulating the expression of integrin α11 [29]. Moreover, we know that sex hormones induce changes in miRNA expression in the endometrium, for example, progesterone induced the expression of miR-125b in human endometrial epithelium cells. The increased expression of miR-125b inhibits cell movement and prevents implantation, by suppressing the expression of Matrix Metallopeptidase 26 (MMP26), a member of the matrix metalloproteinase family, which is involved in degradation of the extracellular matrix [30]. These studies demonstrated that, in addition to the cyclical changes in endometrium morphology and gene expression, there exists a very complex network of miRNAs regulating the expression of many genes encoding proteins related to endometrial receptivity and implantation. 2.3. Extracellular miRNAs in Endometrial Fluid To assess endometrial receptivity, many researchers are focusing their studies on extracellular ncRNAs present in endometrial fluid. Endometrial fluid is a viscous fluid secreted by the endometrial glands into the uterine cavity and its role consists in nurturing the embryo and represents the important microenvironment in which both embryo and endometrium can interact with each other [31]. The embryo, induced by cytokines and different proteins secreted by the endometrium, modulates, in turn, the secretion of integrins β3, α4 and α1, interleukins, chemokines and leptine. Recently, ncRNAs have been added, as new players, in mediating the dialog between embryo and endometrium. It has been demonstrated that endometrial cells secrete exosomes in endometrial fluids and the secretion reaches a peak during WOI [32]. On the other hand, also embryos secrete extracellular vesicles in vitro that are able to be up taken by cultured endometrial cells [33]. During the last few years, the presence of miRNAs in endometrial fluid has been shown in humans and in animal models [27]. Comparing miRNA profiles between endometrial cells and exosomes present in endometrial fluid, 13 miRNAs out of 227 were exclusively present in exosomes/microvesicles and their mRNA targets were involved in several signalling pathways associated with implantation [34]. In 2015, in a very interesting paper, the authors analysed the RNA transcript present in endometrial fluid by microarray and found that a large number of miRNAs were present in the endometrial fluid. They found 27 miRNAs differentially expressed during the WOI with respect to the four different phases of the menstrual cycle and among them miR-30d was the most up-regulated miRNA [32]. The authors also demonstrated that miR-30d is an exosomal miRNA and it is internalized by trophoblastic cells of murine embryos. Moreover, embryos treated with miR-30d exhibited increased expression of ten genes, including those encoding adhesion molecules such as Integrin Subunit Beta 3 (ITGB3), Integrin Subunit Alpha 7 (ITGA7) and Cadherin 5 (CDH5) [32]. These studies demonstrated the molecular dialog, mediated by miRNAs, between embryo and endometrium, in order to promote embryo implantation. 3. ncRNAs and Endometrial Cancer 3.1. Endometrial Cancer Endometrial cancer (EC) is the most common gynaecological tumour in developed countries, it is the fifth most common cancer and the 14th in terms of mortality [35]. It is becoming clear that EC, as well as many other tumours, includes different subtypes having specific genetic and molecular features. Based on histological characteristics, specific protein expression and grade, EC has been classified into two subtypes [36]. Type 1, also called endometrioid type (EEC), represents 70–80% of new sporadic cases. These cancers are typically well differentiated, associated with increasing oestrogen levels and the patients have a favourable prognosis. On the molecular level, more than 80% type I tumours are associated with a decreased or lacking expression of Phosphatase and tensin homolog (PTEN) and to an overexpressed Oestrogen Receptor (ER), which promote a deregulated cellular proliferation [37, 38, 39, 40]. Furthermore, EEC are frequently associated with microsatellite instability (leading to DNA mismatch repair), genetic mutations and epigenetic abnormalities [41]. Type 2, known as non-endometrioid endometrial carcinoma (NEEC), is not related to circulating oestrogen levels and it is less common than type 1 (10–20% of ECs) [42]. Patients with NEEC, usually have a poor prognosis because the tumour is typically diagnosed when metastases are already present [36]. On the molecular level, NEEC tissue samples show a high expression of Tumour Protein p53 (TP53) and Cyclin Dependent Kinase inhibitor 2A (CDKN2A or P16), resulting in non-functional proteins that accumulate in the cell acting as a double negative inhibitor of the wild-type p53, leading to propagation of aberrant cells and to uncontrolled cell growth [43]. Many papers have investigated the role of ncRNA in EC in an attempt to find new molecular markers able to discriminate the different subtypes, predict prognosis and design new drugs for personalized targeted therapies [44, 45, 46, 47]. In this review, we focused our attention on more important biological pathways and processes altered in EC and we discussed the role of ncRNAs in their regulation. 3.2. miRNAs in Endometrial Cancer The phosphatidylinositol 3-kinase (PI3K) pathway is involved in the control of growth, survival, proliferation and apoptosis. Gain- or loss-of-function mutations of genes encoding proteins involved in the pathway lead to neoplastic transformations and it has been demonstrated that several components are dysregulated also in EC [48]. PTEN is a tumour suppressor that negatively regulates the PI3K-AKT signalling pathway. Loss of its function is implicated in the pathogenesis of a number of different tumours, including endometrial carcinoma [49]. PTEN mRNA is negatively regulated by miR-205 and it has been demonstrated that miR-205 up-regulation leads to increased B-cell lymphoma 2 (BCL2) levels and TP53 down-regulation, inhibiting tumour cell apoptosis and enhancing proliferation [50]. Moreover, by stimulating the AKT pathway and inhibiting glycogen synthase kinase 3β (GSK3B), miR-205 suppresses E-cadherin expression and promotes SNAIL expression [51]. Transfecting endometrial cancer cells with LNA-miR-205-inhibitor (Locked Nucleic Acid-inhibitor of miR-205), Torres et al. recently obtained decreased endometrial cancer cell proliferation in vitro and in vivo [52]. PTEN can also be controlled by miR-21, involved in enhanced malignant transformations and proliferation of type 1 EC cells [53]. The transcription factor Forkhead Box O1 (FOXO1), belonging to the Forkhead Box class O family, is a downstream target of the PI3K pathway and it is able to regulate different biological processes involved in endometrial phases such as menstruation, uterine cell regeneration, tissue remodelling and cell differentiation [54, 55]. In EC type 1, FOXO1 is down-regulated and it has been shown that miR-9, miR-27, miR-96, miR-153 and miR-182, able to target FOXO1 mRNA, are over expressed in the same cells [56]. On the other hand, Forkhead Box C1 (FOXC1), another member of the forkhead box transcription factors, is an oncogene controlling tumour cell migration and metastasis and it is inhibited by miR-204 [57]. MiR-204 down-expression has been associated with solid tumour development, such as in the lung and gastric and endometrioid endometrial cancer [57, 58, 59]. The hyperactivation of the PI3K pathway leads to an inhibition of apoptosis by the repression of BCL2 associated agonist of cell death (BAD) and the enhancement of BCL2 activity [60]. On the other hand, apoptosis inhibition in EC can occur through BCL2 overexpression and BCL2 associated X (BAX) downregulation promoted by higher oestrogen levels. One of the most important regulator families involved in this mechanism seems to be the let-7 miRNA family, that is, let-7a, let-7b, let-7c, let7d, let-7e, let-7f and let-7g, whose expression grows in response to oestrogen exposure [61]. Nevertheless, high levels of let-7 are related to low malignancy of type I endometrial cancer, suggesting that these molecules could have dual oncogenic and cancer-suppressive effects in endometrial cancer cells. Another important protein, often evaluated to assess an aetiology-specific therapy in type 1 EC patients, is ERα [62]. Its amplification was regarded as a prevalent event in endometrial cancer, however, a decreased ERα level has been detected in high-stage and poorly differentiated cancers [63, 64, 65]. ERα seems to be suppressed by miR-206, leading to an anti-proliferation effect and decreased invasion capacity [66]. IGF1R, by binding its IGF1 and IGF2 ligands, is able to activate the PI3K pathway, stimulating cell proliferation and inhibiting apoptosis. Therefore, if it is over-expressed, as in most malignant tissues, it plays an anti-apoptotic role, promoting cancer cell survival and tumour metastasis [67, 68]. Recent studies demonstrated that IGF1R was highly expressed in EC tissues and it was inversely correlated with miR-381 levels [69]. Epigenetic mechanisms seem to be involved in EC: silencing of DNA mismatch repair genes by DNA hypermethylation has been demonstrated [70]. Moreover, the study of methylation profiles in endometrial tumorigenesis showed that, among different tumour suppressor genes, the number of methylated promoters increased in the progression of cancer [71]. For example, miR-129-2 seems to be related to microsatellite instability and mutL homolog 1 (MLH1) methylation, which is frequently observed in EC cells, suggesting an important role in early stages of carcinogenesis [72, 73]. Usually, miR-203 suppresses tumour proliferation, invasion and metastasis, through the inactivation of ABL proto-oncogene 1 (ABL1) and BCR-ABL1 oncogenes. In hematopoietic tumours, it has been found silenced by the hypermethylation of its promoter [74]. Huang and collaborators found its hypermethylation in EC cells lines and they observed that it was associated with microsatellite instability and MLH1 methylation in primary endometrioid EC [75]. Mir-152 is a tumour suppressor, which inhibits tumour cell growth both in vitro and in vivo by repressing Cell Division Cycle 25B (CDC25B), an important cell cycle regulator [76]. It is able to target E2F Transcription Factor 3 (E2F3), DNA methyltransferase 1 (DNMT1), met proto-oncogene (MET) and Rapamycin-insensitive companion of mTOR (Rictor) and in EC cells it seems to be down-regulated through CpG hypermethylation, promoting cancer development and progression [77]. Several miRNAs that regulate EMT have been found deregulated in EC. EMT, important in implantation, is also a key process in oncogenesis and tumour metastasis; it has been mainly related to the expression of three markers: epithelial E-cadherin, mesenchymal vimentin and N-cadherin [78]. MiR-93 has been described as a tumour suppressor in ovarian cancer and more recently as an oncomiR in EC cells. MiR-93 up-regulation leads to a downregulated E-cadherin and increased N-cadherin expression [79]. In addition, this miRNA seems to be able to target FOXA1, a negative regulator of EMT, down-regulated in EC cells [79, 80]. Another miRNA involved in EMT processes is miR-30c, which targets Metastasis-associated gene-1 (MTA1) inhibiting cell proliferation, metastasis and invasion in EC cells [81]. It is down-regulated both in type 1 and type 2 EC cells and its expression seems to be related to oestrogen concentration [82]. MiR-106b has been related to EMT, too: it has been associated with Twist family bHLH transcription factor 1 (TWIST1) targeting [83]. TWIST1 contributes to the EMT phenotype in EC cells improving cell invasion and it is more expressed in EC cells than in normal ones [84]. These findings are consistent with miR-106b downregulation in EC cells showing EMT phenotypes; profiling its expression levels might be helpful for predicting the risk of metastasis, especially in patients with type II EC [83]. 3.3. Circulating miRNA and Endometrial Cancer To discriminate the different subtypes and predict prognosis by using circulating miRNAs as non-invasive biomarkers of EC, several investigations have investigated their expression profiles in serum or plasma. High serum levels of miR-155 have been found associated with cancer stage and lymph node metastasis [85]. Another study demonstrated that the association of miR-99a and miR-199b, up-regulated in plasma of EEC patients, better discriminate EEC patients [86]. A recent study demonstrated that down-regulated miR-29b expression in plasma correlates with poor EC prognosis and is helpful to evaluate the EC prognosis [87]. A genome-wide study on miRNA expression profiles from plasma of EEC and subsequent validation by quantitative reverse-transcriptase polymerase chain reaction demonstrated the significant up-regulation of 5 miRNAs (miR-15b, miR-27a, miR-223, miR-3145 and miR-4638). Moreover, the authors found that miR-27a and CA125 together have a considerable clinical value in diagnosing EEC [88]. In summary, a lot of miRNAs, cellular and circulating, have been found dysregulated in endometrial cancer and a comprehensive list and the relative references are reported in Table 2 [89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114]. Table 2. miRNAs differentially expressed in endometrial cancer (EC) patients. miRNAs Expression Sample Reference Has-let-7 miRNAs up EEC [61] miR-let-7a down EC [89] miR-let-7b down EC [90] miR-9 up EEC [56] miR-10b up EC [91] miR-15a-5p down EC [92] miR-15b up Plasma in EEC patients [88] miR-21 up EEC [53] miR-23a down EEC [93] miR-24 down EC [94] miR-27a up EEC, Plasma in EEC patients [56, 88] miR-29b down EC, Plasma [87, 95] miR-30c down EC [81, 82] miR-31 up EC [96] miR-34c down EC [97] miR-96 up EEC [56] miR-99a up Plasma in EEC [86] miR-106b down EEC [83] miR-125b down EEC [98] miR-129 down EC [99] miR-129-2 down EC [72] miR-135a up EC [100] miR-139-5p down EC [101] miR-141 up EEC [102] miR-143 down EC [103] miR-145 down EEC [104] miR-148b down EC [105] miR-152 down EC [76] miR-153 up EC [56] miR-155 Up Serum [85] miR-181c down EEC [106] miR-182-5p up EEC [56, 102] miR-183-5p up EC [108] miR-194 down EC [109] miR-199b up Plasma in EEC patients [86] miR-200a* up EC [102, 108] miR-200b up EEC [102] miR-200c-3p up EC [110] miR-203 down EC [74] miR-204 down EC [57] miR-205 up EEC [50, 51, 102] miR-206 down EC [66] miR-223 up Plasma in EEC [88] miR-361 down EC [90] miR-375 up EC [100] miR-381 down EC [69] miR-424 down EC [111] miR-503 down EEC [112] miR-944 up EC [113] miR-1271 down EC [114] miR-3145 up Plasma in EEC patients [88] miR-4638 up Plasma in EEC patients [88] Open in a new tab 3.4. lncRNAs in Endometrial Cancer To date, there are no studies about lncRNA profiles during physiological endometrial changes but their altered expression in EC as well as in endometriosis has been demonstrated [115, 116, 117]. In fact, different lncRNAs seem to be differentially expressed in EC and some of them could have a prognostic value. Ovarian adenocarcinoma amplified lncRNA (OVAAL) is overexpressed in many ovarian serum carcinoma, as well as in type 1 EC but less frequently in type 2 EC [46, 118]. Its over expression is related to overexpression of p53-regulated genes [118]. Metastasis-associated lung adenocarcinoma transcript 1 (MALAT1) is an 8000 nt lncRNA, upregulated in many different human tumours. It has been positively associated with hyperplasia and negatively with metastasis, thus it can be used as a predictive biomarker. Its expression is regulated by the Wnt/β-catenin signalling pathway, frequently abnormally enhanced in endometrioid type EC. Particularly, MALAT1 has been identified as a functional downstream target of Protocadherin 10 (PCDH10), a tumour suppressor protein, down-regulated in EEC [119]. Growth arrest-specific 5 (GAS5) is a tumour suppressor gene, implicated and aberrantly expressed in multiple cancers [120, 121, 122, 123, 124]. Recent evidences show that this lncRNA is down-regulated in endometrial cancer cells, being able to induce their apoptosis by inducing PTEN expression through inhibiting miR-103, which usually stimulates cell growth and invasion in endometrial carcinoma [125]. HOX transcript antisense intergenic RNA (HOTAIR) is one of the most studied lncRNAs, because of its involvement in genome modification: in fact, it can repress gene expression through the activation of chromatin modifiers [126]. Its overexpression is associated with increasing oestrogen levels and correlated with poor cancer prognosis. HOTAR has been found upregulated in EC compared to normal endometrium samples. HOTAIR overexpression is also associated with increased metastatic spread and a reduced overall-survival rate in EC [127]. Using a mouse xenograft EC model, treated with HOTAIR siRNA lentivirus, a significant tumorigenesis rate and tumour size reduction occurred both in vitro and in vivo [128]. This shows that HOTAIR may be a prognostic molecular marker for EC, even if further studies are required to demonstrate its involvement in endometrial cancer progression and metastasis [127] H19, a paternally imprinted lncRNA, is located on chromosome 11 and lies within 200 kbp downstream of the IFG2 gene [129]. H19 is also known as “oncofetal non-coding RNA” because it is primarily expressed during foetal development, so it is absent or poorly expressed in most normal adult tissues. Its re-expression has been detected in various cancers, including ovarian cancer [130, 131, 132, 133]. H19 expression levels increase throughout endometrial epithelium tumorigenesis: they are low in normal epithelium, higher in hyperplastic endometrium, very high in EC and even higher in dedifferentiated tumour tissues [115]. Urothelial Cancer-Associated 1 (UCA1) is one of the lncRNAs mostly associated with tumour progression, metastasis and chemo-resistance in several cancer types [134, 135, 136, 137, 138]. Lu et al. discovered that UCA1 levels were significantly higher in EC cells than in normal endometrial samples and in metastatic EC they noticed a further level increase if compared to primary tumour, suggesting its involvement in tumour cell migration [139]. The circRNAs, another class of ncRNAs, has been shown to be enriched and stable in exosomes. They act as natural miRNA sponges to decrease miRNA levels, reducing their regulatory effect on mRNAs [140, 141]. In a recent study, Xu et al. found that the number of exosomes isolated from serum of EC patients was higher than those of normal samples. In addition, the expression profile of circRNAs in serum of EC patient was dysregulated [142]. Another analysis found several circRNAs differentially expressed in EC cells when compared to normal endometrium [143]. In Table 3, we present a list of significant lncRNAs deregulated in EC and the relative references [144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156]. Table 3. lncRNAs differentially expressed in EC patients. lncRNAs Expression EC Type Reference ASLNC04080 up EC [144] BANCR up EEC [145] CARLo-5 up EC [146] CCAT2 up EC [147] FER1L4 down EC [148] GAS5 down EC [125] H19 up EC [115] HAND2-AS1 down EEC [149] HOTAIR up EC [127] LA16313D11.11 down EC [150] LINC00672 down EC [151] LINC01016 up EC [152] MALAT1 down EEC [119] MEG3 down EC [153] NEAT1 up EEC [154] OVAL up EEC [46, 118] RP11395G12.3 down EC [150] SNHG8 up EC [155] TUSC7 down EC [156] UCA1 up EC [139] Open in a new tab 4. ncRNAs and Endometriosis 4.1. Endometriosis Endometriosis is a disabling disorder characterized by the presence of endometrial tissues outside of the uterine cavity. The most common symptoms are non-menstrual pelvic pain, dyspareunia and infertility [157]. The exact causes are unknown, even if the most shared theory is retrograde menstruation, that is endometrial cells through menstrual blood move in on pelvic cavity. These cells present different characteristics, as the ability of adhesion, aggression, neo-angiogenesis and inhibition of apoptosis, which make endometriosis similar to cancer [158, 159]. Although decades of research about the pathogenesis of endometriosis have shown the role of hormonal and non-hormonal mechanisms related to disease development, the therapeutic approach and the methods for early diagnosis are still lacking. Today, the best method for the diagnosis of endometriosis remains laparoscopic surgery but the finding new biomarkers for a minimally invasive diagnosis represent an important scientific challenge [160]. 4.2. miRNAs and Endometriosis Different papers have been published about miRNAs potentially involved in endometriosis but unfortunately, many times the concordance among the results is small. This could be due to the heterogeneity of cell types used in the experiments, as illustrated in a recent review [161]. Another possible explanation is that various technologies (deep-sequencing, microarray, Real-Time PCR) have been used by different researchers. The first studies on miRNAs and endometriosis demonstrated that different miRNAs were differentially expressed in endometrial tissues with endometriotic lesions with respect to endometrial tissues of the same woman without endometriotic lesions [162, 163]. Burney et collaborators were among the first authors to publish the difference in expression profiles of miRNAs between eutopic endometrial tissues of woman with and without endometriosis [164]. They found that four miRNAs (miR-34c-5p, miR-9, miR-9*, miR-34b*) were down-regulated in women with endometriosis compared to controls. In another paper, the authors showed that in patients with ovarian endometriosis, miR-483-5p, targeting IGF2 and miR-629-3p were down-regulated in the eutopic endometrium. These authors suggest that the dysregulation of these miRNAs and their target genes could contribute to the overgrowth of endometrial tissue outside the uterus [165]. Further studies have been carried out in the last few years on miRNAs and endometriosis. In 2017, Min Kyoung Kim and collaborators demonstrated that the expression of miR-27b-3p, upregulated in human endometrial stromal cells from patients with endometriosis, is reduced after the treatment with Rg3-enhanced red ginseng extract (Rg3E). The authors showed the same effects in vivo in a mouse model [166]. In 2018, by RNA sequencing, it was demonstrated that 107 miRNAs and 6112 mRNAs were differentially expressed in ectopic endometrium. The authors built regulatory networks among Transcription Factors, miRNAs and mRNA targets identifying the most important points of altered regulation in ectopic tissue. Among the miRNAs, they found that some members of miR-449 and miR-34b/c cluster, miR-200 family, miR-106a-363 cluster were dysregulated [167] (Table 4). Members of miR-200 family are also involved in cell migration and EMT which supposedly occurs in the pathogenesis of endometriosis [161]. The downregulation of members of miR-200 family in endometrial tissues with endometriotic lesion results in increased expression of ZEB1/ZEB2 which are transcriptional repressors of E-cadherin. E-cadherin, in the normal endometrial tissue, is required for maintaining the epithelial nature of cells and their adhesion. Its down-regulation leads to epithelial cells to acquire mesenchymal characteristics [161]. Table 4. miRNAs differentially expressed in Endometriosis patients. miRNAs Expression Sample Reference let-7b down serum [160] let-7d down serum [160] let-7f down serum [160] miR-9 down endometrium [164] miR-9* down endometrium\\\\serum [164, 176] miR-15a-5p down endometrium [171] miR-17-5p down ovarian endometrium\\\\plasma [165, 169, 177] miR-20a down ovarian endometrium\\\\plasma [165, 169, 177] miR-20b-5p down endometrium [167] miR-21 up endometrial exosomes [172] miR-22 down plasma [175] miR-27b-3p up endometrium [166] miR-30d-5p down endometrium [167] miR-34b* down endometrium [164] miR-34c-5p down endometrium [164] miR-106a-5p down endometrium [167] miR-122 up serum [176] miR-133b up endometrium [167] miR-135a down serum [160] miR-141* down serum [176] miR-141-3p down endometrium\\\\plasma [167, 176] miR-145* down serum [176] miR-182-5p down endometrium [167] miR-183-5p down endometrium [167] miR-196b-5p down endometrium [167] miR-199a up serum [176] miR-200a-3p down endometrium\\\\plasma [161, 167, 178] miR-200b-3p down endometrium\\\\plasma [161, 167, 178] miR-200c-3p down endometrium [161, 167] miR-363-3p down endometrium [167] miR-449a down endometrium [167] miR-449b-5p down endometrium [167] miR-483-5p down endometrium [165] miR-542-3p down serum [176] miR-629-3p down endometrium [165] Open in a new tab Neoangiogenesis regulators such as Vascular Endothelial Growth Factor A (VEGFA) and Thrombospondin-1 (THBS1) have been involved in the pathology of endometriosis [168]. Two different groups of researchers reported that miR-17-5p and miR-20a were down-regulated in the ovarian endometrium compared to eutopic endometrium [165, 169] Braza-Boïls and collaborators, in 2013, reported that the miR-17-92 cluster increases tumour neovascularization by decreasing THBS1 expression [170]. MiR-15a-5p, a negative regulator of VEGFA, has been found down-regulated in endometrium samples of woman with endometriosis and VEGFA was found up-regulated in the same endometrial tissues. Moreover, the transfection of endometriosis stromal cells with miR-15a-5p mimics led to reduction in the expression of VEGFA and migration abilities of the endometrial stromal cells, suggesting an important role of this miRNA in the pathogenesis of endometriosis [171]. It has been demonstrated that endometriotic cells in vitro are able to secrete miRNA EV cargo, in culture medium and the role of extracellular miRNAs in the pathogenesis of endometriosis has also been studied. Harp and collaborators observed that miR-21, pro-angiogenic miRNA, is up-regulated in exosomes from endometriosis samples compared with exosomes from controls. The authors suggested that exosomes play autocrine/paracrine roles in the development of endometriosis, potentially by modulating angiogenesis [172]. 4.3. Circulating miRNAs as Biomarkers of Endometriosis Due to the anatomical location of this disease, several closely related biological fluids have been proposed as source for non-invasive biomarkers of endometriosis, for example: urine, plasma/serum and menstrual blood. One potential use of circulating miRNAs as non-invasive biomarkers for endometriosis is an ongoing area of research and its diagnostic and therapeutic implications have been extensively reviewed [164, 173, 174, 175]. In 2013, Wang and collaborators performed a circulating miRNA array profiling in serum and they found different deregulated miRNAs [176] (Table 4). Moreover, a microarray analysis revealed that miR-17-5p, miR-20a and miR-22 were down-regulated in plasma from patients with endometriosis [177]. It is known that prognostic value increases when two or more biomarkers are evaluated at the same time, in fact, two studies performed in serum and plasma demonstrated that let-7b, let-7d and let-7f altered levels, as well as miR-200a-3p, miR-200b-3p and miR-141-3p down-regulation are able to discriminate patients with endometriosis from controls [160, 178] (Table 4). 4.4. lncRNAs and Endometriosis In a paper published in 2015, Wang and collaborators compared the expression of several lncRNAs and mRNAs between eutopic and normal endometrium by microarray analysis and validated the obtained data using RT-qPCR. The authors found several lncRNAs and mRNAs differentially expressed between eutopic and normal endometrium. Specifically, among the lncRNAs, 488 were up-regulated and 789 down-regulated and among the mRNAs, 578 were over expressed and 638 down-regulated [117]. The most significantly up-regulated lncRNA is AC068282.3 (fold change 31.3) while RP11-403H13.1 is the most significantly down-regulated one (fold change 44.3). The differentially expressed lncRNAs were involved in cell cycle regulation and immune response [117]. As previously described, H19 is a lncRNA, which is expressed from the imprinted locus that also contains the reciprocally imprinted IGF2 gene. It has been shown that H19, reducing the bioavailability of miRNA let-7, by acting as a molecular sponge, was significantly down-regulated in the eutopic endometrium of women with endometriosis. Its down regulation, increasing let-7 activity, inhibited IGF1R expression. Accordingly, the proliferation of endometrial stromal cells was reduced. This paper represents the first example of lncRNA involvement in the pathogenesis of endometriosis and its association with infertility [179]. Another lncRNA that is important in endometriosis is MALAT1. This is one of the most known lncRNAs, it is evolutionary well-conserved and is overexpressed in many cancers [180, 181, 182, 183]. Recently, a paper demonstrated that the expression of MALAT1 is significantly increased during endometriosis. Moreover, the authors showed that miR-200c, which is regulated by MALAT1, was significantly down-regulated in endometrial samples from woman with endometriosis and they were significantly negatively correlated. The functional assay showed that overexpression of miR-200c inhibited the proliferation of Human Endometrial Stromal Cells (HESCs) while the inhibition of miR-200c promoted cellular proliferation. Similar results were also obtained from the migration assay in which MALAT1 knockdown inhibits the proliferation and migration of HESCs [184]. Finally, the presence of deregulated lncRNAs in serum as potential non-invasive biomarkers of endometriosis has been investigated. In 2016, it has been reported that a set of lncRNA in serum can discriminate severe versus mild stages of endometriosis and other associated clinical features. Moreover, the combination of five lncRNAs (NR_038395, NR_038452, ENST00000482343, ENST00000544649, ENST00000393610) is able to discriminate patients with and without endometriosis [116]. In Table 5, we show the list of significant lncRNAs deregulated in endometriosis and the relative references. Table 5. lncRNAs differentially expressed in Endometriosis patients. lncRNAs Expression Sample Reference AC068282.3 up endometrium [117] ENST00000393610 up serum [116] ENST00000465368 down serum [116] ENST00000482343 down serum [116] ENST00000529000 down serum [116] ENST00000544649 down serum [116] H19 down endometrium [179] MALAT1 up endometrium [184] NR_033688 down serum [116] NR_038395 down serum [116] NR_038452 up serum [116] RP11-403H13.1 down endometrium [117] Open in a new tab 5. ncRNAs and Chronic Endometritis 5.1. Chronic Endometritis Chronic Endometritis (CE) is a chronic inflammation of the endometrium that is difficult to diagnose, caused by microbial infection sustained by common bacteria such as Escherichia coli, Enterococcus faecalis and Streptococcus agalactiae [185, 186]. It has vague symptoms, such as abnormal uterine bleeding, pelvic pain and leukorrhea and, at the moment, hysteroscopy represents the most reliable diagnostic method [187, 188]. In spite of the benign prognosis, this pathology can impair endometrial receptivity, in spontaneous and in IVF cycles [189, 190]. Moreover, there are evidences showing that CE could be a cause of natural preterm labour, premature birth and recurrent miscarriages [188, 191]. To date, the molecular mechanisms with which CE can cause infertility still remain unknown. In a paper published by our group in 2013, we demonstrated the altered endometrial expression of genes, involved in inflammatory response, proliferation and apoptosis in CE. These findings are in agreement with literature data, showing that the endometrium of CE women results an unusual local microenvironment, due to an altered secretion of paracrine factors [192, 193, 194]. Accordingly, endometrial receptivity may be impaired as well as proliferative processes may be increased [195, 196]. 5.2. miRNAs and Chronic Endometritis At the moment, very few data are present in the literature about the alteration of molecular pathways associated with CE. In spite of CE is considered a mild pathology, it is related to infertility and it has been suggested the association between CE and endometriosis and between CE and the endometrial micro-polyps [193, 195]. Therefore, to investigate the alteration of molecular mechanisms caused by a chronic infection could be interesting. We showed, for the first time, an up-regulation of miR-27a-3p and miR-124-3p in the endometrium from women affected by CE and demonstrated their ability to discriminate CE women when the 2 miRNAs were analysed together in the serum [197]. Previously, miR-27a has been found highly expressed in cows with subclinical endometritis [198]. Finally, we found that miR-27a up-regulation is significantly related to IGF1 down-regulation in the same endometrial samples [197]. Proper IGF1 levels are required for successful embryonic and placental development; consequently, the down-regulation of protein could be associated with endometrial quality and with infertility observed in women with CE [199]. Interestingly, miR-27a and miR-27b, two isoforms of the miR-27 family, seem to be involved in different endometrial pathologies. Their up-regulation in endometrium and serum has been described in EC, in endometriosis and in CE [56, 166, 197] (Figure 3). It would be attractive to demonstrate, by functional experiments, its ability to regulate IGF1 and other messengers, in order to understand its role in endometrial physiopathology and evaluate its prognostic potential and the possibility to plan specific therapy based on RNA interference. Figure 3. Open in a new tab Representative miRNAs involved in endometrial pathologies. Literature analysis revealed that there are no data about lncRNAs in CE (Figure 4). The overlapping among lncRNAs involved in EC and Endometriosis showed that MALAT1 and H19 are shared across the two pathologies, even if, they show a different trend of expression (Figure 4). Figure 4. Open in a new tab Representative lncRNAs involved in endometrial pathologies. Venn diagram summarizing deregulated miRNAs identified in Endometrial Cancer, Endometriosis and Chronic Endometritis. Intersection areas show molecules differentially expressed in more than one pathology. Up and down-regulated miRNA expression is indicated by black arrows. Venn diagram summarizing deregulated lncRNAs identified in Endometrial Cancer and Endometriosis. Intersection areas show molecules differentially expressed in more than one pathology. Up and down-regulated lncRNA expression is indicated by black arrows.',\n",
       "  'Discussion': '',\n",
       "  'Conclusion': '6. Conclusions The discovery of ncRNAs in the regulation of gene expression inside the pathway from genotype (DNA) to phenotype (Proteins) has deeply transformed contemporary biomedicine. The countless studies on pathogenesis of tumours and complex phenotypes revealed the involvement of ncRNAs in different human diseases and postulated their role as molecular biomarkers in diagnosis, prognosis and also as possible therapeutic targets. In this review, we reported that several ncRNAs are involved in endometrial physiology and that their altered expression can be related to different endometrial disorders. High-throughput techniques are able to give a general overview, discovering many molecules involved in a specific biological pathway. Nevertheless, the complexity of ncRNA networks makes difficult to understand if deregulation of specific molecules represents the etiological cause of a disease or, alternatively, a secondary effect, depending on the perturbation of physiological pathways. Even if, EC, endometriosis and CE are dissimilar endometrial disorders, an association among them could be supposed. Comparing literature data, we detected ncRNAs found dysregulated in the three pathologies (Figure 3 and Figure 4). For some of them, there is no expression concordance between EC and Endometriosis, probably due to the different regulated targets but interestingly 4 miRNAs show the same expression trends and the up-regulation of miR-27 family seems to be involved also in CE. We believe that in order to understand and resolve particular biomedical problems, it would be appropriate to focus on a limited number of molecules to perform functional analyses, in vitro and in vivo on animals.',\n",
       "  'URL': 'https://doi.org/10.3390/ijms19072120',\n",
       "  'Journal': 'Int J Mol Sci'},\n",
       " {'PaperTitle': 'Guidelines for biomarker discovery in endometrium: correcting for menstrual cycle bias reveals new genes associated with uterine disorders',\n",
       "  'Publication': '2021',\n",
       "  'Authors': 'Almudena Devesa-Peiro, Patricia Sebastian-Leon, Antonio Pellicer, Patricia Diaz-Gimeno',\n",
       "  'Email': 'patricia.diaz@ivirma.com',\n",
       "  'Abstract': 'Transcriptomic approaches are increasingly used in reproductive medicine to identify candidate endometrial biomarkers. However, it is known that endometrial progression in the molecular biology of the menstrual cycle is a main factor that could affect the discovery of disorder-related genes. Therefore, the aim of this study was to systematically review current practices for considering the menstrual cycle effect and to demonstrate its bias in the identification of potential biomarkers. From the 35 studies meeting the criteria, 31.43% did not register the menstrual cycle phase. We analysed the menstrual cycle effect in 11 papers (including 12 studies) from Gene Expression Omnibus: three evaluating endometriosis, two evaluating recurrent implantation failure, one evaluating recurrent pregnancy loss, one evaluating uterine fibroids and five control studies, which collected endometrial samples throughout menstrual cycle. An average of 44.2% more genes were identified after removing menstrual cycle bias using linear models. This effect was observed even if studies were balanced in the proportion of samples collected at different endometrial stages or only in the mid-secretory phase. Our bias correction method increased the statistical power by retrieving more candidate genes than per-phase independent analyses. Thanks to this practice, we discovered 544 novel candidate genes for eutopic endometriosis, 158 genes for ectopic ovarian endometriosis and 27 genes for recurrent implantation failure. In conclusion, we demonstrate that menstrual cycle progression masks molecular biomarkers, provides new guidelines to unmask them and proposes a new classification that distinguishes between biomarkers of disorder or/and menstrual cycle progression.',\n",
       "  'Introduction': 'The human endometrium is hormonally regulated and changes throughout the menstrual cycle (Noyes et al., 1975; Murphy, 2004; Talbi et al., 2006). During most of the menstrual cycle, the endometrium is not receptive to embryonic implantation; it becomes receptive during a period of two to four-five days within the mid-secretory phase known as the window of implantation (Harper, 1992; Wilcox et al., 1999). To aid in assisted reproduction, endometrial dating methods have been exhaustively investigated over the past 40\\u2009years, as have potential reliable biomarkers of receptive status, including days from luteinising hormone (LH) peak or exogenous progesterone administration, morphological changes detected by ultrasounds, histological features and endometrial gene expression profiles (Noyes et al., 1975; Díaz-Gimeno et al., 2011; Niederberger et al., 2018; Craciunas et al., 2019).\\nSuboptimal endometrial receptivity and altered embryo-endometrial dialogue are considered to be responsible for one third of implantation failures in in-vitro fertilisation (IVF) cycles (Somigliana et al., 2018). Uterine disorders are complex, polygenic and multifactorial gynecological alterations affecting endometrial gene expression. Endometrial pathologies such as endometriosis, uterine fibroids and adenomyosis are associated with infertility, and may impact endometrial receptivity (Devlieger et al., 2003; Dunselman et al., 2014; Zepiridis et al., 2016; Devesa-Peiro et al., 2020). Some disorders remain undiagnosed in IVF patients and current treatments are either invasive (e.g. surgical removal) or have low-to-moderate efficiency (Wang et al., 2009; Dunselman et al., 2014; Harada et al., 2016; Zepiridis et al., 2016; Tanbo and Fedorcsak, 2017). Moreover, issues such as recurrent implantation failure or pregnancy loss of endometrial origin remain incompletely understood with neither an efficient diagnosis nor effective infertility treatment (Jauniaux et al., 2006; Diaz-Gimeno et al., 2017; Sebastian-Leon et al., 2018). Therefore, identifying reliable biomarkers of uterine disorders is a priority to understand the molecular bases of the pathology and to improve diagnosis, prognosis and treatment.\\nWith the advent of high-throughput technologies, uterine disorders are evaluated by transcriptomic analysis to identify genes (Burney et al., 2007; Hawkins et al., 2011; Lédée et al., 2011; Tamaresis et al., 2014; Koot et al., 2016; Pathare et al., 2017) that could be useful as diagnostic biomarkers and/or therapeutic targets. Although many genes are reported as potential uterine disorder biomarkers, the results of individual studies overlap poorly (Altmäe et al., 2017; Lessey and Kim, 2017; Miravet-Valenciano et al., 2017; Sebastian-Leon et al., 2018), and reliable biomarkers and potential therapeutic targets that could be clinically meaningful to address suboptimal endometrial receptivity remain elusive. This lack of reproducibility between studies may arise from low sample sizes, sample heterogeneity, undetected confounding variables in gene expression experimental procedures or differing experimental designs and data analysis protocols (Gurevitch et al., 2018; Suhorutshenko et al., 2018; Craciunas et al., 2019; Devesa-Peiro et al., 2020).\\nMenstrual cycle progression has a profound influence on gene expression (Talbi et al., 2006; Koot et al., 2016; Diaz-Gimeno et al., 2017; Sebastian-Leon et al., 2018; Saare et al., 2019). This effect could mask the discovery of candidate uterine disorder biomarkers whose expression responds to both endometrial progression and alterations in uterine disorders. Consequently, it is unclear whether the observed changes in transcriptomic studies reporting differentially expressed genes (DEGs) reflect variations related to the disorder, to menstrual cycle progression or to both. To address this research gap, we quantified the menstrual cycle effect on biomarkers associated with uterine disorders and described current practices in endometrial transcriptomic analysis. Based on our findings, we propose new guidelines to correct for menstrual cycle bias in biomarker identification.',\n",
       "  'Methods': 'Ethics statement\\nThis is a retrospective study using case and control data from multiple studies of endometrial gene expression in women with uterine disorders and women with no endometrial pathology. The raw gene expression data and patient meta-data were downloaded from National Center for Biotechnology Information (NCBI) functional genomics data repository Gene Expression Omnibus (GEO), where data are made available for the scientific community. Following GEO policies, in these repository patient data are anonymised and encrypted, and no additional institutional review board is required for downloading (Edgar, 2002).\\nStudy search and selection\\nA systematic search and review of individual studies was conducted between October 2016 and January 2019 at the NCBI repository GEO (Edgar, 2002). The search identified experiments involving human transcriptomic case versus control raw data evaluating uterine disorders.\\nKeywords searched included endometrium, endometriosis, uterine fibroids, recurrent implantation failure (RIF) and recurrent pregnancy loss (RPL), among others (see Supplementary Table SI for a full list of search terms). No restrictions were placed on publication date or language. The final inclusion criteria were that: there was at least one uterine disorder evaluated in the study design; RNA was extracted directly from human endometrial biopsies; information regarding the menstrual cycle at the time of biopsy was available for all samples; sample sizes were greater than three for both case and control groups belonging to the same study; microarray or RNA sequencing data were obtained using Affymetrix, Illumina or Agilent gene expression platforms; and raw gene expression data were made freely available to download from GEO. Studies evaluating endometrial gene expression at different times of the menstrual cycle in women with normal endometrium were retrieved from GEO using the same keywords and criteria.\\nPre-processing and exploratory analysis\\nPre-processing and exploratory analyses were completed according to the gene expression platform used: raw data were downloaded and pre-processed using the affy v. 1.52.0 R package (Gautier et al., 2004) for studies measuring endometrial gene expression with Affymetrix microarray platforms and the limma v.3.30.13 R package (Ritchie et al., 2015) was used for studies using Agilent or Illumina devices. Normalisation between samples was applied using quantile normalisation (limma R package v.3.30.13); (Ritchie et al., 2015) and annotation from probeset to gene symbol was established with the biomaRt R package v. 2.30.0 (Durinck et al., 2009). For studies evaluating gene expression through RNA-Seq, low-count filtering and normalisation was achieved with the edgeR R package v. 3.16.5 (Robinson et al., 2010).\\nExploratory analyses were performed to detect batch effects such as sequencing run or microarray slide. Detected effects were treated using linear models [limma R package v.3.30.13 (Ritchie et al., 2015)]. Afterward, menstrual cycle effect was evaluated through principal component analysis plots drawn with the ggplot2 R package v. 3.2.0 (Wickham, 2016). The proportion of endometrial biopsies collected at different stages of the menstrual cycle were compared between case and control groups using Fisher’s exact test (Fisher, 1922) implemented in the R environment (R Core Team, 2016).\\nMenstrual cycle effect correction and differential expression analysis\\nThe effect of menstrual cycle progression on endometrial biopsy collection was removed from gene expression data using the removeBatcheffect function based on linear models implemented in the limma R package v.3.30.13 (Ritchie et al., 2015) because this function enables correcting the bias from a known batch effect (e.g., menstrual cycle) while indicating the group differences to be retained in the data (e.g., uterine disorder versus control). Specifically, we used removeBatcheffect for being a slightly safer option than Combat (Espín-Pérez et al., 2018), specifying the menstrual cycle phase of endometrial biopsy collection as the batch to remove, and defining the design matrix in relation to the condition to be preserved (case versus control samples).\\nCase versus control differential expression analyses were applied (limma R package v.3.30.13 (Ritchie et al., 2015)) with and without removing the menstrual cycle effect; the proportion of differentially expressed genes (false discovery rate; FDR < 0.05) were compared for demonstrating the menstrual cycle bias.\\nTo highlight the advantage of this method and validate its statistical power, an alternative approach was compared evaluating case versus control DEGs within each menstrual cycle phase using the limma R package v.3.30.13 (Ritchie et al., 2015). The proportion of DEGs (FDR < 0.05) obtained was compared with those obtained after menstrual cycle effect correction. All comparisons of DEG proportions were performed with one-sided Fisher’s exact test [adjusted by FDR (Benjamini and Hochberg, 1995)]. The statistical power of both methods was calculated with sizepower R package v.1.52.0 (Qiu et al., 2018).\\nAdditionally, to validate the reliability and robustness of the method used to remove the menstrual cycle effect, the aforementioned approaches of differential expression analysis (with and without menstrual cycle correction) were applied to GEO individual studies comparing endometrial gene expression profiles between different menstrual cycle phases. If the method properly removed the menstrual cycle effect from transcriptomic data, the differential expression analysis between endometrial phases after correcting this effect should indicate no DEGs. The study design is detailed in Fig. 1A.\\nFinally, we compared the fold change (FC), P-value, gene expression average, and standard deviation (SD) before and after menstrual cycle effect correction for understanding the aetiology of the potential endometrial biomarkers. All statistical analyses were run under R software v.3.3.2 (2016-10-31) (R Core Team, 2016).',\n",
       "  'Results': 'Current practices for transcriptomic studies evaluating the endometrium in uterine disorders\\nOf the endometrial studies found in GEO (n\\u2009=\\u2009694), 35 studies met the inclusion criteria (Fig. 1B). Of these, 31.43% (11 studies) did not register the menstrual cycle phase at the time of endometrial biopsy collection and 37.14% (13 studies) collected all endometrial samples at only the proliferative or secretory phases, with no further subdivision for secretory samples into early-, mid- or late-secretory endometrial stages (Fig. 1B). Only six papers with seven case versus control transcriptomic studies reporting the time point in which the biopsy was collected were suitable for analysis: two of eutopic endometrium in stages I–IV endometriosis (n\\u2009=\\u200937, n\\u2009=\\u200981), one of ectopic endometrium in ovarian endometriosis (n\\u2009=\\u200914), one of uterine fibroids (n\\u2009=\\u200943), two of RIF (n\\u2009=\\u2009115, n\\u2009=\\u200918) and one of RPL (n\\u2009=\\u200920) (Table IA; detailed filtering steps in Fig. 1B).\\nWhile studies evaluating endometriosis and uterine fibroids mainly used Noyes histopathological criteria and collected biopsies along both the proliferative and secretory phases (Table IA), those evaluating RIF and RPL used days from LH peak or first administration of human chorionic gonadotropin (hCG) for endometrial biopsies collected in the secretory phase (Table IA). For all studies, case and control groups were balanced in terms of the proportion of samples collected at each endometrial stage (P\\u2009>\\u20090.05; Table IB).\\nMenstrual cycle effect on transcriptomic studies searching for uterine disorder biomarkers\\nAll studies collecting endometrial biopsies at different cycle time points had a menstrual cycle effect on gene expression (see PCAs on Figs 2 and 3).\\nSamples from Burney 2007, Hawkins 2011, Tamaresis 2014 and Koot 2016 were grouped by endometrial phase before menstrual cycle effect correction (Fig. 2). However, samples were mainly grouped by the uterine disorder (Fig. 2) and a significant average of 44.19% more biomarkers were detected (one-sided Fisher’s exact tests FDR ≤ 5.53 × 10–06) when the effect of the menstrual cycle was corrected (Table IIA). These newly revealed uterine disorder biomarkers that were previously masked by the menstrual cycle effect were, in fact, identified despite balanced endometrial cycle stages between case and control groups (Table IB). Masked uterine disorder biomarkers were also detected in Koot 2016, among endometrial biopsies collected at different time points within the mid-secretory phase (LH\\u2009+\\u20095-LH\\u2009+\\u20098) (Table IB).\\nIn contrast, samples from Lucas et al., 2016 and Pathare 2017 were primarily grouped by the disorder rather than by the time point of endometrial biopsy collection before menstrual cycle effect correction; and no more DEGs were identified after correcting the menstrual cycle effect (Table IIA, Fig. 3).\\nWhile in Burney 2007, Hawkins 2011 and Koot 2016, the DEGs detected before menstrual cycle correction where included in those identified after the correction was applied. For Tamaresis 2014, there were DEGs detected only before applying the menstrual cycle correction (Table IIA).\\nA new classification of endometrial transcriptomic biomarkers in uterine disorders\\nComparison of the expression profiles of significant genes identified before and after applying the menstrual cycle correction allowed us to detect their aetiology and define a new classification of endometrial biomarkers (Table IIA, Fig. 4).\\nMenstrual cycle biomarkers (Fig. 4A) are genes detected only before menstrual cycle correction, suggesting that they respond to endometrial progression but not to the uterine disorder itself (higher expression differences were observed between endometrial stages than between cases and controls). This type of biomarker was identified by Tamaresis 2014, where a gene is differentially expressed without being truly affected by the uterine disorder (Fig. 4A). After correcting for the menstrual cycle, the distance between gene expression patterns in different endometrial phases shortened and became non-significant.\\nUterine pathology biomarkers not masked by the menstrual cycle (Fig. 4B) are genes detected with and without menstrual cycle correction because there was no effect of endometrial progression (Fig. 4.B.1, no expression differences were observed between endometrial stages) or the effect was lower than that of the uterine disorder (Fig. 4.B.2, higher expression differences between cases and controls than between endometrial stages).\\nUterine disorder biomarkers masked by the menstrual cycle (Fig. 4C) are genes detected only after menstrual cycle correction because the effect of endometrial progression greatly outweighs that of the uterine disorder (higher expression differences were observed between endometrial phases than between cases and controls). Consequently, these genes remain masked and not significant before menstrual cycle correction and can only be detected as uterine disorder biomarkers after removal of menstrual cycle effect.\\nFor the three types of endometrial biomarkers, the menstrual cycle effect correction method only affected the variability of gene expression explained by endometrial progression, thus changing the P-value when comparing case and control groups (Fig. 5). In contrast, fold changes between cases and controls did not substantially change in any gene belonging to any type of endometrial biomarker (Fig. 5), indicating that the correction method successfully maintained the expression differences associated with uterine disorders. Consequently, P-value changes were not caused by alteration in case versus control mean expression differences but by the removal of menstrual cycle induced variation in gene expression (Fig. 5).\\nDiscovery of new potential biomarkers\\nAmong the type C uterine disorder biomarkers that remained undetected before correcting the menstrual cycle effect on gene expression (Fig. 4C, Supplementary Table SII) (Yu et al., 2010; Stelzer et al., 2016; Piñero et al., 2020), we discovered 544 new candidate biomarkers of eutopic endometriosis (Burney 2007), 158 of ectopic ovarian endometriosis (Hawkins 2011) and 27 of recurrent implantation failure (Koot 2016) that had not been previously reported (Table IIB and Supplementary Table SII) (Burney et al., 2007; Yu et al., 2010; Hawkins et al., 2011; Koot et al., 2016; Stelzer et al., 2016; Piñero et al., 2020). These new biomarkers presented an expression difference between cases and controls of 12–121% for eutopic endometriosis, 15–359% for ectopic ovarian endometriosis and 2–11% for RIF (Supplementary Table SII).\\nTo better understand their functional role in the context of uterine disorder pathophysiology, these new biomarkers were functionally annotated (Fig. 6, Supplementary Table SIII). New candidate endometriosis biomarkers (both ectopic and eutopic) were mainly related to metabolism (19 ectopic and 73 eutopic), transcription regulation processes (14 ectopic and 47 eutopic) and protein-modification processes (14 ectopic and 47 eutopic) (Fig. 6A). In addition, functions widely reported to be involved in endometriosis such as immune response and inflammation (Tomassetti et al., 2006; Burney et al., 2007; Burney and Giudice, 2012; Liu et al., 2015; Patel et al., 2018; Anderson, 2019; Marquardt et al., 2019; Devesa-Peiro et al., 2020) and cell differentiation and development (Tomassetti et al., 2006; Burney et al., 2007; Burney and Giudice, 2012; Crispi et al., 2013; Liu et al., 2015; Patel et al., 2018; Marquardt et al., 2019; Zhang et al., 2019; Devesa-Peiro et al., 2020) were notably annotated to these new potential biomarkers of endometriosis (Fig. 6A). New RIF candidate biomarkers were mostly associated with transcription regulation (3 biomarkers: CHD4, IGHMBP2, ZBTB48), post-transcriptional changes (2 biomarkers: FAM182B and RN7SK), and epigenetics and chromatin remodelling processes (2 biomarkers: CDH4, FAM182B); these functions were previously reported as significantly altered in RIF patients (Cakmak and Taylor, 2011; Koot et al., 2016; Pathare et al., 2017; Devesa-Peiro et al., 2020) (Fig. 6B). Gene names of new candidate biomarkers of ectopic/eutopic endometriosis and RIF belonging to each functional group are listed in Supplementary Table SIII together with literature references supporting the role of each functional group in the context of each uterine disorder.\\nGenes obtained from Tamaresis 2014 were not used to report newly discovered biomarkers because their samples were separated into two groups according to an unknown effect that we were not able to associate with any technical, biological or clinical registered variable in the original study. This unknown effect could be responsible for the remarkably large number of DEGs obtained both before and after applying the correction of the menstrual cycle effect (Table IIA). Considering all this, we cannot assess how this unknown effect on gene expression may impact the results.\\nComparison of the proposed menstrual cycle effect correction method versus transcriptomic analyses within each menstrual cycle phase for the identification of uterine disorder biomarkers in endometrium\\nA significantly lower proportion of DEGs (FDR\\u2009<\\u20090.05) was obtained when analysis was performed independently for each menstrual cycle phase compared to when correcting for the menstrual cycle effect on gene expression (Fisher’s exact test FDR\\u2009<\\u20092.2 × 10−16; Table IIC). Indeed, significant genes from Burney 2007 and Hawkins 2011 were only detected in the early secretory phase and in the proliferative phase, respectively (Table IIC). As expected, due to the reduction in sample size, the statistical power was lower for the per menstrual cycle phase analyses compared to the menstrual cycle correction method (Table IIC).\\nValidation of the menstrual cycle effect correction method\\nTo check the robustness and reliability of the method used to remove the menstrual cycle effect on gene expression, we applied the aforementioned approaches of differential expression analysis (with and without menstrual cycle correction) to five independent endometrial transcriptomic studies that compared endometrial gene expression profiles across different menstrual cycle phases (Table IIIA). Three studies used the LH peak for endometrial dating (Bradley 2010, Altmae 2017, Sigurgeirsson 2016), one used histopathological criteria of Noyes et al., 1975 (Talbi 2006), and the other did not report the dating methodology (Kelleher2017). For the five evaluated studies, samples were mainly grouped by the menstrual cycle phase according to principal component analysis (Supplementary Fig. S1). Consequently, we identified significantly DEGs between endometrial phases in all of them before applying the menstrual cycle effect correction (Table IIIB). However, samples were no longer grouped by endometrial phase (Supplementary Fig. S1) and no DEGs were obtained between the distinct endometrial phases after the menstrual cycle effect was removed (Table IIIB), demonstrating that the correction worked as expected.',\n",
       "  'Discussion': 'This study is the first to assess the current practices in identifying transcriptomic biomarkers of uterine disorders in endometrium, especially in relation to the menstrual cycle phase of endometrial biopsy collection. We found that one third of the studies did not report the menstrual cycle phase of the samples, including all of the suitable studies evaluating endometrial adenocarcinoma, leiomyosarcoma and adenomyosis. Other studies (37.14%) collected endometrial biopsies within the same menstrual cycle phase but with no further sub-classification in early-, mid- or late-secretory stages was made. Noyes histopathological criteria (Noyes et al., 1975) is one of the most utilised methods for endometrial dating, even though endometrial transcriptomics is superior in both accuracy and reproducibility (Díaz-Gimeno et al., 2013).\\nOur description of current practices demonstrated that, although it is widely known that the menstrual cycle progression affects most of the genes expressed in endometrium (Talbi et al., 2006; Koot et al., 2016; Diaz-Gimeno et al., 2017; Sebastian-Leon et al., 2018; Saare et al., 2019), new guidelines are needed for avoiding the menstrual cycle bias in transcriptomic analysis, and a more in-depth registration and consideration of endometrial stage is required.\\nMenstrual cycle correction enabled identification of an average of 44.19% more uterine disorder biomarkers that would have otherwise remained undiscovered. This phenomenon held true regardless of the endometrial dating method or whether the endometrial biopsies were collected along the entire menstrual cycle (Burney 2007, Hawkins 2011, Tamaresis 2014) or only within the secretory phase (Koot 2016). The highest evidence was shown when the menstrual cycle effect correction was needed even if the study design included samples balanced across the cycle between case and control groups. Our findings suggest that the current practice for avoiding menstrual cycle bias in transcriptomic studies is unable to prevent endometrial progression from masking potential uterine disorder biomarkers. In addition, our results showed biopsies collected at the secretory phase must be further subdivided into early-, mid- and late-secretory stages to correct for menstrual cycle bias.\\nOne of the limitations of this study is that it depends on publicly available datasets. Therefore, the effect of the menstrual cycle on biomarker discovery could only be evaluated in a limited number of uterine disorders. However, the effect of the menstrual cycle was present in all the evaluated conditions; thus, this effect is likely also present in other endometrial pathologies not included here. Recently, Suhorutshenko and colleagues demonstrated that endometrial receptivity biomarkers are biased by the distinct proportions of stromal and epithelial cells within the collected endometrial biopsies (Suhorutshenko et al., 2018). Here, we demonstrate that menstrual cycle progression biases the biomarker search and if the proportion of cell types described by Suhorutshenko and colleagues is inherent to menstrual cycle progression, this produces a cycle-based bias rather than a technical bias in biopsy collection as has been suggested (Suhorutshenko et al., 2018). We also demonstrate that menstrual cycle progression affects the discovery of endometrial transcriptomic biomarkers and call for an assessment of best practices of endometrial transcriptomic analysis.\\nWe propose a novel classification of endometrial biomarkers for gene expression studies evaluating uterine disorders. This new classification identifies biomarkers depending on the aetiology of gene expression changes, distinguishing between menstrual cycle biomarkers, uterine disorder biomarkers not masked by the menstrual cycle (which are sub-classified as genes whose expression is not affected by the menstrual cycle and genes with a menstrual cycle effect but in which the phase-dependent expression changes is less than those explained by the uterine disorder) and uterine disorder biomarkers masked by the menstrual cycle (identified after the menstrual cycle effect is corrected). This latter type of endometrial biomarker is likely to remain undetected under current practices of transcriptomic studies that do not control for menstrual cycle bias. Using this methodology, we unveiled new potential biomarkers: 544 for eutopic endometriosis, 158 for ectopic ovarian endometriosis and 27 for recurrent implantation failure, all of which had not been previously reported in the included studies (Burney et al., 2007; Hawkins et al., 2011; Tamaresis et al., 2014; Koot et al., 2016) or other studies consulted through Disgenet (Piñero et al., 2020), Phenopedia (Yu et al., 2010) and GeneCards (Stelzer et al., 2016) databases. These new candidate biomarkers were involved in functions known to be altered by the corresponding uterine disorder, such as immune response and inflammation, cell differentiation and development in endometriosis (Tomassetti et al., 2006; Burney and Giudice, 2012; Crispi et al., 2013; Tamaresis et al., 2014; Liu et al., 2015; Patel et al., 2018; Anderson, 2019; Marquardt et al., 2019; Zhang et al., 2019; Devesa-Peiro et al., 2020) or epigenetics and transcription and post-transcription regulation in RIF (Cakmak and Taylor, 2011; Koot et al., 2016; Pathare et al., 2017; Devesa-Peiro et al., 2020), highlighting their relevance in the pathophysiology of the disease.\\nOur method of menstrual cycle correction proved to be robust and reliable, as samples were no longer grouped by the endometrial phase and we did not identify any DEGs between distinct menstrual cycle phases after we applied this correction to studies evaluating menstrual cycle changes in women with normal endometrium. This was consistent regardless of the statistical method employed for differential expression analysis (ANOVA or pairwise comparisons) and/or endometrial dating method. In addition, the correction method maintained the average differences between case and control samples for studies evaluating uterine disorders, demonstrating that the effect of the condition was not removed in the correction process and that the observed changes in the P-values were explained only by the removal of the gene expression variability explained by menstrual cycle progression.\\nWhen comparing our approach of uterine disorder biomarker detection with those followed in the included studies, we found that only Koot and colleagues corrected for menstrual cycle effect also using linear models (Koot et al., 2016). From the remaining five studies, Burney and colleagues and Tamaresis and colleagues addressed menstrual cycle bias by dividing samples according to the menstrual cycle phase and performing an independent differential expression analysis at a probeset level (Burney et al., 2007; Tamaresis et al., 2014). Unlike our proposed correction method, this strategy allows identification of phase-specific uterine disorder biomarkers (e.g., genes whose expression only differs significantly between cases and controls in the proliferative phase but not in the secretory phase). Although identifying phase-specific uterine disorder biomarkers is useful in understanding the relationship between the disorder and the menstrual cycle, we demonstrated that this strategy retrieves significantly fewer potential uterine disorder biomarkers compared to menstrual cycle effect correction, as it sacrifices statistical power due to lower sample sizes. In contrast, our proposed correction method identifies uterine disorder biomarkers regardless of menstrual cycle phase during biopsy. Notably, removing the menstrual cycle effect does not impede identifying potential biomarker genes that are responding to both the menstrual cycle and the uterine disorder. In fact, those genes more greatly influenced by the menstrual cycle than by the uterine disorder are the ones that the correction method can unmask.\\nConsidering these findings, we define new guidelines for the detection of reliable uterine disorder biomarkers according to distinct scenarios. If endometrial biopsies are collected at different stages of the menstrual cycle, the menstrual cycle effect must be always corrected in the transcriptomic analysis as endometrial timing is masking genes whose expression is affected by the uterine disorder. In unbalanced studies in which the menstrual cycle stage was not corrected, we expected an additional risk of identifying genes as uterine disorder biomarkers whose expression is indeed dependent on the menstrual cycle and not on the evaluated condition. Therefore, applying the correction for the menstrual cycle in these studies is even more crucial. We observed this in Tamaresis 2014, the only study in which menstrual cycle biomarkers were identified and where the endometriosis samples were mostly secretory and the control samples proliferative. However, this hypothesis needs further confirmation, as this study presented an unknown effect on gene expression.\\nSingle-cell studies are increasingly used to evaluate endometrial gene expression changes throughout the menstrual cycle, increasing our molecular understanding of endometrial function and receptivity acquisition (Wang et al., 2020). Although further studies are needed, we could expect from our results that the menstrual cycle effect will also need to be corrected in single-cell studies aimed to identify biomarkers of uterine disorders on specific cell types and in which samples are collected at different endometrial stages. Although the correction method proposed in this study was previously applied to single-cell studies (Tran et al., 2020), other methodologies have recently arisen to specifically correct known effects in this type of gene expression data (Haghverdi et al., 2018; Tran et al., 2020).\\nIn-depth registration and consideration of endometrial stage is needed in any transcriptomic study to optimise the detection of reliable biomarkers of uterine disorders. Here, we introduce a novel classification of endometrial transcriptomic biomarkers depending on the DEG aetiology and set new guidelines to accurately detect uterine disorder biomarkers through differential expression analysis with high reproducibility and statistical power. Using these methods, we unmasked new endometriosis and RIF potential biomarkers to improve diagnosis, prognosis and treatment. The application of these methods in future research on biomarker discovery of uterine disorders would further contribute to delineating their aetiology and progression, and ultimately leading towards improved treatments and increased pregnancy rates in these patients.',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1093/molehr/gaab011',\n",
       "  'Journal': 'Mol Hum Reprod'},\n",
       " {'PaperTitle': 'Differences in the proteomic profiles of the eutopic endometrium in patients with internal and external adenomyosis',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Marta Valdés-Bango, Mikel Azkargorta, Meritxell Gracia, Cristina Ros, Mariona Rius, María-Angeles Martínez-Zamora, Cristian De Guirior, Lara Quintas, Félix Elortza, Francisco Carmona',\n",
       "  'Email': 'martavaldesbc@gmail.com',\n",
       "  'Abstract': 'Two separate phenotypes of adenomyosis have been recognized, determined by the anatomical position of the adenomyotic lesions within the myometrium. This suggests that adenomyosis impacting the inner myometrium and that affecting the outer myometrial layer may have distinct origins and display different clinical and radiological characteristics. We aimed to investigate the endometrial proteomic profiles of patients with both adenomyosis phenotypes to identify differentially expressed proteins and molecular pathways, shedding light on their distinct pathogenic mechanisms. We conducted a cross-sectional study that included thirty-six participants (nine with internal adenomyosis, nine with external adenomyosis, and eighteen healthy controls based on sonographic criteria) from September 2021 to September 2022. Endometrial samples were collected and processed for proteomic analysis. Mass spectrometry and a Data Independent Acquisition strategy were used to identify differentially expressed proteins. Gene Ontology and Ingenuity Pathway Analysis were employed for further functional analysis and network generation. The proteomic profiles of the eutopic endometrium differed significantly among women with internal adenomyosis, external adenomyosis, and controls. Biological functions related to the innate immune response were affected by differentially expressed proteins in patients with both phenotypes of adenomyosis compared to controls. The proteomic profiles of the endometrium of women with external versus internal adenomyosis exhibited significant differences, with external adenomyosis showing a heightened immune response and inflammatory activity, while internal adenomyosis was associated with altered signaling pathways related to cell migration and apoptosis. Upstream regulator analysis predicted the activation of inflammatory mediators like LPS, TGF-β1, IL-4, and IFN-γ in external adenomyosis, and MAPK1 and IRF2BP2 along with multiple microRNAs in internal adenomyosis. Overall, our findings support distinct pathogenic mechanisms for the two adenomyosis phenotypes, highlighting the need for further research to explore their implications for diagnosis, correlation with symptoms, and new potential therapeutic strategies.',\n",
       "  'Introduction': 'Adenomyosis is a common benign gynecological condition causing abnormal uterine bleeding, dysmenorrhea, chronic pelvic pain, as well as infertility [1]. Histologically, adenomyosis is defined by the presence of ectopic endometrial glands and stroma surrounded by hypertrophic and hyperplastic myometrium [2]. Recent studies have defined two distinct phenotypes of adenomyosis based on the location of adenomyotic lesions within the myometrium: either predominantly in the inner layer (internal adenomyosis) or more confined to the outer layer (external adenomyosis) [3–7]. Patients with external and internal adenomyosis phenotypes present radiological and clinical-demographic differences [4, 7]. Although the exact etiology of the disease remains debated, some authors suggest that internal adenomyosis results from endometrial invasion or de novo from metaplasia of embryonic or adult stem cell transformation within the myometrium, while external adenomyosis may occur through invasion from adjacent endometriotic lesions or from differentiation of endometrial stem cells deposited during retrograde menstruation [5, 6, 8–13].\\nMoreover, it is known that patients with adenomyosis present an immune dysregulation leading to chronic inflammation as well as imbalances in steroidal hormones [14]. Previous investigations have shown the existence of intrinsic abnormalities in the eutopic endometrium of patients with adenomyosis and neurogenesis, angiogenesis, fibrosis, cell proliferation and migration, impaired apoptosis and oxidative stress have been recognized as important contributing pathogenic factors [14–17]. In this context, we could hypothesize that the eutopic endometrium of patients with internal and external adenomyosis presents distinct abnormalities and dysregulated pathways.\\nIn the past few years, omics analyses have emerged, offering insights into biological pathways involved in diverse medical conditions, while also aiding in the identification of potential diagnostic and prognostic biomarkers. In this context, various transcriptomic and proteomic studies have been performed to investigate potential mechanisms underlying the pathogenesis and pathophysiology of adenomyosis [18–31].\\nConsidering this, the present study aimed to compare the endometrial proteomic profiles of patients with adenomyosis and controls and to identify differentially expressed proteins (DEPs) and molecular pathways in the eutopic endometrium from patients with internal and external adenomyosis phenotypes at proteomic level, providing new insights into the potentially differing pathogenic mechanisms of these two phenotypes of the disease.',\n",
       "  'Methods': 'Ethics statement\\nEthical approval was obtained from the Ethics Committee for Biomedical Research - Hospital Clínic de Barcelona (Approval Number: HCB/2021/0242). All the study participants provided written informed consent before the study started.\\nThis observational cross-sectional study was performed in a tertiary referral center between 1st September 2021 and 1st September 2022. Patients were included in the study according to the inclusion and exclusion criteria specified below. Nine patients with sonographic diagnosis of internal adenomyosis and 9 with external adenomyosis were included in the study. Control group included 18 asymptomatic women who were visited for routine screening with confirmation for normal uterus and adnexa based on sonographic examination. The exclusion criteria were as follows: (1) postmenopausal status (2) previous diagnosis of autoimmune, inflammatory and/or neoplastic diseases, (3) use of hormone treatment over the last 3 months before sample collection (4) use of antibiotic or pre or probiotics in the 3 months prior to sample collection, (5) presence of myomas or polyps in the ultrasound examination, (6) current menstruation.\\nFor the ultrasound evaluation, we used the same methodology described in our previously published works [7, 32]. Adenomyosis was diagnosed according to the following criteria established by the Morphological Uterus Sonographic Assessment (MUSA) group [33, 34]: myometrial cysts, myometrial hyperechoic islands and echogenic subendometrial lines and buds (direct criteria); and asymmetrical thickening of uterine walls, fan-shaped shadowing, translesional vascularity, irregular or interrupted junctional zone (JZ) and globular uterus (indirect criteria). Adenomyosis was diagnosed in the presence of two or more criteria, with at least one being direct. To distinguish between internal and external adenomyosis we employed the following sonographic categorization of the myometrium into three layers: the inner myometrium (corresponding to the JZ), the middle myometrium extending from the JZ to the venous and arterial arcuate vessels of the uterus, and the outer myometrium located between the arcuate vessels and the uterine serosa. The vascular arcuate, observed in the sagittal plane in two-dimensional transvaginal ultrasound (2D TVUS) with the application of color or power Doppler, served as a reference for the differentiation of uterine layers [35]. Haga clic o pulse aquí para escribir texto. Internal adenomyosis was defined when the JZ (inner myometrium) was affected with or without involvement of the middle myometrium and without affecting the outer myometrium (subserosal layer). External adenomyosis, on the other hand, was considered when it involved the outer myometrium, with or without involvement of the middle myometrium, without altering the JZ. The presence of ovarian and deep endometriosis (DE) was also assessed according to the International Deep Endometriosis Analysis (IDEA) group consensus [36]. For each patient, the following socio-demographic and clinical data were collected: age, ethnicity, body mass index (BMI), infertility, nulliparity. Dysmenorrhea and non-cyclic chronic pelvic pain (NCCPP) were also assessed using a numerical rating scale (NRS), where 0 represented no pain and 10 indicated unbearable pain. Additionally, the presence of heavy menstrual bleeding (HMB) was evaluated based on subjective patient assessments [37].\\nSample collection and processing\\nWomen were scheduled for sample collection between days 3 and 10 of their menstrual cycle, with the proliferative phase confirmed by ultrasound. For sample collection, a sterile speculum was used, and the vagina and external cervix were swabbed with chlorhexidine. Endometrial samples were then collected using a double sheathed, sterile endometrial suction curette that was passed through the cervix to collect an endometrial biopsy, taking care to avoid contact with the vaginal wall and cervix. Biopsies were submerged in 1.0 mL of NaCl (9 mg/mL). Samples were transported to the laboratory on ice and processed within one hour. Endometrial samples were centrifuged at 3000 g at 4 °C for 10 minutes, and pellets were stored at −80 °C until analysis. Pellets were stored for up to six months, from collection and initial processing, until all participant samples were collected, and final analysis conducted.\\nIn-solution tryptic digestion\\nProtein was extracted by incubating the sample in a buffer containing 7M urea, 2M thiourea, and 4% CHAPS. Samples were incubated in this buffer for 30 min at room temperature under agitation and digested following the filter aided sample preparation (FASP) protocol described by Wisniewski et al. [38] with minor modifications. Protein was quantified using the Bio-Rad Protein Assay Dye Reagent Concentrate colorimetric assay (Bio-Rad), following manufacturer’s instructions, and approximately 20 µg per sample were processed. Trypsin was added in 50mM ammonium bicarbonate to a trypsin:protein ratio of 1:10, and the mixture was incubated for overnight at 37oC. Peptides were dried out in an RVC2 25 speedvac concentrator (Christ) and resuspended in 0.1% FA. Peptides were desalted and resuspended in 0.1% FA using C18 stage tips (Millipore, St. Louis, MO, United States) with a maximum peptide load of 2 µg prior to acquisition.\\nMass spectrometry analysis\\nSamples were analyzed in a timsTOF Pro with PASEF (Bruker Daltonics, Bremen, Germany) coupled online to a Evosep ONE liquid chromatograph (Evosep, Odense, Denmark). Approximately 200ng were directly loaded onto the Evosep ONE and resolved using the 60 samples-per-day protocol.\\nData Independent Acquisition (DIA) data was processed with DIA-NN software v.1.8.1 [39] for protein identification and quantification using default parameters. Searches were carried out against a database consisting of Homo sapiens protein entries from Uniprot in library-free mode. Carbamidomethylation of cysteines was considered as fixed modification and oxidation of methionines as variable modification. Proteins identified with a FDR\\u2009<\\u20091% were considered for further analyses. Data was loaded onto Perseus platform [40] for data processing (log2 transformation, imputation) and statistical analysis (Two-tailed Student’s t-test). Proteins with a p-value<0.05 were considered as DEPs for the corresponding comparison.\\nFunctional analysis and network generation\\nGene Ontology analysis was used to identify possible molecular functions and to visualize the potential biological translation of DEPs. DAVID platform (https://david.ncifcrf.gov/) was used for this purpose.\\nIngenuity pathway analysis (IPA) software (Qiagen, Redwood City, CA) was used for characterization of the molecular events lying behind the differential protein patterns under analysis. In this software, the calculated p-values determine the probability that the association between proteins in the dataset and a given process, pathway or upstream regulator is explained by chance alone, based on a Fisher’s exact test (p-value <0.05 being considered significant). The activation z-score represents the bias in gene regulation that predicts whether the upstream regulator exists in an increased (positive values) or decreased (negative values) activation state, based on the knowledge of the relation between the effectors and their target molecule. Z‐score values\\u2009>\\u20092 or <-2 were considered as significantly activated or inhibited, respectively.\\nStatistical analysis\\nPrinciple component analysis (PCA), Heatmaps and Volcano plots were generated online using the SR Plot resource (https://www.bioinformatics.com.cn/) following the instructions provided in each section.',\n",
       "  'Results': 'Samples and participant characteristics\\nA total of 36 women were included in the study: 9 with internal adenomyosis, 9 in the external adenomyosis group, and 18 in the control group. The baseline socio-demographic and clinical characteristics of included women are shown in Table 1. Sonographic findings of included patients are presented in S1 Table.\\nTable 1. Socio-demographic and clinical characteristics.\\nID\\nAdenomyosis\\nAdenomyosis phenotype\\nCaucasian\\nAge\\nBody Mass Index\\nNulliparity\\nDysmenorrhea\\nNon chronic pelvic pain\\nAbnormal uterine bleeding\\nInfertility\\np1\\nNo\\n–\\nYes\\n33\\n24.7\\nYes\\n0\\n3\\nNo\\nNo\\np3\\nNo\\n–\\nYes\\n42\\n23.4\\nNo\\n2\\n2\\nNo\\nNo\\np11\\nNo\\n–\\nYes\\n38\\n24.9\\nNo\\n2\\n2\\nNo\\nNo\\np12\\nNo\\n–\\nYes\\n28\\n22.9\\nYes\\n1\\n0\\nNo\\nNo\\np13\\nNo\\n–\\nNo\\n31\\n27.9\\nNo\\n0\\n0\\nNo\\nNo\\np14\\nNo\\n–\\nYes\\n45\\n20.9\\nNo\\n0\\n0\\nNo\\nNo\\np15\\nNo\\n–\\nNo\\n34\\n25.4\\nNo\\n2\\n0\\nNo\\nNo\\np16\\nNo\\n–\\nYes\\n38\\n22.7\\nYes\\n5\\n0\\nNo\\nNo\\np25\\nNo\\n–\\nNo\\n43\\n26.3\\nYes\\n6\\n5\\nNo\\nNo\\np31\\nNo\\n–\\nYes\\n35\\n30.0\\nNo\\n3\\n0\\nNo\\nNo\\np32\\nNo\\n–\\nYes\\n31\\n21.3\\nNo\\n2\\n0\\nNo\\nNo\\np35\\nNo\\n–\\nYes\\n39\\n18.4\\nNo\\n2\\n0\\nNo\\nNo\\np36\\nNo\\n–\\nYes\\n40\\n24.7\\nNo\\n7\\n5\\nNo\\nNo\\np38\\nNo\\n–\\nYes\\n29\\n30.5\\nNo\\n3\\n0\\nNo\\nNo\\np39\\nNo\\n–\\nYes\\n44\\n26.7\\nNo\\n2\\n0\\nNo\\nNo\\np40\\nNo\\n–\\nYes\\n31\\n19.5\\nYes\\n7\\n0\\nNo\\nNo\\np46\\nNo\\n–\\nYes\\n44\\n19.3\\nNo\\n6\\n0\\nNo\\nNo\\np101\\nNo\\n–\\nYes\\n44\\n22.7\\nYes\\n5\\n0\\nNo\\nNo\\np56\\nYes\\nInternal\\nYes\\n41\\n28.0\\nYes\\n6\\n8\\nYes\\nNo\\np57\\nYes\\nInternal\\nYes\\n37\\n21.5\\nNo\\n0\\n0\\nNo\\nYes\\np58\\nYes\\nInternal\\nYes\\n38\\n17.9\\nYes\\n10\\n8\\nYes\\nYes\\np71\\nYes\\nInternal\\nYes\\n39\\n26.2\\nYes\\n9\\n8\\nNo\\nYes\\np84\\nYes\\nInternal\\nYes\\n42\\n35.9\\nYes\\n7\\n5\\nNo\\nNo\\np87\\nYes\\nInternal\\nNo\\n41\\n34.5\\nYes\\n4\\n4\\nYes\\nNo\\np88\\nYes\\nInternal\\nNo\\n43\\n20.7\\nNo\\n10\\n4\\nNo\\nNo\\np92\\nYes\\nInternal\\nYes\\n43\\n21.3\\nYes\\n0\\n3\\nYes\\nNo\\np98\\nYes\\nInternal\\nYes\\n42\\n20.6\\nYes\\n6\\n5\\nYes\\nYes\\np52\\nYes\\nExternal\\nYes\\n41\\n20.3\\nYes\\n9\\n9\\nYes\\nNo\\np54\\nYes\\nExternal\\nYes\\n39\\n31.6\\nNo\\n9\\n7\\nYes\\nNo\\np65\\nYes\\nExternal\\nYes\\n42\\n26.4\\nYes\\n2\\n8\\nNo\\nNo\\np66\\nYes\\nExternal\\nYes\\n36\\n26.3\\nNo\\n9\\n0\\nYes\\nYes\\np90\\nYes\\nExternal\\nYes\\n44\\n28.3\\nYes\\n0\\n3\\nNo\\nNo\\np91\\nYes\\nExternal\\nNo\\n32\\n24.6\\nYes\\n10\\n9\\nYes\\nYes\\np95\\nYes\\nExternal\\nYes\\n42\\n19.4\\nYes\\n6\\n0\\nNo\\nNo\\np97\\nYes\\nExternal\\nYes\\n29\\n24.2\\nNo\\n7\\n0\\nYes\\nNo\\np99\\nYes\\nExternal\\nYes\\n30\\n22.0\\nNo\\n10\\n4\\nYes\\nNo\\nOpen in a new tab\\nIdentification of differentially expressed proteins\\nWe conducted a comparative proteomic analysis of eutopic endometrium among women with external adenomyosis, internal adenomyosis, and controls (S1 Data).\\nA total of 281 DEPs were identified between women with external adenomyosis and controls, and 91 DEPs were identified between women with internal adenomyosis and controls (Fig 1).\\nFig 1. Volcano-plots illustrating DEPs between (a) external adenomyosis and controls and (b) internal adenomyosis and controls.\\nOpen in a new tab\\nWhen focusing on adenomyosis phenotypes, 323 DEPs were identified between women with internal adenomyosis and those with external adenomyosis. Among these DEPs, 305 proteins were upregulated, and 18 proteins were downregulated in the eutopic endometrium of patients with external adenomyosis compared to those with internal adenomyosis (Student’s t-test, p\\u2009<\\u20090.05). The results reveal that eutopic endometrium from women with internal and external adenomyosis exhibit significantly different proteomic profiles (Fig 2).\\nFig 2. DEPs between external adenomyosis and internal adenomyosis a) Volcano plot external vs internal b) PCA external vs internal c) Heatmaps external vs internal.\\nOpen in a new tab\\nFunctional analysis of differentially expressed proteins\\nTo further characterize the molecular events underlying different adenomyosis subtypes, a functional analysis of the DEPs detected in this work was performed.\\nFirst, NIH’s DAVID tool was used for a broad determination of the biological and molecular processes these DEPs are mainly involved in. Consistent with the differences detected at the expression proteomics analysis, internal adenomyosis provided fewer enriched terms than external adenomyosis when compared to controls (S2 Data). In addition, these terms were appreciably different, except for GO:0045087 (Innate immune response), common to both internal and external adenomyosis.\\nDAVID analysis of the DEPs expressed in the comparison between internal vs external adenomyosis revealed an enrichment of terms related with complement activation, immunoglobulin production, immune response, zymogen activation and blood coagulation in external adenomyosis among others (Fig 3). In summary, these analyses revealed molecular and functional differences in the eutopic endometrium between internal and external adenomyosis.\\nFig 3. Top 10 significantly enriched Gene Ontology Biological Processes (GO_BP) terms in external adenomyosis versus internal adenomyosis.\\nOpen in a new tab\\nTherefore, we decided to further characterize the molecular events underlying these differences using IPA software. This program uses both public and proprietary knowledge for a deeper and more precise characterization of the molecular events underlying the biological systems under analysis. In addition, the analysis of the protein expression fold change patterns adds an additional level of data interpretation through the Z-score. This parameter infers likely activation or inhibition states of the biological functions/pathways/upstream regulators based on the expression pattern afore mentioned. The IPA results for the different analytical categories considered (Canonical Pathways and Upstream Regulators) are summarized in the S3 Data.\\nFig 4 displays the most significantly upregulated and downregulated canonical pathways associated with DEPs in external versus internal adenomyosis. In external adenomyosis, key immune-related pathways were significantly activated, including the “Complement Cascade” (−log(p-value)=27.5; z-score\\u2009=\\u20093.6), “Fc gamma Receptor Dependent Phagocytosis” (−log(p-value)=22.9; z-score\\u2009=\\u20095.3), and “Neutrophil Degranulation” (−log(p-value)=22.7; z-score\\u2009=\\u20096.6); whereas pathways such as “Interleukin 12 (IL-12) Signaling and Production in Macrophages” (−log(p-value)=7.8; z-score\\u2009=\\u2009-3,2); “Programmed cell death protein 1 PD-1)/ Programmed cell death ligand 1 (PD-L1) Pathway” (−log(p- value)=4.1; z-score\\u2009=\\u2009-2.6); and “ Rho GDP-dissociation inhibitor (RhoGDI) Signaling” (−log(p-value)=2.6; z-score\\u2009=\\u2009-2.4) were significantly inhibited.\\nFig 4. Significantly activated (Z score>2) or inhibited (Z score\\u2009<-2) canonical pathways associated with DEPs in external vs internal adenomyosis.\\nOpen in a new tab\\nRegarding significant differences among adenomyosis phenotypes in predicted upstream regulators (Tables 2 and 3), IPA identified notable activation of upstream regulators such as lipopolysaccharide (LPS) (p-value\\u2009=\\u20092.39e-21, z-score\\u2009=\\u20095.1), Transforming Growth Factor Beta 1 (TGFB1) (p-value\\u2009=\\u20091.19e-10, z-score\\u2009=\\u20092.7), Interleukin 4 (IL-4) (p-value\\u2009=\\u20095.84e-10, z-score\\u2009=\\u20094.0), and Interferon Gamma (IFGN) (p-value\\u2009=\\u20091.15e-07, z-score\\u2009=\\u20093.6), and inhibition of others such as Mitogen-Activated Protein Kinase 1 (MAPK1) (p-value\\u2009=\\u20092.03e-02, z-score\\u2009=\\u2009-2.3), Interferon regulatory factor 2-binding protein 2 (IRF2BP2) (p-value\\u2009=\\u20092.23e-02, z-score\\u2009=\\u2009-2.2), and multiple microRNAs in external adenomyosis relative to internal adenomyosis.\\nTable 2. Top 25 activated upstream regulators in external vs internal adenomyosis.\\nUpstream Regulator\\nMolecule Type\\nPredicted State\\nActivation z-score\\np-value of overlap\\nNumber of Target Molecules in Dataset\\nTGFB1\\nGrowth factor\\nActivated\\n2,65\\n1,19E-10\\n58\\nIL4\\nCytokine\\nActivated\\n4,031\\n5,84E-10\\n45\\nCEBPA\\nTranscription regulator\\nActivated\\n2,123\\n1,69E-08\\n23\\nIFNG\\nCytokine\\nActivated\\n3,569\\n1,15E-07\\n46\\nIL2\\nCytokine\\nActivated\\n2,188\\n3,03E-07\\n29\\nIL5\\nCytokine\\nActivated\\n3,925\\n5,09E-07\\n16\\nTNF\\nCytokine\\nActivated\\n3,892\\n8,58E-07\\n53\\nCD38\\nEnzyme\\nActivated\\n2,939\\n9,52E-07\\n12\\nHNF4A\\nTranscription regulator\\nActivated\\n2,308\\n1,24E-06\\n53\\nIL1B\\nCytokine\\nActivated\\n2,229\\n4,27E-06\\n34\\nBCR (complex)\\nComplex\\nActivated\\n3,45\\n6,26E-06\\n15\\nKLF6\\nTranscription regulator\\nActivated\\n2,545\\n6,95E-06\\n14\\nNFKB1\\nTranscription regulator\\nActivated\\n2,779\\n8,79E-06\\n15\\nCSF1\\nCytokine\\nActivated\\n2,359\\n9,20E-06\\n17\\nSTAT1\\nTranscription regulator\\nActivated\\n3,607\\n2,50E-05\\n17\\nSMARCA4\\nTranscription regulator\\nActivated\\n2,714\\n2,86E-05\\n22\\nHNF1A\\nTranscription regulator\\nActivated\\n2,537\\n3,08E-05\\n19\\nCD40\\nTransmembrane receptor\\nActivated\\n3,024\\n4,42E-05\\n14\\nRELA\\nTranscription regulator\\nActivated\\n2,553\\n4,77E-05\\n18\\nBrd4\\nKinase\\nActivated\\n2,795\\n5,54E-05\\n8\\nBHLHE40\\nTranscription regulator\\nActivated\\n3,208\\n6,11E-05\\n14\\nKAT2B\\nEnzyme\\nActivated\\n2,415\\n6,14E-05\\n6\\nCCL20\\nCytokine\\nActivated\\n2,236\\n1,42E-04\\n5\\nCEBPD\\nTranscription regulator\\nActivated\\n2,407\\n1,99E-04\\n8\\nSREBF1\\nTranscription regulator\\nActivated\\n2,387\\n2,04E-04\\n12\\nOpen in a new tab\\nTable 3. Top 25 inhibited upstream regulators in external vs internal adenomyosis.\\nUpstream Regulator\\nMolecule Type\\nPredicted State\\nActivation z-score\\np-value of overlap\\nNumber of Target Molecules in Dataset\\nSKIC2\\nEnzyme\\nInhibited\\n−2,828\\n5,72E-08\\n8\\nETV6-RUNX1\\nFusion gene/product\\nInhibited\\n−2,103\\n1,31E-04\\n13\\nOVOL2\\nTranscription regulator\\nInhibited\\n−2,222\\n3,27E-04\\n5\\nCLPP\\nPeptidase\\nInhibited\\n−2,433\\n3,77E-04\\n6\\nIRGM\\nEnzyme\\nInhibited\\n−2,425\\n8,03E-04\\n6\\nHIBCH\\nEnzyme\\nInhibited\\n−2,236\\n1,03E-03\\n5\\nmiR-155-5p (miRNAs w/seed UAAUGCU)\\nMature microrna\\nInhibited\\n−2,21\\n1,29E-03\\n8\\nNRAS\\nEnzyme\\nInhibited\\n−2,382\\n1,46E-03\\n9\\nFMR1\\nTranslation regulator\\nInhibited\\n−2,178\\n2,29E-03\\n9\\nTREX1\\nEnzyme\\nInhibited\\n−2,359\\n2,64E-03\\n7\\nFOXA1\\nTranscription regulator\\nInhibited\\n−2,355\\n3,02E-03\\n9\\nmiR-16-5p (and other miRNAs w/seed AGCAGCA)\\nMature microrna\\nInhibited\\n−2,784\\n5,83E-03\\n8\\nmiR-291a-3p (and other miRNAs w/seed AAGUGCU)\\nMature microrna\\nInhibited\\n−2,204\\n6,86E-03\\n5\\nmir-122\\nMicrorna\\nInhibited\\n−2,236\\n9,09E-03\\n5\\nlet-7a-5p (and other miRNAs w/seed GAGGUAG)\\nMature microrna\\nInhibited\\n−2,646\\n1,21E-02\\n7\\nPTPN11\\nPhosphatase\\nInhibited\\n−2,213\\n1,35E-02\\n6\\nPD98059\\nChemical - kinase inhibitor\\nInhibited\\n−2,727\\n1,38E-02\\n14\\nmir-1\\nMicrorna\\nInhibited\\n−2,219\\n1,59E-02\\n5\\nMAPK1\\nKinase\\nInhibited\\n−2,333\\n2,03E-02\\n10\\nIRF2BP2\\nTranscription regulator\\nInhibited\\n−2,219\\n2,23E-02\\n6\\nmiR-124-3p (and other miRNAs w/seed AAGGCAC)\\nMature microrna\\nInhibited\\n−2,619\\n2,68E-02\\n7\\nKDM5A\\nTranscription regulator\\nInhibited\\n−2,449\\n3,13E-02\\n6\\nmir-802\\nMicrorna\\nInhibited\\n−2,236\\n3,32E-02\\n5\\nRORC\\nLigand-dependent nuclear receptor\\nInhibited\\n−2,2\\n3,33E-02\\n6\\nGABA\\nChemical - endogenous mammalian\\nInhibited\\n−3,162\\n3,61E-02\\n10\\nOpen in a new tab',\n",
       "  'Discussion': 'The main findings of the present study are that: the proteomic profile of the eutopic endometrium differs significantly among women with internal adenomyosis, external adenomyosis, and controls; biological functions related to the innate immune response are affected by DEPs in patients with both phenotypes of adenomyosis compared to controls; the proteomic profiles of the endometrium of women with external versus internal adenomyosis also exhibit significant differences; multiple canonical pathways, upstream regulators and biological processes are differently affected in patients with external adenomyosis and those with internal adenomyosis.\\nThe eutopic endometrium of women with adenomyosis (both internal and external) showed an enrichment of proteins related to innate immune response compared to controls. These findings align with data from Liu et al. (2008) [21] and Xiaoyu et al. (2013) [24], which identified abnormally expressed proteins associated with cell adhesion, apoptosis, and immune response in serum samples and adenomyotic lesions. Additionally, our results are consistent with Xiang et al. (2019) [18], who reported altered inflammatory and immune-related pathways (‘IL-6 signaling,’ ‘ERK/MAPK signaling,’ ‘MIF regulation of innate immunity,’ and ‘LPS-stimulated MAPK signaling’) in the eutopic endometrium of adenomyosis patients. More recently, Zhai et al. (2022) [27] identified dysfunction in the inflammatory response, extracellular matrix disassembly, and cell proliferation, along with dysregulated TNF, IL-17, and NF-κB signaling pathways in the endometrium of women with adenomyosis.\\nThe eutopic endometrium from women with internal versus external adenomyosis shows significantly different proteomic profiles. We found an enrichment of proteins related with complement activation (by the alternative and classical pathways), immunoglobulin production and with blood coagulation in external adenomyosis. These results are not comparable with what has been reported in the literature because, to the best of our knowledge, this is the first publication comparing external and internal adenomyosis at a proteomic level.\\nCanonical pathway analysis is useful for understanding the biochemical and signal transduction pathways in which DEPs may participate. In external adenomyosis, key immune-related pathways were significantly activated, suggesting a heightened immune response, and increased inflammatory activity, particularly through complement activation, phagocytosis, and neutrophil involvement, when compared to internal adenomyosis. Previous studies have described an increased abundance and activity of neutrophils in the eutopic proliferative endometrium of patients with endometriosis, as well as elevated levels of IL-8, which may enhance neutrophil recruitment [41]. Neutrophils have been associated with promoting angiogenesis through VEGF production and with immune modulation [41]. Furthermore, the complement system plays a crucial role in recognizing and clearing pathogens and apoptotic and necrotic cells, and in modulating the immune response [42].\\nOn the other hand, in internal adenomyosis, IL-12 signaling may contribute to a persistent inflammatory microenvironment. Dysregulation of RhoGDI in internal adenomyosis, which regulates Rho GTPases involved in cell migration, adhesion, and proliferation and is related to multiple human cancers [43], could enhance the invasive properties of endometrial cells, contributing to their ability to penetrate and grow within the myometrium; interestingly, enhanced RhoGDI and PTEN have been recently described in adenomyosis and have been linked to pregnancy disorders in these patients [26]. Additionally, dysregulation of the PD-1/PD-L1 pathway, also observed in various cancers including endometrial cancer [44], might enable immune evasion, promoting the persistence and progression of the disease.\\nFurthermore, IPA results revealed differences in the predicted activity of upstream regulators when comparing the two phenotypes. Upstream regulators analysis helps to identify regulators that might be driving the observed changes in protein expression in our data set. In external adenomyosis, the analysis predicted the activation of multiple upstream regulators such as LPS, TGFB1, IL4, and IFNG. These mediators have been previously associated with adenomyosis [14] and suggest a heightened inflammatory response and fibrosis, in external adenomyosis. Interestingly, TGFB1, IL4, and IFNG are cytokines also involved in the pathogenesis of DE through their effects on inflammation, tissue remodeling, fibrosis, cellular proliferation, and angiogenesis [45]. Conversely, the analysis predicted a notable activation of the upstream regulators MAPK1, IRF2BP2 and multiple microRNAs in internal adenomyosis compared to external adenomyosis. Xiang et al. had earlier reported the enrichment of the ERK/MAPK signaling pathway in adenomyosis [18]; additionally, the ERK/MAPK pathway has been linked to the proliferation of uterine smooth muscle cells in women with adenomyosis [46]. Recently studies published by Zipponi et al. [30] and Jia et al. [31] also suggest that the dysregulation of multiple microRNAs can influence the progression of adenomyosis through the PI3K/AKT/mTOR pathway, which is involved in cell growth, survival, and proliferation. Moreover, recent data suggest IRF2BP2 involvement in the regulation of several cellular functions, such as the cell cycle, cell death, angiogenesis, inflammation, and immune response modulation [47]. Interestingly, microRNA let-7a-5p and microRNA-124-3p have previously been linked to adenomyosis [48] and chronic endometritis [49], respectively.\\nIn recent years, it has been shown that patients with external and internal adenomyosis exhibit radiological and clinical differences. Although symptoms like bleeding and dysmenorrhea are common in both phenotypes, differences exist in other clinical parameters such as age, parity, infertility and the association with leiomyomas and DE [7, 14, 35, 50]. In a previous study published by our team [7], we found that external adenomyosis is independently associated with the presence of DE and nulliparity, both of which support the theory of an ectopic endometrial origin, potentially exacerbated by retrograde menstruation and inflammatory processes. In contrast, internal adenomyosis, was strongly associated with leiomyomas. These findings, along with the results of our study showing DEPs and molecular pathways in the eutopic endometrium of patients with internal and external adenomyosis phenotypes at the proteomic level, support the hypothesis of distinct etiopathogenic mechanisms for both adenomyosis phenotypes.\\nOur study has several strengths that should be considered: to the best of our knowledge, it is the first study comparing endometrial proteomic profiles between external and internal adenomyosis phenotypes. Additionally, the study was performed in a tertiary referral center where the transvaginal ultrasound evaluations were performed by two expert sonographers with more than 10 years of experience and who have previously demonstrated a high diagnostic accuracy with transvaginal ultrasound for determining the presence of adenomyosis [51, 52]. Moreover, all endometrial samples were collected by the same investigator following a strict protocol to avoid contamination. Our analytical approach leverages the timsTOF ion mobility mass spectrometry, enabling rapid and sensitive sample analysis. The use of DIA enhances sensitivity by addressing some of the limitations of the traditional Data-Dependent Acquisition (DDA) method. However, the approach may still be constrained by the sample’s dynamic range, limited sample size, and analytical power, potentially leading to incomplete protein sampling. Functional analysis can help mitigate these limitations by highlighting key processes or regulators that may be crucial for understanding the observed differences. Another limitation of our study is that it would have also been interesting to include adenomyotic lesions and myometrial samples to better characterize both adenomyosis phenotypes at the proteomic level.',\n",
       "  'Conclusion': 'In conclusion, our results support the presence of abnormalities related to immune response, apoptosis, and cell migration in the eutopic endometrium of women with adenomyosis compared to controls. Furthermore, we demonstrated that the proteomic profile of the eutopic endometrium differs significantly between women with internal and external adenomyosis. Multiple canonical pathways, upstream regulators, and biological processes are differentially affected in these two adenomyosis phenotypes. In the eutopic endometrium of patients with external adenomyosis, a heightened immune response and increased inflammatory activity, particularly through complement activation, phagocytosis, and neutrophil involvement, seem to predominate. In the case of internal adenomyosis, other signaling pathways related to cell migration and apoptosis, along with immune dysregulation, appear to play a more significant role. Further studies are warranted to better understand the implications of these differences and their potential utility in diagnosing and treating adenomyosis.',\n",
       "  'URL': 'https://doi.org/10.1371/journal.pgph.0004785',\n",
       "  'Journal': 'PLOS Glob Public Health'},\n",
       " {'PaperTitle': 'Cytoscape.js: a graph theory library for visualisation and analysis',\n",
       "  'Publication': '2015',\n",
       "  'Authors': 'Max Franz, Christian T Lopes, Gerardo Huck, Yue Dong, Onur Sumer, Gary D Bader',\n",
       "  'Email': 'gary.bader@utoronto.ca',\n",
       "  'Abstract': 'Summary: Cytoscape.js is an open-source JavaScript-based graph library. Its most common use case is as a visualization software component, so it can be used to render interactive graphs in a web browser. It also can be used in a headless manner, useful for graph operations on a server, such as Node.js.\\nAvailability and implementation: Cytoscape.js is implemented in JavaScript. Documentation, downloads and source code are available at http://js.cytoscape.org.\\nContact: gary.bader@utoronto.ca',\n",
       "  'Introduction': 'Network information in biology continues to grow in utility in many contexts, from analysis of cellular mechanisms to identifying disease biomarkers. Further, the web is increasingly a platform for apps with complex user interfaces that use standard technologies such as HTML, CSS and JavaScript (JS). Cytoscape.js provides a JS application programming interface (API) to enable software developers to integrate graphs into their data models and web user interfaces. Cytoscape.js can be used in several domains, such as biological networks or social graphs. Cytoscape.js is the modern replacement for the Adobe Flash-based Cytoscape Web (Lopes et\\xa0al., 2010).',\n",
       "  'Methods': 'Cytoscape.js is implemented as a standalone JS library. It has no hard dependencies; neither browser plugins nor other libraries are required. However, it includes hooks to several useful libraries and environments, including CommonJS/Node.js, AMD/Require.js, jQuery, Bower, npm, spm and Meteor. This allows Cytoscape.js to integrate into a wide variety of JS-based software systems.\\nThe architecture of Cytoscape.js allows it to be run headlessly (i.e. without a graphical user interface) or as a visualisation component (Fig. 1), using HTML5 canvas as its underlying implementation. This allows Cytoscape.js to be run on both the client side—i.e. the browser—and the server-side, an important consideration as JS code is increasingly being shared with the client and the server.\\nFig. 1. Open in a new tab A GeneMANIA gene–gene interaction network automatically laid out and visualised with Cytoscape.js, showing interaction strength (edge thickness), interaction type (colour), multiple edges between nodes, protein score (node size) defined using a stylesheet\\nFor increased ease of use, the library shares several concepts with the HTML\\u2009+\\u2009CSS\\u2009+\\u2009JS web model. Styling in Cytoscape.js is specified using CSS-like stylesheets, sharing as much syntax as possible with CSS. Similarly, graph elements are analogous to HTML DOM elements—they are styled by the stylesheets and programmatically accessible via the JS core API.\\nThe Cytoscape.js architecture is composed of the core and the collection. The core is a developer’s main entry point into the library. It represents the graph and is used to run layouts, alter the view, and perform other operations on the graph as a whole. Core functions are available to access graph elements. Each of these functions returns a collection, a set of graph elements. A collection has its own API that can be used to filter, traverse, perform operations on and get data about the elements in the collection. Some core functions take collections as input.',\n",
       "  'Results': '3 Features\\n3.1 Feature set\\nCytoscape.js features include, but are not limited to, the following:\\nGraph types: Cytoscape.js supports several types of graphs, including traditional graphs, directed graphs, undirected graphs, multigraphs and hypergraphs (with compound nodes, but not yet with hyperedges).\\nMutable graphs: The graph can be manipulated by adding, removing, or modifying the state of graph elements. This enables apps to provide highly interactive graphs for the user.\\nGraph traversal: Graph traversal functions are provided, which are useful for both user interface interactions and programmatic graph analysis.\\nGraph theory algorithms: Several well-known graph theory algorithms—such as connectivity search, shortest path, minimum spanning tree, minimum cut, ranking and centrality measures—are included.\\nStylesheets: Stylesheets are used to specify the visual style of elements. Selectors and classes are supported in order to map element state to style. Similar to the Cytoscape desktop app (Shannon P. et\\xa0al., 2003), a functional mapper syntax is provided to map particular style properties based on element data—e.g. node colour mapped from a numerical weight. Stylesheets can be replaced at runtime, changing the overall visual style of the graph.\\nBuilt-in gesture support for mouse and touch based devices: The default Cytoscape.js renderer supports the touch and mouse gestures a user would expect out-of-the-box. Nodes can be dragged. The user can manipulate the viewport with gestures like pinch-to-zoom and drag-to-pan.\\nEvent binding: Events can be bound in several ways. Delegation can be used such that newly added elements trigger bound events. Bindings can be added and removed, and the multiplicity of elements-to-triggered-events (i.e. for which elements and how many times the handler should be triggered) can be specified when binding. Several higher-level events are provided such that the type of user interaction is abstracted, e.g. ‘tap’ works both with computer mice and finger presses on touch devices.\\nAnimations: Animations can be used to increase the salience of particular elements in the graph and to provide visual continuity to the user when programmatic changes to the graph are made.\\nCompound nodes: As an addition to the traditional graph model, compound nodes are a way for the developer to embed nodes within another node. Compound nodes are useful for representing things like biological complexes and their subunits.\\nImport & export: The graph can be exported as an image (PNG or JPG), including at high resolution for publication. Cytoscape.js supports importing and exporting graphs via JSON, thereby allowing for full serialisation and deserialization of graph state.\\nLayouts: Layouts provide a mechanism for automatically positioning nodes in a graph. Alternatively, the developer may specify pre-determined node positions. Default layouts include null, random, preset, grid, circle, concentric, breadthfirst, dagre, cose, cola, spread, arbor and springy.\\nExtensibility: Cytoscape.js provides mechanisms for the developer to extend its behaviour. For instance, user interface widgets can be built on top of the library—several of these extensions, including contributed layout algorithms, exist. Layouts, core and, collection functions and renderers can be written as extensions to add to the library without needing to modify it directly.\\n3.2 Performance\\nCytoscape.js can comfortably render thousands of graph elements on average hardware. Rendering performance is affected by the visual styles used, the graph size and the web browser client. Rendering performance can be improved by using simpler styles—especially for edges. Optional features that improve the real and user-perceived large graph interaction performance are detailed in the online documentation.\\nCytoscape.js is frugal with respect to rendering: A new frame is rendered only when the view needs to be updated. So, a developer can safely use the API for analysis without worrying about overhead caused by rendering. There is no rendering overhead when using Cytoscape.js headlessly.\\n3.3 Documentation\\nCytoscape.js is actively developed as an open-source project, and is freely available at http://js.cytoscape.org. The documentation includes an in-depth description of the API, runnable code examples and demos.\\n3.4 Example applications\\nExamples of Cytoscape.js use include the ‘Export network as a web page’ feature of the Cytoscape desktop application, ConsensusPathDB (Kamburov A et\\xa0al., 2013), InnateDB (Breuer K et\\xa0al., 2013), BioJS library components (Corpas M, 2014), the Saccharomyces genome database (Costanzo MC et\\xa0al., 2013), the RCyjs Bioconductor package (http://www.bioconductor.org/packages/release/bioc/html/RCyjs.html), the upcoming release of GeneMANIA (Warde-Farley et\\xa0al., 2010), cyNetShare (https://idekerlab.github.io/cy-net-share/), NDEx (http://www.ndexbio.org/), Elsevier (http://www.elsevier.com/books-and-journals/content-innovation/cytoscape) and demos linked to on the Cytoscape.js site (http://js.cytoscape.org/).',\n",
       "  'Discussion': '',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1093/bioinformatics/btv557',\n",
       "  'Journal': 'Bioinformatics'},\n",
       " {'PaperTitle': 'Dysregulated miR-124-3p in endometrial epithelial cells reduces endometrial receptivity by altering polarity and adhesion',\n",
       "  'Publication': '2024',\n",
       "  'Authors': 'Wei Zhou, Michelle Van Sinderen, Katarzyna Rainczuk, Ellen Menkhorst, Kelli Sorby, Tiki Osianlis, Mulyoto Pangestu, Leilani Santos, Luk Rombauts, Alberto Rosello-Diez, Evdokia Dimitriadis',\n",
       "  'Email': 'evdokia.dimitriadis@unimelb.edu.au',\n",
       "  'Abstract': 'The endometrium undergoes substantial remodeling in each menstrual cycle to become receptive to an implanting embryo. Abnormal endometrial receptivity is one of the major causes of embryo implantation failure and infertility. MicroRNA-124-3p is elevated in both the serum and endometrial tissue of women with chronic endometritis, a condition associated with infertility. MicroRNA-124-3p also has a role in cell adhesion, a key function during receptivity to allow blastocysts to adhere and implant. In this study, we aimed to determine the function of microRNA-124-3p on endometrial epithelial adhesive capacity during receptivity and effect on embryo implantation. Using a unique inducible, uterine epithelial-specific microRNA overexpression mouse model, we demonstrated that elevated uterine epithelial microRNA-124-3p impaired endometrial receptivity by altering genes associated with cell adhesion and polarity. This resulted in embryo implantation failure. Similarly in a second mouse model, increasing microRNA-124-3p expression only in mouse uterine surface (luminal) epithelium impaired receptivity and led to implantation failure. In humans, we demonstrated that microRNA-124-3p was abnormally increased in the endometrial epithelium of women with unexplained infertility during the receptive window. MicroRNA-124-3p overexpression in primary human endometrial epithelial cells (HEECs) impaired primary human embryo trophectoderm attachment in a 3-dimensional culture model of endometrium. Reduction of microRNA-124-3p in HEECs from infertile women normalized HEEC adhesive capacity. Overexpression of microRNA-124-3p or knockdown of its direct target IQGAP1 reduced fertile HEEC adhesion and its ability to lose polarity. Collectively, our data highlight that microRNA-124-3p and its protein targets contribute to endometrial receptivity by altering cell polarity and adhesion.',\n",
       "  'Introduction': 'The human endometrium undergoes substantial remodeling during each menstrual cycle to become receptive to an implanting embryo only within a narrow window of 2 to 4 d in the mid-secretory phase (1, 2). Inadequate remodeling of endometrium during receptivity is a major cause of embryo implantation failure and female infertility (3). The importance of a receptive uterus for embryo implantation in vivo has been proven by animal models (4). Using reciprocal embryo transfer in rats and mice, it has been shown that blastocysts cannot implant and will degenerate outside the receptive window (5, 6). To attain receptivity the endometrial luminal epithelium must lose its polarity and an apical surface that is repulsive toward the polarized blastocyst trophectoderm (7–9). This allows firm attachment and adhesion between the trophectoderm and luminal epithelium, initiating implantation (3). Inadequate remodeling of the endometrial luminal epithelium results maintenance of polarity, preventing embryo attachment and implantation (9–11). Remodeling of the luminal epithelium involves changes in the expression of adhesive molecules on the apical surface and alterations in cell polarity. However, the mechanisms enabling these changes remain elusive.\\nMicroRNAs (miRs) are dysregulated in the receptive endometrium of women with infertility (12–14). The regulation of protein translation by miRs is an efficient mechanism to control multiple protein-coding genes that may be necessary for specific cell functions. miRs disrupt gene/protein expression by destabilizing the transcript or repressing translation initiation (15, 16). Consequently, dysregulation of a single miR can disrupt the expression of multiple genes/proteins and significantly impact endometrial epithelial function within a short time window (17). However, there is limited evidence on how individual miRs that are dysregulated in endometrial epithelium of women with primary infertility during receptivity contribute to endometrial receptivity.\\nmiR-124-3p has been reported to be elevated in both endometrial tissue and serum of women with chronic endometritis during receptivity (18), a condition associated with infertility and chronic inflammation. miR-124-3p is predicted to regulate genes in the endometrial luminal epithelium vital for endometrial receptivity including leukemia inhibitory factor (LIF) and signal transducer and activator of transcription 3 (STAT3) (19–21). In cancer cells, miR-124-3p can suppress cell adhesive capacity and invasiveness (22, 23), processes known to dysregulate endometrial receptivity. Therefore, in this study, we aimed to investigate the role of miR-124-3p in endometrial receptivity and embryo implantation by focusing on epithelial cell remodeling, as the initiation of implantation depends on the firm adhesion of the blastocyst trophectoderm to the endometrial luminal epithelium.\\nIn this study, we developed and used an inducible, uterine epithelial-specific miR-124-3p overexpression mouse model (transgenic miR-124-3p mice). We determined that uterine epithelial-specific overexpression of miR-124-3p compromised epithelial cell remodeling and receptivity, resulting in implantation failure without affecting blastocysts. In humans, we demonstrated that miR-124-3p was abnormally elevated in the endometrial luminal epithelium of patients with primary unexplained infertility, many of whom exhibit dysregulated receptivity (24). Using four independent in vitro human models, we demonstrated that miR-124-3p negatively affected the adhesive capacity of endometrial epithelial cells. We found that miR-124-3p overexpression in endometrial epithelial cells impaired human embryo trophectoderm adhesion to a 3-D primary human endometrial epithelial-stromal cell bilayer. Mechanistically, overexpression of miR-124-3p disrupted the endometrial epithelial cell changes necessary for adhesion and polarity required for receptivity in both mice and humans. Overall, our findings provide strong evidence that dysregulation of miR-124-3p can block endometrial epithelial cell function during receptivity, leading to implantation failure.',\n",
       "  'Methods': 'Reagents.\\nAntibodies used were summarized in Dataset S3. Primer sequences used were supplied in SI Appendix, Table S1. Collagenase was purchased from Worthington (type 3, Lakewood, NJ). miR primers and Transfection reagents (siRNA mimic, miR-124-3p mimic, miR-124-3p inhibitor, Lipofectamine RNAiMAX, Opti-MEM Reduced serum medium) were supplied by Thermo (Waltham, MA). 2’OMe modified and cholesterol conjugated AgomiR-124 and scrambled control were obtained from RiboBio (Guangzhou, China).\\nAnimal Ethics.\\nAll experimental procedures involving animals were conducted with the approval of the Animal Ethics Committee at the University of Monash (MMCB/2016/36) and the University of Melbourne (project ID: 10216 and 20395).\\nGeneration and Use of The miR-124-3p Transgenic Mice.\\nThree transgenic mice were crossed to produce the miR-124 mice (Ltf-iCre+/−, CAG-rtTA3+/−, Dragon-miR-124-3p+/−) and control mice (Ltf-iCre+/−, CAG-rtTA3−/−, Dragon-miR-124-3p+/−). The Ltf-iCre and CAG-rtTA3 mice were purchased from the Jackson Laboratory (JAX:026030 and 029627). Dragon-miR-124-3p mice were produced at the Monash Genome Modification Platform.\\nDetermination of miR-124 sequence to be included.\\nMature miR-124-3p can be transcribed from three different genomic DNA fragments. To determine which fragment to use, we constructed three plasmids, each containing miR124-3p-stem-loop and 200 bp flanking sequence on each side, derived from each of the three genomic fragments (SI Appendix, Fig. S1A). Using Ishikawa cells and the same doxycycline activation system, we confirmed all three fragments successfully overexpressed mature miR-124-3p in vitro to similar levels (SI Appendix, Fig. S1B). miR-124-3 was chosen to create the Dragon-miR-124-3p mice. Dragon-miR-124-3 Flp-in vector was built by A.R-D. following the design described in ref. 59, by cloning synthetic miR-124-3 cassette into plasmid pDragon-empty (Addgene #155018). G4 embryonic stem cells (kind gift from Andras Nagy) were modified via homologous recombination, to have a Flp-in landing site at the Igs7 locus (chr9:21,271,659) using Addgene plasmid #61580 (kind gift from Hongkui Zeng) as targeting vector. Following this, the landing-site embryonic stem cells underwent recombinase-mediated cassette exchange using the Flp-in vector Dragon-miR-124-3. One of the confirmed clones was selected to generate the Dragon-miR-124-3p mice. In summary, the Dragon-miR-124-3p mouse had a DNA fragment of TRE-Loxp-tdTomato-STOP-Loxp-miR-124-WPRE inserted in an intergenic sequence of chromosome 9 (SI Appendix, Fig. S1A). WPRE (Woodchuck hepatitis virus Posttranscriptional Regulatory Element) is used to enhance miR-124 expression (SI Appendix, Fig. S1A).\\nConfirmation of Dragon system activation in mouse uterine epithelium.\\nLtf-iCre primarily drives recombination in the uterine epithelium (26) and rtTA3 protein activates the Tet-responsive element in the Dragon allele only if bound by doxycycline. Therefore, miR-124 overexpression can be induced specifically in the endometrial epithelium at the desired times by doxycycline administration. Having created the transgenic miR-124-3p mice, we first confirmed the expression of iCre and rtTA3 in the uterus by qPCR (SI Appendix, Fig. S1C). Transgenic control mouse uterus also expressed iCre but not rtTA3 as iCre+/rtTA3− genotype was used as a control. None of them were recorded in wildtype mouse uterus (SI Appendix, Fig. S1C). Using immunofluorescence, we visualized the Ltf-iCre activity in the uterine epithelium by absence of tdTomato fluorescence (SI Appendix, Fig. S1D). These data also confirmed that the activation of Dragon-miR-124-3p was dependent on rtTA3 because no tdTomato fluorescence was observed in the whole uterus of control mice (rtTA3−) after doxycycline administration (SI Appendix, Fig. S1D, Right panel).\\nOptimization of doxycycline administration.\\nDoxycycline is required to activate TRE promoter. To rapidly activate the Dragon system, we chose intraperitoneal injection to deliver doxycycline. A time course experiment was first conducted to optimize time of treatment in which a single intraperitoneal injection of doxycycline (80 µg/g body weight) was given to each unpregnant female. Mice were then culled at 4, 12, 24, and 48 h and uteri were collected to examine miR-124-WPRE transcription by qPCR. We recorded a significant increase in miR-124-WPRE expression in the luminal epithelium and remaining uterus only at 4 and 12 h compared to control respectively (SI Appendix, Fig. S1E). Following optimization, plugged miR-124 or control mice were given four injections of doxycycline (80 µg/g body weight) between E2 9 pm and E4 9 am.\\nIntrauterine Injection.\\nAdult female mice (C57BL6/J) were mated to male mice. Ten µg (in 20 µL saline) AgomiR-124 mimic or scrambled control (2’OMe modified and cholesterol conjugated) were delivered into the uterine horn via intrauterine injection at E3 (Fig. 3A). To confirm the successful delivery, mouse uteri were collected at E4 (prior to implantation) and the luminal epithelium was separated from the uterus via Dispase (0.5% in Hank’s Balanced Salt Solution) digestion for 2 h at room temperature (RT). The luminal epithelium and remaining uterus were used for qPCR to determine miR-124-3p and target genes expression.\\nMouse Embryo Transfection, Hatching, Zona Pellucida Breaking, and Outgrowth.\\nMouse concepti were collected at E1 and cultured in G1 plus medium for 72 h until concepti reached the morulae stage (when in vivo embryos were exposed to miR-124-3p). E3 embryos were treated with fluorescently labeled miR-124-3p mimic or scrambled control (100 nM) using the lipofectamine RNAiMAX based transfection system (1% lipofectamine in blastocyst culture medium). Embryos were then cultured in G1-plus based transfection medium for 24 h and the uptake of miR-124-3p (visualized by fluorescence) was recorded by confocal microscopy. Unhatched embryos (E4) were treated with the same transfection mixture for 72 h to record the effects on hatching, zona pellucida breaking, and outgrowth using the MIRI time-lapse system (Esco). Successful miR-124-3p overexpression was confirmed by qPCR.\\nHuman Sample Collection and Ethics Statement (Endometrial Tissue and Embryo).\\nEndometrial tissue: Written informed consent was obtained from each patient and the study was approved by the Human Research Ethics Committee at Monash Health and the Royal Women’s Hospital (ID: #03066B). All women had regular menstrual cycles, were not using intrauterine contraceptives and had not used hormones for at least 3 mo before surgery. Fertile women had proven parity and infertile women had primary unexplained infertility. The collected endometrium were examined by experienced gynecological pathologists to confirm the cycle stage and absence of apparent endometrial dysfunction such as chronic endometritis and endometriosis. Defective receptivity is a significant factor contributing to the etiology of unexplained infertility (60). Human embryo: The usage of human embryos was approved by the NHMRC Human Embryo Research License (ID: #309722). Embryos (surplus) were created using either conventional IVF or Intracytoplasmic Sperm Injection. The detailed protocol of embryo vitrification, thawing, allotment, and culture was established in our previous study (61) with the modification of removing the inner cell mass due to consent restrictions. The inner cell mass was removed to ensure couples donating excess embryos were comfortable for research to be performed on this material. While the inner cell mass was removed, the embryos were still able to re-expand and we could evaluate embryonic trophectoderm adhesion to the endometrial cocultured cells. To remove inner cell mass, mechanical dissection was performed on fully hatched embryos using a 35 µm diameter biopsy pipette (TPC, LBC-OD35BA90) on a Transferman micromanipulation system. Resultant trophectoderm were allowed to reform into spheroids (occurred within 2 h) and used immediately.\\nHEEC and Stromal Cell Isolation and In Vitro Culture Models.\\nCollected endometrium (day 7 to 18 of the menstrual cycle) were subjected to HEEC and stromal cell isolation as previously published (62). The method of HTR8/SVneo spheroid adhesion on HEEC monolayer was detailed described in our previous study (17). The other two models using the 3D endometrial construct were prepared as previously described (63). 3D endometrial constructs were treated with 0.3 nmol/L estrogen and 900 nmol/L progesterone for 1 d before adding primary human trophectoderm spheroids. The attachment and outgrowth of primary human trophectoderm spheroids on endometrial construct was recorded up to 120 h. Firm attachment of the trophectoderm spheroids was determined by gently shaking the culture plates under microscopy.\\nmiR-124-3p Mimic and Inhibitor, Target Site Blocker, and Target Genes siRNA Transfection.\\nCells were cultured to 70% confluence then transfected with Lipofectamine RNAiMAX and Opti-MEM medium containing miR-124-3p mimic (10 nM), miR-124-3p inhibitor (30 nM), siRNA mimic (10 nM), or scrambled controls. TSB (30 nM) was cotransfected with miR-124-3p mimic (30 nM) to investigate the direct interaction between miR-124-3p and IQGAP1. After 24 h the transfection medium was replaced with fresh culture medium and cells were cultured for 48 h before being subjected to spheroid adhesion assay or other analyses.\\nTransepithelial Electrical Resistance.\\n2 × 105 Ishikawa cells were seeded on each extracellular matrix (1:10 Matrigel/DMEM) coated trans-well insert (6.5 mm, 0.4 µm pore, Corning). Cells were cultured in DMEM containing 10% fetal bovine serum overnight to allow attachment and then switched to 0.5% fetal bovine serum. After 24 h of culturing, the Millipore MilliCell-Electrical Resistance System (Millipore) was used to measure the baseline electrical resistance. Cells were then transfected as described above. Following transfection, cells were maintained under DMEM containing 0.5% fetal bovine serum, and transepithelial electrical resistance was measured daily for three days. Changes in transepithelial electrical resistance were determined relative to the baseline readings, expressed as a percentage for each condition.\\nRNA Isolation, RT-qPCR, and Customized PCR Array.\\nTotal RNA was isolated using the TriReagent or RNeasy Mini Kit (Qiagen) or RNeasy Micro Kit (Qiagen, for mouse embryos). miR-124-3p PCR was conducted using the Taqman miR reverse transcription kit and Universal Master Mix II according to the manufacturer’s instructions (Thermo). For gene targets detection, 300 ng total RNA was converted to cDNA using SuperScript™ III First-Strand Synthesis System. miR-124-3p expression was normalized to U44 (human) or U6 (mouse). Gene expression was normalized to 18S. Relative expression levels were calculated using the comparative cycle threshold method (ΔΔCt). Customized PCR array was conducted according to the manufacturer’s instructions (Qiagen). miRWalk was used to select putative miR-124-3p targets.\\nImmunohistochemistry and Immunofluorescence.\\nTissues were fixed in 10% formalin, embedded in paraffin and sectioned at 4 µm thickness. Slides were dewaxed, rehydrated, and subjected to antigen retrieval as summarized in Dataset S3. Endogenous peroxidase was blocked using 3% hydrogen peroxide in methanol for 15 min. Nonimmune blocking was completed with 10% goat serum and 2% human serum (human tissue) or 2% mouse serum (mouse tissue) in Tris-buffered saline (TBS) for 1 h at RT. Slides were then incubated with primary antibodies at 4 °C overnight (dilution of all antibodies in Dataset S3). An isotype control was included in which sections were incubated with nonimmune antibody of the same isotype as the primary antibody at the same concertation. Positive signal was visualized via Diaminobenzidine. Staining intensity scores were determined by two individual scorers blinded to the patient characteristic.\\nFor immunofluorescent staining, cultured cells were fixed with 4% paraformaldehyde for 15 min. Cells were blocked in 3% bovine serum albumin (BSA)/ Phosphate-buffered saline (PBS) at RT for 1 h and subjected to overnight incubation of primary antibodies at 4 °C. After washing with PBS, cells were incubated with Alexa Fluor 488-conjugated antisera at RT for 1 h. 3D lateral labeling of IQGAP1, E-cadherin, and SCRIBBLE was scanned using Z-stack confocal imaging.\\nImmunoblotting.\\nCells were lysed in ice-cold lysis buffer (50 mM Tris⋅HCl [pH 7.5], 150 mM NaCl, 2 mM EDTA, 2 mM EGTA, 25 mM NaF, 25 mM β-glycerolphosphate, protease inhibitor mixture). Proteins were resolved by SDS-PAGE (150 V, 1 h) and transferred to PVDF membranes (100 V, 1 h). Membranes were blocked with 5% skim milk and incubated with antibodies as summarized in Dataset S3. After overnight incubation at 4 °C, membranes were washed with TBS-Tween 0.1% (v/v) and incubated with HRP conjugated secondary antibodies. Labeled proteins were detected by chemiluminescence (Thermo). Membranes were stripped and incubated with anti-GAPDH as a loading control. Protein band intensity was determined using ImageJ.\\nhCG ELISA.\\nHuman hCG ELISA kit was purchased from Thermo (EHCG) with a sensitivity of 50 pg/mL. ELISA was conducted according to the manufacturer’s instructions. Pregnant human serum was diluted 1:4 and used as a positive control. Absorbance was measured at 450 nm. A concentration of 46,127 pg/mL hCG was detected in diluted pregnant human serum (7,500 pg/mL equals 22.5 mIU/L).\\nStatistical Analysis.\\nExperiments reported in this study were repeated on at least three biological samples (indicated in figure legends). Statistical analysis was carried out using PRISM 8.0 and student’s t test or one-way ANOVA as appropriate. P < 0.05 was considered statistically significant.',\n",
       "  'Results': 'Inducing Overexpression of miR-124-3p in the Uterine Epithelium During the Receptive Window Blocks Receptivity and Leads to Implantation Failure.\\nTo determine whether luminal epithelial expression of miR-124-3p was associated with receptivity we examined its expression in mice just before and during the receptive period. Luminal epithelial miR-124-3p production was significantly down-regulated in the receptive phase (day of implantation, embryonic day [E]4) (day of plug designated E0) compared to the previous day (E3) (Fig. 1A). This suggested that miR-124-3p needs to be reduced in the luminal epithelium during receptivity to enable blastocyst attachment and initiate implantation. To test this, we generated an inducible, uterine epithelium-specific miR-124-3p overexpression mouse model. We used the doxycycline-controlled and cre recombinase activated gene overexpression (Dragon) system (Materials and Methods) (25). Briefly, three transgenic mice were crossed to create transgenic miR-124-3p mice [Ltf-Cre+ (26), CAG-rtTA3+ (27), Dragon-miR-124-3p+]. Activation of the Dragon-miR-124-3p allele and therefore miR-124-3p overexpression depends on both iCre and rtTA3 drivers, such that overexpression only occurs in tissues/cells where both drivers overlap and doxycycline is present (SI Appendix, Fig. S1).\\nFig. 1.\\nOpen in a new tab\\nOverexpression of miR-124-3p in the murine uterine epithelium during endometrial receptivity impaired implantation. (A) Endogenous miR-124-3p levels in uterine luminal epithelium from pregnant wildtype mouse uterine luminal epithelium at E3 and E4 (qPCR; N = 3 to 4). (B–F) Uterine epithelium specific miR-124-3p overexpression significantly reduced implantation success in mice. (B) Timeline of doxycycline treatment to induce miR-124-3p overexpression during endometrial receptivity. (C) miR-124-3p overexpression in the uterus was verified at E4.5 by qPCR (N = 5 to 6).\\n(D) miR-124-3p overexpression significantly reduced implantation site number (IS, arrowhead) at E4.5 (N = 5 to 6). (E) miR-124-3p overexpression was localized to the luminal and glandular epithelium by in situ hybridization. (F) miR-124-3p did not affect embryo viability, IS found in transgenic miR-124-3p mice stained with the cell apoptosis marker Cleaved Caspase-3. Wildtype mouse spleen was used as a positive control for staining. All data are presented as mean ± SEM. *P < 0.05, ****P < 0.0001.\\nTo overexpress miR-124-3p during receptivity, doxycycline was given by intraperitoneal injection every 12 h between E2 9 pm and E4 9 am (Fig. 1B). qPCR analysis identified that mature miR-124-3p was overexpressed in the uterus of transgenic miR-124-3p mice (Fig. 1C). We found that miR-124-3p overexpression significantly reduced the number of implantation sites in the transgenic miR-124-3p mice compared to control (Fig. 1D, P < 0.0001). The levels of miR-124-3p overexpression were positively correlated with the severity of implantation outcomes, as implantation was completely blocked in the two mice with over 10-fold increase of miR-124-3p (Fig. 1C, arrows). In situ hybridization confirmed that miR-124-3p overexpression was localized specifically to the uterine luminal and glandular epithelium in the transgenic miR-124-3p mice (n = 3, Fig. 1E). For the few embryos that implanted in the transgenic mouse uterus [total of four implantation sites in five mice (Fig. 1D)], their viability was confirmed by the absence of cleaved Caspase-3 staining (Fig. 1F).\\nmiR-124-3p Targets a Wide Range of Receptivity-Related Genes in Transgenic miR-124-3p Mouse Uteri.\\nTo determine the gene targets affected by miR-124-3p overexpression during endometrial receptivity, we analyzed E4.5 uteri (just postembryo attachment) from both transgenic control and miR-124-3p mice. qPCR analysis of selected epithelial receptivity markers (Areg, Muc1, and E-cadherin) and stromal decidualization markers (Bmp2, Hand2, and Cox2) revealed significant differences between transgenic control and miR-124-3p mice (Fig. 2A). We therefore expanded the investigation of genes altered in the transgenic miR-124-3p mice using a customized PCR array containing 83 receptivity-associated genes in mice (Dataset S1) (28). Among them, 38 genes showed significant differences in expression in the miR-124-3p overexpression uterus compared to control (Fig. 2B). Furthermore, 29 genes were significantly reduced by miR-124-3p overexpression in the uterus at E4.5 compared to transgenic control mice (Fig. 2B). This included genes known to regulate endometrial receptivity (Hoxa11, Egf, Areg, Ptgs2), extracellular matrix/cell adhesion molecules (Sparc, Gpx3, Col5a2, Vcan, Itga5, Snai2), postimplantation genes (Vegfa, Mid1, Timp1), cytokines (Il15, Il11), leukocyte migration (Mmp2, Fbn1, Icam1, Lama1), cell cycle (Pcna, Stmn1, Mki67, Ccnb1), and signal transduction (Ctnnb1, Il1r1, Sfrp4, Trp53, Il1a). These data suggest that embryos did not firmly adhere to the endometrial luminal epithelium to initiate implantation and the decidual response. miRs act by destabilizing mRNA or repressing mRNA translation, leading to downregulation of target proteins. Here, we confirmed bioinformatically that 21 of the 29 genes reduced have miR-124-3p binding sites (bolded in Fig. 2B) suggesting that they may be direct targets. Conversely the expression of nine genes was significantly increased after miR-124-3p overexpression (Esr1 to Gdf15 in Fig. 2B) suggesting that they may be indirect targets.\\nFig. 2.\\nOpen in a new tab\\nmiR-124-3p overexpression in mouse uterine epithelium impairs receptivity. (A) Representative receptivity and decidualization genes in E4.5 transgenic control and miR-124-3p uteri (qPCR; N = 4 to 6). All data are presented as mean ± SEM. (B) Heatmap of results from customized PCR gene array on E4.5 uteri from transgenic miR-124-3p and control mice. Data presented as dCt value in heat map and ranked by expression level (high to low) (N = 5 to 6). Genes having miR-124-3p binding site(s) highlighted in bold. Original data provided in Dataset S1. *P < 0.05, **P < 0.01, ***P < 0.001, ****P < 0.0001. (C) Immunohistochemistry staining at E4.5 of cell polarity markers (E-cadherin, SCRIBBLE), proliferation marker (KI-67), receptivity marker (P-STAT3), and decidualization marker (DESMIN). As no implantation sites were recorded in some miR-124-3p transgenic mice, uteri at E4.5 were also stained. Arrow: implanting blastocyst. The tissue sections are oriented so that the antimesometrial (AM) sides of the uterus is at the top and the mesometrial (M) is at the bottom.\\nmiR-124-3p Overexpression in the Endometrial Epithelium Compromises Implantation Chamber Formation and the Loss of Luminal Epithelial Cell Polarity.\\nSince implantation occurred in some transgenic miR-124-3p mice [a total of four implantation sites in five mice (Fig. 1D)], when examining mouse uteri by immunohistochemistry, we divided transgenic miR-124-3p mouse uteri into two subgroups: areas with and without implantation sites. Morphologically, the uterine lumen failed to close properly in the implantation sites from transgenic miR-124-3p mice (Fig. 2C, Middle panel). Staining of SCRIBBLE and E-cadherin identified retention of cell polarity in the luminal epithelium in both areas with implantation sites and without implantation site groups in transgenic miR-124-3p mice. This is in contrast to normal pregnancy where uterine luminal epithelial cell polarity should be minimized to allow close contact to the polarized embryo surface (9, 29) (Fig. 2C, Top panel). While there was no difference in localization and staining intensity in areas with implantation sites between transgenic control and miR-124-3p mice, in miR-124-3p mouse uteri without implantation sites, there was no detectable staining for P-STAT3, DESMIN, and KI-67 staining (Fig. 2C), suggesting that where the embryo was present, decidual changes associated with implantation were capable of being initiated.\\nOverexpression of miR-124-3p Specifically in the Mouse Uterine Luminal Epithelium Impairs Receptivity.\\nOur overall aim was to investigate how changes in the uterine luminal epithelium during receptivity impacted embryo attachment and implantation. Since our transgenic mouse model overexpressed miR-124-3p in both luminal and glandular epithelium, we used intrauterine injection to selectively deliver and overexpress AgomiR-124-3p specifically to the luminal epithelium. Briefly each mouse received intrauterine injections to one horn of either AgomiR-124-3p or a control AgomiR mimic at E3, with each mouse therefore serving as both a treatment and control (Fig. 3A). qPCR analysis confirmed a significant fourfold increase in miR-124-3p expression levels in the luminal epithelium at E4 following intrauterine injection compared to AgomiR-control (P < 0.05, Fig. 3B). No significant change in miR-124-3p levels was observed in the remaining uterus (Fig. 3B). In situ hybridization also confirmed that miR-124-3p overexpression was confined to the uterine luminal epithelium (Fig. 3C). Treatment with AgomiR-124-3p significantly reduced (P < 0.01) implantation site number compared to AgomiR-control treatment (Fig. 3D), with five of eight treated mice having no implantation sites in the AgomiR-124-3p treated horn. To confirm that miR-124-3p did not affect embryo viability we transfected wildtype mouse embryos in vitro with miR-124-3p mimic. There was no effect on embryo zona pellucida breaking/demise, hatching, and outgrowth, compared to control (Fig. 3E and SI Appendix, Fig. S2). Therefore, the compromised implantation was likely caused by altered miR-124-3p effects on uterine epithelium during receptivity.\\nFig. 3.\\nOpen in a new tab\\nLuminal epithelium only miR-124-3p overexpression impaired implantation. (A) Schematic diagram of the intrauterine injection to overexpress miR-124-3p in the mouse luminal epithelium. For each individual mouse, AgomiR-124-3p or scrambled mimic was injected into one side of the uterine horn at E3. O: Ovary, F: Fallopian tube, U: Uterus. (B) miR-124-3p was significantly elevated in the luminal epithelium after intrauterine injection compared to the remaining uterus and control uterine horn (qPCR; N = 3). (C) miR-124-3p overexpression localized to the luminal epithelium as determined by in situ hybridization. (D) Intrauterine injection of miR124-3p significantly reduced implantation site number in the treated horn compared to untreated control at E6 (N = 8). (E) Exposure of unhatched wildtype mouse embryos at E4 to miR-124-3p showed no effect on outgrowth compared to control. (F and G) The effect of miR-124-3p overexpression on receptivity and decidualization markers was determined by immunohistochemistry (F) and qPCR (G). Arrow: implanting blastocyst (N = 3 to 4). The tissue sections are oriented so that the AM sides of the uterus is at the top and the M is at the bottom. All data are presented as mean ± SEM. *P < 0.05, **P < 0.01, ***P < 0.001, IS: implantation site.\\nIn uterine horns treated with AgomiR-124-3p where implantation sites were absent, we observed that the luminal epithelium maintained its cell polarity, as revealed by SCRIBBLE staining (Fig. 3F, Top panel), and no decidualization response occurred, as indicated by SCRIBBLE and DESMIN staining (Fig. 3F). Overexpression of miR-124-3p specifically in the uterine luminal epithelium resulted in a significant reduction in the expression of implantation-related genes Bmp2, Wnt4, Hand2, and Cox2 (30–33) compared to AgomiR-control (Fig. 3G), confirming that the embryo failed to firmly attach and initiate implantation, and that the decidualization response was impaired. By contrast, in uterine horns treated with AgomiR-124-3p where implantation sites were present, we saw no differences in the localization and intensity of SCRIBBLE and DESMIN staining compared to AgomiR-control (Fig. 3F). The uterine lumen remained closed in the AgomiR-124-3p treated group with implantation sites, similar to AgomiR-control (Fig. 3F).\\nmiR-124-3p Expression Is Elevated in the Endometrium of Women with Primary Infertility.\\nTo investigate the clinical relevance of miR-124-3p on endometrial function, its expression was assessed in endometrial tissues across the menstrual cycle from women with normal fertility (fertile) and women with primary unexplained infertility (infertile). miR-124-3p was expressed in all phases of the menstrual cycle in both fertile and infertile groups (Fig. 4A). qPCR analysis identified that in fertile women, miR-124-3p expression decreased during the secretory phase compared to the proliferative phase, particularly during the mid-secretory phase (Fig. 4A). This decrease was not observed in infertile women and miR-124-3p expression was significantly increased in infertile endometrium compared to fertile endometrium during the early and mid-secretory phases (Fig. 4A). In situ hybridization of mid-secretory phase endometrium illustrated miR-124-3p localization in the luminal and glandular epithelium of infertile endometrium, whereas minimal fluorescence was observed in the fertile endometrium (Fig. 4B). Overall, these findings suggest that miR-124-3p expression is dysregulated in infertile endometrium.\\nFig. 4.\\nOpen in a new tab\\nmiR-124-3p was overexpressed in infertile human endometrial luminal epithelium during receptive phase. (A) miR-124-3p expression in endometrial tissue (qPCR; N = 6 to 11) from fertile and infertile endometrium. Expression levels were normalized to U44. The unpaired t test was used to compare miR-124-3p levels between each phase and between fertile and infertile groups. Data are presented as mean ± SEM. *P < 0.05, **P < 0.01. (B) In situ hybridization of miR-124-3p on mid-secretory phase endometrial sections revealed strong fluorescence in the luminal epithelial cells of the infertile endometrium, compared to fertile endometrium (green arrows, N = 3). Sections were counterstained with DAPI to reveal the cell nuclei (blue). As controls, endometrial sections were either incubated with U6 probe (positive control) or scrambled sequence probe (scrambled control). L: Luminal epithelium; G: Glandular epithelium; S: Stroma.\\nOverexpression of miR-124-3p in Human Endometrial Epithelial Cells Impairs Their Adhesive Capacity and Receptivity.\\nTo explore how miR-124-3p overexpression affects endometrial epithelial cell adhesive capacity and receptivity, we employed a 3D model mimicking implantation. This model consisted of a layer of primary human endometrial epithelial cells (HEECs) placed on top of a layer of primary human endometrial stromal cells, isolated from women with normal fertility, as previously described with some modifications (34). Initially miR-124-3p was overexpressed in HEECs before they were overlaid on the primary human endometrial stromal cell layer to form a bilayer construct. Primary human embryo trophectoderm spheroids were prepared from Day 6 embryos by removing the inner cell mass (Fig. 5A). The trophectoderm were allowed to reform spheroids for up to 2 h (Fig. 5B). The re-expanded trophectoderm spheroids were added to endometrial constructs and monitored for their ability to firmly adhere to the endometrial epithelium and to outgrow for up to 120 h (equivalent to Day 11 postfertilization embryo).\\nFig. 5.\\nOpen in a new tab\\nmiR-124-3p impaired HEEC receptivity. (A–C) Outgrowth of primary human trophectoderm spheroids on a 3D primary human endometrial cell construct. Primary HEECs in the construct were transfected with miR-124-3p mimic or scrambled control (N = 3). (A) Before culture, inner cell mass (arrow) was removed from day 6 human blastocysts. (B) The remaining blastocyst readily re-formed into a spheroid. (C) Human trophectoderm attachment and outgrowth on the 3D endometrial construct (outlined, cultured up to 120 h) (N = 3). (D and E) Adhesion of HTR8/SVneo spheroid to (D) 3D endometrial construct (N = 3) or (E) HEEC monolayer (N = 3 to 5). (F) miR-124-3p levels in fertile and infertile HEECs (N = 3 to 5). (G) HTR8/SVneo spheroid adhesion to infertile HEECs transfected with miR-124-3p inhibitor or control (N = 5). Data are presented as mean ± SEM. *P < 0.05, **P < 0.01.\\nControl group trophectoderm spheroids firmly attached within 48 h and continued to grow until 120 h (Fig. 5C). Human chorionic gonadotropin (hCG) was detected in the coculture media by ELISA, with an average concentration of 7,040 ± 569 pg/mL at 120 h (N = 3). However, in the miR-124-3p group, all three trophectoderm spheroids failed to firmly attach to the endometrial constructs within 48 h, and two of them showed reduced trophectoderm area and signs of degeneration before 72 h of culture. Only one trophectoderm spheroid could be traced until 120 h, but it appeared to degenerate by 72 h, with very few trophectoderm cells remaining visible by 120 h (Fig. 5C). hCG was undetectable in the coculture media from the miR-124-3p group at 72 and 120 h. Overall, our data indicate that miR-124-3p overexpression in HEECs impaired trophectoderm spheroid attachment, likely leading to cell death or apoptosis.\\nTo further understand the impact of miR-124-3p overexpression on embryo adhesion we used two additional in vitro human models. Primary trophectoderm was replaced with HTR8/SVneo (trophoblast cell line) spheroids and their adhesion on either the 3D endometrial construct or a HEEC monolayer was determined. Both models confirmed that miR-124-3p overexpression in fertile HEECs significantly impaired (P < 0.05) their adhesiveness to HTR8/SVneo spheroids (Fig. 5D and E).\\nBy comparison to fertile HEECs, the levels of miR-124-3p were significantly elevated in infertile HEECs (P < 0.05, Fig. 5F). To investigate whether miR-124-3p inhibition can improve cell adhesive capacity of infertile HEECs, HEECs from infertile women were transfected with a miR-124-3p inhibitor. Following miR-124-3p inhibitor treatment, the adhesive capacity of infertile HEECs to HTR8/SVneo spheroids significantly increased (P < 0.01, Fig. 5G) compared to the control. As the fourth model to assess the adhesive ability of endometrial epithelial cells without using spheroids, Ishikawa cells [a receptive endometrial epithelial cell line (35)] were investigated by xCELLigence to monitor the effects of elevated miR-124-3p on their adhesive capacity in real-time. miR-124-3p mimic treatment of Ishikawa cells significantly reduced their adhesion at 2 h (P < 0.0001) and was reversed after cotreatment with the miR-124-3p inhibitor (SI Appendix, Fig. S3).\\nmiR-124-3p Targets Adhesion Related Molecules in HEECs.\\nAs endometrial epithelial cells must remodel in the receptive phase to facilitate embryo adhesion and initiate implantation, we investigated putative miR-124-3p targets with known functions on cell adhesion. 88 genes (Dataset S2) were identified and screened in HEECs via the Fluidigm BioMark HD System. The list included genes with confirmed functions in regulating embryo implantation in mice (e.g., STAT3 and JAG) (28, 36) and target genes that have not been investigated in endometrial receptivity (e.g., IQGAP1). Of the 88 genes, 20 were significantly downregulated following miR-124-3p overexpression in HEECs compared to control (Fig. 6A). Nine of the 20 miR-124-3p targets identified in HEECs were similarly downregulated in E4.5 transgenic miR-124-3p mouse uterus, suggesting similar mechanisms of action in the epithelium between the two species (Fig. 6B).\\nFig. 6.\\nOpen in a new tab\\nmiR-124-3p targets adhesion genes in human and mouse uterus. (A) Fluidigm BioMark HD System was applied to quantify the expression of 88 genes in HEECs transfected with scrambled control or miR-124-3p mimic. These genes are predicted miR-124-3p targets and fall into broad categories of cell adhesion and attachment. Data were presented as dCt value in heat map (N = 5). Original data provided in Dataset S2. (B) Expression of miR-124-3p gene targets identified in A in E4.5 uterus from transgenic miR-124-3p or control mice (N = 4 to 6). Data are presented as mean ± SEM. *P < 0.05, **P < 0.01, ***P < 0.001, ****P < 0.0001.\\nIQGAP1 was selected for further investigation due to its unknown function in endometrial epithelial cell adhesion. To determine the clinical relevance of IQGAP1 in endometrial receptivity, the localization and production levels of IQGAP1 in the receptive phase endometrium were compared between fertile and infertile women. IQGAP1 localized to the luminal epithelium, glandular epithelium, and stromal cells (Fig. 7A). The staining intensity of IQGAP1 was significantly higher in all cellular compartments in the fertile endometrium compared to the infertile group (Fig. 7A and B).\\nFig. 7.\\nOpen in a new tab\\nmiR-124-3p target IQGAP1 alters HEEC adhesion. (A and B) IQGAP1 immunolocalization in fertile and infertile secretory phase human endometrium. Higher magnification images of the luminal epithelium (L), glandular epithelium (G), and stromal cells (S) are depicted on the right of panels with their original locations indicated by outlines on the left. Sections were counterstained with hematoxylin to indicate cell nuclei (blue). IQGAP1 staining intensity (N = 4) was semiquantitated by scoring. Endometrial tissues were blinded to fertility status. (C) Immunoblot showing reduction in IQGAP1 protein following HEEC transfection with miR-124-3p mimic (10 nM, N = 4 to 6). (D and E) HTR8/SVneo spheroid attachment (D) to HEECs following IQGAP1 knockdown (E, qPCR) (N = 4). (F and G) Target site blocker (TSB) was used to confirm the direct interaction between miR-124-3p and IQGAP1. (F) miR-124-3p and IQGAP1 TSB binding sequences of IQGAP1 mRNA. (G) HEECs were transfected with miR-124-3p mimic (30 nM) in combination with either IQGAP1 or control TSB (30 nM) and subjected to immunoblotting to determine expression of IQGAP1 (N = 6). (H) Protection of IQGAP1 via TSB in miR-124-3p overexpressed HEECs were able to improve their adhesion to HTR8/SVneo spheroids (N = 6). (I) Iqgap1 expression in mouse E4 luminal epithelium (intrauterine injection model). Data are presented as mean ± SEM. *P < 0.05, **P < 0.01, ***P < 0.001.\\nIn HEECs, our results confirmed that miR-124-3p overexpression significantly downregulated IQGAP1 protein levels compared to control (P < 0.01, Fig. 7C). Knockdown of IQGAP1 in HEECs significantly compromised their adhesiveness to HTR8/SVneo spheroids compared to control (Fig. 7D and E). To explore the direct interaction between miR-124-3p and IQGAP1 in HEECs, we applied a “TSB” designed to bind specifically to the miR-124-3p target site of the IQGAP1 mRNA (Fig. 7F), thereby blocking miR-124-3p accessing that site. As expected, cotransfection of HEECs with miR-124-3p and the IQGAP1 TSB significantly increased production of IQGAP1 compared to HEECs transfected with miR-124-3p alone (Fig. 7G, P < 0.01). Similarly, treatment with the TSB restored HEEC adhesive capacity to HTR8/SVneo spheroids (Fig. 7H, P < 0.05). MiR-124-3p overexpression in both human and mouse models significantly reduced Iqgap1 expression (Figs. 6B and 7I), suggesting IQGAP1 is a common miR-124-3p target between humans and mice.\\nmiR-124-3p and IQGAP1 Regulate the Polarity of HEECs.\\nIQGAP1 engages with cytoskeleton proteins to internalize cadherin junction protein E-cadherin in human mammary glandular epithelial cells (37). Here, we sought to test whether IQGAP1 reduces endometrial epithelial cell polarity by internalizing E-cadherin. We first confirmed the reduction of IQGAP1 protein in Ishikawa cells after transfection with either IQGAP1 siRNA or miR-124-3p mimic (Fig. 8A, P < 0.05). Using a transepithelial electrical resistance assay, we demonstrated that loss of IQGAP1 in Ishikawa cell monolayers (either by IQGAP1 siRNA or miR-124-3p mimic treatment, Fig. 8B) increased transepithelial electrical resistance, therefore increasing cell polarity (38) at 72 h posttransfection (P < 0.05) compared to control. In agreement, confocal scanning of the entire lateral layer of the Ishikawa monolayers revealed loss of IQGAP1 resulted in the retention of E-cadherin on the cell lateral surface and the cell polarity marker SCRIBBLE compared to controls (Fig. 8C).',\n",
       "  'Discussion': 'Endometrial receptivity involves the dramatic remodeling of all cellular compartments in the endometrium and is widely recognized to be critical for embryo implantation and the establishment of pregnancy (3, 4). However, how endometrial epithelial cell adhesive capacity during receptivity is regulated in humans remains largely elusive. In this study, we identified elevated levels of miR-124-3p in the endometrial epithelium of women with unexplained infertility. We aimed to investigate the role of miR-124-3p in regulating endometrial epithelial cell adhesive capacity and its subsequent effect on embryo implantation in a unique way. This was achieved by developing a genetically inducible mouse model to overexpress miR-124-3p specifically within the uterine epithelium during the receptive window. This approach allowed us to explore the impact of miR-124-3p on endometrial adhesive capacity and receptivity in a controlled manner. This finding represents a significant advancement in our understanding of how dysregulation of miRs can affect endometrial receptivity and lead to implantation failure.\\nUsing genetic mouse models, studies have provided fundamental information on key uterine genes regulating embryo implantation. Among pioneering examples (39), maternal deletion of Cox2 leads to implantation failure by blocking blastocyst attachment and decidualization (30) while uterine knockout of Bmp2 only affects decidualization (31). Using genetic mouse models to determine miR functions on embryo implantation is rare, previous studies have relied on intrauterine injection of miR mimics (40) which may not accurately represent miR dysregulation in vivo because it bypasses endogenous transcriptional modification and may have unintended off target effects on the uterus. To complement intrauterine injection, here we used a combination of Cre-Loxp and Tet-on systems to genetically overexpress miR-124-3p specifically in the uterine epithelium during the receptive window. This study achieves miR overexpression in a cell-type-specific and time-dependent manner in the uterus. Our findings suggest that there is a critical time window during which miR-124-3p levels in the uterine epithelium need to be downregulated to facilitate implantation. Despite our genetic approach we still found variability in the levels of overexpression achieved, suggesting heterogeneity in the activation and/or turnover of miRs between animals and within the uterus. In addition, we recorded inconsistent results regarding implantation chamber formation in the intrauterine injection model. A potential reason for this is the inherent limitation of intrauterine injection which could fail to uniformly distribute miR-124-3p mimic across the entire uterine luminal surface. Overall, by using these two models, we were able to overexpress miR-124-3p in the luminal and glandular epithelium or in the luminal epithelium only. Increasing miR-124-3p only in the luminal epithelium was sufficient to block almost all embryo implantation. In both models however, where implantation occurred the decidual response was initiated as expected.\\nIn mice, blastocysts enter the uterus from the oviduct on the morning of E3 (E0 designated as morning of plug detection) and rapidly distribute themselves evenly along the uterine horn within a few hours (41). During this period, the uterus must be refractory to prevent early attachment which would result in all blastocysts being clustered together near the oviduct (42). We recorded a relatively high level of endogenous miR-124-3p at E3 wildtype mouse luminal epithelium which is likely a mechanism to prevent early attachment. By contrast, miR-124-3p was reduced at E4 wildtype mouse uterine luminal epithelium during the receptive window. This timing is similar to human endometrium where we demonstrated that miR-124-3p expression was low in the luminal epithelium during the receptive phase in women with normal fertility but abnormally increased in women with primary infertility. Our data suggested miR-124-3p needed to be downregulated in the endometrial epithelium to enable embryo attachment, likely by increasing factors required for endometrial receptivity in both humans and mice. We confirmed this using a customized array that identified the reduction of many adhesion related genes after miR-124-3p overexpression in both humans and mice. Of note, a spike in expression of certain long noncoding RNAs has been recorded during the receptive window in mice and humans (43, 44). Long noncoding RNAs can function as decoys to bind miRs to prevent their binding to target mRNAs essential for the transition of receptivity (45). For example, long noncoding RNA NEAT1 targets miR-124-3p (46) and its expression is significantly increased in the receptive phase human endometrium (44). Progesterone has also been shown to regulate miR expression (47) and conversely the local expression of miRs can attenuate progesterone signaling to prevent endometrial receptivity (48). It will be of interest in future studies to determine whether miR-124-3p expression changes are related to specific local noncoding RNAs and whether estrogen and progesterone directly or indirectly regulate its levels.\\nVery few studies have reported culturing human embryos for extended periods to investigate the effects on trophectoderm attachment and development. The available studies have mostly cultured human embryos on the cell culture plates or in Matrigel (49), up to day 11 to 12 postfertilization, the current legal limit (50, 51). It is known that embryos intimately interact and communicate with endometrial epithelial cells where they appose and adhere, prior to implanting (52). It is therefore highly likely that endometrial cells will influence how embryos attach and progress. While human embryos may have the ability to attach to culture wells and develop, development may be different when endometrial cells are present. Here, we cultured primary human trophectoderm spheroids on top of a primary human endometrial 3D bilayer treated with estrogen and progesterone. We altered the levels of miR-124-3p specifically in the endometrial epithelial cells to investigate how the endometrial cell changes influence early trophectoderm attachment and outgrowth in vitro. Our data overall demonstrated that overexpression of miR-124-3p specifically in the HEECs led to reduced firm attachment of the trophectoderm spheroid blocking its development before 120 h of culture (equivalent to day 11 postfertilization). This suggests that in vivo, high levels of miR-124-3p in endometrial epithelium leads to implantation failure and resorption of embryos. These observations concur with the role of miR-124-3p in cancer studies in which it functions as a tumor suppressor inhibiting cell migration and invasion (53). In comparison with previous studies where embryos are cultured on plastic dishes or in Matrigel (49, 51), our data suggested that the trophectoderm grew faster and expanded more rapidly in our coculture model. This remains to be confirmed by comparing embryos cultured on plastic, Matrigel, and endometrial cells in a single experiment. However, our data suggest that while embryos can develop in vitro without endometrial cellular compartments, their growth and development may not accurately reflect what occurs in utero in women. In our current study, we were unable to utilize human embryos with an intact inner cell mass. However, existing data on anembryonic pregnancies suggest that blastocysts without an inner cell mass are capable of attaching and implanting into the endometrium (54). However it remains unknown whether the attachment and early trophoblast development resemble those of intact embryos with an inner cell mass (55). In future studies it will be important to determine whether there are differences in trophectoderm adhesion between intact embryos to endometrial cells with elevated levels of miR-124-3p.\\nmiR-124-3p downregulated IQGAP1 in both human and mouse endometrial epithelium. IQGAP1 is a scaffold protein that belongs to IQGAP family with three members. While IQGAP2 and -3 have restricted localizations, IQGAP1 is more ubiquitously expressed throughout the body (56). IQGAP1 negatively modulates E-cadherin based cell junctions in the human breast epithelial carcinoma cells (57). Here, we further demonstrated that IQGAP1 reduced apical–basal polarity of the endometrial epithelial layer, which is essential to allow embryo attachment (9).\\nIn summary, this study provides strong evidence to support a key role for miR-124-3p in regulating endometrial receptivity by impairing endometrial epithelial cell adhesion and polarity. Targeting miR-124-3p with an inhibitor in HEECs from women with unexplained infertility improved their compromised adhesion, suggesting its potential as a treatment target for endometrial dysfunction. miR-124-3p may also be useful as a biomarker for diagnosing dysregulated receptivity. Patients diagnosed with endometritis have significantly higher serum levels of miR-124-3p when compared to healthy controls (18). Since some women with endometritis may have compromised fertility (58) this suggests that miR-124-3p levels in serum may reflect endometrial physiology. The insights gained from this study could lead to tests to identify women with infertility due to impaired receptivity and interventions to enhance adhesive capacity of endometrial epithelial cells, facilitating embryo adhesion and establishment of pregnancy.',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1073/pnas.2401071121',\n",
       "  'Journal': 'Proceedings of the National Academy of Sciences of the United States of America'},\n",
       " {'PaperTitle': 'Androgens modulate endometrial function',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Ko Yamagata, Yousuke Mizuno, Yumi Mizuno, Shunnsuke Tamaru, Takeshi Kajihara',\n",
       "  'Email': '',\n",
       "  'Abstract': 'Human endometrium is the major target tissue for sex steroid hormones. The circulating steroid hormones in normal ovulatory cycles accurately control the proliferation and differentiation of the endometrial cells. Androgens, such as testosterone and 5α-dihydrotestosterone, are a type of sex steroid hormones that mainly function in the differentiation, development, and maintenance of male sexual characteristics. Although androgens are best known for their role in male reproduction, the androgen receptor is present in both male and female reproductive organs and is essential for normal reproductive function. Recently, a series of evidence suggests that androgens contribute to endometrial physiology and pathologies. However, the roles of androgens in the endometrium remain insufficiently understood, with contradictory findings being reported. This review summarizes the studies that show the role of androgens in regulating the physiological conditions of the endometrium and the implantation process, and endometrial pathology including endometriosis and others.',\n",
       "  'Introduction': 'The human endometrium is the major target tissue for sex steroid hormones, including estrogen and progesterone. In normal ovulatory cycles, the proliferation and differentiation of the endometrial cells are accurately controlled by the circulating steroid hormones. Notably, the progesterone level increase in the postovulatory phase induces profound remodeling of the estrogen-primed endometrium, characterized by significant growth and coiling of the spiral arteries, secretory transformation of the glands, and decidualization of the stromal compartment. Successful implantation depends on the interaction between a well-developed embryo and a “receptive” endometrium. The duration of endometrial receptivity in the cycle is limited, designated as the “implantation window,” wherein the endometrium allows blastocyst implantation. The endometrium becomes receptive approximately 6\\xa0days after ovulation and remains receptive for up to 2–4\\xa0days [\\n1\\n–\\n4\\n].\\nAlthough androgens are best known for their role in male reproduction, the androgen receptor (AR) is also present in female reproductive organs and is essential for normal reproductive function [\\n5\\n,\\n6\\n]. Furthermore, androgens have been reported to contribute to endometrial physiology and pathologies such as endometriosis and endometrial cancer (EC). However, relative to the role of estrogen and progesterone on physiological pathological endometrium, information on the roles of androgens in the endometrium is still limited, with contradictory findings being reported. In this review, we summarize the studies that describe the role of androgens in regulating the physiological conditions of the endometrium and the implantation process and focus on the association between androgens and endometrial pathology, including endometriosis and EC.',\n",
       "  'Methods': 'Search strategy and selection criteria\\nWe conducted a narrative analysis in the literature review. The PubMed and Google Scholar were searched for literature published up to March 31, 2024, combining the following keywords: “Androgen,” “Endometrium,” “Implantation,” “Decidualization,” “Endometriosis,” and “Endometrial Cancer.”',\n",
       "  'Results': 'Characters and roles of androgen\\nAndrogens, which include testosterone and 5α-dihydrotestosterone (DHT), are a type of sex steroid hormones that mainly participate in the differentiation, development, and maintenance of male sexual characteristics. Androgen has a four-ring structure with C17 carbon, indicating a typical steroid skeleton, which is important in the biological activity as a hormone; it also has ketone and hydroxyl groups. Androgen is mainly produced in the testis; adrenal glands and ovary in females also produce it. Initially, cholesterol is transferred to pregnenolone in the mitochondria by the cholesterol side-chain cleavage enzyme P450scc (CYP11A1) [\\n7\\n]. Then, pregnenolone is hydroxylated to 17α-hydroxypregnenolone by the enzyme 17α-hydroxylase (P450c17) in the adrenal cortex [\\n7\\n]. 17α-hydroxypregnenolone is then converted to dehydroepiandrosterone (DHEA) by the 17,20-lyase activity of P450c17 [\\n7\\n]. DHEA is further converted to androstenedione by 3βHSD, and finally androgens such as testosterone are produced by enzyme AKR1C3 [\\n7\\n]. There are also various mechanisms to degrade or convert androgen to the other metabolites. Those mechanisms are important for maintaining androgens as appropriate levels, and have crucial roles in various physiological processes, such as reproductive health in both male and female, influencing sexual behavior, bone density, and fat distribution, and so on. Testosterone is converted to dihydrotestosterone (DHT) by the enzyme 5α-reductase [\\n8\\n]. Aromatase, encoded by the Cyp19a1 gene, catalyzes the conversion of testosterone to estradiol in various tissues, including the brain and vascular endothelium in mice, which is essential for regulating male sexual behavior, and have crucial process in male vascular endothelium [\\n9\\n,\\n10\\n].\\nAndrogen is widely distributed throughout the body, affecting the entire body. Its function is directly related to male sexual formation, maintenance of male secondary sexual characteristics, development of sexual desire, and sperm development and fertility [\\n11\\n]. The androgen signaling is also involved in various physiological pathways beyond male sex differentiation, such as muscle formation, body hair formation, and vocal cord development, through various tissues and organs. Moreover, it influences red blood cell production through erythropoietin stimulation, and blood lipid amount. Androgen is also produced in females as well [\\n12\\n], performing several functions, such as muscle and bone density maintenance and sexual desire improvement. Moreover, androgen signaling is related to the pathogenesis of several cancer types, especially prostate cancer [\\n13\\n,\\n14\\n]. It stimulates prostate cancer cells to proliferate, resulting in rapid tumor growth. It also inhibits apoptosis, allowing the tumor to survive [\\n13\\n,\\n14\\n]. In addition, some breast and ovary cancer types are promoted by androgen signaling via tumor growth progression and apoptosis inhibition. Several approaches are attempted to prevent and treat cancers by inhibiting androgen signaling. As described above, androgen is related to various physiological and pathological processes in the body.\\nAR and the signaling pathways\\nAR is a nuclear receptor protein. Androgen specifically binds to AR. Once activated, the AR transfers the stimulation of androgens in cells [\\n15\\n]. AR is mainly expressed in the cells of male reproductive organs, including the testes and prostate, but it is also found in other tissues, such as the skeletal muscle, skin, adrenal ground, and ovary in females, transmitting various stimuli within the body. In cells, AR normally exists in cytosols as a monomer. When bound with androgen, AR is activated, subsequently changing to a dimer [\\n16\\n], which subsequently moves into the nucleus. Androgen-bound AR dimer in the nucleus is then bound to specific DNA sequences [androgen response elements (AREs)]; thereafter, the expression of specific genes is activated or inhibited. The target genes whose expression is regulated by androgen signaling can be identified mostly by the chromatin-immune precipitation method [\\n17\\n–\\n19\\n]. Of them, prostate-specific antigen (\\nPSA\\n) is a major target gene of androgen signaling.\\nPSA\\nis mainly expressed in the prostate and is involved in the prostate structure and function by secretion as a part of seminal fluid. In particular, it helps enhance sperm motility and liquefy the seminal fluid to increase the possibility of fertilization [\\n20\\n]. Insulin-like growth factor 1 (\\nIGF-1\\n), erythropoietin (\\nEPO\\n), and succinate dehydrogenase are also the target genes of androgen signaling. Thus, by controlling the expression of multiple target genes, androgen signaling can be involved in several physiological and pathological pathways in living cells, affecting many individual body regulations.\\nAndrogens regulate the menstrual cycle in the endometrium and endometrial decidualization\\nThe endometrium is a major target tissue of sex steroid hormones, including estrogen and progesterone. Androgens play an important role in male reproduction. However, both male and female reproductive organs express AR, which is essential for both normal reproductive functions [\\n5\\n,\\n6\\n]. Although both the epithelial and stromal compartments of the endometrium express the estrogen receptor and progesterone receptor, AR expression is predominantly localized to the stromal cells [\\n21\\n,\\n22\\n]. AR expression in the endometrium decreases steadily from the early proliferative phase to the mid-secretory phase [\\n21\\n]. Serum androgen levels change throughout the menstrual cycle, with levels peaking during ovulation [\\n23\\n,\\n24\\n]. However, tissue dehydroepiandrosterone, androstenediol, androstenedione, and testosterone concentrations increase approximately by fourfold in secretory without significantly changing the plasma level [\\n25\\n]. Taken together, androgens could directly affect human endometrial functions.\\nDecidualization is the morphological and biological differentiation transformation of human endometrial stromal cells (HESCs) into specialized secretory cells during the secretory phase of the menstrual cycle; it is further characterized by the influx of specialized immune cells (uterine natural killer cells and macrophages) into the stroma, and vascular remodeling [\\n26\\n]. This process helps form a functional feto-maternal interface by controlling endovascular trophoblast invasion and tissue homeostasis and conferring resistance to environmental stress signals, including protection against oxidative cell death [\\n27\\n]. If impaired, several reproductive disorders, including implantation failure and recurrent miscarriage, can occur.\\nAs mentioned above, AR expression is confined to the endometrial stromal compartment and is most pronounced during the proliferative phase. After ovulation, the AR decreases during the secretory phase [\\n21\\n,\\n22\\n,\\n28\\n], although its expression remains in the decidua of early pregnancy. Additionally, the role of AR signaling in early pregnancy is brought about by the observation that treatment of rats with anti-androgens delays implantation initiation, fetal development, and parturition. Furthermore, anti-androgens inhibit decidualization in pseudo-pregnant rats [\\n29\\n]. Although the AR decreases, the HESCs become increasingly responsive to androgens as they differentiate into decidual cells. The AR in decidual HESCs also regulates a relatively small but distinct group of genes involved in cytoskeletal organization, cell motility, and cell cycle progression [\\n30\\n]. Androgens were observed to enhance the expression of decidualization markers such as prolactin, IGFBP1, and FOXO1 and promote morphological and ultrastructural changes associated with the decidual process (e.g., expanded endoplasmic reticulum and increased numbers of mitochondria and lipid droplets) (Fig.\\n1\\n) [\\n31\\n,\\n32\\n]. Furthermore, androgens significantly decreased H\\n2\\nO\\n2\\n-induced apoptosis in decidualized HESCs dose-dependently accompanied with the increased expression of superoxide dismutase 2 (SOD2), which protects against oxidative stress [\\n31\\n]. Interestingly, DHT reportedly has no significant effect on trophoblast invasion in a co-culture system using the spheroid of HTR-8/Svneo trophoblast cells and decidualized HESCs. However, if HESCs were first decidualized in the presence of androgen, spheroid expansion was further stimulated [\\n33\\n]. Therefore, androgens may regulate the expansion and invasion of trophoblast cells through the decidual phenotype of HESCs. Taken together, androgens might play an important role in the menstrual cycle and the decidual process of endometrium (Fig.\\n2\\n). However, additional investigations are needed to elucidate the precise role of androgens in endometrial decidualization.\\nAndrogens in the embryo and endometrium during implantation\\nSuccessful human implantation depends on the interaction between a developed embryo and a “receptive” endometrium. Endometrial receptivity occurs during a limited period in the cycle, considered as the “implantation window,” wherein the endometrium allows the blastocyst to be implanted. The endometrium becomes receptive approximately 6\\xa0days after ovulation and remains so for up to 2–4\\xa0days [\\n34\\n]. The implantation process is generally divided into three steps: apposition (the blastocyst is oriented correctly), adhesion (the blastocyst comes into contact with the epithelium and attaches to the endometrial surface), and invasion (the blastocyst penetrates the epithelium, invading the endometrial stroma) [\\n35\\n]. Aside from ovarian estrogen and progesterone, endometrial autocrine or paracrine factors and embryo-derived signals are also important for this process.\\nPolycystic ovaries syndrome (PCOS) affects approximately 5% of all women of reproductive age [\\n36\\n]. Its main clinical characteristics include hyperandrogenemia, oligo- or anovulation, infertility, enlarged polycystic ovary, hirsutism, obesity, and insulin resistance. Although the endocrine and clinical presentation of PCOS can be heterogeneous, hyperandrogenemia is the most responsible biochemical abnormality. Although ovulation occurs in approximately 80% of patients treated for PCOS, the pregnancy rate is only 40–50% [\\n37\\n]. Additionally, the miscarriage rate is reportedly 30–50% of all conceptions [\\n38\\n,\\n39\\n]. The expression level of AR in the endometrium of patients with PCOS was elevated in comparison with that of fertile controls [\\n40\\n]. Furthermore, the putative biomarkers of endometrial receptivity, including αvβ3 integrin, glycodelin, estrogen receptor α, and HOXA10, are aberrantly expressed in patients with PCOS [\\n41\\n–\\n43\\n]. Thus, a disruption of endometrial development and receptivity may link to the decreased fertility and poor reproductive outcome in these patients. Moreover, high plasma concentrations of androgens are associated with adverse reproductive outcomes, including infertility and increased incidence of pregnancy loss with or without PCOS [\\n44\\n,\\n45\\n]. Intriguingly, a recent cohort study based on Swedish nationwide register data showed that early initiation of anti-androgen treatment increased the chance of childbirth in patients with PCOS after spontaneous conception [\\n46\\n]. In pregnant rats, treatment with DHT (a more potent metabolite of testosterone) induced deficiencies in endometrial receptivity and mitochondrial function [\\n47\\n]. Furthermore, the expression of Wilms tumor suppressor (WT1), which is expressed during the window of implantation, in the endometrium was downregulated by androgens [\\n48\\n]. All of these observations point toward an adverse effect of androgens on endometrial function, which, in turn, may account for the association between hyperandrogenemia and subfertility or recurrent miscarriage. However, the precise mechanisms for hyperandrogenism and implantation remain unclear.\\nAndrogens and endometriosis\\nEndometriosis is one of the most prevalent gynecological disorders, affecting approximately 10% of women of reproductive age and with a prevalence rate of as high as 35–50% in women with endometriosis-associated infertility and/or pain [\\n49\\n]. The etiology of endometriosis has been explained by several theories. Of these theories, the most common and acceptable is the retrograde reflux of menstrual blood containing endometrial tissue via the fallopian tubes into the peritoneal cavity where it attaches to the peritoneum, proliferates, differentiates, and eventually invades the underlying tissue [\\n50\\n]. Although 90% of women of reproductive age have retrograde menstruation, only approximately 10% is diagnosed with endometriosis [\\n51\\n]. Retrograde menstrual flow is common, but it does not explain why only some women develop endometriosis. Therefore, other pathologic factors are required to establish this disease. The eutopic endometrium of women with endometriosis is believed to be abnormal, predisposing them to ectopic diseases. The phenotype for the differentiation capacity of ectopic endometrium is significantly varied in comparison with that of eutopic endometrium [\\n52\\n]. However, the pathogenesis of endometriosis has not been fully characterized.\\nThe concentration of testosterone in endometriosis lesions was strikingly higher than that in the corresponding serum concentrations and eutopic endometrium of healthy controls [\\n53\\n]. Carnerio et al. demonstrated that AR and 5α-reductase, which is an enzyme essential for converting testosterone into the more potent androgen DHT, are localized in the cytoplasm of glandular and stromal cells of the ectopic endometrium [\\n54\\n]. Thus, active androgens may be formed in endometriosis tissue, and both local and systemic androgens may contribute to establishing and developing endometriosis. Additionally, a bioinformatic analysis identified AR as a key endometriosis-associated transcription factor, with 373 target AR genes significantly differentially expressed in endometriotic lesions compared with those in the normal endometrium [\\n55\\n]. Interestingly, polymorphic CAG repeats of AR genes may be related to the pathogenesis of endometriosis [\\n56\\n–\\n58\\n].',\n",
       "  'Discussion': '',\n",
       "  'Conclusion': 'This review highlights the studies demonstrating that androgens and their receptors play a crucial role in regulating the endometrial physiological and pathological conditions. Androgens enhance the decidual process in the endometrium. Conversely, several experiments point toward an adverse effect of androgens on implantation, which, in turn, may account for the association between hyperandrogenemia and subfertility or recurrent miscarriage. Therefore, the exact role of androgens on endometrial receptivity remains unclear. Additional studies are needed to confirm whether the physiological and supra-physiological concentrations of androgen have beneficial or adverse effects on endometrial receptivity. Meanwhile, our review also discusses the findings of studies on androgens in gynecological pathological conditions such endometriosis. Currently, the role of androgens in physiological and pathological endometrial functions is still controversial, requiring further clarification. Extensive basic and clinical research studies are required to elucidate the expression, regulation, and functions of androgens under normal and disease conditions to identify new biomarkers and robust therapeutic applications of androgens.',\n",
       "  'URL': 'https://doi.org/10.1007/s00795-025-00430-6',\n",
       "  'Journal': 'Med Mol Morphol'},\n",
       " {'PaperTitle': 'Inactivation of branched-chain amino acid uptake halts Staphylococcus aureus growth and induces bacterial quiescence within macrophages',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Adriana Moldovan, Ronald S Flannagan, Marcel Rühling, Kathrin Stelzner, Clara Hans, Kerstin Paprotka, Tobias C Kunz, David E Heinrichs, Thomas Rudel, Martin J Fraunholz',\n",
       "  'Email': 'martin.fraunholz@uni-wuerzburg.de; thomas.rudel@uni-wuerzburg.de',\n",
       "  'Abstract': 'Staphylococcus aureus is a notorious human pathogen that thrives in macrophages. It resides in mature phagolysosomes, where a subset of the bacteria eventually begin to proliferate. How S. aureus acquires essential nutrients, such as amino acids, for growth in this niche is poorly understood. Using a long-term primary human macrophage infection model, we show that branched-chain amino acid (BCAA) uptake mediated by the major transporter BrnQ1 is required by S. aureus for intracellular replication in macrophages and we provide mechanistic insight into the role of BCAAs in the success of intracellular S. aureus. Loss of BrnQ1 function renders intracellular S. aureus non-replicative and non-cytotoxic. The defective intracellular growth of S. aureus brnQ1 mutants can be rescued by supplementation with BCAAs or by overexpression of the BCAA transporters BrnQ1 or BcaP. Inactivation of the CodY repressor rescues the ability of S. aureus brnQ1 mutants to proliferate intracellularly independent of endogenous BCAA synthesis but dependent on BcaP expression. Non-replicating brnQ1 mutants in primary human macrophages become metabolically quiescent and display aberrant gene expression marked by failure to respond to intraphagosomal iron starvation. The bacteria remain, however, viable for an inordinate length of time. This dormant, yet viable bacterial state is distinct from classical persisters and small colony variants.',\n",
       "  'Introduction': 'Staphylococcus aureus is a Gram-positive prominent human opportunistic pathogen [1], capable of causing a variety of diseases ranging from superficial skin infections to life-threatening conditions (reviewed in [2]). It permanently colonizes 20–30% of the human population [3, 4], with up to 60% carrying S. aureus intermittently during their lifetime [5]. S. aureus survives and replicates in a variety of host cell types including non-professional phagocytes (such as epithelial and endothelial cells) and professional phagocytes such as macrophages (reviewed by [6, 7]). Recently it has been proposed that the ability of S. aureus to exist intracellularly is a prevailing feature of the S. aureus-host interaction [8]. Among professional phagocytes, macrophages play a central role in innate immunity due to their ability to phagocytose bacteria and mount potent microbicidal responses. Indeed, macrophages deploy a multitude of antibacterial effectors such as reactive oxygen species, proteases, lipases, and antimicrobial peptides that function to intoxicate and/or degrade phagocytosed prey [9]. At the same time, macrophages curtail bacterial growth by extruding essential nutrients such as iron and manganese from the phagosome lumen via the host transporter NRAMP1, which is just one aspect of host nutritional immunity [10]. In addition to metals, macrophage phagolysosomes are presumably depleted of several amino acids owing to the activity of host amino acid transporters (e.g., SLC38A9, SLC66A1/PQLC2, SLC36A1, and SLC66A4/cystinosin) localised to phagosomal or lysosomal membranes, where they extrude amino acids including leucine, neutral amino acids, cationic amino acids and cysteine, respectively, into the host cell cytosol [11]. Together, metal and amino acid transport should deplete the phagolysosome of nutrients thereby hindering bacterial proliferation within macrophages. Despite this, macrophages fail to eradicate phagocytosed S. aureus [12–15] and following engulfment, bacteria can begin to replicate within mature phagolysosomes [16, 17] independently of toxin production [18]. Branched-chain amino acids (BCAAs) - isoleucine (Ile), leucine (Leu) and valine (Val) - are essential amino acids that cannot be synthesized by mammals. In S. aureus, BCAAs are not only proteinogenic amino acids, but also serve as precursors for branched-chain fatty acids (BCFAs), which are required for the regulation of membrane fluidity [19]. In addition, Ile functions as a signalling molecule that, together with GTP [20, 21], modulates the activation of CodY, a global transcriptional regulator of metabolism and pathogenesis in most low G + C Gram-positive bacteria including S. aureus [22]. BCAA transport and biosynthesis are therefore closely linked to virulence in S. aureus and other bacterial pathogens, indicating that access to BCAAs as a critical nutrient is required by pathogens to establish infection [23]. For example, when grown in vitro in the absence of Val, S. aureus displays a growth lag of ~20h, until BCAA biosynthesis is initiated [21]. To facilitate immediate BCAA uptake from the environment, S. aureus employs the BCAA transporters, BrnQ1 and BcaP, as well as BrnQ2, a dedicated Ile transporter [24]. In addition, S. aureus harbours BCAA biosynthetic genes encoded by the ilvDBNC-leuABCD-ilvA operon and ilvE. Both BCAA uptake and biosynthesis operons are repressed by CodY when intracellular BCAA levels are sufficient to support growth [21]. CodY controls the expression of over 200 genes in S. aureus [25–27] where it acts primarily as a repressor of metabolic and virulence genes upon binding of Ile and GTP. Upon nutrient depletion, CodY target genes were shown to be de-repressed sequentially, along the BCAA gradient, with genes involved in amino acid uptake (including BCAAs) being turned on much earlier than biosynthesis genes [28].The requirement for BCAA acquisition for S. aureus pathogenesis is underlined by the fact that brnQ1/bcaP mutants are attenuated in murine models of infection [24, 29]. We here reveal a critical role for high-affinity BCAA transport in the growth and adaptation of S. aureus to the macrophage phagolysosome. Using defined mutants lacking specific BCAA transporters or the ability to synthesize BCAA, we uncover a pivotal role for the high-affinity BCAA importer BrnQ1 in the replication and subsequent survival of S. aureus within macrophages. We provide evidence that brnQ1- deficient S. aureus display a phagolysosomal growth defect that can be overcome by inactivation of the transcription factor CodY, by exogenous supply of BCAAs, or by heterologous expression of either BrnQ1 or the secondary BCAA transporter BcaP. In the absence of BCAA supplementation of the macrophage culture medium, brnQ1 bacteria remain intracellular for an inordinate duration and adopt a dormant state, that is distinct from established persister cell or small colony variant (SCV) phenotypes.',\n",
       "  'Methods': 'Ethics Statement Leukocyte samples (i.e., leukoreduction system chambers from anonymous healthy donors) were obtained from the Institute for Clinical Transfusion Medicine and Haemotherapy, of the University of Würzburg, Germany for research purposes and require no specific approval by an ethics committee. Whole human blood was also drawn from healthy adult volunteers upon verbal consent in accordance with protocols approved by the University of Western Ontario’s Research Ethics Board, London, Ontario, Canada. Bacterial culture conditions Bacterial strains are listed in S1 Table. All liquid cultures were grown aerobically at 37°C with 200\\u2009rpm shaking. E. coli was grown in Luria-Bertani broth (LB) or on LB agar containing 100 µg/mL ampicillin for plasmid maintenance when necessary. S. aureus strains were routinely grown in tryptic soy broth (TSB), or on TSB agar plates (TSA) if not stated otherwise. Media were supplemented with appropriate antibiotics, when necessary: 10 µg/ml chloramphenicol (plasmid encoded antibiotic resistance); and/or 5 µg/ml erythromycin, 5 µg/mL tetracycline or 50 µg/mL kanamycin (for chromosomally encoded antibiotic resistance). Construction of bacterial strains and plasmids All used strains (S1 Table), plasmids (S2 Table) and oligonucleotides (S3 Table) can be found in the Supporting material. Strains used for macrophage infection experiments shown in Fig 1c were retrieved from the Nebraska Transposon mutant library [34]. The S. aureus insertional transposon mutant of brnQ1 (NE945) was retrieved from the Nebraska Transposon mutant library, transposon insertion at the correct position was confirmed by PCR, using primers brnQ1_fwd and brnQ1_rev and the mutation was further transduced via phage φ11 or φ80 into the genetic background of wild-type S. aureus strains to yield JE2 brnQ1, 6850 brnQ1, SH1000 brnQ1 and RN4220 brnQ1 respectively. Plasmid pbrnQ1 was generated to complement the brnQ1 gene in trans under the transcriptional control of its native promoter, using Fast Cloning [80]. Briefly, the open reading frame of brnQ1 gene (locus ID: SAUSA300_0188/ SAUSA300_RS00985) was amplified from S. aureus JE2 genomic DNA together with its native promoter region included in the 400\\u2009bp upstream of the start codon, using primer pairs brnQ1_Comp_Inf_fwd and brnQ1_Comp_Inf_rev. The pALC2085 vector [81] was linearized by PCR using primer pairs p2085_Inf_for brnQ1_fwd and p2085_Inf_for brnQ1_rev. Both insert and vector were amplified using an 18-cycle amplification protocol and Phusion DNA-polymerase (Thermo Fisher Scientific, Cat. No. F530L). 1 µL DpnI enzyme (Thermo Fisher Scientific, Cat. No. ER1701) was added to each 50 µL un-purified PCR reaction, vector and insert were mixed in a 1:1 (v/v) ratio and DpnI digest was carried out for 1h at 37°C. 2 µL of the reaction was directly transformed into electrocompetent E. coli DH5α and plated on LB agar plates supplemented with 100 µg/mL ampicillin. The plasmid pbrnQ1 was isolated from E. coli and the sequence was confirmed by Sanger sequencing (SeqLab, Göttingen). Subsequently, plasmids were electroporated into S. aureus RN4220 and further transduced into S. aureus JE2 using phage φ11. A similar procedure was used to generate a control, by amplifying a promoter-less brnQ1 using primer pairs brnQ1_Comp_Inf_fwd and brnQ1_P-less_Comp_Inf_rev and cloning it into the pALC2085 backbone to yield pPless_brnQ1. Plasmids pbcaP and pbrnQ1 shown in Fig 4d were previously described [24, 29] and carry bcaP or brnQ1, respectively, that were cloned into the expression plasmid pRMC2 under the control of their respective native promoters. Red-fluorescent or cyan-fluorescent expressing S. aureus JE2 were generated by transducing the pSarAP1-mRFP or pCerulean, respectively, from recombinant S. aureus RN4220 strains using phage φ11. S. aureus JE2 fur and JE2 brnQ1/fur were generated by transducing the tet cassette from S. aureus MJH010 fur::tet [40] using phage φ11. Strains in USA300 carrying either codY (locus ID: SAUSA300_1148/ SAUSA300_RS06210) or ilvD (locus ID: SAUSA300_2006/ SAUSA300_RS11035) mutations were created by transducing either codY::Tn (NE1555) or ilvD::Tn-KanR using phage φ80 into the recipient strain. The replacement of the erythromycin resistant cassette in strain JE2 NE718 (ilvD) was done using the plasmid pKAN as previously described [82]. In brief, pKAN plasmid was transduced into JE2 NE718 (ilvD) using phage φ80. Replacement of the EryR cassette for KanR was confirmed by resistance phenotype (i.e., the strain became EryR and KanR) and by PCR analysis. Isolation of primary human monocytes from human peripheral blood Primary human macrophages were derived from peripheral blood mononuclear cells (PBMCs) isolated from whole blood leukoreduction system (LRS) chambers [83] using the SepMate-50 system (StemCell Technologies, Cat. No. 85450) and Ficoll-Paque PLUS density gradient medium (Sigma Aldrich, Cat. No. GE17-1440-03), according to manufacturer’s instructions. Monocytes were then purified from the PBMC fraction using the EasySep Human CD14 Positive Selection Kit II (StemCell Technologies, Cat. No. 17858) according to the manufacturer’s instructions. In some instances PBMCs were isolated from whole human blood using lympholyte-poly (Cedarlane labs) according to the manufacturer’s instructions and as described [39]. Macrophage differentiation and polarization All macrophage cultivation steps, and infection experiments were done in RPMI1640 (Thermo Fisher Scientific, Cat. No. 72400054) supplemented with FBS (10% v/v; heat inactivated: 56°, 30\\u2009min) henceforth termed Infection Medium, unless otherwise indicated. Purified monocytes were seeded in Nunc UpCell cell culture plates (Thermo Fisher Scientific, Cat. No. 174902) at a density of 5-7x 106 cells/plate in Infection Medium supplemented with 50\\u2009ng/mL macrophage-colony-stimulating factor (M-CSF; StemCell Technologies, Cat. No. 78057) and incubated at 37°C in a humidified incubator with 5% CO2 for 7 days. Following the 7-day differentiation protocol, macrophages were detached from the Nunc UpCell by incubation at 20°C, following the manufacturer’s instructions. Cells were collected, centrifuged, re-suspended in Infection Medium, counted using a Neubauer counting chamber and seeded in cell culture well plates. In some instances isolated PBMCs were adhered directly to 12 well tissue culture treated plates and differentiated in the presence of recombinant human M-CSF at 10\\u2009ng/mL (PeproTech, Cat. No. 300–25) as previously described [39]. Differentiated macrophages were used for infections from day 7 to day 9. Macrophage infection assay For standard infection experiments, 105 cells were seeded into 24 well microtiter plates, 24h prior to infection. Bacterial overnight cultures (TSB) were diluted to an OD600\\u2009=\\u20090.5 and grown in TSB for 45\\u2009min at 37 °C, under continuous shaking at 200\\u2009rpm. Bacteria were washed with Dulbecco’s phosphate-buffered saline (DPBS, Thermo Fisher Scientific, Cat. No. 14190169), diluted in Infection Medium and counted using a Thoma counting chamber (Hartenstein, Germany, Cat. No. ZK04). For counting, 10 µL of the above-described suspension were loaded onto the Thoma chamber (small counting square area: 0.0025\\u2009mm2; depth 0.1\\u2009mm). Bacteria were allowed to sediment for 10\\u2009min, then counted in at least 16 small squares and the “chamber factor” of 4 x 106 was used to estimate no. of bacteria/mL. Non-opsonized bacteria were used to infect macrophages at a multiplicity of infection (MOI) of 5. For the synchronization of infection, plates were centrifuged at 200 x g, for 5\\u2009min, without brakes and then incubated for 30\\u2009min at 37°C in an incubator with humidified atmosphere with 5% CO2. After 30\\u2009min, extracellular bacteria were removed by 15\\u2009min incubation in Infection Medium containing 30 µg/ml lysostaphin (AMBICIN L, AMBI), followed by washing twice with sterile DPBS and further incubation in Infection Medium without antibiotics until the end of the experiment. For long-term experiments, viable infected macrophages were cultivated for up to 28 days in the absence of lysostaphin (except for the initial 15 min pulse). In some instances, primary human and RAW 264.7 macrophage infections were performed using 12 well tissue culture treated dishes as previously described [12]. Here macrophages were infected at an MOI of 5 in serum-free RPMI1640. After synchronization by centrifugation at 277 x g for 2\\u2009min, phagocytosis was allowed to occur for 30\\u2009min. At this time the medium was replaced with serum-free RPMI1640 containing 100 μg/mL gentamicin for 1\\u2009hr, to kill the extracellular bacteria, followed by washing twice with sterile DPBS and further incubated with Infection Medium. For RAW 264.7 macrophage experiments, Infection Medium with 5% (v/v) FBS was used instead. Intracellular bacterial replication assay (CFU) To enumerate intracellular bacteria, at the required time points, cells were washed once with DPBS and lysed using alkaline water (pH11) for 5\\u2009min at room temperature. Dilution series of the complete cell lysate were plated on TSA and recovered bacterial colony forming units (CFU) were counted. “T0” represented in all graphs corresponds to 45\\u2009min p.i. (i.e., 30\\u2009min phagocytosis and 15\\u2009min lysostaphin treatment). In some instances, after gentamicin treatment and DPBS washing, infected macrophages were lysed using 0.5\\u2009mL of 0.1% (v/v) Triton X-100 diluted in DPBS. Lysis at this time equals 1.5h p.i. and the CFU/mL represents the number of phagocytosed or gentamicin protected bacteria. At 20h p.i. supernatants from all wells were collected individually and centrifuged to pellet any bacteria that may have emerged from infected macrophages. While centrifuging, wells containing macrophages were lysed using Triton X-100 as described immediately above, and this lysate was used to resuspend any pelleted bacteria. At each time point bacteria were serially diluted onto TSA and the CFU determined. To calculate the fold change at 20h p.i., the CFU/mL at 20\\u2009h was divided by CFU/mL at 1.5h. The data were expressed as the Log10 fold change. Exogenous BCAA complementation Macrophages were seeded 24h prior to infection in either Infection Medium or Infection medium supplemented with 1\\u2009mM of each branched-chain amino acid (Ile, Leu, Val) and all infection steps were carried out in the respective medium. Samples were further processed for CFU counting or RNA extraction at 45\\u2009min (“T0”), 10h and 24h p.i. (CFU) or 10h p.i. (RNA). In some instances, 1mM of either Val, Leu or both were added to RPMI1640\\u2009+\\u20095% FBS that was added back to RAW 264.7 macrophages immediately after gentamicin treatment. CFU determination was performed as described above. Live Cell Imaging 4.5 x 104 macrophages/well were seeded in 8 well chamber µ-slides (ibidi, Cat. No. 80826) 24h prior to infection. Infection with mRFP-expressing bacteria was performed at an MOI of 5 as described above. Activation of effector caspases 3/7 in infected macrophages was visualised 5h p.i. by the addition of 5 µM of the fluorogenic substrate Cell Event Caspase-3/7 Green Detection Reagent (Thermo Fisher Scientific, Cat. No. C10723).Time-lapse imaging was performed in Imaging Medium (Infection Medium without phenol red), in the absence of antibiotics, on a Leica TCS SP5 confocal laser scanning microscope (Leica GmbH, Wetzlar, Germany; software LAS AF version 2.7.3.9723) using a 40x oil immersion objective (Leica HC PL APO, NA\\u2009=\\u20091.3), equipped with a pre-warmed (37°C) live-cell incubation chamber (Life Imaging Systems) and perfused with a humidified atmosphere containing 5% CO2. The LAS AF software was used for setting adjustment and image acquisition. All images were acquired at a resolution of 1024x1024 and recorded in 8-bit mode, at 30\\u2009min time intervals. Z-stacks were imaged with a step size of 0.4 µm. Image-processing was performed using Fiji [84]. Phase contrast microscopy Phase contrast microscopic images of live, infected cells were acquired with a LEICA DM IRB microscope connected to a SPOT Pursuit USB camera using a Leica C Plan L 20 x/0.30 PH1 objective and VisiView software (Visitron). Transmission Electron Microscopy (TEM) TEM sample preparation and imaging were carried at the Imaging Core Facility of the Biocenter, University of Würzburg, Germany. Macrophages were seeded on cover slips and infection was performed as described. Infected samples were washed once in DPBS, fixed with 2.5% glutaraldehyde (in 50\\u2009mM sodium cacodylate [pH 7.2], 50\\u2009mM KCl and 2.5\\u2009mM MgCl2) for 2h at 4°C. Samples were subsequently washed twice for 5\\u2009min with 50\\u2009mM cacodylate buffer (pH 7.2) and were further fixed with 2% buffered OsO4 (Science Services) and contrasted with 0.5% uranyl acetate in distilled water and finally embedded in Epon (Serva) after ethanol-based dehydration. Ultrathin sections were generated from embedded samples and contrasted with 2% uranyl acetate in ethanol and lead citrate. TEM imaging was performed on a JEOL JEM-2100 instrument. Preparation of cells for RNA extraction For total RNA extraction (host and pathogen), 3 x 106 macrophages grown in 6-well cell culture plates (i.e., 1.5 x 106 macrophages were seeded per well and 2 wells/condition were pooled) were infected at MOI of 5, as described. At 10h p.i., cells were washed once with DPBS, and RNAprotect Cell Reagent (Qiagen, Cat. No. 76526) was added immediately (1 part Infection Medium and 5 parts RNAprotect). Samples were stored at 4°C for a maximum of one week before further processing. Experiments with bacteria grown in Infection Medium were carried out in parallel, using the same bacterial inoculum. Bacteria were similarly grown in 6-well plates at 37°C in a humidified incubator with 5% CO2, without shaking. After 10h, bacteria were collected by centrifugation and pellets were flash-frozen in liquid N2 and stored at -80°C. For experiments depicted in Fig 3a, bacteria were grown overnight in TSB, washed 2x in DPBS and resuspended in Infection Medium at OD600\\u2009=\\u20091. Bacteria were grown for 1h at 37°C with 200\\u2009rpm shaking, collected and flash-frozen in liquid N2 and stored at -80°C. RNA extraction RNA extraction from either S. aureus-infected macrophages or cultures of S. aureus was done using the RNeasy Mini Kit (Qiagen, Cat. No. 74104). Infected cells stored in RNAprotect Cell Reagent were collected in RNase-free tubes, by centrifugation (6,200 x g, 5\\u2009min, 4°C). Frozen bacterial pellets were allowed to thaw on ice for 5\\u2009min. Next, 600 µL RLT Buffer (kit) supplemented with β-mercaptoethanol (10 µL) was added to the pellets, samples were disrupted in a FastPrep FP120 instrument (6 m/s for 45\\u2009sec, at 4°C) using Lysing Matrix B tubes (MP Biomedicals, Cat. No. 116911100), immediately cooled down and centrifuged (10,000 x g, 2\\u2009min, 4°C). Following lysis, manufacturer’s instructions for the RNeasy Mini kit were followed. TURBO DNA-free DNase (Thermo Fisher Scientific, Cat. No. AM1907) was used to digest contaminating DNA (1h, 37°), according to manufacturer’s instructions. For routine applications, concentration and purity were determined using a Nanodrop instrument and non-denaturing agarose gel electrophoresis, using ethidium bromide staining [85]. The absence of DNA contamination was verified by qPCR (40 cycles using gyrB-specific primers and RNA samples as template). RNA sequencing RNA-seq of S. aureus grown in Infection Medium as well as dual RNAseq of S. aureus-infected macrophages were conducted by the Core Unit SysMed at the University of Würzburg (Germany). RNA quality was checked using a 2100 Bioanalyzer with the RNA 6000 Nano kit (Agilent Technologies, Cat. No. 5067–1511). Ribosomal RNA (rRNA) was depleted from 200\\u2009ng of total RNA as input using Lexogen’s RiboCop rRNA Depletion Kit protocol according to manufacturer’s recommendation with the following modifications: for mixed human and bacterial samples, 5 µl of RiboCOP V1.2 probes were mixed with 5 µL of RiboCOP META probes, 4 µl hybridization buffer, and 21 µL of RNA sample. Following rRNA depletion, DNA libraries suitable for sequencing were prepared using CORALL Total RNA-Seq Library Prep protocol according to manufacturer’s recommendation with 14 PCR cycles. After quality control, libraries were sequenced on an Illumina NextSeq platform (1*75\\u2009bp single end run). Demultiplexed FASTQ files were generated with bcl2fastq2 v2.20.0.422. Sequencing reads were trimmed for Illumina adapter sequences using Cutadapt version 2.5 with default parameters. In addition, Cutadapt was given the –nextseq-trim\\u2009=\\u200920 switch to handle two colour sequencing chemistry and reads that were trimmed to length 0 were discarded. Reads were filtered with FastQScreen [www.bioinformatics.babraham.ac.uk/projects/fastq_screen/]. The splice-aware aligner STAR [86] was used for the mapping of reads originating from human cells to the human genome (GRCh38.p13), whereas the Reademption pipeline [87] was used to map reads originating from S. aureus to the reference (GCF_000013465.1_ASM1346v1). Read counts for each gene were generated and the count output was utilized to identify differentially expressed genes using DESeq2 v1.24.0 [88]. RT-qPCR 1 µg total RNA was converted to cDNA using the RevertAid First Strand cDNA Synthesis kit (Thermo Fisher Scientific, Cat. No. K1622) according to manufacturer’s instructions, using random hexamer primers. RT-qPCR was performed one a StepOne Plus instrument (Applied Biosystems). Each reaction mix consisted of GreenMasterMix (Genaxxon, Cat. No. M3052.0500), 900\\u2009nM of each primer and 50\\u2009ng (infected samples) or 10\\u2009ng (bacteria alone) cDNA. Relative gene expression was normalized to expression of the gyrB reference gene and represented as -ΔCt values [89]. Oligonucleotides used in RT-qPCR experiments are listed in S3 Table. Genome sequencing DNA was isolated using the Promega One4All kit on a KingFisher (ThermoFisher Scientific) device. DNA concentration was measured using Qubit (dsDNA HS assay kit, Thermo Fisher Scientific, Germany) and sequencing libraries were generated using the Nextera XT library Prep Kit (Illumina, USA). Data acquisition was performed on a NextSeq550Dx (Illumina, USA) during a 2\\u2009×\\u2009150\\u2009bp paired-end sequencing run using a mid-output cassette. The derived FastQ files were quality-controlled, and the genomes were assembled using SKESA (V2.4.0 on Seqsphere+ (client and server version 9.0.0, Ridom GmbH, Münster). Contigs were aligned and reordered using progressiveMauve [90] and the genomes were annotated using prokka [91]. Cerulean induction in infected macrophages Macrophages were seeded on coverslips and infected as described with JE2 WT or JE2 brnQ1 carrying pCerulean (anhydrous tetracycline (aTc)-inducible expression of the codon-adapted fluorescent protein Cerulean [81]). At either 1h or 10h p.i. Cerulean was induced by the addition of 300\\u2009ng/mL aTc, and samples were incubated for further 2h before fixing with 4% paraformaldehyde (PFA, Morphisto, Cat. No. 11762.01000). Fixed cells were washed 3x in DPBS, permeabilized with 0.2% Triton-X100 (in DPBS) for 5\\u2009min, followed by incubation with 50\\u2009mM ammonium chloride for 5\\u2009min. Staining was done for 20\\u2009min, with a solution containing 1 µg/mL BODIPY FL Vancomycin (Thermo Fisher Scientific, Cat. No. V34850), Phalloidin-iFluor647 (Abcam, Cat. No. 176759; diluted according to manufacturer’s protocol) and 4 µg/mL Hoechst 34580 (Thermo Fisher Scientific, Cat. No. H21486), in DPBS. Cover-slips were mounted on microscopy slides with Mowiol and air-dried overnight. Image acquisition was performed on a Leica TCS SP5 CLSM using a 40x oil immersion objective (NA 1.4). Per sample at least 10 fields of view were recorded with each covering an area of 388\\u2009×\\u2009388 µm. Images were recorded at a resolution of 1024 x 1024 pixels at 8-bit grey scale depth. Imaging was done using the following excitation/emission (Ex/Em) settings: Cerulean: Ex/Em\\u2009=\\u2009458/460–494\\u2009nm (Argon laser); BODIPY FL Vancomycin: Ex/Em\\u2009=\\u2009488/499–586\\u2009nm; (Argon laser); Hoechst 34580: Ex/Em\\u2009=\\u2009405/423–477\\u2009nm (diode laser); Phalloidin-iFluor 647 Ex/Em\\u2009=\\u2009633/634–710\\u2009nm (HeNe laser). Image analysis was conducted using Fiji [84], using an in-house macro toolset, which uses BODIPY-FL- channel to identify the macrophage-associated bacteria, whose positions are saved as individual regions-of-interest (ROI). Subsequently, the mean fluorescence intensities in each ROI were determined for both the Cerulean channel and the BODIPY-FL channel. The results were exported from Fiji as text files with comma-separated values (csv). The csv-files fere transformed into text-format using RStudio (Integrated Development for R. RStudio, PBC, Boston, MA) and subsequently imported the data as dot plots into the flow cytometry software Flowing2 (Turku Bioscience Centre, Turku Finland) to determine percent of total bacteria expressing Cerulean. Fluorescence-based proliferation assay Fluorescence proliferation assays were performed as previously described [18, 92] using GFP expressing S. aureus USA300 and ∆brnQ1. In brief, bacteria were labelled with Cell Proliferation Dye eFluor 670 |(eBioscience, Cat. No. 65-0840-85) and used to infect macrophages that were grown on 18\\u2009mm glass coverslips (No. 1) at an MOI of 5 as described above. At 1.5, 10 and 20h p.i. macrophages were stained with 1 µg/mL tetramethylrhodamine-conjugated wheat germ agglutinin (TMR-WGA, Thermo Fischer Scientific, Cat. No. W849). Stained cells were fixed with 4% (v/v) PFA at room temperature for 20\\u2009min and mounted on glass microscope slides using ProLong Gold Antifade Mountant (Thermo Fischer Scientific, Cat. No. P36934). Confocal fluorescence microscopy was performed using a Zeiss LSM 880 with Fast AiryScan equipped with a Plan-Apochromat ×63/1.4 oil DIC M27 objective and Definite Focus 2. GFP, TMR-WGA and eFluor 670 were excited with available laser lines at 488\\u2009nm, 561\\u2009nm, and 633\\u2009nm, respectively. Post-image processing (i.e., contrast enhancement and image cropping) was conducted using Fiji [84]. Bacterial colony morphology analysis S. aureus JE2 brnQ1 were recovered from infected macrophages by lysis with pH11 water. Colony pigmentation was assessed by plating the lysate onto TSA plates. Haemolysis was assessed by plating lysates onto Columbia agar with 5% defibrinated sheep blood. After 30h incubation at 37°C, plates were photographed and colony diameter was measured with Fiji [84]. Fresh bacteria grown in late-exponential phase in TSB served as control. Bacterial growth curves Bacterial growth curves were recorded in 48-well plates using a TECAN MPlex plate reader. S. aureus strains were grown overnight in the appropriate culture medium: TSB, RPMI1640 or Infection Medium. Next day, cultures were diluted to an OD600\\u2009=\\u20090.1 in fresh corresponding medium and used to inoculate 400 µL/well. Sterile medium was used as reference. For experiments depicted in S1 and S13a Figs, bacteria were grown overnight in TSB and washed 4x in DPBS before inoculation at OD600\\u2009=\\u20090.1, in a chemically defined medium (CDM) [[93] with 25\\u2009mM glucose]. BCAAs were excluded from the CDM formulation, as needed. Absorbance at 600\\u2009nm (OD600) was measured every 18\\u2009min for 20h or 50h, under continuous shaking at 200\\u2009rpm. For experiments depicted in S1 Fig, bacterial doubling times were estimated from growth curves. Bacterial growth in valine-free CDM For experiments depicted in S14 Fig, CDM was prepared as previously described [24] and Val was omitted when needed. Bacteria were grown overnight in TSB, pelleted and washed in 1\\u2009mL of sterile saline (0.9% w/v NaCl). Bacteria were diluted into saline at an OD600\\u2009nm of 0.1 and 10 μL of this suspension was used to inoculate 2mL culture volumes to give a starting OD600\\u2009nm of ~ 0.0005. The bacteria were grown in 13\\u2009mL polypropylene snap cap tubes at 37°C with constant shaking at 250\\u2009rpm. The end point OD600 was measured at 24h. As controls, either 1\\u2009mM Val or 0.1% (w/v) tryptone was added to the culture medium. S. aureus infection of non-professional phagocytic cells HeLa cells (HeLa 229, ATCC CCL-2.1) and 16HBE14o- human bronchial epithelial cells were routinely cultured in Infection Medium, at 37 °C and 5% CO2. For infection, 105 cells/well were seeded in 12-well microtiter plates 24h prior to infection. Infections were carried out with a multiplicity of infection (MOI) of 5 as described for macrophage infections. In this case, however, bacteria and host cells were co-cultivated for 1h, after which extracellular bacteria were removed by a 30\\u2009min treatment with 20 µg/ml lysostaphin (AMBICIN L, AMBI) followed by further incubation in medium containing 2 µg/ml lysostaphin until the end of the experiment. Phagosomal escape assay Phagosomal escape was determined as previously described [94]. Briefly, HeLa YFP-CWT or 16HBE14o- YFP-CWT cells were infected with S. aureus at MOI of 5 in 24 well µ-plates (ibidi, Cat. No. 82406). For experiments depicted in S4b, S4c and S5b Figs, mRFP-expressing bacterial strains were used. For experiments shown in S5c Fig, bacteria were stained with SNARF-1 (Thermo Fisher Scientific, Cat. No. S22801) prior to infection. For this, logarithmic growth-phase bacteria grown in TSB were washed once with DPBS and stained with 8 µM SNARF-1 in a final volume of 500 µL DPBS, for 20\\u2009min, at RT, protected from light. Afterwards, samples were washed twice in DPBS and de-stained in fresh DPBS for 15\\u2009min, at 37°C, protected from light. After de-staining, samples were again washed twice in DPBS, re-suspended in Infection Medium, counted using a Thoma chamber as described, and used for infection. To ensure that the SNARF-1 staining does not interfere with bacterial growth, growth curves using either SNARF-1 stained or unstained bacteria were performed in both TSB medium and RPMI1640. Infection was performed as described. At 3, 6 and 8h post infection, cells were washed, fixed with 4% PFA overnight at 4°C, treated with 50mM NH4Cl in DPBS for 5\\u2009min, permeabilized with 0.1% (v/v) Triton X-100 and nuclei were stained with 4 µg/mL Hoechst 34580. Images were acquired with an Operetta automated fluorescence microscopy system (Perkin Elmer Operetta High-Content Imaging System) and analysed with the built-in Harmony Software. Phagosomal escape was indicated by the co-localization of the YFP-CWT (phagosomal escape) and mRFP/SNARF signals (total intracellular bacteria). Live cell imaging experiments shown in S4 Fig and S2–S5 Movies were performed as described for macrophages, using mRFP-expressing bacteria. Cytotoxicity assay (LDH release) The cytotoxicity of intracellular bacterial strains against epithelial cells (S5e Fig) and the cytotoxicity of bacterial supernatants against HeLa cells (S6a Fig), were analysed by measuring the amount of released lactate dehydrogenase (LDH). At the required time points, spent medium of infected/treated cells (~2x105 cells/sample) was collected, centrifuged for 5\\u2009min at 100 x g and 100µl of supernatant (in technical duplicates) was used to determine LDH release using the Roche Cytotoxicity Detection Kit PLUS (Sigma Aldrich, Cat. No. 4744934001), according to manufacturer’s instructions. Spent medium of uninfected cells served as negative control and lysed cells served as positive control. S. aureus culture supernatant treatment of epithelial cells Bacteria were grown for 24h in TSB supplemented with the appropriate antibiotics, if necessary. Cultures were adjusted to an OD600\\u2009=\\u20094, in a final volume of 1\\u2009mL, centrifuged at 20,000 x g and supernatants were filtered through 0.22 µm-membrane filters. 2 x 105 seeded HeLa cells were treated with 10% (v/v) supernatants (in Infection Medium) and incubated for 4h at 37°C and 5% CO2. Cytotoxicity was measured by quantifying LDH released into the supernatant. 10% sterile TSB medium was used as negative control. Iron quantification in bacterial pellets A colorimetric Iron Assay kit (Sigma Aldrich Cat. No. MAK025) was used to measure bacteria-associated iron. Overnight S. aureus JE2 cultures in RPMI1640 were transferred to RPMI1640 supplemented with 20 µM FeSO4 (FeSO4 heptahydrate dissolved in technical grade H2SO4), at OD600\\u2009=\\u20090.1 and grown for 24h at 37°C, under shaking at 200\\u2009rpm. OD600\\u2009=\\u20090.5 equivalents of bacterial cultures were pelleted by centrifugation and lysed for 30\\u2009min at 37°C in 110 µL Lysis Buffer (200 µg/mL lysostaphin, 1.2% Triton-X100, in water). Lysed samples were mixed with 110 µL Assay Buffer (kit) and incubated for 10\\u2009min at 80°C. Samples were subsequently centrifuged (13,000 x g, 10\\u2009min, 4°C) and 100 µL supernatant was used to measure total iron (Fe2+ and Fe3+), according to manufacturer’s instructions. Cerulean induction in bacterial cultures Induction of the Cerulean fluorescent protein in S. aureus JE2 during growth in TSB medium was measured by monitoring fluorescence and OD600, in a TECAN MPlex plate reader, in black 96 well plates (Thermo Fisher Scientific, Cat. No. 165305). Overnight bacterial cultures were diluted to OD600\\u2009=\\u20090.1 in fresh TSB, supplemented with 300, 150\\u2009ng/mL aTc or vehicle control (ethanol). Absorbance at 600\\u2009nm and fluorescence (Ex/Em 433/504 nm) were measured every 18\\u2009min for 16h. Results are expressed as arbitrary fluorescence units (Cerulean/OD600 ratio). Random unrestricted growth (URG) Bacterial unrestricted growth (URG) observed in infected macrophages, was analysed as follows: Primary human macrophages were seeded in 96-well plates at a density of 1.5 x 104 cells/well and infected with S. aureus WT or its isogenic brnQ1 mutant as described. Macrophages originating from a single donor were employed for each experiment to exclude donor effects on experimental outcome. Plates were observed daily and the number of wells, as well as the time point when URG occurred were recorded and plotted in a Kaplan-Meier survival plot, where each well represented a test subject. Wild-type S. aureus-infected macrophages and uninfected macrophages were used as positive (URG at 24h) and negative controls (no URG), respectively. Assessment of secreted protein profiles of different S. aureus strains Single bacterial colonies were grown in TSB overnight. Cultures were diluted to 0.1, drop plated on SCMA plates and incubated at 37°C for 18-20h. Precipitation zones were measured with a standard metric ruler. Statistical analysis Data were analysed using GraphPad Prism Software (GraphPad Software, Version 10.2.3). For statistical analysis, at least three independent experiments were performed (sample size is indicated for each graph). All data are presented as means with standard deviation (±SD). P-values ≤0.05 were considered significant. When two groups were analysed, a diagnostic t-test was run first. If residuals were normally distributed (Shapiro-Wilk test), samples were analysed with an unpaired, two-tailed t-test. If residuals were not normally distributed, the non-parametric Mann-Whitney test (unpaired, two-tailed) was used instead. When more than two groups were compared, a diagnostic one-way ANOVA was carried out first to check for normality of residuals (Shapiro-Wilk test), followed by Brown-Forsythe test to check for equal variance. If the ANOVA assumptions were met, the test was followed by an appropriate post-hoc test, as indicated for each graph. The Brown-Forsythe and Welch ANOVA test were used whenever the condition for equal variance was not met. If the ANOVA assumptions were not met, the non-parametric Kruskal-Wallis test, followed by Dunn’s multiple comparisons test was applied instead. A two-way ANOVA was applied for time-course CFU experiments on log10-transformed CFU data, followed by an appropriate post-hoc test, as indicated for each graph. No matching or pairing was assumed in any ANOVA test.',\n",
       "  'Results': 'S. aureus deficient for the BrnQ1 branched-chain amino acid transporter do not replicate in human macrophages S. aureus are internalized by professional phagocytes such as macrophages, where they survive and replicate within acidic phagolysosomes [18]. To test the possible involvement of BCAAs in the S. aureus – macrophage interaction we established a macrophage infection model, involving a short phagocytosis time, followed by a lysostaphin pulse-treatment to eradicate extracellular bacteria (Fig 1a). Here we purposefully avoided prolonged use of lysostaphin or gentamicin as antimicrobials as these can be internalised by macrophages and thus could impact the survival of intracellular bacteria [12, 31–33]. Using macrophage colony-stimulating factor (M-CSF)-differentiated human monocyte-derived macrophages, we analysed the ability of individual S. aureus USA300 mutants retrieved from the Nebraska Transposon Library [34] to replicate and egress from infected macrophages. We analysed mutants deficient for the BCAA transporters BrnQ1, BrnQ2, BrnQ3 and BcaP, as well as mutants lacking the ilvE, ilvD and leuA genes, which are involved in BCAA biosynthesis [21]. A mutant in codY, the repressor of the aforementioned genes [26, 27], known to exhibit a hypervirulent phenotype [35], was included as control (Fig 1b). This analysis revealed that by 24h post-infection (p.i.) all strains except the brnQ1 mutant, had emerged from infected macrophages and massively replicated extracellularly, resulting in host cell death and the formation of micro-colonies within the infected wells of cell culture plates (Fig 1c). Since the high number of bacteria recovered at 24h p.i. is not solely due to intracellular replication, but rather to extracellular replication of those bacteria that escaped macrophage restriction, we termed this phenomenon unrestricted growth (URG). Fig 1. The BCAA transporter BrnQ1 is required for S. aureus replication in human macrophages. Open in a new tab (a) Experimental procedure for macrophage infection: primary human macrophages were infected with S. aureus, at a multiplicity of infection (MOI) of 5 for 30\\u2009min. Extracellular S. aureus were removed by a 15\\u2009min treatment with 30 μg/ml lysostaphin and two subsequent rinses with DPBS. Infected macrophages were further incubated in antibiotic-free media. “T0” corresponds to 45\\u2009min p.i. (b) S. aureus mutants in genes involved in either BCAA-uptake (brnQ1-3 and bcaP) or biosynthesis (ilvE, ilvD, leuA) were chosen for the analysis. Gene identifiers correspond to USA300_FPR3757(old locus ID) [30]. (c) Phase-contrast micrographs of macrophages at 24h p.i., which stayed uninfected, were infected with wild-type S. aureus strains JE2, Cowan I, and RN4220, or mutants in the indicated genes. All mutants were retrieved from the Nebraska Transposon Mutant Library (NTML) generated in the USA300 JE2 background (USA300 LAC-derivative). Unrestricted growth (URG) of the bacteria is observed by formation of bacterial micro-colonies (white arrows). Shown are representative images from two independent experiments. Scale bar\\u2009=\\u2009100 µm. “NE” designates mutant strain identifier in the NTML. (d) Intracellular replication in human macrophages. Data are shown as mean ±SD from independent experiments (JE2 WT and brnQ1: n\\u2009=\\u20097, brnQ1+\\u2009pbrnQ1 and Cowan I: n\\u2009=\\u20093). (e) Macrophage infection with brnQ1 mutants in the genomic backgrounds of S. aureus strains 6850, SH100 and RN4220. White arrows indicate bacterial microcolonies. Shown are representative images from three independent experiments (Scale bar: 100 µm). (f) Live cell imaging of macrophages infected with mRFP-expressing JE2 WT or JE2 brnQ1 (magenta, white arrows). Host cell death (green fluorescence, green arrows) is detected by CellEvent Caspase-3/7 Green Detection Reagent. Shown are micrographs extracted from time-lapse series representative of 3 independent experiments (n\\u2009=\\u20093). Scale bar\\u2009=\\u200950 µm. (see also S1 Movie). (g) Transmission electron micrographs of intracellular JE2 WT or JE2 brnQ1 confined to the phagosomal compartment (white arrows) within the macrophage, at 10h p.i. Shown are representative micrographs, from one experiment. Statistical analysis: (d) two-way ANOVA with Dunnett’s multiple comparisons test (each sample vs JE2 WT, at each time-point), using log10-transformed data; **p\\u2009<\\u20090.01; ***p\\u2009<\\u20090.001; ****p\\u2009<\\u20090.0001. (URG\\u2009=\\u2009unrestricted growth). (a) contains icons which were modified (colour and design) from https://bioicons.com/, as follows: petri-dish-lid-yellow icon by Servier https://smart.servier.com/ is licensed under CC-BY 3.0 Unported https://creativecommons.org/licenses/by/3.0/. Remarkably, even S. aureus strains with reduced virulence, Cowan I and RN4220 [36], emerged from infected macrophages and exhibited URG (Fig 1c). To further investigate the contribution of brnQ1 to S. aureus growth within macrophages, we transduced the brnQ1::Tn transposon insertion (NE945) into a parental S. aureus JE2 strain, hereafter referred to as JE2 brnQ1 and again assessed the ability of this strain to infect macrophages and display URG. These data revealed that, in contrast to the parental JE2 strain, JE2 brnQ1 failed to replicate. Genetic complementation of brnQ1 in trans (JE2 brnQ1 +\\u2009pbrnQ1) (S1 Fig) rescued the ability of JE2 brnQ1 to egress from macrophages (Fig 1d). While at 6h p.i. bacterial loads are comparable among all strains under scrutiny (S2 Fig), at 12h p.i. we recovered approximately 2-fold more bacteria compared with “T0” (after lysostaphin pulse, i.e. at 45\\u2009min p.i.). Both parental JE2 and JE2 brnQ1 +\\u2009pbrnQ1 as well as S. aureus Cowan I reached URG by 24h p.i. In contrast, the number of recovered colony-forming units (CFU) for JE2 brnQ1 dropped to ~4% of the initial infection inoculum by 24h p.i. (Fig 1d). Furthermore, absence of brnQ1 abolished URG in several S. aureus strains: 6850, SH1000 and RN4220 (Fig 1e). Taken together, these data reveal that a functional BrnQ1 transporter is a general requirement for S. aureus growth inside macrophages, and not simply a JE2-specific phenotype. To investigate whether the JE2 brnQ1 growth defect within macrophages is specific to the intracellular environment, we next analysed bacterial growth in the medium used for macrophage infection assays. In an undefined medium such as TSB, which contains an abundance of amino acids and peptides, growth of JE2 brnQ1 is indistinguishable from JE2 WT and the complemented strain JE2 brnQ1 +\\u2009pbrnQ1 (S3a Fig). In contrast, in a chemically defined medium (CDM) containing 1\\u2009mM BCAA, JE2 brnQ1 displayed delayed growth when compared to controls (S1 Fig). In RPMI1640, where the concentrations of Val, Leu and Ile are\\u2009~\\u2009117 µM (Val) and 131 µM (Leu and Ile, each), JE2 brnQ1 also displays a growth delay in comparison to JE2 WT and JE2 brnQ1 +\\u2009pbrnQ1 (S3b Fig). This growth delay is still evident in Infection Medium (i.e., RPMI1640\\u2009+\\u200910% v/v FBS), indicating that even in the presence of FBS the bacterium’s nutritional requirement for BCAAs is not fully met (S3c Fig). Importantly, in all culture media where JE2 brnQ1 initially displays delayed growth, the mutant eventually achieves the same bacterial density as parental JE2 (S3b and S3c Fig). Taken together, these data show that brnQ1-deficient S. aureus displays delayed growth in media with low BCAA availability (≤ 1\\u2009mM), however, in macrophages, growth of the mutant is completely halted. Replication of S. aureus within macrophages precedes cell death of infected phagocytes [12]. We therefore performed live cell imaging to monitor the intracellular replication of JE2 WT and JE2 brnQ1 in the presence of a fluorogenic caspase 3/7 substrate (Fig 1f and S1 Movie). JE2 WT commenced replicating within macrophages at approximately 8h p.i., which was followed by bacterial egress from the infected macrophage and replication in the extracellular milieu. Activation of effector caspases was evident as green fluorescence at ~ 11h p.i. Shortly thereafter, caspase activation was detectable in uninfected bystander cells. In contrast, JE2 brnQ1 did not replicate, remained confined inside macrophages and failed to trigger caspase activation during the assayed time (i.e., up to 20h p.i.) (S1 Movie). In non-professional phagocytes, such as epithelial cells, JE2 brnQ1 exhibited similarly impaired intracellular replication, accompanied by reduced cytotoxicity against the host, despite major differences in infection dynamics in this cell type [i.e., bacterial phagosomal escape precedes cytosolic replication [7, 37]] (S4 and S5 Figs and S2–S5 Movies and S1 Text). Notably, culture supernatants from brnQ1 bacteria are as cytotoxic as JE2 WT supernatants, indicating that in principle, BrnQ1 is dispensable for secreted toxin production if bacteria grow in a rich medium (S6a Fig). Additionally, brnQ1 mutants show wild-type haemolysis, on sheep blood agar, which supports the observation that brnQ1 mutants can be virulent under the right growth conditions (S6b and S6c Fig). Next, we analysed infected macrophages by transmission electron microscopy (TEM). We observed that, at 10h p.i., both JE2 WT and JE2 brnQ1 were localised to the lumen of intact phagosomal compartments (Fig 1g). Overall, these data demonstrate that wild-type S. aureus overcomes intra-macrophage confinement whereas the brnQ1 mutant cannot. Moreover, under our experimental conditions, 10h p.i. is a timepoint at which both wild-type and the brnQ1 mutant, which fails to replicate, are still confined within membrane-bound phagosomes. Transcriptome analysis of intracellular S. aureus reveals dysregulation of iron metabolism in brnQ1 mutants To identify the cause of the restricted intracellular growth of brnQ1 mutants, we next performed dual RNA sequencing (RNAseq) of infected macrophages at 10h p.i. We hypothesized that this time point would be crucial for the bacterial adaptation to the phagolysosomal environment, since S. aureus JE2 WT and brnQ1 mutant still reside within a membrane-bound vacuole but only the WT strain can initiate replication shortly thereafter. Consistent with our previous observations [38], we detected a similar host gene expression signature, irrespective of whether macrophages were infected with JE2 brnQ1 or JE2 WT (Figs 2a and S7a), In contrast, we detected profound differences in gene expression between JE2 WT and JE2 brnQ1. Among bacterial transcripts, we found that 80 genes were significantly downregulated in the JE2 brnQ1 mutant inside macrophages as compared to WT. Only 6 transcripts were upregulated in JE2 brnQ1 (Figs 2b and S7b and Table 1). Most of the downregulated genes in intracellular JE2 brnQ1, belonged to operons that encode iron acquisition systems (Figs 2c and S8). These included several isd heme utilization genes (isdA, isdB, isdCDEF and isdG), srtB, staphyloferrin B siderophore biosynthesis genes (sirAB and sbnABCDEFGH) and the staphyloferrin A biosynthesis gene sfaC (Table 1). Several other genes, including chemotaxis-inhibiting protein (chp), thermonuclease (nuc), von Willebrand factor-binding protein (vwb), superoxide dismutase (sodM), Panton-Valentine leukocidin (PVL) subunit lukS, were downregulated. The cysteine protease staphopain A (scpA), as well as genes encoding superantigen-like proteins Ssl2, 4, 7, 11, 12, 13, 14, which are known virulence factors, were similarly downregulated as compared to wild-type. Fig 2. Intracellular S. aureus JE2 brnQ1 mutants do not respond to iron-starvation inside macrophages. Open in a new tab Dual RNAseq of primary human macrophages infected with either JE2 WT or JE2 brnQ1, at 10h p.i.: (a) Human transcripts and (b) Bacterial transcripts. Horizontal lines represent a p-value cutoff\\u2009<\\u20090.05. Vertical lines represent cutoffs of log2 fold changes of either <-1 and > 1 for human transcripts or <-1.5 and > 1.5 for bacterial transcripts. Transcripts of iron uptake systems, which are prominently downregulated in intracellular JE2 brnQ1 vs JE2 WT, are depicted in green. Differential transcript abundance is expressed as JE2 brnQ1 vs JE2 WT (DESeq2). Data are shown from 3 independent experiments (n\\u2009=\\u20093). (c) Schematic of the ferric uptake regulator Fur regulon of S. aureus (information retrieved from RegPrecise). Coloration of the boxes underneath the open reading frames indicates log2 fold changes in expression of JE2 brnQ1 vs JE2 WT according to the heatmap. Gene identifiers correspond to USA300_FPR3757 (old locus ID). (d) RT-qPCR analysis of selected genes found to be downregulated or upregulated (e) in intracellular JE2 brnQ1. Data are shown as mean -ΔCt values ±SD from independent experiments (n\\u2009=\\u20094, different donors). (f) Venn diagram showing the overlap between the set of significantly downregulated genes in JE2 brnQ1 shown in b and the Fur regulon retrieved from RegPrecise. Statistical analysis: (d, e) one-way ANOVA with Dunnett’s multiple comparisons test (vs JE2 WT for each gene); ns\\u2009=\\u2009not significant; **p\\u2009<\\u20090.01; ****p\\u2009<\\u20090.0001. Table 1. List of regulated genes for S. aureus JE2 in primary human macrophages at 10h p.i. (JE2 brnQ1 compared to JE2 WT). Downregulated Genes Locus ID (old) Locus ID (new) Gene Annotation Log2FC Adj. P value SAUSA300_1034 SAUSA300_RS05565 srtB sortase B -7.34 3.38 x 10–07 SAUSA300_1920 SAUSA300_RS10530 chp chemotaxis-inhibiting protein CHIPS -6.10 4.67 x 10–45 SAUSA300_0407 SAUSA300_RS02185 ssl11 superantigen-like protein SSL11 -5.48 4.10 x 10–12 SAUSA300_0398 SAUSA300_RS02130 ssl4 superantigen-like protein SSL4 -5.05 2.38 x 10–04 SAUSA300_2291 SAUSA300_RS12660 gltS sodium/glutamate symporter -4.80 5.66 x 10–03 SAUSA300_1031 SAUSA300_RS05550 isdD hypothetical protein -4.59 1.49 x 10–06 SAUSA300_0408 SAUSA300_RS02190 – FKLRK protein -4.48 2.99 x 10–22 SAUSA300_0401 SAUSA300_RS02150 ssl7 superantigen-like protein SSL7 -4.37 2.99 x 10–22 SAUSA300_0120 SAUSA300_RS00620 sbnC staphyloferrin B biosynthesis protein SbnC -4.36 1.61 x 10–04 SAUSA300_0774 SAUSA300_RS04175 emp extracellular matrix protein-binding adhesin Emp -4.33 2.99 x 10–22 SAUSA300_1022 SAUSA300_RS05500 – hypothetical protein -4.29 4.43 x 10–02 SAUSA300_1028 SAUSA300_RS05535 isdB heme uptake protein IsdB -4.28 3.56 x 10–20 SAUSA300_0940 SAUSA300_RS05050 – DoxX family protein -4.20 6.60 x 10–03 SAUSA300_1032 SAUSA300_RS05555 isdE heme ABC transporter substrate-binding protein IsdE -4.16 4.45 x 10–05 SAUSA300_1033 SAUSA300_RS05560 isdF iron ABC transporter permease -4.15 2.27 x 10–06 SAUSA300_0775 SAUSA300_RS04180 – hypothetical protein -4.12 3.56 x 10–20 SAUSA300_1929 SAUSA300_RS10585 – phage tail protein -3.93 4.86 x 10–02 SAUSA300_0776 SAUSA300_RS04185 nuc thermonuclease family protein -3.86 1.85 x 10–04 SAUSA300_1030 SAUSA300_RS05545 isdC heme uptake protein IsdC -3.85 3.09 x 10–06 SAUSA300_0815 SAUSA300_RS04395 ear DUF4888 domain-containing protein -3.48 1.03 x 10–15 SAUSA300_1035 SAUSA300_RS05570 isdG staphylobilin-forming heme oxygenase IsdG -3.45 1.04 x 10–02 SAUSA300_0215 SAUSA300_RS01130 – isoprenylcysteine carboxyl methyltransferase family protein -3.36 2.36 x 10–12 SAUSA300_1029 SAUSA300_RS05540 isdA LPXTG-anchored heme-scavenging protein IsdA -3.30 6.07 x 10–26 SAUSA300_0773 SAUSA300_RS04170 vwb von Willebrand factor binding protein Vwb -3.29 3.62 x 10–16 SAUSA300_1060 SAUSA300_RS05745 ssl13 superantigen-like protein SSL13 -3.23 1.84 x 10–02 – SAUSA300_RS09430 – hypothetical protein -3.01 4.62 x 10–02 SAUSA300_2528 SAUSA300_RS14025 – epoxyqueuosine reductase QueH -2.99 1.44 x 10–02 SAUSA300_0124 SAUSA300_RS00640 sbnG staphyloferrin B biosynthesis citrate synthase SbnG -2.98 5.10 x 10–04 SAUSA300_1061 SAUSA300_RS05750 ssl14 superantigen-like protein SSL14 -2.97 1.44 x 10–02 SAUSA300_1059 SAUSA300_RS05740 ssl12 superantigen-like protein SSL12 -2.96 1.82 x 10–02 SAUSA300_0237 SAUSA300_RS01265 – nucleoside hydrolase -2.93 4.86 x 10–02 SAUSA300_0119 SAUSA300_RS00615 sbnB 2,3-diaminopropionate biosynthesis protein SbnB -2.65 4.24 x 10–03 SAUSA300_0116 SAUSA300_RS00600 sirB staphyloferrin B ABC transporter permease subunit SirB -2.62 2.94 x 10–03 SAUSA300_0986 SAUSA300_RS05305 cydA cytochrome ubiquinol oxidase subunit I -2.55 1.14 x 10–05 SAUSA300_0122 SAUSA300_RS00630 sbnE lucA/IucC family siderophore biosynthesis protein -2.55 1.35 x 10–04 SAUSA300_2453 SAUSA300_RS13605 – ATP-binding cassette domain-containing protein -2.49 3.75 x 10–04 SAUSA300_1786 SAUSA300_RS09780 ecsA ABC transporter ATP-binding protein -2.48 1.37 x 10–05 SAUSA300_0814 SAUSA300_RS04390 – Abi family protein -2.46 3.80 x 10–09 SAUSA300_0117 SAUSA300_RS00605 sirA staphyloferrin B ABC transporter substrate-binding protein SirA -2.44 2.38 x 10–09 SAUSA300_0123 SAUSA300_RS00635 sbnF IucA/IucC family siderophore biosynthesis protein -2.42 9.01 x 10–04 SAUSA300_0307 SAUSA300_RS01635 – 5’-nucleotidase, lipoprotein e(P4) family -2.41 4.79 x 10–04 SAUSA300_0902 SAUSA300_RS04855 pepF oligoendopeptidase F -2.38 2.11 x 10–09 SAUSA300_0121 SAUSA300_RS00625 sbnD staphyloferrin B export MFS transporter -2.33 3.05 x 10–02 SAUSA300_2409 SAUSA300_RS13340 cnzC nickel ABC transporter permease -2.33 1.32 x 10–02 SAUSA300_1171 SAUSA300_RS06335 – insulinase family protein -2.30 2.78 x 10–04 SAUSA300_0825 SAUSA300_RS04450 – nitronate monooxygenase -2.26 1.19 x 10–02 SAUSA300_1675 SAUSA300_RS09145 tyrS tyrosine-tRNA ligase -2.25 4.74 x 10–03 SAUSA300_0118 SAUSA300_RS00610 sbnA siderophore biosynthesis protein SbnA -2.20 1.29 x 10–02 SAUSA300_2137 SAUSA300_RS11770 sfaC staphyloferrin A biosynthesis protein SfaC -2.19 6.69 x 10–03 SAUSA300_2262 SAUSA300_RS12500 spdB CPBP family intramembrane metalloprotease SdpB -2.17 1.92 x 10–02 SAUSA300_0125 SAUSA300_RS00645 sbnH staphyloferrin B biosynthesis decarboxylase SbnH -2.16 3.36 x 10–03 SAUSA300_1053 SAUSA300_RS05680 flr formyl peptide receptor-like 1 inhibitory protein -2.14 7.19 x 10–06 SAUSA300_0135 SAUSA300_RS00705 sodM superoxide dismutase -2.14 2.62 x 10–08 SAUSA300_1957 SAUSA300_RS10735 dnaD2 DnaD domain-containing protein -2.14 3.05 x 10–02 SAUSA300_1890 SAUSA300_RS10340 scpA cysteine protease staphopain A -2.13 5.77 x 10–04 SAUSA300_2251 SAUSA300_RS12430 – octopine dehydrogenase -2.13 9.70 x 10–05 SAUSA300_1852 SAUSA300_RS10120 – ABC transporter ATP-binding protein -2.13 3.36 x 10–03 SAUSA300_1964 SAUSA300_RS10775 – DUF771 domain-containing protein -2.10 2.45 x 10–02 SAUSA300_1587 SAUSA300_RS08650 hisS histidine-tRNA ligase -2.04 2.03 x 10–04 SAUSA300_1797 SAUSA300_RS09835 – helix-turn-helix transcriptional regulator -1.99 5.26 x 10–03 SAUSA300_0396 SAUSA300_RS02115 ssl2 superantigen-like protein SSL2 -1.99 1.19 x 10–02 SAUSA300_0254 SAUSA300_RS01355 lytS sensor histidine kinase -1.98 1.46 x 10–03 SAUSA300_0987 SAUSA300_RS05310 cydB cytochrome D ubiquinol oxidase subunit II -1.85 1.67 x 10–03 SAUSA300_2537 SAUSA300_RS14075 ldh2 L-lactate dehydrogenase -1.82 2.29 x 10–04 SAUSA300_0432 SAUSA300_RS02315 – sodium-dependent transporter -1.81 6.37 x 10–03 SAUSA300_2210 SAUSA300_RS12185 glcU glucose uptake protein GlcU -1.79 1.06 x 10–02 SAUSA300_1382 SAUSA300_RS07545 lukS Panton-Valentine bi-component leukocidin subunit LukS-PV -1.78 3.12 x 10–06 SAUSA300_1882 SAUSA300_RS10300 gatC Asp-tRNA(Asn)/Glu-tRNA(Gln) amidotransferase subunit GatC -1.75 2.00 x 10–03 SAUSA300_1381 SAUSA300_RS07540 – Panton-Valentine bi-component leukocidin subunit F -1.75 1.48 x 10–05 SAUSA300_1471 SAUSA300_RS08030 xseB exodeoxyribonuclease VII small subunit -1.74 3.90 x 10–02 – SAUSA300_RS10500 – MAP domain-containing protein -1.74 1.04 x 10–09 SAUSA300_2526 SAUSA300_RS14015 pyrD quinone-dependent dihydroorotate dehydrogenase -1.73 2.21 x 10–03 SAUSA300_2250 SAUSA300_RS12425 nhaC Na+/H+ antiporter NhaC -1.69 3.16 x 10–02 SAUSA300_2328 SAUSA300_RS12865 – DUF4889 domain-containing protein -1.67 2.90 x 10–02 SAUSA300_0107 SAUSA300_RS00555 nptA Na/Pi cotransporter family protein -1.65 1.61 x 10–04 SAUSA300_1153 SAUSA300_RS06245 uppS isoprenyl transferase -1.64 4.06 x 10–03 SAUSA300_1170 SAUSA300_RS06330 – GntR family transcriptional regulator -1.63 2.40 x 10–02 SAUSA300_0025 SAUSA300_RS00130 adsA LPXTG-anchored adenosine synthase AdsA -1.62 5.72 x 10–03 SAUSA300_0620 SAUSA300_RS03325 mntA metal ABC transporter ATP-binding protein -1.57 5.35 x 10–13 SAUSA300_0669 SAUSA300_RS03590 uppP undecaprenyl-diphosphate phosphatase -1.56 3.92 x 10–04 Upregulated Genes Locus ID (old) Locus ID (new) Gene Annotation Log2FC Adj. P value SAUSA300_2417 SAUSA300_RS13380 – AbgT family transporter 1.58 1.83 x 10–03 SAUSA300_0955 SAUSA300_RS05135 atl bifunctional autolysin 1.60 2.24 x 10–04 SAUSA300_0891 SAUSA300_RS04805 opp-3A peptide ABC transporter substrate-binding protein 1.66 4.22 x 10–07 SAUSA300_0890 SAUSA300_RS04800 opp-3F ATP-binding cassette domain-containing protein 1.73 6.41 x 10–05 SAUSA300_2585 SAUSA300_RS14375 asp3 accessory Sec system protein Asp3 1.91 2.45 x 10–02 SAUSA300_2607 SAUSA300_RS14495 hisA 1-(5-phosphoribosyl)-5-((5-phosphoribosylamino)methylideneamino)imidazole-4-carboxamide isomerase 1.94 1.44 x 10–02 Open in a new tab We further validated by RT-qPCR the marked downregulation of heme- and siderophore-mediated iron acquisition genes in intracellular JE2 brnQ1 as compared to intracellular WT JE2 (Fig 2d). It is interesting to note that a few genes showed modest upregulation in the brnQ1 mutant (Table 1). Again, using RT-qPCR we confirmed that, in the brnQ1 mutant, two loci in the opp3 oligopeptide transporter operon [26] and a gene which encodes a putative transporter for p-aminobenzoyl-glutamate (PABA) (SAUSA300_RS13380/ SAUSA300_2417), a precursor of folic acid [28], were indeed upregulated in intracellular JE2 brnQ1 (Fig 2e). Importantly, all changes in gene expression observed in JE2 brnQ1 relative to JE2 WT were restored to wild-type levels when JE2 brnQ1 was complemented with brnQ1 on a plasmid (Fig 2d and 2e). Taken together, these data indicate that wild-type S. aureus can adapt to the intracellular milieu of the macrophage as evidenced, for instance, by upregulation of many genes including Fe-regulated loci, while the brnQ1 mutant displays a global defect in gene expression. The S. aureus-containing phagosome has been shown to be iron depleted [39]. By comparing transcriptomes of intracellular bacteria at 10h p.i. to bacteria grown for 10h in Infection Medium, we reaffirm the observation that S. aureus is indeed iron-starved in this niche, as indicated by upregulation of iron-regulated loci by intracellular JE2 WT (S9 Fig). In S. aureus, transcription of genes encoding iron uptake proteins is controlled by the ferric uptake regulator Fur [40]. Fur is a transcription factor that uses Fe2+ as corepressor. Under conditions of iron starvation, when Fur is not bound to Fe2+, Fur is inactive resulting in de-repression of target genes [41]. The observation that many of the most significantly downregulated genes in the brnQ1 mutant function in iron acquisition, prompted us to consult a list of Fur-regulated targets retrieved from the RegPrecise database [42]. This analysis revealed that 18 of the 80 (22.5%) genes that were significantly downregulated in JE2 brnQ1 indeed belonged to the Fur regulon (Fig 2f). To exclude that inadvertent mutations in fur were responsible for our observations, we sequenced the fur locus from JE2 WT and JE2 brnQ1 and found that the gene was intact in both strains. Intracellular S. aureus brnQ1 mutants are metabolically quiescent at 10h p.i. and fail to respond to environmental cues Since the brnQ1 mutant carries an intact fur gene, yet fails to regulate iron acquisition loci, we speculated that the brnQ1 mutant may lack the ability to respond appropriately to environmental cues within the phagosome. Ostensibly this defect is exclusively due to BCAA starvation. Therefore, we posited that, even upon inactivation of the fur gene, Fur-regulated genes would still not be de-repressed in the intracellular environment. To test this, we generated fur and brnQ1/fur mutants in the JE2 background. To validate our strains, we first analysed isdB expression, as a representative iron-regulated gene within the Fur regulon, in Infection Medium (iron replete; contains BCAAs). Under this condition, the inactivation of fur resulted in increased expression of isdB, that was comparable between both fur mutant backgrounds (Fig 3a). Moreover, when we cultivated both JE2 fur and JE2 brnQ1/fur in iron-rich medium (RPMI1640 supplemented with 20 µM FeSO4), total intracellular iron (Fe2+ and Fe3+) was\\u2009~\\u20093-fold higher for both fur backgrounds as compared to either JE2 WT or JE2 brnQ1 (S10 Fig). This confirms that inactivation of fur, in both wild-type and mutant backgrounds, results in constitutive de-repression of iron-regulated operons in planktonic bacteria. Fig 3. Loss of BrnQ1 function renders intracellular S. aureus unresponsive to environmental cues and metabolically quiescent, but a supply of exogenous BCAAs restores responsiveness and promotes intracellular replication. Open in a new tab (a) RT-qPCR analysis of isdB expression for bacteria grown in Infection Medium for 1 hour or (b) in intracellular bacteria within infected primary human macrophages, at 10h p.i. isdB was selected as representative iron-regulated gene to report on Fur regulon status. Data are shown as mean -ΔCt values ±SD from independent experiments: (a) n\\u2009=\\u20093; (b) n\\u2009=\\u20094. (c) Schematic representation of the set-up for experiments using aTc induction of intracellular JE2 with a plasmid allowing anhydrous tetracycline (aTc)-inducible expression of the cyan-fluorescent protein Cerulean. After 1h p.i. or 10h p.i., infected cells were treated with either vehicle control (-) or 300\\u2009ng/mL aTc (+) for 2h to allow Cerulean expression. (d) Representative confocal images of infected primary human macrophages at 1h p.i. and (f) 10h p.i. upon induction (Magenta: Cerulean; Green: Vancomycin BODIPY-FL/S. aureus; Gray: Phalloidin/Actin; Blue: Hoechst/DNA. Scale bar: overviews: 5 µm; zoomed inset: 1µm). (e) Quantification of intracellular Cerulean-positive bacteria, as percent of total bacteria at 1h p.i. and (g) 10h p.i. in samples exposed to aTc (+) or vehicle control (-) for 2h. Data are shown as mean ±SD from independent experiments (n\\u2009=\\u20093). (h) Intracellular replication of JE2 brnQ1 or (i) JE2 brnQ1/fur in primary human macrophages, without or with BCAA supplementation. Excess BCAAs (1mM each) were supplemented 24h prior to and during the infection. Data are shown as mean ±SD from independent experiments (n\\u2009=\\u20093). Graphed URG value represents an imputed value (median of all quantified URG experiments which are shown in Fig 1d). (j) RT-qPCR analysis of isdB expression in intracellular bacteria in infected primary human macrophages, at 10h p.i. Excess BCAAs (1mM each) were supplemented 24h prior to and during the infection. Data are shown as mean -ΔCt values ±SD from independent experiments: n\\u2009=\\u20094; except JE2 brnQ1/fur +\\u2009BCAA: n\\u2009=\\u20093. Statistical analysis: (a, b) one-way ANOVA with Tukey’s multiple comparisons test; (e, g, h, i, j) one-way ANOVA with Sidak’s multiple comparisons test (h and i: for the 10h p.i. time point only); ns\\u2009=\\u2009not significant; *p\\u2009<\\u20090.05; **p\\u2009<\\u20090.01; ***p\\u2009<\\u20090.001; (URG\\u2009=\\u2009unrestricted growth). (c) contains icons which were modified (colour and design) from https://bioicons.com/, as follows: petri-dish-lid-yellow icon by Servier https://smart.servier.com/ is licensed under CC-BY 3.0 Unported https://creativecommons.org/licenses/by/3.0/. The confocal-scanning-laser-microscope-CSLM icon by DBCLS https://togotv.dbcls.jp/en/pics.html is licensed under CC-BY 4.0 Unported https://creativecommons.org/licenses/by/4.0/. Using these strains, we analysed transcription of isdB inside macrophages and observed that JE2 WT and JE2 fur show comparable isdB expression. This result was expected as under the iron-starved conditions encountered within the phagosome, isdB should be fully de-repressed in JE2 WT similarly to JE2 fur, where iron-regulated genes are constitutively expressed. Intracellular JE2 brnQ1/fur showed isdB expression levels similar to JE2 brnQ1 alone despite lacking the Fur repressor (Fig 3b). Furthermore, isdB expression in both brnQ1 and brnQ1/fur mutants was significantly decreased relative to JE2 WT or JE2 fur, again indicating that brnQ1 deficiency is enough to prevent appropriate induction of gene expression (Fig 3b). The inability of JE2 brnQ1 to remodel its gene expression in the intracellular milieu hinted that the bacteria may enter a state of intracellular quiescence. To test this, we assessed whether JE2 brnQ1 bacteria were capable of de novo protein synthesis within macrophages. We measured de novo synthesis of the cyan-fluorescent protein Cerulean upon intracellular induction with anhydrotetracycline (aTc) as previously described [43]. In a control experiment, both JE2 WT and JE2 brnQ1 responded to aTc induction in TSB medium, showing that, in principle, both strains can produce Cerulean (S11 Fig). To test whether Cerulean production can be induced by intracellular bacteria, aTc was added to infected macrophages at 1h or 10h p.i. for 2h (Fig 3c). Induction with aTc at 1h p.i. resulted in Cerulean fluorescence for both JE2 WT and JE2 brnQ1 (Fig 3d). At this time, approximately 15% of all internalized bacteria, as detected by BODIPY FL Vancomycin staining, also showed Cerulean fluorescence, while in vehicle-treated cells the bacteria remained Cerulean negative (Figs 3e and S12). In contrast, when Cerulean production was induced at 10h p.i., ~10% of the JE2 WT bacteria became Cerulean-positive, whereas only very rarely did JE2 brnQ1 become Cerulean-positive and was comparable to vehicle treated controls (Figs 3f, 3g and S12). The inactivation of brnQ1 alone, the dominant BCAA transporter, was sufficient to halt intracellular bacterial growth. This indicated that the S. aureus-containing phagosome must be scarce in BCAAs. To test this directly, we supplemented the Infection Medium with 1\\u2009mM of each BCAA and observed that the defect in intracellular replication was rescued, both in the case of JE2 brnQ1 (Fig 3h) and JE2 brnQ1/fur (Fig 3i), which now overcame macrophage restriction and grew unrestrictedly. Moreover, addition of excess BCAAs rendered intracellular JE2 brnQ1 and JE2 brnQ1/fur responsive to the iron-deplete state of the phagosome, as shown by significantly higher isdB expression levels 10h p.i. (Fig 3j). During growth in CDM lacking all three BCAAs, the growth of S. aureus is preceded by an extended lag phase of ~20h. However, omission of Val and Leu only, surprisingly completely halted bacterial growth (S13a Fig). We have therefore hypothesized that provision of these 2 amino acids alone would be sufficient to rescue the growth deficit of S. aureus brnQ1 in macrophages. Indeed, supplementation with 1\\u2009mM Val and Leu, but not the single amino acids, restored growth of the brnQ1 mutant in RAW 264.7 macrophages (S13b and S13c Fig). Taken together, we posit that by 10h p.i. S. aureus brnQ1 mutants have assumed a state of quiescence where the bacteria fail to adapt to the intracellular niche as evidenced by altered gene expression and impaired de novo protein synthesis. Importantly, this quiescence is caused by BCAA starvation and can be overcome if Val and Leu become available, indicating that BCAA acquisition underpins S. aureus growth and survival inside macrophages. Inactivation of the CodY repressor rescues growth of brnQ1-deficient S. aureus in an IlvD and BcaP dependent manner The brnQ1 mutant fails to grow in BCAA-depleted phagosomes, although S. aureus carries genes that encode BCAA biosynthesis enzymes (e.g., the ilv operon) [21], as well as a second BCAA transporter (bcaP) [29], which are expressed upon inactivation of the repressor protein CodY [21, 26, 29, 44] and could potentially ensure BCAA supply. To understand the role that CodY plays in the acquisition of BCAAs during infection, we created a brnQ1/codY double mutant and measured bacterial replication in macrophages. In contrast to the brnQ1 single mutant, the brnQ1/codY double mutant behaved similarly to the WT strain and exhibited URG both in primary human (Fig 4a) and murine RAW 264.7 macrophages (Fig 4b). Restored intracellular growth of brnQ1-deficient S. aureus upon codY inactivation, indicated that the bacteria must now be able to meet their BCAA requirement. Fig 4. Intracellular growth deficiency of S. aureus brnQ1 mutants can be rescued by inactivation of the CodY repressor or BcaP overexpression. Open in a new tab Intracellular growth of a brnQ1 mutant in the S. aureus USA300 background (∆brnQ1) and a brnQ1/codY double mutant (∆brnQ1/codY::Tn) in (a) primary human macrophages and (b) murine RAW 264.7 macrophages. (c) Intracellular growth in RAW264.7 macrophages of a triple S. aureus USA300 brnQ1/codY/ilvD mutant (∆brnQ1/codY::Tn/ilvD::Tn-KanR). (d) Growth in infected RAW 264.7 macrophages of S. aureus brnQ1 carrying either a vector control (pRMC2), plasmid-encoded BCAA transporter BcaP (ΔbrnQ1+\\u2009pbcaP) or plasmid encoded brnQ1 (ΔbrnQ1+\\u2009pbrnQ1). (e) Growth in infected RAW 264.7 macrophages of S. aureus brnQ1, a double brnQ1/bcaP mutant (ΔbrnQ1/ ΔbcaP), a triple brnQ1, bcaP, codY mutant (ΔbrnQ1/ΔbcaP/codY::Tn) and a quadruple mutant carrying the additional ilvD mutation (ΔbrnQ1/ΔbcaP/codY::Tn /ilvD::Tn-KanR). USA300 WT serves as control. All data are shown as mean log10 value\\u2009±\\u2009S.D. for the calculated fold change in CFU/mL at 20h relative to 1.5h p.i., for each bacterial strain. Each data point plotted represents a biological replicate derived from at least three independent experiments (n\\u2009≥\\u20093). Statistical analysis: (a, b, c, e) one-way ANOVA with Tukey’s multiple comparison test; (d) Brown-Forsythe and Welch ANOVA with Dunnet’s T3 multiple comparisons test. ns\\u2009=\\u2009not significant; *p\\u2009<\\u20090.05, ***p\\u2009<\\u20090.001, ****p\\u2009<\\u20090.0001. Conceivably, this could be accomplished through endogenous BCAA synthesis. To test this, we created a brnQ1/codY/ilvD triple mutant and assessed its ability to grow in CDM without Val. Growth experiments revealed that the brnQ1/codY mutant, as expected, could grow in the absence of exogenous Val yet the brnQ1/codY/ilvD mutant could not (S14 Fig). This confirms that, as opposed to the triple mutant, the brnQ1/codY mutant must be able to synthesize Val to allow growth in Val-deplete media. Next, we infected RAW 264.7 macrophages with the mutants. As expected, brnQ1-deficient S. aureus failed to grow out of macrophages whereas WT and brnQ1/codY S. aureus overcame macrophage restriction and displayed URG. Remarkably, the brnQ1/codY/ ilvD mutant also grew out of macrophages indicating that, despite the inability to synthesize BCAAs, the bacteria still access these amino acids inside the macrophage (Fig 4c). The observation that the intracellular growth of S. aureus brnQ1 can be restored by exogenous addition of Val and Leu during infection, indicated that the bacteria must retain the ability for BCAA uptake. BcaP is an obvious candidate as it has previously been shown that it transports all three BCAAs and that it can partially compensate for loss of brnQ1 [29]. Therefore, we next infected macrophages with a brnQ1 mutant strain that overexpresses bcaP or a respective vector control and we observed that S. aureus brnQ1 overexpressing bcaP could grow unrestrictedly without BCAA supplementation (Fig 4d). These data suggested that increased bcaP expression, due to codY inactivation [29], could explain why the brnQ1/codY/ilvD triple mutant maintained the ability to overcome macrophage restriction. Hence, we utilized a brnQ1/bcaP double mutant, again inactivated either codY alone, or codY and ilvD in this background and infected RAW 264.7 macrophages. Similar to a brnQ1 single mutant, the brnQ1/bcaP double mutant was unable to replicate (Fig 4e). Inactivation of codY in the brnQ1/bcaP background restored growth in macrophages however, an additional ilvD mutation significantly impaired growth (Fig 4e). Taken together, these data indicate that limited phagosomal BCAA levels can in principle be overcome in S. aureus by increased expression of CodY-regulated BCAA synthesis genes and/or BCAA transporters. S. aureus brnQ1 mutants survive for inordinate duration within human macrophages Since brnQ1-deficient S. aureus failed to replicate in macrophages during the first 24h of infection, we next assessed long-term intracellular survival in primary human macrophages. Often, studies on long-term intracellular S. aureus infections rely on sustained antibiotic treatment of the infected host cells to limit the effects caused by extracellular bacteria that may egress from host cells during the experiment [13, 45–49]. Here we did not maintain antibiotic treatment on infected macrophages, and this allowed us to assess whether URG would occur for S. aureus brnQ1. Bacteria were quantified by CFU counting at 1, 7, 14, 21 and 28 days p.i. unless we observed URG (Fig 5a). Within the first 24h of infection, the number of JE2 brnQ1 bacteria decreased to ~3.5% of the initially phagocytosed bacterial population (i.e., lysostaphin-protected bacteria at “T0”). After this initial decline, the abundance of JE2 brnQ1 remained steady whereas JE2 WT grew unrestrictedly (URG) by day 1 p.i. (Fig 5b). Remarkably, viable JE2 brnQ1 were recovered from infected macrophages at the endpoint of the assay after 28 days p.i. Across 14 independent experiments, we observed that typically JE2 brnQ1 survived for 28 days within the macrophages. However, on occasion, URG was observed at day 2 or 15 p.i. albeit with less frequency (Fig 5b). Consistent with the CFU recovery data, only few bacteria were detected microscopically inside macrophages infected with JE2 brnQ1 from 24h p.i. onwards and ultrastructural analysis revealed brnQ1 mutants to be localised inside intact membrane-enclosed vacuoles (Fig 5c). We hypothesized that the variability in occurrence of URG might arise in a stochastic manner, resulting, for example, from loss of macrophage bactericidal capacity, death of individual host cells with subsequent release of bacteria, or heterogeneity within the bacterial subpopulations. To address this, we set up experiments where macrophages derived from the same donor were seeded in multiple technical replicates (e.g., up to 80 wells/experiment), infected with JE2 brnQ1 and observed daily for 28 days to monitor URG. We observed variability within technical replicates, with URG occurring at various time points during the infection (S15a Fig). Furthermore, brnQ1 mutants in different S. aureus backgrounds similarly survived for up to 28 days without eliciting visible cytotoxic effects against host cells (S15b, S15c and S15d Fig). These data collectively indicate that the duration of infection prior to URG is stochastic and donor-independent. Fig 5. S. aureus brnQ1 mutants survive for at least 28 days inside macrophages but are distinct of small colony variants (SCV). Open in a new tab (a) JE2 brnQ1 intracellular replication in primary human macrophages, assayed over a period of 28 days. Data are shown as mean values ±SD from independent experiments (T0: n\\u2009=\\u20097; T0.5: n\\u2009=\\u20095; Td1: n\\u2009=\\u20096; Td7 and Td21: n\\u2009=\\u20092; Td14 and Td28: n\\u2009=\\u20093). (b) Maximum time until unrestricted growth (URG) in macrophages infected with JE2 brnQ1 across 14 biological replicates (n\\u2009=\\u200914), using 13 different blood donors (arrows: same donor, but used in an independent experiment). JE2 WT serves as control (URG at 24h p.i.). The experiment was terminated 28 days p.i. regardless of URG occurrence. (c) Transmission electron microscopy of JE2 brnQ1-infected macrophages at 24h p.i., 48h p.i., 7 and 28 days p.i. (white arrows). Shown are representative micrographs from one experiment. (d) URG recovery experiment, whereby during a 28-day infection in macrophages, JE2 brnQ1 clones (i.e., micro-colonies formed upon overgrowth from infected macrophages) were recovered upon URG (days 21, 23 and 27 p.i. respectively). Recovered JE2 brnQ1 clones were re-inoculated in Infection Medium and growth was analysed. OD600 was measured every 18\\u2009min, for 20h. (e) Recovered JE2 brnQ1 were used for infection experiments in primary human macrophages, for 48h. Graphed URG value for JE2 WT control, represents an imputed value (median of all quantified URG experiments which are shown in Fig 1d). (f) Circular genome alignment of two JE2 brnQ1 clones recovered from macrophages upon URG (URG 4 and URG5 shown in d and e using the USA300_FPR3757 complete genome and JE2 WT for comparison (parental strain, laboratory stock). (g) JE2 brnQ1 mutants that did not show URG were recovered from viable macrophages at 28 days p.i. and colony size, haemolysis on sheep-blood agar, and colony pigmentation were assessed. Colony diameter measurements of JE2 brnQ1 which were recovered from macrophages, were compared with bacteria grown to late exponential phase in TSB. Data points correspond to single colonies, recovered from independent experiments (n\\u2009=\\u20092). Statistical analysis: (g) two-tailed Mann-Whitney test; ns =\\u2009not significant. S. aureus can grow in the absence of Leu or Val in vitro, albeit after an extended lag phase (S13a Fig). Moreover, growth in the absence of Val was shown to select for heritable loss-of-function mutations in codY or the 5’ untranslated region (5’UTR) of ilvD, to allow de novo BCAA synthesis and subsequent bacterial replication [21]. Spontaneous selection of such suppressor mutants in intracellular JE2 brnQ1 is a possible scenario that could explain the differences in URG observed across experiments and/or technical replicates. Furthermore, our data have revealed that inactivation of CodY is sufficient to rescue the growth of brnQ1 S. aureus in macrophages and in CDM lacking Val (Figs 4 and S14a). To determine whether brnQ1 mutants that initiated URG had acquired genetic changes, we recovered five JE2 brnQ1 clones upon egress from infected macrophages, sub-cultured the clones in Infection Medium and infected again macrophages. Each recovered brnQ1 isolate again displayed delayed growth in Infection Medium (Fig 5d) and impaired growth upon macrophage infection (Fig 5e) suggesting that the egress events were not the result of heritable mutations. Consistently, whole genome sequencing of two such brnQ1 isolates revealed no mutations in either codY or the 5’UTR of ilvD (Fig 5f). Intracellular S. aureus brnQ1 mutants are not classical persisters Within a bacterial subpopulation, transient growth arrest and metabolic quiescence, accompanied by antibiotic tolerance, are hallmarks of persistence [50–53]. Intracellular persisters of S. aureus can arise upon antibiotic pressure [43, 54, 55]. Several mechanisms triggering intracellular S. aureus persisters were proposed, such as a transient activation of the stringent response (SR) [43]. In S. aureus, SR can be induced by amino-acid starvation (especially Leu/Val deprivation [27]) and results in rapid synthesis of the (p)ppGpp alarmone leading to profound transcriptional reprogramming, especially the repression of proliferation-related genes and activation of the stress-response and starvation-related regulons and a halt of bacterial division [27]. However, upon consulting our transcriptomic data (S1 Dataset), we did not observe changes in the expression of translation factors, ribosomal proteins or tRNAs in JE2 brnQ1 compared to wild-type, at 10h p.i. Similarly, the expression of relA, relP and relQ, encoding the bifunctional Rsh (p)ppGpp synthase/hydrolase (relA) and the RelP and RelQ synthases [56] remained unaltered (S8 Fig). A study found that persistence in intracellular S. aureus in the presence of antibiotics involved multiple protective mechanisms: transcriptional reprogramming of the cell wall stress stimulon, SOS response to preserve bacterial genome integrity or the heat-shock stimulon [43]. In case of intracellular JE2 brnQ1, however, we found none of these pathways to be differentially regulated (S1 Dataset and S8 Fig). A particular type of persistence described for S. aureus is the so-called SCV phenotype [45, 57–59]. SCVs exhibit slow growth and metabolic alterations [60] and are often the result of adaption to an intracellular lifestyle [59]. SCVs form slow-growing colonies on agar plates and lack haemolysis and pigmentation [60] however, JE2 brnQ1 recovered from macrophages after 28 days of infection exhibited normal colony size as compared to the control (i.e. brnQ1 bacteria grown in TSB), displayed haemolysis, and were pigmented (Fig 5g) indicating JE2 brnQ1 recovered from macrophages are not SCVs. In summary, growth of S. aureus within macrophages requires high affinity transport of BCAAs. BCAA deprivation, for instance mediated through brnQ1 inactivation, results in S. aureus entering a state of quiescence that is distinct from SCV phenotypes. In the BCAA-starved state, S. aureus brnQ1 stay viable within intact macrophage vacuoles for inordinate duration, even in the absence of antimicrobials.',\n",
       "  'Discussion': 'At the forefront of host-pathogen interaction is a battle for nutrients (reviewed in [10, 61, 62]). The host deploys a plethora of effectors to deprive invading bacteria of essential nutrients whereas the ability of bacteria to overcome nutrient restriction underpins their successful colonization and infection. In this study we demonstrate that the availability of BCAAs and their transport by S. aureus are pivotal to the ability of the pathogen to grow inside macrophages. Here, we found that inactivation of BrnQ1, the main BCAA transporter of S. aureus, is sufficient to arrest intracellular growth within the macrophage phagolysosome. Depriving S. aureus of BCAAs through a brnQ1 mutation dramatically alters their physiology within the phagosome, resulting in major defects in gene expression and de novo protein synthesis. Consequently, S. aureus brnQ1 mutants do not initiate appropriate patterns of gene expression and thus fail to adapt to the intracellular environment. This is emphasized by the inability of brnQ1 bacteria to upregulate the expression of Fur-regulated genes which starkly contrasts wild-type bacteria that appropriately respond to iron limitation after phagocytosis by macrophages (Fig 2). Our data suggest that S. aureus, when deprived of Val and Leu, enter a state of quiescence within macrophages but can be “re-activated” once BCAAs become available. The S. aureus genome encodes a complete BCAA biosynthesis machinery; however, the bacterium prioritises BCAA scavenging over de novo synthesis ostensibly because BCAA synthesis is energetically costly [21, 28]. S. aureus possesses three functional BCAA transporters: BrnQ1 and BcaP, which import all three BCAAs, and BrnQ2, a designated Ile transporter [24, 29]. A fourth gene encoding BrnQ3 bears homology to a BCAA transporter but lacks BCAA transport function [24]. The genes encoding the S. aureus BCAA transporters belong to the CodY regulon [25] and, as such, their transcription is de-repressed upon CodY inactivation [27, 28]. Interestingly, there is a hierarchy of de-represssion that is affected through CodY: modest decrease in intracellular BCAAs causes de-represssion of brnQ1, whereas BCAA biosynthesis, which is also CodY regulated, remains repressed at the same level of CodY inactivation [28, 63]. This is supported by our data showing that inactivation of brnQ1 renders S. aureus unable to grow inside macrophages despite possessing the capability to synthesize the BCAAs required for growth. In this scenario, brnQ1 must be expressed to some level by wild-type bacteria while the BCAA biosynthesis genes encoded by the ilv locus are likely not expressed. In S. aureus, BrnQ1 is the preferred transporter for BCAA uptake [29]. For instance, when grown in CDM lacking Val or Leu (<34 µM), BrnQ1-mediated BCAA uptake is required for growth [24]. Although BcaP can also import all three BCAAs and exhibits similar substrate affinity and transport velocities, BrnQ1 remains the predominant transporter likely due to its higher level of expression as compared to BcaP [29]. Our experiments support this, as the provision of bcaP on a multicopy plasmid is sufficient to rescue the growth of brnQ1 bacteria inside macrophages. This indicates that BCAAs are still available to some extent inside the phagosome lumen, but clearly brnQ1-deficient bacteria cannot acquire sufficient Val and Leu to support growth. This suggests that bcaP is either poorly or not at all expressed in S. aureus brnQ1 mutants. However, our transcriptome analysis did not show differential bcaP expression between JE2 brnQ1 mutant and JE2 WT (Table 1). Since S. aureus brnQ1 mutants fail to replicate inside macrophages, but do not exhibit growth defects in complex media (TSB or CDM with tryptone), the intraphagosomal milieu must be BCAA-deplete although the exact concentrations are unknown. SLC38A9, a human lysosomal membrane transporter, has been shown to remove Leu from lysosomes and it may contribute to BCAA limitation [64]. Val must also be limited in the phagosome. This notion is supported by the fact that simultaneous Leu and Val supplementation is required to rescue the brnQ1 mutant in macrophages (S13 Fig). It is unlikely that the phagosome is depleted of Ile, because the absence of this BCAA would lead to de-repression of CodY targets and subsequent de novo BCAA biosynthesis [21], followed by bacterial growth. Our data show that inactivation of codY, and therefore de-repression of the CodY regulon, is sufficient to allow the growth of brnQ1-deficient S. aureus within macrophages. More specifically, our data reveal that CodY inactivation enables growth of S. aureus brnQ1 mutants in an IlvD- and BcaP-dependent manner, indicating that BCAA biosynthesis and transport are enhanced in the codY background. This raises an important question: why wouldn’t the genes that allow growth simply be de-repressed in brnQ1-deficient S. aureus? One possible explanation is the complexity of CodY regulation, which is influenced by intracellular Ile availability and by GTP levels within the bacteria [20, 21]. Moreover, specific gene targets can be regulated not only by CodY but also by other gene expression modulatory mechanisms. For instance, the ilv operon, which encodes BCAA biosynthesis, is not only subject to CodY regulation but is also influenced by the formation of a leucine-rich attenuator peptide that downregulates the expression of ilv genes [21]. This peptide is encoded by an RNA, which was termed Rli60 in Listeria monocytogenes, where its role in fine-tuning the CodY response has been demonstrated [65]. Previous studies had concluded that S. aureus is auxotrophic for BCAAs Leu and Val [66] because the bacteria delay induction of BCAA synthesis in environments with low BCAA levels [24], especially if Ile is present as a co-repressor with CodY. Macrophage cultivation and infection in the presence of high BCAA concentrations is sufficient to rescue the replication defect displayed by brnQ1 mutants, which highlights the importance of these metabolites for S. aureus pathogenicity. Several human pathogens require functional BCAA transport or biosynthesis for full virulence (reviewed in [23]), and our study reaffirms the pivotal role of these metabolites at the host - pathogen interface. Mature phagolysosomes are highly degradative organelles, in part, due to their acidic pH and enrichment with proteolytic enzymes such as cathepsins that degrade proteins thereby liberating peptides [67]. In addition, S. aureus secretes several proteases (e.g., aureolysin or staphopain) that can degrade various substrates and liberate peptides that could serve as nutritional source [68]. Indeed, the S. aureus-containing phagosome has been shown to be proteolytic [33]. Through the activity of the oligopeptide permease Opp3 [68] and the di-tri-peptide permease DtpT [69], S. aureus could, at least in theory, utilize peptides to satisfy its amino acid requirements under conditions of nutrient limitation. Nevertheless, we found that inactivation of brnQ1 alone is sufficient to inhibit S. aureus growth inside the macrophage phagosome. Although protease expression is enhanced upon codY inactivation [25, 70], we find that simultaneous mutation of brnQ1, bcaP and ilvD is sufficient to render a S. aureus codY mutant unable to grow intracellularly. Our transcriptomic analysis revealed that genes in the opp3 operon were some of the few loci upregulated by intracellular S. aureus brnQ1 mutants suggesting that the bacteria may utilize this peptide transporter to maintain viability. The fact that brnQ1- or brnQ1/bcaP/codY/ilvD/-deficient S. aureus still fail to grow inside macrophages, would indicate that peptides are either not being used by the bacteria or that these peptides do not provide enough BCAAs to support growth. Notably, all strains with a codY mutation display increased proteolytic activity in vitro. Still, S. aureus codY mutants which additionally lack brnQ1, bcaP and ilvD (quadruple mutants), remain unable to acquire sufficient BCAAs to support growth withing macrophages (S16 Fig), indicating that peptides do not provide enough BCAAs to overcome macrophage restriction. These observations provide important insight into the contribution of free amino acids obtained from the extracellular environment versus protein degradation and peptides within the phagosome as a source of nutrition which allows S. aureus replication within the phagosome lumen. The impact of brnQ1 inactivation on the physiology of intracellular S. aureus is striking and forces the bacteria into a non-replicative state, causing them to fail to regulate adaptiveness to environmental cues. Decreased proliferation rates and/ or growth arrest and metabolic quiescence, accompanied by antibiotic tolerance, are hallmarks of persistence [50–53]. So-called persisters can arise stochastically within a bacterial population [50], or in response to environmental cues, such as exposure to antibiotics [43], nutrient starvation, oxidative stress [54], or internalization by host cells, particularly macrophages [53, 71]. Bacterial persisters are thought to constitute a reservoir for relapsing infections, associated with therapeutic failures [43, 71]. Various mechanisms were shown to trigger S. aureus persisters, such as antibiotic exposure of intracellular S. aureus [43, 54]. In macrophages, for example, stringent response induction [43] and ROS-induced defects in translation and ATP synthesis were proposed as mechanisms of persistence [54, 55]. Moreover, ROS was shown to mediate intracellular S. aureus antibiotic tolerance by disrupting the TCA cycle [55]. Antibiotic pressure is a major factor in triggering intracellular persistence of S. aureus [43, 54, 55, 72]. SCVs, an extensively studied form of S. aureus persisters [51, 57, 60, 73–76], may arise in the absence of antibiotic selective pressure [77]. They comprise a slow-growing subpopulation of bacteria with very distinctive phenotypic characteristics: reduced haemolysis and cytotoxicity, variations in pigmentation as compared to parental strains, enhanced antibiotic tolerance and slower growth rates, often due to auxotrophies (e.g., menadione or haemin) [78], resulting in non-haemolytic pinpoint-sized colonies on agar plates. S. aureus brnQ1 mutants recovered here from macrophages, however, despite being growth-arrested and non-responsive to environmental cues (i.e., iron depletion; aTc-induction of the Cerulean fluorescent protein), do not exhibit SCV characteristics, nor do they display other phenotypes reported for persisters (e.g., induction of stringent response [43]). The phenotype reported here, conforms to the definition of dormancy, as stated by Balaban et al.[51]: “Dormancy reflects the state of a bacterium that does not grow and has decreased activity when compared with growing cells”. This phenotype appears to be a non-antibiotic-driven form of intracellular dormancy triggered by BCAA starvation within macrophage phagolysosomes. This dormancy is reversible, as evidenced by the sporadic outgrowth of the bacteria from infected macrophages, in what appears to be, a stochastic event. The environmental cues or genetic changes that enable this process are currently unknown and are of research interest. One possible explanation is that the bacteria inactivate CodY, however re-infection of macrophages with brnQ1 bacteria that displayed URG does not alter or enhance the course of a subsequent infection. Furthermore, genome sequence analysis revealed no SNPs in codY or ilv genes, which were previously reported to be “hot spots” for the accumulation of suppressor mutations that allow S. aureus to eventually replicate under BCAA-deprived conditions [21]. Further investigation into how S. aureus brnQ1 exits dormancy is ongoing and will provide insight into the mechanisms S. aureus employs to modulate its metabolic program for growth and survival under nutrient-limited conditions. In summary, our study revealed a pivotal role for BCAA transport in the growth and survival of S. aureus inside the macrophage phagolysosome. Remarkably, inactivation of BrnQ1 alone is sufficient to arrest intracellular growth of S. aureus and force the bacteria into a state of temporary dormancy. These growth-arrested S. aureus reside within viable phagocytes, in the absence of any antibiotic pressure, and could serve as a long-term reservoir of bacteria with complete virulence potential, should BCAA starvation be alleviated. Understanding the mechanisms underlying this state of dormancy and how the bacteria “reactivate” is important in the rational development anti-staphylococcal therapies especially since bacterial amino acid and peptide transporters are emerging as an attractive target in the fight against virulent or antibiotic-resistant S. aureus [79]. However, our observations suggest that anti-staphylococcal strategies targeting bacterial amino acid transporters such as BrnQ1 should be carefully tailored to avoid selection of dormant intracellular populations prone to long-term intracellular survival and potential dispersal to secondary infection sites.',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1371/journal.ppat.1013291',\n",
       "  'Journal': 'PLoS Pathog'},\n",
       " {'PaperTitle': 'Endometrial Organoids and Their Role in Modeling Human Infertility',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Abdullah Jabri, Mohamed Alsharif, Tasnim Abbad, Bader Taftafa, Abdulaziz Mhannayeh, Abdulrahman Elsalti, Fayrouz Attia, Tanveer Ahmad Mir, Islam Saadeldin, Ahmed Yaqinuddin',\n",
       "  'Email': 'ayaqinuddin@alfaisal.edu',\n",
       "  'Abstract': 'Endometrial organoids (EOs) have emerged as a powerful three-dimensional (3D) model for studying the human endometrium, offering new insights into infertility and reproductive disorders. These self-organizing miniature structures closely mimic the cellular composition, hormonal responsiveness, and functional characteristics of the endometrium, making them valuable preclinical tools for investigating implantation failure, endometrial receptivity, and disease pathophysiology. This review explores the role of EOs in reproductive medicine, with a focus on their applications in infertility research, environmental toxicology, and regenerative therapies. Traditional 2D cell cultures fail to capture the complexity of these physiological and pathological interactions, whereas organoids provide a physiologically relevant system for studying implantation mechanisms. Additionally, co-culture models incorporating stromal and immune cells have further enhanced our understanding of the maternal–fetal interface. Beyond modeling infertility, EOs hold significant promise for therapeutic applications. Advances in organoid transplantation have demonstrated potential for treating endometrial dysfunction-related infertility, including conditions such as Asherman’s syndrome and thin endometrium. Moreover, these models serve as a platform for drug screening and biomarker discovery, paving the way for personalized reproductive medicine. Despite their transformative potential, limitations remain, including the need for improved extracellular matrices, vascularization, and immune system integration. This review emphasizes the significant contributions of EOs to the field of infertility treatment and reproductive biology by examining recent advancements and emerging research. The continued refinement of these models would offer a paradigm for improving assisted reproductive technologies (ARTs) and regenerative medicine outcomes, offering new hope for individuals facing infertility challenges.',\n",
       "  'Introduction': 'One of the major advances that has revolutionized the field of biomedical research is the invention of organoids. Organoids are three-dimensional (3D) in vitro structures formed from cell clusters that are capable of self-organization and differentiation into different cell types [1]. They are considered to be mini-organs as they possess cellular heterogeneity and exhibit functional characteristics of organs they resemble in vivo. The term “organoid” gained popularity in the field in 2009 when Clevers et al. were able to generate intestinal organoids capable of self-organization and differentiation into crypt-villus structures from single leucine-rich repeat containing G protein-coupled receptor 5 (Lgr5)-expressing adult intestinal stem cells [2]. Nevertheless, the history of organoids traces back to the 1900s, when Wilson described his attempt of in vitro regeneration by showing that dissociated sponge cells can self-organize to generate a whole organism [3]. These miniature organ models have been used to study disease mechanisms, drug screening, and regenerative medicine. Organoids mimic the normal behavior of tissues and organs, providing a baseline for understanding organ-specific biology. However, their ability to replicate pathological features makes them indispensable for studying diseases. For example, organoids can model disease mechanisms by mimicking the way normal cellular processes are disrupted in response to genetic mutations, environmental stressors, or other pathological stimuli. By comparing healthy organoids to diseased ones, researchers can uncover the underlying mechanisms of diseases such as cancer, liver cirrhosis, and neurological disorders. Various organoids have been developed, including those representing the intestine, liver, brain, and reproductive system, each providing insights into organ-specific biology and pathology [4, 5, 6]. Among these, endometrial organoids (EOs) have emerged as invaluable models for studying the human endometrium. The endometrium is a highly dynamic tissue that goes through periodic changes in which it becomes receptive to embryo implantation—a process crucial for fertility and successful pregnancy [7]. Though the dysregulation of endometrial receptivity is a major contributor to infertility, a lot remains to be answered. Therefore, Eos, through their unique characteristics, serve as a novel platform for investigating implantation failures and other reproductive disorders. One such application involves investigating the impact of environmental toxins—such as Bisphenol A (BPA)—which have been shown to impair endometrial function and contribute to infertility. Recent studies using EOs as a toxicology model have revealed that BPA exposure induces oxidative stress, apoptosis, and metabolic dysfunction in endometrial tissue, ultimately compromising fertility [8]. Traditional 2D cell cultures fail to capture the complexity of these disruptions, making 3D organoids a valuable tool for studying environmental toxicants and their effects on reproductive health. In this review, we aimed to explore the role of EOs in understanding and addressing infertility by examining how these organoids contribute to the study of endometrial receptivity and implantation and shed light on the molecular and cellular factors that influence successful embryo attachment. Moreover, we highlight the potential of organoid transplantation as a therapeutic strategy for endometrial dysfunction-related infertility. By integrating current knowledge and future directions, this review aims to focus on the transformative impact of EOs in reproductive medicine and their potential to advance infertility treatments.',\n",
       "  'Methods': '3. Development, Structure, and Formation Methods of Organoids Endometrial epithelial cells are isolated and cultured to create EOs; the development and upkeep of these structures rely on particular growth factors and extracellular matrices [15, 16, 17, 18, 19]. Unlike conventional 2D cultures, which lack cellular diversity and long-term viability, 3D EOs provide a more physiologically relevant system for studying uterine function. Recent research highlights how organoids derived from fertile and infertile patients exhibit distinct hormonal responses, making them useful for implantation and infertility studies. Additionally, microfluidic systems have been introduced to enhance the study of maternal–embryonic interactions, offering a new frontier for implantation failure research [20]. EOs can be derived from a variety of sources, such as postpartum placenta, tumor tissues, scratch biopsies, and menstrual flow (Figure 2) [17, 18, 19]. Typically, patient-derived samples are enzymatically dissociated using agents such as collagenase IV, TrypLE, or liberase, followed by mechanical trituration to digest the tissues and generate endometrial and endometriotic organoids [15, 16, 21]. The isolated epithelial cells are then embedded in a 3D matrix, such as Matrigel, and cultured in advanced DMEM/F12 medium supplemented with a ROCK inhibitor to promote cell proliferation and survival. The culture media is additionally enriched with vital growth factors, such as fibroblast growth factor 10 (FGF10), hepatocyte growth factor (HGF), epidermal growth factor (EGF), and R-spondin-1 (RSPO1), which activate important signaling pathways like MAPK and Wnt/β-catenin [22, 23, 24, 25]. Additional components such as Noggin and the TGF-β inhibitor A83-01 promote self-renewal by inhibiting BMP and TGF-β signaling, respectively [26, 27, 28, 29, 30, 31]. Similar to native endometrial glands, EOs are spheroidal structures surrounded by a single columnar epithelium [32]. A mix of epithelial markers, such as EPCAM, PanCK, and E-cadherin, preserves their structural integrity. Progenitor cells (LRIG1, PROM1, AXIN2, SOX9), epithelial cells (EPCAM, KRT7, CDH1, SOX17), secretory cells (PAEP, MUC1, PAX8, HSD17B2), and ciliated cells (FOXJ1, PIFO, RSPH1) are among the several cellular subtypes of EOs that have been identified through genetic studies [11, 19, 33, 34, 35]. Depending on the tissue source and the goals of the investigation, various techniques are used to produce EOs (Figure 3). For example, endometrial samples from hormonally untreated patients undergoing laparoscopy for benign gynecological disorders are mechanically triturated, enzymatically dissociated with collagenase IV, and cultured in specialized media within a 70% Matrigel matrix [21]. After being digested by collagenase I, DNase, and dispase, cell suspensions obtained from premenopausal hysterectomy samples are filtered and cultivated in MammoCultTM medium under epithelial–stromal co-culture conditions [36]. Likewise, single-cell suspensions of human endometrial and decidual samples can be prepared, embedded in hydrogels such as TeloCol-6, and maintained in an ideal organoid culture medium with regular media changes every two days [16, 37]. Organoid development is monitored over a 20–30-day period after single epithelial cells are implanted in Matrigel to determine clonogenic potential [15]. Periodic passaging every 10–20 days ensures the expansion and viability of EOs for biobanking purposes and allows for long-term maintenance [21]. Recent advances in synthetic hydrogel formulations that mimic the endometrial extracellular matrix have enhanced physiological relevance by improving organoid stability and epithelial–stromal interactions [38, 39, 40, 41, 42, 43, 44, 45, 46].',\n",
       "  'Results': '2. Endometrial Receptivity and Implantation Endometrial receptivity is defined as the period when the endometrium permits the trophectoderm of the blastocyst to attach and invade (Figure 1). This receptive phase is tightly regulated by estrogen and progesterone; the latter plays a crucial role in promoting anti-inflammatory actions, immune tolerance, and receptor downregulation, all of which ensure timely implantation and the maintenance of early pregnancy. Receptivity also includes key molecular events like the downregulation of estrogen receptor-alpha (ER-α) and the expression of progesterone receptors (PRs). In women with regular menstrual cycles, the window of receptivity occurs during the mid-luteal phase of the menstrual cycle, approximately 6 to 10 days after the luteinizing hormone (LH) surge, although the precise timing may vary depending on cycle length and hormonal dynamics [9]. The implantation process is divided into three sequential phases: (i) apposition, (ii) adhesion, and (iii) invasion. During these stages, numerous molecular mediators—regulated by ovarian steroid hormones—facilitate the initial maternal–fetal interaction. These include adhesion molecules, cytokines, growth factors, lipids, and other signaling factors [10]. Key markers of endometrial receptivity include endometrial thickness, volume, and morphological appearance. EOs effectively model the cellular environment necessary for implantation by mimicking the structural and functional features of the endometrial epithelium, which undergoes essential transitions during receptivity. Derived from human endometrial epithelial cells, these organoids self-organize into three-dimensional, gland-like structures and exhibit apicobasal polarity similar to that of native endometrial glands. This polarity allows for the separate collection of apical and basal secretions to investigate their respective roles in implantation [6]. EOs closely resemble the mid-secretory phase of the menstrual cycle and exhibit strong responsiveness to hormones like progesterone and estrogen [11]. This includes the upregulation of key receptivity markers such as Heat Shock Protein A9 (HSPA9), which enhances epithelial adhesiveness for blastocyst attachment, and Dipeptidyl Peptidase 4 (DPP4), which promotes glandular development [12]. Furthermore, organoids release apical secretions into their inner lumen that simulate uterine fluid, containing proteins like DPP4 and HSPA9. Dysregulation of these indicators has been observed in organoids derived from infertile patients through high-throughput screening of these secretions, providing a model to investigate causes of faulty receptivity [11]. The basal secretions of EOs resemble the communication between stromal and epithelial cells, by replicating stromal signals necessary for implantation [13]. For instance, EOs secrete extracellular vesicles carrying proteins like cystatin C, which controls stromal decidualization, and microRNAs like miR-92a-3p, which modulate receptivity. Organoids can also be modified to form epithelial monolayers that mimic the luminal epithelium—a surface required for blastocyst attachment. In conjunction with Wnt inhibitors, hormonal therapies improve blastocyst adhesion and allow for the functional modeling of contraceptive drugs by increasing the expression of receptivity genes such as Leukemia Inhibitory Factor (LIF), DPP4, and glutathione peroxidase 3 [14]. 4. Organoid Transplantation as a Therapeutic Strategy for Infertility Endometrial organoid transplantation (EOT) is one of the regeneration techniques being investigated because conventional treatments like hormone therapy or surgical adhesiolysis often fail to restore functioning endometrial tissue [47]. EOT gives patients with refractory infertility hope by promoting structural regeneration, improving vascularization, and improving reproductive outcomes [48]. EOT has demonstrated exceptional effectiveness in restoring endometrial thickness and lowering fibrosis in animal models of endometrial damage. In rats, for instance, endometrial thickness was recovered to 476.80 ± 54.26 µm after receiving a transplant of Müllerian duct-like cells (MDLCs) produced from human pluripotent stem cells (hPSCs), greatly exceeding that of untreated controls [49]. Similarly, He et al. highlighted the role of angiogenesis in tissue repair by observing decreased fibrotic areas and increased vascular endothelial growth factor A (VEGFA) after transplanting stage-specific embryonic antigen-1 positive (SSEA-1+) endometrial epithelial stem cells [50]. Increased micro-vessel density and VEGF expression were noted in treated mice, indicating enhanced vascularization, which is essential for embryo implantation [49]. Moreover, transferring mitochondria from organoids to host cells is a unique repair strategy that improves cellular metabolism and repair. These structural improvements correlate with functional outcomes, including restored hormonal responsiveness and decidualization capacity—key processes for embryo implantation [49]. The therapeutic potential of EOT is further validated by pregnancy outcomes. Zhang H et al. reported successful pregnancies in three out of four mice with intrauterine adhesions (IUAs) post-transplantation [51]. Similarly, Gong et al. reported an 83.33% pregnancy rate in rats treated with MDLC, compared to 50% in injured controls [49]. Litter size improvements were also observed; for example, mice receiving endometrial mesenchymal stem cell spheroids had an increase from roughly three to five pups [52]. Clinically, EOT shows potential for certain causes of infertility. According to Wiweko, autologous cell-based therapies for thin endometrium can improve endometrial receptivity by promoting epithelial regeneration and E-cadherin expression [53]. Another use is postoperative recovery; Park et al. used 3D stem cell-laden constructions to achieve live births in endometrial ablation models, with no morphological or chromosomal abnormalities seen in the offspring [54]. 5. Endometrial Organoids as Models for Infertility and Endometrial Diseases EOs have revolutionized endometrial research by providing a 3D model that mimics the in vivo endometrium. Derived from human endometrial tissue, these organoids can mimic key structural and functional features of the endometrium, making them significantly useful for studying diseases like endometriosis, endometrial cancer, and infertility. Infertility is clinically defined as the failure to achieve pregnancy after 12 months or more of regular, unprotected intercourse [55]. Around one in six couples are affected by infertility globally, with about 30% of the cases attributed solely to the female factor [56]. Traditionally, animal in vivo models have been used to model the human endometrium [57]. Although these studies provided important insights, they did not accurately model the human endometrium [58]. Hence, functional in vitro studies of the human endometrium are needed to explore the different causes and conditions of infertility. 5.1. Endometriosis Endometriosis is a chronic estrogen-dependent condition where functional tissue lining the uterus grows outside the uterine cavity [59]. The main symptoms are chronic pelvic pain and infertility. Endometriosis is a big health issue that is estimated to affect between 5 and 15% of women of reproductive age in the United States and Canada [60]. It is considered incurable, and, despite extensive research, no cause has been found; only theories exist. This situation highlights the need for new investigative and therapeutic approaches, like EOs. The first endometriosis organoid model was cultured in 2013, which more accurately mimicked the molecular and histological features of endometriosis compared to the previous 2D models [61]. Further models also confirmed the similarity between EOs and endometrial tissue [15, 62]. Boretto et al. developed an organoid of eutopic and ectopic endometriosis that possessed the typical features of endometriosis and cancer-linked mutations [15]. Eshfendiari et al. observed significant alterations in methylation levels of HOX clusters and cofactors in both ectopic and eutopic endometrial tissues, as well as in the corresponding organoids, compared to normal endometrial tissue [62]. These findings concluded that EOs can maintain the epigenetic changes and thus can be used to study endometriosis further. Later, the research group established another endometriosis organoid model that showed that progesterone receptor B (PR-B) is downregulated in ectopic and eutopic EOs compared to healthy controls, yet through different mechanisms [63]. Although eutopic endometriosis organoids exhibited the hypermethylation of PR-B and ectopic endometriosis organoids showed the hypomethylation of PR-B, it was downregulated anyhow. This inconsistency suggests that other epigenetic mechanisms, such as histone modifications or micro-RNAs (miRNAs), may be responsible for the downregulation of PR-B in endometriosis. 5.2. Endometrial Dysfunction In 2020, Bui et al. established organoid lines from cryopreserved endometrial tissue obtained from infertile women [64]. Following this, several studies used organoids to compare the endometrium of fertile and infertile women [65, 66, 67]. They noted the upregulation of cell cycle processes during the secretory phase of the infertile organoids. Moreover, it was found that there was increased proliferative activity in the glandular cells of the endometrium in infertile women compared to fertile women. This high proliferative activity in the secretory phase can lead to inadequate differentiation and a poorly receptive endometrium for embryo implantation. Conversely, another study has shown the downregulation of the cell cycle in the luteal phase in the infertile endometrium [68]. These findings suggest the cell cycle is altered in different ways in the infertile endometrium, which indicates that no one cause of infertility applies to all. One of the key applications of EOs is their ability to model implantation disorders. Studies have shown that fertile vs. infertile EOs differ in their response to estrogen and progesterone, suggesting that implantation failure may stem from underlying endometrial receptivity defects [20]. Furthermore, microfluidic-based organoid models now allow for the study of trophoblast attachment and early embryo interactions, making them valuable for research into maternal–fetal health [20]. 5.3. Asherman’s Syndrome Asherman syndrome (AS) is a triad of pain, menstrual irregularities, and infertility caused by intrauterine scar tissue after the instrumentation of the uterus [69]. In 2023, Santamaria et al. investigated AS pathophysiology by comparing organoids from AS patient cells to healthy individuals [70]. The AS organoids had a loss of endometrial epithelium, altered epithelial differentiation signaling pathways (Wnt and Notch), and secretory leukocyte protease inhibitors during the implantation window. They also had changes in cell-to-cell communication and gene expression profiles, showing a dysfunctional pro-fibrotic, pro-inflammatory, and anti-angiogenic environment. Other studies focused on applying EOs in treating Asherman syndrome [71]. For instance, Hwang et al. demonstrated that transplanting endometrial tissue-derived organoids into a murine model of AS reduced the damaged structure of the AS endometrium by dramatically reducing fibrotic lesions and increasing cellular proliferation and vessel formation, eventually resulting in improved embryo implantation [16]. 5.4. Endometrial Cancer Endometrial cancer is one of the most common gynecological malignancies in women [72]. With endometrial cancer rates on the rise, there is an unmet need for biological models to study cancer formation, metastasis, and disease progression and recurrence. There are limited data on the molecular and cellular basis of endometrial cancers, especially in the rarer and more aggressive subtypes. This limitation is partly due to the lack of high-fidelity models that mimic the in vivo properties of human endometrial cancer. Fortunately, 3D organoids have emerged to mimic the microenvironment and cellular milieu more accurately than 2D models, and they also allow for a more in vivo-like representation of the endometrium [15, 73]. Several studies have already used EOs to further investigate endometrial cancer. For example, Chen et al. used mouse models and organoids derived from endometrial cancer patients to study SNORD14E and how it affects those with endometrial cancer [74]. They discovered that SNORD14E was over-expressed in cancer tissues. Moreover, patients with high levels of the gene had worse prognosis without differences in distribution among the biomolecular classification subgroups of endometrial cancer. The authors also established that targeting SNORD14E with an antisense oligonucleotide (ASO) could be a promising treatment for endometrial cancer. Chen et al. created EOs models with mutations in PTEN, PIK3A, and PI3KR1, which are commonly associated with endometrioid adenocarcinoma. They observed that these co-mutations lead to faster tumor development [75]. Additionally, they also established a drug screening pipeline that tested 56 small molecule compounds, revealing varying responses among organoids. Certain organoids displayed drug resistance, while others, such as those treated with dacomitinib, showed promising responses. Other similar studies have highlighted the therapeutic potential of EOs in the novel treatments of endometrial cancer [76, 77, 78, 79]. 5.5. Endometrial Infection Endometrial infections, particularly those caused by chronic or untreated infections, can lead to scarring and dysfunction of the endometrium, resulting in infertility [80]. In this context, the use of EEOs could help elucidate the mechanisms through which infection alters endometrial structure and function, providing insight into therapeutic approaches for managing infertility linked to infection. Interestingly, Łaniewski et al. successfully simulated ascending infections of the female reproductive tract using EOs [81]. The authors infected the EOs with commensal and pathogenic bacteria such as Lactobacillus crispatus, Gardnerella vaginalis, and Neisseria gonorrhoeae. Of all the different bacteria strains, only Neisseria gonorrhoeae triggered a significant proinflammatory reaction and caused significant ultrastructural tissue changes. Łaniewski et al. were the first to describe gonococcal infection of epithelial cells in a 3D cell culture. Other studies later simulated infections of Chlamydia trochamitis using EOs [82]. Dolat et al. observed that Chlamydia trochamitis induced cytoskeletal reorganization and a change in the intracellular organelle positioning [83]. Additionally, the organoids can be co-cultured with neutrophils to reconstruct immune cell responses, revealing that specific effectors like CPAF and TepP limit neutrophil recruitment to infected organoids. 5.6. Environmental and Drug-Induced Toxicity Emerging research highlights the toxic effects of BPA and certain neurotherapeutic drugs on endometrial function. BPA has been shown to disrupt epithelial–mesenchymal transition (EMT), Wnt/β-catenin signaling, and metabolic homeostasis, all of which are essential for maintaining endometrial integrity [8]. Additionally, some drugs can interfere with reproductive function at the molecular level. Notably, melatonin and resveratrol have been found to mitigate BPA-induced damage, restoring cellular integrity and reducing apoptosis [8]. These findings further emphasize the potential of EOs as a model for both environmental and drug-induced toxicology studies, providing valuable insights into reproductive health risks. EOs are increasingly recognized as essential tools in reproductive toxicology research. Traditional preclinical models often fail to predict the reproductive side effects of neurotherapeutic drugs, leading to unforeseen complications in clinical trials [84]. Recent advancements highlight how organoids can serve as alternative models for drug safety testing, helping to assess reproductive toxicity early in drug development. The integration of high-throughput drug screening with organoid platforms, alongside environmental exposure studies, may further enhance personalized medicine approaches in reproductive health. 6. Impact of Organoids on Assisted Reproductive Technologies (ARTs) and Biomarker Development Infertility is a huge problem in today’s societies and has a significant social and psychological impact on women. Treatment is needed—from medical and surgical interventions to assisted reproductive technologies (ARTs)—depending on the cause [32]. Despite the many ways to administer external estrogen and progesterone to prepare the endometrium in women undergoing in vitro fertilization (IVF)—one of the ART methods available today—achieving a clinical pregnancy is not guaranteed. There is ongoing debate about the type, dosage, and timing of these hormonal supplements [85]. For example, high doses of estrogen are essential during the window of implantation, but high doses of estrogen have been associated with implantation failures in mice and increased risk of ovarian hyperstimulation syndrome. On the other hand, high concentrations of progesterone used to treat luteal phase deficiency negatively impact uterine receptivity and decidualization, as shown in mice in vivo and immortalized human endometrial stromal cells in vitro [86, 87, 88]. Another major cause of infertility is the inability of human embryos to implant successfully. This is highlighted by frequent early pregnancy losses and the low success rate of IVF, which is 31% per fresh embryo transfer, as reported by 2022 data from the Human Fertilization and Embryology Authority (HFEA) [89]. Modeling implantation is key to understanding the interactions between the embryo and the mother during early pregnancy. Despite the progress in IVF technology, understanding the implantation process is still a major limitation of IVF treatments [90]. Implantation cannot be studied directly in the human body, so scientists rely on animal models and cell culture methods. However, recent developments in 3D modeling, especially with organoids, offer new avenues to explore previously inaccessible aspects of human reproduction [91]. EOs can be generated from biopsies obtained from infertile women undergoing IVF treatment—using either fresh or frozen samples—and exposed to varying concentrations of estrogen and progesterone, to evaluate their effects on cellular proliferation and differentiation [64]. The expression of estrogen receptor (ER) and progesterone receptor (PR) can also be assessed in these organoid cultures to see if the endometrium can achieve a receptive state in response to progesterone [32]. Various in vitro models have been developed to simulate implantation, which is a key to improving IVF success rates, including primary endometrial epithelial cultures, immortalized epithelial cell lines, and cell lines such as Ishikawa, low-receptivity HEC-1-A, high-receptivity RL95-2 and ECC-1 derived from endometrial carcinoma. Decidualization models use the hormonal stimulation of ESCs, while embryo surrogate models use trophoblastic cell lines and spheroids. More complex assays like Transwell are also used. Although these systems mainly focus on the attachment or invasion of embryo surrogates to a 2D monolayer, they are useful for modeling specific stages of implantation and screening factors that influence embryo adhesion and invasion [92]. Rawlings et al. created an endometrial assembloid to study embryo implantation and decidual senescence, defined as a metabolically active state without cell division [91]. In their model, they embedded a human embryo into the assembloid and added dasatinib, a tyrosine kinase inhibitor that gets rid of senescent decidual cells. Senescence is important to prevent miscarriage and implantation failures, but the removal of these cells, by dasatinib, trapped the embryo in a non-progressive decidual matrix and prevented implantation. However, the model is limited as it does not resemble the real endometrium as it lacks the lining epithelium and immune components. To improve the still-imperfect in vitro conditions for gamete preparation and embryo culture during in vitro fertilization (IVF), researchers created a human fallopian tube organoid from adult stem cells to culture spermatozoa in the apical compartment. Human fallopian tubes are the site of key events for a successful pregnancy, and modeling them could improve IVF outcomes. Their model differentiated into fallopian tube cells, as confirmed by multiple analyses, and worked for sperm culture. Sperm vitality in the human fallopian tube (HFT) organoids was similar to commercial sperm media. Notably, sperm motility was higher in the HFT organoids than in all other conditions and remained higher even after 96 h compared to all other tested environments [93]. In another study, Barry et al. showed that a biphasic (5–2%) O2 concentration during preimplantation embryo culture improves blastulation and live birth rates in IVF [94]. Their study shows that the transcriptomic analysis of trophoblast organoids cultured in different O2 concentrations can be used to mimic in vivo conditions and improve our understanding of embryo development in ARTs. HIF1a was upregulated in biphasic conditions and not in monophasic conditions, and organoid models can provide valuable insights into embryo angiogenesis and vascularization and improve ART outcomes. Repeated Implantation Failure (RIF) is a clinical condition diagnosed in the context of ARTs and is often associated with endometrial receptivity and endometrium–embryo interaction [65, 95]. A previous study on endometrial transcriptomes from whole-tissue biopsies showed numerous differentially expressed genes (DEGs) in the endometrium of women with RIF compared to fertile controls [96]. This means there are differences in endometrial function between the two groups that impact fertility outcomes. Hormone imbalance can affect the timing of endometrial receptivity, which is critical for the synchronization of successful implantation. One study showed that 2D epithelial monolayers treated with estradiol (E2), progesterone (P4), 8- Br-cAMP(C), and XAV939 (X)—shortly EPCX—expressed more receptivity genes and had higher blastoid adhesion rates than controls. Blastoids are structures derived from stem cells that resemble the blastocyst stage of human embryos. This setup allowed for the precise measurement of functional implantation failure and showed significant trophoblast cell migration. These results open up possibilities for the further study of hormonal impact on implantation failure, but the model is limited by its simplicity (lacking stromal, endothelial, and immune cells) [95]. Clinical studies have shown that intrauterine insemination of seminal plasma (SP) during assisted reproductive technologies (ARTs) can improve implantation and pregnancy rates [97]. The exact molecular mechanisms by which SP works is not fully understood. SP contains TGF-b1 or IL-8, which can condition the uterus to be more receptive to pregnancy [98]. SP also induces notable transcriptional changes in 2D endometrial epithelial cells. Recently developed 3D human EOs, which more closely mimic in vivo conditions, can provide a better understanding of how the endometrium interacts with SP. This was tested in a study where endometrial tissue samples from fertile and sub-fertile women (defined as those not achieving pregnancy after at least a year of regular unprotected intercourse) were used to create endometrial epithelial organoids [99]. SP from fertile men was collected, pooled, and used in the experiment. The organoids were grown in a complex medium enriched with 17β-estradiol (E2) and incubated with or without a small amount of SP for six hours. The study showed significant variation between samples from different donors. This variation was attributed more to individual differences than to fertility status or SP treatment. This shows how ARTs can advance by using organoids to study how seminal plasma affects endometrial receptivity. Results can lead to personalized treatment and improved implantation strategies and increase ART success rates. Alternatively, organoids can be derived from non-invasive methods such as menstrual flow-derived organoids, as biopsies of the endometrial tissue may be considered difficult or non-feasible due to ethical considerations. Davies et al. demonstrated that both sets of organoids exhibit identical transcriptome signatures, derivation efficiency, and proliferation rates [18]. Such findings hold significant potential for enhancing assisted reproductive technologies by enabling non-invasive investigations and tailored treatments for reproductive disorders, including failed implantation following IVF and recurrent miscarriage. 7. Biomarker Development A biomarker is a measurable indicator of cellular or organismal state at a given time. Biomarkers are essential for understanding the link between environmental chemicals and human disease, to diagnose, track, and predict disease risk [100]. EOs derived from human biopsies are a great platform for biomarker development in reproductive health (Table 1). They can be grown long-term, are genetically stable, and can be frozen [101, 102]. By exposing them to hormones like estrogen and medroxyprogesterone acetate, changes in gene expression can be simulated, like the endometrial cycle. This is supported by markers like FOXA2 and changes in steroid hormone receptor expression. Techniques like real-time qPCR and bulk RNA sequencing confirm that hormone treatment induces gene expression in a way that mimicked the proliferative and secretory phases of the endometrium [103]. Biomarkers can serve as prognostic tools for the pathologic classification of endometrial carcinoma, which remains inconsistent [104]. To address this, Cochrane et al. and others employed single-cell sequencing to study organoid model systems derived from normal endometrial tissue to identify new markers for endometrial ciliated or secretory cells [113]. Secretory cell marker (MPST) and ciliated cell markers (FAM92B, WDR16, and DYDC2) were validated using immunohistochemistry on organoids and tissue sections. Furthermore, single-cell sequencing of endometrial and ovarian tumors revealed secretory-like and ciliated-like tumor cells. Notably, the expression of ciliated cell markers (DYDC2, CTH, FOXJ1, and p73) and the secretory marker MPST found in endometrial tumors correlated with better survival outcomes, suggesting their utility in stratifying endometrial carcinoma by aggressiveness. Glycodelin-A (GdA), a glycoprotein from the lipocalin superfamily, plays a role in reproduction and fetal–maternal immune tolerance due to its unique glycan structures [114]. Recent studies showed that both the expression and glycoform of GdA are changed in the eutopic endometrium of women with endometriosis [105]. Interestingly, organoids derived from these patients replicate this altered expression while maintaining morphological similarity to a healthy endometrium. This makes organoids a valuable tool for studying implantation biology and for developing biomarkers to aid in the diagnosis and treatment of gynecologic disorders [115].',\n",
       "  'Discussion': '8. Current Limitations and Future Insights of Endometrial Organoids EOs have emerged as a powerful tool for studying the human endometrium, providing new insights into endometrial biology, disease mechanisms, and therapeutic strategies. However, like any model system, they come with inherent limitations that must be addressed to fully realize their potential. 8.1. Technological Limitations Currently, Matrigel, which was isolated from Engelbreth–Holm–Swarm mouse sarcoma, is used as the primary extracellular matrix in the EO culture [116]. For animal-derived and ill-defined traits, it may not fully mirror the human endometrial environment, despite its effectiveness in encouraging organoid growth and self-organization. Several artificial and chemically defined hydrogels, extracellular matrix hydrogels made from decellularized tissues, and microstructured collagen scaffolds have been developed for 3D cell culture and organoid cultures to overcome these limitations [106, 107]. Furthermore, endometrial epithelial and stromal cells have been co-cultured using collagen scaffolds and stromal cell-induced scaffolds using agarose 3D Petri dishes, suggesting that alternative types of culture materials—rather than Matrigel—are equally viable in the EO culture [108, 109]. Recent innovations in microfluidic systems and synthetic extracellular matrices (ECMs) have significantly improved EO models [20]. These systems allow for precise control over hormonal fluctuations, nutrient exchange, and cellular interactions, bringing organoid models closer to clinical applications in fertility research. Because EOs have a basal out/apical phenotype, they are not completely identical to the endometrium in vivo. This makes it difficult to explore their surface markers in binding assays. Furthermore, transfecting nucleic acids into cells is challenging due to the organoid epithelium tight junctions. Simintiras et al. discovered that certain metabolites were specific to extra-organoid fluid, which was the result of the organoid system’s asymmetrical apical and basolateral secretion [110]. Apical proteins in organoids could be studied in this scenario by microinjection [83]. To overcome these difficulties, an organoid-specific transfection methodology should be created, and a matrix gel-independent suspension culture system should be utilized for reverse polarity [111]. It is unclear how organoids interact with other tissues, mostly because of the polarity and the question of whether co-cultured cells, tissues, and organoids would be affected by Matrigel or EO media. The establishment of corresponding co-culture systems is necessary to replicate real in vivo conditions. Moreover, microfluidic technologies may be used to solve the challenging in vitro alignment of various biological systems [117]. For instance, a microfluidic device was developed to replicate the characteristics of the human menstrual cycle in 28 days using various tissues from the female reproductive tract and sex hormones from ovarian follicles [118]. To investigate the communication between the human ovary and endometrium, an organ-on-a-chip system was also created [112]. Chip models and organoids can be used to replace 2D cell cultures and tissue explants, improving the biological relevance to the original tissue. 8.2. Biological Constraints There are technical challenges in the establishment and maintenance of EOs because of the lack of cellular heterogenicity and the limited cell types. Currently, EOs can only be used to study epithelial cells rather than other cell types such as stromal and immune cells which also play an important role in disease development [118]. Co-culturing stromal and endometrial epithelial cells has been accomplished in several experiments, as previously mentioned. Instead of using Matrigel, the EOs were cultivated in novel ECM types. While in these models stromal cells were cultured using agarose-based 3D Petri dishes to generate their own ECM, others used preformed 3D porous collagen scaffolds as the ECM substrate [36, 107]. Thus, it can be seen that co-culture techniques and the investigation of new cultural ECMs benefited from one another. These co-culture models, however, lack maturity and persuasiveness because it was not determined whether they could be long-term or phenotypically and genotypically stable. Additionally, immune system anomalies and vascular dysfunction are intimately linked to endometrial and pregnancy disorders. Pregnancy, for instance, is facilitated by communication between the epithelium, decidualized stromal cells, resident immune cells, vasculature, and placenta trophoblast; any form of cell dysfunction would impact the outcome. The function of immune cells and the vasculature must be taken into account to investigate the pathogenesis. Using agarose 3D Petri dishes to study scaffold-free EOs, no immune cells were virtually present based on the detection of typical surface markers. Additionally, there is no a report on the vasculature’s function in EOs yet. As a result, EOs should enhance the process for co-culturing stromal and epithelial cells while working to investigate the technology for co-culturing immune and endothelial cells [100]. 8.3. Translational Barriers in Endometrial Organoid Transplantation (EOT) The findings of this review, which synthesize a wide range of preclinical evidence in favor of EOT, are primarily descriptive reviews of previous research rather than critical analyses. To put these developments into perspective, a more thorough analysis of methodological limitations and translational barriers is necessary. To generate organoids, for example, several research studies use heterogeneous protocols that vary in scaffold materials, transplantation methods, and cell sources (e.g., primary cells vs. stem cells) [47, 48, 53]. This discrepancy makes direct comparisons more difficult to determine the ideal settings for clinical use. Additionally, whereas preclinical models show encouraging structural and functional recovery, they frequently fail to account for species-specific variations in endometrial physiology, which calls into question whether results can be extrapolated to humans. Clinical translation still faces difficulties in spite of these developments. Human-specific data is still scarce, and the majority of studies use rodent models. Santamaria et al. emphasized the need for patient-specific optimization by highlighting the transcriptome differences between organoids produced from patients with Asherman’s syndrome and healthy controls [70]. Standardization is complicated by protocol variability, including time of transplantation, culture conditions (Matrigel vs. fibrin hydrogels), and organoid sources (primary cells vs. stem cells). Although there are currently no negative results from trials, safety concerns, such as the tumorigenic potential of stem cell-derived organoids, necessitate further long-term research [55]. Another obstacle is scalability because it still requires considerable resources to produce clinical-grade organoids and scaffolds. Future research should focus on mechanistic studies to elucidate organoid–host interactions, standardized techniques to improve reproducibility, and cooperative frameworks to tackle ethical and regulatory issues in order to close these gaps. Prioritizing human trials to assess safety and effectiveness in conditions like AS or thin endometrium is essential for future advancement. Reproducibility will be improved by standardizing procedures for organoid culture, scaffold composition, and transplantation timing. While mechanistic research clarifying mitochondrial transfer and immunomodulatory pathways may improve therapeutic targeting, autologous techniques employing patient-derived cells may reduce the danger of immunological rejection [54]. EOT is a revolutionary approach to treating infertility caused by endometrial malfunction. For patients with few other therapy options, EOT presents a viable solution by promoting embryo implantation, improving vascularization, and repairing tissue architecture. Despite the tremendous preclinical promise of EOT, the transition from lab to bedside runs the risk of halting without a thorough review. 8.4. Ethical Considerations The use of animal-derived materials such as Matrigel raises ethical concerns related to animal welfare, reproducibility, and translational relevance. As Matrigel is sourced from mouse sarcoma, its continued use may limit the clinical applicability of findings and conflicts with efforts to develop fully defined, xeno-free culture systems. Moreover, future applications of EOs in personalized medicine, reproductive health, and genetic manipulation will require careful ethical evaluation, particularly regarding consent, data privacy, and potential off-target effects in gene-editing approaches. Despite the present limitations of EOs, the current results are promising and warrant further experiments and studies in the field. Eos.',\n",
       "  'Conclusion': '9. Conclusions EOs have served as a physiological model, resembling in vivo tissue, for studying endometrial receptivity and implantation. Their ability to respond to hormonal stimuli and mimic the cyclic hormonal changes of the endometrium has made them of great value for studying implantation failures and pregnancy loss. Moreover, they hold a promising future in restoring function in patients with infertility due to endometrial dysregulations. Taking into account the various limitations of current models, future research should focus on integrating microfluidic systems, co-culturing with immune and stromal cells, and improving transplantation techniques to optimize their therapeutic potential. Their therapeutic potential may be further enhanced by developments in bioengineering and personalized medicine, opening the door to patient-specific therapies.',\n",
       "  'URL': 'https://doi.org/10.3390/cells14110829',\n",
       "  'Journal': 'Cells'},\n",
       " {'PaperTitle': 'Differential Resting State Connectivity Responses to Glycemic State in Type 1 Diabetes',\n",
       "  'Publication': '2019',\n",
       "  'Authors': 'Lisa Parikh, Dongju Seo, Cheryl Lacadie, Renata Belfort-Deaguiar, Derek Groskreutz, Muhammad Hamza, Feng Dai, Dustin Scheinost, Rajita Sinha, R Todd Constable, Robert Sherwin, Janice Jin Hwang',\n",
       "  'Email': 'Janice.hwang@yale.edu',\n",
       "  'Abstract': 'Context Individuals with type 1 diabetes mellitus (T1DM) have alterations in brain activity that have been postulated to contribute to the adverse neurocognitive consequences of T1DM; however, the impact of T1DM and hypoglycemic unawareness on the brain’s resting state activity remains unclear. Objective To determine whether individuals with T1DM and hypoglycemia unawareness (T1DM-Unaware) had changes in the brain resting state functional connectivity compared to healthy controls (HC) and those with T1DM and hypoglycemia awareness (T1DM-Aware). Design Observational study. Setting Academic medical center. Participants 27 individuals with T1DM and 12 HC volunteers participated in the study. Intervention All participants underwent blood oxygenation level dependent (BOLD) resting state functional magnetic brain imaging during a 2-step hyperinsulinemic euglycemic (90 mg/dL)–hypoglycemic (60 mg/dL) clamp. Outcome Changes in resting state functional connectivity. Results Using 2 separate methods of functional connectivity analysis, we identified distinct differences in the resting state brain responses to mild hypoglycemia between HC, T1DM-Aware, and T1DM-Unaware participants, particularly in the angular gyrus, an integral component of the default mode network (DMN). Furthermore, changes in angular gyrus connectivity also correlated with greater symptoms of hypoglycemia (r = 0.461, P = 0.003) as well as higher scores of perceived stress (r = 0.531, P = 0.016). Conclusion These findings provide evidence that individuals with T1DM have changes in the brain’s resting state connectivity patterns, which may be further associated with differences in awareness to hypoglycemia. These changes in connectivity may be associated with alterations in functional outcomes among individuals with T1DM.',\n",
       "  'Introduction': 'Type 1 diabetes mellitus (T1DM) is associated with a host of changes in brain function including cognitive deficits (1–4) as well as neuropsychological changes such as increased risk of depression and anxiety (5–7). The mechanisms underlying these associations are incompletely understood but may involve both structural cerebral changes as well as functional changes in brain activity (4). It has been postulated that many of these changes in cerebral function are secondary to exposure to not only chronic hyperglycemia but also frequent episodes of hypoglycemia. While normally the body has multiple layers of defense mechanisms to prevent hypoglycemia, patients with T1DM are reliant on exogenous insulin and have deficits in counterregulatory hormonal responses to prevent hypoglycemia. Patients with T1DM who have frequent episodes of hypoglycemia often develop hypoglycemia unawareness, which is characterized by an inability to mount the normal symptomatic and counterregulatory hormonal responses required to restore plasma glucose levels. Individuals with T1DM also have different responses in regional brain activity in response to working memory tasks (8) as well as visual food cues (9) during mild hypoglycemia. Moreover, these responses are further modulated by the presence or absence of hypoglycemia awareness (9). These studies suggest that brain responses to specific tasks are altered by T1DM and hypoglycemia unawareness. However, over the past 2 decades, there has been a growing understanding that the vast majority of the brain’s energy is devoted to intrinsic neuronal signaling, that is, activity not related to any tasks or in response to environmental stimuli. This activity has been called resting state functional connectivity because different regions of the brain that are functionally coordinated will fluctuate in a synchronous pattern “at rest” or in the absence of stimuli. In fact, it has been suggested that nearly 95% of the brain’s energy budget is utilized during intrinsic neuronal signaling (10, 11). Furthermore, a growing body of work has revealed compelling relationships between alterations in resting state functional connectivity and numerous neurologic and psychiatric diseases including anxiety and depression (12) as well as metabolic diseases such as obesity and type 2 diabetes (13). In addition, children and adults with T1DM appear to have changes in functional connectivity as well. In a study using resting state functional magnetic resonance imaging (rsfMRI), children with T1DM had compensatory hyperconnectivity compared to children without diabetes in the dorsal and ventral attention networks (14). Adults with well-controlled T1DM patients and no microangiopathy had increased connectivity in the visual and sensorimotor networks compared to healthy control (HC) subjects (15). In contrast, among adult T1DM patients with microangiopathy, functional connectivity was reported to be decreased (15). While these studies have laid the groundwork for subsequent investigations of connectivity changes in T1DM, a limitation of many of them was that participants had a wide range of plasma glucose levels ranging from 70 to 300 mg/dL (14) and 72 to 270 mg/dL (15). To our knowledge, only 1 prior study has investigated the impact of hypoglycemia on resting state functional connectivity among individuals with T1DM. In this study by Bolo and colleagues (16), T1DM participants had increased functional connectivity in the prefrontal cortex network compared to control subjects. However, no studies have investigated whether individuals with T1DM and hypoglycemia unawareness have alterations in resting state functional connectivity. Thus, in this study, we utilized a 2-step euglycemia-mild hypoglycemia insulin clamp to assess resting state functional connectivity among HC subjects as well as T1DM individuals with and without hypoglycemia unawareness.',\n",
       "  'Methods': 'Research Design and Methods Participants The participants in this study were part of a larger cohort of participants in a previously published study to investigate the impact of hypoglycemia unawareness on brain responses to visual food tasks (9). Thirty-nine participants from that study also completed rsfMRI scanning and were included in this analysis. Participants were categorized into the following three groups: 12 HC participants, 15 participants with T1DM and hypoglycemia awareness (T1DM-Aware), and 12 participants with T1DM and hypoglycemia unawareness (T1DM-Unaware). The Clarke score (17) was used to differentiate participants with hypoglycemia awareness versus unawareness. If the Clarke score was not classifiable (ie, when individuals reported a score of 3 R) then the Gold (18) method was used to determine whether they had impaired hypoglycemia awareness. All participants underwent a screening history, physical examination, and laboratory testing at the Yale New Haven Hospital Research Unit (HRU). Exclusion criteria included inability to enter the magnetic resonance imaging (MRI); smoking, illegal drug, or recent steroid use; known psychiatric or neurological disorders; active infection; malignancy; abnormal thyroid function; cerebrovascular disease; cardiovascular disease; hepatobiliary disease; or weight change in the last 3 months. The protocol was approved by the Yale University School of Medicine Human Investigation Committee. All subjects provided informed, written consent before participation. Study protocol All participants with T1DM were asked to wear a continuous glucose monitor (Dexcom G4) 1 week prior to their scheduled MRI visit to monitor for antecedent hypoglycemia. If participants had any episodes of hypoglycemia (glucose < 70 mg/dl or a symptomatic episode requiring assistance) in the 5 days prior to MRI scanning, then the scans were postponed to a later date. On the day of the MRI, participants arrived to the HRU at 9:00 am. All participants were instructed to eat breakfast as usual prior to arrival and those with diabetes were further instructed to bolus insulin as usual for breakfast. At 10 am, all participants were provided with a standardized snack consisting of 41 grams of carbohydrate (turkey sandwich, apple, diet ginger ale) to neutralize feeling of hunger as previously described (19). Participants with diabetes were instructed to inject a bolus of insulin as per their home insulin to carbohydrate ratio. Intravenous catheters were placed in antecubital veins bilaterally: one for blood sampling and the other for insulin and glucose infusion. Scanning began in the MRI center at 12:00 pm simultaneously with initiation of an insulin infusion at 2 milliunits/kg/hr. Euglycemia (~90 mg/dL) that was maintained for the first phase of the study, after which glucose was decreased into the mild hypoglycemia range (~60 mg/dL) (Fig. 1A). Glycemic state was not counterbalanced out of concern that antecedent hypoglycemia may alter cerebral energetics thereby effecting resting state connectivity patterns during euglycemia. rsfMRI data were acquired during euglycemia (between time 30–45 minutes) and hypoglycemia (between time 75–90 minutes) sessions. Throughout the MRI scan, blood was periodically sampled for glucose and counterregulatory hormones (epinephrine, norepinephrine, glucagon and cortisol). Hypoglycemia symptom assessments. Participants were asked to verbally rate their sensation of hypoglycemic symptoms (unable to concentrate, blurry vision, anxiety, confusion, difficulty speaking, double vision, drowsiness, tiredness, hunger, weakness, sweating, trembling, warmness, heart racing) on a 7-point Likert scale (1 indicating “not at all” and 7 indicating “a lot”) based upon the Edinburgh hypoglycemia symptom score (20) at 3 separate time points during the study: baseline (prior to entering the scanner) and then at euglycemia (time 30 minutes) and hypoglycemia (at time 75 minutes). Biochemical analysis. Plasma glucose was measured enzymatically using glucose oxidase (YSI, Yellow Springs, OH, US). Plasma free insulin and glucagon were measured by double antibody radioimmunoassay (Millipore, Burlington, MA). Double-antibody radioimmunoassay was used to measure plasma cortisol (MP Biomedicals, Solon, OH, US). Plasma epinephrine and norepinephrine were measured by high-performance liquid chromatography (ESA, Cincinnati, OH). Statistical analysis. One-way analysis of variance was used to determine whether there were statistical differences between the three groups for all normally distributed hormones and variables followed by Fisher’s least significant difference test for pairwise comparisons. Analyses of every repeatedly measured variable, including plasma glucose and hormone levels, were performed using the mixed-effects regression model method, taking into account both between-subject and within-subject correlations of repeated measures using a combination of prespecified compound symmetry covariance matrix and an autoregressive covariance matrix. Age, gender, and body mass index (BMI) were adjusted as covariates (ie, as fixed effects). All analyses were performed using SAS, version 9.4 (SAS, Cary, NC) and SPSS, version 24 (SPSS, Armonk, NY). Unless otherwise stated, data are presented as mean ± standard error. rsfMRI analysis Image acquisition. Participants were scanned in a Siemens 3-T Tim Trio scanner. After structural images were acquired, acquisition of functional data began in the same slice locations as the axial-oblique T1-weighted data. Resting state functional images were collected using an echo-planar image gradient echo pulse sequence (repetition time = 956 ms, echo time = 30 ms, field of vision = 210 mm, matrix size = 84×84, slice thickness = 2.5 mm, flip angle = 62°, bandwidth = 2289 Hz/pixel with 51 slices) with 274 volumes per run. Preprocessing. The digital data (Digital Imaging and Communication in Medicine ) were converted to NlfTI using dcm2nii (21) and then the first 3 images were discarded from each functional run to enable the signal to achieve steady-state equilibrium between radio-frequency pulsing and relaxation leaving 284 images per run for analysis. The data were motion-corrected using SPM8 (www.fil.ion.ucl.ac.uk/spm/software/spm8), and they were discarded if linear motion was greater than 1.5 mm or rotation was greater than 2 degrees. To account for motion confounds, images were iteratively smoothed until the smoothness for any image had a full width half maximum of approximately 6 mm (22, 23). All further analysis was performed using BioImage Suite (24) unless otherwise specified. Several covariates of no interest were regressed from the data including linear and quadratic drift, a 24 parameter model of motion, mean cerebral spinal fluid (CSF) signal, mean white matter signal, and overall global signal. The white matter and CSF areas were defined on the MNI brain, eroded to ensure only white matter or CSF signal would be included and warped to individual-participant space using a series of transformations described in the following text. Finally, the data were temporally smoothed with a 0 mean unit variance Gaussian filter (approximate cutoff frequency = 0.11 Hz). Intrinsic connectivity analyses. After preprocessing, the three resting state runs were concatenated and the functional connectivity of each voxel, as measured by the intrinsic connectivity distribution (ICD), was then calculated in each subject’s individual space (25). First, a gray matter mask was applied to the data so only voxels in the gray matter were considered in the calculation of connectivity. The gray matter mask was defined on the MNI brain and warped to the individual subject’s space using a series of inverse transformations (described in the following text). ICD involves correlating the time course for any given gray-matter voxel with the time course of every other gray-matter voxel in the brain and then summarizing these correlations with a network theory metric. Specifically, ICD models the entire distribution of the network measure of degree, therefore eliminating the need to specify a connection threshold. Because the removal of the global mean makes the signs of the correlations ambiguous, we only focus on the positive correlations (26). A histogram of these positive correlations was constructed to estimate the distribution of connections to the voxel in question. This distribution of connections was converted to a survival function and the survival function was fitted with a stretched exponential with unknown variance, α. As α controls the spread of the distribution of connections, a larger α indicates a greater number of high correlation connections. Finally, this process is repeated for all voxels in the gray matter resulting in a parametric image of the alpha parameter for each subject. Seed connectivity analyses. A reference region in the right posterior cingulate cortex (PCC) was defined on the MNI reference brain as a 9-mm cube surrounding the MNI coordinates of (−6,−43, 38) and transformed back (via the inverse transform described in the following text) into individual subject space. The time-course of the reference region in a given subject was then computed as the average time-course across all pixels in the reference region. This time-course was correlated (using Pearson correlation) with the time-courses of all other voxels in the brain to create a map of r values. These r values were transformed to z-values using Fisher’s transform and averaged across scans in each subject to yield one map for each subject representing the strength of correlation to the PCC in terms of Gaussian variables. Motion analysis. As group differences in motion have been shown to confound connectivity studies, we calculated the average frame-to-frame displacement for each subject’s data. In line with current reports, no subject had an average frame-to-frame displacement greater than 0.20 mm. There were no significant differences for motion between the T1DM-Unaware and HC groups (P = 0.06), between T1DM-Unaware and T1DM-Aware groups (P = 0.06), and T1DM-Aware and HC groups (P = 0.42). Image registration. Single-subject data were warped to a common MNI template through the concatenation of a series of linear and nonlinear registrations. The functional series were linearly registered to the T1 axial-oblique (2D anatomical) image. The 2D anatomical image was linearly registered to the magnetization-prepared rapid acquisition with gradient echo (MPRAGE; 3D anatomical) image. The 3D anatomical image was nonlinearly registered to the template MNI brain using a previously validated algorithm (27). All transformation pairs were calculated independently and combined into a single transform warping the single participant results into common space. It was then inverted to create the reverse transformation from the template MNI brain to the space of the participant. This single transformation allows the single participant images to be transformed with only one transformation, reducing interpolation error. Group analysis. Voxel-wise t tests were used to compare the connectivity data between the study groups. Imaging results are shown at a cluster-level threshold of P < 0.05 familywise error (FWE) correction as determined by AFNI’s 3dClustSim program (version 16.0.09) using a cluster-forming threshold of P = 0.001, 10 000 iterations, a gray matter mask, and a smoothness estimated from the residuals using 3dFWHMx with the “–ACF” option. This stringent cluster-forming threshold was used to avoid the problems of false positives associated with less stringent cluster-forming thresholds and is based on current best practices (28, 29).',\n",
       "  'Results': 'Results Participant characteristics Twelve HC subjects and 27 individuals with T1DM participated in this study. Using the Clarke score (17), individuals with T1DM were further categorized as those with hypoglycemia awareness (T1DM Aware, N = 15) and those with hypoglycemia unawareness (T1DM Unaware, N = 12). Of note, the participants for this study include 39 of 41 subjects from a larger previously published study investigating the impact of hypoglycemia unawareness on brain responses to visual food cues (9). Demographic characteristics are presented in Table 1. Compared to HC subjects, T1DM Aware individuals were similar in gender, age, BMI, and education level. Compared to T1DM-Aware individuals, T1DM-Unaware individuals were older (P = 0.008), had a higher BMI (P = 0.007), and longer duration of T1DM (P = 0.002). There were no differences in hemoglobin A1c (HbA1c) percentage levels between T1DM-Aware and T1DM-Unaware groups (P = 0.26). As expected, T1DM-Unaware individuals had a significantly higher self-reported number of severe hypoglycemic episodes in the preceding year (P = 0.03). Two-step hyperinsulinemic euglycemic-hypoglycemic clamp All participants underwent a 2-stepped hyperinsulinemic euglycemic-hypoglycemic clamp (Fig. 1). A mixed-model linear regression finds a difference in the plasma glucose levels during the clamp across the 3 groups (P = 0.01), which was driven by modestly higher glucose levels during euglycemia amongst the T1DM-Aware group. Averaged plasma glucose levels at the time of rsfMRI acquisitions at euglycemia (between time 30–45 minutes) were 102.4 ± 5.6 mg/dL for T1DM-Aware individuals, 89.2 ± 1.6 mg/dL for T1DM-Unaware individuals, and 88.6 ± 1.9 mg/dL for HC subjects (F[2,36] = 4.1, P = 0.03). During rsfMRI acquisitions at hypoglycemia (time 75–90 minutes), mean plasma glucose levels were nearly identical across the 3 groups (59.0 ± 1.3 mg/dL for T1DM-Aware individuals, 61.0 ± 2.1 mg/dL T1DM-Unaware individuals, and 61.8 ± 1.8 mg/dL for HC subjects (F[2,36] = 0.78, P = 0.47). There were no differences in plasma insulin levels across the 3 groups (mixed model linear regression, P = 0.50). Responses to hypoglycemia Counterregulatory hormonal responses to hypoglycemia are presented in Fig. 2. As expected, T1DM participants had no hypoglycemia-induced increase in glucagon levels. Furthermore, while HC and T1DM-Aware participants had a significant increase in epinephrine levels in response to mild hypoglycemia, T1DM-Unaware participants had no change in epinephrine levels. At both euglycemia and hypoglycemia (times 30 and 75 minutes, respectively), participants were asked to rate their symptoms of hypoglycemia using the Edinburgh hypoglycemia symptom score (20). At euglycemia, mean symptom scores were similar across all 3 groups (HC: 20.8 ± 1.8; T1DM-Aware: 27.7 ± 2.2; T1DM-Unaware: 23.5 ± 2.0, F[2,36] = 2.9, P = 0.07) although T1DM-Aware individuals tended to have higher baseline ratings. At hypoglycemia, symptoms scores were significantly different across groups (HC: 24.3 ± 2.0; T1DM-Aware: 38.9 ± 4.3; T1DM-Unaware: 27.6 ± 3.6, F[2,36] = 4.6, P = 0.02). While HC subjects and T1DM-Aware subjects had significant increases in hypoglycemia symptoms during mild hypoglycemia, T1DM-Unaware individuals had no differences in symptoms scores (Fig. 3A). Furthermore, T1DM-Aware individuals also had a more robust symptom response to mild hypoglycemia (Fig. 3B). rsfMRI responses to mild hypoglycemia Several approaches to assessing resting state functional connectivity have been developed. In this study, we chose to use 2 different approaches: ICD analysis and seed-based analysis. ICD analysis. Although prior work by our group and others have identified numerous distinct brain regions that appear to be altered with T1DM and hypoglycemia unawareness (8, 9, 15), there is a great deal of heterogeneity in which brain regions are different across studies. Thus, to avoid arbitrarily limiting our analysis using predefined regions of interest, we sought to survey potential differences in functional connectivity across the 3 groups. To do this, we measured the ICD, which is a voxel-based method of rsfMRI analysis that models the distribution of all the connections to any voxel. This method has the advantage of allowing assessments of the entire brain and not requiring defined a priori regions of interest. Furthermore, ICD analysis takes into account the entire distribution of connections for each voxel and thus does not set arbitrary thresholds to determine connectivity between 2 regions (25, 30). Across all 3 groups, there was a significant difference between euglycemia and hypoglycemia (Group × Condition Interaction) in the angular gyrus, even after adjusting for age, BMI, and plasma glucose levels (Fig. 4A, initial threshold of P < 0.001, whole brain, FWE corrected at P < 0.05). Connectivity values were extracted from the significant cluster in the angular gyrus and the differences appeared to be driven by the T1DM-Aware group who had a greater change in connectivity between euglycemia and hypoglycemia compared to the other 2 groups (Fig. 4B,C). Of note, 2 T1DM-Aware individuals had plasma glucose levels >150 mg/dL at the start of rsfMRI scanning. ICD analysis was also performed without these 2 individuals and revealed nearly identical findings in the angular gyrus, and thus these individuals were included in the analysis (31). In addition, to assess whether duration of diabetes was driving our findings, we also performed a subanalysis among only the individuals with T1DM with and without covarying for duration of diabetes and found no differences (31). Seed-based connectivity analysis (PCC as a seed). Because the angular gyrus has been shown in numerous studies to be consistently deactivated at rest as part of the DMN (31–34), we next sought to investigate changes in the DMN across groups. The DMN is an important network of discrete brain regions that consistently decrease activity during attention-demanding tasks (35, 36). One standard method to assess the DMN is by using the PCC as a seed region because it directly interacts with the other nodes in the DMN and is not lateralized (36, 37). We again identified a significant Group × Condition interaction in the right angular gyrus (Fig. 5). To further investigate which groups were driving this significant overall interaction, we then examined each group’s resting state functional connectivity patterns separately. The t3 groups had very different responses to mild hypoglycemia (Fig. 6 and digital research materials repository (31)). Specifically, while mild hypoglycemia did not elicit changes in the DMN in HC subjects, T1DM-Aware individuals had a greater change in resting state connectivity in the superior frontal gyrus, medial frontal gyrus, and anterior cingulate cortex, as well as angular gyrus. In contrast, T1DM-Unaware individuals also showed no changes in DMN in response to mild hypoglycemia. Moreover, the change in connectivity value in the angular gyrus correlated positively and significantly with the Edinburgh hypoglycemia symptoms scores at both euglycemia and hypoglycemia (Fig. 7). Next, we assessed whether the observed changes in resting state connectivity could be driven by changes in cerebral perfusion between groups. Across all groups, mild hypoglycemia did elicit changes in cerebral perfusion, as expected; however, there were no significant differences between groups using P < 0.001 or P < 0.05 (31). Relationships between functional connectivity and psychological outcomes Finally, in an exploratory analysis to begin investigating the relationship between resting state functional connectivity and psychological outcomes in T1DM, we administered a battery of psychologic assessments for anxiety and depression. T1DM-Aware individuals were noted to have higher scores on the Perceived Stress Scale (PSS) (38) compared to T1DM-Unaware individuals (P = 0.011) (31). Furthermore, PSS scores correlated positively and significantly with angular gyrus connectivity values (from the ICD analysis) at euglycemia (r = 0.531, P = 0.016) but not at hypoglycemia (r = 0.35, P = 0.129).',\n",
       "  'Discussion': 'Discussion Using 2 different methods of functional connectivity analysis, we identified distinct differences in the resting state brain responses to mild hypoglycemia between HC, T1DM-Aware, and T1DM-Unaware participants. In particular, both ICD and seed-based analysis show changes in angular gyrus connectivity that were consistently different across the groups. Furthermore, changes in angular gyrus connectivity also correlated with greater symptoms of hypoglycemia as well as higher scores of perceived stress. These findings provide evidence that T1DM is associated with changes in the brain’s resting state connectivity patterns and that this may also be associated with awareness of hypoglycemia and with alterations in functional outcomes amongst individuals with T1DM. Anatomically, the angular gyrus sits at the junction of the occipital, temporal, and parietal lobes and is believed to play an important role in the integration of inputs from these brain regions (39). It is involved in theory of mind (34), processing of concrete and abstract concepts (40), and memory (41), as well as attention (42). The angular gyrus is also part of the DMN, a network of discrete brain regions that consistently decrease activity during attention-demanding tasks. In the absence of tasks and when the brain is at wakeful rest, activity in the DMN is believed to reflect the underlying organizational structure of the brain’s intrinsic activity (35). Using the currently recommended thresholds for statistical significance (P < 0.001), we observed that mild hypoglycemia did not elicit changes in functional connectivity HC subjects when using either the ICD analysis approach or a seed-based approach. However, in contrast, T1DM-Aware individuals had greater deactivation of the angular gyrus (Fig. 4B,C) as well as other components of the DMN in response to mild hypoglycemia compared to HC subjects (Fig. 6). Moreover, this difference appears to be driven by relatively higher connectivity values among T1DM-Aware subjects at euglycemia. The T1DM-Unaware cohort did not display this increased connectivity at euglycemia. While the reason for this difference remains unknown, one possible explanation is that the increased connectivity at euglycemia in the T1DM-Aware population reflects a functional reorganization of the brain or an early compensatory hyperconnectivity (15). This is consistent with a magnetoencephalography study that found that patients with T1DM and no microvascular complications had higher DMN connectivity compared to HC subjects (43) as well as a recent fMRI study of children with T1DM, which found compensatory hyperconnectivity at baseline among children with T1DM (14). In addition, van Duinkerken et al (44) showed that patients with T1D without microangiopathy have increased functional connectivity in regions involving in motor and visual processing compared to patients with T1D with microangiopathy. The latter cohort had an earlier age of onset of T1D and a longer duration of diabetes. While our study was not designed to assess markers of cerebral microvascular disease, taken together, these results and ours suggest that perhaps the initial compensatory hyperconnectivity changes are eventually lost with disease progression. This type of pattern has not only been observed in T1D, but in multiple sclerosis patients as well, suggesting that it may be a typical biphasic brain response to certain chronic disease states (45). Future studies will be needed to clarify what factors drive the alteration in functional connectivity over time. Because deactivation of the DMN occurs normally when the brain is engaged in attention-demanding tasks (35, 36), another possible interpretation of our findings could be that T1DM-Aware individuals respond to mild hypoglycemia with greater attention or a heightened awareness of hypoglycemia. That is, T1DM-Aware individuals are more responsive to small changes in plasma glucose levels and that the physiological stress associated with mild hypoglycemia in this group mirrors the effects of a task-oriented cognitive load, which deactivates the DMN and brings the individual out of the wakeful rest, mind-wandering state. This interpretation is supported by our findings that T1DM-Aware individuals have a more robust symptom response to mild hypoglycemia even though they have similar degree of change in counterregulatory hormone responses such as epinephrine compared to HC subjects. Unlike the T1DM-Aware individuals, T1DM-Unaware participants showed no changes in DMN connectivity in response to mild hypoglycemia. Of note, T1DM-Unaware individuals showed no changes in response to mild hypoglycemia, in a similar manner to HC subjects. Whether this is due to adaptive mechanisms to augment glucose and fuel exposure in the brain during hypoglycemia remains to be determined. Frequent exposure to hypoglycemia has been associated with increased cerebral glucose transport in some (46, 47), but not all studies (48). While the pattern of brain connectivity changes appeared similar between T1DM-Unaware and HC participants, they had very different counterregulatory and symptom responses suggesting that there may be T1DM-specific mechanisms driving the differences in brain and peripheral responses to hypoglycemia. To our knowledge, there is only 1 other study that used a hyperinsulinemic euglycemic-hypoglycemic clamp and rsfMRI scanning to investigate the impact of declining glucose levels on functional connectivity. In that study, Bolo and colleagues (16) studied T1DM individuals (but did not distinguish between hypoglycemia aware or unaware) and HC subjects as plasma glucose levels fell from ~90 mg/dL to ~50 mg/dL. They observed that T1DM individuals had an attenuated connectivity response during induction of hypoglycemia compared to HC subjects. Because changes in brain activity are dynamic, it is possible that the change in connectivity we observed among our T1DM-Aware individuals could reflect differences that occur later in time as our participants were scanned after they had been at stable levels of mild hypoglycemia. Moreover, differences in hypoglycemia targets (50 mg/dL vs. 60 mg/dL) or differences in fMRI analysis (ie, different choice of seed regions) may also have contributed to some of the variability in responses. Finally, we observed significant relationships between changes in brain connectivity values (as well as at euglycemia and hypoglycemia) in the angular gyrus and symptom responses to hypoglycemia. Moreover, measures of stress and anxiety, which have been associated with poor glycemic control and less frequent blood glucose monitoring (49), were also associated with angular gyrus connectivity values at euglycemia (but not hypoglycemia). These exploratory findings would be consistent with reports that individuals with hypoglycemia unawareness have low fear of hypoglycemia (50) and individuals with preserved hypoglycemia awareness have high fear of hypoglycemia (51), suggesting that these individuals not only fail to sense hypoglycemia, but also have developed inappropriate emotional valence associated with potential hypoglycemia. Taken together, our findings raise the possibility that functional connectivity in the DMN and specifically the angular gyrus may play a role in modulating these behavioral outcomes and contribute to the inability to take corrective action to prevent or treat hypoglycemia. There are, however, some limitations to the current study. First, our study is limited by the small sample size; however, current recommendations for sample size (52, 53) for fMRI studies were not devised for mechanistic human studies where participants are studies under carefully controlled conditions and compared to themselves at 2 clearly defined states. Another limitation to our study is the modest differences (~10 mg/dL) in plasma glucose levels during the euglycemia phase of the study across the 3 groups. Nevertheless, it is unlikely that this small difference in absolute (or rate of change) of plasma glucose levels would have had a significant impact on our findings given that we observed no differences in results when we performed the analysis while covarying for plasma glucose levels as well as with and without individuals who had plasma glucose levels >150 mg/dL. Furthermore, our findings are consistent with other rsfMRI studies that had a wide range of plasma glucose levels (up to 300 mg/dL) (14, 15). Moreover, although our T1DM participants were matched for HbA1c, there may have been other differences between our groups, which could have played a role in modulating our brain findings. In particular, our T1DM-Unaware participants were slightly older and had longer duration of diabetes compared to the T1DM-Aware participants. Thus, further studies may be needed to clarify the complex interplay between duration of diabetes, glycemic control, and age in the development of hypoglycemia unawareness. In our current study design, glycemic state was not counterbalanced across the 3 groups due to concern that antecedent hypoglycemia may alter cerebral energetics and effect resting state connectivity during euglycemia. Counterbalancing could have been achieved if we conducted a euglycemic and hypoglycemic clamp on 2separate study days for each participant. However, this type of study design was not logistically feasible. Finally, although our participants had well-controlled T1DM, were relatively young (mean age 30–40s), and had no reported history of cerebrovascular disease, occult microvascular disease could also have impacted our findings. However, because individual participants were compared to themselves at euglycemia and hypoglycemia, this would have minimized the impact of within-person variability and would decrease any between-group effects of occult cerebrovascular disease.',\n",
       "  'Conclusion': 'Conclusion In conclusion, our data demonstrate that T1DM-Aware individuals have a heightened responsiveness to mild hypoglycemia leading to deactivation of the DMN including the angular gyrus compared to HC subjects. This response is lost among T1DM individuals who have developed hypoglycemia unawareness. These findings may have implications for understanding the neurocognitive changes that occur during insulin treatment among patients with T1DM.',\n",
       "  'URL': 'https://doi.org/10.1210/clinem/dgz004',\n",
       "  'Journal': 'J Clin Endocrinol Metab'},\n",
       " {'PaperTitle': 'Oestrogen and progesterone action on endometrium: a translational approach to understanding endometrial receptivity',\n",
       "  'Publication': '2013',\n",
       "  'Authors': 'Steven L Young',\n",
       "  'Email': '',\n",
       "  'Abstract': 'Embryo attachment and implantation is critical to successful reproduction of all eutherian mammals, including humans; a better understanding of these processes could lead to improved infertility treatments and novel contraceptive methods. Experience with assisted reproduction, especially oocyte donation cycles, has established that despite the diverse set of hormones produced by the ovary in a cycle-dependent fashion, the sequential actions of only two of them, oestrogen and progesterone, are sufficient to prepare a highly receptive endometrium in humans. Further investigation on the endometrial actions of these two hormones is currently providing significant insight into the implantation process in women, strongly suggesting that an abnormal response to progesterone underlies infertility in some patients.',\n",
       "  'Introduction': 'A thorough understanding of the processes governing human embryo implantation would be of significant benefit for the treatment of infertility and the development of novel contraceptives. However, implantation processes remain poorly understood, largely due to differences between humans and experimental animals and appropriate ethical, moral and legal barriers to direct examination of implanting human embryos. Despite these barriers, significant knowledge has been gained through experience with assisted reproduction coupled with application of improving analytic techniques applied to human tissues and non-human primate models.\\nExperience with donor oocyte IVF cycles has allowed profound clinical insights into the regulation of human endometrial receptivity. Donor oocyte cycles achieve the highest implantation rates of all assisted reproduction approaches (Sunderam et al. 2009), suggesting that the hormonal preparation of the endometrium has been well optimized (van der Linden et al. 2011). In donor oocyte cycles, the endometrium of the recipient is prepared by sequential treatment with oestrogen and progesterone, using protocols that prevent ovulation and corpus luteum formation. Notably, these protocols work just as well in a woman without ovaries. Thus, these two hormones, without any other ovarian or corpus luteum products, are sufficient for excellent preparation of human endometrium to accept an implanting embryo. Their primacy is further supported by the requirement of both hormones for pregnancy initiation and early survival in all eutherian mammals, despite major species-specific differences in ovarian and uterine anatomy and physiology. Given the critical and fundamental role that oestrogen and progesterone play in establishment of receptivity, a deep understanding of the action of these steroid hormones on the human endometrium will allow clear insight into the mechanisms determining endometrial receptivity. This review will attempt to summarize the current, albeit limited, understanding of oestrogen and progesterone action in determination of endometrial receptivity.',\n",
       "  'Methods': 'To avoid the aforementioned barriers to understanding progesterone sufficiency in endometrial function, this study group has utilized a modelled cycle, in which progesterone concentrations are experimentally determined (Figure 3). The controlled cycles are highly similar to endometrial preparation for an oocyte donor IVF cycle, and thus should result in a highly receptive endometrium, if physiological progesterone is provided. The protocol begins with lupron downregulation, followed by transdermal oestrogen replacement at physiological concentrations, followed by oestrogen plus daily i.m. progesterone at physiological and subphysiological concentrations, and subsequent biopsy on day 10 of progesterone treatment.',\n",
       "  'Results': 'Both oestrogen and progesterone act through specific, high-affinity, low-capacity nuclear receptors that function as ligand-activated transcription factors and chromatin modifiers to directly regulate expression of a large number of genes (Cheung and Kraus 2010; Huang et al. 2010). The products of steroid receptor-regulated genes can also act in a downstream, autocrine, paracrine or endocrine fashion to regulate expression of additional genes. It is important to recognize that some non-steroidal ligands can also bind the steroid receptors. Examples of non-steroidal ligands which act through oestrogen receptors include endogenous lipoxin A4 (LXA4), an eicosanoid produced in the endometrium (Russell et al. 2011), bisphenol A, an environmental compound (Li et al. 2012), and clomiphene citrate, a pharmaceutical agent. Thus, nuclear steroid receptors are responsible for the so-called ‘classical’ actions of oestrogen and progesterone (Figure 1).\\nIt is important to point out some significant simplifications made to improve readability in Figure 1. For example, oestrogen receptors and progesterone receptors are bound to chaperone proteins and are released from them after ligand binding. Chaperone binding may regulate steroid receptor availability and access to the nucleus, and therefore function. Another key feature of the classical actions of oestrogen and progesterone, not included in Figure 1, is that there are multiple oestrogen receptor and progesterone receptor isoforms, each having distinct actions on the genome. Differential expression of these isoforms in different cell types and physiological states results in differential effects of the steroids.\\nThere are two nuclear oestrogen receptors – oestrogen receptor α and oestrogen receptor β – each derived from a distinct gene (ESR1 and ESR2, respectively). These genes have high sequence homology, likely resulting from an ancient gene duplication event, since homologous genes are seen in fish and amphibians as well as mammals (Katsu et al. 2008). Although similar in structure, oestrogen receptors and have distinct effects in experimental model organisms and distinct patterns of expression in human disease (Hewitt and Korach 2003). For example, overexpression of oestrogen receptor is observed in endometrioma lesions due to hypomethylation of the promoter leading to a molecular cascade resulting in inflammation and other pathophysiological changes (Bulun et al. 2010).\\nThe progesterone receptors have at least two isoforms – progesterone receptor A and progesterone receptor B. Unlike oestrogen receptors, the progesterone receptor isoforms are derived from alternate transcription and translation start sites in a single gene (PGR; Ogle 2002; Jacobsen and Horwitz 2012). Progesterone receptor A and B are identical in structure except that the progesterone receptor B isoform contains a 164-amino acid N-terminal sequence, which is lacking in the progesterone receptor A isoform. The presence or absence of the N-terminal extension appears to be responsible for the distinct differences in progesterone receptors A and B actions. Truncated isoforms – progesterone receptor C and progesterone receptor M – that retain the progesterone-binding domain but lose the DNA-binding domain have been described as a possible suppressor of progesterone receptors A and B action, but their relevance in vivo is controversial (Wei et al. 1990; Samalecos and Gellersen 2008; Taylor et al. 2009).\\nA further level of complexity is seen in the interaction between steroid receptors and co-activators and co-repressors. These co-activators and repressors mediate the effects of the nuclear receptors on gene transcription (Figure 1). The expression and activity of the co-activators and co-repressors can be determined both developmentally and dynamically in the adult, providing a further basis for the pleiotropic effects of steroid hormones. In this regard, it is important to note that there are distinct mechanistic differences between mammalian species in steroid hormone and co-activator expression. For example, oestrogen receptor appears to be significantly more expressed in human endometrium as opposed to the mouse. A more extreme example is the progesterone receptor B specific co-activator, MAGEA-11, which is only present in primates and appears to play an important role in the human endometrial response to progesterone (Su et al. 2012).\\nThe effects of progesterone via its receptor also depend on other signals and transcription factors. An indisputably critical action of progesterone on endometrial stroma is decidualization. However, full decidualization requires signalling by both progesterone receptor and cAMP (Kajihara et al. 2013). Interestingly, cAMP induces expression of many transcription factors, including FOXO1, C/EBPb (CCAAT/enhancer-binding protein b), STAT5 (signal transducers and activators of transcription 5) and HOXA11, all of which directly interact with and modulate progesterone receptor (Kajihara et al. 2013). These factors, including progesterone receptor, form multimeric complexes at promoters for genes critical to a decidualized phenotype. Without this synergistic interaction between other cellular signals and transcription factors, progesterone would not exert this important effect on endometrial stroma. Emerging data suggesting that progesterone-driven decidualization may act as a biosensor of embryo quality during early implantation is reviewed by Lucas in this issue (Lucas, 2013).\\nAnother simplification in Figure 1 is that steroid receptors dynamically interact with chromatin in a manner regulated by chromatin remodelling, chaperones, the proteasome and binding of other transcription factors (Grontved and Hager 2012). Oestrogen receptor and progesterone receptor isoforms can only bind DNA if the chromatin structure is open enough to allow access. The areas of open and closed chromatin in a particular cell type in a particular physiological environment are yet another mechanism for tissue-specific actions of oestrogen and progesterone.\\nIn this context, it is important to note that epigenetic mechanisms and microRNA expression may be important modifiers of progesterone action. Initial studies in humans have shown epigenetic changes with cycle phase, including alterations in DNA methyltransferase and histone-modifying enzyme expression (Guo 2012). Initial studies have also shown significant cycle-regulated changes in microRNA through the cycle (Sha et al. 2011; Altmae et al. 2013). The role of microRNA in both normal endometrium and in endometriosis are discussed in the review by Hull and Nisenblat (2013, in this issue).\\nIn addition to their direct, genomic effects, both oestrogen and progesterone also exert rapid, ‘non-classical’ effects on the cell via action at the plasma membrane, via nuclear receptors interacting with other transcription factors or via less well-understood effects on mRNA stability (Figure 2). Oestrogen can act through both membrane-associated oestrogen receptor and a structurally unrelated, integral membrane, G-protein coupled oestrogen receptor, GPR30, to stimulate one or more cytoplasmic signalling cascades in response to oestrogen. The effects of signalling via GPR30 in the endometrium are unclear, but there is a profound cyclic regulation of this receptor (Plante et al. 2012).\\nThe non-classical actions of progesterone are less-well understood, but no less complex. As mentioned above, alternative transcription start sites in PGR may result in production of progesterone receptor M or progesterone receptor C, although conflicting evidence exists regarding their relevance in vivo. A separate family of membrane progesterone receptors, mPRα (PAQR VII), mPRβ (PAQR VIII) and mPRγ (PAQR V) that are structurally unrelated to the PGR gene, can also bind progesterone and are thought to activate G-protein coupled signalling pathways (Zhu et al. 2003; Dressing et al. 2011). Significant controversy exists regarding the structure and function of this molecular family. For example, the predicted structure of PAQR family members shows eight transmembrane domains rather than the seven seen in the G-protein coupled receptor family and there is no significant sequence similarity to known G-protein coupled receptors (Moussatche et al. 2012). Furthermore, the PAQR family shows sequence motifs more closely related to alkaline ceramidases and may have similar enzymic activity (Moussatche et al. 2012). Thus, the function of the PAQR family receptors remains to be firmly established and although expression of the mRP family has been shown in the human endometrium, their role in endometrial function remains unclear (Fernandes et al. 2005). Finally, a newly described membrane channel/receptor on human spermatozoa, CatsPer, is capable of binding progesterone (and other compounds released by the cumulus–oocyte complex) and causing calcium influx (Brenker et al. 2012; Lishko et al. 2011). However, CatsPer expression appears to be sperm specific and is, therefore, unlikely to play a role in the endometrium.\\nEndometrial receptivity to embryo implantation exists for a brief period of time and this timing is driven by time of progesterone exposure, only after sufficient exposure to oestrogen. Given this temporally specific process, it is not surprising that expression and localization of steroid receptors and their co-regulators vary markedly in different menstrual cycle phases (Table 1). In all eutherian mammals studied, oestrogen receptor disappears from the endometrial epithelium at the time of embryo implantation (Donaghay and Lessey 2007). In the human endometrial epithelium, both oestrogen receptor and progesterone receptor immunohistochemical staining diminish markedly during the midsecretory implantation window (Lessey et al. 1988; Young and Lessey 2010). Further analysis of the mid- and late proliferative phases shows that progesterone receptors A and B are easily detected in both epithelial and stromal compartments of the human endometrium (Mote et al. 2000; Wang et al. 1998). In the secretory-phase epithelium, progesterone receptor A expression is virtually absent during the mid- and late secretory phases, while progesterone receptor B expression is maintained at low concentrations through the mid-secretory phase and falls to even lower concentrations by the late secretory phase. In the stroma, progesterone receptor A expression is significantly higher than progesterone receptor B throughout the cycle, although present in low abundance in the late secretory phase. Given the absence or paucity of oestrogen receptor and progesterone receptors A and B in the mid- to late secretory endometrial epithelium, it is likely that epithelial effects of oestrogen and progesterone during these cycle phases results from oestrogen- or progesterone-induced paracrine factors, produced in the stroma and acting on the epithelium, termed oestromedins and progestomedins. Potential human endometrial oestromedins and progestomedins include insulin-like growth factor 1 (Giudice et al. 1993), heparin-binding epidermal growth factor (Leach et al. 1999; Young et al. 2002) and fibroblast growth factor 7 (Koji et al. 1994).\\nWhile molecular studies of oestrogen and progesterone receptors provide the mechanistic framework for understanding endometrial function, it is the physiological and clinical studies that provide the most practical insight into implantation mechanisms. Oestrogen is essential for endometrial proliferation, as repeatedly demonstrated in humans and experimental animals lacking ovaries and those in whom oestrogen production or action has been prevented.\\nThe role for oestrogen in the secretory phase and in implantation is less clear. In mice, oestrogen appears to be critical to support implantation and early pregnancy (Dey et al. 2004). Interestingly, the decidualized mouse endometrium appears to produce its own oestradiol and does not require corpus luteum-derived oestrogens (Das et al. 2009). As far as is known, there is no substantive data to support this pathway in human decidua.\\nThere are, of course, many differences between human 28-day menstrual cycle and the mouse 4-day oestrus cycle, including circulating oestradiol concentrations. Mouse peak serum oestradiol concentrations in pro-oestrus are equal to or lower than typical perimenstrual nadir concentrations in the human and 10–20 times lower than peak preovulatory concentrations. However, oestrogen action in the human midsecretory phase could possibly occur through other, non-steroidal oestrogen receptor agonists. An eicosanoid, LXA4, was recently shown to bind oestrogen receptor and act as an agonist, and the biosynthetic pathway for LXA4 appears to be present in the human endometrium (Russell et al. 2011). Further work is needed, however, to determine any role that LXA4 might play in the human endometrium.\\nStudies in women without functional ovaries demonstrate that luteal oestrogen is not necessary for normal day-25 morphology or normal changes in oestrogen receptor and progesterone receptor immunolocalization (de Ziegler et al. 1992). Surprisingly no vaginal spotting was noted in the subjects during the 10 days of progesterone treatment without any oestrogen given. In another study employing oestrogen receptor antagonism with clomiphene begun 2 days after LH surge in a spontaneous cycle and continued until biopsy on day 13 resulted in consistently delayed histological maturation (Fritz et al. 1987). The clomiphene antagonism study findings are echoed by experiments in the bonnet macaque; in these studies, peri-implantation administration of aromatase inhibitor (fadrozole) or oestrogen antagonist (tamoxifen) markedly decreased, but did not eliminate, conception. In another primate study, this time in oophorectomized rhesus macaques, provision of progesterone alone was able to support endometrial receptivity, early post-implantation embryo development and normal pregnancy (Ghosh et al. 1994).\\nIn order to better understand these apparently conflicting data, this study group analysed gonadotrophin-releasing hormone downregulated cycles followed by oestrogen (at varying doses) and progesterone replacement (Groll et al. 2009). Effects on endometrial histology and immunohistochemical staining for integrin subunit β, osteopontin, oestrogen receptor and progesterone receptors A and B were examined. These studies demonstrated no difference in between groups not receiving oestradiol and those receiving physiological or supraphysiological oestradiol.\\nIt is striking that the oestrogen receptor inhibitor studies demonstrate a necessity for luteal-phase oestrogen, while progesterone (with or without oestrogen) replacement studies show no luteal-phase requirement. A possible explanation is that in studies where exogenous progesterone is given, there is sufficient extra-ovarian conversion of progesterone to oestrogen (via testosterone) to maintain endometrial function. The oestradiol antagonism and aromatase inhibition studies might provide a more profound impact by blocking oestrogen action (even that derived in the endometrium). The data in the ovariectomized rhesus macaque, however, remains remarkable, because systemic oestradiol concentrations were measured and shown to be very low, even with administration of progesterone. Taken together, the data suggest that the (human or non-human) primate endometrium appears to function normally with very low concentrations of oestradiol.\\nClinical data are also mixed. It is well known that use of gonadotrophin-releasing hormone agonists or antagonists in non-donor IVF cycles results in a shortened luteal phase and possibly other qualitative luteal defects. Thus, luteal support with progesterone and sometimes oestrogen is given. Clinical outcomes are mixed demonstrating a benefit of luteal oestrogen supplementation in IVF (Farhi et al. 2000; Lukaszuk et al. 2005) or no benefit (Smitz et al. 1993; Lewin et al. 1994; Fatemi et al. 2007). The most recent systematic review suggests no overall benefit (Fatemi et al. 2007). Given the experimental results in women and monkeys with absent luteal function and the mixed evidence in clinical trials, any possible clinical benefit of luteal oestrogen support in IVF must accrue only to a small subset of patients.\\nProgesterone is absolutely required for successful embryo implantation and pregnancy maintenance. In fact, progesterone was discovered because of its effects on the endometrium and early pregnancy survival (Allen and Doisey 1923; Allen and Corner 1929). The effects of progesterone on the endometrium were confirmed in non-human primates (Zuckerman 1937), leading Georgeanna Seeger Jones to characterize patients with possible progesterone deficiency leading to infertility (Jones 1949; Jones 1973). The concept that progesterone insufficiency will cause infertility is logically irrefutable. Progesterone is necessary for implantation and pregnancy survival and thus, at some lower threshold, there will be insufficient progesterone for these functions. However, the methods of diagnosing progesterone insufficiency (or sufficiency) and therefore its role in patients have been controversial.\\nThere are three major contributors to the uncertainty regarding the role of luteal-phase defect in infertility. The first is that the corpus luteum releases progesterone in pulses, which are rapidly cleared from the body, resulting in marked fluctuations of progesterone serum concentrations (Filicori et al. 1984), changing as much as 6-fold within a few hours. The rapidly fluctuating concentrations preclude using individual serum progesterone measurements as a measurement of progesterone sufficiency. Secondly, there is no ‘gold standard’ marker of endometrial receptivity to embryo implantation that would allow evaluation of endometrial function outside of a conception cycle. Current progress in the identification of markers of the receptive endometrium is discussed by Salamonsen et al. (2013, in this issue). Thirdly, there are clear differences between species in the mechanisms regulating embryo implantation, but profound ethical issues prevent systematic study of human embryo and endometrial interactions in vivo.\\nUsing this model, endometria from healthy women exposed to physiological concentrations of progesterone (40 mg dose, steady-state concentration about 15–25 ng/ml) were compared with those exposed to subphysiological (10 mg dose, steady-state concentration about 4–6 ng/ml) and assessed histological dating of endometria, immunohistochemistry for endometrial integrins and quantitative real-time PCR analysis for nine putative functional markers (Usadi et al. 2008). However, despite a 4-fold difference in progesterone, none of the assessed markers of endometrial structure and function showed a significant difference between groups. Given the critical importance of progesterone action in the endometrium and the expectation of a dose-dependent response, a further reduction in dose will certainly have effects on both histology and gene expression. However, the data to date clearly demonstrate that progesterone concentrations in the low end of what is seen in ovulatory women do not cause profound changes in human endometrial structure or function. Thus, it would appear that, in normal women, a progesterone dose threshold can be defined, below which consistent alterations in gene expression and in histological maturation can be seen. Since this threshold concentration is below the lowest serum concentrations encountered clinically, the data strongly suggest the following two conclusions: (i) isolated progesterone deficiency is very unlikely to be a cause of infertility in couples; and (ii) normal secretory-phase endometrial structure and function in young healthy women can be achieved across a wide range of progesterone concentrations. It must be noted that these experiments were performed on young healthy women without any evidence of endometriosis or infertility.\\nIn all of the above studies, it must also be recognized that local effects of sex steroids can be strongly influenced by local metabolism. For example, a recent study examined oestrogen metabolizing enzyme concentrations in human endometrial tissue as well as serum and tissue oestradiol and oestrone concentrations (Huhtinen et al. 2012). These studies showed marked differences between serum and tissue oestradiol/oestrone ratios, which depended on cycle phase and correlated with the type of 17 -hydroxysteroid dehydrogenase expressed.\\nIt has been postulated that women with endometriosis-related infertility may be partially resistant to progesterone actions on the endometrium (Burney et al. 2007; Bulun, Cheng et al. 2010; Fazleabas 2010). Strikingly, the baboon model demonstrates that simply inducing peritoneal lesions can result in changes in progesterone action, consistent with progesterone resistance (Fazleabas, 2010). It is presumed that local inflammation is involved in the observed alterations in progesterone action, although the mechanism for this remains unclear. This hypothesis could explain why some women have persistently delayed histological maturation or persistently abnormal expression of progesterone-regulated genes. If progesterone resistance is truly present in some women, then, depending on the mechanism conferring resistance, such women might achieve normal secretory-phase structure and function with a higher progesterone dose or with treatments targeted at abnormal inflammation.\\nGiven the known mechanisms of progesterone action, resistance might occur through a variety of means. Abnormal expression of specific progesterone receptors is one possible mechanism and women with endometriosis often show failure of mid-secretory downregulation of epithelial progesterone receptor (Lessey, Killam et al. 1988) and evidence for specific suppression of progesterone receptor B, but not progesterone receptor A, at multiple cycle phases (Attia et al. 2000). Another possible mechanism of resistance is an alteration of expression or function of progesterone receptor chaperones and co-chaperones. Overexpression of co-chaperone FKBP51 (Hubler et al. 2003) or lack of co-chaperone FKBP52 (Tranguch et al. 2005; Tranguch et al. 2006; Tranguch et al. 2007) causes progesterone resistance in experimental models. Interestingly, high FKBP51 expression appears to be responsible for the relative progesterone resistance seen in normal squirrel monkeys (Hubler, Denny et al. 2003); however it also leads to glucocorticoid and androgen resistance, which has not been described in women with endometriosis. FKBP52 gene knockout in mice leads to progesterone resistance and embryo implantation failure, which can be overcome with supplemental progesterone (Tranguch, Wang et al. 2007).\\nCo-regulators, which bind steroid receptors and modify their nuclear effects, are also potential modifiers of progesterone resistance. One co-activator, Hic-5, has recently been shown to be deficient in the stroma of proliferative and late-secretory endometria of women with endometriosis (Aghajanova et al. 2009), and null mutations in the progesterone receptor co-activator, steroid receptor co-activator 2 (SRC-2) cause mice to have severe defects in endometrial receptivity. KLF9 is another progesterone receptor co-regulator, whose absence in the mouse results in partial progesterone resistance, subfertility and reduced HOXA10 expression (Simmen and Simmen 2002; Simmen et al. 2002; Zhang et al. 2003). KLF9 was recently shown to be reduced in a mouse model of endometriosis (Lee et al. 2009) and in infertile women with endometriosis (Pabona et al. 2012). Whether these findings are a root cause or an effect of endometriosis remains to be evaluated, but they lend further credence to the concept of progesterone resistance.',\n",
       "  'Discussion': '',\n",
       "  'Conclusion': 'To summarize, although a plethora of hormones are produced by the corpus luteum, the sequential actions of oestrogen and progesterone, without any other corpus luteum hormones, are sufficient to drive a highly receptive endometrium in humans. The mechanisms by which oestrogen and progesterone act are highly complex and involve multiple nuclear receptors as well as recently described membrane receptors. Cell-type specific effects of oestrogen and progesterone depend on differential expression of receptors, chaperones and co-regulators as well as chromatin structure. The role of oestrogen in endometrial proliferation and the importance of that proliferation in embryo implantation are clear. It is also likely that a small amount of oestrogen is necessary for normal luteal-phase endometrium in humans, but the sources of oestrogenic activity and dose requirements remain unclear and the possibility remains that oestrogen or oestrogen-like substances are made locally within the endometrium.\\nProgesterone is absolutely necessary, during the secretory phase, to allow the endometrium to be receptive to the implanting embryo. However, evidence in normal women suggests that only a very small amount of progesterone is necessary, a concentration achieved by the vast majority or perhaps all ovulatory women. Thus, in women with otherwise normal endometrial function, only small amounts of oestrogen and progesterone appear to be required in the luteal phase for full reproductive function. There is also evidence that some women, especially those with endometriosis-related infertility, may be somewhat resistant to the actions of progesterone and it seems that some of these defects are likely to be overcome with higher concentrations of progesterone, but that hypothesis remains to be proven.',\n",
       "  'URL': 'https://doi.org/10.1016/j.rbmo.2013.06.010',\n",
       "  'Journal': 'Reprod Biomed Online'},\n",
       " {'PaperTitle': 'GATA4 and GATA6 regulate pancreatic endoderm identity through inhibition of hedgehog signaling',\n",
       "  'Publication': '2016',\n",
       "  'Authors': 'Shouhong Xuan, Lori Sussel',\n",
       "  'Email': 'lgs2@cumc.columbia.edu',\n",
       "  'Abstract': 'GATA4 and GATA6 are zinc finger transcription factors that have important functions in several mesodermal and endodermal organs, including heart, liver and pancreas. In humans, heterozygous mutations of either factor are associated with pancreatic agenesis; however, homozygous deletion of both Gata4 and Gata6 is necessary to disrupt pancreas development in mice. In this study, we demonstrate that arrested pancreatic development in Gata4fl/fl; Gata6fl/fl; Pdx1:Cre (pDKO) embryos is accompanied by the transition of ventral and dorsal pancreatic fates into intestinal or stomach lineages, respectively. These results indicate that GATA4 and GATA6 play essential roles in maintaining pancreas identity by regulating foregut endodermal fates. Remarkably, pancreatic anlagen derived from pDKO embryos also display a dramatic upregulation of hedgehog pathway components, which are normally absent from the presumptive pancreatic endoderm. Consistent with the erroneous activation of hedgehog signaling, we demonstrate that GATA4 and GATA6 are able to repress transcription through the sonic hedgehog (Shh) endoderm-specific enhancer MACS1 and that GATA-binding sites within this enhancer are necessary for this repressive activity. These studies establish the importance of GATA4/6-mediated inhibition of hedgehog signaling as a major mechanism regulating pancreatic endoderm specification during patterning of the gut tube.',\n",
       "  'Introduction': 'The pancreas is a vital organ that functions to regulate digestive processes and glucose homeostasis (Hegyi and Petersen, 2013; Mastracci and Sussel, 2012). In adults, the three major diseases associated with the pancreas are pancreatitis, pancreatic cancer and diabetes. In addition, congenital genetic defects contribute to defective organ development, such as annular pancreas, and pathological conditions, including pancreatic agenesis and neonatal diabetes (Etienne et al., 2012; Rubio-Cabezas and Ellard, 2013).\\nPancreas development is initiated in two distinct regions of the foregut endoderm in response to signals derived from adjacent tissues (Chen et al., 2004; Kim and MacDonald, 2002; Kumar et al., 2003; Zaret and Grompe, 2008; Zaret et al., 2008). In particular, several studies have shown that a major function of these signals is to repress hedgehog signaling in the presumptive pancreatic endoderm. In the prospective dorsal pancreatic endoderm, notochord-derived activin and fibroblast growth factor (FGF) signals are necessary for the inhibition of sonic hedgehog (Shh) to allow the induction of bud morphogenesis and the pancreatic transcriptional program (Apelqvist et al., 1997; Hebrok et al., 1998). In the prospective ventral pancreatic endoderm, signals secreted by the lateral plate mesoderm, cardiac mesoderm and septum transversum play important roles in inducing pancreatic versus liver fates; the inhibition of WNT and bone morphogenetic protein (BMP) signaling is necessary for ventral pancreas induction, and FGF induces the local expression of Shh to inhibit pancreatic fates in favor of liver lineages (Deutsch et al., 2001; Zaret and Grompe, 2008). As development proceeds, the dorsal and ventral pancreatic buds merge, pancreatic cell specification is initiated and the diverse pancreatic cell types differentiate and proliferate to form the mature functional organ (reviewed by Jorgensen et al., 2007; Pan and Wright, 2011; Pictet and Rutter, 1972).\\nThe GATA regulatory proteins belong to a highly conserved six-member family of zinc finger transcription factors that play essential distinct and overlapping roles during embryonic development, including germ layer specification, organ formation and cell lineage determination. GATA1, GATA2 and GATA3 are important for hematopoiesis, whereas GATA4, GATA5 and GATA6 are important for the development of mesoderm- and endoderm-derived organs, including heart, liver and pancreas (Zhou et al., 2012).\\nGata4 null mice die at around embryonic day (E) 9.5 owing to defects in heart morphogenesis (Kuo et al., 1997; Molkentin et al., 1997; Narita et al., 1997) and Gata6 null mice die before E7.5 due primarily to defects in extra-embryonic endoderm (Koutsourakis et al., 1999; Morrisey et al., 1998). In the pancreas, Gata4 and Gata6 have overlapping expression in the pancreatic endoderm, but gradually become expressed in separate domains: Gata4 becomes restricted to the exocrine compartment and Gata6 is predominantly expressed in the endocrine compartment (Decker et al., 2006; Ketola et al., 2004). Using a Cre-lox approach, we previously demonstrated that simultaneous deletion of Gata4 and Gata6 from pancreatic progenitor cells leads to pancreatic agenesis in newborn mice (Carrasco et al., 2012; Xuan et al., 2012). The importance of GATA4 and GATA6 in human pancreas development has also been highlighted in reports of genetic cases of human pancreatic agenesis: GATA6 haploinsufficiency contributes to the majority of pancreatic agenesis cases (Lango Allen et al., 2012), and GATA4 haploinsufficiency has been documented in a small number of patients with pancreatic agenesis (Shaw-Smith et al., 2014).\\nTo characterize the molecular changes underlying GATA4/6-mediated pancreas development, we performed global transcriptome analysis on pancreatic buds isolated from E12.5 embryos from control and Gata4fl/fl; Gata6fl/fl; Pdx1:Cre (pDKO); R26R:Tomato mice. Consistent with impaired pancreas development, the majority of downregulated genes were pancreatic progenitor markers. Surprisingly, however, many of the upregulated genes included a large number of intestinal and gastric transcription factors. Furthermore, genetic lineage tracing of cells expressing the pancreatic progenitor marker Pdx1 indicated that the pancreatic lineages were converted to stomach and intestinal cell fates. In addition, there was a notable upregulation of many components of the hedgehog pathway, suggesting that the major mechanism of GATA4/6-regulated pancreas development is through suppression of the hedgehog pathway. Previous studies have demonstrated that GATA factors can negatively regulate Shh expression in the stomach, limb bud, and somites (Daoud et al., 2014; Jacobsen et al., 2005; Kozhemyakina et al., 2014). Our studies suggest that GATA4 and GATA6 pattern the foregut endoderm through the repression of hedgehog signaling.',\n",
       "  'Methods': \"RNA-Seq and bioinformatics analysis\\nFluorescently labeled cells from E12.5 control and pDKO pancreata were manually dissected using a Leica MZ16F fluorescence dissecting microscope. Six controls and 12 pDKO pancreata were pooled for RNA-Seq (n=1 pools per genotype). RNA was generated (RNeasy Micro kit; Qiagen) and tested for quality (RIN values >8; Agilent Bioanalyzer 2100). RNA-Seq was performed on the Illumina HISEQ 2000V3 Instrument (Columbia Genome Center) at a depth of 25-30 million 100-bp single-end reads. FPKM values were used to measure RNA expression level and 23,700 genes were compared between control and mutant samples. DEseq analysis (DESeq2, R software package) was performed to identify differentially expressed genes (P<0.0008). RNA-Seq data have been deposited in Gene Expression Omnibus under accession number GSE77083.\\nQuantitative real-time PCR\\nRNA from E12.5 pancreata was isolated (RNeasy Micro, Qiagen) to generate cDNA (Invitrogen). qRT-PCR was performed using Taqman AOD probe sets or SYBR green Premix (Bio-Rad). Taqman AODs: Pdx1, Mm00435565-ml; Ptf1a, Mm04203788_gl; Gata4, Mm00484689_ml; Gata6, Mm00802636_ml; Cdx2, Mm01212280_m1; Sox2, Mm03053820_s1 (Open Biosystems-Thermo Scientific). Specific primers: Shh ex1 F, 5′-GGAGCAGACCGGCTGATGAC-3′; Shh ex2 R, 5′-TCGGTCACTCGCAGCTTCAC-3′; Ihh ex1 F, 5′-TCTTCAAGGACGAGGAGAACACG 3′; Ihh ex2 R, 5′-CACCCGCAGTTTCACACCAG-3′; Gli1 F, 5′TGGTACCATGAGCCCTTCTT 3′; Gli1 R, 5′-GTGGTACACAGGGCTGGACT-3′; Ptch1 F, 5′ATCTCGAGACCAACGTGGAG 3′; Ptch1 R, 5′-GCCTCTTCTCCTATCTTCTGACG-3′; Gata4 F, 5′-TAGTCTGGCAGTTGGCACAG-3′; Gata4 R, 5′-ACGGGACACTACCTGTGCAA-3′; Gata6 F, 5′-AGTTTTCCGGCAGAGCAGTA-3′; Gata6 R, 5′-AGTCAAGGCATCCACTGTC-3′.\\nIn situ hybridization and immunofluorescence analysis\\nRNA in situ hybridization was performed as previously described (Prado et al., 2004). Shh antisense riboprobe was prepared from a pBSK-Shh plasmid containing a full-length Shh cDNA. T3 RNA polymerase was used to transcribe a HindIII-linearized plasmid. Brightfield images were acquired using a Leica DM5500 microscope.\\nAll immunofluorescence analysis was performed on frozen sections as previously described (Xuan et al., 2012). Primary antibodies were: rabbit anti-Pdx1 (1:1000; 07-696, Millipore), mouse anti-Sox2 (1:250; MAB4343, Santa Cruz), mouse anti-Cdx2 (1:80; CDX2-88, BioGenex), mouse anti-Muc5AC (1:500; ab3649, Abcam) and rabbit anti-ATP4b (1:1000; MA3-923, Thermo Fisher Scientific). Secondary antibodies were Alexa Fluor 488-, 594- or 697- conjugates (1:500; Jackson ImmunoResearch). Fluorescence and confocal images were acquired using a Zeiss LSM 710 confocal microscope. Images and z-stack images were analyzed using ZEN software (Zeiss).\\nWestern blot analysis\\nCell lysates from a-TC cells were analyzed using mouse anti-Gata4 (1:1000; SC-25310, Santa Cruz), rabbit anti-Gata6 (1:200; SC-9055, Santa Cruz ) and rabbit anti-Gapdh (1:1000; ab9845, Abcam). Quantification of the relative intensity was performed using ImageJ software.\\nChromatin immunoprecipitation\\nE14.5 pancreata were manually dissected. Chromatin was prepared from these tissues as reported previously (Xuan et al., 2012). The following primers were used for PCR analysis:\\nMACS1 5′ primers: forward 5′-GTGTACAGAAAGCCTAGTGTGTC-3′, reverse 5′-GCAGCAATAAAAGACTATGCCTC-3′;\\nMACS1 3′ primers: forward 5′-ACATTCTGTTGCTAACTTGAAGTG-3′, reverse 5′-AAGCCTGGAATTTATAGCATCTCA-3′; Gata site 2 primers: forward 5′-GAGGCATAGTCTTTTATTGCTGC-3′, reverse 5′-GCATCTGACGGATTGTTAGCCT-3′; Gata site 3 primers: forward 5′-ATGGTCCTGAAGTTTGTTCATCC-3′, reverse 5′-CACTTCAAGTTAGCAACAGAATGT-3′; CPA1exF, CGGAGCTAGTAGCAACCCCT; CPA1exR, CAGGAGCTGGTTCTGATGTG.\\nLuciferase assay\\nThe 806-bp MACS1 genomic fragment was PCR amplified (5′ GGTACCTTGTACTGGTGAGTGT, 3′ AGATCTATAACCAGGAGTCCAGG) and cloned into pGL4.27 luciferase reporter plasmid. A DNA fragment containing the 806-bp MACS1mut with point mutations in the four putative Gata sites was synthesized by Integrated DNA Technologies. Full-length cDNAs of mouse Gata4 and Gata6 were cloned into the pcDNA3 vector. These plasmids were transfected along with Renilla luciferase construct into α-TC6 cells using X-treme (Roche) transfection reagent according to the manufacturer's instructions. Forty-eight hours after transfection, cell lysates were collected and luciferase samples were prepared using the Dual Luciferase Reporter Assay System (Promega). Luciferase activities were measured using an Orion II Luminometer and luciferase activity was normalized to Renilla activity.\\nImage quantification\\nImageJ software was used to quantify the band intensities in the images for PCR gel and western blotting.\\nStatistical analysis\\nAll values are expressed as mean±s.e.m. Statistical analysis was performed using a two-tailed Student's t-test. Results were considered significant at P<0.05.\",\n",
       "  'Results': 'Hedgehog signaling is highly upregulated in E12.5 pDKO pancreata\\nSimultaneous deletion of Gata4 and Gata6 in the pancreatic progenitor domain resulted in severely aplastic pancreatic buds (Carrasco et al., 2012; Xuan et al., 2012). To characterize the molecular pathways that function downstream of GATA4 and GATA6 to mediate the regulation of pancreatic differentiation and morphogenesis, we performed genome-wide transcriptome analysis of the arrested pancreatic rudiments from pDKO embryos compared with littermate control embryos. A R26R;Tomato fluorescent reporter was introduced into the strain to facilitate purification of the PDX1-derived lineages from pooled pancreata of each genotype (Fig. 1A). Quantitative real-time PCR (qPCR) was used to confirm deletion of Gata4 and Gata6, and validate the reduction in pancreatic progenitor markers Pdx1 and Ptf1a (Fig. S1). RNA-Seq analysis of pooled wild-type versus pDKO Tomato+ pancreata identified ∼251 genes that were significantly differentially expressed in the mutant embryos: 118 genes were downregulated and 133 genes were upregulated (Fig. 1B,C). Strikingly, hedgehog signaling was revealed to be the most affected pathway [using ingenuity pathway analysis (IPA) software], with many of the transcriptionally regulated components of the hedgehog pathway being dramatically upregulated (Table 1). qPCR confirmed a 30- to 50-fold upregulation of the two major hedgehog ligands Shh and Indian hedgehog (Ihh), and a 7- to 10-fold upregulation of the receptor patched homolog 1 (Ptch1) and the downstream transcriptional activator Gli1 (Fig. 1D).\\nIn situ hybridization for Shh on E10.5 embryos revealed the erroneous expression of Shh expression throughout the pDKO pancreatic domain (Fig. 1E). As inhibition of Shh expression by notochord-derived activin and FGF or cardiac mesoderm-derived FGF has been postulated to be an early event in pancreas induction (Deutsch et al., 2001; Hebrok et al., 1998), our findings suggest that repression of hedgehog signaling downstream of these mesodermal signals is mediated by GATA4 and GATA6 activity.\\nCell fate switching occurs in the pDKO pancreatic endoderm\\nPrevious studies have demonstrated that the absence of Shh expression in the early pancreatic domain is required for normal pancreas development (Hebrok et al., 2000). Furthermore, ectopic upregulation of Shh (Apelqvist et al., 1997; Haumaitre et al., 2005) is associated with diminished pancreas formation and expanded stomach or gut regionalization. Consistent with the elevation of hedgehog signaling in the pDKO pancreatic endoderm domain, there was a large reduction of pancreatic progenitor markers in the pDKO, whereas there was a notable increase in genes encoding stomach progenitor markers, such as Sox2 and Nkx6-3 (5.8- and 19.7-fold, respectively), and intestinal progenitor marker genes, such as Cdx2 and Isx (20-fold) (Table 2).\\nTo determine whether the reciprocal changes in marker expression resulted from re-specification of the pancreatic endoderm in the pDKO embryos, we assessed pancreatic, stomach and intestinal lineage markers in the pDKO; R26R:Tomato mice using the lineage label to track the potentially re-specified pancreatic lineages. As expected, in control E10.5 pancreatic endoderm, Tomato-labeled pancreatic lineage cells in the dorsal pancreatic anlage express the pancreatic lineage marker Pdx1, but do not express the stomach marker gene Sox2 (Fig. 2A-D′). However, in the pDKO embryos, Tomato-labeled pancreatic lineage cells begin to express Sox2 in the place of Pdx1 (Fig. 2E-H′), indicating that pancreatic lineage cells in the dorsal pancreas have switched to a stomach identity. Similarly, in the ventral pancreatic endoderm, the pancreatic lineages appear to adopt intestinal cell fates (Fig. 2I-Y). In pDKO E9.5 embryos, a small number of pancreatic lineage cells begin to co-express Pdx1 and Cdx2 or express Cdx2 in place of Pdx1 (Fig. 2M-P″). By E10.5, there is an increasing number of Pdx1 lineage-labeled cells that express Cdx2 (Fig. 2V-Y′). At E18.5, conversion of the Pdx1 lineage to mature stomach fates is widespread and many of the Tomato lineage-labeled cells have become incorporated into the stomach epithelium as previously demonstrated (Xuan, et al., 2012) where they co-express mature stomach markers, including MUC5AC and ATP4B (Fig. S2). These results suggest that the erroneous upregulation of the hedgehog pathway in pDKO embryos results in re-specification of the dorsal and ventral pancreatic lineages to the adjacent stomach and intestinal fates, respectively.\\nGATA4 and GATA6 suppress the activity of the Shh foregut endoderm enhancer MACS1\\nGATA regulation of Shh expression has been reported in several organ systems. During gastric development, Gata4 and Shh expression is mutually exclusive, suggesting that GATA4 might inhibit the expression of Shh (Jacobsen et al., 2005). In the limb bud, GATA6 was shown to inhibit Shh gene expression through a limb bud-specific enhancer of the Shh gene (Kozhemyakina et al., 2014), and in the somitic system, GATA4/5/6 were found to inhibit Shh signaling by inhibiting Gli1 expression, although direct inhibition of Shh was not required (Daoud et al., 2014). The upregulation of hedgehog pathway expression in the pDKO mice suggested that Shh might be repressed by GATA4 and/or GATA6. An 806-bp MACS1 enhancer located ∼740\\u2005kb upstream from the Shh transcriptional start site has been shown to be sufficient for driving Shh expression specifically in the foregut endoderm (Anderson et al., 2014; Sagai et al., 2009). Using position weight matrix analysis we identified four putative GATA consensus elements within the MACS1 enhancer (Fig. 3A, red text). Chromatin immunoprecipitation (ChIP) analysis of these sites revealed strong binding of GATA4 and GATA6 to the two most upstream GATA consensus sites (Gata site 1 and Gata site 2) and minimal, if any, binding to the 3′ sites (Gata site 3 and Gata site 4) (Fig. 3B,C).\\nTo determine whether GATA4 and GATA6 could inhibit Shh expression through the MACS1 enhancer element, we performed luciferase experiments in a Shh-expressing pancreatic αTC6 cell line that lacks endogenous Gata4 and Gata6 (Fig. S3A). Transfection of pGL4.27-MACS1 led to an approximately sixfold increase of luciferase activity compared with pGL4.27 vector alone, suggesting that MACS1 has high activity in these cells (Fig. 3C), which is consistent with previous in vivo expression analysis (Kawahira et al., 2005). Furthermore, co-transfection with either Gata4 or Gata6 reduced MACS1 activity by >50%, with GATA6 having the largest effect (∼75%). Furthermore, simultaneous transfection of half the amount each of Gata4 and Gata6, such that the total amount of GATA protein remained constant (Fig. S3B,C) demonstrated that GATA4 and GATA6 could function together to repress MACS1 activity (Fig. 3D). Surprisingly, however, overexpression of Gata4 and/or Gata6 was still able to partially repress a MACS1 fragment that contained mutations in the four putative Gata consensus sites (Fig. 3A, blue text; Fig. 3D, ‘MACS1mut’). This might indicate that overexpression of GATA factors can bind and activate through cryptic Gata sites (Fig. 3A, green text) present within the MACS1 enhancer or that GATA factors mediate MACS1 repression through an indirect mechanism, as previously described in the heart (Rivera-Feliciano et al., 2006).',\n",
       "  'Discussion': 'Repression of the hedgehog pathway in the presumptive ventral and dorsal pancreatic endoderm in response to signals from adjacent tissues has been well documented (Apelqvist et al., 1997; Chung and Stainier, 2008; diIorio et al., 2007; Haumaitre et al., 2005; Hebrok et al., 1998; Roy et al., 2001). However, the transcriptional regulators that mediate these mesodermal signals upstream of the hedgehog pathway have not yet been identified. In this study, we demonstrate the importance of GATA4 and GATA6 in repressing hedgehog pathway components within the pancreatic anlage. Furthermore, our in vitro data suggest GATA-mediated repression of Shh occurs partially through the endoderm-specific MACS1 enhancer. Consistent with a crucial role for GATA4 and GATA6 in repressing hedgehog signaling in the pancreatic endoderm, there is a conversion of pancreatic lineages to stomach or intestinal cell fates when GATA function is absent. The timing of Shh activation relative to the observed fate changes supports a model in which GATA-mediated repression of hedgehog signaling is necessary to delineate and maintain foregut endoderm fates (Fig. 3E). These findings also reinforce the proposed role for GATA4 and GATA6 as pioneer factors that are positioned at the top of the gene regulatory cascade that patterns tissue-specific gene expression pathways (Zaret et al., 2008).\\nRecently, it has been reported that haploinsufficiency of GATA4 or GATA6 genes account for more than half of all human pancreatic agenesis cases (Lango Allen et al., 2012), suggesting their important role in human pancreas development. Our discovery that GATA4 and GATA6 are essential for maintaining repression of hedgehog signaling provides important implications for the pathways regulated by the GATA factors during human pancreas development and could lead to novel strategies to detect and/or prevent pancreatic agenesis before birth.',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1242/dev.127217',\n",
       "  'Journal': 'Development'},\n",
       " {'PaperTitle': 'Improving Replication in Endometrial Omics: Understanding the Influence of the Menstrual Cycle',\n",
       "  'Publication': '2025',\n",
       "  'Authors': 'Jessica Chung, Peter Adrian Rogers',\n",
       "  'Email': 'parogers@unimelb.edu.au',\n",
       "  'Abstract': 'The dynamic nature of human endometrial tissue presents unique challenges in analysis. Despite extensive research into endometrial disorders such as endometriosis and infertility, recent systematic reviews have highlighted concerning issues with the reproducibility of omics studies attempting to identify biomarkers. This review examines factors contributing to poor reproducibility in endometrial omics research. Hormonal fluctuations in the menstrual cycle lead to widespread molecular changes in the endometrium, most notably in gene expression profiles. In this review, we examine the variability in omics data due to the menstrual cycle and highlight the importance of accurate menstrual cycle dating for effective statistical modelling. The current standards of endometrial dating lack precision and we make the case for using molecular-based modelling methods to estimate menstrual cycle time for endometrium tissue samples. Additionally, we discuss statistical considerations such as confounding and interaction effects, as well as the importance of recording the detailed and accurate clinical information of patients. By addressing these methodological challenges, we aim to establish more robust and reproducible research practises, increasing the reliability of endometrial omics research and biomarker discovery.',\n",
       "  'Introduction': 'Endometrial function and fertility can be affected by numerous conditions including endometriosis, adenomyosis, and fibroids. A large component of studying these disorders has attempted to identify biomarkers that are associated with the condition in order to understand the aetiology of the disorder and improve diagnosis and treatment. Unfortunately, systematic reviews of endometrial studies have indicated poor reproducibility with a lack of consensus on genes that are differentially expressed between endometrial disorders [1, 2].\\nThe complex nature of human endometrial tissue renders it particularly challenging to study. In contrast to the relatively homeostatic state of most tissues, the endometrium undergoes a cyclical process of rapid growth, breakdown, and shedding in response to hormonal changes [3]. This process also involves functional changes, allowing for the implantation of an embryo during a narrow time frame following ovulation [4]. Such dynamism entails rapid changes at the molecular level with variations in gene expression, protein expression, DNA methylation, and metabolites observed across different phases of the menstrual cycle [5, 6, 7]. Consequently, accurately identifying the menstrual cycle phase from which endometrial tissue samples are taken is critical for effective statistical modelling due to the large role that the menstrual cycle plays in explaining observed variation.\\nThe aim of this review article is to examine underlying factors contributing to the observed lack of reproducibility in endometrial omics studies, with an emphasis on transcriptomics studies, and provide recommendations to mitigate some of these difficulties. We will also discuss the accuracy of current endometrial dating methods and statistical considerations that are relevant to endometrial analyses. Furthermore, we make the case for using molecular methods to obtain menstrual cycle time estimates for endometrial samples and advocate for their inclusion in statistical models.',\n",
       "  'Methods': '4. Histological and Hormone-Based Endometrium Dating Methods Lack Precision\\nEndometrial tissue dating is the process of assigning a menstrual cycle stage or time point for a given tissue sample. The Noyes criteria [32] for endometrial dating represent the current gold standard and rely on the histological assessment of morphological features. When dating endometrial tissue histologically, samples are typically classified into cycle phases encapsulating several days (e.g., ‘menstrual’, ‘proliferative’, ‘early-secretory’, etc.) [33]. Despite being classified in the same phase, samples can have markedly different gene expression profiles, especially during the early–mid-secretory phase when the endometrium enters the window of implantation [27, 28]. It has been argued previously that histological endometrial dating lacks sufficient accuracy to enable the reliable and precise assignment of a cycle day or narrow time interval [34]. Accuracy and interobserver agreement between pathologists can vary, as demonstrated in a study by Duggan et al. [35], which only found a 68% agreement within one day between experienced pathologists when dating secretory post-ovulatory day samples. Furthermore, while inaccuracies from histological endometrial dating are not uncommon for healthy tissue, the situation is exacerbated by the fact that collected samples may be abnormal in some way due to an underlying condition. Issues can include scant samples with little functional tissue, the inclusion of pathological tissue such as polyps, and the presence of endometritis [36]. A poorly developed secretory endometrium and asynchronous glands, with tissues containing a mix of features from varying points in the menstrual cycle, can also cause ambiguity in dating. In particular, the presence of asynchronous endometrial glands is speculated to be relatively common in women with recurrent reproductive failure [37], adding further difficulty to dating samples for the purpose of researching endometrial disorders. Hormone-based sample-dating methods have also been used to detect where a woman is in the cycle, specifically when examining the secretory phase and fertility (e.g., [38, 39, 40]). In particular, luteinising hormone (LH) typically peaks 10 to 12 h before ovulation with the onset of the surge starting 35 to 44 h prior to ovulation [41]. Detecting LH in urine is a relatively easy task via commercial products such as test strips [42]. However, there is considerable variation in the pattern of LH surges in different women, such as the amplitude of the maximal LH concentration relative to the baseline and the duration of the peak from the onset [43]. Additionally, Park et al. [43] found variability in LH surge configuration, classifying women into three broad categories: spike (a single peak observed, 41.9%), biphasic (two local maxima observed, 44.2%), and plateau (LH levels remained at the peak for 2–3 days, 13.9%). There is also evidence to suggest that an LH surge does not always precede ovulation, with a retrospective cohort study in women who had previously undergone IVF treatment finding that 46.8% had premature LH surges (that did not precede ovulation), and a subset of those (37%) exhibited multiple premature LH surges [44]. Studies may also employ patient-reported cycle time (e.g., counting the number of days since a woman has started her cycle) as part of their methodology to date endometrial tissue. One complication with this method is that women have considerable variability in cycle lengths. Thus, collecting biopsies from women after the same number of days after the onset of menstruation can result in different cycle phases being sampled. For instance, in real-world menstrual cycle data from more than 600,000 menstrual cycles, an estimated 95% of luteal-phase durations ranged between 10.0 and 14.8 days [45]. This consequently affects accuracy when using LH surges to date samples, where studies often collect samples 6–8 days after the LH peak to target the window of implantation. Combined with the inherent difficulties of detecting the LH peak, this method of dating is likely to be relatively inaccurate.\\n\\n5. Molecular Methods for Estimating Menstrual Cycle Time\\nMore recently, endometrial dating approaches based on gene expression have been proposed [26, 27, 46]. These molecular dating approaches have the advantage of not requiring a trained pathologist and also being more reproducible and less subjective, given that variation between the assessments of different pathologists is common [34, 35, 47]. A recent development is the endest R package [27] which estimates time in the menstrual cycle in silico using gene expression data from endometrial biopsies. To develop this method, the authors used 266 histologically dated endometrial samples to characterise the gene expression of all genes throughout the menstrual cycle. Through iterative refinement, the categorical classifications were transformed into a continuous numerical scale and the samples were reassigned to the time point that had the best explanatory power. In essence, this method works by comparing the expression of all genes to the expected expression throughout the menstrual cycle and minimises a loss function to return a time between 0 and 100 that best fits the observed data. To use this method, users must supply either the RNA-seq expression or microarray expression values of their endometrium samples. Another available computational procedure that can generate an estimate for menstrual cycle time is EndoTime [48], where users supply their own data to train an iterative model. Using RTq-PCR, the authors were able to use the expression levels of six genes to build a model for luteal-phase cycle estimation on a continuous numerical scale. As this procedure is generalisable, users have freedom to employ their preferred gene expression technology and select relevant target genes; however, they must build their own model using labelled training data. With every endometrial sample assigned a numerical time value, a large proportion of observed gene variability can be explained by this single covariate. Teh et al. [27] showed that when cycle time was modelled as a spline across the whole menstrual cycle, 50% of gene expression models had an R-squared value above 0.19 and 30% of models had an R-squared value above 0.36. This molecular time also demonstrated how rapidly genes can change during certain parts of the menstrual cycle. Figure 3 shows examples of genes that change rapidly in the cycle and have a high proportion of variance explained by the menstrual cycle.\\n\\n6. Advantages of Molecular Dating Methods over Histological and Hormone-Based Methods\\nThe availability of different endometrial dating methods raises questions about the relative accuracy of each. However, this is a difficult question to answer as there is no ground truth available. Which method is ‘best’ may also be dependent on the specific question being asked. Histological dating is the long-established gold standard and is independent from omics data that are analysed. Some may argue that endometrial tissue structure and cellular organisation best characterises the menstrual stage of a tissue sample. On the other hand, it could be asserted that the molecular phenotype is the best representation of what characterises a tissue sample as it can more accurately reflect molecular function and biological processes. Differences in molecular phenotyping may not be detectable under a microscope and molecular quantification can provide more fine-grained estimates. In the context of differential gene expression, molecular staging is typically superior to histology dating. Capturing as much variation as possible is ideal in order to model the data and maximise statistical power. Consider an RNA-seq endometrium study where the disease state is the variable being studied where all samples are collected during the secretory phase. Any variation observed that is not due to the disease state can be seen as unwanted variation which would be ideally accounted for with covariates. If a PCA plot shows concordance between the position and the estimated time, this indicates that a noteworthy amount of variation can be explained by the estimated time (see Box 1 for examples). A PCA plot showing good concordance with histology dating yet stronger concordance with molecular dating indicates that relying solely on histology dating may leave additional unexplained variance in downstream analyses, variance that could be otherwise captured by using molecular dating. This pattern is likely to emerge even in studies where all samples are collected within a narrow time frame due to how rapidly gene expression can change in the menstrual cycle. Box 1. Comparing provided menstrual cycle information to molecular dating in public endometrial datasets. NCBI’s Gene Expression Omnibus (GEO) database is a public repository containing many thousands of endometrium biopsy samples from transcriptomics studies. When depositing data into GEO, authors typically include sample-associated metadata such as age, BMI, disease status, and some measure of cycle stage. With known issues surrounding accuracy, it follows to question how accurate the cycle stage metadata are for these public datasets. While estimating true accuracy is not possible due to the absence of definitive truth, we can compare the study-provided cycle time with an estimate of the cycle time based on gene expression data [27] and observe concordance across different studies and measures of menstrual cycle time. Table 1 lists nine RNA-seq studies from the GEO database that examined eutopic endometrial tissue and Figure 4 illustrates differences between the provided cycle dating and cycle time es-timated by molecular dating for these studies, with the molecular dating showing greater con-cordance in sample positioning in the PCA space. With molecular dating estimates being derived from omics data, the ability to have multiple estimates from independent methods for a single endometrial sample becomes more feasible. One advantage of having multiple dating estimates is the ability to verify consistency. When estimates diverge significantly, discordant samples can be flagged to identify underlying abnormalities such as an asynchronous endometrium or potential sample mix-ups.\\n\\n7. Other Statistical Considerations for Menstrual Cycle Effects\\n7.1. Sample Size and Chance Confounding\\nEndometrial samples are typically obtained via invasive biopsy procedures, which can make sample collection difficult, leading to studies with small sample sizes. These reduced sample sizes diminish the statistical power of studies; by definition, limiting the potential to detect differentially expressed genes with high confidence. Many case–control studies attempt to limit sample collection to a specific phase (e.g., [55, 56, 57]). Chance confounding, where covariates are imbalanced due to chance, is more likely to occur with small sample sizes [58]. Severe imbalances, such as one group containing all earlier samples and the other group containing all later samples, can result in a differential expression analysis between groups mistakenly identifying genes that change due to normal menstrual cycle effects rather than the condition being studied (Figure 5). Overestimated effect sizes are also more common with small sample sizes due to higher amounts of sampling variability which negatively impacts replicability. Early microarray studies typically had very few patients (e.g., [8, 30, 57]), and while some studies have been cited hundreds of times, there is still no clear consensus for a set of validated biomarkers that can distinguish between endometrial conditions [2, 14]. Small sample sizes, combined with the previously discussed complication of large amounts of gene variation due to cycle stage effects, lead to an increased susceptibility to mistaking noise for signals and publishing flawed results.\\n7.2. Confounding Variables\\nMany omics studies are observational case–control studies and employ some form of multivariate regression analysis. A critical step in statistical modelling is the careful selection of which biological factors to include as covariates, commonly referred to as ‘controlling for a variable’. These may include factors such as patient age, weight, medications, disease comorbidities, and the time in the menstrual cycle when the biopsy was taken. Ideally, the decision to include or exclude a variable in a statistical analysis involves considering the causal pathways that generate gene expression observations related to the endometrial disorder and therefore requires domain expertise. Care must be taken to not ‘overadjust’ when including covariates in models as this can reduce the effects of factors of interest or introduce bias [60]. Frameworks such as Pearl’s back-door criterion [61] can be employed to evaluate which variables to use as controls, while guidelines containing numerous examples can be found in resources such as those developed by Cinelli et al. [62].\\n7.3. Interaction Effects\\nIn addition to confounding effects, interaction effects between the disorder being studied and other variables should be considered. Interaction effects refer to situations where the effects of two variables are not independent from each other and have varying effects depending on their combination. For instance, an endometrial disorder may have an interaction effect with the menstrual cycle stage, leading to different gene expression changes in different cycle phases. For example, Burney et al. [30] identified the dysregulated gene expression of genes involved in cellular proliferation during the early secretory phase in the endometrium of women with endometriosis which was not seen in other phases of the menstrual cycle. Interaction effects can be addressed by including interaction terms in the statistical model or stratifying the samples by the menstrual cycle stage and disease state and performing contrasts between the relevant groups.\\n7.4. Discrete vs. Continuous Cycle Time Modelling\\nAs discussed previously, menstrual cycle data are often recorded as a cycle phase when estimated histologically, which spans several ‘cycle days’, during which many tissue and gene expression changes can occur. This lack of resolution is suboptimal. Time within the menstrual cycle is continuous, and as such, discretising the variable into blocks results in a loss of information that can have adverse effects on statistical power [63]. For example, a sample in the proliferative stage might be closer to other early-secretory-stage samples than other proliferative stage samples but still be classified as proliferative. Examining a PCA plot of the data and identifying outliers can mitigate some of this effect; however, it can also introduce researcher degrees of freedom (where certain decisions can be made to bias towards more favourable results) in classifying which samples belong in which cycle group. On the other hand, one advantage of recording menstrual cycle time as a continuous numerical value allows it to be used in regression models as a numerical variable, with the option of modelling time functions as non-linear, such as by using polynomial or spline functions (see Box 2 for implementation examples). Since gene changes throughout the menstrual cycle are conspicuously non-linear (Figure 3), these methods have considerable utility in explaining observed variance, especially when modelling longer timescales across different parts of the cycle. Box 2. Accounting for cycle effects in linear models. Differential expression analysis is commonly performed using the R programming language [64] using packages such as Limma [65], edgeR [66], and DESeq2 [67]. These methods all require a model formula to set up the statistical model. Here, we create a design matrix which is the approach for model specification in Limma and edgeR. In these examples, ‘endo’ is a binary categorical variable that contains the disease state (case or control) for endometriosis and ‘age’ is a numerical variable. If ‘cycle_stage’ is a categorical variable (which includes phases such as proliferative, early secretory, and mid-secretory), we can simply include it in the model formula: design <- model.matrix(~endo + age + cycle_stage) Where ‘cycle_time’ is a numeric variable, we can simply include it in the model if we want to model the variable as a linear effect. Alternatively, we can use the splines R package to model cycle time as a curve. The ns() function can generate a B-spline basis matrix for use in the design matrix, where ‘df’ is the degree of freedom to control the complexity of the curve: design <- model.matrix(~endo + age + ns(cycle_time, df)) Polynomial modelling can also be performed using the base poly() function where ‘degree’ determines the degree of the polynomial: design <- model.matrix(~endo + age + poly(cycle_time, degree)) Cyclic splines can be used if endometrial samples span the whole menstrual cycle as cyclic splines can account for the biological continuity between the late secretory phase and the beginning of the menstrual phase. The cSplineDes() function from the mgcv R package (≥v1.7) [68] can produce a cyclic B-spline basis that allows for this modelling. Similarly to specifying the degrees of freedom, we supply the location of the spline knots as a vector. We must also remove a column from the basis to remove linear dependence in the model: X <- cSplineDes(cycle_time, knots) design <- model.matrix(~endo + age + X[,-ncol(X)]) Detailed guidance on spline modelling and parameter specifications can be found in the documentation of the respective R packages. Lastly, the amount of variation that is observed for a particular gene may not be constant throughout the menstrual cycle; there may be points in the cycle where expression varies greatly across a cohort. Such behaviour violates homoskedasticity assumptions for linear methods such as t-tests and ANOVAs and should be kept in mind if working with gene expression data across the whole menstrual cycle.\\n\\n8. Replicability Issues in DGE Analysis\\nThe type of data produced by genomic studies is particularly prone to problems in statistical analysis that exacerbate irreproducibility. Analyses of datasets with high dimensionality, where large numbers of potential response and explanatory variables are measured, are prone to becoming ‘fishing expeditions’. Often, thousands of hypotheses are tested in the hope of finding a significant result, without necessarily considering the number of different statistical tests that have been applied along the way or conducting appropriate multiple-testing adjustment. This can be particularly challenging for endometrial studies in which tissue collection occurs often from multiple different menstrual cycle stages that are dissimilar enough to justify separate sub-analyses rather than one combined analysis (e.g., [30, 69]). When performed correctly, statistical methods such as multiple hypothesis testing correction can mitigate reporting incorrect results by reducing false positive rates. However, multiple comparisons are not always as simple as counting the number of statistical tests that are performed and adjusting for this. Gelman and Loken [70] describe ‘The garden of forking paths’ where many informal, post hoc decisions made in the data collection and analysis process are performed after observing data and can cause inadvertent problems for replicability. For example, a researcher could choose to perform a different statistical test based on initial observations, exclude specific data points, include different variables for a regression, or decide to look for interaction effects when main effects do not appear significant. This leads to unintentional effects that resemble ‘p-hacking’, where many statistical tests are performed and only significant results are reported. One way to mitigate this is to preregister hypotheses and analysis methods before observing the data, which would reduce post hoc decisions. Preregistration has been empirically shown to increase the number of null results, with Kaplan and Irvin [71] observing that the number of positive results from 55 large RCTs decreased from 57% to 8% under the requirement for preregistration and transparency measures. The current paradigm of Null Hypothesis Significance testing (NHST) under the frequentist framework can be seen as another factor contributing to poor reproducibility and has been often criticised [72, 73, 74]. NHST is the default method of learning statistics in undergraduate statistical education, the practice of which typically consists of setting up a null hypothesis, calculating a p-value with a statistical test, and then rejecting the null hypothesis if the p-value is below a certain threshold, often p < 0.05. One common charge levelled at NHST is that it asks the wrong question, leading to misinterpretation [74]. Researchers are often interested in the probability of their hypothesis being true given the data, rather than the probability of observing the data (or more extreme) under the null hypothesis. Another important point is that p-values also do not convey effect sizes. An emphasis on searching for statistical significance (“stargazing”) can lose sight of the biological consequence attached to statistically significant findings. With very large datasets, NHST allows minuscule effect sizes to become detectable, perhaps with extremely low p-values, even if they are biologically inconsequential [74]. Moreover, biological systems are complex with often a large number of interacting variables; thus, the presence of confounding variables is likely. Simulations by Bruns and Ioannidis [75] have demonstrated that uncorrected confounding can lead to false positive results that are indistinguishable from true effects. Many researchers have advocated using Bayesian methods as an alternative to NHST, owing to Bayesian methods having numerous advantages [73, 74, 76]. A key feature of the Bayesian framework is the incorporation of existing knowledge into statistical models through the specification of priors, which can lead to more accurate inferences with limited or noisy data. Bayesian methods also result in a probability distribution of parameters, offering a more accurate description of the uncertainty around a measurement. Additionally, hierarchical models can be constructed to model complex scenarios and handle missing data and measurement uncertainty through probabilistic modelling. The growing accessibility of powerful computing resources and the development of probabilistic programming languages such as Stan [77] have facilitated the adoption of these statistical models that use simulation-based methods. Many of the challenges previously mentioned impact the replicability of endometrial omics analysis, but there also exists a widespread bias in DGE publications as a whole. Päll et al. [78] examined differential analysis from RNA-seq data in the NCBI GEO database, estimating a 59% upper limit of reproducibility (i.e., obtaining identical results when using the original workflow and downloaded data), with more recent studies being more reproducible. Analysis of p-value distributions from each experiment revealed that only 23% of experiments had a theoretically expected p-value distribution, again with an improving trend over time. The mean proportion of null effects was also shown to increase over time, indicating that genes were less likely to be identified as differentially expressed between conditions in more recent experiments. Analysis software used in respective experiments was heavily associated with both p-value distribution and proportion null effects, implying that part of the bias was driven by the chosen methodology.\\n\\n9. Pathologies, Comorbidities, and Accurate Diagnoses\\nCase–control studies are commonly employed in omics-based studies, grouping all patients with the pathology in one group and all controls in the other. Classifying patients into analysis groups is a critical part of research and is made difficult by the heterogeneity of endometrial disorders. Placing all women with the disorder into a single, homogeneous ‘case’ group may not be optimal for analysis. For example, the symptoms of women with endometriosis are highly variable, with 30–50% of endometriosis patients being infertile, 40–50% experiencing chronic pelvic pain, and 20–25% being asymptomatic [79]. This multifactorial presentation is likely to be reflected in molecular phenotyping, wherein various pathways are differentially activated in the presence of different symptoms, yielding considerable variability. The application of simple case and control groups without accounting for sub-types would be expected to decrease the statistical power to resolve differential expression signals. Severe cases of endometriosis have been shown to have greater ‘genetic loading’ in GWAS analyses [80, 81], as indicated by an increased number of SNPs conferring higher risk. After excluding mild endometriosis samples, Nyholt et al. [80] were able to discover additional SNPs associated with endometriosis, despite a smaller sample size. Infertility-related disorders may also have diverse aetiologies. Recurrent implantation failure, for example, is hypothesised to have two possible endometrial causes: the displacement of the window of implantation and disruption of the window due to pathologies, both of which could coexist in the same patient [82]. These two aetiologies are likely to confer different gene signatures and pathway involvement. As discussed above for endometriosis, the simple grouping of all infertile samples to form a homogeneous collection regardless of presentation would be expected to reduce the likelihood of identifying differentially expressed genes; there are likely to be subtypes (e.g., the displacement of the window of implantation) of subtypes (e.g., infertile). There exists a trade-off between defining homogeneous disease sub-groups and sample size because the class (sub-group) sample size is reduced with stricter criteria for patient inclusion. While smaller-sized classes generally tend to reduce statistical power, in many cases, more homogenous classes would be expected to more than offset the sample-size effect by reducing variability within groups, thereby allowing biological signals specific to a particular severity or subtype of disease to be revealed; essentially, class-specific signals would become more ‘focused’ [83]. Comorbidities in endometrial disorders are common and represent another important consideration when performing experimental design and analyses. The presence of endometriosis has been associated with an increased risk of other gynecological conditions such as adenomyosis [84], uterine fibroids [85] and ovarian cancers [86]. Non-gynecological comorbidities such as interstitial cystitis and irritable bowel syndrome further complicate matters as they have overlapping pain symptoms caused by shared mechanistic features [87]. These comorbidities can confer statistically confounding effects and, as such, it is vital that they are recorded. Many endometrial pathologies share overlapping symptoms, such as pelvic pain, abnormal uterine bleeding, and infertility [88], which, combined with high symptom heterogeneity, can contribute to difficulties in diagnosis. A retrospective study by Orlov and Jokubikiene [89] found that half of women displaying endometriosis-associated symptoms had no abnormal transvaginal ultrasound findings. The definitive diagnosis of endometriosis requires surgical and histological visualisation and, on average, diagnosis takes approximately 7 years from the onset of symptoms [90, 91]. Similarly, the average time to diagnosis for symptomatic uterine fibroids is approximately 3.6 years [92]. Comorbidities can confound diagnosis, making causal attribution to symptoms difficult. Even in the case of a single observed pathology, a diagnostician cannot ascribe a symptom to that pathology with high confidence. Diagnoses can be missed and pathologies may coexist with symptoms without being responsible for them. To further illustrate the complexity around the diagnosis of endometrial conditions, it has been observed that in over a third of cases of chronic pelvic pain symptoms reappear or persist after diagnosis and treatment [93]. This is also reflected in the non-response rate of endometriosis to laparoscopic surgery, which ranges from 20% to 38% [94, 95]. All bioinformatic analyses are downstream from biological classification. If diagnosis and classification are prone to error, results will be noisy and confer diminished statistical power. Perhaps a better approach to analysis might be to focus on symptoms instead of pathologies. The reason for this is twofold. Firstly, diagnosis is difficult, and symptom classification may lead to cleaner, more homogenous groups. Secondly, many women with pathologies have normal endometrial function, despite the presence of pathologies, and the symptoms, such as pain and infertility, are of most concern to patients. Practically, disorders such as endometriosis and fibroids do not require treatment in asymptomatic cases; thus, focusing on symptoms may better target clinical interventions and efforts towards discovering future treatments.',\n",
       "  'Results': '',\n",
       "  'Discussion': '',\n",
       "  'Conclusion': 'The current state of endometrial omics research suffers from a lack of replicability. A major reason for this is a failure of properly accounting for the menstrual cycle being a major source of variation. It is critical that a measure of the menstrual cycle stage is included in statistical models when modelling endometrial omics data to mitigate possible confounding and increase statistical power to find real effects. The current standards of endometrial omics methodology are inadequate and there is no justification for omitting menstrual cycle time from future analyses. Histological and hormone-based methods are commonly used for endometrial dating; however, we recommend using molecular methods (such as the method implemented in the endest R package [27]) for estimating cycle time in which gene expression data from endometrial tissue are used to obtain an estimate. Molecular methods typically explain a higher proportion of observed variance and have the added advantage of generating continuous variable values, allowing cycle time to be modelled as non-linear in regression models (e.g., using splines or polynomial functions). To improve the replicability of research findings, good statistical analysis is not enough; good data collection is also needed. While clean case–control groups are ideal, such groups are not always feasible. Consequently, it is important to record any phenotypic observation that may impact gene expression such as the presence of endometritis and other abnormalities. To further enhance replicability, pre-registration is encouraged. Decisions such as cycle stage sub-analyses and defining breakpoints when dichotomising variables ought to be decided beforehand. Finally, conducting validation studies with new samples is essential to reduce the propagation of inaccurate research findings. While endometrial tissues present unique challenges with regard to experimental design and bioinformatic analysis, steps can be taken to minimise publishing erroneous conclusions. By considering these factors, researchers can enhance the reproducibility and validity of their findings.',\n",
       "  'URL': 'https://doi.org/10.3390/ijms26020857',\n",
       "  'Journal': 'International Journal of Molecular Sciences'}]"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "my_json_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9a5c59e",
   "metadata": {},
   "source": [
    "## Create database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "id": "1f940678",
   "metadata": {},
   "outputs": [],
   "source": [
    "splitter = RecursiveCharacterTextSplitter(\n",
    "    chunk_size=1000,\n",
    "    chunk_overlap=100,\n",
    "    separators=[\"\\n\\n\", \"\\n\", \".\", \"!\", \"?\", \" \"]  # smart splitting\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "66c60074",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'PaperTitle': 'SARS-CoV-2 infection risk assessment in the endometrium: viral infection-related gene expression across the menstrual cycle',\n",
       " 'Publication': '2020',\n",
       " 'Authors': 'Ismael Henarejos-Castillo, Patricia Sebastian-Leon, Almudena Devesa-Peiro, Antonio Pellicer, Patricia Diaz-Gimeno',\n",
       " 'Email': 'patricia.diaz@ivirma.com',\n",
       " 'Abstract': 'Objective: To determine the susceptibility of the endometrium to infection by- and thereby potential damage from- SARS-CoV-2. Design: Analysis of SARS-Cov-2 infection-related gene expression from endometrial transcriptomic data sets. Setting: Infertility research department affiliated with a public hospital. Patient(s): Gene expression data fromfive studies in 112 patients with normal endometrium collected throughout the menstrual cycle. Intervention(s): None. Main Outcome Measure(s):Gene expression and correlation between viral infectivity genes and age throughout the menstrual cycle. Result(s): Gene expression was high forTMPRSS4, CTSL, CTSB, FURIN, MX1, andBSG; medium forTMPRSS2; and low forACE2. ACE2, TMPRSS4, CTSB, CTSL, andMX1 expression increased toward the window of implantation.TMPRSS4 expression was positively correlated withACE2, CTSB, CTSL, MX1, andFURIN during several cycle phases;TMPRSS2 was not statistically significantly altered across the cycle.ACE2, TMPRSS4, CTSB, CTSL, BSG, andMX1 expression increased with age, especially in early phases of the cycle. Conclusion(s): Endometrial tissue is likely safe from SARS-CoV-2 cell entry based onACE2 and TMPRSS2expression, but susceptibility increases with age. Further,TMPRSS4, along withBSG-mediated viral entry into cells, could imply a susceptible environment for SARSCoV-2 entry via different mechanisms. Additional studies are warranted to determine the true risk of endometrial infection by SARS-CoV2 and implications for fertility treatments. (Fertil Steril /C2102020;114:223-32. /C2112020 by American Society for Reproductive Medicine.) El resumen está disponible en Español alfinal del artículo. Key Words: ACE2, coronavirus, COVID-19, endometrial transcriptomics, SARS-CoV-2',\n",
       " 'Introduction': 'T he rapid international spread of novel coronavirus disease 2019 (COVID-19) has resulted in 5,103,006 con firmed cases and 333,401deaths worldwide as of May 23, 2020 . COVID-19 is caused by SARS-coronavirus 2 (SARS-CoV-2) infection in the lower respiratory tract , but the mechanisms underlying infection remain poorly understood . The rapid spread of SARS-CoV-2, genetically closely related to severe acute respiratory syndrome coronavirus (SARS-CoV-1) , has resulted in a global health emergency with longterm consequences on economics markets, nutritional habits, and mental and physical well-being . As information is still emerging about the health consequences of COVID-19, assisted reproductive treatments (ARTs) have been delayed due to fear of the unknown impact of SARS-Cov-2 on fertility . Fertility is compromised by age, and the longer treatments are delayed, the less likely that successful outcomes will be achieved . Furthermore, effects of SARS-CoV-2 infection increase in severity with host age , meaning that women of more advanced reproductive age undergoing ART could be at higher risk of infection. It is therefore critical to determine exactly how the virus affects reproductive physiology. Although the studies to date have been discordant, vertical transmission during pregnancy seems to occur infrequently, ranging from 0 to 11% . To date there is no information about how infection affects early implantation and development. The ability of SARS-CoV-2 to damage tissue is determined by its capacity to enter and infect cells in that tissue . The SARS-CoV-2 entry point on the cell is angiotensinconverting enzyme 2 (ACE2) , which plays a key role in the renin-angiotensin system, cleaving angiotensin II to angiotensin 1-7. The system is disrupted after SARS-CoV-2 gains cell entry and down-regulates the expression of ACE2, leading to up-regulation of the proin flammatory response by angiotensin II . ACE2 exhibits moderately increased expression with age , which could explain disease severity in older people. Another path of entry, using Basigin (BSG) as receptor instead of ACE2, has been proposed . To enter the cell, SARS-CoV-2 uses its spike S protein to bind to ACE2, which leads to fusion with the cell membrane and endocytosis . TMPRSS2, a transmembrane protease, cleaves the S protein . Cleavage is necessary for the virus to bind to ACE2 and spread through the infected host . Other proteases are under investigation as possible implications in SARS-CoV-2 infectivity related to S protein cleaving. TMPRSS4 increased virus infectivity on its own, at least in gut epithelial cells , while cathepsins B and L (CTSB and CTSL, respectively) had residual cleaving activity of viral S protein in TMPRSS2- cells . FURIN, another protease predicted to cleave S protein, presents alongside ACE2 in epithelial layers of several oral mucosal tissues . MX dynamin-like GTPase 1 (MX1) regulates neutrophil infiltration, favoring infection through protein S modification by neutrophil elastase . The clinical presentation of COVID-19 ranges from mild respiratory symptoms to severe progressive pneumonia, gastrointestinal symptoms, fecal shedding, multiorgan failure, and even death , but few studies have focused on the virus’s effect on fertility and damage to reproductive tissues, or on concerns regarding the use of reproductive treatments. Leydig and Sertoli cells in the testis , oocytes , and ovarian tissue are likely to experience damage due to their medium-high expression of the ACE2 receptor. The endometrium is crucial for human reproduction and embryo implantation, but studies of the effect of SARSCoV-2 infection on menstrual cycle progression have not been performed. Delineating the virus’s impact on the tissue is important for determining risk to ART, given that a healthy endometrium is needed for embryo implantation and growth. The endometrium is a complex tissue subjected to a cycle of cell death and renewal approximately every 28 days . Numerous transcriptomic studies have sought to understand gene expression changes throughout the menstrual cycle , and most of these data sets are available in public repositories, such as the Gene Expression Omnibus (GEO) database . According to the Human Protein Atlas (HPA) , ACE transcript is in low abundance in endometrium and not present as protein. According to HPA expression levels, TMPRSS4 and FURIN RNA expression is low, and protein levels are medium, whereasCTSB, MX1, andBSG have medium RNA expression and high protein expression . However, there is little information on how the virus could affect endometrial receptivity and embryo implantation. We analyzed the impact of SARS-CoV-2 infection on the endometrium by measuring endometrial ACE2, TMPRSS2, TMPRSS4, CTSB, CTSL, FURIN, MX1,a n dBSG gene expression. Transcriptomic data sets available across the phases of endometrial progression were used to evaluate molecularly the risk of SARS-CoV-2 infection during the COVID-19 pandemic.',\n",
       " 'Methods': 'MATERIALS AND METHODS Search and selection of SARS-CoV-2 infectivityrelated proteins A thorough literature search identified proteins related to the SARS-CoV-2 disease-causing mechanism, including genes related to cell entry and genes with implications in health and fertility, among others. The search for relevant genes associated with viral infectivity was made chronologically up to May 10, 2020. The keywords searched in PubMed included all possible combinations between‘‘SARS-CoV-2,’’ ‘‘COVID-19,’’ ‘‘coronavirus,’’ ‘‘cell entry mechanisms,’’ ‘‘long-term implications,’’and ‘‘fertility.’’ Search and selection of endometrial transcriptomic data sets Public transcriptomic data sets were used to analyze expression of SARS-CoV-2 infectivity-related genes throughout the menstrual cycle. Endometrial transcriptomic experiments for control patients (without any known endometrial pathology) were systematically searched in the GEO database with no restrictions on publication date or language and according to preferred reporting items for systematic reviews and meta-analyses (PRISMA) guidelines . Keywords were: uter* OR endometr*,filtered by‘‘homo sapiens.’’Experiments were selected if /C15 RNA was extracted directly from human endometrial biopsies. /C15 Endometrial biopsies were collected at different times during the menstrual cycle. /C15 Cycle phase at the time of biopsy was available for all samples. /C15 Endometrial gene expression was evaluated by microarray or RNA sequencing using Affymetrix, Illumina, or Agilent platforms. /C15 Raw gene expression data were freely available to download from GEO. Preprocessing and integrative analysis Transcriptomic raw data were downloaded from the GEO database and processed according to the standards of the technology used (microarrays or RNA-seq) using limma R-package (version 3.34.9) . Expression data from each 224 VOL. 114 NO. 2 / AUGUST 2020 SEMINAL CONTRIBUTIONS experiment were annotated with biomaRt (version 2.30.0) , log transformed, and quantile normalized using limma . Principal components analysis (PCA) was done using the prcomp() function, and the scores were displayed using the ggplot2 R-package to look for possible batch effects and were corrected if using linear models. Outliers were deleted from posterior analysis if there was a distinct transcriptional behavior according to PCA. To integrate the included endometrial transcriptomic experiments, several steps were followed as recommended by Tajti et al. : after being independently normalized, selected studies were joined in a unique data set, and batch effects were corrected using linear models (limma R package) . A relative expression value of low, medium, and high expression were established. The thresholds respectively correspond to 1% to 10%, 11% to 50%, and 51% to 100% categories of gene expression values of the entire integrated data set. Pairwise differential expression analysis between experiments was performed using limma to identify genes with expression differences.P values were corrected using false discovery rates (FDR) , and genes differentially expressed between experiments were removed from the analysis (FDR<.05). Differential gene expression throughout the menstrual cycle An analysis of variance (ANOVA) was performed for each selected gene related to SARS-CoV-2 infectivity to assess which genes showed statistically significant differences between endometrial phases. Analysis of variance was followed by a pairwiset-test to determine statistically significant differences between phases. Fold changes in gene expression from phase to phase were also calculated. The mean expression and confidence intervals for each gene in each phase were represented using ggplot2 R-package (version 3.0.0) . Coexpression of infectivity genes and the effect of age Pearson’s correlation was used to assess coexpression between each pair of selected proteins related to SARS-CoV-2 infectivity and to analyze the effect of age (from 23 to 50 years) on the expression of each protein. All programming and statistical tests (t-test, Pearson’s correlation, and analysis of variance) were implemented in the R environment (version 3.4.4, 2018-03-15) .',\n",
       " 'Results': 'RESULTS Menstrual cycle clock in integrated endometrial data sets From a total of 694 studies retrieved from GEO, our search identified five unique studies that evaluated endometrial gene expression in women with normal endometrium, comprising 112 samples (Table 1; see Supplemental Fig. 1, available online, for detailed filtering steps). All studies were analyzed separately for correction of possible batch effects, and three samples were excluded from analysis (details in Supplemental Fig. 2, available online). A unique combined data set with a population of 109 patients was obtained, comprising 29 samples in the proliferative phase, 29 in the early secretory phase, 43 in the medium secretory phase, and eight in the late secretory phase (Table 1). The 109 samples were grouped depending on the experiment rather than by the menstrual cycle phase (Fig. 1A, left). To make data sets comparable after the integration of all samples, this batch effect was removed (Fig. 1A, right). The resulting gene expression samples showed a behavior based on menstrual cycle phases rather than study nature (Fig. 1B). Viral infectivity genes show a different expression landscape across the menstrual cycle The gene expression landscapes for each viral gene across the cycle are shown inFigure 2B. TMPRSS4, CTSL, CTSB, FURIN, MX1, and BSG were highly expressed throughout the cycle; TMPRSS2 was moderately expressed, andACE2 was low (Fig. 2A). Gene expression of viral proteins depended on menstrual cycle phase. CalculatedP values and adjustedP values are presented inSupplemental Fig. 3A(available online). All genes exceptTMPRSS2 (P1⁄4.053) had statistically significant changes in expression across the menstrual cycle (Fig. 2A). Specifically, the genes most affected by menstrual cycle progression (P<.0001) were ACE2, which increased expression from early secretory to midsecretory;TMPRSS4, whose expression increased from proliferative to midsecretory and from early secretory to midsecretory (P<.0001); CTSL and CTSB, which increased from early secretory to midsecretory (P<.0001); TMPRSS2 with decreased expression from proliferative to early secretory (P<.05); and BSG and FURIN, which increased from proliferative to midsecretory (P<.01) (Fig. 2A). All genes, including TMPRSS2, showed increased expression from early secretory to midsecretory, as indicated inSupplemental Figure 3B. Detailed expression changes across the menstrual cycle are provided for each gene inSupplemental Figure 4(available online). Correlations between genes showed activations and repressions between them throughout the menstrual cycle (Supplemental Table 1, available online).ACE2 and TMPRSS4 were positively correlated in the early secretory phase (0.58). A weak correlation (activation) was found for ACE2 with TMPRSS4 and CTSL in the window of implantation (0.17 and 0.25, respectively). Coactivations were also detected in the proliferative phase betweenCTSB and TMPRSS4; the early secretory phase betweenFURIN and BSG; the midsecretory phase between CTSB with CTSL,a n dMX1 with CTSB, CTSL,a n d TMPRSS4; and the late secretory phase betweenTMPRSS4 and CTSB, FURIN, CTSB,a n dCTSL. Based on our results, high coactivation values were prioritized to build a molecular scheme of SARS-CoV-2 plausible infection of the endometrium based on two mechanisms of entry: one viaACE2 receptor and the other via BSG receptor. This scheme is shown inFigure 2B. In this model,TMPRSS4, which is highly expressed in all phases of the menstrual cycle, especially during the window of implantation, is the protease responsible for protein S cleaving, giving SARS-CoV-2 the capacity to bind toACE2 and infect cells even ifACE2 is lowly expressed. The model adds CTSL, CTSB, and FURIN as novel actors in the SARSCoV-2 mechanism of cell entry. These proteases are highly VOL. 114 NO. 2 / AUGUST 2020 225 Fertility and Sterility® expressed and coactivate with TMPRSS4. They could also help with the cleaving of protein S in different sites, thereby increasing infectivity. We also includedMX1, which regulates neutrophil activity and is highly expressed, bringing neutrophils into the tissue with elastases to help cleave S protein. Finally, BSG, the alternative receptor for SARS-CoV-2, also showed high expression, which would favor infectivity in this model, and activation withFURIN, which would favor infectivity in this model. Age influences the expression of viral genes throughout the menstrual cycle For this analysis, only samples from the study Talbi et al. were used because it was the only study that provided the age for each sample (in a range of 23-50 years old; n1⁄4 27). ACE2 showed increased expression with age in the proliferative, early secretory, and midsecretory phases (0.4, 0.73, 0.44, respectively; Fig. 3A). Most correlations by age were stronger in early phases of the menstrual cycle (proliferative and early secretory) than in subsequent endometrial stages (Fig. 3B). TMPRSS4 and MX1 were highly correlated with age in the early secretory phase;TMPRSS4, CTSL, andCTSB had positive correlation in the proliferative phase; andTMPRSS4 and BSG had positive correlation in the late secretory phase. In addition, high negative correlations were detected forBSG in the early secretory phase and forFURIN in the midsecretory phase.',\n",
       " 'Discussion': 'DISCUSSION It is important to understand the consequences of SARS-CoV-2 infection as the virus continues to spread worldwide. Fertility, specifically embryo implantation, is impacted by changes in endometrial gene expression throughout the menstrual cycle. Between menstrual cycle phases, gene expression varies to accompany physiologic changes . The differences in gene expression from one phase to another create a changing landscape for viral infection, and viral proteins themselves alter their expression throughout the menstrual cycle, setting the stage for a cyclical landscape of viral infectivity risk. ACE2, the receptor far for SARS-CoV-2 cell entry, showed low expression in the endometrium in our study, as reported previously in the Human Protein Atlas (HPA) . ACE2 is also reduced in critical tissues such as lungs ; TABLE 1 Characterization of endometrial transcriptomic data sets with GEO data sets identifier, experiment name given for this study, cycle type, method of cycle phase dating, population from which samples were collected, age, transcriptomic platform used to measure gene expression, number of genes of each data set, number of samples in each data set, number of samples per cycle phase, and publication in which data were initially employed. GEO ID Experiment name Cycle type Cycle phase dating method Population Age Platform No. genes No. samples No. samples per cycle phase Reference GSE4888 Talbi 2006 Normoovulatory; regular (24-35 d); 3 mo since last hormone treatment Noyes et al. reviewed by four pathologists Caucasian (n 1⁄4 17); black (n 1⁄4 6); Asian (n 1⁄4 1); other (n 1⁄4 2) 23-50 hgu133plus2 Affymetrix 19,361 27 PF (n 1⁄4 6); ESE (n1⁄4 4); MSE (n1⁄4 9); LSE (n1⁄4 8) GSE29981 Bradley 2010 Regular Days from LH peak: PF (LH-14-LH-1), ESE (LHþ1-LHþ4), MSE (LHþ6-LHþ7) Collected in Belgium 20-39 hgu133plus2 Affymetrix 19,361 19 PF (n 1⁄4 10); ESE (n1⁄4 6); MSE (n1⁄4 3) - GSE98386 Altm €ae 2017 Natural cycle Days from LH peak. We classified it in ESE1⁄4 LHþ2; MSE 1⁄4 LHþ8 Collected in Estonia - Illumina HiSeq 2500 16,426 38 ESE (n 1⁄4 19); MSE (n 1⁄4 19) GSE86491 Sigurgeirsson 2017 Regular; 3 mo since last hormone treatment Urinary LH ovulation predictor kit for MSE-LSE, days after the start of the subsequent menstruation for PF. Both confirmed by a gynecologic pathologist through histopathologic examination. Collected in Iceland 24-30 Illumina HiSeq 2500 15,939 14 PF (n 1⁄4 7); MSE (n 1⁄4 7) GSE119209 Kelleher 2017 -- Collected in U.S. - Illumina HiSeq 2500 17,934 11 PF (n 1⁄4 6); MSE (n 1⁄4 5) - Note: Last row indicates the total number of samples accounted and genes in common between all data sets for all menstrual cycle phases . ESE1⁄4 early secretory; GEO1⁄4 Gene Expression Omnibus; LH1⁄4 luteinizing hormone; LSE1⁄4 late secretory; MSE1⁄4 medium secretory; PF1⁄4 proliferative phase. Henarejos-Castillo. COVID-19 effects on the endometrium. Fertil Steril 2020. 226 VOL. 114 NO. 2 / AUGUST 2020 SEMINAL CONTRIBUTIONS however, we cannot suggest that low levels ofACE2 expression imply no effect of the virus on the tissue. Further studies are needed to elucidate the effect of decreasedACE2 in the endometrium. It is interesting that ACE2 increased (fold change 1⁄4 2.47) from the early secretory to midsecretory phases, implying an increase inACE2 in the window of implantation and a high risk of viral infectivity at this stage of the menstrual cycle. Several possible mechanisms of SARS-CoV-2 cell entry have been proposed . TMPRSS2, the most-reported protease involved with SARS-CoV-2 infectivity alongsideACE2, had medium endometrial expression in our study. However, there was no correlation betweenTMPRSS2, ACE2, and the rest of the genes studied. These results imply that the endometrium should be safe against SARS-CoV-2 infectivity mediated by TMPRSS2, though the expression of other proteases associated with S protein cleavage show a different landscape than TMPRSS2. TMPRSS4 statistically significantly changes its expression throughout the menstrual cycle, showing a statistically significant increase in the midsecretory phase. This protein also showed an interesting correlation with other genes, including an increase alongsideACE2 in the early secretory phase. Although TMPRSS2 is implicated in cell entry, the fact that S proteins could be targeted byTMPRSS4 and that TMPRSS4 increased infectivity in gut epithelial cells could mean a vulnerability of the endometrium to infection by SARS-CoV-2 mediated by TMPRSS4. Furthermore, TMPRSS4 up-regulation was correlated withCTSL and CTSB in all phases except the early secretory phase. Cell infection by SARS-CoV-2 was observed inTMRPSS2- cell lines expressing bothCTSL and CTSB, so these proteins may have a residual function of cleaving viral S protein . We propose that a synergy of CTSL and CTSB with TMPRSS4 through most of the cycle favors SARS-CoV-2 FIGURE 1 Endometrial data set integration and menstrual cycle clock. (A) Data set experiment effect and correction. Each point of the principal component analysis (PCA) plots represents endometrial gene expression of one sample and is colored by the endometrial transcriptomic data sets to which it belongs. Principal component 1 (PC1) and principal component 2 (PC2) explain the percentage of variability due to these components for each PCA. Integration of the transcriptomic studies showed a clear batch effect by the nature of each experiment (left PCA plot). After correction (right PCA plot), all genes in common between data sets were retained, amounting to a total of 13,437 genes, as no differentially expressed genes were detected between experiments. (B) Menstrual cycle effect. The samples of the integrated data sets are colored by the menstrual cycle phase to show how they are grouped by phases of the menstrual cycle. ESE1⁄4 early secretory endometrium; LSE1⁄4 late secretory endometrium; MSE1⁄4 midsecretory endometrium; PF1⁄4 proliferative phase. Henarejos-Castillo. COVID-19 effects on the endometrium. Fertil Steril 2020. VOL. 114 NO. 2 / AUGUST 2020 227 Fertility and Sterility® FIGURE 2 Gene expression of viral infection-related genes throughout the menstrual cycle. (A) Landscape of expression changes. Genes were located depending on their relative expression against the whole set. Low, medium, and high expression thresholds correspond to 1% to 10%, 11% to 50%, and 51% to 100% categories of gene expression values of the entire integrated data set, respectively. Analysis of variance results for overall change of expressionduring cycle are shown for each gene. *P<.05; ***P<.0001. (B) Molecular scheme of SARS-CoV-2 endometrial infection.ACE2, TMPRSS4, FURIN,a n dBSGare shown in plasma membrane of an endometrial cell (lower leftfigure).CTSLandCTSBare represented outside the cell.MX1is shown in cytoplasm. Expression of viral genes in comparison to whole transcriptomic setis represented as arrows next to their names: up1⁄4 highly expressed; down1⁄4 lowly expressed. Viral genes are positioned in their schematized cell locations as stablished by GeneCards database>Localization section (release 4.14) . Only maximum confidence levels (5 and 4) for compartments-derived cell locations were used. Proteins were grouped considering the highest coactivation values between pairso fv i r a l genes during the menstrual cycle, which are shown in the lower right table in thefigure. Discontinuous arrow shows less evidence according to our results, given that onlyFURINshowed high activation withBSG and that further studies are needed to understandBSG-related mechanisms of SARS-CoV-2 entry. ESE 1⁄4 early secretory endometrium; LSE1⁄4 late secretory endometrium; MSE1⁄4 midsecretory endometrium; PF1⁄4 proliferative phase. Henarejos-Castillo. COVID-19 effects on the endometrium. Fertil Steril 2020. 228 VOL. 114 NO. 2 / AUGUST 2020 SEMINAL CONTRIBUTIONS infectivity in the endometrium.FURIN, another protease predicted to cleave protein S , also showed a positive correlation with TMPRSS4 in the late secretory phase; with both targeting S protein, higher infectivity may occur. Finally, TMPRSS4 was positively correlated with MX1 in the midsecretory phase.MX1 actively participates in SARSCoV-2 infection by attracting neutrophils to infected tissue and employing their elastases to cleave protein S along with TMPRSS2 ; if TMPRSS4 participates alongside MX1,a more severe infection could occur during the window of implantation. TMPRSS4, CTSB, CTSL, FURIN, andMX1 showed statistically significant expression changes throughout the cycle. Protein expression of TMPRSS4, CTSB, FURIN, and MX1 supports a susceptible environment for infection in the early secretory and midsecretory phases. Future studies are needed to fully understand the mechanism of SARS-CoV2 cell entry and the roles ofTMPRSS4, CTSB, CTSL, FURIN, and MX1 in the endometrium. BSG, the alternative receptor for SARS-CoV-2 entry , showed high expression with statistically significant changes between phases and strong activation withFURIN through most of the cycle; however, its repression was correlated with ACE2 and TMPRSS2. Our results may indicate that only FURIN could be involved in S viral protein cleavage, if necessary, for SARS-CoV-2 binding toBSG, though other unknown proteases could also be involved. Further studies are needed to provide insight onBSG function in SARS-CoV-2 infection in the endometrium. We also investigated the effect of age on viral gene expression throughout the menstrual cycle. In a recent study FIGURE 3 Impact of age on viral-related infectivity gene expression throughout the menstrual cycle. (A) Effect of age onACE2 expression. Gene expression is represented forACE2 in each phase of the cycle according to the age of the sample analyzed. The range of age from patients involved in this study was 23 to 50 years. (B) Effect of age on viral gene expression. Pearson correlation R2 values are shown for each gene studied through of the phases of the menstrual cycle. Gray scale represents the magnitude of the correlation of increase or decrease in expression with age. High values are colored darker, and low values are colored lighter. ESE1⁄4 early secretory endometrium; LSE1⁄4 late secretory endometrium; MSE1⁄4 midsecretory endometrium; PF1⁄4 proliferative phase. Henarejos-Castillo. COVID-19 effects on the endometrium. Fertil Steril 2020. VOL. 114 NO. 2 / AUGUST 2020 229 Fertility and Sterility® in lungs, ACE2 expression was detected as highly variable among individuals ranging from 0.17 transcripts per million to 14.5 transcripts per million in the same demographic group and highlighted a skewness of expression that statistically significantly increased with age . Our results showed a positive correlation between age andACE2 from the proliferative to midsecretory phases, especially in the early secretory phase, meaning that the endometrium in older women (to age 50) could be more susceptible to SARS-CoV-2 infection. TMPRSS2 decreased with age, particularly in the midsecretory and late secretory phases, which may mean that cellmediated entry of the virus is lowered with increased age. However, TMPRSS4 increased in expression from the proliferative to early secretory phases similar toCTSB, CTSL, and MX1. These results, in conjunction with ACE2 expression changes, highlighted the same behavior related to age as in the lungs , implying a higher risk of endometrial infection in older women, especially in the early phases of the menstrual cycle, as well as potential implantation problems. However, our results are limited to only one study with 27 patients, and there are no prospective studies showing the real clinical implications. ACE2 plays an important role in endometrial changes because it cleaves angiotensin II and impacts spiral artery vasoconstriction, posterior proliferation, and renewal of tissue . ACE2 down-regulation by SARS-CoV-2 could alter the endometrial balance ; however, given that the landscape of the tissue is modified with each menstrual cycle renewal, the consequences of SARS-CoV-2 infection could be reset from cycle to cycle. Because our study integrates data sets from GEO, this approach is limited by the study design. This also means that results depend on the experimental transcriptomic procedures of prior reported results and the sample cohort included in each study. However, the inclusion criteria for the selected studies were very careful, and uniform and raw data were preprocessed to control undesired effects. Likewise, the virus differentially affects individuals due to each person’s genetic profile . Further prospective research is needed to determine the mechanisms of SARS-CoV-2 infection in the endometrium and how it could affect the fertility of each patient.',\n",
       " 'Conclusion': 'CONCLUSION Although TMPRSS2 expression implies a safe environment against infection, the roles of proteases and neutrophil regulating proteins TMPRSS4, CTSB, CTSL, FURIN, and MX1 could imply susceptibility of infection, especially during the early secretory and midsecretory phases. Additionally, the high expression ofBSG could imply an alternative mechanism independent of ACE2 for endometrial viral infection. Our findings show that viral gene expression increases with age, suggesting that the endometrium of older women undergoing ART is at higher risk of viral infection. A careful approach is advised, with precautions implemented when resuming ART.',\n",
       " 'URL': 'https://doi.org/10.1016/j.fertnstert.2020.06.026',\n",
       " 'Journal': 'Fertility and Sterility®'}"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "my_json[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "id": "a92a7cfe",
   "metadata": {},
   "outputs": [],
   "source": [
    "info_paper = []\n",
    "\n",
    "for j in my_json:\n",
    "\n",
    "    for key, value in j.items():\n",
    "\n",
    "        if len(value) > 1200:\n",
    "            chunks = splitter.split_text(value)\n",
    "\n",
    "            for i, c in enumerate(chunks):\n",
    "\n",
    "                info_paper.append(\n",
    "                    {\n",
    "                        \"chunk_index\":i,\n",
    "                        \"content\":c,\n",
    "                        \"parent\":key,\n",
    "                        \"split\":True,\n",
    "                        \"DOI\":j.get(\"URL\"),\n",
    "                        \"Reference\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Journal')+\", \"+j.get('Publication')\n",
    "                    }\n",
    "                )\n",
    "        else:\n",
    "\n",
    "            info_paper.append(\n",
    "                    {\n",
    "                        \"chunk_index\":0,\n",
    "                        \"content\":value,\n",
    "                        \"parent\":key,\n",
    "                        \"split\":False,\n",
    "                        \"DOI\":j.get(\"URL\"),\n",
    "                        \"Reference\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Journal')+\", \"+j.get('Publication')\n",
    "                    }\n",
    "                )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2efed7d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# documents = [\n",
    "#     Document(\n",
    "#         page_content=chunk[\"content\"],\n",
    "#         metadata={\n",
    "#             \"parent\": chunk[\"parent\"],\n",
    "#             \"chunk_index\": chunk[\"chunk_index\"],\n",
    "#             \"DOI\":chunk[\"DOI\"],\n",
    "#             \"Reference\":chunk[\"Reference\"]\n",
    "            \n",
    "#         }\n",
    "#     )\n",
    "#     for chunk in info_paper\n",
    "# ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "7ada8805",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1576/401118669.py:1: LangChainDeprecationWarning: The class `HuggingFaceEmbeddings` was deprecated in LangChain 0.2.2 and will be removed in 1.0. An updated version of the class exists in the :class:`~langchain-huggingface package and should be used instead. To use it run `pip install -U :class:`~langchain-huggingface` and import as `from :class:`~langchain_huggingface import HuggingFaceEmbeddings``.\n",
      "  embedding_function = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")\n",
      "/home/antonioparragaleo/miniconda3/envs/RAG/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "embedding_function = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")\n",
    "# client = chromadb.HttpClient(host='localhost', port=7000, settings=Settings(allow_reset=True))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "c4a50048",
   "metadata": {},
   "outputs": [],
   "source": [
    "client = chromadb.PersistentClient(path=\"./chroma_RAG\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "813ab96d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#client.delete_collection(name=\"prueba2\")\n",
    "collection = client.get_or_create_collection(\n",
    "    name=\"prueba2\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "39bcb19c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 2. Prepare documents, metadata, and IDs\n",
    "texts = [chunk[\"content\"] for chunk in info_paper]\n",
    "metadatas = [{\"parent\": chunk[\"parent\"], \"chunk_index\": chunk[\"chunk_index\"],\"DOI\": chunk[\"DOI\"], \"Reference\": chunk[\"Reference\"]} for chunk in info_paper]\n",
    "ids = [str(uuid.uuid1()) for _ in metadatas]\n",
    "#ids = [f'{md[\"parent\"]}_{md[\"chunk_index\"]}' for md in metadatas]  # unique ID per chunk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "eedc1a1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "collection.add(\n",
    "    documents=texts,\n",
    "    metadatas=metadatas,\n",
    "    ids=ids\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "b77d6130",
   "metadata": {},
   "outputs": [],
   "source": [
    "# tell LangChain to use our client and collection name\n",
    "db = Chroma(\n",
    "    client=client,\n",
    "    collection_name=\"prueba\",\n",
    "    embedding_function=embedding_function,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "bb456fa9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('A key factor for reproductive success in assisted reproduction treatments is '\n",
      " 'the status of the maternal endometrium during embryo implantation and fetal '\n",
      " 'development. The endometrial cycle is reflected by the cyclic structural and '\n",
      " 'functional changes of the endometrium across the menstrual cycle, '\n",
      " 'particularly during the mid-secretory phase, to prepare for the‘‘window of '\n",
      " 'implantation’’ (WOI) . After successful embryo implantation, the decidua '\n",
      " '(specialized layer of the endometrium) actively encapsulates the '\n",
      " 'trophectoderm to support placentation and provide adequate vascularization '\n",
      " 'for optimal fetal growth . Thus, approaches that predict and prevent '\n",
      " 'endometrial-factor infertility could substantially improve assisted '\n",
      " 'reproduction treatment outcomes by supporting the establishment and '\n",
      " 'maintenance of pregnancy. There is a lack of consensus on the minimum number '\n",
      " 'of implantation failures with good-quality embryos derived from ovum '\n",
      " 'donation or with confirmed euploid karyotypes required to diagnose recurrent '\n",
      " 'implantation failure (RIF) . The disparities in clinical classification and '\n",
      " 'heterogeneous etiology of RIF confound its diagnosis and leave the exact '\n",
      " 'prevalence unclear . As clinical symptoms cannot be prevented and do not '\n",
      " 'provide sufficient criteria to stratify patients, patients might be '\n",
      " 'misclassi fied and receive inefficient treatment. Indeed, most studies using '\n",
      " 'clinical criteria for RIF identified no or few differentially expressed '\n",
      " 'genes between samples of affected patients , reflecting the molecular '\n",
      " 'heterogeneity of RIF. To overcome these healthcare challenges, '\n",
      " 'high-throughput molecular technologies are combined with artificial '\n",
      " 'intelligence (AI) algorithms to provide a deeper understanding of '\n",
      " 'infertilityrelated pathogenesis. These molecular insights are then applied '\n",
      " 'in genomics-driven precision medicine approaches to characterize complex '\n",
      " 'diseases, and ultimately, improve patient diagnosis and prognosis . '\n",
      " 'Employing genomics in reproductive medicine has the potential to go beyond '\n",
      " 'treating the clinical manifestations of disease progression , by leveraging '\n",
      " 'the patients’ heterogeneous molecular profiles for precise stratification '\n",
      " 'and revealing new targets for preventive measures . In this regard, current '\n",
      " 'diagnostic tools for endometrialfactor infertility use transcriptomics to '\n",
      " 'date the endometrial biopsy in the mid-secretory phase, exceeding the '\n",
      " 'accuracy and objectivity of traditional endometrial progression staging by '\n",
      " 'Noyes’ histological criteria . However, the clinical utility of '\n",
      " 'synchronizing the endometrium by adjusting the timing of progesterone (P4) '\n",
      " 'administration on the basis of the patient’s transcriptomic endometrial '\n",
      " 'dating profile remains controversial. Independent randomized clinical trials '\n",
      " 'found no improvement in reproductive outcomes , indicating the inefficacy of '\n",
      " 'this clinical intervention and highlighting the need for alternative '\n",
      " 'treatments and/or a refined target patient population that bene fits from '\n",
      " 'this strategy. Combining these precision medicine approaches with '\n",
      " 'transcriptomics and AI algorithms; our group recently proposed a novel '\n",
      " 'molecular taxonomy of RIF and a new clinical algorithm for '\n",
      " 'endometrial-factor infertility that considered 2 molecular causes of '\n",
      " 'implantation failure: a WOI displacement that indicates a dysregulated '\n",
      " 'endometrial timing; and a WOI disruption that indicates an endometrial '\n",
      " 'alteration independent of endometrial timing . Transcriptomic analysis '\n",
      " 'revealed these 2 WOI phenotypes were presented exclusively or simultaneously '\n",
      " 'and patients with RIF had significantly more WOI disruptions but a similar '\n",
      " 'frequency of WOI displacements, compared with controls. Thesefindings '\n",
      " 'suggested that WOI disruptions had stronger associations with implantation '\n",
      " 'failure than WOI displacements and shifted the paradigm for the study and '\n",
      " 'treatment of endometrialfactor infertility . In addition, this work by '\n",
      " 'SebastianLeon et al. highlighted the importance of removing the '\n",
      " 'transcriptomic variation due to cyclical endometrial tissue changes (ie, the '\n",
      " 'effect of endometrial timing) that, if not corrected, could mask the '\n",
      " 'significant disruptions in endometrial gene expression. However, given this '\n",
      " 'hypothesis-driven research was conducted retrospectively and in silico with '\n",
      " 'publicly available data, the reproductive outcomes of the cycle after the '\n",
      " 'biopsy collection were not available . This clinical follow-up is required '\n",
      " 'to discover biomarkers for WOI disruption in patients with a higher risk of '\n",
      " 'endometrial failure. On the basis of these genomics-driven precision '\n",
      " 'medicine strategies, this proof-of-concept study aimed to discover a '\n",
      " 'biomarker panel that could identify endometrial disruption, independent of '\n",
      " 'endometrial timing, to predict, and ultimately prevent, poor reproductive '\n",
      " 'outcomes in in vitro fertilization (IVF) patients.')\n"
     ]
    }
   ],
   "source": [
    "a = collection.query(\n",
    "        query_texts=[query],\n",
    "        n_results=4\n",
    "    )\n",
    "parent = a['metadatas'][0][1].get('parent')\n",
    "ref = a['metadatas'][0][1].get('Reference')\n",
    "\n",
    "b = collection.get(\n",
    "    where={\n",
    "\n",
    "        \"$and\" :[\n",
    "             {\"Reference\":ref},\n",
    "             {\"parent\":parent}\n",
    "        ]\n",
    "        },\n",
    "    include=[\"documents\"]\n",
    "        \n",
    "    )\n",
    "\n",
    "pprint.pprint(\"\".join(b['documents']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "87a33f07",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Introduction_Patricia Diaz-Gimeno et al.,Fertil Steril®, 2024_1\n",
      "Methods_Patricia Diaz-Gimeno et al.,Fertil Steril®, 2024_2\n",
      "Results_Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023_8\n"
     ]
    }
   ],
   "source": [
    "for r in a['metadatas'][0]:\n",
    "     if r['parent'] not in ['Journal','URL']:\n",
    "        print( \"_\".join([r['parent'],r['Reference'],str(r['chunk_index']+1)]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "20db46ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[{'parent': 'Journal',\n",
       "   'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023',\n",
       "   'chunk_index': 0,\n",
       "   'DOI': 'https://doi.org/10.1186/s12958-023-01131-4'},\n",
       "  {'DOI': 'https://doi.org/10.1016/j.fertnstert.2024.03.015',\n",
       "   'chunk_index': 0,\n",
       "   'parent': 'Introduction',\n",
       "   'Reference': 'Patricia Diaz-Gimeno et al.,Fertil Steril®, 2024'},\n",
       "  {'Reference': 'Patricia Diaz-Gimeno et al.,Fertil Steril®, 2024',\n",
       "   'DOI': 'https://doi.org/10.1016/j.fertnstert.2024.03.015',\n",
       "   'parent': 'Methods',\n",
       "   'chunk_index': 1},\n",
       "  {'parent': 'Results',\n",
       "   'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023',\n",
       "   'DOI': 'https://doi.org/10.1186/s12958-023-01131-4',\n",
       "   'chunk_index': 7}]]"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a['metadatas']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "723f124a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['A key factor for reproductive success in assisted reproduction treatments is the status of the maternal endometrium during embryo implantation and fetal development. The endometrial cycle is reflected by the cyclic structural and functional changes of the endometrium across the menstrual cycle, particularly during the mid-secretory phase, to prepare for the‘‘window of implantation’’ (WOI) . After successful embryo implantation, the decidua (specialized layer of the endometrium) actively encapsulates the trophectoderm to support placentation and provide adequate vascularization for optimal fetal growth . Thus, approaches that predict and prevent endometrial-factor infertility could substantially improve assisted reproduction treatment outcomes by supporting the establishment and maintenance of pregnancy',\n",
       "  '. There is a lack of consensus on the minimum number of implantation failures with good-quality embryos derived from ovum donation or with confirmed euploid karyotypes required to diagnose recurrent implantation failure (RIF) . The disparities in clinical classification and heterogeneous etiology of RIF confound its diagnosis and leave the exact prevalence unclear . As clinical symptoms cannot be prevented and do not provide sufficient criteria to stratify patients, patients might be misclassi fied and receive inefficient treatment. Indeed, most studies using clinical criteria for RIF identified no or few differentially expressed genes between samples of affected patients , reflecting the molecular heterogeneity of RIF. To overcome these healthcare challenges, high-throughput molecular technologies are combined with artificial intelligence (AI) algorithms to provide a deeper understanding of infertilityrelated pathogenesis'],\n",
       " ['MATERIAL AND METHODS Ethics statement This study was approved by the Ethics Committee of the Instituto Valenciano de Infertilidad, Valencia, Spain (1810-FIVI-066-PD/1810-FIVI-048-PD). Written informed consent was obtained from all recruited patients (n1⁄4 281). Participants, collection of endometrial biopsies, and reproductive outcomes This prospective, multicentric study was conducted between July 2018 and July 2021 at 5 private fertility clinics in Spain. Participants (n1⁄4 281) were scheduled for endometrial evaluation due to medical indications orR2 implantation failures of idiopathic origin. All patients met the following inclusion criteria: 18- 50 years old, with a body mass index (BMI) of 19- 30 kg/m 2, undergoing hormone replacement therapy for single embryo transfer (SET), presenting an endometrial thickness >6',\n",
       "  '.5 mm with trilaminar structure in proliferative phase before P4 administration, and serum levels of estradiol >100 pg/mL and P4<1 ng/mL on the 10th day of estrogen treatment . Endometrial biopsies were collected after an average of 114.5/C6 7.2 h of P4 (duration ranged from 82 to 172 h), and processed as we previously described . Baseline patient characteristics including age, BMI, the number of IVF attempts, the quality of the embryo transferred (defined in theSupplemental Methods, available online), and reproductive outcomes (ie, pregnancy, live birth, and biochemical and clinical miscarriage) for each attempt, were obtained from internal medical records. Information about IVF attempts in other fertility clinics was included if recorded (only in 47.7% of patients)',\n",
       "  '.7% of patients). Study design First, a custom RNA-seq gene panel (the endometrial failure risk [EFR] panel) comprising 404 biomarkers of endometrial luteal phase timing and/or potential disruption was designed and employed to determine the risk of endometrial failure among 217 IVF patients (Supplemental Fig. 1, step 1 available online). The endometrial timing-corrected expression of the EFR panel was used to study the differential expression associated with endometrial disruption together with clinical characteristics of IVF patients to stratify the population as having poor or good endometrial prognoses, according to a semi-supervised learning process (Supplemental Fig. 1, step 2). The distinct transcriptomic patterns of each prognosis group were associated with the differences in reproductive outcomes of SET after biopsy collection, and the differentially expressed gene signature was proposed as a biomarker of endometrial disruption (Supplemental Fig. 1, step 3)'],\n",
       " ['. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53. Notably, GATA6 was selected due to novel association with endometrial progression and function. To infer the expression of CTCF and GATA6 in endometrial tissue throughout the menstrual cycle, we analysed their expression using a relevant endometrial dataset recently created by our group [30] (Table S1 ). We observed that, relative to the global expression of endometrial genes, CTCF was broadly expressed while GATA6 was moderately expressed. Indeed, CTCF was significantly down-regulated between the PF and ESE phases (FDR = 2.10E-03) and ESE to the MSE phase (FDR = 2.10E-04), while GATA6 expression increased from the PF to the MSE phase (FDR = 1.70E-11) and from the ESE to MSE phase (FDR = 9.80E-15)',\n",
       "  '.70E-11) and from the ESE to MSE phase (FDR = 9.80E-15). These findings supported the role of these TFs at the beginning of the menstrual cycle and during the MSE phase (Fig. 3 B). These in-silico validations were then prospectively corroborated by RT-qPCR analysis in an independent sample set of 20 endometrial biopsies. We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with significant changes between the endometrial phases. Notably, CTCF expression was decreased between the PF/ ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C).']]"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ideal_chunks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "3b4092f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[['A key factor for reproductive success in assisted reproduction treatments is the status of the maternal endometrium during embryo implantation and fetal development. The endometrial cycle is reflected by the cyclic structural and functional changes of the endometrium across the menstrual cycle, particularly during the mid-secretory phase, to prepare for the‘‘window of implantation’’ (WOI) . After successful embryo implantation, the decidua (specialized layer of the endometrium) actively encapsulates the trophectoderm to support placentation and provide adequate vascularization for optimal fetal growth . Thus, approaches that predict and prevent endometrial-factor infertility could substantially improve assisted reproduction treatment outcomes by supporting the establishment and maintenance of pregnancy', '. There is a lack of consensus on the minimum number of implantation failures with good-quality embryos derived from ovum donation or with confirmed euploid karyotypes required to diagnose recurrent implantation failure (RIF) . The disparities in clinical classification and heterogeneous etiology of RIF confound its diagnosis and leave the exact prevalence unclear . As clinical symptoms cannot be prevented and do not provide sufficient criteria to stratify patients, patients might be misclassi fied and receive inefficient treatment. Indeed, most studies using clinical criteria for RIF identified no or few differentially expressed genes between samples of affected patients , reflecting the molecular heterogeneity of RIF. To overcome these healthcare challenges, high-throughput molecular technologies are combined with artificial intelligence (AI) algorithms to provide a deeper understanding of infertilityrelated pathogenesis'], ['MATERIAL AND METHODS Ethics statement This study was approved by the Ethics Committee of the Instituto Valenciano de Infertilidad, Valencia, Spain (1810-FIVI-066-PD/1810-FIVI-048-PD). Written informed consent was obtained from all recruited patients (n1⁄4 281). Participants, collection of endometrial biopsies, and reproductive outcomes This prospective, multicentric study was conducted between July 2018 and July 2021 at 5 private fertility clinics in Spain. Participants (n1⁄4 281) were scheduled for endometrial evaluation due to medical indications orR2 implantation failures of idiopathic origin. All patients met the following inclusion criteria: 18- 50 years old, with a body mass index (BMI) of 19- 30 kg/m 2, undergoing hormone replacement therapy for single embryo transfer (SET), presenting an endometrial thickness >6', '.5 mm with trilaminar structure in proliferative phase before P4 administration, and serum levels of estradiol >100 pg/mL and P4<1 ng/mL on the 10th day of estrogen treatment . Endometrial biopsies were collected after an average of 114.5/C6 7.2 h of P4 (duration ranged from 82 to 172 h), and processed as we previously described . Baseline patient characteristics including age, BMI, the number of IVF attempts, the quality of the embryo transferred (defined in theSupplemental Methods, available online), and reproductive outcomes (ie, pregnancy, live birth, and biochemical and clinical miscarriage) for each attempt, were obtained from internal medical records. Information about IVF attempts in other fertility clinics was included if recorded (only in 47.7% of patients)', '.7% of patients). Study design First, a custom RNA-seq gene panel (the endometrial failure risk [EFR] panel) comprising 404 biomarkers of endometrial luteal phase timing and/or potential disruption was designed and employed to determine the risk of endometrial failure among 217 IVF patients (Supplemental Fig. 1, step 1 available online). The endometrial timing-corrected expression of the EFR panel was used to study the differential expression associated with endometrial disruption together with clinical characteristics of IVF patients to stratify the population as having poor or good endometrial prognoses, according to a semi-supervised learning process (Supplemental Fig. 1, step 2). The distinct transcriptomic patterns of each prognosis group were associated with the differences in reproductive outcomes of SET after biopsy collection, and the differentially expressed gene signature was proposed as a biomarker of endometrial disruption (Supplemental Fig. 1, step 3)'], ['. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53. Notably, GATA6 was selected due to novel association with endometrial progression and function. To infer the expression of CTCF and GATA6 in endometrial tissue throughout the menstrual cycle, we analysed their expression using a relevant endometrial dataset recently created by our group [30] (Table S1 ). We observed that, relative to the global expression of endometrial genes, CTCF was broadly expressed while GATA6 was moderately expressed. Indeed, CTCF was significantly down-regulated between the PF and ESE phases (FDR = 2.10E-03) and ESE to the MSE phase (FDR = 2.10E-04), while GATA6 expression increased from the PF to the MSE phase (FDR = 1.70E-11) and from the ESE to MSE phase (FDR = 9.80E-15)', '.70E-11) and from the ESE to MSE phase (FDR = 9.80E-15). These findings supported the role of these TFs at the beginning of the menstrual cycle and during the MSE phase (Fig. 3 B). These in-silico validations were then prospectively corroborated by RT-qPCR analysis in an independent sample set of 20 endometrial biopsies. We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with significant changes between the endometrial phases. Notably, CTCF expression was decreased between the PF/ ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C).']]\n",
      "['Reference:Patricia Diaz-Gimeno et al.,Fertil Steril®, 2024\\n\\nLink (DOI)\\n: https://doi.org/10.1016/j.fertnstert.2024.03.015\\n\\nSummary:\\n\\nA key factor for reproductive success in assisted reproduction treatments is the status of the maternal endometrium during embryo implantation and fetal development. The endometrial cycle is reflected by the cyclic structural and functional changes of the endometrium across the menstrual cycle, particularly during the mid-secretory phase, to prepare for the‘‘window of implantation’’ (WOI) . After successful embryo implantation, the decidua (specialized layer of the endometrium) actively encapsulates the trophectoderm to support placentation and provide adequate vascularization for optimal fetal growth . Thus, approaches that predict and prevent endometrial-factor infertility could substantially improve assisted reproduction treatment outcomes by supporting the establishment and maintenance of pregnancy. There is a lack of consensus on the minimum number of implantation failures with good-quality embryos derived from ovum donation or with confirmed euploid karyotypes required to diagnose recurrent implantation failure (RIF) . The disparities in clinical classification and heterogeneous etiology of RIF confound its diagnosis and leave the exact prevalence unclear . As clinical symptoms cannot be prevented and do not provide sufficient criteria to stratify patients, patients might be misclassi fied and receive inefficient treatment. Indeed, most studies using clinical criteria for RIF identified no or few differentially expressed genes between samples of affected patients , reflecting the molecular heterogeneity of RIF. To overcome these healthcare challenges, high-throughput molecular technologies are combined with artificial intelligence (AI) algorithms to provide a deeper understanding of infertilityrelated pathogenesis\\n\\n', 'Reference:Patricia Diaz-Gimeno et al.,Fertil Steril®, 2024\\n\\nLink (DOI)\\n: https://doi.org/10.1016/j.fertnstert.2024.03.015\\n\\nSummary:\\n\\nMATERIAL AND METHODS Ethics statement This study was approved by the Ethics Committee of the Instituto Valenciano de Infertilidad, Valencia, Spain (1810-FIVI-066-PD/1810-FIVI-048-PD). Written informed consent was obtained from all recruited patients (n1⁄4 281). Participants, collection of endometrial biopsies, and reproductive outcomes This prospective, multicentric study was conducted between July 2018 and July 2021 at 5 private fertility clinics in Spain. Participants (n1⁄4 281) were scheduled for endometrial evaluation due to medical indications orR2 implantation failures of idiopathic origin. All patients met the following inclusion criteria: 18- 50 years old, with a body mass index (BMI) of 19- 30 kg/m 2, undergoing hormone replacement therapy for single embryo transfer (SET), presenting an endometrial thickness >6.5 mm with trilaminar structure in proliferative phase before P4 administration, and serum levels of estradiol >100 pg/mL and P4<1 ng/mL on the 10th day of estrogen treatment . Endometrial biopsies were collected after an average of 114.5/C6 7.2 h of P4 (duration ranged from 82 to 172 h), and processed as we previously described . Baseline patient characteristics including age, BMI, the number of IVF attempts, the quality of the embryo transferred (defined in theSupplemental Methods, available online), and reproductive outcomes (ie, pregnancy, live birth, and biochemical and clinical miscarriage) for each attempt, were obtained from internal medical records. Information about IVF attempts in other fertility clinics was included if recorded (only in 47.7% of patients).7% of patients). Study design First, a custom RNA-seq gene panel (the endometrial failure risk [EFR] panel) comprising 404 biomarkers of endometrial luteal phase timing and/or potential disruption was designed and employed to determine the risk of endometrial failure among 217 IVF patients (Supplemental Fig. 1, step 1 available online). The endometrial timing-corrected expression of the EFR panel was used to study the differential expression associated with endometrial disruption together with clinical characteristics of IVF patients to stratify the population as having poor or good endometrial prognoses, according to a semi-supervised learning process (Supplemental Fig. 1, step 2). The distinct transcriptomic patterns of each prognosis group were associated with the differences in reproductive outcomes of SET after biopsy collection, and the differentially expressed gene signature was proposed as a biomarker of endometrial disruption (Supplemental Fig. 1, step 3)\\n\\n', 'Reference:Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023\\n\\nLink (DOI)\\n: https://doi.org/10.1186/s12958-023-01131-4\\n\\nSummary:\\n\\n. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53. Notably, GATA6 was selected due to novel association with endometrial progression and function. To infer the expression of CTCF and GATA6 in endometrial tissue throughout the menstrual cycle, we analysed their expression using a relevant endometrial dataset recently created by our group [30] (Table S1 ). We observed that, relative to the global expression of endometrial genes, CTCF was broadly expressed while GATA6 was moderately expressed. Indeed, CTCF was significantly down-regulated between the PF and ESE phases (FDR = 2.10E-03) and ESE to the MSE phase (FDR = 2.10E-04), while GATA6 expression increased from the PF to the MSE phase (FDR = 1.70E-11) and from the ESE to MSE phase (FDR = 9.80E-15).70E-11) and from the ESE to MSE phase (FDR = 9.80E-15). These findings supported the role of these TFs at the beginning of the menstrual cycle and during the MSE phase (Fig. 3 B). These in-silico validations were then prospectively corroborated by RT-qPCR analysis in an independent sample set of 20 endometrial biopsies. We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with significant changes between the endometrial phases. Notably, CTCF expression was decreased between the PF/ ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C).\\n\\n']\n"
     ]
    }
   ],
   "source": [
    "results = db.similarity_search(query=query,k=4)\n",
    "\n",
    "selected_index = []\n",
    "ideal_chunks = []\n",
    "meta_selected = []\n",
    "\n",
    "for doc in results:\n",
    "\n",
    "    r = doc.metadata\n",
    "\n",
    "    if r['parent'] not in ['Journal','URL']:\n",
    "\n",
    "        if \"_\".join([r['parent'],r['Reference'],str(r['chunk_index']+1)]) not in selected_index and \"_\".join([r['parent'],r['Reference'],str(r['chunk_index']-1)]) not in selected_index:\n",
    "            \n",
    "            ii = \"_\".join([r['parent'],r['Reference'],str(r['chunk_index'])])\n",
    "            selected_index.append(ii)\n",
    "\n",
    "            candidates = collection.get(\n",
    "            where= {\"$and\" :[\n",
    "                        {\"Reference\":r['Reference']},\n",
    "                        {\"parent\":r['parent']}\n",
    "                ]\n",
    "            })\n",
    "\n",
    "            max_index = len(candidates['metadatas'])-1\n",
    "\n",
    "            meta_selected.append(candidates['metadatas'])\n",
    "            ideal_chunks.append([doc for doc,meta in zip(candidates['documents'], candidates['metadatas'])\n",
    "                        if meta['chunk_index'] in [r[\"chunk_index\"], max(r[\"chunk_index\"]-1,0), min(r[\"chunk_index\"] + 1,max_index)]])\n",
    "\n",
    "context = []\n",
    "print(ideal_chunks)\n",
    "\n",
    "for text, meta in zip(ideal_chunks, meta_selected):\n",
    "    context.append(f'Reference:{meta[0][\"Reference\"]}\\n\\nLink (DOI)\\n: {meta[0][\"DOI\"]}\\n\\nSummary:\\n\\n{\"\".join(text)}\\n\\n')\n",
    "\n",
    "print(context)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "dac7575b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "min(1,len(candidates['metadatas'])-1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "ee4c48eb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ids': ['bd725078-6b0e-11f0-8e08-00155de32492'],\n",
       " 'embeddings': None,\n",
       " 'documents': ['Results Gene signatures associated to endometrial progression and function Of the 19 gene lists used for our analysis, eleven were obtained from published studies evaluating control patients throughout the menstrual cycle [4, 17-19, 34-40], and the other eight were derived from studies comparing patients with RIF or unexplained infertility to controls [21, 41-47] (Table 1). Unifying all the aforementioned signatures, we compiled 3,608 genes related to endometrial progression and function. Hormonal regulation of endometrial progression is largely driven by progesterone We identified 7,540 and 698 genes related to estrogen and progesterone hormones, respectively. However, as determined by the relative contribution of each type of hormone within each gene list (Fig. 1A), 17/19 (89%) signatures favoured regulation by progesterone rather than estrogen. These differences were significant in 47% of the signatures (FDR < 0.05)'],\n",
       " 'uris': None,\n",
       " 'included': ['metadatas', 'documents'],\n",
       " 'data': None,\n",
       " 'metadatas': [{'parent': 'Results',\n",
       "   'DOI': 'https://doi.org/10.1186/s12958-023-01131-4',\n",
       "   'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023',\n",
       "   'chunk_index': 0}]}"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collection.get(\n",
    "    where={\"$and\":[\n",
    "           {\"Reference\":'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023'},\n",
    "           {\"parent\":\"Results\"},\n",
    "           {\"chunk_index\":0}\n",
    "    ]}\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0f49680a",
   "metadata": {},
   "outputs": [],
   "source": [
    "query = \"How many phases does the endoemtrium have?\"\n",
    "docs = db.similarity_search(query,k=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "f54dbad0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Document(metadata={'chunk_index': 7, 'parent': 'Discussion', 'DOI': 'https://doi.org/10.1186/s12958-023-01131-4', 'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023'}, page_content='. Considering these attributed functions and its variable expression throughout the menstrual cycle, we propose that CTCF exerts an inhibitory role in endometrial tissue during the PF phase of the menstrual cycle. With the significant downregulation of CTCF in the secretory endometrium, dually validated in-silico and experimentally herein, its inhibited genes would be derepressed and become transcriptionally active during the WOI. This interpretation supports previous findings from our group demonstrating that, during the WOI, a global transcriptional derepression may be required for implantation and early embryo development [63]. Transcriptional derepression has been associated with multiple human disease states and should be investigated further within the context of endometrial-factor infertility. Despite previous associations of CTCF with endometriosis [67], its implication in endometrial progression and function has not been directly proposed until now'),\n",
       " Document(metadata={'parent': 'Abstract', 'DOI': 'https://doi.org/10.1186/s12958-023-01131-4', 'chunk_index': 3, 'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023'}, page_content='. Keywords CTCF, GATA6, Progesterone, Estrogen, miRNAs, TFs, Endometrial receptivity, Recurrent implantation failure, Infertility, Menstrual cycle regulation'),\n",
       " Document(metadata={'parent': 'Results', 'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023', 'DOI': 'https://doi.org/10.1186/s12958-023-01131-4', 'chunk_index': 7}, page_content='. We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with significant changes between the endometrial phases. Notably, CTCF expression was decreased between the PF/ ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C).'),\n",
       " Document(metadata={'chunk_index': 5, 'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023', 'parent': 'Results', 'DOI': 'https://doi.org/10.1186/s12958-023-01131-4'}, page_content='. CTCF and GATA6 as distinguished transcriptional regulators of the human menstrual cycle We repeated the regulation network analysis with the TFs, to identify the prominent TFs commonly regulating the genes related to endometrial progression and function in the reported signatures. We observed 78.80% of the TFs regulated genes in up to 10 signatures, and only 99 TFs (21.20%) regulated genes in over 10 (Fig. 3A). We highlighted CTCF (a CCCTC-binding factor, which functions as a transcriptomic repressor) as the TF with the most influential regulation of endometrial progression and function across 95% of studies (18 gene lists, excluding that of Punyadeera, Fig. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53')]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "docs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "961a1fd8",
   "metadata": {},
   "outputs": [],
   "source": [
    "texts_re = []\n",
    "metadata_re = []\n",
    "for doc in docs:\n",
    "    texts_re.append(doc.page_content)\n",
    "    metadata_re.append(doc.metadata)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "2f3748d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'parent': 'Discussion',\n",
       "  'chunk_index': 7,\n",
       "  'DOI': 'https://doi.org/10.1186/s12958-023-01131-4'},\n",
       " {'DOI': 'https://doi.org/10.1186/s12958-023-01131-4',\n",
       "  'parent': 'Abstract',\n",
       "  'chunk_index': 3},\n",
       " {'parent': 'Results',\n",
       "  'DOI': 'https://doi.org/10.1186/s12958-023-01131-4',\n",
       "  'chunk_index': 7},\n",
       " {'parent': 'Results',\n",
       "  'chunk_index': 5,\n",
       "  'DOI': 'https://doi.org/10.1186/s12958-023-01131-4'}]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "4390b620",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The reference of article is https://doi.org/10.1186/s12958-023-01131-4, its information is:\n",
      " . Considering these attributed functions and its variable expression throughout the menstrual cycle, we propose that CTCF exerts an inhibitory role in endometrial tissue during the PF phase of the menstrual cycle. With the significant downregulation of CTCF in the secretory endometrium, dually validated in-silico and experimentally herein, its inhibited genes would be derepressed and become transcriptionally active during the WOI. This interpretation supports previous findings from our group demonstrating that, during the WOI, a global transcriptional derepression may be required for implantation and early embryo development [63]. Transcriptional derepression has been associated with multiple human disease states and should be investigated further within the context of endometrial-factor infertility. Despite previous associations of CTCF with endometriosis [67], its implication in endometrial progression and function has not been directly proposed until now\n",
      "\n",
      "The reference of article is https://doi.org/10.1186/s12958-023-01131-4, its information is:\n",
      " . Keywords CTCF, GATA6, Progesterone, Estrogen, miRNAs, TFs, Endometrial receptivity, Recurrent implantation failure, Infertility, Menstrual cycle regulation\n",
      "\n",
      "The reference of article is https://doi.org/10.1186/s12958-023-01131-4, its information is:\n",
      " . We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with significant changes between the endometrial phases. Notably, CTCF expression was decreased between the PF/ ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C).\n",
      "\n",
      "The reference of article is https://doi.org/10.1186/s12958-023-01131-4, its information is:\n",
      " . CTCF and GATA6 as distinguished transcriptional regulators of the human menstrual cycle We repeated the regulation network analysis with the TFs, to identify the prominent TFs commonly regulating the genes related to endometrial progression and function in the reported signatures. We observed 78.80% of the TFs regulated genes in up to 10 signatures, and only 99 TFs (21.20%) regulated genes in over 10 (Fig. 3A). We highlighted CTCF (a CCCTC-binding factor, which functions as a transcriptomic repressor) as the TF with the most influential regulation of endometrial progression and function across 95% of studies (18 gene lists, excluding that of Punyadeera, Fig. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53\n"
     ]
    }
   ],
   "source": [
    "# context = \"\\n\\n---\\n\\n\".join(texts_re)\n",
    "\n",
    "context = []\n",
    "for text, meta in zip(texts_re, metadata_re):\n",
    "\n",
    "  context.append(f'The reference of article is {meta[\"DOI\"]}, its information is:\\n {text}')\n",
    "\n",
    "print(\"\\n\\n\".join(context))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "ec055939",
   "metadata": {},
   "outputs": [],
   "source": [
    "prompt = get_ques_response_prompt(question=query, context=get_system_message_rag(\"\\n\\n\".join(context)))\n",
    "\n",
    "answer = llm.invoke(prompt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "19d90d34",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'CTCF is a conserved zinc finger protein with well-characterized regulatory functions throughout the human body. It acts as a transcriptional repressor in RNA polymerase II (Pol II) pausing, imprinting, and X-chromosome inactivation. Additionally, CTCF functions as an insulator, blocking the interaction between enhancers and the promoters of neighboring genes (https://doi.org/10.1186/s12958‑023‑01131‑4).\\n\\nIn endometrial tissue, CTCF is proposed to exert an inhibitory role during the PF phase of the menstrual cycle. Its significant downregulation in the secretory endometrium leads to the derepression of its inhibited genes, which then become transcriptionally active during the Window of Implantation (WOI) (https://doi.org/10.1186/s12958‑023‑01131‑4).\\n\\nCTCF is highlighted as a CCCTC-binding factor that functions as a transcriptomic repressor and has the most influential regulation of endometrial progression and function across 95% of studies (https://doi.org/10.1186/s12958‑023‑01131‑4). Overexpression of CTCF can lead to a drop in HOXA10 expression, which affects endometrial proliferation and function (https://doi.org/10.1186/s12958‑023‑01131‑4).'"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "answer"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4b4c5adb",
   "metadata": {},
   "source": [
    "## Download reference of each paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9926cb04",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'PaperTitle': 'Deciphering a shared transcriptomic regulation and the relative contribution of each regulator type through endometrial gene expression signatures',\n",
       "  'Publication': '2023',\n",
       "  'Authors': 'Antonio Parraga-Leo, Patricia Sebastian-Leon, Almudena Devesa-Peiro, Diana Marti-Garcia, Nuria Pellicer, Jose Remohi, Francisco Dominguez, Patricia Diaz-Gimeno',\n",
       "  'Email': 'patricia.diaz@ivirma.com',\n",
       "  'Abstract': 'Backgorund While various endometrial biomarkers have been characterized at the transcriptomic and functional level, there is generally a poor overlap among studies, making it unclear to what extent their upstream regulators (e.g., ovarian hormones, transcription factors (TFs) and microRNAs (miRNAs)) realistically contribute to menstrual cycle progression and function. Unmasking the intricacies of the molecular interactions in the endometrium from a novel systemic point of view will help gain a more accurate perspective of endometrial regulation and a better explanation the molecular etiology of endometrial-factor infertility. Methods An in-silico analysis was carried out to identify which regulators consistently target the gene biomarkers proposed in studies related to endometrial progression and implantation failure (19 gene lists/signatures were included). The roles of these regulators, and of genes related to progesterone and estrogens, were then analysed in transcriptomic datasets compiled from samples collected throughout the menstrual cycle (n = 129), and the expression of selected TFs were prospectively validated in an independent cohort of healthy participants (n = 19). Results A total of 3,608 distinct genes from the 19 gene lists were associated with endometrial progression and implantation failure. The lists’ regulation was significantly favoured by TFs (89% (17/19) of gene lists) and progesterone (47% (8 /19) of gene lists), rather than miRNAs (5% (1/19) of gene lists) or estrogen (0% (0/19) of gene lists), respectively (FDR < 0.05). Exceptionally, two gene lists that were previously associated with implantation failure and unexplained infertility were less hormone-dependent, but primarily regulated by estrogen. Although endometrial progression genes were mainly targeted by hormones rather than non-hormonal contributors (odds ratio = 91.94, FDR < 0.05), we identified 311 TFs and 595 miRNAs not previously associated with ovarian hormones. We highlight CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-miR-103a-3p, and hsa-miR-128-3p, as overlapping novel master regulators of endometrial function. The gene expression changes of selected regulators throughout the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their potential regulatory roles in the endometrium. Conclusions This study revealed novel hormonal and non-hormonal regulators and their relative contributions to endometrial progression and pathology, providing new leads for the potential causes of endometrial-factor infertility. Keywords CTCF, GATA6, Progesterone, Estrogen, miRNAs, TFs, Endometrial receptivity, Recurrent implantation failure, Infertility, Menstrual cycle regulation',\n",
       "  'Introduction': 'The endometrium is the innermost layer of the uterus, which undergoes dynamic histological, physiological and molecular changes that allow it to synchronize with embryo development, facilitate embryo implantation, and ultimately, establish a successful pregnancy [1]. Implantation is a crucial and complex limiting step for conception [2] which occurs between days 19 to 24 of a normal menstrual cycle. Consequently, this period of time is called the window of implantation (WOI) [3], and is characterized by abrupt transcriptomic changes in the endometrial tissue [4]. Alterations in either WOI establishment and endometrial progression may lead to implantation failures, which, along with biochemical miscarriage, account for more than 50% of pregnancy losses at pre-clinical stages [5, 6]. These alterations have broadly been classified as being related to a displaced WOI, caused by variable timing of endometrial progression, or a disrupted WOI, where impaired endometrial function prevents the establishment of an effective WOI [7]. Initially, the WOI was postulated to mainly be regu-lated by the ovarian hormones (namely, estrogen and progesterone) acting via their respective nuclear recep-tors (i.e., ESR1/2 and PGR). However, evidence has shown that these receptors cooperate with other transcription factors (TFs) and co-regulators to mediate uterine physi-ology [8]. In fact, TFs (such as homeobox TFs, FOXA2, and KLF9) have recently been reported as key regulators for the establishment of endometrial receptivity [9, 10], and microRNAs (miRNAs) have been found to similarly intervene in this complex and multifactorial process as transcriptional regulators [11]. While the involvement of both types of regulators was widely reported in embryo implantation [12] and other infertility-related diseases, including recurrent implantation failure (RIF) [13], the complex interactions between the different types of regulators and their respective contributions to endo-metrial progression and function have not been investi-gated from a systemic point of view. This paradigm shift approaches genes and regulators in an integrative way, by considering how they interact with and coordinate each other to carry out cellular processes, rather than rein-forcing the premise they work independently [14]. By more accurately reflecting the biological milieu, systemic approaches take gene-based discoveries to the next level, helping to generate hypotheses with relevant clinical and molecular implications [15, 16]. Previous studies of endometrial transcriptomics have been used to define the WOI, compare the endometrium of healthy women as their menstrual cycles progressed [4, 17-19]; address endometrial differences related to age [20] or the dysfunction of the WOI between patients with implantation failure and healthy controls [21]; and finally, correct for the menstrual cycle bias that can mask important biomarkers [22]. Notwithstanding, the candidate biomarkers of endometrial receptivity reported in each of the aforementioned studies overlap poorly [7]. Indeed, recent comparisons conducted by Sebastian-Leon and colleagues [7] found no congruities between 16 different reported gene lists of endometrial receptivity, and only fair to moderate functional agreements between some of the included signatures. Despite previous transcriptional regulation studies, there have been no holistic studies of this regulatory process, which robustly analyse the driv-ers of transcription across a higher number of studies (that each searched for endometrial receptivity biomark-ers) and map the relative contributions of each type of regulator to endometrial progression regulation (Ovarian hormones [i.e., estrogen, progesterone], TFs, miRNAs). Thus, the aim of this study was to utilize available genomic data regarding transcriptional regulators to identify overlapping mediators of endometrial transcrip-tional regulation (i.e., ovarian hormones, TFs, and miR-NAs) and determine the relative contribution of each type of regulator among the previously reported gene signatures. These findings could help unveil the master regulators and principal type of endometrial regulation, paving the way for further research aimed at improving woman ́s reproductive health.',\n",
       "  'Methods': 'A detailed study design is depicted in Supplementary Fig. S1. Annotating gene lists associated with endometrial progression and function Endometrial progression and implantation failure gene lists [7] were retrieved and updated using a public data repository (i.e., Gene Expression Omnibus (GEO)). Keywords used for the GEO search included: endometrial receptivity, mid-secretory endometrium, RIF, recurrent implantation failure, endometrium, unexplained infer-tility, and implantation failure. The resulting datasets were filtered by their publication date (i.e., from January 2018 to October 2020, to expand the systematic search of [7], number of samples (> 3 samples for each condition), and species (i.e., Homo sapiens), with no restrictions on publication language. The original signatures were then retrieved from their corresponding publications. The genes prioritized in the original publication were exclu-sively selected and their names were annotated with HUGO Gene Nomenclature Committee (HGNC) gene name using biomaRt R-package v.3.10 [23]. Each signa-ture was labelled with the name of first author of the cor-responding study. Our annotated gene lists were selected as representative signatures of endometrial progression and function for subsequent analyses (Supplementary Fig. S1A). Identifying hormonal and non-hormonal gene regulators of endometrial progression and function To study the hormonal regulation of endometrial progression, the Kyoto Encyclopaedia of Genes and Genomes (KEGG) [24], and Gene Ontology (GO) [25] databases were consulted for ovarian hormone-related genes, using relevant keywords (i.e., progesterone, estrogen, oestrogen and estradiol) and selecting those pathways and functions that contained them (Supplementary Fig. S1B). Genes associated with the obtained human KEGG pathways and GO functions were grouped according to whether they were unique to progesterone (P4 gene set) or estrogen (E2 gene set) or related to both hormones (P4 and E2 gene set). Gene targets of the nuclear proges-terone (PGR) and estrogen (ESR1, ESR2) receptors were added to their corresponding gene sets using DoRothEA (Discriminant Regulon Expression Analysis) database [26], considering only manually-curated or ChiP-Seq experimentally-validated gene-TF relationships. Finally, the P4 and E2 gene sets were independently mapped to each gene list. Meanwhile, to evaluate the non-hormonal regula-tors of endometrial progression and function, DoRo-thEA [26] and TarBase [27] databases were consulted to obtain TFs and miRNAs. DoRothEA was filtered as pre-viously described herein, while TarBase, which only con-tains miRNA-gene target relationships manually curated from publications or experimentally-validated in highthroughput datasets [27], was filtered by species (i.e., human). Finally, a functional over-representation analysis was carried out to identify which of the total annotated TFs or miRNAs were significantly associated with a par-ticular gene list (Supplementary Fig. S1B). To evaluate the relative contribution of each type of regulator (i.e., P4, E2, TFs, and miRNAs) on endometrial progression and function, three different comparisons were carried out applying Fisher’s exact tests. First, hor-monal regulators were evaluated using the relative pro-portion of P4- and E2-related genes included in each list, with respect to the total of number of genes founded in the P4 or E2 gene sets. Then, non-hormonal regulators were evaluated using the relative proportion of overrepresented miRNAs and TFs with respect to the total of miRNAs and TFs with a at least one target in the gene list. Finally, the relative proportion of genes under hor-monal (i.e., P4 and/or E2-annotated genes) versus nonhormonal (i.e., miRNA and/or TF-annotated genes) regulation was evaluated considering the miRNAs/TFs found within the individual P4 or E2 gene sets as hormo-nal regulators. Prioritizing key regulators in endometrial progression and function We built regulatory networks to systemically analyse endometrial regulation. Nodes represented gene lists and regulators (miRNAs or TFs), while edges indicated significant enrichment among the lists and their corre-sponding regulators (false discovery rate (FDR) ≤ 0.05). To select the most influential regulators (i.e., those which targeted genes in most of our signatures), we studied the degree distribution of the networks (i.e., the number of gene lists regulated by each molecule) and prioritized the miRNAs and TFs which surpassed the relative maximum (i.e., 1.50 times the interquartile range (IQR)) number of relationships (gene list-regulators). All networks were built and analysed with Cytoscape version 3.7 [28] (Sup-plementary Fig. S1B). Dataset construction and processing for in-silico validation To evaluate how the miRNAs’ expression change in the endometrium throughout the menstrual cycle, we used the raw data from GSE44558 GEO dataset [29], deriving from 20 endometrial samples collected throughout the menstrual cycle: four in early-proliferative (EPF), four in late-proliferative (LPF), four in early-secretory (ESE), four in mid-secretory (MSE) and four in late-secretory (LSE). To evaluate how the TFs’ expression change in the endometrium throughout the menstrual cycle, we analysed the integrated endometrial dataset previ-ously created by our group, and reported in [30], which compiled data from five prior publications (GSE98386, GSE29981, GSE4888, GSE119209 and GSE86491). This dataset included endometrial gene expression data from 109 participants with normal endometrium, where biopsies were collected in proliferative (PF) (n = 29), ESE (n = 29), MSE (n = 43) and LSE (n = 8) phases of the menstrual cycle (Table S1 ). Both TF and miRNA datasets were processed using the limma R-package [31]. Expression data were log transformed and quantile normalized, prior to explora-tory analyses which sought out possible outliers and batch effects. Relative gene expression ranges of low (1-10%), medium (11-50%), and high (51-100%) were established based on the expression of all the genes included in our dataset. Wet lab validation cohorts To corroborate the menstrual cycle-related expression changes observed with CTCF and GATA6 in our insilico analysis, we conducted a prospective study as an external validation, using an independent sample set. We included a cohort of 19 healthy Spanish women (obtaining a total of 20 biopsies), between the ages of 22-35 and with a body mass index of 22.80 ± 2.76 kg/ m2. Endometrial samples were collected and staged in the menstrual cycle according to the follicle growth to control ovulation and LH levels in urine. Samples were grouped into PF (n = 5), ESE (n = 5), MSE (n = 5) and LSE (n = 5) phases and categorized into two groups, PF/ ESE and MSE/LSE. RNA extraction and RT-qPCR Total RNA was extracted from the endometrial biopsy samples using the miRNeasy Mini Kit (Qiagen, Ger-many), and reverse transcribed into cDNA using the PrimeScript reagent kit (TAKARA, Japan). RT-qPCR reactions were carried out in duplicate, using fluores-cent Power-up SYBR Green (Thermo Fisher Scientific MA, USA) in a final volume of 10 μL. Primer sequences are shown in Table S2. Samples underwent 40 cycles of amplification, under standard conditions, using a Ste-pOnePlusTM System (Applied Biosystems, MA, USA). Relative mRNA expression was calculated using the 2− ΔΔCt method [32], and normalized to the expression of GAPDH housekeeping gene. Statistical analysis Fisher’s exact test was used to evaluate independent proportions. Mean expression changes across the cycle were studied using an analysis of variance (ANOVA) fol-lowed by a pairwise t-test or the Wilcoxon test when only two groups were available. P-values from multiple test comparisons were corrected by FDR and considered as significant when FDR ≤ 0.05. All statistical analyses were performed through R software (version 3.5) [33].',\n",
       "  'Results': 'Gene signatures associated to endometrial progression and function Of the 19 gene lists used for our analysis, eleven were obtained from published studies evaluating control patients throughout the menstrual cycle [4, 17-19, 34-40], and the other eight were derived from studies comparing patients with RIF or unexplained infertility to controls [21, 41-47] (Table 1). Unifying all the afore-mentioned signatures, we compiled 3,608 genes related to endometrial progression and function. Hormonal regulation of endometrial progression is largely driven by progesterone We identified 7,540 and 698 genes related to estrogen and progesterone hormones, respectively. However, as determined by the relative contribution of each type of hormone within each gene list (Fig. 1A), 17/19 (89%) signatures favoured regulation by progesterone rather than estrogen. These differences were significant in 47% of the signatures (FDR < 0.05). Exceptionally, the genes reported by the Altmae2010 and Koot signa-tures, that were related to unexplained infertility and RIF, had a preference for estrogen regulation (Fig. 1A). With respect to the proportion of progesterone-related genes, Altmae2010 showed 2.1 times more estrogenrelated genes while Koot showed 1.7 times more. More-over, comparing the proportion of genes regulated by both hormones, between signatures revealed that Alt-mae2010 and Koot exhibited a lower hormone-depend-ent behaviour overall, with respect to the remaining signatures (Fig. 1B), highlighting the role of non-hor-monal regulators. Non-hormonal regulation of endometrial progression and function We identified a total of 770 miRNAs and 555 TFs regu-lating at least one of the 3,608 genes reported in the 19 gene lists. Further, we observed that 29% of the miRNAs and 44% of the TFs were associated with the ovarian hormones (i.e., progesterone and/or estrogen). Significantly more genes were targeted by hormo-nally-influenced regulators rather than non-hormonal regulators (P -value < 0.01, odds ratio = 91.94). Never-theless, we also identified 595 miRNAs (77.27%) and 311 TFs (56.04%) that were unrelated to progesterone and/or estrogen, highlighting a substantial proportion of hormonally-independent regulators of the men-strual cycle. Of the respective miRNAs and TFs targeting genes related to endometrial progression and function, 417 miRNAs and 467 TFs were significantly over-represented in at least one gene list (FDR < 0.05, Tables S3-S4). Further, most signatures were significantly governed by TFs rather than miRNAs (4 > odds ratio > 164, FDR < 0.05, 17 lists [89%]), with the excep-tion of Koot’s signature that was governed by miRNAs and showed the opposite (odds ratio = 0.35, FDR < 0.05, Table S5). Hsa-miR-15a-5p, hsa-miR-128-3p, hsa-miR-218-5p, hsa-miR-27b-3p, hsa-miR-107, hsa-miR-424-5p, hsa-miR-195-5p, hsa-miR-103a-3p, has-let-7b-5p and hsa-miR-22-3p are key players in the regulation of endometrial progression and function To bridge the gap of endometrial-based evidence in DoRothEA and TarBase databases, we aimed to iden-tify the common miRNAs that mediate gene expres-sion related to endometrial progression and function. We evaluated the number of endometrial gene lists that each miRNA controlled (Fig. 2A). This strategy highlighted the most influential miRNAs that were commonly found across the endometrial gene lists. Notably, 85.80% of miRNAs regulated genes from one to three signatures, and only 59 miRNAs (14.20%) were associated with more than three signatures. Ultimately, 24 miRNAs were prioritized for regulating genes of at least six signatures. Next, we validated the gene expression trends for these 24 prioritized miRNAs in an independent endo-metrial gene expression dataset (GSE44730) (Fig. 2B). This dataset included the miRNA expression in endo-metrial biopsies from 20 healthy women in differ-ent phases of the menstrual cycle. We found that two of our prioritized miRNAs were not expressed in the endometrial tissue, and only 10 miRNAs (i.e., hsa-miR-15a-5p, hsa-miR-128-3p, hsa-miR-218-5p, hsa-miR-27b-3p, hsa-miR-107, hsa-miR-424-5p, hsamiR-195-5p, hsa-miR-103a-3p, has-let-7b-5p and hsamiR-22-3p) showed significant changes across the menstrual cycle (Fig. 2B). Of these, only the expression of hsa-miR-22-3p significantly increased from ESE to MSE stages (FDR = 0.02, Fig. 2B), highlighting the potential inhibitory role of this miRNA in endometrial function. The expression of the remaining miRNAs decreased during the MSE phase. CTCF and GATA6 as distinguished transcriptional regulators of the human menstrual cycle We repeated the regulation network analysis with the TFs, to identify the prominent TFs commonly regu-lating the genes related to endometrial progression and function in the reported signatures. We observed 78.80% of the TFs regulated genes in up to 10 signa-tures, and only 99 TFs (21.20%) regulated genes in over 10 (Fig. 3A). We highlighted CTCF (a CCCTC-binding factor, which functions as a transcriptomic repressor) as the TF with the most influential regulation of endo-metrial progression and function across 95% of studies (18 gene lists, excluding that of Punyadeera, Fig. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53. Notably, GATA6 was selected due to novel association with endometrial pro-gression and function. To infer the expression of CTCF and GATA6 in endometrial tissue throughout the menstrual cycle, we analysed their expression using a relevant endo-metrial dataset recently created by our group [30] (Table S1 ). We observed that, relative to the global expression of endometrial genes, CTCF was broadly expressed while GATA6 was moderately expressed. Indeed, CTCF was significantly down-regulated between the PF and ESE phases (FDR = 2.10E-03) and ESE to the MSE phase (FDR = 2.10E-04), while GATA6 expression increased from the PF to the MSE phase (FDR = 1.70E-11) and from the ESE to MSE phase (FDR = 9.80E-15). These findings supported the role of these TFs at the beginning of the menstrual cycle and during the MSE phase (Fig. 3 B). These in-silico validations were then prospec-tively corroborated by RT-qPCR analysis in an inde-pendent sample set of 20 endometrial biopsies. We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with signifi-cant changes between the endometrial phases. Nota-bly, CTCF expression was decreased between the PF/ ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C).',\n",
       "  'Discussion': 'This data-driven approach exposed the common tran-scriptional regulators among 19 studies who proposed variable biomarkers of endometrial progression and function. In this study, we focused on understanding the relative contribution of both the hormonal and nonhormonal regulation, from an alternative holistic per-spective. We applied data-driven hypothesis research that, unlike the traditional scientific method, allowed us to generate new hypotheses based on all available bio-logically-relevant knowledge [16], observing the molec-ular relationships from a wider scale view. Besides, some traditional molecular procedures such as PCRs are also performed to corroborate the insight uncovered with this approach. We highlighted a larger influence of progesterone-related genes and TFs in endome-trial progression and observed a larger contribution of miRNAs and estrogen-related genes in endometrial pathology. Furthermore, we unveiled CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsamiR-103a-3p and hsa-miR-128-3p as master regulators in the endometrium, and these findings were reinforced by the dynamic changes in their expression throughout the menstrual cycle (dually validated with in-silico and prospective analyses). The menstrual cycle is perceived to be under tight regulation by the ovarian hormones, progesterone and estrogen [48]. Supporting the role of progester-one in the secretory endometrium [49], and reinforc-ing its indispensable role in endometrial regulation [50, 51], we found progesterone and its related genes had a larger influence in menstrual cycle progression than estrogen. As the RIF-related gene lists reported by Altmae2010 and Koot were generally controlled by estrogen, this further suggested that while progester-one could be crucial for a suitable endometrial progres-sion and mid-secretory phase acquisition, estrogenic irregularities may be more conducive to endometrial function and molecular pathology. Indeed, this hypoth-esis corroborated previous reports of most uterine dis-orders being estrogen-dependent [44, 52]. Supporting the findings from another recent study from our group, where the Altmae2010 and Koot signatures were classi-fied as highly-predictive signatures to identify endome-trial pathology in comparison with a prediction model that detected endometrial progression [7 ]. This study distinguished these signatures (related to implantation failure and unexplained infertility) based on the mag-nitude of their miRNA and estrogen-regulation, which contrasted with the prominent influence of the TFs over the remaining signatures more related to endometrial progression. All these facts reinforce the hypothesis that implantation failure is mainly caused by impaired miRNA expression [53]. Although we observed a clear hormonal influence in numerous annotated regulators, this study transcended the simplistic paradigm of the bihormonal regulation of endometrial function by identifying a plethora of medi-ators, including 595 miRNAs and 311 TFs, that were previously unrelated to these ovarian hormones. These findings set the foundation for new discoveries, reveal-ing alternative pathways and new actors in the regula-tion of the human endometrium, and deepening our understanding of the complex regulatory mechanisms behind endometrial progression and unexplained infertility. MicroRNAs are well known to be involved in embryo implantation [12] and receptivity control [54, 55]. Using network analysis, we unveiled the 10 common miR-NAs regulating endometrial progression and function across published biomarker signatures (hsa-miR-22-3p, hsa-miR-107, hsa-miR-103a-3p, hsa-miR-128-3p, hsamiR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, hsamiR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p). Their significant gene expression changes in the secre-tory endometrium support their potential role in the endometrial regulation through an inhibition during the WOI. Notably, hsa-miR-424-5p, hsa-miR-27b-3p and hsa-miR-195-5p have been previously associated with RIF [11, 54, 56], while has-let-7b-5p was associ-ated with human endometrial receptivity [57], and hsa-miR-22-3p was overexpressed in a decidualization model [58]. Moreover, hsa-miR-15a-5p was previously associated with endometriosis [59]. Notwithstanding, this study is the first to associate hsa-miR-15a-5p, hsamiR-218-5p, hsa-miR-103a-3p, hsa-miR-107, and hsamiR-128-3p to endometrial progression and functional regulation in humans. Due to their direct relationship with progesterone and estrogen [60], we expected to find TFs, such as ESR1 and PGR, involved in endometrial progression, how-ever, we additionally identified other TFs with an even larger influence in this process. For instance, STAT3 was associated with 17/19 signatures (FDR < 0.05), and is known to regulate uterine epithelial junctional reor-ganization and stromal proliferation, which are critical for implantation [61]. SP1, which similarly regulated 17/19 signatures (FDR < 0.05), acts as a downstream paracrine target of progesterone to regulate estrogen inactivation, and could have a predominant role dur-ing the WOI [62, 63]. By focusing on the most univer-sal, overlapping TFs that were not previously related to implantation, we ultimately prioritized CTCF and GATA6 as novel key regulators of endometrial pro-gression, however, their specific molecular actions in endometrial progression and function were beyond the scope of this project and merit further investigation. CTCF is a conserved zinc finger protein whose regu-latory functions are well characterized throughout the human body [64]. CTCF acts as a transcriptional repres-sor in RNA polymerase II (Pol II) pausing and imprint-ing and X-chromosome inactivation [65], as well as an insulator, blocking the interaction between enhancers and the promoters of neighbouring genes [66]. Consider-ing these attributed functions and its variable expression throughout the menstrual cycle, we propose that CTCF exerts an inhibitory role in endometrial tissue during the PF phase of the menstrual cycle. With the significant downregulation of CTCF in the secretory endometrium, dually validated in-silico and experimentally herein, its inhibited genes would be derepressed and become tran-scriptionally active during the WOI. This interpretation supports previous findings from our group demonstrat-ing that, during the WOI, a global transcriptional dere-pression may be required for implantation and early embryo development [63]. Transcriptional derepression has been associated with multiple human disease states and should be investigated further within the context of endometrial-factor infertility. Despite previous asso-ciations of CTCF with endometriosis [67], its implica-tion in endometrial progression and function has not been directly proposed until now. Nevertheless, the role of CTCF in endometrial receptivity has been revealed through its interaction with HOXA10, a gene mainly expressed in endometrium involved in functions such as endometrial proliferation and differentiation, the forma-tion of pinopodes or embryo implantation [68]. Indeed, this study proposed that an overexpression of CTCF can lead to a drop in HOXA10 expression, affecting endome-trial proliferation and endometrial function and corroborating the fact that CTCF must decrease its expression as we can observe in our findings. GATA6, the second prioritized TF that overlapped across the gene lists, is a transcription factor belong-ing to the GATA family, a highly conserved family of six zinc finger proteins [69]. Besides its essential role in embryonic development [70], GATA6 plays a key role in endometriosis and regulates steroidogenic genes [71]. Together with evidence supporting both an activating and inhibitory role for GATA6 [72, 73], the significant changes we observed in its expression across the men-strual cycle reflect its presumed involvement in the endometrium and suggest an activating role in endometrial progression and endometrial receptivity acquisition. Like CTCF, this is the first time that GATA6 is proposed as a key TF in endometrial progression and function. Although the exact molecular mechanisms underlying endometrial regulation remain elusive, emerging exper-imental technologies such as High-throughput Chro-mosome Conformation Capture (Hi-C) or chromatin interaction analysis with paired-end tags (ChIA-PET) that respectively study general chromatin interactions or chromatin interactions involving a specific protein [74, 75] may improve our understanding of genetic regulation in the future. Nevertheless, as collaborative efforts unveil new regulators and gene targets in humans (such as the ENCODE project [76]), the complexities of transcriptional regulation can be exposed by in-silico analyses like those described herein, and we can gain a better understanding of the processes involved in men-strual cycle progression and endometrial competence. Furthermore, it should be noted that results of gene expression studies are influenced by different variables, such as the experimental platforms employed, study designs, or patients’ characteristics of each independent study, adding variability [30, 77, 78]. Despite this fact, we have found that our holistic approach overcomes this undesired variability effect by finding common upstream master regulators across studies and predicting new relationships between hormones, TFs and miRNAs. This approach also provides the basis for future singlemolecule studies that aim to elucidate new regulatory pathways in endometrial progression and function. Our results lay the foundation for further molecular studies that can validate the function(s) of the prevailing endo-metrial regulators we prioritized.',\n",
       "  'Conclusion': 'The study shows for the first time, the relative contri-bution of estrogens, progesterone, TFs and miRNAs in endometrial function and progression. Endometrial pro-gression is mainly influenced by progesterone-related genes and TFs, whereas miRNAs and estrogen-related genes play a larger role in endometrial pathology. Moreover, we highlight novel common transcriptional regulators such as CTCF, GATA6, hsa-miR-15a-5p, hsamiR-218-5p, hsa-miR-107, hsa-miR-103a-3p, and hsamiR-128-3p across 19 studies that propose biomarkers in endometrial regulation. These results reveal the molecu-lar mechanism underlying endometrial regulation and lay the foundation for the development of targeted therapies for patients with endometrial-factor infertility.',\n",
       "  'URL': 'https://doi.org/10.1186/s12958-023-01131-4',\n",
       "  'Journal': 'Reproductive Biology and Endocrinology'},\n",
       " {'PaperTitle': 'Predicting risk of endometrial failure: a biomarker signature that identifies a novel disruption independent of endometrial timing in patients undergoing hormonal replacement cycles',\n",
       "  'Publication': '2024',\n",
       "  'Authors': 'Patricia Diaz-Gimeno, Ph.D., Patricia Sebastian-Leon, Ph.D., Katharina Spath, Ph.D., Diana Marti-Garcia, M.Sc., Josefa Maria Sanchez-Reyes, Ph.D., Maria del Carmen Vidal, Ph.D., Almudena Devesa-Peiro, Ph.D., Immaculada Sanchez-Ribas, Ph.D., Asunta Martinez-Martinez, M.Sc., Nuria Pellicer, M.D., Ph.D., Dagan Wells, Ph.D., and Antonio Pellicer, M.D., Ph.D.',\n",
       "  'Email': 'patricia.diaz@ivirma.com or patricia_diaz@iislafe.es',\n",
       "  'Abstract': 'Objective: To propose a new gene expression signature that identifies endometrial disruptions independent of endometrial luteal phase timing and predicts if patients are at risk of endometrial failure. Design: Multicentric, prospective study. Setting: Reproductive medicine research department in a public hospital affiliated with private fertility clinics and a reproductive genetics laboratory. Patients: Caucasian women (n1⁄4 281; 39.4/C6 4.8 years old with a body mass index of 22.9/C6 3.5 kg/m 2) undergoing hormone replacement therapy between July 2018 and July 2021. Endometrial samples from 217 patients met RNA quality criteria for signature discovery and analysis. Intervention(s): Endometrial biopsies collected in the mid-secretory phase. Main Outcome Measure(s):Endometrial luteal phase timing-corrected expression of 404 genes and reproductive outcomes of thefirst single embryo transfer (SET) after biopsy collection to identify prognostic biomarkers of endometrial failure. Results: Removal of endometrial timing variation from gene expression data allowed patients to be stratified into poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognosis groups on the basis of their clinical and transcriptomic profiles. Significant differences were found between endometrial prognosis groups in terms of reproductive rates: pregnancy (44.6% vs. 79.6%), live birth (25.6% vs. 77.6%), clinical miscarriage (22.2% vs. 2.6%), and biochemical miscarriage (20.4% vs. 0%). The relative risk of endometrial failure for patients predicted as a poor endometrial prognosis was 3.3 times higher than those with a good prognosis. The differences in gene expression between both profiles were proposed as a biomarker, coined the endometrial failure risk (EFR) signature. Poor prognosis profiles were characterized by 59 upregulated and 63 downregulated genes mainly involved in regulation (17.0%), metabolism (8.4%), immune response, and inflammation (7.8%). This EFR signature had a median accuracy of 0.92 (min1⁄40.88, max1⁄40.94), median sensitivity of 0.96 (min1⁄40.91, max 1⁄4 0.98), and median specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88), positioning itself as a promising biomarker for endometrial evaluation. Conclusion(s): The EFR signature revealed a novel endometrial disruption, independent of endometrial luteal phase timing, present in 73.7% of patients. This EFR signature stratified patients into 2 significantly distinct and clinically relevant prognosis profiles providing opportunities for personalized therapy. Nevertheless, further validations are needed before implementing this gene signature as an artificial intelligence (AI)-based tool to reduce the risk of patients experiencing endometrial failure. (Fertil Steril /C2102024;-:-- -. /C2112024 by American Society for Reproductive Medicine.)',\n",
       "  'Introduction': 'A key factor for reproductive success in assisted reproduction treatments is the status of the maternal endometrium during embryo implantation and fetal development. The endometrial cycle is reflected by the cyclic structural and functional changes of the endometrium across the menstrual cycle, particularly during the mid-secretory phase, to prepare for the‘‘window of implantation’’ (WOI) . After successful embryo implantation, the decidua (specialized layer of the endometrium) actively encapsulates the trophectoderm to support placentation and provide adequate vascularization for optimal fetal growth . Thus, approaches that predict and prevent endometrial-factor infertility could substantially improve assisted reproduction treatment outcomes by supporting the establishment and maintenance of pregnancy . There is a lack of consensus on the minimum number of implantation failures with good-quality embryos derived from ovum donation or with confirmed euploid karyotypes required to diagnose recurrent implantation failure (RIF) . The disparities in clinical classification and heterogeneous etiology of RIF confound its diagnosis and leave the exact prevalence unclear . As clinical symptoms cannot be prevented and do not provide sufficient criteria to stratify patients, patients might be misclassi fied and receive inefficient treatment. Indeed, most studies using clinical criteria for RIF identified no or few differentially expressed genes between samples of affected patients , reflecting the molecular heterogeneity of RIF. To overcome these healthcare challenges, high-throughput molecular technologies are combined with artificial intelligence (AI) algorithms to provide a deeper understanding of infertilityrelated pathogenesis . These molecular insights are then applied in genomics-driven precision medicine approaches to characterize complex diseases, and ultimately, improve patient diagnosis and prognosis . Employing genomics in reproductive medicine has the potential to go beyond treating the clinical manifestations of disease progression , by leveraging the patients’ heterogeneous molecular profiles for precise stratification and revealing new targets for preventive measures . In this regard, current diagnostic tools for endometrialfactor infertility use transcriptomics to date the endometrial biopsy in the mid-secretory phase, exceeding the accuracy and objectivity of traditional endometrial progression staging by Noyes’ histological criteria . However, the clinical utility of synchronizing the endometrium by adjusting the timing of progesterone (P4) administration on the basis of the patient’s transcriptomic endometrial dating profile remains controversial . Independent randomized clinical trials found no improvement in reproductive outcomes , indicating the inefficacy of this clinical intervention and highlighting the need for alternative treatments and/or a refined target patient population that bene fits from this strategy. Combining these precision medicine approaches with transcriptomics and AI algorithms; our group recently proposed a novel molecular taxonomy of RIF and a new clinical algorithm for endometrial-factor infertility that considered 2 molecular causes of implantation failure: a WOI displacement that indicates a dysregulated endometrial timing; and a WOI disruption that indicates an endometrial alteration independent of endometrial timing . Transcriptomic analysis revealed these 2 WOI phenotypes were presented exclusively or simultaneously and patients with RIF had significantly more WOI disruptions but a similar frequency of WOI displacements, compared with controls . Thesefindings suggested that WOI disruptions had stronger associations with implantation failure than WOI displacements and shifted the paradigm for the study and treatment of endometrialfactor infertility . In addition, this work by SebastianLeon et al. highlighted the importance of removing the transcriptomic variation due to cyclical endometrial tissue changes (ie, the effect of endometrial timing) that, if not corrected, could mask the significant disruptions in endometrial gene expression. However, given this hypothesis-driven research was conducted retrospectively and in silico with publicly available data, the reproductive outcomes of the cycle after the biopsy collection were not available . This clinical follow-up is required to discover biomarkers for WOI disruption in patients with a higher risk of endometrial failure. On the basis of these genomics-driven precision medicine strategies, this proof-of-concept study aimed to discover a biomarker panel that could identify endometrial disruption, independent of endometrial timing, to predict, and ultimately prevent, poor reproductive outcomes in in vitro fertilization (IVF) patients.',\n",
       "  'Methods': 'MATERIAL AND METHODS Ethics statement This study was approved by the Ethics Committee of the Instituto Valenciano de Infertilidad, Valencia, Spain (1810-FIVI-066-PD/1810-FIVI-048-PD). Written informed consent was obtained from all recruited patients (n1⁄4 281). Participants, collection of endometrial biopsies, and reproductive outcomes This prospective, multicentric study was conducted between July 2018 and July 2021 at 5 private fertility clinics in Spain. Participants (n1⁄4 281) were scheduled for endometrial evaluation due to medical indications orR2 implantation failures of idiopathic origin. All patients met the following inclusion criteria: 18- 50 years old, with a body mass index (BMI) of 19- 30 kg/m 2, undergoing hormone replacement therapy for single embryo transfer (SET), presenting an endometrial thickness >6.5 mm with trilaminar structure in proliferative phase before P4 administration, and serum levels of estradiol >100 pg/mL and P4<1 ng/mL on the 10th day of estrogen treatment . Endometrial biopsies were collected after an average of 114.5/C6 7.2 h of P4 (duration ranged from 82 to 172 h), and processed as we previously described . Baseline patient characteristics including age, BMI, the number of IVF attempts, the quality of the embryo transferred (defined in theSupplemental Methods, available online), and reproductive outcomes (ie, pregnancy, live birth, and biochemical and clinical miscarriage) for each attempt, were obtained from internal medical records. Information about IVF attempts in other fertility clinics was included if recorded (only in 47.7% of patients). Study design First, a custom RNA-seq gene panel (the endometrial failure risk [EFR] panel) comprising 404 biomarkers of endometrial luteal phase timing and/or potential disruption was designed and employed to determine the risk of endometrial failure among 217 IVF patients (Supplemental Fig. 1, step 1 available online). The endometrial timing-corrected expression of the EFR panel was used to study the differential expression associated with endometrial disruption together with clinical characteristics of IVF patients to stratify the population as having poor or good endometrial prognoses, according to a semi-supervised learning process (Supplemental Fig. 1, step 2). The distinct transcriptomic patterns of each prognosis group were associated with the differences in reproductive outcomes of SET after biopsy collection, and the differentially expressed gene signature was proposed as a biomarker of endometrial disruption (Supplemental Fig. 1, step 3). Finally, the gene expression signature was used to supervise machine learning algorithms to evaluate the biomarker capacity of this signature to distinguish endometrial competence. The predictive performance of the models was assessed through the accuracy, sensibility, and specificity over 100 iterations of fivefold strati fied cross-validation ( Supplemental Fig. 1 , step 4). Constructing the EFR gene panel Supplemental Figure 2illustrates the workflow for the EFR panel gene selection. A whole-transcriptome endometrial data set (GSE58144, n1⁄4 115) retrieved from Gene Expression Omnibus was processed to correct the endometrial progression effect, using linear models in limma (version 3.46.0) , and unmask genes related to endometrial disruption . These genes were ranked according to an informativity score (CorrelationAttributeEval) in Weka (version 3.8.2, December 22, 2017) to evaluate their worth by measuring their correlation with endometrial pathology. Gene sets with increasing gene signature size were developed by adding the most informative genes in a stepwise manner, and their predictive performance was analyzed using Support Vector Machine and Random Forest algorithms. After calculating predictive performance with a stratified fivefold cross-validation (default parameters; 80:20; 100 iterations) using the Weka R package , the optimal signature size was determined for each algorithm, as the one with the fewest genes and highest accuracy. Finally, genes we previously prioritized for transcriptomic endometrial dating w e r ea l s oi n c l u d e di nt h e EFR panel to be able to remove the endometrial luteal phase timing variation. Endometrial biopsy processing, sequencing, and transcriptomic preprocessing Total RNA was extracted, and only samples that met our quality criteria were sequenced using the TruSeq Targeted RNA Illumina protocol and our custom EFR panel, in the Illumina NextSeq 550 platform with a paired-end design of 150 cycles and 4 M reads/sample, as previously described . Raw transcriptomic data was processed and normalized using limma, to filter low-quality raw counts (Q< 30) and low-read samples (<2M ). Possible batch effects were detected using principal components analysis and corrected with linear models in limma. Finally, endometrial luteal phase timing effects were detected using our transcriptomic endometrial dating (TED) model (see Supplemental Methods) and removed using limma. EFR biomarker signature identification A preliminary classification of patients was based on the patients’ history of reproductive outcomes and transcriptomic patterns corrected for endometrial luteal phase timing. This approach eliminated the gene expression variation related to the cyclic tissue changes that mask endometrial disruption and was used before applying semi-supervised machine learning on the basis of self-labeled techniques (see Supplemental Methods). Once, the genomic-driven disease taxonomy was established, the differentially expressed genes obtained between the final good and poor prognosis groups were proposed as the EFR biomarker signature. The limma R package was used to identify the statistically signi ficant differentially expressed genes (false discovery rate [FDR]-adjustedP value <.05). Functional characterization of the EFR signature The biomarker signature genes were annotated for functional interpretation databases by consulting Kyoto Encyclopaedia of Genes and Genomes human pathways (release 99.0, August 1, 2021) and Gene Ontology (version July 2, 2021; employing only experimental annotations) . Functional terms were manually summarized into categories according to their description in the database and presented as a bar plot using ggplot2 . EFR signature capability as a biomarker of endometrial disruption To estimate the prediction capability of the EFR signature to distinguish poor and good endometrial prognosis profiles, we implemented predictive models on the basis of the Support Vector Machine algorithm. Using a stratified fivefold crossvalidation process (repeated 100 times), the accuracy, sensitivity, and specificity were estimated. The range of values obtained across the 100 iterations was presented as a boxplot using ggplot2. Statistical analysis Rates for reproductive outcomes (ie, pregnancy rate [PR], cumulative pregnancy rate [CPR], live birth rate [LBR], biochemical miscarriage rate [BMR], and clinical miscarriage rate [CMR]) were calculated as de fined in the Supplemental Methods. Descriptive statistics were used for transcriptomic and clinical characteristics of patients to compare prognosis or WOI displacement groups. Continuous variables were presented as an overall mean /C6 standard deviation whereas discrete variables were presented as counts and percentages. Groups were compared using the Wilcoxon rank-sum test for continuous variables and Fisher’s exact test for discrete variables. All statistical analyses were conducted in R. All graphical results were generated with ggplot2. Gene expression validation by quantitative realtime polymerase chain reaction Quantitative real-time polymerase chain reaction (RT-qPCR) was performed to validate the expression of the top 4 differentially expressed genes between patients with poor (n1⁄4 10) and good (n1⁄4 10) endometrial prognoses. RNA (200- 220 ng) was reverse transcribed into cDNA using the PrimeScript Reagent Kit (Perfect Real Time, Takara, Japan) on a Thermocycler T3000 (Biometra, Ireland). Quantitative real-time polymerase chain reaction was performed on a StepOnePlus Real-Time PCR System (Applied Biosystems, USA) using Power-Up SYBR Green (Thermo Fisher Scientific, USA) and standard conditions.Supplemental Table 1(available online) lists the specific primers (Invitrogen, Thermo Fisher Scientific) used. Universal human RNA (Agilent, Spain) and water were respectively included as positive and negative controls. Each sample was analyzed in duplicate and relative gene expression (normalized to beta actin) was calculated using the DDCt method .',\n",
       "  'Results': 'Preliminary sequencing with the 404-gene EFR panel The EFR panel evaluated 404 genes: 139 that predict endometrial disruption and 310 that serve as biomarkers of endometrial dating or timing. Notably, 45 genes overlapped in these 2 endometrial factors. Of the 281 endometrial samples collected, we excluded 56 samples for having low-quality or insufficient RNA; 6 samples containing a low number of sequencing reads (Supplemental Fig. 3A); and 2 samples that had low gene expression after normalization ( Supplemental Fig. 3B ). Thus, 217 patients were considered in the downstream analysis. After correcting batch effects from sequencing runs (Supplemental Fig. 3C) and removing the transcriptomic variation due to endometrial cyclic changes (endometrial luteal phase timing;Supplemental Fig. 3D), there were no other confounding technical or clinical factors (Supplemental Fig. 4). A genomic-driven endometrial taxonomy for patient stratification on the basis of a novel endometrial disruption independent of endometrial timing Forty-four patients were initially categorized as acute RIF on the basis of the inability to carry a pregnancy to term after 3- 9 SETs. Alternatively, 40 patients were categorized as acute Fertile because they reached full-term pregnancy at thefirst SET after biopsy. Notably, after this clinical acute criteria, 133 (61.3%) patients remained unclassified for having insufficient attempts (Supplemental Fig. 5A), and no transcriptomic differences were found between the acute RIF and acute Fertile groups (the lowest FDR was 0.3027). Incorporation of transcriptomic data with this initial acute clinical classification allowed us to stratify patients with poor (n1⁄4 137, 63.1%) or good (n1⁄4 49, 22.6%) endometrial prognoses; leaving only 31 samples (14.3%) unclassified because of the lack of recognition by the algorithms (Supplemental Fig. 5B). EFR biomarker signature identification and functional characterization A total of 122 genes were identified as significantly different (30.2% of all panel genes, FDR< 0.05) between patients with good and poor endometrial prognoses, including 63 downregulated and 59 upregulated in patients with poor endometrial prognosis (Table 1 and Fig. 1A), and proposed as the EFR biomarker signature. Transcriptomically, endometrial biopsies clustered by prognosis (Fig. 1B), and the signature genes had functions that were mainly related to regulation (17%); metabolism (8.4%); the immune system and inflammation (7.8%); protein processes (7.3%), signaling (6.9%), transport (6.1%), and the cell cycle, proliferation, and differentiation (5.3%) (Fig. 1C andSupplemental Table 2). The poor prognosis profile was distinguished by a downregulation of 27/39 (69.2%) functional groups. We highlight the functions with higher proportions of downregulated genes as steroidogenesis and steroid signaling (77.8%), hormone-related (80.0%), the nervous system (81.3%), gametogenesis and reproduction (85.7%), and oxidative stress and response to O 2 (88.9%), as well as cell action potential and polarization (100%) (Fig. 1D). Few functional groups, including cell migration, extracellular matrix, and glucose homeostasis contained a larger proportion of upregulated genes (60.9%, 62.5%, and 62.5% upregulated genes, respectively;Fig. 1D). Differences in clinical reproductive outcomes authenticated the EFR signature’s clinical relevance There were no significant differences in the age or BMI of patients predicted to have good/poor endometrial prognosis (all P>.05). Women predicted to have poor endometrial prognosis previously underwent significantly more SETs than those with good endometrial prognosis (2.8 vs. 1.6;P<.0001), however, there were no significant differences in embryo quality (P1⁄4.4679) between groups (Supplemental Table 3). The lack of differences in the proportion of out-of-phase endometria (57% in poor vs. 47% in good,P1⁄4.25) between good and poor prognosis groups (Supplemental Table 3) distinguished this signature from established endometrial luteal phase timing phenotypes (displaced or on-time). Therefore, the significant differences (P<.0001) between the reproductive outcomes of women predicted to have poor and good endometrial prognoses validated the clinical relevance of our transcriptomic signature. Specifically, the poor endometrial prognosis was associated with a 34.5% decline in PR (P1⁄43.8E-05), a 52% reduction in LBR (P1⁄45E-10), a 19.6% rise in CMR (P1⁄4.0066), and 20.4% increase in BMR (P1⁄4.0023) (Fig. 2A). Furthermore, patients with a poor endometrial prognosis only achieved a CPR of 42.3% after the ninth SET, whereas patients with a good prognosis had an 83.7% CPR at thefifth attempt (Fig. 2B). The relative risk of implantation failure or miscarriage in women predicted to have poor endometrial prognosis was respectively 2.7 and 11.5 compared with women with good endometrial prognosis. However, when failures in implantation or ongoing pregnancy were considered together, the relative risk of endometrial failure (not achieving a live birth in the next SET) for patients with poor endometrial prognosis was 3.3. EFR signature biomarker’s predictive capability The 186 samples labeled with poor (n1⁄4 137) or good (n1⁄4 49) endometrial prognoses were used to evaluate the signature’s predictive performance. The 122 differentially expressed genes between poor/good endometrial prognoses proposed as the EFR biomarker signature were employed to train the model. Cross-validation yielded a median accuracy of 0.92 (min1⁄4 0.88, max 1⁄4 0.94), sensitivity of 0.96 (min1⁄4 0.91, max 1⁄4 0.98), and specificity of 0.84 (min 1⁄4 0.77, max 1⁄4 0.88) (Fig. 2C). Notably, 163 of 186 (87.6%) samples were classified correctly in almost 90% of all cross-validation models, whereas 151 of 186 (81.2%) samples were well classified correctly in 100% of the iterations. Twelve and eleven samples were misclassified consistently, in <50% or 50%- 90% of models, respectively (Fig. 2D). Clinical algorithm results after evaluating both WOI factors Next, we grouped samples according to both potential causes of implantation failure, endometrial luteal phase timing (displaced or on-time) (Fig. 3A), and endometrial disruption (good or poor prognosis) (Fig. 2A) and compared the patients’ reproductive outcomes. There were no significant differences between the patients with displaced and on-time WOI in terms of BMI (P1⁄4.8740), the average number of SETs (P1⁄4.0879), and embryo quality (P1⁄41). However, patients with a displaced WOI were signi ficantly older (40.4 vs. 38.3 years old; P1⁄4.0044) (Supplemental Table 4). Once patients were stratified by prognosis (endometrial disruption), the proportion of patients with poor prognosis was similar (P1⁄4.2458) between displaced and on-time groups (Fig. 3B). Experimental validation of genes related to the EFR signature The expression of the top 4 differentially expressed genes in the poor prognosis group was validated by RT-qPCR. Progestagen-associated endometrial protein and G-coupled protein receptor 110, which were found upregulated by RNA-seq, were significantly overexpressed in the poor prognosis group, compared with the good prognosis group (fold change [FC]1⁄4 6.157 [P1⁄4.0288] and FC1⁄4 11.803 [P1⁄4.0350], respectively; Supplemental Figs. 6Aand B). Meanwhile, the expression of fibrinogen beta chain and solute carrier 16 member 6, which were found downregulated by RNA-seq, did not reach a statistically significant difference (FC1⁄4 1.582 [P1⁄4.2475] and FC1⁄4 2.311 [P1⁄4.4376], respectively;Supplemental Figs. 6Cand D).',\n",
       "  'Discussion': 'Advances in the diagnosis and treatment of RIF have been limited by its elusive etiology, the discrepancies in clinical classification (mainly due to the lack of consensus over the minimum number of pregnancy attempts) , and the heterogeneous molecular profiles of affected patients . On the basis of current clinical practice, patients with RIF may be misclassified and/or receive inef ficient treatment . Thus, the aim of this study was to propose a novel gene expression signature that could identify endometrial disruptions independent of endometrial luteal phase timing. Using a semi-supervised learning AI strategy together with the endometrial gene expression obtained from our custom EFR panel, we stratified patients according to poor or good endometrial prognoses and demonstrated how this signature reliably predicted gene expression patterns that corresponded with significant differences in PR, LBR, CMR, and BMR. When considering both WOI factors (disruption and timing) in our study cohort, poor endometrial prognoses (disruption) affected 73.7% of patients, whereas WOI displacements only affected 54.3%; 41.9% of patients presented a displaced WOI and were predicted to have a poor endometrial prognosis, whereas only 12.4% of patients presented with a displaced WOI and were predicted to have a good endometrial prognosis. These results offer a possible explanation as to why other studies reported there was no clinical benefito f personalizing embryo transfers on the basis of transcriptomic dating of the endometrium . According to our clinical algorithm, only patients with a displaced WOI and a good endometrial prognosis (12.4% of our cohort) would be better off with this approach. Patients with poor endometrial prognosis (73.7% of our cohort) would have suboptimal reproductive outcomes even though their WOI was on-time. This highlights an unmet healthcare need for IVF patients with endometrial disruptions and underscores the importance of additional prospective studies to further characterize patients on the basis of these 2 molecular causes for trying tailored treatments. Contemplating endometrial luteal phase timing together with a prevalent disruption or poor endometrial prognosis shifts the paradigm in studies about the endometrium. Our innovative methodology based on transcriptomics and AI models led us to develop new endometrial biomarker taxonomies related to endometrial luteal phase timing and/or endometrial disruption . There were 30 genes overlapping between the previously published TED signature and the EFR signature presented herein, emphasizing their correlation despite the lower FC of biomarkers for endometrial disruption versus endometrial timing . AI algorithms offer powerful approaches for patient stratification, by combining high-throughput molecular data and clinical variables . Clinical translation of these approaches will require careful supervision of how the machine learning is initially trained . However, a semi-supervised learning methodology implementing accurate clinical diagnostic criteria and patient-specific genomic data to establish relevant molecular taxonomies will be indispensable for advancing precision medicine , especially within the context of endometrial-factor infertility. Predictive biomarkers can stratify patients with endometrial failure even if the number of their attempts was insufficient to meet clinical RIF classification criteria . To our knowledge, this is thefirst study to apply semi-supervised learning to the endometrium, conferring an advantage over previous models that were only trained with clinical classifications and did not leverage the underlying molecular heterogeneity of the condition . The lack of gene expression differences between RIF and control patients with our initial clinical classification reinforced findings from Macklon’s group and the molecular heterogeneity among patients with RIF . The EFR signature comprised 122 genes, including 44 genes newly associated with an endometrial gene expression signature and a single common gene with Koot’s pioneer endometrial timing-independent signature for endometrial pathology . Here, we only considered biopsies collected in the endometrial mid-secretory phase, when patients had received P4 for 82- 172 h. Although this signature is independent of endometrial luteal phase timing, further studies are required to confirm if the signature is detectable in other phases of the endometrial cycle. Progestagen-associated endometrial protein is well characterized in endometrial tissue and was the most upregulated gene (FC1⁄4 3.18) in the poor endometrial prognosis pro file. However, the levels we observed were 10-fold lower compared with previous studies where changes in endometrial timing were considered . Alternatively,fibrinogen beta chain, related to the plasma membrane and vesicle transport by secretory granules, was the most downregulated gene. This gene accumulated polymorphisms that hinder the production offibrinogen and its associated anti-inflammatory activity and was recently related to the dysregulated pathways in RIF . FOXP1, the most predictive gene, is a transcription factor mediated by estrogen that has established functions in regulation, leukocyte differentiation, and response to lipids and chemokines. This gene is associated with breast endometrial cancer , and endometriosis-related fibrosis , but was never included in an endometrial gene signature until now. With FOXP1 and the FOX transcription factor family being associated with the regulation of the menstrual cycle and implantation , this evidence reinforces our prior study and affirms that alterations in endometrial regulation are an underlying cause of endometrial failure. Our clinical algorithm considers the 2 WOI factors, endometrial timing and disruption, to stratify IVF patients on the basis of 4 possible phenotypes: displaced/on-time and poor/ good prognosis. This model is promising for preventive and precision medicine in endometrial-factor infertility, as it combines the robust prediction parameters of the TED model and the EFR biomarker signature’s predictive capability reported herein. In contrast to other transcriptomic-based endometrial dating tools , our bioinformatic methodology removes the molecular variation associated with the cyclic menstrual changes, providing deeper insights for identifying a new kind of disruption independent of endometrial luteal phase timing by the EFR signature. Furthermore, the EFR biomarker signature was developed on the basis of reproductive outcomes and effectively classifies patients predicted to have positive or negative reproductive outcomes in the first SET after biopsy collection. All these facts, point out to a more potential clinical utility than existing tools that only predict the WOI without showing significant differences in reproductive outcomes as have been reported and we have shown in ourfindings. Although our strategy appears promising, the classification presented herein needs further re finement before clinical implementation. Larger sample sizes and validation in independent samples are required to ensure model generalization . Healthcare professionals will need to be adequately trained to ensure technical reproducibility. Longitudinal assessments including several menstrual cycles in the same patient are recommended to confirm inter-cycle reproducibility and to gain a comprehensive understanding of the reliability of the EFR panel. Additional studies should assess the relationship between a poor endometrial prognosis and the presence of benign disorders that potentially compromise endometrial function and/or embryo implantation . Inspired by the MammaPrint risk signature for breast cancer , the EFR signature could become a nextgeneration tool used to classify patients on the basis of endometrial prognosis, to predict reproductive outcomes before undergoing IVF, potentially experiencing RIF and/or losing valuable embryos. Although treatments for patients with poor endometrial prognosis remain to be determined, clinically identifying these patients is the first step toward improving their care.',\n",
       "  'Conclusion': '',\n",
       "  'URL': 'https://doi.org/10.1016/j.fertnstert.2024.03.015',\n",
       "  'Journal': 'Fertil Steril'}]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "with open(\"info_articles.pkl\",\"rb\") as f:\n",
    "    my_json = pkl.load(f)\n",
    "my_json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "83dd0697",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-10 19:08:04,763 - DEBUG - pubmed2pdf.cli - Full log mode activated\n",
      "2025-08-10 19:08:04,764 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16832043\n",
      "2025-08-10 19:08:05,439 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16832043 failed from error list index out of range\n",
      "2025-08-10 19:08:05,440 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20729534\n",
      "2025-08-10 19:08:06,127 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 20729534 failed from error list index out of range\n",
      "2025-08-10 19:08:06,127 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30929718\n",
      "2025-08-10 19:08:07,885 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30929718 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:07,885 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16306079\n",
      "2025-08-10 19:08:08,525 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16306079 failed from error list index out of range\n",
      "2025-08-10 19:08:08,525 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29102484\n",
      "2025-08-10 19:08:09,753 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29102484 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:09,753 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30624659\n",
      "2025-08-10 19:08:10,451 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30624659 failed from error list index out of range\n",
      "2025-08-10 19:08:10,451 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29452422\n",
      "2025-08-10 19:08:12,160 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29452422 failed from error list index out of range\n",
      "2025-08-10 19:08:12,160 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23994285\n",
      "2025-08-10 19:08:13,535 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23994285 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:13,535 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23290997\n",
      "2025-08-10 19:08:14,874 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23290997 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:14,874 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30951376\n",
      "2025-08-10 19:08:15,703 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30951376 failed from error list index out of range\n",
      "2025-08-10 19:08:15,703 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21849299\n",
      "2025-08-10 19:08:16,383 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 21849299 failed from error list index out of range\n",
      "2025-08-10 19:08:16,383 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24882617\n",
      "2025-08-10 19:08:17,702 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24882617 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:17,702 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18192189\n",
      "2025-08-10 19:08:18,412 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18192189 failed from error list index out of range\n",
      "2025-08-10 19:08:18,413 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31462323\n",
      "2025-08-10 19:08:18,413 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12200466\n",
      "2025-08-10 19:08:19,088 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12200466 failed from error list index out of range\n",
      "2025-08-10 19:08:19,088 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12529417\n",
      "2025-08-10 19:08:19,769 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12529417 failed from error list index out of range\n",
      "2025-08-10 19:08:19,769 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15878921\n",
      "2025-08-10 19:08:20,486 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15878921 failed from error list index out of range\n",
      "2025-08-10 19:08:20,486 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35085395\n",
      "2025-08-10 19:08:21,172 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35085395 failed from error list index out of range\n",
      "2025-08-10 19:08:21,172 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19933690\n",
      "2025-08-10 19:08:21,846 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19933690 failed from error list index out of range\n",
      "2025-08-10 19:08:21,846 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33576824\n",
      "2025-08-10 19:08:22,530 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33576824 failed from error list index out of range\n",
      "2025-08-10 19:08:22,530 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19617889\n",
      "2025-08-10 19:08:24,931 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10592173\n",
      "2025-08-10 19:08:25,615 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10592173 failed from error list index out of range\n",
      "2025-08-10 19:08:25,615 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10802651\n",
      "2025-08-10 19:08:27,841 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31340985\n",
      "2025-08-10 19:08:27,841 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29156006\n",
      "2025-08-10 19:08:28,499 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29156006 failed from error list index out of range\n",
      "2025-08-10 19:08:28,499 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26415722\n",
      "2025-08-10 19:08:29,188 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26415722 failed from error list index out of range\n",
      "2025-08-10 19:08:29,189 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26395145\n",
      "2025-08-10 19:08:31,089 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26395145 failed from error list index out of range\n",
      "2025-08-10 19:08:31,089 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32641214\n",
      "2025-08-10 19:08:32,434 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32641214 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:32,434 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25605792\n",
      "2025-08-10 19:08:33,168 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 25605792 failed from error list index out of range\n",
      "2025-08-10 19:08:33,169 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11846609\n",
      "2025-08-10 19:08:34,466 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 11846609 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:34,467 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28855728\n",
      "2025-08-10 19:08:34,467 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20619403\n",
      "2025-08-10 19:08:35,676 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 20619403 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:35,676 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12021176\n",
      "2025-08-10 19:08:36,360 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12021176 failed from error list index out of range\n",
      "2025-08-10 19:08:36,360 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15501903\n",
      "2025-08-10 19:08:37,055 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15501903 failed from error list index out of range\n",
      "2025-08-10 19:08:37,055 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15666095\n",
      "2025-08-10 19:08:39,277 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12728018\n",
      "2025-08-10 19:08:40,406 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12728018 failed from error list index out of range\n",
      "2025-08-10 19:08:40,406 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30081718\n",
      "2025-08-10 19:08:44,076 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18539642\n",
      "2025-08-10 19:08:44,752 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18539642 failed from error list index out of range\n",
      "2025-08-10 19:08:44,752 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23555582\n",
      "2025-08-10 19:08:44,752 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22025212\n",
      "2025-08-10 19:08:45,688 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 22025212 failed from error list index out of range\n",
      "2025-08-10 19:08:45,688 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28523980\n",
      "2025-08-10 19:08:46,432 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28523980 failed from error list index out of range\n",
      "2025-08-10 19:08:46,432 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16672246\n",
      "2025-08-10 19:08:47,081 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16672246 failed from error list index out of range\n",
      "2025-08-10 19:08:47,081 - INFO - pubmed2pdf.cli - Trying to fetch pmid 7962413\n",
      "2025-08-10 19:08:47,773 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 7962413 failed from error list index out of range\n",
      "2025-08-10 19:08:47,773 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23933037\n",
      "2025-08-10 19:08:48,966 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23933037 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:48,966 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26804062\n",
      "2025-08-10 19:08:49,829 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26804062 failed from error list index out of range\n",
      "2025-08-10 19:08:49,829 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32482256\n",
      "2025-08-10 19:08:51,058 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32482256 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:51,058 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35147192\n",
      "2025-08-10 19:08:51,723 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35147192 failed from error list index out of range\n",
      "2025-08-10 19:08:51,723 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22902743\n",
      "2025-08-10 19:08:54,278 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29663566\n",
      "2025-08-10 19:08:55,126 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29663566 failed from error list index out of range\n",
      "2025-08-10 19:08:55,126 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28229988\n",
      "2025-08-10 19:08:55,995 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28229988 failed from error list index out of range\n",
      "2025-08-10 19:08:55,996 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30037059\n",
      "2025-08-10 19:08:56,737 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30037059 failed from error list index out of range\n",
      "2025-08-10 19:08:56,738 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22911744\n",
      "2025-08-10 19:08:56,738 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34117589\n",
      "2025-08-10 19:08:59,145 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31511876\n",
      "2025-08-10 19:08:59,836 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 31511876 failed from error list index out of range\n",
      "2025-08-10 19:08:59,836 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24100212\n",
      "2025-08-10 19:09:00,550 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24100212 failed from error list index out of range\n",
      "2025-08-10 19:09:00,550 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16807381\n",
      "2025-08-10 19:09:01,210 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16807381 failed from error list index out of range\n",
      "2025-08-10 19:09:01,210 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33830236\n",
      "2025-08-10 19:09:02,014 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33830236 failed from error list index out of range\n",
      "2025-08-10 19:09:02,014 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21964334\n",
      "2025-08-10 19:09:04,499 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19563753\n",
      "2025-08-10 19:09:05,762 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19563753 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:09:05,763 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35574918\n",
      "2025-08-10 19:09:06,450 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35574918 failed from error list index out of range\n",
      "2025-08-10 19:09:06,450 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18174356\n",
      "2025-08-10 19:09:07,116 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18174356 failed from error list index out of range\n",
      "2025-08-10 19:09:07,116 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26932670\n",
      "2025-08-10 19:09:09,248 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26932670 failed from error list index out of range\n",
      "2025-08-10 19:09:09,249 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29402198\n",
      "2025-08-10 19:09:13,253 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15550397\n",
      "2025-08-10 19:09:14,524 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15550397 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:09:14,524 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35088888\n",
      "2025-08-10 19:09:14,524 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19890323\n",
      "2025-08-10 19:09:18,103 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19815776\n",
      "2025-08-10 19:09:18,995 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19815776 failed from error argument of type 'NoneType' is not iterable\n",
      "2025-08-10 19:09:18,996 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17571346\n",
      "2025-08-10 19:09:18,996 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32043035\n",
      "2025-08-10 19:09:20,438 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32043035 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:09:20,438 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36205711\n",
      "2025-08-10 19:09:21,104 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36205711 failed from error list index out of range\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in Data2/\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-10 19:09:48,571 - DEBUG - pubmed2pdf.cli - Full log mode activated\n",
      "2025-08-10 19:09:48,572 - INFO - pubmed2pdf.cli - Trying to fetch pmid 1155504\n",
      "2025-08-10 19:09:49,824 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 1155504 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:09:49,825 - INFO - pubmed2pdf.cli - Trying to fetch pmid 1424330\n",
      "2025-08-10 19:09:50,989 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 1424330 failed from error Invalid URL 'wenmhi4L9dPogJ7fBmofTSqFqabAb1VqRbFzzuvnOkth6PgfEnsViH5uoEhxes7n': No scheme supplied. Perhaps you meant https://wenmhi4L9dPogJ7fBmofTSqFqabAb1VqRbFzzuvnOkth6PgfEnsViH5uoEhxes7n?\n",
      "2025-08-10 19:09:50,989 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10362823\n",
      "2025-08-10 19:09:51,701 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10362823 failed from error list index out of range\n",
      "2025-08-10 19:09:51,701 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15353123\n",
      "2025-08-10 19:09:53,993 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16306079\n",
      "2025-08-10 19:09:54,951 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16306079 failed from error list index out of range\n",
      "2025-08-10 19:09:54,952 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10221616\n",
      "2025-08-10 19:09:56,722 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10221616 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:09:56,722 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28863933\n",
      "2025-08-10 19:09:57,980 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28863933 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:09:57,980 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28923940\n",
      "2025-08-10 19:09:58,644 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28923940 failed from error list index out of range\n",
      "2025-08-10 19:09:58,644 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33313697\n",
      "2025-08-10 19:09:59,315 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33313697 failed from error list index out of range\n",
      "2025-08-10 19:09:59,315 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36822566\n",
      "2025-08-10 19:10:00,634 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36822566 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:00,635 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24269084\n",
      "2025-08-10 19:10:02,258 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24269084 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:02,258 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24581986\n",
      "2025-08-10 19:10:03,662 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24581986 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:03,663 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30624659\n",
      "2025-08-10 19:10:04,327 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30624659 failed from error list index out of range\n",
      "2025-08-10 19:10:04,327 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33077239\n",
      "2025-08-10 19:10:05,586 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33077239 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:05,586 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26797113\n",
      "2025-08-10 19:10:05,586 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33576824\n",
      "2025-08-10 19:10:06,314 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33576824 failed from error list index out of range\n",
      "2025-08-10 19:10:06,314 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34875061\n",
      "2025-08-10 19:10:07,005 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 34875061 failed from error list index out of range\n",
      "2025-08-10 19:10:07,005 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29452422\n",
      "2025-08-10 19:10:07,731 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29452422 failed from error list index out of range\n",
      "2025-08-10 19:10:07,731 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20619403\n",
      "2025-08-10 19:10:09,094 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 20619403 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:09,094 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23102856\n",
      "2025-08-10 19:10:10,409 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23102856 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:10,409 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34199109\n",
      "2025-08-10 19:10:11,104 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 34199109 failed from error list index out of range\n",
      "2025-08-10 19:10:11,104 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12421895\n",
      "2025-08-10 19:10:11,886 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12421895 failed from error list index out of range\n",
      "2025-08-10 19:10:11,886 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22256780\n",
      "2025-08-10 19:10:12,603 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 22256780 failed from error list index out of range\n",
      "2025-08-10 19:10:12,603 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24581625\n",
      "2025-08-10 19:10:13,921 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24581625 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:13,922 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30145981\n",
      "2025-08-10 19:10:13,922 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30713991\n",
      "2025-08-10 19:10:15,115 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36070983\n",
      "2025-08-10 19:10:16,363 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36070983 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:16,364 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32974805\n",
      "2025-08-10 19:10:19,091 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32723696\n",
      "2025-08-10 19:10:20,356 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32723696 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:20,356 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33880419\n",
      "2025-08-10 19:10:21,406 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33067123\n",
      "2025-08-10 19:10:22,521 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33067123 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:22,522 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29737471\n",
      "2025-08-10 19:10:25,069 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33880420\n",
      "2025-08-10 19:10:26,202 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36472596\n",
      "2025-08-10 19:10:26,931 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36472596 failed from error list index out of range\n",
      "2025-08-10 19:10:26,931 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11752295\n",
      "2025-08-10 19:10:27,630 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 11752295 failed from error list index out of range\n",
      "2025-08-10 19:10:27,631 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25605792\n",
      "2025-08-10 19:10:28,314 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 25605792 failed from error list index out of range\n",
      "2025-08-10 19:10:28,314 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17160063\n",
      "2025-08-10 19:10:30,583 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-10 19:10:31,377 - DEBUG - pubmed2pdf.cli - ** fetching of reprint None failed from error list index out of range\n",
      "2025-08-10 19:10:31,377 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-10 19:10:32,180 - DEBUG - pubmed2pdf.cli - ** fetching of reprint None failed from error list index out of range\n",
      "2025-08-10 19:10:32,180 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-10 19:10:32,934 - DEBUG - pubmed2pdf.cli - ** fetching of reprint None failed from error list index out of range\n",
      "2025-08-10 19:10:32,934 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10592173\n",
      "2025-08-10 19:10:33,653 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10592173 failed from error list index out of range\n",
      "2025-08-10 19:10:33,653 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10802651\n",
      "2025-08-10 19:10:35,589 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18546601\n",
      "2025-08-10 19:10:37,781 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26109056\n",
      "2025-08-10 19:10:37,781 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33036008\n",
      "2025-08-10 19:10:39,487 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33036008 failed from error Invalid URL '': No scheme supplied. Perhaps you meant https://?\n",
      "2025-08-10 19:10:39,488 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28370781\n",
      "2025-08-10 19:10:40,447 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28370781 failed from error list index out of range\n",
      "2025-08-10 19:10:40,447 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34947871\n",
      "2025-08-10 19:10:41,158 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 34947871 failed from error list index out of range\n",
      "2025-08-10 19:10:41,158 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35929523\n",
      "2025-08-10 19:10:42,011 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35929523 failed from error list index out of range\n",
      "2025-08-10 19:10:42,011 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27441287\n",
      "2025-08-10 19:10:43,254 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 27441287 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:43,254 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30452456\n",
      "2025-08-10 19:10:43,254 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31540845\n",
      "2025-08-10 19:10:44,502 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 31540845 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:44,502 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33830236\n",
      "2025-08-10 19:10:45,192 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33830236 failed from error list index out of range\n",
      "2025-08-10 19:10:45,192 - INFO - pubmed2pdf.cli - Trying to fetch pmid 37700285\n",
      "2025-08-10 19:10:45,192 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29315421\n",
      "2025-08-10 19:10:45,927 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29315421 failed from error list index out of range\n",
      "2025-08-10 19:10:45,928 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28855728\n",
      "2025-08-10 19:10:45,928 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35092277\n",
      "2025-08-10 19:10:46,606 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35092277 failed from error list index out of range\n",
      "2025-08-10 19:10:46,606 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32482256\n",
      "2025-08-10 19:10:47,970 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32482256 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:47,970 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11823860\n",
      "2025-08-10 19:10:50,115 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12490681\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in Data2/\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-10 19:10:51,291 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12490681 failed from error list index out of range\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'Deciphering a shared transcriptomic regulation and the relative contribution of each regulator type through endometrial gene expression signatures': ['16832043',\n",
       "  '20729534',\n",
       "  '30929718',\n",
       "  '16306079',\n",
       "  '29102484',\n",
       "  '30624659',\n",
       "  '29452422',\n",
       "  '23994285',\n",
       "  '23290997',\n",
       "  '30951376',\n",
       "  '21849299',\n",
       "  '24882617',\n",
       "  '18192189',\n",
       "  '31462323',\n",
       "  '12200466',\n",
       "  '12529417',\n",
       "  '15878921',\n",
       "  '35085395',\n",
       "  '19933690',\n",
       "  '33576824',\n",
       "  '19617889',\n",
       "  '10592173',\n",
       "  '10802651',\n",
       "  '31340985',\n",
       "  '29156006',\n",
       "  '26415722',\n",
       "  '26395145',\n",
       "  '32641214',\n",
       "  '25605792',\n",
       "  '11846609',\n",
       "  '28855728',\n",
       "  '20619403',\n",
       "  '12021176',\n",
       "  '15501903',\n",
       "  '15666095',\n",
       "  '12728018',\n",
       "  '30081718',\n",
       "  '18539642',\n",
       "  '23555582',\n",
       "  '22025212',\n",
       "  '28523980',\n",
       "  '16672246',\n",
       "  '7962413',\n",
       "  '23933037',\n",
       "  '26804062',\n",
       "  '32482256',\n",
       "  '35147192',\n",
       "  '22902743',\n",
       "  '29663566',\n",
       "  '28229988',\n",
       "  '30037059',\n",
       "  '22911744',\n",
       "  '34117589',\n",
       "  '31511876',\n",
       "  '24100212',\n",
       "  '16807381',\n",
       "  '33830236',\n",
       "  '21964334',\n",
       "  '19563753',\n",
       "  '35574918',\n",
       "  '18174356',\n",
       "  '26932670',\n",
       "  '29402198',\n",
       "  '15550397',\n",
       "  '35088888',\n",
       "  '19890323',\n",
       "  '19815776',\n",
       "  '17571346',\n",
       "  '32043035',\n",
       "  '36205711'],\n",
       " 'Predicting risk of endometrial failure: a biomarker signature that identifies a novel disruption independent of endometrial timing in patients undergoing hormonal replacement cycles': ['1155504',\n",
       "  '1424330',\n",
       "  '10362823',\n",
       "  '15353123',\n",
       "  '16306079',\n",
       "  '10221616',\n",
       "  '28863933',\n",
       "  '28923940',\n",
       "  '33313697',\n",
       "  '36822566',\n",
       "  '24269084',\n",
       "  '24581986',\n",
       "  '30624659',\n",
       "  '33077239',\n",
       "  '26797113',\n",
       "  '33576824',\n",
       "  '34875061',\n",
       "  '29452422',\n",
       "  '20619403',\n",
       "  '23102856',\n",
       "  '34199109',\n",
       "  '12421895',\n",
       "  '22256780',\n",
       "  '24581625',\n",
       "  '30145981',\n",
       "  '30713991',\n",
       "  '36070983',\n",
       "  '32974805',\n",
       "  '32723696',\n",
       "  '33880419',\n",
       "  '33067123',\n",
       "  '29737471',\n",
       "  '33880420',\n",
       "  '36472596',\n",
       "  '11752295',\n",
       "  '25605792',\n",
       "  '17160063',\n",
       "  'None',\n",
       "  'None',\n",
       "  'None',\n",
       "  '10592173',\n",
       "  '10802651',\n",
       "  '18546601',\n",
       "  '26109056',\n",
       "  '33036008',\n",
       "  '28370781',\n",
       "  '34947871',\n",
       "  '35929523',\n",
       "  '27441287',\n",
       "  '30452456',\n",
       "  '31540845',\n",
       "  '33830236',\n",
       "  '37700285',\n",
       "  '29315421',\n",
       "  '28855728',\n",
       "  '35092277',\n",
       "  '32482256',\n",
       "  '11823860',\n",
       "  '12490681']}"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pmid = {}\n",
    "url_pmc = f\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC{a}/\"\n",
    "\n",
    "for article in my_json:\n",
    "\n",
    "    pmid[article['PaperTitle']] = []\n",
    "\n",
    "    if 'https://' in my_json[0]['URL']:\n",
    "        doi = re.sub(r'https://doi.org/', '', article['URL'])\n",
    "        response = requests.get('https://api.crossref.org/works/'+doi)\n",
    "        list_reference = response.json()['message'].get('reference',None)\n",
    "        article['List_references'] = [x.get('DOI') for x in list_reference if x.get('DOI')]\n",
    "\n",
    "        # We retrieve the pmid from each DOI\n",
    "\n",
    "        for a in  article['List_references']:\n",
    "            \n",
    "            pmid[article['PaperTitle']].append(get_pmid_from_doi(a))\n",
    "\n",
    "        # We try to get the pdf\n",
    "\n",
    "        command = [\n",
    "        \"python3\", \"-m\", \"pubmed2pdf\", \"pdf\",\n",
    "        \"--out\", \"Data2/\",\n",
    "        \"--pmids\", f\"{', '.join(pmid[article['PaperTitle']])}\",\n",
    "        \"--verbose\",\n",
    "        \"--errors\", \"Data2/pubmed2pdf_log.txt\"\n",
    "        ]\n",
    "        subprocess.run(command)\n",
    "\n",
    "        # We try to get the html otherwise\n",
    "\n",
    "        to_retrieve = pd.read_csv(\"Data2/pubmed2pdf_log.txt\", header=None,dtype='str') # We get those that resulted in a mistake\n",
    "        to_retrieve.columns = [\"PMID\"]\n",
    "        to_retrieve = to_retrieve['PMID'].tolist()\n",
    "        to_retrieve.extend([re.sub(r'.html','',f)  for f in os.listdir('./Data2') if f.endswith('.html')]) # We add those in html format\n",
    "        [os.remove(os.path.join('Data2',f))  for f in os.listdir('./Data2') if f.endswith('.html')] # Then, we remove from directory\n",
    "        \n",
    "        pmcid = []\n",
    "        for pmid in to_retrieve:\n",
    "            pmcid.append(get_pmcid_from_pmid(pmid))\n",
    "        \n",
    "        response = requests.get(url_pmc, headers=HEADERS)\n",
    "        soup = BeautifulSoup(response.text, \"html.parser\")\n",
    "        soup.find(\"article\").get_text(separator=\"\\n\", strip=True)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "67b5e88b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "len(article['List_references'])\n",
    "[os.remove(os.path.join('Data2',f))  for f in os.listdir('./Data2') if f.endswith('.html')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "a4ed29bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_pmid_from_doi(doi, api_key=None):\n",
    "    url = \"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi\"\n",
    "    params = {\n",
    "        \"db\": \"pubmed\",\n",
    "        \"term\": f\"{doi}[DOI]\",\n",
    "        \"retmode\": \"json\"\n",
    "    }\n",
    "    if api_key:\n",
    "        params[\"api_key\"] = api_key\n",
    "\n",
    "    response = requests.get(url, params=params)\n",
    "    result = response.json()\n",
    "    pmids = result.get(\"esearchresult\", {}).get(\"idlist\", [])\n",
    "    return pmids[0] if pmids else 'None'\n",
    "\n",
    "\n",
    "\n",
    "def get_pmcid_from_pmid(pmid, api_key=None):\n",
    "    url = \"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi\"\n",
    "    params = {\n",
    "        \"dbfrom\": \"pubmed\",\n",
    "        \"db\": \"pmc\",\n",
    "        \"id\": pmid,\n",
    "        \"retmode\": \"json\"\n",
    "    }\n",
    "    if api_key:\n",
    "        params[\"api_key\"] = api_key\n",
    "\n",
    "    response = requests.get(url, params=params)\n",
    "    data = response.json()\n",
    "\n",
    "    try:\n",
    "        linksets = data[\"linksets\"]\n",
    "        if linksets and \"linksetdbs\" in linksets[0]:\n",
    "            pmcid_link = linksets[0][\"linksetdbs\"][0][\"links\"][0]\n",
    "            return pmcid_link\n",
    "        else:\n",
    "            return None\n",
    "    except Exception:\n",
    "        return None\n",
    "    \n",
    "def download_pmc_pdf(pmcid, filename=None):\n",
    "    pmc_number = pmcid.replace(\"PMC\", \"\")\n",
    "    pdf_url = f\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC{pmc_number}/pdf/\"\n",
    "\n",
    "    headers = {\n",
    "        \"User-Agent\": (\n",
    "            \"Mozilla/5.0 (Windows NT 10.0; Win64; x64) \"\n",
    "            \"AppleWebKit/537.36 (KHTML, like Gecko) \"\n",
    "            \"Chrome/115.0.0.0 Safari/537.36\"\n",
    "        )\n",
    "    }\n",
    "\n",
    "    response = requests.get(pdf_url, headers=headers)\n",
    "    if \"pdf\" in response.headers.get(\"Content-Type\", \"\"):\n",
    "        if not filename:\n",
    "            filename = f\"{pmcid}.pdf\"\n",
    "        with open(filename, \"wb\") as f:\n",
    "            f.write(response.content)\n",
    "        print(f\"✅ PDF downloaded as {filename}\")\n",
    "    else:\n",
    "        print(f\"⚠️ PDF not directly available at {pdf_url}\")\n",
    "        \n",
    "\n",
    "def download_pdf_from_pmid(pmid, api_key=None):\n",
    "    print(f\"🔎 Checking PMC for PMID: {pmid}\")\n",
    "    pmcid = get_pmcid_from_pmid(pmid, api_key)\n",
    "    if pmcid:\n",
    "        print(f\"✅ Found PMC ID: {pmcid}\")\n",
    "        download_pmc_pdf(pmcid)\n",
    "    else:\n",
    "        print(\"❌ No PMC version available for this PubMed article.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "913d8255",
   "metadata": {},
   "outputs": [],
   "source": [
    "pmids = []\n",
    "for article in my_json:\n",
    "    for doi in article['List_references']:\n",
    "        pmid = get_pmid_from_doi(doi=doi)\n",
    "        pmids.append(pmid)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9a03814",
   "metadata": {},
   "outputs": [],
   "source": [
    "for pmid in list(set(pmids)):\n",
    "    with open('Data2/pmids.txt',\"a\") as f:\n",
    "        f.write(pmid+'\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ea799d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-02 16:38:25,860 - DEBUG - pubmed2pdf.cli - Full log mode activated\n",
      "2025-08-02 16:38:25,861 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28923940\n",
      "2025-08-02 16:38:26,583 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28923940 failed from error list index out of range\n",
      "2025-08-02 16:38:26,583 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18539642\n",
      "2025-08-02 16:38:27,254 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18539642 failed from error list index out of range\n",
      "2025-08-02 16:38:27,254 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26415722\n",
      "2025-08-02 16:38:28,786 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26415722 failed from error list index out of range\n",
      "2025-08-02 16:38:28,786 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31540845\n",
      "2025-08-02 16:38:30,006 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 31540845 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:30,006 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16306079\n",
      "2025-08-02 16:38:30,721 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16306079 failed from error list index out of range\n",
      "2025-08-02 16:38:30,721 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22902743\n",
      "2025-08-02 16:38:32,782 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22025212\n",
      "2025-08-02 16:38:33,661 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 22025212 failed from error list index out of range\n",
      "2025-08-02 16:38:33,661 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19563753\n",
      "2025-08-02 16:38:34,930 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19563753 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:34,930 - INFO - pubmed2pdf.cli - Trying to fetch pmid 1155504\n",
      "2025-08-02 16:38:36,264 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 1155504 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:36,264 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29452422\n",
      "2025-08-02 16:38:36,938 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29452422 failed from error list index out of range\n",
      "2025-08-02 16:38:36,938 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34199109\n",
      "2025-08-02 16:38:37,583 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 34199109 failed from error list index out of range\n",
      "2025-08-02 16:38:37,583 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33077239\n",
      "2025-08-02 16:38:39,043 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33077239 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:39,043 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11846609\n",
      "2025-08-02 16:38:40,418 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 11846609 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:40,418 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12490681\n",
      "2025-08-02 16:38:41,113 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12490681 failed from error list index out of range\n",
      "2025-08-02 16:38:41,113 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26932670\n",
      "2025-08-02 16:38:41,826 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26932670 failed from error list index out of range\n",
      "2025-08-02 16:38:41,826 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18192189\n",
      "2025-08-02 16:38:42,538 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18192189 failed from error list index out of range\n",
      "2025-08-02 16:38:42,538 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28855728\n",
      "2025-08-02 16:38:42,538 - INFO - pubmed2pdf.cli - Trying to fetch pmid 37700285\n",
      "2025-08-02 16:38:42,538 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32641214\n",
      "2025-08-02 16:38:43,806 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32641214 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:43,806 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12529417\n",
      "2025-08-02 16:38:44,447 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12529417 failed from error list index out of range\n",
      "2025-08-02 16:38:44,447 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23290997\n",
      "2025-08-02 16:38:45,826 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23290997 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:45,827 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30713991\n",
      "2025-08-02 16:38:47,005 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15353123\n",
      "2025-08-02 16:38:49,056 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24100212\n",
      "2025-08-02 16:38:49,720 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24100212 failed from error list index out of range\n",
      "2025-08-02 16:38:49,720 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10802651\n",
      "2025-08-02 16:38:51,745 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23102856\n",
      "2025-08-02 16:38:53,923 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23102856 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:53,923 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32043035\n",
      "2025-08-02 16:38:55,160 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32043035 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:55,160 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32482256\n",
      "2025-08-02 16:38:56,401 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32482256 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:56,401 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36070983\n",
      "2025-08-02 16:38:57,955 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36070983 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:57,955 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16807381\n",
      "2025-08-02 16:38:58,613 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16807381 failed from error list index out of range\n",
      "2025-08-02 16:38:58,613 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11752295\n",
      "2025-08-02 16:38:59,291 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 11752295 failed from error list index out of range\n",
      "2025-08-02 16:38:59,291 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33067123\n",
      "2025-08-02 16:39:01,747 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33067123 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:01,747 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26797113\n",
      "2025-08-02 16:39:01,747 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18546601\n",
      "2025-08-02 16:39:03,851 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36205711\n",
      "2025-08-02 16:39:04,539 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36205711 failed from error list index out of range\n",
      "2025-08-02 16:39:04,539 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35929523\n",
      "2025-08-02 16:39:05,385 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35929523 failed from error list index out of range\n",
      "2025-08-02 16:39:05,385 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16832043\n",
      "2025-08-02 16:39:06,266 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16832043 failed from error list index out of range\n",
      "2025-08-02 16:39:06,266 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11823860\n",
      "2025-08-02 16:39:08,266 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21849299\n",
      "2025-08-02 16:39:08,931 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 21849299 failed from error list index out of range\n",
      "2025-08-02 16:39:08,931 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19617889\n",
      "2025-08-02 16:39:11,269 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17160063\n",
      "2025-08-02 16:39:13,388 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31462323\n",
      "2025-08-02 16:39:13,388 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10362823\n",
      "2025-08-02 16:39:14,060 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10362823 failed from error list index out of range\n",
      "2025-08-02 16:39:14,060 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31340985\n",
      "2025-08-02 16:39:14,060 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29156006\n",
      "2025-08-02 16:39:14,717 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29156006 failed from error list index out of range\n",
      "2025-08-02 16:39:14,717 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35574918\n",
      "2025-08-02 16:39:15,407 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35574918 failed from error list index out of range\n",
      "2025-08-02 16:39:15,408 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20619403\n",
      "2025-08-02 16:39:16,743 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 20619403 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:16,743 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17571346\n",
      "2025-08-02 16:39:16,743 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30452456\n",
      "2025-08-02 16:39:16,743 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29102484\n",
      "2025-08-02 16:39:17,960 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29102484 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:17,960 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16672246\n",
      "2025-08-02 16:39:18,631 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16672246 failed from error list index out of range\n",
      "2025-08-02 16:39:18,631 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15501903\n",
      "2025-08-02 16:39:19,290 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15501903 failed from error list index out of range\n",
      "2025-08-02 16:39:19,290 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10221616\n",
      "2025-08-02 16:39:20,660 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10221616 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:20,661 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33576824\n",
      "2025-08-02 16:39:21,352 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33576824 failed from error list index out of range\n",
      "2025-08-02 16:39:21,352 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32723696\n",
      "2025-08-02 16:39:23,654 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32723696 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:23,654 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12200466\n",
      "2025-08-02 16:39:24,318 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12200466 failed from error list index out of range\n",
      "2025-08-02 16:39:24,318 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15666095\n",
      "2025-08-02 16:39:26,541 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35085395\n",
      "2025-08-02 16:39:27,203 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35085395 failed from error list index out of range\n",
      "2025-08-02 16:39:27,203 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30145981\n",
      "2025-08-02 16:39:27,203 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26109056\n",
      "2025-08-02 16:39:27,203 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33036008\n",
      "2025-08-02 16:39:28,940 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33036008 failed from error Invalid URL '': No scheme supplied. Perhaps you meant https://?\n",
      "2025-08-02 16:39:28,941 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24882617\n",
      "2025-08-02 16:39:30,142 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24882617 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:30,142 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30951376\n",
      "2025-08-02 16:39:31,008 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30951376 failed from error list index out of range\n",
      "2025-08-02 16:39:31,008 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19815776\n",
      "2025-08-02 16:39:32,019 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19815776 failed from error argument of type 'NoneType' is not iterable\n",
      "2025-08-02 16:39:32,019 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27441287\n",
      "2025-08-02 16:39:33,538 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 27441287 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:33,538 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21964334\n",
      "2025-08-02 16:39:37,074 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36822566\n",
      "2025-08-02 16:39:39,308 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36822566 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:39,308 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12421895\n",
      "2025-08-02 16:39:39,966 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12421895 failed from error list index out of range\n",
      "2025-08-02 16:39:39,966 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26804062\n",
      "2025-08-02 16:39:40,823 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26804062 failed from error list index out of range\n",
      "2025-08-02 16:39:40,823 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30929718\n",
      "2025-08-02 16:39:42,075 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30929718 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:42,075 - INFO - pubmed2pdf.cli - Trying to fetch pmid 7962413\n",
      "2025-08-02 16:39:42,960 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 7962413 failed from error list index out of range\n",
      "2025-08-02 16:39:42,960 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28863933\n",
      "2025-08-02 16:39:44,131 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28863933 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:44,132 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31511876\n",
      "2025-08-02 16:39:44,747 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 31511876 failed from error list index out of range\n",
      "2025-08-02 16:39:44,747 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35088888\n",
      "2025-08-02 16:39:44,747 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30624659\n",
      "2025-08-02 16:39:45,419 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30624659 failed from error list index out of range\n",
      "2025-08-02 16:39:45,419 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12728018\n",
      "2025-08-02 16:39:46,072 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12728018 failed from error list index out of range\n",
      "2025-08-02 16:39:46,073 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12021176\n",
      "2025-08-02 16:39:46,742 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12021176 failed from error list index out of range\n",
      "2025-08-02 16:39:46,743 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34875061\n",
      "2025-08-02 16:39:47,438 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 34875061 failed from error list index out of range\n",
      "2025-08-02 16:39:47,439 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23555582\n",
      "2025-08-02 16:39:47,439 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26395145\n",
      "2025-08-02 16:39:48,150 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26395145 failed from error list index out of range\n",
      "2025-08-02 16:39:48,150 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18174356\n",
      "2025-08-02 16:39:48,812 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18174356 failed from error list index out of range\n",
      "2025-08-02 16:39:48,812 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24269084\n",
      "2025-08-02 16:39:49,983 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24269084 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:49,983 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28229988\n",
      "2025-08-02 16:39:52,261 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28229988 failed from error Invalid URL '': No scheme supplied. Perhaps you meant https://?\n",
      "2025-08-02 16:39:52,261 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23994285\n",
      "2025-08-02 16:39:54,664 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23994285 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:54,664 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34117589\n",
      "2025-08-02 16:39:56,922 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33880420\n",
      "2025-08-02 16:39:58,022 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19890323\n",
      "2025-08-02 16:40:00,274 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33313697\n",
      "2025-08-02 16:40:00,995 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33313697 failed from error list index out of range\n",
      "2025-08-02 16:40:00,995 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33880419\n",
      "2025-08-02 16:40:02,410 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29315421\n",
      "2025-08-02 16:40:03,134 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29315421 failed from error list index out of range\n",
      "2025-08-02 16:40:03,134 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29737471\n",
      "2025-08-02 16:40:05,587 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30037059\n",
      "2025-08-02 16:40:06,291 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30037059 failed from error list index out of range\n",
      "2025-08-02 16:40:06,291 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15550397\n",
      "2025-08-02 16:40:08,462 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15550397 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:40:08,462 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24581625\n",
      "2025-08-02 16:40:09,708 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24581625 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:40:09,708 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22256780\n",
      "2025-08-02 16:40:10,432 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 22256780 failed from error list index out of range\n",
      "2025-08-02 16:40:10,432 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28370781\n",
      "2025-08-02 16:40:11,304 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28370781 failed from error list index out of range\n",
      "2025-08-02 16:40:11,304 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36472596\n",
      "2025-08-02 16:40:12,050 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36472596 failed from error list index out of range\n",
      "2025-08-02 16:40:12,050 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29402198\n",
      "2025-08-02 16:40:14,215 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-02 16:40:14,977 - DEBUG - pubmed2pdf.cli - ** fetching of reprint None failed from error list index out of range\n",
      "2025-08-02 16:40:14,977 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20729534\n",
      "2025-08-02 16:40:15,669 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 20729534 failed from error list index out of range\n",
      "2025-08-02 16:40:15,669 - INFO - pubmed2pdf.cli - Trying to fetch pmid 1424330\n",
      "2025-08-02 16:40:16,788 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 1424330 failed from error Invalid URL 'HdkXqKLxmGChjnf5zIbHBJtsxfbbs63egf3aCZnpsbYOO9ViFAMaeTPxOpsaXlIo': No scheme supplied. Perhaps you meant https://HdkXqKLxmGChjnf5zIbHBJtsxfbbs63egf3aCZnpsbYOO9ViFAMaeTPxOpsaXlIo?\n",
      "2025-08-02 16:40:16,788 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28523980\n",
      "2025-08-02 16:40:17,445 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28523980 failed from error list index out of range\n",
      "2025-08-02 16:40:17,445 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34947871\n",
      "2025-08-02 16:40:18,334 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 34947871 failed from error list index out of range\n",
      "2025-08-02 16:40:18,334 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35092277\n",
      "2025-08-02 16:40:18,984 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35092277 failed from error list index out of range\n",
      "2025-08-02 16:40:18,985 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35147192\n",
      "2025-08-02 16:40:19,646 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35147192 failed from error list index out of range\n",
      "2025-08-02 16:40:19,646 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24581986\n",
      "2025-08-02 16:40:20,809 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24581986 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:40:20,809 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33830236\n",
      "2025-08-02 16:40:21,463 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33830236 failed from error list index out of range\n",
      "2025-08-02 16:40:21,463 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15878921\n",
      "2025-08-02 16:40:22,177 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15878921 failed from error list index out of range\n",
      "2025-08-02 16:40:22,177 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23933037\n",
      "2025-08-02 16:40:23,424 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23933037 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:40:23,424 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30081718\n",
      "2025-08-02 16:40:26,680 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29663566\n",
      "2025-08-02 16:40:27,547 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29663566 failed from error list index out of range\n",
      "2025-08-02 16:40:27,547 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10592173\n",
      "2025-08-02 16:40:28,194 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10592173 failed from error list index out of range\n",
      "2025-08-02 16:40:28,194 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25605792\n",
      "2025-08-02 16:40:28,837 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 25605792 failed from error list index out of range\n",
      "2025-08-02 16:40:28,837 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22911744\n",
      "2025-08-02 16:40:28,837 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19933690\n",
      "2025-08-02 16:40:30,320 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19933690 failed from error list index out of range\n",
      "2025-08-02 16:40:30,320 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32974805\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in Data2/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['python3', '-m', 'pubmed2pdf', 'pdf', '--out', 'Data2/', '--pmidsfile', 'Data2/pmids.txt', '--verbose', '--errors', 'Data2/pubmed2pdf_log.txt'], returncode=0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import subprocess\n",
    "command = [\n",
    "    \"python3\", \"-m\", \"pubmed2pdf\", \"pdf\",\n",
    "    \"--out\", \"Data2/\",\n",
    "    \"--pmidsfile\", \"Data2/pmids.txt\",\n",
    "    \"--verbose\",\n",
    "    \"--errors\", \"Data2/pubmed2pdf_log.txt\"\n",
    "]\n",
    "subprocess.run(command)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ed1f13d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      1155504\n",
       "1      1424330\n",
       "2     10362823\n",
       "3     16306079\n",
       "4     10221616\n",
       "5     28863933\n",
       "6     28923940\n",
       "7     33313697\n",
       "8     36822566\n",
       "9     24269084\n",
       "10    24581986\n",
       "11    30624659\n",
       "12    33077239\n",
       "13    33576824\n",
       "14    34875061\n",
       "15    29452422\n",
       "16    20619403\n",
       "17    23102856\n",
       "18    34199109\n",
       "19    12421895\n",
       "20    22256780\n",
       "21    24581625\n",
       "22    36070983\n",
       "23    32723696\n",
       "24    33067123\n",
       "25    36472596\n",
       "26    11752295\n",
       "27    25605792\n",
       "28         NaN\n",
       "29         NaN\n",
       "30         NaN\n",
       "31    10592173\n",
       "32    33036008\n",
       "33    28370781\n",
       "34    34947871\n",
       "35    35929523\n",
       "36    27441287\n",
       "37    31540845\n",
       "38    33830236\n",
       "39    29315421\n",
       "40    35092277\n",
       "41    32482256\n",
       "42    12490681\n",
       "Name: PMID, dtype: object"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "\n",
    "to_retrieve = pd.read_csv(\"Data2/pubmed2pdf_log.txt\", header=None,dtype='str')\n",
    "to_retrieve.columns = [\"PMID\"]\n",
    "to_retrieve['PMID']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9473e5a2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'12282335'"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "a = get_pmcid_from_pmid(to_retrieve['PMID'][4])\n",
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "074b68a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "url = f\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC{a}/\"\n",
    "response = requests.get(url, headers=HEADERS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55ced425",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Am J Obstet Gynecol\n",
      ". Author manuscript; available in PMC: 2025 Jul 22.\n",
      "Published in final edited form as:\n",
      "Am J Obstet Gynecol. 2025 Apr;232(4 Suppl):S105–S123. doi:\n",
      "10.1016/j.ajog.2024.08.043\n",
      "Search in PMC\n",
      "Search in PubMed\n",
      "View in NLM Catalog\n",
      "Add to search\n",
      "Endometriosis and Adenomyosis Unveiled Through Single-cell Glasses\n",
      "Linda C Giudice\n",
      "Linda C Giudice\n",
      "1\n",
      "Center for Reproductive Sciences, Department of Obstetrics, Gynecology & Reproductive Sciences, University of California, San Francisco, San Francisco, CA 94143\n",
      "Find articles by\n",
      "Linda C Giudice\n",
      "1\n",
      ",\n",
      "Binya Liu\n",
      "Binya Liu\n",
      "1\n",
      "Center for Reproductive Sciences, Department of Obstetrics, Gynecology & Reproductive Sciences, University of California, San Francisco, San Francisco, CA 94143\n",
      "Find articles by\n",
      "Binya Liu\n",
      "1\n",
      ",\n",
      "Juan C Irwin\n",
      "Juan C Irwin\n",
      "1\n",
      "Center for Reproductive Sciences, Department of Obstetrics, Gynecology & Reproductive Sciences, University of California, San Francisco, San Francisco, CA 94143\n",
      "Find articles by\n",
      "Juan C Irwin\n",
      "1\n",
      "Author information\n",
      "Copyright and License information\n",
      "1\n",
      "Center for Reproductive Sciences, Department of Obstetrics, Gynecology & Reproductive Sciences, University of California, San Francisco, San Francisco, CA 94143\n",
      "✉\n",
      "Correspondence:\n",
      "Linda.Giudice@ucsf.edu\n",
      "PMC Copyright notice\n",
      "PMCID: PMC12282335  NIHMSID: NIHMS2086294  PMID:\n",
      "40253075\n",
      "The publisher's version of this article is available at\n",
      "Am J Obstet Gynecol\n",
      "Abstract\n",
      "Single cell technologies are expanding our understanding of endometriosis and adenomyosis, sister disorders of the uterine endometrium that contain similar complements of lesion cell types but located in different niches – outside and inside the endometrium, respectively. Both diseases cause significant morbidity and impaired quality of life among those affected, and current therapies mitigate most symptoms although with highly variable efficacy, duration of effect, and frequent intolerable side effects. Thus, there is a pressing need for transformative approaches to develop individualized therapies for the variety of presentations of endometriosis and adenomyosis symptoms and heterogeneity of lesion types histologically and architecturally. Single cell technologies are transforming understanding human physiology and pathophysiology in the reproductive system and beyond. This manuscript reviews endometriosis and adenomyosis clinical characteristics and the recent studies focused on eutopic endometrium and ectopic lesions at single cell resolution, the myriad of cell types and subtypes, cell-cell communications, signaling pathways, applications for novel drug discovery and therapeutic approaches, and challenges and opportunities that accompany this type of research.\n",
      "Keywords:\n",
      "endometriosis, adenomyosis, single cell, sequencing, inflammation, cell types, biomarkers, drug discovery, signaling pathways, precision medicine\n",
      "Key take-home messages from the studies reviewed herein include:\n",
      "Evidence strongly supports that superficial and deep endometriosis are related and are distinct entities from ovarian endometriomas, challenging a “one size fits all” approach to these lesions.\n",
      "In the eutopic endometrium from endometriosis patients, there are epithelial abnormalities during the window of implantation, which can compromise fertility, including undetectable progesterone-regulated genes (PAEP and CXCL14), and endometrial subcellular alterations across the cycle that can affect tissue homeostasis, as observed in patients with disease.\n",
      "Menstrual endometrium (as a “liquid biopsy” of the tissue) mimics many of the abnormalities in the late secretory phase of patients with endometriosis, including aberrant progesterone signaling in the stromal fibroblasts, which are key to pregnancy establishment and endometrial shedding and regeneration.\n",
      "Superficial peritoneal endometriosis lesions display basalis epithelial stem cell markers, supporting the role of this tissue compartment in pelvic disease origins.\n",
      "Fibroblasts and immune cells are key contributors to the pro-inflammatory, angiogenic environment in ovarian endometriomas, opening new opportunities for drug discovery.\n",
      "Data from superficial lesions from patients on progestins reveal a unique perivascular cell type with features promoting neoangiogenesis and immune cell trafficking, and an immunotolerant microenvironment consistent with escape from immune clearance and conducive to lesion establishment. These results are remarkable, given progestin exposure of the lesions analyzed - thus raising the question of how efficacious such treatments are in curbing disease progression.\n",
      "The eutopic endometrium from adenomyosis uteri reveal that pericyte progenitors differentiate to steroid hormone receptor positive fibroblasts and to ciliated epithelial cells. The latter strongly supports mesenchymal-epithelial transition in this tissue, opening possible new therapeutic targets.\n",
      "In adenomyosis lesions, fibroblasts do not originate from pericyte progenitors and differentiate to fibroblasts expressing extracellular matrix components and to smooth muscle cells, suggesting these cells lead to the fibrosis and smooth muscle hypertrophy observed clinically.\n",
      "Adenomyosis ciliated and unciliated epithelial cells and fibroblast-like cells displayed abnormal progesterone signaling. In addition, involvement of the WNT signaling pathway in epithelial populations opens new avenues for drug discovery for this disorder.\n",
      "Direct comparison of cell types in endometriosis and adenomyosis lesions is the next step of understanding these sister disorders.\n",
      "We conclude the review with an eye to the future – what Alice might see beyond the single-cell looking glass that connects endometrium and endometrial disorders with the trillions of cells of other tissues and organs in health and disease throughout the human body and the opportunities for novel diagnostic modalities and drug discovery for endometriosis, adenomyosis and related uterine and inflammatory conditions.\n",
      "Introduction\n",
      "Endometriosis and adenomyosis are common estrogen-driven uterine disorders derived mainly from the basalis endometrium, characterized by “ectopic” endometrial-like cells outside their normal location. A key feature distinguishing them is the location of the ectopic lesions. For endometriosis they are outside the uterus - mostly in the pelvis and invading pelvic organs\n",
      "1\n",
      "–\n",
      "4\n",
      ", and for adenomyosis they are within the uterine myometrium\n",
      "5\n",
      "–\n",
      "8\n",
      ". Endometriosis and adenomyosis disease lesions comprise similar phenotypic endometrial epithelium and mesenchymal cell types and display characteristics of inflammation, neoneuroangiogenesis, and fibrosis\n",
      "1\n",
      "–\n",
      "9\n",
      ". However, whether their cellular features are equivalent in different anatomic niches is unknown and is of importance for disease stratification in view of the vastly heterogeneous lesion architectural structures and for developing tailored, personalized, patient-specific treatments and prognostic disease outcome indicators.\n",
      "These complex “sister” conditions often occur concomitantly\n",
      "10\n",
      "and result in similar symptoms of pain and uterine bleeding treated medically (mostly hormonal) with variable efficacies and are associated with infertility and similar poor reproductive outcomes\n",
      "1\n",
      ",\n",
      "4\n",
      "–\n",
      "13\n",
      ". They also greatly impact quality of life of those affected\n",
      "14\n",
      "–\n",
      "18\n",
      ". Given the devastating impact of these disorders and limited success of effective therapies and disease prevention and cure, a transformative approach to disease phenotyping and novel, personalized therapeutics is essential to improve the lives of millions of patients with endometriosis and adenomyosis across the globe.\n",
      "Single cell technologies and precision medicine approaches are early in their application to these disorders and are beginning to provide insights into commonalities/differences of cell types in endometriosis, adenomyosis, and eutopic endometrium, and the role of their unique anatomical niches influencing cell behavior, disease establishment and pathophysiology. Moreover, single cell data and associated signaling pathways are anticipated to lead to novel drug candidate discovery to control symptoms, minimize disease burden resulting ultimately in cure, identifying diagnostic biomarkers, and informing disease classification and prognostics for disease courses. The time is now to move the needle on both disorders through single cell “glasses”!\n",
      "Regarding endometrium, an atlas of normal, cycling endometrium has been derived at single cell resolution by several groups\n",
      "19\n",
      "–\n",
      "22\n",
      "(also see Chapter 3). These studies underscore heterogeneity of specific cell types and altered features in different hormonal states across the menstrual cycle and provide a key backdrop to studies at the single cell level of endometrium of patients with endometriosis and adenomyosis, endometriosis lesions, and adenomyosis lesions. Herein, we begin with an overview of clinical characteristics of endometriosis and adenomyosis (sister disorders but not identical twins), followed by a summary of current advances in scRNA sequencing data of endometriosis and adenomyosis disease lesions, corresponding eutopic endometrium, control endometrium, and disease-specific and common cell features across tissues. We conclude with a review of challenges of conducting research on human tissues, especially those that are steroid hormone responsive, and how these data inform looking to the future through single cell glasses.\n",
      "Endometriosis and adenomyosis – sister disorders, not identical twins\n",
      "Endometriosis and adenomyosis lesions both comprise uterine endometrial cells found outside their normal location (\n",
      "Figure 1\n",
      "). For endometriosis, these cells form lesions - mostly in the pelvis - arriving mainly by retrograde menstruation and causing a profound inflammatory response\n",
      "1\n",
      "–\n",
      "4\n",
      ". Risks of developing endometriosis are ~ 50% genetic and ~50% environmental\n",
      "2\n",
      ",\n",
      "4\n",
      ". In contrast, adenomyosis-associated endometrial cells (epithelium and stroma) are found within the uterus – in the uterine myometrium – where they induce a local inflammatory response, smooth muscle hypertrophy and often an enlarged, globular uterus\n",
      "5\n",
      "–\n",
      "8\n",
      ". Risks of developing adenomyosis are multi-parity, prior uterine surgery, adult exposure to tamoxifen, and in utero exposures to estrogens\n",
      "5\n",
      ",\n",
      "7\n",
      ",\n",
      "23\n",
      ". Endometriosis has been more extensively studied due to ease of obtaining endometrial tissue and sampling disease lesions at surgery. Adenomyosis has had far less molecular, genetic, and genomic analyses, as uterine myometrial lesions and the endometrial-myometrial junction are accessible mainly from hysterectomy specimens, although the eutopic endometrium proper has been studied in adenomyosis patients due to ease of access by office biopsy.\n",
      "Figure 1.\n",
      "Open in a new tab\n",
      "Comparison of endometriosis and adenomyosis pathogenesis, risk factors, diagnostic approaches, pathophysiology, symptoms, and therapies. Adapted from Bulun et al, 2023\n",
      "9\n",
      ", with permission.\n",
      "Despite their differences in anatomic location, the two disorders share common origins, some molecular characteristics, symptoms, and treatment strategies\n",
      "5\n",
      "–\n",
      "9\n",
      ",\n",
      "12\n",
      "(\n",
      "Figure 1\n",
      "). Both are diseases of the basalis endometrium with collective cell migration out of the uterus into the pelvis or into the muscle that may also involve epithelial to mesenchymal cell transition\n",
      "5\n",
      ",\n",
      "6\n",
      ",\n",
      "9\n",
      ". They share postulated origins of Mullerian remnant transformation and differentiation of endometrial stem/progenitor cells, and some common epithelial cancer driver mutations although they rarely transform into cancers\n",
      "6\n",
      ",\n",
      "9\n",
      ". They share some histologic features (e.g., deep infiltrating endometriosis and adenomyosis\n",
      "24\n",
      ", both comprise endometrial epithelial cells, stromal cells, vascular, smooth muscle, stem/progenitor, and immune cells, and have heterogeneous lesion phenotypes\n",
      "5\n",
      ",\n",
      "7\n",
      ". They are estrogen-dependent, progesterone (P\n",
      "4\n",
      ")-resistant disorders, display increased estradiol (E\n",
      "2\n",
      ") production in disease (due to aberrant aromatase expression in the epithelium and some stromal cells) and decreased E\n",
      "2\n",
      "metabolism and conversion to testosterone\n",
      "2\n",
      ",\n",
      "4\n",
      ",\n",
      "6\n",
      ",\n",
      "9\n",
      ",\n",
      "23\n",
      ",\n",
      "25\n",
      ",\n",
      "26\n",
      ". Both are also inflammatory disorders and have the activated inflammasome as the final common pathway for local and systemic inflammation\n",
      "2\n",
      ",\n",
      "3\n",
      ",\n",
      "27\n",
      ". And both can be induced in the developing fetus and/or adult (in animal models) and are associated in humans with endocrine disrupting chemicals found in the environment and personal care products\n",
      "23\n",
      ".\n",
      "Endometriosis and adenomyosis share similar clinical features (\n",
      "Figure 1\n",
      ") including dysmenorrhea, chronic pelvic pain, infertility, and poor pregnancy outcomes, and they often co-exist in the same patient and relapse post fertility-preserving surgery\n",
      "5\n",
      "–\n",
      "7\n",
      ",\n",
      "9\n",
      ",\n",
      "12\n",
      ",\n",
      "28\n",
      ". Symptom histories are key in diagnosing both disorders, but many symptoms overlap between them and with other disorders\n",
      "2\n",
      ",\n",
      "4\n",
      ". Standards for endometriosis and adenomyosis diagnosis are histologic confirmation of endometrial-like cells in lesions, and pelvic MRI and transvaginal ultrasound are useful for detecting endometriomas and deep disease as well as adenomyosis-associated widened endometrial-myometrial junction and adenomyomas\n",
      "4\n",
      ",\n",
      "29\n",
      ".\n",
      "Treatment paradigms are similar for both disorders and include combined estrogen/ progestin contraceptive steroids, progestins alone, progestin-containing intrauterine systems, selective progesterone receptor modulators (SPRMs), selective estrogen receptor modulators (SERMs), GnRH analogs, aromatase inhibitors accompanied by ovulation suppression, and dopamine agonists, and medically assisted and advanced reproductive technologies for infertility\n",
      "1\n",
      ",\n",
      "4\n",
      ",\n",
      "8\n",
      ",\n",
      "12\n",
      ". Overall, medical therapies for pain have limited efficacy and/or intolerability for many patients. Endometriosis lesion removal or hysterectomy is largely effective for symptom management, although ~ 50% of patients recur within 5 years after conservative surgery and ~15% after hysterectomy\n",
      "1\n",
      ",\n",
      "4\n",
      ",\n",
      "12\n",
      ". Adenomyosis, on the other hand, can be excised but this is rarely performed due to technical challenges, and while hysterectomy is curative, it compromises fertility pursuits\n",
      "30\n",
      ".\n",
      "Thus, developing non-invasive diagnostics as stand-alone assessments or as adjuncts to imaging technologies, and innovative and personalized therapies are major goals for both disorders.\n",
      "Endometriosis through single cell glasses\n",
      "Subsequent to the single cell cartography of human endometrium published by Wang et al\n",
      "19\n",
      "(see Chapter 3), several groups have recently reported eutopic endometrial scRNAseq data of patients with endometriosis, endometriosis lesions and control eutopic endometrium\n",
      "20\n",
      ",\n",
      "31\n",
      "–\n",
      "36\n",
      "(\n",
      "Table 1\n",
      "). Types of samples, inclusion/exclusion criteria, choice of controls, study designs, numbers of cells sequenced, and number of reads (i.e., base pairs sequenced) are key to data integrity and interpretation and to comparative analyses across tissues and studies. They have provided insights into origins of ectopic disease, the heterogeneity of cell types/subtypes, unique clusters and signatures of endometrium and ectopic lesions informing mechanisms and pathways involved in cellular dysfunctions relevant to pain and fertility compromise, novel cell-cell communications, relationships between lesions and eutopic tissue/cells, and biomarker discovery. Across these studies, samples were obtained in different hormonal milieu (menstrual cycle phase, exogenous hormones), and between 55,000 to 378,000 cells were sequenced (\n",
      "Table 1\n",
      "). As cycle phase and exogenous hormone use are key drivers of endometrial gene expression\n",
      "37\n",
      "–\n",
      "44\n",
      ", comparison of data across studies can be challenging and mixed hormonal milieu are sometimes experienced within the same study in both cases and/or controls. Below we summarize results of studies focused exclusively on eutopic endometrium of patients with endometriosis relevant to endometrial function and dysfunction and biomarker discovery, followed by studies focused on ovarian endometriomas, superficial, and/or deep infiltrating disease, and eutopic endometrium, relevant to disease heterogeneity, pathophysiology, pathogenesis, and in the setting of progestins – commonly prescribed for symptom management of patients with endometriosis and dysmenorrhea and chronic pelvic pain.\n",
      "Table 1.\n",
      "Highlights of recent endometriosis-related scRNAseq studies.\n",
      "Reference\n",
      "Tissue Analyzed\n",
      "Total cells sequenced\n",
      "Technology/Platform\n",
      "Total subjects\n",
      "Endometriosis Type and ASRM Stage\n",
      "Controls\n",
      "Race/ Ethnicity\n",
      "Hormones/IUD\n",
      "Cycle Phase Endometrial histology\n",
      "Ma 2021\n",
      "endometrium, ovarian endometrioma\n",
      "55,000\n",
      "scRNAseq (10X)\n",
      "n=6 subjects (n=3 cases, n=3 controls)\n",
      "ASRM Stage III, IV ovarian endometriomas\n",
      "Healthy controls without endometriosis\n",
      "N/A\n",
      "None\n",
      "All in prolifeative phase\n",
      "Garcia-Alonso 2022\n",
      "endometrium (functionalis, full thickness), endometriosis peritoneal lesions (red, white, black)\n",
      "98,569 sequenced\n",
      "scRNAseq, snRNA seq, spatial profilng. Lesion microarray data (\n",
      "GSE141549\n",
      ")\n",
      "n=3 functionalis n=6 full thickness; Microarray data controls: endom n=42, peritoneum n=12; n=9 red, 9 white, 11 black\n",
      "Peritoneal disease: red, white, black\n",
      "Healthy controls (functionalis layer) n=6 full thickness without reproductive disorders; normal peritoneum\n",
      "N/A\n",
      "None\n",
      "Proliferative, secretory; Microarray samples: Control Endo 17PE, 25SE; Perit 4PE, 8SE; Lesions: Red 2PE, 7 SE; White 5PE, 4SE; Black 6PE, 5 SE\n",
      "Shih 2022\n",
      "menstrual endometrium\n",
      "43,054\n",
      "scRNAseq (10X)\n",
      "n=33 subjects (n=11 Dx, n=13 sx, n=9 controls)\n",
      "N/A; another group with Sx but no Dx.\n",
      "No endometriosis diagnosis. ? Other GYN disorders\n",
      "White: 10 Dx, 13 sx, 7 controls Black: 0 Dx, 0 sx, 1 control\n",
      "Hispanic: 0 Dx, 0 sx, 0 cont\n",
      "Mixed: 0 Dx,0 sx, 1 control\n",
      "Other: 1 Dx,0 sx, 0 control\n",
      "None on hormones, except 1 case used vaginal P4. Re-analysis showed no impact on results\n",
      "Menstrual (heaviest flow, mostly CD 1 or 2)\n",
      "Tan 2022\n",
      "endometrium, enodmetriosis lesions\n",
      "122,000\n",
      "scRNAseq (10X) IMC organoids\n",
      "n=27 subjects (n=19 cases, n=8 controls; n=14 sequenced\n",
      "ASRM Stage II-IV, peritoneal lesions, ovarian “lesions”, organoids\n",
      "No endometriosis or infammatory conditions\n",
      "White: 9 cases, 3 controls\n",
      "Asian: 4 cases, 2 controls\n",
      "Hisp: 5 cases, 3 controls\n",
      "Black: 1 case, 0 controls\n",
      "Progestin [NETA, LVN oral, IUD, drosperinone, norelgestromin] +/− ethinyl E2, medroxy-progesterone acetate, copperIUD\n",
      "wkly PE: 3 cases, 2 control; inactive 3 cases, 0 control; mens: 1 case, 0 control, exog hormone effect: 7 cases, 7 control\n",
      "PE: 0 cases, 2 controls\n",
      "IE: 0 cases, 1 control\n",
      "ESE: 4 cases, 0 control\n",
      "N/A: 1 case, 1 control\n",
      "Fonseca 2023\n",
      "endometrium, endometriosis lesions, unaffected ovary and peritoneum\n",
      "373,851\n",
      "digital scRNAseq\n",
      "n=21 subjects n=17 cases, n=4 controls n-54 specimens collected\n",
      "Cases: n=17 Endometrium Endometrioma Superficial and deep disesase n=9 w/o GYN disorders; n=8 w/ adenomyosis, uterine fibroids +/− polyp\n",
      "Controls n=4 n=3 PMP, 1 peri-menopause All no evidence of endometriosis, all w/ leiomyoma +/− adenomyosis +/− uterine polyp\n",
      "White: 13 cases,2 controls\n",
      "Black: 2 cases, 0 controls\n",
      "Asian: 1 case, 0 control\n",
      "Other: 1 case, 0 control\n",
      "Cases\n",
      ": n=14 on no hormones. n=1 w/ vaginal ring E+P; n=1 on E/T, P4; n=1 on E2+P4.\n",
      "Controls\n",
      ": 3 of 4 on E +/− P4; NETA; n=1 peri-menopause in luteal phase\n",
      "Cases\n",
      ": Proliferative phase: n=7; secretory phase: n=9; N/A (on hormones) n=3.\n",
      "Controls\n",
      ": Proliferative n=0, Secretory n=1, N/A (on hormones) n=3\n",
      "Huang 2023\n",
      "endometrium\n",
      "128,243\n",
      "scRNAseq (10X)\n",
      "n=10 sequenced n=6 cases, n=4 controls\n",
      "ASRM Stage I,II\n",
      "No endometriosis, benign ovarian cysts\n",
      "N/A\n",
      "None\n",
      "EPE: 3 cases, 3 controls\n",
      "MSE: 3 cases, 3 controls\n",
      "LSE: 1 control\n",
      "Almonte 2023\n",
      "endometrium\n",
      ">250,000\n",
      "scRNseq (10X)\n",
      "n=29 cases, 7 controls w/o disease and 8 controls with fibroids\n",
      "ASRM stage I-IV\n",
      "No endometriosis or other GYN d/o; No endo but + uterine fibroids\n",
      "N/A\n",
      "None\n",
      "Proliferative:12 cases, 5 controls. Secretory: 14 cases, 3 controls.\n",
      "Open in a new tab\n",
      "Abbreviations:\n",
      "CD=cycle day; dx=diagnosis; E\n",
      "2\n",
      "=estradiol; ESE=early secretory phase; exog=exogenous; IE=interval endometrium; IMC=imaging mass cytometry; IUD=intrauterine device; LSE=late secretory phase; LVN=levonorgestrel; MSE=mid-secretory phase; N/A=not available; NETA=norethindrone acetate; P=progestin; P\n",
      "4\n",
      "=progesterone; PE=proliferative phase; SE=secretory phase; sx=symptoms, w/= with; +/− = with and/or without. Adapted from Giudice et al\n",
      "4\n",
      ", with permission.\n",
      "Eutopic endometrium\n",
      "Window of implantation epithelial abnormalities in infertile endometriosis patients\n",
      "Endometrial biopsies of cases with stage I/II disease (revised American Society for Reproductive Medicine (rASRM) scoring system)\n",
      "45\n",
      "and infertility (n=3 WOI, n=3 early proliferative phase) versus n=7 fertile controls without disease (3 WOI, 3 proliferative, 1 late secretory phase) were studied by Huang et al\n",
      "34\n",
      ". A total of 128,243 cells were sequenced, and 30 type/cell type clusters were identified. Notably, a unique epithelial cell cluster in WOI endometrium of endometriosis patients was identified that lacked expression of two key genes associated with implantation:\n",
      "PAEP\n",
      "(progesterone-associated endometrial protein) and\n",
      "CXCL14\n",
      "(chemokine (CXC motif) ligand 14) (\n",
      "Figure 2\n",
      "). PAEP, also known as glycodelin A and placental protein 14, is involved in immune response regulation, promoting opening of the WOI by lowering maternal immune responses to an implanting embryo\n",
      "46\n",
      ". CXCL14 is involved in activating the innate immune response stimulating chemotaxis of uterine natural killer cells (uNK) and other monocytes to epithelial glands for implantation\n",
      "46\n",
      ". Other findings included absence of normal variation of endometrial uNK and T cell numbers across the cycle, and lower anti-inflammatory IL-10 and more pro-inflammatory cytokines expressed in endometrial immune cells in the WOI versus in the proliferative phase in cases and the opposite in controls. Notably, 11 ligand–receptor pairs were upregulated between endometrial immune and epithelial cells during the WOI of cases.\n",
      "Figure 2. Absence of key genes that open the window of implantation in eutopic endometrial epithelial cluster in infertile patients with Stage I/II endometriosis.\n",
      "Open in a new tab\n",
      "Panel A\n",
      ". t-SNE projection of endometrial cells colored based on their sample origin and cycle phase.\n",
      "Panel B\n",
      ". Expression patterns of PAEP and CXCL14 projected on t-distributed stochastic neighbour embedding (t-SNE) plot. From Huang et al, 2023\n",
      "34\n",
      ", with permission.\n",
      "Thus, lower epithelial receptivity markers, abnormal immune cell frequencies, a pro- inflammatory WOI, and new insights into cell-cell communications between epithelium and immune cells support an important role of the immune system and an adverse environment for embryo implantation for fertility and potentially for poor pregnancy outcomes in patients with stage I/II endometriosis who do achieve pregnancy versus fertile controls.\n",
      "Endometrial subcellular alterations across the cycle in endometriosis patients\n",
      "In a recent study, our group sequenced eutopic endometrium (237,620 cells) from 29 subjects with endometriosis (stages I/II and III/IV) and 15 controls without disease (7 healthy controls, 8 with symptomatic uterine fibroids), across the menstrual cycle\n",
      "36\n",
      ". We identified 5 main cell groups: stromal fibroblasts, smooth muscle, endothelial, epithelial, and immune. In the latter, further analysis revealed 18 sub-populations including uterine NK cells, T cell subsets, monocytes, macrophages, and dendritic cells. Epithelial subpopulations comprised glandular, luminal and ciliated cells, with abnormal expression of\n",
      "CXCL14, PAEP, CCL20\n",
      ",\n",
      "MMP7,\n",
      "and\n",
      "SFRP4\n",
      "genes in cases, regardless of phase. Endometrium in controls with FIGO Type 2 uterine fibroids revealed overexpression of\n",
      "WFDC2\n",
      "and\n",
      "DAXX\n",
      "genes in cases.\n",
      "In immune cells, fibroblasts, and epithelial cells, cytokine production, stress signal and pro-inflammatory pathways in cases were noted, similar to Huang et al\n",
      "34\n",
      "findings, and enriched pathways in epithelial and ciliated cells related to migration and cell motility were higher in cases. Cell communication analysis revealed a complex network of inflammatory pathways, with significant involvement of classical monocytes and NK cells. Finally, genes implicated in ovarian cancer were upregulated in endometrium of cases compared to controls, confirming known association of endometriosis with specific subtypes of ovarian cancer\n",
      "12\n",
      ". Thus, in the eutopic endometrium in patients with endometriosis, there are numerous alterations in specific cell types across the cycle versus controls without disease and disease-free controls with uterine fibroids\n",
      "36\n",
      ". These can predispose to impaired fertility and tissue dyshomeostasis.\n",
      "Menstrual endometrium, disease pathophysiology, candidate biomarker discovery\n",
      "A recent scRNAseq study\n",
      "32\n",
      "of endometrial tissue in menstrual efflux (not biopsies) of n=11 cases with laparoscopic evidence of endometriosis (undefined stage) (11,924 cells) and n=9 healthy controls with no disease or symptoms (14,327 cells) revealed 18 cell clusters. These included abundant uNKs, stromal fibroblast, epithelial, B, T, and myeloid cell clusters, and a small plasmacytoid dendritic cell (pDC) cluster. Further analysis separated data by cases vs controls and revealed a marked decrease in uNK1 and uNK2 cells and significantly enriched B cells in cases (\n",
      "Figure 3A\n",
      "). Log2 odds ratio (OR) analysis confirmed uNK1/2 significantly enriched in controls and B cells greatly enriched in cases (\n",
      "Figure 3B\n",
      "). Moreover, menstrual endometrium scRNA seq revealed decreased decidualization markers in stromal fibroblasts (reflecting P\n",
      "4\n",
      "-resistance) and pro-inflammatory stromal cells, as observed by others in FACS-sorted cell analyses and\n",
      "in vitro\n",
      "cell culture studies\n",
      "6\n",
      ",\n",
      "47\n",
      ".\n",
      "Figure 3. Distinct Cell Compositions and Cell Enrichment in Menstrual Endometrium from patients with Endometriosis versus Controls and Model of Disease.\n",
      "Open in a new tab\n",
      "Panels A\n",
      "and\n",
      "B\n",
      "show low abundance of uNK1 and uNK2 cells and high abundance of B cells in patients with endometriosis.\n",
      "Panel C\n",
      "shows proposed pathogenesis and pathophysiology implied by the data. From Shih et al, 2022\n",
      "32\n",
      ", with permission.\n",
      "Overall, this study demonstrates that menstrual endometrium (commonly referred to as a “liquid biopsy of the endometrium”) reflects secretory endometrium abnormalities as possible biomarkers of disease. The proposed disease model is shown in\n",
      "Figure 3C\n",
      "wherein defective endometrial stromal fibroblast decidualization is driven by multiple factors including inflammation, chronic endometritis, stress, and/or P\n",
      "4\n",
      "-resistance – which all have clinical relevance for pregnancy establishment and success, as well as tissue homeostasis. This, in turn, may direct fibroblast differentiation to a chronic inflammatory phenotype and senescence, which may also impair decidualization. Reduced decidualization may also compromise the infiltration and proliferation of uNK cells, likely to be important for senescent cell removal and important for an optimal setting for embryo nidation\n",
      "32\n",
      ".\n",
      "Ectopic endometrium\n",
      "Superficial peritoneal lesions display basalis epithelial stem cell markers\n",
      "In their landmark spatial-temporal single cell transcriptomic study\n",
      "48\n",
      ", García-Alonso et al\n",
      "20\n",
      "analyzed endometrium (3 functionalis and 6 full thickness (functionalis and basalis) tissues) across the cycle and identified 14 unique cell clusters that localized to corresponding endometrial compartments and niches. The proposed endometrial epithelial stem-cell niche resides within the basalis layer and contains SOX9+ cells and other “stem” markers (e.g., LGR5, SSEA1, β-catenin, N-cadherin)\n",
      "49\n",
      ",\n",
      "50\n",
      ", and Garcia-Alonso et al\n",
      "20\n",
      "found proliferative SOX9+LGR5+ cells mapped to the basalis stem cell niche. Epithelial stem cells are believed to play a role in endometriosis pathogenesis, and by leveraging bulk (microarray) transcriptomic data of superficial peritoneal lesions (9 red, 9 white, 11 black) (\n",
      "GSE141549\n",
      "), this team identified enrichment of SOX9+LGR5+ markers in peritoneal lesions in the proliferative phase, compared to normal endometrium (n=42) and peritoneum (n=12)\n",
      "20\n",
      ". A proliferative phenotype of endometriosis lesion epithelial cells is consistent with P\n",
      "4\n",
      "-resistance noted in endometriosis\n",
      "4\n",
      ",\n",
      "6\n",
      ". Moreover, dysfunctional epithelium as a major driver of endometrial diseases, including endometriosis, and discovery of SOX9+LGR5+ stem-like populations dominant in endometriosis peritoneal lesions are foundational observations about pathogenesis and pathophysiology of superficial endometriosis disease lesions deriving from the basalis layer epithelium.\n",
      "Endometrioma, superficial and deep infiltrating endometriosis cell atlas\n",
      "A comprehensive scRNAseq analysis of >370,000 individual cells from 8 ovarian endometriomas, 28 superficial and deep disease lesions, 10 eutopic endometrium, 4 unaffected ovary, and 4 endometriosis-free peritoneum samples, has resulted in a cell atlas of endometrial-type epithelial cells, stromal cells, and microenvironment cell populations across tissue types\n",
      "35\n",
      ". Samples were derived from 17 cases (9 only endometriosis; 8 endometriosis and adenomyosis, uterine fibroids, and/or uterine polyps). 14 were cycling (7 proliferative and 7 secretory phases), and 3 were on contraceptive steroids (\n",
      "Table 1\n",
      "). Controls (n=4) were postmenopausal (3) or perimenopausal (1), without endometriosis but with uterine fibroids +/− adenomyosis +/− uterine polyps, and all were on hormonal replacement regimens. Histologic and macroscopic features of specimens from one patient, shown in\n",
      "Figure 4A\n",
      ", underscore the heterogeneity of the lesions by cell abundance, cell types, and tissue architecture.\n",
      "Figure 4. Endometriosis lesion scRNAseq analysis.\n",
      "Open in a new tab\n",
      "Panel A.\n",
      "Histologic (H&E staining) and architectural heterogeneity of endometriosis lesions.\n",
      "Panel B.\n",
      "Correlations Based on Cluster Frequency Across All Specimens Profiled by scRNAseq.\n",
      "Panel C.\n",
      "Epithelial differential gene expression and signaling pathways in endometriomas and superficial and deep disease (P < 0.05 and log2 FC > 1). From Fonseca et al, 2023\n",
      "35\n",
      ", with permission.\n",
      "The major cell types identified from scRNAseq of all cells from n=49 samples comprising the 5 major tissue classes (endometrioma, peritoneal endometriosis (deep and superficial), unaffected peritoneum, eutopic endometrium, unaffected ovary) included epithelial, smooth muscle, mesenchymal, endothelial, mast, myeloid, B, and T/NKT cells, and erythrocytes. Correlations based on cluster frequency across all specimens profiled by scRNAseq are shown in\n",
      "Figure 4B\n",
      "and reveal 3 main clusters: 1) mainly endometriosis specimens; 2) mix of endometriomas (5 of the 8 endometrioma specimens) and eutopic endometrium (5 of the 10 eutopic samples; 3) only controls: unaffected ovary and 5 of the 10 eutopic endometrium samples\n",
      "35\n",
      ".\n",
      "Endometrioma, endometrium, superficial and deep endometriosis epithelia revealed distinct cell subtypes and signaling pathways, and endometriomas displayed markedly different epithelial differential gene expression and signaling pathways compared to deep and superficial peritoneal disease and eutopic endometrium (\n",
      "Figure 4C\n",
      "). Interestingly, endometriomas displayed immune cell and complement activation and were enriched in B cells and plasma cells, suggesting possible infection in endometriomas and a unique role for B cells which have received limited attention in endometriosis pathophysiology\n",
      "51\n",
      "–\n",
      "53\n",
      ". Peritoneal disease was enriched in mast cells and T/NK-T cells engaging novel immune targets, and some histologically negative mesothelium surprisingly had disease signatures, although depth of lesion evaluation for endometriosis tissue may vary among pathologists.\n",
      "Endometrial-type epithelial cells and fibroblasts exhibited differential gene expression associated with deep or superficial disease status (\n",
      "Figure 5\n",
      "). Interestingly, deep infiltrating disease epithelium displayed upregulation of nerve growth factor, suggesting a role of the epithelium in promoting innervation of this disease type which is particularly painful\n",
      "12\n",
      ".\n",
      "Figure 5. Molecular signatures of epithelial and stromal cells in superficial peritoneal and deep endometriosis differ.\n",
      "Open in a new tab\n",
      "Endometrial-type epithelial cells (\n",
      "Panel A\n",
      ") and fibroblasts (\n",
      "Panel B\n",
      ") exhibit DGE associated with deep or superficial status. From Fonseca et al, 2023\n",
      "35\n",
      ", with permission.\n",
      "While somatic mutations have been reported in endometrium and endometriosis lesions, mainly involving KRAS activating mutations and ARID1A loss of function mutations\n",
      "54\n",
      ",\n",
      "55\n",
      ", Fonseca et al\n",
      "35\n",
      "confirmed these somatic mutations in various lesions (\n",
      "Figure 6A\n",
      ") and demonstrated\n",
      "in vivo\n",
      "transcriptional consequences of these mutations at the single cell level (\n",
      "Figure 6B\n",
      "). ARID1A mutation was associated with pro-lymphangiogenic stroma (\n",
      "Figure 6C\n",
      ") and adjacent mesothelium pro-inflammatory features and ciliated epithelial signatures - consistent with ovarian oncogenic potential. It is well-known that endometrioid ovarian cancer (EOC) and clear cell ovarian cancer (CCOC) arise in association with endometriosis, suggesting endometrial-type epithelial cells may be precursors for these tumors. A multi-subject single cell deconvolution analysis of the single cell data (\n",
      "Figure 7\n",
      ") revealed strong enrichment of signatures for\n",
      "ciliated\n",
      "endometrial-type epithelial cells compared with the other keratin-positive clusters. In contrast, high grade serous ovarian cancer, an epithelial ovarian cancer type,\n",
      "not\n",
      "believed to originate from endometriosis epithelial cells, showed no pattern for preferential enrichment of any epithelial cluster.\n",
      "Figure 6. Endometrium, endometriosis, and endometriomas share somatic mutations with transcriptional consequences.\n",
      "Open in a new tab\n",
      "Panel A.\n",
      "Mutations detected in each lesion.\n",
      "Panel B.\n",
      "Differential gene expression by mutation state.\n",
      "Panel C.\n",
      "Epithelial expression of lymph-angiogenesis markers by ARID1A staining status. From Fonseca et al, 2023\n",
      "35\n",
      ", with permission.\n",
      "Figure 7. Deconvolution Analysis Reveals Endometrial-type Epithelium Cell-type Signatures in Endometriosis-associated Ovarian Cancers.\n",
      "Open in a new tab\n",
      "Multi-subject single-cell deconvolution revealed strong enrichment of signatures for\n",
      "ciliated endometrial-type epithelial cells\n",
      "compared with the other keratin-positive clusters. HGSOC, an epithelial ovarian tumor type thought\n",
      "not\n",
      "to originate from endometriosis epithelial cells showed no pattern for preferential enrichment of any epithelial cluster. From Fonseca et al, 2023\n",
      "35\n",
      ", with permission.\n",
      "Key takeaways:\n",
      "Endometrium and all three lesion types versus control endometrium displayed different cell/molecular signatures across tissues, consistent with restructuring/transcriptional reprogramming in lesions.\n",
      "Epithelial and stroma differed by site with striking differential gene expression in endometriomas and peritoneal lesions suggesting these are distinct disease entities.\n",
      "Superficial peritoneal and deep infiltrative endometriosis epithelium and fibroblasts had some differential gene expression and pathways, but similar structural genes to each other, suggesting they are part of the same continuum and distinct from endometrioma.\n",
      "While this study is a tour de force of endometriosis lesion and eutopic endometrium single cell transcriptomic analysis giving major insights into cell- and tissue-specific features and functions, including oncogenic potential of endometrioma epithelial cells, it underscores some of the challenges in conducting these types of studies. These include the presence of co-existing gynecologic disorders that can confound data interpretation; cells sequenced comprise a minor population in most samples (e.g., <1% of cells profiled), and controls and normal tissues are challenging to obtain e.g., controls were peri- or post-menopausal on hormone replacement therapies and had other uterine disorders. Nonetheless, these data are important and provide foundational information for comparison with other studies on eutopic and ectopic lesions in endometriosis patients and controls without disease.\n",
      "Fibroblasts and immune cells: key contributors to a pro-inflammatory, angiogenic environment in endometriomas.\n",
      "Sequencing 55,000 single cells from 3 endometriomas, 3 matched eutopic endometrium, and 3 endometrial samples from controls without disease in the proliferative phase, Ma et al\n",
      "31\n",
      "found 9 cell type clusters: fibroblasts, macrophages/monocytes, neutrophils, and T, NK, epithelial, endothelial, and mast cells, and unknown. Thirteen fibroblast subtype clusters were identified with significant differences in subtype composition among ectopic endometrium, eutopic endometrium, and normal endometrium. Fibroblast subtypes from normal proliferative phase endometrium displayed enriched functions of DNA replication, repair, growth regulation, and metabolic processes. However, endometrioma fibroblasts comprised 4 subtypes: 1) cytokine, inflammatory response; 2) FGF stimulation, immune response; 3) extracellular matrix (ECM) organization, cell adhesion; 4) angiogenesis, hypoxia response, and epithelial-mesenchymal transition (EMT), all of which were enriched in MAPK, TNF, IL-17, TGF-β signaling pathways. Fibroblasts from eutopic endometrium of cases displayed many pathways similar to normal tissue, but some features suggested a “transition state” from normal endometrium to endometrioma, involving TGF-β, MAPK, Rho-Rack, NF-kB, JAK-STAT, and EMT. A developmental trajectory was postulated based on these fibroblast features, with endometrioma fibroblasts derived from endometrial fibroblasts\n",
      "31\n",
      ", much as the epithelial cell population has been proposed, based on shared\n",
      "epithelial\n",
      "somatic mutations in these two tissues\n",
      "54\n",
      ". Thus, endometrioma and paired eutopic endometrium fibroblasts differ, and the single cell data support their key roles in disease pathogenesis.\n",
      "Analysis of immune cells revealed they differed significantly in normal endometrium, endometrioma, and paired eutopic tissue (\n",
      "Figure 8A\n",
      "). Specifically, T cell and uNK cell frequencies were lower, uNK cells were more active, and macrophages (Mφ) were enriched and had features of tissue remodeling versus eutopic endometrium\n",
      "31\n",
      ", similar to data from Fonseca et al\n",
      "35\n",
      ". Thus, in the proliferative phase, fibroblasts and immune cell subpopulations contribute to a pro-inflammatory, angiogenic environment in endometriomas. Moreover cell-to-cell communication/interactome analysis revealed that immune cells stimulate fibroblast adhesion and growth in endometriomas (\n",
      "Figure 8B\n",
      "), underscoring the power of single cell analyses to inform the pathogenesis and pathophysiology of endometriomas and cell-cell communications. Furthermore, recent observations indicated that Fusobacterium\n",
      "56\n",
      "and dysbiosis of gut\n",
      "57\n",
      ",\n",
      "58\n",
      "altered systemic and potentially local immune functions, contributing to the development of endometriosis. Notably, while transcriptomic alterations of specific populations and sub- populations of the endometrium, lesions, and immune cells (local and peripheral) under these conditions awaits further study, antibiotic and nutritional therapies for dysbiosis is emerging as a new paradigm for treating endometriosis development and symptom management.\n",
      "Figure 8. Immune cell frequencies in endometriomas and communication networks of immune cell types with endometrioma fibroblasts.\n",
      "Open in a new tab\n",
      "The network supports interactions between immune and fibroblast cells in ovarian endometrioma pathogenesis and pathophysiology. From Ma et al, 2021\n",
      "31\n",
      ", with permission.\n",
      "Endometriosis peritoneal and ovary lesions of patients on progestins\n",
      "As progestins are a mainstay of medical therapy for endometriosis-related pain\n",
      "12\n",
      ", analysis of lesions and endometrium of subjects taking progestins is highly valuable in understanding their effects on cell types and tissue features. Tan et al\n",
      "33\n",
      "performed scRNAseq on 122,000 cells from 14 individuals, selective cell localization by imaging mass cytometry, and validations in endometrial epithelial organoid cultures. Peritoneal and/or ovarian disease and endometrium from 19 cases (rASRM stage II-IV) included 14 on norethindrone (NETA) + ethinyl estradiol (EthE\n",
      "2\n",
      "), 2 on drosperinone + EthE\n",
      "2\n",
      ", 1 on levonorgestrel + EthE\n",
      "2\n",
      ", 1 on only NETA, 1 menstrual phase, and 8 controls with no endometriosis or inflammatory conditions: 4 on NETA + EthE\n",
      "2\n",
      ", 2 in proliferative phase, 1 on medroxyprogesterone acetate, 1 on norgestromin + EthE\n",
      "2\n",
      ". The peritoneal lesions had similar cell compositions as eutopic endometrium but dysregulated innate immune and vascular components - vastly different from ovarian endometriosis lesions. The authors identified a unique perivascular mural cell specific to the peritoneal lesions with features promoting neo- angiogenesis and immune cell trafficking. They also identified a progenitor-like epithelial subpopulation, different from the SOX9+LGR5+ epithelial subpopulation found by Garcia-Alfonso et al\n",
      "20\n",
      ". Analysis of peritoneal lesion myeloid populations of immunomodulatory Mφ and DCs revealed a coordinated immunotolerant phenotype in the disease microenvironment, promoting immunosurveillance escape and thus conducive to lesion establishment.\n",
      "Overall, the data demonstrate that immune and vascular components of peritoneal endometriosis favor neo-angiogenesis and an immune tolerant niche in the peritoneal cavity, beneficial for lesion establishment and growth. These are remarkable findings given that most subjects were on hormone suppressive therapies for pain [progestin (+ estrogen)] and raises the issue of how efficacious these therapies are on specific cells and the role of estrogenic components in disease management. This valuable single cell atlas of disease from patients largely on hormonal treatments offers the opportunity for comparisons of cell types and features in peritoneal lesions and endometriomas in the setting of other hormonal treatments and/or in various phases of the cycle to interpret efficacies of selective therapies on cell-specific pathways and cell-cell communications.\n",
      "Adenomyosis through single cell glasses\n",
      "Uterine compartments relevant to adenomyosis pathogenesis include the endometrial-myometrial junction (EMJ) [basalis and lower functionalis endometrium + sub-endometrial inner myometrium], all of Mullerian origin and displaying cycling estrogen receptors (ER) and progesterone receptors (PR)\n",
      "59\n",
      "–\n",
      "61\n",
      "(\n",
      "Figure 9\n",
      "). Few molecular and genomic studies of the EMJ in adenomyosis uteri have been conducted, likely as access to basalis and inner myometrium requires hysterectomy versus the functionalis, which can be readily obtained by office biopsy.\n",
      "Figure 9. The endometrial-myometrial junction (EMJ).\n",
      "Open in a new tab\n",
      "Panel A.\n",
      "Histology of human uterine endometrium and myometrium compartments (H&E staining).\n",
      "Panel B\n",
      ": Transvaginal ultrasound midsagittal image of uterine compartments.\n",
      "Panel C.\n",
      "Schematic of uterine compartments. Adapted from Naftlin and Jurkovic, 2009\n",
      "61\n",
      ", with permission.\n",
      "Bulk analyses:\n",
      "Studies to date in adenomyosis patients using immunohistochemistry and qPCR reveal abnormal\n",
      "endometrial\n",
      "functionalis epithelial aromatase expression (thus high E\n",
      "2\n",
      "milieu), elevated stromal fibroblast IL-6, IL-8, IL-17 secretion, TH17-Treg imbalance, altered uNK receptors, and aberrant P\n",
      "4\n",
      "-signaling\n",
      "5\n",
      ",\n",
      "26\n",
      ",\n",
      "62\n",
      "–\n",
      "64\n",
      "– very similar to endometrium of patients with endometriosis except for IL-17 and TH17-Tregulatory cell imbalance. We conducted bulk RNA seq of endometrial\n",
      "functionalis\n",
      "65\n",
      "and a study on separate compartments of EMJ inner myometrium and endometrium from patients with diffuse adenomyosis versus controls in the proliferative phase\n",
      "66\n",
      ". ECM remodeling, inflammation, IL-17 signaling, and mTOR signaling were enriched in endometrium, and inflammation, pain, and neuronal signaling pathways in myometrium, consistent with clinical symptoms of this disorder.\n",
      "Single cell analyses:\n",
      "Two recent publications have analyzed adenomyosis lesions\n",
      "67\n",
      ",\n",
      "68\n",
      ". They are important beginnings to understanding the pathogenesis and pathophysiology of adenomyosis, building upon bulk transcriptomic data, which could be further mined in deconvolution studies to validate new and likely forthcoming additional single cell data.\n",
      "Liu et al\n",
      "67\n",
      "sequenced 42,292 cells derived from a control endometrium of a patient with uterine fibroids and eutopic (functionalis) endometrium and matched ectopic lesion from a patient with adenomyosis (type and menstrual cycle phase undefined). The data revealed seven distinct cell types. Comparison of the data from the lesion with those from eutopic matched endometrium and control endometrium revealed processes involved in progression of normal→endometrial→adenomyosis epithelial phenotypes. These included motility, proliferation, angiogenesis, inflammation, and cancer, consistent with the theory of collective cell migration and invasion into the myometrium of endometrial-derived cells. The novel data set revealed vascular mimicry (important in tumor formation) and epithelial transformation to endothelial cells. It is anticipated that the novel epithelial→endothelial transition phenotype will likely be mined in the future and expanded studies of adenomyosis and perhaps other gynecologic disorders and may serve as a unique cell type for drug targeting.\n",
      "Bulun and colleagues\n",
      "68\n",
      "reported results of single cell analyses of adenomyosis lesions and matched eutopic endometrium and myometrium from 3 subjects, all with diffuse adenomyosis and in the proliferative phase (total of 9 samples and 66,000 cells sequenced). They identified 11 cell types in eutopic endometrium of patients with adenomyosis (\n",
      "Figure 10 A\n",
      "). Fibroblast-like cells comprised a large fraction (36%), and pseudotime analysis revealed they originate from pericyte progenitors that differentiate to ER+ and PR+ endometrial stromal fibroblasts. Notably, the pseudotime trajectory was found to end with ciliated epithelial cells, suggesting mesenchymal- epithelial transition at play in this tissue. In adenomyosis lesions, they identified 13 cell types including 2 (smooth muscle cells and neurons) not found in the endometrium of the same patients (\n",
      "Figure 10 B\n",
      "). The adenomyosis lesions contained even more fibroblast-like cells compared to endometrium (50% versus 36%, respectively). In contrast to matched endometrium, the fibroblast-like cells did not originate from pericyte progenitors and differentiated to ECM-expressing fibroblasts and smooth muscle cells (SMC). Also, myometrial SMC were found to derive from fibroblast-like clusters. Moreover, the WNT inhibitor, secreted frizzled related protein (SFRP) family expressed in fibroblast-like clusters were similar across tissues - endometrium, adenomyosis and myometrium\n",
      "68\n",
      ".\n",
      "Figure 10. Fibroblasts and epithelial cell subtypes differ in endometrium and adenomyosis.\n",
      "Open in a new tab\n",
      "Panel A.\n",
      "UMAP of cell types and fibroblast trajectories in endometrium.\n",
      "B.\n",
      "UMAP of cell types and fibroblast trajectories in adenomyosis.\n",
      "Panel C.\n",
      "Progesterone resistance of epithelial subtypes in adenomyosis.\n",
      "Panel D.\n",
      "WNT signaling pathway network. From Yildiz et al, 2023\n",
      "68\n",
      ", with permission.\n",
      "Regarding the epithelial cells, P\n",
      "4\n",
      "-resistance was found in ciliated and unciliated cells in adenomyosis tissue versus matched eutopic endometrium (\n",
      "Figure 10 C\n",
      ") – this is a key finding regarding treatment efficiencies. Further data analysis revealed that the canonical WNT/SFRP pathway signaling was uniquely activated in adenomyosis unciliated and ciliated epithelium and paracrine signaling fibroblast-like mesenchymal: epithelial interaction networks (\n",
      "Figure 10D\n",
      "), opening new avenues for drug discovery for this disorder. The epithelial→endothelial transition reported by Liu et al\n",
      "67\n",
      "was not detected in this study and may be due to different adenomyosis lesion types, cycle phase, patient age, and other variables. Whether the findings of these two studies involving patients with diffuse adenomyosis also apply to other forms of adenomyosis (\n",
      "Figure 1\n",
      ") is unknown and is an opportunity to investigate cell features wherein novel therapeutics could be derived. Moreover, recent data suggest that the endometrial and vaginal microbiomes differ in patients with versus without adenomyosis\n",
      "69\n",
      ",\n",
      "70\n",
      ", raising the possibility of adjunctive antibiotic therapies for symptomatic patients with disease. Additionally, further study of the local and systemic immune system cells and their subtypes as well as in affected uterine tissues at single cell resolution are warranted.\n",
      "Challenges and opportunities in endometriosis and adenomyosis at the single cell level\n",
      "While single cell RNA sequencing and related innovative technologies and single cell atlases are providing unprecedented insights into human physiology and pathophysiology\n",
      "71\n",
      ",\n",
      "72\n",
      ", there are some unique challenges and opportunities in studying the steroid-hormone responsive human uterine compartments and their associated disorders. All research on human tissues, obtained under approved IRB protocols and written informed consent, requires accurate information about the study cohorts (cases and controls) to minimize confounders and maximize data accuracy and interpretation. Key are well annotated, de-identified patient metadata - e.g., demographics (age, race/ethnicity, BMI), pregnancy history, family history, personal habits, indication for surgical procedure, medical and surgical histories, current hormonal and nonhormonal medications, supplements, co-morbidities, the surgical procedure in which tissue samples were acquired, histologic confirmation that samples contain relevant tissues/cells (e.g., endometrium, myometrium, EMJ, endometriosis, adenomyosis, fibrosis, other), and hormonal status at the time of sampling, documented in a comprehensive database. Acquiring such information and processing samples for sequencing and data management demand teams of experienced study directors, clinical, surgical, laboratory, and pathology colleagues and services, genomics and sequencing cores, laboratory staff, clinical research coordinators, data entry staff, biostatisticians, and computational scientists. Some challenges and opportunities regarding studies specifically on the female reproductive tract are described below.\n",
      "The hormonal status of cases and controls whose tissues are under study is critical in studies on endometrium, endometriosis, and adenomyosis, as these tissues are estrogen- and progesterone-responsive. The context of the steroid hormone milieu (cycle phase, hormonal medications, pre- and post-menopausal status) is essential for data analysis and interpretation. If patients are not using hormonal preparations, then menstrual cycle phase determination must distinguish the estrogen-dominant proliferative phase and the sub-phases of the progesterone-dominant secretory phase\n",
      "44\n",
      ". Many patients, however, with symptomatic endometriosis and/or adenomyosis undergo surgery for pain and/or heavy menstrual bleeding for which combined contraceptive steroids, progestins, gonadotropin releasing hormone analogues, and/or aromatase inhibitors have been administered\n",
      "5\n",
      ",\n",
      "12\n",
      ". While synthetic progestins and progesterone signal through PR, they also elicit unique signaling pathways and transcriptomes in endometrial cells when administered systemically and in\n",
      "in vitro\n",
      "models\n",
      "39\n",
      "–\n",
      "43\n",
      ", which can impact some or most cell features discovered by scRNAseq analysis. It is also important to have hormonally similar controls without disease so hormonal status is not a confounder. Cohort studies restricted to a single menstrual cycle phase avoid hormonal confounders due to similar hormonal environments of the samples\n",
      "in situ\n",
      "at the time of acquisition. Also, precise timing is key, as the endometrial and endometriosis and/or adenomyosis lesion transcriptomes may change within a given cycle phase (e.g., early vs late secretory phase and, strikingly, at the opening of the implantation window in normal patients without disease\n",
      "19\n",
      "). A few scRNAseq studies have focused exclusively on one cycle phase versus another e.g., window of implantation (WOI) mid-secretory phase versus proliferative phase\n",
      "31\n",
      ", or timed menstrual effluent sampling\n",
      "32\n",
      ", which reduces confounding. Notably, studies comparing hormone treatments with no treatment among cases or controls are highly valuable to understand the efficacy of therapies on disease cell pathophysiology and could direct individualized cell-responsive therapeutics in the future.\n",
      "Lesion type, location and co-morbidities are additional key components for context and data interpretation. Age of the lesions is largely unknown, and thus comparing a lesion “type” within the same patient or between patients may be affected by different temporal histories of the lesions. Challenges also include the high prevalence of co-existing common gynecologic disorders (uterine fibroids, uterine polyps) in both cases and controls and defining the control group (no endometriosis or adenomyosis and without or with other gynecologic disorders). Standard operating procedures for biospecimen (tissue, blood, peritoneal fluid, other) collection, processing, storage, and distribution, as developed by Sheldon et al\n",
      "73\n",
      "and the World Endometriosis Research Foundation\n",
      "74\n",
      "–\n",
      "77\n",
      ", choice of sequencing technologies (Chapter 1), and pursuit of multiple computational strategies for comprehensive data analysis and validation of findings are key to success in working with these tissues and looking through single cell glasses.\n",
      "While numbers of cells sequenced enrich the phenotypic features of individual cell types, the numbers of subjects recruited is, by comparison, low, and notable is the limited diversity of the cohorts. There is a preponderance of White subjects recruited for scRNA sequencing of endometriosis and adenomyosis\n",
      "4\n",
      "(\n",
      "Table 1\n",
      "), opening a great opportunity to expand studies to include more diverse populations and comparative analyses of data across ethnicities in the future.\n",
      "Through the looking glass and beyond\n",
      "The analyses of human endometrium and the endometrial disorders - endometriosis and adenomyosis - at single cell resolution, described herein, complement and extend bulk tissue and isolated cell transcriptomics, epigenomics, and metabolomics, as well as blood and peritoneal fluid analyses, and\n",
      "in vitro\n",
      "cellular and tissue organoid/assembloid models\n",
      "1\n",
      "–\n",
      "4\n",
      ",\n",
      "78\n",
      "–\n",
      "82\n",
      ". Single cell transcriptomics is taking the field of “endometromics” to new heights – demonstrating commonalities and differences among cells of the same lineage and in different locations within a given tissue and across tissue types, inferred cell-cell communications, and insights into inflammation as a driver of endometriosis and adenomyosis, as well as the consequences of endometrial epithelial somatic mutations (shared in both disorders) on individual cell gene expression and a unique epithelial cell subtype associated with the risk of ovarian cancer in patients with endometriosis. The recent discovery linking Fusobacterium in ovarian endometriosis in women and antibiotic therapy reducing lesions in a mouse model of endometriosis and peritoneal colonization with Fusobacterium has set the stage for well controlled clinical trials for possible antibiotic therapies to reduce disease burden and for symptom management in patients. Also, the associated dysbiosis of the gut microbiome in mouse models opens opportunities for nutritional interventions in patients that could reduce the disease burden and ameliorate symptoms\n",
      "56\n",
      "–\n",
      "58\n",
      ". These new directions warrant further analysis of the circulating metabolome and single cell analyses of endometriosis in the setting of associated tissue microbiota.\n",
      "So, what will we see beyond the looking glass and into the future? The advances noted above will no doubt serve as launchpads for new and novel biomarker discovery and noninvasive diagnostic development and ultimately prognostic indicators of disease and treatment outcomes\n",
      "83\n",
      ", as well as individualized, patient-tailored therapeutics for symptoms associated with both disorders. Mapping human reproductive tissues and disorders with single cell genomics is certainly coming into its own\n",
      "84\n",
      ", and it is anticipated to benefit from the ever-growing single cell atlases (e.g.,\n",
      "www.humancellatlas.org\n",
      ";\n",
      "www.reproductivecellatlas.org\n",
      ") that are providing immensely rich, publicly available datasets. Indeed, current and future studies on endometriosis and adenomyosis will greatly benefit from the abundant single cell data on fundamental cell types and subtypes in other reproductive\n",
      "and non-reproductive\n",
      "tissues and diseases. This is likely, as cells in different tissues and disorders share some fundamental codes that identify their cell type, and that there are tissue-specific codes and tissue-specialized or altered codes (e.g., in the presence of inflammation or hormonal and other medical therapies)\n",
      "72\n",
      ",\n",
      "83\n",
      ",\n",
      "85\n",
      ". Data integration and comparative analyses of these codes and features across a multitude of endometrial and uterine disorders and their co-morbidities, as well as other tissues and diseases has the possibility to break the silos of discipline-specific study and capture the breadth of genomic features integrated with holistic features of patient metadata to render individualized health care and well-being across the life span and across diverse populations. Moreover, the gut microbiome-brain axis has been recognized as a key regulator of health and disease\n",
      "86\n",
      ", and, recently, dysbioses of the gut\n",
      "87\n",
      "and of the female reproductive tract\n",
      "88\n",
      "have been reported in patients with endometriosis and gastrointestinal (GI) symptoms. Notably, circulating- and lesion-derived estrogens are related to these symptoms and with pelvic pain\n",
      "89\n",
      ". Additionally, dysbiotic gut bacteria (the “estrobolome”) that convert conjugated estrogens to active estrogens and other bioactive metabolites are postulated to communicate with the brain regarding pain perception and GI symptoms and may also stimulate disease\n",
      "87\n",
      ",\n",
      "88\n",
      ",\n",
      "90\n",
      ". Studies on adenomyosis-associated altered microbiota in the endometrium and vagina are more limited compared to endometriosis-associated dysbiosis\n",
      "91\n",
      "–\n",
      "93\n",
      ". However, they are exciting as the paradigm of altered female reproductive tract microbiota in both these estrogen-dependent disorders is rapidly evolving. Identifying cell populations and sub-populations through single cell analyses in these various uterine and gut compartments and understanding mechanisms underlying their role in endometriosis-associated symptoms could open a novel approach to non-hormonal antibiotic and nutritional interventions for symptom relief, in well-designed clinical studies\n",
      "87\n",
      ",\n",
      "94\n",
      ". Data integration across compartments and the circulating metabolome, proteome, and immunome are additional players in the path forward.\n",
      "We have much to do to enable teams and new investigators to join this unprecedented effort in science and medicine beyond the looking glass – as technology progresses, so will biology, and then so will technology, and so will biology\n",
      "95\n",
      ".\n",
      "Acknowledgments\n",
      "Supported by the National Institutes of Health (NIH) Eunice Kennedy Shriver National Institute of Child Health & Human Development (NICHD) Center Grant P01 HD 106414\n",
      "REFERENCES\n",
      "1.\n",
      "Zondervan KT, Becker CM, Koga K, Missmer SA, Taylor RN, Viganò P. Endometriosis. Nat Rev Dis Primers. 2018;4(1):9. doi: 10.1038/s41572-018-0008-5\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "2.\n",
      "Zondervan KT, Becker CM, Missmer SA. Endometriosis. N Engl J Med. 2020;382(13):1244–1256. doi: 10.1056/NEJMra1810764\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "3.\n",
      "Bulun SE, Yilmaz BD, Sison C, et al. \n",
      "Endometriosis. Endocr Rev. 2019;40(4):1048–1079. doi: 10.1210/er.2018-00242\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "4.\n",
      "Giudice LC, Oskotsky TT, Falako S, Opoku-Anane J, Sirota M. Endometriosis in the era of precision medicine and impact on sexual and reproductive health across the lifespan and in diverse populations. FASEB Journal. 2023;37(9). doi: 10.1096/fj.202300907\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "5.\n",
      "Zhai J, Vannuccini S, Petraglia F, Giudice LC. Adenomyosis: Mechanisms and Pathogenesis. Semin Reprod Med. 2020;38(2-03):129–143. doi: 10.1055/s-0040-1716687\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "6.\n",
      "Bulun SE, Yildiz S, Adli M, Wei JJ. Adenomyosis pathogenesis: Insights from next-generation sequencing. Hum Reprod Update. 2021;27(6):1086–1097. doi: 10.1093/humupd/dmab017\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "7.\n",
      "Rossi M, Vannuccini S, Capezzuoli T, et al. \n",
      "Mechanisms and Pathogenesis of Adenomyosis. Curr Obstet Gynecol Rep. 2022;11(2):95–102. doi: 10.1007/s13669-022-00326-7\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "8.\n",
      "Shehzad S\n",
      "Adenomyosis: Current knowledge, Recent Advances and Future Perspective. doi: 10.33425/2639-9342.12223\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "9.\n",
      "Bulun SE, Yildiz S, Adli M, et al. \n",
      "Endometriosis and adenomyosis: shared pathophysiology. Fertil Steril. 2023;119(5):746–750. doi: 10.1016/j.fertnstert.2023.03.006\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "10.\n",
      "Upson K, Missmer SA. Epidemiology of Adenomyosis. Semin Reprod Med. 2020;38(02/03):089–107. doi: 10.1055/s-0040-1718920\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "11.\n",
      "Cope AG, Ainsworth AJ, Stewart EA. Current and Future Medical Therapies for Adenomyosis. Semin Reprod Med. 2020;38(02/03):151–156. doi: 10.1055/s-0040-1719016\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "12.\n",
      "Becker CM, Bokor A, Heikinheimo O, et al. \n",
      "ESHRE guideline: endometriosis. Hum Reprod Open. 2022;2022(2). doi: 10.1093/hropen/hoac009\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "13.\n",
      "Park BY, Yao R, Rossi J, Lee AW. Severe maternal morbidity associated with endometriosis: a population-based, retrospective cohort study. Fertil Steril. 2023;120(2):360–368. doi: 10.1016/j.fertnstert.2023.03.033\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "14.\n",
      "Culley L, Law C, Hudson N, et al. \n",
      "The social and psychological impact of endometriosis on women’s lives: A critical narrative review. Hum Reprod Update. 2013;19(6):625–639. doi: 10.1093/humupd/dmt027\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "15.\n",
      "Aerts L, Grangier L, Streuli I, et al. \n",
      "Psychosocial impact of endometriosis: From co-morbidity to intervention. Best Pract Res Clin Obstet Gynaecol. 2018;50:2–10. doi: 10.1016/j.bpobgyn.2018.01.008\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "16.\n",
      "Sperschneider ML, Hengartner MP, Kohl-Schwartz A, et al. \n",
      "Does endometriosis affect professional life? A matched case-control study in Switzerland, Germany and Austria. BMJ Open. 2019;9(1):e019570. doi: 10.1136/bmjopen-2017-019570\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "17.\n",
      "La Rosa VL, De Franciscis P, Barra F, et al. \n",
      "Quality of life in women with endometriosis: a narrative overview. Minerva Med. 2020;111(1). doi: 10.23736/S0026-4806.19.06298-0\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "18.\n",
      "Nelsen LM, Lenderking WR, Pokrzywinski R, et al. \n",
      "Experience of Symptoms and Disease Impact in Patients with Adenomyosis. The Patient - Patient-Centered Outcomes Research. 2018;11(3):319–328. doi: 10.1007/s40271-017-0284-2\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "19.\n",
      "Wang W, Vilella F, Alama P, et al. \n",
      "Single-cell transcriptomic atlas of the human endometrium during the menstrual cycle. Nat Med. 2020;26(10):1644–1653. doi: 10.1038/s41591-020-1040-z\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "20.\n",
      "Garcia-Alonso L, Handfield LF, Roberts K, et al. \n",
      "Mapping the temporal and spatial dynamics of the human endometrium in vivo and in vitro. Nat Genet. 2021;53(12):1698–1711. doi: 10.1038/s41588-021-00972-2\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "21.\n",
      "Queckbörner S, von Grothusen C, Boggavarapu NR, Francis RM, Davies LC, Gemzell-Danielsson K. Stromal Heterogeneity in the Human Proliferative Endometrium—A Single-Cell RNA Sequencing Study. J Pers Med. 2021;11(6):448. doi: 10.3390/jpm11060448\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "22.\n",
      "Jones RC, Karkanias J, Krasnow MA, et al. \n",
      "The Tabula Sapiens: A multiple-organ, single-cell transcriptomic atlas of humans. Science (1979). 2022;376(6594). doi: 10.1126/science.abl4896\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "23.\n",
      "Stephens VR, Rumph JT, Ameli S, Bruner-Tran KL, Osteen KG. The Potential Relationship Between Environmental Endocrine Disruptor Exposure and the Development of Endometriosis and Adenomyosis. Front Physiol. 2022;12. doi: 10.3389/fphys.2021.807685\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "24.\n",
      "Marshall A, Kommoss KF, Ortmann H, et al. \n",
      "Comparing gene expression in deep infiltrating endometriosis with adenomyosis uteri: evidence for dysregulation of oncogene pathways. Reproductive Biology and Endocrinology. 2023;21(1). doi: 10.1186/s12958-023-01083-9\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "25.\n",
      "Jiang Y, Jiang R, Cheng X, et al. \n",
      "Decreased expression of NR4A nuclear receptors in adenomyosis impairs endometrial decidualization. Mol Hum Reprod. 2016;22(9):655–668. doi: 10.1093/molehr/gaw042\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "26.\n",
      "Peng Y, Jin Z, Liu H, Xu C. Impaired decidualization of human endometrial stromal cells from women with adenomyosis†. Biol Reprod. 2021;104(5):1034–1044. doi: 10.1093/biolre/ioab017\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "27.\n",
      "Taylor HS, Kotlyar AM, Flores VA. Endometriosis is a chronic systemic disease: clinical challenges and novel innovations. The Lancet. 2021;397(10276):839–852. doi: 10.1016/S0140-6736(21)00389-5\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "28.\n",
      "Isaacson K, Loring M. Symptoms of Adenomyosis and Overlapping Diseases. Semin Reprod Med. 2020;38(02/03):144–150. doi: 10.1055/s-0040-1721795\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "29.\n",
      "O’Shea A, Figueiredo G, Lee SI. Imaging Diagnosis of Adenomyosis. Semin Reprod Med. 2020;38(02/03):119–128. doi: 10.1055/s-0040-1719017\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "30.\n",
      "Chen J, Porter AE, Kho KA. Current and Future Surgical and Interventional Management Options for Adenomyosis. Semin Reprod Med. 2020;38(02/03):157–167. doi: 10.1055/s-0040-1718921\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "31.\n",
      "Ma J, Zhang L, Zhan H, et al. \n",
      "Single-cell transcriptomic analysis of endometriosis provides insights into fibroblast fates and immune cell heterogeneity. Cell Biosci. 2021;11(1):125. doi: 10.1186/s13578-021-00637-x\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "32.\n",
      "Shih AJ, Adelson RP, Vashistha H, et al. \n",
      "Single-cell analysis of menstrual endometrial tissues defines phenotypes associated with endometriosis. BMC Med. 2022;20(1):315. doi: 10.1186/s12916-022-02500-3\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "33.\n",
      "Tan Y, Flynn WF, Sivajothi S, et al. \n",
      "Single-cell analysis of endometriosis reveals a coordinated transcriptional programme driving immunotolerance and angiogenesis across eutopic and ectopic tissues. Nat Cell Biol. 2022;24(8):1306–1318. doi: 10.1038/s41556-022-00961-5\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "34.\n",
      "Huang X, Wu L, Pei T, et al. \n",
      "Single-cell transcriptome analysis reveals endometrial immune microenvironment in minimal/mild endometriosis. Clin Exp Immunol. 2023;212(3):285–295. doi: 10.1093/cei/uxad029\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "35.\n",
      "Fonseca MAS, Haro M, Wright KN, et al. \n",
      "Single-cell transcriptomic analysis of endometriosis. Nat Genet. 2023;55(2):255–267. doi: 10.1038/s41588-022-01254-1\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "36.\n",
      "Almonte AL, Wang W, Flynn E, et al. \n",
      "ENDOMETRIAL SUBCELLULAR ALTERATIONS IN ENDOMETRIOSIS IDENTIFIED BY SINGLE-CELL ANALYSIS ACROSS THE MENSTRUAL CYCLE. Fertil Steril. 2023;120(4):e61–e62. doi: 10.1016/j.fertnstert.2023.08.204\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "37.\n",
      "Ponnampalam AP, Weston GC, Trajstman AC, Susil B, Rogers PAW. Molecular classification of human endometrial cycle stages by transcriptional profiling. Mol Hum Reprod. 2004;10(12):879–893. doi: 10.1093/molehr/gah121\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "38.\n",
      "Talbi S, Hamilton AE, Vo KC, et al. \n",
      "Molecular Phenotyping of Human Endometrium Distinguishes Menstrual Cycle Phases and Underlying Biological Processes in Normo-Ovulatory Women. Endocrinology. 2006;147(3):1097–1121. doi: 10.1210/en.2005-1076\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "39.\n",
      "Goldfien GA, Barragan F, Chen J, et al. \n",
      "Progestin-Containing Contraceptives Alter Expression of Host Defense-Related Genes of the Endometrium and Cervix. Reproductive Sciences. 2015;22(7):814–828. doi: 10.1177/1933719114565035\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "40.\n",
      "Shanmugasundaram U, Hilton JF, Critchfield JW, et al. \n",
      "Effects of the levonorgestrel-releasing intrauterine device on the immune microenvironment of the human cervix and endometrium. American Journal of Reproductive Immunology. 2016;76(2):137–148. doi: 10.1111/aji.12535\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "41.\n",
      "Smith-McCune KK, Hilton JF, Shanmugasundaram U, et al. \n",
      "Effects of depot-medroxyprogesterone acetate on the immune microenvironment of the human cervix and endometrium: implications for HIV susceptibility. Mucosal Immunol. 2017;10(5):1270–1278. doi: 10.1038/mi.2016.121\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "42.\n",
      "Smith-McCune K, Thomas R, Averbach S, et al. \n",
      "Differential Effects of the Hormonal and Copper Intrauterine Device on the Endometrial Transcriptome. Sci Rep. 2020;10(1):6888. doi: 10.1038/s41598-020-63798-8\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "43.\n",
      "Houshdaran S, Chen JC, Vallvé-Juanico J, et al. \n",
      "Progestins Related to Progesterone and Testosterone Elicit Divergent Human Endometrial Transcriptomes and Biofunctions. Int J Mol Sci. 2020;21(7):2625. doi: 10.3390/ijms21072625\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "44.\n",
      "Teh WT, Chung J, Holdsworth-Carson SJ, et al. \n",
      "A molecular staging model for accurately dating the endometrial biopsy. Nat Commun. 2023;14(1). doi: 10.1038/s41467-023-41979-z\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "45.\n",
      "Revised American Fertility Society Classification of Endometriosis: 1985. Fertil Steril. 1985;43(3):351–352. doi: 10.1016/S0015-0282(16)48430-X\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "46.\n",
      "Miravet-Valenciano J, Ruiz-Alonso M, Simón C. Transcriptomics of the human endometrium and embryo implantation. In: Human Reproductive and Prenatal Genetics. Elsevier; 2018:271–291. doi: 10.1016/B978-0-12-813570-9.00012-7\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "47.\n",
      "Barragan F, Irwin JC, Balayan S, et al. \n",
      "Human Endometrial Fibroblasts Derived from Mesenchymal Progenitors Inherit Progesterone Resistance and Acquire an Inflammatory Phenotype in the Endometrial Niche in Endometriosis. Biol Reprod. 2016;94(5). doi: 10.1095/biolreprod.115.136010\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "48.\n",
      "Wang X, Hawkins SM. Using advanced spatial and single-cell transcriptomics to characterize the human endometrium. Nat Genet. 2021;53(12):1628–1630. doi: 10.1038/s41588-021-00982-0\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "49.\n",
      "Tempest N, Maclean A, Hapangama DK. Endometrial Stem Cell Markers: Current Concepts and Unresolved Questions. Int J Mol Sci. 2018;19(10):3240. doi: 10.3390/ijms19103240\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "50.\n",
      "Cousins FL, Pandoy R, Jin S, Gargett CE. The Elusive Endometrial Epithelial Stem/Progenitor Cells. Front Cell Dev Biol. 2021;9. doi: 10.3389/fcell.2021.640319\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "51.\n",
      "Riccio LGC, Baracat EC, Chapron C, Batteux F, Abrão MS. The role of the B lymphocytes in endometriosis: A systematic review. J Reprod Immunol. 2017;123:29–34. doi: 10.1016/j.jri.2017.09.001\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "52.\n",
      "Vallvé-Juanico J, Houshdaran S, Giudice LC. The endometrial immune environment of women with endometriosis. Hum Reprod Update. 2019;25(5):565–592. doi: 10.1093/humupd/dmz018\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "53.\n",
      "Shen M, O’Donnell E, Leon G, et al. \n",
      "The role of endometrial B cells in normal endometrium and benign female reproductive pathologies: a systematic review. Hum Reprod Open. 2022;2022(1). doi: 10.1093/hropen/hoab043\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "54.\n",
      "Anglesio MS, Papadopoulos N, Ayhan A, et al. \n",
      "Cancer-Associated Mutations in Endometriosis without Cancer. New England Journal of Medicine. 2017;376(19):1835–1848. doi: 10.1056/nejmoa1614814\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "55.\n",
      "Yong PJ, Talhouk A, Anglesio MS. Somatic Genomic Events in Endometriosis: Review of the Literature and Approach to Phenotyping. Reproductive Sciences. 2021;28(10):2743–2757. doi: 10.1007/s43032-020-00451-9\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "56.\n",
      "Muraoka A, Suzuki M, Hamaguchi T, et al. \n",
      "Fusobacterium Infection Facilitates the Development of Endometriosis through the Phenotypic Transition of Endometrial Fibroblasts.; 2023.\n",
      "https://www.science.org\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "57.\n",
      "Chadchan SB, Naik SK, Popli P, et al. \n",
      "Gut microbiota and microbiota-derived metabolites promotes endometriosis. Cell Death Discov. 2023;9(1). doi: 10.1038/s41420-023-01309-0\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "58.\n",
      "Lucarini E, Di Pilato V, Parisio C, et al. \n",
      "Visceral sensitivity modulation by faecal microbiota transplantation: The active role of gut bacteria in pain persistence. Pain. 2022;163(5):861–877. doi: 10.1097/j.pain.0000000000002438\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "59.\n",
      "Brosens JJ, de Souza NM, Barker FG. Uterine junctional zone: function and disease. The Lancet. 1995;346(8974):558–560. doi: 10.1016/S0140-6736(95)91387-4\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "60.\n",
      "Noe M, Kunz G, Herbertz M, Mall G, Leyendecker G. The Cyclic Pattern of the Immunocytochemical Expression of Oestrogen and Progesterone Receptors in Human Myometrial and Endometrial Layers: Characterization of the Endometrial-Subendometrial Unit. Vol 14.; 1999.\n",
      "https://academic.oup.com/humrep/article/14/1/190/588131\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "61.\n",
      "Naftalin J, Jurkovic D. The endometrial–myometrial junction: a fresh look at a busy crossing. Ultrasound in Obstetrics & Gynecology. 2009;34(1):1–11. doi: 10.1002/uog.6432\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "62.\n",
      "Bourdon M, Santulli P, Jeljeli M, et al. \n",
      "Immunological changes associated with adenomyosis: A systematic review. Hum Reprod Update. 2021;27(1):108–129. doi: 10.1093/humupd/dmaa038\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "63.\n",
      "Gui T, Chen C, Zhang Z, et al. \n",
      "The disturbance of TH17-Treg cell balance in adenomyosis. Fertil Steril. 2014;101(2):506–514. doi: 10.1016/j.fertnstert.2013.10.050\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "64.\n",
      "Yang JH. Decreased expression of killer cell inhibitory receptors on natural killer cells in eutopic endometrium in women with adenomyosis. Human Reproduction. 2004;19(9):1974–1978. doi: 10.1093/humrep/deh372\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "65.\n",
      "Herndon CN, Aghajanova L, Balayan S, et al. \n",
      "Global Transcriptome Abnormalities of the Eutopic Endometrium From Women With Adenomyosis. Reproductive Sciences. 2016;23(10):1289–1303. doi: 10.1177/1933719116650758\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "66.\n",
      "Zhai J, Li S, Sen S, et al. \n",
      "Transcriptomic analysis supports collective endometrial cell migration in the pathogenesis of adenomyosis. Reprod Biomed Online. 2022;45(3):519–530. doi: 10.1016/j.rbmo.2022.05.007\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "67.\n",
      "Liu Z, Sun Z, Liu H, et al. \n",
      "Single-cell transcriptomic analysis of eutopic endometrium and ectopic lesions of adenomyosis. Cell Biosci. 2021;11(1):51. doi: 10.1186/s13578-021-00562-z\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "68.\n",
      "Yildiz S, Kinali M, Wei JJ, et al. \n",
      "Adenomyosis: single-cell transcriptomic analysis reveals a paracrine mesenchymal–epithelial interaction involving the WNT/SFRP pathway. Fertil Steril. 2023;119(5):869–882. doi: 10.1016/j.fertnstert.2023.01.041\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "69.\n",
      "Kunaseth J, Waiyaput W, Chanchaem P, et al. \n",
      "Vaginal microbiome of women with adenomyosis: A case-control study. PLoS One. 2022;17(2\n",
      "February). doi: 10.1371/journal.pone.0263283\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "70.\n",
      "Lin Q, Duan H, Wang S, et al. \n",
      "Endometrial microbiota in women with and without adenomyosis: A pilot study. Front Microbiol. 2023;14. doi: 10.3389/fmicb.2023.1075900\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "71.\n",
      "Baysoy A, Bai Z, Satija R, Fan R. The technological landscape and applications of single-cell multi-omics. Nat Rev Mol Cell Biol. 2023;24(10):695–713. doi: 10.1038/s41580-023-00615-w\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "72.\n",
      "Elmentaite R, Domínguez Conde C, Yang L, Teichmann SA. Single-cell atlases: shared and tissue-specific cell types across human organs. Nat Rev Genet. 2022;23(7):395–410. doi: 10.1038/s41576-022-00449-w\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "73.\n",
      "Sheldon E, Vo KC, McIntire RA, et al. \n",
      "Biobanking human endometrial tissue and blood specimens: standard operating procedure and importance to reproductive biology research and diagnostic development. Fertil Steril. 2011;95(6):2120–2122.e12. doi: 10.1016/j.fertnstert.2011.01.164\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "74.\n",
      "Becker CM, Laufer MR, Stratton P, et al. \n",
      "World Endometriosis Research Foundation Endometriosis Phenome and Biobanking Harmonisation Project: I. Surgical phenotype data collection in endometriosis research. Fertil Steril. 2014;102(5):1213–1222. doi: 10.1016/j.fertnstert.2014.07.709\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "75.\n",
      "Vitonis AF, Vincent K, Rahmioglu N, et al. \n",
      "World Endometriosis Research Foundation Endometriosis Phenome and biobanking harmonization project: II. Clinical and covariate phenotype data collection in endometriosis research. Fertil Steril. 2014;102(5):1223–1232. doi: 10.1016/j.fertnstert.2014.07.1244\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "76.\n",
      "Rahmioglu N, Fassbender A, Vitonis AF, et al. \n",
      "World Endometriosis Research Foundation Endometriosis Phenome and Biobanking Harmonization Project: III. Fluid biospecimen collection, processing, and storage in endometriosis research. Fertil Steril. 2014;102(5):1233–1243. doi: 10.1016/j.fertnstert.2014.07.1208\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "77.\n",
      "Fassbender A, Rahmioglu N, Vitonis AF, et al. \n",
      "World Endometriosis Research Foundation Endometriosis Phenome and Biobanking Harmonisation Project: IV. Tissue collection, processing, and storage in endometriosis research. Fertil Steril. 2014;102(5):1244–1253. doi: 10.1016/j.fertnstert.2014.07.1209\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "78.\n",
      "Boretto M, Maenhoudt N, Luo X, et al. \n",
      "Patient-derived organoids from endometrial disease capture clinical heterogeneity and are amenable to drug screening. Nat Cell Biol. 2019;21(8):1041–1051. doi: 10.1038/s41556-019-0360-z\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "79.\n",
      "Rawlings TM, Makwana K, Taylor DM, et al. \n",
      "Modelling the impact of decidual senescence on embryo implantation in human endometrial assembloids. Elife. 2021;10. doi: 10.7554/eLife.69603\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "80.\n",
      "Lou L, Kong S, Sun Y, Zhang Z, Wang H. Human Endometrial Organoids: Recent Research Progress and Potential Applications. Front Cell Dev Biol. 2022;10. doi: 10.3389/fcell.2022.844623\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "81.\n",
      "Juárez-barber E, Francés-herrero E, Corachán A, et al. \n",
      "Establishment of Adenomyosis Organoids as a Preclinical Model to Study Infertility. J Pers Med. 2022;12(2). doi: 10.3390/jpm12020219\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "82.\n",
      "Monnin N, Fattet AJ, Koscinski I. Endometriosis: Update of Pathophysiology, (Epi) Genetic and Environmental Involvement. Biomedicines. 2023;11(3):978. doi: 10.3390/biomedicines11030978\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "83.\n",
      "Hu W, Zhang Y, Mei J, Fang X. Spatial transcriptomics in human biomedical research and clinical application. Current Medicine. 2023;2(1). doi: 10.1007/s44194-023-00023-4\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "84.\n",
      "Marečková M, Massalha H, Lorenzi V, Vento-Tormo R. Mapping Human Reproduction with Single-Cell Genomics. Published online 2022. doi: 10.1146/annurev-genom-120121\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "85.\n",
      "Buechler MB, Pradhan RN, Krishnamurty AT, et al. \n",
      "Cross-tissue organization of the fibroblast lineage. Nature. 2021;593(7860):575–579. doi: 10.1038/s41586-021-03549-5\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "86.\n",
      "Cryan JF, O KJ, M Cowan CS, et al. \n",
      "The Microbiota-Gut-Brain Axis. Physiol Rev. 2019;99:1877–2013. doi: 10.1152/physrev.00018.2018.-The\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "87.\n",
      "Salliss ME, Farland LV., Mahnert ND, Herbst-Kralovetz MM. The role of gut and genital microbiota and the estrobolome in endometriosis, infertility and chronic pelvic pain. Hum Reprod Update. 2022;28(1):92–131. doi: 10.1093/humupd/dmab035\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "88.\n",
      "Uzuner C, Mak J, El-Assaad F, Condous G. The bidirectional relationship between endometriosis and microbiome. Front Endocrinol (Lausanne). 2023;14. doi: 10.3389/fendo.2023.1110824\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "89.\n",
      "Greaves E, Temp J, Esnal-Zufiurre A, Mechsner S, Horne AW, Saunders PTK. Estradiol Is a Critical Mediator of Macrophage-Nerve Cross Talk in Peritoneal Endometriosis. American Journal of Pathology. 2015;185(8):2286–2297. doi: 10.1016/j.ajpath.2015.04.012\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "90.\n",
      "Pai AHY, Wang YW, Lu PC, Wu HM, Xu JL, Huang HY. Gut Microbiome–Estrobolome Profile in Reproductive-Age Women with Endometriosis. Int J Mol Sci. 2023;24(22). doi: 10.3390/ijms242216301\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "91.\n",
      "Chen S, Gu Z, Zhang W, et al. \n",
      "The study of endometriosis and adenomyosis related microbiota in female lower genital tract in Northern Chinese population. Gynecology and Obstetrics Clinical Medicine. 2021;1(3):119–129. doi: 10.1016/j.gocm.2021.07.007\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "92.\n",
      "Lin Q, Duan H, Wang S, et al. \n",
      "Endometrial microbiota in women with and without adenomyosis: A pilot study. Front Microbiol. 2023;14. doi: 10.3389/fmicb.2023.1075900\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "93.\n",
      "Valdés-Bango M, Gracia M, Rubio E, et al. \n",
      "Comparative analysis of endometrial, vaginal, and gut microbiota in patients with and without adenomyosis. Acta Obstet Gynecol Scand. 2024;103(7):1271–1282. doi: 10.1111/aogs.14847\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "94.\n",
      "Griffiths MJ, Horne AW, Gibson DA, Roberts N, Saunders PTK. Endometriosis: recent advances that could accelerate diagnosis and improve care. Trends Mol Med. Published online 2024. doi: 10.1016/j.molmed.2024.06.008\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "95.\n",
      "The Cell editorial team. The integral partnership of biology and technology. Cell. 2022;185(15):2609–2610. doi: 10.1016/j.cell.2022.06.053\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n"
     ]
    }
   ],
   "source": [
    "soup = BeautifulSoup(response.text, \"html.parser\")\n",
    "print(soup.find(\"article\").get_text(separator=\"\\n\", strip=True))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6448c1bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "prueba = chain.invoke({\"query\": soup.find(\"article\").get_text(separator=\"\\n\", strip=True)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "85f263d4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'PaperTitle': 'Endometriosis and Adenomyosis Unveiled Through Single-cell Glasses',\n",
       " 'Publication': '2025',\n",
       " 'Authors': 'Linda C Giudice, Binya Liu, Juan C Irwin',\n",
       " 'Email': 'Linda.Giudice@ucsf.edu',\n",
       " 'Abstract': 'Single cell technologies are expanding our understanding of endometriosis and adenomyosis, sister disorders of the uterine endometrium that contain similar complements of lesion cell types but located in different niches – outside and inside the endometrium, respectively. Both diseases cause significant morbidity and impaired quality of life among those affected, and current therapies mitigate most symptoms although with highly variable efficacy, duration of effect, and frequent intolerable side effects. Thus, there is a pressing need for transformative approaches to develop individualized therapies for the variety of presentations of endometriosis and adenomyosis symptoms and heterogeneity of lesion types histologically and architecturally. Single cell technologies are transforming understanding human physiology and pathophysiology in the reproductive system and beyond. This manuscript reviews endometriosis and adenomyosis clinical characteristics and the recent studies focused on eutopic endometrium and ectopic lesions at single cell resolution, the myriad of cell types and subtypes, cell-cell communications, signaling pathways, applications for novel drug discovery and therapeutic approaches, and challenges and opportunities that accompany this type of research.',\n",
       " 'Introduction': 'Endometriosis and adenomyosis are common estrogen-driven uterine disorders derived mainly from the basalis endometrium, characterized by “ectopic” endometrial-like cells outside their normal location. A key feature distinguishing them is the location of the ectopic lesions. For endometriosis they are outside the uterus - mostly in the pelvis and invading pelvic organs 1 – 4 , and for adenomyosis they are within the uterine myometrium 5 – 8 . Endometriosis and adenomyosis disease lesions comprise similar phenotypic endometrial epithelium and mesenchymal cell types and display characteristics of inflammation, neoneuroangiogenesis, and fibrosis 1 – 9 . However, whether their cellular features are equivalent in different anatomic niches is unknown and is of importance for disease stratification in view of the vastly heterogeneous lesion architectural structures and for developing tailored, personalized, patient-specific treatments and prognostic disease outcome indicators.\\nThese complex “sister” conditions often occur concomitantly 10 and result in similar symptoms of pain and uterine bleeding treated medically (mostly hormonal) with variable efficacies and are associated with infertility and similar poor reproductive outcomes 1 , 4 – 13 . They also greatly impact quality of life of those affected 14 – 18 . Given the devastating impact of these disorders and limited success of effective therapies and disease prevention and cure, a transformative approach to disease phenotyping and novel, personalized therapeutics is essential to improve the lives of millions of patients with endometriosis and adenomyosis across the globe.\\nSingle cell technologies and precision medicine approaches are early in their application to these disorders and are beginning to provide insights into commonalities/differences of cell types in endometriosis, adenomyosis, and eutopic endometrium, and the role of their unique anatomical niches influencing cell behavior, disease establishment and pathophysiology. Moreover, single cell data and associated signaling pathways are anticipated to lead to novel drug candidate discovery to control symptoms, minimize disease burden resulting ultimately in cure, identifying diagnostic biomarkers, and informing disease classification and prognostics for disease courses. The time is now to move the needle on both disorders through single cell “glasses”!\\nRegarding endometrium, an atlas of normal, cycling endometrium has been derived at single cell resolution by several groups 19 – 22 (also see Chapter 3). These studies underscore heterogeneity of specific cell types and altered features in different hormonal states across the menstrual cycle and provide a key backdrop to studies at the single cell level of endometrium of patients with endometriosis and adenomyosis, endometriosis lesions, and adenomyosis lesions. Herein, we begin with an overview of clinical characteristics of endometriosis and adenomyosis (sister disorders but not identical twins), followed by a summary of current advances in scRNA sequencing data of endometriosis and adenomyosis disease lesions, corresponding eutopic endometrium, control endometrium, and disease-specific and common cell features across tissues. We conclude with a review of challenges of conducting research on human tissues, especially those that are steroid hormone responsive, and how these data inform looking to the future through single cell glasses.\\nEndometriosis and adenomyosis – sister disorders, not identical twins\\nEndometriosis and adenomyosis lesions both comprise uterine endometrial cells found outside their normal location ( Figure 1 ). For endometriosis, these cells form lesions - mostly in the pelvis - arriving mainly by retrograde menstruation and causing a profound inflammatory response 1 – 4 . Risks of developing endometriosis are ~ 50% genetic and ~50% environmental 2 , 4 . In contrast, adenomyosis-associated endometrial cells (epithelium and stroma) are found within the uterus – in the uterine myometrium – where they induce a local inflammatory response, smooth muscle hypertrophy and often an enlarged, globular uterus 5 – 8 . Risks of developing adenomyosis are multi-parity, prior uterine surgery, adult exposure to tamoxifen, and in utero exposures to estrogens 5 , 7 , 23 . Endometriosis has been more extensively studied due to ease of obtaining endometrial tissue and sampling disease lesions at surgery. Adenomyosis has had far less molecular, genetic, and genomic analyses, as uterine myometrial lesions and the endometrial-myometrial junction are accessible mainly from hysterectomy specimens, although the eutopic endometrium proper has been studied in adenomyosis patients due to ease of access by office biopsy.\\nFigure 1. Open in a new tab Comparison of endometriosis and adenomyosis pathogenesis, risk factors, diagnostic approaches, pathophysiology, symptoms, and therapies. Adapted from Bulun et al, 2023 9 , with permission.\\nDespite their differences in anatomic location, the two disorders share common origins, some molecular characteristics, symptoms, and treatment strategies 5 – 9 , 12 ( Figure 1 ). Both are diseases of the basalis endometrium with collective cell migration out of the uterus into the pelvis or into the muscle that may also involve epithelial to mesenchymal cell transition 5 , 6 , 9 . They share postulated origins of Mullerian remnant transformation and differentiation of endometrial stem/progenitor cells, and some common epithelial cancer driver mutations although they rarely transform into cancers 6 , 9 . They are estrogen-dependent, progesterone (P 4 )-resistant disorders, display increased estradiol (E 2 ) production in disease (due to aberrant aromatase expression in the epithelium and some stromal cells) and decreased E 2 metabolism and conversion to testosterone 2 , 4 , 6 , 9 , 23 , 25 , 26 . Both are also inflammatory disorders and have the activated inflammasome as the final common pathway for local and systemic inflammation 2 , 3 , 27 . And both can be induced in the developing fetus and/or adult (in animal models) and are associated in humans with endocrine disrupting chemicals found in the environment and personal care products 23 .\\nEndometriosis and adenomyosis share similar clinical features ( Figure 1 ) including dysmenorrhea, chronic pelvic pain, infertility, and poor pregnancy outcomes, and they often co-exist in the same patient and relapse post fertility-preserving surgery 5 – 7 , 9 , 12 , 28 . Symptom histories are key in diagnosing both disorders, but many symptoms overlap between them and with other disorders 2 , 4 . Standards for endometriosis and adenomyosis diagnosis are histologic confirmation of endometrial-like cells in lesions, and pelvic MRI and transvaginal ultrasound are useful for detecting endometriomas and deep disease as well as adenomyosis-associated widened endometrial-myometrial junction and adenomyomas 4 , 29 .\\nTreatment paradigms are similar for both disorders and include combined estrogen/ progestin contraceptive steroids, progestins alone, progestin-containing intrauterine systems, selective progesterone receptor modulators (SPRMs), selective estrogen receptor modulators (SERMs), GnRH analogs, aromatase inhibitors accompanied by ovulation suppression, and dopamine agonists, and medically assisted and advanced reproductive technologies for infertility 1 , 4 , 8 , 12 . Overall, medical therapies for pain have limited efficacy and/or intolerability for many patients. Endometriosis lesion removal or hysterectomy is largely effective for symptom management, although ~ 50% of patients recur within 5 years after conservative surgery and ~15% after hysterectomy 1 , 4 , 12 . Adenomyosis, on the other hand, can be excised but this is rarely performed due to technical challenges, and while hysterectomy is curative, it compromises fertility pursuits 30 .\\nThus, developing non-invasive diagnostics as stand-alone assessments or as adjuncts to imaging technologies, and innovative and personalized therapies are major goals for both disorders.',\n",
       " 'Methods': '',\n",
       " 'Results': 'Endometriosis through single cell glasses Subsequent to the single cell cartography of human endometrium published by Wang et al 19 (see Chapter 3), several groups have recently reported eutopic endometrial scRNAseq data of patients with endometriosis, endometriosis lesions and control eutopic endometrium 20 , 31 – 36 ( Table 1 ). Types of samples, inclusion/exclusion criteria, choice of controls, study designs, numbers of cells sequenced, and number of reads (i.e., base pairs sequenced) are key to data integrity and interpretation and to comparative analyses across tissues and studies. They have provided insights into origins of ectopic disease, the heterogeneity of cell types/subtypes, unique clusters and signatures of endometrium and ectopic lesions informing mechanisms and pathways involved in cellular dysfunctions relevant to pain and fertility compromise, novel cell-cell communications, relationships between lesions and eutopic tissue/cells, and biomarker discovery. Across these studies, samples were obtained in different hormonal milieu (menstrual cycle phase, exogenous hormones), and between 55,000 to 378,000 cells were sequenced ( Table 1 ). As cycle phase and exogenous hormone use are key drivers of endometrial gene expression 37 – 44 , comparison of data across studies can be challenging and mixed hormonal milieu are sometimes experienced within the same study in both cases and/or controls. Below we summarize results of studies focused exclusively on eutopic endometrium of patients with endometriosis relevant to endometrial function and dysfunction and biomarker discovery, followed by studies focused on ovarian endometriomas, superficial, and/or deep infiltrating disease, and eutopic endometrium, relevant to disease heterogeneity, pathophysiology, pathogenesis, and in the setting of progestins – commonly prescribed for symptom management of patients with endometriosis and dysmenorrhea and chronic pelvic pain.\\nTable 1. Highlights of recent endometriosis-related scRNAseq studies.\\nReference Tissue Analyzed Total cells sequenced Technology/Platform Total subjects Endometriosis Type and ASRM Stage Controls Race/ Ethnicity Hormones/IUD Cycle Phase Endometrial histology Ma 2021 endometrium, ovarian endometrioma 55,000 scRNAseq (10X) n=6 subjects (n=3 cases, n=3 controls) ASRM Stage III, IV ovarian endometriomas Healthy controls without endometriosis N/A None All in prolifeative phase Garcia-Alonso 2022 endometrium (functionalis, full thickness), endometriosis peritoneal lesions (red, white, black) 98,569 sequenced scRNAseq, snRNA seq, spatial profilng. Lesion microarray data ( GSE141549 ) n=3 functionalis n=6 full thickness; Microarray data controls: endom n=42, peritoneum n=12; n=9 red, 9 white, 11 black Peritoneal disease: red, white, black Healthy controls (functionalis layer) n=6 full thickness without reproductive disorders; normal peritoneum N/A None Proliferative, secretory; Microarray samples: Control Endo 17PE, 25SE; Perit 4PE, 8SE; Lesions: Red 2PE, 7 SE; White 5PE, 4SE; Black 6PE, 5 SE Shih 2022 menstrual endometrium 43,054 scRNAseq (10X) n=33 subjects (n=11 Dx, n=13 sx, n=9 controls) N/A; another group with Sx but no Dx. No endometriosis diagnosis. ? Other GYN disorders White: 10 Dx, 13 sx, 7 controls Black: 0 Dx, 0 sx, 1 control Hispanic: 0 Dx, 0 sx, 0 cont Mixed: 0 Dx,0 sx, 1 control Other: 1 Dx,0 sx, 0 control None on hormones, except 1 case used vaginal P4. Re-analysis showed no impact on results Menstrual (heaviest flow, mostly CD 1 or 2) Tan 2022 endometrium, enodmetriosis lesions 122,000 scRNAseq (10X) IMC organoids n=27 subjects (n=19 cases, n=8 controls; n=14 sequenced ASRM Stage II-IV, peritoneal lesions, ovarian “lesions”, organoids No endometriosis or infammatory conditions White: 9 cases, 3 controls Asian: 4 cases, 2 controls Hisp: 5 cases, 3 controls Black: 1 case, 0 controls Progestin [NETA, LVN oral, IUD, drosperinone, norelgestromin] +/− ethinyl E2, medroxy-progesterone acetate, copperIUD wkly PE: 3 cases, 2 control; inactive 3 cases, 0 control; mens: 1 case, 0 control, exog hormone effect: 7 cases, 7 control PE: 0 cases, 2 controls IE: 0 cases, 1 control ESE: 4 cases, 0 control N/A: 1 case, 1 control Fonseca 2023 endometrium, endometriosis lesions, unaffected ovary and peritoneum 373,851 digital scRNAseq n=21 subjects n=17 cases, n=4 controls n-54 specimens collected Cases: n=17 Endometrium Endometrioma Superficial and deep disesase n=9 w/o GYN disorders; n=8 w/ adenomyosis, uterine fibroids +/− polyp Controls n=4 n=3 PMP, 1 peri-menopause All no evidence of endometriosis, all w/ leiomyoma +/− adenomyosis +/− uterine polyp White: 13 cases,2 controls Black: 2 cases, 0 controls Asian: 1 case, 0 control Other: 1 case, 0 control Cases : n=14 on no hormones. n=1 w/ vaginal ring E+P; n=1 on E/T, P4; n=1 on E2+P4. Controls : 3 of 4 on E +/− P4; NETA; n=1 peri-menopause in luteal phase Cases : Proliferative phase: n=7; secretory phase: n=9; N/A (on hormones) n=3. Controls : Proliferative n=0, Secretory n=1, N/A (on hormones) n=3 Huang 2023 endometrium 128,243 scRNAseq (10X) n=10 sequenced n=6 cases, n=4 controls ASRM Stage I,II No endometriosis, benign ovarian cysts N/A None EPE: 3 cases, 3 controls MSE: 3 cases, 3 controls LSE: 1 control Almonte 2023 endometrium >250,000 scRNseq (10X) n=29 cases, 7 controls w/o disease and 8 controls with fibroids ASRM stage I-IV No endometriosis or other GYN d/o; No endo but + uterine fibroids N/A None Proliferative:12 cases, 5 controls. Secretory: 14 cases, 3 controls. Open in a new tab Abbreviations: CD=cycle day; dx=diagnosis; E=estradiol; ESE=early secretory phase; exog=exogenous; IE=interval endometrium; IMC=imaging mass cytometry; IUD=intrauterine device; LSE=late secretory phase; LVN=levonorgestrel; MSE=mid-secretory phase; N/A=not available; NETA=norethindrone acetate; P=progestin; P=progesterone; PE=proliferative phase; SE=secretory phase; sx=symptoms, w/= with; +/− = with and/or without. Adapted from Giudice et al 4 , with permission. Eutopic endometrium Window of implantation epithelial abnormalities in infertile endometriosis patients Endometrial biopsies of cases with stage I/II disease (revised American Society for Reproductive Medicine (rASRM) scoring system) 45 and infertility (n=3 WOI, n=3 early proliferative phase) versus n=7 fertile controls without disease (3 WOI, 3 proliferative, 1 late secretory phase) were studied by Huang et al 34 . A total of 128,243 cells were sequenced, and 30 type/cell type clusters were identified. Notably, a unique epithelial cell cluster in WOI endometrium of endometriosis patients was identified that lacked expression of two key genes associated with implantation: PAEP (progesterone-associated endometrial protein) and CXCL14 (chemokine (CXC motif) ligand 14) ( Figure 2 ). PAEP, also known as glycodelin A and placental protein 14, is involved in immune response regulation, promoting opening of the WOI by lowering maternal immune responses to an implanting embryo 46 . CXCL14 is involved in activating the innate immune response stimulating chemotaxis of uterine natural killer cells (uNK) and other monocytes to epithelial glands for implantation 46 . Other findings included absence of normal variation of endometrial uNK and T cell numbers across the cycle, and lower anti-inflammatory IL-10 and more pro-inflammatory cytokines expressed in endometrial immune cells in the WOI versus in the proliferative phase in cases and the opposite in controls. Notably, 11 ligand–receptor pairs were upregulated between endometrial immune and epithelial cells during the WOI of cases. Figure 2. Absence of key genes that open the window of implantation in eutopic endometrial epithelial cluster in infertile patients with Stage I/II endometriosis. Open in a new tab Panel A . t-SNE projection of endometrial cells colored based on their sample origin and cycle phase. Panel B . Expression patterns of PAEP and CXCL14 projected on t-distributed stochastic neighbour embedding (t-SNE) plot. From Huang et al, 2023 34 , with permission. Thus, lower epithelial receptivity markers, abnormal immune cell frequencies, a pro- inflammatory WOI, and new insights into cell-cell communications between epithelium and immune cells support an important role of the immune system and an adverse environment for embryo implantation for fertility and potentially for poor pregnancy outcomes in patients with stage I/II endometriosis who do achieve pregnancy versus fertile controls. Endometrial subcellular alterations across the cycle in endometriosis patients In a recent study, our group sequenced eutopic endometrium (237,620 cells) from 29 subjects with endometriosis (stages I/II and III/IV) and 15 controls without disease (7 healthy controls, 8 with symptomatic uterine fibroids), across the menstrual cycle 36 . We identified 5 main cell groups: stromal fibroblasts, smooth muscle, endothelial, epithelial, and immune. In the latter, further analysis revealed 18 sub-populations including uterine NK cells, T cell subsets, monocytes, macrophages, and dendritic cells. Epithelial subpopulations comprised glandular, luminal and ciliated cells, with abnormal expression of CXCL14, PAEP, CCL20 , MMP7, and SFRP4 genes in cases, regardless of phase. Endometrium in controls with FIGO Type 2 uterine fibroids revealed overexpression of WFDC2 and DAXX genes in cases. In immune cells, fibroblasts, and epithelial cells, cytokine production, stress signal and pro-inflammatory pathways in cases were noted, similar to Huang et al 34 findings, and enriched pathways in epithelial and ciliated cells related to migration and cell motility were higher in cases. Cell communication analysis revealed a complex network of inflammatory pathways, with significant involvement of classical monocytes and NK cells. Finally, genes implicated in ovarian cancer were upregulated in endometrium of cases compared to controls, confirming known association of endometriosis with specific subtypes of ovarian cancer 12 . Thus, in the eutopic endometrium in patients with endometriosis, there are numerous alterations in specific cell types across the cycle versus controls without disease and disease-free controls with uterine fibroids 36 . These can predispose to impaired fertility and tissue dyshomeostasis. Menstrual endometrium, disease pathophysiology, candidate biomarker discovery A recent scRNAseq study 32 of endometrial tissue in menstrual efflux (not biopsies) of n=11 cases with laparoscopic evidence of endometriosis (undefined stage) (11,924 cells) and n=9 healthy controls with no disease or symptoms (14,327 cells) revealed 18 cell clusters. These included abundant uNKs, stromal fibroblast, epithelial, B, T, and myeloid cell clusters, and a small plasmacytoid dendritic cell (pDC) cluster. Further analysis separated data by cases vs controls and revealed a marked decrease in uNK1 and uNK2 cells and significantly enriched B cells in cases ( Figure 3A ). Log2 odds ratio (OR) analysis confirmed uNK1/2 significantly enriched in controls and B cells greatly enriched in cases ( Figure 3B ). Moreover, menstrual endometrium scRNA seq revealed decreased decidualization markers in stromal fibroblasts (reflecting P 4 -resistance) and pro-inflammatory stromal cells, as observed by others in FACS-sorted cell analyses and in vitro cell culture studies 6 , 47 . Figure 3. Distinct Cell Compositions and Cell Enrichment in Menstrual Endometrium from patients with Endometriosis versus Controls and Model of Disease. Open in a new tab Panels A and B show low abundance of uNK1 and uNK2 cells and high abundance of B cells in patients with endometriosis. Panel C shows proposed pathogenesis and pathophysiology implied by the data. From Shih et al, 2022 32 , with permission. Overall, this study demonstrates that menstrual endometrium (commonly referred to as a “liquid biopsy of the endometrium”) reflects secretory endometrium abnormalities as possible biomarkers of disease. The proposed disease model is shown in Figure 3C wherein defective endometrial stromal fibroblast decidualization is driven by multiple factors including inflammation, chronic endometritis, stress, and/or P 4 -resistance – which all have clinical relevance for pregnancy establishment and success, as well as tissue homeostasis. This, in turn, may direct fibroblast differentiation to a chronic inflammatory phenotype and senescence, which may also impair decidualization. Reduced decidualization may also compromise the infiltration and proliferation of uNK cells, likely to be important for senescent cell removal and important for an optimal setting for embryo nidation 32 . Ectopic endometrium Superficial peritoneal lesions display basalis epithelial stem cell markers In their landmark spatial-temporal single cell transcriptomic study 48 , García-Alonso et al 20 analyzed endometrium (3 functionalis and 6 full thickness (functionalis and basalis) tissues) across the cycle and identified 14 unique cell clusters that localized to corresponding endometrial compartments and niches. The proposed endometrial epithelial stem-cell niche resides within the basalis layer and contains SOX9+ cells and other “stem” markers (e.g., LGR5, SSEA1, β-catenin, N-cadherin) 49 , 50 , and Garcia-Alonso et al 20 found proliferative SOX9+LGR5+ cells mapped to the basalis stem cell niche. Epithelial stem cells are believed to play a role in endometriosis pathogenesis, and by leveraging bulk (microarray) transcriptomic data of superficial peritoneal lesions (9 red, 9 white, 11 black) ( GSE141549 ), this team identified enrichment of SOX9+LGR5+ markers in peritoneal lesions in the proliferative phase, compared to normal endometrium (n=42) and peritoneum (n=12) 20 . A proliferative phenotype of endometriosis lesion epithelial cells is consistent with P 4 -resistance noted in endometriosis 4 , 6 . Moreover, dysfunctional epithelium as a major driver of endometrial diseases, including endometriosis, and discovery of SOX9+LGR5+ stem-like populations dominant in endometriosis peritoneal lesions are foundational observations about pathogenesis and pathophysiology of superficial endometriosis disease lesions deriving from the basalis layer epithelium. Endometrioma, superficial and deep infiltrating endometriosis cell atlas A comprehensive scRNAseq analysis of >370,000 individual cells from 8 ovarian endometriomas, 28 superficial and deep disease lesions, 10 eutopic endometrium, 4 unaffected ovary, and 4 endometriosis-free peritoneum samples, has resulted in a cell atlas of endometrial-type epithelial cells, stromal cells, and microenvironment cell populations across tissue types 35 . Samples were derived from 17 cases (9 only endometriosis; 8 endometriosis and adenomyosis, uterine fibroids, and/or uterine polyps). 14 were cycling (7 proliferative and 7 secretory phases), and 3 were on contraceptive steroids ( Table 1 ). Controls (n=4) were postmenopausal (3) or perimenopausal (1), without endometriosis but with uterine fibroids +/− adenomyosis +/− uterine polyps, and all were on hormonal replacement regimens. Histologic and macroscopic features of specimens from one patient, shown in Figure 4A , underscore the heterogeneity of the lesions by cell abundance, cell types, and tissue architecture. Figure 4. Endometriosis lesion scRNAseq analysis. Open in a new tab Panel A. Histologic (H&E staining) and architectural heterogeneity of endometriosis lesions. Panel B. Correlations Based on Cluster Frequency Across All Specimens Profiled by scRNAseq. Panel C. Epithelial differential gene expression and signaling pathways in endometriomas and superficial and deep disease (P < 0.05 and log2 FC > 1). From Fonseca et al, 2023 35 , with permission. The major cell types identified from scRNAseq of all cells from n=49 samples comprising the 5 major tissue classes (endometrioma, peritoneal endometriosis (deep and superficial), unaffected peritoneum, eutopic endometrium, unaffected ovary) included epithelial, smooth muscle, mesenchymal, endothelial, mast, myeloid, B, and T/NKT cells, and erythrocytes. Correlations based on cluster frequency across all specimens profiled by scRNAseq are shown in Figure 4B and reveal 3 main clusters: 1) mainly endometriosis specimens; 2) mix of endometriomas (5 of the 8 endometrioma specimens) and eutopic endometrium (5 of the 10 eutopic samples; 3) only controls: unaffected ovary and 5 of the 10 eutopic endometrium samples 35 . Endometrioma, endometrium, superficial and deep endometriosis epithelia revealed distinct cell subtypes and signaling pathways, and endometriomas displayed markedly different epithelial differential gene expression and signaling pathways compared to deep and superficial peritoneal disease and eutopic endometrium ( Figure 4C ). Interestingly, endometriomas displayed immune cell and complement activation and were enriched in B cells and plasma cells, suggesting possible infection in endometriomas and a unique role for B cells which have received limited attention in endometriosis pathophysiology 51 – 53 . Peritoneal disease was enriched in mast cells and T/NK-T cells engaging novel immune targets, and some histologically negative mesothelium surprisingly had disease signatures, although depth of lesion evaluation for endometriosis tissue may vary among pathologists. Endometrial-type epithelial cells and fibroblasts exhibited differential gene expression associated with deep or superficial disease status ( Figure 5 ). Interestingly, deep infiltrating disease epithelium displayed upregulation of nerve growth factor, suggesting a role of the epithelium in promoting innervation of this disease type which is particularly painful 12 . Figure 5. Molecular signatures of epithelial and stromal cells in superficial peritoneal and deep endometriosis differ. Open in a new tab Endometrial-type epithelial cells ( Panel A ) and fibroblasts ( Panel B ) exhibit DGE associated with deep or superficial status. From Fonseca et al, 2023 35 , with permission. While somatic mutations have been reported in endometrium and endometriosis lesions, mainly involving KRAS activating mutations and ARID1A loss of function mutations 54 , 55 , Fonseca et al 35 confirmed these somatic mutations in various lesions ( Figure 6A ) and demonstrated in vivo transcriptional consequences of these mutations at the single cell level ( Figure 6B ). ARID1A mutation was associated with pro-lymphangiogenic stroma ( Figure 6C ) and adjacent mesothelium pro-inflammatory features and ciliated epithelial signatures - consistent with ovarian oncogenic potential. It is well-known that endometrioid ovarian cancer (EOC) and clear cell ovarian cancer (CCOC) arise in association with endometriosis, suggesting endometrial-type epithelial cells may be precursors for these tumors. A multi-subject single cell deconvolution analysis of the single cell data ( Figure 7 ) revealed strong enrichment of signatures for ciliated endometrial-type epithelial cells compared with the other keratin-positive clusters. In contrast, high grade serous ovarian cancer, an epithelial ovarian cancer type, not believed to originate from endometriosis epithelial cells, showed no pattern for preferential enrichment of any epithelial cluster. Figure 6. Endometrium, endometriosis, and endometriomas share somatic mutations with transcriptional consequences. Open in a new tab Panel A. Mutations detected in each lesion. Panel B. Differential gene expression by mutation state. Panel C. Epithelial expression of lymph-angiogenesis markers by ARID1A staining status. From Fonseca et al, 2023 35 , with permission. Figure 7. Deconvolution Analysis Reveals Endometrial-type Epithelium Cell-type Signatures in Endometriosis-associated Ovarian Cancers. Open in a new tab Multi-subject single-cell deconvolution revealed strong enrichment of signatures for ciliated endometrial-type epithelial cells compared with the other keratin-positive clusters. HGSOC, an epithelial ovarian tumor type thought not to originate from endometriosis epithelial cells showed no pattern for preferential enrichment of any epithelial cluster. From Fonseca et al, 2023 35 , with permission. Key takeaways: Endometrium and all three lesion types versus control endometrium displayed different cell/molecular signatures across tissues, consistent with restructuring/transcriptional reprogramming in lesions. Epithelial and stroma differed by site with striking differential gene expression in endometriomas and peritoneal lesions suggesting these are distinct disease entities. Superficial peritoneal and deep infiltrative endometriosis epithelium and fibroblasts had some differential gene expression and pathways, but similar structural genes to each other, suggesting they are part of the same continuum and distinct from endometrioma. While this study is a tour de force of endometriosis lesion and eutopic endometrium single cell transcriptomic analysis giving major insights into cell- and tissue-specific features and functions, including oncogenic potential of endometrioma epithelial cells, it underscores some of the challenges in conducting these types of studies. These include the presence of co-existing gynecologic disorders that can confound data interpretation; cells sequenced comprise a minor population in most samples (e.g., <1% of cells profiled), and controls and normal tissues are challenging to obtain e.g., controls were peri- or post-menopausal on hormone replacement therapies and had other uterine disorders. Nonetheless, these data are important and provide foundational information for comparison with other studies on eutopic and ectopic lesions in endometriosis patients and controls without disease. Fibroblasts and immune cells: key contributors to a pro-inflammatory, angiogenic environment in endometriomas. Sequencing 55,000 single cells from 3 endometriomas, 3 matched eutopic endometrium, and 3 endometrial samples from controls without disease in the proliferative phase, Ma et al 31 found 9 cell type clusters: fibroblasts, macrophages/monocytes, neutrophils, and T, NK, epithelial, endothelial, and mast cells, and unknown. Thirteen fibroblast subtype clusters were identified with significant differences in subtype composition among ectopic endometrium, eutopic endometrium, and normal endometrium. Fibroblast subtypes from normal proliferative phase endometrium displayed enriched functions of DNA replication, repair, growth regulation, and metabolic processes. However, endometrioma fibroblasts comprised 4 subtypes: 1) cytokine, inflammatory response; 2) FGF stimulation, immune response; 3) extracellular matrix (ECM) organization, cell adhesion; 4) angiogenesis, hypoxia response, and epithelial-mesenchymal transition (EMT), all of which were enriched in MAPK, TNF, IL-17, TGF-β signaling pathways. Fibroblasts from eutopic endometrium of cases displayed many pathways similar to normal tissue, but some features suggested a “transition state” from normal endometrium to endometrioma, involving TGF-β, MAPK, Rho-Rack, NF-kB, JAK-STAT, and EMT. A developmental trajectory was postulated based on these fibroblast features, with endometrioma fibroblasts derived from endometrial fibroblasts 31 , much as the epithelial cell population has been proposed, based on shared epithelial somatic mutations in these two tissues 54 . Thus, endometrioma and paired eutopic endometrium fibroblasts differ, and the single cell data support their key roles in disease pathogenesis. Analysis of immune cells revealed they differed significantly in normal endometrium, endometrioma, and paired eutopic tissue ( Figure 8A ). Specifically, T cell and uNK cell frequencies were lower, uNK cells were more active, and macrophages (Mφ) were enriched and had features of tissue remodeling versus eutopic endometrium 31 , similar to data from Fonseca et al 35 . Thus, in the proliferative phase, fibroblasts and immune cell subpopulations contribute to a pro-inflammatory, angiogenic environment in endometriomas. Moreover cell-to-cell communication/interactome analysis revealed that immune cells stimulate fibroblast adhesion and growth in endometriomas ( Figure 8B ), underscoring the power of single cell analyses to inform the pathogenesis and pathophysiology of endometriomas and cell-cell communications. Furthermore, recent observations indicated that Fusobacterium 56 and dysbiosis of gut 57 , 58 altered systemic and potentially local immune functions, contributing to the development of endometriosis. Notably, while transcriptomic alterations of specific populations and sub- populations of the endometrium, lesions, and immune cells (local and peripheral) under these conditions awaits further study, antibiotic and nutritional therapies for dysbiosis is emerging as a new paradigm for treating endometriosis development and symptom management. Figure 8. Immune cell frequencies in endometriomas and communication networks of immune cell types with endometrioma fibroblasts. Open in a new tab The network supports interactions between immune and fibroblast cells in ovarian endometrioma pathogenesis and pathophysiology. From Ma et al, 2021 31 , with permission. Endometriosis peritoneal and ovary lesions of patients on progestins As progestins are a mainstay of medical therapy for endometriosis-related pain 12 , analysis of lesions and endometrium of subjects taking progestins is highly valuable in understanding their effects on cell types and tissue features. Tan et al 33 performed scRNAseq on 122,000 cells from 14 individuals, selective cell localization by imaging mass cytometry, and validations in endometrial epithelial organoid cultures. Peritoneal and/or ovarian disease and endometrium from 19 cases (rASRM stage II-IV) included 14 on norethindrone (NETA) + ethinyl estradiol (EthE 2 ), 2 on drosperinone + EthE 2 , 1 on levonorgestrel + EthE 2 , 1 on only NETA, 1 menstrual phase, and 8 controls with no endometriosis or inflammatory conditions: 4 on NETA + EthE 2 , 2 in proliferative phase, 1 on medroxyprogesterone acetate, 1 on norgestromin + EthE 2 . The peritoneal lesions had similar cell compositions as eutopic endometrium but dysregulated innate immune and vascular components - vastly different from ovarian endometriosis lesions. The authors identified a unique perivascular mural cell specific to the peritoneal lesions with features promoting neo- angiogenesis and immune cell trafficking. They also identified a progenitor-like epithelial subpopulation, different from the SOX9+LGR5+ epithelial subpopulation found by Garcia-Alfonso et al 20 . Analysis of peritoneal lesion myeloid populations of immunomodulatory Mφ and DCs revealed a coordinated immunotolerant phenotype in the disease microenvironment, promoting immunosurveillance escape and thus conducive to lesion establishment. Overall, the data demonstrate that immune and vascular components of peritoneal endometriosis favor neo-angiogenesis and an immune tolerant niche in the peritoneal cavity, beneficial for lesion establishment and growth. These are remarkable findings given that most subjects were on hormone suppressive therapies for pain [progestin (+ estrogen)] and raises the issue of how efficacious these therapies are on specific cells and the role of estrogenic components in disease management. This valuable single cell atlas of disease from patients largely on hormonal treatments offers the opportunity for comparisons of cell types and features in peritoneal lesions and endometriomas in the setting of other hormonal treatments and/or in various phases of the cycle to interpret efficacies of selective therapies on cell-specific pathways and cell-cell communications. Adenomyosis through single cell glasses Uterine compartments relevant to adenomyosis pathogenesis include the endometrial-myometrial junction (EMJ) [basalis and lower functionalis endometrium + sub-endometrial inner myometrium], all of Mullerian origin and displaying cycling estrogen receptors (ER) and progesterone receptors (PR) 59 – 61 ( Figure 9 ). Few molecular and genomic studies of the EMJ in adenomyosis uteri have been conducted, likely as access to basalis and inner myometrium requires hysterectomy versus the functionalis, which can be readily obtained by office biopsy. Figure 9. The endometrial-myometrial junction (EMJ). Open in a new tab Panel A. Histology of human uterine endometrium and myometrium compartments (H&E staining). Panel B : Transvaginal ultrasound midsagittal image of uterine compartments. Panel C. Schematic of uterine compartments. Adapted from Naftlin and Jurkovic, 2009 61 , with permission. Bulk analyses: Studies to date in adenomyosis patients using immunohistochemistry and qPCR reveal abnormal endometrial functionalis epithelial aromatase expression (thus high E 2 milieu), elevated stromal fibroblast IL-6, IL-8, IL-17 secretion, TH17-Treg imbalance, altered uNK receptors, and aberrant P 4 -signaling 5 , 26 , 62 – 64 – very similar to endometrium of patients with endometriosis except for IL-17 and TH17-Tregulatory cell imbalance. We conducted bulk RNA seq of endometrial functionalis 65 and a study on separate compartments of EMJ inner myometrium and endometrium from patients with diffuse adenomyosis versus controls in the proliferative phase 66 . ECM remodeling, inflammation, IL-17 signaling, and mTOR signaling were enriched in endometrium, and inflammation, pain, and neuronal signaling pathways in myometrium, consistent with clinical symptoms of this disorder. Single cell analyses: Two recent publications have analyzed adenomyosis lesions 67 , 68 . They are important beginnings to understanding the pathogenesis and pathophysiology of adenomyosis, building upon bulk transcriptomic data, which could be further mined in deconvolution studies to validate new and likely forthcoming additional single cell data. Liu et al 67 sequenced 42,292 cells derived from a control endometrium of a patient with uterine fibroids and eutopic (functionalis) endometrium and matched ectopic lesion from a patient with adenomyosis (type and menstrual cycle phase undefined). The data revealed seven distinct cell types. Comparison of the data from the lesion with those from eutopic matched endometrium and control endometrium revealed processes involved in progression of normal→endometrial→adenomyosis epithelial phenotypes. These included motility, proliferation, angiogenesis, inflammation, and cancer, consistent with the theory of collective cell migration and invasion into the myometrium of endometrial-derived cells. The novel data set revealed vascular mimicry (important in tumor formation) and epithelial transformation to endothelial cells. It is anticipated that the novel epithelial→endothelial transition phenotype will likely be mined in the future and expanded studies of adenomyosis and perhaps other gynecologic disorders and may serve as a unique cell type for drug targeting. Bulun and colleagues 68 reported results of single cell analyses of adenomyosis lesions and matched eutopic endometrium and myometrium from 3 subjects, all with diffuse adenomyosis and in the proliferative phase (total of 9 samples and 66,000 cells sequenced). They identified 11 cell types in eutopic endometrium of patients with adenomyosis ( Figure 10 A ). Fibroblast-like cells comprised a large fraction (36%), and pseudotime analysis revealed they originate from pericyte progenitors that differentiate to ER+ and PR+ endometrial stromal fibroblasts. Notably, the pseudotime trajectory was found to end with ciliated epithelial cells, suggesting mesenchymal- epithelial transition at play in this tissue. In adenomyosis lesions, they identified 13 cell types including 2 (smooth muscle cells and neurons) not found in the endometrium of the same patients ( Figure 10 B ). The adenomyosis lesions contained even more fibroblast-like cells compared to endometrium (50% versus 36%, respectively). In contrast to matched endometrium, the fibroblast-like cells did not originate from pericyte progenitors and differentiated to ECM-expressing fibroblasts and smooth muscle cells (SMC). Also, myometrial SMC were found to derive from fibroblast-like clusters. Moreover, the WNT inhibitor, secreted frizzled related protein (SFRP) family expressed in fibroblast-like clusters were similar across tissues - endometrium, adenomyosis and myometrium 68 . Figure 10. Fibroblasts and epithelial cell subtypes differ in endometrium and adenomyosis. Open in a new tab Panel A. UMAP of cell types and fibroblast trajectories in endometrium. B. UMAP of cell types and fibroblast trajectories in adenomyosis. Panel C. Progesterone resistance of epithelial subtypes in adenomyosis. Panel D. WNT signaling pathway network. From Yildiz et al, 2023 68 , with permission. Regarding the epithelial cells, P 4 -resistance was found in ciliated and unciliated cells in adenomyosis tissue versus matched eutopic endometrium ( Figure 10 C ) – this is a key finding regarding treatment efficiencies. Further data analysis revealed that the canonical WNT/SFRP pathway signaling was uniquely activated in adenomyosis unciliated and ciliated epithelium and paracrine signaling fibroblast-like mesenchymal: epithelial interaction networks ( Figure 10D ), opening new avenues for drug discovery for this disorder. The epithelial→endothelial transition reported by Liu et al 67 was not detected in this study and may be due to different adenomyosis lesion types, cycle phase, patient age, and other variables. Whether the findings of these two studies involving patients with diffuse adenomyosis also apply to other forms of adenomyosis ( Figure 1 ) is unknown and is an opportunity to investigate cell features wherein novel therapeutics could be derived. Moreover, recent data suggest that the endometrial and vaginal microbiomes differ in patients with versus without adenomyosis 69 , 70 , raising the possibility of adjunctive antibiotic therapies for symptomatic patients with disease. Additionally, further study of the local and systemic immune system cells and their subtypes as well as in affected uterine tissues at single cell resolution are warranted.',\n",
       " 'Discussion': '',\n",
       " 'Conclusion': '',\n",
       " 'URL': 'https://doi.org/10.1016/j.ajog.2024.08.043',\n",
       " 'Journal': 'Am J Obstet Gynecol'}"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "prueba"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "917d03c0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Endometriosis through single cell glasses Subsequent to the single cell '\n",
      " 'cartography of human endometrium published by Wang et al 19 (see Chapter 3), '\n",
      " 'several groups have recently reported eutopic endometrial scRNAseq data of '\n",
      " 'patients with endometriosis, endometriosis lesions and control eutopic '\n",
      " 'endometrium 20 , 31 – 36 ( Table 1 ). Types of samples, inclusion/exclusion '\n",
      " 'criteria, choice of controls, study designs, numbers of cells sequenced, and '\n",
      " 'number of reads (i.e., base pairs sequenced) are key to data integrity and '\n",
      " 'interpretation and to comparative analyses across tissues and studies. They '\n",
      " 'have provided insights into origins of ectopic disease, the heterogeneity of '\n",
      " 'cell types/subtypes, unique clusters and signatures of endometrium and '\n",
      " 'ectopic lesions informing mechanisms and pathways involved in cellular '\n",
      " 'dysfunctions relevant to pain and fertility compromise, novel cell-cell '\n",
      " 'communications, relationships between lesions and eutopic tissue/cells, and '\n",
      " 'biomarker discovery. Across these studies, samples were obtained in '\n",
      " 'different hormonal milieu (menstrual cycle phase, exogenous hormones), and '\n",
      " 'between 55,000 to 378,000 cells were sequenced ( Table 1 ). As cycle phase '\n",
      " 'and exogenous hormone use are key drivers of endometrial gene expression 37 '\n",
      " '– 44 , comparison of data across studies can be challenging and mixed '\n",
      " 'hormonal milieu are sometimes experienced within the same study in both '\n",
      " 'cases and/or controls. Below we summarize results of studies focused '\n",
      " 'exclusively on eutopic endometrium of patients with endometriosis relevant '\n",
      " 'to endometrial function and dysfunction and biomarker discovery, followed by '\n",
      " 'studies focused on ovarian endometriomas, superficial, and/or deep '\n",
      " 'infiltrating disease, and eutopic endometrium, relevant to disease '\n",
      " 'heterogeneity, pathophysiology, pathogenesis, and in the setting of '\n",
      " 'progestins – commonly prescribed for symptom management of patients with '\n",
      " 'endometriosis and dysmenorrhea and chronic pelvic pain.\\n'\n",
      " 'Table 1. Highlights of recent endometriosis-related scRNAseq studies.\\n'\n",
      " 'Reference Tissue Analyzed Total cells sequenced Technology/Platform Total '\n",
      " 'subjects Endometriosis Type and ASRM Stage Controls Race/ Ethnicity '\n",
      " 'Hormones/IUD Cycle Phase Endometrial histology Ma 2021 endometrium, ovarian '\n",
      " 'endometrioma 55,000 scRNAseq (10X) n=6 subjects (n=3 cases, n=3 controls) '\n",
      " 'ASRM Stage III, IV ovarian endometriomas Healthy controls without '\n",
      " 'endometriosis N/A None All in prolifeative phase Garcia-Alonso 2022 '\n",
      " 'endometrium (functionalis, full thickness), endometriosis peritoneal lesions '\n",
      " '(red, white, black) 98,569 sequenced scRNAseq, snRNA seq, spatial profilng. '\n",
      " 'Lesion microarray data ( GSE141549 ) n=3 functionalis n=6 full thickness; '\n",
      " 'Microarray data controls: endom n=42, peritoneum n=12; n=9 red, 9 white, 11 '\n",
      " 'black Peritoneal disease: red, white, black Healthy controls (functionalis '\n",
      " 'layer) n=6 full thickness without reproductive disorders; normal peritoneum '\n",
      " 'N/A None Proliferative, secretory; Microarray samples: Control Endo 17PE, '\n",
      " '25SE; Perit 4PE, 8SE; Lesions: Red 2PE, 7 SE; White 5PE, 4SE; Black 6PE, 5 '\n",
      " 'SE Shih 2022 menstrual endometrium 43,054 scRNAseq (10X) n=33 subjects (n=11 '\n",
      " 'Dx, n=13 sx, n=9 controls) N/A; another group with Sx but no Dx. No '\n",
      " 'endometriosis diagnosis. ? Other GYN disorders White: 10 Dx, 13 sx, 7 '\n",
      " 'controls Black: 0 Dx, 0 sx, 1 control Hispanic: 0 Dx, 0 sx, 0 cont Mixed: 0 '\n",
      " 'Dx,0 sx, 1 control Other: 1 Dx,0 sx, 0 control None on hormones, except 1 '\n",
      " 'case used vaginal P4. Re-analysis showed no impact on results Menstrual '\n",
      " '(heaviest flow, mostly CD 1 or 2) Tan 2022 endometrium, enodmetriosis '\n",
      " 'lesions 122,000 scRNAseq (10X) IMC organoids n=27 subjects (n=19 cases, n=8 '\n",
      " 'controls; n=14 sequenced ASRM Stage II-IV, peritoneal lesions, ovarian '\n",
      " '“lesions”, organoids No endometriosis or infammatory conditions White: 9 '\n",
      " 'cases, 3 controls Asian: 4 cases, 2 controls Hisp: 5 cases, 3 controls '\n",
      " 'Black: 1 case, 0 controls Progestin [NETA, LVN oral, IUD, drosperinone, '\n",
      " 'norelgestromin] +/− ethinyl E2, medroxy-progesterone acetate, copperIUD wkly '\n",
      " 'PE: 3 cases, 2 control; inactive 3 cases, 0 control; mens: 1 case, 0 '\n",
      " 'control, exog hormone effect: 7 cases, 7 control PE: 0 cases, 2 controls IE: '\n",
      " '0 cases, 1 control ESE: 4 cases, 0 control N/A: 1 case, 1 control Fonseca '\n",
      " '2023 endometrium, endometriosis lesions, unaffected ovary and peritoneum '\n",
      " '373,851 digital scRNAseq n=21 subjects n=17 cases, n=4 controls n-54 '\n",
      " 'specimens collected Cases: n=17 Endometrium Endometrioma Superficial and '\n",
      " 'deep disesase n=9 w/o GYN disorders; n=8 w/ adenomyosis, uterine fibroids '\n",
      " '+/− polyp Controls n=4 n=3 PMP, 1 peri-menopause All no evidence of '\n",
      " 'endometriosis, all w/ leiomyoma +/− adenomyosis +/− uterine polyp White: 13 '\n",
      " 'cases,2 controls Black: 2 cases, 0 controls Asian: 1 case, 0 control Other: '\n",
      " '1 case, 0 control Cases : n=14 on no hormones. n=1 w/ vaginal ring E+P; n=1 '\n",
      " 'on E/T, P4; n=1 on E2+P4. Controls : 3 of 4 on E +/− P4; NETA; n=1 '\n",
      " 'peri-menopause in luteal phase Cases : Proliferative phase: n=7; secretory '\n",
      " 'phase: n=9; N/A (on hormones) n=3. Controls : Proliferative n=0, Secretory '\n",
      " 'n=1, N/A (on hormones) n=3 Huang 2023 endometrium 128,243 scRNAseq (10X) '\n",
      " 'n=10 sequenced n=6 cases, n=4 controls ASRM Stage I,II No endometriosis, '\n",
      " 'benign ovarian cysts N/A None EPE: 3 cases, 3 controls MSE: 3 cases, 3 '\n",
      " 'controls LSE: 1 control Almonte 2023 endometrium >250,000 scRNseq (10X) n=29 '\n",
      " 'cases, 7 controls w/o disease and 8 controls with fibroids ASRM stage I-IV '\n",
      " 'No endometriosis or other GYN d/o; No endo but + uterine fibroids N/A None '\n",
      " 'Proliferative:12 cases, 5 controls. Secretory: 14 cases, 3 controls. Open in '\n",
      " 'a new tab Abbreviations: CD=cycle day; dx=diagnosis; E=estradiol; ESE=early '\n",
      " 'secretory phase; exog=exogenous; IE=interval endometrium; IMC=imaging mass '\n",
      " 'cytometry; IUD=intrauterine device; LSE=late secretory phase; '\n",
      " 'LVN=levonorgestrel; MSE=mid-secretory phase; N/A=not available; '\n",
      " 'NETA=norethindrone acetate; P=progestin; P=progesterone; PE=proliferative '\n",
      " 'phase; SE=secretory phase; sx=symptoms, w/= with; +/− = with and/or without. '\n",
      " 'Adapted from Giudice et al 4 , with permission. Eutopic endometrium Window '\n",
      " 'of implantation epithelial abnormalities in infertile endometriosis patients '\n",
      " 'Endometrial biopsies of cases with stage I/II disease (revised American '\n",
      " 'Society for Reproductive Medicine (rASRM) scoring system) 45 and infertility '\n",
      " '(n=3 WOI, n=3 early proliferative phase) versus n=7 fertile controls without '\n",
      " 'disease (3 WOI, 3 proliferative, 1 late secretory phase) were studied by '\n",
      " 'Huang et al 34 . A total of 128,243 cells were sequenced, and 30 type/cell '\n",
      " 'type clusters were identified. Notably, a unique epithelial cell cluster in '\n",
      " 'WOI endometrium of endometriosis patients was identified that lacked '\n",
      " 'expression of two key genes associated with implantation: PAEP '\n",
      " '(progesterone-associated endometrial protein) and CXCL14 (chemokine (CXC '\n",
      " 'motif) ligand 14) ( Figure 2 ). PAEP, also known as glycodelin A and '\n",
      " 'placental protein 14, is involved in immune response regulation, promoting '\n",
      " 'opening of the WOI by lowering maternal immune responses to an implanting '\n",
      " 'embryo 46 . CXCL14 is involved in activating the innate immune response '\n",
      " 'stimulating chemotaxis of uterine natural killer cells (uNK) and other '\n",
      " 'monocytes to epithelial glands for implantation 46 . Other findings included '\n",
      " 'absence of normal variation of endometrial uNK and T cell numbers across the '\n",
      " 'cycle, and lower anti-inflammatory IL-10 and more pro-inflammatory cytokines '\n",
      " 'expressed in endometrial immune cells in the WOI versus in the proliferative '\n",
      " 'phase in cases and the opposite in controls. Notably, 11 ligand–receptor '\n",
      " 'pairs were upregulated between endometrial immune and epithelial cells '\n",
      " 'during the WOI of cases. Figure 2. Absence of key genes that open the window '\n",
      " 'of implantation in eutopic endometrial epithelial cluster in infertile '\n",
      " 'patients with Stage I/II endometriosis. Open in a new tab Panel A . t-SNE '\n",
      " 'projection of endometrial cells colored based on their sample origin and '\n",
      " 'cycle phase. Panel B . Expression patterns of PAEP and CXCL14 projected on '\n",
      " 't-distributed stochastic neighbour embedding (t-SNE) plot. From Huang et al, '\n",
      " '2023 34 , with permission. Thus, lower epithelial receptivity markers, '\n",
      " 'abnormal immune cell frequencies, a pro- inflammatory WOI, and new insights '\n",
      " 'into cell-cell communications between epithelium and immune cells support an '\n",
      " 'important role of the immune system and an adverse environment for embryo '\n",
      " 'implantation for fertility and potentially for poor pregnancy outcomes in '\n",
      " 'patients with stage I/II endometriosis who do achieve pregnancy versus '\n",
      " 'fertile controls. Endometrial subcellular alterations across the cycle in '\n",
      " 'endometriosis patients In a recent study, our group sequenced eutopic '\n",
      " 'endometrium (237,620 cells) from 29 subjects with endometriosis (stages I/II '\n",
      " 'and III/IV) and 15 controls without disease (7 healthy controls, 8 with '\n",
      " 'symptomatic uterine fibroids), across the menstrual cycle 36 . We identified '\n",
      " '5 main cell groups: stromal fibroblasts, smooth muscle, endothelial, '\n",
      " 'epithelial, and immune. In the latter, further analysis revealed 18 '\n",
      " 'sub-populations including uterine NK cells, T cell subsets, monocytes, '\n",
      " 'macrophages, and dendritic cells. Epithelial subpopulations comprised '\n",
      " 'glandular, luminal and ciliated cells, with abnormal expression of CXCL14, '\n",
      " 'PAEP, CCL20 , MMP7, and SFRP4 genes in cases, regardless of phase. '\n",
      " 'Endometrium in controls with FIGO Type 2 uterine fibroids revealed '\n",
      " 'overexpression of WFDC2 and DAXX genes in cases. In immune cells, '\n",
      " 'fibroblasts, and epithelial cells, cytokine production, stress signal and '\n",
      " 'pro-inflammatory pathways in cases were noted, similar to Huang et al 34 '\n",
      " 'findings, and enriched pathways in epithelial and ciliated cells related to '\n",
      " 'migration and cell motility were higher in cases. Cell communication '\n",
      " 'analysis revealed a complex network of inflammatory pathways, with '\n",
      " 'significant involvement of classical monocytes and NK cells. Finally, genes '\n",
      " 'implicated in ovarian cancer were upregulated in endometrium of cases '\n",
      " 'compared to controls, confirming known association of endometriosis with '\n",
      " 'specific subtypes of ovarian cancer 12 . Thus, in the eutopic endometrium in '\n",
      " 'patients with endometriosis, there are numerous alterations in specific cell '\n",
      " 'types across the cycle versus controls without disease and disease-free '\n",
      " 'controls with uterine fibroids 36 . These can predispose to impaired '\n",
      " 'fertility and tissue dyshomeostasis. Menstrual endometrium, disease '\n",
      " 'pathophysiology, candidate biomarker discovery A recent scRNAseq study 32 of '\n",
      " 'endometrial tissue in menstrual efflux (not biopsies) of n=11 cases with '\n",
      " 'laparoscopic evidence of endometriosis (undefined stage) (11,924 cells) and '\n",
      " 'n=9 healthy controls with no disease or symptoms (14,327 cells) revealed 18 '\n",
      " 'cell clusters. These included abundant uNKs, stromal fibroblast, epithelial, '\n",
      " 'B, T, and myeloid cell clusters, and a small plasmacytoid dendritic cell '\n",
      " '(pDC) cluster. Further analysis separated data by cases vs controls and '\n",
      " 'revealed a marked decrease in uNK1 and uNK2 cells and significantly enriched '\n",
      " 'B cells in cases ( Figure 3A ). Log2 odds ratio (OR) analysis confirmed '\n",
      " 'uNK1/2 significantly enriched in controls and B cells greatly enriched in '\n",
      " 'cases ( Figure 3B ). Moreover, menstrual endometrium scRNA seq revealed '\n",
      " 'decreased decidualization markers in stromal fibroblasts (reflecting P 4 '\n",
      " '-resistance) and pro-inflammatory stromal cells, as observed by others in '\n",
      " 'FACS-sorted cell analyses and in vitro cell culture studies 6 , 47 . Figure '\n",
      " '3. Distinct Cell Compositions and Cell Enrichment in Menstrual Endometrium '\n",
      " 'from patients with Endometriosis versus Controls and Model of Disease. Open '\n",
      " 'in a new tab Panels A and B show low abundance of uNK1 and uNK2 cells and '\n",
      " 'high abundance of B cells in patients with endometriosis. Panel C shows '\n",
      " 'proposed pathogenesis and pathophysiology implied by the data. From Shih et '\n",
      " 'al, 2022 32 , with permission. Overall, this study demonstrates that '\n",
      " 'menstrual endometrium (commonly referred to as a “liquid biopsy of the '\n",
      " 'endometrium”) reflects secretory endometrium abnormalities as possible '\n",
      " 'biomarkers of disease. The proposed disease model is shown in Figure 3C '\n",
      " 'wherein defective endometrial stromal fibroblast decidualization is driven '\n",
      " 'by multiple factors including inflammation, chronic endometritis, stress, '\n",
      " 'and/or P 4 -resistance – which all have clinical relevance for pregnancy '\n",
      " 'establishment and success, as well as tissue homeostasis. This, in turn, may '\n",
      " 'direct fibroblast differentiation to a chronic inflammatory phenotype and '\n",
      " 'senescence, which may also impair decidualization. Reduced decidualization '\n",
      " 'may also compromise the infiltration and proliferation of uNK cells, likely '\n",
      " 'to be important for senescent cell removal and important for an optimal '\n",
      " 'setting for embryo nidation 32 . Ectopic endometrium Superficial peritoneal '\n",
      " 'lesions display basalis epithelial stem cell markers In their landmark '\n",
      " 'spatial-temporal single cell transcriptomic study 48 , García-Alonso et al '\n",
      " '20 analyzed endometrium (3 functionalis and 6 full thickness (functionalis '\n",
      " 'and basalis) tissues) across the cycle and identified 14 unique cell '\n",
      " 'clusters that localized to corresponding endometrial compartments and '\n",
      " 'niches. The proposed endometrial epithelial stem-cell niche resides within '\n",
      " 'the basalis layer and contains SOX9+ cells and other “stem” markers (e.g., '\n",
      " 'LGR5, SSEA1, β-catenin, N-cadherin) 49 , 50 , and Garcia-Alonso et al 20 '\n",
      " 'found proliferative SOX9+LGR5+ cells mapped to the basalis stem cell niche. '\n",
      " 'Epithelial stem cells are believed to play a role in endometriosis '\n",
      " 'pathogenesis, and by leveraging bulk (microarray) transcriptomic data of '\n",
      " 'superficial peritoneal lesions (9 red, 9 white, 11 black) ( GSE141549 ), '\n",
      " 'this team identified enrichment of SOX9+LGR5+ markers in peritoneal lesions '\n",
      " 'in the proliferative phase, compared to normal endometrium (n=42) and '\n",
      " 'peritoneum (n=12) 20 . A proliferative phenotype of endometriosis lesion '\n",
      " 'epithelial cells is consistent with P 4 -resistance noted in endometriosis 4 '\n",
      " ', 6 . Moreover, dysfunctional epithelium as a major driver of endometrial '\n",
      " 'diseases, including endometriosis, and discovery of SOX9+LGR5+ stem-like '\n",
      " 'populations dominant in endometriosis peritoneal lesions are foundational '\n",
      " 'observations about pathogenesis and pathophysiology of superficial '\n",
      " 'endometriosis disease lesions deriving from the basalis layer epithelium. '\n",
      " 'Endometrioma, superficial and deep infiltrating endometriosis cell atlas A '\n",
      " 'comprehensive scRNAseq analysis of >370,000 individual cells from 8 ovarian '\n",
      " 'endometriomas, 28 superficial and deep disease lesions, 10 eutopic '\n",
      " 'endometrium, 4 unaffected ovary, and 4 endometriosis-free peritoneum '\n",
      " 'samples, has resulted in a cell atlas of endometrial-type epithelial cells, '\n",
      " 'stromal cells, and microenvironment cell populations across tissue types 35 '\n",
      " '. Samples were derived from 17 cases (9 only endometriosis; 8 endometriosis '\n",
      " 'and adenomyosis, uterine fibroids, and/or uterine polyps). 14 were cycling '\n",
      " '(7 proliferative and 7 secretory phases), and 3 were on contraceptive '\n",
      " 'steroids ( Table 1 ). Controls (n=4) were postmenopausal (3) or '\n",
      " 'perimenopausal (1), without endometriosis but with uterine fibroids +/− '\n",
      " 'adenomyosis +/− uterine polyps, and all were on hormonal replacement '\n",
      " 'regimens. Histologic and macroscopic features of specimens from one patient, '\n",
      " 'shown in Figure 4A , underscore the heterogeneity of the lesions by cell '\n",
      " 'abundance, cell types, and tissue architecture. Figure 4. Endometriosis '\n",
      " 'lesion scRNAseq analysis. Open in a new tab Panel A. Histologic (H&E '\n",
      " 'staining) and architectural heterogeneity of endometriosis lesions. Panel B. '\n",
      " 'Correlations Based on Cluster Frequency Across All Specimens Profiled by '\n",
      " 'scRNAseq. Panel C. Epithelial differential gene expression and signaling '\n",
      " 'pathways in endometriomas and superficial and deep disease (P < 0.05 and '\n",
      " 'log2 FC > 1). From Fonseca et al, 2023 35 , with permission. The major cell '\n",
      " 'types identified from scRNAseq of all cells from n=49 samples comprising the '\n",
      " '5 major tissue classes (endometrioma, peritoneal endometriosis (deep and '\n",
      " 'superficial), unaffected peritoneum, eutopic endometrium, unaffected ovary) '\n",
      " 'included epithelial, smooth muscle, mesenchymal, endothelial, mast, myeloid, '\n",
      " 'B, and T/NKT cells, and erythrocytes. Correlations based on cluster '\n",
      " 'frequency across all specimens profiled by scRNAseq are shown in Figure 4B '\n",
      " 'and reveal 3 main clusters: 1) mainly endometriosis specimens; 2) mix of '\n",
      " 'endometriomas (5 of the 8 endometrioma specimens) and eutopic endometrium (5 '\n",
      " 'of the 10 eutopic samples; 3) only controls: unaffected ovary and 5 of the '\n",
      " '10 eutopic endometrium samples 35 . Endometrioma, endometrium, superficial '\n",
      " 'and deep endometriosis epithelia revealed distinct cell subtypes and '\n",
      " 'signaling pathways, and endometriomas displayed markedly different '\n",
      " 'epithelial differential gene expression and signaling pathways compared to '\n",
      " 'deep and superficial peritoneal disease and eutopic endometrium ( Figure 4C '\n",
      " '). Interestingly, endometriomas displayed immune cell and complement '\n",
      " 'activation and were enriched in B cells and plasma cells, suggesting '\n",
      " 'possible infection in endometriomas and a unique role for B cells which have '\n",
      " 'received limited attention in endometriosis pathophysiology 51 – 53 . '\n",
      " 'Peritoneal disease was enriched in mast cells and T/NK-T cells engaging '\n",
      " 'novel immune targets, and some histologically negative mesothelium '\n",
      " 'surprisingly had disease signatures, although depth of lesion evaluation for '\n",
      " 'endometriosis tissue may vary among pathologists. Endometrial-type '\n",
      " 'epithelial cells and fibroblasts exhibited differential gene expression '\n",
      " 'associated with deep or superficial disease status ( Figure 5 ). '\n",
      " 'Interestingly, deep infiltrating disease epithelium displayed upregulation '\n",
      " 'of nerve growth factor, suggesting a role of the epithelium in promoting '\n",
      " 'innervation of this disease type which is particularly painful 12 . Figure '\n",
      " '5. Molecular signatures of epithelial and stromal cells in superficial '\n",
      " 'peritoneal and deep endometriosis differ. Open in a new tab Endometrial-type '\n",
      " 'epithelial cells ( Panel A ) and fibroblasts ( Panel B ) exhibit DGE '\n",
      " 'associated with deep or superficial status. From Fonseca et al, 2023 35 , '\n",
      " 'with permission. While somatic mutations have been reported in endometrium '\n",
      " 'and endometriosis lesions, mainly involving KRAS activating mutations and '\n",
      " 'ARID1A loss of function mutations 54 , 55 , Fonseca et al 35 confirmed these '\n",
      " 'somatic mutations in various lesions ( Figure 6A ) and demonstrated in vivo '\n",
      " 'transcriptional consequences of these mutations at the single cell level ( '\n",
      " 'Figure 6B ). ARID1A mutation was associated with pro-lymphangiogenic stroma '\n",
      " '( Figure 6C ) and adjacent mesothelium pro-inflammatory features and '\n",
      " 'ciliated epithelial signatures - consistent with ovarian oncogenic '\n",
      " 'potential. It is well-known that endometrioid ovarian cancer (EOC) and clear '\n",
      " 'cell ovarian cancer (CCOC) arise in association with endometriosis, '\n",
      " 'suggesting endometrial-type epithelial cells may be precursors for these '\n",
      " 'tumors. A multi-subject single cell deconvolution analysis of the single '\n",
      " 'cell data ( Figure 7 ) revealed strong enrichment of signatures for ciliated '\n",
      " 'endometrial-type epithelial cells compared with the other keratin-positive '\n",
      " 'clusters. In contrast, high grade serous ovarian cancer, an epithelial '\n",
      " 'ovarian cancer type, not believed to originate from endometriosis epithelial '\n",
      " 'cells, showed no pattern for preferential enrichment of any epithelial '\n",
      " 'cluster. Figure 6. Endometrium, endometriosis, and endometriomas share '\n",
      " 'somatic mutations with transcriptional consequences. Open in a new tab Panel '\n",
      " 'A. Mutations detected in each lesion. Panel B. Differential gene expression '\n",
      " 'by mutation state. Panel C. Epithelial expression of lymph-angiogenesis '\n",
      " 'markers by ARID1A staining status. From Fonseca et al, 2023 35 , with '\n",
      " 'permission. Figure 7. Deconvolution Analysis Reveals Endometrial-type '\n",
      " 'Epithelium Cell-type Signatures in Endometriosis-associated Ovarian Cancers. '\n",
      " 'Open in a new tab Multi-subject single-cell deconvolution revealed strong '\n",
      " 'enrichment of signatures for ciliated endometrial-type epithelial cells '\n",
      " 'compared with the other keratin-positive clusters. HGSOC, an epithelial '\n",
      " 'ovarian tumor type thought not to originate from endometriosis epithelial '\n",
      " 'cells showed no pattern for preferential enrichment of any epithelial '\n",
      " 'cluster. From Fonseca et al, 2023 35 , with permission. Key takeaways: '\n",
      " 'Endometrium and all three lesion types versus control endometrium displayed '\n",
      " 'different cell/molecular signatures across tissues, consistent with '\n",
      " 'restructuring/transcriptional reprogramming in lesions. Epithelial and '\n",
      " 'stroma differed by site with striking differential gene expression in '\n",
      " 'endometriomas and peritoneal lesions suggesting these are distinct disease '\n",
      " 'entities. Superficial peritoneal and deep infiltrative endometriosis '\n",
      " 'epithelium and fibroblasts had some differential gene expression and '\n",
      " 'pathways, but similar structural genes to each other, suggesting they are '\n",
      " 'part of the same continuum and distinct from endometrioma. While this study '\n",
      " 'is a tour de force of endometriosis lesion and eutopic endometrium single '\n",
      " 'cell transcriptomic analysis giving major insights into cell- and '\n",
      " 'tissue-specific features and functions, including oncogenic potential of '\n",
      " 'endometrioma epithelial cells, it underscores some of the challenges in '\n",
      " 'conducting these types of studies. These include the presence of co-existing '\n",
      " 'gynecologic disorders that can confound data interpretation; cells sequenced '\n",
      " 'comprise a minor population in most samples (e.g., <1% of cells profiled), '\n",
      " 'and controls and normal tissues are challenging to obtain e.g., controls '\n",
      " 'were peri- or post-menopausal on hormone replacement therapies and had other '\n",
      " 'uterine disorders. Nonetheless, these data are important and provide '\n",
      " 'foundational information for comparison with other studies on eutopic and '\n",
      " 'ectopic lesions in endometriosis patients and controls without disease. '\n",
      " 'Fibroblasts and immune cells: key contributors to a pro-inflammatory, '\n",
      " 'angiogenic environment in endometriomas. Sequencing 55,000 single cells from '\n",
      " '3 endometriomas, 3 matched eutopic endometrium, and 3 endometrial samples '\n",
      " 'from controls without disease in the proliferative phase, Ma et al 31 found '\n",
      " '9 cell type clusters: fibroblasts, macrophages/monocytes, neutrophils, and '\n",
      " 'T, NK, epithelial, endothelial, and mast cells, and unknown. Thirteen '\n",
      " 'fibroblast subtype clusters were identified with significant differences in '\n",
      " 'subtype composition among ectopic endometrium, eutopic endometrium, and '\n",
      " 'normal endometrium. Fibroblast subtypes from normal proliferative phase '\n",
      " 'endometrium displayed enriched functions of DNA replication, repair, growth '\n",
      " 'regulation, and metabolic processes. However, endometrioma fibroblasts '\n",
      " 'comprised 4 subtypes: 1) cytokine, inflammatory response; 2) FGF '\n",
      " 'stimulation, immune response; 3) extracellular matrix (ECM) organization, '\n",
      " 'cell adhesion; 4) angiogenesis, hypoxia response, and epithelial-mesenchymal '\n",
      " 'transition (EMT), all of which were enriched in MAPK, TNF, IL-17, TGF-β '\n",
      " 'signaling pathways. Fibroblasts from eutopic endometrium of cases displayed '\n",
      " 'many pathways similar to normal tissue, but some features suggested a '\n",
      " '“transition state” from normal endometrium to endometrioma, involving TGF-β, '\n",
      " 'MAPK, Rho-Rack, NF-kB, JAK-STAT, and EMT. A developmental trajectory was '\n",
      " 'postulated based on these fibroblast features, with endometrioma fibroblasts '\n",
      " 'derived from endometrial fibroblasts 31 , much as the epithelial cell '\n",
      " 'population has been proposed, based on shared epithelial somatic mutations '\n",
      " 'in these two tissues 54 . Thus, endometrioma and paired eutopic endometrium '\n",
      " 'fibroblasts differ, and the single cell data support their key roles in '\n",
      " 'disease pathogenesis. Analysis of immune cells revealed they differed '\n",
      " 'significantly in normal endometrium, endometrioma, and paired eutopic tissue '\n",
      " '( Figure 8A ). Specifically, T cell and uNK cell frequencies were lower, uNK '\n",
      " 'cells were more active, and macrophages (Mφ) were enriched and had features '\n",
      " 'of tissue remodeling versus eutopic endometrium 31 , similar to data from '\n",
      " 'Fonseca et al 35 . Thus, in the proliferative phase, fibroblasts and immune '\n",
      " 'cell subpopulations contribute to a pro-inflammatory, angiogenic environment '\n",
      " 'in endometriomas. Moreover cell-to-cell communication/interactome analysis '\n",
      " 'revealed that immune cells stimulate fibroblast adhesion and growth in '\n",
      " 'endometriomas ( Figure 8B ), underscoring the power of single cell analyses '\n",
      " 'to inform the pathogenesis and pathophysiology of endometriomas and '\n",
      " 'cell-cell communications. Furthermore, recent observations indicated that '\n",
      " 'Fusobacterium 56 and dysbiosis of gut 57 , 58 altered systemic and '\n",
      " 'potentially local immune functions, contributing to the development of '\n",
      " 'endometriosis. Notably, while transcriptomic alterations of specific '\n",
      " 'populations and sub- populations of the endometrium, lesions, and immune '\n",
      " 'cells (local and peripheral) under these conditions awaits further study, '\n",
      " 'antibiotic and nutritional therapies for dysbiosis is emerging as a new '\n",
      " 'paradigm for treating endometriosis development and symptom management. '\n",
      " 'Figure 8. Immune cell frequencies in endometriomas and communication '\n",
      " 'networks of immune cell types with endometrioma fibroblasts. Open in a new '\n",
      " 'tab The network supports interactions between immune and fibroblast cells in '\n",
      " 'ovarian endometrioma pathogenesis and pathophysiology. From Ma et al, 2021 '\n",
      " '31 , with permission. Endometriosis peritoneal and ovary lesions of patients '\n",
      " 'on progestins As progestins are a mainstay of medical therapy for '\n",
      " 'endometriosis-related pain 12 , analysis of lesions and endometrium of '\n",
      " 'subjects taking progestins is highly valuable in understanding their effects '\n",
      " 'on cell types and tissue features. Tan et al 33 performed scRNAseq on '\n",
      " '122,000 cells from 14 individuals, selective cell localization by imaging '\n",
      " 'mass cytometry, and validations in endometrial epithelial organoid cultures. '\n",
      " 'Peritoneal and/or ovarian disease and endometrium from 19 cases (rASRM stage '\n",
      " 'II-IV) included 14 on norethindrone (NETA) + ethinyl estradiol (EthE 2 ), 2 '\n",
      " 'on drosperinone + EthE 2 , 1 on levonorgestrel + EthE 2 , 1 on only NETA, 1 '\n",
      " 'menstrual phase, and 8 controls with no endometriosis or inflammatory '\n",
      " 'conditions: 4 on NETA + EthE 2 , 2 in proliferative phase, 1 on '\n",
      " 'medroxyprogesterone acetate, 1 on norgestromin + EthE 2 . The peritoneal '\n",
      " 'lesions had similar cell compositions as eutopic endometrium but '\n",
      " 'dysregulated innate immune and vascular components - vastly different from '\n",
      " 'ovarian endometriosis lesions. The authors identified a unique perivascular '\n",
      " 'mural cell specific to the peritoneal lesions with features promoting neo- '\n",
      " 'angiogenesis and immune cell trafficking. They also identified a '\n",
      " 'progenitor-like epithelial subpopulation, different from the SOX9+LGR5+ '\n",
      " 'epithelial subpopulation found by Garcia-Alfonso et al 20 . Analysis of '\n",
      " 'peritoneal lesion myeloid populations of immunomodulatory Mφ and DCs '\n",
      " 'revealed a coordinated immunotolerant phenotype in the disease '\n",
      " 'microenvironment, promoting immunosurveillance escape and thus conducive to '\n",
      " 'lesion establishment. Overall, the data demonstrate that immune and vascular '\n",
      " 'components of peritoneal endometriosis favor neo-angiogenesis and an immune '\n",
      " 'tolerant niche in the peritoneal cavity, beneficial for lesion establishment '\n",
      " 'and growth. These are remarkable findings given that most subjects were on '\n",
      " 'hormone suppressive therapies for pain [progestin (+ estrogen)] and raises '\n",
      " 'the issue of how efficacious these therapies are on specific cells and the '\n",
      " 'role of estrogenic components in disease management. This valuable single '\n",
      " 'cell atlas of disease from patients largely on hormonal treatments offers '\n",
      " 'the opportunity for comparisons of cell types and features in peritoneal '\n",
      " 'lesions and endometriomas in the setting of other hormonal treatments and/or '\n",
      " 'in various phases of the cycle to interpret efficacies of selective '\n",
      " 'therapies on cell-specific pathways and cell-cell communications. '\n",
      " 'Adenomyosis through single cell glasses Uterine compartments relevant to '\n",
      " 'adenomyosis pathogenesis include the endometrial-myometrial junction (EMJ) '\n",
      " '[basalis and lower functionalis endometrium + sub-endometrial inner '\n",
      " 'myometrium], all of Mullerian origin and displaying cycling estrogen '\n",
      " 'receptors (ER) and progesterone receptors (PR) 59 – 61 ( Figure 9 ). Few '\n",
      " 'molecular and genomic studies of the EMJ in adenomyosis uteri have been '\n",
      " 'conducted, likely as access to basalis and inner myometrium requires '\n",
      " 'hysterectomy versus the functionalis, which can be readily obtained by '\n",
      " 'office biopsy. Figure 9. The endometrial-myometrial junction (EMJ). Open in '\n",
      " 'a new tab Panel A. Histology of human uterine endometrium and myometrium '\n",
      " 'compartments (H&E staining). Panel B : Transvaginal ultrasound midsagittal '\n",
      " 'image of uterine compartments. Panel C. Schematic of uterine compartments. '\n",
      " 'Adapted from Naftlin and Jurkovic, 2009 61 , with permission. Bulk analyses: '\n",
      " 'Studies to date in adenomyosis patients using immunohistochemistry and qPCR '\n",
      " 'reveal abnormal endometrial functionalis epithelial aromatase expression '\n",
      " '(thus high E 2 milieu), elevated stromal fibroblast IL-6, IL-8, IL-17 '\n",
      " 'secretion, TH17-Treg imbalance, altered uNK receptors, and aberrant P 4 '\n",
      " '-signaling 5 , 26 , 62 – 64 – very similar to endometrium of patients with '\n",
      " 'endometriosis except for IL-17 and TH17-Tregulatory cell imbalance. We '\n",
      " 'conducted bulk RNA seq of endometrial functionalis 65 and a study on '\n",
      " 'separate compartments of EMJ inner myometrium and endometrium from patients '\n",
      " 'with diffuse adenomyosis versus controls in the proliferative phase 66 . ECM '\n",
      " 'remodeling, inflammation, IL-17 signaling, and mTOR signaling were enriched '\n",
      " 'in endometrium, and inflammation, pain, and neuronal signaling pathways in '\n",
      " 'myometrium, consistent with clinical symptoms of this disorder. Single cell '\n",
      " 'analyses: Two recent publications have analyzed adenomyosis lesions 67 , 68 '\n",
      " '. They are important beginnings to understanding the pathogenesis and '\n",
      " 'pathophysiology of adenomyosis, building upon bulk transcriptomic data, '\n",
      " 'which could be further mined in deconvolution studies to validate new and '\n",
      " 'likely forthcoming additional single cell data. Liu et al 67 sequenced '\n",
      " '42,292 cells derived from a control endometrium of a patient with uterine '\n",
      " 'fibroids and eutopic (functionalis) endometrium and matched ectopic lesion '\n",
      " 'from a patient with adenomyosis (type and menstrual cycle phase undefined). '\n",
      " 'The data revealed seven distinct cell types. Comparison of the data from the '\n",
      " 'lesion with those from eutopic matched endometrium and control endometrium '\n",
      " 'revealed processes involved in progression of normal→endometrial→adenomyosis '\n",
      " 'epithelial phenotypes. These included motility, proliferation, angiogenesis, '\n",
      " 'inflammation, and cancer, consistent with the theory of collective cell '\n",
      " 'migration and invasion into the myometrium of endometrial-derived cells. The '\n",
      " 'novel data set revealed vascular mimicry (important in tumor formation) and '\n",
      " 'epithelial transformation to endothelial cells. It is anticipated that the '\n",
      " 'novel epithelial→endothelial transition phenotype will likely be mined in '\n",
      " 'the future and expanded studies of adenomyosis and perhaps other gynecologic '\n",
      " 'disorders and may serve as a unique cell type for drug targeting. Bulun and '\n",
      " 'colleagues 68 reported results of single cell analyses of adenomyosis '\n",
      " 'lesions and matched eutopic endometrium and myometrium from 3 subjects, all '\n",
      " 'with diffuse adenomyosis and in the proliferative phase (total of 9 samples '\n",
      " 'and 66,000 cells sequenced). They identified 11 cell types in eutopic '\n",
      " 'endometrium of patients with adenomyosis ( Figure 10 A ). Fibroblast-like '\n",
      " 'cells comprised a large fraction (36%), and pseudotime analysis revealed '\n",
      " 'they originate from pericyte progenitors that differentiate to ER+ and PR+ '\n",
      " 'endometrial stromal fibroblasts. Notably, the pseudotime trajectory was '\n",
      " 'found to end with ciliated epithelial cells, suggesting mesenchymal- '\n",
      " 'epithelial transition at play in this tissue. In adenomyosis lesions, they '\n",
      " 'identified 13 cell types including 2 (smooth muscle cells and neurons) not '\n",
      " 'found in the endometrium of the same patients ( Figure 10 B ). The '\n",
      " 'adenomyosis lesions contained even more fibroblast-like cells compared to '\n",
      " 'endometrium (50% versus 36%, respectively). In contrast to matched '\n",
      " 'endometrium, the fibroblast-like cells did not originate from pericyte '\n",
      " 'progenitors and differentiated to ECM-expressing fibroblasts and smooth '\n",
      " 'muscle cells (SMC). Also, myometrial SMC were found to derive from '\n",
      " 'fibroblast-like clusters. Moreover, the WNT inhibitor, secreted frizzled '\n",
      " 'related protein (SFRP) family expressed in fibroblast-like clusters were '\n",
      " 'similar across tissues - endometrium, adenomyosis and myometrium 68 . Figure '\n",
      " '10. Fibroblasts and epithelial cell subtypes differ in endometrium and '\n",
      " 'adenomyosis. Open in a new tab Panel A. UMAP of cell types and fibroblast '\n",
      " 'trajectories in endometrium. B. UMAP of cell types and fibroblast '\n",
      " 'trajectories in adenomyosis. Panel C. Progesterone resistance of epithelial '\n",
      " 'subtypes in adenomyosis. Panel D. WNT signaling pathway network. From Yildiz '\n",
      " 'et al, 2023 68 , with permission. Regarding the epithelial cells, P 4 '\n",
      " '-resistance was found in ciliated and unciliated cells in adenomyosis tissue '\n",
      " 'versus matched eutopic endometrium ( Figure 10 C ) – this is a key finding '\n",
      " 'regarding treatment efficiencies. Further data analysis revealed that the '\n",
      " 'canonical WNT/SFRP pathway signaling was uniquely activated in adenomyosis '\n",
      " 'unciliated and ciliated epithelium and paracrine signaling fibroblast-like '\n",
      " 'mesenchymal: epithelial interaction networks ( Figure 10D ), opening new '\n",
      " 'avenues for drug discovery for this disorder. The epithelial→endothelial '\n",
      " 'transition reported by Liu et al 67 was not detected in this study and may '\n",
      " 'be due to different adenomyosis lesion types, cycle phase, patient age, and '\n",
      " 'other variables. Whether the findings of these two studies involving '\n",
      " 'patients with diffuse adenomyosis also apply to other forms of adenomyosis ( '\n",
      " 'Figure 1 ) is unknown and is an opportunity to investigate cell features '\n",
      " 'wherein novel therapeutics could be derived. Moreover, recent data suggest '\n",
      " 'that the endometrial and vaginal microbiomes differ in patients with versus '\n",
      " 'without adenomyosis 69 , 70 , raising the possibility of adjunctive '\n",
      " 'antibiotic therapies for symptomatic patients with disease. Additionally, '\n",
      " 'further study of the local and systemic immune system cells and their '\n",
      " 'subtypes as well as in affected uterine tissues at single cell resolution '\n",
      " 'are warranted.')\n"
     ]
    }
   ],
   "source": [
    "pprint.pprint(prueba['Results'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7733fdb9",
   "metadata": {},
   "source": [
    "## Others"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a10b4d6d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Devesa-Peiro2020.pdf\n",
      "Henarejos-Castillo2020.pdf\n",
      "Sebastian-Leon2021.pdf\n",
      "Marti-Garcia2024(1).pdf\n",
      "Henarejos-Castillo2021.pdf\n",
      "Henarejos-Castillo2022.pdf\n",
      "Diaz-Gimeno2024.pdf\n",
      "Devesa-Peiro2021.pdf\n",
      "Henarejos-Castillo2024.pdf\n",
      "parraga-leo_2023.pdf\n",
      "Marti-Garcia2024.pdf\n",
      "Diaz-Gimeno2022.pdf\n",
      "Devesa-Peiro2022.pdf\n",
      "Sebastian-Leon2018.pdf\n",
      "Diaz-Gimeno2017.pdf\n"
     ]
    }
   ],
   "source": [
    "# Load several documents\n",
    "\n",
    "storage = []\n",
    "\n",
    "for root, dirs, files in os.walk(\"Data\",): # for server all the path\n",
    "\n",
    "    for file in files:\n",
    "        print(file)\n",
    "\n",
    "        loader = PyPDFLoader(os.path.join(root,file))\n",
    "        pages = loader.load_and_split()\n",
    "\n",
    "        # split it into chunks\n",
    "        text_splitter = CharacterTextSplitter(chunk_size=1000, chunk_overlap=50)\n",
    "        docs = text_splitter.split_documents(pages)\n",
    "        storage.extend(docs) # This  is a list of lists\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "4e00cd7b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load several documents\n",
    "\n",
    "storage = []\n",
    "\n",
    "for root, dirs, files in os.walk(\"Data\",): # for server all the path\n",
    "\n",
    "    for file in files:\n",
    "\n",
    "        loader = PyPDFLoader(os.path.join(root,file),mode='page')\n",
    "        pages = loader.load() # List of pages\n",
    "        \n",
    "        text_splitter = RecursiveCharacterTextSplitter(chunk_size=500, chunk_overlap=100)\n",
    "        docs = text_splitter.split_documents(pages)\n",
    "\n",
    "        storage.extend(docs) # Add several items at the same time\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "04f51224",
   "metadata": {},
   "outputs": [],
   "source": [
    "# load the document and split it into pages\n",
    "loader = PyPDFLoader(\"Publications/Parraga-Leo2023.pdf\")\n",
    "pages = loader.load_and_split()\n",
    "\n",
    "# split it into chunks\n",
    "text_splitter = RecursiveCharacterTextSplitter(chunk_size=3000, chunk_overlap=100)\n",
    "storage = text_splitter.split_documents(pages)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "fcdd0572",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('out the menstrual cycle (FDR < 0.05), dually validated in-silico and through '\n",
      " 'endometrial biopsies, corroborated their \\n'\n",
      " 'potential regulatory roles in the endometrium.\\n'\n",
      " '*Correspondence:\\n'\n",
      " 'Patricia Diaz‑Gimeno\\n'\n",
      " 'patricia.diaz@ivirma.com; patricia_diaz@iislafe.es\\n'\n",
      " 'Full list of author information is available at the end of the article')\n"
     ]
    }
   ],
   "source": [
    "pprint.pprint(storage[2].page_content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "7725a5a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "model = ChatOllama(\n",
    "    model=\"gemma3:4b\",\n",
    "    temperature=0.1,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "61f23c4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "class paper(BaseModel):\n",
    "    text: str = Field(description=\"Main text that explains the reuslts, introductions, methods or conclusions\")\n",
    "    doi: str = Field(description=\"link with the url of scientifc article known as doi.\")\n",
    "    citation: str = Field(description=\"add the citation of the manuscript is presented in the text.\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "88db094b",
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = JsonOutputParser(pydantic_object=paper)\n",
    "\n",
    "prompt = PromptTemplate(\n",
    "    template=\"\"\"\n",
    "    You are an expert analyzing scientifc research papers. Read the text carefully to provide the request information.\n",
    "    Remove the potential information that is not related with the scientific article.\n",
    "\n",
    "    If it appears a link with the doi of the artcile, take it. Otherwise write down UNKOWN.\n",
    "\n",
    "    Help me to extract the information and create an json file with it. If you don't know the\n",
    "    answer, say UNKNOWN.\n",
    "    \n",
    "    Format instructions: \\n{format_instructions}\\n\n",
    "    \n",
    "    Case:\\n{query}\\n\"\"\",\n",
    "\n",
    "    input_variables=[\"query\"],\n",
    "    \n",
    "    partial_variables={\"format_instructions\": parser.get_format_instructions()},\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "8ccc261f",
   "metadata": {},
   "outputs": [],
   "source": [
    "chain = prompt | model | parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "94b83fdd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Parraga‑Leo et al. \n",
      "Reproductive Biology and Endocrinology           (2023) 21:84  \n",
      "https://doi.org/10.1186/s12958‑023‑01131‑4\n",
      "RESEARCH Open Access\n",
      "© The Author(s) 2023. Open Access This article is licensed under a Creative Commons Attribution 4.0 International License, which \n",
      "permits use, sharing, adaptation, distribution and reproduction in any medium or format, as long as you give appropriate credit to the \n",
      "original author(s) and the source, provide a link to the Creative Commons licence, and indicate if changes were made. The images or \n",
      "other third party material in this article are included in the article’s Creative Commons licence, unless indicated otherwise in a credit line \n",
      "to the material. If material is not included in the article’s Creative Commons licence and your intended use is not permitted by statutory \n",
      "regulation or exceeds the permitted use, you will need to obtain permission directly from the copyright holder. To view a copy of this \n",
      "licence, visit http://creativecommons.org/licenses/by/4.0/. The Creative Commons Public Domain Dedication waiver (http://creativecom‑\n",
      "mons.org/publicdomain/zero/1.0/) applies to the data made available in this article, unless otherwise stated in a credit line to the data.\n",
      "Reproductive Biology\n",
      "and Endocrinology\n",
      "Deciphering a shared transcriptomic \n",
      "regulation and the relative contribution of each \n",
      "regulator type through endometrial gene \n",
      "expression signatures\n",
      "Antonio Parraga‑Leo1,2, Patricia Sebastian‑Leon1, Almudena Devesa‑Peiro1,2, Diana Marti‑Garcia1,2, \n",
      "Nuria Pellicer1,3, Jose Remohi2,3, Francisco Dominguez1 and Patricia Diaz‑Gimeno1*   \n",
      "Abstract \n",
      "Backgorund While various endometrial biomarkers have been characterized at the transcriptomic and functional \n",
      "level, there is generally a poor overlap among studies, making it unclear to what extent their upstream regulators \n",
      "(e.g., ovarian hormones, transcription factors (TFs) and microRNAs (miRNAs)) realistically contribute to menstrual cycle \n",
      "progression and function. Unmasking the intricacies of the molecular interactions in the endometrium from a novel \n",
      "systemic point of view will help gain a more accurate perspective of endometrial regulation and a better explanation \n",
      "the molecular etiology of endometrial‑factor infertility.\n",
      "Methods An in-silico analysis was carried out to identify which regulators consistently target the gene biomark‑\n",
      "ers proposed in studies related to endometrial progression and implantation failure (19 gene lists/signatures were \n",
      "included). The roles of these regulators, and of genes related to progesterone and estrogens, were then analysed \n",
      "in transcriptomic datasets compiled from samples collected throughout the menstrual cycle (n = 129), and the expres‑\n",
      "sion of selected TFs were prospectively validated in an independent cohort of healthy participants (n = 19).\n",
      "Results A total of 3,608 distinct genes from the 19 gene lists were associated with endometrial progression\n",
      "and implantation failure. The lists’ regulation was significantly favoured by TFs (89% (17/19) of gene lists) and pro‑\n",
      "gesterone (47% (8 /19) of gene lists), rather than miRNAs (5% (1/19) of gene lists) or estrogen (0% (0/19) of gene \n",
      "lists), respectively (FDR < 0.05). Exceptionally, two gene lists that were previously associated with implantation failure \n",
      "and unexplained infertility were less hormone‑dependent, but primarily regulated by estrogen. Although endometrial \n",
      "progression genes were mainly targeted by hormones rather than non‑hormonal contributors (odds ratio = 91.94, \n",
      "FDR < 0.05), we identified 311 TFs and 595 miRNAs not previously associated with ovarian hormones. We highlight \n",
      "CTCF, GATA6, hsa‑miR‑15a‑5p, hsa‑miR‑218‑5p, hsa‑miR‑107, hsa‑miR‑103a‑3p, and hsa‑miR‑128‑3p, as overlapping \n",
      "novel master regulators of endometrial function. The gene expression changes of selected regulators through‑\n",
      "out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their\n",
      "out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their \n",
      "potential regulatory roles in the endometrium.\n",
      "*Correspondence:\n",
      "Patricia Diaz‑Gimeno\n",
      "patricia.diaz@ivirma.com; patricia_diaz@iislafe.es\n",
      "Full list of author information is available at the end of the article\n",
      "Page 2 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "Conclusions This study revealed novel hormonal and non‑hormonal regulators and their relative contributions \n",
      "to endometrial progression and pathology, providing new leads for the potential causes of endometrial‑factor \n",
      "infertility.\n",
      "Keywords CTCF, GATA6, Progesterone, Estrogen, miRNAs, TFs, Endometrial receptivity, Recurrent implantation failure, \n",
      "Infertility, Menstrual cycle regulation\n",
      "Background\n",
      "The endometrium is the innermost layer of the uterus, \n",
      "which undergoes dynamic histological, physiological \n",
      "and molecular changes that allow it to synchronize with \n",
      "embryo development, facilitate embryo implantation, \n",
      "and ultimately, establish a successful pregnancy [1].\n",
      "Implantation is a crucial and complex limiting step \n",
      "for conception [2] which occurs between days 19 to 24 \n",
      "of a normal menstrual cycle. Consequently, this period \n",
      "of time is called the window of implantation (WOI) [3], \n",
      "and is characterized by abrupt transcriptomic changes \n",
      "in the endometrial tissue [4]. Alterations in either WOI \n",
      "establishment and endometrial progression may lead to \n",
      "implantation failures, which, along with biochemical mis-\n",
      "carriage, account for more than 50% of pregnancy losses \n",
      "at pre-clinical stages [5, 6]. These alterations have broadly \n",
      "been classified as being related to a displaced WOI, \n",
      "caused by variable timing of endometrial progression, or \n",
      "a disrupted WOI, where impaired endometrial function \n",
      "prevents the establishment of an effective WOI [7].\n",
      "Initially, the WOI was postulated to mainly be regu -\n",
      "lated by the ovarian hormones (namely, estrogen and \n",
      "progesterone) acting via their respective nuclear recep -\n",
      "tors (i.e., ESR1/2 and PGR). However, evidence has shown \n",
      "that these receptors cooperate with other transcription \n",
      "factors (TFs) and co-regulators to mediate uterine physi -\n",
      "ology [8]. In fact, TFs (such as homeobox TFs, FOXA2, \n",
      "and KLF9) have recently been reported as key regulators \n",
      "for the establishment of endometrial receptivity [9, 10], \n",
      "and microRNAs (miRNAs) have been found to similarly \n",
      "intervene in this complex and multifactorial process as \n",
      "transcriptional regulators [11]. While the involvement of \n",
      "both types of regulators was widely reported in embryo \n",
      "implantation [12] and other infertility-related diseases, \n",
      "including recurrent implantation failure (RIF) [13], the \n",
      "complex interactions between the different types of \n",
      "regulators and their respective contributions to endo -\n",
      "metrial progression and function have not been investi -\n",
      "gated from a systemic point of view. This paradigm shift \n",
      "approaches genes and regulators in an integrative way, by \n",
      "considering how they interact with and coordinate each \n",
      "other to carry out cellular processes, rather than rein -\n",
      "forcing the premise they work independently [14]. By \n",
      "more accurately reflecting the biological milieu, systemic \n",
      "approaches take gene-based discoveries to the next level,\n",
      "helping to generate hypotheses with relevant clinical and \n",
      "molecular implications [15, 16].\n",
      "Previous studies of endometrial transcriptomics have \n",
      "been used to define the WOI, compare the endometrium \n",
      "of healthy women as their menstrual cycles progressed \n",
      "[4, 17–19]; address endometrial differences related to age \n",
      "[20] or the dysfunction of the WOI between patients with \n",
      "implantation failure and healthy controls [21]; and finally, \n",
      "correct for the menstrual cycle bias that can mask impor-\n",
      "tant biomarkers [22]. Notwithstanding, the candidate \n",
      "biomarkers of endometrial receptivity reported in each \n",
      "of the aforementioned studies overlap poorly [7]. Indeed, \n",
      "recent comparisons conducted by Sebastian-Leon and \n",
      "colleagues [7] found no congruities between 16 different \n",
      "reported gene lists of endometrial receptivity, and only \n",
      "fair to moderate functional agreements between some of \n",
      "the included signatures. Despite previous transcriptional \n",
      "regulation studies, there have been no holistic studies of\n",
      "fair to moderate functional agreements between some of \n",
      "the included signatures. Despite previous transcriptional \n",
      "regulation studies, there have been no holistic studies of \n",
      "this regulatory process, which robustly analyse the driv -\n",
      "ers of transcription across a higher number of studies \n",
      "(that each searched for endometrial receptivity biomark -\n",
      "ers) and map the relative contributions of each type of \n",
      "regulator to endometrial progression regulation (Ovarian \n",
      "hormones [i.e., estrogen, progesterone], TFs, miRNAs).\n",
      "Thus, the aim of this study was to utilize available \n",
      "genomic data regarding transcriptional regulators to \n",
      "identify overlapping mediators of endometrial transcrip -\n",
      "tional regulation (i.e., ovarian hormones, TFs, and miR -\n",
      "NAs) and determine the relative contribution of each \n",
      "type of regulator among the previously reported gene \n",
      "signatures. These findings could help unveil the master \n",
      "regulators and principal type of endometrial regulation, \n",
      "paving the way for further research aimed at improving \n",
      "woman´s reproductive health.\n",
      "Methods\n",
      "A detailed study design is depicted in Supplementary \n",
      "Fig. S1.\n",
      "Annotating gene lists associated with endometrial \n",
      "progression and function\n",
      "Endometrial progression and implantation failure gene \n",
      "lists [7] were retrieved and updated using a public data \n",
      "repository (i.e., Gene Expression Omnibus (GEO)).\n",
      "Page 3 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Keywords used for the GEO search included: endometrial \n",
      "receptivity, mid-secretory endometrium, RIF, recurrent \n",
      "implantation failure, endometrium, unexplained infer -\n",
      "tility, and implantation failure. The resulting datasets \n",
      "were filtered by their publication date (i.e., from January \n",
      "2018 to October 2020, to expand the systematic search of \n",
      "[7], number of samples (> 3 samples for each condition), \n",
      "and species (i.e., Homo sapiens), with no restrictions on \n",
      "publication language. The original signatures were then \n",
      "retrieved from their corresponding publications. The \n",
      "genes prioritized in the original publication were exclu -\n",
      "sively selected and their names were annotated with \n",
      "HUGO Gene Nomenclature Committee (HGNC) gene \n",
      "name using biomaRt R-package v.3.10 [23]. Each signa -\n",
      "ture was labelled with the name of first author of the cor -\n",
      "responding study. Our annotated gene lists were selected \n",
      "as representative signatures of endometrial progression \n",
      "and function for subsequent analyses (Supplementary \n",
      "Fig. S1A).\n",
      "Identifying hormonal and non‑hormonal gene regulators \n",
      "of endometrial progression and function\n",
      "To study the hormonal regulation of endometrial \n",
      "progression, the Kyoto Encyclopaedia of Genes and \n",
      "Genomes (KEGG) [24], and Gene Ontology (GO) [25] \n",
      "databases were consulted for ovarian hormone-related \n",
      "genes, using relevant keywords (i.e., progesterone, estro-\n",
      "gen, oestrogen and estradiol) and selecting those pathways \n",
      "and functions that contained them (Supplementary Fig. \n",
      "S1B). Genes associated with the obtained human KEGG \n",
      "pathways and GO functions were grouped according to \n",
      "whether they were unique to progesterone (P4 gene set) \n",
      "or estrogen (E2 gene set) or related to both hormones \n",
      "(P4 and E2 gene set). Gene targets of the nuclear proges -\n",
      "terone (PGR) and estrogen (ESR1, ESR2) receptors were \n",
      "added to their corresponding gene sets using DoRothEA \n",
      "(Discriminant Regulon Expression Analysis) database \n",
      "[26], considering only manually-curated or ChiP-Seq \n",
      "experimentally-validated gene-TF relationships. Finally, \n",
      "the P4 and E2 gene sets were independently mapped to \n",
      "each gene list.\n",
      "Meanwhile, to evaluate the non-hormonal regula -\n",
      "tors of endometrial progression and function, DoRo -\n",
      "thEA [26] and TarBase [27] databases were consulted to \n",
      "obtain TFs and miRNAs. DoRothEA was filtered as pre -\n",
      "viously described herein, while TarBase, which only con -\n",
      "tains miRNA-gene target relationships manually curated \n",
      "from publications or experimentally-validated in high-\n",
      "throughput datasets [27], was filtered by species (i.e., \n",
      "human). Finally, a functional over-representation analysis \n",
      "was carried out to identify which of the total annotated \n",
      "TFs or miRNAs were significantly associated with a par -\n",
      "ticular gene list (Supplementary Fig. S1B).\n",
      "To evaluate the relative contribution of each type of \n",
      "regulator (i.e., P4, E2, TFs, and miRNAs) on endometrial\n",
      "progression and function, three different comparisons \n",
      "were carried out applying Fisher’s exact tests. First, hor -\n",
      "monal regulators were evaluated using the relative pro -\n",
      "portion of P4- and E2-related genes included in each list, \n",
      "with respect to the total of number of genes founded in \n",
      "the P4 or E2 gene sets. Then, non-hormonal regulators \n",
      "were evaluated using the relative proportion of over-\n",
      "represented miRNAs and TFs with respect to the total of \n",
      "miRNAs and TFs with a at least one target in the gene \n",
      "list. Finally, the relative proportion of genes under hor -\n",
      "monal (i.e., P4 and/or E2-annotated genes) versus non-\n",
      "hormonal (i.e., miRNA and/or TF-annotated genes) \n",
      "regulation was evaluated considering the miRNAs/TFs \n",
      "found within the individual P4 or E2 gene sets as hormo -\n",
      "nal regulators.\n",
      "Prioritizing key regulators in endometrial progression \n",
      "and function\n",
      "We built regulatory networks to systemically analyse \n",
      "endometrial regulation. Nodes represented gene lists\n",
      "nal regulators.\n",
      "Prioritizing key regulators in endometrial progression \n",
      "and function\n",
      "We built regulatory networks to systemically analyse \n",
      "endometrial regulation. Nodes represented gene lists \n",
      "and regulators (miRNAs or TFs), while edges indicated \n",
      "significant enrichment among the lists and their corre -\n",
      "sponding regulators (false discovery rate (FDR) ≤ 0.05). \n",
      "To select the most influential regulators (i.e., those which \n",
      "targeted genes in most of our signatures), we studied the \n",
      "degree distribution of the networks (i.e., the number of \n",
      "gene lists regulated by each molecule) and prioritized the \n",
      "miRNAs and TFs which surpassed the relative maximum \n",
      "(i.e., 1.50 times the interquartile range (IQR)) number of \n",
      "relationships (gene list–regulators). All networks were \n",
      "built and analysed with Cytoscape version 3.7 [28] (Sup -\n",
      "plementary Fig. S1B).\n",
      "Dataset construction and processing for in‑silico validation\n",
      "To evaluate how the miRNAs’ expression change in the \n",
      "endometrium throughout the menstrual cycle, we used \n",
      "the raw data from GSE44558 GEO dataset [29], deriving \n",
      "from 20 endometrial samples collected throughout the \n",
      "menstrual cycle: four in early-proliferative (EPF), four \n",
      "in late-proliferative (LPF), four in early-secretory (ESE), \n",
      "four in mid-secretory (MSE) and four in late-secretory \n",
      "(LSE).\n",
      "To evaluate how the TFs’ expression change in the \n",
      "endometrium throughout the menstrual cycle, we \n",
      "analysed the integrated endometrial dataset previ -\n",
      "ously created by our group, and reported in [30], which \n",
      "compiled data from five prior publications (GSE98386, \n",
      "GSE29981, GSE4888, GSE119209 and GSE86491). \n",
      "This dataset included endometrial gene expression\n",
      "Page 4 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "data from 109 participants with normal endometrium, \n",
      "where biopsies were collected in proliferative (PF) \n",
      "(n = 29), ESE (n  = 29), MSE (n  = 43) and LSE (n  = 8) \n",
      "phases of the menstrual cycle (Table S1 ).\n",
      "Both TF and miRNA datasets were processed using \n",
      "the limma R-package [31]. Expression data were log \n",
      "transformed and quantile normalized, prior to explora -\n",
      "tory analyses which sought out possible outliers and \n",
      "batch effects. Relative gene expression ranges of low \n",
      "(1–10%), medium (11–50%), and high (51–100%) were \n",
      "established based on the expression of all the genes \n",
      "included in our dataset.\n",
      "Wet lab validation cohorts\n",
      "To corroborate the menstrual cycle-related expression \n",
      "changes observed with CTCF and GATA6 in our in-\n",
      "silico analysis, we conducted a prospective study as an \n",
      "external validation, using an independent sample set. \n",
      "We included a cohort of 19 healthy Spanish women \n",
      "(obtaining a total of 20 biopsies), between the ages of \n",
      "22–35 and with a body mass index of 22.80 ± 2.76  kg/\n",
      "m2. Endometrial samples were collected and staged in \n",
      "the menstrual cycle according to the follicle growth to \n",
      "control ovulation and LH levels in urine. Samples were \n",
      "grouped into PF (n  = 5), ESE (n  = 5), MSE (n  = 5) and \n",
      "LSE (n = 5) phases and categorized into two groups, PF/\n",
      "ESE and MSE/LSE.\n",
      "RNA extraction and RT‑qPCR\n",
      "Total RNA was extracted from the endometrial biopsy \n",
      "samples using the miRNeasy Mini Kit (Qiagen, Ger -\n",
      "many), and reverse transcribed into cDNA using the \n",
      "PrimeScript reagent kit (TAKARA, Japan). RT-qPCR \n",
      "reactions were carried out in duplicate, using fluores -\n",
      "cent Power-up SYBR Green (Thermo Fisher Scientific \n",
      "MA, USA) in a final volume of 10 µL. Primer sequences \n",
      "are shown in Table S2. Samples underwent 40 cycles of \n",
      "amplification, under standard conditions, using a Ste -\n",
      "pOnePlus™ System (Applied Biosystems, MA, USA). \n",
      "Relative mRNA expression was calculated using the  2−\n",
      "ΔΔCt method [32], and normalized to the expression of \n",
      "GAPDH housekeeping gene.\n",
      "Statistical analysis\n",
      "Fisher’s exact test was used to evaluate independent \n",
      "proportions. Mean expression changes across the cycle \n",
      "were studied using an analysis of variance (ANOVA) fol -\n",
      "lowed by a pairwise t-test or the Wilcoxon test when only \n",
      "two groups were available. P-values from multiple test \n",
      "comparisons were corrected by FDR and considered as \n",
      "significant when FDR ≤ 0.05. All statistical analyses were \n",
      "performed through R software (version 3.5) [33].\n",
      "Results\n",
      "Gene signatures associated to endometrial progression \n",
      "and function\n",
      "Of the 19 gene lists used for our analysis, eleven were \n",
      "obtained from published studies evaluating control \n",
      "patients throughout the menstrual cycle [4, 17–19, \n",
      "34–40], and the other eight were derived from studies \n",
      "comparing patients with RIF or unexplained infertility \n",
      "to controls [21,  41–47] (Table 1). Unifying all the afore -\n",
      "mentioned signatures, we compiled 3,608 genes related \n",
      "to endometrial progression and function.\n",
      "Hormonal regulation of endometrial progression is largely \n",
      "driven by progesterone\n",
      "We identified 7,540 and 698 genes related to estrogen \n",
      "and progesterone hormones, respectively. However, as \n",
      "determined by the relative contribution of each type of \n",
      "hormone within each gene list (Fig.  1A), 17/19 (89%) \n",
      "signatures favoured regulation by progesterone rather \n",
      "than estrogen. These differences were significant in \n",
      "47% of the signatures (FDR < 0.05). Exceptionally, the \n",
      "genes reported by the Altmae2010 and Koot signa -\n",
      "tures, that were related to unexplained infertility and \n",
      "RIF, had a preference for estrogen regulation (Fig.  1A). \n",
      "With respect to the proportion of progesterone-related \n",
      "genes, Altmae2010 showed 2.1 times more estrogen-\n",
      "related genes while Koot showed 1.7 times more. More -\n",
      "over, comparing the proportion of genes regulated by \n",
      "both hormones, between signatures revealed that Alt -\n",
      "related genes while Koot showed 1.7 times more. More -\n",
      "over, comparing the proportion of genes regulated by \n",
      "both hormones, between signatures revealed that Alt -\n",
      "mae2010 and Koot exhibited a lower hormone-depend -\n",
      "ent behaviour overall, with respect to the remaining \n",
      "signatures (Fig.  1B), highlighting the role of non-hor -\n",
      "monal regulators.\n",
      "Non‑hormonal regulation of endometrial progression \n",
      "and function\n",
      "We identified a total of 770 miRNAs and 555 TFs regu -\n",
      "lating at least one of the 3,608 genes reported in the \n",
      "19 gene lists. Further, we observed that 29% of the \n",
      "miRNAs and 44% of the TFs were associated with the \n",
      "ovarian hormones (i.e., progesterone and/or estrogen). \n",
      "Significantly more genes were targeted by hormo -\n",
      "nally-influenced regulators rather than non-hormonal \n",
      "regulators (P -value < 0.01, odds ratio =  91.94). Never -\n",
      "theless, we also identified 595 miRNAs (77.27%) and \n",
      "311 TFs (56.04%) that were unrelated to progesterone \n",
      "and/or estrogen, highlighting a substantial proportion \n",
      "of hormonally-independent regulators of the men -\n",
      "strual cycle.\n",
      "Page 5 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Table 1 Characteristics of the 19 gene lists included in this study. Gene lists were divided according to their evaluated condition. Each \n",
      "signature was labelled using the name of first author, followed by the publication year in the case of duplicates. Endometrial staging \n",
      "was based on different variables including urinary LH, histological analysis, ultrasound evaluation, or the number of days since the hCG \n",
      "trigger. Participants included healthy controls and women with infertility‑related conditions. Various platforms were used to evaluate \n",
      "gene expression. The thresholds applied for gene refinement and the final number of genes per signature are also presented\n",
      "Abbreviations: Δ delta  Ct, adj-P-value Adjusted P‑value, C Control group, d Day of menstrual cycle, EPF Early proliferative, ESE Early secretory, |FC| Absolute fold‑change, \n",
      "FC Fold change, HRT Hormone replacement therapy, IF Implantation failure, LH Luteinizing hormone, LH + X X days post LH surge, LPF Late proliferative, LSE Late‑\n",
      "secretory, M Miscarriage, MPF Mid‑proliferative, MSE Mid‑secretory, n number of samples, N/A Not applicable, OP Ongoing pregnancy, PF Proliferative, PFP Proportion \n",
      "of false positives, PR Pre‑receptive, R Receptive, RIF Recurrent implantation failure, RRA  Robust Rank Aggregation algorithm, SF Spontaneously fertile women, UI \n",
      "Women with unexplained infertility, vs versus\n",
      "Condition Gene signature Endometrial staging Participants Platform Threshold No. genes References\n",
      "Endometrial progres‑\n",
      "sion\n",
      "Altmäe2017 N/A N/A N/A RRA 57 [34]\n",
      "Borthwick Urinary LH; Histologi‑\n",
      "cally confirmed\n",
      "PF d9‑11 (n = 5) vs R \n",
      "LH + 6–8 (n = 5)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "N/A 116 [18]\n",
      "Carrascosa N/A N/A Quantitative RT‑PCR Based on literature 187 [35]\n",
      "Carson Urinary LH; Histologi‑\n",
      "cally confirmed\n",
      "ESE LH + 2–4 (n = \n",
      "3) vs MSE LH + 7–9 \n",
      "(n = 3)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "|FC|> 2 695 [17]\n",
      "Diaz‑Gimeno Urinary LH PR LH + 1–5 (n = \n",
      "15) vs R\n",
      "Agilent custom gene \n",
      "expression microarray\n",
      "|FC| > 2 234 [36]\n",
      "Kao Urinary LH; Histologi‑\n",
      "cally confirmed\n",
      "LPF d8‑10 (n = 4) vs \n",
      "MSE LH + 8‑10 (n = 7)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "|FC|> 2 and P‑value \n",
      "<0.05\n",
      "340 [37]\n",
      "Mirkin Urinary LH; Histologi‑\n",
      "cal dating\n",
      "ESE LH + 3 (n = 3) vs \n",
      "MSE LH + 8 (n = 5)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "|FC|>2 and adj‑P‑\n",
      "value < 0.05\n",
      "105 [19]\n",
      "Ponnampalam Histological dating EPF (n = 5) MPF (n = 7) \n",
      "LPF (n = 3) ESE (n = 7) \n",
      "MSE (n = 8) LS (n = 7) \n",
      "Mense (n = 6)\n",
      "Custom Adj‑P‑value <0.05 306 [38]\n",
      "Punyadeera Histological dating Mense (n = 2) vs LPF \n",
      "(n=2)\n",
      "Affymetrix HG‑U133A FC and P‑value 50 [39]\n",
      "Riesewijk Urinary LH; Histologi‑\n",
      "cal dating\n",
      "PR LH + 2 (n = 5) vs R \n",
      "LH + 7 (n = 5)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "|FC|> 3 in at least four \n",
      "out of five women.\n",
      "196 [40]\n",
      "Talbi Histological dating P (n = 6) vs ESE (n = 3) \n",
      "vs MSE (n = 8) vs LSE \n",
      "(n = 6)\n",
      "Affymetrix HG‑U133 \n",
      "Plus 2.0\n",
      "|FC|> 1.5 and adj‑P‑\n",
      "value < 0.05\n",
      "317 [4]\n",
      "|FC|> 1.5 and adj‑P‑\n",
      "value < 0.05\n",
      "317 [4]\n",
      "Implantation failure Altmäe2010 Urinary LH UI (n = 4) vs C (n = 5) Whole Human \n",
      "Genome Oligo \n",
      "Microarray (Agilent \n",
      "Technologies) \n",
      "|FC|> 2 and PFP < 0.05 184 [21]\n",
      "Bastu 6‑7 days after LH \n",
      "surge\n",
      "IF (n = 24) vs Fertile \n",
      "women (n = 24)\n",
      "Agilent‑039494\n",
      "SurePrint G3 Human\n",
      "GE v2 8x60K\n",
      "Log2(FC) ≥ 1 and adj‑\n",
      "P‑value ≤0.05 |FC|> 2 \n",
      "and PFP < 0.05\n",
      "524 [41]\n",
      "Bersinger N/A IF (n = 3) vs M (n = 3) \n",
      "vs OP (n = 3)\n",
      "Affymetrix Human \n",
      "Genome U133A 2.0 \n",
      "Chips\n",
      "Not specified 93 [42]\n",
      "Bhagwat N/A PR vs R Multiple platforms Based on literature 177 [43]\n",
      "Koot Urinary LH RIF (n = 43) vs C (n \n",
      "= 72)\n",
      "Human whole \n",
      "genome gene expres‑\n",
      "sion microarrays V2 \n",
      "(Agilent, Belgium)\n",
      "Machine learning \n",
      "predictor\n",
      "310 [44]\n",
      "Lédée Ultrasound evaluation IF (n = 30) vs Fertile \n",
      "women (n = 15)\n",
      "Affymetrix Human\n",
      "Genome U133 Plus \n",
      "2.0 Array\n",
      "Adj‑P‑value <0.01 \n",
      "and Δ > 1.35\n",
      "322 [45]\n",
      "Pathare 6‑7 days after hCG \n",
      "administration\n",
      "IF (n = 10) vs Healthy \n",
      "oocyte donors (n = 8)\n",
      "Illumina HumanHT‑12 \n",
      "V4.0 expression \n",
      "beadchip\n",
      "|FC|> 2 818 [46]\n",
      "Shi Urinary LH; Ultrasound \n",
      "evaluation\n",
      "Pathare 6‑7 days after hCG \n",
      "administration\n",
      "IF (n = 10) vs Healthy \n",
      "oocyte donors (n = 8)\n",
      "Illumina HumanHT‑12 \n",
      "V4.0 expression \n",
      "beadchip\n",
      "|FC|> 2 818 [46]\n",
      "Shi Urinary LH; Ultrasound \n",
      "evaluation\n",
      "RIF (n = 12) vs C (n \n",
      "= 10)\n",
      "Agilent‑052909\n",
      "CBC lncRNAmRNA V3\n",
      "|FC|≥ 2 and adj‑P‑\n",
      "value ≤ 0.05\n",
      "281 [47]\n",
      "Page 6 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "Of the respective miRNAs and TFs targeting genes \n",
      "related to endometrial progression and function, \n",
      "417 miRNAs and 467 TFs were significantly over-  \n",
      "represented in at least one gene list (FDR < 0.05, Tables \n",
      "S3-S4). Further, most signatures were significantly \n",
      "governed by TFs rather than miRNAs (4 > odds \n",
      "ratio > 164, FDR < 0.05, 17 lists [89%]), with the excep -\n",
      "tion of Koot’s signature that was governed by miRNAs \n",
      "and showed the opposite (odds ratio = 0.35, FDR < 0.05, \n",
      "Table S5).\n",
      "Fig. 1 Estrogen and progesterone‑mediated regulation of endometrial progression and function. A Eight out of nineteen gene lists (47%) \n",
      "had significant differences between the proportion of genes regulated by progesterone versus estrogen. (*FDR ≤ 0.05). B Descriptive analysis \n",
      "of hormonal regulation, normalized by the size of each gene list. Gene lists clustered according to higher (blue) or lower (orange) contribution \n",
      "of hormonal regulation\n",
      "Page 7 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Hsa‑miR‑15a‑5p, hsa‑miR‑128‑3p, hsa‑miR‑218‑5p, \n",
      "hsa‑miR‑27b‑3p, hsa‑miR‑107, hsa‑miR‑424‑5p, \n",
      "hsa‑miR‑195‑5p, hsa‑miR‑103a‑3p, has‑let‑7b‑5p \n",
      "and hsa‑miR‑22‑3p are key players in the regulation \n",
      "of endometrial progression and function\n",
      "To bridge the gap of endometrial-based evidence in \n",
      "DoRothEA and TarBase databases, we aimed to iden -\n",
      "tify the common miRNAs that mediate gene expres -\n",
      "sion related to endometrial progression and function. \n",
      "We evaluated the number of endometrial gene lists \n",
      "that each miRNA controlled (Fig.  2A). This strategy \n",
      "highlighted the most influential miRNAs that were \n",
      "commonly found across the endometrial gene lists. \n",
      "Notably, 85.80% of miRNAs regulated genes from one \n",
      "to three signatures, and only 59 miRNAs (14.20%) were \n",
      "associated with more than three signatures.  Ultimately, \n",
      "24 miRNAs were prioritized for regulating genes of at \n",
      "least six signatures.\n",
      "Next, we validated the gene expression trends for \n",
      "these 24 prioritized miRNAs in an independent endo -\n",
      "metrial gene expression dataset (GSE44730) (Fig.  2B). \n",
      "This dataset included the miRNA expression in endo -\n",
      "metrial biopsies from 20 healthy women in differ -\n",
      "ent phases of the menstrual cycle. We found that \n",
      "two of our prioritized miRNAs were not expressed \n",
      "in the endometrial tissue, and only 10 miRNAs (i.e., \n",
      "hsa-miR-15a-5p, hsa-miR-128-3p, hsa-miR-218-5p, \n",
      "hsa-miR-27b-3p, hsa-miR-107, hsa-miR-424-5p, hsa-\n",
      "miR-195-5p, hsa-miR-103a-3p, has-let-7b-5p and hsa-\n",
      "miR-22-3p) showed significant changes across the \n",
      "menstrual cycle (Fig.  2B). Of these, only the expression \n",
      "of hsa-miR-22-3p significantly increased from ESE \n",
      "to MSE stages (FDR = 0.02, Fig.  2B), highlighting the \n",
      "potential inhibitory role of this miRNA in endometrial \n",
      "function. The expression of the remaining miRNAs \n",
      "decreased during the MSE phase.\n",
      "CTCF and GATA6 as distinguished transcriptional \n",
      "regulators of the human menstrual cycle\n",
      "We repeated the regulation network analysis with the \n",
      "TFs, to identify the prominent TFs commonly regu -\n",
      "lating the genes related to endometrial progression \n",
      "and function in the reported signatures. We observed \n",
      "78.80% of the TFs regulated genes in up to 10 signa -\n",
      "tures, and only 99 TFs (21.20%) regulated genes in over \n",
      "10 (Fig.  3A). We highlighted CTCF (a CCCTC-binding \n",
      "factor, which functions as a transcriptomic repressor) \n",
      "as the TF with the most influential regulation of endo -\n",
      "metrial progression and function across 95% of studies \n",
      "(18 gene lists, excluding that of Punyadeera, Fig.  3A). \n",
      "Other distinguished TFs that regulated up to 15 gene \n",
      "lists included AR, CEBPA, CEBPB, CEBPD, CREB1, \n",
      "EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, \n",
      "GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, \n",
      "NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, \n",
      "TCF7L2, TFAP2C and TP53. Notably, GATA6 was \n",
      "selected due to novel association with endometrial pro -\n",
      "gression and function.\n",
      "gression and function.\n",
      "To infer the expression of CTCF and GATA6  in \n",
      "endometrial tissue throughout the menstrual cycle, \n",
      "we analysed their expression using a relevant endo -\n",
      "metrial dataset recently created by our group [30] \n",
      "(Table S1 ). We observed that, relative to the global \n",
      "expression of endometrial genes, CTCF was broadly \n",
      "expressed while GATA6  was moderately expressed. \n",
      "Indeed, CTCF was significantly down-regulated \n",
      "between the PF and ESE phases (FDR = 2.10E-03)  \n",
      "and ESE to the MSE phase (FDR = 2.10E-04), while \n",
      "GATA6 expression increased from the PF to the \n",
      "MSE phase (FDR = 1.70E-11) and from the ESE  \n",
      "to MSE phase (FDR = 9.80E-15). These findings \n",
      "supported the role of these TFs at the beginning \n",
      "of the menstrual cycle and during the MSE phase \n",
      "(Fig. 3 B).\n",
      "These in-silico  validations were then prospec -\n",
      "tively corroborated by RT-qPCR analysis in an inde -\n",
      "pendent sample set of 20 endometrial biopsies. We \n",
      "observed a similar expression profile of CTCF and \n",
      "GATA6 throughout the menstrual cycle, with signifi -\n",
      "pendent sample set of 20 endometrial biopsies. We \n",
      "observed a similar expression profile of CTCF and \n",
      "GATA6 throughout the menstrual cycle, with signifi -\n",
      "cant changes between the endometrial phases. Nota -\n",
      "bly, CTCF expression was decreased between the PF/\n",
      "ESE and MSE/LSE phases (FDR = 0.01) while GATA6 \n",
      "expression increased from the PF/ESE to MSE/LSE \n",
      "phases (FDR = 1.55E-04) (Fig. 3 C).\n",
      "Fig. 2 MicroRNAs universally regulating endometrial progression and function and their expression throughout the menstrual cycle. A Regulatory \n",
      "network of miRNAs forming relationships with numerous gene lists (2–12); miRNAs targeting specific gene lists were placed in the centre \n",
      "of the network. The prioritized miRNAs (outlined in red) included has‑miR‑16‑5p, has‑miR‑138‑5p, has‑miR‑21‑3p, has‑miR‑205‑5p, has‑miR‑27a‑5p, \n",
      "has‑miR‑15a‑5p, has‑miR‑155‑5p, has‑miR‑129–2‑3p, has‑miR‑147a, has‑miR‑146a‑5p, has‑miR‑107, has‑miR‑424‑5p, has‑miR‑203a‑3p, \n",
      "has‑miR‑195‑5p, has‑miR‑124‑3p, has‑miR‑128‑3p, has‑miR‑22‑3p, has‑miR‑27b‑3p, has‑let‑7b‑5p, has‑miR‑1343‑3p, has‑miR‑1‑3p, has‑miR‑103a‑3p, \n",
      "has‑miR‑23b‑3p, has‑miR‑218‑5p. B Expression profiles of the 10 prioritized miRNAs at different phases of the menstrual cycle. Significant changes \n",
      "between phases are denoted with asterisks (* FDR ≤ 0.05, ** FDR ≤ 0.01. Abbreviations: EPF, early proliferative phase; LPF, late proliferative phase; ESE, \n",
      "early‑secretory phase; MSE, mid‑secretory phase; LSE, late‑secretory phase\n",
      "(See figure on next page.)\n",
      "Page 8 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "Fig. 2 (See legend on previous page.)\n",
      "Page 9 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Fig. 3 Transcription factors universally regulating endometrial progression and function and their expression throughout the menstrual \n",
      "cycle. A Regulatory network of TFs forming relationships with numerous gene lists (2–18). TFs targeting specific gene lists were placed \n",
      "in the center of the network. CTCF stood out as the only TF that targeted genes belonging to 18/19 (95%) different signatures. B In-silico analysis \n",
      "of CTCF and GATA6 expression throughout the menstrual cycle. Significant changes between phases are marked with asterisks (**FDR ≤ 0.01, \n",
      "***FDR ≤ 0.001). C RT‑qPCR validation. Bar plot shows the relative mRNA expression of CTCF and GATA6 in endometrial biopsies collected \n",
      "throughout the menstrual cycle. Data is presented as a mean ± SD. Significant differences of CTCF (*FDR < 0.05) and GATA6 (***FDR < 0.001) \n",
      "expression were observed between proliferative/early‑secretory (PF/ESE) and mid‑secretory/late‑secretory phases (MSE/LSE). Abbreviations in (B \n",
      "and C): PF, Proliferative; ESE, early‑secretory; MSE, mid‑secretory; LSE, late‑secretory; SD, standard deviation\n",
      "Page 10 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "Discussion\n",
      "This data-driven approach exposed the common tran -\n",
      "scriptional regulators among 19 studies who proposed \n",
      "variable biomarkers of endometrial progression and \n",
      "function. In this study, we focused on understanding \n",
      "the relative contribution of both the hormonal and non-\n",
      "hormonal regulation, from an alternative holistic per -\n",
      "spective. We applied data-driven hypothesis research \n",
      "that, unlike the traditional scientific method, allowed us \n",
      "to generate new hypotheses based on all available bio -\n",
      "logically-relevant knowledge [16], observing the molec -\n",
      "ular relationships from a wider scale view. Besides, \n",
      "some traditional molecular procedures such as PCRs \n",
      "are also performed to corroborate the insight uncovered \n",
      "with this approach. We highlighted a larger influence \n",
      "of progesterone-related genes and TFs in endome -\n",
      "trial progression and observed a larger contribution \n",
      "of miRNAs and estrogen-related genes in endometrial \n",
      "pathology. Furthermore, we unveiled CTCF, GATA6, \n",
      "hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-\n",
      "miR-103a-3p and hsa-miR-128-3p as master regulators \n",
      "in the endometrium, and these findings were reinforced \n",
      "by the dynamic changes in their expression throughout \n",
      "the menstrual cycle (dually validated with in-silico  and \n",
      "prospective analyses).\n",
      "The menstrual cycle is perceived to be under tight \n",
      "regulation by the ovarian hormones, progesterone \n",
      "and estrogen [48]. Supporting the role of progester -\n",
      "one in the secretory endometrium [49], and reinforc -\n",
      "ing its indispensable role in endometrial regulation \n",
      "[50, 51], we found progesterone and its related genes \n",
      "had a larger influence in menstrual cycle progression \n",
      "than estrogen. As the RIF-related gene lists reported \n",
      "by Altmae2010 and Koot were generally controlled by \n",
      "estrogen, this further suggested that while progester -\n",
      "one could be crucial for a suitable endometrial progres -\n",
      "sion and mid-secretory phase acquisition, estrogenic \n",
      "irregularities may be more conducive to endometrial \n",
      "function and molecular pathology. Indeed, this hypoth -\n",
      "esis corroborated previous reports of most uterine dis -\n",
      "orders being estrogen-dependent [44, 52]. Supporting \n",
      "the findings from another recent study from our group, \n",
      "where the Altmae2010 and Koot signatures were classi -\n",
      "fied as highly-predictive signatures to identify endome -\n",
      "trial pathology in comparison with a prediction model \n",
      "that detected endometrial progression [7 ]. This study \n",
      "distinguished these signatures (related to implantation \n",
      "failure and unexplained infertility) based on the mag -\n",
      "nitude of their miRNA and estrogen-regulation, which \n",
      "contrasted with the prominent influence of the TFs over \n",
      "the remaining signatures more related to endometrial \n",
      "progression. All these facts reinforce the hypothesis \n",
      "that implantation failure is mainly caused by impaired \n",
      "miRNA expression [53].\n",
      "Although we observed a clear hormonal influence in\n",
      "numerous annotated regulators, this study transcended \n",
      "the simplistic paradigm of the bihormonal regulation of \n",
      "endometrial function by identifying a plethora of medi -\n",
      "ators, including 595 miRNAs and 311 TFs, that were \n",
      "previously unrelated to these ovarian hormones. These \n",
      "findings set the foundation for new discoveries, reveal -\n",
      "ing alternative pathways and new actors in the regula -\n",
      "tion of the human endometrium, and deepening our \n",
      "understanding of the complex regulatory mechanisms \n",
      "behind endometrial progression and unexplained \n",
      "infertility.\n",
      "MicroRNAs are well known to be involved in embryo \n",
      "implantation [12] and receptivity control [54, 55]. Using \n",
      "network analysis, we unveiled the 10 common miR -\n",
      "NAs regulating endometrial progression and function \n",
      "across published biomarker signatures (hsa-miR-22-3p, \n",
      "hsa-miR-107, hsa-miR-103a-3p, hsa-miR-128-3p, hsa-\n",
      "miR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, hsa-\n",
      "miR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p).\n",
      "hsa-miR-107, hsa-miR-103a-3p, hsa-miR-128-3p, hsa-\n",
      "miR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, hsa-\n",
      "miR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p). \n",
      "Their significant gene expression changes in the secre -\n",
      "tory endometrium support their potential role in the \n",
      "endometrial regulation through an inhibition during \n",
      "the WOI. Notably, hsa-miR-424-5p, hsa-miR-27b-3p \n",
      "and hsa-miR-195-5p have been previously associated \n",
      "with RIF [11, 54, 56], while has-let-7b-5p was associ -\n",
      "ated with human endometrial receptivity [57], and \n",
      "hsa-miR-22-3p was overexpressed in a decidualization \n",
      "model [58]. Moreover, hsa-miR-15a-5p was previously \n",
      "associated with endometriosis [59]. Notwithstanding, \n",
      "this study is the first to associate hsa-miR-15a-5p, hsa-\n",
      "miR-218-5p, hsa-miR-103a-3p, hsa-miR-107, and hsa-\n",
      "miR-128-3p to endometrial progression and functional \n",
      "regulation in humans.\n",
      "Due to their direct relationship with progesterone and \n",
      "estrogen [60], we expected to find TFs, such as ESR1  \n",
      "and PGR, involved in endometrial progression, how -\n",
      "ever, we additionally identified other TFs with an even \n",
      "larger influence in this process. For instance, STAT3  \n",
      "was associated with 17/19 signatures (FDR < 0.05), and \n",
      "is known to regulate uterine epithelial junctional reor -\n",
      "ganization and stromal proliferation, which are critical \n",
      "for implantation [61]. SP1, which similarly regulated \n",
      "17/19 signatures (FDR < 0.05), acts as a downstream \n",
      "paracrine target of progesterone to regulate estrogen \n",
      "inactivation, and could have a predominant role dur -\n",
      "ing the WOI [62, 63]. By focusing on the most univer -\n",
      "sal, overlapping TFs that were not previously related \n",
      "to implantation, we ultimately prioritized CTCF and \n",
      "GATA6 as novel key regulators of endometrial pro -\n",
      "gression, however, their specific molecular actions in\n",
      "Page 11 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "endometrial progression and function were beyond the \n",
      "scope of this project and merit further investigation.\n",
      "CTCF is a conserved zinc finger protein whose regu -\n",
      "latory functions are well characterized throughout the \n",
      "human body [64]. CTCF acts as a transcriptional repres -\n",
      "sor in RNA polymerase II (Pol II) pausing and imprint -\n",
      "ing and  X-chromosome inactivation [65], as well as an \n",
      "insulator, blocking the interaction between enhancers \n",
      "and the promoters of neighbouring genes [66]. Consider -\n",
      "ing these attributed functions and its variable expression \n",
      "throughout the menstrual cycle, we propose that CTCF \n",
      "exerts an inhibitory role in endometrial tissue during \n",
      "the PF phase of the menstrual cycle. With the significant \n",
      "downregulation of CTCF in the secretory endometrium, \n",
      "dually validated in-silico  and experimentally herein, its \n",
      "inhibited genes would be derepressed and become tran -\n",
      "scriptionally active during the WOI. This interpretation \n",
      "supports previous findings from our group demonstrat -\n",
      "ing that, during the WOI, a global transcriptional dere -\n",
      "pression may be required for implantation and early \n",
      "embryo development [63]. Transcriptional derepression \n",
      "has been associated with multiple human disease states \n",
      "and should be investigated further within the context \n",
      "of endometrial-factor infertility. Despite previous asso -\n",
      "ciations of CTCF with endometriosis [67], its implica -\n",
      "tion in endometrial progression and function has not \n",
      "been directly proposed until now. Nevertheless, the role \n",
      "of CTCF in endometrial receptivity has been revealed \n",
      "through its interaction with HOXA10, a gene mainly \n",
      "expressed in endometrium involved in functions such as \n",
      "endometrial proliferation and differentiation, the forma -\n",
      "tion of pinopodes or embryo implantation [68]. Indeed, \n",
      "this study proposed that an overexpression of CTCF can \n",
      "lead to a drop in HOXA10 expression, affecting endome -\n",
      "trial proliferation and endometrial function and corrobo-\n",
      "rating the fact that CTCF must decrease its expression as \n",
      "we can observe in our findings.\n",
      "GATA6, the second prioritized TF that overlapped \n",
      "across the gene lists, is a transcription factor belong -\n",
      "ing to the GATA family, a highly conserved family of \n",
      "six zinc finger proteins [69]. Besides its essential role in \n",
      "embryonic development [70], GATA6 plays a key role \n",
      "in endometriosis and regulates steroidogenic genes [71]. \n",
      "Together with evidence supporting both an activating \n",
      "and inhibitory role for GATA6 [72, 73], the significant \n",
      "changes we observed in its expression across the men -\n",
      "strual cycle reflect its presumed involvement in the endo-\n",
      "metrium and suggest an activating role in endometrial \n",
      "progression and endometrial receptivity acquisition. Like \n",
      "CTCF, this is the first time that GATA6 is proposed as a \n",
      "key TF in endometrial progression and function.\n",
      "Although the exact molecular mechanisms underlying\n",
      "endometrial regulation remain elusive, emerging exper -\n",
      "imental technologies such as High-throughput Chro -\n",
      "mosome Conformation Capture (Hi-C) or chromatin \n",
      "interaction analysis with paired-end tags (ChIA-PET) \n",
      "that respectively study general chromatin interactions \n",
      "or chromatin interactions involving a specific protein \n",
      "[74, 75] may improve our understanding of genetic \n",
      "regulation in the future. Nevertheless, as collaborative \n",
      "efforts unveil new regulators and gene targets in humans \n",
      "(such as the ENCODE project [76]), the complexities of \n",
      "transcriptional regulation can be exposed by in-silico  \n",
      "analyses like those described herein, and we can gain a \n",
      "better understanding of the processes involved in men -\n",
      "strual cycle progression and endometrial competence.\n",
      "Furthermore, it should be noted that results of gene \n",
      "expression studies are influenced by different variables, \n",
      "such as the experimental platforms employed, study \n",
      "designs, or patients’ characteristics of each independent\n",
      "expression studies are influenced by different variables, \n",
      "such as the experimental platforms employed, study \n",
      "designs, or patients’ characteristics of each independent \n",
      "study, adding variability [30, 77, 78]. Despite this fact, \n",
      "we have found that our holistic approach overcomes this \n",
      "undesired variability effect by finding common upstream \n",
      "master regulators across studies and predicting new \n",
      "relationships between hormones, TFs and miRNAs. \n",
      "This approach also provides the basis for future single-\n",
      "molecule studies that aim to elucidate new regulatory \n",
      "pathways in endometrial progression and function. Our \n",
      "results lay the foundation for further molecular studies \n",
      "that can validate the function(s) of the prevailing endo -\n",
      "metrial regulators we prioritized.\n",
      "Conclusion\n",
      "The study shows for the first time, the relative contri -\n",
      "bution of estrogens, progesterone, TFs and miRNAs in \n",
      "endometrial function and progression. Endometrial pro -\n",
      "gression is mainly influenced by progesterone-related \n",
      "genes and TFs, whereas miRNAs and estrogen-related \n",
      "genes play a larger role in endometrial pathology. \n",
      "Moreover, we highlight novel common transcriptional \n",
      "regulators such as CTCF, GATA6, hsa-miR-15a-5p, hsa-\n",
      "miR-218-5p, hsa-miR-107, hsa-miR-103a-3p, and hsa-\n",
      "miR-128-3p across 19 studies that propose biomarkers in \n",
      "endometrial regulation. These results reveal the molecu -\n",
      "lar mechanism underlying endometrial regulation and lay \n",
      "the foundation for the development of targeted therapies \n",
      "for patients with endometrial-factor infertility.\n",
      "Abbreviations\n",
      "ANOVA  Analysis of variance\n",
      "ChIA‑PET  Chromatin interaction analysis with paired‑end tags\n",
      "EPF  Early proliferative\n",
      "ESE  Early secretory\n",
      "FDR  False Discovery Rate\n",
      "GEO  Gene Expression Omnibus\n",
      "Page 12 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "GO  Gene Ontology\n",
      "HGNC  HUGO Gene Nomenclature Committee\n",
      "Hi‑C  High‑throughput Chromosome Conformation Capture\n",
      "IQR  Interquartile range\n",
      "KEGG  Kyoto Encyclopaedia of Genes and Genomes\n",
      "miRNAs  MicroRNAs\n",
      "MSE  Mid Secretory\n",
      "LPF  Late proliferative\n",
      "LSE  Late secretory\n",
      "RIF  Recurrent Implantation Failure\n",
      "Pol II  RNA polymerase II\n",
      "TFs  Transcription Factors\n",
      "WOI  Window of implantation\n",
      "Supplementary Information\n",
      "The online version contains supplementary material available at https:// doi. \n",
      "org/ 10. 1186/ s12958‑ 023‑ 01131‑4.\n",
      "Additional file 1: Supplementary Figure S1. Study design and \n",
      "workflow. Steps performed throughout the study. (A) We reviewed 19 \n",
      "publications related to endometrial progression and function, extracted \n",
      "the gene signatures from the original publication, and created corre‑\n",
      "sponding gene lists that we could annotate. (B) We performed different \n",
      "functional analyses based on the type of regulator (i.e., ovarian hormones, \n",
      "transcription factors and miRNAs). We used online databases to obtain the \n",
      "information of the regulatory process, at the hormonal (Gene Ontology, \n",
      "Kyoto Encyclopedia of Genes and Genomes, Dorothea), transcription \n",
      "factor (Dorothea) and miRNA (Tarbase) level and analysed the results \n",
      "accordingly. Finally, we employed an integrative analysis and prioritized \n",
      "the most relevant regulators for subsequent validation. TF, Transcription \n",
      "factors; miRNA, microRNA.\n",
      "Additional file 2: Supplementary Table S1. Characteristics of endo‑\n",
      "metrial transcriptomic datasets used to build the integrated dataset. ID, \n",
      "identifier of Gene Expression Omnibus database; Source, first author and \n",
      "year of the publication if it is available; No. samples; number of samples \n",
      "included; Age, range of patient age; Participants, reason for endome‑\n",
      "trial sample collection; Cycle Type, type of menstrual cycle of subjects \n",
      "included; Platform, platform used to measure gene expression; No. genes, \n",
      "number of genes measured in each dataset, Endometrial phase, number \n",
      "of samples in each phase of menstrual cycle. PF, proliferative; ESE, early \n",
      "secretory; MSE, mid secretory, LSE, late secretory. NA, Not Available.\n",
      "Additional file 3: Supplementary Table S2. Primer sequences for CTCF \n",
      "and GATA6. Forward and reverse primers used for RT‑qPCR validation of \n",
      "CTCF and GATA6 transcription factors.\n",
      "Additional file 4: Supplementary Table S3. Results of miRNA enrich‑\n",
      "ment. Excel file with the over‑representation analysis results of the micro‑\n",
      "RNAs (miRNAs). Each sheet contains the results of a specific list of genes. \n",
      "In each sheet, the following information is included: ID, gene symbol ID of \n",
      "the miRNA; GeneRatio, the proportion of miRNA targets in the gene list; \n",
      "BgRatio, the proportion of miRNA targets in the whole genome; OddsRa‑\n",
      "tio; the odd ratios for each miRNA, pvalue; the P‑value obtained for each \n",
      "miRNA; Padjust, P‑value adjusted by FDR for each miRNA; geneID, gene\n",
      "symbol ID of the miRNA targets.\n",
      "Additional file 5: Supplementary Table S4. Results of TFs enrichment. \n",
      "Excel file with the over‑representation analysis results of the transcription \n",
      "factors (TFs). Each sheet contains the results of a specific list of genes. In \n",
      "each sheet, the following information is included: ID, gene symbol of the \n",
      "TF; GeneRatio, the proportion of TF targets in the gene list; BgRatio, the \n",
      "proportion of TF targets in the whole genome; OddsRatio; the odd ratios \n",
      "for each TF, pvalue; the P‑value obtained for each TF; padjust, P‑value \n",
      "adjusted by FDR for each TF; geneID, the gene symbol ID of the TF targets.\n",
      "Additional file 6: Supplementary Table S5. Relative contribution of \n",
      "each regulator type (TFs or miRNAs) in each gene list. A Fisher’s test was \n",
      "performed to calculate the proportional differences between transcrip‑\n",
      "tion factors (TFs) and microRNAs (miRNAs). Resulting P‑values were \n",
      "adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene\n",
      "tion factors (TFs) and microRNAs (miRNAs). Resulting P‑values were \n",
      "adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene \n",
      "lists were mainly regulated by TFs, whereas odds ratios < 1 indicate a \n",
      "higher regulation by miRNAs. Odds Ratio with a value of “Inf” indicate \n",
      "that miRNAs were not found for this gene list. Percentages indicate the \n",
      "proportion between the significant regulators in each gene signature and \n",
      "regulators with at least one target gene in the corresponding signature. \n",
      "(***FDR ≤0.001).\n",
      "Acknowledgements\n",
      "The authors would like to thank Yassmin Medina‑Laver and Alicia Quiñonero \n",
      "for their advice and technical support during the experimental validation. \n",
      "Moreover, the authors thank Rosalba Lopez for her constructive criticism and \n",
      "editorial services in preparing the manuscript for publication.\n",
      "Author’s contribution\n",
      "P . D.‑G. conceived the original idea of this work and designed the study with the \n",
      "help of P .S.‑L. P .D.‑G. with the help of J.R. promoted the study and P .D.‑G. supervised \n",
      "it. All the in‑silico and experimental procedures were done by A.P .‑L. and supervised \n",
      "by P .D.‑G. In‑silico procedures were also supervised by P .S.‑L. Gene lists were col‑\n",
      "lected by A.P .‑L. with help from P .S.‑L. and A.D.‑P . miRNA dataset was collected and \n",
      "pre‑processed by A.P .‑L. with the help from P .S.‑L. and A.D.‑P . The collection of genes \n",
      "related to ovarian hormones was carried out by A.P .‑L. with the supervision of P .D.‑G. \n",
      "Statistical analyses were carried out by A.P .‑L. with the help of P .S.‑L. and supervised \n",
      "by P .D.‑G. miRNAs and TFs analysis was done by A.P .‑L. supervised by P .S.‑L. Networks \n",
      "and figures were performed by A.P .‑L., with the help of A.D.‑P . being supervised by \n",
      "P .D.‑G. Samples were collected by J.R. and N.P . Experimental validation was carried \n",
      "out by A.P .‑L. and D.M.‑G. and co‑supervised by F.D. and P .D‑G. Interpretation regard‑\n",
      "ing the results and their effect in the receptivity was carried out by A.P .‑L. and P .D.‑G. \n",
      "with the help of A.D.‑P . The manuscript was written by A.P .‑L., P .D.‑G., and A.D‑P , and \n",
      "was revised by all co‑authors (P .S.‑L., J.R., D.M.‑G., N.P . and F.D.).\n",
      "Funding\n",
      "This study was supported by Fundación IVI (1706‑FIVI‑048‑PD); the Instituto \n",
      "de Salud Carlos III (ISCIII) and co‑funded by the European Regional Develop‑\n",
      "ment Fund “A way to make Europe” (PI19/00537 [P .D.‑G]). Patricia Diaz‑Gimeno is \n",
      "supported by Instituto de Salud Carlos III (ISCIII) through Miguel Servet program \n",
      "(CP20/00118 [P .D.‑G]) and co‑funded by the European Union. Spanish Ministry of \n",
      "Science, Innovation and Universities (FPU18/01777 [A.P .‑L.], FPU15/01398 [A.D.‑P .], \n",
      "FPU19/03247 [D.M.‑G). Patricia Sebastian‑Leon is funded by Instituto de Salud \n",
      "Carlos III (ISCIII) through Sara Borrell program (CD21/00132[P .S.‑L]) and co‑funded \n",
      "by the European Union. Francisco Dominguez was supported by Instituto de \n",
      "Salud Carlos III(ISCIII) through Miguel Servet program CPII18/00002 and co‑funded\n",
      "by the European Union (Fonda Social Europeo), “El FSE invierte en tu futuro” .\n",
      "Availability of data and materials\n",
      "The data used to analyze the TF and miRNA expression throughout the men‑\n",
      "strual cycle and in infertility are available in Gene Expression Omnibus, with \n",
      "IDs GSE98386, GSE29981, GSE4888, GSE119209, GSE86491, and GSE44558.\n",
      "Declarations\n",
      "Ethics approval and consent to participate\n",
      "This research evaluated endometrial gene expression data from women throughout \n",
      "the menstrual cycle. Raw endometrial gene expression data (from previous studies) \n",
      "was downloaded from the publicly‑available functional genomic dataset repository \n",
      "Gene Expression Omnibus (GEO) [79] for curation/processing. In accordance with \n",
      "GEO policies, all patient data was anonymized and encrypted, and therefore an addi‑\n",
      "tional institutional review board approval was not required for our in-silico validation \n",
      "of regulator gene expression changes in endometrial tissue. On the other hand, for\n",
      "tional institutional review board approval was not required for our in-silico validation \n",
      "of regulator gene expression changes in endometrial tissue. On the other hand, for \n",
      "the experimental validation of gene expression changes in endometrial biopsies, \n",
      "all individuals gave written informed consent for endometrial gene expression \n",
      "evaluation prior to participating (Ethics Committee of the Instituto Valenciano de \n",
      "Infertilidad [IVI Valencia], Valencia, Spain [1810‑FIVI‑066‑PD/1706‑FIVI‑048‑PD]).\n",
      "Consent for publication\n",
      "Not applicable.\n",
      "Competing interests\n",
      "The authors declare no competing interests.\n",
      "Page 13 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Author details\n",
      "1 IVIRMA Global Research Alliance, IVI Foundation, Instituto de Investigación \n",
      "Sanitaria La Fe (IIS La Fe), Av. Fernando Abril Martorell 106, Torre A, Planta \n",
      "1ª, 46026 Valencia, Valencia, Spain. 2 Department of Pediatrics, Obstetrics \n",
      "and Gynaecology, Universidad de Valencia, Av. Blasco Ibáñez 15, 46010 Valen‑\n",
      "cia, Valencia, Spain. 3 IVIRMA Global Research Alliance, IVIRMA Valencia, Plaza de \n",
      "La Policia Local 3, 46015 Valencia, Spain. \n",
      "Received: 21 June 2023   Accepted: 22 August 2023\n",
      "References\n",
      " 1. Strowitzki T, Germeyer A, Popovici R, von Wolff M. The human endo‑\n",
      "metrium as a fertility‑determining factor. Hum Reprod Update. \n",
      "2006;12:617–30.\n",
      " 2. Cakmak H, Taylor HS. Implantation failure: Molecular mechanisms and \n",
      "clinical treatment. Hum Reprod Update. 2011;17:242–53.\n",
      " 3. Lessey BA, Young SL. What exactly is endometrial receptivity? Fertil Steril. \n",
      "2019;111:611–7.\n",
      " 4. Talbi S, Hamilton AE, Vo KC, Tulac S, Overgaard MT, Dosiou C, et al. \n",
      "Molecular phenotyping of human endometrium distinguishes menstrual \n",
      "cycle phases and underlying biological processes in normo‑ovulatory \n",
      "women. Endocrinology. 2006;147:1097–121.\n",
      " 5. Somigliana E, Vigano P , Busnelli A, Paffoni A, Vegetti W, Vercellini P . \n",
      "Repeated implantation failure at the crossroad between statistics, clinics \n",
      "and over‑diagnosis. Reprod Biomed Online. 2018;36:32–8.\n",
      " 6. Craciunas L, Gallos I, Chu J, Bourne T, Quenby S, Brosens JJ, et al. Con‑\n",
      "ventional and modern markers of endometrial receptivity: a systematic \n",
      "review and meta‑analysis. Hum Reprod Update. 2019;25:202–23.\n",
      " 7. Sebastian‑Leon P , Garrido N, Remohí J, Pellicer A, Diaz‑Gimeno P . Asyn‑\n",
      "chronous and pathological windows of implantation: Two causes of \n",
      "recurrent implantation failure. Hum Reprod. 2018;33:626–35.\n",
      " 8. Vasquez YM, DeMayo FJ. Role of nuclear receptors in blastocyst implanta‑\n",
      "tion. Semin Cell Dev Biol. 2013;24:724–35.\n",
      " 9. Wang H, Zhang S, Lin H, Kong S, Wang S, Wang H, et al. Physiological \n",
      "and molecular determinants of embryo implantation. Mol Aspects Med. \n",
      "2013;34:939–80.\n",
      " 10. Kelleher AM, Behura SK, Burns GW, Young SL, Demayo FJ, Spencer TE. Inte‑\n",
      "grative analysis of the forkhead box A2 (FOXA2) cistrome for the human \n",
      "endometrium. FASEB J. 2019;33:8543–54.\n",
      " 11. Revel A, Achache H, Stevens J, Smith Y, Reich R. MicroRNAs are \n",
      "associated with human embryo implantation defects. Hum Reprod. \n",
      "2011;26:2830–40.\n",
      " 12. Galliano D, Pellicer A. MicroRNA and implantation. Fertil Steril. \n",
      "2014;101:1531–44.\n",
      " 13. Xu H, Zhou M, Cao Y, Zhang D, Han M, Gao X, et al. Genome‑wide analysis \n",
      "of long noncoding RNAs, microRNAs, and mRNAs forming a compet‑\n",
      "ing endogenous RNA network in repeated implantation failure. Gene. \n",
      "2019;720: 144056.\n",
      " 14. Copley RR. The animal in the genome: comparative genomics and evolu‑\n",
      "tion. Philos Trans R Soc Lond B: Biol Sci. 2008;363:1453–61.\n",
      "15. Jacob F. The statue within: an autobiography. CSHL Press; 1995.\n",
      " 16. Yanai I, Lercher M. Night science. Genome Biol. 2019;20:179.\n",
      " 17. Carson DD, Lagow E, Thathiah A, Al‑Shami R, Farach‑Carson MC, Vernon \n",
      "M, et al. Changes in gene expression during the early to mid‑luteal \n",
      "(receptive phase) transition in human endometrium detected by high‑\n",
      "density microarray screening. Mol Hum Reprod. 2002;8:871–9.\n",
      " 18. Borthwick JM, Charnock‑Jones DS, Tom BD, Hull ML, Teirney R, Phillips SC, \n",
      "et al. Determination of the transcript profile of human endometrium. Mol \n",
      "Hum Reprod. 2003;9:19–33.\n",
      " 19. Mirkin S, Arslan M, Churikov D, Corica A, Diaz JI, Williams S, et al. In search \n",
      "of candidate genes critically expressed in the human endometrium dur‑\n",
      "ing the window of implantation. Hum Reprod. 2005;20:2104–17.\n",
      " 20. Devesa‑Peiro A, Sebastian‑Leon P , Parraga‑Leo A, Pellicer A, Diaz‑Gimeno \n",
      "P . Breaking the ageing paradigm in endometrium: endometrial gene \n",
      "expression related to cilia and ageing hallmarks in women over 35 years. \n",
      "Hum Reprod. 2022;37:762–76.\n",
      "P . Breaking the ageing paradigm in endometrium: endometrial gene \n",
      "expression related to cilia and ageing hallmarks in women over 35 years. \n",
      "Hum Reprod. 2022;37:762–76.\n",
      " 21. Altmae S, Martinez‑Conejero JA, Salumets A, Simon C, Horcajadas JA, \n",
      "Stavreus‑Evers A. Endometrial gene expression analysis at the time of \n",
      "embryo implantation in women with unexplained infertility. Mol Hum \n",
      "Reprod. 2010;16:178–87.\n",
      " 22. Devesa‑Peiro A, Sebastian‑Leon P , Pellicer A, Diaz‑Gimeno P . Guidelines for \n",
      "biomarker discovery in endometrium: correcting for menstrual cycle bias \n",
      "reveals new genes associated with uterine disorders. Mol Hum Reprod. \n",
      "2021;27:11.\n",
      " 23. Durinck S, Spellman PT, Birney E, Huber W. Mapping identifiers for the \n",
      "integration of genomic datasets with the R/ Bioconductor package \n",
      "biomaRt. Nat Protoc. 2009;4:1184–91.\n",
      " 24. Kanehisa M, Goto S. KEGG: Kyoto encyclopedia of genes and genomes. \n",
      "Nucleic Acids Res. 2000;27:27–30.\n",
      " 25. Ashburner M, Ball CA, Blake JA, Botstein D, Butler H, Cherry JM, et al. Gene \n",
      "ontology: tool for the unification of biology. Nat Genet. 2000;25:25–9.\n",
      " 26. Garcia‑Alonso L, Holland CH, Ibrahim MM, Turei D, Saez‑Rodriguez J. \n",
      "Benchmark and integration of resources for the estimation of human \n",
      "transcription factor activities. Genome Res. 2019;29:1363. https:// doi. org/ \n",
      "10. 1101/ gr. 240663. 118.\n",
      " 27. Karagkouni D, Paraskevopoulou MD, Chatzopoulos S, Vlachos IS, Tast‑\n",
      "soglou S, Kanellos I, et al. DIANA‑TarBase v8: a decade‑long collection of \n",
      "experimentally supported miRNA‑gene interactions. Nucleic Acids Res. \n",
      "2018;46:239–45.\n",
      " 28. Franz M, Lopes CT, Huck G, Dong Y, Sumer O, Bader GD. Cytoscape. \n",
      "js: a graph theory library for visualisation and analysis. Bioinformatics. \n",
      "2016;32:309–11.\n",
      " 29. Vilella F, Moreno‑Moya JM, Balaguer N, Grasso A, Herrero M, Martínez S, \n",
      "et al. Hsa‑miR‑30d, secreted by the human endometrium, is taken up \n",
      "by the pre‑implantation embryo and might modify its transcriptome. \n",
      "Development. 2015;142:3210–21.\n",
      " 30. Henarejos‑Castillo I, Sebastian‑Leon P , Devesa‑Peiro A, Pellicer A, Diaz‑\n",
      "Gimeno P . SARS‑CoV‑2 infection risk assessment in the endometrium: \n",
      "viral infection‑related gene expression across the menstrual cycle. Fertil \n",
      "Steril. 2020;114:223–32.\n",
      " 31. Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, et al. Limma powers \n",
      "differential expression analyses for RNA sequencing and microarray stud‑\n",
      "ies. Nucleic Acids Res. 2015;43:e47.\n",
      " 32. Livak KJ, Schmittgen TD. Analysis of relative gene expression data \n",
      "using real‑time quantitative PCR and the 2‑ΔΔCT method. Methods. \n",
      "2001;25:402–8.\n",
      " 33. R Development Core Team, R Core Team, R Development Core Team. A \n",
      "Language and Environment for Statistical Computing. R Foundation for \n",
      "Statistical Computing. 2018;2. https:// www. R‑‑ proje ct. org.\n",
      " 34. Altmäe S, Koel M, Võsa U, Adler P , Suhorutšenko M, Laisk‑Podar T, et al. \n",
      "Meta‑signature of human endometrial receptivity: a meta‑analysis and \n",
      "validation study of transcriptomic biomarkers. Scient Rep. 2017;7:10077.\n",
      "35. Carrascosa J, Horcajadas J, Moreno‑Moya JM. Chapter 15 ‑ the molecular \n",
      "signature of the endometrial receptivity: research and clinical application. \n",
      "In: Horcajadas JA, Gosálvez J, editors. Reproductomics: the omics revolu‑\n",
      "tion and its impact on human reproductive medicine. Academic Press; \n",
      "2018. p. 301–279.\n",
      " 36. Díaz‑Gimeno P , Horcajadas JA, Martínez‑Conejero JA, Esteban FJ, Alamá \n",
      "P , Pellicer A, et al. A genomic diagnostic tool for human endometrial \n",
      "receptivity based on the transcriptomic signature. Fertil Steril. 2011;95:50.\n",
      " 37. Kao LC, Tulac S, Lobo S, Imani B, Yang JP , Germeyer A, et al. Global gene \n",
      "profiling in human endometrium during the window of implantation. \n",
      "Endocrinology. 2002;143:2119–38.\n",
      " 38. Ponnampalam AP , Weston GC, Trajstman AC, Susil B, Rogers PAW. Molecu‑\n",
      "lar classification of human endometrial cycle stages by transcriptional \n",
      "profiling. Mol Hum Reprod. 2004;10:879–93.\n",
      " 39. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al.\n",
      "lar classification of human endometrial cycle stages by transcriptional \n",
      "profiling. Mol Hum Reprod. 2004;10:879–93.\n",
      " 39. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al. \n",
      "Oestrogen‑modulated gene expression in the human endometrium. Cell \n",
      "Mol Life Sci. 2005;62:239–50.\n",
      " 40. Riesewijk A, Martín J, van Os R, Horcajadas JA, Polman J, Pellicer A, et al. \n",
      "Gene expression profiling of human endometrial receptivity on days LH+2 \n",
      "versus LH+7 by microarray technology. Mol Hum Reprod. 2003;9:253–64.\n",
      " 41. Bastu E, Demiral I, Gunel T, Ulgen E, Gumusoglu E, Hosseini MK, et al. \n",
      "Potential marker pathways in the endometrium that may cause recurrent \n",
      "implantation failure. Reprod Sci. 2019;26:879–90.\n",
      " 42. Bersinger NA, Wunder DM, Birkhäuser MH, Mueller MD. Gene expression \n",
      "in cultured endometrium from women with different outcomes follow‑\n",
      "ing IVF. Mol Hum Reprod. 2008;14:475–84.\n",
      "Page 14 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "•\n",
      " \n",
      "fast, convenient online submission\n",
      " •\n",
      "  \n",
      "thorough peer review by experienced researchers in your ﬁeld\n",
      "• \n",
      " \n",
      "rapid publication on acceptance\n",
      "• \n",
      " \n",
      "support for research data, including large and complex data types\n",
      "•\n",
      "  \n",
      "gold Open Access which fosters wider collaboration and increased citations \n",
      " \n",
      "maximum visibility for your research: over 100M website views per year •\n",
      "  At BMC, research is always in progress.\n",
      "Learn more biomedcentral.com/submissions\n",
      "Ready to submit y our researc hReady to submit y our researc h  ?  Choose BMC and benefit fr om: ?  Choose BMC and benefit fr om: \n",
      " 43. Bhagwat SR, Chandrashekar DS, Kakar R, Davuluri S, Bajpai AK, Nayak S, \n",
      "et al. Endometrial receptivity: a revisit to functional genomics studies \n",
      "on human endometrium and creation of HGEx ‑ERdb. PLoS ONE. 2013;8: \n",
      "e58419.\n",
      " 44. Koot YEM, Van Hooff SR, Boomsma CM, Van Leenen D, Koerkamp \n",
      "MJAG, Goddijn M, et al. An endometrial gene expression signature \n",
      "accurately predicts recurrent implantation failure after IVF. Sci Rep. \n",
      "2015;2016(6):1–12.\n",
      " 45. Lédée N, Munaut C, Aubert J, Sérazin V, Rahmati M, Chaouat GG, et al. \n",
      "Specific and extensive endometrial deregulation is present before \n",
      "conception in IVF/ICSI repeated implantation failures (IF) or recurrent \n",
      "miscarriages. J Pathol. 2011;225:554–64.\n",
      " 46. Pathare ADS, Zaveri K, Hinduja I. Downregulation of genes related to \n",
      "immune and inflammatory response in IVF implantation failure cases under \n",
      "controlled ovarian stimulation. Am J Reprod Immunol. 2017;78: e12679.\n",
      " 47. Shi C, Han HJ, Fan LJ, Guan J, Zheng XB, Chen X, et al. Diverse endometrial \n",
      "mRNA signatures during the window of implantation in patients with \n",
      "repeated implantation failure. Hum Fertil. 2018;21:183–94.\n",
      " 48. Messinisi IE. Ovarian feedback, mechanism of action and possible clinical \n",
      "implications. Hum Reprod Update. 2006;12:557–71.\n",
      " 49. Edwards RG. Implantation, interception and contraception. Hum Reprod \n",
      "(Oxford, England). 1994;9:985–95.\n",
      " 50. Young SL. Oestrogen and progesterone action on endometrium: a \n",
      "translational approach to understanding endometrial receptivity. Reprod \n",
      "Biomed Online. 2013;27:497–505.\n",
      " 51. Bhurke AS, Bagchi IC, Bagchi MK. Progesterone‑regulated endometrial \n",
      "factors controlling implantation. Am J Reprod Immunol. 2016;75:237–45.\n",
      " 52. Devesa‑Peiro A, Sebastian‑Leon P , Garcia‑ Garcia F, Arnau V, Aleman \n",
      "A, Pellicer A, et al. Uterine disorders affecting female fertility: what \n",
      "are the molecular functions altered in endometrium? Fertil Steril. \n",
      "2020;113:1261–74.\n",
      " 53. von Grothusen C, Frisendahl C, Modhukur V, Lalitkumar PG, Peters M, \n",
      "Faridani OR, et al. Uterine fluid microRNAs are dysregulated in women \n",
      "with recurrent implantation failure. Hum Reprod. 2022;37:734–46.\n",
      " 54. Altmäe S, Martinez‑Conejero JA, Esteban FJ, Ruiz‑Alonso M, Stavreus‑\n",
      "Evers A, Horcajadas JA, et al. MicroRNAs miR‑30b, miR‑30d, and miR‑494\n",
      "regulate human endometrial receptivity. Reprod Sci. 2013;20:308–17.\n",
      " 55. Di Pietro C, Caruso S, Battaglia R, IraciSareri M, La Ferlita A, Strino F, et al. \n",
      "MiR‑27a‑3p and miR‑124–3p, upregulated in endometrium and serum \n",
      "from women affected by Chronic Endometritis, are new potential \n",
      "molecular markers of endometrial receptivity. Am J Reprod Immunol. \n",
      "2018;80:12858.\n",
      " 56. Shi C, Shen H, Fan L ‑ JJ, Guan J, Zheng X ‑BB, Chen X, et al. Endometrial \n",
      "MicroRNA signature during the window of implantation changed in \n",
      "patients with repeated implantation failure. Chin Med J. 2017;130:566–73.\n",
      " 57. La Ferlita A, Battaglia R, Andronico F, Caruso S, Cianci A, Purrello M, \n",
      "et al. Non‑coding RNAs in endometrial physiopathology. Int J Mol Sci. \n",
      "2018;19:2120.\n",
      " 58. Estella C, Herrer I, Moreno‑Moya JM, Quiñonero A, Martínez S, Pellicer A, \n",
      "et al. miRNA signature and dicer requirement during human endometrial \n",
      "stromal decidualization in vitro. PLoS ONE. 2012;7:41080.\n",
      " 59. Liu D, Liang Y, Chen M, Yang F, Yao S. Knockdown of circ_0075503 sup ‑\n",
      "stromal decidualization in vitro. PLoS ONE. 2012;7:41080.\n",
      " 59. Liu D, Liang Y, Chen M, Yang F, Yao S. Knockdown of circ_0075503 sup ‑\n",
      "presses cell migration and invasion by regulating miR‑15a‑5p and KLF12 \n",
      "in endometriosis. Mol Cell Biochem. 2021;476:3845–56.\n",
      " 60. Chi RPA, Wang T, Adams N, Wu SP , Young SL, Spencer TE, et al. Human \n",
      "endometrial transcriptome and progesterone receptor cistrome reveal \n",
      "important pathways and epithelial regulators. J Clin Endocrinol Metab. \n",
      "2020;105:1–21.\n",
      " 61. Pawar S, Starosvetsky E, Orvis GD, Behringer RR, Bagchi IC, Bagchi MK. \n",
      "STAT3 regulates uterine epithelial remodeling and epithelial‑stromal \n",
      "crosstalk during implantation. Mol Endocrinol. 2013;27:1996–2012.\n",
      " 62. Cheng YH, Imir A, Suzuki T, Fenkci V, Yilmaz B, Sasano H, et al. SP1 and SP3 \n",
      "mediate progesterone‑ dependent induction of the 17beta hydroxyster ‑\n",
      "oid dehydrogenase type 2 gene in human endometrium. Biol Reprod. \n",
      "2006;75:605–14.\n",
      " 63. Sebastian‑Leon P , Devesa‑Peiro A, Aleman A, Parraga‑Leo A, Arnau V, \n",
      "Pellicer A, et al. Transcriptional changes through menstrual cycle reveal a \n",
      "global transcriptional derepression underlying the molecular mechanism \n",
      "involved in the window of implantation. Mol Hum Reprod. 2021. https:// \n",
      "doi. org/ 10. 1093/ molehr/ gaab0 27.\n",
      " 64. Kim S, Yu NK, Kaang BK. CTCF as a multifunctional protein in genome \n",
      "regulation and gene expression. Exp Mol Med. 2015;47: e166.\n",
      " 65. Shukla S, Kavak E, Gregory M, Imashimizu M, Shutinoski B, Kashlev M, \n",
      "et al. CTCF‑promoted RNA polymerase II pausing links DNA methylation \n",
      "to splicing. Nature. 2011;479:74–9.\n",
      " 66. Phillips JE, Corces VGCTCF. Master weaver of the genome. Cell. \n",
      "2009;137:1194–211.\n",
      " 67. Guo J, Cao B, Xu X, Wu F, Zhu B. Novel CTCF mutations in Chinese \n",
      "patients with ovarian endometriosis. Mol Med Rep. 2018;18:1031–6.\n",
      " 68. Geng J, Cui C, Yin Y, Zhao Y, Zhang C. LncRNA NEAT1 affects endome ‑\n",
      "trial receptivity by regulating HOXA10 promoter activity. Cell Cycle. \n",
      "2022;21:1932–44.\n",
      " 69. Viger RS, Guittot SM, Anttonen M, Wilson DB, Heikinheimo M. Role of the \n",
      "GATA family of transcription factors in endocrine development, function, \n",
      "and disease. Mol Endocrinol. 2008;22:781.\n",
      " 70. Xuan S, Sussel L. GATA4 and GATA6 regulate pancreatic endoderm \n",
      "identity through inhibition of hedgehog signaling. Development (Cam‑\n",
      "bridge). 2016;143:780–6.\n",
      " 71. Bernardi LA, Dyson MT, Tokunaga H, Sison C, Oral M, Robins JC, et al. The \n",
      "essential role of GATA6 in the activation of estrogen synthesis in endome‑\n",
      "triosis. Reprod Sci. 2019;26:60.\n",
      " 72. Yin F, Herring BP . GATA‑6 can act as a positive or negative regula‑\n",
      "tor of smooth muscle‑specific gene expression *. J Biol Chem. \n",
      "2005;280:4745–52.\n",
      " 73. Zhai J, Luo G. GATA6‑induced FN1 activation promotes the proliferation, \n",
      "invasion and migration of oral squamous cell carcinoma cells. Mol Med \n",
      "Rep. 2022;25:102.\n",
      " 74. Fullwood MJ, Liu MH, Pan YF, Liu J, Xu H, Mohamed YB, et al. An \n",
      "oestrogen‑receptor‑α‑bound human chromatin interactome. Nature. \n",
      "2009;462:58–64.\n",
      "2009;462:58–64.\n",
      " 75. Lieberman‑Aiden E, Van Berkum NL, Williams L, Imakaev M, Ragoczy \n",
      "T, Telling A, et al. Comprehensive mapping of long range interactions \n",
      "reveals folding principles of the human genome. Science. 2009;326:289.\n",
      " 76. Birney E, Stamatoyannopoulos JA, Dutta A, Guigó R, Gingeras TR, \n",
      "Margulies EH, et al. Identification and analysis of functional elements \n",
      "in 1% of the human genome by the ENCODE pilot project. Nature. \n",
      "2007;447:799–816.\n",
      " 77. Tajti F, Kuppe C, Antoranz A, Ibrahim MM, Kim H, Ceccarelli F, et al. A func‑\n",
      "tional landscape of CKD entities from public transcriptomic data. Kidney \n",
      "Int Rep. 2020. https:// doi. org/ 10. 1016/j. ekir. 2019. 11. 005.\n",
      " 78. Henarejos‑ Castillo I, Devesa‑Peiro A, de Miguel‑ Gomez L, Sebastian‑Leon \n",
      "P , Romeu M, Aleman A, et al. Predicted COVID ‑19 molecular effects on \n",
      "endometrium reveal key dysregulated genes and functions. Mol Hum \n",
      "Reprod. 2022;28:gaac035.\n",
      " 79. Barrett T, Wilhite SE, Ledoux P , Evangelista C, Kim IF, Tomashevsky M, et al.\n",
      "endometrium reveal key dysregulated genes and functions. Mol Hum \n",
      "Reprod. 2022;28:gaac035.\n",
      " 79. Barrett T, Wilhite SE, Ledoux P , Evangelista C, Kim IF, Tomashevsky M, et al. \n",
      "NCBI GEO: archive for functional genomics data sets–update. Nucleic \n",
      "Acids Res. 2013;41(Database issue):D991‑5.\n",
      "Publisher’s Note\n",
      "Springer Nature remains neutral with regard to jurisdictional claims in pub‑\n",
      "lished maps and institutional affiliations.\n"
     ]
    }
   ],
   "source": [
    "my_json = []\n",
    "\n",
    "for docs in storage:\n",
    "    print(docs.page_content)\n",
    "\n",
    "    my_json.append(chain.invoke({\"query\": docs.page_content}))\n",
    "\n",
    "\n",
    "#chain.invoke({\"query\": storage[0].page_content})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "19cb09a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'text': 'A total of 3,608 distinct genes from the 19 gene lists were associated with endometrial progression',\n",
       "  'doi': '10.1186/s12958‑023‑01131‑4',\n",
       "  'citation': 'Parraga‑Leo et al. (2023)'},\n",
       " {'text': 'The lists’ regulation was significantly favoured by TFs (89% (17/19) of gene lists) and progesterone (47% (8 /19) of gene lists), rather than miRNAs (5% (1/19) of gene lists) or estrogen (0% (0/19) of gene lists), respectively (FDR < 0.05). Exceptionally, two gene lists that were previously associated with implantation failure and unexplained infertility were less hormone‑dependent, but primarily regulated by estrogen. Although endometrial progression genes were mainly targeted by hormones rather than non‑hormonal contributors (odds ratio = 91.94, FDR < 0.05), we identified 311 TFs and 595 miRNAs not previously associated with ovarian hormones. We highlight CTCF, GATA6, hsa‑miR‑15a‑5p, hsa‑miR‑218‑5p, hsa‑miR‑107, hsa‑miR‑103a‑3p, and hsa‑miR‑128‑3p, as overlapping novel master regulators of endometrial function. The gene expression changes of selected regulators through‑out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their potential regulatory roles in the endometrium.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'This study revealed novel hormonal and non‑hormonal regulators and their relative contributions to endometrial progression and pathology, providing new leads for the potential causes of endometrial‑factor infertility. The endometrium is the innermost layer of the uterus, which undergoes dynamic histological, physiological and molecular changes that allow it to synchronize with embryo development, facilitate embryo implantation, and ultimately, establish a successful pregnancy. Implantation is a crucial and complex limiting step for conception which occurs between days 19 to 24 of a normal menstrual cycle. Consequently, this period of time is called the window of implantation (WOI) and is characterized by abrupt transcriptomic changes in the endometrial tissue. Alterations in either WOI establishment and endometrial progression may lead to implantation failures, which, along with biochemical miscarriage, account for more than 50% of pregnancy losses at pre-clinical stages. These alterations have broadly been classified as being related to a displaced WOI, caused by variable timing of endometrial progression, or a disrupted WOI, where impaired endometrial function prevents the establishment of an effective WOI. Initially, the WOI was postulated to mainly be regu -lated by the ovarian hormones (namely, estrogen and progesterone) acting via their respective nuclear receptors (i.e., ESR1/2 and PGR). However, evidence has shown that these receptors cooperate with other transcription factors (TFs) and co-regulators to mediate uterine physiology [8]. In fact, TFs (such as homeobox TFs, FOXA2, and KLF9) have recently been reported as key regulators for the establishment of endometrial receptivity [9, 10], and microRNAs (miRNAs) have been found to similarly intervene in this complex and multifactorial process as transcriptional regulators [11]. While the involvement of both types of regulators was widely reported in embryo implantation [12] and other infertility-related diseases, including recurrent implantation failure (RIF) [13], the complex interactions between the different types of regulators and their respective contributions to endometrial progression and function have not been investigated from a systemic point of view. This paradigm shift approaches genes and regulators in an integrative way, by considering how they interact with and coordinate each other to carry out cellular processes, rather than reinforcing the premise they work independently [14]. By more accurately reflecting the biological milieu, systemic approaches take gene-based discoveries to the next level.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Previous studies of endometrial transcriptomics have been used to define the WOI, compare the endometrium of healthy women as their menstrual cycles progressed [4, 17–19]; address endometrial differences related to age [20] or the dysfunction of the WOI between patients with implantation failure and healthy controls [21]; and finally, correct for the menstrual cycle bias that can mask important biomarkers [22]. Notwithstanding, the candidate biomarkers of endometrial receptivity reported in each of the aforementioned studies overlap poorly [7]. Indeed, recent comparisons conducted by Sebastian-Leon and colleagues [7] found no congruities between 16 different reported gene lists of endometrial receptivity, and only fair to moderate functional agreements between some of the included signatures. Despite previous transcriptional regulation studies, there have been no holistic studies of endometrial receptivity.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'fair to moderate functional agreements between some of the included signatures. Despite previous transcriptional regulation studies, there have been no holistic studies of this regulatory process, which robustly analyse the drivers of transcription across a higher number of studies (that each searched for endometrial receptivity biomarkers) and map the relative contributions of each type of regulator to endometrial progression regulation (Ovarian hormones [i.e., estrogen, progesterone], TFs, miRNAs). Thus, the aim of this study was to utilize available genomic data regarding transcriptional regulators to identify overlapping mediators of endometrial transcriptional regulation (i.e., ovarian hormones, TFs, and miRNAs) and determine the relative contribution of each type of regulator among the previously reported gene signatures. These findings could help unveil the master regulators and principal type of endometrial regulation, paving the way for further research aimed at improving woman´s reproductive health. Annotating gene lists associated with endometrial progression and function. Endometrial progression and implantation failure gene lists [7] were retrieved and updated using a public data repository (i.e., Gene Expression Omnibus (GEO)).',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'To study the hormonal regulation of endometrial progression, the Kyoto Encyclopaedia of Genes and Genomes (KEGG) [24], and Gene Ontology (GO) [25] databases were consulted for ovarian hormone-related genes, using relevant keywords (i.e., progesterone, estrogen, oestrogen and estradiol) and selecting those pathways and functions that contained them (Supplementary Fig. S1B). Genes associated with the obtained human KEGG pathways and GO functions were grouped according to whether they were unique to progesterone (P4 gene set) or estrogen (E2 gene set) or related to both hormones (P4 and E2 gene set). Gene targets of the nuclear progesterone (PGR) and estrogen (ESR1, ESR2) receptors were added to their corresponding gene sets using DoRothEA (Discriminant Regulon Expression Analysis) database [26], considering only manually-curated or ChiP-Seq experimentally-validated gene-TF relationships. Finally, the P4 and E2 gene sets were independently mapped to each gene list.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Prioritizing key regulators in endometrial progression and function\\nWe built regulatory networks to systemically analyse endometrial regulation.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'We built regulatory networks to systemically analyse endometrial regulation. Nodes represented gene lists and regulators (miRNAs or TFs), while edges indicated significant enrichment among the lists and their corresponding regulators (false discovery rate (FDR) ≤ 0.05). To select the most influential regulators (i.e., those which targeted genes in most of our signatures), we studied the degree distribution of the networks (i.e., the number of gene lists regulated by each molecule) and prioritized the miRNAs and TFs which surpassed the relative maximum (i.e., 1.50 times the interquartile range (IQR)) number of relationships (gene list–regulators). All networks were built and analysed with Cytoscape version 3.7 [28] (Supplementary Fig. S1B). To evaluate how the miRNAs’ expression change in the endometrium throughout the menstrual cycle, we used the raw data from GSE44558 GEO dataset [29], deriving from 20 endometrial samples collected throughout the menstrual cycle: four in early-proliferative (EPF), four in late-proliferative (LPF), four in early-secretory (ESE), four in mid-secretory (MSE) and four in late-secretory (LSE). To evaluate how the TFs’ expression change in the endometrium throughout the menstrual cycle, we analysed the integrated endometrial dataset previously created by our group, and reported in [30], which compiled data from five prior publications (GSE98386, GSE29981, GSE4888, GSE119209 and GSE86491). This dataset included endometrial gene expression',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Of the 19 gene lists used for our analysis, eleven were obtained from published studies evaluating control patients throughout the menstrual cycle [4, 17–19, 34–40], and the other eight were derived from studies comparing patients with RIF or unexplained infertility to controls [21, 41–47] (Table 1). Unifying all the afore-',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'Table 1'},\n",
       " {'text': 'mentioned signatures, we compiled 3,608 genes related to endometrial progression and function. Hormonal regulation of endometrial progression is largely driven by progesterone. We identified 7,540 and 698 genes related to estrogen and progesterone hormones, respectively. However, as determined by the relative contribution of each type of hormone within each gene list (Fig. 1A), 17/19 (89%) signatures favoured regulation by progesterone rather than estrogen. These differences were significant in 47% of the signatures (FDR < 0.05). Exceptionally, the genes reported by the Altmae2010 and Koot signatures, that were related to unexplained infertility and RIF, had a preference for estrogen regulation (Fig. 1A). With respect to the proportion of progesterone-related genes, Altmae2010 showed 2.1 times more estrogen-related genes while Koot showed 1.7 times more. More over, comparing the proportion of genes regulated by both hormones, between signatures revealed that Alt -',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'We identified a total of 770 miRNAs and 555 TFs regulating at least one of the 3,608 genes reported in the 19 gene lists. Further, we observed that 29% of the miRNAs and 44% of the TFs were associated with the ovarian hormones (i.e., progesterone and/or estrogen). Significantly more genes were targeted by hormonally-influenced regulators rather than non-hormonal regulators (P -value < 0.01, odds ratio =  91.94). Nevertheless, we also identified 595 miRNAs (77.27%) and 311 TFs (56.04%) that were unrelated to progesterone and/or estrogen, highlighting a substantial proportion of hormonally-independent regulators of the menstrual cycle.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Table 1 Characteristics of the 19 gene lists included in this study. Gene lists were divided according to their evaluated condition. Each signature was labelled using the name of first author, followed by the publication year in the case of duplicates. Endometrial staging was based on different variables including urinary LH, histological analysis, ultrasound evaluation, or the number of days since the hCG trigger. Participants included healthy controls and women with infertility‑related conditions. Various platforms were used to evaluate gene expression. The thresholds applied for gene refinement and the final number of genes per signature are also presented\\nAbbreviations: Δ delta  Ct, adj-P-value Adjusted P‑value, C Control group, d Day of menstrual cycle, EPF Early proliferative, ESE Early secretory, |FC| Absolute fold‑change, FC Fold change, HRT Hormone replacement therapy, IF Implantation failure, LH Luteinizing hormone, LH + X X days post LH surge, LPF Late proliferative, LSE Late‑secretory, M Miscarriage, MPF Mid‑proliferative, MSE Mid‑secretory, n number of samples, N/A Not applicable, OP Ongoing pregnancy, PF Proliferative, PFP Proportion of false positives, PR Pre‑receptive, R Receptive, RIF Recurrent implantation failure, RRA Robust Rank Aggregation algorithm, SF Spontaneously fertile women, UI Women with unexplained infertility, vs versus\\nCondition Gene signature Endometrial staging Participants Platform Threshold No. genes References\\nEndometrial progres‑ sion\\nAltmäe2017 N/A N/A N/A RRA 57 [34]\\nBorthwick Urinary LH; Histologi‑ cally confirmed PF d9‑11 (n = 5) vs R LH + 6–8 (n = 5)\\nAffymetrix Genechip Hu95A N/A 116 [18]\\nCarrascosa N/A N/A Quantitative RT‑PCR Based on literature 187 [35]\\nCarson Urinary LH; Histologically confirmed ESE LH + 2–4 (n = 3) vs MSE LH + 7–9 (n = 3)\\nAffymetrix Genechip Hu95A |FC|> 2 695 [17]\\nDiaz‑Gimeno Urinary LH PR LH + 1–5 (n = 15) vs R\\nAgilent custom gene expression microarray |FC| > 2 234 [36]\\nKao Urinary LH; Histologically confirmed LPF d8‑10 (n = 4) vs MSE LH + 8‑10 (n = 7)\\nAffymetrix Genechip Hu95A |FC|> 2 and P‑value <0.05 340 [37]\\nMirkin Urinary LH; Histological dating ESE LH + 3 (n = 3) vs MSE LH + 8 (n = 5)\\nAffymetrix Genechip Hu95A |FC|>2 and adj‑P‑value < 0.05 105 [19]\\nPonnampalam Histological dating EPF (n = 5) MPF (n = 7) LPF (n = 3) ESE (n = 7) MSE (n = 8) LS (n = 7) Mense (n = 6)\\nCustom Adj‑P‑value <0.05 306 [38]\\nPunyadeera Histological dating Mense (n = 2) vs LPF (n=2)\\nAffymetrix HG‑U133A FC and P‑value 50 [39]\\nRiesewijk Urinary LH; Histological dating PR LH + 2 (n = 5) vs R LH + 7 (n = 5)\\nAffymetrix Genechip Hu95A |FC|> 3 in at least four out of five women. 196 [40]\\nTalbi Histological dating P (n = 6) vs ESE (n = 3) vs MSE (n = 8) vs LSE (n = 6)\\nAffymetrix HG‑U133 Plus 2.0 |FC|> 1.5 and adj‑P‑value < 0.05 317 [4]',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Implantation failure Altmäe2010 Urinary LH UI (n = 4) vs C (n = 5) Whole Human Genome Oligo Microarray (Agilent Technologies) |FC|> 2 and PFP < 0.05 184 [21] Bastu 6‑7 days after LH surge IF (n = 24) vs Fertile women (n = 24) Agilent‑039494 SurePrint G3 Human GE v2 8x60K Log2(FC) ≥ 1 and adj‑P‑value ≤0.05 |FC|> 2 and PFP < 0.05 524 [41] Bersinger N/A IF (n = 3) vs M (n = 3) vs OP (n = 3) Affymetrix Human Genome U133A 2.0 Chips Not specified 93 [42] Bhagwat N/A PR vs R Multiple platforms Based on literature 177 [43] Koot Urinary LH RIF (n = 43) vs C (n = 72) Human whole genome gene expres‑sion microarrays V2 (Agilent, Belgium) Machine learning predictor 310 [44] Lédée Ultrasound evaluation IF (n = 30) vs Fertile women (n = 15) Affymetrix Human Genome U133 Plus 2.0 Array Adj‑P‑value <0.01 and Δ > 1.35 322 [45] Pathare 6‑7 days after hCG administration IF (n = 10) vs Healthy oocyte donors (n = 8) Illumina HumanHT‑12 V4.0 expression beadchip |FC|> 2 818 [46] Shi Urinary LH; Ultrasound evaluation',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'IF (n = 10) vs Healthy oocyte donors (n = 8) Illumina HumanHT‑12 V4.0 expression beadchip |FC|> 2 818 [46] Shi Urinary LH; Ultrasound evaluation RIF (n = 12) vs C (n \\n= 10) Agilent‑052909 CBC lncRNAmRNA V3 |FC|≥ 2 and adj‑P‑value ≤ 0.05 281 [47]',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Of the respective miRNAs and TFs targeting genes related to endometrial progression and function, 417 miRNAs and 467 TFs were significantly over- represented in at least one gene list (FDR < 0.05, Tables S3-S4). Further, most signatures were significantly governed by TFs rather than miRNAs (4 > odds ratio > 164, FDR < 0.05, 17 lists [89%]), with the exception of Koot’s signature that was governed by miRNAs and showed the opposite (odds ratio = 0.35, FDR < 0.05, Table S5). Eight out of nineteen gene lists (47%) had significant differences between the proportion of genes regulated by progesterone versus estrogen. (*FDR ≤ 0.05). Descriptive analysis of hormonal regulation, normalized by the size of each gene list. Gene lists clustered according to higher (blue) or lower (orange) contribution of hormonal regulation',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Parraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology (2023) 21:84 \\n\\nHsa‑miR‑15a‑5p, hsa‑miR‑128‑3p, hsa‑miR‑218‑5p, \\nhsa‑miR‑27b‑3p, hsa‑miR‑107, hsa‑miR‑424‑5p, \\nhsa‑miR‑195‑5p, hsa‑miR‑103a‑3p, has‑let‑7b‑5p and hsa‑miR‑22‑3p are key players in the regulation of endometrial progression and function\\n\\nTo bridge the gap of endometrial-based evidence in DoRothEA and TarBase databases, we aimed to identify the common miRNAs that mediate gene expression related to endometrial progression and function. We evaluated the number of endometrial gene lists that each miRNA controlled (Fig. 2A). This strategy highlighted the most influential miRNAs that were commonly found across the endometrial gene lists. Notably, 85.80% of miRNAs regulated genes from one to three signatures, and only 59 miRNAs (14.20%) were associated with more than three signatures. Ultimately, 24 miRNAs were prioritized for regulating genes of at least six signatures.\\n\\nNext, we validated the gene expression trends for these 24 prioritized miRNAs in an independent endometrial gene expression dataset (GSE44730) (Fig. 2B). This dataset included the miRNA expression in endometrial biopsies from 20 healthy women in different phases of the menstrual cycle. We found that two of our prioritized miRNAs were not expressed in the endometrial tissue, and only 10 miRNAs (i.e., hsa-miR-15a-5p, hsa-miR-128-3p, hsa-miR-218-5p, hsa-miR-27b-3p, hsa-miR-107, hsa-miR-424-5p, hsa-miR-195-5p, hsa-miR-103a-3p, has-let-7b-5p and hsa-miR-22-3p) showed significant changes across the menstrual cycle (Fig. 2B). Of these, only the expression of hsa-miR-22-3p significantly increased from ESE to MSE stages (FDR = 0.02, Fig. 2B), highlighting the potential inhibitory role of this miRNA in endometrial function. The expression of the remaining miRNAs decreased during the MSE phase.\\n\\nCTCF and GATA6 as distinguished transcriptional regulators of the human menstrual cycle\\n\\nWe repeated the regulation network analysis with the TFs, to identify the prominent TFs commonly regulating the genes related to endometrial progression and function in the reported signatures. We observed 78.80% of the TFs regulated genes in up to 10 signatures, and only 99 TFs (21.20%) regulated genes in over 10 (Fig. 3A). We highlighted CTCF (a CCCTC-binding factor, which functions as a transcriptomic repressor) as the TF with the most influential regulation of endometrial progression and function across 95% of studies (18 gene lists, excluding that of Punyadeera, Fig. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53. Notably, GATA6 was selected due to novel association with endometrial progression and function.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'Parraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology (2023) 21:84'},\n",
       " {'text': 'To infer the expression of CTCF and GATA6  in \\nendometrial tissue throughout the menstrual cycle, \\nwe analysed their expression using a relevant endo - \\nmentrial dataset recently created by our group [30] \\n(Table S1 ). We observed that, relative to the global \\nexpression of endometrial genes, CTCF was broadly \\nexpressed while GATA6  was moderately expressed. \\nIndeed, CTCF was significantly down-regulated \\nbetween the PF and ESE phases (FDR = 2.10E-03)  \\nand ESE to the MSE phase (FDR = 2.10E-04), while \\nGATA6 expression increased from the PF to the \\nMSE phase (FDR = 1.70E-11) and from the ESE  \\nto MSE phase (FDR = 9.80E-15). These findings \\nsupported the role of these TFs at the beginning \\nof the menstrual cycle and during the MSE phase \\n(Fig.\\xa03 B). \\nThese in-silico  validations were then prospec - \\ntively corroborated by RT-qPCR analysis in an inde - \\npendent sample set of 20 endometrial biopsies. We \\nobserved a similar expression profile of CTCF and \\nGATA6 throughout the menstrual cycle, with signifi-\\n',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with significant changes between the endometrial phases. Notably, CTCF expression was decreased between the PF/ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C). The prioritized miRNAs included has‑miR‑16‑5p, has‑miR‑138‑5p, has‑miR‑21‑3p, has‑miR‑205‑5p, has‑miR‑27a‑5p, has‑miR‑15a‑5p, has‑miR‑155‑5p, has‑miR‑129–2‑3p, has‑miR‑147a, has‑miR‑146a‑5p, has‑miR‑107, has‑miR‑424‑5p, has‑miR‑203a‑3p, has‑miR‑195‑5p, has‑miR‑124‑3p, has‑miR‑128‑3p, has‑miR‑22‑3p, has‑miR‑27b‑3p, has‑let‑7b‑5p, has‑miR‑1343‑3p, has‑miR‑1‑3p, has‑miR‑103a‑3p, has‑miR‑23b‑3p, has‑miR‑218‑5p. Significant changes between phases are denoted with asterisks (* FDR ≤ 0.05, ** FDR ≤ 0.01.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Fig. 2 (See legend on previous page.)',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Page 9 of 14\\nParraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology (2023) 21:84 \\n\\nFig. 3 Transcription factors universally regulating endometrial progression and function and their expression throughout the menstrual cycle. A Regulatory network of TFs forming relationships with numerous gene lists (2–18). TFs targeting specific gene lists were placed in the center of the network. CTCF stood out as the only TF that targeted genes belonging to 18/19 (95%) different signatures. B In-silico analysis of CTCF and GATA6 expression throughout the menstrual cycle. Significant changes between phases are marked with asterisks (**FDR ≤ 0.01, ***FDR ≤ 0.001). C RT‑qPCR validation. Bar plot shows the relative mRNA expression of CTCF and GATA6 in endometrial biopsies collected throughout the menstrual cycle. Data is presented as a mean ± SD. Significant differences of CTCF (*FDR < 0.05) and GATA6 (***FDR < 0.001) expression were observed between proliferative/early‑secretory (PF/ESE) and mid‑secretory/late‑secretory phases (MSE/LSE). Abbreviations in (B and C): PF, Proliferative; ESE, early‑secretory; MSE, mid‑secretory; LSE, late‑secretory; SD, standard deviation',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'This data-driven approach exposed the common transcriptional regulators among 19 studies who proposed variable biomarkers of endometrial progression and function. In this study, we focused on understanding the relative contribution of both the hormonal and non-hormonal regulation, from an alternative holistic perspective. We applied data-driven hypothesis research that, unlike the traditional scientific method, allowed us to generate new hypotheses based on all available biologically-relevant knowledge [16], observing the molecular relationships from a wider scale view. Besides, some traditional molecular procedures such as PCRs are also performed to corroborate the insight uncovered with this approach. We highlighted a larger influence of progesterone-related genes and TFs in endometrial progression and observed a larger contribution of miRNAs and estrogen-related genes in endometrial pathology. Furthermore, we unveiled CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-miR-103a-3p and hsa-miR-128-3p as master regulators in the endometrium, and these findings were reinforced by the dynamic changes in their expression throughout the menstrual cycle (dually validated with in-silico and prospective analyses). The menstrual cycle is perceived to be under tight regulation by the ovarian hormones, progesterone and estrogen [48]. Supporting the role of progesterone in the secretory endometrium [49], and reinforcing its indispensable role in endometrial regulation [50, 51], we found progesterone and its related genes had a larger influence in menstrual cycle progression than estrogen. As the RIF-related gene lists reported by Altmae2010 and Koot were generally controlled by estrogen, this further suggested that while progesterone could be crucial for a suitable endometrial progression and mid-secretory phase acquisition, estrogenic irregularities may be more conducive to endometrial function and molecular pathology. Indeed, this hypothesis corroborated previous reports of most uterine disorders being estrogen-dependent [44, 52]. Supporting the findings from another recent study from our group, where the Altmae2010 and Koot signatures were classified as highly-predictive signatures to identify endometrial pathology in comparison with a prediction model that detected endometrial progression [7 ]. This study distinguished these signatures (related to implantation failure and unexplained infertility) based on the magnitude of their miRNA and estrogen-regulation, which contrasted with the prominent influence of the TFs over the remaining signatures more related to endometrial progression. All these facts reinforce the hypothesis that implantation failure is mainly caused by impaired miRNA expression [53].',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'This study transcended the simplistic paradigm of the bihormonal regulation of endometrial function by identifying a plethora of mediators, including 595 miRNAs and 311 TFs, that were previously unrelated to these ovarian hormones. These findings set the foundation for new discoveries, revealing alternative pathways and new actors in the regulation of the human endometrium, and deepening our understanding of the complex regulatory mechanisms behind endometrial progression and unexplained infertility. MicroRNAs are well known to be involved in embryo implantation and receptivity control. Using network analysis, we unveiled the 10 common miRNAs regulating endometrial progression and function across published biomarker signatures (hsa-miR-22-3p, hsa-miR-107, hsa-miR-103a-3p, hsa-miR-128-3p, hsa-miR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, hsa-miR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p).',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Their significant gene expression changes in the secre - \\ntory endometrium support their potential role in the \\nendometrial regulation through an inhibition during \\nthe WOI. Notably, hsa-miR-424-5p, hsa-miR-27b-3p \\nand hsa-miR-195-5p have been previously associated \\nwith RIF [11, 54, 56], while has-let-7b-5p was associ - \\nated with human endometrial receptivity [57], and \\nhsa-miR-22-3p was overexpressed in a decidualization \\nmodel [58]. Moreover, hsa-miR-15a-5p was previously \\nassociated with endometriosis [59]. Notwithstanding, \\nthis study is the first to associate hsa-miR-15a-5p, hsa- \\nmiR-218-5p, hsa-miR-103a-3p, hsa-miR-107, and hsa- \\nmiR-128-3p to endometrial progression and functional \\nregulation in humans.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Page 11 of 14\\nParraaga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \\n\\nendometrial progression and function were beyond the scope of this project and merit further investigation. CTCF is a conserved zinc finger protein whose regulatory functions are well characterized throughout the human body [64]. CTCF acts as a transcriptional repressor in RNA polymerase II (Pol II) pausing and imprinting and X-chromosome inactivation [65], as well as an insulator, blocking the interaction between enhancers and the promoters of neighbouring genes [66]. Considering these attributed functions and its variable expression throughout the menstrual cycle, we propose that CTCF exerts an inhibitory role in endometrial tissue during the PF phase of the menstrual cycle. With the significant downregulation of CTCF in the secretory endometrium, dually validated in-silico and experimentally herein, its inhibited genes would be derepressed and become transcriptionally active during the WOI. This interpretation supports previous findings from our group demonstrating that, during the WOI, a global transcriptional derepression may be required for implantation and early embryo development [63]. Transcriptional derepression has been associated with multiple human disease states and should be investigated further within the context of endometrial-factor infertility. Despite previous associations of CTCF with endometriosis [67], its implication in endometrial progression and function has not been directly proposed until now. Nevertheless, the role of CTCF in endometrial receptivity has been revealed through its interaction with HOXA10, a gene mainly expressed in endometrium involved in functions such as endometrial proliferation and differentiation, the formation of pinopodes or embryo implantation [68]. Indeed, this study proposed that an overexpression of CTCF can lead to a drop in HOXA10 expression, affecting endometrial proliferation and endometrial function and corroborating the fact that CTCF must decrease its expression as we can observe in our findings. GATA6, the second prioritized TF that overlapped across the gene lists, is a transcription factor belonging to the GATA family, a highly conserved family of six zinc finger proteins [69]. Besides its essential role in embryonic development [70], GATA6 plays a key role in endometriosis and regulates steroidogenic genes [71]. Together with evidence supporting both an activating and inhibitory role for GATA6 [72, 73], the significant changes we observed in its expression across the menstrual cycle reflect its presumed involvement in the endometrium and suggest an activating role in endometrial progression and endometrial receptivity acquisition. Like CTCF, this is the first time that GATA6 is proposed as a key TF in endometrial progression and function.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'endometrial regulation remain elusive, emerging experimental technologies such as High-throughput Chromosome Conformation Capture (Hi-C) or chromatin interaction analysis with paired-end tags (ChIA-PET) that respectively study general chromatin interactions or chromatin interactions involving a specific protein [74, 75] may improve our understanding of genetic regulation in the future. Nevertheless, as collaborative efforts unveil new regulators and gene targets in humans (such as the ENCODE project [76]), the complexities of transcriptional regulation can be exposed by in-silico analyses like those described herein, and we can gain a better understanding of the processes involved in menstrual cycle progression and endometrial competence. Furthermore, it should be noted that results of gene expression studies are influenced by different variables, such as the experimental platforms employed, study designs, or patients’ characteristics of each independent',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'expression studies are influenced by different variables, such as the experimental platforms employed, study designs, or patients’ characteristics of each independent study, adding variability [30, 77, 78]. Despite this fact, we have found that our holistic approach overcomes this undesired variability effect by finding common upstream master regulators across studies and predicting new relationships between hormones, TFs and miRNAs. This approach also provides the basis for future single-molecule studies that aim to elucidate new regulatory pathways in endometrial progression and function. Our results lay the foundation for further molecular studies that can validate the function(s) of the prevailing endometrial regulators we prioritized. The study shows for the first time, the relative contribution of estrogens, progesterone, TFs and miRNAs in endometrial function and progression. Endometrial progression is mainly influenced by progesterone-related genes and TFs, whereas miRNAs and estrogen-related genes play a larger role in endometrial pathology. Moreover, we highlight novel common transcriptional regulators such as CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-miR-103a-3p, and hsa-miR-128-3p across 19 studies that propose biomarkers in endometrial regulation. These results reveal the molecular mechanism underlying endometrial regulation and lay the foundation for the development of targeted therapies for patients with endometrial-factor infertility.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'We reviewed 19 publications related to endometrial progression and function, extracted the gene signatures from the original publication, and created corresponding gene lists that we could annotate. We performed different functional analyses based on the type of regulator (i.e., ovarian hormones, transcription factors and miRNAs). We used online databases to obtain the information of the regulatory process, at the hormonal (Gene Ontology, Kyoto Encyclopedia of Genes and Genomes, Dorothea), transcription factor (Dorothea) and miRNA (Tarbase) level and analysed the results accordingly. Finally, we employed an integrative analysis and prioritized the most relevant regulators for subsequent validation. TF, Transcription factors; miRNA, microRNA.',\n",
       "  'doi': 'https://doi.org/10.1186/s12958-023-01131-4',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Additional file 5: Supplementary Table S4. Results of TFs enrichment. Excel file with the over‑representation analysis results of the transcription factors (TFs). Each sheet contains the results of a specific list of genes. In each sheet, the following information is included: ID, gene symbol of the TF; GeneRatio, the proportion of TF targets in the gene list; BgRatio, the proportion of TF targets in the whole genome; OddsRatio; the odd ratios for each TF, pvalue; the P‑value obtained for each TF; padjust, P‑value adjusted by FDR for each TF; geneID, the gene symbol ID of the TF targets. Additional file 6: Supplementary Table S5. Relative contribution of each regulator type (TFs or miRNAs) in each gene list. A Fisher’s test was performed to calculate the proportional differences between transcription factors (TFs) and microRNAs (miRNAs). Resulting P‑values were adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Resulting P‑values were adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene lists were mainly regulated by TFs, whereas odds ratios < 1 indicate a higher regulation by miRNAs. Odds Ratio with a value of “Inf” indicate that miRNAs were not found for this gene list. Percentages indicate the proportion between the significant regulators in each gene signature and regulators with at least one target gene in the corresponding signature. (***FDR ≤0.001).',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'The data used to analyze the TF and miRNA expression throughout the men‑\\ntstrual cycle and in infertility are available in Gene Expression Omnibus, with \\nIDs GSE98386, GSE29981, GSE4888, GSE119209, GSE86491, and GSE44558. The raw endometrial gene expression data (from previous studies) was downloaded from the publicly‑available functional genomic dataset repository Gene Expression Omnibus (GEO) [79] for curation/processing. In accordance with GEO policies, all patient data was anonymized and encrypted, and therefore an additional institutional review board approval was not required for our in-silico validation of regulator gene expression changes in endometrial tissue.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'tional institutional review board approval was not required for our in-silico validation of regulator gene expression changes in endometrial tissue. On the other hand, for the experimental validation of gene expression changes in endometrial biopsies, all individuals gave written informed consent for endometrial gene expression evaluation prior to participating (Ethics Committee of the Instituto Valenciano de Infertilidad [IVI Valencia], Valencia, Spain [1810‑FIVI‑066‑PD/1706‑FIVI‑048‑PD]).',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'Ethics Committee of the Instituto Valenciano de Infertilidad [IVI Valencia], Valencia, Spain [1810‑FIVI‑066‑PD/1706‑FIVI‑048‑PD]'},\n",
       " {'text': 'Parraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology (2023) 21:84 \\n\\nAuthor details\\n1 IVIRMA Global Research Alliance, IVI Foundation, Instituto de Investigación Sanitaria La Fe (IIS La Fe), Av. Fernando Abril Martorell 106, Torre A, Planta 1ª, 46026 Valencia, Valencia, Spain. 2 Department of Pediatrics, Obstetrics and Gynaecology, Universidad de Valencia, Av. Blasco Ibáñez 15, 46010 Valen‑cia, Valencia, Spain. 3 IVIRMA Global Research Alliance, IVIRMA Valencia, Plaza de La Policia Local 3, 46015 Valencia, Spain.\\nReceived: 21 June 2023 Accepted: 22 August 2023\\n\\nReferences\\n1. Strowitzki T, Germeyer A, Popovici R, von Wolff M. The human endometrium as a fertility-determining factor. Hum Reprod Update. 2006;12:617–30.\\n2. Cakmak H, Taylor HS. Implantation failure: Molecular mechanisms and clinical treatment. Hum Reprod Update. 2011;17:242–53.\\n3. Lessey BA, Young SL. What exactly is endometrial receptivity? Fertil Steril. 2019;111:611–7.\\n4. Talbi S, Hamilton AE, Vo KC, Tulac S, Overgaard MT, Dosiou C, et al. Molecular phenotyping of human endometrium distinguishes menstrual cycle phases and underlying biological processes in normo‑ovulatory women. Endocrinology. 2006;147:1097–121.\\n5. Somigliana E, Vigano P , Busnelli A, Paffoni A, Vegetti W, Vercellini P . Repeated implantation failure at the crossroad between statistics, clinics and over‑diagnosis. Reprod Biomed Online. 2018;36:32–8.\\n6. Craciunas L, Gallos I, Chu J, Bourne T, Quenby S, Brosens JJ, et al. Conventional and modern markers of endometrial receptivity: a systematic review and meta‑analysis. Hum Reprod Update. 2019;25:202–23.\\n7. Sebastian‑Leon P , Garrido N, Remohí J, Pellicer A, Diaz‑Gimeno P . Asynchronous and pathological windows of implantation: Two causes of recurrent implantation failure. Hum Reprod. 2018;33:626–35.\\n8. Vasquez YM, DeMayo FJ. Role of nuclear receptors in blastocyst implantation. Semin Cell Dev Biol. 2013;24:724–35.\\n9. Wang H, Zhang S, Lin H, Kong S, Wang S, Wang H, et al. Physiological and molecular determinants of embryo implantation. Mol Aspects Med. 2013;34:939–80.\\n10. Kelleher AM, Behura SK, Burns GW, Young SL, Demayo FJ, Spencer TE. Integrative analysis of the forkhead box A2 (FOXA2) cistrome for the human endometrium. FASEB J. 2019;33:8543–54.\\n11. Revel A, Achache H, Stevens J, Smith Y, Reich R. MicroRNAs are associated with human embryo implantation defects. Hum Reprod. 2011;26:2830–40.\\n12. Galliano D, Pellicer A. MicroRNA and implantation. Fertil Steril. 2014;101:1531–44.\\n13. Xu H, Zhou M, Cao Y, Zhang D, Han M, Gao X, et al. Genome‑wide analysis of long noncoding RNAs, microRNAs, and mRNAs forming a competing endogenous RNA network in repeated implantation failure. Gene. 2019;720: 144056.\\n14. Copley RR. The animal in the genome: comparative genomics and evolution. Philos Trans R Soc Lond B: Biol Sci. 2008;363:1453–61.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Breaking the ageing paradigm in endometrium: endometrial gene expression related to cilia and ageing hallmarks in women over 35 years.',\n",
       "  'doi': '37:762–76',\n",
       "  'citation': 'Devesa‑Peiro A, Sebastian‑Leon P , Parraga‑Leo A, Pellicer A, Diaz‑Gimeno P .'},\n",
       " {'text': 'Meta‑signature of human endometrial receptivity: a meta‑analysis and validation study of transcriptomic biomarkers.',\n",
       "  'doi': 'https:// doi. org/ 10. 1101/ gr. 240663.',\n",
       "  'citation': 'Altmäe S, Koel M, Võsa U, Adler P , Suhorutšenko M, Laisk‑Podar T, et al. Scient Rep. 2017;7:10077.'},\n",
       " {'text': 'Chapter 15 ‑ the molecular signature of the endometrial receptivity: research and clinical application. In: Horcajadas JA, Gosálvez J, editors. Reproductomics: the omics revolu‑tion and its impact on human reproductive medicine. Academic Press; 2018. p. 301–279.\\n\\n36. Díaz‑Gimeno P , Horcajadas JA, Martínez‑Conejero JA, Esteban FJ, Alamá P , Pellicer A, et al. A genomic diagnostic tool for human endometrial receptivity based on the transcriptomic signature. Fertil Steril. 2011;95:50.\\n\\n37. Kao LC, Tulac S, Lobo S, Imani B, Yang JP , Germeyer A, et al. Global gene profiling in human endometrium during the window of implantation. Endocrinology. 2002;143:2119–38.\\n\\n38. Ponnampalam AP , Weston GC, Trajstman AC, Susil B, Rogers PAW. Molecular classification of human endometrial cycle stages by transcriptional profiling. Mol Hum Reprod. 2004;10:879–93.\\n\\n39. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'lar classification of human endometrial cycle stages by transcriptional profiling. Mol Hum Reprod. 2004;10:879–93. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al. Oestrogen‑modulated gene expression in the human endometrium. Cell Mol Life Sci. 2005;62:239–50. Riesewijk A, Martín J, van Os R, Horcajadas JA, Polman J, Pellicer A, et al. Gene expression profiling of human endometrial receptivity on days LH+2 versus LH+7 by microarray technology. Mol Hum Reprod. 2003;9:253–64. Bastu E, Demiral I, Gunel T, Ulgen E, Gumusoglu E, Hosseini MK, et al. Potential marker pathways in the endometrium that may cause recurrent implantation failure. Reprod Sci. 2019;26:879–90. Bersinger NA, Wunder DM, Birkhäuser MH, Mueller MD. Gene expression in cultured endometrium from women with different outcomes follow‑following IVF. Mol Hum Reprod. 2008;14:475–84.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Page 14 of 14Parraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology           (2023) 21:84 \\n•\\n\\nfast, convenient online submission\\n •\\n\\nthorough peer review by experienced researchers in your ﬁeld\\n• \\n\\nrapid publication on acceptance\\n• \\n\\nsupport for research data, including large and complex data types\\n\\ngold Open Access which fosters wider collaboration and increased citations\\n\\nmaximum visibility for your research: over 100M website views per year •\\n  At BMC, research is always in progress. Learn more biomedcentral.com/submissions\\nReady to submit y our researc hReady to submit y our researc h  ?  Choose BMC and benefit fr om: ?  Choose BMC and benefit fr om: 43. Bhagwat SR, Chandrashekar DS, Kakar R, Davuluri S, Bajpai AK, Nayak S, et al. Endometrial receptivity: a revisit to functional genomics studies on human endometrium and creation of HGEx ‑ERdb. PLoS ONE. 2013;8: e58419.\\n44. Koot YEM, Van Hooff SR, Boomsma CM, Van Leenen D, Koerkamp MJAG, Goddijn M, et al. An endometrial gene expression signature accurately predicts recurrent implantation failure after IVF. Sci Rep. 2015;2016(6):1–12.\\n45. Lédée N, Munaut C, Aubert J, Sérazin V, Rahmati M, Chaouat GG, et al. Specific and extensive endometrial deregulation is present before conception in IVF/ICSI repeated implantation failures (IF) or recurrent miscarriages. J Pathol. 2011;225:554–64.\\n46. Pathare ADS, Zaveri K, Hinduja I. Downregulation of genes related to immune and inflammatory response in IVF implantation failure cases under controlled ovarian stimulation. Am J Reprod Immunol. 2017;78: e12679.\\n47. Shi C, Han HJ, Fan LJ, Guan J, Zheng XB, Chen X, et al. Diverse endometrial mRNA signatures during the window of implantation in patients with repeated implantation failure. Hum Fertil. 2018;21:183–94.\\n48. Messinisi IE. Ovarian feedback, mechanism of action and possible clinical implications. Hum Reprod Update. 2006;12:557–71.\\n49. Edwards RG. Implantation, interception and contraception. Hum Reprod (Oxford, England). 1994;9:985–95.\\n50. Young SL. Oestrogen and progesterone action on endometrium: a translational approach to understanding endometrial receptivity. Reprod Biomed Online. 2013;27:497–505.\\n51. Bhurke AS, Bagchi IC, Bagchi MK. Progesterone‑regulated endometrial factors controlling implantation. Am J Reprod Immunol. 2016;75:237–45.\\n52. Devesa‑Peiro A, Sebastian‑Leon P , Garcia‑ Garcia F, Arnau V, Aleman A, Pellicer A, et al. Uterine disorders affecting female fertility: what are the molecular functions altered in endometrium? Fertil Steril. 2020;113:1261–74.\\n53. von Grothusen C, Frisendahl C, Modhukur V, Lalitkumar PG, Peters M, Faridani OR, et al. Uterine fluid microRNAs are dysregulated in women with recurrent implantation failure. Hum Reprod. 2022;37:734–46.\\n54. Altmäe S, Martinez‑Conejero JA, Esteban FJ, Ruiz‑Alonso M, Stavreus‑Evers A, Horcajadas JA, et al. MicroRNAs miR‑30b, miR‑30d, and miR‑494',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'regulate human endometrial receptivity.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'Reprod Sci. 2013;20:308–17.'},\n",
       " {'text': 'stromal decidualization in vitro. PLoS ONE. 2012;7:41080. Liu D, Liang Y, Chen M, Yang F, Yao S. Knockdown of circ_0075503 sup ‑ presses cell migration and invasion by regulating miR‑15a‑5p and KLF12 in endometriosis. Mol Cell Biochem. 2021;476:3845–56. Chi RPA, Wang T, Adams N, Wu SP , Young SL, Spencer TE, et al. Human endometrial transcriptome and progesterone receptor cistrome reveal important pathways and epithelial regulators. J Clin Endocrinol Metab. 2020;105:1–21. Pawar S, Starosvetsky E, Orvis GD, Behringer RR, Bagchi IC, Bagchi MK. STAT3 regulates uterine epithelial remodeling and epithelial‑stromal crosstalk during implantation. Mol Endocrinol. 2013;27:1996–2012. Cheng YH, Imir A, Suzuki T, Fenkci V, Yilmaz B, Sasano H, et al. SP1 and SP3 mediate progesterone‑ dependent induction of the 17beta hydroxyster ‑ oid dehydrogenase type 2 gene in human endometrium. Biol Reprod. 2006;75:605–14. Sebastian‑Leon P , Devesa‑Peiro A, Aleman A, Parraga‑Leo A, Arnau V, Pellicer A, et al. Transcriptional changes through menstrual cycle reveal a global transcriptional derepression underlying the molecular mechanism involved in the window of implantation. Mol Hum Reprod. 2021. https:// doi. org/ 10. 1093/ molehr/ gaab0 27. Kim S, Yu NK, Kaang BK. CTCF as a multifunctional protein in genome regulation and gene expression. Exp Mol Med. 2015;47: e166. Shukla S, Kavak E, Gregory M, Imashimizu M, Shutinoski B, Kashlev M, et al. CTCF‑promoted RNA polymerase II pausing links DNA methylation to splicing. Nature. 2011;479:74–9. Phillips JE, Corces VGCTCF. Master weaver of the genome. Cell. 2009;137:1194–211. Guo J, Cao B, Xu X, Wu F, Zhu B. Novel CTCF mutations in Chinese patients with ovarian endometriosis. Mol Med Rep. 2018;18:1031–6. Geng J, Cui C, Yin Y, Zhao Y, Zhang C. LncRNA NEAT1 affects endome ‑ trial receptivity by regulating HOXA10 promoter activity. Cell Cycle. 2022;21:1932–44. Viger RS, Guittot SM, Anttonen M, Wilson DB, Heikinheimo M. Role of the GATA family of transcription factors in endocrine development, function, and disease. Mol Endocrinol. 2008;22:781. Xuan S, Sussel L. GATA4 and GATA6 regulate pancreatic endoderm identity through inhibition of hedgehog signaling. Development (Cam‑ bridge). 2016;143:780–6. Bernardi LA, Dyson MT, Tokunaga H, Sison C, Oral M, Robins JC, et al. The essential role of GATA6 in the activation of estrogen synthesis in endome ‑ triosis. Reprod Sci. 2019;26:60. Yin F, Herring BP . GATA‑6 can act as a positive or negative regula ‑ tor of smooth muscle‑specific gene expression *. J Biol Chem. 2005;280:4745–52. Zhai J, Luo G. GATA6‑induced FN1 activation promotes the proliferation, invasion and migration of oral squamous cell carcinoma cells. Mol Med ‑ Rep. 2022;25:102. Fullwood MJ, Liu MH, Pan YF, Liu J, Xu H, Mohamed YB, et al. An oestrogen‑receptor‑α‑bound human chromatin interactome. Nature. 2009;462:58–64.',\n",
       "  'doi': 'https://doi.org/10.1093/molehr/gaab027',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': '75. Lieberman‑Aiden E, Van Berkum NL, Williams L, Imakaev M, Ragoczy T, Telling A, et al. Comprehensive mapping of long range interactions reveals folding principles of the human genome. Science. 2009;326:289.\\n76. Birney E, Stamatoyannopoulos JA, Dutta A, Guigó R, Gingeras TR, Margulies EH, et al. Identification and analysis of functional elements in 1% of the human genome by the ENCODE pilot project. Nature. 2007;447:799–816.\\n77. Tajti F, Kuppe C, Antoranz A, Ibrahim MM, Kim H, Ceccarelli F, et al. A func‑\\n tional landscape of CKD entities from public transcriptomic data. Kidney Int Rep. 2020. https:// doi. org/ 10. 1016/j. ekir. 2019. 11. 005.\\n78. Henarejos‑ Castillo I, Devesa‑Peiro A, de Miguel‑ Gomez L, Sebastian‑Leon P , Romeu M, Aleman A, et al. Predicted COVID ‑19 molecular effects on \\n endometrium reveal key dysregulated genes and functions. Mol Hum Reprod. 2022.gaac035.\\n79. Barrett T, Wilhite SE, Ledoux P , Evangelista C, Kim IF, Tomashevsky M, et al.',\n",
       "  'doi': 'https:// doi. org/ 10. 1016/j. ekir. 2019. 11. 005',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'endometrium reveal key dysregulated genes and functions.',\n",
       "  'doi': 'gaac035',\n",
       "  'citation': 'UNKOWN'}]"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "my_json\n",
    "\n",
    "# my_json['author'] = \"parraga-leo\"\n",
    "# my_json['title'] = storage[0].metadata['title']\n",
    "# my_json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a7cfb13b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2455317/1728583740.py:10: LangChainDeprecationWarning: The class `HuggingFaceEmbeddings` was deprecated in LangChain 0.2.2 and will be removed in 1.0. An updated version of the class exists in the :class:`~langchain-huggingface package and should be used instead. To use it run `pip install -U :class:`~langchain-huggingface` and import as `from :class:`~langchain_huggingface import HuggingFaceEmbeddings``.\n",
      "  embedding_function = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")\n",
      "/home/aparraga/anaconda3/envs/RAG/lib/python3.9/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "# load the document and split it into pages\n",
    "loader = PyPDFLoader(\"/data/local/aparraga/Bioinformatician/RAG/Publications/Parraga-Leo2023.pdf\")\n",
    "pages = loader.load_and_split()\n",
    "\n",
    "# split it into chunks\n",
    "text_splitter = CharacterTextSplitter(chunk_size=1000, chunk_overlap=50)\n",
    "storage = text_splitter.split_documents(pages)\n",
    "\n",
    "# create the open-source embedding function\n",
    "embedding_function = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a0b29cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A total of 3,608 distinct genes from the 19 gene lists were associated with endometrial progression\n",
      "The lists’ regulation was significantly favoured by TFs (89% (17/19) of gene lists) and progesterone (47% (8 /19) of gene lists), rather than miRNAs (5% (1/19) of gene lists) or estrogen (0% (0/19) of gene lists), respectively (FDR < 0.05). Exceptionally, two gene lists that were previously associated with implantation failure and unexplained infertility were less hormone‑dependent, but primarily regulated by estrogen. Although endometrial progression genes were mainly targeted by hormones rather than non‑hormonal contributors (odds ratio = 91.94, FDR < 0.05), we identified 311 TFs and 595 miRNAs not previously associated with ovarian hormones. We highlight CTCF, GATA6, hsa‑miR‑15a‑5p, hsa‑miR‑218‑5p, hsa‑miR‑107, hsa‑miR‑103a‑3p, and hsa‑miR‑128‑3p, as overlapping novel master regulators of endometrial function. The gene expression changes of selected regulators through‑out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their\n",
      "out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their potential regulatory roles in the endometrium.\n",
      "This study revealed novel hormonal and non‑hormonal regulators and their relative contributions to endometrial progression and pathology, providing new leads for the potential causes of endometrial‑factor infertility. The endometrium is the innermost layer of the uterus, which undergoes dynamic histological, physiological and molecular changes that allow it to synchronize with embryo development, facilitate embryo implantation, and ultimately, establish a successful pregnancy. Implantation is a crucial and complex limiting step for conception which occurs between days 19 to 24 of a normal menstrual cycle. Consequently, this period of time is called the window of implantation (WOI) and is characterized by abrupt transcriptomic changes in the endometrial tissue. Alterations in either WOI establishment and endometrial progression may lead to implantation failures, which, along with biochemical miscarriage, account for more than 50% of pregnancy losses at pre-clinical stages. These alterations have broadly been classified as being related to a displaced WOI, caused by variable timing of endometrial progression, or a disrupted WOI, where impaired endometrial function prevents the establishment of an effective WOI. Initially, the WOI was postulated to mainly be regu -lated by the ovarian hormones (namely, estrogen and progesterone) acting via their respective nuclear receptors (i.e., ESR1/2 and PGR). However, evidence has shown that these receptors cooperate with other transcription factors (TFs) and co-regulators to mediate uterine physiology [8]. In fact, TFs (such as homeobox TFs, FOXA2, and KLF9) have recently been reported as key regulators for the establishment of endometrial receptivity [9, 10], and microRNAs (miRNAs) have been found to similarly intervene in this complex and multifactorial process as transcriptional regulators [11]. While the involvement of both types of regulators was widely reported in embryo implantation [12] and other infertility-related diseases, including recurrent implantation failure (RIF) [13], the complex interactions between the different types of regulators and their respective contributions to endometrial progression and function have not been investigated from a systemic point of view. This paradigm shift approaches genes and regulators in an integrative way, by considering how they interact with and coordinate each other to carry out cellular processes, rather than reinforcing the premise they work independently [14]. By more accurately reflecting the biological milieu, systemic approaches take gene-based discoveries to the next level.\n",
      "Previous studies of endometrial transcriptomics have been used to define the WOI, compare the endometrium of healthy women as their menstrual cycles progressed [4, 17–19]; address endometrial differences related to age [20] or the dysfunction of the WOI between patients with implantation failure and healthy controls [21]; and finally, correct for the menstrual cycle bias that can mask important biomarkers [22]. Notwithstanding, the candidate biomarkers of endometrial receptivity reported in each of the aforementioned studies overlap poorly [7]. Indeed, recent comparisons conducted by Sebastian-Leon and colleagues [7] found no congruities between 16 different reported gene lists of endometrial receptivity, and only fair to moderate functional agreements between some of the included signatures. Despite previous transcriptional regulation studies, there have been no holistic studies of endometrial receptivity.\n",
      "fair to moderate functional agreements between some of the included signatures. Despite previous transcriptional regulation studies, there have been no holistic studies of this regulatory process, which robustly analyse the drivers of transcription across a higher number of studies (that each searched for endometrial receptivity biomarkers) and map the relative contributions of each type of regulator to endometrial progression regulation (Ovarian hormones [i.e., estrogen, progesterone], TFs, miRNAs). Thus, the aim of this study was to utilize available genomic data regarding transcriptional regulators to identify overlapping mediators of endometrial transcriptional regulation (i.e., ovarian hormones, TFs, and miRNAs) and determine the relative contribution of each type of regulator among the previously reported gene signatures. These findings could help unveil the master regulators and principal type of endometrial regulation, paving the way for further research aimed at improving woman´s reproductive health. Annotating gene lists associated with endometrial progression and function. Endometrial progression and implantation failure gene lists [7] were retrieved and updated using a public data repository (i.e., Gene Expression Omnibus (GEO)).\n",
      "To study the hormonal regulation of endometrial progression, the Kyoto Encyclopaedia of Genes and Genomes (KEGG) [24], and Gene Ontology (GO) [25] databases were consulted for ovarian hormone-related genes, using relevant keywords (i.e., progesterone, estrogen, oestrogen and estradiol) and selecting those pathways and functions that contained them (Supplementary Fig. S1B). Genes associated with the obtained human KEGG pathways and GO functions were grouped according to whether they were unique to progesterone (P4 gene set) or estrogen (E2 gene set) or related to both hormones (P4 and E2 gene set). Gene targets of the nuclear progesterone (PGR) and estrogen (ESR1, ESR2) receptors were added to their corresponding gene sets using DoRothEA (Discriminant Regulon Expression Analysis) database [26], considering only manually-curated or ChiP-Seq experimentally-validated gene-TF relationships. Finally, the P4 and E2 gene sets were independently mapped to each gene list.\n",
      "Prioritizing key regulators in endometrial progression and function\n",
      "We built regulatory networks to systemically analyse endometrial regulation.\n",
      "We built regulatory networks to systemically analyse endometrial regulation. Nodes represented gene lists and regulators (miRNAs or TFs), while edges indicated significant enrichment among the lists and their corresponding regulators (false discovery rate (FDR) ≤ 0.05). To select the most influential regulators (i.e., those which targeted genes in most of our signatures), we studied the degree distribution of the networks (i.e., the number of gene lists regulated by each molecule) and prioritized the miRNAs and TFs which surpassed the relative maximum (i.e., 1.50 times the interquartile range (IQR)) number of relationships (gene list–regulators). All networks were built and analysed with Cytoscape version 3.7 [28] (Supplementary Fig. S1B). To evaluate how the miRNAs’ expression change in the endometrium throughout the menstrual cycle, we used the raw data from GSE44558 GEO dataset [29], deriving from 20 endometrial samples collected throughout the menstrual cycle: four in early-proliferative (EPF), four in late-proliferative (LPF), four in early-secretory (ESE), four in mid-secretory (MSE) and four in late-secretory (LSE). To evaluate how the TFs’ expression change in the endometrium throughout the menstrual cycle, we analysed the integrated endometrial dataset previously created by our group, and reported in [30], which compiled data from five prior publications (GSE98386, GSE29981, GSE4888, GSE119209 and GSE86491). This dataset included endometrial gene expression\n",
      "Of the 19 gene lists used for our analysis, eleven were obtained from published studies evaluating control patients throughout the menstrual cycle [4, 17–19, 34–40], and the other eight were derived from studies comparing patients with RIF or unexplained infertility to controls [21, 41–47] (Table 1). Unifying all the afore-\n",
      "mentioned signatures, we compiled 3,608 genes related to endometrial progression and function. Hormonal regulation of endometrial progression is largely driven by progesterone. We identified 7,540 and 698 genes related to estrogen and progesterone hormones, respectively. However, as determined by the relative contribution of each type of hormone within each gene list (Fig. 1A), 17/19 (89%) signatures favoured regulation by progesterone rather than estrogen. These differences were significant in 47% of the signatures (FDR < 0.05). Exceptionally, the genes reported by the Altmae2010 and Koot signatures, that were related to unexplained infertility and RIF, had a preference for estrogen regulation (Fig. 1A). With respect to the proportion of progesterone-related genes, Altmae2010 showed 2.1 times more estrogen-related genes while Koot showed 1.7 times more. More over, comparing the proportion of genes regulated by both hormones, between signatures revealed that Alt -\n",
      "We identified a total of 770 miRNAs and 555 TFs regulating at least one of the 3,608 genes reported in the 19 gene lists. Further, we observed that 29% of the miRNAs and 44% of the TFs were associated with the ovarian hormones (i.e., progesterone and/or estrogen). Significantly more genes were targeted by hormonally-influenced regulators rather than non-hormonal regulators (P -value < 0.01, odds ratio =  91.94). Nevertheless, we also identified 595 miRNAs (77.27%) and 311 TFs (56.04%) that were unrelated to progesterone and/or estrogen, highlighting a substantial proportion of hormonally-independent regulators of the menstrual cycle.\n",
      "Table 1 Characteristics of the 19 gene lists included in this study. Gene lists were divided according to their evaluated condition. Each signature was labelled using the name of first author, followed by the publication year in the case of duplicates. Endometrial staging was based on different variables including urinary LH, histological analysis, ultrasound evaluation, or the number of days since the hCG trigger. Participants included healthy controls and women with infertility‑related conditions. Various platforms were used to evaluate gene expression. The thresholds applied for gene refinement and the final number of genes per signature are also presented\n",
      "Abbreviations: Δ delta  Ct, adj-P-value Adjusted P‑value, C Control group, d Day of menstrual cycle, EPF Early proliferative, ESE Early secretory, |FC| Absolute fold‑change, FC Fold change, HRT Hormone replacement therapy, IF Implantation failure, LH Luteinizing hormone, LH + X X days post LH surge, LPF Late proliferative, LSE Late‑secretory, M Miscarriage, MPF Mid‑proliferative, MSE Mid‑secretory, n number of samples, N/A Not applicable, OP Ongoing pregnancy, PF Proliferative, PFP Proportion of false positives, PR Pre‑receptive, R Receptive, RIF Recurrent implantation failure, RRA Robust Rank Aggregation algorithm, SF Spontaneously fertile women, UI Women with unexplained infertility, vs versus\n",
      "Condition Gene signature Endometrial staging Participants Platform Threshold No. genes References\n",
      "Endometrial progres‑ sion\n",
      "Altmäe2017 N/A N/A N/A RRA 57 [34]\n",
      "Borthwick Urinary LH; Histologi‑ cally confirmed PF d9‑11 (n = 5) vs R LH + 6–8 (n = 5)\n",
      "Affymetrix Genechip Hu95A N/A 116 [18]\n",
      "Carrascosa N/A N/A Quantitative RT‑PCR Based on literature 187 [35]\n",
      "Carson Urinary LH; Histologically confirmed ESE LH + 2–4 (n = 3) vs MSE LH + 7–9 (n = 3)\n",
      "Affymetrix Genechip Hu95A |FC|> 2 695 [17]\n",
      "Diaz‑Gimeno Urinary LH PR LH + 1–5 (n = 15) vs R\n",
      "Agilent custom gene expression microarray |FC| > 2 234 [36]\n",
      "Kao Urinary LH; Histologically confirmed LPF d8‑10 (n = 4) vs MSE LH + 8‑10 (n = 7)\n",
      "Affymetrix Genechip Hu95A |FC|> 2 and P‑value <0.05 340 [37]\n",
      "Mirkin Urinary LH; Histological dating ESE LH + 3 (n = 3) vs MSE LH + 8 (n = 5)\n",
      "Affymetrix Genechip Hu95A |FC|>2 and adj‑P‑value < 0.05 105 [19]\n",
      "Ponnampalam Histological dating EPF (n = 5) MPF (n = 7) LPF (n = 3) ESE (n = 7) MSE (n = 8) LS (n = 7) Mense (n = 6)\n",
      "Custom Adj‑P‑value <0.05 306 [38]\n",
      "Punyadeera Histological dating Mense (n = 2) vs LPF (n=2)\n",
      "Affymetrix HG‑U133A FC and P‑value 50 [39]\n",
      "Riesewijk Urinary LH; Histological dating PR LH + 2 (n = 5) vs R LH + 7 (n = 5)\n",
      "Affymetrix Genechip Hu95A |FC|> 3 in at least four out of five women. 196 [40]\n",
      "Talbi Histological dating P (n = 6) vs ESE (n = 3) vs MSE (n = 8) vs LSE (n = 6)\n",
      "Affymetrix HG‑U133 Plus 2.0 |FC|> 1.5 and adj‑P‑value < 0.05 317 [4]\n",
      "Implantation failure Altmäe2010 Urinary LH UI (n = 4) vs C (n = 5) Whole Human Genome Oligo Microarray (Agilent Technologies) |FC|> 2 and PFP < 0.05 184 [21] Bastu 6‑7 days after LH surge IF (n = 24) vs Fertile women (n = 24) Agilent‑039494 SurePrint G3 Human GE v2 8x60K Log2(FC) ≥ 1 and adj‑P‑value ≤0.05 |FC|> 2 and PFP < 0.05 524 [41] Bersinger N/A IF (n = 3) vs M (n = 3) vs OP (n = 3) Affymetrix Human Genome U133A 2.0 Chips Not specified 93 [42] Bhagwat N/A PR vs R Multiple platforms Based on literature 177 [43] Koot Urinary LH RIF (n = 43) vs C (n = 72) Human whole genome gene expres‑sion microarrays V2 (Agilent, Belgium) Machine learning predictor 310 [44] Lédée Ultrasound evaluation IF (n = 30) vs Fertile women (n = 15) Affymetrix Human Genome U133 Plus 2.0 Array Adj‑P‑value <0.01 and Δ > 1.35 322 [45] Pathare 6‑7 days after hCG administration IF (n = 10) vs Healthy oocyte donors (n = 8) Illumina HumanHT‑12 V4.0 expression beadchip |FC|> 2 818 [46] Shi Urinary LH; Ultrasound evaluation\n",
      "IF (n = 10) vs Healthy oocyte donors (n = 8) Illumina HumanHT‑12 V4.0 expression beadchip |FC|> 2 818 [46] Shi Urinary LH; Ultrasound evaluation RIF (n = 12) vs C (n \n",
      "= 10) Agilent‑052909 CBC lncRNAmRNA V3 |FC|≥ 2 and adj‑P‑value ≤ 0.05 281 [47]\n",
      "Of the respective miRNAs and TFs targeting genes related to endometrial progression and function, 417 miRNAs and 467 TFs were significantly over- represented in at least one gene list (FDR < 0.05, Tables S3-S4). Further, most signatures were significantly governed by TFs rather than miRNAs (4 > odds ratio > 164, FDR < 0.05, 17 lists [89%]), with the exception of Koot’s signature that was governed by miRNAs and showed the opposite (odds ratio = 0.35, FDR < 0.05, Table S5). Eight out of nineteen gene lists (47%) had significant differences between the proportion of genes regulated by progesterone versus estrogen. (*FDR ≤ 0.05). Descriptive analysis of hormonal regulation, normalized by the size of each gene list. Gene lists clustered according to higher (blue) or lower (orange) contribution of hormonal regulation\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \n",
      "\n",
      "Hsa‑miR‑15a‑5p, hsa‑miR‑128‑3p, hsa‑miR‑218‑5p, \n",
      "hsa‑miR‑27b‑3p, hsa‑miR‑107, hsa‑miR‑424‑5p, \n",
      "hsa‑miR‑195‑5p, hsa‑miR‑103a‑3p, has‑let‑7b‑5p and hsa‑miR‑22‑3p are key players in the regulation of endometrial progression and function\n",
      "\n",
      "To bridge the gap of endometrial-based evidence in DoRothEA and TarBase databases, we aimed to identify the common miRNAs that mediate gene expression related to endometrial progression and function. We evaluated the number of endometrial gene lists that each miRNA controlled (Fig. 2A). This strategy highlighted the most influential miRNAs that were commonly found across the endometrial gene lists. Notably, 85.80% of miRNAs regulated genes from one to three signatures, and only 59 miRNAs (14.20%) were associated with more than three signatures. Ultimately, 24 miRNAs were prioritized for regulating genes of at least six signatures.\n",
      "\n",
      "Next, we validated the gene expression trends for these 24 prioritized miRNAs in an independent endometrial gene expression dataset (GSE44730) (Fig. 2B). This dataset included the miRNA expression in endometrial biopsies from 20 healthy women in different phases of the menstrual cycle. We found that two of our prioritized miRNAs were not expressed in the endometrial tissue, and only 10 miRNAs (i.e., hsa-miR-15a-5p, hsa-miR-128-3p, hsa-miR-218-5p, hsa-miR-27b-3p, hsa-miR-107, hsa-miR-424-5p, hsa-miR-195-5p, hsa-miR-103a-3p, has-let-7b-5p and hsa-miR-22-3p) showed significant changes across the menstrual cycle (Fig. 2B). Of these, only the expression of hsa-miR-22-3p significantly increased from ESE to MSE stages (FDR = 0.02, Fig. 2B), highlighting the potential inhibitory role of this miRNA in endometrial function. The expression of the remaining miRNAs decreased during the MSE phase.\n",
      "\n",
      "CTCF and GATA6 as distinguished transcriptional regulators of the human menstrual cycle\n",
      "\n",
      "We repeated the regulation network analysis with the TFs, to identify the prominent TFs commonly regulating the genes related to endometrial progression and function in the reported signatures. We observed 78.80% of the TFs regulated genes in up to 10 signatures, and only 99 TFs (21.20%) regulated genes in over 10 (Fig. 3A). We highlighted CTCF (a CCCTC-binding factor, which functions as a transcriptomic repressor) as the TF with the most influential regulation of endometrial progression and function across 95% of studies (18 gene lists, excluding that of Punyadeera, Fig. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53. Notably, GATA6 was selected due to novel association with endometrial progression and function.\n",
      "To infer the expression of CTCF and GATA6  in \n",
      "endometrial tissue throughout the menstrual cycle, \n",
      "we analysed their expression using a relevant endo - \n",
      "mentrial dataset recently created by our group [30] \n",
      "(Table S1 ). We observed that, relative to the global \n",
      "expression of endometrial genes, CTCF was broadly \n",
      "expressed while GATA6  was moderately expressed. \n",
      "Indeed, CTCF was significantly down-regulated \n",
      "between the PF and ESE phases (FDR = 2.10E-03)  \n",
      "and ESE to the MSE phase (FDR = 2.10E-04), while \n",
      "GATA6 expression increased from the PF to the \n",
      "MSE phase (FDR = 1.70E-11) and from the ESE  \n",
      "to MSE phase (FDR = 9.80E-15). These findings \n",
      "supported the role of these TFs at the beginning \n",
      "of the menstrual cycle and during the MSE phase \n",
      "(Fig. 3 B). \n",
      "These in-silico  validations were then prospec - \n",
      "tively corroborated by RT-qPCR analysis in an inde - \n",
      "pendent sample set of 20 endometrial biopsies. We \n",
      "observed a similar expression profile of CTCF and \n",
      "GATA6 throughout the menstrual cycle, with signifi-\n",
      "\n",
      "We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with significant changes between the endometrial phases. Notably, CTCF expression was decreased between the PF/ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C). The prioritized miRNAs included has‑miR‑16‑5p, has‑miR‑138‑5p, has‑miR‑21‑3p, has‑miR‑205‑5p, has‑miR‑27a‑5p, has‑miR‑15a‑5p, has‑miR‑155‑5p, has‑miR‑129–2‑3p, has‑miR‑147a, has‑miR‑146a‑5p, has‑miR‑107, has‑miR‑424‑5p, has‑miR‑203a‑3p, has‑miR‑195‑5p, has‑miR‑124‑3p, has‑miR‑128‑3p, has‑miR‑22‑3p, has‑miR‑27b‑3p, has‑let‑7b‑5p, has‑miR‑1343‑3p, has‑miR‑1‑3p, has‑miR‑103a‑3p, has‑miR‑23b‑3p, has‑miR‑218‑5p. Significant changes between phases are denoted with asterisks (* FDR ≤ 0.05, ** FDR ≤ 0.01.\n",
      "Fig. 2 (See legend on previous page.)\n",
      "Page 9 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \n",
      "\n",
      "Fig. 3 Transcription factors universally regulating endometrial progression and function and their expression throughout the menstrual cycle. A Regulatory network of TFs forming relationships with numerous gene lists (2–18). TFs targeting specific gene lists were placed in the center of the network. CTCF stood out as the only TF that targeted genes belonging to 18/19 (95%) different signatures. B In-silico analysis of CTCF and GATA6 expression throughout the menstrual cycle. Significant changes between phases are marked with asterisks (**FDR ≤ 0.01, ***FDR ≤ 0.001). C RT‑qPCR validation. Bar plot shows the relative mRNA expression of CTCF and GATA6 in endometrial biopsies collected throughout the menstrual cycle. Data is presented as a mean ± SD. Significant differences of CTCF (*FDR < 0.05) and GATA6 (***FDR < 0.001) expression were observed between proliferative/early‑secretory (PF/ESE) and mid‑secretory/late‑secretory phases (MSE/LSE). Abbreviations in (B and C): PF, Proliferative; ESE, early‑secretory; MSE, mid‑secretory; LSE, late‑secretory; SD, standard deviation\n",
      "This data-driven approach exposed the common transcriptional regulators among 19 studies who proposed variable biomarkers of endometrial progression and function. In this study, we focused on understanding the relative contribution of both the hormonal and non-hormonal regulation, from an alternative holistic perspective. We applied data-driven hypothesis research that, unlike the traditional scientific method, allowed us to generate new hypotheses based on all available biologically-relevant knowledge [16], observing the molecular relationships from a wider scale view. Besides, some traditional molecular procedures such as PCRs are also performed to corroborate the insight uncovered with this approach. We highlighted a larger influence of progesterone-related genes and TFs in endometrial progression and observed a larger contribution of miRNAs and estrogen-related genes in endometrial pathology. Furthermore, we unveiled CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-miR-103a-3p and hsa-miR-128-3p as master regulators in the endometrium, and these findings were reinforced by the dynamic changes in their expression throughout the menstrual cycle (dually validated with in-silico and prospective analyses). The menstrual cycle is perceived to be under tight regulation by the ovarian hormones, progesterone and estrogen [48]. Supporting the role of progesterone in the secretory endometrium [49], and reinforcing its indispensable role in endometrial regulation [50, 51], we found progesterone and its related genes had a larger influence in menstrual cycle progression than estrogen. As the RIF-related gene lists reported by Altmae2010 and Koot were generally controlled by estrogen, this further suggested that while progesterone could be crucial for a suitable endometrial progression and mid-secretory phase acquisition, estrogenic irregularities may be more conducive to endometrial function and molecular pathology. Indeed, this hypothesis corroborated previous reports of most uterine disorders being estrogen-dependent [44, 52]. Supporting the findings from another recent study from our group, where the Altmae2010 and Koot signatures were classified as highly-predictive signatures to identify endometrial pathology in comparison with a prediction model that detected endometrial progression [7 ]. This study distinguished these signatures (related to implantation failure and unexplained infertility) based on the magnitude of their miRNA and estrogen-regulation, which contrasted with the prominent influence of the TFs over the remaining signatures more related to endometrial progression. All these facts reinforce the hypothesis that implantation failure is mainly caused by impaired miRNA expression [53].\n",
      "This study transcended the simplistic paradigm of the bihormonal regulation of endometrial function by identifying a plethora of mediators, including 595 miRNAs and 311 TFs, that were previously unrelated to these ovarian hormones. These findings set the foundation for new discoveries, revealing alternative pathways and new actors in the regulation of the human endometrium, and deepening our understanding of the complex regulatory mechanisms behind endometrial progression and unexplained infertility. MicroRNAs are well known to be involved in embryo implantation and receptivity control. Using network analysis, we unveiled the 10 common miRNAs regulating endometrial progression and function across published biomarker signatures (hsa-miR-22-3p, hsa-miR-107, hsa-miR-103a-3p, hsa-miR-128-3p, hsa-miR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, hsa-miR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p).\n",
      "Their significant gene expression changes in the secre - \n",
      "tory endometrium support their potential role in the \n",
      "endometrial regulation through an inhibition during \n",
      "the WOI. Notably, hsa-miR-424-5p, hsa-miR-27b-3p \n",
      "and hsa-miR-195-5p have been previously associated \n",
      "with RIF [11, 54, 56], while has-let-7b-5p was associ - \n",
      "ated with human endometrial receptivity [57], and \n",
      "hsa-miR-22-3p was overexpressed in a decidualization \n",
      "model [58]. Moreover, hsa-miR-15a-5p was previously \n",
      "associated with endometriosis [59]. Notwithstanding, \n",
      "this study is the first to associate hsa-miR-15a-5p, hsa- \n",
      "miR-218-5p, hsa-miR-103a-3p, hsa-miR-107, and hsa- \n",
      "miR-128-3p to endometrial progression and functional \n",
      "regulation in humans.\n",
      "Page 11 of 14\n",
      "Parraaga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \n",
      "\n",
      "endometrial progression and function were beyond the scope of this project and merit further investigation. CTCF is a conserved zinc finger protein whose regulatory functions are well characterized throughout the human body [64]. CTCF acts as a transcriptional repressor in RNA polymerase II (Pol II) pausing and imprinting and X-chromosome inactivation [65], as well as an insulator, blocking the interaction between enhancers and the promoters of neighbouring genes [66]. Considering these attributed functions and its variable expression throughout the menstrual cycle, we propose that CTCF exerts an inhibitory role in endometrial tissue during the PF phase of the menstrual cycle. With the significant downregulation of CTCF in the secretory endometrium, dually validated in-silico and experimentally herein, its inhibited genes would be derepressed and become transcriptionally active during the WOI. This interpretation supports previous findings from our group demonstrating that, during the WOI, a global transcriptional derepression may be required for implantation and early embryo development [63]. Transcriptional derepression has been associated with multiple human disease states and should be investigated further within the context of endometrial-factor infertility. Despite previous associations of CTCF with endometriosis [67], its implication in endometrial progression and function has not been directly proposed until now. Nevertheless, the role of CTCF in endometrial receptivity has been revealed through its interaction with HOXA10, a gene mainly expressed in endometrium involved in functions such as endometrial proliferation and differentiation, the formation of pinopodes or embryo implantation [68]. Indeed, this study proposed that an overexpression of CTCF can lead to a drop in HOXA10 expression, affecting endometrial proliferation and endometrial function and corroborating the fact that CTCF must decrease its expression as we can observe in our findings. GATA6, the second prioritized TF that overlapped across the gene lists, is a transcription factor belonging to the GATA family, a highly conserved family of six zinc finger proteins [69]. Besides its essential role in embryonic development [70], GATA6 plays a key role in endometriosis and regulates steroidogenic genes [71]. Together with evidence supporting both an activating and inhibitory role for GATA6 [72, 73], the significant changes we observed in its expression across the menstrual cycle reflect its presumed involvement in the endometrium and suggest an activating role in endometrial progression and endometrial receptivity acquisition. Like CTCF, this is the first time that GATA6 is proposed as a key TF in endometrial progression and function.\n",
      "endometrial regulation remain elusive, emerging experimental technologies such as High-throughput Chromosome Conformation Capture (Hi-C) or chromatin interaction analysis with paired-end tags (ChIA-PET) that respectively study general chromatin interactions or chromatin interactions involving a specific protein [74, 75] may improve our understanding of genetic regulation in the future. Nevertheless, as collaborative efforts unveil new regulators and gene targets in humans (such as the ENCODE project [76]), the complexities of transcriptional regulation can be exposed by in-silico analyses like those described herein, and we can gain a better understanding of the processes involved in menstrual cycle progression and endometrial competence. Furthermore, it should be noted that results of gene expression studies are influenced by different variables, such as the experimental platforms employed, study designs, or patients’ characteristics of each independent\n",
      "expression studies are influenced by different variables, such as the experimental platforms employed, study designs, or patients’ characteristics of each independent study, adding variability [30, 77, 78]. Despite this fact, we have found that our holistic approach overcomes this undesired variability effect by finding common upstream master regulators across studies and predicting new relationships between hormones, TFs and miRNAs. This approach also provides the basis for future single-molecule studies that aim to elucidate new regulatory pathways in endometrial progression and function. Our results lay the foundation for further molecular studies that can validate the function(s) of the prevailing endometrial regulators we prioritized. The study shows for the first time, the relative contribution of estrogens, progesterone, TFs and miRNAs in endometrial function and progression. Endometrial progression is mainly influenced by progesterone-related genes and TFs, whereas miRNAs and estrogen-related genes play a larger role in endometrial pathology. Moreover, we highlight novel common transcriptional regulators such as CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-miR-103a-3p, and hsa-miR-128-3p across 19 studies that propose biomarkers in endometrial regulation. These results reveal the molecular mechanism underlying endometrial regulation and lay the foundation for the development of targeted therapies for patients with endometrial-factor infertility.\n",
      "We reviewed 19 publications related to endometrial progression and function, extracted the gene signatures from the original publication, and created corresponding gene lists that we could annotate. We performed different functional analyses based on the type of regulator (i.e., ovarian hormones, transcription factors and miRNAs). We used online databases to obtain the information of the regulatory process, at the hormonal (Gene Ontology, Kyoto Encyclopedia of Genes and Genomes, Dorothea), transcription factor (Dorothea) and miRNA (Tarbase) level and analysed the results accordingly. Finally, we employed an integrative analysis and prioritized the most relevant regulators for subsequent validation. TF, Transcription factors; miRNA, microRNA.\n",
      "Additional file 5: Supplementary Table S4. Results of TFs enrichment. Excel file with the over‑representation analysis results of the transcription factors (TFs). Each sheet contains the results of a specific list of genes. In each sheet, the following information is included: ID, gene symbol of the TF; GeneRatio, the proportion of TF targets in the gene list; BgRatio, the proportion of TF targets in the whole genome; OddsRatio; the odd ratios for each TF, pvalue; the P‑value obtained for each TF; padjust, P‑value adjusted by FDR for each TF; geneID, the gene symbol ID of the TF targets. Additional file 6: Supplementary Table S5. Relative contribution of each regulator type (TFs or miRNAs) in each gene list. A Fisher’s test was performed to calculate the proportional differences between transcription factors (TFs) and microRNAs (miRNAs). Resulting P‑values were adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene\n",
      "Resulting P‑values were adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene lists were mainly regulated by TFs, whereas odds ratios < 1 indicate a higher regulation by miRNAs. Odds Ratio with a value of “Inf” indicate that miRNAs were not found for this gene list. Percentages indicate the proportion between the significant regulators in each gene signature and regulators with at least one target gene in the corresponding signature. (***FDR ≤0.001).\n",
      "The data used to analyze the TF and miRNA expression throughout the men‑\n",
      "tstrual cycle and in infertility are available in Gene Expression Omnibus, with \n",
      "IDs GSE98386, GSE29981, GSE4888, GSE119209, GSE86491, and GSE44558. The raw endometrial gene expression data (from previous studies) was downloaded from the publicly‑available functional genomic dataset repository Gene Expression Omnibus (GEO) [79] for curation/processing. In accordance with GEO policies, all patient data was anonymized and encrypted, and therefore an additional institutional review board approval was not required for our in-silico validation of regulator gene expression changes in endometrial tissue.\n",
      "tional institutional review board approval was not required for our in-silico validation of regulator gene expression changes in endometrial tissue. On the other hand, for the experimental validation of gene expression changes in endometrial biopsies, all individuals gave written informed consent for endometrial gene expression evaluation prior to participating (Ethics Committee of the Instituto Valenciano de Infertilidad [IVI Valencia], Valencia, Spain [1810‑FIVI‑066‑PD/1706‑FIVI‑048‑PD]).\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \n",
      "\n",
      "Author details\n",
      "1 IVIRMA Global Research Alliance, IVI Foundation, Instituto de Investigación Sanitaria La Fe (IIS La Fe), Av. Fernando Abril Martorell 106, Torre A, Planta 1ª, 46026 Valencia, Valencia, Spain. 2 Department of Pediatrics, Obstetrics and Gynaecology, Universidad de Valencia, Av. Blasco Ibáñez 15, 46010 Valen‑cia, Valencia, Spain. 3 IVIRMA Global Research Alliance, IVIRMA Valencia, Plaza de La Policia Local 3, 46015 Valencia, Spain.\n",
      "Received: 21 June 2023 Accepted: 22 August 2023\n",
      "\n",
      "References\n",
      "1. Strowitzki T, Germeyer A, Popovici R, von Wolff M. The human endometrium as a fertility-determining factor. Hum Reprod Update. 2006;12:617–30.\n",
      "2. Cakmak H, Taylor HS. Implantation failure: Molecular mechanisms and clinical treatment. Hum Reprod Update. 2011;17:242–53.\n",
      "3. Lessey BA, Young SL. What exactly is endometrial receptivity? Fertil Steril. 2019;111:611–7.\n",
      "4. Talbi S, Hamilton AE, Vo KC, Tulac S, Overgaard MT, Dosiou C, et al. Molecular phenotyping of human endometrium distinguishes menstrual cycle phases and underlying biological processes in normo‑ovulatory women. Endocrinology. 2006;147:1097–121.\n",
      "5. Somigliana E, Vigano P , Busnelli A, Paffoni A, Vegetti W, Vercellini P . Repeated implantation failure at the crossroad between statistics, clinics and over‑diagnosis. Reprod Biomed Online. 2018;36:32–8.\n",
      "6. Craciunas L, Gallos I, Chu J, Bourne T, Quenby S, Brosens JJ, et al. Conventional and modern markers of endometrial receptivity: a systematic review and meta‑analysis. Hum Reprod Update. 2019;25:202–23.\n",
      "7. Sebastian‑Leon P , Garrido N, Remohí J, Pellicer A, Diaz‑Gimeno P . Asynchronous and pathological windows of implantation: Two causes of recurrent implantation failure. Hum Reprod. 2018;33:626–35.\n",
      "8. Vasquez YM, DeMayo FJ. Role of nuclear receptors in blastocyst implantation. Semin Cell Dev Biol. 2013;24:724–35.\n",
      "9. Wang H, Zhang S, Lin H, Kong S, Wang S, Wang H, et al. Physiological and molecular determinants of embryo implantation. Mol Aspects Med. 2013;34:939–80.\n",
      "10. Kelleher AM, Behura SK, Burns GW, Young SL, Demayo FJ, Spencer TE. Integrative analysis of the forkhead box A2 (FOXA2) cistrome for the human endometrium. FASEB J. 2019;33:8543–54.\n",
      "11. Revel A, Achache H, Stevens J, Smith Y, Reich R. MicroRNAs are associated with human embryo implantation defects. Hum Reprod. 2011;26:2830–40.\n",
      "12. Galliano D, Pellicer A. MicroRNA and implantation. Fertil Steril. 2014;101:1531–44.\n",
      "13. Xu H, Zhou M, Cao Y, Zhang D, Han M, Gao X, et al. Genome‑wide analysis of long noncoding RNAs, microRNAs, and mRNAs forming a competing endogenous RNA network in repeated implantation failure. Gene. 2019;720: 144056.\n",
      "14. Copley RR. The animal in the genome: comparative genomics and evolution. Philos Trans R Soc Lond B: Biol Sci. 2008;363:1453–61.\n",
      "Breaking the ageing paradigm in endometrium: endometrial gene expression related to cilia and ageing hallmarks in women over 35 years.\n",
      "Meta‑signature of human endometrial receptivity: a meta‑analysis and validation study of transcriptomic biomarkers.\n",
      "Chapter 15 ‑ the molecular signature of the endometrial receptivity: research and clinical application. In: Horcajadas JA, Gosálvez J, editors. Reproductomics: the omics revolu‑tion and its impact on human reproductive medicine. Academic Press; 2018. p. 301–279.\n",
      "\n",
      "36. Díaz‑Gimeno P , Horcajadas JA, Martínez‑Conejero JA, Esteban FJ, Alamá P , Pellicer A, et al. A genomic diagnostic tool for human endometrial receptivity based on the transcriptomic signature. Fertil Steril. 2011;95:50.\n",
      "\n",
      "37. Kao LC, Tulac S, Lobo S, Imani B, Yang JP , Germeyer A, et al. Global gene profiling in human endometrium during the window of implantation. Endocrinology. 2002;143:2119–38.\n",
      "\n",
      "38. Ponnampalam AP , Weston GC, Trajstman AC, Susil B, Rogers PAW. Molecular classification of human endometrial cycle stages by transcriptional profiling. Mol Hum Reprod. 2004;10:879–93.\n",
      "\n",
      "39. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al.\n",
      "lar classification of human endometrial cycle stages by transcriptional profiling. Mol Hum Reprod. 2004;10:879–93. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al. Oestrogen‑modulated gene expression in the human endometrium. Cell Mol Life Sci. 2005;62:239–50. Riesewijk A, Martín J, van Os R, Horcajadas JA, Polman J, Pellicer A, et al. Gene expression profiling of human endometrial receptivity on days LH+2 versus LH+7 by microarray technology. Mol Hum Reprod. 2003;9:253–64. Bastu E, Demiral I, Gunel T, Ulgen E, Gumusoglu E, Hosseini MK, et al. Potential marker pathways in the endometrium that may cause recurrent implantation failure. Reprod Sci. 2019;26:879–90. Bersinger NA, Wunder DM, Birkhäuser MH, Mueller MD. Gene expression in cultured endometrium from women with different outcomes follow‑following IVF. Mol Hum Reprod. 2008;14:475–84.\n",
      "Page 14 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "•\n",
      "\n",
      "fast, convenient online submission\n",
      " •\n",
      "\n",
      "thorough peer review by experienced researchers in your ﬁeld\n",
      "• \n",
      "\n",
      "rapid publication on acceptance\n",
      "• \n",
      "\n",
      "support for research data, including large and complex data types\n",
      "\n",
      "gold Open Access which fosters wider collaboration and increased citations\n",
      "\n",
      "maximum visibility for your research: over 100M website views per year •\n",
      "  At BMC, research is always in progress. Learn more biomedcentral.com/submissions\n",
      "Ready to submit y our researc hReady to submit y our researc h  ?  Choose BMC and benefit fr om: ?  Choose BMC and benefit fr om: 43. Bhagwat SR, Chandrashekar DS, Kakar R, Davuluri S, Bajpai AK, Nayak S, et al. Endometrial receptivity: a revisit to functional genomics studies on human endometrium and creation of HGEx ‑ERdb. PLoS ONE. 2013;8: e58419.\n",
      "44. Koot YEM, Van Hooff SR, Boomsma CM, Van Leenen D, Koerkamp MJAG, Goddijn M, et al. An endometrial gene expression signature accurately predicts recurrent implantation failure after IVF. Sci Rep. 2015;2016(6):1–12.\n",
      "45. Lédée N, Munaut C, Aubert J, Sérazin V, Rahmati M, Chaouat GG, et al. Specific and extensive endometrial deregulation is present before conception in IVF/ICSI repeated implantation failures (IF) or recurrent miscarriages. J Pathol. 2011;225:554–64.\n",
      "46. Pathare ADS, Zaveri K, Hinduja I. Downregulation of genes related to immune and inflammatory response in IVF implantation failure cases under controlled ovarian stimulation. Am J Reprod Immunol. 2017;78: e12679.\n",
      "47. Shi C, Han HJ, Fan LJ, Guan J, Zheng XB, Chen X, et al. Diverse endometrial mRNA signatures during the window of implantation in patients with repeated implantation failure. Hum Fertil. 2018;21:183–94.\n",
      "48. Messinisi IE. Ovarian feedback, mechanism of action and possible clinical implications. Hum Reprod Update. 2006;12:557–71.\n",
      "49. Edwards RG. Implantation, interception and contraception. Hum Reprod (Oxford, England). 1994;9:985–95.\n",
      "50. Young SL. Oestrogen and progesterone action on endometrium: a translational approach to understanding endometrial receptivity. Reprod Biomed Online. 2013;27:497–505.\n",
      "51. Bhurke AS, Bagchi IC, Bagchi MK. Progesterone‑regulated endometrial factors controlling implantation. Am J Reprod Immunol. 2016;75:237–45.\n",
      "52. Devesa‑Peiro A, Sebastian‑Leon P , Garcia‑ Garcia F, Arnau V, Aleman A, Pellicer A, et al. Uterine disorders affecting female fertility: what are the molecular functions altered in endometrium? Fertil Steril. 2020;113:1261–74.\n",
      "53. von Grothusen C, Frisendahl C, Modhukur V, Lalitkumar PG, Peters M, Faridani OR, et al. Uterine fluid microRNAs are dysregulated in women with recurrent implantation failure. Hum Reprod. 2022;37:734–46.\n",
      "54. Altmäe S, Martinez‑Conejero JA, Esteban FJ, Ruiz‑Alonso M, Stavreus‑Evers A, Horcajadas JA, et al. MicroRNAs miR‑30b, miR‑30d, and miR‑494\n",
      "regulate human endometrial receptivity.\n",
      "stromal decidualization in vitro. PLoS ONE. 2012;7:41080. Liu D, Liang Y, Chen M, Yang F, Yao S. Knockdown of circ_0075503 sup ‑ presses cell migration and invasion by regulating miR‑15a‑5p and KLF12 in endometriosis. Mol Cell Biochem. 2021;476:3845–56. Chi RPA, Wang T, Adams N, Wu SP , Young SL, Spencer TE, et al. Human endometrial transcriptome and progesterone receptor cistrome reveal important pathways and epithelial regulators. J Clin Endocrinol Metab. 2020;105:1–21. Pawar S, Starosvetsky E, Orvis GD, Behringer RR, Bagchi IC, Bagchi MK. STAT3 regulates uterine epithelial remodeling and epithelial‑stromal crosstalk during implantation. Mol Endocrinol. 2013;27:1996–2012. Cheng YH, Imir A, Suzuki T, Fenkci V, Yilmaz B, Sasano H, et al. SP1 and SP3 mediate progesterone‑ dependent induction of the 17beta hydroxyster ‑ oid dehydrogenase type 2 gene in human endometrium. Biol Reprod. 2006;75:605–14. Sebastian‑Leon P , Devesa‑Peiro A, Aleman A, Parraga‑Leo A, Arnau V, Pellicer A, et al. Transcriptional changes through menstrual cycle reveal a global transcriptional derepression underlying the molecular mechanism involved in the window of implantation. Mol Hum Reprod. 2021. https:// doi. org/ 10. 1093/ molehr/ gaab0 27. Kim S, Yu NK, Kaang BK. CTCF as a multifunctional protein in genome regulation and gene expression. Exp Mol Med. 2015;47: e166. Shukla S, Kavak E, Gregory M, Imashimizu M, Shutinoski B, Kashlev M, et al. CTCF‑promoted RNA polymerase II pausing links DNA methylation to splicing. Nature. 2011;479:74–9. Phillips JE, Corces VGCTCF. Master weaver of the genome. Cell. 2009;137:1194–211. Guo J, Cao B, Xu X, Wu F, Zhu B. Novel CTCF mutations in Chinese patients with ovarian endometriosis. Mol Med Rep. 2018;18:1031–6. Geng J, Cui C, Yin Y, Zhao Y, Zhang C. LncRNA NEAT1 affects endome ‑ trial receptivity by regulating HOXA10 promoter activity. Cell Cycle. 2022;21:1932–44. Viger RS, Guittot SM, Anttonen M, Wilson DB, Heikinheimo M. Role of the GATA family of transcription factors in endocrine development, function, and disease. Mol Endocrinol. 2008;22:781. Xuan S, Sussel L. GATA4 and GATA6 regulate pancreatic endoderm identity through inhibition of hedgehog signaling. Development (Cam‑ bridge). 2016;143:780–6. Bernardi LA, Dyson MT, Tokunaga H, Sison C, Oral M, Robins JC, et al. The essential role of GATA6 in the activation of estrogen synthesis in endome ‑ triosis. Reprod Sci. 2019;26:60. Yin F, Herring BP . GATA‑6 can act as a positive or negative regula ‑ tor of smooth muscle‑specific gene expression *. J Biol Chem. 2005;280:4745–52. Zhai J, Luo G. GATA6‑induced FN1 activation promotes the proliferation, invasion and migration of oral squamous cell carcinoma cells. Mol Med ‑ Rep. 2022;25:102. Fullwood MJ, Liu MH, Pan YF, Liu J, Xu H, Mohamed YB, et al. An oestrogen‑receptor‑α‑bound human chromatin interactome. Nature. 2009;462:58–64.\n",
      "75. Lieberman‑Aiden E, Van Berkum NL, Williams L, Imakaev M, Ragoczy T, Telling A, et al. Comprehensive mapping of long range interactions reveals folding principles of the human genome. Science. 2009;326:289.\n",
      "76. Birney E, Stamatoyannopoulos JA, Dutta A, Guigó R, Gingeras TR, Margulies EH, et al. Identification and analysis of functional elements in 1% of the human genome by the ENCODE pilot project. Nature. 2007;447:799–816.\n",
      "77. Tajti F, Kuppe C, Antoranz A, Ibrahim MM, Kim H, Ceccarelli F, et al. A func‑\n",
      " tional landscape of CKD entities from public transcriptomic data. Kidney Int Rep. 2020. https:// doi. org/ 10. 1016/j. ekir. 2019. 11. 005.\n",
      "78. Henarejos‑ Castillo I, Devesa‑Peiro A, de Miguel‑ Gomez L, Sebastian‑Leon P , Romeu M, Aleman A, et al. Predicted COVID ‑19 molecular effects on \n",
      " endometrium reveal key dysregulated genes and functions. Mol Hum Reprod. 2022.gaac035.\n",
      "79. Barrett T, Wilhite SE, Ledoux P , Evangelista C, Kim IF, Tomashevsky M, et al.\n",
      "endometrium reveal key dysregulated genes and functions.\n"
     ]
    }
   ],
   "source": [
    "# create the chroma client\n",
    "import uuid\n",
    "import chromadb\n",
    "from chromadb.config import Settings\n",
    "\n",
    "# create the open-source embedding function\n",
    "embedding_function = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")\n",
    "\n",
    "client = chromadb.HttpClient(host='localhost', port=7000, settings=Settings(allow_reset=True))\n",
    "# client.list_collections()\n",
    "# client.reset()  # resets the database\n",
    "collection = client.get_or_create_collection(\"tfm_APL\")\n",
    "# collection = client.get_collection('tfm')\n",
    "\n",
    "for i, doc in enumerate(my_json): #storage:\n",
    "    print(doc['text'])\n",
    "    collection.add(\n",
    "        ids=[str(uuid.uuid1())], metadatas=storage[i].metadata, documents=str(doc['text'])\n",
    "    )\n",
    "\n",
    "# tell LangChain to use our client and collection name\n",
    "db = Chroma(\n",
    "    client=client,\n",
    "    collection_name=\"tfm\",\n",
    "    embedding_function=embedding_function,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bec23937",
   "metadata": {},
   "outputs": [],
   "source": [
    "# from ollama import Client\n",
    "# client = Client(host='http://localhost:11434')\n",
    "# stream = client.chat(model='gemma3:12b', messages=[\n",
    "# {\"role\": \"system\", \"content\": get_system_message_rag(fullcontent)},            \n",
    "# {\"role\": \"user\", \"content\": get_ques_response_prompt(query)}\n",
    "# ],stream=True)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "RAG",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
