{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "18772ea0",
   "metadata": {},
   "source": [
    "# Prepare documents for RAG"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "579d11f6",
   "metadata": {},
   "source": [
    "## Libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c3b57a69",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/antonioparragaleo/miniconda3/envs/RAG/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "import langchain_community\n",
    "import langchain_text_splitters\n",
    "from langchain_community.document_loaders import PyPDFLoader, pdf\n",
    "from langchain_community.embeddings.sentence_transformer import (\n",
    "    SentenceTransformerEmbeddings,\n",
    ")\n",
    "from langchain_community.vectorstores import Chroma\n",
    "from langchain_text_splitters import CharacterTextSplitter, RecursiveCharacterTextSplitter\n",
    "import os\n",
    "import pprint\n",
    "import re\n",
    "from langchain_core.documents import Document\n",
    "from langchain_core.prompts.prompt import PromptTemplate\n",
    "from langchain_core.output_parsers import JsonOutputParser\n",
    "from langchain_ollama import ChatOllama\n",
    "from pydantic import BaseModel, Field\n",
    "import json\n",
    "import uuid\n",
    "import chromadb\n",
    "from chromadb.config import Settings\n",
    "import unicodedata\n",
    "from langchain_google_genai import GoogleGenerativeAI\n",
    "import uuid\n",
    "# from chromadb.utils.embedding_functions import SentenceTransformerEmbeddingFunction\n",
    "import pickle as pkl\n",
    "import requests\n",
    "import subprocess\n",
    "import pandas as pd\n",
    "from bs4 import BeautifulSoup\n",
    "import tqdm\n",
    "import time\n",
    "from sentence_transformers import SentenceTransformer\n",
    "from langchain_huggingface import HuggingFaceEmbeddings\n",
    "import datetime\n",
    "import time"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "328c2061",
   "metadata": {},
   "source": [
    "## Google option"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9bc6ea96",
   "metadata": {},
   "source": [
    "#### Config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7be2125f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/antonioparragaleo/Documents/RAG'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "357e9e5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('api_google.txt') as f:\n",
    "    \n",
    "    api_key = json.load(f)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9b413d0a",
   "metadata": {},
   "source": [
    "#### Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "7784f8c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def clean_text(text):\n",
    "    # Normalize weird Unicode characters to their closest ASCII equivalent\n",
    "    text = unicodedata.normalize(\"NFKC\", text)\n",
    "    # Replace non-breaking hyphens and dashes with ASCII hyphen\n",
    "    text = re.sub(r'[\\u2010-\\u2015]', '-', text)\n",
    "     # Remove form feed characters (\\x0c), common page breaks\n",
    "    text = text.replace('\\x0c', ' ')\n",
    "    # Replace non-breaking spaces (\\xa0) with regular space\n",
    "    text = text.replace('\\xa0', ' ')\n",
    "    # 3. Fix hyphenated line breaks (e.g., treat-\\nment -> treatment)\n",
    "    text = re.sub(r\"(\\w+)-\\n(\\w+)\", r\"\\1\\2\", text)\n",
    "    # Remove numbered citations like (1), (9, 10), (5–7), etc.\n",
    "    text = re.sub(r\"\\(\\s?\\d+(?:\\s?(?:,|-)\\s?\\d+)*\\s?\\)\", \"\", text)\n",
    "    # Remove mid-sentence line breaks: \"word\\nword\" -> \"word word\"\n",
    "    text = re.sub(r'(?<!\\n)\\n(?!\\n)', ' ', text)\n",
    "    # Replace multiple newlines with just two (preserve paragraphs)\n",
    "    text = re.sub(r'\\n{2,}', '\\n\\n', text)\n",
    "    return text.strip()\n",
    "\n",
    "def get_system_message_rag(content):\n",
    "        return f\"\"\"You are an expert consultant helping executive advisors to get relevant information from scientific articles and code related to reproduction and bioinformatics.\n",
    "\n",
    "        Generate your response by following the steps below:\n",
    "        1. Recursively break down the question into smaller questions to better understand it.\n",
    "        2. For each question/directive:\n",
    "            2a. Select the most relevant information from the context in light of the conversation history.\n",
    "        3. Generate a draft response using selected information.\n",
    "        4. Remove duplicate content from draft response.\n",
    "        5. Generate your final response after adjusting it to increase accuracy and relevance.\n",
    "        6. Do not try to summarize the answers, explain it properly.\n",
    "        7. When you provide information, you must also provide the reference of the article.\n",
    "        8. Do not look up on internet.\n",
    "        9. Only show your final response! \n",
    "        \n",
    "        Constraints:\n",
    "        1. DO NOT PROVIDE ANY EXPLANATION OR DETAILS OR MENTION THAT YOU WERE GIVEN CONTEXT. Only do that when questions are related to coding.\n",
    "        2. Don't mention that you are not able to find the answer in the provided context.\n",
    "        3. Ignore the part of the content that only contains references.\n",
    "        3. Don't make up the answers by yourself.\n",
    "        4. Try your best to provide answer from the given context.\n",
    "\n",
    "        CONTENT:\n",
    "        {content}\n",
    "        \"\"\"\n",
    "\n",
    "def get_ques_response_prompt(question, context):\n",
    "    return f\"\"\"\n",
    "    Context\\n:\n",
    "    {context}\n",
    "    ==============================================================\n",
    "    Based on the above context, please provide the answer to the following question\\n:\n",
    "    {question}\n",
    "    \"\"\"\n",
    "\n",
    "def get_pmid_from_doi(doi, api_key=None):\n",
    "    url = \"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi\"\n",
    "    params = {\n",
    "        \"db\": \"pubmed\",\n",
    "        \"term\": f\"{doi}[DOI]\",\n",
    "        \"retmode\": \"json\"\n",
    "    }\n",
    "    if api_key:\n",
    "        params[\"api_key\"] = api_key\n",
    "\n",
    "    response = requests.get(url, params=params)\n",
    "    result = response.json()\n",
    "    pmids = result.get(\"esearchresult\", {}).get(\"idlist\", [])\n",
    "    return pmids[0] if pmids else 'None'\n",
    "\n",
    "\n",
    "\n",
    "def get_pmcid_from_pmid(pmid, api_key=None):\n",
    "    url = \"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi\"\n",
    "    params = {\n",
    "        \"dbfrom\": \"pubmed\",\n",
    "        \"db\": \"pmc\",\n",
    "        \"id\": pmid,\n",
    "        \"retmode\": \"json\"\n",
    "    }\n",
    "    if api_key:\n",
    "        params[\"api_key\"] = api_key\n",
    "\n",
    "    response = requests.get(url, params=params)\n",
    "    \n",
    "\n",
    "    try:\n",
    "        data = response.json()\n",
    "        pmcid_link = None\n",
    "        linksets = data['linksets'][0]['linksetdbs']\n",
    "\n",
    "        for j in linksets:\n",
    "            if j.get('linkname') =='pubmed_pmc':\n",
    "                pmcid_link = j['links'][0]\n",
    "                return pmcid_link\n",
    "    except Exception:\n",
    "        return None\n",
    "\n",
    "def clean_doi(doi):\n",
    "    \"\"\"Remove prefixes from DOI links and standardize format.\"\"\"\n",
    "    return re.sub(\"(https://doi\\.org/|http://dx\\.doi\\.org/)\", \"\", doi.strip())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b11837db",
   "metadata": {},
   "source": [
    "## Timing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "01e40cf4",
   "metadata": {},
   "outputs": [],
   "source": [
    "time_management = {}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d6746c38",
   "metadata": {},
   "source": [
    "#### Model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "23cd2e8b",
   "metadata": {},
   "outputs": [],
   "source": [
    "llm = GoogleGenerativeAI(model=\"gemini-2.0-flash\",api_key=api_key['key'],temperature=0.2)\n",
    "# from langchain_google_genai import ChatGoogleGenerativeAI\n",
    "# llm = ChatGoogleGenerativeAI(model=\"gemma-3-27b-it\",api_key=api_key['key'], temperature=0.2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a914329c",
   "metadata": {},
   "source": [
    "#### Load PDFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8d164fb0",
   "metadata": {},
   "outputs": [],
   "source": [
    "paths = []\n",
    "paths_folder_create = []\n",
    "for root, dirs, files in os.walk(\"Data\"):\n",
    "    for f in files:\n",
    "        if f.endswith('.pdf'):\n",
    "            paths.append(os.path.join(root,f))\n",
    "            paths_folder_create.append(os.path.join(os.getcwd(),\"ToDataBase\",f))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "96d7da10",
   "metadata": {},
   "source": [
    "We create a folder with each article to pre-process the references"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "0e4874f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None]"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.mkdir(os.path.join(os.getcwd(),\"ToDataBase\"))\n",
    "[os.mkdir(re.sub(r'.pdf','',p)) for p in paths_folder_create]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a0f6a65c",
   "metadata": {},
   "outputs": [],
   "source": [
    "articles = []\n",
    "for file in paths:\n",
    "    loader = loader = PyPDFLoader(file, mode=\"single\")\n",
    "    doc = loader.load()[0] # As this function provides a list, we select the first element.\n",
    "    doc.page_content = clean_text(doc.page_content)\n",
    "    \n",
    "    articles.append(doc)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "62bb9420",
   "metadata": {},
   "source": [
    "#### Json generation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "cd521499",
   "metadata": {},
   "outputs": [],
   "source": [
    "class paper(BaseModel):\n",
    "\n",
    "    PaperTitle: str = Field(description=\"The full title of the research paper\")\n",
    "    Publication: str = Field(description=\"Year: The year the paper was published\")\n",
    "    Authors: str = Field(description=\"The full names of all authors of the paper\")\n",
    "    Email: str = Field(description=\"The email address of the author (if provided)\")\n",
    "    Abstract: str = Field(description=\"The full text of the paper's abstract.\")\n",
    "    Introduction: str = Field(description=\"The full text fo the paper's introduction. Bear in mind that it can have other names such as background.\")\n",
    "    Methods: str = Field(description=\"The full text fo the paper's methods. Don't take the information from abstract.\")\n",
    "    Results: str = Field(description=\"The full text fo the paper's results. Don't take the information from abstract. Please take all possible text of results, this section could be divided into different sections.\")\n",
    "    Discussion: str = Field(description=\"The full text fo the paper's discussion if provided. Otherwise leave the filed blank. It should be entitled 'Discussion'\")\n",
    "    Conclusion: str = Field(description=\"The full text fo the paper's conclusion if provided. Otherwise leave the filed blank. Please don't take this information form abstract.\")\n",
    "    DOI: str = Field(description=\"Provide the Digital Object Identifier (DOI) for the paper. If a DOI is not available, provide a URL for the paper. This field is compulsory. Example of a DOI format: 10.1007/s10814-017-9105-3, 10.1016/j.cell.2023.08.001, 10.1038/nature12345.\")\n",
    "    Journal: str = Field(description=\"provide the name of the journal, e.g Nature.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6f763428",
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = JsonOutputParser(pydantic_object=paper)\n",
    "\n",
    "prompt = PromptTemplate(\n",
    "    template=\"\"\"\n",
    "You are an expert in analyzing scientific research papers. Please carefully read the provided research paper above and extract the following key information:\n",
    "Extract these nine (10) properties from the research paper:\n",
    "\n",
    "Paper Title: The full title of the research paper\n",
    "\n",
    "Publication Year: The year the paper was published\n",
    "\n",
    "Authors: The full names of all authors of the paper\n",
    "\n",
    "Email: The email address of the author (if provided)\n",
    "\n",
    "Abstract: The full text of the paper's abstract\n",
    "\n",
    "Introduction: The full text fo the paper's introduction. Bear in mind that it can have other names such as background.\n",
    "\n",
    "Methods: The full text fo the paper's methods. Don't take the information from abstract.\n",
    "\n",
    "Results: The full text fo the paper's results. Don't take the information from abstract. Please take all possible text of results, this section could be divided into different sections.\n",
    "\n",
    "Discussion:The full text fo the paper's discussion if provided. Otherwise leave the filed blank. It shoudl be entitled \"Discussion\"\n",
    "\n",
    "Conclusion: The full text fo the paper's conclusion if provided. Otherwise leave the filed blank. Please don't take this information form abstract.\n",
    "\n",
    "DOI: Provide the Digital Object Identifier (DOI) for the paper. If a DOI is not available, provide a URL for the paper. This field is compulsory. Example of a DOI format: 10.1007/s10814-017-9105-3, 10.1016/j.cell.2023.08.001, 10.1038/nature12345.\n",
    "\n",
    "Journal: provide the name of the journal, e.g Nature.\n",
    "\n",
    "\n",
    "Guidelines:\n",
    "\n",
    "\n",
    "The extracted information should be factual and accurate to the document. Be extremely concise, except for the Abstract, Introduction, Methods, Results. Discussion and Conclusion which should be copied in full.\n",
    "The extracted entities should be self-contained and easily understood without the rest of the paper. If a property is missing from the paper, please leave the field empty rather than guessing.\n",
    "Answer in JSON format. The JSON should contain 9 keys: \"PaperTitle\", \"PublicationYear\", \"Authors\", \"Email\", \"Abstract\", \"Introduction\", \"Methods\",\"Results\", \"DOI\", \"Reference\". You MUST add the DOI since is key to identify the article.\n",
    "\n",
    "Format instructions: \\n{format_instructions}\\n\n",
    "    \n",
    "The article is this:\\n{query}\\n\"\"\",\n",
    "\n",
    "    input_variables=[\"query\"],\n",
    "\n",
    "    partial_variables={\"format_instructions\": parser.get_format_instructions()}\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "aabf31d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "chain = prompt | llm | parser\n",
    "\n",
    "doi_prompt = PromptTemplate(\n",
    "    template=(\n",
    "        \"\"\"Extract the Digital Object Identifier (DOI) from the following paper.\n",
    "        \"The DOI must be returned in its raw form (e.g., 10.1007/s10814-017-9105-3,\n",
    "        \"10.1016/j.cell.2023.08.001, 10.1038/nature12345).\n",
    "        \"Return only the DOI string with no additional text.\n",
    "        \"If no DOI is found, return None.\\n\\n\n",
    "        \"Paper:\\n{query}\"\"\"\n",
    "    ),\n",
    "    input_variables=[\"query\"]\n",
    ")\n",
    "\n",
    "chain_doi = doi_prompt | llm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e1f44809",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 18/18 [11:53<00:00, 39.62s/it]\n"
     ]
    }
   ],
   "source": [
    "start = time.time()\n",
    "my_json = []\n",
    "\n",
    "for doc in tqdm.tqdm(articles, total=len(articles)):\n",
    "    info_article = chain.invoke({\"query\": doc.page_content})\n",
    "\n",
    "    if 'DOI' not in info_article.keys():\n",
    "        previous_doi= chain_doi.invoke({\"query\":doc.page_content})\n",
    "        if 'https://doi.org/' not in previous_doi:\n",
    "            info_article['DOI'] = 'https://doi.org/' + previous_doi\n",
    "        else:\n",
    "            info_article['DOI'] = previous_doi\n",
    "    \n",
    "    my_json.append(info_article)\n",
    "\n",
    "time_management[\"ParserMain\"] = time.time()-start\n",
    "\n",
    "with open('info_articles_main.pkl','wb') as f:\n",
    "    pkl.dump(my_json,f)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7147d0f3",
   "metadata": {},
   "source": [
    "## References"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "4d4b49c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# already DOI (To avoid repeat)\n",
    "\n",
    "includedDOI = [ re.sub('(https://doi\\.org/|http://dx\\.doi\\.org/)', '',a['DOI']) for a in my_json]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "973d0869",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We process the article: Devesa-Peiro2020\n",
      "Getting PDFs-----------------------------\n",
      "\n",
      "We process the article: Sebastian-Leon2025\n",
      "Getting PDFs-----------------------------\n",
      "\n",
      "We process the article: Henarejos-Castillo2020\n",
      "Getting PDFs-----------------------------\n",
      "\n",
      "We process the article: Parraga-Leo2023\n",
      "Getting PDFs-----------------------------\n",
      "\n",
      "We process the article: Garcia-Acero2025\n",
      "Getting PDFs-----------------------------\n",
      "\n",
      "We process the article: Sebastian-Leon2021\n",
      "Getting PDFs-----------------------------\n",
      "\n",
      "We process the article: Henarejos-Castillo2021\n",
      "Getting PDFs-----------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-17 21:24:52,774 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21989058\n",
      "2025-08-17 21:24:53,518 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27901055\n",
      "2025-08-17 21:24:53,518 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24020646\n",
      "2025-08-17 21:24:53,518 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25227694\n",
      "2025-08-17 21:24:55,788 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11125122\n",
      "2025-08-17 21:24:56,503 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 21:24:57,337 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26243799\n",
      "2025-08-17 21:24:58,065 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30276597\n",
      "2025-08-17 21:25:00,376 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32461654\n",
      "2025-08-17 21:25:00,376 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26620551\n",
      "2025-08-17 21:25:01,311 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 21:25:02,131 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16722528\n",
      "2025-08-17 21:25:02,131 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21478889\n",
      "2025-08-17 21:25:04,611 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27690531\n",
      "2025-08-17 21:25:05,499 - INFO - pubmed2pdf.cli - Trying to fetch pmid 2061095\n",
      "2025-08-17 21:25:05,499 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29225794\n",
      "2025-08-17 21:25:08,015 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30395287\n",
      "2025-08-17 21:25:08,714 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 21:25:09,550 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26667040\n",
      "2025-08-17 21:25:10,509 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26463876\n",
      "2025-08-17 21:25:12,807 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31077743\n",
      "2025-08-17 21:25:14,254 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22248077\n",
      "2025-08-17 21:25:14,255 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25906823\n",
      "2025-08-17 21:25:14,255 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 21:25:15,031 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30304577\n",
      "2025-08-17 21:25:15,952 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18493590\n",
      "2025-08-17 21:25:15,952 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 21:25:16,820 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23307926\n",
      "2025-08-17 21:25:17,554 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21299886\n",
      "2025-08-17 21:25:17,554 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31427081\n",
      "2025-08-17 21:25:19,312 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23153500\n",
      "2025-08-17 21:25:20,742 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28031467\n",
      "2025-08-17 21:25:22,077 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26918822\n",
      "2025-08-17 21:25:22,077 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30023082\n",
      "/home/antonioparragaleo/miniconda3/envs/RAG/lib/python3.11/site-packages/pubmed2pdf/utils.py:57: XMLParsedAsHTMLWarning: It looks like you're using an HTML parser to parse an XML document.\n",
      "\n",
      "Assuming this really is an XML document, what you're doing might work, but you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the Python package 'lxml' installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "\n",
      "If you want or need to use an HTML parser on this document, you can make this warning go away by filtering it. To do that, run this code before calling the BeautifulSoup constructor:\n",
      "\n",
      "    from bs4 import XMLParsedAsHTMLWarning\n",
      "    import warnings\n",
      "\n",
      "    warnings.filterwarnings(\"ignore\", category=XMLParsedAsHTMLWarning)\n",
      "\n",
      "  soup = BeautifulSoup(req.content, 'lxml')\n",
      "2025-08-17 21:25:27,514 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 21:25:28,381 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22728672\n",
      "2025-08-17 21:25:29,124 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21565265\n",
      "2025-08-17 21:25:30,469 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30406445\n",
      "2025-08-17 21:25:32,950 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15280409\n",
      "2025-08-17 21:25:36,505 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26215093\n",
      "2025-08-17 21:25:36,506 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31584097\n",
      "2025-08-17 21:25:37,234 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19451168\n",
      "2025-08-17 21:25:37,960 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19508998\n",
      "2025-08-17 21:25:38,716 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22726442\n",
      "2025-08-17 21:25:41,068 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31440958\n",
      "2025-08-17 21:25:43,611 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29251670\n",
      "2025-08-17 21:25:44,558 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30357004\n",
      "2025-08-17 21:25:44,558 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19546856\n",
      "2025-08-17 21:25:46,779 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30357029\n",
      "2025-08-17 21:25:46,779 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 21:25:47,623 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28938739\n",
      "2025-08-17 21:25:48,339 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20890277\n",
      "2025-08-17 21:25:51,778 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30972108\n",
      "2025-08-17 21:25:51,778 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29706485\n",
      "2025-08-17 21:25:53,015 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22256780\n",
      "2025-08-17 21:25:53,787 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26911863\n",
      "2025-08-17 21:25:54,500 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29704323\n",
      "2025-08-17 21:25:55,481 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30224786\n",
      "2025-08-17 21:25:56,824 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 21:25:57,689 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28505269\n",
      "2025-08-17 21:25:58,408 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11089562\n",
      "2025-08-17 21:25:59,135 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31787151\n",
      "2025-08-17 21:26:00,524 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28971280\n",
      "2025-08-17 21:26:03,758 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26113766\n",
      "2025-08-17 21:26:04,462 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30734403\n",
      "2025-08-17 21:26:05,344 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30078697\n",
      "2025-08-17 21:26:06,645 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28388912\n",
      "2025-08-17 21:26:06,645 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22546560\n",
      "2025-08-17 21:26:08,515 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23407710\n",
      "2025-08-17 21:26:09,372 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30603774\n",
      "2025-08-17 21:26:11,690 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 21:26:12,504 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29259429\n",
      "2025-08-17 21:26:13,736 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12091911\n",
      "2025-08-17 21:26:15,986 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17554680\n",
      "2025-08-17 21:26:15,986 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11912187\n",
      "2025-08-17 21:26:16,665 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29019003\n",
      "2025-08-17 21:26:22,220 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26432246\n",
      "2025-08-17 21:26:22,220 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21900680\n",
      "2025-08-17 21:26:22,910 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23505417\n",
      "2025-08-17 21:26:22,910 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26432245\n",
      "2025-08-17 21:26:22,910 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30395331\n",
      "2025-08-17 21:26:23,584 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29100496\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in ToDataBase/Henarejos-Castillo2021\n",
      "Turn PDFs into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 20/20 [09:05<00:00, 27.29s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting xml from PMC--------------------\n",
      "\n",
      "Turn XML into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 62/62 [20:21<00:00, 19.71s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We process the article: Henarejos-Castillo2022\n",
      "Getting PDFs-----------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-17 21:56:26,230 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18586725\n",
      "2025-08-17 21:56:26,911 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19628854\n",
      "2025-08-17 21:56:27,900 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30817321\n",
      "2025-08-17 21:56:28,605 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17960529\n",
      "2025-08-17 21:56:32,202 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31885807\n",
      "2025-08-17 21:56:33,032 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32898168\n",
      "2025-08-17 21:56:36,176 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32918567\n",
      "2025-08-17 21:56:38,758 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22287627\n",
      "2025-08-17 21:56:39,504 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32552854\n",
      "2025-08-17 21:56:40,864 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30612956\n",
      "2025-08-17 21:56:43,235 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29259471\n",
      "2025-08-17 21:56:44,503 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33568179\n",
      "2025-08-17 21:56:47,152 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32164226\n",
      "2025-08-17 21:56:48,918 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32749712\n",
      "2025-08-17 21:56:49,745 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33307546\n",
      "2025-08-17 21:56:52,351 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15297299\n",
      "2025-08-17 21:56:53,017 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17447836\n",
      "2025-08-17 21:56:54,965 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24485249\n",
      "2025-08-17 21:56:56,426 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30344300\n",
      "2025-08-17 21:56:57,155 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32883349\n",
      "2025-08-17 21:56:58,483 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24368037\n",
      "2025-08-17 21:57:00,059 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22455463\n",
      "2025-08-17 21:57:00,784 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33077917\n",
      "2025-08-17 21:57:03,244 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32468072\n",
      "2025-08-17 21:57:05,357 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17360645\n",
      "2025-08-17 21:57:06,082 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24959817\n",
      "2025-08-17 21:57:08,464 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24837612\n",
      "2025-08-17 21:57:09,897 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31861484\n",
      "2025-08-17 21:57:10,698 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21164525\n",
      "2025-08-17 21:57:13,947 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29264977\n",
      "2025-08-17 21:57:14,650 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31738970\n",
      "2025-08-17 21:57:15,984 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24600449\n",
      "2025-08-17 21:57:18,752 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32684080\n",
      "2025-08-17 21:57:19,493 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28770908\n",
      "2025-08-17 21:57:21,337 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24285824\n",
      "2025-08-17 21:57:23,836 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30929717\n",
      "2025-08-17 21:57:25,195 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24762139\n",
      "2025-08-17 21:57:25,880 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33975619\n",
      "2025-08-17 21:57:27,262 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32251791\n",
      "2025-08-17 21:57:28,653 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19954523\n",
      "2025-08-17 21:57:30,108 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18765015\n",
      "2025-08-17 21:57:31,281 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32466458\n",
      "2025-08-17 21:57:31,957 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21540246\n",
      "2025-08-17 21:57:32,630 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28901407\n",
      "2025-08-17 21:57:34,801 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29324807\n",
      "2025-08-17 21:57:37,729 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32791974\n",
      "2025-08-17 21:57:39,202 - INFO - pubmed2pdf.cli - Trying to fetch pmid 14644817\n",
      "2025-08-17 21:57:39,954 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32401299\n",
      "2025-08-17 21:57:40,677 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32680910\n",
      "2025-08-17 21:57:41,727 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28378640\n",
      "2025-08-17 21:57:42,554 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27830058\n",
      "2025-08-17 21:57:44,524 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32514174\n",
      "2025-08-17 21:57:47,448 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12810542\n",
      "2025-08-17 21:57:48,439 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33298930\n",
      "2025-08-17 21:57:52,311 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32608159\n",
      "2025-08-17 21:57:53,741 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30509602\n",
      "2025-08-17 21:57:55,135 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32443177\n",
      "2025-08-17 21:57:56,166 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32299776\n",
      "2025-08-17 21:57:57,504 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29671401\n",
      "2025-08-17 21:57:58,961 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32818434\n",
      "2025-08-17 21:58:00,290 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27840831\n",
      "2025-08-17 21:58:00,975 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34688597\n",
      "2025-08-17 21:58:02,260 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27428722\n",
      "2025-08-17 21:58:02,981 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31953174\n",
      "2025-08-17 21:58:04,342 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32872526\n",
      "2025-08-17 21:58:06,086 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32274342\n",
      "2025-08-17 21:58:10,222 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32404170\n",
      "2025-08-17 21:58:11,754 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25521941\n",
      "2025-08-17 21:58:13,422 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24959814\n",
      "2025-08-17 21:58:15,471 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26446766\n",
      "2025-08-17 21:58:17,459 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24320616\n",
      "2025-08-17 21:58:18,251 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24947308\n",
      "2025-08-17 21:58:19,683 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34867824\n",
      "2025-08-17 21:58:23,038 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30833554\n",
      "2025-08-17 21:58:25,116 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24412827\n",
      "2025-08-17 21:58:25,822 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33258995\n",
      "2025-08-17 21:58:28,130 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22192140\n",
      "2025-08-17 21:58:29,485 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32921216\n",
      "2025-08-17 21:58:30,164 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17462640\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in ToDataBase/Henarejos-Castillo2022\n",
      "Turn PDFs into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 26/26 [12:39<00:00, 29.21s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting xml from PMC--------------------\n",
      "\n",
      "Turn XML into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 53/53 [18:13<00:00, 20.63s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We process the article: Diaz-Gimeno2024\n",
      "Getting PDFs-----------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-17 22:29:39,976 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33880419\n",
      "2025-08-17 22:29:41,121 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36822566\n",
      "2025-08-17 22:29:42,415 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18546601\n",
      "2025-08-17 22:29:44,658 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36472596\n",
      "2025-08-17 22:29:45,352 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28370781\n",
      "2025-08-17 22:29:46,334 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35929523\n",
      "2025-08-17 22:29:47,132 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33036008\n",
      "2025-08-17 22:29:48,727 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26109056\n",
      "2025-08-17 22:29:50,228 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32723696\n",
      "2025-08-17 22:29:51,467 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24581625\n",
      "2025-08-17 22:29:52,774 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29737471\n",
      "2025-08-17 22:29:55,961 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 22:29:56,712 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17160063\n",
      "2025-08-17 22:29:58,877 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27441287\n",
      "2025-08-17 22:30:00,183 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30713991\n",
      "2025-08-17 22:30:01,413 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33880420\n",
      "2025-08-17 22:30:02,608 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33067123\n",
      "2025-08-17 22:30:03,850 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11823860\n",
      "2025-08-17 22:30:06,312 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12421895\n",
      "2025-08-17 22:30:07,540 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33313697\n",
      "2025-08-17 22:30:08,618 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28923940\n",
      "2025-08-17 22:30:09,584 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31540845\n",
      "2025-08-17 22:30:10,990 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32974805\n",
      "2025-08-17 22:30:13,380 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30452456\n",
      "2025-08-17 22:30:15,900 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24581986\n",
      "2025-08-17 22:30:17,093 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29315421\n",
      "2025-08-17 22:30:17,797 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 22:30:18,610 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 22:30:19,353 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34947871\n",
      "2025-08-17 22:30:20,118 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10221616\n",
      "2025-08-17 22:30:21,564 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35092277\n",
      "2025-08-17 22:30:22,281 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30145981\n",
      "2025-08-17 22:30:23,855 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12490681\n",
      "2025-08-17 22:30:24,576 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36070983\n",
      "2025-08-17 22:30:25,916 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15353123\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in ToDataBase/Diaz-Gimeno2024\n",
      "Turn PDFs into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 3/3 [02:09<00:00, 43.30s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting xml from PMC--------------------\n",
      "\n",
      "Turn XML into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 32/32 [08:47<00:00, 16.49s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We process the article: Devesa-Peiro2021\n",
      "Getting PDFs-----------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-17 22:41:40,580 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16707507\n",
      "2025-08-17 22:41:42,320 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27323161\n",
      "2025-08-17 22:41:43,472 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 22:41:44,302 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22819144\n",
      "2025-08-17 22:41:45,575 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26856931\n",
      "2025-08-17 22:41:46,857 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29576469\n",
      "2025-08-17 22:41:48,184 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26366788\n",
      "2025-08-17 22:41:49,534 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30588329\n",
      "2025-08-17 22:41:52,025 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27998009\n",
      "2025-08-17 22:41:52,740 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19910308\n",
      "2025-08-17 22:41:53,436 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30477193\n",
      "2025-08-17 22:41:54,306 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25243856\n",
      "2025-08-17 22:41:55,031 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16632515\n",
      "2025-08-17 22:41:55,753 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30161168\n",
      "2025-08-17 22:41:58,476 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30798642\n",
      "2025-08-17 22:42:00,235 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16820110\n",
      "2025-08-17 22:42:01,461 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31387263\n",
      "2025-08-17 22:42:02,122 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16725353\n",
      "2025-08-17 22:42:03,261 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19527389\n",
      "2025-08-17 22:42:04,087 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31339848\n",
      "2025-08-17 22:42:06,050 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21436257\n",
      "2025-08-17 22:42:07,759 - INFO - pubmed2pdf.cli - Trying to fetch pmid 9130884\n",
      "2025-08-17 22:42:08,981 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19864262\n",
      "2025-08-17 22:42:09,645 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18765014\n",
      "2025-08-17 22:42:10,931 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20391347\n",
      "2025-08-17 22:42:11,956 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27640610\n",
      "2025-08-17 22:42:14,114 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29608177\n",
      "2025-08-17 22:42:16,436 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12751776\n",
      "2025-08-17 22:42:17,135 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31004479\n",
      "2025-08-17 22:42:17,855 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31948481\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in ToDataBase/Devesa-Peiro2021\n",
      "Turn PDFs into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 4/4 [03:03<00:00, 45.83s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting xml from PMC--------------------\n",
      "\n",
      "Turn XML into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 26/26 [05:43<00:00, 13.20s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We process the article: Henarejos-Castillo2024\n",
      "Getting PDFs-----------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-17 22:51:34,072 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33025164\n",
      "2025-08-17 22:51:36,722 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36055201\n",
      "2025-08-17 22:51:37,967 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32943985\n",
      "2025-08-17 22:51:39,633 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24995866\n",
      "2025-08-17 22:51:40,818 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19810025\n",
      "2025-08-17 22:51:41,656 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28574608\n",
      "2025-08-17 22:51:42,454 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29566152\n",
      "2025-08-17 22:51:44,199 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29425284\n",
      "2025-08-17 22:51:44,905 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34597585\n",
      "2025-08-17 22:51:56,971 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30207912\n",
      "2025-08-17 22:51:57,652 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25326635\n",
      "2025-08-17 22:51:58,385 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15769979\n",
      "2025-08-17 22:51:59,057 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32139519\n",
      "2025-08-17 22:52:00,037 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15355784\n",
      "2025-08-17 22:52:01,425 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22441326\n",
      "2025-08-17 22:52:02,430 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31605095\n",
      "2025-08-17 22:52:04,990 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31096927\n",
      "2025-08-17 22:52:06,693 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34373449\n",
      "2025-08-17 22:52:09,230 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32099950\n",
      "2025-08-17 22:52:10,418 - INFO - pubmed2pdf.cli - Trying to fetch pmid 39185493\n",
      "2025-08-17 22:52:11,580 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27341508\n",
      "2025-08-17 22:52:14,470 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25344409\n",
      "2025-08-17 22:52:16,440 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20453035\n",
      "2025-08-17 22:52:17,147 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29367954\n",
      "2025-08-17 22:52:18,286 - INFO - pubmed2pdf.cli - Trying to fetch pmid 38310280\n",
      "2025-08-17 22:52:20,157 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30371827\n",
      "2025-08-17 22:52:21,187 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35311454\n",
      "2025-08-17 22:52:21,892 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33411734\n",
      "2025-08-17 22:52:24,669 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24597867\n",
      "2025-08-17 22:52:25,363 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32657361\n",
      "2025-08-17 22:52:26,065 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27153000\n",
      "2025-08-17 22:52:26,744 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30742610\n",
      "2025-08-17 22:52:29,243 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30423088\n",
      "2025-08-17 22:52:29,949 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21297384\n",
      "2025-08-17 22:52:32,262 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28550012\n",
      "2025-08-17 22:52:34,505 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34480478\n",
      "2025-08-17 22:52:35,191 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28535294\n",
      "2025-08-17 22:52:35,852 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29318599\n",
      "2025-08-17 22:52:36,616 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34408140\n",
      "2025-08-17 22:52:39,146 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 22:52:39,917 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27899662\n",
      "2025-08-17 22:52:40,566 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18691683\n",
      "2025-08-17 22:52:51,692 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33558758\n",
      "2025-08-17 22:52:54,161 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25738998\n",
      "2025-08-17 22:52:55,778 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29231814\n",
      "2025-08-17 22:52:57,009 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20823910\n",
      "2025-08-17 22:52:59,167 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33751459\n",
      "2025-08-17 22:53:02,529 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31484976\n",
      "2025-08-17 22:53:04,924 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35388217\n",
      "2025-08-17 22:53:07,453 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30952665\n",
      "2025-08-17 22:53:08,198 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19461434\n",
      "2025-08-17 22:53:09,002 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30964354\n",
      "2025-08-17 22:53:09,702 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22197887\n",
      "2025-08-17 22:53:11,056 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26935102\n",
      "2025-08-17 22:53:12,475 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29290392\n",
      "2025-08-17 22:53:13,819 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36732629\n",
      "2025-08-17 22:53:17,554 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33613687\n",
      "2025-08-17 22:53:18,258 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28253086\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in ToDataBase/Henarejos-Castillo2024\n",
      "Turn PDFs into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 13/13 [08:26<00:00, 38.93s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting xml from PMC--------------------\n",
      "\n",
      "Turn XML into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 45/45 [18:56<00:00, 25.26s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We process the article: Sanchez-Reyes2025\n",
      "Getting PDFs-----------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-17 23:20:48,511 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35085989\n",
      "2025-08-17 23:20:49,870 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28627518\n",
      "2025-08-17 23:20:52,818 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33489905\n",
      "2025-08-17 23:20:55,638 - INFO - pubmed2pdf.cli - Trying to fetch pmid 38906211\n",
      "2025-08-17 23:20:57,151 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35409214\n",
      "2025-08-17 23:20:57,830 - INFO - pubmed2pdf.cli - Trying to fetch pmid 2903323\n",
      "2025-08-17 23:21:00,118 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32992208\n",
      "2025-08-17 23:21:01,469 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24227677\n",
      "2025-08-17 23:21:02,182 - INFO - pubmed2pdf.cli - Trying to fetch pmid 37355665\n",
      "2025-08-17 23:21:03,827 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23104886\n",
      "2025-08-17 23:21:04,500 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31467207\n",
      "2025-08-17 23:21:05,484 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12215322\n",
      "2025-08-17 23:21:06,756 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36482358\n",
      "2025-08-17 23:21:08,300 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31932747\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in ToDataBase/Sanchez-Reyes2025\n",
      "Turn PDFs into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 3/3 [01:41<00:00, 33.81s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting xml from PMC--------------------\n",
      "\n",
      "Turn XML into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 11/11 [01:39<00:00,  9.03s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We process the article: Marti-Garcia2024\n",
      "Getting PDFs-----------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-17 23:24:47,325 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29250769\n",
      "2025-08-17 23:24:48,023 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35351010\n",
      "2025-08-17 23:24:49,415 - INFO - pubmed2pdf.cli - Trying to fetch pmid 1521649\n",
      "2025-08-17 23:24:50,722 - INFO - pubmed2pdf.cli - Trying to fetch pmid 1838082\n",
      "2025-08-17 23:24:52,045 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15368600\n",
      "2025-08-17 23:24:53,835 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21435901\n",
      "2025-08-17 23:24:55,210 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31085094\n",
      "2025-08-17 23:24:56,452 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19948745\n",
      "2025-08-17 23:24:57,169 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25298042\n",
      "2025-08-17 23:24:59,493 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30283331\n",
      "2025-08-17 23:25:03,714 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 23:25:04,487 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28854727\n",
      "2025-08-17 23:25:05,174 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32544881\n",
      "2025-08-17 23:25:05,889 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29414008\n",
      "2025-08-17 23:25:07,228 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 23:25:09,016 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29955789\n",
      "2025-08-17 23:25:09,707 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16495411\n",
      "2025-08-17 23:25:10,446 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12519859\n",
      "2025-08-17 23:25:11,149 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35710597\n",
      "2025-08-17 23:25:12,476 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16772154\n",
      "2025-08-17 23:25:13,883 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 23:25:14,697 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18487213\n",
      "2025-08-17 23:25:15,405 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12743149\n",
      "2025-08-17 23:25:17,258 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30360364\n",
      "2025-08-17 23:25:18,431 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34120144\n",
      "2025-08-17 23:25:21,110 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11020522\n",
      "2025-08-17 23:25:22,366 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36681695\n",
      "2025-08-17 23:25:24,521 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27981182\n",
      "2025-08-17 23:25:25,896 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22246449\n",
      "2025-08-17 23:25:26,585 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22969138\n",
      "2025-08-17 23:25:27,279 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36849755\n",
      "2025-08-17 23:25:29,842 - INFO - pubmed2pdf.cli - Trying to fetch pmid 9786459\n",
      "2025-08-17 23:25:31,178 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28760748\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in ToDataBase/Marti-Garcia2024\n",
      "Turn PDFs into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 4/4 [02:04<00:00, 31.10s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting xml from PMC--------------------\n",
      "\n",
      "Turn XML into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 29/29 [08:27<00:00, 17.50s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We process the article: Marti-Garcia2024_review\n",
      "Getting PDFs-----------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-17 23:37:02,776 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33258951\n",
      "2025-08-17 23:37:03,484 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23754297\n",
      "2025-08-17 23:37:04,182 - INFO - pubmed2pdf.cli - Trying to fetch pmid 9241295\n",
      "2025-08-17 23:37:05,045 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 23:37:05,801 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34061977\n",
      "2025-08-17 23:37:06,856 - INFO - pubmed2pdf.cli - Trying to fetch pmid 6053626\n",
      "2025-08-17 23:37:07,965 - INFO - pubmed2pdf.cli - Trying to fetch pmid 4347566\n",
      "2025-08-17 23:37:08,694 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25785919\n",
      "2025-08-17 23:37:09,388 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18454478\n",
      "2025-08-17 23:37:11,506 - INFO - pubmed2pdf.cli - Trying to fetch pmid 8801134\n",
      "2025-08-17 23:37:12,812 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12641627\n",
      "2025-08-17 23:37:13,506 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27178763\n",
      "2025-08-17 23:37:14,701 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20664389\n",
      "2025-08-17 23:37:15,516 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20129426\n",
      "2025-08-17 23:37:16,844 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15568328\n",
      "2025-08-17 23:37:17,925 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34295308\n",
      "2025-08-17 23:37:20,335 - INFO - pubmed2pdf.cli - Trying to fetch pmid 3755843\n",
      "2025-08-17 23:37:21,281 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31077426\n",
      "2025-08-17 23:37:22,034 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23663223\n",
      "2025-08-17 23:37:24,441 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32532669\n",
      "2025-08-17 23:37:25,816 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16054962\n",
      "2025-08-17 23:37:27,009 - INFO - pubmed2pdf.cli - Trying to fetch pmid 4877579\n",
      "2025-08-17 23:37:28,345 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23327588\n",
      "2025-08-17 23:37:30,071 - INFO - pubmed2pdf.cli - Trying to fetch pmid 38401570\n",
      "2025-08-17 23:37:31,426 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36599349\n",
      "2025-08-17 23:37:34,016 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35222264\n",
      "2025-08-17 23:37:36,755 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29683366\n",
      "2025-08-17 23:37:37,518 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11211132\n",
      "2025-08-17 23:37:38,186 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34234149\n",
      "2025-08-17 23:37:40,823 - INFO - pubmed2pdf.cli - Trying to fetch pmid 8499679\n",
      "2025-08-17 23:37:41,990 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12213366\n",
      "2025-08-17 23:37:43,236 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10575580\n",
      "2025-08-17 23:37:44,349 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31895813\n",
      "2025-08-17 23:37:45,211 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33782057\n",
      "2025-08-17 23:37:48,610 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30834625\n",
      "2025-08-17 23:37:49,317 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25779030\n",
      "2025-08-17 23:37:50,583 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35773140\n",
      "2025-08-17 23:37:51,807 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35295981\n",
      "2025-08-17 23:37:54,055 - INFO - pubmed2pdf.cli - Trying to fetch pmid 8752624\n",
      "2025-08-17 23:37:55,379 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34600855\n",
      "2025-08-17 23:37:57,578 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36515535\n",
      "2025-08-17 23:37:58,242 - INFO - pubmed2pdf.cli - Trying to fetch pmid 6320581\n",
      "2025-08-17 23:37:58,958 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35264839\n",
      "2025-08-17 23:37:59,976 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33502634\n",
      "2025-08-17 23:38:03,250 - INFO - pubmed2pdf.cli - Trying to fetch pmid 3137101\n",
      "2025-08-17 23:38:04,532 - INFO - pubmed2pdf.cli - Trying to fetch pmid 37852992\n",
      "2025-08-17 23:38:06,943 - INFO - pubmed2pdf.cli - Trying to fetch pmid 8690114\n",
      "2025-08-17 23:38:08,293 - INFO - pubmed2pdf.cli - Trying to fetch pmid 9853881\n",
      "2025-08-17 23:38:08,988 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27295032\n",
      "2025-08-17 23:38:11,082 - INFO - pubmed2pdf.cli - Trying to fetch pmid 37383030\n",
      "2025-08-17 23:38:12,205 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35535879\n",
      "2025-08-17 23:38:12,988 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15690442\n",
      "2025-08-17 23:38:13,847 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17207799\n",
      "2025-08-17 23:38:15,231 - INFO - pubmed2pdf.cli - Trying to fetch pmid 760768\n",
      "2025-08-17 23:38:16,331 - INFO - pubmed2pdf.cli - Trying to fetch pmid 2056019\n",
      "2025-08-17 23:38:17,002 - INFO - pubmed2pdf.cli - Trying to fetch pmid 9696210\n",
      "2025-08-17 23:38:18,350 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18991682\n",
      "2025-08-17 23:38:21,686 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16373406\n",
      "2025-08-17 23:38:22,339 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36040386\n",
      "2025-08-17 23:38:26,375 - INFO - pubmed2pdf.cli - Trying to fetch pmid 8193098\n",
      "2025-08-17 23:38:27,579 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31547808\n",
      "2025-08-17 23:38:28,998 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10625188\n",
      "2025-08-17 23:38:29,669 - INFO - pubmed2pdf.cli - Trying to fetch pmid 8018616\n",
      "2025-08-17 23:38:30,991 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17371803\n",
      "2025-08-17 23:38:31,781 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10920085\n",
      "2025-08-17 23:38:32,462 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25553089\n",
      "2025-08-17 23:38:34,935 - INFO - pubmed2pdf.cli - Trying to fetch pmid 1427629\n",
      "2025-08-17 23:38:35,714 - INFO - pubmed2pdf.cli - Trying to fetch pmid 7650143\n",
      "2025-08-17 23:38:36,517 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15564650\n",
      "2025-08-17 23:38:38,953 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32147172\n",
      "2025-08-17 23:38:40,281 - INFO - pubmed2pdf.cli - Trying to fetch pmid 7732803\n",
      "2025-08-17 23:38:40,976 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33821427\n",
      "2025-08-17 23:38:44,447 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34980280\n",
      "2025-08-17 23:38:46,085 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23746838\n",
      "2025-08-17 23:38:47,307 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31398256\n",
      "2025-08-17 23:38:48,007 - INFO - pubmed2pdf.cli - Trying to fetch pmid 7978918\n",
      "2025-08-17 23:38:48,717 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36362169\n",
      "2025-08-17 23:38:49,393 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36151546\n",
      "2025-08-17 23:38:51,387 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18500453\n",
      "2025-08-17 23:38:53,883 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32061522\n",
      "2025-08-17 23:38:55,106 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32970718\n",
      "2025-08-17 23:38:57,366 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18328483\n",
      "2025-08-17 23:38:58,658 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33394226\n",
      "2025-08-17 23:39:00,887 - INFO - pubmed2pdf.cli - Trying to fetch pmid 39185250\n",
      "2025-08-17 23:39:02,034 - INFO - pubmed2pdf.cli - Trying to fetch pmid 14747176\n",
      "2025-08-17 23:39:02,714 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24138928\n",
      "2025-08-17 23:39:04,497 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33716956\n",
      "2025-08-17 23:39:06,906 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 23:39:07,672 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17418150\n",
      "2025-08-17 23:39:09,026 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29929546\n",
      "2025-08-17 23:39:10,509 - INFO - pubmed2pdf.cli - Trying to fetch pmid 8932630\n",
      "2025-08-17 23:39:11,244 - INFO - pubmed2pdf.cli - Trying to fetch pmid 336898\n",
      "2025-08-17 23:39:12,364 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32945210\n",
      "2025-08-17 23:39:13,089 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30142863\n",
      "2025-08-17 23:39:13,981 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32410619\n",
      "2025-08-17 23:39:16,038 - INFO - pubmed2pdf.cli - Trying to fetch pmid 8906001\n",
      "2025-08-17 23:39:16,714 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32132441\n",
      "2025-08-17 23:39:17,575 - INFO - pubmed2pdf.cli - Trying to fetch pmid 37248206\n",
      "2025-08-17 23:39:18,440 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15136099\n",
      "2025-08-17 23:39:19,671 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15533340\n",
      "2025-08-17 23:39:20,878 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21652599\n",
      "2025-08-17 23:39:21,571 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32077310\n",
      "2025-08-17 23:39:22,231 - INFO - pubmed2pdf.cli - Trying to fetch pmid 8150907\n",
      "2025-08-17 23:39:22,905 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12615813\n",
      "2025-08-17 23:39:23,547 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35382992\n",
      "2025-08-17 23:39:24,847 - INFO - pubmed2pdf.cli - Trying to fetch pmid 7868691\n",
      "2025-08-17 23:39:25,498 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32487204\n",
      "2025-08-17 23:39:27,064 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34552924\n",
      "2025-08-17 23:39:29,424 - INFO - pubmed2pdf.cli - Trying to fetch pmid 9988411\n",
      "2025-08-17 23:39:30,671 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32741623\n",
      "2025-08-17 23:39:31,993 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-17 23:39:32,781 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11117092\n",
      "2025-08-17 23:39:33,489 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30008696\n",
      "2025-08-17 23:39:35,726 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10966991\n",
      "2025-08-17 23:39:36,593 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35301622\n",
      "2025-08-17 23:39:38,667 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35044694\n",
      "2025-08-17 23:39:39,552 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34400878\n",
      "2025-08-17 23:39:42,239 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19131054\n",
      "2025-08-17 23:39:43,563 - INFO - pubmed2pdf.cli - Trying to fetch pmid 8984921\n",
      "2025-08-17 23:39:44,673 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25967104\n",
      "2025-08-17 23:39:46,165 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24700054\n",
      "2025-08-17 23:39:48,620 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32308115\n",
      "2025-08-17 23:39:49,407 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32567320\n",
      "2025-08-17 23:39:50,116 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11907929\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in ToDataBase/Marti-Garcia2024_review\n",
      "Turn PDFs into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 24/24 [14:26<00:00, 36.11s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting xml from PMC--------------------\n",
      "\n",
      "Turn XML into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 100/100 [13:14<00:00,  7.95s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We process the article: Diaz-Gimeno2022\n",
      "Getting PDFs-----------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-18 00:07:39,716 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19246470\n",
      "2025-08-18 00:07:40,422 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22285995\n",
      "2025-08-18 00:07:41,098 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28443690\n",
      "2025-08-18 00:07:41,776 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-18 00:07:42,585 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18077318\n",
      "2025-08-18 00:07:43,269 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-18 00:07:44,051 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27386492\n",
      "/home/antonioparragaleo/miniconda3/envs/RAG/lib/python3.11/site-packages/pubmed2pdf/utils.py:57: XMLParsedAsHTMLWarning: It looks like you're using an HTML parser to parse an XML document.\n",
      "\n",
      "Assuming this really is an XML document, what you're doing might work, but you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the Python package 'lxml' installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "\n",
      "If you want or need to use an HTML parser on this document, you can make this warning go away by filtering it. To do that, run this code before calling the BeautifulSoup constructor:\n",
      "\n",
      "    from bs4 import XMLParsedAsHTMLWarning\n",
      "    import warnings\n",
      "\n",
      "    warnings.filterwarnings(\"ignore\", category=XMLParsedAsHTMLWarning)\n",
      "\n",
      "  soup = BeautifulSoup(req.content, 'lxml')\n",
      "2025-08-18 00:07:48,478 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-18 00:07:49,245 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29093507\n",
      "2025-08-18 00:07:51,699 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21785142\n",
      "2025-08-18 00:07:52,366 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27172870\n",
      "2025-08-18 00:07:53,762 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20676074\n",
      "2025-08-18 00:07:55,975 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19542175\n",
      "2025-08-18 00:07:56,667 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25150838\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in ToDataBase/Diaz-Gimeno2022\n",
      "Turn PDFs into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 1/1 [00:28<00:00, 28.22s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting xml from PMC--------------------\n",
      "\n",
      "Turn XML into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 13/13 [03:19<00:00, 15.33s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We process the article: Devesa-Peiro2022\n",
      "Getting PDFs-----------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-18 00:11:59,858 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31806903\n",
      "2025-08-18 00:12:02,097 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21245076\n",
      "2025-08-18 00:12:02,778 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17604715\n",
      "2025-08-18 00:12:04,163 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23725226\n",
      "2025-08-18 00:12:05,627 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15797956\n",
      "2025-08-18 00:12:06,282 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-18 00:12:07,041 - INFO - pubmed2pdf.cli - Trying to fetch pmid 8903775\n",
      "2025-08-18 00:12:08,291 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32173784\n",
      "2025-08-18 00:12:10,755 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12969699\n",
      "2025-08-18 00:12:11,984 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24894503\n",
      "2025-08-18 00:12:13,657 - INFO - pubmed2pdf.cli - Trying to fetch pmid 14559028\n",
      "2025-08-18 00:12:14,900 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27271600\n",
      "2025-08-18 00:12:15,582 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19668215\n",
      "2025-08-18 00:12:17,777 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11278680\n",
      "2025-08-18 00:12:19,159 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28698599\n",
      "2025-08-18 00:12:22,585 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18270434\n",
      "2025-08-18 00:12:33,512 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24055232\n",
      "2025-08-18 00:12:34,765 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32000714\n",
      "2025-08-18 00:12:36,291 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20948989\n",
      "2025-08-18 00:12:39,159 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15661136\n",
      "2025-08-18 00:12:40,704 - INFO - pubmed2pdf.cli - Trying to fetch pmid 5675864\n",
      "2025-08-18 00:12:41,927 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29278305\n",
      "2025-08-18 00:12:42,620 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31004829\n",
      "2025-08-18 00:12:43,990 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34857954\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in ToDataBase/Devesa-Peiro2022\n",
      "Turn PDFs into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 5/5 [02:56<00:00, 35.37s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting xml from PMC--------------------\n",
      "\n",
      "Turn XML into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 19/19 [05:17<00:00, 16.71s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We process the article: Sebastian-Leon2018\n",
      "Getting PDFs-----------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-18 00:21:11,348 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-18 00:21:12,143 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24737781\n",
      "2025-08-18 00:21:12,869 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24082038\n",
      "2025-08-18 00:21:13,581 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15539444\n",
      "2025-08-18 00:21:14,327 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18035563\n",
      "2025-08-18 00:21:15,620 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23756099\n",
      "2025-08-18 00:21:17,027 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15924536\n",
      "2025-08-18 00:21:18,020 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22080510\n",
      "2025-08-18 00:21:18,696 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-18 00:21:19,460 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22683339\n",
      "2025-08-18 00:21:20,767 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23664094\n",
      "2025-08-18 00:21:22,064 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28710396\n",
      "2025-08-18 00:21:24,632 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22119322\n",
      "2025-08-18 00:21:25,848 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17674185\n",
      "2025-08-18 00:21:27,943 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16905766\n",
      "2025-08-18 00:21:28,604 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18697870\n",
      "2025-08-18 00:21:29,277 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16960016\n",
      "2025-08-18 00:21:31,043 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28668151\n",
      "2025-08-18 00:21:32,341 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22634130\n",
      "2025-08-18 00:21:33,671 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22028643\n",
      "2025-08-18 00:21:35,929 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in ToDataBase/Sebastian-Leon2018\n",
      "Turn PDFs into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 2/2 [01:11<00:00, 35.53s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting xml from PMC--------------------\n",
      "\n",
      "Turn XML into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 19/19 [02:24<00:00,  7.61s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We process the article: Diaz-Gimeno2017\n",
      "Getting PDFs-----------------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-18 00:25:18,509 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27128483\n",
      "2025-08-18 00:25:21,327 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11120680\n",
      "2025-08-18 00:25:22,005 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-18 00:25:22,764 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-18 00:25:24,516 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16005454\n",
      "2025-08-18 00:25:25,886 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18252602\n",
      "2025-08-18 00:25:27,146 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-18 00:25:27,907 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26385059\n",
      "2025-08-18 00:25:29,064 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20689021\n",
      "2025-08-18 00:25:29,812 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24706003\n",
      "2025-08-18 00:25:30,940 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27122490\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in ToDataBase/Diaz-Gimeno2017\n",
      "Turn PDFs into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 1/1 [00:41<00:00, 41.76s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting xml from PMC--------------------\n",
      "\n",
      "Turn XML into JSON--------------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 10/10 [01:43<00:00, 10.39s/it]\n"
     ]
    }
   ],
   "source": [
    "my_json_ref = []\n",
    "pmid_dict = {}\n",
    "# tocheck={}\n",
    "not_available = []\n",
    "\n",
    "HEADERS = {\n",
    "        \"User-Agent\": (\n",
    "            \"Mozilla/5.0 (Windows NT 10.0; Win64; x64) \"\n",
    "            \"AppleWebKit/537.36 (KHTML, like Gecko) \"\n",
    "            \"Chrome/115.0.0.0 Safari/537.36\"\n",
    "        )\n",
    "    }\n",
    "\n",
    "for i, article in enumerate(my_json):\n",
    "    \n",
    "    path_name = re.sub(r'.pdf','',paths[i]).split('/')[1]\n",
    "    pmid_dict[path_name] = []\n",
    "    #tocheck[path_name] = []\n",
    "\n",
    "    print(f\"We process the article: {path_name}\")\n",
    "\n",
    "    doi = clean_doi(article['DOI'])\n",
    "    \n",
    "    cross_ref = requests.get('https://api.crossref.org/works/'+doi)\n",
    "    list_reference = cross_ref.json()['message'].get('reference',None)\n",
    "    article['List_references'] = list(set([x.get('DOI') for x in list_reference if x.get('DOI') and x.get('DOI') not in includedDOI]))\n",
    "\n",
    "        # for x in list_reference:\n",
    "\n",
    "        #     if x.get('DOI'):\n",
    "\n",
    "        #         title = x.get('article-title')\n",
    "        #         doi = x.get(\"DOI\")\n",
    "        #         pmid = get_pmid_from_doi(doi)\n",
    "        #         pmcid = get_pmcid_from_pmid(pmid)\n",
    "        #         year = x.get(\"year\")\n",
    "        #         tocheck[path_name].append({title:[doi,pmid, year,pmcid]})\n",
    "\n",
    "\n",
    "        # tocheck[path_name] = [{'List_references' : list(set([(x.get('DOI')) for x in list_reference if x.get('DOI') and x.get('DOI') not in includedDOI]))}]\n",
    "        # tocheck[path_name] = [{'List_references' : list(set([(x.get('DOI')) for x in list_reference if x.get('DOI') and x.get('DOI') not in includedDOI]))}]\n",
    "\n",
    "        # We retrieve the pmid from each DOI\n",
    "\n",
    "    for ref_doi in article['List_references']:\n",
    "            \n",
    "        pmid_dict[path_name].append(get_pmid_from_doi(ref_doi))\n",
    "\n",
    "    # We try to get the pdf\n",
    "\n",
    "    print(\"Getting PDFs-----------------------------\\n\")\n",
    "\n",
    "    output_dir = os.path.join(\"ToDataBase\", path_name)\n",
    "\n",
    "    if len(pmid_dict[path_name]) > 0:\n",
    "\n",
    "        command = [\n",
    "        \"python3\", \"-m\", \"pubmed2pdf\", \"pdf\",\n",
    "        \"--out\", output_dir,\n",
    "        \"--pmids\", f\"{', '.join(pmid_dict[path_name])}\",\n",
    "        \"--maxtries\",\"1\",\n",
    "        \"--errors\", os.path.join(output_dir,\"pubmed2pdf_log.txt\")\n",
    "        ]\n",
    "        subprocess.run(command)\n",
    "\n",
    "        # We process pdfs\n",
    "\n",
    "        pdfs2retrieve = [os.path.join(output_dir,f) for f in os.listdir(output_dir) if f.endswith('.pdf')]\n",
    "\n",
    "        print(\"Turn PDFs into JSON--------------------\\n\")\n",
    "\n",
    "        for pdf in tqdm.tqdm(pdfs2retrieve, total=len(pdfs2retrieve)):\n",
    "\n",
    "            loader = loader = PyPDFLoader(pdf, mode=\"single\")\n",
    "            doc = loader.load()[0] # As this function provides a list, we select the first element.\n",
    "            doc.page_content = clean_text(doc.page_content)\n",
    "\n",
    "            try:\n",
    "                info_article = chain.invoke({\"query\": doc.page_content})\n",
    "                doi_candidate = info_article.get(\"DOI\",None)\n",
    "\n",
    "                if not doi_candidate:\n",
    "\n",
    "                    retry_doi = chain_doi.invoke({\"query\":doc.page_content})\n",
    "                    info_article['DOI'] = 'https://doi.org/' + retry_doi if 'https://doi.org/' not in retry_doi else retry_doi\n",
    "\n",
    "                else:\n",
    "                    info_article['DOI'] = 'https://doi.org/' + info_article['DOI'] if 'https://doi.org/' not in info_article['DOI'] else info_article['DOI']\n",
    "\n",
    "                my_json_ref.append(info_article)\n",
    "            except:\n",
    "                not_available.append(re.sub(r\".pdf\",\"\",os.path.basename(pdf)))\n",
    "\n",
    "        # We try to get the html otherwise\n",
    "\n",
    "        print(\"Getting xml from PMC--------------------\\n\")\n",
    "\n",
    "        log_file = os.path.join(output_dir,\"pubmed2pdf_log.txt\")\n",
    "        if os.path.exists(log_file):\n",
    "            to_retrieve = pd.read_csv(log_file, header=None,dtype='str')[0].tolist() # We get those files that could not be retrieved\n",
    "        else:\n",
    "            to_retrieve = []\n",
    "\n",
    "        to_retrieve.extend([re.sub(r'.html','',f)  for f in os.listdir(output_dir) if f.endswith('.html')]) # We add those in html format since they do not have information\n",
    "\n",
    "\n",
    "        # Clean information\n",
    "        for f in os.listdir(output_dir):\n",
    "            if f.endswith('html'):\n",
    "                os.remove(os.path.join(output_dir,f)) # Then, we remove from directory\n",
    "        \n",
    "        print(\"Turn XML into JSON--------------------\\n\")\n",
    "        \n",
    "        for pmid in tqdm.tqdm(to_retrieve, total=len(to_retrieve)):\n",
    "\n",
    "            try:\n",
    "                pmcid = get_pmcid_from_pmid(pmid)\n",
    "                response = requests.get(url = f\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC{pmcid}/\", headers=HEADERS)\n",
    "                soup = BeautifulSoup(response.text, \"html.parser\")\n",
    "                text_paper = soup.find(\"article\").get_text(separator=\"\\n\", strip=True)\n",
    "                info_article = chain.invoke({\"query\": text_paper})\n",
    "\n",
    "                doi_candidate = info_article.get(\"DOI\",None)\n",
    "                if not doi_candidate:\n",
    "                    retry_doi= chain_doi.invoke({\"query\": text_paper})\n",
    "                    info_article['DOI'] = 'https://doi.org/' + retry_doi if 'https://doi.org/' not in retry_doi else retry_doi\n",
    "                        \n",
    "                else:\n",
    "                    info_article['DOI'] = 'https://doi.org/' + info_article['DOI'] if 'https://doi.org/' not in info_article['DOI'] else info_article['DOI']\n",
    "\n",
    "                my_json_ref.append(info_article)\n",
    "                \n",
    "            except:\n",
    "                not_available.append(pmid)\n",
    "\n",
    "            time.sleep(1)\n",
    "    \n",
    "\n",
    "    includedDOI.extend(article['List_references'])\n",
    "    with open(\"info_articles_ref_final.pkl\",\"wb\") as f:\n",
    "        pkl.dump(my_json_ref, f)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9a5c59e",
   "metadata": {},
   "source": [
    "## Create database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "1f940678",
   "metadata": {},
   "outputs": [],
   "source": [
    "splitter = RecursiveCharacterTextSplitter(\n",
    "    chunk_size=1000,\n",
    "    chunk_overlap=100,\n",
    "    separators=[\"\\n\\n\", \"\\n\", \".\", \"!\", \"?\", \" \"]  # smart splitting\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a92a7cfe",
   "metadata": {},
   "outputs": [],
   "source": [
    "info_paper = []\n",
    "\n",
    "for j in my_json_final:\n",
    "\n",
    "    for key, value in j.items():\n",
    "    \n",
    "        if key in ['Abstract', 'Introduction', 'Methods', 'Results', 'Discussion', 'Conclusion',] and value != \"\":\n",
    "\n",
    "            if len(value) > 1200:\n",
    "                chunks = splitter.split_text(value)\n",
    "\n",
    "                for i, c in enumerate(chunks):\n",
    "\n",
    "                    info_paper.append(\n",
    "                        {\n",
    "                            \"chunk_index\":i,\n",
    "                            \"content\":c,\n",
    "                            \"parent\":key,\n",
    "                            \"split\":True,\n",
    "                            \"DOI\":j.get(\"URL\"),\n",
    "                            \"Reference\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Journal')+\", \"+j.get('Publication')\n",
    "                        }\n",
    "                    )\n",
    "            else:\n",
    "\n",
    "                info_paper.append(\n",
    "                        {\n",
    "                            \"chunk_index\":0,\n",
    "                            \"content\":value,\n",
    "                            \"parent\":key,\n",
    "                            \"split\":False,\n",
    "                            \"DOI\":j.get(\"URL\"),\n",
    "                            \"Reference\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Journal')+\", \"+j.get('Publication')\n",
    "                        }\n",
    "                    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "7ada8805",
   "metadata": {},
   "outputs": [],
   "source": [
    "# embedding_function = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")\n",
    "# client = chromadb.HttpClient(host='localhost', port=7000, settings=Settings(allow_reset=True))\n",
    "emb_f= SentenceTransformer(\"avsolatorio/GIST-small-Embedding-v0\", device=\"cuda\")\n",
    "embedding_function2 = HuggingFaceEmbeddings(model_name=\"avsolatorio/GIST-small-Embedding-v0\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "c4a50048",
   "metadata": {},
   "outputs": [],
   "source": [
    "# os.makedirs(\"./chroma_RAG\", exist_ok=True)\n",
    "# client = chromadb.HttpClient(host='localhost', port=7000, settings=Settings(allow_reset=True))\n",
    "client = chromadb.PersistentClient(path=\"./chroma_RAG\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "813ab96d",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'function' object has no attribute 'name'",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mAttributeError\u001b[39m                            Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[51]\u001b[39m\u001b[32m, line 2\u001b[39m\n\u001b[32m      1\u001b[39m \u001b[38;5;66;03m# client.delete_collection(name=\"ReproRAG\")\u001b[39;00m\n\u001b[32m----> \u001b[39m\u001b[32m2\u001b[39m collection = \u001b[43mclient\u001b[49m\u001b[43m.\u001b[49m\u001b[43mget_or_create_collection\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m      3\u001b[39m \u001b[43m    \u001b[49m\u001b[43mname\u001b[49m\u001b[43m=\u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mReproRAG\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[32m      4\u001b[39m \u001b[43m    \u001b[49m\u001b[43membedding_function\u001b[49m\u001b[43m=\u001b[49m\u001b[43membedding_function\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m      5\u001b[39m \u001b[43m    \u001b[49m\u001b[43mmetadata\u001b[49m\u001b[43m=\u001b[49m\u001b[43m{\u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mDescription\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mDatabase containing scientific articles in reproductive field\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[32m      6\u001b[39m \u001b[43m              \u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mCreated\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[38;5;28;43mstr\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mdatetime\u001b[49m\u001b[43m.\u001b[49m\u001b[43mdatetime\u001b[49m\u001b[43m.\u001b[49m\u001b[43mtoday\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\u001b[43m}\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m      7\u001b[39m \u001b[43m    \u001b[49m\u001b[43mconfiguration\u001b[49m\u001b[43m=\u001b[49m\u001b[43m{\u001b[49m\n\u001b[32m      8\u001b[39m \u001b[43m         \u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mhnsw\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43m{\u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mspace\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mcosine\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m}\u001b[49m\n\u001b[32m      9\u001b[39m \u001b[43m    \u001b[49m\u001b[43m}\u001b[49m\n\u001b[32m     10\u001b[39m \n\u001b[32m     11\u001b[39m \u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/chromadb/api/client.py:242\u001b[39m, in \u001b[36mClient.get_or_create_collection\u001b[39m\u001b[34m(self, name, configuration, metadata, embedding_function, data_loader)\u001b[39m\n\u001b[32m    232\u001b[39m model = \u001b[38;5;28mself\u001b[39m._server.get_or_create_collection(\n\u001b[32m    233\u001b[39m     name=name,\n\u001b[32m    234\u001b[39m     metadata=metadata,\n\u001b[32m   (...)\u001b[39m\u001b[32m    237\u001b[39m     configuration=configuration,\n\u001b[32m    238\u001b[39m )\n\u001b[32m    240\u001b[39m persisted_ef_config = model.configuration_json.get(\u001b[33m\"\u001b[39m\u001b[33membedding_function\u001b[39m\u001b[33m\"\u001b[39m)\n\u001b[32m--> \u001b[39m\u001b[32m242\u001b[39m \u001b[43mvalidate_embedding_function_conflict_on_get\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    243\u001b[39m \u001b[43m    \u001b[49m\u001b[43membedding_function\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpersisted_ef_config\u001b[49m\n\u001b[32m    244\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    246\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m Collection(\n\u001b[32m    247\u001b[39m     client=\u001b[38;5;28mself\u001b[39m._server,\n\u001b[32m    248\u001b[39m     model=model,\n\u001b[32m    249\u001b[39m     embedding_function=embedding_function,\n\u001b[32m    250\u001b[39m     data_loader=data_loader,\n\u001b[32m    251\u001b[39m )\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/RAG/lib/python3.11/site-packages/chromadb/api/collection_configuration.py:762\u001b[39m, in \u001b[36mvalidate_embedding_function_conflict_on_get\u001b[39m\u001b[34m(embedding_function, persisted_ef_config)\u001b[39m\n\u001b[32m    756\u001b[39m \u001b[38;5;250m\u001b[39m\u001b[33;03m\"\"\"\u001b[39;00m\n\u001b[32m    757\u001b[39m \u001b[33;03mValidates that there are no conflicting embedding functions between function parameter\u001b[39;00m\n\u001b[32m    758\u001b[39m \u001b[33;03mand collection configuration.\u001b[39;00m\n\u001b[32m    759\u001b[39m \u001b[33;03m\"\"\"\u001b[39;00m\n\u001b[32m    760\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m persisted_ef_config \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;129;01mand\u001b[39;00m embedding_function \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[32m    761\u001b[39m     \u001b[38;5;28;01mif\u001b[39;00m (\n\u001b[32m--> \u001b[39m\u001b[32m762\u001b[39m         \u001b[43membedding_function\u001b[49m\u001b[43m.\u001b[49m\u001b[43mname\u001b[49m() != \u001b[33m\"\u001b[39m\u001b[33mdefault\u001b[39m\u001b[33m\"\u001b[39m\n\u001b[32m    763\u001b[39m         \u001b[38;5;129;01mand\u001b[39;00m persisted_ef_config.get(\u001b[33m\"\u001b[39m\u001b[33mname\u001b[39m\u001b[33m\"\u001b[39m) \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[32m    764\u001b[39m         \u001b[38;5;129;01mand\u001b[39;00m persisted_ef_config.get(\u001b[33m\"\u001b[39m\u001b[33mname\u001b[39m\u001b[33m\"\u001b[39m) != embedding_function.name()\n\u001b[32m    765\u001b[39m     ):\n\u001b[32m    766\u001b[39m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[32m    767\u001b[39m             \u001b[33mf\u001b[39m\u001b[33m\"\u001b[39m\u001b[33mAn embedding function already exists in the collection configuration, and a new one is provided. If this is intentional, please embed documents separately. Embedding function conflict: new: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00membedding_function.name()\u001b[38;5;132;01m}\u001b[39;00m\u001b[33m vs persisted: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mpersisted_ef_config.get(\u001b[33m'\u001b[39m\u001b[33mname\u001b[39m\u001b[33m'\u001b[39m)\u001b[38;5;132;01m}\u001b[39;00m\u001b[33m\"\u001b[39m\n\u001b[32m    768\u001b[39m         )\n\u001b[32m    769\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "\u001b[31mAttributeError\u001b[39m: 'function' object has no attribute 'name'"
     ]
    }
   ],
   "source": [
    "# client.delete_collection(name=\"ReproRAG\")\n",
    "collection = client.get_or_create_collection(\n",
    "    name=\"ReproRAG\",\n",
    "    embedding_function=embedding_function,\n",
    "    metadata={\"Description\": \"Database containing scientific articles in reproductive field\",\n",
    "              \"Created\":str(datetime.datetime.today())},\n",
    "    configuration={\n",
    "         \"hnsw\": {\"space\": \"cosine\"}\n",
    "    }\n",
    "\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "39bcb19c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 2. Prepare documents, metadata, and IDs\n",
    "texts = [chunk[\"content\"] for chunk in info_paper]\n",
    "metadatas = [{\"parent\": chunk[\"parent\"], \"chunk_index\": chunk[\"chunk_index\"],\"DOI\": chunk[\"DOI\"], \"Reference\": chunk[\"Reference\"]} for chunk in info_paper]\n",
    "ids = [str(uuid.uuid1()) for _ in metadatas]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "5dd72a75",
   "metadata": {},
   "outputs": [],
   "source": [
    "db = Chroma.from_texts(\n",
    "    texts=texts,\n",
    "    embedding=embedding_function2,\n",
    "    metadatas=metadatas,\n",
    "    ids=ids,\n",
    "    collection_name=\"ReproRAG\",\n",
    "    persist_directory=\"./chromaRepro\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "eedc1a1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "batch_size = 100  # or smaller if needed\n",
    "\n",
    "for i in range(0, len(texts), batch_size):\n",
    "    batch_texts = texts[i:i+batch_size]\n",
    "    batch_vectors = vectors[i:i+batch_size]\n",
    "    batch_metadatas = metadatas[i:i+batch_size]\n",
    "    batch_ids = ids[i:i+batch_size]\n",
    "    \n",
    "    collection.add(\n",
    "        documents=batch_texts,\n",
    "        embeddings=batch_vectors,\n",
    "        metadatas=batch_metadatas,\n",
    "        ids=batch_ids\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "id": "b77d6130",
   "metadata": {},
   "outputs": [],
   "source": [
    "# tell LangChain to use our client and collection name\n",
    "db = Chroma(\n",
    "    client=client,\n",
    "    collection_name=\"ReproRAG\",\n",
    "    embedding_function=embedding_function2,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "882e1768",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on class Chroma in module langchain_community.vectorstores.chroma:\n",
      "\n",
      "class Chroma(langchain_core.vectorstores.base.VectorStore)\n",
      " |  Chroma(collection_name: 'str' = 'langchain', embedding_function: 'Optional[Embeddings]' = None, persist_directory: 'Optional[str]' = None, client_settings: 'Optional[chromadb.config.Settings]' = None, collection_metadata: 'Optional[Dict]' = None, client: 'Optional[chromadb.Client]' = None, relevance_score_fn: 'Optional[Callable[[float], float]]' = None) -> 'None'\n",
      " |  \n",
      " |  .. deprecated:: 0.2.9 Use ``:class:`~langchain_chroma.Chroma``` instead. It will not be removed until langchain-community==1.0.\n",
      " |  \n",
      " |  `ChromaDB` vector store.\n",
      " |  \n",
      " |  To use, you should have the ``chromadb`` python package installed.\n",
      " |  \n",
      " |  Example:\n",
      " |      .. code-block:: python\n",
      " |  \n",
      " |              from langchain_community.vectorstores import Chroma\n",
      " |              from langchain_community.embeddings.openai import OpenAIEmbeddings\n",
      " |  \n",
      " |              embeddings = OpenAIEmbeddings()\n",
      " |              vectorstore = Chroma(\"langchain_store\", embeddings)\n",
      " |  \n",
      " |  Method resolution order:\n",
      " |      Chroma\n",
      " |      langchain_core.vectorstores.base.VectorStore\n",
      " |      abc.ABC\n",
      " |      builtins.object\n",
      " |  \n",
      " |  Methods defined here:\n",
      " |  \n",
      " |  __init__(self, collection_name: 'str' = 'langchain', embedding_function: 'Optional[Embeddings]' = None, persist_directory: 'Optional[str]' = None, client_settings: 'Optional[chromadb.config.Settings]' = None, collection_metadata: 'Optional[Dict]' = None, client: 'Optional[chromadb.Client]' = None, relevance_score_fn: 'Optional[Callable[[float], float]]' = None) -> 'None'\n",
      " |      Initialize with a Chroma client.\n",
      " |  \n",
      " |  __len__(self) -> 'int'\n",
      " |      Count the number of documents in the collection.\n",
      " |  \n",
      " |  add_images(self, uris: 'List[str]', metadatas: 'Optional[List[dict]]' = None, ids: 'Optional[List[str]]' = None, **kwargs: 'Any') -> 'List[str]'\n",
      " |      Run more images through the embeddings and add to the vectorstore.\n",
      " |      \n",
      " |      Args:\n",
      " |          uris List[str]: File path to the image.\n",
      " |          metadatas (Optional[List[dict]], optional): Optional list of metadatas.\n",
      " |          ids (Optional[List[str]], optional): Optional list of IDs.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List[str]: List of IDs of the added images.\n",
      " |  \n",
      " |  add_texts(self, texts: 'Iterable[str]', metadatas: 'Optional[List[dict]]' = None, ids: 'Optional[List[str]]' = None, **kwargs: 'Any') -> 'List[str]'\n",
      " |      Run more texts through the embeddings and add to the vectorstore.\n",
      " |      \n",
      " |      Args:\n",
      " |          texts (Iterable[str]): Texts to add to the vectorstore.\n",
      " |          metadatas (Optional[List[dict]], optional): Optional list of metadatas.\n",
      " |          ids (Optional[List[str]], optional): Optional list of IDs.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List[str]: List of IDs of the added texts.\n",
      " |  \n",
      " |  delete(self, ids: 'Optional[List[str]]' = None, **kwargs: 'Any') -> 'None'\n",
      " |      Delete by vector IDs.\n",
      " |      \n",
      " |      Args:\n",
      " |          ids: List of ids to delete.\n",
      " |  \n",
      " |  delete_collection(self) -> 'None'\n",
      " |      Delete the collection.\n",
      " |  \n",
      " |  encode_image(self, uri: 'str') -> 'str'\n",
      " |      Get base64 string from image URI.\n",
      " |  \n",
      " |  get(self, ids: 'Optional[OneOrMany[ID]]' = None, where: 'Optional[Where]' = None, limit: 'Optional[int]' = None, offset: 'Optional[int]' = None, where_document: 'Optional[WhereDocument]' = None, include: 'Optional[List[str]]' = None) -> 'Dict[str, Any]'\n",
      " |      Gets the collection.\n",
      " |      \n",
      " |      Args:\n",
      " |          ids: The ids of the embeddings to get. Optional.\n",
      " |          where: A Where type dict used to filter results by.\n",
      " |                 E.g. `{\"color\" : \"red\", \"price\": 4.20}`. Optional.\n",
      " |          limit: The number of documents to return. Optional.\n",
      " |          offset: The offset to start returning results from.\n",
      " |                  Useful for paging results with limit. Optional.\n",
      " |          where_document: A WhereDocument type dict used to filter by the documents.\n",
      " |                          E.g. `{$contains: \"hello\"}`. Optional.\n",
      " |          include: A list of what to include in the results.\n",
      " |                   Can contain `\"embeddings\"`, `\"metadatas\"`, `\"documents\"`.\n",
      " |                   Ids are always included.\n",
      " |                   Defaults to `[\"metadatas\", \"documents\"]`. Optional.\n",
      " |  \n",
      " |  max_marginal_relevance_search(self, query: 'str', k: 'int' = 4, fetch_k: 'int' = 20, lambda_mult: 'float' = 0.5, filter: 'Optional[Dict[str, str]]' = None, where_document: 'Optional[Dict[str, str]]' = None, **kwargs: 'Any') -> 'List[Document]'\n",
      " |      Return docs selected using the maximal marginal relevance.\n",
      " |      Maximal marginal relevance optimizes for similarity to query AND diversity\n",
      " |      among selected documents.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Text to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          fetch_k: Number of Documents to fetch to pass to MMR algorithm.\n",
      " |          lambda_mult: Number between 0 and 1 that determines the degree\n",
      " |                      of diversity among the results with 0 corresponding\n",
      " |                      to maximum diversity and 1 to minimum diversity.\n",
      " |                      Defaults to 0.5.\n",
      " |          filter (Optional[Dict[str, str]]): Filter by metadata. Defaults to None.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Documents selected by maximal marginal relevance.\n",
      " |  \n",
      " |  max_marginal_relevance_search_by_vector(self, embedding: 'List[float]', k: 'int' = 4, fetch_k: 'int' = 20, lambda_mult: 'float' = 0.5, filter: 'Optional[Dict[str, str]]' = None, where_document: 'Optional[Dict[str, str]]' = None, **kwargs: 'Any') -> 'List[Document]'\n",
      " |      Return docs selected using the maximal marginal relevance.\n",
      " |      Maximal marginal relevance optimizes for similarity to query AND diversity\n",
      " |      among selected documents.\n",
      " |      \n",
      " |      Args:\n",
      " |          embedding: Embedding to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          fetch_k: Number of Documents to fetch to pass to MMR algorithm.\n",
      " |          lambda_mult: Number between 0 and 1 that determines the degree\n",
      " |                      of diversity among the results with 0 corresponding\n",
      " |                      to maximum diversity and 1 to minimum diversity.\n",
      " |                      Defaults to 0.5.\n",
      " |          filter (Optional[Dict[str, str]]): Filter by metadata. Defaults to None.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Documents selected by maximal marginal relevance.\n",
      " |  \n",
      " |  persist(self) -> 'None'\n",
      " |      .. deprecated:: 0.1.17 Since Chroma 0.4.x the manual persistence method is no longer supported as docs are automatically persisted. It will not be removed until langchain-community==1.0.\n",
      " |      \n",
      " |      Persist the collection.\n",
      " |      \n",
      " |      This can be used to explicitly persist the data to disk.\n",
      " |      It will also be called automatically when the object is destroyed.\n",
      " |      \n",
      " |      Since Chroma 0.4.x the manual persistence method is no longer\n",
      " |      supported as docs are automatically persisted.\n",
      " |  \n",
      " |  similarity_search(self, query: 'str', k: 'int' = 4, filter: 'Optional[Dict[str, str]]' = None, **kwargs: 'Any') -> 'List[Document]'\n",
      " |      Run similarity search with Chroma.\n",
      " |      \n",
      " |      Args:\n",
      " |          query (str): Query text to search for.\n",
      " |          k (int): Number of results to return. Defaults to 4.\n",
      " |          filter (Optional[Dict[str, str]]): Filter by metadata. Defaults to None.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List[Document]: List of documents most similar to the query text.\n",
      " |  \n",
      " |  similarity_search_by_image(self, uri: 'str', k: 'int' = 4, filter: 'Optional[Dict[str, str]]' = None, **kwargs: 'Any') -> 'List[Document]'\n",
      " |      Search for similar images based on the given image URI.\n",
      " |      \n",
      " |      Args:\n",
      " |          uri (str): URI of the image to search for.\n",
      " |          k (int, optional): Number of results to return. Defaults to DEFAULT_K.\n",
      " |          filter (Optional[Dict[str, str]], optional): Filter by metadata.\n",
      " |          **kwargs (Any): Additional arguments to pass to function.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Images most similar to the provided image.\n",
      " |          Each element in list is a Langchain Document Object.\n",
      " |          The page content is b64 encoded image, metadata is default or\n",
      " |          as defined by user.\n",
      " |      \n",
      " |      Raises:\n",
      " |          ValueError: If the embedding function does not support image embeddings.\n",
      " |  \n",
      " |  similarity_search_by_image_with_relevance_score(self, uri: 'str', k: 'int' = 4, filter: 'Optional[Dict[str, str]]' = None, **kwargs: 'Any') -> 'List[Tuple[Document, float]]'\n",
      " |      Search for similar images based on the given image URI.\n",
      " |      \n",
      " |      Args:\n",
      " |          uri (str): URI of the image to search for.\n",
      " |          k (int, optional): Number of results to return.\n",
      " |          Defaults to DEFAULT_K.\n",
      " |          filter (Optional[Dict[str, str]], optional): Filter by metadata.\n",
      " |          **kwargs (Any): Additional arguments to pass to function.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List[Tuple[Document, float]]: List of tuples containing documents similar\n",
      " |          to the query image and their similarity scores.\n",
      " |          0th element in each tuple is a Langchain Document Object.\n",
      " |          The page content is b64 encoded img, metadata is default or defined by user.\n",
      " |      \n",
      " |      Raises:\n",
      " |          ValueError: If the embedding function does not support image embeddings.\n",
      " |  \n",
      " |  similarity_search_by_vector(self, embedding: 'List[float]', k: 'int' = 4, filter: 'Optional[Dict[str, str]]' = None, where_document: 'Optional[Dict[str, str]]' = None, **kwargs: 'Any') -> 'List[Document]'\n",
      " |      Return docs most similar to embedding vector.\n",
      " |      Args:\n",
      " |          embedding (List[float]): Embedding to look up documents similar to.\n",
      " |          k (int): Number of Documents to return. Defaults to 4.\n",
      " |          filter (Optional[Dict[str, str]]): Filter by metadata. Defaults to None.\n",
      " |      Returns:\n",
      " |          List of Documents most similar to the query vector.\n",
      " |  \n",
      " |  similarity_search_by_vector_with_relevance_scores(self, embedding: 'List[float]', k: 'int' = 4, filter: 'Optional[Dict[str, str]]' = None, where_document: 'Optional[Dict[str, str]]' = None, **kwargs: 'Any') -> 'List[Tuple[Document, float]]'\n",
      " |      Return docs most similar to embedding vector and similarity score.\n",
      " |      \n",
      " |      Args:\n",
      " |          embedding (List[float]): Embedding to look up documents similar to.\n",
      " |          k (int): Number of Documents to return. Defaults to 4.\n",
      " |          filter (Optional[Dict[str, str]]): Filter by metadata. Defaults to None.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List[Tuple[Document, float]]: List of documents most similar to\n",
      " |          the query text and cosine distance in float for each.\n",
      " |          Lower score represents more similarity.\n",
      " |  \n",
      " |  similarity_search_with_score(self, query: 'str', k: 'int' = 4, filter: 'Optional[Dict[str, str]]' = None, where_document: 'Optional[Dict[str, str]]' = None, **kwargs: 'Any') -> 'List[Tuple[Document, float]]'\n",
      " |      Run similarity search with Chroma with distance.\n",
      " |      \n",
      " |      Args:\n",
      " |          query (str): Query text to search for.\n",
      " |          k (int): Number of results to return. Defaults to 4.\n",
      " |          filter (Optional[Dict[str, str]]): Filter by metadata. Defaults to None.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List[Tuple[Document, float]]: List of documents most similar to\n",
      " |          the query text and cosine distance in float for each.\n",
      " |          Lower score represents more similarity.\n",
      " |  \n",
      " |  update_document(self, document_id: 'str', document: 'Document') -> 'None'\n",
      " |      Update a document in the collection.\n",
      " |      \n",
      " |      Args:\n",
      " |          document_id (str): ID of the document to update.\n",
      " |          document (Document): Document to update.\n",
      " |  \n",
      " |  update_documents(self, ids: 'List[str]', documents: 'List[Document]') -> 'None'\n",
      " |      Update a document in the collection.\n",
      " |      \n",
      " |      Args:\n",
      " |          ids (List[str]): List of ids of the document to update.\n",
      " |          documents (List[Document]): List of documents to update.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Class methods defined here:\n",
      " |  \n",
      " |  from_documents(documents: 'List[Document]', embedding: 'Optional[Embeddings]' = None, ids: 'Optional[List[str]]' = None, collection_name: 'str' = 'langchain', persist_directory: 'Optional[str]' = None, client_settings: 'Optional[chromadb.config.Settings]' = None, client: 'Optional[chromadb.Client]' = None, collection_metadata: 'Optional[Dict]' = None, **kwargs: 'Any') -> 'Chroma' from abc.ABCMeta\n",
      " |      Create a Chroma vectorstore from a list of documents.\n",
      " |      \n",
      " |      If a persist_directory is specified, the collection will be persisted there.\n",
      " |      Otherwise, the data will be ephemeral in-memory.\n",
      " |      \n",
      " |      Args:\n",
      " |          collection_name (str): Name of the collection to create.\n",
      " |          persist_directory (Optional[str]): Directory to persist the collection.\n",
      " |          ids (Optional[List[str]]): List of document IDs. Defaults to None.\n",
      " |          documents (List[Document]): List of documents to add to the vectorstore.\n",
      " |          embedding (Optional[Embeddings]): Embedding function. Defaults to None.\n",
      " |          client_settings (Optional[chromadb.config.Settings]): Chroma client settings\n",
      " |          collection_metadata (Optional[Dict]): Collection configurations.\n",
      " |                                                Defaults to None.\n",
      " |      \n",
      " |      Returns:\n",
      " |          Chroma: Chroma vectorstore.\n",
      " |  \n",
      " |  from_texts(texts: 'List[str]', embedding: 'Optional[Embeddings]' = None, metadatas: 'Optional[List[dict]]' = None, ids: 'Optional[List[str]]' = None, collection_name: 'str' = 'langchain', persist_directory: 'Optional[str]' = None, client_settings: 'Optional[chromadb.config.Settings]' = None, client: 'Optional[chromadb.Client]' = None, collection_metadata: 'Optional[Dict]' = None, **kwargs: 'Any') -> 'Chroma' from abc.ABCMeta\n",
      " |      Create a Chroma vectorstore from a raw documents.\n",
      " |      \n",
      " |      If a persist_directory is specified, the collection will be persisted there.\n",
      " |      Otherwise, the data will be ephemeral in-memory.\n",
      " |      \n",
      " |      Args:\n",
      " |          texts (List[str]): List of texts to add to the collection.\n",
      " |          collection_name (str): Name of the collection to create.\n",
      " |          persist_directory (Optional[str]): Directory to persist the collection.\n",
      " |          embedding (Optional[Embeddings]): Embedding function. Defaults to None.\n",
      " |          metadatas (Optional[List[dict]]): List of metadatas. Defaults to None.\n",
      " |          ids (Optional[List[str]]): List of document IDs. Defaults to None.\n",
      " |          client_settings (Optional[chromadb.config.Settings]): Chroma client settings\n",
      " |          collection_metadata (Optional[Dict]): Collection configurations.\n",
      " |                                                Defaults to None.\n",
      " |      \n",
      " |      Returns:\n",
      " |          Chroma: Chroma vectorstore.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Readonly properties defined here:\n",
      " |  \n",
      " |  embeddings\n",
      " |      Access the query embedding object if available.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data and other attributes defined here:\n",
      " |  \n",
      " |  __abstractmethods__ = frozenset()\n",
      " |  \n",
      " |  __annotations__ = {'_LANGCHAIN_DEFAULT_COLLECTION_NAME': 'str'}\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Methods inherited from langchain_core.vectorstores.base.VectorStore:\n",
      " |  \n",
      " |  async aadd_documents(self, documents: 'list[Document]', **kwargs: 'Any') -> 'list[str]'\n",
      " |      Async run more documents through the embeddings and add to the vectorstore.\n",
      " |      \n",
      " |      Args:\n",
      " |          documents: Documents to add to the vectorstore.\n",
      " |          kwargs: Additional keyword arguments.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of IDs of the added texts.\n",
      " |      \n",
      " |      Raises:\n",
      " |          ValueError: If the number of IDs does not match the number of documents.\n",
      " |  \n",
      " |  async aadd_texts(self, texts: 'Iterable[str]', metadatas: 'Optional[list[dict]]' = None, *, ids: 'Optional[list[str]]' = None, **kwargs: 'Any') -> 'list[str]'\n",
      " |      Async run more texts through the embeddings and add to the vectorstore.\n",
      " |      \n",
      " |      Args:\n",
      " |          texts: Iterable of strings to add to the vectorstore.\n",
      " |          metadatas: Optional list of metadatas associated with the texts.\n",
      " |              Default is None.\n",
      " |          ids: Optional list\n",
      " |          **kwargs: vectorstore specific parameters.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of ids from adding the texts into the vectorstore.\n",
      " |      \n",
      " |      Raises:\n",
      " |          ValueError: If the number of metadatas does not match the number of texts.\n",
      " |          ValueError: If the number of ids does not match the number of texts.\n",
      " |  \n",
      " |  add_documents(self, documents: 'list[Document]', **kwargs: 'Any') -> 'list[str]'\n",
      " |      Add or update documents in the vectorstore.\n",
      " |      \n",
      " |      Args:\n",
      " |          documents: Documents to add to the vectorstore.\n",
      " |          kwargs: Additional keyword arguments.\n",
      " |              if kwargs contains ids and documents contain ids,\n",
      " |              the ids in the kwargs will receive precedence.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of IDs of the added texts.\n",
      " |      \n",
      " |      Raises:\n",
      " |          ValueError: If the number of ids does not match the number of documents.\n",
      " |  \n",
      " |  async adelete(self, ids: 'Optional[list[str]]' = None, **kwargs: 'Any') -> 'Optional[bool]'\n",
      " |      Async delete by vector ID or other criteria.\n",
      " |      \n",
      " |      Args:\n",
      " |          ids: List of ids to delete. If None, delete all. Default is None.\n",
      " |          **kwargs: Other keyword arguments that subclasses might use.\n",
      " |      \n",
      " |      Returns:\n",
      " |          Optional[bool]: True if deletion is successful,\n",
      " |          False otherwise, None if not implemented.\n",
      " |  \n",
      " |  async aget_by_ids(self, ids: 'Sequence[str]', /) -> 'list[Document]'\n",
      " |      Async get documents by their IDs.\n",
      " |      \n",
      " |      The returned documents are expected to have the ID field set to the ID of the\n",
      " |      document in the vector store.\n",
      " |      \n",
      " |      Fewer documents may be returned than requested if some IDs are not found or\n",
      " |      if there are duplicated IDs.\n",
      " |      \n",
      " |      Users should not assume that the order of the returned documents matches\n",
      " |      the order of the input IDs. Instead, users should rely on the ID field of the\n",
      " |      returned documents.\n",
      " |      \n",
      " |      This method should **NOT** raise exceptions if no documents are found for\n",
      " |      some IDs.\n",
      " |      \n",
      " |      Args:\n",
      " |          ids: List of ids to retrieve.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Documents.\n",
      " |      \n",
      " |      .. versionadded:: 0.2.11\n",
      " |  \n",
      " |  async amax_marginal_relevance_search(self, query: 'str', k: 'int' = 4, fetch_k: 'int' = 20, lambda_mult: 'float' = 0.5, **kwargs: 'Any') -> 'list[Document]'\n",
      " |      Async return docs selected using the maximal marginal relevance.\n",
      " |      \n",
      " |      Maximal marginal relevance optimizes for similarity to query AND diversity\n",
      " |      among selected documents.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Text to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          fetch_k: Number of Documents to fetch to pass to MMR algorithm.\n",
      " |              Default is 20.\n",
      " |          lambda_mult: Number between 0 and 1 that determines the degree\n",
      " |              of diversity among the results with 0 corresponding\n",
      " |              to maximum diversity and 1 to minimum diversity.\n",
      " |              Defaults to 0.5.\n",
      " |          **kwargs: Arguments to pass to the search method.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Documents selected by maximal marginal relevance.\n",
      " |  \n",
      " |  async amax_marginal_relevance_search_by_vector(self, embedding: 'list[float]', k: 'int' = 4, fetch_k: 'int' = 20, lambda_mult: 'float' = 0.5, **kwargs: 'Any') -> 'list[Document]'\n",
      " |      Async return docs selected using the maximal marginal relevance.\n",
      " |      \n",
      " |      Maximal marginal relevance optimizes for similarity to query AND diversity\n",
      " |      among selected documents.\n",
      " |      \n",
      " |      Args:\n",
      " |          embedding: Embedding to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          fetch_k: Number of Documents to fetch to pass to MMR algorithm.\n",
      " |              Default is 20.\n",
      " |          lambda_mult: Number between 0 and 1 that determines the degree\n",
      " |              of diversity among the results with 0 corresponding\n",
      " |              to maximum diversity and 1 to minimum diversity.\n",
      " |              Defaults to 0.5.\n",
      " |          **kwargs: Arguments to pass to the search method.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Documents selected by maximal marginal relevance.\n",
      " |  \n",
      " |  as_retriever(self, **kwargs: 'Any') -> 'VectorStoreRetriever'\n",
      " |      Return VectorStoreRetriever initialized from this VectorStore.\n",
      " |      \n",
      " |      Args:\n",
      " |          **kwargs: Keyword arguments to pass to the search function.\n",
      " |              Can include:\n",
      " |              search_type (Optional[str]): Defines the type of search that\n",
      " |                  the Retriever should perform.\n",
      " |                  Can be \"similarity\" (default), \"mmr\", or\n",
      " |                  \"similarity_score_threshold\".\n",
      " |              search_kwargs (Optional[Dict]): Keyword arguments to pass to the\n",
      " |                  search function. Can include things like:\n",
      " |                      k: Amount of documents to return (Default: 4)\n",
      " |                      score_threshold: Minimum relevance threshold\n",
      " |                          for similarity_score_threshold\n",
      " |                      fetch_k: Amount of documents to pass to MMR algorithm\n",
      " |                          (Default: 20)\n",
      " |                      lambda_mult: Diversity of results returned by MMR;\n",
      " |                          1 for minimum diversity and 0 for maximum. (Default: 0.5)\n",
      " |                      filter: Filter by document metadata\n",
      " |      \n",
      " |      Returns:\n",
      " |          VectorStoreRetriever: Retriever class for VectorStore.\n",
      " |      \n",
      " |      Examples:\n",
      " |      \n",
      " |      .. code-block:: python\n",
      " |      \n",
      " |          # Retrieve more documents with higher diversity\n",
      " |          # Useful if your dataset has many similar documents\n",
      " |          docsearch.as_retriever(\n",
      " |              search_type=\"mmr\",\n",
      " |              search_kwargs={'k': 6, 'lambda_mult': 0.25}\n",
      " |          )\n",
      " |      \n",
      " |          # Fetch more documents for the MMR algorithm to consider\n",
      " |          # But only return the top 5\n",
      " |          docsearch.as_retriever(\n",
      " |              search_type=\"mmr\",\n",
      " |              search_kwargs={'k': 5, 'fetch_k': 50}\n",
      " |          )\n",
      " |      \n",
      " |          # Only retrieve documents that have a relevance score\n",
      " |          # Above a certain threshold\n",
      " |          docsearch.as_retriever(\n",
      " |              search_type=\"similarity_score_threshold\",\n",
      " |              search_kwargs={'score_threshold': 0.8}\n",
      " |          )\n",
      " |      \n",
      " |          # Only get the single most similar document from the dataset\n",
      " |          docsearch.as_retriever(search_kwargs={'k': 1})\n",
      " |      \n",
      " |          # Use a filter to only retrieve documents from a specific paper\n",
      " |          docsearch.as_retriever(\n",
      " |              search_kwargs={'filter': {'paper_title':'GPT-4 Technical Report'}}\n",
      " |          )\n",
      " |  \n",
      " |  async asearch(self, query: 'str', search_type: 'str', **kwargs: 'Any') -> 'list[Document]'\n",
      " |      Async return docs most similar to query using a specified search type.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Input text.\n",
      " |          search_type: Type of search to perform. Can be \"similarity\",\n",
      " |              \"mmr\", or \"similarity_score_threshold\".\n",
      " |          **kwargs: Arguments to pass to the search method.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Documents most similar to the query.\n",
      " |      \n",
      " |      Raises:\n",
      " |          ValueError: If search_type is not one of \"similarity\",\n",
      " |              \"mmr\", or \"similarity_score_threshold\".\n",
      " |  \n",
      " |  async asimilarity_search(self, query: 'str', k: 'int' = 4, **kwargs: 'Any') -> 'list[Document]'\n",
      " |      Async return docs most similar to query.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Input text.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          **kwargs: Arguments to pass to the search method.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Documents most similar to the query.\n",
      " |  \n",
      " |  async asimilarity_search_by_vector(self, embedding: 'list[float]', k: 'int' = 4, **kwargs: 'Any') -> 'list[Document]'\n",
      " |      Async return docs most similar to embedding vector.\n",
      " |      \n",
      " |      Args:\n",
      " |          embedding: Embedding to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          **kwargs: Arguments to pass to the search method.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Documents most similar to the query vector.\n",
      " |  \n",
      " |  async asimilarity_search_with_relevance_scores(self, query: 'str', k: 'int' = 4, **kwargs: 'Any') -> 'list[tuple[Document, float]]'\n",
      " |      Async return docs and relevance scores in the range [0, 1].\n",
      " |      \n",
      " |      0 is dissimilar, 1 is most similar.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Input text.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          **kwargs: kwargs to be passed to similarity search. Should include:\n",
      " |              score_threshold: Optional, a floating point value between 0 to 1 to\n",
      " |                  filter the resulting set of retrieved docs\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Tuples of (doc, similarity_score)\n",
      " |  \n",
      " |  async asimilarity_search_with_score(self, *args: 'Any', **kwargs: 'Any') -> 'list[tuple[Document, float]]'\n",
      " |      Async run similarity search with distance.\n",
      " |      \n",
      " |      Args:\n",
      " |          *args: Arguments to pass to the search method.\n",
      " |          **kwargs: Arguments to pass to the search method.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Tuples of (doc, similarity_score).\n",
      " |  \n",
      " |  get_by_ids(self, ids: 'Sequence[str]', /) -> 'list[Document]'\n",
      " |      Get documents by their IDs.\n",
      " |      \n",
      " |      The returned documents are expected to have the ID field set to the ID of the\n",
      " |      document in the vector store.\n",
      " |      \n",
      " |      Fewer documents may be returned than requested if some IDs are not found or\n",
      " |      if there are duplicated IDs.\n",
      " |      \n",
      " |      Users should not assume that the order of the returned documents matches\n",
      " |      the order of the input IDs. Instead, users should rely on the ID field of the\n",
      " |      returned documents.\n",
      " |      \n",
      " |      This method should **NOT** raise exceptions if no documents are found for\n",
      " |      some IDs.\n",
      " |      \n",
      " |      Args:\n",
      " |          ids: List of ids to retrieve.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Documents.\n",
      " |      \n",
      " |      .. versionadded:: 0.2.11\n",
      " |  \n",
      " |  search(self, query: 'str', search_type: 'str', **kwargs: 'Any') -> 'list[Document]'\n",
      " |      Return docs most similar to query using a specified search type.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Input text\n",
      " |          search_type: Type of search to perform. Can be \"similarity\",\n",
      " |              \"mmr\", or \"similarity_score_threshold\".\n",
      " |          **kwargs: Arguments to pass to the search method.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Documents most similar to the query.\n",
      " |      \n",
      " |      Raises:\n",
      " |          ValueError: If search_type is not one of \"similarity\",\n",
      " |              \"mmr\", or \"similarity_score_threshold\".\n",
      " |  \n",
      " |  similarity_search_with_relevance_scores(self, query: 'str', k: 'int' = 4, **kwargs: 'Any') -> 'list[tuple[Document, float]]'\n",
      " |      Return docs and relevance scores in the range [0, 1].\n",
      " |      \n",
      " |      0 is dissimilar, 1 is most similar.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Input text.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          **kwargs: kwargs to be passed to similarity search. Should include:\n",
      " |              score_threshold: Optional, a floating point value between 0 to 1 to\n",
      " |                  filter the resulting set of retrieved docs.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Tuples of (doc, similarity_score).\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Class methods inherited from langchain_core.vectorstores.base.VectorStore:\n",
      " |  \n",
      " |  async afrom_documents(documents: 'list[Document]', embedding: 'Embeddings', **kwargs: 'Any') -> 'Self' from abc.ABCMeta\n",
      " |      Async return VectorStore initialized from documents and embeddings.\n",
      " |      \n",
      " |      Args:\n",
      " |          documents: List of Documents to add to the vectorstore.\n",
      " |          embedding: Embedding function to use.\n",
      " |          kwargs: Additional keyword arguments.\n",
      " |      \n",
      " |      Returns:\n",
      " |          VectorStore: VectorStore initialized from documents and embeddings.\n",
      " |  \n",
      " |  async afrom_texts(texts: 'list[str]', embedding: 'Embeddings', metadatas: 'Optional[list[dict]]' = None, *, ids: 'Optional[list[str]]' = None, **kwargs: 'Any') -> 'Self' from abc.ABCMeta\n",
      " |      Async return VectorStore initialized from texts and embeddings.\n",
      " |      \n",
      " |      Args:\n",
      " |          texts: Texts to add to the vectorstore.\n",
      " |          embedding: Embedding function to use.\n",
      " |          metadatas: Optional list of metadatas associated with the texts.\n",
      " |              Default is None.\n",
      " |          ids: Optional list of IDs associated with the texts.\n",
      " |          kwargs: Additional keyword arguments.\n",
      " |      \n",
      " |      Returns:\n",
      " |          VectorStore: VectorStore initialized from texts and embeddings.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors inherited from langchain_core.vectorstores.base.VectorStore:\n",
      " |  \n",
      " |  __dict__\n",
      " |      dictionary for instance variables (if defined)\n",
      " |  \n",
      " |  __weakref__\n",
      " |      list of weak references to the object (if defined)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(Chroma)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "a081d3bf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Document(metadata={'chunk_index': 7, 'DOI': 'https://doi.org/10.1186/s12958-023-01131-4', 'parent': 'Discussion', 'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023'}, page_content='. Consider-ing these attributed functions and its variable expression throughout the menstrual cycle, we propose that CTCF exerts an inhibitory role in endometrial tissue during the PF phase of the menstrual cycle. With the significant downregulation of CTCF in the secretory endometrium, dually validated in-silico and experimentally herein, its inhibited genes would be derepressed and become tran-scriptionally active during the WOI. This interpretation supports previous findings from our group demonstrat-ing that, during the WOI, a global transcriptional dere-pression may be required for implantation and early embryo development [63]. Transcriptional derepression has been associated with multiple human disease states and should be investigated further within the context of endometrial-factor infertility. Despite previous asso-ciations of CTCF with endometriosis [67], its implica-tion in endometrial progression and function has not been directly proposed until now'),\n",
       " Document(metadata={'parent': 'Results', 'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023', 'chunk_index': 7, 'DOI': 'https://doi.org/10.1186/s12958-023-01131-4'}, page_content='.70E-11) and from the ESE to MSE phase (FDR = 9.80E-15). These findings supported the role of these TFs at the beginning of the menstrual cycle and during the MSE phase (Fig. 3 B). These in-silico validations were then prospec-tively corroborated by RT-qPCR analysis in an inde-pendent sample set of 20 endometrial biopsies. We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with signifi-cant changes between the endometrial phases. Nota-bly, CTCF expression was decreased between the PF/ ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C).'),\n",
       " Document(metadata={'parent': 'Discussion', 'chunk_index': 8, 'DOI': 'https://doi.org/10.1186/s12958-023-01131-4', 'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023'}, page_content='. Nevertheless, the role of CTCF in endometrial receptivity has been revealed through its interaction with HOXA10, a gene mainly expressed in endometrium involved in functions such as endometrial proliferation and differentiation, the forma-tion of pinopodes or embryo implantation [68]. Indeed, this study proposed that an overexpression of CTCF can lead to a drop in HOXA10 expression, affecting endome-trial proliferation and endometrial function and corroborating the fact that CTCF must decrease its expression as we can observe in our findings. GATA6, the second prioritized TF that overlapped across the gene lists, is a transcription factor belong-ing to the GATA family, a highly conserved family of six zinc finger proteins [69]. Besides its essential role in embryonic development [70], GATA6 plays a key role in endometriosis and regulates steroidogenic genes [71]'),\n",
       " Document(metadata={'chunk_index': 6, 'parent': 'Results', 'DOI': 'https://doi.org/10.1186/s12958-023-01131-4', 'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023'}, page_content='. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53. Notably, GATA6 was selected due to novel association with endometrial pro-gression and function. To infer the expression of CTCF and GATA6 in endometrial tissue throughout the menstrual cycle, we analysed their expression using a relevant endo-metrial dataset recently created by our group [30] (Table S1 ). We observed that, relative to the global expression of endometrial genes, CTCF was broadly expressed while GATA6 was moderately expressed. Indeed, CTCF was significantly down-regulated between the PF and ESE phases (FDR = 2.10E-03) and ESE to the MSE phase (FDR = 2.10E-04), while GATA6 expression increased from the PF to the MSE phase (FDR = 1.70E-11) and from the ESE to MSE phase (FDR = 9.80E-15)')]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db.similarity_search(\"What is the function of CTCF in endometrium?\", 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "bb456fa9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'parent': 'Discussion', 'DOI': 'https://doi.org/10.1186/s12958-023-01131-4', 'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023', 'chunk_index': 7}\n",
      "Discussion\n",
      "('This data-driven approach exposed the common tran-scriptional regulators '\n",
      " 'among 19 studies who proposed variable biomarkers of endometrial progression '\n",
      " 'and function. In this study, we focused on understanding the relative '\n",
      " 'contribution of both the hormonal and nonhormonal regulation, from an '\n",
      " 'alternative holistic per-spective. We applied data-driven hypothesis '\n",
      " 'research that, unlike the traditional scientific method, allowed us to '\n",
      " 'generate new hypotheses based on all available bio-logically-relevant '\n",
      " 'knowledge [16], observing the molec-ular relationships from a wider scale '\n",
      " 'view. Besides, some traditional molecular procedures such as PCRs are also '\n",
      " 'performed to corroborate the insight uncovered with this approach. We '\n",
      " 'highlighted a larger influence of progesterone-related genes and TFs in '\n",
      " 'endome-trial progression and observed a larger contribution of miRNAs and '\n",
      " 'estrogen-related genes in endometrial pathology. Furthermore, we unveiled '\n",
      " 'CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsamiR-103a-3p and '\n",
      " 'hsa-miR-128-3p as master regulators in the endometrium, and these findings '\n",
      " 'were reinforced by the dynamic changes in their expression throughout the '\n",
      " 'menstrual cycle (dually validated with in-silico and prospective analyses). '\n",
      " 'The menstrual cycle is perceived to be under tight regulation by the ovarian '\n",
      " 'hormones, progesterone and estrogen [48]. Supporting the role of '\n",
      " 'progester-one in the secretory endometrium [49], and reinforc-ing its '\n",
      " 'indispensable role in endometrial regulation [50, 51], we found progesterone '\n",
      " 'and its related genes had a larger influence in menstrual cycle progression '\n",
      " 'than estrogen. As the RIF-related gene lists reported by Altmae2010 and Koot '\n",
      " 'were generally controlled by estrogen, this further suggested that while '\n",
      " 'progester-one could be crucial for a suitable endometrial progres-sion and '\n",
      " 'mid-secretory phase acquisition, estrogenic irregularities may be more '\n",
      " 'conducive to endometrial function and molecular pathology. Indeed, this '\n",
      " 'hypoth-esis corroborated previous reports of most uterine dis-orders being '\n",
      " 'estrogen-dependent [44, 52]. Supporting the findings from another recent '\n",
      " 'study from our group, where the Altmae2010 and Koot signatures were '\n",
      " 'classi-fied as highly-predictive signatures to identify endome-trial '\n",
      " 'pathology in comparison with a prediction model that detected endometrial '\n",
      " 'progression [7 ]. This study distinguished these signatures (related to '\n",
      " 'implantation failure and unexplained infertility) based on the mag-nitude of '\n",
      " 'their miRNA and estrogen-regulation, which contrasted with the prominent '\n",
      " 'influence of the TFs over the remaining signatures more related to '\n",
      " 'endometrial progression. All these facts reinforce the hypothesis that '\n",
      " 'implantation failure is mainly caused by impaired miRNA expression [53]. '\n",
      " 'Although we observed a clear hormonal influence in numerous annotated '\n",
      " 'regulators, this study transcended the simplistic paradigm of the bihormonal '\n",
      " 'regulation of endometrial function by identifying a plethora of medi-ators, '\n",
      " 'including 595 miRNAs and 311 TFs, that were previously unrelated to these '\n",
      " 'ovarian hormones. These findings set the foundation for new discoveries, '\n",
      " 'reveal-ing alternative pathways and new actors in the regula-tion of the '\n",
      " 'human endometrium, and deepening our understanding of the complex regulatory '\n",
      " 'mechanisms behind endometrial progression and unexplained infertility. '\n",
      " 'MicroRNAs are well known to be involved in embryo implantation [12] and '\n",
      " 'receptivity control [54, 55]. Using network analysis, we unveiled the 10 '\n",
      " 'common miR-NAs regulating endometrial progression and function across '\n",
      " 'published biomarker signatures (hsa-miR-22-3p, hsa-miR-107, hsa-miR-103a-3p, '\n",
      " 'hsa-miR-128-3p, hsamiR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, '\n",
      " 'hsamiR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p). Their significant gene '\n",
      " 'expression changes in the secre-tory endometrium support their potential '\n",
      " 'role in the endometrial regulation through an inhibition during the WOI. '\n",
      " 'Notably, hsa-miR-424-5p, hsa-miR-27b-3p and hsa-miR-195-5p have been '\n",
      " 'previously associated with RIF [11, 54, 56], while has-let-7b-5p was '\n",
      " 'associ-ated with human endometrial receptivity [57], and hsa-miR-22-3p was '\n",
      " 'overexpressed in a decidualization model [58]. Moreover, hsa-miR-15a-5p was '\n",
      " 'previously associated with endometriosis [59]. Notwithstanding, this study '\n",
      " 'is the first to associate hsa-miR-15a-5p, hsamiR-218-5p, hsa-miR-103a-3p, '\n",
      " 'hsa-miR-107, and hsamiR-128-3p to endometrial progression and functional '\n",
      " 'regulation in humans. Due to their direct relationship with progesterone and '\n",
      " 'estrogen [60], we expected to find TFs, such as ESR1 and PGR, involved in '\n",
      " 'endometrial progression, how-ever, we additionally identified other TFs with '\n",
      " 'an even larger influence in this process. For instance, STAT3 was associated '\n",
      " 'with 17/19 signatures (FDR < 0.05), and is known to regulate uterine '\n",
      " 'epithelial junctional reor-ganization and stromal proliferation, which are '\n",
      " 'critical for implantation [61]. SP1, which similarly regulated 17/19 '\n",
      " 'signatures (FDR < 0.05), acts as a downstream paracrine target of '\n",
      " 'progesterone to regulate estrogen inactivation, and could have a predominant '\n",
      " 'role dur-ing the WOI [62, 63]. By focusing on the most univer-sal, '\n",
      " 'overlapping TFs that were not previously related to implantation, we '\n",
      " 'ultimately prioritized CTCF and GATA6 as novel key regulators of endometrial '\n",
      " 'pro-gression, however, their specific molecular actions in endometrial '\n",
      " 'progression and function were beyond the scope of this project and merit '\n",
      " 'further investigation. CTCF is a conserved zinc finger protein whose '\n",
      " 'regu-latory functions are well characterized throughout the human body [64]. '\n",
      " 'CTCF acts as a transcriptional repres-sor in RNA polymerase II (Pol II) '\n",
      " 'pausing and imprint-ing and X-chromosome inactivation [65], as well as an '\n",
      " 'insulator, blocking the interaction between enhancers and the promoters of '\n",
      " 'neighbouring genes [66]. Consider-ing these attributed functions and its '\n",
      " 'variable expression throughout the menstrual cycle, we propose that CTCF '\n",
      " 'exerts an inhibitory role in endometrial tissue during the PF phase of the '\n",
      " 'menstrual cycle. With the significant downregulation of CTCF in the '\n",
      " 'secretory endometrium, dually validated in-silico and experimentally herein, '\n",
      " 'its inhibited genes would be derepressed and become tran-scriptionally '\n",
      " 'active during the WOI. This interpretation supports previous findings from '\n",
      " 'our group demonstrat-ing that, during the WOI, a global transcriptional '\n",
      " 'dere-pression may be required for implantation and early embryo development '\n",
      " '[63]. Transcriptional derepression has been associated with multiple human '\n",
      " 'disease states and should be investigated further within the context of '\n",
      " 'endometrial-factor infertility. Despite previous asso-ciations of CTCF with '\n",
      " 'endometriosis [67], its implica-tion in endometrial progression and function '\n",
      " 'has not been directly proposed until now. Nevertheless, the role of CTCF in '\n",
      " 'endometrial receptivity has been revealed through its interaction with '\n",
      " 'HOXA10, a gene mainly expressed in endometrium involved in functions such as '\n",
      " 'endometrial proliferation and differentiation, the forma-tion of pinopodes '\n",
      " 'or embryo implantation [68]. Indeed, this study proposed that an '\n",
      " 'overexpression of CTCF can lead to a drop in HOXA10 expression, affecting '\n",
      " 'endome-trial proliferation and endometrial function and corroborating the '\n",
      " 'fact that CTCF must decrease its expression as we can observe in our '\n",
      " 'findings. GATA6, the second prioritized TF that overlapped across the gene '\n",
      " 'lists, is a transcription factor belong-ing to the GATA family, a highly '\n",
      " 'conserved family of six zinc finger proteins [69]. Besides its essential '\n",
      " 'role in embryonic development [70], GATA6 plays a key role in endometriosis '\n",
      " 'and regulates steroidogenic genes [71]. Together with evidence supporting '\n",
      " 'both an activating and inhibitory role for GATA6 [72, 73], the significant '\n",
      " 'changes we observed in its expression across the men-strual cycle reflect '\n",
      " 'its presumed involvement in the endometrium and suggest an activating role '\n",
      " 'in endometrial progression and endometrial receptivity acquisition. Like '\n",
      " 'CTCF, this is the first time that GATA6 is proposed as a key TF in '\n",
      " 'endometrial progression and function. Although the exact molecular '\n",
      " 'mechanisms underlying endometrial regulation remain elusive, emerging '\n",
      " 'exper-imental technologies such as High-throughput Chro-mosome Conformation '\n",
      " 'Capture (Hi-C) or chromatin interaction analysis with paired-end tags '\n",
      " '(ChIA-PET) that respectively study general chromatin interactions or '\n",
      " 'chromatin interactions involving a specific protein [74, 75] may improve our '\n",
      " 'understanding of genetic regulation in the future. Nevertheless, as '\n",
      " 'collaborative efforts unveil new regulators and gene targets in humans (such '\n",
      " 'as the ENCODE project [76]), the complexities of transcriptional regulation '\n",
      " 'can be exposed by in-silico analyses like those described herein, and we can '\n",
      " 'gain a better understanding of the processes involved in men-strual cycle '\n",
      " 'progression and endometrial competence. Furthermore, it should be noted that '\n",
      " 'results of gene expression studies are influenced by different variables, '\n",
      " 'such as the experimental platforms employed, study designs, or patients’ '\n",
      " 'characteristics of each independent study, adding variability [30, 77, 78]. '\n",
      " 'Despite this fact, we have found that our holistic approach overcomes this '\n",
      " 'undesired variability effect by finding common upstream master regulators '\n",
      " 'across studies and predicting new relationships between hormones, TFs and '\n",
      " 'miRNAs. This approach also provides the basis for future singlemolecule '\n",
      " 'studies that aim to elucidate new regulatory pathways in endometrial '\n",
      " 'progression and function. Our results lay the foundation for further '\n",
      " 'molecular studies that can validate the function(s) of the prevailing '\n",
      " 'endo-metrial regulators we prioritized.')\n"
     ]
    }
   ],
   "source": [
    "a = db.similarity_search(\"What is the function of CTCF in endometrium?\", 4)\n",
    "print(a[0].metadata)\n",
    "parent = a[0].metadata.get('parent')\n",
    "print(parent)\n",
    "ref = a[0].metadata.get('Reference')\n",
    "\n",
    "b = db.get(\n",
    "    where={\n",
    "\n",
    "        \"$and\" :[\n",
    "             {\"Reference\":ref},\n",
    "             {\"parent\":parent}\n",
    "        ]\n",
    "        },\n",
    "    include=[\"documents\"]\n",
    "        \n",
    "    )\n",
    "\n",
    "pprint.pprint(\"\".join(b['documents']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "87a33f07",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Introduction_Patricia Diaz-Gimeno et al.,Fertil Steril®, 2024_1\n",
      "Methods_Patricia Diaz-Gimeno et al.,Fertil Steril®, 2024_2\n",
      "Results_Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023_8\n"
     ]
    }
   ],
   "source": [
    "for r in a['metadatas'][0]:\n",
    "     if r['parent'] not in ['Journal','URL']:\n",
    "        print( \"_\".join([r['parent'],r['Reference'],str(r['chunk_index']+1)]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "20db46ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[{'parent': 'Journal',\n",
       "   'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023',\n",
       "   'chunk_index': 0,\n",
       "   'DOI': 'https://doi.org/10.1186/s12958-023-01131-4'},\n",
       "  {'DOI': 'https://doi.org/10.1016/j.fertnstert.2024.03.015',\n",
       "   'chunk_index': 0,\n",
       "   'parent': 'Introduction',\n",
       "   'Reference': 'Patricia Diaz-Gimeno et al.,Fertil Steril®, 2024'},\n",
       "  {'Reference': 'Patricia Diaz-Gimeno et al.,Fertil Steril®, 2024',\n",
       "   'DOI': 'https://doi.org/10.1016/j.fertnstert.2024.03.015',\n",
       "   'parent': 'Methods',\n",
       "   'chunk_index': 1},\n",
       "  {'parent': 'Results',\n",
       "   'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023',\n",
       "   'DOI': 'https://doi.org/10.1186/s12958-023-01131-4',\n",
       "   'chunk_index': 7}]]"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a['metadatas']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "723f124a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['A key factor for reproductive success in assisted reproduction treatments is the status of the maternal endometrium during embryo implantation and fetal development. The endometrial cycle is reflected by the cyclic structural and functional changes of the endometrium across the menstrual cycle, particularly during the mid-secretory phase, to prepare for the‘‘window of implantation’’ (WOI) . After successful embryo implantation, the decidua (specialized layer of the endometrium) actively encapsulates the trophectoderm to support placentation and provide adequate vascularization for optimal fetal growth . Thus, approaches that predict and prevent endometrial-factor infertility could substantially improve assisted reproduction treatment outcomes by supporting the establishment and maintenance of pregnancy',\n",
       "  '. There is a lack of consensus on the minimum number of implantation failures with good-quality embryos derived from ovum donation or with confirmed euploid karyotypes required to diagnose recurrent implantation failure (RIF) . The disparities in clinical classification and heterogeneous etiology of RIF confound its diagnosis and leave the exact prevalence unclear . As clinical symptoms cannot be prevented and do not provide sufficient criteria to stratify patients, patients might be misclassi fied and receive inefficient treatment. Indeed, most studies using clinical criteria for RIF identified no or few differentially expressed genes between samples of affected patients , reflecting the molecular heterogeneity of RIF. To overcome these healthcare challenges, high-throughput molecular technologies are combined with artificial intelligence (AI) algorithms to provide a deeper understanding of infertilityrelated pathogenesis'],\n",
       " ['MATERIAL AND METHODS Ethics statement This study was approved by the Ethics Committee of the Instituto Valenciano de Infertilidad, Valencia, Spain (1810-FIVI-066-PD/1810-FIVI-048-PD). Written informed consent was obtained from all recruited patients (n1⁄4 281). Participants, collection of endometrial biopsies, and reproductive outcomes This prospective, multicentric study was conducted between July 2018 and July 2021 at 5 private fertility clinics in Spain. Participants (n1⁄4 281) were scheduled for endometrial evaluation due to medical indications orR2 implantation failures of idiopathic origin. All patients met the following inclusion criteria: 18- 50 years old, with a body mass index (BMI) of 19- 30 kg/m 2, undergoing hormone replacement therapy for single embryo transfer (SET), presenting an endometrial thickness >6',\n",
       "  '.5 mm with trilaminar structure in proliferative phase before P4 administration, and serum levels of estradiol >100 pg/mL and P4<1 ng/mL on the 10th day of estrogen treatment . Endometrial biopsies were collected after an average of 114.5/C6 7.2 h of P4 (duration ranged from 82 to 172 h), and processed as we previously described . Baseline patient characteristics including age, BMI, the number of IVF attempts, the quality of the embryo transferred (defined in theSupplemental Methods, available online), and reproductive outcomes (ie, pregnancy, live birth, and biochemical and clinical miscarriage) for each attempt, were obtained from internal medical records. Information about IVF attempts in other fertility clinics was included if recorded (only in 47.7% of patients)',\n",
       "  '.7% of patients). Study design First, a custom RNA-seq gene panel (the endometrial failure risk [EFR] panel) comprising 404 biomarkers of endometrial luteal phase timing and/or potential disruption was designed and employed to determine the risk of endometrial failure among 217 IVF patients (Supplemental Fig. 1, step 1 available online). The endometrial timing-corrected expression of the EFR panel was used to study the differential expression associated with endometrial disruption together with clinical characteristics of IVF patients to stratify the population as having poor or good endometrial prognoses, according to a semi-supervised learning process (Supplemental Fig. 1, step 2). The distinct transcriptomic patterns of each prognosis group were associated with the differences in reproductive outcomes of SET after biopsy collection, and the differentially expressed gene signature was proposed as a biomarker of endometrial disruption (Supplemental Fig. 1, step 3)'],\n",
       " ['. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53. Notably, GATA6 was selected due to novel association with endometrial progression and function. To infer the expression of CTCF and GATA6 in endometrial tissue throughout the menstrual cycle, we analysed their expression using a relevant endometrial dataset recently created by our group [30] (Table S1 ). We observed that, relative to the global expression of endometrial genes, CTCF was broadly expressed while GATA6 was moderately expressed. Indeed, CTCF was significantly down-regulated between the PF and ESE phases (FDR = 2.10E-03) and ESE to the MSE phase (FDR = 2.10E-04), while GATA6 expression increased from the PF to the MSE phase (FDR = 1.70E-11) and from the ESE to MSE phase (FDR = 9.80E-15)',\n",
       "  '.70E-11) and from the ESE to MSE phase (FDR = 9.80E-15). These findings supported the role of these TFs at the beginning of the menstrual cycle and during the MSE phase (Fig. 3 B). These in-silico validations were then prospectively corroborated by RT-qPCR analysis in an independent sample set of 20 endometrial biopsies. We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with significant changes between the endometrial phases. Notably, CTCF expression was decreased between the PF/ ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C).']]"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ideal_chunks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "4c06b00c",
   "metadata": {},
   "outputs": [],
   "source": [
    "query = \"What is the function of CTCF in endometrium?\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "3b4092f3",
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "list indices must be integers or slices, not str",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mTypeError\u001b[39m                                 Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[74]\u001b[39m\u001b[32m, line 25\u001b[39m\n\u001b[32m     16\u001b[39m selected_index.append(ii)\n\u001b[32m     18\u001b[39m candidates = db.get(\n\u001b[32m     19\u001b[39m where= {\u001b[33m\"\u001b[39m\u001b[33m$and\u001b[39m\u001b[33m\"\u001b[39m :[\n\u001b[32m     20\u001b[39m             {\u001b[33m\"\u001b[39m\u001b[33mReference\u001b[39m\u001b[33m\"\u001b[39m:r[\u001b[33m'\u001b[39m\u001b[33mReference\u001b[39m\u001b[33m'\u001b[39m]},\n\u001b[32m     21\u001b[39m             {\u001b[33m\"\u001b[39m\u001b[33mparent\u001b[39m\u001b[33m\"\u001b[39m:r[\u001b[33m'\u001b[39m\u001b[33mparent\u001b[39m\u001b[33m'\u001b[39m]}\n\u001b[32m     22\u001b[39m     ]\n\u001b[32m     23\u001b[39m })\n\u001b[32m---> \u001b[39m\u001b[32m25\u001b[39m \u001b[38;5;28mprint\u001b[39m(\u001b[33mf\u001b[39m\u001b[33m\"\u001b[39m\u001b[33mCandidates\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[33m: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[43mcandidates\u001b[49m\u001b[43m[\u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mmetadatas\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m[\u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mchunk_index\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;132;01m}\u001b[39;00m\u001b[33m\"\u001b[39m)\n\u001b[32m     26\u001b[39m max_index = \u001b[38;5;28mlen\u001b[39m(candidates[\u001b[33m'\u001b[39m\u001b[33mmetadatas\u001b[39m\u001b[33m'\u001b[39m])-\u001b[32m1\u001b[39m\n\u001b[32m     28\u001b[39m meta_selected.append(candidates[\u001b[33m'\u001b[39m\u001b[33mmetadatas\u001b[39m\u001b[33m'\u001b[39m])\n",
      "\u001b[31mTypeError\u001b[39m: list indices must be integers or slices, not str"
     ]
    }
   ],
   "source": [
    "results = db.similarity_search(query=query,k=4)\n",
    "\n",
    "selected_index = []\n",
    "ideal_chunks = []\n",
    "meta_selected = []\n",
    "\n",
    "for doc in results:\n",
    "\n",
    "    r = doc.metadata\n",
    "\n",
    "    if r['parent'] not in ['Journal','URL']:\n",
    "\n",
    "        if \"_\".join([r['parent'],r['Reference'],str(r['chunk_index']+1)]) not in selected_index and \"_\".join([r['parent'],r['Reference'],str(r['chunk_index']-1)]) not in selected_index:\n",
    "            \n",
    "            ii = \"_\".join([r['parent'],r['Reference'],str(r['chunk_index'])])\n",
    "            selected_index.append(ii)\n",
    "\n",
    "            candidates = db.get(\n",
    "            where= {\"$and\" :[\n",
    "                        {\"Reference\":r['Reference']},\n",
    "                        {\"parent\":r['parent']}\n",
    "                ]\n",
    "            })\n",
    "\n",
    "            print(f\"Candidates\\n: {candidates['metadatas']['chunk_index']}\")\n",
    "            max_index = len(candidates['metadatas'])-1\n",
    "\n",
    "            meta_selected.append(candidates['metadatas'])\n",
    "            ideal_chunks.append([doc for doc,meta in zip(candidates['documents'], candidates['metadatas'])\n",
    "                        if meta['chunk_index'] in [r[\"chunk_index\"], max(r[\"chunk_index\"]-1,0), min(r[\"chunk_index\"] + 1,max_index)]])\n",
    "\n",
    "context = []\n",
    "print(ideal_chunks)\n",
    "\n",
    "for text, meta in zip(ideal_chunks, meta_selected):\n",
    "    context.append(f'Reference:{meta[0][\"Reference\"]}\\n\\nLink (DOI)\\n: {meta[0][\"DOI\"]}\\n\\nSummary:\\n\\n{\"\".join(text)}\\n\\n')\n",
    "\n",
    "print(context)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "dac7575b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "min(1,len(candidates['metadatas'])-1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "c0d5a492",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7\n",
      "4\n"
     ]
    }
   ],
   "source": [
    "for doc  in results:\n",
    "    r = doc.metadata\n",
    "    if r['parent'] not in ['Journal','URL']:\n",
    "        \n",
    "        if \"_\".join([r['parent'],r['Reference'],str(r['chunk_index']+1)]) not in selected_index and \"_\".join([r['parent'],r['Reference'],str(r['chunk_index']-1)]) not in selected_index:\n",
    "\n",
    "            ii = \"_\".join([r['parent'],r['Reference'],str(r['chunk_index'])])\n",
    "            selected_index.append(ii)\n",
    "\n",
    "            candidates = db.get(\n",
    "            where= {\"$and\" :[\n",
    "                        {\"Reference\":r['Reference']},\n",
    "                        {\"parent\":r['parent']}\n",
    "                ]\n",
    "            })\n",
    "\n",
    "            max_index = len(candidates['metadatas'])-1\n",
    "            meta_selected.append(candidates['metadatas'])\n",
    "            ideal_chunks.append([doc for doc,meta in zip(candidates['documents'], candidates['metadatas'])\n",
    "                        if meta['chunk_index'] in [r[\"chunk_index\"], max(r[\"chunk_index\"]-1,0), min(r[\"chunk_index\"] + 1,max_index)]])\n",
    "\n",
    "print(max_index)\n",
    "print(len(ideal_chunks))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "f1e29da9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Document(metadata={'DOI': 'https://doi.org/10.1186/s12958-023-01131-4', 'chunk_index': 7, 'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023', 'parent': 'Discussion'}, page_content='. Consider-ing these attributed functions and its variable expression throughout the menstrual cycle, we propose that CTCF exerts an inhibitory role in endometrial tissue during the PF phase of the menstrual cycle. With the significant downregulation of CTCF in the secretory endometrium, dually validated in-silico and experimentally herein, its inhibited genes would be derepressed and become tran-scriptionally active during the WOI. This interpretation supports previous findings from our group demonstrat-ing that, during the WOI, a global transcriptional dere-pression may be required for implantation and early embryo development [63]. Transcriptional derepression has been associated with multiple human disease states and should be investigated further within the context of endometrial-factor infertility. Despite previous asso-ciations of CTCF with endometriosis [67], its implica-tion in endometrial progression and function has not been directly proposed until now')"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "f7891139",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['. SP1, which similarly regulated 17/19 signatures (FDR < 0.05), acts as a downstream paracrine target of progesterone to regulate estrogen inactivation, and could have a predominant role dur-ing the WOI [62, 63]. By focusing on the most univer-sal, overlapping TFs that were not previously related to implantation, we ultimately prioritized CTCF and GATA6 as novel key regulators of endometrial pro-gression, however, their specific molecular actions in endometrial progression and function were beyond the scope of this project and merit further investigation. CTCF is a conserved zinc finger protein whose regu-latory functions are well characterized throughout the human body [64]. CTCF acts as a transcriptional repres-sor in RNA polymerase II (Pol II) pausing and imprint-ing and X-chromosome inactivation [65], as well as an insulator, blocking the interaction between enhancers and the promoters of neighbouring genes [66]',\n",
       " '. Consider-ing these attributed functions and its variable expression throughout the menstrual cycle, we propose that CTCF exerts an inhibitory role in endometrial tissue during the PF phase of the menstrual cycle. With the significant downregulation of CTCF in the secretory endometrium, dually validated in-silico and experimentally herein, its inhibited genes would be derepressed and become tran-scriptionally active during the WOI. This interpretation supports previous findings from our group demonstrat-ing that, during the WOI, a global transcriptional dere-pression may be required for implantation and early embryo development [63]. Transcriptional derepression has been associated with multiple human disease states and should be investigated further within the context of endometrial-factor infertility. Despite previous asso-ciations of CTCF with endometriosis [67], its implica-tion in endometrial progression and function has not been directly proposed until now',\n",
       " '. Nevertheless, the role of CTCF in endometrial receptivity has been revealed through its interaction with HOXA10, a gene mainly expressed in endometrium involved in functions such as endometrial proliferation and differentiation, the forma-tion of pinopodes or embryo implantation [68]. Indeed, this study proposed that an overexpression of CTCF can lead to a drop in HOXA10 expression, affecting endome-trial proliferation and endometrial function and corroborating the fact that CTCF must decrease its expression as we can observe in our findings. GATA6, the second prioritized TF that overlapped across the gene lists, is a transcription factor belong-ing to the GATA family, a highly conserved family of six zinc finger proteins [69]. Besides its essential role in embryonic development [70], GATA6 plays a key role in endometriosis and regulates steroidogenic genes [71]']"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ideal_chunks[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "ee4c48eb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ids': ['bd725078-6b0e-11f0-8e08-00155de32492'],\n",
       " 'embeddings': None,\n",
       " 'documents': ['Results Gene signatures associated to endometrial progression and function Of the 19 gene lists used for our analysis, eleven were obtained from published studies evaluating control patients throughout the menstrual cycle [4, 17-19, 34-40], and the other eight were derived from studies comparing patients with RIF or unexplained infertility to controls [21, 41-47] (Table 1). Unifying all the aforementioned signatures, we compiled 3,608 genes related to endometrial progression and function. Hormonal regulation of endometrial progression is largely driven by progesterone We identified 7,540 and 698 genes related to estrogen and progesterone hormones, respectively. However, as determined by the relative contribution of each type of hormone within each gene list (Fig. 1A), 17/19 (89%) signatures favoured regulation by progesterone rather than estrogen. These differences were significant in 47% of the signatures (FDR < 0.05)'],\n",
       " 'uris': None,\n",
       " 'included': ['metadatas', 'documents'],\n",
       " 'data': None,\n",
       " 'metadatas': [{'parent': 'Results',\n",
       "   'DOI': 'https://doi.org/10.1186/s12958-023-01131-4',\n",
       "   'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023',\n",
       "   'chunk_index': 0}]}"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collection.get(\n",
    "    where={\"$and\":[\n",
    "           {\"Reference\":'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023'},\n",
    "           {\"parent\":\"Results\"},\n",
    "           {\"chunk_index\":0}\n",
    "    ]}\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0f49680a",
   "metadata": {},
   "outputs": [],
   "source": [
    "query = \"How many phases does the endoemtrium have?\"\n",
    "docs = db.similarity_search(query,k=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "f54dbad0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Document(metadata={'chunk_index': 7, 'parent': 'Discussion', 'DOI': 'https://doi.org/10.1186/s12958-023-01131-4', 'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023'}, page_content='. Considering these attributed functions and its variable expression throughout the menstrual cycle, we propose that CTCF exerts an inhibitory role in endometrial tissue during the PF phase of the menstrual cycle. With the significant downregulation of CTCF in the secretory endometrium, dually validated in-silico and experimentally herein, its inhibited genes would be derepressed and become transcriptionally active during the WOI. This interpretation supports previous findings from our group demonstrating that, during the WOI, a global transcriptional derepression may be required for implantation and early embryo development [63]. Transcriptional derepression has been associated with multiple human disease states and should be investigated further within the context of endometrial-factor infertility. Despite previous associations of CTCF with endometriosis [67], its implication in endometrial progression and function has not been directly proposed until now'),\n",
       " Document(metadata={'parent': 'Abstract', 'DOI': 'https://doi.org/10.1186/s12958-023-01131-4', 'chunk_index': 3, 'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023'}, page_content='. Keywords CTCF, GATA6, Progesterone, Estrogen, miRNAs, TFs, Endometrial receptivity, Recurrent implantation failure, Infertility, Menstrual cycle regulation'),\n",
       " Document(metadata={'parent': 'Results', 'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023', 'DOI': 'https://doi.org/10.1186/s12958-023-01131-4', 'chunk_index': 7}, page_content='. We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with significant changes between the endometrial phases. Notably, CTCF expression was decreased between the PF/ ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C).'),\n",
       " Document(metadata={'chunk_index': 5, 'Reference': 'Antonio Parraga-Leo et al.,Reproductive Biology and Endocrinology, 2023', 'parent': 'Results', 'DOI': 'https://doi.org/10.1186/s12958-023-01131-4'}, page_content='. CTCF and GATA6 as distinguished transcriptional regulators of the human menstrual cycle We repeated the regulation network analysis with the TFs, to identify the prominent TFs commonly regulating the genes related to endometrial progression and function in the reported signatures. We observed 78.80% of the TFs regulated genes in up to 10 signatures, and only 99 TFs (21.20%) regulated genes in over 10 (Fig. 3A). We highlighted CTCF (a CCCTC-binding factor, which functions as a transcriptomic repressor) as the TF with the most influential regulation of endometrial progression and function across 95% of studies (18 gene lists, excluding that of Punyadeera, Fig. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53')]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "docs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "961a1fd8",
   "metadata": {},
   "outputs": [],
   "source": [
    "texts_re = []\n",
    "metadata_re = []\n",
    "for doc in docs:\n",
    "    texts_re.append(doc.page_content)\n",
    "    metadata_re.append(doc.metadata)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "2f3748d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'parent': 'Discussion',\n",
       "  'chunk_index': 7,\n",
       "  'DOI': 'https://doi.org/10.1186/s12958-023-01131-4'},\n",
       " {'DOI': 'https://doi.org/10.1186/s12958-023-01131-4',\n",
       "  'parent': 'Abstract',\n",
       "  'chunk_index': 3},\n",
       " {'parent': 'Results',\n",
       "  'DOI': 'https://doi.org/10.1186/s12958-023-01131-4',\n",
       "  'chunk_index': 7},\n",
       " {'parent': 'Results',\n",
       "  'chunk_index': 5,\n",
       "  'DOI': 'https://doi.org/10.1186/s12958-023-01131-4'}]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "4390b620",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The reference of article is https://doi.org/10.1186/s12958-023-01131-4, its information is:\n",
      " . Considering these attributed functions and its variable expression throughout the menstrual cycle, we propose that CTCF exerts an inhibitory role in endometrial tissue during the PF phase of the menstrual cycle. With the significant downregulation of CTCF in the secretory endometrium, dually validated in-silico and experimentally herein, its inhibited genes would be derepressed and become transcriptionally active during the WOI. This interpretation supports previous findings from our group demonstrating that, during the WOI, a global transcriptional derepression may be required for implantation and early embryo development [63]. Transcriptional derepression has been associated with multiple human disease states and should be investigated further within the context of endometrial-factor infertility. Despite previous associations of CTCF with endometriosis [67], its implication in endometrial progression and function has not been directly proposed until now\n",
      "\n",
      "The reference of article is https://doi.org/10.1186/s12958-023-01131-4, its information is:\n",
      " . Keywords CTCF, GATA6, Progesterone, Estrogen, miRNAs, TFs, Endometrial receptivity, Recurrent implantation failure, Infertility, Menstrual cycle regulation\n",
      "\n",
      "The reference of article is https://doi.org/10.1186/s12958-023-01131-4, its information is:\n",
      " . We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with significant changes between the endometrial phases. Notably, CTCF expression was decreased between the PF/ ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C).\n",
      "\n",
      "The reference of article is https://doi.org/10.1186/s12958-023-01131-4, its information is:\n",
      " . CTCF and GATA6 as distinguished transcriptional regulators of the human menstrual cycle We repeated the regulation network analysis with the TFs, to identify the prominent TFs commonly regulating the genes related to endometrial progression and function in the reported signatures. We observed 78.80% of the TFs regulated genes in up to 10 signatures, and only 99 TFs (21.20%) regulated genes in over 10 (Fig. 3A). We highlighted CTCF (a CCCTC-binding factor, which functions as a transcriptomic repressor) as the TF with the most influential regulation of endometrial progression and function across 95% of studies (18 gene lists, excluding that of Punyadeera, Fig. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53\n"
     ]
    }
   ],
   "source": [
    "# context = \"\\n\\n---\\n\\n\".join(texts_re)\n",
    "\n",
    "context = []\n",
    "for text, meta in zip(texts_re, metadata_re):\n",
    "\n",
    "  context.append(f'The reference of article is {meta[\"DOI\"]}, its information is:\\n {text}')\n",
    "\n",
    "print(\"\\n\\n\".join(context))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "ec055939",
   "metadata": {},
   "outputs": [],
   "source": [
    "prompt = get_ques_response_prompt(question=query, context=get_system_message_rag(\"\\n\\n\".join(context)))\n",
    "\n",
    "answer = llm.invoke(prompt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "19d90d34",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'CTCF is a conserved zinc finger protein with well-characterized regulatory functions throughout the human body. It acts as a transcriptional repressor in RNA polymerase II (Pol II) pausing, imprinting, and X-chromosome inactivation. Additionally, CTCF functions as an insulator, blocking the interaction between enhancers and the promoters of neighboring genes (https://doi.org/10.1186/s12958‑023‑01131‑4).\\n\\nIn endometrial tissue, CTCF is proposed to exert an inhibitory role during the PF phase of the menstrual cycle. Its significant downregulation in the secretory endometrium leads to the derepression of its inhibited genes, which then become transcriptionally active during the Window of Implantation (WOI) (https://doi.org/10.1186/s12958‑023‑01131‑4).\\n\\nCTCF is highlighted as a CCCTC-binding factor that functions as a transcriptomic repressor and has the most influential regulation of endometrial progression and function across 95% of studies (https://doi.org/10.1186/s12958‑023‑01131‑4). Overexpression of CTCF can lead to a drop in HOXA10 expression, which affects endometrial proliferation and function (https://doi.org/10.1186/s12958‑023‑01131‑4).'"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "answer"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4b4c5adb",
   "metadata": {},
   "source": [
    "## Download reference of each paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "9926cb04",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"info_articles.pkl\",\"rb\") as f:\n",
    "    my_json = pkl.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "83dd0697",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-10 19:08:04,763 - DEBUG - pubmed2pdf.cli - Full log mode activated\n",
      "2025-08-10 19:08:04,764 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16832043\n",
      "2025-08-10 19:08:05,439 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16832043 failed from error list index out of range\n",
      "2025-08-10 19:08:05,440 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20729534\n",
      "2025-08-10 19:08:06,127 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 20729534 failed from error list index out of range\n",
      "2025-08-10 19:08:06,127 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30929718\n",
      "2025-08-10 19:08:07,885 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30929718 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:07,885 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16306079\n",
      "2025-08-10 19:08:08,525 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16306079 failed from error list index out of range\n",
      "2025-08-10 19:08:08,525 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29102484\n",
      "2025-08-10 19:08:09,753 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29102484 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:09,753 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30624659\n",
      "2025-08-10 19:08:10,451 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30624659 failed from error list index out of range\n",
      "2025-08-10 19:08:10,451 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29452422\n",
      "2025-08-10 19:08:12,160 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29452422 failed from error list index out of range\n",
      "2025-08-10 19:08:12,160 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23994285\n",
      "2025-08-10 19:08:13,535 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23994285 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:13,535 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23290997\n",
      "2025-08-10 19:08:14,874 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23290997 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:14,874 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30951376\n",
      "2025-08-10 19:08:15,703 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30951376 failed from error list index out of range\n",
      "2025-08-10 19:08:15,703 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21849299\n",
      "2025-08-10 19:08:16,383 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 21849299 failed from error list index out of range\n",
      "2025-08-10 19:08:16,383 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24882617\n",
      "2025-08-10 19:08:17,702 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24882617 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:17,702 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18192189\n",
      "2025-08-10 19:08:18,412 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18192189 failed from error list index out of range\n",
      "2025-08-10 19:08:18,413 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31462323\n",
      "2025-08-10 19:08:18,413 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12200466\n",
      "2025-08-10 19:08:19,088 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12200466 failed from error list index out of range\n",
      "2025-08-10 19:08:19,088 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12529417\n",
      "2025-08-10 19:08:19,769 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12529417 failed from error list index out of range\n",
      "2025-08-10 19:08:19,769 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15878921\n",
      "2025-08-10 19:08:20,486 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15878921 failed from error list index out of range\n",
      "2025-08-10 19:08:20,486 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35085395\n",
      "2025-08-10 19:08:21,172 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35085395 failed from error list index out of range\n",
      "2025-08-10 19:08:21,172 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19933690\n",
      "2025-08-10 19:08:21,846 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19933690 failed from error list index out of range\n",
      "2025-08-10 19:08:21,846 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33576824\n",
      "2025-08-10 19:08:22,530 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33576824 failed from error list index out of range\n",
      "2025-08-10 19:08:22,530 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19617889\n",
      "2025-08-10 19:08:24,931 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10592173\n",
      "2025-08-10 19:08:25,615 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10592173 failed from error list index out of range\n",
      "2025-08-10 19:08:25,615 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10802651\n",
      "2025-08-10 19:08:27,841 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31340985\n",
      "2025-08-10 19:08:27,841 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29156006\n",
      "2025-08-10 19:08:28,499 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29156006 failed from error list index out of range\n",
      "2025-08-10 19:08:28,499 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26415722\n",
      "2025-08-10 19:08:29,188 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26415722 failed from error list index out of range\n",
      "2025-08-10 19:08:29,189 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26395145\n",
      "2025-08-10 19:08:31,089 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26395145 failed from error list index out of range\n",
      "2025-08-10 19:08:31,089 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32641214\n",
      "2025-08-10 19:08:32,434 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32641214 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:32,434 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25605792\n",
      "2025-08-10 19:08:33,168 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 25605792 failed from error list index out of range\n",
      "2025-08-10 19:08:33,169 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11846609\n",
      "2025-08-10 19:08:34,466 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 11846609 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:34,467 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28855728\n",
      "2025-08-10 19:08:34,467 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20619403\n",
      "2025-08-10 19:08:35,676 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 20619403 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:35,676 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12021176\n",
      "2025-08-10 19:08:36,360 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12021176 failed from error list index out of range\n",
      "2025-08-10 19:08:36,360 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15501903\n",
      "2025-08-10 19:08:37,055 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15501903 failed from error list index out of range\n",
      "2025-08-10 19:08:37,055 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15666095\n",
      "2025-08-10 19:08:39,277 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12728018\n",
      "2025-08-10 19:08:40,406 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12728018 failed from error list index out of range\n",
      "2025-08-10 19:08:40,406 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30081718\n",
      "2025-08-10 19:08:44,076 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18539642\n",
      "2025-08-10 19:08:44,752 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18539642 failed from error list index out of range\n",
      "2025-08-10 19:08:44,752 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23555582\n",
      "2025-08-10 19:08:44,752 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22025212\n",
      "2025-08-10 19:08:45,688 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 22025212 failed from error list index out of range\n",
      "2025-08-10 19:08:45,688 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28523980\n",
      "2025-08-10 19:08:46,432 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28523980 failed from error list index out of range\n",
      "2025-08-10 19:08:46,432 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16672246\n",
      "2025-08-10 19:08:47,081 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16672246 failed from error list index out of range\n",
      "2025-08-10 19:08:47,081 - INFO - pubmed2pdf.cli - Trying to fetch pmid 7962413\n",
      "2025-08-10 19:08:47,773 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 7962413 failed from error list index out of range\n",
      "2025-08-10 19:08:47,773 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23933037\n",
      "2025-08-10 19:08:48,966 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23933037 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:48,966 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26804062\n",
      "2025-08-10 19:08:49,829 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26804062 failed from error list index out of range\n",
      "2025-08-10 19:08:49,829 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32482256\n",
      "2025-08-10 19:08:51,058 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32482256 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:08:51,058 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35147192\n",
      "2025-08-10 19:08:51,723 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35147192 failed from error list index out of range\n",
      "2025-08-10 19:08:51,723 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22902743\n",
      "2025-08-10 19:08:54,278 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29663566\n",
      "2025-08-10 19:08:55,126 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29663566 failed from error list index out of range\n",
      "2025-08-10 19:08:55,126 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28229988\n",
      "2025-08-10 19:08:55,995 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28229988 failed from error list index out of range\n",
      "2025-08-10 19:08:55,996 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30037059\n",
      "2025-08-10 19:08:56,737 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30037059 failed from error list index out of range\n",
      "2025-08-10 19:08:56,738 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22911744\n",
      "2025-08-10 19:08:56,738 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34117589\n",
      "2025-08-10 19:08:59,145 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31511876\n",
      "2025-08-10 19:08:59,836 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 31511876 failed from error list index out of range\n",
      "2025-08-10 19:08:59,836 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24100212\n",
      "2025-08-10 19:09:00,550 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24100212 failed from error list index out of range\n",
      "2025-08-10 19:09:00,550 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16807381\n",
      "2025-08-10 19:09:01,210 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16807381 failed from error list index out of range\n",
      "2025-08-10 19:09:01,210 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33830236\n",
      "2025-08-10 19:09:02,014 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33830236 failed from error list index out of range\n",
      "2025-08-10 19:09:02,014 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21964334\n",
      "2025-08-10 19:09:04,499 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19563753\n",
      "2025-08-10 19:09:05,762 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19563753 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:09:05,763 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35574918\n",
      "2025-08-10 19:09:06,450 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35574918 failed from error list index out of range\n",
      "2025-08-10 19:09:06,450 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18174356\n",
      "2025-08-10 19:09:07,116 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18174356 failed from error list index out of range\n",
      "2025-08-10 19:09:07,116 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26932670\n",
      "2025-08-10 19:09:09,248 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26932670 failed from error list index out of range\n",
      "2025-08-10 19:09:09,249 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29402198\n",
      "2025-08-10 19:09:13,253 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15550397\n",
      "2025-08-10 19:09:14,524 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15550397 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:09:14,524 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35088888\n",
      "2025-08-10 19:09:14,524 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19890323\n",
      "2025-08-10 19:09:18,103 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19815776\n",
      "2025-08-10 19:09:18,995 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19815776 failed from error argument of type 'NoneType' is not iterable\n",
      "2025-08-10 19:09:18,996 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17571346\n",
      "2025-08-10 19:09:18,996 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32043035\n",
      "2025-08-10 19:09:20,438 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32043035 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:09:20,438 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36205711\n",
      "2025-08-10 19:09:21,104 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36205711 failed from error list index out of range\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in Data2/\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-10 19:09:48,571 - DEBUG - pubmed2pdf.cli - Full log mode activated\n",
      "2025-08-10 19:09:48,572 - INFO - pubmed2pdf.cli - Trying to fetch pmid 1155504\n",
      "2025-08-10 19:09:49,824 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 1155504 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:09:49,825 - INFO - pubmed2pdf.cli - Trying to fetch pmid 1424330\n",
      "2025-08-10 19:09:50,989 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 1424330 failed from error Invalid URL 'wenmhi4L9dPogJ7fBmofTSqFqabAb1VqRbFzzuvnOkth6PgfEnsViH5uoEhxes7n': No scheme supplied. Perhaps you meant https://wenmhi4L9dPogJ7fBmofTSqFqabAb1VqRbFzzuvnOkth6PgfEnsViH5uoEhxes7n?\n",
      "2025-08-10 19:09:50,989 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10362823\n",
      "2025-08-10 19:09:51,701 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10362823 failed from error list index out of range\n",
      "2025-08-10 19:09:51,701 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15353123\n",
      "2025-08-10 19:09:53,993 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16306079\n",
      "2025-08-10 19:09:54,951 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16306079 failed from error list index out of range\n",
      "2025-08-10 19:09:54,952 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10221616\n",
      "2025-08-10 19:09:56,722 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10221616 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:09:56,722 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28863933\n",
      "2025-08-10 19:09:57,980 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28863933 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:09:57,980 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28923940\n",
      "2025-08-10 19:09:58,644 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28923940 failed from error list index out of range\n",
      "2025-08-10 19:09:58,644 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33313697\n",
      "2025-08-10 19:09:59,315 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33313697 failed from error list index out of range\n",
      "2025-08-10 19:09:59,315 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36822566\n",
      "2025-08-10 19:10:00,634 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36822566 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:00,635 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24269084\n",
      "2025-08-10 19:10:02,258 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24269084 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:02,258 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24581986\n",
      "2025-08-10 19:10:03,662 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24581986 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:03,663 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30624659\n",
      "2025-08-10 19:10:04,327 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30624659 failed from error list index out of range\n",
      "2025-08-10 19:10:04,327 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33077239\n",
      "2025-08-10 19:10:05,586 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33077239 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:05,586 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26797113\n",
      "2025-08-10 19:10:05,586 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33576824\n",
      "2025-08-10 19:10:06,314 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33576824 failed from error list index out of range\n",
      "2025-08-10 19:10:06,314 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34875061\n",
      "2025-08-10 19:10:07,005 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 34875061 failed from error list index out of range\n",
      "2025-08-10 19:10:07,005 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29452422\n",
      "2025-08-10 19:10:07,731 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29452422 failed from error list index out of range\n",
      "2025-08-10 19:10:07,731 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20619403\n",
      "2025-08-10 19:10:09,094 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 20619403 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:09,094 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23102856\n",
      "2025-08-10 19:10:10,409 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23102856 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:10,409 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34199109\n",
      "2025-08-10 19:10:11,104 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 34199109 failed from error list index out of range\n",
      "2025-08-10 19:10:11,104 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12421895\n",
      "2025-08-10 19:10:11,886 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12421895 failed from error list index out of range\n",
      "2025-08-10 19:10:11,886 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22256780\n",
      "2025-08-10 19:10:12,603 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 22256780 failed from error list index out of range\n",
      "2025-08-10 19:10:12,603 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24581625\n",
      "2025-08-10 19:10:13,921 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24581625 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:13,922 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30145981\n",
      "2025-08-10 19:10:13,922 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30713991\n",
      "2025-08-10 19:10:15,115 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36070983\n",
      "2025-08-10 19:10:16,363 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36070983 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:16,364 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32974805\n",
      "2025-08-10 19:10:19,091 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32723696\n",
      "2025-08-10 19:10:20,356 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32723696 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:20,356 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33880419\n",
      "2025-08-10 19:10:21,406 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33067123\n",
      "2025-08-10 19:10:22,521 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33067123 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:22,522 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29737471\n",
      "2025-08-10 19:10:25,069 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33880420\n",
      "2025-08-10 19:10:26,202 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36472596\n",
      "2025-08-10 19:10:26,931 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36472596 failed from error list index out of range\n",
      "2025-08-10 19:10:26,931 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11752295\n",
      "2025-08-10 19:10:27,630 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 11752295 failed from error list index out of range\n",
      "2025-08-10 19:10:27,631 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25605792\n",
      "2025-08-10 19:10:28,314 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 25605792 failed from error list index out of range\n",
      "2025-08-10 19:10:28,314 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17160063\n",
      "2025-08-10 19:10:30,583 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-10 19:10:31,377 - DEBUG - pubmed2pdf.cli - ** fetching of reprint None failed from error list index out of range\n",
      "2025-08-10 19:10:31,377 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-10 19:10:32,180 - DEBUG - pubmed2pdf.cli - ** fetching of reprint None failed from error list index out of range\n",
      "2025-08-10 19:10:32,180 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-10 19:10:32,934 - DEBUG - pubmed2pdf.cli - ** fetching of reprint None failed from error list index out of range\n",
      "2025-08-10 19:10:32,934 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10592173\n",
      "2025-08-10 19:10:33,653 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10592173 failed from error list index out of range\n",
      "2025-08-10 19:10:33,653 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10802651\n",
      "2025-08-10 19:10:35,589 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18546601\n",
      "2025-08-10 19:10:37,781 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26109056\n",
      "2025-08-10 19:10:37,781 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33036008\n",
      "2025-08-10 19:10:39,487 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33036008 failed from error Invalid URL '': No scheme supplied. Perhaps you meant https://?\n",
      "2025-08-10 19:10:39,488 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28370781\n",
      "2025-08-10 19:10:40,447 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28370781 failed from error list index out of range\n",
      "2025-08-10 19:10:40,447 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34947871\n",
      "2025-08-10 19:10:41,158 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 34947871 failed from error list index out of range\n",
      "2025-08-10 19:10:41,158 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35929523\n",
      "2025-08-10 19:10:42,011 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35929523 failed from error list index out of range\n",
      "2025-08-10 19:10:42,011 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27441287\n",
      "2025-08-10 19:10:43,254 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 27441287 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:43,254 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30452456\n",
      "2025-08-10 19:10:43,254 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31540845\n",
      "2025-08-10 19:10:44,502 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 31540845 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:44,502 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33830236\n",
      "2025-08-10 19:10:45,192 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33830236 failed from error list index out of range\n",
      "2025-08-10 19:10:45,192 - INFO - pubmed2pdf.cli - Trying to fetch pmid 37700285\n",
      "2025-08-10 19:10:45,192 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29315421\n",
      "2025-08-10 19:10:45,927 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29315421 failed from error list index out of range\n",
      "2025-08-10 19:10:45,928 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28855728\n",
      "2025-08-10 19:10:45,928 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35092277\n",
      "2025-08-10 19:10:46,606 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35092277 failed from error list index out of range\n",
      "2025-08-10 19:10:46,606 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32482256\n",
      "2025-08-10 19:10:47,970 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32482256 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-10 19:10:47,970 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11823860\n",
      "2025-08-10 19:10:50,115 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12490681\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in Data2/\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-10 19:10:51,291 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12490681 failed from error list index out of range\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'Deciphering a shared transcriptomic regulation and the relative contribution of each regulator type through endometrial gene expression signatures': ['16832043',\n",
       "  '20729534',\n",
       "  '30929718',\n",
       "  '16306079',\n",
       "  '29102484',\n",
       "  '30624659',\n",
       "  '29452422',\n",
       "  '23994285',\n",
       "  '23290997',\n",
       "  '30951376',\n",
       "  '21849299',\n",
       "  '24882617',\n",
       "  '18192189',\n",
       "  '31462323',\n",
       "  '12200466',\n",
       "  '12529417',\n",
       "  '15878921',\n",
       "  '35085395',\n",
       "  '19933690',\n",
       "  '33576824',\n",
       "  '19617889',\n",
       "  '10592173',\n",
       "  '10802651',\n",
       "  '31340985',\n",
       "  '29156006',\n",
       "  '26415722',\n",
       "  '26395145',\n",
       "  '32641214',\n",
       "  '25605792',\n",
       "  '11846609',\n",
       "  '28855728',\n",
       "  '20619403',\n",
       "  '12021176',\n",
       "  '15501903',\n",
       "  '15666095',\n",
       "  '12728018',\n",
       "  '30081718',\n",
       "  '18539642',\n",
       "  '23555582',\n",
       "  '22025212',\n",
       "  '28523980',\n",
       "  '16672246',\n",
       "  '7962413',\n",
       "  '23933037',\n",
       "  '26804062',\n",
       "  '32482256',\n",
       "  '35147192',\n",
       "  '22902743',\n",
       "  '29663566',\n",
       "  '28229988',\n",
       "  '30037059',\n",
       "  '22911744',\n",
       "  '34117589',\n",
       "  '31511876',\n",
       "  '24100212',\n",
       "  '16807381',\n",
       "  '33830236',\n",
       "  '21964334',\n",
       "  '19563753',\n",
       "  '35574918',\n",
       "  '18174356',\n",
       "  '26932670',\n",
       "  '29402198',\n",
       "  '15550397',\n",
       "  '35088888',\n",
       "  '19890323',\n",
       "  '19815776',\n",
       "  '17571346',\n",
       "  '32043035',\n",
       "  '36205711'],\n",
       " 'Predicting risk of endometrial failure: a biomarker signature that identifies a novel disruption independent of endometrial timing in patients undergoing hormonal replacement cycles': ['1155504',\n",
       "  '1424330',\n",
       "  '10362823',\n",
       "  '15353123',\n",
       "  '16306079',\n",
       "  '10221616',\n",
       "  '28863933',\n",
       "  '28923940',\n",
       "  '33313697',\n",
       "  '36822566',\n",
       "  '24269084',\n",
       "  '24581986',\n",
       "  '30624659',\n",
       "  '33077239',\n",
       "  '26797113',\n",
       "  '33576824',\n",
       "  '34875061',\n",
       "  '29452422',\n",
       "  '20619403',\n",
       "  '23102856',\n",
       "  '34199109',\n",
       "  '12421895',\n",
       "  '22256780',\n",
       "  '24581625',\n",
       "  '30145981',\n",
       "  '30713991',\n",
       "  '36070983',\n",
       "  '32974805',\n",
       "  '32723696',\n",
       "  '33880419',\n",
       "  '33067123',\n",
       "  '29737471',\n",
       "  '33880420',\n",
       "  '36472596',\n",
       "  '11752295',\n",
       "  '25605792',\n",
       "  '17160063',\n",
       "  'None',\n",
       "  'None',\n",
       "  'None',\n",
       "  '10592173',\n",
       "  '10802651',\n",
       "  '18546601',\n",
       "  '26109056',\n",
       "  '33036008',\n",
       "  '28370781',\n",
       "  '34947871',\n",
       "  '35929523',\n",
       "  '27441287',\n",
       "  '30452456',\n",
       "  '31540845',\n",
       "  '33830236',\n",
       "  '37700285',\n",
       "  '29315421',\n",
       "  '28855728',\n",
       "  '35092277',\n",
       "  '32482256',\n",
       "  '11823860',\n",
       "  '12490681']}"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pmid = {}\n",
    "url_pmc = f\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC{a}/\"\n",
    "\n",
    "for article in my_json:\n",
    "\n",
    "    pmid[article['PaperTitle']] = []\n",
    "\n",
    "    if 'https://' in my_json[0]['URL']:\n",
    "        doi = re.sub(r'https://doi.org/', '', article['URL'])\n",
    "        response = requests.get('https://api.crossref.org/works/'+doi)\n",
    "        list_reference = response.json()['message'].get('reference',None)\n",
    "        article['List_references'] = [x.get('DOI') for x in list_reference if x.get('DOI')]\n",
    "\n",
    "        # We retrieve the pmid from each DOI\n",
    "\n",
    "        for a in  article['List_references']:\n",
    "            \n",
    "            pmid[article['PaperTitle']].append(get_pmid_from_doi(a))\n",
    "\n",
    "        # We try to get the pdf\n",
    "\n",
    "        command = [\n",
    "        \"python3\", \"-m\", \"pubmed2pdf\", \"pdf\",\n",
    "        \"--out\", \"Data2/\",\n",
    "        \"--pmids\", f\"{', '.join(pmid[article['PaperTitle']])}\",\n",
    "        \"--verbose\",\n",
    "        \"--errors\", \"Data2/pubmed2pdf_log.txt\"\n",
    "        ]\n",
    "        subprocess.run(command)\n",
    "\n",
    "        # We try to get the html otherwise\n",
    "\n",
    "        to_retrieve = pd.read_csv(\"Data2/pubmed2pdf_log.txt\", header=None,dtype='str') # We get those that resulted in a mistake\n",
    "        to_retrieve.columns = [\"PMID\"]\n",
    "        to_retrieve = to_retrieve['PMID'].tolist()\n",
    "        to_retrieve.extend([re.sub(r'.html','',f)  for f in os.listdir('./Data2') if f.endswith('.html')]) # We add those in html format\n",
    "        [os.remove(os.path.join('Data2',f))  for f in os.listdir('./Data2') if f.endswith('.html')] # Then, we remove from directory\n",
    "        \n",
    "        pmcid = []\n",
    "        for pmid in to_retrieve:\n",
    "            pmcid.append(get_pmcid_from_pmid(pmid))\n",
    "        \n",
    "        response = requests.get(url_pmc, headers=HEADERS)\n",
    "        soup = BeautifulSoup(response.text, \"html.parser\")\n",
    "        soup.find(\"article\").get_text(separator=\"\\n\", strip=True)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "67b5e88b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "len(article['List_references'])\n",
    "[os.remove(os.path.join('Data2',f))  for f in os.listdir('./Data2') if f.endswith('.html')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "a4ed29bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_pmid_from_doi(doi, api_key=None):\n",
    "    url = \"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi\"\n",
    "    params = {\n",
    "        \"db\": \"pubmed\",\n",
    "        \"term\": f\"{doi}[DOI]\",\n",
    "        \"retmode\": \"json\"\n",
    "    }\n",
    "    if api_key:\n",
    "        params[\"api_key\"] = api_key\n",
    "\n",
    "    response = requests.get(url, params=params)\n",
    "    result = response.json()\n",
    "    pmids = result.get(\"esearchresult\", {}).get(\"idlist\", [])\n",
    "    return pmids[0] if pmids else 'None'\n",
    "\n",
    "\n",
    "\n",
    "def get_pmcid_from_pmid(pmid, api_key=None):\n",
    "    url = \"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi\"\n",
    "    params = {\n",
    "        \"dbfrom\": \"pubmed\",\n",
    "        \"db\": \"pmc\",\n",
    "        \"id\": pmid,\n",
    "        \"retmode\": \"json\"\n",
    "    }\n",
    "    if api_key:\n",
    "        params[\"api_key\"] = api_key\n",
    "\n",
    "    response = requests.get(url, params=params)\n",
    "    data = response.json()\n",
    "\n",
    "    try:\n",
    "        linksets = data[\"linksets\"]\n",
    "        if linksets and \"linksetdbs\" in linksets[0]:\n",
    "            pmcid_link = linksets[0][\"linksetdbs\"][0][\"links\"][0]\n",
    "            return pmcid_link\n",
    "        else:\n",
    "            return None\n",
    "    except Exception:\n",
    "        return None\n",
    "    \n",
    "def download_pmc_pdf(pmcid, filename=None):\n",
    "    pmc_number = pmcid.replace(\"PMC\", \"\")\n",
    "    pdf_url = f\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC{pmc_number}/pdf/\"\n",
    "\n",
    "    headers = {\n",
    "        \"User-Agent\": (\n",
    "            \"Mozilla/5.0 (Windows NT 10.0; Win64; x64) \"\n",
    "            \"AppleWebKit/537.36 (KHTML, like Gecko) \"\n",
    "            \"Chrome/115.0.0.0 Safari/537.36\"\n",
    "        )\n",
    "    }\n",
    "\n",
    "    response = requests.get(pdf_url, headers=headers)\n",
    "    if \"pdf\" in response.headers.get(\"Content-Type\", \"\"):\n",
    "        if not filename:\n",
    "            filename = f\"{pmcid}.pdf\"\n",
    "        with open(filename, \"wb\") as f:\n",
    "            f.write(response.content)\n",
    "        print(f\"✅ PDF downloaded as {filename}\")\n",
    "    else:\n",
    "        print(f\"⚠️ PDF not directly available at {pdf_url}\")\n",
    "        \n",
    "\n",
    "def download_pdf_from_pmid(pmid, api_key=None):\n",
    "    print(f\"🔎 Checking PMC for PMID: {pmid}\")\n",
    "    pmcid = get_pmcid_from_pmid(pmid, api_key)\n",
    "    if pmcid:\n",
    "        print(f\"✅ Found PMC ID: {pmcid}\")\n",
    "        download_pmc_pdf(pmcid)\n",
    "    else:\n",
    "        print(\"❌ No PMC version available for this PubMed article.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "913d8255",
   "metadata": {},
   "outputs": [],
   "source": [
    "pmids = []\n",
    "for article in my_json:\n",
    "    for doi in article['List_references']:\n",
    "        pmid = get_pmid_from_doi(doi=doi)\n",
    "        pmids.append(pmid)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9a03814",
   "metadata": {},
   "outputs": [],
   "source": [
    "for pmid in list(set(pmids)):\n",
    "    with open('Data2/pmids.txt',\"a\") as f:\n",
    "        f.write(pmid+'\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ea799d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-08-02 16:38:25,860 - DEBUG - pubmed2pdf.cli - Full log mode activated\n",
      "2025-08-02 16:38:25,861 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28923940\n",
      "2025-08-02 16:38:26,583 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28923940 failed from error list index out of range\n",
      "2025-08-02 16:38:26,583 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18539642\n",
      "2025-08-02 16:38:27,254 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18539642 failed from error list index out of range\n",
      "2025-08-02 16:38:27,254 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26415722\n",
      "2025-08-02 16:38:28,786 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26415722 failed from error list index out of range\n",
      "2025-08-02 16:38:28,786 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31540845\n",
      "2025-08-02 16:38:30,006 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 31540845 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:30,006 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16306079\n",
      "2025-08-02 16:38:30,721 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16306079 failed from error list index out of range\n",
      "2025-08-02 16:38:30,721 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22902743\n",
      "2025-08-02 16:38:32,782 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22025212\n",
      "2025-08-02 16:38:33,661 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 22025212 failed from error list index out of range\n",
      "2025-08-02 16:38:33,661 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19563753\n",
      "2025-08-02 16:38:34,930 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19563753 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:34,930 - INFO - pubmed2pdf.cli - Trying to fetch pmid 1155504\n",
      "2025-08-02 16:38:36,264 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 1155504 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:36,264 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29452422\n",
      "2025-08-02 16:38:36,938 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29452422 failed from error list index out of range\n",
      "2025-08-02 16:38:36,938 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34199109\n",
      "2025-08-02 16:38:37,583 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 34199109 failed from error list index out of range\n",
      "2025-08-02 16:38:37,583 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33077239\n",
      "2025-08-02 16:38:39,043 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33077239 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:39,043 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11846609\n",
      "2025-08-02 16:38:40,418 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 11846609 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:40,418 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12490681\n",
      "2025-08-02 16:38:41,113 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12490681 failed from error list index out of range\n",
      "2025-08-02 16:38:41,113 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26932670\n",
      "2025-08-02 16:38:41,826 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26932670 failed from error list index out of range\n",
      "2025-08-02 16:38:41,826 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18192189\n",
      "2025-08-02 16:38:42,538 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18192189 failed from error list index out of range\n",
      "2025-08-02 16:38:42,538 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28855728\n",
      "2025-08-02 16:38:42,538 - INFO - pubmed2pdf.cli - Trying to fetch pmid 37700285\n",
      "2025-08-02 16:38:42,538 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32641214\n",
      "2025-08-02 16:38:43,806 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32641214 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:43,806 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12529417\n",
      "2025-08-02 16:38:44,447 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12529417 failed from error list index out of range\n",
      "2025-08-02 16:38:44,447 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23290997\n",
      "2025-08-02 16:38:45,826 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23290997 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:45,827 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30713991\n",
      "2025-08-02 16:38:47,005 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15353123\n",
      "2025-08-02 16:38:49,056 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24100212\n",
      "2025-08-02 16:38:49,720 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24100212 failed from error list index out of range\n",
      "2025-08-02 16:38:49,720 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10802651\n",
      "2025-08-02 16:38:51,745 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23102856\n",
      "2025-08-02 16:38:53,923 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23102856 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:53,923 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32043035\n",
      "2025-08-02 16:38:55,160 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32043035 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:55,160 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32482256\n",
      "2025-08-02 16:38:56,401 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32482256 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:56,401 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36070983\n",
      "2025-08-02 16:38:57,955 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36070983 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:38:57,955 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16807381\n",
      "2025-08-02 16:38:58,613 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16807381 failed from error list index out of range\n",
      "2025-08-02 16:38:58,613 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11752295\n",
      "2025-08-02 16:38:59,291 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 11752295 failed from error list index out of range\n",
      "2025-08-02 16:38:59,291 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33067123\n",
      "2025-08-02 16:39:01,747 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33067123 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:01,747 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26797113\n",
      "2025-08-02 16:39:01,747 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18546601\n",
      "2025-08-02 16:39:03,851 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36205711\n",
      "2025-08-02 16:39:04,539 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36205711 failed from error list index out of range\n",
      "2025-08-02 16:39:04,539 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35929523\n",
      "2025-08-02 16:39:05,385 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35929523 failed from error list index out of range\n",
      "2025-08-02 16:39:05,385 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16832043\n",
      "2025-08-02 16:39:06,266 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16832043 failed from error list index out of range\n",
      "2025-08-02 16:39:06,266 - INFO - pubmed2pdf.cli - Trying to fetch pmid 11823860\n",
      "2025-08-02 16:39:08,266 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21849299\n",
      "2025-08-02 16:39:08,931 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 21849299 failed from error list index out of range\n",
      "2025-08-02 16:39:08,931 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19617889\n",
      "2025-08-02 16:39:11,269 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17160063\n",
      "2025-08-02 16:39:13,388 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31462323\n",
      "2025-08-02 16:39:13,388 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10362823\n",
      "2025-08-02 16:39:14,060 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10362823 failed from error list index out of range\n",
      "2025-08-02 16:39:14,060 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31340985\n",
      "2025-08-02 16:39:14,060 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29156006\n",
      "2025-08-02 16:39:14,717 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29156006 failed from error list index out of range\n",
      "2025-08-02 16:39:14,717 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35574918\n",
      "2025-08-02 16:39:15,407 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35574918 failed from error list index out of range\n",
      "2025-08-02 16:39:15,408 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20619403\n",
      "2025-08-02 16:39:16,743 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 20619403 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:16,743 - INFO - pubmed2pdf.cli - Trying to fetch pmid 17571346\n",
      "2025-08-02 16:39:16,743 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30452456\n",
      "2025-08-02 16:39:16,743 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29102484\n",
      "2025-08-02 16:39:17,960 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29102484 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:17,960 - INFO - pubmed2pdf.cli - Trying to fetch pmid 16672246\n",
      "2025-08-02 16:39:18,631 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 16672246 failed from error list index out of range\n",
      "2025-08-02 16:39:18,631 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15501903\n",
      "2025-08-02 16:39:19,290 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15501903 failed from error list index out of range\n",
      "2025-08-02 16:39:19,290 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10221616\n",
      "2025-08-02 16:39:20,660 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10221616 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:20,661 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33576824\n",
      "2025-08-02 16:39:21,352 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33576824 failed from error list index out of range\n",
      "2025-08-02 16:39:21,352 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32723696\n",
      "2025-08-02 16:39:23,654 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 32723696 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:23,654 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12200466\n",
      "2025-08-02 16:39:24,318 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12200466 failed from error list index out of range\n",
      "2025-08-02 16:39:24,318 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15666095\n",
      "2025-08-02 16:39:26,541 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35085395\n",
      "2025-08-02 16:39:27,203 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35085395 failed from error list index out of range\n",
      "2025-08-02 16:39:27,203 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30145981\n",
      "2025-08-02 16:39:27,203 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26109056\n",
      "2025-08-02 16:39:27,203 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33036008\n",
      "2025-08-02 16:39:28,940 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33036008 failed from error Invalid URL '': No scheme supplied. Perhaps you meant https://?\n",
      "2025-08-02 16:39:28,941 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24882617\n",
      "2025-08-02 16:39:30,142 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24882617 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:30,142 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30951376\n",
      "2025-08-02 16:39:31,008 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30951376 failed from error list index out of range\n",
      "2025-08-02 16:39:31,008 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19815776\n",
      "2025-08-02 16:39:32,019 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19815776 failed from error argument of type 'NoneType' is not iterable\n",
      "2025-08-02 16:39:32,019 - INFO - pubmed2pdf.cli - Trying to fetch pmid 27441287\n",
      "2025-08-02 16:39:33,538 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 27441287 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:33,538 - INFO - pubmed2pdf.cli - Trying to fetch pmid 21964334\n",
      "2025-08-02 16:39:37,074 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36822566\n",
      "2025-08-02 16:39:39,308 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36822566 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:39,308 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12421895\n",
      "2025-08-02 16:39:39,966 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12421895 failed from error list index out of range\n",
      "2025-08-02 16:39:39,966 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26804062\n",
      "2025-08-02 16:39:40,823 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26804062 failed from error list index out of range\n",
      "2025-08-02 16:39:40,823 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30929718\n",
      "2025-08-02 16:39:42,075 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30929718 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:42,075 - INFO - pubmed2pdf.cli - Trying to fetch pmid 7962413\n",
      "2025-08-02 16:39:42,960 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 7962413 failed from error list index out of range\n",
      "2025-08-02 16:39:42,960 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28863933\n",
      "2025-08-02 16:39:44,131 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28863933 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:44,132 - INFO - pubmed2pdf.cli - Trying to fetch pmid 31511876\n",
      "2025-08-02 16:39:44,747 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 31511876 failed from error list index out of range\n",
      "2025-08-02 16:39:44,747 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35088888\n",
      "2025-08-02 16:39:44,747 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30624659\n",
      "2025-08-02 16:39:45,419 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30624659 failed from error list index out of range\n",
      "2025-08-02 16:39:45,419 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12728018\n",
      "2025-08-02 16:39:46,072 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12728018 failed from error list index out of range\n",
      "2025-08-02 16:39:46,073 - INFO - pubmed2pdf.cli - Trying to fetch pmid 12021176\n",
      "2025-08-02 16:39:46,742 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 12021176 failed from error list index out of range\n",
      "2025-08-02 16:39:46,743 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34875061\n",
      "2025-08-02 16:39:47,438 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 34875061 failed from error list index out of range\n",
      "2025-08-02 16:39:47,439 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23555582\n",
      "2025-08-02 16:39:47,439 - INFO - pubmed2pdf.cli - Trying to fetch pmid 26395145\n",
      "2025-08-02 16:39:48,150 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 26395145 failed from error list index out of range\n",
      "2025-08-02 16:39:48,150 - INFO - pubmed2pdf.cli - Trying to fetch pmid 18174356\n",
      "2025-08-02 16:39:48,812 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 18174356 failed from error list index out of range\n",
      "2025-08-02 16:39:48,812 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24269084\n",
      "2025-08-02 16:39:49,983 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24269084 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:49,983 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28229988\n",
      "2025-08-02 16:39:52,261 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28229988 failed from error Invalid URL '': No scheme supplied. Perhaps you meant https://?\n",
      "2025-08-02 16:39:52,261 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23994285\n",
      "2025-08-02 16:39:54,664 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23994285 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:39:54,664 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34117589\n",
      "2025-08-02 16:39:56,922 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33880420\n",
      "2025-08-02 16:39:58,022 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19890323\n",
      "2025-08-02 16:40:00,274 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33313697\n",
      "2025-08-02 16:40:00,995 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33313697 failed from error list index out of range\n",
      "2025-08-02 16:40:00,995 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33880419\n",
      "2025-08-02 16:40:02,410 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29315421\n",
      "2025-08-02 16:40:03,134 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29315421 failed from error list index out of range\n",
      "2025-08-02 16:40:03,134 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29737471\n",
      "2025-08-02 16:40:05,587 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30037059\n",
      "2025-08-02 16:40:06,291 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 30037059 failed from error list index out of range\n",
      "2025-08-02 16:40:06,291 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15550397\n",
      "2025-08-02 16:40:08,462 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15550397 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:40:08,462 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24581625\n",
      "2025-08-02 16:40:09,708 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24581625 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:40:09,708 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22256780\n",
      "2025-08-02 16:40:10,432 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 22256780 failed from error list index out of range\n",
      "2025-08-02 16:40:10,432 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28370781\n",
      "2025-08-02 16:40:11,304 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28370781 failed from error list index out of range\n",
      "2025-08-02 16:40:11,304 - INFO - pubmed2pdf.cli - Trying to fetch pmid 36472596\n",
      "2025-08-02 16:40:12,050 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 36472596 failed from error list index out of range\n",
      "2025-08-02 16:40:12,050 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29402198\n",
      "2025-08-02 16:40:14,215 - INFO - pubmed2pdf.cli - Trying to fetch pmid None\n",
      "2025-08-02 16:40:14,977 - DEBUG - pubmed2pdf.cli - ** fetching of reprint None failed from error list index out of range\n",
      "2025-08-02 16:40:14,977 - INFO - pubmed2pdf.cli - Trying to fetch pmid 20729534\n",
      "2025-08-02 16:40:15,669 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 20729534 failed from error list index out of range\n",
      "2025-08-02 16:40:15,669 - INFO - pubmed2pdf.cli - Trying to fetch pmid 1424330\n",
      "2025-08-02 16:40:16,788 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 1424330 failed from error Invalid URL 'HdkXqKLxmGChjnf5zIbHBJtsxfbbs63egf3aCZnpsbYOO9ViFAMaeTPxOpsaXlIo': No scheme supplied. Perhaps you meant https://HdkXqKLxmGChjnf5zIbHBJtsxfbbs63egf3aCZnpsbYOO9ViFAMaeTPxOpsaXlIo?\n",
      "2025-08-02 16:40:16,788 - INFO - pubmed2pdf.cli - Trying to fetch pmid 28523980\n",
      "2025-08-02 16:40:17,445 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 28523980 failed from error list index out of range\n",
      "2025-08-02 16:40:17,445 - INFO - pubmed2pdf.cli - Trying to fetch pmid 34947871\n",
      "2025-08-02 16:40:18,334 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 34947871 failed from error list index out of range\n",
      "2025-08-02 16:40:18,334 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35092277\n",
      "2025-08-02 16:40:18,984 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35092277 failed from error list index out of range\n",
      "2025-08-02 16:40:18,985 - INFO - pubmed2pdf.cli - Trying to fetch pmid 35147192\n",
      "2025-08-02 16:40:19,646 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 35147192 failed from error list index out of range\n",
      "2025-08-02 16:40:19,646 - INFO - pubmed2pdf.cli - Trying to fetch pmid 24581986\n",
      "2025-08-02 16:40:20,809 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 24581986 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:40:20,809 - INFO - pubmed2pdf.cli - Trying to fetch pmid 33830236\n",
      "2025-08-02 16:40:21,463 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 33830236 failed from error list index out of range\n",
      "2025-08-02 16:40:21,463 - INFO - pubmed2pdf.cli - Trying to fetch pmid 15878921\n",
      "2025-08-02 16:40:22,177 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 15878921 failed from error list index out of range\n",
      "2025-08-02 16:40:22,177 - INFO - pubmed2pdf.cli - Trying to fetch pmid 23933037\n",
      "2025-08-02 16:40:23,424 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 23933037 failed from error direct_pdf_link() takes 1 positional argument but 3 were given\n",
      "2025-08-02 16:40:23,424 - INFO - pubmed2pdf.cli - Trying to fetch pmid 30081718\n",
      "2025-08-02 16:40:26,680 - INFO - pubmed2pdf.cli - Trying to fetch pmid 29663566\n",
      "2025-08-02 16:40:27,547 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 29663566 failed from error list index out of range\n",
      "2025-08-02 16:40:27,547 - INFO - pubmed2pdf.cli - Trying to fetch pmid 10592173\n",
      "2025-08-02 16:40:28,194 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 10592173 failed from error list index out of range\n",
      "2025-08-02 16:40:28,194 - INFO - pubmed2pdf.cli - Trying to fetch pmid 25605792\n",
      "2025-08-02 16:40:28,837 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 25605792 failed from error list index out of range\n",
      "2025-08-02 16:40:28,837 - INFO - pubmed2pdf.cli - Trying to fetch pmid 22911744\n",
      "2025-08-02 16:40:28,837 - INFO - pubmed2pdf.cli - Trying to fetch pmid 19933690\n",
      "2025-08-02 16:40:30,320 - DEBUG - pubmed2pdf.cli - ** fetching of reprint 19933690 failed from error list index out of range\n",
      "2025-08-02 16:40:30,320 - INFO - pubmed2pdf.cli - Trying to fetch pmid 32974805\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done downloading. All downloaded can be found in Data2/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['python3', '-m', 'pubmed2pdf', 'pdf', '--out', 'Data2/', '--pmidsfile', 'Data2/pmids.txt', '--verbose', '--errors', 'Data2/pubmed2pdf_log.txt'], returncode=0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import subprocess\n",
    "command = [\n",
    "    \"python3\", \"-m\", \"pubmed2pdf\", \"pdf\",\n",
    "    \"--out\", \"Data2/\",\n",
    "    \"--pmidsfile\", \"Data2/pmids.txt\",\n",
    "    \"--verbose\",\n",
    "    \"--errors\", \"Data2/pubmed2pdf_log.txt\"\n",
    "]\n",
    "subprocess.run(command)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ed1f13d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      1155504\n",
       "1      1424330\n",
       "2     10362823\n",
       "3     16306079\n",
       "4     10221616\n",
       "5     28863933\n",
       "6     28923940\n",
       "7     33313697\n",
       "8     36822566\n",
       "9     24269084\n",
       "10    24581986\n",
       "11    30624659\n",
       "12    33077239\n",
       "13    33576824\n",
       "14    34875061\n",
       "15    29452422\n",
       "16    20619403\n",
       "17    23102856\n",
       "18    34199109\n",
       "19    12421895\n",
       "20    22256780\n",
       "21    24581625\n",
       "22    36070983\n",
       "23    32723696\n",
       "24    33067123\n",
       "25    36472596\n",
       "26    11752295\n",
       "27    25605792\n",
       "28         NaN\n",
       "29         NaN\n",
       "30         NaN\n",
       "31    10592173\n",
       "32    33036008\n",
       "33    28370781\n",
       "34    34947871\n",
       "35    35929523\n",
       "36    27441287\n",
       "37    31540845\n",
       "38    33830236\n",
       "39    29315421\n",
       "40    35092277\n",
       "41    32482256\n",
       "42    12490681\n",
       "Name: PMID, dtype: object"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "\n",
    "to_retrieve = pd.read_csv(\"Data2/pubmed2pdf_log.txt\", header=None,dtype='str')\n",
    "to_retrieve.columns = [\"PMID\"]\n",
    "to_retrieve['PMID']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9473e5a2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'12282335'"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "a = get_pmcid_from_pmid(to_retrieve['PMID'][4])\n",
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "074b68a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "url = f\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC{a}/\"\n",
    "response = requests.get(url, headers=HEADERS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55ced425",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Am J Obstet Gynecol\n",
      ". Author manuscript; available in PMC: 2025 Jul 22.\n",
      "Published in final edited form as:\n",
      "Am J Obstet Gynecol. 2025 Apr;232(4 Suppl):S105–S123. doi:\n",
      "10.1016/j.ajog.2024.08.043\n",
      "Search in PMC\n",
      "Search in PubMed\n",
      "View in NLM Catalog\n",
      "Add to search\n",
      "Endometriosis and Adenomyosis Unveiled Through Single-cell Glasses\n",
      "Linda C Giudice\n",
      "Linda C Giudice\n",
      "1\n",
      "Center for Reproductive Sciences, Department of Obstetrics, Gynecology & Reproductive Sciences, University of California, San Francisco, San Francisco, CA 94143\n",
      "Find articles by\n",
      "Linda C Giudice\n",
      "1\n",
      ",\n",
      "Binya Liu\n",
      "Binya Liu\n",
      "1\n",
      "Center for Reproductive Sciences, Department of Obstetrics, Gynecology & Reproductive Sciences, University of California, San Francisco, San Francisco, CA 94143\n",
      "Find articles by\n",
      "Binya Liu\n",
      "1\n",
      ",\n",
      "Juan C Irwin\n",
      "Juan C Irwin\n",
      "1\n",
      "Center for Reproductive Sciences, Department of Obstetrics, Gynecology & Reproductive Sciences, University of California, San Francisco, San Francisco, CA 94143\n",
      "Find articles by\n",
      "Juan C Irwin\n",
      "1\n",
      "Author information\n",
      "Copyright and License information\n",
      "1\n",
      "Center for Reproductive Sciences, Department of Obstetrics, Gynecology & Reproductive Sciences, University of California, San Francisco, San Francisco, CA 94143\n",
      "✉\n",
      "Correspondence:\n",
      "Linda.Giudice@ucsf.edu\n",
      "PMC Copyright notice\n",
      "PMCID: PMC12282335  NIHMSID: NIHMS2086294  PMID:\n",
      "40253075\n",
      "The publisher's version of this article is available at\n",
      "Am J Obstet Gynecol\n",
      "Abstract\n",
      "Single cell technologies are expanding our understanding of endometriosis and adenomyosis, sister disorders of the uterine endometrium that contain similar complements of lesion cell types but located in different niches – outside and inside the endometrium, respectively. Both diseases cause significant morbidity and impaired quality of life among those affected, and current therapies mitigate most symptoms although with highly variable efficacy, duration of effect, and frequent intolerable side effects. Thus, there is a pressing need for transformative approaches to develop individualized therapies for the variety of presentations of endometriosis and adenomyosis symptoms and heterogeneity of lesion types histologically and architecturally. Single cell technologies are transforming understanding human physiology and pathophysiology in the reproductive system and beyond. This manuscript reviews endometriosis and adenomyosis clinical characteristics and the recent studies focused on eutopic endometrium and ectopic lesions at single cell resolution, the myriad of cell types and subtypes, cell-cell communications, signaling pathways, applications for novel drug discovery and therapeutic approaches, and challenges and opportunities that accompany this type of research.\n",
      "Keywords:\n",
      "endometriosis, adenomyosis, single cell, sequencing, inflammation, cell types, biomarkers, drug discovery, signaling pathways, precision medicine\n",
      "Key take-home messages from the studies reviewed herein include:\n",
      "Evidence strongly supports that superficial and deep endometriosis are related and are distinct entities from ovarian endometriomas, challenging a “one size fits all” approach to these lesions.\n",
      "In the eutopic endometrium from endometriosis patients, there are epithelial abnormalities during the window of implantation, which can compromise fertility, including undetectable progesterone-regulated genes (PAEP and CXCL14), and endometrial subcellular alterations across the cycle that can affect tissue homeostasis, as observed in patients with disease.\n",
      "Menstrual endometrium (as a “liquid biopsy” of the tissue) mimics many of the abnormalities in the late secretory phase of patients with endometriosis, including aberrant progesterone signaling in the stromal fibroblasts, which are key to pregnancy establishment and endometrial shedding and regeneration.\n",
      "Superficial peritoneal endometriosis lesions display basalis epithelial stem cell markers, supporting the role of this tissue compartment in pelvic disease origins.\n",
      "Fibroblasts and immune cells are key contributors to the pro-inflammatory, angiogenic environment in ovarian endometriomas, opening new opportunities for drug discovery.\n",
      "Data from superficial lesions from patients on progestins reveal a unique perivascular cell type with features promoting neoangiogenesis and immune cell trafficking, and an immunotolerant microenvironment consistent with escape from immune clearance and conducive to lesion establishment. These results are remarkable, given progestin exposure of the lesions analyzed - thus raising the question of how efficacious such treatments are in curbing disease progression.\n",
      "The eutopic endometrium from adenomyosis uteri reveal that pericyte progenitors differentiate to steroid hormone receptor positive fibroblasts and to ciliated epithelial cells. The latter strongly supports mesenchymal-epithelial transition in this tissue, opening possible new therapeutic targets.\n",
      "In adenomyosis lesions, fibroblasts do not originate from pericyte progenitors and differentiate to fibroblasts expressing extracellular matrix components and to smooth muscle cells, suggesting these cells lead to the fibrosis and smooth muscle hypertrophy observed clinically.\n",
      "Adenomyosis ciliated and unciliated epithelial cells and fibroblast-like cells displayed abnormal progesterone signaling. In addition, involvement of the WNT signaling pathway in epithelial populations opens new avenues for drug discovery for this disorder.\n",
      "Direct comparison of cell types in endometriosis and adenomyosis lesions is the next step of understanding these sister disorders.\n",
      "We conclude the review with an eye to the future – what Alice might see beyond the single-cell looking glass that connects endometrium and endometrial disorders with the trillions of cells of other tissues and organs in health and disease throughout the human body and the opportunities for novel diagnostic modalities and drug discovery for endometriosis, adenomyosis and related uterine and inflammatory conditions.\n",
      "Introduction\n",
      "Endometriosis and adenomyosis are common estrogen-driven uterine disorders derived mainly from the basalis endometrium, characterized by “ectopic” endometrial-like cells outside their normal location. A key feature distinguishing them is the location of the ectopic lesions. For endometriosis they are outside the uterus - mostly in the pelvis and invading pelvic organs\n",
      "1\n",
      "–\n",
      "4\n",
      ", and for adenomyosis they are within the uterine myometrium\n",
      "5\n",
      "–\n",
      "8\n",
      ". Endometriosis and adenomyosis disease lesions comprise similar phenotypic endometrial epithelium and mesenchymal cell types and display characteristics of inflammation, neoneuroangiogenesis, and fibrosis\n",
      "1\n",
      "–\n",
      "9\n",
      ". However, whether their cellular features are equivalent in different anatomic niches is unknown and is of importance for disease stratification in view of the vastly heterogeneous lesion architectural structures and for developing tailored, personalized, patient-specific treatments and prognostic disease outcome indicators.\n",
      "These complex “sister” conditions often occur concomitantly\n",
      "10\n",
      "and result in similar symptoms of pain and uterine bleeding treated medically (mostly hormonal) with variable efficacies and are associated with infertility and similar poor reproductive outcomes\n",
      "1\n",
      ",\n",
      "4\n",
      "–\n",
      "13\n",
      ". They also greatly impact quality of life of those affected\n",
      "14\n",
      "–\n",
      "18\n",
      ". Given the devastating impact of these disorders and limited success of effective therapies and disease prevention and cure, a transformative approach to disease phenotyping and novel, personalized therapeutics is essential to improve the lives of millions of patients with endometriosis and adenomyosis across the globe.\n",
      "Single cell technologies and precision medicine approaches are early in their application to these disorders and are beginning to provide insights into commonalities/differences of cell types in endometriosis, adenomyosis, and eutopic endometrium, and the role of their unique anatomical niches influencing cell behavior, disease establishment and pathophysiology. Moreover, single cell data and associated signaling pathways are anticipated to lead to novel drug candidate discovery to control symptoms, minimize disease burden resulting ultimately in cure, identifying diagnostic biomarkers, and informing disease classification and prognostics for disease courses. The time is now to move the needle on both disorders through single cell “glasses”!\n",
      "Regarding endometrium, an atlas of normal, cycling endometrium has been derived at single cell resolution by several groups\n",
      "19\n",
      "–\n",
      "22\n",
      "(also see Chapter 3). These studies underscore heterogeneity of specific cell types and altered features in different hormonal states across the menstrual cycle and provide a key backdrop to studies at the single cell level of endometrium of patients with endometriosis and adenomyosis, endometriosis lesions, and adenomyosis lesions. Herein, we begin with an overview of clinical characteristics of endometriosis and adenomyosis (sister disorders but not identical twins), followed by a summary of current advances in scRNA sequencing data of endometriosis and adenomyosis disease lesions, corresponding eutopic endometrium, control endometrium, and disease-specific and common cell features across tissues. We conclude with a review of challenges of conducting research on human tissues, especially those that are steroid hormone responsive, and how these data inform looking to the future through single cell glasses.\n",
      "Endometriosis and adenomyosis – sister disorders, not identical twins\n",
      "Endometriosis and adenomyosis lesions both comprise uterine endometrial cells found outside their normal location (\n",
      "Figure 1\n",
      "). For endometriosis, these cells form lesions - mostly in the pelvis - arriving mainly by retrograde menstruation and causing a profound inflammatory response\n",
      "1\n",
      "–\n",
      "4\n",
      ". Risks of developing endometriosis are ~ 50% genetic and ~50% environmental\n",
      "2\n",
      ",\n",
      "4\n",
      ". In contrast, adenomyosis-associated endometrial cells (epithelium and stroma) are found within the uterus – in the uterine myometrium – where they induce a local inflammatory response, smooth muscle hypertrophy and often an enlarged, globular uterus\n",
      "5\n",
      "–\n",
      "8\n",
      ". Risks of developing adenomyosis are multi-parity, prior uterine surgery, adult exposure to tamoxifen, and in utero exposures to estrogens\n",
      "5\n",
      ",\n",
      "7\n",
      ",\n",
      "23\n",
      ". Endometriosis has been more extensively studied due to ease of obtaining endometrial tissue and sampling disease lesions at surgery. Adenomyosis has had far less molecular, genetic, and genomic analyses, as uterine myometrial lesions and the endometrial-myometrial junction are accessible mainly from hysterectomy specimens, although the eutopic endometrium proper has been studied in adenomyosis patients due to ease of access by office biopsy.\n",
      "Figure 1.\n",
      "Open in a new tab\n",
      "Comparison of endometriosis and adenomyosis pathogenesis, risk factors, diagnostic approaches, pathophysiology, symptoms, and therapies. Adapted from Bulun et al, 2023\n",
      "9\n",
      ", with permission.\n",
      "Despite their differences in anatomic location, the two disorders share common origins, some molecular characteristics, symptoms, and treatment strategies\n",
      "5\n",
      "–\n",
      "9\n",
      ",\n",
      "12\n",
      "(\n",
      "Figure 1\n",
      "). Both are diseases of the basalis endometrium with collective cell migration out of the uterus into the pelvis or into the muscle that may also involve epithelial to mesenchymal cell transition\n",
      "5\n",
      ",\n",
      "6\n",
      ",\n",
      "9\n",
      ". They share postulated origins of Mullerian remnant transformation and differentiation of endometrial stem/progenitor cells, and some common epithelial cancer driver mutations although they rarely transform into cancers\n",
      "6\n",
      ",\n",
      "9\n",
      ". They share some histologic features (e.g., deep infiltrating endometriosis and adenomyosis\n",
      "24\n",
      ", both comprise endometrial epithelial cells, stromal cells, vascular, smooth muscle, stem/progenitor, and immune cells, and have heterogeneous lesion phenotypes\n",
      "5\n",
      ",\n",
      "7\n",
      ". They are estrogen-dependent, progesterone (P\n",
      "4\n",
      ")-resistant disorders, display increased estradiol (E\n",
      "2\n",
      ") production in disease (due to aberrant aromatase expression in the epithelium and some stromal cells) and decreased E\n",
      "2\n",
      "metabolism and conversion to testosterone\n",
      "2\n",
      ",\n",
      "4\n",
      ",\n",
      "6\n",
      ",\n",
      "9\n",
      ",\n",
      "23\n",
      ",\n",
      "25\n",
      ",\n",
      "26\n",
      ". Both are also inflammatory disorders and have the activated inflammasome as the final common pathway for local and systemic inflammation\n",
      "2\n",
      ",\n",
      "3\n",
      ",\n",
      "27\n",
      ". And both can be induced in the developing fetus and/or adult (in animal models) and are associated in humans with endocrine disrupting chemicals found in the environment and personal care products\n",
      "23\n",
      ".\n",
      "Endometriosis and adenomyosis share similar clinical features (\n",
      "Figure 1\n",
      ") including dysmenorrhea, chronic pelvic pain, infertility, and poor pregnancy outcomes, and they often co-exist in the same patient and relapse post fertility-preserving surgery\n",
      "5\n",
      "–\n",
      "7\n",
      ",\n",
      "9\n",
      ",\n",
      "12\n",
      ",\n",
      "28\n",
      ". Symptom histories are key in diagnosing both disorders, but many symptoms overlap between them and with other disorders\n",
      "2\n",
      ",\n",
      "4\n",
      ". Standards for endometriosis and adenomyosis diagnosis are histologic confirmation of endometrial-like cells in lesions, and pelvic MRI and transvaginal ultrasound are useful for detecting endometriomas and deep disease as well as adenomyosis-associated widened endometrial-myometrial junction and adenomyomas\n",
      "4\n",
      ",\n",
      "29\n",
      ".\n",
      "Treatment paradigms are similar for both disorders and include combined estrogen/ progestin contraceptive steroids, progestins alone, progestin-containing intrauterine systems, selective progesterone receptor modulators (SPRMs), selective estrogen receptor modulators (SERMs), GnRH analogs, aromatase inhibitors accompanied by ovulation suppression, and dopamine agonists, and medically assisted and advanced reproductive technologies for infertility\n",
      "1\n",
      ",\n",
      "4\n",
      ",\n",
      "8\n",
      ",\n",
      "12\n",
      ". Overall, medical therapies for pain have limited efficacy and/or intolerability for many patients. Endometriosis lesion removal or hysterectomy is largely effective for symptom management, although ~ 50% of patients recur within 5 years after conservative surgery and ~15% after hysterectomy\n",
      "1\n",
      ",\n",
      "4\n",
      ",\n",
      "12\n",
      ". Adenomyosis, on the other hand, can be excised but this is rarely performed due to technical challenges, and while hysterectomy is curative, it compromises fertility pursuits\n",
      "30\n",
      ".\n",
      "Thus, developing non-invasive diagnostics as stand-alone assessments or as adjuncts to imaging technologies, and innovative and personalized therapies are major goals for both disorders.\n",
      "Endometriosis through single cell glasses\n",
      "Subsequent to the single cell cartography of human endometrium published by Wang et al\n",
      "19\n",
      "(see Chapter 3), several groups have recently reported eutopic endometrial scRNAseq data of patients with endometriosis, endometriosis lesions and control eutopic endometrium\n",
      "20\n",
      ",\n",
      "31\n",
      "–\n",
      "36\n",
      "(\n",
      "Table 1\n",
      "). Types of samples, inclusion/exclusion criteria, choice of controls, study designs, numbers of cells sequenced, and number of reads (i.e., base pairs sequenced) are key to data integrity and interpretation and to comparative analyses across tissues and studies. They have provided insights into origins of ectopic disease, the heterogeneity of cell types/subtypes, unique clusters and signatures of endometrium and ectopic lesions informing mechanisms and pathways involved in cellular dysfunctions relevant to pain and fertility compromise, novel cell-cell communications, relationships between lesions and eutopic tissue/cells, and biomarker discovery. Across these studies, samples were obtained in different hormonal milieu (menstrual cycle phase, exogenous hormones), and between 55,000 to 378,000 cells were sequenced (\n",
      "Table 1\n",
      "). As cycle phase and exogenous hormone use are key drivers of endometrial gene expression\n",
      "37\n",
      "–\n",
      "44\n",
      ", comparison of data across studies can be challenging and mixed hormonal milieu are sometimes experienced within the same study in both cases and/or controls. Below we summarize results of studies focused exclusively on eutopic endometrium of patients with endometriosis relevant to endometrial function and dysfunction and biomarker discovery, followed by studies focused on ovarian endometriomas, superficial, and/or deep infiltrating disease, and eutopic endometrium, relevant to disease heterogeneity, pathophysiology, pathogenesis, and in the setting of progestins – commonly prescribed for symptom management of patients with endometriosis and dysmenorrhea and chronic pelvic pain.\n",
      "Table 1.\n",
      "Highlights of recent endometriosis-related scRNAseq studies.\n",
      "Reference\n",
      "Tissue Analyzed\n",
      "Total cells sequenced\n",
      "Technology/Platform\n",
      "Total subjects\n",
      "Endometriosis Type and ASRM Stage\n",
      "Controls\n",
      "Race/ Ethnicity\n",
      "Hormones/IUD\n",
      "Cycle Phase Endometrial histology\n",
      "Ma 2021\n",
      "endometrium, ovarian endometrioma\n",
      "55,000\n",
      "scRNAseq (10X)\n",
      "n=6 subjects (n=3 cases, n=3 controls)\n",
      "ASRM Stage III, IV ovarian endometriomas\n",
      "Healthy controls without endometriosis\n",
      "N/A\n",
      "None\n",
      "All in prolifeative phase\n",
      "Garcia-Alonso 2022\n",
      "endometrium (functionalis, full thickness), endometriosis peritoneal lesions (red, white, black)\n",
      "98,569 sequenced\n",
      "scRNAseq, snRNA seq, spatial profilng. Lesion microarray data (\n",
      "GSE141549\n",
      ")\n",
      "n=3 functionalis n=6 full thickness; Microarray data controls: endom n=42, peritoneum n=12; n=9 red, 9 white, 11 black\n",
      "Peritoneal disease: red, white, black\n",
      "Healthy controls (functionalis layer) n=6 full thickness without reproductive disorders; normal peritoneum\n",
      "N/A\n",
      "None\n",
      "Proliferative, secretory; Microarray samples: Control Endo 17PE, 25SE; Perit 4PE, 8SE; Lesions: Red 2PE, 7 SE; White 5PE, 4SE; Black 6PE, 5 SE\n",
      "Shih 2022\n",
      "menstrual endometrium\n",
      "43,054\n",
      "scRNAseq (10X)\n",
      "n=33 subjects (n=11 Dx, n=13 sx, n=9 controls)\n",
      "N/A; another group with Sx but no Dx.\n",
      "No endometriosis diagnosis. ? Other GYN disorders\n",
      "White: 10 Dx, 13 sx, 7 controls Black: 0 Dx, 0 sx, 1 control\n",
      "Hispanic: 0 Dx, 0 sx, 0 cont\n",
      "Mixed: 0 Dx,0 sx, 1 control\n",
      "Other: 1 Dx,0 sx, 0 control\n",
      "None on hormones, except 1 case used vaginal P4. Re-analysis showed no impact on results\n",
      "Menstrual (heaviest flow, mostly CD 1 or 2)\n",
      "Tan 2022\n",
      "endometrium, enodmetriosis lesions\n",
      "122,000\n",
      "scRNAseq (10X) IMC organoids\n",
      "n=27 subjects (n=19 cases, n=8 controls; n=14 sequenced\n",
      "ASRM Stage II-IV, peritoneal lesions, ovarian “lesions”, organoids\n",
      "No endometriosis or infammatory conditions\n",
      "White: 9 cases, 3 controls\n",
      "Asian: 4 cases, 2 controls\n",
      "Hisp: 5 cases, 3 controls\n",
      "Black: 1 case, 0 controls\n",
      "Progestin [NETA, LVN oral, IUD, drosperinone, norelgestromin] +/− ethinyl E2, medroxy-progesterone acetate, copperIUD\n",
      "wkly PE: 3 cases, 2 control; inactive 3 cases, 0 control; mens: 1 case, 0 control, exog hormone effect: 7 cases, 7 control\n",
      "PE: 0 cases, 2 controls\n",
      "IE: 0 cases, 1 control\n",
      "ESE: 4 cases, 0 control\n",
      "N/A: 1 case, 1 control\n",
      "Fonseca 2023\n",
      "endometrium, endometriosis lesions, unaffected ovary and peritoneum\n",
      "373,851\n",
      "digital scRNAseq\n",
      "n=21 subjects n=17 cases, n=4 controls n-54 specimens collected\n",
      "Cases: n=17 Endometrium Endometrioma Superficial and deep disesase n=9 w/o GYN disorders; n=8 w/ adenomyosis, uterine fibroids +/− polyp\n",
      "Controls n=4 n=3 PMP, 1 peri-menopause All no evidence of endometriosis, all w/ leiomyoma +/− adenomyosis +/− uterine polyp\n",
      "White: 13 cases,2 controls\n",
      "Black: 2 cases, 0 controls\n",
      "Asian: 1 case, 0 control\n",
      "Other: 1 case, 0 control\n",
      "Cases\n",
      ": n=14 on no hormones. n=1 w/ vaginal ring E+P; n=1 on E/T, P4; n=1 on E2+P4.\n",
      "Controls\n",
      ": 3 of 4 on E +/− P4; NETA; n=1 peri-menopause in luteal phase\n",
      "Cases\n",
      ": Proliferative phase: n=7; secretory phase: n=9; N/A (on hormones) n=3.\n",
      "Controls\n",
      ": Proliferative n=0, Secretory n=1, N/A (on hormones) n=3\n",
      "Huang 2023\n",
      "endometrium\n",
      "128,243\n",
      "scRNAseq (10X)\n",
      "n=10 sequenced n=6 cases, n=4 controls\n",
      "ASRM Stage I,II\n",
      "No endometriosis, benign ovarian cysts\n",
      "N/A\n",
      "None\n",
      "EPE: 3 cases, 3 controls\n",
      "MSE: 3 cases, 3 controls\n",
      "LSE: 1 control\n",
      "Almonte 2023\n",
      "endometrium\n",
      ">250,000\n",
      "scRNseq (10X)\n",
      "n=29 cases, 7 controls w/o disease and 8 controls with fibroids\n",
      "ASRM stage I-IV\n",
      "No endometriosis or other GYN d/o; No endo but + uterine fibroids\n",
      "N/A\n",
      "None\n",
      "Proliferative:12 cases, 5 controls. Secretory: 14 cases, 3 controls.\n",
      "Open in a new tab\n",
      "Abbreviations:\n",
      "CD=cycle day; dx=diagnosis; E\n",
      "2\n",
      "=estradiol; ESE=early secretory phase; exog=exogenous; IE=interval endometrium; IMC=imaging mass cytometry; IUD=intrauterine device; LSE=late secretory phase; LVN=levonorgestrel; MSE=mid-secretory phase; N/A=not available; NETA=norethindrone acetate; P=progestin; P\n",
      "4\n",
      "=progesterone; PE=proliferative phase; SE=secretory phase; sx=symptoms, w/= with; +/− = with and/or without. Adapted from Giudice et al\n",
      "4\n",
      ", with permission.\n",
      "Eutopic endometrium\n",
      "Window of implantation epithelial abnormalities in infertile endometriosis patients\n",
      "Endometrial biopsies of cases with stage I/II disease (revised American Society for Reproductive Medicine (rASRM) scoring system)\n",
      "45\n",
      "and infertility (n=3 WOI, n=3 early proliferative phase) versus n=7 fertile controls without disease (3 WOI, 3 proliferative, 1 late secretory phase) were studied by Huang et al\n",
      "34\n",
      ". A total of 128,243 cells were sequenced, and 30 type/cell type clusters were identified. Notably, a unique epithelial cell cluster in WOI endometrium of endometriosis patients was identified that lacked expression of two key genes associated with implantation:\n",
      "PAEP\n",
      "(progesterone-associated endometrial protein) and\n",
      "CXCL14\n",
      "(chemokine (CXC motif) ligand 14) (\n",
      "Figure 2\n",
      "). PAEP, also known as glycodelin A and placental protein 14, is involved in immune response regulation, promoting opening of the WOI by lowering maternal immune responses to an implanting embryo\n",
      "46\n",
      ". CXCL14 is involved in activating the innate immune response stimulating chemotaxis of uterine natural killer cells (uNK) and other monocytes to epithelial glands for implantation\n",
      "46\n",
      ". Other findings included absence of normal variation of endometrial uNK and T cell numbers across the cycle, and lower anti-inflammatory IL-10 and more pro-inflammatory cytokines expressed in endometrial immune cells in the WOI versus in the proliferative phase in cases and the opposite in controls. Notably, 11 ligand–receptor pairs were upregulated between endometrial immune and epithelial cells during the WOI of cases.\n",
      "Figure 2. Absence of key genes that open the window of implantation in eutopic endometrial epithelial cluster in infertile patients with Stage I/II endometriosis.\n",
      "Open in a new tab\n",
      "Panel A\n",
      ". t-SNE projection of endometrial cells colored based on their sample origin and cycle phase.\n",
      "Panel B\n",
      ". Expression patterns of PAEP and CXCL14 projected on t-distributed stochastic neighbour embedding (t-SNE) plot. From Huang et al, 2023\n",
      "34\n",
      ", with permission.\n",
      "Thus, lower epithelial receptivity markers, abnormal immune cell frequencies, a pro- inflammatory WOI, and new insights into cell-cell communications between epithelium and immune cells support an important role of the immune system and an adverse environment for embryo implantation for fertility and potentially for poor pregnancy outcomes in patients with stage I/II endometriosis who do achieve pregnancy versus fertile controls.\n",
      "Endometrial subcellular alterations across the cycle in endometriosis patients\n",
      "In a recent study, our group sequenced eutopic endometrium (237,620 cells) from 29 subjects with endometriosis (stages I/II and III/IV) and 15 controls without disease (7 healthy controls, 8 with symptomatic uterine fibroids), across the menstrual cycle\n",
      "36\n",
      ". We identified 5 main cell groups: stromal fibroblasts, smooth muscle, endothelial, epithelial, and immune. In the latter, further analysis revealed 18 sub-populations including uterine NK cells, T cell subsets, monocytes, macrophages, and dendritic cells. Epithelial subpopulations comprised glandular, luminal and ciliated cells, with abnormal expression of\n",
      "CXCL14, PAEP, CCL20\n",
      ",\n",
      "MMP7,\n",
      "and\n",
      "SFRP4\n",
      "genes in cases, regardless of phase. Endometrium in controls with FIGO Type 2 uterine fibroids revealed overexpression of\n",
      "WFDC2\n",
      "and\n",
      "DAXX\n",
      "genes in cases.\n",
      "In immune cells, fibroblasts, and epithelial cells, cytokine production, stress signal and pro-inflammatory pathways in cases were noted, similar to Huang et al\n",
      "34\n",
      "findings, and enriched pathways in epithelial and ciliated cells related to migration and cell motility were higher in cases. Cell communication analysis revealed a complex network of inflammatory pathways, with significant involvement of classical monocytes and NK cells. Finally, genes implicated in ovarian cancer were upregulated in endometrium of cases compared to controls, confirming known association of endometriosis with specific subtypes of ovarian cancer\n",
      "12\n",
      ". Thus, in the eutopic endometrium in patients with endometriosis, there are numerous alterations in specific cell types across the cycle versus controls without disease and disease-free controls with uterine fibroids\n",
      "36\n",
      ". These can predispose to impaired fertility and tissue dyshomeostasis.\n",
      "Menstrual endometrium, disease pathophysiology, candidate biomarker discovery\n",
      "A recent scRNAseq study\n",
      "32\n",
      "of endometrial tissue in menstrual efflux (not biopsies) of n=11 cases with laparoscopic evidence of endometriosis (undefined stage) (11,924 cells) and n=9 healthy controls with no disease or symptoms (14,327 cells) revealed 18 cell clusters. These included abundant uNKs, stromal fibroblast, epithelial, B, T, and myeloid cell clusters, and a small plasmacytoid dendritic cell (pDC) cluster. Further analysis separated data by cases vs controls and revealed a marked decrease in uNK1 and uNK2 cells and significantly enriched B cells in cases (\n",
      "Figure 3A\n",
      "). Log2 odds ratio (OR) analysis confirmed uNK1/2 significantly enriched in controls and B cells greatly enriched in cases (\n",
      "Figure 3B\n",
      "). Moreover, menstrual endometrium scRNA seq revealed decreased decidualization markers in stromal fibroblasts (reflecting P\n",
      "4\n",
      "-resistance) and pro-inflammatory stromal cells, as observed by others in FACS-sorted cell analyses and\n",
      "in vitro\n",
      "cell culture studies\n",
      "6\n",
      ",\n",
      "47\n",
      ".\n",
      "Figure 3. Distinct Cell Compositions and Cell Enrichment in Menstrual Endometrium from patients with Endometriosis versus Controls and Model of Disease.\n",
      "Open in a new tab\n",
      "Panels A\n",
      "and\n",
      "B\n",
      "show low abundance of uNK1 and uNK2 cells and high abundance of B cells in patients with endometriosis.\n",
      "Panel C\n",
      "shows proposed pathogenesis and pathophysiology implied by the data. From Shih et al, 2022\n",
      "32\n",
      ", with permission.\n",
      "Overall, this study demonstrates that menstrual endometrium (commonly referred to as a “liquid biopsy of the endometrium”) reflects secretory endometrium abnormalities as possible biomarkers of disease. The proposed disease model is shown in\n",
      "Figure 3C\n",
      "wherein defective endometrial stromal fibroblast decidualization is driven by multiple factors including inflammation, chronic endometritis, stress, and/or P\n",
      "4\n",
      "-resistance – which all have clinical relevance for pregnancy establishment and success, as well as tissue homeostasis. This, in turn, may direct fibroblast differentiation to a chronic inflammatory phenotype and senescence, which may also impair decidualization. Reduced decidualization may also compromise the infiltration and proliferation of uNK cells, likely to be important for senescent cell removal and important for an optimal setting for embryo nidation\n",
      "32\n",
      ".\n",
      "Ectopic endometrium\n",
      "Superficial peritoneal lesions display basalis epithelial stem cell markers\n",
      "In their landmark spatial-temporal single cell transcriptomic study\n",
      "48\n",
      ", García-Alonso et al\n",
      "20\n",
      "analyzed endometrium (3 functionalis and 6 full thickness (functionalis and basalis) tissues) across the cycle and identified 14 unique cell clusters that localized to corresponding endometrial compartments and niches. The proposed endometrial epithelial stem-cell niche resides within the basalis layer and contains SOX9+ cells and other “stem” markers (e.g., LGR5, SSEA1, β-catenin, N-cadherin)\n",
      "49\n",
      ",\n",
      "50\n",
      ", and Garcia-Alonso et al\n",
      "20\n",
      "found proliferative SOX9+LGR5+ cells mapped to the basalis stem cell niche. Epithelial stem cells are believed to play a role in endometriosis pathogenesis, and by leveraging bulk (microarray) transcriptomic data of superficial peritoneal lesions (9 red, 9 white, 11 black) (\n",
      "GSE141549\n",
      "), this team identified enrichment of SOX9+LGR5+ markers in peritoneal lesions in the proliferative phase, compared to normal endometrium (n=42) and peritoneum (n=12)\n",
      "20\n",
      ". A proliferative phenotype of endometriosis lesion epithelial cells is consistent with P\n",
      "4\n",
      "-resistance noted in endometriosis\n",
      "4\n",
      ",\n",
      "6\n",
      ". Moreover, dysfunctional epithelium as a major driver of endometrial diseases, including endometriosis, and discovery of SOX9+LGR5+ stem-like populations dominant in endometriosis peritoneal lesions are foundational observations about pathogenesis and pathophysiology of superficial endometriosis disease lesions deriving from the basalis layer epithelium.\n",
      "Endometrioma, superficial and deep infiltrating endometriosis cell atlas\n",
      "A comprehensive scRNAseq analysis of >370,000 individual cells from 8 ovarian endometriomas, 28 superficial and deep disease lesions, 10 eutopic endometrium, 4 unaffected ovary, and 4 endometriosis-free peritoneum samples, has resulted in a cell atlas of endometrial-type epithelial cells, stromal cells, and microenvironment cell populations across tissue types\n",
      "35\n",
      ". Samples were derived from 17 cases (9 only endometriosis; 8 endometriosis and adenomyosis, uterine fibroids, and/or uterine polyps). 14 were cycling (7 proliferative and 7 secretory phases), and 3 were on contraceptive steroids (\n",
      "Table 1\n",
      "). Controls (n=4) were postmenopausal (3) or perimenopausal (1), without endometriosis but with uterine fibroids +/− adenomyosis +/− uterine polyps, and all were on hormonal replacement regimens. Histologic and macroscopic features of specimens from one patient, shown in\n",
      "Figure 4A\n",
      ", underscore the heterogeneity of the lesions by cell abundance, cell types, and tissue architecture.\n",
      "Figure 4. Endometriosis lesion scRNAseq analysis.\n",
      "Open in a new tab\n",
      "Panel A.\n",
      "Histologic (H&E staining) and architectural heterogeneity of endometriosis lesions.\n",
      "Panel B.\n",
      "Correlations Based on Cluster Frequency Across All Specimens Profiled by scRNAseq.\n",
      "Panel C.\n",
      "Epithelial differential gene expression and signaling pathways in endometriomas and superficial and deep disease (P < 0.05 and log2 FC > 1). From Fonseca et al, 2023\n",
      "35\n",
      ", with permission.\n",
      "The major cell types identified from scRNAseq of all cells from n=49 samples comprising the 5 major tissue classes (endometrioma, peritoneal endometriosis (deep and superficial), unaffected peritoneum, eutopic endometrium, unaffected ovary) included epithelial, smooth muscle, mesenchymal, endothelial, mast, myeloid, B, and T/NKT cells, and erythrocytes. Correlations based on cluster frequency across all specimens profiled by scRNAseq are shown in\n",
      "Figure 4B\n",
      "and reveal 3 main clusters: 1) mainly endometriosis specimens; 2) mix of endometriomas (5 of the 8 endometrioma specimens) and eutopic endometrium (5 of the 10 eutopic samples; 3) only controls: unaffected ovary and 5 of the 10 eutopic endometrium samples\n",
      "35\n",
      ".\n",
      "Endometrioma, endometrium, superficial and deep endometriosis epithelia revealed distinct cell subtypes and signaling pathways, and endometriomas displayed markedly different epithelial differential gene expression and signaling pathways compared to deep and superficial peritoneal disease and eutopic endometrium (\n",
      "Figure 4C\n",
      "). Interestingly, endometriomas displayed immune cell and complement activation and were enriched in B cells and plasma cells, suggesting possible infection in endometriomas and a unique role for B cells which have received limited attention in endometriosis pathophysiology\n",
      "51\n",
      "–\n",
      "53\n",
      ". Peritoneal disease was enriched in mast cells and T/NK-T cells engaging novel immune targets, and some histologically negative mesothelium surprisingly had disease signatures, although depth of lesion evaluation for endometriosis tissue may vary among pathologists.\n",
      "Endometrial-type epithelial cells and fibroblasts exhibited differential gene expression associated with deep or superficial disease status (\n",
      "Figure 5\n",
      "). Interestingly, deep infiltrating disease epithelium displayed upregulation of nerve growth factor, suggesting a role of the epithelium in promoting innervation of this disease type which is particularly painful\n",
      "12\n",
      ".\n",
      "Figure 5. Molecular signatures of epithelial and stromal cells in superficial peritoneal and deep endometriosis differ.\n",
      "Open in a new tab\n",
      "Endometrial-type epithelial cells (\n",
      "Panel A\n",
      ") and fibroblasts (\n",
      "Panel B\n",
      ") exhibit DGE associated with deep or superficial status. From Fonseca et al, 2023\n",
      "35\n",
      ", with permission.\n",
      "While somatic mutations have been reported in endometrium and endometriosis lesions, mainly involving KRAS activating mutations and ARID1A loss of function mutations\n",
      "54\n",
      ",\n",
      "55\n",
      ", Fonseca et al\n",
      "35\n",
      "confirmed these somatic mutations in various lesions (\n",
      "Figure 6A\n",
      ") and demonstrated\n",
      "in vivo\n",
      "transcriptional consequences of these mutations at the single cell level (\n",
      "Figure 6B\n",
      "). ARID1A mutation was associated with pro-lymphangiogenic stroma (\n",
      "Figure 6C\n",
      ") and adjacent mesothelium pro-inflammatory features and ciliated epithelial signatures - consistent with ovarian oncogenic potential. It is well-known that endometrioid ovarian cancer (EOC) and clear cell ovarian cancer (CCOC) arise in association with endometriosis, suggesting endometrial-type epithelial cells may be precursors for these tumors. A multi-subject single cell deconvolution analysis of the single cell data (\n",
      "Figure 7\n",
      ") revealed strong enrichment of signatures for\n",
      "ciliated\n",
      "endometrial-type epithelial cells compared with the other keratin-positive clusters. In contrast, high grade serous ovarian cancer, an epithelial ovarian cancer type,\n",
      "not\n",
      "believed to originate from endometriosis epithelial cells, showed no pattern for preferential enrichment of any epithelial cluster.\n",
      "Figure 6. Endometrium, endometriosis, and endometriomas share somatic mutations with transcriptional consequences.\n",
      "Open in a new tab\n",
      "Panel A.\n",
      "Mutations detected in each lesion.\n",
      "Panel B.\n",
      "Differential gene expression by mutation state.\n",
      "Panel C.\n",
      "Epithelial expression of lymph-angiogenesis markers by ARID1A staining status. From Fonseca et al, 2023\n",
      "35\n",
      ", with permission.\n",
      "Figure 7. Deconvolution Analysis Reveals Endometrial-type Epithelium Cell-type Signatures in Endometriosis-associated Ovarian Cancers.\n",
      "Open in a new tab\n",
      "Multi-subject single-cell deconvolution revealed strong enrichment of signatures for\n",
      "ciliated endometrial-type epithelial cells\n",
      "compared with the other keratin-positive clusters. HGSOC, an epithelial ovarian tumor type thought\n",
      "not\n",
      "to originate from endometriosis epithelial cells showed no pattern for preferential enrichment of any epithelial cluster. From Fonseca et al, 2023\n",
      "35\n",
      ", with permission.\n",
      "Key takeaways:\n",
      "Endometrium and all three lesion types versus control endometrium displayed different cell/molecular signatures across tissues, consistent with restructuring/transcriptional reprogramming in lesions.\n",
      "Epithelial and stroma differed by site with striking differential gene expression in endometriomas and peritoneal lesions suggesting these are distinct disease entities.\n",
      "Superficial peritoneal and deep infiltrative endometriosis epithelium and fibroblasts had some differential gene expression and pathways, but similar structural genes to each other, suggesting they are part of the same continuum and distinct from endometrioma.\n",
      "While this study is a tour de force of endometriosis lesion and eutopic endometrium single cell transcriptomic analysis giving major insights into cell- and tissue-specific features and functions, including oncogenic potential of endometrioma epithelial cells, it underscores some of the challenges in conducting these types of studies. These include the presence of co-existing gynecologic disorders that can confound data interpretation; cells sequenced comprise a minor population in most samples (e.g., <1% of cells profiled), and controls and normal tissues are challenging to obtain e.g., controls were peri- or post-menopausal on hormone replacement therapies and had other uterine disorders. Nonetheless, these data are important and provide foundational information for comparison with other studies on eutopic and ectopic lesions in endometriosis patients and controls without disease.\n",
      "Fibroblasts and immune cells: key contributors to a pro-inflammatory, angiogenic environment in endometriomas.\n",
      "Sequencing 55,000 single cells from 3 endometriomas, 3 matched eutopic endometrium, and 3 endometrial samples from controls without disease in the proliferative phase, Ma et al\n",
      "31\n",
      "found 9 cell type clusters: fibroblasts, macrophages/monocytes, neutrophils, and T, NK, epithelial, endothelial, and mast cells, and unknown. Thirteen fibroblast subtype clusters were identified with significant differences in subtype composition among ectopic endometrium, eutopic endometrium, and normal endometrium. Fibroblast subtypes from normal proliferative phase endometrium displayed enriched functions of DNA replication, repair, growth regulation, and metabolic processes. However, endometrioma fibroblasts comprised 4 subtypes: 1) cytokine, inflammatory response; 2) FGF stimulation, immune response; 3) extracellular matrix (ECM) organization, cell adhesion; 4) angiogenesis, hypoxia response, and epithelial-mesenchymal transition (EMT), all of which were enriched in MAPK, TNF, IL-17, TGF-β signaling pathways. Fibroblasts from eutopic endometrium of cases displayed many pathways similar to normal tissue, but some features suggested a “transition state” from normal endometrium to endometrioma, involving TGF-β, MAPK, Rho-Rack, NF-kB, JAK-STAT, and EMT. A developmental trajectory was postulated based on these fibroblast features, with endometrioma fibroblasts derived from endometrial fibroblasts\n",
      "31\n",
      ", much as the epithelial cell population has been proposed, based on shared\n",
      "epithelial\n",
      "somatic mutations in these two tissues\n",
      "54\n",
      ". Thus, endometrioma and paired eutopic endometrium fibroblasts differ, and the single cell data support their key roles in disease pathogenesis.\n",
      "Analysis of immune cells revealed they differed significantly in normal endometrium, endometrioma, and paired eutopic tissue (\n",
      "Figure 8A\n",
      "). Specifically, T cell and uNK cell frequencies were lower, uNK cells were more active, and macrophages (Mφ) were enriched and had features of tissue remodeling versus eutopic endometrium\n",
      "31\n",
      ", similar to data from Fonseca et al\n",
      "35\n",
      ". Thus, in the proliferative phase, fibroblasts and immune cell subpopulations contribute to a pro-inflammatory, angiogenic environment in endometriomas. Moreover cell-to-cell communication/interactome analysis revealed that immune cells stimulate fibroblast adhesion and growth in endometriomas (\n",
      "Figure 8B\n",
      "), underscoring the power of single cell analyses to inform the pathogenesis and pathophysiology of endometriomas and cell-cell communications. Furthermore, recent observations indicated that Fusobacterium\n",
      "56\n",
      "and dysbiosis of gut\n",
      "57\n",
      ",\n",
      "58\n",
      "altered systemic and potentially local immune functions, contributing to the development of endometriosis. Notably, while transcriptomic alterations of specific populations and sub- populations of the endometrium, lesions, and immune cells (local and peripheral) under these conditions awaits further study, antibiotic and nutritional therapies for dysbiosis is emerging as a new paradigm for treating endometriosis development and symptom management.\n",
      "Figure 8. Immune cell frequencies in endometriomas and communication networks of immune cell types with endometrioma fibroblasts.\n",
      "Open in a new tab\n",
      "The network supports interactions between immune and fibroblast cells in ovarian endometrioma pathogenesis and pathophysiology. From Ma et al, 2021\n",
      "31\n",
      ", with permission.\n",
      "Endometriosis peritoneal and ovary lesions of patients on progestins\n",
      "As progestins are a mainstay of medical therapy for endometriosis-related pain\n",
      "12\n",
      ", analysis of lesions and endometrium of subjects taking progestins is highly valuable in understanding their effects on cell types and tissue features. Tan et al\n",
      "33\n",
      "performed scRNAseq on 122,000 cells from 14 individuals, selective cell localization by imaging mass cytometry, and validations in endometrial epithelial organoid cultures. Peritoneal and/or ovarian disease and endometrium from 19 cases (rASRM stage II-IV) included 14 on norethindrone (NETA) + ethinyl estradiol (EthE\n",
      "2\n",
      "), 2 on drosperinone + EthE\n",
      "2\n",
      ", 1 on levonorgestrel + EthE\n",
      "2\n",
      ", 1 on only NETA, 1 menstrual phase, and 8 controls with no endometriosis or inflammatory conditions: 4 on NETA + EthE\n",
      "2\n",
      ", 2 in proliferative phase, 1 on medroxyprogesterone acetate, 1 on norgestromin + EthE\n",
      "2\n",
      ". The peritoneal lesions had similar cell compositions as eutopic endometrium but dysregulated innate immune and vascular components - vastly different from ovarian endometriosis lesions. The authors identified a unique perivascular mural cell specific to the peritoneal lesions with features promoting neo- angiogenesis and immune cell trafficking. They also identified a progenitor-like epithelial subpopulation, different from the SOX9+LGR5+ epithelial subpopulation found by Garcia-Alfonso et al\n",
      "20\n",
      ". Analysis of peritoneal lesion myeloid populations of immunomodulatory Mφ and DCs revealed a coordinated immunotolerant phenotype in the disease microenvironment, promoting immunosurveillance escape and thus conducive to lesion establishment.\n",
      "Overall, the data demonstrate that immune and vascular components of peritoneal endometriosis favor neo-angiogenesis and an immune tolerant niche in the peritoneal cavity, beneficial for lesion establishment and growth. These are remarkable findings given that most subjects were on hormone suppressive therapies for pain [progestin (+ estrogen)] and raises the issue of how efficacious these therapies are on specific cells and the role of estrogenic components in disease management. This valuable single cell atlas of disease from patients largely on hormonal treatments offers the opportunity for comparisons of cell types and features in peritoneal lesions and endometriomas in the setting of other hormonal treatments and/or in various phases of the cycle to interpret efficacies of selective therapies on cell-specific pathways and cell-cell communications.\n",
      "Adenomyosis through single cell glasses\n",
      "Uterine compartments relevant to adenomyosis pathogenesis include the endometrial-myometrial junction (EMJ) [basalis and lower functionalis endometrium + sub-endometrial inner myometrium], all of Mullerian origin and displaying cycling estrogen receptors (ER) and progesterone receptors (PR)\n",
      "59\n",
      "–\n",
      "61\n",
      "(\n",
      "Figure 9\n",
      "). Few molecular and genomic studies of the EMJ in adenomyosis uteri have been conducted, likely as access to basalis and inner myometrium requires hysterectomy versus the functionalis, which can be readily obtained by office biopsy.\n",
      "Figure 9. The endometrial-myometrial junction (EMJ).\n",
      "Open in a new tab\n",
      "Panel A.\n",
      "Histology of human uterine endometrium and myometrium compartments (H&E staining).\n",
      "Panel B\n",
      ": Transvaginal ultrasound midsagittal image of uterine compartments.\n",
      "Panel C.\n",
      "Schematic of uterine compartments. Adapted from Naftlin and Jurkovic, 2009\n",
      "61\n",
      ", with permission.\n",
      "Bulk analyses:\n",
      "Studies to date in adenomyosis patients using immunohistochemistry and qPCR reveal abnormal\n",
      "endometrial\n",
      "functionalis epithelial aromatase expression (thus high E\n",
      "2\n",
      "milieu), elevated stromal fibroblast IL-6, IL-8, IL-17 secretion, TH17-Treg imbalance, altered uNK receptors, and aberrant P\n",
      "4\n",
      "-signaling\n",
      "5\n",
      ",\n",
      "26\n",
      ",\n",
      "62\n",
      "–\n",
      "64\n",
      "– very similar to endometrium of patients with endometriosis except for IL-17 and TH17-Tregulatory cell imbalance. We conducted bulk RNA seq of endometrial\n",
      "functionalis\n",
      "65\n",
      "and a study on separate compartments of EMJ inner myometrium and endometrium from patients with diffuse adenomyosis versus controls in the proliferative phase\n",
      "66\n",
      ". ECM remodeling, inflammation, IL-17 signaling, and mTOR signaling were enriched in endometrium, and inflammation, pain, and neuronal signaling pathways in myometrium, consistent with clinical symptoms of this disorder.\n",
      "Single cell analyses:\n",
      "Two recent publications have analyzed adenomyosis lesions\n",
      "67\n",
      ",\n",
      "68\n",
      ". They are important beginnings to understanding the pathogenesis and pathophysiology of adenomyosis, building upon bulk transcriptomic data, which could be further mined in deconvolution studies to validate new and likely forthcoming additional single cell data.\n",
      "Liu et al\n",
      "67\n",
      "sequenced 42,292 cells derived from a control endometrium of a patient with uterine fibroids and eutopic (functionalis) endometrium and matched ectopic lesion from a patient with adenomyosis (type and menstrual cycle phase undefined). The data revealed seven distinct cell types. Comparison of the data from the lesion with those from eutopic matched endometrium and control endometrium revealed processes involved in progression of normal→endometrial→adenomyosis epithelial phenotypes. These included motility, proliferation, angiogenesis, inflammation, and cancer, consistent with the theory of collective cell migration and invasion into the myometrium of endometrial-derived cells. The novel data set revealed vascular mimicry (important in tumor formation) and epithelial transformation to endothelial cells. It is anticipated that the novel epithelial→endothelial transition phenotype will likely be mined in the future and expanded studies of adenomyosis and perhaps other gynecologic disorders and may serve as a unique cell type for drug targeting.\n",
      "Bulun and colleagues\n",
      "68\n",
      "reported results of single cell analyses of adenomyosis lesions and matched eutopic endometrium and myometrium from 3 subjects, all with diffuse adenomyosis and in the proliferative phase (total of 9 samples and 66,000 cells sequenced). They identified 11 cell types in eutopic endometrium of patients with adenomyosis (\n",
      "Figure 10 A\n",
      "). Fibroblast-like cells comprised a large fraction (36%), and pseudotime analysis revealed they originate from pericyte progenitors that differentiate to ER+ and PR+ endometrial stromal fibroblasts. Notably, the pseudotime trajectory was found to end with ciliated epithelial cells, suggesting mesenchymal- epithelial transition at play in this tissue. In adenomyosis lesions, they identified 13 cell types including 2 (smooth muscle cells and neurons) not found in the endometrium of the same patients (\n",
      "Figure 10 B\n",
      "). The adenomyosis lesions contained even more fibroblast-like cells compared to endometrium (50% versus 36%, respectively). In contrast to matched endometrium, the fibroblast-like cells did not originate from pericyte progenitors and differentiated to ECM-expressing fibroblasts and smooth muscle cells (SMC). Also, myometrial SMC were found to derive from fibroblast-like clusters. Moreover, the WNT inhibitor, secreted frizzled related protein (SFRP) family expressed in fibroblast-like clusters were similar across tissues - endometrium, adenomyosis and myometrium\n",
      "68\n",
      ".\n",
      "Figure 10. Fibroblasts and epithelial cell subtypes differ in endometrium and adenomyosis.\n",
      "Open in a new tab\n",
      "Panel A.\n",
      "UMAP of cell types and fibroblast trajectories in endometrium.\n",
      "B.\n",
      "UMAP of cell types and fibroblast trajectories in adenomyosis.\n",
      "Panel C.\n",
      "Progesterone resistance of epithelial subtypes in adenomyosis.\n",
      "Panel D.\n",
      "WNT signaling pathway network. From Yildiz et al, 2023\n",
      "68\n",
      ", with permission.\n",
      "Regarding the epithelial cells, P\n",
      "4\n",
      "-resistance was found in ciliated and unciliated cells in adenomyosis tissue versus matched eutopic endometrium (\n",
      "Figure 10 C\n",
      ") – this is a key finding regarding treatment efficiencies. Further data analysis revealed that the canonical WNT/SFRP pathway signaling was uniquely activated in adenomyosis unciliated and ciliated epithelium and paracrine signaling fibroblast-like mesenchymal: epithelial interaction networks (\n",
      "Figure 10D\n",
      "), opening new avenues for drug discovery for this disorder. The epithelial→endothelial transition reported by Liu et al\n",
      "67\n",
      "was not detected in this study and may be due to different adenomyosis lesion types, cycle phase, patient age, and other variables. Whether the findings of these two studies involving patients with diffuse adenomyosis also apply to other forms of adenomyosis (\n",
      "Figure 1\n",
      ") is unknown and is an opportunity to investigate cell features wherein novel therapeutics could be derived. Moreover, recent data suggest that the endometrial and vaginal microbiomes differ in patients with versus without adenomyosis\n",
      "69\n",
      ",\n",
      "70\n",
      ", raising the possibility of adjunctive antibiotic therapies for symptomatic patients with disease. Additionally, further study of the local and systemic immune system cells and their subtypes as well as in affected uterine tissues at single cell resolution are warranted.\n",
      "Challenges and opportunities in endometriosis and adenomyosis at the single cell level\n",
      "While single cell RNA sequencing and related innovative technologies and single cell atlases are providing unprecedented insights into human physiology and pathophysiology\n",
      "71\n",
      ",\n",
      "72\n",
      ", there are some unique challenges and opportunities in studying the steroid-hormone responsive human uterine compartments and their associated disorders. All research on human tissues, obtained under approved IRB protocols and written informed consent, requires accurate information about the study cohorts (cases and controls) to minimize confounders and maximize data accuracy and interpretation. Key are well annotated, de-identified patient metadata - e.g., demographics (age, race/ethnicity, BMI), pregnancy history, family history, personal habits, indication for surgical procedure, medical and surgical histories, current hormonal and nonhormonal medications, supplements, co-morbidities, the surgical procedure in which tissue samples were acquired, histologic confirmation that samples contain relevant tissues/cells (e.g., endometrium, myometrium, EMJ, endometriosis, adenomyosis, fibrosis, other), and hormonal status at the time of sampling, documented in a comprehensive database. Acquiring such information and processing samples for sequencing and data management demand teams of experienced study directors, clinical, surgical, laboratory, and pathology colleagues and services, genomics and sequencing cores, laboratory staff, clinical research coordinators, data entry staff, biostatisticians, and computational scientists. Some challenges and opportunities regarding studies specifically on the female reproductive tract are described below.\n",
      "The hormonal status of cases and controls whose tissues are under study is critical in studies on endometrium, endometriosis, and adenomyosis, as these tissues are estrogen- and progesterone-responsive. The context of the steroid hormone milieu (cycle phase, hormonal medications, pre- and post-menopausal status) is essential for data analysis and interpretation. If patients are not using hormonal preparations, then menstrual cycle phase determination must distinguish the estrogen-dominant proliferative phase and the sub-phases of the progesterone-dominant secretory phase\n",
      "44\n",
      ". Many patients, however, with symptomatic endometriosis and/or adenomyosis undergo surgery for pain and/or heavy menstrual bleeding for which combined contraceptive steroids, progestins, gonadotropin releasing hormone analogues, and/or aromatase inhibitors have been administered\n",
      "5\n",
      ",\n",
      "12\n",
      ". While synthetic progestins and progesterone signal through PR, they also elicit unique signaling pathways and transcriptomes in endometrial cells when administered systemically and in\n",
      "in vitro\n",
      "models\n",
      "39\n",
      "–\n",
      "43\n",
      ", which can impact some or most cell features discovered by scRNAseq analysis. It is also important to have hormonally similar controls without disease so hormonal status is not a confounder. Cohort studies restricted to a single menstrual cycle phase avoid hormonal confounders due to similar hormonal environments of the samples\n",
      "in situ\n",
      "at the time of acquisition. Also, precise timing is key, as the endometrial and endometriosis and/or adenomyosis lesion transcriptomes may change within a given cycle phase (e.g., early vs late secretory phase and, strikingly, at the opening of the implantation window in normal patients without disease\n",
      "19\n",
      "). A few scRNAseq studies have focused exclusively on one cycle phase versus another e.g., window of implantation (WOI) mid-secretory phase versus proliferative phase\n",
      "31\n",
      ", or timed menstrual effluent sampling\n",
      "32\n",
      ", which reduces confounding. Notably, studies comparing hormone treatments with no treatment among cases or controls are highly valuable to understand the efficacy of therapies on disease cell pathophysiology and could direct individualized cell-responsive therapeutics in the future.\n",
      "Lesion type, location and co-morbidities are additional key components for context and data interpretation. Age of the lesions is largely unknown, and thus comparing a lesion “type” within the same patient or between patients may be affected by different temporal histories of the lesions. Challenges also include the high prevalence of co-existing common gynecologic disorders (uterine fibroids, uterine polyps) in both cases and controls and defining the control group (no endometriosis or adenomyosis and without or with other gynecologic disorders). Standard operating procedures for biospecimen (tissue, blood, peritoneal fluid, other) collection, processing, storage, and distribution, as developed by Sheldon et al\n",
      "73\n",
      "and the World Endometriosis Research Foundation\n",
      "74\n",
      "–\n",
      "77\n",
      ", choice of sequencing technologies (Chapter 1), and pursuit of multiple computational strategies for comprehensive data analysis and validation of findings are key to success in working with these tissues and looking through single cell glasses.\n",
      "While numbers of cells sequenced enrich the phenotypic features of individual cell types, the numbers of subjects recruited is, by comparison, low, and notable is the limited diversity of the cohorts. There is a preponderance of White subjects recruited for scRNA sequencing of endometriosis and adenomyosis\n",
      "4\n",
      "(\n",
      "Table 1\n",
      "), opening a great opportunity to expand studies to include more diverse populations and comparative analyses of data across ethnicities in the future.\n",
      "Through the looking glass and beyond\n",
      "The analyses of human endometrium and the endometrial disorders - endometriosis and adenomyosis - at single cell resolution, described herein, complement and extend bulk tissue and isolated cell transcriptomics, epigenomics, and metabolomics, as well as blood and peritoneal fluid analyses, and\n",
      "in vitro\n",
      "cellular and tissue organoid/assembloid models\n",
      "1\n",
      "–\n",
      "4\n",
      ",\n",
      "78\n",
      "–\n",
      "82\n",
      ". Single cell transcriptomics is taking the field of “endometromics” to new heights – demonstrating commonalities and differences among cells of the same lineage and in different locations within a given tissue and across tissue types, inferred cell-cell communications, and insights into inflammation as a driver of endometriosis and adenomyosis, as well as the consequences of endometrial epithelial somatic mutations (shared in both disorders) on individual cell gene expression and a unique epithelial cell subtype associated with the risk of ovarian cancer in patients with endometriosis. The recent discovery linking Fusobacterium in ovarian endometriosis in women and antibiotic therapy reducing lesions in a mouse model of endometriosis and peritoneal colonization with Fusobacterium has set the stage for well controlled clinical trials for possible antibiotic therapies to reduce disease burden and for symptom management in patients. Also, the associated dysbiosis of the gut microbiome in mouse models opens opportunities for nutritional interventions in patients that could reduce the disease burden and ameliorate symptoms\n",
      "56\n",
      "–\n",
      "58\n",
      ". These new directions warrant further analysis of the circulating metabolome and single cell analyses of endometriosis in the setting of associated tissue microbiota.\n",
      "So, what will we see beyond the looking glass and into the future? The advances noted above will no doubt serve as launchpads for new and novel biomarker discovery and noninvasive diagnostic development and ultimately prognostic indicators of disease and treatment outcomes\n",
      "83\n",
      ", as well as individualized, patient-tailored therapeutics for symptoms associated with both disorders. Mapping human reproductive tissues and disorders with single cell genomics is certainly coming into its own\n",
      "84\n",
      ", and it is anticipated to benefit from the ever-growing single cell atlases (e.g.,\n",
      "www.humancellatlas.org\n",
      ";\n",
      "www.reproductivecellatlas.org\n",
      ") that are providing immensely rich, publicly available datasets. Indeed, current and future studies on endometriosis and adenomyosis will greatly benefit from the abundant single cell data on fundamental cell types and subtypes in other reproductive\n",
      "and non-reproductive\n",
      "tissues and diseases. This is likely, as cells in different tissues and disorders share some fundamental codes that identify their cell type, and that there are tissue-specific codes and tissue-specialized or altered codes (e.g., in the presence of inflammation or hormonal and other medical therapies)\n",
      "72\n",
      ",\n",
      "83\n",
      ",\n",
      "85\n",
      ". Data integration and comparative analyses of these codes and features across a multitude of endometrial and uterine disorders and their co-morbidities, as well as other tissues and diseases has the possibility to break the silos of discipline-specific study and capture the breadth of genomic features integrated with holistic features of patient metadata to render individualized health care and well-being across the life span and across diverse populations. Moreover, the gut microbiome-brain axis has been recognized as a key regulator of health and disease\n",
      "86\n",
      ", and, recently, dysbioses of the gut\n",
      "87\n",
      "and of the female reproductive tract\n",
      "88\n",
      "have been reported in patients with endometriosis and gastrointestinal (GI) symptoms. Notably, circulating- and lesion-derived estrogens are related to these symptoms and with pelvic pain\n",
      "89\n",
      ". Additionally, dysbiotic gut bacteria (the “estrobolome”) that convert conjugated estrogens to active estrogens and other bioactive metabolites are postulated to communicate with the brain regarding pain perception and GI symptoms and may also stimulate disease\n",
      "87\n",
      ",\n",
      "88\n",
      ",\n",
      "90\n",
      ". Studies on adenomyosis-associated altered microbiota in the endometrium and vagina are more limited compared to endometriosis-associated dysbiosis\n",
      "91\n",
      "–\n",
      "93\n",
      ". However, they are exciting as the paradigm of altered female reproductive tract microbiota in both these estrogen-dependent disorders is rapidly evolving. Identifying cell populations and sub-populations through single cell analyses in these various uterine and gut compartments and understanding mechanisms underlying their role in endometriosis-associated symptoms could open a novel approach to non-hormonal antibiotic and nutritional interventions for symptom relief, in well-designed clinical studies\n",
      "87\n",
      ",\n",
      "94\n",
      ". Data integration across compartments and the circulating metabolome, proteome, and immunome are additional players in the path forward.\n",
      "We have much to do to enable teams and new investigators to join this unprecedented effort in science and medicine beyond the looking glass – as technology progresses, so will biology, and then so will technology, and so will biology\n",
      "95\n",
      ".\n",
      "Acknowledgments\n",
      "Supported by the National Institutes of Health (NIH) Eunice Kennedy Shriver National Institute of Child Health & Human Development (NICHD) Center Grant P01 HD 106414\n",
      "REFERENCES\n",
      "1.\n",
      "Zondervan KT, Becker CM, Koga K, Missmer SA, Taylor RN, Viganò P. Endometriosis. Nat Rev Dis Primers. 2018;4(1):9. doi: 10.1038/s41572-018-0008-5\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "2.\n",
      "Zondervan KT, Becker CM, Missmer SA. Endometriosis. N Engl J Med. 2020;382(13):1244–1256. doi: 10.1056/NEJMra1810764\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "3.\n",
      "Bulun SE, Yilmaz BD, Sison C, et al. \n",
      "Endometriosis. Endocr Rev. 2019;40(4):1048–1079. doi: 10.1210/er.2018-00242\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "4.\n",
      "Giudice LC, Oskotsky TT, Falako S, Opoku-Anane J, Sirota M. Endometriosis in the era of precision medicine and impact on sexual and reproductive health across the lifespan and in diverse populations. FASEB Journal. 2023;37(9). doi: 10.1096/fj.202300907\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "5.\n",
      "Zhai J, Vannuccini S, Petraglia F, Giudice LC. Adenomyosis: Mechanisms and Pathogenesis. Semin Reprod Med. 2020;38(2-03):129–143. doi: 10.1055/s-0040-1716687\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "6.\n",
      "Bulun SE, Yildiz S, Adli M, Wei JJ. Adenomyosis pathogenesis: Insights from next-generation sequencing. Hum Reprod Update. 2021;27(6):1086–1097. doi: 10.1093/humupd/dmab017\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "7.\n",
      "Rossi M, Vannuccini S, Capezzuoli T, et al. \n",
      "Mechanisms and Pathogenesis of Adenomyosis. Curr Obstet Gynecol Rep. 2022;11(2):95–102. doi: 10.1007/s13669-022-00326-7\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "8.\n",
      "Shehzad S\n",
      "Adenomyosis: Current knowledge, Recent Advances and Future Perspective. doi: 10.33425/2639-9342.12223\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "9.\n",
      "Bulun SE, Yildiz S, Adli M, et al. \n",
      "Endometriosis and adenomyosis: shared pathophysiology. Fertil Steril. 2023;119(5):746–750. doi: 10.1016/j.fertnstert.2023.03.006\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "10.\n",
      "Upson K, Missmer SA. Epidemiology of Adenomyosis. Semin Reprod Med. 2020;38(02/03):089–107. doi: 10.1055/s-0040-1718920\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "11.\n",
      "Cope AG, Ainsworth AJ, Stewart EA. Current and Future Medical Therapies for Adenomyosis. Semin Reprod Med. 2020;38(02/03):151–156. doi: 10.1055/s-0040-1719016\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "12.\n",
      "Becker CM, Bokor A, Heikinheimo O, et al. \n",
      "ESHRE guideline: endometriosis. Hum Reprod Open. 2022;2022(2). doi: 10.1093/hropen/hoac009\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "13.\n",
      "Park BY, Yao R, Rossi J, Lee AW. Severe maternal morbidity associated with endometriosis: a population-based, retrospective cohort study. Fertil Steril. 2023;120(2):360–368. doi: 10.1016/j.fertnstert.2023.03.033\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "14.\n",
      "Culley L, Law C, Hudson N, et al. \n",
      "The social and psychological impact of endometriosis on women’s lives: A critical narrative review. Hum Reprod Update. 2013;19(6):625–639. doi: 10.1093/humupd/dmt027\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "15.\n",
      "Aerts L, Grangier L, Streuli I, et al. \n",
      "Psychosocial impact of endometriosis: From co-morbidity to intervention. Best Pract Res Clin Obstet Gynaecol. 2018;50:2–10. doi: 10.1016/j.bpobgyn.2018.01.008\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "16.\n",
      "Sperschneider ML, Hengartner MP, Kohl-Schwartz A, et al. \n",
      "Does endometriosis affect professional life? A matched case-control study in Switzerland, Germany and Austria. BMJ Open. 2019;9(1):e019570. doi: 10.1136/bmjopen-2017-019570\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "17.\n",
      "La Rosa VL, De Franciscis P, Barra F, et al. \n",
      "Quality of life in women with endometriosis: a narrative overview. Minerva Med. 2020;111(1). doi: 10.23736/S0026-4806.19.06298-0\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "18.\n",
      "Nelsen LM, Lenderking WR, Pokrzywinski R, et al. \n",
      "Experience of Symptoms and Disease Impact in Patients with Adenomyosis. The Patient - Patient-Centered Outcomes Research. 2018;11(3):319–328. doi: 10.1007/s40271-017-0284-2\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "19.\n",
      "Wang W, Vilella F, Alama P, et al. \n",
      "Single-cell transcriptomic atlas of the human endometrium during the menstrual cycle. Nat Med. 2020;26(10):1644–1653. doi: 10.1038/s41591-020-1040-z\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "20.\n",
      "Garcia-Alonso L, Handfield LF, Roberts K, et al. \n",
      "Mapping the temporal and spatial dynamics of the human endometrium in vivo and in vitro. Nat Genet. 2021;53(12):1698–1711. doi: 10.1038/s41588-021-00972-2\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "21.\n",
      "Queckbörner S, von Grothusen C, Boggavarapu NR, Francis RM, Davies LC, Gemzell-Danielsson K. Stromal Heterogeneity in the Human Proliferative Endometrium—A Single-Cell RNA Sequencing Study. J Pers Med. 2021;11(6):448. doi: 10.3390/jpm11060448\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "22.\n",
      "Jones RC, Karkanias J, Krasnow MA, et al. \n",
      "The Tabula Sapiens: A multiple-organ, single-cell transcriptomic atlas of humans. Science (1979). 2022;376(6594). doi: 10.1126/science.abl4896\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "23.\n",
      "Stephens VR, Rumph JT, Ameli S, Bruner-Tran KL, Osteen KG. The Potential Relationship Between Environmental Endocrine Disruptor Exposure and the Development of Endometriosis and Adenomyosis. Front Physiol. 2022;12. doi: 10.3389/fphys.2021.807685\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "24.\n",
      "Marshall A, Kommoss KF, Ortmann H, et al. \n",
      "Comparing gene expression in deep infiltrating endometriosis with adenomyosis uteri: evidence for dysregulation of oncogene pathways. Reproductive Biology and Endocrinology. 2023;21(1). doi: 10.1186/s12958-023-01083-9\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "25.\n",
      "Jiang Y, Jiang R, Cheng X, et al. \n",
      "Decreased expression of NR4A nuclear receptors in adenomyosis impairs endometrial decidualization. Mol Hum Reprod. 2016;22(9):655–668. doi: 10.1093/molehr/gaw042\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "26.\n",
      "Peng Y, Jin Z, Liu H, Xu C. Impaired decidualization of human endometrial stromal cells from women with adenomyosis†. Biol Reprod. 2021;104(5):1034–1044. doi: 10.1093/biolre/ioab017\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "27.\n",
      "Taylor HS, Kotlyar AM, Flores VA. Endometriosis is a chronic systemic disease: clinical challenges and novel innovations. The Lancet. 2021;397(10276):839–852. doi: 10.1016/S0140-6736(21)00389-5\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "28.\n",
      "Isaacson K, Loring M. Symptoms of Adenomyosis and Overlapping Diseases. Semin Reprod Med. 2020;38(02/03):144–150. doi: 10.1055/s-0040-1721795\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "29.\n",
      "O’Shea A, Figueiredo G, Lee SI. Imaging Diagnosis of Adenomyosis. Semin Reprod Med. 2020;38(02/03):119–128. doi: 10.1055/s-0040-1719017\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "30.\n",
      "Chen J, Porter AE, Kho KA. Current and Future Surgical and Interventional Management Options for Adenomyosis. Semin Reprod Med. 2020;38(02/03):157–167. doi: 10.1055/s-0040-1718921\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "31.\n",
      "Ma J, Zhang L, Zhan H, et al. \n",
      "Single-cell transcriptomic analysis of endometriosis provides insights into fibroblast fates and immune cell heterogeneity. Cell Biosci. 2021;11(1):125. doi: 10.1186/s13578-021-00637-x\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "32.\n",
      "Shih AJ, Adelson RP, Vashistha H, et al. \n",
      "Single-cell analysis of menstrual endometrial tissues defines phenotypes associated with endometriosis. BMC Med. 2022;20(1):315. doi: 10.1186/s12916-022-02500-3\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "33.\n",
      "Tan Y, Flynn WF, Sivajothi S, et al. \n",
      "Single-cell analysis of endometriosis reveals a coordinated transcriptional programme driving immunotolerance and angiogenesis across eutopic and ectopic tissues. Nat Cell Biol. 2022;24(8):1306–1318. doi: 10.1038/s41556-022-00961-5\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "34.\n",
      "Huang X, Wu L, Pei T, et al. \n",
      "Single-cell transcriptome analysis reveals endometrial immune microenvironment in minimal/mild endometriosis. Clin Exp Immunol. 2023;212(3):285–295. doi: 10.1093/cei/uxad029\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "35.\n",
      "Fonseca MAS, Haro M, Wright KN, et al. \n",
      "Single-cell transcriptomic analysis of endometriosis. Nat Genet. 2023;55(2):255–267. doi: 10.1038/s41588-022-01254-1\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "36.\n",
      "Almonte AL, Wang W, Flynn E, et al. \n",
      "ENDOMETRIAL SUBCELLULAR ALTERATIONS IN ENDOMETRIOSIS IDENTIFIED BY SINGLE-CELL ANALYSIS ACROSS THE MENSTRUAL CYCLE. Fertil Steril. 2023;120(4):e61–e62. doi: 10.1016/j.fertnstert.2023.08.204\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "37.\n",
      "Ponnampalam AP, Weston GC, Trajstman AC, Susil B, Rogers PAW. Molecular classification of human endometrial cycle stages by transcriptional profiling. Mol Hum Reprod. 2004;10(12):879–893. doi: 10.1093/molehr/gah121\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "38.\n",
      "Talbi S, Hamilton AE, Vo KC, et al. \n",
      "Molecular Phenotyping of Human Endometrium Distinguishes Menstrual Cycle Phases and Underlying Biological Processes in Normo-Ovulatory Women. Endocrinology. 2006;147(3):1097–1121. doi: 10.1210/en.2005-1076\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "39.\n",
      "Goldfien GA, Barragan F, Chen J, et al. \n",
      "Progestin-Containing Contraceptives Alter Expression of Host Defense-Related Genes of the Endometrium and Cervix. Reproductive Sciences. 2015;22(7):814–828. doi: 10.1177/1933719114565035\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "40.\n",
      "Shanmugasundaram U, Hilton JF, Critchfield JW, et al. \n",
      "Effects of the levonorgestrel-releasing intrauterine device on the immune microenvironment of the human cervix and endometrium. American Journal of Reproductive Immunology. 2016;76(2):137–148. doi: 10.1111/aji.12535\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "41.\n",
      "Smith-McCune KK, Hilton JF, Shanmugasundaram U, et al. \n",
      "Effects of depot-medroxyprogesterone acetate on the immune microenvironment of the human cervix and endometrium: implications for HIV susceptibility. Mucosal Immunol. 2017;10(5):1270–1278. doi: 10.1038/mi.2016.121\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "42.\n",
      "Smith-McCune K, Thomas R, Averbach S, et al. \n",
      "Differential Effects of the Hormonal and Copper Intrauterine Device on the Endometrial Transcriptome. Sci Rep. 2020;10(1):6888. doi: 10.1038/s41598-020-63798-8\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "43.\n",
      "Houshdaran S, Chen JC, Vallvé-Juanico J, et al. \n",
      "Progestins Related to Progesterone and Testosterone Elicit Divergent Human Endometrial Transcriptomes and Biofunctions. Int J Mol Sci. 2020;21(7):2625. doi: 10.3390/ijms21072625\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "44.\n",
      "Teh WT, Chung J, Holdsworth-Carson SJ, et al. \n",
      "A molecular staging model for accurately dating the endometrial biopsy. Nat Commun. 2023;14(1). doi: 10.1038/s41467-023-41979-z\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "45.\n",
      "Revised American Fertility Society Classification of Endometriosis: 1985. Fertil Steril. 1985;43(3):351–352. doi: 10.1016/S0015-0282(16)48430-X\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "46.\n",
      "Miravet-Valenciano J, Ruiz-Alonso M, Simón C. Transcriptomics of the human endometrium and embryo implantation. In: Human Reproductive and Prenatal Genetics. Elsevier; 2018:271–291. doi: 10.1016/B978-0-12-813570-9.00012-7\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "47.\n",
      "Barragan F, Irwin JC, Balayan S, et al. \n",
      "Human Endometrial Fibroblasts Derived from Mesenchymal Progenitors Inherit Progesterone Resistance and Acquire an Inflammatory Phenotype in the Endometrial Niche in Endometriosis. Biol Reprod. 2016;94(5). doi: 10.1095/biolreprod.115.136010\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "48.\n",
      "Wang X, Hawkins SM. Using advanced spatial and single-cell transcriptomics to characterize the human endometrium. Nat Genet. 2021;53(12):1628–1630. doi: 10.1038/s41588-021-00982-0\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "49.\n",
      "Tempest N, Maclean A, Hapangama DK. Endometrial Stem Cell Markers: Current Concepts and Unresolved Questions. Int J Mol Sci. 2018;19(10):3240. doi: 10.3390/ijms19103240\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "50.\n",
      "Cousins FL, Pandoy R, Jin S, Gargett CE. The Elusive Endometrial Epithelial Stem/Progenitor Cells. Front Cell Dev Biol. 2021;9. doi: 10.3389/fcell.2021.640319\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "51.\n",
      "Riccio LGC, Baracat EC, Chapron C, Batteux F, Abrão MS. The role of the B lymphocytes in endometriosis: A systematic review. J Reprod Immunol. 2017;123:29–34. doi: 10.1016/j.jri.2017.09.001\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "52.\n",
      "Vallvé-Juanico J, Houshdaran S, Giudice LC. The endometrial immune environment of women with endometriosis. Hum Reprod Update. 2019;25(5):565–592. doi: 10.1093/humupd/dmz018\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "53.\n",
      "Shen M, O’Donnell E, Leon G, et al. \n",
      "The role of endometrial B cells in normal endometrium and benign female reproductive pathologies: a systematic review. Hum Reprod Open. 2022;2022(1). doi: 10.1093/hropen/hoab043\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "54.\n",
      "Anglesio MS, Papadopoulos N, Ayhan A, et al. \n",
      "Cancer-Associated Mutations in Endometriosis without Cancer. New England Journal of Medicine. 2017;376(19):1835–1848. doi: 10.1056/nejmoa1614814\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "55.\n",
      "Yong PJ, Talhouk A, Anglesio MS. Somatic Genomic Events in Endometriosis: Review of the Literature and Approach to Phenotyping. Reproductive Sciences. 2021;28(10):2743–2757. doi: 10.1007/s43032-020-00451-9\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "56.\n",
      "Muraoka A, Suzuki M, Hamaguchi T, et al. \n",
      "Fusobacterium Infection Facilitates the Development of Endometriosis through the Phenotypic Transition of Endometrial Fibroblasts.; 2023.\n",
      "https://www.science.org\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "57.\n",
      "Chadchan SB, Naik SK, Popli P, et al. \n",
      "Gut microbiota and microbiota-derived metabolites promotes endometriosis. Cell Death Discov. 2023;9(1). doi: 10.1038/s41420-023-01309-0\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "58.\n",
      "Lucarini E, Di Pilato V, Parisio C, et al. \n",
      "Visceral sensitivity modulation by faecal microbiota transplantation: The active role of gut bacteria in pain persistence. Pain. 2022;163(5):861–877. doi: 10.1097/j.pain.0000000000002438\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "59.\n",
      "Brosens JJ, de Souza NM, Barker FG. Uterine junctional zone: function and disease. The Lancet. 1995;346(8974):558–560. doi: 10.1016/S0140-6736(95)91387-4\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "60.\n",
      "Noe M, Kunz G, Herbertz M, Mall G, Leyendecker G. The Cyclic Pattern of the Immunocytochemical Expression of Oestrogen and Progesterone Receptors in Human Myometrial and Endometrial Layers: Characterization of the Endometrial-Subendometrial Unit. Vol 14.; 1999.\n",
      "https://academic.oup.com/humrep/article/14/1/190/588131\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "61.\n",
      "Naftalin J, Jurkovic D. The endometrial–myometrial junction: a fresh look at a busy crossing. Ultrasound in Obstetrics & Gynecology. 2009;34(1):1–11. doi: 10.1002/uog.6432\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "62.\n",
      "Bourdon M, Santulli P, Jeljeli M, et al. \n",
      "Immunological changes associated with adenomyosis: A systematic review. Hum Reprod Update. 2021;27(1):108–129. doi: 10.1093/humupd/dmaa038\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "63.\n",
      "Gui T, Chen C, Zhang Z, et al. \n",
      "The disturbance of TH17-Treg cell balance in adenomyosis. Fertil Steril. 2014;101(2):506–514. doi: 10.1016/j.fertnstert.2013.10.050\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "64.\n",
      "Yang JH. Decreased expression of killer cell inhibitory receptors on natural killer cells in eutopic endometrium in women with adenomyosis. Human Reproduction. 2004;19(9):1974–1978. doi: 10.1093/humrep/deh372\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "65.\n",
      "Herndon CN, Aghajanova L, Balayan S, et al. \n",
      "Global Transcriptome Abnormalities of the Eutopic Endometrium From Women With Adenomyosis. Reproductive Sciences. 2016;23(10):1289–1303. doi: 10.1177/1933719116650758\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "66.\n",
      "Zhai J, Li S, Sen S, et al. \n",
      "Transcriptomic analysis supports collective endometrial cell migration in the pathogenesis of adenomyosis. Reprod Biomed Online. 2022;45(3):519–530. doi: 10.1016/j.rbmo.2022.05.007\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "67.\n",
      "Liu Z, Sun Z, Liu H, et al. \n",
      "Single-cell transcriptomic analysis of eutopic endometrium and ectopic lesions of adenomyosis. Cell Biosci. 2021;11(1):51. doi: 10.1186/s13578-021-00562-z\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "68.\n",
      "Yildiz S, Kinali M, Wei JJ, et al. \n",
      "Adenomyosis: single-cell transcriptomic analysis reveals a paracrine mesenchymal–epithelial interaction involving the WNT/SFRP pathway. Fertil Steril. 2023;119(5):869–882. doi: 10.1016/j.fertnstert.2023.01.041\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "69.\n",
      "Kunaseth J, Waiyaput W, Chanchaem P, et al. \n",
      "Vaginal microbiome of women with adenomyosis: A case-control study. PLoS One. 2022;17(2\n",
      "February). doi: 10.1371/journal.pone.0263283\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "70.\n",
      "Lin Q, Duan H, Wang S, et al. \n",
      "Endometrial microbiota in women with and without adenomyosis: A pilot study. Front Microbiol. 2023;14. doi: 10.3389/fmicb.2023.1075900\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "71.\n",
      "Baysoy A, Bai Z, Satija R, Fan R. The technological landscape and applications of single-cell multi-omics. Nat Rev Mol Cell Biol. 2023;24(10):695–713. doi: 10.1038/s41580-023-00615-w\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "72.\n",
      "Elmentaite R, Domínguez Conde C, Yang L, Teichmann SA. Single-cell atlases: shared and tissue-specific cell types across human organs. Nat Rev Genet. 2022;23(7):395–410. doi: 10.1038/s41576-022-00449-w\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "73.\n",
      "Sheldon E, Vo KC, McIntire RA, et al. \n",
      "Biobanking human endometrial tissue and blood specimens: standard operating procedure and importance to reproductive biology research and diagnostic development. Fertil Steril. 2011;95(6):2120–2122.e12. doi: 10.1016/j.fertnstert.2011.01.164\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "74.\n",
      "Becker CM, Laufer MR, Stratton P, et al. \n",
      "World Endometriosis Research Foundation Endometriosis Phenome and Biobanking Harmonisation Project: I. Surgical phenotype data collection in endometriosis research. Fertil Steril. 2014;102(5):1213–1222. doi: 10.1016/j.fertnstert.2014.07.709\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "75.\n",
      "Vitonis AF, Vincent K, Rahmioglu N, et al. \n",
      "World Endometriosis Research Foundation Endometriosis Phenome and biobanking harmonization project: II. Clinical and covariate phenotype data collection in endometriosis research. Fertil Steril. 2014;102(5):1223–1232. doi: 10.1016/j.fertnstert.2014.07.1244\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "76.\n",
      "Rahmioglu N, Fassbender A, Vitonis AF, et al. \n",
      "World Endometriosis Research Foundation Endometriosis Phenome and Biobanking Harmonization Project: III. Fluid biospecimen collection, processing, and storage in endometriosis research. Fertil Steril. 2014;102(5):1233–1243. doi: 10.1016/j.fertnstert.2014.07.1208\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "77.\n",
      "Fassbender A, Rahmioglu N, Vitonis AF, et al. \n",
      "World Endometriosis Research Foundation Endometriosis Phenome and Biobanking Harmonisation Project: IV. Tissue collection, processing, and storage in endometriosis research. Fertil Steril. 2014;102(5):1244–1253. doi: 10.1016/j.fertnstert.2014.07.1209\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "78.\n",
      "Boretto M, Maenhoudt N, Luo X, et al. \n",
      "Patient-derived organoids from endometrial disease capture clinical heterogeneity and are amenable to drug screening. Nat Cell Biol. 2019;21(8):1041–1051. doi: 10.1038/s41556-019-0360-z\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "79.\n",
      "Rawlings TM, Makwana K, Taylor DM, et al. \n",
      "Modelling the impact of decidual senescence on embryo implantation in human endometrial assembloids. Elife. 2021;10. doi: 10.7554/eLife.69603\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "80.\n",
      "Lou L, Kong S, Sun Y, Zhang Z, Wang H. Human Endometrial Organoids: Recent Research Progress and Potential Applications. Front Cell Dev Biol. 2022;10. doi: 10.3389/fcell.2022.844623\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "81.\n",
      "Juárez-barber E, Francés-herrero E, Corachán A, et al. \n",
      "Establishment of Adenomyosis Organoids as a Preclinical Model to Study Infertility. J Pers Med. 2022;12(2). doi: 10.3390/jpm12020219\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "82.\n",
      "Monnin N, Fattet AJ, Koscinski I. Endometriosis: Update of Pathophysiology, (Epi) Genetic and Environmental Involvement. Biomedicines. 2023;11(3):978. doi: 10.3390/biomedicines11030978\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "83.\n",
      "Hu W, Zhang Y, Mei J, Fang X. Spatial transcriptomics in human biomedical research and clinical application. Current Medicine. 2023;2(1). doi: 10.1007/s44194-023-00023-4\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "84.\n",
      "Marečková M, Massalha H, Lorenzi V, Vento-Tormo R. Mapping Human Reproduction with Single-Cell Genomics. Published online 2022. doi: 10.1146/annurev-genom-120121\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "85.\n",
      "Buechler MB, Pradhan RN, Krishnamurty AT, et al. \n",
      "Cross-tissue organization of the fibroblast lineage. Nature. 2021;593(7860):575–579. doi: 10.1038/s41586-021-03549-5\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "86.\n",
      "Cryan JF, O KJ, M Cowan CS, et al. \n",
      "The Microbiota-Gut-Brain Axis. Physiol Rev. 2019;99:1877–2013. doi: 10.1152/physrev.00018.2018.-The\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "87.\n",
      "Salliss ME, Farland LV., Mahnert ND, Herbst-Kralovetz MM. The role of gut and genital microbiota and the estrobolome in endometriosis, infertility and chronic pelvic pain. Hum Reprod Update. 2022;28(1):92–131. doi: 10.1093/humupd/dmab035\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "88.\n",
      "Uzuner C, Mak J, El-Assaad F, Condous G. The bidirectional relationship between endometriosis and microbiome. Front Endocrinol (Lausanne). 2023;14. doi: 10.3389/fendo.2023.1110824\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "89.\n",
      "Greaves E, Temp J, Esnal-Zufiurre A, Mechsner S, Horne AW, Saunders PTK. Estradiol Is a Critical Mediator of Macrophage-Nerve Cross Talk in Peritoneal Endometriosis. American Journal of Pathology. 2015;185(8):2286–2297. doi: 10.1016/j.ajpath.2015.04.012\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "90.\n",
      "Pai AHY, Wang YW, Lu PC, Wu HM, Xu JL, Huang HY. Gut Microbiome–Estrobolome Profile in Reproductive-Age Women with Endometriosis. Int J Mol Sci. 2023;24(22). doi: 10.3390/ijms242216301\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "91.\n",
      "Chen S, Gu Z, Zhang W, et al. \n",
      "The study of endometriosis and adenomyosis related microbiota in female lower genital tract in Northern Chinese population. Gynecology and Obstetrics Clinical Medicine. 2021;1(3):119–129. doi: 10.1016/j.gocm.2021.07.007\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "92.\n",
      "Lin Q, Duan H, Wang S, et al. \n",
      "Endometrial microbiota in women with and without adenomyosis: A pilot study. Front Microbiol. 2023;14. doi: 10.3389/fmicb.2023.1075900\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "93.\n",
      "Valdés-Bango M, Gracia M, Rubio E, et al. \n",
      "Comparative analysis of endometrial, vaginal, and gut microbiota in patients with and without adenomyosis. Acta Obstet Gynecol Scand. 2024;103(7):1271–1282. doi: 10.1111/aogs.14847\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PMC free article\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "94.\n",
      "Griffiths MJ, Horne AW, Gibson DA, Roberts N, Saunders PTK. Endometriosis: recent advances that could accelerate diagnosis and improve care. Trends Mol Med. Published online 2024. doi: 10.1016/j.molmed.2024.06.008\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n",
      "95.\n",
      "The Cell editorial team. The integral partnership of biology and technology. Cell. 2022;185(15):2609–2610. doi: 10.1016/j.cell.2022.06.053\n",
      "[\n",
      "DOI\n",
      "] [\n",
      "PubMed\n",
      "] [\n",
      "Google Scholar\n",
      "]\n"
     ]
    }
   ],
   "source": [
    "soup = BeautifulSoup(response.text, \"html.parser\")\n",
    "print(soup.find(\"article\").get_text(separator=\"\\n\", strip=True))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6448c1bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "prueba = chain.invoke({\"query\": soup.find(\"article\").get_text(separator=\"\\n\", strip=True)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "85f263d4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'PaperTitle': 'Endometriosis and Adenomyosis Unveiled Through Single-cell Glasses',\n",
       " 'Publication': '2025',\n",
       " 'Authors': 'Linda C Giudice, Binya Liu, Juan C Irwin',\n",
       " 'Email': 'Linda.Giudice@ucsf.edu',\n",
       " 'Abstract': 'Single cell technologies are expanding our understanding of endometriosis and adenomyosis, sister disorders of the uterine endometrium that contain similar complements of lesion cell types but located in different niches – outside and inside the endometrium, respectively. Both diseases cause significant morbidity and impaired quality of life among those affected, and current therapies mitigate most symptoms although with highly variable efficacy, duration of effect, and frequent intolerable side effects. Thus, there is a pressing need for transformative approaches to develop individualized therapies for the variety of presentations of endometriosis and adenomyosis symptoms and heterogeneity of lesion types histologically and architecturally. Single cell technologies are transforming understanding human physiology and pathophysiology in the reproductive system and beyond. This manuscript reviews endometriosis and adenomyosis clinical characteristics and the recent studies focused on eutopic endometrium and ectopic lesions at single cell resolution, the myriad of cell types and subtypes, cell-cell communications, signaling pathways, applications for novel drug discovery and therapeutic approaches, and challenges and opportunities that accompany this type of research.',\n",
       " 'Introduction': 'Endometriosis and adenomyosis are common estrogen-driven uterine disorders derived mainly from the basalis endometrium, characterized by “ectopic” endometrial-like cells outside their normal location. A key feature distinguishing them is the location of the ectopic lesions. For endometriosis they are outside the uterus - mostly in the pelvis and invading pelvic organs 1 – 4 , and for adenomyosis they are within the uterine myometrium 5 – 8 . Endometriosis and adenomyosis disease lesions comprise similar phenotypic endometrial epithelium and mesenchymal cell types and display characteristics of inflammation, neoneuroangiogenesis, and fibrosis 1 – 9 . However, whether their cellular features are equivalent in different anatomic niches is unknown and is of importance for disease stratification in view of the vastly heterogeneous lesion architectural structures and for developing tailored, personalized, patient-specific treatments and prognostic disease outcome indicators.\\nThese complex “sister” conditions often occur concomitantly 10 and result in similar symptoms of pain and uterine bleeding treated medically (mostly hormonal) with variable efficacies and are associated with infertility and similar poor reproductive outcomes 1 , 4 – 13 . They also greatly impact quality of life of those affected 14 – 18 . Given the devastating impact of these disorders and limited success of effective therapies and disease prevention and cure, a transformative approach to disease phenotyping and novel, personalized therapeutics is essential to improve the lives of millions of patients with endometriosis and adenomyosis across the globe.\\nSingle cell technologies and precision medicine approaches are early in their application to these disorders and are beginning to provide insights into commonalities/differences of cell types in endometriosis, adenomyosis, and eutopic endometrium, and the role of their unique anatomical niches influencing cell behavior, disease establishment and pathophysiology. Moreover, single cell data and associated signaling pathways are anticipated to lead to novel drug candidate discovery to control symptoms, minimize disease burden resulting ultimately in cure, identifying diagnostic biomarkers, and informing disease classification and prognostics for disease courses. The time is now to move the needle on both disorders through single cell “glasses”!\\nRegarding endometrium, an atlas of normal, cycling endometrium has been derived at single cell resolution by several groups 19 – 22 (also see Chapter 3). These studies underscore heterogeneity of specific cell types and altered features in different hormonal states across the menstrual cycle and provide a key backdrop to studies at the single cell level of endometrium of patients with endometriosis and adenomyosis, endometriosis lesions, and adenomyosis lesions. Herein, we begin with an overview of clinical characteristics of endometriosis and adenomyosis (sister disorders but not identical twins), followed by a summary of current advances in scRNA sequencing data of endometriosis and adenomyosis disease lesions, corresponding eutopic endometrium, control endometrium, and disease-specific and common cell features across tissues. We conclude with a review of challenges of conducting research on human tissues, especially those that are steroid hormone responsive, and how these data inform looking to the future through single cell glasses.\\nEndometriosis and adenomyosis – sister disorders, not identical twins\\nEndometriosis and adenomyosis lesions both comprise uterine endometrial cells found outside their normal location ( Figure 1 ). For endometriosis, these cells form lesions - mostly in the pelvis - arriving mainly by retrograde menstruation and causing a profound inflammatory response 1 – 4 . Risks of developing endometriosis are ~ 50% genetic and ~50% environmental 2 , 4 . In contrast, adenomyosis-associated endometrial cells (epithelium and stroma) are found within the uterus – in the uterine myometrium – where they induce a local inflammatory response, smooth muscle hypertrophy and often an enlarged, globular uterus 5 – 8 . Risks of developing adenomyosis are multi-parity, prior uterine surgery, adult exposure to tamoxifen, and in utero exposures to estrogens 5 , 7 , 23 . Endometriosis has been more extensively studied due to ease of obtaining endometrial tissue and sampling disease lesions at surgery. Adenomyosis has had far less molecular, genetic, and genomic analyses, as uterine myometrial lesions and the endometrial-myometrial junction are accessible mainly from hysterectomy specimens, although the eutopic endometrium proper has been studied in adenomyosis patients due to ease of access by office biopsy.\\nFigure 1. Open in a new tab Comparison of endometriosis and adenomyosis pathogenesis, risk factors, diagnostic approaches, pathophysiology, symptoms, and therapies. Adapted from Bulun et al, 2023 9 , with permission.\\nDespite their differences in anatomic location, the two disorders share common origins, some molecular characteristics, symptoms, and treatment strategies 5 – 9 , 12 ( Figure 1 ). Both are diseases of the basalis endometrium with collective cell migration out of the uterus into the pelvis or into the muscle that may also involve epithelial to mesenchymal cell transition 5 , 6 , 9 . They share postulated origins of Mullerian remnant transformation and differentiation of endometrial stem/progenitor cells, and some common epithelial cancer driver mutations although they rarely transform into cancers 6 , 9 . They are estrogen-dependent, progesterone (P 4 )-resistant disorders, display increased estradiol (E 2 ) production in disease (due to aberrant aromatase expression in the epithelium and some stromal cells) and decreased E 2 metabolism and conversion to testosterone 2 , 4 , 6 , 9 , 23 , 25 , 26 . Both are also inflammatory disorders and have the activated inflammasome as the final common pathway for local and systemic inflammation 2 , 3 , 27 . And both can be induced in the developing fetus and/or adult (in animal models) and are associated in humans with endocrine disrupting chemicals found in the environment and personal care products 23 .\\nEndometriosis and adenomyosis share similar clinical features ( Figure 1 ) including dysmenorrhea, chronic pelvic pain, infertility, and poor pregnancy outcomes, and they often co-exist in the same patient and relapse post fertility-preserving surgery 5 – 7 , 9 , 12 , 28 . Symptom histories are key in diagnosing both disorders, but many symptoms overlap between them and with other disorders 2 , 4 . Standards for endometriosis and adenomyosis diagnosis are histologic confirmation of endometrial-like cells in lesions, and pelvic MRI and transvaginal ultrasound are useful for detecting endometriomas and deep disease as well as adenomyosis-associated widened endometrial-myometrial junction and adenomyomas 4 , 29 .\\nTreatment paradigms are similar for both disorders and include combined estrogen/ progestin contraceptive steroids, progestins alone, progestin-containing intrauterine systems, selective progesterone receptor modulators (SPRMs), selective estrogen receptor modulators (SERMs), GnRH analogs, aromatase inhibitors accompanied by ovulation suppression, and dopamine agonists, and medically assisted and advanced reproductive technologies for infertility 1 , 4 , 8 , 12 . Overall, medical therapies for pain have limited efficacy and/or intolerability for many patients. Endometriosis lesion removal or hysterectomy is largely effective for symptom management, although ~ 50% of patients recur within 5 years after conservative surgery and ~15% after hysterectomy 1 , 4 , 12 . Adenomyosis, on the other hand, can be excised but this is rarely performed due to technical challenges, and while hysterectomy is curative, it compromises fertility pursuits 30 .\\nThus, developing non-invasive diagnostics as stand-alone assessments or as adjuncts to imaging technologies, and innovative and personalized therapies are major goals for both disorders.',\n",
       " 'Methods': '',\n",
       " 'Results': 'Endometriosis through single cell glasses Subsequent to the single cell cartography of human endometrium published by Wang et al 19 (see Chapter 3), several groups have recently reported eutopic endometrial scRNAseq data of patients with endometriosis, endometriosis lesions and control eutopic endometrium 20 , 31 – 36 ( Table 1 ). Types of samples, inclusion/exclusion criteria, choice of controls, study designs, numbers of cells sequenced, and number of reads (i.e., base pairs sequenced) are key to data integrity and interpretation and to comparative analyses across tissues and studies. They have provided insights into origins of ectopic disease, the heterogeneity of cell types/subtypes, unique clusters and signatures of endometrium and ectopic lesions informing mechanisms and pathways involved in cellular dysfunctions relevant to pain and fertility compromise, novel cell-cell communications, relationships between lesions and eutopic tissue/cells, and biomarker discovery. Across these studies, samples were obtained in different hormonal milieu (menstrual cycle phase, exogenous hormones), and between 55,000 to 378,000 cells were sequenced ( Table 1 ). As cycle phase and exogenous hormone use are key drivers of endometrial gene expression 37 – 44 , comparison of data across studies can be challenging and mixed hormonal milieu are sometimes experienced within the same study in both cases and/or controls. Below we summarize results of studies focused exclusively on eutopic endometrium of patients with endometriosis relevant to endometrial function and dysfunction and biomarker discovery, followed by studies focused on ovarian endometriomas, superficial, and/or deep infiltrating disease, and eutopic endometrium, relevant to disease heterogeneity, pathophysiology, pathogenesis, and in the setting of progestins – commonly prescribed for symptom management of patients with endometriosis and dysmenorrhea and chronic pelvic pain.\\nTable 1. Highlights of recent endometriosis-related scRNAseq studies.\\nReference Tissue Analyzed Total cells sequenced Technology/Platform Total subjects Endometriosis Type and ASRM Stage Controls Race/ Ethnicity Hormones/IUD Cycle Phase Endometrial histology Ma 2021 endometrium, ovarian endometrioma 55,000 scRNAseq (10X) n=6 subjects (n=3 cases, n=3 controls) ASRM Stage III, IV ovarian endometriomas Healthy controls without endometriosis N/A None All in prolifeative phase Garcia-Alonso 2022 endometrium (functionalis, full thickness), endometriosis peritoneal lesions (red, white, black) 98,569 sequenced scRNAseq, snRNA seq, spatial profilng. Lesion microarray data ( GSE141549 ) n=3 functionalis n=6 full thickness; Microarray data controls: endom n=42, peritoneum n=12; n=9 red, 9 white, 11 black Peritoneal disease: red, white, black Healthy controls (functionalis layer) n=6 full thickness without reproductive disorders; normal peritoneum N/A None Proliferative, secretory; Microarray samples: Control Endo 17PE, 25SE; Perit 4PE, 8SE; Lesions: Red 2PE, 7 SE; White 5PE, 4SE; Black 6PE, 5 SE Shih 2022 menstrual endometrium 43,054 scRNAseq (10X) n=33 subjects (n=11 Dx, n=13 sx, n=9 controls) N/A; another group with Sx but no Dx. No endometriosis diagnosis. ? Other GYN disorders White: 10 Dx, 13 sx, 7 controls Black: 0 Dx, 0 sx, 1 control Hispanic: 0 Dx, 0 sx, 0 cont Mixed: 0 Dx,0 sx, 1 control Other: 1 Dx,0 sx, 0 control None on hormones, except 1 case used vaginal P4. Re-analysis showed no impact on results Menstrual (heaviest flow, mostly CD 1 or 2) Tan 2022 endometrium, enodmetriosis lesions 122,000 scRNAseq (10X) IMC organoids n=27 subjects (n=19 cases, n=8 controls; n=14 sequenced ASRM Stage II-IV, peritoneal lesions, ovarian “lesions”, organoids No endometriosis or infammatory conditions White: 9 cases, 3 controls Asian: 4 cases, 2 controls Hisp: 5 cases, 3 controls Black: 1 case, 0 controls Progestin [NETA, LVN oral, IUD, drosperinone, norelgestromin] +/− ethinyl E2, medroxy-progesterone acetate, copperIUD wkly PE: 3 cases, 2 control; inactive 3 cases, 0 control; mens: 1 case, 0 control, exog hormone effect: 7 cases, 7 control PE: 0 cases, 2 controls IE: 0 cases, 1 control ESE: 4 cases, 0 control N/A: 1 case, 1 control Fonseca 2023 endometrium, endometriosis lesions, unaffected ovary and peritoneum 373,851 digital scRNAseq n=21 subjects n=17 cases, n=4 controls n-54 specimens collected Cases: n=17 Endometrium Endometrioma Superficial and deep disesase n=9 w/o GYN disorders; n=8 w/ adenomyosis, uterine fibroids +/− polyp Controls n=4 n=3 PMP, 1 peri-menopause All no evidence of endometriosis, all w/ leiomyoma +/− adenomyosis +/− uterine polyp White: 13 cases,2 controls Black: 2 cases, 0 controls Asian: 1 case, 0 control Other: 1 case, 0 control Cases : n=14 on no hormones. n=1 w/ vaginal ring E+P; n=1 on E/T, P4; n=1 on E2+P4. Controls : 3 of 4 on E +/− P4; NETA; n=1 peri-menopause in luteal phase Cases : Proliferative phase: n=7; secretory phase: n=9; N/A (on hormones) n=3. Controls : Proliferative n=0, Secretory n=1, N/A (on hormones) n=3 Huang 2023 endometrium 128,243 scRNAseq (10X) n=10 sequenced n=6 cases, n=4 controls ASRM Stage I,II No endometriosis, benign ovarian cysts N/A None EPE: 3 cases, 3 controls MSE: 3 cases, 3 controls LSE: 1 control Almonte 2023 endometrium >250,000 scRNseq (10X) n=29 cases, 7 controls w/o disease and 8 controls with fibroids ASRM stage I-IV No endometriosis or other GYN d/o; No endo but + uterine fibroids N/A None Proliferative:12 cases, 5 controls. Secretory: 14 cases, 3 controls. Open in a new tab Abbreviations: CD=cycle day; dx=diagnosis; E=estradiol; ESE=early secretory phase; exog=exogenous; IE=interval endometrium; IMC=imaging mass cytometry; IUD=intrauterine device; LSE=late secretory phase; LVN=levonorgestrel; MSE=mid-secretory phase; N/A=not available; NETA=norethindrone acetate; P=progestin; P=progesterone; PE=proliferative phase; SE=secretory phase; sx=symptoms, w/= with; +/− = with and/or without. Adapted from Giudice et al 4 , with permission. Eutopic endometrium Window of implantation epithelial abnormalities in infertile endometriosis patients Endometrial biopsies of cases with stage I/II disease (revised American Society for Reproductive Medicine (rASRM) scoring system) 45 and infertility (n=3 WOI, n=3 early proliferative phase) versus n=7 fertile controls without disease (3 WOI, 3 proliferative, 1 late secretory phase) were studied by Huang et al 34 . A total of 128,243 cells were sequenced, and 30 type/cell type clusters were identified. Notably, a unique epithelial cell cluster in WOI endometrium of endometriosis patients was identified that lacked expression of two key genes associated with implantation: PAEP (progesterone-associated endometrial protein) and CXCL14 (chemokine (CXC motif) ligand 14) ( Figure 2 ). PAEP, also known as glycodelin A and placental protein 14, is involved in immune response regulation, promoting opening of the WOI by lowering maternal immune responses to an implanting embryo 46 . CXCL14 is involved in activating the innate immune response stimulating chemotaxis of uterine natural killer cells (uNK) and other monocytes to epithelial glands for implantation 46 . Other findings included absence of normal variation of endometrial uNK and T cell numbers across the cycle, and lower anti-inflammatory IL-10 and more pro-inflammatory cytokines expressed in endometrial immune cells in the WOI versus in the proliferative phase in cases and the opposite in controls. Notably, 11 ligand–receptor pairs were upregulated between endometrial immune and epithelial cells during the WOI of cases. Figure 2. Absence of key genes that open the window of implantation in eutopic endometrial epithelial cluster in infertile patients with Stage I/II endometriosis. Open in a new tab Panel A . t-SNE projection of endometrial cells colored based on their sample origin and cycle phase. Panel B . Expression patterns of PAEP and CXCL14 projected on t-distributed stochastic neighbour embedding (t-SNE) plot. From Huang et al, 2023 34 , with permission. Thus, lower epithelial receptivity markers, abnormal immune cell frequencies, a pro- inflammatory WOI, and new insights into cell-cell communications between epithelium and immune cells support an important role of the immune system and an adverse environment for embryo implantation for fertility and potentially for poor pregnancy outcomes in patients with stage I/II endometriosis who do achieve pregnancy versus fertile controls. Endometrial subcellular alterations across the cycle in endometriosis patients In a recent study, our group sequenced eutopic endometrium (237,620 cells) from 29 subjects with endometriosis (stages I/II and III/IV) and 15 controls without disease (7 healthy controls, 8 with symptomatic uterine fibroids), across the menstrual cycle 36 . We identified 5 main cell groups: stromal fibroblasts, smooth muscle, endothelial, epithelial, and immune. In the latter, further analysis revealed 18 sub-populations including uterine NK cells, T cell subsets, monocytes, macrophages, and dendritic cells. Epithelial subpopulations comprised glandular, luminal and ciliated cells, with abnormal expression of CXCL14, PAEP, CCL20 , MMP7, and SFRP4 genes in cases, regardless of phase. Endometrium in controls with FIGO Type 2 uterine fibroids revealed overexpression of WFDC2 and DAXX genes in cases. In immune cells, fibroblasts, and epithelial cells, cytokine production, stress signal and pro-inflammatory pathways in cases were noted, similar to Huang et al 34 findings, and enriched pathways in epithelial and ciliated cells related to migration and cell motility were higher in cases. Cell communication analysis revealed a complex network of inflammatory pathways, with significant involvement of classical monocytes and NK cells. Finally, genes implicated in ovarian cancer were upregulated in endometrium of cases compared to controls, confirming known association of endometriosis with specific subtypes of ovarian cancer 12 . Thus, in the eutopic endometrium in patients with endometriosis, there are numerous alterations in specific cell types across the cycle versus controls without disease and disease-free controls with uterine fibroids 36 . These can predispose to impaired fertility and tissue dyshomeostasis. Menstrual endometrium, disease pathophysiology, candidate biomarker discovery A recent scRNAseq study 32 of endometrial tissue in menstrual efflux (not biopsies) of n=11 cases with laparoscopic evidence of endometriosis (undefined stage) (11,924 cells) and n=9 healthy controls with no disease or symptoms (14,327 cells) revealed 18 cell clusters. These included abundant uNKs, stromal fibroblast, epithelial, B, T, and myeloid cell clusters, and a small plasmacytoid dendritic cell (pDC) cluster. Further analysis separated data by cases vs controls and revealed a marked decrease in uNK1 and uNK2 cells and significantly enriched B cells in cases ( Figure 3A ). Log2 odds ratio (OR) analysis confirmed uNK1/2 significantly enriched in controls and B cells greatly enriched in cases ( Figure 3B ). Moreover, menstrual endometrium scRNA seq revealed decreased decidualization markers in stromal fibroblasts (reflecting P 4 -resistance) and pro-inflammatory stromal cells, as observed by others in FACS-sorted cell analyses and in vitro cell culture studies 6 , 47 . Figure 3. Distinct Cell Compositions and Cell Enrichment in Menstrual Endometrium from patients with Endometriosis versus Controls and Model of Disease. Open in a new tab Panels A and B show low abundance of uNK1 and uNK2 cells and high abundance of B cells in patients with endometriosis. Panel C shows proposed pathogenesis and pathophysiology implied by the data. From Shih et al, 2022 32 , with permission. Overall, this study demonstrates that menstrual endometrium (commonly referred to as a “liquid biopsy of the endometrium”) reflects secretory endometrium abnormalities as possible biomarkers of disease. The proposed disease model is shown in Figure 3C wherein defective endometrial stromal fibroblast decidualization is driven by multiple factors including inflammation, chronic endometritis, stress, and/or P 4 -resistance – which all have clinical relevance for pregnancy establishment and success, as well as tissue homeostasis. This, in turn, may direct fibroblast differentiation to a chronic inflammatory phenotype and senescence, which may also impair decidualization. Reduced decidualization may also compromise the infiltration and proliferation of uNK cells, likely to be important for senescent cell removal and important for an optimal setting for embryo nidation 32 . Ectopic endometrium Superficial peritoneal lesions display basalis epithelial stem cell markers In their landmark spatial-temporal single cell transcriptomic study 48 , García-Alonso et al 20 analyzed endometrium (3 functionalis and 6 full thickness (functionalis and basalis) tissues) across the cycle and identified 14 unique cell clusters that localized to corresponding endometrial compartments and niches. The proposed endometrial epithelial stem-cell niche resides within the basalis layer and contains SOX9+ cells and other “stem” markers (e.g., LGR5, SSEA1, β-catenin, N-cadherin) 49 , 50 , and Garcia-Alonso et al 20 found proliferative SOX9+LGR5+ cells mapped to the basalis stem cell niche. Epithelial stem cells are believed to play a role in endometriosis pathogenesis, and by leveraging bulk (microarray) transcriptomic data of superficial peritoneal lesions (9 red, 9 white, 11 black) ( GSE141549 ), this team identified enrichment of SOX9+LGR5+ markers in peritoneal lesions in the proliferative phase, compared to normal endometrium (n=42) and peritoneum (n=12) 20 . A proliferative phenotype of endometriosis lesion epithelial cells is consistent with P 4 -resistance noted in endometriosis 4 , 6 . Moreover, dysfunctional epithelium as a major driver of endometrial diseases, including endometriosis, and discovery of SOX9+LGR5+ stem-like populations dominant in endometriosis peritoneal lesions are foundational observations about pathogenesis and pathophysiology of superficial endometriosis disease lesions deriving from the basalis layer epithelium. Endometrioma, superficial and deep infiltrating endometriosis cell atlas A comprehensive scRNAseq analysis of >370,000 individual cells from 8 ovarian endometriomas, 28 superficial and deep disease lesions, 10 eutopic endometrium, 4 unaffected ovary, and 4 endometriosis-free peritoneum samples, has resulted in a cell atlas of endometrial-type epithelial cells, stromal cells, and microenvironment cell populations across tissue types 35 . Samples were derived from 17 cases (9 only endometriosis; 8 endometriosis and adenomyosis, uterine fibroids, and/or uterine polyps). 14 were cycling (7 proliferative and 7 secretory phases), and 3 were on contraceptive steroids ( Table 1 ). Controls (n=4) were postmenopausal (3) or perimenopausal (1), without endometriosis but with uterine fibroids +/− adenomyosis +/− uterine polyps, and all were on hormonal replacement regimens. Histologic and macroscopic features of specimens from one patient, shown in Figure 4A , underscore the heterogeneity of the lesions by cell abundance, cell types, and tissue architecture. Figure 4. Endometriosis lesion scRNAseq analysis. Open in a new tab Panel A. Histologic (H&E staining) and architectural heterogeneity of endometriosis lesions. Panel B. Correlations Based on Cluster Frequency Across All Specimens Profiled by scRNAseq. Panel C. Epithelial differential gene expression and signaling pathways in endometriomas and superficial and deep disease (P < 0.05 and log2 FC > 1). From Fonseca et al, 2023 35 , with permission. The major cell types identified from scRNAseq of all cells from n=49 samples comprising the 5 major tissue classes (endometrioma, peritoneal endometriosis (deep and superficial), unaffected peritoneum, eutopic endometrium, unaffected ovary) included epithelial, smooth muscle, mesenchymal, endothelial, mast, myeloid, B, and T/NKT cells, and erythrocytes. Correlations based on cluster frequency across all specimens profiled by scRNAseq are shown in Figure 4B and reveal 3 main clusters: 1) mainly endometriosis specimens; 2) mix of endometriomas (5 of the 8 endometrioma specimens) and eutopic endometrium (5 of the 10 eutopic samples; 3) only controls: unaffected ovary and 5 of the 10 eutopic endometrium samples 35 . Endometrioma, endometrium, superficial and deep endometriosis epithelia revealed distinct cell subtypes and signaling pathways, and endometriomas displayed markedly different epithelial differential gene expression and signaling pathways compared to deep and superficial peritoneal disease and eutopic endometrium ( Figure 4C ). Interestingly, endometriomas displayed immune cell and complement activation and were enriched in B cells and plasma cells, suggesting possible infection in endometriomas and a unique role for B cells which have received limited attention in endometriosis pathophysiology 51 – 53 . Peritoneal disease was enriched in mast cells and T/NK-T cells engaging novel immune targets, and some histologically negative mesothelium surprisingly had disease signatures, although depth of lesion evaluation for endometriosis tissue may vary among pathologists. Endometrial-type epithelial cells and fibroblasts exhibited differential gene expression associated with deep or superficial disease status ( Figure 5 ). Interestingly, deep infiltrating disease epithelium displayed upregulation of nerve growth factor, suggesting a role of the epithelium in promoting innervation of this disease type which is particularly painful 12 . Figure 5. Molecular signatures of epithelial and stromal cells in superficial peritoneal and deep endometriosis differ. Open in a new tab Endometrial-type epithelial cells ( Panel A ) and fibroblasts ( Panel B ) exhibit DGE associated with deep or superficial status. From Fonseca et al, 2023 35 , with permission. While somatic mutations have been reported in endometrium and endometriosis lesions, mainly involving KRAS activating mutations and ARID1A loss of function mutations 54 , 55 , Fonseca et al 35 confirmed these somatic mutations in various lesions ( Figure 6A ) and demonstrated in vivo transcriptional consequences of these mutations at the single cell level ( Figure 6B ). ARID1A mutation was associated with pro-lymphangiogenic stroma ( Figure 6C ) and adjacent mesothelium pro-inflammatory features and ciliated epithelial signatures - consistent with ovarian oncogenic potential. It is well-known that endometrioid ovarian cancer (EOC) and clear cell ovarian cancer (CCOC) arise in association with endometriosis, suggesting endometrial-type epithelial cells may be precursors for these tumors. A multi-subject single cell deconvolution analysis of the single cell data ( Figure 7 ) revealed strong enrichment of signatures for ciliated endometrial-type epithelial cells compared with the other keratin-positive clusters. In contrast, high grade serous ovarian cancer, an epithelial ovarian cancer type, not believed to originate from endometriosis epithelial cells, showed no pattern for preferential enrichment of any epithelial cluster. Figure 6. Endometrium, endometriosis, and endometriomas share somatic mutations with transcriptional consequences. Open in a new tab Panel A. Mutations detected in each lesion. Panel B. Differential gene expression by mutation state. Panel C. Epithelial expression of lymph-angiogenesis markers by ARID1A staining status. From Fonseca et al, 2023 35 , with permission. Figure 7. Deconvolution Analysis Reveals Endometrial-type Epithelium Cell-type Signatures in Endometriosis-associated Ovarian Cancers. Open in a new tab Multi-subject single-cell deconvolution revealed strong enrichment of signatures for ciliated endometrial-type epithelial cells compared with the other keratin-positive clusters. HGSOC, an epithelial ovarian tumor type thought not to originate from endometriosis epithelial cells showed no pattern for preferential enrichment of any epithelial cluster. From Fonseca et al, 2023 35 , with permission. Key takeaways: Endometrium and all three lesion types versus control endometrium displayed different cell/molecular signatures across tissues, consistent with restructuring/transcriptional reprogramming in lesions. Epithelial and stroma differed by site with striking differential gene expression in endometriomas and peritoneal lesions suggesting these are distinct disease entities. Superficial peritoneal and deep infiltrative endometriosis epithelium and fibroblasts had some differential gene expression and pathways, but similar structural genes to each other, suggesting they are part of the same continuum and distinct from endometrioma. While this study is a tour de force of endometriosis lesion and eutopic endometrium single cell transcriptomic analysis giving major insights into cell- and tissue-specific features and functions, including oncogenic potential of endometrioma epithelial cells, it underscores some of the challenges in conducting these types of studies. These include the presence of co-existing gynecologic disorders that can confound data interpretation; cells sequenced comprise a minor population in most samples (e.g., <1% of cells profiled), and controls and normal tissues are challenging to obtain e.g., controls were peri- or post-menopausal on hormone replacement therapies and had other uterine disorders. Nonetheless, these data are important and provide foundational information for comparison with other studies on eutopic and ectopic lesions in endometriosis patients and controls without disease. Fibroblasts and immune cells: key contributors to a pro-inflammatory, angiogenic environment in endometriomas. Sequencing 55,000 single cells from 3 endometriomas, 3 matched eutopic endometrium, and 3 endometrial samples from controls without disease in the proliferative phase, Ma et al 31 found 9 cell type clusters: fibroblasts, macrophages/monocytes, neutrophils, and T, NK, epithelial, endothelial, and mast cells, and unknown. Thirteen fibroblast subtype clusters were identified with significant differences in subtype composition among ectopic endometrium, eutopic endometrium, and normal endometrium. Fibroblast subtypes from normal proliferative phase endometrium displayed enriched functions of DNA replication, repair, growth regulation, and metabolic processes. However, endometrioma fibroblasts comprised 4 subtypes: 1) cytokine, inflammatory response; 2) FGF stimulation, immune response; 3) extracellular matrix (ECM) organization, cell adhesion; 4) angiogenesis, hypoxia response, and epithelial-mesenchymal transition (EMT), all of which were enriched in MAPK, TNF, IL-17, TGF-β signaling pathways. Fibroblasts from eutopic endometrium of cases displayed many pathways similar to normal tissue, but some features suggested a “transition state” from normal endometrium to endometrioma, involving TGF-β, MAPK, Rho-Rack, NF-kB, JAK-STAT, and EMT. A developmental trajectory was postulated based on these fibroblast features, with endometrioma fibroblasts derived from endometrial fibroblasts 31 , much as the epithelial cell population has been proposed, based on shared epithelial somatic mutations in these two tissues 54 . Thus, endometrioma and paired eutopic endometrium fibroblasts differ, and the single cell data support their key roles in disease pathogenesis. Analysis of immune cells revealed they differed significantly in normal endometrium, endometrioma, and paired eutopic tissue ( Figure 8A ). Specifically, T cell and uNK cell frequencies were lower, uNK cells were more active, and macrophages (Mφ) were enriched and had features of tissue remodeling versus eutopic endometrium 31 , similar to data from Fonseca et al 35 . Thus, in the proliferative phase, fibroblasts and immune cell subpopulations contribute to a pro-inflammatory, angiogenic environment in endometriomas. Moreover cell-to-cell communication/interactome analysis revealed that immune cells stimulate fibroblast adhesion and growth in endometriomas ( Figure 8B ), underscoring the power of single cell analyses to inform the pathogenesis and pathophysiology of endometriomas and cell-cell communications. Furthermore, recent observations indicated that Fusobacterium 56 and dysbiosis of gut 57 , 58 altered systemic and potentially local immune functions, contributing to the development of endometriosis. Notably, while transcriptomic alterations of specific populations and sub- populations of the endometrium, lesions, and immune cells (local and peripheral) under these conditions awaits further study, antibiotic and nutritional therapies for dysbiosis is emerging as a new paradigm for treating endometriosis development and symptom management. Figure 8. Immune cell frequencies in endometriomas and communication networks of immune cell types with endometrioma fibroblasts. Open in a new tab The network supports interactions between immune and fibroblast cells in ovarian endometrioma pathogenesis and pathophysiology. From Ma et al, 2021 31 , with permission. Endometriosis peritoneal and ovary lesions of patients on progestins As progestins are a mainstay of medical therapy for endometriosis-related pain 12 , analysis of lesions and endometrium of subjects taking progestins is highly valuable in understanding their effects on cell types and tissue features. Tan et al 33 performed scRNAseq on 122,000 cells from 14 individuals, selective cell localization by imaging mass cytometry, and validations in endometrial epithelial organoid cultures. Peritoneal and/or ovarian disease and endometrium from 19 cases (rASRM stage II-IV) included 14 on norethindrone (NETA) + ethinyl estradiol (EthE 2 ), 2 on drosperinone + EthE 2 , 1 on levonorgestrel + EthE 2 , 1 on only NETA, 1 menstrual phase, and 8 controls with no endometriosis or inflammatory conditions: 4 on NETA + EthE 2 , 2 in proliferative phase, 1 on medroxyprogesterone acetate, 1 on norgestromin + EthE 2 . The peritoneal lesions had similar cell compositions as eutopic endometrium but dysregulated innate immune and vascular components - vastly different from ovarian endometriosis lesions. The authors identified a unique perivascular mural cell specific to the peritoneal lesions with features promoting neo- angiogenesis and immune cell trafficking. They also identified a progenitor-like epithelial subpopulation, different from the SOX9+LGR5+ epithelial subpopulation found by Garcia-Alfonso et al 20 . Analysis of peritoneal lesion myeloid populations of immunomodulatory Mφ and DCs revealed a coordinated immunotolerant phenotype in the disease microenvironment, promoting immunosurveillance escape and thus conducive to lesion establishment. Overall, the data demonstrate that immune and vascular components of peritoneal endometriosis favor neo-angiogenesis and an immune tolerant niche in the peritoneal cavity, beneficial for lesion establishment and growth. These are remarkable findings given that most subjects were on hormone suppressive therapies for pain [progestin (+ estrogen)] and raises the issue of how efficacious these therapies are on specific cells and the role of estrogenic components in disease management. This valuable single cell atlas of disease from patients largely on hormonal treatments offers the opportunity for comparisons of cell types and features in peritoneal lesions and endometriomas in the setting of other hormonal treatments and/or in various phases of the cycle to interpret efficacies of selective therapies on cell-specific pathways and cell-cell communications. Adenomyosis through single cell glasses Uterine compartments relevant to adenomyosis pathogenesis include the endometrial-myometrial junction (EMJ) [basalis and lower functionalis endometrium + sub-endometrial inner myometrium], all of Mullerian origin and displaying cycling estrogen receptors (ER) and progesterone receptors (PR) 59 – 61 ( Figure 9 ). Few molecular and genomic studies of the EMJ in adenomyosis uteri have been conducted, likely as access to basalis and inner myometrium requires hysterectomy versus the functionalis, which can be readily obtained by office biopsy. Figure 9. The endometrial-myometrial junction (EMJ). Open in a new tab Panel A. Histology of human uterine endometrium and myometrium compartments (H&E staining). Panel B : Transvaginal ultrasound midsagittal image of uterine compartments. Panel C. Schematic of uterine compartments. Adapted from Naftlin and Jurkovic, 2009 61 , with permission. Bulk analyses: Studies to date in adenomyosis patients using immunohistochemistry and qPCR reveal abnormal endometrial functionalis epithelial aromatase expression (thus high E 2 milieu), elevated stromal fibroblast IL-6, IL-8, IL-17 secretion, TH17-Treg imbalance, altered uNK receptors, and aberrant P 4 -signaling 5 , 26 , 62 – 64 – very similar to endometrium of patients with endometriosis except for IL-17 and TH17-Tregulatory cell imbalance. We conducted bulk RNA seq of endometrial functionalis 65 and a study on separate compartments of EMJ inner myometrium and endometrium from patients with diffuse adenomyosis versus controls in the proliferative phase 66 . ECM remodeling, inflammation, IL-17 signaling, and mTOR signaling were enriched in endometrium, and inflammation, pain, and neuronal signaling pathways in myometrium, consistent with clinical symptoms of this disorder. Single cell analyses: Two recent publications have analyzed adenomyosis lesions 67 , 68 . They are important beginnings to understanding the pathogenesis and pathophysiology of adenomyosis, building upon bulk transcriptomic data, which could be further mined in deconvolution studies to validate new and likely forthcoming additional single cell data. Liu et al 67 sequenced 42,292 cells derived from a control endometrium of a patient with uterine fibroids and eutopic (functionalis) endometrium and matched ectopic lesion from a patient with adenomyosis (type and menstrual cycle phase undefined). The data revealed seven distinct cell types. Comparison of the data from the lesion with those from eutopic matched endometrium and control endometrium revealed processes involved in progression of normal→endometrial→adenomyosis epithelial phenotypes. These included motility, proliferation, angiogenesis, inflammation, and cancer, consistent with the theory of collective cell migration and invasion into the myometrium of endometrial-derived cells. The novel data set revealed vascular mimicry (important in tumor formation) and epithelial transformation to endothelial cells. It is anticipated that the novel epithelial→endothelial transition phenotype will likely be mined in the future and expanded studies of adenomyosis and perhaps other gynecologic disorders and may serve as a unique cell type for drug targeting. Bulun and colleagues 68 reported results of single cell analyses of adenomyosis lesions and matched eutopic endometrium and myometrium from 3 subjects, all with diffuse adenomyosis and in the proliferative phase (total of 9 samples and 66,000 cells sequenced). They identified 11 cell types in eutopic endometrium of patients with adenomyosis ( Figure 10 A ). Fibroblast-like cells comprised a large fraction (36%), and pseudotime analysis revealed they originate from pericyte progenitors that differentiate to ER+ and PR+ endometrial stromal fibroblasts. Notably, the pseudotime trajectory was found to end with ciliated epithelial cells, suggesting mesenchymal- epithelial transition at play in this tissue. In adenomyosis lesions, they identified 13 cell types including 2 (smooth muscle cells and neurons) not found in the endometrium of the same patients ( Figure 10 B ). The adenomyosis lesions contained even more fibroblast-like cells compared to endometrium (50% versus 36%, respectively). In contrast to matched endometrium, the fibroblast-like cells did not originate from pericyte progenitors and differentiated to ECM-expressing fibroblasts and smooth muscle cells (SMC). Also, myometrial SMC were found to derive from fibroblast-like clusters. Moreover, the WNT inhibitor, secreted frizzled related protein (SFRP) family expressed in fibroblast-like clusters were similar across tissues - endometrium, adenomyosis and myometrium 68 . Figure 10. Fibroblasts and epithelial cell subtypes differ in endometrium and adenomyosis. Open in a new tab Panel A. UMAP of cell types and fibroblast trajectories in endometrium. B. UMAP of cell types and fibroblast trajectories in adenomyosis. Panel C. Progesterone resistance of epithelial subtypes in adenomyosis. Panel D. WNT signaling pathway network. From Yildiz et al, 2023 68 , with permission. Regarding the epithelial cells, P 4 -resistance was found in ciliated and unciliated cells in adenomyosis tissue versus matched eutopic endometrium ( Figure 10 C ) – this is a key finding regarding treatment efficiencies. Further data analysis revealed that the canonical WNT/SFRP pathway signaling was uniquely activated in adenomyosis unciliated and ciliated epithelium and paracrine signaling fibroblast-like mesenchymal: epithelial interaction networks ( Figure 10D ), opening new avenues for drug discovery for this disorder. The epithelial→endothelial transition reported by Liu et al 67 was not detected in this study and may be due to different adenomyosis lesion types, cycle phase, patient age, and other variables. Whether the findings of these two studies involving patients with diffuse adenomyosis also apply to other forms of adenomyosis ( Figure 1 ) is unknown and is an opportunity to investigate cell features wherein novel therapeutics could be derived. Moreover, recent data suggest that the endometrial and vaginal microbiomes differ in patients with versus without adenomyosis 69 , 70 , raising the possibility of adjunctive antibiotic therapies for symptomatic patients with disease. Additionally, further study of the local and systemic immune system cells and their subtypes as well as in affected uterine tissues at single cell resolution are warranted.',\n",
       " 'Discussion': '',\n",
       " 'Conclusion': '',\n",
       " 'URL': 'https://doi.org/10.1016/j.ajog.2024.08.043',\n",
       " 'Journal': 'Am J Obstet Gynecol'}"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "prueba"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "917d03c0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Endometriosis through single cell glasses Subsequent to the single cell '\n",
      " 'cartography of human endometrium published by Wang et al 19 (see Chapter 3), '\n",
      " 'several groups have recently reported eutopic endometrial scRNAseq data of '\n",
      " 'patients with endometriosis, endometriosis lesions and control eutopic '\n",
      " 'endometrium 20 , 31 – 36 ( Table 1 ). Types of samples, inclusion/exclusion '\n",
      " 'criteria, choice of controls, study designs, numbers of cells sequenced, and '\n",
      " 'number of reads (i.e., base pairs sequenced) are key to data integrity and '\n",
      " 'interpretation and to comparative analyses across tissues and studies. They '\n",
      " 'have provided insights into origins of ectopic disease, the heterogeneity of '\n",
      " 'cell types/subtypes, unique clusters and signatures of endometrium and '\n",
      " 'ectopic lesions informing mechanisms and pathways involved in cellular '\n",
      " 'dysfunctions relevant to pain and fertility compromise, novel cell-cell '\n",
      " 'communications, relationships between lesions and eutopic tissue/cells, and '\n",
      " 'biomarker discovery. Across these studies, samples were obtained in '\n",
      " 'different hormonal milieu (menstrual cycle phase, exogenous hormones), and '\n",
      " 'between 55,000 to 378,000 cells were sequenced ( Table 1 ). As cycle phase '\n",
      " 'and exogenous hormone use are key drivers of endometrial gene expression 37 '\n",
      " '– 44 , comparison of data across studies can be challenging and mixed '\n",
      " 'hormonal milieu are sometimes experienced within the same study in both '\n",
      " 'cases and/or controls. Below we summarize results of studies focused '\n",
      " 'exclusively on eutopic endometrium of patients with endometriosis relevant '\n",
      " 'to endometrial function and dysfunction and biomarker discovery, followed by '\n",
      " 'studies focused on ovarian endometriomas, superficial, and/or deep '\n",
      " 'infiltrating disease, and eutopic endometrium, relevant to disease '\n",
      " 'heterogeneity, pathophysiology, pathogenesis, and in the setting of '\n",
      " 'progestins – commonly prescribed for symptom management of patients with '\n",
      " 'endometriosis and dysmenorrhea and chronic pelvic pain.\\n'\n",
      " 'Table 1. Highlights of recent endometriosis-related scRNAseq studies.\\n'\n",
      " 'Reference Tissue Analyzed Total cells sequenced Technology/Platform Total '\n",
      " 'subjects Endometriosis Type and ASRM Stage Controls Race/ Ethnicity '\n",
      " 'Hormones/IUD Cycle Phase Endometrial histology Ma 2021 endometrium, ovarian '\n",
      " 'endometrioma 55,000 scRNAseq (10X) n=6 subjects (n=3 cases, n=3 controls) '\n",
      " 'ASRM Stage III, IV ovarian endometriomas Healthy controls without '\n",
      " 'endometriosis N/A None All in prolifeative phase Garcia-Alonso 2022 '\n",
      " 'endometrium (functionalis, full thickness), endometriosis peritoneal lesions '\n",
      " '(red, white, black) 98,569 sequenced scRNAseq, snRNA seq, spatial profilng. '\n",
      " 'Lesion microarray data ( GSE141549 ) n=3 functionalis n=6 full thickness; '\n",
      " 'Microarray data controls: endom n=42, peritoneum n=12; n=9 red, 9 white, 11 '\n",
      " 'black Peritoneal disease: red, white, black Healthy controls (functionalis '\n",
      " 'layer) n=6 full thickness without reproductive disorders; normal peritoneum '\n",
      " 'N/A None Proliferative, secretory; Microarray samples: Control Endo 17PE, '\n",
      " '25SE; Perit 4PE, 8SE; Lesions: Red 2PE, 7 SE; White 5PE, 4SE; Black 6PE, 5 '\n",
      " 'SE Shih 2022 menstrual endometrium 43,054 scRNAseq (10X) n=33 subjects (n=11 '\n",
      " 'Dx, n=13 sx, n=9 controls) N/A; another group with Sx but no Dx. No '\n",
      " 'endometriosis diagnosis. ? Other GYN disorders White: 10 Dx, 13 sx, 7 '\n",
      " 'controls Black: 0 Dx, 0 sx, 1 control Hispanic: 0 Dx, 0 sx, 0 cont Mixed: 0 '\n",
      " 'Dx,0 sx, 1 control Other: 1 Dx,0 sx, 0 control None on hormones, except 1 '\n",
      " 'case used vaginal P4. Re-analysis showed no impact on results Menstrual '\n",
      " '(heaviest flow, mostly CD 1 or 2) Tan 2022 endometrium, enodmetriosis '\n",
      " 'lesions 122,000 scRNAseq (10X) IMC organoids n=27 subjects (n=19 cases, n=8 '\n",
      " 'controls; n=14 sequenced ASRM Stage II-IV, peritoneal lesions, ovarian '\n",
      " '“lesions”, organoids No endometriosis or infammatory conditions White: 9 '\n",
      " 'cases, 3 controls Asian: 4 cases, 2 controls Hisp: 5 cases, 3 controls '\n",
      " 'Black: 1 case, 0 controls Progestin [NETA, LVN oral, IUD, drosperinone, '\n",
      " 'norelgestromin] +/− ethinyl E2, medroxy-progesterone acetate, copperIUD wkly '\n",
      " 'PE: 3 cases, 2 control; inactive 3 cases, 0 control; mens: 1 case, 0 '\n",
      " 'control, exog hormone effect: 7 cases, 7 control PE: 0 cases, 2 controls IE: '\n",
      " '0 cases, 1 control ESE: 4 cases, 0 control N/A: 1 case, 1 control Fonseca '\n",
      " '2023 endometrium, endometriosis lesions, unaffected ovary and peritoneum '\n",
      " '373,851 digital scRNAseq n=21 subjects n=17 cases, n=4 controls n-54 '\n",
      " 'specimens collected Cases: n=17 Endometrium Endometrioma Superficial and '\n",
      " 'deep disesase n=9 w/o GYN disorders; n=8 w/ adenomyosis, uterine fibroids '\n",
      " '+/− polyp Controls n=4 n=3 PMP, 1 peri-menopause All no evidence of '\n",
      " 'endometriosis, all w/ leiomyoma +/− adenomyosis +/− uterine polyp White: 13 '\n",
      " 'cases,2 controls Black: 2 cases, 0 controls Asian: 1 case, 0 control Other: '\n",
      " '1 case, 0 control Cases : n=14 on no hormones. n=1 w/ vaginal ring E+P; n=1 '\n",
      " 'on E/T, P4; n=1 on E2+P4. Controls : 3 of 4 on E +/− P4; NETA; n=1 '\n",
      " 'peri-menopause in luteal phase Cases : Proliferative phase: n=7; secretory '\n",
      " 'phase: n=9; N/A (on hormones) n=3. Controls : Proliferative n=0, Secretory '\n",
      " 'n=1, N/A (on hormones) n=3 Huang 2023 endometrium 128,243 scRNAseq (10X) '\n",
      " 'n=10 sequenced n=6 cases, n=4 controls ASRM Stage I,II No endometriosis, '\n",
      " 'benign ovarian cysts N/A None EPE: 3 cases, 3 controls MSE: 3 cases, 3 '\n",
      " 'controls LSE: 1 control Almonte 2023 endometrium >250,000 scRNseq (10X) n=29 '\n",
      " 'cases, 7 controls w/o disease and 8 controls with fibroids ASRM stage I-IV '\n",
      " 'No endometriosis or other GYN d/o; No endo but + uterine fibroids N/A None '\n",
      " 'Proliferative:12 cases, 5 controls. Secretory: 14 cases, 3 controls. Open in '\n",
      " 'a new tab Abbreviations: CD=cycle day; dx=diagnosis; E=estradiol; ESE=early '\n",
      " 'secretory phase; exog=exogenous; IE=interval endometrium; IMC=imaging mass '\n",
      " 'cytometry; IUD=intrauterine device; LSE=late secretory phase; '\n",
      " 'LVN=levonorgestrel; MSE=mid-secretory phase; N/A=not available; '\n",
      " 'NETA=norethindrone acetate; P=progestin; P=progesterone; PE=proliferative '\n",
      " 'phase; SE=secretory phase; sx=symptoms, w/= with; +/− = with and/or without. '\n",
      " 'Adapted from Giudice et al 4 , with permission. Eutopic endometrium Window '\n",
      " 'of implantation epithelial abnormalities in infertile endometriosis patients '\n",
      " 'Endometrial biopsies of cases with stage I/II disease (revised American '\n",
      " 'Society for Reproductive Medicine (rASRM) scoring system) 45 and infertility '\n",
      " '(n=3 WOI, n=3 early proliferative phase) versus n=7 fertile controls without '\n",
      " 'disease (3 WOI, 3 proliferative, 1 late secretory phase) were studied by '\n",
      " 'Huang et al 34 . A total of 128,243 cells were sequenced, and 30 type/cell '\n",
      " 'type clusters were identified. Notably, a unique epithelial cell cluster in '\n",
      " 'WOI endometrium of endometriosis patients was identified that lacked '\n",
      " 'expression of two key genes associated with implantation: PAEP '\n",
      " '(progesterone-associated endometrial protein) and CXCL14 (chemokine (CXC '\n",
      " 'motif) ligand 14) ( Figure 2 ). PAEP, also known as glycodelin A and '\n",
      " 'placental protein 14, is involved in immune response regulation, promoting '\n",
      " 'opening of the WOI by lowering maternal immune responses to an implanting '\n",
      " 'embryo 46 . CXCL14 is involved in activating the innate immune response '\n",
      " 'stimulating chemotaxis of uterine natural killer cells (uNK) and other '\n",
      " 'monocytes to epithelial glands for implantation 46 . Other findings included '\n",
      " 'absence of normal variation of endometrial uNK and T cell numbers across the '\n",
      " 'cycle, and lower anti-inflammatory IL-10 and more pro-inflammatory cytokines '\n",
      " 'expressed in endometrial immune cells in the WOI versus in the proliferative '\n",
      " 'phase in cases and the opposite in controls. Notably, 11 ligand–receptor '\n",
      " 'pairs were upregulated between endometrial immune and epithelial cells '\n",
      " 'during the WOI of cases. Figure 2. Absence of key genes that open the window '\n",
      " 'of implantation in eutopic endometrial epithelial cluster in infertile '\n",
      " 'patients with Stage I/II endometriosis. Open in a new tab Panel A . t-SNE '\n",
      " 'projection of endometrial cells colored based on their sample origin and '\n",
      " 'cycle phase. Panel B . Expression patterns of PAEP and CXCL14 projected on '\n",
      " 't-distributed stochastic neighbour embedding (t-SNE) plot. From Huang et al, '\n",
      " '2023 34 , with permission. Thus, lower epithelial receptivity markers, '\n",
      " 'abnormal immune cell frequencies, a pro- inflammatory WOI, and new insights '\n",
      " 'into cell-cell communications between epithelium and immune cells support an '\n",
      " 'important role of the immune system and an adverse environment for embryo '\n",
      " 'implantation for fertility and potentially for poor pregnancy outcomes in '\n",
      " 'patients with stage I/II endometriosis who do achieve pregnancy versus '\n",
      " 'fertile controls. Endometrial subcellular alterations across the cycle in '\n",
      " 'endometriosis patients In a recent study, our group sequenced eutopic '\n",
      " 'endometrium (237,620 cells) from 29 subjects with endometriosis (stages I/II '\n",
      " 'and III/IV) and 15 controls without disease (7 healthy controls, 8 with '\n",
      " 'symptomatic uterine fibroids), across the menstrual cycle 36 . We identified '\n",
      " '5 main cell groups: stromal fibroblasts, smooth muscle, endothelial, '\n",
      " 'epithelial, and immune. In the latter, further analysis revealed 18 '\n",
      " 'sub-populations including uterine NK cells, T cell subsets, monocytes, '\n",
      " 'macrophages, and dendritic cells. Epithelial subpopulations comprised '\n",
      " 'glandular, luminal and ciliated cells, with abnormal expression of CXCL14, '\n",
      " 'PAEP, CCL20 , MMP7, and SFRP4 genes in cases, regardless of phase. '\n",
      " 'Endometrium in controls with FIGO Type 2 uterine fibroids revealed '\n",
      " 'overexpression of WFDC2 and DAXX genes in cases. In immune cells, '\n",
      " 'fibroblasts, and epithelial cells, cytokine production, stress signal and '\n",
      " 'pro-inflammatory pathways in cases were noted, similar to Huang et al 34 '\n",
      " 'findings, and enriched pathways in epithelial and ciliated cells related to '\n",
      " 'migration and cell motility were higher in cases. Cell communication '\n",
      " 'analysis revealed a complex network of inflammatory pathways, with '\n",
      " 'significant involvement of classical monocytes and NK cells. Finally, genes '\n",
      " 'implicated in ovarian cancer were upregulated in endometrium of cases '\n",
      " 'compared to controls, confirming known association of endometriosis with '\n",
      " 'specific subtypes of ovarian cancer 12 . Thus, in the eutopic endometrium in '\n",
      " 'patients with endometriosis, there are numerous alterations in specific cell '\n",
      " 'types across the cycle versus controls without disease and disease-free '\n",
      " 'controls with uterine fibroids 36 . These can predispose to impaired '\n",
      " 'fertility and tissue dyshomeostasis. Menstrual endometrium, disease '\n",
      " 'pathophysiology, candidate biomarker discovery A recent scRNAseq study 32 of '\n",
      " 'endometrial tissue in menstrual efflux (not biopsies) of n=11 cases with '\n",
      " 'laparoscopic evidence of endometriosis (undefined stage) (11,924 cells) and '\n",
      " 'n=9 healthy controls with no disease or symptoms (14,327 cells) revealed 18 '\n",
      " 'cell clusters. These included abundant uNKs, stromal fibroblast, epithelial, '\n",
      " 'B, T, and myeloid cell clusters, and a small plasmacytoid dendritic cell '\n",
      " '(pDC) cluster. Further analysis separated data by cases vs controls and '\n",
      " 'revealed a marked decrease in uNK1 and uNK2 cells and significantly enriched '\n",
      " 'B cells in cases ( Figure 3A ). Log2 odds ratio (OR) analysis confirmed '\n",
      " 'uNK1/2 significantly enriched in controls and B cells greatly enriched in '\n",
      " 'cases ( Figure 3B ). Moreover, menstrual endometrium scRNA seq revealed '\n",
      " 'decreased decidualization markers in stromal fibroblasts (reflecting P 4 '\n",
      " '-resistance) and pro-inflammatory stromal cells, as observed by others in '\n",
      " 'FACS-sorted cell analyses and in vitro cell culture studies 6 , 47 . Figure '\n",
      " '3. Distinct Cell Compositions and Cell Enrichment in Menstrual Endometrium '\n",
      " 'from patients with Endometriosis versus Controls and Model of Disease. Open '\n",
      " 'in a new tab Panels A and B show low abundance of uNK1 and uNK2 cells and '\n",
      " 'high abundance of B cells in patients with endometriosis. Panel C shows '\n",
      " 'proposed pathogenesis and pathophysiology implied by the data. From Shih et '\n",
      " 'al, 2022 32 , with permission. Overall, this study demonstrates that '\n",
      " 'menstrual endometrium (commonly referred to as a “liquid biopsy of the '\n",
      " 'endometrium”) reflects secretory endometrium abnormalities as possible '\n",
      " 'biomarkers of disease. The proposed disease model is shown in Figure 3C '\n",
      " 'wherein defective endometrial stromal fibroblast decidualization is driven '\n",
      " 'by multiple factors including inflammation, chronic endometritis, stress, '\n",
      " 'and/or P 4 -resistance – which all have clinical relevance for pregnancy '\n",
      " 'establishment and success, as well as tissue homeostasis. This, in turn, may '\n",
      " 'direct fibroblast differentiation to a chronic inflammatory phenotype and '\n",
      " 'senescence, which may also impair decidualization. Reduced decidualization '\n",
      " 'may also compromise the infiltration and proliferation of uNK cells, likely '\n",
      " 'to be important for senescent cell removal and important for an optimal '\n",
      " 'setting for embryo nidation 32 . Ectopic endometrium Superficial peritoneal '\n",
      " 'lesions display basalis epithelial stem cell markers In their landmark '\n",
      " 'spatial-temporal single cell transcriptomic study 48 , García-Alonso et al '\n",
      " '20 analyzed endometrium (3 functionalis and 6 full thickness (functionalis '\n",
      " 'and basalis) tissues) across the cycle and identified 14 unique cell '\n",
      " 'clusters that localized to corresponding endometrial compartments and '\n",
      " 'niches. The proposed endometrial epithelial stem-cell niche resides within '\n",
      " 'the basalis layer and contains SOX9+ cells and other “stem” markers (e.g., '\n",
      " 'LGR5, SSEA1, β-catenin, N-cadherin) 49 , 50 , and Garcia-Alonso et al 20 '\n",
      " 'found proliferative SOX9+LGR5+ cells mapped to the basalis stem cell niche. '\n",
      " 'Epithelial stem cells are believed to play a role in endometriosis '\n",
      " 'pathogenesis, and by leveraging bulk (microarray) transcriptomic data of '\n",
      " 'superficial peritoneal lesions (9 red, 9 white, 11 black) ( GSE141549 ), '\n",
      " 'this team identified enrichment of SOX9+LGR5+ markers in peritoneal lesions '\n",
      " 'in the proliferative phase, compared to normal endometrium (n=42) and '\n",
      " 'peritoneum (n=12) 20 . A proliferative phenotype of endometriosis lesion '\n",
      " 'epithelial cells is consistent with P 4 -resistance noted in endometriosis 4 '\n",
      " ', 6 . Moreover, dysfunctional epithelium as a major driver of endometrial '\n",
      " 'diseases, including endometriosis, and discovery of SOX9+LGR5+ stem-like '\n",
      " 'populations dominant in endometriosis peritoneal lesions are foundational '\n",
      " 'observations about pathogenesis and pathophysiology of superficial '\n",
      " 'endometriosis disease lesions deriving from the basalis layer epithelium. '\n",
      " 'Endometrioma, superficial and deep infiltrating endometriosis cell atlas A '\n",
      " 'comprehensive scRNAseq analysis of >370,000 individual cells from 8 ovarian '\n",
      " 'endometriomas, 28 superficial and deep disease lesions, 10 eutopic '\n",
      " 'endometrium, 4 unaffected ovary, and 4 endometriosis-free peritoneum '\n",
      " 'samples, has resulted in a cell atlas of endometrial-type epithelial cells, '\n",
      " 'stromal cells, and microenvironment cell populations across tissue types 35 '\n",
      " '. Samples were derived from 17 cases (9 only endometriosis; 8 endometriosis '\n",
      " 'and adenomyosis, uterine fibroids, and/or uterine polyps). 14 were cycling '\n",
      " '(7 proliferative and 7 secretory phases), and 3 were on contraceptive '\n",
      " 'steroids ( Table 1 ). Controls (n=4) were postmenopausal (3) or '\n",
      " 'perimenopausal (1), without endometriosis but with uterine fibroids +/− '\n",
      " 'adenomyosis +/− uterine polyps, and all were on hormonal replacement '\n",
      " 'regimens. Histologic and macroscopic features of specimens from one patient, '\n",
      " 'shown in Figure 4A , underscore the heterogeneity of the lesions by cell '\n",
      " 'abundance, cell types, and tissue architecture. Figure 4. Endometriosis '\n",
      " 'lesion scRNAseq analysis. Open in a new tab Panel A. Histologic (H&E '\n",
      " 'staining) and architectural heterogeneity of endometriosis lesions. Panel B. '\n",
      " 'Correlations Based on Cluster Frequency Across All Specimens Profiled by '\n",
      " 'scRNAseq. Panel C. Epithelial differential gene expression and signaling '\n",
      " 'pathways in endometriomas and superficial and deep disease (P < 0.05 and '\n",
      " 'log2 FC > 1). From Fonseca et al, 2023 35 , with permission. The major cell '\n",
      " 'types identified from scRNAseq of all cells from n=49 samples comprising the '\n",
      " '5 major tissue classes (endometrioma, peritoneal endometriosis (deep and '\n",
      " 'superficial), unaffected peritoneum, eutopic endometrium, unaffected ovary) '\n",
      " 'included epithelial, smooth muscle, mesenchymal, endothelial, mast, myeloid, '\n",
      " 'B, and T/NKT cells, and erythrocytes. Correlations based on cluster '\n",
      " 'frequency across all specimens profiled by scRNAseq are shown in Figure 4B '\n",
      " 'and reveal 3 main clusters: 1) mainly endometriosis specimens; 2) mix of '\n",
      " 'endometriomas (5 of the 8 endometrioma specimens) and eutopic endometrium (5 '\n",
      " 'of the 10 eutopic samples; 3) only controls: unaffected ovary and 5 of the '\n",
      " '10 eutopic endometrium samples 35 . Endometrioma, endometrium, superficial '\n",
      " 'and deep endometriosis epithelia revealed distinct cell subtypes and '\n",
      " 'signaling pathways, and endometriomas displayed markedly different '\n",
      " 'epithelial differential gene expression and signaling pathways compared to '\n",
      " 'deep and superficial peritoneal disease and eutopic endometrium ( Figure 4C '\n",
      " '). Interestingly, endometriomas displayed immune cell and complement '\n",
      " 'activation and were enriched in B cells and plasma cells, suggesting '\n",
      " 'possible infection in endometriomas and a unique role for B cells which have '\n",
      " 'received limited attention in endometriosis pathophysiology 51 – 53 . '\n",
      " 'Peritoneal disease was enriched in mast cells and T/NK-T cells engaging '\n",
      " 'novel immune targets, and some histologically negative mesothelium '\n",
      " 'surprisingly had disease signatures, although depth of lesion evaluation for '\n",
      " 'endometriosis tissue may vary among pathologists. Endometrial-type '\n",
      " 'epithelial cells and fibroblasts exhibited differential gene expression '\n",
      " 'associated with deep or superficial disease status ( Figure 5 ). '\n",
      " 'Interestingly, deep infiltrating disease epithelium displayed upregulation '\n",
      " 'of nerve growth factor, suggesting a role of the epithelium in promoting '\n",
      " 'innervation of this disease type which is particularly painful 12 . Figure '\n",
      " '5. Molecular signatures of epithelial and stromal cells in superficial '\n",
      " 'peritoneal and deep endometriosis differ. Open in a new tab Endometrial-type '\n",
      " 'epithelial cells ( Panel A ) and fibroblasts ( Panel B ) exhibit DGE '\n",
      " 'associated with deep or superficial status. From Fonseca et al, 2023 35 , '\n",
      " 'with permission. While somatic mutations have been reported in endometrium '\n",
      " 'and endometriosis lesions, mainly involving KRAS activating mutations and '\n",
      " 'ARID1A loss of function mutations 54 , 55 , Fonseca et al 35 confirmed these '\n",
      " 'somatic mutations in various lesions ( Figure 6A ) and demonstrated in vivo '\n",
      " 'transcriptional consequences of these mutations at the single cell level ( '\n",
      " 'Figure 6B ). ARID1A mutation was associated with pro-lymphangiogenic stroma '\n",
      " '( Figure 6C ) and adjacent mesothelium pro-inflammatory features and '\n",
      " 'ciliated epithelial signatures - consistent with ovarian oncogenic '\n",
      " 'potential. It is well-known that endometrioid ovarian cancer (EOC) and clear '\n",
      " 'cell ovarian cancer (CCOC) arise in association with endometriosis, '\n",
      " 'suggesting endometrial-type epithelial cells may be precursors for these '\n",
      " 'tumors. A multi-subject single cell deconvolution analysis of the single '\n",
      " 'cell data ( Figure 7 ) revealed strong enrichment of signatures for ciliated '\n",
      " 'endometrial-type epithelial cells compared with the other keratin-positive '\n",
      " 'clusters. In contrast, high grade serous ovarian cancer, an epithelial '\n",
      " 'ovarian cancer type, not believed to originate from endometriosis epithelial '\n",
      " 'cells, showed no pattern for preferential enrichment of any epithelial '\n",
      " 'cluster. Figure 6. Endometrium, endometriosis, and endometriomas share '\n",
      " 'somatic mutations with transcriptional consequences. Open in a new tab Panel '\n",
      " 'A. Mutations detected in each lesion. Panel B. Differential gene expression '\n",
      " 'by mutation state. Panel C. Epithelial expression of lymph-angiogenesis '\n",
      " 'markers by ARID1A staining status. From Fonseca et al, 2023 35 , with '\n",
      " 'permission. Figure 7. Deconvolution Analysis Reveals Endometrial-type '\n",
      " 'Epithelium Cell-type Signatures in Endometriosis-associated Ovarian Cancers. '\n",
      " 'Open in a new tab Multi-subject single-cell deconvolution revealed strong '\n",
      " 'enrichment of signatures for ciliated endometrial-type epithelial cells '\n",
      " 'compared with the other keratin-positive clusters. HGSOC, an epithelial '\n",
      " 'ovarian tumor type thought not to originate from endometriosis epithelial '\n",
      " 'cells showed no pattern for preferential enrichment of any epithelial '\n",
      " 'cluster. From Fonseca et al, 2023 35 , with permission. Key takeaways: '\n",
      " 'Endometrium and all three lesion types versus control endometrium displayed '\n",
      " 'different cell/molecular signatures across tissues, consistent with '\n",
      " 'restructuring/transcriptional reprogramming in lesions. Epithelial and '\n",
      " 'stroma differed by site with striking differential gene expression in '\n",
      " 'endometriomas and peritoneal lesions suggesting these are distinct disease '\n",
      " 'entities. Superficial peritoneal and deep infiltrative endometriosis '\n",
      " 'epithelium and fibroblasts had some differential gene expression and '\n",
      " 'pathways, but similar structural genes to each other, suggesting they are '\n",
      " 'part of the same continuum and distinct from endometrioma. While this study '\n",
      " 'is a tour de force of endometriosis lesion and eutopic endometrium single '\n",
      " 'cell transcriptomic analysis giving major insights into cell- and '\n",
      " 'tissue-specific features and functions, including oncogenic potential of '\n",
      " 'endometrioma epithelial cells, it underscores some of the challenges in '\n",
      " 'conducting these types of studies. These include the presence of co-existing '\n",
      " 'gynecologic disorders that can confound data interpretation; cells sequenced '\n",
      " 'comprise a minor population in most samples (e.g., <1% of cells profiled), '\n",
      " 'and controls and normal tissues are challenging to obtain e.g., controls '\n",
      " 'were peri- or post-menopausal on hormone replacement therapies and had other '\n",
      " 'uterine disorders. Nonetheless, these data are important and provide '\n",
      " 'foundational information for comparison with other studies on eutopic and '\n",
      " 'ectopic lesions in endometriosis patients and controls without disease. '\n",
      " 'Fibroblasts and immune cells: key contributors to a pro-inflammatory, '\n",
      " 'angiogenic environment in endometriomas. Sequencing 55,000 single cells from '\n",
      " '3 endometriomas, 3 matched eutopic endometrium, and 3 endometrial samples '\n",
      " 'from controls without disease in the proliferative phase, Ma et al 31 found '\n",
      " '9 cell type clusters: fibroblasts, macrophages/monocytes, neutrophils, and '\n",
      " 'T, NK, epithelial, endothelial, and mast cells, and unknown. Thirteen '\n",
      " 'fibroblast subtype clusters were identified with significant differences in '\n",
      " 'subtype composition among ectopic endometrium, eutopic endometrium, and '\n",
      " 'normal endometrium. Fibroblast subtypes from normal proliferative phase '\n",
      " 'endometrium displayed enriched functions of DNA replication, repair, growth '\n",
      " 'regulation, and metabolic processes. However, endometrioma fibroblasts '\n",
      " 'comprised 4 subtypes: 1) cytokine, inflammatory response; 2) FGF '\n",
      " 'stimulation, immune response; 3) extracellular matrix (ECM) organization, '\n",
      " 'cell adhesion; 4) angiogenesis, hypoxia response, and epithelial-mesenchymal '\n",
      " 'transition (EMT), all of which were enriched in MAPK, TNF, IL-17, TGF-β '\n",
      " 'signaling pathways. Fibroblasts from eutopic endometrium of cases displayed '\n",
      " 'many pathways similar to normal tissue, but some features suggested a '\n",
      " '“transition state” from normal endometrium to endometrioma, involving TGF-β, '\n",
      " 'MAPK, Rho-Rack, NF-kB, JAK-STAT, and EMT. A developmental trajectory was '\n",
      " 'postulated based on these fibroblast features, with endometrioma fibroblasts '\n",
      " 'derived from endometrial fibroblasts 31 , much as the epithelial cell '\n",
      " 'population has been proposed, based on shared epithelial somatic mutations '\n",
      " 'in these two tissues 54 . Thus, endometrioma and paired eutopic endometrium '\n",
      " 'fibroblasts differ, and the single cell data support their key roles in '\n",
      " 'disease pathogenesis. Analysis of immune cells revealed they differed '\n",
      " 'significantly in normal endometrium, endometrioma, and paired eutopic tissue '\n",
      " '( Figure 8A ). Specifically, T cell and uNK cell frequencies were lower, uNK '\n",
      " 'cells were more active, and macrophages (Mφ) were enriched and had features '\n",
      " 'of tissue remodeling versus eutopic endometrium 31 , similar to data from '\n",
      " 'Fonseca et al 35 . Thus, in the proliferative phase, fibroblasts and immune '\n",
      " 'cell subpopulations contribute to a pro-inflammatory, angiogenic environment '\n",
      " 'in endometriomas. Moreover cell-to-cell communication/interactome analysis '\n",
      " 'revealed that immune cells stimulate fibroblast adhesion and growth in '\n",
      " 'endometriomas ( Figure 8B ), underscoring the power of single cell analyses '\n",
      " 'to inform the pathogenesis and pathophysiology of endometriomas and '\n",
      " 'cell-cell communications. Furthermore, recent observations indicated that '\n",
      " 'Fusobacterium 56 and dysbiosis of gut 57 , 58 altered systemic and '\n",
      " 'potentially local immune functions, contributing to the development of '\n",
      " 'endometriosis. Notably, while transcriptomic alterations of specific '\n",
      " 'populations and sub- populations of the endometrium, lesions, and immune '\n",
      " 'cells (local and peripheral) under these conditions awaits further study, '\n",
      " 'antibiotic and nutritional therapies for dysbiosis is emerging as a new '\n",
      " 'paradigm for treating endometriosis development and symptom management. '\n",
      " 'Figure 8. Immune cell frequencies in endometriomas and communication '\n",
      " 'networks of immune cell types with endometrioma fibroblasts. Open in a new '\n",
      " 'tab The network supports interactions between immune and fibroblast cells in '\n",
      " 'ovarian endometrioma pathogenesis and pathophysiology. From Ma et al, 2021 '\n",
      " '31 , with permission. Endometriosis peritoneal and ovary lesions of patients '\n",
      " 'on progestins As progestins are a mainstay of medical therapy for '\n",
      " 'endometriosis-related pain 12 , analysis of lesions and endometrium of '\n",
      " 'subjects taking progestins is highly valuable in understanding their effects '\n",
      " 'on cell types and tissue features. Tan et al 33 performed scRNAseq on '\n",
      " '122,000 cells from 14 individuals, selective cell localization by imaging '\n",
      " 'mass cytometry, and validations in endometrial epithelial organoid cultures. '\n",
      " 'Peritoneal and/or ovarian disease and endometrium from 19 cases (rASRM stage '\n",
      " 'II-IV) included 14 on norethindrone (NETA) + ethinyl estradiol (EthE 2 ), 2 '\n",
      " 'on drosperinone + EthE 2 , 1 on levonorgestrel + EthE 2 , 1 on only NETA, 1 '\n",
      " 'menstrual phase, and 8 controls with no endometriosis or inflammatory '\n",
      " 'conditions: 4 on NETA + EthE 2 , 2 in proliferative phase, 1 on '\n",
      " 'medroxyprogesterone acetate, 1 on norgestromin + EthE 2 . The peritoneal '\n",
      " 'lesions had similar cell compositions as eutopic endometrium but '\n",
      " 'dysregulated innate immune and vascular components - vastly different from '\n",
      " 'ovarian endometriosis lesions. The authors identified a unique perivascular '\n",
      " 'mural cell specific to the peritoneal lesions with features promoting neo- '\n",
      " 'angiogenesis and immune cell trafficking. They also identified a '\n",
      " 'progenitor-like epithelial subpopulation, different from the SOX9+LGR5+ '\n",
      " 'epithelial subpopulation found by Garcia-Alfonso et al 20 . Analysis of '\n",
      " 'peritoneal lesion myeloid populations of immunomodulatory Mφ and DCs '\n",
      " 'revealed a coordinated immunotolerant phenotype in the disease '\n",
      " 'microenvironment, promoting immunosurveillance escape and thus conducive to '\n",
      " 'lesion establishment. Overall, the data demonstrate that immune and vascular '\n",
      " 'components of peritoneal endometriosis favor neo-angiogenesis and an immune '\n",
      " 'tolerant niche in the peritoneal cavity, beneficial for lesion establishment '\n",
      " 'and growth. These are remarkable findings given that most subjects were on '\n",
      " 'hormone suppressive therapies for pain [progestin (+ estrogen)] and raises '\n",
      " 'the issue of how efficacious these therapies are on specific cells and the '\n",
      " 'role of estrogenic components in disease management. This valuable single '\n",
      " 'cell atlas of disease from patients largely on hormonal treatments offers '\n",
      " 'the opportunity for comparisons of cell types and features in peritoneal '\n",
      " 'lesions and endometriomas in the setting of other hormonal treatments and/or '\n",
      " 'in various phases of the cycle to interpret efficacies of selective '\n",
      " 'therapies on cell-specific pathways and cell-cell communications. '\n",
      " 'Adenomyosis through single cell glasses Uterine compartments relevant to '\n",
      " 'adenomyosis pathogenesis include the endometrial-myometrial junction (EMJ) '\n",
      " '[basalis and lower functionalis endometrium + sub-endometrial inner '\n",
      " 'myometrium], all of Mullerian origin and displaying cycling estrogen '\n",
      " 'receptors (ER) and progesterone receptors (PR) 59 – 61 ( Figure 9 ). Few '\n",
      " 'molecular and genomic studies of the EMJ in adenomyosis uteri have been '\n",
      " 'conducted, likely as access to basalis and inner myometrium requires '\n",
      " 'hysterectomy versus the functionalis, which can be readily obtained by '\n",
      " 'office biopsy. Figure 9. The endometrial-myometrial junction (EMJ). Open in '\n",
      " 'a new tab Panel A. Histology of human uterine endometrium and myometrium '\n",
      " 'compartments (H&E staining). Panel B : Transvaginal ultrasound midsagittal '\n",
      " 'image of uterine compartments. Panel C. Schematic of uterine compartments. '\n",
      " 'Adapted from Naftlin and Jurkovic, 2009 61 , with permission. Bulk analyses: '\n",
      " 'Studies to date in adenomyosis patients using immunohistochemistry and qPCR '\n",
      " 'reveal abnormal endometrial functionalis epithelial aromatase expression '\n",
      " '(thus high E 2 milieu), elevated stromal fibroblast IL-6, IL-8, IL-17 '\n",
      " 'secretion, TH17-Treg imbalance, altered uNK receptors, and aberrant P 4 '\n",
      " '-signaling 5 , 26 , 62 – 64 – very similar to endometrium of patients with '\n",
      " 'endometriosis except for IL-17 and TH17-Tregulatory cell imbalance. We '\n",
      " 'conducted bulk RNA seq of endometrial functionalis 65 and a study on '\n",
      " 'separate compartments of EMJ inner myometrium and endometrium from patients '\n",
      " 'with diffuse adenomyosis versus controls in the proliferative phase 66 . ECM '\n",
      " 'remodeling, inflammation, IL-17 signaling, and mTOR signaling were enriched '\n",
      " 'in endometrium, and inflammation, pain, and neuronal signaling pathways in '\n",
      " 'myometrium, consistent with clinical symptoms of this disorder. Single cell '\n",
      " 'analyses: Two recent publications have analyzed adenomyosis lesions 67 , 68 '\n",
      " '. They are important beginnings to understanding the pathogenesis and '\n",
      " 'pathophysiology of adenomyosis, building upon bulk transcriptomic data, '\n",
      " 'which could be further mined in deconvolution studies to validate new and '\n",
      " 'likely forthcoming additional single cell data. Liu et al 67 sequenced '\n",
      " '42,292 cells derived from a control endometrium of a patient with uterine '\n",
      " 'fibroids and eutopic (functionalis) endometrium and matched ectopic lesion '\n",
      " 'from a patient with adenomyosis (type and menstrual cycle phase undefined). '\n",
      " 'The data revealed seven distinct cell types. Comparison of the data from the '\n",
      " 'lesion with those from eutopic matched endometrium and control endometrium '\n",
      " 'revealed processes involved in progression of normal→endometrial→adenomyosis '\n",
      " 'epithelial phenotypes. These included motility, proliferation, angiogenesis, '\n",
      " 'inflammation, and cancer, consistent with the theory of collective cell '\n",
      " 'migration and invasion into the myometrium of endometrial-derived cells. The '\n",
      " 'novel data set revealed vascular mimicry (important in tumor formation) and '\n",
      " 'epithelial transformation to endothelial cells. It is anticipated that the '\n",
      " 'novel epithelial→endothelial transition phenotype will likely be mined in '\n",
      " 'the future and expanded studies of adenomyosis and perhaps other gynecologic '\n",
      " 'disorders and may serve as a unique cell type for drug targeting. Bulun and '\n",
      " 'colleagues 68 reported results of single cell analyses of adenomyosis '\n",
      " 'lesions and matched eutopic endometrium and myometrium from 3 subjects, all '\n",
      " 'with diffuse adenomyosis and in the proliferative phase (total of 9 samples '\n",
      " 'and 66,000 cells sequenced). They identified 11 cell types in eutopic '\n",
      " 'endometrium of patients with adenomyosis ( Figure 10 A ). Fibroblast-like '\n",
      " 'cells comprised a large fraction (36%), and pseudotime analysis revealed '\n",
      " 'they originate from pericyte progenitors that differentiate to ER+ and PR+ '\n",
      " 'endometrial stromal fibroblasts. Notably, the pseudotime trajectory was '\n",
      " 'found to end with ciliated epithelial cells, suggesting mesenchymal- '\n",
      " 'epithelial transition at play in this tissue. In adenomyosis lesions, they '\n",
      " 'identified 13 cell types including 2 (smooth muscle cells and neurons) not '\n",
      " 'found in the endometrium of the same patients ( Figure 10 B ). The '\n",
      " 'adenomyosis lesions contained even more fibroblast-like cells compared to '\n",
      " 'endometrium (50% versus 36%, respectively). In contrast to matched '\n",
      " 'endometrium, the fibroblast-like cells did not originate from pericyte '\n",
      " 'progenitors and differentiated to ECM-expressing fibroblasts and smooth '\n",
      " 'muscle cells (SMC). Also, myometrial SMC were found to derive from '\n",
      " 'fibroblast-like clusters. Moreover, the WNT inhibitor, secreted frizzled '\n",
      " 'related protein (SFRP) family expressed in fibroblast-like clusters were '\n",
      " 'similar across tissues - endometrium, adenomyosis and myometrium 68 . Figure '\n",
      " '10. Fibroblasts and epithelial cell subtypes differ in endometrium and '\n",
      " 'adenomyosis. Open in a new tab Panel A. UMAP of cell types and fibroblast '\n",
      " 'trajectories in endometrium. B. UMAP of cell types and fibroblast '\n",
      " 'trajectories in adenomyosis. Panel C. Progesterone resistance of epithelial '\n",
      " 'subtypes in adenomyosis. Panel D. WNT signaling pathway network. From Yildiz '\n",
      " 'et al, 2023 68 , with permission. Regarding the epithelial cells, P 4 '\n",
      " '-resistance was found in ciliated and unciliated cells in adenomyosis tissue '\n",
      " 'versus matched eutopic endometrium ( Figure 10 C ) – this is a key finding '\n",
      " 'regarding treatment efficiencies. Further data analysis revealed that the '\n",
      " 'canonical WNT/SFRP pathway signaling was uniquely activated in adenomyosis '\n",
      " 'unciliated and ciliated epithelium and paracrine signaling fibroblast-like '\n",
      " 'mesenchymal: epithelial interaction networks ( Figure 10D ), opening new '\n",
      " 'avenues for drug discovery for this disorder. The epithelial→endothelial '\n",
      " 'transition reported by Liu et al 67 was not detected in this study and may '\n",
      " 'be due to different adenomyosis lesion types, cycle phase, patient age, and '\n",
      " 'other variables. Whether the findings of these two studies involving '\n",
      " 'patients with diffuse adenomyosis also apply to other forms of adenomyosis ( '\n",
      " 'Figure 1 ) is unknown and is an opportunity to investigate cell features '\n",
      " 'wherein novel therapeutics could be derived. Moreover, recent data suggest '\n",
      " 'that the endometrial and vaginal microbiomes differ in patients with versus '\n",
      " 'without adenomyosis 69 , 70 , raising the possibility of adjunctive '\n",
      " 'antibiotic therapies for symptomatic patients with disease. Additionally, '\n",
      " 'further study of the local and systemic immune system cells and their '\n",
      " 'subtypes as well as in affected uterine tissues at single cell resolution '\n",
      " 'are warranted.')\n"
     ]
    }
   ],
   "source": [
    "pprint.pprint(prueba['Results'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7733fdb9",
   "metadata": {},
   "source": [
    "## Others"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a10b4d6d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Devesa-Peiro2020.pdf\n",
      "Henarejos-Castillo2020.pdf\n",
      "Sebastian-Leon2021.pdf\n",
      "Marti-Garcia2024(1).pdf\n",
      "Henarejos-Castillo2021.pdf\n",
      "Henarejos-Castillo2022.pdf\n",
      "Diaz-Gimeno2024.pdf\n",
      "Devesa-Peiro2021.pdf\n",
      "Henarejos-Castillo2024.pdf\n",
      "parraga-leo_2023.pdf\n",
      "Marti-Garcia2024.pdf\n",
      "Diaz-Gimeno2022.pdf\n",
      "Devesa-Peiro2022.pdf\n",
      "Sebastian-Leon2018.pdf\n",
      "Diaz-Gimeno2017.pdf\n"
     ]
    }
   ],
   "source": [
    "# Load several documents\n",
    "\n",
    "storage = []\n",
    "\n",
    "for root, dirs, files in os.walk(\"Data\",): # for server all the path\n",
    "\n",
    "    for file in files:\n",
    "        print(file)\n",
    "\n",
    "        loader = PyPDFLoader(os.path.join(root,file))\n",
    "        pages = loader.load_and_split()\n",
    "\n",
    "        # split it into chunks\n",
    "        text_splitter = CharacterTextSplitter(chunk_size=1000, chunk_overlap=50)\n",
    "        docs = text_splitter.split_documents(pages)\n",
    "        storage.extend(docs) # This  is a list of lists\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "4e00cd7b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load several documents\n",
    "\n",
    "storage = []\n",
    "\n",
    "for root, dirs, files in os.walk(\"Data\",): # for server all the path\n",
    "\n",
    "    for file in files:\n",
    "\n",
    "        loader = PyPDFLoader(os.path.join(root,file),mode='page')\n",
    "        pages = loader.load() # List of pages\n",
    "        \n",
    "        text_splitter = RecursiveCharacterTextSplitter(chunk_size=500, chunk_overlap=100)\n",
    "        docs = text_splitter.split_documents(pages)\n",
    "\n",
    "        storage.extend(docs) # Add several items at the same time\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "04f51224",
   "metadata": {},
   "outputs": [],
   "source": [
    "# load the document and split it into pages\n",
    "loader = PyPDFLoader(\"Publications/Parraga-Leo2023.pdf\")\n",
    "pages = loader.load_and_split()\n",
    "\n",
    "# split it into chunks\n",
    "text_splitter = RecursiveCharacterTextSplitter(chunk_size=3000, chunk_overlap=100)\n",
    "storage = text_splitter.split_documents(pages)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "fcdd0572",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('out the menstrual cycle (FDR < 0.05), dually validated in-silico and through '\n",
      " 'endometrial biopsies, corroborated their \\n'\n",
      " 'potential regulatory roles in the endometrium.\\n'\n",
      " '*Correspondence:\\n'\n",
      " 'Patricia Diaz‑Gimeno\\n'\n",
      " 'patricia.diaz@ivirma.com; patricia_diaz@iislafe.es\\n'\n",
      " 'Full list of author information is available at the end of the article')\n"
     ]
    }
   ],
   "source": [
    "pprint.pprint(storage[2].page_content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "7725a5a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "model = ChatOllama(\n",
    "    model=\"gemma3:4b\",\n",
    "    temperature=0.1,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "61f23c4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "class paper(BaseModel):\n",
    "    text: str = Field(description=\"Main text that explains the reuslts, introductions, methods or conclusions\")\n",
    "    doi: str = Field(description=\"link with the url of scientifc article known as doi.\")\n",
    "    citation: str = Field(description=\"add the citation of the manuscript is presented in the text.\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "88db094b",
   "metadata": {},
   "outputs": [],
   "source": [
    "parser = JsonOutputParser(pydantic_object=paper)\n",
    "\n",
    "prompt = PromptTemplate(\n",
    "    template=\"\"\"\n",
    "    You are an expert analyzing scientifc research papers. Read the text carefully to provide the request information.\n",
    "    Remove the potential information that is not related with the scientific article.\n",
    "\n",
    "    If it appears a link with the doi of the artcile, take it. Otherwise write down UNKOWN.\n",
    "\n",
    "    Help me to extract the information and create an json file with it. If you don't know the\n",
    "    answer, say UNKNOWN.\n",
    "    \n",
    "    Format instructions: \\n{format_instructions}\\n\n",
    "    \n",
    "    Case:\\n{query}\\n\"\"\",\n",
    "\n",
    "    input_variables=[\"query\"],\n",
    "    \n",
    "    partial_variables={\"format_instructions\": parser.get_format_instructions()},\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "8ccc261f",
   "metadata": {},
   "outputs": [],
   "source": [
    "chain = prompt | model | parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "94b83fdd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Parraga‑Leo et al. \n",
      "Reproductive Biology and Endocrinology           (2023) 21:84  \n",
      "https://doi.org/10.1186/s12958‑023‑01131‑4\n",
      "RESEARCH Open Access\n",
      "© The Author(s) 2023. Open Access This article is licensed under a Creative Commons Attribution 4.0 International License, which \n",
      "permits use, sharing, adaptation, distribution and reproduction in any medium or format, as long as you give appropriate credit to the \n",
      "original author(s) and the source, provide a link to the Creative Commons licence, and indicate if changes were made. The images or \n",
      "other third party material in this article are included in the article’s Creative Commons licence, unless indicated otherwise in a credit line \n",
      "to the material. If material is not included in the article’s Creative Commons licence and your intended use is not permitted by statutory \n",
      "regulation or exceeds the permitted use, you will need to obtain permission directly from the copyright holder. To view a copy of this \n",
      "licence, visit http://creativecommons.org/licenses/by/4.0/. The Creative Commons Public Domain Dedication waiver (http://creativecom‑\n",
      "mons.org/publicdomain/zero/1.0/) applies to the data made available in this article, unless otherwise stated in a credit line to the data.\n",
      "Reproductive Biology\n",
      "and Endocrinology\n",
      "Deciphering a shared transcriptomic \n",
      "regulation and the relative contribution of each \n",
      "regulator type through endometrial gene \n",
      "expression signatures\n",
      "Antonio Parraga‑Leo1,2, Patricia Sebastian‑Leon1, Almudena Devesa‑Peiro1,2, Diana Marti‑Garcia1,2, \n",
      "Nuria Pellicer1,3, Jose Remohi2,3, Francisco Dominguez1 and Patricia Diaz‑Gimeno1*   \n",
      "Abstract \n",
      "Backgorund While various endometrial biomarkers have been characterized at the transcriptomic and functional \n",
      "level, there is generally a poor overlap among studies, making it unclear to what extent their upstream regulators \n",
      "(e.g., ovarian hormones, transcription factors (TFs) and microRNAs (miRNAs)) realistically contribute to menstrual cycle \n",
      "progression and function. Unmasking the intricacies of the molecular interactions in the endometrium from a novel \n",
      "systemic point of view will help gain a more accurate perspective of endometrial regulation and a better explanation \n",
      "the molecular etiology of endometrial‑factor infertility.\n",
      "Methods An in-silico analysis was carried out to identify which regulators consistently target the gene biomark‑\n",
      "ers proposed in studies related to endometrial progression and implantation failure (19 gene lists/signatures were \n",
      "included). The roles of these regulators, and of genes related to progesterone and estrogens, were then analysed \n",
      "in transcriptomic datasets compiled from samples collected throughout the menstrual cycle (n = 129), and the expres‑\n",
      "sion of selected TFs were prospectively validated in an independent cohort of healthy participants (n = 19).\n",
      "Results A total of 3,608 distinct genes from the 19 gene lists were associated with endometrial progression\n",
      "and implantation failure. The lists’ regulation was significantly favoured by TFs (89% (17/19) of gene lists) and pro‑\n",
      "gesterone (47% (8 /19) of gene lists), rather than miRNAs (5% (1/19) of gene lists) or estrogen (0% (0/19) of gene \n",
      "lists), respectively (FDR < 0.05). Exceptionally, two gene lists that were previously associated with implantation failure \n",
      "and unexplained infertility were less hormone‑dependent, but primarily regulated by estrogen. Although endometrial \n",
      "progression genes were mainly targeted by hormones rather than non‑hormonal contributors (odds ratio = 91.94, \n",
      "FDR < 0.05), we identified 311 TFs and 595 miRNAs not previously associated with ovarian hormones. We highlight \n",
      "CTCF, GATA6, hsa‑miR‑15a‑5p, hsa‑miR‑218‑5p, hsa‑miR‑107, hsa‑miR‑103a‑3p, and hsa‑miR‑128‑3p, as overlapping \n",
      "novel master regulators of endometrial function. The gene expression changes of selected regulators through‑\n",
      "out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their\n",
      "out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their \n",
      "potential regulatory roles in the endometrium.\n",
      "*Correspondence:\n",
      "Patricia Diaz‑Gimeno\n",
      "patricia.diaz@ivirma.com; patricia_diaz@iislafe.es\n",
      "Full list of author information is available at the end of the article\n",
      "Page 2 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "Conclusions This study revealed novel hormonal and non‑hormonal regulators and their relative contributions \n",
      "to endometrial progression and pathology, providing new leads for the potential causes of endometrial‑factor \n",
      "infertility.\n",
      "Keywords CTCF, GATA6, Progesterone, Estrogen, miRNAs, TFs, Endometrial receptivity, Recurrent implantation failure, \n",
      "Infertility, Menstrual cycle regulation\n",
      "Background\n",
      "The endometrium is the innermost layer of the uterus, \n",
      "which undergoes dynamic histological, physiological \n",
      "and molecular changes that allow it to synchronize with \n",
      "embryo development, facilitate embryo implantation, \n",
      "and ultimately, establish a successful pregnancy [1].\n",
      "Implantation is a crucial and complex limiting step \n",
      "for conception [2] which occurs between days 19 to 24 \n",
      "of a normal menstrual cycle. Consequently, this period \n",
      "of time is called the window of implantation (WOI) [3], \n",
      "and is characterized by abrupt transcriptomic changes \n",
      "in the endometrial tissue [4]. Alterations in either WOI \n",
      "establishment and endometrial progression may lead to \n",
      "implantation failures, which, along with biochemical mis-\n",
      "carriage, account for more than 50% of pregnancy losses \n",
      "at pre-clinical stages [5, 6]. These alterations have broadly \n",
      "been classified as being related to a displaced WOI, \n",
      "caused by variable timing of endometrial progression, or \n",
      "a disrupted WOI, where impaired endometrial function \n",
      "prevents the establishment of an effective WOI [7].\n",
      "Initially, the WOI was postulated to mainly be regu -\n",
      "lated by the ovarian hormones (namely, estrogen and \n",
      "progesterone) acting via their respective nuclear recep -\n",
      "tors (i.e., ESR1/2 and PGR). However, evidence has shown \n",
      "that these receptors cooperate with other transcription \n",
      "factors (TFs) and co-regulators to mediate uterine physi -\n",
      "ology [8]. In fact, TFs (such as homeobox TFs, FOXA2, \n",
      "and KLF9) have recently been reported as key regulators \n",
      "for the establishment of endometrial receptivity [9, 10], \n",
      "and microRNAs (miRNAs) have been found to similarly \n",
      "intervene in this complex and multifactorial process as \n",
      "transcriptional regulators [11]. While the involvement of \n",
      "both types of regulators was widely reported in embryo \n",
      "implantation [12] and other infertility-related diseases, \n",
      "including recurrent implantation failure (RIF) [13], the \n",
      "complex interactions between the different types of \n",
      "regulators and their respective contributions to endo -\n",
      "metrial progression and function have not been investi -\n",
      "gated from a systemic point of view. This paradigm shift \n",
      "approaches genes and regulators in an integrative way, by \n",
      "considering how they interact with and coordinate each \n",
      "other to carry out cellular processes, rather than rein -\n",
      "forcing the premise they work independently [14]. By \n",
      "more accurately reflecting the biological milieu, systemic \n",
      "approaches take gene-based discoveries to the next level,\n",
      "helping to generate hypotheses with relevant clinical and \n",
      "molecular implications [15, 16].\n",
      "Previous studies of endometrial transcriptomics have \n",
      "been used to define the WOI, compare the endometrium \n",
      "of healthy women as their menstrual cycles progressed \n",
      "[4, 17–19]; address endometrial differences related to age \n",
      "[20] or the dysfunction of the WOI between patients with \n",
      "implantation failure and healthy controls [21]; and finally, \n",
      "correct for the menstrual cycle bias that can mask impor-\n",
      "tant biomarkers [22]. Notwithstanding, the candidate \n",
      "biomarkers of endometrial receptivity reported in each \n",
      "of the aforementioned studies overlap poorly [7]. Indeed, \n",
      "recent comparisons conducted by Sebastian-Leon and \n",
      "colleagues [7] found no congruities between 16 different \n",
      "reported gene lists of endometrial receptivity, and only \n",
      "fair to moderate functional agreements between some of \n",
      "the included signatures. Despite previous transcriptional \n",
      "regulation studies, there have been no holistic studies of\n",
      "fair to moderate functional agreements between some of \n",
      "the included signatures. Despite previous transcriptional \n",
      "regulation studies, there have been no holistic studies of \n",
      "this regulatory process, which robustly analyse the driv -\n",
      "ers of transcription across a higher number of studies \n",
      "(that each searched for endometrial receptivity biomark -\n",
      "ers) and map the relative contributions of each type of \n",
      "regulator to endometrial progression regulation (Ovarian \n",
      "hormones [i.e., estrogen, progesterone], TFs, miRNAs).\n",
      "Thus, the aim of this study was to utilize available \n",
      "genomic data regarding transcriptional regulators to \n",
      "identify overlapping mediators of endometrial transcrip -\n",
      "tional regulation (i.e., ovarian hormones, TFs, and miR -\n",
      "NAs) and determine the relative contribution of each \n",
      "type of regulator among the previously reported gene \n",
      "signatures. These findings could help unveil the master \n",
      "regulators and principal type of endometrial regulation, \n",
      "paving the way for further research aimed at improving \n",
      "woman´s reproductive health.\n",
      "Methods\n",
      "A detailed study design is depicted in Supplementary \n",
      "Fig. S1.\n",
      "Annotating gene lists associated with endometrial \n",
      "progression and function\n",
      "Endometrial progression and implantation failure gene \n",
      "lists [7] were retrieved and updated using a public data \n",
      "repository (i.e., Gene Expression Omnibus (GEO)).\n",
      "Page 3 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Keywords used for the GEO search included: endometrial \n",
      "receptivity, mid-secretory endometrium, RIF, recurrent \n",
      "implantation failure, endometrium, unexplained infer -\n",
      "tility, and implantation failure. The resulting datasets \n",
      "were filtered by their publication date (i.e., from January \n",
      "2018 to October 2020, to expand the systematic search of \n",
      "[7], number of samples (> 3 samples for each condition), \n",
      "and species (i.e., Homo sapiens), with no restrictions on \n",
      "publication language. The original signatures were then \n",
      "retrieved from their corresponding publications. The \n",
      "genes prioritized in the original publication were exclu -\n",
      "sively selected and their names were annotated with \n",
      "HUGO Gene Nomenclature Committee (HGNC) gene \n",
      "name using biomaRt R-package v.3.10 [23]. Each signa -\n",
      "ture was labelled with the name of first author of the cor -\n",
      "responding study. Our annotated gene lists were selected \n",
      "as representative signatures of endometrial progression \n",
      "and function for subsequent analyses (Supplementary \n",
      "Fig. S1A).\n",
      "Identifying hormonal and non‑hormonal gene regulators \n",
      "of endometrial progression and function\n",
      "To study the hormonal regulation of endometrial \n",
      "progression, the Kyoto Encyclopaedia of Genes and \n",
      "Genomes (KEGG) [24], and Gene Ontology (GO) [25] \n",
      "databases were consulted for ovarian hormone-related \n",
      "genes, using relevant keywords (i.e., progesterone, estro-\n",
      "gen, oestrogen and estradiol) and selecting those pathways \n",
      "and functions that contained them (Supplementary Fig. \n",
      "S1B). Genes associated with the obtained human KEGG \n",
      "pathways and GO functions were grouped according to \n",
      "whether they were unique to progesterone (P4 gene set) \n",
      "or estrogen (E2 gene set) or related to both hormones \n",
      "(P4 and E2 gene set). Gene targets of the nuclear proges -\n",
      "terone (PGR) and estrogen (ESR1, ESR2) receptors were \n",
      "added to their corresponding gene sets using DoRothEA \n",
      "(Discriminant Regulon Expression Analysis) database \n",
      "[26], considering only manually-curated or ChiP-Seq \n",
      "experimentally-validated gene-TF relationships. Finally, \n",
      "the P4 and E2 gene sets were independently mapped to \n",
      "each gene list.\n",
      "Meanwhile, to evaluate the non-hormonal regula -\n",
      "tors of endometrial progression and function, DoRo -\n",
      "thEA [26] and TarBase [27] databases were consulted to \n",
      "obtain TFs and miRNAs. DoRothEA was filtered as pre -\n",
      "viously described herein, while TarBase, which only con -\n",
      "tains miRNA-gene target relationships manually curated \n",
      "from publications or experimentally-validated in high-\n",
      "throughput datasets [27], was filtered by species (i.e., \n",
      "human). Finally, a functional over-representation analysis \n",
      "was carried out to identify which of the total annotated \n",
      "TFs or miRNAs were significantly associated with a par -\n",
      "ticular gene list (Supplementary Fig. S1B).\n",
      "To evaluate the relative contribution of each type of \n",
      "regulator (i.e., P4, E2, TFs, and miRNAs) on endometrial\n",
      "progression and function, three different comparisons \n",
      "were carried out applying Fisher’s exact tests. First, hor -\n",
      "monal regulators were evaluated using the relative pro -\n",
      "portion of P4- and E2-related genes included in each list, \n",
      "with respect to the total of number of genes founded in \n",
      "the P4 or E2 gene sets. Then, non-hormonal regulators \n",
      "were evaluated using the relative proportion of over-\n",
      "represented miRNAs and TFs with respect to the total of \n",
      "miRNAs and TFs with a at least one target in the gene \n",
      "list. Finally, the relative proportion of genes under hor -\n",
      "monal (i.e., P4 and/or E2-annotated genes) versus non-\n",
      "hormonal (i.e., miRNA and/or TF-annotated genes) \n",
      "regulation was evaluated considering the miRNAs/TFs \n",
      "found within the individual P4 or E2 gene sets as hormo -\n",
      "nal regulators.\n",
      "Prioritizing key regulators in endometrial progression \n",
      "and function\n",
      "We built regulatory networks to systemically analyse \n",
      "endometrial regulation. Nodes represented gene lists\n",
      "nal regulators.\n",
      "Prioritizing key regulators in endometrial progression \n",
      "and function\n",
      "We built regulatory networks to systemically analyse \n",
      "endometrial regulation. Nodes represented gene lists \n",
      "and regulators (miRNAs or TFs), while edges indicated \n",
      "significant enrichment among the lists and their corre -\n",
      "sponding regulators (false discovery rate (FDR) ≤ 0.05). \n",
      "To select the most influential regulators (i.e., those which \n",
      "targeted genes in most of our signatures), we studied the \n",
      "degree distribution of the networks (i.e., the number of \n",
      "gene lists regulated by each molecule) and prioritized the \n",
      "miRNAs and TFs which surpassed the relative maximum \n",
      "(i.e., 1.50 times the interquartile range (IQR)) number of \n",
      "relationships (gene list–regulators). All networks were \n",
      "built and analysed with Cytoscape version 3.7 [28] (Sup -\n",
      "plementary Fig. S1B).\n",
      "Dataset construction and processing for in‑silico validation\n",
      "To evaluate how the miRNAs’ expression change in the \n",
      "endometrium throughout the menstrual cycle, we used \n",
      "the raw data from GSE44558 GEO dataset [29], deriving \n",
      "from 20 endometrial samples collected throughout the \n",
      "menstrual cycle: four in early-proliferative (EPF), four \n",
      "in late-proliferative (LPF), four in early-secretory (ESE), \n",
      "four in mid-secretory (MSE) and four in late-secretory \n",
      "(LSE).\n",
      "To evaluate how the TFs’ expression change in the \n",
      "endometrium throughout the menstrual cycle, we \n",
      "analysed the integrated endometrial dataset previ -\n",
      "ously created by our group, and reported in [30], which \n",
      "compiled data from five prior publications (GSE98386, \n",
      "GSE29981, GSE4888, GSE119209 and GSE86491). \n",
      "This dataset included endometrial gene expression\n",
      "Page 4 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "data from 109 participants with normal endometrium, \n",
      "where biopsies were collected in proliferative (PF) \n",
      "(n = 29), ESE (n  = 29), MSE (n  = 43) and LSE (n  = 8) \n",
      "phases of the menstrual cycle (Table S1 ).\n",
      "Both TF and miRNA datasets were processed using \n",
      "the limma R-package [31]. Expression data were log \n",
      "transformed and quantile normalized, prior to explora -\n",
      "tory analyses which sought out possible outliers and \n",
      "batch effects. Relative gene expression ranges of low \n",
      "(1–10%), medium (11–50%), and high (51–100%) were \n",
      "established based on the expression of all the genes \n",
      "included in our dataset.\n",
      "Wet lab validation cohorts\n",
      "To corroborate the menstrual cycle-related expression \n",
      "changes observed with CTCF and GATA6 in our in-\n",
      "silico analysis, we conducted a prospective study as an \n",
      "external validation, using an independent sample set. \n",
      "We included a cohort of 19 healthy Spanish women \n",
      "(obtaining a total of 20 biopsies), between the ages of \n",
      "22–35 and with a body mass index of 22.80 ± 2.76  kg/\n",
      "m2. Endometrial samples were collected and staged in \n",
      "the menstrual cycle according to the follicle growth to \n",
      "control ovulation and LH levels in urine. Samples were \n",
      "grouped into PF (n  = 5), ESE (n  = 5), MSE (n  = 5) and \n",
      "LSE (n = 5) phases and categorized into two groups, PF/\n",
      "ESE and MSE/LSE.\n",
      "RNA extraction and RT‑qPCR\n",
      "Total RNA was extracted from the endometrial biopsy \n",
      "samples using the miRNeasy Mini Kit (Qiagen, Ger -\n",
      "many), and reverse transcribed into cDNA using the \n",
      "PrimeScript reagent kit (TAKARA, Japan). RT-qPCR \n",
      "reactions were carried out in duplicate, using fluores -\n",
      "cent Power-up SYBR Green (Thermo Fisher Scientific \n",
      "MA, USA) in a final volume of 10 µL. Primer sequences \n",
      "are shown in Table S2. Samples underwent 40 cycles of \n",
      "amplification, under standard conditions, using a Ste -\n",
      "pOnePlus™ System (Applied Biosystems, MA, USA). \n",
      "Relative mRNA expression was calculated using the  2−\n",
      "ΔΔCt method [32], and normalized to the expression of \n",
      "GAPDH housekeeping gene.\n",
      "Statistical analysis\n",
      "Fisher’s exact test was used to evaluate independent \n",
      "proportions. Mean expression changes across the cycle \n",
      "were studied using an analysis of variance (ANOVA) fol -\n",
      "lowed by a pairwise t-test or the Wilcoxon test when only \n",
      "two groups were available. P-values from multiple test \n",
      "comparisons were corrected by FDR and considered as \n",
      "significant when FDR ≤ 0.05. All statistical analyses were \n",
      "performed through R software (version 3.5) [33].\n",
      "Results\n",
      "Gene signatures associated to endometrial progression \n",
      "and function\n",
      "Of the 19 gene lists used for our analysis, eleven were \n",
      "obtained from published studies evaluating control \n",
      "patients throughout the menstrual cycle [4, 17–19, \n",
      "34–40], and the other eight were derived from studies \n",
      "comparing patients with RIF or unexplained infertility \n",
      "to controls [21,  41–47] (Table 1). Unifying all the afore -\n",
      "mentioned signatures, we compiled 3,608 genes related \n",
      "to endometrial progression and function.\n",
      "Hormonal regulation of endometrial progression is largely \n",
      "driven by progesterone\n",
      "We identified 7,540 and 698 genes related to estrogen \n",
      "and progesterone hormones, respectively. However, as \n",
      "determined by the relative contribution of each type of \n",
      "hormone within each gene list (Fig.  1A), 17/19 (89%) \n",
      "signatures favoured regulation by progesterone rather \n",
      "than estrogen. These differences were significant in \n",
      "47% of the signatures (FDR < 0.05). Exceptionally, the \n",
      "genes reported by the Altmae2010 and Koot signa -\n",
      "tures, that were related to unexplained infertility and \n",
      "RIF, had a preference for estrogen regulation (Fig.  1A). \n",
      "With respect to the proportion of progesterone-related \n",
      "genes, Altmae2010 showed 2.1 times more estrogen-\n",
      "related genes while Koot showed 1.7 times more. More -\n",
      "over, comparing the proportion of genes regulated by \n",
      "both hormones, between signatures revealed that Alt -\n",
      "related genes while Koot showed 1.7 times more. More -\n",
      "over, comparing the proportion of genes regulated by \n",
      "both hormones, between signatures revealed that Alt -\n",
      "mae2010 and Koot exhibited a lower hormone-depend -\n",
      "ent behaviour overall, with respect to the remaining \n",
      "signatures (Fig.  1B), highlighting the role of non-hor -\n",
      "monal regulators.\n",
      "Non‑hormonal regulation of endometrial progression \n",
      "and function\n",
      "We identified a total of 770 miRNAs and 555 TFs regu -\n",
      "lating at least one of the 3,608 genes reported in the \n",
      "19 gene lists. Further, we observed that 29% of the \n",
      "miRNAs and 44% of the TFs were associated with the \n",
      "ovarian hormones (i.e., progesterone and/or estrogen). \n",
      "Significantly more genes were targeted by hormo -\n",
      "nally-influenced regulators rather than non-hormonal \n",
      "regulators (P -value < 0.01, odds ratio =  91.94). Never -\n",
      "theless, we also identified 595 miRNAs (77.27%) and \n",
      "311 TFs (56.04%) that were unrelated to progesterone \n",
      "and/or estrogen, highlighting a substantial proportion \n",
      "of hormonally-independent regulators of the men -\n",
      "strual cycle.\n",
      "Page 5 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Table 1 Characteristics of the 19 gene lists included in this study. Gene lists were divided according to their evaluated condition. Each \n",
      "signature was labelled using the name of first author, followed by the publication year in the case of duplicates. Endometrial staging \n",
      "was based on different variables including urinary LH, histological analysis, ultrasound evaluation, or the number of days since the hCG \n",
      "trigger. Participants included healthy controls and women with infertility‑related conditions. Various platforms were used to evaluate \n",
      "gene expression. The thresholds applied for gene refinement and the final number of genes per signature are also presented\n",
      "Abbreviations: Δ delta  Ct, adj-P-value Adjusted P‑value, C Control group, d Day of menstrual cycle, EPF Early proliferative, ESE Early secretory, |FC| Absolute fold‑change, \n",
      "FC Fold change, HRT Hormone replacement therapy, IF Implantation failure, LH Luteinizing hormone, LH + X X days post LH surge, LPF Late proliferative, LSE Late‑\n",
      "secretory, M Miscarriage, MPF Mid‑proliferative, MSE Mid‑secretory, n number of samples, N/A Not applicable, OP Ongoing pregnancy, PF Proliferative, PFP Proportion \n",
      "of false positives, PR Pre‑receptive, R Receptive, RIF Recurrent implantation failure, RRA  Robust Rank Aggregation algorithm, SF Spontaneously fertile women, UI \n",
      "Women with unexplained infertility, vs versus\n",
      "Condition Gene signature Endometrial staging Participants Platform Threshold No. genes References\n",
      "Endometrial progres‑\n",
      "sion\n",
      "Altmäe2017 N/A N/A N/A RRA 57 [34]\n",
      "Borthwick Urinary LH; Histologi‑\n",
      "cally confirmed\n",
      "PF d9‑11 (n = 5) vs R \n",
      "LH + 6–8 (n = 5)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "N/A 116 [18]\n",
      "Carrascosa N/A N/A Quantitative RT‑PCR Based on literature 187 [35]\n",
      "Carson Urinary LH; Histologi‑\n",
      "cally confirmed\n",
      "ESE LH + 2–4 (n = \n",
      "3) vs MSE LH + 7–9 \n",
      "(n = 3)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "|FC|> 2 695 [17]\n",
      "Diaz‑Gimeno Urinary LH PR LH + 1–5 (n = \n",
      "15) vs R\n",
      "Agilent custom gene \n",
      "expression microarray\n",
      "|FC| > 2 234 [36]\n",
      "Kao Urinary LH; Histologi‑\n",
      "cally confirmed\n",
      "LPF d8‑10 (n = 4) vs \n",
      "MSE LH + 8‑10 (n = 7)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "|FC|> 2 and P‑value \n",
      "<0.05\n",
      "340 [37]\n",
      "Mirkin Urinary LH; Histologi‑\n",
      "cal dating\n",
      "ESE LH + 3 (n = 3) vs \n",
      "MSE LH + 8 (n = 5)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "|FC|>2 and adj‑P‑\n",
      "value < 0.05\n",
      "105 [19]\n",
      "Ponnampalam Histological dating EPF (n = 5) MPF (n = 7) \n",
      "LPF (n = 3) ESE (n = 7) \n",
      "MSE (n = 8) LS (n = 7) \n",
      "Mense (n = 6)\n",
      "Custom Adj‑P‑value <0.05 306 [38]\n",
      "Punyadeera Histological dating Mense (n = 2) vs LPF \n",
      "(n=2)\n",
      "Affymetrix HG‑U133A FC and P‑value 50 [39]\n",
      "Riesewijk Urinary LH; Histologi‑\n",
      "cal dating\n",
      "PR LH + 2 (n = 5) vs R \n",
      "LH + 7 (n = 5)\n",
      "Affymetrix Genechip \n",
      "Hu95A\n",
      "|FC|> 3 in at least four \n",
      "out of five women.\n",
      "196 [40]\n",
      "Talbi Histological dating P (n = 6) vs ESE (n = 3) \n",
      "vs MSE (n = 8) vs LSE \n",
      "(n = 6)\n",
      "Affymetrix HG‑U133 \n",
      "Plus 2.0\n",
      "|FC|> 1.5 and adj‑P‑\n",
      "value < 0.05\n",
      "317 [4]\n",
      "|FC|> 1.5 and adj‑P‑\n",
      "value < 0.05\n",
      "317 [4]\n",
      "Implantation failure Altmäe2010 Urinary LH UI (n = 4) vs C (n = 5) Whole Human \n",
      "Genome Oligo \n",
      "Microarray (Agilent \n",
      "Technologies) \n",
      "|FC|> 2 and PFP < 0.05 184 [21]\n",
      "Bastu 6‑7 days after LH \n",
      "surge\n",
      "IF (n = 24) vs Fertile \n",
      "women (n = 24)\n",
      "Agilent‑039494\n",
      "SurePrint G3 Human\n",
      "GE v2 8x60K\n",
      "Log2(FC) ≥ 1 and adj‑\n",
      "P‑value ≤0.05 |FC|> 2 \n",
      "and PFP < 0.05\n",
      "524 [41]\n",
      "Bersinger N/A IF (n = 3) vs M (n = 3) \n",
      "vs OP (n = 3)\n",
      "Affymetrix Human \n",
      "Genome U133A 2.0 \n",
      "Chips\n",
      "Not specified 93 [42]\n",
      "Bhagwat N/A PR vs R Multiple platforms Based on literature 177 [43]\n",
      "Koot Urinary LH RIF (n = 43) vs C (n \n",
      "= 72)\n",
      "Human whole \n",
      "genome gene expres‑\n",
      "sion microarrays V2 \n",
      "(Agilent, Belgium)\n",
      "Machine learning \n",
      "predictor\n",
      "310 [44]\n",
      "Lédée Ultrasound evaluation IF (n = 30) vs Fertile \n",
      "women (n = 15)\n",
      "Affymetrix Human\n",
      "Genome U133 Plus \n",
      "2.0 Array\n",
      "Adj‑P‑value <0.01 \n",
      "and Δ > 1.35\n",
      "322 [45]\n",
      "Pathare 6‑7 days after hCG \n",
      "administration\n",
      "IF (n = 10) vs Healthy \n",
      "oocyte donors (n = 8)\n",
      "Illumina HumanHT‑12 \n",
      "V4.0 expression \n",
      "beadchip\n",
      "|FC|> 2 818 [46]\n",
      "Shi Urinary LH; Ultrasound \n",
      "evaluation\n",
      "Pathare 6‑7 days after hCG \n",
      "administration\n",
      "IF (n = 10) vs Healthy \n",
      "oocyte donors (n = 8)\n",
      "Illumina HumanHT‑12 \n",
      "V4.0 expression \n",
      "beadchip\n",
      "|FC|> 2 818 [46]\n",
      "Shi Urinary LH; Ultrasound \n",
      "evaluation\n",
      "RIF (n = 12) vs C (n \n",
      "= 10)\n",
      "Agilent‑052909\n",
      "CBC lncRNAmRNA V3\n",
      "|FC|≥ 2 and adj‑P‑\n",
      "value ≤ 0.05\n",
      "281 [47]\n",
      "Page 6 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "Of the respective miRNAs and TFs targeting genes \n",
      "related to endometrial progression and function, \n",
      "417 miRNAs and 467 TFs were significantly over-  \n",
      "represented in at least one gene list (FDR < 0.05, Tables \n",
      "S3-S4). Further, most signatures were significantly \n",
      "governed by TFs rather than miRNAs (4 > odds \n",
      "ratio > 164, FDR < 0.05, 17 lists [89%]), with the excep -\n",
      "tion of Koot’s signature that was governed by miRNAs \n",
      "and showed the opposite (odds ratio = 0.35, FDR < 0.05, \n",
      "Table S5).\n",
      "Fig. 1 Estrogen and progesterone‑mediated regulation of endometrial progression and function. A Eight out of nineteen gene lists (47%) \n",
      "had significant differences between the proportion of genes regulated by progesterone versus estrogen. (*FDR ≤ 0.05). B Descriptive analysis \n",
      "of hormonal regulation, normalized by the size of each gene list. Gene lists clustered according to higher (blue) or lower (orange) contribution \n",
      "of hormonal regulation\n",
      "Page 7 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Hsa‑miR‑15a‑5p, hsa‑miR‑128‑3p, hsa‑miR‑218‑5p, \n",
      "hsa‑miR‑27b‑3p, hsa‑miR‑107, hsa‑miR‑424‑5p, \n",
      "hsa‑miR‑195‑5p, hsa‑miR‑103a‑3p, has‑let‑7b‑5p \n",
      "and hsa‑miR‑22‑3p are key players in the regulation \n",
      "of endometrial progression and function\n",
      "To bridge the gap of endometrial-based evidence in \n",
      "DoRothEA and TarBase databases, we aimed to iden -\n",
      "tify the common miRNAs that mediate gene expres -\n",
      "sion related to endometrial progression and function. \n",
      "We evaluated the number of endometrial gene lists \n",
      "that each miRNA controlled (Fig.  2A). This strategy \n",
      "highlighted the most influential miRNAs that were \n",
      "commonly found across the endometrial gene lists. \n",
      "Notably, 85.80% of miRNAs regulated genes from one \n",
      "to three signatures, and only 59 miRNAs (14.20%) were \n",
      "associated with more than three signatures.  Ultimately, \n",
      "24 miRNAs were prioritized for regulating genes of at \n",
      "least six signatures.\n",
      "Next, we validated the gene expression trends for \n",
      "these 24 prioritized miRNAs in an independent endo -\n",
      "metrial gene expression dataset (GSE44730) (Fig.  2B). \n",
      "This dataset included the miRNA expression in endo -\n",
      "metrial biopsies from 20 healthy women in differ -\n",
      "ent phases of the menstrual cycle. We found that \n",
      "two of our prioritized miRNAs were not expressed \n",
      "in the endometrial tissue, and only 10 miRNAs (i.e., \n",
      "hsa-miR-15a-5p, hsa-miR-128-3p, hsa-miR-218-5p, \n",
      "hsa-miR-27b-3p, hsa-miR-107, hsa-miR-424-5p, hsa-\n",
      "miR-195-5p, hsa-miR-103a-3p, has-let-7b-5p and hsa-\n",
      "miR-22-3p) showed significant changes across the \n",
      "menstrual cycle (Fig.  2B). Of these, only the expression \n",
      "of hsa-miR-22-3p significantly increased from ESE \n",
      "to MSE stages (FDR = 0.02, Fig.  2B), highlighting the \n",
      "potential inhibitory role of this miRNA in endometrial \n",
      "function. The expression of the remaining miRNAs \n",
      "decreased during the MSE phase.\n",
      "CTCF and GATA6 as distinguished transcriptional \n",
      "regulators of the human menstrual cycle\n",
      "We repeated the regulation network analysis with the \n",
      "TFs, to identify the prominent TFs commonly regu -\n",
      "lating the genes related to endometrial progression \n",
      "and function in the reported signatures. We observed \n",
      "78.80% of the TFs regulated genes in up to 10 signa -\n",
      "tures, and only 99 TFs (21.20%) regulated genes in over \n",
      "10 (Fig.  3A). We highlighted CTCF (a CCCTC-binding \n",
      "factor, which functions as a transcriptomic repressor) \n",
      "as the TF with the most influential regulation of endo -\n",
      "metrial progression and function across 95% of studies \n",
      "(18 gene lists, excluding that of Punyadeera, Fig.  3A). \n",
      "Other distinguished TFs that regulated up to 15 gene \n",
      "lists included AR, CEBPA, CEBPB, CEBPD, CREB1, \n",
      "EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, \n",
      "GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, \n",
      "NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, \n",
      "TCF7L2, TFAP2C and TP53. Notably, GATA6 was \n",
      "selected due to novel association with endometrial pro -\n",
      "gression and function.\n",
      "gression and function.\n",
      "To infer the expression of CTCF and GATA6  in \n",
      "endometrial tissue throughout the menstrual cycle, \n",
      "we analysed their expression using a relevant endo -\n",
      "metrial dataset recently created by our group [30] \n",
      "(Table S1 ). We observed that, relative to the global \n",
      "expression of endometrial genes, CTCF was broadly \n",
      "expressed while GATA6  was moderately expressed. \n",
      "Indeed, CTCF was significantly down-regulated \n",
      "between the PF and ESE phases (FDR = 2.10E-03)  \n",
      "and ESE to the MSE phase (FDR = 2.10E-04), while \n",
      "GATA6 expression increased from the PF to the \n",
      "MSE phase (FDR = 1.70E-11) and from the ESE  \n",
      "to MSE phase (FDR = 9.80E-15). These findings \n",
      "supported the role of these TFs at the beginning \n",
      "of the menstrual cycle and during the MSE phase \n",
      "(Fig. 3 B).\n",
      "These in-silico  validations were then prospec -\n",
      "tively corroborated by RT-qPCR analysis in an inde -\n",
      "pendent sample set of 20 endometrial biopsies. We \n",
      "observed a similar expression profile of CTCF and \n",
      "GATA6 throughout the menstrual cycle, with signifi -\n",
      "pendent sample set of 20 endometrial biopsies. We \n",
      "observed a similar expression profile of CTCF and \n",
      "GATA6 throughout the menstrual cycle, with signifi -\n",
      "cant changes between the endometrial phases. Nota -\n",
      "bly, CTCF expression was decreased between the PF/\n",
      "ESE and MSE/LSE phases (FDR = 0.01) while GATA6 \n",
      "expression increased from the PF/ESE to MSE/LSE \n",
      "phases (FDR = 1.55E-04) (Fig. 3 C).\n",
      "Fig. 2 MicroRNAs universally regulating endometrial progression and function and their expression throughout the menstrual cycle. A Regulatory \n",
      "network of miRNAs forming relationships with numerous gene lists (2–12); miRNAs targeting specific gene lists were placed in the centre \n",
      "of the network. The prioritized miRNAs (outlined in red) included has‑miR‑16‑5p, has‑miR‑138‑5p, has‑miR‑21‑3p, has‑miR‑205‑5p, has‑miR‑27a‑5p, \n",
      "has‑miR‑15a‑5p, has‑miR‑155‑5p, has‑miR‑129–2‑3p, has‑miR‑147a, has‑miR‑146a‑5p, has‑miR‑107, has‑miR‑424‑5p, has‑miR‑203a‑3p, \n",
      "has‑miR‑195‑5p, has‑miR‑124‑3p, has‑miR‑128‑3p, has‑miR‑22‑3p, has‑miR‑27b‑3p, has‑let‑7b‑5p, has‑miR‑1343‑3p, has‑miR‑1‑3p, has‑miR‑103a‑3p, \n",
      "has‑miR‑23b‑3p, has‑miR‑218‑5p. B Expression profiles of the 10 prioritized miRNAs at different phases of the menstrual cycle. Significant changes \n",
      "between phases are denoted with asterisks (* FDR ≤ 0.05, ** FDR ≤ 0.01. Abbreviations: EPF, early proliferative phase; LPF, late proliferative phase; ESE, \n",
      "early‑secretory phase; MSE, mid‑secretory phase; LSE, late‑secretory phase\n",
      "(See figure on next page.)\n",
      "Page 8 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "Fig. 2 (See legend on previous page.)\n",
      "Page 9 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Fig. 3 Transcription factors universally regulating endometrial progression and function and their expression throughout the menstrual \n",
      "cycle. A Regulatory network of TFs forming relationships with numerous gene lists (2–18). TFs targeting specific gene lists were placed \n",
      "in the center of the network. CTCF stood out as the only TF that targeted genes belonging to 18/19 (95%) different signatures. B In-silico analysis \n",
      "of CTCF and GATA6 expression throughout the menstrual cycle. Significant changes between phases are marked with asterisks (**FDR ≤ 0.01, \n",
      "***FDR ≤ 0.001). C RT‑qPCR validation. Bar plot shows the relative mRNA expression of CTCF and GATA6 in endometrial biopsies collected \n",
      "throughout the menstrual cycle. Data is presented as a mean ± SD. Significant differences of CTCF (*FDR < 0.05) and GATA6 (***FDR < 0.001) \n",
      "expression were observed between proliferative/early‑secretory (PF/ESE) and mid‑secretory/late‑secretory phases (MSE/LSE). Abbreviations in (B \n",
      "and C): PF, Proliferative; ESE, early‑secretory; MSE, mid‑secretory; LSE, late‑secretory; SD, standard deviation\n",
      "Page 10 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "Discussion\n",
      "This data-driven approach exposed the common tran -\n",
      "scriptional regulators among 19 studies who proposed \n",
      "variable biomarkers of endometrial progression and \n",
      "function. In this study, we focused on understanding \n",
      "the relative contribution of both the hormonal and non-\n",
      "hormonal regulation, from an alternative holistic per -\n",
      "spective. We applied data-driven hypothesis research \n",
      "that, unlike the traditional scientific method, allowed us \n",
      "to generate new hypotheses based on all available bio -\n",
      "logically-relevant knowledge [16], observing the molec -\n",
      "ular relationships from a wider scale view. Besides, \n",
      "some traditional molecular procedures such as PCRs \n",
      "are also performed to corroborate the insight uncovered \n",
      "with this approach. We highlighted a larger influence \n",
      "of progesterone-related genes and TFs in endome -\n",
      "trial progression and observed a larger contribution \n",
      "of miRNAs and estrogen-related genes in endometrial \n",
      "pathology. Furthermore, we unveiled CTCF, GATA6, \n",
      "hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-\n",
      "miR-103a-3p and hsa-miR-128-3p as master regulators \n",
      "in the endometrium, and these findings were reinforced \n",
      "by the dynamic changes in their expression throughout \n",
      "the menstrual cycle (dually validated with in-silico  and \n",
      "prospective analyses).\n",
      "The menstrual cycle is perceived to be under tight \n",
      "regulation by the ovarian hormones, progesterone \n",
      "and estrogen [48]. Supporting the role of progester -\n",
      "one in the secretory endometrium [49], and reinforc -\n",
      "ing its indispensable role in endometrial regulation \n",
      "[50, 51], we found progesterone and its related genes \n",
      "had a larger influence in menstrual cycle progression \n",
      "than estrogen. As the RIF-related gene lists reported \n",
      "by Altmae2010 and Koot were generally controlled by \n",
      "estrogen, this further suggested that while progester -\n",
      "one could be crucial for a suitable endometrial progres -\n",
      "sion and mid-secretory phase acquisition, estrogenic \n",
      "irregularities may be more conducive to endometrial \n",
      "function and molecular pathology. Indeed, this hypoth -\n",
      "esis corroborated previous reports of most uterine dis -\n",
      "orders being estrogen-dependent [44, 52]. Supporting \n",
      "the findings from another recent study from our group, \n",
      "where the Altmae2010 and Koot signatures were classi -\n",
      "fied as highly-predictive signatures to identify endome -\n",
      "trial pathology in comparison with a prediction model \n",
      "that detected endometrial progression [7 ]. This study \n",
      "distinguished these signatures (related to implantation \n",
      "failure and unexplained infertility) based on the mag -\n",
      "nitude of their miRNA and estrogen-regulation, which \n",
      "contrasted with the prominent influence of the TFs over \n",
      "the remaining signatures more related to endometrial \n",
      "progression. All these facts reinforce the hypothesis \n",
      "that implantation failure is mainly caused by impaired \n",
      "miRNA expression [53].\n",
      "Although we observed a clear hormonal influence in\n",
      "numerous annotated regulators, this study transcended \n",
      "the simplistic paradigm of the bihormonal regulation of \n",
      "endometrial function by identifying a plethora of medi -\n",
      "ators, including 595 miRNAs and 311 TFs, that were \n",
      "previously unrelated to these ovarian hormones. These \n",
      "findings set the foundation for new discoveries, reveal -\n",
      "ing alternative pathways and new actors in the regula -\n",
      "tion of the human endometrium, and deepening our \n",
      "understanding of the complex regulatory mechanisms \n",
      "behind endometrial progression and unexplained \n",
      "infertility.\n",
      "MicroRNAs are well known to be involved in embryo \n",
      "implantation [12] and receptivity control [54, 55]. Using \n",
      "network analysis, we unveiled the 10 common miR -\n",
      "NAs regulating endometrial progression and function \n",
      "across published biomarker signatures (hsa-miR-22-3p, \n",
      "hsa-miR-107, hsa-miR-103a-3p, hsa-miR-128-3p, hsa-\n",
      "miR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, hsa-\n",
      "miR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p).\n",
      "hsa-miR-107, hsa-miR-103a-3p, hsa-miR-128-3p, hsa-\n",
      "miR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, hsa-\n",
      "miR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p). \n",
      "Their significant gene expression changes in the secre -\n",
      "tory endometrium support their potential role in the \n",
      "endometrial regulation through an inhibition during \n",
      "the WOI. Notably, hsa-miR-424-5p, hsa-miR-27b-3p \n",
      "and hsa-miR-195-5p have been previously associated \n",
      "with RIF [11, 54, 56], while has-let-7b-5p was associ -\n",
      "ated with human endometrial receptivity [57], and \n",
      "hsa-miR-22-3p was overexpressed in a decidualization \n",
      "model [58]. Moreover, hsa-miR-15a-5p was previously \n",
      "associated with endometriosis [59]. Notwithstanding, \n",
      "this study is the first to associate hsa-miR-15a-5p, hsa-\n",
      "miR-218-5p, hsa-miR-103a-3p, hsa-miR-107, and hsa-\n",
      "miR-128-3p to endometrial progression and functional \n",
      "regulation in humans.\n",
      "Due to their direct relationship with progesterone and \n",
      "estrogen [60], we expected to find TFs, such as ESR1  \n",
      "and PGR, involved in endometrial progression, how -\n",
      "ever, we additionally identified other TFs with an even \n",
      "larger influence in this process. For instance, STAT3  \n",
      "was associated with 17/19 signatures (FDR < 0.05), and \n",
      "is known to regulate uterine epithelial junctional reor -\n",
      "ganization and stromal proliferation, which are critical \n",
      "for implantation [61]. SP1, which similarly regulated \n",
      "17/19 signatures (FDR < 0.05), acts as a downstream \n",
      "paracrine target of progesterone to regulate estrogen \n",
      "inactivation, and could have a predominant role dur -\n",
      "ing the WOI [62, 63]. By focusing on the most univer -\n",
      "sal, overlapping TFs that were not previously related \n",
      "to implantation, we ultimately prioritized CTCF and \n",
      "GATA6 as novel key regulators of endometrial pro -\n",
      "gression, however, their specific molecular actions in\n",
      "Page 11 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "endometrial progression and function were beyond the \n",
      "scope of this project and merit further investigation.\n",
      "CTCF is a conserved zinc finger protein whose regu -\n",
      "latory functions are well characterized throughout the \n",
      "human body [64]. CTCF acts as a transcriptional repres -\n",
      "sor in RNA polymerase II (Pol II) pausing and imprint -\n",
      "ing and  X-chromosome inactivation [65], as well as an \n",
      "insulator, blocking the interaction between enhancers \n",
      "and the promoters of neighbouring genes [66]. Consider -\n",
      "ing these attributed functions and its variable expression \n",
      "throughout the menstrual cycle, we propose that CTCF \n",
      "exerts an inhibitory role in endometrial tissue during \n",
      "the PF phase of the menstrual cycle. With the significant \n",
      "downregulation of CTCF in the secretory endometrium, \n",
      "dually validated in-silico  and experimentally herein, its \n",
      "inhibited genes would be derepressed and become tran -\n",
      "scriptionally active during the WOI. This interpretation \n",
      "supports previous findings from our group demonstrat -\n",
      "ing that, during the WOI, a global transcriptional dere -\n",
      "pression may be required for implantation and early \n",
      "embryo development [63]. Transcriptional derepression \n",
      "has been associated with multiple human disease states \n",
      "and should be investigated further within the context \n",
      "of endometrial-factor infertility. Despite previous asso -\n",
      "ciations of CTCF with endometriosis [67], its implica -\n",
      "tion in endometrial progression and function has not \n",
      "been directly proposed until now. Nevertheless, the role \n",
      "of CTCF in endometrial receptivity has been revealed \n",
      "through its interaction with HOXA10, a gene mainly \n",
      "expressed in endometrium involved in functions such as \n",
      "endometrial proliferation and differentiation, the forma -\n",
      "tion of pinopodes or embryo implantation [68]. Indeed, \n",
      "this study proposed that an overexpression of CTCF can \n",
      "lead to a drop in HOXA10 expression, affecting endome -\n",
      "trial proliferation and endometrial function and corrobo-\n",
      "rating the fact that CTCF must decrease its expression as \n",
      "we can observe in our findings.\n",
      "GATA6, the second prioritized TF that overlapped \n",
      "across the gene lists, is a transcription factor belong -\n",
      "ing to the GATA family, a highly conserved family of \n",
      "six zinc finger proteins [69]. Besides its essential role in \n",
      "embryonic development [70], GATA6 plays a key role \n",
      "in endometriosis and regulates steroidogenic genes [71]. \n",
      "Together with evidence supporting both an activating \n",
      "and inhibitory role for GATA6 [72, 73], the significant \n",
      "changes we observed in its expression across the men -\n",
      "strual cycle reflect its presumed involvement in the endo-\n",
      "metrium and suggest an activating role in endometrial \n",
      "progression and endometrial receptivity acquisition. Like \n",
      "CTCF, this is the first time that GATA6 is proposed as a \n",
      "key TF in endometrial progression and function.\n",
      "Although the exact molecular mechanisms underlying\n",
      "endometrial regulation remain elusive, emerging exper -\n",
      "imental technologies such as High-throughput Chro -\n",
      "mosome Conformation Capture (Hi-C) or chromatin \n",
      "interaction analysis with paired-end tags (ChIA-PET) \n",
      "that respectively study general chromatin interactions \n",
      "or chromatin interactions involving a specific protein \n",
      "[74, 75] may improve our understanding of genetic \n",
      "regulation in the future. Nevertheless, as collaborative \n",
      "efforts unveil new regulators and gene targets in humans \n",
      "(such as the ENCODE project [76]), the complexities of \n",
      "transcriptional regulation can be exposed by in-silico  \n",
      "analyses like those described herein, and we can gain a \n",
      "better understanding of the processes involved in men -\n",
      "strual cycle progression and endometrial competence.\n",
      "Furthermore, it should be noted that results of gene \n",
      "expression studies are influenced by different variables, \n",
      "such as the experimental platforms employed, study \n",
      "designs, or patients’ characteristics of each independent\n",
      "expression studies are influenced by different variables, \n",
      "such as the experimental platforms employed, study \n",
      "designs, or patients’ characteristics of each independent \n",
      "study, adding variability [30, 77, 78]. Despite this fact, \n",
      "we have found that our holistic approach overcomes this \n",
      "undesired variability effect by finding common upstream \n",
      "master regulators across studies and predicting new \n",
      "relationships between hormones, TFs and miRNAs. \n",
      "This approach also provides the basis for future single-\n",
      "molecule studies that aim to elucidate new regulatory \n",
      "pathways in endometrial progression and function. Our \n",
      "results lay the foundation for further molecular studies \n",
      "that can validate the function(s) of the prevailing endo -\n",
      "metrial regulators we prioritized.\n",
      "Conclusion\n",
      "The study shows for the first time, the relative contri -\n",
      "bution of estrogens, progesterone, TFs and miRNAs in \n",
      "endometrial function and progression. Endometrial pro -\n",
      "gression is mainly influenced by progesterone-related \n",
      "genes and TFs, whereas miRNAs and estrogen-related \n",
      "genes play a larger role in endometrial pathology. \n",
      "Moreover, we highlight novel common transcriptional \n",
      "regulators such as CTCF, GATA6, hsa-miR-15a-5p, hsa-\n",
      "miR-218-5p, hsa-miR-107, hsa-miR-103a-3p, and hsa-\n",
      "miR-128-3p across 19 studies that propose biomarkers in \n",
      "endometrial regulation. These results reveal the molecu -\n",
      "lar mechanism underlying endometrial regulation and lay \n",
      "the foundation for the development of targeted therapies \n",
      "for patients with endometrial-factor infertility.\n",
      "Abbreviations\n",
      "ANOVA  Analysis of variance\n",
      "ChIA‑PET  Chromatin interaction analysis with paired‑end tags\n",
      "EPF  Early proliferative\n",
      "ESE  Early secretory\n",
      "FDR  False Discovery Rate\n",
      "GEO  Gene Expression Omnibus\n",
      "Page 12 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "GO  Gene Ontology\n",
      "HGNC  HUGO Gene Nomenclature Committee\n",
      "Hi‑C  High‑throughput Chromosome Conformation Capture\n",
      "IQR  Interquartile range\n",
      "KEGG  Kyoto Encyclopaedia of Genes and Genomes\n",
      "miRNAs  MicroRNAs\n",
      "MSE  Mid Secretory\n",
      "LPF  Late proliferative\n",
      "LSE  Late secretory\n",
      "RIF  Recurrent Implantation Failure\n",
      "Pol II  RNA polymerase II\n",
      "TFs  Transcription Factors\n",
      "WOI  Window of implantation\n",
      "Supplementary Information\n",
      "The online version contains supplementary material available at https:// doi. \n",
      "org/ 10. 1186/ s12958‑ 023‑ 01131‑4.\n",
      "Additional file 1: Supplementary Figure S1. Study design and \n",
      "workflow. Steps performed throughout the study. (A) We reviewed 19 \n",
      "publications related to endometrial progression and function, extracted \n",
      "the gene signatures from the original publication, and created corre‑\n",
      "sponding gene lists that we could annotate. (B) We performed different \n",
      "functional analyses based on the type of regulator (i.e., ovarian hormones, \n",
      "transcription factors and miRNAs). We used online databases to obtain the \n",
      "information of the regulatory process, at the hormonal (Gene Ontology, \n",
      "Kyoto Encyclopedia of Genes and Genomes, Dorothea), transcription \n",
      "factor (Dorothea) and miRNA (Tarbase) level and analysed the results \n",
      "accordingly. Finally, we employed an integrative analysis and prioritized \n",
      "the most relevant regulators for subsequent validation. TF, Transcription \n",
      "factors; miRNA, microRNA.\n",
      "Additional file 2: Supplementary Table S1. Characteristics of endo‑\n",
      "metrial transcriptomic datasets used to build the integrated dataset. ID, \n",
      "identifier of Gene Expression Omnibus database; Source, first author and \n",
      "year of the publication if it is available; No. samples; number of samples \n",
      "included; Age, range of patient age; Participants, reason for endome‑\n",
      "trial sample collection; Cycle Type, type of menstrual cycle of subjects \n",
      "included; Platform, platform used to measure gene expression; No. genes, \n",
      "number of genes measured in each dataset, Endometrial phase, number \n",
      "of samples in each phase of menstrual cycle. PF, proliferative; ESE, early \n",
      "secretory; MSE, mid secretory, LSE, late secretory. NA, Not Available.\n",
      "Additional file 3: Supplementary Table S2. Primer sequences for CTCF \n",
      "and GATA6. Forward and reverse primers used for RT‑qPCR validation of \n",
      "CTCF and GATA6 transcription factors.\n",
      "Additional file 4: Supplementary Table S3. Results of miRNA enrich‑\n",
      "ment. Excel file with the over‑representation analysis results of the micro‑\n",
      "RNAs (miRNAs). Each sheet contains the results of a specific list of genes. \n",
      "In each sheet, the following information is included: ID, gene symbol ID of \n",
      "the miRNA; GeneRatio, the proportion of miRNA targets in the gene list; \n",
      "BgRatio, the proportion of miRNA targets in the whole genome; OddsRa‑\n",
      "tio; the odd ratios for each miRNA, pvalue; the P‑value obtained for each \n",
      "miRNA; Padjust, P‑value adjusted by FDR for each miRNA; geneID, gene\n",
      "symbol ID of the miRNA targets.\n",
      "Additional file 5: Supplementary Table S4. Results of TFs enrichment. \n",
      "Excel file with the over‑representation analysis results of the transcription \n",
      "factors (TFs). Each sheet contains the results of a specific list of genes. In \n",
      "each sheet, the following information is included: ID, gene symbol of the \n",
      "TF; GeneRatio, the proportion of TF targets in the gene list; BgRatio, the \n",
      "proportion of TF targets in the whole genome; OddsRatio; the odd ratios \n",
      "for each TF, pvalue; the P‑value obtained for each TF; padjust, P‑value \n",
      "adjusted by FDR for each TF; geneID, the gene symbol ID of the TF targets.\n",
      "Additional file 6: Supplementary Table S5. Relative contribution of \n",
      "each regulator type (TFs or miRNAs) in each gene list. A Fisher’s test was \n",
      "performed to calculate the proportional differences between transcrip‑\n",
      "tion factors (TFs) and microRNAs (miRNAs). Resulting P‑values were \n",
      "adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene\n",
      "tion factors (TFs) and microRNAs (miRNAs). Resulting P‑values were \n",
      "adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene \n",
      "lists were mainly regulated by TFs, whereas odds ratios < 1 indicate a \n",
      "higher regulation by miRNAs. Odds Ratio with a value of “Inf” indicate \n",
      "that miRNAs were not found for this gene list. Percentages indicate the \n",
      "proportion between the significant regulators in each gene signature and \n",
      "regulators with at least one target gene in the corresponding signature. \n",
      "(***FDR ≤0.001).\n",
      "Acknowledgements\n",
      "The authors would like to thank Yassmin Medina‑Laver and Alicia Quiñonero \n",
      "for their advice and technical support during the experimental validation. \n",
      "Moreover, the authors thank Rosalba Lopez for her constructive criticism and \n",
      "editorial services in preparing the manuscript for publication.\n",
      "Author’s contribution\n",
      "P . D.‑G. conceived the original idea of this work and designed the study with the \n",
      "help of P .S.‑L. P .D.‑G. with the help of J.R. promoted the study and P .D.‑G. supervised \n",
      "it. All the in‑silico and experimental procedures were done by A.P .‑L. and supervised \n",
      "by P .D.‑G. In‑silico procedures were also supervised by P .S.‑L. Gene lists were col‑\n",
      "lected by A.P .‑L. with help from P .S.‑L. and A.D.‑P . miRNA dataset was collected and \n",
      "pre‑processed by A.P .‑L. with the help from P .S.‑L. and A.D.‑P . The collection of genes \n",
      "related to ovarian hormones was carried out by A.P .‑L. with the supervision of P .D.‑G. \n",
      "Statistical analyses were carried out by A.P .‑L. with the help of P .S.‑L. and supervised \n",
      "by P .D.‑G. miRNAs and TFs analysis was done by A.P .‑L. supervised by P .S.‑L. Networks \n",
      "and figures were performed by A.P .‑L., with the help of A.D.‑P . being supervised by \n",
      "P .D.‑G. Samples were collected by J.R. and N.P . Experimental validation was carried \n",
      "out by A.P .‑L. and D.M.‑G. and co‑supervised by F.D. and P .D‑G. Interpretation regard‑\n",
      "ing the results and their effect in the receptivity was carried out by A.P .‑L. and P .D.‑G. \n",
      "with the help of A.D.‑P . The manuscript was written by A.P .‑L., P .D.‑G., and A.D‑P , and \n",
      "was revised by all co‑authors (P .S.‑L., J.R., D.M.‑G., N.P . and F.D.).\n",
      "Funding\n",
      "This study was supported by Fundación IVI (1706‑FIVI‑048‑PD); the Instituto \n",
      "de Salud Carlos III (ISCIII) and co‑funded by the European Regional Develop‑\n",
      "ment Fund “A way to make Europe” (PI19/00537 [P .D.‑G]). Patricia Diaz‑Gimeno is \n",
      "supported by Instituto de Salud Carlos III (ISCIII) through Miguel Servet program \n",
      "(CP20/00118 [P .D.‑G]) and co‑funded by the European Union. Spanish Ministry of \n",
      "Science, Innovation and Universities (FPU18/01777 [A.P .‑L.], FPU15/01398 [A.D.‑P .], \n",
      "FPU19/03247 [D.M.‑G). Patricia Sebastian‑Leon is funded by Instituto de Salud \n",
      "Carlos III (ISCIII) through Sara Borrell program (CD21/00132[P .S.‑L]) and co‑funded \n",
      "by the European Union. Francisco Dominguez was supported by Instituto de \n",
      "Salud Carlos III(ISCIII) through Miguel Servet program CPII18/00002 and co‑funded\n",
      "by the European Union (Fonda Social Europeo), “El FSE invierte en tu futuro” .\n",
      "Availability of data and materials\n",
      "The data used to analyze the TF and miRNA expression throughout the men‑\n",
      "strual cycle and in infertility are available in Gene Expression Omnibus, with \n",
      "IDs GSE98386, GSE29981, GSE4888, GSE119209, GSE86491, and GSE44558.\n",
      "Declarations\n",
      "Ethics approval and consent to participate\n",
      "This research evaluated endometrial gene expression data from women throughout \n",
      "the menstrual cycle. Raw endometrial gene expression data (from previous studies) \n",
      "was downloaded from the publicly‑available functional genomic dataset repository \n",
      "Gene Expression Omnibus (GEO) [79] for curation/processing. In accordance with \n",
      "GEO policies, all patient data was anonymized and encrypted, and therefore an addi‑\n",
      "tional institutional review board approval was not required for our in-silico validation \n",
      "of regulator gene expression changes in endometrial tissue. On the other hand, for\n",
      "tional institutional review board approval was not required for our in-silico validation \n",
      "of regulator gene expression changes in endometrial tissue. On the other hand, for \n",
      "the experimental validation of gene expression changes in endometrial biopsies, \n",
      "all individuals gave written informed consent for endometrial gene expression \n",
      "evaluation prior to participating (Ethics Committee of the Instituto Valenciano de \n",
      "Infertilidad [IVI Valencia], Valencia, Spain [1810‑FIVI‑066‑PD/1706‑FIVI‑048‑PD]).\n",
      "Consent for publication\n",
      "Not applicable.\n",
      "Competing interests\n",
      "The authors declare no competing interests.\n",
      "Page 13 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      " \n",
      "Author details\n",
      "1 IVIRMA Global Research Alliance, IVI Foundation, Instituto de Investigación \n",
      "Sanitaria La Fe (IIS La Fe), Av. Fernando Abril Martorell 106, Torre A, Planta \n",
      "1ª, 46026 Valencia, Valencia, Spain. 2 Department of Pediatrics, Obstetrics \n",
      "and Gynaecology, Universidad de Valencia, Av. Blasco Ibáñez 15, 46010 Valen‑\n",
      "cia, Valencia, Spain. 3 IVIRMA Global Research Alliance, IVIRMA Valencia, Plaza de \n",
      "La Policia Local 3, 46015 Valencia, Spain. \n",
      "Received: 21 June 2023   Accepted: 22 August 2023\n",
      "References\n",
      " 1. Strowitzki T, Germeyer A, Popovici R, von Wolff M. The human endo‑\n",
      "metrium as a fertility‑determining factor. Hum Reprod Update. \n",
      "2006;12:617–30.\n",
      " 2. Cakmak H, Taylor HS. Implantation failure: Molecular mechanisms and \n",
      "clinical treatment. Hum Reprod Update. 2011;17:242–53.\n",
      " 3. Lessey BA, Young SL. What exactly is endometrial receptivity? Fertil Steril. \n",
      "2019;111:611–7.\n",
      " 4. Talbi S, Hamilton AE, Vo KC, Tulac S, Overgaard MT, Dosiou C, et al. \n",
      "Molecular phenotyping of human endometrium distinguishes menstrual \n",
      "cycle phases and underlying biological processes in normo‑ovulatory \n",
      "women. Endocrinology. 2006;147:1097–121.\n",
      " 5. Somigliana E, Vigano P , Busnelli A, Paffoni A, Vegetti W, Vercellini P . \n",
      "Repeated implantation failure at the crossroad between statistics, clinics \n",
      "and over‑diagnosis. Reprod Biomed Online. 2018;36:32–8.\n",
      " 6. Craciunas L, Gallos I, Chu J, Bourne T, Quenby S, Brosens JJ, et al. Con‑\n",
      "ventional and modern markers of endometrial receptivity: a systematic \n",
      "review and meta‑analysis. Hum Reprod Update. 2019;25:202–23.\n",
      " 7. Sebastian‑Leon P , Garrido N, Remohí J, Pellicer A, Diaz‑Gimeno P . Asyn‑\n",
      "chronous and pathological windows of implantation: Two causes of \n",
      "recurrent implantation failure. Hum Reprod. 2018;33:626–35.\n",
      " 8. Vasquez YM, DeMayo FJ. Role of nuclear receptors in blastocyst implanta‑\n",
      "tion. Semin Cell Dev Biol. 2013;24:724–35.\n",
      " 9. Wang H, Zhang S, Lin H, Kong S, Wang S, Wang H, et al. Physiological \n",
      "and molecular determinants of embryo implantation. Mol Aspects Med. \n",
      "2013;34:939–80.\n",
      " 10. Kelleher AM, Behura SK, Burns GW, Young SL, Demayo FJ, Spencer TE. Inte‑\n",
      "grative analysis of the forkhead box A2 (FOXA2) cistrome for the human \n",
      "endometrium. FASEB J. 2019;33:8543–54.\n",
      " 11. Revel A, Achache H, Stevens J, Smith Y, Reich R. MicroRNAs are \n",
      "associated with human embryo implantation defects. Hum Reprod. \n",
      "2011;26:2830–40.\n",
      " 12. Galliano D, Pellicer A. MicroRNA and implantation. Fertil Steril. \n",
      "2014;101:1531–44.\n",
      " 13. Xu H, Zhou M, Cao Y, Zhang D, Han M, Gao X, et al. Genome‑wide analysis \n",
      "of long noncoding RNAs, microRNAs, and mRNAs forming a compet‑\n",
      "ing endogenous RNA network in repeated implantation failure. Gene. \n",
      "2019;720: 144056.\n",
      " 14. Copley RR. The animal in the genome: comparative genomics and evolu‑\n",
      "tion. Philos Trans R Soc Lond B: Biol Sci. 2008;363:1453–61.\n",
      "15. Jacob F. The statue within: an autobiography. CSHL Press; 1995.\n",
      " 16. Yanai I, Lercher M. Night science. Genome Biol. 2019;20:179.\n",
      " 17. Carson DD, Lagow E, Thathiah A, Al‑Shami R, Farach‑Carson MC, Vernon \n",
      "M, et al. Changes in gene expression during the early to mid‑luteal \n",
      "(receptive phase) transition in human endometrium detected by high‑\n",
      "density microarray screening. Mol Hum Reprod. 2002;8:871–9.\n",
      " 18. Borthwick JM, Charnock‑Jones DS, Tom BD, Hull ML, Teirney R, Phillips SC, \n",
      "et al. Determination of the transcript profile of human endometrium. Mol \n",
      "Hum Reprod. 2003;9:19–33.\n",
      " 19. Mirkin S, Arslan M, Churikov D, Corica A, Diaz JI, Williams S, et al. In search \n",
      "of candidate genes critically expressed in the human endometrium dur‑\n",
      "ing the window of implantation. Hum Reprod. 2005;20:2104–17.\n",
      " 20. Devesa‑Peiro A, Sebastian‑Leon P , Parraga‑Leo A, Pellicer A, Diaz‑Gimeno \n",
      "P . Breaking the ageing paradigm in endometrium: endometrial gene \n",
      "expression related to cilia and ageing hallmarks in women over 35 years. \n",
      "Hum Reprod. 2022;37:762–76.\n",
      "P . Breaking the ageing paradigm in endometrium: endometrial gene \n",
      "expression related to cilia and ageing hallmarks in women over 35 years. \n",
      "Hum Reprod. 2022;37:762–76.\n",
      " 21. Altmae S, Martinez‑Conejero JA, Salumets A, Simon C, Horcajadas JA, \n",
      "Stavreus‑Evers A. Endometrial gene expression analysis at the time of \n",
      "embryo implantation in women with unexplained infertility. Mol Hum \n",
      "Reprod. 2010;16:178–87.\n",
      " 22. Devesa‑Peiro A, Sebastian‑Leon P , Pellicer A, Diaz‑Gimeno P . Guidelines for \n",
      "biomarker discovery in endometrium: correcting for menstrual cycle bias \n",
      "reveals new genes associated with uterine disorders. Mol Hum Reprod. \n",
      "2021;27:11.\n",
      " 23. Durinck S, Spellman PT, Birney E, Huber W. Mapping identifiers for the \n",
      "integration of genomic datasets with the R/ Bioconductor package \n",
      "biomaRt. Nat Protoc. 2009;4:1184–91.\n",
      " 24. Kanehisa M, Goto S. KEGG: Kyoto encyclopedia of genes and genomes. \n",
      "Nucleic Acids Res. 2000;27:27–30.\n",
      " 25. Ashburner M, Ball CA, Blake JA, Botstein D, Butler H, Cherry JM, et al. Gene \n",
      "ontology: tool for the unification of biology. Nat Genet. 2000;25:25–9.\n",
      " 26. Garcia‑Alonso L, Holland CH, Ibrahim MM, Turei D, Saez‑Rodriguez J. \n",
      "Benchmark and integration of resources for the estimation of human \n",
      "transcription factor activities. Genome Res. 2019;29:1363. https:// doi. org/ \n",
      "10. 1101/ gr. 240663. 118.\n",
      " 27. Karagkouni D, Paraskevopoulou MD, Chatzopoulos S, Vlachos IS, Tast‑\n",
      "soglou S, Kanellos I, et al. DIANA‑TarBase v8: a decade‑long collection of \n",
      "experimentally supported miRNA‑gene interactions. Nucleic Acids Res. \n",
      "2018;46:239–45.\n",
      " 28. Franz M, Lopes CT, Huck G, Dong Y, Sumer O, Bader GD. Cytoscape. \n",
      "js: a graph theory library for visualisation and analysis. Bioinformatics. \n",
      "2016;32:309–11.\n",
      " 29. Vilella F, Moreno‑Moya JM, Balaguer N, Grasso A, Herrero M, Martínez S, \n",
      "et al. Hsa‑miR‑30d, secreted by the human endometrium, is taken up \n",
      "by the pre‑implantation embryo and might modify its transcriptome. \n",
      "Development. 2015;142:3210–21.\n",
      " 30. Henarejos‑Castillo I, Sebastian‑Leon P , Devesa‑Peiro A, Pellicer A, Diaz‑\n",
      "Gimeno P . SARS‑CoV‑2 infection risk assessment in the endometrium: \n",
      "viral infection‑related gene expression across the menstrual cycle. Fertil \n",
      "Steril. 2020;114:223–32.\n",
      " 31. Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, et al. Limma powers \n",
      "differential expression analyses for RNA sequencing and microarray stud‑\n",
      "ies. Nucleic Acids Res. 2015;43:e47.\n",
      " 32. Livak KJ, Schmittgen TD. Analysis of relative gene expression data \n",
      "using real‑time quantitative PCR and the 2‑ΔΔCT method. Methods. \n",
      "2001;25:402–8.\n",
      " 33. R Development Core Team, R Core Team, R Development Core Team. A \n",
      "Language and Environment for Statistical Computing. R Foundation for \n",
      "Statistical Computing. 2018;2. https:// www. R‑‑ proje ct. org.\n",
      " 34. Altmäe S, Koel M, Võsa U, Adler P , Suhorutšenko M, Laisk‑Podar T, et al. \n",
      "Meta‑signature of human endometrial receptivity: a meta‑analysis and \n",
      "validation study of transcriptomic biomarkers. Scient Rep. 2017;7:10077.\n",
      "35. Carrascosa J, Horcajadas J, Moreno‑Moya JM. Chapter 15 ‑ the molecular \n",
      "signature of the endometrial receptivity: research and clinical application. \n",
      "In: Horcajadas JA, Gosálvez J, editors. Reproductomics: the omics revolu‑\n",
      "tion and its impact on human reproductive medicine. Academic Press; \n",
      "2018. p. 301–279.\n",
      " 36. Díaz‑Gimeno P , Horcajadas JA, Martínez‑Conejero JA, Esteban FJ, Alamá \n",
      "P , Pellicer A, et al. A genomic diagnostic tool for human endometrial \n",
      "receptivity based on the transcriptomic signature. Fertil Steril. 2011;95:50.\n",
      " 37. Kao LC, Tulac S, Lobo S, Imani B, Yang JP , Germeyer A, et al. Global gene \n",
      "profiling in human endometrium during the window of implantation. \n",
      "Endocrinology. 2002;143:2119–38.\n",
      " 38. Ponnampalam AP , Weston GC, Trajstman AC, Susil B, Rogers PAW. Molecu‑\n",
      "lar classification of human endometrial cycle stages by transcriptional \n",
      "profiling. Mol Hum Reprod. 2004;10:879–93.\n",
      " 39. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al.\n",
      "lar classification of human endometrial cycle stages by transcriptional \n",
      "profiling. Mol Hum Reprod. 2004;10:879–93.\n",
      " 39. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al. \n",
      "Oestrogen‑modulated gene expression in the human endometrium. Cell \n",
      "Mol Life Sci. 2005;62:239–50.\n",
      " 40. Riesewijk A, Martín J, van Os R, Horcajadas JA, Polman J, Pellicer A, et al. \n",
      "Gene expression profiling of human endometrial receptivity on days LH+2 \n",
      "versus LH+7 by microarray technology. Mol Hum Reprod. 2003;9:253–64.\n",
      " 41. Bastu E, Demiral I, Gunel T, Ulgen E, Gumusoglu E, Hosseini MK, et al. \n",
      "Potential marker pathways in the endometrium that may cause recurrent \n",
      "implantation failure. Reprod Sci. 2019;26:879–90.\n",
      " 42. Bersinger NA, Wunder DM, Birkhäuser MH, Mueller MD. Gene expression \n",
      "in cultured endometrium from women with different outcomes follow‑\n",
      "ing IVF. Mol Hum Reprod. 2008;14:475–84.\n",
      "Page 14 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "•\n",
      " \n",
      "fast, convenient online submission\n",
      " •\n",
      "  \n",
      "thorough peer review by experienced researchers in your ﬁeld\n",
      "• \n",
      " \n",
      "rapid publication on acceptance\n",
      "• \n",
      " \n",
      "support for research data, including large and complex data types\n",
      "•\n",
      "  \n",
      "gold Open Access which fosters wider collaboration and increased citations \n",
      " \n",
      "maximum visibility for your research: over 100M website views per year •\n",
      "  At BMC, research is always in progress.\n",
      "Learn more biomedcentral.com/submissions\n",
      "Ready to submit y our researc hReady to submit y our researc h  ?  Choose BMC and benefit fr om: ?  Choose BMC and benefit fr om: \n",
      " 43. Bhagwat SR, Chandrashekar DS, Kakar R, Davuluri S, Bajpai AK, Nayak S, \n",
      "et al. Endometrial receptivity: a revisit to functional genomics studies \n",
      "on human endometrium and creation of HGEx ‑ERdb. PLoS ONE. 2013;8: \n",
      "e58419.\n",
      " 44. Koot YEM, Van Hooff SR, Boomsma CM, Van Leenen D, Koerkamp \n",
      "MJAG, Goddijn M, et al. An endometrial gene expression signature \n",
      "accurately predicts recurrent implantation failure after IVF. Sci Rep. \n",
      "2015;2016(6):1–12.\n",
      " 45. Lédée N, Munaut C, Aubert J, Sérazin V, Rahmati M, Chaouat GG, et al. \n",
      "Specific and extensive endometrial deregulation is present before \n",
      "conception in IVF/ICSI repeated implantation failures (IF) or recurrent \n",
      "miscarriages. J Pathol. 2011;225:554–64.\n",
      " 46. Pathare ADS, Zaveri K, Hinduja I. Downregulation of genes related to \n",
      "immune and inflammatory response in IVF implantation failure cases under \n",
      "controlled ovarian stimulation. Am J Reprod Immunol. 2017;78: e12679.\n",
      " 47. Shi C, Han HJ, Fan LJ, Guan J, Zheng XB, Chen X, et al. Diverse endometrial \n",
      "mRNA signatures during the window of implantation in patients with \n",
      "repeated implantation failure. Hum Fertil. 2018;21:183–94.\n",
      " 48. Messinisi IE. Ovarian feedback, mechanism of action and possible clinical \n",
      "implications. Hum Reprod Update. 2006;12:557–71.\n",
      " 49. Edwards RG. Implantation, interception and contraception. Hum Reprod \n",
      "(Oxford, England). 1994;9:985–95.\n",
      " 50. Young SL. Oestrogen and progesterone action on endometrium: a \n",
      "translational approach to understanding endometrial receptivity. Reprod \n",
      "Biomed Online. 2013;27:497–505.\n",
      " 51. Bhurke AS, Bagchi IC, Bagchi MK. Progesterone‑regulated endometrial \n",
      "factors controlling implantation. Am J Reprod Immunol. 2016;75:237–45.\n",
      " 52. Devesa‑Peiro A, Sebastian‑Leon P , Garcia‑ Garcia F, Arnau V, Aleman \n",
      "A, Pellicer A, et al. Uterine disorders affecting female fertility: what \n",
      "are the molecular functions altered in endometrium? Fertil Steril. \n",
      "2020;113:1261–74.\n",
      " 53. von Grothusen C, Frisendahl C, Modhukur V, Lalitkumar PG, Peters M, \n",
      "Faridani OR, et al. Uterine fluid microRNAs are dysregulated in women \n",
      "with recurrent implantation failure. Hum Reprod. 2022;37:734–46.\n",
      " 54. Altmäe S, Martinez‑Conejero JA, Esteban FJ, Ruiz‑Alonso M, Stavreus‑\n",
      "Evers A, Horcajadas JA, et al. MicroRNAs miR‑30b, miR‑30d, and miR‑494\n",
      "regulate human endometrial receptivity. Reprod Sci. 2013;20:308–17.\n",
      " 55. Di Pietro C, Caruso S, Battaglia R, IraciSareri M, La Ferlita A, Strino F, et al. \n",
      "MiR‑27a‑3p and miR‑124–3p, upregulated in endometrium and serum \n",
      "from women affected by Chronic Endometritis, are new potential \n",
      "molecular markers of endometrial receptivity. Am J Reprod Immunol. \n",
      "2018;80:12858.\n",
      " 56. Shi C, Shen H, Fan L ‑ JJ, Guan J, Zheng X ‑BB, Chen X, et al. Endometrial \n",
      "MicroRNA signature during the window of implantation changed in \n",
      "patients with repeated implantation failure. Chin Med J. 2017;130:566–73.\n",
      " 57. La Ferlita A, Battaglia R, Andronico F, Caruso S, Cianci A, Purrello M, \n",
      "et al. Non‑coding RNAs in endometrial physiopathology. Int J Mol Sci. \n",
      "2018;19:2120.\n",
      " 58. Estella C, Herrer I, Moreno‑Moya JM, Quiñonero A, Martínez S, Pellicer A, \n",
      "et al. miRNA signature and dicer requirement during human endometrial \n",
      "stromal decidualization in vitro. PLoS ONE. 2012;7:41080.\n",
      " 59. Liu D, Liang Y, Chen M, Yang F, Yao S. Knockdown of circ_0075503 sup ‑\n",
      "stromal decidualization in vitro. PLoS ONE. 2012;7:41080.\n",
      " 59. Liu D, Liang Y, Chen M, Yang F, Yao S. Knockdown of circ_0075503 sup ‑\n",
      "presses cell migration and invasion by regulating miR‑15a‑5p and KLF12 \n",
      "in endometriosis. Mol Cell Biochem. 2021;476:3845–56.\n",
      " 60. Chi RPA, Wang T, Adams N, Wu SP , Young SL, Spencer TE, et al. Human \n",
      "endometrial transcriptome and progesterone receptor cistrome reveal \n",
      "important pathways and epithelial regulators. J Clin Endocrinol Metab. \n",
      "2020;105:1–21.\n",
      " 61. Pawar S, Starosvetsky E, Orvis GD, Behringer RR, Bagchi IC, Bagchi MK. \n",
      "STAT3 regulates uterine epithelial remodeling and epithelial‑stromal \n",
      "crosstalk during implantation. Mol Endocrinol. 2013;27:1996–2012.\n",
      " 62. Cheng YH, Imir A, Suzuki T, Fenkci V, Yilmaz B, Sasano H, et al. SP1 and SP3 \n",
      "mediate progesterone‑ dependent induction of the 17beta hydroxyster ‑\n",
      "oid dehydrogenase type 2 gene in human endometrium. Biol Reprod. \n",
      "2006;75:605–14.\n",
      " 63. Sebastian‑Leon P , Devesa‑Peiro A, Aleman A, Parraga‑Leo A, Arnau V, \n",
      "Pellicer A, et al. Transcriptional changes through menstrual cycle reveal a \n",
      "global transcriptional derepression underlying the molecular mechanism \n",
      "involved in the window of implantation. Mol Hum Reprod. 2021. https:// \n",
      "doi. org/ 10. 1093/ molehr/ gaab0 27.\n",
      " 64. Kim S, Yu NK, Kaang BK. CTCF as a multifunctional protein in genome \n",
      "regulation and gene expression. Exp Mol Med. 2015;47: e166.\n",
      " 65. Shukla S, Kavak E, Gregory M, Imashimizu M, Shutinoski B, Kashlev M, \n",
      "et al. CTCF‑promoted RNA polymerase II pausing links DNA methylation \n",
      "to splicing. Nature. 2011;479:74–9.\n",
      " 66. Phillips JE, Corces VGCTCF. Master weaver of the genome. Cell. \n",
      "2009;137:1194–211.\n",
      " 67. Guo J, Cao B, Xu X, Wu F, Zhu B. Novel CTCF mutations in Chinese \n",
      "patients with ovarian endometriosis. Mol Med Rep. 2018;18:1031–6.\n",
      " 68. Geng J, Cui C, Yin Y, Zhao Y, Zhang C. LncRNA NEAT1 affects endome ‑\n",
      "trial receptivity by regulating HOXA10 promoter activity. Cell Cycle. \n",
      "2022;21:1932–44.\n",
      " 69. Viger RS, Guittot SM, Anttonen M, Wilson DB, Heikinheimo M. Role of the \n",
      "GATA family of transcription factors in endocrine development, function, \n",
      "and disease. Mol Endocrinol. 2008;22:781.\n",
      " 70. Xuan S, Sussel L. GATA4 and GATA6 regulate pancreatic endoderm \n",
      "identity through inhibition of hedgehog signaling. Development (Cam‑\n",
      "bridge). 2016;143:780–6.\n",
      " 71. Bernardi LA, Dyson MT, Tokunaga H, Sison C, Oral M, Robins JC, et al. The \n",
      "essential role of GATA6 in the activation of estrogen synthesis in endome‑\n",
      "triosis. Reprod Sci. 2019;26:60.\n",
      " 72. Yin F, Herring BP . GATA‑6 can act as a positive or negative regula‑\n",
      "tor of smooth muscle‑specific gene expression *. J Biol Chem. \n",
      "2005;280:4745–52.\n",
      " 73. Zhai J, Luo G. GATA6‑induced FN1 activation promotes the proliferation, \n",
      "invasion and migration of oral squamous cell carcinoma cells. Mol Med \n",
      "Rep. 2022;25:102.\n",
      " 74. Fullwood MJ, Liu MH, Pan YF, Liu J, Xu H, Mohamed YB, et al. An \n",
      "oestrogen‑receptor‑α‑bound human chromatin interactome. Nature. \n",
      "2009;462:58–64.\n",
      "2009;462:58–64.\n",
      " 75. Lieberman‑Aiden E, Van Berkum NL, Williams L, Imakaev M, Ragoczy \n",
      "T, Telling A, et al. Comprehensive mapping of long range interactions \n",
      "reveals folding principles of the human genome. Science. 2009;326:289.\n",
      " 76. Birney E, Stamatoyannopoulos JA, Dutta A, Guigó R, Gingeras TR, \n",
      "Margulies EH, et al. Identification and analysis of functional elements \n",
      "in 1% of the human genome by the ENCODE pilot project. Nature. \n",
      "2007;447:799–816.\n",
      " 77. Tajti F, Kuppe C, Antoranz A, Ibrahim MM, Kim H, Ceccarelli F, et al. A func‑\n",
      "tional landscape of CKD entities from public transcriptomic data. Kidney \n",
      "Int Rep. 2020. https:// doi. org/ 10. 1016/j. ekir. 2019. 11. 005.\n",
      " 78. Henarejos‑ Castillo I, Devesa‑Peiro A, de Miguel‑ Gomez L, Sebastian‑Leon \n",
      "P , Romeu M, Aleman A, et al. Predicted COVID ‑19 molecular effects on \n",
      "endometrium reveal key dysregulated genes and functions. Mol Hum \n",
      "Reprod. 2022;28:gaac035.\n",
      " 79. Barrett T, Wilhite SE, Ledoux P , Evangelista C, Kim IF, Tomashevsky M, et al.\n",
      "endometrium reveal key dysregulated genes and functions. Mol Hum \n",
      "Reprod. 2022;28:gaac035.\n",
      " 79. Barrett T, Wilhite SE, Ledoux P , Evangelista C, Kim IF, Tomashevsky M, et al. \n",
      "NCBI GEO: archive for functional genomics data sets–update. Nucleic \n",
      "Acids Res. 2013;41(Database issue):D991‑5.\n",
      "Publisher’s Note\n",
      "Springer Nature remains neutral with regard to jurisdictional claims in pub‑\n",
      "lished maps and institutional affiliations.\n"
     ]
    }
   ],
   "source": [
    "my_json = []\n",
    "\n",
    "for docs in storage:\n",
    "    print(docs.page_content)\n",
    "\n",
    "    my_json.append(chain.invoke({\"query\": docs.page_content}))\n",
    "\n",
    "\n",
    "#chain.invoke({\"query\": storage[0].page_content})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "19cb09a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'text': 'A total of 3,608 distinct genes from the 19 gene lists were associated with endometrial progression',\n",
       "  'doi': '10.1186/s12958‑023‑01131‑4',\n",
       "  'citation': 'Parraga‑Leo et al. (2023)'},\n",
       " {'text': 'The lists’ regulation was significantly favoured by TFs (89% (17/19) of gene lists) and progesterone (47% (8 /19) of gene lists), rather than miRNAs (5% (1/19) of gene lists) or estrogen (0% (0/19) of gene lists), respectively (FDR < 0.05). Exceptionally, two gene lists that were previously associated with implantation failure and unexplained infertility were less hormone‑dependent, but primarily regulated by estrogen. Although endometrial progression genes were mainly targeted by hormones rather than non‑hormonal contributors (odds ratio = 91.94, FDR < 0.05), we identified 311 TFs and 595 miRNAs not previously associated with ovarian hormones. We highlight CTCF, GATA6, hsa‑miR‑15a‑5p, hsa‑miR‑218‑5p, hsa‑miR‑107, hsa‑miR‑103a‑3p, and hsa‑miR‑128‑3p, as overlapping novel master regulators of endometrial function. The gene expression changes of selected regulators through‑out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their potential regulatory roles in the endometrium.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'This study revealed novel hormonal and non‑hormonal regulators and their relative contributions to endometrial progression and pathology, providing new leads for the potential causes of endometrial‑factor infertility. The endometrium is the innermost layer of the uterus, which undergoes dynamic histological, physiological and molecular changes that allow it to synchronize with embryo development, facilitate embryo implantation, and ultimately, establish a successful pregnancy. Implantation is a crucial and complex limiting step for conception which occurs between days 19 to 24 of a normal menstrual cycle. Consequently, this period of time is called the window of implantation (WOI) and is characterized by abrupt transcriptomic changes in the endometrial tissue. Alterations in either WOI establishment and endometrial progression may lead to implantation failures, which, along with biochemical miscarriage, account for more than 50% of pregnancy losses at pre-clinical stages. These alterations have broadly been classified as being related to a displaced WOI, caused by variable timing of endometrial progression, or a disrupted WOI, where impaired endometrial function prevents the establishment of an effective WOI. Initially, the WOI was postulated to mainly be regu -lated by the ovarian hormones (namely, estrogen and progesterone) acting via their respective nuclear receptors (i.e., ESR1/2 and PGR). However, evidence has shown that these receptors cooperate with other transcription factors (TFs) and co-regulators to mediate uterine physiology [8]. In fact, TFs (such as homeobox TFs, FOXA2, and KLF9) have recently been reported as key regulators for the establishment of endometrial receptivity [9, 10], and microRNAs (miRNAs) have been found to similarly intervene in this complex and multifactorial process as transcriptional regulators [11]. While the involvement of both types of regulators was widely reported in embryo implantation [12] and other infertility-related diseases, including recurrent implantation failure (RIF) [13], the complex interactions between the different types of regulators and their respective contributions to endometrial progression and function have not been investigated from a systemic point of view. This paradigm shift approaches genes and regulators in an integrative way, by considering how they interact with and coordinate each other to carry out cellular processes, rather than reinforcing the premise they work independently [14]. By more accurately reflecting the biological milieu, systemic approaches take gene-based discoveries to the next level.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Previous studies of endometrial transcriptomics have been used to define the WOI, compare the endometrium of healthy women as their menstrual cycles progressed [4, 17–19]; address endometrial differences related to age [20] or the dysfunction of the WOI between patients with implantation failure and healthy controls [21]; and finally, correct for the menstrual cycle bias that can mask important biomarkers [22]. Notwithstanding, the candidate biomarkers of endometrial receptivity reported in each of the aforementioned studies overlap poorly [7]. Indeed, recent comparisons conducted by Sebastian-Leon and colleagues [7] found no congruities between 16 different reported gene lists of endometrial receptivity, and only fair to moderate functional agreements between some of the included signatures. Despite previous transcriptional regulation studies, there have been no holistic studies of endometrial receptivity.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'fair to moderate functional agreements between some of the included signatures. Despite previous transcriptional regulation studies, there have been no holistic studies of this regulatory process, which robustly analyse the drivers of transcription across a higher number of studies (that each searched for endometrial receptivity biomarkers) and map the relative contributions of each type of regulator to endometrial progression regulation (Ovarian hormones [i.e., estrogen, progesterone], TFs, miRNAs). Thus, the aim of this study was to utilize available genomic data regarding transcriptional regulators to identify overlapping mediators of endometrial transcriptional regulation (i.e., ovarian hormones, TFs, and miRNAs) and determine the relative contribution of each type of regulator among the previously reported gene signatures. These findings could help unveil the master regulators and principal type of endometrial regulation, paving the way for further research aimed at improving woman´s reproductive health. Annotating gene lists associated with endometrial progression and function. Endometrial progression and implantation failure gene lists [7] were retrieved and updated using a public data repository (i.e., Gene Expression Omnibus (GEO)).',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'To study the hormonal regulation of endometrial progression, the Kyoto Encyclopaedia of Genes and Genomes (KEGG) [24], and Gene Ontology (GO) [25] databases were consulted for ovarian hormone-related genes, using relevant keywords (i.e., progesterone, estrogen, oestrogen and estradiol) and selecting those pathways and functions that contained them (Supplementary Fig. S1B). Genes associated with the obtained human KEGG pathways and GO functions were grouped according to whether they were unique to progesterone (P4 gene set) or estrogen (E2 gene set) or related to both hormones (P4 and E2 gene set). Gene targets of the nuclear progesterone (PGR) and estrogen (ESR1, ESR2) receptors were added to their corresponding gene sets using DoRothEA (Discriminant Regulon Expression Analysis) database [26], considering only manually-curated or ChiP-Seq experimentally-validated gene-TF relationships. Finally, the P4 and E2 gene sets were independently mapped to each gene list.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Prioritizing key regulators in endometrial progression and function\\nWe built regulatory networks to systemically analyse endometrial regulation.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'We built regulatory networks to systemically analyse endometrial regulation. Nodes represented gene lists and regulators (miRNAs or TFs), while edges indicated significant enrichment among the lists and their corresponding regulators (false discovery rate (FDR) ≤ 0.05). To select the most influential regulators (i.e., those which targeted genes in most of our signatures), we studied the degree distribution of the networks (i.e., the number of gene lists regulated by each molecule) and prioritized the miRNAs and TFs which surpassed the relative maximum (i.e., 1.50 times the interquartile range (IQR)) number of relationships (gene list–regulators). All networks were built and analysed with Cytoscape version 3.7 [28] (Supplementary Fig. S1B). To evaluate how the miRNAs’ expression change in the endometrium throughout the menstrual cycle, we used the raw data from GSE44558 GEO dataset [29], deriving from 20 endometrial samples collected throughout the menstrual cycle: four in early-proliferative (EPF), four in late-proliferative (LPF), four in early-secretory (ESE), four in mid-secretory (MSE) and four in late-secretory (LSE). To evaluate how the TFs’ expression change in the endometrium throughout the menstrual cycle, we analysed the integrated endometrial dataset previously created by our group, and reported in [30], which compiled data from five prior publications (GSE98386, GSE29981, GSE4888, GSE119209 and GSE86491). This dataset included endometrial gene expression',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Of the 19 gene lists used for our analysis, eleven were obtained from published studies evaluating control patients throughout the menstrual cycle [4, 17–19, 34–40], and the other eight were derived from studies comparing patients with RIF or unexplained infertility to controls [21, 41–47] (Table 1). Unifying all the afore-',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'Table 1'},\n",
       " {'text': 'mentioned signatures, we compiled 3,608 genes related to endometrial progression and function. Hormonal regulation of endometrial progression is largely driven by progesterone. We identified 7,540 and 698 genes related to estrogen and progesterone hormones, respectively. However, as determined by the relative contribution of each type of hormone within each gene list (Fig. 1A), 17/19 (89%) signatures favoured regulation by progesterone rather than estrogen. These differences were significant in 47% of the signatures (FDR < 0.05). Exceptionally, the genes reported by the Altmae2010 and Koot signatures, that were related to unexplained infertility and RIF, had a preference for estrogen regulation (Fig. 1A). With respect to the proportion of progesterone-related genes, Altmae2010 showed 2.1 times more estrogen-related genes while Koot showed 1.7 times more. More over, comparing the proportion of genes regulated by both hormones, between signatures revealed that Alt -',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'We identified a total of 770 miRNAs and 555 TFs regulating at least one of the 3,608 genes reported in the 19 gene lists. Further, we observed that 29% of the miRNAs and 44% of the TFs were associated with the ovarian hormones (i.e., progesterone and/or estrogen). Significantly more genes were targeted by hormonally-influenced regulators rather than non-hormonal regulators (P -value < 0.01, odds ratio =  91.94). Nevertheless, we also identified 595 miRNAs (77.27%) and 311 TFs (56.04%) that were unrelated to progesterone and/or estrogen, highlighting a substantial proportion of hormonally-independent regulators of the menstrual cycle.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Table 1 Characteristics of the 19 gene lists included in this study. Gene lists were divided according to their evaluated condition. Each signature was labelled using the name of first author, followed by the publication year in the case of duplicates. Endometrial staging was based on different variables including urinary LH, histological analysis, ultrasound evaluation, or the number of days since the hCG trigger. Participants included healthy controls and women with infertility‑related conditions. Various platforms were used to evaluate gene expression. The thresholds applied for gene refinement and the final number of genes per signature are also presented\\nAbbreviations: Δ delta  Ct, adj-P-value Adjusted P‑value, C Control group, d Day of menstrual cycle, EPF Early proliferative, ESE Early secretory, |FC| Absolute fold‑change, FC Fold change, HRT Hormone replacement therapy, IF Implantation failure, LH Luteinizing hormone, LH + X X days post LH surge, LPF Late proliferative, LSE Late‑secretory, M Miscarriage, MPF Mid‑proliferative, MSE Mid‑secretory, n number of samples, N/A Not applicable, OP Ongoing pregnancy, PF Proliferative, PFP Proportion of false positives, PR Pre‑receptive, R Receptive, RIF Recurrent implantation failure, RRA Robust Rank Aggregation algorithm, SF Spontaneously fertile women, UI Women with unexplained infertility, vs versus\\nCondition Gene signature Endometrial staging Participants Platform Threshold No. genes References\\nEndometrial progres‑ sion\\nAltmäe2017 N/A N/A N/A RRA 57 [34]\\nBorthwick Urinary LH; Histologi‑ cally confirmed PF d9‑11 (n = 5) vs R LH + 6–8 (n = 5)\\nAffymetrix Genechip Hu95A N/A 116 [18]\\nCarrascosa N/A N/A Quantitative RT‑PCR Based on literature 187 [35]\\nCarson Urinary LH; Histologically confirmed ESE LH + 2–4 (n = 3) vs MSE LH + 7–9 (n = 3)\\nAffymetrix Genechip Hu95A |FC|> 2 695 [17]\\nDiaz‑Gimeno Urinary LH PR LH + 1–5 (n = 15) vs R\\nAgilent custom gene expression microarray |FC| > 2 234 [36]\\nKao Urinary LH; Histologically confirmed LPF d8‑10 (n = 4) vs MSE LH + 8‑10 (n = 7)\\nAffymetrix Genechip Hu95A |FC|> 2 and P‑value <0.05 340 [37]\\nMirkin Urinary LH; Histological dating ESE LH + 3 (n = 3) vs MSE LH + 8 (n = 5)\\nAffymetrix Genechip Hu95A |FC|>2 and adj‑P‑value < 0.05 105 [19]\\nPonnampalam Histological dating EPF (n = 5) MPF (n = 7) LPF (n = 3) ESE (n = 7) MSE (n = 8) LS (n = 7) Mense (n = 6)\\nCustom Adj‑P‑value <0.05 306 [38]\\nPunyadeera Histological dating Mense (n = 2) vs LPF (n=2)\\nAffymetrix HG‑U133A FC and P‑value 50 [39]\\nRiesewijk Urinary LH; Histological dating PR LH + 2 (n = 5) vs R LH + 7 (n = 5)\\nAffymetrix Genechip Hu95A |FC|> 3 in at least four out of five women. 196 [40]\\nTalbi Histological dating P (n = 6) vs ESE (n = 3) vs MSE (n = 8) vs LSE (n = 6)\\nAffymetrix HG‑U133 Plus 2.0 |FC|> 1.5 and adj‑P‑value < 0.05 317 [4]',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Implantation failure Altmäe2010 Urinary LH UI (n = 4) vs C (n = 5) Whole Human Genome Oligo Microarray (Agilent Technologies) |FC|> 2 and PFP < 0.05 184 [21] Bastu 6‑7 days after LH surge IF (n = 24) vs Fertile women (n = 24) Agilent‑039494 SurePrint G3 Human GE v2 8x60K Log2(FC) ≥ 1 and adj‑P‑value ≤0.05 |FC|> 2 and PFP < 0.05 524 [41] Bersinger N/A IF (n = 3) vs M (n = 3) vs OP (n = 3) Affymetrix Human Genome U133A 2.0 Chips Not specified 93 [42] Bhagwat N/A PR vs R Multiple platforms Based on literature 177 [43] Koot Urinary LH RIF (n = 43) vs C (n = 72) Human whole genome gene expres‑sion microarrays V2 (Agilent, Belgium) Machine learning predictor 310 [44] Lédée Ultrasound evaluation IF (n = 30) vs Fertile women (n = 15) Affymetrix Human Genome U133 Plus 2.0 Array Adj‑P‑value <0.01 and Δ > 1.35 322 [45] Pathare 6‑7 days after hCG administration IF (n = 10) vs Healthy oocyte donors (n = 8) Illumina HumanHT‑12 V4.0 expression beadchip |FC|> 2 818 [46] Shi Urinary LH; Ultrasound evaluation',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'IF (n = 10) vs Healthy oocyte donors (n = 8) Illumina HumanHT‑12 V4.0 expression beadchip |FC|> 2 818 [46] Shi Urinary LH; Ultrasound evaluation RIF (n = 12) vs C (n \\n= 10) Agilent‑052909 CBC lncRNAmRNA V3 |FC|≥ 2 and adj‑P‑value ≤ 0.05 281 [47]',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Of the respective miRNAs and TFs targeting genes related to endometrial progression and function, 417 miRNAs and 467 TFs were significantly over- represented in at least one gene list (FDR < 0.05, Tables S3-S4). Further, most signatures were significantly governed by TFs rather than miRNAs (4 > odds ratio > 164, FDR < 0.05, 17 lists [89%]), with the exception of Koot’s signature that was governed by miRNAs and showed the opposite (odds ratio = 0.35, FDR < 0.05, Table S5). Eight out of nineteen gene lists (47%) had significant differences between the proportion of genes regulated by progesterone versus estrogen. (*FDR ≤ 0.05). Descriptive analysis of hormonal regulation, normalized by the size of each gene list. Gene lists clustered according to higher (blue) or lower (orange) contribution of hormonal regulation',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Parraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology (2023) 21:84 \\n\\nHsa‑miR‑15a‑5p, hsa‑miR‑128‑3p, hsa‑miR‑218‑5p, \\nhsa‑miR‑27b‑3p, hsa‑miR‑107, hsa‑miR‑424‑5p, \\nhsa‑miR‑195‑5p, hsa‑miR‑103a‑3p, has‑let‑7b‑5p and hsa‑miR‑22‑3p are key players in the regulation of endometrial progression and function\\n\\nTo bridge the gap of endometrial-based evidence in DoRothEA and TarBase databases, we aimed to identify the common miRNAs that mediate gene expression related to endometrial progression and function. We evaluated the number of endometrial gene lists that each miRNA controlled (Fig. 2A). This strategy highlighted the most influential miRNAs that were commonly found across the endometrial gene lists. Notably, 85.80% of miRNAs regulated genes from one to three signatures, and only 59 miRNAs (14.20%) were associated with more than three signatures. Ultimately, 24 miRNAs were prioritized for regulating genes of at least six signatures.\\n\\nNext, we validated the gene expression trends for these 24 prioritized miRNAs in an independent endometrial gene expression dataset (GSE44730) (Fig. 2B). This dataset included the miRNA expression in endometrial biopsies from 20 healthy women in different phases of the menstrual cycle. We found that two of our prioritized miRNAs were not expressed in the endometrial tissue, and only 10 miRNAs (i.e., hsa-miR-15a-5p, hsa-miR-128-3p, hsa-miR-218-5p, hsa-miR-27b-3p, hsa-miR-107, hsa-miR-424-5p, hsa-miR-195-5p, hsa-miR-103a-3p, has-let-7b-5p and hsa-miR-22-3p) showed significant changes across the menstrual cycle (Fig. 2B). Of these, only the expression of hsa-miR-22-3p significantly increased from ESE to MSE stages (FDR = 0.02, Fig. 2B), highlighting the potential inhibitory role of this miRNA in endometrial function. The expression of the remaining miRNAs decreased during the MSE phase.\\n\\nCTCF and GATA6 as distinguished transcriptional regulators of the human menstrual cycle\\n\\nWe repeated the regulation network analysis with the TFs, to identify the prominent TFs commonly regulating the genes related to endometrial progression and function in the reported signatures. We observed 78.80% of the TFs regulated genes in up to 10 signatures, and only 99 TFs (21.20%) regulated genes in over 10 (Fig. 3A). We highlighted CTCF (a CCCTC-binding factor, which functions as a transcriptomic repressor) as the TF with the most influential regulation of endometrial progression and function across 95% of studies (18 gene lists, excluding that of Punyadeera, Fig. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53. Notably, GATA6 was selected due to novel association with endometrial progression and function.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'Parraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology (2023) 21:84'},\n",
       " {'text': 'To infer the expression of CTCF and GATA6  in \\nendometrial tissue throughout the menstrual cycle, \\nwe analysed their expression using a relevant endo - \\nmentrial dataset recently created by our group [30] \\n(Table S1 ). We observed that, relative to the global \\nexpression of endometrial genes, CTCF was broadly \\nexpressed while GATA6  was moderately expressed. \\nIndeed, CTCF was significantly down-regulated \\nbetween the PF and ESE phases (FDR = 2.10E-03)  \\nand ESE to the MSE phase (FDR = 2.10E-04), while \\nGATA6 expression increased from the PF to the \\nMSE phase (FDR = 1.70E-11) and from the ESE  \\nto MSE phase (FDR = 9.80E-15). These findings \\nsupported the role of these TFs at the beginning \\nof the menstrual cycle and during the MSE phase \\n(Fig.\\xa03 B). \\nThese in-silico  validations were then prospec - \\ntively corroborated by RT-qPCR analysis in an inde - \\npendent sample set of 20 endometrial biopsies. We \\nobserved a similar expression profile of CTCF and \\nGATA6 throughout the menstrual cycle, with signifi-\\n',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with significant changes between the endometrial phases. Notably, CTCF expression was decreased between the PF/ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C). The prioritized miRNAs included has‑miR‑16‑5p, has‑miR‑138‑5p, has‑miR‑21‑3p, has‑miR‑205‑5p, has‑miR‑27a‑5p, has‑miR‑15a‑5p, has‑miR‑155‑5p, has‑miR‑129–2‑3p, has‑miR‑147a, has‑miR‑146a‑5p, has‑miR‑107, has‑miR‑424‑5p, has‑miR‑203a‑3p, has‑miR‑195‑5p, has‑miR‑124‑3p, has‑miR‑128‑3p, has‑miR‑22‑3p, has‑miR‑27b‑3p, has‑let‑7b‑5p, has‑miR‑1343‑3p, has‑miR‑1‑3p, has‑miR‑103a‑3p, has‑miR‑23b‑3p, has‑miR‑218‑5p. Significant changes between phases are denoted with asterisks (* FDR ≤ 0.05, ** FDR ≤ 0.01.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Fig. 2 (See legend on previous page.)',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Page 9 of 14\\nParraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology (2023) 21:84 \\n\\nFig. 3 Transcription factors universally regulating endometrial progression and function and their expression throughout the menstrual cycle. A Regulatory network of TFs forming relationships with numerous gene lists (2–18). TFs targeting specific gene lists were placed in the center of the network. CTCF stood out as the only TF that targeted genes belonging to 18/19 (95%) different signatures. B In-silico analysis of CTCF and GATA6 expression throughout the menstrual cycle. Significant changes between phases are marked with asterisks (**FDR ≤ 0.01, ***FDR ≤ 0.001). C RT‑qPCR validation. Bar plot shows the relative mRNA expression of CTCF and GATA6 in endometrial biopsies collected throughout the menstrual cycle. Data is presented as a mean ± SD. Significant differences of CTCF (*FDR < 0.05) and GATA6 (***FDR < 0.001) expression were observed between proliferative/early‑secretory (PF/ESE) and mid‑secretory/late‑secretory phases (MSE/LSE). Abbreviations in (B and C): PF, Proliferative; ESE, early‑secretory; MSE, mid‑secretory; LSE, late‑secretory; SD, standard deviation',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'This data-driven approach exposed the common transcriptional regulators among 19 studies who proposed variable biomarkers of endometrial progression and function. In this study, we focused on understanding the relative contribution of both the hormonal and non-hormonal regulation, from an alternative holistic perspective. We applied data-driven hypothesis research that, unlike the traditional scientific method, allowed us to generate new hypotheses based on all available biologically-relevant knowledge [16], observing the molecular relationships from a wider scale view. Besides, some traditional molecular procedures such as PCRs are also performed to corroborate the insight uncovered with this approach. We highlighted a larger influence of progesterone-related genes and TFs in endometrial progression and observed a larger contribution of miRNAs and estrogen-related genes in endometrial pathology. Furthermore, we unveiled CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-miR-103a-3p and hsa-miR-128-3p as master regulators in the endometrium, and these findings were reinforced by the dynamic changes in their expression throughout the menstrual cycle (dually validated with in-silico and prospective analyses). The menstrual cycle is perceived to be under tight regulation by the ovarian hormones, progesterone and estrogen [48]. Supporting the role of progesterone in the secretory endometrium [49], and reinforcing its indispensable role in endometrial regulation [50, 51], we found progesterone and its related genes had a larger influence in menstrual cycle progression than estrogen. As the RIF-related gene lists reported by Altmae2010 and Koot were generally controlled by estrogen, this further suggested that while progesterone could be crucial for a suitable endometrial progression and mid-secretory phase acquisition, estrogenic irregularities may be more conducive to endometrial function and molecular pathology. Indeed, this hypothesis corroborated previous reports of most uterine disorders being estrogen-dependent [44, 52]. Supporting the findings from another recent study from our group, where the Altmae2010 and Koot signatures were classified as highly-predictive signatures to identify endometrial pathology in comparison with a prediction model that detected endometrial progression [7 ]. This study distinguished these signatures (related to implantation failure and unexplained infertility) based on the magnitude of their miRNA and estrogen-regulation, which contrasted with the prominent influence of the TFs over the remaining signatures more related to endometrial progression. All these facts reinforce the hypothesis that implantation failure is mainly caused by impaired miRNA expression [53].',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'This study transcended the simplistic paradigm of the bihormonal regulation of endometrial function by identifying a plethora of mediators, including 595 miRNAs and 311 TFs, that were previously unrelated to these ovarian hormones. These findings set the foundation for new discoveries, revealing alternative pathways and new actors in the regulation of the human endometrium, and deepening our understanding of the complex regulatory mechanisms behind endometrial progression and unexplained infertility. MicroRNAs are well known to be involved in embryo implantation and receptivity control. Using network analysis, we unveiled the 10 common miRNAs regulating endometrial progression and function across published biomarker signatures (hsa-miR-22-3p, hsa-miR-107, hsa-miR-103a-3p, hsa-miR-128-3p, hsa-miR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, hsa-miR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p).',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Their significant gene expression changes in the secre - \\ntory endometrium support their potential role in the \\nendometrial regulation through an inhibition during \\nthe WOI. Notably, hsa-miR-424-5p, hsa-miR-27b-3p \\nand hsa-miR-195-5p have been previously associated \\nwith RIF [11, 54, 56], while has-let-7b-5p was associ - \\nated with human endometrial receptivity [57], and \\nhsa-miR-22-3p was overexpressed in a decidualization \\nmodel [58]. Moreover, hsa-miR-15a-5p was previously \\nassociated with endometriosis [59]. Notwithstanding, \\nthis study is the first to associate hsa-miR-15a-5p, hsa- \\nmiR-218-5p, hsa-miR-103a-3p, hsa-miR-107, and hsa- \\nmiR-128-3p to endometrial progression and functional \\nregulation in humans.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Page 11 of 14\\nParraaga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \\n\\nendometrial progression and function were beyond the scope of this project and merit further investigation. CTCF is a conserved zinc finger protein whose regulatory functions are well characterized throughout the human body [64]. CTCF acts as a transcriptional repressor in RNA polymerase II (Pol II) pausing and imprinting and X-chromosome inactivation [65], as well as an insulator, blocking the interaction between enhancers and the promoters of neighbouring genes [66]. Considering these attributed functions and its variable expression throughout the menstrual cycle, we propose that CTCF exerts an inhibitory role in endometrial tissue during the PF phase of the menstrual cycle. With the significant downregulation of CTCF in the secretory endometrium, dually validated in-silico and experimentally herein, its inhibited genes would be derepressed and become transcriptionally active during the WOI. This interpretation supports previous findings from our group demonstrating that, during the WOI, a global transcriptional derepression may be required for implantation and early embryo development [63]. Transcriptional derepression has been associated with multiple human disease states and should be investigated further within the context of endometrial-factor infertility. Despite previous associations of CTCF with endometriosis [67], its implication in endometrial progression and function has not been directly proposed until now. Nevertheless, the role of CTCF in endometrial receptivity has been revealed through its interaction with HOXA10, a gene mainly expressed in endometrium involved in functions such as endometrial proliferation and differentiation, the formation of pinopodes or embryo implantation [68]. Indeed, this study proposed that an overexpression of CTCF can lead to a drop in HOXA10 expression, affecting endometrial proliferation and endometrial function and corroborating the fact that CTCF must decrease its expression as we can observe in our findings. GATA6, the second prioritized TF that overlapped across the gene lists, is a transcription factor belonging to the GATA family, a highly conserved family of six zinc finger proteins [69]. Besides its essential role in embryonic development [70], GATA6 plays a key role in endometriosis and regulates steroidogenic genes [71]. Together with evidence supporting both an activating and inhibitory role for GATA6 [72, 73], the significant changes we observed in its expression across the menstrual cycle reflect its presumed involvement in the endometrium and suggest an activating role in endometrial progression and endometrial receptivity acquisition. Like CTCF, this is the first time that GATA6 is proposed as a key TF in endometrial progression and function.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'endometrial regulation remain elusive, emerging experimental technologies such as High-throughput Chromosome Conformation Capture (Hi-C) or chromatin interaction analysis with paired-end tags (ChIA-PET) that respectively study general chromatin interactions or chromatin interactions involving a specific protein [74, 75] may improve our understanding of genetic regulation in the future. Nevertheless, as collaborative efforts unveil new regulators and gene targets in humans (such as the ENCODE project [76]), the complexities of transcriptional regulation can be exposed by in-silico analyses like those described herein, and we can gain a better understanding of the processes involved in menstrual cycle progression and endometrial competence. Furthermore, it should be noted that results of gene expression studies are influenced by different variables, such as the experimental platforms employed, study designs, or patients’ characteristics of each independent',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'expression studies are influenced by different variables, such as the experimental platforms employed, study designs, or patients’ characteristics of each independent study, adding variability [30, 77, 78]. Despite this fact, we have found that our holistic approach overcomes this undesired variability effect by finding common upstream master regulators across studies and predicting new relationships between hormones, TFs and miRNAs. This approach also provides the basis for future single-molecule studies that aim to elucidate new regulatory pathways in endometrial progression and function. Our results lay the foundation for further molecular studies that can validate the function(s) of the prevailing endometrial regulators we prioritized. The study shows for the first time, the relative contribution of estrogens, progesterone, TFs and miRNAs in endometrial function and progression. Endometrial progression is mainly influenced by progesterone-related genes and TFs, whereas miRNAs and estrogen-related genes play a larger role in endometrial pathology. Moreover, we highlight novel common transcriptional regulators such as CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-miR-103a-3p, and hsa-miR-128-3p across 19 studies that propose biomarkers in endometrial regulation. These results reveal the molecular mechanism underlying endometrial regulation and lay the foundation for the development of targeted therapies for patients with endometrial-factor infertility.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'We reviewed 19 publications related to endometrial progression and function, extracted the gene signatures from the original publication, and created corresponding gene lists that we could annotate. We performed different functional analyses based on the type of regulator (i.e., ovarian hormones, transcription factors and miRNAs). We used online databases to obtain the information of the regulatory process, at the hormonal (Gene Ontology, Kyoto Encyclopedia of Genes and Genomes, Dorothea), transcription factor (Dorothea) and miRNA (Tarbase) level and analysed the results accordingly. Finally, we employed an integrative analysis and prioritized the most relevant regulators for subsequent validation. TF, Transcription factors; miRNA, microRNA.',\n",
       "  'doi': 'https://doi.org/10.1186/s12958-023-01131-4',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Additional file 5: Supplementary Table S4. Results of TFs enrichment. Excel file with the over‑representation analysis results of the transcription factors (TFs). Each sheet contains the results of a specific list of genes. In each sheet, the following information is included: ID, gene symbol of the TF; GeneRatio, the proportion of TF targets in the gene list; BgRatio, the proportion of TF targets in the whole genome; OddsRatio; the odd ratios for each TF, pvalue; the P‑value obtained for each TF; padjust, P‑value adjusted by FDR for each TF; geneID, the gene symbol ID of the TF targets. Additional file 6: Supplementary Table S5. Relative contribution of each regulator type (TFs or miRNAs) in each gene list. A Fisher’s test was performed to calculate the proportional differences between transcription factors (TFs) and microRNAs (miRNAs). Resulting P‑values were adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Resulting P‑values were adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene lists were mainly regulated by TFs, whereas odds ratios < 1 indicate a higher regulation by miRNAs. Odds Ratio with a value of “Inf” indicate that miRNAs were not found for this gene list. Percentages indicate the proportion between the significant regulators in each gene signature and regulators with at least one target gene in the corresponding signature. (***FDR ≤0.001).',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'The data used to analyze the TF and miRNA expression throughout the men‑\\ntstrual cycle and in infertility are available in Gene Expression Omnibus, with \\nIDs GSE98386, GSE29981, GSE4888, GSE119209, GSE86491, and GSE44558. The raw endometrial gene expression data (from previous studies) was downloaded from the publicly‑available functional genomic dataset repository Gene Expression Omnibus (GEO) [79] for curation/processing. In accordance with GEO policies, all patient data was anonymized and encrypted, and therefore an additional institutional review board approval was not required for our in-silico validation of regulator gene expression changes in endometrial tissue.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'tional institutional review board approval was not required for our in-silico validation of regulator gene expression changes in endometrial tissue. On the other hand, for the experimental validation of gene expression changes in endometrial biopsies, all individuals gave written informed consent for endometrial gene expression evaluation prior to participating (Ethics Committee of the Instituto Valenciano de Infertilidad [IVI Valencia], Valencia, Spain [1810‑FIVI‑066‑PD/1706‑FIVI‑048‑PD]).',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'Ethics Committee of the Instituto Valenciano de Infertilidad [IVI Valencia], Valencia, Spain [1810‑FIVI‑066‑PD/1706‑FIVI‑048‑PD]'},\n",
       " {'text': 'Parraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology (2023) 21:84 \\n\\nAuthor details\\n1 IVIRMA Global Research Alliance, IVI Foundation, Instituto de Investigación Sanitaria La Fe (IIS La Fe), Av. Fernando Abril Martorell 106, Torre A, Planta 1ª, 46026 Valencia, Valencia, Spain. 2 Department of Pediatrics, Obstetrics and Gynaecology, Universidad de Valencia, Av. Blasco Ibáñez 15, 46010 Valen‑cia, Valencia, Spain. 3 IVIRMA Global Research Alliance, IVIRMA Valencia, Plaza de La Policia Local 3, 46015 Valencia, Spain.\\nReceived: 21 June 2023 Accepted: 22 August 2023\\n\\nReferences\\n1. Strowitzki T, Germeyer A, Popovici R, von Wolff M. The human endometrium as a fertility-determining factor. Hum Reprod Update. 2006;12:617–30.\\n2. Cakmak H, Taylor HS. Implantation failure: Molecular mechanisms and clinical treatment. Hum Reprod Update. 2011;17:242–53.\\n3. Lessey BA, Young SL. What exactly is endometrial receptivity? Fertil Steril. 2019;111:611–7.\\n4. Talbi S, Hamilton AE, Vo KC, Tulac S, Overgaard MT, Dosiou C, et al. Molecular phenotyping of human endometrium distinguishes menstrual cycle phases and underlying biological processes in normo‑ovulatory women. Endocrinology. 2006;147:1097–121.\\n5. Somigliana E, Vigano P , Busnelli A, Paffoni A, Vegetti W, Vercellini P . Repeated implantation failure at the crossroad between statistics, clinics and over‑diagnosis. Reprod Biomed Online. 2018;36:32–8.\\n6. Craciunas L, Gallos I, Chu J, Bourne T, Quenby S, Brosens JJ, et al. Conventional and modern markers of endometrial receptivity: a systematic review and meta‑analysis. Hum Reprod Update. 2019;25:202–23.\\n7. Sebastian‑Leon P , Garrido N, Remohí J, Pellicer A, Diaz‑Gimeno P . Asynchronous and pathological windows of implantation: Two causes of recurrent implantation failure. Hum Reprod. 2018;33:626–35.\\n8. Vasquez YM, DeMayo FJ. Role of nuclear receptors in blastocyst implantation. Semin Cell Dev Biol. 2013;24:724–35.\\n9. Wang H, Zhang S, Lin H, Kong S, Wang S, Wang H, et al. Physiological and molecular determinants of embryo implantation. Mol Aspects Med. 2013;34:939–80.\\n10. Kelleher AM, Behura SK, Burns GW, Young SL, Demayo FJ, Spencer TE. Integrative analysis of the forkhead box A2 (FOXA2) cistrome for the human endometrium. FASEB J. 2019;33:8543–54.\\n11. Revel A, Achache H, Stevens J, Smith Y, Reich R. MicroRNAs are associated with human embryo implantation defects. Hum Reprod. 2011;26:2830–40.\\n12. Galliano D, Pellicer A. MicroRNA and implantation. Fertil Steril. 2014;101:1531–44.\\n13. Xu H, Zhou M, Cao Y, Zhang D, Han M, Gao X, et al. Genome‑wide analysis of long noncoding RNAs, microRNAs, and mRNAs forming a competing endogenous RNA network in repeated implantation failure. Gene. 2019;720: 144056.\\n14. Copley RR. The animal in the genome: comparative genomics and evolution. Philos Trans R Soc Lond B: Biol Sci. 2008;363:1453–61.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Breaking the ageing paradigm in endometrium: endometrial gene expression related to cilia and ageing hallmarks in women over 35 years.',\n",
       "  'doi': '37:762–76',\n",
       "  'citation': 'Devesa‑Peiro A, Sebastian‑Leon P , Parraga‑Leo A, Pellicer A, Diaz‑Gimeno P .'},\n",
       " {'text': 'Meta‑signature of human endometrial receptivity: a meta‑analysis and validation study of transcriptomic biomarkers.',\n",
       "  'doi': 'https:// doi. org/ 10. 1101/ gr. 240663.',\n",
       "  'citation': 'Altmäe S, Koel M, Võsa U, Adler P , Suhorutšenko M, Laisk‑Podar T, et al. Scient Rep. 2017;7:10077.'},\n",
       " {'text': 'Chapter 15 ‑ the molecular signature of the endometrial receptivity: research and clinical application. In: Horcajadas JA, Gosálvez J, editors. Reproductomics: the omics revolu‑tion and its impact on human reproductive medicine. Academic Press; 2018. p. 301–279.\\n\\n36. Díaz‑Gimeno P , Horcajadas JA, Martínez‑Conejero JA, Esteban FJ, Alamá P , Pellicer A, et al. A genomic diagnostic tool for human endometrial receptivity based on the transcriptomic signature. Fertil Steril. 2011;95:50.\\n\\n37. Kao LC, Tulac S, Lobo S, Imani B, Yang JP , Germeyer A, et al. Global gene profiling in human endometrium during the window of implantation. Endocrinology. 2002;143:2119–38.\\n\\n38. Ponnampalam AP , Weston GC, Trajstman AC, Susil B, Rogers PAW. Molecular classification of human endometrial cycle stages by transcriptional profiling. Mol Hum Reprod. 2004;10:879–93.\\n\\n39. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'lar classification of human endometrial cycle stages by transcriptional profiling. Mol Hum Reprod. 2004;10:879–93. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al. Oestrogen‑modulated gene expression in the human endometrium. Cell Mol Life Sci. 2005;62:239–50. Riesewijk A, Martín J, van Os R, Horcajadas JA, Polman J, Pellicer A, et al. Gene expression profiling of human endometrial receptivity on days LH+2 versus LH+7 by microarray technology. Mol Hum Reprod. 2003;9:253–64. Bastu E, Demiral I, Gunel T, Ulgen E, Gumusoglu E, Hosseini MK, et al. Potential marker pathways in the endometrium that may cause recurrent implantation failure. Reprod Sci. 2019;26:879–90. Bersinger NA, Wunder DM, Birkhäuser MH, Mueller MD. Gene expression in cultured endometrium from women with different outcomes follow‑following IVF. Mol Hum Reprod. 2008;14:475–84.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'Page 14 of 14Parraga‑Leo\\xa0et\\xa0al. Reproductive Biology and Endocrinology           (2023) 21:84 \\n•\\n\\nfast, convenient online submission\\n •\\n\\nthorough peer review by experienced researchers in your ﬁeld\\n• \\n\\nrapid publication on acceptance\\n• \\n\\nsupport for research data, including large and complex data types\\n\\ngold Open Access which fosters wider collaboration and increased citations\\n\\nmaximum visibility for your research: over 100M website views per year •\\n  At BMC, research is always in progress. Learn more biomedcentral.com/submissions\\nReady to submit y our researc hReady to submit y our researc h  ?  Choose BMC and benefit fr om: ?  Choose BMC and benefit fr om: 43. Bhagwat SR, Chandrashekar DS, Kakar R, Davuluri S, Bajpai AK, Nayak S, et al. Endometrial receptivity: a revisit to functional genomics studies on human endometrium and creation of HGEx ‑ERdb. PLoS ONE. 2013;8: e58419.\\n44. Koot YEM, Van Hooff SR, Boomsma CM, Van Leenen D, Koerkamp MJAG, Goddijn M, et al. An endometrial gene expression signature accurately predicts recurrent implantation failure after IVF. Sci Rep. 2015;2016(6):1–12.\\n45. Lédée N, Munaut C, Aubert J, Sérazin V, Rahmati M, Chaouat GG, et al. Specific and extensive endometrial deregulation is present before conception in IVF/ICSI repeated implantation failures (IF) or recurrent miscarriages. J Pathol. 2011;225:554–64.\\n46. Pathare ADS, Zaveri K, Hinduja I. Downregulation of genes related to immune and inflammatory response in IVF implantation failure cases under controlled ovarian stimulation. Am J Reprod Immunol. 2017;78: e12679.\\n47. Shi C, Han HJ, Fan LJ, Guan J, Zheng XB, Chen X, et al. Diverse endometrial mRNA signatures during the window of implantation in patients with repeated implantation failure. Hum Fertil. 2018;21:183–94.\\n48. Messinisi IE. Ovarian feedback, mechanism of action and possible clinical implications. Hum Reprod Update. 2006;12:557–71.\\n49. Edwards RG. Implantation, interception and contraception. Hum Reprod (Oxford, England). 1994;9:985–95.\\n50. Young SL. Oestrogen and progesterone action on endometrium: a translational approach to understanding endometrial receptivity. Reprod Biomed Online. 2013;27:497–505.\\n51. Bhurke AS, Bagchi IC, Bagchi MK. Progesterone‑regulated endometrial factors controlling implantation. Am J Reprod Immunol. 2016;75:237–45.\\n52. Devesa‑Peiro A, Sebastian‑Leon P , Garcia‑ Garcia F, Arnau V, Aleman A, Pellicer A, et al. Uterine disorders affecting female fertility: what are the molecular functions altered in endometrium? Fertil Steril. 2020;113:1261–74.\\n53. von Grothusen C, Frisendahl C, Modhukur V, Lalitkumar PG, Peters M, Faridani OR, et al. Uterine fluid microRNAs are dysregulated in women with recurrent implantation failure. Hum Reprod. 2022;37:734–46.\\n54. Altmäe S, Martinez‑Conejero JA, Esteban FJ, Ruiz‑Alonso M, Stavreus‑Evers A, Horcajadas JA, et al. MicroRNAs miR‑30b, miR‑30d, and miR‑494',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'regulate human endometrial receptivity.',\n",
       "  'doi': 'UNKNOWN',\n",
       "  'citation': 'Reprod Sci. 2013;20:308–17.'},\n",
       " {'text': 'stromal decidualization in vitro. PLoS ONE. 2012;7:41080. Liu D, Liang Y, Chen M, Yang F, Yao S. Knockdown of circ_0075503 sup ‑ presses cell migration and invasion by regulating miR‑15a‑5p and KLF12 in endometriosis. Mol Cell Biochem. 2021;476:3845–56. Chi RPA, Wang T, Adams N, Wu SP , Young SL, Spencer TE, et al. Human endometrial transcriptome and progesterone receptor cistrome reveal important pathways and epithelial regulators. J Clin Endocrinol Metab. 2020;105:1–21. Pawar S, Starosvetsky E, Orvis GD, Behringer RR, Bagchi IC, Bagchi MK. STAT3 regulates uterine epithelial remodeling and epithelial‑stromal crosstalk during implantation. Mol Endocrinol. 2013;27:1996–2012. Cheng YH, Imir A, Suzuki T, Fenkci V, Yilmaz B, Sasano H, et al. SP1 and SP3 mediate progesterone‑ dependent induction of the 17beta hydroxyster ‑ oid dehydrogenase type 2 gene in human endometrium. Biol Reprod. 2006;75:605–14. Sebastian‑Leon P , Devesa‑Peiro A, Aleman A, Parraga‑Leo A, Arnau V, Pellicer A, et al. Transcriptional changes through menstrual cycle reveal a global transcriptional derepression underlying the molecular mechanism involved in the window of implantation. Mol Hum Reprod. 2021. https:// doi. org/ 10. 1093/ molehr/ gaab0 27. Kim S, Yu NK, Kaang BK. CTCF as a multifunctional protein in genome regulation and gene expression. Exp Mol Med. 2015;47: e166. Shukla S, Kavak E, Gregory M, Imashimizu M, Shutinoski B, Kashlev M, et al. CTCF‑promoted RNA polymerase II pausing links DNA methylation to splicing. Nature. 2011;479:74–9. Phillips JE, Corces VGCTCF. Master weaver of the genome. Cell. 2009;137:1194–211. Guo J, Cao B, Xu X, Wu F, Zhu B. Novel CTCF mutations in Chinese patients with ovarian endometriosis. Mol Med Rep. 2018;18:1031–6. Geng J, Cui C, Yin Y, Zhao Y, Zhang C. LncRNA NEAT1 affects endome ‑ trial receptivity by regulating HOXA10 promoter activity. Cell Cycle. 2022;21:1932–44. Viger RS, Guittot SM, Anttonen M, Wilson DB, Heikinheimo M. Role of the GATA family of transcription factors in endocrine development, function, and disease. Mol Endocrinol. 2008;22:781. Xuan S, Sussel L. GATA4 and GATA6 regulate pancreatic endoderm identity through inhibition of hedgehog signaling. Development (Cam‑ bridge). 2016;143:780–6. Bernardi LA, Dyson MT, Tokunaga H, Sison C, Oral M, Robins JC, et al. The essential role of GATA6 in the activation of estrogen synthesis in endome ‑ triosis. Reprod Sci. 2019;26:60. Yin F, Herring BP . GATA‑6 can act as a positive or negative regula ‑ tor of smooth muscle‑specific gene expression *. J Biol Chem. 2005;280:4745–52. Zhai J, Luo G. GATA6‑induced FN1 activation promotes the proliferation, invasion and migration of oral squamous cell carcinoma cells. Mol Med ‑ Rep. 2022;25:102. Fullwood MJ, Liu MH, Pan YF, Liu J, Xu H, Mohamed YB, et al. An oestrogen‑receptor‑α‑bound human chromatin interactome. Nature. 2009;462:58–64.',\n",
       "  'doi': 'https://doi.org/10.1093/molehr/gaab027',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': '75. Lieberman‑Aiden E, Van Berkum NL, Williams L, Imakaev M, Ragoczy T, Telling A, et al. Comprehensive mapping of long range interactions reveals folding principles of the human genome. Science. 2009;326:289.\\n76. Birney E, Stamatoyannopoulos JA, Dutta A, Guigó R, Gingeras TR, Margulies EH, et al. Identification and analysis of functional elements in 1% of the human genome by the ENCODE pilot project. Nature. 2007;447:799–816.\\n77. Tajti F, Kuppe C, Antoranz A, Ibrahim MM, Kim H, Ceccarelli F, et al. A func‑\\n tional landscape of CKD entities from public transcriptomic data. Kidney Int Rep. 2020. https:// doi. org/ 10. 1016/j. ekir. 2019. 11. 005.\\n78. Henarejos‑ Castillo I, Devesa‑Peiro A, de Miguel‑ Gomez L, Sebastian‑Leon P , Romeu M, Aleman A, et al. Predicted COVID ‑19 molecular effects on \\n endometrium reveal key dysregulated genes and functions. Mol Hum Reprod. 2022.gaac035.\\n79. Barrett T, Wilhite SE, Ledoux P , Evangelista C, Kim IF, Tomashevsky M, et al.',\n",
       "  'doi': 'https:// doi. org/ 10. 1016/j. ekir. 2019. 11. 005',\n",
       "  'citation': 'UNKNOWN'},\n",
       " {'text': 'endometrium reveal key dysregulated genes and functions.',\n",
       "  'doi': 'gaac035',\n",
       "  'citation': 'UNKOWN'}]"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "my_json\n",
    "\n",
    "# my_json['author'] = \"parraga-leo\"\n",
    "# my_json['title'] = storage[0].metadata['title']\n",
    "# my_json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a7cfb13b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2455317/1728583740.py:10: LangChainDeprecationWarning: The class `HuggingFaceEmbeddings` was deprecated in LangChain 0.2.2 and will be removed in 1.0. An updated version of the class exists in the :class:`~langchain-huggingface package and should be used instead. To use it run `pip install -U :class:`~langchain-huggingface` and import as `from :class:`~langchain_huggingface import HuggingFaceEmbeddings``.\n",
      "  embedding_function = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")\n",
      "/home/aparraga/anaconda3/envs/RAG/lib/python3.9/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "# load the document and split it into pages\n",
    "loader = PyPDFLoader(\"/data/local/aparraga/Bioinformatician/RAG/Publications/Parraga-Leo2023.pdf\")\n",
    "pages = loader.load_and_split()\n",
    "\n",
    "# split it into chunks\n",
    "text_splitter = CharacterTextSplitter(chunk_size=1000, chunk_overlap=50)\n",
    "storage = text_splitter.split_documents(pages)\n",
    "\n",
    "# create the open-source embedding function\n",
    "embedding_function = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a0b29cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A total of 3,608 distinct genes from the 19 gene lists were associated with endometrial progression\n",
      "The lists’ regulation was significantly favoured by TFs (89% (17/19) of gene lists) and progesterone (47% (8 /19) of gene lists), rather than miRNAs (5% (1/19) of gene lists) or estrogen (0% (0/19) of gene lists), respectively (FDR < 0.05). Exceptionally, two gene lists that were previously associated with implantation failure and unexplained infertility were less hormone‑dependent, but primarily regulated by estrogen. Although endometrial progression genes were mainly targeted by hormones rather than non‑hormonal contributors (odds ratio = 91.94, FDR < 0.05), we identified 311 TFs and 595 miRNAs not previously associated with ovarian hormones. We highlight CTCF, GATA6, hsa‑miR‑15a‑5p, hsa‑miR‑218‑5p, hsa‑miR‑107, hsa‑miR‑103a‑3p, and hsa‑miR‑128‑3p, as overlapping novel master regulators of endometrial function. The gene expression changes of selected regulators through‑out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their\n",
      "out the menstrual cycle (FDR < 0.05), dually validated in-silico and through endometrial biopsies, corroborated their potential regulatory roles in the endometrium.\n",
      "This study revealed novel hormonal and non‑hormonal regulators and their relative contributions to endometrial progression and pathology, providing new leads for the potential causes of endometrial‑factor infertility. The endometrium is the innermost layer of the uterus, which undergoes dynamic histological, physiological and molecular changes that allow it to synchronize with embryo development, facilitate embryo implantation, and ultimately, establish a successful pregnancy. Implantation is a crucial and complex limiting step for conception which occurs between days 19 to 24 of a normal menstrual cycle. Consequently, this period of time is called the window of implantation (WOI) and is characterized by abrupt transcriptomic changes in the endometrial tissue. Alterations in either WOI establishment and endometrial progression may lead to implantation failures, which, along with biochemical miscarriage, account for more than 50% of pregnancy losses at pre-clinical stages. These alterations have broadly been classified as being related to a displaced WOI, caused by variable timing of endometrial progression, or a disrupted WOI, where impaired endometrial function prevents the establishment of an effective WOI. Initially, the WOI was postulated to mainly be regu -lated by the ovarian hormones (namely, estrogen and progesterone) acting via their respective nuclear receptors (i.e., ESR1/2 and PGR). However, evidence has shown that these receptors cooperate with other transcription factors (TFs) and co-regulators to mediate uterine physiology [8]. In fact, TFs (such as homeobox TFs, FOXA2, and KLF9) have recently been reported as key regulators for the establishment of endometrial receptivity [9, 10], and microRNAs (miRNAs) have been found to similarly intervene in this complex and multifactorial process as transcriptional regulators [11]. While the involvement of both types of regulators was widely reported in embryo implantation [12] and other infertility-related diseases, including recurrent implantation failure (RIF) [13], the complex interactions between the different types of regulators and their respective contributions to endometrial progression and function have not been investigated from a systemic point of view. This paradigm shift approaches genes and regulators in an integrative way, by considering how they interact with and coordinate each other to carry out cellular processes, rather than reinforcing the premise they work independently [14]. By more accurately reflecting the biological milieu, systemic approaches take gene-based discoveries to the next level.\n",
      "Previous studies of endometrial transcriptomics have been used to define the WOI, compare the endometrium of healthy women as their menstrual cycles progressed [4, 17–19]; address endometrial differences related to age [20] or the dysfunction of the WOI between patients with implantation failure and healthy controls [21]; and finally, correct for the menstrual cycle bias that can mask important biomarkers [22]. Notwithstanding, the candidate biomarkers of endometrial receptivity reported in each of the aforementioned studies overlap poorly [7]. Indeed, recent comparisons conducted by Sebastian-Leon and colleagues [7] found no congruities between 16 different reported gene lists of endometrial receptivity, and only fair to moderate functional agreements between some of the included signatures. Despite previous transcriptional regulation studies, there have been no holistic studies of endometrial receptivity.\n",
      "fair to moderate functional agreements between some of the included signatures. Despite previous transcriptional regulation studies, there have been no holistic studies of this regulatory process, which robustly analyse the drivers of transcription across a higher number of studies (that each searched for endometrial receptivity biomarkers) and map the relative contributions of each type of regulator to endometrial progression regulation (Ovarian hormones [i.e., estrogen, progesterone], TFs, miRNAs). Thus, the aim of this study was to utilize available genomic data regarding transcriptional regulators to identify overlapping mediators of endometrial transcriptional regulation (i.e., ovarian hormones, TFs, and miRNAs) and determine the relative contribution of each type of regulator among the previously reported gene signatures. These findings could help unveil the master regulators and principal type of endometrial regulation, paving the way for further research aimed at improving woman´s reproductive health. Annotating gene lists associated with endometrial progression and function. Endometrial progression and implantation failure gene lists [7] were retrieved and updated using a public data repository (i.e., Gene Expression Omnibus (GEO)).\n",
      "To study the hormonal regulation of endometrial progression, the Kyoto Encyclopaedia of Genes and Genomes (KEGG) [24], and Gene Ontology (GO) [25] databases were consulted for ovarian hormone-related genes, using relevant keywords (i.e., progesterone, estrogen, oestrogen and estradiol) and selecting those pathways and functions that contained them (Supplementary Fig. S1B). Genes associated with the obtained human KEGG pathways and GO functions were grouped according to whether they were unique to progesterone (P4 gene set) or estrogen (E2 gene set) or related to both hormones (P4 and E2 gene set). Gene targets of the nuclear progesterone (PGR) and estrogen (ESR1, ESR2) receptors were added to their corresponding gene sets using DoRothEA (Discriminant Regulon Expression Analysis) database [26], considering only manually-curated or ChiP-Seq experimentally-validated gene-TF relationships. Finally, the P4 and E2 gene sets were independently mapped to each gene list.\n",
      "Prioritizing key regulators in endometrial progression and function\n",
      "We built regulatory networks to systemically analyse endometrial regulation.\n",
      "We built regulatory networks to systemically analyse endometrial regulation. Nodes represented gene lists and regulators (miRNAs or TFs), while edges indicated significant enrichment among the lists and their corresponding regulators (false discovery rate (FDR) ≤ 0.05). To select the most influential regulators (i.e., those which targeted genes in most of our signatures), we studied the degree distribution of the networks (i.e., the number of gene lists regulated by each molecule) and prioritized the miRNAs and TFs which surpassed the relative maximum (i.e., 1.50 times the interquartile range (IQR)) number of relationships (gene list–regulators). All networks were built and analysed with Cytoscape version 3.7 [28] (Supplementary Fig. S1B). To evaluate how the miRNAs’ expression change in the endometrium throughout the menstrual cycle, we used the raw data from GSE44558 GEO dataset [29], deriving from 20 endometrial samples collected throughout the menstrual cycle: four in early-proliferative (EPF), four in late-proliferative (LPF), four in early-secretory (ESE), four in mid-secretory (MSE) and four in late-secretory (LSE). To evaluate how the TFs’ expression change in the endometrium throughout the menstrual cycle, we analysed the integrated endometrial dataset previously created by our group, and reported in [30], which compiled data from five prior publications (GSE98386, GSE29981, GSE4888, GSE119209 and GSE86491). This dataset included endometrial gene expression\n",
      "Of the 19 gene lists used for our analysis, eleven were obtained from published studies evaluating control patients throughout the menstrual cycle [4, 17–19, 34–40], and the other eight were derived from studies comparing patients with RIF or unexplained infertility to controls [21, 41–47] (Table 1). Unifying all the afore-\n",
      "mentioned signatures, we compiled 3,608 genes related to endometrial progression and function. Hormonal regulation of endometrial progression is largely driven by progesterone. We identified 7,540 and 698 genes related to estrogen and progesterone hormones, respectively. However, as determined by the relative contribution of each type of hormone within each gene list (Fig. 1A), 17/19 (89%) signatures favoured regulation by progesterone rather than estrogen. These differences were significant in 47% of the signatures (FDR < 0.05). Exceptionally, the genes reported by the Altmae2010 and Koot signatures, that were related to unexplained infertility and RIF, had a preference for estrogen regulation (Fig. 1A). With respect to the proportion of progesterone-related genes, Altmae2010 showed 2.1 times more estrogen-related genes while Koot showed 1.7 times more. More over, comparing the proportion of genes regulated by both hormones, between signatures revealed that Alt -\n",
      "We identified a total of 770 miRNAs and 555 TFs regulating at least one of the 3,608 genes reported in the 19 gene lists. Further, we observed that 29% of the miRNAs and 44% of the TFs were associated with the ovarian hormones (i.e., progesterone and/or estrogen). Significantly more genes were targeted by hormonally-influenced regulators rather than non-hormonal regulators (P -value < 0.01, odds ratio =  91.94). Nevertheless, we also identified 595 miRNAs (77.27%) and 311 TFs (56.04%) that were unrelated to progesterone and/or estrogen, highlighting a substantial proportion of hormonally-independent regulators of the menstrual cycle.\n",
      "Table 1 Characteristics of the 19 gene lists included in this study. Gene lists were divided according to their evaluated condition. Each signature was labelled using the name of first author, followed by the publication year in the case of duplicates. Endometrial staging was based on different variables including urinary LH, histological analysis, ultrasound evaluation, or the number of days since the hCG trigger. Participants included healthy controls and women with infertility‑related conditions. Various platforms were used to evaluate gene expression. The thresholds applied for gene refinement and the final number of genes per signature are also presented\n",
      "Abbreviations: Δ delta  Ct, adj-P-value Adjusted P‑value, C Control group, d Day of menstrual cycle, EPF Early proliferative, ESE Early secretory, |FC| Absolute fold‑change, FC Fold change, HRT Hormone replacement therapy, IF Implantation failure, LH Luteinizing hormone, LH + X X days post LH surge, LPF Late proliferative, LSE Late‑secretory, M Miscarriage, MPF Mid‑proliferative, MSE Mid‑secretory, n number of samples, N/A Not applicable, OP Ongoing pregnancy, PF Proliferative, PFP Proportion of false positives, PR Pre‑receptive, R Receptive, RIF Recurrent implantation failure, RRA Robust Rank Aggregation algorithm, SF Spontaneously fertile women, UI Women with unexplained infertility, vs versus\n",
      "Condition Gene signature Endometrial staging Participants Platform Threshold No. genes References\n",
      "Endometrial progres‑ sion\n",
      "Altmäe2017 N/A N/A N/A RRA 57 [34]\n",
      "Borthwick Urinary LH; Histologi‑ cally confirmed PF d9‑11 (n = 5) vs R LH + 6–8 (n = 5)\n",
      "Affymetrix Genechip Hu95A N/A 116 [18]\n",
      "Carrascosa N/A N/A Quantitative RT‑PCR Based on literature 187 [35]\n",
      "Carson Urinary LH; Histologically confirmed ESE LH + 2–4 (n = 3) vs MSE LH + 7–9 (n = 3)\n",
      "Affymetrix Genechip Hu95A |FC|> 2 695 [17]\n",
      "Diaz‑Gimeno Urinary LH PR LH + 1–5 (n = 15) vs R\n",
      "Agilent custom gene expression microarray |FC| > 2 234 [36]\n",
      "Kao Urinary LH; Histologically confirmed LPF d8‑10 (n = 4) vs MSE LH + 8‑10 (n = 7)\n",
      "Affymetrix Genechip Hu95A |FC|> 2 and P‑value <0.05 340 [37]\n",
      "Mirkin Urinary LH; Histological dating ESE LH + 3 (n = 3) vs MSE LH + 8 (n = 5)\n",
      "Affymetrix Genechip Hu95A |FC|>2 and adj‑P‑value < 0.05 105 [19]\n",
      "Ponnampalam Histological dating EPF (n = 5) MPF (n = 7) LPF (n = 3) ESE (n = 7) MSE (n = 8) LS (n = 7) Mense (n = 6)\n",
      "Custom Adj‑P‑value <0.05 306 [38]\n",
      "Punyadeera Histological dating Mense (n = 2) vs LPF (n=2)\n",
      "Affymetrix HG‑U133A FC and P‑value 50 [39]\n",
      "Riesewijk Urinary LH; Histological dating PR LH + 2 (n = 5) vs R LH + 7 (n = 5)\n",
      "Affymetrix Genechip Hu95A |FC|> 3 in at least four out of five women. 196 [40]\n",
      "Talbi Histological dating P (n = 6) vs ESE (n = 3) vs MSE (n = 8) vs LSE (n = 6)\n",
      "Affymetrix HG‑U133 Plus 2.0 |FC|> 1.5 and adj‑P‑value < 0.05 317 [4]\n",
      "Implantation failure Altmäe2010 Urinary LH UI (n = 4) vs C (n = 5) Whole Human Genome Oligo Microarray (Agilent Technologies) |FC|> 2 and PFP < 0.05 184 [21] Bastu 6‑7 days after LH surge IF (n = 24) vs Fertile women (n = 24) Agilent‑039494 SurePrint G3 Human GE v2 8x60K Log2(FC) ≥ 1 and adj‑P‑value ≤0.05 |FC|> 2 and PFP < 0.05 524 [41] Bersinger N/A IF (n = 3) vs M (n = 3) vs OP (n = 3) Affymetrix Human Genome U133A 2.0 Chips Not specified 93 [42] Bhagwat N/A PR vs R Multiple platforms Based on literature 177 [43] Koot Urinary LH RIF (n = 43) vs C (n = 72) Human whole genome gene expres‑sion microarrays V2 (Agilent, Belgium) Machine learning predictor 310 [44] Lédée Ultrasound evaluation IF (n = 30) vs Fertile women (n = 15) Affymetrix Human Genome U133 Plus 2.0 Array Adj‑P‑value <0.01 and Δ > 1.35 322 [45] Pathare 6‑7 days after hCG administration IF (n = 10) vs Healthy oocyte donors (n = 8) Illumina HumanHT‑12 V4.0 expression beadchip |FC|> 2 818 [46] Shi Urinary LH; Ultrasound evaluation\n",
      "IF (n = 10) vs Healthy oocyte donors (n = 8) Illumina HumanHT‑12 V4.0 expression beadchip |FC|> 2 818 [46] Shi Urinary LH; Ultrasound evaluation RIF (n = 12) vs C (n \n",
      "= 10) Agilent‑052909 CBC lncRNAmRNA V3 |FC|≥ 2 and adj‑P‑value ≤ 0.05 281 [47]\n",
      "Of the respective miRNAs and TFs targeting genes related to endometrial progression and function, 417 miRNAs and 467 TFs were significantly over- represented in at least one gene list (FDR < 0.05, Tables S3-S4). Further, most signatures were significantly governed by TFs rather than miRNAs (4 > odds ratio > 164, FDR < 0.05, 17 lists [89%]), with the exception of Koot’s signature that was governed by miRNAs and showed the opposite (odds ratio = 0.35, FDR < 0.05, Table S5). Eight out of nineteen gene lists (47%) had significant differences between the proportion of genes regulated by progesterone versus estrogen. (*FDR ≤ 0.05). Descriptive analysis of hormonal regulation, normalized by the size of each gene list. Gene lists clustered according to higher (blue) or lower (orange) contribution of hormonal regulation\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \n",
      "\n",
      "Hsa‑miR‑15a‑5p, hsa‑miR‑128‑3p, hsa‑miR‑218‑5p, \n",
      "hsa‑miR‑27b‑3p, hsa‑miR‑107, hsa‑miR‑424‑5p, \n",
      "hsa‑miR‑195‑5p, hsa‑miR‑103a‑3p, has‑let‑7b‑5p and hsa‑miR‑22‑3p are key players in the regulation of endometrial progression and function\n",
      "\n",
      "To bridge the gap of endometrial-based evidence in DoRothEA and TarBase databases, we aimed to identify the common miRNAs that mediate gene expression related to endometrial progression and function. We evaluated the number of endometrial gene lists that each miRNA controlled (Fig. 2A). This strategy highlighted the most influential miRNAs that were commonly found across the endometrial gene lists. Notably, 85.80% of miRNAs regulated genes from one to three signatures, and only 59 miRNAs (14.20%) were associated with more than three signatures. Ultimately, 24 miRNAs were prioritized for regulating genes of at least six signatures.\n",
      "\n",
      "Next, we validated the gene expression trends for these 24 prioritized miRNAs in an independent endometrial gene expression dataset (GSE44730) (Fig. 2B). This dataset included the miRNA expression in endometrial biopsies from 20 healthy women in different phases of the menstrual cycle. We found that two of our prioritized miRNAs were not expressed in the endometrial tissue, and only 10 miRNAs (i.e., hsa-miR-15a-5p, hsa-miR-128-3p, hsa-miR-218-5p, hsa-miR-27b-3p, hsa-miR-107, hsa-miR-424-5p, hsa-miR-195-5p, hsa-miR-103a-3p, has-let-7b-5p and hsa-miR-22-3p) showed significant changes across the menstrual cycle (Fig. 2B). Of these, only the expression of hsa-miR-22-3p significantly increased from ESE to MSE stages (FDR = 0.02, Fig. 2B), highlighting the potential inhibitory role of this miRNA in endometrial function. The expression of the remaining miRNAs decreased during the MSE phase.\n",
      "\n",
      "CTCF and GATA6 as distinguished transcriptional regulators of the human menstrual cycle\n",
      "\n",
      "We repeated the regulation network analysis with the TFs, to identify the prominent TFs commonly regulating the genes related to endometrial progression and function in the reported signatures. We observed 78.80% of the TFs regulated genes in up to 10 signatures, and only 99 TFs (21.20%) regulated genes in over 10 (Fig. 3A). We highlighted CTCF (a CCCTC-binding factor, which functions as a transcriptomic repressor) as the TF with the most influential regulation of endometrial progression and function across 95% of studies (18 gene lists, excluding that of Punyadeera, Fig. 3A). Other distinguished TFs that regulated up to 15 gene lists included AR, CEBPA, CEBPB, CEBPD, CREB1, EGR1, ELF3, ESR1, ETV4, FOS, FOXA1, GATA2, GATA3, GATA6, HNF4A, JUN, JUND, NFKB1, NR1H2, NR3C1, RELA, SMAD3, SP1, SPI1, STAT1, STAT3, TCF7L2, TFAP2C and TP53. Notably, GATA6 was selected due to novel association with endometrial progression and function.\n",
      "To infer the expression of CTCF and GATA6  in \n",
      "endometrial tissue throughout the menstrual cycle, \n",
      "we analysed their expression using a relevant endo - \n",
      "mentrial dataset recently created by our group [30] \n",
      "(Table S1 ). We observed that, relative to the global \n",
      "expression of endometrial genes, CTCF was broadly \n",
      "expressed while GATA6  was moderately expressed. \n",
      "Indeed, CTCF was significantly down-regulated \n",
      "between the PF and ESE phases (FDR = 2.10E-03)  \n",
      "and ESE to the MSE phase (FDR = 2.10E-04), while \n",
      "GATA6 expression increased from the PF to the \n",
      "MSE phase (FDR = 1.70E-11) and from the ESE  \n",
      "to MSE phase (FDR = 9.80E-15). These findings \n",
      "supported the role of these TFs at the beginning \n",
      "of the menstrual cycle and during the MSE phase \n",
      "(Fig. 3 B). \n",
      "These in-silico  validations were then prospec - \n",
      "tively corroborated by RT-qPCR analysis in an inde - \n",
      "pendent sample set of 20 endometrial biopsies. We \n",
      "observed a similar expression profile of CTCF and \n",
      "GATA6 throughout the menstrual cycle, with signifi-\n",
      "\n",
      "We observed a similar expression profile of CTCF and GATA6 throughout the menstrual cycle, with significant changes between the endometrial phases. Notably, CTCF expression was decreased between the PF/ESE and MSE/LSE phases (FDR = 0.01) while GATA6 expression increased from the PF/ESE to MSE/LSE phases (FDR = 1.55E-04) (Fig. 3 C). The prioritized miRNAs included has‑miR‑16‑5p, has‑miR‑138‑5p, has‑miR‑21‑3p, has‑miR‑205‑5p, has‑miR‑27a‑5p, has‑miR‑15a‑5p, has‑miR‑155‑5p, has‑miR‑129–2‑3p, has‑miR‑147a, has‑miR‑146a‑5p, has‑miR‑107, has‑miR‑424‑5p, has‑miR‑203a‑3p, has‑miR‑195‑5p, has‑miR‑124‑3p, has‑miR‑128‑3p, has‑miR‑22‑3p, has‑miR‑27b‑3p, has‑let‑7b‑5p, has‑miR‑1343‑3p, has‑miR‑1‑3p, has‑miR‑103a‑3p, has‑miR‑23b‑3p, has‑miR‑218‑5p. Significant changes between phases are denoted with asterisks (* FDR ≤ 0.05, ** FDR ≤ 0.01.\n",
      "Fig. 2 (See legend on previous page.)\n",
      "Page 9 of 14\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \n",
      "\n",
      "Fig. 3 Transcription factors universally regulating endometrial progression and function and their expression throughout the menstrual cycle. A Regulatory network of TFs forming relationships with numerous gene lists (2–18). TFs targeting specific gene lists were placed in the center of the network. CTCF stood out as the only TF that targeted genes belonging to 18/19 (95%) different signatures. B In-silico analysis of CTCF and GATA6 expression throughout the menstrual cycle. Significant changes between phases are marked with asterisks (**FDR ≤ 0.01, ***FDR ≤ 0.001). C RT‑qPCR validation. Bar plot shows the relative mRNA expression of CTCF and GATA6 in endometrial biopsies collected throughout the menstrual cycle. Data is presented as a mean ± SD. Significant differences of CTCF (*FDR < 0.05) and GATA6 (***FDR < 0.001) expression were observed between proliferative/early‑secretory (PF/ESE) and mid‑secretory/late‑secretory phases (MSE/LSE). Abbreviations in (B and C): PF, Proliferative; ESE, early‑secretory; MSE, mid‑secretory; LSE, late‑secretory; SD, standard deviation\n",
      "This data-driven approach exposed the common transcriptional regulators among 19 studies who proposed variable biomarkers of endometrial progression and function. In this study, we focused on understanding the relative contribution of both the hormonal and non-hormonal regulation, from an alternative holistic perspective. We applied data-driven hypothesis research that, unlike the traditional scientific method, allowed us to generate new hypotheses based on all available biologically-relevant knowledge [16], observing the molecular relationships from a wider scale view. Besides, some traditional molecular procedures such as PCRs are also performed to corroborate the insight uncovered with this approach. We highlighted a larger influence of progesterone-related genes and TFs in endometrial progression and observed a larger contribution of miRNAs and estrogen-related genes in endometrial pathology. Furthermore, we unveiled CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-miR-103a-3p and hsa-miR-128-3p as master regulators in the endometrium, and these findings were reinforced by the dynamic changes in their expression throughout the menstrual cycle (dually validated with in-silico and prospective analyses). The menstrual cycle is perceived to be under tight regulation by the ovarian hormones, progesterone and estrogen [48]. Supporting the role of progesterone in the secretory endometrium [49], and reinforcing its indispensable role in endometrial regulation [50, 51], we found progesterone and its related genes had a larger influence in menstrual cycle progression than estrogen. As the RIF-related gene lists reported by Altmae2010 and Koot were generally controlled by estrogen, this further suggested that while progesterone could be crucial for a suitable endometrial progression and mid-secretory phase acquisition, estrogenic irregularities may be more conducive to endometrial function and molecular pathology. Indeed, this hypothesis corroborated previous reports of most uterine disorders being estrogen-dependent [44, 52]. Supporting the findings from another recent study from our group, where the Altmae2010 and Koot signatures were classified as highly-predictive signatures to identify endometrial pathology in comparison with a prediction model that detected endometrial progression [7 ]. This study distinguished these signatures (related to implantation failure and unexplained infertility) based on the magnitude of their miRNA and estrogen-regulation, which contrasted with the prominent influence of the TFs over the remaining signatures more related to endometrial progression. All these facts reinforce the hypothesis that implantation failure is mainly caused by impaired miRNA expression [53].\n",
      "This study transcended the simplistic paradigm of the bihormonal regulation of endometrial function by identifying a plethora of mediators, including 595 miRNAs and 311 TFs, that were previously unrelated to these ovarian hormones. These findings set the foundation for new discoveries, revealing alternative pathways and new actors in the regulation of the human endometrium, and deepening our understanding of the complex regulatory mechanisms behind endometrial progression and unexplained infertility. MicroRNAs are well known to be involved in embryo implantation and receptivity control. Using network analysis, we unveiled the 10 common miRNAs regulating endometrial progression and function across published biomarker signatures (hsa-miR-22-3p, hsa-miR-107, hsa-miR-103a-3p, hsa-miR-128-3p, hsa-miR-195-5p, hsa-miR-218-5p, hsa-miR-15a-5p, hsa-miR-27b-3p, hsa-miR-424-5p, and has-let-7b-5p).\n",
      "Their significant gene expression changes in the secre - \n",
      "tory endometrium support their potential role in the \n",
      "endometrial regulation through an inhibition during \n",
      "the WOI. Notably, hsa-miR-424-5p, hsa-miR-27b-3p \n",
      "and hsa-miR-195-5p have been previously associated \n",
      "with RIF [11, 54, 56], while has-let-7b-5p was associ - \n",
      "ated with human endometrial receptivity [57], and \n",
      "hsa-miR-22-3p was overexpressed in a decidualization \n",
      "model [58]. Moreover, hsa-miR-15a-5p was previously \n",
      "associated with endometriosis [59]. Notwithstanding, \n",
      "this study is the first to associate hsa-miR-15a-5p, hsa- \n",
      "miR-218-5p, hsa-miR-103a-3p, hsa-miR-107, and hsa- \n",
      "miR-128-3p to endometrial progression and functional \n",
      "regulation in humans.\n",
      "Page 11 of 14\n",
      "Parraaga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \n",
      "\n",
      "endometrial progression and function were beyond the scope of this project and merit further investigation. CTCF is a conserved zinc finger protein whose regulatory functions are well characterized throughout the human body [64]. CTCF acts as a transcriptional repressor in RNA polymerase II (Pol II) pausing and imprinting and X-chromosome inactivation [65], as well as an insulator, blocking the interaction between enhancers and the promoters of neighbouring genes [66]. Considering these attributed functions and its variable expression throughout the menstrual cycle, we propose that CTCF exerts an inhibitory role in endometrial tissue during the PF phase of the menstrual cycle. With the significant downregulation of CTCF in the secretory endometrium, dually validated in-silico and experimentally herein, its inhibited genes would be derepressed and become transcriptionally active during the WOI. This interpretation supports previous findings from our group demonstrating that, during the WOI, a global transcriptional derepression may be required for implantation and early embryo development [63]. Transcriptional derepression has been associated with multiple human disease states and should be investigated further within the context of endometrial-factor infertility. Despite previous associations of CTCF with endometriosis [67], its implication in endometrial progression and function has not been directly proposed until now. Nevertheless, the role of CTCF in endometrial receptivity has been revealed through its interaction with HOXA10, a gene mainly expressed in endometrium involved in functions such as endometrial proliferation and differentiation, the formation of pinopodes or embryo implantation [68]. Indeed, this study proposed that an overexpression of CTCF can lead to a drop in HOXA10 expression, affecting endometrial proliferation and endometrial function and corroborating the fact that CTCF must decrease its expression as we can observe in our findings. GATA6, the second prioritized TF that overlapped across the gene lists, is a transcription factor belonging to the GATA family, a highly conserved family of six zinc finger proteins [69]. Besides its essential role in embryonic development [70], GATA6 plays a key role in endometriosis and regulates steroidogenic genes [71]. Together with evidence supporting both an activating and inhibitory role for GATA6 [72, 73], the significant changes we observed in its expression across the menstrual cycle reflect its presumed involvement in the endometrium and suggest an activating role in endometrial progression and endometrial receptivity acquisition. Like CTCF, this is the first time that GATA6 is proposed as a key TF in endometrial progression and function.\n",
      "endometrial regulation remain elusive, emerging experimental technologies such as High-throughput Chromosome Conformation Capture (Hi-C) or chromatin interaction analysis with paired-end tags (ChIA-PET) that respectively study general chromatin interactions or chromatin interactions involving a specific protein [74, 75] may improve our understanding of genetic regulation in the future. Nevertheless, as collaborative efforts unveil new regulators and gene targets in humans (such as the ENCODE project [76]), the complexities of transcriptional regulation can be exposed by in-silico analyses like those described herein, and we can gain a better understanding of the processes involved in menstrual cycle progression and endometrial competence. Furthermore, it should be noted that results of gene expression studies are influenced by different variables, such as the experimental platforms employed, study designs, or patients’ characteristics of each independent\n",
      "expression studies are influenced by different variables, such as the experimental platforms employed, study designs, or patients’ characteristics of each independent study, adding variability [30, 77, 78]. Despite this fact, we have found that our holistic approach overcomes this undesired variability effect by finding common upstream master regulators across studies and predicting new relationships between hormones, TFs and miRNAs. This approach also provides the basis for future single-molecule studies that aim to elucidate new regulatory pathways in endometrial progression and function. Our results lay the foundation for further molecular studies that can validate the function(s) of the prevailing endometrial regulators we prioritized. The study shows for the first time, the relative contribution of estrogens, progesterone, TFs and miRNAs in endometrial function and progression. Endometrial progression is mainly influenced by progesterone-related genes and TFs, whereas miRNAs and estrogen-related genes play a larger role in endometrial pathology. Moreover, we highlight novel common transcriptional regulators such as CTCF, GATA6, hsa-miR-15a-5p, hsa-miR-218-5p, hsa-miR-107, hsa-miR-103a-3p, and hsa-miR-128-3p across 19 studies that propose biomarkers in endometrial regulation. These results reveal the molecular mechanism underlying endometrial regulation and lay the foundation for the development of targeted therapies for patients with endometrial-factor infertility.\n",
      "We reviewed 19 publications related to endometrial progression and function, extracted the gene signatures from the original publication, and created corresponding gene lists that we could annotate. We performed different functional analyses based on the type of regulator (i.e., ovarian hormones, transcription factors and miRNAs). We used online databases to obtain the information of the regulatory process, at the hormonal (Gene Ontology, Kyoto Encyclopedia of Genes and Genomes, Dorothea), transcription factor (Dorothea) and miRNA (Tarbase) level and analysed the results accordingly. Finally, we employed an integrative analysis and prioritized the most relevant regulators for subsequent validation. TF, Transcription factors; miRNA, microRNA.\n",
      "Additional file 5: Supplementary Table S4. Results of TFs enrichment. Excel file with the over‑representation analysis results of the transcription factors (TFs). Each sheet contains the results of a specific list of genes. In each sheet, the following information is included: ID, gene symbol of the TF; GeneRatio, the proportion of TF targets in the gene list; BgRatio, the proportion of TF targets in the whole genome; OddsRatio; the odd ratios for each TF, pvalue; the P‑value obtained for each TF; padjust, P‑value adjusted by FDR for each TF; geneID, the gene symbol ID of the TF targets. Additional file 6: Supplementary Table S5. Relative contribution of each regulator type (TFs or miRNAs) in each gene list. A Fisher’s test was performed to calculate the proportional differences between transcription factors (TFs) and microRNAs (miRNAs). Resulting P‑values were adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene\n",
      "Resulting P‑values were adjusted by False Discovery Rate (FDR). Odds Ratios > 1 indicate gene lists were mainly regulated by TFs, whereas odds ratios < 1 indicate a higher regulation by miRNAs. Odds Ratio with a value of “Inf” indicate that miRNAs were not found for this gene list. Percentages indicate the proportion between the significant regulators in each gene signature and regulators with at least one target gene in the corresponding signature. (***FDR ≤0.001).\n",
      "The data used to analyze the TF and miRNA expression throughout the men‑\n",
      "tstrual cycle and in infertility are available in Gene Expression Omnibus, with \n",
      "IDs GSE98386, GSE29981, GSE4888, GSE119209, GSE86491, and GSE44558. The raw endometrial gene expression data (from previous studies) was downloaded from the publicly‑available functional genomic dataset repository Gene Expression Omnibus (GEO) [79] for curation/processing. In accordance with GEO policies, all patient data was anonymized and encrypted, and therefore an additional institutional review board approval was not required for our in-silico validation of regulator gene expression changes in endometrial tissue.\n",
      "tional institutional review board approval was not required for our in-silico validation of regulator gene expression changes in endometrial tissue. On the other hand, for the experimental validation of gene expression changes in endometrial biopsies, all individuals gave written informed consent for endometrial gene expression evaluation prior to participating (Ethics Committee of the Instituto Valenciano de Infertilidad [IVI Valencia], Valencia, Spain [1810‑FIVI‑066‑PD/1706‑FIVI‑048‑PD]).\n",
      "Parraga‑Leo et al. Reproductive Biology and Endocrinology (2023) 21:84 \n",
      "\n",
      "Author details\n",
      "1 IVIRMA Global Research Alliance, IVI Foundation, Instituto de Investigación Sanitaria La Fe (IIS La Fe), Av. Fernando Abril Martorell 106, Torre A, Planta 1ª, 46026 Valencia, Valencia, Spain. 2 Department of Pediatrics, Obstetrics and Gynaecology, Universidad de Valencia, Av. Blasco Ibáñez 15, 46010 Valen‑cia, Valencia, Spain. 3 IVIRMA Global Research Alliance, IVIRMA Valencia, Plaza de La Policia Local 3, 46015 Valencia, Spain.\n",
      "Received: 21 June 2023 Accepted: 22 August 2023\n",
      "\n",
      "References\n",
      "1. Strowitzki T, Germeyer A, Popovici R, von Wolff M. The human endometrium as a fertility-determining factor. Hum Reprod Update. 2006;12:617–30.\n",
      "2. Cakmak H, Taylor HS. Implantation failure: Molecular mechanisms and clinical treatment. Hum Reprod Update. 2011;17:242–53.\n",
      "3. Lessey BA, Young SL. What exactly is endometrial receptivity? Fertil Steril. 2019;111:611–7.\n",
      "4. Talbi S, Hamilton AE, Vo KC, Tulac S, Overgaard MT, Dosiou C, et al. Molecular phenotyping of human endometrium distinguishes menstrual cycle phases and underlying biological processes in normo‑ovulatory women. Endocrinology. 2006;147:1097–121.\n",
      "5. Somigliana E, Vigano P , Busnelli A, Paffoni A, Vegetti W, Vercellini P . Repeated implantation failure at the crossroad between statistics, clinics and over‑diagnosis. Reprod Biomed Online. 2018;36:32–8.\n",
      "6. Craciunas L, Gallos I, Chu J, Bourne T, Quenby S, Brosens JJ, et al. Conventional and modern markers of endometrial receptivity: a systematic review and meta‑analysis. Hum Reprod Update. 2019;25:202–23.\n",
      "7. Sebastian‑Leon P , Garrido N, Remohí J, Pellicer A, Diaz‑Gimeno P . Asynchronous and pathological windows of implantation: Two causes of recurrent implantation failure. Hum Reprod. 2018;33:626–35.\n",
      "8. Vasquez YM, DeMayo FJ. Role of nuclear receptors in blastocyst implantation. Semin Cell Dev Biol. 2013;24:724–35.\n",
      "9. Wang H, Zhang S, Lin H, Kong S, Wang S, Wang H, et al. Physiological and molecular determinants of embryo implantation. Mol Aspects Med. 2013;34:939–80.\n",
      "10. Kelleher AM, Behura SK, Burns GW, Young SL, Demayo FJ, Spencer TE. Integrative analysis of the forkhead box A2 (FOXA2) cistrome for the human endometrium. FASEB J. 2019;33:8543–54.\n",
      "11. Revel A, Achache H, Stevens J, Smith Y, Reich R. MicroRNAs are associated with human embryo implantation defects. Hum Reprod. 2011;26:2830–40.\n",
      "12. Galliano D, Pellicer A. MicroRNA and implantation. Fertil Steril. 2014;101:1531–44.\n",
      "13. Xu H, Zhou M, Cao Y, Zhang D, Han M, Gao X, et al. Genome‑wide analysis of long noncoding RNAs, microRNAs, and mRNAs forming a competing endogenous RNA network in repeated implantation failure. Gene. 2019;720: 144056.\n",
      "14. Copley RR. The animal in the genome: comparative genomics and evolution. Philos Trans R Soc Lond B: Biol Sci. 2008;363:1453–61.\n",
      "Breaking the ageing paradigm in endometrium: endometrial gene expression related to cilia and ageing hallmarks in women over 35 years.\n",
      "Meta‑signature of human endometrial receptivity: a meta‑analysis and validation study of transcriptomic biomarkers.\n",
      "Chapter 15 ‑ the molecular signature of the endometrial receptivity: research and clinical application. In: Horcajadas JA, Gosálvez J, editors. Reproductomics: the omics revolu‑tion and its impact on human reproductive medicine. Academic Press; 2018. p. 301–279.\n",
      "\n",
      "36. Díaz‑Gimeno P , Horcajadas JA, Martínez‑Conejero JA, Esteban FJ, Alamá P , Pellicer A, et al. A genomic diagnostic tool for human endometrial receptivity based on the transcriptomic signature. Fertil Steril. 2011;95:50.\n",
      "\n",
      "37. Kao LC, Tulac S, Lobo S, Imani B, Yang JP , Germeyer A, et al. Global gene profiling in human endometrium during the window of implantation. Endocrinology. 2002;143:2119–38.\n",
      "\n",
      "38. Ponnampalam AP , Weston GC, Trajstman AC, Susil B, Rogers PAW. Molecular classification of human endometrial cycle stages by transcriptional profiling. Mol Hum Reprod. 2004;10:879–93.\n",
      "\n",
      "39. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al.\n",
      "lar classification of human endometrial cycle stages by transcriptional profiling. Mol Hum Reprod. 2004;10:879–93. Punyadeera C, Dassen H, Klomp J, Dunselman G, Kamps R, Dijcks F, et al. Oestrogen‑modulated gene expression in the human endometrium. Cell Mol Life Sci. 2005;62:239–50. Riesewijk A, Martín J, van Os R, Horcajadas JA, Polman J, Pellicer A, et al. Gene expression profiling of human endometrial receptivity on days LH+2 versus LH+7 by microarray technology. Mol Hum Reprod. 2003;9:253–64. Bastu E, Demiral I, Gunel T, Ulgen E, Gumusoglu E, Hosseini MK, et al. Potential marker pathways in the endometrium that may cause recurrent implantation failure. Reprod Sci. 2019;26:879–90. Bersinger NA, Wunder DM, Birkhäuser MH, Mueller MD. Gene expression in cultured endometrium from women with different outcomes follow‑following IVF. Mol Hum Reprod. 2008;14:475–84.\n",
      "Page 14 of 14Parraga‑Leo et al. Reproductive Biology and Endocrinology           (2023) 21:84 \n",
      "•\n",
      "\n",
      "fast, convenient online submission\n",
      " •\n",
      "\n",
      "thorough peer review by experienced researchers in your ﬁeld\n",
      "• \n",
      "\n",
      "rapid publication on acceptance\n",
      "• \n",
      "\n",
      "support for research data, including large and complex data types\n",
      "\n",
      "gold Open Access which fosters wider collaboration and increased citations\n",
      "\n",
      "maximum visibility for your research: over 100M website views per year •\n",
      "  At BMC, research is always in progress. Learn more biomedcentral.com/submissions\n",
      "Ready to submit y our researc hReady to submit y our researc h  ?  Choose BMC and benefit fr om: ?  Choose BMC and benefit fr om: 43. Bhagwat SR, Chandrashekar DS, Kakar R, Davuluri S, Bajpai AK, Nayak S, et al. Endometrial receptivity: a revisit to functional genomics studies on human endometrium and creation of HGEx ‑ERdb. PLoS ONE. 2013;8: e58419.\n",
      "44. Koot YEM, Van Hooff SR, Boomsma CM, Van Leenen D, Koerkamp MJAG, Goddijn M, et al. An endometrial gene expression signature accurately predicts recurrent implantation failure after IVF. Sci Rep. 2015;2016(6):1–12.\n",
      "45. Lédée N, Munaut C, Aubert J, Sérazin V, Rahmati M, Chaouat GG, et al. Specific and extensive endometrial deregulation is present before conception in IVF/ICSI repeated implantation failures (IF) or recurrent miscarriages. J Pathol. 2011;225:554–64.\n",
      "46. Pathare ADS, Zaveri K, Hinduja I. Downregulation of genes related to immune and inflammatory response in IVF implantation failure cases under controlled ovarian stimulation. Am J Reprod Immunol. 2017;78: e12679.\n",
      "47. Shi C, Han HJ, Fan LJ, Guan J, Zheng XB, Chen X, et al. Diverse endometrial mRNA signatures during the window of implantation in patients with repeated implantation failure. Hum Fertil. 2018;21:183–94.\n",
      "48. Messinisi IE. Ovarian feedback, mechanism of action and possible clinical implications. Hum Reprod Update. 2006;12:557–71.\n",
      "49. Edwards RG. Implantation, interception and contraception. Hum Reprod (Oxford, England). 1994;9:985–95.\n",
      "50. Young SL. Oestrogen and progesterone action on endometrium: a translational approach to understanding endometrial receptivity. Reprod Biomed Online. 2013;27:497–505.\n",
      "51. Bhurke AS, Bagchi IC, Bagchi MK. Progesterone‑regulated endometrial factors controlling implantation. Am J Reprod Immunol. 2016;75:237–45.\n",
      "52. Devesa‑Peiro A, Sebastian‑Leon P , Garcia‑ Garcia F, Arnau V, Aleman A, Pellicer A, et al. Uterine disorders affecting female fertility: what are the molecular functions altered in endometrium? Fertil Steril. 2020;113:1261–74.\n",
      "53. von Grothusen C, Frisendahl C, Modhukur V, Lalitkumar PG, Peters M, Faridani OR, et al. Uterine fluid microRNAs are dysregulated in women with recurrent implantation failure. Hum Reprod. 2022;37:734–46.\n",
      "54. Altmäe S, Martinez‑Conejero JA, Esteban FJ, Ruiz‑Alonso M, Stavreus‑Evers A, Horcajadas JA, et al. MicroRNAs miR‑30b, miR‑30d, and miR‑494\n",
      "regulate human endometrial receptivity.\n",
      "stromal decidualization in vitro. PLoS ONE. 2012;7:41080. Liu D, Liang Y, Chen M, Yang F, Yao S. Knockdown of circ_0075503 sup ‑ presses cell migration and invasion by regulating miR‑15a‑5p and KLF12 in endometriosis. Mol Cell Biochem. 2021;476:3845–56. Chi RPA, Wang T, Adams N, Wu SP , Young SL, Spencer TE, et al. Human endometrial transcriptome and progesterone receptor cistrome reveal important pathways and epithelial regulators. J Clin Endocrinol Metab. 2020;105:1–21. Pawar S, Starosvetsky E, Orvis GD, Behringer RR, Bagchi IC, Bagchi MK. STAT3 regulates uterine epithelial remodeling and epithelial‑stromal crosstalk during implantation. Mol Endocrinol. 2013;27:1996–2012. Cheng YH, Imir A, Suzuki T, Fenkci V, Yilmaz B, Sasano H, et al. SP1 and SP3 mediate progesterone‑ dependent induction of the 17beta hydroxyster ‑ oid dehydrogenase type 2 gene in human endometrium. Biol Reprod. 2006;75:605–14. Sebastian‑Leon P , Devesa‑Peiro A, Aleman A, Parraga‑Leo A, Arnau V, Pellicer A, et al. Transcriptional changes through menstrual cycle reveal a global transcriptional derepression underlying the molecular mechanism involved in the window of implantation. Mol Hum Reprod. 2021. https:// doi. org/ 10. 1093/ molehr/ gaab0 27. Kim S, Yu NK, Kaang BK. CTCF as a multifunctional protein in genome regulation and gene expression. Exp Mol Med. 2015;47: e166. Shukla S, Kavak E, Gregory M, Imashimizu M, Shutinoski B, Kashlev M, et al. CTCF‑promoted RNA polymerase II pausing links DNA methylation to splicing. Nature. 2011;479:74–9. Phillips JE, Corces VGCTCF. Master weaver of the genome. Cell. 2009;137:1194–211. Guo J, Cao B, Xu X, Wu F, Zhu B. Novel CTCF mutations in Chinese patients with ovarian endometriosis. Mol Med Rep. 2018;18:1031–6. Geng J, Cui C, Yin Y, Zhao Y, Zhang C. LncRNA NEAT1 affects endome ‑ trial receptivity by regulating HOXA10 promoter activity. Cell Cycle. 2022;21:1932–44. Viger RS, Guittot SM, Anttonen M, Wilson DB, Heikinheimo M. Role of the GATA family of transcription factors in endocrine development, function, and disease. Mol Endocrinol. 2008;22:781. Xuan S, Sussel L. GATA4 and GATA6 regulate pancreatic endoderm identity through inhibition of hedgehog signaling. Development (Cam‑ bridge). 2016;143:780–6. Bernardi LA, Dyson MT, Tokunaga H, Sison C, Oral M, Robins JC, et al. The essential role of GATA6 in the activation of estrogen synthesis in endome ‑ triosis. Reprod Sci. 2019;26:60. Yin F, Herring BP . GATA‑6 can act as a positive or negative regula ‑ tor of smooth muscle‑specific gene expression *. J Biol Chem. 2005;280:4745–52. Zhai J, Luo G. GATA6‑induced FN1 activation promotes the proliferation, invasion and migration of oral squamous cell carcinoma cells. Mol Med ‑ Rep. 2022;25:102. Fullwood MJ, Liu MH, Pan YF, Liu J, Xu H, Mohamed YB, et al. An oestrogen‑receptor‑α‑bound human chromatin interactome. Nature. 2009;462:58–64.\n",
      "75. Lieberman‑Aiden E, Van Berkum NL, Williams L, Imakaev M, Ragoczy T, Telling A, et al. Comprehensive mapping of long range interactions reveals folding principles of the human genome. Science. 2009;326:289.\n",
      "76. Birney E, Stamatoyannopoulos JA, Dutta A, Guigó R, Gingeras TR, Margulies EH, et al. Identification and analysis of functional elements in 1% of the human genome by the ENCODE pilot project. Nature. 2007;447:799–816.\n",
      "77. Tajti F, Kuppe C, Antoranz A, Ibrahim MM, Kim H, Ceccarelli F, et al. A func‑\n",
      " tional landscape of CKD entities from public transcriptomic data. Kidney Int Rep. 2020. https:// doi. org/ 10. 1016/j. ekir. 2019. 11. 005.\n",
      "78. Henarejos‑ Castillo I, Devesa‑Peiro A, de Miguel‑ Gomez L, Sebastian‑Leon P , Romeu M, Aleman A, et al. Predicted COVID ‑19 molecular effects on \n",
      " endometrium reveal key dysregulated genes and functions. Mol Hum Reprod. 2022.gaac035.\n",
      "79. Barrett T, Wilhite SE, Ledoux P , Evangelista C, Kim IF, Tomashevsky M, et al.\n",
      "endometrium reveal key dysregulated genes and functions.\n"
     ]
    }
   ],
   "source": [
    "# create the chroma client\n",
    "import uuid\n",
    "import chromadb\n",
    "from chromadb.config import Settings\n",
    "\n",
    "# create the open-source embedding function\n",
    "embedding_function = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")\n",
    "\n",
    "client = chromadb.HttpClient(host='localhost', port=7000, settings=Settings(allow_reset=True))\n",
    "# client.list_collections()\n",
    "# client.reset()  # resets the database\n",
    "collection = client.get_or_create_collection(\"tfm_APL\")\n",
    "# collection = client.get_collection('tfm')\n",
    "\n",
    "for i, doc in enumerate(my_json): #storage:\n",
    "    print(doc['text'])\n",
    "    collection.add(\n",
    "        ids=[str(uuid.uuid1())], metadatas=storage[i].metadata, documents=str(doc['text'])\n",
    "    )\n",
    "\n",
    "# tell LangChain to use our client and collection name\n",
    "db = Chroma(\n",
    "    client=client,\n",
    "    collection_name=\"tfm\",\n",
    "    embedding_function=embedding_function,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bec23937",
   "metadata": {},
   "outputs": [],
   "source": [
    "# from ollama import Client\n",
    "# client = Client(host='http://localhost:11434')\n",
    "# stream = client.chat(model='gemma3:12b', messages=[\n",
    "# {\"role\": \"system\", \"content\": get_system_message_rag(fullcontent)},            \n",
    "# {\"role\": \"user\", \"content\": get_ques_response_prompt(query)}\n",
    "# ],stream=True)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "RAG",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
