{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2fcec673",
   "metadata": {},
   "source": [
    "# Chroma Database Generation"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "932c4587",
   "metadata": {},
   "source": [
    "# Library"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2df9b888",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/antonioparragaleo/miniconda3/envs/RAG/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "import langchain_community\n",
    "import langchain_text_splitters\n",
    "from langchain_community.document_loaders import PyPDFLoader, pdf\n",
    "from langchain_community.embeddings.sentence_transformer import (\n",
    "    SentenceTransformerEmbeddings,\n",
    ")\n",
    "from langchain_community.vectorstores import Chroma\n",
    "from langchain_text_splitters import CharacterTextSplitter, RecursiveCharacterTextSplitter\n",
    "import os\n",
    "import pprint\n",
    "import re\n",
    "from langchain_core.documents import Document\n",
    "from langchain_core.prompts.prompt import PromptTemplate\n",
    "from langchain_core.output_parsers import JsonOutputParser\n",
    "from langchain_ollama import ChatOllama\n",
    "from pydantic import BaseModel, Field\n",
    "import json\n",
    "import uuid\n",
    "import chromadb\n",
    "from chromadb.config import Settings\n",
    "import unicodedata\n",
    "from langchain_google_genai import GoogleGenerativeAI\n",
    "import uuid\n",
    "# from chromadb.utils.embedding_functions import SentenceTransformerEmbeddingFunction\n",
    "import pickle as pkl\n",
    "import requests\n",
    "import subprocess\n",
    "import pandas as pd\n",
    "from bs4 import BeautifulSoup\n",
    "import tqdm\n",
    "import time\n",
    "from sentence_transformers import SentenceTransformer\n",
    "from langchain_huggingface import HuggingFaceEmbeddings\n",
    "import datetime\n",
    "import time\n",
    "import random\n",
    "from tqdm import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "be4ea84b",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('api_google.txt') as f:\n",
    "    \n",
    "    api_key = json.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ed9cd500",
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.environ.get(\"GOOGLE_API_KEY\"):\n",
    "  os.environ[\"GOOGLE_API_KEY\"] = api_key['key']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "22d295dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "from langchain.chat_models import init_chat_model\n",
    "llm = init_chat_model(\"gemini-2.0-flash\", model_provider=\"google_genai\",temperature=0,max_output_tokens=1024) # gemma-3-27b-it"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc8a4892",
   "metadata": {},
   "source": [
    "## Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5c7558ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"info_articles_main.pkl\",\"rb\") as f:\n",
    "    info_articles_main = pkl.load(f)\n",
    "with open(\"info_articles_ref_final.pkl\",\"rb\") as f:\n",
    "    info_articles_ref = pkl.load(f)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "53b641d8",
   "metadata": {},
   "source": [
    "## Database"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "04ef787c",
   "metadata": {},
   "source": [
    "### Split the text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e689c386",
   "metadata": {},
   "outputs": [],
   "source": [
    "splitter = RecursiveCharacterTextSplitter(\n",
    "    chunk_size=1000,\n",
    "    chunk_overlap=100,\n",
    "    separators=[\"\\n\\n\", \"\\n\", \".\", \"!\", \"?\", \" \"]  # smart splitting\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "14fa4e3c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "265"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_articles_final = info_articles_main + info_articles_ref\n",
    "len(info_articles_final)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d2f19ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "info_splitted = []\n",
    "\n",
    "for j in info_articles_final:\n",
    "\n",
    "    for key, value in j.items():\n",
    "    \n",
    "        if key in ['Abstract', 'Introduction', 'Methods', 'Results', 'Discussion', 'Conclusion',] and value:\n",
    "\n",
    "            if len(value) > 1200:\n",
    "                chunks = splitter.split_text(value)\n",
    "\n",
    "                for i, c in enumerate(chunks):\n",
    "\n",
    "                    info_splitted.append(\n",
    "                        {\n",
    "                            \"chunk_index\":i,\n",
    "                            \"content\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")+\", DOI:\"+j.get(\"DOI\")+\":\\n \"+c,\n",
    "                            \"parent\":key,\n",
    "                            \"split\":True,\n",
    "                            \"DOI\":j.get(\"DOI\"),\n",
    "                            \"Reference\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")\n",
    "                        }\n",
    "                    )\n",
    "            else:\n",
    "\n",
    "                info_splitted.append(\n",
    "                        {\n",
    "                            \"chunk_index\":0,\n",
    "                            \"content\":j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")+\", DOI:\"+j.get(\"DOI\")+\":\\n \"+value,\n",
    "                            \"parent\":key,\n",
    "                            \"split\":False,\n",
    "                            \"DOI\":j.get(\"DOI\"),\n",
    "                            \"Reference\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")\n",
    "                        }\n",
    "                    )"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0809f73",
   "metadata": {},
   "source": [
    "## Embedding"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6065057d",
   "metadata": {},
   "source": [
    "We chose this embedding according to leaderboard of HuggingFace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4bae4ce3",
   "metadata": {},
   "outputs": [],
   "source": [
    "embedding_function2 = HuggingFaceEmbeddings(model_name=\"avsolatorio/GIST-small-Embedding-v0\", model_kwargs={'device': 'cuda'}) # model_kwargs={'device': 'cuda'}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2786e097",
   "metadata": {},
   "source": [
    "## Chroma"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "961b72f8",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'info_splitted' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mNameError\u001b[39m                                 Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[9]\u001b[39m\u001b[32m, line 2\u001b[39m\n\u001b[32m      1\u001b[39m \u001b[38;5;66;03m# 2. Prepare documents, metadata, and IDs\u001b[39;00m\n\u001b[32m----> \u001b[39m\u001b[32m2\u001b[39m texts = [chunk[\u001b[33m\"\u001b[39m\u001b[33mcontent\u001b[39m\u001b[33m\"\u001b[39m] \u001b[38;5;28;01mfor\u001b[39;00m chunk \u001b[38;5;129;01min\u001b[39;00m \u001b[43minfo_splitted\u001b[49m]\n\u001b[32m      3\u001b[39m metadatas = [{\u001b[33m\"\u001b[39m\u001b[33mparent\u001b[39m\u001b[33m\"\u001b[39m: chunk[\u001b[33m\"\u001b[39m\u001b[33mparent\u001b[39m\u001b[33m\"\u001b[39m], \u001b[33m\"\u001b[39m\u001b[33mchunk_index\u001b[39m\u001b[33m\"\u001b[39m: chunk[\u001b[33m\"\u001b[39m\u001b[33mchunk_index\u001b[39m\u001b[33m\"\u001b[39m],\u001b[33m\"\u001b[39m\u001b[33mDOI\u001b[39m\u001b[33m\"\u001b[39m: chunk[\u001b[33m\"\u001b[39m\u001b[33mDOI\u001b[39m\u001b[33m\"\u001b[39m], \u001b[33m\"\u001b[39m\u001b[33mReference\u001b[39m\u001b[33m\"\u001b[39m: chunk[\u001b[33m\"\u001b[39m\u001b[33mReference\u001b[39m\u001b[33m\"\u001b[39m]} \u001b[38;5;28;01mfor\u001b[39;00m chunk \u001b[38;5;129;01min\u001b[39;00m info_splitted]\n\u001b[32m      4\u001b[39m ids = [\u001b[38;5;28mstr\u001b[39m(uuid.uuid1()) \u001b[38;5;28;01mfor\u001b[39;00m _ \u001b[38;5;129;01min\u001b[39;00m metadatas]\n",
      "\u001b[31mNameError\u001b[39m: name 'info_splitted' is not defined"
     ]
    }
   ],
   "source": [
    "# 2. Prepare documents, metadata, and IDs\n",
    "texts = [chunk[\"content\"] for chunk in info_splitted]\n",
    "metadatas = [{\"parent\": chunk[\"parent\"], \"chunk_index\": chunk[\"chunk_index\"],\"DOI\": chunk[\"DOI\"], \"Reference\": chunk[\"Reference\"]} for chunk in info_splitted]\n",
    "ids = [str(uuid.uuid1()) for _ in metadatas]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "1b79161a",
   "metadata": {},
   "outputs": [],
   "source": [
    "db = Chroma.from_texts(\n",
    "    texts=texts,\n",
    "    embedding=embedding_function2,\n",
    "    metadatas=metadatas,\n",
    "    ids=ids,\n",
    "    collection_name=\"ReproRAG\",\n",
    "    persist_directory=\"./chromaRepro\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "469cb121",
   "metadata": {},
   "source": [
    "We check that the search works properly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "1b92eabd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Document(metadata={'chunk_index': 10, 'Reference': 'P. Diaz-Gimeno et al.,2022', 'DOI': 'https://doi.org/10.1093/humrep/deab262', 'parent': 'Results'}, page_content='P. Diaz-Gimeno et al.,2022, DOI:https://doi.org/10.1093/humrep/deab262\\nPredictive performance comparison of signatures and consistency across endometrial datasets'),\n",
       " Document(metadata={'DOI': 'https://doi.org/10.1016/j.fertnstert.2024.03.015', 'parent': 'Abstract', 'Reference': 'Patricia Diaz-Gimeno et al.,2024', 'chunk_index': 0}, page_content='Patricia Diaz-Gimeno et al.,2024, DOI:https://doi.org/10.1016/j.fertnstert.2024.03.015\\nObjective: To propose a new gene expression signature that identifies endometrial disruptions independent of endometrial luteal phase timing and predicts if patients are at risk of endometrial failure. Design: Multicentric, prospective study. Setting: Reproductive medicine research department in a public hospital affiliated with private fertility clinics and a reproductive genetics laboratory. Patients: Caucasian women (n1⁄4 281; 39.4/C6 4.8 years old with a body mass index of 22.9/C6 3.5 kg/m 2) undergoing hormone replacement therapy between July 2018 and July 2021. Endometrial samples from 217 patients met RNA quality criteria for signature discovery and analysis. Intervention(s): Endometrial biopsies collected in the mid-secretory phase. Main Outcome Measure(s):Endometrial luteal phase timing-corrected expression of 404 genes and reproductive outcomes of thefirst single embryo transfer (SET) after biopsy collection to identify prognostic biomarkers of endometrial failure')]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db.similarity_search(\"Is there a signature to predict endometrial disruption?\", 2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "95c90458",
   "metadata": {},
   "source": [
    "## Evaluation"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aceba79f",
   "metadata": {},
   "source": [
    "### Agents for questions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4c9c2bd5",
   "metadata": {},
   "outputs": [],
   "source": [
    "def call_llm(llm, prompt):\n",
    "    response = llm.invoke(prompt)\n",
    "    return response.content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "9c74c378",
   "metadata": {},
   "outputs": [],
   "source": [
    "QA_generation_prompt = \"\"\"\n",
    "You are given a piece of scientific text (context).\n",
    "Your task is to generate ONE question and ONE answer from it.\n",
    "\n",
    "Guidelines for the question:\n",
    "- It must be factual and answerable using the context only.\n",
    "- Phrase it naturally, as if a researcher typed it into a search engine.\n",
    "- Do NOT mention \"context\", \"passage\", or \"according to the text\".\n",
    "- The question should be specific and concise.\n",
    "\n",
    "Guidelines for the answer:\n",
    "- The answer must be a short, factual statement directly supported by the context.\n",
    "- Do not add explanations, speculation, or references to the text.\n",
    "\n",
    "Formatting rules (strict):\n",
    "Output:::\n",
    "Question: <your question here>\n",
    "Answer: <your answer here>\n",
    "\n",
    "Now here is the context:\n",
    "\n",
    "{context}\n",
    "\n",
    "Output:::\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "787c51ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "info_splitted_evaluation = [d for d in info_splitted if d['parent'] in ['Abstract','Introduction','Results','Conclusion','Discussion','Methods']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "69dd44e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'chunk_index': 1, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n . The search identified experiments involving human endometrial transcriptomic case versus control raw data related to uterine pathologies and implantation alterations. The keywords employed in the search included endometriosis, endometrial adenocarcinoma (ADC), recurrent implantation failure (RIF), and recurrent pregnancy loss (RPL), among others (Supplemental Table 1A, available online, for a full list of search terms). No restrictions were placed on publication date or language. Uterine leiomyoma, adenomyosis, and uterine leiomyosarcoma data were not included due to a lack of suitable studies meeting our criteria. For each sample cohort belonging to the same individual study, 39 variables were evaluated (see Supplemental Table 1B), including clinical characteristics of the participants (e.g., age and body mass index), experimental design (e.g., endometrial biopsy preprocessing procedure and transcriptomic platform), and additional study information (e.g', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 0, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n MATERIALS AND METHODS This is a retrospective study integrating case and control data from multiple cohorts with endometrium gene expression in women with uterine disorders. The raw gene expression data and patient meta-data were downloaded from the National Center for Biotechnology Information (NCBI) functional genomics data repository Gene Expression Omnibus (GEO), where data are made available for the scientific community. Following GEO policies, in this repository patient data are anonymized and encrypted, and no additional institutional review board is required for downloading . Study Search and Selection Systematic searches and reviews of individual studies were conducted between October 2016 and June 2017 at the NCBI functional genomics data repository GEO  according to preferred reporting items for systematic reviews and metaanalysis (PRISMA) guidelines', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 1, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n . The search identified experiments involving human endometrial transcriptomic case versus control raw data related to uterine pathologies and implantation alterations. The keywords employed in the search included endometriosis, endometrial adenocarcinoma (ADC), recurrent implantation failure (RIF), and recurrent pregnancy loss (RPL), among others (Supplemental Table 1A, available online, for a full list of search terms). No restrictions were placed on publication date or language. Uterine leiomyoma, adenomyosis, and uterine leiomyosarcoma data were not included due to a lack of suitable studies meeting our criteria. For each sample cohort belonging to the same individual study, 39 variables were evaluated (see Supplemental Table 1B), including clinical characteristics of the participants (e.g., age and body mass index), experimental design (e.g., endometrial biopsy preprocessing procedure and transcriptomic platform), and additional study information (e.g', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 2, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n .g., GEO identifier and associated publication). Thefinal inclusion criteria were that at least one uterine disorder was evaluated in the sample cohort study design, RNA was extracted directly from human endometrial biopsies (samples collected from eutopic endometrium in endometriosis patients), the sample size was greater than three for both case and control groups belonging to the same study, microarray or RNA sequencing data were obtained using Affymetrix, Illumina, or Agilent gene expression platforms, and raw gene expression data were made freely available to download from GEO. Individual Transcriptomic Analysis: Making Functional Results Comparable The same transcriptomic analysis procedure was applied independently to each of the selected individual data sets for making the functional results comparable to one another before being integrated through functional meta-analysis', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 3, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n . The individual transcriptomic analysis consisted of three consecutive steps run in R software  and applied separately to compare cases versus controls within samples from the same study to avoid variability among sample cohorts related to different uterine disorders. The same preprocessing of raw gene expression data, differential expression analysis, and functional enrichment analysis was performed in a sample cohort comparing cases versus controls ( Supplemental Fig. 1A , available online). The preprocessing step was performed using the affy v.1.52.0  R package for studies using Affymetrix platforms to measure gene expression and thelimma v.3.34.5 1262 VOL. 113 NO. 6 / JUNE 2020 ORIGINAL ARTICLE: GENETICS   R package for the remaining platforms. Normalization between samples was done using quantile normalization (limma R package) . Annotation from probe set to gene symbol was implemented with the biomaRt R package v.2.30.0', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 4, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n . Annotation from probe set to gene symbol was implemented with the biomaRt R package v.2.30.0 . Next, exploratory analyses were performed to detect outliers and batch effects. The effect of the cycle phase of endometrial biopsy collection on the data was removed using linear models (limma v.3.34.5) as previously described elsewhere  because the samples were collected throughout the menstrual cycle . This correction ensures that the functional alterations obtained are due to the uterine disorder instead of the different proportions of cell types (stromal, epithelial) and/or the distinct menstrual cycle stages in which the endometrial biopsy samples were obtained . Differential expression analyses were applied using the limma R package  and functional enrichment analyses were executed using the gene set enrichment analysis (GSEA) logistic model implemented in themdgsa R package . TheP values for each function were calculated and corrected for false discovery rate (FDR)', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 5, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n . TheP values for each function were calculated and corrected for false discovery rate (FDR) . Two functional databases were consulted: the Kyoto Encyclopedia of Genes and Genomes (KEGG)  (release 81.0/2017-1-1) and Gene Ontology (GO)  (Release 1.2/2017-03-31). Functional Integration A functional meta-analysis using mdgsa  and metafor , R packages  was applied for each gene function evaluated, integrating each function among the different patient cohorts belonging to the same disorder (see Supplemental Fig. 1A). This methodology robustly highlights the common alterations across a set of studies, thus discarding those alterations that are exclusive of each individual study’s characteristics . In addition, it increases the statistical power through the integration of a higher number of samples and overcomes the limitations of individual studies’ heterogeneity regarding their distinct clinical population characteristics and experimental procedures', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 6, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n . All integrations were performed under the DerSimonian and Laird  random effects model to account for individual study heterogeneity in the R environment . This model weighs the functional results of each study according to their inherent variability measured by the standard deviation (SD) of the logarithm of odds ratio (LOR) obtained in the GSEA analysis for each evaluated function. Consequently, studies with less variability are weighted more in the computation of thefinal meta-analysis results than those with high variability. Thus, meta-analysis highlights the most robust endometrial functions associated with each uterine disorder. A total of 6,390 functions (226 KEGG pathways, 4,405 GO biological processes (BP) terms, 1,093 GO molecular functions (MF) terms, and 666 GO cellular components (CC) terms) were evaluated through meta-analysis for each uterine disorder', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 7, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n .P values were corrected for multiple testing by FDR , and functions were considered statistically signi ficant in the meta-analysis when FDR<0.05. The P values, summary LOR, and 95% confidence interval (CI) were calculated for all functions evaluated in the metaanalysis. To ensure unbiased results, funnel plots were analyzed for each function evaluated in the meta-analysis. The standard error (SE) of the individual LOR for each integrated study was calculated as an accuracy measure of the study performance. The LOR represents the effect size of the function associated with each uterine disorder compared with controls. If study heterogeneity and publication bias did not affect the functional meta-analysis, the individual LOR of all the integrated studies will be inside the 95% CI area of the funnel plot .', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\n",
      " MATERIALS AND METHODS This is a retrospective study integrating case and control data from multiple cohorts with endometrium gene expression in women with uterine disorders. The raw gene expression data and patient meta-data were downloaded from the National Center for Biotechnology Information (NCBI) functional genomics data repository Gene Expression Omnibus (GEO), where data are made available for the scientific community. Following GEO policies, in this repository patient data are anonymized and encrypted, and no additional institutional review board is required for downloading . Study Search and Selection Systematic searches and reviews of individual studies were conducted between October 2016 and June 2017 at the NCBI functional genomics data repository GEO  according to preferred reporting items for systematic reviews and metaanalysis (PRISMA) guidelines. The search identified experiments involving human endometrial transcriptomic case versus control raw data related to uterine pathologies and implantation alterations. The keywords employed in the search included endometriosis, endometrial adenocarcinoma (ADC), recurrent implantation failure (RIF), and recurrent pregnancy loss (RPL), among others (Supplemental Table 1A, available online, for a full list of search terms). No restrictions were placed on publication date or language. Uterine leiomyoma, adenomyosis, and uterine leiomyosarcoma data were not included due to a lack of suitable studies meeting our criteria. For each sample cohort belonging to the same individual study, 39 variables were evaluated (see Supplemental Table 1B), including clinical characteristics of the participants (e.g., age and body mass index), experimental design (e.g., endometrial biopsy preprocessing procedure and transcriptomic platform), and additional study information (e.g.g., GEO identifier and associated publication). Thefinal inclusion criteria were that at least one uterine disorder was evaluated in the sample cohort study design, RNA was extracted directly from human endometrial biopsies (samples collected from eutopic endometrium in endometriosis patients), the sample size was greater than three for both case and control groups belonging to the same study, microarray or RNA sequencing data were obtained using Affymetrix, Illumina, or Agilent gene expression platforms, and raw gene expression data were made freely available to download from GEO. Individual Transcriptomic Analysis: Making Functional Results Comparable The same transcriptomic analysis procedure was applied independently to each of the selected individual data sets for making the functional results comparable to one another before being integrated through functional meta-analysis. The individual transcriptomic analysis consisted of three consecutive steps run in R software  and applied separately to compare cases versus controls within samples from the same study to avoid variability among sample cohorts related to different uterine disorders. The same preprocessing of raw gene expression data, differential expression analysis, and functional enrichment analysis was performed in a sample cohort comparing cases versus controls ( Supplemental Fig. 1A , available online). The preprocessing step was performed using the affy v.1.52.0  R package for studies using Affymetrix platforms to measure gene expression and thelimma v.3.34.5 1262 VOL. 113 NO. 6 / JUNE 2020 ORIGINAL ARTICLE: GENETICS   R package for the remaining platforms. Normalization between samples was done using quantile normalization (limma R package) . Annotation from probe set to gene symbol was implemented with the biomaRt R package v.2.30.0. Annotation from probe set to gene symbol was implemented with the biomaRt R package v.2.30.0 . Next, exploratory analyses were performed to detect outliers and batch effects. The effect of the cycle phase of endometrial biopsy collection on the data was removed using linear models (limma v.3.34.5) as previously described elsewhere  because the samples were collected throughout the menstrual cycle . This correction ensures that the functional alterations obtained are due to the uterine disorder instead of the different proportions of cell types (stromal, epithelial) and/or the distinct menstrual cycle stages in which the endometrial biopsy samples were obtained . Differential expression analyses were applied using the limma R package  and functional enrichment analyses were executed using the gene set enrichment analysis (GSEA) logistic model implemented in themdgsa R package . TheP values for each function were calculated and corrected for false discovery rate (FDR). TheP values for each function were calculated and corrected for false discovery rate (FDR) . Two functional databases were consulted: the Kyoto Encyclopedia of Genes and Genomes (KEGG)  (release 81.0/2017-1-1) and Gene Ontology (GO)  (Release 1.2/2017-03-31). Functional Integration A functional meta-analysis using mdgsa  and metafor , R packages  was applied for each gene function evaluated, integrating each function among the different patient cohorts belonging to the same disorder (see Supplemental Fig. 1A). This methodology robustly highlights the common alterations across a set of studies, thus discarding those alterations that are exclusive of each individual study’s characteristics . In addition, it increases the statistical power through the integration of a higher number of samples and overcomes the limitations of individual studies’ heterogeneity regarding their distinct clinical population characteristics and experimental procedures. All integrations were performed under the DerSimonian and Laird  random effects model to account for individual study heterogeneity in the R environment . This model weighs the functional results of each study according to their inherent variability measured by the standard deviation (SD) of the logarithm of odds ratio (LOR) obtained in the GSEA analysis for each evaluated function. Consequently, studies with less variability are weighted more in the computation of thefinal meta-analysis results than those with high variability. Thus, meta-analysis highlights the most robust endometrial functions associated with each uterine disorder. A total of 6,390 functions (226 KEGG pathways, 4,405 GO biological processes (BP) terms, 1,093 GO molecular functions (MF) terms, and 666 GO cellular components (CC) terms) were evaluated through meta-analysis for each uterine disorder.P values were corrected for multiple testing by FDR , and functions were considered statistically signi ficant in the meta-analysis when FDR<0.05. The P values, summary LOR, and 95% confidence interval (CI) were calculated for all functions evaluated in the metaanalysis. To ensure unbiased results, funnel plots were analyzed for each function evaluated in the meta-analysis. The standard error (SE) of the individual LOR for each integrated study was calculated as an accuracy measure of the study performance. The LOR represents the effect size of the function associated with each uterine disorder compared with controls. If study heterogeneity and publication bias did not affect the functional meta-analysis, the individual LOR of all the integrated studies will be inside the 95% CI area of the funnel plot .\n"
     ]
    }
   ],
   "source": [
    "mynumb=10\n",
    "\n",
    "print(info_splitted_evaluation[mynumb])\n",
    "\n",
    "for d in info_splitted_evaluation:\n",
    "    \n",
    "    if d['Reference'] == info_splitted_evaluation[mynumb]['Reference'] and d['parent'] ==  info_splitted_evaluation[mynumb]['parent']:\n",
    "\n",
    "        print(d)\n",
    "\n",
    "        if d['chunk_index'] == 0:\n",
    "            info = d['content']\n",
    "        else:\n",
    "            info+=d['content'].split(\":\\n \")[-1]\n",
    "\n",
    "print(info)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "4e93f034",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_context(piece_of_paper, all_papers):\n",
    "\n",
    "    for d in all_papers:\n",
    "    \n",
    "        if d['Reference'] == piece_of_paper['Reference'] and d['parent'] ==  piece_of_paper['parent']:\n",
    "\n",
    "            if d['chunk_index'] == 0:\n",
    "                info = d['content']\n",
    "            else:\n",
    "                info+=d['content'].split(\":\\n \")[-1]\n",
    "\n",
    "    return(info)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "0ed9c7ef",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 500/500 [05:50<00:00,  1.43it/s]\n"
     ]
    }
   ],
   "source": [
    "N = 500\n",
    "examples = []\n",
    "for sample in tqdm(random.sample(info_splitted_evaluation,N), total=N):\n",
    "    context = get_context(piece_of_paper=sample, all_papers=info_splitted_evaluation)\n",
    "    response = call_llm(llm=llm,prompt=QA_generation_prompt.format(context=context))\n",
    "    \n",
    "    try:\n",
    "        question = response.split(\"Question:\")[-1].split(\"Answer: \")[0].strip()\n",
    "        answer = response.split(\"Answer: \")[-1].strip()\n",
    "        examples.append({\n",
    "            \"context\" : context,\n",
    "            \"question\" : question,\n",
    "            \"answer\" : answer\n",
    "        })\n",
    "    except Exception as e:\n",
    "        continue\n",
    "\n",
    "with open(\"examples_evaluation.pkl\",\"wb\") as f:\n",
    "    pkl.dump(examples, f)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a2f1d753",
   "metadata": {},
   "source": [
    "## Evaluation of questions generated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "efe8cfae",
   "metadata": {},
   "outputs": [],
   "source": [
    "question_groundedness_critique_prompt = \"\"\"\n",
    "You will be given a context and a question.\n",
    "Your task is to provide a 'total rating' scoring how well one can answer the given question unambiguously with the given context.\n",
    "Give your answer on a scale of 1 to 5, where 1 means that the question is not answerable at all given the context, and 5 means that the question is clearly and unambiguously answerable with the context.\n",
    "\n",
    "Provide your answer as follows:\n",
    "\n",
    "Answer:::\n",
    "Evaluation: (your rationale for the rating, as a text)\n",
    "Total rating: (your rating, as a number between 1 and 5)\n",
    "\n",
    "You MUST provide values for 'Evaluation:' and 'Total rating:' in your answer.\n",
    "\n",
    "Now here are the question and context.\n",
    "\n",
    "Question: {question}\\n\n",
    "Context: {context}\\n\n",
    "Answer::: \"\"\"\n",
    "\n",
    "question_relevance_critique_prompt = \"\"\"\n",
    "You will be given a question.\n",
    "Your task is to provide a 'total rating' representing how useful this question can be to researchers in the reproductive medicine field.\n",
    "Give your answer on a scale of 1 to 5, where 1 means that the question is not useful at all, and 5 means that the question is extremely useful.\n",
    "\n",
    "Provide your answer as follows:\n",
    "\n",
    "Answer:::\n",
    "Evaluation: (your rationale for the rating, as a text)\n",
    "Total rating: (your rating, as a number between 1 and 5)\n",
    "\n",
    "You MUST provide values for 'Evaluation:' and 'Total rating:' in your answer.\n",
    "\n",
    "Now here is the question.\n",
    "\n",
    "Question: {question}\\n\n",
    "Answer::: \"\"\"\n",
    "\n",
    "question_standalone_critique_prompt = \"\"\"\n",
    "You will be given a question.\n",
    "Your task is to provide a 'total rating' representing how context-independent this question is.\n",
    "Give your answer on a scale of 1 to 5, where 1 means that the question depends on additional information to be understood, and 5 means that the question makes sense by itself.\n",
    "For instance, if the question refers to a particular setting, like 'in the context' or 'in the document', the rating must be 1.\n",
    "The questions can contain obscure technical nouns or acronyms like Gradio, Hub, Hugging Face or Space and still be a 5: it must simply be clear to an operator with access to documentation what the question is about.\n",
    "\n",
    "For instance, \"What is the name of the checkpoint from which the ViT model is imported?\" should receive a 1, since there is an implicit mention of a context, thus the question is not independent from the context.\n",
    "\n",
    "Provide your answer as follows:\n",
    "\n",
    "Answer:::\n",
    "Evaluation: (your rationale for the rating, as a text)\n",
    "Total rating: (your rating, as a number between 1 and 5)\n",
    "\n",
    "You MUST provide values for 'Evaluation:' and 'Total rating:' in your answer.\n",
    "\n",
    "Now here is the question.\n",
    "\n",
    "Question: {question}\\n\n",
    "Answer::: \"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "5d199ed2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generating critique for each QA couple...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 500/500 [22:08<00:00,  2.66s/it]\n"
     ]
    }
   ],
   "source": [
    "print(\"Generating critique for each QA couple...\")\n",
    "for output in tqdm(examples,total=len(examples)):\n",
    "\n",
    "    evaluations = {\n",
    "        \"groundedness\": call_llm(\n",
    "            llm,\n",
    "            question_groundedness_critique_prompt.format(context=output[\"context\"], question=output[\"question\"]),\n",
    "        ),\n",
    "        \"relevance\": call_llm(\n",
    "            llm,\n",
    "            question_relevance_critique_prompt.format(question=output[\"question\"]),\n",
    "        ),\n",
    "        \"standalone\": call_llm(\n",
    "            llm,\n",
    "            question_standalone_critique_prompt.format(question=output[\"question\"]),\n",
    "        ),\n",
    "    }\n",
    "    try:\n",
    "        for criterion, evaluation in evaluations.items():\n",
    "            score, eval = (\n",
    "                int(evaluation.split(\"Total rating: \")[-1].strip()),\n",
    "                evaluation.split(\"Total rating: \")[-2].split(\"Evaluation: \")[1],\n",
    "            )\n",
    "            output.update(\n",
    "                {\n",
    "                    f\"{criterion}_score\": score,\n",
    "                    f\"{criterion}_eval\": eval,\n",
    "                }\n",
    "            )\n",
    "    except:\n",
    "        continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "8a312109",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "question",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "context",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "answer",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "groundedness_score",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "relevance_score",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "standalone_score",
         "rawType": "float64",
         "type": "float"
        }
       ],
       "ref": "bdb8d30d-0e9e-4cf3-bd7a-95e5681641d7",
       "rows": [
        [
         "0",
         "What do disease-associated alleles commonly alter, rather than grossly affecting protein folding and stability?",
         "Feixiong Cheng et al.,2021, DOI:https://doi.org/10.1038/s41588-020-00774-y:\n Interpretation of the clinical pathogenic effects of variants is crucial for the advancement of precision medicine. However, our ability to understand the functional and biological consequences of genetic variants identified by human genome sequencing projects is limited. Many computational approaches can identify only a small proportion of pathogenic variants with the high confidence required in clinical settings. Human genome sequencing studies have reported potential mutation-disease associations with the functional regions altered by somatic mutations, such as molecular drivers in cancers.\n1\n,\n2\nHowever, many important issues in the field remain unclear, including the phenotypic consequences of different mutations within the same gene and the same mutation across different cell types.\nRecent efforts using systematic analyses of 1,000–3,000 missense mutations in Mendelian disorders\n3\n,\n4\nand ~2,000\nde novo\nmissense mutations in developmental disorders\n53\n,\n4\nand ~2,000\nde novo\nmissense mutations in developmental disorders\n5\ndemonstrate that disease-associated alleles commonly alter distinct protein-protein interactions (PPIs) rather than grossly affecting the folding and stability of proteins.\n3\n,\n4\nNetwork-based approaches provide novel insights into disease-disease\n6\nand drug-disease\n7\n–\n9\nrelationships within the human interactome. Yet, the functional consequences of disease mutations on the comprehensive human interactome and their implications for therapeutic development remain understudied. Several studies have suggested that protein structure-based mutation enrichment analysis offers a potential tool for identification of possible cancer driver genes\n10\n, such as hotspot mutation regions in three-dimensional (3D) protein structures\n11\n–\n1410\n, such as hotspot mutation regions in three-dimensional (3D) protein structures\n11\n–\n14\n. Development of novel computational and experimental approaches for the study of functional consequences of mutations at single amino acid residue resolution is crucial for our understanding of the pleiotropic effects of disease risk genes and offers potential strategies for accelerating precision medicine\n15\n,\n16\n.\nIn this study, we investigated the network effects of disease-associated mutations at amino acid resolution within the 3D macromolecular interactome of structurally-resolved and computationally-predicted PPI interfaces. We provide evidence for widespread perturbations of PPIs in human diseases caused by both germline and somatic mutations identified in large-scale sequencing studies.",
         "Disease-associated alleles commonly alter distinct protein-protein interactions.",
         null,
         null,
         null
        ],
        [
         "1",
         "In protein networks, what is the difference in coexpression with neighbors between hub-bottlenecks and hub-nonbottlenecks?",
         "Haiyuan Yu et al.,2007, DOI:https://doi.org/10.1371/journal.pcbi.0030059:\n Bottlenecks Tend To Be Essential\nBecause bottlenecks are key connectors in protein networks, we hypothesized that these proteins would represent important points in networks. Therefore, we first examined the essentiality of bottlenecks in different networks in yeast (see Methods). We found that bottlenecks in both regulatory and interaction networks indeed tend to be essential proteins with high significance (see Figure 2A), in agreement with previous studies [18,19].\nBottlenecks Dictate the Essentiality of Networks with Directed EdgesAs discussed above, previous studies have shown that, in biological networks, hubs tend to be essential [7,9], and betweenness of a node is correlated with its degree [20]. We found that degree and betweenness are indeed highly correlated quantities in the networks we analyzed (Pearson correlation coefficient of 0.49,p , 10 /C0 15 for the interaction network; Pearson correlation coefficient of 0.67,p , 10/C0 15 for the regulatory network;p-values measure the significance of the Pearson correlation coefficient scores according to t distributions; i.e., many bottlenecks also tend to be hubs). Therefore, we further investigate which one of these two quantities is a better predictor of protein essentiality in both regulatory and interaction networks. To disentangle the effects of betweenness and degree, we divided all proteins in a certain network into four categories:  nonhub-nonbottlenecks;  hub-nonbottlenecks;  nonhub-bottlenecks; and  hub-bottlenecks (see Figure 1). Even though the two quantities are highly correlated, the number of hub-nonbottlenecks and nonhub-bottlenecks is enough for reliable statistics (see Table S1). This is in agreement with the previous observation by Huang and his colleagues, who found that proteins with high betweenness but low degree (i.e., nonhub-bottlenecks) are abundant in the yeast protein interaction network [18]. As we discussed above, numerous previous studies have shown that the degree of a protein determines its essentiality in scale-free networks (i.e., proteins with higher degrees are more likely to be essential) [7,9]. Both interaction and regulatory networks have been shown to be scale-free networks [7,9,22]. Here, we observed that bottlenecks (both nonhub-bottlenecks and hub-bottlenecks) have a strong tendency to be products of essential genes, whereas hubnonbottlenecks are surprisin gly not essential. Thus, we determined that it is the betweenness that is a stronger determinant of the essentiality of a protein in the regulatory network, not the degree (Figure 2B). In contrast to regulatory networks, the interaction network is undirected with no obvious information flow. Furthermore, nonneighboring pairs in the interaction network have no noticeable relationships, as they are neither coregulated nor coexpressed [23]. Therefore, it is reasonable to assume that in Figure 1. Schematic Showing a Bottleneck and the Four Categories of Nodes in a Network Four nodes with different colors represent examples of the four categories defined by degree and betweenness. Please note that every node in the network belongs to one of the four categories. However, in this schematic, we only point out the categories of the four example nodes. doi:10.1371/journal.pcbi.0030059.g001 PLoS Computational Biology | www.ploscompbiol. doi:10.1371/journal.pcbi.0030059.g001 PLoS Computational Biology | www.ploscompbiol.org April 2007 | Volume 3 | Issue 4 | e590714 Author Summary A network is a graph consisting of a number of nodes with edges connecting them. Recently, network models have been widely applied to biological systems. Here, we are mainly interested in two types of biological networks: the interaction network, where nodes are proteins and edges connect interacting partners; and the regulatory network, where nodes are proteins and edges connect transcription factors and their targets. Betweenness is one of the most important topological properties of a network. It measures the number of shortest paths going through a certain node. Therefore, nodes with the highest betweenness control most of the information flow in the network, representing the critical points of the network. We thus call these nodes the ‘‘bottlenecks’’ of the network. Here, we focus on bottlenecks in protein networks. Here, we focus on bottlenecks in protein networks. We find that, in the regulatory network, where there is a clear concept of information flow, protein bottlenecks indeed have a much higher tendency to be essential genes. In this type of network, betweenness is a good predictor of essentiality. Biological researchers can therefore use the betweenness as one more feature to choose potential targets for detailed analysis. Importance of Protein Bottlenecks  interaction networks, hubs are more important than bottlenecks. Our calculations confirmed this hypothesis (see Figure 2B): although nonhub-bottlenecks are significantly more likely to be essential than nonhub-nonbottlenecks (p , 10 /C0 5; see Methods for the calculation ofp-values in Figures 2 and 3), the difference is not nearly as substantial as that between hubnonbottlenecks and nonhub-nonbottlenecks ( p , 10 /C0 267). Similar results were also found in different interaction networks (see Figure S4). Similar results were also found in different interaction networks (see Figure S4). The difference between nonhubbottlenecks (low essentiality) and hub-nonbottlenecks (much higher essentiality) confirms that degree is a much better predictor of essentiality in the interaction network.Signal transduction pathways are a special case of proteinprotein interactions [24]. There are well-defined information flows in these pathways. Nonhub-bottlenecks participating in signaling transduction pathways clearly are more likely to be products of essential genes (see Figure 3).\nBottlenecks within Permanent Undirected Interactions Are Also ImportantBesides directionality, another important but often overlooked aspect of interaction networks is that there are two major classes of interactions: permanent and transient [25,26]. Within permanent interactions, bottlenecks are connectors holding different, functionally important complexes together. However, within transient interactions, bottlenecks merely interact with different complexes at different times. In this sense, ‘‘transient’’ bottlenecks are not really bottlenecks. They are classified as‘‘bottlenecks’’ by our algorithm simply because of the fact that current interaction networks are a collection of individual networks under various conditions. As a result, the function of these transient bottlenecks is likely to be not as important as that of permanent ones. Therefore, we hypothesized that bottlenecks would be more likely to be essential in permanent rather than in transient interactions. We tested our hypothesis in the yeast interaction network. We tested our hypothesis in the yeast interaction network. Defining permanent interactions as those participating in protein complexes, we analyzed all complexes from the Munich Information Center for Protein Sequences (MIPS) complex catalog [27]. (Previous studies have shown that most of the MIPS complexes are stable, permanent complexes. However, there are 52 proteins in this catalog without direct evidence of stable interactions with others [26]. We removed these proteins from our analysis.) Because the catalog is far from complete, we also considered all interactions forming a clique (a complete subgraph) of size 5 or bigger as permanent, because protein complexes are often considered as cliques in interaction networks [28,29]. Any interaction not participating in a clique of size 3 or bigger was considered transient. Any interaction not participating in a clique of size 3 or bigger was considered transient. Our calculations confirmed our hypothesis: nonhub-bottlenecks within permanent interactions tend to be essential, while those within transient ones do not (see Figure 3).For completeness, we also analyzed hubs and bottlenecks in other kinds of protein networks. Specifically, we analyzed the topology of three very different kinds of protein networks, namely the metabolic network (where links connect enzymes that share a metabolite) [23], the genetic network (where links connect proteins that have genetic interactions) [30], and the phosphorylation network (where links connect a kinase with its substrates) [31]. The edges in the phosphorylation and metabolic networks are directed, whereas those in the genetic network are undirected (see Table S2 and http://www. gersteinlab.org/proj/bottleneck).\nBottlenecks Constitute the Dynamic Components of NetworksThe correspondence of protein interaction bottlenecks to connectors both in complexes and in pathways leads us to investigate their dynamic properties. To this end, we examined their coexpression with their neighbors. It has been previously observed that interacting protein pairs are more likely to coexpress than noninteracting protein pairs [32]. Likewise, protein complex members have been shown to be highly coexpressed [25]. Given this information, we hypothesized that bottlenecks would tend to have a belowaverage expression correlation with their neighbors, since they tend to represent proteins that connect different complexes or pathways. Indeed, in all the datasets examined, we find that bottlenecks have a much lower average expression correlation with their neighbors than other nodes. Surprisingly, the difference is much more pronounced when focusing on hubs only (i.e. Surprisingly, the difference is much more pronounced when focusing on hubs only (i.e., the difference is more significant between hub-nonbottlenecks and hub-bottlenecks than between nonhub-nonbottlenecks and nonhubbottlenecks). The majority of hub-nonbottlenecks are relatively well coexpressed with their neighbors, whereas most hub-bottlenecks are not very well coexpressed (see Figure 4). What appears especially striking is that bottlenecks always have low coexpression with their neighbors, whereas hubs can have a relatively high average coexpression with their neighbors, but only if they are nonbottlenecks. We find that while nonbottlenecks simply follow the same distribution as the rest of the datasets, the nonhub-bottlenecks tend to have a lower expression correlation. Central complex members have a low betweenness and are hub-nonbottlenecks. Because of the high connectivity inside these complexes, paths can go through them and all their neighbors. On the other hand, hub-bottlenecks tend to correspond to highly central proteins that connect several complexes or are peripheral members of central complexes. The fact that they have a high betweenness suggests that these proteins are not, however, simply members of large protein complexes (which is true for nonbottleneck-hubs), but are those members that connect the complex to the rest of the graph; in a sense, real connectivity bottlenecks. While hubnonbottlenecks mainly consist of structural proteins, hubbottlenecks are more likely to be part of signal transduction pathways (see Table S3). Furthermore, hub-bottlenecks are (by construction) the most efficient in disrupting the network upon hub removal (see Figure S3). This relates nicely to the date/ party-hub concept by Han et al. [21]: hub-bottlenecks tend to be date-hubs, whereas hub-nonbottlenecks tend to be party-hubs. [21]: hub-bottlenecks tend to be date-hubs, whereas hub-nonbottlenecks tend to be party-hubs. Nonhub-bottlenecks generally coexpress less well with their neighbors than nonhub-nonbottlenecks, in line with the observation that betweenness is a good predictor of average correlation with neighbors. Nonhub-bottlenecks also rarely are complex members and are in large part made up of regulatory proteins and signal transduction machinery.",
         "Hub-bottlenecks have low coexpression with their neighbors, whereas hub-nonbottlenecks have relatively high average coexpression.",
         "5.0",
         "4.0",
         "5.0"
        ],
        [
         "2",
         "What are the main children of the new top level term GO:0140110 transcription regulator activity?",
         "The Gene Ontology Consortium et al.,2018, DOI:https://doi.org/10.1093/nar/gky1055:\n GO resource content\nThe GO knowledgebase consists of the ontology and the annotations made using the ontology. As of the 5 September 2018 release (doi:10.5281/zenodo.1410625), there were ∼45 000 terms in GO: 29 698 biological processes, 11 147 molecular functions and 4201 cellular components, linked by almost 134 000 relationships. The number of annotations (as well as the percentage change since 2016 (\n1\n)) are shown in Table\n1\n. It is important to understand that the changes reflect two distinct processes: addition of annotations based on new evidence, and obsoleting of annotations that have been superseded by newer studies. We expect the number of obsoleted annotations to increase, due to our increasing annotation quality assurance efforts, described in more detail below.\nNew framework and repository, for gene function ‘models’New framework and repository, for gene function ‘models’\nWe have developed a more expressive computational framework for representing gene functions, which subsumes our current GO annotation framework, while maintaining compatibility. We refer to the framework as GO-\nC\nausal\nA\nctivity\nM\nodeling (GO-CAM, formerly referred to as ‘LEGO’ (\n1\n)) and to GO-CAMs as\nmodels\nto distinguish them from\nstandard annotations\n. A paper detailing GO-CAM is in preparation, but we summarize some properties here. In GO-CAM, each model is represented as a set of triples (\nsubject\n-\nrelation\n-object, with brackets {} as a set container), e.g. {\nABCA1\nenables\ncholesterol transporter activity\n;\ncholesterol transporter activity\noccurs in\nplasma membrane\n, and\ncholesterol transporter activity\npart_ofoccurs in\nplasma membrane\n, and\ncholesterol transporter activity\npart_of\ncholesterol homeostasis}. Each triple is supported by one or more pieces of evidence, consisting of a class from ECO and a citable source, usually a scientific publication. GO-CAM specifies the semantics of GO annotations, and how standard GO annotations can be combined into a larger model. Each GO-CAM model is represented using the Web Ontology Language (OWL), which is converted computationally to standard GO annotations (GAF format), ensuring backward compatibility. Users can browse, view, and download the available models in different formats at:\nhttp://geneontology.org/go-camhttp://geneontology.org/go-cam\n. The number of models is currently small, and most models contain only one gene product (standard annotations extended with additional contextual information, such as cell type). The GO is currently rapidly increasing the curation of GO-CAM models, and the model repository is growing. In particular, models are now available that each represent an entire regulatory or metabolic pathway.\nChanges to data access: new production pipelineChanges to data access: new production pipeline\nStarting in March 2018, the releases of the GO resource have been generated by a new data production pipeline using a refreshed software stack. This system allows for easily extensible error checking and improved reporting of quality assurance checks that ensure the quality and integrity of the released ontology and annotations. For users, one of the most important aspects of this pipeline is that the GO resource now produces monthly releases (named by release date) that are available at the GO site and can be referenced and obtained as stable Document Object Identifiers (DOIs) via Zenodo. This feature is critical for ensuring that GO-based analyses can be replicated in a consistent and referenceable manner through the inclusion of these DOIs and/or version of both the ontology and annotation files used (\n66\n). Our data production pipeline is currently hosted at the Lawrence Berkeley National Laboratory. We provide GO annotations in multiple formats: as standard GAF (Gene Association Format) and GPAD (Gene Product Association Data) annotation files, in Turtle (OWL serialization format) [\nhttp://current.geneontology.org/products/ttl\n], and a Blazegraph [\nhttp://current.geneontology.org/products/blazegraph\n] journal, replacing the legacy MySQL output.\nInteractions with the GO user community\nGO is an open project, and we encourage community contributions to the knowledgebase and software.\nAll users\n: GO can be contacted using the GO Helpdesk (\nhttp://help.geneontology.orgAll users\n: GO can be contacted using the GO Helpdesk (\nhttp://help.geneontology.org\n) for any questions or feedback about the annotations, the ontology, software, or other GO resources. If users notice an annotation that may not be correct, they should first review the original publication or data source. If the annotation still seems inaccurate, users are encouraged to report this to the GO helpdesk, and GOC members will review the annotation and remove or modify it if justified.\nAuthors\n: Authors can now see if their paper has been used for GO annotation directly in PubMed. Under the ‘LinkOut - more resources’ section of the PubMed abstract page, papers with annotations will have a link labeled ‘Gene Ontology annotations from this paper - Gene Ontology’ (see, e.g.\nhttps://www.ncbi.nlm.nih.gov/pubmed/3357510https://www.ncbi.nlm.nih.gov/pubmed/3357510\nwhich links to a web page on the GO site that shows the annotations based on evidence in that paper). If no GO LinkOut is present, that may indicate that the publication has not been used for GO annotation. Authors can contact the helpdesk at the GO website to suggest new annotations or changes to existing annotations.\nResources or consortia\n: The GOC collaborates with established data resources and other groups and consortia representing a particular area of biology. Recent examples include cilium biology (\n7\n,\n8\n), autophagy (\n9\n) and cardiac phenotypes (\n10\n). We encourage members of other interest groups to contact us to improve the ontology and annotations in their areas of expertise.\nTracking all contributions:\nMost aspects of the GO project management are now based in GitHub (\nhttps://github.com/geneontology\n). In addition to tracking ontology change requests (\nhttps://github.com/geneontology/go-ontology). In addition to tracking ontology change requests (\nhttps://github.com/geneontology/go-ontology\n), we now use GitHub to collect feedback on annotations (\nhttps://github.com/geneontology/go-annotation\n). For users familiar with GitHub, we encourage them to submit any requests directly to GitHub, where they can follow all further discussion and actions. Otherwise, issues and queries can be submitted to our helpdesk (\nhttp://help.geneontology.org\n).\nIncreased focus on annotation quality controlThe GO resource is now 20 years old. The longevity of the resource adds a challenge to maintain and update the many existing annotations, as many of the findings published during that time have become much more precise, or were reinterpreted or superseded. We have made it a high priority to identify and correct inaccurate and out-of-date legacy annotations to make sure that GO continues to consistently reflect current knowledge. We have taken a number of different approaches to tackle this challenge. First, to ensure consistency and quality, GO biocurators meet regularly for training, establishment of annotation guidelines, and coordinated review of specific areas of biology. More recently, we have made significant efforts to integrate annotation review with ontology improvements, taking advantage of suggested changes to the ontology to clarify term definitions, intended usage, and coordinate annotation practices with curators. In addition, quality assurance is performed centrally, both computationally, to ensure annotations are valid, and manually, to ensure they accurately represent the experimental findings.We have discovered that one of the most powerful approaches to quality control and consistency is the phylogenetic approach. Originally developed as a means of propagating annotations from experimentally studied genes to evolutionarily related genes in other species, the phylogenetic perspective provides a unified view of all experimental annotations within a evolutionarily-related protein family, allowing curators to more easily find outlier annotations (see e.g. (\n1111\n)). In parallel, the development of GO-CAM models has been useful in identifying inconsistent annotation practices, and has provided opportunities to develop consortium-wide annotation guidelines. Another observation that emerged is that older annotations from isolated phenotypic observations, taken outside of other contextual data, often do not provide evidence of direct involvement of a gene in a biological process. If inconsistencies are noticed, they are reported to the contributing group for verification and correction as appropriate.In a pilot quality assurance effort, we have requested the review (by GO Consortium biocurators) of ∼2500 manual annotations (<0.01% of the total corpus) that were judged questionable by one of the strategies above. Approximately 70–80% of the annotations flagged for review were modified to a more appropriate term or removed. We will continue to work on improving the quality of the annotations and reviewing legacy data when appropriate. As a result, we expect that the increase in annotations and new ontology terms may not be as rapid as in the past, at least for the main species annotated by the consortium members, as a greater proportion of our efforts will be dedicated to reviewing and revising older annotations.\nOntology revision and integrationOntology revision and integration\nSince our last update article, we have developed an entirely new process for ontology editing and maintenance that has dramatically increased efficiency and enabled extensive real-time quality checks. Ontology editing is now performed in an OWL-based environment using the ontology editing tool Protégé (\n12\n). The ontology is versioned and tracked using a GitHub repository (\nhttps://github.com/geneontology/go-ontology\n). One major advantage of the new ontology management process is that the work can be parallelized among multiple editors, thus increasing efficiency. In addition, real-time quality checks prevent errors that would otherwise require revisiting the same editing task more than once to correct them.GO continues to integrate and align with external ontology resources in two main ways: import of sub-ontologies used to define GO terms, and inclusion of external cross-references. GO utilizes the structure of external ontologies to aid in reasoning and in the automatic inference of relations between GO terms (\n13\n). GO imports subsets of these external ontologies that include information about anatomical structures, cell types, chemicals and taxonomic groupings: Uberon (\n14\n), Protein Ontology (\n15\n), Plant Ontology (\n16\n), ChEBI (\n17\n), Relations Ontology (\n18\n), NCBI Taxonomy (\n19\n), Sequence Ontology (\n20\n), Ontology of Biological Attributes (\nhttp://www.obofoundry.org/ontology/oba.html\n), Fungal Anatomy Ontology (\nhttp://www.obofoundry.org/ontology/fao.html\n), Phenotypic Quality Ontology (\nhttp://obofoundry.org/ontology/pato.html\n), and Common Anatomy Reference Ontology (\nhttp://www.obofoundry.org/ontology/caro.html), and Common Anatomy Reference Ontology (\nhttp://www.obofoundry.org/ontology/caro.html\n). GO also maintains cross-references between terms and multiple widely-used external resources, including Reactome (\n21\n), The Annotated Reactions Database (Rhea) (\n22\n), Enzyme Commission (EC;\nhttp://www.sbcs.qmul.ac.uk/iubmb/enzyme/\n), IntAct, Complex Portal (\n23\n) and MetaCyc (\n24\n).\nRefactoring the molecular function branch of GO23\n) and MetaCyc (\n24\n).\nRefactoring the molecular function branch of GO\nPreviously, there was a trend in GO molecular function ontology development to focus on adding terms that describe molecular binding activities of specific gene products. The advantage of such terms is that annotations can often be made unequivocally based on results from a single experiment. However, this approach has led to a complex ontology structure and a proliferation of annotations that individually represent only a partial functional description of a gene product. Annotations to binding terms can obscure annotations to more informative function terms, making annotations more difficult to interpret. For example, one can annotate\nCDK1\n(UniProtKB:\nP06493CDK1\n(UniProtKB:\nP06493\n) separately with ‘GO:0030332 cyclin binding’, ‘GO:0005524 ATP binding’, ‘GO:0005515 protein binding’, and ‘GO:0004674 protein serine/threonine kinase activity’. However, these are all aspects of a more precise molecular function that is more informative than the sum of these parts: ‘GO:0004693 cyclin-dependent protein serine/threonine kinase activity’. In the GO molecular function refactoring, we recognized that, while specific binding events are an essential mechanism by which gene products function, an individual binding activity is almost never sufficient in itself to describe molecular function in a larger biological context (\n2525\n). One of the primary goals of our refactoring was to ensure that the ontology contains the terms necessary to describe these higher-level functions, and that they have a path to the root of the ontology that is not simply under the generic ‘GO:0005488 binding’ term. Accordingly, we reinstated some previously obsoleted terms and added new terms, as well as additional relations. We also addressed the structure of the ontology so that the upper-level terms would be more biologically meaningful and have more uniform specificity. We removed 8 terms from the top level and added four new terms (Figure\n2\n). Most of the terms that were formerly direct children of ‘GO:0003674 molecular function’ were moved under more biologically meaningful terms: for example, ‘GO:0042056 chemoattractant activity’ and ‘GO:0045499 chemorepellent activity’ were moved under ‘GO:0048018 receptor ligand activity’, while ‘GO:0036370\ndd\n-alanyl carrier activity’ and ‘GO:0016530 metallochaperone activity’ were moved under the new term ‘GO:0140104 molecular carrier activity’ (representing an activity of ‘directly binding to a specific ion or molecule and delivering it either to an acceptor molecule or to a specific location’). We have also created a new term ‘GO:0104005 hijacked molecular function’ as a parent of terms such as ‘GO:0001618 virus receptor activity’, which is, from the standpoint of the protein being annotated, not a normal function, but nevertheless relevant for some of our users.\nFigure 2.\nOpen in a new tab\nThe\nmolecular function\nbranch, before and after refactoring. The term marked with an ‘x’ in the left-hand panel has been obsoleted. Terms moved (assigned to a new parent) are indicated by arrows. New terms (right panel) are marked with ‘NEW’.\n1\nThe class label ‘electron carrier activity’ was changed to ‘electron transfer activity’.1\nThe class label ‘electron carrier activity’ was changed to ‘electron transfer activity’.\nFinally, we have made significant changes to the structure representing the molecular functions of transcription factors (Figure\n3\n). This refactoring was carried out in collaboration with experts in transcription factors and gene regulation from the Gene Regulation Consortium (GRECO;\nhttp://thegreco.orghttp://thegreco.org\n). In keeping with our design principle of having terms that describe higher-level functions, we have created a new parent term to group all functions that directly regulate transcription, ‘GO:0140110 transcription regulator activity’. The formerly top-level term ‘GO:0000988 transcription factor activity, protein binding’ has been obsoleted because this activity was partly covered by other terms in the ontology and its usage was inconsistent. Accordingly, its children have either been obsoleted or subsumed under different terms (merged or moved). The new top level term ‘GO:0140110 transcription regulator activity’ has three main children - ‘GO:0003700 DNA-binding transcription factor activity’ (formerly labeled ‘transcription factor activity, sequence-specific DNA binding), ‘GO:0140223 general transcription initiation factor activity’ and ‘GO:0003712 transcription coregulator activity’.\nFigure 3.\nOpen in a new tabFigure 3.\nOpen in a new tab\nCurrent structure of the ‘GO:0140110 transcription regulator activity’ branch of the ontology.\nThe transcription factor areas of GO had previously been refactored between 2010 and 2012 (\n26\n,\n27\n) with the aim of more finely capturing all combinations of different types of protein and DNA binding activities (e.g. binding to different types of regulatory regions such as promoters and enhancers) and transcription regulation processes (positive and negative regulation). However, this structure, while very precise, has proved very difficult to use by biocurators, resulting in inconsistent annotations. Additionally, end-users have had difficulty with common queries, such as comprehensively identifying the set of all transcription factors in a given species. We expect that even more improvements to the ontology structure, as well as more consistent annotations to transcription regulator terms, will be available in 2019.Defining the boundaries of biological processes: MAP kinase signaling and extracellular matrix as examplesWe have used our integrated annotation review and ontology development methodology to refine two areas of the ontology, the MAP kinase signaling pathway and the representation of the extracellular matrix. Refinements to the MAPK cascade included defining the molecular functions that are parts of the process: ‘GO:0004707 MAP kinase activity’, ‘GO:0004708 MAP kinase kinase activity, ‘GO:0004709 MAP kinase kinase kinase activity’ and ‘GO:0008349 MAP kinase kinase kinase kinase activity’. All other upstream and downstream molecular functions/biological processes will be modeled in GO-CAM with causal relationships between them and the MAPK cascade. We also enumerated the types of cascades based on current literature and on discussions among expert model-organism curators, trying to keep the distinctions useful across multiple taxa. There are four direct subtypes of ‘GO:0000165 MAPK cascade’: ‘GO:0070371 ERK1 and ERK2 cascade’, ‘GO:0070375 ERK5 cascade’, ‘GO:0071507 pheromone response MAPK cascade’ and ‘GO:0051403 stress-activated MAPK cascade’. Some other MAPK processes such as ‘GO:1903616 MAPK cascade involved in axon regeneration’ will eventually be obsoleted, as these combine two or more other GO terms and can be composed in GO-CAM. For the refinement of the extracellular matrix area of the ontology, we worked with external experts to add terms that were useful grouping terms such as ‘GO:0062023 collagen-containing extracellular matrix’. We also obsoleted or merged terms that were poorly annotated and thought to represent an outdated view.GO subsets (slims)\nA GO subset (or slim) is a set of GO terms selected to provide an overview of the functions, locations or roles of a set of genes. The subset can be developed for high coverage of specific species, or to represent only certain areas of the ontology, and in most cases, contain only high-level GO terms to provide a broad biological overview. Another use of subsets is to blacklist certain terms for annotation: GO has two such subsets, one to flag terms that should not be used for manual annotation, and one for terms that should not be used at all. GO maintains two additional subsets, the Generic GO slim and the Alliance of Genome Resources (\nhttps://www.alliancegenome.org/\n) slim. GO also hosts subsets useful to groups using GO; we currently have 11 such subsets (Table\n2\n;\nhttp://www.geneontology.org/page/go-subset-guide2\n;\nhttp://www.geneontology.org/page/go-subset-guide\n). Each subset provides a global overview of gene functions. Each subset now has a designated contact person to resolve any issue resulting from ontology changes (see Ontology revision and integration).\nOther developments\nThe GO ribbon: a configurable tool for visualizing GO annotations\nMany genes have large numbers of annotations, making it difficult to get a quick overview of a gene function, or the functions of gene sets. We have developed the GO ribbon specifically to help users visualize and explore the functions of a gene. The GO ribbon visualization metaphor borrows from a viewer originally developed by the Mouse Genome Database team (\n28\n), but in contrast, the GO ribbon was developed as a lightweight, reusable widget that can be embedded in any website, and retrieves data directly from the GO resource via API.To generate a GO ribbon, all the functions (GO terms) associated with a gene of interest are mapped onto a specified GO subset using the ontology structure. The end result is a simple graphical representation of a gene's functions (Figure\n4\n). The ribbon is interactive, allowing users to drill down to more specific functions by selecting a high-level category such as ‘GO:0030154 cell differentiation’, ‘GO:0050877 nervous system process’, or ‘GO:0003700 DNA-binding transcription factor activity’, and to filter the functions based on the evidence codes provided in the GO annotations. This overview of gene functions is particularly useful when comparing the functions of different genes in the same species, or the functions of orthologous genes across different species.\nFigure 4.\nOpen in a new tab\nGO ribbon representation. Darker boxes indicate terms with the most annotations; white boxes represent terms that are not annotated for this protein (\nMus musculusMus musculus\nSox7, MGI:98369). Screenshot obtained from\nhttps://www.alliancegenome.org/gene/MGI:98369\n.\nThe GO ribbon is a React component available on GitHub (\nhttps://github.com/geneontology/ribbon\n) and as a NPM package\nhttps://www.npmjs.com/package/@geneontology/ribbon\n). The GO ribbon widget is currently used by the Alliance of Genome Resources.\nGO annotations from high-throughput experiments\nData from high-throughput experiments are generally collected in a hypothesis-free manner, and consequently do not generally provide as strong evidence of gene function as small-scale molecular biology experiments that currently support most of the experimental GO annotations. In addition, high-throughput experiments can be subject to relatively high false positive rates. Users may therefore want to filter out these experimental annotations in some applications of the GO. To make this possible, starting in 2018, in collaboration with the Evidence and Conclusions Ontology (\n29\n) (\n2), the GO has added several new evidence codes to describe high-throughput experiments: ‘ECO:0006056 high throughput evidence used in manual assertion’ (HTP), and the subclasses: ‘ECO:0007005 high throughput direct assay evidence used in manual assertion’ (HDA), ‘ECO:0007001 high throughput mutant phenotype evidence used in manual assertion’ (HMP), ‘ECO:0007003 high throughput genetic interaction evidence used in manual assertion’ (HGI) and ‘ECO:0007007 high throughput expression pattern evidence used in manual assertion’ (HEP). To accompany the new evidence codes, we have provided annotation guidelines to help identify and curate high-throughput datasets that meet the GO Consortium annotation criteria. Consortium members have reviewed papers with more than 40 annotations using a single evidence code, and updated the evidence codes, or removed the annotations if appropriate. There are currently over 31 000 annotations that have HTP evidence codes from 140 research articles, representing <5% of experimental GO annotations. The identification of annotations derived from high-throughput experiments allows users to choose to exclude these from their analyses, if they are concerned that these annotations may lead to an increased bias in data analysis. This is likely to be particularly important, as is often the case, when GO is used to interpret types of data similar to those on which the annotations are based.",
         "The main children are GO:0003700 DNA-binding transcription factor activity, GO:0140223 general transcription initiation factor activity and GO:0003712 transcription coregulator activity.",
         null,
         null,
         null
        ],
        [
         "3",
         "How did ApEn vary with sex and age in the study?",
         "Ferdinand Roelfsema et al.,2017, DOI:https://doi.org/10.1530/EC-17-0160:\n This study quantifies simple and complex measures of cortisol secretion with respect to sex, age and BMI utilizing 10-min sampling for 24 h in 143 adults ages 19–78 years and validated analytical measures of putative feedback control (ApEn), pulsatile and nonpulsatile secretion (deconvolution) and diurnal rhythmicity (Cosinor analysis). Total 24-h cortisol secretion estimated by deconvolution of intensively sampled profiles was independent of sex, age and BMI. In subanalyses, cortisol dynamics exhibited a higher pulse frequency, shorter half-life and lower basal secretion in women than men. Nycthemeral variations differed by sex and age, with higher mean cortisol in the daytime in men than women, and in late evening and early night in older than young subjects. This result seems to be at variance with a study in which cortisol levels were stimulated with low-dose and high-dose ACTH and by glucagon administration in 55 healthy individuals (\n3838\n). In this limited study BMI, age and sex were not significant variables. However, acute stimulation tests are strictly non-physiological, and therefore, any comparison is hampered by these considerations. Moreover, for a fair comparison between such studies dose–response curves are required, but such expensive and labor-intensive studies are far beyond the clinical requirements.\nApEn was higher in women than men, higher in the daytime and independent of aging. The nycthemeral maximum (acrophase) was advanced to an earlier clock time by 40 min/decade and increased by about 10 nmol/L/decade.\nHPA axis activity ostensibly varies with age, sex, BMI, socioeconomic status, chronic illness, stress, psychiatric disorders and sleep disruption (\n1\n,\n13\n,\n181\n,\n13\n,\n18\n). Comparisons among studies, however, are hampered by different experimental designs, including number and spacing of blood samples, salivary samples or urinary excretion of cortisol and/or cortisol metabolites, size of groups, sex, BMI and age distributions. Blood sampling intensity in studies in which age was related to mean serum cortisol levels varied from 4 to 72 samples/24 h. The study of 145 samples/24 h could not demonstrate age effects on mean (median) cortisol concentrations or distinct cortisol secretion parameters. Three studies reported an increase in cortisol area under the curve (\n5\n,\n6\n,\n13\n), a fourth study a decrease in mean 24-h cortisol (\n10\n), and a fifth and sixth no influence of age (\n10\n,\n11\n,\n2710\n), and a fifth and sixth no influence of age (\n10\n,\n11\n,\n27\n). The present approach differs by using 10-min sampling, 24-h data, a larger cohort of subjects, and all three of deconvolution, Cosinor and ApEn analyses. Nonetheless, the age range of 19–78 years, like most other studies, limits conclusions in the very old.\nCortisol’s regulation was examined further using consecutive 3-h time windows over 24 h, since higher cortisol in the late evening-early night quiescent period has been associated with sleep disorders and the metabolic syndrome (\n7\n,\n28\n). By cohort comparisons, two studies found higher cortisol concentrations in older than young subjects in the quiescent period (\n6\n,\n13\n), whereas one study did not (\n12\n). The present analysis across continuous ages delineates a late-day rise in mean cortisol concentrations of 10 nmol/L/decade. This effect could be due to diminished feedback of cortisol on pituitary corticotropes and/or brain CRH/AVP-secreting neurons (\n29\n,\n3029\n,\n30\n) and may have health implications. Thus, the Leiden Longevity Study of 303 subjects disclosed lower salivary cortisol in the late evening in offspring of healthy nonagenarian siblings compared to their partners (\n8\n). In another study, hydrocortisone administration during the quiescent phase in metyrapone-clamped volunteers reduced insulin sensitivity and glucose tolerance (\n31\n). However, in this study, serum cortisol levels reached about 800 nmol/L, exceeding physiological levels. Conversely, continuous sc infusion of cortisol in Addison’s patients was sufficient to prevent the nocturnal glucose decrease (\n32\n). Direct experimental data are needed on the impact of glucocorticoid levels in this time window over days or weeks on other health indices, such as blood pressure, abdominal adiposity, diabetes control and systemic inflammation.Few studies have assessed whether 24-h cortisol secretion is sex dependent or estradiol dependent. One small isotope dilution study in 7 men and 7 women under age 40 years estimated nearly 50% lower cortisol production in women than men (\n14\n). Another study reported lower mean 24-h cortisol concentrations in women than men, supporting the present findings in volunteers under age 50 years (\n13\n). Using 24-h sampling, we could show that cortisol concentrations were higher in men only during daytime. This insight would agree with higher single-morning total and free cortisol levels reported in men earlier (\n33\n).\nThe present analysis revealed that cortisol secretory-burst mass is smaller and half-life shorter in premenopausal women than men under age 50 years. Because of a higher pulse frequency, no",
         "ApEn was higher in women than men, higher in the daytime, and independent of aging.",
         "5.0",
         "4.0",
         "1.0"
        ],
        [
         "4",
         "What average implantation rate is achieved with existing morpho-kinetic markers in in vitro fertilization?",
         "Rehannah Borup et al.,2016, DOI:https://doi.org/10.1371/journal.pone.0153562:\n In order to improve the efficacy of in vitro fertilisation treatment, there is ongoing research to define good non-invasive embryo markers to increase the implantation rate from an average of around 25% achieved with the existing morpho-kinetic markers [1]. The developmental competence of the mammalian oocyte is achieved during folliculogenesis, where cross-talk between oocyte and somatic cells ensures oocyte growth and maturation and the subsequent potential to sustain fertilisation and produce viable embryos [2]. Thus, several attempts have been made to identify oocyte competence markers in the various follicle compartments. Specific functions of the somatic cells from the pre-ovulatory follicles just prior to ovulation (e.g. progesterone and oestrogen production from the cumulus-oocyte-complex (COC) [3] and expression levels of specific genes in the mural granulosa cells (MGC) and cumulus cells (CC) have been shown to reflect the developmental competence of the corresponding oocyte. Several authors have used a single gene approach by analyzing few genes by RT-PCR [4-7], while others have used global gene expression profiling analysis to find differences in somatic cell gene expression between cells connected to competent and non-competent oocytes [8-16]. These studies represent different end-points for competence that vary from oocyte maturational stage to the cleavage stage embryo morphology, positive hCG, ongoing pregnancy and finally birth of a healthy child. The difference in competence measures may partly explain the predominant lack of overlap in suggested competence marker genes, as may the difference in isolation technique, and the in vitro exposure time before isolation of cells. Taking the complexity of the preovulatory follicle into account, the aim of identifying single marker genes seems to be less robust than a marker based on the scenario of multi interacting genes, which may be identified by the high trough put techniques such as the expression array. Therefore the present study focus on differences in the MGC and CC transcriptomes between oocytes from competent and non-competent follicles and the aim is to develop classifier profiles in the two cellular compartments in order to be able to select the embryos with the best potential to result in life offspring. To approach the scenario for selecting a competent embryo for transfer in a cohort of embryos, a case control design was chosen matching women’s age and embryo morphology. Thus, in the present study, gene expression profiling of corresponding CC and MGC of individual follicles obtained from women undergoing IVF with elective single embryo Day 3 transfer were associated to no pregnancy after IVF or life birth of a healthy baby.",
         "An average of around 25% implantation rate is achieved with existing morpho-kinetic markers.",
         null,
         null,
         null
        ],
        [
         "5",
         "What range of Pearson correlation values between WNC scores and P values were found in different networks?",
         "Francisco J Azuaje et al.,2014, DOI:https://doi.org/None:\n Methods Weighted node connectivity One way to account for co-expression intensity consists of assigning weights to each edge in the network. In the case of gene co-expression networks, an edge weight, between genes i and j, can represent the value of the expression correlation coefficient between genes i and j.Is t u d i e da weighted version of degree, which has previously been applied to the analysis of traffic control and social networks [12,13]. In bioinformatics it has been applied to compare the global structure of networks [1,5,14]. Previous investigations, however, have mainly addressed gene selection based on the analysis of the observed connectivity values only, without statistical significance inference at the level of individual genes. The method reported here goes beyond this by implementing a non-parametric statistical assessment procedure. The weighted node connectivity, WNC, score can be specified as: WNC i 1⁄4 XN j wi;j ð1Þ where node i is connected to nodej, and wi,j reflects the strength of the connection of nodei with node j (input information provided by the user). The algorithm is not constrained by the correlation measurement used to generate the input network. In this paper, w i,j is computed as the absolute value of the Pearson correlation coefficient between genesi and j. Previous research, including those cited above, shows that WNC is a reliable indicator of connectivity based on co-expression information. More importantly, these investigations have shown that the score is useful to explore the potential biological significance of genes. For instance, they showed that the score is a good indicator of biological relevance when searching for possibly interesting groups of interconnected genes, i.e., network modules [1,14].e., network modules [1,14]. For each gene, I calculated theWNC i score and support each value with an indicator of its statistical significance. This is needed for reducing noise and possible bias toward many connections with weak wi,j values, as well as for ranking the selected genes. Thus, I estimatedP values for each WNCi score. This was done by implementing a Azuaje Biology Direct 2014, 9:12 Page 2 of 23 http://www.biologydirect.com/content/9/1/12  permutation-based test that compares the observedWNCi with an (empirical null) distribution of randomWNCi,random values. The latter was estimated by randomly swapping the edges in the network, while preserving the distributions of observed edges and weights in the network. Given two randomly selected edges: e(x,y)a n de (v,z) with corresponding weight valueswx,y and wv,z,w h e r e :x,y,v,zrepresent network nodes and x ≠ y ≠ v ≠ z,t h ee d g ew e i g h t swx,y and wv,z are assigned to e(v,z)a n de (x,y) respectively. Thus, the resulting null distribution is both random and feasible. This edge randomization procedure has been applied to estimate null distributions in different network research applications [15-17]. AP value is obtained by calculating the proportion of WNCi ≥ WNCi,random,w h e r et h eWNCi,random values were obtained from 100E + 3 permuted samples. This permutation procedure is suitable to generate a null model because the resulting networks are random and comparable with the observed (original) network. In particular, this procedure guarantees the preservation of fundamental properties of the original network: size, degree distribution and weight distribution [18]. Furthermore, eachP value is adjusted with a Bonferroni correction to account for multiple testing. This correction provides sufficiently conservative estimates of “significance” among the many statistically detectable scores. Those WNC i with (adjusted)P < 0. Those WNC i with (adjusted)P < 0.05 are here reported as relevant and considered for further analysis. Ranking and selection of genes should be based on their P values. This is because, in comparison withWNC scores, P values provide a more reliable estimator of gene connectivity. Indeed, it is possible to obtain genes with relatively high WNC scores that are detected as statistically spurious after P value calculation. This is reflected in the observation that WNC scores and their correspondingP values are not perfectly linearly correlated (Additional files 1, 2 and 3). In this study, Pearson correlations betweenWNC scores and P values range from−0.53 to −0.85 in different networks. Moreover, P values represent more interpretable values to bioinformaticians and experimental biologists alike, e.g., values between 0 and 1 with well-defined statistical meaning.g., values between 0 and 1 with well-defined statistical meaning. Software and published work based on random permutation methods, including network-based research, typically define between 1 K and 10 K random samples for networks of similar or larger size [17,19]. In the case of my algorithm, when tested on different datasets, a number of permuted samples above 10 K gives highly consistent estimations of P values. Indeed, there are no detectable differences in the datasets investigated here when generating more than 50 K permutated samples. Thus, although there is no guarantee that my test can estimate the true exact P values, these observations at least indicate that the selected number of permutated samples is sufficiently large. Also it is important to highlight that the accompanying software allows users to define this parameter according to their needs and computing resources. Other indicators of gene centrality I also investigated 4 standard indicators of node centrality, w h i c hh a v ep r e v i o u s l yb e e na p p l i e dt od i v e r s eb i o l o g i c a lr e - search applications. These scores are applied to networks in which edges are defined as binary interactions, i.e., an edge, a ij, connecting genesi and j is represented with a value of 1 if the expression correlation between genesi and j is above a predefined value, otherwiseaij is equal to 0. a. The degree, di, of gene i in a network of N genes, represents the number of nodes connected to gene i [20]. Genes with large degrees are commonly referred to as hubs. This indicator can be defined as: di 1⁄4 XN j aij ð2Þ Where aij is 1 if gene i is linked to gene j, and 0 otherwise. b. The closeness centrality, ci, encodes the capacity of node i to interact with all the other nodes in the network, including those that are not directly connected to node i [21]. This measure is calculated as the inverse sum of the shortest distances from i to all the other nodes in the network: ci 1⁄4 1 XN j hi ; jðÞ ð3Þ Where h(i,j) represents the shortest distance between genes i and j. Equation  will only return values for nodes that are connected to other nodes. Also the closeness centrality value of a node is calculated in relation to the connected graph in which the node is located. This is a key limitation of this approach, which alsoimplies that nodes located in small sub-networks (separated from the largest connected sub-network) may report relatively high closeness centrality values. In this investigation, all unweighted networks analyzed with this score consistedo fal a r g ec o n n e c t e dg r a p h containing the vast majority of the nodes, and all of the nodes had at least 1 edge and reported closeness centrality v a l u e sb e t w e e n0a n d1 . c. c. The betweenness centrality, b i,e n c a p s u l a t e st h ep r o p -erty of node i a sab r i d g i n gn o d ei nt h en e t w o r k ,i . e . ,a measurement of the number of shortest paths connecting any two nodes,j and k, which pass through nodei [22]. Nodes with large b i values are often called“high traffic” nodes. Azuaje Biology Direct 2014, 9:12 Page 3 of 23 http://www.biologydirect.com/content/9/1/12  bi 1⁄4 X i≠j≠k nj;k iðÞ nj;k ð4Þ Where nj,k is the total number of shortest paths connecting nodes j and k, and nj,k(i) is the number of shortest paths that pass through nodei. d. The clustering coefficient , ui, indicates the edge density in the neighborhood of nodei. d. The clustering coefficient , ui, indicates the edge density in the neighborhood of nodei. This is calculated as the proportion of edges that exist between the nodes contained in its neighborhood, divided by the total number of edges that can potentially be found between these nodes [23]: u i 1⁄4 mi j; kðÞ mi ð5Þ Where mi(j,k) is the number of edges connecting all nodes j and k in the neighborhood of nodei, and mi is the total number of edges that could be seen if all the nodes in the neighborhood ofi were fully connected. Implementation of scores The WNC score was implemented in Java. Source code and an executable program, WiPer, are freely available (see Software Availability). The centrality measures specified in formula  to  were computed with the Network Analyzer plugin [24] of Cytoscape [25]. Datasets Gene expression datasets from published studies on: glioblastoma treatment (GBM), kidney versus liver tissue comparisons (KL) and heart regeneration after injury in zebrafish (ZF) were analyzed [26-28] (see Table 1). To minimize noise and facilitate interpretation of results, genes showing significant expression changes across samples were selected for network generation. In the GBM dataset, I focused on 198 genes displaying │log 2Fold-Change │ ≥ 1 between treatment and control samples. In the KL dataset, 211 genes showing high differential expression between kidney and liver samples were selected (those with adjustedP = 0). In the ZF dataset, I selected 221 genes, which showed statistically detectable differential expression between 1- and 7-day post-injury in relation to uninjured (control) samples (adjusted P < 0.05). “Adjusted P” refers to P values obtained after correcting for multiple testing.05). “Adjusted P” refers to P values obtained after correcting for multiple testing. In the GBM dataset, log 2Fold-Change was used to select genes, instead of P values, because the latter detected relatively low numbers of differentially expressed genes. In the other datasets, the chosenP values allowed the selection of gene sets consisting of hundreds of genes instead of thousands. In all datasets, the latter was the main selection requirement. Microarray data analyses were implemented with the limma package (MEV platform, v4.9.0) [29,30]. DESeq tools (version 1.10.1) were applied to the RNA-Seq dataset under the R platform (v. 2.15.3) [31]. Generation of gene co-expression networks For all pairs of genes in the datasets, the Pearson correlation coefficient,ρ, was calculated to define the strength of the gene-gene connections in the network. In the analyses of WNC scores, I considered all the obtainedρ values to generate the networks (weighted networks). As both negative and positive correlations may be biologically interesting, network edges were re presented with absolute ρ values, │ρ i,j │. For the analyses of standard centrality measures (unweighted networks), I concentrated on gene coexpression values sufficiently high to reduce the possibility of including spurious associations and to focus on the most likely relevant co-expression patterns. Here I report results in which a network edge was established, between nodesi and j,i f │ρ i,j │ ≥ 0.95-percentile, i.e., those│ρi,j│ values falling above the 95th percentile of the observed data. Although this procedure is expected to reduce the number of edges in comparison to the weighted network, the total number of genes is not substantially affected. The characteristics of the resulting networks are summarized in Table 2. Gene co-expression values were computed with the MINE application [32], and networks were visualized with Cytoscape [25]. Note that the proposedWNC score does not require the application of specific gene selection criterion or correlation measures to generate the networks. Here, to facilitate comparisons, the same gene selection criterion (inputs to network generation) and correlation measure were applied to generate the weighted and unweighted networks in each dataset. To illustrate the application of the algorithm, other co-expression cut-off criteria for network generation are feasible as well. Here, changes in these thresholds naturally will have the effect of generating different network sizes and, therefore, different numbers of putative hubs. Nevertheless, for conservative threshold selections, e.g., cut-offs above the 75th percentile, I have found that hubs detected as highly significant will consistently Table 1 The gene expression datasets investigated Dataset Biological problem Tissue origin No. genes No. genes No. samples Platform Source GBM Glioblastoma resistance to treatment Brain 198 10 Microarrays [ 26] KL Comparison of tissue types Kidney and liver 211 14 RNA-Seq [ 27] ZF Zebrafish heart regeneration after amputation Heart 221 9 Microarrays [ 28] Azuaje Biology Direct 2014, 9:12 Page 4 of 23 http://www.biologydirect.com/content/9/1/12  appear as top-ranked genes in different network generation settings. Exploration of potential biological relevance Here, a group of genes are considered as biologically informative if at least there are statistically detectable associations between the genes and biological processes relevant to the phenotype under investigation. The potential biological relevance of the genes identified as significant with the centrality scores was first estimated through an enrichment analysis of Gene Ontology (GO) terms. Using Fisher ’s exact test and multiple-test corrections, this procedure aimed to detect biological processes that are statistically associated with the genes selected. This was done for functional terms in the biological process hierarchy of GO, from levels 3 to 9, and considered as significant whenP < 0.05. Analyses were implemented with Babelomics (v. 4.3) [33]. An independent estimation of GO biological process enrichment (at all levels) was completed with DAVID (v. 6.7), with those terms reporting FDR < 0.05 considered as statistically significant [34]. This analysis was extended in a deeper investigation of the application case in zebrafish heart regeneration. Based on the integration of independent datasets, I computationally predicted biological processes that are statistically associated with the top-ranked genes ( P <0 . 0 5 ). 0 5 ) . These analyses were performed with the IMP (Integrative Multi-species Prediction) system [35], and associations with biological processes were deemed significant at P < 0 . 0 5 .T h eI M Pm o d e li sb a s e do nt h ei n t e g r a t i o n of multiple sources of biolog ical information, including protein-protein interactions and gene expression profiles [35].",
         "Pearson correlations between WNC scores and P values ranged from −0.53 to −0.85 in different networks.",
         null,
         null,
         null
        ],
        [
         "6",
         "What is the network parsimony principle in network medicine?",
         "Albert-László Barabási et al.,2011, DOI:https://doi.org/10.1038/nrg2918:\n Location of disease genes within networks\nAn unexpected property of biological networks is the emergence of hub proteins (\nBox 2\n), suggesting that hubs must play a special biological role. Indeed, evidence from model organisms indicates that hub proteins tend to be encoded by essential genes\n29\n, and that genes encoding hubs are older and evolve more slowly than genes encoding non-hub proteins\n30\n–\n32\n. The deletion of genes encoding hubs also leads to a larger number of phenotypic outcomes than the deletion of genes encoding less connected proteins\n21\n. While the strength of evidence for some of these effects is still debated\n21\n,\n3321\n. While the strength of evidence for some of these effects is still debated\n21\n,\n33\n, by virtue of the many interactions they have, one expects that the absence of a hub would affect the function of an exceptional number of other proteins. This assumption has led to the hypothesis that, in humans, hubs should typically be associated with disease genes. Indeed, the protein products of up-regulated genes in lung squamous cell carcinoma tend to have a high degree of connectivity\n34\nand 346 proteins implicated in cancer have, on average, twice as many interaction partners as non-cancer proteins\n35\n. Moving beyond cancer, one study found that disease proteins in the OMIM Morbid Map\n28\nhave more protein-protein interactions than non-disease proteins in a literature-curated protein-protein interaction datase\n36\n.36\n.\nNote, however, that the essential gene concept in simple organisms does not map uniquely into disease genes in humans. Indeed, some human genes are essential in early development, so functional changes in them often lead to first-trimester spontaneous abortions (embryonic lethality). Mutations in such ‘essential’ genes cannot propagate in the population, as individuals carrying them cannot reproduce. In contrast, individuals can tolerate for a long time the disease-causing mutations, often past their reproductive age. The question is, are both (disease and essential) genes associated with hubs? Goh\net al\n37\nfound that essential genes show a strong tendency to be associated with hubs and expressed in multiple tissues, i.e., they tend to be located at the functional center of the interactome (\nFig. 1Fig. 1\n). Yet, in contrast with our initial hypothesis, non-essential disease genes do not show a tendency to encode hubs and tend to be tissue-specific. That is, from a network perspective, these genes segregate at the functional periphery of the interactome (\nFig. 1b\n). In summary, in human cells it is the essential genes, and not the disease genes, that are encoding hubs. This difference can be understood from an evolutionary perspective: mutations that disrupt hubs have difficulty propagating in the population, as the absence of hubs create so many disruptions that the host may not survive long enough to reproduce. Thus, only mutations that impair functionally or topologically peripheral genes can persist, accounting for the family of heritable diseases, especially those that appear in adulthood.\nLocal clustering of disease genes – disease modulesLocal clustering of disease genes – disease modules\nIf a gene or molecule is involved in a specific biochemical process or disease, its direct interactors might also be suspected to play some role in the same biochemical process. In line with this “local” hypothesis (\nBox 3\n), proteins involved in the same disease show a high propensity to interact with each other\n37\n,\n38\n. For example, Goh\net al.\n37\nobserved 290 physical interactions between the products of genes associated with the same disorder, representing a 10-fold increase relative to random expectation (\nP\n< 10\n−6\n). Furthermore, Gandhi\net al.\n39\nand Xu and Li\n36P\n< 10\n−6\n). Furthermore, Gandhi\net al.\n39\nand Xu and Li\n36\nfound that genes linked to diseases with similar phenotypes have a significantly increased tendency to interact directly with each other. These observations indicate that if we identify a few disease components, the other disease-related components will likely be in their network-based vicinity. That is, we expect that each disease can be linked to a well-defined neighborhood of the interactome, often referred to as a\ndisease module\n.\nBOX 3: Hypotheses of Network Medicine.\nNetwork medicine is based on a series of widely used (and often unspoken) hypotheses and organizing principles that link network structure to biological function and disease. Next, we summarize some of the most frequently utilized hypotheses, their use being discussed in more detail in the main text.\nHubs:Hubs:\nNon-essential disease genes (representing the majority of all known disease genes) tend to avoid hubs and segregate at the functional periphery of the interactome.\nIn utero\nessential genes tend to associated with hubs.\nLocal hypothesis:\nProteins involved in the same disease have an increased tendency to interact with each other.\nCorollary of the local hypothesis:\nMutations in interacting proteins often lead to similar disease phenotypes.\nDisease module hypothesis:\nCellular components associated with a specific disease phenotype show a tendency to cluster in the same network neighborhood.\nNetwork parsimony principle\n: Causal molecular pathways often coincide with the shortest molecular paths between known disease-associated components.\nShared components hypothesis:\nDiseases that share disease-associated cellular components (genes, proteins, metabolites, miRNAs) show phenotypic similarity and comorbidity.As we try to understand the network-based position of disease genes, we need to distinguish among three distinct phenomena (\nFig. 2\n). A\ntopological module\nrepresents a locally dense neighborhood in a network, such that nodes have a higher tendency to link to nodes within the same local neighborhood than to nodes outside of it. It can be identified using various network clustering algorithms\n40\n–\n43\nthat are blind to the function of individual nodes. By contrast, a\nfunctional module\nrepresents the aggregation of nodes of similar or related function in the same network neighborhood. Finally, a\ndisease module\nrepresents a group of network components that together contribute to a cellular function whose disruption results in a particular disease phenotype.In the biological literature, there is a tacit assumption that these three concepts are interrelated: cellular components that form a topological module have closely related functions, thus corresponding to a functional module; and a disease is a result of the breakdown of a particular functional module, intimating that a functional module is also a disease module. However, several unique characteristics of disease modules are important to bear in mind. First, a disease module may not be identical to, but likely overlaps with, the topological and/or functional modules. Second, a disease module is defined in relation to a particular disease, and, accordingly, each disease has its own unique module. Finally, a gene, protein, or metabolite can be implicated in several disease modules, which means that different disease modules can overlap.The disease module hypothesis represents a network-level expansion of the disease gene hypothesis. The emergence of a disease is viewed as a combinatorial problem in which many different defects and perturbations result in a similar disease phenotype, provided they alter the activity of the disease module. Such combinatorial disease mechanisms are well documented in cancer\n4444\n, but the utility of the disease module hypothesis extends beyond polygenic diseases, and is important even in some monogenic diseases. For example, sickle cell disease, a classic Mendelian disorder, is caused by a single point mutation at position 6 of the beta-chain of hemoglobin. Yet, this simple biochemical phenotype and its corresponding monogenotype do not yield a single pathophenotype: individuals with sickle cell disease can present with painful crises, osteonecrosis, acute chest syndrome, stroke, profound anemia, or mild asymptomatic anemia. Thus, the underlying disease module will likely include all disease modifying genes (e.g., hemoglobin F) that mediate various epigenetic, transcriptional, and post-translational phenomena. An important step of network-based approaches to disease is, therefore, to identify the disease module for the pathophenotype of interest, which, in turn, can guide further experimental work and influence drug development.\nIdentifying disease modulesIdentifying disease modules\nBioinformatics approaches\nDisease modules can be identified on the basis of currently available data using bioinformatics approaches, whose main steps are described in\nFigure 3\n. Variants of this methodology have been applied to a wide range of diseases and pathophenotypes, from cardiovascular disease to various forms of cancer. For example, Taylor\net al.\n45\nstudied the dynamic modular structure of the protein interaction network in adenocarcinoma of the breast, finding that hub proteins that displayed altered modularity in the human interactome were useful indicators for predicting breast cancer outcome. Similarly, Chen\net al.\n46et al.\n46\nrelied on co-expression networks constructed from liver and adipose tissues, facilitating the identification of sub-networks associated with genetic loci linked to obesity- and diabetes-related DNA variations. The results confirmed the connection between obesity and a macrophage-enriched metabolic subnetwork, validating three previously unknown genes,\nLPL\n,\nLACTB\n, and\nPPM1L\n, as obesity genes in transgenic mice. An extensive list of diseases to which the disease module has been identified, with the pertinent references, can be found in\nTable I online\n.\nThe disease module-based approach is useful in exploring pathogen-induced phenotypes, as well. Pathogens often disrupt the cell’s normal activity by interacting with and transferring their genes or proteins to the host cell; thus, a thorough understanding of this process requires a map of the interactions between the molecular components of the virus and the host cell\n47\n,\n4847\n,\n48\n. A recent study has shown that the array of diseases associated with Epstein-Barr virus and human papillomavirus are linked to genes whose protein products lie in topological proximity to the viral target proteins\n49\n, leading to the identification of the viral disease module. In light of these advances, an area ripe for network-based approaches is the bacterial microbiome (and other metagenomes) and its relationship to human disease\n50\n.\nExperimental mapping of disease modules\nOften the rate-limiting step in mapping a disease module is the small coverage of the available cellular interaction maps in the vicinity of the known disease components, requiring additional experimental efforts to identify relevant interactions. This approach was successfully applied to several diseases, including Huntington disease\n51\n, spinocerebellar ataxia\n52\n, breast cancer\n53\n, and schizophrenia\n54\n. For example, starting from 23 known ataxia-causing genes, Lim\net al\n.\n52, and schizophrenia\n54\n. For example, starting from 23 known ataxia-causing genes, Lim\net al\n.\n52\nused yeast two-hybrid assays to map out their interactions with other human proteins; the interactions of this second group of proteins were then used to build a dense subnetwork two degrees removed from the known ataxia genes. A member of the predicted ataxia disease module, puratrophin-1, a common binding partner to many of the known ataxia genes, which were not previously recognized as having any commonality, was later shown to lead to ataxia-like phenotypes in mice upon its deletion\n55\n.\nPredicting disease genes\nTraditionally, disease-associated genes were discovered by linking genomic intervals containing hundreds of genes to a particular phenotype, or, more recently, with genome-wide association studies (GWAS)\n5656\n, identifying SNPs that have a statistically significant correlation with the disease. Both methodologies can offer a large number of disease-gene candidates, but identifying the particular gene and the mutation that is causal to the disease remains a difficult undertaking. Recently, a series of increasingly sophisticated network-based tools have been developed to predict potential disease genes, integrating in the network context knowledge about a particular disease, whether it derives from GWAS, full-genome sequencing, linkage methods, or individual studies. The existing tools can be loosely grouped into three categories (\nFig. 4\n):\nLinkage methods\nassume that the direct interaction partners of a disease protein are likely candidates to be associated with the same disease phenotype\n38\n,\n57\n,\n58\n. Indeed, Oti\net al\n.\n3838\n,\n57\n,\n58\n. Indeed, Oti\net al\n.\n38\nshowed that the set of genes that fell within one of the known disease loci and whose products interacted with a known disease protein were 10-fold enriched in true disease-causing genes; and by considering cellular localization as well, the network information lead to a 1000-fold enrichment over a random selection. Using this feature, they predicted (and confirmed using independent data) that Janus kinase 3 (JAK3) was a candidate protein for severe combined immunodeficiency syndrome, due to its interaction with lymphocyte specific protein-tyrosine (LCK), protein-tyrosine phosphatase (PTPRC), and interleukin 2 receptor (IL2RG), known disease-associated proteins.\nModularity or pathway-based methods\nassume that all cellular components that belong to the same topological/functional/disease module have a high likelihood of being involved in the same disease\n59\n,\n60\n. Thus these methods start with identifying the disease modules (\nFig 359\n,\n60\n. Thus these methods start with identifying the disease modules (\nFig 3\n), and inspecting their members as potential disease genes. For example, Wu et al.\n61\nshowed that many of the recently identified breast cancer susceptibility genes were ranked highly by an algorithm which took advantage of the positive correlation between modularity in the protein interaction network and phenotype-phenotype similarity network.\nDiffusion-based methodsDiffusion-based methods\naim to identify the pathways that are closest to the known disease genes. By releasing hypothetical random walkers from the protein products of the known disease genes, that are then allowed to diffuse along the links of the interactome (moving to any neighboring node with equal probability), one can identify the nodes and links that are closest to the known disease genes, as they will be those most often visited by the random walkers during their random walks. Proteins that interact with several disease proteins will gain a high probabilistic weight, as will those that may not directly interact with any disease proteins but are in close network proximity thereto, helping prioritize proteins and interactions based on their potential involvement in the particular disease. Variants of this methodology have been applied to detect disease genes related to a wide range of diseases, from diabetes mellitus to prostate cancer and Alzheimer disease\n62\n,\n63\n.62\n,\n63\n.\nThese methodologies (A–C) exploit to an increasing degree the topological and functional information encoded by the interactome. Method A involves only pair-wise linkage information (local hypothesis,\nBox 3\n), while the modularity-based method (B) exploits the full network neighborhood of disease genes (disease module hypothesis,\nBox 3\n). Finally, diffusion-based methods (C) use the information encoded in the full network topology and the placement of the known disease genes, thereby simultaneously exploiting both topological and functional modularity (together with the parsimony principle,\nBox 3\n). It is not surprising, therefore, that a recent comparative study found, that, on the same dataset, linkage-based methods have the least predictive power and that diffusion-based methods offer the best predictive performance\n59\n.\nHuman Diseasome59\n.\nHuman Diseasome\nThe highly interconnected nature of the interactome means that at the molecular level, it is difficult, if not counter-intuitive, to consider diseases as being invariably independent of one another. Indeed, different disease modules can overlap, so that perturbations caused by one disease can affect other disease modules. The systematic mapping of such network-based dependencies between the pathophenotypes and their disease modules has culminated in the concept of the diseasome\n3737\n, representing disease maps whose nodes are diseases and whose links represent various molecular relationships between the disease-associated cellular components. Uncovering such links between diseases not only helps us understand how different phenotypes, often addressed by different medical sub-disciplines, are linked at the molecular level, but can also help us comprehend why certain groups of diseases arise together. The co-morbidity of conditions culled from the diseasome offers insights that may yield novel approaches to disease prevention, diagnosis, and treatment. Diseasome-based approaches could also aid drug discovery, in particular when it comes to the use of approved drugs to treat molecularly linked diseases. Next, we review the construction of such disease maps and the consequences of the observed disease associations.\nShared Gene Hypothesis and the Human Diseasome Network (HDN)Shared Gene Hypothesis and the Human Diseasome Network (HDN)\nIf the same gene is linked to two different disease pathophenotypes, this linkage is often an indication that the two diseases have a common genetic origin. Motivated by this hypothesis, Goh\net al.\n37\nused the gene-disease associations collected in the OMIM database to build a network of diseases that are linked if they share one or several genes. In the obtained HDN, 867 of 1,284 diseases with an associated gene are connected to at least one other disease, and 516 of them belong to a single disease cluster (\nFig. 5\n). The clustering of nodes of similar color in\nFigure 5\n, denoting the disease class, reflects the fact that similar pathophenotypes have a higher likelihood of sharing genes than pathophenotypes that belong to different disease classes. For example, cancers form a tightly interconnected and easily detectable cluster, held together by a small group of genes associated with multiple cancers, such as\nP53\n,\nKRAS\n,P53\n,\nKRAS\n,\nERBB2\nor\nNF1\n.\nTo test if the shared gene relationship has epidemiological consequences in disease occurrence in the population\n64\n, we show the comorbidity between linked disease pairs in\nFigure 5\n. This analysis indicates that a patient is twice as likely to develop a (comorbid) disease if that disease shares a gene with the primary disease than if that disease does not share a gene with the primary disease. Yet, many disease pairs that share genes do not show significant comorbidity. This lack of comorbidity may occur, in part, because different mutations on the same gene can have different effects on the function of the gene product and on its organ-based expression, therefore, different pathological consequences\n65\nthat are context-dependent. Such ‘edgetic’ alleles affect a specific subset of links in the interactome\n6666\n, and individuals who harbor different mutations in the same gene can develop different disorders. Consistent with this view, disease pairs associated with mutations that affect the same functional domain of a protein show higher comorbidity than disease pairs whose mutations occur in different functional domains\n64\n(\nFig. 5\n).\nShared metabolic pathway hypothesis and the Metabolic Disease Network (MDN)\nAn enzymatic defect that affects the flux of one reaction may potentially affect the fluxes of all downstream reactions in the same pathway, leading to disease phenotypes that are normally associated with these downstream reactions. Thus, for metabolic diseases, links induced by shared metabolic pathways are expected to be more relevant than the links based on shared genes. In support of this hypothesis, Lee\net al\n.\n67\nconstructed a metabolic disease network in which two disorders are connected if the enzymes associated with them catalyze adjacent reactions (\nFig. 5Fig. 5\n). The visually apparent clustering of the MDN mirrors distinct metabolic pathways. For example, purine metabolism consists of 62 reactions associated with 33 diseases, including nucleoside phosphorylase deficiency and congenital dyserythropoietic anemia. These diseases form a visually distinct cluster, highlighted with blue shading in\nFigure 5\n. Comorbidity analysis confirms the functional relevance of metabolic coupling: disease pairs linked in the MDN have a 1.8-fold increased comorbidity compared to disease pairs that are not linked metabolically\n67\n. Comorbidity is even more pronounced if the fluxes of the reactions catalyzed by the respective disease genes are themselves coupled, i.e., changes in one flux induce significant changes in the other flux, even if the corresponding reactions are not adjacent.\nShared microRNA hypothesis\nPrompted by the increasing evidence of the role of miRNAs in human disease, Lu\net al.\n68Prompted by the increasing evidence of the role of miRNAs in human disease, Lu\net al.\n68\nconnected diseases pairs whose associated genes are targeted by at least one common miRNA molecule. The obtained network displays a disease class-based segregation: for example, cancers share similar associations at the miRNA level, leading to a distinct cancer cluster, which, for example, differs from the cluster associated with cardiovascular diseases, in the miRNA-based disease network.\nPhenotypic Disease Network (PDNs)\nOne can also link disease pairs based on the directly observed comorbidity between them, obtaining a phenotypic disease network. For example, Rzhetsky\net al.\n69\ninferred the comorbidity links between 161 disorders from the disease history of 1.5 million patients at the Columbia University Medical Center, and Hidalgo\net al.\n70et al.\n70\nbuilt a network involving 657 diseases from the disease history of over 30 million Medicare patients. In these maps, two diseases are connected if their comorbidity exceeds a predefined threshold. The PDN is blind to the mechanism underlying the observed comorbidity, which may be rooted in molecular-level dependencies (as seen for HDN, MDN or mi-RNA based disease networks), or in environmental or treatment-related perturbations of the network. Yet, PDN captures disease progression, as patients tend to develop diseases in the network vicinity of diseases they have already had\n70\n. Furthermore, patients who are diagnosed with diseases with more links in the PDN show a higher mortality than those diagnosed with less connected diseases\n70\n. Another use of phenotypic information to build a disease network was suggested by Van Driel\net al.\n71et al.\n71\n, who employed text mining to assign to over 5,000 human phenotypes in the OMIM database a string of phenotypic features from the medical subject heading vocabulary. The overlap of their phenotypic descriptions was used to link various diseases, finding that phenotypic similarity positively correlates with the molecular signatures of two linked diseases, from relatedness at the level of protein sequence to protein motifs and direct protein–protein interactions between the disease-associated proteins.These studies indicate that the molecular-level links between the known disease components have direct epidemiological consequences, leading to observable comorbidity patterns. While most efforts focused on the role of single molecular or phenotypic measure to capture disease-disease relationships (such as shared genes or metabolites), a comprehensive understanding requires us to inspect multiple sources of evidence, from shared genes to protein-protein interaction based relationships, shared environmental factors, common treatments, affected tissues and organs, and phenotypic manifestations. In line with such integrated approaches, Suthram\net al\n.\n72\nbuilt a disease network by linking two diseases for which the same modules were activated in the specific disease states and Liu\net al.\n73et al.\n73\nlinked diseases with common environmental influences. While a comprehensive program towards understanding all causal links between diseases is still in its infancy, it will be essential if we seek a deeper understanding of human disease.\nApplying network-based knowledge of disease\nNetwork pharmacologyApplying network-based knowledge of disease\nNetwork pharmacology\nOwing to the often unknown interactions between drug targets and other cellular components, drugs whose efficacy was predicted by specific target-binding experiments may not have the same effect in different clinical settings in which that target is of modified contextual importance (e.g., tissue-specific isoform compensates for the loss of function of the inhibited protein). Furthermore, single-target drugs may, perhaps, correct some dysfunctional aspects of the disease module, but could alter the activity of other network neighborhoods, leading to detectable side effects. This network-based view of drug action implies that most disease phenotypes are difficult to reverse through the use of a single ‘magic bullet,’ i.e., an intervention that affects a single node in the network\n7474\n. While network-based approaches represent a relatively recent trend in drug discovery, given the intricate network effects drug development must face, the nascent field of network pharmacology\n75\n, at the intersection of network medicine and polypharmacology, is poised to become an essential component of drug development strategies.\nThe utility of network-based approaches in drug discovery has been demonstrated in the search for antibiotic targets against bacterial metabolism. Given the relatively accurate metabolic maps (see\nBox 1\n), and that in bacteria flux balance analysis\n76\nand other flux-based methods\n77\nallow the prediction of the flux changes induced by drug-altered enzymatic activity, the metabolic impact of a hypothetical enzyme-blocking drug can be explicitly explored. This process has recently led to the identification and testing of potential new antibacterial targets\n7878\n. Furthermore, the coupled nature of metabolic fluxes allows for the possibility of rescuing a lost metabolic function through the blocking of additional enzymes, selected to re-route metabolic activity to compensate for the original loss of function, an intriguing alternative to gene therapy\n79\n.\nThere is increasing attention paid to therapies involving multiple targets that may be potentially more effective in reversing the disease phenotype than single drugs\n80\n. The efficacy of this approach has been demonstrated by combinatorial therapies of AIDS, cancer, or depression, raising an important question: can one systematically identify multiple drug targets with optimal impact on the disease phenotype? This is an archetypical network problem, leading to methods to",
         "Causal molecular pathways often coincide with the shortest molecular paths between known disease-associated components.",
         null,
         null,
         null
        ],
        [
         "7",
         "What is the reported range for the incidence of premature progesterone rise (PPR)?",
         "Yu-Ju Lin et al.,2015, DOI:https://doi.org/10.1186/s12958-015-0037-9:\n BackgroundIt is well known that gonadotropin-releasing hormone (GnRH) agonists and antagonists, known as GnRH analogues, can effectively eradicate the premature luteinizing hormone surge, and they are almost routinely used during controlled ovarian hyperstimulation (COH) in modern in vitro fertilization (IVF) treatment. However, unpredicted elevation of serum progesterone without an increase luteinizing hormone (LH) in the late follicular phase still occurs in patients co-treated with GnRH-a and COH, a phenomenon called premature progesterone rise (PPR) [1]. The reported incidence of PPR varies greatly from 3 % to 71 %, and it has been reported to be as high as 35 % in GnRH agonist cycles and 38 % in antagonist cycles [2 -5]. PPR has previously been defined based on absolute progesterone concentration or as a progesterone/ estradiol (P/E 2) ratio on the day of human chorionic gonadotropin (hCG) administration [6 -8]. The effects of elevated serum progesterone on oocyte maturation, endometrial development and receptivity, and prognostic outcomes after embryo transfer are controversial. An increasing number of retrospective studies suggest that progesterone elevation on the day of hCG administration may have a negative effect on pregnancy rate, and this hypothesis was confirmed by a meta-analysis based on the analysis of 60,000 IVF cycles [9 -11]. However, PPR remains a controversial issue because several studies did not find this association [6, 12, 13]. In addition, whether the level of serum progesterone in consecutive cycles remains constant or fluctuates in an individual undergoing repeated COH-IVF treatment is unknown. In this study, we aimed to explore the inter-cycle reproducibility of serum progesterone and P/E 2 ratio in the final step of triggering oocyte maturation in patients undergoing repeated consecutive COH-IVF treatment. The association of both serum progesterone concentration and P/E 2 ratio, and the patient ’s clinical characteristics and ovarian stimulation responses were also assessed.",
         "The reported incidence of PPR varies greatly from 3 % to 71 %.",
         null,
         null,
         null
        ],
        [
         "8",
         "What percentage of personalized embryo transfers were performed in the S-RIF group?",
         "Mauro Cozzolino et al.,2020, DOI:https://doi.org/10.1007/s10815-020-01948-7:\n We identified 2110 patients with M-RIF and 488 with S-RIF (Fig.\n1\n). The general and clinical features of the cycles and patients included in the study are shown in Table\n1\n. Table\n2\nsummarizes the parameters of the test-guided IVF cycles among the different groups of patients. The retrospective nature of our study resulted in the differential distribution of some relevant variables (Tables\n1\nand\n2\n). To avoid bias, we included parameters with statistically significant differences and/or clinical relevance within a multivariate analysis model using generalized estimating equations. Some patients included in the M-RIF and S-RIF groups had infertility associated with uterine factors (Supplemental Table\n1\n). Patients who received the ERA test had a higher percentage of uterine pathologies than control patients or those undergoing PGT-A (Supplemental Table\n1\n). Patients scheduled for personalized ET after the ERA are shown in Supplemental Table\n21\n). Patients scheduled for personalized ET after the ERA are shown in Supplemental Table\n2\n. A large percentage of patients had an asynchronous or displaced WOI, particularly in the M-RIF group. The percentages of personalized ET (pET) were 25.7% and 39.3% in the S-RIF and M-RIF groups, respectively. The rate of euploid embryos in M-RIF for patients with PGT-A and ERA+PGT-A was 48.13%, 95% CI (46.56–49.69) and 39.35%, 95%CI (34.74–43.96), respectively,\nP\n= 0.225. In the S-RIF group, the rate of euploid embryos with PGT-A and ERA+PGT-A was 57.18%, 95%CI (54.71–59.65) and 62.13%, 95%CI (53.57–70.70), respectively,\nP\n= 0.708.\nFig. 1.\nOpen in a new tab\nPatient selection flowchart.\nIVF\nin vitro fertilization,\nM-RIF\nmoderate recurrent implantation failure,\nS-RIF\nsevere recurrent implantation failure,\npET\npersonalized embryo transfer,\nERA\nendometrial receptivity assay,\nPGT-A\npreimplantation genetic testing for aneuploidy\nTable 1.ERA\nendometrial receptivity assay,\nPGT-A\npreimplantation genetic testing for aneuploidy\nTable 1.\nNumber of embryo transfers (fresh and frozen) and general characteristics of patients before undergoing ERA, PGT-A, or ERA+PGT-A. Data are expressed as means or percentages with corresponding 95% confidence intervals within brackets. Superscripts in the\nP\nvalue column indicate statistically significant differences between groups with the same letter. The\nP\nvalues without superscripts indicate ANOVA results. Patients with no ERA, PGT-A, or ERA+PGT-A cycles are presented in the standard group\nM-RIF (\nN\n= 2110)\nS-RIF (\nN\n= 488)\nStandard\nPGT-A\nERA\nERA+PGT-A\nP\nvalue\nStandard\nPGT-A\nERA\nERA+PGT-A\nP\nvalue\nAge (mean)\n37.9\n[37.7–38.1]\na,c\n38.2\n[38.0–38.5]\na,d\n38.6\n[38.3–38.9]\nd,f\n38.5\n[38.1–39.0]\nc,f\n< 0.001\na\n< 0.001\nc\n0.0013\nd\n< 0.001\nf\n< 0.001\n38.5\n[38.2–38.7]\na,c\n38.3\n[38.1–38.6]\na,d,e\n38.9\n[37.8–41.9]\nd,f\n37.9\n[37.7–38.5]\nc,e,f\n< 0.001\na\n< 0.001\nc\n< 0.001\nd\n0.0082\ne\n< 0.001\nf\n< 0.001[37.8–41.9]\nd,f\n37.9\n[37.7–38.5]\nc,e,f\n< 0.001\na\n< 0.001\nc\n< 0.001\nd\n0.0082\ne\n< 0.001\nf\n< 0.001\nYears of infertility\n2.6\n[2.5–2.7]a\n2.2\n[1.9–2.5]\na\n2.6\n[2.2–3]\n-\n1.9\n[1.3-2.8]\n-\n0.0178\na\n0.0216\n2.7\n[2.4–2.9]\nc\n2.1\n[1.8–2.5]\n-\n2.2\n[1.7–2.9]\n-\n1.3 [0.9–2]\nc\n0.0016\nc\n0.0059\nBody mass index (kg/m\n2\n)\n23.2\n[23–23.4]\n-\n22.6\n[21.9–23.2]\n-\n22.7\n[22.1–23.2]\n-\n23.6\n[21.2–25.9]\n-\n0.1196\n23.2\n[22.8–23.6]\n-\n22.2\n[21.1–23.3]\n-\n22.9\n[21.5–24.2]\n-\n24.2\n[19.8–28.6]\n-\n0.3272\nMean number of previous fresh embryos transferred per patient\n2.22\n[2.22–2.24]\nb\n2.24\n[2.17–2.31]\nd\n2.4\n[2.31–2.49]\nb,d\n2.33\n[2.11–2.58]\n-\n0.0015\nb\n< 0.001\nd\n0.0412\n3.31\n[3.25–3.37]\n-\n3.19\n[3.06–3.32]\n-\n3.35\n[3.13–3.58]\n-\n3.25\n[2.85–3.70]\nc\n0.3708\nMean number of previous frozen embryos transferred per patient\n1.35\n[1.32–1.38]\na,b\n1.55\n[1.44–1.66]\na\n1.57\n[1.44–1.71]\nb\n1.64\n[1.35–2.01]\n-\n<0.001\na\n< 0.001\nb\n0.0029\n1.82\n[1.73–1.91]\na\n2.16\n[1.95–2.4]\na\n1.86\n[1.854–2.26]\n-\n2.25\n[1.86–2.72]\n-\n0.0046\na\n0.0101\nOpen in a new tab\nTable 2.2.16\n[1.95–2.4]\na\n1.86\n[1.854–2.26]\n-\n2.25\n[1.86–2.72]\n-\n0.0046\na\n0.0101\nOpen in a new tab\nTable 2.\nCharacteristics of IVF cycles in patients who underwent ERA, PGT-A, or ERA+PGT-A or standard IVF treatments. Data are expressed as means or percentages with corresponding 95% confidence intervals in brackets. Superscripts in the\nP\nvalue column indicate statistically significant differences between groups with the same letter. The\nP\nvalues without superscripts indicate ANOVA results. Patients with no ERA, PGT-A, or ERA+PGT-A cycles are presented in the standard group\nM-RIF\nS-RIF\nStandard\nPGT-A\nERA\nERA+PGT-A\nP\nvalue\nStandard\nPGT-A\nERA\nERA+PGT-A\nP\nvalue\nNumber of total embryo transfers\n2636\n183\n160\n21\n591\n72\n35\n6\nEmbryo transfers with donated oocytes\n1593\n15\n124\n1\n427\n9\n26\n0\nPercentage of donated oocyte-derived embryos among all embryos\n60.43\n[58.06–62.75]\na,b,c\n8.2\n[4.04–15.91]\na,d\n77.5\n[68.41–84.56]\nb,d,f\n4.76\n[0.64–27.8]\nc,f\n< 0.001\na\n< 0.001\nb\n0.0038\nc\n0.0057\nd\n< 0.001\nf\n< 0.001\n72.25b,d,f\n4.76\n[0.64–27.8]\nc,f\n< 0.001\na\n< 0.001\nb\n0.0038\nc\n0.0057\nd\n< 0.001\nf\n< 0.001\n72.25\n[67.54–76.52]\na,c\n12.5\n[4.4–30.71]\na,d,e\n74.29\n[50.55–89.09]\nd,f\n0\n[0–0]\nc,e,f\n< 0.001\na\n< 0.001\nc\n< 0.001\nc\n< 0.001\ne\n< 0.001\nf\n< 0.001\nNumber frozen embryos transferred\n1774\n183\n124\n21\n407\n72\n23\n6\nPercentage of transferred frozen embryos among all transferred embryos\n67.3\n[65.52–69.03]\na,b,c\n100\n[100–100]\na,e\n77.5\n[70.97–82.91]\nb,f\n100\n[100–100]\nc,e,f\n< 0.001\na\n< 0.001\nb\n0.0216\nc\n< 0.001\ne\n< 0.001\nf\n< 0.001\n68.87\n[65.09–72.41]\na,c\n100\n[100–100]\na,d,e\n65.71\n[52.0–77.2]\nd,f\n100\n[100–100]\nc,e,f\n< 0.001\na\n0.0182\nc\n< 0.001\nd\n0.039\ne\n< 0.001\nf\n< 0.001\nMean number of oocytes retrieved or donated\n14.1\n[13.71–14.51]\n-\n14.58\n[13.1–16.23]\n-\n14.6\n[13.28–16.06]\n-\n14.07\n[10.98–18.03]\n-\n0.8405\n15.26\n[14.38–16.2]\n-\n15.65\n[13.42–18.26]\n-\n13.18\n[11.31–15.36]\n-\n15.25\n[9.22–25.21]\n-\n0.3376\nMean number of MII oocytes inseminated\n12.48\n[12.18–12.78]\n-\n12.26\n[11.08–13.58]\n-\n13.09\n[12.13–14.13]\n-\n11.87\n[9.54–14.76]\n-12.48\n[12.18–12.78]\n-\n12.26\n[11.08–13.58]\n-\n13.09\n[12.13–14.13]\n-\n11.87\n[9.54–14.76]\n-\n0.6261\n13.77\n[13.1–14.48]\n-\n12.94\n[11.09–15.11]\n-\n13.17\n[11.47–15.13]\n-\n13.25\n[8.49–20.67]\n-\n0.8318\nFertilization rate per inseminated oocyte\n76.59\n[75.65–77.55]\n-\n76.16\n[73.34–79.09]\n-\n75.29\n[71.86–78.89]\n-\n77.53\n[65.83–91.3]\n-\n0.9024\n75.89\n[74.16–77.66]\n77.11\n[72.83–81.65]\n77.89\n[71.71–84.59]\n79.25\n[70.38–89.23]\n0.8022\nMean number of embryos available per oocyte pick-up\n5.05\n[4.83–5.27]\na,c\n3.29\n[2.78–3.9]\na,e\n4.28\n[3.54–5.17]\nf\n1.47\n[1.18–1.82]\nc,e,f\n<0.001\na\n< 0.001\nc\n< 0.001\ne\n< 0.001\nf\n< 0.001\n5.31\n[4.84–5.82]\na,c\n3.17\n[2.46–4.08]\na,d\n6.17\n[4.47–8.53]\nd,f\n1.5\n[0.85–2.64]\nc,f\n< 0.001\na\n< 0.001\nb\n0.8163\nc\n< 0.001\nd\n0.007\ne\n0.0782\nf\n< 0.001\nMean endometrial thickness (mm)\n8.9\n[8.8–9]\na,b\n9.4\n[9.1–9.8]\na,d\n8.3\n[8–8.6]\nb,d\n8.7\n[8.2–9.3]\n-\n< 0.001\na\n0.0301\nb\n0.0013\nd\n< 0.001\n8.8\n[8.6–9]\n-\n9.2\n[8.7–9.7]\n-\n8.5\n[7.6–9.4]\n-\n8.5\n[7.2–9.8]\n-\n0.4423\nMean number of days for endometrial preparation\n11.1-\n8.5\n[7.6–9.4]\n-\n8.5\n[7.2–9.8]\n-\n0.4423\nMean number of days for endometrial preparation\n11.1\n[10.9–11.3]\nb\n11.1\n[10.5–11.7]\nd\n12.3\n[11.7–12.9]\nb,d\n11.4\n[10.5–12.4]\n-\n0.002\nb\n< 0.001*\nd\n0.038*\n13.7\n[12.3–15.2]\nc\n11.4\n[10.7–12.1]\ne\n11.8\n[10.7–13.1]\n-\n13.7\n[12.3–15.2]\nc,e\n0.0095\nc\n0.0052\ne\n0.0128\nOpen in a new tab\nUnivariate ANOVA of the M-RIF group revealed a statistically significant difference in the overall mean implantation rates of the subgroups (\nP\n= 0.0053). The use of PGT-A yielded a better implantation rate per transfer (45.9%) than standard IVF (35.9%) with an OR of 1.34, 95% CI: 1.17–1.55,\nP\n< 0.001. Implantation rates per transfer were not improved over standard rates by ERA, OR 1.03, 95% CI: 0.85–1.24,\nP\n= 0.9926, and ERA+PGT-A, OR 0.94, 95% CI: 0.53–1.65,\nPP\n= 0.9926, and ERA+PGT-A, OR 0.94, 95% CI: 0.53–1.65,\nP\n= 0.9954. Significant differences were not detected between subgroups subjected to different tests (ERA vs PGT-A, ERA vs. PGT-A+ERA, PGT-A vs. PGT-A+ERA). Logistic regression models adjusted for control variables confirmed that within the M-RIF group, only the PGT-A test yielded significant improvement (AdjOR 1.22, 95% CI: 1.14–1.30,\nP\n< 0.001) over standard treatment, without statistical significance over ERA and ERA+PGT-A (Supplemental Table\n3\n). When comparing the other subgroups after adjusting for control variables, we found a significant difference between the ERA+PGT-A and ERA (OR 0.84, 95% CI: 0.77–0.92,\nP\n< 0.001), but no other subgroup comparison reached significance.\nUnivariate ANOVA of the implantation rate per transfer as calculated by the number of gestational sacs per number of embryos transferred revealed statistically significant differences between the M-RIF subgroups (\nPP\n= 0.005). The highest implantation rate was in PGT-A (47.2%) versus the control group (35.8%), ERA (35.6%), and ERA+PGT-A (31.82%). For M-RIF patients, the implantation rate per transfer was higher after PGT-A testing than after standard IVF (OR 1.61, 95% CI: 1.24–2.11,\nP\n= 0.002). There were no statistically significant differences between standard over ERA (OR 1.00, 95% CI: 0.74–1.34,\nP\n= 1) and ERA+ PGT-A (OR 0.84, 95% CI: 0.34–2.07,\nP\n= 0.9797). When the ORs were adjusted by logistic regression models for control variables, the PGT-A subgroup was significantly different from the control subgroup (AdjOR 2.69, 95% CI: 1.99–3.66,\nP\n< 0.001), while no statistically confirmed differences were identified between ERA and ERA+PGT-A and controls (Supplemental Table\n3\n). In addition, the ERA subgroup significantly differed from the PGT-A subgroup (AdjOR 0.40, 95% CI: 0.26–0.62,\nP\n< 0.001).\nTable\n3P\n< 0.001).\nTable\n3\nshows the ongoing pregnancy rates for all study groups based on the number of embryos transferred. Univariate ANOVA revealed statistically significant differences between M-RIF subgroups (\nP\n= 0.05). Post hoc testing between M-RIF subgroups revealed that only the PGT-A subgroup differed from the control group (OR 1.51, 95% CI: 1.12–2.05,\nP\n= 0.029). There were no significant differences between the standard group and ERA (OR 1.02, 95% CI: 0.73–1.42,\nP\n= 0.9997) and ERA+PGT-A (OR 0.89, 95% CI: 0.36–2.22,\nP\n= 0.9942). The PGT-A group differed from the control group (AdjOR 2.19, 95% CI: 1.55–3.07,\nP\n< 0.0001) when multivariate analysis was applied to adjust for control variables, while no statistical differences were detected for ERA and ERA+PGT-A (Supplemental Table\n3\n). The ERA group showed lower ongoing pregnancy rates compared with the PGT-A group (AdjOR 0.51, 95% CI:",
         "25.7% of embryo transfers were personalized in the S-RIF group.",
         null,
         null,
         null
        ],
        [
         "9",
         "What software was used for variant annotation in this study?",
         "Ismael Henarejos-Castillo et al.,2021, DOI:https://doi.org/10.3390/jpm11070609:\n 2. Materials and Methods 2.1. Participants and Inclusion Criteria A WES-based case-control study was conducted between 2017 and 2019 at infertility clinics in collaboration with a genetic diagnosis and reproductive medicine research department. The study recruited 118 women diagnosed with OF and 32 women as controls from Spain. Women exhibiting amenorrhea for >6 months with AMH values < 0.3 ng/mL, FSH values > 20 IU, and <5 follicles upon antral follicle count (AFC) via transvaginal ultrasound were classified as OF. Controls had AMH values > 1.5 ng/mL, FSH < 10 IU, and AFC > 10. Clinical outcome (live birth (LB)) for both groups was determined. All patients were <40 years old at the time of recruitment and were selected following clinical criteria with idiopathic disease and normal karyotypes, no FRM1 permutations, and no history of pelvic surgery, radiotherapy, chemotherapy, or autoimmune disorders. The International Genome Sample Resource (IGSR) database, which contains genomic variant information including allele frequencies, normal genomic variability, and ethnicity from 1271 healthy female individuals (OF was not considered an exclusion criterion), was used as a pseudocontrol population to optimise the study complementing control population [36-38]. The Shapiro-Wilk test was used to check normality of clinical variables, hormone levels, age, body mass index, AFC, and LB, while the Wilcoxon and Fisher tests were used to evaluate clinical ranges of AMH and presence or absence of variants, respectively, between cases and controls [39-41]. The study was approved by the institutional review board of the Instituto Valenciano de Infertilidad and Hospital La Fe (1709-PAM-090-PR).  J. Pers. Med. 2021, 11, 609 3 of 17 2.2.  J. Pers. Med. 2021, 11, 609 3 of 17 2.2. Pre-Processing, NGS, and Variant Calling Peripheral blood genomic DNA was isolated (Maxwell 16 lev blood DNA, Promega, Madison, WI, USA) and quantified using fluorescence spectroscopy (Qubit). Absorbance readings with Nanodrop confirmed the purity of DNA, with all samples yielding a 260:280 ratio of >1.8. DNA integrity was evaluated using TapeStation (Agilent Technologies, Santa Clara, CA, USA), and the DNA integrity number (DIN) was >7 (recommended threshold for NGS library prep strategies) for each sample. WES was performed on all DNA samples using SureSelect Clinical Research Exome V2 (Agilent Technologies) and Illumina sequencers (MiSeq or NextSeq). Reads from the 18,311 sequenced genes were aligned to the human reference genome (hg19) using the Burrows-Wheeler algorithm (version 0.7.17) mapper [42]. Subsequent variant calling was performed using GATK software (version 3.6.7.17) mapper [42]. Subsequent variant calling was performed using GATK software (version 3.6.0) following the standard pipeline the standard pipeline recommended by the developers of the software [ 43]. Variants were annotated using SnpEff software (version 4.3) [44] to obtain information on which position of the genome is affected by the variant, including if it is a protein coding sequence, which gene and in what position of said gene is located, and the biological consequences expected from the changes (e.g., if the variant disrupts the triplet reading frame of the DNA, is called a Frameshift variant). Furthermore, SnpEff retrieves information from the IGSR that allow the user to know if a certain variant is registered or not in the database [44]. 2.3. Variant Filtering, Processing, and Prioritisation V ariants were filtered based on several criteria (further detailed in Supplemental Figure S1): 1. A moderate or deleterious effect on protein coding sequence (according to SnpEff annotations). Moderate effect included Missense variants, UTR (5 ′+ 3′) and splice (acceptor or donor) variants; while deleterious effects included Frameshift, Nosense (stop codon gain/loss) variants, protein to protein contact modifier variants, structural interaction modifier variants and disruptive inframe variants. 2. Variants absent from the IGSR database were retained for downstream analysis. Meanwhile, variants present in the IGSR were only kept if their minor allele frequency (MAF) was lower than <0.05, based on the premise that purifying selection decreases allele frequency of variants that confer less fitness. 3. 3. Passage of quality criteria for coverage (>100×) as well as several parameters evaluated by GATK: Genotype Quality (GQ), which evaluates the confidence of the genotype attributed to a patient for a certain variant (homozygous for the reference allele, heterozygous or homozygous for the disease-associated allele); Position depth (DP) or total number of reads detected at a given position of the genome; Allele Depth (AD), the number of reads for the variant in that position. Further information is annexed with Supplementary Figure S1. After applying these filters, remaining variants were used to (i) identify variants with significant differences in frequency between controls and cases and (ii) classify variants if they were present in at least 10% of cases and completely absent from controls. This last point was based on the premise that variants found only in one individual are related to individual variation while variants shared by a subgroup have a higher probability of being biomarkers related to the disease. Prioritized variants in this step were researched to find possible links with fertility. In addition to IGSR, GnomAD, and dbSNP databases were consulted to determine if identified variants were already reported [45,46], and Genecards, Uniprot, and Gene Ontology databases were used to annotate gene function [ 47-49]. Case and control variant frequencies were tested by Fisher test in the R environment (version 3.4.4, 15 March 2018) [50], while variant processing and prioritisation were done in the Python environment (version 3.5.2, 26 July 2016) [51]. 2.4.5.2, 26 July 2016) [51]. 2.4. Patient Stratification and Ovarian Failure Subtypes Prediction Prioritised variants were used to stratify the study population based on patient genotype (homozygous for the reference allele, heterozygous or homozygous for the disease-associated  J. Pers. Med. 2021, 11, 609 4 of 17 allele). In order to find different subtypes of ovarian failure in the study population, unsupervised hierarchical clustering-with similarity and genomic distance values calculated with the Jaccard coefficient [52]-was used to group patients based on genomic variant profile. Jaccard’s formula is J (X, Y) = |X∩Y|/|X∪Y| where X is the genotype profile of the variants of one patient, while Y is the same for another patient. Calculating the similarity coefficient with Jaccard allow grouping together patients that share not only the same variants but also the same genotype for said variants. Optimal clustering was achieved with Ball and Hartigan indexes implemented in Nbclust (version 3.0) in R [53], which increase differences inter-cluster while trying to maximize similarities intra-cluster; this is, trying to find clusters of patients that are really similar by the genomic profile while separating them as much as possible from other patients that do not share similar profiles. Using the subtypes of ovarian failure generated and to generate a model capable of distinguish patients with OF, a Random Forest algorithm was trained with a 10-fold cross-validation (CV) 100 times in WEKA platform software (version 3.8.2, 22 December 2017) [54], WEKA by default stratified the folds maintaining the proportion of cases and controls (79%/21%). With Random Forest, the model created 500 forest each iteration and selected the best consensual tree [55,56]. Random forest also assigned a score to each variant based on the mean decrease in impurity (MDI), which represents how informative each variant is for stratifying the population according to genotype of said variant [57]. Key variants in the stratification were analysed to look for potential relations with fertility . Finally , to ascertain if the genomic variant profile found in our population is detected in an independent population and ensure that a determined subtype is reproducible, the female population from the IGSR database (n = 1271) [36] was evaluated using our predictive model.",
         "SnpEff software (version 4.3) was used for variant annotation.",
         null,
         null,
         null
        ],
        [
         "10",
         "What sugars besides glucose are transported by GLUT1?",
         "Ivana Vrhovac Madunić et al.,2021, DOI:https://doi.org/10.3389/fcell.2021.703671:\n GLUT1GLUT1 (SLC2A1) was the first facilitative glucose transporter to be discovered, having been purified from human erythrocytes (Kasahara and Hinkle, 1977) and cloned from HepG2 cells (Mueckler et al., 1985). GLUT1 is ubiquitously expressed in all organs, ranging from endothelial cells of the blood-brain barrier to fetal tissues; however, it is most abundant in erythrocytes, where it increases the cells’ glucose carrying capacity (Mueckler and Thorens, 2013). Besides glucose, it transports other sugars including mannose, galactose, glucosamine, and reduced ascorbate (Carruthers et al., 2009; Mueckler and Thorens, 2013). Several research groups reported GLUT1 expression in the placenta (Fukumoto et al., 1988; Bell et al., 1990; Farell et al., 1992; Takata et al., 1992; Jansson et al., 1993; Clarson et al., 1997). In different cells and tissues, GLUT1 is regulated by glucose concentration, insulin, and growth factors (Mueckler and Thorens, 2013). GLUT1 was the first GLUT to be found in the endometrium; more precisely, its mRNA is abundant in ESCs of both human and rodents. High GLUT1 levels have been detected in decidua during pregnancy, implying the role ofGLUT1 in maintaining pregnancy and fetal development (Sakata et al., 1995; Y amaguchi et al., 1996; Frolova et al., 2009). Furthermore, it was shown that during decidualization, GLUT1 protein levels can increase up to 10 times and are accompanied by an increase in the glucose uptake in vivo (Frolova et al., 2009; Neff et al., 2020). GLUT1 upregulation has also been reported in decidualized hESCs (Tamura et al., 2014). In contrast, downregulated GLUT1 in ESCs lead to a reduced glucose uptake and suppressed decidualization (Huang et al., 2019). Accordingly, in cultured hESCs, the GLUT1 knockdown impaired decidualization (Frolova and Moley, 2011a). A recent study reported that hESCs are able to survive the hypoxic environment by activating metabolic pathwaysvia GLUTs, which could be relevant for the menstrual and implantation period (Kido et al., 2020). In the human endometrium,GLUT1 expression is upregulated during the peri-implantation period, and is higher during the secretory phase compared to the proliferative phase (Korgun et al., 2001; Strowitzki et al., 2001; Zhang et al., 2020). In contrast, no difference was observed in the expression of GLUT3 in different phases. This result is consistent with immunohistochemical analysis (Zhang et al., 2020). An adequate glucose metabolism is crucial for endometrial differentiation and decidualization processes. These processes are mainly mediated by progesterone (P4) and partially by 17 β-estradiol (E2). These processes are mainly mediated by progesterone (P4) and partially by 17 β-estradiol (E2). In vitro studies demonstrated that GLUT1 is upregulated by P4 in murine ESCs, and furthermore that 17 β-estradiol E2 reverses the effect of P4, bringing GLUT1 expression back to its basic level (Frolova et al., 2009). Although this finding indicates the opposite effect of E2 and P4 on GLUT1, both hormones are present at the time of implantation, indicating their interplay Frontiers in Cell and Developmental Biology | www.frontiersin.org 3 September 2021 | Volume 9 | Article 703671 fcell-09-703671 September 1, 2021 Time: 16:21 # 4 Vrhovac Maduni ć et al. Endometrial Glucose Transporters in the regulation of GLUT1. A recent study confirmed these findings in vivo, showing that P4 was responsible for the upregulation of GLUT1 protein in the murine uterus (Zhang et al., 2020)., 2020). In women with idiopathic infertility, GLUT1 is significantly reduced, suggesting its enrolment in the regulation of endometrial function (von Wolff et al., 2003). Further, a GLUT1 knockdown resulted in failed embryo implantation in vivo, indicating that GLUT1 silencing affects the endometrium, rather than the embryo (Zhang et al., 2020). These findings propose that pregnancy failure may be associated to a lower GLUT1-mediated glucose metabolism, resulting in reduced endometrial receptivity. A recent in vitro study showed that high insulin concentrations downregulate the GLUT1 gene and protein expression, which is accompanied by a slight reduction in the glucose uptake in decidualizing hESCs (Jakson et al., 2020). Further clinical studies are necessary to determine the importance of insulin regulation in endometrial function and decidualization in patients with insulin resistance and hyperinsulinemia.GLUT3GLUT3 (SLC2A3) is a high-affinity glucose transporter expressed in tissues with an intensive glucose metabolism, such as the brain, placenta, preimplantation embryos, sperm, and testes (Maher et al., 1992; Shepherd et al., 1992; Haber et al., 1993; Burant and Davidson, 1994; Boileau et al., 1995; Hauguel-de Mouzon et al., 1997; Pantaleon et al., 1997). In the human endometrium, GLUT3 is expressed in the proliferative phase and decidua at a constant level throughout different menstrual stages or in early pregnancy (Hahn et al., 2001; von Wolff et al., 2003; Korgun et al., 2005). GLUT3 and GLUT1 are differentially expressed in utero-placental modulation. In mouse decidua, GLUT1 is upregulated after midpregnancy, whereas GLUT3 is slightly downregulated, indicating that this glucose transporter is not crucial after mid-pregnancy (Y amaguchi et al., 1996)., 1996). This decrease was confirmed in rats, where GLUT3 is found in the uterine epithelium and stroma, and its expression increases through day 4 since gestation (Korgun et al., 2001). A correlation between placental and uterine GLUT3 expression with serum progesterone levels has been shown during early pregnancy (Ferre-Dolcet et al., 2018). Studies using GLUT3 KO mice showed that GLUT3 is not essential for fertilization or implantation of blastocysts, but for the development of early post-implanted embryos. The deletion of GLUT3 inhibits early embryonic development due to apoptosis of ectodermal cells shortly after implantation (Schmidt et al., 2009). In summary, GLUT1 is a basal transporter that is responsible for constant glucose intake during the entire pregnancy, whereas GLUT3 is required for optimizing glucose uptake in the first trimester when maternal circulation through the placenta is not yet established. Putatively, GLUT3 may function as a “scavenger” glucose isoform, acting in circumstances when extracellular glucose levels are lower than the normal circulating ones (Brown et al., 2011). Therefore, GLUT3 may represent an important transporter for fetal supply in early gestation. However, further studies on GLUT3 expression and function in the endometrium are required.GLUT4GLUT4 ( SLC2A4) is an insulin-regulated transporter found in the heart, skeletal muscle, adipose tissue and brain (Bryant et al., 2002; McKinnon et al., 2014). Reversible translocation of GLUT4 to the cell surface, stimulated by insulin, leads to rapid glucose uptake into the cell (Pessin et al., 1999; Bryant et al., 2002). Several studies found muscle and fat cell insulin resistance to be associated with decreased GLUT4 expression and its impaired translocation (Dunaif, 1997; Shepherd and Kahn, 1999; Mueckler, 2001), contributing to insulin resistance and Type II diabetes (Bryant et al., 2002). Data on GLUT4 in endometrial tissue are scarce and inconsistent. Endometrial GLUT4 was detected only in epithelial cells by some research groups (Mioni et al., 2012), while others found it in both stromal and epithelial cells (Cui et al., 2015). In contrast, some reported the absence of GLUT4 in endometrial tissue and stromal cells (von Wolff et al., 2003; Frolova and Moley, 2011a)., 2003; Frolova and Moley, 2011a). Insulin-dependent endometrial GLUT4 was found maximally expressed in the follicular phase of menstrual cycle, with a decline in the luteal phase (Mioni et al., 2012; Zhai et al., 2012; Cui et al., 2015). In contrast, GLUT1, which is an insulin independent transporter, was found maximally expressed in the luteal phase and mainly localized in the cytoplasm of stromal cells predetermined to transform into decidual cells (Mioni et al., 2012). Although little is known about the insulin dependence of the endometrium, GLUT4 was found in endometrial epithelial cells (Mioni et al., 2004; Mozzanega et al., 2004; Fornes et al., 2010; Zhai et al., 2012). Moreover, Mioni et al. demonstrated that GLUT4 is dysregulated in the endometrium of PCOS patients. This is important, as PCOS is the most common metabolic and endocrine disorder affecting 5-10% of all women of reproductive age (Diamanti-Kandarakis, 2008). Chronic anovulation is one of its hallmarks, and about 75% of infertility due to anovulation is attributed to PCOS (Costello and Eden, 2003). Other important clinical manifestations of PCOS may also include obesity, insulin resistance (IR), and signs of androgen excess (Rotterdam ESHRE/ASRM-Sponsored PCOS Consensus Workshop Group, 2004). Endometrial GLUT4 is significantly decreased in hyperinsulinemic women with PCOS, compared to those with PCOS but without IR. This implies a link between hyperinsulinism and GLUT4 downregulation. In women with PCOS and no IR, endometrial GLUT4 was significantly decreased only in obese women (Mioni et al., 2004). This is important, as IR is found in 50-70% of all women with PCOS (Sirmans and Pate, 2013), consequently leading to hyperinsulinism (Dunaif, 1997) in both obese and non-obese patients, with obesity aggravating IR (Maison et al., 2001). It is noteworthy that 30-50% patients with PCOS are obese (Sam, 2007; Barber et al., 2019)., 2001). It is noteworthy that 30-50% patients with PCOS are obese (Sam, 2007; Barber et al., 2019). Hyperandrogenemia negatively affects glucose uptake in endometrial cells regulated by GLUT4 (Rosas et al., 2010; Zhang and Liao, 2010; Rivero et al., 2012; Li et al., 2015). However, Frontiers in Cell and Developmental Biology | www.frontiersin.org 4 September 2021 | Volume 9 | Article 703671 fcell-09-703671 September 1, 2021 Time: 16:21 # 5 Vrhovac Maduni ć et al. Endometrial Glucose Transporters one study reported no connection between the hyperandrogenic syndrome per se and GLUT4-regulated glucose uptake (Mioni et al., 2004). All these findings can possibly explain the reduced pregnancy rate in women with PCOS at only 30-40%, although their ovulation rate reaches up to 80% (Imani et al., 2002). Furthermore, they experience a higher rate of spontaneous and recurrent abortions (Giudice, 2006). Furthermore, they experience a higher rate of spontaneous and recurrent abortions (Giudice, 2006). It has been shown that hyperinsulinemia does not have a direct effect on LH secretion, and therefore does not cause the hypothalamic-pituitary-ovarian axis dysfunction in PCOS. However, in vivo and in vitro studies confirmed that insulin excess stimulates further androgen production in ovaries via insulin receptors on theca/interstitial ovarian stromal cells, therefore aggravating already increased ovarian androgen production in PCOS and lowering the hepatic sex-hormone binding globulin (Smith et al., 1987; Plymate et al., 1988; Nestler et al., 1998; Baillargeon, 2007; Rosenfield and Ehrmann, 2016). IR in PCOS patients is commonly treated with antidiabetic drug metformin (1,1-dimethylbuguanide hydrochloride), which is known to improve insulin sensitivity and decrease insulin and androgen levels, thus promoting ovulation (Harborne et al., 2003; Lord et al., 2003)., 2003; Lord et al., 2003). Endometrial GLUT4 is lower in PCOS than in non-PCOS patients. However, after 3 months of metformin treatment, GLUT4 mRNA and protein expression is increased in the endometrium of PCOS patients (Zhai et al., 2012). Moreover, metformin directly restores endometrial function by increasing GLUT4 expression (Carvajal et al., 2013). In addition, metformin increases serum IGF-1 and glycodelin during the luteal phase (Jakubowicz et al., 2001; Palomba et al., 2006), which may contribute to an improved endometrial environment for the establishment and maintenance of pregnancy (Carvajal et al., 2013). GLUT4 may be regulated by steroid hormones. Progesterone (P4) alone, and in combination with estradiol (E2), decreases endometrial GLUT4, suggesting that an abnormal hormonal secretion pattern, such as in PCOS, could disrupt normal GLUT4 expression (Cui et al., 2015)., 2015). The importance of GLUT4 in the endometrium is well established, although numerous studies were performed on only a small number of patients. Therefore, the elucidation of endometrial GLUT4 function and its molecular mechanisms in physiological and pathophysiological (such as PCOS) states is of high interest. Furthermore, due to its limited cellular expression and inducible nature, GLUT4 could be a desirable target for new non-hormone-based drugs (McKinnon et al., 2014).GLUT8The GLUT8 ( SLC2A8) transporter is mostly present in the brain and testes (Doege et al., 2000; Carosa et al., 2005), but is also detected in the uterus, ovary, and term placenta (Carayannopoulos et al., 2000; Ibberson et al., 2000; Limesand et al., 2004; Schmidt et al., 2009). It is localized in intracellular membranes of predominantly late endosomes and lysosomes, as well as in the ER and trans-Golgi (Stanirowski et al., 2017). The late-endosomal/lysosomal limiting membrane is the cellular hub for various metabolic reactions (e.g., mTORC1 recruitment to the lysosomal membrane, modulation of mitochondrial homeostasis, regulation of protein, and lipid content of the cellular membrane and membranes of various vesicles). Therefore, besides transporting glucose, GLUT8 may also function as a sensor for various metabolites in cellular metabolic homeostasis (Alexander et al., 2020). Some of the GLUT8 is cleaved into N-terminal and C-terminal peptides., 2020). Some of the GLUT8 is cleaved into N-terminal and C-terminal peptides. Of its 12 trans-membrane domains, the cleavage occurs at domain 10, which happens to be the only one that is 100% conserved across the species. The full-length and N-terminal cleaved proteins are retained at the endosomal/lysosomal boundary, whereas the carboxy domain accumulates in a different population of vesicles and is unlikely to perform transport (Alexander et al., 2020). GLUT8 is one of several GLUT species that exists as several alternatively spliced mRNA variants, and there is few available data on its functionality. Alexander et al. concentrated on its expression in breast cancer and found that GLUT8 has three such variants: v1 (full-length), v2 (missing exon 9 and part of the C-terminus) and v3 (missing exon 2 and 3). In normal human tissue, by far the highest expressions of v1 and v3 mRNA variants are found in the testis, followed by liver and muscle, and only very little in the mammary gland, with the full-length version being predominant and v3 accounting for only about 5-10% of the total GLUT8 mRNA. The expression survey found that only the fulllength GLUT8 variant is translated into stable, functional protein in both normal and tumor tissues (Alexander et al., 2020). In the endometrium, GLUT8 can provide glucose required for protein glycosylation in the ER during decidualization (Frolova and Moley, 2011b). GLUT8 is elevated in ESCs upon decidualization in mice and humans, whereas the protein expression is not increased (Frolova and Moley, 2011a). Mouse Slc2a8-/- mutants exhibited disordered decidualization, indicating that SLC2A8 plays an important role in this process (Adastra et al., 2012). Among other GLUTs, only GLUT8 was found to be reduced in hESCs in a hypoxic environment., 2012). Among other GLUTs, only GLUT8 was found to be reduced in hESCs in a hypoxic environment. Recently, it was reported that suppressing GLUT8 under hypoxia might promote the preferential use of glucose for glycolysis (Kido et al., 2020). Hence, gaining an understanding of hESCs behavior in the hypoxic conditions of menstrual and implantation periods is of great importance. GLUT8 is anchored at intracellular membranes, and it is not sensitive to insulin in most tissues, unlike certain other GLUT family members (e.g., GLUT4 and GLUT12). The blastocyst is an exception. Namely, it has been shown that in the blastocyst, GLUT8 translocates to the plasma membrane in response to insulin stimulation and imports glucose. Thus, it is considered critical for murine blastocyst survival (Carayannopoulos et al., 2000; Pinto et al., 2002; von Wolff et al., 2003).Sodium-Glucose Transporter 1Sodium-glucose transporter 1 is primarily expressed in the small intestine and kidneys, but it is also found in the heart, liver, pancreas, brain, lungs, prostate, and uterus (Wright et al., 2004; Vallon et al., 2011; Gorboulev et al., 2012; Kashiwagi et al., 2015; Vrhovac et al., 2015; Salker et al., 2017; Sharma et al., 2017; Vrhovac Madunic et al., 2017). The first report of SGLT1 expression and activity in murine and human endometrial epithelial cells showed that it is Frontiers in Cell and Developmental Biology | www.frontiersin.org 5 September 2021 | Volume 9 | Article 703671 fcell-09-703671 September 1, 2021 Time: 16:21 # 6 Vrhovac Maduni ć et al. Endometrial Glucose Transporters responsible for controlling glycogen accumulation essential for embryo implantation (Salker et al., 2017). Endometrial glycogen is associated with fertility. During peri-implantation and in early pregnancy, the increase in endometrial glycogen represents a highly important source of glucose. Glycogen production requires the cellular uptake of glucose, mostly by passive transporter GLUTs. Recently, another mechanism of accumulating glucose has been demonstrated via the secondary active transporter SGLT1, which can mediate glucose uptake even at low extracellular glucose concentrations (Salker et al., 2017). Glucose transport across the endometrium (at the estrus phase of the cycle) was detected in wild type, but not in SGLT1knockout mice. The SGLT1 deficiency notably decreased the endometrial glycogen, litter size, and weight of the offspring. Moreover, SGLT1 was upregulated upon the decidualization of primary hESCs, which is crucial for embryo implantation. Furthermore, endometrial SGLT1 levels are significantly lower in women with recurrent pregnancy loss when compared to women with a healthy pregnancy. The study reveals a novel mechanism for adequate endometrial glycogen storage for pregnancy as a key factor for embryo implantation. Overall, endometrial SGLT1 deficiency in the human during implantation may lead to early pregnancy failure and obstetrical complications, including low fetal growth (Salker et al., 2017). These results suggest that caution is needed in the period of preconception and during pregnancy if SGLT1 inhibitors are considered for the treatment of diabetes (Vallon, 2015; Song et al., 2016).Signaling Pathways in GLUT ExpressionOne of the crucial signaling pathways in the glucose metabolism is the insulin pathway, with insulin acting as a main regulator of blood glucose concentration by increasing glucose uptake in muscles and fatty tissue, while inhibiting its production in liver (Saltiel and Kahn, 2001). The primary function of insulin is to maintain glucose homeostasis by translocating GLUT4 to the cell surface via activation of the PI3K/Akt pathway (Bryant et al., 2002). Maintaining the physiologically relevant narrow range of blood glucose concentration requires tight and rapid control of insulin signaling. An essential step in successful pregnancy is decidualization of the human endometrium, a process consisting of many morphological changes and a functional differentiation of ESCs. This process depends on a complex interaction of transcription factors, cytokines, cell cycle regulators, and signaling pathways (Okada et al., 2018)., 2018). Proliferation of the endometrium is induced by insulin through the insulin-like growth factor 1 (IGF-1), induced by estrogen in the proliferative phase (Strowitzki et al., 2006). A study by Nandi et al. reported that impaired insulin signaling affects female reproduction health through the hypothalamic-pituitary-gonadal axis, resulting in altered cycles and follicular development. Authors also observed unsuccessful pregnancies, increased calcification of placentae, and small embryos in insulin-resistant and hyperinsulinemic model mice. Other studies proposed that hyperandrogenism in the ovaries of PCOS patients could disrupt the insulin pathway (Zhang and Liao, 2010) and affect the function of the endometrium (Diamanti-Kandarakis and Papavassiliou, 2006). Furthermore, hyperinsulinism observed in PCOS patients lowers the concentration of circulating glycodelin (biomarker of endometrial function) and insulin-like growth factor-binding protein-1 (IGFBP-1), a feto-maternal interface adhesion factor. This contributes to early pregnancy loss due to diminished endometrial receptivity caused by decreased uterine vascularity and endometrial blood flow (Jakubowicz et al., 2001). A recent study showed that the deletion of ovary-specific insulin and/or IGF-1 receptor(s) caused infertility in mice as a consequence of impaired ovulation, luteal differentiation, uterine receptivity, and steroid hormone signaling (Sekulovski et al., 2020). Furthermore, dual receptor ablation compromised decidualization, reduced endometrial thickness, and inhibited embryo implantation. Authors further observed decreased activation of ERK/MAPK signaling (Sekulovski et al., 2021). Authors further observed decreased activation of ERK/MAPK signaling (Sekulovski et al., 2021). Other pathological states, such as inflammation and obesity, cause insulin resistance by serine phosphorylation of the insulin receptor or insulin receptor substrate (IRS) proteins by TNF- α and PKC ε (Hotamisligil, 1999). A subsequent decrease in tyrosine phosphorylation inhibits kinase activity of IR. A study by Bergman et al. suggested that IR and type 2 diabetes can be caused by reduced hepatic clearance of insulin. Samuel and Shulman reported ectopic lipid accumulation, and the activation of unfolded protein response likewise led to the development of IR. Other signaling pathways have been linked to the regulation of GLUT expression. Other signaling pathways have been linked to the regulation of GLUT expression. Models of artificial decidualization and delayed/activated implantation have demonstrated that the expression of murine uterine epithelial GLUT1, GLUT8, and GLUT9B are regulated by estradiol and progesterone through Akt/MAPK/PRKAA signaling (Kim and Moley, 2009). This result implies that the hormonal changes associated with early pregnancy affect GLUT expression and indicate an important role of uterine glucose utilization in endometrial decidualization, embryo implantation, and maintenance of pregnancy. Similarly, GLUT1 has been shown to be induced by progesterone-activated Hif1α and c-Myc via the PI3K/Akt signaling pathway during decidualization at the sites of murine embryo implantation. The corresponding increase in GLUT1 expression was one of the drivers of decidual glycolysis taking place under normoxic conditions, a phenomenon the authors termed Warburg-like glycolysis (Zuo et al., 2015)., 2015). Another study analyzed the GLUT isoform expression in a PCOS-like model of uterine IR and hyperandrogenism (Zhang et al., 2016). Their results revealed both an increase and decrease in the mRNAs expression in rats treated with insulin. In contrast, most GLUT genes (including GLUT4) were downregulated in hyperandrogenic conditions. The authors suggested that chronic treatment with insulin and hCG causes changes in the GLUT expression due to altered PI3K/Akt and MAPK/ERK signaling pathways, which contributes to the initiation and development of local uterine IR. Another signaling pathway connecting GLUT expression and uterine decidualization is Notch signaling. In vitro and in vivo experiments demonstrated that inhibition of Notch signaling through endometrial loss of the RBPJ transcription factor leads to impaired decidualization in the murine uterus and in human endometrial fibroblasts associated with reduced progesterone receptor and GLUT1 expression (Strug et al., 2019).Frontiers in Cell and Developmental Biology | www.frontiersin.org 6 September 2021 | Volume 9 | Article 703671 fcell-09-703671 September 1, 2021 Time: 16:21 # 7 Vrhovac Maduni ć et al. Endometrial Glucose Transporters Epigenetic Regulation of Glucose Transporters Knowledge about epigenetic regulation of endometrial GLUTs is limited, as there is almost no experimental data on the epigenetic regulation of these transporters in the endometrium. Epigenetic modifications, including histone modification, DNA methylation, and non-coding RNAs, play an important role in the decidualization of hESCs by influencing the expression of target genes (Sugino et al., 2016; Liu et al., 2",
         "Mannose, galactose, glucosamine, and reduced ascorbate are transported by GLUT1.",
         null,
         null,
         null
        ],
        [
         "11",
         "What are the main children of the new top level term GO:0140110 transcription regulator activity?",
         "The Gene Ontology Consortium et al.,2018, DOI:https://doi.org/10.1093/nar/gky1055:\n GO resource content\nThe GO knowledgebase consists of the ontology and the annotations made using the ontology. As of the 5 September 2018 release (doi:10.5281/zenodo.1410625), there were ∼45 000 terms in GO: 29 698 biological processes, 11 147 molecular functions and 4201 cellular components, linked by almost 134 000 relationships. The number of annotations (as well as the percentage change since 2016 (\n1\n)) are shown in Table\n1\n. It is important to understand that the changes reflect two distinct processes: addition of annotations based on new evidence, and obsoleting of annotations that have been superseded by newer studies. We expect the number of obsoleted annotations to increase, due to our increasing annotation quality assurance efforts, described in more detail below.\nNew framework and repository, for gene function ‘models’New framework and repository, for gene function ‘models’\nWe have developed a more expressive computational framework for representing gene functions, which subsumes our current GO annotation framework, while maintaining compatibility. We refer to the framework as GO-\nC\nausal\nA\nctivity\nM\nodeling (GO-CAM, formerly referred to as ‘LEGO’ (\n1\n)) and to GO-CAMs as\nmodels\nto distinguish them from\nstandard annotations\n. A paper detailing GO-CAM is in preparation, but we summarize some properties here. In GO-CAM, each model is represented as a set of triples (\nsubject\n-\nrelation\n-object, with brackets {} as a set container), e.g. {\nABCA1\nenables\ncholesterol transporter activity\n;\ncholesterol transporter activity\noccurs in\nplasma membrane\n, and\ncholesterol transporter activity\npart_ofoccurs in\nplasma membrane\n, and\ncholesterol transporter activity\npart_of\ncholesterol homeostasis}. Each triple is supported by one or more pieces of evidence, consisting of a class from ECO and a citable source, usually a scientific publication. GO-CAM specifies the semantics of GO annotations, and how standard GO annotations can be combined into a larger model. Each GO-CAM model is represented using the Web Ontology Language (OWL), which is converted computationally to standard GO annotations (GAF format), ensuring backward compatibility. Users can browse, view, and download the available models in different formats at:\nhttp://geneontology.org/go-camhttp://geneontology.org/go-cam\n. The number of models is currently small, and most models contain only one gene product (standard annotations extended with additional contextual information, such as cell type). The GO is currently rapidly increasing the curation of GO-CAM models, and the model repository is growing. In particular, models are now available that each represent an entire regulatory or metabolic pathway.\nChanges to data access: new production pipelineChanges to data access: new production pipeline\nStarting in March 2018, the releases of the GO resource have been generated by a new data production pipeline using a refreshed software stack. This system allows for easily extensible error checking and improved reporting of quality assurance checks that ensure the quality and integrity of the released ontology and annotations. For users, one of the most important aspects of this pipeline is that the GO resource now produces monthly releases (named by release date) that are available at the GO site and can be referenced and obtained as stable Document Object Identifiers (DOIs) via Zenodo. This feature is critical for ensuring that GO-based analyses can be replicated in a consistent and referenceable manner through the inclusion of these DOIs and/or version of both the ontology and annotation files used (\n66\n). Our data production pipeline is currently hosted at the Lawrence Berkeley National Laboratory. We provide GO annotations in multiple formats: as standard GAF (Gene Association Format) and GPAD (Gene Product Association Data) annotation files, in Turtle (OWL serialization format) [\nhttp://current.geneontology.org/products/ttl\n], and a Blazegraph [\nhttp://current.geneontology.org/products/blazegraph\n] journal, replacing the legacy MySQL output.\nInteractions with the GO user community\nGO is an open project, and we encourage community contributions to the knowledgebase and software.\nAll users\n: GO can be contacted using the GO Helpdesk (\nhttp://help.geneontology.orgAll users\n: GO can be contacted using the GO Helpdesk (\nhttp://help.geneontology.org\n) for any questions or feedback about the annotations, the ontology, software, or other GO resources. If users notice an annotation that may not be correct, they should first review the original publication or data source. If the annotation still seems inaccurate, users are encouraged to report this to the GO helpdesk, and GOC members will review the annotation and remove or modify it if justified.\nAuthors\n: Authors can now see if their paper has been used for GO annotation directly in PubMed. Under the ‘LinkOut - more resources’ section of the PubMed abstract page, papers with annotations will have a link labeled ‘Gene Ontology annotations from this paper - Gene Ontology’ (see, e.g.\nhttps://www.ncbi.nlm.nih.gov/pubmed/3357510https://www.ncbi.nlm.nih.gov/pubmed/3357510\nwhich links to a web page on the GO site that shows the annotations based on evidence in that paper). If no GO LinkOut is present, that may indicate that the publication has not been used for GO annotation. Authors can contact the helpdesk at the GO website to suggest new annotations or changes to existing annotations.\nResources or consortia\n: The GOC collaborates with established data resources and other groups and consortia representing a particular area of biology. Recent examples include cilium biology (\n7\n,\n8\n), autophagy (\n9\n) and cardiac phenotypes (\n10\n). We encourage members of other interest groups to contact us to improve the ontology and annotations in their areas of expertise.\nTracking all contributions:\nMost aspects of the GO project management are now based in GitHub (\nhttps://github.com/geneontology\n). In addition to tracking ontology change requests (\nhttps://github.com/geneontology/go-ontology). In addition to tracking ontology change requests (\nhttps://github.com/geneontology/go-ontology\n), we now use GitHub to collect feedback on annotations (\nhttps://github.com/geneontology/go-annotation\n). For users familiar with GitHub, we encourage them to submit any requests directly to GitHub, where they can follow all further discussion and actions. Otherwise, issues and queries can be submitted to our helpdesk (\nhttp://help.geneontology.org\n).\nIncreased focus on annotation quality controlThe GO resource is now 20 years old. The longevity of the resource adds a challenge to maintain and update the many existing annotations, as many of the findings published during that time have become much more precise, or were reinterpreted or superseded. We have made it a high priority to identify and correct inaccurate and out-of-date legacy annotations to make sure that GO continues to consistently reflect current knowledge. We have taken a number of different approaches to tackle this challenge. First, to ensure consistency and quality, GO biocurators meet regularly for training, establishment of annotation guidelines, and coordinated review of specific areas of biology. More recently, we have made significant efforts to integrate annotation review with ontology improvements, taking advantage of suggested changes to the ontology to clarify term definitions, intended usage, and coordinate annotation practices with curators. In addition, quality assurance is performed centrally, both computationally, to ensure annotations are valid, and manually, to ensure they accurately represent the experimental findings.We have discovered that one of the most powerful approaches to quality control and consistency is the phylogenetic approach. Originally developed as a means of propagating annotations from experimentally studied genes to evolutionarily related genes in other species, the phylogenetic perspective provides a unified view of all experimental annotations within a evolutionarily-related protein family, allowing curators to more easily find outlier annotations (see e.g. (\n1111\n)). In parallel, the development of GO-CAM models has been useful in identifying inconsistent annotation practices, and has provided opportunities to develop consortium-wide annotation guidelines. Another observation that emerged is that older annotations from isolated phenotypic observations, taken outside of other contextual data, often do not provide evidence of direct involvement of a gene in a biological process. If inconsistencies are noticed, they are reported to the contributing group for verification and correction as appropriate.In a pilot quality assurance effort, we have requested the review (by GO Consortium biocurators) of ∼2500 manual annotations (<0.01% of the total corpus) that were judged questionable by one of the strategies above. Approximately 70–80% of the annotations flagged for review were modified to a more appropriate term or removed. We will continue to work on improving the quality of the annotations and reviewing legacy data when appropriate. As a result, we expect that the increase in annotations and new ontology terms may not be as rapid as in the past, at least for the main species annotated by the consortium members, as a greater proportion of our efforts will be dedicated to reviewing and revising older annotations.\nOntology revision and integrationOntology revision and integration\nSince our last update article, we have developed an entirely new process for ontology editing and maintenance that has dramatically increased efficiency and enabled extensive real-time quality checks. Ontology editing is now performed in an OWL-based environment using the ontology editing tool Protégé (\n12\n). The ontology is versioned and tracked using a GitHub repository (\nhttps://github.com/geneontology/go-ontology\n). One major advantage of the new ontology management process is that the work can be parallelized among multiple editors, thus increasing efficiency. In addition, real-time quality checks prevent errors that would otherwise require revisiting the same editing task more than once to correct them.GO continues to integrate and align with external ontology resources in two main ways: import of sub-ontologies used to define GO terms, and inclusion of external cross-references. GO utilizes the structure of external ontologies to aid in reasoning and in the automatic inference of relations between GO terms (\n13\n). GO imports subsets of these external ontologies that include information about anatomical structures, cell types, chemicals and taxonomic groupings: Uberon (\n14\n), Protein Ontology (\n15\n), Plant Ontology (\n16\n), ChEBI (\n17\n), Relations Ontology (\n18\n), NCBI Taxonomy (\n19\n), Sequence Ontology (\n20\n), Ontology of Biological Attributes (\nhttp://www.obofoundry.org/ontology/oba.html\n), Fungal Anatomy Ontology (\nhttp://www.obofoundry.org/ontology/fao.html\n), Phenotypic Quality Ontology (\nhttp://obofoundry.org/ontology/pato.html\n), and Common Anatomy Reference Ontology (\nhttp://www.obofoundry.org/ontology/caro.html), and Common Anatomy Reference Ontology (\nhttp://www.obofoundry.org/ontology/caro.html\n). GO also maintains cross-references between terms and multiple widely-used external resources, including Reactome (\n21\n), The Annotated Reactions Database (Rhea) (\n22\n), Enzyme Commission (EC;\nhttp://www.sbcs.qmul.ac.uk/iubmb/enzyme/\n), IntAct, Complex Portal (\n23\n) and MetaCyc (\n24\n).\nRefactoring the molecular function branch of GO23\n) and MetaCyc (\n24\n).\nRefactoring the molecular function branch of GO\nPreviously, there was a trend in GO molecular function ontology development to focus on adding terms that describe molecular binding activities of specific gene products. The advantage of such terms is that annotations can often be made unequivocally based on results from a single experiment. However, this approach has led to a complex ontology structure and a proliferation of annotations that individually represent only a partial functional description of a gene product. Annotations to binding terms can obscure annotations to more informative function terms, making annotations more difficult to interpret. For example, one can annotate\nCDK1\n(UniProtKB:\nP06493CDK1\n(UniProtKB:\nP06493\n) separately with ‘GO:0030332 cyclin binding’, ‘GO:0005524 ATP binding’, ‘GO:0005515 protein binding’, and ‘GO:0004674 protein serine/threonine kinase activity’. However, these are all aspects of a more precise molecular function that is more informative than the sum of these parts: ‘GO:0004693 cyclin-dependent protein serine/threonine kinase activity’. In the GO molecular function refactoring, we recognized that, while specific binding events are an essential mechanism by which gene products function, an individual binding activity is almost never sufficient in itself to describe molecular function in a larger biological context (\n2525\n). One of the primary goals of our refactoring was to ensure that the ontology contains the terms necessary to describe these higher-level functions, and that they have a path to the root of the ontology that is not simply under the generic ‘GO:0005488 binding’ term. Accordingly, we reinstated some previously obsoleted terms and added new terms, as well as additional relations. We also addressed the structure of the ontology so that the upper-level terms would be more biologically meaningful and have more uniform specificity. We removed 8 terms from the top level and added four new terms (Figure\n2\n). Most of the terms that were formerly direct children of ‘GO:0003674 molecular function’ were moved under more biologically meaningful terms: for example, ‘GO:0042056 chemoattractant activity’ and ‘GO:0045499 chemorepellent activity’ were moved under ‘GO:0048018 receptor ligand activity’, while ‘GO:0036370\ndd\n-alanyl carrier activity’ and ‘GO:0016530 metallochaperone activity’ were moved under the new term ‘GO:0140104 molecular carrier activity’ (representing an activity of ‘directly binding to a specific ion or molecule and delivering it either to an acceptor molecule or to a specific location’). We have also created a new term ‘GO:0104005 hijacked molecular function’ as a parent of terms such as ‘GO:0001618 virus receptor activity’, which is, from the standpoint of the protein being annotated, not a normal function, but nevertheless relevant for some of our users.\nFigure 2.\nOpen in a new tab\nThe\nmolecular function\nbranch, before and after refactoring. The term marked with an ‘x’ in the left-hand panel has been obsoleted. Terms moved (assigned to a new parent) are indicated by arrows. New terms (right panel) are marked with ‘NEW’.\n1\nThe class label ‘electron carrier activity’ was changed to ‘electron transfer activity’.1\nThe class label ‘electron carrier activity’ was changed to ‘electron transfer activity’.\nFinally, we have made significant changes to the structure representing the molecular functions of transcription factors (Figure\n3\n). This refactoring was carried out in collaboration with experts in transcription factors and gene regulation from the Gene Regulation Consortium (GRECO;\nhttp://thegreco.orghttp://thegreco.org\n). In keeping with our design principle of having terms that describe higher-level functions, we have created a new parent term to group all functions that directly regulate transcription, ‘GO:0140110 transcription regulator activity’. The formerly top-level term ‘GO:0000988 transcription factor activity, protein binding’ has been obsoleted because this activity was partly covered by other terms in the ontology and its usage was inconsistent. Accordingly, its children have either been obsoleted or subsumed under different terms (merged or moved). The new top level term ‘GO:0140110 transcription regulator activity’ has three main children - ‘GO:0003700 DNA-binding transcription factor activity’ (formerly labeled ‘transcription factor activity, sequence-specific DNA binding), ‘GO:0140223 general transcription initiation factor activity’ and ‘GO:0003712 transcription coregulator activity’.\nFigure 3.\nOpen in a new tabFigure 3.\nOpen in a new tab\nCurrent structure of the ‘GO:0140110 transcription regulator activity’ branch of the ontology.\nThe transcription factor areas of GO had previously been refactored between 2010 and 2012 (\n26\n,\n27\n) with the aim of more finely capturing all combinations of different types of protein and DNA binding activities (e.g. binding to different types of regulatory regions such as promoters and enhancers) and transcription regulation processes (positive and negative regulation). However, this structure, while very precise, has proved very difficult to use by biocurators, resulting in inconsistent annotations. Additionally, end-users have had difficulty with common queries, such as comprehensively identifying the set of all transcription factors in a given species. We expect that even more improvements to the ontology structure, as well as more consistent annotations to transcription regulator terms, will be available in 2019.Defining the boundaries of biological processes: MAP kinase signaling and extracellular matrix as examplesWe have used our integrated annotation review and ontology development methodology to refine two areas of the ontology, the MAP kinase signaling pathway and the representation of the extracellular matrix. Refinements to the MAPK cascade included defining the molecular functions that are parts of the process: ‘GO:0004707 MAP kinase activity’, ‘GO:0004708 MAP kinase kinase activity, ‘GO:0004709 MAP kinase kinase kinase activity’ and ‘GO:0008349 MAP kinase kinase kinase kinase activity’. All other upstream and downstream molecular functions/biological processes will be modeled in GO-CAM with causal relationships between them and the MAPK cascade. We also enumerated the types of cascades based on current literature and on discussions among expert model-organism curators, trying to keep the distinctions useful across multiple taxa. There are four direct subtypes of ‘GO:0000165 MAPK cascade’: ‘GO:0070371 ERK1 and ERK2 cascade’, ‘GO:0070375 ERK5 cascade’, ‘GO:0071507 pheromone response MAPK cascade’ and ‘GO:0051403 stress-activated MAPK cascade’. Some other MAPK processes such as ‘GO:1903616 MAPK cascade involved in axon regeneration’ will eventually be obsoleted, as these combine two or more other GO terms and can be composed in GO-CAM. For the refinement of the extracellular matrix area of the ontology, we worked with external experts to add terms that were useful grouping terms such as ‘GO:0062023 collagen-containing extracellular matrix’. We also obsoleted or merged terms that were poorly annotated and thought to represent an outdated view.GO subsets (slims)\nA GO subset (or slim) is a set of GO terms selected to provide an overview of the functions, locations or roles of a set of genes. The subset can be developed for high coverage of specific species, or to represent only certain areas of the ontology, and in most cases, contain only high-level GO terms to provide a broad biological overview. Another use of subsets is to blacklist certain terms for annotation: GO has two such subsets, one to flag terms that should not be used for manual annotation, and one for terms that should not be used at all. GO maintains two additional subsets, the Generic GO slim and the Alliance of Genome Resources (\nhttps://www.alliancegenome.org/\n) slim. GO also hosts subsets useful to groups using GO; we currently have 11 such subsets (Table\n2\n;\nhttp://www.geneontology.org/page/go-subset-guide2\n;\nhttp://www.geneontology.org/page/go-subset-guide\n). Each subset provides a global overview of gene functions. Each subset now has a designated contact person to resolve any issue resulting from ontology changes (see Ontology revision and integration).\nOther developments\nThe GO ribbon: a configurable tool for visualizing GO annotations\nMany genes have large numbers of annotations, making it difficult to get a quick overview of a gene function, or the functions of gene sets. We have developed the GO ribbon specifically to help users visualize and explore the functions of a gene. The GO ribbon visualization metaphor borrows from a viewer originally developed by the Mouse Genome Database team (\n28\n), but in contrast, the GO ribbon was developed as a lightweight, reusable widget that can be embedded in any website, and retrieves data directly from the GO resource via API.To generate a GO ribbon, all the functions (GO terms) associated with a gene of interest are mapped onto a specified GO subset using the ontology structure. The end result is a simple graphical representation of a gene's functions (Figure\n4\n). The ribbon is interactive, allowing users to drill down to more specific functions by selecting a high-level category such as ‘GO:0030154 cell differentiation’, ‘GO:0050877 nervous system process’, or ‘GO:0003700 DNA-binding transcription factor activity’, and to filter the functions based on the evidence codes provided in the GO annotations. This overview of gene functions is particularly useful when comparing the functions of different genes in the same species, or the functions of orthologous genes across different species.\nFigure 4.\nOpen in a new tab\nGO ribbon representation. Darker boxes indicate terms with the most annotations; white boxes represent terms that are not annotated for this protein (\nMus musculusMus musculus\nSox7, MGI:98369). Screenshot obtained from\nhttps://www.alliancegenome.org/gene/MGI:98369\n.\nThe GO ribbon is a React component available on GitHub (\nhttps://github.com/geneontology/ribbon\n) and as a NPM package\nhttps://www.npmjs.com/package/@geneontology/ribbon\n). The GO ribbon widget is currently used by the Alliance of Genome Resources.\nGO annotations from high-throughput experiments\nData from high-throughput experiments are generally collected in a hypothesis-free manner, and consequently do not generally provide as strong evidence of gene function as small-scale molecular biology experiments that currently support most of the experimental GO annotations. In addition, high-throughput experiments can be subject to relatively high false positive rates. Users may therefore want to filter out these experimental annotations in some applications of the GO. To make this possible, starting in 2018, in collaboration with the Evidence and Conclusions Ontology (\n29\n) (\n2), the GO has added several new evidence codes to describe high-throughput experiments: ‘ECO:0006056 high throughput evidence used in manual assertion’ (HTP), and the subclasses: ‘ECO:0007005 high throughput direct assay evidence used in manual assertion’ (HDA), ‘ECO:0007001 high throughput mutant phenotype evidence used in manual assertion’ (HMP), ‘ECO:0007003 high throughput genetic interaction evidence used in manual assertion’ (HGI) and ‘ECO:0007007 high throughput expression pattern evidence used in manual assertion’ (HEP). To accompany the new evidence codes, we have provided annotation guidelines to help identify and curate high-throughput datasets that meet the GO Consortium annotation criteria. Consortium members have reviewed papers with more than 40 annotations using a single evidence code, and updated the evidence codes, or removed the annotations if appropriate. There are currently over 31 000 annotations that have HTP evidence codes from 140 research articles, representing <5% of experimental GO annotations. The identification of annotations derived from high-throughput experiments allows users to choose to exclude these from their analyses, if they are concerned that these annotations may lead to an increased bias in data analysis. This is likely to be particularly important, as is often the case, when GO is used to interpret types of data similar to those on which the annotations are based.",
         "The main children are GO:0003700 DNA-binding transcription factor activity, GO:0140223 general transcription initiation factor activity and GO:0003712 transcription coregulator activity.",
         null,
         null,
         null
        ],
        [
         "12",
         "What complement components were found deposited in the lungs of COVID-19 patients?",
         "Cynthia Magro et al.,2020, DOI:https://doi.org/10.1016/j.trsl.2020.04.007:\n Using pulmonary and cutaneous biopsy and autopsy samples from 5 individuals with severe COVID-19, we document that at least some SARS-CoV-2-infected patients who become critically ill suffer a generalized thrombotic microvascular injury. Such pathology involves at least the lung and skin, and appears mediated by intense complement activation. Specifically, we found striking septal capillary injury accompanied by extensive deposits of the terminal complement complex C5b-9 as well as C4d and MASP2 in the lungs of 2 cases examined, and a similar pattern of pauci-inflammatory complement mediated microthrombotic disease in the skin of 3 cases with retiform and purpuric lesions, with C5b-9 and C4d deposition in samples taken from both cutaneous lesions and normal-appearing skin.Our histologic findings are consistent with emerging observations suggesting that COVID-19 has clinical features distinct from typical ARDS. That is, COVID-19-related severe respiratory distress can be manifest by relatively well-preserved lung mechanics, despite the severity of hypoxemia, characterized by high respiratory compliance, high shunt fraction, and prolonged requirement for mechanical ventilation.\n10\n,\n11\nThe pathology in these cases might therefore be expected to differ from the diffuse alveolar damage and hyaline membrane formation which are hallmarks of typical ARDS. Albeit preliminary pathology studies of lungs from COVID-19 cases described DAD with edema, hyaline membranes, and inflammation, followed by type II pneumocyte hyperplasia, features characteristic of typical ARDS,\n8\n,\n98\n,\n9\nthe pulmonary abnormalities in our patients appear largely restricted to the alveolar capillaries, that is, more of a thrombotic microvascular injury with few signs of viral cytopathic or fibroproliferative changes. An increase in the dead space fraction might be anticipated with this type of pathology, i.e., respiratory failure accompanied by greater lung compliance and less pulmonary consolidation than is characteristic of typical ARDS. Indeed, the histologic pattern of pauci-cellular terminal lung parenchymal injury with septal capillary damage we document resembles images captured in early case reports in the Chinese language literature of severe COVID-19 pneumonitis.\n1616\nWe now show that this pathologic pattern, atypical for classic ARDS, is accompanied by extensive deposition of complement components within the lung septal microvasculature. With such extensive complement involvement, membrane attack complex-mediated microvascular endothelial cell injury and subsequent activation of the clotting pathway, leading to fibrin deposition, might be anticipated,\n17\nas observed in our cases. It is also consistent with the very high d-dimer levels found in the 3 cases in which it was assessed. Vascular deposition of C5b-9 is a key feature of many microthrombotic syndromes, regardless of the particular syndromic complex, including catastrophic antiphospholipid antibody syndrome, atypical hemolytic uremic syndrome, purpura fulminans and severe multiorgan malignant atrophic papulosis, and they may respond to anticomplement therapies.\n18\n,\n19\n,\n20\n,\n21\n,\n22\n,\n2318\n,\n19\n,\n20\n,\n21\n,\n22\n,\n23\nA preclinical model of SARS infection emphasizes the potential role of complement in the parenchymal lung injury of coronavirus infection.\n7\nMouse-adapted SARS-CoV MA15 infection of C3\n−/−\nmice led to significantly less weight loss and respiratory dysfunction than seen in wild-type mice, and this occurred despite equivalent viral loads in the lung.\n7\nAnother feature reflective of a prominence of complement-mediated pathology rather than simply inflammation or vasculitis in our COVID-19 patients was the extent of septal and intra-alveolar neutrophilia. It was present, but pauci-inflammatory. SARS-CoV-infected wild-type mice also had higher levels of neutrophils in the lung than their C3\n−/−\ncounterparts.\n7−/−\ncounterparts.\n7\nTissue neutrophilia may be attributable to the neutrophil chemoattractant properties of complement. Both neutrophils and complement are key sentinels of innate immunity and also modulate thrombogenic pathways, the latter thought related to C5a receptor/tissue factor cross-talk mediated by neutrophils.\n24\nNeutrophilia in human SARS-CoV patients is associated with a poor outcome and could be an index of the extent of complement activation.\n7\n,\n25\nPeripheral neutrophilia is not a significant observation in limited series of COVID-19 patients from China\n26\nor the United States,\n27\nalthough there are insufficient numbers to assess its correlation with severity of disease. We have observed an increase in peripheral blood neutrophil vacuolization and granule content in our patients, consistent with an activated state, even if absolute numbers are not elevated.Systemic activation of complement was reflected by elevated complement levels in the sera of SARS-CoV-infected wild-type mice. We measured complement levels in only one of our cases, Case 3. They were only moderately elevated, perhaps reflecting consumption in tissues, with reciprocal depression of circulating levels. Indeed, we found deposition of AP and LP components in normal-appearing skin in 3 cases, not only in biopsies of retiform cutaneous lesions from those patients. Retiform purpura reflects the cutaneous manifestation of an occlusive microthrombotic process.\n28\nIn summary, while complement does not appear to play a major role in controlling CoV replication, it may have a critical role in its pathogenicity.There are several intriguing questions that remain to be addressed as additional cases are investigated. For example, one might have anticipated thrombocytopenia in the context of a systemic microvascular thrombosis in our cases. Although mild thrombocytopenia (100–150 × 10\n9\n/L) has been recorded in 20% of COVID-19 patients,\n26\nsuch values were seen in only 2 of our 5 patients, nor were platelets a prominent component of the fibrin microthrombi seen histologically. It is possible that infection-related thrombocytosis was mitigated by the development of severe vasculopathy. However, there is also precedent for such platelet values in an atypical hemolytic-uremic syndrome (aHUS)-type of microangiopathy. Platelet counts may be in the normal range despite severe thrombotic disease.\n2929\nSecond, there appears to have been a delay of 5–9 days in most, if not all, of our patients between the onset of typical respiratory symptoms, including a nonproductive cough, myalgias, fatigue, fever, and mild dyspnea, and a catastrophic change in respiratory status. This may reflect an evolution of immune processes—antiviral IgM levels would expect to emerge at this point—or perhaps the fact that the complement cascade is a threshold pathway. When activated to a great extent it may exceed the capacity of complement regulatory proteins, both soluble and normally present in abundance on the microvasculature.\n29\nThat raises the issue of why only a subset of SARS-CoV2-infected patients develops such severe disease with features atypical for ARDS. The age range was wide among our cases, as was the degree of pre-existing immune suppression. But there are a variety of complement regulatory factor polymorphisms or mutations\n2121\nas well as coagulation pathway mutations which could promote susceptibility to enhanced complement activation and thrombosis in a given individual that should be investigated.\nOur results, particularly when viewed in the context of murine models of SARS-CoV infection, suggest the hypothesis that intervention with inhibitors of the alternative or lectin complement pathways, or both, may be relevant to severe COVID-19 in humans. Multiple studies have documented a role for the AP in microvascular injury with C5b-9 deposition,\n13\n,\n30\n31\nbut the lectin pathway has not been similarly investigated. Support for the involvement of the LP in COVID-19 comes from the discovery that MBL binds to the SARS-CoV spike glycoprotein.\n32\nA complex of MBL with MASP2 is the first step in LP activation, and part of a positive feedback loop leading to sustained AP activation, with inflammation and concurrent activation of the coagulation cascade.\n33\n,\n3433\n,\n34\nAlthough documented experimentally for SARS-CoV, this binding is speculative in terms of SARS-CoV-2, as certain sites on its glycoprotein spikes are mutated compared to SARS-CoV.\n35\nHowever, glycosylation sites for high-mannose structures with the potential to similarly engage MBL, and thus activate MASP2, have been identified for SARS-CoV-2.\n35\nOur data showing co-localization of products linked to activation of the AP and LP with SARS-CoV-2 spike glycoproteins are consistent with this hypothesis.\nThere may also be pathways apart from virus spike engagement by which LP and AP are activated. First, MBL and ficolins bind to altered glycan structures present on injured cells, enabling complex formation with MASP2.\n36\nIn addition, SARS-Cov1 and SARS-CoV use Angiotensin Converting Enzyme (ACE2) as an entry point to cells.\n6\nAngiotensin I and angiotensin II have been associated with inflammation, oxidative stress, and fibrosis, and ACE2 is involved in their deactivation.\n3737\nIf overwhelming coronavirus infection, with binding to ACE2 on epithelial targets not only in the lung but in other tissues expressing these proteins, including the kidney, intestines, and brain, were to interfere with ACE2 activity, the resulting increases in angiotensin II could lead to reactive oxygen species formation and interference with antioxidant and vasodilatory signals such as NOX2 and eNOS, with further complement activation. This has been demonstrated in a rat model of angiotensin II overexpression linked to other disorders.\n38\nThe potential loss of auto-vasoconstriction and regulation of lung blood flow through injured vascular segments would also lead to increased shunting and severe hypoxemia.\nFig 10\nsummarizes the potential mechanisms of SARS-CoV2 associated complement activation, and its possible interaction with coagulation pathways.\nFig 10.\nOpen in a new tab\nModel for AP and LP complement activation by SARS-CoV2, and its interaction with coagulation cascades.Given these data, use of agents that block LP activation such as narsoplimab (OMS721), a human monoclonal antibody against MASP2 which recently received Breakthrough Therapy Designation for hematopoietic stem cell transplant-linked thrombotic microangiopathies,\n3939\nor eculizumab, a humanized anti-C5 monoclonal antibody FDA-approved for aHUS, might be considered on a case-by-case basis in severe COVID-19. Investigating critical biomarkers consistent with complement-mediated microvascular injury and thrombosis in COVID-19 patients would be important. A panel to study might include: d-dimers; factor VIII, fibrinogen, and other coagulation factors; antiphospholipid antibodies; C-reactive protein; pro-inflammatory cytokines, particularly IL-1 and IL-6; circulating complement proteins including C3, C4, C5b-9, and Bb (the latter remaining in circulation longer than other components, as a marker of AP activation); and tissue biopsy, the skin being most readily accessible. This could enable establishment of criteria for clinical trials with anti-complement and/or anticoagulants, and perhaps open the possibility for earlier intervention than at the end-stages of severe COVID-19.",
         "The terminal complement complex C5b-9, C4d, and MASP2 were found in the lungs.",
         null,
         null,
         null
        ],
        [
         "13",
         "How does MediBoost differ from traditional decision tree learning algorithms?",
         "Gilmer Valdes et al.,2016, DOI:https://doi.org/10.1038/srep37854:\n Traditional decision trees perform recursive partitioning in order to arrive at a prediction. At each node of the tree, the observed data are further subdivided so that as one goes farther down the tree, each branch has fewer and fewer observations, as illustrated in Fig. 1. This strongly limits the possible depth of the tree as the number of available observations typically shrinks exponentially with tree depth. In this ‘greedy search’ over data par - titions, assigning an observation on the first few nodes of the tree to incorrect branches can greatly reduce the accuracy of the resulting model 24. MediBoost trees are constructed using a different mathematical framework, called boosting, in which each node focuses on observations that previous nodes have not separated correctly21-23. Additionally, in order to obtain a smaller tree, which is a key issue in maintaining interpretability, MediBoost penalizes the weights of observations assigned to different branches through the novel introduction of a membership function, forming a relative “soft” recursive partition similar to decision trees grown using fuzzy logic25. In MediBoost, no hard partitioning is performed, and all observations contribute to all decision nodes. The specialized reader will identify that each path through a MediBoost tree represents a different ensemble, similar to those generated by AdaBoost or gradient boosting, as illustrated in Fig. S1 21-23. This is fundamentally different from previous decision tree learning algorithms29 and is the primary reason for the improved accuracy of MediBoost with respect to current decision tree algorithms. We conclude that MediBoost in its various forms is significantly better than standard decision tree induction algorithms and has comparable accuracy to ensemble methods, based on the two way sign-to-sign, three-way ANOV A, Friedman and Wilcoxon tests shown above. In two of the statistical tests for both balanced cross-validation error and AUC, our decision tree algorithm, in its current form was inferior to random forests. This is consistent with the observations of Caruana et al. 30, who showed that boosted stumps, the structure currently used in MediBoost nodes, are inferior to random forests. Although we present MediBoost algorithms with only two branches per node in this paper (stumps) for simplicity, it could easily be extended to multi-branch nodes, where each node will represent a tree similar to those used in boosted trees with the corresponding improvement in accuracy as shown by Caruana et al. 30. 30. When only stumps are used, MediBoost only takes into account additive effects but random forests is taking into account both additive and interaction effects. If multi-branch nodes are used, however, interaction effects will be taken in to account by MediBoost. In this case, it is expected that MediBoost will be on average equally or more accurate than random forests 30. Additionally, the magnitude of the difference was bigger than 0.03 in only 4 problems out of 13 which might indicate that MediBoost might still be the prefer option in most cases, Table S5. Moreover, MediBoost has been generalized to any loss functions, it can also be easily extended to regression, multi-class or survival analysis. This is one of the advantages over other methods like Bayesian Rule Lists, though MediBoost rules could be larger and more complex in this case 31. Finally, healthcare providers, patients, and biomedical researchers should not be discouraged by the mathematical complexity of the underlying our method-while the mathematical framework of MediBoost is complex, its output, a single tree for any given problem, can be understood with little mathematical knowledge. In fact, MediBoost produces decision trees that can immediately replace those used in current clinical practice/research, a sub-sample of which are referenced in this paper. If MAB and LMB are applied to these previously published medical problems, we predict that more accurate decision trees will be obtained in the majority of problems, with a corresponding positive impact on clinical practice/research. MediBoost thus gives the best of both worlds: it grows a single, highly interpretable tree that has the high accuracy of ensemble methods.",
         "MediBoost focuses on observations that previous nodes have not separated correctly.",
         "5.0",
         "3.0",
         "5.0"
        ],
        [
         "14",
         "In thyroid cancer cells, what effect does upregulating MAPK8IP1P2 have on the activity of caspase-3 or -9?",
         "Xiaoli Liu et al.,2021, DOI:https://doi.org/10.3389/fonc.2020.600927:\n MAPK8IP1P2 Is Downregulated in Thyroid Cancer With Lymph Node Metastasis By analyzing RNA sequencing dataset of thyroid cancer from The Cancer Genome Atlas (TCGA), we found that MAPK8IP1P2 was marked downregulated in thyroid cancer tissues compared with that in the adjacent normal tissues (ANT) (Figure 1A).Consistently, MAPK8IP1P2 expression was reduced in our 24 paired thyroid cancer tissues compared with the matched ANT (Figure 1B). Interestingly, we found that downexpression of MAPK8IP1P2 occurred in 4/9 (44.4%) thyroid cancer tissues without lymph node metastasis, and was 10/15 (66.7%) in thyroid cancer tissues with lymph node metastasis (Figure 1B). Our results further Liu et al. MAPK8IP1P2 Inhibits Metastasis of Thyroid Cancer Frontiers in Oncology | www.frontiersin. MAPK8IP1P2 Inhibits Metastasis of Thyroid Cancer Frontiers in Oncology | www.frontiersin.org January 2021 | Volume 10 | Article 6009274  indicated that MAPK8IP1P2 expression in thyroid cancer tissues without lymph node metastasis had no significant difference compared with that in ANT ( Figures 1C, D ), but was dramatically and significantly downregulated in thyroid cancer tissues with lymph node metastasis (Figure 1C), even in lymph node metastatic thyroid cancer tissues with T1-T2 (Figure 1D). However, there was no significant difference of MAPK8IP1P2 expression between in T1-2 thyroid cancer tissues and in T3-4 thyroid cancer tissues, although MAPK8IP1P2 was downregulated in both compared with that in ANT (Figure 1E). Thesefindings suggested that downexpression of MAPK8IP1P2 may play an important role in lymphatic metastasis of thyroid cancer. TCGA analysis further supported this finding that MAPK8IP1P2 expression was reduced in thyroid cancer tissues compared with that in ANT, especially in thyroid cancer tissues with lymph node metastasis (Figures 1F-H). Therefore, our results combined with TCGA analysis suggest that downexpression of MAPK8IP1P2 may be implicated in lymphatic metastasis of thyroid cancer. Upregulating MAPK8IP1P2 Inhibits Lymphatic MetastasisIn Vivo Then, the effect of MAPK8IP1P2 on lymphatic metastasis of thyroid cancer cells in vivo was further investigated using the inguinal lymph node metastasis model. First, the expression levels of MAPK8IP1P2 in 7 thyroid cancer cell lines and a normal thyroid follicular epithelial cell line PTFE were measured. As shown in Figure 2A ,M A P K 8 I P 1 P 2w a s differentially downregulated in thyroid cancer cells compared A B DE FG H C FIGURE 1 | MAPK8IP1P2 is downregulated in thyroid cancer with lymphatic metastasis.(A) MAPK8IP1P2 expression in 59 paired thyroid cancer tissues and the matched adjacent normal tissues in the thyroid cancer dataset from TCGA.(B) Real-time PCR analysis of MAPK8IP1P2 expression in our 24 paired thyroid cancer tissues and their matched adjacent normal tissues, including 9 thyroid cancer tissues without lymphatic metastasis and 15 thyroid cancer tissues with lymphatic metastasis. The number on the abscissa indicated the patient number according to our record when collecting patient information. GAPDH was used as endogenous controls. *P < 0.05.(C) Real-time PCR analysis of MAPK8IP1P2 expression in ANT (n = 24), thyroid cancer tissues without lymphatic metastasis (n = 10), and thyroid cancer tissues with lymphatic metastasis (n = 38). GAPDH was used as endogenous controls. n.s. indicates no significance. GAPDH was used as endogenous controls. n.s. indicates no significance. (D) Real-time PCR analysis of MAPK8IP1P2 expression in ANT (n = 24), thyroid cancer tissues of T1-T2 grade without lymphatic metastasis (n = 10), and thyroid cancer tissues of T1T2 grade with lymphatic metastasis (n = 27). GAPDH was used as endogenous controls. n.s. indicates no significance. (E) Real-time PCR analysis of MAPK8IP1P2 expression in ANT (n = 24), thyroid cancer tissues with T1-T2 grade (n = 37), and thyroid cancer tissues with T3-T4 grade (n = 11). GAPDH was used as endogenous controls. n.s. indicates no significance. (F) MAPK8IP1P2 expression in ANT (n = 59), thyroid cancer tissues without lymphatic metastasis (n = 230), and thyroid cancer tissues with lymphatic metastasis (n = 225) in the thyroid cancer dataset from TCGA.(G) MAPK8IP1P2 expression in ANT (n = 59), thyroid cancer tissues with T1-T2 grade (n = 309), and thyroid cancer tissues with T3-T4 grade (n = 194) in the thyroid cancer dataset from TCGA.(H) MAPK8IP1P2 expression in ANT (n = 59), thyroid cancer tissues of T1-T2 grade without lymphatic metastasis (n = 164), and thyroid cancer tissues of T1-T2 grade with lymphatic metastasis (n = 110) in the thyroid cancer dataset from TCGA. Liu et al. MAPK8IP1P2 Inhibits Metastasis of Thyroid Cancer Frontiers in Oncology | www.frontiersin.org January 2021 | Volume 10 | Article 6009275  with that in PTFE cells. We further constructed MAPK8IP1P2stably overexpressing B-CPAP and K1 cells and endogenously knocked down MAPK8IP1P2 expression in B-CPAP and K1 cells, both of which expressed moderate levels of MAPK8IP1P2 compared with that in other thyroid cancer cell lines (Figure 2B). Then, Vector-, MAPK8IP1P2-overexpressing, scramble and MAPK8IP1P2-downexpressing K1 cells were injected into the surrounding tissues in the foot pads of the mice (n = 6/group) using the inguinal lymph node metastasis model (Figure 2C). The metastatic inguinal lymph nodes were excised after 4 weeks and analyzed by H & E staining. As show inFigures 2D, E, histological examination of the lymph node metastatic tumors revealed that the tumors in lymph nodes formed from MAPK8IP1P2-overexpressing c ells exhibited reduce tumor burden and the decreased number of tumor cells compared with those injected with the vector-control cells. Conversely, the tumors in lymph nodes formed from MAPK8IP1P2downexpressing cells had larger tumor burden and more number of tumor cells than those inoculated with the scramble cells (Figures 2D, E). These findings indicate that upregulating MAPK8IP1P2 inhibits lymphatic metastasisin vivo. These findings indicate that upregulating MAPK8IP1P2 inhibits lymphatic metastasisin vivo. Upregulating MAPK8IP1P2 Improves Anoikis Resistance in Thyroid Cancer Cells The biological function of MAPK8IP1P2 in lymphatic metastasis of thyroid cancer was further CCK-8 assay showed that either upregulating or downregulating MAPK8IP1P2 had no significant effect on the cell growth of B-CPAP and K1 cells (Figures 3A-D). Similarly, neither colony-formation ability nor cell cycle progression was impeded by the changed expression of MAPK8IP1P2 in thyroid cancer cells (Figures 3E, F). However, upregulating MAPK8IP1P2 inhibited, while silencing MAPK8IP1P2 increased anchorage-independent growth capability of thyroid cancer cells (Figure 3G). These results indicate that the proliferation ability of thyroid cancer cells was not impeded by MAPK8IP1P2in vitro. Notably, upregulating MAPK8IP1P2 represses anchorageindependent growth capability of thyroid cancer cells as demonstrated above. Accumulating studies have shown that the capacity of cancer cells to survive under suspension conditions, namely anoikis resistance, is an important characteristic contributing to tumor progression and metastasis , including thyroid cancer . Therefore, the effect of MAPK8IP1P2 on anoikis resistance in thyroid cancer cells was further evaluated. As shown in Figure 4A, upregulating MAPK8IP1P2 enhanced, while silencing MAPK8IP1P2 reduced the apoptosis rate of thyroid cancer cells. Mitochondrial potential assay showed that upregulating MAPK8IP1P2 attenuated, while silencing MAPK8IP1P2 elevated the mitochondrial potential of thyroid cancer cells (Figure 4B). The results of caspase activity assay and western blot analysis revealed that upregulating MAPK8IP1P2 increased the activity of caspase-3 or -9 and expression of pro-apoptotic proteins BAD and BAX, but reduced expression of anti-apoptotic proteins BCL2 and BCL2L1 ( Figures 4C -E); conversely, A B D E C FIGURE 2 | Upregulating MAPK8IP1P2 inhibits cancer stem cell characteristics in thyroid cancer cells.(A) Real-time PCR analysis of MAPK8IP1P2 expression in 7 thyroid cancer cells, including 4 PTC cell lines, B-CPAP, BHT101, KTC-1, and K1, and 2 ATC cell lines, CAL-62 and 8305C, and 1 thyroid duct cell carcinoma cells, TT, and a normal thyroid follicular epithelial cell line PTFE. GAPDH was used as endogenous controls.*P < 0.05.(B) MAPK8IP1P2 expression in the vector, MAPK8IP1P2 overexpression scramble, MAPK8IP1P2 shRNA#1, and MAPK8IP1P2 shRNA#2 thyroid cancer cells using real-time PCR. Transcript levels were normalized by GAPDH expression.*P < 0.05. Transcript levels were normalized by GAPDH expression.*P < 0.05.(C) Schematic model of lymphatic metastasis modelin vivo. (D) H & E staining analysis of tumors in lymph node from the indicated mice group.(E) The count of tumor cells in the tumor areas of lymph node from the indicated mice group.*P < 0.05. Liu et al. MAPK8IP1P2 Inhibits Metastasis of Thyroid Cancer Frontiers in Oncology | www.frontiersin.org January 2021 | Volume 10 | Article 6009276  silencing MAPK8IP1P2 yielded the opposite effect in thyroid cancer cells (Figures 4C-E). Taken together, our results indicate that upregulating MAPK8IP1P2 abrogates anoikis resistance in thyroid cancer cells. MAPK8IP1P2 Activates Hippo Signaling Pathway in Thyroid Cancer Cells To determine the underlying mechanism implicated in antilymphatic metastatic role of MAPK8IP1P2 in thyroid cancer, Gene Set Enrichment Analysis (GSEA) was performed based on MAPK8IP1P2 expression in the thyroid cancer dataset from TCGA. As shown inFigure 5A, MAPK8IP1P2 overexpression was positively correlated with activity of Hippo signaling pathway, but negatively associa ted with the transcriptional activity of downstream co-activators YAP1/TAZ of Hippo signaling. Inactivation of Hippo signaling has been widely reported to be implicated in anoikis resistance and metastatic thyroid cancer , as well as in lymphatic metastasis process of cancers , suggesting that Hippo signaling may mediate the functional role of MAPK8IP1P2 in lymphatic metastasis of thyroid cancer. Luciferase reporter assay showed that upregulating MAPK8IP1P 2 reduced, while silencing MAPK8IP1P2 increased the luci ferase reporter activity of HOP-Flash, but not the HIP-Flash (Figure 5B), suggesting that upregulating MAPK8IP1P2 inhibits the TEAD-dependent luciferase activity in thyroid cancer cells. Furthermore, upregulating MAPK8IP1P2 enhanced the expression of phophorylated MST1/2 (p-MST1/2), phophorylated LATS1 A BD E F G C FIGURE 3 | Upregulating MAPK8IP1P2 does not affect proliferation of thyroid cancer cells.(A-D) The effect of overexpression or silencing MAPK8IP1P2 on the cell growth in the indicated thyroid cancer cells by CCK-8 assay.(E) The effect of overexpression or silencing MAPK8IP1P2 on colony-formation ability of the indicated thyroid cancer cells by colony-formation assay.(F) The effect of overexpression or silencing MAPK8IP1P2 on cell cycle progression of the indicated thyroid cancer cells byflow cytometry.(G) The effect of overexpression or silencing MAPK8IP1P2 on survival ability in the indicated thyroid cancer cells by anchorage-independent growth assay. *P < 0.05. Liu et al. MAPK8IP1P2 Inhibits Metastasis of Thyroid Cancer Frontiers in Oncology | www.frontiersin. MAPK8IP1P2 Inhibits Metastasis of Thyroid Cancer Frontiers in Oncology | www.frontiersin.org January 2021 | Volume 10 | Article 6009277  (p-LATS1) and phophorylated YAP1 (p-YAP1), reduced the nuclear translocation of YAP1 and TAZ, but had no effect on total level of MST1 and LATS1 in thyroid cancer cells (Figure 5C). In contrast, silencing MAPK8IP1P2 reduced p-MST1/2, pLATS1, and p-YAP1 expression, and increased the nuclear expression of YAP1 and TAZ ( Figure 5C). Real-time PCR analysis showed that upregulating MAPK8IP1P2 decreased, while silencing MAPK8IP1P2 increased the expression levels of multiple downstream genes of Hippo pathway, including CTGF, CYR61, HOXA1, PPIA, RPL13A, and SOX9 , in thyroid cancer cells (Figure 5D). Therefore, thesefindings indicate that MAPK8IP1P2 activates Hippo signaling in thyroid cancer cells. MAPK8IP1P2 Activates Hippo Signaling by Sponging miR-146b-3p Accumulating studies have reported that lncRNAs can serve as competitive endogenous RNAs (ceRNAs) to de-repress miRNAtargeted mRNA expression . Therefore, we further explored the potential binding miRNAs of MAPK8IP1P2 by analyzing the correlation of MAPK8IP1P2 with all reported miRNAs in the thyroid cancer dataset from TCGA. As shown in Figure 6A , the only miR-146b-3p expression level was negatively correlated with MAPK8IP1P2 expression, and was upregulated in thyroid cancer tissues compared with that in ANT. Using several publicly available algorithms, including miRanda and targetscan, we found that miR-146b-3p had the potential recognition sequences on MAPK8IP1P2, and NF2, RASSF1, and RASSF5 were the potential targets of miR146b-3p (Figure 6B). NF2, RASSF1, and RASSF5 have been reported to promote activity of Hippo signaling by varying mechanism. Luciferase assay demonstrated that miR146b-3p mimics suppressed the 3 ’UTR reporter activity of MAPK8IP1P2, NF2, RASSF1, and RASSF5, but not of the mutant 3’UTR of MAPK8IP1P2 (Figures 6C, D). RT-PCR and Western blot analysis revealed that upregulating MAPK8IP1P2 increased, while silencing MAPK8IP1P2 decreased the mRNA and protein levels of NF2, RASSF1, and RASSF",
         "Upregulating MAPK8IP1P2 increased the activity of caspase-3 or -9.",
         "5.0",
         "3.0",
         "5.0"
        ],
        [
         "15",
         "What percentage of errors in medical laboratories are attributed to the preanalytical phase?",
         "Nadin Younes et al.,Not identified, DOI:https://doi.org/10.3390/v12060582:\n 3. Implications and Challenges of Current COVID-19 Diagnostic Tests\n3.1. Preanalytical and Analytical Errors\nAlthough medical diagnostic errors can happen almost always and everywhere [\n17\n], the fragility of diagnostic laboratories is significantly magnified when the healthcare workers were required to face high workload and work in high-throughput settings due to the increasing number of cases [\n18\n]. Although the consequences of laboratory errors are often substantial [\n1918\n]. Although the consequences of laboratory errors are often substantial [\n19\n], the consequences in the current SARS-CoV-2 pandemic are certainly amplified. Unfortunately, false-positive and false-negative results do not only possess a threat to the health of the individual, but may also disrupt the efficiency of emergency plans, public health policies, and preventive measures applied for containing the pandemic. A false-positive test result not only leads to unrequired treatment but may cause societal problems as it may undermine the workforce available for facing this pandemic if attributed to people working in public facilities. Nevertheless, a false-negative test result may potentially contribute to further spread of the SARS-CoV-2 virus within the community. Therefore, accurate and precise laboratory technologies play a vital role in diagnosing and managing the current SARS-CoV-2 outbreak [\n2020\n]. However, there are a number of potential preanalytical and analytical errors that must be taken into consideration by clinicians, clinical microbiology laboratories, and public health authorities to avoid false test results.\nThere is undeniable evidence that the preanalytical phase is the main source of errors in medical laboratories [\n21\n,\n22\n], accounting for approximately 46% to 68.2% of errors observed during the whole testing process [\n23\n], despite continuous improvements in pre-analytical automation. It is estimated that more than one-fourth of all pre-analytical errors result in an unnecessary investigation or inappropriate patient care, substantially magnifying the financial burden on the healthcare system [\n2424\n], and thus resulting in inadequate and slow healthcare. The safety and quality of diagnostic testing may be endangered by misidentification of the patient and/or sample, collection of an inappropriate or insufficient sample, inaccurate conditions of sample transportation and storage (e.g., prolonged transportation time and injury exposure), presence of interfering substances (e.g., cellular components due to whole blood freezing and inappropriate additives) [\n25\n,\n26\n,\n27\n], and finally, procedural issues occurring during sample preparation, including pipetting errors during manual sample preparation or aliquoting, cross-contamination and sample mismatch [\n2828\n]. Although analytical errors are believed to be the smallest contributors to laboratory errors, there are several potential analytical problems that could significantly jeopardize the quality of testing, and thus need to be considered. Analytical errors include equipment malfunction, non-adequately validated assays, undetected failure of quality control, active viral recombination, testing carried outside the diagnostic window, poor harmonization of primers or probes, and non-specific rRT-PCR annealing, along with other technical issues [\n25\n,\n26\n,\n27\n].\n3.2. Chest Computerized Tomography (CT)25\n,\n26\n,\n27\n].\n3.2. Chest Computerized Tomography (CT)\nChest computerized tomography (CT) is a conventional, non-invasive imaging technology with high accuracy and speed. The sensitivity to detect SARS-CoV-2 using chest CT is reported to be higher than that of real-time reverse transcription-polymerase chain reaction (rRT-PCR). Recent evidence has shown that asymptomatic patients with COVID-19 may show paradigmatic CT changes very early and even before being positive with rRT-PCR [\n29\n,\n30\n,\n31\n]. For instance, a case was reported in Wuhan city with a history of chills and fever of unknown cause and tested negative four times for SARS-CoV-2 with rRT-PCR from the disease onset [\n32\n]. Thus, the clinical physician could not diagnose the patient with COVID-19 at an early stage because of the false negative rRT-PCR results [\n3232\n]. Therefore, according to Feng et al., patients showing symptoms of fever, dry cough, fatigue, or dyspnea along with recent exposure to SARS-CoV-2 infected patients should be diagnosed with CT despite negative rRT-PCR test results [\n32\n]. These pieces of evidence support the advice that the most efficient approach for diagnosing suspected patients with COVID-19 in suspected patients shall encompass a combination of rRT-PCR with clinical and epidemiologic evidence (such as the probability of exposure with infected patient, signs, and symptoms) and chest CT findings.\n3.3. Nucleic Acid Amplification Testing (NAAT)Rapid and accurate detection of positive COVID-19 cases is crucial to control the viral outbreak in the community and health care facilities. In general, studies have shown that molecular technologies are more accurate than CT scans and serological tests for the definitive diagnosis of COVID-19, as they can target and identify the specific antigen of SARS-CoV-2. The development of molecular diagnostic technologies against SARS-CoV-2 is dependent upon the understanding of the proteomic and genomic composition of the virus and the viral induction of changes in proteins and genes expression in the patient during and after infection. According to the World Health Organization (WHO), 104 strains of SARS-CoV-2 virus were isolated and sequenced using Illumina and Oxford nanopore sequencing by the 15th of February 2020. By the 24th of March, the genomic and proteomic compositions of SARS-CoV-2 had been identified. However, the host response to the virus is still under investigation. However, the host response to the virus is still under investigation. Currently, the NAAT available for SARS-CoV-2 includes rRT-PCR (Laboratory-based) and reverse transcription loop-mediated isothermal amplification (RT-LAMP) (POC) [33\n,\n34\n]. Unfortunately, the currently available diagnostic tests are labor-intensive and time-consuming, and a shortage of commercial kits delays diagnosis.\n3.3.1. Manual Laboratory Based NAAT: Real-Time Reverse Transcription-Polymerase Chain Reaction (rRT-PCR)\nThe current gold standard for the etiological diagnosis of SARS-CoV-2 infection is rRT-PCR on a variety of clinical specimens, including bronchoalveolar lavage fluid, fiber bronchoscope brush biopsies, sputum, nasal swabs, pharyngeal swabs, feces, or blood [\n3535\n]. The rRT-PCR tests offer several benefits. First of all, rRT-PCR is especially valuable at the early stage of infection, when the viral load is lowest and can differentiate it from other similar viruses, due to its sensitivity and specificity, respectively. Thus, as opposed to serology, rRT-PCR provides more valuable information at the initial stages of infection, as it detects the pathogen directly by detecting its RNA when the aim is to prevent infectivity and antibodies have not yet been built. In addition, rRT-PCR results are generally available within a few hours to 2 days. Moreover, it can be easily operated on a large scale.Although rRT-PCR offers many benefits, it has some limitations. Its low sensitivity, low stability, and long processing time were detrimental to the health care efforts to contain the outbreak. Also, several external factors may affect rRT-PCR testing results accuracy, including sampling operations, specimen source (e.g., upper or lower respiratory tract), sampling timing (before and after symptoms onset), and the performance of detection kits. Most importantly, recent evidence has shown that the diagnostic accuracy of many of the available commercial rRT-PCR kits for detecting SARS-CoV-2 may be lower than optimal (i.e., <100%), and there are reports where it has given false negatives in subjects for up to two weeks [\n36\n,\n37\n,\n38\n]. This high incidence of false negative diagnosis was observed specifically in SARS-CoV-2 testing. The largest study on coronaviruses testing to date estimated a rate of 41% false negatives on RT PCR diagnostic tests used in China [\n39\n,\n40\n,\n4139\n,\n40\n,\n41\n]. However, still more research is needed to determine the true prevalence of such false-negative rRT-PCR results; scientists and researchers agree that the problem is significant, which not only impedes the diagnosis of the disease in patients but also risk patients who assume they are uninfected further transmitting the virus in the community. Moreover, using PCR, codetection with other respiratory viruses is frequently encountered in coronaviruses, and the contribution of positive CoV PCR results to disease severity is not always explicitly exhibited [\n4242\n]. Furthermore, rRT-PCR requires professionally trained staff to operate sophisticated laboratory facilities, which are usually located at a central laboratory (biosafety level 2 or above), and is often time-consuming, requiring from few hours up to 2 or 3 days to obtain laboratory results. This often leaves a rapidly rising number of potential cases untested and thus opening a gaping hole in SARS-CoV-2 prevention efforts. Furthermore, this time-consuming process of sample testing is not only extremely disadvantageous but also unsafe since the virus needs to be contained. Finally, the US Food and Drug Administration (FDA) concluded that a negative rRT-PCR test result does not completely rule out SARS-CoV-2 infection and shall not be used as a single element for patient management decisions, and re-testing shall be considered in consultation with public health authorities [\n43\n]. The information summarized in\nTable A1\n(\nAppendix A43\n]. The information summarized in\nTable A1\n(\nAppendix A\n) was extracted from the manufacturer package inserts or their websites.\nProtocols for rRT-PCR testing developed by several countries and entities, including Germany, Hong Kong, China CDC, Thailand, and Japan, have been posted to the WHO’s website [\n44\n], and the protocol for testing in the United States has been posted to CDC’s website [\n45\n].\nTable 2\nis a comparison between the available rRT-PCR protocols.\nTable 2.\nSummary table of available protocols posted to the WHO’s website.\nInstitute\nGene Targets\nAmplicon Size (bp)\nSensitivity\nSpecificity\nConcentration/Volume of Reagents\nDoes the Protocol Recommend Specific Kits?\nChina CDC\nORF1ab gene\nNR\nNR\nNR\nNR\nNR\nN gene\nNR\nNR\nNR\nNR\nNR\nInstitute Pasteur, Paris, France\nRdRp: nCoV_IP2 gene\n108 bp\n95% hit rate for approx. 100 copies of RNA GE.\nLOD for 1x10\n7\nRNA copies is ~21 cycles\nLOD for 1x10\n4\nRNA copies is ~30 cycles\nNo cross reactivityLOD for 1x10\n7\nRNA copies is ~21 cycles\nLOD for 1x10\n4\nRNA copies is ~30 cycles\nNo cross reactivity\nFinal concentration of 0.4 μM of each primer and 0.2 μM of probe\nRNA extraction via NucleoSpin Dx Virus and Invitrogen SuperscriptTM III Platinum\n®\nRdRp: nCoV_IP4 gene\n107 bp\nE gene\n125 bp\nUS CDC, USA\nN1 gene\n71 bp\nLOD: 1x10\n0.5\nRNA copies/μL and 10 RNA copies/ μL for Qiagen EZ1 and Qiagen respectively.\nProbe showed high sequence homology with SARS coronavirus and Bat Sars-like coronavirus\n20 μM primers, 5 μM probe; 15 μL total volume\nFor the RT-qPCR TaqPathTM 1- Step RT-qPCR Master Mix. For extraction, they recommend bioMérieux NucliSens\n®\nsystems, QIAamp\n®\nkits, QIAGEN kits, Roche Kits and Invitrogen kits\nN2 gene\n67 bp\nN3 gene (removed from diagnostic panel 3/15/20)\n72 bp\nNational Institute of Infectious Diseases, Japan\nN gene\nNR\nAverage Cq value of specimen was 36.7 and 35.0 for the positive control (500 copies of RNA transcript)\nNRNR\n1 μL of 20 xprimer and probe mix in a 20 μL reaction with 5 μL of RNA. F primer at 500 nM, R primer at 700 nM, probe at 200 nM.\nRNA extracted using QIAamp viral RNA mini kit (Qiagen). Reverse transcription via Super Script IV Reverse Transcriptase (Thermo). RT-PCR via QuantiTect Probe RT-PCR Kit (Qiagen)\nCharité, Germany\nRdRp gene\nNR\nLOD: 3.8 RNA copies/ reaction, 95% hit rate; 95% CI: 2.7-7.6 RNA copies/reaction\nNo reactivity with other human respiratory viruses\nRdRP: F-600 nM/reaction, R-800 nM/rxn, P-100 nM each/ reaction,\nRNA extracted using MagNA Pure 96 system (Roche), RT- PCR via Superscript III one step RT-PCR system with Platinum Taq Polymerase (Invitrogen).\nE gene\nNR\nLOD: 5.2 RNA copies/reaction, at 95% hit rate; CI: 3.7-9.6 RNA copies/reaction\nE gene: F-400 nM/r reaction, R-400 nM/ reaction, P-200 nM/ reaction\nHKU, Hong Kong SAR\nORF1b-nsp14 gene\n132 bp\nNR\nNo reactivity with respiratory cultured viruses and clinical samples.\n10 μM primers, 10 μM probesNo reactivity with respiratory cultured viruses and clinical samples.\n10 μM primers, 10 μM probes\nQIAamp Viral RNA Mini Kit or equivalent and TaqMan Fast Virus Master mix.\nN gene\n110 bp\nNational Institute of Health, Thailand\nN gene\nNR\nPositive control detected at less than 38 cycles.\nNR\n40 μM primers, 10 μM probe\nMacherey-Nagel Nucleospin RNA virus and Invitrogen superscriptTM III Platinum One-Step Quantitative\nOpen in a new tab\nNR: not reported; GE: genome equivalent; LOD: Limit of detection.\n3.3.2. Rapid and Point of Care NAAT: Reverse Transcription-Loop-Mediated Isothermal Amplification\nTransforming the molecular diagnostic testing for SARS-CoV-2 from laboratory settings to point of care (POC) is potentially important to increase the quantity of testing that can be conducted [\n39\n,\n4239\n,\n42\n], potentially reducing the time to obtain an actionable result, and thus supporting earlier identification of positive cases. Most importantly, POC testing will support the suitable use of quarantine resources, infection control measures, and patient recruitment into clinical trials of treatments. Most of the available molecular POC tests have either gained Conformité Européenne (CE) marking or emergency Food and Drug Administration (FDA) approval [\n46\n]. Molecular POC testing utilizes the same basic technology as the laboratory-based assays, but with automating various number of the steps. Therefore, molecular POC tests could be operated in near-patient settings rather than on the laboratory bench, which is expected to reduce the turnaround time and rapidly provides the result. Some of the molecular POC tests utilize isothermal nucleic acid amplification techniques, such as MicrosensDx RapiPrep\n©©\nCOVID-19 and Abbott ID NOW COVID-19, while others utilize PCR technology, such as Cepheid Xpert SARS-CoV-2, Credo VitaPC R COVID-19 assay, GenMark ePlex SARS-CoV-2, MesaBioTech Accula SARS-CoV-2, which utilizes lateral flow technology, and the very recent Spartan Cube CYP2C19 System (Canada) [\n46\n].\nLoop-mediated isothermal amplification (LAMP) was developed as a rapid, accurate, reliable, and cheaper technique to amplify the target sequence at a single reaction temperature instead of sophisticated thermal cycling equipment needed in rRT-PCR [\n4747\n]. The advantage of using LAMP is that the amount of DNA produced is much higher than in rRT-PCR and a positive test result can be seen visually without requiring a machine to read the results. In addition, it is simple, cheap, and rapid. Several studies evaluated the use of a novel RT-LAMP method against the gold standard rRT-PCR. Two studies showed evidence that RT-LAMP methods demonstrated more than 97% sensitivity targeting the ORF1ab gene compared to rRT-PCR [\n48\n,\n49\n]. Yang et al. showed that RT-LAMP and rRT-PCR have the same sensitivity and both can detect a 20-fold diluted sample [\n50\n]. Additionally, according to Yang et al., the detection limit of LAMP is 1000 copies/mL, which is equal to the rRT-PCR kits [\n50\n]. Most importantly, studies have shown that RT-LAMP analysis is extremely specific because it uses six to eight primers to identify eight different regions on the target DNA [\n50\n,\n5150\n,\n51\n]. However, unlike rRT-PCR, LAMP technology does not have such a large background of literature behind it. Thus, tests using LAMP technology for COVID-19 are still being assessed in clinical settings.\nAlmost all molecular POC described devices are portable benchtop-sized analyzers, except the MesaBioTech Accula and MicrosensDx RapiPrep\n©\nCOVID-19 tests, which are smaller, handheld devices. A variety of clinical sample types may be used, including oral, throat, nasal, or nasopharyngeal swabs. All tests require a similar sample preparation procedure that involves placing the swab sample into the viral transport media and pipetting the sample into a single-use disposable cartridge—this sample preparation step takes approximately 2–10 min [\n46\n]. The time to result varies from 13 min in Abbott Diagnostics ID NOW COVID-19 to 45 min in Cepheid Xpert SARS-CoV-2 [\n46\n]. The information and validation of each device are summarized in\nTable A1\n(\nAppendix A\n).46\n]. The information and validation of each device are summarized in\nTable A1\n(\nAppendix A\n).\n3.4. Serological Testing for COVID-19 Diagnosis3.4. Serological Testing for COVID-19 Diagnosis\nrRT-PCR–based assays performed on respiratory specimens remain the gold standard for COVID-19 diagnostics, as mentioned previously. However, point-of-care technologies and serologic immunoassays are rapidly emerging, with high serological tests for SARS-CoV-2 being at increased demand for better quantification of SARS-CoV-2-positive cases, including asymptomatic and recovered cases. Serological tests are blood-based tests that measure antibodies or antigens present in the blood when the body is responding to a particular infection. Thus, it could identify previous exposure to a particular pathogen as well as the production of the body’s immune system-specific antibodies. Two types of serology test, in particular, are becoming more widely available, namely laboratory-based enzyme immunoassays (EIA) on high throughput automated platforms and rapid, point of care (POC) tests, which are similar to a blood glucose test or home pregnancy test.Serological tests offer a number of advantages compared to rRT-PCR. First of all, serological testing can provide further details by identifying individuals who have developed virus-specific antibodies, and thus can detect past infection and give better information regarding the disease prevalence in a population. Unlike viral RNA, virus-specific antibodies stay in the blood for several weeks to months after symptom onset. According to the FDA, IgM antibodies to SARS-CoV-2 are detectable in the blood just a few days after initial infection [\n52\n]. However, IgM levels throughout the course of infection are not well characterized. IgG becomes detectable three days from symptom onset or at least 7–10 days after infection [\n1616\n]. It worth mentioning that when the result is negative for COVID-19, the patient was probably not infected at the time of sample collection. However, that does not mean that he will not get sick. In addition, the detection of SARS-CoV-2 antibodies does not guarantee the protection against COVID-19 infection, as many types of anti-SARS-CoV-2 are not neutralizing antibodies [\n53\n]. Considering the fact that 20%–80% of SARS-CoV-2-positive cases are estimated to be asymptomatic, serological tests are especially beneficial because of their scalability, which allows their use on a large scale to assess the overall immune response in a population [\n5454\n]. In addition, human antibodies are known to be more stable compared to viral RNA, and thus serological samples are less prone to deterioration during sample collection, preparation, transport, storage, and testing compared to rRT-PCR samples. Moreover, serological samples have less variations compared to nasopharyngeal specimens because antibodies are usually homogeneously dispersed in the blood. Furthermore, serological samples can be collected easily with minimal discomfort to the patient during phlebotomy. On the other hand, serological tests have some disadvantages, mainly involving the slow antibody response to SARS-CoV-2 virus, as they may not be detectable until three days from symptom onset or at least 7–10 days after infection (\nFigure 2\n) [\n16Figure 2\n) [\n16\n]. In addition, these tests are not designed to detect individuals in the early stages of COVID-19 infection. For instance, less than 40% of infected individuals are seropositive (IgM/IgA) in the first seven days, making it unreliable for the detection of acutely infected individuals. Importantly, there have been reports that those with mild cases of COVID-19 infection do not produce antibodies. It was proposed that their innate immune system (cell-mediated immunity) wiped out the virus before the adaptive immune system (antibodies) had to produce antibodies [\n55\n]. Since serological tests alone may not be enough to diagnose SARS-CoV-2, combining both serological and molecular techniques would give a valuable diagnostic result.\nFigure 2.\nOpen in a new tab\nEstimation of biomarker levels during the COVID-19/SARS-CoV-2 infection.\n3.4.1. Manual ELISAEstimation of biomarker levels during the COVID-19/SARS-CoV-2 infection.\n3.4.1. Manual ELISA\nA variety of CE-marked manual enzyme-linked immunosorbent assays (ELISA) have been developed for the rapid detection of neutralizing antibodies (IgM, IgG, and IgA) against the novel coronavirus by many IVD companies such as Euroimmun, Epitope Diagnostics, DRG Diagnostics GmbH, IBL International, Creative Diagnostics and others (\nTable A2\n,\nAppendix A\n). There are also some commercially available manual ELISA kits for detecting SARS-CoV-2 viral antigens (SP and NP); however, these are mainly used in research and not for clinical diagnosis [\n5656\n]. Manual ELISA provides accurate and valuable information regarding the immune response to the virus; however, unlike rRT-PCR, it cannot be used for screening or diagnosis of early infection, since specific IgM and IgG antibodies are not detectable at this phase. IgM antibody response occurs earlier than that of IgG, with positive IgM antibodies in 70% of symptomatic patients after 8–14 days and about 90% of total antibodies test positive within 11–24 days [\n57\n]. On the other hand, IgG antibodies can be detected around 20 days after viral infection and they persist for a long time [\n58\n]. The reactivity of IgG is assumed to reach more than 98% after several weeks, but the extent of this antibody response is yet to be determined [\n5959\n]. According to recent reports of the WHO, only 2% or 3% of infected COVID-19 individuals appear to have antibodies in their blood. “There is simply not enough data yet to determine if protective immunity is achieved after infection,” says Jennifer Rychert, the medical director of microbial immunology at ARUP Laboratories.\nAnother challenge of using manual ELISA for SARS-CoV-2 detection is that IgM antibodies are notoriously non-specific, and given the time it takes for the development of specific IgG antibodies, serology testing will not likely play an active role in the detection of early cases (\nFigure 1\n) except for diagnosis/confirming late cases or to determine the immunity of healthcare personnel as the outbreak progresses [\n6060\n]. Furthermore, manual ELISA kits are subject to many interferences, including a specific binding and cross-reaction with other coronaviruses, such as MERS-CoV, SARS-CoV-1, and endemic coronavirus. This depends on the type of antigen used to coat the plates. For instance, an ELISA method based on bat SARSr-CoV Rp3 N protein was successfully developed to detect IgM and IgG antibodies against SARS-CoV-2 in early cases of COVID-19 [\n59\n]. A caveat in this ELISA method is that it may produce false positive results since nucleocapsid protein (NP) is the most conserved viral protein among human betacoronaviruses [\n61\n]. Hence, antigens used in this ELISA may react with antibodies against other types of coronavirus (HKU1, 229E, OC43, NL63) that are known to cause the common cold [\n6262\n]. On the other hand, spike protein (SP) is the most diverse protein and several companies have focused on developing ELISA methods for detecting serum antibodies against two domains in the S protein (S1 and S2). The coronavirus envelope spike is responsible for viral entry and it determines the host tropism and virus transmission, which makes it a good candidate for ELISA development [\n60\n]. Still, the evaluation of the clinical performance of manual ELISA kits is imperative before using them for COVID-19 diagnosis.\nAlthough many challenges exist, serology testing using ELISA offers great benefits as a therapeutic option to control the current pandemic and possible re-emergence of coronavirus in the future. Hence, the development of manual ELISA kits remains a high priority, as they can complement the existing testing of SARS-CoV-2 by rRT-PCR (the gold standard) and overcome some of its limitations [\n63\n].\n3.4.2. Automated SerologyThe increased demand to perform diagnostic tests on the population imposes a huge clinical and financial burden on diagnostic laboratories. The implementation of automated serological testing has increased the quality assurance and lowered the turn-around-time (TAT) as well as false positive and negative results. Automated techniques are currently adopted for the most commonly used serological methods. Regular serology tests, which are more amenable to automation, are best deployed in the laboratory setting where they can be used to identify immune individuals and for population-level seroprevalence studies. These will be most useful later in the outbreak when the prevalence of the disease increases. In fact, the healthcare market has been flushed with SARS-CoV-2 laboratory testing platforms just a few months into the COVID-19 pandemic. The laboratory-based EIA automated platforms offer high efficiency, high throughput, and improved quality of the results. However, this expansion of newly developed platforms makes it challenging to critically evaluate SARS-CoV-2 laboratory automated tests. Most of the available SARS-CoV-2 manual ELISA kits use the standard 96-microplate as a solid phase and also the standard spectrophotometry/colorimetric method for signal detection, while in the automated EIA assay, the solid phase materials are different, such as polystyrene (PS-COOH) or metal-based nanoparticles (magnetic nanobeads). Further, more sensitive detection systems such as chemiluminescence technology are usually sued in the automated assays.In April 2020, a fully automated serology test was launched by DiaSorin (Saluggia,\nItaly\n) to detect SARS-CoV-2 antibodies [\n64\n]. The test was developed to detect SARS-CoV-2 IgG antibodies against both the S1 and S2 domain of the spike protein. This increases the specificity of the test and prevents cross-reaction and false-positive results due to other coronaviruses. The LIAISON\n®\nXL platform is a chemiluminescence analyzer that is used to perform a fully automated diagnostic tests process with a minimum level of laboratory personnel intervention. The system could perform up to 170 samples per hour to fulfill the need for large population screening for SARS-CoV-2 and identify infected individuals. By the end of April, the DiaSorin test obtained the FDA emergency use authorization (EUA). Similarly, Bio-Rad (Hercules, CA, USA) has developed a test that is blood-based EIA to detect antibodies against SARS-CoV-2 virus [\n2\n]. The test could be",
         "Approximately 46% to 68.2% of errors observed during the whole testing process occur in the preanalytical phase.",
         null,
         null,
         null
        ],
        [
         "16",
         "What is the typical turnaround time for rRT-PCR tests for SARS-CoV-2?",
         "Nadin Younes et al.,Not identified, DOI:https://doi.org/10.3390/v12060582:\n 2. The Roles of Diagnostic Testing in the SARS-CoV-2 Pandemic\nThe primary goal of the epidemic containment of COVID-19 is to reduce the infection transmission in the population by reducing the number of susceptible persons or by reducing the basic reproductive number (R0). The R0 is modulated by several factors, including the duration of viral shedding, the infectiousness of the organism, and the contact matrix between infected and susceptible persons [\n8\n]. Due to the lack of effective vaccines or treatments, the only available method to reduce SARS-CoV-2 transmission as much as possible is by identifying and isolating infected patients who are contagious and can transmit the diseases. Unfortunately, the rapid spread of COVID-19 outbreak across the globe has exposed the major gaps and vulnerabilities in the abilities of healthcare systems of most countries to successfully contain the outbreak.The deployment of COVID-19 diagnostic testing has varied widely across the globe. A few countries in Asia showed the power of investment in pandemic preparedness, flexible isolation systems, and intensive case finding in the epidemic containment. For example, in South Korea, they dramatically hindered the COVID-19 outbreak by establishing an unprecedented national testing effort [\n9\n] as they successfully managed to perform more than 300,000 tests in the first 9 weeks after identifying the first case of COVID-19 [\n9\n]. Similarly, in Singapore, they implemented different protective measures including a broader case definition, aggressive contact tracing, and strict patient isolation [\n10\n]. Most importantly, to identify asymptomatic patients who did not meet the case definition, a Singapore-wide screening program on patients with pneumonia, influenza-like illnesses, severely ill patients in ICU, and deaths with a possible infectious cause was performed [\n1111\n]. Similar approaches were implemented in Taiwan and Hong Kong [\n12\n]. These countries successfully contained the COVID-19 outbreak by rapidly deploying resource-intensive strategies that prioritized aggressive testing and isolation to interrupt transmission [\n12\n].\nDue to the rapid transmission of SARS-CoV-2, the role of diagnostic testing is dependent on the types of test available, the resources required for testing, and the time to obtain results. In other words, the rapid identification of suspected cases remains a high priority to properly allocate personal protective equipment (PPE) and to prevent nosocomial spread with subsequent community transmission [\n13\n,\n14\n]. Thus, many diagnostic tests for COVID-19 are available so far, with more gaining emergency approval every day. These tests are largely based on four different techniques, as shown in\nTable 1\n.\nTable 1.\nFour main type of technologies used for identification of SARS-CoV-2.\nTechnology\nMolecular TestedFour main type of technologies used for identification of SARS-CoV-2.\nTechnology\nMolecular Tested\nLaboratory or Point of Care\nTime to Results\nTypical Sample Site\nNumber of Samples/Batches\nrRT-PCR\nViral RNA\nLaboratory-based\n3–4 h\nNasopharyngeal swab, sputum\nUp to 96 samples\nLAMP\nViral RNA\nPOC\n2–3 h\nNasopharyngeal swab, sputum\n1–4 samples\nLateral Flow\nAntibody or Antigen\nPOC\n15–20 min\nBlood\n1 patient sample\nELISA\nAntibody or Antigen\nLaboratory-based\n1–3 h\nBlood\nUp to 96 samples\nOpen in a new tab\nThe current diagnosis of COVID-19 infection relies mainly on the centralized laboratory-based rRT-PCR. Although rRT-PCR provides a relatively rapid result (average 3–4 h), it is limited by transportation to the laboratory and the requirement to batch samples in a large run, as shown in\nTable 1Table 1\n. Thus, public health sectors are in deep need for fast and reliable tests for SARS-CoV-2 to be able to effectively contain the pandemic. Cost-effective and efficient diagnostic techniques as near to the POC as possible would be a game-changer in the current situation. Some of the currently available POC diagnostic devices utilize molecular-based techniques, and thus are more suitable for diagnosing new COVID-19 cases, while others utilize serological techniques, and thus are better suited to determining whether an individual has previously been infected, to ascertain their suitability to return to frontline services.\nFigure 1\nsummarizes the diagnostic window for molecular-based techniques and serological testing.\nFigure 1.\nOpen in a new tabFigure 1.\nOpen in a new tab\nRepresentative figure showing the correspondence between the viral load during SARS-CoV-2 infection and the clinical course of the disease. The diagnostic windows of nucleic acid amplification tests (NAAT) and serology test are shown. Testing before and after the NAAT diagnostic window will show a false negative result [\n15\n]. Nevertheless, testing before the serology diagnosing window will show in false negative results [\n16\n].",
         "rRT-PCR tests typically take 3–4 hours.",
         null,
         null,
         null
        ],
        [
         "17",
         "What are some examples of sensory ciliopathies linked to defects in ciliary signaling?",
         "Jeremy F Reiter et al.,2017, DOI:https://doi.org/10.1038/nrm.2017.60:\n Motile ciliopathies\nMotile cilia on different cell types have different waveforms and functions\n20\n. Similar to choanoflagellates, sperm use a specialized cilium (flagellum) to locomote. Other motile cilia do not propel cells but instead move the overlying fluid. Such cilia are found on airway epithelial cells, oviduct cells, ependymal cells that line the brain ventricles, and on the node, which is a transient developmental structure that is crucial for left–right axis determination\n21\n.\nPrimary Ciliary Dyskinesia (PCD)\nImpaired ciliary motility almost exclusively results in a motile ciliopathy (\nBOX 1\n) that is known as primary ciliary dyskinesia (PCD)\n20\n(\nFIG. 3\n). PCD results in situs inversus (a left–right patterning anomaly), chronic bronchitis, sinusitis and atelectasis (attributable to defective clearance of lung mucus) and male infertility (owing to defective sperm locomotion)\n2020\n. Less common manifestations of PCD include decreased female fertility (resulting from improper oocyte transport through the oviducts) and a disposition to headaches and hydrocephalus (presumably resulting from impaired cerebrospinal fluid movement by ependymal cilia).\nPCD is an inherited autosomal recessive disease. It is usually caused by the impaired formation or function of the inner or outer dynein arms, dynein regulatory complex or central pair, all of which are required for ciliary motility\n20\n, and does not normally affect ciliary signalling (\nFIG. 1\n). As in many ciliopathies, PCD displays considerable genetic heterogeneity, and mutations in at least 37 separate loci have been linked to the syndrome (\nFIG. 3\n; see\nSupplementary information S1 (table)\n). Such heterogeneity probably reflects the complexity of the structures required to generate ciliary motility.\nFirst-order and second-order PCD ciliopathiesFirst-order and second-order PCD ciliopathies\nMost forms of PCD are linked to genes that encode ciliary components that are directly required for motility, and thus can be considered first-order ciliopathies (\nBOX 1\n). However, not all PCD-associated genes encode ciliary proteins. For example, mutations in\nDNAAF2, DNAAF3\nand\nDYX1C1\ncause PCD but encode cytoplasmic proteins that are involved in the pre-assembly of axonemal dynein complexes before their import into cilia\n22\n–\n24\n. PCDs that are a result of the loss of these non-ciliary components are examples of second-order ciliopathies (\nBOX 1\n).\nSensory ciliopathies\nSensory ciliopathies result specifically from defects in the sensory and/or signalling functions of cilia, and are primarily caused by defects in non-motile cilia (although either motile or non-motile cilia can be involved) (\nFIG. 3\n).\nDefects in primary cilium structure or signalling cause sensory ciliopathiesFIG. 3\n).\nDefects in primary cilium structure or signalling cause sensory ciliopathies\nAlthough motile cilia also have sensory capabilities, such as sensing noxious chemicals in the respiratory airway\n25\n, the phenotypic presentations of PCD are distinct from those exhibited when nonmotile primary cilia functions are disrupted. Metazoan non-motile primary cilia have evolved different sensory modalities for environmental cues and intercellular cues. Therefore, defects in primary cilia function lead to more varied sensory, physiological and developmental anomalies than do defects in motile cilia (\nFIGS 1\n,\n3\n). Sensory ciliopathies have several possible molecular aetiologies, including impaired cilium formation or maintenance; abrogation of ciliary signal transduction pathway components; or trafficking defects that prevent the signalling machinery from being localized to, or removed from, cilia.\nSensory ciliopathies can impair the perception of environmental cuesSensory ciliopathies can impair the perception of environmental cues\nDifferent cell types make use of different forms of ciliary signalling. The determinants of this ciliary signalling specificity are only beginning to be understood, but are probably determined by the nature of the signal transduction machinery that is expressed by different cell types.\nFor example, in the human retina, photoreceptor cells that are responsible for vision express proteins that are specialized for phototransduction. In the photoreceptor outer segment, which is part of a modified cilium\n2626\n, light interacts with opsin to activate cGMP-specific phosphodiesterase via the G protein transducin; this reduces intracellular levels of cGMP and leads to the closing of ion channels on the cell membrane. Consequent decreases in the intracellular calcium concentration outside the outer segment then promote neurotransmitter release. Therefore, for vision, the cilium is the site of both signal reception and initial transduction, with the subsequent transmission of the information to the cell body being required for interpretation and communication with other cells.\nSimilarly, in olfaction\n27Similarly, in olfaction\n27\n, an odorant is detected by its G protein-coupled receptor (GPCR) on the ciliary membrane. However, instead of being transmitted via cGMP, the activation of the olfactory receptor stimulates an adenylate cyclase (ADCY3) to generate a different second messenger, cAMP. In further contrast to vision, olfaction uses the G protein Golf and opens ion channels to trigger an action potential. Thus, although the machinery is different, sight and smell rely on conceptually similar mechanisms for signal reception and transduction.\nDefects in ciliary signalling through opsins and olfactory receptors are linked to sensory ciliopathies such as retinal degeneration and anosmia (olfaction impairment), respectively\n26\n–\n28\n. Three different molecular aetiologies for a sensory ciliopathy can be observed in retinal degeneration (retinitis pigmentosa (RP)): cilium formation or length control may be impaired\n29\n; an enzyme that lowers ciliary cGMP concentration may be disrupted29\n; an enzyme that lowers ciliary cGMP concentration may be disrupted\n30\n; or mislocalization of opsin or other phototransduction components may occur\n31\n,\n32\n. In these cases, the ciliary defects promote apoptotic cell death through a mechanism that is unknown but may involve the accumulation of opsin in the endoplasmic reticulum and the subsequent activation of the unfolded protein response\n33\n.\nSensory defects can also result from anomalies in the structure or signalling functions of olfactory epithelial cell cilia\n34\n. In olfactory cells, mistrafficking of ciliary proteins does not cause apoptosis but impairs their function, resulting in anosmia. Consistently, anosmia is one hallmark of ciliopathies, such as BBS\n35\nor those caused by transition zone dysfunction\n36\n.\nImpaired ciliary signalling impacts development\nSensory ciliopathies extend beyond defects in interpreting environmental cues. Primary cilia also regulate intercellular signalling pathways (\nFIG. 1FIG. 1\n) that, when impaired, result in defects that affect physiology (for example, weight control) or the function of various organs, including the heart, kidney, skeleton and brain (\nFIG. 2\n).\nAlthough cilia participate in multiple intercellular signalling pathways (\nFIG. 1\n), Hedgehog signalling is one of the pathways that has been most strongly linked to ciliary function. Hedgehog is a lipoprotein morphogen that participates in the developmental patterning of many vertebrate tissues, including the neural tube and limb buds\n37\n. We focus below on the Hedgehog signal transduction pathway, as its relationship to cilia function may elucidate general principles by which other signalling systems use cilia. Many components of the Hedgehog signal transduction pathway, including the Hedgehog receptor PTCH1, localize to cilia\n3838\n. Binding of Hedgehog to PTCH1 allows the downstream seven-pass transmembrane protein SMO to accumulate inside cilia, where it converts its transcriptional effectors, the GLI proteins, from repressors to activators\n39\n. GLI proteins regulate transcription in the nucleus, but also localize to cilia to function in signalling\n40\n,\n41\n. As discussed below, both IFT (trafficking) and the transition zone (gating) have important roles in dynamically modulating the localization of Hedgehog (and other) signalling components to the cilia. Therefore, many developmental abnormalities that are associated with syndromic ciliopathies, such as polydactyly in BBS and the neural tube defects of Meckel syndrome (MKS), may result from compromised ciliary Hedgehog signalling\n42\n.42\n.\nOther ciliary proteins help to modulate the output of Hedgehog signalling. These include the GPCR GPR161, disruption of which can cause pituitary stalk interruption syndrome, and the EVC–EVC2 complex, dysfunction of which can cause two ciliopathies that are associated with skeletal malformations: Ellis–van Creveld syndrome (EVC) and Weyers acrofacial dysostosis\n43\n–\n45\n(\nFIG. 3\n).\nHedgehog signalling in\nDrosophila melanogaster\nimaginal discs and cuticle does not require cilia, indicating that at least some organisms have evolved ciliumindependent mechanisms for Hedgehog signalling\n37\n. Cilia form on some sensory neurons in\nD. melanogaster\n, including those involved in olfaction, and these neurons use cilium-dependent Hedgehog signalling\n46\n. Thus, within a single organism, a signal transduction pathway can be deployed in both ciliary-dependent and ciliary-independent manners.\nIn vitroIn vitro\nevidence suggests that mammals may also be able to interpret Hedgehog signals through a cilium-independent mechanism, with different outputs from those of cilium-dependent signalling\n47\n.\nEarly ciliogenis is linked to ciliopathies\nDefects in specific ciliary signal transduction components, such as those required for olfaction, phototransduction and Hedgehog signalling, can result in ciliopathies without impairing cilium structure. However, many ciliopathies are caused by the disruption of a specific aspect of the ciliogenic programme, such as the transcriptional regulation of ciliogenesis, basal body formation and the early ciliogenesis pathway, the formation of the transition zone (ciliary gate), and the trafficking machinery responsible for building the ciliary axoneme.\nInitiation of the ciliogenic programmeInitiation of the ciliogenic programme\nIn metazoans, ciliogenesis is initiated by a transcriptional cascade that involves one or more RFX transcription factors, namely, RFX2, RFX3, RFX4 and RFX7 in vertebrates, DAF-19 in\nCaenorhabditis elegans\nand Rfx in\nD. melanogaster\n48\n–\n55\n. These transcription factors bind to X box regulatory motifs to activate the transcription of many genes that are required to build cilia\n52\n,\n55\n,\n56\n. RFX targets include genes that encode components of the transition zone and the IFT–BBSome system. The formation of specialized forms of cilia also requires other transcriptional regulators that may cooperate with RFX transcription factors\n48\n. Examples include forkhead box protein J1 (FOXJ1) for generating motile cilia, GEMC1 (also known as GMNC) and MCIDAS for producing multiciliated cells, the homeobox transcription factor NOTO for generating nodal cilia, and CRX for producing photoreceptors\n57\n–\n6057\n–\n60\n. CRX is associated with two retinal ciliopathies: Leber congenital amaurosis (LCA) and cone–rod dystrophy (CRD)\n61\n. Recently, the TAp73 isoform of TP73 was found to function upstream of RFX and FOXJ1 to contribute to motile multiciliogenesis\n62\n. At least two proteins that are needed for multiciliated cell differentiation, MCIDAS and the regulator of centriole duplication cyclin O (CCNO), underlie motile ciliopathies\n63\n,\n64\n(\nFIG. 3\n; see\nSupplementary information S1 (table)\n). Additional regulators of ciliogenesis probably await discovery, and it is possible that mutations in gene-regulatory elements or in non-coding genes (\nBOX 1\n) may also cause ciliopathies.\nBasal body and initiation of ciliogenesis\nCiliogenesis has long been known to involve basal body docking to the incipient ciliary membrane, after which the transition zone forms and IFT extends the axoneme (\nFIGS 4\n,\n5FIGS 4\n,\n5\n). Further mechanistic details have described, for example, how binding of a ciliary vesicle to distal appendages may be a prerequisite for basal body migration to the cell surface\n65\n. More recent analyses of ciliogenesis have uncovered proteins that functionally connect basal bodies via their distal appendages to membranes, including small GTPases that regulate vesicular trafficking (such as RAB8 and RAB11)\n66\n–\n68\n, proteins that shape membranes (such as EHD1 and EHD3)\n69\nand proteins that promote membrane fusion (for example, the exocyst complex)\n70\n,\n71\n. Consistent with their essential roles in an early step in ciliogenesis, five distal appendage components (CEP164, CEP89, CEP83 (also known as CCDC41), FBF1 and SCLT1) are required for ciliogenesis\n72\n–\n74\n.\nMutations in genes encoding distal appendage components can cause a variety of ciliopathies. For example, the disruption of\nCEP164\nor\nCEP83\ncauses nephronophthisis (NPHP)\n75\n,\n76CEP164\nor\nCEP83\ncauses nephronophthisis (NPHP)\n75\n,\n76\n, a cystic kidney disease, and mutations in\nSCLT1\nmay result in orofaciodigital syndrome (OFD)\n77\n, which is characterized by polydactyly and craniofacial abnormalities (\nFIG. 4\n).\nOther ciliopathy-associated proteins localize to the distal basal body region and are essential for distal appendage formation or function (\nFIG. 4\n). One such protein, HYLS1, is associated with hydrolethalus syndrome, which is a perinatal lethal syndrome that is characterized by hydrocephalus and brain malformation, or the milder ciliopathy Joubert syndrome (JBTS)\n78\n–\n80\n. Other proteins include OFD1 (associated with JBTS, RP, OFD and Simpson–Golabi–Behmel syndrome\n81\n–\n83\n) and C2CD3 (which is linked to OFD\n84\n). Yet another distal basal body component, TALPID3 (also known as KIAA0586), supports ciliogenesis and underlies some cases of JBTS, hydrolethalus syndrome and short-rib polydactyly syndrome\n85\n.85\n.\nHow mutations in genes encoding either distal appendage components or distal basal body components engender such a pleiotropic range of human syndromes is unclear. Hydrolethalus syndrome may result from strong loss-of-function alleles, as the absence of distal appendages in mice severely impairs ciliogenesis and is incompatible with life\n85\n. Other non-lethal ciliopathies are likely to be caused by hypomorphic alleles or mosaicism (\nBOX 2\n). For example, null alleles of mouse\nOfd1\n, an X-linked gene, are lethal in males but recapitulate many human OFD phenotypes in heterozygous females, which are epigenetic mosaics owing to X inactivation\n81\n.\nBox 2 | The complexity of ciliopathies: multigenicity, allelism, cell type specificity, redundancy and modifiers.As the number of ciliopathy-associated genes grows and the range and overlap between ciliopathy phenotypes increase, it is clear that the relationship between a ciliary gene and a ciliopathy is often more complex than a deterministic, Mendelian one-gene-to-one-phenotype relationship.\nFor example, a gene can be implicated in multiple ciliopathies with no, or limited, phenotypic overlap. A single gene can be linked to multiple phenotypes if the alleles are of differing strength. For example, presumed nonsense mutations in\nCC2D2A\nmay cause Meckel syndrome (MKS) (MKS6 subtype), whereas missense mutations in the same gene lead to Joubert syndrome (JBTS) (JBTS9 subtype)\n188\n, suggesting that MKS and JBTS are caused by an allelic series that affects the same essential ciliary function. Similarly, different alleles of\nTMEM231\nare associated with MKS, orofaciodigital syndrome (OFD) and JBTS, even within one family\n101\n,\n111\n,\n189\n.101\n,\n111\n,\n189\n.\nDifferent missense mutations in the same gene can also result in ciliopathies that are associated with distinct ciliary functions. For example, hypomorphic mutations that affect the core IFT-B protein IFT172 result in a skeletal ciliopathy, whereas other mutations cause retinitis pigmentosa (RP) or Bardet–Biedl syndrome (BBS)\n190\n. An intriguing hypothesis to explain how IFT172 can give rise to disparate ciliopathies is that certain mutations do not impair core IFT-B functions but specifically disrupt the association of IFT172 with the BBSome and thus cause BBS. Mutations in\nCEP290\nprovide another example, as they are associated with JBTS, BBS, Leber congenital amaurosis (LCA), MKS and Senior–Løken syndrome (SLSN)\n186\n,\n191\n–\n193\n(\nSupplementary information S1 (table)\n). As it is not clear whether these different ciliopathy-associated mutations form an allelic series, it is possible that they affect distinct functions of CEP290 at the transition zone\n172\n,\n194172\n,\n194\nand centriolar satellites\n88\n,\n195\n(\nFIGS 4\n,\n5\n).\nAnother way in which different mutations in the same gene can result in distinct phenotypes is by affecting protein isoforms that have different functions. For example, disruption of the BBSome-associated protein ARL6 (also known as BBS3) causes typical BBS phenotypes, whereas a longer isoform (BBS3L) is specifically required for photoreceptor maintenance in mice and zebrafish\n196\n.\nAdditionally, genetic modifiers influence the clinical manifestation of mutations in ciliopathy-associated genes. Such modifiers help to identify genes with overlapping or antagonistic functions. For example, mutations in\nRPGRIP1L\n, which encodes a transition zone component, are associated with MKS, JBTS and COACH (cerebellar vermis hypo/aplasia, oligophrenia (mental retardation), ataxia, ocular coloboma, and hepatic fibrosis) syndrome\n197\n,\n198\n. Mutations in a paralogue,\nRPGRIP1197\n,\n198\n. Mutations in a paralogue,\nRPGRIP1\n, cause isolated retinal phenotypes (cone–rod dystrophy (CRD) and LCA)\n199\n,\n200\n.\nCaenorhabditis elegans\nhas only one orthologue of RPGRIP1L and RPGRIP1, which is crucial for transition zone assembly\n10\n,\n99\n. In mammals,\nRPGRIP1\nand\nRPGRIP1L\nmight have overlapping functions, and non-pathogenic alleles may modify the phenotypes that are caused by the pathogenic alleles. Indeed, components of different complexes (transition zone and BBS) have overlapping functions in cilium formation in\nC. elegans\nand mice\n10\n,\n99\n,\n106\n. These findings in model organisms indicate that the type of alleles, the modifiers present in different genetic backgrounds, overlapping protein functions and cell-type specificity can all influence the phenotypic outcome, suggesting that similar genetic complexities underlie human ciliopathies.\nCentriolar satellites\nIn addition to the distal centriole, OFD1, C2CD3 and TALPID3 also localize to centriolar satellites\n86\n–86\n–\n88\n(\nFIG. 4\n). Various centrosomal and ciliary proteins partially localize to these regions, which are found near centrosomes or basal bodies\n89\n. Like OFD1 and C2CD3, many centriolar satellite proteins are essential for cilium formation\n89\n–\n91\n. However, it remains unclear whether such proteins simply localize within the centriolar satellites before their transit to other locations that are directly relevant to ciliogenesis. Some proteins that are involved in centriole duplication and microcephaly, a disorder that is mainly associated with centrosomal dysfunction and with possible ramifications for ciliary signalling\n92\n,\n93\n, also require centriolar satellites\n94\n. Thus, centriolar satellites have roles that are both relevant to, and potentially independent of, ciliary function.\nIn summary, basal body-associated defects can compromise cilium formation or function, resulting in diverse ciliopathies that are often characterized by developmental abnormalities (\nFIG. 4FIG. 4\n). Numerous basal body proteins that are relevant to cilia function have been identified",
         "Retinal degeneration and anosmia are examples of sensory ciliopathies linked to defects in ciliary signaling.",
         null,
         null,
         null
        ],
        [
         "18",
         "What endometrial changes are associated with the beginning of the implantation window?",
         "L Aghajanova et al.,2008, DOI:https://doi.org/10.1016/j.semcdb.2007.10.008:\n 1.1 Approaches to Studying Human Implantation\nEndometrial Tissue\nA common way to study human implantation is to study human endometrium\nin vitro\n, usually obtained by biopsies or from uteri after hysterectomies. A disadvantage of a biopsy in the implantation window is a possible disturbance of an ongoing implantation. There are limited, and hence controversial, reports about the effect of the procedure on pregnancy rates. Some authors [11] claim that the endometrial biopsy at the time of embryo transfer does not exclude viable pregnancy. Others suggest that analysis of protein patterns in endometrial secretions may offer a more safe and non-invasive method of assessing endometrial receptivity during treatment cycles [12].\nBlood and Uterine Fluid BiomarkersThe least invasive method of endometrial fertility assessment would be blood sampling and analysis of a biomarker or a panel of biomarkers predictive of optimal endometrial receptivity. Leukaemia inhibitor factor (LIF) is an example of such a biomarker of the “window of implantation” [13, 14]. While LIF measurements in serum do not reflect fertility status [15], its low concentrations in uterine flushings are predictive of unsuccessful implantation [15]. Similar data were reported for glycodelin, the main secretory product of luteal phase endometrial epithelium and one of the markers of a receptive endometrium. Its concentration in serum and uterine fluid increase in ovulatory cycles [16, 17] and decrease in uterine fluid of infertile women, with no difference in serum levels in those patients [18, 19]. Indeed, most proposed endometrial receptivity markers are not specific to the uterus, and thus it would be difficult to determine whether their serum concentrations play a role in predicting successful implantation. An alternative to peripheral blood analysis could be investigation of uterine flushings; however, this technique is not standardized, and the concentrations of components therein may vary, depending on variations in the procedure [10].Cervical MucusExamination of cervical mucus affords another non–invasive method to study the cytokines and growth factors produced by a receptive endometrium (and transport to the cervical mucus). To the best of our knowledge, there are only two studies that aimed to correlate endometrial expression of cytokines implicated in endometrial receptivity with their levels in cervico-vaginal secretions. The earlier study focused on LIF and demonstrated its presence in cervical mucus [20]. In that analysis, levels of LIF in cervical mucus were highly correlated with its production by endometrial tissue in the periovulatory period. However, in another study LIF was not detectable in cervical secretions throughout the menstrual cycle except for menses [21]. Macrophage-colony stimulating factor (M-CSF), epidermal growth factor (EGF), interleukin-1 beta (IL-1beta), transforming growth factor beta –1 (TGF-beta1) and TGF-beta 2 were detected in cervico-vaginal secretions [21]. However, no correlations between cytokine levels in cervico-vaginal secretions and serum, and between the level of cytokine gene expression in secretory endometrium and the concentration of cytokines in serum collected on the day of endometrial biopsy were found. Only EGF exhibited a significant positive correlation between the level of gene expression in secretory endometrium and its concentration in cervico-vaginal secretions collected on the day of endometrial biopsy, suggesting that EGF may have a predictive value as a minimally invasive marker of uterine receptivity [21].UltrasoundAnother approach has been to use ultrasound evaluation as a predictive method for a receptive endometrium. Jarvela et al [22] reported that in women undergoing IVF, when using three-dimensional Doppler ultrasound, a triple-line pattern after FSH stimulation and a decrease in endometrial volume was associated with conception. However, Sterzik et al. [23] concluded that ultrasonography is an inadequate method to predict endometrial receptivity in IVF cycles, since neither the endometrial thickness nor the echo pattern correlate with the histologic findings. Earlier, Hambartsoumian [24] evaluated a possible relationship between endometrial LIF secretion (endometrial biopsy at cycle day 10) and endometrial growth during IVF cycles. It was demonstrated that LIF production was negatively correlated with endometrial thickness and echo pattern, specifically, and suggested that over-expression of LIF might lead to an inhibitory effect on the endometrium. Taken together, prospective studies on biomarkers as predictors of conception are needed to develop non-invasive methods in assessment of endometrial receptivity.2. Endometrial dating and the window of implantationA normal, ovulatory menstrual cycle (natural, spontaneous cycle) has approximately the same length in each cycle, but can vary in length between women. A standardized (ideal) menstrual cycle that lasts 28 days can be morphologically “dated” according to criteria of Noyes et al. [25]. The histological maturation of the endometrium develops in a distinct pattern that follows the progression of the menstrual cycle. Recently, on a large number of women, it was demonstrated that histological dating of the endometrium does not have the accuracy or the precision necessary for the diagnosis of luteal phase deficiency, or to guide clinical management of women with reproductive failure [26, 27]. It was proposed that the dating of endometrium should be related to the serum LH surge, which corresponds to ovulation, rather than to the “ideal” 28-day cycle [28]. Endometrial histology is most consistent in biopsies from days LH −3/−2 to days LH +7/+8, when changes occur with a high degree of regularity regardless of the length of the preovulatory and postovulatory phases [28]. It is now a common practice to date endometrium according to days after the LH peak [29] [30]. Ultrastructural changes in human endometrium throughout the cycle also allow dating of the endometrium [31, 32] and assist in determining the “implantation window”. The implantation window is a short interval during the mid-secretory phase, when the endometrium is most receptive to blastocyst implantation. It begins on days 20–24 of an ideal menstrual cycle or 6–10 days after the LH surge and is believed to last less than 48 hours [33–35]. Endometrium appears to be the only tissue in which period of embryo implantation is time restricted, i.e.e. embryo can implant only during a specific time period – “implantation window”, whereas extrauterine pregnancy with embryo implanting in various tissues in peritoneal cavity can occur at any time [36, 37].3. Molecular and structural markers of endometrial receptivity\nA principal event of mammalian implantation is the development of a receptive endometrium and its subsequent differentiation into decidua. Many molecular markers such as integrins, IL-1, calcitonin, amphiregulin, EGF, HB (heparin binding)-EGF, colony stimulating factor-1, LIF, mucins, leptin, selectin-L ligands, Hoxa genes, and COX (cyclooxygenase) have been proposed to identify this period of receptivity [9, 38–42]. The beginning of the implantation window is also characterized by remarkable ultrastructural changes in endometrial epithelial cell morphology [31, 32]. However, the implantation process is a complex and multifactorial event, with association and interplay of the different factors involved. Therefore, it is important to know how different markers of implantation correlate with each other. Several studies have demonstrated correlation between pinopode appearance and expression of integrin α\nvv\nβ3 [43], osteopontin [44], LIF and LIF receptor [45], HB-EGF [46], glycodelin [47], progesterone receptors [48], antioxidant enzyme glutaredoxin [49] and cyclooxygenases 1 and 2 [50] in the same endometrial tissue samples. These observations are consistent with an important role of the cytokines, growth factors, and other molecules in the implantation process and reveal interesting spatio-temporal aspects of the expression of some of these.\n4. Transcriptomic approaches to markers of uterine receptivity\nWith the development of microarray technology, global approaches have been pursued to identify novel pathways involved in implantation events [51]. Five studies were published, within a short time frame, reporting the transcriptome human endometrium during the window of implantation [52–56] (\nTable 1Table 1\n). Remarkably, all studies used the same microarray platform to perform their analysis. This makes a comparison between the studies possible; however, there are differences in the type of data analysis, as well as number of patient samples used, ages of subjects, cycle phases compared, and pooling or not pooling samples (\nTable I). Two studies [53, 55] pooled RNA from different patients to run on microarrays. Four out of five studies [52, 53, 55, 56] used pre-defined fold change cut off of 2.0 as evidence of change in gene regulation, while Riesewijk et al. [54] used a more stringent approach and a 3.0 fold change as a cutoff. The latter study is the only one that analyzed samples from the same woman in the early and the mid-secretory phase of the menstrual cycle. These differences in study design may be responsible for the differences in the list of regulated genes identified. The main difference between the studies comes from the analysis of functionally and structurally different stages of the menstrual cycle (proliferative and secretory), which is reflected by the large variability of the up- and down-regulated genes. Interestingly, however, as noted above, Riesewijk et al. [54] performed microarray analysis of endometrium from the same women at LH+2 and at LH+7, which minimizes inter-patient variability. However, caveats include reactive inflammatory changes and subsequent gene alterations as a result of the first biopsy [56], and different microenvironments and/or complement of cell types may be represented in the two biopsies from the same subject.Table I.\nComparison of five studies on human implantation window using the microarray technology approach.\nReference\nN of samples biopsy1/biopsy2\nAge (median)\nEndometrial biopsy 1\nEndometrial biopsy 2\nArray\nN of genes on the array\nDefined fold change\nN of genes up-regulated\nN of genes down-regulated\nKao et al, 2002\n4/7\n28–39\nCycle day 8–10\nLH + 8 - + 10\nAffymetrix HG-U95A\n12 686\n2\n156\n377\nCarson et al., 2002\n3/3\nnot mentioned\nLH+ 2 - + 4\nLH+ 7 - + 9\nAffymetrix HG- U95A\n~ 12 000\n2\n323\n370\nRiesewijk et al., 2003\n5/5\n23–39\nLH+ 2\nLH+ 7\nAffymetrix HG- U95A\n~ 12 000\n3\n153\n58\nBorthwick et al., 2003\n5/5\n23–44 (35.7)\nCycle day 9-11\nLH+ 6 - +8\nAffymetrix HG- U95A-E\n60 000\n2\n90\n46\nMirkin et al., 2005\n3/5\n24–32\nLH+ 3\nLH+ 8\nAffymetrix HG-U95A\n12 686\n2\n49\n58\nOpen in a new tab\nFigure 1Affymetrix HG-U95A\n12 686\n2\n49\n58\nOpen in a new tab\nFigure 1\nrepresents the distribution of significantly changed genes among different groups of genes in the studies by Kao et al. [52], Carson et al [53], Borthwick et al [55], Riesewijk et al. [54] and Mirkin et al. [56]. Genes with unknown biological function represented one of the biggest groups of regulated genes. Cell surface proteins, extracellular marker components, and growth factors/cytokines represent a large fraction of the up-regulated genes in mid-secretory phase in these studies, as well as genes encoding intracellular signaling and cell cycle proteins. Immune genes were also represented in four out of five studies, implicating an important role for the immune cells and processes in endometrium prior to and during embryonic implantation. Genes encoding DNA binding proteins, transcription factors and DNA modifying enzymes were highly represented among down-regulated genes, as well as genes with unknown function and ESTs (Figure 1\n). These approaches have enabled identification of some genes not previously known to be involved in the implantation process [56], and further studies are required to elucidate their potential roles.\nFigure 1.\nOpen in a new tab\nDistribution of changes in gene expression among different functional classes. A – results from Kao et al., 2002; B – Riesewijk et al., 2003; C - Carson et al., 2002; D –Borthwick et al., 2003; E – Mirkin et al., 2005.We have analyzed the studies above in an effort to identify common genes present in all studies and which change in similar way during the receptive period in mid-secretory endometrium. Similar analyses were performed earlier by Horcajadas et al. [57] and Mirkin et al. [56]. There are very few genes that were significantly and similarly regulated during the window of implantation in at least four out of the five reports. These surprising results could be explained by some differences in study design among all studies (in particular, comparison of mid-secretory phase endometrium to either proliferative or early secretory endometrium). Up-regulated transcripts in common between studies are: osteopontin (SPPI) – the only gene up-regulated according to all studies, decay accelerating factor for complement (CD55, Cromer blood group system), growth arrest and DNA damage-inducible protein (GADD45), apolipoprotein D, Dickkopf/DKK1, monoamine oxidase A (MAOA), interleukin 15 (IL15) and mitogen-activated protein kinase kinase kinase 5 (MAP3K5); the down-regulated gene in common in those studies was olfactomedin-related ER localized protein. A brief overview of those factors is presented.The significant up-regulation of osteopontin in all five studies is remarkable, since it is acknowledged to be involved in the implantation process. This glycoprotein is a ligand for αvβ3 integrin [58, 59], it mediates cellular adhesion and migration during embryo implantation, is regulated by progesterone, and its maximal expression in endometrial epithelial cells has been observed in the window of implantation [60–62]. Moreover, the localization of osteopontin on the surface of endometrial pinopodes has been demonstrated [44].Decay accelerating factor for complement (DAF) is a complement protective protein, highly expressed in the human endometrial epithelium in mid- to late secretory phase and regulated by HB-EGF and EGF [63–65]. In was highly up-regulated in our recent microarray study in normal mid-secretory endometrium with subsequent decrease in the late secretory phase [62] and was decreased in endometrium from women with luteal phase defect and patients with recurrent pregnancy loss associated with antiphospholipid syndrome [66, 67]. It functions probably by protecting the endometrial integrity from complement components increased in secretory endometrium [64].\nGADD45 is member of a group of stress inducible genes. Its up-regulation in mid-secretory endometrium reflects the escalation of endometrial protective mechanisms in anticipation of implantation and invasion of a blastocyst. High expression of GADD45 in high receptivity cell lines compared to the low receptivity lines was reported [29].Apolipoprotein D is a multi-ligand, multi-functional transporter involved in lipid metabolism and cholesterol transport; it can bind cholesterol, progesterone, pregnenolone, bilirubin and arachidonic acid, but it is unclear which of these represent its physiological ligands [68]. Up-regulation of this gene in four out of five studies underscores the regulation of lipid metabolism during endometrial maturation.\nDKK-1 is a potent inhibitor of the Wnt signaling pathway [69]. It was recently found to be specifically up-regulated by progesterone in\nin vitro\ndecidualized endometrial stromal cells [70], was up-regulated in microarray study of mid-secretory endometrium [62] and down-regulated in eutopic endometrium from women with endometriosis [52].Monoamine oxidase A (MAOA) is localized in the outer membrane of mitochondria and catalyses the oxidative deamination of a variety of monoamines using flavin adenine dinucleotide as cofactor [71]. Henriques et al [72] reported up-regulation of MAOA gene and protein in mid-secretory compared to early secretory endometrium, and decreased MAOA expression in donor oocyte recipients with implantation failure. Interleukin 15 is progesterone regulated gene in endometrial stromal cells and is important as chemoattractant and stimulator of natural killer (NK) cell replication [73, 74]. Interestingly, in women with unexplained recurrent abortion, there are elevated levels of endometrial IL-15 compared with control endometrium [75, 76]. This chemokine was shown to play a central role in postovulatory recruitment of CD16(−) NK cells from peripheral blood into human endometrium [77].MAP3K5 is a member of mitogen-activated protein kinase (MAPK) signaling cascades. Expression and activation of this pathway in human endometrium is increased in secretory phase, in particular in mid-secretory phase [78] and alteration in it’s phosphorylation may be involved in pathogenesis of endometriosis [79].\nOlfactomedin-related ER localized protein, or olfactomedin 1, is a down-regulated gene in common in the five microarray studies. The exact function of the protein is not known. Its expression was increased in endometrium from women with unexplained recurrent spontaneous abortion, and it induced cell arrest in a human endometrial cell line [80].Unexpectedly, several biomolecules with an established role in human implantation process by one-to-one approach (LIF, HOXA10, L-selectin, HB-EGF etc) did not pass the defined fold-change threshold in the microarray studies described above. The existing discrepancies among the array studies confound identification of definitive markers of uterine receptivity. Analysis of endometrial tissue obtained by standard operating procedures shared by all investigators and from large numbers of well-characterized subjects, with the same study design is anticipated to be extremely informative.\n5. Model of early embryonic implantation\nThere are multiple interactions that must occur for successful embryonic implantation into a receptive endometrium, and the transcriptome of the endometrium in the implantation window provides the beginning to understanding this complex process (\nFigure 2Figure 2\n). Notably, the endometrial transcriptome in the mid-secretory phase studied to date is a static snapshot of events occurring in this tissue as it awaits embryonic implantation. Of equal importance are the close spatio-temporal relationship between the conceptus and the maternal compartment and how their physical interactions affect/change their respective transcriptomes. A few studies have used\nin vitroin vitro\nmodels of human implantation that focused mainly on ultrastructural relationships between the attached blastocyst and the endometrial epithelium, e.g. in a three dimensional cell culture system [81–84]. Carver et al. [85] demonstrated hCG-secreting activity of a blastocyst implanted into endometrial stromal cells monolayer. Mercader et al [86] reported results of a 5 year-long study, where blastocysts and endometrial epithelial cells were co-cultured in IVF and oocyte donation programs. Such an approach dramatically increased implantation and pregnancy rates in IVF patients. Further research by the same group, studying the secretome of human blastocysts cultured in sequential (routinely used) media versus co-culture with endometrial epithelial cells, identified, in a preliminary report, IL-6 as a predictor of implanting versus non-implanting blastocysts [87]. Microarray studies of the whole genome of endometrial cells during/after physical contact with the human blastocystin vitro\nwould be valuable and informative about the molecular pathways driving the initiation of pregnancy, although this is a challenge experimentally. Finally,\nin vitro\nmodels of the invasive phase of implantation, such as co-cultures of endometrial and trophoblast cells and treatment of endometrial cells with trophoblast conditioned medium, are informative about the invasive phase of implantation (and are reviewed elsewhere [\n88\n].\nFigure 2. Model of events in early human implantation.\nOpen in a new tab\nLIF- leukemia inhibitory factor; N-Ac6-ST - N-acetylglucosamine-6-O-sulfotransferase; CPE receptor - Clostridium perfringens enterotoxin receptor; IDO - Indoleamine 2,3-dioxygenase; IL-15 – interleukin 15.",
         "The beginning of the implantation window is characterized by ultrastructural changes in endometrial epithelial cell morphology.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "19",
         "What device was used to take endometrial biopsies in the study?",
         "Linden Stocker et al.,2017, DOI:https://doi.org/10.1038/s41598-017-07901-6:\n The study was conducted under local ethical (REC number 12/SC/0548) and R&D approval (RHM O&G 0197).  Patients were recruited from either elective gynecological theatre day lists or In Vitro -Fertilisation (IVF) clinics  at Princess Anne Hospital, Southampton, UK in accordance with local guidelines. Informed consent was gained  from all participants. Participants and Tissue Collection. Eligible participants (n = 45) were women aged between 25-45 years,  attending for gynaecological procedures or fertility clinics from August 2012 to December 2013. Baseline demographics and fertility characteristics were collected for all patients. There were two independent groups of cases.  The first were women who had suffered with recurrent implantation failure (women with failure of clinical pregnancy following the transfer of three or more good quality fresh or frozen embryos transferred over two or more  IVF or ICSI cycles) (n = 15). The second group were women who had undergone recurrent miscarriage (women  who had suffered three or more spontaneous pregnancy losses at fewer than 24 completed weeks of pregnancy)  (n = 15). The third group, who served as controls, were women who were attending for elective procedures for  non-endometrial pathology (n = 15) and has had at least one full term pregnancy ending in live birth without a  history of recurrent implantation failure or recurrent miscarriage as previously described. Endometrial biopsies  were taken by suction curette (Pipelle device, Laboratoire CCD, Paris, France), washed in sterile saline, divided  into uniform size (10mm lengths), snap frozen and stored at 80 °C in separate aliquots. RNA Extraction and cdna Synthesis. Total RNA was extracted from endometrial biopsies using the TRIzol  (Thermo Fisher Scientific, USA). The A260/280 ratio of each sample was measured using mass spectrometry  (NanoDrop; Thermo Fisher Scientific USA) and the total RNA concentration for each sample was calculated.  Samples were stored at −80 °C until use. Total RNA was reverse transcribed to produce cDNA (Precision nanoScript RT kit; Primerdesign Ltd, UK) with Oligo-dT primers. The mRNA expression of the 10 genes listed  in Table 3 were measured by qRT-PCR using primers and probes designed and made by Primerdesign Ltd  (Southampton, UK). These sets of genes are from the geNorm Plus Reference Gene Selection kits. All samples  were measured in duplicate using a LightCycler 480 Instrument (Roche Diagnostics, Germany). The optimised  cycle parameters were 95 °C for 2 min, followed by 40 cycles of 95 °C for 15 seconds, 60 °C for 60 s. Statistical Analysis.  Quantification cycle (Cq) values were transformed into relative quantification data  using the delta-delta-CT method 38. These were then converted into relative quantities (RQs) by calculating the  average Cq value for replicates and using the average Cq for the given gene and the amplification efficiency.  Samples for the reference gene stability analyses were divided into three groups (RIF , RM and control endometrium). To determine the stability of the HKGs a computer algorithm the geNorm Software was used (geNorm  software qbasePLUS, Version: 2.6.1, Southampton UK) 2. Part of the functionality of this program, as well as  processing the transformed data and measuring gene expression is calculating gene stability. This is described as  stability measure ‘M’ for a given reference gene and is generated as the average pairwise variation for that gene  compared with all other tested reference genes (with a lower M value indicating more stability). Pairwise variation  (V , V(n/nþ1)) determines the number of HKG required for accurate normalization and the benefit gained from  additional HKG. A V score of 0.15 or below indicates that the additional gene has no significant contribution to  the newly calculated normalization factor and is therefore not needed 2,9. M values were further analysed using  the Prism Software (Version 7.0b, GraphPad Software, Inc). Group differences were evaluated using independent  samples t-test to elucidate differences in both patient groups and differences in M-Values. P < 0.05 was considered  statistically significant. Data values are represented as mean ± standard deviation (SD).",
         "A suction curette (Pipelle device, Laboratoire CCD, Paris, France) was used.",
         null,
         null,
         null
        ],
        [
         "20",
         "What were the P-value thresholds used for identifying new potential TAS loci in Col-0 and rdr2?",
         "Ho-Ming Chen et al.,2007, DOI:https://doi.org/10.1073/pnas.0611119104:\n Computational Prediction of\nTAS\nGenes.\nThe\nArabidopsis\nMPSS small RNA signatures was downloaded from\nhttp://mpss.udel.edu/at\n(\n16\n,\n19\n) (as of September 2005 for Col-0 and November 2006 for\nrdr2\n). A total of 73,086 and 18,927 distinct signatures corresponding to the first 17-nt sequence of 21- to 24-nt small RNA sequences were extracted from the Col-0 and\nrdr2\ndata sets, respectively. Small RNA signatures with greater than six hits in the\nArabidopsis\ngenome were further filtered and removed because of mapping ambiguity. The coordinates of MPSS small RNA signatures were then subjected to the analyses using the\nTAS\nprediction algorithm developed based on Perl. For\nTable 1\n, the fragment size is determined by multiplying the phasing interval size by 11 for calculating small RNAs phased in 19- to 20- or 22- to 24-nt intervals. The above equation is also adjusted accordingly to reflect the numbers of phased and nonphased positions. New potential\nTAS\nloci with\nP\n< 0.0006 for Col-0 and\nPTAS\nloci with\nP\n< 0.0006 for Col-0 and\nP\n< 0.005 for\nrdr2\n, respectively, were searched against the\nArabidopsis\nSmall RNA Project (ASRP) data set (\n24\n) (\nhttp://asrp.cgrb.oregonstate.edu/db/\n) for potential small RNA(s) that could target and set the phase for the production of small RNA clusters identified in our study.\nPlant Materials and Growth Conditions.\nSeeds of\nArabidopsis\nmutant lines\ndcl1\n-\n9\n(CS3828) (\n35\n),\ndcl2\n-\n1\n(SALK_064627),\ndcl3\n-\n1\n(SALK_005512),\ndcl4\n-\n2\n(GABI_160G05),\nrdr6\n-\n11\n(CS24285) (\n8\n),\nsgs3\n-\n11\n(CS24289) (\n8\n) and\ntas2\n(SALK_014168) were obtained from the\nArabidopsis\nBiological Research Center or Nottingham\nArabidopsis\nStock Center. Seedlings of Col-0, Ler, and all mutants used in this study were grown in 1% agar plates containing 1× MS and 1% sucrose under a 16/8-h light/dark cycle at 22°C.\nSmall RNA Northern Blot Analyses.Small RNA Northern Blot Analyses.\nTotal RNA was extracted from 14-d-old seedlings with use of TRIZOL reagent (Invitrogen, Carlsbad, CA). Fifty micrograms of total RNA was separated by 15% denaturing polyacrylamide TBE-Urea gels (Invitrogen) and transferred to Hybond-N\n+\nmembranes (GE Healthcare, Piscataway, NJ) by use of a transblot semidry transfer cell (Bio-Rad, Hercules, CA). Antisense DNA oligonucleotides of 21 nt complementary to predicted ta-siRNAs were used as probes (for At1g63130-siR5s, 5′-CACAACCATCCGGTCAACTAA-3′; At1g63130-siR9as, 5′-GTGATATTGATTTGGCTTTGA-3′; and At1g63080-siR3as, 5′- TCATGGGCTTTTTCAACACAA-3′). The probes were end-labeled with [γ-\n32\nP]ATP by T4 polynucleotide kinase. The membrane was hybridized with ULTRAhyb-Oligo buffer (Ambion, Austin, TX) and exposed to Kodak BioMAX MS x-ray films for 3–5 days.\nValidation of ta-siRNA Targets.Validation of ta-siRNA Targets.\nModified 5′ RACE with the GeneRacer kit (Invitrogen) was adapted to validate the cleavage site determined by ta-siRNA targeting. Nested primers were used in PCRs and the cleavage sites were revealed by sequence analyses of the PCR product. Two nested primers for At1g63080 were primary, 5′-ACATATCTGTGACCAAGCCATAAGTTG-3′, and secondary, 5′-CGTCTGGCAGACAATCTTTGCTAACCAT-3′. Three nested primers for At1g62930 were primary, 5′-CTGACTGCAGAGAACTGTACCAGTCATG-3′; secondary, 5′-GGAAGAGTCCCATCTTCTTTCATTTCTC-3′; and tertiary, 5′-CAAAGGCATCTTATGAGGGAGTTGTAGG-3′.",
         "New potential TAS loci had P < 0.0006 for Col-0 and P < 0.005 for rdr2.",
         null,
         null,
         null
        ],
        [
         "21",
         "What were the overall implantation, clinical pregnancy, and live birth rates in the analyzed HRT-FET cycles?",
         "Zhang Shaodi et al.,2020, DOI:https://doi.org/10.1371/journal.pone.0239120:\n A total of 10,165 HRT-FET cycles were analyzed and reported here with the overall implantation rate of 46.45%, clinical pregnancy rate of 62.11%, and live birth rate of 51.08%. All patients were grouped according to their endometrial thickness on the ET day. Patient demographics and characteristics are shown in Table 1. There were significant differences between the age, infertility duration, BMI and percentage of primary infertility of the women in each group, but there was no difference in the number and type of embryos transferred. (Table 1) Comparison of clinical outcomes between different groups The implantation rate, the early miscarriage rate, clinical pregnancy rate and live birth rate were significantly different among groups that had different endometrial thickness (P < 0.001). In addition, there were no statistical significances among in the ectopic pregnancy rate, delivery gestational weeks and newborn birth weight between groups (p > 0.05) (Table 2). Table 1.05) (Table 2). Table 1. Characteri stics and periodic features in patients grouped by endometrial thickn ess on the embryo transfer day. Group EN 6 mm EN 6-7 mm EN 7-8 mm EN 8-9 mm EN 9-10 mm EN 10-11 mm EN 11-12 mm EN 12-13 mm EN>13 mm P-value N 235 445 1526 3338 2299 1218 615 291 198 Age 34.06 ± 6.19 33.54 ± 6.09 32.36 ± 5.92 31.17 ± 5.60 30.99 ± 5.42 31.08 ± 5.27 31.69 ± 5.71 31.84 ± 5.49 32.67 ± 6.50 <0.001 Duration of infertility 3.44 ± 2.82 3.73 ± 3.13 4.06 ± 2.99 4.05 ± 2.89 4.30 ± 3.30 4.31 ± 3.27 4.62 ± 3.44 4.38 ± 3.53 5.11 ± 3.57 <0.001 BMI 22.74 ± 3.07 23.18 ± 3.74 23.02 ± 3.42 22.87 ± 4.24 23.04 ± 4.63 23.33 ± 7.47 23.47 ± 3.32 23.69 ± 3.70 23.80 ± 3.64 <0.001 No. of embryo transferre d 1.81 ± 0.43 1.82 ± 0.43 1.83 ± 0.41 1.85 ± 0.38 1.85 ± 0.38 1.84 ± 0.37 1.86 ± 0.35 1.85 ± 0.37 1.81 ± 0.43 0.273 Infertility type <0.001 Primary Infertility 25.33% (58/229) 26.09% (114/437 ) 41.26% (621/1 505) 52.78% (1727/ 3272) 56.01% (1267/2262) 59.00% (705/1195) 60.23% (362/601) 57.26% (621/1 505) 52.78% (1727/ 3272) 56.01% (1267/2262) 59.00% (705/1195) 60.23% (362/601) 57.19% (163/285 ) 56.48% (109/1 93) Secondary infertility 74.67% (171/229) 73.91% (323/437 ) 58.74% (884/1 505) 47.22% (1545/ 3272) 43.99% (995/2262) 41.00% (490/1195) 39.77% (239/601) 42.81% (122/285 ) 43.52% (84/19 3) Embryo type 0.433 Embryos in cleavage stage 76.17% (179/235) 74.83% (333/445 ) 72.94% (1113/ 1526) 72.44% (2418/ 3338) 71.58% (1645/2298) 70.11% (854/1218) 73.98% (455/615) 71.82% (209/291 ) 73.23% (145/1 98) Blastocyst 23.83% (56/235) 25.17% (112 /445) 27.06% (413/1 526) 27.56% (920/3 338) 28.42% (653/2298) 29.89% (854/1218) 26.02% (160/615) 28.18% (82/291) 26.77% (53/19 8) https://do i.org/10.1371/j ournal.pone .0239120.t001 PLOS ONE The effect of endometr ial thickness on pregnanc y outcomes PLOS ONE | https://doi.or g/10.137 1/journal.po ne.02391 20 September 24, 2020 4 / 12  Table 2. Clinical outcome s in patients grouped by endom etrial thickn ess on the embryo transfer day. Clinical outcome s in patients grouped by endom etrial thickn ess on the embryo transfer day. Group EN  6 mm EN 6-7 mm EN 7-8 mm EN 8-9 mm EN 9-10 mm EN 10-11 mm EN 11-12 mm EN 12-13 mm EN >13 mm P-value N 235 445 1526 3338 2299 1218 615 291 198 Implantatio n rate 27.76% (118/425) 28.94% (235/8 12) 39.26% (1097/ 2794) 49.50% (3059/6180) 48.81% (2078/4257) 51.09% (1098/2245) 51.31% (587/1144) 50.65% (273/539 ) 46.37% (166/358) <0.001 Clinical pregnancy rate 38.46% (90/234) 44.14% (196/4 80) 54.24% (825/1 521) 65.83% (2192/3330) 65.22% (1498/2297) 64.86% (788/1215) 67.48% (415/615) 67.01% (195/291 ) 62.63% (124/198) <0.001 Ectopic pregnancy rate 4.44% (4/90) 2.55% (5/196) 1.82% (15/825) 1.32% (29/2192) 1.67% (25/1498) 1.52% (12/788) 0.72% (3/415) 1.54% (3/195) 1.61% (2/124) 0.313 Early miscarriage rate 25.56% (23/90) 17.86% (35/196) 13.58% (112/8 25) 10.40% (228/2192) 10.35% (155/1498) 9.90% (78/788) 10.12% (42/415) 12.82% (25/195) 16.94% (21/124) <0.001 Live birth rate 25.90% (78/788) 10.12% (42/415) 12.82% (25/195) 16.94% (21/124) <0.001 Live birth rate 25.11% (59/235) 32.13% (143/4 45) 42.60% (650/1 526) 54.70% (1826/3338) 54.76% (1259/2299) 54.35% (662/1218) 55.77% (343/615) 53.26% (155/291 ) 47.98% (95/198) <0.001 Gestationa l week at delivery 37.34 ± 2.35 37.67 ± 2.36 37.44 ± 2.64 37.57 ± 2.50 37.61 ± 2.43 37.54 ± 2.49 37.49 ± 2.49 37.60 ± 2.35 37.66 ± 1.63 0.912 Newborn birth weight 3114.66 ± 684.58 3196.92 ± 655.47 3099.89 ± 731.15 3106.46 ± 691.47 3119.72 ± 678.25 3153.67 ± 698.83 3084.11 ± 674.71 3137.08 ± 658.12 3224.74 ± 668.59 0.456 https://doi.o rg/10.1371/j ournal.pone .0239120.t002 PLOS ONE The effect of endometr ial thickness on pregnanc y outcomes PLOS ONE | https://doi.or g/10.137 1/journal.po ne.or g/10.137 1/journal.po ne.02391 20 September 24, 2020 5 / 12  Multivariable logistic regression used to evaluate the association between endometrial thickness and pregnancy outcomes In order to control the impact of age, the duration of infertility, BMI and the infertility type (primary or secondary) on pregnancy outcomes, we performed multivariable logistic regression analysis to evaluate the association between endometrial thickness and pregnancy outcomes. After making adjustments based on the above factors, significant associations were found between endometrial thickness and implantation rate(adjusted odds ratio [aOR]: 1.08; 95% confidence interval [CI]: 1.06-1.10, p < 0.0001), clinical pregnancy rate(aOR: 1.10; 95% CI: 1.07-1.14, p < 0.0001) and live birth rate (aOR: 1.09; 95% CI: 1.06-1.12, p < 0.0001).10; 95% CI: 1.07-1.14, p < 0.0001) and live birth rate (aOR: 1.09; 95% CI: 1.06-1.12, p < 0.0001). The associations between endometrial thickness and early miscarriage rate, ectopic pregnancy rate, gestational weeks at delivery and newborn birth weight were not significant (Table 3). Curve fitting For continuous variables such as endometrial thickness, the implantation rate, clinical pregnancy rate and live birth rate, the fitted curves are presented in Figs 1-3, respectively. With the increase of endometrial thickness, the implantation rate, clinical pregnancy rate and live birth rate initially went up and then down (Figs 1-3). Threshold effect analysis results The threshold effect analysis of endometrial thickness and the implantation rate, clinical pregnancy rate and live birth rate are presented in Table 4. The endometrial thickness is a non-linear significant predictor of clinical outcomes, and its turning point is 8.7 mm.7 mm. The implantation rate, clinical pregnancy rate and live birth rate increased by 32%, 36% and 45%, respectively with the increase of each millimeter increment of endometrial thickness up to 8.7 mm. When the endometrial thickness was  8.7 mm, the clinical outcomes did not increase significantly with but rather tended to be stable. Influence of the delta between the endometrial thickness on the transfer day compared with the thickness at the starting of progesterone day on the clinical outcome Between the endometrial thickness on the transfer day compared with the thickness at the starting of progesterone day, endometrial thickness had no changed cycles, the increased and compaction cycles accounted for 18.88%, 38.05%, 47.80% of the total number of cycles, respectively. The clinical pregnancy rate of cycles with compaction endometrial thickness is lower than the other two groups (P<0.001) (Table 5). Table 3.001) (Table 5). Table 3. Associati ons between endom etrial thickne ss and clinical outcomes of HRT-FET cycles using multivar iable logistic regress ion analysi s. All adjusted for age, the duration of infertility, body mass index, infertility type, and the number type of embryos transfe rred. Item Non-adjus ted Adjusted I Adjusted II OR 95% CI P OR 95% CI P OR 95% CI P Implantat ion rate 1.10 (1.08, 1.12) < 0.0001 1.08 (1.06, 1.10) <0.0001 1.08 (1.06, 1.10) < 0.001 Clinical pregnancy rate 1.13 (1.10, 1.15) < 0.0001 1.10 (1.07, 1.14) <0.0001 1.10 (1.07, 1.14) < 0.001 Ectopic pregnancy rate 0.90 (0.79, 1.02) 0.1075 0.92 (0.80, 1.06) 0.2369 0.92 (0.80, 1.06) 0.2427 Early miscarriage rate 0.95 (0.90, 0.99) 0.0284 0.96 (0.91, 1.01) 0.1101 0.96 (0.91, 1.01) 0.1307 Live birth rate 1.12 (1.09, 1.14) <0.0001 1.09 (1.07, 1.12) <0.0001 1.09 (1.06, 1.12) <0.001 Gestation al age at delivery 0.0 (-0.0, 0.1) 0.5420 0.0 (-0.0, 0.1) 0.2283 0.0 (-0.0, 0.1) 0.2252 Newborn birth weight 7.2 (-4.9, 19.0, 0.1) 0.5420 0.0 (-0.0, 0.1) 0.2283 0.0 (-0.0, 0.1) 0.2252 Newborn birth weight 7.2 (-4.9, 19.3) 0.2426 10.3 (-2.1, 22.7) 0.1027 10.2 (-2.2, 22.6) 0.1055 https://d oi.org/10.1371/j ournal.pon e.0239120.t00 3 PLOS ONE The effect of endometr ial thickness on pregnanc y outcomes PLOS ONE | https://doi.or g/10.137 1/journal.po ne.02391 20 September 24, 2020 6 / 12",
         "The overall implantation rate was 46.45%, the clinical pregnancy rate was 62.11%, and the live birth rate was 51.08%.",
         null,
         null,
         null
        ],
        [
         "22",
         "What statistical model does edgeR use to account for variability in count data?",
         "Mark D Robinson et al.,2009, DOI:https://doi.org/10.1093/bioinformatics/btp616:\n Summary:\nIt is expected that emerging digital gene expression (DGE) technologies will overtake microarray technologies in the near future for many functional genomics applications. One of the fundamental data analysis tasks, especially for gene expression studies, involves determining whether there is evidence that\ncounts\nfor a transcript or exon are significantly different across experimental conditions.\nedgeR\nis a Bioconductor software package for examining differential expression of replicated count data. An overdispersed Poisson model is used to account for both biological and technical variability. Empirical Bayes methods are used to moderate the degree of overdispersion across transcripts, improving the reliability of inference. The methodology can be used even with the most minimal levels of replication, provided at least one phenotype or experimental condition is replicated. The software may have other applications beyond sequencing data, such as proteome peptide count data.\nAvailability:\nThe package is freely available under the LGPL licence from the Bioconductor web site (\nhttp://bioconductor.org\n).\nContact:\nmrobinson@wehi.edu.au",
         "An overdispersed Poisson model is used.",
         null,
         null,
         null
        ],
        [
         "23",
         "What is the function of the fly heart's aorta?",
         "Berrak Ugur et al.,2016, DOI:https://doi.org/10.1242/dmm.023762:\n Nervous system assaysThe nervous system in Drosophila is required for sensing and processing information related to vision, hearing, olfaction, proprioception and taste. Just like in humans, this information is conveyed to the CNS and processed to provide a motor output. Although the gross anatomy of Drosophila and human brain is very different, they share numerous conserved genetic, cellular, electrophysiological and chemical properties. As in vertebrates, many different types of neurons are required to process information in fruit flies. For example, in the visual system, ∼115 different types of neurons have been identified in Drosophila , which is a very similar number to what has been estimated in vertebrates ( Venken et al., 2011 ). However, there are probably a million-fold fewer neurons overall in flies than in vertebrates. The reduced complexity and ease to study the Drosophila nervous system allows an in-depth assessment of the function of genes and neuronal networks. Many different assays have been developed to assess neuronal function. These include hearing, flight, learning and memory, and diurnal rhythmicity assays, as well as numerous highly specific behavioral assays ( Branson et al., 2009 ; Inagaki et al., 2010 ; McGuire et al., 2005 ; Simon and Dickinson, 2010 ). In the last two decades, Drosophila has been increasingly used to model neurological dysfunction, including neurodegeneration, epilepsy, dementias, stroke, traumatic brain injury and brain tumors. In this section, we provide a snapshot of the different assays that can be used to study neurological disorders and describe the specific contexts in which they have been most useful so far. These tools allow researchers to gain novel insights into pathogenic mechanisms and might help to provide new therapeutic strategies for different neurological diseases.Electroretinogram\nUnlike vertebrate photoreceptors (PRs), which are light-sensory cells in the retina that connect to a neuron, fly PRs themselves are light-sensory neurons that project axons into the deep lamina and medulla layers of the adult brain. The ease of accessibility of fly PRs and their regular arrangement have facilitated the development of several assays to study retinal disorders and other more general neurological phenotypes. The electroretinogram (ERG) records the change in electrical activity of the PRs in response to a flash of light. The ERG depolarization amplitude provides a read-out of the phototransduction process, whereas the on/off transient spikes at the onset and offset of a light flash correspond to postsynaptic potential changes (see poster) ( Hardie and Raghu, 2001 ; Stark and Wasserman, 1972 ).ERG recordings were adapted to assess mutant phenotypes for Drosophila in the late 1960s ( Hotta and Benzer, 1969 ; Pak et al., 1969 ). Given the ease of the assay, forward genetic screens based on ERGs in homozygous viable mutants permitted the identification of numerous genes that control the phototransduction pathway ( Wang and Montell, 2007 ). Impaired phototransduction results in aberrant ERG recordings; for example, reduced depolarization amplitude and/or loss of the on and off transients (see example in the poster). Most of the mutations that affect the phototransduction pathway do not affect viability because the eye is not an essential organ in the fly. To circumvent the issue of lethality and study the role of essential genes in the eye, a technique was developed to generate eye-specific mosaic clones ( Newsome et al., 2000 ; Stowers and Schwarz, 1999 )., 2000 ; Stowers and Schwarz, 1999 ). This allowed the identification of genes that are more broadly implicated in neurodegeneration by measuring ERGs in young and old animals to document the time-dependent change in neuronal function ( Haelterman et al., 2014 ; Yamamoto et al., 2014 ). Using this approach, the Bellen lab identified several fly mutants in which the human homologs have been implicated by other studies to play a role in various neurodegenerative diseases, including Leigh syndrome ( C8orf38/sicily ) ( Zhang et al., 2013b ), Charcot-Marie-Tooth disease type 2A ( MFN2/Marf ) ( Sandoval et al., 2014 ) and autosomal recessive spastic ataxia with leukoencephalopathy ( MARS2 / Aats-met ) ( Bayat et al., 2012 ). In addition, we recently determined that reactive oxygen species (ROS) accumulate in the PRs of these mutants and, in turn, this triggers lipid-droplet formation that eventually contributes to PR neurodegeneration ( Liu et al., 2015 )., 2015 ). These examples illustrate how ERGs can be effective tools to identify genes and elucidate molecular mechanisms underlying neurodegeneration.Neuromuscular junction electrophysiologyThe neuromuscular junction (NMJ) is the connection (synapse) between the motor neuron and the muscle. Disorders of the NMJ span a variety of phenotypes and can be due to genetic or acquired causes ( Rodríguez Cruz et al., 2014 ). The Drosophila NMJ provides a powerful platform to study neuromuscular diseases because it allows detailed analyses of structural connections between the neuron and the muscle as well as their electrophysiological properties. The fly larval NMJ consists of arrays of overlapping striated muscle fibers that are innervated by motor neurons that form synaptic boutons ( Prokop and Meinertzhagen, 2006 ). The fly NMJ is a large glutamatergic synapse that is easy to access, thereby permitting a detailed characterization of the properties of synaptic transmission, including assessment of excitatory junctional potentials (EJPs), spontaneous miniature EJPs, synaptic plasticity, transmission electron microscopy imaging and pre- versus postsynaptic phenotypic analysis. Below, we discuss the NMJ electrophysiological assay (see poster).The electrophysiological recording assay for the third instar larval NMJ was described in the seventies ( Jan and Jan, 1976 ), and a parallel assay based on patch clamping in embryos was developed later ( Broadie and Bate, 1993 ). Briefly, the larva or embryo is filleted to expose the muscles along the body wall. A motor neuron is cut posterior to the ventral ganglion and drawn into an electrode in order to induce action potentials. A second electrode is placed in or on the muscle to measure the response. Both assays permit a very accurate characterization of a mutant's neuronal function but they are quite labor-intensive and thus are not the best tools to perform genetic screens. Instead, genetic screens for altered NMJ morphology have been performed to identify genes regulating synaptic bouton morphology ( Aberle et al., 2002 ) because the reiterated pattern of NMJs along the body wall lends itself to rapid visual inspection. NMJ electrophysiological assays have allowed detailed characterization of numerous mutant genes whose human homologs were later shown to cause different human diseases. For example, Shaker ( Sh ) and ether-a-go-go ( eag ) mutations were isolated owing to the shaking phenotype of the legs ( Kaplan and Trout, 1969 ). Subsequently, further characterization of these mutants using NMJ electrophysiological recordings revealed that the mutations affect potassium channel function ( Jan et al., 1977 ; Wu et al., 1983 ). The vertebrate homologs of these potassium channels were later identified and shown to be involved in many human diseases, including cardiac arrhythmias, deafness and epilepsy ( Jentsch, 2000 ). Another example is the Drosophila homolog of the human genes synaptotagmin 1 and 2, which was first shown in flies to be the calcium (Ca 2+ ) sensor for synaptic transmission, based on altered NMJ recordings (see poster) ( Littleton et al., 1994 )., 1994 ). Recently, mutations in human synaptotagmin 2 have been discovered to cause Lambert-Eaton myasthenic syndrome, a rare autoimmune neuromuscular disorder ( Herrmann et al., 2014 ).Of note, the properties of Drosophila larval NMJs can be further studied using a variety of different assays, allowing an in-depth analysis that is not possible at any other synapse. These include live imaging of protein or organelle trafficking in the motor neuron axon or synapse ( Andlauer and Sigrist, 2012 ), Ca 2+ imaging of synaptic boutons ( Macleod, 2012 ) and focal patch recordings from single boutons ( Kurdyak et al., 1994 ). Given that fly NMJs are glutamatergic in nature, the similarities to mammalian CNS synapses provide us with a tool to study pathogenic mechanisms associated with neurological diseases, such as amyotrophic lateral sclerosis (ALS), spinal muscular atrophy and certain encephalopathies ( Nahm et al., 2010 ; Pennetta et al., 2002 ; Sandoval et al., 2014 ; Sherwood et al., 2004 ; West et al., 2015 ). Collectively, these examples portray the instructive role of fly NMJ studies in identifying the pathogenic mechanisms of neuronal disorders.Giant-fiber-system recordings\nAlthough NMJ electrophysiological assays in flies are very useful to study synaptic development and transmission, they do not enable the analysis of neuronal function over time. Adult-specific assays are more appropriate to document gradual changes that occur in neuronal function with age, and are thus particularly useful for the study of neurodegenerative disorders. The giant fiber system (GFS) is one of the few neuronal circuits in adult flies that is amenable to electrophysiological recordings ( Tanouye and Wyman, 1980 ). When flies detect a change in luminescence, the giant-fiber neurons in the brain signal to the thoracic ganglion to activate flight muscles [dorsal longitudinal muscles (DLMs)] and jump muscles [tergotrochanteral muscles (TTMs)] ( King and Wyman, 1980 ). Electrophysiological recordings of the GFS can be made by stimulating the eyes while recording the depolarization in DLMs and TTMs ( Tanouye and Wyman, 1980 ) (see poster).The GFS assay has been extensively used to study features of epilepsy and seizures by high-frequency stimulation in the eye and detection of seizure-like electrical activity in the muscles. For example, gain-of-function mutations in the fly gene paralytic ( para bss ) ( Parker et al., 2011 ; Pavlidis and Tanouye, 1995 ) and loss-of-function alleles of easily shocked ( Pavlidis et al., 1994 ) have a much lower threshold voltage of high-frequency stimulation to evoke seizure-like activity ( Pavlidis and Tanouye, 1995 ). para encodes a voltage-gated sodium channel ( Feng et al., 1995 ), and its human homologs are either associated with or shown to cause diseases such as encephalomyopathy, neuropathy and myoclonic epilepsy ( Escayg et al., 2000 ; Meisler and Kearney, 2005 ). Recently, GFS assays have been used to document the demise of neurons in neurodegenerative studies and provide insight into underlying pathologies ( Kerr et al., 2011 ; Lee et al., 2004 ; Watson et al., 2008 ; Dutta et al., 2016 ). For example, mutations in superoxide dismutase 1 (SOD1) in humans cause ALS ( Rosen, 1993 ). Flies that overexpress wild-type human SOD1 are unable to follow high-frequency stimulation in DLMs but exhibit a normal TTM response, indicative of interference of the GFS (see poster). Flies that overexpress human mutant SOD1 have defective TTM and DLM responses, which worsen with age, consistent with age-dependent degeneration in ALS ( Watson et al., 2008 )., 2008 ). This fly ALS model demonstrates the convenience of the GFS in neurodegeneration studies because it detects the progressive decline in motor function through electrophysiology.Wing injury assayThe recently developed Drosophila wing injury assay is an elegant approach to study axonal degeneration and regeneration in vivo ( Fang et al., 2012 ). The goal of these studies is to identify genes that are required for axonal degeneration and regeneration, and to identify the regulatory processes that are involved in spinal cord and nerve injuries. The fly peripheral nervous system has been intensively studied, which provides a great platform for this purpose. In the fly wing, mechanosensory and chemosensory neurons reside in the wing margin and project their axons toward the thoracic ganglion. Severing these axons using scissors or lasers causes degeneration of the distal portions of the axons (see poster) ( Fang et al., 2012 ). After ∼7 days, the proximal portion of the injured axons regenerates by extending sprouts toward the lesion site. Alternatively, the axons regrow but invade another wing vein ( Soares et al., 2014 ). Alternatively, the axons regrow but invade another wing vein ( Soares et al., 2014 ). These processes can be visualized by expressing a cytoplasmic green fluorescent protein (GFP) marker under the control of a neuronal GAL4 driver. This simple assay is suitable for a large-scale forward genetic screen of viable as well as lethal mutations to identify genes that have not previously been implicated in degeneration or regeneration processes, because many of the genes and proteins required for these processes are evolutionarily conserved. By using this assay, a forward genetic screen identified mutations in highwire , a RING domain E3 ubiquitin ligase, that suppresses the degeneration of the distal portion of the axons upon axotomy ( Neukomm et al., 2014 ). Furthermore, another screen based on this wing injury assay led to the discovery that downregulation of JNK signaling promotes axon regeneration ( Soares et al., 2014 )., 2014 ). In summary, this novel technique capitalizes on the well-characterized fly peripheral nervous system and can provide clues about the molecular mechanisms that underlie the degeneration and regeneration of neurons.Cardiovascular assaysFly models of cardiovascular diseases first emerged 20 years ago together with the development of assays to measure heart development and function ( Bodmer, 1993 ; Ocorr et al., 2014 ). The Drosophila heart, called the dorsal vessel, differs from the human heart in that it is an open circulatory system consisting of a hollow, muscular tube closed at the posterior end. The vessel runs longitudinally from the posterior abdomen (heart proper) into the thorax (aorta). Similar to the human heart, which consists of distinct chambers, the fly heart is also divided into four chambers that are separated by small valve-like openings through which blood, or rather the analogous fluid in insects, hemolymph, enters the heart ( Lehmacher et al., 2012 ). Each chamber consists of six myocardial cells to facilitate the flow of hemolymph through the dorsal vessel. The aorta, which is made up of myocardial cells that do not contract very much, is a tube that facilitates the transport of hemolymph to the head, from where it flows into the body cavity ( Bier and Bodmer, 2004 ). The molecular pathways underlying the development of the fly heart and its function have provided valuable information relevant to human heart development and physiology. For example, tinman ( Nkx2 -5 in humans), a homeobox transcription factor identified in flies, is required for heart development ( Bodmer, 1993 ). Mutations in the human homolog of this gene were later shown to cause congenital heart disease and have subsequently been shown to cause sudden cardiac arrest in middle age ( Schott et al., 1998 ). Moreover, the discovery of pannier (GATA4) and neuromancer (Tbx20) transcription factors revealed a conserved cardiogenic network, which enabled the study of these factors in human heart development and function (reviewed in Qian and Bodmer, 2012 ). The fly heart proves to be a convenient invertebrate heart disease model owing to conserved molecular pathways and the variety of assays to study different aspects of heart disease. In the following section, we will discuss a few of the cardiovascular assays and how they provide mechanistic insight about human heart disease. We refer readers to excellent reviews on different aspects of the topic for further information ( Diop and Bodmer, 2015 ; Choma et al., 2011 ; Ocorr et al., 2014 ; Wolf and Rockman, 2011 ).Heartbeat measurement\nSimilar to mammals, the fly heartbeat consists of a cardiac cycle that includes diastolic and systolic periods. Interestingly, the cardiac cycle in adult flies is composed of alternating anterograde and retrograde beats ( Wasserthal, 2007 ), leading to a periodic change in the flow of hemolymph ( Dulcis and Levine, 2005 ). Measuring the heartbeat rate and rhythmicity is one of the fundamental assays to determine heart function. Measurements of the heartbeat, either in the dissected dorsal vessel of the larva or the adult abdomen, are based on visual recordings. These methods rely on the optical intensity of light passing through the heart while it beats ( Gu and Singh, 1995 ). The approaches are relatively fast and can be used to assess the function of individual genes while also being appropriate for forward genetic screens.Nowadays, further insight into cardiac function can be obtained by combining the standard heartbeat measurement assay with the use of a high-resolution camera and computer algorithms to detect optical intensity changes. This improvement of the method, semi-automated optical heartbeat analysis (SOHA), allows simultaneous assessment of systole, diastole and rhythmicity ( Fink et al., 2009 ; Ocorr et al., 2009 ). This assay has been instructive in the study of various disease models, including channelopathies ( Ocorr et al., 2007b ), cardiomyopathies, age-dependent heart defects ( Gill et al., 2015 ) and heart dysfunction associated with non-cardiac conditions, such as myotonic dystrophy ( Chakraborty et al., 2015 ). Young wild-type flies monitored with SOHA show rhythmic cardiac contraction that gives rise to a characteristic M (motion)-mode trace. As these animals age, the contraction becomes arrhythmic ( Ocorr et al. As these animals age, the contraction becomes arrhythmic ( Ocorr et al., 2007a ), analogous to cardiac arrhythmias observed in elderly humans ( Jones, 2006 ). Apart from aging, the genetic cardiac fly models also recapitulate human cardiac disorders and allow in-depth analysis of genes involved in cardiovascular disease. For instance, dominant mutations in the human gene alpha-B crystallin ( CryAB R120G ), which encodes a chaperone, cause defects that underlie a range of diseases, including cardiomyopathies. Similarly, overexpression of the human CryAB R120G mutant transgene in the fly results in dilation of the dorsal vessel (see poster), mimicking dilated cardiomyopathy in humans ( Xie et al., 2013 ).Optical coherence tomography (OCT)The heartbeat assays described above all require the dissection of a semi-intact fly heart. More sophisticated tools such as optical coherence tomography (OCT), the equivalent of echocardiography in humans, permit the non-invasive characterization of the Drosophila heartbeat in vivo ( Wolf et al., 2006 ). OCT uses laser beams to scan the entire tissue and subsequently uses the scattered light to produce subsurface images (see poster). This technique allows imaging of the heart in awake adult flies. This method is also convenient to produce M-mode images that reveal subtle changes in cardiac movement, similar to other video-based techniques. Yet, it should be kept in mind that, because this method requires downstream processing, it is relatively slow and is not very convenient to measure heart rhythmicity. Although it is costly and specialized, once established, it can be used for screening, however. Although it is costly and specialized, once established, it can be used for screening, however. Indeed, a deletion screen designed to identify cardiomyopathy genes via OCT led to the discovery that a component of Notch signaling, weary (wry) , could play a role in dilated cardiomyopathy ( Kim et al., 2010 ). The protein encoded by wry has non-canonical Notch motifs, and overexpression of Serrate, a ligand of Notch, can rescue the cardiomyopathy in wry mutants. The results of this study indicate a potential role of Notch signaling in the adult heart and suggest that it can be a therapeutic target for dilated cardiomyopathies. In humans, Notch signaling is important for the organogenesis of the heart ( de la Pompa and Epstein, 2012 ) and mutations in NOTCH1 cause aortic valve disease ( Garg et al., 2005 )., 2005 ). Although the importance of Notch signaling in mammalian heart development has been studied, its role in the adult heart, especially in a regenerative context, is an emerging area that has many unexplored questions ( Ferrari and Rizzo, 2014 ).Ca 2+ measurement\nNormal cardiac physiology requires controlled Ca 2+ handling for proper contraction of cardiomyocytes, in flies as well as in humans. Defects in cardiac Ca 2+ homeostasis are observed in cardiomyopathies and heart failures ( Guo et al., 2006 ). Intracellular Ca 2+ levels can be quantified using genetically encoded Ca 2+ indicators (GECIs), such as GCaMP (a fusion protein of green fluorescent protein, calmodulin and a peptide sequence from myosin light chain kinase). These Ca 2+ sensors have been successfully used to assess cardiac Ca 2+ pulses by expressing them specifically in the fly heart (see poster) ( Lin et al., 2011 ). Dilated cardiomyopathies are associated with altered Ca 2+ homeostasis. For example, held-up mutant flies ( Hdp 2 ) contain a point mutation in troponin, a protein required for proper cardiac muscle contraction, and have enlarged diastolic chambers ( Wolf et al., 2006 ).h dp 2 mutants display a prolonged duration of the Ca 2+ peak intensity (see poster), indicating that the late cytosolic Ca 2+ rise is delayed. This delay, along with protracted Ca 2+ indicator fluorescence, is speculated to be the result of impaired Ca 2+ reuptake into the sarcoplasmic reticulum ( Lin et al., 2011 ). Moreover, Ca 2+ measurements are also useful in the study of age-related cardiac defects, because aging flies display a decrease in the maximal rate of Ca 2+ fluorescence decay, indicating that spontaneous cardiac frequency is reduced ( Santalla et al., 2014 ). Older flies have variable spontaneous cardiac frequencies, which indicates an arrhythmia in the aging heart. As mentioned previously, elderly humans also demonstrate cardiac arrhythmias. Hence, the result of this study proposes that alterations in Ca 2+ signaling might be related to arrhythmias observed in the elderly.\nField potential and intracellular recordingHeart rhythmicity is maintained through electrical conduction. The assays described above to monitor fly heart rhythm do not directly measure electric conduction. For this purpose, electrophysiological methods have been developed to record heart field potential directly, and these approaches can be applied in both larval and adult fly hearts ( Papaefthimiou and Theophilidis, 2001 ). Spontaneous heart field potentials can be measured via a thin glass electrode, which is placed in contact with the semi-intact fly heart (see poster) ( Cooper et al., 2009 ). In larvae, the heart is loosely attached to the body wall, and a floating electrode technique can therefore be used to minimize damage ( Lalevée et al., 2006 ). By contrast, the adult heart adheres more tightly to the body wall, and hence recordings are performed from the myocardium of the heart chamber ( Dulcis and Levine, 2005 ) by stimulating a local glutamatergic input. The action potential recorded from both the larval and adult hearts display a pacemaker potential that is a feature of the myogenic heart. Although these electrophysiological methods are not very convenient for high-throughput screening, they are particularly useful to understand the mechanistic properties of the heart. For example, KCNQ1 (a gene that encodes a voltage-gated potassium channel) mutant flies display cardiac arrhythmia that worsens with age ( Ocorr et al., 2007b ). KCNQ1 is partially responsible for the rhythmic contraction of the heart muscle. The electrophysiological recordings from the KCNQ1 mutant fly hearts show reduced repolarization ability, most probably due to a decrease in the repolarizing K +current (see poster). Interestingly, in humans, mutations in KCNQ1 cause many cardiac disorders, such as familial atrial fibrillation, which is characterized by uncoordinated cardiac electrical activity, and long-QT syndrome, which is also characterized by rapid heartbeats ( Bellocq et al., 2004 ; Johnson et al., 2008 ). These studies collectively support the use of Drosophila as an effective model to study heart disease.\nOenocyte and fat body assaysLiver disease causes millions of deaths per year worldwide ( Byass, 2014 ). Nonalcoholic fatty liver disease (NAFLD) is the most common form of liver disease, affecting 75-million to 100-million individuals in the USA ( Rinella and Sanyal, 2015 ). Because the burden of this disease is large and costly, the elucidation of pathogenic mechanisms underlying liver disease, using model organisms, is a key healthcare priority. In humans, the liver has many metabolic functions, including detoxification of metabolites, protein synthesis, synthesis of digestive metabolites and maintenance of blood glucose levels. These functions are performed by highly specialized cells named hepatocytes. To regulate fat usage during starvation, adipocytes – the body's major fat-storing cells – break down lipids into fatty acids (FAs) via adipocyte triglyceride lipase (ATGL) ( Zechner et al., 2005 ). The FAs are secreted in the bloodstream, taken up by the liver and processed via hepatocytes. The FAs are secreted in the bloodstream, taken up by the liver and processed via hepatocytes. During prolonged starvation, hepatocytes synthesize water-soluble ketone bodies from the FAs, and these are released into the bloodstream to be used as an energy source for other tissues ( Green et al., 2015 ).Fasting fly larvae similarly release lipids from the fat body – the organ responsible for energy storage and utilization – and these lipids are taken up by specialized cells named oenocytes ( Chatterjee et al., 2014 ). Until recently, the fly fat body was thought to be the functional homolog of the human liver ( Baker and Thummel, 2007 ). However, studies have shown that fly oenocytes are more similar to hepatocytes than is the fat body, based on their response to starvation. Furthermore, oenocytes express 22 homologs of human fat-metabolizing genes expressed in hepatocytes, and also express genes involved in hepatocyte differentiation, including hepatocyte nuclear factor 4-a ( Hnf4-a ) and COUP-transcription factor ( COUP-TF ) ( Gutierrez et al., 2007 ). Finally, oenocyte-specific knockdown of acetyl-coenzyme A-carboxylase (ACC), a rate-limiting enzyme in FA synthesis, results in lethality, demonstrating the importance of oenocytes for FA synthesis ( Parvy et al., 2012 )., 2012 ). Overall, recent data demonstrate that the fat body and oenocytes in flies are the functional homologs of the vertebrate liver (see poster). In the next section, we will review the fundamental fat body and oenocyte assays to model liver diseases in flies.Drosophila has, in our opinion, not been exploited to its full potential as a model system in liver disease research, and there is scope for the development of new assays and the improvement of existing ones.\nLipid-droplet accumulation\nThe assays that are used to study fat body and oenocyte function in flies typically depend on visualizing lipid storage in response to differential nutritional conditions. Generally, dyes such as Oil Red-O ( Gutierrez et al., 2007 ) or BODIPY ( Kohyama-Koganeya et al., 2008 ) are used to visualize lipids in oenocytes and the fat body of the fly. An in vivo assay to determine the presence of lipids is based on stimulated Raman scattering (SRS) microscopy or coherent anti-Stokes Raman scattering (CARS) microscopy ( Chien et al., 2012 ). These spectroscopic assays detect the vibrational signature of molecules and allows labeling-free, live imaging of lipids (see poster).Mutations in human HNF4A are associated with a type of inherited diabetes known as maturity-onset diabetes of the young, type 1 (MODY1) ( Yamagata et al., 1996 ). These mutations lead to decreased serum triglyceride (TAG) levels, and the onset of diabetes, in affected individuals ( Fajans et al., 2001 ). The functional homolog of HNF4A in flies is Hnf4 . Hnf4 -null flies are very sensitive to starvation because they are unable to harness energy from stored lipids (see poster) ( Palanker et al., 2009 ). Interestingly, many enzymes involved in lipid catabolism are upregulated in starved\nH nf 4 -null animals. Investigation of these mutant flies has provided supporting evidence that a specific FA activates the nuclear hormone receptor, which in turn stimulates energy production by activating FA oxidation.In a forward genetics study, a genome-wide RNAi screen performed in fly oenocytes identified multiple obesity-related genes that are also associated with obesity in mice ( Pospisilik et al., 2010 ). One of the candidates from this screen is the fly homolog of fatty-acid elongase ( ELOVL6 ), baldspot . Interestingly, mouse mutants of Elovl6 also develop obesity and hepatosteosis ( Matsuzaka et al., 2007 ). To date, mutations in human\nE LOVL 6 have not been linked with obesity or liver disease, however. Another obesity model in flies is the fat-body-specific overexpression ofL ipid storage droplet 2 ( Lsd-2 ) ( Grönke et al., 2003 ). These flies are obese, resistant to starvation and have elevated TAG storage. Furthermore, CARS microscopy revealed that starved flies that overexpress Lsd-2 in the fat body have reduced lipids in oenocytes, showing that they are unable to store lipids in oenocytes as a starvation response (see poster) ( Chien et al., 2012 ). The functional homolog of Lsd-2 in humans is perilipin 2 ( Plin2 ) ( Rajan and Perrimon, 2013 ). Perilipins coat intracellular lipid droplets and are involved in lipolysis. Interestingly, the liver biopsies from nonalcoholic steatohepatitis patients show increased levels of PLIN2 in lipid droplets, indicating the importance ofPlin2 for individuals with NAFLD ( Fujii et al., 2009 ). In addition to genetic risk factors, environmental factors that result in liver disease can also be studied in flies. Obesity and type 2 diabetes are the major risk factors for NAFLD ( Neuschwander-Tetri et al., 2010 ). Fly larvae that are fed with a high-sugar diet mimic the hallmarks of both obesity and type 2 diabetes ( Owusu-Ansah and Perrimon, 2014 ). The high-sugar-fed larvae have increased body fat along with the accumulation of large lipid droplets in the fat bodies ( Musselman et al., 2011 ). The transcriptional profile of these larvae demonstrates an increase in the expression of genes related to lipid catabolism. Of note, the transcriptional profile of these larvae is similar toH nf 4 -null flies, indicating a conserved transcriptional signature that might be instrumental in diabetes and liver disease. In summary, these results provide compelling evidence that the fly can help provide a valuable mechanistic understanding of processes that cause liver diseases.\nNephrocyte and Malpighian tubule assaysThe function of the human excretory system is to eliminate metabolic waste and maintain a homeostatic ion balance. The nephron is the basic structural and functional unit of the human kidney. It is composed of a glomerulus, glomerular capsule and renal tube. The glomerular podocyte is an epithelial cell that wraps around capillaries in the glomerulus and plays an important role in the filtration of blood to produce urine. Glomerular podocytes create the filtration barrier by sending out interdigitating processes that are separated by 30- to 50-nm-wide slit pores, called the slit diaphragm (see poster) ( Wartiovaara et al., 2004 ). Blood is filtered through these slit diaphragms, and mutations in several genes that disrupt the filtration barrier lead to kidney failure ( Kestilä et al., 1998 ; Patrakka et al., 2000 ). Although invertebrates lack nephrons, Drosophila nephrocytes share remarkable similarity to podocytes. There are two different nephrocytes in Drosophila : the pericardial nephrocytes, which flank the fly aorta, and garland nephrocytes, which form a ring around the proventriculus ( Denholm et al., 2013 ). These nephrocytes exhibit extensive folds of the plasma membrane and create ∼30-nm slit pores ( Weavers et al., 2009 ), forming the nephrocyte diaphragm, which shares functional and molecular similarities with the slit diaphragm of the mammalian podocyte (see poster). Indeed, similar to the",
         "It facilitates the transport of hemolymph to the head.",
         null,
         null,
         null
        ],
        [
         "24",
         "What were the systolic and diastolic blood pressure criteria for preeclamptic women in this study?",
         "Karen C. Wheeler et al.,2018, DOI:https://doi.org/10.1371/journal.pone.0191040:\n Human subjectsTissue samples were collected after obtaining informed consent under protocols approved by the Wayne State University and Stanford University IRBs on the use of Human Subjects in Decidual VEGF and M2 macrophage polarization PLOS ONE | https://doi.org/10.1371/journal.pone.0191040 January 11, 2018 2 / 18 Competing interests : The authors have declared that no competing interests exist. Medical Research for our previous studies [22, 30, 31]. Briefly, secretory phase endometrium was obtained from subjects undergoing hysterectomy for benign indications, such as fibroids or chronic pelvic pain, in the age group between 28 and 39 years during the secretory phase of the cycle, as described previously [31]. Determination of the stage of the menstrual cycle was based on the day of last menstruation and histological evaluation by pathologists [31]. Fullthickness endometrial OCT blocks (2-3 mm) and cryosections (10μm) were prepared for in situ hybridization studies and RNA extracted for qPCR analysis as described below (n = 5). Decidual samples from early pregnancy (5-9 weeks) were collected from subjects undergoing elective termination of pregnancy following the same procedures as described previously [30, 32]. After dilation and curettage, deciduas were separated from chorionic villi and other fetal tissues under a dissecting microscope and processed for RNA extraction and preparation of OCT blocks. Decidual samples from term pregnancies were used from a cohort of 19 preeclamptic and 24 gestational age-matched controls, and 4 samples per group were used for this study. Decidual tissues were carefully separated from attached placentas or placental bed biopsies and used for RNA extraction as described below [33]. Details of clinical evaluations have been reported previously [33]. Details of clinical evaluations have been reported previously [33]. Preeclamptic women had systolic blood pressures of \u0015 140 mm Hg and diastolic blood pressures of \u0015 90 mm Hg more than once, with proteinuria of \u0015 300 mg/24h [33]. All endometrial/decidual samples were stored at -80  ̊C and processed for RNA extraction, in situ hybridization, and immunohistochemistry as described below.In situ hybridization\nIn situ hybridization (ISH) for detection of human VEGF mRNA expression was performed in frozen endometrial/decidual sections following the same procedures and probes as described previously [22, 31, 34]. 35 S-UTP-labeled (PerkinElmer, Hopkinton, MA) sense and antisense riboprobes from VEGF cDNA templates were prepared with the MAXIscript in vitro transcription kit from Applied Biosystems (Waltham, MA), and purified using mini Quick Spin TM RNA Columns (Roche, Indianapolis, IN). Sections were fixed in 4% paraformaldehyde and hybridized with 5x10 6 cpm/ml S 35 -labeled sense and antisense probes. Sections treated with RNase-A or treated with the sense probe were used as negative controls. Hybrids resistant to RNase-A were detected by autoradiography, and sections were post-stained with hematoxylin. Images were taken on a Zeiss Axioskop 2 microscope equipped with a Zeiss AxioCam camera (Carl Zeiss).\nImmunohistochemistryImmunohistochemistry\nImmunohistochemistry (IHC) was performed using 10 μm OCT-embedded frozen tissue sections fixed with 4% paraformaldehyde following our previously published methods [22, 35]. After washing and incubation with horse blocking serum, the sections were incubated with a mouse monoclonal primary antibody to CD68 (1:200, Agilent, Santa Clara, CA) overnight at 4 ̊C, followed by blocking and incubation with a biotinylated anti-mouse secondary antibody (Vector Laboratories, Burlingame, CA) at room temperature. The sections were visualized by immunofluorescence imaging using fluorescein-labeled avidin and Vectashield mounting medium with DAPI (Vector Laboratories).\nRNA extraction and quantitative real time PCR (qPCR)Total RNA extraction and qPCR were performed following the same methods as previously published [35]. Total RNA from tissues/cells was isolated using TRIZOL (Invitrogen, Carlsbad, CA), treated with DNase (Roche), and purified using RNeasy Spin Columns (Roche). Two micrograms of RNA were used for cDNA synthesis using a kit from Promega (Madison, WI). Decidual VEGF and M2 macrophage polarization PLOS ONE | https://doi.org/10.1371/journal.pone.0191040 January 11, 2018 3 / 18  Real-time PCR was performed with the 7500 Fast instrument (Applied Biosystems) using SYBR Green PCR master mix (Applied Biosystems) and the respective primer pairs for each gene (sequences presented in Table 1). All experiments were performed in triplicate, and data were normalized against beta-actin. Water was used as a negative control, and the size and sequence of each PCR product were used for validation of the results. The relative expression of mRNAs for various genes is presented as fold change using the 2 ^ (-ΔΔCT) method as previously described [36].THP-1 cell cultureHuman monocytic THP-1 cells (ATCC, Rockville, MD) were cultured in RPMI-1640 medium (Life Technologies, Carlsbad, CA) supplemented with 10% FBS, 2mM L-Glutamine and 1% antibiotics (penicillin, streptomycin and amphotericin-B) (Sigma-Aldrich, St. Louis, MO). Differentiation of THP-1 cells into M0, M1, and M2 macrophages was performed as previously described [37]. Briefly, THP-1 cells were differentiated into M0 macrophages by incubation with 100 nM phorbol 12-myristate 13-acetate (PMA) for 48 h (Sigma-Aldrich). Once the cells were adherent, they were transferred to PMA-free media to obtain resting macrophages (M0). These cells were then polarized to M1 macrophages by incubation with LPS (100 ng/ml) and IFN-γ (20 ng/ml) for 48 h and M2 macrophages by incubation with IL-4 (100ng/ml) for 48 h. In a subset of experiments, to determine the effective dose of VEGF, M1 cells were treated with different doses of rhVEGF (0.1-20 μg/mL) (R&D systems, Minneapolis, MN).1-20 μg/mL) (R&D systems, Minneapolis, MN). Based on the results of this experiment, the cells were treated with/without 4 μg/mL of rhVEGF (most effective dose) for subsequent experiments.In vitro decidualization of endometrial stromal cellsPrimary endometrial stromal cells were isolated from endometrial tissues obtained from normally cycling women undergoing hysterectomy for benign conditions as indicated above and described previously [38-40]. Briefly, endometrial tissues were digested with collagenase, and stromal cells were separated from the epithelium and cultured in DMEM (Gibco, Gaithersburg, MD) supplemented with 5 μg/ml insulin (Sigma, St. Louis, MO) and 10% heatTable 1. Primers for quantitative real-time PCR. Louis, MO) and 10% heatTable 1. Primers for quantitative real-time PCR. Gene Primer M1 CXCL10 F 5’-GAAAG CAGTTAGC AAGGAAAGG TC-3’ R 5’-ATGTA GGGAAGTG ATGGGAGAG G-3’ CCR7 F 5’-TGGTG GTGGCTCT CCTTGTC-3 ’ R 5’-TGTGG TGTTGTCT CCGATGTAA TC-3’ IL-12 F 5’-AAAGG ACATCTGC GAGGAAAGT TC-3’ R 5’-CGAGG TGAGGTGC GTTTATGC- 3’ M2 CD206 F 5’-ACCTC ACAAGTAT CCACACCAT C-3’ R 5’-CTTTC ATCACCAC ACAATCCTC -3’ CD163 F 5’-GTCGC TCATCCCG TCAGTCATC -3’ R 5’-GCCGC TGTCTCTG TCTTCGC-3 ’ CCL-17 F 5’-GAGCC ATTCCCCT TAGAAAG-3 ’ R 5’-AGGCT TCAAGACC TCTCAAG-3 ’ VEGFR1 F 5’ -TGGCAG CGAGAAAC ATTCTTTTA T-3’ R 5’-CAGCA ATACTCCG TAAGACCAC AC-3’ VEGFR2 F 5’-CTCTT GGCCGTGG TGCCTTTG- 3’ R 5’-GTGTG TTGCTCCT TCTTTCAAC -3’ β-Actin F 5’-ATTGC CGACAGGA TGCAGAA-3 ’ R 5’-GCTGA TCCACATC TGCTGGAA- 3’ https://doi.org/10.1371/journal.pone.0191040.t001 Decidual VEGF and M2 macrophage polarization PLOS ONE | https://doi.org/10.1371/journal.pone.0191040 January 11, 2018 4 / 18  inactivated charcoal-stripped fetal bovine serum (FBS, Gibco).pone.0191040 January 11, 2018 4 / 18  inactivated charcoal-stripped fetal bovine serum (FBS, Gibco). At passages 3-4, the cells were grown to confluence and then decidualized with 10 nm estradiol, 1 μm progesterone, and 20 ng/ml epidermal growth factor in serum-free and insulin-free decidualization medium (75% DMEM, 25% MCDB-105, 50 μg/ml ascorbic acid, 1 mg/ml BSA, 5 μg/ml transferrin) (Sigma) [40]. Decidualization occurred after 7-14 days in culture and was confirmed by documenting increased secretion of insulin-like growth factor-binding protein-1 into the medium, as described previously [38, 39]. Cells treated with decidualization medium only (without steroid hormones) were used as controls. RNA and conditioned media from both decidualized and non-decidualized cells were collected. The conditioned media were concentrated ten-fold using a Centricon filter (10 kD MWCO, Millipore, Billerica, MA) and used in subsequent experiments with or without the VEGF inhibitor sFlt1 (1.5 μg/ml, R&D systems) to examine the role of VEGF secreted into the conditioned media on differentiation and migration of macrophages.In vitro cell migration assayIn vitro cell migration assays were performed in 24-well plates using transwell inserts with 8 μm pores (Corning, Corning, NY) following the procedures described previously [41]. Briefly, M0 cells were seeded (5 X 10 4 cells/insert) onto the upper well of the chamber, and the inserts were placed in 24-well cell culture plates (Corning) containing 500 μL of RPMI-1640 medium with or without 2 μg/ml rhVEGF, and incubated at 37 ̊C in 5% CO2 for 16 hours. In another set of experiments, the cells were incubated in RPMI-1640 medium with concentrated supernatants from decidualized endometrial stromal cells and with or without sFlt1 (1.5 μg/ml; R&D systems), as described above. Concentrated supernatants from nondecidualized stromal cells were used as controls. The cells in the upper chambers were removed carefully with a wet cotton swab. The cells in the upper chambers were removed carefully with a wet cotton swab. The migrated cells on the bottom surface were stained with Diff-Quick Hema3 stain following the manufacturer’s instructions (Fisher Scientific, Kalamazoo, MI), and six evenly spaced fields of cells were counted in each well using a Leica DMIRES2 microscope.Statistical analysis\nComparisons were made using an analysis of variance (ANOVA) with Tukey post-test or Students’ t-test, as appropriate. A p-value of equal to or less than 0.05 was considered statistically significant. Statistical analyses were performed using GraphPad Prism (GraphPad Software, Inc.; La Jolla, CA, USA) or Microsoft Excel (Microsoft Corporation; Redmond, WA, USA), as described previously.",
         "Preeclamptic women had systolic blood pressures of \u0015 140 mm Hg and diastolic blood pressures of \u0015 90 mm Hg more than once.",
         null,
         null,
         null
        ],
        [
         "25",
         "What kit was used for total RNA sequencing?",
         "Niyati Desai et al.,2020, DOI:https://doi.org/10.1038/s41467-020-20139-7:\n Immunohistochemistry and RNA-ISH. Analysis of patient autopsy material was reviewed and approved by the Partners Human Research IRB (Protocol #: 2020P001001). Autopsy consent was per clinical care as directed by the patient or health care proxy. Given this is not human subject research and a discarded tissue protocol, no additional research consent was required. We evaluated hematoxylin and eosin (H&E) stained sections from FFPE tissue from the lungs, heart, liver, intestine, bone marrow, adipose tissue, and kidney. We performed immunohistochemistry (IHC) for CD3, CD8, CD20, CD163, CD123, IDO1, PD-L1, Napsin A, keratin, and SARS-CoV N protein on an immunohistochemical platform. RNAISH was performed on FFPE sections using a SARS-CoV-2 RNA specific probe on an automated Leica BOND RX (RNAscope 2.5 LS Probe V-nCoV2019-S, #848568, and RNAscope 2.5 LS Reagent Kit-RED, #322150; Advanced Cell Diagnostics).5 LS Reagent Kit-RED, #322150; Advanced Cell Diagnostics). Slides were imaged using a Leica Aperio CS-O slide scanning microscope at 40x magnification. Image quantification was performed using Halo software (Indica Labs). Tissue regions of interest were annotated by hand, excluding any folds or debris. The Multiplex IHC module was used to calculate the number of positive cells per square millimeter of tissue. Areas of pulmonary parenchyma with positive SARS-CoV-2 RNA ISH signal were annotated manually. The percentage of ISH signal was calculated as follows: area of lung parenchyma with positive signal/area of total lung parenchyma. All the IHC staining was performed on Leica Biosystem Bond III with Leica Bond Polymer Refine Detection (Catalog number DS9800) Supplementary Table 9. Molecular RNA analysis. We performed total RNA-seq and quantitative reverse transcriptase-polymerase chain reaction (qRT-PCR). RNA extraction from FFPE slides was done using the FormaPure Total nucleic acid extraction kit (C16675, Beckman Coulter) according to manufacturer instructions. Three 5μm thin tissue sections from areas devoid of acute inflammation were used per sample. SARSCoV-2 RNA was detected in extracts from FFPE samples with qRT-PCR by following the guidelines of the Center for Disease Control and Prevention for the qRT-PCR diagnostic panel. A total of 1 ng of RNA input was used per reaction. 1step qRT-PCR was performed with the GoTaq Probe 1-Step RT-qPCR kit (A6120, Promega) using the CDC-approved 2019-nCoV RUO primer-probe kit (10006713, IDT). Supplementary Table 10. For Total RNA-sequencing, The Smarter Stranded Total RNA-Seq kit v2 (634413, Takara) was used with 10 ng RNA input, according to the manufacturer instructions to generate libraries. Dual-indexed pooled libraries were sequenced on the Illumina NextSeq 500 platform using a 150 cycles kit with paired end read mode. For RNA expression analysis, the initial quality control of sequencing data were carried out using the tool FASTQC and alignment of sequencing reads to the reference genome was carried out using STAR aligner, version 2.7 47. We used the genome annotation and GTF for SARS-CoV-2 available on NCBI. A joint annotation was created by adding the COVID19 genome to the HG38 genome sequence and the GTF sequence. A new index for STAR aligner was created using this new annotation. Post alignment using this new annotation, the duplicate reads were marked using PICARD and removed using SAMtools. The resulting BAM files were used to quantify the read counts per gene using HTSeqcount program. The downstream analysis was carried out in the R statistical programming language including hierarchical clustering. The DESeq2 package 48 was used for differential expression analysis between samples. The DESeq2 package 48 was used for differential expression analysis between samples. Cell type deconvolution from gene expression was performed using CIBERSORTx 49 using the LM22 signature matrix and batch correction for bulk sorted reference profiles. Plots were made using the heatmap.2 function in the gplots package in R. GeoMx DSP for protein profiling. Autopsy tissues from COVID-19 infected patents were processed following the GeoMx DSP slide prep user manual (MAN10087-04). Autopsy FFPE slides were baked in oven at 60 °C for at least 1 h, and then rehydrated and blocked by Nanostring block buffer for 1 h. CD68-594 (Novus Bio, NBP2-34736AF647), CD45-647 (Novus Bio, NBP2-34527AF647), and PanCK-488 (eBioscience, 53-9003-82) were added on the sections along with the Nanostring protein cocktail for overnight incubation Supplementary Table 11. The slides were washed and stained with Syto83 (ThermoFisher, S11364) on the next day. The slides were washed and stained with Syto83 (ThermoFisher, S11364) on the next day. 20X fluorescent images were scanned after loading the slides to GeoMx machine. Regions of interest (ROIs) in Alveoli were selected in both COVID19 high and COVID19 low regions based on the COVID-19 ISH staining in the serial section. Oligos from antibodies were cleaved and collected into 96-well plates. Then these oligos were hybridized with NanoString barcodes overnight and read with an nCounter machine. Digital accounts of each antibody in each ROI were generated for data analysis. GeoMx DSP for CTA profiling. Autopsy FFPE tissues from COVID-19 infected patents were processed following the GeoMx DSP slide prep user manual (MAN10087-04). Autopsy slides were baked in oven at 60 °C for at least 1 h, and then deparaffinized and hydrated by Leica Biosystems BOND RX. Proteinase K was added prior to the incubation of incubated with RNA probe mix (CTA and COVID-19 spikein panel). After overnight incubation, slides were washed with buffer and stained with CD68-594 (Novus Bio, NBP2-34736AF647), CD45-647 (Novus Bio, NBP234527AF647), and PanCK-488 (eBioscience, 53-9003-82) and Syto83 (ThermoFisher, S11364) for 1 h, and loaded to the GeoMx DSP machine to scan 20×fluorescent images. Regions of interest (ROIs) were placed by aligning to the ROIs placed during protein profiling. Oligos were cleaved and collected into 96-well plates. Oligos from each AOI was uniquely indexed using Illumina’s i5 × i7 dual-indexing system. 4μLo fa GeoMx DSP sample was used in the PCR reaction. PCR reactions were purified with two rounds of AMPure XP beads (Beckman Coulter) at 1.2× bead-to-sample ratio. Libraries were paired-end sequenced (2 × 75) on a NextSeq550 up to 400 M total aligned reads. Fastqfiles were processed by the NanoString DND pipeline to generate count files for each target probe, and saved as DCCfiles. The NCBI GEO accession number for the DSP experiments is GSE159788. The NCBI GEO accession number for the DSP experiments is GSE159788. Analysis of GeoMx protein data. Per manufacturer’s recommendations, the data were normalized by scaling to the negative control IgG probes, which reflect the rate at which antibodies bind to each region. The Ms IgG2a probe was excluded from this calculation due to poor concordance with the other IgGs. 6 ROIs were removed due to low signal. 10 proteins were excluded from analysis due to lack of above-background signal. For each tissue and each protein, differential expression vs. SARS-CoV-2 presence/absence was evaluated with an unpaired, heteroscedastic t-test of the protein’s log2-transformed normalized data. GeoMx RNA data normalization and background estimation. Probes were collapsed to the geometric mean of for each target after removing outliers using a Grubbs test per gene across AOIs. Each AOI’s data was scaled to have the same 75 th percentile of expression. Each AOI’s data was scaled to have the same 75 th percentile of expression. For each data point (gene × AOI), the expected background was estimated as the geometric mean of the negative control probes from the appropriate probe pool (CTA or COVID-19 spike-in) within the ROI in question. These expected background values were used as an input in mixed cell deconvolution and for background-subtraction in the plot of spatially-resolved expression. Deconvolution of cell proportions from GeoMx RNA data. Cell mixing proportions were estimated using the SpatialDecon R library 50, which performs mixture deconvolution using constrained log-normal regression. The algorithm was run using a cell profile matrix derived from the Human Cell Atlas adult lung 10× dataset and appended with a neutrophil profile derived from snRNA-seq of lung tumors51. The neutrophil profile was scaled to have the same 75th percentile expression value as the average cell type’s profile in the Human Cell Atlas lung dataset. Differential expression analysis of GeoMx RNA data. Only genes that rose 2fold above-background in at least one ROI were considered. For each tissue and each gene, differential expression vs. SARS-CoV-2 presence/absence was evaluated with an unpaired, heteroscedastic t-test of the gene’s log2-transformed normalized data. For analysis of expression within tissues, genes were defined as consistently up-regulated if they 1. had a log2 fold-change >0.2 and a Benjamini-Hochberg FDR < 0.1 in at least 2 tissues, and 2. never had a log2 fold-change <0 and a Benjamini-Hochberg FDR < 0.1 in any tissue. Genes were defined as consistently downregulated by an equivalent rule. For analysis across patient samples, expression was modeled using linear mixed effect models allowing for random slope and intercept terms per patient sample.Pvalues were estimated using Satterthwaite’s method for approximation, and adjusting using the Benjamini-Hochberg FDR. Statistics and reproducibility. All statistics are described as above and otherwise parametric data were analyzed using GraphPad PRISM software (v8) or Microsoft Excel (v16.42). For all RNA-ISH and IHC staining, given the limited resource of these human autopsy samples, these assays were performed on a single slide per specimen. However, multiple specimens from the same patient were analyzed in this work and for each specimen there were multiple orthogonal assays performed to confirm the findings as presented. Reporting summary. Further information on research design is available in the Nature Research Reporting Summary linked to this article. Data availability RNA-seq data have been deposited in NCBI GEO database under the accession code GSE150316. The NanoString GeoMX DSP data have been deposited in the NCBI GEO database under the accession codeGSE159788. All other data are available in the article and its Supplementaryfiles or from the corresponding author upon reasonable request.",
         "The Smarter Stranded Total RNA-Seq kit v2 was used.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "26",
         "Which variant interpretation tools showed the highest specificities on data for ancestry groups?",
         "Abhishek Niroula et al.,2019, DOI:https://doi.org/10.1371/journal.pcbi.1006481:\n Specificity of tolerance predictors To assess the quality of variant pathogenicity/tolerance prediction methods we collected from the ExAC database all variants that had AF 1% and <25%. Because of their high frequency, these variants are usually considered to be neutral and were used in here to assess the specificity of prediction methods. We tested whether 10 widely used methods having different background and design principles showed differences in their performance for benign variants. The predictions for 9 tools were collected from the dbNSFP database [36]. For PON-P2 [22], we run the predictions using the Application Programming Interface. We could not evaluate four tools MetaLR [9], MetaSVM [9], M-CAP [46], and REVEL [11]. MetaLR and MetaSVM are meta predictors, after excluding the training datasets of their constituent tools no variants were left for evaluation. REVEL has been trained with several datasets including Exome Sequencing Project and The 1000 Genomes project that form a substantial part of the ExAC dataset that we used for testing. Thus, analysis of the performance with ExAC data would introduce circularity and not indicate true performance, instead denote how well the methods have learned the training data. M-CAP is aimed for rare variants, therefore predictions for common variants were not available and the method performance could not be assessed. The tools are based on different principles and include those based on evolutionary information only, LRT [39], PROVEAN [43], and SIFT [44], and those combining different types of features, CADD [37], FATHMM [38], MutationAssessor [40], MutationTaster2 [47], PolyPhen2 [42], PON-P2 [48], and VEST [45]. Most of the investigated tools have been trained with known disease-causing and benign variants. The methods that use only sequence conservation information have not been trained. The methods that use only sequence conservation information have not been trained. If variants used for training are used for assessing the methods, the obtained performance measures are likely inflated [20, 26, 49]. Hence, we excluded the training datasets for FATHMM, PON-P2, PolyPhen-2, and VEST. The remaining tools were either not trained or the training datasets were not available. All the tested tools classify variants into pathogenic and benign classes except for CADD and VEST. CADD predicts a continuous phred-like score that ranges from 1 to 99, higher values indicating more deleterious cases. The score for VEST indicates benign when 0 and pathogenic when 1. For CADD we used the highest phred-like score cutoff recommended by the authors i.e. 20. For VEST, we classified the variants into two classes using VEST score cutoff of 0.5.e. 20. For VEST, we classified the variants into two classes using VEST score cutoff of 0.5. To evaluate usability of the CADD and VEST cutoffs, we analyzed the sensitivities and specificities of the tools at different cutoffs which showed that the optimal VEST score cutoff is between 0.45 and 0.5 and phred-like score cutoff is between 20 and 25 (S1A and S1B Fig). The performances of some of these tools have been assessed previously several times, however not with this kind of high-quality and large dataset for benign variants. It is important both in research and clinical practice to be able to sort out variants that have no relevance for the condition under investigation. The specificities of the methods range from 0.63 for SIFT and 0.64 for MutationTaster2 to 0.96 for PON-P2 (Table 1). FATHMM and VEST have the second and third highest performance i.e. 0.86 and 0.84, respectively. FATHMM and VEST have the second and third highest performance i.e. 0.86 and 0.84, respectively. It should be noted that variants are classified into three classes by PON-P2 and two classes by FATHMM, and VEST, and CADD does not group variants into pathogenic and benign categories, instead predicts continuous probabilities. For VEST, we classified the variants into two classes using a cutoff of 0.5. The methods that use evolutionary data only are towards the end of the list (Table 1). Their specificities are 0.724 for LRT, 0.774 for PROVEAN and 0.634 for SIFT. Machine learning methods populate both ends of specificity spectrum. PON-P2, FATHMM and VEST have the highest scores while the specificities for MutationTaster2 and CADD are 0.640 and 0.643, How good are pathoge nicity predicto rs in detecting benign variants? PLOS Computationa l Biology | https:/ /doi.org/10.13 71/journal.p cbi.1006481 February 11, 2019 5 / 17  respectively.org/10.13 71/journal.p cbi.1006481 February 11, 2019 5 / 17  respectively. It is not possible to draw definitive conclusions from the ways methods have been implemented, except saying that machine learning methods can reach much higher specificities in the best installations. To systematically assess the performance of prediction tools, it would be important to include both pathogenic and benign variants. However, since there is no dataset of pathogenic variants that has not been used for training any of the tools, we could not perform a similar analysis for the pathogenic variants. Therefore, we used a small set of pathogenic and likely pathogenic variants from ClinVar to compare sensitivities of the tools side by side with the specificities (S1C Fig). Since we could not filter out variants used for training of all the tools, we did not do this for any of the methods. High sensitivities indicate that the tools with high specificities are not overfitted towards predicting all the variants as benign. Apart from that, we do not recommend to use the sensitivity scores presented here as reliable estimates of performance. S1D Fig shows almost identical results to those in S1C Fig when the ClinVar variants were evaluated together with the variants predicted by all the methods. PON-P2 had the highest proportion of unclassified variants, however with far better specificity compared to the other tools (Fig 1 and Table 1). The end users have to decide what is most relevant for them-large coverage with additional false positives or lower coverage but highly reliable predictions. One percent difference in specificity means >100 false positives more or less per exome, thus the differences accumulate very fast. Table 1. Specificitie s of variant interpretation tools. Table 1. Specificitie s of variant interpretation tools. All variants (n = 63,197) a Variants predicted by all tools (n = 7,268) b Tools VUS c Benign Harmful Specific ity Benign Harmful Specificity PON-P2 d 21,373 34,529 1,626 0.955 6655 613 0.916 VEST d,e 1,168 22,614 4,480 0.835 5984 1284 0.823 FATHMM d 5,531 43,005 6,766 0.864 6287 981 0.865 PROVEAN 3,908 45,868 13,421 0.774 5712 1556 0.786 PPH2 d,f 6,386 37,124 13,602 0.732 5404 1864 0.744 LRT 19,333 31,736 12,128 0.724 5465 1803 0.752 MA 8,044 39,493 15,660 0.716 5306 1962 0.730 CADD g 0 40,659 22,538 0.643 4539 2729 0.625 SIFT 5,099 36,808 21,290 0.634 4868 2400 0.670 MT2 h 15,313 30,632 17,252 0.640 4764 2504 0.655 a All variants having AF> = 1% and <25% in at least one population and not present in the training dataset for the method. After excluding cases in the training datasets, the total number of variants was 57,528 for PON-P2, 28,262 for VEST, 55,302 for FATHMM , and 57,112 for PPH2. b Variants classified as benign or harmful . Variants present in training dataset of any of the tools were excluded. All variants that were automati cally annotated without making predictions were exclude d. c Variants for which the predictions were not available, were ambiguous, or were predicte d to have unknown significance. d Variants present in the training datasets were excluded . e Variants were not classifie d into benign and harmful by the program . A cutoff of 0.5 was used so that variants with score greater than or equal to 0.5 were classified as harmful, otherwise benign. f HumVa r version of PolyPhen-2 was used as the performan ce was higher than for HumDiv version. g Variants were not classified into benign and harmful by the program. A cutoff of 20 was used so that variants with score greater than or equal to 20 were grouped as harmful and otherwi se benign. The authors have recommend ed a cutoff ranging from 10 to 20. The authors have recommend ed a cutoff ranging from 10 to 20. The highest cutoff was used so that the highest possible specificity was obtained. h Variants that were automati cally detected to be harmful or benign were not included in the classifie d cases as they are not real predictions by the tool, instead annotations based on known data. https://do i.org/10.1371/j ournal.pc bi.1006481. t001 How good are pathoge nicity predicto rs in detecting benign variants? PLOS Computationa l Biology | https:/ /doi.org/10.13 71/journal.p cbi.1006481 February 11, 2019 6 / 17  To compare the performance of tools on the same set of variants, we computed the specificities of the tools on variants for which all tools made predictions (Table 1). The scores are somewhat different for all the methods and that is normal for different test datasets. The largest difference is seen for PON-P2, however, it is still the best predictor also on this dataset. The number of variants predicted by all the tools, 7,268, is only 11.5% of the total number of cases. There are various reasons for differences in coverage, some data items may be missing, some sequences are unique for human and may therefore not be evaluated, etc. All the methods have their limitations. Comparison of both the sets in Table 1 shows that the ranking order of the methods remains practically the same. The major differences are that FATHMM has higher specificity than VEST, and CADD has the lowest specificity of all, for the variants that all the tools can predict. The other analyses are reported for all the variants that each method predicted to cover as many variants as possible. Next we investigated whether the differences in specificities could be due to certain types of variations or whether they were due to differences in the methods. To assess the performance of tools for variants with different AFs, we divided the dataset into groups based on adjusted AF on the whole dataset. The predictor performance is higher for variants with higher AFs for all the tools (Fig 1). The specificity differences between the AF bins are the smallest for PON-P2 and FATHMM while several other methods, including CADD, LRT, PolyPhen, SIFT and VEST, had very strong correlation between specificity score and allele frequency. As mentioned above, 1% difference in specificity means a difference of over 100 false classifications in an exome. Since the dataset is so large even a small difference in specificity is statistically significant. Results for Fisher exact tests between the pairs of tools show that the differences are significant for all variants (S2A Fig) as well as for variants predicted by all the tools (S2B Fig). The tools with similar performances have high p-value (low negative logarithm of p-value). The tools with similar performances have high p-value (low negative logarithm of p-value). CADD, SIFT and MT2 form one group where the results are somewhat similar, PolyPhen2, LRT and MutationAssesso r form another group, The rest of the tools have significantly different performances for all variants, VEST, FATHMM and PROVEAN have similar Fig 1. Performan ce of variant tolerance predictors. Specifici ties of 10 prediction tools for variants with different AFs. The black horizontal line indicates performance for all variants (AF 1% and <25%). The variants with AF <1% have low AF in the whole dataset but have higher AF in at least one of the populations. MA, MutationA ssessor; MT2, MutationTas ter2; PPH2, PolyPhe n-2. https://doi.o rg/10.1371/j ournal.pcbi. 1006481.g001 How good are pathoge nicity predicto rs in detecting benign variants? PLOS Computationa l Biology | https:/ /doi.org/10.13 71/journal.p cbi.1006481 February 11, 2019 7 / 17  performances.org/10.13 71/journal.p cbi.1006481 February 11, 2019 7 / 17  performances. The differences are large as the p value scale ranges from 1 to 10 −16 . Thus, practically all the differences are statistically highly significant. Population-speci fic performance ExAC database contains information for the genetic ancestry of the individuals. Thus, in addition to the general performance, we were able to analyze also ancestry-based assessment. The same three tools, i.e. PON-P2, FATHMM, and VEST, showed the highest specificities also on the data for the ancestry groups (called for populations hereafter) (Fig 2). The methods, however, show somewhat different performances for different populations. PON-P2 and FATHMM have small performance differences between the populations, 2 and 1%, respectively, while VEST has bigger performance differences, 11% between FIN and AFR. Interestingly, all the tools have the lowest specificity for the Finnish population. Interestingly, all the tools have the lowest specificity for the Finnish population. This is presumably because the small, and in the past rather closed population passed through a narrow bottleneck some 300 years ago during which certain unique alleles were highly enriched. We analyzed whether the differences in specificities in the populations were due to the differences in the percentages of variants predicted as unknown (S1 Table). The percentages of the predicted unknown variants for most tools are similar across all populations except for the Finnish population. Most tools, except for PON-P2, have the lowest percentage of variants that could not be predicted for the Finnish population. On the other hand, PON-P2 has slightly higher percentage of unknown variants in Finnish population. The difference in performance between the populations is much bigger than the difference in the percentage of unknown variants. Next, we identified population-specific common variants which have AF 1% and <25% in one population but have AF <1% in all the other populations. These are referred to as population-specific unique variants and the remaining variants for the population are referred to as non-unique variants. The proportions of unique variants vary in the populations, ranging from 6.8% in European population (excluding Finnish) to 62.4% in the Fig 2. Performan ce of variant tolerance predictors for variants in ethnic groups. Specificit ies of prediction tools for common variants (AF 1% and <25%) in different populations . AFR, African; AMR, American; EAS, East Asian; FIN, Finnish; NFE, Non-Finni sh Europea n; OTH, Other; SAS, South Asian; MA, MutationAsse ssor; MT2, MutationTast er2; PPH2, PolyPhe n-2. https://doi.o rg/10.1371/j ournal.pcbi. 1006481.g002 How good are pathoge nicity predicto rs in detecting benign variants? PLOS Computationa l Biology | https:/ /doi.org/10.13 71/journal.p cbi.org/10.13 71/journal.p cbi.1006481 February 11, 2019 8 / 17  African population (S2 Table). Humans have their origin in Africa and it is well known that the African population has the highest variation as most variants are recent, see e.g. [50]. The tools showed lower specificities for the unique variants than for the non-unique variants in the populations (Fig 3A). The lowest performance is seen for the unique variants in the Finnish population. Performance differences vary largely depending on the tools and the populations. The performance differences between the unique and non-unique variants are the lowest in the African population (0.6-3.5%) and the highest in the Finnish population (3.2-12.1%) (S3 Table). With respect to the tools, the differences for unique variants are the lowest for FATHMM (ranging from 1.3 to 4.1%) and PON-P2 (ranging from 1.2 to 8.0%) and the largest for MutationTaster2 (18.4%), VEST (16.4%) and LRT (14.9%).2 to 8.0%) and the largest for MutationTaster2 (18.4%), VEST (16.4%) and LRT (14.9%). The differences for the unique and nonunique variants in each population are visualized in Fig 3A. The differences are the smallest for FATHMM and PROVEAN, up to 3.6 and 6.6%, and the largest for LRT and CADD, up to 18.7 and 12.2%. As the tools have lower performances for unique variants, we investigated the frequencies of unique variants and those that were not unique (i.e. non-unique). Most unique variants have low AF, between 1% and 5%, while the proportions of non-unique variants with different AFs are similar (Fig 3B). Since many predictors have been trained with variants with high allele frequencies as benign variants, the lower specificities for unique variants could be due to disparity in the frequencies. To control the bias due to frequency, we compared the performance of the tools for unique and non-unique variants with AF in the same range (i.e. 1-5%) in each population.e. 1-5%) in each population. The comparison showed that the tools indeed have poorer performance for unique variants than for non-unique variants (Fig 3C). The differences are the smallest for FATHMM, PON-P2 and PROVEAN, up to 3.7, 6.7 and 6.7%, and the largest for CADD and MutationTaster, up to 12.2% for both (S4 Table, Fig 3C). For Finnish population there are generally the largest differences (3.2 to 12.2%). Effects of the sex and chromosomal location on prediction performance Finally, we evaluated the performance for variants from males and females in the populations. No differences were observed in predictor performance. Most of the variants in these two datasets are overlapping. The proportions of unique variants in male (AF 1% in male but <1% in female) and female (AF 1% in female but <1% in male) populations are 5.6% and 16.9%, respectively (S5 Table). The number of unique variants in females is 3.4 times higher than the unique variants in males. The number of unique variants in females is 3.4 times higher than the unique variants in males. This may be because of the larger numbers of females than males with African ancestry (1.75 times) in the ExAC dataset. The AFR population has the largest percentage of unique variants compared to the other groups. The performance for unique variants in male is lower than for the common variants and unique variants in female (Fig 4). To assess the influence of variants in sex chromosomes for the lower performance of tools for unique variants in males, we examined the proportions of variants for females and males in all chromosomes. As there were only 3 variants in Y chromosome we could not investigate performance for variants in this chromosome. In the remaining chromosomes, the ratio of unique variants in males to females range from 0.17 to 0.39, with a median of 0.30. The ratio is 0.28 in the X chromosome, i.e. very close to the median (S5 Table).30. The ratio is 0.28 in the X chromosome, i.e. very close to the median (S5 Table). The tools show only minor differences in the specificities for variants in different chromosomes (Fig 5).",
         "PON-P2, FATHMM, and VEST showed the highest specificities on the data for the ancestry groups.",
         null,
         null,
         null
        ],
        [
         "27",
         "What percentage of twin pregnancies experience vanishing twin syndrome?",
         "Jialyu Huang et al.,2020, DOI:https://doi.org/10.1186/s12958-020-00597-w:\n In this retrospective analysis of 1016 live-born singletons, we provided the first evidence that EMT was not associated with adverse perinatal outcomes in IUI cycles. Neither a thin or thickened endometrial lining (EMT ≤7.6 or ≥ 13.1 mm) at trigger day would increase the infant risks of PTB, LBW, SGA and major congenital anomalies. Despite significant progress in ultrasonographic, immunological and molecular diagnostic techniques of Fig. 1 Flow chart of the study. IUI, intrauterine insemination; CC, clomiphene citrate; hMG, human menopausal gonadotropin Huang et al. Reproductive Biology and Endocrinology            18:48 Page 4 of 9  endometrial receptivity, the EMT still remains to be the mainly used marker as part of standard cycle monitoring in assisted reproductive treatment [ 12, 13]. Over the past decades, much efforts have been made to explore its prognostic capacity in pregnancy chances. While it has been well-established that the probability of clinical pregnancy was lower in IVF/ICSI patients with thin EMT defined by different cut-off values from 6 to 10 mm [ 12], only few studies have investigated this issue in the setting of IUI treatment with conflicting results [ 13, 22]. The pooled data revealed no significant difference in mean EMT between pregnant and non-pregnant IUI women, and no difference in clinical pregnancy was observed after association analysis using EMT cut-offs ranging from 3 to 12 mm [ 13, 22]. Although the discrepancy remains unclarified, it does highlight the necessity to separately assess the influence of EMT on reproductive outcomes between IUI and IVF/ICSI treatment. The relationship between EMT and neonatal morbidity in IVF/ICSI cycles has been evaluated in several prior studies [ 11, 14- 17]. In 2006, Chung et al. In 2006, Chung et al. [ 14] first reported that patients with an EMT ≤10 mm had a twofold higher risk of experiencing PTB, LBW or intrauterine fatal demise beyond first trimester than those with an EMT > 12 mm. This study, however, was flawed by a high proportion of multiple gestations (37%) out of 435 Table 1 Baseline characteristics according to endometrial thickness at trigger day ≤7.6 mm (n = 109) 7.7- 13.0 mm (n = 798) ≥13.1 mm (n = 109) P-value Maternal age (years) 30.7 ± 3.7 30.2 ± 3.7 30.0 ± 3.3 0.313 Maternal BMI (kg/m 2) 21.76 ± 2.56 22.34 ± 3.35 22.13 ± 2.87 0.401 Paternal age (years) 32.9 ± 5.4 32.1 ± 4.4 32.1 ± 4.9 0.465 Paternal BMI (kg/m 2) 25.09 ± 2.95 24.50 ± 3.08 24.35 ± 3.21 0.224 Nulligravida, n (%) 53 (48.6) 551 (69.0) 90 (82.6) < 0.001 Nulliparity, n (%) 101 (92.7) 761 (95.4) 103 (94.5) 0.465 Duration of infertility (years) 2.8 ± 1.8 2.9 ± 2.0 3.0 ± 2.2 0.828 Infertility diagnosis, n (%) 0.610 Anovulation 19 (17.4) 143 (17.9) 16 (14.7) Male factor 21 (19.610 Anovulation 19 (17.4) 143 (17.9) 16 (14.7) Male factor 21 (19.3) 192 (24.1) 28 (25.7) Endometriosis 0  18 (2.3) 3 (2.8) Mixed 3 (2.8) 26 (3.3) 5 (4.6) Unexplained 66 (60.6) 419 (52.5) 57 (52.3) Rank of IUI attempts, n (%) 0.652 1st cycle 70 (64.2) 485 (60.8) 61 (56.0) 2nd cycle 30 (27.5) 246 (30.8) 35 (32.1) 3rd or more 9 (8.3) 67 (8.4) 13 (11.9) Stimulation protocol, n (%) 0.661 LE 13 (11.9) 79 (9.9) 9 (8.3) LE + hMG 96 (88.1) 719 (90.1) 100 (91.7) Length of treatment (days) 10.3 ± 2.0 11.5 ± 2.9 12.5 ± 2.7 < 0.001 Peak E 2 level (pg/mL) 256.6 ± 197.9 287.9 ± 205.8 346.3 ± 248.7 < 0.001 Endometrial thickness (mm) 6.60 ± 0.96 10.06 ± 1.43 14.42 ± 1.27 < 0.001 Postprocessing TMSC (millions) 27.8 ± 27.4 32.6 ± 37.4 33.2 ± 33.8 0.430 Vanishing twin syndrome, n (%) 4 (3.7) 15 (1.9) 2 (1.8) 0.469 Pregnancy complications, n (%) 7 (6.4) 60 (7.5) 9 (8.3) 0.875 Year of treatment 0.131 2008- 2011 6 (5.5) 72 (9.0) 14 (12.8) 2012- 2015 39 (35.8) 345 (43.2) 43 (39.4) 2016- 2018 64 (58.5) 72 (9.0) 14 (12.8) 2012- 2015 39 (35.8) 345 (43.2) 43 (39.4) 2016- 2018 64 (58.7) 381 (47.7) 52 (47.7) Data are presented as mean ± standard deviation or number (percentage). BMI body mass index, IUI intrauterine insemination, LE letrozole, hMG human menopausal gonadotropin, E2 estradiol, TMSC total motile sperm count Huang et al. Reproductive Biology and Endocrinology            18:48 Page 5 of 9  viable pregnancies. By retrospectively analyzing 764 singleton deliveries, a later study by Moffat et al. [ 15] showed that EMT was strongly predictive for neonatal birthweight in pregnancies with obstetric complications but not in uneventful pregnancies. Consistently, Ribeiro and colleagues [ 16] found that an EMT less than 7 mm was significantly associated with a decrease in neonatal birthweight Z-score. However, they did not detect an increase in clinically relevant outcomes such as PTB or LBW, and attributed it to the limited size of the liveborn singleton sample subset. This was subsequently validated in another study by Oron et al. [ 11] showing that women with an EMT < 7.5 mm had a higher incidence of placenta-related pregnancy complications including SGA. In addition to the aforementioned four studies on fresh embryo transfer cycles, a recent analysis of 6181 singletons resulting from FET also demonstrated that a thin endometrium (EMT < 8 mm) was independently Table 2 Neonatal outcomes according to endometrial thickness at trigger day ≤7.6 mm (n = 109) 7.7- 13.0 mm (n = 798) ≥13.1 mm (n = 109) P-value Mode of delivery, n (%) 0.533 Vaginal 48 (44.0) 326 (40.9) 50 (45.9) Cesarean section 61 (56.0) 472 (59.1) 59 (54.1) Gender, n (%) 0.493 Female 60 (55.0) 395 (49.5) 52 (47.7) Male 49 (45.0) 403 (50.5) 57 (52.3) Gestational age (days) 274.3 ± 8.4 273.3 ± 11.7 274.5 ± 9.4 0.503 Preterm birth, n (%) 6 (5.5) 59 (7.4) 5 (4.6) 0.462 Very preterm birth, n (%) 0  8 (1.0) 0  0.530 Birthweight (g) 3319.8 ± 499.9 3304.3 ± 511.6 3323.3 ± 442.5 0.0) 0  0.530 Birthweight (g) 3319.8 ± 499.9 3304.3 ± 511.6 3323.3 ± 442.5 0.732 Low birthweight, n (%) 7 (6.4) 37 (4.6) 3 (2.8) 0.435 Very low birthweight, n (%) 0  6 (0.8) 0  1.000 Macrosomia, n (%) 6 (5.5) 54 (6.8) 4 (3.7) 0.430 Z-score 0.25 ± 1.07 0.25 ± 1.08 0.23 ± 1.01 0.914 Small-for-gestational age, n (%) 8 (7.3) 49 (6.1) 5 (4.6) 0.694 Large-for-gestational age, n (%) 19 (17.4) 123 (15.4) 13 (11.9) 0.509 Major congenital malformations, n (%) 4 (3.7) 10 (1.3) 2 (1.8) 0.119 Data are presented as mean ± standard deviation or number (percentage) Table 3 Crude and adjusted analysis of neonatal outcomes in endometrial thickness categories ≤7.6 vs. 7.7 - 13.0 mm ≥13.1 vs. 7.7 - 13.0 mm Crude Adjusted a Crude Adjusted a Newborn parameters, MD (95% CI) Gestational age (days) 1.0 ( −1.2- 3.2) 0.7 ( −1.5- 3.0) 1.2 ( −1.0- 3.4) 1.5 ( −0.7- 3.7) Birthweight (g) 15.6 ( −85.3- 116.5) 16.4 ( −86.6- 119.4) 19.0 ( −81.9- 119.9) 28.6 ( −73.2- 130.3) Z-score −0.01 ( − 0.22- 0.21) 0 ( − 0.21- 0.6- 119.4) 19.0 ( −81.9- 119.9) 28.6 ( −73.2- 130.3) Z-score −0.01 ( − 0.22- 0.21) 0 ( − 0.21- 0.22) −0.02 ( − 0.23- 0.20) 0 ( − 0.22- 0.21) Adverse outcomes, OR (95% CI) Preterm birth 0.73 (0.31 - 1.73) 0.81 (0.33 - 2.01) 0.60 (0.24 - 1.54) 0.63 (0.24 - 1.65) Low birthweight 1.41 (0.61 - 3.25) 1.44 (0.58 - 3.58) 0.58 (0.18 - 1.92) 0.57 (0.17 - 1.95) Macrosomia 0.80 (0.34 - 1.91) 0.91 (0.37 - 2.25) 0.53 (0.19 - 1.48) 0.53 (0.18 - 1.54) Small-for-gestational age 1.21 (0.56 - 2.63) 1.21 (0.53 - 2.76) 0.74 (0.29 - 1.89) 0.73 (0.28 - 1.92) Large-for-gestational age 1.16 (0.68 - 1.97) 1.21 (0.69 - 2.11) 0.74 (0.40 - 1.37) 0.70 (0.37 - 1.34) Major congenital malformations 3.00 (0.93 - 9.74) 3.18 (0.82 - 12.28) 1.47 (0.32 - 6.81) 1.32 (0.24 - 7.00 (0.93 - 9.74) 3.18 (0.82 - 12.28) 1.47 (0.32 - 6.81) 1.32 (0.24 - 7.13) MD mean difference, CI confidence interval, OR odds ratio a Adjusted for parental age, parental body mass index, gravidity, parity, infertility duration, infertility diagnosis, rank of cycle, stimulation p rotocol, length of treatment, peak estradiol level, postprocessing total motile sperm count, vanishing twin syndrome, pregnancy complications, and year of treatmen t Huang et al. Reproductive Biology and Endocrinology            18:48 Page 6 of 9  related to a lower birthweight and Z-score, while the modest difference failed to translate into a significantly higher risk of SGA [ 17]. Similar to the contradictory findings on pregnancy outcomes, the present study reported no evidence for an association between EMT and IUI infant outcomes as in IVF/ICSI cycles. However, the underlying reasons are still unclear as to why a thin endometrial lining is more clinically relevant in IVF/ICSI than in IUI treatment. During either natural or stimulated cycles, continuous E2 production from growing follicles induces endometrial proliferation, thus leading to increased EMT as measured by TVU. Compared with a thin endometrium developing under mild ovarian stimulation conditions as in IUI, a thin endometrium developing under maximal stimulation conditions in IVF/ICSI cycles may imply genuine diminished potential for implantation and placentation. Moreover, in the presence of thin endometrium, the embryos are implanted much closer to the spiral arteries of the endometrial basal layer, whose higher vascularity and oxygen concentrations could be detrimental to embryos due to the production of reactive oxygen species [ 28, 29]. Since embryos are developed in vivo in IUI cycles, we speculate that they may be less susceptible to high oxygen tensions in comparison with in vitro cultured or further vitrified/thawed embryos in IVF laboratories. Nevertheless, these explanations are merely theoretical and further investigations are warranted to provide stronger and more direct evidence. VTS, defined as the spontaneous reduction of multiple pregnancies to singleton birth [ 30], was identified as an independent predictor of adverse neonatal outcomes after multivariable regression analysis. This phenomenon has been estimated to occur in 50% of pregnancies with initially three or more gestational sacs, 36% of twin pregnancies and 10 - 30% of IVF/ ICSI pregnancies [ 31, 32]. In the present study, we reported for the first time that the prevalence of VTS was about 2.1% in IUI treatment, much lower than that in IVF/ICSI cycles. Consistent with previous findings on IVF/ICSI infants [ 30, 33, 34], VTS is associated with higher risks of PTB and LBW in IUI newborns as well. However, the adjusted odds of SGA and birth defects did not reach statistical significance possibly due to the limitation of relatively small sample size, as reflected by the wide confidence intervals. Therefore, more studies with larger datasets are needed to further assess the incidence, risk factors as well as neonatal outcomes of VTS in IUI cycles. A major weakness of the study relies on its retrospective design. Some baseline characteristics were not balanced between the stratified groups. Notably, the fact that more patients with thin EMT had been pregnant before but the frequency of nulliparity was similar among the three groups suggested a higher rate of prior pregnancy loss, which was associated with increased chances of adverse neonatal outcomes such as PTB and LBW [ 35]. While the present study included gravidity and pari ty number for adjustment, we were unable to retrieve data on the actual causes of pregnancy loss history (e.g.g., miscarriage, abortion or ectopic pregnancy). In addition, possible unknown or unavailable confounders may not be accounted for in the regression model. For example, over 50% of singletons were delivered via cesarean section, but the indications and timing in relation to neonatal mortality remained unclear [ 36, 37]. Also, information on endometrial morphology, which has been reported to influence pregnancy outcomes [ 13], was incomplete in our database and thus not analyzed in this study. Other risk factors for adverse neonatal outcomes, including previous delivery history, medication use, metabolic status, nutrition intake and lifestyle habit [38], were not adequately controlled due to the lack of detailed records. Therefore, future studies with a comprehensive evaluation of women before and during pregnancy are needed for further investigation. Secondly, previous studies have observed significant differences in EMT among various ovarian stimulation protocols and in neonatal outcomes of different sperm sources [ 7, 22, 23, 39]. To minimize the likelihood of confounding, only IUI cycles using LE and partner semen were included for analysis. Thus, our conclusion should not be extrapolated to IUI singletons born after treatment with donor semen or with clomiphene citrate, gonadotropin only and natural cycles. The single-center nature should also caution generalization of the study finding to other areas in China and other parts of the world. Finally, while the measurement of EMT was performed by highly experienced doctors at our center, some inter-observer inconsistency may still be present. Moreover, most of the neonatal data were collected from parental questionnaires instead of direct access to medical records. In this regard, the detection of some minor problems could be comprised and the actual incidence of congenital malformations may",
         "Vanishing twin syndrome occurs in 36% of twin pregnancies.",
         null,
         null,
         null
        ],
        [
         "28",
         "What effect does chrysin have on the expression of follicle-stimulating hormone receptor in D-gal-induced mice?",
         "Xinning Li et al.,2022, DOI:https://doi.org/10.1080/21655979.2021.2005991:\n Results\nChrysin regulated the body weight of D-gal-induced mice\nAs\nFigure 1\ndepicts, in the D-gal group, there was an obvious weight loss compared with control group. Chrysin treatment significantly and concentration-dependently increased the mice weight compared to the D-gal group.\nFigure 1.\nOpen in a new tab\nChrysin regulated the body weight of D-gal-induced mice\n. The weight of D-gal-induced mice. ***P < 0.001.\nChrysin regulated the levels of hormones in D-gal-induced mice\nWith the application of ELISA, the levels of E2, FSH, P and AMH were evaluated. The D-gal treatment group had significantly increased serum FSH level and drastically decreased E2, P and AMH levels compared to those in the control (\nFigure 2a-d\n). Compared with the POF model group, chrysin treatment markedly and concentration-dependently decreased the serum FSH level and increased the serum E2, P and AMH levels.\nFigure 2.\nOpen in a new tab\nChrysin regulated the levels of hormones in D-gal-induced miceFigure 2.\nOpen in a new tab\nChrysin regulated the levels of hormones in D-gal-induced mice\n. (a-d) The levels of E2, FSH, P and AMH were detected using ELISA. ***P < 0.001 vs Control;\n##\nP < 0.01,\n###\nP < 0.001 vs D-gal.\nChrysin inhibited the inflammation and oxidative stress in D-gal-induced mice\nAs a key inflammatory cytokine the levels of TNF-α, IL-1β and IL-6 in ovarian tissues were assessed using ELISA. Compared with control group, the levels of TNF-α, IL-6 and IL-1β were significantly elevated in the ovaries of mice treated with D-gal. Conversely, chrysin treatment significantly inhibited the elevation of TNF-α, IL-1β and IL-6 levels in D-gal group and reversed them close to that of controls, D-gal+Chrysin144.4 group was closer to controls compared with D-gal+Chrysin36.1 and D-gal+Chrysin72.2 groups (\nFigure 3(a\n)).\nFigure 3.\nOpen in a new tab\nChrysin inhibited the inflammation and oxidative stress in D-gal-induced miceOpen in a new tab\nChrysin inhibited the inflammation and oxidative stress in D-gal-induced mice\n. The expression of TNF-a, IL-1β and IL-6 was measured using ELISA. (b) The expression of MDA, GSH-px and SOD was detected using corresponding kits. (c) The expression of ROS was detected by DCH-DA fluorescence staining. ***P < 0.001 vs Control;\n#\nP < 0.05,\n##\nP < 0.01,\n###\nP < 0.001 vs D-gal.\nTo further investigate the effects of chrysin on oxidative stress, the activities of MDA, GSH-px and SOD in D-gal-induced mice were assessed with corresponding kits. Compared with Control, D-gal upregulated MDA expression but downregulated GSH-px and SOD expression (\nFigure 3(bFigure 3(b\n)). However, chrysin treatment reversed the effects of D-gal induction on MDA, GSH-px and SOD, evidenced by the downregulated MDA expression and upregulated GSH-px and SOD expression. In addition, we observed an increase in ROS production after D-gal stimulation, which was obviously and concentration-dependently reversed by chrysin treatment (\nFigure 3(c\n)).\nChrysin improved follicular development and increased follicular number in D-gal-induced mice\nFemale mice treated with D-gal showed significant reduction in ovaries. On the other hand, chrysin treatment significantly counteracted these effects and maintained ovarian weights (\nFigure 4(a\n)). It was noted that the effects of chrysin on ovaries weight were in a concentration-dependent manner. The classification of follicle was based on the characteristics proposed by Hirshfield & Midgley [1978,\n23\n]. According to\nFigure 4(b23\n]. According to\nFigure 4(b\n), counting of the primordial follicles showed that D-gal treatment reduced the proportion of primordial follicles compared to that in the control group. Significantly more follicles were counted at the different developmental stages of maturation in the chrysin treatment group than in the D-gal model group.\nFigure 4.\nOpen in a new tab\nChrysin improved follicular development and increased follicular number in D-gal-induced mice\n. (a) The ovarian weight of D-gal-induced mice. (b) The number of follicles in D-gal-induced mice. (c) Pathological changes of ovaries in mice were detected by HE staining. D. Western blot was used to detect the expression of FSH receptor. **P < 0.01, ***P < 0.001 vs Control;\n#\nP < 0.05,\n##\nP < 0.01,\n###\nP < 0.001 vs D-gal.\nIn\nFigure 4(c#\nP < 0.05,\n##\nP < 0.01,\n###\nP < 0.001 vs D-gal.\nIn\nFigure 4(c\n), the control group and the D-gal+ chrysin group showed normal ovarian morphological structure with different stages of follicular development. However, in the D-gal group, almost all follicles were atresia, and no primordial follicles were observed. In addition, it was found that the granular layer in the atresia follicle was generally loose. The growing follicles have a relatively irregular arrangement with discrete membrane layers, and the thickness of zona pellucida is uneven in some follicles. It was found that most primary follicles lacked oocytes and the interstitial structure of ovary was destroyed. In addition, collagen fiber accumulation in the form of vascular congestion, hemorrhage and fibrosis were detected in the matrix.According to western blot, the results indicated that the expression of follicle-stimulating hormone receptor (FSHR) in D-gal group was extensively dropped off in comparison with the control. As revealed in\nFigure 4(d\n), FSHR was significantly and concentration-dependently risen compared with D-gal group after chrysin treatment.\nChrysin inhibited the apoptosis of ovarian granulosa cells in D-gal-induced mice\nWith the application of TUNEL assay, the apoptosis of ovarian granulosa cells was detected. As\nFigure 5(a-B)\ndisplayed, D-gal induction greatly enhanced the number of TUNEL-positive cells compared with that in Control group. However, the increased cell apoptosis was then significantly decreased by chrysin, suggesting that chrysin exerted protective effects against D-gal-induced ovarian cell apoptosis. Compared with the Control, D-gal induction downregulated Bcl2 expression but upregulated the expression of Bax and cleaved caspase3, which were then reversed by chrysin (Figure 5(c-D\n)). Notably, chrysin inhibited the apoptosis of ovarian granulosa cells in POF.\nFigure 5.\nOpen in a new tab\nChrysin inhibited the apoptosis of ovarian granulosa cells in D-gal-induced mice\n. (a-b) TUNEL was used to detect the cell apoptosis. (c-d) The levels of apoptosis-related proteins were detected using western blot. ***P < 0.001 vs Control;\n#\nP < 0.05,\n##\nP < 0.01,\n###\nP < 0.001 vs D-gal.\nChrysin molecule targeted ERβ and activated ERβ expression in POF\nOne previous study has found that genistein can activate the expression of ERβ when improving radiation-induced POF, which indicates that ERβ activation plays a vital role in POF [\n24\n]. Besides, according to Targetnet database (\nhttp://targetnet.scbdd.com/\n), chrysin could target ERβ (\nFigure 6\n). In view of this, we also investigated the relationship between chrysin and ERβ in POF. As shown in\nFigure 7(aFigure 7(a\n), chrysin could be docked with ERβ, and the docking result is −8.3. In addition, the decreased ERβ expression in D-gal-induced mice was then increased by chrysin administration (\nFigure 7(b\n)).\nFigure 6.\nOpen in a new tab\nMolecular docking diagram\n. Chrysin was verified to be docked with ERβ. In the activate site of LEU-339, LEU-343, HIS-475, PHE-356, LEU-476 and LEU-301 bonds with chrysin.\nFigure 7.\nOpen in a new tab\nChrysin molecule targeted ERβ and activated ERβ expression in premature ovarian failure\n. The mRNA (a) and protein (b) levels of ERβ were detected using RT-qPCR and western blot, respectively.",
         "Chrysin significantly and concentration-dependently increased FSHR expression compared with the D-gal group.",
         "5.0",
         "4.0",
         "5.0"
        ],
        [
         "29",
         "What databases were used for functional annotation and to identify microRNAs and transcription factors involved in the fertile WOI regulation?",
         "P. Sebastian-Leon et al.,2021, DOI:https://doi.org/10.1093/molehr/gaab027:\n Ethics statement This retrospective study was approved by the institutional review board of the Instituto Valenciano de Infertilidad, Valencia, Spain (1401FIVI-002-CS). Menstrual cycle endometrial expression dataset Previous transcriptomic data of the 238 endometrial receptivity dating genes (ERarray) (Diaz-Gimeno et al., 2011) measured throughout the menstrual cycle for 523 women constituted the reference set for this study that evaluates gene pair correlations across the menstrual cycle. This sample cohort was published between 2011 and 2017 (DiazGimeno et al., 2011, 2013, 2017)( Supplementary Fig. S1is a summary of clinical characteristics of the participants) and was deposited in Gene expression Omnibus (GEO) (Edgar, 2002) (GSE126581). This data set comprised a reference set that was previously classified according to the transcriptomic taxonomy of menstrual cycle progression, especially of the WOI (details about how this transcriptomic taxonomy was built are available in Diaz-Gimeno et al.  and summarized in Supplementary Fig. S1). Briefly, the endometrial dating of these transcriptomic profiles was applied rigorously using blood LH and transcriptomics as gold standards, with machine learning predictors to assign each menstrual cycle profile, from the proliferative to late secretory endometrial stage. Then, these transcriptomic profiles were tested clinically against embryo transfer results. Embryo transfer occurred during the same cycle day on which transcriptomic profiles were obtained, without personalized embryo transfer or change in progesterone administration. We identified the profiles with the highest probability of achieving pregnancy-defined by live birth-as the fertile WOI (detailed inDiaz-Gimeno et al.  and summarized in Supplementary Fig. S1). Here, we used the raw gene expression data of this reference set of samples, and their endometrial dating based on the previously defined transcriptomic profiles, to evaluate gene coexpression and regulation throughout the menstrual cycle. 2 Sebastian-Leon et al. Downloaded from https://academic.oup.com/molehr/article/27/5/gaab027/6217366 by guest on 08 February 2023  Outlier treatment Population outliers on each transcriptomic profile were removed when falling outside of normal 99% tolerance ellipses (Hadi et al., 2009) estimated with the ggplot2 R-package (Wickham, 2009). The normalization process between the arrays used in this study was implemented with the quantile method included in thepreprocessCore R-package (Bolstad, 2013). Co-expression matrices across the menstrual cycle Gene co-expression was calculated between each gene pair across menstrual cycle progression using Pearson’s correlation coefficient included in the R environment (R Core Team 2016). In molecular biology, WGCNA is the best method to understand how genes work together as a team comprising modules of highly correlated genes (Zhang and Horvath, 2005). WGCNA converts co-expression values into Topology Overlap Measures (TOM) and uses these measures to group highly co-expressed genes ( Ravasz et al.,2 0 0 2). WGCNA (Zhang and Horvath, 2005) was performed for each transcriptomic profile considering a minimum module size of 10 genes and a dendrogram cut height of 0.25 for module merging (WGCNA R-package; Langfelder and Horvath, 2008 ). Networks were represented in Cytoscape (Shannon et al. Networks were represented in Cytoscape (Shannon et al., 2003) based on WGCNA modules and co-expression matrices calculated using the Pearson’s correlation coefficient included in R environment (RC o r eT e a m ,2 0 1 6). Network topological parameters by profile were estimated using the Cytoscape Network Analyser plugin ( Assenov et al., 2007). Global network parameters by profile, such as mean of co-expressions by gene in the network, co-expression modules, density, centralization and the total number of clustered and unclustered genes of each menstrual cycle stage and in each module were estimated as described inHorvath and Dong . Network density refers to a normalized version of the average number of neighbor genes, which shows how densely the network is related to edges connecting genes. Centralization describes the extent to which a network contains many nodes that connect to a central hub node, but do not connect to their neighbors. Hub genes are nodes with high connectivity (high number of correlations) within a module. Statistical approaches Positive and negative correlation proportions in each transcriptomic profile were contrasted using Cochran’s Q test ( Cochran, 1950) implemented in the RVAideMemoire R package (Herve ́, 2018), which also performs a Wilcoxon test for pairwise comparisons if there are significant differences between two or more transcriptomic profiles (Herve ́, 2018 ). Profile comparisons with a false discovery rate (Benjamini and Hochberg, 1995)( F D R )<0.05 were considered as significantly different in the proportion of positive and negative correlations. Gene pairs relevant to the fertile WOI were then prioritized based on these proportions and according to their gene co-expression behavior in relation to the other endometrial profiles (Fig. 1B). Figure 1 Study design and workflow. Endometrial biopsies were collected at different times of the menstrual cycle (Supplementary Fig. Endometrial biopsies were collected at different times of the menstrual cycle (Supplementary Fig. S1) for gene expression assessment and endometrial profile dating through the endometrial receptivity array. Gene co-expression matrices obtained for each endometrial profile were then used to study the regulation of menstrual cycle progression (A), with a special focus on the fertile window of implantation (WOI) profiles (B). (A) The menstrual cycle regulation. Gene modules and key drivers by profile, co-expression matrices, functional and regulatory information are available to stimulate hypothesisdriven single-molecule endometrial studies in a user-friendly database: the Menstrual Cycle Gene Co-expression Network (www.men strualcyclegcn.com). (B) The fertile WOI regulation. The proportion of negative/positive co-expressions was compared between profiles, and gene pair prioritization according to these co-expression behaviors was carried out for the fertile WOI profiles. The prioritized genes were then functionally annotated and microRNAs (miRNAs) and transcription factors (TFs) potentially regulating them were included in the network. Gene Ontology (GO), Gene cards, Kyoto Enciclopedy of Genes and Genomes (KEGG), TRRUST and mBISON were the databases consulted for functional annotation and miRNA and TFs involved. The resulting regulatory network was used to derive molecular mechanisms behind the fertile WOI profiles and their main players were validated in patients with recurrent implantation failure (RIF) of endometrial origin, because these patients have an altered WOI. Menstrual cycle progression gene co-expression 3 Downloaded from https://academic.oup.com/molehr/article/27/5/gaab027/6217366 by guest on 08 February 2023  .............. ................................................. ........................................ . .......... ................................................. .................................... . .......... ................................................. ....... Functional genomic characterization and regulation Functional annotation for co-expressed genes was consulted in Gene Ontology (GO) biological processes (BP) and molecular functions (MF), and Kyoto Encyclopaedia of Genes and Genomes (KEGG) pathways retrieved from GeneCards (Stelzer et al.,2 0 1 6) (Release 4.12). Additionally, cell locations for gene products were manually added from the GeneCards sections Compartments (confidence levels 5 and 4) and GO Cellular Components (CC) (only experimental evidence) (Stelzer et al., 2016) (Release 4.12). Then, TFs regulating co-expressed genes were searched in the TRRUST (Transcriptional Regulatory Relationships Unravelled by Sentence-based Text mining) database (Han et al., 2017), which contains only experimentally validated TF-target associations., 2017), which contains only experimentally validated TF-target associations. Predicted miRNAs regulating these genes were obtained from mBISON (analysis on miRNA binding site over-representation) database (Gebhardt et al.,2 0 1 5). Genes co-expressed (activations and repressions) and their gene regulators were represented using network models visualized and analyzed in Cytoscape (Shannon et al.,2 0 0 3) to unveil the molecular regulatory mechanisms behind endometrial receptivity and menstrual cycle progression. Validation in patients with recurrent implantation failure To validate the relevance of the prioritized WOI genes and TFs in endometrial receptivity acquisition, we evaluated whether these features were dysregulated in patients with recurrent implantation failure (RIF), who are known to have a dysregulated endometrium in terms of receptivity. For this purpose, the expression dataset GSE111974 (Bastu et al. For this purpose, the expression dataset GSE111974 (Bastu et al., 2019; Sezerman and Ulgen, 2018) was downloaded from the NCBI repository GEO (Edgar, 2002). This study includes endometrial gene expression data from 24 patients with RIF and 24 fertile controls with a history of at least one live birth and with no associated comorbidities. Endometrial biopsies were collected at the expected WOI (LH þ7t oL Hþ10) and RIF was determined as failure of pregnancy in /C21 3 consecutive IVF cycles with/C21 1 transfer(s) of good quality embryos in each cycle. Exclusion criteria included active pelvic infections, undiagnosed vaginal bleeding, uterine anomalies, endometriosis and karyotype anomalies in one or both partners. Raw data were preprocessed and normalized by quantile method using limma R-package v.3.46.0 (Ritchie et al., 2015); and annotation from probeset to gene symbol was downloaded from GEO (GPL17077)., 2015); and annotation from probeset to gene symbol was downloaded from GEO (GPL17077). Exploratory analysis was performed to detect batch effects using principal components analysis (PCA). Detected batch effects were corrected using linear models (limma R-package v.3.46.0) (Ritchie et al.,2 0 1 5). Unsupervised clustering using k-means (R environment v. 3.5.1) (R Core Team, 2016) was applied to establish sample groups for those detected batches with no association with registered study variables. Optimal numbers of batch clusters were defined by the silhouette method (cluster v.2.0.7-1 R-package) (Ma ̈chler et al.,2 0 1 2). Association between each measured gene/TF and the evaluated condition (RIF versus control) was estimated usingWGCNA R-package based on correlations (Langfelder and Horvath, 2008), as the WOI gene pairs were prioritizated based on correlation changes. Nevertheless, a differential expression analysis using linear models (limma R-package v.3.46.0) (Ritchie et al.3.46.0) (Ritchie et al.,2 0 1 5) was also applied as an independent validation method to test whether these WOI genes were differentially expressed between RIF and controls. A gene/TF was considered to be associated with RIF/differentially expressed when FDR<0.05. A general scheme of the study design is provided inFig. 1. MenstrualCycleGCN website Detailed results for each gene across menstrual cycle progression are described in the web-tool Menstrual Cycle Gene Co-expression Network (MenstrualCycleGCN) (www.menstrualcyclegcn.com ) (Fig. 1A), a public and user-friendly website based on HTML5 and JavaScript. The application user interface was developed using Polymer 2, and the networks using Cytoscape.js framework (Franz et al., 2016). MenstrualCycleGCN makes an intensive use of new web technologies and standards; therefore, only new browsers are fully supported. These include the following: Chrome 67þ, Firefox 60þ, Edge þ41, Safari 11þ and Opera 53þ. These include the following: Chrome 67þ, Firefox 60þ, Edge þ41, Safari 11þ and Opera 53þ. Older versions of these browsers may yield errors. Internet Explorer is not supported. Further information about the methodology behind prioritization of key drivers (genes with high connectivity and centralization within a module) and party hubs (genes strongly connecting different modules), functional annotation of the 238 ERarray genes, and functional and regulation enrichment analyses of each endometrial profile and WGCNA module are available at www.menstrualcyclegcn.com > Documentation.",
         "Gene Ontology, Gene cards, Kyoto Enciclopedy of Genes and Genomes, TRRUST and mBISON were the databases consulted.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "30",
         "What sugars besides glucose does GLUT1 transport?",
         "Ivana Vrhovac Madunić et al.,2021, DOI:https://doi.org/10.3389/fcell.2021.703671:\n GLUT1GLUT1 (SLC2A1) was the first facilitative glucose transporter to be discovered, having been purified from human erythrocytes (Kasahara and Hinkle, 1977) and cloned from HepG2 cells (Mueckler et al., 1985). GLUT1 is ubiquitously expressed in all organs, ranging from endothelial cells of the blood-brain barrier to fetal tissues; however, it is most abundant in erythrocytes, where it increases the cells’ glucose carrying capacity (Mueckler and Thorens, 2013). Besides glucose, it transports other sugars including mannose, galactose, glucosamine, and reduced ascorbate (Carruthers et al., 2009; Mueckler and Thorens, 2013). Several research groups reported GLUT1 expression in the placenta (Fukumoto et al., 1988; Bell et al., 1990; Farell et al., 1992; Takata et al., 1992; Jansson et al., 1993; Clarson et al., 1997). In different cells and tissues, GLUT1 is regulated by glucose concentration, insulin, and growth factors (Mueckler and Thorens, 2013). GLUT1 was the first GLUT to be found in the endometrium; more precisely, its mRNA is abundant in ESCs of both human and rodents. High GLUT1 levels have been detected in decidua during pregnancy, implying the role ofGLUT1 in maintaining pregnancy and fetal development (Sakata et al., 1995; Y amaguchi et al., 1996; Frolova et al., 2009). Furthermore, it was shown that during decidualization, GLUT1 protein levels can increase up to 10 times and are accompanied by an increase in the glucose uptake in vivo (Frolova et al., 2009; Neff et al., 2020). GLUT1 upregulation has also been reported in decidualized hESCs (Tamura et al., 2014). In contrast, downregulated GLUT1 in ESCs lead to a reduced glucose uptake and suppressed decidualization (Huang et al., 2019). Accordingly, in cultured hESCs, the GLUT1 knockdown impaired decidualization (Frolova and Moley, 2011a). A recent study reported that hESCs are able to survive the hypoxic environment by activating metabolic pathwaysvia GLUTs, which could be relevant for the menstrual and implantation period (Kido et al., 2020). In the human endometrium,GLUT1 expression is upregulated during the peri-implantation period, and is higher during the secretory phase compared to the proliferative phase (Korgun et al., 2001; Strowitzki et al., 2001; Zhang et al., 2020). In contrast, no difference was observed in the expression of GLUT3 in different phases. This result is consistent with immunohistochemical analysis (Zhang et al., 2020). An adequate glucose metabolism is crucial for endometrial differentiation and decidualization processes. These processes are mainly mediated by progesterone (P4) and partially by 17 β-estradiol (E2). These processes are mainly mediated by progesterone (P4) and partially by 17 β-estradiol (E2). In vitro studies demonstrated that GLUT1 is upregulated by P4 in murine ESCs, and furthermore that 17 β-estradiol E2 reverses the effect of P4, bringing GLUT1 expression back to its basic level (Frolova et al., 2009). Although this finding indicates the opposite effect of E2 and P4 on GLUT1, both hormones are present at the time of implantation, indicating their interplay Frontiers in Cell and Developmental Biology | www.frontiersin.org 3 September 2021 | Volume 9 | Article 703671 fcell-09-703671 September 1, 2021 Time: 16:21 # 4 Vrhovac Maduni ć et al. Endometrial Glucose Transporters in the regulation of GLUT1. A recent study confirmed these findings in vivo, showing that P4 was responsible for the upregulation of GLUT1 protein in the murine uterus (Zhang et al., 2020)., 2020). In women with idiopathic infertility, GLUT1 is significantly reduced, suggesting its enrolment in the regulation of endometrial function (von Wolff et al., 2003). Further, a GLUT1 knockdown resulted in failed embryo implantation in vivo, indicating that GLUT1 silencing affects the endometrium, rather than the embryo (Zhang et al., 2020). These findings propose that pregnancy failure may be associated to a lower GLUT1-mediated glucose metabolism, resulting in reduced endometrial receptivity. A recent in vitro study showed that high insulin concentrations downregulate the GLUT1 gene and protein expression, which is accompanied by a slight reduction in the glucose uptake in decidualizing hESCs (Jakson et al., 2020). Further clinical studies are necessary to determine the importance of insulin regulation in endometrial function and decidualization in patients with insulin resistance and hyperinsulinemia.GLUT3GLUT3 (SLC2A3) is a high-affinity glucose transporter expressed in tissues with an intensive glucose metabolism, such as the brain, placenta, preimplantation embryos, sperm, and testes (Maher et al., 1992; Shepherd et al., 1992; Haber et al., 1993; Burant and Davidson, 1994; Boileau et al., 1995; Hauguel-de Mouzon et al., 1997; Pantaleon et al., 1997). In the human endometrium, GLUT3 is expressed in the proliferative phase and decidua at a constant level throughout different menstrual stages or in early pregnancy (Hahn et al., 2001; von Wolff et al., 2003; Korgun et al., 2005). GLUT3 and GLUT1 are differentially expressed in utero-placental modulation. In mouse decidua, GLUT1 is upregulated after midpregnancy, whereas GLUT3 is slightly downregulated, indicating that this glucose transporter is not crucial after mid-pregnancy (Y amaguchi et al., 1996)., 1996). This decrease was confirmed in rats, where GLUT3 is found in the uterine epithelium and stroma, and its expression increases through day 4 since gestation (Korgun et al., 2001). A correlation between placental and uterine GLUT3 expression with serum progesterone levels has been shown during early pregnancy (Ferre-Dolcet et al., 2018). Studies using GLUT3 KO mice showed that GLUT3 is not essential for fertilization or implantation of blastocysts, but for the development of early post-implanted embryos. The deletion of GLUT3 inhibits early embryonic development due to apoptosis of ectodermal cells shortly after implantation (Schmidt et al., 2009). In summary, GLUT1 is a basal transporter that is responsible for constant glucose intake during the entire pregnancy, whereas GLUT3 is required for optimizing glucose uptake in the first trimester when maternal circulation through the placenta is not yet established. Putatively, GLUT3 may function as a “scavenger” glucose isoform, acting in circumstances when extracellular glucose levels are lower than the normal circulating ones (Brown et al., 2011). Therefore, GLUT3 may represent an important transporter for fetal supply in early gestation. However, further studies on GLUT3 expression and function in the endometrium are required.GLUT4GLUT4 ( SLC2A4) is an insulin-regulated transporter found in the heart, skeletal muscle, adipose tissue and brain (Bryant et al., 2002; McKinnon et al., 2014). Reversible translocation of GLUT4 to the cell surface, stimulated by insulin, leads to rapid glucose uptake into the cell (Pessin et al., 1999; Bryant et al., 2002). Several studies found muscle and fat cell insulin resistance to be associated with decreased GLUT4 expression and its impaired translocation (Dunaif, 1997; Shepherd and Kahn, 1999; Mueckler, 2001), contributing to insulin resistance and Type II diabetes (Bryant et al., 2002). Data on GLUT4 in endometrial tissue are scarce and inconsistent. Endometrial GLUT4 was detected only in epithelial cells by some research groups (Mioni et al., 2012), while others found it in both stromal and epithelial cells (Cui et al., 2015). In contrast, some reported the absence of GLUT4 in endometrial tissue and stromal cells (von Wolff et al., 2003; Frolova and Moley, 2011a)., 2003; Frolova and Moley, 2011a). Insulin-dependent endometrial GLUT4 was found maximally expressed in the follicular phase of menstrual cycle, with a decline in the luteal phase (Mioni et al., 2012; Zhai et al., 2012; Cui et al., 2015). In contrast, GLUT1, which is an insulin independent transporter, was found maximally expressed in the luteal phase and mainly localized in the cytoplasm of stromal cells predetermined to transform into decidual cells (Mioni et al., 2012). Although little is known about the insulin dependence of the endometrium, GLUT4 was found in endometrial epithelial cells (Mioni et al., 2004; Mozzanega et al., 2004; Fornes et al., 2010; Zhai et al., 2012). Moreover, Mioni et al. demonstrated that GLUT4 is dysregulated in the endometrium of PCOS patients. This is important, as PCOS is the most common metabolic and endocrine disorder affecting 5-10% of all women of reproductive age (Diamanti-Kandarakis, 2008). Chronic anovulation is one of its hallmarks, and about 75% of infertility due to anovulation is attributed to PCOS (Costello and Eden, 2003). Other important clinical manifestations of PCOS may also include obesity, insulin resistance (IR), and signs of androgen excess (Rotterdam ESHRE/ASRM-Sponsored PCOS Consensus Workshop Group, 2004). Endometrial GLUT4 is significantly decreased in hyperinsulinemic women with PCOS, compared to those with PCOS but without IR. This implies a link between hyperinsulinism and GLUT4 downregulation. In women with PCOS and no IR, endometrial GLUT4 was significantly decreased only in obese women (Mioni et al., 2004). This is important, as IR is found in 50-70% of all women with PCOS (Sirmans and Pate, 2013), consequently leading to hyperinsulinism (Dunaif, 1997) in both obese and non-obese patients, with obesity aggravating IR (Maison et al., 2001). It is noteworthy that 30-50% patients with PCOS are obese (Sam, 2007; Barber et al., 2019)., 2001). It is noteworthy that 30-50% patients with PCOS are obese (Sam, 2007; Barber et al., 2019). Hyperandrogenemia negatively affects glucose uptake in endometrial cells regulated by GLUT4 (Rosas et al., 2010; Zhang and Liao, 2010; Rivero et al., 2012; Li et al., 2015). However, Frontiers in Cell and Developmental Biology | www.frontiersin.org 4 September 2021 | Volume 9 | Article 703671 fcell-09-703671 September 1, 2021 Time: 16:21 # 5 Vrhovac Maduni ć et al. Endometrial Glucose Transporters one study reported no connection between the hyperandrogenic syndrome per se and GLUT4-regulated glucose uptake (Mioni et al., 2004). All these findings can possibly explain the reduced pregnancy rate in women with PCOS at only 30-40%, although their ovulation rate reaches up to 80% (Imani et al., 2002). Furthermore, they experience a higher rate of spontaneous and recurrent abortions (Giudice, 2006). Furthermore, they experience a higher rate of spontaneous and recurrent abortions (Giudice, 2006). It has been shown that hyperinsulinemia does not have a direct effect on LH secretion, and therefore does not cause the hypothalamic-pituitary-ovarian axis dysfunction in PCOS. However, in vivo and in vitro studies confirmed that insulin excess stimulates further androgen production in ovaries via insulin receptors on theca/interstitial ovarian stromal cells, therefore aggravating already increased ovarian androgen production in PCOS and lowering the hepatic sex-hormone binding globulin (Smith et al., 1987; Plymate et al., 1988; Nestler et al., 1998; Baillargeon, 2007; Rosenfield and Ehrmann, 2016). IR in PCOS patients is commonly treated with antidiabetic drug metformin (1,1-dimethylbuguanide hydrochloride), which is known to improve insulin sensitivity and decrease insulin and androgen levels, thus promoting ovulation (Harborne et al., 2003; Lord et al., 2003)., 2003; Lord et al., 2003). Endometrial GLUT4 is lower in PCOS than in non-PCOS patients. However, after 3 months of metformin treatment, GLUT4 mRNA and protein expression is increased in the endometrium of PCOS patients (Zhai et al., 2012). Moreover, metformin directly restores endometrial function by increasing GLUT4 expression (Carvajal et al., 2013). In addition, metformin increases serum IGF-1 and glycodelin during the luteal phase (Jakubowicz et al., 2001; Palomba et al., 2006), which may contribute to an improved endometrial environment for the establishment and maintenance of pregnancy (Carvajal et al., 2013). GLUT4 may be regulated by steroid hormones. Progesterone (P4) alone, and in combination with estradiol (E2), decreases endometrial GLUT4, suggesting that an abnormal hormonal secretion pattern, such as in PCOS, could disrupt normal GLUT4 expression (Cui et al., 2015)., 2015). The importance of GLUT4 in the endometrium is well established, although numerous studies were performed on only a small number of patients. Therefore, the elucidation of endometrial GLUT4 function and its molecular mechanisms in physiological and pathophysiological (such as PCOS) states is of high interest. Furthermore, due to its limited cellular expression and inducible nature, GLUT4 could be a desirable target for new non-hormone-based drugs (McKinnon et al., 2014).GLUT8The GLUT8 ( SLC2A8) transporter is mostly present in the brain and testes (Doege et al., 2000; Carosa et al., 2005), but is also detected in the uterus, ovary, and term placenta (Carayannopoulos et al., 2000; Ibberson et al., 2000; Limesand et al., 2004; Schmidt et al., 2009). It is localized in intracellular membranes of predominantly late endosomes and lysosomes, as well as in the ER and trans-Golgi (Stanirowski et al., 2017). The late-endosomal/lysosomal limiting membrane is the cellular hub for various metabolic reactions (e.g., mTORC1 recruitment to the lysosomal membrane, modulation of mitochondrial homeostasis, regulation of protein, and lipid content of the cellular membrane and membranes of various vesicles). Therefore, besides transporting glucose, GLUT8 may also function as a sensor for various metabolites in cellular metabolic homeostasis (Alexander et al., 2020). Some of the GLUT8 is cleaved into N-terminal and C-terminal peptides., 2020). Some of the GLUT8 is cleaved into N-terminal and C-terminal peptides. Of its 12 trans-membrane domains, the cleavage occurs at domain 10, which happens to be the only one that is 100% conserved across the species. The full-length and N-terminal cleaved proteins are retained at the endosomal/lysosomal boundary, whereas the carboxy domain accumulates in a different population of vesicles and is unlikely to perform transport (Alexander et al., 2020). GLUT8 is one of several GLUT species that exists as several alternatively spliced mRNA variants, and there is few available data on its functionality. Alexander et al. concentrated on its expression in breast cancer and found that GLUT8 has three such variants: v1 (full-length), v2 (missing exon 9 and part of the C-terminus) and v3 (missing exon 2 and 3). In normal human tissue, by far the highest expressions of v1 and v3 mRNA variants are found in the testis, followed by liver and muscle, and only very little in the mammary gland, with the full-length version being predominant and v3 accounting for only about 5-10% of the total GLUT8 mRNA. The expression survey found that only the fulllength GLUT8 variant is translated into stable, functional protein in both normal and tumor tissues (Alexander et al., 2020). In the endometrium, GLUT8 can provide glucose required for protein glycosylation in the ER during decidualization (Frolova and Moley, 2011b). GLUT8 is elevated in ESCs upon decidualization in mice and humans, whereas the protein expression is not increased (Frolova and Moley, 2011a). Mouse Slc2a8-/- mutants exhibited disordered decidualization, indicating that SLC2A8 plays an important role in this process (Adastra et al., 2012). Among other GLUTs, only GLUT8 was found to be reduced in hESCs in a hypoxic environment., 2012). Among other GLUTs, only GLUT8 was found to be reduced in hESCs in a hypoxic environment. Recently, it was reported that suppressing GLUT8 under hypoxia might promote the preferential use of glucose for glycolysis (Kido et al., 2020). Hence, gaining an understanding of hESCs behavior in the hypoxic conditions of menstrual and implantation periods is of great importance. GLUT8 is anchored at intracellular membranes, and it is not sensitive to insulin in most tissues, unlike certain other GLUT family members (e.g., GLUT4 and GLUT12). The blastocyst is an exception. Namely, it has been shown that in the blastocyst, GLUT8 translocates to the plasma membrane in response to insulin stimulation and imports glucose. Thus, it is considered critical for murine blastocyst survival (Carayannopoulos et al., 2000; Pinto et al., 2002; von Wolff et al., 2003).Sodium-Glucose Transporter 1Sodium-glucose transporter 1 is primarily expressed in the small intestine and kidneys, but it is also found in the heart, liver, pancreas, brain, lungs, prostate, and uterus (Wright et al., 2004; Vallon et al., 2011; Gorboulev et al., 2012; Kashiwagi et al., 2015; Vrhovac et al., 2015; Salker et al., 2017; Sharma et al., 2017; Vrhovac Madunic et al., 2017). The first report of SGLT1 expression and activity in murine and human endometrial epithelial cells showed that it is Frontiers in Cell and Developmental Biology | www.frontiersin.org 5 September 2021 | Volume 9 | Article 703671 fcell-09-703671 September 1, 2021 Time: 16:21 # 6 Vrhovac Maduni ć et al. Endometrial Glucose Transporters responsible for controlling glycogen accumulation essential for embryo implantation (Salker et al., 2017). Endometrial glycogen is associated with fertility. During peri-implantation and in early pregnancy, the increase in endometrial glycogen represents a highly important source of glucose. Glycogen production requires the cellular uptake of glucose, mostly by passive transporter GLUTs. Recently, another mechanism of accumulating glucose has been demonstrated via the secondary active transporter SGLT1, which can mediate glucose uptake even at low extracellular glucose concentrations (Salker et al., 2017). Glucose transport across the endometrium (at the estrus phase of the cycle) was detected in wild type, but not in SGLT1knockout mice. The SGLT1 deficiency notably decreased the endometrial glycogen, litter size, and weight of the offspring. Moreover, SGLT1 was upregulated upon the decidualization of primary hESCs, which is crucial for embryo implantation. Furthermore, endometrial SGLT1 levels are significantly lower in women with recurrent pregnancy loss when compared to women with a healthy pregnancy. The study reveals a novel mechanism for adequate endometrial glycogen storage for pregnancy as a key factor for embryo implantation. Overall, endometrial SGLT1 deficiency in the human during implantation may lead to early pregnancy failure and obstetrical complications, including low fetal growth (Salker et al., 2017). These results suggest that caution is needed in the period of preconception and during pregnancy if SGLT1 inhibitors are considered for the treatment of diabetes (Vallon, 2015; Song et al., 2016).Signaling Pathways in GLUT ExpressionOne of the crucial signaling pathways in the glucose metabolism is the insulin pathway, with insulin acting as a main regulator of blood glucose concentration by increasing glucose uptake in muscles and fatty tissue, while inhibiting its production in liver (Saltiel and Kahn, 2001). The primary function of insulin is to maintain glucose homeostasis by translocating GLUT4 to the cell surface via activation of the PI3K/Akt pathway (Bryant et al., 2002). Maintaining the physiologically relevant narrow range of blood glucose concentration requires tight and rapid control of insulin signaling. An essential step in successful pregnancy is decidualization of the human endometrium, a process consisting of many morphological changes and a functional differentiation of ESCs. This process depends on a complex interaction of transcription factors, cytokines, cell cycle regulators, and signaling pathways (Okada et al., 2018)., 2018). Proliferation of the endometrium is induced by insulin through the insulin-like growth factor 1 (IGF-1), induced by estrogen in the proliferative phase (Strowitzki et al., 2006). A study by Nandi et al. reported that impaired insulin signaling affects female reproduction health through the hypothalamic-pituitary-gonadal axis, resulting in altered cycles and follicular development. Authors also observed unsuccessful pregnancies, increased calcification of placentae, and small embryos in insulin-resistant and hyperinsulinemic model mice. Other studies proposed that hyperandrogenism in the ovaries of PCOS patients could disrupt the insulin pathway (Zhang and Liao, 2010) and affect the function of the endometrium (Diamanti-Kandarakis and Papavassiliou, 2006). Furthermore, hyperinsulinism observed in PCOS patients lowers the concentration of circulating glycodelin (biomarker of endometrial function) and insulin-like growth factor-binding protein-1 (IGFBP-1), a feto-maternal interface adhesion factor. This contributes to early pregnancy loss due to diminished endometrial receptivity caused by decreased uterine vascularity and endometrial blood flow (Jakubowicz et al., 2001). A recent study showed that the deletion of ovary-specific insulin and/or IGF-1 receptor(s) caused infertility in mice as a consequence of impaired ovulation, luteal differentiation, uterine receptivity, and steroid hormone signaling (Sekulovski et al., 2020). Furthermore, dual receptor ablation compromised decidualization, reduced endometrial thickness, and inhibited embryo implantation. Authors further observed decreased activation of ERK/MAPK signaling (Sekulovski et al., 2021). Authors further observed decreased activation of ERK/MAPK signaling (Sekulovski et al., 2021). Other pathological states, such as inflammation and obesity, cause insulin resistance by serine phosphorylation of the insulin receptor or insulin receptor substrate (IRS) proteins by TNF- α and PKC ε (Hotamisligil, 1999). A subsequent decrease in tyrosine phosphorylation inhibits kinase activity of IR. A study by Bergman et al. suggested that IR and type 2 diabetes can be caused by reduced hepatic clearance of insulin. Samuel and Shulman reported ectopic lipid accumulation, and the activation of unfolded protein response likewise led to the development of IR. Other signaling pathways have been linked to the regulation of GLUT expression. Other signaling pathways have been linked to the regulation of GLUT expression. Models of artificial decidualization and delayed/activated implantation have demonstrated that the expression of murine uterine epithelial GLUT1, GLUT8, and GLUT9B are regulated by estradiol and progesterone through Akt/MAPK/PRKAA signaling (Kim and Moley, 2009). This result implies that the hormonal changes associated with early pregnancy affect GLUT expression and indicate an important role of uterine glucose utilization in endometrial decidualization, embryo implantation, and maintenance of pregnancy. Similarly, GLUT1 has been shown to be induced by progesterone-activated Hif1α and c-Myc via the PI3K/Akt signaling pathway during decidualization at the sites of murine embryo implantation. The corresponding increase in GLUT1 expression was one of the drivers of decidual glycolysis taking place under normoxic conditions, a phenomenon the authors termed Warburg-like glycolysis (Zuo et al., 2015)., 2015). Another study analyzed the GLUT isoform expression in a PCOS-like model of uterine IR and hyperandrogenism (Zhang et al., 2016). Their results revealed both an increase and decrease in the mRNAs expression in rats treated with insulin. In contrast, most GLUT genes (including GLUT4) were downregulated in hyperandrogenic conditions. The authors suggested that chronic treatment with insulin and hCG causes changes in the GLUT expression due to altered PI3K/Akt and MAPK/ERK signaling pathways, which contributes to the initiation and development of local uterine IR. Another signaling pathway connecting GLUT expression and uterine decidualization is Notch signaling. In vitro and in vivo experiments demonstrated that inhibition of Notch signaling through endometrial loss of the RBPJ transcription factor leads to impaired decidualization in the murine uterus and in human endometrial fibroblasts associated with reduced progesterone receptor and GLUT1 expression (Strug et al., 2019).Frontiers in Cell and Developmental Biology | www.frontiersin.org 6 September 2021 | Volume 9 | Article 703671 fcell-09-703671 September 1, 2021 Time: 16:21 # 7 Vrhovac Maduni ć et al. Endometrial Glucose Transporters Epigenetic Regulation of Glucose Transporters Knowledge about epigenetic regulation of endometrial GLUTs is limited, as there is almost no experimental data on the epigenetic regulation of these transporters in the endometrium. Epigenetic modifications, including histone modification, DNA methylation, and non-coding RNAs, play an important role in the decidualization of hESCs by influencing the expression of target genes (Sugino et al., 2016; Liu et al., 2",
         "GLUT1 transports mannose, galactose, glucosamine, and reduced ascorbate.",
         "5.0",
         "3.0",
         "5.0"
        ],
        [
         "31",
         "What cutoff value was used for the Benjamini-Hochberg adjusted p-value in the enrichment analysis?",
         "Shuqin Zhang et al.,2018, DOI:https://doi.org/10.1186/s12918-018-0530-9:\n Datasets\nThelevel3geneexpressiondataforbreastcancer(BRCA) andovariancancer(OV)weredownloadedfromTheCancer Genome Atlas (TCGA). The gene expression data were generated with UNC AgilentG4502A. We chose the samplesfromthesolidtissuesonly.Thereare526and572 samplesforbothBRCAandOV,respectively.Theexpression value of 17,814 genes was measured. The missing data for each specific gene was imputed with the mean value of the known samples. We also downloaded the most updated protein-protein interaction (PPI) data for humans from BioGrid https://thebiogrid.org.W ec h o s e the genes having PPIs in the gene expression data, and 9603geneswereselected.\nTo choose the differential expressed genes for BRCA and OV, t-test and Kolmogorov-Smirnov test were applied. The genes having p- v a l u e sl e s s0 . 0 1i nb o t h tests were chosen, where the p-values were adjusted by controlling the false discovery rate. We then got 7742genes.\nCoexpressionnetworkconstructionWe first computed the pairwise Pearson correlation coefficient rij to measure the coexpression levels between gene i and gene j. To make the correlations across the two networks comparable, we ‘normalized’ the Pearson correlation coefficients with the same method a ss h o w ni n[25]. Fisher’s z transformation score for each rij was first computed as zij = 0.5log 1+rij 1−rij .T h e z-scores were then normalized to makezij follow normal distribution. After this, the ‘normalized’ correlations were obtained by transforming back to rij. With these steps, the correlations in the two networks will be on the same level. We then did hard thresholding to make the network be unweighted. We took 99.5% quantile of the transformed correlation coefficients as the threshold. If the absolute value of the correlation coefficient is greater than the threshold, we assigned an edge between the corresponding genes, otherwise, there is no edge. With this method, the average degree is about 37 in bothnetworks.ModuleidentificationinBRCAandOVcoexpression networks\nBefore we did module identification, we first removed the genes having no links with any other genes in both networks. The left gene number became 6779. Consider the constructed gene coexpression networkG1 (BRCA), G2 (OV) consisting of 6779 genes. We let the adjacency matrices for both networks beA1,A2,w h e r eAk(i,j) = 1 represents there is an edge between genei and gene j. We first applied the model proposed in [24]t oc l u s t e rt h e genes in lower subspace for both networks. This model aims at finding the clusters in multiple networks and aligning the clusters at the same time. The main idea is to use spectral clustering to find the cluster in each network,andalignthembymaximizingtheclustersimilarity of multiple networks. Here, we only have two networks.We assume the putative number of clustersM in both networksisgivenfirst. Let Sk be the assignment of the 6779 vertices intoM clusters for the networkGk,w h e r eSk im = 1, if i ∈ Gk belongs to them-th cluster, otherwiseSk im = 0f o ri = 1,2, ··· ,6779, m = 1,2, ··· ,M,k = 1,2. The optimizationmodelisformulatedas: min ∑ k=1,2 M∑ m=1 Sk ,m T (Dk − Ak)Sk ,m Sk,m TSk,m − β ∑ k,l=1,2 M∑ m=1 Sk ,m T Sl ,m ∥Sk,m∥2∥Sl,m∥2 s.t. Sk i,m ∈{ 0,1},i = 1,2, ··· ,6779; m = 1,2, ··· ,M;k = 1,2; M∑ m=1 Sk ,m = 1,for k = 1,2.  The first term in the objective function is to do clustering in both networks separately, and the second term defines the similarity of the clusters in both networks measuredbycosinefunction. β balancesthecontributions frominter-networkandintra-network. Tosolvetheoptimizationproblem,thevariable Sk im is relaxed. Using the same technique as spectral clustering, theoptimizationproblemistransformedto: min  ̃/Psi1( ̃S) = Tr( ̃STC ̃S),s . t. t . ̃ST  ̃S = 2IM, whereC = ( L1 0 0 L2 ) − β ( 0 In In 0 ) ,  ̃S = (  ̃S1  ̃S2 ) ,  ̃Sk ,m = Sk ,m ∥Sk,m∥2 , andLk = Dk −Ak,0isan n×nmatrixwithallentriesbeing zero. By computing the eigenvectors corresponding to theM smallest eigenvalues of matrixC, the original vertices in the networks are projected to a space of dimensionM. To get the clusters, we may usek-means clustering to cluster the data points similar to spectral clustering. Due to the large size of the network,k-means does not work well.Instead,weappliedhierarchicalclusteringwithcompletelinkagetoclusterthevertices.Thedistanceischosen to be the spearman distance. This is because when the size of matrix C is large, the range of the eigenvector entriesislarge,buttheirorderiscomparativelystable.The algorithmissummarizedin‘ Algorithm’.Selection of parameterβ and MThe parameterβ controls the connections between the vertices in both networks. When β = 0, it is equivalent to finding the clusters in two networks separately. Whenβ becomes larger, the corresponding vertices in both networks tend to belong to the same cluster. We note that even when a group of vertices are densely connected in the first network, while their corresponding parts are isolated in the second network, the method will put all the isolated vertices in the same cluster as in the first network. Here, since both networks were controlled to have a close number of total connections, we directly set β = 1, which means the connection weight between two networks is the same as that within Algorithm: Input: Adjacency matrixA k,k = 1,2, andM, which is the putativenumberofclusters. 1. Constructthematrix C; 2. Computethe M eigenvectorsu1,u2,··· ,uM correspondingtothe M smallesteigenvaluesofmatrix C; 3. Computethe M eigenvectorsu1,u2,··· ,uM correspondingtothe M smallesteigenvaluesofmatrix C; 3. Constructanewmatrix T ∈ R2·6779×M withcolumns u1,u2,··· ,uM; 4. Clusterthepointsconstructedfromeachrowofmatrix T intoM clusters usinghierarchicalclusteringwith spearmandistance. Output:Clusterlabelinbothnetworks. bothnetworks.Wenotethatwhen β> 1 anditiswithina reasonablerange,theresultsdonotchangemuch.The number of clustersM was chosen according to the eigenvalues of matrixC. M corresponds to the first big eigengap [26]. We note that hereM is not the number of clustersineitherofthetwonetworksbecausebychoosing β, the isolated vertices in one network can also be clustered together depending on the other network. It should be the number of the union clusters in both networks. Thusthemethodcanfindboththeconsistentclustersand the differential clusters having quite different connection probabilities. Aftertheaboveclusteringprocedures,wecangetaclusterlabelforeachgene.Thecorrespondingclustersinboth networks may include different genes. We take the union of the genes as the cluster. Due to the large number of genes, the clustering method may have some bias. Some unconnected subnetworks may be clustered together. Before going to further analysis, we need to check each cluster such that it cannot include unconnected subnetworks. These resulted subnetworks are defined as the modulesweidentified.Enrichmentanalysis\nTo see the associations between the identified gene modulesandgenefunctions,wedidGeneOntology(GO)[ 27], KEGG pathway [28], and Disease Ontology (DO) [29] enrichmentanalysisforeachmodule.GOannotatesgenes to biological processes (BP), molecular functions (MF), and cellular components (CC) in a directed acyclic graph structure. We only considered BP terms here. KEGG annotates genes to pathways, and Disease Ontology (DO) annotates genes with human disease associations. To see whether the identified modules show statistically significant, concordant differences between the two diseases, we also did Gene Set Enrichment Analysis (GSEA) [30]. GSEA was also done for the three types of enrichment analysis: GO, KEGG, and DO. We implemented the enrichment analysis with ‘clusterProfiler’ [31]. For all the cases, we let the cutoff be the Benjamini-Hochberg adjusted p-value0.05,andrecordedalltheenrichedterms withp-valuelessthan0.05.",
         "The cutoff was a Benjamini-Hochberg adjusted p-value of 0.05.",
         null,
         null,
         null
        ],
        [
         "32",
         "What are some databases used for pathway and network knowledge in biomarker discovery?",
         "Jinlian Wang et al.,2015, DOI:https://doi.org/10.7150/jca.10631:\n Summary of pathways and networks\ncentric approaches for cancer biomarker\ndiscovery\nAvailability of biomedical pathways and ne tworks based on large -scale data gathering through\ndiverse omics  data sources offers new opportunities\nto explain the causality of relationships between bi ological entities and cancers [15]. As shown in Figure 1,\nthe general steps of the biomarker discovery include\nthe following: 1) Define precisely a well -framed, relevant clinical problem and focus the experimental\ndesign around appropriate study populations and\nsamples; 2) Collect tissue samples or fluids from p atients and suitable assays; 3) Acquire high-throughput\ndata from the omics technologies; 4) Analyze the data\nusing signal processing, statistical and machine\nlearning methods to select relevant features from the\ndata; 5) Integrate the pathway/network knowledge\nfrom databases such as KEGG, HMDB and Reatcome\nmapping candidate biomarkers to the correspondingfrom databases such as KEGG, HMDB and Reatcome\nmapping candidate biomarkers to the corresponding\npathways or networks; 6) Evaluate biomarkers to estimate their diagnostic or prognostic capability and\nclinical validity using alternative technologies such as\nWesten blot, ELSA, and RT -PCR. In computational\naspect, cross -validation and independent validation\nare the commonly used methods to evaluate the pe rformance of a biomarkers. P-values, sensitivity,\nspecificity and the area under receiver operating curves\n(AUC) are used as quantitative indicators of the pe rformance of the methods [16]; 7) U se the biomarkers\nfor clinical applications after reliable pre -clinical tests\nand validation of the markers in a large population.Statistics methods\nStatistical methods test scientific theories when\nobservations, processes or boundary conditions are\nsubject to stochasticity. For examples, the classical\nt-test has been extensively used for testing differential\ngene expression in microarray data [35]. However,\nthis kind of procedure relies on reasonable estimates\nof reproducibility or within -gene error, requiring a\nlarge number of replicated arrays. Thus, several\nmethods for improving estimates of variability and\nstatistical tests of differential expression have been\nproposed. For example, Significance Analysis of M icroarrays (SAM) aimed to improve the unstable error\nestimation in the two -sample t-test by adding a var iance stabilization factor which minimizes the variance\nvariability across different intensity ranges [36, 37].\nANOVA model approach is widely used in multiple\nkinds of omic data. For example, it was used to model\nmicroarray data with the effects of array, condition,microarray data with the effects of array, condition,\nand condition- array interaction and then to fit the\nresiduals with the effects of gene, gene -condition interaction, and gene -array interaction [38,39]. Also, it\nwas applied to capture the effects of controlled\ngroups, batches, condition, alias of experimental\nequipment, and condition -metabolite interaction separately on LC -MS data [40]. To improve the accuracy\nand sensitivity of analytic results, false discovery rate\n(FDR) [41] and its refinement, q -value, (q -value\npackage, www.bioconductor.org) have been rapidly\nJournal of Cancer 2015, Vol. 6    http://www.jcancer.org  56  adopted for genomic, proteomics a nd metabolic data\nanalysis including the widely-used SAM, DAVID [42]\nand other approaches [36]. Another statistics method\nfor biomarker discovery is l inear discriminant anal ysis (LDA), one of the classical statistical classificationtechniques based on the  multivariate normal distr ibution assumption, is quite robust and powerful to\ndiscover biomarker or pathways between omics data\nfor many different applications despite the distrib utional assumption. Compared to LDA, quadratic di scriminant analysis (QDA) requires more observations\nto estimate each variance -covariance matrix for each\nclass [43]. In addition, logistic regression analysis has\nbeen successfully used to  evaluate biomarker performance of prostate cancer with mRNA profiling\n[44]. Logistic regression (LR) model based on the r egression fit on probabilistic odds between  comparing\nconditions requires no specific distribution assum ption (e.g. Gaussian distribution) but is often found to\nbe less sensitive than other approaches[42,43].Graph theory based network and visualization\nThe modeling fundamentals of  graph theory are\noften used to describe the global  topology, structure\nor the community of a complex system. It emphasizes\non entities (e.g, genes, proteins, diseases, biological\nprocess) and the relat ions between them. The complexity of graphical modeling can be either simple\nonly with nodes and edge s or more complex where\nedges have weights, and nodes and edges can be of\ndifferent types. Recent publications have applied\ngraphical modeling in computational biology to study\nbiological networks, enhance the ability to draw\ncausal inferences from functional MRI experiments,\nsupport the early detection of disconnection and the\nmodeling of pathology spread in neurodegenerative\ndisease such as Alzheimer's disease [45 -49]. For e xample, in mammalian cells, Bleris et al. have had early\nsuccess in characterizing the dynamics of key feed\nforward modules and motifs, helping to enable theforward modules and motifs, helping to enable the\ncircuit design of adaptive gene expression [50]. Using\ngraph based approaches, Ma'ayan et al. model cellular\nmachinery including genes, proteins and other su bcellular compartments [51], in which the interactions\nbetween components are drawn as edge connections\nbetween the relevant nodes [51]. Gene expression data\ncombined with networ k analysis can yield important\ninformation on how expression variation relates to\ndifferences between observed states [52]. As closely\nconnected genes tend to be involved in similar functions, network annotation can complement clusters\nobtained via fold ch ange analysis [7]. A standard\nsystems-based approach to biomarker and drug target\ndiscovery consists of placing putative or known b iomarkers in the context of a network of biological\ninteractions, followed by different ‘guilt -by-\nassociation’ analyses [53].Figure 1. The pipeline  of pathway/networks  centric approach for  cancer biomarker   discovery. A variety of  computational tools  and algorithms have  been proposed for  biomarker discovery  based on pathway and  network methods .\nThe most commonly  used methods are  categorized roughly  into statistical [17],   graph theory [18],  Bayesian methods  [19], text mining [20],  machine learning  [21-23] and integrative  methods summarized  in Table 1....... Sample preparation and process Study design Data analysis KEGG Reactome PubMed Pathways/network analysis Biomarkers identificationBiomarkers validationClinical application Databases • Fluid selection (blood, tissue, cell, urine..) • Sample collection, storage, freeze, assay  design and processing • Patient population • Noise filtering, background subtraction,  alignment, normalization, transformation,  feature selection, visualization. • Feature selection, visualization •Clinical question •Experiment workflow design •Patient population HMDB A C B Transcriptomics ProteomicsMetabolomics epigeomics Genomics PID\nJournal of Cancer 2015, Vol. 6    http://www.jcancer.org  57  Table 1. Computational methods for biomarker discovery categorized by their application, examplary tools and URLs.\nApproaches Technique & Application Examples Exemplary Tools &URLApproaches Technique & Application Examples Exemplary Tools &URL\nStatistical analysis  Hypothesis testing, random sampling. ANOVA. Detection of  differentially expressed genes/proteins, genotypes, biomarker  filtering/selection[24]   BRB:http://linus.nci.nih.gov/BRB-ArrayTools.html   PAM: http://www-stat.stanford.edu/~tibs/PAM/  SAM: http://www-stat.stanford.edu/~tibs/SAM/\nPattern recognition   Machine learning, Probabilistic, instance-based, kernel classification models. Clustering, multi-source data classification,  biomarker selection and associations [25]  Bayesian regression  models [26], partial least squares [27], and Genetic Algorithm/KNN [28].   Weka: http://weka.wikispaces.com/  LIBSVM: http://www.csie.ntu.edu.tw/~cjlin/libsvm/  PRTools: http://prtools.org/  R package: http://cran.r-project.org/web/views/Bayesian.htmlGraph/network  theory  Network topology analysis, network visualization and data  integration, clustering. Genetic, regulatory, protein-protein,  signaling network analysis, biomarker/target identification  [29]  BioNet[4]  :http://www.fda.gov/ScienceResearch/BioinformaticsTools/ucm285284.htm  Jung: http://jung.sourceforge.net/  http://bioinfo.mc.vanderbilt.edu/dmGWAS.html [30]\nData visualization and imaging  Sequence and cluster visualization, interactive visualization,  statistical analysis graphs. Data exploration, biomarker visualization, model explanation, in vivo/in vitro imaging of molecules and cells [29]   Cytoscape [31]: http://www.biotapestry.org/  Medusa: http://coot.embl.de/medusa/  Graphviz: http://www.graphviz.org/  Osprey:http://biodata.mshri.on.ca/osprey/servlet/Index  Pajek: http://vlado.fmf.uni-lj.si/pub/networks/pajek/  3Omics: http://3omics.cmdm.twNatural language processing and  information  retrieval  Ontologies, text mining, information representation standards,  information retrieval and extraction. Inference of functional  associations from publications, automated annotation and  characterization [32,33]  iHOP: http://www.ihop-net.org/UniPub/iHOP  CoPub: http://services.nbic.nl/copub/portal  PolySearch: http://wishart.biology.ualberta.ca/polysearch/  index.htm  Open Biomedical Annotator: http://bioportal.bioontology. org/annotator  GeneSeeker: http://www.cmbi.ru.nl/GeneSeeker/\nSoftware development,  Internet technologies  Data warehouses and distributed information systems, semantic Web tools, information retrieval, extraction and curation.  Biomarker discovery and validation platforms, data mining  tools, search and reasoning engines [34]   IPA:http://www.ingenuity.com/products/pathways_analysis.htm  GO: http://www.geneontology.org/GO.tools.shtml   MiMI: http://mimi.ncibi.org/MimiWeb/main-page.jspThe goal of visualization is to find patterns and\nstructures that remain hidden in the raw unstructured\ndatasets. Graph visualization is key to display directly\nthe various relationships between entities (e.g., genes,\nproteins). C hallenges of graph visualization lie in 1)\nthe high false positive rate of incorporating heter ogeneous multi-omic datasets; 2) Visual representation\nof the logical structure tran sformed from the raw data;\n3) Graph manipulation  and layout algorithm for\nrepresenting the complicated relationships between\nbiological entities. 4) Heterogeneous  omic data from\ndifferent level visualization needs more flexibility for\nlayered representation. A number of commercial and\nfree sourced graph visualization tools and platforms\nhave been extensively developed. For example, C ytoscape [31] , one of the free open source platform sproviding biological network analysis  and visualization with more than 17 2 registered plugins contri buted by the community , is very versatile in network\napplications, such as network importing,  network\nintegrating, inference customization, literature\nmining, topological clustering, functional enrichment,\nnetwork comparison, and  programmatic access [54] .\n3DScapeCS, a Cytoscape plugin providing   three-dimensional, dynamic, parallel network visua lization for Mass Spectrometry (MS) molecular ne twork [55]. IPA  [56], a commercial software tool for\npathway analysis with omics  data provides powerful\ngraphical visualized pathways and networks overlaid\nby diseases, drugs and biological process etc. PathwayStudio provide abstractive graphical interface for\nusers to analyze gene expression, protein interaction\nand metabolic data to  analyze and explore the pat hways and networks identified from data. STRING not\nonly gives the graphical visualized protein interactiononly gives the graphical visualized protein interaction\nof both known and predicted but also quantifies  each\npair of proteins by their interaction types such as\nphysical interaction and gene fusion etc. [57].Bayesian methods and its derivatives.\nBayesian methods allow informative priors  so\nthat prior knowledge or results of a previous model\ncan be used to inform the current model . In cancer\nbioinformatics and systems biology, the  primary application of Bayesian methods include\nBayesian i nference, Bayesian network, Naive Bayes classifier and\nBayesian variable selection . Among these methods,\nBayesian network is one of the  most common modeling tools for  pathway and network analysi s [19].\nBayesian network is a form of directed statistical\nmodeling designed to capture conditional dependencies between probabilistic events [58]. It consists of a\ndependency structure and local probability model\nalso named probabilistic graph models which include\nHierarchical Bayesian Networks (HBN), Probabilistic\nBoolean Networks (PBN), Hidden Markov Models\n(HMM), and Markov Logic Networks (MLN) [59 -61].\nThe dependency structure specifies how the variablesThe dependency structure specifies how the variables\nare related to each other by drawing directed edges\nbetween the variables without creating directed c ycles. Each variable depends on a possibly set of other\nJournal of Cancer 2015, Vol. 6    http://www.jcancer.org  58  variables, termed \"parents.\" Compared with other\npathway/network centric method, Bayesian network\nmodel is capable of integrating heterogeneous dat a,\nmissing value and dependent relationships between\nvariables [62].\nIn a Bayesian network model, probabilities d efine the relationship between the current node and its\npredecessor or parent in a graph [63]. The power of\nthese methods lies in their ability  to facilitate the r everse engineering of multiplex networks based on\nmolecular expression, molecular activity and/or cell\nbehavior data, serving as a precursor to synthetic\nmodifications of existing molecular pathways\n[64]. Bayesian inference is one of the very importantmodifications of existing molecular pathways\n[64]. Bayesian inference is one of the very important\nBayesian methods widely used in cancer biomarker\ndiscovery, signaling pathway and network inference\n[65,66]. It has previously been applied to gene expression data for inference of gene regulatory ne tworks [67,68], infer both protein signa ling networks\n[69,70] and gene regulatory networks  [71]. To incorporate an explicit time element, dynamic Bayesian\nInference was proposed to interrogate dynamic si gnaling responses within a Bayesian framework, with\nexisting signaling biology incorporated t hrough an\ninformative prior distribution on networks [66].\nIn\naddition, Bayesian variable selection aims at solving\nthe problems of “large p, small n” existing in omic\ndata set and using prior knowledge such as pathway\nand protein interaction to estimate the posterior\nprobability by Markov Chain Monte Carlo (MCMC)\nalso widely used to infer functional interactions inprobability by Markov Chain Monte Carlo (MCMC)\nalso widely used to infer functional interactions in\nbiochemical pathway, model the interactions between\ndifferent functional modules of a biological network\n[72] and pathway based cancer biomarke r discovery\n[73,74]. For example, Yang et al. [21] used a Bayesian\nnetwork to construct HCC cell networks and identify\nfunctional modules and interactions between these\nmodules. Stochastic simulation models offer an a lternative, but they are hitherto assoc iated with a m ajor disadvantage: their likelihood functions cannot be\ncalculated explicitly, and thus it is difficult to couple\nthem to well -established statistical theory such as\nmaximum likelihood and  Bayesian statistics. A nu mber of new methods, among t hem Approximate\nBayesian Computing and Pattern-Oriented Modeling,\nbypass this limitation. The difference between Baye sian and frequentist inference lies in the following: 1)\nBayesian inference provides answers conditional onBayesian inference provides answers conditional on\nthe observed data and not based on the distribution of\nestimators or test statistics over imaginary samples\nnot observed (Rossi et al., 2005, p. 4); 2) It includes\nuncertainty in the probability model, yielding more\nrealistic predictions. 3) It safeguards against overfitting by integrating over model parameters. But the\nquality of the prior information directly impacts the\nperformance of the Bayesian methods. Also, they are\nunable to account for feed - back regulation, a hal lmark of signaling networks.Text mining\nWith the growth of information in literature and\nbiomedical databases, biological and clinical scientists\nneed efficient means of handling and extracting d iagnostic methods and prognostic terms and information from scientific literature. For this purpose, text\nmining that comprises the discovery and extraction of\nknowledge from free text to generate new hypotheses\nparticularly relevant and helpful in biomedical r esearch [14]. Text mining complements the reading of\nscientific literature by individual researchers, allows\nrapid access to information contained in large volume\nof documents and increase s the reproducibility of\nliterature searches by enabling users to process all\ndocuments for a specific result. T he primary application of text-mining in biomedical research roughly lies\nin three aspects: 1) Simple text- mining such as\ntransforming textual information into database content and  integrating with existing knowledge resources tosuggest novel hypotheses ; 2) Literature analysis including clustering and classification of entities  or\ndiseases; 3) Integrative biology for producing or tes ting hypotheses against knowledge bases.Currently, text mining is being successfully a pplied to the identification of molecular causes of di seases using facts from databases and literature [75-77].\nFor example, text -mining has been used to suggest\ndisease biomarkers from the scientific literature, and\nmade on the basis of the assumption that two proteins\nare likely to interact with each other if they share a\nsubstantial amount of contextual information [78,79].\nBy defining a gene of interest , a  network is constructed from all scien tific publications related to the\nquery-defined gene. The results  can be browsed by\nnavigating through the visualized network . CoPub\nmakes use s of lexical resources for genes, proteins,\nGene Ontology labels, diseases, pathways, drugs and\ntissues to identify and statistically to qualify the significance of a specific term for a gene or a set of genes\n[80]. The results return  a set of annotations f or their\ngenes of interest. Besides, text mining has been widelygenes of interest. Besides, text mining has been widely\nused in industrial large scale knowledge base for\nquery genes, proteins, metabolic compounds and\ndrugs functional analysis.  To visualize knowledge\ncontained in the scientific literature, software tools\nhave been developed that provide improved integr ation of text -mining results with other data resources.\nFor example, IPA (Ingenuity) [56], KEGG [81], Pat hway Studio [82] and HPRD [83] use text -mining to\nintegrate gene/protein-phenotype associati ons lin king genes and protein variants to the diseases, toxic\nJournal of Cancer 2015, Vol. 6    http://www.jcancer.org  59  effects and drug response  to their knowledge dat abases.\nDepending on the tasks researchers address,\ntext-mining can achieve different objectives. This i nclude primarily the following: 1) retrieval information\nfrom relevant documents ; 2) Identification of entities\nsuch as genes, diseases, complex relationship betweensuch as genes, diseases, complex relationship between\nentities and diseases and interactions between proteins and genes [80]; 3) Deposit extracted information\ninto database or used to su pport manual database\ncuration efforts [15] ; 4)  Generation hypothesis\n[79]  and test  novel research questions  [78]. The trend of\ntext-mining technique is  shifting from the analysis of\nonly abstracts to the full text of papers, from the\nanalysis of gene and protein -related information to\nthe information about cells, tissues and whole orga nisms. The most prominent  shift is to integrate info rmation from the literature with data sets from other\ndomains such as gene expression profiles [84],\ngenome-wide associati on studies (GWASs), biochemistry and phenotype  [84,85]. Text-mining is prone to\nintegration with machine learning, statistical tec hniques. In the future, text -mining might face several\nmajor challenges such as improve literature analysis,major challenges such as improve literature analysis,\nintegrate to existing knowledge base, visualization of\nextracted information.\nMachine learning\nMachine learning methods have been used for\nthe biomarker discovery from high-throughput omics\ndata, inferring causal relations between mutations\nand diseases [21] , interactions  between genes and\nproteins [86-88] and relations between environ mental\nfeatures and cancer [89] as well as pathway and ne twork modeling. There are two kinds of basic machine\nlearning techniques, one is unsupervised machine\nlearning such as hierarchical c lustering,  self-organizing mapping (SOM) etc. [90]. The other is\nsupervised machine learning which needs known\nknowledge from data train a model and then apply\nthis model to predict the output variables [3]. A\nnumber of machine learning such as SVM [14], A rtificial Neural Network [ 91], decision tree and randomforests (RFs) etc. have been widely for various appl ications including identification of breast cancer b iomarkers [92], diagnosis biomarker of Parkinson di sorders [93], subcellular locations of protei ns [94,95],\nthe prediction of protein functions on the basis of\nprotein structures [96,97], the annotation of mutations\n[98,99]. For example, Han proposed a machine lear ning based derivative component analysis method to\nselect implicit feature by capturing  subtle data b ehaviors and removing system noises from a proteomic\nprofile to overcome the  reproducibility problem for\nbiomarker discovery in proteomics [100]. Another\ninteresting study by Hoshida et al [101] combined\neight independent cohorts of gene expr ession profiles\nto reveal the subclass of HCC and their related pat hways using unsupervised machine learning methods.\nThey found that three common subclasses (S1 -S3) of\nhepatocellular carcinoma (HCC) were significantlyhepatocellular carcinoma (HCC) were significantly\ncorrelated to Wnt pathway, MYC, AKT a nd hepat ocyte differentiation respectively. Westen blotting;\nknockout and immunohistochemical staining were\nused for experimental validation of their discovery.\nAnother framework called knowledge -driven matrix\nfactorization (KMF) proposed by Yang et al. was used\nto reconstruct phenotype -specific modular gene ne tworks [21].Integrative methods\nIntegration of data from multiple omic  studies\nnot only can help unravel the underlying molecular\nmechanism of carcinogenesis but also identify the\nsignature of signaling pathway/networks characteristic for specific cancer types that can be used for d iagnosis, prognosis and guidance for targeted therapy.\nThe methods described in Sections A -E have proven\nuseful for discovering biomarkers from  high-throughput omic data, analyzing pr otein-protein, protein -DNA, and kinase -substrate i nteractions, as well as for genetic interactions among\ngenes [102]. These efforts have yielded good results in\ncancer biomarker discovery, protein interaction and\ninteraction between genotype and diseases [103].\nHowever, current omic technologies provide only\nlimited fragmented reality of the biological functions\nwithin cell or cancer mechanism. Separate analysis of\nthe data generated from each of these technologies is\nlimited to revealing only partial aberrant molecularlimited to revealing only partial aberrant molecular\nchanges, because the interaction of multiple molecules\ncannot be modeled by isolated analysis of genes ,\nproteins or metabolites. Furthermore, limitations such\nas intrinsic high noise, incomplete data, small sa mple-size, bias have motivated the use of integrative\nomic analysis and use of prior biological knowledge\nand information bases, rather than as mere collections\nof single large -scale omic studies [14 , 34, 104]. However, integration of multiple disparate data types r emains a significant challenge in systems biology r esearch. Most recently, attempts at integration of mu ltiple high -throughput omics data have concentrated\non capturing regulatory associations between genes\nand proteins by comparing expression patterns across\nmultiple conditions [105 -107], combining functional\ncharacterization and quantitative evidence extractedcharacterization and quantitative evidence extracted\nfrom different data sources of  all levels of gene pro ducts, mRNA, proteins and metabolites, as well as their\ninteraction [108 -110]. Some previous works [81,\n111-113] in integrative analysis utilize pathways in the\nJournal of Cancer 2015, Vol. 6    http://www.jcancer.org  60  form of connected routes through a graph -based representation of the met abolic network [114]. Other a pproaches focus on the functional module of protein\ninteraction network and analyze experimental data in\nthe context of pathways using multiple source omics\ndata [14,115,116]. We and others have developed a dvanced bioinformatics tools and algorithms to facil itate the integration of diverse data types [34,\n110,\n117-120].\nDifferent biological types of data, such as s equences, protein structures and families, proteomicsDifferent biological types of data, such as s equences, protein structures and families, proteomics\ndata, ontologies, gene expression and other experimental data sets show a growing complexity pr oduced by numerous heterogeneous application areas.\nThe integration of heterogeneous data is therefore\nbecoming more and more important. In order to gain\ninsights into the complexity and dynamics of biological systems, the information stored in these data r epositories needs to be linked and combined in eff icient ways.\nApplication of biomarker discovery in\nHCC\nHepatocellular carcinoma (HCC) is the fifth most\ncommon malignancy and the third leading cause of\ncancer death in the world, with the five-year survival\nrate approaching 7% [33]. Treatments of HCC include\nsurgical resection and transplantation, ablation and\ntransarterial chemoembolization, and systemic che motherapy. Even so, no existing systemic chemotherapy\nis effective for advanced HCC [121,122]. For example,is effective for advanced HCC [121,122]. For example,\nLovet et al. [123] reported that targeted therapy\nwith  sorafenib which inhibits multiple tyrosine kinase r eceptors (RAS/VEGFR) may prolong survival by about\nthree months. However, due to the redundancy and\ncompensation of the signaling network in HCC , a\nsignificant reorganization of the signaling network\nobserved such as down regulation of tumor suppre ssors (p53 and CHK1 when XIAP silenced or p -RB\nwhen CDK6 silenced) and upregulation of tumor\npromoting proteins (ETS 1 when XIAP silenced or\np-CREB when CDK6 silenced) may confer the growth\nbenefit for cancer cells  [124]. This example suggests\nproviding pathways and network information may\nimprove the efficacy of systemic chemotherapy of\nHCC. Chang et al [125] partitione d the complex oncogenic signaling networks into basic units, or functional modules, of signaling activity (e.g., a protein\nphosphorylating another protein to activate its kinasephosphorylating another protein to activate its kinase\nactivity) and demonstrated that gene expression si gnatures based on these modules can predict the effectiveness of pathway -specific therapeutics  [125]. Except for surgical resection/transplantation of early\nstage HCC, the survival time is not significantly pr olonged by any of these treatments. Added to pathway\nand network centric method making use of omics data\nwith systematic chemotherapy will benefit  the development of newer therapeutic targets for HCC\ntreatment.In recent years, computational methods  for\nmodels take more and more important roles in the\nHCC investigations [114,126 ,127]. Some computer\nsystems have also been developed. For example,\nShannon et al. [128] developed a java based tool Ga ggle by integrati ng diverse databases (e.g., KEGG, B ioCyc, String) and software (e.g., Cytoscape, R ) to\nsimultaneously explore the experimental data (e.g.,\nmRNA and protein abundance, protein -protein and\nprotein-DNA interactions), functional associations,\nmetabolic pathways (KEGG) and Pubmed abstracts.\nRecently, Zheng et al [129] identified the molecular\nevents underlying the development o f HCV induced\nHCC by integrating gene expression profile and protein interaction data. To get the subnetworks, they\nrefined the network by removing a network comp onent if the number of nodes is smaller than five. They\nfound four subnetworks called normal -cirrhosis, ci rrhosis-dysplasia, dysplasia-early and early -advancedHCC networks. From each of the sub networks they\nidentified functional modules and hub genes. By\ncomparing the pathways in each sub networks, they\nobserved changes of pathways and network activities.\nTheir findings were validated by literature. Even\nthough the types of omics data they used only include\ngene expression and protein interactions, they pr ovide a way to study the changes of network activities\nby analysis of omic data. Zhang et al.  used systematical method including partial least squares, literature\nmining technique and with GeneGO Meta -Core to\ndiscover the biomarkers of HCC with gene expression\nas well as protein data. Based on these marker genes,\nthey constructed down regulated an d up regulated\nnetworks. In the former, they identified 10 up reg ulated hub genes (MAPK1, SP1, HDAC1, YY1, ABL1,\nPTK2, SMAD2, NCOA3; CDC25A and NCOA2). They\nidentified 7 hub genes (FOS, ESR1, JUNB, EGFR,\nSOCS3; FOLH1 and IGF1) in the latter. Partial leastidentified 7 hub genes (FOS, ESR1, JUNB, EGFR,\nSOCS3; FOLH1 and IGF1) in the latter. Partial least\nsquares were employed to construct a classifier with\nthese biomarkers. They used five-fold cross-validation\nand two independent datasets to evaluate the performance of the classifier. Furthermore, they used\nexperimental immunohistochemistry and western\nblot measurements to verify the marker genes\npredicted by the classifier. Their results show that the\nnetwork-based approach facilitates biomarker ident ification and improves classification accuracy [130].\nHollywod et al [131] identified driver genes which are\npotent diagnosis markers and mechanism study of\nHCC using t -statistic map (TM) and transcriptome\ncorrelation map (TCM) approaches with integration\nJournal of Cancer 2015, Vol. 6    http://www.jcancer.org  61  of DNA copy number measured by genomics CGH\narray and gene expression. They found 50 driver\ngenes with sign ificant prognostic relevance to HCCarray and gene expression. They found 50 driver\ngenes with sign ificant prognostic relevance to HCC\nkey signaling pathways such as mTOR, AMPK, and\nEGFR. siRNA-mediated knockdown experiments was\nused to evaluate the functional significance of the",
         "KEGG, HMDB and Reatcome are databases used for pathway and network knowledge.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "33",
         "How does age correlate with ACE2 expression in the endometrium during the menstrual cycle?",
         "Ismael Henarejos-Castillo et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.06.026:\n DISCUSSIONIt is important to understand the consequences of SARS-CoV-2 infection as the virus continues to spread worldwide. Fertility, specifically embryo implantation, is impacted by changes in endometrial gene expression throughout the menstrual cycle. Between menstrual cycle phases, gene expression varies to accompany physiologic changes . The differences in gene expression from one phase to another create a changing landscape for viral infection, and viral proteins themselves alter their expression throughout the menstrual cycle, setting the stage for a cyclical landscape of viral infectivity risk. ACE2, the receptor far for SARS-CoV-2 cell entry, showed low expression in the endometrium in our study, as reported previously in the Human Protein Atlas (HPA). ACE2 is also reduced in critical tissues such as lungs ; TABLE 1 Characterization of endometrial transcriptomic data sets with GEO data sets identifier, experiment name given for this study, cycle type, method of cycle phase dating, population from which samples were collected, age, transcriptomic platform used to measure gene expression, number of genes of each data set, number of samples in each data set, number of samples per cycle phase, and publication in which data were initially employed. GEO ID Experiment name Cycle type Cycle phase dating method Population Age Platform No. genes No. samples No. samples per cycle phase Reference GSE4888 Talbi 2006 Normoovulatory; regular (24-35 d); 3 mo since last hormone treatment Noyes et al.  reviewed by four pathologists Caucasian (n 1⁄4 17); black (n 1⁄4 6); Asian (n 1⁄4 1); other (n 1⁄4 2) 23-50 hgu133plus2 Affymetrix 19,361 27 PF (n 1⁄4 6); ESE (n1⁄4 4); MSE (n1⁄4 9); LSE (n1⁄4 8)GSE29981 Bradley 2010 Regular Days from LH peak: PF (LH-14-LH-1), ESE (LHþ1-LHþ4), MSE (LHþ6-LHþ7) Collected in Belgium 20-39 hgu133plus2 Affymetrix 19,361 19 PF (n 1⁄4 10); ESE (n1⁄4 6); MSE (n1⁄4 3) -\nGSE98386 Altm €ae 2017 Natural cycle Days from LH peak. We classified it in ESE1⁄4 LHþ2; MSE 1⁄4 LHþ8 Collected in Estonia - Illumina HiSeq 2500 16,426 38 ESE (n 1⁄4 19); MSE (n 1⁄4 19)\n\nGSE86491 Sigurgeirsson 2017 Regular; 3 mo since last hormone treatment Urinary LH ovulation predictor kit for MSE-LSE, days after the start of the subsequent menstruation for PF. Both confirmed by a gynecologic pathologist through histopathologic examination. Collected in Iceland 24-30 Illumina HiSeq 2500 15,939 14 PF (n 1⁄4 7); MSE (n 1⁄4 7)GSE119209 Kelleher 2017 -- Collected in U.S. - Illumina HiSeq 2500 17,934 11 PF (n 1⁄4 6); MSE (n 1⁄4 5) - Note: Last row indicates the total number of samples accounted and genes in common between all data sets for all menstrual cycle phases . ESE1⁄4 early secretory; GEO1⁄4 Gene Expression Omnibus; LH1⁄4 luteinizing hormone; LSE1⁄4 late secretory; MSE1⁄4 medium secretory; PF1⁄4 proliferative phase. Henarejos-Castillo. COVID-19 effects on the endometrium. Fertil Steril 2020. 226 VOL. 114 NO. 2 / AUGUST 2020 SEMINAL CONTRIBUTIONS  however, we cannot suggest that low levels ofACE2 expression imply no effect of the virus on the tissue. Further studies are needed to elucidate the effect of decreasedACE2 in the endometrium. It is interesting that ACE2 increased (fold change 1⁄4 2.47) from the early secretory to midsecretory phases, implying an increase inACE2 in the window of implantation and a high risk of viral infectivity at this stage of the menstrual cycle. Several possible mechanisms of SARS-CoV-2 cell entry have been proposed . TMPRSS2, the most-reported protease involved with SARS-CoV-2 infectivity alongsideACE2, had medium endometrial expression in our study. However, there was no correlation betweenTMPRSS2, ACE2, and the rest of the genes studied. These results imply that the endometrium should be safe against SARS-CoV-2 infectivity mediated by TMPRSS2, though the expression of other proteases associated with S protein cleavage show a different landscape than TMPRSS2. TMPRSS4 statistically significantly changes its expression throughout the menstrual cycle, showing a statistically significant increase in the midsecretory phase. This protein also showed an interesting correlation with other genes, including an increase alongsideACE2 in the early secretory phase. Although TMPRSS2 is implicated in cell entry, the fact that S proteins could be targeted byTMPRSS4 and that TMPRSS4 increased infectivity in gut epithelial cells  could mean a vulnerability of the endometrium to infection by SARS-CoV-2 mediated by TMPRSS4. Furthermore, TMPRSS4 up-regulation was correlated withCTSL and CTSB in all phases except the early secretory phase. Cell infection by SARS-CoV-2 was observed inTMRPSS2- cell lines expressing bothCTSL and CTSB, so these proteins may have a residual function of cleaving viral S protein . We propose that a synergy of CTSL and CTSB with TMPRSS4 through most of the cycle favors SARS-CoV-2 FIGURE 1 Endometrial data set integration and menstrual cycle clock. (A) Data set experiment effect and correction. Each point of the principal component analysis (PCA) plots represents endometrial gene expression of one sample and is colored by the endometrial transcriptomic data sets to which it belongs. Principal component 1 (PC1) and principal component 2 (PC2) explain the percentage of variability due to these components for each PCA. Integration of the transcriptomic studies showed a clear batch effect by the nature of each experiment (left PCA plot). After correction (right PCA plot), all genes in common between data sets were retained, amounting to a total of 13,437 genes, as no differentially expressed genes were detected between experiments. (B) Menstrual cycle effect. The samples of the integrated data sets are colored by the menstrual cycle phase to show how they are grouped by phases of the menstrual cycle. ESE1⁄4 early secretory endometrium; LSE1⁄4 late secretory endometrium; MSE1⁄4 midsecretory endometrium; PF1⁄4 proliferative phase. Henarejos-Castillo. COVID-19 effects on the endometrium. Fertil Steril 2020. VOL. 114 NO. 2 / AUGUST 2020 227 Fertility and Sterility®  FIGURE 2 Gene expression of viral infection-related genes throughout the menstrual cycle. (A) Landscape of expression changes. Genes were located depending on their relative expression against the whole set. Low, medium, and high expression thresholds correspond to 1% to 10%, 11% to 50%, and 51% to 100% categories of gene expression values of the entire integrated data set, respectively. Analysis of variance results for overall change of expressionduring cycle are shown for each gene. *P<.05; ***P<.0001. (B) Molecular scheme of SARS-CoV-2 endometrial infection.ACE2, TMPRSS4, FURIN,a n dBSGare shown in plasma membrane of an endometrial cell (lower leftfigure).CTSLandCTSBare represented outside the cell.MX1is shown in cytoplasm. Expression of viral genes in comparison to whole transcriptomic setis represented as arrows next to their names: up1⁄4 highly expressed; down1⁄4 lowly expressed. Viral genes are positioned in their schematized cell locations as stablished by GeneCards database>Localization section (release 4.14).14) . Only maximum confidence levels (5 and 4) for compartments-derived cell locations were used. Proteins were grouped considering the highest coactivation values between pairso fv i r a l genes during the menstrual cycle, which are shown in the lower right table in thefigure. Discontinuous arrow shows less evidence according to our results, given that onlyFURINshowed high activation withBSG and that further studies are needed to understandBSG-related mechanisms of SARS-CoV-2 entry. ESE 1⁄4 early secretory endometrium; LSE1⁄4 late secretory endometrium; MSE1⁄4 midsecretory endometrium; PF1⁄4 proliferative phase. Henarejos-Castillo. COVID-19 effects on the endometrium. Fertil Steril 2020. 228 VOL. 114 NO. 2 / AUGUST 2020 SEMINAL CONTRIBUTIONS  infectivity in the endometrium.FURIN, another protease predicted to cleave protein S , also showed a positive correlation with TMPRSS4 in the late secretory phase; with both targeting S protein, higher infectivity may occur. Finally, TMPRSS4 was positively correlated with MX1 in the midsecretory phase.MX1 actively participates in SARSCoV-2 infection by attracting neutrophils to infected tissue and employing their elastases to cleave protein S along with TMPRSS2 ; if TMPRSS4 participates alongside MX1,a more severe infection could occur during the window of implantation. TMPRSS4, CTSB, CTSL, FURIN, andMX1 showed statistically significant expression changes throughout the cycle. Protein expression of TMPRSS4, CTSB, FURIN, and MX1  supports a susceptible environment for infection in the early secretory and midsecretory phases. Future studies are needed to fully understand the mechanism of SARS-CoV2 cell entry and the roles ofTMPRSS4, CTSB, CTSL, FURIN, and MX1 in the endometrium. BSG, the alternative receptor for SARS-CoV-2 entry , showed high expression with statistically significant changes between phases and strong activation withFURIN through most of the cycle; however, its repression was correlated with ACE2 and TMPRSS2. Our results may indicate that only FURIN could be involved in S viral protein cleavage, if necessary, for SARS-CoV-2 binding toBSG, though other unknown proteases could also be involved. Further studies are needed to provide insight onBSG function in SARS-CoV-2 infection in the endometrium. We also investigated the effect of age on viral gene expression throughout the menstrual cycle. In a recent study FIGURE 3 Impact of age on viral-related infectivity gene expression throughout the menstrual cycle. (A) Effect of age onACE2 expression. Gene expression is represented forACE2 in each phase of the cycle according to the age of the sample analyzed. The range of age from patients involved in this study was 23 to 50 years. The range of age from patients involved in this study was 23 to 50 years. (B) Effect of age on viral gene expression. Pearson correlation R2 values are shown for each gene studied through of the phases of the menstrual cycle. Gray scale represents the magnitude of the correlation of increase or decrease in expression with age. High values are colored darker, and low values are colored lighter. ESE1⁄4 early secretory endometrium; LSE1⁄4 late secretory endometrium; MSE1⁄4 midsecretory endometrium; PF1⁄4 proliferative phase. Henarejos-Castillo. COVID-19 effects on the endometrium. Fertil Steril 2020. VOL. 114 NO. 2 / AUGUST 2020 229 Fertility and Sterility®  in lungs, ACE2 expression was detected as highly variable among individuals ranging from 0.17 transcripts per million to 14.5 transcripts per million in the same demographic group and highlighted a skewness of expression that statistically significantly increased with age. Our results showed a positive correlation between age andACE2 from the proliferative to midsecretory phases, especially in the early secretory phase, meaning that the endometrium in older women (to age 50) could be more susceptible to SARS-CoV-2 infection. TMPRSS2 decreased with age, particularly in the midsecretory and late secretory phases, which may mean that cellmediated entry of the virus is lowered with increased age. However, TMPRSS4 increased in expression from the proliferative to early secretory phases similar toCTSB, CTSL, and MX1. These results, in conjunction with ACE2 expression changes, highlighted the same behavior related to age as in the lungs , implying a higher risk of endometrial infection in older women, especially in the early phases of the menstrual cycle, as well as potential implantation problems. However, our results are limited to only one study with 27 patients, and there are no prospective studies showing the real clinical implications. ACE2 plays an important role in endometrial changes because it cleaves angiotensin II and impacts spiral artery vasoconstriction, posterior proliferation, and renewal of tissue . ACE2 down-regulation by SARS-CoV-2 could alter the endometrial balance ; however, given that the landscape of the tissue is modified with each menstrual cycle renewal, the consequences of SARS-CoV-2 infection could be reset from cycle to cycle. Because our study integrates data sets from GEO, this approach is limited by the study design. This also means that results depend on the experimental transcriptomic procedures of prior reported results and the sample cohort included in each study. However, the inclusion criteria for the selected studies were very careful, and uniform and raw data were preprocessed to control undesired effects. Likewise, the virus differentially affects individuals due to each person’s genetic profile. Likewise, the virus differentially affects individuals due to each person’s genetic profile . Further prospective research is needed to determine the mechanisms of SARS-CoV-2 infection in the endometrium and how it could affect the fertility of each patient.",
         "ACE2 expression shows a positive correlation with age from the proliferative to midsecretory phases, especially in the early secretory phase.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "34",
         "What do vertices and edges represent in correlation-based network analysis?",
         "Albert Batushansky et al.,2016, DOI:https://doi.org/10.1155/2016/8313272:\n Advanced technology methods for high-throughput biological studies, such as metabolomics and transcriptomics developed during the last decades, are successfully applied in biomedical research [\n1\n], plant studies [\n2\n], and microbiology [\n3\n]. The wide use of these technologies led to the accumulation of data on biological processes at their multiple levels (metabolic, genetic, enzymatic, physiological, phenotypical, etc.) and called for the development of tools to ease the visualization, analysis, and interpretation of an often complex and multidimensional matrix. Furthermore, the readily available “\nomics\n” technologies in biological laboratories prompted biologists to enter a field often needing extensive computational knowhow and led to the increased interest in biological interaction networks [\n4\n]. Thus, in the recent decades networks describing cellular processes were generated for human [\n5\n], yeast [\n6\n], and plants [\n7\n].5\n], yeast [\n6\n], and plants [\n7\n].\nNetworks can be presented as graphs, that is, a set of vertices (V) connected by edges (E), and consequently can be analyzed using graph theory, an approach that has been increasingly implemented in biological studies during the last decade. It is commonly accepted that graph theory as a scientific discipline was first used by the Swiss mathematician Leonhard Euler in 1735-1736, tackling the Königsberg bridge problem. Later, in the 19th and 20th centuries, graph theory was formulated and eventually introduced for applied fields, such as physics, computer science, and biology [\n8\n]. Today, graph theory consists of many tens of basic definitions and properties [\n98\n]. Today, graph theory consists of many tens of basic definitions and properties [\n9\n]. The understanding of the biological networks lies in the nature of the vertices and edges between them; that is, the vertices may represent one of the components of the three major molecular levels: genes, proteins, or metabolites, while the edges between them represent gene coexpression, protein-protein interactions, or biochemical conversions of metabolites, respectively [\n10\n]. However, molecular networks are not delimited to illustrate single-level component interactions. They can also show cross-level interactions. Alternatively, and perhaps a little counterintuitive, a network may incorporate vertices representing a set of metabolic reactions, where the connection between a pair of vertices is established if the reactions share one or multiple metabolites used or produced by these reactions [\n11\n,\n1211\n,\n12\n]. In other networks, vertices represent a community of molecular components, especially used with very vast data sets (>1000 of components) such as in weighted gene coexpression network analysis (WGCNA). Here, a single vertex delineates a module of genes and edges between vertices represent the correlation between them. This allows reducing the complexity of the network and simultaneously retains most of the information used for the interpretation of the gene coexpression results [\n13\n]. In simple words, vertices and edges represent the information as defined by the creator/user of the network.\nIn the last decade, correlation-based network analysis (CNA) has become a popular data-mining tool for visualizing and analyzing biological relationships within large data sets [\n13\n,\n14\n]. In this type of networks, vertices and edges represent molecular elements (e.g., metabolites or genes) and their correlation coefficient (strength and sign), respectively [\n10\n,\n15\n,\n1610\n,\n15\n,\n16\n]. Edges inferred by correlation analyses reflect a coordinated behavior between vertices across the data set (treatments, genotypes, conditions, and time). The type of correlation has to be selected based on the parametrical distribution of the data. In large population studies, data has to be tested for normality using existing tests, for example, the Shapiro-Wilk test. The Pearson correlation should be applied to normally distributed data, while Spearman's rank correlation should be used for data violating the assumption of normal distribution. CNA was successfully applied to various biological systems; it revealed, for example, metabolic markers related to plant growth and biomass in\nArabidopsis thaliana\nrecombinant inbred lines (RIL) and introgression lines (IL) [\n17\n,\n18\n], the role of gene Col5a2 in myocardial infarction [\n19\n], effect of hypoxia on tumor cell biochemistry [\n2019\n], effect of hypoxia on tumor cell biochemistry [\n20\n], and recently, identification of genetically based mechanism of the regulation of amino acid metabolism [\n2\n].\nGraph theory defines a number of network properties that allow successful analysis and interpretation of correlation networks (CN). These properties are a set of measures that describe the graph topology from different vantage points. CNs are undirected graphs, reflecting the coordinated behavior of two or more adjacent vertices (connected vertices) and the biological components they represent and not the effect of one vertex/component onto another, that is, a directed network. Properties that may have biological significance have been reviewed by Toubiana et al. [\n10\n]; they include (a)\nvertex degree\n: the number of edges incident on a given vertex [\n21\n], (b)\ncentrality score\n: reflecting the number of shortest paths between a vertex and any other vertex in the network, (c)\nnetwork diameternetwork diameter\n: the maximal shortest path between any two vertices in the graph, (d)\nnetwork density\n: the ratio of existing edges to the number of all possible edges of a network, (e)\nvertex betweenness centrality\n: the relative number of the shortest paths between any two vertices that pass via a specific vertex, and (f)\nmodules\n: subgraphs, within a global network characterized by higher connectivity (biologically interpreted as possible tighter coordination) between their components compared to other regions of the network. The analysis of these modules within the obtained network helped in the prediction of diseases [\n22\n,\n23\n]. In this contribution we aim at providing an easy-to-implement pipeline for the generation of CNs for biologists without extensive computational skills. To do so, we are demonstrating the potential use of CNs in cancer studies.Nowadays, there exist a number of software tools that allow researchers to generate networks, visualize them, and analyze their structure, via the calculation of a number of network properties, based on their own experimental data. Commonly known tools are Cytoscape [\n24\n], Gephi [\n25\n], and iGraph [\n26]. Each software has its benefits and disadvantages. For example, while iGraph requires programming skills and knowledge of the R programming language syntax, graphical-user-interface (GUI) based programs, such as Gephi and Cytoscape, do not, simplifying the interaction with the user. On the other hand, while script-based programs allow for the extension of existing functions and integration of compatible libraries, increasing the number of potential properties to be calculated, GUI programs are bound to the functionalities of the version of the software the researcher is using. However, Cytoscape and Gephi both offer a greater and easier-to-use set of visualization tools for networks, whereas the visualization functionalities of iGraph are rather limited and difficult to handle. Cytoscape allows for the integration of externally developed plugins, exerting functionality as desired by its developer. However, this option requires knowledge of the Java programming language and an understanding of how to interface it with the Cytoscape software.The current proposed stepwise pipeline allows the user to control each step of the network creation, as it provides flexibility in selection of correlation methods and thresholds and describes easy-to-handle options to analyze the network topology. The pipeline works irrespective of the nature of the data set and can be implemented by a combined use of the freely distributed Apache OpenOffice software (\nhttp://www.openoffice.org/\n), built-in packages within the R-environment [\n27\n], and Cytoscape [\n24\n].",
         "Molecular elements and their correlation coefficient.",
         null,
         null,
         null
        ],
        [
         "35",
         "What percentage of Taf4b-deficient oocytes exhibit some degree of asynapsis during pachytene?",
         "Kathryn J. Grive et al.,2016, DOI:https://doi.org/10.1371/journal.pgen.1006128:\n Coordinate expression of HumanTAF4B with expression of meiotic regulators and effectorsTo gain a better understanding of the potential molecular functions of TAF4B in human oogenesis, we examined coordinateTAF4B gene expression profiles in the human fetal ovary over gestational time [22], reasoning that the most essential functions of TAF4B may be highly conserved between mice and humans. We identified the genes that are most correlated with TAF4B expression during human ovarian development (S1 Table). To test if the list of genes highly correlated withTAF4B is enriched for specific functions, we evaluated the top 624 genes with Pearson correlations>0.85 for enriched pathways. Enrichment determined using Ingenuity Pathway Analysis (IPA) found thatTAF4B expression is most highly correlated with the expression of a network of meiotic regulators and effectors during human fetal ovarian development (Fig 1A). Pearson Coefficient values for a number of notable genes implicated in meiosis were calculated (Fig 1B), includingSYCP3 (R 2 = 0.87,P-value = 0.004),YBX2 (R2 = 0.87,P-value = 0.004),YBX2 (R2 = 0.90,P-value = 0.004), DAZL (R2 = 0.95,P-value = 0.004), the human ortholog ofDazl, MAEL (R2 = 0.93,Pvalue = 0.00398), a piRNA-pathway regulating gene essential for prophase progression and oocyte survival [23], andSTAG3 (R2 = 0.92,P-value = 0.00393), a cohesin required for proper synapsis [24- 26]. Interestingly, genes involved in primordial follicle formation were also highly correlated includingTEX14 (R2 = 0.91,P-value = 0.004), a component of intercellular bridges in germ cell cysts [27,28], DDX4 (R2 = 0.92,P-value = 0.004), the human ortholog ofVasa [29,30], andFMR1 (R2 = 0.97,P-value = 0.004), an RNA-binding protein associated with premature ovarian failure [31,32], among others. Together these data suggest that TAF4B may execute a highly conserved role in mammalian meiosis I regulation that is critical for healthy oocyte development and ovarian aging in women.Taf4b-deficient oocytes experience defective meiotic progression and chromosome synapsis at E16.5As many genes coordinately expressed with humanTAF4B in the human fetal ovary are critical for the fidelity of meiosis I, we analyzed prophase I progression inTaf4b-deficient mouse oocytes. Strikingly, when visualized by two different immunofluorescence methods, E16.5 Taf4b-deficient oocytes exhibit high degrees of asynapsis and aberrant meiosis. One means to test synapsis is by visualizing chromosomal co-localization of Synaptonemal Complex Proteins 1 and 3, which coat the central and lateral elements, respectively [33]. Complete co-localization indicates faithful synapsis, as seen in wild-type oocytes (Fig 2A- 2D), while regions of asynapsis only stain positively for SYCP3 and lack SYCP1, as observed in mostTaf4b-deficient oocytes (Fig 2E- 2H). Another means of testing synapsis is by counting chromosomal centromeric foci marked by Centromere Protein A (CENP-A) staining [34]. Complete synapsis of XX pachytene chromosomes will result in the appearance of 20 centromeric foci, as seen in wild-type oocytes (Fig 2I- 2L), while regions of asynapsis will result in non-overlapped centromeres and the appearance of greater than 20 CENP-A foci as seen in mostTaf4b-deficient oocytes (Fig 2M2P). Persistent double strand breaks are also evident inTaf4b-deficient oocytes as visualized by the presence of the histone variantγH2AX on pachytene chromosomes (Fig 2Q and 2R). Notably, nearly 75% ofTaf4b-deficient oocytes exhibit some degree of asynapsis during pachytene, a percentage significantly greater than that ever observed in wild-type oocytes (Fig 2S, p < 0.05).In addition to clear deficits in synapsis during pachytene,Taf4b-deficient oocytes also experience a prophase I progression defect. Meiotic stages of oocytes were determined by the configuration of SYCP3 staining at E16.5 (S1 Fig). While nearly 90% of wild-type oocytes have reached pachytene by E16.5,Taf4b-deficient oocytes persist in the earlier stages of leptotene and zygotene with only about 50% reaching pachytene at the same time (Fig 2T). Moreover, using MSY2 as a marker of diplotene arrest at PND0, we found that while about 40% of wildtype oocytes (visualized by TRA98 staining) have reached this developmental milestone, fewer than 10% ofTaf4b-deficient oocytes have reached this arrest (Fig 3). Strikingly, just one day later, at PND1, mostTaf4b-deficient oocytes undergo apoptotic cell death [19]. Finally, as proper synapsis is known to initiate at sites of recombination, the fidelity of meiotic recombination was tested by visualizing MLH1 foci on pachytene chromosomes. While wild-type oocytes possess one or two strongly-stained foci per homologous pair,Taf4b-deficient oocytes lack many of these sites as well as an overall decreased intensity of staining (Fig 4A). Quantification revealed a significant reduction of overall number of MLH1 foci on chromosomes fromTaf4b-deficient oocytes (p<0.0001) (Fig 4B). Wild-type oocytes average about 23 foci per cell, whileTaf4b-deficient oocytes average about 8 foci per cell with increased variability including many oocytes with no quantifiable foci. Given the loss ofTaf4b-deficient oocytes just days after these phenotypes are observed, the data suggest a TAF4b-dependent link between the ability to correctly progress and arrest in prophase I during embryogenesis and postnatal oocyte survival.TAF4b directly occupies promoter regions of critical meiosis and oogenesis regulatorsWhile TAF4b is evidently required for proper regulation of meiosis I progression, the nature of this regulation was unclear. To test if TAF4b directly occupies the proximal promoters of essential meiosis genes, we performed ChIP from wild-type E18.5 ovaries using a validated anti-TAF4b antibody (S2 Fig) or a negative control IgG antibody, and performed quantitative PCR. Strikingly, immunoprecipitated chromatin bound by TAF4b included the proximal promoters ofStra8 and Dazl (Fig 5A). As oogenesis regulatorsFiglα and Nobox are known downstream targets of DAZL [35], these promoters were also tested and found to be directly bound by TAF4b (Fig 5A and 5B). TAF4b occupancy at these important loci is specific, as genomic regions not expected to be occupied by TAF4b, including a non-genic region 50kb upstream of Nobox were not enriched for TAF4b (Fig 5B). Quantitative PCR results were validated by gel electrophoresis and visualization of amplified proximal promoters (S3 Fig). To determine whether TAF4b occupancy at some of these promoters correlates with differences in their protein expression, E18.5Taf4b-heterozygous andTaf4b-deficient ovaries were stained by immunohistochemistry for NOBOX protein. Although there are no statistically significant differences in germ cell density at this time point [19], NOBOX expression is dramatically reduced inTaf4b-deficient oocytes (Fig 6A and 6B). Additionally, PND0Taf4b-heterozygous and Taf4b-deficient ovaries were stained by immunofluorescence for DAZL protein, which was also observed to be dramatically reduced inTaf4b-deficient oocytes (S4A and S4B Fig), a result which was confirmed by immunoblotting (S4C Fig). To better understand if meiotic onset may also be affected inTaf4b-deficient oocytes, mRNA expression ofStra8 was tested at E13.5, the developmental time at which it is first activated by RA [3].5, the developmental time at which it is first activated by RA [3]. Notably,Stra8 expression inTaf4b-deficient ovaries is approximately 25% of that expressed in wild-type ovaries at the same time (Fig 7). Genes downstream ofStra8, including Sycp1 and Sycp2, exhibit similar reduced expression, suggesting that TAF4b may be important for meiotic onset as well as progression through direct transcriptional modulation of Stra8. In contrast to these oocyte-specific genes,Wnt4 expression [36] is not significantly changed inTaf4b-deficient ovaries at this time, suggesting that TAF4b is specifically promoting oocyte-specific gene expression programs. Given that oocytes are not mitotic after E13.5, and as we have previously demonstrated equivalent oocyte densities inTaf4b-deficient and wildtype ovaries at E18.5 [19], these data likely reflect alterations in gene expression and not reduced germ cell numbers. Accordingly, MVH-stained E13. Accordingly, MVH-stained E13.5 ovaries display relatively equivalent numbers of germ cells acrossTaf4b genotypes (S5 Fig). Proper TAF4b regulation thus leads to the proper expression of downstream genes regulating multiple stages of meiosis I including onset, synapsis, recombination and arrest.",
         "Nearly 75% of Taf4b-deficient oocytes exhibit some degree of asynapsis during pachytene.",
         null,
         null,
         null
        ],
        [
         "36",
         "How many genes and transcripts are expressed in neuroblastoma, according to RNA-seq?",
         "Wenqian Zhang et al.,2015, DOI:https://doi.org/10.1186/s13059-015-0694-1:\n Background: Gene expression profiling is being widely applied in cancer research to identify biomarkers for clinical endpoint prediction. Since RNA-seq provides a powerful tool for transcriptome-based applications beyond the limitations of microarrays, we sought to systematically evaluate the performance of RNA-seq-based and microarraybased classifiers in this MAQC-III/SEQC study for clinical endpoint prediction using neuroblastoma as a model.Results: We generate gene expression profiles from 498 primary neuroblastomas using both RNA-seq and 44 k microarrays. Characterization of the neuroblastoma transcriptome by RNA-seq reveals that more than 48,000 genes and 200,000 transcripts are being expressed in this malignancy. We also find that RNA-seq provides much more detailed information on specific transcript expression patterns in clinico-genetic neuroblastoma subgroups than microarrays. To systematically compare the power of RNA-seq and microarray-based models in predicting clinical endpoints, we divide the cohort randomly into training and validation sets and develop 360 predictive models on six clinical endpoints of varying predictability. Evaluation of factors potentially affecting model performances reveals that prediction accuracies are most strongly influenced by the nature of the clinical endpoint, whereas technological platforms (RNA-seq vs. microarrays), RNA-seq data analysi s pipelines, and feature levels (gene vs. microarrays), RNA-seq data analysi s pipelines, and feature levels (gene vs. transcript vs. exon-junction level) do not significan tly affect performances of the models. (Continued on next page)Conclusions: We demonstrate that RNA-seq outperforms microarrays in determining the transcriptomic characteristics of cancer, while RNA-seq and microarray-based models perform similarly in clinical endpoint prediction. Our findings may be valuable to guide future studies on the development of gene expression-based predictive models and their implementation in clinical practice.",
         "More than 48,000 genes and 200,000 transcripts are expressed.",
         null,
         null,
         null
        ],
        [
         "37",
         "What databases were searched for articles related to COVID-19 in pregnant women?",
         "Jianhua Chi et al.,2020, DOI:https://doi.org/10.1007/s00404-020-05889-5:\n Search strategy and selection criteria An electronic search of PubMed, Embase, Medline,  MedRxiv, CNKI (China National Knowledge Infrastructure),  and the Chinese Medical Journal Full Text Database (http:// journ al.yiigl e.com/) was performed following Preferred  Reporting Items for Systematic reviews and Meta-Analyses  (PRISMA) guidelines [4] through to April 18, 2020. Search  terms were (COVID-19 OR 2019-nCoV OR SARS-CoV-2  OR 2019 Novel coronavirus) AND (pregnant women OR  Maternal OR parturient OR mothers OR Infant OR Newborn  OR neonate OR Infectious Disease Transmission, Vertical). The titles and abstracts in the search results were independently reviewed by two authors (Jianhua Chi and Wenjian Gong) to determine whether the article was relevant  to our research topic. The full texts of the literature we  identified were further searched, and we read the articles  to identify those eligible that ultimately met the inclusion  and exclusion criteria. The inclusion criteria were as follows: 1, pregnant women were diagnosed with COVID-19  by dual fluorescence polymerase chain reaction (PCR) or  quantitative real-time polymerase chain reaction (qRT-PCR),  or were clinically diagnosed cases according to the latest  clinical guidelines at the time of the article’s publication; 2,  articles that described clinical characteristics and maternal  and newborn outcomes. The exclusion criteria included, 1,  duplicated cases and 2, newborns who tested positive for  COVID-19 but lived with COVID-19 relatives to avoid confounding factors when assessing vertical transmission. We  abandoned duplicated cases by referring to the birthplace,  times of admission and delivery, specific maternal characteristics of cases, as well as dates and authors of publications  to eliminate potential overlap of cases. Data extraction, assessment of risk of bias, and data  synthesis The following variables were extracted from eligible literature: study type, time, region, inclusion criteria, sample size,  epidemiological history of mothers and newborns related to  COVID-19, delivery mode (cesarean section/vaginal deliv - ery), cesarean section indications, fetal sex (male/female),  gestational age, birth weight, clinical diagnosis, imaging,  oxygen therapy, antiviral treatment, breastfeeding, maternal  and newborn outcomes, case admission time and hospital  name, number of pregnant women, clinical characteristics  and outcomes of pregnant women, number of newborns,  characteristics and outcomes, method of diagnosis and number of positive diagnoses. To avoid bias, the authors received  training, established a consensus, read all articles separately,  and discussed any ambiguities. We adopted an approach specially used for case reports  and case series that was proposed by Murad et al. [5 ] to  evaluate the quality of the included articles. According to  this method, two authors read all articles and considered four  aspects, selection, ascertainment, causality, and reporting.",
         "PubMed, Embase, Medline, MedRxiv, CNKI, and the Chinese Medical Journal Full Text Database were searched.",
         null,
         null,
         null
        ],
        [
         "38",
         "What percentage of patients in the study used transdermal estrogen therapy?",
         "Monika Obara-Moszynska et al.,2021, DOI:https://doi.org/10.3389/fendo.2021.707031:\n Our study evaluated uterine de velopment during pubertal induction in a homogenous group of girls with TS with hypergonadotropic hypogonadism. Most of the patients (87.5%) used transdermal estrogen therapy. The studied group was compared to the healthy population at prepubertal and pubertal ages. The mean age of TS patients at ERT initiation was more advanced than that of the prepubertal control group. The time of puberty induction (12.9 years) was in line with the recommendations , and the control group had to be younger, as puberty in healthy girls starts at an average age of 10.5 years . Most studies estimating uterine development during ERT in individuals with hypogonadism have analysed the effects of oral estrogens  and/or late-onset pubertal induction ; some present patients with different hypogonadism aetiologies  or do not compare the studied group to the control group. The optimal imaging technique and the most objective and useful measurements of the uterus for monitoring uterine development have not yet been determined. The criteria comprising uterine maturity vary (FCR, uterine length, and UV). In our opinion, the appreciation of uterine development based on the entire size is more reliable than approaches based on a single measurement. The research conducted by Hagen et al.  in healthy girls showed a strong correlation between the UV calculated with the ellipsoid formula from the 2D TAUS measurements (length, depth and width) and the volume assessed using both the threedimensional (3D) TAUS model and the ellipsoid MRI model; however, the differences between the methods were significant FIGURE 2 |U V  and FCR: (A) at the start of ERT, (B) after 6-12 months of ERT, (C) after 36 months of ERT or 12 months after menarche. TABLE 2 |Ultrasound parameters of the uterus in Turner syndrome patients and in the control group. TABLE 2 |Ultrasound parameters of the uterus in Turner syndrome patients and in the control group. TS at the initiation of ERT Prepubertal control group TS after 6-12 months of ERT TS mature uterus Pubertal control group UV [cm3] Mean ± SD 1.55 ± 1.22 1.96 ± 0.94 9.27 ± 5.63 31.04 ± 11.78 45.68 ± 12.51 Median 1.11 1.78 8.65 29.87 43.25 Range 0.33-4.76 0.93-4.7 1.65-25.05 13.42-64.29 27.5-74.5 p-value 0.09 < 0.001 FCR Mean ± SD 0,96 ± 1.22 1.04 ± 0.24 9.27 ± 5.63 1.4 ± 0.33 1.47 ± 0.27 Median 1.0 1.0 1.04 1.37 1.45 Range 0.67-1.67 0.66-1.8 1.44-1.83 0.95-2.56 0.93-2.06 p-value 0.22 0.18 TABLE 1 |Clinical characteristics of the study group. Number of patients N = 40 Age at initiation of E2 therapy, years (range) 12.9 ± 1.2 (10.5-15.4) Bone age at initiation of E2 therapy, years (range) 11.7 ± 1.0  Age at the last visit, years (range) 16.1 ± 1.6 (13.8-18.0) Height at the last visit, cm 153.8 ± 6.5 Height SD at last visit -1.8 ± 1.1 BMI at the last visit, kg/m 2 23.3 ± 4.8 ± 6.5 Height SD at last visit -1.8 ± 1.1 BMI at the last visit, kg/m 2 23.3 ± 4.0 BMI centile at the last visit, kg/m2 72.5 ± 24.0 Body surface at the last visit, m2 1.5 ± 0,2 Growth hormone therapy,n (%) 39/40 (97.5%) Karyotype 45, X,n (%) 21/40 (52.5%) Obara-Moszynska et al. Uterine Development in Turner Syndrome Frontiers in Endocrinology | www.frontiersin.org July 2021 | Volume 12 | Article 7070315  and increased with increasing volume. Considering the shape of the uterus, we decided to use other geometric formulas (elliptical cylinder for the prepubertal uterus and combined ellipsoid and elliptical cylinder for the pubertal and mature uterus) to obtain the most reliable UV with 2D TAUS. Further studies with 3D US are necessary to verify this approach. Nevertheless, in contrast to 3D US, 2D TAUS is widely available in clinical practice and therefore is the standard imaging modality used to evaluate internal genital organs in girls. A validated normative model of UV, elaborated by Kelsey et al. , shows that the predicted volume of the uterus in healthy young women at the age of 16 years is 48.2 cm 3 (68% prediction limit: 27.7-77.83 cm3), which is comparable to that in our pos tmenarcheal controls. To derive the model, the authors used the results of pelvic MRI from 87 paediatric patients and the results of uterine US from 1331 females aged 0-40 years. In MRI, the UV was obtained from the region of interest (ROI) drawn, whereas in US, UV was calculated using the modi fied prolate ellipsoid formula . The lack of differences in uterine size between girls with TS prior to the beginning of pubertal induction and healthy prepubertal controls aged 5-8 years suggests that the very low estrogen secretion, which is observed in healthy ovaries in the prepubertal years , exerts a rather insignificant effect on uterine development. This result suggests that promoting uterine development with very-low-dose estrogen in the prepubertal years is unfounded. Trials using ultralow-dose oral ethinylestradiol as a growth-promoting agent during the prepubertal period combined with GH demonstrated a modest increase in adult height, normalization of the timing of thelarche for approximately onequarter of the girls, and modest improvements in cognition and memory . However, as the dosing and administration of childhood estrogen have not been optimized and long-term safety has not been assessed, the addition of very-low-dose estrogen replacement in the prepubertal years in TS patients is currently not recommended . In healthy girls, a significant increase in the size of the uterus is concurrent with pubertal development, and the maximum velocity of the increase in UV occurs less than a year after peak height velocity. Pubertal induction in our TS patients resulted in a significant increase in UV, which was already evident after thefirst 6-12 months of ERT with low E2 doses. The fundal-cervical ratio, the other parameter evaluating uterine growth, did not change significantly during thisfirst period of treatment. The discrepancy between these two uterine size parameters was also noticed at the final stage of uterine development: the last evaluated UV in girls with TS was significantly smaller than that in the age-matched controls, in contrast to the FCRs, which were similar in both groups. In the majority of subjects, uterine assessment was performed when the girls had a full urinary bladder. In TAUS, the full urinary bladder creates an acoustic window that facilitates evaluation of the uterus; however, it may alsoflatten the uterus and modify the measurements, especially when thefilling is excessive. For this reason, we consider UV, the calculation of which involves all uterine dimensions (longitudinal, AP and transverse of the uterine body and the cervix) to be the more reliable parameter of uterine development and size. Because in our study, the uterus appeared sensitive to very low estrogen levels, we suggest that proper uterine measurements might be used to titrate estradiol dosages during the induction of puberty. In contrast to serum E2 concentrations, uterine growth also re flects tissue sensitivity to estrogens. In the healthy population up to the age of 40, approximately 84% of the variation in UV is due to age alone, with a dramatic increase in uterine size observed from age 10 to 15 . In a study published by Burt et al. In a study published by Burt et al. , which involved women with hypogonadism who received exogenous estrogens for pubertal induction from the mean age of 15 years, 84% of patients on full estrogen replacement attained FCR>1, and 48% had UVs at less than the 5th percentile of the reference group. Compared to the participants in the aforementioned study, our patients started ERT earlier (mean age 12.9 years), and although they were younger at the final evaluation (mean age 16.1 versus 24 years), they achieved better uterine development: all but two (95%) attained a mature uterine configuration with FCR>1, and 61% had afinal UV above the 5th percentile of the reference group. In a study by Gawlik et al. , which involved girls with TS with late-onset ERT (mean age 15.1 years), the initial UV was comparable with our results, but the final UV seemed to be significantly smaller than that in our subjects (10vs 30 cm 3). Considering the postulated existence of an optimal time window for uterine development in adolescence, we can assume that in our patients, we probably applied ERT closer to this optimal window; nevertheless, we found no correlation between age at the onset of ERT andfinal uterine size. Karyotype, body size parameters andfinal estradiol dose also had no impact on thefinal UV. It must also be emphasized that the smaller UV of the TS patients after puberty induction compared to the healthy population may result from a shorter period of estrogen exposure. Girls with TS usually started ERT later than natural puberty and at thefinal visit, although the chronological ages of the TS and control groups were comparable, their gynaecological ages were different. The average age at menarche in the Polish population is 12.6 years, similar to the mean age at the onset of ERT in our study group.6 years, similar to the mean age at the onset of ERT in our study group. Uterine growth and maturation may continue beyond the time of menarche; therefore, patients with TS should be monitored for further development of the uterus. There are several limitations of our study. Thefirst one is study design (retrospective analysis). The second limitation, which was already mentioned, is the imaging technique used to evaluate UV. Further studies with the use of 3D US, which facilitates the acquisition of volumes that are more reliable and less dependent on the examiner, could provide more accurate data. The third limitation is the size of the study group and of the control groups and the relatively short time of observation. As the maturation of the uter us continues beyond the first menstrual bleeding, extension of this study and evaluation of the uterus in patients at the childhood-adulthood transition should be performed. Finally, we cannot exclude periodic noncompliance with ERT in some of our girls with TS, which could have affected the results.",
         "87.5% of the patients used transdermal estrogen therapy.",
         null,
         null,
         null
        ],
        [
         "39",
         "What role does embryo-endometrial communication play in pregnancy?",
         "Hsien-Ming Wu et al.,2022, DOI:https://doi.org/10.3390/ijms232113382:\n The embryo–endometrial communication has a critical role in the successful embryo implantation and pregnancy [1]. Maternal immune modulation during embryo implantation and subsequent pregnancy provides a tolerant microenvironment to keep the fetus from rejection [2]. Since the first human IVF baby was born in 1978, pregnancy and live birth rates following assisted reproductive technologies have improved significantly. Even so, there is still a gap in achieving higher pregnancy and live birth rates, attributed to unresolved problems in embryo quality, endometrium receptivity, and embryo–endometrium interaction [3,4]. Usually, a steroid hormone primes the endometrium for the receptive phase; after that, it is further immunomodulated by embryo signals for subsequent embryo implantation and maintenance of pregnancy [5]. Therefore, successful embryo implantation and the ensuing pregnancy depend on the coordinated interaction between the endometrium and embryos. The classical embryo–endometrial interactions through dynamic intracellular and secreted protein alterations in the stage of embryo implantation are performed to promote the successful maintenance of pregnancy [6]. The role of extracellular vesicles (EVs) and microRNAs in the decidualization for the embryo–maternal interactions of decidual endometrial stromal cells has been clarified in the implantation of the embryos [7]. EVs are cell-originated membranous vesicles that deliver bioactive molecules from cells to cells [8]. The cargo of EVs mediates cell–cell communication by delivering numerous factors, including DNA, microRNAs, proteins, and lipids [9]. Innovative insights and approaches have explored the broad features of EVs, including ligands, surface receptors, and cargo from the mother cells. Moreover, the lipid bilayer membranes of EVs contain relatively high levels of ceramide, cholesterol, and sphingomyelin, causing EVs to be steady in extracellular spaces [10]. The biochemical and molecular characteristics of EV cargo from cells of the female reproductive tissue, placenta, and embryo have been recently demonstrated [7,11,12]. In this review, we will highlight the roles of human decidual stromal cell-associated EVs in the immune tolerance of embryo implantation and pregnancy.",
         "It has a critical role in successful embryo implantation and pregnancy.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "40",
         "How many genes from the COSMIC database were initially identified as causally implicated in cancer?",
         "Anja C Gumpinger et al.,2020, DOI:https://doi.org/10.1093/bioinformatics/btaa452:\n 3 Results\n3.1 Dataset description\nIn order to find novel cancer driver genes, we combine data from a The Cancer Genome Atlas (TCGA) pan-cancer study of 9 423 tumor exomes (comprising all 33 of TCGA projects;\nBailey\net al.\n,\n2018\n) with the well-established InBio Map PPI network (\nLage\net al.\n,\n2007\n;\nLi\net al.\n,\n2017\n). The network constitutes our view of interactions between genes on a protein level. The network has an average degree of 61.02 (±128.33), and the sizes of the\nk\n-hop neighborhoods are illustrated in\nFigure 2a\n. We represent each node in the network with its −log\n10\ntransformed MutSig\nP\n-value (\nLawrence\net al.\n,\n2014\n). Those\nP\n-values measure whether a gene shows significantly different mutational patterns in tumor versus normal tissues. In total, we have access to\nP\n-values for 18 154 genes. As a pre-processing step, we remove all nodes from the network that cannot be represented with a MutSig\nPP\n-value, as well as all isolated nodes. This results in a total of 11 449 genes that are present in the InBio Map network, are connected to at least one other node and have been tested with the MutSig tool. Those constitute our candidates for network-based prediction of cancer driver genes.\nFig. 2.\nOpen in a new tab\nDataset description: (\na\n) the distribution of neighborhood sizes, for neighborhoods defined as in\nEquation (1)\n, for\nk\n=\n1\n,\n2\n,\n3\n. (\nb\n) The distribution of the node degree, shown for 635 cancer genes and 10 816 unlabeled genes. (\nc\n) The distribution of the MutSig\nP\n-values, in cancer genes and unlabeled genes. (\nd\n) The correlation between the degree and the MutSig\nP\n-values for cancer genes and for unlabeled genes\nClass labels\nIn general, supervised machine learning requires access to labeled data to train a classifier. To obtain labels for the genes, we use the CGC data from the COSMIC database (\nSondka\net al.\n,\n2018Sondka\net al.\n,\n2018\n). We downloaded a list of 723 genes that have been causally implicated in cancer, and use this set as our ground truth. Genes in the CGC are categorized into Tiers 1 and 2, where genes in Tier 1 show a documented activity relevant to cancer, and genes in Tier 2 show strong indications to play a role in cancer. For our analysis we treat both tiers equally. We overlap the set of 723 genes with our network, giving a total of 635 cancer genes. This leads to a dataset, in which ‘positive’ samples make up <6.0% of our dataset. We refer to the remaining genes as\nunlabeled genes\n, and we are interested in finding new cancer genes among them.\nUsing the CGC genes, we observe a knowledge-bias in the InBio Map PPI network (see\nFig. 2b\n), that is cancer genes tend to have higher degrees in the network. We furthermore observe an increased correlation between degree and MutSig\nPP\n-values for cancer genes (Pearson correlation: 0.17) compared with unlabeled genes (Pearson correlation: 0.10), as can be seen in\nFigure 2c and d\n. Although this indicates that MutSig can identify the highly mutated cancer genes, there exist many well-established cancer genes whose mutation rates lie within the background distribution (i.e. their MutSig\nPP\n-values are undistinguishable between cancer genes and unlabeled genes). This poses three challenges that have to be addressed: (i) we do not have a high-quality negative class, i.e. in general any gene not classified as a cancer gene might potentially be a cancer driver, (ii) the dataset is imbalanced, a fact that requires attention during supervised classification and (iii) the dataset is affected by knowledge contamination. We address challenges (i) and (ii) with an elaborate and unbiased cross-validation procedure to train and test a classifier, as well as to predict cancer driver genes from the unlabeled genes. The third challenge is addressed by using the moments in the MoPro embeddings. Although we observe that moments such as skewness and kurtosis exhibit positive correlations with the node-degree, this is the case for both, cancer genes and unlabeled genes (see\nSupplementary Fig. S1\n).\n3.2 Experimental setup\n3.2.1 Cross-validation for one-class, imbalanced learning).\n3.2 Experimental setup\n3.2.1 Cross-validation for one-class, imbalanced learning\nTo address the above mentioned challenges imposed by the class imbalance and the lack of a negative class, we developed a cross-validation procedure that is based on the repeated undersampling of the majority class. The cross-validation procedure is illustrated in\nFigure 3\n, and a pseudocode can be found in the\nSupplementary Algorithm S1\n. The dataset can be represented as a matrix\nD\n∈\nℝ\n11\n449\n×\nd\n, where\nd\nis the number of node features (\nFig. 3a\n). The cross-validation procedure consists of three main steps:\nFig. 3.\nOpen in a new tab\n(\na\n) The dataset consists of 11 449 genes, each one represented by a set of features. 635 of those genes are classified as cancer genes (\nSondka\net al.\n,\n2018\n), highlighted in yellow. The remaining 10 814 genes are unlabeled (green). Cross-validation scheme on\none\ndata split, resulting in one best classifier\nC\n′\n. (\nbone\ndata split, resulting in one best classifier\nC\n′\n. (\nb\n) The unlabeled genes are sub-sampled at random and combined with the cancer genes, giving rise to the labeled dataset\nD\nl\n. The unlabeled genes in\nD\nl\nare assigned a negative class label. The remaining unlabeled genes make up the set\nD\nu\n. Those are the genes for which the cancer status will be predicted in the current split. (\nc\n) The\nD\nl\ndataset is split, and 80% of the data are used to find the best hyperparameters of the classifier via 5-fold CV, resulting in the classifier\nC\n′\n. The remaining 20% are used as test set for evaluation of the classifier\nC\n′\n. (\nd\n) The classifier\nC\n′\nthat has been trained on the cross-validation set in\nD\nl\nis used to predict the cancer status of genes in\nD\nu\nStep 1: Data splits\nWe split the dataset\nD\ninto two disjoint datasets,\nD\nl\nand\nD\nu\n(\nFig. 3b\n), where\nD\nlWe split the dataset\nD\ninto two disjoint datasets,\nD\nl\nand\nD\nu\n(\nFig. 3b\n), where\nD\nl\nconsists of all genes in the positive class, and a random subsample of the unlabeled genes. We undersample the majority class such that 10% of samples in\nD\nl\nare cancer genes. For the sake of training a classifier, we assign the unlabeled samples in\nD\nl\nto the negative class, and assume this to be the ground truth for the current split. This dataset will be used in the second step to train and evaluate a classifier. The genes in\nD\nu\nremain unlabeled, and we use the classifier trained on\nD\nl\nto predict their cancer status.\nStep 2: Training and evaluation of the classifier\nNext, the dataset\nD\nl\nis split into a cross-validation (80% of data) and a hold-out test set (20% of data;\nFig. 3c\n). On the cross-validation set, we do a 5-fold stratified cross-validation to find the best hyperparameters of the classifier\nC\n, resulting in\nC\n′C\n, resulting in\nC\n′\n. We retrain the classifier on the complete cross-validation set, and evaluate the predictive performance of\nC\n′\non the hold-out test set. Importantly, the cross-validation and hold-out test sets are disjoint. This implies that samples in the hold-out test set have never been seen during training, nor were they used to choose the best hyperparameters of the classifier. This set is solely used to evaluate the ability of the classifier to generalize to unseen samples. The strict separation",
         "A list of 723 genes from the COSMIC database have been causally implicated in cancer.",
         null,
         null,
         null
        ],
        [
         "41",
         "What are the three methods offered by clusterProfiler for gene classification and enrichment analyses?",
         "Guangchuang Yu et al.,2012, DOI:https://doi.org/10.1089/omi.2011.0118:\n In this work, we present a new ontology-based tool, clusterProfiler, that offers three methods,\ngroupGO\n,\nenrichGO\nand\nenrichKEGG\n, for gene classification and enrichment analyses. Most importantly, clusterProfiler applies biological term classification and enrichment analyses to gene cluster comparison, helping to better understand higher order functions of biological system. We also adopted a model design with more flexibility. The comparison function of clusterProfiler is designed as a general framework for comparing any kind of biological and biomedical ontologies as demonstrated in the online vignette. In addition, clusterProfiler provides a visualization module for displaying analysis results.This package is a simple-to-use tool, specifically designed for biologists who wish to analyze high-throughput data obtained from transcriptomics or proteomics. It can be easily extended to support new organisms, and integrated into pipelines for high-throughput data analysis, especially in cooperating with other Bioconductor packages.We plan to employ three strategies to improve this package in the future. First, we will use semantic similarity among KEGG pathways and GO terms to aggregate closely related categories. This can be anticipated to result in more interpretable outcomes. Second, we will rank gene similarities in each cluster and then correlate them to search for enriched categories. This will provide more sensibility for finding active gene modules. Third, we will develop a statistical model based on the induced directed acyclic graph to compare functional profiles as a whole rather than a set of unrelated categories. These strategies will increase the versatility of clusterProfiler.",
         "groupGO, enrichGO, and enrichKEGG.",
         null,
         null,
         null
        ],
        [
         "42",
         "How did hormonal contraceptives affect serum E2 and E1 concentrations?",
         "Kaisa Huhtinen et al.,2012, DOI:https://doi.org/10.1210/jc.2012-1154:\n Tissue E2 and E1 concentrations and the ratio of local to systemic E2 and E1\nEndometrium\nEndometrial intratissue E2 concentration was up to 10-fold higher in the proliferative phase compared with the secretory phase (\nP\n= 0.003;\nTable 1\n). No difference in tissue E2 concentrations was observed between healthy and eutopic endometrium. Interestingly, a similar strong cycle dependent change was not observed in the E1 levels (\nP\n= 0.751 for PE;\nTable 1\n). As a result, the ratio of E2 to E1 in the endometrium tissue decreased markedly from proliferative to secretory phase without a difference between women with and without endometriosis (\nP\n< 0.001 for PE;\nTable 1\n).\nTable 1.\nSystemic and local estrogen concentrations in women with and without endometriosis\nSample type\nHormonal status\nn\nE2 (pg/ml)\nE1 (pg/ml)\nE2/E1 ratio\nMedian\n25/75%\nP\nMedian\n25/75%\nP\nMedian\n25/75%\nP\nSerum\nHealthy\nProliferative\n9\n77.1\n48.0/110.0\n59.3\n48.1/77.0\n1.37\n0.97/1.81\nSecretory\n7\n157.3\n117.2/174.7\n#\n78.4\n68.3/96.09\n77.1\n48.0/110.0\n59.3\n48.1/77.0\n1.37\n0.97/1.81\nSecretory\n7\n157.3\n117.2/174.7\n#\n78.4\n68.3/96.0\n1.76\n1.55/2.02\nEndometriosis\nProliferative\n19\n68.7\n52.1/195.0\n63.4\n49.1/90.8\n1.46\n0.89/1.98\nSecretory\n20\n109.8\n76.8/176.0\n74.9\n51.7/97.5\n1.50\n1.16/2.14\nMedication\n21\n2.5\n2.5/58.6\n†\n21.3\n16.6/96.1\n†\n0.13\n0.12/0.80\n†\nTissue\nHealthy endometrium\nProliferative\n9\n532.0\n334.7/736.0\n55.6\n41.7/67.1\n8.34\n7.33/9.27\nSecretory\n7\n66.0\n52.5/100.8\n##\n85.1\n72.1/106.0\n#\n0.69\n0.50/1.40\n###\nPatient endometrium\nProliferative\n9\n649.3\n404.0/1168.7\n57.3\n44.0/152.8\n8.64\n4.72/13.11\nSecretory\n9\n68.5\n25.0/157.3\n##\n117.6\n61.1/194.7\n0.80\n0.69/0.99\n###\nMedication\n9\n25.0\n25.0/115.5\n12.5\n12.5/96.3\n2.00\n2.00/3.50\n†\nPeritoneal endometrium\nProliferative\n3\n238.0\n78.9/397.0\n†\n55.1\n43.3/96.5\n5.50\n0.82/7.21\nSecretory\n3\n176.0\n49.4/355.0\n37.5\n12.5/129.0\n3.95\n1.36/9.47\nMedication\n3\n25.0\n25.0/25.0\n12.5\n12.5/32.8\n2.00\n0.76/2.00\nOvarian endometrium\nProliferative\n9\n3430.0\n1809.7/21600.0\n*\n1380.0\n631.5/4280.0\n*\n4.35\n1.69/5.53\nSecretory\n9Proliferative\n9\n3430.0\n1809.7/21600.0\n*\n1380.0\n631.5/4280.0\n*\n4.35\n1.69/5.53\nSecretory\n9\n305.0\n199.0/758.0\n##\n262.0\n164.1/441.5\n##\n1.34\n1.04/3.28\n#\nMedication\n8\n25.0\n25.0/564.8\n†\n75.2\n12.5/1709.0\n†\n2.00\n0.62/2.58\nDeep endometrium\nProliferative\n8\n112.0\n53.9/162.0\n74.3\n36.7/194.0\n1.06\n0.60/2.60\n*\nSecretory\n9\n25.0\n25.0/117.3\n12.5\n12.5/34.4\n##\n2.00\n1.44/4.77\nMedication\n8\n38.5\n25.0/115.6\n23.4\n12.5/128.8\n2.00\n0.65/2.00\nOpen in a new tab\n*\n,\nP\n< 0.05\nvs\n. PE within cycle phase.\n†\n,\nP\n< 0.05\nvs\n. proliferative phase.\n#, #, ###,\nP\n< 0.05, < 0.01, and < 0.001, respectively,\nvs\n. serum within the cycle phase.\nIn the proliferative phase, the median for the E2/E1 ratio in the tissue concentrations was 8.3 and 8.6 in healthy and eutopic endometrium, respectively, whereas the ratio was 0.7 and 0.8 in the secretory phase. Thus, the E2/E1 ratio was 10 to 12 times higher in the proliferative phase compared with the secretory phase (\nTable 1Table 1\n). This shows that the balance between tissue E2 and E1 concentrations changes dramatically during the cycle, with E2 being the dominant steroid in the proliferative phase and E1 in the secretory phase. The endometrial intratissue E2 concentration was markedly higher (\nP\n= 0.003) than that observed in the serum during the proliferative phase (the median being eight and five times higher for women with and without endometriosis, respectively), whereas in the secretory phase no significant difference was observed between endometrial and serum E2 concentration (\nTable 1\n). A strong correlation (r = 0.746;\nP\n< 10\n−5\n) was observed between paired endometrial intratissue E1 and serum E1 concentrations, whereas a remarkably weaker correlation was observed between the corresponding E2 concentrations (r = 0.452;\nP\n= 0.021), which provides further evidence for the observation that intratissue E2 in the endometrium is more strongly regulated compared with that of E1.\nEndometriosisEndometriosis\nThe intratissue E2 and E1 concentrations in the ovarian endometriosis tissue were far higher than those measured in the other types of endometriosis lesions or in eutopic endometrium, and they were four to five times higher than in eutopic endometrium depending on the cycle phase (\nP\n< 0.05). The local E2 and E1 concentrations as well as E2/E1 ratio in the ovarian endometriosis tissue were also significantly higher than the corresponding serum concentrations in the proliferative phase of the menstrual cycle (\nP\n< 0.001). Similar to that observed in the endometrium, a cycle-dependent change in intratissue E2 concentration was detected in ovarian endometriosis lesions (\nP\n< 0.001), the value being 11 times higher in the proliferative phase than in the secretory phase. Thus, between the proliferative and secretory phase, a higher drop was observed for E2 than for E1, and especially in the proliferative phase the ovarian endometriosis is under a very strong E2 stimulus (Table 1\n).\nOf the different lesion types, the intratissue E2 concentration was lowest in the deep lesions, and the concentrations did not vary significantly between the cycle phases, whereas the E1 concentration in the secretory phase was lower than that in the proliferative phase specimens (\nP\n= 0.026). Furthermore, in the proliferative phase, the intratissue E2/E1 ratio was significantly (\nP\n< 0.001) lower in deep lesions compared with eutopic endometrium of women with endometriosis. In the peritoneal lesions, we did not observe any significant cycle-dependent variation in E2 or E1 concentrations (\nTable 1\n).\nEffect of hormonal treatment on local and systemic E2 and E1 concentrations\nExpectedly, suppression of ovarian steroid synthesis by hormonal contraceptives significantly decreased serum E2 [fold change (FC) = −43.9;\nP\n< 0.001] and E1 (FC = −3.5;\nP\n= 0.021) concentrations (\nTable 1P\n< 0.001] and E1 (FC = −3.5;\nP\n= 0.021) concentrations (\nTable 1\n). Interestingly, the E2 concentration was more strongly altered, and the serum E2/E1 ratio was 0.13 in the patients on hormonal medication, whereas it was between 1.46 and 1.50 at the different menstrual phases in the women without medication. Accordingly, the medication also decreased both E2 and E1 concentrations in the eutopic endometrium to concentrations being lower than those measured in the proliferative or secretory phase. The strong suppression of ovarian steroidogenesis reflected into a dramatic drop also in the intratissue concentration of E2 and E1 in ovarian endometriosis lesions (FC = −137.2 and −18.4, compared with proliferative phase specimens;\nTable 1\n).\nComparison of the gene expression of estrogen-metabolizing enzymes in endometrium and different types of endometriosisThe expression profiles of AKR1C3, CYP19A1, and HSD17B1, -2, -4, -6, -7, -10, -12, and -14 were analyzed using both the microarray and qRT-PCR analyses (\nFig. 1\n). The qRT-PCR data demonstrated that among the HSD17B genes, the most striking difference between the healthy or eutopic endometrium and different types of endometriosis specimens was observed for HSD17B2 and HSD17B6.\nFig. 1.\nOpen in a new tab\nmRNA expression of AKR1C, HSD17B1, -2, -4, -6, -7, -10, -12, and -14, as well as ESR1 and ESR2 in healthy endometrium (CE) and eutopic endometrium of endometriosis patients (PE), and PERIT, OV, and DEEP endometriosis by qRT-PCR analysis in proliferative and secretory phases of the menstrual cycle (median with 25th/75th percentiles). Statistically significant changes compared with CE within the cycle phase are labeled as: *,\nP\n≤ 0.05; **,\nP\n≤ 0.01; ***,\nP\n≤ 0.001. The significant difference between proliferative and secretory phases within each sample group is labeled as: ¤,\nPP\n≤ 0.05; ¤¤,\nP\n≤ 0.01; ¤¤¤,\nP\n≤ 0.001 (two-way ANOVA with multiple comparisons with\npost hoc\nTukey-Kramer test on log-transformed data). P, Proliferative; S, secretory.\nThe level of expression of HSD17B2 was significantly higher in healthy (FC = 17.1;\nP\n< 0.001) and eutopic endometrium (FC = 6.7;\nP\n< 0.001) during the secretory phase, compared with the proliferative endometrium. However, no cycle-dependent change in the expression of HSD17B2 was observed in the endometriosis lesions. Thus, the level of HSD17B2 mRNA was markedly lower in PERIT, OV, and DEEP endometriosis (\nPP\n< 0.05) compared with that in the CE (FC = 0.11, 0.13, and 0.10, respectively). Furthermore, in the proliferative phase, we did not detect a difference in the HSD17B2 expression between the CE or PE and endometriosis specimens. By comparing the E2 concentration and the mRNA expression analyzed from the same specimen by the microarrays, we observed a significant negative correlation between the E2 concentration and the HSD17B2 mRNA expression in endometrium of women with and without endometriosis. However, HSD17B2 mRNA and E2 concentration did not correlate in peritoneal, ovarian or deep lesions (Supplemental Table 2). Identical data were obtained by verifying the microarray data by qRT-PCR analysis, whereas the other E2 inactivating HSD17Bs were only weakly regulated in the endometrium or endometriosis.\nThe expression of HSD17B6 was remarkably higher in endometriosis lesions compared with eutopic endometrium of patients with endometriosis or healthy controls (\nFig. 1Fig. 1\n). Compared with the healthy endometrium, the median expression in different types of endometriosis was 8- to 9-fold higher in the proliferative phase and 3- to 11-fold higher in the secretory phase (FC = 11.2, 2.7, and 7.9 for peritoneal, ovarian, and deep lesions, respectively;\nP\n< 0.01). Furthermore, the HSD17B6 expression did not vary during the menstrual cycle in any of the sample groups. Of interest, HSD17B6 was strongly but negatively correlated with E2 levels in healthy endometrium, whereas no correlation was observed in endometriosis lesions with high mRNA expression (Supplemental Table 2). Only minor differences in mRNA expression for the E1 activating HSD17Bs, including HSD17B1, -5, -7, and -12, were observed between the endometrium and endometriosis specimens. However, it is worth noting that HSD17B1 and HSD17B7 expression was higher (\nPP\n< 0.05) specifically in the ovarian endometriosis lesions during the secretory phase, but only HSD17B7 expression showed correlation with intratissue E2 and E1. The microarray data were validated by the strong positive correlation observed between the progesterone receptor expression and tissue E2 concentration (Supplemental Table 2), and with its correlation with the intratissue E2/E1 ratio in the healthy endometrium.\nThe mRNA expression of ESR1 (\nFig. 1\n) was significantly lower specifically in ovarian endometriosis compared with endometrium, whereas no statistical difference was observed between endometrium and other endometriosis lesion types. In contrast, ESR2 mRNA expression was significantly higher in both ovarian and deep endometriosis compared with the endometrium.",
         "Hormonal contraceptives significantly decreased serum E2 and E1 concentrations.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "43",
         "What percentage of errors in medical laboratories are attributed to the preanalytical phase?",
         "Nadin Younes et al.,Not identified, DOI:https://doi.org/10.3390/v12060582:\n 3. Implications and Challenges of Current COVID-19 Diagnostic Tests\n3.1. Preanalytical and Analytical Errors\nAlthough medical diagnostic errors can happen almost always and everywhere [\n17\n], the fragility of diagnostic laboratories is significantly magnified when the healthcare workers were required to face high workload and work in high-throughput settings due to the increasing number of cases [\n18\n]. Although the consequences of laboratory errors are often substantial [\n1918\n]. Although the consequences of laboratory errors are often substantial [\n19\n], the consequences in the current SARS-CoV-2 pandemic are certainly amplified. Unfortunately, false-positive and false-negative results do not only possess a threat to the health of the individual, but may also disrupt the efficiency of emergency plans, public health policies, and preventive measures applied for containing the pandemic. A false-positive test result not only leads to unrequired treatment but may cause societal problems as it may undermine the workforce available for facing this pandemic if attributed to people working in public facilities. Nevertheless, a false-negative test result may potentially contribute to further spread of the SARS-CoV-2 virus within the community. Therefore, accurate and precise laboratory technologies play a vital role in diagnosing and managing the current SARS-CoV-2 outbreak [\n2020\n]. However, there are a number of potential preanalytical and analytical errors that must be taken into consideration by clinicians, clinical microbiology laboratories, and public health authorities to avoid false test results.\nThere is undeniable evidence that the preanalytical phase is the main source of errors in medical laboratories [\n21\n,\n22\n], accounting for approximately 46% to 68.2% of errors observed during the whole testing process [\n23\n], despite continuous improvements in pre-analytical automation. It is estimated that more than one-fourth of all pre-analytical errors result in an unnecessary investigation or inappropriate patient care, substantially magnifying the financial burden on the healthcare system [\n2424\n], and thus resulting in inadequate and slow healthcare. The safety and quality of diagnostic testing may be endangered by misidentification of the patient and/or sample, collection of an inappropriate or insufficient sample, inaccurate conditions of sample transportation and storage (e.g., prolonged transportation time and injury exposure), presence of interfering substances (e.g., cellular components due to whole blood freezing and inappropriate additives) [\n25\n,\n26\n,\n27\n], and finally, procedural issues occurring during sample preparation, including pipetting errors during manual sample preparation or aliquoting, cross-contamination and sample mismatch [\n2828\n]. Although analytical errors are believed to be the smallest contributors to laboratory errors, there are several potential analytical problems that could significantly jeopardize the quality of testing, and thus need to be considered. Analytical errors include equipment malfunction, non-adequately validated assays, undetected failure of quality control, active viral recombination, testing carried outside the diagnostic window, poor harmonization of primers or probes, and non-specific rRT-PCR annealing, along with other technical issues [\n25\n,\n26\n,\n27\n].\n3.2. Chest Computerized Tomography (CT)25\n,\n26\n,\n27\n].\n3.2. Chest Computerized Tomography (CT)\nChest computerized tomography (CT) is a conventional, non-invasive imaging technology with high accuracy and speed. The sensitivity to detect SARS-CoV-2 using chest CT is reported to be higher than that of real-time reverse transcription-polymerase chain reaction (rRT-PCR). Recent evidence has shown that asymptomatic patients with COVID-19 may show paradigmatic CT changes very early and even before being positive with rRT-PCR [\n29\n,\n30\n,\n31\n]. For instance, a case was reported in Wuhan city with a history of chills and fever of unknown cause and tested negative four times for SARS-CoV-2 with rRT-PCR from the disease onset [\n32\n]. Thus, the clinical physician could not diagnose the patient with COVID-19 at an early stage because of the false negative rRT-PCR results [\n3232\n]. Therefore, according to Feng et al., patients showing symptoms of fever, dry cough, fatigue, or dyspnea along with recent exposure to SARS-CoV-2 infected patients should be diagnosed with CT despite negative rRT-PCR test results [\n32\n]. These pieces of evidence support the advice that the most efficient approach for diagnosing suspected patients with COVID-19 in suspected patients shall encompass a combination of rRT-PCR with clinical and epidemiologic evidence (such as the probability of exposure with infected patient, signs, and symptoms) and chest CT findings.\n3.3. Nucleic Acid Amplification Testing (NAAT)Rapid and accurate detection of positive COVID-19 cases is crucial to control the viral outbreak in the community and health care facilities. In general, studies have shown that molecular technologies are more accurate than CT scans and serological tests for the definitive diagnosis of COVID-19, as they can target and identify the specific antigen of SARS-CoV-2. The development of molecular diagnostic technologies against SARS-CoV-2 is dependent upon the understanding of the proteomic and genomic composition of the virus and the viral induction of changes in proteins and genes expression in the patient during and after infection. According to the World Health Organization (WHO), 104 strains of SARS-CoV-2 virus were isolated and sequenced using Illumina and Oxford nanopore sequencing by the 15th of February 2020. By the 24th of March, the genomic and proteomic compositions of SARS-CoV-2 had been identified. However, the host response to the virus is still under investigation. However, the host response to the virus is still under investigation. Currently, the NAAT available for SARS-CoV-2 includes rRT-PCR (Laboratory-based) and reverse transcription loop-mediated isothermal amplification (RT-LAMP) (POC) [33\n,\n34\n]. Unfortunately, the currently available diagnostic tests are labor-intensive and time-consuming, and a shortage of commercial kits delays diagnosis.\n3.3.1. Manual Laboratory Based NAAT: Real-Time Reverse Transcription-Polymerase Chain Reaction (rRT-PCR)\nThe current gold standard for the etiological diagnosis of SARS-CoV-2 infection is rRT-PCR on a variety of clinical specimens, including bronchoalveolar lavage fluid, fiber bronchoscope brush biopsies, sputum, nasal swabs, pharyngeal swabs, feces, or blood [\n3535\n]. The rRT-PCR tests offer several benefits. First of all, rRT-PCR is especially valuable at the early stage of infection, when the viral load is lowest and can differentiate it from other similar viruses, due to its sensitivity and specificity, respectively. Thus, as opposed to serology, rRT-PCR provides more valuable information at the initial stages of infection, as it detects the pathogen directly by detecting its RNA when the aim is to prevent infectivity and antibodies have not yet been built. In addition, rRT-PCR results are generally available within a few hours to 2 days. Moreover, it can be easily operated on a large scale.Although rRT-PCR offers many benefits, it has some limitations. Its low sensitivity, low stability, and long processing time were detrimental to the health care efforts to contain the outbreak. Also, several external factors may affect rRT-PCR testing results accuracy, including sampling operations, specimen source (e.g., upper or lower respiratory tract), sampling timing (before and after symptoms onset), and the performance of detection kits. Most importantly, recent evidence has shown that the diagnostic accuracy of many of the available commercial rRT-PCR kits for detecting SARS-CoV-2 may be lower than optimal (i.e., <100%), and there are reports where it has given false negatives in subjects for up to two weeks [\n36\n,\n37\n,\n38\n]. This high incidence of false negative diagnosis was observed specifically in SARS-CoV-2 testing. The largest study on coronaviruses testing to date estimated a rate of 41% false negatives on RT PCR diagnostic tests used in China [\n39\n,\n40\n,\n4139\n,\n40\n,\n41\n]. However, still more research is needed to determine the true prevalence of such false-negative rRT-PCR results; scientists and researchers agree that the problem is significant, which not only impedes the diagnosis of the disease in patients but also risk patients who assume they are uninfected further transmitting the virus in the community. Moreover, using PCR, codetection with other respiratory viruses is frequently encountered in coronaviruses, and the contribution of positive CoV PCR results to disease severity is not always explicitly exhibited [\n4242\n]. Furthermore, rRT-PCR requires professionally trained staff to operate sophisticated laboratory facilities, which are usually located at a central laboratory (biosafety level 2 or above), and is often time-consuming, requiring from few hours up to 2 or 3 days to obtain laboratory results. This often leaves a rapidly rising number of potential cases untested and thus opening a gaping hole in SARS-CoV-2 prevention efforts. Furthermore, this time-consuming process of sample testing is not only extremely disadvantageous but also unsafe since the virus needs to be contained. Finally, the US Food and Drug Administration (FDA) concluded that a negative rRT-PCR test result does not completely rule out SARS-CoV-2 infection and shall not be used as a single element for patient management decisions, and re-testing shall be considered in consultation with public health authorities [\n43\n]. The information summarized in\nTable A1\n(\nAppendix A43\n]. The information summarized in\nTable A1\n(\nAppendix A\n) was extracted from the manufacturer package inserts or their websites.\nProtocols for rRT-PCR testing developed by several countries and entities, including Germany, Hong Kong, China CDC, Thailand, and Japan, have been posted to the WHO’s website [\n44\n], and the protocol for testing in the United States has been posted to CDC’s website [\n45\n].\nTable 2\nis a comparison between the available rRT-PCR protocols.\nTable 2.\nSummary table of available protocols posted to the WHO’s website.\nInstitute\nGene Targets\nAmplicon Size (bp)\nSensitivity\nSpecificity\nConcentration/Volume of Reagents\nDoes the Protocol Recommend Specific Kits?\nChina CDC\nORF1ab gene\nNR\nNR\nNR\nNR\nNR\nN gene\nNR\nNR\nNR\nNR\nNR\nInstitute Pasteur, Paris, France\nRdRp: nCoV_IP2 gene\n108 bp\n95% hit rate for approx. 100 copies of RNA GE.\nLOD for 1x10\n7\nRNA copies is ~21 cycles\nLOD for 1x10\n4\nRNA copies is ~30 cycles\nNo cross reactivityLOD for 1x10\n7\nRNA copies is ~21 cycles\nLOD for 1x10\n4\nRNA copies is ~30 cycles\nNo cross reactivity\nFinal concentration of 0.4 μM of each primer and 0.2 μM of probe\nRNA extraction via NucleoSpin Dx Virus and Invitrogen SuperscriptTM III Platinum\n®\nRdRp: nCoV_IP4 gene\n107 bp\nE gene\n125 bp\nUS CDC, USA\nN1 gene\n71 bp\nLOD: 1x10\n0.5\nRNA copies/μL and 10 RNA copies/ μL for Qiagen EZ1 and Qiagen respectively.\nProbe showed high sequence homology with SARS coronavirus and Bat Sars-like coronavirus\n20 μM primers, 5 μM probe; 15 μL total volume\nFor the RT-qPCR TaqPathTM 1- Step RT-qPCR Master Mix. For extraction, they recommend bioMérieux NucliSens\n®\nsystems, QIAamp\n®\nkits, QIAGEN kits, Roche Kits and Invitrogen kits\nN2 gene\n67 bp\nN3 gene (removed from diagnostic panel 3/15/20)\n72 bp\nNational Institute of Infectious Diseases, Japan\nN gene\nNR\nAverage Cq value of specimen was 36.7 and 35.0 for the positive control (500 copies of RNA transcript)\nNRNR\n1 μL of 20 xprimer and probe mix in a 20 μL reaction with 5 μL of RNA. F primer at 500 nM, R primer at 700 nM, probe at 200 nM.\nRNA extracted using QIAamp viral RNA mini kit (Qiagen). Reverse transcription via Super Script IV Reverse Transcriptase (Thermo). RT-PCR via QuantiTect Probe RT-PCR Kit (Qiagen)\nCharité, Germany\nRdRp gene\nNR\nLOD: 3.8 RNA copies/ reaction, 95% hit rate; 95% CI: 2.7-7.6 RNA copies/reaction\nNo reactivity with other human respiratory viruses\nRdRP: F-600 nM/reaction, R-800 nM/rxn, P-100 nM each/ reaction,\nRNA extracted using MagNA Pure 96 system (Roche), RT- PCR via Superscript III one step RT-PCR system with Platinum Taq Polymerase (Invitrogen).\nE gene\nNR\nLOD: 5.2 RNA copies/reaction, at 95% hit rate; CI: 3.7-9.6 RNA copies/reaction\nE gene: F-400 nM/r reaction, R-400 nM/ reaction, P-200 nM/ reaction\nHKU, Hong Kong SAR\nORF1b-nsp14 gene\n132 bp\nNR\nNo reactivity with respiratory cultured viruses and clinical samples.\n10 μM primers, 10 μM probesNo reactivity with respiratory cultured viruses and clinical samples.\n10 μM primers, 10 μM probes\nQIAamp Viral RNA Mini Kit or equivalent and TaqMan Fast Virus Master mix.\nN gene\n110 bp\nNational Institute of Health, Thailand\nN gene\nNR\nPositive control detected at less than 38 cycles.\nNR\n40 μM primers, 10 μM probe\nMacherey-Nagel Nucleospin RNA virus and Invitrogen superscriptTM III Platinum One-Step Quantitative\nOpen in a new tab\nNR: not reported; GE: genome equivalent; LOD: Limit of detection.\n3.3.2. Rapid and Point of Care NAAT: Reverse Transcription-Loop-Mediated Isothermal Amplification\nTransforming the molecular diagnostic testing for SARS-CoV-2 from laboratory settings to point of care (POC) is potentially important to increase the quantity of testing that can be conducted [\n39\n,\n4239\n,\n42\n], potentially reducing the time to obtain an actionable result, and thus supporting earlier identification of positive cases. Most importantly, POC testing will support the suitable use of quarantine resources, infection control measures, and patient recruitment into clinical trials of treatments. Most of the available molecular POC tests have either gained Conformité Européenne (CE) marking or emergency Food and Drug Administration (FDA) approval [\n46\n]. Molecular POC testing utilizes the same basic technology as the laboratory-based assays, but with automating various number of the steps. Therefore, molecular POC tests could be operated in near-patient settings rather than on the laboratory bench, which is expected to reduce the turnaround time and rapidly provides the result. Some of the molecular POC tests utilize isothermal nucleic acid amplification techniques, such as MicrosensDx RapiPrep\n©©\nCOVID-19 and Abbott ID NOW COVID-19, while others utilize PCR technology, such as Cepheid Xpert SARS-CoV-2, Credo VitaPC R COVID-19 assay, GenMark ePlex SARS-CoV-2, MesaBioTech Accula SARS-CoV-2, which utilizes lateral flow technology, and the very recent Spartan Cube CYP2C19 System (Canada) [\n46\n].\nLoop-mediated isothermal amplification (LAMP) was developed as a rapid, accurate, reliable, and cheaper technique to amplify the target sequence at a single reaction temperature instead of sophisticated thermal cycling equipment needed in rRT-PCR [\n4747\n]. The advantage of using LAMP is that the amount of DNA produced is much higher than in rRT-PCR and a positive test result can be seen visually without requiring a machine to read the results. In addition, it is simple, cheap, and rapid. Several studies evaluated the use of a novel RT-LAMP method against the gold standard rRT-PCR. Two studies showed evidence that RT-LAMP methods demonstrated more than 97% sensitivity targeting the ORF1ab gene compared to rRT-PCR [\n48\n,\n49\n]. Yang et al. showed that RT-LAMP and rRT-PCR have the same sensitivity and both can detect a 20-fold diluted sample [\n50\n]. Additionally, according to Yang et al., the detection limit of LAMP is 1000 copies/mL, which is equal to the rRT-PCR kits [\n50\n]. Most importantly, studies have shown that RT-LAMP analysis is extremely specific because it uses six to eight primers to identify eight different regions on the target DNA [\n50\n,\n5150\n,\n51\n]. However, unlike rRT-PCR, LAMP technology does not have such a large background of literature behind it. Thus, tests using LAMP technology for COVID-19 are still being assessed in clinical settings.\nAlmost all molecular POC described devices are portable benchtop-sized analyzers, except the MesaBioTech Accula and MicrosensDx RapiPrep\n©\nCOVID-19 tests, which are smaller, handheld devices. A variety of clinical sample types may be used, including oral, throat, nasal, or nasopharyngeal swabs. All tests require a similar sample preparation procedure that involves placing the swab sample into the viral transport media and pipetting the sample into a single-use disposable cartridge—this sample preparation step takes approximately 2–10 min [\n46\n]. The time to result varies from 13 min in Abbott Diagnostics ID NOW COVID-19 to 45 min in Cepheid Xpert SARS-CoV-2 [\n46\n]. The information and validation of each device are summarized in\nTable A1\n(\nAppendix A\n).46\n]. The information and validation of each device are summarized in\nTable A1\n(\nAppendix A\n).\n3.4. Serological Testing for COVID-19 Diagnosis3.4. Serological Testing for COVID-19 Diagnosis\nrRT-PCR–based assays performed on respiratory specimens remain the gold standard for COVID-19 diagnostics, as mentioned previously. However, point-of-care technologies and serologic immunoassays are rapidly emerging, with high serological tests for SARS-CoV-2 being at increased demand for better quantification of SARS-CoV-2-positive cases, including asymptomatic and recovered cases. Serological tests are blood-based tests that measure antibodies or antigens present in the blood when the body is responding to a particular infection. Thus, it could identify previous exposure to a particular pathogen as well as the production of the body’s immune system-specific antibodies. Two types of serology test, in particular, are becoming more widely available, namely laboratory-based enzyme immunoassays (EIA) on high throughput automated platforms and rapid, point of care (POC) tests, which are similar to a blood glucose test or home pregnancy test.Serological tests offer a number of advantages compared to rRT-PCR. First of all, serological testing can provide further details by identifying individuals who have developed virus-specific antibodies, and thus can detect past infection and give better information regarding the disease prevalence in a population. Unlike viral RNA, virus-specific antibodies stay in the blood for several weeks to months after symptom onset. According to the FDA, IgM antibodies to SARS-CoV-2 are detectable in the blood just a few days after initial infection [\n52\n]. However, IgM levels throughout the course of infection are not well characterized. IgG becomes detectable three days from symptom onset or at least 7–10 days after infection [\n1616\n]. It worth mentioning that when the result is negative for COVID-19, the patient was probably not infected at the time of sample collection. However, that does not mean that he will not get sick. In addition, the detection of SARS-CoV-2 antibodies does not guarantee the protection against COVID-19 infection, as many types of anti-SARS-CoV-2 are not neutralizing antibodies [\n53\n]. Considering the fact that 20%–80% of SARS-CoV-2-positive cases are estimated to be asymptomatic, serological tests are especially beneficial because of their scalability, which allows their use on a large scale to assess the overall immune response in a population [\n5454\n]. In addition, human antibodies are known to be more stable compared to viral RNA, and thus serological samples are less prone to deterioration during sample collection, preparation, transport, storage, and testing compared to rRT-PCR samples. Moreover, serological samples have less variations compared to nasopharyngeal specimens because antibodies are usually homogeneously dispersed in the blood. Furthermore, serological samples can be collected easily with minimal discomfort to the patient during phlebotomy. On the other hand, serological tests have some disadvantages, mainly involving the slow antibody response to SARS-CoV-2 virus, as they may not be detectable until three days from symptom onset or at least 7–10 days after infection (\nFigure 2\n) [\n16Figure 2\n) [\n16\n]. In addition, these tests are not designed to detect individuals in the early stages of COVID-19 infection. For instance, less than 40% of infected individuals are seropositive (IgM/IgA) in the first seven days, making it unreliable for the detection of acutely infected individuals. Importantly, there have been reports that those with mild cases of COVID-19 infection do not produce antibodies. It was proposed that their innate immune system (cell-mediated immunity) wiped out the virus before the adaptive immune system (antibodies) had to produce antibodies [\n55\n]. Since serological tests alone may not be enough to diagnose SARS-CoV-2, combining both serological and molecular techniques would give a valuable diagnostic result.\nFigure 2.\nOpen in a new tab\nEstimation of biomarker levels during the COVID-19/SARS-CoV-2 infection.\n3.4.1. Manual ELISAEstimation of biomarker levels during the COVID-19/SARS-CoV-2 infection.\n3.4.1. Manual ELISA\nA variety of CE-marked manual enzyme-linked immunosorbent assays (ELISA) have been developed for the rapid detection of neutralizing antibodies (IgM, IgG, and IgA) against the novel coronavirus by many IVD companies such as Euroimmun, Epitope Diagnostics, DRG Diagnostics GmbH, IBL International, Creative Diagnostics and others (\nTable A2\n,\nAppendix A\n). There are also some commercially available manual ELISA kits for detecting SARS-CoV-2 viral antigens (SP and NP); however, these are mainly used in research and not for clinical diagnosis [\n5656\n]. Manual ELISA provides accurate and valuable information regarding the immune response to the virus; however, unlike rRT-PCR, it cannot be used for screening or diagnosis of early infection, since specific IgM and IgG antibodies are not detectable at this phase. IgM antibody response occurs earlier than that of IgG, with positive IgM antibodies in 70% of symptomatic patients after 8–14 days and about 90% of total antibodies test positive within 11–24 days [\n57\n]. On the other hand, IgG antibodies can be detected around 20 days after viral infection and they persist for a long time [\n58\n]. The reactivity of IgG is assumed to reach more than 98% after several weeks, but the extent of this antibody response is yet to be determined [\n5959\n]. According to recent reports of the WHO, only 2% or 3% of infected COVID-19 individuals appear to have antibodies in their blood. “There is simply not enough data yet to determine if protective immunity is achieved after infection,” says Jennifer Rychert, the medical director of microbial immunology at ARUP Laboratories.\nAnother challenge of using manual ELISA for SARS-CoV-2 detection is that IgM antibodies are notoriously non-specific, and given the time it takes for the development of specific IgG antibodies, serology testing will not likely play an active role in the detection of early cases (\nFigure 1\n) except for diagnosis/confirming late cases or to determine the immunity of healthcare personnel as the outbreak progresses [\n6060\n]. Furthermore, manual ELISA kits are subject to many interferences, including a specific binding and cross-reaction with other coronaviruses, such as MERS-CoV, SARS-CoV-1, and endemic coronavirus. This depends on the type of antigen used to coat the plates. For instance, an ELISA method based on bat SARSr-CoV Rp3 N protein was successfully developed to detect IgM and IgG antibodies against SARS-CoV-2 in early cases of COVID-19 [\n59\n]. A caveat in this ELISA method is that it may produce false positive results since nucleocapsid protein (NP) is the most conserved viral protein among human betacoronaviruses [\n61\n]. Hence, antigens used in this ELISA may react with antibodies against other types of coronavirus (HKU1, 229E, OC43, NL63) that are known to cause the common cold [\n6262\n]. On the other hand, spike protein (SP) is the most diverse protein and several companies have focused on developing ELISA methods for detecting serum antibodies against two domains in the S protein (S1 and S2). The coronavirus envelope spike is responsible for viral entry and it determines the host tropism and virus transmission, which makes it a good candidate for ELISA development [\n60\n]. Still, the evaluation of the clinical performance of manual ELISA kits is imperative before using them for COVID-19 diagnosis.\nAlthough many challenges exist, serology testing using ELISA offers great benefits as a therapeutic option to control the current pandemic and possible re-emergence of coronavirus in the future. Hence, the development of manual ELISA kits remains a high priority, as they can complement the existing testing of SARS-CoV-2 by rRT-PCR (the gold standard) and overcome some of its limitations [\n63\n].\n3.4.2. Automated SerologyThe increased demand to perform diagnostic tests on the population imposes a huge clinical and financial burden on diagnostic laboratories. The implementation of automated serological testing has increased the quality assurance and lowered the turn-around-time (TAT) as well as false positive and negative results. Automated techniques are currently adopted for the most commonly used serological methods. Regular serology tests, which are more amenable to automation, are best deployed in the laboratory setting where they can be used to identify immune individuals and for population-level seroprevalence studies. These will be most useful later in the outbreak when the prevalence of the disease increases. In fact, the healthcare market has been flushed with SARS-CoV-2 laboratory testing platforms just a few months into the COVID-19 pandemic. The laboratory-based EIA automated platforms offer high efficiency, high throughput, and improved quality of the results. However, this expansion of newly developed platforms makes it challenging to critically evaluate SARS-CoV-2 laboratory automated tests. Most of the available SARS-CoV-2 manual ELISA kits use the standard 96-microplate as a solid phase and also the standard spectrophotometry/colorimetric method for signal detection, while in the automated EIA assay, the solid phase materials are different, such as polystyrene (PS-COOH) or metal-based nanoparticles (magnetic nanobeads). Further, more sensitive detection systems such as chemiluminescence technology are usually sued in the automated assays.In April 2020, a fully automated serology test was launched by DiaSorin (Saluggia,\nItaly\n) to detect SARS-CoV-2 antibodies [\n64\n]. The test was developed to detect SARS-CoV-2 IgG antibodies against both the S1 and S2 domain of the spike protein. This increases the specificity of the test and prevents cross-reaction and false-positive results due to other coronaviruses. The LIAISON\n®\nXL platform is a chemiluminescence analyzer that is used to perform a fully automated diagnostic tests process with a minimum level of laboratory personnel intervention. The system could perform up to 170 samples per hour to fulfill the need for large population screening for SARS-CoV-2 and identify infected individuals. By the end of April, the DiaSorin test obtained the FDA emergency use authorization (EUA). Similarly, Bio-Rad (Hercules, CA, USA) has developed a test that is blood-based EIA to detect antibodies against SARS-CoV-2 virus [\n2\n]. The test could be",
         "Approximately 46% to 68.2% of errors observed during the whole testing process occur in the preanalytical phase.",
         null,
         null,
         null
        ],
        [
         "44",
         "What endometrial cell types are enriched in the proliferative phase?",
         "Luz Garcia-Alonso et al.,2021, DOI:https://doi.org/10.1038/s41588-021-00972-2:\n The human endometrium is the site of implantation that pro - vides nutritional support to the placenta throughout preg - nancy. Unlike other mucosal tissues, it undergoes dynamic, cyclical changes of shedding, regeneration and differentiation throughout reproductive life coordinated by the hypothalamicpituitary-ovarian axis. Endometrial dysfunction underpins many common disorders, including abnormal uterine bleeding, infertil - ity, miscarriage, pre-eclampsia, endometriosis and endometrial carcinoma, that collectively affect many women across the world 1-5. Throughout reproductive years, the functional upper layer of the endometrium, the stratum functionalis, is shed at menstruation. The subsequent tissue repair and proliferation are driven by rising levels of estrogen, secreted by the ovarian follicle during the first half of the menstrual cycle (proliferative phase). Following ovula - tion, progesterone, produced by the corpus luteum, induces the secretory phase during which the initial changes of decidualization occur. Menstruation and spontaneous decidualization are unique to higher simian primates 6-8. Thus, dissecting the mechanisms that regulate cellular differentiation across the menstrual cycle in humans is crucial for understanding how normal endometrium is regulated. Essential to endometrial function are the lumenal and glandu - lar epithelia, composed of a mixture of ciliated and secretory cells. The lumenal epithelium is the site of embryo attachment covering the endometrial surface. Long tubular glands open into the lume - nal epithelium and produce secretions that are rich in growth fac - tors and lipids necessary for placental growth9. We and others have established a three-dimensional in vitro organoid culture model of human endometrial epithelium 10,11. These organoids are generated from dissociated endometrial tissue and menstrual fluid samples 12; they retain the morphology, function and gene signature of the tissue in vivo and respond functionally to ovarian hormones with differentiation into ciliated and secretory cells. They are therefore powerful platforms to investigate endometrial disorders and study mechanisms regulating endometrial differentiation in humans. However, a systematic, quantitative comparison of endometrial organoids at a single-cell level with epithelial cell states in vivo is lacking. This is required to confirm their suitability for exploring the cellular pathways and processes involved in normal and patho - logical endometrial function. The explosion in spatial transcriptomics technologies13-16 pro - vides a unique opportunity to resolve tissue architecture in conjunction with underlying cellular interactions. The spatial arrangement of cells is key to understanding a morphologically complex tissue such as the endometrium, where a cell’s function may differ depend - ing on signals it receives from neighboring cells17. Many spatially resolved transcriptomics methods are not quite at single-cell resolution and rely on the computational integration of coupled single-cell (or single-nucleus) transcriptomes to achieve this level of detail 18-20. These genomic technologies are the basis of the Human Cell Atlas initiative, which aims to map all cells in the human body 21. In this study, by using single-cell and spatial transcriptional profiling, we interrogate the cellular states and spatial localization of human endometrial cells during the proliferative and secretory phases of the menstrual cycle in women of reproductive age. We develop CellPhoneDB v.3. We develop CellPhoneDB v.3.0 to measure intercellular communica - tion taking into account spatial coordinates of cells and use this tool to define cell signaling in both lumenal and glandular epi - thelial microenvironments. We define a complementary role for WNT and NOTCH signaling in regulating differentiation toward the two main epithelial lineages (ciliated and secretory). We pro - file three-dimensional endometrial organoids at single-cell resolution to characterize their hormonal responses in vitro and design a computational toolkit to compare the results with those observed in vivo to benchmark this model system. Finally, by modulating WNT and NOTCH pathways in the organoid cultures, we develop lineage-specific endometrial epithelial cells and define the molecu - lar events involved in their response to ovarian hormones. Results A single-cell map of the full-thickness human uterus. Results A single-cell map of the full-thickness human uterus. To gener - ate a cellular map of the human endometrium that accounts for the temporal and spatial changes across the menstrual cycle, uterine samples were analyzed by single-cell transcriptomics-single-cell RNA sequencing (scRNA-seq) and single-nucleus RNA sequenc - ing (snRNA-seq)-alongside spatial transcriptomics methods (10x Genomics Visium slides and high-resolution microscopy) (Fig. 1a and Extended Data Fig. 1a). Two different types of samples from women of reproductive age were integrated in our analysis: endo - metrial biopsies from live donors screened for potential endometrial disorders (n = 3) and the whole endometrium with attached subjacent myometrium from the uteri of donors who died of nongynecological causes (n = 6) (Supplementary Table 1). This latter approach allows sampling of the endometrial basal layer and myometrium, which are absent from endometrial biopsies. We were able to generate a uterine map of 98,568 cells from 15 individuals by integrating our dataset with previous scRNA-seq data obtained from superficial endometrial biopsies 22 (Fig. 1b, Extended Data Fig. 1b-f and Supplementary Table 2). We identify 14 clusters that were assigned cell identity based on their expression of known markers (Extended Data Fig. 1g). These clusters can be grouped into five main cellular categories:  immune (lymphoid and myeloid);  epithelial (SOX9 +, lumenal, glandular and ciliated);   endothelial (arterial and venous);  supporting-perivascular cells (PV STEAP4 and PV MYH11), smooth muscle cells and fibro - blasts expressing C7 (fibroblasts C7); and  stromal-nondecidualized endometrial (eS) and decidualized endometrial (dS). SOX9+ epithelial cells and eS are characteristic of the regenerating proliferative phase (Fig. 1c). SOX9+ epithelial cells and eS are characteristic of the regenerating proliferative phase (Fig. 1c). Subclustering of immune cells resolves their heterogeneity, including identification of the three uterine natural killer cell subsets we have previously defined in the early pregnant uterus 23 (Fig. 1d and Extended Data Fig. 1h). snRNA-seq data from four additional full-thickness uterine samples of proliferative endometrium confirm the populations found in the proliferative phase (Fig. 1e,f and Extended Data Fig. 2a-d). We additionally found lym - phatic endothelial and mast cells enriched in these samples; these are likely to originate from the myometrium (Fig. 1e and Extended Data Fig. 2d). To systematically map the location of the cell types identified by scRNA-seq within the endometrium and myometrium, we used Visium Spatial Transcriptomics technology. We examined four fullthickness uterine samples in the proliferative and secretory phases from two individuals (Extended Data Fig. 3a-e). After integrat - ing single-cell transcriptomics and Visium data using our recently developed cell2location algorithm 18, cell states were mapped to the endometrium and/or the myometrium. We identify specific perivascular cells: PV MYH11 are characteristic of myometrium while PV STEAP4 are only present in the endometrium (Fig. 1g and Extended Data Fig. 3f). In addition, we find that fibroblasts C7 are enriched in the basal layer of the endometrium in both the proliferative and secretory phases (Fig. 1h,i and Extended Data Fig. 3f). Altogether, our analysis yields a comprehensive catalog of the major subsets of uterine cells together with their cellular position in endometrium and myometrium. We have made an open-source web server available at www.reproductivecellatlas.org. Spatiotemporal characterization of proliferative epithelium.reproductivecellatlas.org. Spatiotemporal characterization of proliferative epithelium. We next focused on the two main lineages of endometrial epithelial cells, secretory and ciliated, across the menstrual cycle and analyzed these subsets individually (Fig. 2a). Epithelial cells are classified into four main groups based on their marker expression:  SOX9 pop - ulations, enriched in the proliferative phase and expressing genes characteristic of rising estrogen levels (MMP7, ESR1);  ciliated cells (PIFO, TPPP3);  lumenal cells (LGR5); and  glandular cells (SCGB2A2) (Fig. 2b-d and Extended Data Fig. 4a,b). A frac - tion of the glandular epithelial cells express molecules characteristic of uterine milk in the secretory stage (PAEP, CXCL8). With our integrative scRNA-seq maps, we can now resolve three clusters within the SOX9 population:  SOX9 +LGR5+ cells, expressing KRT17 and WNT7A;  SOX9+LGR5− cells, expressing IHH; and  proliferative SOX9+ cells, including both LGR5+ and LGR5− cycling cells (that is, cells in G2M/S phase). By integrating scRNA-seq and Visium data (Fig. 2e and Extended Data Fig. 4c), we define specific spatial coordinates for these SOX9 subsets:  noncycling SOX9+LGR5+ cells are enriched in the surface epithelium;   noncycling SOX9+LGR5− cells are located in the basal glands; and   cycling SOX9+ cells map to glands in the regenerating superficial layer. Single-molecule fluorescence in situ hybridization (smFISH) with RNAscope probes shows higher expression of the proliferative marker MKI67 in the superficial layer of the endometrium dur - ing the proliferative phase, validating Visium data (Extended Data Fig. 5a). 5a). We also validated the presence of the markers LGR5 and WNT7A, characteristic of the SOX9 +LGR5+ cells present in surface epithelium during the proliferative phase (Fig. 2f and Extended Data Fig. 5b,c). SOX9 and LGR5 are expressed in stem cells/pro - genitors in several tissues including gut, kidney, skin and ovaries and may label similar populations in the human endometrium 24-29. Ciliated cells are present in both the proliferative and secretory phases, but, as expected, PAEP secretory cells are only present following ovulation (Fig. 2b,c and Extended Data Fig. 4b). This indicates that estrogen alone can induce ciliary differentiation, while secretory differentiation depends on the addition of progesterone. During the proliferative phase, in addition to the FOXJ1, PIFO, TP73 ciliated population, we define a distinct subset of preciliated cells that express MUC12, HES6 and cell cycle genes (CDC20B, CCNO) (Fig. 2d). 2d). After ovulation, secretion of progesterone induces the differentiation of SOX9+ cells into specialized secretory cells found at the surface and in glands (Extended Data Fig. 5d). To date, it has not been possible to distinguish between differentiated lumenal and glandular subsets using specific transcriptomic signatures and markers. We can now identify these subsets in our scRNA-seq data, confirmed by integration with spatial transcriptomic data (Fig. 2g and Extended Data Fig. 4d,e) and revealing markers that are vali - dated at the protein level (Fig. 2h). There is enriched expression of both COX1 (encoded by PTGS1) and KRT5 in the lumenal epithelium and SCGB2A2 in the glandular epithelium. SOX9+ epithelial cells in endometrial disorders. Disorders in endometrial function have a profound impact on women’s health and reproductive outcomes. There has been limited progress in the study of these disorders over the past decade, partly due to the challenges in analyzing this highly dynamic and complex tissue. To identify the cells involved in these disorders, we looked for tran - scriptomic signatures in bulk RNA data from endometrial cancers and endometriotic lesions. We first deconvoluted the transcriptomes from bulk RNA sequenc - ing data from serous and endometrioid endometrial adenocarcinomas available in The Cancer Genome Atlas (TCGA), using our single-cell atlas as a reference. Deconvolution reveals that the signature of the SOX9 + epithelial population is dominant in tumors (Fig. 3a). Based on expression signals in epithelial cells, endometrial adenocarcinomas resolve into three patterns. The first, characterized by SOX9+LGR5+, occurs in 63% of serous and 24% of endometrioid adenocarcinomas. The second, SOX9 +LGR5−, is found in 33% of endometrioid adenocarcinomas but is absent from serous adenocarcinomas. The third pattern, with expression of differentiated ciliated or glandular signals, is present in <10% of endometrial carcinomas. In line with these results, markers characteristic of the SOX9 +LGR5+ subset (for example, WNT7A, MSLN, KRT17, PTGS1 and VTCN1) are all upregulated in tumors with a SOX9+LGR5+ transcriptomic signature compared to tumors with a SOX9+LGR5− signature (Extended Data Fig. 6a). We then correlated the clinical stages of endometrial adenocarcinomas with our cell signals. The more advanced stages of endometrial adenocarcinomas (stages III and IV) have a greater SOX9+LGR5+ signal (Wilcoxon test, stages I + II against stages III + IV , P value 7.54 × 10−6) (Fig. 3b and Supplementary Table 3). We also linked the SOX9+LGR5+ signals to the four molecular subtypes of endometrial cancer defined by TCGA30. The SOX9+LGR5+ signature is stronger in tumors characterized by high copy number (copy-number-high) alterations (Wilcoxon text, P value 6.72 × 10 −13), typical of serous endometrial adenocarcinomas and linked with a worse prognosis (Supplementary Table 3). Indeed, all the copy-number-high tumors considered in our analysis (31 serous and 7 serous-like endome - trioid adenocarcinomas) were classified as SOX9+LGR5+. In con - trast, copy-number-low tumors have a lower SOX9+LGR5+ signal (Wilcoxon text, P value 5.91 × 10−6) (Supplementary Table 3). We then explored the expression of specific epithelial markers in microarray expression data available from peritoneal biopsies from donors with endometriosis 31 (Supplementary Table 3). As expected, endometriotic peritoneal lesions upregulate markers characteristic of proliferative endometrium (SOX9+ and preciliated markers) compared with normal peritoneum (Fig. 3c). In particular, peritoneal lesions upregulate markers specific for the SOX9+LGR5+ subset (such as WNT7A and KRT17) with expression levels similar to those in proliferative endometrium (Fig. 3c). 3c). In contrast, markers of secretory epithelial cells, PAEP and SCGB2A2, or ciliated cells, PIFO and TP73, are expressed at similar levels as in normal peritoneum (Extended Data Fig. 6b). In summary, our analysis suggests that dysfunctional epithelium is a major driver of endometrial disease. By defining the transcrip - tomes of our two SOX9 populations, we can show the specific cell signals that dominate in endometrial carcinomas and endometriosis. Effect of microenvironments on epithelial identity. Having defined these distinct epithelial cell states and their potential role in pathology, we focused next on the transcription factors (TFs) that regulate epithelial differentiation throughout the menstrual cycle by analyzing expression of TFs and their consensus target genes32 (Fig. 4a and Supplementary Table 4). There is high activity of WNT targets (for example, FOXJ1) in the ciliated epithelium. There is high activity of WNT targets (for example, FOXJ1) in the ciliated epithelium. In contrast, the glandular subsets show high expression for TFs induced by WNT inhibition (for example, CSRNP1 and FOXO1) and NOTCH activation (for example, HES1 and HEY1). This suggests different roles for NOTCH and WNT in shaping the identity and function of ciliated versus secretory cells. To investigate the cell signals operating in the lumenal and glandular microenvironments that could influence differentiation into ciliated and secretory lineages, we used spatial transcriptomics and performed clustering on the 10x Genomics Visium spots assigned to epithelial subsets. We resolve five clusters corresponding to cells in the lumenal (one cluster), functional (two clusters) and basal (two clusters) layers (Fig. 4b). In addition to cell-type-specific markers, signatures of WNT and NOTCH signaling pathways are present in distinct endometrial regions (Fig. 4c and Supplementary Table 5). 4c and Supplementary Table 5). Genes involved in the WNT pathway, FOXJ1 and LGR5, are highly expressed at the lumenal surface while NOTCH2 is enriched in glands in the functional layer. To validate expression of NOTCH2 and WNT7A in these compartments, we stained uterine tissue with smFISH probes for both genes alongside EPCAM using immunohis - tochemistry (IHC) (Fig. 4d,e). Lumenal and glandular epithelial cells were classified automatically based on EPCAM expression, and the distance of the signal from the lumen was then measured (Methods). Our results show that NOTCH2 expression increases in glands mov - ing away from the lumen while WNT7A expression is higher in the lumenal epithelium compared with glands (Fig. 4e). By contrast, the noncanonical WNT molecule WNT5A was mainly expressed in stro - mal cells surrounding the glands (Extended Data Fig. 7a). 7a). These findings suggest that canonical WNT is downregulated in the glandular microenvironment where noncanonical WNT pathways dominate. To investigate how surrounding cells may shape signaling in the surface and glandular compartments, we developed CellPhoneDB v.3.0, an updated version of our cell-cell communication pipeline that takes into account spatial cellular colocalization when map - ping ligand-receptor pairs33 (Methods and Fig. 5a). CellPhoneDB considers the multimeric composition of the majority of ligands and receptors, which is highly relevant for the complex regulation of WNT signaling (Fig. 5b). 5b). We define three endometrial microen - vironments centered on epithelial cells based on the cellular coordinates provided by cell2location:  lumenal-preciliated, ciliated and SOX9+LGR5+ epithelium (proliferative phase) and ciliated and lumenal (secretory phase);  functional-SOX9+ proliferative epithelium, immune and eS (proliferative phase) and immune, glandular and dS (secretory phase); and  basal-SOX9+LGR5− and fibroblasts C7. We ran CellPhoneDB on each of the microenvironments (Supplementary Table 6) and found significant epithelialstromal interactions (log fold change > 0.02 and false discovery rate (FDR) < 0.005). NOTCH interactions are mainly mediated by the epithe - lial compartment (Fig. 5c), with the NOTCH ligand JAG1 more highly expressed in the lumen than in glands, as shown by smFISH (Extended Data Fig. 7b,c). As well as JAG1 being coexpressed with HEY1, sparse JAG1 high epithelial cells are often adjacent to JAG1low cells expressing HEY1 (Extended Data Fig. 7c). 7c). A similar lateral NOTCH inhibition model has been described in the gut 34. WNT ligands are expressed by both epithelial and stromal cells (Fig. 5c); the latter express WNT agonists that can potentially bind the cognate WNT receptors expressed by all epithelial subsets during the proliferative phase (Fig. 5c). Focusing on genes that are differentially expressed following ovulation, glandular secretory subsets show a dramatic decline in WNT receptor expression, poten - tially limiting the activity of this pathway (Fig. 5c). They also show a decrease in the WNT target AXIN2 when compared with their lumenal counterparts (Extended Data Fig. 7d). In addition, decidu - alized stromal cells express significantly higher levels of DKK1, a potent inhibitor of the WNT pathway, than their nondecidualized counterparts. Expression of DKK1 surrounding the glands of the secretory endothelium is also found in spatial transcriptomics (Fig. 5d). 5d). Overall, these findings strongly suggest that WNT signaling is inhibited in the secretory cell lineages, meaning that NOTCH sig - naling will then dominate (Fig. 5e). Response of endometrial organoids to ovarian hormones. To test our predictions on the potential roles of WNT and NOTCH signaling pathways on endometrial epithelium in vitro, we first profiled endometrial organoids at a single-cell level to benchmark this model system against our in vivo data. We derived organ - oids10 from three different donors and primed them as previously described with estrogen for 48 h, followed by stimulation with progesterone, prolactin and cyclic AMP (cAMP) in the presence of estrogen for 4 d (ref. 10) (Fig. 6a and Supplementary Tables 7 and 8). To identify epithelial cells in the organoids, we looked for markers specific to the clusters. To identify epithelial cells in the organoids, we looked for markers specific to the clusters. Before hormonal treatment, the majority of cells within the organoids are proliferative (expressing TOP2A, PCNA) and all express the estrogen receptor ESR1 (Fig. 6b-d and Extended Data Fig. 8a,b). Two additional populations emerge when the organoids are treated with estrogen:  an estrogen-induced population expressing the progesterone receptor (PGR), a target gene of estrogen, and  a preciliated population sharing markers with the equivalent cluster defined in vivo. Upon further stimulation with progesterone, markers of more advanced stages of differentiation emerge in both secretory and ciliated populations. Markers of secreted products (PAEP, DEFB1) and glands (SCGB2A2) are both seen. Ciliated cells express typical markers (FOXJ1, TP73) and closely match their in vivo counterparts. Ciliated cells express typical markers (FOXJ1, TP73) and closely match their in vivo counterparts. IHC of endometrial organoids confirmed the expression of glandular and ciliary markers after hormonal stimulation (Fig. 6e). We next assessed how closely the hormonal responses of the organoids that generate glandular and ciliated cells relate to their in vivo counterparts. To focus on the hormonal responses to estrogen and progesterone, we performed a quantitative assessment of organoids exposed to estrogen and progesterone while maintaining them in their expansion medium (ExM)10. Next, we projected the epithelial in vivo reference data onto the hormone-treated in vitro epithelial subsets. Assignments were made based on logistic regression predic - tions. In the absence of progesterone, the SOX9+ populations are the best match for estrogen-induced cells while preciliated organoid cells align with their preciliated in vivo counterparts (Fig. 6f, Extended Data Fig. 6f, Extended Data Fig. 8c and Supplementary Table 9). In response to hormones, a large fraction (over a quarter of secretory cells) correspond to glan - dular epithelium, and all ciliated cells match perfectly with their in vivo counterparts, indicating that organoids respond similarly to hormones as in vivo (Fig. 6f and Supplementary Table 9). To determine whether the pathways driving hormonally induced differentiation toward the ciliated and secretory lineages are similar in vitro and in vivo, we looked at the differential expression and activity of lineage-specific TFs (Fig. 6g and Supplementary Table 10). We calculated differentially expressed or active TFs in the hor - monal subsets in vivo and in vitro. WNT-activated TFs (FOXJ1) are present in the ciliated lineage while WNT-inhibitory TFs (CSRNP1, FOXO1) are in the secretory lineage. NOTCH-induced TFs, HEY1 and HES1, are activated in the secretory lineage. NOTCH-induced TFs, HEY1 and HES1, are activated in the secretory lineage. These results indi - cate that ovarian hormones activate similar pathways both in vivo and in vitro. These results mean that it is possible to reconstruct pseudotime and recapitulate cell fate decisions of epithelial cells in response to hormones using endometrial organoids. We performed scRNA-seq on two clonal organoids from the same individual (Extended Data Fig. 8d). Both clones showed similar behavior and were integrated under the same manifold (Extended Data Fig. 8e,f). Annotation of clusters was performed based on known markers, and then clusters expressing the progesterone receptor were selected to reconstruct epithelial differentiation in response to hormones (Fig. 6h and Extended Data Fig. 8g). A subset of cells emerging from the estrogen-induced population differentiate into preciliated cells in response to estrogen and, following additional progesterone, into ciliated cells. The secretory lineage also emerges from the estrogen-induced population. Thus, there is a common progenitor for both lineages. WNT and NOTCH inhibition mediates epithelial differentia - tion. To test the roles of the WNT and NOTCH pathways in ciliated and secretory differentiation, we cultured organoids in the presence of inhibitors of either NOTCH (DBZ or DAPT) or WNT (IWP-2 or XAV939). We used functional, histological and single-cell transcriptomic assays to assess outcomes (Fig. 7a). Organoid viability is high under all conditions (Extended Data Fig. 9a). scRNA-seq analysis reveals a higher proportion of preciliated and ciliated cells and a lower proportion of secretory cells in the presence of DBZ (NOTCH inhibitor) (Fig. 7b-d , Extended Data Fig. 9b,c and Supplementary Table 11). This finding was validated by IHC and quantitative PCR with reverse transcription (RT-qPCR) using two NOTCH inhibitors (DBZ and DAPT) (Fig. 7e and Extended Data Fig. 9d). 7e and Extended Data Fig. 9d). Ciliated cells are virtually absent when WNT is inhibited (presence of XAV939), highlighting the strong dependence on this pathway for ciliary commitment, while the proportion of secre - tory cells is increased under these conditions (Fig. 7b-d, Extended Data Fig. 9b,c and Supplementary Table 11). The drive toward the secretory lineage under WNT-inhibitory conditions (IWP-2 and XAV939) was further validated by RT-qPCR and IHC (Fig. 7e and Extended Data Fig. 9d). These results demonstrate the fine balance between NOTCH and WNT signaling to regulate commitment to endometrial secretory or ciliary lineages. Hormonal stimulation in the presence of WNT and NOTCH inhibitors modified the secretory cell transcriptome (Fig. 7b). To quantify the similarity of the secretory populations that emerged in the presence of inhibitors with their in vivo counterparts, we measured the expression levels of genes encoding secretory products. Expression levels for PAEP and DEFB1 are higher when WNT is inhibited and downregulated with NOTCH inhibition (Fig. 7f ). We validated the increase of one secretory product, PAEP , under WNT-inhibitory conditions by ELISA (Extended Data Fig. 9e). Although PAEP expression levels increase slightly in the presence of WNT inhibitors even in the absence of hormones, although this is not significant (Fig. 7f). In addition, we built a logistic regression model training the data on the organoids stimulated with distinct hormonal and inhibitory conditions. Computational projection of the in vitro dataset onto the in vivo dataset shows similar overlaps of the secretory populations emerging from WNT-inhibitory con - ditions and controls with their in vivo counterparts (Fig. 7g and Supplementary Table 12). This shows that WNT inhibitors target a specific gene module relevant for glandular secretions. This shows that WNT inhibitors target a specific gene module relevant for glandular secretions. We next dissected the regulatory programs in the secretory lin - eage by comparing expression and activity of TFs between populations emerging after treatment with and without hormones and WNT inhibition. The expression of NOTCH-regulated TFs (HEY1) is only upregulated in the presence of WNT inhibitors when hor - mones are present. This probably explains why WNT inhibitors are not sufficient to induce the secretory lineage on their own (Fig. 7h,i and Supplementary Table 13). In the presence of hormones, WNT inhibition represses TFs characteristic of the ciliated lineage (FOXJ1, TP73, RFX2) (Fig. 7h,i and Supplementary Table 13). Switching off these genes therefore drives secretory lineage differentiation. To summarize, we show that by inhibiting NOTCH and WNT pathways it is possible to influence cell fate decisions between cili - ary and secretory differentiation. This effect depends on the presence of estrogen and progesterone, and we can dissect the dialog between signaling pathways and hormonal stimulation in endome - trial differentiation.",
         "SOX9+ epithelial cells and eS are characteristic of the regenerating proliferative phase.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "45",
         "What is the role of Rec8-cohesins in meiosis?",
         "Tomoya S Kitajima et al.,2018, DOI:https://doi.org/10.1111/dgd.12410:\n Overall chromosome dynamics during meiosis I in mouse oocytes\nIn mammals, oocytes undergo premeiotic S‐phase, followed by meiotic prophase, including the leptotene, zygotene and pachytene stages, before arresting at the dictyate stage in fetal ovaries. This cell cycle arrest at meiotic prophase is maintained for a long period including the oocyte growth phase in which the oocyte grows into an extraordinarily large cell (approximately 80\nμ\nm in diameter in mouse oocytes and approximately 130\nμ\nm in human oocytes), and the arrest is released shortly before ovulation in adult stages (Chiang\net al\n.\n2012\n; Nagaoka\net al\n.\n2012\n; Jones & Lane\n2013\n; Webster & Schuh\n2017\n). Fully grown oocytes have a characteristic nucleus called the germinal vesicle (GV) in which chromosomes are compacted around the nucleolus and nuclear membrane. These GV‐stage oocytes are competent for M‐phase entry; they enter M‐phase around the time of ovulation or when they are experimentally isolated and culturedin vitro\n(Jaffe & Norris\n2010\n).\nUpon entry into M‐phase of meiosis I, which is accompanied by nuclear envelope breakdown (NEBD; also called GV breakdown, GVBD), chromosomes are condensed and microtubule polymerization is activated (Dumont\net al\n.\n2007\n; Schuh & Ellenberg\n2007\n) (Fig.\n1\n). Polymerized microtubules initially self‐assemble an apolar spindle and relocate chromosomes to the surface of the ball‐like spindle. The spindle gradually establishes bipolarity, pushing the minus‐ends of anti‐parallel microtubules into opposite directions (Schuh & Ellenberg\n2007\n; Breuer\net al\n.\n2010\n). As the spindle bipolarizes, the chromosomes move toward the spindle equator through the spindle surface, resulting in a characteristic chromosome arrangement called the prometaphase belt (Kitajima\net al\n.\n2011et al\n.\n2011\n). The microtubules then stretch the chromosomes, orienting homologous kinetochores into opposite spindle poles. Stably stretched chromosomes align at the spindle equator, resulting in a chromosome arrangement called the metaphase plate. Finally, homologous chromosomes are separated and segregated toward opposite spindle poles, allocating one set of chromosomes into an egg and the other into a polar body (Fig.\n1\n). The egg is arrested at metaphase of meiosis II, a stage for sperm entry.\nFigure 1.\nOpen in a new tabFigure 1.\nOpen in a new tab\nAcentrosomal spindle assembly in oocytes. Multiple microtubule‐organizing center (MTOC) clusters are located around the nucleus and in the cytoplasm in germinal vesicle (GV)‐stage oocytes. Upon nuclear envelop breakdown (NEBD), the MTOCs are fragmented and distributed around chromosomes, forming an apolar spindle. The spindle gradually sort anti‐parallel microtubules and push their minus‐ends into opposite poles, which is accompanied with MTOC regrouping into the two poles and chromosome congression into the spindle equator. The spindle microtubules then stretch chromosomes by pulling their homologous kinetochores toward opposite spindle poles. The chromosomes are eventually segregated into an egg and a polar body.\nGradual spindle assembly without centrosomesGradual spindle assembly without centrosomes\nOne essential process for correct chromosome segregation in both mitosis and meiosis is the formation of a bipolar spindle. In mitosis of somatic cells, two centrosomes containing centrioles serve as locations where a majority of spindle microtubules are polymerized (Prosser & Pelletier\n2017\n) and thus, immediately determine two spindle poles after entry to M‐phase. This process provides a spatial organization in which microtubules can attach chromosomes to opposite spindle poles, which is essential for equal chromosome segregation into two daughter cells.\nIn oocytes, the process of bipolar spindle formation is unique. Mammalian oocytes have no centrioles (Szollosi\net al\n.\n1972\n) and thus, have no centrosomes that predetermine two spindle poles. In mouse oocytes, multiple clusters of microtubule‐organizing centers (MTOCs) are distributed around the nucleus and in the cytoplasm before entry into M‐phase (Calarco\n2000\n; Schuh & Ellenberg\n20072000\n; Schuh & Ellenberg\n2007\n; Clift & Schuh\n2015\n) (Fig.\n1\n). Upon entry into M‐phase, the MTOCs are fragmented into small pieces and randomly distributed around chromosomes (Clift & Schuh\n2015\n). A majority of microtubules are polymerized at these MTOCs, depending on the activity of RanGTP, and they initially self‐assemble an apolar spindle (Schuh & Ellenberg\n2007\n). The fragmented MTOCs are gradually regrouped and segregated toward two spindle poles, depending on the activity of Kif11 (an orthologue of Xenopus Eg5) (Breuer\net al\n.\n2010\n; Clift & Schuh\n2015\n), a kinesin that crosslinks anti‐parallel microtubules and pushes their minus‐ends into opposite directions (Kapitein\net al\n.\n2005\n). The segregated MTOCs are distributed around the spindle poles, which are broader than those defined by point centrosomes in mitosis (Breuer\net al\n.\n2010\n; Clift & Schuh\n2015\n). This gradual spindle bipolarization needs a lengthy period of 3–5 h in mouse oocytes in an\nin vitro\nculture (Dumontin vitro\nculture (Dumont\net al\n.\n2007\n; Schuh & Ellenberg\n2007\n; Breuer\net al\n.\n2010\n; Clift & Schuh\n2015\n).\nIn human oocytes, the bipolar spindle assembly requires an even longer period (approximately 16 h) (Holubcová\net al\n.\n2015\n). A key difference between human and mouse oocytes is the absence of MTOC clusters in human oocytes. Live imaging experiments using human oocytes that could not be used for\nin vitro\nfertilization revealed that a majority of microtubules are polymerized around chromosomes, which depends on chromosome‐mediated activation of RanGTP (Holubcová\net al\n.\n2015et al\n.\n2015\n). The polymerized microtubules self‐assemble initially an apolar spindle, which is slowly converted into a bipolar spindle. The transition from an apolar to a bipolar state exhibits unstable bipolarity in the spindle. A bipolar spindle frequently fails to maintain its bipolarity and becomes apolar or multipolar. This spindle instability in human oocytes, which is more pronounced than that observed in mouse oocytes, may be attributable to the absence of MTOC clusters (Holubcová\net al\n.\n2015\n).\nThe microtubule organization in an apolar spindle is not suited for the formation of proper kinetochore‐microtubule attachments because kinetochores are at risk for microtubule attachments in random directions (Kitajima\net al\n.\n2011\n; Davydenko\net al\n.\n2013et al\n.\n2011\n; Davydenko\net al\n.\n2013\n). Assuming that a fraction of kinetochore‐microtubule attachments form in the apolar spindle and could fail to be removed through spindle bipolarization, such attachments would result in merotelic attachment in which a single kinetochore is erroneously attached by microtubules to opposite spindle poles (Holubcová\net al\n.\n2015\n; Webster & Schuh\n2017\n). Thus, the gradual assembly of the acentrosomal spindle provides an explanation for the higher frequency of errors in kinetochore‐microtubule attachments at the early stages of M‐phase in oocytes than that in centrosomal somatic cells and spermatocytes.\nLess focused spindle poles\nThe acentrosomal bipolar spindle in oocytes exhibits a barrel‐like shape, which is different not only from centrosomal spindles in somatic cells but also from acentrosomal spindles in mouse early embryos. The barrel‐shaped acentrosomal spindle in mouse oocytes has a broad distribution of MTOCs around the spindle poles (Fig.1\n), whereas acentrosomal spindles in embryos at the 8‐cell stage exhibit well‐focused spindle poles with a confined distribution of MTOCs (Courtois\net al\n.\n2012\n). Thus, the absence of centrosomes alone does not account for the less focused spindle poles in oocytes. The difference in the spindle poles between oocytes and 8‐cell‐stage embryos is at least partly attributed to a difference in cytoplasmic size. Every cell division during early embryogenesis reduces cytoplasmic size by approximately half, which is accompanied by a reduction in spindle volume and focusing of spindle poles (Courtois\net al\n.\n2012\n). If half the cytoplasmic volume is artificially removed from oocytes, the oocyte assembles a smaller spindle with a restricted distribution of MTOCs at the poles (Kyogoku & Kitajima\n20172017\n). Conversely, if the cytoplasmic volume is increased through cell fusion with an enucleated oocyte, the fused oocyte exhibits a larger spindle with a broader distribution of MTOCs at the poles. The spindle with broad MTOC distributions at the poles is prone to chromosome misalignment, likely through improper attachments of kinetochores with microtubules emanating from the broadly distributed MTOCs within one pole (Kyogoku & Kitajima\n2017\n). These data suggest that the large cytoplasm, a unique feature of fully grown oocytes, is limiting for the spatial focusing of spindle poles and for the efficiency of chromosome alignment, a prerequisite for correct chromosome segregation.\nChromosome cohesion\nChromosome cohesion is essential for accomplishing correct chromosome segregation in both mitosis and meiosis (Nasmyth & Haering\n2009\n). Cohesion between sister chromatids, which contain DNAs duplicated during S‐phase, is mediated by four‐subunit protein complexes called cohesins (Fig.\n22\n). In somatic cells, a cohesin complex is composed of structural maintenance of chromosomes (SMC) proteins Smc1\nα\nand Smc3, the kleisin subunit Rad21, and either STAG1/SA1 or STAG2/SA2 (Losada\net al\n.\n1998\n,\n2000\n; Waizenegger\net al\n.\n2000\n). In mitosis of somatic cells, sister chromatids are segregated into daughter cells. This process requires the establishment of sister kinetochore biorientation in which sister kinetochores are stably attached by microtubules to opposite spindle poles before entry into anaphase onset (Lampson & Cheeseman\n2011\n; Sarangapani & Asbury\n2014\n). The stabilization of kinetochore‐microtubule attachment requires tension across paired kinetochores (Nicklas & Koch\n1969\n; King & Nicklas\n20001969\n; King & Nicklas\n2000\n). Cohesin‐mediated cohesion between sister chromatids enables generation of tension across sister kinetochores by providing a resistant force against the microtubule‐mediated forces that pull sister kinetochores toward opposite spindle poles. Improper attachments that fail to pull sister kinetochores into opposite spindle poles lack tension, and such tension‐less attachments are unstable and subsequently removed for error correction (Lampson & Cheeseman\n2011\n; Sarangapani & Asbury\n2014\n). Thus, cohesin is required for error correction and stabilization of proper kinetochore‐microtubule attachments.\nFigure 2.\nOpen in a new tabFigure 2.\nOpen in a new tab\nChromosome structures in mitosis and meiosis I. In mitosis, sister chromatids are held together by cohesin. Sister kinetochores are pulled by spindle microtubules into the opposite spindle poles (sister biorientation). In meiosis I, homologous chromosomes are held together through chiasmata that are supported by cohesin‐mediated sister chromatid cohesion along chromosome arms. Sister kinetochores are associated with each other and thus co‐oriented. Homologous kinetochores are pulled by spindle microtubules into the opposite spindle poles (homologue biorientation).\nMeiotic chromosome cohesion\nIn meiotic cells, meiosis‐specific cohesin subunits, such as the kleisin subunits Rec8 and Rad21L, Smc1\nβ\n, and STAG3/SA3, are expressed and form meiotic cohesin complexes (Prieto\net al\n.\n2001\n; Revenkova\net al\n.\n2001\n; Lee\net al\n.\n2003\n; Ishiguro\net al\n.\n2011\n; Lee & Hirano\n2011.\n2001\n; Revenkova\net al\n.\n2001\n; Lee\net al\n.\n2003\n; Ishiguro\net al\n.\n2011\n; Lee & Hirano\n2011\n). Rec8‐containing meiotic cohesins are responsible for the maintenance of cohesion between sister chromatids that are generated during premeiotic S‐phase (Tachibana‐Konwalski\net al\n.\n2010\n). In addition, Rec8‐cohesins also support chiasmata, physical links generated by meiotic recombination between homologous chromosomes, through mediating sister chromatid cohesion along chromosome arms (Fig.\n2\n). Thus, Rec8‐cohesin supports the meiosis‐specific structure of chromosomes in which both homologous chromosomes and sister chromatids are held together.In contrast to mitosis in which sister chromatids are segregated, during meiosis I, homologous chromosomes are segregated. To accomplish this meiosis I‐specific pattern of chromosome segregation, homologous kinetochores but not sister kinetochores must be pulled by spindle microtubules into opposite spindle poles prior to the onset of anaphase (Watanabe\n20122012\n). As in mitosis, the stabilization of kinetochore‐microtubule attachments requires tension. Supported by Rec8‐cohesins along chromosome arms, chiasmata enable the generation of tension across homologous kinetochores when they are correctly pulled by microtubules into opposite spindle poles. Correct attachments are therefore stabilized, whereas improper attachments, which fail to generate tension, are unstable and are thus corrected. Thus, Rec8‐containing cohesin is essential for the establishment of correct kinetochore‐microtubule attachments that enable meiotic chromosome segregation. Moreover, Rec8‐containing cohesin at kinetochores may mediate the establishment of the meiotic kinetochore geometry that facilitates correct microtubule attachments, as discussed later.\nTension‐sensing mechanisms in mitosisTension‐sensing mechanisms in mitosis\nHow do kinetochores sense tension? It is likely that multiple tension‐sensing mechanisms function in mitosis. Thus far, several lines of experimental evidence have been provided for at least three proposed models – the spatial position model, spatial separation model and kinetochore‐intrinsic model (Lampson & Cheeseman\n2011\n; Sarangapani & Asbury\n2014\n; Monda & Cheeseman\n2015\n).\nFirst, the spatial position model suggests that tension‐less kinetochore‐microtubule attachments are destabilized when they are located in the vicinity of spindle poles (Monda & Cheeseman\n2015\n). Because tension‐less attachments would apply imbalanced spindle forces on kinetochores, the kinetochores would move toward one spindle pole. The Aurora A kinase, which is enriched at the spindle pole, generates a spatial gradient of its kinase activity around the pole and preferentially phosphorylates the substrates on kinetochores located in the vicinity of spindle poles (Fig.\n33\nA) (Anna\net al\n.\n2015\n; Chmátal\net al\n.\n2015\n). Aurora A phosphorylates kinetochore proteins that anchor microtubules, such as the subunits of the Ndc80 kinetochore complex (Anna\net al\n.\n2015\n), which exhibits reduced affinity for microtubules when phosphorylated (Cheeseman\net al\n.\n2006\n; DeLuca\net al\n.\n2006\n). Thus, this mechanism can preferentially destabilize kinetochore‐microtubule attachments that fail to generate balanced tension using a spatial cue within the spindle.\nFigure 3.\nOpen in a new tabModels of Aurora kinase activities on mitotic and meiotic kinetochores. In both mitosis and meiosis I, Aurora A activity is enriched around spindle poles, whereas Aurora B (and Aurora C in meiosis I of oocytes) activity is enriched at the inner centromere. (A) In mitosis, kinetochores on misaligned chromosomes are highly phosphorylated by both Aurora A and Aurora B because the kinetochores are positioned near spindle poles and the microtubule attachment sites are not separated from the inner centromere. This phosphorylation reduces the stability of microtubule attachments. On aligned chromosomes, kinetochores are positioned far from spindle poles and the microtubule attachment sites are spatially separated from the inner centromere because sister kinetochores are pulled into opposite poles. Therefore, the phosphorylation levels of these kinetochores are low, and thus kinetochore‐microtubule attachments are stable. (B) In meiosis I, kinetochores on misaligned chromosomes are highly phosphorylated by both Aurora A and Aurora B/C as in mitosis. On aligned chromosomes, kinetochores are positioned far from spindle poles and thus Aurora A activity is low. However, kinetochore‐microtubule attachment sites are not separated from the inner centromere because sister kinetochores are pulled into one pole. Therefore, Aurora B/C activity at the kinetochores remain high upon chromosome alignment, and thus the attachment is still not stabilized. During metaphase, the level of PP2A‐B56 phosphatase at kinetochores progressively increases and counteracts Aurora B/C activity, which eventually stabilizes kinetochore‐microtubule attachments.Second, the spatial separation model is based on the fact that Aurora B kinase, which partly shares kinetochore substrates with Aurora A and destabilizes kinetochore‐microtubule attachment, is located at a centromeric domain between sister kinetochores (Tanaka\net al\n.\n2002\n; Andrews\net al\n.\n2004\n; Cimini\net al\n.\n2006\n). This domain, called the inner centromere, is spatially separated from kinetochore‐microtubule attachment sites when sister kinetochores are correctly pulled toward opposite spindle poles. This spatial separation allows kinetochore substrates to avoid Aurora B‐dependent phosphorylation (Liu\net al\n.\n2009\n; Welburn\net al\n.\n2009\n), leading to the stabilization of kinetochore‐microtubule attachment (Fig.\n3\nA). Thus, the spatial separation mechanism can selectively stabilize correct kinetochore‐microtubule attachments by sensing tension‐dependent changes in the spatial organization of sister kinetochores.Last, the kinetochore‐intrinsic model assumes that kinetochores have an intrinsic capacity to stabilize microtubule attachment when mechanical tension is applied (Sarangapani & Asbury\n2014\n). In support of this model, purified kinetochores, which do not contain Aurora kinases, exhibit the capacity to stabilize microtubule attachment in response to mechanical tension (Akiyoshi\net al\n.\n2010\n). This capacity depends on the XMAP215 family member Stu2, which binds to the Ndc80 kinetochore complex, in budding yeast (Miller\net al\n.\n2016\n). How Stu2 strengthens tension‐bearing kinetochore‐microtubule attachments remains to be elucidated. Although kinetochore‐intrinsic mechanisms for sensing tension are far from understood, it is likely that they contribute to the stabilization of correct kinetochore‐microtubule attachments.In summary, in mitosis, the establishment of correct kinetochore‐microtubule attachment can be ensured by at least three different mechanisms. Importantly, however, the relative physiological importance of these different mechanisms remains unclear and may differ among different cell types and different stages within M‐phase.\nUnique geometry of meiotic kinetochores\nMeiotic kinetochores have unique geometry to accomplish homologous chromosome segregation, a pattern specific for meiosis I (Watanabe\n2012\n). In this meiotic geometry, sister kinetochores are spatially associated with each other, which presumably facilitates the orientation of microtubule‐attachment surfaces of the sister kinetochores toward the same direction (Fig.\n22\n). This sister kinetochore co‐orientation would help microtubules attach the sister kinetochores to a same spindle pole (monopolar attachment). The monopolar attachment of sister kinetochores is needed to pull homologous kinetochores into opposite spindle poles, which is a prerequisite for correct homologous chromosome segregation. A protein required for the maintenance of sister kinetochore association in mammals is MEIKIN (Kim\net al\n.\n2015\n), which is specifically expressed in meiotic cells and localizes to kinetochores through binding to the constitutive kinetochore component CENP‐C. MEIKIN recruits Polo‐like kinase Plk1 to kinetochores, another factor needed for the maintenance of the sister kinetochore co‐orientation (Kim\net al\n.\n2015et al\n.\n2015\n). Although how MEIKIN and Plk1 maintain the sister kinetochore association is unclear, it is possible that they achieve this goal through regulating cohesion between sister kinetochores mediated by Rec8‐cohesin. In mouse oocytes, artificial cleavage of Rec8 at kinetochores results in a loss of sister kinetochore association (Tachibana‐Konwalski\net al\n.\n2013\n). In fission yeast, Rec8‐cohesin mediates the cohesion between sister kinetochores, which is established specifically in meiotic cells (Sakuno\net al\n.\n2009\n). The maintenance of sister kinetochore cohesion depends on Moa1 and Plo1, the functional homologues of MEIKIN and Polo‐like kinase, respectively (Yokobayashi & Watanabe\n2005\n; Sakuno\net al\n.\n2009\n). In budding yeast, the monopolin complex is needed for monopolar attachment of sister kinetochores (Tóth\net al\n.\n2000\n). Structural studies suggest that monopolin crosslinks sister kinetochores to form meiosis I‐specific kinetochore geometry (Corbett\net al\n.\n2010et al\n.\n2010\n). Thus, the unique geometry of meiotic kinetochores is established by factors that physically mediate sister kinetochore association.\nEffect of meiotic kinetochore geometry on spatial separation‐based tension‐sensing mechanisms\nThe unique geometry of meiotic kinetochores is essential for accomplishing the meiosis I‐specific pattern of chromosome segregation but may influence the tension‐sensing mechanisms for error correction and selective stabilization of kinetochore‐microtubule attachments. In particular, among the three tension‐sensing mechanisms in mitosis described above, the spatial separation model relies on the mitotic geometry of sister kinetochores (Fig.\n3\nA). Nevertheless, a spatial separation mechanism similar to that in mitotic kinetochores may work on meiotic kinetochores (Watanabe\n20122012\n). In support of this idea, in fission yeast meiosis I, the inner centromere at which Aurora kinase is located appears to be spatially separated from kinetochore‐microtubule attachment sites when homologous kinetochores are pulled into opposite spindle poles (Sakuno\net al\n.\n2011\n). However, in mouse oocytes, Aurora B and Aurora C (a paralogue expressed in meiotic cells) remain in close proximity to kinetochore‐microtubule attachment sites even after homologous kinetochores are pulled into opposite spindle poles (Fig.\n3\nB) (Yoshida\net al\n.\n2015\n). Acute inhibition of Aurora B and C on the stretched chromosomes increases stable correct kinetochore‐microtubule attachments and decreases incorrect attachments (Yoshida\net al\n.\n2015et al\n.\n2015\n). These results suggest that the activity of Aurora B and C lowers the stability of correct kinetochore‐microtubule attachments and allows the formation of incorrect attachments on stretched chromosomes. Thus, it is likely that the contribution of the spatial separation mechanism for the stabilization of microtubule attachments on meiotic kinetochores in mammalian oocytes is low compared with that on mitotic kinetochores in somatic cells. This hypothesis explains why meiotic kinetochores in oocytes exhibit more frequent attachment errors than do mitotic kinetochores in somatic cells.\nSpindle pole‐based mechanisms correct kinetochore‐microtubule attachments during prometaphaseSpindle pole‐based mechanisms correct kinetochore‐microtubule attachments during prometaphase\nGiven that the contribution of the spatial separation mechanism on meiotic kinetochores to the stabilization of microtubule attachments is small in oocytes, what are the principal mechanisms? As in mitosis, several lines of evidence suggest that spindle pole‐based mechanisms destabilize kinetochore‐microtubule attachments during meiosis I in acentrosomal oocytes. High‐resolution live imaging studies indicate that during prometaphase I, chromosomes frequently exhibit vigorous movements between spindle poles, likely due to imbalanced spindle forces applied to kinetochores through improper kinetochore‐microtubule attachments (Kitajima\net al\n.\n2011\n). These attachments are destabilized as kinetochores approach the spindle poles, depending on the Aurora A activity enriched around the acentrosomal poles of the meiotic spindle (Fig.\n3\nB) (Chmátal\net al\n.\n20153\nB) (Chmátal\net al\n.\n2015\n). Accordingly, the vast majority of chromosomes can stay no longer than 30 min within 4\nμ\nm of the spindle poles (Kitajima\net al\n.\n2011\n; Chmátal\net al\n.\n2015\n; Kyogoku & Kitajima\n2017\n). This removal of attachments near the spindle pole would allow the chromosome to reorient and move toward the spindle equator, which would promote subsequent attempts for the formation of correct kinetochore‐microtubule attachments (Chmátal\net al\n.\n2015\n).\nA mechanism for coordinated stabilization of kinetochore‐microtubule attachments in oocytes\nA marked difference between mitosis in somatic cells and meiosis I in oocytes is temporal changes in the stability of kinetochore‐microtubule attachments through prometaphase and metaphase. In mitosis, the stability of kinetochore‐microtubule attachments is thought to sharply increase when sister kinetochores establish biorientation (Lampson & Cheeseman\n2011\n; Godek\net al\n.\n20152011\n; Godek\net al\n.\n2015\n). In contrast, during meiosis I in oocytes, the stability of kinetochore‐microtubule attachments gradually increases during prometaphase and metaphase (Brunet\net al\n.\n1999\n; Kitajima\net al\n.\n2011\n; Gui & Homer\n2012\n; Lane\net al\n.\n2012\n; Davydenko\net al\n.\n2013\n), which takes approximately 6–12 h in mice and 15–20 h in humans. This gradual increase in the stability of kinetochore‐microtubule attachments is controlled by temporal changes in Cdk1 activity (Davydenko\net al\n.\n2013\n). In contrast to mitosis in which Cdk1 activity reaches maximum immediately after entry to M‐phase, in meiosis I, the Cdk1 activity of oocytes gradually increases during prometaphase and metaphase (Choi\net al\n.\n1991\n; Gavin\net al\n.\n1994\n; Polanski\net al\n.\n1998\n; Davydenko\net al\n.\n2013\n). Cdk1 phosphorylates the kinetochore component BubR1 (Elowe\net al\n.\n2010\n; Kruse\net al\n.\n2013\n), which binds and recruits the phosphatase PP2A‐B56 to kinetochores upon phosphorylation (Huang\net al\n.\n2008et al\n.\n2008\n; Elowe\net al\n.\n2010\n; Suijkerbuijk\net al\n.\n2012\n; Kruse\net al\n.\n2013\n; Yoshida\net al\n.\n2015\n). PP2A‐B56 antagonizes the attachment destabilizers Aurora B and Aurora C on kinetochores (Foley\net al\n.\n2011\n; Suijkerbuijk\net al\n.\n2012\n; Yoshida\net al\n.\n2015\n). Thus, the gradual increase in Cdk1 activity results in a gradual stabilization of kinetochore‐microtubule attachments during prometaphase and metaphase in oocytes (Fig.\n3\nB). This pathway likely affects the stability of the attachments in a coordinated fashion among all chromosomes. Such coordinated stabilization of kinetochore‐microtubule attachments is also observed in mitosis in somatic cells (Godek\net al\n.\n2015\n). In mitosis, however, coordinated stabilization sharply occurs at the transition of prometaphase to metaphase, depending on a progressive reduction in Cdk1 activity through Cyclin A degradation during prometaphase (Kabeche & Compton\n20132013\n). This process is in striking contrast to the gradual coordinated stabilization by the progressive increase in Cdk1 activity in meiosis I of oocytes.\nGradual increase in the stability of kinetochore‐microtubule attachments allows temporal coordination with the progression of acentrosomal spindle assembly\nKinetochore‐microtubule attachments are sharply stabilized in somatic cells where bipolar spindle formation is rapid. On the other hand, attachments are gradually stabilized in oocytes where bipolar spindle formation is slow. Thus, mitosis of somatic cells and meiosis I of oocytes are similar in that the timing of attachment stabilization matches the timing of bipolar spindle formation. It is possible that this similarity reflects the importance of temporal coordination of attachment stabilization with bipolar spindle formation (Fig.\n33\n). In oocytes, at early prometaphase I, when the acentrosomal spindle has not yet established a bipolarized state, the activity of the Cdk1‐BubR1‐PP2A‐B56 pathway is relatively low, and thus, Aurora B/C is allowed to phosphorylate kinetochore substrates (Fig.\n3\nB). This high phosphorylation state prevents kinetochores from precocious stabilization of microtubule attachments in random directions in an apolar spindle. From late prometaphase I to early metaphase I, when the spindle bipolarizes, the activity of Cdk1‐BubR1‐PP2A‐B56 is balanced with the activity of Aurora B/C at kinetochores (Fig.\n3\nB). This balance would moderate the phosphorylation levels of Aurora B/C substrates at kinetochores, which would allow the formation of kinetochore‐microtubule attachments with a relatively low stability. If kinetochores are moved toward spindle poles by imbalanced forces through these attachments,",
         "Rec8-cohesins maintain cohesion between sister chromatids and support chiasmata.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "46",
         "How many viral complete proteomes were in UniProt release 2018_07?",
         "The UniProt Consortium et al.,2018, DOI:https://doi.org/10.1093/nar/gky1049:\n PROGRESS AND DEVELOPMENTS\nGrowth of sequences in UniProt\nThe UniProtKB Proteomes portal (\nhttps://www.uniprot.org/proteomes/\n) provides access to proteomes for over 84 thousand (84 387, release 2018_07) species with completely sequenced genomes. The majority of these proteomes are based on the translation of genome sequence submissions to the INSDC source databases—ENA, GenBank and the DDBJ (\n2\n). To ensure comprehensiveness, complementary pipelines have been developed to supplement these with genomes sequenced and/or annotated by groups such as Ensembl (\n3\n), Vectorbase (\n4\n) and WormBase ParaSite (\n53\n), Vectorbase (\n4\n) and WormBase ParaSite (\n5\n). This has been extended to RefSeq genomes, allowing us to import key genomes of special interest annotated by NCBI’s Eukaryotic Genome Annotation Pipeline. Release 2018_03, for example, saw the inclusion of a set of primate proteomes and a dozen genomes for marine mammals were imported from RefSeq for release 2018_08. Taxa may be specifically targeted by curators to fill gaps in the taxonomic space and additional proteome import can be requested by individual users.\nThe continuing growth of sequenced genomes is a challenge for databases as much of this growth is being driven by sequencing of very similar and almost identical strains (90% of proteins have >90% identity) of the same species. We continue to see exponential growth in many of our datasets, see Figure\n11\n. We are managing this growth by a number of processes. Most critically, a redundancy removal process was first introduced in 2015. This process identifies and removes almost identical proteomes of the same species before their inclusion in UniProtKB (\nhttps://www.uniprot.org/help/proteome_redundancy\n) and places their sequences in UniParc. Currently this process has removed ∼38% all complete proteomes (∼241 million proteins) from UniProtKB. As can be seen in Figure\n1\n, the redundancy reduction both greatly reduced the size of UniProtKB as well as made its growth more scalable. This approach has now been extended from prokaryotes to fungi, resulting in the deprecation of ∼1 million fungal protein records in release 2016_08.\nFigure 1.\nOpen in a new tab\nGrowth of UniProt sequences over the last decade.\nReference proteomesFigure 1.\nOpen in a new tab\nGrowth of UniProt sequences over the last decade.\nReference proteomes\nFor the remaining proteomes we provide a Reference Proteome set (∼9% of total proteomes) selected by the research community and supplemented with selected proteomes from a computational clustering (\n6\n) to provide the best annotated proteome in their cluster. Recently, we have added virus Reference Proteomes (described below) to this list. The growth of Reference Proteome sets is shown in Figure\n2\n. Programmatic access to the non-redundant proteomes included in UniProtKB is provided via the Proteins API (\nhttps://www.ebi.ac.uk/proteins/api/doc/\n) (\n7\n) whilst all proteomes reference, non-reference and redundant in UniProtKB and UniParc are accessible via the Proteome section of the UniProt website (\nhttps://www.uniprot.org/proteomes/\n).\nFigure 2.\nOpen in a new tab\nGrowth of the total number of Complete Proteomes and Reference Proteomes since 2015.Growth of the total number of Complete Proteomes and Reference Proteomes since 2015.\nUniProt reference proteomes are used by user communities such as the Quest for Orthologs (QfO) group as the ‘gold standard’ reference proteome dataset for orthologue benchmarking (\n8\n). The QfO reference proteomes datasets are a compiled subset of the UniProt reference proteomes, comprising well-annotated model organisms and organisms of interest for biomedical research and phylogeny, with the intention to provide broad coverage of the tree of life whilst maintaining a low number of proteomes for the benchmark. Such datasets have been generated annually from the UniProt Knowledgebase (UniProtKB) database for the past seven years. To this end, a gene-centric pipeline has been developed and enhanced over the past year by UniProt. These complete, non-redundant reference proteomes are publicly available at\nftp://ftp.ebi.ac.uk/pub/databases/reference_proteomes/QfOftp://ftp.ebi.ac.uk/pub/databases/reference_proteomes/QfO\n. The datasets are provided either in SeqXML format or as a collection of FASTA files and include genomics coordinates for most proteins. In the last release for QfO in April, the number of species increased from 66 to 78 comprising 1,311,679 non-redundant protein sequences (48 Eukaryotes and 30 bacteria-archaea).At the time of our previous publication, the number of virus Reference Proteomes in UniProtKB stood at 428—practically an order of magnitude less than the number of known viral species (which stands at 4405 according to the International Committee on Taxonomy of Viruses, or ICTV). To improve Reference Proteome Coverage of viruses in UniProt we have undertaken a concerted effort to curate Complete Proteomes and to use these as input for the computational selection of Reference Proteomes. We compiled and curated complete viral genome (proteome) data from INSDC (which feeds into UniProt) as well as a range of other specialized viral genome resources including ViPr (\n9\n), IRD (\n10\n), the HIV database (\nhttps://www.hiv.lanl.gov/\n), Papillomavirus Episteme (\n11\n) and the HCV database (\nhttps://hcv.lanl.gov/\n) and HBV database (\n1211\n) and the HCV database (\nhttps://hcv.lanl.gov/\n) and HBV database (\n12\n). This curation effort included the identification and removal of segmented viruses lacking one or more segments, and increased the number of the Complete Proteomes of viral origin in UniProt to 86 951 (UniProt release 2018_07). Computational clustering of this enhanced viral Complete Proteome set (\n13\n) produced 5,887 viral Reference Proteomes. Note that redundancy removal procedures are not currently applied to viruses, due to the challenges posed by the high number of variants in the small genomes of these species.Technological advances have enabled the sequencing of the genetic material from all the microorganisms in a particular environment without the cultivation of any of the community members. Concurrent advances in bioinformatics have enabled the rapid assembly of genomes from metagenomes (MAGs) and a corresponding input of sequence data into the UniProt database with >4800 assembled proteomes in the database (4839, release 2018_07). The existing data input pipeline is currently based on those submissions to the INSDC which fulfil certain threshold criteria but future plans are to move to using the EBI Metagenomics resource, MGnify (\n14\n), as the main source of metagenome derived assemblies. We will include only those MAGs that show a high level of completeness and a low level of contamination.\nExpert curation progressExpert curation progress\nExpert curation of the literature is critically important to the UniProt databases. The information extracted from scientific publications is stored in the UniProtKB/Swiss-Prot section of the UniProt Knowledgebase and describes functional information both in the form of human readable free-text/controlled syntax summaries and via structured vocabularies such as the Gene Ontology (GO) (\n15\n) or ChEBI (\n16\n). Expert curation is labour intensive, with curators assimilating and evaluating multiple lines of evidence from the text and figures of relevant publications, but this has repeatedly proven to be the most efficient method of extracting all relevant data from a paper. We have previously demonstrated that UniProtKB expert biocuration captures between 35 and 45% of the curatable literature for any given species, rising to 50% of the curatable literature for\nHomo sapiens\n(\n17Homo sapiens\n(\n17\n). UniProtKB/Swiss-Prot entries serve as a source of functional data for the development and enhancement of bioinformatics prediction tools, so we prioritize the capture of functional data that cannot currently be correctly predicted by computational methods, such as proteins which predictive protein signatures would describe as being part of an enzyme family but are in fact non-functional due to the loss of specific amino acid residues. An example of this is PLC1 (UniProtKB:\nQ15111Q15111\n), a member of the phosphoinositide phospholipase C family (InterPro:IPR001192) in which the existence of an asparagine residue in the active site instead of the conserved histidine residue suggests a non-catalytic role for this protein. This has broader applications than the annotation of just one protein as information on protein function is computationally transferred from UniProtKB/Swiss-Prot to sequence-related, but less well studied, proteins in UniProtKB/TrEMBL. Therefore, it is important to ensure we have a broad enough collection of annotated proteins in UniProtKB/Swiss-Prot to add value to new entries as the taxonomic range of fully sequenced proteomes continues to expand.Once a record has been moved into UniProtKB/Swiss-Prot from UniProtKB/TrEMBL, there is a concerted effort to ensure that it is regularly updated to ensure the current understanding of a protein’s activity reflects that in the scientific literature. This task can be difficult, as our knowledge of protein function continues to evolve and finer grained experimental techniques provide new knowledge that may appear to contradict previous observations. An example of this is provided by the proteins involved in N\n6\n-methyladenosine (m\n6\nA) modification which takes place on both coding and non-coding RNAs containing a Pu[G>A]m\n66\nA) modification which takes place on both coding and non-coding RNAs containing a Pu[G>A]m\n6\nAC[A/C/U] (Pu = purine) sequence, the nucleotide A3 of which becomes N6-methylated. The modification acts as a key regulator of mRNA stability: methylation is completed upon the release of mRNA into the nucleoplasm and affects various processes, such as mRNA stability, processing, translation efficiency and editing. The enzymes that catalyse this process were believed to have been fully characterized some years ago, but more recent data has changed our understanding of how these molecules work.\nm\n6\nA methylation is performed by a heteromeric WMM N\n6\n-adenosine-methyltransferase complex, originally thought to be a heterotrimer of METTL3 (UniProtKB:\nQ86U44\n), METTL14 (UniProtKB:\nQ9HCE5\n) and WTAP (UniProtKB:\nQ15007\n), but additional component subunits have recently been identified (\n18Q15007\n), but additional component subunits have recently been identified (\n18\n). Initial work indicated that METTL3 and METTL14 form a tight heterodimer in order to perform catalysis on the preferred motif sequence. Both proteins are members of the MT-A70 family and are classified as S-adenosyl-L-methionine-dependent methyltransferases by sequence prediction resources. Methyltransferase activity has been reported for both (\n19\n). Subsequent structural studies have now shown that that only one protein, METTL3, constitutes the catalytic subunit (\n20–22\n). The other subunit, METTL14, has a degenerate active site that is unable to accommodate donor and acceptor substrates and plays a non-catalytic role in maintaining complex integrity and substrate RNA binding (\n22\n). WTAP appears to serve as a regulatory subunit. A newly identified subunit of the WMM methyltransferase complex, VIRMA (UniProtKB:\nQ69YN4\n) plays a role in mediating mRNA m\n6Q69YN4\n) plays a role in mediating mRNA m\n6\nA methylation in 3′UTRs and near stop codons and in recruiting the methyltransferase core components.\nExpert biocuration has been essential in keeping the relevant UniProtKB/Swiss-Prot entries up to date with each step in the pathway to reflect our understanding of the function of these proteins and the molecules they associate with. Every piece of knowledge we capture is associated with evidences (see\nhttps://www.uniprot.org/help/evidences\nfor further details) to indicate the source of information and we have added a CAUTION comment to the METTL14 entry to both provide some background and inform users that the initially reported methyltransferase activity is an unsafe observation (Figure\n33\n). The correct annotation on the UniProtKB/Swiss-Prot entry will ensure that misleading data is not computationally added to orthologous proteins in UniProtKB/TrEMBL on the basis of an over-predictive protein signature, again highlighting the importance of expert biocuration. Future plans for the manual curation activities in UniProtKB include the development of mechanisms to identify and highlight contradictory information in existing protein entries in order to improve rigor and reproducibility. This example also illustrates the collaborative nature of the UniProt Consortium in that the molecular interactions involved have also been added to the IMEx Consortium (\nwww.imexconsortium.org\n) dataset by UniProt curators (\n23\n), the protein complexes involved curated into the Complex Portal (\nhttps://www.ebi.ac.uk/complexportal/\n) (\n24\n) and both the proteins and complexes have been further annotated using GO (\nhttp://www.geneontology.orghttp://www.geneontology.org\n). New terms were added to the GO to enable this and others were updated. All of these data are subsequently reimported back to enrich the appropriate entries in UniProtKB/Swiss-Prot but are stored and maintained by these domain-specific resources.\nFigure 3.\nOpen in a new tab\nFunctional annotation describing human METTL14 (UniProtKB\nQ9HCE5\n).\nAutomatic annotation progressUniProt's automatic annotation pipelines enrich the unreviewed records in UniProtKB/TrEMBL with classification and functional annotations. InterPro is used to classify sequences at superfamily, family and subfamily levels and to predict the occurrence of functional domains and important sites. InterPro integrates predictive models of protein function, so-called ‘signatures’, from a number of member databases. In UniProtKB/TrEMBL entries, domains from the InterPro member databases PROSITE, SMART or Pfam are predicted and used to automatically provide domain annotations. All automatic annotations are labelled with their evidence/source. UniProt has developed two complementary rule-based prediction systems, UniRule and the Statistical Automatic Annotation System (SAAS) to automatically annotate UniProtKB/TrEMBL in an efficient and scalable manner with a high degree of accuracy. These prediction systems can annotate protein properties such as protein names, function, catalytic activity, pathway membership and subcellular location, along with sequence-specific information, such as the positions of post-translational modifications and active sites. We continue to increase the number of Rules used for annotation and this has now grown to over 6000 in total as shown in Figure4\n.\nFigure 4.\nOpen in a new tab\nGrowth of curated automatic annotation rules within the UniRule system.\nOngoing developments in automatic annotation\nWe are evaluating new methods for computational annotation of function based on based specifically on their domain architecture (DAAC, or Domain Architecture Alignment and Classification) (\n22\n,\n25\n) or any combination of sequence properties and features (ARBA, or Association-Rule-Based Annotator).\nThe Domain Architecture Alignment and Classification system (DAAC, (\n25)) is based on the assumption that not only the presence of domains is important for the protein to perform its function but also their architecture i.e. their arrangement and order. It performs the prediction in four major steps. Firstly, it generates the different ordered combinations of InterPro domains, these combinations are termed domain architectures (DA). Secondly, it calculates the pairwise similarity between DAs using alignment. The third step is that of training. In this phase, clusters of proteins are created, where each cluster represents a single annotation. A threshold of belonging is determined for each cluster while maximising the classification performance, and then annotations corresponding to low performance clusters are discarded. In the final step, the clusters along with their corresponding thresholds serve as models to annotate other query proteins. The fact that DAAC takes into account the arrangement of domains within proteins allows it to make more accurate predictions for specific cases where proteins can have the same set of domains but with different arrangements and hence could perform different functions. Based on UniProt release 2018_05, DAAC generates 7 095 966 EC predictions. 3 880 775 of them are new predictions (not predicted by UniRule or SAAS). Furthermore, DAAC generates 82 285 267 predictions of GO Terms for 20 827 747 distinct proteins in UniProtKB/TrEMBL. In total, 47 992 323 of the predictions are new (not predicted by UniRule or SAAS).Association rule mining and selection techniques can be used effectively as computational methods for functional prediction. ARBA (Association-Rule-Based Annotator) is based on these techniques and can be used to enhance the quality of automatically generated annotations as well as annotating proteins with unknown functions. It learns on data from UniProtKB/Swiss-Prot and uses InterPro signatures and organism taxonomy as attributes to predict most of the protein functional annotations including GO terms, metabolic pathways, EC numbers, etc. With respect to certain quality measures, ARBA finds all rules which would define significant relationships between attributes and functional annotations in UniProtKB/Swiss-Prot entries. The set of extracted rules represent the comprehensive knowledge which could explain protein functions. However, these rules comprise redundant information and their high number makes it infeasible to apply them on large sets of data such as UniProtKB/TrEMBL. To address this issue, ARBA puts these rules into a fast competition process called SkyRule based on two concepts, namely dominance and comparability. Rules are then elegantly and considerably reduced in number and aggregated to form concise prediction models that assign functional annotations to UniProtKB entries.In order to share our knowledge in computational annotation and our rule-based systems, we are working on developing standard formats for rule annotation and tools to apply them. UniFire (the UNIprot Functional annotation Inference Rule Engine) is a standalone tool to apply the UniProt annotation rules on third party data. This engine is based on a Business Rules Management System (BRMS) named Drools and is developed in conjunction with an exchangeable format for UniProt rules and the protein data to be annotated. we termed it URML (Uniprot Rule Markup Language). URML respects the definition of business rules, and is therefore executable. When applied on third party proteins, UniFire will transform the data according to the URML format and execute the UniProt rules on them to generate functional predictions. We would like to work with the scientific community in this development and encourage users to register their interest in the links provide in our blog Inside UniProt (https://insideuniprot.blogspot.com/2018/03/\n).\nA new change in UniRef to provide GO annotation was based on user feedback. The UniRef databases (UniProt Reference Clusters) provide clustered sets of sequences from the UniProt Knowledgebase and selected UniParc records to obtain complete coverage of sequence space at several resolutions while hiding redundant sequences. Our study indicated functional annotations are generally preserved in UniRef clusters due to the intra-cluster homogeneity (\n2626\n). The GO terms of member protein sequences are highly consistent at UniRef90/50 levels in each of the three GO domains: Molecular Function, Biological Process and Cellular Component. We have started to compute GO annotations last year for UniRef90 and UniRef50 clusters: A GO term is assigned to a cluster when it is found in every UniProtKB member that has GO annotation in this cluster, or when it is a common ancestor of at least one GO term from each such member. We evaluated the GO term prediction by comparing the protein sequences that newly acquired GO annotation with the ones that GO terms were assigned by UniRef in previous releases. The overall rate of prediction precision is higher than 97%, in all three GO domains, indicating the confidence to propagate UniRef GO annotation to member sequences. Thus, UniRef provides an annotation source to our users. They can search against UniRef and then obtain GO annotation from their results.GO annotations of UniRef can be found in the UniRef XML files from the UniProt ftp site:\nftp://ftp.uniprot.org/pub/databases/uniprot/uniref/uniref50/\n.\nFor the release 2018_06, 25 300 GO terms were assigned to 33 814 200 UniRef90 clusters and 21 000 GO terms were assigned to 11 235 770 UniRef50 Clusters.\nUniProt bibliography\nUniProt compiles additional bibliography from external sources to complement the curated literature set in UniProtKB/Swiss-Prot with additional publications and to add relevant literature to UniProtKB/TrEMBL entries not yet curated. The sources of literature are of two types: biological databases (currently 22 sources, including model organism, structure, post-translational modification, function, disease and interaction databases) where the literature mapped to UniProt entries are added in a collaborative manner, and secondly, text mining (currently PubTator_Tmvar for literature about human variants in disease (\n27\n) and pGenN (\n2827\n) and pGenN (\n28\n) for literature related to plant proteins). Altogether these sources add 902 956 unique papers covering 363 591 entries, with a total of 36 147 401 UniProt accession-PubMed identifier (accession/PMID) pairs (Release 2018_08).\nIn the entry publication section, publications providing evidence for a specific annotation type are organized across different categories, such as function, interaction and expression, based on the type of data they contain. This categorization facilitates navigation through the protein entry literature by offering users quick access to literature of interest related to a given protein-topic. The classification of the UniProt references is automatically done based on the topics found in the flat file RP lines that are linked to the references, whereas the computationally mapped bibliography has to be categorized in a different way.As a first approach, we rely on the information provided by the underlying database sources for this categorization (e.g. the literature provided by iPTMnet (\n29\n) will fall into PTM/Processing category and literature provided by Intact will be classified as Interaction). However, this approach is not optimal for all literature, thus we are now systematically classifying the articles into the UniProt main categories present in the entry using UPCLASS, a neural network based pipeline to classify publications for UniProtKB protein entries, developed in SIB Text Mining group (\nhttps://www.sib.swiss/ruch-patrick/patrick-ruch-subhttps://www.sib.swiss/ruch-patrick/patrick-ruch-sub\n). UPCLASS has been trained and evaluated with the curated literature and categories in UniProt entries (with precision, recall and F-score of 0.80, 0.62 and 0.70, respectively). The model used for UniProt classification considers the protein names corresponding to the accession-PMID pair, resulting in a more accurate protein-centric classification. Implementation of UPCLASS has enabled the classification of 32 196 740 UniProt accession-PMID pairs that have been previously displayed as unclassified, representing 254 038 protein entries and 768 505 PMIDs (Release 2018_08).\nOutreach and trainingOutreach and training\nThe training of our users to make the best use of our data, whether accessed via the web, the API or downloaded from the FTP site, is a key mission of the UniProt Consortium. We have played an active role in the provision of hands-on training workshops across the globe, supporting both early-career researchers and domain specialists such as clinicians and proteomic scientists. Whilst face-to-face training will remain an important route for disseminating very detailed information, it is not always the most efficient method of reaching out to large numbers of users or of penetrating new user groups. We have therefore scheduled a series of webinars, with basic modules on searching protein sequence and function repeated regularly and more specialist units, for example on protein structure, machine learning protein function and linking protein and genome annotation delivered less often. The webinars are recorded, and are subsequently made available online (https://www.ebi.ac.uk/training/online/\n) where they are supported by related online training materials and YouTube videos (\nhttps://www.youtube.com/user/uniprotvideos/\n). In order to reach new communities, the webinars are widely advertised, making pro-active use of social media forums such as Twitter (@uniprot) and FaceBook (\nhttps://www.facebook.com/uniprot.org/\n) as well as established mailing lists.\nWebsite developments\nDuring the last year we have added three new visualizations to the UniProt website. Firstly, we have added a method for viewing molecular interactions, secondly a method for viewing the subcellular localization of the proteins and finally we have added a molecular structure viewer. Together these enable users to rapidly understand the molecular context of UniProt entries.For UniProtKB entries that include an Interaction section, we show details of the protein's binary interactions with other proteins, using a high-quality dataset supplied by the IMEx Consortium. The binary interactions of a protein are now shown as a matrix that shows the interaction partners of your protein and also shows which of those partners interact with each other. For example, the interaction matrix for the human E3 ubiquitin–protein ligase parkin protein (UniProtKB:\nO60260\n) is shown in Figure\n5\n.\nFigure 5.\nOpen in a new tab\nInteraction matrix of the human Parkin protein.\nOne of the sections on the protein entry pages is Subcellular Location. This section provides information on the location and the topology of the mature protein in the cell. We now allow users to visually explore the subcellular location in UniProtKB entries. The visualization presents image templates from COMPARTMENTS (\nhttps://compartments.jensenlab.org/\n) (\n28\n,\n30https://compartments.jensenlab.org/\n) (\n28\n,\n30\n) combined with protein location data from UniProt (expert annotation, rule-based automatic annotation) and imported from GO annotation. Figure\n6\nbelow shows the subcellular location view from the Human Copper-transporting ATPase 2 protein (UniProtKB:\nP35670\n).\nFigure 6.\nOpen in a new tab\nThe subcellular localization view of a UniProt entry (UniProtKB\nP35670\n).\nStructural information is important in understanding the molecular mechanisms that allow proteins to perform their specific functions. UniProt now provides a protein structure viewer in the ‘Structure’ section of the entry view of the website as well as in the ProtVista protein viewer (see Figure\n7\n). The structures are rendered using the Litemol viewer. This innovation helps users to connect protein information in UniProt with structural data.\nFigure 7.\nOpen in a new tabFigure 7.\nOpen in a new tab\nThe molecular structure of the Spike protein of the Human SARS coronavirus (PDB ID: 1WNC) structure as shown in the ProtVista protein viewer. The 3D viewer is interactively connected with the sequence level annotations in UniProt e.g. domains, PTMs and mutations. Note that the user can select from any of the structures that map to the protein entry.",
         "There were 86,951 complete proteomes of viral origin.",
         null,
         null,
         null
        ],
        [
         "47",
         "What were the intra-assay and interassay CVs for cortisol as measured by solid-phase RIA in the Leiden studies?",
         "Ferdinand Roelfsema et al.,2017, DOI:https://doi.org/10.1530/EC-17-0160:\n Subjects and methods\nClinical protocol\nThe 109 volunteers from Leiden, The Netherlands, had participated as controls in studies on neurological disorders, obesity, longevity and ACTH–cortisol relationships (\n3\n,\n4\n,\n9\n,\n21\n). In addition, 34 healthy male participants of a not yet published sleep interference study were included from Rochester, MN, USA. Informed consent was obtained from each subject after full explanation of the purpose and nature of all procedures used as approved by the Mayo Clinic Internal Review Board and Leiden University Ethics Committee. Clinical characteristics are listed in\nTable 1Table 1\n. Postmenopausal individuals studied here did not use estrogen therapy. Premenopausal women were studied in the early follicular phase of the menstrual cycle. Participants maintained conventional work and sleeping patterns and reported no recent transmeridian travel, weight change (>2 kg in 6 week), shift work, psychosocial stress, prescription medication use, including contraceptive pills in women, substance abuse, neuropsychiatric illness or acute or chronic systemic disease. A complete medical history, physical examination and screening biochemistry tests were normal. No subject had been exposed to glucocorticoids within the preceding 3 months. Volunteers were admitted to the Study Unit in the evening or stayed overnight in a nearby hotel (\n33\n). Ambulation was permitted to the lavatory only. Vigorous exercise, daytime sleep, snacks, caffeinated beverages and cigarette smoking were disallowed. Meals were provided at 09:00, 12:30 and 17:30 h, and room lights were turned off between 22:00 and 24:00 h. Blood samples (2.0 mL) were withdrawn at 10-min intervals for 24 h. Total blood loss was less than 430 mL. Volunteers were compensated for the time spent in the study as approved by the IRB.\nTable 1.\nDemographic properties of the volunteers.\nMen\n(79)\nWomen\n(64)\nP\nvalue\nAge (year)\n47.2 ± 1.9\n45.4 ± 1.7\n0.93\nBMI (kg/m\n2\n)\n26.8 ± 0.6\n27.9 ± 0.7\n0.28\nIGF-1 (nmol/L)\n19.9 ± 0.7\n19.4 ± 0.8\n0.62\nFree T4 (nmol/L)\n15.8 ± 0.4\n15.0 ± 0.2\n0.09\nOpen in a new tab\nData are shown as mean and\ns.e.m.\nStatistical comparisons were done with the 2-tailed Student’s\nt\n-test.\nAssayst\n-test.\nAssays\nSerum cortisol concentrations were measured with different assays. In the majority of the Leiden studies, cortisol was assayed by solid-phase RIA (Sorin Biomedical, Milan, Italy). Intra-assay and interassay CVs were 5.1 and 6.4%, respectively. Cortisol in the Leiden Longevity Study was measured by the electrochemiluminescence immunoassay on the Modular E-170 immunoanalyzer (Roche Diagnostics). The detection limit of the first generation of the assay was 5.5 nmol/L, and intra- and interassay CVs were 3.5–6.3% and 3.1–5.1%, respectively. The Mayo study used Immulite 2000 (Siemens Healthcare Diagnostics, Flanders, NJ, USA). This chemiluminescent assay has a detection range of 5.5–1380 nmol/L with intra- and interassay coefficients of variation of 7.2–9.4% and 6.3–9.4% at respective concentrations of 10.5–122 and 10.2–113 nmol/L. All samples of each individual were measured the same assay. No samples were undetectable in any assay.Average, peak and nadir GH cortisol concentrations\nMean (24-h and 3-h averages), median, absolute peak (single daily maximum) and absolute nadir (single daily minimum) cortisol concentrations were determined in each subject.\nDeconvolution analysisDeconvolution analysis\nCortisol concentration time series were analyzed using a recently developed automated deconvolution method. Pulse detection was empirically validated using hypothalamo pituitary sampling and simulated pulsatile time series. Sensitivity and specificity both exceed 93%. The Matlab-based algorithm first detrends the data and normalizes concentrations to the unit interval (0, 1). Second, the program creates multiple successively decremental potential pulse-onset time sets, each containing one fewer pulse by a smoothing process (a nonlinear adaptation of the heat-diffusion equation). Third, a maximum-likelihood expectation (MLE) estimation method calculates all secretion and elimination parameters simultaneously conditional on each of the candidate pulse-time sets. Deconvolution parameters comprise basal secretion, secretory-burst mass (concentration units), mode (time delay in min to maximum value) and frequency (number of bursts per sampling duration,\nlambdalambda\nof Weibull distribution). Pulsatile secretion is the sum of secretory-burst mass and total secretion of the sum of basal and pulsatile secretion. The fast half-life of cortisol was represented as 2.41 min, constituting 37% of the decay amplitude and the slow half-life ranging between 40 and 70 min. Statistical model selection was performed using the Akaike information criterion (\n22\n,\n23\n).\nApproximate entropy\nAdaptive ensemble (network-like) control of hormone release patterns was appraised via the approximate entropy statistic (ApEn) (\n24\n). ApEn provides a model-free and scale-invariant statistical estimate of relative randomness in the data, wherein larger values denote greater process randomness and less feedback control (\n25\n,\n26\n).\nDiurnal variations25\n,\n26\n).\nDiurnal variations\nDiurnal variations of cortisol levels were studied in two ways: first, by relating mean concentrations in successive 3-h windows to age, BMI and sex; and secondly, by Cosinor analysis, which quantitates the nyctohemeral variation of cortisol concentrations by using a nonlinear unweighted least-squares cosine regression. Ninety-five percent statistical confidence intervals were determined for the 24-h cosineamplitude (50% of the zenith-nadir difference), mesor (rhythmic mean) and acrophase (clock time of maximal value).\nStatistical analysis\nThe primary endpoint was the 24-h mean cortisol concentration with segmentation into 3-h bins across the 24 h. Secondary mechanistic outcomes were (1) deconvolution, (2) ApEn and (3) Cosinor parameters.\nThe\na priori\nclinical hypothesis was that sex, age and/or BMI determine time-dependent cortisol concentrations. Comparisons between sexes were done with the Student’s\ntt\n-test for unpaired data. Data were analyzed using a General Linear Model (GLM) for repeat measurements with categorical variables, sex and assay method. The effect of age was analyzed by linear regression. Calculations were performed using Systat 13 (Systat Software, Inc, San Jose, CA, USA) and Matlab 8.6 (The MathWorks, Inc, Natick, MA, USA). Cosinor analysis was carried out with Sigmaplot 13 (Systat Software, Inc).\nP\n< 0.05 was construed as significant for the overall study.\nPost hoc\nmultiple comparisons used Sidak’s correction.",
         "Intra-assay and interassay CVs were 5.1 and 6.4%, respectively.",
         null,
         null,
         null
        ],
        [
         "48",
         "How many viral complete proteomes were in UniProt release 2018_07?",
         "The UniProt Consortium et al.,2018, DOI:https://doi.org/10.1093/nar/gky1049:\n PROGRESS AND DEVELOPMENTS\nGrowth of sequences in UniProt\nThe UniProtKB Proteomes portal (\nhttps://www.uniprot.org/proteomes/\n) provides access to proteomes for over 84 thousand (84 387, release 2018_07) species with completely sequenced genomes. The majority of these proteomes are based on the translation of genome sequence submissions to the INSDC source databases—ENA, GenBank and the DDBJ (\n2\n). To ensure comprehensiveness, complementary pipelines have been developed to supplement these with genomes sequenced and/or annotated by groups such as Ensembl (\n3\n), Vectorbase (\n4\n) and WormBase ParaSite (\n53\n), Vectorbase (\n4\n) and WormBase ParaSite (\n5\n). This has been extended to RefSeq genomes, allowing us to import key genomes of special interest annotated by NCBI’s Eukaryotic Genome Annotation Pipeline. Release 2018_03, for example, saw the inclusion of a set of primate proteomes and a dozen genomes for marine mammals were imported from RefSeq for release 2018_08. Taxa may be specifically targeted by curators to fill gaps in the taxonomic space and additional proteome import can be requested by individual users.\nThe continuing growth of sequenced genomes is a challenge for databases as much of this growth is being driven by sequencing of very similar and almost identical strains (90% of proteins have >90% identity) of the same species. We continue to see exponential growth in many of our datasets, see Figure\n11\n. We are managing this growth by a number of processes. Most critically, a redundancy removal process was first introduced in 2015. This process identifies and removes almost identical proteomes of the same species before their inclusion in UniProtKB (\nhttps://www.uniprot.org/help/proteome_redundancy\n) and places their sequences in UniParc. Currently this process has removed ∼38% all complete proteomes (∼241 million proteins) from UniProtKB. As can be seen in Figure\n1\n, the redundancy reduction both greatly reduced the size of UniProtKB as well as made its growth more scalable. This approach has now been extended from prokaryotes to fungi, resulting in the deprecation of ∼1 million fungal protein records in release 2016_08.\nFigure 1.\nOpen in a new tab\nGrowth of UniProt sequences over the last decade.\nReference proteomesFigure 1.\nOpen in a new tab\nGrowth of UniProt sequences over the last decade.\nReference proteomes\nFor the remaining proteomes we provide a Reference Proteome set (∼9% of total proteomes) selected by the research community and supplemented with selected proteomes from a computational clustering (\n6\n) to provide the best annotated proteome in their cluster. Recently, we have added virus Reference Proteomes (described below) to this list. The growth of Reference Proteome sets is shown in Figure\n2\n. Programmatic access to the non-redundant proteomes included in UniProtKB is provided via the Proteins API (\nhttps://www.ebi.ac.uk/proteins/api/doc/\n) (\n7\n) whilst all proteomes reference, non-reference and redundant in UniProtKB and UniParc are accessible via the Proteome section of the UniProt website (\nhttps://www.uniprot.org/proteomes/\n).\nFigure 2.\nOpen in a new tab\nGrowth of the total number of Complete Proteomes and Reference Proteomes since 2015.Growth of the total number of Complete Proteomes and Reference Proteomes since 2015.\nUniProt reference proteomes are used by user communities such as the Quest for Orthologs (QfO) group as the ‘gold standard’ reference proteome dataset for orthologue benchmarking (\n8\n). The QfO reference proteomes datasets are a compiled subset of the UniProt reference proteomes, comprising well-annotated model organisms and organisms of interest for biomedical research and phylogeny, with the intention to provide broad coverage of the tree of life whilst maintaining a low number of proteomes for the benchmark. Such datasets have been generated annually from the UniProt Knowledgebase (UniProtKB) database for the past seven years. To this end, a gene-centric pipeline has been developed and enhanced over the past year by UniProt. These complete, non-redundant reference proteomes are publicly available at\nftp://ftp.ebi.ac.uk/pub/databases/reference_proteomes/QfOftp://ftp.ebi.ac.uk/pub/databases/reference_proteomes/QfO\n. The datasets are provided either in SeqXML format or as a collection of FASTA files and include genomics coordinates for most proteins. In the last release for QfO in April, the number of species increased from 66 to 78 comprising 1,311,679 non-redundant protein sequences (48 Eukaryotes and 30 bacteria-archaea).At the time of our previous publication, the number of virus Reference Proteomes in UniProtKB stood at 428—practically an order of magnitude less than the number of known viral species (which stands at 4405 according to the International Committee on Taxonomy of Viruses, or ICTV). To improve Reference Proteome Coverage of viruses in UniProt we have undertaken a concerted effort to curate Complete Proteomes and to use these as input for the computational selection of Reference Proteomes. We compiled and curated complete viral genome (proteome) data from INSDC (which feeds into UniProt) as well as a range of other specialized viral genome resources including ViPr (\n9\n), IRD (\n10\n), the HIV database (\nhttps://www.hiv.lanl.gov/\n), Papillomavirus Episteme (\n11\n) and the HCV database (\nhttps://hcv.lanl.gov/\n) and HBV database (\n1211\n) and the HCV database (\nhttps://hcv.lanl.gov/\n) and HBV database (\n12\n). This curation effort included the identification and removal of segmented viruses lacking one or more segments, and increased the number of the Complete Proteomes of viral origin in UniProt to 86 951 (UniProt release 2018_07). Computational clustering of this enhanced viral Complete Proteome set (\n13\n) produced 5,887 viral Reference Proteomes. Note that redundancy removal procedures are not currently applied to viruses, due to the challenges posed by the high number of variants in the small genomes of these species.Technological advances have enabled the sequencing of the genetic material from all the microorganisms in a particular environment without the cultivation of any of the community members. Concurrent advances in bioinformatics have enabled the rapid assembly of genomes from metagenomes (MAGs) and a corresponding input of sequence data into the UniProt database with >4800 assembled proteomes in the database (4839, release 2018_07). The existing data input pipeline is currently based on those submissions to the INSDC which fulfil certain threshold criteria but future plans are to move to using the EBI Metagenomics resource, MGnify (\n14\n), as the main source of metagenome derived assemblies. We will include only those MAGs that show a high level of completeness and a low level of contamination.\nExpert curation progressExpert curation progress\nExpert curation of the literature is critically important to the UniProt databases. The information extracted from scientific publications is stored in the UniProtKB/Swiss-Prot section of the UniProt Knowledgebase and describes functional information both in the form of human readable free-text/controlled syntax summaries and via structured vocabularies such as the Gene Ontology (GO) (\n15\n) or ChEBI (\n16\n). Expert curation is labour intensive, with curators assimilating and evaluating multiple lines of evidence from the text and figures of relevant publications, but this has repeatedly proven to be the most efficient method of extracting all relevant data from a paper. We have previously demonstrated that UniProtKB expert biocuration captures between 35 and 45% of the curatable literature for any given species, rising to 50% of the curatable literature for\nHomo sapiens\n(\n17Homo sapiens\n(\n17\n). UniProtKB/Swiss-Prot entries serve as a source of functional data for the development and enhancement of bioinformatics prediction tools, so we prioritize the capture of functional data that cannot currently be correctly predicted by computational methods, such as proteins which predictive protein signatures would describe as being part of an enzyme family but are in fact non-functional due to the loss of specific amino acid residues. An example of this is PLC1 (UniProtKB:\nQ15111Q15111\n), a member of the phosphoinositide phospholipase C family (InterPro:IPR001192) in which the existence of an asparagine residue in the active site instead of the conserved histidine residue suggests a non-catalytic role for this protein. This has broader applications than the annotation of just one protein as information on protein function is computationally transferred from UniProtKB/Swiss-Prot to sequence-related, but less well studied, proteins in UniProtKB/TrEMBL. Therefore, it is important to ensure we have a broad enough collection of annotated proteins in UniProtKB/Swiss-Prot to add value to new entries as the taxonomic range of fully sequenced proteomes continues to expand.Once a record has been moved into UniProtKB/Swiss-Prot from UniProtKB/TrEMBL, there is a concerted effort to ensure that it is regularly updated to ensure the current understanding of a protein’s activity reflects that in the scientific literature. This task can be difficult, as our knowledge of protein function continues to evolve and finer grained experimental techniques provide new knowledge that may appear to contradict previous observations. An example of this is provided by the proteins involved in N\n6\n-methyladenosine (m\n6\nA) modification which takes place on both coding and non-coding RNAs containing a Pu[G>A]m\n66\nA) modification which takes place on both coding and non-coding RNAs containing a Pu[G>A]m\n6\nAC[A/C/U] (Pu = purine) sequence, the nucleotide A3 of which becomes N6-methylated. The modification acts as a key regulator of mRNA stability: methylation is completed upon the release of mRNA into the nucleoplasm and affects various processes, such as mRNA stability, processing, translation efficiency and editing. The enzymes that catalyse this process were believed to have been fully characterized some years ago, but more recent data has changed our understanding of how these molecules work.\nm\n6\nA methylation is performed by a heteromeric WMM N\n6\n-adenosine-methyltransferase complex, originally thought to be a heterotrimer of METTL3 (UniProtKB:\nQ86U44\n), METTL14 (UniProtKB:\nQ9HCE5\n) and WTAP (UniProtKB:\nQ15007\n), but additional component subunits have recently been identified (\n18Q15007\n), but additional component subunits have recently been identified (\n18\n). Initial work indicated that METTL3 and METTL14 form a tight heterodimer in order to perform catalysis on the preferred motif sequence. Both proteins are members of the MT-A70 family and are classified as S-adenosyl-L-methionine-dependent methyltransferases by sequence prediction resources. Methyltransferase activity has been reported for both (\n19\n). Subsequent structural studies have now shown that that only one protein, METTL3, constitutes the catalytic subunit (\n20–22\n). The other subunit, METTL14, has a degenerate active site that is unable to accommodate donor and acceptor substrates and plays a non-catalytic role in maintaining complex integrity and substrate RNA binding (\n22\n). WTAP appears to serve as a regulatory subunit. A newly identified subunit of the WMM methyltransferase complex, VIRMA (UniProtKB:\nQ69YN4\n) plays a role in mediating mRNA m\n6Q69YN4\n) plays a role in mediating mRNA m\n6\nA methylation in 3′UTRs and near stop codons and in recruiting the methyltransferase core components.\nExpert biocuration has been essential in keeping the relevant UniProtKB/Swiss-Prot entries up to date with each step in the pathway to reflect our understanding of the function of these proteins and the molecules they associate with. Every piece of knowledge we capture is associated with evidences (see\nhttps://www.uniprot.org/help/evidences\nfor further details) to indicate the source of information and we have added a CAUTION comment to the METTL14 entry to both provide some background and inform users that the initially reported methyltransferase activity is an unsafe observation (Figure\n33\n). The correct annotation on the UniProtKB/Swiss-Prot entry will ensure that misleading data is not computationally added to orthologous proteins in UniProtKB/TrEMBL on the basis of an over-predictive protein signature, again highlighting the importance of expert biocuration. Future plans for the manual curation activities in UniProtKB include the development of mechanisms to identify and highlight contradictory information in existing protein entries in order to improve rigor and reproducibility. This example also illustrates the collaborative nature of the UniProt Consortium in that the molecular interactions involved have also been added to the IMEx Consortium (\nwww.imexconsortium.org\n) dataset by UniProt curators (\n23\n), the protein complexes involved curated into the Complex Portal (\nhttps://www.ebi.ac.uk/complexportal/\n) (\n24\n) and both the proteins and complexes have been further annotated using GO (\nhttp://www.geneontology.orghttp://www.geneontology.org\n). New terms were added to the GO to enable this and others were updated. All of these data are subsequently reimported back to enrich the appropriate entries in UniProtKB/Swiss-Prot but are stored and maintained by these domain-specific resources.\nFigure 3.\nOpen in a new tab\nFunctional annotation describing human METTL14 (UniProtKB\nQ9HCE5\n).\nAutomatic annotation progressUniProt's automatic annotation pipelines enrich the unreviewed records in UniProtKB/TrEMBL with classification and functional annotations. InterPro is used to classify sequences at superfamily, family and subfamily levels and to predict the occurrence of functional domains and important sites. InterPro integrates predictive models of protein function, so-called ‘signatures’, from a number of member databases. In UniProtKB/TrEMBL entries, domains from the InterPro member databases PROSITE, SMART or Pfam are predicted and used to automatically provide domain annotations. All automatic annotations are labelled with their evidence/source. UniProt has developed two complementary rule-based prediction systems, UniRule and the Statistical Automatic Annotation System (SAAS) to automatically annotate UniProtKB/TrEMBL in an efficient and scalable manner with a high degree of accuracy. These prediction systems can annotate protein properties such as protein names, function, catalytic activity, pathway membership and subcellular location, along with sequence-specific information, such as the positions of post-translational modifications and active sites. We continue to increase the number of Rules used for annotation and this has now grown to over 6000 in total as shown in Figure4\n.\nFigure 4.\nOpen in a new tab\nGrowth of curated automatic annotation rules within the UniRule system.\nOngoing developments in automatic annotation\nWe are evaluating new methods for computational annotation of function based on based specifically on their domain architecture (DAAC, or Domain Architecture Alignment and Classification) (\n22\n,\n25\n) or any combination of sequence properties and features (ARBA, or Association-Rule-Based Annotator).\nThe Domain Architecture Alignment and Classification system (DAAC, (\n25)) is based on the assumption that not only the presence of domains is important for the protein to perform its function but also their architecture i.e. their arrangement and order. It performs the prediction in four major steps. Firstly, it generates the different ordered combinations of InterPro domains, these combinations are termed domain architectures (DA). Secondly, it calculates the pairwise similarity between DAs using alignment. The third step is that of training. In this phase, clusters of proteins are created, where each cluster represents a single annotation. A threshold of belonging is determined for each cluster while maximising the classification performance, and then annotations corresponding to low performance clusters are discarded. In the final step, the clusters along with their corresponding thresholds serve as models to annotate other query proteins. The fact that DAAC takes into account the arrangement of domains within proteins allows it to make more accurate predictions for specific cases where proteins can have the same set of domains but with different arrangements and hence could perform different functions. Based on UniProt release 2018_05, DAAC generates 7 095 966 EC predictions. 3 880 775 of them are new predictions (not predicted by UniRule or SAAS). Furthermore, DAAC generates 82 285 267 predictions of GO Terms for 20 827 747 distinct proteins in UniProtKB/TrEMBL. In total, 47 992 323 of the predictions are new (not predicted by UniRule or SAAS).Association rule mining and selection techniques can be used effectively as computational methods for functional prediction. ARBA (Association-Rule-Based Annotator) is based on these techniques and can be used to enhance the quality of automatically generated annotations as well as annotating proteins with unknown functions. It learns on data from UniProtKB/Swiss-Prot and uses InterPro signatures and organism taxonomy as attributes to predict most of the protein functional annotations including GO terms, metabolic pathways, EC numbers, etc. With respect to certain quality measures, ARBA finds all rules which would define significant relationships between attributes and functional annotations in UniProtKB/Swiss-Prot entries. The set of extracted rules represent the comprehensive knowledge which could explain protein functions. However, these rules comprise redundant information and their high number makes it infeasible to apply them on large sets of data such as UniProtKB/TrEMBL. To address this issue, ARBA puts these rules into a fast competition process called SkyRule based on two concepts, namely dominance and comparability. Rules are then elegantly and considerably reduced in number and aggregated to form concise prediction models that assign functional annotations to UniProtKB entries.In order to share our knowledge in computational annotation and our rule-based systems, we are working on developing standard formats for rule annotation and tools to apply them. UniFire (the UNIprot Functional annotation Inference Rule Engine) is a standalone tool to apply the UniProt annotation rules on third party data. This engine is based on a Business Rules Management System (BRMS) named Drools and is developed in conjunction with an exchangeable format for UniProt rules and the protein data to be annotated. we termed it URML (Uniprot Rule Markup Language). URML respects the definition of business rules, and is therefore executable. When applied on third party proteins, UniFire will transform the data according to the URML format and execute the UniProt rules on them to generate functional predictions. We would like to work with the scientific community in this development and encourage users to register their interest in the links provide in our blog Inside UniProt (https://insideuniprot.blogspot.com/2018/03/\n).\nA new change in UniRef to provide GO annotation was based on user feedback. The UniRef databases (UniProt Reference Clusters) provide clustered sets of sequences from the UniProt Knowledgebase and selected UniParc records to obtain complete coverage of sequence space at several resolutions while hiding redundant sequences. Our study indicated functional annotations are generally preserved in UniRef clusters due to the intra-cluster homogeneity (\n2626\n). The GO terms of member protein sequences are highly consistent at UniRef90/50 levels in each of the three GO domains: Molecular Function, Biological Process and Cellular Component. We have started to compute GO annotations last year for UniRef90 and UniRef50 clusters: A GO term is assigned to a cluster when it is found in every UniProtKB member that has GO annotation in this cluster, or when it is a common ancestor of at least one GO term from each such member. We evaluated the GO term prediction by comparing the protein sequences that newly acquired GO annotation with the ones that GO terms were assigned by UniRef in previous releases. The overall rate of prediction precision is higher than 97%, in all three GO domains, indicating the confidence to propagate UniRef GO annotation to member sequences. Thus, UniRef provides an annotation source to our users. They can search against UniRef and then obtain GO annotation from their results.GO annotations of UniRef can be found in the UniRef XML files from the UniProt ftp site:\nftp://ftp.uniprot.org/pub/databases/uniprot/uniref/uniref50/\n.\nFor the release 2018_06, 25 300 GO terms were assigned to 33 814 200 UniRef90 clusters and 21 000 GO terms were assigned to 11 235 770 UniRef50 Clusters.\nUniProt bibliography\nUniProt compiles additional bibliography from external sources to complement the curated literature set in UniProtKB/Swiss-Prot with additional publications and to add relevant literature to UniProtKB/TrEMBL entries not yet curated. The sources of literature are of two types: biological databases (currently 22 sources, including model organism, structure, post-translational modification, function, disease and interaction databases) where the literature mapped to UniProt entries are added in a collaborative manner, and secondly, text mining (currently PubTator_Tmvar for literature about human variants in disease (\n27\n) and pGenN (\n2827\n) and pGenN (\n28\n) for literature related to plant proteins). Altogether these sources add 902 956 unique papers covering 363 591 entries, with a total of 36 147 401 UniProt accession-PubMed identifier (accession/PMID) pairs (Release 2018_08).\nIn the entry publication section, publications providing evidence for a specific annotation type are organized across different categories, such as function, interaction and expression, based on the type of data they contain. This categorization facilitates navigation through the protein entry literature by offering users quick access to literature of interest related to a given protein-topic. The classification of the UniProt references is automatically done based on the topics found in the flat file RP lines that are linked to the references, whereas the computationally mapped bibliography has to be categorized in a different way.As a first approach, we rely on the information provided by the underlying database sources for this categorization (e.g. the literature provided by iPTMnet (\n29\n) will fall into PTM/Processing category and literature provided by Intact will be classified as Interaction). However, this approach is not optimal for all literature, thus we are now systematically classifying the articles into the UniProt main categories present in the entry using UPCLASS, a neural network based pipeline to classify publications for UniProtKB protein entries, developed in SIB Text Mining group (\nhttps://www.sib.swiss/ruch-patrick/patrick-ruch-subhttps://www.sib.swiss/ruch-patrick/patrick-ruch-sub\n). UPCLASS has been trained and evaluated with the curated literature and categories in UniProt entries (with precision, recall and F-score of 0.80, 0.62 and 0.70, respectively). The model used for UniProt classification considers the protein names corresponding to the accession-PMID pair, resulting in a more accurate protein-centric classification. Implementation of UPCLASS has enabled the classification of 32 196 740 UniProt accession-PMID pairs that have been previously displayed as unclassified, representing 254 038 protein entries and 768 505 PMIDs (Release 2018_08).\nOutreach and trainingOutreach and training\nThe training of our users to make the best use of our data, whether accessed via the web, the API or downloaded from the FTP site, is a key mission of the UniProt Consortium. We have played an active role in the provision of hands-on training workshops across the globe, supporting both early-career researchers and domain specialists such as clinicians and proteomic scientists. Whilst face-to-face training will remain an important route for disseminating very detailed information, it is not always the most efficient method of reaching out to large numbers of users or of penetrating new user groups. We have therefore scheduled a series of webinars, with basic modules on searching protein sequence and function repeated regularly and more specialist units, for example on protein structure, machine learning protein function and linking protein and genome annotation delivered less often. The webinars are recorded, and are subsequently made available online (https://www.ebi.ac.uk/training/online/\n) where they are supported by related online training materials and YouTube videos (\nhttps://www.youtube.com/user/uniprotvideos/\n). In order to reach new communities, the webinars are widely advertised, making pro-active use of social media forums such as Twitter (@uniprot) and FaceBook (\nhttps://www.facebook.com/uniprot.org/\n) as well as established mailing lists.\nWebsite developments\nDuring the last year we have added three new visualizations to the UniProt website. Firstly, we have added a method for viewing molecular interactions, secondly a method for viewing the subcellular localization of the proteins and finally we have added a molecular structure viewer. Together these enable users to rapidly understand the molecular context of UniProt entries.For UniProtKB entries that include an Interaction section, we show details of the protein's binary interactions with other proteins, using a high-quality dataset supplied by the IMEx Consortium. The binary interactions of a protein are now shown as a matrix that shows the interaction partners of your protein and also shows which of those partners interact with each other. For example, the interaction matrix for the human E3 ubiquitin–protein ligase parkin protein (UniProtKB:\nO60260\n) is shown in Figure\n5\n.\nFigure 5.\nOpen in a new tab\nInteraction matrix of the human Parkin protein.\nOne of the sections on the protein entry pages is Subcellular Location. This section provides information on the location and the topology of the mature protein in the cell. We now allow users to visually explore the subcellular location in UniProtKB entries. The visualization presents image templates from COMPARTMENTS (\nhttps://compartments.jensenlab.org/\n) (\n28\n,\n30https://compartments.jensenlab.org/\n) (\n28\n,\n30\n) combined with protein location data from UniProt (expert annotation, rule-based automatic annotation) and imported from GO annotation. Figure\n6\nbelow shows the subcellular location view from the Human Copper-transporting ATPase 2 protein (UniProtKB:\nP35670\n).\nFigure 6.\nOpen in a new tab\nThe subcellular localization view of a UniProt entry (UniProtKB\nP35670\n).\nStructural information is important in understanding the molecular mechanisms that allow proteins to perform their specific functions. UniProt now provides a protein structure viewer in the ‘Structure’ section of the entry view of the website as well as in the ProtVista protein viewer (see Figure\n7\n). The structures are rendered using the Litemol viewer. This innovation helps users to connect protein information in UniProt with structural data.\nFigure 7.\nOpen in a new tabFigure 7.\nOpen in a new tab\nThe molecular structure of the Spike protein of the Human SARS coronavirus (PDB ID: 1WNC) structure as shown in the ProtVista protein viewer. The 3D viewer is interactively connected with the sequence level annotations in UniProt e.g. domains, PTMs and mutations. Note that the user can select from any of the structures that map to the protein entry.",
         "There were 86,951 complete proteomes of viral origin.",
         null,
         null,
         null
        ],
        [
         "49",
         "What are the main children of the new top level term GO:0140110 transcription regulator activity?",
         "The Gene Ontology Consortium et al.,2018, DOI:https://doi.org/10.1093/nar/gky1055:\n GO resource content\nThe GO knowledgebase consists of the ontology and the annotations made using the ontology. As of the 5 September 2018 release (doi:10.5281/zenodo.1410625), there were ∼45 000 terms in GO: 29 698 biological processes, 11 147 molecular functions and 4201 cellular components, linked by almost 134 000 relationships. The number of annotations (as well as the percentage change since 2016 (\n1\n)) are shown in Table\n1\n. It is important to understand that the changes reflect two distinct processes: addition of annotations based on new evidence, and obsoleting of annotations that have been superseded by newer studies. We expect the number of obsoleted annotations to increase, due to our increasing annotation quality assurance efforts, described in more detail below.\nNew framework and repository, for gene function ‘models’New framework and repository, for gene function ‘models’\nWe have developed a more expressive computational framework for representing gene functions, which subsumes our current GO annotation framework, while maintaining compatibility. We refer to the framework as GO-\nC\nausal\nA\nctivity\nM\nodeling (GO-CAM, formerly referred to as ‘LEGO’ (\n1\n)) and to GO-CAMs as\nmodels\nto distinguish them from\nstandard annotations\n. A paper detailing GO-CAM is in preparation, but we summarize some properties here. In GO-CAM, each model is represented as a set of triples (\nsubject\n-\nrelation\n-object, with brackets {} as a set container), e.g. {\nABCA1\nenables\ncholesterol transporter activity\n;\ncholesterol transporter activity\noccurs in\nplasma membrane\n, and\ncholesterol transporter activity\npart_ofoccurs in\nplasma membrane\n, and\ncholesterol transporter activity\npart_of\ncholesterol homeostasis}. Each triple is supported by one or more pieces of evidence, consisting of a class from ECO and a citable source, usually a scientific publication. GO-CAM specifies the semantics of GO annotations, and how standard GO annotations can be combined into a larger model. Each GO-CAM model is represented using the Web Ontology Language (OWL), which is converted computationally to standard GO annotations (GAF format), ensuring backward compatibility. Users can browse, view, and download the available models in different formats at:\nhttp://geneontology.org/go-camhttp://geneontology.org/go-cam\n. The number of models is currently small, and most models contain only one gene product (standard annotations extended with additional contextual information, such as cell type). The GO is currently rapidly increasing the curation of GO-CAM models, and the model repository is growing. In particular, models are now available that each represent an entire regulatory or metabolic pathway.\nChanges to data access: new production pipelineChanges to data access: new production pipeline\nStarting in March 2018, the releases of the GO resource have been generated by a new data production pipeline using a refreshed software stack. This system allows for easily extensible error checking and improved reporting of quality assurance checks that ensure the quality and integrity of the released ontology and annotations. For users, one of the most important aspects of this pipeline is that the GO resource now produces monthly releases (named by release date) that are available at the GO site and can be referenced and obtained as stable Document Object Identifiers (DOIs) via Zenodo. This feature is critical for ensuring that GO-based analyses can be replicated in a consistent and referenceable manner through the inclusion of these DOIs and/or version of both the ontology and annotation files used (\n66\n). Our data production pipeline is currently hosted at the Lawrence Berkeley National Laboratory. We provide GO annotations in multiple formats: as standard GAF (Gene Association Format) and GPAD (Gene Product Association Data) annotation files, in Turtle (OWL serialization format) [\nhttp://current.geneontology.org/products/ttl\n], and a Blazegraph [\nhttp://current.geneontology.org/products/blazegraph\n] journal, replacing the legacy MySQL output.\nInteractions with the GO user community\nGO is an open project, and we encourage community contributions to the knowledgebase and software.\nAll users\n: GO can be contacted using the GO Helpdesk (\nhttp://help.geneontology.orgAll users\n: GO can be contacted using the GO Helpdesk (\nhttp://help.geneontology.org\n) for any questions or feedback about the annotations, the ontology, software, or other GO resources. If users notice an annotation that may not be correct, they should first review the original publication or data source. If the annotation still seems inaccurate, users are encouraged to report this to the GO helpdesk, and GOC members will review the annotation and remove or modify it if justified.\nAuthors\n: Authors can now see if their paper has been used for GO annotation directly in PubMed. Under the ‘LinkOut - more resources’ section of the PubMed abstract page, papers with annotations will have a link labeled ‘Gene Ontology annotations from this paper - Gene Ontology’ (see, e.g.\nhttps://www.ncbi.nlm.nih.gov/pubmed/3357510https://www.ncbi.nlm.nih.gov/pubmed/3357510\nwhich links to a web page on the GO site that shows the annotations based on evidence in that paper). If no GO LinkOut is present, that may indicate that the publication has not been used for GO annotation. Authors can contact the helpdesk at the GO website to suggest new annotations or changes to existing annotations.\nResources or consortia\n: The GOC collaborates with established data resources and other groups and consortia representing a particular area of biology. Recent examples include cilium biology (\n7\n,\n8\n), autophagy (\n9\n) and cardiac phenotypes (\n10\n). We encourage members of other interest groups to contact us to improve the ontology and annotations in their areas of expertise.\nTracking all contributions:\nMost aspects of the GO project management are now based in GitHub (\nhttps://github.com/geneontology\n). In addition to tracking ontology change requests (\nhttps://github.com/geneontology/go-ontology). In addition to tracking ontology change requests (\nhttps://github.com/geneontology/go-ontology\n), we now use GitHub to collect feedback on annotations (\nhttps://github.com/geneontology/go-annotation\n). For users familiar with GitHub, we encourage them to submit any requests directly to GitHub, where they can follow all further discussion and actions. Otherwise, issues and queries can be submitted to our helpdesk (\nhttp://help.geneontology.org\n).\nIncreased focus on annotation quality controlThe GO resource is now 20 years old. The longevity of the resource adds a challenge to maintain and update the many existing annotations, as many of the findings published during that time have become much more precise, or were reinterpreted or superseded. We have made it a high priority to identify and correct inaccurate and out-of-date legacy annotations to make sure that GO continues to consistently reflect current knowledge. We have taken a number of different approaches to tackle this challenge. First, to ensure consistency and quality, GO biocurators meet regularly for training, establishment of annotation guidelines, and coordinated review of specific areas of biology. More recently, we have made significant efforts to integrate annotation review with ontology improvements, taking advantage of suggested changes to the ontology to clarify term definitions, intended usage, and coordinate annotation practices with curators. In addition, quality assurance is performed centrally, both computationally, to ensure annotations are valid, and manually, to ensure they accurately represent the experimental findings.We have discovered that one of the most powerful approaches to quality control and consistency is the phylogenetic approach. Originally developed as a means of propagating annotations from experimentally studied genes to evolutionarily related genes in other species, the phylogenetic perspective provides a unified view of all experimental annotations within a evolutionarily-related protein family, allowing curators to more easily find outlier annotations (see e.g. (\n1111\n)). In parallel, the development of GO-CAM models has been useful in identifying inconsistent annotation practices, and has provided opportunities to develop consortium-wide annotation guidelines. Another observation that emerged is that older annotations from isolated phenotypic observations, taken outside of other contextual data, often do not provide evidence of direct involvement of a gene in a biological process. If inconsistencies are noticed, they are reported to the contributing group for verification and correction as appropriate.In a pilot quality assurance effort, we have requested the review (by GO Consortium biocurators) of ∼2500 manual annotations (<0.01% of the total corpus) that were judged questionable by one of the strategies above. Approximately 70–80% of the annotations flagged for review were modified to a more appropriate term or removed. We will continue to work on improving the quality of the annotations and reviewing legacy data when appropriate. As a result, we expect that the increase in annotations and new ontology terms may not be as rapid as in the past, at least for the main species annotated by the consortium members, as a greater proportion of our efforts will be dedicated to reviewing and revising older annotations.\nOntology revision and integrationOntology revision and integration\nSince our last update article, we have developed an entirely new process for ontology editing and maintenance that has dramatically increased efficiency and enabled extensive real-time quality checks. Ontology editing is now performed in an OWL-based environment using the ontology editing tool Protégé (\n12\n). The ontology is versioned and tracked using a GitHub repository (\nhttps://github.com/geneontology/go-ontology\n). One major advantage of the new ontology management process is that the work can be parallelized among multiple editors, thus increasing efficiency. In addition, real-time quality checks prevent errors that would otherwise require revisiting the same editing task more than once to correct them.GO continues to integrate and align with external ontology resources in two main ways: import of sub-ontologies used to define GO terms, and inclusion of external cross-references. GO utilizes the structure of external ontologies to aid in reasoning and in the automatic inference of relations between GO terms (\n13\n). GO imports subsets of these external ontologies that include information about anatomical structures, cell types, chemicals and taxonomic groupings: Uberon (\n14\n), Protein Ontology (\n15\n), Plant Ontology (\n16\n), ChEBI (\n17\n), Relations Ontology (\n18\n), NCBI Taxonomy (\n19\n), Sequence Ontology (\n20\n), Ontology of Biological Attributes (\nhttp://www.obofoundry.org/ontology/oba.html\n), Fungal Anatomy Ontology (\nhttp://www.obofoundry.org/ontology/fao.html\n), Phenotypic Quality Ontology (\nhttp://obofoundry.org/ontology/pato.html\n), and Common Anatomy Reference Ontology (\nhttp://www.obofoundry.org/ontology/caro.html), and Common Anatomy Reference Ontology (\nhttp://www.obofoundry.org/ontology/caro.html\n). GO also maintains cross-references between terms and multiple widely-used external resources, including Reactome (\n21\n), The Annotated Reactions Database (Rhea) (\n22\n), Enzyme Commission (EC;\nhttp://www.sbcs.qmul.ac.uk/iubmb/enzyme/\n), IntAct, Complex Portal (\n23\n) and MetaCyc (\n24\n).\nRefactoring the molecular function branch of GO23\n) and MetaCyc (\n24\n).\nRefactoring the molecular function branch of GO\nPreviously, there was a trend in GO molecular function ontology development to focus on adding terms that describe molecular binding activities of specific gene products. The advantage of such terms is that annotations can often be made unequivocally based on results from a single experiment. However, this approach has led to a complex ontology structure and a proliferation of annotations that individually represent only a partial functional description of a gene product. Annotations to binding terms can obscure annotations to more informative function terms, making annotations more difficult to interpret. For example, one can annotate\nCDK1\n(UniProtKB:\nP06493CDK1\n(UniProtKB:\nP06493\n) separately with ‘GO:0030332 cyclin binding’, ‘GO:0005524 ATP binding’, ‘GO:0005515 protein binding’, and ‘GO:0004674 protein serine/threonine kinase activity’. However, these are all aspects of a more precise molecular function that is more informative than the sum of these parts: ‘GO:0004693 cyclin-dependent protein serine/threonine kinase activity’. In the GO molecular function refactoring, we recognized that, while specific binding events are an essential mechanism by which gene products function, an individual binding activity is almost never sufficient in itself to describe molecular function in a larger biological context (\n2525\n). One of the primary goals of our refactoring was to ensure that the ontology contains the terms necessary to describe these higher-level functions, and that they have a path to the root of the ontology that is not simply under the generic ‘GO:0005488 binding’ term. Accordingly, we reinstated some previously obsoleted terms and added new terms, as well as additional relations. We also addressed the structure of the ontology so that the upper-level terms would be more biologically meaningful and have more uniform specificity. We removed 8 terms from the top level and added four new terms (Figure\n2\n). Most of the terms that were formerly direct children of ‘GO:0003674 molecular function’ were moved under more biologically meaningful terms: for example, ‘GO:0042056 chemoattractant activity’ and ‘GO:0045499 chemorepellent activity’ were moved under ‘GO:0048018 receptor ligand activity’, while ‘GO:0036370\ndd\n-alanyl carrier activity’ and ‘GO:0016530 metallochaperone activity’ were moved under the new term ‘GO:0140104 molecular carrier activity’ (representing an activity of ‘directly binding to a specific ion or molecule and delivering it either to an acceptor molecule or to a specific location’). We have also created a new term ‘GO:0104005 hijacked molecular function’ as a parent of terms such as ‘GO:0001618 virus receptor activity’, which is, from the standpoint of the protein being annotated, not a normal function, but nevertheless relevant for some of our users.\nFigure 2.\nOpen in a new tab\nThe\nmolecular function\nbranch, before and after refactoring. The term marked with an ‘x’ in the left-hand panel has been obsoleted. Terms moved (assigned to a new parent) are indicated by arrows. New terms (right panel) are marked with ‘NEW’.\n1\nThe class label ‘electron carrier activity’ was changed to ‘electron transfer activity’.1\nThe class label ‘electron carrier activity’ was changed to ‘electron transfer activity’.\nFinally, we have made significant changes to the structure representing the molecular functions of transcription factors (Figure\n3\n). This refactoring was carried out in collaboration with experts in transcription factors and gene regulation from the Gene Regulation Consortium (GRECO;\nhttp://thegreco.orghttp://thegreco.org\n). In keeping with our design principle of having terms that describe higher-level functions, we have created a new parent term to group all functions that directly regulate transcription, ‘GO:0140110 transcription regulator activity’. The formerly top-level term ‘GO:0000988 transcription factor activity, protein binding’ has been obsoleted because this activity was partly covered by other terms in the ontology and its usage was inconsistent. Accordingly, its children have either been obsoleted or subsumed under different terms (merged or moved). The new top level term ‘GO:0140110 transcription regulator activity’ has three main children - ‘GO:0003700 DNA-binding transcription factor activity’ (formerly labeled ‘transcription factor activity, sequence-specific DNA binding), ‘GO:0140223 general transcription initiation factor activity’ and ‘GO:0003712 transcription coregulator activity’.\nFigure 3.\nOpen in a new tabFigure 3.\nOpen in a new tab\nCurrent structure of the ‘GO:0140110 transcription regulator activity’ branch of the ontology.\nThe transcription factor areas of GO had previously been refactored between 2010 and 2012 (\n26\n,\n27\n) with the aim of more finely capturing all combinations of different types of protein and DNA binding activities (e.g. binding to different types of regulatory regions such as promoters and enhancers) and transcription regulation processes (positive and negative regulation). However, this structure, while very precise, has proved very difficult to use by biocurators, resulting in inconsistent annotations. Additionally, end-users have had difficulty with common queries, such as comprehensively identifying the set of all transcription factors in a given species. We expect that even more improvements to the ontology structure, as well as more consistent annotations to transcription regulator terms, will be available in 2019.Defining the boundaries of biological processes: MAP kinase signaling and extracellular matrix as examplesWe have used our integrated annotation review and ontology development methodology to refine two areas of the ontology, the MAP kinase signaling pathway and the representation of the extracellular matrix. Refinements to the MAPK cascade included defining the molecular functions that are parts of the process: ‘GO:0004707 MAP kinase activity’, ‘GO:0004708 MAP kinase kinase activity, ‘GO:0004709 MAP kinase kinase kinase activity’ and ‘GO:0008349 MAP kinase kinase kinase kinase activity’. All other upstream and downstream molecular functions/biological processes will be modeled in GO-CAM with causal relationships between them and the MAPK cascade. We also enumerated the types of cascades based on current literature and on discussions among expert model-organism curators, trying to keep the distinctions useful across multiple taxa. There are four direct subtypes of ‘GO:0000165 MAPK cascade’: ‘GO:0070371 ERK1 and ERK2 cascade’, ‘GO:0070375 ERK5 cascade’, ‘GO:0071507 pheromone response MAPK cascade’ and ‘GO:0051403 stress-activated MAPK cascade’. Some other MAPK processes such as ‘GO:1903616 MAPK cascade involved in axon regeneration’ will eventually be obsoleted, as these combine two or more other GO terms and can be composed in GO-CAM. For the refinement of the extracellular matrix area of the ontology, we worked with external experts to add terms that were useful grouping terms such as ‘GO:0062023 collagen-containing extracellular matrix’. We also obsoleted or merged terms that were poorly annotated and thought to represent an outdated view.GO subsets (slims)\nA GO subset (or slim) is a set of GO terms selected to provide an overview of the functions, locations or roles of a set of genes. The subset can be developed for high coverage of specific species, or to represent only certain areas of the ontology, and in most cases, contain only high-level GO terms to provide a broad biological overview. Another use of subsets is to blacklist certain terms for annotation: GO has two such subsets, one to flag terms that should not be used for manual annotation, and one for terms that should not be used at all. GO maintains two additional subsets, the Generic GO slim and the Alliance of Genome Resources (\nhttps://www.alliancegenome.org/\n) slim. GO also hosts subsets useful to groups using GO; we currently have 11 such subsets (Table\n2\n;\nhttp://www.geneontology.org/page/go-subset-guide2\n;\nhttp://www.geneontology.org/page/go-subset-guide\n). Each subset provides a global overview of gene functions. Each subset now has a designated contact person to resolve any issue resulting from ontology changes (see Ontology revision and integration).\nOther developments\nThe GO ribbon: a configurable tool for visualizing GO annotations\nMany genes have large numbers of annotations, making it difficult to get a quick overview of a gene function, or the functions of gene sets. We have developed the GO ribbon specifically to help users visualize and explore the functions of a gene. The GO ribbon visualization metaphor borrows from a viewer originally developed by the Mouse Genome Database team (\n28\n), but in contrast, the GO ribbon was developed as a lightweight, reusable widget that can be embedded in any website, and retrieves data directly from the GO resource via API.To generate a GO ribbon, all the functions (GO terms) associated with a gene of interest are mapped onto a specified GO subset using the ontology structure. The end result is a simple graphical representation of a gene's functions (Figure\n4\n). The ribbon is interactive, allowing users to drill down to more specific functions by selecting a high-level category such as ‘GO:0030154 cell differentiation’, ‘GO:0050877 nervous system process’, or ‘GO:0003700 DNA-binding transcription factor activity’, and to filter the functions based on the evidence codes provided in the GO annotations. This overview of gene functions is particularly useful when comparing the functions of different genes in the same species, or the functions of orthologous genes across different species.\nFigure 4.\nOpen in a new tab\nGO ribbon representation. Darker boxes indicate terms with the most annotations; white boxes represent terms that are not annotated for this protein (\nMus musculusMus musculus\nSox7, MGI:98369). Screenshot obtained from\nhttps://www.alliancegenome.org/gene/MGI:98369\n.\nThe GO ribbon is a React component available on GitHub (\nhttps://github.com/geneontology/ribbon\n) and as a NPM package\nhttps://www.npmjs.com/package/@geneontology/ribbon\n). The GO ribbon widget is currently used by the Alliance of Genome Resources.\nGO annotations from high-throughput experiments\nData from high-throughput experiments are generally collected in a hypothesis-free manner, and consequently do not generally provide as strong evidence of gene function as small-scale molecular biology experiments that currently support most of the experimental GO annotations. In addition, high-throughput experiments can be subject to relatively high false positive rates. Users may therefore want to filter out these experimental annotations in some applications of the GO. To make this possible, starting in 2018, in collaboration with the Evidence and Conclusions Ontology (\n29\n) (\n2), the GO has added several new evidence codes to describe high-throughput experiments: ‘ECO:0006056 high throughput evidence used in manual assertion’ (HTP), and the subclasses: ‘ECO:0007005 high throughput direct assay evidence used in manual assertion’ (HDA), ‘ECO:0007001 high throughput mutant phenotype evidence used in manual assertion’ (HMP), ‘ECO:0007003 high throughput genetic interaction evidence used in manual assertion’ (HGI) and ‘ECO:0007007 high throughput expression pattern evidence used in manual assertion’ (HEP). To accompany the new evidence codes, we have provided annotation guidelines to help identify and curate high-throughput datasets that meet the GO Consortium annotation criteria. Consortium members have reviewed papers with more than 40 annotations using a single evidence code, and updated the evidence codes, or removed the annotations if appropriate. There are currently over 31 000 annotations that have HTP evidence codes from 140 research articles, representing <5% of experimental GO annotations. The identification of annotations derived from high-throughput experiments allows users to choose to exclude these from their analyses, if they are concerned that these annotations may lead to an increased bias in data analysis. This is likely to be particularly important, as is often the case, when GO is used to interpret types of data similar to those on which the annotations are based.",
         "The main children are GO:0003700 DNA-binding transcription factor activity, GO:0140223 general transcription initiation factor activity and GO:0003712 transcription coregulator activity.",
         null,
         null,
         null
        ]
       ],
       "shape": {
        "columns": 6,
        "rows": 500
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>question</th>\n",
       "      <th>context</th>\n",
       "      <th>answer</th>\n",
       "      <th>groundedness_score</th>\n",
       "      <th>relevance_score</th>\n",
       "      <th>standalone_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>What do disease-associated alleles commonly al...</td>\n",
       "      <td>Feixiong Cheng et al.,2021, DOI:https://doi.or...</td>\n",
       "      <td>Disease-associated alleles commonly alter dist...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>In protein networks, what is the difference in...</td>\n",
       "      <td>Haiyuan Yu et al.,2007, DOI:https://doi.org/10...</td>\n",
       "      <td>Hub-bottlenecks have low coexpression with the...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>What are the main children of the new top leve...</td>\n",
       "      <td>The Gene Ontology Consortium et al.,2018, DOI:...</td>\n",
       "      <td>The main children are GO:0003700 DNA-binding t...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>How did ApEn vary with sex and age in the study?</td>\n",
       "      <td>Ferdinand Roelfsema et al.,2017, DOI:https://d...</td>\n",
       "      <td>ApEn was higher in women than men, higher in t...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>What average implantation rate is achieved wit...</td>\n",
       "      <td>Rehannah Borup et al.,2016, DOI:https://doi.or...</td>\n",
       "      <td>An average of around 25% implantation rate is ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>What overall accuracy was achieved using the l...</td>\n",
       "      <td>Rehannah Borup et al.,2016, DOI:https://doi.or...</td>\n",
       "      <td>The LOOCV accuracy was 0.81 and 0.85, respecti...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>What percentage of patients in the whole-exome...</td>\n",
       "      <td>Yaping Yang et al.,2014, DOI:https://doi.org/1...</td>\n",
       "      <td>4.6% of patients had medically actionable inci...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>What effect does the decrease of γδ T cells in...</td>\n",
       "      <td>Xue Feng et al.,2022, DOI:https://doi.org/10.1...</td>\n",
       "      <td>It may damage the establishment of the local i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>498</th>\n",
       "      <td>What were the mean uterine dimensions observed...</td>\n",
       "      <td>Hong Gao et al.,2018, DOI:https://doi.org/10.1...</td>\n",
       "      <td>The mean uterine size was 49.33 mm × 39.94 mm ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>What percentage of embryonic aneuploidies are ...</td>\n",
       "      <td>Danilo Cimadomo et al.,2018, DOI:https://doi.o...</td>\n",
       "      <td>More than 90% of embryonic aneuploidies are of...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>500 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              question  \\\n",
       "0    What do disease-associated alleles commonly al...   \n",
       "1    In protein networks, what is the difference in...   \n",
       "2    What are the main children of the new top leve...   \n",
       "3     How did ApEn vary with sex and age in the study?   \n",
       "4    What average implantation rate is achieved wit...   \n",
       "..                                                 ...   \n",
       "495  What overall accuracy was achieved using the l...   \n",
       "496  What percentage of patients in the whole-exome...   \n",
       "497  What effect does the decrease of γδ T cells in...   \n",
       "498  What were the mean uterine dimensions observed...   \n",
       "499  What percentage of embryonic aneuploidies are ...   \n",
       "\n",
       "                                               context  \\\n",
       "0    Feixiong Cheng et al.,2021, DOI:https://doi.or...   \n",
       "1    Haiyuan Yu et al.,2007, DOI:https://doi.org/10...   \n",
       "2    The Gene Ontology Consortium et al.,2018, DOI:...   \n",
       "3    Ferdinand Roelfsema et al.,2017, DOI:https://d...   \n",
       "4    Rehannah Borup et al.,2016, DOI:https://doi.or...   \n",
       "..                                                 ...   \n",
       "495  Rehannah Borup et al.,2016, DOI:https://doi.or...   \n",
       "496  Yaping Yang et al.,2014, DOI:https://doi.org/1...   \n",
       "497  Xue Feng et al.,2022, DOI:https://doi.org/10.1...   \n",
       "498  Hong Gao et al.,2018, DOI:https://doi.org/10.1...   \n",
       "499  Danilo Cimadomo et al.,2018, DOI:https://doi.o...   \n",
       "\n",
       "                                                answer  groundedness_score  \\\n",
       "0    Disease-associated alleles commonly alter dist...                 NaN   \n",
       "1    Hub-bottlenecks have low coexpression with the...                 5.0   \n",
       "2    The main children are GO:0003700 DNA-binding t...                 NaN   \n",
       "3    ApEn was higher in women than men, higher in t...                 5.0   \n",
       "4    An average of around 25% implantation rate is ...                 NaN   \n",
       "..                                                 ...                 ...   \n",
       "495  The LOOCV accuracy was 0.81 and 0.85, respecti...                 NaN   \n",
       "496  4.6% of patients had medically actionable inci...                 NaN   \n",
       "497  It may damage the establishment of the local i...                 NaN   \n",
       "498  The mean uterine size was 49.33 mm × 39.94 mm ...                 NaN   \n",
       "499  More than 90% of embryonic aneuploidies are of...                 NaN   \n",
       "\n",
       "     relevance_score  standalone_score  \n",
       "0                NaN               NaN  \n",
       "1                4.0               5.0  \n",
       "2                NaN               NaN  \n",
       "3                4.0               1.0  \n",
       "4                NaN               NaN  \n",
       "..               ...               ...  \n",
       "495              NaN               NaN  \n",
       "496              NaN               NaN  \n",
       "497              NaN               NaN  \n",
       "498              NaN               NaN  \n",
       "499              NaN               NaN  \n",
       "\n",
       "[500 rows x 6 columns]"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "generated_questions = pd.DataFrame.from_dict(examples)\n",
    "generated_questions.loc[:,[\"question\",\"context\",\"answer\",\"groundedness_score\",\"relevance_score\",\"standalone_score\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "02a98086",
   "metadata": {},
   "outputs": [],
   "source": [
    "generated_questions_final = generated_questions.loc[\n",
    "    (generated_questions[\"groundedness_score\"] >= 4)\n",
    "    & (generated_questions[\"relevance_score\"] >= 4)\n",
    "    & (generated_questions[\"standalone_score\"] >= 4)\n",
    "]\n",
    "with open(\"generated_questions_final.pkl\",\"wb\") as f:\n",
    "    pkl.dump(generated_questions_final, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "7651e27f",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"generated_questions_final.pkl\",\"rb\") as f:\n",
    "    generated_questions_final = pkl.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "15684535",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>question</th>\n",
       "      <th>answer</th>\n",
       "      <th>groundedness_score</th>\n",
       "      <th>relevance_score</th>\n",
       "      <th>standalone_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>In protein networks, what is the difference in...</td>\n",
       "      <td>Hub-bottlenecks have low coexpression with the...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>What endometrial changes are associated with t...</td>\n",
       "      <td>The beginning of the implantation window is ch...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>What kit was used for total RNA sequencing?</td>\n",
       "      <td>The Smarter Stranded Total RNA-Seq kit v2 was ...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>What effect does chrysin have on the expressio...</td>\n",
       "      <td>Chrysin significantly and concentration-depend...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>What databases were used for functional annota...</td>\n",
       "      <td>Gene Ontology, Gene cards, Kyoto Enciclopedy o...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>469</th>\n",
       "      <td>What effect does the FGB rs1800790G &gt; A minor ...</td>\n",
       "      <td>It results in an up to 8-fold higher increase ...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476</th>\n",
       "      <td>What immunity processes are related to up-regu...</td>\n",
       "      <td>The up-regulated genes in the thin endometrium...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481</th>\n",
       "      <td>What bacterial phyla are more abundant in the ...</td>\n",
       "      <td>The uterine cavity demonstrates higher amounts...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>484</th>\n",
       "      <td>What effect does VEGF treatment have on the mi...</td>\n",
       "      <td>VEGF treatment significantly increases the mig...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485</th>\n",
       "      <td>What are some of the human diseases that can r...</td>\n",
       "      <td>Beckwith-Wiedemann syndrome, Silver-Russell sy...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>87 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              question  \\\n",
       "1    In protein networks, what is the difference in...   \n",
       "18   What endometrial changes are associated with t...   \n",
       "25         What kit was used for total RNA sequencing?   \n",
       "28   What effect does chrysin have on the expressio...   \n",
       "29   What databases were used for functional annota...   \n",
       "..                                                 ...   \n",
       "469  What effect does the FGB rs1800790G > A minor ...   \n",
       "476  What immunity processes are related to up-regu...   \n",
       "481  What bacterial phyla are more abundant in the ...   \n",
       "484  What effect does VEGF treatment have on the mi...   \n",
       "485  What are some of the human diseases that can r...   \n",
       "\n",
       "                                                answer  groundedness_score  \\\n",
       "1    Hub-bottlenecks have low coexpression with the...                 5.0   \n",
       "18   The beginning of the implantation window is ch...                 5.0   \n",
       "25   The Smarter Stranded Total RNA-Seq kit v2 was ...                 5.0   \n",
       "28   Chrysin significantly and concentration-depend...                 5.0   \n",
       "29   Gene Ontology, Gene cards, Kyoto Enciclopedy o...                 5.0   \n",
       "..                                                 ...                 ...   \n",
       "469  It results in an up to 8-fold higher increase ...                 4.0   \n",
       "476  The up-regulated genes in the thin endometrium...                 5.0   \n",
       "481  The uterine cavity demonstrates higher amounts...                 5.0   \n",
       "484  VEGF treatment significantly increases the mig...                 5.0   \n",
       "485  Beckwith-Wiedemann syndrome, Silver-Russell sy...                 5.0   \n",
       "\n",
       "     relevance_score  standalone_score  \n",
       "1                4.0               5.0  \n",
       "18               5.0               5.0  \n",
       "25               5.0               5.0  \n",
       "28               4.0               5.0  \n",
       "29               5.0               5.0  \n",
       "..               ...               ...  \n",
       "469              5.0               5.0  \n",
       "476              5.0               5.0  \n",
       "481              5.0               5.0  \n",
       "484              4.0               5.0  \n",
       "485              5.0               5.0  \n",
       "\n",
       "[87 rows x 5 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "generated_questions_final.loc[:,[\"question\",\"answer\",\"groundedness_score\",\"relevance_score\",\"standalone_score\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "69babbd7",
   "metadata": {},
   "outputs": [],
   "source": [
    "eval_dataset = generated_questions_final.to_dict(\"records\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "763d2a80",
   "metadata": {},
   "source": [
    "### Create a function to check RAG performance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "878aa04c",
   "metadata": {},
   "outputs": [],
   "source": [
    "llm_reader = init_chat_model(\"gemini-2.0-flash\", model_provider=\"google_genai\",temperature=0.5,max_output_tokens=1024) # gemma-3-27b-it"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e0c5ef00",
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_embeddings(\n",
    "    documents,\n",
    "    chunk_size: int,\n",
    "    embedding_model\n",
    "):\n",
    "    splitter = RecursiveCharacterTextSplitter(\n",
    "    chunk_size=chunk_size,\n",
    "    chunk_overlap=chunk_size//10,\n",
    "    separators=[\"\\n\\n\", \"\\n\", \".\", \"!\", \"?\", \" \"]  # smart splitting\n",
    "    )\n",
    "\n",
    "    info_splitted = []\n",
    "\n",
    "    for j in documents:\n",
    "\n",
    "        for key, value in j.items():\n",
    "        \n",
    "            if key in ['Abstract', 'Introduction', 'Methods', 'Results', 'Discussion', 'Conclusion',] and value:\n",
    "\n",
    "                if len(value) > 1200:\n",
    "                    chunks = splitter.split_text(value)\n",
    "\n",
    "                    for i, c in enumerate(chunks):\n",
    "\n",
    "                        info_splitted.append(\n",
    "                            {\n",
    "                                \"chunk_index\":i,\n",
    "                                \"content\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")+\", DOI:\"+j.get(\"DOI\")+\"\\n\"+c,\n",
    "                                \"parent\":key,\n",
    "                                \"split\":True,\n",
    "                                \"DOI\":j.get(\"DOI\"),\n",
    "                                \"Reference\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")\n",
    "                            }\n",
    "                        )\n",
    "                else:\n",
    "\n",
    "                    info_splitted.append(\n",
    "                            {\n",
    "                                \"chunk_index\":0,\n",
    "                                \"content\":j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")+\", DOI:\"+j.get(\"DOI\")+\"\\n\"+value,\n",
    "                                \"parent\":key,\n",
    "                                \"split\":False,\n",
    "                                \"DOI\":j.get(\"DOI\"),\n",
    "                                \"Reference\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")\n",
    "                            }\n",
    "                        )\n",
    "    \n",
    "    texts = [chunk[\"content\"] for chunk in info_splitted]\n",
    "    metadatas = [{\"parent\": chunk[\"parent\"], \"chunk_index\": chunk[\"chunk_index\"],\"DOI\": chunk[\"DOI\"], \"Reference\": chunk[\"Reference\"]} for chunk in info_splitted]\n",
    "    ids = [str(uuid.uuid1()) for _ in metadatas]\n",
    "\n",
    "    db = Chroma.from_texts(\n",
    "    texts=texts,\n",
    "    embedding=embedding_model,\n",
    "    metadatas=metadatas,\n",
    "    ids=ids,\n",
    ")\n",
    "\n",
    "    return(db)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "064d4ff5",
   "metadata": {},
   "outputs": [],
   "source": [
    "RAG_PROMPT_TEMPLATE = \"\"\"\n",
    "<|system|>\n",
    "Using the information contained in the context,\n",
    "give a comprehensive answer to the question.\n",
    "Respond only to the question asked, response should be concise and relevant to the question.\n",
    "Provide the number of the source document when relevant.\n",
    "If the answer cannot be deduced from the context, do not give an answer.</s>\n",
    "<|user|>\n",
    "Context:\n",
    "{context}\n",
    "---\n",
    "Now here is the question you need to answer.\n",
    "\n",
    "Question: {question}\n",
    "</s>\n",
    "<|assistant|>\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "0e054c3c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def clean_doi_links(text):\n",
    "    \"\"\"\n",
    "    Replace problematic Unicode dashes (like non-breaking hyphen) with normal ASCII dashes.\n",
    "    \"\"\"\n",
    "    return re.sub(r\"[\\u2010-\\u2015\\u2212]\", \"-\", text)\n",
    "\n",
    "def retrieve_context(question, k, database):\n",
    "    results = database.similarity_search(question, k)\n",
    "    selected_index = []\n",
    "    ideal_chunks = []\n",
    "    meta_selected = []\n",
    "\n",
    "    def is_new_chunk(r, selected_index):\n",
    "        next_chunk = \"_\".join([r[\"parent\"], r[\"Reference\"], str(r[\"chunk_index\"] + 1)])\n",
    "        prev_chunk = \"_\".join([r[\"parent\"], r[\"Reference\"], str(r[\"chunk_index\"] - 1)])\n",
    "        return next_chunk not in selected_index and prev_chunk not in selected_index\n",
    "\n",
    "    for doc in results:\n",
    "        r = doc.metadata\n",
    "\n",
    "        if r[\"parent\"] not in [\"Journal\", \"DOI\"] and is_new_chunk(r, selected_index):\n",
    "            ii = \"_\".join([r[\"parent\"], r[\"Reference\"], str(r[\"chunk_index\"])])\n",
    "            selected_index.append(ii)\n",
    "\n",
    "            candidates = db.get(\n",
    "                where={\"$and\": [{\"Reference\": r[\"Reference\"]}, {\"parent\": r[\"parent\"]}]}\n",
    "            )\n",
    "\n",
    "            max_index = len(candidates[\"metadatas\"]) - 1\n",
    "\n",
    "            meta_selected.append(candidates[\"metadatas\"])\n",
    "            ideal_chunks.append(\n",
    "                [\n",
    "                    doc\n",
    "                    for doc, meta in zip(\n",
    "                        candidates[\"documents\"], candidates[\"metadatas\"]\n",
    "                    )\n",
    "                    if meta[\"chunk_index\"]\n",
    "                    in [\n",
    "                        r[\"chunk_index\"],\n",
    "                        max(r[\"chunk_index\"] - 1, 0),\n",
    "                        min(r[\"chunk_index\"] + 1, max_index),\n",
    "                    ]\n",
    "                ]\n",
    "                )\n",
    "\n",
    "        context = []\n",
    "        for text, meta in zip(ideal_chunks, meta_selected):\n",
    "            if meta:  # Only proceed if meta is not empty\n",
    "                doi = (\n",
    "                    clean_doi_links(meta[0][\"DOI\"])\n",
    "                    if \"DOI\" in meta[0]\n",
    "                    else \"DOI not available\"\n",
    "                )\n",
    "                context.append(\n",
    "                    f\"Summary:\\n\\n{''.join(text)}\\n\\n\"\n",
    "                )\n",
    "\n",
    "    return(context)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "ff0a7875",
   "metadata": {},
   "outputs": [],
   "source": [
    "def answer_with_rag(\n",
    "    question: str,\n",
    "    llm,\n",
    "    database,\n",
    "    num_docs_final: int = 7,\n",
    "    recursive_chunk = False\n",
    "):\n",
    "    \"\"\"Answer a question using RAG with the given knowledge index.\"\"\"\n",
    "    # Gather documents with retriever\n",
    "    \n",
    "    if  recursive_chunk:\n",
    "        relevant_docs = retrieve_context(question=question, database=database,k=num_docs_final)\n",
    "    else:\n",
    "        relevant_docs = database.similarity_search(query=question, k=num_docs_final)\n",
    "        relevant_docs = [doc.page_content for doc in relevant_docs]  # keep only the text\n",
    "\n",
    "\n",
    "    # Build the final prompt\n",
    "    context = \"\\nExtracted documents:\\n\"\n",
    "    context += \"\".join([f\"Document {str(i)}:::\\n\" + doc for i, doc in enumerate(relevant_docs)])\n",
    "\n",
    "    final_prompt = RAG_PROMPT_TEMPLATE.format(question=question, context=context)\n",
    "\n",
    "    # Redact an answer\n",
    "    answer = llm.invoke(final_prompt)\n",
    "\n",
    "    return answer.content, relevant_docs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "d66498c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_rag_tests(\n",
    "    eval_dataset,\n",
    "    llm,\n",
    "    database,\n",
    "    output_file,\n",
    "    verbose=False,\n",
    "    test_settings = None,  # To document the test settings used\n",
    "):\n",
    "    \"\"\"Runs RAG tests on the given dataset and saves the results to the given output file.\"\"\"\n",
    "    try:  # load previous generations if they exist\n",
    "        with open(output_file, \"r\") as f:\n",
    "            outputs = json.load(f)\n",
    "    except:\n",
    "        outputs = []\n",
    "\n",
    "    for example in tqdm(eval_dataset):\n",
    "        question = example[\"question\"]\n",
    "        if question in [output[\"question\"] for output in outputs]:\n",
    "            continue\n",
    "\n",
    "        answer, relevant_docs = answer_with_rag(question, llm, database,recursive_chunk=True)\n",
    "        if verbose:\n",
    "            print(\"=======================================================\")\n",
    "            print(f\"Question: {question}\")\n",
    "            print(f\"Answer: {answer}\")\n",
    "            print(f'True answer: {example[\"answer\"]}')\n",
    "        result = {\n",
    "            \"question\": question,\n",
    "            \"true_answer\": example[\"answer\"],\n",
    "            \"generated_answer\": answer,\n",
    "            \"retrieved_docs\": [doc for doc in relevant_docs],\n",
    "        }\n",
    "        if test_settings:\n",
    "            result[\"test_settings\"] = test_settings\n",
    "        outputs.append(result)\n",
    "\n",
    "        with open(output_file, \"w\") as f:\n",
    "            json.dump(outputs, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "cb5f5681",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ChatPromptTemplate(input_variables=['instruction', 'reference_answer', 'response'], input_types={}, partial_variables={}, messages=[SystemMessage(content='You are a fair evaluator language model.', additional_kwargs={}, response_metadata={}), HumanMessagePromptTemplate(prompt=PromptTemplate(input_variables=['instruction', 'reference_answer', 'response'], input_types={}, partial_variables={}, template='###Task Description:\\nAn instruction (might include an Input inside it), a response to evaluate, a reference answer that gets a score of 5, and a score rubric representing a evaluation criteria are given.\\n1. Write a detailed feedback that assess the quality of the response strictly based on the given score rubric, not evaluating in general.\\n2. After writing a feedback, write a score that is an integer between 1 and 5. You should refer to the score rubric.\\n3. The output format should look as follows: \"Feedback: {{write a feedback for criteria}} [RESULT] {{an integer number between 1 and 5}}\"\\n4. Please do not generate any other opening, closing, and explanations. Be sure to include [RESULT] in your output.\\n\\n###The instruction to evaluate:\\n{instruction}\\n\\n###Response to evaluate:\\n{response}\\n\\n###Reference Answer (Score 5):\\n{reference_answer}\\n\\n###Score Rubrics:\\n[Is the response correct, accurate, and factual based on the reference answer?]\\nScore 1: The response is completely incorrect, inaccurate, and/or not factual.\\nScore 2: The response is mostly incorrect, inaccurate, and/or not factual.\\nScore 3: The response is somewhat correct, accurate, and/or factual.\\nScore 4: The response is mostly correct, accurate, and factual.\\nScore 5: The response is completely correct, accurate, and factual.\\n\\n###Feedback:'), additional_kwargs={})])"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from langchain.prompts.chat import (\n",
    "    ChatPromptTemplate,\n",
    "    HumanMessagePromptTemplate,\n",
    ")\n",
    "from langchain.schema import SystemMessage\n",
    "\n",
    "EVALUATION_PROMPT = \"\"\"###Task Description:\n",
    "An instruction (might include an Input inside it), a response to evaluate, a reference answer that gets a score of 5, and a score rubric representing a evaluation criteria are given.\n",
    "1. Write a detailed feedback that assess the quality of the response strictly based on the given score rubric, not evaluating in general.\n",
    "2. After writing a feedback, write a score that is an integer between 1 and 5. You should refer to the score rubric.\n",
    "3. The output format should look as follows: \\\"Feedback: {{write a feedback for criteria}} [RESULT] {{an integer number between 1 and 5}}\\\"\n",
    "4. Please do not generate any other opening, closing, and explanations. Be sure to include [RESULT] in your output.\n",
    "\n",
    "###The instruction to evaluate:\n",
    "{instruction}\n",
    "\n",
    "###Response to evaluate:\n",
    "{response}\n",
    "\n",
    "###Reference Answer (Score 5):\n",
    "{reference_answer}\n",
    "\n",
    "###Score Rubrics:\n",
    "[Is the response correct, accurate, and factual based on the reference answer?]\n",
    "Score 1: The response is completely incorrect, inaccurate, and/or not factual.\n",
    "Score 2: The response is mostly incorrect, inaccurate, and/or not factual.\n",
    "Score 3: The response is somewhat correct, accurate, and/or factual.\n",
    "Score 4: The response is mostly correct, accurate, and factual.\n",
    "Score 5: The response is completely correct, accurate, and factual.\n",
    "\n",
    "###Feedback:\"\"\"\n",
    "\n",
    "evaluation_prompt_template = ChatPromptTemplate.from_messages(\n",
    "    [\n",
    "        SystemMessage(content=\"You are a fair evaluator language model.\"),\n",
    "        HumanMessagePromptTemplate.from_template(EVALUATION_PROMPT),\n",
    "    ]\n",
    ")\n",
    "\n",
    "evaluation_prompt_template"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "19ca2c79",
   "metadata": {},
   "outputs": [],
   "source": [
    "def evaluate_answers(\n",
    "    answer_path: str,\n",
    "    eval_chat_model,\n",
    "    evaluator_name: str,\n",
    "    evaluation_prompt_template: ChatPromptTemplate,\n",
    ") -> None:\n",
    "    \"\"\"Evaluates generated answers. Modifies the given answer file in place for better checkpointing.\"\"\"\n",
    "    answers = []\n",
    "    if os.path.isfile(answer_path):  # load previous generations if they exist\n",
    "        answers = json.load(open(answer_path, \"r\"))\n",
    "\n",
    "    for experiment in tqdm(answers):\n",
    "        if f\"eval_score_{evaluator_name}\" in experiment:\n",
    "            continue\n",
    "\n",
    "        eval_prompt = evaluation_prompt_template.format_messages(\n",
    "            instruction=experiment[\"question\"],\n",
    "            response=experiment[\"generated_answer\"],\n",
    "            reference_answer=experiment[\"true_answer\"],\n",
    "        )\n",
    "        eval_result = eval_chat_model.invoke(eval_prompt)\n",
    "        feedback, score = [item.strip() for item in eval_result.content.split(\"[RESULT]\")]\n",
    "        experiment[f\"eval_score_{evaluator_name}\"] = score\n",
    "        experiment[f\"eval_feedback_{evaluator_name}\"] = feedback\n",
    "\n",
    "        with open(answer_path, \"w\") as f:\n",
    "            json.dump(answers, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "e159aff1",
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.exists(\"./output\"):\n",
    "    os.mkdir(\"./output\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "b312c9c8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation for chunk:1000_reader-model:gemini-2.0-flash:\n",
      "Loading knowledge base embeddings...\n",
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 87/87 [00:57<00:00,  1.52it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 70/70 [00:52<00:00,  1.34it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk:1500_reader-model:gemini-2.0-flash:\n",
      "Loading knowledge base embeddings...\n",
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 87/87 [00:53<00:00,  1.63it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 70/70 [00:55<00:00,  1.27it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk:1700_reader-model:gemini-2.0-flash:\n",
      "Loading knowledge base embeddings...\n",
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 87/87 [00:52<00:00,  1.65it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 70/70 [00:52<00:00,  1.33it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk:1800_reader-model:gemini-2.0-flash:\n",
      "Loading knowledge base embeddings...\n",
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 87/87 [00:54<00:00,  1.59it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 70/70 [00:56<00:00,  1.24it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk:1900_reader-model:gemini-2.0-flash:\n",
      "Loading knowledge base embeddings...\n",
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 87/87 [00:53<00:00,  1.63it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 70/70 [00:54<00:00,  1.28it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk:2000_reader-model:gemini-2.0-flash:\n",
      "Loading knowledge base embeddings...\n",
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 87/87 [00:52<00:00,  1.65it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 70/70 [00:53<00:00,  1.30it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "for chunk_size in [1000,1500,1700,1800,1900,2000]:  # Add other chunk sizes (in tokens) as needed\n",
    "        \n",
    "    settings_name = f\"chunk:{chunk_size}_reader-model:gemini-2.0-flash\"\n",
    "    output_file_name = f\"./output/rag_{settings_name}.json\"\n",
    "\n",
    "    print(f\"Running evaluation for {settings_name}:\")\n",
    "\n",
    "    print(\"Loading knowledge base embeddings...\")\n",
    "\n",
    "    db = load_embeddings(\n",
    "        info_articles_final,\n",
    "        chunk_size=chunk_size,\n",
    "        embedding_model=embedding_function2,\n",
    "    )\n",
    "\n",
    "    print(\"Running RAG...\")\n",
    "    run_rag_tests(\n",
    "        eval_dataset=eval_dataset,\n",
    "        llm=llm_reader,\n",
    "        database=db,\n",
    "        output_file=output_file_name,\n",
    "        verbose=False,\n",
    "        test_settings=settings_name,\n",
    "    )\n",
    "\n",
    "    print(\"Running evaluation...\")\n",
    "    evaluate_answers(\n",
    "        output_file_name,\n",
    "        llm_reader,\n",
    "        \"gemini-2.0-flash\",\n",
    "        evaluation_prompt_template,\n",
    "    )\n",
    "    print(\"Removing database\")\n",
    "    db.delete_collection()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "8f6b57c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob\n",
    "\n",
    "outputs = []\n",
    "for file in glob.glob(\"./output/*.json\"):\n",
    "    output = pd.DataFrame(json.load(open(file, \"r\")))\n",
    "    output[\"settings\"] = file\n",
    "    outputs.append(output)\n",
    "result = pd.concat(outputs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "23117f2b",
   "metadata": {},
   "outputs": [],
   "source": [
    "result[\"eval_score_gemini-2.0-flash\"] = result[\"eval_score_gemini-2.0-flash\"].apply(lambda x: int(x) if isinstance(x, str) else 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "6d5ebaa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "result[\"eval_score_gemini-2.0-flash\"] = result[\"eval_score_gemini-2.0-flash\"]/5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "5e8551da",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "settings\n",
       "./output/rag_chunk:2000_reader-model:gemini-2.0-flash.json    0.831429\n",
       "./output/rag_chunk:1900_reader-model:gemini-2.0-flash.json    0.834286\n",
       "./output/rag_chunk:1800_reader-model:gemini-2.0-flash.json    0.837143\n",
       "./output/rag_chunk:1500_reader-model:gemini-2.0-flash.json    0.840000\n",
       "./output/rag_chunk:1700_reader-model:gemini-2.0-flash.json    0.842857\n",
       "./output/rag_chunk:1000_reader-model:gemini-2.0-flash.json    0.854286\n",
       "Name: eval_score_gemini-2.0-flash, dtype: float64"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "average_scores = result.groupby(\"settings\")[\"eval_score_gemini-2.0-flash\"].mean()\n",
    "average_scores.sort_values()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "RAG",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
