{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2fcec673",
   "metadata": {},
   "source": [
    "# Chroma Database Generation"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "932c4587",
   "metadata": {},
   "source": [
    "# Library"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2df9b888",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/aparraga/anaconda3/envs/RAG/lib/python3.9/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "import langchain_community\n",
    "import langchain_text_splitters\n",
    "from langchain_community.document_loaders import PyPDFLoader, pdf\n",
    "from langchain_community.embeddings.sentence_transformer import (\n",
    "    SentenceTransformerEmbeddings,\n",
    ")\n",
    "from langchain_community.vectorstores import Chroma\n",
    "from langchain_text_splitters import CharacterTextSplitter, RecursiveCharacterTextSplitter\n",
    "import os\n",
    "import pprint\n",
    "import re\n",
    "from langchain_core.documents import Document\n",
    "from langchain_core.prompts.prompt import PromptTemplate\n",
    "from langchain_core.output_parsers import JsonOutputParser\n",
    "from langchain_ollama import ChatOllama\n",
    "from pydantic import BaseModel, Field\n",
    "import json\n",
    "import uuid\n",
    "import chromadb\n",
    "from chromadb.config import Settings\n",
    "import unicodedata\n",
    "from langchain_google_genai import GoogleGenerativeAI\n",
    "import uuid\n",
    "# from chromadb.utils.embedding_functions import SentenceTransformerEmbeddingFunction\n",
    "import pickle as pkl\n",
    "import requests\n",
    "import subprocess\n",
    "import pandas as pd\n",
    "from bs4 import BeautifulSoup\n",
    "import tqdm\n",
    "import time\n",
    "from sentence_transformers import SentenceTransformer\n",
    "from langchain_huggingface import HuggingFaceEmbeddings\n",
    "import datetime\n",
    "import time\n",
    "import random\n",
    "from tqdm import tqdm\n",
    "from langchain.chat_models import init_chat_model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "be4ea84b",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('api_google.txt') as f:\n",
    "    \n",
    "    api_key = json.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ed9cd500",
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.environ.get(\"GOOGLE_API_KEY\"):\n",
    "  os.environ[\"GOOGLE_API_KEY\"] = api_key['key']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "22d295dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "llm = init_chat_model(\"gemini-2.5-flash\", model_provider=\"google_genai\",temperature=0,max_output_tokens=1024) # gemma-3-27b-it"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc8a4892",
   "metadata": {},
   "source": [
    "## Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "5c7558ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"info_articles_main.pkl\",\"rb\") as f:\n",
    "    info_articles_main = pkl.load(f)\n",
    "with open(\"info_articles_ref_final.pkl\",\"rb\") as f:\n",
    "    info_articles_ref = pkl.load(f)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "53b641d8",
   "metadata": {},
   "source": [
    "## Database"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "04ef787c",
   "metadata": {},
   "source": [
    "### Split the text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "e689c386",
   "metadata": {},
   "outputs": [],
   "source": [
    "splitter = RecursiveCharacterTextSplitter(\n",
    "    chunk_size=1000,\n",
    "    chunk_overlap=100,\n",
    "    separators=[\"\\n\\n\", \"\\n\", \".\", \"!\", \"?\", \" \"]  # smart splitting\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "14fa4e3c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "265"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_articles_final = info_articles_main + info_articles_ref\n",
    "len(info_articles_final)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3d2f19ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "info_splitted = []\n",
    "\n",
    "for j in info_articles_final:\n",
    "\n",
    "    for key, value in j.items():\n",
    "    \n",
    "        if key in ['Abstract', 'Introduction', 'Methods', 'Results', 'Discussion', 'Conclusion',] and value:\n",
    "\n",
    "            if len(value) > 1200:\n",
    "                chunks = splitter.split_text(value)\n",
    "\n",
    "                for i, c in enumerate(chunks):\n",
    "\n",
    "                    info_splitted.append(\n",
    "                        {\n",
    "                            \"chunk_index\":i,\n",
    "                            \"content\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")+\", DOI:\"+j.get(\"DOI\")+\":\\n \"+c,\n",
    "                            \"parent\":key,\n",
    "                            \"split\":True,\n",
    "                            \"DOI\":j.get(\"DOI\"),\n",
    "                            \"Reference\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")\n",
    "                        }\n",
    "                    )\n",
    "            else:\n",
    "\n",
    "                info_splitted.append(\n",
    "                        {\n",
    "                            \"chunk_index\":0,\n",
    "                            \"content\":j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")+\", DOI:\"+j.get(\"DOI\")+\":\\n \"+value,\n",
    "                            \"parent\":key,\n",
    "                            \"split\":False,\n",
    "                            \"DOI\":j.get(\"DOI\"),\n",
    "                            \"Reference\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")\n",
    "                        }\n",
    "                    )"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0809f73",
   "metadata": {},
   "source": [
    "## Embedding"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6065057d",
   "metadata": {},
   "source": [
    "We chose this embedding according to leaderboard of HuggingFace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4bae4ce3",
   "metadata": {},
   "outputs": [],
   "source": [
    "embedding_function2 = HuggingFaceEmbeddings(model_name=\"avsolatorio/GIST-small-Embedding-v0\") # model_kwargs={'device': 'cuda'}\n",
    "# embedding_function2 = HuggingFaceEmbeddings(model_name=\"Snowflake/snowflake-arctic-embed-s\")\n",
    "# embedding_function2 = HuggingFaceEmbeddings(model_name=\"intfloat/e5-small-v2\")  #intfloat/e5-small-v2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2786e097",
   "metadata": {},
   "source": [
    "## Chroma"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "961b72f8",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'info_splitted' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mNameError\u001b[39m                                 Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[9]\u001b[39m\u001b[32m, line 2\u001b[39m\n\u001b[32m      1\u001b[39m \u001b[38;5;66;03m# 2. Prepare documents, metadata, and IDs\u001b[39;00m\n\u001b[32m----> \u001b[39m\u001b[32m2\u001b[39m texts = [chunk[\u001b[33m\"\u001b[39m\u001b[33mcontent\u001b[39m\u001b[33m\"\u001b[39m] \u001b[38;5;28;01mfor\u001b[39;00m chunk \u001b[38;5;129;01min\u001b[39;00m \u001b[43minfo_splitted\u001b[49m]\n\u001b[32m      3\u001b[39m metadatas = [{\u001b[33m\"\u001b[39m\u001b[33mparent\u001b[39m\u001b[33m\"\u001b[39m: chunk[\u001b[33m\"\u001b[39m\u001b[33mparent\u001b[39m\u001b[33m\"\u001b[39m], \u001b[33m\"\u001b[39m\u001b[33mchunk_index\u001b[39m\u001b[33m\"\u001b[39m: chunk[\u001b[33m\"\u001b[39m\u001b[33mchunk_index\u001b[39m\u001b[33m\"\u001b[39m],\u001b[33m\"\u001b[39m\u001b[33mDOI\u001b[39m\u001b[33m\"\u001b[39m: chunk[\u001b[33m\"\u001b[39m\u001b[33mDOI\u001b[39m\u001b[33m\"\u001b[39m], \u001b[33m\"\u001b[39m\u001b[33mReference\u001b[39m\u001b[33m\"\u001b[39m: chunk[\u001b[33m\"\u001b[39m\u001b[33mReference\u001b[39m\u001b[33m\"\u001b[39m]} \u001b[38;5;28;01mfor\u001b[39;00m chunk \u001b[38;5;129;01min\u001b[39;00m info_splitted]\n\u001b[32m      4\u001b[39m ids = [\u001b[38;5;28mstr\u001b[39m(uuid.uuid1()) \u001b[38;5;28;01mfor\u001b[39;00m _ \u001b[38;5;129;01min\u001b[39;00m metadatas]\n",
      "\u001b[31mNameError\u001b[39m: name 'info_splitted' is not defined"
     ]
    }
   ],
   "source": [
    "# 2. Prepare documents, metadata, and IDs\n",
    "texts = [chunk[\"content\"] for chunk in info_splitted]\n",
    "metadatas = [{\"parent\": chunk[\"parent\"], \"chunk_index\": chunk[\"chunk_index\"],\"DOI\": chunk[\"DOI\"], \"Reference\": chunk[\"Reference\"]} for chunk in info_splitted]\n",
    "ids = [str(uuid.uuid1()) for _ in metadatas]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "1b79161a",
   "metadata": {},
   "outputs": [],
   "source": [
    "db = Chroma.from_texts(\n",
    "    texts=texts,\n",
    "    embedding=embedding_function2,\n",
    "    metadatas=metadatas,\n",
    "    ids=ids,\n",
    "    collection_name=\"ReproRAG\",\n",
    "    persist_directory=\"./chromaRepro\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "469cb121",
   "metadata": {},
   "source": [
    "We check that the search works properly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "1b92eabd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Document(metadata={'chunk_index': 10, 'Reference': 'P. Diaz-Gimeno et al.,2022', 'DOI': 'https://doi.org/10.1093/humrep/deab262', 'parent': 'Results'}, page_content='P. Diaz-Gimeno et al.,2022, DOI:https://doi.org/10.1093/humrep/deab262\\nPredictive performance comparison of signatures and consistency across endometrial datasets'),\n",
       " Document(metadata={'DOI': 'https://doi.org/10.1016/j.fertnstert.2024.03.015', 'parent': 'Abstract', 'Reference': 'Patricia Diaz-Gimeno et al.,2024', 'chunk_index': 0}, page_content='Patricia Diaz-Gimeno et al.,2024, DOI:https://doi.org/10.1016/j.fertnstert.2024.03.015\\nObjective: To propose a new gene expression signature that identifies endometrial disruptions independent of endometrial luteal phase timing and predicts if patients are at risk of endometrial failure. Design: Multicentric, prospective study. Setting: Reproductive medicine research department in a public hospital affiliated with private fertility clinics and a reproductive genetics laboratory. Patients: Caucasian women (n1⁄4 281; 39.4/C6 4.8 years old with a body mass index of 22.9/C6 3.5 kg/m 2) undergoing hormone replacement therapy between July 2018 and July 2021. Endometrial samples from 217 patients met RNA quality criteria for signature discovery and analysis. Intervention(s): Endometrial biopsies collected in the mid-secretory phase. Main Outcome Measure(s):Endometrial luteal phase timing-corrected expression of 404 genes and reproductive outcomes of thefirst single embryo transfer (SET) after biopsy collection to identify prognostic biomarkers of endometrial failure')]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db.similarity_search(\"Is there a signature to predict endometrial disruption?\", 2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "95c90458",
   "metadata": {},
   "source": [
    "## Evaluation"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aceba79f",
   "metadata": {},
   "source": [
    "### Agents for questions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4c9c2bd5",
   "metadata": {},
   "outputs": [],
   "source": [
    "def call_llm(llm, prompt):\n",
    "    response = llm.invoke(prompt)\n",
    "    return response.content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "9c74c378",
   "metadata": {},
   "outputs": [],
   "source": [
    "QA_generation_prompt = \"\"\"\n",
    "You are given a piece of scientific text (context).\n",
    "Your task is to generate ONE question and ONE answer from it.\n",
    "\n",
    "Guidelines for the question:\n",
    "- It must be factual and answerable using the context only.\n",
    "- Phrase it naturally, as if a researcher typed it into a search engine.\n",
    "- Do NOT mention \"context\", \"passage\", or \"according to the text\".\n",
    "- The question should be specific and concise.\n",
    "\n",
    "Guidelines for the answer:\n",
    "- The answer must be a short, factual statement directly supported by the context.\n",
    "- Do not add explanations, speculation, or references to the text.\n",
    "\n",
    "Formatting rules (strict):\n",
    "Output:::\n",
    "Question: <your question here>\n",
    "Answer: <your answer here>\n",
    "\n",
    "Now here is the context:\n",
    "\n",
    "{context}\n",
    "\n",
    "Output:::\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "787c51ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "info_splitted_evaluation = [d for d in info_splitted if d['parent'] in ['Abstract','Introduction','Results','Conclusion','Discussion','Methods']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "69dd44e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'chunk_index': 1, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n . The search identified experiments involving human endometrial transcriptomic case versus control raw data related to uterine pathologies and implantation alterations. The keywords employed in the search included endometriosis, endometrial adenocarcinoma (ADC), recurrent implantation failure (RIF), and recurrent pregnancy loss (RPL), among others (Supplemental Table 1A, available online, for a full list of search terms). No restrictions were placed on publication date or language. Uterine leiomyoma, adenomyosis, and uterine leiomyosarcoma data were not included due to a lack of suitable studies meeting our criteria. For each sample cohort belonging to the same individual study, 39 variables were evaluated (see Supplemental Table 1B), including clinical characteristics of the participants (e.g., age and body mass index), experimental design (e.g., endometrial biopsy preprocessing procedure and transcriptomic platform), and additional study information (e.g', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 0, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n MATERIALS AND METHODS This is a retrospective study integrating case and control data from multiple cohorts with endometrium gene expression in women with uterine disorders. The raw gene expression data and patient meta-data were downloaded from the National Center for Biotechnology Information (NCBI) functional genomics data repository Gene Expression Omnibus (GEO), where data are made available for the scientific community. Following GEO policies, in this repository patient data are anonymized and encrypted, and no additional institutional review board is required for downloading . Study Search and Selection Systematic searches and reviews of individual studies were conducted between October 2016 and June 2017 at the NCBI functional genomics data repository GEO  according to preferred reporting items for systematic reviews and metaanalysis (PRISMA) guidelines', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 1, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n . The search identified experiments involving human endometrial transcriptomic case versus control raw data related to uterine pathologies and implantation alterations. The keywords employed in the search included endometriosis, endometrial adenocarcinoma (ADC), recurrent implantation failure (RIF), and recurrent pregnancy loss (RPL), among others (Supplemental Table 1A, available online, for a full list of search terms). No restrictions were placed on publication date or language. Uterine leiomyoma, adenomyosis, and uterine leiomyosarcoma data were not included due to a lack of suitable studies meeting our criteria. For each sample cohort belonging to the same individual study, 39 variables were evaluated (see Supplemental Table 1B), including clinical characteristics of the participants (e.g., age and body mass index), experimental design (e.g., endometrial biopsy preprocessing procedure and transcriptomic platform), and additional study information (e.g', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 2, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n .g., GEO identifier and associated publication). Thefinal inclusion criteria were that at least one uterine disorder was evaluated in the sample cohort study design, RNA was extracted directly from human endometrial biopsies (samples collected from eutopic endometrium in endometriosis patients), the sample size was greater than three for both case and control groups belonging to the same study, microarray or RNA sequencing data were obtained using Affymetrix, Illumina, or Agilent gene expression platforms, and raw gene expression data were made freely available to download from GEO. Individual Transcriptomic Analysis: Making Functional Results Comparable The same transcriptomic analysis procedure was applied independently to each of the selected individual data sets for making the functional results comparable to one another before being integrated through functional meta-analysis', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 3, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n . The individual transcriptomic analysis consisted of three consecutive steps run in R software  and applied separately to compare cases versus controls within samples from the same study to avoid variability among sample cohorts related to different uterine disorders. The same preprocessing of raw gene expression data, differential expression analysis, and functional enrichment analysis was performed in a sample cohort comparing cases versus controls ( Supplemental Fig. 1A , available online). The preprocessing step was performed using the affy v.1.52.0  R package for studies using Affymetrix platforms to measure gene expression and thelimma v.3.34.5 1262 VOL. 113 NO. 6 / JUNE 2020 ORIGINAL ARTICLE: GENETICS   R package for the remaining platforms. Normalization between samples was done using quantile normalization (limma R package) . Annotation from probe set to gene symbol was implemented with the biomaRt R package v.2.30.0', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 4, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n . Annotation from probe set to gene symbol was implemented with the biomaRt R package v.2.30.0 . Next, exploratory analyses were performed to detect outliers and batch effects. The effect of the cycle phase of endometrial biopsy collection on the data was removed using linear models (limma v.3.34.5) as previously described elsewhere  because the samples were collected throughout the menstrual cycle . This correction ensures that the functional alterations obtained are due to the uterine disorder instead of the different proportions of cell types (stromal, epithelial) and/or the distinct menstrual cycle stages in which the endometrial biopsy samples were obtained . Differential expression analyses were applied using the limma R package  and functional enrichment analyses were executed using the gene set enrichment analysis (GSEA) logistic model implemented in themdgsa R package . TheP values for each function were calculated and corrected for false discovery rate (FDR)', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 5, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n . TheP values for each function were calculated and corrected for false discovery rate (FDR) . Two functional databases were consulted: the Kyoto Encyclopedia of Genes and Genomes (KEGG)  (release 81.0/2017-1-1) and Gene Ontology (GO)  (Release 1.2/2017-03-31). Functional Integration A functional meta-analysis using mdgsa  and metafor , R packages  was applied for each gene function evaluated, integrating each function among the different patient cohorts belonging to the same disorder (see Supplemental Fig. 1A). This methodology robustly highlights the common alterations across a set of studies, thus discarding those alterations that are exclusive of each individual study’s characteristics . In addition, it increases the statistical power through the integration of a higher number of samples and overcomes the limitations of individual studies’ heterogeneity regarding their distinct clinical population characteristics and experimental procedures', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 6, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n . All integrations were performed under the DerSimonian and Laird  random effects model to account for individual study heterogeneity in the R environment . This model weighs the functional results of each study according to their inherent variability measured by the standard deviation (SD) of the logarithm of odds ratio (LOR) obtained in the GSEA analysis for each evaluated function. Consequently, studies with less variability are weighted more in the computation of thefinal meta-analysis results than those with high variability. Thus, meta-analysis highlights the most robust endometrial functions associated with each uterine disorder. A total of 6,390 functions (226 KEGG pathways, 4,405 GO biological processes (BP) terms, 1,093 GO molecular functions (MF) terms, and 666 GO cellular components (CC) terms) were evaluated through meta-analysis for each uterine disorder', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "{'chunk_index': 7, 'content': 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\\n .P values were corrected for multiple testing by FDR , and functions were considered statistically signi ficant in the meta-analysis when FDR<0.05. The P values, summary LOR, and 95% confidence interval (CI) were calculated for all functions evaluated in the metaanalysis. To ensure unbiased results, funnel plots were analyzed for each function evaluated in the meta-analysis. The standard error (SE) of the individual LOR for each integrated study was calculated as an accuracy measure of the study performance. The LOR represents the effect size of the function associated with each uterine disorder compared with controls. If study heterogeneity and publication bias did not affect the functional meta-analysis, the individual LOR of all the integrated studies will be inside the 95% CI area of the funnel plot .', 'parent': 'Methods', 'split': True, 'DOI': 'https://doi.org/10.1016/j.fertnstert.2020.01.025', 'Reference': 'Almudena Devesa-Peiro et al.,2020'}\n",
      "Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025:\n",
      " MATERIALS AND METHODS This is a retrospective study integrating case and control data from multiple cohorts with endometrium gene expression in women with uterine disorders. The raw gene expression data and patient meta-data were downloaded from the National Center for Biotechnology Information (NCBI) functional genomics data repository Gene Expression Omnibus (GEO), where data are made available for the scientific community. Following GEO policies, in this repository patient data are anonymized and encrypted, and no additional institutional review board is required for downloading . Study Search and Selection Systematic searches and reviews of individual studies were conducted between October 2016 and June 2017 at the NCBI functional genomics data repository GEO  according to preferred reporting items for systematic reviews and metaanalysis (PRISMA) guidelines. The search identified experiments involving human endometrial transcriptomic case versus control raw data related to uterine pathologies and implantation alterations. The keywords employed in the search included endometriosis, endometrial adenocarcinoma (ADC), recurrent implantation failure (RIF), and recurrent pregnancy loss (RPL), among others (Supplemental Table 1A, available online, for a full list of search terms). No restrictions were placed on publication date or language. Uterine leiomyoma, adenomyosis, and uterine leiomyosarcoma data were not included due to a lack of suitable studies meeting our criteria. For each sample cohort belonging to the same individual study, 39 variables were evaluated (see Supplemental Table 1B), including clinical characteristics of the participants (e.g., age and body mass index), experimental design (e.g., endometrial biopsy preprocessing procedure and transcriptomic platform), and additional study information (e.g.g., GEO identifier and associated publication). Thefinal inclusion criteria were that at least one uterine disorder was evaluated in the sample cohort study design, RNA was extracted directly from human endometrial biopsies (samples collected from eutopic endometrium in endometriosis patients), the sample size was greater than three for both case and control groups belonging to the same study, microarray or RNA sequencing data were obtained using Affymetrix, Illumina, or Agilent gene expression platforms, and raw gene expression data were made freely available to download from GEO. Individual Transcriptomic Analysis: Making Functional Results Comparable The same transcriptomic analysis procedure was applied independently to each of the selected individual data sets for making the functional results comparable to one another before being integrated through functional meta-analysis. The individual transcriptomic analysis consisted of three consecutive steps run in R software  and applied separately to compare cases versus controls within samples from the same study to avoid variability among sample cohorts related to different uterine disorders. The same preprocessing of raw gene expression data, differential expression analysis, and functional enrichment analysis was performed in a sample cohort comparing cases versus controls ( Supplemental Fig. 1A , available online). The preprocessing step was performed using the affy v.1.52.0  R package for studies using Affymetrix platforms to measure gene expression and thelimma v.3.34.5 1262 VOL. 113 NO. 6 / JUNE 2020 ORIGINAL ARTICLE: GENETICS   R package for the remaining platforms. Normalization between samples was done using quantile normalization (limma R package) . Annotation from probe set to gene symbol was implemented with the biomaRt R package v.2.30.0. Annotation from probe set to gene symbol was implemented with the biomaRt R package v.2.30.0 . Next, exploratory analyses were performed to detect outliers and batch effects. The effect of the cycle phase of endometrial biopsy collection on the data was removed using linear models (limma v.3.34.5) as previously described elsewhere  because the samples were collected throughout the menstrual cycle . This correction ensures that the functional alterations obtained are due to the uterine disorder instead of the different proportions of cell types (stromal, epithelial) and/or the distinct menstrual cycle stages in which the endometrial biopsy samples were obtained . Differential expression analyses were applied using the limma R package  and functional enrichment analyses were executed using the gene set enrichment analysis (GSEA) logistic model implemented in themdgsa R package . TheP values for each function were calculated and corrected for false discovery rate (FDR). TheP values for each function were calculated and corrected for false discovery rate (FDR) . Two functional databases were consulted: the Kyoto Encyclopedia of Genes and Genomes (KEGG)  (release 81.0/2017-1-1) and Gene Ontology (GO)  (Release 1.2/2017-03-31). Functional Integration A functional meta-analysis using mdgsa  and metafor , R packages  was applied for each gene function evaluated, integrating each function among the different patient cohorts belonging to the same disorder (see Supplemental Fig. 1A). This methodology robustly highlights the common alterations across a set of studies, thus discarding those alterations that are exclusive of each individual study’s characteristics . In addition, it increases the statistical power through the integration of a higher number of samples and overcomes the limitations of individual studies’ heterogeneity regarding their distinct clinical population characteristics and experimental procedures. All integrations were performed under the DerSimonian and Laird  random effects model to account for individual study heterogeneity in the R environment . This model weighs the functional results of each study according to their inherent variability measured by the standard deviation (SD) of the logarithm of odds ratio (LOR) obtained in the GSEA analysis for each evaluated function. Consequently, studies with less variability are weighted more in the computation of thefinal meta-analysis results than those with high variability. Thus, meta-analysis highlights the most robust endometrial functions associated with each uterine disorder. A total of 6,390 functions (226 KEGG pathways, 4,405 GO biological processes (BP) terms, 1,093 GO molecular functions (MF) terms, and 666 GO cellular components (CC) terms) were evaluated through meta-analysis for each uterine disorder.P values were corrected for multiple testing by FDR , and functions were considered statistically signi ficant in the meta-analysis when FDR<0.05. The P values, summary LOR, and 95% confidence interval (CI) were calculated for all functions evaluated in the metaanalysis. To ensure unbiased results, funnel plots were analyzed for each function evaluated in the meta-analysis. The standard error (SE) of the individual LOR for each integrated study was calculated as an accuracy measure of the study performance. The LOR represents the effect size of the function associated with each uterine disorder compared with controls. If study heterogeneity and publication bias did not affect the functional meta-analysis, the individual LOR of all the integrated studies will be inside the 95% CI area of the funnel plot .\n"
     ]
    }
   ],
   "source": [
    "mynumb=10\n",
    "\n",
    "print(info_splitted_evaluation[mynumb])\n",
    "\n",
    "for d in info_splitted_evaluation:\n",
    "    \n",
    "    if d['Reference'] == info_splitted_evaluation[mynumb]['Reference'] and d['parent'] ==  info_splitted_evaluation[mynumb]['parent']:\n",
    "\n",
    "        print(d)\n",
    "\n",
    "        if d['chunk_index'] == 0:\n",
    "            info = d['content']\n",
    "        else:\n",
    "            info+=d['content'].split(\":\\n \")[-1]\n",
    "\n",
    "print(info)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "4e93f034",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_context(piece_of_paper, all_papers):\n",
    "\n",
    "    for d in all_papers:\n",
    "    \n",
    "        if d['Reference'] == piece_of_paper['Reference'] and d['parent'] ==  piece_of_paper['parent']:\n",
    "\n",
    "            if d['chunk_index'] == 0:\n",
    "                info = d['content']\n",
    "            else:\n",
    "                info+=d['content'].split(\":\\n \")[-1]\n",
    "\n",
    "    return(info)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "0ed9c7ef",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 600/600 [48:13<00:00,  4.82s/it]\n"
     ]
    }
   ],
   "source": [
    "N = 600\n",
    "examples = []\n",
    "for sample in tqdm(random.sample(info_splitted_evaluation,N), total=N):\n",
    "    context = get_context(piece_of_paper=sample, all_papers=info_splitted_evaluation)\n",
    "    response = call_llm(llm=llm,prompt=QA_generation_prompt.format(context=context))\n",
    "    \n",
    "    try:\n",
    "        question = response.split(\"Question:\")[-1].split(\"Answer: \")[0].strip()\n",
    "        answer = response.split(\"Answer: \")[-1].strip()\n",
    "        examples.append({\n",
    "            \"context\" : context,\n",
    "            \"question\" : question,\n",
    "            \"answer\" : answer\n",
    "        })\n",
    "    except Exception as e:\n",
    "        continue\n",
    "\n",
    "with open(\"examples_evaluation_gemini_2-5.pkl\",\"wb\") as f:\n",
    "    pkl.dump(examples, f)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a2f1d753",
   "metadata": {},
   "source": [
    "## Evaluation of questions generated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "efe8cfae",
   "metadata": {},
   "outputs": [],
   "source": [
    "question_groundedness_critique_prompt = \"\"\"\n",
    "You will be given a context and a question.\n",
    "Your task is to provide a 'total rating' scoring how well one can answer the given question unambiguously with the given context.\n",
    "Give your answer on a scale of 1 to 5, where 1 means that the question is not answerable at all given the context, and 5 means that the question is clearly and unambiguously answerable with the context.\n",
    "\n",
    "Provide your answer as follows:\n",
    "\n",
    "Answer:::\n",
    "Evaluation: (your rationale for the rating, as a text)\n",
    "Total rating: (your rating, as a number between 1 and 5)\n",
    "\n",
    "You MUST provide values for 'Evaluation:' and 'Total rating:' in your answer.\n",
    "\n",
    "Now here are the question and context.\n",
    "\n",
    "Question: {question}\\n\n",
    "Context: {context}\\n\n",
    "Answer::: \"\"\"\n",
    "\n",
    "question_relevance_critique_prompt = \"\"\"\n",
    "You will be given a question.\n",
    "Your task is to provide a 'total rating' representing how useful this question can be to researchers in the reproductive medicine field.\n",
    "Give your answer on a scale of 1 to 5, where 1 means that the question is not useful at all, and 5 means that the question is extremely useful.\n",
    "\n",
    "Provide your answer as follows:\n",
    "\n",
    "Answer:::\n",
    "Evaluation: (your rationale for the rating, as a text)\n",
    "Total rating: (your rating, as a number between 1 and 5)\n",
    "\n",
    "You MUST provide values for 'Evaluation:' and 'Total rating:' in your answer.\n",
    "\n",
    "Now here is the question.\n",
    "\n",
    "Question: {question}\\n\n",
    "Answer::: \"\"\"\n",
    "\n",
    "question_standalone_critique_prompt = \"\"\"\n",
    "You will be given a question.\n",
    "Your task is to provide a 'total rating' representing how context-independent this question is.\n",
    "Give your answer on a scale of 1 to 5, where 1 means that the question depends on additional information to be understood, and 5 means that the question makes sense by itself.\n",
    "For instance, if the question refers to a particular setting, like 'in the context' or 'in the document', the rating must be 1.\n",
    "The questions can contain obscure technical nouns or acronyms like Gradio, Hub, Hugging Face or Space and still be a 5: it must simply be clear to an operator with access to documentation what the question is about.\n",
    "\n",
    "For instance, \"What is the name of the checkpoint from which the ViT model is imported?\" should receive a 1, since there is an implicit mention of a context, thus the question is not independent from the context.\n",
    "\n",
    "Provide your answer as follows:\n",
    "\n",
    "Answer:::\n",
    "Evaluation: (your rationale for the rating, as a text)\n",
    "Total rating: (your rating, as a number between 1 and 5)\n",
    "\n",
    "You MUST provide values for 'Evaluation:' and 'Total rating:' in your answer.\n",
    "\n",
    "Now here is the question.\n",
    "\n",
    "Question: {question}\\n\n",
    "Answer::: \"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "5d199ed2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generating critique for each QA couple...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 600/600 [56:53<00:00,  5.69s/it]  \n"
     ]
    }
   ],
   "source": [
    "print(\"Generating critique for each QA couple...\")\n",
    "for output in tqdm(examples,total=len(examples)):\n",
    "\n",
    "    evaluations = {\n",
    "        \"groundedness\": call_llm(\n",
    "            llm,\n",
    "            question_groundedness_critique_prompt.format(context=output[\"context\"], question=output[\"question\"]),\n",
    "        ),\n",
    "        \"relevance\": call_llm(\n",
    "            llm,\n",
    "            question_relevance_critique_prompt.format(question=output[\"question\"]),\n",
    "        ),\n",
    "        \"standalone\": call_llm(\n",
    "            llm,\n",
    "            question_standalone_critique_prompt.format(question=output[\"question\"]),\n",
    "        ),\n",
    "    }\n",
    "    try:\n",
    "        for criterion, evaluation in evaluations.items():\n",
    "            score, eval = (\n",
    "                int(evaluation.split(\"Total rating: \")[-1].strip()),\n",
    "                evaluation.split(\"Total rating: \")[-2].split(\"Evaluation: \")[1],\n",
    "            )\n",
    "            output.update(\n",
    "                {\n",
    "                    f\"{criterion}_score\": score,\n",
    "                    f\"{criterion}_eval\": eval,\n",
    "                }\n",
    "            )\n",
    "    except:\n",
    "        continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "8a312109",
   "metadata": {},
   "outputs": [],
   "source": [
    "generated_questions = pd.DataFrame.from_dict(examples)\n",
    "generated_questions.loc[:,[\"question\",\"context\",\"answer\",\"groundedness_score\",\"relevance_score\",\"standalone_score\"]]\n",
    "with open(\"generated_questions_gemini_2-5.pkl\",\"wb\") as f:\n",
    "    pkl.dump(generated_questions, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "02a98086",
   "metadata": {},
   "outputs": [],
   "source": [
    "generated_questions_final = generated_questions.loc[\n",
    "    (generated_questions[\"groundedness_score\"] >= 4)\n",
    "    & (generated_questions[\"relevance_score\"] >= 4)\n",
    "    & (generated_questions[\"standalone_score\"] >= 4)\n",
    "]\n",
    "with open(\"generated_questions_final_gemini_2-5.pkl\",\"wb\") as f:\n",
    "    pkl.dump(generated_questions_final, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7651e27f",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"generated_questions_final_gemini_2-5.pkl\",\"rb\") as f:\n",
    "    generated_questions_final = pkl.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "15684535",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "question",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "answer",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "groundedness_score",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "relevance_score",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "standalone_score",
         "rawType": "float64",
         "type": "float"
        }
       ],
       "ref": "b435fd07-5c35-464e-ab2f-4098073a1f26",
       "rows": [
        [
         "6",
         "How does the FGB rs1800790A allele affect fibrinogen levels in F13A 34Val/Val wildtypes?",
         "In F13A 34Val/Val wildtypes, carriage of the FGB rs1800790A allele is related to significantly lower fibrinogen levels compared with FGB rs1800790GG wildtypes.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "20",
         "For which patient group might the ERA test be helpful?",
         "The ERA test may be helpful for women with suspected recurrent implantation failure.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "25",
         "What kind of values does the Color Pathway tool in KEGG Mapper accept?",
         "The Color Pathway tool accepts numerical values.",
         "5.0",
         "4.0",
         "5.0"
        ],
        [
         "32",
         "What is the implantation potential of an euploid blastocyst?",
         "Once an euploid blastocyst is identified, its implantation potential is independent of maternal age, at 45-50%.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "38",
         "What does the PRISMA 2020 statement reflect?",
         "The PRISMA 2020 statement reflects advances in methods to identify, select, appraise, and synthesise studies.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "40",
         "What percentage of individuals with Mendelian disorders have pathology contributed by de novo variants?",
         "De novo variants contribute pathology to 40% of individuals with Mendelian disorders.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "75",
         "What are some exogenous factors that can lead to primary ovarian insufficiency?",
         "Viral infections, metabolomic dysfunction, autoimmune diseases, and environmental or iatrogenic factors.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "80",
         "What is recommended when endometrial receptiveness is seriously compromised by a controlled ovarian stimulation protocol?",
         "Fresh embryo replacement should be cancelled, the embryo frozen, and thawed embryo replacement should be performed under natural cycles.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "81",
         "What was the accuracy range of the best performing classifiers in identifying endometriosis?",
         "The best performing classifiers identified endometriosis with 90%–100% accuracy.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "88",
         "Which two Cytoscape plugins compute various topological network parameters?",
         "NetworkAnalyzer and CentiScaPe.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "96",
         "How long does the human endometrium remain receptive to blastocyst implantation during the implantation window?",
         "The implantation window is limited to approximately 48 hours.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "99",
         "Which rare variant test is generally more powerful, RVT1 or RVT2?",
         "Output:::\nQuestion: Which rare variant test is generally more powerful, RVT1 or RVT2?",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "118",
         "How many genes were up-regulated in the thin endometrium group compared to the control group",
         "Output:::\nQuestion: How many genes were up-regulated in the thin endometrium group compared to the control group",
         "5.0",
         "4.0",
         "5.0"
        ],
        [
         "120",
         "Is increased endometrial thickness (greater than 14 mm) associated with decreased implantation or pregnancy rates in assisted reproduction?",
         "No, it is not associated with decreased implantation or pregnancy rates.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "122",
         "What statistical model does edgeR use for count data?",
         "edgeR models count data using an overdispersed Poisson model.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "130",
         "How much did R-spondin1-Fc pretreatment increase mature oocyte retrieval in adult mice?",
         "R-spondin1-",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "131",
         "What are the characteristics of Type I endometrial cancer?",
         "Type I endometrial cancer is estrogen-dependent, arises before and at menopause, is low grade, shows good response to treatment, and is the most prevalent histological type.",
         "5.0",
         "4.0",
         "5.0"
        ],
        [
         "135",
         "How much more correct predictions did REVEL thresholds yield compared to SIFT/PolyPhen2 for all 66 genes?",
         "REVEL thresholds yielded 12.4% more correct predictions than SIFT/PolyPhen2.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "137",
         "Which type of stem cells has been most widely used for strategies aimed at regenerating the ovarian niche?",
         "Mesenchymal stem cells (MSCs).",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "154",
         "How does the ERA computational model classify the endometrium?",
         "The ERA computational model classifies the endometrium as pre-receptive, receptive, or post-receptive based on its transcriptomic profile.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "157",
         "How do RNA-seq and microarray-based models compare in clinical endpoint prediction?",
         "RNA-seq and microarray-based models perform similarly in clinical endpoint prediction.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "161",
         "What is an important prognosticator of endometrial receptivity on the day of oocyte retrieval?",
         "Endometrial pattern.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "170",
         "What is the implantation potential of an euploid blastocyst?",
         "Once an euploid blastocyst is identified, its implantation potential is independent of maternal age, at 45-50%.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "173",
         "What should guide the restart of ART services?",
         "Advanced planning.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "179",
         "How does CFTR expression in ovarian cancer specimens compare to benign or normal ovaries?",
         "CFTR expression in ovarian cancer",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "190",
         "Which human genome build does CADD version 1.4 support?",
         "CADD version 1.4 supports the human genome build GRCh38.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "193",
         "Which type of stem cells has been most widely used for strategies aimed at regenerating the ovarian niche?",
         "Mesenchymal stem cells (MSCs).",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "199",
         "What are the two main steps of the STAR algorithm?",
         "The STAR algorithm consists of a seed searching step and a clustering/stitching/scoring step.",
         "5.0",
         "4.0",
         "5.0"
        ],
        [
         "204",
         "Which outer dynein arm (ODA) components can Dnah6 genetically interact with?",
         "Dnah6 can genetically interact with Dnai1 and Dnah5.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "244",
         "What percentage of additional biomarkers were detected after correcting for the menstrual cycle effect?",
         "A significant average of 44.19% more biomarkers were detected.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "259",
         "How much more correct predictions did REVEL thresholds yield compared to SIFT/PolyPhen2 for all 66 genes?",
         "REVEL thresholds yielded 12.4% more correct predictions than SIFT/PolyPhen2.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "266",
         "Which 'omics' technique has been most commonly applied to the human endometrium?",
         "Analysis of the transcriptome pattern.",
         "5.0",
         "4.0",
         "5.0"
        ],
        [
         "283",
         "What happens to oocytes in CDC25B-deficient mice?",
         "They do not resume meiosis and remain arrested at prophase I.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "292",
         "Which patients are most likely to benefit from preimplantation genetic testing for aneuploidy?",
         "Infertile women of advanced maternal age.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "308",
         "What method was used for the analysis of urine cortisol?",
         "Urine cortisol was analyzed using a commercially available enzyme-linked immunosorbent assay kit.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "309",
         "Where is R-spondin2 exclusively expressed in ovarian follicles?",
         "R-spondin2 is exclusively expressed in oocytes of ovarian follicles.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "314",
         "How much was estradiol concentration increased in obese women compared to the lightest women?",
         "Estradiol concentration was increased by 130% in obese women compared with the lightest women.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "325",
         "What is considered the normal range of CGG nucleotide repeats for the FMR1 gene's ovarian function?",
         "The normal range for the FMR1 gene's ovarian function is 26 to 34 CGG nucleotide repeats.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "336",
         "What methods does clusterProfiler offer for gene classification and enrichment analyses?",
         "clusterProfiler offers groupGO, enrichGO, and enrichKEGG.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "351",
         "What parameters does the NetworkAnalyzer plugin in Cytoscape calculate?",
         "The NetworkAnalyzer plugin calculates the degree of vertices, vertex betweenness centrality, vertex clustering coefficient, and edge betweenness.",
         "5.0",
         "4.0",
         "5.0"
        ],
        [
         "358",
         "What is maSigPro-GLM suitable for identifying?",
         "maSigPro-GLM is suitable for the identification of DEGs from time-course RNA-seq data.",
         "5.0",
         "4.0",
         "5.0"
        ],
        [
         "362",
         "Which aquaporin mRNA expressions were significantly decreased in neonatal AQP8-deficient ovaries?",
         "The expression of AQP7, AQP11, and AQP12 mRNAs were significantly decreased in neonatal AQP8-deficient ovaries.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "374",
         "What steps are required for sequencing-based association studies?",
         "Sequencing-based association studies require platform selection, quality control, choice of analysis units, assignment of variants to analysis units with bioinformatic tools, selection of methods for testing rare-variant effects, and prioritization and replication of top signals.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "376",
         "What is considered the normal range of CGG nucleotide repeats for FMR1 gene ovarian function?",
         "26 to 34 CGG nucleotide repeats.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "380",
         "What functional alteration was statistically significantly dysregulated as a shared alteration across uterine disorders at the endometrial level?",
         "Cell cycle alterations.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "384",
         "What are the well-supported molecular hallmarks of endometriosis?",
         "Genetic predisposition, estrogen dependence, progesterone resistance, and inflammation.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "385",
         "What is the increased likelihood of advanced age mothers experiencing pregnancy-induced hypertension compared to adult mothers?",
         "Advanced age mothers are four times more likely to encounter pregnancy-induced hypertension than their adult counterparts.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "391",
         "Where does age-related hypermethylation preferentially occur?",
         "At CpG islands, at bivalent chromatin domain promoters that are associated with key developmental genes, and at Polycomb-group protein targets.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "397",
         "What do embryonic signals act on to induce immune tolerance?",
         "They act on maternal immune cells.",
         "5.0",
         "5.0",
         "5.0"
        ],
        [
         "417",
         "What kind of values does the Color Pathway tool in KEGG Mapper accept?",
         "The Color Pathway tool accepts numerical values.",
         "5.0",
         "4.0",
         "5.0"
        ]
       ],
       "shape": {
        "columns": 5,
        "rows": 70
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>question</th>\n",
       "      <th>answer</th>\n",
       "      <th>groundedness_score</th>\n",
       "      <th>relevance_score</th>\n",
       "      <th>standalone_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>How does the FGB rs1800790A allele affect fibr...</td>\n",
       "      <td>In F13A 34Val/Val wildtypes, carriage of the F...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>For which patient group might the ERA test be ...</td>\n",
       "      <td>The ERA test may be helpful for women with sus...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>What kind of values does the Color Pathway too...</td>\n",
       "      <td>The Color Pathway tool accepts numerical values.</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>What is the implantation potential of an euplo...</td>\n",
       "      <td>Once an euploid blastocyst is identified, its ...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>What does the PRISMA 2020 statement reflect?</td>\n",
       "      <td>The PRISMA 2020 statement reflects advances in...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>566</th>\n",
       "      <td>Which genes share genetic susceptibility for A...</td>\n",
       "      <td>The ESR1, HK3, and BRSK1 genes share genetic s...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>573</th>\n",
       "      <td>What database were the GSE26787 and GSE63901 d...</td>\n",
       "      <td>The Gene Expression Omnibus (GEO) database.</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>576</th>\n",
       "      <td>What percentage of women globally are affected...</td>\n",
       "      <td>3.7% of women globally.</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>589</th>\n",
       "      <td>What is the purpose of unique molecular identi...</td>\n",
       "      <td>Unique molecular identifiers are applied to ov...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596</th>\n",
       "      <td>What are pinopodes?</td>\n",
       "      <td>Pinopodes are apical protrusions on the epithe...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>70 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              question  \\\n",
       "6    How does the FGB rs1800790A allele affect fibr...   \n",
       "20   For which patient group might the ERA test be ...   \n",
       "25   What kind of values does the Color Pathway too...   \n",
       "32   What is the implantation potential of an euplo...   \n",
       "38        What does the PRISMA 2020 statement reflect?   \n",
       "..                                                 ...   \n",
       "566  Which genes share genetic susceptibility for A...   \n",
       "573  What database were the GSE26787 and GSE63901 d...   \n",
       "576  What percentage of women globally are affected...   \n",
       "589  What is the purpose of unique molecular identi...   \n",
       "596                                What are pinopodes?   \n",
       "\n",
       "                                                answer  groundedness_score  \\\n",
       "6    In F13A 34Val/Val wildtypes, carriage of the F...                 5.0   \n",
       "20   The ERA test may be helpful for women with sus...                 5.0   \n",
       "25    The Color Pathway tool accepts numerical values.                 5.0   \n",
       "32   Once an euploid blastocyst is identified, its ...                 5.0   \n",
       "38   The PRISMA 2020 statement reflects advances in...                 5.0   \n",
       "..                                                 ...                 ...   \n",
       "566  The ESR1, HK3, and BRSK1 genes share genetic s...                 5.0   \n",
       "573        The Gene Expression Omnibus (GEO) database.                 5.0   \n",
       "576                            3.7% of women globally.                 5.0   \n",
       "589  Unique molecular identifiers are applied to ov...                 5.0   \n",
       "596  Pinopodes are apical protrusions on the epithe...                 5.0   \n",
       "\n",
       "     relevance_score  standalone_score  \n",
       "6                5.0               5.0  \n",
       "20               5.0               5.0  \n",
       "25               4.0               5.0  \n",
       "32               5.0               5.0  \n",
       "38               5.0               5.0  \n",
       "..               ...               ...  \n",
       "566              5.0               5.0  \n",
       "573              4.0               5.0  \n",
       "576              5.0               5.0  \n",
       "589              5.0               5.0  \n",
       "596              4.0               5.0  \n",
       "\n",
       "[70 rows x 5 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "generated_questions_final.loc[:,[\"question\",\"answer\",\"groundedness_score\",\"relevance_score\",\"standalone_score\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "69babbd7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'context': 'Christian Schwedler et al.,2021, DOI:https://doi.org/10.3390/life11121340:\\n Fibrinogen synthesis is stimulated by proinflammatory triggers and depends on α-, β- and γ-fibrinogen (\\nFGA\\n,\\nFGB\\n,\\nFGG\\n) genotypes. Constellations of fibrinogen, factor XIII A-subunit (\\nF13A\\n) and α\\n2\\n-antiplasmin (\\nA2AP\\n) genotypes predisposing for dense fibrin gels with high antifibrinolytic capacity (e.g.,\\nFGB\\nrs1800790 A-allele carriage in\\nF13A\\n34Val/Val or\\nA2AP\\n6Arg/Arg wildtypes) are related with reduced inflammation. As both relationships are likely to influence each other, we tested whether the association of fibrinogen genotypes with fibrinogen levels is influenced by\\nF13A\\nVal34Leu and\\nA2AP\\nArg6Trp genotypes in a population under proinflammatory stress. In total, 639 women were followed during pregnancy (2218 observations). The relationship between fibrinogen genotypes and levels was statistically assessed in univariate and multivariate analyses without and with stratification for\\nF13A\\nVal34Leu and\\nA2AP\\nArg6Trp. Strong associations with fibrinogen levels could be found forF13A\\nVal34Leu and\\nA2AP\\nArg6Trp. Strong associations with fibrinogen levels could be found for\\nFGB\\nrs1800790G > A,\\nFGA\\nrs2070016T > C and\\nFGG\\nrs1049636T > C. For\\nFGB\\nrs1800790G > A and\\nFGA\\nrs2070016T > C, this relationship significantly depended on\\nF13A\\nVal34Leu and\\nA2AP\\nArg6Trp genotypes. Specifically, in\\nF13A\\n34Val/Val wildtypes, carriage of\\nFGB\\nrs1800790A was related to significantly lower fibrinogen levels compared with\\nFGB\\nrs1800790GG wildtypes (\\np\\n< 0.01). For\\nA2AP\\n6Arg/Arg wildtypes, a comparable relationship could be found (\\np\\n< 0.04). As these genotype constellations related to lower fibrinogen levels have previously been shown to be associated with reduced inflammatory activity, these findings suggest that the influence of fibrinogen,\\nF13A\\nand\\nA2AP\\ngenotypes on inflammation could affect the control of fibrinogen levels and vice versa.',\n",
       " 'question': 'How does the FGB rs1800790A allele affect fibrinogen levels in F13A 34Val/Val wildtypes?',\n",
       " 'answer': 'In F13A 34Val/Val wildtypes, carriage of the FGB rs1800790A allele is related to significantly lower fibrinogen levels compared with FGB rs1800790GG wildtypes.',\n",
       " 'groundedness_score': 5.0,\n",
       " 'groundedness_eval': 'The context explicitly states: \"Specifically, in F13A 34Val/Val wildtypes, carriage of FGB rs1800790A was related to significantly lower fibrinogen levels compared with FGB rs1800790GG wildtypes (p < 0.01).\" This directly and unambiguously answers how the FGB rs1800790A allele affects fibrinogen levels in F13A 34Val/Val wildtypes.\\n',\n",
       " 'relevance_score': 5.0,\n",
       " 'relevance_eval': \"This question is highly relevant and potentially extremely useful for researchers in reproductive medicine. Coagulation disorders, including those related to fibrinogen and Factor XIII, are significant risk factors for various adverse pregnancy outcomes such as recurrent pregnancy loss, preeclampsia, intrauterine growth restriction, and placental abruption. Understanding how specific genetic variants (like FGB rs1800790A) interact with other genetic factors (like F13A 34Val/Val genotype) to influence fibrinogen levels can provide crucial insights into the genetic predisposition to thrombophilia in pregnant women. This knowledge can contribute to:\\n1.  **Improved risk stratification:** Identifying women at higher genetic risk for thrombotic complications during pregnancy.\\n2.  **Personalized medicine:** Guiding prophylactic interventions based on an individual's genetic profile.\\n3.  **Pathophysiological understanding:** Elucidating the molecular mechanisms underlying coagulation abnormalities in reproductive health.\\nTherefore, investigating this specific genetic interaction and its impact on a key coagulation parameter like fibrinogen levels has direct clinical and research implications for reproductive medicine.\\n\",\n",
       " 'standalone_score': 5.0,\n",
       " 'standalone_eval': 'The question uses highly specific scientific and genetic terminology (FGB rs1800790A allele, fibrinogen levels, F13A 34Val/Val wildtypes). While these terms are obscure to a general audience, they are well-defined concepts within the field of genetics and biology. The question itself does not refer to any external document, context, or specific setting (like \"in this text\" or \"from the provided data\") that would be necessary to understand the question\\'s intent. An operator with access to scientific documentation or databases would understand what information is being sought. Thus, the question makes sense by itself, even if the answer requires specialized knowledge.\\n'}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eval_dataset = generated_questions_final.to_dict(\"records\")\n",
    "eval_dataset[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "763d2a80",
   "metadata": {},
   "source": [
    "### Create a function to check RAG performance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e0c5ef00",
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_embeddings(\n",
    "    documents,\n",
    "    chunk_size: int,\n",
    "    embedding_model\n",
    "):\n",
    "    splitter = RecursiveCharacterTextSplitter(\n",
    "    chunk_size=chunk_size,\n",
    "    chunk_overlap=chunk_size//10,\n",
    "    separators=[\"\\n\\n\", \"\\n\", \".\", \"!\", \"?\", \" \"]  # smart splitting\n",
    "    )\n",
    "\n",
    "    info_splitted = []\n",
    "\n",
    "    for j in documents:\n",
    "\n",
    "        for key, value in j.items():\n",
    "        \n",
    "            if key in ['Abstract', 'Introduction', 'Methods', 'Results', 'Discussion', 'Conclusion',] and value:\n",
    "\n",
    "                if len(value) > 1200:\n",
    "                    chunks = splitter.split_text(value)\n",
    "\n",
    "                    for i, c in enumerate(chunks):\n",
    "\n",
    "                        info_splitted.append(\n",
    "                            {\n",
    "                                \"chunk_index\":i,\n",
    "                                \"content\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")+\", DOI:\"+j.get(\"DOI\")+\"\\n\"+c,\n",
    "                                \"parent\":key,\n",
    "                                \"split\":True,\n",
    "                                \"DOI\":j.get(\"DOI\"),\n",
    "                                \"Reference\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")\n",
    "                            }\n",
    "                        )\n",
    "                else:\n",
    "\n",
    "                    info_splitted.append(\n",
    "                            {\n",
    "                                \"chunk_index\":0,\n",
    "                                \"content\":j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")+\", DOI:\"+j.get(\"DOI\")+\"\\n\"+value,\n",
    "                                \"parent\":key,\n",
    "                                \"split\":False,\n",
    "                                \"DOI\":j.get(\"DOI\"),\n",
    "                                \"Reference\": j.get('Authors').split(\",\")[0]+\" et al.,\"+j.get('Publication',\"Not identified\")\n",
    "                            }\n",
    "                        )\n",
    "    \n",
    "    texts = [chunk[\"content\"] for chunk in info_splitted]\n",
    "    metadatas = [{\"parent\": chunk[\"parent\"], \"chunk_index\": chunk[\"chunk_index\"],\"DOI\": chunk[\"DOI\"], \"Reference\": chunk[\"Reference\"]} for chunk in info_splitted]\n",
    "    ids = [str(uuid.uuid1()) for _ in metadatas]\n",
    "\n",
    "    db = Chroma.from_texts(\n",
    "    texts=texts,\n",
    "    embedding=embedding_model,\n",
    "    metadatas=metadatas,\n",
    "    ids=ids,\n",
    ")\n",
    "\n",
    "    return(db)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "064d4ff5",
   "metadata": {},
   "outputs": [],
   "source": [
    "RAG_PROMPT_TEMPLATE = \"\"\"\n",
    "<|system|>\n",
    "Using the information contained in the context,\n",
    "give a comprehensive answer to the question.\n",
    "Respond only to the question asked, response should be concise and relevant to the question.\n",
    "Provide the number of the source document when relevant.\n",
    "If the answer cannot be deduced from the context, do not give an answer.</s>\n",
    "<|user|>\n",
    "Context:\n",
    "{context}\n",
    "---\n",
    "Now here is the question you need to answer.\n",
    "\n",
    "Question: {question}\n",
    "</s>\n",
    "<|assistant|>\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0e054c3c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def clean_doi_links(text):\n",
    "    \"\"\"\n",
    "    Replace problematic Unicode dashes (like non-breaking hyphen) with normal ASCII dashes.\n",
    "    \"\"\"\n",
    "    return re.sub(r\"[\\u2010-\\u2015\\u2212]\", \"-\", text)\n",
    "\n",
    "def retrieve_context(question, k, database):\n",
    "    results = database.similarity_search(question, k)\n",
    "    selected_index = []\n",
    "    ideal_chunks = []\n",
    "    meta_selected = []\n",
    "\n",
    "    def is_new_chunk(r, selected_index):\n",
    "        next_chunk = \"_\".join([r[\"parent\"], r[\"Reference\"], str(r[\"chunk_index\"] + 1)])\n",
    "        prev_chunk = \"_\".join([r[\"parent\"], r[\"Reference\"], str(r[\"chunk_index\"] - 1)])\n",
    "        return next_chunk not in selected_index and prev_chunk not in selected_index\n",
    "\n",
    "    for doc in results:\n",
    "        r = doc.metadata\n",
    "\n",
    "        if r[\"parent\"] not in [\"Journal\", \"DOI\"] and is_new_chunk(r, selected_index):\n",
    "            ii = \"_\".join([r[\"parent\"], r[\"Reference\"], str(r[\"chunk_index\"])])\n",
    "            selected_index.append(ii)\n",
    "\n",
    "            candidates = db.get(\n",
    "                where={\"$and\": [{\"Reference\": r[\"Reference\"]}, {\"parent\": r[\"parent\"]}]}\n",
    "            )\n",
    "\n",
    "            max_index = len(candidates[\"metadatas\"]) - 1\n",
    "\n",
    "            meta_selected.append(candidates[\"metadatas\"])\n",
    "            ideal_chunks.append(\n",
    "                [\n",
    "                    doc\n",
    "                    for doc, meta in zip(\n",
    "                        candidates[\"documents\"], candidates[\"metadatas\"]\n",
    "                    )\n",
    "                    if meta[\"chunk_index\"]\n",
    "                    in [\n",
    "                        r[\"chunk_index\"],\n",
    "                        max(r[\"chunk_index\"] - 1, 0),\n",
    "                        min(r[\"chunk_index\"] + 1, max_index),\n",
    "                    ]\n",
    "                ]\n",
    "                )\n",
    "\n",
    "        context = []\n",
    "        for text, meta in zip(ideal_chunks, meta_selected):\n",
    "            if meta:  # Only proceed if meta is not empty\n",
    "                doi = (\n",
    "                    clean_doi_links(meta[0][\"DOI\"])\n",
    "                    if \"DOI\" in meta[0]\n",
    "                    else \"DOI not available\"\n",
    "                )\n",
    "                context.append(\n",
    "                    f\"Summary:\\n\\n{''.join(text)}\\n\\n\"\n",
    "                )\n",
    "\n",
    "    return(context)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ff0a7875",
   "metadata": {},
   "outputs": [],
   "source": [
    "def answer_with_rag(\n",
    "    question: str,\n",
    "    llm,\n",
    "    database,\n",
    "    num_docs_final=7,\n",
    "    recursive_chunk = False\n",
    "):\n",
    "    \"\"\"Answer a question using RAG with the given knowledge index.\"\"\"\n",
    "    # Gather documents with retriever\n",
    "    \n",
    "    if  recursive_chunk:\n",
    "        relevant_docs = retrieve_context(question=question, database=database,k=num_docs_final)\n",
    "    else:\n",
    "        relevant_docs = database.similarity_search(query=question, k=num_docs_final)\n",
    "        relevant_docs = [doc.page_content for doc in relevant_docs]  # keep only the text\n",
    "    \n",
    "\n",
    "    # Build the final prompt\n",
    "    context = \"\\nExtracted documents:\\n\"\n",
    "    context += \"\".join([f\"Document {str(i)}:::\\n\" + doc for i, doc in enumerate(relevant_docs)])\n",
    "\n",
    "    final_prompt = RAG_PROMPT_TEMPLATE.format(question=question, context=context)\n",
    "\n",
    "    # Redact an answer\n",
    "    answer = llm.invoke(final_prompt)\n",
    "\n",
    "    return answer.content, relevant_docs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d66498c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_rag_tests(\n",
    "    eval_dataset,\n",
    "    llm,\n",
    "    database,\n",
    "    output_file,\n",
    "    recursive_chunk,\n",
    "    verbose=False,\n",
    "    test_settings = None,\n",
    "    num_docs_final = 7 # To document the test settings used\n",
    "):\n",
    "    \"\"\"Runs RAG tests on the given dataset and saves the results to the given output file.\"\"\"\n",
    "    try:  # load previous generations if they exist\n",
    "        with open(output_file, \"r\") as f:\n",
    "            outputs = json.load(f)\n",
    "    except:\n",
    "        outputs = []\n",
    "\n",
    "    for example in tqdm(eval_dataset):\n",
    "        question = example[\"question\"]\n",
    "        if question in [output[\"question\"] for output in outputs]:\n",
    "            continue\n",
    "            \n",
    "        answer, relevant_docs = answer_with_rag(question=question,llm=llm, database=database,recursive_chunk=recursive_chunk,num_docs_final=num_docs_final)\n",
    "\n",
    "        if verbose:\n",
    "            print(\"=======================================================\")\n",
    "            print(f\"Question: {question}\")\n",
    "            print(f\"Answer: {answer}\")\n",
    "            print(f'True answer: {example[\"answer\"]}')\n",
    "        result = {\n",
    "            \"question\": question,\n",
    "            \"true_answer\": example[\"answer\"],\n",
    "            \"generated_answer\": answer,\n",
    "            \"retrieved_docs\": [doc for doc in relevant_docs],\n",
    "        }\n",
    "        if test_settings:\n",
    "            result[\"test_settings\"] = test_settings\n",
    "        outputs.append(result)\n",
    "\n",
    "        with open(output_file, \"w\") as f:\n",
    "            json.dump(outputs, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "cb5f5681",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ChatPromptTemplate(input_variables=['instruction', 'reference_answer', 'response'], input_types={}, partial_variables={}, messages=[SystemMessage(content='You are a fair evaluator language model.', additional_kwargs={}, response_metadata={}), HumanMessagePromptTemplate(prompt=PromptTemplate(input_variables=['instruction', 'reference_answer', 'response'], input_types={}, partial_variables={}, template='###Task Description:\\nAn instruction (might include an Input inside it), a response to evaluate, a reference answer that gets a score of 5, and a score rubric representing a evaluation criteria are given.\\n1. Write a detailed feedback that assess the quality of the response strictly based on the given score rubric, not evaluating in general.\\n2. After writing a feedback, write a score that is an integer between 1 and 5. You should refer to the score rubric.\\n3. The output format MUST look as follows: \"Feedback: {{write a feedback for criteria}} [RESULT] {{an integer number between 1 and 5}}\"\\n4. Please do not generate any other opening, closing, and explanations. Be sure to include [RESULT] in your output, it is required.\\n\\n###The instruction to evaluate:\\n{instruction}\\n\\n###Response to evaluate:\\n{response}\\n\\n###Reference Answer (Score 5):\\n{reference_answer}\\n\\n###Score Rubrics:\\n[Is the response correct, accurate, and factual based on the reference answer?]\\nScore 1: The response is completely incorrect, inaccurate, and/or not factual.\\nScore 2: The response is mostly incorrect, inaccurate, and/or not factual.\\nScore 3: The response is somewhat correct, accurate, and/or factual.\\nScore 4: The response is mostly correct, accurate, and factual.\\nScore 5: The response is completely correct, accurate, and factual.\\n\\n###Feedback:'), additional_kwargs={})])"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from langchain.prompts.chat import (\n",
    "    ChatPromptTemplate,\n",
    "    HumanMessagePromptTemplate,\n",
    ")\n",
    "from langchain.schema import SystemMessage\n",
    "\n",
    "EVALUATION_PROMPT = \"\"\"###Task Description:\n",
    "An instruction (might include an Input inside it), a response to evaluate, a reference answer that gets a score of 5, and a score rubric representing a evaluation criteria are given.\n",
    "1. Write a detailed feedback that assess the quality of the response strictly based on the given score rubric, not evaluating in general.\n",
    "2. After writing a feedback, write a score that is an integer between 1 and 5. You should refer to the score rubric.\n",
    "3. The output format MUST look as follows: \\\"Feedback: {{write a feedback for criteria}} [RESULT] {{an integer number between 1 and 5}}\\\"\n",
    "4. Please do not generate any other opening, closing, and explanations. Be sure to include [RESULT] in your output, it is required.\n",
    "\n",
    "###The instruction to evaluate:\n",
    "{instruction}\n",
    "\n",
    "###Response to evaluate:\n",
    "{response}\n",
    "\n",
    "###Reference Answer (Score 5):\n",
    "{reference_answer}\n",
    "\n",
    "###Score Rubrics:\n",
    "[Is the response correct, accurate, and factual based on the reference answer?]\n",
    "Score 1: The response is completely incorrect, inaccurate, and/or not factual.\n",
    "Score 2: The response is mostly incorrect, inaccurate, and/or not factual.\n",
    "Score 3: The response is somewhat correct, accurate, and/or factual.\n",
    "Score 4: The response is mostly correct, accurate, and factual.\n",
    "Score 5: The response is completely correct, accurate, and factual.\n",
    "\n",
    "###Feedback:\"\"\"\n",
    "\n",
    "evaluation_prompt_template = ChatPromptTemplate.from_messages(\n",
    "    [\n",
    "        SystemMessage(content=\"You are a fair evaluator language model.\"),\n",
    "        HumanMessagePromptTemplate.from_template(EVALUATION_PROMPT),\n",
    "    ]\n",
    ")\n",
    "\n",
    "evaluation_prompt_template"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "19ca2c79",
   "metadata": {},
   "outputs": [],
   "source": [
    "def evaluate_answers(\n",
    "    answer_path: str,\n",
    "    eval_chat_model,\n",
    "    evaluator_name: str,\n",
    "    evaluation_prompt_template: ChatPromptTemplate,\n",
    ") -> None:\n",
    "    \"\"\"Evaluates generated answers. Modifies the given answer file in place for better checkpointing.\"\"\"\n",
    "    answers = []\n",
    "    if os.path.isfile(answer_path):  # load previous generations if they exist\n",
    "        answers = json.load(open(answer_path, \"r\"))\n",
    "\n",
    "    for experiment in tqdm(answers):\n",
    "        if f\"eval_score_{evaluator_name}\" in experiment:\n",
    "            continue\n",
    "\n",
    "        eval_prompt = evaluation_prompt_template.format_messages(\n",
    "            instruction=experiment[\"question\"],\n",
    "            response=experiment[\"generated_answer\"],\n",
    "            reference_answer=experiment[\"true_answer\"],\n",
    "        )\n",
    "        \n",
    "        eval_result = eval_chat_model.invoke(eval_prompt)\n",
    "        feedback, score = [item.strip() for item in eval_result.content.split(\"[RESULT]\")]\n",
    "        experiment[f\"eval_score_{evaluator_name}\"] = score\n",
    "        experiment[f\"eval_feedback_{evaluator_name}\"] = feedback\n",
    "\n",
    "        with open(answer_path, \"w\") as f:\n",
    "            json.dump(answers, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "e159aff1",
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.exists(\"./output\"):\n",
    "    os.mkdir(\"./output\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "acee4bab",
   "metadata": {},
   "outputs": [],
   "source": [
    "embedding_function2 = HuggingFaceEmbeddings(model_name=\"avsolatorio/GIST-small-Embedding-v0\") # model_kwargs={'device': 'cuda'}\n",
    "# embedding_function2 = HuggingFaceEmbeddings(model_name=\"Snowflake/snowflake-arctic-embed-s\")\n",
    "# embedding_function2 = HuggingFaceEmbeddings(model_name=\"intfloat/e5-small-v2\")  #intfloat/e5-small-v2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "b312c9c8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation for chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_4:\n",
      "Loading knowledge base embeddings...\n",
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 70/70 [00:00<00:00, 171196.08it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 63/63 [00:00<00:00, 684562.57it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_5:\n",
      "Loading knowledge base embeddings...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 70/70 [00:00<00:00, 189664.91it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 63/63 [00:00<00:00, 836206.18it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_6:\n",
      "Loading knowledge base embeddings...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 70/70 [00:00<00:00, 144845.23it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 63/63 [00:00<00:00, 763702.75it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7:\n",
      "Loading knowledge base embeddings...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 70/70 [00:00<00:00, 187365.21it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 63/63 [00:00<00:00, 1065488.52it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_8:\n",
      "Loading knowledge base embeddings...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 70/70 [00:00<00:00, 193668.39it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 63/63 [00:00<00:00, 830947.02it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_9:\n",
      "Loading knowledge base embeddings...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 70/70 [00:00<00:00, 198245.29it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 63/63 [00:00<00:00, 828342.17it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_10:\n",
      "Loading knowledge base embeddings...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 70/70 [00:00<00:00, 199864.72it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 63/63 [00:00<00:00, 1048576.00it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_True_k_4:\n",
      "Loading knowledge base embeddings...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 70/70 [00:00<00:00, 178156.12it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 63/63 [00:00<00:00, 866364.43it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_True_k_5:\n",
      "Loading knowledge base embeddings...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 70/70 [00:00<00:00, 198245.29it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 63/63 [00:00<00:00, 889700.85it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_True_k_6:\n",
      "Loading knowledge base embeddings...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 70/70 [00:00<00:00, 202623.38it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running evaluation...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 63/63 [00:00<00:00, 828342.17it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing database\n",
      "Running evaluation for chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_True_k_7:\n",
      "Loading knowledge base embeddings...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running RAG...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 80%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▏                                  | 56/70 [01:18<00:19,  1.40s/it]\n"
     ]
    },
    {
     "ename": "InternalError",
     "evalue": "Database error: error returned from database: (code: 1) no such table: collections",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mInternalError\u001b[0m                             Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[17], line 30\u001b[0m\n\u001b[1;32m     23\u001b[0m db \u001b[38;5;241m=\u001b[39m load_embeddings(\n\u001b[1;32m     24\u001b[0m     info_articles_final,\n\u001b[1;32m     25\u001b[0m     chunk_size\u001b[38;5;241m=\u001b[39mchunk_size,\n\u001b[1;32m     26\u001b[0m     embedding_model\u001b[38;5;241m=\u001b[39membedding_function,\n\u001b[1;32m     27\u001b[0m )\n\u001b[1;32m     29\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRunning RAG...\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m---> 30\u001b[0m \u001b[43mrun_rag_tests\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     31\u001b[0m \u001b[43m    \u001b[49m\u001b[43meval_dataset\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43meval_dataset\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     32\u001b[0m \u001b[43m    \u001b[49m\u001b[43mllm\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mllm_reader\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     33\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdatabase\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdb\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     34\u001b[0m \u001b[43m    \u001b[49m\u001b[43moutput_file\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43moutput_file_name\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     35\u001b[0m \u001b[43m    \u001b[49m\u001b[43mverbose\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m     36\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtest_settings\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msettings_name\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     37\u001b[0m \u001b[43m    \u001b[49m\u001b[43mrecursive_chunk\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrecursive_chunk\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     38\u001b[0m \u001b[43m    \u001b[49m\u001b[43mnum_docs_final\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mk\u001b[49m\n\u001b[1;32m     39\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     41\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRunning evaluation...\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     42\u001b[0m evaluate_answers(\n\u001b[1;32m     43\u001b[0m     output_file_name,\n\u001b[1;32m     44\u001b[0m     llm_reader,\n\u001b[1;32m     45\u001b[0m     llm_model,\n\u001b[1;32m     46\u001b[0m     evaluation_prompt_template,\n\u001b[1;32m     47\u001b[0m )\n",
      "Cell \u001b[0;32mIn[11], line 23\u001b[0m, in \u001b[0;36mrun_rag_tests\u001b[0;34m(eval_dataset, llm, database, output_file, recursive_chunk, verbose, test_settings, num_docs_final)\u001b[0m\n\u001b[1;32m     20\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m question \u001b[38;5;129;01min\u001b[39;00m [output[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mquestion\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;28;01mfor\u001b[39;00m output \u001b[38;5;129;01min\u001b[39;00m outputs]:\n\u001b[1;32m     21\u001b[0m     \u001b[38;5;28;01mcontinue\u001b[39;00m\n\u001b[0;32m---> 23\u001b[0m answer, relevant_docs \u001b[38;5;241m=\u001b[39m \u001b[43manswer_with_rag\u001b[49m\u001b[43m(\u001b[49m\u001b[43mquestion\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquestion\u001b[49m\u001b[43m,\u001b[49m\u001b[43mllm\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mllm\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdatabase\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdatabase\u001b[49m\u001b[43m,\u001b[49m\u001b[43mrecursive_chunk\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrecursive_chunk\u001b[49m\u001b[43m,\u001b[49m\u001b[43mnum_docs_final\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnum_docs_final\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     25\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m verbose:\n\u001b[1;32m     26\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m=======================================================\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "Cell \u001b[0;32mIn[10], line 12\u001b[0m, in \u001b[0;36manswer_with_rag\u001b[0;34m(question, llm, database, num_docs_final, recursive_chunk)\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[38;5;66;03m# Gather documents with retriever\u001b[39;00m\n\u001b[1;32m     11\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m  recursive_chunk:\n\u001b[0;32m---> 12\u001b[0m     relevant_docs \u001b[38;5;241m=\u001b[39m \u001b[43mretrieve_context\u001b[49m\u001b[43m(\u001b[49m\u001b[43mquestion\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquestion\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdatabase\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdatabase\u001b[49m\u001b[43m,\u001b[49m\u001b[43mk\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnum_docs_final\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     13\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m     14\u001b[0m     relevant_docs \u001b[38;5;241m=\u001b[39m database\u001b[38;5;241m.\u001b[39msimilarity_search(query\u001b[38;5;241m=\u001b[39mquestion, k\u001b[38;5;241m=\u001b[39mnum_docs_final)\n",
      "Cell \u001b[0;32mIn[9], line 25\u001b[0m, in \u001b[0;36mretrieve_context\u001b[0;34m(question, k, database)\u001b[0m\n\u001b[1;32m     22\u001b[0m ii \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m_\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m.\u001b[39mjoin([r[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mparent\u001b[39m\u001b[38;5;124m\"\u001b[39m], r[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mReference\u001b[39m\u001b[38;5;124m\"\u001b[39m], \u001b[38;5;28mstr\u001b[39m(r[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mchunk_index\u001b[39m\u001b[38;5;124m\"\u001b[39m])])\n\u001b[1;32m     23\u001b[0m selected_index\u001b[38;5;241m.\u001b[39mappend(ii)\n\u001b[0;32m---> 25\u001b[0m candidates \u001b[38;5;241m=\u001b[39m \u001b[43mdb\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     26\u001b[0m \u001b[43m    \u001b[49m\u001b[43mwhere\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m{\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m$and\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43m[\u001b[49m\u001b[43m{\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mReference\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mr\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mReference\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m}\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m{\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mparent\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mr\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mparent\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m}\u001b[49m\u001b[43m]\u001b[49m\u001b[43m}\u001b[49m\n\u001b[1;32m     27\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     29\u001b[0m max_index \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlen\u001b[39m(candidates[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmetadatas\u001b[39m\u001b[38;5;124m\"\u001b[39m]) \u001b[38;5;241m-\u001b[39m \u001b[38;5;241m1\u001b[39m\n\u001b[1;32m     31\u001b[0m meta_selected\u001b[38;5;241m.\u001b[39mappend(candidates[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmetadatas\u001b[39m\u001b[38;5;124m\"\u001b[39m])\n",
      "File \u001b[0;32m~/anaconda3/envs/RAG/lib/python3.9/site-packages/langchain_community/vectorstores/chroma.py:700\u001b[0m, in \u001b[0;36mChroma.get\u001b[0;34m(self, ids, where, limit, offset, where_document, include)\u001b[0m\n\u001b[1;32m    697\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m include \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    698\u001b[0m     kwargs[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124minclude\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m include\n\u001b[0;32m--> 700\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_collection\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/anaconda3/envs/RAG/lib/python3.9/site-packages/chromadb/api/models/Collection.py:131\u001b[0m, in \u001b[0;36mCollection.get\u001b[0;34m(self, ids, where, limit, offset, where_document, include)\u001b[0m\n\u001b[1;32m    109\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Get embeddings and their associate data from the data store. If no ids or where filter is provided returns\u001b[39;00m\n\u001b[1;32m    110\u001b[0m \u001b[38;5;124;03mall embeddings up to limit starting at offset.\u001b[39;00m\n\u001b[1;32m    111\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    122\u001b[0m \n\u001b[1;32m    123\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    124\u001b[0m get_request \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_validate_and_prepare_get_request(\n\u001b[1;32m    125\u001b[0m     ids\u001b[38;5;241m=\u001b[39mids,\n\u001b[1;32m    126\u001b[0m     where\u001b[38;5;241m=\u001b[39mwhere,\n\u001b[1;32m    127\u001b[0m     where_document\u001b[38;5;241m=\u001b[39mwhere_document,\n\u001b[1;32m    128\u001b[0m     include\u001b[38;5;241m=\u001b[39minclude,\n\u001b[1;32m    129\u001b[0m )\n\u001b[0;32m--> 131\u001b[0m get_results \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_client\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    132\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcollection_id\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mid\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    133\u001b[0m \u001b[43m    \u001b[49m\u001b[43mids\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mget_request\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mids\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    134\u001b[0m \u001b[43m    \u001b[49m\u001b[43mwhere\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mget_request\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mwhere\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    135\u001b[0m \u001b[43m    \u001b[49m\u001b[43mwhere_document\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mget_request\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mwhere_document\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    136\u001b[0m \u001b[43m    \u001b[49m\u001b[43minclude\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mget_request\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43minclude\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    137\u001b[0m \u001b[43m    \u001b[49m\u001b[43msort\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    138\u001b[0m \u001b[43m    \u001b[49m\u001b[43mlimit\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mlimit\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    139\u001b[0m \u001b[43m    \u001b[49m\u001b[43moffset\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43moffset\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    140\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtenant\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mtenant\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    141\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdatabase\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdatabase\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    142\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    143\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_transform_get_response(\n\u001b[1;32m    144\u001b[0m     response\u001b[38;5;241m=\u001b[39mget_results, include\u001b[38;5;241m=\u001b[39mget_request[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124minclude\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[1;32m    145\u001b[0m )\n",
      "File \u001b[0;32m~/anaconda3/envs/RAG/lib/python3.9/site-packages/chromadb/api/rust.py:354\u001b[0m, in \u001b[0;36mRustBindingsAPI._get\u001b[0;34m(self, collection_id, ids, where, sort, limit, offset, page, page_size, where_document, include, tenant, database)\u001b[0m\n\u001b[1;32m    342\u001b[0m ids_amount \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlen\u001b[39m(ids) \u001b[38;5;28;01mif\u001b[39;00m ids \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;241m0\u001b[39m\n\u001b[1;32m    343\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mproduct_telemetry_client\u001b[38;5;241m.\u001b[39mcapture(\n\u001b[1;32m    344\u001b[0m     CollectionGetEvent(\n\u001b[1;32m    345\u001b[0m         collection_uuid\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mstr\u001b[39m(collection_id),\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    351\u001b[0m     )\n\u001b[1;32m    352\u001b[0m )\n\u001b[0;32m--> 354\u001b[0m rust_response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbindings\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    355\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mstr\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mcollection_id\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    356\u001b[0m \u001b[43m    \u001b[49m\u001b[43mids\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    357\u001b[0m \u001b[43m    \u001b[49m\u001b[43mjson\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdumps\u001b[49m\u001b[43m(\u001b[49m\u001b[43mwhere\u001b[49m\u001b[43m)\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mif\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mwhere\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01melse\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    358\u001b[0m \u001b[43m    \u001b[49m\u001b[43mlimit\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    359\u001b[0m \u001b[43m    \u001b[49m\u001b[43moffset\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01mor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    360\u001b[0m \u001b[43m    \u001b[49m\u001b[43mjson\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdumps\u001b[49m\u001b[43m(\u001b[49m\u001b[43mwhere_document\u001b[49m\u001b[43m)\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mif\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mwhere_document\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01melse\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    361\u001b[0m \u001b[43m    \u001b[49m\u001b[43minclude\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    362\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtenant\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    363\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdatabase\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    364\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    366\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m GetResult(\n\u001b[1;32m    367\u001b[0m     ids\u001b[38;5;241m=\u001b[39mrust_response\u001b[38;5;241m.\u001b[39mids,\n\u001b[1;32m    368\u001b[0m     embeddings\u001b[38;5;241m=\u001b[39mrust_response\u001b[38;5;241m.\u001b[39membeddings,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    373\u001b[0m     metadatas\u001b[38;5;241m=\u001b[39mrust_response\u001b[38;5;241m.\u001b[39mmetadatas,\n\u001b[1;32m    374\u001b[0m )\n",
      "\u001b[0;31mInternalError\u001b[0m: Database error: error returned from database: (code: 1) no such table: collections"
     ]
    }
   ],
   "source": [
    "for chunk_size in [1000,1250,1500,1750,2000]:  # Add other chunk sizes (in tokens) as needed\n",
    "\n",
    "    for llm_model in [\"gemini-2.0-flash\",\"gemma-3-27b-it\"]:\n",
    "\n",
    "        llm_reader = init_chat_model(llm_model, model_provider=\"google_genai\",temperature=0.5,max_output_tokens=1024)\n",
    "\n",
    "        for embedding_model in [\"avsolatorio/GIST-small-Embedding-v0\",\"Snowflake/snowflake-arctic-embed-s\",\"intfloat/e5-small-v2\"]:\n",
    "\n",
    "            name_model = embedding_model.split(\"/\")[1]\n",
    "            embedding_function = HuggingFaceEmbeddings(model_name=embedding_model)\n",
    "\n",
    "            for recursive_chunk in [False, True]:\n",
    "\n",
    "                for k in range(4,11):\n",
    "            \n",
    "                    settings_name = f\"chunk_{chunk_size}_reader-model_{llm_model}_emnedding_model_{name_model}recursive_{recursive_chunk}_k_{k}\"\n",
    "                    output_file_name = f\"./output/rag_{settings_name}.json\"\n",
    "\n",
    "                    print(f\"Running evaluation for {settings_name}:\")\n",
    "\n",
    "                    print(\"Loading knowledge base embeddings...\")\n",
    "\n",
    "                    db = load_embeddings(\n",
    "                        info_articles_final,\n",
    "                        chunk_size=chunk_size,\n",
    "                        embedding_model=embedding_function,\n",
    "                    )\n",
    "\n",
    "                    print(\"Running RAG...\")\n",
    "                    run_rag_tests(\n",
    "                        eval_dataset=eval_dataset,\n",
    "                        llm=llm_reader,\n",
    "                        database=db,\n",
    "                        output_file=output_file_name,\n",
    "                        verbose=False,\n",
    "                        test_settings=settings_name,\n",
    "                        recursive_chunk=recursive_chunk,\n",
    "                        num_docs_final=k\n",
    "                    )\n",
    "\n",
    "                    print(\"Running evaluation...\")\n",
    "                    evaluate_answers(\n",
    "                        output_file_name,\n",
    "                        llm_reader,\n",
    "                        llm_model,\n",
    "                        evaluation_prompt_template,\n",
    "                    )\n",
    "                    print(\"Removing database\")\n",
    "                    db.delete_collection()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "f523874a",
   "metadata": {},
   "outputs": [
    {
     "ename": "InternalError",
     "evalue": "Error getting collection: Database error: error returned from database: (code: 1) no such table: collections",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mInternalError\u001b[0m                             Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[19], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mdb\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msimilarity_search\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mEndometrium\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43mk\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m2\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/anaconda3/envs/RAG/lib/python3.9/site-packages/langchain_community/vectorstores/chroma.py:350\u001b[0m, in \u001b[0;36mChroma.similarity_search\u001b[0;34m(self, query, k, filter, **kwargs)\u001b[0m\n\u001b[1;32m    333\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21msimilarity_search\u001b[39m(\n\u001b[1;32m    334\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m    335\u001b[0m     query: \u001b[38;5;28mstr\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    338\u001b[0m     \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs: Any,\n\u001b[1;32m    339\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m List[Document]:\n\u001b[1;32m    340\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"Run similarity search with Chroma.\u001b[39;00m\n\u001b[1;32m    341\u001b[0m \n\u001b[1;32m    342\u001b[0m \u001b[38;5;124;03m    Args:\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    348\u001b[0m \u001b[38;5;124;03m        List[Document]: List of documents most similar to the query text.\u001b[39;00m\n\u001b[1;32m    349\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m--> 350\u001b[0m     docs_and_scores \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msimilarity_search_with_score\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    351\u001b[0m \u001b[43m        \u001b[49m\u001b[43mquery\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mk\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mfilter\u001b[39;49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mfilter\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\n\u001b[1;32m    352\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    353\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m [doc \u001b[38;5;28;01mfor\u001b[39;00m doc, _ \u001b[38;5;129;01min\u001b[39;00m docs_and_scores]\n",
      "File \u001b[0;32m~/anaconda3/envs/RAG/lib/python3.9/site-packages/langchain_community/vectorstores/chroma.py:440\u001b[0m, in \u001b[0;36mChroma.similarity_search_with_score\u001b[0;34m(self, query, k, filter, where_document, **kwargs)\u001b[0m\n\u001b[1;32m    438\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    439\u001b[0m     query_embedding \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_embedding_function\u001b[38;5;241m.\u001b[39membed_query(query)\n\u001b[0;32m--> 440\u001b[0m     results \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m__query_collection\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    441\u001b[0m \u001b[43m        \u001b[49m\u001b[43mquery_embeddings\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m[\u001b[49m\u001b[43mquery_embedding\u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    442\u001b[0m \u001b[43m        \u001b[49m\u001b[43mn_results\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mk\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    443\u001b[0m \u001b[43m        \u001b[49m\u001b[43mwhere\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mfilter\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    444\u001b[0m \u001b[43m        \u001b[49m\u001b[43mwhere_document\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mwhere_document\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    445\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    446\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    448\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m _results_to_docs_and_scores(results)\n",
      "File \u001b[0;32m~/anaconda3/envs/RAG/lib/python3.9/site-packages/langchain_core/utils/utils.py:54\u001b[0m, in \u001b[0;36mxor_args.<locals>.decorator.<locals>.wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m     48\u001b[0m     msg \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m     49\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mExactly one argument in each of the following\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     50\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m groups must be defined:\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     51\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m, \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;241m.\u001b[39mjoin(invalid_group_names)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     52\u001b[0m     )\n\u001b[1;32m     53\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(msg)\n\u001b[0;32m---> 54\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/anaconda3/envs/RAG/lib/python3.9/site-packages/langchain_community/vectorstores/chroma.py:157\u001b[0m, in \u001b[0;36mChroma.__query_collection\u001b[0;34m(self, query_texts, query_embeddings, n_results, where, where_document, **kwargs)\u001b[0m\n\u001b[1;32m    152\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mImportError\u001b[39;00m:\n\u001b[1;32m    153\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mImportError\u001b[39;00m(\n\u001b[1;32m    154\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCould not import chromadb python package. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    155\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mPlease install it with `pip install chromadb`.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    156\u001b[0m     )\n\u001b[0;32m--> 157\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_collection\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mquery\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    158\u001b[0m \u001b[43m    \u001b[49m\u001b[43mquery_texts\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquery_texts\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    159\u001b[0m \u001b[43m    \u001b[49m\u001b[43mquery_embeddings\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquery_embeddings\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    160\u001b[0m \u001b[43m    \u001b[49m\u001b[43mn_results\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mn_results\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    161\u001b[0m \u001b[43m    \u001b[49m\u001b[43mwhere\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mwhere\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    162\u001b[0m \u001b[43m    \u001b[49m\u001b[43mwhere_document\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mwhere_document\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    163\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    164\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/anaconda3/envs/RAG/lib/python3.9/site-packages/chromadb/api/models/Collection.py:219\u001b[0m, in \u001b[0;36mCollection.query\u001b[0;34m(self, query_embeddings, query_texts, query_images, query_uris, n_results, where, where_document, include)\u001b[0m\n\u001b[1;32m    185\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Get the n_results nearest neighbor embeddings for provided query_embeddings or query_texts.\u001b[39;00m\n\u001b[1;32m    186\u001b[0m \n\u001b[1;32m    187\u001b[0m \u001b[38;5;124;03mArgs:\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    205\u001b[0m \n\u001b[1;32m    206\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    208\u001b[0m query_request \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_validate_and_prepare_query_request(\n\u001b[1;32m    209\u001b[0m     query_embeddings\u001b[38;5;241m=\u001b[39mquery_embeddings,\n\u001b[1;32m    210\u001b[0m     query_texts\u001b[38;5;241m=\u001b[39mquery_texts,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    216\u001b[0m     include\u001b[38;5;241m=\u001b[39minclude,\n\u001b[1;32m    217\u001b[0m )\n\u001b[0;32m--> 219\u001b[0m query_results \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_client\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_query\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    220\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcollection_id\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mid\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    221\u001b[0m \u001b[43m    \u001b[49m\u001b[43mquery_embeddings\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquery_request\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43membeddings\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    222\u001b[0m \u001b[43m    \u001b[49m\u001b[43mn_results\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquery_request\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mn_results\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    223\u001b[0m \u001b[43m    \u001b[49m\u001b[43mwhere\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquery_request\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mwhere\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    224\u001b[0m \u001b[43m    \u001b[49m\u001b[43mwhere_document\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquery_request\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mwhere_document\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    225\u001b[0m \u001b[43m    \u001b[49m\u001b[43minclude\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquery_request\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43minclude\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    226\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtenant\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mtenant\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    227\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdatabase\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdatabase\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    228\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    230\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_transform_query_response(\n\u001b[1;32m    231\u001b[0m     response\u001b[38;5;241m=\u001b[39mquery_results, include\u001b[38;5;241m=\u001b[39mquery_request[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124minclude\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[1;32m    232\u001b[0m )\n",
      "File \u001b[0;32m~/anaconda3/envs/RAG/lib/python3.9/site-packages/chromadb/api/rust.py:493\u001b[0m, in \u001b[0;36mRustBindingsAPI._query\u001b[0;34m(self, collection_id, query_embeddings, n_results, where, where_document, include, tenant, database)\u001b[0m\n\u001b[1;32m    478\u001b[0m query_amount \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlen\u001b[39m(query_embeddings)\n\u001b[1;32m    479\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mproduct_telemetry_client\u001b[38;5;241m.\u001b[39mcapture(\n\u001b[1;32m    480\u001b[0m     CollectionQueryEvent(\n\u001b[1;32m    481\u001b[0m         collection_uuid\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mstr\u001b[39m(collection_id),\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    490\u001b[0m     )\n\u001b[1;32m    491\u001b[0m )\n\u001b[0;32m--> 493\u001b[0m rust_response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbindings\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mquery\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    494\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mstr\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mcollection_id\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    495\u001b[0m \u001b[43m    \u001b[49m\u001b[43mquery_embeddings\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    496\u001b[0m \u001b[43m    \u001b[49m\u001b[43mn_results\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    497\u001b[0m \u001b[43m    \u001b[49m\u001b[43mjson\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdumps\u001b[49m\u001b[43m(\u001b[49m\u001b[43mwhere\u001b[49m\u001b[43m)\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mif\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mwhere\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01melse\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    498\u001b[0m \u001b[43m    \u001b[49m\u001b[43mjson\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdumps\u001b[49m\u001b[43m(\u001b[49m\u001b[43mwhere_document\u001b[49m\u001b[43m)\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mif\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mwhere_document\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01melse\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    499\u001b[0m \u001b[43m    \u001b[49m\u001b[43minclude\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    500\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtenant\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    501\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdatabase\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    502\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    504\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m QueryResult(\n\u001b[1;32m    505\u001b[0m     ids\u001b[38;5;241m=\u001b[39mrust_response\u001b[38;5;241m.\u001b[39mids,\n\u001b[1;32m    506\u001b[0m     embeddings\u001b[38;5;241m=\u001b[39mrust_response\u001b[38;5;241m.\u001b[39membeddings,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    512\u001b[0m     distances\u001b[38;5;241m=\u001b[39mrust_response\u001b[38;5;241m.\u001b[39mdistances,\n\u001b[1;32m    513\u001b[0m )\n",
      "\u001b[0;31mInternalError\u001b[0m: Error getting collection: Database error: error returned from database: (code: 1) no such table: collections"
     ]
    }
   ],
   "source": [
    "db.similarity_search(\"Endometrium\",k=2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "8f6b57c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob\n",
    "\n",
    "outputs = []\n",
    "for file in glob.glob(\"./output/*.json\"):\n",
    "    output = pd.DataFrame(json.load(open(file, \"r\")))\n",
    "    output[\"settings\"] = file\n",
    "    outputs.append(output)\n",
    "result = pd.concat(outputs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "53ccea67",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "question",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "true_answer",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "generated_answer",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "retrieved_docs",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "test_settings",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "eval_score_gemini-2.0-flash",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "eval_feedback_gemini-2.0-flash",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "settings",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "4a991a48-5e67-49d1-8558-6ec42ab86a89",
       "rows": [
        [
         "0",
         "How does the FGB rs1800790A allele affect fibrinogen levels in F13A 34Val/Val wildtypes?",
         "In F13A 34Val/Val wildtypes, carriage of the FGB rs1800790A allele is related to significantly lower fibrinogen levels compared with FGB rs1800790GG wildtypes.",
         "In F13A 34Val/Val wildtypes, carriage of FGB rs1800790A was related to significantly lower fibrinogen levels (p < 0.01) (5). In F13A 34Val/Val wildtypes, FGB rs1800790G > A minor allele carriers exhibited significantly lower fibrinogen levels (2).",
         "['Christian Schwedler et al.,2021, DOI:https://doi.org/10.3390/life11121340\\nAs described previously, a strong correlation between fibrinogen level and fibrin clot density depends on the presence of the\\nF13A\\n34Val/Val wildtype [\\n7\\n]. Furthermore, reduced inflammatory activity in\\nFGB\\nrs1800790 A-allele carriers could be found only in\\nF13A\\n34Val/Val wildtypes, thus,\\nF13A\\n34Val/Val wildtype seems to be a necessary precondition for this effect [\\n9\\n]. We therefore tested the relationship between fibrinogen genotypes and fibrinogen levels in\\nF13A\\nVal34Leu defined strata (\\nTable 4\\n).\\nTable 4.\\nAssociation between fibrinogen genotypes and fibrinogen levels in dependence of\\nF13A\\nVal34Leu and\\nA2AP\\nArg6Trp genotypes.\\nFibrinogen Level [g/L], Mean ± SD\\nGenotype\\nWildtype\\nMinor Allele\\np\\nValue\\nFGB rs1800790G > A\\nF13A\\n34Val/Val\\n4.62 ± 1.23\\n4.45 ± 1.12\\n<0.01 *\\nF13A\\n34Leu\\n4.50 ± 1.18\\n4.58 ± 1.11\\n0.15 *\\nA2AP\\n6Arg/Arg\\n4.56 ± 1.25\\n4.44 ± 1.07\\n<0.04\\n#\\nA2AP\\n6Trp\\n4.52 ± 1.14\\n4.59 ± 1.17\\n0.22\\n#\\nFGA rs2070016T > C\\nF13A\\n34Val/Val\\n4.60 ± 1.26\\n4.38 ± 1.04\\n<0.004\\nF13A\\n34Leu\\n4.56 ± 1.16', 'Christian Schwedler et al.,2021, DOI:https://doi.org/10.3390/life11121340\\nF13A\\n34Leu this relation vanished. Heterogeneity for this effect between both\\nF13A\\nVal34Leu defined strata was statistically significant in univariate as well as in multivariate analyses (\\np\\nfor homogeneity: 0.019 and 0.002, respectively). A similar tendency could be found for\\nFGA\\nrs2070016T > C. In\\nF13A\\n34Val/Val wildtypes, carriage of the\\nFGA\\nrs2070016T > C minor allele was associated with lower fibrinogen levels compared with\\nFGA\\nrs2070016T > C wildtypes. In\\nF13A\\n34Leu carriers, this effect of\\nFGA\\nrs2070016T > C genotype on fibrinogen levels was considerably diminished. Heterogeneity between both\\nF13A\\nVal34Leu strata was not statistically significant. For\\nFGG\\nrs1049636T > C, minor allele carriage was robustly associated with higher fibrinogen levels in both\\nF13A\\nVal34Leu strata.\\nWe additionally evaluated whether\\nF13A\\nVal34Leu or\\nA2AP\\nArg6Trp would have an influence on fibrinogen levels. Neither for\\nF13A', 'Christian Schwedler et al.,2021, DOI:https://doi.org/10.3390/life11121340\\nF13A\\n34Val/Val wildtypes and/or\\nA2AP\\n6Arg/Arg wildtypes),\\nFGB\\nrs1800790G > A or\\nFGA\\nrs2070016T > C minor allele carriers exhibited significantly lower fibrinogen levels (\\nTable 4\\n) compared with the corresponding wildtypes. When comparing the relationship between\\nFGB\\nrs1800790G > A or\\nFGA\\nrs2070016T > C genotype and fibrinogen level in strata, defined by absence or presence of\\nF13A\\n34Leu, a significant heterogeneity was detectable (\\nTable 4\\n). The same holds true for strata defined by\\nA2AP\\n6Trp carriage (\\np\\n= 0.09) (\\nTable 4\\n).\\nAs reflected by the hypotheses of our study, we think the findings are plausible when considering the interactive effects of fibrinogen,\\nF13A\\nand\\nA2AP\\ngenetics, described previously [\\n9\\n,\\n10\\n,\\n11\\n,\\n27\\n]. Carriage of\\nFGB\\nrs1800790 A allele(s) or\\nFGA\\nrs2070016 C allele(s) is associated with significantly lower inflammatory activity, if\\nF13A\\n34Val/Val wildtypes and/or\\nA2AP', 'Christian Schwedler et al.,2021, DOI:https://doi.org/10.3390/life11121340\\n10\\n]. The results of these analyses are given in\\nTable 4\\n. The findings resembled those for\\nF13A\\nVal34Leu. In individuals predisposed for a higher antifibrinolytic capacity (\\nA2AP\\n6Arg/Arg wildtypes), carriage of the\\nFGB\\nrs1800790 A allele was related to lower fibrinogen levels. The same holds true for\\nFGA\\nrs2070016T > C. Here, the differences in fibrinogen levels were especially pronounced, and heterogeneity between both\\nA2AP\\nArg6Trp defined strata was significant (\\np\\nfor homogeneity: 0.005). For\\nFGG\\nrs1049636T > C, again, carriage of the minor allele was robustly associated with higher fibrinogen levels in both\\nA2AP\\nArg6Trp defined strata.', 'Christian Schwedler et al.,2021, DOI:https://doi.org/10.3390/life11121340\\nF13B\\n, factor XIII B-subunit,\\nA2AP\\n, α\\n2\\n-antiplasmin.\\nWhile carriage of the\\nFGB\\nrs1800790 A allele exhibited only a very small effect on fibrinogen concentration, carriage of the minor alleles of\\nFGA\\nrs2070016T > C or\\nFGG\\nrs1049636T > C influenced fibrinogen levels significantly. Interestingly, in our cohort, carriage of the minor alleles of\\nFGB\\nrs1800790G > A and\\nFGA\\nrs2070016T > C was related to lower fibrinogen levels compared with the corresponding wildtypes, and for\\nFGG\\nrs1049636T > C higher fibrinogen levels could be detected. For\\nF13A\\nVal34Leu,\\nF13B\\nHis95Arg and\\nA2AP\\nArg6Trp fibrinogen levels were equal for wildtypes and minor allele carriers.\\nAs expected, CRP elevation itself was very strongly associated with fibrinogen levels in the total study population (CRP <5 mg/L, 4.25 ± 1.06 mg/L, CRP ≥5 mg/L, 5.18 ± 1.14 mg/L,\\np\\n< 0.0001; CRP <10 mg/L, 4.41 ± 1.10 mg/L, CRP ≥10 mg/L, 5.52 ± 1.23 mg/L,\\np', 'Christian Schwedler et al.,2021, DOI:https://doi.org/10.3390/life11121340\\nF13A\\nVal34Leu and\\nA2AP\\nArg6Trp. Strong associations with fibrinogen levels could be found for\\nFGB\\nrs1800790G > A,\\nFGA\\nrs2070016T > C and\\nFGG\\nrs1049636T > C. For\\nFGB\\nrs1800790G > A and\\nFGA\\nrs2070016T > C, this relationship significantly depended on\\nF13A\\nVal34Leu and\\nA2AP\\nArg6Trp genotypes. Specifically, in\\nF13A\\n34Val/Val wildtypes, carriage of\\nFGB\\nrs1800790A was related to significantly lower fibrinogen levels compared with\\nFGB\\nrs1800790GG wildtypes (\\np\\n< 0.01). For\\nA2AP\\n6Arg/Arg wildtypes, a comparable relationship could be found (\\np\\n< 0.04). As these genotype constellations related to lower fibrinogen levels have previously been shown to be associated with reduced inflammatory activity, these findings suggest that the influence of fibrinogen,\\nF13A\\nand\\nA2AP\\ngenotypes on inflammation could affect the control of fibrinogen levels and vice versa.', 'Christian Schwedler et al.,2021, DOI:https://doi.org/10.3390/life11121340\\nWe tested the association of fibrinogen genotypes as well as of\\nF13A\\n,\\nF13B\\nand\\nA2AP\\nwith fibrinogen levels. The findings are given in\\nTable 2\\n.\\nTable 2.\\nAssociation of genotypes of fibrinogen,\\nF13A\\n,\\nF13B\\nand\\nA2AP\\npolymorphisms with fibrinogen levels.\\nFibrinogen Level [g/L], Mean ± SD\\nGenotype\\nWildtype\\nMinor Allele\\np\\nValue\\nFGB\\nrs1800790G > A\\n4.56 ± 1.21\\n4.51 ± 1.11\\n0.15\\nFGB\\nrs1800788C > T\\n4.53 ± 1.13\\n4.51 ± 1.21\\n0.41\\nFGA\\nrs6050A > G\\n4.53 ± 1.15\\n4.54 ± 1.18\\n0.59\\nFGA\\nrs2070016T > C\\n4.58 ± 1.21\\n4.42 ± 1.07\\n0.002\\nFGA\\nrs2070006G > A\\n4.51 ± 1.15\\n4.53 ± 1.17\\n0.58\\nFGG\\nrs1049636T > C\\n4.43 ± 1.16\\n4.63 ± 1.15\\n0.0002\\nF13A\\nVal34Leu\\n4.54 ± 1.18\\n4.53 ± 1.16\\n0.42\\nF13B\\nHis95Arg\\n4.53 ± 1.15\\n4.53 ± 1.21\\n0.51\\nA2AP\\nArg6Trp\\n4.51 ± 1.17\\n4.55 ± 1.15\\n0.21\\nOpen in a new tab\\nSD, standard deviation,\\nFGB\\n, β-fibrinogen,\\nFGA\\n, α-fibrinogen,\\nFGG\\n, γ-fibrinogen,\\nF13A\\n, factor XIII A-subunit,\\nF13B\\n, factor XIII B-subunit,\\nA2AP\\n, α\\n2\\n-antiplasmin.\\nWhile carriage of the\\nFGB']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "4",
         "The response accurately states that the FGB rs1800790A allele is related to significantly lower fibrinogen levels in F13A 34Val/Val wildtypes. It also mentions that FGB rs1800790G > A minor allele carriers exhibited significantly lower fibrinogen levels. The reference answer specifies the comparison is with FGB rs1800790GG wildtypes, which provides more context. The response is mostly correct and factual.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "1",
         "For which patient group might the ERA test be helpful?",
         "The ERA test may be helpful for women with suspected recurrent implantation failure.",
         "The ERA test may be helpful in women with suspected recurrent implantation failure (2).",
         "['Zion Ben Rafael et al.,2021, DOI:https://doi.org/10.1093/hropen/hoab010\\nMahajan (2015)\\n, the ERA test is probably applicable to a marginal number of patients. They have used the test in a group 186 patients, dividing them into two groups:', 'Rawad Bassil et al.,2018, DOI:https://doi.org/10.1007/s10815-018-1190-9\\nConclusions\\nPerforming the ERA test in a mock cycle prior to a FET does not seem to improve the ongoing pregnancy rate in good prognosis patients. Further large prospective studies are needed to elucidate the role of ERA testing in both good prognosis patients and in patients with recurrent implantation failure.', 'Rawad Bassil et al.,2018, DOI:https://doi.org/10.1007/s10815-018-1190-9\\nIn conclusion, the results of the present study do not support the use of the ERA test in a mock cycle prior to FET as a way to improve pregnancy outcomes in good prognosis patients. Further large prospective studies are needed to elucidate the role of ERA testing or histological endometrial dating in good prognosis patients. Additional large studies are required for patients with suspected recurrent implantation failure since the ERA test may be helpful in those women. These studies will aid in patient counseling and in tailoring the appropriate timing of embryo transfer in women undergoing infertility treatment.', 'Zion Ben Rafael et al.,2021, DOI:https://doi.org/10.1093/hropen/hoab010\\nCozzolino\\net al.\\n,\\n2020\\n) has concluded that the use of the ERA test in 488 patients who underwent preimplantation genetic testing for aneuploidy (PGT-A), ERA or both, could not show any advantage to the use of ERA test. In short, ERA did not appear to improve outcomes in a group of patients who failed three or more ETs with a total of three or five embryos, and strangely PGT-A per se was effective in those failing three ETs, but was ineffective after failures of transfer of five embryos (\\nCozzolino\\net al.\\n, 2020\\n). Similarly, others (\\nNeves\\net al.\\n, 2019\\n) in a multivariate analysis\\nhave confirmed that performing an ERA test did not influence the pregnancy rate, in the tested euploid ET arm and was even associated with a diminished pregnancy rate in the donor ET arm. Taken together, these observations coupled with the lack of intra-patient consistency, bare witness to the randomity of the current state of research in the field.', 'Zion Ben Rafael et al.,2021, DOI:https://doi.org/10.1093/hropen/hoab010\\nCozzolino\\net al.\\n,\\n2020\\n) has concluded that the use of the ERA test in 488 patients who underwent preimplantation genetic testing for aneuploidy (PGT-A), ERA or both, could not show any advantage to the use of ERA test. In short, ERA did not appear to improve outcomes in a group of patients who failed three or more ETs with a total of three or five embryos, and strangely PGT-A per se was effective in those failing three ETs, but was ineffective after failures of transfer of five embryos (\\nCozzolino\\net al.\\n, 2020\\n). Similarly, others (\\nNeves\\net al.\\n, 2019\\n) in a multivariate analysis\\nhave confirmed that performing an ERA test did not influence the pregnancy rate, in the tested euploid ET arm and was even associated with a diminished pregnancy rate in the donor ET arm. Taken together, these observations coupled with the lack of intra-patient consistency, bare witness to the randomity of the current state of research in the field.', 'Rawad Bassil et al.,2018, DOI:https://doi.org/10.1007/s10815-018-1190-9\\nThis single-center retrospective cohort study included 53 consecutive patients admitted to our IVF unit, between April 2016 and March 2017 for a mock cycle prior their frozen day-5 embryo (blastocyst) transfer cycle that included an endometrial biopsy for the ERA test.\\nAll the patients in the clinic, during this period, were offered to undergo ERA during a mock cycle, but only 53 patients decided to perform the ERA test.', 'Rawad Bassil et al.,2018, DOI:https://doi.org/10.1007/s10815-018-1190-9\\n9\\n]. This test is done by obtaining endometrial biopsy samples on day LH+ 7 in a natural cycle or on the 6th day of progesterone supplementation during an HRT cycle. Results are expressed as pre-receptive, receptive, or post-receptive. If the result is non-receptive, the embryo replacement timing may be adjusted in a subsequent cycle enabling personalized embryo transfer [\\n10\\n]. The ERA test is offered commercially but appears to be based on a relatively small number of samples.\\nThe aim of the present study was to determine the percentage of good prognosis (0–2 prior FETs) infertility patients who were determined to have a non-receptive endometrium according to the ERA test and to examine whether adjusting the suggested day of transfer according to the ERA test increases the pregnancy rate compared to a similar group of women without ERA testing.']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response accurately identifies the patient group for whom the ERA test might be helpful, mirroring the reference answer. The inclusion of \"(2)\" at the end is irrelevant to the correctness of the answer but does not detract from its accuracy.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "2",
         "What kind of values does the Color Pathway tool in KEGG Mapper accept?",
         "The Color Pathway tool accepts numerical values.",
         "The Color Pathway tool in KEGG Mapper accepts numerical values associated with genes (Document 1, 6).",
         "['Minoru Kanehisa et al.,2011, DOI:https://doi.org/10.1093/nar/gkr988\\nKEGG Mapper\\nKEGG PATHWAY, KEGG BRITE and KEGG MODULE constitute the KEGG reference knowledge base for biological interpretation of molecular data sets, especially large-scale data sets generated by high-throughput experimental technologies. This is accomplished by the process of KEGG mapping, which is to map, for example, a genomic content of genes to generate organism-specific versions of pathways as mentioned above or a transcriptomic content of genes to see which parts of pathways are up/down regulated. The mapping process is considered a set operation between the query data set and the target data of the KEGG reference knowledge base.\\nKEGG Mapper is the user interface for KEGG mapping. It currently consists of seven tools as shown in\\nTable 3', 'Minoru Kanehisa et al.,2011, DOI:https://doi.org/10.1093/nar/gkr988\\nSearch Module\\nObjects\\nKEGG MODULE database\\nOpen in a new tab\\nFigure 1.\\nOpen in a new tab\\nThe Color Pathway tool in KEGG Mapper accepts numerical values associated with genes, which can be displayed by color-shading or 3D mapping. Here somatic mutation frequency observed in chronic myeloid leukemia (obtained from the COSMIC database,\\nhttp://www.sanger.ac.uk/genetics/CGP/cosmic/\\n) is shown on top of the global cancer pathway map (hsa05200).\\nGenome comparison and combination\\nFor the organisms (GENES, DGENES and EGENES) and environmental samples (MGENES) available in KEGG, it is now possible to map multiple data sets against KEGG pathway maps and BRITE functional hierarchies. The user interface may be found in the KEGG GENOME page, and the result is displayed using multiple color coding. This feature can be used, for example, to compare metabolic capabilities of different organisms (see\\nFigure 2\\nfor human and\\nEscherichia coli', 'Minoru Kanehisa et al.,2011, DOI:https://doi.org/10.1093/nar/gkr988\\nspecific pathways in pink, and shared pathways in blue or split coloring.\\nKnowledge base extension\\nThe query data in KEGG Mapper is usually raw and uninterpreted data, but the KEGG mapping set operation may also be applied to well curated data sets. In fact it was the original concept of the KEGG project to automatically generate organism-specific pathways through the set operation between manually annotated genome data and manually created pathway maps. We have extended this approach as more data on diseases and drugs are accumulated in KEGG, effectively enhancing the KEGG knowledge base and enabling KEGG Mapper and other applications to easily integrate disease and drug information.', 'Minoru Kanehisa et al.,2016, DOI:https://doi.org/10.1093/nar/gkw1092\\n10\\n).\\nHIGHER-LEVEL FUNCTIONS\\nImprovement of PATHWAY database\\nThe KEGG PATHWAY database has been and will continue to be the main database in KEGG. It consists of manually drawn reference pathway maps together with organism-specific pathway maps that are computationally generated by matching KO assignments in the genome with reference pathway maps. In regular maps rectangular nodes are linked to KOs and the matching is shown by coloring. For example, the organism-specific glycolysis pathway hsa00010 with green coloring is generated from the hsa (\\nHomo sapiens\\n) genome and the reference pathway map00010. Table\\n6', 'Minoru Kanehisa et al.,2011, DOI:https://doi.org/10.1093/nar/gkr988\\nThe KEGG pathway maps are graphical diagrams representing knowledge on molecular interaction and reaction networks for metabolism, genetic information processing, environmental information processing, cellular processes, organismal systems, human diseases and drug development. Each map is manually drawn with in-house software called KegSketch, summarizing experimental evidence in published literature. Basic graphics objects in the KEGG pathway maps are: (i) boxes for KEGG Orthology (KO) groups identified by K numbers, (ii) circles for small molecules identified by C numbers, glycans identified by G numbers, and other molecules, and (iii) lines for KO groups in global metabolism maps. Boxes in regular metabolic maps and lines in global metabolic maps, both for KO groups, are also linked to enzymes identified by EC numbers and reactions identified by R numbers', 'Minoru Kanehisa et al.,2016, DOI:https://doi.org/10.1093/nar/gkw1092\\n7\\nshows the current list of KEGG mapping tools including the new Search Disease tool. Two pathway mapping tools, Search Pathway and Search&Color Pathway, were made available from the beginning of the KEGG project, and they are still the most widely used. The mapping of genes and proteins can be made either in reference mode using KOs or in organism-specific mode using gene identifiers, such as human gene identifiers used to search against human pathway maps. The three tools, Reconstruct Pathway, Reconstruct Brite and Reconstruct Module, accept KOs only. They are linked from the BlastKOALA and GhostKOALA servers for interpretation of the KO assignment results.\\nTable 7. KEGG Mapper tools.\\nTool\\nQuery dataset\\nDatabase\\nSearch Pathway\\nKOs, gene identifiers, C numbers, etc.\\nPATHWAY\\nSearch&Color Pathway\\nKOs, gene identifiers, C numbers, etc.\\nPATHWAY\\nColor Pathway\\nKOs, gene identifiers\\nsingle KEGG pathway map\\nColor Pathway WebGL\\nKOs, gene identifiers\\nsingle KEGG pathway map\\nSearch Brite', 'Minoru Kanehisa et al.,2011, DOI:https://doi.org/10.1093/nar/gkr988\\nTable 3\\n. For the three basic tools (Search Pathway, Search Brite and Search Module) the query data set is a collection of molecular objects (genes, proteins, small molecules, etc.), and the user would simply query the presence or absence of objects, resulting in pathway and ontology enrichment. In the advanced tools (Search&Color Pathway, Search&Color Brite, Color Pathway and Join Brite) the query data can consist of molecular objects and their attributes; for example, genes with up/down expression levels and genes with disease names known to be associated. The attributes need to be processed and color-coded when using the Search&Color Pathway and Search&Color Brite tools. The Color Pathway tool accepts numerical values, which may be displayed by color shading or by overlaying 3D bar graphs on top of the KEGG pathway map.\\nFigure 1']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response accurately states that the Color Pathway tool accepts numerical values, which aligns with the reference answer. The inclusion of document references is extra information but does not detract from the correctness.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "3",
         "What is the implantation potential of an euploid blastocyst?",
         "Once an euploid blastocyst is identified, its implantation potential is independent of maternal age, at 45-50%.",
         "An euploid blastocyst transfer results in a healthy live birth in approximately 50% of cases, regardless of the woman's age (1). Once an euploid blastocyst is identified, its implantation potential is independent of maternal age (45-50%) (3).",
         "['Andres Reig et al.,2020, DOI:https://doi.org/10.1007/s10815-020-01739-0\\nA prior study evaluating the effect of maternal age at the time of oocyte retrieval on implantation potential of euploid blastocysts concluded that age had no impact on the said potential after adjusting for embryo morphology [18]. Of note, of the 785 transfers included in the analysis, approximately 10% were double embryo transfers. This was appropriately controlled for in the statistical analysis, but it is important to note that IR was defined as percentage of embryos transferred, and not of transfers performed. This publication contributed very valuable insight, particularly regarding the weight of morphology on implantation potential of euploid embryos.', 'Danilo Cimadomo et al.,2018, DOI:https://doi.org/10.3389/fendo.2018.00327\\na euploid blastocyst transfer results into a healthy live birth in\\n~50% of cases, regardless woman age . To conclude, evidence-based data should always guide the\\ncounseling and the patients should be scrupulously informed about\\ntheir estimated chance to conceive, especially if older than 35.\\nIndeed, 35  years should be the lowest age-threshold to define\\nAMA and 45 years should be considered the highest age-threshold\\nto undergo IVF with own eggs, at least according to the latest\\npublished report .', 'L Aghajanova et al.,2008, DOI:https://doi.org/10.1016/j.semcdb.2007.10.008\\n1. Challenges and approaches in studying the process of human implantation\\nClinical implications\\nSuccessful embryo implantation is a crucial event in natural and assisted human reproduction. Blastocyst implantation is a dynamic process, involving embryo apposition, attachment to the maternal endometrial epithelium, and invasion into the endometrial stroma [1]. With\\nin vitro fertilization', 'Danilo Cimadomo et al.,2018, DOI:https://doi.org/10.3389/fendo.2018.00327\\nAMA patients. In fact, the goal of ART is to achieve the birth\\nof a healthy child minimizing the risks for the patient, and this\\nis particularly true in AMA when the incidence of aneuploidies\\ndramatically increases . This approach, by avoiding the\\ntransfer of aneuploid blastocysts and their related risks (i.e.,\\nimplantation failures, miscarriages, and affected child), might\\nresult in an increased efficiency of each IVF treatment .   Importantly, once an euploid blastocyst is identified, its implantation potential is independent of maternal age (45-50%),\\nthere by allowing the adoption of a single-embryo-transfer policy\\nalso in AMA patients, concurrently lowering the risk for multiple\\ngestations and their related obstetrical/perinatal risks .  Soon, the implementation of -omic sciences and the pursuit of\\nnon-invasiveness and higher cost-effectiveness in this field may\\nconverge and bring about intriguing avant-gardes to further', 'Andres Reig et al.,2020, DOI:https://doi.org/10.1007/s10815-020-01739-0\\nOur research demonstrates a declining implantation potential with age after controlling for aneuploidy, as well as several other associated factors. Further research is needed to clarify the reasons behind the age-related decline in implantation of euploid embryos. Similarly, studies analyzing the potential effect of age on the ability of a uterus to carry a pregnancy could be beneficial, but should be undertaken in the setting of known euploid embryo transfers.', 'Andres Reig et al.,2020, DOI:https://doi.org/10.1007/s10815-020-01739-0\\nOur research demonstrates a declining implantation potential with age after controlling for aneuploidy, as well as several other associated factors. Further research is needed to clarify the reasons behind the age-related decline in implantation of euploid embryos. Similarly, studies analyzing the potential effect of age on the ability of a uterus to carry a pregnancy could be beneficial, but should be undertaken in the setting of known euploid embryo transfers.', 'L Aghajanova et al.,2008, DOI:https://doi.org/10.1016/j.semcdb.2007.10.008\\nin vitro fertilization\\n(IVF), implantation failure can occur due to several factors [2], including embryonic defects such as chromosomal abnormalities, which are the most common cause of implantation failure [3, 4]. Another widely acknowledged barrier to successful blastocyst implantation and, therefore, successful treatment of infertile women with implantation failure, is an inappropriately developed endometrium. It is well established that embryos cannot implant in a poorly matured endometrium [5], and this may be responsible for low implantation rates with transfer of “good quality” embryos. The success of embryonic implantation further relies upon cross-talk between the embryo and a receptive endometrium [5]. Optimization of IVF results are important for many reasons, and achieving optimal implantation is even more important in the setting of single embryo transfer, a common practice aimed to avoid multiple pregnancies.\\nThe implantation window and challenges of studying it']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "4",
         "Feedback: The response provides a similar answer to the reference, mentioning that the implantation potential of a euploid blastocyst is independent of maternal age and providing a percentage. However, the response includes additional information about live birth rates that is not directly relevant to the question about implantation potential.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "4",
         "What does the PRISMA 2020 statement reflect?",
         "The PRISMA 2020 statement reflects advances in methods to identify, select, appraise, and synthesise studies.",
         "The PRISMA 2020 statement reflects advances in methods to identify, select, appraise, and synthesise studies (2).",
         "['Matthew J Page et al.,2021, DOI:https://doi.org/10.1136/bmj.n71\\n. We discussed proposed  content and wording of the PRISMA 2020 statement,  as informed by the review and survey results, at a  21-member, two-day, in-person meeting in September  2018 in Edinburgh, Scotland. Throughout 2019 and  2020, we circulated an initial draft and five revisions  of the checklist and explanation and elaboration paper  to co-authors for feedback. In April 2020, we invited  22 systematic reviewers who had expressed interest in  providing feedback on the PRISMA 2020 checklist to  share their views (via an online survey) on the layout  and terminology used in a preliminary version of the  checklist. Feedback was received from 15 individuals  and considered by the first author, and any revisions  deemed necessary were incorporated before the final  version was approved and endorsed by all co-authors.', 'Matthew J Page et al.,2021, DOI:https://doi.org/10.1136/bmj.n71\\nThe PRISMA 2020 statement\\nScope of the guideline\\nThe PRISMA 2020 statement has been designed\\nprimarily for systematic reviews of studies that\\nevaluate the effects of health interventions, irrespective\\nof the design of the included studies. However, the\\nchecklist items are applicable to reports of systematic\\nreviews evaluating other interventions (such as social\\nor educational interventions), and many items are\\napplicable to systematic reviews with objectives other\\nthan evaluating interventions (such as evaluating\\naetiology, prevalence, or prognosis). PRISMA 2020\\nis intended for use in systematic reviews that include\\nsynthesis (such as pairwise meta-analysis or other\\nstatistical synthesis methods) or do not include\\nsynthesis (for example, because only one eligible study\\nis identified). The PRISMA 2020 items are relevant for\\nmixed-methods systematic reviews (which include\\nquantitative and qualitative studies), but reporting\\nguidelines addressing the presentation and synthesis', 'Matthew J Page et al.,2021, DOI:https://doi.org/10.1136/bmj.n71\\n.n71 Accepted: 4 January 2021 SUMMARY POINTS To ensure a systematic review is valuable to users, authors should prepare a  transparent, complete, and accurate account of why the review was done, what  they did, and what they found The PRISMA 2020 statement provides updated reporting guidance for systematic  reviews that reflects advances in methods to identify, select, appraise, and  synthesise studies The PRISMA 2020 statement consists of a 27-item checklist, an expanded  checklist that details reporting recommendations for each item, the PRISMA  2020 abstract checklist, and revised flow diagrams for original and updated  reviews We anticipate that the PRISMA 2020 statement will benefit authors, editors, and  peer reviewers of systematic reviews, and different users of reviews, including  guideline developers, policy makers, healthcare providers, patients, and other  stakeholders Protected by copyright, including for uses related to text and data mining, AI training, and similar', 'Matthew J Page et al.,2021, DOI:https://doi.org/10.1136/bmj.n71\\nprisma-statement.org/). To enhance accessibility of  PRISMA 2020, several translations of the guideline are  under way (see available translations at the PRISMA  statement website). We encourage journal editors and  publishers to raise awareness of PRISMA 2020 (for  example, by referring to it in journal “Instructions to  authors”), endorsing its use, advising editors and peer  reviewers to evaluate submitted systematic reviews  against the PRISMA 2020 checklists, and making  changes to journal policies to accommodate the new  reporting recommendations. We recommend existing  PRISMA extensions 47 49 50 51-53 71 72 be updated to reflect  PRISMA 2020 and advise developers of new PRISMA  extensions to use PRISMA 2020 as the foundation  document.', 'Matthew J Page et al.,2021, DOI:https://doi.org/10.1136/bmj.n71\\nWe anticipate that the PRISMA 2020 statement  will benefit authors, editors, and peer reviewers of  systematic reviews, and different users of reviews,  including guideline developers, policy makers,  healthcare providers, patients, and other stakeholders.  Ultimately, we hope that uptake of the guideline will  lead to more transparent, complete, and accurate  reporting of systematic reviews, thus facilitating  evidence based decision making.', 'Matthew J Page et al.,2021, DOI:https://doi.org/10.1136/bmj.n71\\nUse of PRISMA 2020 has the potential to benefit  many stakeholders. Complete reporting allows  readers to assess the appropriateness of the methods,  and therefore the trustworthiness of the findings.  Presenting and summarising characteristics of studies  contributing to a synthesis allows healthcare providers  and policy makers to evaluate the applicability of the  findings to their setting. Describing the certainty in the  body of evidence for an outcome and the implications  of findings should help policy makers, managers,  and other decision makers formulate appropriate  recommendations for practice or policy. Complete  reporting of all PRISMA 2020 items also facilitates  replication and review updates, as well as inclusion of  systematic reviews in overviews (of systematic reviews)  and guidelines, so teams can leverage work that is  already done and decrease research waste', 'Matthew J Page et al.,2021, DOI:https://doi.org/10.1136/bmj.n71\\nThe PRISMA 2020 statement (including the checklists,\\nexplanation and elaboration, and flow diagram)\\nreplaces the PRISMA 2009 statement, which should\\nno longer be used. Box 2 summarises noteworthy\\nchanges from the PRISMA 2009 statement. The\\nPRISMA 2020 checklist includes seven sections with\\n27 items, some of which include sub-items (table 1).\\nA checklist for journal and conference abstracts for\\nsystematic reviews is included in PRISMA 2020. This\\nabstract checklist is an update of the 2013 PRISMA\\nfor Abstracts statement, 54 reflecting new and modified\\nProtected by copyright, including for uses related to text and data mining, AI training, and similar technologies.\\n. by guest on 17 August 2025 https://www.bmj.com/Downloaded from 29 March 2021. 10.1136/bmj.n71 on BMJ: first published as\\nRESEARCH METHODS AND REPORTING the bmj | BMJ 2021;372:n71 | doi: 10.1136/bmj.n71 3\\ncontent in PRISMA 2020 (table 2). A template PRISMA\\nflow diagram is provided, which can be modified']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response accurately reflects the content of the reference answer, stating that the PRISMA 2020 statement reflects advances in methods to identify, select, appraise, and synthesise studies. The inclusion of \"(2)\" at the end doesn't detract from the accuracy of the statement.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "5",
         "What percentage of individuals with Mendelian disorders have pathology contributed by de novo variants?",
         "De novo variants contribute pathology to 40% of individuals with Mendelian disorders.",
         "De novo variants contribute pathology to 40% of individuals with Mendelian disorders [0].",
         "['Xiang Yang et al.,2018, DOI:https://doi.org/10.1007/s10815-018-1349-4\\n. This is especially important since de novo variants contribute pathology to 40% of individuals with Mendelian disorders [', 'E. Mossotto et al.,2019, DOI:https://doi.org/10.1186/s12859-019-2877-3\\n. This remains the case when studying rare Mendelian disease where single gene mutations imparting severe consequence are expected to represent the most extreme set of deleterious variants. In contrast to rare diseases, common genetic diseases such as ischemic heart disease, asthma, inflammatory bowel disease (IBD) or Alzheimer ’disease are caused by the combined action of multiple genetic variants each differentially impacting risk and disease severity while working in combination with environmental exposures [26]. Collectively, common diseases impose an enormous economic burden and arguably have the greatest unmet need for diagnosis and stratified treatment [ 27]. The set of genes and variants imparting increased susceptibility vary from one patient to the next even when clinical presentation and molecular pathology appear indistinct', 'Yaping Yang et al.,2014, DOI:https://doi.org/10.1001/jama.2014.14601\\n1\\n; however, fundamental questions remained unanswered. The robustness of the 25% frequency rate for attaining a molecular diagnosis, the contribution of rare variants, modes of inheritance in the patient population, and the precise rate at which rare genetic events such as mosaicism, multiple loci with contributing mutations, and new mutations contribute to disease remained to be established. Refinement of the coupling between clinical data and molecular interpretation is of particular interest because current methods include considerable expert human involvement and are not readily scalable without further automation. Knowledge of pathogenic variation in an ever-increasing number of Mendelian disease genes is growing,\\n8\\nas well as an increasing understanding of tolerated loss of function mutations in healthy controls.\\n9\\nThis study reports findings from clinical whole-exome sequencing evaluations for 2000 consecutive patients.', 'Feixiong Cheng et al.,2021, DOI:https://doi.org/10.1038/s41588-020-00774-y\\nInterpretation of the clinical pathogenic effects of variants is crucial for the advancement of precision medicine. However, our ability to understand the functional and biological consequences of genetic variants identified by human genome sequencing projects is limited. Many computational approaches can identify only a small proportion of pathogenic variants with the high confidence required in clinical settings. Human genome sequencing studies have reported potential mutation-disease associations with the functional regions altered by somatic mutations, such as molecular drivers in cancers.\\n1\\n,\\n2\\nHowever, many important issues in the field remain unclear, including the phenotypic consequences of different mutations within the same gene and the same mutation across different cell types.\\nRecent efforts using systematic analyses of 1,000–3,000 missense mutations in Mendelian disorders\\n3\\n,\\n4\\nand ~2,000\\nde novo\\nmissense mutations in developmental disorders\\n5', 'Marzia Antonella Scelsi et al.,2021, DOI:https://doi.org/10.1371/journal.pcbi.1008517\\n. These types of rare variants have the possibility to exercise a large effect on disease risk and often show a Mendelian inheritance pattern, thus being at the core of familial forms of many disorders; prominent examples are rare variants in APP, PSEN1, and PSEN2 in familial AD [5] or variants in MAPT, GRN and C9orf72 in frontotemporal dementia (FTD) [6]. The decreasing costs for NGS have enabled the establishment of large whole genome (WGS) or whole exome (WES) sequencing studies such as the AD Sequencing Project (ADSP) [7]; still the largest studies are two orders of magnitude smaller than the largest GWAS (i.e., 10,000s vs a million participants). Moreover, by definition, rare variants are not frequent in the population and it is unlikely for the same rare variant to be shared by many subjects in a study with limited sample size. Therefore, the resulting data matrix of subjects × rare variants is sparse', 'Philipp Rentzsch et al.,2018, DOI:https://doi.org/10.1093/nar/gky1016\\n59\\n), LINSIGHT (\\n60\\n) and CDTS (\\n61\\n). As a general statement, we believe that CADD and tools like it that: (i) integrate many correlated genomic annotations in a principled fashion; (ii) rely on large training datasets to minimize the risk of overfitting; and (iii) avoid curated sets of pathogenic and benign variants during training, represent the best path forward for predicting the relative pathogenicity or functional importance of human genetic variants on a genome-wide basis.\\nAs genomic annotations grow in depth and breadth, CADD and CADD-inspired variant scores will continue to improve and provide utility across a wide range of analytical scenarios. While this is particularly true for studies of Mendelian disease, many complex-trait, comparative genomic, population genetic and functional genomic studies are likely to also benefit from current and future versions of CADD and related frameworks.', 'Abhishek Niroula et al.,2019, DOI:https://doi.org/10.1371/journal.pcbi.1006481\\nNext Generation Sequencing (NGS) is widely used in clinical diagnosis as well as in population genetics to investigate patterns of genetic variants in healthy individuals. The large numbers of variants, millions per genome in comparison to reference sequences, pose challenges for detecting disease-causing variants. There are on average about 10,000 variants per genome that cause amino acid substitutions [1]. Several databases enable annotation of disease relevance of variants and frequencies among healthy individuals. These include numerous locus specific variation databases (LSDBs) that are curated by experts in the genes and diseases. While LSDBs typically concentrate on individual genes and proteins or diseases, the general databases have much wider scope such as ClinVar [2], Online Mendelian Inheritance in Man (OMIM) [3] and the UniProt Knowledgebase (UniProtKB) [4]']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response is completely correct, accurate, and factual, mirroring the reference answer exactly.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "6",
         "What are some exogenous factors that can lead to primary ovarian insufficiency?",
         "Viral infections, metabolomic dysfunction, autoimmune diseases, and environmental or iatrogenic factors.",
         "Viral infections, metabolomic dysfunction, autoimmune diseases, and environmental or iatrogenic factors can lead to primary ovarian insufficiency (Document 0).",
         "['Cristina Fortuño et al.,2014, DOI:https://doi.org/10.1007/s10815-014-0342-9\\nPrimary ovarian insufficiency is one of the main causes of female infertility owing to an abnormal ovarian reserve. Its relevance has increased in more recent years due to the fact that age of motherhood is being delayed in developed countries, with the risk of having either primary ovarian insufficiency or less chances of pregnancy when women consider the option of having their first baby. Several exogenous factors can lead to this event, such us viral infections, metabolomic dysfunction, autoimmune diseases, and environmental or iatrogenic factors, although in most cases the mechanism that leads to the disorder is unknown. Genetic factors represent the most commonly identified cause and the impact of sex chromosome abnormalities (e.g., Turner syndrome or X structural abnormalities), autosomal and X-linked mutations on the genesis of primary ovarian insufficiency has also been well described. Yet in most cases, the genetic origin remains unknown and there are multiple candidate genes', 'Anna Calik-Ksepka et al.,2018, DOI:https://doi.org/10.5114/pm.2018.78560\\nPremature ovarian insufficiency (POI) affects approximately 1% of women under the age of 40 years  and is associated with female infertility [1]. The symptoms are similar to those of menopause: oligo/amenorrhea, hypoestrogenism, elevated level of gonadotropins, and diminished number of follicles within the  ovaries. Although the causes of POI are largely unknown, genetic defects such as Turner syndrome and  fragile X syndrome premutation, environmental factors,  and autoimmune diseases seem to play an important  role in the aetiopathology. Primary ovarian insufficiency is generally diagnosed  as part of the evaluation for frequent gynaecological  complaints, such as irregular menses, so the diagnosis  usually comes unexpectedly', 'Monica Malheiros França et al.,2019, DOI:https://doi.org/10.1210/jendso/bvz037\\n1. Definition\\nPrimary ovarian insufficiency (POI) is defined by the depletion of ovary follicles, leading to infertility before the age of 40 years [\\n1\\n]. This condition is characterized by the cessation of menses (amenorrhea or oligomenorrhea) for at least 4 months, increased gonadotropin levels (FSH > LH), and hypoestrogenism [\\n2\\n].\\nIn 1942, Albright and colleagues [\\n3\\n] reported the first case of primary ovarian insufficiency. There is no consensus regarding the name of this disorder. On the one hand, the European Society of Human Reproduction and Embryology guidelines [\\n2\\n] recommend the term “premature ovarian insufficiency” for describing this disorder in research and clinical practice [\\n2\\n]. On the other hand, the American College of Obstetricians and Gynecologists (ACOG) committee is in favor of “primary ovarian insufficiency” [\\n1', 'Cristina Fortuño et al.,2014, DOI:https://doi.org/10.1007/s10815-014-0342-9\\nPrimary ovarian insufficiency (POI) is a condition caused by the absence, non functionality or early depletion of the ovarian reserve that leads to infertility, instead of the gradual process of follicular atresia, typical of fertile women, until the menopause is reached. It is characterized by primary or secondary amenorrhea for at least 4\\xa0months in women under the age of 40, menopausal serum FSH levels (above 40\\xa0IU/L) obtained on two occasions at least 1\\xa0month apart and low estradiol levels (below 50\\xa0pg/mL) [\\n1\\n]. Another indicator is the anti-Müllerian hormone (AMH), whose serum levels can help assess the state of follicular senescence, which is a possible predictor of risk for POI [\\n2\\n]. However, there are presently no single screening tests that can predict a woman’s reproductive lifespan [\\n3', 'Alba M. Polonio et al.,2021, DOI:https://doi.org/10.3389/fendo.2020.626322\\nIn humans, oocyte development begins during fetal life and follicle pool reaches its maximum at 16/ 20 weeks of fetal development . Follicular decline is initiated before birth, so that, at the time of delivery, only about 1 million of follicles remain in the ovary of the baby. By the time of menarche, each ovary contains about 400,000 follicles and ovarian reserve continues decreasing as women age . Thereby, the decline in oocyte quantity and quality during women reproductive life is a physiological process; however, in some women, ovary deterioration occurs in an abrupt way and they become prematurely infertile. Primary Ovarian Insufficiency (POI), also known as Premature Ovarian Failure (POF) or premature menopause, is a reproductive disorder, characterized by oligo-amenorrhea and high levels of serum FSH, leading to a cessation of ovarian function before the age of 40 . This condition affects 1% of women under the age of 40 years, and 1 out of 250 women under the age of 35 years', 'Michal Kirshenbaum et al.,2019, DOI:https://doi.org/10.1007/s10815-019-01572-0\\nPurpose\\nPrimary ovarian insufficiency (POI) represents ovarian dysfunction related to very early aging of the ovaries. While the cause of POI in a majority of clinical cases remains undefined, autoimmunity is responsible for approximately 4–30% of POI cases. In the present paper, we aim to provide a critical appraisal and update review on the role of autoimmunity in POI patients.\\nMethods\\nA literature review was conducted for all relevant articles reporting on POI and autoimmunity. PubMed/MEDLINE and the Cochrane library were searched for the best available evidence on this topic.\\nResults', 'Anlu Chen et al.,2018, DOI:https://doi.org/10.1093/hmg/ddy098\\nPrimary ovarian insufficiency (POI) is defined by the loss or dysfunction of ovarian follicles associated with amenorrhea before the age of 40 (\\n1\\n). POI is a major cause of female infertility with a prevalence greater than 1%. There is a strong genetic component to the development of POI, both in the form of monogenic and multigenic disorders, however, in most cases the genetic etiology of POI remains unclear (\\n2\\n). Among the most common genetic defects associated with POI are X chromosome defects, which collectively account for approximately 10–25% of POI cases (\\n2\\n). These include Turner’s syndrome, Triple X syndrome and Fragile X syndrome. A number of monogenic disorders resulting in POI have also been identified, including those with variants in\\nBMP15\\nand\\nPGRMC1\\n, both located on Chromosome X, as well as those with variants in\\nGDF9\\n,\\nFOXO3\\n,\\nFIGLA\\nand\\nNR5A1\\n, among others, that are autosomal (\\n2']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response lists viral infections, metabolomic dysfunction, autoimmune diseases, and environmental or iatrogenic factors, which are all exogenous factors that can lead to primary ovarian insufficiency. The response is completely correct, accurate, and factual.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "7",
         "What is recommended when endometrial receptiveness is seriously compromised by a controlled ovarian stimulation protocol?",
         "Fresh embryo replacement should be cancelled, the embryo frozen, and thawed embryo replacement should be performed under natural cycles.",
         "It is recommended that fresh embryo replacement should be canceled, the embryo frozen, and thawed embryo replacement should be performed under natural cycles when the receptiveness of the endometrium is seriously compromised by the COS protocol (4).",
         "['Delphine Haouzi et al.,2009, DOI:https://doi.org/10.1093/humrep/dep039\\nBackground\\nThe adjunction of exogenous hormones for controlled ovarian stimulation (COS) may alter endometrial receptiveness. In order to identify the genes mis-regulated under COS, we compared the endometrium gene expression profiles, from the same patients, in a natural cycle and in a subsequent COS cycle.\\nMethods\\nFor the same normal-responder patients (n=21), endometrial biopsies (n=84) were collected during the pre-receptive (LH+2) and receptive stages (LH+7) of a natural cycle and, subsequently, on oocyte retrieval day (hCG+2) and on transfer day (hCG+5) of a stimulated cycle. Samples were analyzed using DNA microarrays. Gene expression profiles and biological pathways involved in the endometrial receptivity were analyzed.\\nResults', \"Patricia Díaz-Gimeno et al.,2017, DOI:http://dx.doi.org/10.1016/j.fertnstert.2017.07.007\\nMATERIALS AND METHODS This retrospective study was approved by the institutional review board of the Instituto Valenciano de Infertilidad, Valencia, Spain (1401-FIVI-002-CS). Endometrial Sample Cohort Initial training set. The initial training set comprised 79 healthy, regularly cycling oocyte donors aged 20-34 years with a body mass index (BMI) of 19-25 kg/m2. Each donor's endometrial sample was timed based on the LH peak determined from the menstrual cycle of fertile women. The receptive (R) (n 1⁄4 39) group was formed from samples obtained at day LHþ7, and the prereceptive (PR) (n1⁄4 14) group comprised samples from days LHþ1t oL Hþ4. The proliferative (PF) group (n1⁄4 14) included samples collected on days 8-12 of the menstrual cycle, and the postsecretory group (n1⁄4 12) consisted of samples from LHþ11 to LHþ13. The sample cohort was published in Díaz-Gimeno et al.. New training set\", 'Nicole Doyle et al.,2022, DOI:https://doi.org/10.1016/j.fertnstert.2022.05.013\\nTable 2:\\nEndometrial Receptivity Analysis Results\\nERA RESULT\\nExposed Group N=307\\nReceptive\\n125 (40.7)\\nNon-Receptive\\n*\\n182 (59.3)\\n• Pre-receptive\\n105 (34.2)\\n• Early receptive\\n60 (19.5)\\n• Late receptive\\n9 (2.9)\\n• Post-receptive\\n8 (2.6)\\nSecond Biopsy Needed for Informative Result\\n12 (3.9)\\nOpen in a new tab\\nData are expressed as N (%);\\n*\\nERA result recommended change in timing of FET from standard 123 hrs +/− 3 hours of progesterone exposure.\\nERA results according to the number of previous unsuccessful FET cycles', 'Zion Ben Rafael et al.,2021, DOI:https://doi.org/10.1093/hropen/hoab010\\nGarcia-Velasco\\net al.\\n, 2015\\n).\\nThey are not self-explanatory for the clinician’s (or client’s) judgment, bearing witness to the randomity of the search for a credible signature of the receptive, pre-receptive or post-receptive endometrium. This issue cannot be separated from the question of consistency of the methods used and the repeatability of the test in the same patients.\\nCho\\net al.\\n(2018)\\nhave shown a large intra-patient variability in the results of gene profiles which might be due to the methods used or the hormonal status.\\nThe same group (\\nGarcia-Velasco\\net\\nal.\\n, 2015', 'Delphine Haouzi et al.,2009, DOI:https://doi.org/10.1093/humrep/dep039\\nConclusion\\nGonadotropin treatments in COS cycles led to disruptions of the transcriptional activation of genes involved in normal endometrial receptivity. We propose that when the receptiveness of the endometrium is seriously compromised by the COS protocol, fresh embryo replacement should be cancelled, the embryo frozen, and thawed embryo replacement should be performed under natural cycles.', 'Zion Ben Rafael et al.,2021, DOI:https://doi.org/10.1093/hropen/hoab010\\n.5% women not pregnant after three failed cycles, of whom 27.5% were non-receptive (a similar percentage of non-receptive endometrium was suggested after three failures also by others;', 'Delphine Haouzi et al.,2009, DOI:https://doi.org/10.1093/humrep/dep039\\nKrikun\\net al\\n., 2005\\n;\\nLiu\\net al\\n., 2008\\n).\\nThe divergence between the reports analyzing the effects of stimulation protocols on endometrial receptivity may have several explanations: differences in the day of the endometrial biopsies, different patient profiles, different COS protocols, and inadequate numbers of endometrial samples studied. In addition, no study has compared the early and the mid-secretory phases during a natural and stimulated cycle for the same patients, which seems an essential condition to minimize the impact of inter-patient variability.\\nThe aim of this study was to evaluate the global gene expression profile of human endometrial biopsies of normal responder patients during the early-secretory phase and the mid-secretory phase of natural and stimulated cycles in the same patients.']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response accurately identifies the recommendation for compromised endometrial receptiveness due to controlled ovarian stimulation, aligning with the reference answer. It correctly suggests canceling fresh embryo replacement, freezing the embryo, and performing thawed embryo replacement under natural cycles.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "8",
         "What was the accuracy range of the best performing classifiers in identifying endometriosis?",
         "The best performing classifiers identified endometriosis with 90%–100% accuracy.",
         "The best performing classifiers achieved greater than 90% accuracy, with some phase-specific classifiers achieving 100% accuracy on independent validation samples (3).",
         "['John S Tamaresis et al.,2014, DOI:https://doi.org/10.1210/en.2014-1490\\nFigure 1\\nA), and classifiers were developed with construction sets, underwent\\nk\\n-fold cross-validation to minimize overfitting, and were scored for accuracy with validation sets not used during construction.\\nTable 1.\\nEndometrial Samples by Menstrual Cycle Phase, Group, and Disease Severity\\nGroup\\nPE\\nESE\\nMSE\\nTotal\\nNE.NUPP\\n20\\n6\\n8\\n34\\nNE.UPP\\n15\\n6\\n14\\n35\\na\\nE\\n29\\n18\\n28\\n75\\nb\\nMinimal/mild (E.Min/Mild)\\n11\\n6\\n10\\n27\\nb\\nModerate/severe (E.Mod/Severe)\\n18\\n12\\n18\\n48\\nb\\nOpen in a new tab\\na\\nTwo additional NE.UPP samples had indeterminate cycle phase assignment.\\nb\\nTwo additional E samples had indeterminate cycle phase assignment: 1 E.Min/Mild and 1 E.Mod/Severe sample had indeterminate cycle phase assignments.\\nDecision tree\\nHighest accuracies were obtained with composite classifiers comprising a disease component and a severity component, corresponding to the decision tree shown in\\nFigure 1', 'John S Tamaresis et al.,2014, DOI:https://doi.org/10.1210/en.2014-1490\\nEndometriosis (E), an estrogen-dependent, progesterone-resistant, inflammatory disorder, affects 10% of reproductive-age women. It is diagnosed and staged at surgery, resulting in an 11-year latency from symptom onset to diagnosis, underscoring the need for less invasive, less expensive approaches. Because the uterine lining (endometrium) in women with E has altered molecular profiles, we tested whether molecular classification of this tissue can distinguish and stage disease. We developed classifiers using genomic data from n = 148 archived endometrial samples from women with E or without E (normal controls or with other common uterine/pelvic pathologies) across the menstrual cycle and evaluated their performance on independent sample sets. Classifiers were trained separately on samples in specific hormonal milieu, using margin tree classification, and accuracies were scored on independent validation samples', 'John S Tamaresis et al.,2014, DOI:https://doi.org/10.1210/en.2014-1490\\nFigure 2.\\nOpen in a new tab\\nA, E disease classifier margin tree dendrogram. B, Venn diagrams showing number of differentially expressed probe sets in the comparison groups shown. E.ModSevere, moderate/severe stage endometriosis; E.MinMild, minimal/mild stage endometriosis.\\nThe core genes in Supplemental Table 4 correspond to the 3 decision levels (\\nFigure 1\\nB) for the highest accuracy phase-specific composite classifiers. Importantly, each level reflects margin tree algorithm decisions for optimal high-dimensional partition of the classes into two groups at each juncture (\\n32', 'John S Tamaresis et al.,2014, DOI:https://doi.org/10.1210/en.2014-1490\\nDisease component\\nDisease classifiers discriminated among NE.NUPP, NE.UPP, and E. We developed menstrual cycle phase-unrestricted classifiers diagnosing samples in all cycle phase categories, phase-restricted classifiers diagnosing samples in both PE and ESE, and phase-specific classifiers (PE, ESE, or MSE) diagnosing samples in the corresponding cycle phase. Resampling yielded multiple high (>90%) accuracy classifiers, the best performing being the PE/ESE phase restricted, and all phase-specific classifiers achieving 100% accuracy on independent validation samples. Altogether, 75 high accuracy disease classifiers were discovered (\\nTable 2\\n). Characteristics of individual disease classifiers (accuracy and number of probe sets for each decision) from all diagnostic variants are in\\nSupplemental Table 3A\\n, wherein 2 particular PE and ESE phase-specific disease classifiers achieved 100% accuracy using less than 100 genes for each disease classification decision.\\nTable 2.', 'P. Diaz-Gimeno et al.,2022, DOI:https://doi.org/10.1093/humrep/deab262\\n., 2018; Craciunas et al., 2019; Cozzolino et al., 2020; Simo ́n et al., 2020; Lensen et al.,2 0 2 1), these TED tools can offer accuracy for staging the endometrial biopsy based on the analysis of a Transcriptomic endometrial dating 285 Downloaded from https://academic.oup.com/humrep/article/37/2/284/6454792 by guest on 08 February 2023  ......................................... ................................................... ........... . ..................................... ................................. gene expression signature in combination with simple mathematical algorithms or more sophisticated artificial intelligence methods. Best practices and procedures for applying artificial intelligence algorithms to transcriptomic data for clinical diagnosis include random forest (RF), support vector machine (SVM) and k-nearest neighbour (kNN) algorithms (Shi et al., 2010; Su et al.,2 0 1 4). In line with this, algorithms applied to gene expression data from known samples (e.g', 'P. Diaz-Gimeno et al.,2022, DOI:https://doi.org/10.1093/humrep/deab262\\nPredictive performance comparison across different independent endometrial datasets', 'John S Tamaresis et al.,2014, DOI:https://doi.org/10.1210/en.2014-1490\\n16\\n–\\n20\\n), herein we applied machine learning and high dimensional analysis to leverage these observed differences towards developing classifiers for E diagnosis and stage and pursue further insights into the pathobiology of endometrium in women with disease. We report highly accurate diagnostic classifiers that use sequential binary decisions, each based on specific gene sets, that distinguish E (disease and stage) and are menstrual cycle stage specific or hormonal milieu independent. Furthermore, differential gene expression and pathway analyses, based on these binary decisions, provided insight into steroid hormone signaling and other molecular and cellular dysfunctions in endometrium of women with E and also dysfunctions with other uterine/pelvic disorders.']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response accurately identifies the accuracy range of the best performing classifiers as greater than 90% to 100%. It correctly reflects the information provided in the reference answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "9",
         "Which two Cytoscape plugins compute various topological network parameters?",
         "NetworkAnalyzer and CentiScaPe.",
         "NetworkAnalyzer and CentiScaPe compute various topological network parameters for undirected and/ or directed networks [0].",
         "['Chia-Hao Chin et al.,2014, DOI:https://doi.org/10.1186/1752-0509-8-S4-S11\\n. Cytoscape [2] is an open platform with many plugins to expand both the visualization options and the network analysis power. Via Cytoscape, the graphical view of a network is easy accessed, and multiple layers of information including large-scale, genome-wise experiments, and protein function annotations can be granted on the interactome. Several Cytoscape plugins can score and rank the nodes by network features. For example, NetworkAnalyzer [3] and CentiScaPe [4] computes various topological network parameters for undirected and/ or directed networks. These plugins provide more centrality measures than other commonly used, but some other important features and recent developed methods are not included. Different methods focus on different topological features, or similar features with different scoring strategies', 'Chia-Hao Chin et al.,2014, DOI:https://doi.org/10.1186/1752-0509-8-S4-S11\\nConclusions: CytoHubba provide a user-friendly interface to explore important nodes in biological networks. It computes all eleven methods in one stop shopping way. Besides, researchers are able to combinecytoHubba with and other plugins into a novel analysis scheme. The network and sub-networks caught by this topological analysis strategy will lead to new insights on essential regulatory networks and protein drug targets for experimental biologists. According to cytoscape plugin download statistics, the accumulated number ofcytoHubba is around 6,700 times since 2010.', 'Albert Batushansky et al.,2016, DOI:https://doi.org/10.1155/2016/8313272\\nNowadays, there exist a number of software tools that allow researchers to generate networks, visualize them, and analyze their structure, via the calculation of a number of network properties, based on their own experimental data. Commonly known tools are Cytoscape [\\n24\\n], Gephi [\\n25\\n], and iGraph [\\n26', 'Chia-Hao Chin et al.,2014, DOI:https://doi.org/10.1186/1752-0509-8-S4-S11\\nIn this study, we implement our network scoring methods, MCC, MNC and DMNC, and eight other popular methods into a Cytoscape plugin, cytoHubba .T h r o u g h the extendable, flexible an d modulated properties of Cytoscape, cytoHubba can work together with other plugins. The computing processes had been optimized and can complete all eleven analysis on a common desktop/ notebook in a reasonable time cost. We also improve the network retrieving function in cytoHubba control panel. Therefore, users can utilize a PPIs network from public domain and extract sub-networks based on users’ domain-knowledge. Among the 11 methods, the newly proposed method MCC performs better than the others. MCC captures more essential proteins in the top ranked list in both high-degree and low-degree proteins. Another method, DMNC, catches different set of essential proteins suggesting it scores the network in different way', 'Chia-Hao Chin et al.,2014, DOI:https://doi.org/10.1186/1752-0509-8-S4-S11\\nBackground: Network is a useful way for presenting many types of biological data including protein-protein interactions, gene regulations, cellular pathways, and signal transductions. We can measure nodes by their network features to infer their importance in the network, and it can help us identify central elements of biological networks.\\nResults: We introduce a novel Cytoscape plugincytoHubba for ranking nodes in a network by their network features. CytoHubba provides 11 topological analysis methods including Degree, Edge Percolated Component, Maximum Neighborhood Component, Density of Maximum Neighborhood Component, Maximal Clique Centrality and six centralities (Bottleneck, EcCentricity, Closeness, Radiality, Betweenness, and Stress) based on shortest paths. Among the eleven methods, the new proposed method, MCC, has a better performance on the precision of predicting essential proteins from the yeast PPI network.', 'Chia-Hao Chin et al.,2014, DOI:https://doi.org/10.1186/1752-0509-8-S4-S11\\nThe cytoHubba plugin is implemented in Java, based on the Cytoscape API. The plugin implements eleven node ranking methods to evaluate the importance of nodes in a biological network including Degree [1], Edge Percolated Component [9], Maximum Neighborhood Component [10], Density of Maximum Neighborhood Component [10], Maximal Clique Centrality (proposed in this paper), Bottleneck [11], EcCentricity [12], Closeness [13], Radiality [14], Betweenness [15], and Stress [16]. Each method is associated with a functionF which assigns every nodev a numeric valueF(v). We say that the ranking of a nodeu is greater than that of another nodev if the score ofu (i.e. F(u)) is greater that ofv (i.e. F(v)). The 11 methods can be divided into two major categories: local and global methods', \"Albert Batushansky et al.,2016, DOI:https://doi.org/10.1155/2016/8313272\\n29\\n]. The selected significant correlation values can be easily converted to a table, listing in three columns the vertices that are adjacent to each other. This table is subsequently used as a template to illustrate the network using Cytoscape. We have chosen Cytoscape out of the list of network software as it was specifically developed for biological data, because of its intuitively understandable interface, wide range of visualization options, and available additional plugins for calculations of the main network properties. The method's workflow is presented in\\nFigure 1\\n.\\nFigure 1.\\nOpen in a new tab\\nCorrelation-based network, pipeline flowchart.\\n2.1. Method Pipeline\\n2.1.1. Download R-Environment and Required R-Packages\\nTo start the workflow, first download and install the latest version of R-environment from the following website:\\nhttps://www.r-project.org/\\n. For the processes described here two R-packages will be used: “psych” [\\n28\\n] and “reshape2” [\\n30\"]",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response is completely correct and factual, accurately identifying the two Cytoscape plugins, NetworkAnalyzer and CentiScaPe.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "10",
         "How long does the human endometrium remain receptive to blastocyst implantation during the implantation window?",
         "The implantation window is limited to approximately 48 hours.",
         "The implantation window, during which the human endometrium is receptive to blastocyst implantation, lasts approximately 48 hours (1, 4).",
         "['Almudena Devesa-Peiro et al.,2021, DOI:https://doi.org/10.1093/molehr/gaab011\\nIntroduction The human endometrium is hormonally regulated and changes throughout the menstrual cycle ( Noyes et al.,1 9 7 5; Murphy, 2004 ; Talbi et al.,2 0 0 6). During most of the menstrual cycle, the endometrium is not receptive to embryonic implantation; it becomes receptive during a period of two to four-five days within the mid-secretory phase known as the window of implantation ( Harper, 1992 ; Wilcox et al.,1 9 9 9 ). To aid in assisted reproduction, endometrial dating methods have been exhaustively investigated over the past 40 years, as have potential reliable biomarkers of receptive status, including days from luteinising hormone (LH) peak or exogenous progesterone administration, morphological changes detected by ultrasounds, histological features and endometrial gene expression profiles ( Noyes et al., 1975; Dı ́az-Gimeno et al.,2 0 1 1; Niederberger et al.,2 0 1 8; Craciunas et al.,2 0 1 9)', 'L Aghajanova et al.,2008, DOI:https://doi.org/10.1016/j.semcdb.2007.10.008\\nThe implantation window and challenges of studying it\\nThe endometrium is receptive to blastocyst implantation during a spatially and temporally restricted “window”, when the luminal epithelium is favorable for blastocyst implantation even in the absence of an implanting blastocyst [6, 7]. This period, at the mid-secretory phase, is limited to approximately 48 hours and is characterized by up-regulation of several endometrial growth factors, cytokines and adhesion molecules [8-10]. For ethical and practical reasons, it is not possible or extremely difficult to study the human implantation process\\nin vivo\\n. The implantation process varies among species, and results from animal studies cannot always be extrapolated to humans. Therefore, investigation about the human implantation process is limited to evaluation of human endometrium in non-pregnant cycles or responses of endometrial cellular components to placental and/or embryonic secreted products.', 'Vamberto O A Maia-Filho et al.,2014, DOI:https://doi.org/10.1177/1933719114529373\\nIn spite of the development of ovarian stimulation and embryo culture technology, assisted reproductive techniques (ARTs) have a success rate of around 30% to 50%. One of the main limiting factors of ART is the failure of embryo implantation, which is a complex and poorly understood process. Approximately 70% of morphologically normal embryos transferred to the uterus are reabsorbed before implantation, without showing any sign of trophoblast action.\\n1\\n–\\n3\\nThe endometrium is receptive to embryo adhesion and invasion only during a certain time in the menstrual cycle, deemed the implantation window. Although there is still no consensus on the exact time of embryo implantation, human clinical studies suggest the implantation window is limited to between days 20 and 24 in ideal ovulatory cycles of 28 days.\\n4\\n–\\n9', 'P. Sebastian-Leon et al.,2021, DOI:https://doi.org/10.1093/molehr/gaab027\\nThe human endometrium is a dynamic tissue that hosts the implanted embryo and is, therefore, critical to reproduction. Notably, for most of the menstrual cycle, the endometrium is not receptive to embryo implantation; receptivity in humans is induced in each cycle via an active process that only occurs during a brief time in the middle secretory phase, called the window of implantation (WOI) (Wilcox et al., 1999). Endometrial changes that prepare the WOI are complex and occur at the molecular, cellular and tissue levels in each cycle (Noyes et al., 1975; Talbi et al., 2006). Given the complexity of this regenerating tissue, researchers have used transcriptomics to characterize endometrial changes (Talbi et al.,2 0 0 6; Wang et al.,2 0 1 8), particularly to identify endometrial receptivity (Craciunas et al.,2 0 1 9)', 'L Aghajanova et al.,2008, DOI:https://doi.org/10.1016/j.semcdb.2007.10.008\\n. Endometrial histology is most consistent in biopsies from days LH −3/−2 to days LH +7/+8, when changes occur with a high degree of regularity regardless of the length of the preovulatory and postovulatory phases [28]. It is now a common practice to date endometrium according to days after the LH peak [29] [30]. Ultrastructural changes in human endometrium throughout the cycle also allow dating of the endometrium [31, 32] and assist in determining the “implantation window”. The implantation window is a short interval during the mid-secretory phase, when the endometrium is most receptive to blastocyst implantation. It begins on days 20–24 of an ideal menstrual cycle or 6–10 days after the LH surge and is believed to last less than 48 hours [33–35]. Endometrium appears to be the only tissue in which period of embryo implantation is time restricted, i.e', 'Kelsey E Quinn et al.,2019, DOI:https://doi.org/10.1016/j.mce.2019.110644\\nHawkins and Matzuk, 2008\\n). The menstrual cycle is comprised of two parallel cycles: the ovarian cycle, which includes the follicular and luteal phases, and the uterine cycle consisting of menses and proliferative and secretory phases (\\nMessinis et al., 2014\\n). In the mid-luteal phase, the endometrium undergoes changes in response to a rise in progesterone levels to ensure receptivity of the fertilized embryo (\\nHawkins and Matzuk, 2008\\n). There is a finite time during which the endometrium becomes receptive for blastocyst attachment, known as the “window of implantation” (WOI) (\\nHarper, 1992\\n).\\nThe events leading up to the WOI are critically important for the establishment of a healthy pregnancy and are of clinical interest to the field of reproductive endocrinology and infertility (\\nMurray et al., 2004', 'Hiroshi Fujiwara et al.,2020, DOI:https://doi.org/10.3390/ijms21051885\\n3\\n,\\n4\\n,\\n5\\n,\\n6\\n].\\nIt is widely believed that such an implantation window, which spans from a few days after ovulation to several days prior to menstruation, also exists in women [\\n7\\n]. The human implantation window is estimated to correspond to cycle days 20 to 24 of the menstrual cycle [\\n8\\n]. However, there has been no study to directly confirm this window. Therefore, we developed an attachment assay using a human choriocarcinoma cell line, BeWo cells, and human primary endometrial epithelial cell culture to examine whether human endometrial receptivity changes during the menstrual cycle [\\n9\\n]. In this assay, high attachment rates were observed in endometrial culture derived from the mid-luteal phase. Of note, except for the mid-luteal phase, the attachment rates were low, suggesting that human endometrial receptivity changes during the menstrual cycle [\\n10']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response accurately states the duration of the implantation window as approximately 48 hours, which aligns with the reference answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "11",
         "Which rare variant test is generally more powerful, RVT1 or RVT2?",
         "Output:::\nQuestion: Which rare variant test is generally more powerful, RVT1 or RVT2?",
         "RVT1 is generally more powerful than RVT2 (Document 0).",
         "['Andrew P Morris et al.,2009, DOI:https://doi.org/10.1002/gepi.20450\\nFig. 2\\n). However, even when the trait mean is determined by the presence or absence of a minor allele at any causal variant, RVT1 is generally more powerful than RVT2. This would suggest that RVT2 is less robust to the presence of minor alleles at non-causal rare variants than is RVT1.\\nNext, there is a clear gain in power for tests based on rare variants identified through re-sequencing over analyses of SNPs or low-frequency variants present on the GWA chip. The greatest gains are observed in the presence of substantial allelic heterogeneity (\\nFig. 1a\\n), where rare causal loci are less likely to be captured by SNPs as a result of linkage disequilibrium [\\nThe International HapMap Consortium, 2005\\n;\\nZeggini et al., 2005\\n]. However, the differences in power between the tests are less noticeable when there is less allelic heterogeneity (\\nFig. 1b', 'Xiaowei Zhan et al.,2016, DOI:https://doi.org/10.1093/bioinformatics/btw079\\nNext, we compared the association analyses results on simulated datasets. We performed single variant tests, as well as simple burden, SKAT and VT tests analyzing rare variants with MAF\\u2009 <\\u20095%. RVTESTS attains a good balance of time and memory usage compared with other tools, e.g. completing the simple burden test took RVTESTS/PSEQ/VAT a 15.7/60.2/374.8\\u2009h, and required 1.6/10.0/0.8\\u2009GB of RAM (\\nSupplementary Table S3\\n).', 'Xiaowei Zhan et al.,2016, DOI:https://doi.org/10.1093/bioinformatics/btw079\\nZhan and Liu, 2015\\n) in R\\nOpen in a new tab\\n(1) A comprehensive set of rare variant association tests were implemented for both autosomal and X-linked variants. These tests range from the simple burden test, which assumes rare variants in the gene region have similar effect sizes (\\nLi and Leal, 2008\\n), to sequence-kernel association tests that can handle scenarios where variants in the same gene region affect the phenotype in opposite directions (\\nWu\\net al.\\n, 2011\\n), e.g. when both hypermorphic and hypomorphic alleles are present. For a comprehensive description of rare variant association analysis methods, the readers may refer to a recent review (\\nLee\\net al.\\n, 2014\\n).\\nRVTESTS can perform linear/logistic regression analyses for unrelated samples, as well as efficient linear mixed model analyses of (cryptically) related individuals (with either empirical or pedigree kinships;\\nLippert\\net al.\\n, 2011\\n). To permit the more accurate assessment of\\nP', 'Xiaowei Zhan et al.,2016, DOI:https://doi.org/10.1093/bioinformatics/btw079\\nIn summary, RVTESTS is an efficient and comprehensive tool for sequence-based association analyses. As the cost of sequencing continues to decrease (e.g. a milestone of whole-genome sequencing for $1000 has been recently attained), population-based genetic studies will soon reach another scale. The speed and scalability advantage of RVTESTS will be even more critical when applied to these ultra-large scale datasets. We envision that our program will continue to make valuable contributions to the study of rare genetic variants in complex human diseases.', 'Marzia Antonella Scelsi et al.,2021, DOI:https://doi.org/10.1371/journal.pcbi.1008517\\nGene-based rare variant association testing', 'E. Mossotto et al.,2019, DOI:https://doi.org/10.1186/s12859-019-2877-3\\n. Currently, SKAT (and SKAT-O optimised for small sample size) [ 30] represents the most sensitive approach to test for association between a genomic region and a phenotype. SKAT jointly assesses both rare and common variants maximizing the statistical power and representing a new class of analysis lying between burden and association tests and has been successfully applied to a large variety of complex diseases [ 31- 35]. While NGS is proving a transformative technology for the diagnosis and treatment of rare diseases, its relatively modest application in common diseases is limited by a lack of analytical approaches that incorporate individual profiles of genetic variation ascertained through NGS annotated with biologically meaningful information on their frequency and consequence', 'Andrew P Morris et al.,2009, DOI:https://doi.org/10.1002/gepi.20450\\nFig. 1b\\n). Our results also confirm previous findings that haplotype-based analyses of SNPs have greater power to detect rare variant associations than single-locus tests, unless there is substantial allelic heterogeneity [\\nMorris and Kaplan, 2002\\n]. Finally, low-frequency variants (MAF<5%) on GWA chips are too scarce to detect accumulations of minor alleles, and thus RVT1 and RVT2 have minimal power to identify rare variant associations with this type of data.']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response accurately states that RVT1 is generally more powerful than RVT2, aligning with the reference answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "12",
         "How many genes were up-regulated in the thin endometrium group compared to the control group",
         "Output:::\nQuestion: How many genes were up-regulated in the thin endometrium group compared to the control group",
         "In the thin-endometrium group, 318 genes were up-regulated compared to the control endometrium group (3, Tables 2 and S1).",
         "['Ryo Maekawa et al.,2017, DOI:https://doi.org/10.1002/rmb2.12030\\nFigure 2.\\nOpen in a new tab\\nPrincipal component analysis of the mRNA expression profiles of patients with thin and normal (control) endometria. The mRNA expression profiles of the thin endometrium (Thin 1, Thin 2, and Thin 3) and the normal‐thickness endometrium (Control 1, Control 2, and Control 3) were compared. The\\ny\\n‐axis and the\\nx\\n‐axis show the principal component numbers, respectively\\n3.2. Gene Ontology and Kyoto Encyclopedia of Genes and Genomes pathway analyses of the up‐regulated genes in the thin endometrium\\nIn the thin‐endometrium group, 318 genes were up‐regulated and 322 genes were down‐regulated, compared to the control endometrium group (Tables\\nS5\\nand\\nS6\\n). The up‐regulated genes in the thin endometrium were related to immunity processes, such as the “response to external stimulus,” “defense response,” “leukocyte mediated immunity,” “immune response,” “immune effector process,” and “regulation of immune system process” (Tables\\n2\\nand\\nS1', \"Ryo Maekawa et al.,2017, DOI:https://doi.org/10.1002/rmb2.12030\\n0.759\\nThin 2\\n30\\n30\\n5.8\\n22\\n193.9\\n25.3\\n0.750\\nThin 3\\n27\\n28\\n6.1\\n21\\n104.8\\n12.7\\n0.781\\nOpen in a new tab\\nControl, normal‐thickness endometrium; LMP, last menstrual period; thin, thin endometrium.\\n2.2. Transcriptome analysis\\nThe total RNAs were isolated from the tissues by using TRIzol reagent (Invitrogen, Carlsbad, CA, USA) and they were reverse‐transcribed by using a QuantiTect Reverse Transcription Kit (Qiagen, Valencia, CA, USA), according to the manufacturer's protocol. The transcriptome analysis gene expression was analyzed by using a GeneChip Human Genome U133 Plus 2.0 Array (Affymetrix, Santa Clara, CA, USA) that contained 54\\xa0120 probes supporting 18\\xa0599 genes, as previously reported.\\n9\\n,\\n10\", 'Delphine Haouzi et al.,2009, DOI:https://doi.org/10.1093/humrep/dep039\\nFigure 2B\\n). Interestingly, more down-regulated genes were observed during endometrial receptivity of the stimulated cycle in comparison with the natural cycle.\\nFigure 4. Venn diagram of transcripts up-regulated and down regulated during endometrial receptivity in the natural cycle compared to the stimulated cycle.\\nOpen in a new tab\\nEach list of genes was determined with the SAM software with a FDR < 5% and a fold change > 2.\\nFigure 2. Supervised hierarchical clustering of hCG+2 and hCG+5 samples from stimulated endometrium cycles.\\nOpen in a new tab\\n(A) The majority of the genes specifically modulated during endometrial receptivity in stimulated cycles (998 genes) were up-regulated (777 up-regulated genes, 221 down-regulated genes). Red, up-regulated genes; green, down-regulated genes. (B) A hierarchical clustering was performed with this gene list, allowing the separation of the two sample groups. Rhomb black outline: “Atypical” samples.', 'Ryo Maekawa et al.,2017, DOI:https://doi.org/10.1002/rmb2.12030\\nAim\\nAlthough a thin endometrium has been well recognized as a critical factor in implantation failure, little information is available regarding the molecular mechanisms. The present study investigated these mechanisms by using genome‐wide mRNA expression analysis.\\nMethods\\nThin and normal endometrial tissue was obtained from a total of six women during the mid‐luteal phase of the menstrual cycle. The transcriptomes were analyzed with a microarray. Differentially expressed genes were classified according to Gene Ontology (GO) terms and Kyoto Encyclopedia of Genes and Genomes (KEGG) pathways.\\nResults\\nThe study identified 318 up‐regulated genes and 322 down‐regulated genes in the thin endometrium, compared to the control endometrium. The GO and KEGG pathway analyses indicated that the thin endometrium possessed aberrantly activated immunity and natural killer cell cytotoxicity that was accompanied by an increased number of inflammatory cytokines, such as\\nIFN‐', 'Antonio Parraga-Leo et al.,2023, DOI:https://doi.org/10.1186/s12958-023-01131-4\\n. The roles of these regulators, and of genes related to progesterone and estrogens, were then analysed  in transcriptomic datasets compiled from samples collected throughout the menstrual cycle (n = 129), and the expression of selected TFs were prospectively validated in an independent cohort of healthy participants (n = 19). Results A total of 3,608 distinct genes from the 19 gene lists were associated with endometrial progression  and implantation failure. The lists’ regulation was significantly favoured by TFs (89% (17/19) of gene lists) and progesterone (47% (8 /19) of gene lists), rather than miRNAs (5% (1/19) of gene lists) or estrogen (0% (0/19) of gene  lists), respectively (FDR < 0.05). Exceptionally, two gene lists that were previously associated with implantation failure  and unexplained infertility were less hormone-dependent, but primarily regulated by estrogen', 'P. Sebastian-Leon et al.,2021, DOI:https://doi.org/10.1093/molehr/gaab027\\n. These include the following: Chrome 67þ, Firefox 60þ, Edge þ41, Safari 11þ and Opera 53þ. Older versions of these browsers may yield errors. Internet Explorer is not supported. Further information about the methodology behind prioritization of key drivers (genes with high connectivity and centralization within a module) and party hubs (genes strongly connecting different modules), functional annotation of the 238 ERarray genes, and functional and regulation enrichment analyses of each endometrial profile and WGCNA module are available at www.menstrualcyclegcn.com > Documentation.', 'L Aghajanova et al.,2008, DOI:https://doi.org/10.1016/j.semcdb.2007.10.008\\nAffymetrix HG-U95A\\n12 686\\n2\\n49\\n58\\nOpen in a new tab\\nFigure 1\\nrepresents the distribution of significantly changed genes among different groups of genes in the studies by Kao et al. [52], Carson et al [53], Borthwick et al [55], Riesewijk et al. [54] and Mirkin et al. [56]. Genes with unknown biological function represented one of the biggest groups of regulated genes. Cell surface proteins, extracellular marker components, and growth factors/cytokines represent a large fraction of the up-regulated genes in mid-secretory phase in these studies, as well as genes encoding intracellular signaling and cell cycle proteins. Immune genes were also represented in four out of five studies, implicating an important role for the immune cells and processes in endometrium prior to and during embryonic implantation. Genes encoding DNA binding proteins, transcription factors and DNA modifying enzymes were highly represented among down-regulated genes, as well as genes with unknown function and ESTs (']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response accurately states that 318 genes were up-regulated in the thin-endometrium group compared to the control group, which aligns with the reference answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "13",
         "Is increased endometrial thickness (greater than 14 mm) associated with decreased implantation or pregnancy rates in assisted reproduction?",
         "No, it is not associated with decreased implantation or pregnancy rates.",
         "No, increased endometrial thickness (>14 mm) is not associated with decreased implantation or pregnancy rates in assisted reproduction [0, 1].",
         "['Rakefet Yoeli et al.,2004, DOI:https://doi.org/10.1023/B:JARG.0000043701.22835.56\\nIncreased endometrial thickness (>14 mm) is not associated with decreased implantation or pregnancy rates in assisted reproduction.', \"Rakefet Yoeli et al.,2004, DOI:https://doi.org/10.1023/B:JARG.0000043701.22835.56\\nResults:\\nIn all, 1218 cycles were included in the study (50 in Group A and 1168 in Group B). There was no significant difference between the groups in mean patient or mean number of embryos transferred. Similar pregnancy and implantation rates were noted in Group A (24 and 11.3%, respectively) and Group B (27.7 and 14.7%, respectively). Endometrial thickness was found to have a significant positive correlation with the duration of follicular stimulation, and an inverse correlation with woman's age.\\nConclusions:\\nIncreased endometrial thickness (>14 mm) is not associated with decreased implantation or pregnancy rates in assisted reproduction.\", 'Zhang Shaodi et al.,2020, DOI:https://doi.org/10.1371/journal.pone.0239120\\n. In our study, we found significant associations between endometrial thickness and the rates of implantation (aOR: 1.08; 95% CI: 1.06-1.10, p < 0.0001), clinical pregnancy (aOR: 1.10; 95% CI: 1.07-1.14, p < 0.0001) and live birth (aOR: 1.09;95% CI: 1.06-1.12, p < 0.0001). The curve fitting analysis further revealed a quantitative relationship between endometrial thickness and clinical outcomes. The cut-off value of the endometrial thickness was 8.7 mm. With every millimeter increment of endometrial thickness up to 8.7 mm, the implantation rate, clinical pregnancy rate and live birth rate increased by 32%, 36% and 45%, respectively. However, the clinical outcomes tended to remain stable without further increasing when the endometrial thickness is  8.7 mm. It is noteworthy that this study is the first to report a minimum threshold of endometrial thickness for optimal pregnancy outcomes', 'Zhang Shaodi et al.,2020, DOI:https://doi.org/10.1371/journal.pone.0239120\\n. The endometrial thickness was measured at different time points during previous studies, such as on the day of hCG administration, on the day of oocyte retrieval or on the day embryos transferred [11-14]. In addition to this, the published studies also varied in other factors, for example, controlled ovarian stimulation protocols, FET protocols, number of embryos transferred, and the type of embryo transferred (cleavage stage embryo or blastocyst) [14-17]. Many factors could affect endometrial receptivity in fresh IVF cycles and natural cycle FETs, such as excessive estrogen levels, elevated endogenous progesterone, and LH surge. Since such factors can be confounding and affect the reliability of the study results, the relationship between endometrial thickness and IVF outcomes has been a subject of much debate for several decades', 'Rakefet Yoeli et al.,2004, DOI:https://doi.org/10.1023/B:JARG.0000043701.22835.56\\nPurpose:\\nThe purpose of this study was to evaluate the relationship between thick endometrium and both implantation and pregnancy rates in ART treatments.\\nMethods:\\nThe study group was composed of consecutive women undergoing ultrasonographic evaluation on the day of hCG administration in our ART unit. Endometrial thickness was measured at the thickest part of the midsagittal plane. On the basis of the findings, patients were divided into two groups: A—endometrial thickness >14 mm (above the 95th percentile) and B—endometrial thickness 7–14 mm (between the 5th and 95th percentiles). Patients with a thickness of less than 7 mm were excluded from the study.\\nResults:', 'Fady I Sharara et al.,1999, DOI:https://doi.org/10.1023/A:1020545120256\\nMultiple studies have evaluated the role of endometrial thickness and pattern as predictors of implantation and pregnancy rates (PR) in women undergoing in vitro fertilization and embryo transfer (IVF/ET) (1–32). Several studies have demonstrated a lower PR in women with thin endometria (less than 8 mm) (9, 21, 32). However, other studies have failed to confirm these findings (7, 15, 24). Similarly, multiple studies have confirmed a lower implantation rate (IR) and PR in women who exhibit a homogeneous pattern (pattern II) of the endometrium compared to a triple-line pattern (pattern I) on the day of human chorionic gonadotropin (hCG) administration (5, 7, 15, 19, 24). However, no data are available to evaluate if patients alter their endometrial thickness and pattern between the day of hCG administration (DhCG) and the day of oocyte retrieval (DRET) and whether these changes adversely affect endometrial receptivity. The purpose of this study was to prospectively evaluate endometrial thickness and pattern on DhCG and DRET and correlate these findings with the outcome of IVF/ET cycles.', 'Zhang Shaodi et al.,2020, DOI:https://doi.org/10.1371/journal.pone.0239120\\n. But in HRT-FET cycles, the major factor affecting the endometrial implantation window is the time of progesterone administration. Endometrial thickness (EMT) is the most widely used prognostic indicator for measuring endometrial receptivity. Although observations suggest that the association between EMT and IVF outcomes is still contradictory and inconsistent [4, 5], there is a general consensus that thin endometrium is associated with low pregnancy rate. Furthermore, several studies have shown that there is a correlation between EMT and clinical pregnancy outcome after conducting FET [6, 7]. However, very little research has focused on the relationship between endometrial thickness and clinical outcomes of the HRT-FET cycle']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response accurately answers the question and aligns with the reference answer. It correctly states that increased endometrial thickness is not associated with decreased implantation or pregnancy rates.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "14",
         "What statistical model does edgeR use for count data?",
         "edgeR models count data using an overdispersed Poisson model.",
         "edgeR models count data using an overdispersed Poisson model (2). It uses an empirical Bayes procedure to moderate the degree of overdispersion across genes (2, 3). The data is modeled as negative binomial (NB) distributed (2).",
         "['Mark D Robinson et al.,2009, DOI:https://doi.org/10.1093/bioinformatics/btp616\\n2007\\n,\\n2008\\n). Although initially developed for serial analysis of gene expression (SAGE), the methods and software should be equally applicable to emerging technologies such as RNA-seq (Li\\net al.\\n,\\n2008\\n; Marioni\\net al.\\n,\\n2008\\n) giving rise to digital expression data.\\nedgeR\\nmay also be useful in other experiments that generate counts, such as ChIP-seq, in proteomics experiments where spectral counts are used to summarize the peptide abundance (Wong\\net al.\\n,\\n2008\\n), or in barcoding experiments where several species are counted (Andersson\\net al.\\n,\\n2008\\n). The software is designed for finding changes between two or more groups when at least one of the groups has replicated measurements.', \"Mark D Robinson et al.,2009, DOI:https://doi.org/10.1093/bioinformatics/btp616\\n4 DISCUSSION\\nWe have developed a Bioconductor package\\nedgeR\\nthat addresses one of the fundamental downstream data analysis tasks for count-based expression data: determining differential expression. The package and methods are general, and can work on other sources of count data, such as barcoding experiments and peptide counts. To the authors' knowledge,\\nedgeR\\nis the only software for SAGE or DGE data at this time which can account for biological variability when there are only one or two replicate samples.\", 'Mark D Robinson et al.,2009, DOI:https://doi.org/10.1093/bioinformatics/btp616\\n2 MODEL\\nBioinformatics researchers have learned many things from the analysis of microarray data. For instance, power to detect differential expression can be improved and false discoveries reduced by sharing information across all probes. One such procedure is\\nlimma\\n(Smyth,\\n2004\\n), where an empirical Bayes model is used to moderate the probe-wise variances. The moderated variances replace the probe-wise variances in the\\nt\\n- and\\nF\\n-statistic calculations. In a closely analogous but mathematically more complex procedure,\\nedgeR\\nmodels count data using an overdispersed Poisson model, and uses an empirical Bayes procedure to moderate the degree of overdispersion across genes.\\nWe assume the data can be summarized into a table of counts, with rows corresponding to genes (or tags or exons or transcripts) and columns to samples. For RNA-seq experiments, these may be counts at the exon, transcript or gene-level. We model the data as negative binomial (NB) distributed,\\n(1)\\nfor gene\\ng\\nand sample', 'Mark D Robinson et al.,2009, DOI:https://doi.org/10.1093/bioinformatics/btp616\\nSummary:\\nIt is expected that emerging digital gene expression (DGE) technologies will overtake microarray technologies in the near future for many functional genomics applications. One of the fundamental data analysis tasks, especially for gene expression studies, involves determining whether there is evidence that\\ncounts\\nfor a transcript or exon are significantly different across experimental conditions.\\nedgeR\\nis a Bioconductor software package for examining differential expression of replicated count data. An overdispersed Poisson model is used to account for both biological and technical variability. Empirical Bayes methods are used to moderate the degree of overdispersion across transcripts, improving the reliability of inference. The methodology can be used even with the most minimal levels of replication, provided at least one phenotype or experimental condition is replicated. The software may have other applications beyond sequencing data, such as proteome peptide count data.\\nAvailability:\\nThe package is freely available under the LGPL licence from the Bioconductor web site (\\nhttp://bioconductor.org\\n).\\nContact:\\nmrobinson@wehi.edu.au', 'Mark D Robinson et al.,2009, DOI:https://doi.org/10.1093/bioinformatics/btp616\\n1 INTRODUCTION\\nModern molecular biology data present major challenges for the statistical methods that are used to detect differential expression, such as the requirement of multiple testing procedures and increasingly, empirical Bayes or similar methods that share information across all observations to improve inference. For microarrays, the abundance of a particular transcript is measured as a fluorescence intensity, effectively a continuous response, whereas for digital gene expression (DGE) data the abundance is observed as a count. Therefore, procedures that are successful for microarray data are not directly applicable to DGE data.\\nThis note describes the software package\\nedgeR\\n(empirical analysis of DGE in\\nR\\n), which forms part of the Bioconductor project (Gentleman\\net al.\\n,\\n2004\\n).\\nedgeR\\nis designed for the analysis of replicated count-based expression data and is an implementation of methology developed by Robinson and Smyth (\\n2007\\n,\\n2008', 'María José Nueda et al.,2014, DOI:https://doi.org/10.1093/bioinformatics/btu333\\nBoth maSigPro and edgeR methods are based on the GLMs but with a different approach. The major difference between the maSigPro and edgeR methods is that maSigPro is specialized in the estimation of serial data, i.e. when the independent variable is quantitative such as time. This is achieved by providing an easy way to define a polynomial model for the data. Another important difference is that maSigPro follows a second stepwise regression that obtains the best model for each gene and retains only significant coefficients in each model, whereas edgeR applies the same model to each gene.\\n2.2.1 Simulated data\\nSimulations have been created using NB distributions with a parametrization based on the mean μ and size θ. In each sample\\ni\\n, where the targeted total number of reads is\\nN\\n, and the relative abundance of each gene\\ng\\nis\\np\\ngi\\n, the expected gene counts, μ\\ngi\\n, can be computed as\\nμ\\ng\\ni\\n=\\nN\\n×\\np\\ng\\ni', 'Mark D Robinson et al.,2009, DOI:https://doi.org/10.1093/bioinformatics/btp616\\n3 FEATURES\\nThe required inputs for\\nedgeR\\nare the table of counts and two vectors annotating the samples: the vector of the library sizes (i.e. total number of reads) and a factor specifying the experimental group or condition for each sample.\\nFor users of\\nlimma\\n, the\\nedgeR\\npackage has a number of analogous functions. Once the data have been processed and the dispersion estimates are moderated, the\\ntopTags\\nfunction can be used to tabulate the top differentially expressed genes (or tags or exons, etc.). Also, MA (log ratio versus abundance) plots can be created using the\\nplotSmear\\nfunction, allowing the same visualizations for DGE data as used for microarray data analysis (\\nFig. 1\\n).\\nFig. 1.\\nOpen in a new tab\\nDGE data can be visualized as ‘MA’ plots (log ratio versus abundance), just as with microarray data where each dot represents a gene. This plot shows RNA-seq gene expression for DHT-stimulated versus Control LNCaP cells, as described in Li\\net al.\\n(\\n2008']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "4",
         "Feedback: The response is mostly correct and factual, but it includes some extra information that is not necessary for answering the question directly. The core of the answer is correct, mentioning the overdispersed Poisson model, but it also adds details about empirical Bayes procedures and negative binomial distribution, which are related but not the primary model used.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "15",
         "How much did R-spondin1-Fc pretreatment increase mature oocyte retrieval in adult mice?",
         "R-spondin1-",
         "R-spondin1-Fc pretreatment increased mature oocyte retrieval from oviducts of adult mice by 74% (2).",
         "['Yuan Cheng et al.,2013, DOI:https://doi.org/10.1096/fj.12-223412\\nFig. 4\\nD\\n). After mating with adult males, viable pups were delivered from mice pretreated with R-spondin1-Fc.\\nFigure 4.\\nOpen in a new tab\\nTreatment with R-Spondin1-Fc stimulated follicle development in adult mice to generate mature oocytes for fertilization and early embryonic development. Adult mice were treated with a GnRH antagonist (1 μg/g body weight) for 4 d, followed by treatment with the GnRH antagonist together with R-spondin1-Fc (20 μg/d) or saline for another 4 d. Ovaries were obtained for weighing, followed by serial sectioning and follicle counting. Some animals pretreated with R-spondin1-Fc or saline were further treated with eCG for 48 h, followed by hCG to induce ovulation and oocyte maturation. At 16 h after hCG injection, numbers of ovulated mature oocyte in oviducts were determined. Ovulated oocytes were inseminated with sperm\\nin vitro', 'Yuan Cheng et al.,2013, DOI:https://doi.org/10.1096/fj.12-223412\\nP\\n< 0.05\\nvs\\n. control.\\nTo test whether R-spondin could promote follicle development\\nin vivo\\n, prepubertal mice at d 10 of age were injected i.p. with R-spondin1-Fc (10 μg/d) daily for 5 d. As shown in\\nFig. 3\\nB\\n, increases in ovarian weight were found after R-spondin1-Fc treatment. Analyses of follicle dynamics indicated decreases in primary follicles, accompanied by increases in antral follicles (\\nFig. 3\\nC\\n). Some of these animals were further treated with eCG for 48 h, followed by an ovulating dose (7 IU) of hCG, before evaluating the number of ovulated eggs in oviducts 16 h later. As shown in\\nFig. 3\\nD\\n, the number of ovulated mature eggs retrieved was 58% higher in R-spondin1-Fc-pretreated group as compared with the controls. These data demonstrated the ability of R-spondin1-Fc to promote the development of primary follicles, leading to the formation of antral follicles capable of responding to gonadotropins to generate mature oocytes.', 'Yuan Cheng et al.,2013, DOI:https://doi.org/10.1096/fj.12-223412\\nTo further demonstrate the stimulatory effects of the R-spondin agonist in adult animals, mice at 9–10 wk of age were treated with a GnRH antagonist (1 μg/g body weight) daily for 4 d. After pretreatment with the GnRH antagonist, animals were treated with R-spondin1-Fc or saline for another 4 d. As shown in\\nFig. 4\\nA\\n, ovarian weight was increased after R-spondin1-Fc treatment. Follicle counting showed growth of primary follicles to the antral stage (\\nFig. 4\\nB\\n). Some animals were further treated with eCG to induce the final stage of follicle maturation. After 2 d, animals were injected with a single dose of hCG to induce ovulation. As shown in\\nFig. 4\\nC\\n, we retrieved 74% more mature oocytes from oviducts of R-spondin1-Fc-pretreated mice. Subsequent\\nin vitro\\nfertilization of ovulated oocytes and culturing of early embryos indicated that similar percentages of oocytes obtained from control and R-spondin1-Fc-pretreated animals progressed to the blastocyst stage (\\nFig. 4\\nD', 'Yuan Cheng et al.,2013, DOI:https://doi.org/10.1096/fj.12-223412\\nIn vivo\\nadministration of an R-spondin agonist stimulated the development of primary follicles to the antral stage in both immature mice and gonadotropin releasing hormone antagonist-treated adult mice. Subsequent treatment with gonadotropins allowed the generation of mature oocytes capable of undergoing early embryonic development and successful pregnancy. Furthermore, R-spondin agonist treatment of immune-deficient mice grafted with human cortical fragments stimulated the development of primary follicles to the secondary stage. Thus, oocyte-derived R-spondin2 is a paracrine factor essential for primary follicle development, and R-spondin agonists could provide a new treatment regimen for infertile women with low responses to the traditional gonadotropin therapy.—Cheng, Y., Kawamura, K., Takae, S., Deguchi, M., Yang, Q., Kuo, C., Hsueh, A. J. W. Oocyte-derived R-spondin2 promotes ovarian follicle development.', 'Yuan Cheng et al.,2013, DOI:https://doi.org/10.1096/fj.12-223412\\nin vitro\\n, and early embryonic development was monitored daily for 4 d. Percentages of mature oocytes developed into different stages of early embryos were determined.\\nA\\n) Increases in ovarian weight after treatment with R-spondin1-Fc.\\nB\\n) Follicle dynamics after R-spondin1-Fc treatment\\nin vivo\\n;\\nn\\n= 5–6.\\nC\\n) Treatment with R-spondin1-Fc increased ovulation efficiency.\\nD\\n) Development of early embryos after\\nin vitro\\nfertilization of mature oocytes;\\nn\\n= 70–150. *\\nP\\n< 0.05\\nvs\\n. control.\\nTreatment with an R-spondin agonist promoted human early follicle development\\nTo test the ability of R-spondin to stimulate human follicle development, we obtained ovarian cortical fragments from a patient with informed consent. Ovarian fragments were dissected into 1-mm\\n3\\ncubes and xenotransplanted into kidney capsules of SCID mice. At the day after transplantation, animals were treated with saline or R-spondin1-Fc (20 μg/animal/d) for 7 d. As shown in\\nFig. 5\\nA', 'Yuan Cheng et al.,2013, DOI:https://doi.org/10.1096/fj.12-223412\\nA\\n) Purification of recombinant R-spondin1-Fc and its stimulation of the canonical Wnt signaling pathway. Left panel: Coomassie blue staining of R-spondin1-Fc. Right panel: TCF-luciferase assay. 293T cells were transfected with TCF-luciferase reporter plasmids before treatment with different doses of R-spondin2 or R-spondin1-Fc.\\nn\\n= 5. For testing the\\nin vivo\\nactions of R-spondin 1-Fc, prepubertal mice at d 10 of age were injected ip with R-spondin1-Fc (10 μg/d) or saline daily for 5 d for ovarian weight determination and follicle counting. Some animals were further treated with eCG (7 IU) for 48 h, followed by a single dose (7 IU) of hCG to induce ovulation.\\nB\\n) Increases in ovarian weight after treatment with R-spondin1-Fc.\\nC\\n) Follicle dynamics after R-spondin1-Fc treatment\\nin vivo\\n;\\nn\\n= 5–6.\\nD\\n) Ovulation efficiency. *\\nP\\n< 0.05\\nvs\\n. control.\\nTo test whether R-spondin could promote follicle development\\nin vivo', 'Yuan Cheng et al.,2013, DOI:https://doi.org/10.1096/fj.12-223412\\nin vivo\\nstudies, R-spondin1-Fc was injected intraperitoneally into d 10 mice (10 μg/d) daily for 5 d, followed by a single injection of eCG (7 IU) and, at 48 h later, with an ovulatory dose (7 IU) of hCG. At 16 h after treatlater, numbers of ovulated oocytes in oviducts were determined. For studies in adult animals, BDF1 mice at 9 wk of age were pretreated s.c. with a gonadotropin-releasing hormone (GnRH) antagonist (1 μg/g body weight; Cetrotide, Shionogi & Co., Tokyo, Japan) daily for 4 d, followed by daily i.p. treatment with saline or R-spondin1-Fc (20 μg/animal/d), together with the GnRH antagonist for another 4 d. Animals were then treated i.p. with a single injection of eCG (5 IU) to stimulate preovulatory follicles and, at 48 h later, with hCG (5 IU) to induce ovulation. At 16 h after treatment, the number of ovulated oocytes was determined. For\\nin vitro']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response accurately states that R-spondin1-Fc pretreatment increased mature oocyte retrieval in adult mice. The response also provides the percentage increase (74%) and a citation. This aligns with the information expected in a complete and accurate answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "16",
         "What are the characteristics of Type I endometrial cancer?",
         "Type I endometrial cancer is estrogen-dependent, arises before and at menopause, is low grade, shows good response to treatment, and is the most prevalent histological type.",
         "Type I endometrial cancer is estrogen-dependent, low grade, shows good response to treatment and is the most prevalent histological type of endometrial cancers (2, 3).",
         "['Makito Mizunuma et al.,2016, DOI:https://doi.org/10.1016/j.heliyon.2016.e00116\\n3. Results\\n3.1. Patient characteristics\\nClinical characteristics of patients are shown in\\nTable 1\\n. Grade 1 and grade 2 endometrioid adenocarcinoma were at a significantly earlier stage compared to grade 3 endometrioid adenocarcinoma (P < 0.005).\\nTable 1.\\nPatient characteristics.\\nClinical factor\\nNormal endometrium\\nAtypical endometrial hyperplasia (n = 8)\\nEndometrial adenocarcinoma\\nProliferative (n = 5)\\nSecretory (n = 3)\\nGrade 1 (n = 35)\\nGrade 2 (n = 13)\\nGrade 3 (n = 11)\\nMean age\\n38.4 ± 3.9\\n32.3 ± 1.8\\n47.6 ± 7.3\\n54.3 ± 12.2\\n57.8 ± 11.7\\n53.9 ± 10.5\\nFIGO Stage\\nI/II\\n32 (94 %)*\\n10 (77 %)*\\n5 (45 %)\\nIII/IV\\n2 (6 %)*\\n3 (23 %)*\\n6 (55 %)\\nOpen in a new tab\\n*\\np < 0.005 vs. grade 3 according to one-way ANOVA.\\n3.2. FOXP1 expression\\nTable 2', 'Delphine Haouzi et al.,2009, DOI:https://doi.org/10.1093/humrep/dep039\\nPatient characteristics and endometrial biopsies', 'Makito Mizunuma et al.,2016, DOI:https://doi.org/10.1016/j.heliyon.2016.e00116\\nEndometrial cancer is one of the most common malignancies of female reproductive tract and has been increasing in developed countries\\n[1]\\n. Endometrial cancer is categorized into two types, Type I and Type II based on clinical features and pathogenesis\\n[2]\\n. Type I endometrial cancer is estrogen-dependent and arises in most cases before and at the time of menopause. In addition, Type I endometrial cancer is low grade, shows good response to treatment and is the most prevalent histological type of endometrial cancers.\\nWith regard to pathogenesis of endometrial carcinoma, presence of abnormal PTEN expression [\\n3\\n,\\n4\\n,\\n5\\n] and KRAS mutations [\\n5\\n,\\n6\\n] has been proposed. PTEN plays an important role in cell apoptosis and decreased expression of PTEN has been shown in endometrial carcinoma [\\n3\\n,\\n4\\n], while KRAS is an oncogene and stimulates proliferation of endometrial cancers [\\n5\\n,\\n6', 'Kun Yu et al.,2022, DOI:https://doi.org/10.3389/fendo.2022.827724\\n. Some endometrial atypical hyperplasias can evolve into EC over a long period of time. EC is subdivided into two types on the basis of histopathology. Type I endometrial tumors, also known as low-grade endometrioid, which make up account for 75% of endometrial cases, are usually associated with high levels of ERa . Type II tumors include high-grade endometrioid tumors, serous tumors, clear cell tumors, carcinosarcomas, and mixed histology tumors. Carcinomas can typically be classified into two types according to their estrogenic state, type 1 (ER-positive EC) and type 2 (ER-negative EC). In a previous study, the expression of GPER, estrogen, progesterone, EGFR, and Ki-67 in 47 EC patients treated between 1997 and 2001 based on early immunohistochemical methods showed that GPER was overexpressed in high-risk EC patients and was negatively correlated with PR expression', 'Meenakshi Swain et al.,2022, DOI:https://doi.org/10.4103/jmh.jmh_218_21\\nFigure 13\\n].[\\n8\\n]\\nFigure 12.\\nOpen in a new tab\\n(a) Photomicrograph of a case of endometrioid carcinoma showing high-grade features, characterized by solid nests (blue arrow) with surrounding desmoplastic stroma (HE stain, ×10). (b) On immunohistochemistry, this tumor showed a diffuse, strong nuclear positivity for ER (IHC, ×20 for ER). (c) p53 also showed strong nuclear positivity in more than 75% of cells, consistent with mutant phenotype (IHC, ×10 for p53). (d) p16 showed a patchy positivity (IHC, ×10 for p16). This immunohistochemical pattern is diagnostic of an endometrioid carcinoma, p53-mutant molecular subtype\\nTable 1.\\nMolecular subtypes of endometrioid carcinoma\\nFeatures\\nPOLE ultramutated EC\\nMMR-deficient EC\\np53-mutant EC\\nNSMP EC\\nAssociated clinical features\\nYounger age at presentation\\nAssociation with Lynch syndrome\\nAdvanced stage at presentation\\nHigh BMI\\nAssociated histologic features\\nOften high-grade; scattered tumor GC; TILs\\nOften high grade; TILs; invasion', 'George Anderson et al.,2019, DOI:https://doi.org/10.1515/bmc-2019-0017\\n.  Such a frame of reference better integrates wide bodies  of data on the pathoetiology and pathophysiology of  endometriosis, including its association with heightened  cancer risk.', 'Makito Mizunuma et al.,2016, DOI:https://doi.org/10.1016/j.heliyon.2016.e00116\\n2.1. Subjects and tissue samples\\nSeventy-five tissue samples were collected from patients seen at Hirosaki University Hospital from 2009 to 2013. Thirty-five patients had grade 1 endometrioid adenocarcinoma, 13 had grade 2 endometrioid adenocarcinoma, 11 had grade 3 endometrioid adenocarcinoma, 8 had atypical endometrial hyperplasia, and 8 had normal endometrium (5 patients were in the proliferative phase and 3 were in the secretory phase). Samples of histologically normal endometrium were obtained from patients who underwent a hysterectomy due to a benign condition such as uterine myoma. Endometrial phases were determined based on hematoxylin and eosin-stained sections assessed in accordance with the histological criteria of Noyes et al.\\n[26]']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "4",
         "Feedback: The response accurately identifies several key characteristics of Type I endometrial cancer, including its estrogen dependence, low grade, good treatment response, and prevalence. However, it misses the detail that Type I endometrial cancer arises before and at menopause.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "17",
         "How much more correct predictions did REVEL thresholds yield compared to SIFT/PolyPhen2 for all 66 genes?",
         "REVEL thresholds yielded 12.4% more correct predictions than SIFT/PolyPhen2.",
         "REVEL thresholds yielded 12.4% more correct predictions than SIFT/PolyPhen2 for all 66 genes (0).",
         "['Yuan tian et al.,2019, DOI:https://doi.org/10.1038/s41598-019-49224-8\\n.6% more correct predictions than SIFT/PolyPhen2,  respectively (Supplementary Tables S7 and S8). Comparisons of REVEL or BayesDel with SIFT/PolyPhen2 agreement for all 66 genes yielded similar results (Supplemental Tables S9 and S10). Using REVEL or BayesDel thresholds yielded 12.4% or 11.7% more correct predictions than SIFT/PolyPhen2, respectively.', 'Yuan tian et al.,2019, DOI:https://doi.org/10.1038/s41598-019-49224-8\\nSimilar trends in prediction performance were observed for thresholds estimated across 66 genes compared to the 20-gene analysis. Using a combination of gene-level and generalized thresholds, REVEL and  BayesDel achieved higher OPP than those of CADD, MetaSVM and Eigen (0.8% to 9.7% higher, all p  < 0.32)  (Supplementary Table S5). REVEL and BayesDel were discordant on assigned evidence for 14.5% of variants,  roughly half of which were better predicted with REVEL and half with BayesDel (Supplementary Table S6).', 'Yuan tian et al.,2019, DOI:https://doi.org/10.1038/s41598-019-49224-8\\nevidence assignment by alternative methods. Our 20-gene analysis also demonstrated that evidence  assigned using REVEL or BayesDel gene-level and generalized thresholds each had consistently better prediction  performance than those derived using SIFT/PolyPhen2 agreement (Table 1). Evidence assigned using REVEL vs.  SIFT/PolyPhen2 agreement was concordant for 1,463 (68.0%) variants (Supplementary Table S7). Among the 690  variants for which evidence assignment was discordant, use of thresholds based on SIFT/PolyPhen2 agreement  resulted in 27.5% false positives/negatives, in contrast to REVEL ’s 4.6% false positives/negatives. Proportions of  true and false predictions similarly favored BayesDel over SIFT/PolyPhen2 agreement (Supplementary Table S8).  Using REVEL or BayesDel thresholds achieved 4.9% or 5.6% more correct predictions than SIFT/PolyPhen2,  respectively (Supplementary Tables S7 and S8)', 'Yuan tian et al.,2019, DOI:https://doi.org/10.1038/s41598-019-49224-8\\nAlternative methods. For comparison, we evaluated the performance of in silico prediction using generalized thresholds from a combination of 20 genes and that of SIFT/PolyPhen2 agreement (Table  1). Evidence  for SIFT/PolyPhen2 agreement was assessed as deleterious if SIFT < 0.05 and PolyPhen2 = “possibly/probably  damaging” , or benign if SIFT ≥ 0.05 and PolyPhen2 = “benign” . All analyses were conducted with R for Statistical  Computing version 3.3.3.', 'Yuan tian et al.,2019, DOI:https://doi.org/10.1038/s41598-019-49224-8\\n. 2). While these findings support the use of gene-level thresholds for in silico evidence, many clinically actionable  genes currently lack a sufficient number of classified variants for gene-level analysis. Given the NPV , PPV , and  OPP for generalized thresholds are high (all >90%; Supplementary Table S5), and their prediction performance  is superior to that of SIFT/PolyPhen2 agreement, generalized thresholds derived from the full set of 66 genes and  their 90% CI are also presented (Supplementary Table S12). For REVEL, the 90% CI estimated for generalized  thresholds for benign and deleterious evidence overlap with those estimated at the gene-level for 7/20 and 8/20  genes, respectively. For BayesDel, overlapping 90% CI for generalized and gene-level thresholds indicating benign  or deleterious evidence were observed for 9/20 and 7/20 genes, respectively', 'Yuan tian et al.,2019, DOI:https://doi.org/10.1038/s41598-019-49224-8\\nAssigning in silico evidence using gene-level 2-sided thresholds from REVEL or BayesDel scores achieved higher  predictive accuracy and yield compared to the use of other in silico predictors. The REVEL and BayesDel thresholds we report can serve as a viable resource for assigning in silico  evidence to missense variants in the genes  examined herein, to be used in conjunction with other lines of evidence in variant assessment as recommended  in ACMG/AMP guidelines.', 'Yuan tian et al.,2019, DOI:https://doi.org/10.1038/s41598-019-49224-8\\nevidence assignment using gene-level and generalized thresholds. Using gene-level thresholds,  REVEL and BayesDel achieved the highest OPP , 0.907 and 0.908, respectively (Table 1). Compared to CADD,  MetaSVM, and Eigen, predictions using REVEL had equivalent NPV (−0.6% to 1.1% relative change, all p > 0.33)  and improved PPV (1.3% to 3.2% higher, all p < 0.28), YR (1.5% to 10.6% higher, all p < 0.37) and OPP (0.6% to  4.1% higher, all p < 0.60) (Table 1; Fig. 1). Like REVEL, BayesDel outperformed CADD, MetaSVM, and Eigen.  REVEL and BayesDel had nearly equivalent PPV , NPV , YR and OPP (all p > 0.44 for differences between the two). For each of 5 meta-predictors in 20 genes, evidence assignment using gene-level thresholds consistently  outperformed that of generalized thresholds (2.2% to 5.9% higher OPP , all p  < 0.09) (Table 1). While REVEL  and BayesDel were concordant for 86']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response is completely correct, accurate, and factual, as it accurately states the percentage increase in correct predictions yielded by REVEL thresholds compared to SIFT/PolyPhen2.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "18",
         "Which type of stem cells has been most widely used for strategies aimed at regenerating the ovarian niche?",
         "Mesenchymal stem cells (MSCs).",
         "Mesenchymal stem cells (MSCs) have been the most widely used for strategies aimed at regenerating the ovarian niche (0).",
         "['Alba M. Polonio et al.,2021, DOI:https://doi.org/10.3389/fendo.2020.626322\\nMost of these strategies highlighted the relevance of the ovarian niche as a key parameter to promote an adequate follicle development in order to restore ovarian function. Following this idea, one of the most promising strategies pursues the regeneration of ovarian niche using Stem Cells (SCs) in order to promote development of remaining follicles within the ovary. With the rise of regenerative medicine, different types of SCs have been tested for follicular rescue and regeneration of the ovarian niche. Among them,mesenchymal stem cells (MSCs) have been the most widely used for these strategies. MSCs are a population of SCs that can be derived from different adult tissues, and that have proliferative, self-renewal and differentiation to different lineages properties . Different studies in animal models with different degrees of ovarian damage describe the ability of MSCs to restore ovarian function in these animals', 'Alba M. Polonio et al.,2021, DOI:https://doi.org/10.3389/fendo.2020.626322\\nPremature ovarian insufficiency is a common disorder affecting young women and represents the worst-case ovarian scenario due to the substantial impact on the reproductive lifespan of these patients. Due to the complexity of this condition, which is not fully understood, non-effective treatments have yet been established for these patients. Different experimental approaches are being explored and strategies based on stem cells deserve special attention. The regenerative and immunomodulatory properties of stem cells have been successfully tested in different tissues, including ovary. Numerous works point out to the efficacy of stem cells in POI treatment, and a wide range of clinical trials have been developed in order to prove safety and effectiveness of stem cells therapy- in diminished ovarian reserve and POI women. The main purpose of this review is to describe the state of the art of the treatment of POI involving stem cells, especially those that use mobilization of stem cells or paracrine signaling.', 'Alba M. Polonio et al.,2021, DOI:https://doi.org/10.3389/fendo.2020.626322\\n. It has also been reported a long-term survival of UCMSCs in the rat ovary after the transplant  and stabilization of the ovarian epithelium by these SCs . Both human UCMSCs and AMSCs interventions restore ovarian morphology elasticity and toughness, and involve a slight recovery of ovarian function in QT damaged ovaries . Menstrual blood is another possible source of MSCs.Human menstrual blood-derived stem cells (MenSCs)are endometrial MSCs, which have also been used for the treatment of POI. Thus, the ability of these cells to migrate to the ovary has been reported, and ovary infiltration by MenSCs is followed by hormone levels restoration and follicular co u n ti n c r e a s ei nm i c ew i t hQ T - induced POF , as well as the restoration of fertility in these mice . The reduction of bothfibrosis and apoptosis through cytokine secretion ha s been suggested as possible mechanisms to restore ovarian function by MenSCs . Feng et al', 'Alba M. Polonio et al.,2021, DOI:https://doi.org/10.3389/fendo.2020.626322\\n. In the context of ovarian tissue, stem cell paracrine actions should be evaluated for their capacity to activate the preexisting quiescent follicles based on the ability of BMDSCs to produce and secrete a broad variety of growth factors involved in follicular growth, angiogenesis, viability, and ovarian response to Controlled Ovarian Stimulation (COS) . In fact, our results suggest that ASCOT optimized the growth of existing follicles, mediated the presence of specific stem cell secreted factors such as FGF-2 and THSP-1 within aphaeresis. Based on that, a randomized pilot study (NCT03535480) has been designed with 20 POI women younger than 39 . Patients will be randomized to the ASCOT or only stem cell mobilization based on the ability of ovarian niche to attract undifferentiated cells from BM in a process known as“homing’’. It is important to highlight that many other stem cell origins are also being tested worldwide in several RCT involving POI women', 'Saioa Torrealday et al.,2017, DOI:https://doi.org/10.12688/f1000research.11948.1\\nAn area of research that has the potential for revolutionizing the course of POI is that of stem cell technology. Therapeutic potential of hematopoietic and mesenchymal stem cells in humans has progressed over recent years, and much effort has focused on establishing the existence of oogonial stem cells (OSCs), challenging the central dogma that women are born with a fixed number of oocytes that steadily declines until menopause is reached. Indeed, a number of investigators have now reported success in identifying OSCs in experimental models allowing postulation that mitotically active OSCs may also be present in human ovarian cortex\\n89\\n–\\n93\\n. OSCs have been detected in women and have been able to form oocyte-like structures in a xenotransplant model\\n92\\n,\\n94\\n; however, the physiological relevance of OSCs in human female remains unclear\\n91', 'Alba M. Polonio et al.,2021, DOI:https://doi.org/10.3389/fendo.2020.626322\\n. Feng et al. also described a possible role of these cells in the regulation of folliculogenesis . Recovery of ovarian function has been also achieved by injecting only the culture medium of MenSCs, which reinforces the idea of paracrine action of these cells . Thus, MesSCs would represent an interesting alternative due to the possibility of non-invasive collection. Nevertheless, most of POI patients present amenorrhea or oligomenorrhea, which reduces the application of this strategy. Adipose tissue-derived mesenchymal stem cells (ADMSCs) have also shown the ability to restore ovarian function, increasing the number of follicles after injection into the ovary in QT induced POF mice and rats . An improvement in estradiol serum levels and an increase in the gestation rate have been reported after ADMSCs transplant . Fouad et al', 'Alba M. Polonio et al.,2021, DOI:https://doi.org/10.3389/fendo.2020.626322\\n. Although promising results obtained for ovarian rescue with different types of MSC, their clinical application requires cell culture procedures to reach clinically relevant cell numbers for transplant. This represents a main limitation for use, as the accumulation of genomic and epigenomic alterations and degeneration in progenitor potency in human SCs have been associated to cell expansion procedures . Bone marrow derived stem cells (BMDSCs) present an interesting alternative for transplantation in women with POI (Table 1). The possibility of obtaining a large number of BMDSCs, from an autologous source, by means of wellestablished clinical protocols - used for BM transplant after QT- makes them a valuable candidate . This possibility lets us avoid cell expansion steps, which is associated with genetic instability . In 2007, Lee et al. described fertility rescue in a QT-induced POF rat model after injection into the tail of BMDSCs derived from another rat']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response accurately identifies Mesenchymal stem cells (MSCs) as the most widely used stem cells for ovarian niche regeneration strategies.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "19",
         "How does the ERA computational model classify the endometrium?",
         "The ERA computational model classifies the endometrium as pre-receptive, receptive, or post-receptive based on its transcriptomic profile.",
         "The ERA computational model classifies the endometrium as pre-receptive, receptive, or post-receptive based on its transcriptomic profile (2).",
         "['P. Diaz-Gimeno et al.,2022, DOI:https://doi.org/10.1093/humrep/deab262\\nPredictive performance comparison across different independent endometrial datasets', 'P. Diaz-Gimeno et al.,2022, DOI:https://doi.org/10.1093/humrep/deab262\\nPredictive performance comparison of signatures and consistency across endometrial datasets', 'Nicole Doyle et al.,2022, DOI:https://doi.org/10.1016/j.fertnstert.2022.05.013\\n6\\n). ERA employs a computational predicter model that classifies the endometrium as pre-receptive, receptive or post-receptive based on its transcriptomic profile. The RNA panel employed by ERA was developed via assessment of: (1) natural cycles (optimal fertile group), (2) controlled ovarian hyperstimulation cycles as (subfertile) and (3) intrauterine device-induced refractory endometrium in otherwise fertile patients (negative control) (\\n7', 'Maria Ruiz-Alonso et al.,2021, DOI:https://doi.org/10.1093/hropen/hoab011\\nThis article summarises and contextualises the accumulated basic and clinical data on the\\nERA test and addresses specific comments and opinions presented by the opponent as part of\\nan invited debate. Progress in medicine depends on new technologies and concepts that\\ntranslate to practice to solve long-standing problems. In a key example, combining RNA\\nsequencing data (transcriptomics) with artificial intelligence (AI) led to a clinical\\nrevolution in personalising disease diagnosis and fostered the concept of precision\\nmedicine. The reproductive field is no exception. Translation of endometrial\\ntranscriptomics to the clinic yielded an objective definition of the limited time period\\nduring which the maternal endometrium is receptive to an embryo, known as the window of\\nimplantation (WOI). The WOI is induced by the presence of exogenous and/or endogenous\\nprogesterone (P) after proper oestradiol (E\\n2\\n) priming. The window lasts', 'Maria Ruiz-Alonso et al.,2021, DOI:https://doi.org/10.1093/hropen/hoab011\\nin endometrial biopsies collected throughout the menstrual cycle (n\\u2009=\\u2009128), and results were\\nmeasured by the quadratic weighted Kappa index (\\nDiaz-Gimeno\\net al.\\n, 2013\\n). For the accuracy study, biopsies were\\ngrouped into two cohorts: the training set (n\\u2009=\\u200979) for ERA machine-learning training and\\ndating, and a test set (n\\u2009=\\u200949) for comparison between histological and ERA dating. For the\\nreproducibility study, seven women underwent one ERA test and a repeat test 29–40\\u2009months\\nlater on the same day of their cycle. Concordance values following luteinising hormone (LH)\\npeak were 0.618 (0.446–0.791) and 0.685 (0.545–0.824) for the two pathologists. Further, the\\nKappa index for inter-observer variability (0.622; 0.435–0.839) was sub-optimal. ERA dating\\nachieved a concordance of 0.922 (0.815–1.000) with LH peak. ERA test reproducibility in the\\nindicated subgroup was consistent in all patients (\\nDiaz-Gimeno\\net al.\\n, 2013\\n). These data provided robust indicators\\nfor the utility of ERA.', 'P. Diaz-Gimeno et al.,2022, DOI:https://doi.org/10.1093/humrep/deab262\\n. 3E). Thus, together, the TED tool comprises the expression of selected genes measured with the TruSeq panel with Illumina assays (Supplementary Table SIII) and the machine learning models that stage the endometrial tissue as C1, C2, C3 and C4 according to secretory phase progression.', 'Maria Ruiz-Alonso et al.,2021, DOI:https://doi.org/10.1093/hropen/hoab011\\nhttp://www.stat.stanford.edu/_tibs/SAM/\\n), intersected with the multitest gene\\nlist (\\nhttp://www.bioconductor.org/\\n).\\nMathematically, the approach can be written as: [T-Rex U SAM]Xmulttest.\\nInitially, the ERA was created as a customised array containing 238 differentially\\nexpressed genes that were coupled to a computational predictor able to identify the\\ntranscriptomic profiles of proliferative (PRO), pre-receptive (PRE), receptive (R) or\\npost-receptive (POST) endometrial samples, regardless of their histological appearance.\\nThese 238 genes were presented to the scientific community in\\nDíaz-Gimeno\\net al.\\n(2011)\\n. But even more\\nimportant than the genes implicated is the prediction algorithm, which enables combining the\\nexpression of all 238 analysed genes to reach a consensus clinical diagnosis.\\nTo test its accuracy and reproducibility, ERA was compared to standard histological methods\\nin endometrial biopsies collected throughout the menstrual cycle (n\\u2009=\\u2009128), and results were']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response accurately classifies the endometrium based on the ERA computational model, aligning with the reference answer. The inclusion of \"pre-receptive, receptive, or post-receptive\" and the mention of \"transcriptomic profile\" are key components present in both the response and the reference.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "20",
         "How do RNA-seq and microarray-based models compare in clinical endpoint prediction?",
         "RNA-seq and microarray-based models perform similarly in clinical endpoint prediction.",
         "RNA-seq and microarray-based models perform similarly in clinical endpoint prediction (1, 3). RNA-seq-based models performed significantly better than microarray-based models in predicting endpoint EFS ALL (P = 0.043), while models based on the two platforms performed similarly well in predicting the remaining five endpoints (4).",
         "['Wenqian Zhang et al.,2015, DOI:https://doi.org/10.1186/s13059-015-0694-1\\nClinical endpoint prediction by RNA-seq- and microarray-based models', 'Wenqian Zhang et al.,2015, DOI:https://doi.org/10.1186/s13059-015-0694-1\\nConclusions: We demonstrate that RNA-seq outperforms microarrays in determining the transcriptomic characteristics of cancer, while RNA-seq and microarray-based models perform similarly in clinical endpoint prediction. Our findings may be valuable to guide future studies on the development of gene expression-based predictive models and their implementation in clinical practice.', 'Wenqian Zhang et al.,2015, DOI:https://doi.org/10.1186/s13059-015-0694-1\\nGiven the comprehensive transcriptomic information provided by RNA-seq and its power to identify DEGs, we hypothesized that RNA-seq may improve gene expression-based clinical endpoint prediction over microarrays', 'Wenqian Zhang et al.,2015, DOI:https://doi.org/10.1186/s13059-015-0694-1\\n. On average, models comprising 100 to 1,000 features gave the best prediction results (Fig. 4b). The model size effect, however, differed between the endpoints analyzed: For prediction of the two endpoints that were easy to predict (SEX and CLASS LABEL), only one to 10 features were required, while more complex models improved prediction results for the remaining endpoints (Additional file 1: Figure S11a). In addition, different analysis teams and different modeling methods had varying performances across the endpoints (Additional file 1: Figure S11b and c). The remaining 3.5 % were residual variance not explained by any of the factors under investigation (Fig. 4a). Taken together, our data demonstrate that microarray an d RNA-seq models perform similar in clinical endpoint prediction, and that the RNA-seq processing pipeline and the feature level do not influence prediction performances systematically', 'Wenqian Zhang et al.,2015, DOI:https://doi.org/10.1186/s13059-015-0694-1\\nWe determined prediction performances of all models in terms of MCC for each endpoint in the validation set, and compared performances of RNA-Seq and microarraybased models. RNA-seq-based models performed significantly better than microarray-based models in predicting endpoint EFS ALL ( P = 0.043), while models based on the two platforms performed similarly well in predicting the remaining five endpoints (Fig. 2b). In line with our findings from the MAQC II study [7], we noticed that prediction performances were strongly influenced by the respective clinical endpoint. While patient’s SEX and CLASS LABEL were predicted accurately, performances were substantially inferior for predicting EFS and OS in the entire cohort, and even worse in the high-risk subgroup by both RNA-seq and microarray-based models (Fig. 2b)', 'Wenqian Zhang et al.,2015, DOI:https://doi.org/10.1186/s13059-015-0694-1\\n. RNA-seq data have also been used to develop gene expression-based predictive models in cancer research [13, 14]. Considering the vast amount of additional information provided by RNA-seq in comparison to microarrays, it is tempting to speculate that RNAseq-based models may outperform microarray-based models for clinical endpoint prediction. A comprehensive comparison of RNA-seq and microarray-based predictive models, however, is lacking to date. In this study of the Sequencing Quality Control (SEQC) consortium, we therefore aimed to systematically investigate the potential of RNA-seq-based classifiers in predicting clinical endpoints in comparison to microarrays. To this end, we selected neuroblastoma as a model, a pediatric malignancy arising from the developing sympathetic nervous system [15]. The clinical courses of neuroblastoma are remarkably heterogeneous ranging from spontaneous regression to relentless progression', 'Wenqian Zhang et al.,2015, DOI:https://doi.org/10.1186/s13059-015-0694-1\\nWe next aimed to identify additional variables that may affect prediction performances of RNA-seq-based models. In general, models derived from distinct processing pipelines and distinct feature levels did not differ significantly in their prediction performances with few exceptions (Fig. 3a and b). To investigate systematically which variables may affect model performances, we performed variance component analysis using Mixed Modeling [29]. We found that 95 % of the variation across MCC values was caused by effects of the endpoints, with lowest variances Fig. 2 Performances of RNA-seq- and microarray-based models to predict clinical endpoints in the validation cohorts. a Schematic overview of gene expression profiles generated by RNA-seq (n = 9 per sample) and microarray (n = 1 per sample). CL, Cufflinks; MAV, Magic-AceView; TAV, TopHat-AceView; TUC, TopHat-UCSC. b Distribution of MCC values of all models for each endpoint according to the technical platform (MA, microarray)']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "4",
         "The response is mostly correct and factual, but it includes additional information that is not present in the reference answer. The core statement about the similarity in performance is accurate.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "21",
         "What is an important prognosticator of endometrial receptivity on the day of oocyte retrieval?",
         "Endometrial pattern.",
         "Endometrial pattern, rather than thickness, on the day of oocyte retrieval appears to be an important prognosticator of endometrial receptivity (0).",
         "['Fady I Sharara et al.,1999, DOI:https://doi.org/10.1023/A:1020545120256\\nEndometrial pattern, rather than thickness, on the day of oocyte retrieval appears to be an important prognosticator of endometrial receptivity.', 'L Aghajanova et al.,2008, DOI:https://doi.org/10.1016/j.semcdb.2007.10.008\\n. Taken together, prospective studies on biomarkers as predictors of conception are needed to develop non-invasive methods in assessment of endometrial receptivity.', \"Yu-Ju Lin et al.,2015, DOI:https://doi.org/10.1186/s12958-015-0037-9\\n. When the lead follicle reached 16- 18 mm in diameter, hMG and FSH were discontinued, and hCG (Ovidrel®; Serono, Modugno, Italy) was administered. Oocyte retrieval was scheduled 36 - 38 h later. Standard IVF or ICSI procedures, depending on semen parameters, were used to achieve oocyte fertilization, as previously described [14, 15]. On day 3, all embryos were graded on a scale of 0 to 4, which was based on a modification of Veeck's morphological grading system and our previous report [16, 17]. Embryo transfer was performed 3 to 5 days after oocyte retrieval, and no more than 4 embryos were transferred per cycle. During the luteal phase, each patient received 800 mg/day of micronized progesterone (Utrogestan; Piette Internatio nal Laboratories, Brussels, Belgium) intravaginally or 90 mg of progesterone vaginal gel once daily (Crinone 8 %; Serono Pharmaceuticals Ltd., UK) starting on the day after oocyte retrieval\", 'Nicole Doyle et al.,2022, DOI:https://doi.org/10.1001/jama.2022.20438\\nDesign, Setting, and Participants\\nDouble-blind, randomized clinical trial at 30 sites within a multicenter private fertility practice in the Eastern US. Enrollment was from May 2018 to September 2020; follow-up concluded in August 2021. Participants underwent in vitro fertilization, preimplantation genetic testing for aneuploidy, endometrial receptivity testing, and frozen embryo transfer. Those with euploid blastocyst(s) and an informative receptivity result were randomized. Exclusion criteria included recurrent pregnancy loss, recurrent implantation failure, surgically aspirated sperm, donor egg(s), and unmitigated anatomic uterine cavity defects.\\nInterventions\\nThe intervention group (n\\u2009=\\u2009381) underwent receptivity-timed frozen embryo transfer, with adjusted duration of progesterone exposure prior to transfer, if indicated by receptivity testing. The control group (n\\u2009=\\u2009386) underwent transfer at standard timing, regardless of receptivity test results.\\nMain Outcomes and Measures', 'Yu-Ju Lin et al.,2015, DOI:https://doi.org/10.1186/s12958-015-0037-9\\n. The relationship between PPR and pregnancy outcome remains the subject of much debate, and some studies have hypothesized that the possible detrimental effect of PPR is mainly through a negative impact on endometrial development at the end of the follicular phase rather than by negatively affecting embryo quality. This may result in embryo-endometrial asynchrony with poor endometrial receptivity [10, 34, 35], and this may reduce embryo implantation and pregnancy rates [24, 25]. Thus, freezing embryos and transferring them in a subsequent frozen-thawed cycle may avoid impairments in endometrial receptivity, which is why it is the method most frequently used to manage PPR. The results of an oocyte donation program yielded similar observations as the present study. In that study, among 120 women who each donated oocytes on two occasions, the serum P level was ≥ 1.2 ng/mL in the first donation cycle but < 1.2 ng/mL (no further serum P elevation) in the second cycle', 'Inmaculada Moreno et al.,2021, DOI:https://doi.org/10.1186/s40168-021-01184-w\\nStudy design and sample size calculation This was a multicentre prospective observational study  analysing the EM of infertile patients with maternal age  ≤ 40 years undergoing IVF or ≤ 50 years undergoing  ovum donation. The EM was analysed in the same endo - metrial sample in which endometrial receptivity analysis  (ERA) was conducted in a cycle prior to embryo trans - fer of frozen blastocyst stage embryos (day 5/6). Patients  were all assessed in a hormone replacement therapy cycle  after 120 h of progesterone administration. The EF sam - ple was aspirated before EB collection for EM analysis  by 16S rRNA sequencing. The EB was divided into two  pieces for the study of endometrial receptivity and EM  analysis. Patients were treated following the standard  protocols in each clinic and the embryo was transferred  on the day recommended by the ERA test result using the  same hormonal protocol as for the diagnostic cycle', 'Nicole Doyle et al.,2022, DOI:https://doi.org/10.1001/jama.2022.20438\\nAllocation Concealment\\nEndometrial receptivity cycle results were received only by the study coordinators and not disclosed to the patient or clinical team. Prior to frozen embryo transfer, the study team assigned progesterone start and frozen embryo transfer times, calculated based on the patient’s study group. Timing was based on receptivity testing only for the intervention group. Although the physician, nurse, and patient were not able to access receptivity results and randomization group, they may have been able to deduce assignment to the intervention group based on the frozen embryo transfer time if the progesterone exposure time prior to frozen embryo transfer was other than 123 ± 3 hours. The assessor (statistician) remained blinded throughout.\\nSample Size Calculation']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response includes the correct answer, endometrial pattern, and correctly identifies it as an important prognosticator of endometrial receptivity. The response includes additional information that is not incorrect, but is not necessary to answer the question.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "22",
         "What should guide the restart of ART services?",
         "Advanced planning.",
         "Advanced planning should guide the restart of ART services (0).",
         "['Carlo Alviggi et al.,2020, DOI:https://doi.org/10.1186/s12958-020-00605-z\\n13. Advanced planning should guide the restart of ART services. Working groups and quality managers should design scenarios on which patients to prioritize, how working lists should be filled and staff scheduling.', 'Carlo Alviggi et al.,2020, DOI:https://doi.org/10.1186/s12958-020-00605-z\\nFuture perspectives\\nWhat we visualize as a future scenario is a gradual restart of “less urgent ” infertility treatments, which will follow different phases. The dilemma of how ART services are restarted after the present lockdown is genuine, as every country will follow a different recovery curve. Importantly, it should be considered that contamination of patients and medical staff could occur with COVID-19 when ART care is restarted, particularly in the context of asymptomatic shedding. Thus, we reiterate the recommendations given above that care should only be restarted if social distancing can be maintained, proper PPE be available and used, areas regularly disinfected, and screening for signs and symptoms of the infection undertaken before allowing patients embark on ART treatments.', 'Evelyn Lee1 and Jinhui Zhang2 et al.,2022, DOI:https://doi.org/10.1186/s12913-022-08485-2\\n. This also can have a significant impact of the costeffectiveness of a strategy, as shown in our study where  the out-of-pocket cost for the alternate treatment strate - gies was almost more than twice the cost of undertaking  autologous ART treatment.', 'Evelyn Lee1 and Jinhui Zhang2 et al.,2022, DOI:https://doi.org/10.1186/s12913-022-08485-2\\nAll age and transition probabilities and live-birth rates  were sourced from ANZARD.\\nCosts\\nA summary of cost estimates of resources consumed in  the delivery of ART services is listed in Table  2. Costs  reflected the direct healthcare costs of ART treatment  including public costs covered by Medicare (Australia’s  universal health insurance scheme), private health insur - ers and patient out-of-pockets (OOP) expenses.', 'Evelyn Lee1 and Jinhui Zhang2 et al.,2022, DOI:https://doi.org/10.1186/s12913-022-08485-2\\nART strategies\\n1) Standard autologous ART: this treatment strategy  involves undertaking two ‘complete autologous’ ART  cycles. A ‘complete autologous’ cycle in the stand - ard strategy is defined as a fresh autologous cycle  followed by two subsequent FET cycles resulting  from one episode of ovarian stimulation. The model  started with ART naïve women aged between 35 and  45years commencing treatment by entering the ‘fresh  cycle’ state where she could either proceed to an  oocyte pick-up (OPU) or the cycle is cancelled (i.e.,  due to poor response or hyper-stimulation). If the  cycle was cancelled, she could either end her treat - ment or undertake a second fresh cycle. If no live  birth was achieved after her first complete cycle, she  could end her treatment or commence a second fresh  cycle. The strategy pathway ends when two ‘complete  autologous’ cycles have been undertaken.', 'Evelyn Lee1 and Jinhui Zhang2 et al.,2022, DOI:https://doi.org/10.1186/s12913-022-08485-2\\n4) Donor ART: this strategy started with an infertile  ART naïve woman aged between 35 and 45 years  entering the model to undertake two ‘complete autologous’ cycles followed by two donor ART cycles. In  this strategy, a ‘complete autologous’ ART cycle refers  to a fresh autologous cycle followed by two subse - quent FET cycles resulting from one episode of ovar - ian stimulation. If the woman was unsuccessful in  achieving a live birth after two ‘complete autologous’  cycles, she could either end her treatment or under - take up to two FET cycles using donated oocytes  (donor ART cycles). The ‘donor ART’ strategy path - way ends after two ‘complete autologous’ cycles fol - lowed by two donor ART cycles have been under - taken.\\nHealth states', 'Carlo Alviggi et al.,2020, DOI:https://doi.org/10.1186/s12958-020-00605-z\\n9. Adherence to regulatory infection prevention recommendations should be of utmost importance for patients and health practitioners [ 28]. This advice includes the use of appropriate PPE by staff, adherence to social distancing measures for staff and patients, and staggering appointments so that no patients are waiting together in the clinic waiting area.\\n10. The importance of training of ART staff (clinicians, embryologists, nurses) on PPE needs and usage is stressed.\\n11. Good standard laboratory practices should be strictly applied when handling gametes and embryos within the embryology and andrology laboratories.\\n12. A thorough discussion between patients and health care professionals should be made for responsible shared decisions, and psychological support should be an option for those in need.']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "4",
         "Feedback: The response includes the correct answer, \"Advanced planning,\" but adds additional information that, while not incorrect, makes the response less concise than the reference answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "23",
         "How does CFTR expression in ovarian cancer specimens compare to benign or normal ovaries?",
         "CFTR expression in ovarian cancer",
         "CFTR expression in ovarian cancer specimens was significantly higher than in benign and normal ovaries (P<0.05) (0, 3). In a majority of benign tumors (77.8%) and normal ovaries (90.1%), it was difficult to detect CFTR, which was in contrast to the cancer group (0).",
         "['JIAO Xu et al.,2015, DOI:https://doi.org/10.3892/or.2015.3829\\nResults CFTR exhibits a high expression in ovarian cancer and is  correlated with clinicopathological characteristics. CFTR  staining was predominantly detected in the cell membrane  and cytoplasm (Fig. 1A). The expression of CFTR in ovarian  cancer specimens was significantly higher than that in benign  (P<0.05) and normal ovaries (P<0.05, Fig. 1B). Moreover,  in a majority of benign tumors (77.8%) and normal ovaries  (90.1%), it was difficult to detect CFTR, which was in contrast  to the cancer group (Fig. 1B). Regarding the clinicopatho - logical characteristics, the statistical analysis revealed that  the CFTR protein level was well-related to advanced clinical  stages (stage III/I v vs. I/II, P<0.05), poor histological grade  (grade  2-3 vs. 1, p<0.05), and a higher serum Ca-125 level  (P<0.05, Table I). Furthermore, we observed that CFTR  staining was stronger in the serous type and clear cell type as  compared to the remaining types (P<0.05, Table I)', 'JIAO Xu et al.,2015, DOI:https://doi.org/10.3892/or.2015.3829\\n. The present study  demonstrated that CFTR expression of ovarian cancer, which  was associated with clinical features, was significantly higher  than that in benign ovarian tumor and normal ovaries. CFTR  knockdown suppressed the malignant behavior of ovarian  cancer cells, including cell invasion, motility and proliferation.', 'JIAO Xu et al.,2015, DOI:https://doi.org/10.3892/or.2015.3829\\nIn conclusion, to the best of our knowledge, our data have  provided clinical and laboratorial evidence for the first time  that CFTR expression is well correlated with ovarian cancer  progression and aggressive behaviors, indicating CFTR may  be a novel tumor marker for ovarian cancer, particularly  for serous carcinoma. However, a greater number of cases  should be included to make the prospective studies valid.  Nevertheless, our result supports the important role of CFTR  in ovarian cancer.', 'JIAO Xu et al.,2015, DOI:https://doi.org/10.3892/or.2015.3829\\n.  Cell proliferation and motility were observed and compared  with CFTR-RNAi cells. Tumorigenicity of CFTR-RNAi cells  was investigated by tumor xenograft experiments conducted  subcutaneously in nude mice. The expresssion of CFTR in  ovarian cancer was significantly higher than that in benign  ovarian tumor and normal ovaries (P<0.05). In ovarian cancer,  CFTR expression was significantly associated with advanced  FIGO stage, poor histopathological grade and serum Ca-125  (P<0.05). Furthermore, we observed that CFTR staining was  stronger in the serous type as compared to the other types  (P<0.05). Compared with the negative control, decreased cell  invasion, migration, proliferation, adhesion and colony formation were observed in CFTR-RNAi cells in vitro. In vivo,  tumorigenic abilities of CFTR-RNAi cells were significantly  repressed compared with that of the control groups. CFTR  overexpression may play an important role in the develop - ment and progression of ovarian cancer', 'JIAO Xu et al.,2015, DOI:https://doi.org/10.3892/or.2015.3829\\n. Another study proposed that CFTR  expression was significantly downregulated in breast cancer,  which promotes epithelial-to-mesenchymal transition and is  associated with poor prognosis . CFTR is also known to be  involved in modulating signaling pathways in cell inflamma - tion and apoptosis . Although previous studies suggested  a role for CFTR in various types of cancer, the relationship  between CFTR and ovarian cancer remains to be determined. Therefore, in the present study, to identify the potential  role of CFTR in the pathogenesis of ovarian cancer, CFTR  expression was evaluated in human epithelial ovarian cancer,  benign epithelial ovarian tumor and normal ovarian tissues  using immunohistochemical staining. The role of CFTR in the  malignancy of ovarian cancer was observed in CFTR -RNAi  SKOV3 and A2780 cells in vitro and in vivo', 'JIAO Xu et al.,2015, DOI:https://doi.org/10.3892/or.2015.3829\\n. These results indicated  that the expression of CFTR had a positive correlation with the  progression state and malignancy degree of ovarian cancer,  suggesting that CFTR is a potential biomarker in progression  of ovarian cancer. In subsequent experiments, we investigated  the expression of CFTR in metastasis outside the pelvis or  retroperitoneal or inguinal node of patients, and measured  the conditions of prognosis and survival data of patients to  evaluate the prognostic value of CFTR. Notably, we found that  CFTR expression was significantly higher in the serous carci - noma and clear cell carcinoma as compared to the remaining  carcinomas used in our study. Serous ovarian cancer is the  most common histotype and accounts for more than half of  epithelial ovarian cancer patients . Therefore, we selected  the serous ovarian cancer cell lines for further study. In our experiments, the expression of CFTR in SKOV3  and A2780 cells was knocked down by shRNA-mediated  gene silencing', \"JIAO Xu et al.,2015, DOI:https://doi.org/10.3892/or.2015.3829\\n. The  cells were cultured in Dulbecco's modified Eagle's medium  supplemented with 10% fetal bovine serum and antibiotics  (100 U/ml penicillin and 100 μg/ml streptomycin) (all from  Invitrogen, Carlsbad, CA, uSA). The cells were cultured in an  incubator with 5% CO 2 at 37 ̊C. Table I. Association of CFTR expression with clinicopatho - logical characteristics in 83 cases of human epithelial ovarian  cancer.  CFTR expression  -------------------------------------  No. of pts Low no. High no. Characteristics (n=83) (%) (%) P-value Age (years)   <51.0 47 16  31  0.654   >51.0 36 14  22  Serum Ca-125 level (U/ml)   <35 17 11  6  0.010 a   >35 66 19  47  FIGO stage   I/II 31 18  13  0.002 a   III/Iv 52 12  40  Histological grade   1 18 14  4  0.000 a   2 29 9  20    3 36 7  29  Grades 2-3 vs. 1 Ascites (ml)   <100 15 7  8  0.384   >100 68 23  45  Tumor type   Serous 54 12  42    Mucinous 8 6  2  0.006 a   Clear cell 14 4  10  0.726   Endometrioid 7 5  2  0\"]",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response accurately states that CFTR expression is higher in ovarian cancer specimens compared to benign or normal ovaries. It also includes supporting details about the difficulty in detecting CFTR in benign tumors and normal ovaries, contrasting with the cancer group.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "24",
         "Which human genome build does CADD version 1.4 support?",
         "CADD version 1.4 supports the human genome build GRCh38.",
         "CADD version 1.4 supports the human genome build GRCh38 (Document 1).",
         "['Philipp Rentzsch et al.,2018, DOI:https://doi.org/10.1093/nar/gky1016\\nCADD v1.0-v1.3 made use of the human genome build GRCh37. In the latest release, v1.4, we also provide scores for the human genome build GRCh38. Because new annotations primarily support GRCh38, and coordinate liftovers are limited to regions well characterized in both genome builds, the new model is based almost entirely on annotations generated on GRCh38 (see\\nSupplementary Materials\\n). We chose annotations that are identical or similar to those used in the CADD GRCh37-v1.4 model. Although training and parameter optimization were performed independently on GRCh37 and GRCh38 models, for regions well-represented in both genome builds, the fitted models provided very similar variant scores (Figure\\n3\\n). In total, CADD v1.4 covers 2 937 639 113 bases on GRCh38 compared to 2 858 658 094 bases on GRCh37. When compared through coordinate liftover on a random sample of sites, the two different releases show very similar score distributions with Pearson correlation of 0.79 (', 'Philipp Rentzsch et al.,2018, DOI:https://doi.org/10.1093/nar/gky1016\\nde novo\\nvariants and variants that have arisen and become fixed in human populations since the split between humans and chimpanzees; the former are free of selective pressure and may thus include both neutral and deleterious alleles, while the latter are overwhelmingly neutral (or, at most, weakly deleterious) by virtue of having survived millions of years of purifying selection. Here we review the latest updates to CADD, including the most recent version, 1.4, which supports the human genome build GRCh38. We also present updates to our website that include simplified variant lookup, extended documentation, an Application Program Interface and improved mechanisms for integrating CADD scores into other tools or applications. CADD scores, software and documentation are available at\\nhttps://cadd.gs.washington.edu\\n.', 'Philipp Rentzsch et al.,2018, DOI:https://doi.org/10.1093/nar/gky1016\\n22\\n). Lists of annotations used in CADD v1.4 are available as\\nSupplementary Tables S1 and S2\\n. For InDels, variant effects are used as predicted from VEP. For all other annotations, the extreme values are selected from the two neighboring positions for insertions and across the bases of the removed range for deletions. After model training, the fitted model is applied to all ∼9 billion potential SNVs of the human reference genome in order to calculate raw CADD scores. A PHRED conversion table is derived from the relative ranking of model scores across all potential SNVs (−10 log\\n10\\nrank/total number of potential substitutions). Details on the different usage of these scores is available in the section ‘Raw versus scaled scores’.', 'Philipp Rentzsch et al.,2018, DOI:https://doi.org/10.1093/nar/gky1016\\n23\\n,\\n25–28\\n), as well as variants identified in population-based studies (e.g.\\n29\\n). Since its introduction in 2014, CADD has become one of the most widely used tools to assess human genetic variation, and other tools and scores often use CADD to benchmark their performance; according to Google Scholar CADD has been cited 1984 times (as of 15 September\\xa02018) with about 24 000 unique users of its website over the last year.\\nFurthermore, CADD has also seen applications in evolutionary studies, ranging from the interpretation of evolutionary changes (\\n30–32\\n) to the theoretical investigation of variant fitness effects in human populations (\\n33\\n).\\nThe release of CADD has also spurred the development of several other genome-wide predictors. For instance, the feature set from CADD has been used to train Deep Neural Networks (e.g. DANN (\\n34\\n)), and CADD’s underlying approach and training set definition methodology has been adapted for other model organisms (\\n35', 'Philipp Rentzsch et al.,2018, DOI:https://doi.org/10.1093/nar/gky1016\\n35\\n). A similar approach based on ape-lineage-derived variants has been used to score non-synonymous variants (\\n36\\n). CADD has also been used to develop tools for complex variants, like scoring the effect of larger structural variants (e.g. SVScore (\\n37\\n)). Some recently developed predictors are ensemble learners, which combine CADD and other scores (\\n38–41\\n). However, we are not aware of any competing tool for variant-scoring that consistently outperforms CADD in comprehensive testing across diverse use cases in human genetics.\\nCADD UPDATES AND SUPPORT OF GRCh38\\nSince the initial release of CADD in 2014, we have published four score updates. Besides, minor bug fixes and adjustments to the genomic features (\\nSupplementary Table S3', 'Philipp Rentzsch et al.,2018, DOI:https://doi.org/10.1093/nar/gky1016\\n59\\n), LINSIGHT (\\n60\\n) and CDTS (\\n61\\n). As a general statement, we believe that CADD and tools like it that: (i) integrate many correlated genomic annotations in a principled fashion; (ii) rely on large training datasets to minimize the risk of overfitting; and (iii) avoid curated sets of pathogenic and benign variants during training, represent the best path forward for predicting the relative pathogenicity or functional importance of human genetic variants on a genome-wide basis.\\nAs genomic annotations grow in depth and breadth, CADD and CADD-inspired variant scores will continue to improve and provide utility across a wide range of analytical scenarios. While this is particularly true for studies of Mendelian disease, many complex-trait, comparative genomic, population genetic and functional genomic studies are likely to also benefit from current and future versions of CADD and related frameworks.', 'Philipp Rentzsch et al.,2018, DOI:https://doi.org/10.1093/nar/gky1016\\nIn this manuscript, we presented an overview of recent updates to CADD, as well as the services that we provide in order to make those scores available and maximally useful to the scientific community. In addition to better documentation and a fresh web layout, we substantially expanded the options for how users can access scores by providing website and API lookups, genome browser tracks and an easy-to-install offline scoring script. With the release of CADD v1.4, we support direct (non-lifted) variant interpretation on GRCh38\\xa0and show that the available annotations provide a similar level of accuracy to those generated for GRCh37.\\nA key strength of CADD is that the model is trained on a very large training set that does not suffer from ascertainment bias inherent to curated sets of pathogenic and benign variants such as ClinVar (\\n7\\n) or HGMD (\\n58\\n). CADD shares this strength with only a few other scores, such as Eigen (\\n59\\n), LINSIGHT (\\n60\\n) and CDTS (\\n61']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response is completely correct and factual, as it accurately states that CADD version 1.4 supports the human genome build GRCh38. The inclusion of \"(Document 1)\" doesn't detract from the correctness.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "25",
         "What are the two main steps of the STAR algorithm?",
         "The STAR algorithm consists of a seed searching step and a clustering/stitching/scoring step.",
         "The two main steps of the STAR algorithm are seed searching and clustering/stitching/scoring (Document 0).",
         "['Alexander Dobin et al.,2012, DOI:https://doi.org/10.1093/bioinformatics/bts635\\n2 ALGORITHM\\nMany previously described RNA-seq aligners were developed as extensions of contiguous (DNA) short read mappers, which were used to either align short reads to a database of splice junctions or align split-read portions contiguously to a reference genome, or a combination thereof. In contrast to these approaches, STAR was designed to align the non-contiguous sequences directly to the reference genome. STAR algorithm consists of two major steps: seed searching step and clustering/stitching/scoring step.\\n2.1 Seed search\\nThe central idea of the STAR seed finding phase is the sequential search for a Maximal Mappable Prefix (\\nMMP\\n).\\nMMP\\nis similar to the Maximal Exact (Unique) Match concept used by the large-scale genome alignment tools Mummer (\\nDelcher\\net al.\\n, 1999\\n,\\n2002\\n; Kurtz\\net al.\\n) and MAUVE (\\nDarling\\net al.\\n, 2004\\n,\\n2010\\n).\\nGiven a read sequence\\nR\\n, read location\\ni\\nand a reference genome sequence\\nG\\n, the\\nMMP(R\\n,\\ni\\n,\\nG)\\nis defined as the longest substring (\\nR\\ni\\n,\\nR\\ni\\n+1', 'Alexander Dobin et al.,2012, DOI:https://doi.org/10.1093/bioinformatics/bts635\\nThe algorithm extensibility to long reads shows that STAR has a potential to serve as a universal alignment tool across a broad spectrum of emerging sequencing platforms. STAR can align reads in a continuous streaming mode, which makes it compatible with novel sequencing technologies such as the one recently announced by Oxford Nanopore Technologies. As the sequencing technologies and protocols evolve, new mapping strategies will have to be developed, and STAR core algorithm can provide a flexible framework to address arising alignment challenges.', 'Pauline Vabre et al.,2017, DOI:https://doi.org/10.1186/s12940-017-0242-4\\n. In all, we have combined the results from search 1 with those from search 2 according to the following algorithm: Search 1 OR Search 2.', 'Alexander Dobin et al.,2012, DOI:https://doi.org/10.1093/bioinformatics/bts635\\nSupplementary Section\\n1.8). This speed advantage is traded off against the increased memory usage by uncompressed arrays, which is assessed further in Section 3.3.\\n2.2 Clustering, stitching and scoring\\nIn the second phase of the algorithm, STAR builds alignments of the entire read sequence by stitching together all the seeds that were aligned to the genome in the first phase. First, the seeds are clustered together by proximity to a selected set of ‘anchor’ seeds. We found that an optimal procedure for anchor selection is through limiting the number of genomic loci the anchors align to. All the seeds that map within user-defined genomic windows around the anchors are stitched together assuming a local linear transcription model. The size of the genomic windows determines the maximum intron size for the spliced alignments. A frugal dynamic programming algorithm (see\\nSupplementary Section', 'Jinlian Wang et al.,2015, DOI:https://doi.org/10.7150/jca.10631\\nFigure 1. The pipeline  of pathway/networks  centric approach for  cancer biomarker   discovery. A variety of  computational tools  and algorithms have  been proposed for  biomarker discovery  based on pathway and  network methods .\\nThe most commonly  used methods are  categorized roughly  into statistical [17],   graph theory [18],  Bayesian methods  [19], text mining [20],  machine learning  [21-23] and integrative  methods summarized  in Table 1.', 'Matthew J Page et al.,2021, DOI:https://doi.org/10.1136/bmj.n71\\nby copyright, including for uses related to text and data mining, AI training, and similar technologies', \"Davis J McCarthy et al.,2012, DOI:https://doi.org/10.1093/nar/gks042\\ng\\nis therefore\\nwith δ\\u2009=\\u2009(\\nX\\nT\\nW\\ng\\nX\\n)\\n−1\\nX\\nT\\nz\\ng\\n. This iteration usually produces an increase in the likelihood function, but the likelihood can also decrease representing divergence from the required solution. On the other hand, there always exists a stepsize modifier α with 0\\u2009<\\nα\\u2009<\\n1 such that\\nproduces an increase in the likelihood. Choosing α so that this is so at each iteration is known as a line search strategy (\\n44\\n,\\n45\\n).\\nFisher's scoring iteration can be viewed as an approximate Newton-Raphson algorithm, with the Fisher information matrix approximating the second derivative matrix. The line search strategy may be used with any approximation to the second derivative matrix that is positive definite. Our implemention uses a computationally convenient approximation. Without loss of generality, the linear model can be parametrized so that\\nX\\nT\\nX\\n=\\nI\\n. If this is done, and if the\\u2009μ\\ngi\\nalso happen to be constant over\\ni\\nfor a given gene\\ng\"]",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response accurately identifies the two main steps of the STAR algorithm: seed searching and clustering/stitching/scoring, mirroring the reference answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "26",
         "Which outer dynein arm (ODA) components can Dnah6 genetically interact with?",
         "Dnah6 can genetically interact with Dnai1 and Dnah5.",
         "Dnah6 can genetically interact with Dnai1 and Dnah5, which are two outer dynein arm (ODA) components (0).",
         "['You Li et al.,2016, DOI:https://doi.org/10.1371/journal.pgen.1005821\\n. This enrichment for PCD gene mutations prompted us to experimentally interrogate for potential trans-heterozygous interactions of DNAH6 with DNAI1 or DNAH5. In zebrafish, heterotaxy was observed upon subthreshold knockdown of dnah6/dnai1 or dnah6/dnah5. Subthreshold knockdown of Dnah6 in respiratory epithelia from heterozygous Dnai1 or Dnah5 mutant mice caused motile cilia defects in the airway epithelia. Together, these studies showed Dnah6,a putative IDA component, can genetically interact with two ODA components, Dnai1 and Dnah5. We note synthetic interactions between inner and outer arm mutations are well known in Chlamydomonas. This is associated with immotile cilia and is observed in double mutants lacking the outer arm and inner arm species I1/f, between the outer arm and some monomeric inner arm dyneins, or between dimeric inner arm I1/f dynein and some monomeric inner arm dyneins[21-23]', 'You Li et al.,2016, DOI:https://doi.org/10.1371/journal.pgen.1005821\\n. In combination, these observations suggest functional interdependence between different types of dyneins, further supporting a role for oligogenic interactions functionally linking outer/inner dynein arm components. Interestingly, we note Chlamydanomonas ODA/IDA double mutants can exhibit shorter cilia, consistent with the observation of shorter cilia in the respiratory epithelia of heterotaxy patient 9002 harboring the DNAH6/DNAI1 mutations. We also observed shorter cilia after Dnah6 knockdown in the human/mouse airway and in zebrafish Kupffer ’s vesicle. In contrast, PCD patients with ODA and IDA defects associated with mutations in DNAAF2 and DNAAF3 do not exhibit shorter cilia [ 24, 25]. This suggests DNAH6 may have a unique role in mediating axonemal doublet stability. An oligogenic model of disease for heterotaxy is compelling, given the complexity of ciliogenesis', 'You Li et al.,2016, DOI:https://doi.org/10.1371/journal.pgen.1005821\\nWe also examined the effects of DNAH6 knockdown on cilia ultrastructure in the respiratory epithelia, as cilia ultrastructural defects are often observed with PCD[ 12], Analysis of cilia ultrastructure by electron microscopy (EM) showed the abundance of outer dynein arm (ODA) was not affected by DNAH6 knockdown (Fig 2F and 2H ). This was confirmed by the finding of robust DNAH5 immunostaining in the ciliary axoneme in the reciliated airway cultures subject to DNAH6 knockdown (S3 Fig). Assessment of the inner dynein arm component (IDA) by immunostaining with an antibody to the IDA component DNALI1 also showed robust DNALI localization in the ciliary axoneme in respiratory epithelia subjected to DNAH6 knockdown (S3 Fig). Together these findings show ODA and IDA assembly were not affected by DNAH6 deficiency. However, EM ultrastructural analysis showed an unexpected cilia central pair defect with DNAH6 knockdown', 'You Li et al.,2016, DOI:https://doi.org/10.1371/journal.pgen.1005821\\n. It is an ortholog of Chlamydomonas DHC2, an inner dynein arm component essential for motile cilia function. Interestingly, one of the two patients identified with a DNAH6 mutation is 9002, a patient previously found to have a heterozygous DNAI1 PCD founder mutation. Genotyping analysis showed the DNAH6 mutation in patient 9002 was inherited from the unaffected father, while the pathogenic DNAI1 mutation was inherited from the unaffected mother ( Fig 2B and S1 Fig).', 'You Li et al.,2016, DOI:https://doi.org/10.1371/journal.pgen.1005821\\n. First we note the phenotypes observed with MO knockdown included curly tail and left-right patterning defects, both phenotypes robustly associated with disruption of motile cilia function in zebrafish. Second, we independently confirmed with gene knockdown analysis that Dnah6 is required for motile cilia function not only in zebrafish Kupffer ’s vesicle, but also in the human and mouse airway ( S3 Table). DNAH6 is a putative inner dynein arm component (IDA), and is the ortholog of Chlamydomonas DHC2encoding the heavy chain of inner arm species d[ 16] (component of inner arm group 4). While no mutation in this Chlamydonomas heavy chain gene is known, this species is missing in several mutant strains with defects in non-motor subunits. Our EM analysis showed DNAH6 knockdown caused unexpected central pair defects in human respiratory cilia', 'You Li et al.,2016, DOI:https://doi.org/10.1371/journal.pgen.1005821\\nAssaying dnah6 interactions withdnah5 or dnahi1', 'You Li et al.,2016, DOI:https://doi.org/10.1371/journal.pgen.1005821\\nTo investigate whether trans-heterozygous interactions between DNAH6 and other PCD genes can perturb motile cilia function and contribute to heterotaxy and PCD, we examined digenic interactions between DNAH6 and DNAI1 or DNAH5. This analysis focused on the possible effects of haploinsufficiency using subthreshold morpholino or siRNA gene knockdown, as PCD mutations are often loss of function, such as the splicing defect DNAI1 founder mutation in patient 9002. Dual haploinsufficiency ofdnah6 with dnai1 or dnah5 in heterotaxy. To functionally assay for possible digenic interactions in causing heterotaxy, we used subthreshold antisense MO knockdown in zebrafish. Thus we carried out injections of dnah6/dnai1 or dnah6/dnah5 MO in zebrafish embryos using subthreshold MO doses that individually had no effect on development']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response accurately identifies Dnai1 and Dnah5 as components that Dnah6 genetically interacts with, mirroring the reference answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "27",
         "What percentage of additional biomarkers were detected after correcting for the menstrual cycle effect?",
         "A significant average of 44.19% more biomarkers were detected.",
         "A significant average of 44.19% more biomarkers were detected when the effect of the menstrual cycle was corrected (Document 3).",
         "['Almudena Devesa-Peiro et al.,2021, DOI:https://doi.org/10.1093/molehr/gaab011\\n. Study UD UD biomarkers masked by the menstrual cycle Previously reported Newly discovered Total Burney 2007 EU 141 544 685 Hawkins 2011 EC 144 158 302 Koot 2016 RIF 3 27 30 C) Differential expression analysis (DEA) for each menstrual cycle phase Study DEA for each menstrual cycle phase DEA with menstrual cycle correction Menstrual cycle phase (n) N /C14 DEGs Statistical Power % Statistical Power % (n) Burney 2007 PF  0 29.74% 98.7%  ESE  100 9.28% MSE  0 65.5% Hawkins 2011 PF  541 40.01% 52.1%  S  - - Koot 2016 LH þ5  0 3.21% 100%  LH þ6  0 9', 'Almudena Devesa-Peiro et al.,2021, DOI:https://doi.org/10.1093/molehr/gaab011\\n. The study design is detailed in Fig. 1A. Finally, we compared the fold change (FC), P-value, gene expression average, and standard deviation (SD) before and after menstrual cycle effect correction for understanding the aetiology of the potential endometrial biomarkers. All statistical analyses were run under R software v.3.3.2  ( R Core Team, 2016 ).', 'Almudena Devesa-Peiro et al.,2021, DOI:https://doi.org/10.1093/molehr/gaab011\\n. 3). While in Burney 2007 , Hawkins 2011 and Koot 2016 ,t h eD E G s detected before menstrual cycle correction where included in those identified after the correction was applied. For Tamaresis 2014,t h e r e were DEGs detected only before applying the menstrual cycle correction ( Table IIA). A new classification of endometrial transcriptomic biomarkers in uterine disorders Comparison of the expression profiles of significant genes identified before and after applying the menstrual cycle correction allowed us to detect their aetiology and define a new classification of endometrial biomarkers (Table IIA, Fig. 4 ). Menstrual cycle biomarkers ( Fig. 4A) are genes detected only before menstrual cycle correction, suggesting that they respond to Transcriptomic correction of menstrual cycle bias 3 Downloaded from https://academic.oup.com/molehr/article/27/4/gaab011/6134138 by guest on 08 February 2023  Figure 1', 'Almudena Devesa-Peiro et al.,2021, DOI:https://doi.org/10.1093/molehr/gaab011\\n. 2 ). However, samples were mainly grouped by the uterine disorder ( Fig. 2 ) and a significant average of 44.19% more biomarkers were detected (one-sided Fisher’s exact tests FDR /C20 5.53 /C2 10-06) when the effect of the menstrual cycle was corrected ( Table IIA ). These newly revealed uterine disorder biomarkers that were previously masked by the menstrual cycle effect were, in fact, identified despite balanced endometrial cycle stages between case and control groups ( Table IB ). Masked uterine disorder biomarkers were also detected in Koot 2016 , among endometrial biopsies collected at different time points within the mid-secretory phase (LH þ5LH þ8) ( Table IB). In contrast, samples from Lucas et al., 2016 and Pathare 2017 were primarily grouped by the disorder rather than by the time point of endometrial biopsy collection before menstrual cycle effect correction; and no more DEGs were identified after correcting the menstrual cycle effect ( Table IIA, Fig. 3)', 'A. Devesa-Peiro et al.,2022, DOI:https://doi.org/10.1093/humrep/deac010\\nRaw data were downloaded and pre-processed using the affy v.1.52.0 Rp a c k a g e(Gautier et al.,2 0 0 4) and annotated with the biomaRt R package v.2.30.0 (Durinck et al., 2009). Normalization among samples was achieved by the quantile method implemented in the limma R package v. 3.38.3 (Ritchie et al.,2 0 1 5). Exploratory analyses were performed to detect batch effects and outliers. Detected effects were corrected using theremovebatcheffect() function based on linear models (limma R package v. 3.38.3;Ritchie et al.,2 0 1 5). The same correction method was used to remove the effect of the menstrual cycle (Talbi et al.,2 0 0 6; D/C19 ıaz-Gimeno et al., 2011; Sebastian-Leon et al., 2018), which can bias biomarker discovery (Devesa-Peiro et al., 2021). The effect of presence of benign pathologies was also corrected for by the same methodology', 'Almudena Devesa-Peiro et al.,2021, DOI:https://doi.org/10.1093/molehr/gaab011\\n. Study UD N /C14 DEGs without menstrual cycle correction N /C14 DEGs with menstrual cycle correction % of newly detected disorder biomarkers Fisher’s test FDR Menstrual cycle biomarkers UD biomarkers not masked by the menstrual cycle UD biomarkers masked by the menstrual cycle Burney 2007 EU 127 812 84.36% 6.6 /C2 1016 0 127 685 Hawkins 2011 EC 903 1205 25.06% 1.92 /C2 10-11 0 903 302 Tamaresis 2014 EU 13 397 13 797 6.05% 5.53 /C2 10-06 435 12 962 835 UF 9715 10 909 11.71% 6.6 /C2 10-16 83 9632 1277 Koot 2016 RIF 2 32 93.75% 5.15 /C2 10-08 02 3 0 Pathare 2017 RIF 2783 2492 /C0 2.29% 1 - - - Lucas et al., 2016 RPL 0 0 0% - - - - B) Newly discovered uterine disorder biomarkers previously masked by the menstrual cycle effect', 'Almudena Devesa-Peiro et al.,2021, DOI:https://doi.org/10.1093/molehr/gaab011\\n. This type of biomarker was identified by Tamaresis 2014, where a gene is differentially expressed without being truly affected by the uterine disorder ( Fig. 4A ). After correcting for the menstrual cycle, the distance between gene expression patterns in different endometrial phases shortened and became non-significant. Uterine pathology biomarkers not masked by the menstrual cycle (Fig. 4B ) are genes detected with and without menstrual cycle correction because there was no effect of endometrial progression (Fig. 4.B.1 , no expression differences were observed between endometrial stages) or the effect was lower than that of the uterine disorder ( Fig. 4.B.2 , higher expression differences between cases and controls than between endometrial stages). Figure 3.Menstrual cycle effect on endometrial gene expression forLucas et al., 2016and Pathare 2017']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response accurately states the percentage of additional biomarkers detected after correcting for the menstrual cycle effect, mirroring the reference answer. It also correctly identifies the source document.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "28",
         "Which 'omics' technique has been most commonly applied to the human endometrium?",
         "Analysis of the transcriptome pattern.",
         "This question cannot be answered from the given context.",
         "['Signe Altmäe et al.,2013, DOI:https://doi.org/10.1093/humupd/dmt048\\nThe current review aims to provide guidelines for future ‘omics’ studies on human endometrium, together with a summary of the status and trends, promise and shortcomings in the high-throughput technologies. In addition, the approaches presented here can be adapted to other areas of high-throughput ‘omics’ studies.', 'Signe Altmäe et al.,2013, DOI:https://doi.org/10.1093/humupd/dmt048\\nMETHODS\\nSystematic review of the literature in PubMed, and references from relevant articles were investigated up to March 2013.\\nRESULTS\\nThe current review aims to provide guidelines for future ‘omics’ studies on human endometrium, together with a summary of the status and trends, promise and shortcomings in the high-throughput technologies. In addition, the approaches presented here can be adapted to other areas of high-throughput ‘omics’ studies.\\nCONCLUSION\\nA highly rigorous approach to future studies, based on the guidelines provided here, is a prerequisite for obtaining data on biological systems which can be shared among researchers worldwide and will ultimately be of clinical benefit.', 'Signe Altmäe et al.,2013, DOI:https://doi.org/10.1093/humupd/dmt048\\nThus, the future of the endometrial ‘omics’ field lies in well designed, sufficiently powered studies together with the application of new-generation technologies, complex data analyses and integrated systems biology approaches. Provision of integrated databases and multicentre collaboration will enable new insights, and provide valid and reliable biomarkers. We are on the threshold of realizing the promise of the ‘omics’ technologies in endometrial research.', 'Signe Altmäe et al.,2013, DOI:https://doi.org/10.1093/humupd/dmt048\\nThe current review focuses on all ‘omics’ studies applied to human endometrium. While the ‘omics’ technologies have undoubtedly advanced our knowledge of human endometrium in health and different diseases, the challenges arising from the huge amount of data generated and broad variation of ‘omics’ profiling according to different environments and stimuli make it difficult to determine the use, validity, reproducibility and interpretation of ‘omics’ data. The current review provides guidelines for future ‘omics’ studies on human endometrium and summarizes the current status together with the promise and shortcomings of the high-throughput technologies.\\n‘Omics’ studies applied to human endometrium\\nThe human endometrium is a dynamic tissue that undergoes cyclic growth, differentiation, desquamation and regeneration that are driven by the ovarian steroidal hormones oestrogen and progesterone as well as other hormones, cytokines and chemokines (\\nWilcox\\net al.\\n, 1999\\n;\\nLessey, 2000\\n;', 'Signe Altmäe et al.,2013, DOI:https://doi.org/10.1093/humupd/dmt048\\n. With the expansion of ‘omics’ analyses in the study of the endometrium, there is a growing need to develop guidelines for the design of studies, and the analysis and interpretation of ‘omics’ data.', 'Signe Altmäe et al.,2013, DOI:https://doi.org/10.1093/humupd/dmt048\\nin vivo\\ntissue.\\nSample processing for different ‘omics’ techniques has been previously discussed in detail (\\nWhite and Salamonsen, 2005\\n;\\nHorgan\\net al.\\n, 2009\\n;\\nSavaris and Giudice, 2009\\n;\\nGunaratne\\net al.\\n, 2010\\n;\\nRobert, 2010\\n;\\nPritchard\\net al.\\n, 2012\\n;\\nSlattery\\net al.\\n, 2012\\n;\\nFassbender\\net al.\\n, 2013\\n;\\nUlbrich\\net al.\\n, 2013\\n). In terms of study design and sample collection, researchers are encouraged to take careful note of these guidelines. In Table\\nII\\n, we provide an outline of the ‘omics’ techniques and respective sample sources that can be utilized in endometrial studies.\\nData processing and analysis', 'Signe Altmäe et al.,2013, DOI:https://doi.org/10.1093/humupd/dmt048\\nData processing and analysis\\nAdequate data analysis is always crucial for providing conclusive results, which can guide us to an appropriate interpretation of the data, in terms of both biological sense and clearly defined target clinical applications. Several excellent papers provide both protocols and instructions for data analysis for each ‘omic’ (see detailed references below, and general standards for ‘omic’ data description, exchange, terminology and experimental execution have also been described;\\nChervitz\\net al.\\n, 2011\\n). The aim of these data standards is not only to improve reproducibility and to avoid discrepancies but also to determine a true clinical application. Thus, researchers are encouraged to read and follow such standards (\\nChervitz\\net al.\\n, 2011\\n). In this section, an overview of the analysis and interpretation of ‘omics’ data provides some specific quality criteria details for each ‘omic’, with a special focus on human endometrium (see Table\\nIII']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "1",
         "The response states that the question cannot be answered from the given context, but the reference answer provides a specific 'omics' technique. Therefore, the response is incorrect.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "29",
         "What happens to oocytes in CDC25B-deficient mice?",
         "They do not resume meiosis and remain arrested at prophase I.",
         "Oocytes in CDC25B-deficient mice do not resume meiosis and remain arrested at prophase I (0).",
         "['Petr Solc et al.,2010, DOI:https://doi.org/10.1093/molehr/gaq034\\nCDC25\\ngenes: A, B and C.\\nCDC25C\\n-deficient mice are viable and fertile, and mouse embryo fibroblasts (MEFs) from these mice have normal timing of entry into mitosis and a normal response to DNA damage (\\nChen\\net al\\n., 2001\\n). Although\\nCDC25B\\n-deficient mice are also viable and their MEFs proliferate normally and mount a strong DNA damage response, oocytes do not resume meiosis and remain arrested at prophase I (\\nLincoln\\net al\\n., 2002\\n).', 'Enrico Maria Daldello et al.,2019, DOI:https://doi.org/10.1242/dev.172734\\nCcnb2\\n−/−\\nmice (\\nBrandeis et al., 1998\\n), we show that oocytes deficient in CCNB2 have compromised nuclear maturation, resulting in the subfertility of these mice. Thus, we conclude that CCNB2 has a significant role during mouse oocyte meiosis, which cannot be compensated for by endogenous CCNB1.', 'Petr Solc et al.,2010, DOI:https://doi.org/10.1093/molehr/gaq034\\nBrunet and Maro, 2005\\n).\\nIn this minireview, we focus on the signalling pathways responsible for prophase I arrest and resumption of meiosis in mouse oocytes. Resumption of meiosis in oocytes and recovery from G2-arrest of somatic cells have many similarities, and accordingly we highlight some common features.\\nCDK1 regulation\\nAlthough oocytes are arrested in the first meiotic prophase, resumption of meiosis has historically been viewed as a model system to study the G2-M transition, because oocytes have a 4C DNA content and the chromosomes remain relatively decondensed. The G2-M transition is largely governed by activating CDK1. CDK1 is positively regulated by CCNB1 binding but also negatively regulated by WEE1/MYT kinase family-mediated phosphorylation on Thr14 and Tyr15 (Fig.\\n1\\nA). Dephosphorylation of these residues is mediated by CDC25 phosphatases. The mammalian genome contains three\\nCDC25\\ngenes: A, B and C.\\nCDC25C', 'Sandrine Caburet et al.,2014, DOI:https://doi.org/10.1056/NEJMoa1309635\\nFig. 2B\\n). We concluded that synapsis between homo-logues was disrupted at an early leptotene-like stage and that this premature arrest did not impair the formation of programmed double-strand breaks or the recruitment of the RAD51 recombinase on recombination foci, because these processes appeared to be similar in normal and mutant oocytes (\\nFig. S6 in the Supplementary Appendix\\n).\\nFigure 2. Premature Ovarian Failure in\\nStag3\\n−/−</mtext>\\nMice.\\nOpen in a new tab\\nPanel A shows a comparative histologic analysis of ovarian sections from wild-type and\\nStag3\\n−/−</mtext>\\nmice at 7 days, 15 days, and 6 weeks of age. Growing oocytes and follicle development are evident in wild-type animals at each age. Ovaries with no follicles or oocytes were observed in their\\nStag3\\n-deficient littermates. At 6 weeks,\\nStag3', 'Enrico Maria Daldello et al.,2019, DOI:https://doi.org/10.1242/dev.172734\\nIn vivo\\n, CCNB2 depletion caused ovulation of immature oocytes, premature ovarian failure, and compromised female fecundity. These findings demonstrate that CCNB2 is required to assemble sufficient pre-MPF for timely meiosis re-entry and progression. Although endogenous cyclins cannot compensate, overexpression of CCNB1/2 rescues the meiotic phenotypes, indicating similar molecular properties but divergent modes of regulation of these cyclins.\\nKEY WORDS:\\nCyclins, Fertility, MPF, Meiosis, Mouse oocytes\\nSummary:\\nThe altered timing of maturation caused by CyclinB2 loss results in aneuploidy, ovulation of immature oocytes and decreased fecundity, indicating that CyclinB2 is essential for oocyte re-entry and progression through meiosis.', 'Enrico Maria Daldello et al.,2019, DOI:https://doi.org/10.1242/dev.172734\\n−/−\\nfemale mice display decreased fecundity\\nTogether with a previous report (\\nGui and Homer, 2013\\n), the above findings are at odds with the widely held notion that CCNB2 is dispensable for oocyte maturation. Therefore, we re-evaluated the fertility phenotype of the previously generated\\nCcnb2\\n−/−\\nmice (\\nBrandeis et al., 1998\\n).\\nWhereas both\\nCcnb2\\n+/+\\nand\\nCcnb2\\n−/−\\nmice were fertile and produced pups,\\nCcnb2\\n−/−\\nmice had decreased fecundity (\\nFig.\\xa02\\nA). Fecundity declined rapidly with age, suggesting premature ovarian failure (\\nFig.\\xa02\\nA and\\nFig.\\xa0S1C,D\\n). This phenotype was not the result of the embryonic lethality of the\\nCcnb2\\n−/−\\npups because there was no statistically significant deviation from the expected Mendelian ratio when heterozygous males and females were mated (\\nFig.\\xa0S1E\\n), suggesting that CCNB2 is dispensable for embryo development. Furthermore,\\nCcnb2\\n−/−\\nfemales generated fewer pups even when crossed with\\nCcnb2\\n+/+', 'Sandrine Caburet et al.,2014, DOI:https://doi.org/10.1056/NEJMoa1309635\\nStag3\\n−/−</mtext>\\nmice were infertile (\\nFig. S10 in the Supplementary Appendix\\n).\\nMutations in mice that affect other meiotic cohesin subunits also result in abnormal oocyte development. Specifically, oocytes show meiotic defects because of abnormal formation of axial elements (in mutant REC8),\\n17\\nimpaired sister chromatid cohesion leading to massive aneuploidy during the second meiotic division (in mutant SMC1β),\\n21\\nor mild defects in synapsis resulting in premature depletion of the oocyte pool (in mutant RAD21L).\\n18\\nFurthermore, the gradual loss of meiotic cohesins or their protector, shugoshin-like 2, is an important mechanism of meiotic nondis-junction and age-related aneuploidy and infertility, as shown in mice and postulated in humans.\\n22\\n-\\n25\\nAlthough some human cases of premature ovarian failure have been attributed to defective chromosomal cohesion at later stages of meiosis because of supernumerary chromosomes,\\n26']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response accurately states that oocytes in CDC25B-deficient mice do not resume meiosis and remain arrested at prophase I, mirroring the reference answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "30",
         "Which patients are most likely to benefit from preimplantation genetic testing for aneuploidy?",
         "Infertile women of advanced maternal age.",
         "Older women are most likely to benefit from preimplantation genetic testing for aneuploidy (PGT-A) (4). Patients aged 30 to 40 years at the time of egg retrieval and likely to produce at least 1 euploid blastocyst based on ovarian reserve testing are also candidates for PGT-A (2).",
         "['Nicole Doyle et al.,2022, DOI:https://doi.org/10.1001/jama.2022.20438\\nOpen in a new tab\\nPGT-A indicates preimplantation genetic testing for aneuploidy.\\na\\nPatients were screened against their eligibility criteria; inclusion criteria were 30 to 40 years at time of egg retrieval and likely to produce at least 1 euploid blastocyst based on ovarian reserve testing. Exclusion criteria were surgically aspirated sperm, donor egg(s), recurrent implantation failure, recurrent pregnancy loss, planned preimplantation genetic testing for monogenic disorders or structural rearrangements, unmitigated uterine cavity, body mass index (calculated as weight in kilograms divided by height in meters squared) greater than 40, currently breastfeeding, pregnant, or any contraindication to pregnancy.\\nb\\nTwo participants were withdrawn due to an endometrial receptivity cycle medication error and 1 participant was withdrawn due to a biopsy timing error.\\nc', 'Nicole Doyle et al.,2022, DOI:https://doi.org/10.1001/jama.2022.20438\\nStudy Participants\\nA total of 978 of 1887 eligible participants were enrolled. Unavailability of blastocyst(s) for preimplantation genetic testing (n\\u2009=\\u200939) and all aneuploid embryos (n\\u2009=\\u2009115) accounted for the majority of prerandomization withdrawal. A total of 767 patients were randomized into the intervention group (n\\u2009=\\u2009381) and control group (n\\u2009=\\u2009386), respectively. Included in the analysis were 12 participants withdrawn after randomization (\\nFigure 1\\n); these patients were excluded from the per-protocol analysis. No patients were lost to follow-up and there were no missing outcome data.\\nAs shown in\\nTable 1', 'Nicole Doyle et al.,2022, DOI:https://doi.org/10.1001/jama.2022.20438\\nEligible participants were planning in vitro fertilization, preimplantation genetic testing for aneuploidy, and frozen embryo transfer, aged 30 to 40 years at time of egg retrieval and likely to produce at least 1 euploid blastocyst based on ovarian reserve testing. Exclusion criteria were surgically aspirated sperm or donor egg(s), recurrent implantation failure (>2 embryo transfers not resulting in ongoing pregnancy since the participant’s last live birth [if any]), recurrent pregnancy loss (≥2 clinical pregnancy losses without live birth), preimplantation genetic testing for monogenic disorders or structural rearrangements, unmitigated uterine cavity defect (normal uterine cavity on saline sonogram or hysterosalpingogram was required), body mass index (calculated as weight in kilograms divided by height in meters squared) greater than 40 at the start of in vitro fertilization cycle, current pregnancy or breastfeeding, and any contraindication to in vitro fertilization or pregnancy.', 'Saioa Torrealday et al.,2017, DOI:https://doi.org/10.12688/f1000research.11948.1\\nin vitro\\nfertilization (IVF) with preconception genetic diagnosing can minimize transgenerational transmission of identifiable genetic mutations in families that are deemed at an increased risk of passing genetic mutations to future progeny.\\nFor women with diagnosed or suspected POI for whom fertility is a priority, prompt referral to a reproductive endocrinologist is recommended. For those with established POI, while the opportunity for fertility preservation is missed, assisted conception by IVF using donor gametes or embryos provides the patient with the best option for biological parenting. Oocyte donation cycles generally result in high pregnancy rates given that the success rates are attributed to the age of the preselected and young oocyte donor. Alternatively, some couples prefer to pursue adoption to complete their families. All options should be offered to couples with a clear appreciation that this topic is sensitive and decisions should be individualized.', 'Evelyn Lee1 and Jinhui Zhang2 et al.,2022, DOI:https://doi.org/10.1186/s12913-022-08485-2\\nThree treatment strategies that are increasingly used  to augment ART in older women are preimplantation  genetic testing for aneuploidy (PGT-A), cryopreservation  of autologous oocytes at a younger age for potential later  use (‘social freezing’), and the use of donated oocytes  (donor ART).\\nPreimplantation genetic testing for aneuploidy in all  24 chromosomes using advanced genetic techniques has  been shown in observational studies and randomised  trials to improve implantation, clinical pregnancy and  live-birth rates in women of AMA [2, 10-12]. For example, Lee and colleagues reported older women (mean  age 40.1 years) who used PGT-A and blastomere aCGH  (array comparative genomic hybridization) analysis for  blastocyst transfer achieved a higher live birth rate using  fewer ART cycle compared with standard morphologic  embryo selection at the blastocyst stage (14.5% vs 9.1%  respectively) [2].', 'Zion Ben Rafael et al.,2021, DOI:https://doi.org/10.1093/hropen/hoab010\\nCozzolino\\net al.\\n,\\n2020\\n) has concluded that the use of the ERA test in 488 patients who underwent preimplantation genetic testing for aneuploidy (PGT-A), ERA or both, could not show any advantage to the use of ERA test. In short, ERA did not appear to improve outcomes in a group of patients who failed three or more ETs with a total of three or five embryos, and strangely PGT-A per se was effective in those failing three ETs, but was ineffective after failures of transfer of five embryos (\\nCozzolino\\net al.\\n, 2020\\n). Similarly, others (\\nNeves\\net al.\\n, 2019\\n) in a multivariate analysis\\nhave confirmed that performing an ERA test did not influence the pregnancy rate, in the tested euploid ET arm and was even associated with a diminished pregnancy rate in the donor ET arm. Taken together, these observations coupled with the lack of intra-patient consistency, bare witness to the randomity of the current state of research in the field.', 'Zion Ben Rafael et al.,2021, DOI:https://doi.org/10.1093/hropen/hoab010\\nCozzolino\\net al.\\n,\\n2020\\n) has concluded that the use of the ERA test in 488 patients who underwent preimplantation genetic testing for aneuploidy (PGT-A), ERA or both, could not show any advantage to the use of ERA test. In short, ERA did not appear to improve outcomes in a group of patients who failed three or more ETs with a total of three or five embryos, and strangely PGT-A per se was effective in those failing three ETs, but was ineffective after failures of transfer of five embryos (\\nCozzolino\\net al.\\n, 2020\\n). Similarly, others (\\nNeves\\net al.\\n, 2019\\n) in a multivariate analysis\\nhave confirmed that performing an ERA test did not influence the pregnancy rate, in the tested euploid ET arm and was even associated with a diminished pregnancy rate in the donor ET arm. Taken together, these observations coupled with the lack of intra-patient consistency, bare witness to the randomity of the current state of research in the field.']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "4",
         "Feedback: The response identifies older women as benefiting from PGT-A, which aligns with the reference answer regarding advanced maternal age. However, it includes additional details about age ranges and ovarian reserve, which, while potentially relevant, are not the primary focus of the question. The response is mostly correct.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "31",
         "What method was used for the analysis of urine cortisol?",
         "Urine cortisol was analyzed using a commercially available enzyme-linked immunosorbent assay kit.",
         "The residual urine specimens were transferred to the Frye Lab at the University at Albany (NY) in July, 2011 for cortisol analysis using a commercial enzyme-linked immunosorbent kit-based assay specific for measurement of urinary cortisol (Oxford Biomedical Research. Oxford, MI) (0).",
         "['Celeste D Butts et al.,2014, DOI:https://doi.org/10.1007/s10815-014-0359-0\\nWe collected fasting whole blood and urine specimens from women and non-fasting whole blood and urine specimens from male partners, on the day of oocyte retrieval. These were immediately processed, transferred into 1.8\\xa0mL cryovials, and frozen at−80\\xa0°C. Specimens were shipped on dry ice to the Laboratory of Inorganic and Nuclear Chemistry at the Wadsworth Center, New York State Department of Health (Albany, NY) in May, 2008 for measurement of toxic metals. Residual urine specimens were transferred to the Frye Lab at the University at Albany (NY) in July, 2011 for cortisol analysis using a commercial enzyme-linked immunosorbent kit-based assay specific for measurement of urinary cortisol (Oxford Biomedical Research. Oxford, MI). The current study comprised 52 women and 28 male partners for which cortisol levels were available.\\nIVF outcomes included: number of oocytes retrieved, oocyte maturity (defined as an oocyte retrieved in metaphase-2-arrest among\\nn', 'Pablo A Nepomnaschy et al.,2006, DOI:https://doi.org/10.1073/pnas.0511183103\\n63\\n–\\n67\\n). None of the participants smoked or consumed alcohol. To reduce the influence of the other confounding variables, participants were requested to collect their urine samples as soon as they woke up each morning, before they consumed food or performed any major physical activity. Sometimes women forgot to collect their first-morning urine before they began with their daily chores or consumed one of the substances mentioned above (besides tobacco or alcohol) before producing their sample. In such cases, the specimen was discarded (20.7% of 1,645 samples).\\nNulliparous women tend to present higher cortisol levels than parous women (\\n68\\n); our protocol, however, included only parous women. Cortisol levels may be affected by age (\\n69\\n), but in our sample, age was not associated with cortisol (mixed-model ANOVA,\\nP', 'Ferdinand Roelfsema et al.,2017, DOI:https://doi.org/10.1530/EC-17-0160\\n1\\n,\\n13\\n,\\n18\\n). Comparisons among studies, however, are hampered by different experimental designs, including number and spacing of blood samples, salivary samples or urinary excretion of cortisol and/or cortisol metabolites, size of groups, sex, BMI and age distributions. Blood sampling intensity in studies in which age was related to mean serum cortisol levels varied from 4 to 72 samples/24\\u2009h. The study of 145 samples/24\\u2009h could not demonstrate age effects on mean (median) cortisol concentrations or distinct cortisol secretion parameters. Three studies reported an increase in cortisol area under the curve (\\n5\\n,\\n6\\n,\\n13\\n), a fourth study a decrease in mean 24-h cortisol (\\n10\\n), and a fifth and sixth no influence of age (\\n10\\n,\\n11\\n,\\n27', 'Ferdinand Roelfsema et al.,2017, DOI:https://doi.org/10.1530/EC-17-0160\\nAverage, peak and nadir GH cortisol concentrations\\nMean (24-h and 3-h averages), median, absolute peak (single daily maximum) and absolute nadir (single daily minimum) cortisol concentrations were determined in each subject.\\nDeconvolution analysis', 'Celeste D Butts et al.,2014, DOI:https://doi.org/10.1007/s10815-014-0359-0\\n43\\n]. Hence, our study would have been better suited to an average of cortisol over repeated urine collections [\\n44\\n], or within a more restricted early morning time frame; however, conducting our study within the usual clinical context of IVF precluded this strategy. Still, a\\npost hoc\\npower analysis indicated that 2,762 women were needed to detect the cortisol difference between pregnant versus non-pregnant women. Thus, any potential undetected difference is likely to have been very modest and of questionable clinical relevance.\\nDespite the aforementioned limitations, our study has several strengths, including its prospective nature, the use of a widely accepted biomarker of psychological stress [\\n45\\n,\\n46', 'Ferdinand Roelfsema et al.,2017, DOI:https://doi.org/10.1530/EC-17-0160\\nProgressively reducing sampling intensity affected estimates of mean 24-h cortisol concentration, as portrayed in\\nFig. 5\\n. The squared correlation coefficient decreased from 0.998 (20 vs 10-min sampling) to 0.91 (120 vs 20-min sampling). Means of two samples per 24\\u2009h, when taken 12\\u2009h apart, yielded\\nR\\n2\\n=\\u20090.51, thus explaining 51% of the variability.\\nFigure 5.\\nOpen in a new tab\\nRelationship between the mean cortisol concentration determined from decreasing sampling interval (\\nx\\n-axis) and mean cortisol level based on 144 samples taken at 10\\u2009min intervals for 24\\u2009h. The squared correlation coefficients determined by linear regressions are shown on the\\ny\\n-axis.', 'Celeste D Butts et al.,2014, DOI:https://doi.org/10.1007/s10815-014-0359-0\\nOur study results should be interpreted cautiously given several limitations. Foremost, the limited sample size precluded simultaneous consideration of women and men, and moreover may have reduced statistical power to detect subtle associations. The QC data suggested increased variability at low cortisol concentrations, which could also have undermined study power secondary to exposure measurement misclassification error, non-differential by outcome. Furthermore, we collected spot urine samples between 9\\xa0AM and 1\\xa0PM, without a record of the precise collection times for each sample. As such, we were unable to adjust for diurnal variation, which is likely to have further compromised study power. Cortisol follows a circadian rhythm that peaks at approximately 8:30\\xa0AM and subsides as the day progresses, reaching a nadir at approximately 12:00\\xa0AM [\\n42\\n]. For example, average evening values were approximately 28\\xa0% of morning values in a study comparing one hour to 24\\xa0h urine collections [']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response accurately identifies the method used for urine cortisol analysis as a commercially available enzyme-linked immunosorbent assay kit. It also provides additional details about the specific kit and the lab where the analysis was performed.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "32",
         "Where is R-spondin2 exclusively expressed in ovarian follicles?",
         "R-spondin2 is exclusively expressed in oocytes of ovarian follicles.",
         "R-spondin2 is exclusively expressed in oocytes of ovarian follicles (3).",
         "['Yuan Cheng et al.,2013, DOI:https://doi.org/10.1096/fj.12-223412\\nExpression of R-spondin2 in oocytes of primary and larger follicles\\nAs shown in\\nFig. 1\\nA\\n, ovaries from neonatal mice at d 3 of age contained multiple primordial follicles in the ovarian cortex and some primary follicles migrated into the medulla region (\\nFig. 1\\nAa\\n).\\nIn situ\\nhybridization analyses showed R-spondin2 mRNA signals in oocytes of primary follicles but not in cortical primordial follicles (\\nFig. 1\\nAb\\n). In ovaries from juvenile mice at d 15 of age, follicles at primary, secondary, and early antral stages were present (\\nFig. 1\\nAc\\n), and R-spondin2 signals were found in oocytes of primary and larger follicles (\\nFig. 1\\nAd\\n). In ovaries of prepubertal mice (d 23 of age), the largest follicles reached the antral stage (\\nFig. 1\\nAe\\n). Again, R-spondin2 expression was found in oocytes of primary and larger follicles (\\nFig. 1\\nAf\\n). No signals were found for sections with the sense probes (data not shown).\\nFigure 1.\\nOpen in a new tab', 'Yuan Cheng et al.,2013, DOI:https://doi.org/10.1096/fj.12-223412\\nFig. 2\\nB\\n, treatment of ovarian explants with R-spondin 2 increased ovarian weight in a dose-dependent manner, with further increases when both R-spondin2 and FSH were added together. In contrast, cotreatment with Wnt3a and R-spondin2 did not further increase R-spondin2-stimulated weight gain, suggesting actions of endogenous Wnt factors. Furthermore, cotreatment with the antagonist DKK1 inhibited basal and R-spondin2-stimulated ovarian weight gain (\\nFig. 2\\nC\\n). To neutralize endogenous R-spondin2 actions, some ovarian explants were incubated with affinity-purified R-spondin2 antibodies (\\nFig. 2\\nC\\n). Treatment with R-spondin2 antibodies decreased basal ovarian weights, whereas treatment with nonimmune IgG or boiled R-spondin2 antibodies were ineffective, suggesting a role of endogenous R-spondin2 in follicle growth. Morphological analyses demonstrated the promotion of ovarian follicle development following treatment with either R-spondin2 and/or FSH (\\nFig. 2\\nD', 'Yuan Cheng et al.,2013, DOI:https://doi.org/10.1096/fj.12-223412\\nIn vivo\\nadministration of an R-spondin agonist stimulated the development of primary follicles to the antral stage in both immature mice and gonadotropin releasing hormone antagonist-treated adult mice. Subsequent treatment with gonadotropins allowed the generation of mature oocytes capable of undergoing early embryonic development and successful pregnancy. Furthermore, R-spondin agonist treatment of immune-deficient mice grafted with human cortical fragments stimulated the development of primary follicles to the secondary stage. Thus, oocyte-derived R-spondin2 is a paracrine factor essential for primary follicle development, and R-spondin agonists could provide a new treatment regimen for infertile women with low responses to the traditional gonadotropin therapy.—Cheng, Y., Kawamura, K., Takae, S., Deguchi, M., Yang, Q., Kuo, C., Hsueh, A. J. W. Oocyte-derived R-spondin2 promotes ovarian follicle development.', 'Yuan Cheng et al.,2013, DOI:https://doi.org/10.1096/fj.12-223412\\nR-spondin proteins are adult stem cell growth factors capable of stimulating gut development by activating LGR4, 5, and 6 receptors to promote Wnt signaling. Although multiple Wnt ligands and cognate Frizzled receptors are expressed in the ovary, their physiological roles are unclear. Based on bioinformatic and\\nin situ\\nhybridization analyses, we demonstrated the exclusive expression of\\nR-spondin\\n2 in oocytes of ovarian follicles. In cultured somatic cells from preantral follicles, R-spondin2 treatment (ED\\n50\\n: 3 ng/ml) synergized with Wnt3a to stimulate Wnt signaling. In cultured ovarian explants from prepubertal mice containing preantral follicles, treatment with R-spondin2, similar to follicle stimulating hormone, promoted the development of primary follicles to the secondary stage.\\nIn vivo', 'Yuan Cheng et al.,2013, DOI:https://doi.org/10.1096/fj.12-223412\\nP\\n<0.05).\\nC\\n) Antagonistic effects of DKK1 and R-spondin2 antibodies on basal and R-spondin2-stimulated ovarian weight. Ovarian explants were treated with different reagents for 4 d. Treatment with nonimmune IgG (500 ng/ml) and boiled R-spondin2 antibodies (500 ng/ml) served as controls. Ab, antibodies; R-spo2, R-spondin2.\\nn\\n= 9–18. *\\nP\\n< 0.05\\nvs\\n. control or as indicated.\\nD\\n) Histology of ovaries treated with R-spondin2 and/or FSH. Arrowheads indicate secondary follicles. Scale bars = 100 μm.\\nE\\n) Follicle dynamics following treatment with R-spondin2, FSH, and Wnt3a. Serial sections of ovarian explants were used for detailed counting of follicles at different developmental stages. R-spondin2, 10 ng/ml; FSH, 25 ng/ml; Wnt3a, 3 ng/ml.\\nn\\n= 5–10. *\\nP\\n< 0.05\\nvs\\n. control.\\nOvaries from d 10 mice were cultured with different factors for 4 d with medium changes at d 2 of culture. As shown in\\nFig. 2\\nB', 'Yuan Cheng et al.,2013, DOI:https://doi.org/10.1096/fj.12-223412\\nFigure 1.\\nOpen in a new tab\\nExpression of R-spondin2 together with different Wnt ligands, as well as Frizzled and LGR receptors, in different ovarian cell types.\\nA\\n)\\nIn situ\\nhybridization analyses of R-spondin2 transcripts in ovaries of neonatal (d 3,\\na\\n,\\nb\\n), juvenile (d 15,\\nc\\n,\\nd\\n), and prepubertal (d 23,\\ne\\n,\\nf\\n) mice were analyzed.\\na\\n,\\nc\\n,\\ne\\n) Bright-field images.\\nb\\n,\\nd\\n,\\nf\\n) Dark-field images. Scale bars = 100 μm. Arrowheads indicate oocytes.\\nB\\n) Real-time RT-PCR analyses of R-spondin2 transcripts in isolated oocytes and somatic cells of preantral follicles. Ovaries from immature mice at 10 d of age containing secondary and smaller follicles were dissociated to obtain oocytes and somatic cells. The purity of these cells was confirmed using specific markers (GDF9 for oocyte, FSH receptor for somatic cells). For all Wnt genes, negligible expression was found in oocytes. For FZD genes, negligible expression was found for FZD2, 4, 6, and 9.\\nn\\n= 6.\\nC', 'Yuan Cheng et al.,2013, DOI:https://doi.org/10.1096/fj.12-223412\\nFig. 2\\nD\\n). Histological analyses were performed using serial sections to evaluate follicles at different developmental stages. Although negligible changes in ratios of primordial and degenerated follicles were detected after treatment with R-spondin2, we found decreases in the percentage of primary follicles accompanied by increases in the percentage of secondary follicles (\\nFig. 2\\nE\\n). In contrast, FSH or Wnt3a treatment did not affect primordial, primary, or secondary follicle numbers but promoted early secondary follicles to the late secondary stage (data not shown). When explants were treated with either Wnt3a or FSH, together with R-spondin2 also stimulated the growth of primary follicles to the secondary stage.\\nIn vivo\\ntreatment with R-spondin1-Fc promoted follicle development to allow gonadotropin stimulation of ovulation and early embryonic development\\nBecause R-spondin1–3 are equipotent in stimulating the canonical Wnt signaling pathway (\\n14\\n), we used R-spondin1-Fc for']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response completely and accurately answers the question, matching the reference answer exactly.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "33",
         "How much was estradiol concentration increased in obese women compared to the lightest women?",
         "Estradiol concentration was increased by 130% in obese women compared with the lightest women.",
         "Estradiol concentration was increased by 130% in obese women compared with the lightest women (p trend = 0.002) (Document 0, 5).",
         "['Anne McTiernan et al.,2003, DOI:https://doi.org/10.1200/JCO.2003.07.057\\n). Estradiol concentration was increased by 130% in obese women compared with the lightest women (p trend = 0.002). Increasing adiposity was associated with increasing testosterone concentrations; obese women had testosterone concentrations that were almost twice as high as those of the lightest women (p trend, 0.0001). Concentrations of free estradiol and free testosterone were doubled to tripled in overweight and obese women compared with the lightest women (p trend=0.0001). Concentrations of SHBG decreased with increasing BMI; obese women had an average SHBG concentration that was half that of the women with BMI < 22 (p trend, p= 0.0001). Concentrations of DHEAS increased with increasing adiposity, but the test for trend was not statistically significant. When we categorized women by the common cutpoints for “normal” (BMI < 25.0), “overweight” (BMI 25.0–29.9), and “obese” (BMI ≥ 30', 'Anne McTiernan et al.,2003, DOI:https://doi.org/10.1200/JCO.2003.07.057\\n.0), “overweight” (BMI 25.0–29.9), and “obese” (BMI ≥ 30.0), a similar gradient of increasing estrogen and decreasing SHBG concentrations was seen, and the results for individual hormones were similarly statistically significant as for the data categorized by the more refined categories (data not shown).', 'Anne McTiernan et al.,2003, DOI:https://doi.org/10.1200/JCO.2003.07.057\\nData Analysis\\nWe categorized BMI (kg/m\\n2\\n) as light (< 22.0), normal weight (22.0 ≤ BMI ≤ 24.9), moderately overweight (25.0 ≤ BMI ≤ 27.5), severely overweight (27.6 ≤ BMI ≤ 29.9), or obese (BMI ≥ 30.0). We also categorized BMI using the World Health Organization public health cutpoints for obese (BMI ≥ 30.0), overweight (BMI ≥ 25.0) and normal or underweight (BMI < 25.0).\\n8\\nWe categorized percent body fat, waist circumference and waist-to-hip ratio into quartiles. We calculated total fat mass by multiplying DEXA-derived percent body fat by weight, and divided participants into quartiles of this variable.\\nWe calculated free estradiol and free testosterone concentrations (unbound to either SHBG or albumin) using values for estradiol, testosterone, and SHBG with the equations of Sodergard et al..\\n9', 'Hannah P Yang et al.,2011, DOI:https://doi.org/10.1016/j.annepidem.2011.01.003\\nOur analysis demonstrates that concentration and mass of estrogens among postmenopausal control women are highly correlated and relate similarly to BMI. However, as BMI increases, total body mass of hormones rises more steeply than concentration. In contrast to relationships for estrogens, higher BMI was related only to androstenedione mass, but not to concentration. This may represent a chance finding, but could indicate that concentration and mass associations with BMI vary by analyte, reflecting differences in homeostatic mechanisms, hormone interconversions or other factors. In particular, conversion of androgens to estrogens in peripheral adipose tissue may blunt the rise in androgen concentrations relative to total body mass among heavier women.\\nThe mathematical conversion of concentration to mass is derived from computation of body surface area and plasma volume (\\n13', 'Hannah P Yang et al.,2011, DOI:https://doi.org/10.1016/j.annepidem.2011.01.003\\n13\\n). We acknowledge that the masses of the sex steroid hormones were calculated and not measured. However, we suspect that our estimation is based on a reasonable formula and assessment using an alternative method (\\n20\\n) yielded the same interpretation. Estimation of plasma volume is more strongly related to height than weight, raising the possibility that differences in body habitus might differentially influence total body mass of analytes. Recently, Baglietto et al reported that fat mass was the strongest anthropometric determinant of estradiol concentration, whereas waist circumference was the best indicator of estrone sulfate and androstenedione (\\n2\\n). Given that anthropometry adds information about the variance in hormone concentrations, continued assessment of different hormone metrics may also be warranted.', 'Anne McTiernan et al.,2003, DOI:https://doi.org/10.1200/JCO.2003.07.057\\nResults\\nObese women (BMI ≥ 30) had 35% higher concentrations of estrone and 130% higher concentrations of estradiol, compared with lighter women (BMI < 22.0) (p trend, 0.005 and 0.002, respectively). Similar associations were observed for body fat mass, percent body fat and waist circumference. Testosterone concentrations also increased with increasing levels of adiposity (p trend, 0.0001). Concentrations of free estradiol and free testosterone were doubled to tripled in overweight and obese women compared with lighter-weight women (p trend=0.0001).\\nConclusions\\nThese data provide information about potential hormonal explanations for the association between adiposity and breast cancer prognosis. These sex hormones may be useful biomarkers for weight loss intervention studies in women with breast cancer.', 'Hannah P Yang et al.,2011, DOI:https://doi.org/10.1016/j.annepidem.2011.01.003\\n4\\n). However, an alternative proposal is that values for hormone concentration and mass may diverge among obese women secondary to expanded blood volume, and that mass may be a better marker of risk than concentration.\\nSex steroid hormone levels are highly regulated among normal cycling women through complex feedback loops involving the ovaries and hypothalamic-pituitary axis. After menopause, these homeostatic mechanisms are no longer operative, and excess hormone production among obese women may lead to increased endometrial cancer risk. Studies demonstrate that BMI and vascular volume are positively correlated, although relationships vary by body habitus (\\n9\\n–\\n10']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response accurately states that estradiol concentration was increased by 130% in obese women compared with the lightest women, which aligns with the reference answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "34",
         "What is considered the normal range of CGG nucleotide repeats for the FMR1 gene's ovarian function?",
         "The normal range for the FMR1 gene's ovarian function is 26 to 34 CGG nucleotide repeats.",
         "The normal range for CGG nucleotide repeats is 26 to 34 (median 30) for the FMR1 gene's ovarian function [3].",
         "['Norbert Gleicher et al.,2011, DOI:https://doi.org/10.1186/1477-7827-9-23\\nControlling OR Wallace and Kelsey recently reaffirmed that ovarian aging varies between individuals [12]. Using age at menopause as end points, they determined that speed of follicle recruitment and follicle numbers vary significantly at different stages of life. We have come to similar conclusions, recently describing effects of theFMR1 gene on the ovary [46]. Evaluations of the FMR1 gene have become increasingly popular because of the gene ’s neuro/psychiatric effects [47]. It, however, also, independently, demonstrates specific ovarian effects [46]. Increased risk for POF/POI in women with premutation-range (ca. 55-200) CGG triple nucleotide repeats has been known for decades [47] but that even lower CGG repeat numbers may also denote risk towards premature ovarian senescence, at times representing milder forms of so-called POA/OPO I, is a more recent discovery [48-51]', 'Norbert Gleicher et al.,2011, DOI:https://doi.org/10.1186/1477-7827-9-23\\n. FMR1 can, therefore, be viewed as an“ovarian aging gene.” Norm women at younger ages appear to recruit actively and, therefore, likely deplete TOR quicker than het and hom females, who from young age on recruit at much slower pace. The latter two genotypes, therefore, demonstrate much slower and steadier declines in AMH levels. Still prelimin a r yd a t as u g g e s tt h a thet patients can be further subdivided into het-norm/high and hetnorm/low sub-genotypes (depending on whether the abnormal allele is abnormally high or low), which may further differ in respective aging curves [52,53]. These observations confirm statistical associations between FMR1 genotypes (based on normal CGG range of 26 to 34) and FOR, based on FOR assessment with AMH. They, thus, suggest significant relevance of triple CGG repeat counts to clinical practice [49,50], and that FMR1 genotypes, within reasons, already at young ages allow predictions about “ ovarian aging ” patterns', 'Norbert Gleicher et al.,2011, DOI:https://doi.org/10.1186/1477-7827-9-23\\n. Chen and associates reported that maximal translation of the gene product and that the switching point between positive and negative effects of the gene occurs at 30 CGG repeats [57]. This exactly reflects the median of the normal range, reported for the gene ’so v a r i a n function [46,58]. This normal range of 26 to 34 repeats also contains at midpoint the tall distribution peak of CGG repeats in the general population, reported by Fu and associates at 29 to 30 repeats [59]. Combined, these observations suggest that FMRP, the gene product ofFMR1,a n d / o ri t s translation, may play a role in follicular recruitment and “ovarian aging.” Mouse models have allowed progress in understanding follicular recruitment and, by extension, OR regulation. We noted earlier the important recent paper by Reddy et al [4]. In humans, this area, however, still largely represents a black box [3], resulting in many models of “ ovarian aging ” [11,12,43] but little factual data', 'Norbert Gleicher et al.,2011, DOI:https://doi.org/10.1186/1477-7827-9-23\\n. The genotypes relating to ovarian function are distinctly different from the genotypes historically reported to define the gene’s neuro/psychiatric risks. In regard to ovarian function, 26 to 34 CGG nucleotide repeats represent normal (median 30), independent of ethnicity/ race [49]. Using this range to define genotypes, women can be designated as normal (norm), when both alleles are in normal range,heterozygous (het) if one is normal and the other abnormal and homozygous (hom) if both alleles are outside normal range. Figure 2 demonstrates linear regressions of AMH over age, depending on whether women are norm , hetabnormal or hom- abnormal: Depending on FMR1 genotype, ovarian aging patterns differ. Before physiologic ovarian aging significantly contributes to OR at young ages, differences are most obvious. As expected, norm women demonstrate better OR than het females, with hom women demonstrating the lowest OR', 'Ewa Rudnicka et al.,2018, DOI:https://doi.org/10.5114/pm.2018.78550\\nOn the other hand, the presence of X trisomy may  also result in ovarian dysfunction [12, 13].\\nGene abnormalities can pertain to mutation on X as well as on autosomal chromosomes. The most common is FMR1 premutation located  on X chromosome. The mutation rests upon increased  expansion of CGG triplets in the untranslated region.  Patients with FMR1 premutation (CGG present 55-199  times) are at increased risk of developing POI [12]. The  incidence of ovarian dysfunction among carriers reaches 12.9% [14]. Full mutations characterised by more  than 200 repetitions inactivate the gene by methylation  and lead to fragile X syndrome, and are also the most  common inherited cause of disability among men [15].', 'Norbert Gleicher et al.,2011, DOI:https://doi.org/10.1186/1477-7827-9-23\\n. Described FMR1 genotypes raise additional questions about these proposed human models of ovarian aging normal ovarian aging, quite obviously, has to be defined separately for individualFMR1 genotypes. In some aspects, Wallace and Kelsey ’s model [12] appears superior to others since it detected rapid declines in OR at younger ages than Faddy and coworkers, who reported an accelerated declines only at age 37 to 38 years [43]. Considering varying “ ovarian aging ” patterns of different FMR1 genotypes [46], Faddy and Gosden ’s later timing, however becomes understandable as norm women demonstrate accelerated declines in FOR at approximately age 35 years (Figure 2), while het and hom females demonstrate a more gradual decrease. Math ematically combined, all three genotypes, indeed, resemble Faddy and Gosden’s model. Ultimate purposes of FMR1 genotypes remain to be determined', 'Norbert Gleicher et al.,2011, DOI:https://doi.org/10.1186/1477-7827-9-23\\n. Paradoxically, d ue to rapid recruitment at young ages, the norm population appears at greatest risk for early follicle depletion and, possibly, early menopause. Whether early depletion can be equated with early menopause has, however, so far not been established. FMR1 , thus, is involved in regulating FOR but may also affect TOR by regulating follicular recruitment. How FMR1 actually regulates physiologic functions has remained controversial: In association with neuro/psychiatric consequences, the number of CGG triple repeats is a determining factor. Risks increase with increasing expansion sizes up to approximately the lower half of the so-called premutation range (ca. 55-200 repeats), though not necessarily in linear fashion [47,54]. The gene codes for an RNA-binding protein, the so-called fragile X mental retardation protein (FMRP) [55], important in synaptic physiology and apparently demonstrating RNA toxicity [56]']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response accurately states the normal range of CGG nucleotide repeats for the FMR1 gene's ovarian function as 26 to 34. It includes additional information (median 30) which does not detract from the correctness.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "35",
         "What methods does clusterProfiler offer for gene classification and enrichment analyses?",
         "clusterProfiler offers groupGO, enrichGO, and enrichKEGG.",
         "The clusterProfiler package offers a gene classification method, namely groupGO, to classify genes based on their projection at a specific level of the GO corpus, and provides functions, enrichGO and enrichKEGG, to calculate enrichment test for GO terms and KEGG pathways based on hypergeometric distribution (1).",
         "['Guangchuang Yu et al.,2012, DOI:https://doi.org/10.1089/omi.2011.0118\\nIncreasing quantitative data generated from transcriptomics and proteomics require integrative strategies for analysis. Here, we present an R package, clusterProfiler that automates the process of biological-term classification and the enrichment analysis of gene clusters. The analysis module and visualization module were combined into a reusable workflow. Currently, clusterProfiler supports three species, including humans, mice, and yeast. Methods provided in this package can be easily extended to other species and ontologies. The clusterProfiler package is released under Artistic-2.0 License within Bioconductor project. The source code and vignette are freely available at\\nhttp://bioconductor.org/packages/release/bioc/html/clusterProfiler.html', 'Guangchuang Yu et al.,2012, DOI:https://doi.org/10.1089/omi.2011.0118\\nThe clusterProfiler was implemented in R, an open-source programming environment (Ihaka and Gentleman,\\n1996\\n), and was released under Artistic License 2.0 within Bioconductor project (Gentleman et al.,\\n2004\\n). The clusterProfiler package depends on the Bioconductor annotation data GO.db and KEGG.db to obtain the maps of the entire GO and KEGG corpus. Bioconductor annotation packages org.Hs.eg.db, org.Mm.eg.db, and org.Sc.sgd.db were imported for genome-wide annotation of mapping Entrez gene identifiers or ORF identifiers for humans, mice, and yeast, respectively.\\nThe clusterProfiler package offers a gene classification method, namely\\ngroupGO\\n, to classify genes based on their projection at a specific level of the GO corpus, and provides functions,\\nenrichGO\\nand\\nenrichKEGG\\n, to calculate enrichment test for GO terms and KEGG pathways based on hypergeometric distribution. To prevent high false discovery rate (FDR) in multiple testing,\\nq\\n-values (Storey,\\n2002', 'Guangchuang Yu et al.,2012, DOI:https://doi.org/10.1089/omi.2011.0118\\nThe clusterProfiler package is implemented for gene cluster comparison. It is not limited to gene clusters obtained from gene expression data, but can also be applied to gene clusters from other approaches, such as PPI modules and miRNA target genes (Yu and He,\\n2011\\n).\\nTo illustrate how clusterProfiler assesses and compares biological themes among gene clusters, we analyzed the publicly available expression dataset of breast tumour tissues from 200 patients (\\nGSE11121\\n, Gene Expression Omnibus) (Schmidt et al.,\\n2008\\n). We first identified 5230 differentially expressed genes (DEGs) using the significant analysis of microarray (SAM) algorithm (Schwender and Ickstadt,\\n2008\\n), with cutoff of\\np\\n-values <0.01 and FDR <0.05, and then identified eight gene clusters from these DEGs, which have different expression patterns across samples, using soft clustering algorithm (Kumar and Futschik,\\n2007', 'Guangchuang Yu et al.,2012, DOI:https://doi.org/10.1089/omi.2011.0118\\n2009\\n) for detecting enriched pathways.\\nAlthough these tools can automatically calculate the statistically significant categories, the manual step to choose interesting clusters followed by enrichment analysis on each selected cluster is still slow and tedious. Therefore, several new tools such as ClueGO (Bindea et al.,\\n2009\\n) and goProfiles (Sánchez et al.,\\n2007\\n) were developed to partially resolve this problem. However, they only support comparison of two clusters of genes. TM4 MultiExperiment Viewer (MeV) (Saeed et al.,\\n2003\\n), which implements the Tree-EASE (TEASE) algorithm by combining hierarchical clustering with EASE (Hosack et al.,\\n2003\\n), can perform clustering analysis followed by GO enrichment calculation, but without tools designed for comparing and visualizing functional differences among clusters. Here, we present an R package called clusterProfiler for statistical analysis of GO and KEGG, allowing biological theme comparison among gene clusters.', 'Guangchuang Yu et al.,2012, DOI:https://doi.org/10.1089/omi.2011.0118\\nWe plan to employ three strategies to improve this package in the future. First, we will use semantic similarity among KEGG pathways and GO terms to aggregate closely related categories. This can be anticipated to result in more interpretable outcomes. Second, we will rank gene similarities in each cluster and then correlate them to search for enriched categories. This will provide more sensibility for finding active gene modules. Third, we will develop a statistical model based on the induced directed acyclic graph to compare functional profiles as a whole rather than a set of unrelated categories. These strategies will increase the versatility of clusterProfiler.', 'Guangchuang Yu et al.,2012, DOI:https://doi.org/10.1089/omi.2011.0118\\nWe plan to employ three strategies to improve this package in the future. First, we will use semantic similarity among KEGG pathways and GO terms to aggregate closely related categories. This can be anticipated to result in more interpretable outcomes. Second, we will rank gene similarities in each cluster and then correlate them to search for enriched categories. This will provide more sensibility for finding active gene modules. Third, we will develop a statistical model based on the induced directed acyclic graph to compare functional profiles as a whole rather than a set of unrelated categories. These strategies will increase the versatility of clusterProfiler.', 'Guangchuang Yu et al.,2012, DOI:https://doi.org/10.1089/omi.2011.0118\\n2007\\n). Finally, we used clusterProfiler to compare these gene clusters by their enriched biological processes, with the strict cutoff of\\np\\n-values <0.01 and FDR <0.05.\\nAs illustrated in\\nFigure 1\\n, clusterProfiler pinpointed different clusters of DEGs related to different biological processes, such as cluster 2 and cluster 5 related to cellular component organization, cluster 3 related to developmental process, and cluster 6 related to cell cycle.\\nFIG. 1.\\nOpen in a new tab\\nComparison of GO enrichment of gene clusters.\\nDots represent term enrichment with color coding: red indicates high enrichment, blue indicates low enrichment. The sizes of the dots represent the percentage of each row (GO category).\\nThese results are consistent with the main findings of the previous studies, showing that high expression of proliferation-associated genes confers a good prognosis in breast cancer (Schmidt et al.,\\n2008\\n). This analysis provides valuable information for further investigations.']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response correctly identifies groupGO, enrichGO, and enrichKEGG as methods offered by clusterProfiler. It also accurately describes their functions.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "36",
         "What parameters does the NetworkAnalyzer plugin in Cytoscape calculate?",
         "The NetworkAnalyzer plugin calculates the degree of vertices, vertex betweenness centrality, vertex clustering coefficient, and edge betweenness.",
         "NetworkAnalyzer computes various topological network parameters for undirected and/ or directed networks [0].",
         "['Chia-Hao Chin et al.,2014, DOI:https://doi.org/10.1186/1752-0509-8-S4-S11\\n. Cytoscape [2] is an open platform with many plugins to expand both the visualization options and the network analysis power. Via Cytoscape, the graphical view of a network is easy accessed, and multiple layers of information including large-scale, genome-wise experiments, and protein function annotations can be granted on the interactome. Several Cytoscape plugins can score and rank the nodes by network features. For example, NetworkAnalyzer [3] and CentiScaPe [4] computes various topological network parameters for undirected and/ or directed networks. These plugins provide more centrality measures than other commonly used, but some other important features and recent developed methods are not included. Different methods focus on different topological features, or similar features with different scoring strategies', 'Albert Batushansky et al.,2016, DOI:https://doi.org/10.1155/2016/8313272\\nNowadays, there exist a number of software tools that allow researchers to generate networks, visualize them, and analyze their structure, via the calculation of a number of network properties, based on their own experimental data. Commonly known tools are Cytoscape [\\n24\\n], Gephi [\\n25\\n], and iGraph [\\n26', \"Albert Batushansky et al.,2016, DOI:https://doi.org/10.1155/2016/8313272\\n29\\n]. The selected significant correlation values can be easily converted to a table, listing in three columns the vertices that are adjacent to each other. This table is subsequently used as a template to illustrate the network using Cytoscape. We have chosen Cytoscape out of the list of network software as it was specifically developed for biological data, because of its intuitively understandable interface, wide range of visualization options, and available additional plugins for calculations of the main network properties. The method's workflow is presented in\\nFigure 1\\n.\\nFigure 1.\\nOpen in a new tab\\nCorrelation-based network, pipeline flowchart.\\n2.1. Method Pipeline\\n2.1.1. Download R-Environment and Required R-Packages\\nTo start the workflow, first download and install the latest version of R-environment from the following website:\\nhttps://www.r-project.org/\\n. For the processes described here two R-packages will be used: “psych” [\\n28\\n] and “reshape2” [\\n30\", 'Albert Batushansky et al.,2016, DOI:https://doi.org/10.1155/2016/8313272\\n=\\nif\\nX\\n=\\n1\\n,\\nconcatenate\\nY\\n,\\n“\\n−\\n”\\n,\\nZ\\n,\\n“\\n”\\n.\\n(2)\\nIn this formula\\nX\\nis a number of 1st cell of the obtained column (usually 3rd) on the current sheet;\\nY\\nand\\nZ\\nare the numbers of 1st cell in the 1st and 2nd column on the current sheet, respectively. At this stage copy 1st, 2nd, and 4th columns as the values to the new sheet, filter out and remove rows with empty last cell, and save the obtained fully filled three-column table in\\u2009\\u2009.\\ntxt tab delimited format\\n.\\nThis file can be imported as a network to the Cytoscape software and analyzed using the built-in NetworkAnalyzer plugin. To import the file run the Cytoscape and select “\\nimport\\n” from the main menu bar, then locate the previously saved three-column file in\\u2009\\u2009.\\ntxt\\nformat, and import it as\\n“table”\\n(Supplementary Figure\\u2009\\u20092). To run the plugin locate\\nNetworlAnalyzer\\nin the Tool menu and execute\\n“Analyze Network', 'Chia-Hao Chin et al.,2014, DOI:https://doi.org/10.1186/1752-0509-8-S4-S11\\nIn this study, we implement our network scoring methods, MCC, MNC and DMNC, and eight other popular methods into a Cytoscape plugin, cytoHubba .T h r o u g h the extendable, flexible an d modulated properties of Cytoscape, cytoHubba can work together with other plugins. The computing processes had been optimized and can complete all eleven analysis on a common desktop/ notebook in a reasonable time cost. We also improve the network retrieving function in cytoHubba control panel. Therefore, users can utilize a PPIs network from public domain and extract sub-networks based on users’ domain-knowledge. Among the 11 methods, the newly proposed method MCC performs better than the others. MCC captures more essential proteins in the top ranked list in both high-degree and low-degree proteins. Another method, DMNC, catches different set of essential proteins suggesting it scores the network in different way', 'Chia-Hao Chin et al.,2014, DOI:https://doi.org/10.1186/1752-0509-8-S4-S11\\nConclusions: CytoHubba provide a user-friendly interface to explore important nodes in biological networks. It computes all eleven methods in one stop shopping way. Besides, researchers are able to combinecytoHubba with and other plugins into a novel analysis scheme. The network and sub-networks caught by this topological analysis strategy will lead to new insights on essential regulatory networks and protein drug targets for experimental biologists. According to cytoscape plugin download statistics, the accumulated number ofcytoHubba is around 6,700 times since 2010.', 'Albert Batushansky et al.,2016, DOI:https://doi.org/10.1155/2016/8313272\\nThe current proposed stepwise pipeline allows the user to control each step of the network creation, as it provides flexibility in selection of correlation methods and thresholds and describes easy-to-handle options to analyze the network topology. The pipeline works irrespective of the nature of the data set and can be implemented by a combined use of the freely distributed Apache OpenOffice software (\\nhttp://www.openoffice.org/\\n), built-in packages within the R-environment [\\n27\\n], and Cytoscape [\\n24\\n].']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "2",
         "The response is too general and doesn't specify which parameters are calculated. It lacks the specific details provided in the reference answer, such as degree of vertices, betweenness centrality, clustering coefficient, and edge betweenness.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "37",
         "What is maSigPro-GLM suitable for identifying?",
         "maSigPro-GLM is suitable for the identification of DEGs from time-course RNA-seq data.",
         "maSigPro-GLM is suitable for the identification of DEGs from time-course RNA-seq data under a wide range of experimental settings (0).",
         "['María José Nueda et al.,2014, DOI:https://doi.org/10.1093/bioinformatics/btu333\\nIn conclusion, we show that maSigPro-GLM is suitable for the identification of DEGs from time-course RNA-seq data under a wide range of experimental settings. The updated package successfully controls both false-positive and false-negative detection rates.', 'María José Nueda et al.,2014, DOI:https://doi.org/10.1093/bioinformatics/btu333\\nFNR is properly controlled both by maSigPro-GLM and edgeR. This last method has a zero false call rate in our simulations, whereas maSigPro-GLM shows FNR <1%. Results were basically similar considering one or two series and different expression patterns for DEGs.\\nFig. 1.\\nOpen in a new tab\\nFDR and FNR for maSigPro-GLM at different levels of\\nR\\n2\\nwith 1 and 2 series\\nTable 2.\\nSimulated experiments results with scenarios A, B and C for maSigPro-LM, maSigPro-GLM and edgeR\\n(Scenario) # Series\\nRep\\nmaSigPro-LM\\nmaSigPro-GLM\\nedgeR\\nSel\\nFP\\nFN\\nSel\\nFP\\nFN\\nSel\\nFP\\nFN\\n(A)\\n1 Series\\n1\\n0\\n0\\n999\\n2210\\n1496\\n286\\n2\\n533\\n25\\n492\\n976\\n52\\n76\\n1135\\n135\\n0\\n3\\n589\\n5\\n416\\n975\\n2\\n27\\n1173\\n173\\n0\\n5\\n515\\n0\\n485\\n997\\n0\\n3\\n1170\\n170\\n0\\n2 Series\\n1\\n471\\n34\\n563\\n1969\\n972\\n3\\n2\\n981\\n5\\n24\\n1001\\n1\\n0\\n1267\\n267\\n0\\n3\\n985\\n1\\n16\\n1000\\n0\\n0\\n1278\\n278\\n0\\n5\\n995\\n0\\n5\\n1000\\n0\\n0\\n1219\\n219\\n0\\n(B)\\n1 Series\\n1\\n0\\n0\\n1000\\n1592\\n741\\n149\\n2\\n723\\n46\\n323\\n990\\n34\\n44\\n1158\\n158\\n0\\n3\\n750\\n2\\n252\\n978\\n1\\n23\\n1155\\n155\\n0\\n5\\n751\\n0\\n249\\n994\\n0\\n6\\n1136\\n136\\n0\\n2 Series\\n1\\n253\\n14\\n761\\n1351\\n411\\n60\\n2\\n672\\n4\\n332\\n951', 'María José Nueda et al.,2014, DOI:https://doi.org/10.1093/bioinformatics/btu333\\n1\\n23\\n1155\\n155\\n0\\n5\\n751\\n0\\n249\\n994\\n0\\n6\\n1136\\n136\\n0\\n2 Series\\n1\\n253\\n14\\n761\\n1351\\n411\\n60\\n2\\n672\\n4\\n332\\n951\\n1\\n50\\n1240\\n240\\n0\\n3\\n592\\n0\\n408\\n963\\n0\\n37\\n1225\\n225\\n0\\n5\\n538\\n0\\n462\\n978\\n0\\n22\\n1138\\n138\\n0\\n(C)\\n1 Series\\n1\\n0\\n0\\n1000\\n1427\\n764\\n337\\n2\\n284\\n14\\n730\\n972\\n37\\n65\\n1166\\n166\\n0\\n3\\n433\\n3\\n570\\n945\\n0\\n55\\n1125\\n125\\n0\\n5\\n357\\n0\\n643\\n963\\n0\\n37\\n1134\\n134\\n0\\n2 Series\\n1\\n222\\n12\\n790\\n1458\\n471\\n13\\n2\\n684\\n9\\n325\\n996\\n2\\n6\\n1284\\n284\\n0\\n3\\n378\\n0\\n322\\n999\\n0\\n1\\n1201\\n201\\n0\\n5\\n681\\n0\\n319\\n998\\n0\\n2\\n1209\\n209\\n0\\nOpen in a new tab\\nNote\\n: Number of replicates (Rep), number of selected genes (Sel), false positives (FP) and false negatives (FN).\\n3.2 Experimental study\\nWe applied both edgeR and maSigPro-GLM to the\\nA\\n.\\nthaliana\\ntime-course data considering the two series defined by the Bgh isolate. An\\nR\\n2\\nthreshold of 0.5 was chosen for the second maSigPro-GLM step, according to the results presented in\\nFigure 1', 'María José Nueda et al.,2014, DOI:https://doi.org/10.1093/bioinformatics/btu333\\nFigure 1\\n. Genes with <100 reads in all samples were discarded, resulting in a dataset containing 5838 genes. edgeR identified 2870 DEGs across the different time points, whereas maSigPro-GLM selected 2158 DEGs (FDR = 0.05). There were 1629 genes in common between the two methods, 529 specifically found by maSigPro and 1241 identified only by edgeR. Out of these 1241 edgeR exclusive DEGs, 1194 were identified as significant in the first maSigPro but finally not selected in the second regression step because their\\nR\\n2\\n< 0.5, while the remaining 47 genes were not preselected by maSigPro in the first step. To better understand the gene expression patterns associated to similarities and differences between the two methods, we randomly selected three genes belonging to each of these sets (\\nFig. 2', 'María José Nueda et al.,2014, DOI:https://doi.org/10.1093/bioinformatics/btu333\\nFinally, although significance thresholds in maSigPro-GLM maintain their statistical meaning, the goodness of fit, which is used in the second step of maSigPro to select genes with well-fitted models, is evaluated in GLMs in terms of the deviance: the percentage of deviance explained by the model. We conducted experiments with simulated data to understand how this parameter behaves in different experimental settings. Our results indicated that similar to the recommended threshold in the LM version of maSigPro, a cutoff value of 0.7 is valid in most scenarios. However, when data are abundant, i.e. triplicated measurements and multiple series, this threshold could be lowered to 0.5. Indeed, this value was used in the analysis of the real Arabidopsis dataset. The comparison with edgeR, which solely selects genes on the basis of a significant\\nP\\n-value, showed that the maSigPro filtering based on a\\nR\\n2', 'María José Nueda et al.,2014, DOI:https://doi.org/10.1093/bioinformatics/btu333\\nBoth maSigPro and edgeR methods are based on the GLMs but with a different approach. The major difference between the maSigPro and edgeR methods is that maSigPro is specialized in the estimation of serial data, i.e. when the independent variable is quantitative such as time. This is achieved by providing an easy way to define a polynomial model for the data. Another important difference is that maSigPro follows a second stepwise regression that obtains the best model for each gene and retains only significant coefficients in each model, whereas edgeR applies the same model to each gene.\\n2.2.1 Simulated data\\nSimulations have been created using NB distributions with a parametrization based on the mean μ and size θ. In each sample\\ni\\n, where the targeted total number of reads is\\nN\\n, and the relative abundance of each gene\\ng\\nis\\np\\ngi\\n, the expected gene counts, μ\\ngi\\n, can be computed as\\nμ\\ng\\ni\\n=\\nN\\n×\\np\\ng\\ni', 'María José Nueda et al.,2014, DOI:https://doi.org/10.1093/bioinformatics/btu333\\nmaSigPro applies GLMs to model RNA-seq as do other dedicated statistical packages such as edgeR, included for comparison in this study. The major difference between maSigPro and edegR methods is that maSigPro is specialized in parameter estimation of serial data, i.e. when the independent variable is quantitative such as time. This is achieved by providing an easy way to define a polynomial model for the data that have the flexibility to fit different time-course patterns. In contrast, edgeR treats time not as a continuous variable but as multifactor. Another important difference is that maSigPro follows a second step that obtains the best model for each gene such that only significant coefficients are retained in each model, whereas edgeR applies the same model to each gene under the multifactor consideration. This results in models with more variables that might be prone to give false calls. Moreover, we apply in the second step a filter on gene selection that takes into account the']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "4",
         "Feedback: The response is correct and factual but includes additional information that is not necessary to answer the question. The reference answer is more concise.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "38",
         "Which aquaporin mRNA expressions were significantly decreased in neonatal AQP8-deficient ovaries?",
         "The expression of AQP7, AQP11, and AQP12 mRNAs were significantly decreased in neonatal AQP8-deficient ovaries.",
         "The mRNA expression of AQP7, AQP11 and AQP12 were significantly decreased in neonatal AQP8-deficient ovaries (1, 3).",
         "['Weiheng Su et al.,2013, DOI:https://doi.org/10.1186/1477-7827-11-88\\nBackground: Granulosa cells play a key role in folliculogenesis and female reproduction. Our previous study demonstrated that water channel aquaporin-8 (AQP8) is expressed in mouse follicular granulosa cells and is an important determinant of granulosa cell apoptosis and follicular maturation. More roles of AQP8 in folliculogenesis remain to be determined. Findings: The present study reports the increased occurrence of multi-oocyte follicles (MOFs) in ovaries of AQP8 knockout mice. The MOFs in AQP8-deficient ovaries contained two or three oocytes, and distributed at various follicle stages including primary (12.5%), secondary (50%), antral (18.8%) and atretic (18.8%) follicles in 5-week ovaries. The MOF is occasionally seen in wild-type ovary only in primary and secondary follicles. The number of MOFs in AQP8-deficient ovary reduced with age (26.7 +/− 5.2 per ovary at 5 weeks old, 14 +/− 5.5 at 10 weeks old, and 3.3 +/− 5.1 at 20 weeks old)', 'Weiheng Su et al.,2013, DOI:https://doi.org/10.1186/1477-7827-11-88\\n.7 +/− 5.2 per ovary at 5 weeks old, 14 +/− 5.5 at 10 weeks old, and 3.3 +/− 5.1 at 20 weeks old). mRNA expression of AQP5, AQP7, AQP8, AQP11 and AQP12 was detected in neonatal mouse ovaries and in granulosa cells in 4 week old mouse ovaries. The expression of AQP7, AQP11 and AQP12 mRNAs are decreased significantly in neonatal AQP8-deficient ovaries, whereas AQP5 mRNA expression remains unchanged. Conclusions: The emergence of MOFs is associated with AQP8 deficiency. The study suggested the involvement of AQP8 in the formation of follicles and provided new insight into the molecular mechanisms of folliculogenesis.', 'Weiheng Su et al.,2013, DOI:https://doi.org/10.1186/1477-7827-11-88\\nFollicles are functional units of mammalian ovary that develop through primordial, primary, secondary, antral, and preovulatory follicle stages, and finally ovulate in response to the stimulation of luteinizing hormone. Disorders of folliculogenesis in ovary lead to gynecological diseases such as polycystic ovary syndrome, ovarian cyst, premature ovarian failure and even ovarian carcinoma, which impair female reproductive function [1,2]. In some cases abnormal folliculogenesis also can cause increased ovulation [3,4]. However, the mechanisms underlying the altered folliculogenesis are largely unknown. Aquaporins (AQPs) are a family of water channel proteins selectively expressed on the plasma membrane of some cell types where they may have important physiological functions [5]. Follicles, especially antral follicles are fluid-containing structures', 'Weiheng Su et al.,2013, DOI:https://doi.org/10.1186/1477-7827-11-88\\n. AQP8 was reported to express in granulosa cells of formed follicles, we first investigate the expression of AQPs family in granulosa cells of 4-week-old mice. RTPCR results indicated that AQP5, AQP7, AQP8, AQP11 and AQP12 are expressed (Figure 2A), and this is the first time to report the expression of AQP5, AQP11 and AQP12 in mouse granulosa cells. For reasons that neonatal pre-granulosa cells are difficult to isolate, and neonatal ovaries are almost simply composed by germ cells and pre-granulosa cells, neonatal ovaries were used to represent pre-granulosa cells in this study. RT-PCR results identified that AQP8 is also expressed in neonatal wild-type ovaries, but not in AQP8−/− ovaries (Figure 2B). Furthermore, relative expression levels of the five AQPs in wild-type and AQP8−/− neonatal ovaries by qRT-PCR indicated significantly decreased AQP7, AQP11 and AQP12 (but not AQP5) expression in AQP8−/− neonatal ovaries compared to wild-type ovaries (Figure 2C).', 'Weiheng Su et al.,2013, DOI:https://doi.org/10.1186/1477-7827-11-88\\n. In this context, the migration of pre-granulosa cells is probably impaired in AQP8-deficient mice. Therefore the invasion and encapsulation of pre-granulosa cells may be slowed down by AQP8 deletion, resulting abnormal encapsulation and formation of MOFs. In addition, other factors such as the transport property of AQP8 to urea, ammonia and H 2O2 [25-27] and the decreased expression of AQP7, AQP11 and AQP12 may also contribute to the increased formation of MOFs in AQP8 −/− mice. So far, there is not report for observation of MOFs in other AQP-deficient mice. Further studies are required to determine the mechanisms of increased MOFs formation in aquaporin deficient ovary. The MOFs in AQP8 −/− ovary happened more frequently in prepubertal mice, decreased with age, and nearly disappeared in old mice. A similar age-related change of MOFs quantity was also observed in dog ovary [28]. Such characteristics of MOFs appearance support their formation in the early stage of folliculogenesis', 'Weiheng Su et al.,2013, DOI:https://doi.org/10.1186/1477-7827-11-88\\n. The incomplete penetration suggests that the AQP8 null allele is haploinsufficient or semi dominant for the MOFs phenotype. We didn’t find morphological trace of joining between adjacent follicles, suggesting MOFs are not formed during follicle development but during primordial follicle formation in neonatal ovary. AQP8 is a major plasma membrane water channel of mouse granulosa cells, as AQP8 deletion reduced the water permeability by 45% [3]. Reduced plasma membrane water permeability in AQP8 −/− pregranulosa cells may have two consequences: decreased apoptosis and impaired cell migration [5,10,24] that are likely to change the behavior of pre-granulosa cells during folliculogenesis. Saadoun et al. reported that cell migration of AQP1- or AQP4-transfected cells are faster than that of control cells [24]. In migrating cells which express AQP1, the localization of AQP1 polarized into the lamellipodia, where rapid water fluxes occur', 'Weiheng Su et al.,2013, DOI:https://doi.org/10.1186/1477-7827-11-88\\n. Follicles, especially antral follicles are fluid-containing structures. Expression and function of several AQPs in follicular granulosa cells have been reported in human [6], rat [7], mouse [3,8], and pig [9] ovaries. AQPs 7- 9 expressed in granulosa cells may provide a transcellular water transport pathway into antrum in the rat follicle [7]. AQP-mediated plasma membrane water permeability can control the rate of apoptosis by regulating apoptotic volume decrease in rat granulosa cells [10]. Thoroddsen et al. reported that AQPs1-4 are differentially expressed in human granulosa and theca cells of the preovulatory follicle during ovulation and may be involved in follicular rupture and corpus luteum formation [11]. Another study by Qu et al. suggested that hyperandrogenism in follicular fluid of women with polycystic ovary syndrome inhibited the expression of AQP9 in granulosa cells through the PI3K pathway [6]']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response accurately identifies the aquaporin mRNA expressions (AQP7, AQP11, and AQP12) that were significantly decreased in neonatal AQP8-deficient ovaries, mirroring the reference answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "39",
         "What steps are required for sequencing-based association studies?",
         "Sequencing-based association studies require platform selection, quality control, choice of analysis units, assignment of variants to analysis units with bioinformatic tools, selection of methods for testing rare-variant effects, and prioritization and replication of top signals.",
         "Sequencing-based association studies require several data-processing and -analysis steps, including platform selection, quality control, choice of analysis units, assignment of variants to analysis units with bioinformatic tools, selection of methods for testing rare-variant effects, and prioritization and replication of top signals (Document 0).",
         "['Seunggeung Lee et al.,2014, DOI:https://doi.org/10.1016/j.ajhg.2014.06.009\\nSequencing-based association studies require several data-processing and -analysis steps, including platform selection, quality control, choice of analysis units, assignment of variants to analysis units with bioinformatic tools, selection of methods for testing rare-variant effects, and prioritization and replication of top signals (\\nFigure\\xa01\\nand\\nBox 1\\n). In this section, we focus on choices of different sequencing designs and platforms, and we discuss association analysis in the next section.', 'Seunggeung Lee et al.,2014, DOI:https://doi.org/10.1016/j.ajhg.2014.06.009\\nIn this review, we provide an overview of the current status of sequencing-based association studies and discuss\\xa0the statistical issues that arise. We first discuss strategies for study design, proceed to association testing and meta-analysis methods for sequencing-based rare-variant association studies, and conclude with other important issues, including population-stratification adjustment, genotype imputation, and the design of follow-up studies.', 'Daniel Sik Wai Ho et al.,2019, DOI:https://doi.org/10.3389/fgene.2019.00267\\nGENOME-WIDE ASSOCIATION STUDIES', 'Xiaowei Zhan et al.,2016, DOI:https://doi.org/10.1093/bioinformatics/btw079\\nLiu\\net al.\\n, 2014\\n).\\nSoftware packages that accompany methodology papers are often available (\\nHu\\net al.\\n, 2014\\n;\\nWu\\net al.\\n, 2011\\n;\\nYandell\\net al.\\n, 2011\\n). While they implement useful methods, they typically do not provide sufficient features that are necessary for practical data analyses. It is necessary to develop software tools and pipelines that enable statistical analysis of sequence data which is otherwise cumbersome to perform. A few software packages are available for performing comprehensive sequence-based association analyses, including Variant Association Tools (VAT;\\nWang\\net al.\\n, 2014\\n) and PSEQ (\\nhttps://atgu.mgh.harvard.edu/plinkseq/index.shtml', 'Seunggeung Lee et al.,2014, DOI:https://doi.org/10.1016/j.ajhg.2014.06.009\\n37,124\\nwill facilitate future genome-wide sequencing-based association studies.\\nWe have provided a review of numerous recently developed methods for rare-variant association testing. Given that the relative performance of these methods depends on the underlying genetic architectures of complex traits, it is difficult to have a test that is optimal for all scenarios. Omnibus tests that combine different tests provide an attractive alternative for balancing power and robustness. When more is learned about genetic architectures of complex diseases and traits, this knowledge can be incorporated in association tests to increase power and prioritize variants for replication studies and functional analysis.', 'Seunggeung Lee et al.,2014, DOI:https://doi.org/10.1016/j.ajhg.2014.06.009\\n37,124\\nwill facilitate future genome-wide sequencing-based association studies.\\nWe have provided a review of numerous recently developed methods for rare-variant association testing. Given that the relative performance of these methods depends on the underlying genetic architectures of complex traits, it is difficult to have a test that is optimal for all scenarios. Omnibus tests that combine different tests provide an attractive alternative for balancing power and robustness. When more is learned about genetic architectures of complex diseases and traits, this knowledge can be incorporated in association tests to increase power and prioritize variants for replication studies and functional analysis.', 'Marzia Antonella Scelsi et al.,2021, DOI:https://doi.org/10.1371/journal.pcbi.1008517\\n. We present a framework that enables enhanced genetic association testing for a wide range of traits, diseases, and sample sizes.']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response is completely correct, accurate, and factual, mirroring the reference answer. It correctly identifies the steps involved in sequencing-based association studies.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "40",
         "What is considered the normal range of CGG nucleotide repeats for FMR1 gene ovarian function?",
         "26 to 34 CGG nucleotide repeats.",
         "The normal range of CGG nucleotide repeats for FMR1 gene ovarian function is 26 to 34 (Document 3).",
         "['Norbert Gleicher et al.,2011, DOI:https://doi.org/10.1186/1477-7827-9-23\\nControlling OR Wallace and Kelsey recently reaffirmed that ovarian aging varies between individuals [12]. Using age at menopause as end points, they determined that speed of follicle recruitment and follicle numbers vary significantly at different stages of life. We have come to similar conclusions, recently describing effects of theFMR1 gene on the ovary [46]. Evaluations of the FMR1 gene have become increasingly popular because of the gene ’s neuro/psychiatric effects [47]. It, however, also, independently, demonstrates specific ovarian effects [46]. Increased risk for POF/POI in women with premutation-range (ca. 55-200) CGG triple nucleotide repeats has been known for decades [47] but that even lower CGG repeat numbers may also denote risk towards premature ovarian senescence, at times representing milder forms of so-called POA/OPO I, is a more recent discovery [48-51]', 'Norbert Gleicher et al.,2011, DOI:https://doi.org/10.1186/1477-7827-9-23\\n. FMR1 can, therefore, be viewed as an“ovarian aging gene.” Norm women at younger ages appear to recruit actively and, therefore, likely deplete TOR quicker than het and hom females, who from young age on recruit at much slower pace. The latter two genotypes, therefore, demonstrate much slower and steadier declines in AMH levels. Still prelimin a r yd a t as u g g e s tt h a thet patients can be further subdivided into het-norm/high and hetnorm/low sub-genotypes (depending on whether the abnormal allele is abnormally high or low), which may further differ in respective aging curves [52,53]. These observations confirm statistical associations between FMR1 genotypes (based on normal CGG range of 26 to 34) and FOR, based on FOR assessment with AMH. They, thus, suggest significant relevance of triple CGG repeat counts to clinical practice [49,50], and that FMR1 genotypes, within reasons, already at young ages allow predictions about “ ovarian aging ” patterns', 'Norbert Gleicher et al.,2011, DOI:https://doi.org/10.1186/1477-7827-9-23\\n. Chen and associates reported that maximal translation of the gene product and that the switching point between positive and negative effects of the gene occurs at 30 CGG repeats [57]. This exactly reflects the median of the normal range, reported for the gene ’so v a r i a n function [46,58]. This normal range of 26 to 34 repeats also contains at midpoint the tall distribution peak of CGG repeats in the general population, reported by Fu and associates at 29 to 30 repeats [59]. Combined, these observations suggest that FMRP, the gene product ofFMR1,a n d / o ri t s translation, may play a role in follicular recruitment and “ovarian aging.” Mouse models have allowed progress in understanding follicular recruitment and, by extension, OR regulation. We noted earlier the important recent paper by Reddy et al [4]. In humans, this area, however, still largely represents a black box [3], resulting in many models of “ ovarian aging ” [11,12,43] but little factual data', 'Norbert Gleicher et al.,2011, DOI:https://doi.org/10.1186/1477-7827-9-23\\n. The genotypes relating to ovarian function are distinctly different from the genotypes historically reported to define the gene’s neuro/psychiatric risks. In regard to ovarian function, 26 to 34 CGG nucleotide repeats represent normal (median 30), independent of ethnicity/ race [49]. Using this range to define genotypes, women can be designated as normal (norm), when both alleles are in normal range,heterozygous (het) if one is normal and the other abnormal and homozygous (hom) if both alleles are outside normal range. Figure 2 demonstrates linear regressions of AMH over age, depending on whether women are norm , hetabnormal or hom- abnormal: Depending on FMR1 genotype, ovarian aging patterns differ. Before physiologic ovarian aging significantly contributes to OR at young ages, differences are most obvious. As expected, norm women demonstrate better OR than het females, with hom women demonstrating the lowest OR', 'Ewa Rudnicka et al.,2018, DOI:https://doi.org/10.5114/pm.2018.78550\\nOn the other hand, the presence of X trisomy may  also result in ovarian dysfunction [12, 13].\\nGene abnormalities can pertain to mutation on X as well as on autosomal chromosomes. The most common is FMR1 premutation located  on X chromosome. The mutation rests upon increased  expansion of CGG triplets in the untranslated region.  Patients with FMR1 premutation (CGG present 55-199  times) are at increased risk of developing POI [12]. The  incidence of ovarian dysfunction among carriers reaches 12.9% [14]. Full mutations characterised by more  than 200 repetitions inactivate the gene by methylation  and lead to fragile X syndrome, and are also the most  common inherited cause of disability among men [15].', 'Norbert Gleicher et al.,2011, DOI:https://doi.org/10.1186/1477-7827-9-23\\n. Described FMR1 genotypes raise additional questions about these proposed human models of ovarian aging normal ovarian aging, quite obviously, has to be defined separately for individualFMR1 genotypes. In some aspects, Wallace and Kelsey ’s model [12] appears superior to others since it detected rapid declines in OR at younger ages than Faddy and coworkers, who reported an accelerated declines only at age 37 to 38 years [43]. Considering varying “ ovarian aging ” patterns of different FMR1 genotypes [46], Faddy and Gosden ’s later timing, however becomes understandable as norm women demonstrate accelerated declines in FOR at approximately age 35 years (Figure 2), while het and hom females demonstrate a more gradual decrease. Math ematically combined, all three genotypes, indeed, resemble Faddy and Gosden’s model. Ultimate purposes of FMR1 genotypes remain to be determined', 'Norbert Gleicher et al.,2011, DOI:https://doi.org/10.1186/1477-7827-9-23\\n. Paradoxically, d ue to rapid recruitment at young ages, the norm population appears at greatest risk for early follicle depletion and, possibly, early menopause. Whether early depletion can be equated with early menopause has, however, so far not been established. FMR1 , thus, is involved in regulating FOR but may also affect TOR by regulating follicular recruitment. How FMR1 actually regulates physiologic functions has remained controversial: In association with neuro/psychiatric consequences, the number of CGG triple repeats is a determining factor. Risks increase with increasing expansion sizes up to approximately the lower half of the so-called premutation range (ca. 55-200 repeats), though not necessarily in linear fashion [47,54]. The gene codes for an RNA-binding protein, the so-called fragile X mental retardation protein (FMRP) [55], important in synaptic physiology and apparently demonstrating RNA toxicity [56]']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response accurately states the normal range of CGG nucleotide repeats for the FMR1 gene, aligning with the reference answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "41",
         "What functional alteration was statistically significantly dysregulated as a shared alteration across uterine disorders at the endometrial level?",
         "Cell cycle alterations.",
         "Cell cycle alterations were statistically significantly dysregulated as a shared functional alteration across uterine disorders at the endometrial level (1).",
         "['Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025\\nThis study represents the most exhaustive functional analysis performed for robustly identifying functional dysregulations at the endometrial level in patients affected by uterine disorders. Indeed, to our knowledge, this work is the first to compare and highlight functional similarities between different uterine disorders. Our results also highlight specific gene functions, some of which were not previously reported even in the individual patient cohort studies. These newfindings demonstrate the advantage of applying this novel procedure, in which integrating different sample cohorts using the same procedure at the functional level within each cohort ultimately achieves greater statistical power. This approach also enabled us to uncover newfindings by observing functional processes affecting the endometrium from a genomescale perspective', 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025\\n. 2B andSupplemental Table 2). Meta-analyses results for these statistically significant functions, including the LOR and its 95% CI, SE of the LOR, and the FDR associated with each function are reported inSupplemental Table 2and Supplemental Fig. 2.F u n n e lp l o t ss h o w e da l ls t u d i e s inside the 95% CI, suggesting that study origin and heterogeneity did not bias the results (Supplemental Fig. 3, available online). Statistically significantly altered gene functions that presented contradictory results (up- or down-regulated) between sample cohorts evaluating the same uterine disorder were not considered in our analysis. Comparison of Dysregulated Processes in the Endometrium Among Uterine Disorders Cell cycle alterations were statistically significantly dysregulated as a shared functional alteration across uterine disorders at the endometrial level (Fig. 3A). Additionally, each uterine disorder shared statistically significantly altered gene functions (FDR<0', 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025\\n. All integrations were performed under the DerSimonian and Laird  random effects model to account for individual study heterogeneity in the R environment . This model weighs the functional results of each study according to their inherent variability measured by the standard deviation (SD) of the logarithm of odds ratio (LOR) obtained in the GSEA analysis for each evaluated function. Consequently, studies with less variability are weighted more in the computation of thefinal meta-analysis results than those with high variability. Thus, meta-analysis highlights the most robust endometrial functions associated with each uterine disorder. A total of 6,390 functions (226 KEGG pathways, 4,405 GO biological processes (BP) terms, 1,093 GO molecular functions (MF) terms, and 666 GO cellular components (CC) terms) were evaluated through meta-analysis for each uterine disorder', 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025\\n. 2). The 483 statistically significantly altered gene functions identified in the metaanalyses were classified into 37 functional groups according to the descriptions provided in the functional databases (Supplemental Table 2, available online). Following this classification, we obtained 12 statistically significantly dysregulated functional groups for RIF, 11 for RPL, 10 for endometriosis, and 34 for ADC (seeFig. 1and Fig. 2). ‘‘Chromosome parts and DNA binding’’was the most represented functional group among the RIF sample cohorts, comprising 25% of affected endometrial functions.‘‘Phosphorylation’’ was also dysregulated in RIF, representing 15% of altered gene functions (all of them up-regulated). VOL. 113 NO. 6 / JUNE 2020 1263 Fertility and Sterility®  TABLE 1 Characterization of uterine disorders studies and participants: disorders', 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025\\n. Detailed information for each significant function about the logarithm of odds ratio (LOR), 95% confidence interval (CI), standard error (SE) of the LOR, the FDR and the functional database are presented inSupplemental Table 2. Devesa-Peiro. Endometrial factor in uterine disorders. Fertil Steril 2020. VOL. 113 NO. 6 / JUNE 2020 1269 Fertility and Sterility®  FIGURE 3 Functional comparison of dysregulated processes in the endometrium of uterine pathologies and implantation alterations. (A) Similarities between uterine disorders. Statistically significant functional groups (rows) (false discovery rate [FDR]<0.05) commonly dysregulated between uterine disorders (columns). Box color represents each dysregulated functional group and the direction of the dysregulation: up-regulated in the uterine disorder versus control (red), down-regulated (blue), or both (grey). ( B) Functional dysregulations associated with endometrial receptivity', 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025\\n. 3). The functional processes involved in embryo implantation are reported with their associated references and indication of whether they were up-regulated or down-regulated in each uterine disorder (seeFig. 3B) . Forest and funnel plots of the most statistically significantly altered functions (those with the lowest FDR) for each uterine disorder among the 12 highlighted functional groups (see Fig. 3A) are depicted in Supplemental Figs. 2 and 3, respectively. These plots ensure the meta-analyses quality and unbiased results for the main functions reported as our principalfindings.', \"Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025\\n. A genome-scale perspective provides a bird's eye view of functional processes without deepening in the molecular details behind these functionals alterations. Despite the lack of molecular details and experimental validations, the approach performed in this study employs a powerful methodology to highlight similarities among different sources of information, providing newfindings in the endometrium of patients with uterine disorders. In this research, we generated new hypotheses, which is the whole point rather than only useful tools as commonly considered in the traditional research process . On the other hand, the functional alterations reported in this study are based on functional genomics approaches and on the nontissue and noncell type specific information available in KEGG and GO. However, these results are inferred from transcriptomics of whole endometrial tissue, and only those gene-function associations with experimental evidence were considered\"]",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response includes the correct answer, \"Cell cycle alterations,\" and accurately reflects the information provided in the reference answer. Therefore, it is completely correct and factual.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "42",
         "What are the well-supported molecular hallmarks of endometriosis?",
         "Genetic predisposition, estrogen dependence, progesterone resistance, and inflammation.",
         "In endometriosis, dysregulation of progesterone and estrogen signaling commonly results in progesterone resistance and estrogen dominance (3). This hormone imbalance leads to heightened inflammation and may also increase pelvic pain and decrease endometrial receptivity to embryo implantation (3). Endometriosis is also characterized by the down-regulation of ciliary processes (6).",
         "['Richard O Burney et al.,2012, DOI:https://doi.org/10.1016/j.fertnstert.2012.06.029\\nOriginally described over three hundred years ago, endometriosis is classically defined by the presence of endometrial glands and stroma in extrauterine locations. Endometriosis is an inflammatory, estrogen dependent condition associated with pelvic pain and infertility. This work reviews the disease process from theories regarding origin to the molecular basis for disease sequelae. A thorough understanding of the histopathogenesis and pathophysiology of endometriosis is essential toward the development of novel diagnostic and treatment approaches for this debilitating condition.', 'Richard O Burney et al.,2012, DOI:https://doi.org/10.1016/j.fertnstert.2012.06.029\\n5\\n). The significant individual and public health concerns associated with endometriosis underscore the importance of understanding its pathogenesis and pathophysiology toward prevention and the development of sensitive non-surgical diagnostic assays and effective treatments. The past several decades have witnessed substantial progress toward unraveling the enigma associated with this disorder. Herein, we outline the current understanding of the pathogenesis and pathophysiology of endometriosis.', 'George Anderson et al.,2019, DOI:https://doi.org/10.1515/bmc-2019-0017\\n.  Such a frame of reference better integrates wide bodies  of data on the pathoetiology and pathophysiology of  endometriosis, including its association with heightened  cancer risk.', 'Ryan M Marquardt et al.,2019, DOI:https://doi.org/10.3390/ijms20153822\\nIn the healthy endometrium, progesterone and estrogen signaling coordinate in a tightly regulated, dynamic interplay to drive a normal menstrual cycle and promote an embryo-receptive state to allow implantation during the window of receptivity. It is well-established that progesterone and estrogen act primarily through their cognate receptors to set off cascades of signaling pathways and enact large-scale gene expression programs. In endometriosis, when endometrial tissue grows outside the uterine cavity, progesterone and estrogen signaling are disrupted, commonly resulting in progesterone resistance and estrogen dominance. This hormone imbalance leads to heightened inflammation and may also increase the pelvic pain of the disease and decrease endometrial receptivity to embryo implantation. This review focuses on the molecular mechanisms governing progesterone and estrogen signaling supporting endometrial function and how they become dysregulated in endometriosis', 'Richard O Burney et al.,2012, DOI:https://doi.org/10.1016/j.fertnstert.2012.06.029\\n45\\n).\\nTable 1.\\nCandidate factors implicated in the pathophysiology of endometriosis\\nGene\\nFunction\\nReference(s)\\n17βHSD-2\\nHydroxysteroid dehydrogenase\\nZeitoun et al 1998 (\\n47\\n)\\nBCL-2\\nAnti-apoptosis\\nJones et al 1998 (\\n31\\n)\\nCYP19\\nAromatase enzyme\\nNoble et al 1996 (\\n101\\n)\\nHOXA10\\nTranscription factor\\nTaylor et al 1999 (\\n102\\n)\\nIL-6\\nCytokine\\nHarada et al 1997 (\\n81\\n)\\nKRAS\\nOncogene\\nDinulescu et al 2005 (\\n30\\n)\\nMMP 3,7\\nMatrix metalloproteinases\\nBruner-Tran et al 2002 (\\n65\\n)\\nNF-KB\\nTranscription factor\\nGonzalez-Ramos et al 2010 (\\n103\\n)\\nPGE2\\nProstaglandin\\nBadawy et al 1984 (\\n104\\n)\\nPTEN\\nTumor suppressor gene\\nDinulescu et al 2005 (\\n30\\n)\\nTGF-B\\nCytokine\\nOosterlynck et al 1994 (\\n61\\n)\\nTNF-α\\nCytokine\\nEisermann et al 1988 (\\n80\\n)\\nOpen in a new tab\\nAltered hormonal milieu: estrogen dependence and progesterone resistance', 'Ryan M Marquardt et al.,2019, DOI:https://doi.org/10.3390/ijms20153822\\n3. Dysregulation of Progesterone and Estrogen Signaling in Endometriosis\\nAs the work reviewed in the previous sections has demonstrated, tightly regulated signaling pathways governed by P4 and E2 in a stromal and epithelial compartment-specific manner are key to maintaining endometrial homeostasis and supporting female fertility. Dysregulation of steroid hormone signaling is common in many uterine pathologies such as endometriosis, infertility, endometrial cancer, uterine leiomyoma, and recurrent pregnancy loss [\\n12\\n]. For the remainder of this review, we will focus on the molecular pathophysiology and treatment of endometriosis with particular focus on recent findings that shed light on the contribution of P4 and E2 signaling dysregulation to the infertility and pelvic pain women with this disease often experience.\\nEndometriosis is classically defined as the presence of endometrium-like tissue located outside the uterine cavity [\\n8', 'Almudena Devesa-Peiro et al.,2020, DOI:https://doi.org/10.1016/j.fertnstert.2020.01.025\\n. Cell cycle alterations were shared among all the pathologies investigated. Endometriosis was characterized by the down-regulation of ciliary processes. Among the endometriosis, ADC, and RIF samples, mitochondrial dysfunction and protein degradation were shared dysregulated processes. In addition, RPL had the most distinct functional profile, and 95% of affected functions were down-regulated. Conclusion(s): The most robust functions dysregulated in the endometrium of patients with uterine disorders across sample cohorts implicated an endometrial factor at the gene expression level. This shared endometrial factor affects endometrial receptivity processes. (Fertil Steril /C2102020;113:1261- 74. /C2112020 by American Society for Reproductive Medicine.) El resumen está disponible en Español alfinal del artículo']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "3",
         "Feedback: The response mentions hormone imbalance, inflammation, and downregulation of ciliary processes, which are related to endometriosis but doesn't directly address the core molecular hallmarks like genetic predisposition, estrogen dependence, and progesterone resistance as the reference answer does. It touches upon related concepts but misses key elements.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "43",
         "What is the increased likelihood of advanced age mothers experiencing pregnancy-induced hypertension compared to adult mothers?",
         "Advanced age mothers are four times more likely to encounter pregnancy-induced hypertension than their adult counterparts.",
         "Advanced age mothers were four times more likely to encounter pregnancy-induced hypertension than their adult counterparts (OR 4.15, (95% CI 2.272 - 7.575), p < 0.001) (0). Advanced age mothers were 4.15 times more likely to get their pregnancy complicated by pregnancy-induced hypertension than the reference group (AOR 4.15, (95% CI 2.272 - 7.575), p < 0.001) (1). Advanced maternal age pregnancy was significantly associated with pregnancy induced hypertension [AOR 4.15, 95% CI (2.272 - 7.575), p < 0.001] (2).",
         "['Mihret-ab Mehari et al.,2020, DOI:https://doi.org/10.1186/s12884-020-2740-6\\nThis study has indicated that advanced maternal age is an important risk factor for many adverse obstetrical and perinatal outcomes. One of the main findings of this study was the association of maternal age with pregnancyinduced hypertension in which advanced age mothers were four times more likely to encounter pregnancyinduced hypertension than their adult counterparts (OR 4.15, (95% CI 2.272 - 7.575), p < 0.001). This result is congruent with the study done in Malaysia and UK with an Table 3 magnitude of adverse perinatal outcomes of advanced and adult mothers, Ayder Comprehensive Specialized Hospital, Northern Ethiopia, 2017/18 ( n = 752) Perinatal outcomes Maternal age 20- 34 years 35+ years Total N % N % N % Perinatal Death No 364 96.8 338 89.9 702 93.4 Yes 12 3.2 38 10.1 50 6.6 Low Fifth Minute Apgar Score No 354 94.4 288 76.6 643 91.1 Yes 10 2.4 54 14.4 63 8.9 Congenital Anomalies No 374 99.5 371 98.7 745 99.1 Yes 2 0.5 5 1.3 7 0.9 Gestational Age Pre-term 10 2.7 42 11.3 52 7', 'Mihret-ab Mehari et al.,2020, DOI:https://doi.org/10.1186/s12884-020-2740-6\\n. Amniotic fluid disorders, premature rupture of membranes and post-term pregnancy were not found to have a significant association with any of the independent variables. Gestational DM, postpartum hemorrhage, macrosomia, and congenital anomalies were not analyzed using logistic regression because the estimated count of cells was less than 10. Advanced age mothers were 4.15 times more likely to get their pregnancy complicated by pregnancy-induced hypertension than the reference group (AOR 4.15, (95% CI 2.272 - 7.575), p < 0.001). Besides APH was 2.54 times more likely to happen among the study group than the reference group (AOR 2.54 (95% CI 1.32 - 4.91), P = 0.005). Furthermore, advanced age mothers were 2.7 times more probable to undergo cesarean section than their adult counters (AOR 2.722, (95% CI 1.777 - 4.170), p < 0.001). The regression table is shown in Table 4', 'Mihret-ab Mehari et al.,2020, DOI:https://doi.org/10.1186/s12884-020-2740-6\\n. Results: This study revealed that advanced maternal age pregnancy was significantly associated with pregnancy induced hypertension [AOR 4.15, 95% CI (2.272 - 7.575), p < 0.001], ante partum hemorrhage [AOR 2.54, 95% CI (1.32- 4.91), P = 0.005] & cesarean delivery [AOR 2.722, 95% CI (1.777 - 4.170), p < 0.001]. Furthermore, advanced maternal age pregnancy was also increasingly associated with adverse perinatal outcomes like preterm delivery [AOR 3.622, 95% CI (1.469 - 8.930), p = 0.005], low birth weight [AOR 3.137, 95% CI (1.324 - 7.433), p = 0.009], perinatal death [AOR 2.54, 95% CI (1.141 - 5.635), p = 0.022] and low fifth minute APGAR score [AOR 7.507, 95% CI (3.134 - 17.98), p < 0.001]. Notwithstanding this, maternal age was not found to be associated with amniotic fluid disturbances, premature rupture of membranes and post-term pregnancy. Conclusions: Advanced maternal age is markedly linked with adverse obstetrical and perinatal outcomes', 'Mihret-ab Mehari et al.,2020, DOI:https://doi.org/10.1186/s12884-020-2740-6\\nSocio-demographic and obstetric characteristics of study participants A total of 752 charts of mothers were reviewed in this study. The median age of the study group ( ≥ 35 years old) was found to be 37 years with (IQR of 35 - 40 years) whereas the mean age of the reference group (20 - 34 years old) was 26.6 years with a standard deviation of 3.5 years. Almost one fifth (19.1%) of advanced maternal age mothers in this study had ANC follow up of less than four times. Nearly half of adult pregnant mothers (43.1%) were pregnant for the first time. The magnitude of malpresentation among advanced age mothers and adult mothers was 9.6 and 4.3%, respectively. Sociodemographic and obstetric characteristics of study participants are listed in Table 1. Magnitude of adverse obstetrical outcomes This study revealed that the magnitude of pregnancyinduced hypertension and antepartum hemorrhage was 87 (11.6%) and 76 (10.1%), respectively. There were only 3 (0', 'Mihret-ab Mehari et al.,2020, DOI:https://doi.org/10.1186/s12884-020-2740-6\\n. Various studies conducted on the topic elucidated that the aftermath of advanced maternal age (AMA) is highly linked with the occurrence of pregnancy-induced hypertension, DM, maternal near-miss, increased cesarean delivery, malpresentation, and maternal death. Alongside with this, AMA also results in neonatal complications, such as Low Apgar score, NICU admission, preterm delivery, low birth weight, birth defects, chromosomal abnormalities and perinatal death [ 2, 4- 7]. Fertility in women starts to decline in the early thirties and even decreases faster after mid and late thirties. Women with advanced age usually have a relatively lower tendency to achieve pregnancy within a short period. The probability of achieving pregnancy in a single menstrual cycle, fecundability, is decreased in these age groups [ 8, 9]. Globally, the academic community has extensively explored the association of AMA with adverse obstetrical and perinatal outcomes', 'Gabriele Saccone et al.,2022, DOI:https://doi.org/10.1002/ijgo.14100\\n4.3. Conclusions\\nIn summary, women with advanced maternal age have an increased risk of maternal and perinatal complications. Our meta‐analysis showed that the higher the maternal age the higher the risk of adverse pregnancy outcomes. These data should be used when women with advanced maternal age are counseled regarding their risk in pregnancy.', 'Mihret-ab Mehari et al.,2020, DOI:https://doi.org/10.1186/s12884-020-2740-6\\n. This similarity might be mainly attributed to the fact that endothelial response to vasodilators diminishes as mothers get older [ 14]. Nevertheless, a study conducted in India contradicts this study as it claimed that PIH has no significant association with maternal age [ 15]. This can be accounted for by the difference in sample size. In addition, maternal age was found to be significantly associated with APH (OR 2.54 (95% CI 1.32 - 4.91), P = 0.005) in this study. This finding is supported by studies done in Amman Jordan [ 16]. This could be ascribed to increased gravidity in advanced age mothers which in turn make them at greater risk of having placenta Previa [ 14]. In contrast, studies from Turkey and India showed no significant association of maternal age with APH [ 12, 15]']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response accurately states that advanced age mothers are four times more likely to experience pregnancy-induced hypertension compared to adult mothers, aligning perfectly with the reference answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "44",
         "Where does age-related hypermethylation preferentially occur?",
         "At CpG islands, at bivalent chromatin domain promoters that are associated with key developmental genes, and at Polycomb-group protein targets.",
         "Age-related hypermethylation occurs preferentially at CpG islands [8], at bivalent chromatin domain promoters that are associated with key developmental genes [15], and at Polycomb-group protein targets [10] (Document 0).",
         "['Steve Horvath et al.,2013, DOI:https://doi.org/10.1186/gb-2013-14-10-r115\\nAn increasing body of evidence suggests that many manifestations of aging are epigenetic [1,2]. This article focuses on one particular type of epigenetic control: cytosine-5 methylation within CpG dinucleotides (also known as DNA methylation). Age-related DNA hypomethylation has long been observed in a variety of species, including salmon [3], rats [4], and mice [5]. More recent studies have shown that many CpGs are subject to age-related hypermethylation. A vast literature characterizes genes or genomic regions that either get hypermethylated or hypomethylated with age [6-14]. Previous studies have shown that age-related hypermethylation occurs preferentially at CpG islands [8], at bivalent chromatin domain promoters that are associated with key developmental genes [15], and at Polycomb-group protein targets [10]. The epigenomic landscape varies markedly across tissue types [16-18] and many age-related changes depend on tissue type [8,19]', 'Steve Horvath et al.,2013, DOI:https://doi.org/10.1186/gb-2013-14-10-r115\\n. Analysis of 6,000 cancer samples from 32 datasets showed that all of the considered 20 cancer types exhibit significant age acceleration, with an average of 36 years. Low age-acceleration of cancer tissue is associated with a high number of somatic mutations andTP53 mutations, while mutations in steroid receptors greatly accelerate DNA methylation age in breast cancer. Finally, I characterize the 353 CpG sites that together form an aging clock in terms of chromatin states and tissue variance. Conclusions:I propose that DNA methylation age measures the cumulative effect of an epigenetic maintenance system. This novel epigenetic clock can be used to address a host of questions in developmental biology, cancer and aging research.', 'Steve Horvath et al.,2013, DOI:https://doi.org/10.1186/gb-2013-14-10-r115\\n. DNA methylation profiling and normalization steps All of the public Illumina DNA data were generated by following the standard protocol of Illumina methylation assays, which quantifies DNA methylation levels by the β value. A detailed description of the pre-processing and data normalization steps is provided in Additional file 2. Meta analysis for measuring pure age effects (irrespective of tissue type) Iu s e dt h emetaAnalysis Rf u n c t i o ni nt h eW G C N AR package [86] to measure pure age effects (Additional file 9) as detailed in Additional file 2. Analysis of variance for measuring tissue variation To measure tissue effects in the training data (Additional file 8), I used analysis of variance (ANOVA) to calculate an F statistic as follows. First, a multivariate regression model was used to regress each CpG (dependent variable) on age and tissue type. The analysis adjusted for age since the different data sets have very different mean ages (Additional file 1)', 'Steve Horvath et al.,2013, DOI:https://doi.org/10.1186/gb-2013-14-10-r115\\nBackground: It is not yet known whether DNA methylation levels can be used to accurately predict age across a broad spectrum of human tissues and cell types, nor whether the resulting age prediction is a biologically meaningful measure. Results:I developed a multi-tissue predictor of age that allows one to estimate the DNA methylation age of most tissues and cell types. The predictor, which is freely available, wasdeveloped using 8,000 samples from 82 Illumina DNA methylation array datasets, encompassing 51 healthy tissues and celltypes. I found that DNA methylation age has the following properties: first, it is close to zero for embryonic and induced pluripotent stem cells; second, it correlates with cell passage number; third, it gives rise to a highly heritable measure of age acceleration; and, fourth, it is applicable to chimpanzee tissues', 'Steve Horvath et al.,2013, DOI:https://doi.org/10.1186/gb-2013-14-10-r115\\n. But several recent studies have shown that age-dependent CpG signatures can be defined independently of sex, tissue type, disease state and array platform [10,13-15,20-22]. While several recent articles describe age predictors based on DNAm levels in specific tissues (for example, saliva or blood [23,24]), it is not yet known whether age can be predicted irrespective of tissue type using a single predictor. Here I use an unprecedented collection of publicly available DNA methylation data sets for defining and evaluating an age predictor. Its astonishing accuracy across most tissues and cell types justifies its designation as a multi-tissue age predictor. Its age prediction, referred to as DNAm age, can be used as a biomarker for addressing a host of questions arising in aging research and related fields', 'Carlos López-Otín et al.,2013, DOI:https://doi.org/10.1016/j.cell.2013.05.039\\nSomeya et al., 2010\\n). Very recently, overexpression of SIRT3 has been reported to reverse the regenerative capacity of aged hematopoietic stem cells (\\nBrown et al., 2013\\n). Therefore, in mammals, at least three members of the sirtuin family, SIRT1, SIRT3 and SIRT6, contribute to healthy aging.\\nDNA methylation\\nThe relationship between DNA methylation and aging is complex. Early studies described an age-associated global hypomethylation, but subsequent analyses revealed that several loci, including those corresponding to various tumor suppressor genes and Polycomb target genes, actually become hypermethylated with age (\\nMaegawa et al., 2010\\n). Cells from patients and mice with progeroid syndromes exhibit DNA methylation patterns and histone modifications that largely recapitulate those found in normal aging (\\nOsorio et al., 2010\\n;\\nShumaker et al., 2006', 'Steve Horvath et al.,2013, DOI:https://doi.org/10.1186/gb-2013-14-10-r115\\n. Myosatellite cells may effectively rejuvenate the DNAm age of skeletal muscle tissue. Similarly, the recruitment of stem cells into cardiomyocytes for new cardiac muscle formation could explain why human heart tissue tends to have a low DNAm age. Carefully designed studies will be needed to test these hypotheses. The age correlation in a data set is determined by the standard deviation of age In the following, I describe non-biological reasons that affect the accuracy (age correlation) of the age predictor. To address how well the age predictor works in individual data sets, I used two different approaches. First, I applied the age predictor to individual data sets (see columns‘Cor Figure 1Chronological age (y-axis) versus DNAm age (x-axis) in the training data.Each point corresponds to a DNA methylation sample (human subject). Points are colored and labeled according to the underlying data set as described in Additional file 1']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response accurately identifies the three locations where age-related hypermethylation preferentially occurs: CpG islands, bivalent chromatin domain promoters associated with key developmental genes, and Polycomb-group protein targets. The response is completely correct and factual.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "45",
         "What do embryonic signals act on to induce immune tolerance?",
         "They act on maternal immune cells.",
         "Embryonic signals act on maternal immune cells to induce immune tolerance (0).",
         "['Hiroshi Fujiwara et al.,2020, DOI:https://doi.org/10.3390/ijms21051885\\nEmbryo implantation in the uterus is an essential process for successful pregnancy in mammals. In general, the endocrine system induces sufficient embryo receptivity in the endometrium, where adhesion-promoting molecules increase and adhesion-inhibitory molecules decrease. Although the precise mechanisms remain unknown, it is widely accepted that maternal–embryo communications, including embryonic signals, improve the receptive ability of the sex steroid hormone-primed endometrium. The embryo may utilize repulsive forces produced by an Eph–ephrin system for its timely attachment to and subsequent invasion through the endometrial epithelial layer. Importantly, the embryonic signals are considered to act on maternal immune cells to induce immune tolerance. They also elicit local inflammation that promotes endometrial differentiation and maternal tissue remodeling during embryo implantation and placentation', 'Hiroshi Fujiwara et al.,2020, DOI:https://doi.org/10.3390/ijms21051885\\n. Additional clarification of the immune control mechanisms by embryonic signals, such as human chorionic gonadotropin, pre-implantation factor, zona pellucida degradation products, and laeverin, will aid in the further development of immunotherapy to minimize implantation failure in the future.', 'Micol Massimiani et al.,2019, DOI:https://doi.org/10.3390/ijms21010023\\n193\\n]. During early pregnancy, in the decidua there is an increase in Tregs, which produce immunosuppressive cytokines, such as IL-10, for inducing immune tolerance [\\n194\\n,\\n195\\n,\\n196\\n,\\n197\\n]. Other cells involved in maternal immune tolerance are the uNK, a particular type of NK cells, which lose their cytotoxic functions during pregnancy andplay a supportive role by enhancing angiogenesis. uNK cells induce immune tolerance by reducing inflammation through interferon-γ (IFN-γ) [\\n198\\n] and by inhibiting the function of T cells through the expression of immunomodulatory molecules such as galectin-1 and glycodelin A [\\n199\\n].\\n2.10. Endometrial Receptivity Array\\nRecently, a customized endometrial receptivity array (ERA), containing 238 genes related to endometrial receptivity, was created [\\n200', 'Hsien-Ming Wu et al.,2022, DOI:https://doi.org/10.3390/ijms232113382\\nFunctional embryo–maternal interactions occur during embryo implantation and placentation, contributing to a successful pregnancy [64]. Embryo implantation is a vital step in the complex embryo–maternal interactions, indicating that a well-ready and receptive endometrium recognizes and tolerates a good-quality embryo [65]. Cytokines released from the endometrium regulate these complex embryo–maternal interactions throughout pregnancy as the immunoregulatory roles in decidualization and placentation [66]. In order to prime a receptive endometrium, several immune cells, including uNK cells and macrophages, are initiated to release growth factors and proinflammatory cytokines modulating receptive decidualization and maternal tolerance to embryo invasion [67,68,69]. All through the pregnancy, cytokines establish complex immunological stability between inflammatory and anti-inflammatory reactions essential for balancing embryo–maternal interactions [70]', 'Hiroshi Fujiwara et al.,2020, DOI:https://doi.org/10.3390/ijms21051885\\n107\\n]. To achieve this, the developing embryo should convey species- and embryo-specific signals, i.e., embryonic signals, to the maternal immune system. Early pregnancy factors that inhibit T-cell-induced rosette formation were reported to be detected just after fertilization [\\n111\\n]. Later, several substances, such as platelet activating factor, thioredoxin, and chaperonin 10, were proposed as early pregnancy factors [\\n112\\n,\\n113\\n,\\n114\\n]. It was also reported that the embryo-derived pre-implantation factor (PIF), a novel embryo-specific 18-kDa peptide that is specifically expressed on the fetus and placenta [\\n115\\n], was secreted at the two-cell stage and was able to be detected in the maternal sera prior to implantation [\\n116\\n]. Its many biological roles are under investigation [\\n115\\n,\\n117\\n,\\n118\\n].', 'Xiaoxiao Li et al.,2020, DOI:https://doi.org/10.1042/BSR20193938\\nThe MEred module which was significantly associated with the effect of progesterone was found to be mainly enriched in pathways related to cellular defense response, NK cell-mediated cytotoxicity, and antigen processing and presentation. All these pathways were related to immune response. Multiple studies have revealed that REPL is closely related to the dysregulation of immune response [\\n27\\n,\\n28\\n]. Our result suggested that the hub genes in the MEred module may regulate cellular defense response to affect the early embryo implantation. In addition, NK cells are a class of cytotoxic effector cells involved in the innate immune system that respond to a variety of cytokines such as type I interferon and interleukins, which help the host to produce immune responses [\\n29\\n]. In lymph nodes, NK cell-mediated killing of target cells affects T-cell responses, possibly by reducing antigen loading and/or target cell debris to promote antigen cross-presentation to CD8\\n+\\ncytotoxic T cells [\\n30', 'Hsien-Ming Wu et al.,2022, DOI:https://doi.org/10.3390/ijms232113382\\nThe embryo–endometrial communication has a critical role in the successful embryo implantation and pregnancy [1]. Maternal immune modulation during embryo implantation and subsequent pregnancy provides a tolerant microenvironment to keep the fetus from rejection [2]. Since the first human IVF baby was born in 1978, pregnancy and live birth rates following assisted reproductive technologies have improved significantly. Even so, there is still a gap in achieving higher pregnancy and live birth rates, attributed to unresolved problems in embryo quality, endometrium receptivity, and embryo–endometrium interaction [3,4]. Usually, a steroid hormone primes the endometrium for the receptive phase; after that, it is further immunomodulated by embryo signals for subsequent embryo implantation and maintenance of pregnancy [5]. Therefore, successful embryo implantation and the ensuing pregnancy depend on the coordinated interaction between the endometrium and embryos']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response is completely correct and factual. The response identifies that embryonic signals act on maternal immune cells.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "46",
         "What percentage of infants born globally each year are conceived using assisted reproductive technology?",
         "Approximately 0.3% of all infants born every year are conceived using assisted reproductive technology (ART) treatments.",
         "Globally, ART babies represent about 0.3% of the total live birth rate every year [4, 5, 8].",
         "['Andres Reig et al.,2020, DOI:https://doi.org/10.1007/s10815-020-01739-0\\nn= 3789 embryos transferred), 35–37 (\\nn= 2200), 38–40 (\\nn= 1624), 41–42 (\\nn= 319), and > 42 (\\nn= 243). Implantation rate (IR), clinical pregnancy rate (CPR), and LBR were calculated for each group as a percentage of embryos transferred and compared. CPR was also analyzed as a percentage of implanted pregnancies, and LBR as a percentage of clinical pregnancies, to determine when age has the greatest impact. These results were then adjusted for confounding variables via a multivariate logistic regression model.\\nResults', 'Evelyn Lee1 and Jinhui Zhang2 et al.,2022, DOI:https://doi.org/10.1186/s12913-022-08485-2\\nData sources\\nThe model inputs are summarised in Table 1. As live birth rates for different strategies (standard  autologous ART, PGT-A, social freezing and donor ART)  were summarised and reported by 5-year interval in the  literature, logistic regression was used to obtain age-spe - cific live birth rate. The regression formulas are presented  in Appendix A (Supplementary Materials).\\nUnless otherwise stated, live-birth rates using autolo - gous fresh, FET and donated ART cycles, OPU rate, cycle  cancelation, transition probabilities and discontinuation  rates were sourced from the Australian and New Zea - land Assisted Reproduction Database (ANZARD) held  at the National Perinatal Epidemiology and Statistics  Unit of the University of New South Wales, Sydney [18].  ANZARD collects information on all ART treatment  cycles undertaken in Australia and New Zealand, includ - ing the resulting treatment and pregnancy outcomes.\\nPGT-A', 'Evelyn Lee1 and Jinhui Zhang2 et al.,2022, DOI:https://doi.org/10.1186/s12913-022-08485-2\\nAssisted reproductive technology (ART) has revolution - ised the treatment of infertility over the last three dec - ades, with more than 2.5 million ART cycles performed  annually and the birth of over 6.5 million children con - ceived worldwide to date [1]. Despite the mainstream  acceptance of ART treatment and improvements in  overall success rates, women of advanced maternal age  (AMA) defined as ≥35 years of age, remain a challenge  for fertility physician [2, 3]. According to the latest report  from the Centers for Disease Control and Prevention on  ART cycles performed in the United States (US) in 2018,  the live-birth rate per initiated cycle for women aged  41-42 years was 13.2% compared to 55.1% for women  aged below 35 years [4].', 'Mauro Cozzolino et al.,2020, DOI:https://doi.org/10.1007/s10815-020-01948-7\\nCategorical and continuous variables are presented as percentages or means with standard deviations or 95% confidence intervals (95% CI). To compare means, ANOVA tests were used with Bonferroni post-hoc tests. To control possible bias in multivariate analyses and to calculate adjusted ORs, logistic regression analysis was performed considering the proportion of ET that used donated oocytes, the day of ET, the mean number of embryos transferred per procedure, age, fresh/frozen status of the embryo, and the mean of the prior number of embryos transferred per patient as clinically relevant variables. Because this study was conducted over a 5-year period and couples may have had consecutive ET with different diagnostic techniques (e.g., PGT-A and then ERA+PGT-A), the groups lack independence and there is potential correlation between data from each group', 'Carlo Alviggi et al.,2020, DOI:https://doi.org/10.1186/s12958-020-00605-z\\nWhile it is prudent to advocate temporary social distancing and closure of non-emergency health services, we would like to point at what a prolonged lockdown of fertility treatment might mean for the society as a whole, and infertility patients in general. This consideration will focus on assisted reproductive technology (ART) treatment, particularly in the low prognosis patients who represent approximately 30- 50% of patients seeking ART [ 4]. Conservative estimates indicate that over 1.5 million IVF cycles are carried out every year worldwide, resulting in approximately 400,000 babies born [ 5]. In the United Kingdom and the United States, approximately 3 and 2%, respectively, of all infants born every year are conceived using ART [ 6, 7]. Globally, ART babies represent about 0.3% of the total live birth rate every year [ 5, 8]', 'Patricia Sebastian-Leon et al.,2025, DOI:https://doi.org/10.1016/j.fertnstert.2025.05.150\\nI n the last decades, the average age at first childbirth and the number of births in women over the age of 35 have risen  because of modern women’s professional aspirations, socioeconomic factors, and the improvement of assisted reproductive technologies (ART) , among other reasons. The age-related decline in women’s fertility begins at the age of 30 and is exacerbated between the ages of 35- 40 . In this context, the consistently increasing number of couples undergoing ART  Received October 23, 2024; revised May 9, 2025; accepted May 12, 2025. P.S.-L. and F.J.-S. contributed equally to this work and should be regarded as jointfirst authors. Funded by IVI Foundation (2203-FIVI-030-PD) and Instituto de Salud Carlos III (ISCIII; PI23/00806) cofunded by the European Union. P.D.-G. was supported by the Instituto de Salud Carlos III (ISCIII; Spanish Ministry of Science and Innovation) through the Miguel Servet program (CP20/00118) cofounded by the European Union. P.S.-L. and F.J.-S', 'Andres Reig et al.,2020, DOI:https://doi.org/10.1007/s10815-020-01739-0\\nWhile these changes have obvious positive effects on society, the age-related decline in female fertility poses a significant challenge regarding the ability to conceive later in life [4]. A response to this societal need has been the increased utilization of assisted reproductive technologies (ART) in older reproductive age women. In 2003, the number of fresh, non-donor cycles performed in women older than 42 in the USA was 3203 [5]. This number increased by more than 300% to 9996 in 2016 [6]. Importantly, despite significant advances in ART, live birth rate (LBR) per retrieval in women older than 42 has not improved and remains at approximately 4%. These dynamics also contribute to the rapid increase in the number of IVF cycles performed using donor oocytes, from 11,627 in 2003 to 20,391 in 2016 [5–7].']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response accurately states that ART babies represent about 0.3% of the total live birth rate every year, which aligns with the reference answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "47",
         "What new functionality was introduced with the release of CADD v1.4?",
         "CADD v1.4 supports direct variant interpretation on GRCh38.",
         "CADD v1.4 supports direct (non-lifted) variant interpretation on GRCh38 and shows that the available annotations provide a similar level of accuracy to those generated for GRCh37 (6).",
         "['Philipp Rentzsch et al.,2018, DOI:https://doi.org/10.1093/nar/gky1016\\n23\\n,\\n25–28\\n), as well as variants identified in population-based studies (e.g.\\n29\\n). Since its introduction in 2014, CADD has become one of the most widely used tools to assess human genetic variation, and other tools and scores often use CADD to benchmark their performance; according to Google Scholar CADD has been cited 1984 times (as of 15 September\\xa02018) with about 24 000 unique users of its website over the last year.\\nFurthermore, CADD has also seen applications in evolutionary studies, ranging from the interpretation of evolutionary changes (\\n30–32\\n) to the theoretical investigation of variant fitness effects in human populations (\\n33\\n).\\nThe release of CADD has also spurred the development of several other genome-wide predictors. For instance, the feature set from CADD has been used to train Deep Neural Networks (e.g. DANN (\\n34\\n)), and CADD’s underlying approach and training set definition methodology has been adapted for other model organisms (\\n35', 'Philipp Rentzsch et al.,2018, DOI:https://doi.org/10.1093/nar/gky1016\\nSupplementary Table S3\\n), the main change between these releases was the choice of the machine learning algorithm and software library. A major challenge in training a CADD model is the size of the fully annotated training dataset, which comprises hundreds of gigabytes if stored naively. This is difficult to handle in active working memory, and therefore needs to be kept in a sparse matrix representation or handled using other computational techniques. While CADD v1.0 used a linear support vector machine implemented in the LIBOCAS library (\\n42\\n), later models used L\\n2\\n-regularized logistic regression implemented in GraphLab Create (\\n43\\n). For the latest release, CADD v1.4, a logistic regression model was fit using a fully open source pipeline based on SciPy (\\n44\\n) and scikit-learn (\\n45\\n). All libraries permit model training in sparse matrix format, with major benefits in terms of run time and memory requirements.', 'Philipp Rentzsch et al.,2018, DOI:https://doi.org/10.1093/nar/gky1016\\nhttps://cadd.gs.washington.edu/\\nor else by using a local CADD installation. In order to upload data to our web server, users must confirm that they are authorized to upload the data, that their upload does not contain any identifiable information, and that they understand that our server does not require user registration and that therefore data is accessible by decrypting URLs. Users, who are unable to confirm this, have the option to score variants offline, using a local CADD installation. Given a variant to be scored from a variant set, the CADD score is either retrieved from an already pre-computed file (e.g. a file of CADD scores for all ∼9 billion potential SNVs) or else obtained by annotating the variant and applying the previously-fitted model. The PHRED-scaled score is looked up in a conversion table and both scores are returned to the user. In addition, the user may request that the output files contain the variant annotations used to create the CADD score.', 'Philipp Rentzsch et al.,2018, DOI:https://doi.org/10.1093/nar/gky1016\\n56\\n,\\n57\\n) may also be integrated into CADD in the near future.\\nFurther improvement of CADD could also come in terms of a more complex, structured model that combines features via linear or non-linear interactions. Currently, CADD includes features obtained by taking the product of VEP-predicted variant consequences with a number of annotations, such as conservation and transcript position. In the future, a more sophisticated and streamlined approach could be applied in order to allow for non-linearity and interactions within CADD. However, this must be performed with care, as the risk of overfitting such complex models is high.', 'Philipp Rentzsch et al.,2018, DOI:https://doi.org/10.1093/nar/gky1016\\n. However, for an alternative view on this topic, we refer the reader to recent methods that use CADD scores in conjunction with hard cutoffs; see GAVIN (', 'Chia-Hao Chin et al.,2014, DOI:https://doi.org/10.1186/1752-0509-8-S4-S11\\nImplementation', 'Philipp Rentzsch et al.,2018, DOI:https://doi.org/10.1093/nar/gky1016\\nIn this manuscript, we presented an overview of recent updates to CADD, as well as the services that we provide in order to make those scores available and maximally useful to the scientific community. In addition to better documentation and a fresh web layout, we substantially expanded the options for how users can access scores by providing website and API lookups, genome browser tracks and an easy-to-install offline scoring script. With the release of CADD v1.4, we support direct (non-lifted) variant interpretation on GRCh38\\xa0and show that the available annotations provide a similar level of accuracy to those generated for GRCh37.\\nA key strength of CADD is that the model is trained on a very large training set that does not suffer from ascertainment bias inherent to curated sets of pathogenic and benign variants such as ClinVar (\\n7\\n) or HGMD (\\n58\\n). CADD shares this strength with only a few other scores, such as Eigen (\\n59\\n), LINSIGHT (\\n60\\n) and CDTS (\\n61']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response accurately states that CADD v1.4 supports direct variant interpretation on GRCh38. It also provides additional accurate information, which doesn't detract from the correctness.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "48",
         "What can TFF2 and MUC6 be used as a biological marker for?",
         "TFF2 and MUC6 may be used as a biological marker for the process of antralisation of the gastric incisura.",
         "Aberrant TFF2 and MUC6 expression might be a useful biological marker for antralisation (Document 4).",
         "['H H-X Xia et al.,Not identified, DOI:https://doi.org/10.1136/jcp.2003.015487\\n26\\nImmunohistochemistry for TFF2 and MUC6 in gastric epithelium\\nBiopsy specimens taken from the antral, body, and incisura of all patients were subjected to immunohistochemistry for the assessment of TFF2 and MUC6 expression in the gastric epithelium. Briefly, tissues embedded in paraffin wax were cut into sections 4 µm thick, dewaxed, and dehydrated for 10 minutes. The sections were treated in 3% hydrogen peroxide for five minutes, and then incubated with monoclonal antibody against TFF2 (undiluted), MUC6 (diluted 1/100; both from Novocastra Laboratories Ltd, Newcastle upon Tyne, UK) overnight at 4°C, followed by the application of biotinylated link antibody and streptavidin–horseradish peroxidase (LSAB kit; Dako Corporation, Carpinteria, California, USA). Finally, the sections were developed with diaminobenzidine–hydrogen peroxidase substrate, and lightly counterstained with haematoxylin.', 'H H-X Xia et al.,Not identified, DOI:https://doi.org/10.1136/jcp.2003.015487\\nIn conclusion,\\nH\\xa0pylori\\ninfection increases the expression of TFF2 and MUC6 in the gastric epithelium, which may be a protective response of the host to the mucosal damage induced by\\nH\\xa0pylori\\ninfection. Aberrant expression of TFF2 and MUC6 is associated with antralisation of the gastric incisura, and thus may be involved in and used as a biological marker for the process.', 'H H-X Xia et al.,Not identified, DOI:https://doi.org/10.1136/jcp.2003.015487\\nH\\xa0pylori\\nnegative patients, but that 72% of\\nH\\xa0pylori\\npositive patients expressed MUC6 in the surface mucous cells.\\n3\\nMoreover, our present study showed that both TFF2 and MUC6 were also expressed in the deeper glands of the body, especially the incisura mucosa, in the presence of\\nH\\xa0pylori\\ninfection.\\nTFF2 is upregulated in diverse pathological conditions, and its expression profile broadens to include the regenerative epithelia of the entire gastrointestinal tract.\\n27\\nFor example, expression of TFF2 and other trefoil peptides is increased at sites of gastric ulceration, duodenal ulceration, and Crohn’s disease.\\n28–\\n30\\nThus, increased expression of TFF2 may be a potentially important mucosal defence mechanism. Previous studies have shown that there is an association between TFFs and mucins.\\n1,\\n9–\\n11\\nTFFs interact with mucous, and stabilise the mucous gel by binding the long mucous molecules together.\\n17', 'H H-X Xia et al.,Not identified, DOI:https://doi.org/10.1136/jcp.2003.015487\\n17\\nIt has been suggested that TFF2 may function with MUC6 to strengthen the mucous barrier.\\n17\\nMoreover, Hoosein\\net al\\nshowed that a highly purified preparation of porcine TFF2 was able to stimulate [3H] thymidine incorporation in HCT116 colon carcinoma cells, suggesting a role in cell proliferation.\\n31\\nRecently, Farrell\\net al\\nalso showed that TFF2 deficient mice have a 35% reduction in the fundic proliferation rate compared with wild-type mice.\\n18\\nThose results suggest a physiological role for TFF2 in promoting mucosal healing through the stimulation of cell proliferation.\\n“Our present study showed that both trefoil family factor 2 and MUC6 were also expressed in the deeper glands of the body, especially the incisura mucosa, in the presence of\\nHelicobacter pylori\\ninfection”\\nHelicobacter pylori\\ninfection is associated with antralisation (or so called pseudopyloric metaplasia) at the gastric incisura, body, and fundus.\\n23,\\n24,\\n32\\nIt has been established that\\nH\\xa0pylori', 'H H-X Xia et al.,Not identified, DOI:https://doi.org/10.1136/jcp.2003.015487\\nH\\xa0pylori\\ninfection\\nAberrant TFF2 and MUC6 expression is associated with antralisation of gastric incisura\\nThis increased expression of TFF2 and MUC6 might be a useful biological marker for antralisation', 'H H-X Xia et al.,Not identified, DOI:https://doi.org/10.1136/jcp.2003.015487\\nStatistical analysis\\nData were expressed as the percentage of positive cases over the total number of cases. Statistical analysis was performed using SPSS system (version 10.0; SPSS Inc, Chicago, Illinois, USA). The association between the expression of TFF2 and MUC6 and\\nH\\xa0pylori\\ninfection at different gastric sites was assessed using the χ\\n2\\ntest, with Yates’s correction if required, or Fisher’s exact test; odds ratios (OR), 95% confidence intervals (CI), and two tailed p values were calculated. The α level of significance was set at p < 0.05.', 'H H-X Xia et al.,Not identified, DOI:https://doi.org/10.1136/jcp.2003.015487\\nRESULTS\\nExpression of TFF2 and MUC6 in normal gastric epithelium at different gastric sites\\nIn the normal stomach, staining for TFF2 and MUC6 in the gastric foveola was very weak or not discernable. In the gastric antrum, there was uniform staining of TFF2 and MUC6 in the regenerative zone and the deeper mucous glands (fig 1). In gastric body mucosa, TFF2 and MUC6 were expressed together only in the regenerative zone, where mucous neck cells were present (fig 2). In gastric incisura mucosa, the expression of TFF2 and MUC6 was limited to the regenerative zone in most cases (fig 3; table 1).\\nFigure 1.\\nOpen in a new tab\\nThe expression of trefoil family factor 2 (TFF2) and MUC6 in normal antral mucosa. There is uniform staining of (A) TFF2 and (B) MUC6 in the regenerative zone and the deeper mucous glands (arrows). Original magnification, ×100.\\nFigure 2.\\nOpen in a new tab']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "Feedback: The response accurately identifies TFF2 and MUC6 as potential biological markers for antralisation, aligning with the reference answer.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ],
        [
         "49",
         "What distinguishes the 1000 Genomes Project's data sharing policy from other large WGS datasets?",
         "Samples within the 1kGP cohort were consented for full public release of genetic information, allowing unrestricted sharing.",
         "The 1000 Genomes Project (1kGP) samples have been consented for full public release of genetic information, allowing unrestricted sharing of complete sample-level GT data (2). Unlike the 1kGP, other resources often have restrictions on public data sharing because they are linked to clinical data, which amplifies privacy concerns (2).",
         "['Marta Byrska-Bishop et al.,2022, DOI:https://doi.org/10.1016/j.cell.2022.08.004\\nThe 1000 Genomes Project (1kGP) was the first large-scale whole-genome sequencing (WGS) effort to deliver a catalog of human genetic variation (\\nSudmant et\\xa0al., 2015\\n;\\nThe 1000 Genomes Project Consortium, 2010\\n,\\n2012\\n,\\n2015\\n). The project sampled participants from 26 populations across five continental regions of the world. It culminated in 2015 with publication of the\\xa0final, phase 3 variant call set (\\nSudmant et\\xa0al., 2015\\n;\\nThe 1000 Genomes Project Consortium, 2015\\n) consisting of 2,504 unrelated samples, a subset of which is from the HapMap collection (\\nThe International HapMap 3 Consortium, 2010', 'Marta Byrska-Bishop et al.,2022, DOI:https://doi.org/10.1016/j.cell.2022.08.004\\nSince the completion of phase 3, much larger WGS datasets have been released such as the Genome Aggregation Database (gnomAD; 76,156 WGS samples) (\\nKarczewski et\\xa0al., 2020\\n), Trans-Omics for Precision Medicine (TOPMed, ∼180,000 samples) (\\nTaliun et\\xa0al., 2021\\n), All of Us (∼100,000 samples), or the UK Biobank (UKBB, 200,000 samples) (\\nHalldorsson et\\xa0al., 2022', 'Marta Byrska-Bishop et al.,2022, DOI:https://doi.org/10.1016/j.cell.2022.08.004\\nHalldorsson et\\xa0al., 2022\\n). Unlike the 1kGP, these resources have restrictions on public data sharing as they are often linked to clinical data, which amplifies privacy concerns. In contrast, samples within the 1kGP cohort have been consented for full public release of genetic information, which allows for unrestricted sharing of the complete sample-level GT data. This enables granting access to a downloadable reference imputation panel, as well as use of the dataset for methods development and benchmarking, among other applications. A small subset of the 602 pedigrees that are now part of the expanded 1kGP cohort have been sequenced previously as part of various efforts, such as Platinum Genomes (\\nEberle et\\xa0al., 2017\\n), Complete Genomics (\\nThe 1000 Genomes Project Consortium, 2015\\n), and the Human Genome Structural Variation Consortium (HGSVC) (\\nChaisson et\\xa0al., 2019\\n;\\nEbert et\\xa0al., 2021', 'The 1000 Genomes Project Consortium et al.,2015, DOI:https://doi.org/10.1038/nature15393\\n. Because of the wide availability of the data and samples, these samples have been and will continue to be used for studying many molecular phenotypes. Thus, we predict that the samples will accumulate many types of data that will allow connections to be drawn between variants and both molecular and disease phenotypes.', 'The 1000 Genomes Project Consortium et al.,2015, DOI:https://doi.org/10.1038/nature15393\\nOver the course of the 1000 Genomes Project there have been substantial advances in sequence data generation, archiving and analysis. Primary sequence data production improved with increased read length and depth, reduced per-base errors, and the introduction of paired-end sequencing. Sequence analysis methods improved with the development of strategies for identifying and filtering poor-quality data, for more accurate mapping of sequence reads (particularly in repetitive regions), for exchanging data between analysis tools and enabling ensemble analyses, and for capturing more diverse types of variants. Importantly, each release has examined larger numbers of individuals, aiding population-based analyses that identify and leverage shared haplotypes during genotyping', 'Javier Lopez et al.,2017, DOI:https://doi.org/10.1093/nar/gkx445\\nBy providing efficient remote access HGVA prevents massive downloads of data required to use genomic allele frequencies in prioritization pipelines. Data size and complexity will increase as technologies and general reference resources improve. Movements of large datasets should be avoided since it is an expensive operation in terms of time and storage space. Local installations of databases are difficult to maintain and tracking of source database versions used in the analysis affects long term reproducibility of analysis results. Users of HGVA will always automatically access the most recent versions of reference data without the need of periodical downloads and re-installations involving hundreds of Gigabytes. The current release comprises over 650GB of compressed data. Moreover, novel communication protocols such as the gRPC, rarely used in Bioinformatics, offer powerful streaming features with high performance, scalability and flexibility', 'Peter H. Sudmant et al.,2015, DOI:https://doi.org/10.1038/nature15394\\nWe mapped Illumina WGS data (,100 bp reads, mean 7.4-fold coverage) from 2,504 individuals onto an amended version 8 of the GRCh37 reference assembly using two independent mapping algorithms-BWA17 and mrsFAST18-and performed SV discovery and genotyping using an ensemble of nine different algorithms (Extended Data Fig. 1 and Supplementary Note). We applied several orthogonal experimental platforms for SV set assessment, refinement and characterization (Supplementary Table 2) and to calculate the false discovery rate (FDR) for each SV class (Table 1). Callset refinements facilitated through long-read sequencing enabled us to incorporate a number of additional SVs into our callset, including an additional 698 inversions and 9,132 small (,1 kbp) deletions, compared to the SV set released with the 1000 Genomes Project marker paper 16. As a result, our callset differs slightly relative to the marker paper’s SV set16 (see Supplementary Table 2)']",
         "chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7",
         "5",
         "The response accurately identifies the key distinction of the 1000 Genomes Project's data sharing policy, which is the unrestricted sharing of genetic information due to informed consent. It also correctly points out that other datasets often have restrictions due to links with clinical data and associated privacy concerns.",
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json"
        ]
       ],
       "shape": {
        "columns": 8,
        "rows": 630
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>question</th>\n",
       "      <th>true_answer</th>\n",
       "      <th>generated_answer</th>\n",
       "      <th>retrieved_docs</th>\n",
       "      <th>test_settings</th>\n",
       "      <th>eval_score_gemini-2.0-flash</th>\n",
       "      <th>eval_feedback_gemini-2.0-flash</th>\n",
       "      <th>settings</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>How does the FGB rs1800790A allele affect fibr...</td>\n",
       "      <td>In F13A 34Val/Val wildtypes, carriage of the F...</td>\n",
       "      <td>In F13A 34Val/Val wildtypes, carriage of FGB r...</td>\n",
       "      <td>[Christian Schwedler et al.,2021, DOI:https://...</td>\n",
       "      <td>chunk_1000_reader-model_gemini-2.0-flash_emned...</td>\n",
       "      <td>4</td>\n",
       "      <td>The response accurately states that the FGB rs...</td>\n",
       "      <td>./output/rag_chunk_1000_reader-model_gemini-2....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>For which patient group might the ERA test be ...</td>\n",
       "      <td>The ERA test may be helpful for women with sus...</td>\n",
       "      <td>The ERA test may be helpful in women with susp...</td>\n",
       "      <td>[Zion Ben Rafael et al.,2021, DOI:https://doi....</td>\n",
       "      <td>chunk_1000_reader-model_gemini-2.0-flash_emned...</td>\n",
       "      <td>5</td>\n",
       "      <td>Feedback: The response accurately identifies t...</td>\n",
       "      <td>./output/rag_chunk_1000_reader-model_gemini-2....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>What kind of values does the Color Pathway too...</td>\n",
       "      <td>The Color Pathway tool accepts numerical values.</td>\n",
       "      <td>The Color Pathway tool in KEGG Mapper accepts ...</td>\n",
       "      <td>[Minoru Kanehisa et al.,2011, DOI:https://doi....</td>\n",
       "      <td>chunk_1000_reader-model_gemini-2.0-flash_emned...</td>\n",
       "      <td>5</td>\n",
       "      <td>The response accurately states that the Color ...</td>\n",
       "      <td>./output/rag_chunk_1000_reader-model_gemini-2....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>What is the implantation potential of an euplo...</td>\n",
       "      <td>Once an euploid blastocyst is identified, its ...</td>\n",
       "      <td>An euploid blastocyst transfer results in a he...</td>\n",
       "      <td>[Andres Reig et al.,2020, DOI:https://doi.org/...</td>\n",
       "      <td>chunk_1000_reader-model_gemini-2.0-flash_emned...</td>\n",
       "      <td>4</td>\n",
       "      <td>Feedback: The response provides a similar answ...</td>\n",
       "      <td>./output/rag_chunk_1000_reader-model_gemini-2....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>What does the PRISMA 2020 statement reflect?</td>\n",
       "      <td>The PRISMA 2020 statement reflects advances in...</td>\n",
       "      <td>The PRISMA 2020 statement reflects advances in...</td>\n",
       "      <td>[Matthew J Page et al.,2021, DOI:https://doi.o...</td>\n",
       "      <td>chunk_1000_reader-model_gemini-2.0-flash_emned...</td>\n",
       "      <td>5</td>\n",
       "      <td>Feedback: The response accurately reflects the...</td>\n",
       "      <td>./output/rag_chunk_1000_reader-model_gemini-2....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>Which genes share genetic susceptibility for A...</td>\n",
       "      <td>The ESR1, HK3, and BRSK1 genes share genetic s...</td>\n",
       "      <td>The ESR1, HK3 and BRSK1 genes share genetic su...</td>\n",
       "      <td>[Yingying Qin et al.,2012, DOI:https://doi.org...</td>\n",
       "      <td>chunk_1000_reader-model_gemini-2.0-flash_emned...</td>\n",
       "      <td>5</td>\n",
       "      <td>The response accurately identifies the ESR1, H...</td>\n",
       "      <td>./output/rag_chunk_1000_reader-model_gemini-2....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>What database were the GSE26787 and GSE63901 d...</td>\n",
       "      <td>The Gene Expression Omnibus (GEO) database.</td>\n",
       "      <td>This question cannot be answered from the give...</td>\n",
       "      <td>[Minoru Kanehisa et al.,2011, DOI:https://doi....</td>\n",
       "      <td>chunk_1000_reader-model_gemini-2.0-flash_emned...</td>\n",
       "      <td>1</td>\n",
       "      <td>The response states that the question cannot b...</td>\n",
       "      <td>./output/rag_chunk_1000_reader-model_gemini-2....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>What percentage of women globally are affected...</td>\n",
       "      <td>3.7% of women globally.</td>\n",
       "      <td>Premature ovarian insufficiency affects 3.7% o...</td>\n",
       "      <td>[Ewa Rudnicka et al.,2018, DOI:https://doi.org...</td>\n",
       "      <td>chunk_1000_reader-model_gemini-2.0-flash_emned...</td>\n",
       "      <td>5</td>\n",
       "      <td>Feedback: The response is completely correct, ...</td>\n",
       "      <td>./output/rag_chunk_1000_reader-model_gemini-2....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>What is the purpose of unique molecular identi...</td>\n",
       "      <td>Unique molecular identifiers are applied to ov...</td>\n",
       "      <td>Unique molecular identifiers (UMIs) are applie...</td>\n",
       "      <td>[Hindrek Teder et al.,2018, DOI:https://doi.or...</td>\n",
       "      <td>chunk_1000_reader-model_gemini-2.0-flash_emned...</td>\n",
       "      <td>5</td>\n",
       "      <td>Feedback: The response is completely correct a...</td>\n",
       "      <td>./output/rag_chunk_1000_reader-model_gemini-2....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>What are pinopodes?</td>\n",
       "      <td>Pinopodes are apical protrusions on the epithe...</td>\n",
       "      <td>Pinopodes are uniquely large, transient plasma...</td>\n",
       "      <td>[Kelsey E Quinn et al.,2019, DOI:https://doi.o...</td>\n",
       "      <td>chunk_1000_reader-model_gemini-2.0-flash_emned...</td>\n",
       "      <td>3</td>\n",
       "      <td>The response mentions that pinopodes are relat...</td>\n",
       "      <td>./output/rag_chunk_1000_reader-model_gemini-2....</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>630 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             question  \\\n",
       "0   How does the FGB rs1800790A allele affect fibr...   \n",
       "1   For which patient group might the ERA test be ...   \n",
       "2   What kind of values does the Color Pathway too...   \n",
       "3   What is the implantation potential of an euplo...   \n",
       "4        What does the PRISMA 2020 statement reflect?   \n",
       "..                                                ...   \n",
       "58  Which genes share genetic susceptibility for A...   \n",
       "59  What database were the GSE26787 and GSE63901 d...   \n",
       "60  What percentage of women globally are affected...   \n",
       "61  What is the purpose of unique molecular identi...   \n",
       "62                                What are pinopodes?   \n",
       "\n",
       "                                          true_answer  \\\n",
       "0   In F13A 34Val/Val wildtypes, carriage of the F...   \n",
       "1   The ERA test may be helpful for women with sus...   \n",
       "2    The Color Pathway tool accepts numerical values.   \n",
       "3   Once an euploid blastocyst is identified, its ...   \n",
       "4   The PRISMA 2020 statement reflects advances in...   \n",
       "..                                                ...   \n",
       "58  The ESR1, HK3, and BRSK1 genes share genetic s...   \n",
       "59        The Gene Expression Omnibus (GEO) database.   \n",
       "60                            3.7% of women globally.   \n",
       "61  Unique molecular identifiers are applied to ov...   \n",
       "62  Pinopodes are apical protrusions on the epithe...   \n",
       "\n",
       "                                     generated_answer  \\\n",
       "0   In F13A 34Val/Val wildtypes, carriage of FGB r...   \n",
       "1   The ERA test may be helpful in women with susp...   \n",
       "2   The Color Pathway tool in KEGG Mapper accepts ...   \n",
       "3   An euploid blastocyst transfer results in a he...   \n",
       "4   The PRISMA 2020 statement reflects advances in...   \n",
       "..                                                ...   \n",
       "58  The ESR1, HK3 and BRSK1 genes share genetic su...   \n",
       "59  This question cannot be answered from the give...   \n",
       "60  Premature ovarian insufficiency affects 3.7% o...   \n",
       "61  Unique molecular identifiers (UMIs) are applie...   \n",
       "62  Pinopodes are uniquely large, transient plasma...   \n",
       "\n",
       "                                       retrieved_docs  \\\n",
       "0   [Christian Schwedler et al.,2021, DOI:https://...   \n",
       "1   [Zion Ben Rafael et al.,2021, DOI:https://doi....   \n",
       "2   [Minoru Kanehisa et al.,2011, DOI:https://doi....   \n",
       "3   [Andres Reig et al.,2020, DOI:https://doi.org/...   \n",
       "4   [Matthew J Page et al.,2021, DOI:https://doi.o...   \n",
       "..                                                ...   \n",
       "58  [Yingying Qin et al.,2012, DOI:https://doi.org...   \n",
       "59  [Minoru Kanehisa et al.,2011, DOI:https://doi....   \n",
       "60  [Ewa Rudnicka et al.,2018, DOI:https://doi.org...   \n",
       "61  [Hindrek Teder et al.,2018, DOI:https://doi.or...   \n",
       "62  [Kelsey E Quinn et al.,2019, DOI:https://doi.o...   \n",
       "\n",
       "                                        test_settings  \\\n",
       "0   chunk_1000_reader-model_gemini-2.0-flash_emned...   \n",
       "1   chunk_1000_reader-model_gemini-2.0-flash_emned...   \n",
       "2   chunk_1000_reader-model_gemini-2.0-flash_emned...   \n",
       "3   chunk_1000_reader-model_gemini-2.0-flash_emned...   \n",
       "4   chunk_1000_reader-model_gemini-2.0-flash_emned...   \n",
       "..                                                ...   \n",
       "58  chunk_1000_reader-model_gemini-2.0-flash_emned...   \n",
       "59  chunk_1000_reader-model_gemini-2.0-flash_emned...   \n",
       "60  chunk_1000_reader-model_gemini-2.0-flash_emned...   \n",
       "61  chunk_1000_reader-model_gemini-2.0-flash_emned...   \n",
       "62  chunk_1000_reader-model_gemini-2.0-flash_emned...   \n",
       "\n",
       "   eval_score_gemini-2.0-flash  \\\n",
       "0                            4   \n",
       "1                            5   \n",
       "2                            5   \n",
       "3                            4   \n",
       "4                            5   \n",
       "..                         ...   \n",
       "58                           5   \n",
       "59                           1   \n",
       "60                           5   \n",
       "61                           5   \n",
       "62                           3   \n",
       "\n",
       "                       eval_feedback_gemini-2.0-flash  \\\n",
       "0   The response accurately states that the FGB rs...   \n",
       "1   Feedback: The response accurately identifies t...   \n",
       "2   The response accurately states that the Color ...   \n",
       "3   Feedback: The response provides a similar answ...   \n",
       "4   Feedback: The response accurately reflects the...   \n",
       "..                                                ...   \n",
       "58  The response accurately identifies the ESR1, H...   \n",
       "59  The response states that the question cannot b...   \n",
       "60  Feedback: The response is completely correct, ...   \n",
       "61  Feedback: The response is completely correct a...   \n",
       "62  The response mentions that pinopodes are relat...   \n",
       "\n",
       "                                             settings  \n",
       "0   ./output/rag_chunk_1000_reader-model_gemini-2....  \n",
       "1   ./output/rag_chunk_1000_reader-model_gemini-2....  \n",
       "2   ./output/rag_chunk_1000_reader-model_gemini-2....  \n",
       "3   ./output/rag_chunk_1000_reader-model_gemini-2....  \n",
       "4   ./output/rag_chunk_1000_reader-model_gemini-2....  \n",
       "..                                                ...  \n",
       "58  ./output/rag_chunk_1000_reader-model_gemini-2....  \n",
       "59  ./output/rag_chunk_1000_reader-model_gemini-2....  \n",
       "60  ./output/rag_chunk_1000_reader-model_gemini-2....  \n",
       "61  ./output/rag_chunk_1000_reader-model_gemini-2....  \n",
       "62  ./output/rag_chunk_1000_reader-model_gemini-2....  \n",
       "\n",
       "[630 rows x 8 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "23117f2b",
   "metadata": {},
   "outputs": [],
   "source": [
    "result[\"eval_score_gemini-2.0-flash\"] = result[\"eval_score_gemini-2.0-flash\"].apply(lambda x: int(x) if isinstance(x, str) else 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "6d5ebaa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "result[\"eval_score_gemini-2.0-flash\"] = result[\"eval_score_gemini-2.0-flash\"]/5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "5e8551da",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "settings",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "eval_score_gemini-2.0-flash",
         "rawType": "float64",
         "type": "float"
        }
       ],
       "ref": "6ccca069-d92e-45da-990f-5ba9729de1ad",
       "rows": [
        [
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_4.json",
         "0.8444444444444444"
        ],
        [
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_5.json",
         "0.8761904761904762"
        ],
        [
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_6.json",
         "0.8825396825396825"
        ],
        [
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_True_k_4.json",
         "0.8857142857142857"
        ],
        [
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json",
         "0.8984126984126984"
        ],
        [
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_True_k_6.json",
         "0.9047619047619048"
        ],
        [
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_True_k_5.json",
         "0.9079365079365079"
        ],
        [
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_8.json",
         "0.9396825396825397"
        ],
        [
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_9.json",
         "0.9428571428571428"
        ],
        [
         "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_10.json",
         "0.9587301587301587"
        ]
       ],
       "shape": {
        "columns": 1,
        "rows": 10
       }
      },
      "text/plain": [
       "settings\n",
       "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_4.json     0.844444\n",
       "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_5.json     0.876190\n",
       "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_6.json     0.882540\n",
       "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_True_k_4.json      0.885714\n",
       "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_7.json     0.898413\n",
       "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_True_k_6.json      0.904762\n",
       "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_True_k_5.json      0.907937\n",
       "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_8.json     0.939683\n",
       "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_9.json     0.942857\n",
       "./output/rag_chunk_1000_reader-model_gemini-2.0-flash_emnedding_model_GIST-small-Embedding-v0recursive_False_k_10.json    0.958730\n",
       "Name: eval_score_gemini-2.0-flash, dtype: float64"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "average_scores = result.groupby(\"settings\")[\"eval_score_gemini-2.0-flash\"].mean()\n",
    "average_scores.sort_values()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
